From d281d6804b98d9c67c73f004931e6f11ccce3f13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=2E=20Neusch=C3=A4fer?= <j.ne@posteo.net>
Date: Wed, 31 Dec 2025 17:14:06 +0000
Subject: [PATCH] test: Don't download any files

---
 test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 7727016..8281e24 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -54,7 +54,7 @@ if(NOT PYTHONINTERP_FOUND)
   return()
 endif()
 
-find_program(WGET_EXE wget)
+set(WGET_EXE OFF)
 if(WGET_EXE)
   message(STATUS "Found WGet: ${WGET_EXE}")
   add_custom_target(wget_cppcheck
-- 
2.51.0

