From 8b0d26300acb9b304259214fb4cb7c3a49fedb59 Mon Sep 17 00:00:00 2001
From: Ralf Rachinger <git@ralfrachinger.de>
Date: Thu, 28 Apr 2022 01:04:43 +0200
Subject: [PATCH] Change Test FindLibrary to libc

---
 tests/base/test_os.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/base/test_os.lua b/tests/base/test_os.lua
index c4d9a4a0..5f1e5d12 100644
--- a/tests/base/test_os.lua
+++ b/tests/base/test_os.lua
@@ -31,7 +31,7 @@
 		elseif os.istarget("haiku") then
 			test.istrue(os.findlib("root"))
 		else
-			test.istrue(os.findlib("m"))
+			test.istrue(os.findlib("libc"))
 		end
 	end
 
-- 
2.36.0

