From 7bb635ffcb0fcff4e61f2ae7fc2b1aa6a699b52f Mon Sep 17 00:00:00 2001
From: Ralf Rachinger <ralfrachinger@gmail.com>
Date: Tue, 10 Aug 2021 02:09:47 +0200
Subject: [PATCH 2/2] Change Test FindLibrary to libc

---
 tests/base/test_os.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/base/test_os.lua b/tests/base/test_os.lua
index 62315dea..23c5b966 100644
--- a/tests/base/test_os.lua
+++ b/tests/base/test_os.lua
@@ -28,7 +28,7 @@
 		elseif os.istarget("haiku") then
 			test.istrue(os.findlib("root"))
 		else
-			test.istrue(os.findlib("m"))
+			test.istrue(os.findlib("libc"))
 		end
 	end
 
-- 
2.32.0

