diff --git a/lib/help_chrome.lua b/lib/help_chrome.lua
index ac43f07..9d3406f 100644
--- a/lib/help_chrome.lua
+++ b/lib/help_chrome.lua
@@ -310,7 +310,7 @@ local help_doc_page = function (v, path, request)
     if not doc_html then
         local file = doc_style
         error_page.show_error_page(v, {
-            heading = "Documentation not found",
+            heading = "Documentation not found (install luakit-doc)",
             content = "Opening <code>" .. file .. "</code> failed",
             buttons = { path ~= "index.html" and {
                 label = "Return to API Index",
