--- ./config.mk.in.orig	2013-08-16 09:27:58.138960649 +0000
+++ ./config.mk.in	2013-08-16 09:30:47.452830344 +0000
@@ -166,7 +166,7 @@
 # HOST_OS is usually has a version number suffix, e.g. "aix5.3.0.0", so
 # we compare based on prefix.
 
-ifeq ($(patsubst linux-gnu%,linux-gnu,$(HOST_OS)),linux-gnu)
+ifeq ($(patsubst linux%,linux,$(HOST_OS)),linux)
   # Assume linker is GNU Compiler (gcc)
   SHARED_LIB_TYPE = unix
   MUST_BUILD_SHLIB = Y
