Not applicable upstream

not everything is inside a git repository

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e72f08..a7c5c23 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,18 +11,8 @@ set(PROJECT_BUGS https://github.com/midori-browser/core/issues)
 set(PROJECT_WEBSITE https://www.midori-browser.org)
 set(CORE_VERSION 9.0)
 
-execute_process(COMMAND "git" "describe" "--tags"
-                OUTPUT_VARIABLE REVISION
-                ERROR_QUIET
-                OUTPUT_STRIP_TRAILING_WHITESPACE)
-if (REVISION)
-    set(CORE_VERSION "${REVISION}")
-    # All warnings are errors in development builds
-    set(VALAFLAGS ${VALAFLAGS} --fatal-warnings)
-else ()
-    # No runtime type checks
-    set(VALAFLAGS ${VALAFLAGS} --disable-assert)
-endif ()
+# No runtime type checks
+set(VALAFLAGS ${VALAFLAGS} --disable-assert)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake)
 # Disallow building during install to avoid permission problems
