=== modified file 'extensions/transfers.vala'
--- a/extensions/transfers.vala	2016-04-02 08:15:11 +0000
+++ b/extensions/transfers.vala	2017-02-15 18:50:47 +0000
@@ -455,7 +455,7 @@
                 if (notifications.length () == 1)
                     msg = _("The file '<b>%s</b>' has been downloaded.").printf (filename);
                 else
-                    msg = _("'<b>%s</b>' and %d other files have been downloaded.").printf (filename, notifications.length ());
+                    msg = _("'<b>%s</b>' and %d other files have been downloaded.").printf (filename, (int) notifications.length ());
                 get_app ().send_notification (_("Transfer completed"), msg);
                 notifications = new GLib.List<string> ();
             }

=== modified file 'midori/midori-notebook.vala'
--- a/midori/midori-notebook.vala	2016-04-02 08:15:11 +0000
+++ b/midori/midori-notebook.vala	2017-02-15 18:50:47 +0000
@@ -10,7 +10,7 @@
 */
 
 namespace Midori {
-    protected class Tally : Gtk.EventBox {
+    internal class Tally : Gtk.EventBox {
         public Midori.Tab tab { get; set; }
         Gtk.Spinner spinner;
         public Gtk.Label label;
@@ -22,7 +22,7 @@
         public bool close_button_left { get; set; default = false; }
         public bool close_button_visible { get; set; default = false; }
 
-        protected Tally (Midori.Tab tab) {
+        internal Tally (Midori.Tab tab) {
             this.tab = tab;
             box = new Gtk.HBox (false, 1);
             add (box);
@@ -175,7 +175,7 @@
         int last_tab_size = 0;
 
 #if !HAVE_GTK3
-        static const string style_fixup = """
+        const string style_fixup = """
             style "midori-close-button-style"
             {
             GtkWidget::focus-padding = 0

=== modified file 'extensions/transfers.vala'
--- a/extensions/transfers.vala	2015-08-20 06:18:50 +0000
+++ b/extensions/transfers.vala	2016-04-02 08:15:11 +0000
@@ -342,7 +342,7 @@
             icon = new Gtk.Image ();
             button = new Gtk.Button ();
             button.relief = Gtk.ReliefStyle.NONE;
-            button.focus_on_click = false;
+            button.set_focus_on_click (false);
             button.clicked.connect (button_clicked);
             button.add (icon);
             box.pack_start (button, false, false, 0);

=== modified file 'midori/midori-notebook.vala'
--- a/midori/midori-notebook.vala	2016-02-20 23:47:11 +0000
+++ b/midori/midori-notebook.vala	2016-04-02 08:15:11 +0000
@@ -41,7 +41,7 @@
             box.pack_start (label, true, true, 0);
             close = new Gtk.Button ();
             close.relief = Gtk.ReliefStyle.NONE;
-            close.focus_on_click = false;
+            close.set_focus_on_click (false);
 #if !HAVE_GTK3
             close.name = "midori-close-button";
             close.style_set.connect (close_style_set);

diff --git a/extensions/adblock/widgets.vala b/extensions/adblock/widgets.vala
index 757d35b..45880df 100644
--- a/extensions/adblock/widgets.vala
+++ b/extensions/adblock/widgets.vala
@@ -227,7 +227,7 @@ namespace Adblock {
                             liststore.get (iter, 0, out sub);
                             if (sub.mutable) {
                                 config.remove (sub);
-                                liststore.remove (iter);
+                                liststore.remove (ref iter);
                                 return true;
                             }
                         }
diff --git a/extensions/apps.vala b/extensions/apps.vala
index 16e2286..f81d351 100644
--- a/extensions/apps.vala
+++ b/extensions/apps.vala
@@ -265,7 +265,7 @@ namespace Apps {
                             store.get (iter, 0, out launcher);
                             try {
                                 launcher.file.trash (null);
-                                store.remove (iter);
+                                store.remove (ref iter);
 
                                 string filename = Midori.Download.clean_filename (launcher.name);
 #if HAVE_WIN32
diff --git a/extensions/history-list.vala b/extensions/history-list.vala
index db71927..23d8263 100644
--- a/extensions/history-list.vala
+++ b/extensions/history-list.vala
@@ -223,7 +223,7 @@ namespace HistoryList {
                     FixMe: the retrun value of `Gtk.ListStore.remove` should be checked
                     Note:  in some cases the return value of `Gtk.ListStore.remove` is wrong
                 */
-                model.remove (iter);
+                model.remove (ref iter);
                 this.browser.close_tab (view);
                 if (length > 2)
                     this.resize_treeview ();
diff --git a/extensions/notes.vala b/extensions/notes.vala
index f0ccfe6..c8d4c4a 100644
--- a/extensions/notes.vala
+++ b/extensions/notes.vala
@@ -116,7 +116,7 @@ namespace ClipNotes {
                     if (current_note == note) {
                         current_note = null;
                     }
-                    notes_list_store.remove (iter);
+                    notes_list_store.remove (ref iter);
                     break;
                 }
             } while (notes_list_store.iter_next (ref iter));
diff --git a/extensions/transfers.vala b/extensions/transfers.vala
index ae95fa2..ad063b2 100644
--- a/extensions/transfers.vala
+++ b/extensions/transfers.vala
@@ -275,7 +275,7 @@ namespace Transfers {
                     Transfer found;
                     store.get (iter, 0, out found);
                     if (transfer == found) {
-                        store.remove (iter);
+                        store.remove (ref iter);
                         break;
                     }
                 } while (store.iter_next (ref iter));
