diff --git a/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h b/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
index 1bd2970b..a1c23415 100644
--- a/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
+++ b/ThirdParty/doubleconversion/vtkdoubleconversion/double-conversion/utils.h
@@ -99,6 +99,7 @@ int main(int argc, char** argv) {
     defined(__ARMEL__) || defined(__avr32__) || defined(_M_ARM) || defined(_M_ARM64) || \
     defined(__hppa__) || defined(__ia64__) || \
     defined(__mips__) || \
+    defined(__loongarch__) || \
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
     defined(__sparc__) || defined(__sparc) || defined(__s390__) || \
