--- ./core/atomic_types.h.orig
+++ ./core/atomic_types.h
@@ -13,7 +13,7 @@
 		  ) || \
 		  (defined(__linux__) && \
 		    ( \
-		      (defined(__i386__) && (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4))) || \
+		      (defined(__i386__) && (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4) && (defined(__GCC_HAVE_SYNC_COMPARE_AND_SWAP_8)))) || \
 		      defined(__ia64__) || \
 		      defined(__x86_64__) || \
 		      (defined(__powerpc__) && !defined(__powerpc64__)) || \
