diff --git a/core/AmBasicSipDialog.cpp b/core/AmBasicSipDialog.cpp
index f0aeef4..c177766 100644
--- a/core/AmBasicSipDialog.cpp
+++ b/core/AmBasicSipDialog.cpp
@@ -577,7 +577,7 @@ int AmBasicSipDialog::reply(const AmSipRequest& req,
 	  "remote_tag=%s\n",
 	  req.cseq,code,callid.c_str(),
 	  local_tag.c_str(),remote_tag.c_str());
-    log_stacktrace(L_ERR);
+    /* log_stacktrace(L_ERR); */
     return -1;
   }
   DBG("reply: transaction found!\n");
diff --git a/core/log.cpp b/core/log.cpp
index d1de02f..9e59464 100644
--- a/core/log.cpp
+++ b/core/log.cpp
@@ -209,6 +209,7 @@ void register_log_hook(AmLoggingFacility* fac)
 /**
  * Print stack-trace through logging function
  */
+/* 
 void log_stacktrace(int ll)
 {
    void* callstack[128];
@@ -219,3 +220,4 @@ void log_stacktrace(int ll)
    }
    free(strs);
 }
+*/
diff --git a/core/log.h b/core/log.h
index 7eb2480..892c926 100644
--- a/core/log.h
+++ b/core/log.h
@@ -33,8 +33,9 @@
 #include <stdio.h>
 #include <unistd.h>	/* getpid() */
 #include <pthread.h>	/* pthread_self() */
+#if defined (__GLIBC__)
 #include <execinfo.h>   /* backtrace_symbols() */
-
+#endif
 
 #ifdef __cplusplus
 extern "C" {
@@ -174,7 +175,9 @@ void run_log_hooks(int, pid_t, pthread_t, const char*, const char*, int, char*);
 int set_syslog_facility(const char*);
 #endif
 
+/*
 void log_stacktrace(int ll);
+*/
 
 #ifdef __cplusplus
 }
