--- ./Makefile.orig
+++ ./Makefile
@@ -33,7 +33,7 @@
 reader/dff: reader/dff.c; $(CC) -c $(CFLAGS) $^ -o $(VPATH)/$@.o
 reader/dsf: reader/dsf.c; $(CC) -c $(CFLAGS) $^ -o $(VPATH)/$@.o
 libodiosacd: libodiosacd.c; $(CC) -c $(CFLAGS) $^ -o $(VPATH)/$@.o
-odio-libsacd: $(foreach object, $(OBJECTS), $(VPATH)/$(object).o); $(CC) -o data/usr/lib/libodiosacd.so $^ $(LDFLAGS)
+odio-libsacd: $(foreach object, $(OBJECTS), $(VPATH)/$(object).o); $(CC) -o data/usr/lib/libodiosacd.so.1 $^ $(LDFLAGS)
 
 clean:
 
@@ -42,8 +42,8 @@
 
 install:
 
-	$(shell install -Dt $(DESTDIR)/usr/$(LIBDIR) ./data/usr/lib/libodiosacd.so)
-	$(shell ln -sf ./libodiosacd.so $(DESTDIR)/usr/$(LIBDIR)/libodiosacd.so.1)
+	$(shell install -Dt $(DESTDIR)/usr/$(LIBDIR) ./data/usr/lib/libodiosacd.so.1)
+	$(shell ln -sf ./libodiosacd.so.1 $(DESTDIR)/usr/$(LIBDIR)/libodiosacd.so)
 	$(shell install -Dt $(DESTDIR)/usr/include/libodiosacd ./src/libodiosacd.h)
 	$(shell install -Dt $(DESTDIR)/usr/include/libodiosacd/reader ./src/reader/disc.h)
 	$(shell install -Dt $(DESTDIR)/usr/include/libodiosacd/reader ./src/reader/media.h)
