diff --git a/setup.py b/setup.py
index f852d0f..b7f5a57 100644
--- a/setup.py
+++ b/setup.py
@@ -44,15 +44,12 @@ setup(
         ],
     python_requires='~=3.8',
     install_requires=['PyYAML', 'pymisp', 'progressbar2', 'ruamel.yaml', 'termcolor'],
-    extras_require={
-        'test': ['coverage', 'yamllint', 'attackcti'],
-    },
-    data_files=[
-        ('etc/sigma', [ str(p) for p in Path('config/').glob('*.yml') ]),
-        ('etc/sigma/generic', [ str(p) for p in Path('config/generic/').glob('*.yml') ])],
-    entry_points={
-        'console_scripts': [
-            'sigmac = sigma.sigmac:main',
+    extras_require={
+        'test': ['coverage', 'yamllint', 'attackcti'],
+    },
+    entry_points={
+        'console_scripts': [
+            'sigmac = sigma.sigmac:main',
             'merge_sigma = sigma.merge_sigma:main',
             'sigma2misp = sigma.sigma2misp:main',
             'sigma2attack = sigma.sigma2attack:main',
