Upstream vendors packaging, use system packaging instead

diff --git a/setup.py b/setup.py
index 8157d44..a5990e1 100644
--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@
 from distutils.core import setup
 
 packages = \
-['wheel', 'wheel.cli', 'wheel.vendored', 'wheel.vendored.packaging']
+['wheel', 'wheel.cli']
 
 package_data = \
 {'': ['*']}
diff --git a/src/wheel/_bdist_wheel.py b/src/wheel/_bdist_wheel.py
index 6b811ee..45540ed 100644
--- a/src/wheel/_bdist_wheel.py
+++ b/src/wheel/_bdist_wheel.py
@@ -20,6 +20,7 @@
 from shutil import rmtree
 from typing import TYPE_CHECKING, Callable, Iterable, Literal, Sequence, cast
 from zipfile import ZIP_DEFLATED, ZIP_STORED
+from packaging import tags
 
 import setuptools
 from setuptools import Command
@@ -27,8 +28,7 @@
 from . import __version__ as wheel_version
 from .metadata import pkginfo_to_metadata
 from .util import log
-from .vendored.packaging import tags
-from .vendored.packaging import version as _packaging_version
+from packaging import version as _packaging_version
 from .wheelfile import WheelFile
 
 if TYPE_CHECKING:
diff --git a/src/wheel/metadata.py b/src/wheel/metadata.py
index 6aa4362..27cc7fa 100644
--- a/src/wheel/metadata.py
+++ b/src/wheel/metadata.py
@@ -13,7 +13,7 @@ from email.message import Message
 from email.parser import Parser
 from typing import Generator, Iterable, Iterator, Literal
 
-from .vendored.packaging.requirements import Requirement
+from packaging.requirements import Requirement
 
 
 def _nonblank(str: str) -> bool | Literal[""]:
diff --git a/tests/test_bdist_wheel.py b/tests/test_bdist_wheel.py
index b440b64..f94d752 100644
--- a/tests/test_bdist_wheel.py
+++ b/tests/test_bdist_wheel.py
@@ -23,7 +23,7 @@ from wheel.bdist_wheel import (
     remove_readonly,
     remove_readonly_exc,
 )
-from wheel.vendored.packaging import tags
+from packaging import tags
 from wheel.wheelfile import WheelFile
 
 DEFAULT_FILES = {
--- a/src/wheel/cli/convert.py
+++ b/src/wheel/cli/convert.py
@@ -15,7 +15,7 @@
 
 from .. import __version__
 from ..metadata import generate_requirements
-from ..vendored.packaging.tags import parse_tag
+from packaging.tags import parse_tag
 from ..wheelfile import WheelFile
 
 egg_filename_re = re.compile(
