Upstream vendors packaging, use system packaging instead

diff --git a/setup.py b/setup.py
index 8157d44..a5990e1 100644
--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,7 @@
 from distutils.core import setup
 
 packages = \
-['wheel', 'wheel.cli', 'wheel.vendored', 'wheel.vendored.packaging']
+['wheel', 'wheel.cli']
 
 package_data = \
 {'': ['*']}
diff --git a/src/wheel/bdist_wheel.py b/src/wheel/bdist_wheel.py
index 6b811ee..45540ed 100644
--- a/src/wheel/bdist_wheel.py
+++ b/src/wheel/bdist_wheel.py
@@ -19,6 +19,7 @@ from email.policy import EmailPolicy
 from glob import iglob
 from shutil import rmtree
 from zipfile import ZIP_DEFLATED, ZIP_STORED
+from packaging import tags
 
 import setuptools
 from setuptools import Command
@@ -27,8 +28,7 @@ from . import __version__ as wheel_version
 from .macosx_libfile import calculate_macosx_platform_tag
 from .metadata import pkginfo_to_metadata
 from .util import log
-from .vendored.packaging import tags
-from .vendored.packaging import version as _packaging_version
+from packaging import version as _packaging_version
 from .wheelfile import WheelFile
 
 
diff --git a/src/wheel/metadata.py b/src/wheel/metadata.py
index 6aa4362..27cc7fa 100644
--- a/src/wheel/metadata.py
+++ b/src/wheel/metadata.py
@@ -13,7 +13,7 @@ from email.message import Message
 from email.parser import Parser
 from typing import Iterator
 
-from .vendored.packaging.requirements import Requirement
+from packaging.requirements import Requirement
 
 
 def _nonblank(str):
diff --git a/tests/test_bdist_wheel.py b/tests/test_bdist_wheel.py
index b440b64..f94d752 100644
--- a/tests/test_bdist_wheel.py
+++ b/tests/test_bdist_wheel.py
@@ -21,7 +21,7 @@ from wheel.bdist_wheel import (
     remove_readonly,
     remove_readonly_exc,
 )
-from wheel.vendored.packaging import tags
+from packaging import tags
 from wheel.wheelfile import WheelFile
 
 DEFAULT_FILES = {
