Upstream vendors packaging, use system packaging instead

diff --git a/src/wheel/bdist_wheel.py b/src/wheel/bdist_wheel.py
index 80e43d0..6bb272c 100644
--- a/src/wheel/bdist_wheel.py
+++ b/src/wheel/bdist_wheel.py
@@ -19,13 +19,13 @@ from glob import iglob
 from shutil import rmtree
 from sysconfig import get_config_var
 from zipfile import ZIP_DEFLATED, ZIP_STORED
+from packaging import tags
 
 import pkg_resources
 
 from .pkginfo import write_pkg_info
 from .macosx_libfile import calculate_macosx_platform_tag
 from .metadata import pkginfo_to_metadata
-from .vendored.packaging import tags
 from .wheelfile import WheelFile
 from . import __version__ as wheel_version
 
