Upstream vendors packaging, use system packaging instead

diff --git a/src/wheel/bdist_wheel.py b/src/wheel/bdist_wheel.py
index 4754fd1..31dc391 100644
--- a/src/wheel/bdist_wheel.py
+++ b/src/wheel/bdist_wheel.py
@@ -19,6 +19,7 @@ from glob import iglob
 from io import BytesIO
 from shutil import rmtree
 from zipfile import ZIP_DEFLATED, ZIP_STORED
+from packaging import tags
 
 import pkg_resources
 from setuptools import Command
@@ -27,7 +28,6 @@ from . import __version__ as wheel_version
 from .macosx_libfile import calculate_macosx_platform_tag
 from .metadata import pkginfo_to_metadata
 from .util import log
-from .vendored.packaging import tags
 from .wheelfile import WheelFile
 
 safe_name = pkg_resources.safe_name
diff --git a/tests/test_bdist_wheel.py b/tests/test_bdist_wheel.py
index 531d9e6..30940c5 100644
--- a/tests/test_bdist_wheel.py
+++ b/tests/test_bdist_wheel.py
@@ -11,7 +11,7 @@ from zipfile import ZipFile
 import pytest
 
 from wheel.bdist_wheel import bdist_wheel, get_abi_tag
-from wheel.vendored.packaging import tags
+from packaging import tags
 from wheel.wheelfile import WheelFile
 
 DEFAULT_FILES = {
