commit 3136f9ead217b68d00d66ef26ae9ff8f354f5d91
Author: q66 <q66@chimera-linux.org>
Date:   Sat Oct 14 16:36:01 2023 +0200

    use system spirv-headers

diff --git a/src/meson.build b/src/meson.build
index c63cbd7..0afe796 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -94,7 +94,7 @@ reshade_src = [
 reshade_include = include_directories([
   'reshade/source',
   'reshade/include',
-  '../thirdparty/SPIRV-Headers/include/spirv/unified1'
+  '/usr/include/spirv/unified1'
 ])
 
 src = [
