diff -upr multitail-6.4.2.orig/mt.h multitail-6.4.2/mt.h
--- multitail-6.4.2.orig/mt.h	2015-10-03 01:53:00.408700746 +0200
+++ multitail-6.4.2/mt.h	2015-10-03 01:53:03.935011348 +0200
@@ -60,8 +60,8 @@ typedef enum { SCHEME_TYPE_EDIT = 0, SCH
 #endif
 
 #ifdef UTF8_SUPPORT
-	#include <ncursesw/panel.h>
-	#include <ncursesw/ncurses.h>
+	#include <panel.h>
+	#include <ncurses.h>
 #else
 	#if defined(sun) || defined(__sun) || defined(scoos) || defined(_HPUX_SOURCE) || defined(AIX) || defined(__CYGWIN__)
 		#include <ncurses/panel.h>
