diff --git a/scripts/extratest.sh b/scripts/extratest.sh
index a230662..191c3e0 100755
--- a/scripts/extratest.sh
+++ b/scripts/extratest.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 set -o errexit
 set -o pipefail
 set -o nounset
@@ -80,11 +80,11 @@ elif [ "$1" = test ]; then
 
     set +x
     if ! which docker; then
-        echo -e "WARNING: =====\n\ndocker not present, some tests will fail\n\n=====\n\n\n\n\n"
+        printf "WARNING: =====\n\ndocker not present, some tests will fail\n\n=====\n\n\n\n\n"
         sleep 5
     fi
     if ! which icecc-create-env; then
-        echo -e "WARNING: =====\n\nicecc-create-env not present, some tests will fail\n\n=====\n\n\n\n\n"
+        printf "WARNING: =====\n\nicecc-create-env not present, some tests will fail\n\n=====\n\n\n\n\n"
         sleep 5
     fi
     set -x
diff --git a/tests/sccache_rustc.rs b/tests/sccache_rustc.rs
index 70ad65e..f8ffd70 100644
--- a/tests/sccache_rustc.rs
+++ b/tests/sccache_rustc.rs
@@ -87,12 +87,12 @@ fn create_mock_rustc(dir: PathBuf) {
     let rustc = bin.join("rustc");
     write!(
         File::create(&rustc).unwrap(),
-        r#"#!/usr/bin/env bash
+        r#"#!/usr/bin/env sh
 
 set -e
 build=0
 
-while (( "$#" )); do
+while [ "$#" -gt 0 ]; do
     case "$1" in
         -vV)
             echo rustc 1.0.0
@@ -124,7 +124,7 @@ while (( "$#" )); do
     shift
 done
 
-if (( build )); then
+if [ "$build" -eq 1 ]; then
     echo $(($(cat counter) + 1)) > counter
     cp counter RUST_FILE
 fi
