--- a/pam/pam_fprintd.c
+++ b/pam/pam_fprintd.c
@@ -31,8 +31,7 @@
 #include <errno.h>
 
 #include <libintl.h>
-#include <systemd/sd-bus.h>
-#include <systemd/sd-login.h>
+#include <basu/sd-bus.h>
 #include <signal.h>
 #include <sys/signalfd.h>
 #include <poll.h>
@@ -775,8 +774,9 @@
       strcmp (rhost, "localhost") != 0)
     return true;
 
-  if (sd_session_is_remote (NULL) > 0)
-    return true;
+// XXX-Patched: Remove dependency on elogind.
+//if (sd_session_is_remote (NULL) > 0)
+//  return true;
 
   return false;
 }
