--- omxplayer-b34143cb3e678453ccae968d3a65c0cb5a34333e.orig/Makefile
+++ omxplayer-b34143cb3e678453ccae968d3a65c0cb5a34333e/Makefile
@@ -46,7 +46,7 @@
 	$(CXX) $(CFLAGS) $(INCLUDES) -c $< -o $@ -Wno-deprecated-declarations
 
 version:
-	bash gen_version.sh > version.h 
+	sh gen_version.sh > version.h 
 
 omxplayer.bin: version $(OBJS)
 	$(CXX) $(LDFLAGS) -o omxplayer.bin $(OBJS) -lvchiq_arm -lvcos -lrt -lpthread -lavutil -lavcodec -lavformat -lswscale -lswresample -lpcre
@@ -65,11 +65,9 @@
 	make -f Makefile.ffmpeg install
 
 dist: omxplayer.bin
-	mkdir -p $(DIST)/usr/lib/omxplayer
 	mkdir -p $(DIST)/usr/bin
 	mkdir -p $(DIST)/usr/share/doc
 	cp omxplayer omxplayer.bin $(DIST)/usr/bin
 	cp COPYING $(DIST)/usr/share/doc/
 	cp README.md $(DIST)/usr/share/doc/README
-	cp -a ffmpeg_compiled/usr/local/lib/*.so* $(DIST)/usr/lib/omxplayer/
 	tar -czf omxplayer-dist.tar.gz $(DIST)
