--- ./configure.ac.orig
+++ ./configure.ac
@@ -361,13 +361,13 @@
   else
     include=stdlib.h
   fi
-  AC_COMPILE_IFELSE(AC_LANG_PROGRAM([#include <$include>],
+  AC_LINK_IFELSE(AC_LANG_PROGRAM([#include <$include>],
                                     [char* e; return $fn("100", &e, 10)]),
                     [AC_MSG_RESULT(yes)
                      AC_DEFINE_UNQUOTED(HAVE_`echo $fn | tr abcdefghijklmnopqrstuvwxyz ABCDEFGHIJKLMNOPQRSTUVWXYZ`, 1,
                                         [Define to 1 if you have `$fn'.])
                      have_strto_fn=1
-                     break],
+                     ],
                     [AC_MSG_RESULT(no)])
 done
 
