diff --git a/util/dependencies.lua b/util/dependencies.lua
index 4d50cf6..9ea211d 100644
--- a/util/dependencies.lua
+++ b/util/dependencies.lua
@@ -99,6 +99,9 @@ function check_dependencies()
 				["luarocks"] = "luarocks install luasec";
 				["Source"] = "http://www.inf.puc-rio.br/~brunoos/luasec/";
 			}, "SSL/TLS support will not be available");
+	elseif not _G.ssl then
+		_G.ssl = ssl;
+		_G.ssl.context = require "ssl.context";
 	end
 	
 	local encodings, err = softreq "util.encodings"
