diff --git a/simpleble/CMakeLists.txt b/simpleble/CMakeLists.txt
index 29d1929..2b90a38 100644
--- a/simpleble/CMakeLists.txt
+++ b/simpleble/CMakeLists.txt
@@ -20,10 +20,6 @@ option(SIMPLEBLE_PLAIN "Use plain version of SimpleBLE" OFF)
 if(NOT TARGET fmt::fmt-header-only)
     option(LIBFMT_VENDORIZE "Enable vendorized libfmt" ON)
     find_package(fmt REQUIRED)
-
-    if(TARGET fmt)
-        set_target_properties(fmt PROPERTIES EXCLUDE_FROM_ALL TRUE)
-    endif()
 endif()
 
 if(SIMPLEBLE_TEST)
diff --git a/simplebluez/CMakeLists.txt b/simplebluez/CMakeLists.txt
index 53cd80d..bc77361 100644
--- a/simplebluez/CMakeLists.txt
+++ b/simplebluez/CMakeLists.txt
@@ -21,7 +21,6 @@ include(${CMAKE_CURRENT_SOURCE_DIR}/../cmake/epilogue.cmake)
 option(LIBFMT_VENDORIZE "Enable vendorized libfmt" ON)
 
 find_package(fmt REQUIRED)
-set_target_properties(fmt PROPERTIES EXCLUDE_FROM_ALL TRUE)
 
 find_package(DBus1 REQUIRED)
