--- openswan-2.6.35-orig/programs/setup/Makefile
+++ openswan-2.6.35/programs/setup/Makefile
@@ -36,8 +36,7 @@
 	# check the directories we need exist (ie., make install DESTDIR=/tmp/IPsec)
 	@mkdir -p $(RCDIR) $(BINDIR)
 	# install and link everything
-	@$(INSTALL) $(INSTBINFLAGS) setup $(RCDIR)/ipsec
-	@ln -s $(FINALRCDIR)/ipsec $(BINDIR)/setup
+	@$(INSTALL) $(INSTBINFLAGS) setup $(BINDIR)/setup
 
 install_file_list::
 	@echo $(RCDIR)/ipsec
