diff --git a/test/unittests/test_numeric.py b/test/unittests/test_numeric.py
index a32fe5e..cd1e781 100644
--- a/test/unittests/test_numeric.py
+++ b/test/unittests/test_numeric.py
@@ -11,9 +11,9 @@ class TestEnglish(unittest.TestCase):
 
-        self.assertEquals(parser.convert_words_to_numbers("this is test number two"),
+        self.assertEqual(parser.convert_words_to_numbers("this is test number two"),
                           "this is test number 2")
-        self.assertEquals(parser.convert_words_to_numbers("this is test number two and a half"),
+        self.assertEqual(parser.convert_words_to_numbers("this is test number two and a half"),
                           "this is test number 2.5")
-        self.assertEquals(parser.convert_words_to_numbers("this is the first test"),
+        self.assertEqual(parser.convert_words_to_numbers("this is the first test"),
                           "this is the first test")
-        self.assertEquals(parser.convert_words_to_numbers("this is the first test", ordinals=True),
+        self.assertEqual(parser.convert_words_to_numbers("this is the first test", ordinals=True),
                           "this is the 1 test")
