--- ./packages/flutter_tools/lib/src/linux/build_linux.dart.orig
+++ ./packages/flutter_tools/lib/src/linux/build_linux.dart
@@ -135,9 +135,9 @@
       if (needCrossBuild)
         '-DFLUTTER_TARGET_PLATFORM_SYSROOT=$targetSysroot',
       if (needCrossBuildOptionsForArm64)
-        '-DCMAKE_C_COMPILER_TARGET=aarch64-linux-gnu',
+        '-DCMAKE_C_COMPILER_TARGET=aarch64-alpine-linux-musl',
       if (needCrossBuildOptionsForArm64)
-        '-DCMAKE_CXX_COMPILER_TARGET=aarch64-linux-gnu',
+        '-DCMAKE_CXX_COMPILER_TARGET=aarch64-alpine-linux-musl',
       sourceDir.path,
     ],
     workingDirectory: buildDir.path,
