--- a/packages/flutter_tools/lib/src/build_system/targets/linux.dart
+++ b/packages/flutter_tools/lib/src/build_system/targets/linux.dart
@@ -18,7 +18,7 @@
 import 'native_assets.dart';
 
 /// The only files/subdirectories we care out.
-const List<String> _kLinuxArtifacts = <String>['libflutter_linux_gtk.so'];
+const List<String> _kLinuxArtifacts = <String>[];
 
 const String _kLinuxDepfile = 'linux_engine_sources.d';
 
@@ -76,10 +76,6 @@
       outputDirectory: outputDirectory,
       artifacts: _kLinuxArtifacts,
       clientSourcePaths: <String>[headersPath],
-      icuDataPath: environment.artifacts.getArtifactPath(
-        Artifact.icuData,
-        platform: targetPlatform,
-      ),
     );
     environment.depFileService.writeToFile(depfile, environment.buildDir.childFile(_kLinuxDepfile));
   }
