--- a/packages/flutter_tools/lib/src/build_system/targets/linux.dart
+++ b/packages/flutter_tools/lib/src/build_system/targets/linux.dart
@@ -19,7 +19,6 @@
 
 /// The only files/subdirectories we care out.
 const List<String> _kLinuxArtifacts = <String>[
-  'libflutter_linux_gtk.so',
 ];
 
 const String _kLinuxDepfile = 'linux_engine_sources.d';
@@ -79,10 +78,6 @@
       outputDirectory: outputDirectory,
       artifacts: _kLinuxArtifacts,
       clientSourcePaths: <String>[headersPath],
-      icuDataPath: environment.artifacts.getArtifactPath(
-        Artifact.icuData,
-        platform: targetPlatform,
-      )
     );
     final DepfileService depfileService = DepfileService(
       fileSystem: environment.fileSystem,
