--- ./althttpd.c.orig
+++ ./althttpd.c
@@ -3425,7 +3425,7 @@
         NotFound(400); /* LOG: URI is a directory w/o index.html */
       }
       zRealScript = StrDup(&zLine[j0]);
-      if( zScript[i]==0 ){
+      if( zScript[i]==0 && strcmp(zScript, "/")!=0 ){
         /* If the requested URL does not end with "/" but we had to
         ** append "index.html", then a redirect is necessary.  Otherwise
         ** none of the relative URLs in the delivered document will be
