--- a/nsd.conf.sample.in
+++ b/nsd.conf.sample.in
@@ -79,9 +79,6 @@
 	# facility LOG_DAEMON).  stderr disappears when daemon goes to bg.
 	# logfile: "@logfile@"
 
-	# File to store pid for nsd in.
-	# pidfile: "@pidfile@"
-
 	# The file where secondary zone refresh and expire timeouts are kept.
 	# If you delete this file, all secondary zones are forced to be 
 	# 'refreshing' (as if nsd got a notify).  Set to "" to disable.
@@ -222,11 +219,14 @@
 remote-control:
 	# Enable remote control with nsd-control(8) here.
 	# set up the keys and certificates with nsd-control-setup.
-	# control-enable: no
+	# (This needs to be enabled to use init script's commands reload and
+	# reopen)
+	control-enable: yes
 
 	# what interfaces are listened to for control, default is on localhost.
 	# with an absolute path, a unix local named pipe is used for control
 	# (and key and cert files are not needed, use directory permissions).
+	control-interface: /run/nsd.control.sock
 	# control-interface: 127.0.0.1
 	# control-interface: ::1
 
