From d7824289f40aeaa221f401007ab195ba599d94e3 Mon Sep 17 00:00:00 2001
From: Michael Klishin <michael@clojurewerkz.org>
Date: Tue, 20 Oct 2020 13:28:58 +0300
Subject: [PATCH] List Elixir 1.11 as supported

---
 deps/rabbitmq_cli/mix.exs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/rabbitmq_cli/mix.exs b/deps/rabbitmq_cli/mix.exs
index 7f531d0757a..3f542e49b09 100644
--- a/deps/rabbitmq_cli/mix.exs
+++ b/deps/rabbitmq_cli/mix.exs
@@ -11,7 +11,7 @@ defmodule RabbitMQCtl.MixfileBase do
     [
       app: :rabbitmqctl,
       version: "3.8.0-dev",
-      elixir: ">= 1.8.0 and < 1.11.0",
+      elixir: ">= 1.10.4 and < 1.12.0",
       build_embedded: Mix.env == :prod,
       start_permanent: Mix.env == :prod,
       escript: [main_module: RabbitMQCtl,
