From e322f2e00e48b639b5d87ed06738aa24d99396bf Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 31 Mar 2010 18:56:39 +0400
Subject: [PATCH 03/11] Set CFG_PREFIX to empty string by default

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef3934e..c36ce62 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,7 @@ ENDIF (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} STREQUAL "2.4")
 SET(CMAKE_DL_LIBS ${CMAKE_DL_LIBS} m pthread)
 
 IF (NOT SEMS_CFG_PREFIX)
-	SET(SEMS_CFG_PREFIX "/usr/local")
+	SET(SEMS_CFG_PREFIX "")
 ENDIF (NOT SEMS_CFG_PREFIX)
 MESSAGE(STATUS "SEMS cfg prefix was set to: ${SEMS_CFG_PREFIX}")
 
-- 
1.7.3.4

