diff --git a/Telegram/gyp/lib_lottie.gyp b/Telegram/gyp/lib_lottie.gyp
index ada149e..870f4e6 100644
--- a/Telegram/gyp/lib_lottie.gyp
+++ b/Telegram/gyp/lib_lottie.gyp
@@ -24,21 +24,18 @@
       'official_build_target%': '',
       'submodules_loc': '../ThirdParty',
       'rlottie_loc': '<(submodules_loc)/rlottie/inc',
-      'lz4_loc': '<(submodules_loc)/lz4/lib',
     },
     'dependencies': [
       'crl.gyp:crl',
       'lib_base.gyp:lib_base',
       'lib_rlottie.gyp:lib_rlottie',
       'lib_ffmpeg.gyp:lib_ffmpeg',
-      'lib_lz4.gyp:lib_lz4',
     ],
     'export_dependent_settings': [
       'crl.gyp:crl',
       'lib_base.gyp:lib_base',
       'lib_rlottie.gyp:lib_rlottie',
       'lib_ffmpeg.gyp:lib_ffmpeg',
-      'lib_lz4.gyp:lib_lz4',
     ],
     'defines': [
       'LOT_BUILD',
@@ -50,7 +47,6 @@
       '<(libs_loc)/zlib',
       '<(libs_loc)/ffmpeg',
       '<(rlottie_loc)',
-      '<(lz4_loc)',
       '<(submodules_loc)/GSL/include',
       '<(submodules_loc)/variant/include',
       '<(submodules_loc)/crl/src',

diff --git a/Telegram/gyp/telegram/linux.gypi b/Telegram/gyp/telegram/linux.gypi
index 9349c68..087c424 100644
--- a/Telegram/gyp/telegram/linux.gypi
+++ b/Telegram/gyp/telegram/linux.gypi
@@ -34,6 +34,7 @@
       '-lopus',
       '-lz',
       '-llzma',
+	  '-llz4',
 #      '<!(pkg-config 2> /dev/null --libs <@(pkgconfig_libs))',
     ],
     'cflags_cc': [

