diff --git a/Telegram/gyp/lib_rlottie.gyp b/Telegram/gyp/lib_rlottie.gyp
index efc208500..8f2a66f5a 100644
--- a/Telegram/gyp/lib_rlottie.gyp
+++ b/Telegram/gyp/lib_rlottie.gyp
@@ -113,7 +113,7 @@
     ],
     'conditions': [[ 'not build_win', {
       'defines': [
-        'RLOTTIE_WITH_STATIC_QT',
+#        'RLOTTIE_WITH_STATIC_QT',
       ]
     }], [ 'build_macold', {
       'xcode_settings': {

