diff --git a/cmake/init_target.cmake b/cmake/init_target.cmake
index c7a1244..2824a60 100644
--- a/cmake/init_target.cmake
+++ b/cmake/init_target.cmake
@@ -27,7 +27,7 @@ function(init_target target_name) # init_target(my_target folder_name)
     endif()
     target_link_libraries(${target_name} PUBLIC desktop-app::common_options)
     set_target_properties(${target_name} PROPERTIES
-        LINK_SEARCH_START_STATIC 1
+        LINK_SEARCH_START_STATIC 0
         XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_WEAK YES
         XCODE_ATTRIBUTE_GCC_INLINES_ARE_PRIVATE_EXTERN YES
         XCODE_ATTRIBUTE_GCC_SYMBOLS_PRIVATE_EXTERN YES
