diff --git a/cmake/external/ffmpeg/CMakeLists.txt b/cmake/external/ffmpeg/CMakeLists.txt
index 014fb06..84a3011 100644
--- a/cmake/external/ffmpeg/CMakeLists.txt
+++ b/cmake/external/ffmpeg/CMakeLists.txt
@@ -7,32 +7,8 @@
 add_library(external_ffmpeg INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_ffmpeg ALIAS external_ffmpeg)
 
-target_include_directories(external_ffmpeg SYSTEM
-INTERFACE
-    ${libs_loc}/ffmpeg
-)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(FFMPEG REQUIRED libavcodec libavformat libavutil libswresample libswscale)
 
-set(ffmpeg_lib_loc ${libs_loc}/ffmpeg)
-
-target_link_libraries(external_ffmpeg
-INTERFACE
-    ${ffmpeg_lib_loc}/libavformat/libavformat.a
-    ${ffmpeg_lib_loc}/libavcodec/libavcodec.a
-    ${ffmpeg_lib_loc}/libswresample/libswresample.a
-    ${ffmpeg_lib_loc}/libswscale/libswscale.a
-    ${ffmpeg_lib_loc}/libavutil/libavutil.a
-)
-if (LINUX)
-    target_link_static_libraries(external_ffmpeg
-    INTERFACE
-        va-x11
-        va-drm
-        va
-        vdpau
-        drm
-        Xi
-        Xext
-        Xfixes
-        Xrender
-    )
-endif()
+target_include_directories(external_ffmpeg SYSTEM INTERFACE ${FFMPEG_INCLUDE_DIR})
+target_link_libraries(external_ffmpeg INTERFACE ${FFMPEG_LIBRARIES})
