--- a/Telegram/ThirdParty/libtgvoip/config.h.in
+++ b/Telegram/ThirdParty/libtgvoip/config.h.in
@@ -197,7 +197,7 @@
 #undef WITHOUT_ALSA
 
 /* Define to disable PulseAudio support */
-#undef WITHOUT_PULSE
+#define WITHOUT_PULSE
 
 /* Define for Solaris 2.5.1 so the uint32_t typedef from <sys/synch.h>,
    <pthread.h>, or <semaphore.h> is not used. If the typedef were allowed, the
--- a/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
+++ b/Telegram/ThirdParty/libtgvoip/libtgvoip.gyp
@@ -10,6 +10,7 @@
           'WEBRTC_APM_DEBUG_DUMP=0',
           'TGVOIP_USE_DESKTOP_DSP',
           'WEBRTC_NS_FLOAT',
+          'WITHOUT_PULSE',
         ],
         'variables': {
           'tgvoip_src_loc': '.',
@@ -105,12 +106,6 @@
           '<(tgvoip_src_loc)/os/linux/AudioInputALSA.h',
           '<(tgvoip_src_loc)/os/linux/AudioOutputALSA.cpp',
           '<(tgvoip_src_loc)/os/linux/AudioOutputALSA.h',
-          '<(tgvoip_src_loc)/os/linux/AudioOutputPulse.cpp',
-          '<(tgvoip_src_loc)/os/linux/AudioOutputPulse.h',
-          '<(tgvoip_src_loc)/os/linux/AudioInputPulse.cpp',
-          '<(tgvoip_src_loc)/os/linux/AudioInputPulse.h',
-          '<(tgvoip_src_loc)/os/linux/AudioPulse.cpp',
-          '<(tgvoip_src_loc)/os/linux/AudioPulse.h',
 
           # POSIX
           '<(tgvoip_src_loc)/os/posix/NetworkSocketPosix.cpp',
