diff --git a/cmake/external/gsl/CMakeLists.txt b/cmake/external/gsl/CMakeLists.txt
index aa8b09a..0bbfb8e 100644
--- a/cmake/external/gsl/CMakeLists.txt
+++ b/cmake/external/gsl/CMakeLists.txt
@@ -7,15 +7,7 @@
 add_library(external_gsl INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_gsl ALIAS external_gsl)
 
-if (DESKTOP_APP_USE_PACKAGED)
-    find_path(GSL_INCLUDE_DIRS gsl/gsl)
-
-    if (NOT GSL_INCLUDE_DIRS)
-        message(FATAL_ERROR "Guidelines Support Library not found")
-    endif()
-else()
     target_include_directories(external_gsl SYSTEM
     INTERFACE
         ${third_party_loc}/GSL/include
     )
-endif()
diff --git a/cmake/external/variant/CMakeLists.txt b/cmake/external/variant/CMakeLists.txt
index a198f90..88a1d4c 100644
--- a/cmake/external/variant/CMakeLists.txt
+++ b/cmake/external/variant/CMakeLists.txt
@@ -7,14 +7,7 @@
 add_library(external_variant INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_variant ALIAS external_variant)
 
-if (DESKTOP_APP_USE_PACKAGED_VARIANT)
-    find_path(VARIANT_INCLUDE_DIRS mapbox/variant.hpp)
-    if (NOT VARIANT_INCLUDE_DIRS)
-        message(FATAL_ERROR "Packaged version of mapbox-variant library not found!")
-    endif()
-else()
     target_include_directories(external_variant SYSTEM
     INTERFACE
         ${third_party_loc}/variant/include
     )
-endif()
