diff --git a/cmake/external/openal/CMakeLists.txt b/cmake/external/openal/CMakeLists.txt
index e2e4992..d047e96 100644
--- a/cmake/external/openal/CMakeLists.txt
+++ b/cmake/external/openal/CMakeLists.txt
@@ -7,37 +7,7 @@
 add_library(external_openal INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_openal ALIAS external_openal)
 
-if (WIN32)
-    target_include_directories(external_openal SYSTEM
-    INTERFACE
-        ${libs_loc}/openal-soft/include
-    )
-    target_link_libraries(external_openal
-    INTERFACE
-        ${libs_loc}/openal-soft/build/$<IF:$<CONFIG:Debug>,Debug,Release>/OpenAL32.lib
-    )
-elseif(APPLE AND NOT build_osx)
-    target_include_directories(external_openal SYSTEM
-    INTERFACE
-        /usr/local/macos/include
-    )
-    target_link_libraries(external_openal
-    INTERFACE
-        /usr/local/macos/lib/libopenal.a
-    )
-else()
-    target_include_directories(external_openal SYSTEM
-    INTERFACE
-        /usr/local/include
-    )
-    target_link_static_libraries(external_openal
-    INTERFACE
-        openal
-    )
-endif()
+find_package(OpenAL REQUIRED)
 
-target_compile_definitions(external_openal
-INTERFACE
-    AL_LIBTYPE_STATIC
-    AL_ALEXT_PROTOTYPES
-)
+target_include_directories(external_openal SYSTEM INTERFACE ${OPENAL_INCLUDE_DIR})
+target_link_libraries(external_openal INTERFACE ${OPENAL_LIBRARY})
