diff --git a/cmake/external/opus/CMakeLists.txt b/cmake/external/opus/CMakeLists.txt
index 00c2508..35c029a 100644
--- a/cmake/external/opus/CMakeLists.txt
+++ b/cmake/external/opus/CMakeLists.txt
@@ -7,39 +7,8 @@
 add_library(external_opus INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_opus ALIAS external_opus)
 
-target_include_directories(external_opus SYSTEM
-INTERFACE
-    ${libs_loc}/opus/include
-)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(OPUS REQUIRED opus)
 
-if (WIN32)
-    set(opus_lib_loc ${libs_loc}/opus/win32/VS2015/Win32/$<IF:$<CONFIG:Debug>,Debug,Release>)
-
-    target_link_libraries(external_opus
-    INTERFACE
-        ${opus_lib_loc}/opus.lib
-    )
-elseif (APPLE AND NOT build_osx)
-    target_include_directories(external_opus
-    INTERFACE
-        /usr/local/macos/include
-    )
-    target_link_libraries(external_opus
-    INTERFACE
-        /usr/local/macos/lib/libopus.a
-    )
-elseif (APPLE)
-    target_include_directories(external_opus
-    INTERFACE
-        /usr/local/include
-    )
-    target_link_libraries(external_opus
-    INTERFACE
-        /usr/local/lib/libopus.a
-    )
-else()
-    target_link_static_libraries(external_opus
-    INTERFACE
-        opus
-    )
-endif()
+target_include_directories(external_opus SYSTEM INTERFACE ${OPUS_INCLUDE_DIRS})
+target_link_libraries(external_opus INTERFACE ${OPUS_LIBRARIES})
