--- a/Telegram/SourceFiles/base/build_config.h
+++ b/Telegram/SourceFiles/base/build_config.h
@@ -51,6 +51,8 @@
 #define ARCH_CPU_64_BITS 1
 #elif defined(_M_ARM) || defined(__arm__)
 #define ARCH_CPU_32_BITS 1
+#elif defined(__PPC64__)
+#define ARCH_CPU_64_BITS 1
 #else
 #error Please add support for your architecture in base/build_config.h
 #endif
--- a/Telegram/ThirdParty/libtgvoip/webrtc_dsp/rtc_base/system/arch.h
+++ b/Telegram/ThirdParty/libtgvoip/webrtc_dsp/rtc_base/system/arch.h
@@ -27,6 +27,10 @@
 #define WEBRTC_ARCH_ARM_FAMILY
 #define WEBRTC_ARCH_64_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__PPC64__)
+#define WEBRTC_ARCH_PPC64
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_LITTLE_ENDIAN
 #elif defined(_M_IX86) || defined(__i386__)
 #define WEBRTC_ARCH_X86_FAMILY
 #define WEBRTC_ARCH_X86
--- a/Telegram/ThirdParty/libtgvoip/webrtc_dsp/typedefs.h
+++ b/Telegram/ThirdParty/libtgvoip/webrtc_dsp/typedefs.h
@@ -27,6 +27,10 @@
 #define WEBRTC_ARCH_ARM_FAMILY
 #define WEBRTC_ARCH_64_BITS
 #define WEBRTC_ARCH_LITTLE_ENDIAN
+#elif defined(__PPC64__)
+#define WEBRTC_ARCH_PPC64
+#define WEBRTC_ARCH_64_BITS
+#define WEBRTC_ARCH_LITTLE_ENDIAN
 #elif defined(_M_IX86) || defined(__i386__)
 #define WEBRTC_ARCH_X86_FAMILY
 #define WEBRTC_ARCH_X86
