diff --git a/Telegram/SourceFiles/core/launcher.cpp b/Telegram/SourceFiles/core/launcher.cpp
index 2547342..8a14dcb 100644
--- a/Telegram/SourceFiles/core/launcher.cpp
+++ b/Telegram/SourceFiles/core/launcher.cpp
@@ -278,6 +278,19 @@ int Launcher::exec() {
 	Platform::start();
 	Ui::DisableCustomScaling();
 
+	// I don't know why path is not in QT_PLUGIN_PATH by default
+	QCoreApplication::addLibraryPath("/usr/lib/qt5/plugins");
+	// without this Telegram doesn't start on Ubuntu 17.04 due GTK errors
+	setenv("QT_STYLE_OVERRIDE", "qwerty", false);
+	// Telegram doesn't start when extraordinary theme is set, see launchpad.net/bugs/1680943
+	unsetenv("QT_QPA_PLATFORMTHEME");
+
+	// unset QT screen scale related envvars
+	unsetenv("QT_SCREEN_SCALE_FACTORS");
+	unsetenv("QT_AUTO_SCREEN_SCALE_FACTOR");
+	unsetenv("QT_SCALE_FACTOR");
+	unsetenv("QT_DEVICE_PIXEL_RATIO");
+
 	auto result = executeApplication();
 
 	DEBUG_LOG(("Telegram finished, result: %1").arg(result));
