diff --git a/Telegram/CMakeLists.txt b/Telegram/CMakeLists.txt
index d8c623a..bd7cea0 100644
--- a/Telegram/CMakeLists.txt
+++ b/Telegram/CMakeLists.txt
@@ -1117,43 +1117,3 @@ else()
 endif()
 
 set_target_properties(Telegram PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${output_folder})
-
-if (NOT build_macstore AND NOT build_winstore)
-    add_executable(Updater WIN32)
-    init_target(Updater)
-
-    add_dependencies(Telegram Updater)
-
-    nice_target_sources(Updater ${src_loc}
-    PRIVATE
-        _other/updater_win.cpp
-        _other/updater_linux.cpp
-        _other/updater_osx.m
-        _other/updater.h
-    )
-
-    set_target_properties(Updater PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${output_folder})
-
-    if (DESKTOP_APP_SPECIAL_TARGET)
-        add_executable(Packer WIN32)
-        init_target(Packer)
-
-        add_dependencies(Telegram Packer)
-
-        nice_target_sources(Packer ${src_loc}
-        PRIVATE
-            _other/packer.cpp
-            _other/packer.h
-        )
-
-        target_link_libraries(Packer
-        PRIVATE
-            desktop-app::external_qt
-            desktop-app::external_zlib
-            desktop-app::external_auto_updates
-            desktop-app::external_openssl
-        )
-
-        set_target_properties(Packer PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${output_folder})
-    endif()
-endif()
