diff --git a/cmake/options_linux.cmake b/cmake/options_linux.cmake
index a6a13c9..20af0ae 100644
--- a/cmake/options_linux.cmake
+++ b/cmake/options_linux.cmake
@@ -9,7 +9,7 @@ INTERFACE
     $<IF:$<CONFIG:Debug>,,-Ofast -fno-strict-aliasing>
     -pipe
     -Wall
-    -Werror
+	-O2
     -W
     -fPIC
     -Wno-unused-variable
@@ -28,13 +28,5 @@ INTERFACE
 )
 target_link_options(common_options
 INTERFACE
-    $<IF:$<CONFIG:Debug>,,-Ofast>
+    $<IF:$<CONFIG:Debug>,,-O2>
 )
-if (build_linux32)
-    target_compile_options(common_options INTERFACE -g0)
-    target_link_options(common_options INTERFACE -g0)
-else()
-    target_compile_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto>)
-    target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
-endif()
-
