diff -urp patchelf-0.9-orig/tests/set-rpath-library.sh patchelf-0.9/tests/set-rpath-library.sh
--- patchelf-0.9-orig/tests/set-rpath-library.sh	2016-02-29 12:15:58.000000000 +0200
+++ patchelf-0.9/tests/set-rpath-library.sh	2016-07-10 03:09:55.365997370 +0300
@@ -26,7 +26,9 @@ exitCode=0
 
 if test "$exitCode" = 46; then
     echo "expected failure"
-    exit 1
+#    exit 1
+# As it is an "expected failure" the test should not exit but should go on.
+# See the comment below.
 fi
 
 # So set an RUNPATH on libfoo as well.
