diff --git a/test_project/pyproject.toml.orig b/test_project/pyproject.toml
index 6284d4cb1d5..1a70c595702 100644
--- a/test_project/pyproject.toml.orig
+++ b/test_project/pyproject.toml
@@ -19,3 +19,5 @@ py-gte-38 = "sys_version_info >= (3, 8)"
 py-gte-39 = "sys_version_info >= (3, 9)"
 py-gte-310 = "sys_version_info >= (3, 10)"
 py-gte-311 = "sys_version_info >= (3, 11)"
+py-gte-312 = "sys_version_info >= (3, 12)"
+py-gte-313 = "sys_version_info >= (3, 13)"
diff --git a/test_project/example.py.orig b/test_project/example.py
index 132d3f5100c..aa259cb5856 100644
--- a/test_project/example.py.orig
+++ b/test_project/example.py
@@ -1,5 +1,13 @@
 from typing import Tuple
 
+def if_gte_python313() -> Tuple[int, int]:  # pragma: py-gte-313
+    """Test function for pragma ``py-gte-313``."""
+    return (3, 13)
+
+def if_gte_python312() -> Tuple[int, int]:  # pragma: py-gte-312
+    """Test function for pragma ``py-gte-312``."""
+    return (3, 12)
+
 def if_gte_python311() -> Tuple[int, int]:  # pragma: py-gte-311
     """Test function for pragma ``py-gte-311``."""
     return (3, 11)
diff --git a/tests/test_integration.py.orig b/tests/test_integration.py
index 8123609e7c9..117814e8708 100644
--- a/tests/test_integration.py.orig
+++ b/tests/test_integration.py
@@ -13,6 +13,8 @@ from test_project.example import (
     if_gte_python39,
     if_gte_python310,
     if_gte_python311,
+    if_gte_python312,
+    if_gte_python313,
 )
 
 #: This is just our specific example.
@@ -41,6 +43,8 @@ def test_integration(cov, capsys):
     if_gte_python39()
     if_gte_python310()
     if_gte_python311()
+    if_gte_python312()
+    if_gte_python313()
     always()
 
     cov.json_report(outfile='-')
diff --git a/test_project/.coveragerc.orig b/test_project/.coveragerc
index d95926f2066..80591b3dacf 100644
--- a/test_project/.coveragerc.orig
+++ b/test_project/.coveragerc
@@ -22,3 +22,5 @@ rules =
   "sys_version_info >= (3, 9)": py-gte-39
   "sys.version_info >= (3, 10)": py-gte-310
   "sys.version_info >= (3, 11)": py-gte-311
+  "sys.version_info >= (3, 12)": py-gte-312
+  "sys.version_info >= (3, 13)": py-gte-313
