diff --git a/Makefile b/Makefile
index 216a251..cffc2ad 100644
--- a/Makefile
+++ b/Makefile
@@ -1008,6 +1008,7 @@ check-symbols: $(STARTUP_FILES) libc
 	    | grep -v '^#define assert' \
 	    | grep -v '^#define __NO_INLINE__' \
 	    | grep -v '^#define __U\?INT.*_C(' \
+	    | grep -v '^#define __wasip[12]__' \
 	    > "$(SYSROOT_SHARE)/predefined-macros.txt"
 
 	# Check that the computed metadata matches the expected metadata.
