--- fuse-2.9.3.orig/util/fusermount.c	2015-02-07 08:58:54.000000000 -0200
+++ fuse-2.9.3/util/fusermount.c	2015-02-09 09:15:16.737367957 -0200
@@ -1255,19 +1255,16 @@
 
 	origmnt = argv[optind];
 
-	drop_privs();
 	mnt = fuse_mnt_resolve_path(progname, origmnt);
-	if (mnt != NULL) {
-		res = chdir("/");
-		if (res == -1) {
-			fprintf(stderr, "%s: failed to chdir to '/'\n", progname);
-			exit(1);
-		}
-	}
-	restore_privs();
 	if (mnt == NULL)
 		exit(1);
 
+	res = chdir("/");
+	if (res == -1) {
+		fprintf(stderr, "%s: failed to chdir to '/'\n", progname);
+		exit(1);
+	}
+
 	umask(033);
 	if (unmount)
 		goto do_unmount;
