--- fuse-2.9.3.orig/lib/fuse.c
+++ fuse-2.9.3/lib/fuse.c
@@ -35,6 +35,7 @@
 #include <poll.h>
 #include <sys/param.h>
 #include <sys/uio.h>
+#include <sys/file.h>
 #include <sys/time.h>
 #include <sys/mman.h>
 
--- fuse-2.9.3.orig/lib/fuse_lowlevel.c
+++ fuse-2.9.3/lib/fuse_lowlevel.c
@@ -24,6 +24,7 @@
 #include <limits.h>
 #include <errno.h>
 #include <assert.h>
+#include <sys/file.h>
 
 #ifndef F_LINUX_SPECIFIC_BASE
 #define F_LINUX_SPECIFIC_BASE       1024
--- fuse-2.9.3.orig/lib/mount_util.c
+++ fuse-2.9.3/lib/mount_util.c
@@ -15,6 +15,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <paths.h>
 #include <limits.h>
 #ifndef __NetBSD__
 #include <mntent.h>
--- fuse-2.9.3.orig/util/fusermount.c
+++ fuse-2.9.3/util/fusermount.c
@@ -19,6 +19,7 @@
 #include <getopt.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <paths.h>
 #include <pwd.h>
 #include <mntent.h>
 #include <sys/wait.h>
--- fuse-2.9.3.orig/util/mount_util.c
+++ fuse-2.9.3/util/mount_util.c
@@ -15,6 +15,7 @@
 #include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <paths.h>
 #include <limits.h>
 #ifndef __NetBSD__
 #include <mntent.h>
