diff --git a/lib/Makefile.am b/lib/Makefile.am
index 3eec119..c7414ac 100644
--- a/lib/Makefile.am
+++ b/lib/Makefile.am
@@ -11,8 +11,10 @@ else
 mount_source = mount.c mount_util.c mount_util.h
 endif
 
+libfuse_la_LDFLAGS =
 if ICONV
 iconv_source = modules/iconv.c
+libfuse_la_LDFLAGS += -liconv
 else
 iconv_source =
 endif
@@ -35,7 +37,7 @@ libfuse_la_SOURCES = 		\
 	$(iconv_source)		\
 	$(mount_source)
 
-libfuse_la_LDFLAGS = @libfuse_libs@ -version-number 2:8:1 \
+libfuse_la_LDFLAGS += @libfuse_libs@ -version-number 2:8:1 \
 	-Wl,--version-script,$(srcdir)/fuse_versionscript
 
 libulockmgr_la_SOURCES = ulockmgr.c
