--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,6 +140,7 @@
 
 	ProcessPackage(LIBLUA LIBRARY	"lua"
 			"lua-5.3"
+			LIB_SUFFIXES	"lua5.3"
 			INCLUDE lua.h INCLUDE_SUFFIXES
 			"include/lua-5.3"
 			"include/lua5.3"
@@ -152,6 +153,7 @@
 	IF (NOT WITH_LIBLUA)
 		ProcessPackage(LIBLUA LIBRARY	"lua"
 				"lua-5.2"
+				LIB_SUFFIXES	"lua5.2"
 				INCLUDE lua.h INCLUDE_SUFFIXES
 				"include/lua-5.2"
 				"include/lua5.2"
