Distro-specific adjustments of the default configs.

--- a/conf/options.inc
+++ b/conf/options.inc
@@ -8,7 +8,7 @@
 map_watch_interval = 5min;
 # Multiplier for watch interval for files
 map_file_watch_multiplier = 0.1;
-dynamic_conf = "$DBDIR/rspamd_dynamic";
+dynamic_conf = "$DBDIR/dynamic";
 history_file = "$DBDIR/rspamd.history";
 check_all_filters = false;
 dns {
@@ -25,7 +25,7 @@
 	"X-MimeOLE",
 ];
 
-control_socket = "$DBDIR/rspamd.sock mode=0600";
+control_socket = "$RUNDIR/rspamd.sock mode=0600";
 history_rows = 200;
 explicit_modules = ["settings", "bayes_expiry"];
 
--- a/conf/rspamd.conf
+++ b/conf/rspamd.conf
@@ -18,7 +18,6 @@
 .include "$CONFDIR/common.conf"
 
 options {
-    pidfile = "$RUNDIR/rspamd.pid";
     .include "$CONFDIR/options.inc"
     .include(try=true; priority=1,duplicate=merge) "$LOCAL_CONFDIR/local.d/options.inc"
     .include(try=true; priority=10) "$LOCAL_CONFDIR/override.d/options.inc"
