From b563e16adb19086f99f9ecf6b3c3636f44229758 Mon Sep 17 00:00:00 2001
From: Vsevolod Stakhov <vsevolod@highsecure.ru>
Date: Tue, 8 May 2018 20:44:10 +0100
Subject: [PATCH] [Fix] Fix build on non-x86 platforms

Issue: #2226
---
 contrib/t1ha/t1ha.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/contrib/t1ha/t1ha.h b/contrib/t1ha/t1ha.h
index 30c408c87..191758c57 100644
--- a/contrib/t1ha/t1ha.h
+++ b/contrib/t1ha/t1ha.h
@@ -217,13 +217,13 @@
 #define T1HA_API
 #endif /* T1HA_API */
 
-#if defined(_MSC_VER) && defined(__ia32__)
+#if defined(_MSC_VER)
 #define T1HA_ALIGN_PREFIX __declspec(align(32)) /* required only for SIMD */
 #else
 #define T1HA_ALIGN_PREFIX
 #endif /* _MSC_VER */
 
-#if defined(__GNUC__) && defined(__ia32__)
+#if defined(__GNUC__)
 #define T1HA_ALIGN_SUFFIX                                                      \
   __attribute__((aligned(32))) /* required only for SIMD */
 #else
