diff --git a/test/lua/unit/utf.lua b/test/lua/unit/utf.lua
index dbdab7f..5d59d14 100644
--- a/test/lua/unit/utf.lua
+++ b/test/lua/unit/utf.lua
@@ -78,7 +78,7 @@ context("UTF8 check functions", function()
   end
 
   -- Enable sse and avx2
-  ffi.C.rspamd_fast_utf8_library_init(3)
+  ffi.C.rspamd_fast_utf8_library_init(0)
   local valid_cases = {
     "a",
     "\xc3\xb1",
@@ -192,14 +192,6 @@ context("UTF8 check functions", function()
           local res = test_size(sz, false, 'sse')
           assert_equal(res, 0)
         end)
-        test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'valid'), function()
-          local res = test_size(sz, true, 'avx2')
-          assert_equal(res, 0)
-        end)
-        test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'invalid'), function()
-          local res = test_size(sz, false, 'avx2')
-          assert_equal(res, 0)
-        end)
       end
     end
   end
