diff --git a/test/lua/unit/utf.lua b/test/lua/unit/utf.lua
index 67ccfb4..ab9c1a7 100644
--- a/test/lua/unit/utf.lua
+++ b/test/lua/unit/utf.lua
@@ -191,15 +191,15 @@ context("UTF8 check functions", function()
         local res = test_size(sz, false, 'sse')
         assert_equal(res, 0)
       end)
-      test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'valid'), function()
-        local res = test_size(sz, true, 'avx2')
-        assert_equal(res, 0)
-      end)
-      test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'invalid'), function()
-        local res = test_size(sz, false, 'avx2')
-        assert_equal(res, 0)
-      end)
+      -- test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'valid'), function()
+      --   local res = test_size(sz, true, 'avx2')
+      --   assert_equal(res, 0)
+      -- end)
+      -- test(string.format("Utf8 test %s %d buffer, %s", 'avx2', sz, 'invalid'), function()
+      --   local res = test_size(sz, false, 'avx2')
+      --   assert_equal(res, 0)
+      -- end)
     end
   end
 
-end)
\ No newline at end of file
+end)
