diff -Nurp a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-10-16 10:33:53.000000000 +0000
+++ b/CMakeLists.txt	2024-10-17 05:28:54.816502585 +0000
@@ -799,10 +799,13 @@ ADD_CUSTOM_TARGET(dist ${CMAKE_SOURCE_DI
         WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
 
 ADD_CUSTOM_TARGET(check DEPENDS rspamd-test-cxx rspamd-test)
-ADD_CUSTOM_TARGET(run-test DEPENDS check
-        COMMAND test/rspamd-test-cxx
+ADD_CUSTOM_TARGET(run-test DEPENDS check test-cmd)
+ADD_CUSTOM_COMMAND(OUTPUT test-cmd COMMAND test/rspamd-test-cxx)
+IF (ENABLE_LUAJIT MATCHES "ON")
+	ADD_CUSTOM_COMMAND(OUTPUT test-cmd APPEND
         COMMAND sh -c 'LUA_PATH="${CMAKE_SOURCE_DIR}/lualib/?.lua\;${CMAKE_SOURCE_DIR}/lualib/?/?.lua\;${CMAKE_SOURCE_DIR}/lualib/?/init.lua\;${CMAKE_SOURCE_DIR}/contrib/lua-?/?.lua"
         test/rspamd-test -p /rspamd/lua')
+ENDIF ()
 
 
 # PVS Studio
