diff --git a/docs/cookbook/container/default.sha256.do b/docs/cookbook/container/default.sha256.do
index d3007e9..b414d02 100644
--- a/docs/cookbook/container/default.sha256.do
+++ b/docs/cookbook/container/default.sha256.do
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Calculate the sha256 digest of a given file."""
 import hashlib, os, subprocess, sys
 
diff --git a/docs/cookbook/container/dockjson.py b/docs/cookbook/container/dockjson.py
index 11a4bec..133a55f 100755
--- a/docs/cookbook/container/dockjson.py
+++ b/docs/cookbook/container/dockjson.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Generate a docker 1.0-style manifest for a docker image."""
 import json, os, sys, time
 
diff --git a/docs/cookbook/container/fileids.py b/docs/cookbook/container/fileids.py
index 68d9982..667c54c 100755
--- a/docs/cookbook/container/fileids.py
+++ b/docs/cookbook/container/fileids.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import hashlib, os, stat, sys
 
 for name in sys.stdin:
diff --git a/docs/cookbook/container/memcalc.py b/docs/cookbook/container/memcalc.py
index 2301874..7c43e11 100755
--- a/docs/cookbook/container/memcalc.py
+++ b/docs/cookbook/container/memcalc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import os, sys
 st = os.stat(sys.argv[1])
 megabytes = st.st_size // 1024 // 1024
diff --git a/docs/cookbook/defaults/test.py b/docs/cookbook/defaults/test.py
index fa679d2..868cae5 100644
--- a/docs/cookbook/defaults/test.py
+++ b/docs/cookbook/defaults/test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Test program for auto-generated version.py"""
 import version

diff --git a/redo/py.do b/redo/py.do
index 11d4bf4..d293d09 100644
--- a/redo/py.do
+++ b/redo/py.do
@@ -1,7 +1,5 @@
-redo-ifchange whichpython
-read py <whichpython
 cat >$3 <<-EOF
 	#!/bin/sh
-	exec $py "\$@"
+	exec /usr/bin/python3 "\$@"
 EOF
 chmod a+x $3
