commit 462cd6c81a2321872a5807e652780eb90a0fb40e
Author: Isaac Dunham <ibid.ag@gmail.com>
Date:   Tue Jul 15 14:39:37 2014 -0700

    Add -Iheaders so we can find gpm.h

diff --git a/src/Makefile.in b/src/Makefile.in
index d3b1dcd..1f8a4dc 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -65,7 +65,7 @@ prog/%:	prog/%.o
 	$(CC) @LDFLAGS@ $(LDFLAGS) -o $@ $^ @LIBS@ $(LIBS)
 
 %.o: %.c
-	$(CC) @CPPFLAGS@ $(CPPFLAGS) @CFLAGS@ $(CFLAGS) -c -o $@ $<
+	$(CC) -Iheaders @CPPFLAGS@ $(CPPFLAGS) @CFLAGS@ $(CFLAGS) -c -o $@ $<
 
 %.lo:	%.c
 	$(CC) @CPPFLAGS@ $(CPPFLAGS) @PICFLAGS@ @CFLAGS@ $(CFLAGS) -c -o $@ $<
@@ -96,7 +96,7 @@ $(DEPFILE) dep: prog/gpm-root.c
 
 	# create dependencies
 	for DEPS in `echo *.c */*.c`; do \
-	$(CC) -I. -I $(srcdir) -M @CPPFLAGS@ $(CPPFLAGS) $$DEPS | \
+	$(CC) -I. -Iheaders -I $(srcdir) -M @CPPFLAGS@ $(CPPFLAGS) $$DEPS | \
 	$(SED) 's/^\(.*\)\.o\([ :]+\)/\1.o \1.lo\2/g' >> $(DEPFILE) ; done
 
 ### INSTALL
diff --git a/src/daemon/open_console.c b/src/daemon/open_console.c
index 98297c9..f7df875 100644
--- a/src/daemon/open_console.c
+++ b/src/daemon/open_console.c
@@ -22,6 +22,7 @@
 #include <fcntl.h>                  /* open and co.      */
 #include <sys/stat.h>               /* stat()            */
 #include <sys/ioctl.h>              /* ioctl             */
+#include <sys/sysmacros.h>          /* major()           */
 
 /* Linux specific (to be outsourced in gpm2 */
 #include <linux/serial.h>           /* for serial console check */
