--- a/modules/afamqp/rabbitmq-c/librabbitmq/amqp_openssl.c
+++ b/modules/afamqp/rabbitmq-c/librabbitmq/amqp_openssl.c
@@ -664,7 +664,10 @@ destroy_openssl(void)
      * an issue if you repeatedly unload and load the library
      */
     ERR_remove_state(0);
+/* LibreSSL patch for Alpine */
+#ifndef LIBRESSL_VERSION_NUMBER
     FIPS_mode_set(0);
+#endif
     CRYPTO_set_locking_callback(NULL);
     CRYPTO_set_id_callback(NULL);
     ENGINE_cleanup();
