diff --git a/modules/cloud-auth/Makefile.am b/modules/cloud-auth/Makefile.am
index 488043e..dea82c1 100644
--- a/modules/cloud-auth/Makefile.am
+++ b/modules/cloud-auth/Makefile.am
@@ -38,7 +38,8 @@ modules_cloud_auth_libcloud_auth_la_CPPFLAGS = \
 
 modules_cloud_auth_libcloud_auth_la_LIBADD = \
   $(MODULE_DEPS_LIBS) \
-  $(top_builddir)/modules/cloud-auth/libcloud_auth_cpp.la
+  $(top_builddir)/modules/cloud-auth/libcloud_auth_cpp.la \
+  -lstdc++
 
 modules_cloud_auth_libcloud_auth_la_LDFLAGS = \
   $(MODULE_LDFLAGS)
