diff --git a/google/cloud/testing_util/command_line_parsing.cc b/google/cloud/testing_util/command_line_parsing.cc
index 330d6d402b..92d90842a6 100644
--- a/google/cloud/testing_util/command_line_parsing.cc
+++ b/google/cloud/testing_util/command_line_parsing.cc
@@ -16,6 +16,7 @@
 #include "google/cloud/internal/throw_delegate.h"
 #include "absl/strings/match.h"
 #include "absl/time/time.h"
+#include <algorithm>
 #include <cctype>
 #include <iomanip>
 #include <sstream>
