diff -Nur mozilla-1.9.2.orig/browser/app/profile/firefox.js mozilla-1.9.2/browser/app/profile/firefox.js
--- mozilla-1.9.2.orig/browser/app/profile/firefox.js	2010-04-14 01:21:35.000000000 +0300
+++ mozilla-1.9.2/browser/app/profile/firefox.js	2010-04-21 00:16:59.673528609 +0300
@@ -183,7 +183,7 @@
 
 pref("general.useragent.locale", "@AB_CD@");
 pref("general.skins.selectedSkin", "classic/1.0");
-pref("general.useragent.extra.firefox", "@APP_UA_NAME@/@APP_VERSION@");
+pref("general.useragent.extra.firefox", "Firefox/@APP_VERSION@");
 
 pref("general.smoothScroll", false);
 #ifdef UNIX_BUT_NOT_MAC
