diff -rc /tmp/mozilla-release/browser/base/content/browser.js mozilla-release/browser/base/content/browser.js
*** /tmp/mozilla-release/browser/base/content/browser.js        2013-05-11 14:19:21.000000000 -0500
--- mozilla-release/browser/base/content/browser.js     2013-05-19 01:42:13.778359190 -0500
***************
*** 3559,3568 ****
     */
    recordSearchInHealthReport: function (engine, source) {
  #ifdef MOZ_SERVICES_HEALTHREPORT
!     let reporter = Cc["@mozilla.org/datareporting/service;1"]
!                      .getService()
!                      .wrappedJSObject
!                      .healthReporter;
  
      // This can happen if the FHR component of the data reporting service is
      // disabled. This is controlled by a pref that most will never use.
--- 3559,3565 ----
     */
    recordSearchInHealthReport: function (engine, source) {
  #ifdef MOZ_SERVICES_HEALTHREPORT
!       return;
  
      // This can happen if the FHR component of the data reporting service is
      // disabled. This is controlled by a pref that most will never use.
