diff --git a/panel-plugin/simple-libmpd.c b/panel-plugin/simple-libmpd.c
index 29104f9..f33b98d 100644
--- a/panel-plugin/simple-libmpd.c
+++ b/panel-plugin/simple-libmpd.c
@@ -39,7 +39,7 @@
 
 int mpd_wait_for_answer(MpdObj *);
 int mpd_send_single_cmd(MpdObj*, char*);
-void send_complex_cmd(MpdObj*, char*, void (*)(), void *);
+void send_complex_cmd(MpdObj*, char*, void (*)(MpdObj *, void *), void *);
 void parse_status_answer(MpdObj*, void*);
 void parse_one_song(MpdObj*, void*);
 void parse_playlistinfo_answer(MpdObj*, void*);
@@ -273,7 +273,7 @@ int mpd_send_single_cmd(MpdObj*mo, char* cmd)
    return ((!mo->error) ? MPD_OK : MPD_FAILED);
 }
 
-void send_complex_cmd(MpdObj* mo, char* cmd, void (*parse_answer_fct)(), void *res)
+void send_complex_cmd(MpdObj* mo, char* cmd, void (*parse_answer_fct)(MpdObj *, void *), void *res)
 {
    int nbwri, nbread, tmp_bufsize;
    char *ptr;
