diff --git a/scc/parser.py b/scc/parser.py
index 78633f0e..4dcf3fa3 100644
--- a/scc/parser.py
+++ b/scc/parser.py
@@ -109,7 +109,7 @@ class ActionParser(object):
 				ActionParser.Token(type, string)
 				for type, string, *_
 				in generate_tokens( iter([s]).__next__ )
-				if type != TokenType.ENDMARKER
+				if type != TokenType.ENDMARKER and string != ''
 			]
 		except TokenError:
 			self.tokens = None
