--- a/config/config.inc.php.sample
+++ b/config/config.inc.php.sample
@@ -74,7 +74,7 @@
 // in the session record (and the client cookie if remember password is enabled).
 // please provide a string of exactly 24 chars.
 // YOUR KEY MUST BE DIFFERENT THAN THE SAMPLE VALUE FOR SECURITY REASONS
-$config['des_key'] = 'rcmail-!24ByteDESkey*Str';
+$config['des_key'] = trim(file(RCMAIL_CONFIG_DIR . '/session_key')[0]);
 
 // List of active plugins (in plugins/ directory)
 $config['plugins'] = array(
