--- a/program/include/iniset.php
+++ b/program/include/iniset.php
@@ -44,7 +44,7 @@
 }
 
 if (!defined('RCMAIL_CONFIG_DIR')) {
-    define('RCMAIL_CONFIG_DIR', INSTALL_PATH . 'config');
+    define('RCMAIL_CONFIG_DIR', '/etc/roundcube');
 }
 
 if (!defined('RCUBE_LOCALIZATION_DIR')) {
--- a/program/lib/Roundcube/bootstrap.php
+++ b/program/lib/Roundcube/bootstrap.php
@@ -65,7 +65,7 @@
 }
 
 if (!defined('RCUBE_CONFIG_DIR')) {
-    define('RCUBE_CONFIG_DIR', RCUBE_INSTALL_PATH . 'config/');
+    define('RCUBE_CONFIG_DIR', '/etc/roundcube');
 }
 
 if (!defined('RCUBE_PLUGINS_DIR')) {
