--- a/installer/index.php
+++ b/installer/index.php
@@ -41,7 +41,7 @@
 
 define('INSTALL_PATH', realpath(dirname(__FILE__) . '/../').'/');
 define('RCUBE_INSTALL_PATH', INSTALL_PATH);
-define('RCUBE_CONFIG_DIR', INSTALL_PATH . 'config/');
+define('RCUBE_CONFIG_DIR', '/etc/roundcube');
 
 $include_path  = INSTALL_PATH . 'program/lib' . PATH_SEPARATOR;
 $include_path .= INSTALL_PATH . 'program/include' . PATH_SEPARATOR;
--- a/program/include/iniset.php
+++ b/program/include/iniset.php
@@ -44,7 +44,7 @@
 }
 
 if (!defined('RCMAIL_CONFIG_DIR')) {
-    define('RCMAIL_CONFIG_DIR', INSTALL_PATH . 'config');
+    define('RCMAIL_CONFIG_DIR', '/etc/roundcube');
 }
 
 if (!defined('RCUBE_LOCALIZATION_DIR')) {
