--- pptpd-1.3.4/Makefile.am
+++ pptpd-1.3.4-orig/Makefile.am
@@ -11,7 +11,8 @@
 ##  warning with -Wmissing-prototypes).
 ## -Wmissing-prototypes removed (eg, Linux 2.2.6 headers
 ##  aren't up to it).
-CFLAGS = -O2 -fno-builtin -Wall -DSBINDIR='"$(sbindir)"'
+OPTFLAGS=-O2
+CFLAGS = $(OPTFLAGS) -fno-builtin -Wall -DSBINDIR='"$(sbindir)"'
 #CFLAGS = -O2 -fno-builtin -Wall -ansi -DSBINDIR='"$(sbindir)"'
 #CFLAGS = -O2 -fno-builtin -Wall -ansi -pedantic -Wmissing-prototypes -Werror -DSBINDIR='"$(sbindir)"'
 
--- pptpd-1.3.4/plugins/Makefile
+++ pptpd-1.3.4-orig/plugins/Makefile
@@ -3,7 +3,7 @@
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 LDADD   = -lutil
-INSTALL	= install -o root
+INSTALL	= install
 prefix  = /usr/local
 
 PLUGINS = pptpd-logwtmp.so
@@ -18,7 +18,7 @@
 %.so: %.c
 	$(CC) -o $@ $(LDFLAGS) $(CFLAGS) $^ $(LDADD)
 
-LIBDIR	?= $(DESTDIR)$(prefix)/lib/pptpd
+LIBDIR	= $(DESTDIR)$(prefix)/lib/pptpd
 
 install: $(PLUGINS)
 	$(INSTALL) -d $(LIBDIR)
