From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 15 Nov 2024 01:42:00 +0100
Subject: Log to syslog even when not started with -d (daemon).

--- a/src/core/main.cpp
+++ b/src/core/main.cpp
@@ -269,9 +269,10 @@
         return EXIT_SUCCESS;
     }
 
+    ipx_verb_syslog(true);
+
     if (daemon_en) {
         // Run as a standalone daemon process
-        ipx_verb_syslog(true);
         if (daemon(1, 0) == -1) {
             const char *err_str;
             ipx_strerror(errno, err_str);
