From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 15 Nov 2024 01:30:00 +0100
Subject: Don't use git

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,7 +20,6 @@
     ${IPFIXCOL_VERSION_MAJOR}.${IPFIXCOL_VERSION_MINOR}.${IPFIXCOL_VERSION_PATCH})
 
 # Include modules
-include(CMakeModules/git_info.cmake)
 include(CMakeModules/install_dirs.cmake)
 include(CheckCCompilerFlag)
 include(CheckCXXCompilerFlag)
