fixes maybe-used-uninitialised errors
--- a/tests/bigint_test.c
+++ b/tests/bigint_test.c
@@ -210,7 +210,7 @@
 	static const uint8_t addend_raw[] = addend;			\
 	static const uint8_t value_raw[] = value;			\
 	static const uint8_t expected_raw[] = expected;			\
-	uint8_t result_raw[ sizeof ( expected_raw ) ];			\
+	uint8_t result_raw[ sizeof ( expected_raw ) ] = {0};		\
 	unsigned int size =						\
 		bigint_required_size ( sizeof ( value_raw ) );		\
 	bigint_t ( size ) addend_temp;					\
--- a/drivers/net/ath/ath5k/ath5k_eeprom.c
+++ b/drivers/net/ath/ath5k/ath5k_eeprom.c
@@ -410,7 +410,7 @@
 {
 	struct ath5k_eeprom_info *ee = &ah->ah_capabilities.cap_eeprom;
 	u32 o = *offset;
-	u16 val;
+	u16 val = 0;
 	int ret;
 
 	if (ee->ee_version < AR5K_EEPROM_VERSION_5_0)
