--- a/sources/Archive.h
+++ b/sources/Archive.h
@@ -18,6 +18,8 @@
 #include "Rule.h"
 #include "Setup.h"
 
+#define uint unsigned int
+
 using namespace std;
 
 /**
--- a/sources/Problem.h
+++ b/sources/Problem.h
@@ -18,6 +18,7 @@
 #include "Rule.h"
 #include "Setup.h"
 
+#define uint unsigned int
 
 using namespace std;
 
--- a/sources/Rule.h
+++ b/sources/Rule.h
@@ -14,6 +14,8 @@
 
 #include "Feature.h"
 
+#define uint unsigned int
+
 using namespace std;
 
 /**
--- a/sources/Setup.cpp
+++ b/sources/Setup.cpp
@@ -2,6 +2,8 @@
 
 #include <bits/stdc++.h>
 
+#define uint unsigned int
+
 Setup::Setup(string fname)
 {
 	// TODO Auto-generated constructor stub
--- a/sources/Squash.cpp
+++ b/sources/Squash.cpp
@@ -7,6 +7,8 @@
 
 #include "Squash.h"
 
+#define uint unsigned int
+
 Squash::Squash() {
 	// TODO Auto-generated constructor stub
 
