--- shorewall-common-4.2.11.orig/lib.cli
+++ shorewall-common-4.2.11/lib.cli
@@ -315,8 +315,6 @@
 					grep 'loadmodule ip_set' ${VARDIR}/.modules >> $f
 					echo "__EOF__" >> $f
 					echo  >> $f
-					echo "ipset -U :all: :all:" >> $f
-					echo "ipset -U :all: :default:" >> $f
 					echo "ipset -F" >> $f
 					echo "ipset -X" >> $f
 					echo "ipset -R << __EOF__" >> $f
--- shorewall-common-4.2.11.orig/lib.config
+++ shorewall-common-4.2.11/lib.config
@@ -884,7 +884,7 @@
 	    ;;
     esac
 
-    echo "--set ${setname#+} $options"
+    echo "--match-set ${setname#+} $options"
 }
 
 #
