diff --git a/Makefile.linux b/Makefile.linux
index 7b27008..0afa70c 100644
--- a/Makefile.linux
+++ b/Makefile.linux
@@ -17,10 +17,11 @@
 WHALE="+"
 ONI="-"
 COMMANDS += containerd-shim containerd-shim-runc-v1 containerd-shim-runc-v2
+BUILDMODE := pie
 
 # check GOOS for cross compile builds
 ifeq ($(GOOS),linux)
-	GO_GCFLAGS += -buildmode=pie
+	GO_GCFLAGS += -buildmode=$(BUILDMODE)
 endif
 
 # amd64 supports go test -race
