From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 20 Feb 2023 18:09:42 +0100
Subject: [PATCH] Remove RPATH

--- a/configure
+++ b/configure
@@ -518,7 +518,6 @@
 
 if ($sysname eq "linux") {
 	$DEFS{"DEF"} .= " -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64";
-	$DEFS{"LIB_MAVIS"} = "$DEFS{'LDOPT_R'}\$\$ORIGIN/../lib$DEFS{'LIBARCH'}"
 }
 
 if (exists $DEFS{"DEBUG"}) {
@@ -528,7 +527,7 @@
 }
 
 $DEFS{"DIR_MAVIS"} = "$cwd/mavis";
-$DEFS{"LIB_MAVIS"} .= " -L$cwd/build/$DEFS{'OS'}/mavis $DEFS{'LDOPT_R'}$DEFS{'LIBARCHDIR_DEST'} -lmavis";
+$DEFS{"LIB_MAVIS"} .= " -L$cwd/build/$DEFS{'OS'}/mavis -lmavis";
 $DEFS{"DIR_MISC"} = "$cwd/misc";
 $DEFS{"INC"} .= " -I$cwd";
 
@@ -616,7 +615,7 @@
 	if (exists $DEFS{"WITH_$PKG"} && $DEFS{"WITH_$PKG"} == 1) {
 		$DEFS{"INC_$PKG"} = "-I" . $DEFS{"INCDIR_$PKG"} if exists $DEFS{"INCDIR_$PKG"};
 		$DEFS{"LIB_$PKG"} = "";
-		$DEFS{"LIB_$PKG"} .= "-L" . $DEFS{"LIBDIR_$PKG"} . " " . $DEFS{'LDOPT_R'} . $DEFS{"LIBDIR_$PKG"} . " " if exists $DEFS{"LIBDIR_$PKG"};
+		$DEFS{"LIB_$PKG"} .= "-L" . $DEFS{"LIBDIR_$PKG"} . " " if exists $DEFS{"LIBDIR_$PKG"};
 		$DEFS{"LIB_$PKG"} .= "-l$lib" if defined $lib;
 		$DEFS{'DEF'} .= " -DWITH_$PKG";
 		$DEFS{'DEF'} .= " $defs" if defined $defs;;
