diff -rupN a/mopidy_local/__init__.py b/mopidy_local/__init__.py
--- a/mopidy_local/__init__.py	2024-12-17 23:00:55.000000000 +0000
+++ b/mopidy_local/__init__.py	2026-03-31 02:05:19.880000000 +0000
@@ -1,10 +1,10 @@
 import pathlib
 
-import pkg_resources
+from importlib.metadata import version
 
 from mopidy import config, ext
 
-__version__ = pkg_resources.get_distribution("Mopidy-Local").version
+__version__ = version("Mopidy-Local")
 
 
 class Extension(ext.Extension):
