From 7577fa21177442aacc2d86640ef28cebf1c4aaef Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@barrera.io>
Date: Tue, 9 Jun 2020 17:04:46 +0200
Subject: [PATCH] Update usage of deprecated method

---
 vdirsyncer/storage/dav.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vdirsyncer/storage/dav.py b/vdirsyncer/storage/dav.py
index 07d164fb..a7e2422c 100644
--- a/vdirsyncer/storage/dav.py
+++ b/vdirsyncer/storage/dav.py
@@ -124,7 +124,7 @@ def _merge_xml(items):
         return None
     rv = items[0]
     for item in items[1:]:
-        rv.extend(item.getiterator())
+        rv.extend(item.iter())
     return rv
 
 
