--- gcc-4.3.2.orig/gcc/common.opt	Tue Jan 22 14:11:44 2008
+++ gcc-4.3.2/gcc/common.opt	Tue Nov 18 10:00:49 2008
@@ -147,7 +147,7 @@
 Warn when one local variable shadows another
 
 Wstack-protector
-Common Var(warn_stack_protect) Warning
+Common Var(warn_stack_protect) Warning Init(1)
 Warn when not issuing stack smashing protection for some reason
 
 Wstrict-aliasing
@@ -766,7 +766,7 @@
 Generate position-independent code if possible (large mode)
 
 fPIE
-Common Report Var(flag_pie,2)
+Common Report Var(flag_pie,2) Init(2)
 Generate position-independent code for executables if possible (large mode)
 
 fpic
@@ -978,7 +978,7 @@
 Use propolice as a stack protection method
 
 fstack-protector-all
-Common Report RejectNegative Var(flag_stack_protect, 2) VarExists
+Common Report RejectNegative Var(flag_stack_protect, 2) init(2)
 Use a stack protection method for every function
 
 fstrength-reduce
