Remove bundled zlib.
--- a/Makefile.common
+++ b/Makefile.common
@@ -8,7 +8,6 @@
 			-I$(CORE_DIR)/libretro/emu \
 			-I$(CORE_DIR)/libretro \
 			-I$(CORE_DIR)/Src \
-			-I$(CORE_DIR)/Src/zlib \
 			-I$(LIBRETRO_COMM_DIR)/include \
 			-I$(CORE_DIR)
 
@@ -30,22 +29,6 @@
 SOURCES_C += $(LIBRETRO_COMM_DIR)/libco/libco.c 
 endif
 
-SOURCES_C += \
-				 $(CORE_DIR)/Src/zlib/adler32.c \
-				 $(CORE_DIR)/Src/zlib/compress.c \
-				 $(CORE_DIR)/Src/zlib/crc32.c \
-				 $(CORE_DIR)/Src/zlib/deflate.c \
-				 $(CORE_DIR)/Src/zlib/gzclose.c \
-				 $(CORE_DIR)/Src/zlib/gzlib.c \
-				 $(CORE_DIR)/Src/zlib/gzread.c \
-				 $(CORE_DIR)/Src/zlib/gzwrite.c \
-				 $(CORE_DIR)/Src/zlib/inffast.c \
-				 $(CORE_DIR)/Src/zlib/inflate.c \
-				 $(CORE_DIR)/Src/zlib/inftrees.c \
-				 $(CORE_DIR)/Src/zlib/trees.c \
-				 $(CORE_DIR)/Src/zlib/uncompr.c \
-				 $(CORE_DIR)/Src/zlib/zutil.c
-
 ifeq ($(NOLIBCO), 1)
 	CPPFLAGS += -DNO_LIBCO
 endif
@@ -113,3 +96,5 @@
 					$(GUI_DIR)/dlgMain.cpp \
 					$(GUI_DIR)/dlgSnapshot.cpp \
 					$(GUI_DIR)/sdlgui.cpp
+
+LDFLAGS += -lz
