diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e98b7f..d4fb02c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -284,7 +284,7 @@ install(FILES "share/pixmaps/scap-workbench.png"
 install(FILES "share/pixmaps/scap-workbench.svg"
     DESTINATION "${CMAKE_INSTALL_DATADIR}/pixmaps")
 install(FILES "scap-workbench.appdata.xml"
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/appdata")
+    DESTINATION "${CMAKE_INSTALL_DATADIR}/metainfo")
 
 if (ASCIIDOC_EXECUTABLE)
     file(GLOB USER_MANUAL_SCREENSHOTS "${CMAKE_CURRENT_SOURCE_DIR}/doc/user_manual/*.png")
