API changes

--- a/src/RemediationRoleSaver.cpp
+++ b/src/RemediationRoleSaver.cpp
@@ -129,7 +129,7 @@ void ProfileBasedRemediationSaver::saveToFile(const QString& filename)
     outputFile.open(QIODevice::WriteOnly);
     struct xccdf_session* session = mScanningSession->getXCCDFSession();
     struct xccdf_policy* policy = xccdf_session_get_xccdf_policy(session);
-    const int result = xccdf_policy_generate_fix(policy, NULL, mTemplateString.toUtf8().constData(), outputFile.handle());
+    const int result = xccdf_policy_generate_fix(policy, NULL, mTemplateString.toUtf8().constData(), NULL, NULL, outputFile.handle(), 0);
     if (!outputFile.flush())
     {
         throw std::runtime_error("Could not write to the destination location.");
@@ -224,7 +224,7 @@ void ResultBasedLibraryRemediationSaver::saveToFile(const QString& filename)
     QFile outputFile(filename);
     outputFile.open(QIODevice::WriteOnly);
 
-    const int rc = xccdf_policy_generate_fix(policy, result, mTemplateString.toUtf8().constData(), outputFile.handle());
+    const int rc = xccdf_policy_generate_fix(policy, result, mTemplateString.toUtf8().constData(), NULL, NULL, outputFile.handle(), 0);
     if (!outputFile.flush())
     {
         throw std::runtime_error("Could not write to the destination location.");
