From e5863102bde02c729bf5bd176393b5ef583a2885 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Paku=C5=82a?= <tomasz.pakula.oficjalny@gmail.com>
Date: Mon, 16 Jun 2025 21:00:50 +0200
Subject: [PATCH 1/2] Remove redundant if

---
 transport/dongle.c | 15 ++++++---------
 1 file changed, 6 insertions(+), 9 deletions(-)

diff --git a/transport/dongle.c b/transport/dongle.c
index 1a24a54..0fc6824 100644
--- a/transport/dongle.c
+++ b/transport/dongle.c
@@ -887,15 +887,12 @@ static int xone_dongle_init(struct xone_dongle *dongle,
 	dev_dbg(dongle->mt.dev, "%s: trying to load firmware %s\n",
 			__func__, fwname);
 	err = request_firmware(&fw, fwname, mt->dev);
-	if (err) {
-		if (err == -ENOENT) {
-			snprintf(fwname, 25, "xow_dongle.bin",
-				 id->idVendor, id->idProduct);
-			dev_dbg(dongle->mt.dev, "%s: trying to load firmware %s\n",
-				__func__, fwname);
-			err = request_firmware(&fw, fwname, mt->dev);
-
-		}
+	if (err == -ENOENT) {
+		snprintf(fwname, 25, "xow_dongle.bin",
+				id->idVendor, id->idProduct);
+		dev_dbg(dongle->mt.dev, "%s: trying to load firmware %s\n",
+			__func__, fwname);
+		err = request_firmware(&fw, fwname, mt->dev);
 	}
 
 	if (err) {

From dba5d3425eafef1a48454c4fc48cd639164f3148 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tomasz=20Paku=C5=82a?= <tomasz.pakula.oficjalny@gmail.com>
Date: Mon, 16 Jun 2025 21:04:12 +0200
Subject: [PATCH 2/2] Fix warning: too many arguments for format

Seems like this was a copy-paste of snprintf form line 885.
---
 transport/dongle.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/transport/dongle.c b/transport/dongle.c
index 0fc6824..a98858d 100644
--- a/transport/dongle.c
+++ b/transport/dongle.c
@@ -888,8 +888,7 @@ static int xone_dongle_init(struct xone_dongle *dongle,
 			__func__, fwname);
 	err = request_firmware(&fw, fwname, mt->dev);
 	if (err == -ENOENT) {
-		snprintf(fwname, 25, "xow_dongle.bin",
-				id->idVendor, id->idProduct);
+		snprintf(fwname, 15, "xow_dongle.bin");
 		dev_dbg(dongle->mt.dev, "%s: trying to load firmware %s\n",
 			__func__, fwname);
 		err = request_firmware(&fw, fwname, mt->dev);
