diff --git a/src/common/gl/context.cpp b/src/common/gl/context.cpp
index a6d2c92..c681f0e 100644
--- a/src/common/gl/context.cpp
+++ b/src/common/gl/context.cpp
@@ -12,6 +12,8 @@
 #endif
 Log_SetChannel(GL::Context);
 
+#include <cstdio>
+
 #if defined(_WIN32) && !defined(_M_ARM64)
 #include "context_wgl.h"
 #elif defined(__APPLE__)
diff --git a/src/common/log.h b/src/common/log.h
index 49179a9..077e861 100644
--- a/src/common/log.h
+++ b/src/common/log.h
@@ -4,6 +4,7 @@
 #pragma once
 #include "types.h"
 #include <cinttypes>
+#include <stdarg.h>
 #include <mutex>
 
 enum LOGLEVEL
