diff --git a/src/common/file_system.cpp b/src/common/file_system.cpp
index 87cbbed..79cab7a 100644
--- a/src/common/file_system.cpp
+++ b/src/common/file_system.cpp
@@ -1480,8 +1480,8 @@ static u32 RecursiveFindFiles(const char* OriginPath, const char* ParentPath, co
       continue;
 
 #else
-    struct stat64 sDir;
-    if (stat64(full_path.c_str(), &sDir) < 0)
+    struct stat sDir;
+    if (stat(full_path.c_str(), &sDir) < 0)
       continue;
 #endif
 
@@ -1591,8 +1591,8 @@ bool FileSystem::StatFile(const char* path, FILESYSTEM_STAT_DATA* sd)
   struct stat sysStatData;
   if (stat(path, &sysStatData) < 0)
 #else
-  struct stat64 sysStatData;
-  if (stat64(path, &sysStatData) < 0)
+  struct stat sysStatData;
+  if (stat(path, &sysStatData) < 0)
 #endif
     return false;
 
@@ -1624,8 +1624,8 @@ bool FileSystem::StatFile(std::FILE* fp, FILESYSTEM_STAT_DATA* sd)
   struct stat sysStatData;
   if (fstat(fd, &sysStatData) < 0)
 #else
-  struct stat64 sysStatData;
-  if (fstat64(fd, &sysStatData) < 0)
+  struct stat sysStatData;
+  if (fstat(fd, &sysStatData) < 0)
 #endif
     return false;
 
@@ -1657,8 +1657,8 @@ bool FileSystem::FileExists(const char* path)
   struct stat sysStatData;
   if (stat(path, &sysStatData) < 0)
 #else
-  struct stat64 sysStatData;
-  if (stat64(path, &sysStatData) < 0)
+  struct stat sysStatData;
+  if (stat(path, &sysStatData) < 0)
 #endif
     return false;
 
@@ -1679,8 +1679,8 @@ bool FileSystem::DirectoryExists(const char* path)
   struct stat sysStatData;
   if (stat(path, &sysStatData) < 0)
 #else
-  struct stat64 sysStatData;
-  if (stat64(path, &sysStatData) < 0)
+  struct stat sysStatData;
+  if (stat(path, &sysStatData) < 0)
 #endif
     return false;
 
diff --git a/src/util/memory_arena.cpp b/src/util/memory_arena.cpp
index 2f5b2df..07612c9 100644
--- a/src/util/memory_arena.cpp
+++ b/src/util/memory_arena.cpp
@@ -166,9 +166,9 @@ bool MemoryArena::Create(size_t size, bool writable, bool executable)
   shm_unlink(file_mapping_name.c_str());
 
   // ensure it's the correct size
-  if (ftruncate64(m_shmem_fd, static_cast<off64_t>(size)) < 0)
+  if (ftruncate(m_shmem_fd, static_cast<off_t>(size)) < 0)
   {
-    Log_ErrorPrintf("ftruncate64(%zu) failed: %d", size, errno);
+    Log_ErrorPrintf("ftruncate(%zu) failed: %d", size, errno);
     return false;
   }
 
