diff --git a/dep/CMakeLists.txt b/dep/CMakeLists.txt
index b5dc473..5f28344 100644
--- a/dep/CMakeLists.txt
+++ b/dep/CMakeLists.txt
@@ -1,9 +1,9 @@
-add_subdirectory(fmt)
+# add_subdirectory(fmt)
 add_subdirectory(glad)
 add_subdirectory(stb)
-add_subdirectory(zlib)
-add_subdirectory(zstd)
-add_subdirectory(minizip)
+# add_subdirectory(zlib)
+# add_subdirectory(zstd)
+# add_subdirectory(minizip)
 add_subdirectory(lzma)
 add_subdirectory(libchdr)
 add_subdirectory(xxhash)
@@ -12,8 +12,8 @@ add_subdirectory(glslang)
 add_subdirectory(imgui)
 add_subdirectory(simpleini)
 add_subdirectory(vulkan)
-add_subdirectory(soundtouch)
-add_subdirectory(tinyxml2)
+# add_subdirectory(soundtouch)
+# add_subdirectory(tinyxml2)
 add_subdirectory(googletest)
 add_subdirectory(cpuinfo)
 
@@ -33,3 +33,11 @@ if(${CPU_ARCH} STREQUAL "aarch32" OR ${CPU_ARCH} STREQUAL "aarch64")
   add_subdirectory(vixl)
 endif()
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(zstd REQUIRED libzstd IMPORTED_TARGET GLOBAL)
+add_library(Zstd::Zstd ALIAS PkgConfig::zstd)
+pkg_check_modules(fmt REQUIRED fmt IMPORTED_TARGET GLOBAL)
+pkg_check_modules(minizip REQUIRED minizip IMPORTED_TARGET GLOBAL)
+pkg_check_modules(soundtouch REQUIRED soundtouch IMPORTED_TARGET GLOBAL)
+pkg_check_modules(tinyxml2 REQUIRED tinyxml2 IMPORTED_TARGET GLOBAL)
+pkg_check_modules(zlib REQUIRED zlib IMPORTED_TARGET GLOBAL)
diff --git a/dep/libchdr/CMakeLists.txt b/dep/libchdr/CMakeLists.txt
index 05b8c7a..9fe2b3e 100644
--- a/dep/libchdr/CMakeLists.txt
+++ b/dep/libchdr/CMakeLists.txt
@@ -15,5 +15,5 @@ add_library(libchdr
 )
 
 target_include_directories(libchdr PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
-target_link_libraries(libchdr PRIVATE zlib lzma)
+target_link_libraries(libchdr PRIVATE z lzma)
 
diff --git a/dep/stb/CMakeLists.txt b/dep/stb/CMakeLists.txt
index 8206b10..dda493e 100644
--- a/dep/stb/CMakeLists.txt
+++ b/dep/stb/CMakeLists.txt
@@ -10,4 +10,4 @@ set(SRCS
 add_library(stb ${SRCS})
 target_include_directories(stb PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/include" "${CMAKE_CURRENT_SOURCE_DIR}/src")
 target_include_directories(stb INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/include")
-target_link_libraries(stb zlib Threads::Threads "${CMAKE_DL_LIBS}")
+target_link_libraries(stb z Threads::Threads "${CMAKE_DL_LIBS}")
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 43cd717..efdd4e8 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -62,7 +62,7 @@ add_library(common
 target_include_directories(common PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/..")
 target_include_directories(common PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/..")
 target_link_libraries(common PUBLIC fmt Threads::Threads vulkan-headers)
-target_link_libraries(common PRIVATE stb libchdr zlib minizip Zstd::Zstd "${CMAKE_DL_LIBS}")
+target_link_libraries(common PRIVATE stb libchdr z minizip Zstd::Zstd "${CMAKE_DL_LIBS}")
 
 if(WIN32)
   target_sources(common PRIVATE
diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 29c4b22..48a2cfc 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -116,7 +116,7 @@ set(RECOMPILER_SRCS
 
 target_include_directories(core PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/..")
 target_include_directories(core PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/..")
-target_link_libraries(core PUBLIC Threads::Threads common util zlib)
+target_link_libraries(core PUBLIC Threads::Threads common util z)
 target_link_libraries(core PRIVATE stb xxhash imgui rapidjson tinyxml2)
 
 if(WIN32)
diff --git a/src/util/CMakeLists.txt b/src/util/CMakeLists.txt
index 81a33f7..719bd99 100644
--- a/src/util/CMakeLists.txt
+++ b/src/util/CMakeLists.txt
@@ -42,4 +42,4 @@ add_library(util
 target_include_directories(util PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}/..")
 target_include_directories(util PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/..")
 target_link_libraries(util PUBLIC common simpleini)
-target_link_libraries(util PRIVATE libchdr zlib soundtouch)
+target_link_libraries(util PRIVATE libchdr z SoundTouch)
