diff --git a/eolie.in b/eolie.in
index 837ba81..2a2d29a 100755
--- a/eolie.in
+++ b/eolie.in
@@ -34,15 +34,6 @@ def install_excepthook():
 if __name__ == "__main__":
     install_excepthook()
     
-    try:
-        locale.bindtextdomain('eolie', localedir)
-        locale.textdomain('eolie')
-    except AttributeError as e:
-        # Python built without gettext support doesn't have bindtextdomain()
-        # and textdomain()
-        print("Couldn't bind the gettext translation domain. Some translations"
-        " won't work.\n{}".format(e))
-
     gettext.bindtextdomain('eolie', localedir)
     gettext.textdomain('eolie')
 
