diff --git a/python-webextension/meson.build b/python-webextension/meson.build
index 06b7cbb..2a1c9b6 100644
--- a/python-webextension/meson.build
+++ b/python-webextension/meson.build
@@ -2,7 +2,7 @@ extension_dir = join_paths(get_option('libdir'), 'eolie', 'webkitextension')
 
 webkitextension = dependency('webkit2gtk-web-extension-4.0')
 pyobject = dependency('pygobject-3.0')
-python3 = dependency('python3')
+python3 = dependency('python3-embed')
 
 pythonloader = configure_file(
   input: 'pythonloader.c.in',
