From 96aa7d8650f4de9f1d009e10d45df01d5e7a1331 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Mon, 21 Feb 2011 10:11:48 -0600
Subject: [PATCH] health model: work with procps and busybox 1.18

---
 health-model.lua |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/health-model.lua b/health-model.lua
index 2749e2a..bb70073 100644
--- a/health-model.lua
+++ b/health-model.lua
@@ -49,8 +49,8 @@ get_system = function (self)
 	system.version = cfe({ value=indexver or fs.read_file("/etc/alpine-release") or "Unknown", label="Version" })
 	system.timezone = cfe({ value=date.what_tz(), label="Time Zone" })
 	system.uname = cfe({ value=querycmd("uname -a"), label="UName" })
-	system.memory = cfe({ value=querycmd("free"), label="Memory usage" })
-	system.memory.used = getpercentage(querycmd("free"), "Total:", 3, 4)
+	system.memory = cfe({ value=querycmd("free -o"), label="Memory usage" })
+	system.memory.used = getpercentage(querycmd("free -o"), "Total:", 3, 4)
 	return cfe({ type="group", value=system })
 end
 
-- 
1.7.4.1

