diff -ru e2fsprogs-1.42.8.orig/lib/ext2fs/unix_io.c e2fsprogs-1.42.8/lib/ext2fs/unix_io.c
--- e2fsprogs-1.42.8.orig/lib/ext2fs/unix_io.c	2013-09-27 17:26:55.203155268 +0300
+++ e2fsprogs-1.42.8/lib/ext2fs/unix_io.c	2013-09-27 17:27:27.782978767 +0300
@@ -23,6 +23,7 @@
 
 #include "config.h"
 #include <stdio.h>
+#include <stdint.h>
 #include <string.h>
 #if HAVE_UNISTD_H
 #include <unistd.h>
@@ -931,10 +932,10 @@
 
 	if (channel->flags & CHANNEL_FLAGS_BLOCK_DEVICE) {
 #ifdef BLKDISCARD
-		__uint64_t range[2];
+		uint64_t range[2];
 
-		range[0] = (__uint64_t)(block) * channel->block_size;
-		range[1] = (__uint64_t)(count) * channel->block_size;
+		range[0] = (uint64_t)(block) * channel->block_size;
+		range[1] = (uint64_t)(count) * channel->block_size;
 
 		ret = ioctl(data->dev, BLKDISCARD, &range);
 #else
