From 584fd01d78e94d98ea2cb21954789fcc844fabcb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Hor=C3=A1k?= <dan@danny.cz>
Date: Fri, 22 Apr 2016 13:21:50 +0200
Subject: [PATCH] zipl: disable strict aliasing for bootloader

---
 zipl/boot/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/zipl/boot/Makefile b/zipl/boot/Makefile
index bb8a19d..f1f4401 100644
--- a/zipl/boot/Makefile
+++ b/zipl/boot/Makefile
@@ -4,7 +4,7 @@ include ../../common.mak
 CFLAGS_BOOT = -Os -g -I../include -D__ASSEMBLY__ \
 	  -DS390_TOOLS_RELEASE=$(S390_TOOLS_RELEASE) \
 	  -fno-builtin -ffreestanding -fno-asynchronous-unwind-tables \
-	  -fno-delete-null-pointer-checks \
+	  -fno-delete-null-pointer-checks -fno-strict-aliasing \
 	  -fexec-charset=IBM1047 -m64 -mpacked-stack \
 	  -mstack-size=8192 -mstack-guard=128 -msoft-float $(WARNFLAGS)
 
-- 
2.7.4

