--- a/vmur/vmur.c
+++ b/vmur/vmur.c
@@ -1503,7 +1503,7 @@
 				     struct splink_page *buf)
 {
 	struct splink_record *rec;
-	char netdata_id[5] = {0xc9, 0xd5, 0xd4, 0xd9, 0xf0}; /* EBCDIC: INMR0 */
+	unsigned char netdata_id[5] = {0xc9, 0xd5, 0xd4, 0xd9, 0xf0}; /* EBCDIC: INMR0 */
 	unsigned int i;
 
 	rec = (struct splink_record *) &buf->data;
--- a/include/lib/util_sys.h
+++ b/include/lib/util_sys.h
@@ -14,6 +14,10 @@
 
 #include <stdbool.h>
 
+#ifndef	__dev_t_defined
+#include <sys/types.h>
+#endif /* __dev_t_defined */
+
 int util_sys_get_dev_addr(const char *dev, char *addr);
 bool util_sys_dev_is_partition(dev_t dev);
 int util_sys_get_partnum(dev_t dev);
--- a/include/lib/vtoc.h
+++ b/include/lib/vtoc.h
@@ -16,6 +16,7 @@
 #include <string.h>
 #include <sys/ioctl.h>
 #include <sys/stat.h>
+#include <sys/types.h>
 #include <time.h>
 #include <unistd.h>
 
--- a/zkey/keystore.c
+++ b/zkey/keystore.c
@@ -23,6 +23,7 @@
 #include <sys/file.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <fcntl.h>
 
 #include "lib/util_base.h"
 #include "lib/util_file.h"
--- a/include/boot/s390.h
+++ b/include/boot/s390.h
@@ -29,6 +29,7 @@
 #define UNPACK_FACILITY		_AC(161, U)
 
 #ifndef __ASSEMBLER__
+#include <asm/types.h>
 #include <stdint.h>
 
 /*
