diff --git a/fdasd/fdasd.c b/fdasd/fdasd.c
index fb28b1e..218dca3 100644
--- a/fdasd/fdasd.c
+++ b/fdasd/fdasd.c
@@ -9,6 +9,7 @@
 
 #include <getopt.h>
 #include <stdio.h>
+#include <sys/sysmacros.h>
 
 #include "lib/dasd_sys.h"
 #include "lib/util_opt.h"
diff --git a/tape390/tape390_common.c b/tape390/tape390_common.c
index 90954c0..ffe72bf 100644
--- a/tape390/tape390_common.c
+++ b/tape390/tape390_common.c
@@ -13,6 +13,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #include "tape390_common.h"
 
