diff --git a/mon_tools/mon_procd.c b/mon_tools/mon_procd.c
index 9b52057..d7dbee3 100644
--- a/mon_tools/mon_procd.c
+++ b/mon_tools/mon_procd.c
@@ -8,6 +8,7 @@
  */
 #include <unistd.h>
 #include <sys/vfs.h>
+#include <sys/time.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
diff --git a/ziomon/ziomon_msg_tools.h b/ziomon/ziomon_msg_tools.h
index 383cdd4..1aa16ff 100644
--- a/ziomon/ziomon_msg_tools.h
+++ b/ziomon/ziomon_msg_tools.h
@@ -10,6 +10,7 @@
 #ifndef ZIOMON_MSG__TOOLS_H
 #define ZIOMON_MSG__TOOLS_H
 
+#include <time.h>
 #include "ziomon_dacc.h"
 
 void conv_blkiomon_v2_to_v3(struct message *msg);
