diff --git a/zipl/boot/Makefile b/zipl/boot/Makefile
index 83771f1..f221dbf 100644
--- a/zipl/boot/Makefile
+++ b/zipl/boot/Makefile
@@ -1,6 +1,8 @@
 # Common definitions
 include ../../common.mak
 
+ALL_CPPFLAGS := $(filter-out -Os,$(ALL_CPPFLAGS))
+ALL_CFLAGS := $(filter-out -Os,$(ALL_CFLAGS))
 CFLAGS_BOOT = -fno-pie -Os -g -I../include -D__ASSEMBLY__ \
 	  -DS390_TOOLS_RELEASE=$(S390_TOOLS_RELEASE) \
 	  -fno-builtin -ffreestanding -fno-asynchronous-unwind-tables \
diff --git a/zipl/src/Makefile b/zipl/src/Makefile
index c526a46..24ec04f 100644
--- a/zipl/src/Makefile
+++ b/zipl/src/Makefile
@@ -8,6 +8,8 @@ ALL_CPPFLAGS += -I../include -I../boot -I../../include \
 	    -DZFCPDUMP_PART_IMAGE=$(ZFCPDUMP_PART_IMAGE) \
 	    -DZFCPDUMP_PART_RD=$(ZFCPDUMP_PART_RD) \
 	    -D_FILE_OFFSET_BITS=64 -fno-pie
+ALL_CPPFLAGS := $(filter-out -Os,$(ALL_CPPFLAGS))
+ALL_CFLAGS := $(filter-out -Os,$(ALL_CFLAGS))
 ALL_LDFLAGS += -Wl,-z,noexecstack -no-pie
 
 libs =	$(rootdir)/libutil/libutil.a \
