--- a/common.mak
+++ b/common.mak
@@ -168,7 +168,7 @@ USRSBINDIR      = $(INSTALLDIR)/usr/sbin
 USRBINDIR       = $(INSTALLDIR)/usr/bin
 BINDIR          = $(INSTALLDIR)/sbin
 LIBDIR          = $(INSTALLDIR)/lib
-USRLIB64DIR     = $(INSTALLDIR)/usr/lib64
+USRLIBDIR       = $(INSTALLDIR)/usr/lib
 SYSCONFDIR      = $(INSTALLDIR)/etc
 MANDIR          = $(INSTALLDIR)/usr/share/man
 VARDIR          = $(INSTALLDIR)/var
@@ -179,10 +179,10 @@ ZFCPDUMP_DIR    = $(TOOLS_LIBDIR)/zfcpdu
 # for SYSTEMDSYSTEMUNITDIR (e.g. /lib/systemd/system)
 SYSTEMDSYSTEMUNITDIR =
 USRINCLUDEDIR   = $(INSTALLDIR)/usr/include
-ZKEYKMSPLUGINDIR = $(USRLIB64DIR)/zkey
+ZKEYKMSPLUGINDIR = $(USRLIBDIR)/zkey
 
 ifeq ($(LIBDIR),$(INSTALLDIR)/lib)
-SOINSTALLDIR = $(USRLIB64DIR)
+SOINSTALLDIR = $(USRLIBDIR)
 else
 SOINSTALLDIR = $(LIBDIR)
 endif
@@ -191,11 +191,11 @@ INSTDIRS        = $(USRSBINDIR) $(USRBIN
 			$(SYSCONFDIR) $(SYSCONFDIR)/sysconfig \
 			$(TOOLS_LIBDIR) $(TOOLS_DATADIR) \
 			$(ZFCPDUMP_DIR) $(SYSTEMDSYSTEMUNITDIR) \
-			$(USRLIB64DIR) $(USRINCLUDEDIR) $(ZKEYKMSPLUGINDIR) \
+			$(USRLIBDIR) $(USRINCLUDEDIR) $(ZKEYKMSPLUGINDIR) \
 			$(SOINSTALLDIR)
 OWNER           = $(shell id -un)
 GROUP		= $(shell id -gn)
-export INSTALLDIR BINDIR LIBDIR USRLIB64DIR MANDIR OWNER GROUP
+export INSTALLDIR BINDIR LIBDIR USRLIBDIR MANDIR OWNER GROUP
 
 # Special defines for zfcpdump
 ZFCPDUMP_IMAGE	= zfcpdump-image
--- a/genprotimg/src/Makefile
+++ b/genprotimg/src/Makefile
@@ -43,7 +43,7 @@ LIBCRYPTO_LIBS := $(shell pkg-config --s
 LIBCURL_CFLAGS := $(shell pkg-config --silence-errors --cflags libcurl)
 LIBCURL_LIBS := $(shell pkg-config --silence-errors --libs libcurl)
 else
-GLIB2_CFLAGS := -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include
+GLIB2_CFLAGS := -I/usr/include/glib-2.0
 GLIB2_LIBS := -lglib-2.0
 LIBCRYPTO_CFLAGS :=
 LIBCRYPTO_LIBS := -lcrypto
