diff --git a/common.mak b/common.mak
index 78c678e..ba7cbb4 100644
--- a/common.mak
+++ b/common.mak
@@ -181,7 +181,7 @@ USRBINDIR       = $(INSTALLDIR)/usr/bin
 BINDIR          = $(INSTALLDIR)/sbin
 LIBDIR          = $(INSTALLDIR)/lib
 USRLIBDIR	= $(INSTALLDIR)/usr/lib
-USRLIB64DIR     = $(INSTALLDIR)/usr/lib64
+USRLIB64DIR     = $(INSTALLDIR)/usr/lib
 SYSCONFDIR      = $(INSTALLDIR)/etc
 MANDIR          = $(INSTALLDIR)/usr/share/man
 VARDIR          = $(INSTALLDIR)/var
