--- a/zkey/kms.c
+++ b/zkey/kms.c
@@ -40,7 +40,7 @@
 
 #define ENVVAR_ZKEY_KMS_PLUGINS		"ZKEY_KMS_PLUGINS"
 #define DEFAULT_KMS_PLUGINS		"/etc/zkey/kms-plugins.conf"
-#define KMS_PLUGIN_LOCATION		"/usr/lib64/zkey"
+#define KMS_PLUGIN_LOCATION		"/usr/lib/zkey"
 
 #define KMS_CONFIG_FILE			"kms.conf"
 #define KMS_CONFIG_PROP_KMS		"kms"
--- a/libekmfweb/Makefile
+++ b/libekmfweb/Makefile
@@ -77,9 +77,9 @@
 	ln -sf libekmfweb.so.$(VERSION) libekmfweb.so
 
 install-libekmfweb.so.$(VERSION): libekmfweb.so.$(VERSION)
-	$(INSTALL) -g $(GROUP) -o $(OWNER) -m 755 libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIB64DIR)/libekmfweb.so.$(VERSION)
-	ln -sf $(DESTDIR)$(USRLIB64DIR)/libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIB64DIR)/libekmfweb.so.$(VERM)
-	ln -sf $(DESTDIR)$(USRLIB64DIR)/libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIB64DIR)/libekmfweb.so
+	$(INSTALL) -g $(GROUP) -o $(OWNER) -m 755 libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIBDIR)/libekmfweb.so.$(VERSION)
+	ln -sf $(DESTDIR)$(USRLIBDIR)/libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIBDIR)/libekmfweb.so.$(VERM)
+	ln -sf $(DESTDIR)$(USRLIBDIR)/libekmfweb.so.$(VERSION) $(DESTDIR)$(USRLIBDIR)/libekmfweb.so
 	$(INSTALL) -d -m 770 $(DESTDIR)$(USRINCLUDEDIR)/ekmfweb
 	$(INSTALL) -g $(GROUP) -o $(OWNER) -m 755 $(rootdir)include/ekmfweb/ekmfweb.h $(DESTDIR)$(USRINCLUDEDIR)/ekmfweb
 
--- a/zkey/ekmfweb/Makefile
+++ b/zkey/ekmfweb/Makefile
@@ -46,9 +46,9 @@
 install: all install-libekmfweb.dep zkey-ekmfweb.so
 	$(INSTALL) -d -m 755 $(DESTDIR)$(MANDIR)/man1
 	$(INSTALL) -m 644 -c zkey-ekmfweb.1 $(DESTDIR)$(MANDIR)/man1
-	$(INSTALL) -d -m 755 $(DESTDIR)$(USRLIB64DIR)
-	$(INSTALL) -d -m 755 $(DESTDIR)$(USRLIB64DIR)/zkey
-	$(INSTALL) -g $(GROUP) -o $(OWNER) -m 755 zkey-ekmfweb.so $(DESTDIR)$(USRLIB64DIR)/zkey/zkey-ekmfweb.so
+	$(INSTALL) -d -m 755 $(DESTDIR)$(USRLIBDIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(USRLIBDIR)/zkey
+	$(INSTALL) -g $(GROUP) -o $(OWNER) -m 755 zkey-ekmfweb.so $(DESTDIR)$(USRLIBDIR)/zkey/zkey-ekmfweb.so
 
 clean:
 	rm -f *.o zkey-ekmfweb.so install-libekmfweb.dep libekmfweb.dep
--- a/common.mak
+++ b/common.mak
@@ -163,7 +163,7 @@
 USRBINDIR       = $(INSTALLDIR)/usr/bin
 BINDIR          = $(INSTALLDIR)/sbin
 LIBDIR          = $(INSTALLDIR)/lib
-USRLIB64DIR     = $(INSTALLDIR)/usr/lib64
+USRLIBDIR       = $(INSTALLDIR)/usr/lib
 SYSCONFDIR      = $(INSTALLDIR)/etc
 MANDIR          = $(INSTALLDIR)/usr/share/man
 VARDIR          = $(INSTALLDIR)/var
@@ -179,10 +179,10 @@
 			$(SYSCONFDIR) $(SYSCONFDIR)/sysconfig \
 			$(TOOLS_LIBDIR) $(TOOLS_DATADIR) \
 			$(ZFCPDUMP_DIR) $(SYSTEMDSYSTEMUNITDIR) \
-			$(USRLIB64DIR) $(USRINCLUDEDIR)
+			$(USRLIBDIR) $(USRINCLUDEDIR)
 OWNER           = $(shell id -un)
 GROUP		= $(shell id -gn)
-export INSTALLDIR BINDIR LIBDIR USRLIB64DIR MANDIR OWNER GROUP
+export INSTALLDIR BINDIR LIBDIR USRLIBDIR MANDIR OWNER GROUP
 
 # Special defines for zfcpdump
 ZFCPDUMP_IMAGE	= zfcpdump-image
