diff --git a/common.mak b/common.mak
index 78c678e..ba7cbb4 100644
--- a/common.mak
+++ b/common.mak
@@ -181,7 +181,7 @@ USRBINDIR       = $(INSTALLDIR)/usr/bin
 BINDIR          = $(INSTALLDIR)/sbin
 LIBDIR          = $(INSTALLDIR)/lib
 USRLIBDIR	= $(INSTALLDIR)/usr/lib
-USRLIB64DIR     = $(INSTALLDIR)/usr/lib64
+USRLIB64DIR     = $(INSTALLDIR)/usr/lib
 SYSCONFDIR      = $(INSTALLDIR)/etc
 MANDIR          = $(INSTALLDIR)/usr/share/man
 VARDIR          = $(INSTALLDIR)/var
--- a/genprotimg/src/Makefile
+++ b/genprotimg/src/Makefile
@@ -43,7 +43,7 @@ LIBCRYPTO_LIBS := $(shell pkg-config --s
 LIBCURL_CFLAGS := $(shell pkg-config --silence-errors --cflags libcurl)
 LIBCURL_LIBS := $(shell pkg-config --silence-errors --libs libcurl)
 else
-GLIB2_CFLAGS := -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include
+GLIB2_CFLAGS := -I/usr/include/glib-2.0
 GLIB2_LIBS := -lglib-2.0
 LIBCRYPTO_CFLAGS :=
 LIBCRYPTO_LIBS := -lcrypto
