diff --git a/libutil/util_panic.c b/libutil/util_panic.c
index 59a9a9e..cc21e3d 100644
--- a/libutil/util_panic.c
+++ b/libutil/util_panic.c
@@ -9,7 +9,6 @@
  * it under the terms of the MIT license. See LICENSE for details.
  */
 
-#include <execinfo.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -26,6 +25,7 @@
  */
 static void print_backtrace(void)
 {
+	#if 0
 	void *array[256];
 	size_t i, size;
 	char **strings;
@@ -41,6 +41,7 @@ static void print_backtrace(void)
 		fprintf(stderr, "  %s\n", strings[i]);
 
 	free(strings);
+	#endif
 }
 
 /*
diff --git a/zdev/src/misc.c b/zdev/src/misc.c
index 2f5cbc2..08a2524 100644
--- a/zdev/src/misc.c
+++ b/zdev/src/misc.c
@@ -10,7 +10,6 @@
 #include <ctype.h>
 #include <dirent.h>
 #include <errno.h>
-#include <execinfo.h>
 #include <limits.h>
 #include <signal.h>
 #include <stdarg.h>
@@ -1258,6 +1257,7 @@ void line_free(int argc, char **argv)
 /* Used for debugging. */
 void print_trace(void)
 {
+	#if 0
 	void *bt[BACKTRACE_MAX];
 	char **bt_sym;
 	int num, i;
@@ -1268,6 +1268,7 @@ void print_trace(void)
 	for (i = 0; i < num; i++)
 		fprintf(stderr, "DEBUG:  %s\n", bt_sym[i]);
 	free(bt_sym);
+	#endif
 }
 
 /* Return a textual representation of @config. */
