diff --git a/zipl/src/Makefile b/zipl/src/Makefile
index 786bb7f..2a1e10a 100644
--- a/zipl/src/Makefile
+++ b/zipl/src/Makefile
@@ -4,7 +4,7 @@ include ../../common.mak
 ALL_CPPFLAGS += -I../include -I../boot \
 	    -DZFCPDUMP_IMAGE="STRINGIFY($(ZFCPDUMP_DIR)/$(ZFCPDUMP_IMAGE))" \
 	    -DZFCPDUMP_INITRD="STRINGIFY($(ZFCPDUMP_DIR)/$(ZFCPDUMP_INITRD))" \
-	    -D_FILE_OFFSET_BITS=64 $(NO_PIE_CFLAGS)
+	    -D_FILE_OFFSET_BITS=64 -U_FORTIFY_SOURCE $(NO_PIE_CFLAGS)
 ALL_LDFLAGS += -Wl,-z,noexecstack $(NO_PIE_LDFLAGS)
 
 libs = $(rootdir)/libutil/libutil.a
