--- ./src/ngircd/irc-login.c.orig	2013-02-08 09:59:13.313191929 +0000
+++ ./src/ngircd/irc-login.c	2013-02-08 09:59:53.153525988 +0000
@@ -444,7 +444,7 @@
 		ptr = Req->argv[0];
 		while (*ptr) {
 			if (!isalnum((int)*ptr) &&
-			    *ptr != '+' && *ptr != '-' &&
+			    *ptr != '+' && *ptr != '-' && *ptr != '@' &&
 			    *ptr != '.' && *ptr != '_') {
 				Conn_Close(Client_Conn(Client), NULL,
 					   "Invalid user name", true);
