this can have a stale nonsense and fail the assert
--
diff --git a/tests/test_dict.py b/tests/test_dict.py
index 24ac44f..410072d 100644
--- a/tests/test_dict.py
+++ b/tests/test_dict.py
@@ -92,6 +92,7 @@ def test_session(en_us_dict):
 def test_add_remove(en_us_dict):
     """Test adding/removing from default user dictionary."""
     nonsense = "kxhjsddsi"
+    en_us_dict.remove(nonsense)
     assert not en_us_dict.check(nonsense)
     en_us_dict.add(nonsense)
     assert en_us_dict.is_added(nonsense)
