--- catimg-2.7.0-origin/src/catimg.c
+++ catimg-2.7.0/src/catimg.c
@@ -3,6 +3,8 @@
 #include <string.h>
 #include "sh_image.h"
 #include "sh_utils.h"
+// Enable usleep()
+#define _GNU_SOURCE
 #include <unistd.h>
 #include <signal.h>
 #include <sys/ioctl.h>
--- catimg-2.7.0-origin/src/sh_utils.c
+++ catimg-2.7.0/src/sh_utils.c
@@ -11,6 +11,9 @@
 #include <unistd.h>
 #endif
 
+// Enable fileno()
+#define _POSIX_C_SOURCE
+
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
