diff --git a/papershell/CMakeLists.txt b/papershell/CMakeLists.txt
index 0b84494..fc4cdf5 100644
--- a/papershell/CMakeLists.txt
+++ b/papershell/CMakeLists.txt
@@ -25,6 +25,7 @@ set( CMAKE_AUTOUIC ON )
 
 find_package( Qt5Core REQUIRED )
 find_package( Qt5Gui REQUIRED )
+find_package( Qt5Network REQUIRED )
 find_package( Qt5Widgets REQUIRED )
 find_package( Qt5DBus REQUIRED )
 find_package( dbusmenu-qt5 REQUIRED )
@@ -121,7 +122,7 @@ set ( RESOURCES
 add_executable( papershell ${SOURCES} ${HEADERS} ${UIS} ${RESOURCES}  )
 target_link_libraries(
     papershell
-    Qt5::Core Qt5::Gui Qt5::Widgets Qt5::DBus           # Qt5 Libs
+    Qt5::Core Qt5::Gui Qt5::Widgets Qt5::DBus Qt5::Network           # Qt5 Libs
     dbusmenu-qt5                                        # DBusMenu Qt5 for tray menu
     cprime-widgets cprime-gui cprime-core               # CuboCore Libs
     paperdegui paperdecore paperdewl                    # PaperDE Libs
diff --git a/sessionmanager/CMakeLists.txt b/sessionmanager/CMakeLists.txt
index 0e981e4..ca4daa5 100644
--- a/sessionmanager/CMakeLists.txt
+++ b/sessionmanager/CMakeLists.txt
@@ -25,6 +25,7 @@ set( CMAKE_AUTOUIC ON )
 
 find_package( Qt5Core REQUIRED )
 find_package( Qt5DBus REQUIRED )
+find_package( Qt5Network REQUIRED )
 
 set( HEADERS
 	sessionmgr.h
@@ -38,7 +39,7 @@ set( SOURCES
 )
 
 add_executable( papersessionmanager ${SOURCES} ${HEADERS}  )
-target_link_libraries( papersessionmanager Qt5::Core Qt5::DBus cprime-core csys paperdecore )
+target_link_libraries( papersessionmanager Qt5::Core Qt5::DBus Qt5::Network cprime-core csys paperdecore )
 
 install( TARGETS papersessionmanager DESTINATION bin )
 install( FILES paperdesktop.desktop DESTINATION share/wayland-sessions/ )
diff --git a/settings/CMakeLists.txt b/settings/CMakeLists.txt
index 16e567b..0c1627b 100644
--- a/settings/CMakeLists.txt
+++ b/settings/CMakeLists.txt
@@ -25,6 +25,7 @@ set( CMAKE_AUTOUIC ON )
 
 find_package( Qt5Core REQUIRED )
 find_package( Qt5Gui REQUIRED )
+find_package( Qt5Network REQUIRED )
 find_package( Qt5Widgets REQUIRED )
 find_package( Qt5UiTools REQUIRED )
 
@@ -50,7 +51,7 @@ set ( RESOURCES
 add_executable( papersettings ${SOURCES} ${HEADERS} ${UIS} ${RESOURCES}  )
 target_link_libraries(
     papersettings
-    Qt5::Core Qt5::Gui Qt5::Widgets Qt5::UiTools
+    Qt5::Core Qt5::Gui Qt5::Widgets Qt5::UiTools Qt5::Network
     cprime-widgets cprime-core
     paperdegui paperdecore paperdewl
 )
