--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2262,7 +2262,6 @@
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "0c10584274047cb335c23d3e61bcef8e323adae7c5c8c760540f73610177fc3f"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -181,7 +181,7 @@
 nu-system = { path = "./crates/nu-system", version = "0.95.0" }
 nu-utils = { path = "./crates/nu-utils", version = "0.95.0" }
 
-reedline = { workspace = true, features = ["bashisms", "sqlite"] }
+reedline = { workspace = true, features = ["bashisms", "sqlite-dynlib"] }
 
 crossterm = { workspace = true }
 ctrlc = { workspace = true }
--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -25,7 +25,7 @@
 nu-utils = { path = "../nu-utils", version = "0.95.0" }
 nu-color-config = { path = "../nu-color-config", version = "0.95.0" }
 nu-ansi-term = { workspace = true }
-reedline = { workspace = true, features = ["bashisms", "sqlite"] }
+reedline = { workspace = true, features = ["bashisms", "sqlite-dynlib"] }
 
 chrono = { default-features = false, features = ["std"], workspace = true }
 crossterm = { workspace = true }
--- a/crates/nu-command/Cargo.toml
+++ b/crates/nu-command/Cargo.toml
@@ -72,7 +72,7 @@
 rayon = { workspace = true }
 regex = { workspace = true }
 roxmltree = { workspace = true }
-rusqlite = { workspace = true, features = ["bundled", "backup", "chrono"], optional = true }
+rusqlite = { workspace = true, features = ["backup", "chrono"], optional = true }
 rmp = { workspace = true }
 same-file = { workspace = true }
 serde = { workspace = true, features = ["derive"] }
