--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2465,7 +2465,6 @@ version = "0.28.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "0c10584274047cb335c23d3e61bcef8e323adae7c5c8c760540f73610177fc3f"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -205,7 +205,7 @@ nu-protocol = { path = "./crates/nu-protocol", version = "0.98.0" }
 nu-std = { path = "./crates/nu-std", version = "0.98.0" }
 nu-system = { path = "./crates/nu-system", version = "0.98.0" }
 nu-utils = { path = "./crates/nu-utils", version = "0.98.0" }
-reedline = { workspace = true, features = ["bashisms", "sqlite"] }
+reedline = { workspace = true, features = ["bashisms", "sqlite-dynlib"] }

 crossterm = { workspace = true }
 ctrlc = { workspace = true }
--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -27,7 +27,7 @@ nu-protocol = { path = "../nu-protocol", version = "0.98.0" }
 nu-utils = { path = "../nu-utils", version = "0.98.0" }
 nu-color-config = { path = "../nu-color-config", version = "0.98.0" }
 nu-ansi-term = { workspace = true }
-reedline = { workspace = true, features = ["bashisms", "sqlite"] }
+reedline = { workspace = true, features = ["bashisms", "sqlite-dynlib"] }

 chrono = { default-features = false, features = ["std"], workspace = true }
 crossterm = { workspace = true }
--- a/crates/nu-command/Cargo.toml
+++ b/crates/nu-command/Cargo.toml
@@ -77,7 +77,7 @@ rand = { workspace = true }
 rayon = { workspace = true }
 regex = { workspace = true }
 roxmltree = { workspace = true }
-rusqlite = { workspace = true, features = ["bundled", "backup", "chrono"], optional = true }
+rusqlite = { workspace = true, features = ["backup", "chrono"], optional = true }
 rmp = { workspace = true }
 serde = { workspace = true, features = ["derive"] }
 serde_json = { workspace = true, features = ["preserve_order"] }
