diff -up kbd-1.15.3/configure.ac.orig kbd-1.15.3/configure.ac
--- kbd-1.15.3/configure.ac.orig	2011-08-22 16:01:11.456165128 +0200
+++ kbd-1.15.3/configure.ac	2011-08-22 16:01:38.878163675 +0200
@@ -60,6 +60,7 @@ AM_CONDITIONAL(KEYCODES_PROGS, test "$KE
 
 case $host_cpu in
 	i?86*) RESIZECONS_PROGS=yes ;;
+	x86_64*) RESIZECONS_PROGS=yes ;;
 	*)     RESIZECONS_PROGS=no ;;
 esac
 AM_CONDITIONAL(RESIZECONS_PROGS, test "$RESIZECONS_PROGS" = "yes")
