From 3a62a43ef758ec4d973032a9aaeb6ed8d1be630e Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Tue, 8 Oct 2019 19:33:19 +0300
Subject: [PATCH] build: workaround for meson disabler object not working with
 if not

---
 examples/meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/examples/meson.build b/examples/meson.build
index 8a60e1edd..869384c29 100644
--- a/examples/meson.build
+++ b/examples/meson.build
@@ -20,7 +20,8 @@ foreach dep : ['libpng', 'libavutil', 'libavcodec', 'libavformat']
 	endif
 endforeach
 
-if not cc.has_header('libavutil/hwcontext_drm.h', dependencies: libavutil)
+# Check if libavutil is found because of https://github.com/mesonbuild/meson/issues/6010
+if libavutil.found() and not cc.has_header('libavutil/hwcontext_drm.h', dependencies: libavutil)
 	libavutil = disabler()
 endif
 
