Description: change default ctl file
Author: Victor Seva <linuxmaniac@torreviejawireless.org>

--- a/etc/kamailio-basic.cfg
+++ b/etc/kamailio-basic.cfg
@@ -193,6 +193,8 @@
 # ----- mi_fifo params -----
 modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
--- a/etc/kamailio-oob.cfg
+++ b/etc/kamailio-oob.cfg
@@ -317,6 +317,8 @@
 # ----- mi_fifo params -----
 modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
--- a/etc/kamailio.cfg
+++ b/etc/kamailio.cfg
@@ -290,6 +290,8 @@
 # ----- mi_fifo params -----
 modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
--- a/modules/ctl/ctl_defaults.h
+++ b/modules/ctl/ctl_defaults.h
@@ -6,7 +6,7 @@
 /*listen by default on: */
 #ifdef SRNAME
 /* this is used when compiling sercmd tool */
-#define DEFAULT_CTL_SOCKET  "unixs:/tmp/" SRNAME "_ctl"
+#define DEFAULT_CTL_SOCKET  "unixs:/var/run/" SRNAME "/" SRNAME "_ctl"
 #else
 /* this is used when compiling sip server */
 #define DEFAULT_CTL_SOCKET  "unixs:/tmp/" NAME "_ctl"
