--- kamailio-4.2.3.orig/etc/kamailio-basic.cfg
+++ kamailio-4.2.3/etc/kamailio-basic.cfg
@@ -191,8 +191,10 @@
 
 
 # ----- mi_fifo params -----
-modparam("mi_fifo", "fifo_name", "/tmp/kamailio_fifo")
+modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
Only in kamailio-4.2.3: etc/kamailio-basic.cfg.orig
--- kamailio-4.2.3.orig/etc/kamailio-oob.cfg
+++ kamailio-4.2.3/etc/kamailio-oob.cfg
@@ -315,8 +315,10 @@
 
 
 # ----- mi_fifo params -----
-modparam("mi_fifo", "fifo_name", "/tmp/kamailio_fifo")
+modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
Only in kamailio-4.2.3: etc/kamailio-oob.cfg.orig
--- kamailio-4.2.3.orig/etc/kamailio.cfg
+++ kamailio-4.2.3/etc/kamailio.cfg
@@ -288,8 +288,10 @@
 
 
 # ----- mi_fifo params -----
-modparam("mi_fifo", "fifo_name", "/tmp/kamailio_fifo")
+modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 
+# ----- ctl params -----
+modparam("ctl", "binrpc", "unix:/var/run/kamailio/kamailio_ctl")
 
 # ----- tm params -----
 # auto-discard branches from previous serial forking leg
Only in kamailio-4.2.3: etc/kamailio.cfg.orig
--- kamailio-4.2.3.orig/modules/ctl/ctl_defaults.h
+++ kamailio-4.2.3/modules/ctl/ctl_defaults.h
@@ -6,10 +6,10 @@
 /*listen by default on: */
 #ifdef SRNAME
 /* this is used when compiling sercmd tool */
-#define DEFAULT_CTL_SOCKET  "unixs:/tmp/" SRNAME "_ctl"
+#define DEFAULT_CTL_SOCKET  "unixs:/var/run/" SRNAME "/" SRNAME "_ctl"
 #else
 /* this is used when compiling sip server */
-#define DEFAULT_CTL_SOCKET  "unixs:/tmp/" NAME "_ctl"
+#define DEFAULT_CTL_SOCKET  "unixs:/var/run/" NAME "/" NAME "_ctl"
 #endif
 /* port used by default for tcp/udp if no port is explicitely specified */
 #define DEFAULT_CTL_PORT 2049
--- kamailio-4.2.3.orig/modules/mi_fifo/README
+++ kamailio-4.2.3/modules/mi_fifo/README
@@ -117,7 +117,7 @@
 
    Example 1.1. Set fifo_name parameter
 ...
-modparam("mi_fifo", "fifo_name", "/tmp/kamailio_fifo")
+modparam("mi_fifo", "fifo_name", "/var/run/kamailio/kamailio_fifo")
 ...
 
 4.2. fifo_mode (integer)
@@ -160,7 +160,7 @@
 
    Directory to be used for creating the reply FIFO files.
 
-   Default value is "/tmp/"
+   Default value is "/var/run/kamailio/"
 
    Example 1.5. Set reply_dir parameter
 ...
--- kamailio-4.2.3.orig/modules/mi_fifo/mi_fifo.h
+++ kamailio-4.2.3/modules/mi_fifo/mi_fifo.h
@@ -34,7 +34,7 @@
 #ifndef _MI_FIFO_H_
 #define _MI_FIFO_H_
 
-#define DEFAULT_MI_REPLY_DIR "/tmp/"
+#define DEFAULT_MI_REPLY_DIR "/var/run/kamailio/"
 
 #define DEFAULT_MI_REPLY_IDENT "\t"
 
