diff --git a/utils/kamctl/kamctl.db_berkeley b/utils/kamctl/kamctl.db_berkeley
index c8b77ec..76faebb 100644
--- a/utils/kamctl/kamctl.db_berkeley
+++ b/utils/kamctl/kamctl.db_berkeley
@@ -6,7 +6,7 @@
 
 # path to the db_berkeley directory
 if [ -z "$DB_PATH" ] ; then
-	DB_PATH="/usr/local/share/kamailio/db_berkeley/kamailio"
+	DB_PATH="/usr/share/kamailio/db_berkeley/kamailio"
 fi
 
 #===================================================================
diff --git a/utils/kamctl/kamctl.dbtext b/utils/kamctl/kamctl.dbtext
index 0e5f9c4..b4c2c01 100644
--- a/utils/kamctl/kamctl.dbtext
+++ b/utils/kamctl/kamctl.dbtext
@@ -20,7 +20,7 @@ fi;
 # path to the dbtext database
 if [ -z "$DBTEXT_PATH" ] ; then
 	if [ -z "$DB_PATH" ] ; then
-		DBTEXT_PATH="/usr/local/share/kamailio/dbtext/kamailio"
+		DBTEXT_PATH="/usr/share/kamailio/dbtext/kamailio"
 	else
 		DBTEXT_PATH=$DB_PATH
 	fi
diff --git a/utils/kamctl/kamdbctl.base b/utils/kamctl/kamdbctl.base
index e129629..090ab48 100644
--- a/utils/kamctl/kamdbctl.base
+++ b/utils/kamctl/kamdbctl.base
@@ -63,7 +63,7 @@ FOREVER=${FOREVER:-2030-05-28 21:32:15}
 DEFAULT_CFG_DIR=/usr/local/etc/kamailio
 
 # default location for data files
-DEFAULT_DATA_DIR=/usr/local/share/kamailio
+DEFAULT_DATA_DIR=/usr/share/kamailio
 
 # Needed programs
 MD5=${MD5:-md5sum}
diff --git a/utils/kamctl/kamdbctl.db_berkeley b/utils/kamctl/kamdbctl.db_berkeley
index 666122a..f8f6a97 100644
--- a/utils/kamctl/kamdbctl.db_berkeley
+++ b/utils/kamctl/kamdbctl.db_berkeley
@@ -108,7 +108,7 @@ if [ $ret -eq 0 ] ; then
 fi ;
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/db_berkeley/kamailio" ] ; then
 	DB_SCHEMA="$DATA_DIR/db_berkeley/kamailio"
 else
diff --git a/utils/kamctl/kamdbctl.dbtext b/utils/kamctl/kamdbctl.dbtext
index 8c2019e..d6c3645 100644
--- a/utils/kamctl/kamdbctl.dbtext
+++ b/utils/kamctl/kamdbctl.dbtext
@@ -9,7 +9,7 @@
 #             control tool for database tasks, like the kamctl (henning)
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/dbtext/kamailio" ] ; then
 	DB_SCHEMA="$DATA_DIR/dbtext/kamailio"
 else
diff --git a/utils/kamctl/kamdbctl.mysql b/utils/kamctl/kamdbctl.mysql
index 49915dc..275ed71 100644
--- a/utils/kamctl/kamdbctl.mysql
+++ b/utils/kamctl/kamdbctl.mysql
@@ -16,7 +16,7 @@
 # 2007-06-11  Use a common control tool for database tasks, like the kamctl
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/mysql" ] ; then
 	DB_SCHEMA="$DATA_DIR/mysql"
 else
diff --git a/utils/kamctl/kamdbctl.oracle b/utils/kamctl/kamdbctl.oracle
index e6de52f..63b7e94 100644
--- a/utils/kamctl/kamdbctl.oracle
+++ b/utils/kamctl/kamdbctl.oracle
@@ -11,7 +11,7 @@ DBSYSUSER="sys"
 
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/oracle" ] ; then
 	DB_SCHEMA="$DATA_DIR/oracle"
 else
diff --git a/utils/kamctl/kamdbctl.pgsql b/utils/kamctl/kamdbctl.pgsql
index 3b46c04..a4b6702 100644
--- a/utils/kamctl/kamdbctl.pgsql
+++ b/utils/kamctl/kamdbctl.pgsql
@@ -28,7 +28,7 @@
 # 2007-06-11  Use a common control tool for database tasks, like the kamctl
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/postgres" ] ; then
 	DB_SCHEMA="$DATA_DIR/postgres"
 else
diff --git a/utils/kamctl/kamdbctl.sqlite b/utils/kamctl/kamdbctl.sqlite
index 1aeae55..880b261 100644
--- a/utils/kamctl/kamdbctl.sqlite
+++ b/utils/kamctl/kamdbctl.sqlite
@@ -3,7 +3,7 @@
 #
 
 # path to the database schemas
-DATA_DIR="/usr/local/share/kamailio"
+DATA_DIR="/usr/share/kamailio"
 if [ -d "$DATA_DIR/db_sqlite" ] ; then
 	DB_SCHEMA="$DATA_DIR/db_sqlite"
 else
