--- kamailio-4.0.4.orig/modules/ldap/ld_session.h
+++ kamailio-4.0.4/modules/ldap/ld_session.h
@@ -33,6 +33,7 @@
 #ifndef LD_SESSION_H
 #define LD_SESSION_H
 
+#include <sys/time.h>
 #include <ldap.h>
 
 #include "iniparser.h"
--- kamailio-4.0.4.orig/modules/xhttp_pi/xhttp_pi_fnc.c
+++ kamailio-4.0.4/modules/xhttp_pi/xhttp_pi_fnc.c
@@ -836,7 +836,7 @@
 			if(vals==NULL||ids==NULL) {LM_ERR("oom\n"); return -1;}
 			col_vals = vals; col_ids = ids;
 			vals = &col_vals[size]; ids = &col_ids[size];
-			memset(vals, 0, sizeof(str*)); memset(ids, 0, sizeof(str*));
+			memset(vals, 0, sizeof(str)); memset(ids, 0, sizeof(str));
 			/* Retrieve the node attribute */
 			attr.s = ph_xmlNodeGetAttrContentByName(node,
 							XHTTP_PI_XML_ID_ATTR);
--- kamailio-4.0.4.orig/io_wait.h
+++ kamailio-4.0.4/io_wait.h
@@ -63,7 +63,7 @@
 #endif
 
 #define _GNU_SOURCE  /* for POLLRDHUP on linux */
-#include <sys/poll.h>
+#include <poll.h>
 #include <fcntl.h>
 
 #ifdef HAVE_EPOLL
--- kamailio-4.0.4.orig/modules/tls/tls_server.c
+++ kamailio-4.0.4/modules/tls/tls_server.c
@@ -35,7 +35,7 @@
  */
 
 
-#include <sys/poll.h>
+#include <poll.h>
 #include <openssl/err.h>
 #include <openssl/ssl.h>
 #include "../../dprint.h"
--- kamailio-4.0.4.orig/stats.h
+++ kamailio-4.0.4/stats.h
@@ -32,7 +32,7 @@
 
 #include <ctype.h>
 #include <sys/mman.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/time.h>
 #include <time.h>
 #include <sys/utsname.h>
--- kamailio-4.0.4.orig/tcp_main.c
+++ kamailio-4.0.4/tcp_main.c
@@ -150,7 +150,7 @@
 #ifdef HAVE_SELECT
 #include <sys/select.h>
 #endif
-#include <sys/poll.h>
+#include <poll.h>
 
 
 #include "ip_addr.h"
