diff --git a/tests/first_run.rs b/tests/first_run.rs
index 39427fb..acde304 100644
--- a/tests/first_run.rs
+++ b/tests/first_run.rs
@@ -40,7 +40,7 @@ fn test_first_run() {
     let shell = test.shell.clone();
     test.start("first-run", glib::spawn_future_local(clone!(@weak shell => async move {
         let _ = ready_rx.recv().await.unwrap();
-        glib::timeout_future(Duration::from_millis(2000)).await;
+        glib::timeout_future(Duration::from_millis(20000)).await;
 
         // The first-run script should have started.
         assert!(touch.exists());
