diff -rupN a/app/app.go b/app/app.go
--- a/app/app.go	2025-10-24 08:49:19.000000000 +0000
+++ b/app/app.go	2026-01-16 18:27:57.990000000 +0000
@@ -29,7 +29,7 @@ func Branch() string    { return branch
 func BuildDate() string { return strings.ReplaceAll(buildDate, "_", " ") }
 
 // Copyright is the copyright owner string
-const Copyright = "2018-2025 Laurence Morgan"
+const Copyright = "2018-2026 Laurence Morgan"
 
 // License is the projects software license
 const License = "GPL v2"
diff -rupN a/utils/docgen/main.go b/utils/docgen/main.go
--- a/utils/docgen/main.go	2025-10-24 08:49:19.000000000 +0000
+++ b/utils/docgen/main.go	2026-01-16 18:27:17.400000000 +0000
@@ -14,7 +14,7 @@ const (
 	Version = "5.0.1"
 
 	// Copyright is the copyright owner string
-	Copyright = "(c) 2018-2025 Laurence Morgan"
+	Copyright = "(c) 2018-2026 Laurence Morgan"
 
 	// License is the projects software license
 	License = "License GPL v2"
