--- a/src/bootstrap/bin/rustc.rs.orig	2019-05-03 14:49:24.660025779 +0200
+++ b/src/bootstrap/bin/rustc.rs	2019-05-03 14:51:22.606695185 +0200
@@ -221,11 +221,11 @@
                 // so. Note that this is definitely a hack, and we should likely
                 // flesh out rpath support more fully in the future.
                 cmd.arg("-Z").arg("osx-rpath-install-name");
-                Some("-Wl,-rpath,@loader_path/../lib")
+                Some("-Wl,-rpath,@loader_path/../lib".to_string())
             } else if !target.contains("windows") &&
                       !target.contains("wasm32") &&
                       !target.contains("fuchsia") {
-                Some("-Wl,-rpath,$ORIGIN/../lib")
+                Some(format!("-Wl,-rpath,$ORIGIN/../lib/rustlib/{}/lib", target))
             } else {
                 None
             };
