--- a/src/etc/rust-gdb
+++ b/src/etc/rust-gdb
@@ -14,7 +14,7 @@
 
 # Find out where the pretty printer Python module is
 RUSTC_SYSROOT=`rustc --print=sysroot`
-GDB_PYTHON_MODULE_DIRECTORY="$RUSTC_SYSROOT/lib/rustlib/etc"
+GDB_PYTHON_MODULE_DIRECTORY="$RUSTC_SYSROOT/share/rust/etc"
 
 # Run GDB with the additional arguments that load the pretty printers
 # Set the environment variable `RUST_GDB` to overwrite the call to a
--- a/src/etc/rust-lldb
+++ b/src/etc/rust-lldb
@@ -33,7 +33,7 @@
 RUSTC_SYSROOT=`rustc --print sysroot`
 
 # Write the LLDB script to the tempfile
-echo "command script import \"$RUSTC_SYSROOT/lib/rustlib/etc/lldb_rust_formatters.py\"" >> $TMPFILE
+echo "command script import \"$RUSTC_SYSROOT/share/rust/etc/lldb_rust_formatters.py\"" >> $TMPFILE
 echo "type summary add --no-value --python-function lldb_rust_formatters.print_val -x \".*\" --category Rust" >> $TMPFILE
 echo "type category enable Rust" >> $TMPFILE
 
