--- rustc-1.58.0-src.orig/vendor/libc/src/unix/mod.rs
+++ rustc-1.58.0-src/vendor/libc/src/unix/mod.rs
@@ -329,11 +329,11 @@
         #[link(name = "c", cfg(not(target_feature = "crt-static")))]
         extern {}
     } else if #[cfg(target_env = "musl")] {
+	#[link(name = "c")]
+	extern {}
         #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", kind = "static", modifiers = "-bundle",
+                   link(name = "gcc", kind = "static", modifiers = "-bundle",
                         cfg(target_feature = "crt-static")))]
-        #[cfg_attr(feature = "rustc-dep-of-std",
-                   link(name = "c", cfg(not(target_feature = "crt-static"))))]
         extern {}
     } else if #[cfg(target_os = "emscripten")] {
         #[link(name = "c")]
