--- rustc-1.50.0-src.orig/compiler/rustc_target/src/spec/mips64_alpine_linux_musl.rs
+++ rustc-1.50.0-src/compiler/rustc_target/src/spec/mips64_alpine_linux_musl.rs
@@ -4,6 +4,7 @@
     let mut base = super::mips64_unknown_linux_muslabi64::target();
 
     base.llvm_target = "mips64-alpine-linux-musl".to_string();
+    base.options.features = "+mips64r2,+soft-float".to_string();
     base.options.vendor = "alpine".to_string();
     base.options.crt_static_default = false;
     base.options.static_position_independent_executables = true;
