--- a/t/51_table_column_metadata.t
+++ b/t/51_table_column_metadata.t
@@ -19,7 +19,7 @@
 		my $data = $dbh->$call_func(undef, 'foo', 'id', 'table_column_metadata');
 		ok $data && ref $data eq ref {}, "got a metadata";
 		ok $data->{auto_increment}, "id is auto incremental";
-		is $data->{data_type} => 'integer', "data type is correct";
+		is $data->{data_type} => 'INTEGER', "data type is correct";
 		ok $data->{primary}, "id is a primary key";
 		ok !$data->{not_null}, "id is not null";
 	}
