--- a/Makefile
+++ b/Makefile
@@ -5,11 +5,11 @@
 
 LIBS = toxcore ncursesw libconfig libcurl
 
-CFLAGS ?= -std=c99 -pthread -Wall -Wpedantic -Wunused -fstack-protector-all -Wvla -Wno-missing-braces
+CFLAGS = -std=gnu99 -pthread -Wall -Wpedantic -Wunused -fstack-protector-all -Wvla -Wno-missing-braces
 CFLAGS += '-DTOXICVER="$(VERSION)"' -DHAVE_WIDECHAR -D_XOPEN_SOURCE_EXTENDED -D_FILE_OFFSET_BITS=64
 CFLAGS += '-DPACKAGE_DATADIR="$(abspath $(DATADIR))"'
 CFLAGS += ${USER_CFLAGS}
-LDFLAGS ?=
+LDFLAGS =
 LDFLAGS += ${USER_LDFLAGS}
 
 OBJ = autocomplete.o avatars.o bootstrap.o chat.o chat_commands.o conference.o configdir.o curl_util.o execute.o
@@ -72,19 +72,16 @@
 all: $(BUILD_DIR)/toxic
 
 $(BUILD_DIR)/toxic: $(OBJ)
-	@echo "  LD    $(@:$(BUILD_DIR)/%=%)"
-	@$(CC) $(CFLAGS) -o $(BUILD_DIR)/toxic $(OBJ) $(LDFLAGS)
+	$(CC) $(CFLAGS) -o $(BUILD_DIR)/toxic $(OBJ) $(LDFLAGS)
 
 $(BUILD_DIR)/osx_video.o: $(SRC_DIR)/$(OSX_VIDEO)
-	@echo "  CC    $(@:$(BUILD_DIR)/)osx_video.o"
-	@$(CC) $(CFLAGS) -o $(BUILD_DIR)/osx_video.o -c $(SRC_DIR)/$(OSX_VIDEO)
+	$(CC) $(CFLAGS) -o $(BUILD_DIR)/osx_video.o -c $(SRC_DIR)/$(OSX_VIDEO)
 
 $(BUILD_DIR)/%.o: $(SRC_DIR)/%.c
 	@if [ ! -e $(BUILD_DIR) ]; then \
 		mkdir -p $(BUILD_DIR) ;\
 	fi
-	@echo "  CC    $(@:$(BUILD_DIR)/%=%)"
-	@$(CC) $(CFLAGS) -o $(BUILD_DIR)/$*.o -c $(SRC_DIR)/$*.c
+	$(CC) $(CFLAGS) -o $(BUILD_DIR)/$*.o -c $(SRC_DIR)/$*.c
 	@$(CC) -MM $(CFLAGS) $(SRC_DIR)/$*.c >$(BUILD_DIR)/$*.d
 
 clean:
