diff --git a/Makefile b/Makefile
index 7e09923..93153ac 100644
--- a/Makefile
+++ b/Makefile
@@ -120,15 +120,8 @@ ifdef LOW_MEMORY
  endif
 endif
 
-# Use jody_hash instead of xxHash if requested
-ifdef USE_JODY_HASH
- COMPILER_OPTIONS += -DUSE_JODY_HASH -DNO_XXHASH2
- OBJS_CLEAN += xxhash.o
- else
- ifndef EXTERNAL_HASH_LIB
-  OBJS += xxhash.o
- endif
-endif  # USE_JODY_HASH
+CFLAGS += `pkg-config --cflags libxxhash`
+LDFLAGS += `pkg-config --libs libxxhash`
 
 # Stack size limit can be too small for deep directory trees, so set to 16 MiB
 # The ld syntax for Windows is the same for both Cygwin and MinGW
diff --git a/jdupes.c b/jdupes.c
index e3e5762..bc66c05 100644
--- a/jdupes.c
+++ b/jdupes.c
@@ -62,7 +62,7 @@
 #include "version.h"
 
 #ifndef USE_JODY_HASH
- #include "xxhash.h"
+ #include <xxhash.h>
 #endif
 #ifdef ENABLE_DEDUPE
  #ifdef __linux__
diff --git a/filehash.c b/filehash.c
index 8b05d87..2975fbb 100644
--- a/filehash.c
+++ b/filehash.c
@@ -21,7 +21,7 @@
 #include "interrupt.h"
 #include "progress.h"
 #include "jdupes.h"
-#include "xxhash.h"
+#include <xxhash.h>
 
 const char *hash_algo_list[2] = {
   "xxHash64 v2",
