Author: Pellegrino Prevete <pellegrinoprevete@gmail.com>
Summary: fix missing locale in Alpine
----

--- lollypop-1.3.2/lollypop.in.orig
+++ lollypop-1.3.2/lollypop.in
@@ -34,8 +34,6 @@
 if __name__ == "__main__":
     install_excepthook()
 
-    locale.bindtextdomain('lollypop', localedir)
-    locale.textdomain('lollypop')
     gettext.bindtextdomain('lollypop', localedir)
     gettext.textdomain('lollypop')
 
