diff -ru freeradius-server-2.1.7.orig/Make.inc.in freeradius-server-2.1.7/Make.inc.in
--- freeradius-server-2.1.7.orig/Make.inc.in	2009-10-09 08:38:58.000000000 +0000
+++ freeradius-server-2.1.7/Make.inc.in	2009-10-09 08:39:15.000000000 +0000
@@ -10,6 +10,7 @@
 sysconfdir	= @sysconfdir@
 localstatedir	= @localstatedir@
 libdir		= @libdir@
+pkglibdir	= @libdir@/freeradius
 bindir		= @bindir@
 sbindir		= @sbindir@
 docdir		= @docdir@
diff -ru freeradius-server-2.1.7.orig/raddb/radiusd.conf.in freeradius-server-2.1.7/raddb/radiusd.conf.in
--- freeradius-server-2.1.7.orig/raddb/radiusd.conf.in	2009-10-09 08:38:58.000000000 +0000
+++ freeradius-server-2.1.7/raddb/radiusd.conf.in	2009-10-09 08:39:15.000000000 +0000
@@ -103,7 +103,7 @@
 #	make
 #	make install
 #
-libdir = @libdir@
+libdir = @libdir@/freeradius
 
 #  pidfile: Where to place the PID of the RADIUS server.
 #
diff -ru freeradius-server-2.1.7.orig/src/modules/Makefile freeradius-server-2.1.7/src/modules/Makefile
--- freeradius-server-2.1.7.orig/src/modules/Makefile	2009-10-09 08:38:58.000000000 +0000
+++ freeradius-server-2.1.7/src/modules/Makefile	2009-10-09 08:39:15.000000000 +0000
@@ -12,7 +12,7 @@
 	@$(MAKE) $(MFLAGS) WHAT_TO_MAKE=$@ common
 
 install:
-	$(INSTALL) -d -m 755 $(R)$(libdir)
+	$(INSTALL) -d -m 755 $(R)$(pkglibdir)
 	@$(MAKE) $(MFLAGS) WHAT_TO_MAKE=$@ common
 
 clean:
diff -ru freeradius-server-2.1.7.orig/src/modules/rules.mak freeradius-server-2.1.7/src/modules/rules.mak
--- freeradius-server-2.1.7.orig/src/modules/rules.mak	2009-10-09 08:38:58.000000000 +0000
+++ freeradius-server-2.1.7/src/modules/rules.mak	2009-10-09 08:40:56.000000000 +0000
@@ -123,7 +123,7 @@
 $(TARGET).la: $(LT_OBJS)
 	$(LIBTOOL) --mode=link $(CC) -release $(RADIUSD_VERSION) \
 	-module $(LINK_MODE) $(LDFLAGS) $(RLM_LDFLAGS) -o $@     \
-	-rpath $(libdir) $^ $(LIBRADIUS) $(RLM_LIBS) $(LIBS)
+	-rpath $(pkglibdir) $^ $(LIBRADIUS) $(RLM_LIBS) $(LIBS)
 
 #######################################################################
 #
@@ -164,13 +164,13 @@
 #  Do any module-specific installation.
 #
 #  If there isn't a TARGET defined, then don't do anything.
-#  Otherwise, install the libraries into $(libdir)
+#  Otherwise, install the libraries into $(pkglibdir)
 #
 install:
 	@[ "x$(RLM_INSTALL)" = "x" ] || $(MAKE) $(MFLAGS) $(RLM_INSTALL)
 	if [ "x$(TARGET)" != "x" ]; then \
 	    $(LIBTOOL) --mode=install $(INSTALL) -c \
-		$(TARGET).la $(R)$(libdir)/$(TARGET).la || exit $$?; \
+ 		$(TARGET).la $(R)$(pkglibdir)/$(TARGET).la || exit $$?; \
 	    rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la; \
 	    ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la || exit $$?; \
 	fi
