--- freeradius-server-3.0.13.orig/src/main/tls.c
+++ freeradius-server-3.0.13/src/main/tls.c
@@ -2131,7 +2131,7 @@
 	}
 
 	if (lookup == 0) {
-#if OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if OPENSSL_VERSION_NUMBER >= 0x10100000L && OPENSSL_VERSION_NUMBER < 0x20000000L
 		ext_list = X509_get0_extensions(client_cert);
 #else
 		X509_CINF	*client_inf;
@@ -2903,6 +2903,7 @@
 		SSL_CTX_set_verify_depth(ctx, conf->verify_depth);
 	}
 
+#if OPENSSL_VERSION_NUMBER < 0x20000000L
 	/* Load randomness */
 	if (conf->random_file) {
 		if (!(RAND_load_file(conf->random_file, 1024*10))) {
@@ -2910,6 +2911,7 @@
 			return NULL;
 		}
 	}
+#endif
 
 	/*
 	 * Set the cipher list if we were told to
