diff --git a/tools/patchwork-update-commits b/tools/patchwork-update-commits
index 820fd1c..ca0022a 100755
--- a/tools/patchwork-update-commits
+++ b/tools/patchwork-update-commits
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 toolsdir="$(dirname "$0")"
 pwpath="${toolsdir}"/../patchwork
diff --git a/tools/post-receive.hook b/tools/post-receive.hook
index 8f05b8d..3d6ec1b 100755
--- a/tools/post-receive.hook
+++ b/tools/post-receive.hook
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Git post-receive hook to update Patchwork patches after Git pushes
 #
@@ -15,7 +15,7 @@ STATE_MAP="refs/heads/master:Accepted"
 #
 EXCLUDE=""
 
-PWDIR=/srv/patchwork/patchwork
+PWDIR=/usr/share/webapps/patchwork/patchwork
 
 do_exit=0
 trap "do_exit=1" INT
