diff --git a/include/spdlog/details/pattern_formatter-inl.h b/include/spdlog/details/pattern_formatter-inl.h
index 1b95eb6..a77a472 100644
--- a/include/spdlog/details/pattern_formatter-inl.h
+++ b/include/spdlog/details/pattern_formatter-inl.h
@@ -15,6 +15,7 @@
 
 #include <array>
 #include <chrono>
+#include <cassert>
 #include <ctime>
 #include <cctype>
 #include <cstring>
diff --git a/include/spdlog/details/thread_pool-inl.h b/include/spdlog/details/thread_pool-inl.h
index 77d0b1f..c773149 100644
--- a/include/spdlog/details/thread_pool-inl.h
+++ b/include/spdlog/details/thread_pool-inl.h
@@ -9,6 +9,8 @@
 
 #include "spdlog/common.h"
 
+#include <cassert>
+
 namespace spdlog {
 namespace details {

