diff --git a/Samba/source/pidl/pidl b/Samba/source/pidl/pidl
index 8084213..59e467f 100755
--- a/Samba/source/pidl/pidl
+++ b/Samba/source/pidl/pidl
@@ -580,7 +580,7 @@ sub process_file($)
 		require Parse::Pidl::IDL;
 
 		$pidl = Parse::Pidl::IDL::parse_file($idl_file, \@opt_incdirs);
-		defined @$pidl || die "Failed to parse $idl_file";
+		@$pidl || die "Failed to parse $idl_file";
 		require Parse::Pidl::Typelist;
 		Parse::Pidl::Typelist::LoadIdl($pidl);
 	}
