diff --git a/Samba/source/pidl/idl.yp b/Samba/source/pidl/idl.yp
index b5c5185..6469068 100644
--- a/Samba/source/pidl/idl.yp
+++ b/Samba/source/pidl/idl.yp
@@ -483,7 +483,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{INPUT_FILENAME} = $2;
 				goto again;
diff --git a/Samba/source/pidl/lib/Parse/Pidl/IDL.pm b/Samba/source/pidl/lib/Parse/Pidl/IDL.pm
index d4820ff..da7e3c8 100644
--- a/Samba/source/pidl/lib/Parse/Pidl/IDL.pm
+++ b/Samba/source/pidl/lib/Parse/Pidl/IDL.pm
@@ -3169,7 +3169,7 @@
 
 	for ($parser->YYData->{INPUT}) {
 		if (/^\#/) {
-			if (s/^\# (\d+) \"(.*?)\"( \d+|)//) {
+			if (s/^\# (\d+) \"(.*?)\"(( \d+){1,4}|)//) {
 				$parser->YYData->{LINE} = $1-1;
 				$parser->YYData->{INPUT_FILENAME} = $2;
 				goto again;
