diff --git a/source/musicformats/music_midi.cpp b/source/musicformats/music_midi.cpp
index 0e82849..f1c9ec3 100644
--- a/source/musicformats/music_midi.cpp
+++ b/source/musicformats/music_midi.cpp
@@ -36,6 +36,7 @@
 
 #include <string>
 #include <algorithm>
+#include <stdexcept>
 #include <assert.h>
 #include "zmusic/zmusic_internal.h"
 #include "zmusic/musinfo.h"
diff --git a/source/streamsources/music_gme.cpp b/source/streamsources/music_gme.cpp
index c6248d9..366aa99 100644
--- a/source/streamsources/music_gme.cpp
+++ b/source/streamsources/music_gme.cpp
@@ -38,6 +38,7 @@
 //#define GME_DLL
 
 #include <algorithm>
+#include <stdexcept>
 #include "streamsource.h"
 #include <gme/gme.h>
 #include <mutex>
diff --git a/source/zmusic/fileio.h b/source/zmusic/fileio.h
index 87a363d..85dccc1 100644
--- a/source/zmusic/fileio.h
+++ b/source/zmusic/fileio.h
@@ -23,6 +23,7 @@
 
 
 #pragma once
+#include <cstdint>
 #include <stdio.h>
 #include <string.h>
 #include <vector>
diff --git a/source/zmusic/zmusic_internal.h b/source/zmusic/zmusic_internal.h
index 0de89e9..021a968 100644
--- a/source/zmusic/zmusic_internal.h
+++ b/source/zmusic/zmusic_internal.h
@@ -25,6 +25,7 @@ typedef class MusInfo *ZMusic_MusicStream;
 #define HAVE_WILDMIDI	// LGPL v3.0
 #endif
 
+#include <stdexcept>
 #include "zmusic.h"
 #include "fileio.h"
 
