diff --git a/cmake/Modules/DefineCompilerFlags.cmake b/cmake/Modules/DefineCompilerFlags.cmake
index 4d90f36..cf6e459 100644
--- a/cmake/Modules/DefineCompilerFlags.cmake
+++ b/cmake/Modules/DefineCompilerFlags.cmake
@@ -52,11 +52,6 @@ if (UNIX)
         endif (WITH_FPIC)
     endif (NOT CYGWIN)
 
-    check_c_compiler_flag("-D_FORTIFY_SOURCE=2" WITH_FORTIFY_SOURCE)
-    if (WITH_FORTIFY_SOURCE)
-      add_definitions(-D_FORTIFY_SOURCE=2)
-    endif (WITH_FORTIFY_SOURCE)
-
     if (NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
       add_definitions(-O)
     endif ()
