diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 24ad308..94d3d7a 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -218,69 +218,5 @@ if (CGREEN_WITH_LIBXML2)
   target_link_libraries(${libxml_output_library} ${CGREEN_LIBRARY})
 endif (CGREEN_WITH_LIBXML2)
 
-# TODO We should not add these if the cgreen-runner was not built e.g. no 'nm' available
-macro_add_test(
-    NAME constraint_messsages
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            constraint_messages_tests       # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${constraint_messages_library}.expected
-)
-
-macro_add_test(
-    NAME custom_constraint_messsages
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            custom_constraint_messages_tests       # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${custom_constraint_messages_library}.expected
-)
-
-macro_add_test(
-    NAME mock_messsages
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            mock_messages_tests             # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${mock_messages_library}.expected
-)
-
-macro_add_test(NAME failure_messages
-    COMMAND env "CGREEN_PER_TEST_TIMEOUT=2" ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            failure_messages_tests          # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${failure_messages_library}.expected
-)
-
-macro_add_test(NAME assertion_messages
-    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            assertion_messages_tests        # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${assertion_messages_library}.expected
-)
-
-macro_add_test(NAME ignore_messages
-    COMMAND env "CGREEN_PER_TEST_TIMEOUT=2" ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_runner_output_diff
-            ignore_messages_tests           # Name
-            ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-            ${ignore_messages_library}.expected
-)
-
-if (CGREEN_WITH_XML)
-  macro_add_test(NAME xml_output
-      COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_xml_output_diff
-              xml_output_tests                # Name
-              ${CMAKE_CURRENT_SOURCE_DIR}     # Where sources are
-              ${xml_output_library}.expected
-  )
-endif (CGREEN_WITH_XML)
-
-if (CGREEN_WITH_LIBXML2)
-  macro_add_test(NAME libxml_output
-      COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/../tools/cgreen_libxml_output_diff
-      	libxml_output_tests
-      	${CMAKE_CURRENT_SOURCE_DIR}
-      	${libxml_output_library}.expected
-  )
-endif (CGREEN_WITH_LIBXML2)
-
 # add verification that all public api is available as it should
 add_subdirectory(api)
