--- a/tests/tester.nim
+++ b/tests/tester.nim
@@ -278,7 +278,7 @@
     cd "invalidPackage":
       let (output, exitCode) = execNimble("check")
       let lines = output.strip.processOutput()
-      check(lines[^2].contains("undeclared identifier: 'thisFieldDoesNotExist'"))
+      check(lines[^5].contains("undeclared identifier: 'thisFieldDoesNotExist'"))
       check exitCode == QuitFailure
 
   test "can accept short flags (#329)":
