diff -Nurp a/src/rgw/rgw_d3n_datacache.cc b/src/rgw/rgw_d3n_datacache.cc
--- a/src/rgw/rgw_d3n_datacache.cc	2022-02-26 09:14:35.977956275 +0000
+++ b/src/rgw/rgw_d3n_datacache.cc	2022-02-26 14:05:24.935725031 +0000
@@ -97,7 +97,7 @@ void D3nDataCache::init(CephContext *_cc
   if (conf_eviction_policy == "random")
     eviction_policy = _eviction_policy::RANDOM;
 
-#if defined(HAVE_LIBAIO)
+#if defined(HAVE_LIBAIO_BROKEN)
   // libaio setup
   struct aioinit ainit{0};
   ainit.aio_threads = cct->_conf.get_val<int64_t>("rgw_d3n_libaio_aio_threads");
