diff --git a/version.sh b/version.sh
index 281de74..22cd96d 100755
--- a/version.sh
+++ b/version.sh
@@ -1,10 +1 @@
-if test "x$LIBSASS_VERSION" = "x"; then
-  LIBSASS_VERSION=`git describe --abbrev=4 --dirty --always --tags 2>/dev/null`
-fi
-if test "x$LIBSASS_VERSION" = "x"; then
-  LIBSASS_VERSION=`cat VERSION 2>/dev/null`
-fi
-if test "x$LIBSASS_VERSION" = "x"; then
-  LIBSASS_VERSION="[na]"
-fi
-echo $LIBSASS_VERSION
+echo 3.6.3
