diff --git a/coverage/tomlconfig.py b/coverage/tomlconfig.py
index 148c34f..4a43d5b 100644
--- a/coverage/tomlconfig.py
+++ b/coverage/tomlconfig.py
@@ -62,7 +62,7 @@ class TomlConfigParser:
             has_toml = re.search(r"^\[tool\.coverage\.", toml_text, flags=re.MULTILINE)
             if self.our_file or has_toml:
                 # Looks like they meant to read TOML, but we can't read it.
-                msg = "Can't read {!r} without TOML support. Install with [toml] extra"
+                msg = "Can't read {!r} without TOML support. Install with `apk add py3-tomli` or [toml] extra"
                 raise ConfigError(msg.format(filename))
             return []
 
