diff --git a/coverage/tomlconfig.py b/coverage/tomlconfig.py
index 3ad5815..931192d 100644
--- a/coverage/tomlconfig.py
+++ b/coverage/tomlconfig.py
@@ -58,7 +58,7 @@ class TomlConfigParser:
             has_toml = re.search(r"^\[tool\.coverage\.", toml_text, flags=re.MULTILINE)
             if self.our_file or has_toml:
                 # Looks like they meant to read TOML, but we can't read it.
-                msg = "Can't read {!r} without TOML support. Install with [toml] extra"
+                msg = "Can't read {!r} without TOML support. Install with `apk add py3-toml` or [toml] extra"
                 raise CoverageException(msg.format(filename))
             return []
 
