--- ./src/openssl.c.orig
+++ ./src/openssl.c
@@ -36,7 +36,7 @@
 #include <errno.h>        /* errno */
 
 #include <sys/types.h>    /* ssize_t pid_t */
-#if !defined __sun && !defined _AIX
+#if defined __GLIBC__
 #include <sys/sysctl.h>   /* CTL_KERN KERN_RANDOM RANDOM_UUID KERN_URND KERN_ARND sysctl(2) */
 #endif
 #include <sys/time.h>     /* struct timeval gettimeofday(2) */
@@ -4609,7 +4609,7 @@
 
 
 #ifndef HAVE_RANDOM_UUID
-#define HAVE_RANDOM_UUID (defined __linux) /* RANDOM_UUID is an enum, not macro */
+#define HAVE_RANDOM_UUID (defined __GLIBC__) /* RANDOM_UUID is an enum, not macro */
 #endif
 
 #ifndef HAVE_KERN_URND
