--- rocksdb-7.10.2/db/db_write_test.cc
+++ rocksdb-8.3.2/db/db_write_test.cc
@@ -4,6 +4,7 @@
 //  (found in the LICENSE.Apache file in the root directory).
 
 #include <atomic>
+#include <cstdint>
 #include <fstream>
 #include <memory>
 #include <thread>
--- rocksdb-7.10.2/table/block_based/data_block_hash_index.h
+++ rocksdb-8.3.2/table/block_based/data_block_hash_index.h
@@ -5,6 +5,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- rocksdb-7.10.2/util/string_util.h
+++ rocksdb-8.3.2/util/string_util.h
@@ -6,6 +6,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <sstream>
 #include <string>
 #include <unordered_map>
