diff --git a/meek-client/torrc b/meek-client/torrc
index 9272271..366d632 100644
--- a/meek-client/torrc
+++ b/meek-client/torrc
@@ -5,4 +5,4 @@ UseBridges 1
 # HTTPSProxy localhost:8080
 
 Bridge meek 0.0.2.0:3 url=https://meek.azureedge.net/ front=ajax.aspnetcdn.com
-ClientTransportPlugin meek exec ./meek-client --log meek-client.log
+ClientTransportPlugin meek exec /usr/bin/meek-client --log meek-client.log
diff --git a/meek-server/torrc b/meek-server/torrc
index 613cb9c..0ad1e6c 100644
--- a/meek-server/torrc
+++ b/meek-server/torrc
@@ -3,5 +3,5 @@ ORPort 9001
 ExtORPort auto
 SocksPort 0
 ServerTransportListenAddr meek 0.0.0.0:7002
-ServerTransportPlugin meek exec ./meek-server --log meek-server.log --cert cert.pem --key key.pem
+ServerTransportPlugin meek exec /usr/bin/meek-server --log meek-server.log --cert cert.pem --key key.pem
 ExitPolicy reject *:*
