diff --git a/t/026FileApp.t b/t/026FileApp.t
index b3ae4f4..6c3b345 100644
--- a/t/026FileApp.t
+++ b/t/026FileApp.t
@@ -484,7 +484,7 @@ EOT
   
   my ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size, $atime,$mtime,$ctime,$blksize,$blocks) = stat("${testpath}_1");
   
-  is($mode & 07777,0750); #Win32 777
+  is($mode & 0777,0750); #Win32 777
   
    ($dev,$ino,$mode,$nlink,$uid,$gid,$rdev,$size, $atime,$mtime,$ctime,$blksize,$blocks) = stat($testmkpathfile);
   
