diff --git a/Source/WebKit/qt/tests/util.h b/Source/WebKit/qt/tests/util.h
index 4925aa4..9f834f7 100644
--- a/Source/WebKit/qt/tests/util.h
+++ b/Source/WebKit/qt/tests/util.h
@@ -23,7 +23,7 @@
 #endif
 
 #include <QEventLoop>
-#include <QSignalSpy>
+#include <QtTest/QSignalSpy>
 #include <QTimer>
 
 #if !defined(TESTS_SOURCE_DIR)
diff --git a/Source/WebKit2/UIProcess/API/qt/tests/util.h b/Source/WebKit2/UIProcess/API/qt/tests/util.h
index 8711d5b..6bb97fa 100644
--- a/Source/WebKit2/UIProcess/API/qt/tests/util.h
+++ b/Source/WebKit2/UIProcess/API/qt/tests/util.h
@@ -19,7 +19,7 @@
 // Functions and macros that really need to be in QTestLib
 
 #include <QEventLoop>
-#include <QSignalSpy>
+#include <QtTest/QSignalSpy>
 #include <QTimer>
 
 class QQuickWebView;
