--- Python-3.6.3.orig/Lib/test/test_ssl.py
+++ Python-3.6.3/Lib/test/test_ssl.py
@@ -3303,6 +3303,7 @@
             self.assertIs(stats['client_npn_protocol'], None)
 
         @unittest.skipUnless(ssl.HAS_NPN, "NPN support needed for this test")
+        @unittest.skipIf(IS_LIBRESSL, "LibreSSL doesn't support NPN anymore")
         def test_npn_protocols(self):
             server_protocols = ['http/1.1', 'spdy/2']
             protocol_tests = [
