diff -Nurp a-Python-3.8.7/Modules/_decimal/_decimal.c b-Python-3.8.7/Modules/_decimal/_decimal.c
--- a-Python-3.8.7/Modules/_decimal/_decimal.c	2020-12-21 16:25:24.000000000 +0000
+++ b-Python-3.8.7/Modules/_decimal/_decimal.c	2021-02-10 15:45:09.976876631 +0000
@@ -3284,7 +3284,7 @@ dec_format(PyObject *dec, PyObject *args
     }
     else {
         size_t n = strlen(spec.dot);
-        if (n > 1 || (n == 1 && !isascii((uchar)spec.dot[0]))) {
+        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.dot[0]))) {
             /* fix locale dependent non-ascii characters */
             dot = dotsep_as_utf8(spec.dot);
             if (dot == NULL) {
@@ -3293,7 +3293,7 @@ dec_format(PyObject *dec, PyObject *args
             spec.dot = PyBytes_AS_STRING(dot);
         }
         n = strlen(spec.sep);
-        if (n > 1 || (n == 1 && !isascii((uchar)spec.sep[0]))) {
+        if (n > 1 || (n == 1 && !isascii((unsigned char)spec.sep[0]))) {
             /* fix locale dependent non-ascii characters */
             sep = dotsep_as_utf8(spec.sep);
             if (sep == NULL) {
diff -Nurp a-Python-3.8.7/setup.py b-Python-3.8.7/setup.py
--- a-Python-3.8.7/setup.py	2020-12-21 16:25:24.000000000 +0000
+++ b-Python-3.8.7/setup.py	2021-02-10 15:45:09.977876640 +0000
@@ -2077,7 +2077,7 @@ class PyBuildExt(build_ext):
         undef_macros = []
         if '--with-system-libmpdec' in sysconfig.get_config_var("CONFIG_ARGS"):
             include_dirs = []
-            libraries = [':libmpdec.so.2']
+            libraries = ['mpdec']
             sources = ['_decimal/_decimal.c']
             depends = ['_decimal/docstrings.h']
         else:
