diff --git a/setup.cfg b/setup.cfg
index c56ca83..31bacd8 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -3,6 +3,7 @@ name = platformdirs
 description = A small Python module for determining appropriate platform-specific dirs, e.g. a "user data dir".
 long_description = file: README.rst
 long_description_content_type = text/x-rst
+version = attr: platformdirs.__version__
 url = https://github.com/platformdirs/platformdirs
 maintainer = Bernát Gábor, Julian Berman, Ofek Lev, Ronny Pfannschmidt
 maintainer_email = gaborjbernat@gmail.com, Julian@GrayVines.com, oss@ofek.dev, opensource@ronnypfannschmidt.de
