From 4e5c7a2176773c9a1564c6603240337b3ad6b496 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Cl=C3=A9ment=20Vasseur?= <clement.vasseur@gmail.com>
Date: Fri, 13 Jun 2014 18:27:56 +0200
Subject: [PATCH] fix missing argument to syscall in fanotify_mark

---
 src/linux/fanotify.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/linux/fanotify.c b/src/linux/fanotify.c
index 1f4ef93..c6211af 100644
--- a/src/linux/fanotify.c
+++ b/src/linux/fanotify.c
@@ -9,6 +9,6 @@ int fanotify_init(unsigned flags, unsigned event_f_flags)
 int fanotify_mark(int fanotify_fd, unsigned flags, unsigned long long mask,
 	          int dfd, const char *pathname)
 {
-	return syscall(SYS_fanotify_mark, flags, __SYSCALL_LL_O(mask), dfd, pathname);
+	return syscall(SYS_fanotify_mark, fanotify_fd, flags, __SYSCALL_LL_E(mask), dfd, pathname);
 }
 
-- 
2.0.0

