From 349381aa8c0fc385e54e1068dd5f2b27af55cd12 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Fri, 11 Jul 2014 00:26:12 -0400
Subject: [PATCH] make dynamic linker accept colon as a separator for
 LD_PRELOAD

---
 src/ldso/dynlink.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ldso/dynlink.c b/src/ldso/dynlink.c
index 55124ff..6b7f0f3 100644
--- a/src/ldso/dynlink.c
+++ b/src/ldso/dynlink.c
@@ -862,8 +862,8 @@ static void load_preload(char *s)
 	int tmp;
 	char *z;
 	for (z=s; *z; s=z) {
-		for (   ; *s && isspace(*s); s++);
-		for (z=s; *z && !isspace(*z); z++);
+		for (   ; *s && (isspace(*s) || *s==':'); s++);
+		for (z=s; *z && !isspace(*z) && *z!=':'; z++);
 		tmp = *z;
 		*z = 0;
 		load_library(s, 0);
-- 
2.0.2

