From 9134a8ca79fa9c79eada278b10f4d6b1eaf271d3 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Tue, 9 Jul 2013 02:11:52 -0400
Subject: [PATCH 6/9] fix off-by-one array bound in strsignal

---
 src/string/strsignal.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/string/strsignal.c b/src/string/strsignal.c
index 3b0ae6d..d70982a 100644
--- a/src/string/strsignal.c
+++ b/src/string/strsignal.c
@@ -48,7 +48,7 @@ static const char map[] = {
 	[SIGSYS]    = 31
 };
 
-#define sigmap(x) ((unsigned)(x) > sizeof map ? 0 : map[(unsigned)(x)])
+#define sigmap(x) ((unsigned)(x) >= sizeof map ? 0 : map[(unsigned)(x)])
 
 #endif
 
-- 
1.8.3.2

