From 1fa2a32d1e2d25f493058812008face73d411b2b Mon Sep 17 00:00:00 2001
From: Daniel Sabogal <dsabogalcc@gmail.com>
Date: Sun, 4 Sep 2016 10:42:48 -0400
Subject: [PATCH] fix typo in utmpx.h

---
 include/utmpx.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/utmpx.h b/include/utmpx.h
index 9e5cc955..0429014d 100644
--- a/include/utmpx.h
+++ b/include/utmpx.h
@@ -38,7 +38,7 @@ struct utmpx *getutxline(const struct utmpx *);
 struct utmpx *pututxline(const struct utmpx *);
 void          setutxent(void);
 
-#if defined(_BSD_SOURCE) | defined(_GNU_SOURCE)
+#if defined(_BSD_SOURCE) || defined(_GNU_SOURCE)
 #define e_exit __e_exit
 #define e_termination __e_termination
 void updwtmpx(const char *, const struct utmpx *);
-- 
2.11.0

