From 4b8f94c2501e6fea3db631b73b0c91e34e2134cf Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Thu, 20 Oct 2016 17:01:56 -0400
Subject: [PATCH] fix misspelling of a legacy macro name in sys/param.h

---
 include/sys/param.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/sys/param.h b/include/sys/param.h
index 344c0d23..ce6b8019 100644
--- a/include/sys/param.h
+++ b/include/sys/param.h
@@ -7,7 +7,7 @@
 #define MAXPATHLEN 4096
 #define NBBY 8
 #define NGROUPS 32
-#define CANBSIZE 255
+#define CANBSIZ 255
 #define NOFILE 256
 #define NCARGS 131072
 #define DEV_BSIZE 512
-- 
2.11.0

