From c76c5b275a8dfa06468605369a5b025a68696183 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Mon, 14 Oct 2013 08:28:19 +0300
Subject: [PATCH] add linux tcp state enums

---
 include/netinet/tcp.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/include/netinet/tcp.h b/include/netinet/tcp.h
index 8266f21..6f75eb3 100644
--- a/include/netinet/tcp.h
+++ b/include/netinet/tcp.h
@@ -27,6 +27,18 @@
 #define TCP_FASTOPEN     23
 #define TCP_TIMESTAMP    24
 
+#define TCP_ESTABLISHED  1
+#define TCP_SYN_SENT     2
+#define TCP_SYN_RECV     3
+#define TCP_FIN_WAIT1    4
+#define TCP_FIN_WAIT2    5
+#define TCP_TIME_WAIT    6
+#define TCP_CLOSE        7
+#define TCP_CLOSE_WAIT   8
+#define TCP_LAST_ACK     9
+#define TCP_LISTEN       10
+#define TCP_CLOSING      11
+
 #if defined(_GNU_SOURCE) || defined(_BSD_SOURCE)
 #define SOL_TCP 6
 #include <sys/types.h>
-- 
1.8.4

