From f89e29829029c5003450f7b58bd00fe5d1049d09 Mon Sep 17 00:00:00 2001
From: Rich Felker <dalias@aerifal.cx>
Date: Thu, 19 Dec 2013 10:05:13 +0000
Subject: fix hangs in localtime for near-overflowing time_t values on 64-bit archs

---
diff --git a/src/time/localtime_r.c b/src/time/localtime_r.c
index c52678f..1d43d9f 100644
--- a/src/time/localtime_r.c
+++ b/src/time/localtime_r.c
@@ -4,6 +4,12 @@
 
 struct tm *__localtime_r(const time_t *restrict t, struct tm *restrict tm)
 {
+	/* Reject time_t values whose year would overflow int because
+	 * __secs_to_zone cannot safely handle them. */
+	if (*t < INT_MIN * 31622400LL || *t > INT_MAX * 31622400LL) {
+		errno = EOVERFLOW;
+		return 0;
+	}
 	__secs_to_zone(*t, 0, &tm->tm_isdst, &tm->__tm_gmtoff, 0, &tm->__tm_zone);
 	if (__secs_to_tm((long long)*t - tm->__tm_gmtoff, tm) < 0) {
 		errno = EOVERFLOW;
--
cgit v0.9.0.3-65-g4555
