diff --git a/ldso/dynlink.c b/ldso/dynlink.c
index 7b47b163..753de91d 100644
--- a/ldso/dynlink.c
+++ b/ldso/dynlink.c
@@ -552,6 +552,7 @@ static void do_relocs(struct dso *dso, size_t *rel, size_t rel_size, size_t stri
 
 static void do_relr_relocs(struct dso *dso, size_t *relr, size_t relr_size)
 {
+	if (dso == &ldso) return; // self-relocation already done a entry point
 	unsigned char *base = dso->base;
 	size_t *reloc_addr;
 	for (; relr_size; relr++, relr_size-=sizeof(size_t))
