Fixes configure.ac for use with newer autoconf versions

diff --git a/configure.ac b/configure.ac
index d488c23..91da296 100644
--- a/configure.ac
+++ b/configure.ac
@@ -89,7 +89,7 @@ AC_SUBST(PCRE_LIBS)
 AC_MSG_RESULT(found)
 
 AC_MSG_CHECKING(for suboptarg)
-AC_RUN_IFELSE(AC_LANG_PROGRAM(
+AC_RUN_IFELSE([AC_LANG_SOURCE(
 [[#include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -101,7 +101,7 @@ extern char *suboptarg;
   char *value;
 
   getsubopt (&subopts, empty_subopts, &value);
-  exit (!suboptarg);]]), [have_suboptarg="yes"],
+  exit (!suboptarg);]])], [have_suboptarg="yes"],
 [have_suboptarg="no"], [have_suboptarg="no"])
 AC_MSG_RESULT(${have_suboptarg})
 case ${have_suboptarg} in
