--- a/thirdparty/pdo_sqlite/sqlite_driver.c
+++ b/thirdparty/pdo_sqlite/sqlite_driver.c
@@ -163,6 +163,11 @@
 	return zend_i64_to_str(sqlite3_last_insert_rowid(H->db));
 }
 
+#if PHP_VERSION_ID < 80200
+static inline bool zend_str_has_nul_byte(const zend_string *str) {
+    return strlen(ZSTR_VAL(str)) != ZSTR_LEN(str);
+}
+#endif
 /* NB: doesn't handle binary strings... use prepared stmts for that */
 static zend_string* sqlite_handle_quoter(pdo_dbh_t *dbh, const zend_string *unquoted, enum pdo_param_type paramtype)
 {
