--- mongodb-src-r4.0.0/src/mongo/util/processinfo_linux.cpp.orig
+++ mongodb-src-r4.0.0/src/mongo/util/processinfo_linux.cpp
@@ -42,13 +42,7 @@
 #include <sys/time.h>
 #include <sys/utsname.h>
 #include <unistd.h>
-#ifdef __BIONIC__
-#include <android/api-level.h>
-#elif __UCLIBC__
 #include <features.h>
-#else
-#include <gnu/libc-version.h>
-#endif
 
 #include <boost/filesystem.hpp>
 #include <boost/none.hpp>
@@ -495,17 +489,9 @@
 
     BSONObjBuilder bExtra;
     bExtra.append("versionString", LinuxSysHelper::readLineFromFile("/proc/version"));
-#ifdef __BIONIC__
     stringstream ss;
-    ss << "bionic (android api " << __ANDROID_API__ << ")";
+    ss << "musl-1.1.19";
     bExtra.append("libcVersion", ss.str());
-#elif __UCLIBC__
-    stringstream ss;
-    ss << "uClibc-" << __UCLIBC_MAJOR__ << "." << __UCLIBC_MINOR__ << "." << __UCLIBC_SUBLEVEL__;
-    bExtra.append("libcVersion", ss.str());
-#else
-    bExtra.append("libcVersion", gnu_get_libc_version());
-#endif
     if (!verSig.empty())
         // optional
         bExtra.append("versionSignature", verSig);
