From b7d0c3abe2df11174cf275f06bb3e0210047cfc3 Mon Sep 17 00:00:00 2001
From: Roland Kammerer <roland.kammerer@linbit.com>
Date: Tue, 16 Jan 2018 14:28:47 +0100
Subject: [PATCH 1/1] build: fix building on alpine

---
 drbd/Kbuild | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drbd/Kbuild b/drbd/Kbuild
index e63cefc1..e7263600 100644
--- a/drbd/Kbuild
+++ b/drbd/Kbuild
@@ -150,4 +150,4 @@ $(obj)/drbd_buildtag.c: $(filter-out drbd_buildtag.o,$(drbd-y))
 		buildinfo="reproducible build, `date -u -d@${SOURCE_DATE_EPOCH} "+%F %T"`" ; 	\
 	fi ; 											\
 	echo -e "\t\t\" $$buildinfo\";\n}";							\
-	mv --force $@.new $@
+	mv -f $@.new $@
-- 
2.15.1

