diff --git a/Makefile.in b/Makefile.in
index 57cf591..624a59e 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -143,6 +143,7 @@ install: sslh-fork $(MAN)
 	mkdir -p $(DESTDIR)/$(BINDIR)
 	mkdir -p $(DESTDIR)/$(MANDIR)
 	install -p sslh-fork $(DESTDIR)/$(BINDIR)/sslh
+	install -p sslh-select $(DESTDIR)/$(BINDIR)/sslh-select
 	install -p -m 0644 $(MAN) $(DESTDIR)/$(MANDIR)/$(MAN)
 
 # "extended" install for Debian: install startup script
