diff -Naur doc/Makefile.in.orig doc/Makefile.in
--- doc/Makefile.in.orig	2004-07-27 05:57:48.000000000 -0700
+++ doc/Makefile.in	2005-02-11 22:44:49.000000000 -0800
@@ -220,7 +220,7 @@
 		$(SHELL) $(SUPPORT_SRCDIR)/mkinstalldirs $(DESTDIR)$(htmldir) ; \
 	fi
 	
-install: info installdirs bash.info
+install: installdirs
 	-$(INSTALL_DATA) $(srcdir)/bash.1 $(DESTDIR)$(man1dir)/bash${man1ext}
 	-$(INSTALL_DATA) $(srcdir)/bashbug.1 $(DESTDIR)$(man1dir)/bashbug${man1ext}
 # uncomment the next line to install the builtins man page
