diff --git a/Makefile b/Makefile
index 79c993c..c5b0f73 100644
--- a/Makefile
+++ b/Makefile
@@ -20,9 +20,9 @@ SDL_LIB=$(shell sdl-config --libs) $(MIXER) -lSDL_image
 NOSOUNDFLAG=__SOUND
 MIXER=-lSDL_mixer
 
-CFLAGS=-Wall -Wno-long-long -pedantic -ansi -O2 \
+CFLAGS := -Wall -Wno-long-long -pedantic -ansi -O2 \
 	$(shell sdl-config --cflags) -D$(NOSOUNDFLAG) \
-	-DDATA_PREFIX=\"$(DATA_PREFIX)\" -DJOY_$(JOY) -D$(TARGET_DEF)
+	-DDATA_PREFIX=\"$(DATA_PREFIX)\" -DJOY_$(JOY) -D$(TARGET_DEF) $(CFLAGS)
 
 
 all:	vectoroids
@@ -60,7 +60,7 @@ clean:
 
 
 vectoroids:	vectoroids.o
-	$(CC) $(CFLAGS) vectoroids.o -o vectoroids $(SDL_LIB)
+	$(CC) $(CFLAGS) vectoroids.o -o vectoroids $(LDFLAGS) $(SDL_LIB)
 
 
 vectoroids.o:	vectoroids.c
