From 0a001a728d1fe844258b006cf298e69cbdcf9b2e Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Tue, 24 Oct 2023 21:09:54 +0200
Subject: [PATCH] Install pil-splitter during 'make install'

---
 Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b047c49..f32b982 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,8 @@ all: $(OUT)
 	$(CC) $(LDFLAGS) -o $@ $^
 
 install: $(OUT)
-	install -D -m 755 $< $(DESTDIR)$(prefix)/bin/$<
+	install -D -m 755 pil-squasher $(DESTDIR)$(prefix)/bin/pil-squasher
+	install -D -m 755 pil-splitter $(DESTDIR)$(prefix)/bin/pil-splitter
 
 clean:
 	rm -f $(OUT) $(OBJS)
-- 
2.42.0

