From 82127427579ddcb7cfb4091968102b311214a026 Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Tue, 10 Jan 2023 06:54:05 +0100
Subject: [PATCH] Fix test compilation after b5fa401

---
 tests/src/collectionbackend_test.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tests/src/collectionbackend_test.cpp b/tests/src/collectionbackend_test.cpp
index 36bcbad8..ef1cba67 100644
--- a/tests/src/collectionbackend_test.cpp
+++ b/tests/src/collectionbackend_test.cpp
@@ -83,10 +83,10 @@ TEST_F(CollectionBackendTest, AddDirectory) {
 
   // Check the signal was emitted correctly
   ASSERT_EQ(1, spy.count());
-  CollectionDirectory dir = spy[0][0].value<Directory>();
+  CollectionDirectory dir = spy[0][0].value<CollectionDirectory>();
   EXPECT_EQ(QFileInfo("/tmp").canonicalFilePath(), dir.path);
   EXPECT_EQ(1, dir.id);
-  EXPECT_EQ(0, spy[0][1].value<SubdirectoryList>().size());
+  EXPECT_EQ(0, spy[0][1].value<CollectionSubdirectoryList>().size());
 
 }
 
@@ -105,7 +105,7 @@ TEST_F(CollectionBackendTest, RemoveDirectory) {
 
   // Check the signal was emitted correctly
   ASSERT_EQ(1, spy.count());
-  dir = spy[0][0].value<Directory>();
+  dir = spy[0][0].value<CollectionDirectory>();
   EXPECT_EQ("/tmp", dir.path);
   EXPECT_EQ(1, dir.id);
 
-- 
2.39.0

