From 963370a09dba86b661d4fec6f9152c219419f861 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 17 Jul 2009 12:35:21 +0000
Subject: [PATCH] profile: change default path

---
 profile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/profile b/profile
index ef97cc8..a292008 100644
--- a/profile
+++ b/profile
@@ -1,4 +1,4 @@
-export PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin
+export PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
 export PAGER=less
 export PS1='\h:\w\$ '
 umask 022
-- 
1.6.3.3

