diff --git a/configure.ac b/configure.ac
index 997c0d6..9030642 100644
--- a/configure.ac
+++ b/configure.ac
@@ -135,6 +135,9 @@ PKG_PROG_PKG_CONFIG
 PKG_CHECK_MODULES([TINYXML], [tinyxml], [HAVE_TINYXML=yes], [HAVE_TINYXML=no])
 AM_CONDITIONAL([HAVE_TINYXML], [test "$HAVE_TINYXML" = yes])
 
+PKG_CHECK_MODULES([NSPR], [nspr >= 4.0], [HAVE_SYSTEM_NSPR=yes], [HAVE_SYSTEM_NSPR=no])
+AM_CONDITIONAL([HAVE_SYSTEM_NSPR], [test "$HAVE_SYSTEM_NSPR" = yes])
+
 dnl Darwin -> .dylib  Linux -> .so
 if test "x$DARWIN_TRUE" = "x" ; then
     MODULE_SHARED_LDFLAGS="-dynamiclib"
@@ -342,10 +345,20 @@ else
 	CB_TINYXML_LIBS="\$(top_builddir)/src/base/tinyxml/libtinyxml.la"
 fi
 
+if test "x$HAVE_SYSTEM_NSPR" = "xyes" ; then
+	CB_NSPR_CFLAGS="$NSPR_CFLAGS"
+	CB_NSPR_LIBS="$NSPR_LIBS"
+else
+	CB_NSPR_CFLAGS="-I\$(top_srcdir)/src/sdk/mozilla_chardet/include/nsprpub/pr/include"
+	CB_NSPR_LIBS=""
+fi
+
 AC_SUBST(CB_SQUIRREL_CFLAGS)
 AC_SUBST(CB_SQUIRREL_LIBS)
 AC_SUBST(CB_TINYXML_CFLAGS)
 AC_SUBST(CB_TINYXML_LIBS)
+AC_SUBST(CB_NSPR_CFLAGS)
+AC_SUBST(CB_NSPR_LIBS)
 AC_SUBST(WX_GTK_CFLAGS)
 AC_SUBST(WX_GTK_LIBS)
 AC_SUBST(SCINTILLA_GTK_CFLAGS)
diff --git a/src/sdk/Makefile.am b/src/sdk/Makefile.am
index ae746be..cd7917c 100644
--- a/src/sdk/Makefile.am
+++ b/src/sdk/Makefile.am
@@ -8,7 +8,7 @@ AM_CPPFLAGS = $(WX_CXXFLAGS) \
 			-I$(top_srcdir)/src/sdk/wxscintilla/include \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/mfbt \
-			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/nsprpub/pr/include \
+			$(CB_NSPR_CFLAGS) \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/xpcom \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/xpcom/base \
 			-I$(top_srcdir)/src/sdk/mozilla_chardet/include/xpcom/glue
@@ -27,7 +27,8 @@ libcodeblocks_la_LIBADD = scripting/bindings/libsqbindings.la \
 			$(SCINTILLA_GTK_LIBS) \
 			$(CB_GLIB2_LIBS) \
 			$(CB_TINYXML_LIBS) \
-			$(CB_SQUIRREL_LIBS)
+			$(CB_SQUIRREL_LIBS) \
+			$(CB_NSPR_LIBS)
 
 if CODEBLOCKS_NT
 libcodeblocks_la_LIBADD += -lgdi32
