From b0dcc6f6cd404d28836a852e53872200929efb7d Mon Sep 17 00:00:00 2001
From: Stacy Harper <contact@stacyharper.net>
Date: Sat, 15 Jan 2022 17:39:47 +0100
Subject: [PATCH] Fix printf using size_t lineno variable

Signed-off-by: Stacy Harper <contact@stacyharper.net>
---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 9ea3f0b..b524e73 100644
--- a/main.c
+++ b/main.c
@@ -1004,7 +1004,7 @@ static int load_config(const char *config_path) {
 			parse_idlehint(p.we_wordc, p.we_wordv);
 		} else {
 			line[i] = 0;
-			swayidle_log(LOG_ERROR, "Unexpected keyword \"%s\" in line %lu", line, lineno);
+			swayidle_log(LOG_ERROR, "Unexpected keyword \"%s\" in line %zu", line, lineno);
 			free(line);
 			return -EINVAL;
 		}
-- 
2.34.1

