Patch-Source: https://github.com/jbeder/yaml-cpp/pull/1374
---
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 693139f..5f97ca9 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -43,15 +43,9 @@ target_compile_options(yaml-cpp-tests
 target_link_libraries(yaml-cpp-tests
   PRIVATE
     Threads::Threads
-    yaml-cpp
-    gtest
-    gmock)
-
-set_property(TARGET yaml-cpp-tests PROPERTY CXX_STANDARD_REQUIRED ON)
-if (NOT DEFINED CMAKE_CXX_STANDARD)
-  set_target_properties(yaml-cpp-tests PROPERTIES CXX_STANDARD 11)
-endif()
-
+    yaml-cpp::yaml-cpp
+    GTest::gtest
+    GTest::gmock)
 
 add_test(NAME yaml-cpp::test COMMAND yaml-cpp-tests)
 
