From 93ef480062d1db7da02696dfe60793362ab0b2b3 Mon Sep 17 00:00:00 2001
From: Andy Postnikov <apostnikov@gmail.com>
Date: Tue, 21 Jan 2025 22:31:02 +0100
Subject: [PATCH] Fix int-bool usage for s390x

Follow-up #156
---
 ext/otel_observer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/otel_observer.c b/ext/otel_observer.c
index 4acc94c..e5f9b26 100644
--- a/otel_observer.c
+++ b/otel_observer.c
@@ -976,7 +976,7 @@ static otel_observer *resolve_observer(zend_execute_data *execute_data) {
     bool has_withspan_attribute = func_has_withspan_attribute(execute_data);
 
     if (OTEL_G(attr_hooks_enabled) == false && has_withspan_attribute &&
-        OTEL_G(display_warnings) == true) {
+        OTEL_G(display_warnings)) {
         php_error_docref(NULL, E_CORE_WARNING,
                          "OpenTelemetry: WithSpan attribute found but "
                          "attribute hooks disabled");
