diff --git a/hack/lib/test.sh b/hack/lib/test.sh
index 36ea5f04..045de151 100644
--- a/hack/lib/test.sh
+++ b/hack/lib/test.sh
@@ -78,12 +78,12 @@ kube::test::object_assert() {
   local object=$2
   local request=$3
   local expected=$4
-  local args=${5:-}
+  local get_args=${5:-}
 
   for j in $(seq 1 "${tries}"); do
     # shellcheck disable=SC2086
     # Disabling because to allow for expansion here
-    res=$(kubectl get "${kube_flags[@]}" ${args} ${object} -o go-template="${request}")
+	res=$(eval kubectl ${args} get "${kube_flags[@]}" ${get_args} "${object}" -o go-template=\""${request}"\")
     if [[ "${res}" =~ ^$expected$ ]]; then
         echo -n "${green}"
         echo "$(kube::test::get_caller 3): Successful get ${object} ${request}: ${res}"
