diff -rupN a/setup.py b/setup.py
--- a/setup.py	2017-08-14 01:35:04.000000000 +0000
+++ b/setup.py	2026-03-25 05:45:51.360000000 +0000
@@ -26,9 +26,8 @@ params = dict(
 	description=description or name,
 	long_description=long_description,
 	url="https://github.com/yougov/" + name,
-	packages=setuptools.find_packages(),
+	packages=["yg.lockfile"],
 	include_package_data=True,
-	namespace_packages=name.split('.')[:-1],
 	python_requires='>=2.7',
 	install_requires=[
 		'zc.lockfile',
diff -rupN a/yg/__init__.py b/yg/__init__.py
--- a/yg/__init__.py	2017-08-14 01:35:04.000000000 +0000
+++ b/yg/__init__.py	2026-03-25 05:17:00.170000000 +0000
@@ -1 +0,0 @@
-__import__("pkg_resources").declare_namespace(__name__)
