Ref https://gitlab.gnome.org/GNOME/glycin/-/merge_requests/380

--- a/glycin/src/sandbox.rs
+++ b/glycin/src/sandbox.rs
@@ -748,6 +748,7 @@
                     // Wording of an old Debian patch
                     "No permissions to create new namespace",
                     "bwrap: setting up uid map: Permission denied",
+                    "bwrap: Failed to make / slave: Permission denied",
                 ]
                 .iter()
                 .any(|y| x.contains(y))
