diff --git a/delfin/meson.build b/delfin/meson.build
index ef37520..fa88f52 100644
--- a/delfin/meson.build
+++ b/delfin/meson.build
@@ -21,10 +21,11 @@ env.set('MESON_BUILD_ROOT', meson.project_build_root())
 env.set('FLATPAK', get_option('flatpak') ? 'true' : 'false')
 
 cargo = find_program('cargo', required: true)
+cargo_auditable = find_program('cargo-auditable', required: true)
 clang = find_program('clang', required: false)
 mold = find_program('mold', required: false)
 
-cargo_options = ['--target-dir', meson.project_build_root() / 'delfin']
+cargo_options = ['--target-dir', meson.project_build_root() / 'delfin', '--frozen']
 rust_target = 'debug'
 
 if get_option('profile') == 'release'
@@ -45,6 +46,7 @@ cargo_build = custom_target(
   env: env,
   command: [
     cargo,
+    'auditable',
     'build',
     cargo_options,
   ],
