diff --git a/delfin/meson.build b/delfin/meson.build
index 61283cd..0af6cd9 100644
--- a/delfin/meson.build
+++ b/delfin/meson.build
@@ -17,6 +17,7 @@ run_command(
 )
 
 cargo = find_program('cargo', required: true)
+cargo_auditable = find_program('cargo-auditable', required: true)
 
-cargo_options = ['--target-dir', meson.project_build_root() / 'delfin']
+cargo_options = ['--target-dir', meson.project_build_root() / 'delfin', '--frozen']
 rust_target = 'debug'
 
 if get_option('profile') == 'release'
@@ -33,7 +33,7 @@ custom_target(
   output: meson.project_name(),
   console: true,
   command: [
-    cargo, 'build', cargo_options,
+    cargo, 'auditable', 'build', cargo_options,
     '&&', 'cp', 'delfin' / rust_target / meson.project_name(), '@OUTPUT@',
   ],
   install: true,
