From 70089a405c006c6cae4480bd0307954983c48733 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Wed, 14 Feb 2018 03:30:50 +0000
Subject: [PATCH] build: use absolute path for LD_LIBRARY_PATH when testing

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cc02c35..b524a2b 100644
--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,7 @@ install: all
 	ln -sf ${LIBUCONTEXT_SONAME} ${DESTDIR}/lib/${LIBUCONTEXT_NAME}
 
 check: test_libucontext ${LIBUCONTEXT_SONAME}
-	env LD_LIBRARY_PATH=. ./test_libucontext
+	env LD_LIBRARY_PATH=$(shell pwd) ./test_libucontext
 
 test_libucontext: test_libucontext.c ${LIBUCONTEXT_NAME}
 	$(CC) -std=c99 -D_BSD_SOURCE ${CFLAGS} ${CPPFLAGS} $@.c -o $@ ${LIBUCONTEXT_NAME}
-- 
2.16.1

