From 0143bbac98dfd55c4a42652bd1cce8d35047a861 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Wed, 14 Feb 2018 03:37:25 +0000
Subject: [PATCH 2/2] x86: trampoline: kill textrel

---
 arch/x86/startcontext.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/startcontext.S b/arch/x86/startcontext.S
index 165804d..2bb87ee 100644
--- a/arch/x86/startcontext.S
+++ b/arch/x86/startcontext.S
@@ -18,7 +18,7 @@ __start_context:
 	je	hosed
 
 	/* call setcontext to switch to the linked context */
-	call	__setcontext
+	call	__setcontext@plt
 	movl	%eax, (%esp)
 
 hosed:
-- 
2.16.1

