diff --git a/initscript/gentoo/avahi-daemon.in b/initscript/gentoo/avahi-daemon.in
index f12d0e4..c038c75 100644
--- a/initscript/gentoo/avahi-daemon.in
+++ b/initscript/gentoo/avahi-daemon.in
@@ -1,4 +1,4 @@
-#!/sbin/runscript
+#!/sbin/openrc-run
 # Copyright 1999-2007 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
diff --git a/initscript/gentoo/avahi-dnsconfd.in b/initscript/gentoo/avahi-dnsconfd.in
index dff914c..48f3b35 100644
--- a/initscript/gentoo/avahi-dnsconfd.in
+++ b/initscript/gentoo/avahi-dnsconfd.in
@@ -1,4 +1,4 @@
-#!/sbin/runscript
+#!/sbin/openrc-run
 # Copyright 1999-2007 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
