diff --git a/client/CMakeLists.txt b/client/CMakeLists.txt
index 836f814..7047365 100644
--- a/client/CMakeLists.txt
+++ b/client/CMakeLists.txt
@@ -68,7 +68,6 @@ add_compile_options(
   "-Wno-unused-parameter"
   "$<$<COMPILE_LANGUAGE:C>:-Wstrict-prototypes>"
   "$<$<C_COMPILER_ID:GNU>:-Wimplicit-fallthrough=2>"
-  "-Werror"
   "-Wfatal-errors"
   "-ffast-math"
   "-fdata-sections"
diff --git a/obs/CMakeLists.txt b/obs/CMakeLists.txt
index 0491e65..60b37ff 100644
--- a/obs/CMakeLists.txt
+++ b/obs/CMakeLists.txt
@@ -18,7 +18,6 @@ add_feature_info(ENABLE_BACKTRACE ENABLE_BACKTRACE "Backtrace support.")
 
 add_compile_options(
   "-Wall"
-  "-Werror"
   "-Wfatal-errors"
   "-ffast-math"
   "-fdata-sections"
