diff --git a/etc/Makefile.in b/etc/Makefile.in
index c5f647a..22aa277 100644
--- a/etc/Makefile.in
+++ b/etc/Makefile.in
@@ -56,7 +56,7 @@ DIALRULES=dialrules \
	dialrules.ext \
	dialrules.world \
	dialrules.uk
-TARGETS	= config.fax hosts.hfaxd xferfaxlog lockname ondelay
+TARGETS	= config.fax hosts.hfaxd lockname ondelay

 default all::
	@${MAKE} targets
@@ -68,8 +68,6 @@ targets: ${TARGETS}
 hosts.hfaxd:
	${ECHO} "localhost" >hosts.hfaxd
	${ECHO} "127.0.0.1" >>hosts.hfaxd
-xferfaxlog:
-	${ECHO} >xferfaxlog
 config.fax:
	${ECHO} on >config.fax
 lockname: ${SRCDIR}/lockname.c
@@ -89,7 +87,6 @@ install: default
		-src ${SRCDIR}/faxsetup.$$i -O faxsetup.$$i; \
	done
	${PUTSERVER} -root ${INSTALLROOT} -F ${SBIN} -m 755 -src probemodem.sh -O probemodem
-	${PUTSAMPLE} -root ${INSTALLROOT} -F ${SPOOL}/etc -m 644 -o xferfaxlog
	${PUTSAMPLE} -root ${INSTALLROOT} -F ${SPOOL}/etc -m 600 -o hosts.hfaxd
	for i in ${DIALRULES}; do \
	    ${PUTSERVER} -root ${INSTALLROOT} -F ${SPOOL}/etc -m 444 -src ${SRCDIR}/$$i \
@@ -108,7 +105,6 @@ uninstall::
		${RM} -f ${SBIN}/faxsetup.$$i; \
	done
	${RM} -f ${SBIN}/probemodem
-	${RM} -f ${SPOOL}/etc/xferfaxlog
	${RM} -f ${SPOOL}/etc/hosts.hfaxd
	for i in ${DIALRULES}; do \
	    ${RM} -f ${SPOOL}/etc/$$i; \
