--- a/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp
+++ b/gsoap/VisualStudio2005/wsdl2h/wsdl2h/stdsoap2.cpp
@@ -58,7 +58,7 @@
 # pragma convert(819)   /* EBCDIC to ASCII */
 #endif
 
-#if defined(__gnu_linux__) && !defined(_GNU_SOURCE)
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 # define _GNU_SOURCE 1
 #endif
 
--- a/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp
+++ b/gsoap/samples/calc_vs2005/calc_vs2005/stdsoap2.cpp
@@ -58,7 +58,7 @@
 # pragma convert(819)   /* EBCDIC to ASCII */
 #endif
 
-#if defined(__gnu_linux__) && !defined(_GNU_SOURCE)
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 # define _GNU_SOURCE 1
 #endif
 
--- a/gsoap/stdsoap2.c
+++ b/gsoap/stdsoap2.c
@@ -58,7 +58,7 @@
 # pragma convert(819)   /* EBCDIC to ASCII */
 #endif
 
-#if defined(__gnu_linux__) && !defined(_GNU_SOURCE)
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 # define _GNU_SOURCE 1
 #endif
 
--- a/gsoap/stdsoap2.cpp
+++ b/gsoap/stdsoap2.cpp
@@ -58,7 +58,7 @@
 # pragma convert(819)   /* EBCDIC to ASCII */
 #endif
 
-#if defined(__gnu_linux__) && !defined(_GNU_SOURCE)
+#if defined(__linux__) && !defined(_GNU_SOURCE)
 # define _GNU_SOURCE 1
 #endif
 
