diff --git a/src/dep_logic/tags/platform.py b/src/dep_logic/tags/platform.py
index 7772867..b39492d 100644
--- a/src/dep_logic/tags/platform.py
+++ b/src/dep_logic/tags/platform.py
@@ -292,12 +292,14 @@ class Arch(Enum):
     Aarch64 = "aarch64"
     Armv6L = "armv6l"
     Armv7L = "armv7l"
+    Armv8L = "armv8l"
     Powerpc64Le = "ppc64le"
     Powerpc64 = "ppc64"
     X86 = "x86"
     X86_64 = "x86_64"
     S390X = "s390x"
     RISCV64 = "riscv64"
+    Loongarch64 = "loongarch64"
 
     def __str__(self) -> str:
         return self.value
@@ -306,6 +308,7 @@ class Arch(Enum):
         if self in [
             Arch.Aarch64,
             Arch.Armv7L,
+            Arch.Armv8L,
             Arch.Powerpc64,
             Arch.Powerpc64Le,
             Arch.S390X,
