diff --git a/src/dep_logic/tags/platform.py b/src/dep_logic/tags/platform.py
index 0d7c336..e4d3f71 100644
--- a/src/dep_logic/tags/platform.py
+++ b/src/dep_logic/tags/platform.py
@@ -331,6 +331,7 @@ class Arch(Enum):
     Aarch64 = "aarch64"
     Armv6L = "armv6l"
     Armv7L = "armv7l"
+    Armv8L = "armv8l"
     Powerpc64Le = "ppc64le"
     Powerpc64 = "ppc64"
     X86 = "x86"
@@ -346,6 +347,7 @@ class Arch(Enum):
         if self in [
             Arch.Aarch64,
             Arch.Armv7L,
+            Arch.Armv8L,
             Arch.Powerpc64,
             Arch.Powerpc64Le,
             Arch.S390X,
