--- ./applets/notification_area/Makefile.in.orig
+++ ./applets/notification_area/Makefile.in
@@ -403,17 +403,17 @@
 
 NOTIFICATION_AREA_SOURCES = main.c
 NOTIFICATION_AREA_LDADD = \
+	libtray.la					\
 	../../libpanel-applet/libpanel-applet-3.la	\
 	$(X_LIBS)					\
 	$(NOTIFICATION_AREA_LIBS)			\
-	$(LIBPANEL_APPLET_LIBS)				\
-	libtray.la
+	$(LIBPANEL_APPLET_LIBS)
 
 testtray_SOURCES = testtray.c
 testtray_LDADD = \
+	libtray.la 			\
 	$(X_LIBS)			\
-	$(NOTIFICATION_AREA_LIBS)	\
-	libtray.la
+	$(NOTIFICATION_AREA_LIBS)
 
 @NOTIFICATION_AREA_INPROCESS_FALSE@APPLET_IN_PROCESS = false
 @NOTIFICATION_AREA_INPROCESS_TRUE@APPLET_IN_PROCESS = true
