From 939338c137f69a818c42b9654acb5a4edbd3981b Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Fri, 6 May 2022 10:00:06 +0200
Subject: [PATCH] cmake fixes

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 82a58a1..af27a3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -196,6 +196,7 @@ set(OUTPUT_DEBUG Debug/bin)
 set(OUTPUT_RELEASE Release/bin)
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/bin )
 
+if(false)
 if(CMAKE_BUILD_TYPE)
     # Validate build type
     set(CONFIGURATION_TYPES "Debug;Release;RelWithDebInfo")
@@ -208,6 +209,7 @@ else(NOT CMAKE_BUILD_TYPE)
     set(CMAKE_BUILD_TYPE "Release")
     message(STATUS "Build type not specified: Use Release by default.")
 endif(CMAKE_BUILD_TYPE)
+endif()
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/ispcrt/cmake)
 include(${CMAKE_CURRENT_SOURCE_DIR}/cmake/FixWindowsPath.cmake)
-- 
2.36.0

