From e90ed08643fde558f6974e56b085c7a2b45bd563 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Mon, 23 Jan 2023 12:40:04 -0800
Subject: [PATCH] dbus: allow setting filter-A and filter-AAAA options

---
 dbus/DBus-interface |  8 ++++++++
 src/dbus.c          | 14 ++++++++++++++
 2 files changed, 22 insertions(+)

diff --git a/dbus/DBus-interface b/dbus/DBus-interface
index 59b41b1..00b4465 100644
--- a/dbus/DBus-interface
+++ b/dbus/DBus-interface
@@ -44,6 +44,14 @@ SetFilterWin2KOption
 --------------------
 Takes boolean, sets or resets the --filterwin2k option.
 
+SetFilterA
+------------------------
+Takes boolean, sets or resets the --filter-A option.
+
+SetFilterAAAA
+------------------------
+Takes boolean, sets or resets the --filter-AAAA option.
+
 SetBogusPrivOption
 ------------------
 Takes boolean, sets or resets the --bogus-priv option.
diff --git a/src/dbus.c b/src/dbus.c
index fd5d1ca..be2c506 100644
--- a/src/dbus.c
+++ b/src/dbus.c
@@ -52,6 +52,12 @@ const char* introspection_xml_template =
 "    <method name=\"SetFilterWin2KOption\">\n"
 "      <arg name=\"filterwin2k\" direction=\"in\" type=\"b\"/>\n"
 "    </method>\n"
+"    <method name=\"SetFilterA\">\n"
+"      <arg name=\"filter-a\" direction=\"in\" type=\"b\"/>\n"
+"    </method>\n"
+"    <method name=\"SetFilterAAAA\">\n"
+"      <arg name=\"filter-aaaa\" direction=\"in\" type=\"b\"/>\n"
+"    </method>\n"
 "    <method name=\"SetLocaliseQueriesOption\">\n"
 "      <arg name=\"localise-queries\" direction=\"in\" type=\"b\"/>\n"
 "    </method>\n"
@@ -817,6 +823,14 @@ DBusHandlerResult message_handler(DBusConnection *connection,
     {
       reply = dbus_set_bool(message, OPT_FILTER, "filterwin2k");
     }
+  else if (strcmp(method, "SetFilterA") == 0)
+    {
+      reply = dbus_set_bool(message, OPT_FILTER_A, "filter-a");
+    }
+  else if (strcmp(method, "SetFilterAAAA") == 0)
+    {
+      reply = dbus_set_bool(message, OPT_FILTER_AAAA, "filter-aaaa");
+    }
   else if (strcmp(method, "SetLocaliseQueriesOption") == 0)
     {
       reply = dbus_set_bool(message, OPT_LOCALISE, "localise-queries");
-- 
2.39.1

