--- a/dnsmasq.conf.example	2016-08-02 11:30:24.207034952 +0200
+++ b/dnsmasq.conf.example	2016-08-02 11:30:34.094695721 +0200
@@ -671,7 +671,7 @@
 #conf-dir=/etc/dnsmasq.d,.bak
 
 # Include all files in a directory which end in .conf
-#conf-dir=/etc/dnsmasq.d/,*.conf
+conf-dir=/etc/dnsmasq.d/,*.conf
 
 # If a DHCP client claims that its name is "wpad", ignore that.
 # This fixes a security hole. see CERT Vulnerability VU#598349
