--- ./gas/configure.orig
+++ ./gas/configure
@@ -3221,7 +3221,7 @@
 fi
 
 
-BFD_VERSION=`${srcdir}/../bfd/configure --version | sed -n -e '1s,.* ,,p'`
+BFD_VERSION=`sed -n 's/AC_INIT(\[bfd\], *\[\([0-9.]*\)\])/\1/p' <${srcdir}/../bfd/configure.in`
 am__api_version='1.11'
 
 # Find a good install program.  We prefer a C program (faster),
--- ./gprof/configure.orig
+++ ./gprof/configure
@@ -3147,7 +3147,7 @@
 fi
 
 
-BFD_VERSION=`${srcdir}/../bfd/configure --version | sed -n -e '1s,.* ,,p'`
+BFD_VERSION=`sed -n 's/AC_INIT(\[bfd\], *\[\([0-9.]*\)\])/\1/p' <${srcdir}/../bfd/configure.in`
 am__api_version='1.11'
 
 # Find a good install program.  We prefer a C program (faster),
--- ./binutils/configure.orig
+++ ./binutils/configure
@@ -3238,7 +3238,7 @@
 fi
 
 
-BFD_VERSION=`${srcdir}/../bfd/configure --version | sed -n -e '1s,.* ,,p'`
+BFD_VERSION=`sed -n 's/AC_INIT(\[bfd\], *\[\([0-9.]*\)\])/\1/p' <${srcdir}/../bfd/configure.in`
 am__api_version='1.11'
 
 # Find a good install program.  We prefer a C program (faster),
--- ./ld/configure.orig
+++ ./ld/configure
@@ -3556,7 +3556,7 @@
 fi
 
 
-BFD_VERSION=`${srcdir}/../bfd/configure --version | sed -n -e '1s,.* ,,p'`
+BFD_VERSION=`sed -n 's/AC_INIT(\[bfd\], *\[\([0-9.]*\)\])/\1/p' <${srcdir}/../bfd/configure.in`
 am__api_version='1.11'
 
 # Find a good install program.  We prefer a C program (faster),
