diff -upr qt-wayland-shell-helpers-0.1.1.orig/src/library/meson.build qt-wayland-shell-helpers-0.1.1/src/library/meson.build
--- qt-wayland-shell-helpers-0.1.1.orig/src/library/meson.build	2021-07-09 14:55:47.174112496 +0200
+++ qt-wayland-shell-helpers-0.1.1/src/library/meson.build	2021-07-09 14:57:24.824233259 +0200
@@ -1,4 +1,4 @@
-private_module_includes = '/usr/include/@0@/@1@/@0@'.format('QtGui', qt.version())
+private_module_includes = '/usr/include/qt5/@0@/@1@/@0@'.format('QtGui', qt.version())
 
 qt_private = include_directories(private_module_includes)
 
