diff -up net-snmp-5.7.3/perl/ASN/Makefile.PL.orig net-snmp-5.7.3/perl/ASN/Makefile.PL
--- net-snmp-5.7.3/perl/ASN/Makefile.PL.orig	2016-05-06 10:35:30.005905040 +0200
+++ net-snmp-5.7.3/perl/ASN/Makefile.PL	2016-05-06 10:35:57.518776400 +0200
@@ -3,6 +3,7 @@ require 5;
 use Config;
 use Getopt::Long;
 my $lib_version;
+my %MakeParams = ();
 
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
diff -up net-snmp-5.7.3/perl/Makefile.PL.orig net-snmp-5.7.3/perl/Makefile.PL
--- net-snmp-5.7.3/perl/Makefile.PL.orig	2016-05-06 10:34:38.975143641 +0200
+++ net-snmp-5.7.3/perl/Makefile.PL	2016-05-06 10:35:13.326983024 +0200
@@ -2,6 +2,7 @@ use ExtUtils::MakeMaker;
 use Config;
 use Getopt::Long;
 require 5;
+my %MakeParams = ();
 
 %MakeParams = InitMakeParams();
 
