diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 9df03eb..040e184 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -111,7 +111,6 @@ endif()
 
 add_test_macro(ExecuteCoreModules dummy)
 add_test_macro(ExecuteKDEModules dummy)
-add_test_macro(KDEFetchTranslations dummy)
 
 add_test_macro(KDEInstallDirsTest.relative_or_absolute dummy)
 add_test_variant(KDEInstallDirsTest.vars_in_sync_no_args

