--- ./Tools/qmake/mkspecs/features/unix/default_post.prf.orig	2013-10-10 13:50:38.663745911 +0000
+++ ./Tools/qmake/mkspecs/features/unix/default_post.prf	2013-10-10 13:50:53.253909607 +0000
@@ -32,8 +32,6 @@
     QMAKE_OBJECTIVE_CFLAGS += -Wno-c++11-extensions -Wno-c++0x-extensions
 }
 
-contains(TEMPLATE, app): CONFIG += rpath
-
 isEqual(QT_ARCH,i386):CONFIG(debug, debug|release) {
   # Make ld not cache the symbol tables of input files in memory to avoid memory exhaustion during the linking phase.
   config_gnuld: QMAKE_LFLAGS += -Wl,--no-keep-memory
