diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5974b5c..cf91527 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -211,12 +211,7 @@ set(PKG_CONFIG_LIBS "-L\${libdir} -llwan ${ADDITIONAL_LIBRARIES_STR}")
 unset(ADDITIONAL_LIBRARIES_STR)
 set(PKG_CONFIG_CFLAGS "-I\${includedir}")
 
-execute_process(
-	COMMAND git log -1 --format=%h
-	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-	OUTPUT_VARIABLE PROJECT_VERSION
-	OUTPUT_STRIP_TRAILING_WHITESPACE
-)
+set(PROJECT_VERSION $ENV{VERSION})
 configure_file(
 	"${CMAKE_CURRENT_SOURCE_DIR}/src/cmake/lwan.pc.cmake"
 	"${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.pc"
