Upstream-Issue: https://github.com/lpereira/lwan/issues/219

--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -90,19 +90,7 @@
 
 set(ADDITIONAL_LIBRARIES ${ADDITIONAL_LIBRARIES} -lm PARENT_SCOPE)
 
-
-get_property(LIB64 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS)
-if ("${LIB64}" STREQUAL "TRUE")
-       set(LIBSUFFIX 64)
-else ()
-       get_property(LIB32 GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS)
-       if ("${LIB32}" STREQUAL "TRUE")
-	       set(LIBSUFFIX 32)
-       else ()
-	       set(LIBSUFFIX "")
-       endif ()
-endif ()
-set(INSTALL_LIB_DIR lib${LIBSUFFIX} CACHE PATH "Installation directory for libraries")
+set(INSTALL_LIB_DIR lib CACHE PATH "Installation directory for libraries")
 mark_as_advanced(INSTALL_LIB_DIR)
 message(STATUS "Library install directory: ${CMAKE_INSTALL_PREFIX}/${INSTALL_LIB_DIR}")
 
