--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ TAR ?= tar
 BISON ?= bison
 FLEX ?= flex
 
-GCC_WARNINGS1=-Wall -Wextra -Werror -Wpointer-arith -Wstrict-prototypes
+GCC_WARNINGS1=-Wall -Wextra -Wpointer-arith -Wstrict-prototypes
 GCC_WARNINGS3=-Wno-format-zero-length
 GCC_WARNINGS=$(GCC_WARNINGS1) $(GCC_WARNINGS2) $(GCC_WARNINGS3)
 CFLAGS += $(GCC_WARNINGS)
@@ -70,7 +70,7 @@ CPPFLAGS += -DTHREADS
 # CFLAGS += -pg
 # Warning: Using -O3 has been seen to cause core dumps on repositories with
 # very long revision names - some bounds check gets optimized away. Don't do that.
-CFLAGS += -O2
+# CFLAGS += -O2
 # If your toolchain supports link time optimization this is a cheap speedup
 # CFLAGS += -flto
 # Bounds checks at runtime. Not enabled by default for performance reasons
