--- a/cvsconvert
+++ b/cvsconvert
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 """
 cvsconvert - convert a CVS repo and check against the original
 
--- a/cvsreduce
+++ b/cvsreduce
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # Runs under both Python 2 and Python 3: preserve this property!
 # SPDX-License-Identifier: GPL-2.0+
 """
--- a/tests/at.tst
+++ b/tests/at.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # -*- coding: latin-1 -*-
 ## Verify parsing of escaped at on final line
 
--- a/tests/basic.tst
+++ b/tests/basic.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## basic test for CVS master parsing
 
 import sys, testlifter
--- a/tests/branchy.tst
+++ b/tests/branchy.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## A branchy repo with deletions and only valid tags
 
 import sys, testlifter
--- a/tests/daughterbranch.tst
+++ b/tests/daughterbranch.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Test for the daughter-branch bug
 
 # This was the description:
--- a/tests/exec.tst
+++ b/tests/exec.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Test handling of executable bit
 
 import sys, testlifter
--- a/tests/expand.tst
+++ b/tests/expand.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Test keyword expansion
 
 import sys, testlifter
--- a/tests/hack1.tst
+++ b/tests/hack1.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## First example from the Hacking Guide
 
 import sys, testlifter
--- a/tests/hack2.tst
+++ b/tests/hack2.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Second example from the Hacking Guide
 
 import sys, testlifter
--- a/tests/hack3.tst
+++ b/tests/hack3.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Third example from the Hacking Guide
 
 import sys, testlifter
--- a/tests/linear.tst
+++ b/tests/linear.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## simplest possible linear repository with multiple commits
 
 import sys, testlifter
--- a/tests/longrev.tst
+++ b/tests/longrev.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## A widely branched repo with long file revision strings.
 
 import sys, testlifter
--- a/tests/postbranch.tst
+++ b/tests/postbranch.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Ilya Basin's test, failed by cvsps-3.x
 """
 Date: Sat, 20 Apr 2013 14:38:55 +0400
--- a/tests/t9601.py
+++ b/tests/t9601.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Test handling of vendor branches
 #
 # This test was swiped from the git 1.8.1 tree, then modified to exercise
--- a/tests/t9602.py
+++ b/tests/t9602.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Test handling of pathological tags
 #
 # This test was swiped from the git 1.8.1 tree, then modified to exercise
--- a/tests/t9603.py
+++ b/tests/t9603.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Testing for correct patchset estimation
 
 # Structure of the test cvs repository
--- a/tests/t9604.py
+++ b/tests/t9604.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Testing for correct timestamp handling in author maps.
 import sys, testlifter, tempfile, os
 
--- a/tests/t9605.py
+++ b/tests/t9605.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Testing for correct patchset estimation
 
 # Structure of the test cvs repository
--- a/tests/tagbug.tst
+++ b/tests/tagbug.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Tricky tag corner case
 
 import sys, testlifter
--- a/tests/twobranch.tst
+++ b/tests/twobranch.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## Two-branch repo to test incremental dumping
 
 import sys, testlifter
--- a/tests/twotag.tst
+++ b/tests/twotag.tst
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 ## A repo with identical tags attached to different changesets
 
 import sys, testlifter, time
--- a/cvssync
+++ b/cvssync
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # Runs under both Python 2 and Python 3: preserve this property!
 # SPDX-License-Identifier: GPL-2.0+
 """
