diff -up octave-4.0.0/liboctave/cruft/Faddeeva/Faddeeva.hh.gnulib octave-4.0.0/liboctave/cruft/Faddeeva/Faddeeva.hh
--- octave-4.0.0/liboctave/cruft/Faddeeva/Faddeeva.hh.gnulib	2015-05-23 08:21:53.000000000 -0600
+++ octave-4.0.0/liboctave/cruft/Faddeeva/Faddeeva.hh	2016-02-19 19:50:04.693811663 -0700
@@ -27,6 +27,7 @@
 #ifndef FADDEEVA_HH
 #define FADDEEVA_HH 1
 
+#include <math.h>
 #include <complex>
 
 namespace Faddeeva {
