--- octave-4.2.0/doc/interpreter/module.mk.orig
+++ octave-4.2.0/doc/interpreter/module.mk
@@ -250,12 +250,7 @@
 
 DOC_TARGETS += \
   $(srcdir)/doc/interpreter/octave.info \
-  doc/interpreter/doc-cache \
-  doc/interpreter/octave.ps \
-  doc/interpreter/octave.pdf \
-  $(OCTAVE_HTML_STAMP) \
-  $(HTMLDIR_IMAGES) \
-  $(HTMLDIR_CSS)
+  doc/interpreter/doc-cache
 
 ## Distribute both OCTAVE_CSS and HTMLDIR_CSS so that the rules for
 ## building HTMLDIR_CSS work properly.
