From 51b5bd7bc8bb7a410400667e00d0813603df32bd Mon Sep 17 00:00:00 2001
From: Bruno BELANYI <bruno@belanyi.fr>
Date: Fri, 23 Sep 2022 15:40:43 +0200
Subject: [PATCH] fix: update tests for Django 4 compatibility

---
 tests/urls.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/urls.py b/tests/urls.py
index c949612..9c4e361 100644
--- a/tests/urls.py
+++ b/tests/urls.py
@@ -1,7 +1,10 @@
 """URLs for django-annoying's tests"""
 from __future__ import absolute_import
 
-from django.conf.urls import url
+try:
+    from django.conf.urls import url
+except ImportError:
+    from django.urls import re_path as url
 
 from . import views
 
