Remove bundled zlib.
--- a/Makefile.common
+++ b/Makefile.common
@@ -24,8 +24,7 @@
 				-I$(CORE_DIR)/Src/VideoRender \
 				-I$(CORE_DIR)/Src/Z80 \
 				-I$(CORE_DIR)/Src/Input \
-				-I$(CORE_DIR)/Src/Libretro \
-				-I$(DEPS_DIR)/zlib
+				-I$(CORE_DIR)/Src/Libretro
 
 ifneq (,$(findstring msvc2003,$(platform)))
 INCFLAGS += -I$(LIBRETRO_COMM_DIR)/include/compat/msvc
@@ -332,18 +331,6 @@
 SOURCES_C += $(LIBRETRO_COMM_DIR)/file/retro_dirent.c \
 				 $(LIBRETRO_COMM_DIR)/compat/compat_snprintf.c \
 				 $(LIBRETRO_COMM_DIR)/compat/compat_strcasestr.c
-
-#include zlib
-SOURCES_C += $(DEPS_DIR)/zlib/adler32.c \
-				 $(DEPS_DIR)/zlib/compress.c \
-				 $(DEPS_DIR)/zlib/crc32.c \
-				 $(DEPS_DIR)/zlib/deflate.c \
-				 $(DEPS_DIR)/zlib/inffast.c \
-				 $(DEPS_DIR)/zlib/inflate.c \
-				 $(DEPS_DIR)/zlib/inftrees.c \
-				 $(DEPS_DIR)/zlib/trees.c \
-				 $(DEPS_DIR)/zlib/uncompr.c \
-				 $(DEPS_DIR)/zlib/zutil.c
 endif
 
 ifeq ($(HAVE_COMPAT), 1)
@@ -355,3 +342,5 @@
 	COREDEFINES  += -DHAVE_WIN32_MSX_MANAGER
 	SOURCES_CXX  += $(CORE_DIR)/Src/IoDevice/msxgr.cpp
 endif
+
+LIBS += -lz
