--- ./tools/clang/lib/Driver/ToolChains.cpp.add_gcc_vers	2011-03-21 22:29:27.000000000 +0100
+++ ./tools/clang/lib/Driver/ToolChains.cpp	2011-08-01 18:20:29.504194241 +0200
@@ -1449,7 +1449,8 @@
       GccTriple = "i586-suse-linux";
   }
 
-  const char* GccVersions[] = {"4.5.2", "4.5.1", "4.5", "4.4.5", "4.4.4",
+  const char* GccVersions[] = {"4.6.1", "4.6.0",
+                               "4.5.2", "4.5.1", "4.5", "4.4.5", "4.4.4",
                                "4.4.3", "4.4", "4.3.4", "4.3.3", "4.3.2",
                                "4.3", "4.2.4", "4.2.3", "4.2.2", "4.2.1",
                                "4.2"};
