--- a/lib/actions/test.bash
+++ a/lib/actions/test.bash
@@ -69,6 +69,7 @@
 
 	# Add the Rakudo bin directories from RSTAR_PREFIX
 	PATH+=":$(readlink -f "$RSTAR_PREFIX/bin")"
+	PATH+=":$(readlink -f "$RSTAR_PREFIX/vendor/bin")"
 	PATH+=":$(readlink -f "$RSTAR_PREFIX/share/perl6/site/bin")"
 	PATH+=":$(readlink -f "$RSTAR_PREFIX/share/perl6/vendor/bin")"
 	PATH+=":$(readlink -f "$RSTAR_PREFIX/share/perl6/core/bin")"
@@ -84,7 +85,7 @@
 	while read -r module
 	do
 		chgdir "$BASEDIR/src/rakudo-star-modules/$module"
-		prove6 -v . && continue
+		RAKUDO_PREFIX="$RSTAR_PREFIX" prove6 -v . && continue
 
 		failures+=("$module")
 	done < "$modules"
