--- a/src/rakudo-star-modules/rakudoc/Build.pm6
+++ b/src/rakudo-star-modules/rakudoc/Build.pm6
@@ -4,7 +4,7 @@
 class Build {
     method build($workdir) {
         my $doc-dir   = $workdir.IO.child('doc');
-        my $dest-pref = $*REPO.repo-chain.grep(/site/).first.prefix.child("doc");
+        my $dest-pref = %*ENV<RSTAR_PREFIX>.IO.child('site').child('doc');
         mkdir($dest-pref) unless $dest-pref.d;
 
         my @files = find(dir => "$workdir/doc", type => 'file').list;
