--- a/lib/install-module.raku
+++ b/lib/install-module.raku
@@ -20,7 +20,11 @@
 		default { $_.say; exit 1; }
 	}
 
-	my $repository = CompUnit::RepositoryRegistry.repository-for-name($repo);
+	my $repository;
+	{
+		temp %*ENV<RAKUDO_PREFIX> = %*ENV<RSTAR_PREFIX>;
+		$repository = CompUnit::RepositoryRegistry.repository-for-name($repo);
+	}
 	my $meta-file = $path.add('META6.json');
 	my $dist = Distribution::Path.new($path, :$meta-file);
 
