x[strlen(x)] does not get the last char.
--
--- a/src/xheader.c
+++ b/src/xheader.c
@@ -561,7 +561,7 @@ locate_handler (char const *keyword)
     if (p->prefix)
       {
 	size_t kwlen = strlen (p->keyword);
-	if (strncmp (p->keyword, keyword, kwlen) == 0 && keyword[kwlen] == '.')
+	if (strncmp (p->keyword, keyword, kwlen) == 0 && keyword[kwlen - 1] == '.')
           return p;
       }
     else
