diff --git a/sni/statusnotifierbutton.cpp b/sni/statusnotifierbutton.cpp
index d07eb6b0163fc33db8e4e03731df7ee31ce671b2..c339657d538ded7544520386a31e01237e423304 100644
--- a/sni/statusnotifierbutton.cpp
+++ b/sni/statusnotifierbutton.cpp
@@ -267,7 +267,7 @@ void StatusNotifierButton::mouseReleaseEvent(QMouseEvent *event)
 {
     if (event->button() == Qt::LeftButton)
         interface->Activate(QCursor::pos().x(), QCursor::pos().y());
-    else if (event->button() == Qt::MidButton)
+    else if (event->button() == Qt::MiddleButton)
         interface->SecondaryActivate(QCursor::pos().x(), QCursor::pos().y());
     else if (Qt::RightButton == event->button())
     {
