diff --git a/sim/arm/armdefs.h b/sim/arm/armdefs.h
index d87220bf29b..016c18e189b 100644
--- a/sim/arm/armdefs.h
+++ b/sim/arm/armdefs.h
@@ -264,7 +264,7 @@ extern ARMword ARMul_DoInstr (ARMul_State * state);
 \***************************************************************************/
 
 extern void ARMul_ScheduleEvent (ARMul_State * state, unsigned long delay,
-				 unsigned (*func) ());
+				 unsigned (*func) (ARMul_State *));
 extern void ARMul_EnvokeEvent (ARMul_State * state);
 extern unsigned long ARMul_Time (ARMul_State * state);
 
--- a/sim/sh/interp.c
+++ n/sim/sh/interp.c
@@ -1712,7 +1712,7 @@
   register int endianw = global_endianw;
 
   int tick_start = get_now ();
-  void (*prev_fpe) ();
+  void (*prev_fpe) (int);
 
   register unsigned short *jump_table = sh_jump_table;
 
--- a/sim/d10v/gencode.c
+++ b/sim/d10v/gencode.c
@@ -150,5 +150,5 @@
 	  printf ("},\n");
 	}
     }
-  printf ("{ 0,0,0,0,0,0,0,(void (*)())0,0,{0,0,0}},\n};\n");
+  printf ("{ 0,0,0,0,0,0,0,NULL,0,{0,0,0}},\n};\n");
 }
