diff --git a/gdb/gcore.in b/gdb/gcore.in
index 34860de..eaba70d 100644
--- a/gdb/gcore.in
+++ b/gdb/gcore.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/sh
 
 #   Copyright (C) 2003-2023 Free Software Foundation, Inc.
 
@@ -25,15 +25,14 @@ prefix=core
 
 # When the -a option is present, this may hold additional commands
 # to ensure gdb dumps all mappings (OS dependent).
-dump_all_cmds=()
+dump_all_cmds=""
 
 while getopts :ao: opt; do
     case "$opt" in
         a)
             case "$OSTYPE" in
                 linux*)
-                    dump_all_cmds=("-ex" "set use-coredump-filter off")
-                    dump_all_cmds+=("-ex" "set dump-excluded-mappings on")
+                    dump_all_cmds="-ex set use-coredump-filter off -ex set dump-excluded-mappings on"
                     ;;
             esac
             ;;
@@ -100,7 +99,7 @@ do
 	"$binary_path/@GDB_TRANSFORM_NAME@" </dev/null \
 	    --nx --batch --readnever -iex 'set debuginfod enabled off' \
 	    -ex "set pagination off" -ex "set height 0" -ex "set width 0" \
-	    "${dump_all_cmds[@]}" \
+	    $dump_all_cmds \
 	    -ex "attach $pid" -ex "gcore $prefix.$pid" -ex detach -ex quit
 
 	if [ -r "$prefix.$pid" ] ; then
