diff -u -r -N gdb-7.3.1-org//gdb/linux-thread-db.c gdb-7.3.1-patched//gdb/linux-thread-db.c
--- gdb-7.3.1-org//gdb/linux-thread-db.c	1970-01-01 00:00:00.000000000 +0000
+++ gdb-7.3.1-patched//gdb/linux-thread-db.c	2011-09-22 22:17:38.648841814 +0000
@@ -47,6 +47,8 @@
 #include <gnu/libc-version.h>
 #endif
 
+#include "threadpathhack.h"
+
 /* GNU/Linux libthread_db support.
 
    libthread_db is a library, provided along with libpthread.so, which
diff -u -r -N gdb-7.3.1-org//gdb/threadpathhack.h gdb-7.3.1-patched//gdb/threadpathhack.h
--- gdb-7.3.1-org//gdb/threadpathhack.h	1970-01-01 00:00:00.000000000 +0000
+++ gdb-7.3.1-patched//gdb/threadpathhack.h	2011-09-22 22:17:38.648841814 +0000
@@ -0,0 +1,12 @@
+#ifndef HACK_PATH_SO
+#define HACK_PATH_SO
+
+#ifndef LIBTHREAD_DB_SO
+#define LIBTHREAD_DB_SO "bug.so"
+#endif
+#ifndef LIBTHREAD_DB_SEARCH_PATH
+#define LIBTHREAD_DB_SEARCH_PATH "/lib"
+#endif
+
+#endif
+
