diff --git a/getopt.c b/getopt.c
index 0a24ce5..0bc26ce 100644
--- a/getopt.c
+++ b/getopt.c
@@ -211,7 +211,7 @@ static char *posixly_correct;
 /* Avoid depending on library functions or files
    whose names are inconsistent.  */
 
-char *getenv();
+char *getenv(const char *);
 
 static char *
      my_index(str, chr)
diff --git a/getopt.h b/getopt.h
index 2fa12f7..2e2d8c2 100644
--- a/getopt.h
+++ b/getopt.h
@@ -106,7 +106,7 @@ extern "C"
    errors, only prototype getopt for the GNU C library.  */
 	extern int getopt(int argc, char *const *argv, const char *shortopts);
 #else				/* not __GNU_LIBRARY__ */
-	extern int getopt();
+	extern int getopt(int argc, char *const *argv, const char *shortopts);
 #endif				/* __GNU_LIBRARY__ */
 	extern int getopt_long(int argc, char *const *argv, const char *shortopts,
 			       const struct option *longopts, int *longind);
diff --git a/mbswidth.c b/mbswidth.c
index 2162205..df2fa10 100644
--- a/mbswidth.c
+++ b/mbswidth.c
@@ -63,7 +63,7 @@
 #	warn "this configure-time declaration test was not run"
 #endif
 #if !HAVE_DECL_WCWIDTH
-int wcwidth ();
+int wcwidth (wchar_t);
 #endif
 
 #ifndef wcwidth
