diff -up abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.c.gcc5 abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.c
--- abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.c.gcc5	2014-11-28 16:29:52.000000000 +0100
+++ abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.c	2015-02-11 16:02:15.837789303 +0100
@@ -859,7 +859,7 @@ item_merge(list_item dest, list_item src
  */
 
 /* quick lookup by "standard" field number */
-inline int
+extern inline int
 field_id(int i)
 {
 	assert((i >= 0) && (i < ITEM_FIELDS));
diff -up abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.h.gcc5 abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.h
--- abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.h.gcc5	2014-11-28 16:29:52.000000000 +0100
+++ abook-git-6e550af8a907a680dc845c22a7d6c36d89e900c1/database.h	2015-02-11 16:02:23.465818915 +0100
@@ -61,7 +61,7 @@ struct db_enumerator {
 /*
  * Field operations
  */
-inline int field_id(int i);
+extern inline int field_id(int i);
 abook_field *find_standard_field(char *key, int do_declare);
 abook_field *real_find_field(char *key, abook_field_list *list, int *nb);
 #define find_field(key, list)		real_find_field(key, list, NULL)
