--- ./hatch_build.py.orig
+++ ./hatch_build.py
@@ -7,22 +7,11 @@
 
 class CustomBuildHook(BuildHookInterface):
     def initialize(self, version, build_data):
-        if shutil.which("deno"):
-            print("Building with deno...", flush=True)
-            os.environ["DENO_NO_UPDATE_CHECK"] = "1"
-            subprocess.run(["deno", "install", "--frozen"], check=True)
-            subprocess.run(["deno", "task", "bundle"], check=True)
-
-        elif shutil.which("bun"):
-            print("Building with bun...", flush=True)
-            subprocess.run(["bun", "install"], check=True)
-            subprocess.run(["bun", "--bun", "run", "bundle"], check=True)
-
-        elif shutil.which("npm"):
+        if shutil.which("npm"):
             print("Building with npm...", flush=True)
             # npm is a batch file (`npm.cmd`) on windows, which requires `shell=True`
             requires_shell = os.name == "nt"
-            subprocess.run(["npm", "install"], check=True, shell=requires_shell)
+            subprocess.run(["npm", "ci", "--ignore-scripts", "--omit=optional"], check=True, shell=requires_shell)
             subprocess.run(["npm", "run", "bundle"], check=True, shell=requires_shell)
 
         else:
