--- a/lib/clean-temp.c
+++ b/lib/clean-temp.c
@@ -231,7 +231,7 @@ create_temp_dir (const char *prefix, const char *parentdir,
   xtemplate = (char *) xmalloca (PATH_MAX);
   if (path_search (xtemplate, PATH_MAX, parentdir, prefix, parentdir == NULL))
     {
-      error (0, errno,
+      error (0, errno, "%s",
              _("cannot find a temporary directory, try setting $TMPDIR"));
       goto quit;
     }
--- a/lib/spawn-pipe.c
+++ b/lib/spawn-pipe.c
@@ -215,10 +215,10 @@ create_pipe (const char *progname,
        process is running.  */
   if (pipe_stdout)
     if (pipe2_safer (ifd, O_BINARY | O_CLOEXEC) < 0)
-      error (EXIT_FAILURE, errno, _("cannot create pipe"));
+      error (EXIT_FAILURE, errno, "%s", _("cannot create pipe"));
   if (pipe_stdin)
     if (pipe2_safer (ofd, O_BINARY | O_CLOEXEC) < 0)
-      error (EXIT_FAILURE, errno, _("cannot create pipe"));
+      error (EXIT_FAILURE, errno, "%s", _("cannot create pipe"));
 /* Data flow diagram:
  *
  *           write        system         read
