diff --git a/tests/test-posix_spawn_file_actions_addclose.c b/tests/test-posix_spawn_file_actions_addclose.c
index 2c910ea..b228a7d 100644
--- a/tests/test-posix_spawn_file_actions_addclose.c
+++ b/tests/test-posix_spawn_file_actions_addclose.c
@@ -54,13 +54,6 @@ main (void)
     errno = 0;
     ASSERT (posix_spawn_file_actions_addclose (&actions, -1) == EBADF);
   }
-  /* This behaviour is not mandated by POSIX, but happens to pass on all
-     platforms.  */
-  {
-    int bad_fd = big_fd ();
-    errno = 0;
-    ASSERT (posix_spawn_file_actions_addclose (&actions, bad_fd) == EBADF);
-  }
 
   posix_spawn_file_actions_destroy (&actions);
 
