--- a/src/NetworkManagerUtils.c	2020-07-17 20:05:45.382005791 +0200
+++ b/src/NetworkManagerUtils.c	2020-07-17 20:05:39.721463542 +0200
@@ -4,6 +4,7 @@
  * Copyright (C) 2005 - 2008 Novell, Inc.
  */
 
+#include <netinet/if_ether.h>
 #include "nm-default.h"
 
 #include "NetworkManagerUtils.h"
diff --git a/src/devices/nm-device.c b/src/devices/nm-device.c
index e7a4a05..7578abf 100644
--- a/src/devices/nm-device.c
+++ b/src/devices/nm-device.c
@@ -17,7 +17,11 @@
 #include <arpa/inet.h>
 #include <fcntl.h>
 #include <linux/if_addr.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/rtnetlink.h>
 #include <linux/pkt_sched.h>
 
diff --git a/src/nm-manager.c b/src/nm-manager.c
index 132cf5a..0dd71dd 100644
--- a/src/nm-manager.c
+++ b/src/nm-manager.c
@@ -15,6 +15,7 @@
 #include <sys/stat.h>
 #include <sys/sendfile.h>
 #include <limits.h>
+#include <asm/types.h>
 
 #include "nm-glib-aux/nm-c-list.h"
 
diff --git a/src/platform/nm-linux-platform.c b/src/platform/nm-linux-platform.c
index 305ae52..10cfec6 100644
--- a/src/platform/nm-linux-platform.c
+++ b/src/platform/nm-linux-platform.c
@@ -6,6 +6,7 @@
 #include "nm-default.h"
 
 #include "nm-linux-platform.h"
+#include "wpan/nm-wpan-utils.h"
 
 #include <arpa/inet.h>
 #include <dlfcn.h>
@@ -14,7 +15,11 @@
 #include <libudev.h>
 #include <linux/fib_rules.h>
 #include <linux/ip.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/if_bridge.h>
 #include <linux/if_link.h>
 #include <linux/if_tun.h>
diff --git a/src/systemd/src/libsystemd-network/sd-dhcp6-client.c b/src/systemd/src/libsystemd-network/sd-dhcp6-client.c
index e1150f9..3cb8390 100644
--- a/src/systemd/src/libsystemd-network/sd-dhcp6-client.c
+++ b/src/systemd/src/libsystemd-network/sd-dhcp6-client.c
@@ -7,7 +7,11 @@
 
 #include <errno.h>
 #include <sys/ioctl.h>
+#if defined(__GLIBC__)
 #include <linux/if_arp.h>
+#else
+#include <linux/if.h>
+#endif
 #include <linux/if_infiniband.h>
 
 #include "sd-dhcp6-client.h"
diff --git a/src/devices/nm-device-bridge.c b/src/devices/nm-device-bridge.c
index c93c6a5..66d2532 100644
--- a/src/devices/nm-device-bridge.c
+++ b/src/devices/nm-device-bridge.c
@@ -15,6 +15,8 @@
 #include "nm-device-factory.h"
 #include "nm-core-internal.h"
 
+#include <linux/if_ether.h>
+
 #include "nm-device-logging.h"
 _LOG_DECLARE_SELF(NMDeviceBridge);
