From ac8b551d16feb2fd9dbc5b1bfac4057ce6e9ef87 Mon Sep 17 00:00:00 2001
From: "j.r" <j.r@jugendhacker.de>
Date: Tue, 29 Dec 2020 14:44:26 +0100
Subject: [PATCH] Change setup.py to accept new python-telegram ver

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 2455e78..011eed7 100644
--- a/setup.py
+++ b/setup.py
@@ -22,5 +22,5 @@ setup(
     packages=["tg"],
     entry_points={"console_scripts": ["tg = tg.__main__:main"]},
     python_requires=">=3.8",
-    install_requires=["python-telegram==0.14.0"],
+    install_requires=["python-telegram==0.15.0"],
 )
-- 
2.34.1

