--- a/src/lanes_private.h	2025-10-23 02:21:41.000000000 -0400
+++ b/src/lanes_private.h	2025-12-22 02:29:21.425242307 -0500
@@ -79,7 +79,7 @@
 // 'Lane' are malloc/free'd and the handle only carries a pointer.
 // This is not deep userdata since the handle's not portable among lanes.
 //
-inline Lane* lua_toLane(lua_State* L, int i_)
+static inline Lane* lua_toLane(lua_State* L, int i_)
 {
     return *(Lane**)(luaL_checkudata(L, i_, "Lane"));
 }
