--- v4l-utils-1.26.1.orig/lib/include/libv4l1.h
+++ v4l-utils-1.26.1/lib/include/libv4l1.h
@@ -62,7 +62,7 @@
 LIBV4L_PUBLIC int v4l1_open(const char *file, int oflag, ...);
 LIBV4L_PUBLIC int v4l1_close(int fd);
 LIBV4L_PUBLIC int v4l1_dup(int fd);
-LIBV4L_PUBLIC int v4l1_ioctl(int fd, unsigned long int request, ...);
+LIBV4L_PUBLIC int v4l1_ioctl(int fd, int request, ...);
 LIBV4L_PUBLIC ssize_t v4l1_read(int fd, void *buffer, size_t n);
 LIBV4L_PUBLIC void *v4l1_mmap(void *start, size_t length, int prot, int flags,
 		int fd, int64_t offset);
--- v4l-utils-1.26.1.orig/lib/include/libv4l2.h
+++ v4l-utils-1.26.1/lib/include/libv4l2.h
@@ -63,7 +63,7 @@
 LIBV4L_PUBLIC int v4l2_open(const char *file, int oflag, ...);
 LIBV4L_PUBLIC int v4l2_close(int fd);
 LIBV4L_PUBLIC int v4l2_dup(int fd);
-LIBV4L_PUBLIC int v4l2_ioctl(int fd, unsigned long int request, ...);
+LIBV4L_PUBLIC int v4l2_ioctl(int fd, int request, ...);
 LIBV4L_PUBLIC ssize_t v4l2_read(int fd, void *buffer, size_t n);
 LIBV4L_PUBLIC ssize_t v4l2_write(int fd, const void *buffer, size_t n);
 LIBV4L_PUBLIC void *v4l2_mmap(void *start, size_t length, int prot, int flags,
--- v4l-utils-1.26.1.orig/lib/libv4l1/libv4l1.c
+++ v4l-utils-1.26.1/lib/libv4l1/libv4l1.c
@@ -478,7 +478,7 @@
 	return v4l2_dup(fd);
 }
 
-int v4l1_ioctl(int fd, unsigned long int request, ...)
+int v4l1_ioctl(int fd, int request, ...)
 {
 	void *arg;
 	va_list ap;
--- v4l-utils-1.26.1.orig/lib/libv4l2/libv4l2.c
+++ v4l-utils-1.26.1/lib/libv4l2/libv4l2.c
@@ -1051,7 +1051,7 @@
 	return 0;
 }
 
-int v4l2_ioctl(int fd, unsigned long int request, ...)
+int v4l2_ioctl(int fd, int request, ...)
 {
 	void *arg;
 	va_list ap;
