diff --git a/repo-projects/Directory.Build.props.orig b/repo-projects/Directory.Build.props
index b5d4fa43c88..3db84f2e63e 100644
--- a/repo-projects/Directory.Build.props.orig
+++ b/repo-projects/Directory.Build.props
@@ -92,6 +92,7 @@
     <BuildArgs>$(BuildArgs) /p:DotNetBuildSourceOnly=true</BuildArgs>
     <BuildArgs>$(BuildArgs) /p:PreviouslySourceBuiltNupkgCacheDir="$(PrebuiltSourceBuiltPackagesPath)"</BuildArgs>
     <BuildArgs>$(BuildArgs) /p:ReferencePackageNupkgCacheDir="$(ReferencePackagesDir)"</BuildArgs>
+    <BuildArgs>$(BuildArgs) /consoleLoggerParameters:ShowTimestamp</BuildArgs>
   </PropertyGroup>
 
   <PropertyGroup>

diff --git a/src/fsharp/eng/build.sh b/src/fsharp/eng/build.sh
index fff8414b3ef..d470fc6ed56 100755
--- a/src/fsharp/eng/build.sh
+++ b/src/fsharp/eng/build.sh
@@ -144,7 +144,7 @@ while [[ $# > 0 ]]; do
     --sourcebuild)
       source_build=true
       ;;
-    /p:*)
+    /*)
       properties="$properties $1"
       ;;
     *)
diff --git a/src/roslyn/eng/build.sh b/src/roslyn/eng/build.sh
index a08cbea099d..d3ec235732f 100755
--- a/src/roslyn/eng/build.sh
+++ b/src/roslyn/eng/build.sh
@@ -167,7 +167,7 @@ while [[ $# > 0 ]]; do
       args="$args $1"
       shift
       ;;
-    /p:*)
+    /*)
       properties="$properties $1"
       ;;
     *)
