diff --git a/scripts/xdg-open b/scripts/xdg-open
index 4cc18a8..e7ed9c3 100755
--- a/scripts/xdg-open
+++ b/scripts/xdg-open
@@ -303,7 +303,7 @@ detectDE()
 {
     if [ x"$KDE_FULL_SESSION" = x"true" ]; then DE=kde;
     elif [ x"$GNOME_DESKTOP_SESSION_ID" != x"" ]; then DE=gnome;
-    elif xprop -root _DT_SAVE_MODE | grep ' = \"xfce4\"$' >/dev/null 2>&1; then DE=xfce;
+    elif [ -x /usr/bin/exo-open ]; then DE=xfce;
     fi
 }
 
