From 8370499d4035ec7a5c8e1f1b04d5a6c404883442 Mon Sep 17 00:00:00 2001
From: Matteo Delfino <kendatsuba@gmail.com>
Date: Wed, 21 Oct 2009 14:26:45 +0000
Subject: Fix BIOS read on 650 and 760. (#19070)

Signed-off-by: Matteo Delfino <kendatsuba@gmail.com>
Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/sis_driver.c b/src/sis_driver.c
index 994b02d..b19e7a9 100644
--- a/src/sis_driver.c
+++ b/src/sis_driver.c
@@ -4099,6 +4099,8 @@ SISPreInit(ScrnInfoPtr pScrn, int flags)
 	     case SIS_315H:
 	     case SIS_330:
 	     case SIS_340:
+	     case SIS_650:
+	     case SIS_760:
 	     case XGI_40:     readpci = TRUE;
 			      break;
 	     case XGI_20:     readpci = TRUE;
--
cgit v0.8.2
