From dd7f7953d01a3e1b8bd78334aed05c016558df85 Mon Sep 17 00:00:00 2001
From: Andreas Nilsson <nisses.mail@home.se>
Date: Tue, 26 Oct 2010 15:09:15 +0000
Subject: Remove duplicate label in File System tab

https://bugzilla.gnome.org/show_bug.cgi?id=597250
---
diff --git a/src/disks.cpp b/src/disks.cpp
index 172fc7a..c694c0a 100644
--- a/src/disks.cpp
+++ b/src/disks.cpp
@@ -288,7 +288,6 @@ GtkWidget *
 create_disk_view(ProcData *procdata)
 {
 	GtkWidget *disk_box;
-	GtkWidget *label;
 	GtkWidget *scrolled;
 	GtkWidget *disk_tree;
 	GtkListStore *model;
@@ -310,9 +309,6 @@ create_disk_view(ProcData *procdata)
 
 	gtk_container_set_border_width(GTK_CONTAINER(disk_box), 12);
 
-	label = make_title_label(_("File Systems"));
-	gtk_box_pack_start(GTK_BOX(disk_box), label, FALSE, FALSE, 0);
-
 	scrolled = gtk_scrolled_window_new(NULL, NULL);
 	gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(scrolled),
 				       GTK_POLICY_AUTOMATIC,
--
cgit v0.9
