diff --git a/doc/asn2deb.pod b/doc/asn2deb.pod
index 0d88a95..3ad1a59 100644
--- a/doc/asn2deb.pod
+++ b/doc/asn2deb.pod
@@ -1,4 +1,4 @@
-
+=encoding utf8
 =head1 NAME
 
 asn2deb - Create a Debian package for BER monitoring from ASN.1
diff --git a/doc/idl2deb.pod b/doc/idl2deb.pod
index 7daa798..6d44491 100644
--- a/doc/idl2deb.pod
+++ b/doc/idl2deb.pod
@@ -1,4 +1,4 @@
-
+=encoding utf8
 =head1 NAME
 
 idl2deb - Create a Debian package for CORBA monitoring from IDL
diff --git a/doc/reordercap.pod b/doc/reordercap.pod
index 08a1c9a..3278a5b 100644
--- a/doc/reordercap.pod
+++ b/doc/reordercap.pod
@@ -40,6 +40,8 @@ the same way B<reordercap> handles this.
 When the B<-n> option is used, B<reordercap> will not write out the output
 file if it finds that the input file is already in order.
 
+=back
+
 =head1 SEE ALSO
 
 pcap(3), wireshark(1), tshark(1), dumpcap(1), editcap(1), mergecap(1),
