https://gitlab.torproject.org/tpo/core/arti/-/work_items/2543

--- a/crates/arti/Cargo.toml
+++ b/crates/arti/Cargo.toml
@@ -103,7 +103,7 @@ ctor-keystore = ["arti-client/ctor-keyst
 flowctl-cc = ["arti-client/flowctl-cc"]
 counter-galois-onion = ["arti-client/counter-galois-onion", "flowctl-cc", "__is_experimental"]
 
-rpc = ["arti-rpcserver", "tor-rpcbase", "tor-rpc-connect", "arti-client/rpc", "derive-deftly"]
+rpc = ["arti-rpcserver", "dep:tor-rpcbase", "tor-rpc-connect", "arti-client/rpc", "derive-deftly"]
 
 # These are not nonadditive from a software POV, but we mark them as such because
 # they introduce a different backend, and only one backend can actually be used.
--- a/crates/arti-client/Cargo.toml
+++ b/crates/arti-client/Cargo.toml
@@ -118,7 +118,7 @@ vanguards = ["tor-guardmgr/vanguards", "
 
 flowctl-cc = ["tor-circmgr/flowctl-cc"]
 
-rpc = ["dyn-clone", "tor-rpcbase"]
+rpc = ["dyn-clone", "dep:tor-rpcbase"]
 
 __is_nonadditive = []
 
