https://gitlab.torproject.org/tpo/core/arti/-/work_items/2545

--- a/crates/tor-async-utils/Cargo.toml
+++ b/crates/tor-async-utils/Cargo.toml
@@ -20,11 +20,11 @@ full = [
     "oneshot-fused-workaround/full",
     "slotmap-careful/full",
     "tokio",
+    "tokio-util",
     "tor-basic-utils/full",
     "tor-rtcompat/full",
     "web-time-compat/full",
 ]
-tokio = ["dep:tokio", "dep:tokio-util"]
 # Use visibility dependency so we can change the some functions that our
 # benchmarks require from pub(crate) to pub.
 bench = ["dep:visibility"]
