diff --git a/tests/wp/meson.build b/tests/wp/meson.build
index 178564d..adc9cef 100644
--- a/tests/wp/meson.build
+++ b/tests/wp/meson.build
@@ -62,13 +62,6 @@ test(
   env: common_env,
 )
 
-test(
-  'test-spa-json',
-  executable('test-spa-json', 'spa-json.c',
-      dependencies: common_deps, c_args: common_args),
-  env: common_env,
-)
-
 test(
   'test-spa-pod',
   executable('test-spa-pod', 'spa-pod.c',
