--- a/libfreeipmi/driver/ipmi-kcs-driver.c
+++ b/libfreeipmi/driver/ipmi-kcs-driver.c
@@ -162,7 +162,7 @@
 #if defined(__FreeBSD__) || defined(__NetBSD__) || defined(__OpenBSD__)
 # define _INB(port)  inb (port)
 # define _OUTB(data, port)  outb (port, data)
-#elif (defined(HAVE_INB) && defined(HAVE_OUTB)) || (defined(HAVE_DECL_INB) && defined(HAVE_DECL_OUTB))
+#elif (defined(HAVE_INB) && defined(HAVE_OUTB)) || (HAVE_DECL_INB && HAVE_DECL_OUTB)
 # define _INB(port)  inb (port)
 # define _OUTB(data, port)  outb (data, port)
 #else
