--- PHP_5_3/Zend/zend_execute_API.c	2009/12/18 18:41:53	292294
+++ PHP_5_3/Zend/zend_execute_API.c	2009/12/18 19:12:11	292295
@@ -838,6 +838,7 @@
 		zval *param;
 
 		if (EX(function_state).function->type == ZEND_INTERNAL_FUNCTION 
+			&& (EX(function_state).function->common.fn_flags & ZEND_ACC_CALL_VIA_HANDLER) == 0 
 			&& !ARG_SHOULD_BE_SENT_BY_REF(EX(function_state).function, i + 1)
 			&& PZVAL_IS_REF(*fci->params[i])) {
 			SEPARATE_ZVAL(fci->params[i]);
