--- a/python/samba/provision/sambadns.py
+++ b/python/samba/provision/sambadns.py
@@ -943,7 +943,8 @@
             bind9_9 = ''
         elif bind_info.upper().find('BIND 9.10') != -1:
             bind9_10 = ''
-        elif bind_info.upper().find('BIND 9.11') != -1:
+        elif bind_info.upper().find('BIND 9.11') != -1 \
+            or bind_info.upper().find('BIND 9.12') != -1:
             bind9_11 = ''
         elif bind_info.upper().find('BIND 9.7') != -1:
             raise ProvisioningError("DLZ option incompatible with BIND 9.7.")
--- a/source4/setup/named.conf.dlz
+++ b/source4/setup/named.conf.dlz
@@ -19,7 +19,7 @@
     # For BIND 9.10.x
     ${BIND9_10} database "dlopen ${MODULESDIR}/bind9/dlz_bind9_10.so";
 
-    # For BIND 9.11.x
+    # For BIND 9.11.x, 9.12.x
     ${BIND9_11} database "dlopen ${MODULESDIR}/bind9/dlz_bind9_11.so";
 };
 
