--- ./python/samba/netcmd/domain.py.orig
+++ ./python/samba/netcmd/domain.py
@@ -388,8 +388,10 @@
             def ask(prompt, default=None):
                 if default is not None:
                     print("%s [%s]: " % (prompt, default), end=' ')
+                    sys.stdout.flush()
                 else:
                     print("%s: " % (prompt,), end=' ')
+                    sys.stdout.flush()
                 return sys.stdin.readline().rstrip("\n") or default
 
             try:
