diff --git a/source3/wscript b/source3/wscript
index b664b6f..db1fc8b 100644
--- a/source3/wscript
+++ b/source3/wscript
@@ -629,8 +629,8 @@ msg.msg_accrightslen = sizeof(fd);
                                 headers='unistd.h sys/types.h dirent.h',
                                 define='HAVE_DIRENT_D_OFF')
 
-    if (conf.CONFIG_SET('HAVE_GETDOMAINNAME')):
-        conf.DEFINE('HAVE_NETGROUP', '1')
+ #   if (conf.CONFIG_SET('HAVE_GETDOMAINNAME')):
+ #       conf.DEFINE('HAVE_NETGROUP', '1')
 
     # Look for CUPS
     if Options.options.with_cups: