diff --git a/BSM Simple/xfwm4/bottom-active.xpm b/BSM Simple/xfwm4/bottom-active.xpm
index 4096e85..49d4eab 100644
--- a/BSM Simple/xfwm4/bottom-active.xpm	
+++ b/BSM Simple/xfwm4/bottom-active.xpm	
@@ -1,6 +1,10 @@
 /* XPM */
 static char * bottom_active_xpm[] = {
-"17 1 2 1",
+"16 3 4 1",
 " 	c None",
 ".	c #888888",
-"................."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+"0000000000000000",
+"oooooooooooooooo",
+"................"};
diff --git a/BSM Simple/xfwm4/bottom-inactive.xpm b/BSM Simple/xfwm4/bottom-inactive.xpm
index fbfbe7b..2ebcdd5 100644
--- a/BSM Simple/xfwm4/bottom-inactive.xpm	
+++ b/BSM Simple/xfwm4/bottom-inactive.xpm	
@@ -1,6 +1,10 @@
 /* XPM */
 static char * bottom_inactive_xpm[] = {
-"17 1 2 1",
+"16 3 4 1",
 " 	c None",
 ".	c #888888",
-"................."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+"0000000000000000",
+"oooooooooooooooo",
+"................"};
diff --git a/BSM Simple/xfwm4/bottom-left-active.xpm b/BSM Simple/xfwm4/bottom-left-active.xpm
index 5bd7b94..b5a975b 100644
--- a/BSM Simple/xfwm4/bottom-left-active.xpm	
+++ b/BSM Simple/xfwm4/bottom-left-active.xpm	
@@ -1,22 +1,23 @@
 /* XPM */
 static char * bottom_left_active_xpm[] = {
-"1 17 2 1",
+"16 16 4 1",
 " 	c None",
 ".	c #888888",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".000000000000000",
+".ooooooooooooooo",
+" ..............."};
diff --git a/BSM Simple/xfwm4/bottom-left-inactive.xpm b/BSM Simple/xfwm4/bottom-left-inactive.xpm
index cbd3b99..1a03ca7 100644
--- a/BSM Simple/xfwm4/bottom-left-inactive.xpm	
+++ b/BSM Simple/xfwm4/bottom-left-inactive.xpm	
@@ -1,22 +1,23 @@
 /* XPM */
 static char * bottom_left_inactive_xpm[] = {
-"1 17 2 1",
+"16 16 4 1",
 " 	c None",
 ".	c #888888",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".               ",
+".000000000000000",
+".ooooooooooooooo",
+" ..............."};
diff --git a/BSM Simple/xfwm4/bottom-right-active.xpm b/BSM Simple/xfwm4/bottom-right-active.xpm
index fb3039f..6b16404 100644
--- a/BSM Simple/xfwm4/bottom-right-active.xpm	
+++ b/BSM Simple/xfwm4/bottom-right-active.xpm	
@@ -1,22 +1,23 @@
 /* XPM */
 static char * bottom_right_active_xpm[] = {
-"1 17 2 1",
+"16 16 4 1",
 " 	c None",
 ".	c #888888",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"000000000000000.",
+"ooooooooooooooo.",
+"............... "};
diff --git a/BSM Simple/xfwm4/bottom-right-inactive.xpm b/BSM Simple/xfwm4/bottom-right-inactive.xpm
index 527219a..002cc24 100644
--- a/BSM Simple/xfwm4/bottom-right-inactive.xpm	
+++ b/BSM Simple/xfwm4/bottom-right-inactive.xpm	
@@ -1,22 +1,23 @@
 /* XPM */
 static char * bottom_right_inactive_xpm[] = {
-"1 17 2 1",
+"16 16 4 1",
 " 	c None",
 ".	c #888888",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-".",
-"."};
+"o	c #C0C0C0",
+"0	c #d0d0d0",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"               .",
+"000000000000000.",
+"ooooooooooooooo.",
+"............... "};
