--- a/src/iwd-dbus.conf	2018-09-20 19:43:09.000000000 +0200
+++ b/src/iwd-dbus.conf	2018-09-20 21:02:03.000000000 +0100
@@ -18,6 +18,11 @@
     <allow send_interface="net.connman.iwd.Agent"/>
   </policy>
 
+  <policy group="netdev">
+    <allow send_destination="net.connman.iwd"/>
+    <allow send_interface="net.connman.iwd.Agent"/>
+  </policy>
+
   <policy at_console="true">
     <allow send_destination="net.connman.iwd"/>
     <allow send_interface="net.connman.iwd.Agent"/>
