--- a/Makefile.am	2019-08-20 17:25:10.375741405 +0200
+++ b/Makefile.am	2019-08-20 17:27:19.502589536 +0200
@@ -395,8 +395,7 @@
 				src/ie.h src/ie.c
 unit_test_mpdu_LDADD = $(ell_ldadd)
 
-unit_test_eapol_SOURCES = unit/test-eapol.c \
-				src/crypto.h src/crypto.c \
+unit_test_eapol_SOURCES = src/crypto.h src/crypto.c \
 				src/ie.h src/ie.c \
 				src/watchlist.h src/watchlist.c \
 				src/eapol.h src/eapol.c \

@@ -326,7 +326,7 @@
 unit_tests = unit/test-cmac-aes \
 		unit/test-hmac-md5 unit/test-hmac-sha1 unit/test-hmac-sha256 \
 		unit/test-prf-sha1 unit/test-kdf-sha256 \
-		unit/test-crypto unit/test-eapol unit/test-mpdu \
+		unit/test-crypto unit/test-mpdu \
 		unit/test-ie unit/test-util unit/test-ssid-security \
 		unit/test-arc4 unit/test-wsc unit/test-eap-mschapv2 \
 		unit/test-eap-sim unit/test-sae unit/test-p2p
