--- a/Makefile.in	2019-08-28 21:45:47.000000000 +0200
+++ b/Makefile.in	2019-08-29 00:48:01.030221354 +0200
@@ -209,7 +209,7 @@
 am__EXEEXT_8 = unit/test-cmac-aes$(EXEEXT) unit/test-hmac-md5$(EXEEXT) \
 	unit/test-hmac-sha1$(EXEEXT) unit/test-hmac-sha256$(EXEEXT) \
 	unit/test-prf-sha1$(EXEEXT) unit/test-kdf-sha256$(EXEEXT) \
-	unit/test-crypto$(EXEEXT) unit/test-eapol$(EXEEXT) \
+	unit/test-crypto$(EXEEXT) \
 	unit/test-mpdu$(EXEEXT) unit/test-ie$(EXEEXT) \
 	unit/test-util$(EXEEXT) unit/test-ssid-security$(EXEEXT) \
 	unit/test-arc4$(EXEEXT) unit/test-wsc$(EXEEXT) \
@@ -1148,7 +1148,7 @@
 @TOOLS_TRUE@tools_test_runner_LDADD = $(ell_ldadd)
 unit_tests = unit/test-cmac-aes unit/test-hmac-md5 unit/test-hmac-sha1 \
 	unit/test-hmac-sha256 unit/test-prf-sha1 unit/test-kdf-sha256 \
-	unit/test-crypto unit/test-eapol unit/test-mpdu unit/test-ie \
+	unit/test-crypto unit/test-mpdu unit/test-ie \
 	unit/test-util unit/test-ssid-security unit/test-arc4 \
 	unit/test-wsc unit/test-eap-mschapv2 unit/test-eap-sim \
 	unit/test-sae unit/test-p2p $(am__append_20)
@@ -1205,8 +1205,7 @@
 				src/ie.h src/ie.c
 
 unit_test_mpdu_LDADD = $(ell_ldadd)
-unit_test_eapol_SOURCES = unit/test-eapol.c \
-				src/crypto.h src/crypto.c \
+unit_test_eapol_SOURCES = src/crypto.h src/crypto.c \
 				src/ie.h src/ie.c \
 				src/watchlist.h src/watchlist.c \
 				src/eapol.h src/eapol.c \
