diff --git a/tests/test___init__.py b/tests/test___init__.py
index 25c008c..56bff35 100644
--- a/tests/test___init__.py
+++ b/tests/test___init__.py
@@ -10,7 +10,7 @@ EMOJI_LEVELS = ('CRITICAL', 'ERROR', 'FATAL', 'WARN', 'WARNING', 'INFO', 'DEBUG'
 def test_version():
     """Make sure that version matches semver.
     """
-    assert semver.VersionInfo.isvalid(milc.__VERSION__)
+    assert semver.VersionInfo.is_valid(milc.__VERSION__)
 
 
 def test_emojis():
