diff --git a/Makefile b/Makefile
index 47f9ee9..7091128 100644
--- a/Makefile
+++ b/Makefile
@@ -10,13 +10,7 @@
 # To manually invoke the locally installed Go, use ./go
 
 # Go installation config.
-GO_VER=1.11.5
-SYSTEM_NAME:=$(shell uname -s | tr '[:upper:]' '[:lower:]')
-SYSTEM_ARCH:=$(shell uname -m)
-GO_ARCH:=$(if $(filter x86_64, $(SYSTEM_ARCH)),amd64,386)
-GO_VERSION:=$(GO_VER).$(SYSTEM_NAME)-$(GO_ARCH)
-GO_DOWNLOAD_URL:=https://dl.google.com/go/go$(GO_VERSION).tar.gz
-GO_DIR:=go-$(GO_VER)
+GO_DIR:=/usr
 
 # Build websocketd binary
 websocketd: $(GO_DIR)/bin/go $(wildcard *.go) $(wildcard libwebsocketd/*.go)
@@ -24,13 +18,6 @@ websocketd: $(GO_DIR)/bin/go $(wildcard *.go) $(wildcard libwebsocketd/*.go)
 
 localgo: $(GO_DIR)/bin/go
 
-# Download and unpack Go distribution.
-$(GO_DIR)/bin/go:
-	mkdir -p $(GO_DIR)
-	rm -f $@
-	@echo Downloading and unpacking Go $(GO_VERSION) to $(GO_DIR)
-	curl -s $(GO_DOWNLOAD_URL) | tar xfz - --strip-components=1 -C $(GO_DIR)
-
 # Clean up binary
 clean:
 	rm -rf websocketd
