diff --git a/mupen64plus-core/src/device/r4300/new_dynarec/new_dynarec.c b/mupen64plus-core/src/device/r4300/new_dynarec/new_dynarec.c
index 9fd6054..c6fe0f7 100644
--- a/mupen64plus-core/src/device/r4300/new_dynarec/new_dynarec.c
+++ b/mupen64plus-core/src/device/r4300/new_dynarec/new_dynarec.c
@@ -24,10 +24,8 @@
 #include <string.h>
 #include <assert.h>
 
-#if defined(__APPLE__)
 #include <sys/types.h> // needed for u_int, u_char, etc
 #define MAP_ANONYMOUS MAP_ANON
-#endif
 
 #include "new_dynarec.h"
 #include "api/m64p_types.h"
