diff --git a/setup.cfg b/setup.cfg
index 45e8221..6af011f 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -26,6 +26,7 @@ replace = version="{new_version}"
 test = pytest
 
 [metadata]
+version= %%PKGVER%%
 license_file = LICENSE
 
 [tool:pytest]
diff --git a/setup.py b/setup.py
index 9e692e7..391fd20 100644
--- a/setup.py
+++ b/setup.py
@@ -139,7 +139,6 @@ with open('README.rst', 'r') as f:
 
 setup_params = dict(
     name="compreffor",
-    use_scm_version={"write_to": "src/python/compreffor/_version.py"},
     description="A CFF subroutinizer for fontTools.",
     long_description=long_description,
     long_description_content_type="text/x-rst",
@@ -152,7 +151,7 @@ setup_params = dict(
         'build_ext': custom_build_ext,
     },
     setup_requires=(
-        ["setuptools_scm"] + pytest_runner + wheel
+        pytest_runner + wheel
     ),
     tests_require=[
         'pytest>=2.8',
