From a0ee588b62cca56c756fa90d0c04a15b12d59e9a Mon Sep 17 00:00:00 2001
From: Drew DeVault <sir@cmpwn.com>
Date: Mon, 8 Feb 2021 20:40:32 -0500
Subject: [PATCH] setup.cfg: remove scm nonsense

---
 pyproject.toml | 4 +---
 setup.cfg      | 1 -
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index b6ebc0b..8ca88ac 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,12 +1,10 @@
 [build-system]
-requires = ["setuptools>=42", "wheel", "setuptools_scm[toml]>=3.4.1"]
+requires = ["setuptools>=42", "wheel"]>=3.4.1"]
 build-backend = "setuptools.build_meta"
 
 [tool.black]
 skip-string-normalization = true
 
-[tool.setuptools_scm]
-
 [pytest.enabler.black]
 addopts = "--black"
 
diff --git a/setup.cfg b/setup.cfg
index 7d9424d..2c336ea 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -23,7 +23,6 @@ include_package_data = true
 python_requires = >=3.6
 install_requires =
     zipp >= 0.4; python_version < '3.8'
-setup_requires = setuptools_scm[toml] >= 3.4.1
 
 [options.packages.find]
 exclude =
-- 
2.30.1

