diff --git a/setup.py b/setup.py
index 3de9c1c..8833d4a 100644
--- a/setup.py
+++ b/setup.py
@@ -14,6 +14,4 @@ setup(
     license='MIT',
 
     packages=['speg'],
-
-    install_requires=['six']
     )
diff --git a/speg/peg.py b/speg/peg.py
index a355a1e..c3085c0 100644
--- a/speg/peg.py
+++ b/speg/peg.py
@@ -1,8 +1,6 @@
 import re
 import sys
 
-import six
-
 from .errors import ExpectedExpr, UnexpectedExpr, SemanticFailure, raise_parsing_error
 from .position import Position, get_line_at_position
 from .rules import eof
@@ -66,7 +64,7 @@ class ParsingState(object):
             if st.position.offset != len(self._s):
                 self._raise(ExpectedExpr, eof)
             return ''
-        elif isinstance(r, six.string_types):
+        elif isinstance(r, str):
             flags = args[0] if args else 0
             compiled = self._re_cache.get((r, flags))
             if not compiled:
diff --git a/speg/rules.py b/speg/rules.py
index 9ce0aed..a86df5f 100644
--- a/speg/rules.py
+++ b/speg/rules.py
@@ -1,12 +1,10 @@
-from six import string_types
-
 class Eof: pass
 eof = Eof()
 
 def rule_to_str(rule):
     if rule is eof:
         return 'eof'
-    if isinstance(rule, string_types):
+    if isinstance(rule, str):
         return repr(rule)
     fn_name = rule.__name__
     if fn_name.startswith('_'):
diff --git a/tox.ini b/tox.ini
index 0262ef6..085a6e6 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,5 +1,5 @@
 [tox]
-envlist=py27,py34,py35,py36
+envlist=py34,py35,py36
 
 [testenv]
 usedevelop=True
@@ -7,7 +7,6 @@ usedevelop=True
 deps=
     pytest
     pytest-cov
-    six
 
 commands=
     py.test --cov=speg --cov-report term-missing {posargs}
