--- midori-0.5.6.orig/CMakeLists.txt
+++ midori-0.5.6/CMakeLists.txt
@@ -97,13 +97,8 @@
 endif ()
 
 # Check if execinfo.h header exists
-string(FIND ${CMAKE_SYSTEM_NAME} "BSD" BEGIN)
-if (${BEGIN} GREATER 0)
-    string(SUBSTRING ${CMAKE_SYSTEM_NAME} ${BEGIN} 3 BSD)
-else()
-    set(BSD 0)
-endif()
-if (UNIX AND NOT BSD)
+find_file (EXECINFO execinfo.h)
+if (EXECINFO)
     set(VALAFLAGS ${VALAFLAGS} -D HAVE_EXECINFO_H)
 endif ()
 
