--- secpwgen-1.3/Makefile.proto	2015-07-21 20:15:10.000000000 +0000
+++ secpwgen-1.3/Makefile.proto.new	2015-07-21 22:03:47.511905311 +0000
@@ -30,9 +30,9 @@
 ##
 # If using OpenSSL, uncomment the following 3 lines.
 ##
-#CRYPTO_CFLAGS =
-#CRYPTO_OBJS   = secure_random_openssl.o
-#CRYPTO_LIBS   = -lcrypto
+CRYPTO_CFLAGS =
+CRYPTO_OBJS   = secure_random_openssl.o
+CRYPTO_LIBS   = -lcrypto
 
 ##
 # If using cryptlib, uncomment the following 3 lines.
@@ -45,7 +45,7 @@
 # Change PREFIX to install to different directories. The binary is installed
 # in $PREFIX/bin, and the man in $PREFIX/man/man1
 ##
-PREFIX = /usr/local
+PREFIX = /usr
 
 ##
 # On some operating systems (most notably Linux 2.6 kernels) using mlockall
@@ -60,7 +60,7 @@ PREFIX = /usr/local
 # Sometimes you have only dynamic libraries available. In that case COMMENT
 # the following line.
 ##
-LINK_STATIC = -static
+#LINK_STATIC = -static
 
 ##############################################################################
 # NO USER MODIFIABLE PARTS AFTER THIS POINT
