diff --git a/tests/031_Imagick_affineTransformImage_basic.phpt b/tests/031_Imagick_affineTransformImage_basic.phpt
index d64fa05..27841d8 100644
--- a/tests/031_Imagick_affineTransformImage_basic.phpt
+++ b/tests/031_Imagick_affineTransformImage_basic.phpt
@@ -14,7 +14,7 @@ function affineTransformImage() {
     $imagick->newPseudoImage(640, 480, "magick:logo");
     $draw = new \ImagickDraw();
 
-    $angle = 40 ;
+    $angle = deg2rad(40);
 
     $affineRotate = array(
         "sx" => cos($angle), "sy" => cos($angle), 
@@ -34,4 +34,4 @@ affineTransformImage() ;
 echo "Ok";
 ?>
 --EXPECTF--
-Ok
\ No newline at end of file
+Ok
