From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 23 Apr 2023 22:12:30 +0200
Subject: [PATCH] Build with ureq native-tls

https://github.com/ramsayleung/rspotify/pull/402 (rebased)

--- a/vendor/rspotify/Cargo.toml
+++ b/vendor/rspotify/Cargo.toml
@@ -71,6 +71,7 @@
 reqwest-native-tls-vendored = ["rspotify-http/reqwest-native-tls-vendored"]
 # Same for ureq.
 ureq-rustls-tls = ["rspotify-http/ureq-rustls-tls"]
+ureq-native-tls = ["rspotify-http/ureq-native-tls"]
 
 # Internal features for checking async or sync compilation
 __async = ["futures", "async-stream", "async-trait"]
--- a/vendor/rspotify/rspotify-http/Cargo.toml
+++ b/vendor/rspotify/rspotify-http/Cargo.toml
@@ -46,6 +46,7 @@
 reqwest-native-tls-vendored = ["reqwest/native-tls-vendored"]
 # Same for ureq.
 ureq-rustls-tls = ["ureq/tls"]
+ureq-native-tls = ["ureq/native-tls"]
 
 # Internal features for checking async or sync compilation
 __async = ["async-trait"]
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,7 +61,7 @@
 
 [dependencies.rspotify]
 default-features = false
-features = ["client-ureq", "ureq-rustls-tls"]
+features = ["client-ureq", "ureq-native-tls"]
 version = "0.11.5"
 
 [dependencies.cursive]
