diff -upr cryptsetup-2.4.0.orig/src/utils_tools.c cryptsetup-2.4.0/src/utils_tools.c
--- cryptsetup-2.4.0.orig/src/utils_tools.c	2021-08-29 15:17:46.989874344 +0200
+++ cryptsetup-2.4.0/src/utils_tools.c	2021-08-29 15:18:45.543241866 +0200
@@ -79,10 +79,13 @@ void tool_log(int level, const char *msg
 
 	case CRYPT_LOG_NORMAL:
 		fprintf(stdout, "%s", msg);
+		fflush(stdout);
 		break;
 	case CRYPT_LOG_VERBOSE:
-		if (params && params->verbose)
+		if (params && params->verbose) {
 			fprintf(stdout, "%s", msg);
+			fflush(stdout);
+		}
 		break;
 	case CRYPT_LOG_ERROR:
 		fprintf(stderr, "%s", msg);
