diff -upr cryptsetup-2.1.0.orig/src/utils_tools.c cryptsetup-2.1.0/src/utils_tools.c
--- cryptsetup-2.1.0.orig/src/utils_tools.c	2019-01-31 21:37:12.000000000 +0100
+++ cryptsetup-2.1.0/src/utils_tools.c	2019-03-16 16:33:13.000000000 +0100
@@ -105,10 +105,13 @@ void tool_log(int level, const char *msg
 
 	case CRYPT_LOG_NORMAL:
 		fprintf(stdout, "%s", msg);
+		fflush(stdout);
 		break;
 	case CRYPT_LOG_VERBOSE:
-		if (opt_verbose)
+		if (opt_verbose) {
 			fprintf(stdout, "%s", msg);
+			fflush(stdout);
+		}
 		break;
 	case CRYPT_LOG_ERROR:
 		fprintf(stderr, "%s", msg);
