--- sysstat-10.2.0.old/common.c
+++ sysstat-10.2.0/common.c
@@ -29,6 +29,7 @@
 #include <sys/types.h>
 #include <dirent.h>
 #include <ctype.h>
+#include <limits.h>
 #include <libgen.h>
 
 #include "version.h"
--- sysstat-10.2.0.old/ioconf.c
+++ sysstat-10.2.0/ioconf.c
@@ -26,6 +26,7 @@
 #include <string.h>
 #include <errno.h>
 #include <dirent.h>
+#include <sys/types.h>
 #include <sys/stat.h>
 
 #include "ioconf.h"
--- sysstat-10.2.0.old/sa_common.c
+++ sysstat-10.2.0/sa_common.c
@@ -27,6 +27,7 @@
 #include <unistd.h>	/* For STDOUT_FILENO, among others */
 #include <dirent.h>
 #include <fcntl.h>
+#include <limits.h>
 #include <libgen.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
