--- ./gtk/Makefile.in.orig
+++ ./gtk/Makefile.in
@@ -895,7 +895,7 @@
 @WITH_PYTHON_TRUE@SpiceClientGtk_la_LDFLAGS = -module -avoid-version -fPIC
 @WITH_PYTHON_TRUE@SpiceClientGtk_la_SOURCES = spice-client-gtk-module.c
 @WITH_PYTHON_TRUE@nodist_SpiceClientGtk_la_SOURCES = spice-client-gtk-module.defs.c
-@WITH_PYTHON_TRUE@CODEGENDIR = `pkg-config --variable=codegendir pygtk-2.0`
+@WITH_PYTHON_TRUE@CODEGENDIR = `pkg-config --variable=codegendir pygobject-2.0`
 @WITH_PYTHON_TRUE@DEFSDIR = `pkg-config --variable=defsdir pygtk-2.0`
 @G_IR_SCANNER_SYMBOL_PREFIX_FALSE@PREFIX_ARGS = --strip-prefix=Spice
 @G_IR_SCANNER_SYMBOL_PREFIX_TRUE@PREFIX_ARGS = --symbol-prefix=spice --identifier-prefix=Spice
