diff -Naur js-1.8.5.orig/js/src/assembler/wtf/Platform.h js-1.8.5/js/src/assembler/wtf/Platform.h
--- js-1.8.5.orig/js/src/assembler/wtf/Platform.h
+++ js-1.8.5/js/src/assembler/wtf/Platform.h
@@ -129,16 +129,22 @@
    || defined(__POWERPC__) \
    || defined(_M_PPC)      \
    || defined(__PPC)
+#if !defined(__ppc64__) && !defined(__PPC64__)
 #define WTF_CPU_PPC 1
+#endif
+#if !defined(__LITTLE_ENDIAN__)
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
+#endif
 
 /* CPU(PPC64) - PowerPC 64-bit */
 #if   defined(__ppc64__) \
    || defined(__PPC64__)
 #define WTF_CPU_PPC64 1
+#if !defined(__LITTLE_ENDIAN__)
 #define WTF_CPU_BIG_ENDIAN 1
 #endif
+#endif
 
 /* CPU(SH4) - SuperH SH-4 */
 #if defined(__SH4__)
diff -Naur js-1.8.5.orig/js/src/configure.in js-1.8.5/js/src/configure.in
--- js-1.8.5.orig/js/src/configure.in
+++ js-1.8.5/js/src/configure.in
@@ -1498,7 +1498,7 @@
     CPU_ARCH=x86
     ;;
 
-powerpc64 | ppc64)
+powerpc64 | ppc64 | powerpc64le | ppc64le)
     CPU_ARCH=ppc64
     ;;
 
diff -Naur js-1.8.5/js/src/Makefile.in.ori js-1.8.5/js/src/Makefile.in
--- js-1.8.5/js/src/Makefile.in.ori
+++ js-1.8.5/js/src/Makefile.in
@@ -382,7 +382,7 @@ CPPSRCS += 	checks.cc \
 # END enclude sources for V8 dtoa
 #############################################
 
-ifeq (,$(filter-out aarch64 powerpc powerpc64 sparc sparc64 s390 s390x,$(TARGET_CPU)))
+ifeq (,$(filter-out aarch64 powerpc powerpc64 powerpc64le sparc sparc64 s390 s390x,$(TARGET_CPU)))
 
 VPATH +=	$(srcdir)/assembler \
 		$(srcdir)/assembler/wtf \
