From 77b8f1f26305141853cc324e4cfb1c62ac87e17b Mon Sep 17 00:00:00 2001
From: Jingyun Hua <huajingyun@loongson.cn>
Date: Sat, 16 Nov 2024 16:29:56 +0800
Subject: [PATCH] Skia:add target_sources for loongarch64

---
 Source/ThirdParty/skia/CMakeLists.txt | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/Source/ThirdParty/skia/CMakeLists.txt b/Source/ThirdParty/skia/CMakeLists.txt
index dca346d4..d0854578 100644
--- a/Source/ThirdParty/skia/CMakeLists.txt
+++ b/Source/ThirdParty/skia/CMakeLists.txt
@@ -895,6 +895,15 @@ add_library(Skia STATIC
     modules/skcms/src/skcms_TransformSkx.cc
 )
 
+
+if (WTF_CPU_LOONGARCH64)
+    target_sources(Skia PRIVATE src/opts/SkOpts_lasx.cpp)
+    target_sources(Skia PRIVATE src/core/SkBitmapProcState_opts_lasx.cpp)
+    target_sources(Skia PRIVATE src/core/SkSwizzler_opts_lasx.cpp)
+    target_sources(Skia PRIVATE src/core/SkBlitRow_opts_lasx.cpp)
+	
+endif ()
+
 if (ANDROID)
     target_sources(Skia PRIVATE src/ports/SkDebug_android.cpp)
     target_sources(Skia PRIVATE src/ports/SkFontMgr_android.cpp)
-- 
2.46.2

