diff --git a/Source/WebKit/UIProcess/gtk/PointerLockManagerX11.cpp b/Source/WebKit/UIProcess/gtk/PointerLockManagerX11.cpp
index e1d35d4a3808..b36a5417dabd 100644
--- a/Source/WebKit/UIProcess/gtk/PointerLockManagerX11.cpp
+++ b/Source/WebKit/UIProcess/gtk/PointerLockManagerX11.cpp
@@ -29,11 +29,14 @@
 #if PLATFORM(X11)
 
 #include "WebPageProxy.h"
-#include <X11/Xlib.h>
 #include <gtk/gtk.h>
 #include <wtf/TZoneMallocInlines.h>
 #include <wtf/glib/GRefPtr.h>
 
+// Some of the macros in X11/X.h conflict with definitions in the
+// bmalloc code so this include must come after them (#278858).
+#include <X11/Xlib.h>
+
 #if USE(GTK4)
 #include <gdk/x11/gdkx.h>
 #else
