--- audacious-2.4.2.orig/src/libaudcore/audstrings.c
+++ audacious-2.4.2/src/libaudcore/audstrings.c
@@ -230,9 +230,8 @@
     return dirname;
 }
 
-#if defined(__GLIBC__) && (__GLIBC__ >= 2)
-#define HAVE_EXECINFO 1
-#include <execinfo.h>
+#ifdef HAVE_EXECINFO_H
+# include <execinfo.h>
 #endif
 
 /**
@@ -258,7 +257,7 @@
 
     /* already UTF-8? */
     if (!g_utf8_validate(str, -1, NULL)) {
-#ifdef HAVE_EXECINFO
+#ifdef HAVE_EXECINFO_H
 		gint i, nsymbols;
 		const gint nsymmax = 50;
 		void *addrbuf[nsymmax];
