--- a/Rakefile
+++ b/Rakefile
@@ -2,13 +2,8 @@
 # Copyright (c) 2010-2017 VMware, Inc.  All Rights Reserved.
 # SPDX-License-Identifier: MIT
 
-require 'bundler/gem_tasks'
 require 'rake/testtask'
-require 'yard'
-require 'rubocop/rake_task'
 
-Dir.glob(File.join('lib/tasks/**/*.rake')).each { |file| load file }
-
 task(default: :test)
 
 Rake::TestTask.new do |t|
@@ -17,7 +12,3 @@
   t.verbose = true
   t.warning = true
 end
-
-RuboCop::RakeTask.new
-
-YARD::Rake::YardocTask.new
--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -2,9 +2,6 @@
 # Copyright (c) 2011-2017 VMware, Inc.  All Rights Reserved.
 # SPDX-License-Identifier: MIT
 
-require 'simplecov'
-SimpleCov.start { add_filter '/test/' }
-
 require 'rbvmomi2'
 VIM = RbVmomi::VIM
 
