--- ./src/profiles.c.orig
+++ ./src/profiles.c
@@ -205,13 +205,13 @@
   for (i = 0; i < ctx->nb_streams; i++)
   {
     if (audio_stream == -1 &&
-        ctx->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO)
+        ctx->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO)
     {
       audio_stream = i;
       continue;
     }
     else if (video_stream == -1 &&
-             ctx->streams[i]->codec->codec_type == CODEC_TYPE_VIDEO)
+             ctx->streams[i]->codec->codec_type == AVMEDIA_TYPE_VIDEO)
     {
       video_stream = i;
       continue;
