--- a/crypto/conf/conf_mod.c	2022-10-12 20:29:28.444106891 +0900
+++ b/crypto/conf/conf_mod.c	2022-10-12 20:29:34.577488900 +0900
@@ -545,7 +545,7 @@
 {
 	char *file = NULL;
 
-	if (asprintf(&file, "%s/openssl.cnf",
+	if (asprintf(&file, "%s/libressl.cnf",
 	    X509_get_default_cert_area()) == -1)
 		return (NULL);
 	return file;
--- a/apps/openssl/apps.c	2022-10-12 20:33:17.555896661 +0900
+++ b/apps/openssl/apps.c	2022-10-12 20:33:39.866067425 +0900
@@ -1134,7 +1134,7 @@
 	const char *t = X509_get_default_cert_area();
 	char *p;
 
-	if (asprintf(&p, "%s/openssl.cnf", t) == -1)
+	if (asprintf(&p, "%s/libressl.cnf", t) == -1)
 		return NULL;
 	return p;
 }
