diff --git a/include/fmt/core.h b/include/fmt/core.h
index 6a0846f..39d3e0e 100644
--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -224,7 +224,7 @@ namespace internal {
 // A workaround for gcc 4.8 to make void_t work in a SFINAE context.
 template <typename... Ts> struct void_t_impl { using type = void; };
 
-void assert_fail(const char* file, int line, const char* message);
+FMT_API void assert_fail(const char* file, int line, const char* message);
 
 #ifndef FMT_ASSERT
 #  ifdef NDEBUG
