diff --git a/tests/dependencies/test_closure.py b/tests/dependencies/test_closure.py
index 6231438cd..b5f2b0515 100644
--- a/tests/dependencies/test_closure.py
+++ b/tests/dependencies/test_closure.py
@@ -83,7 +83,7 @@ class Package:
         snake_case_name = name.replace("-", "_")
         for sitepackages in site.getsitepackages():
             for filename in os.listdir(sitepackages):
-                if fnmatch.fnmatch(filename, f"{snake_case_name}-*.dist-info"):
+                if fnmatch.fnmatch(filename, f"{snake_case_name}-*.dist-info") or fnmatch.fnmatch(filename, f"{snake_case_name}-*.egg-info"):
                     return importlib.metadata.Distribution.at(
                         os.path.join(sitepackages, filename)
                     )
