--- a/awscli/autoprompt/prompttoolkit.py  2025-04-25 20:05:42.000000000 +0200
+++ b/awscli/autoprompt/prompttoolkit.py  2025-05-12 18:22:21.449461488 +0200
@@ -15,6 +15,19 @@
 import sys
 from contextlib import contextmanager, nullcontext

+import importlib.metadata
+
+# Patch metadata.version to prevent crash due to missing dist-info
+real_metadata_version = importlib.metadata.version
+
+def safe_version(name):
+    if name == "prompt_toolkit":
+        return "0.0.0"  # dummy version
+    return real_metadata_version(name)
+
+importlib.metadata.version = safe_version
+
+
 from prompt_toolkit.application import Application
 from prompt_toolkit.completion import Completer, Completion, ThreadedCompleter
 from prompt_toolkit.document import Document
