diff -up dosbox-0.74/src/gui/sdl_mapper.cpp~ dosbox-0.74/src/gui/sdl_mapper.cpp
--- dosbox-0.74/src/gui/sdl_mapper.cpp~	2010-05-10 20:58:06.000000000 +0200
+++ dosbox-0.74/src/gui/sdl_mapper.cpp	2013-05-10 20:53:02.133112189 +0200
@@ -2384,9 +2384,11 @@ void MAPPER_StartUp(Section * sec) {
 	mapper.sticks.num=0;
 	mapper.sticks.num_groups=0;
 	Bitu i;
-	for (i=0; i<16; i++) {
+	for (i=0; i<MAX_VJOY_BUTTONS; i++) {
 		virtual_joysticks[0].button_pressed[i]=false;
 		virtual_joysticks[1].button_pressed[i]=false;
+	}
+	for (i=0; i<16; i++) {
 		virtual_joysticks[0].hat_pressed[i]=false;
 		virtual_joysticks[1].hat_pressed[i]=false;
 	}
