diff -up ortp-0.16.1/src/stun.c.unusedvar ortp-0.16.1/src/stun.c
--- ortp-0.16.1/src/stun.c.unusedvar	2011-03-15 12:35:33.000000000 +0000
+++ ortp-0.16.1/src/stun.c	2011-03-15 12:35:33.000000000 +0000
@@ -1191,7 +1191,7 @@ stunRand(void)
 	int maxFd=0;
 	struct timeval tv;
 	int e;
-	int res;
+	__attribute__((__unused__)) int res;
 
         int fd=open("/dev/random",O_RDONLY);
 
@@ -1985,7 +1985,7 @@ stunSendTest( Socket myFd, StunAddress4 
 	
    bool_t changePort=FALSE;
    bool_t changeIP=FALSE;
-   bool_t discard=FALSE;
+   __attribute__((__unused__)) bool_t discard=FALSE;
 
    StunMessage req;
    char buf[STUN_MAX_MESSAGE_SIZE];
@@ -2154,7 +2154,7 @@ stunNatType( StunAddress4 *dest, 
 
    bool_t respTestI=FALSE;
    bool_t isNat=TRUE;
-   StunAddress4 testIchangedAddr;
+   __attribute__((__unused__)) StunAddress4 testIchangedAddr;
    StunAddress4 testImappedAddr;
    bool_t respTestI2=FALSE; 
    bool_t mappedIpSame = TRUE;
diff -up ortp-0.16.1/src/tests/rtprecv.c.unusedvar ortp-0.16.1/src/tests/rtprecv.c
--- ortp-0.16.1/src/tests/rtprecv.c.unusedvar	2011-03-15 12:36:28.000000000 +0000
+++ ortp-0.16.1/src/tests/rtprecv.c	2011-03-15 12:36:53.000000000 +0000
@@ -162,7 +162,7 @@ int main(int argc, char*argv[])
 			if (err>0) stream_received=1;
 			/* this is to avoid to write to disk some silence before the first RTP packet is returned*/	
 			if ((stream_received) && (err>0)) {
-				size_t ret = fwrite(buffer,1,err,outfile);
+				__attribute__((__unused__)) size_t ret = fwrite(buffer,1,err,outfile);
 				if (sound_fd>0)
 					ret = write(sound_fd,buffer,err);
 			}
