diff -up ortp-0.16.1/src/stun.c.retval ortp-0.16.1/src/stun.c
--- ortp-0.16.1/src/stun.c.retval	2010-09-04 10:14:53.000000000 +0200
+++ ortp-0.16.1/src/stun.c	2010-09-04 10:15:10.000000000 +0200
@@ -1191,6 +1191,7 @@ stunRand(void)
 	int maxFd=0;
 	struct timeval tv;
 	int e;
+	int res;
 
         int fd=open("/dev/random",O_RDONLY);
 
@@ -1213,7 +1214,7 @@ stunRand(void)
            closesocket(fd);
 	   return random();
 	}
-	read(fd,&tick,sizeof(tick));
+	res = read(fd,&tick,sizeof(tick));
 	closesocket(fd);
       }
 #else
