From 11ef1f4508b89f76ad1040942c5a90e0eb33f3fe Mon Sep 17 00:00:00 2001
From: Wojtek Kaniewski <wojtekka@toxygen.net>
Date: Thu, 10 Jul 2025 21:27:48 +0200
Subject: [PATCH] Strict prototypes (#19)

---
 6tunnel.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/6tunnel.c b/6tunnel.c
index cf880e7..e40008b 100644
--- a/6tunnel.c
+++ b/6tunnel.c
@@ -556,7 +556,7 @@ void map_read(void)
 	fclose(f);
 }
 
-void sighup()
+void sighup(int unused)
 {
 	source_map_destroy();
 	map_read();
@@ -564,7 +564,7 @@ void sighup()
 	signal(SIGHUP, sighup);
 }
 
-void sigchld()
+void sigchld(int unused)
 {
 	while (waitpid(-1, NULL, WNOHANG) > 0) {
 		debug("child process exited\n");
@@ -574,7 +574,7 @@ void sigchld()
 	signal(SIGCHLD, sigchld);
 }
 
-void sigterm()
+void sigterm(int unused)
 {
 	if (pid_file != NULL)
 		unlink(pid_file);
