gtest needs 14
--
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9af4c45..597af4a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -135,7 +135,7 @@ add_library(${PROJECT_NAME} ${SOURCE_LITEHTML})
 set_target_properties(${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_LIB_VERSION} SOVERSION ${PROJECT_SO_VERSION})
 
 set_target_properties(${PROJECT_NAME} PROPERTIES
-    CXX_STANDARD 11
+    CXX_STANDARD 14
     C_STANDARD 99
     PUBLIC_HEADER "${HEADER_LITEHTML}"
 )
@@ -189,7 +189,7 @@ if (BUILD_TESTING)
     )
 
     set_target_properties(${TEST_NAME} PROPERTIES
-        CXX_STANDARD 11
+        CXX_STANDARD 14
         C_STANDARD 99
         PUBLIC_HEADER "${HEADER_LITEHTML}"
     )
