From a788e5f7b449dd701adc642c7cfb129f1895bd3e Mon Sep 17 00:00:00 2001
From: fwcd <fwcd@fwcd.dev>
Date: Sat, 18 Jan 2025 23:56:53 +0100
Subject: [PATCH] Fix unix permission literal

---
 server/build.gradle.kts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/build.gradle.kts b/server/build.gradle.kts
index 9b4c9bb4..420dde3b 100644
--- a/server/build.gradle.kts
+++ b/server/build.gradle.kts
@@ -20,7 +20,7 @@ application {
     description = "Code completions, diagnostics and more for Kotlin"
     applicationDefaultJvmArgs = listOf("-DkotlinLanguageServer.version=$version")
     applicationDistribution.into("bin") {
-        filePermissions { unix(755) }
+        filePermissions { unix("755".toInt(radix = 8)) }
     }
 }
 
