diff -urw orig/src/Makefile.in src/src/Makefile.in
--- orig/src/Makefile.in	2015-02-09 14:14:38.576648826 +0000
+++ src/src/Makefile.in	2015-02-09 14:15:11.662322918 +0000
@@ -455,6 +455,7 @@
 ifeq ($(CANNOT_DUMP),yes)
 	ln -f temacs$(EXEEXT) $@
 else
+	paxmark -r temacs$(EXEEXT)
 	LC_ALL=C $(RUN_TEMACS) -batch -l loadup dump
 	$(PAXCTL_if_present) -zex $@
 	ln -f $@ bootstrap-emacs$(EXEEXT)
@@ -659,6 +660,7 @@
 ifeq ($(CANNOT_DUMP),yes)
 	ln -f temacs$(EXEEXT) $@
 else
+	paxmark -r temacs$(EXEEXT)
 	$(RUN_TEMACS) --batch --load loadup bootstrap
 	$(PAXCTL_if_present) -zex emacs$(EXEEXT)
 	mv -f emacs$(EXEEXT) $@
