--- a/mozilla/nsprpub/lib/libc/src/Makefile.in.orig	2005-06-01 14:28:26.000000000 +0000
+++ b/mozilla/nsprpub/lib/libc/src/Makefile.in	2006-01-15 02:50:39.000000000 +0000
@@ -112,6 +112,10 @@
 MKSHLIB += -R '$$ORIGIN'
 endif
 
+ifeq ($(OS_ARCH), Linux)
+DSO_LDOPTS +=-Wl,-R,'$$ORIGIN'
+endif
+
 ifeq ($(OS_ARCH),OS2)
 MAPFILE = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).def
 GARBAGE += $(MAPFILE)
--- a/mozilla/nsprpub/lib/ds/Makefile.in.orig	2005-06-01 14:28:25.000000000 +0000
+++ b/mozilla/nsprpub/lib/ds/Makefile.in	2006-01-15 02:52:30.000000000 +0000
@@ -102,6 +102,10 @@
 MKSHLIB += -R '$$ORIGIN'
 endif
 
+ifeq ($(OS_ARCH), Linux)
+DSO_LDOPTS += -Wl,-R,'$$ORIGIN'
+endif
+
 ifeq ($(OS_ARCH),OS2)
 MAPFILE = $(OBJDIR)/$(LIBRARY_NAME)$(LIBRARY_VERSION).def
 GARBAGE += $(MAPFILE)
--- a/mozilla/nsprpub/pr/src/Makefile.in.orig	2005-06-01 14:28:27.000000000 +0000
+++ b/mozilla/nsprpub/pr/src/Makefile.in	2006-01-15 03:29:36.000000000 +0000
@@ -168,6 +168,7 @@
 else
 OS_LIBS		= -ldl
 endif
+DSO_LDOPTS +=-Wl,-R,'$$ORIGIN'
 endif
 
 ifeq ($(OS_ARCH),HP-UX)
--- a/mozilla/nsprpub/config/nspr-config.in.orig	2005-05-11 00:53:41.000000000 +0000
+++ b/mozilla/nsprpub/config/nspr-config.in	2006-01-15 06:37:58.000000000 +0000
@@ -122,7 +122,7 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-      libdirs=-L$libdir
+      libdirs="-Wl,-R$libdir -L$libdir"
       if test -n "$lib_plds"; then
 	libdirs="$libdirs -lplds${major_version}"
       fi
