diff --git a/macros/largefile-check.m4 b/macros/largefile-check.m4
index 8610f5e..f3282ca 100644
--- a/macros/largefile-check.m4
+++ b/macros/largefile-check.m4
@@ -60,7 +60,7 @@ if test "$enable_largefile" != no; then
     AC_TRY_RUN([#include <stdio.h>
 #include <stdlib.h>
 #include <sys/stat.h>
-main() { exit((sizeof(off_t) == 8) ? 0 : 1); }],
+int main() { exit((sizeof(off_t) == 8) ? 0 : 1); }],
 netatalk_cv_SIZEOF_OFF_T=yes,netatalk_cv_SIZEOF_OFF_T=no,netatalk_cv_SIZEOF_OFF_T=cross)])
 
     AC_MSG_CHECKING([if large file support is available])
