--- ./Gemfile.orig
+++ ./Gemfile
@@ -4,11 +4,11 @@
   abort "Redmine requires Bundler 1.5.0 or higher (you're using #{Bundler::VERSION}).\nPlease update with 'gem update bundler'." 
 end
 
-gem "rails", "4.2.0"
+gem "rails", "~> 4.2.0"
 gem "jquery-rails", "~> 3.1.1"
 gem "coderay", "~> 1.1.0"
 gem "builder", ">= 3.0.4"
-gem "request_store", "1.0.5"
+gem "request_store", "~> 1.0.5"
 gem "mime-types"
 gem "protected_attributes"
 gem "actionpack-action_caching"
@@ -89,15 +89,15 @@
 end
 
 group :test do
-  gem "minitest"
-  gem "rails-dom-testing"
-  gem "mocha"
-  gem "simplecov", "~> 0.9.1", :require => false
+#  gem "minitest"
+#  gem "rails-dom-testing"
+#  gem "mocha"
+#  gem "simplecov", "~> 0.9.1", :require => false
   # For running UI tests
-  gem "capybara"
-  gem "selenium-webdriver"
+#  gem "capybara"
+#  gem "selenium-webdriver"
   # building ffi 1.9.7 fails in Ubuntu: https://github.com/ffi/ffi/issues/414
-  gem "ffi", "1.9.6"
+#  gem "ffi", "1.9.6"
 end
 
 local_gemfile = File.join(File.dirname(__FILE__), "Gemfile.local")
