--- a/bin/common/check_unique_constraint.py
+++ b/bin/common/check_unique_constraint.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import argparse, collections, psycopg2, sys, tempfile
 
--- a/bin/common/dump_partition.py
+++ b/bin/common/dump_partition.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import argparse, hashlib, os, os.path, psycopg2, subprocess, sys
 
--- a/bin/common/vacuum_maintenance.py
+++ b/bin/common/vacuum_maintenance.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 import argparse, psycopg2, sys
 
