Taken from https://github.com/notandy/ympd/pull/191/files

diff --git a/src/mpd_client.c b/src/mpd_client.c
index 7271984..1f086e4 100644
--- a/src/mpd_client.c
+++ b/src/mpd_client.c
@@ -29,6 +29,7 @@
 
 /* forward declaration */
 static int mpd_notify_callback(struct mg_connection *c, enum mg_event ev);
+struct t_mpd mpd;
 
 const char * mpd_cmd_strs[] = {
     MPD_CMDS(GEN_STR)
diff --git a/src/mpd_client.h b/src/mpd_client.h
index dd78af9..386a31d 100644
--- a/src/mpd_client.h
+++ b/src/mpd_client.h
@@ -92,7 +92,9 @@ struct t_mpd {
 
     int song_id;
     unsigned queue_version;
-} mpd;
+};
+
+extern struct t_mpd mpd;
 
 struct t_mpd_client_session {
     int song_id;
