See https://github.com/OpenNebula/one/commit/030b5770336f61f69273275ae0e10d7014376d63

--- a/share/rubygems/generate
+++ b/share/rubygems/generate
@@ -97,7 +97,7 @@
         :summary => 'OpenNebula Client API',
         :description => 'Libraries needed to talk to OpenNebula',
         :dependencies => [
-            ['nokogiri','<1.16'],
+            'nokogiri',
             'json',
             'rbvmomi2',
             'xmlrpc',
