--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@
 	CGO_ENABLED=0 $(GOBUILD) -trimpath $(GOFLAGS) -ldflags '${LDFLAGS}' $(GOBUILD_ADDITIONAL_ARGS) \
 		 -o '${GOBUILD_OUTPUT}' $(GOBUILD_PACKAGES)
 
-build: GOFLAGS = -v -pgo=auto
+build: GOFLAGS += -v -pgo=auto
 build: GOBUILD_OUTPUT = ./bin/nuclei
 build: GOBUILD_PACKAGES = cmd/nuclei/main.go
 build: go-build
@@ -75,7 +75,7 @@
 syntax-docs:
 	./bin/docgen SYNTAX-REFERENCE.md nuclei-jsonschema.json
 
-test: GOFLAGS = -race -v
+test: GOFLAGS += -race -v
 test:
 	$(GOTEST) $(GOFLAGS) ./...
 
