--- a/configure	Tue Jan 12 20:44:32 2010
+++ b/configure	Tue Jan 12 20:46:59 2010
@@ -21380,63 +21380,6 @@
   fi
 
 
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libxul >= " >&5
-$as_echo_n "checking for libxul >= ... " >&6; }
-  if $PKG_CONFIG "--atleast-version=" "libxul" >/dev/null 2>&1; then
-    GECKO_VERSION=`$PKG_CONFIG --modversion "libxul"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $GECKO_VERSION" >&5
-$as_echo "$GECKO_VERSION" >&6; }
-
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking GECKO_CFLAGS" >&5
-$as_echo_n "checking GECKO_CFLAGS... " >&6; }
-    GECKO_CFLAGS=`$PKG_CONFIG --cflags "libxul"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $GECKO_CFLAGS" >&5
-$as_echo "$GECKO_CFLAGS" >&6; }
-
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking GECKO_LIBS" >&5
-$as_echo_n "checking GECKO_LIBS... " >&6; }
-    GECKO_LIBS=`$PKG_CONFIG --libs "libxul"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: $GECKO_LIBS" >&5
-$as_echo "$GECKO_LIBS" >&6; }
-
-    GECKO_REQUIRED_VERSION=
-
-
-
-
-
-
-
-  elif $PKG_CONFIG --exists "libxul" >/dev/null 2>&1; then
-    xdt_cv_version=`$PKG_CONFIG --modversion "libxul"`
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: found, but $xdt_cv_version" >&5
-$as_echo "found, but $xdt_cv_version" >&6; }
-
-
-      echo "*** The required package libxul was found on your system,"
-      echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade libxul to atleast version , or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you installed"
-      echo "*** the new version of the package in a nonstandard prefix so"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  else
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: not found" >&5
-$as_echo "not found" >&6; }
-
-
-      echo "*** The required package libxul was not found on your system."
-      echo "*** Please install libxul (atleast version ) or adjust"
-      echo "*** the PKG_CONFIG_PATH environment variable if you"
-      echo "*** installed the package in a nonstandard prefix so that"
-      echo "*** pkg-config is able to find it."
-      exit 1
-
-  fi
-
-
-
 $as_echo "#define BUILD_BROWSER_PLUGIN 1" >>confdefs.h
 
 fi
