--- a/cachefilesd.c
+++ b/cachefilesd.c
@@ -53,6 +53,14 @@
 #include <sys/vfs.h>
 #include <sys/stat.h>
 
+/* explicit musl 64-bit definitions no longer exist */
+#ifndef stat64
+#define stat64 stat
+#endif
+#ifndef fstatat64
+#define fstatat64 fstatat
+#endif
+
 typedef enum objtype {
 	OBJTYPE_INDEX,
 	OBJTYPE_DATA,
Common subdirectories: cachefilesd-inotify-0.11.0.orig/redhat and cachefilesd-inotify-0.11.0/redhat
Common subdirectories: cachefilesd-inotify-0.11.0.orig/selinux and cachefilesd-inotify-0.11.0/selinux
