--- edk2-edk2-stable202508.orig/edk2-platforms/Features/Ext4Pkg/Ext4Dxe/Collation.c
+++ edk2-edk2-stable202508/edk2-platforms/Features/Ext4Pkg/Ext4Dxe/Collation.c
@@ -66,7 +66,6 @@
   UINTN                           NumHandles;
   EFI_HANDLE                      *Handles;
   EFI_UNICODE_COLLATION_PROTOCOL  *Uci;
-  BOOLEAN                         Iso639Language;
   CHAR8                           *Language;
   EFI_STATUS                      RetStatus;
   EFI_STATUS                      Status;
