diff --git a/repo-projects/Directory.Build.targets.orig b/repo-projects/Directory.Build.targets
index e1a5734..610f9cd 100644
--- a/repo-projects/Directory.Build.targets.orig
+++ b/repo-projects/Directory.Build.targets
@@ -238,8 +238,6 @@
           Inputs="$(MSBuildProjectFullPath)"
           Outputs="$(RepoCompletedSemaphorePath)Build.complete">
 
-    <Exec Command="$(DotnetToolCommand) build-server shutdown" />
-
     <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Building $(ProjectBuildReason)" />
     <Message Importance="High" Text="Running command:" />
     <Message Importance="High" Text="  $(BuildCommand)" Condition="'$(BuildCommand)' != ''" />
diff --git a/build.sh.orig b/build.sh
index a1224e4..068c8e0 100755
--- a/build.sh.orig
+++ b/build.sh
@@ -253,15 +253,10 @@ initDistroRidGlobal "$os" "$arch" ""
 
 LogDateStamp=$(date +"%m%d%H%M%S")
 
-"$CLI_ROOT/dotnet" build-server shutdown
-
 if [ "$alternateTarget" == "true" ]; then
   export NUGET_PACKAGES=$NUGET_PACKAGES/smoke-tests
   "$CLI_ROOT/dotnet" msbuild "$SCRIPT_ROOT/build.proj" -bl:"$SCRIPT_ROOT/artifacts/log/Debug/BuildTests_$LogDateStamp.binlog" -flp:"LogFile=$SCRIPT_ROOT/artifacts/logs/BuildTests_$LogDateStamp.log" -clp:v=m ${MSBUILD_ARGUMENTS[@]} "$@"
 else
   "$CLI_ROOT/dotnet" msbuild "$SCRIPT_ROOT/eng/tools/init-build.proj" -bl:"$SCRIPT_ROOT/artifacts/log/Debug/BuildXPlatTasks_$LogDateStamp.binlog" -flp:LogFile="$SCRIPT_ROOT/artifacts/logs/BuildXPlatTasks_$LogDateStamp.log" -t:PrepareOfflineLocalTools ${MSBUILD_ARGUMENTS[@]} "$@"
-  # kill off the MSBuild server so that on future invocations we pick up our custom SDK Resolver
-  "$CLI_ROOT/dotnet" build-server shutdown
-
   "$CLI_ROOT/dotnet" msbuild "$SCRIPT_ROOT/build.proj" -bl:"$SCRIPT_ROOT/artifacts/log/Debug/Build_$LogDateStamp.binlog" -flp:"LogFile=$SCRIPT_ROOT/artifacts/logs/Build_$LogDateStamp.log" ${MSBUILD_ARGUMENTS[@]} "$@"
 fi
