diff -rupN libretro-bsnes.orig/.git/FETCH_HEAD libretro-bsnes/.git/FETCH_HEAD
--- libretro-bsnes.orig/.git/FETCH_HEAD	1970-01-01 01:00:00.000000000 +0100
+++ libretro-bsnes/.git/FETCH_HEAD	2018-12-07 18:31:25.943337424 +0100
@@ -0,0 +1,3 @@
+2a0b1a4f0b47100a781e025b085cb7ed88798527	not-for-merge	branch '096' of /home/alucryd/Packages/src/libretro-bsnes
+c6e90ddcdfdc5ca1b56319df662912b3c026d78f	not-for-merge	branch 'libretro' of /home/alucryd/Packages/src/libretro-bsnes
+95350550af6c76b57593dea980b5c3a27e5bf09e	not-for-merge	branch 'master' of /home/alucryd/Packages/src/libretro-bsnes
Binary files libretro-bsnes.orig/.git/index and libretro-bsnes/.git/index differ
diff -rupN libretro-bsnes.orig/Makefile libretro-bsnes/Makefile
--- libretro-bsnes.orig/Makefile	2018-12-07 18:09:48.766873570 +0100
+++ libretro-bsnes/Makefile	2018-12-07 18:22:39.006021877 +0100
@@ -32,7 +32,8 @@ sfc_lagfix := 1
 ifeq ($(DEBUG), 1)
   flags := -I. -Ilibco -O0 -g
 else
-  flags := -I. -Ilibco -O3 -fomit-frame-pointer
+  flags := $(CFLAGS) -I. -Ilibco -fomit-frame-pointer
+  link := $(LDFLAGS)
 endif
 
 cflags := -std=gnu99 -xc
diff -rupN libretro-bsnes.orig/target-libretro/Makefile libretro-bsnes/target-libretro/Makefile
--- libretro-bsnes.orig/target-libretro/Makefile	2018-12-07 18:09:48.813540589 +0100
+++ libretro-bsnes/target-libretro/Makefile	2018-12-07 18:31:23.149982988 +0100
@@ -46,7 +46,7 @@ obj/libretro-$(profile).o: $(ui)/libretr
 #targets
 build: $(objects)
 ifeq ($(platform),linux)
-	$(compiler) -o out/bsnes_$(profile)_libretro.so -shared $(objects) -ldl -Wl,--no-undefined -Wl,--version-script=$(ui)/link.T
+	$(compiler) -o out/bsnes_$(profile)_libretro.so $(link) -shared $(objects) -ldl -Wl,--no-undefined -Wl,--version-script=$(ui)/link.T
 else ifneq (,$(findstring ios,$(platform)))
   ifeq ($(platform),ios-arm64)
 		$(compiler) -o out/bsnes_$(profile)_libretro_ios.dylib -dynamiclib $(objects) -isysroot $(IOSSDK) -arch arm64
