--- a/Cargo.toml
+++ b/Cargo.toml
@@ -68,7 +68,7 @@
 path-slash = "0.1.4"
 rayon = "1.5.1"
 regex = "1.5.4"
-rusqlite = { version = "0.26.1", features = ["bundled"] }
+rusqlite = { version = "0.26.1" }
 tempfile = "3.2.0"
 textwrap = "0.14.2"
 tracing = "0.1.26"
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1284,7 +1284,6 @@
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "abd5850c449b40bacb498b2bbdfaff648b1b055630073ba8db499caf2d0ea9f2"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
