diff --git a/data/meson.build b/data/meson.build
index cc924ae..63dc69b 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -86,16 +86,18 @@ if get_option('with_sysprofd') == 'bundled'
   )
 
   systemdunitdir = get_option('systemdunitdir')
-  if systemdunitdir == ''
-    systemdunitdir = dependency('systemd').get_pkgconfig_variable('systemdsystemunitdir')
+  if systemdunitdir != 'no'
+    if systemdunitdir == ''
+      systemdunitdir = dependency('systemd').get_pkgconfig_variable('systemdsystemunitdir')
+    endif
+    configure_file(
+      input: 'sysprof2.service.in',
+      output: 'sysprof2.service',
+      configuration: sysprofdconf,
+      install: true,
+      install_dir: systemdunitdir,
+    )
   endif
-  configure_file(
-    input: 'sysprof2.service.in',
-    output: 'sysprof2.service',
-    configuration: sysprofdconf,
-    install: true,
-    install_dir: systemdunitdir,
-  )
 
   i18n.merge_file(
     input: 'org.gnome.sysprof2.policy.in',
