--- a/variables.cmake
+++ b/variables.cmake
@@ -428,9 +428,9 @@ endif ()
 # udev rules
 if(UNIX AND NOT APPLE)
     if (${INSTALL_ROOT} STREQUAL "/")
-        set(UDEVRULESDIR "/etc/udev/rules.d")
+        set(UDEVRULESDIR "/usr/lib/udev/rules.d")
     else()
-        set(UDEVRULESDIR "${INSTALL_ROOT}/etc/udev/rules.d")
+        set(UDEVRULESDIR "${INSTALL_ROOT}/usr/lib/udev/rules.d")
     endif()
 endif()
 
