--- a/meson.build
+++ b/meson.build
@@ -260,7 +260,7 @@ glycin_opt = get_option('glycin').enable_auto_if(host_machine.system() == 'linux
 if glycin_opt.disabled()
   glycin_dep = disabler()
 else
-  glycin_dep = dependency('glycin-2', version: '>= 2.2.alpha.7', required: glycin_opt)
+  glycin_dep = dependency('glycin-2', version: '>= 2.2.alpha7', required: glycin_opt)
 endif
 
 # Check which loaders should be built into gdk-pixbuf
