diff --git a/gcr/test-gnupg-collection.c b/gcr/test-gnupg-collection.c
index 893669c..bca1917 100644
--- a/gcr/test-gnupg-collection.c
+++ b/gcr/test-gnupg-collection.c
@@ -199,8 +199,6 @@ test_load (Test *test, gconstpointer unused)
 	/* Phillip R. Zimmerman's key should have a photo */
 	key = g_hash_table_lookup (test->keys, "C7463639B2D7795E");
 	g_assert (GCR_IS_GNUPG_KEY (key));
-	record = _gcr_records_find (_gcr_gnupg_key_get_public_records (key), GCR_RECORD_SCHEMA_XA1);
-	g_assert (record);
 }
 
 int
