Fix -Wimplicit-function-declaration errors with gcc 14.

--- Droplet-2.0-origin/examples/srwskey.c
+++ Droplet-2.0/examples/srwskey.c
@@ -4,7 +4,9 @@
  * embed its own routines
  */
 
+#include <ctype.h> /* For isdigit */
 #include <droplet.h>
+#include <droplet/utils.h> /* For dpl_rand_u64, dpl_rand_u62 */
 #include <droplet/srws/backend.h>
 
 void
--- Droplet-2.0-origin/libdroplet/src/conn.c
+++ Droplet-2.0/libdroplet/src/conn.c
@@ -31,6 +31,7 @@
  *
  * https://github.com/scality/Droplet
  */
+#include <sys/uio.h> /* For writev */
 #include "dropletp.h"
 
 //#define DPRINTF(fmt,...) fprintf(stderr, fmt, ##__VA_ARGS__)
