diff --git a/libdroplet/cdmi/replyparser.c b/libdroplet/cdmi/replyparser.c
index d5791cc..1689dff 100644
--- a/libdroplet/cdmi/replyparser.c
+++ b/libdroplet/cdmi/replyparser.c
@@ -32,7 +32,7 @@
  * https://github.com/scality/Droplet
  */
 #include "dropletp.h"
-#include <json/json.h>
+#include <json-c/json.h>
 
 //#define DPRINTF(fmt,...) fprintf(stderr, fmt, ##__VA_ARGS__)
 #define DPRINTF(fmt,...)
diff --git a/libdroplet/include/dropletp.h b/libdroplet/include/dropletp.h
index f765c59..3867c43 100644
--- a/libdroplet/include/dropletp.h
+++ b/libdroplet/include/dropletp.h
@@ -58,7 +58,7 @@
 #include <openssl/rand.h>
 #include <libxml/parser.h>
 #include <libxml/tree.h>
-#include <json/json.h>
+#include <json-c/json.h>
 #include <sys/types.h>
 #include <pwd.h>
 #include <fcntl.h>
