--- ghostscript-9.18/base/lib.mak.orig	2017-04-14 19:02:04.166047598 +0000
+++ ghostscript-9.18/base/lib.mak	2017-04-14 19:03:54.614782450 +0000
@@ -1794,7 +1794,7 @@
  $(memory__h) $(gserror_h) $(gserrors_h) \
  $(gdebug_h) $(strimpl_h) $(sjpx_openjpeg_h) $(MAKEDIRS)
 	$(GLJPXOPJCC) $(GLO_)sjpx_openjpeg.$(OBJ) \
-		$(C_) -DOPJ_STATIC $(GLSRC)sjpx_openjpeg.c
+		$(C_) $(GLSRC)sjpx_openjpeg.c
 
 # ---------------- Pixel-difference filters ---------------- #
 # The Predictor facility of the LZW and Flate filters uses these.
