diff -up ghostscript-9.16/base/openjpeg.mak.system-openjpeg2 ghostscript-9.16/base/openjpeg.mak
--- ghostscript-9.16/base/openjpeg.mak.system-openjpeg2	2015-03-30 09:21:24.000000000 +0100
+++ ghostscript-9.16/base/openjpeg.mak	2015-03-30 17:05:55.160034798 +0100
@@ -16,6 +16,7 @@
 # makefile for Luratech lwf_jp2 library code.
 # Users of this makefile must define the following:
 #       SHARE_JPX - whether to compile in or link to the library
+#       SHARE_JPX_LIB - name of external library to link to
 #       JPXSRCDIR - the library source directory
 #
 # gs.mak and friends define the following:
@@ -105,7 +106,7 @@ $(OPEN_JPEG_GEN)openjpeg.dev : $(TOP_MAK
 # external link .dev
 $(OPEN_JPEG_GEN)openjpeg_1.dev : $(TOP_MAKEFILES) $(OPEN_JPEG_MAK) $(ECHOGS_XE) \
  $(MAKEDIRS)
-	$(SETMOD) $(OPEN_JPEG_GEN)openjpeg_1 -lib lib_openjpeg
+	$(SETMOD) $(OPEN_JPEG_GEN)openjpeg_1 -lib $(SHARE_JPX_LIB)
 
 # compile our own .dev
 $(OPEN_JPEG_GEN)openjpeg_0.dev : $(TOP_MAKEFILES) $(OPEN_JPEG_MAK) $(ECHOGS_XE) $(open_jpeg_OBJS) \
diff -up ghostscript-9.16/configure.ac.system-openjpeg2 ghostscript-9.16/configure.ac
--- ghostscript-9.16/configure.ac.system-openjpeg2	2015-03-30 09:21:24.000000000 +0100
+++ ghostscript-9.16/configure.ac	2015-03-30 17:05:08.563193858 +0100
@@ -1531,6 +1531,17 @@ if test "x$JPX_DECODER" = "x"; then
       JPXDEVS='$(PSD)jpx.dev'
     else
       AC_MSG_RESULT([no])
+      AC_MSG_CHECKING([for system OpenJPEG library])
+      if $PKGCONFIG --exists libopenjp2; then
+        AC_MSG_RESULT(yes)
+        JPX_DECODER=openjpeg
+        SHARE_JPX=1
+        SHARE_JPX_LIB="`$PKGCONFIG libopenjp2 --libs | sed -e 's,^-l,,'`"
+        JPX_AUTOCONF_CFLAGS="-DUSE_OPENJPEG_JP2 `$PKGCONFIG libopenjp2 --cflags`"
+        JPXDEVS='$(PSD)jpx.dev'
+      else
+        AC_MSG_RESULT(no)
+      fi
     fi
   fi
 fi
@@ -1539,6 +1550,7 @@ AC_SUBST(JPX_DECODER)
 AC_SUBST(JPX_AUTOCONF_CFLAGS)
 AC_SUBST(JPXDIR)
 AC_SUBST(SHARE_JPX)
+AC_SUBST(SHARE_JPX_LIB)
 AC_SUBST(JPXDEVS)
 
 dnl check if we can/should build the gtk loader
diff -up ghostscript-9.16/Makefile.in.system-openjpeg2 ghostscript-9.16/Makefile.in
--- ghostscript-9.16/Makefile.in.system-openjpeg2	2015-03-30 17:02:51.000000000 +0100
+++ ghostscript-9.16/Makefile.in	2015-03-30 17:05:08.563193858 +0100
@@ -248,6 +248,7 @@ JBIG2_CFLAGS=@JBIG2_AUTOCONF_CFLAGS@
 # and source location and configuration flags for compiling in
 JPX_LIB=@JPX_DECODER@
 SHARE_JPX=@SHARE_JPX@
+SHARE_JPX_LIB=@SHARE_JPX_LIB@
 JPXSRCDIR=@JPXDIR@
 JPX_CFLAGS=@JPX_AUTOCONF_CFLAGS@
 
