diff -Nur ghostscript-8.70.orig/base/configure.ac ghostscript-8.70/base/configure.ac
--- ghostscript-8.70.orig/base/configure.ac	2009-07-27 19:59:50.000000000 +0300
+++ ghostscript-8.70/base/configure.ac	2010-01-17 02:20:52.000000000 +0200
@@ -402,7 +402,7 @@
 	PNGDEVS="$PNGDEVS_ALL"
 else
 	AC_MSG_RESULT([no])
-	AC_CHECK_LIB(png, png_check_sig, [
+	AC_CHECK_LIB(png, png_sig_cmp, [
 	  AC_CHECK_HEADERS(png.h, [
 		SHARE_LIBPNG=1
 		PNGDEVS="$PNGDEVS_ALL"
diff -Nur ghostscript-8.70.orig/jbig2dec/configure.ac ghostscript-8.70/jbig2dec/configure.ac
--- ghostscript-8.70.orig/jbig2dec/configure.ac	2009-05-29 09:48:44.000000000 +0300
+++ ghostscript-8.70/jbig2dec/configure.ac	2010-01-17 02:21:25.000000000 +0200
@@ -34,7 +34,7 @@
   fi
   dnl libpng requires pow() which may be in libm
   AC_SEARCH_LIBS([pow], [m])
-  AC_CHECK_LIB([png], [png_check_sig], [
+  AC_CHECK_LIB([png], [png_sig_cmp], [
     AC_CHECK_LIB([z], [deflate], [
       AC_DEFINE(HAVE_LIBPNG, 1, [Define if libpng is available (-lpng)])
       LIBS="-lpng -lz $LIBS"
