--- a/CairoSVG.egg-info/requires.txt
+++ b/CairoSVG.egg-info/requires.txt
@@ -11,5 +11,3 @@
 [test]
 pytest-runner
 pytest-cov
-pytest-flake8
-pytest-isort
--- a/setup.cfg
+++ b/setup.cfg
@@ -45,8 +45,6 @@
 	tinycss2
 tests_require = 
 	pytest-cov
-	pytest-flake8
-	pytest-isort
 	pytest-runner
 python_requires = >= 3.4
 
@@ -64,8 +62,6 @@
 test = 
 	pytest-runner
 	pytest-cov
-	pytest-flake8
-	pytest-isort
 
 [bdist_wheel]
 python-tag = py3
@@ -75,8 +71,6 @@
 
 [tool:pytest]
 addopts = 
-	--flake8
-	--isort
 	--cov=cairosvg
 	--cov=test_non_regression.test_non_regression
 	--cov=test_non_regression.test_helpers
