diff --git a/repo-projects/Directory.Build.props.orig b/repo-projects/Directory.Build.props
index ca50071..782abb6 100644
--- a/repo-projects/Directory.Build.props.orig
+++ b/repo-projects/Directory.Build.props
@@ -144,7 +144,7 @@
     <CommonArgs>$(CommonArgs) /p:PreviouslySourceBuiltNupkgCacheDir="$(PreviouslySourceBuiltPackagesPath)"</CommonArgs>
     <CommonArgs Condition="Exists('$(SharedComponentsArtifactsPath)')">$(CommonArgs) /p:AdditionalSourceBuiltNupkgCacheDir="$(SharedComponentsArtifactsPath)"</CommonArgs>
     <CommonArgs>$(CommonArgs) /p:ReferencePackageNupkgCacheDir="$(ReferencePackagesDir)"</CommonArgs>
-    <CommonArgs>$(CommonArgs) /p:TrackPrebuiltUsageReportDir="$(ArtifactsLogRepoDir)"</CommonArgs>
+    <CommonArgs>$(CommonArgs) /p:TrackPrebuiltUsage="false"</CommonArgs>
   </PropertyGroup>
 
   <!-- Build specific args -->
