ax25-tools-0.0.10-rc5/tcpip/Makefile.in.orig
diff -uN a/ax25-tools-0.0.10-rc5/tcpip/Makefile.in.orig b/ax25-tools-0.0.10-rc5/tcpip/Makefile.in
--- ax25-tools-0.0.10-rc5/tcpip/Makefile.in.orig
+++ ax25-tools-0.0.10-rc5/tcpip/Makefile.in
@@ -14,7 +14,6 @@
 
 @SET_MAKE@
 
-
 VPATH = @srcdir@
 am__is_gnu_make = { \
   if test -z '$(MAKELEVEL)'; then \
@@ -87,28 +86,24 @@ POST_INSTALL = :
 NORMAL_UNINSTALL = :
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
-sbin_PROGRAMS = rip98d$(EXEEXT) ttylinkd$(EXEEXT)
+sbin_PROGRAMS = rip98d$(EXEEXT)
 subdir = tcpip
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
 am__aclocal_m4_deps = $(top_srcdir)/configure.ac
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
 	$(ACLOCAL_M4)
-DIST_COMMON = $(srcdir)/Makefile.am $(dist_doc_DATA) \
-	$(am__DIST_COMMON)
+DIST_COMMON = $(srcdir)/Makefile.am $(am__DIST_COMMON)
 mkinstalldirs = $(install_sh) -d
 CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 am__installdirs = "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(man5dir)" \
-	"$(DESTDIR)$(man8dir)" "$(DESTDIR)$(docdir)"
+	"$(DESTDIR)$(man8dir)"
 PROGRAMS = $(sbin_PROGRAMS)
 am_rip98d_OBJECTS = rip98d.$(OBJEXT) rip98r.$(OBJEXT) rip98t.$(OBJEXT)
 rip98d_OBJECTS = $(am_rip98d_OBJECTS)
 am__DEPENDENCIES_1 =
 rip98d_DEPENDENCIES = $(am__DEPENDENCIES_1)
-ttylinkd_SOURCES = ttylinkd.c
-ttylinkd_OBJECTS = ttylinkd.$(OBJEXT)
-ttylinkd_LDADD = $(LDADD)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -125,7 +120,7 @@ DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/rip98d.Po ./$(DEPDIR)/rip98r.Po \
-	./$(DEPDIR)/rip98t.Po ./$(DEPDIR)/ttylinkd.Po
+	./$(DEPDIR)/rip98t.Po
 am__mv = mv -f
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
@@ -139,8 +134,8 @@ AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
-SOURCES = $(rip98d_SOURCES) ttylinkd.c
-DIST_SOURCES = $(rip98d_SOURCES) ttylinkd.c
+SOURCES = $(rip98d_SOURCES)
+DIST_SOURCES = $(rip98d_SOURCES)
 am__can_run_installinfo = \
   case $$AM_UPDATE_INFO_DIR in \
     n|no|NO) false;; \
@@ -177,7 +172,6 @@ man5dir = $(mandir)/man5
 man8dir = $(mandir)/man8
 NROFF = nroff
 MANS = $(dist_man_MANS)
-DATA = $(dist_doc_DATA)
 am__tagged_files = $(HEADERS) $(SOURCES) $(TAGS_FILES) $(LISP)
 # Read a list of newline-separated strings from the standard input,
 # and print each of them once, without duplicates.  Input order is
@@ -294,6 +288,7 @@ pdfdir = @pdfdir@
 prefix = @prefix@
 program_transform_name = @program_transform_name@
 psdir = @psdir@
+runstatedir = @runstatedir@
 sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 srcdir = @srcdir@
@@ -302,9 +297,8 @@ target_alias = @target_alias@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-etcfiles = rip98d.conf ttylinkd.conf
-dist_man_MANS = rip98d.conf.5 rip98d.8 ttylinkd.conf.5 ttylinkd.8
-dist_doc_DATA = ttylinkd.README ttylinkd.INSTALL
+etcfiles = rip98d.conf 
+dist_man_MANS = rip98d.conf.5 rip98d.8 
 EXTRA_DIST = $(etcfiles)
 rip98d_SOURCES = \
 	rip98d.c	\
@@ -399,10 +393,6 @@ rip98d$(EXEEXT): $(rip98d_OBJECTS) $(rip98d_DEPENDENCIES) $(EXTRA_rip98d_DEPENDE
 	@rm -f rip98d$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(rip98d_OBJECTS) $(rip98d_LDADD) $(LIBS)
 
-ttylinkd$(EXEEXT): $(ttylinkd_OBJECTS) $(ttylinkd_DEPENDENCIES) $(EXTRA_ttylinkd_DEPENDENCIES) 
-	@rm -f ttylinkd$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(ttylinkd_OBJECTS) $(ttylinkd_LDADD) $(LIBS)
-
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
@@ -412,7 +402,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rip98d.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rip98r.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rip98t.Po@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/ttylinkd.Po@am__quote@ # am--include-marker
 
 $(am__depfiles_remade):
 	@$(MKDIR_P) $(@D)
@@ -519,27 +508,6 @@ uninstall-man8:
 	} | sed -e 's,.*/,,;h;s,.*\.,,;s,^[^8][0-9a-z]*$$,8,;x' \
 	      -e 's,\.[0-9a-z]*$$,,;$(transform);G;s,\n,.,'`; \
 	dir='$(DESTDIR)$(man8dir)'; $(am__uninstall_files_from_dir)
-install-dist_docDATA: $(dist_doc_DATA)
-	@$(NORMAL_INSTALL)
-	@list='$(dist_doc_DATA)'; test -n "$(docdir)" || list=; \
-	if test -n "$$list"; then \
-	  echo " $(MKDIR_P) '$(DESTDIR)$(docdir)'"; \
-	  $(MKDIR_P) "$(DESTDIR)$(docdir)" || exit 1; \
-	fi; \
-	for p in $$list; do \
-	  if test -f "$$p"; then d=; else d="$(srcdir)/"; fi; \
-	  echo "$$d$$p"; \
-	done | $(am__base_list) | \
-	while read files; do \
-	  echo " $(INSTALL_DATA) $$files '$(DESTDIR)$(docdir)'"; \
-	  $(INSTALL_DATA) $$files "$(DESTDIR)$(docdir)" || exit $$?; \
-	done
-
-uninstall-dist_docDATA:
-	@$(NORMAL_UNINSTALL)
-	@list='$(dist_doc_DATA)'; test -n "$(docdir)" || list=; \
-	files=`for p in $$list; do echo $$p; done | sed -e 's|^.*/||'`; \
-	dir='$(DESTDIR)$(docdir)'; $(am__uninstall_files_from_dir)
 
 ID: $(am__tagged_files)
 	$(am__define_uniq_tagged_files); mkid -fID $$unique
@@ -628,9 +596,9 @@ distdir-am: $(DISTFILES)
 	done
 check-am: all-am
 check: check-am
-all-am: Makefile $(PROGRAMS) $(MANS) $(DATA)
+all-am: Makefile $(PROGRAMS) $(MANS)
 installdirs:
-	for dir in "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(man5dir)" "$(DESTDIR)$(man8dir)" "$(DESTDIR)$(docdir)"; do \
+	for dir in "$(DESTDIR)$(sbindir)" "$(DESTDIR)$(man5dir)" "$(DESTDIR)$(man8dir)"; do \
 	  test -z "$$dir" || $(MKDIR_P) "$$dir"; \
 	done
 install: install-am
@@ -671,7 +639,6 @@ distclean: distclean-am
 		-rm -f ./$(DEPDIR)/rip98d.Po
 	-rm -f ./$(DEPDIR)/rip98r.Po
 	-rm -f ./$(DEPDIR)/rip98t.Po
-	-rm -f ./$(DEPDIR)/ttylinkd.Po
 	-rm -f Makefile
 distclean-am: clean-am distclean-compile distclean-generic \
 	distclean-tags
@@ -688,7 +655,7 @@ info: info-am
 
 info-am:
 
-install-data-am: install-dist_docDATA install-man
+install-data-am: install-man
 
 install-dvi: install-dvi-am
 
@@ -720,7 +687,6 @@ maintainer-clean: maintainer-clean-am
 		-rm -f ./$(DEPDIR)/rip98d.Po
 	-rm -f ./$(DEPDIR)/rip98r.Po
 	-rm -f ./$(DEPDIR)/rip98t.Po
-	-rm -f ./$(DEPDIR)/ttylinkd.Po
 	-rm -f Makefile
 maintainer-clean-am: distclean-am maintainer-clean-generic
 
@@ -736,8 +702,7 @@ ps: ps-am
 
 ps-am:
 
-uninstall-am: uninstall-dist_docDATA uninstall-man \
-	uninstall-sbinPROGRAMS
+uninstall-am: uninstall-man uninstall-sbinPROGRAMS
 
 uninstall-man: uninstall-man5 uninstall-man8
 
@@ -747,16 +712,15 @@ uninstall-man: uninstall-man5 uninstall-man8
 	clean-generic clean-sbinPROGRAMS cscopelist-am ctags ctags-am \
 	distclean distclean-compile distclean-generic distclean-tags \
 	distdir dvi dvi-am html html-am info info-am install \
-	install-am install-data install-data-am install-dist_docDATA \
-	install-dvi install-dvi-am install-exec install-exec-am \
-	install-html install-html-am install-info install-info-am \
-	install-man install-man5 install-man8 install-pdf \
-	install-pdf-am install-ps install-ps-am install-sbinPROGRAMS \
-	install-strip installcheck installcheck-am installdirs \
-	maintainer-clean maintainer-clean-generic mostlyclean \
-	mostlyclean-compile mostlyclean-generic pdf pdf-am ps ps-am \
-	tags tags-am uninstall uninstall-am uninstall-dist_docDATA \
-	uninstall-man uninstall-man5 uninstall-man8 \
+	install-am install-data install-data-am install-dvi \
+	install-dvi-am install-exec install-exec-am install-html \
+	install-html-am install-info install-info-am install-man \
+	install-man5 install-man8 install-pdf install-pdf-am \
+	install-ps install-ps-am install-sbinPROGRAMS install-strip \
+	installcheck installcheck-am installdirs maintainer-clean \
+	maintainer-clean-generic mostlyclean mostlyclean-compile \
+	mostlyclean-generic pdf pdf-am ps ps-am tags tags-am uninstall \
+	uninstall-am uninstall-man uninstall-man5 uninstall-man8 \
 	uninstall-sbinPROGRAMS
 
 .PRECIOUS: Makefile
