https://github.com/raspberrypi/linux/issues/731

diff --git a/drivers/video/bcm2708_fb.c b/drivers/video/bcm2708_fb.c
index b3b1e04..e791b62 100644
--- a/drivers/video/bcm2708_fb.c
+++ b/drivers/video/bcm2708_fb.c
@@ -625,6 +625,7 @@ static int bcm2708_fb_register(struct bcm2708_fb *fb)
 	 */
 
 	fb_set_var(&fb->fb, &fb->fb.var);
+	bcm2708_fb_set_par(&fb->fb);
 
 	print_debug("BCM2708FB: registering framebuffer (%dx%d@%d) (%d)\n", fbwidth
 		fbheight, fbdepth, fbswap);
