diff --git a/src/vtp.c b/src/vtp.c
index 0f918ff..ba51761 100644
--- a/src/vtp.c
+++ b/src/vtp.c
@@ -1526,7 +1526,7 @@ vtp_load_values(struct pcap_data *data, void *values)
             aux_short = (u_int16_t *)ptr;
 #ifdef LBL_ALIGN
             memcpy((void *)&aux_short2, (void *)aux_short, 4);
-            vtp->start_val = ntohs(aux_short2);
+            vtp->start_val = ntohs((uint16_t)aux_short2);
 #else
             vtp->start_val = ntohs(*aux_short);
 #endif            
