diff -ru xfsprogs-3.0.0.orig/doc/Makefile xfsprogs-3.0.0/doc/Makefile
--- xfsprogs-3.0.0.orig/doc/Makefile	2009-02-18 13:42:47.000000000 +0000
+++ xfsprogs-3.0.0/doc/Makefile	2009-02-18 13:43:10.000000000 +0000
@@ -13,7 +13,7 @@
 include $(BUILDRULES)
 
 CHANGES.gz:
-	$(ZIP) --best -c < CHANGES > $@
+	$(ZIP) -c < CHANGES > $@
 
 install: default
 	$(INSTALL) -m 755 -d $(PKG_DOC_DIR)
diff -ru xfsprogs-3.0.0.orig/include/buildmacros xfsprogs-3.0.0/include/buildmacros
--- xfsprogs-3.0.0.orig/include/buildmacros	2009-02-18 13:42:47.000000000 +0000
+++ xfsprogs-3.0.0/include/buildmacros	2009-02-18 13:43:46.000000000 +0000
@@ -128,7 +128,7 @@
 MAN_MAKERULE = \
 	@for f in *.[12345678] ""; do \
 		if test ! -z "$$f"; then \
-			$(ZIP) --best -c < $$f > $$f.gz; \
+			$(ZIP) -c < $$f > $$f.gz; \
 		fi; \
 	done
 
