--- a/test/runtests.py
+++ b/test/runtests.py
@@ -29,10 +29,10 @@
     sys.path.insert(0, joinpath(dn,'..'))
     sys.path.insert(0, dn)
     # it's okay to import now that sys.path is setup.
-    import test_grabber, test_byterange, test_mirror
+    import test_grabber, test_byterange
     suite = TestSuite( (test_grabber.suite(),
                         test_byterange.suite(),
-                        test_mirror.suite()) )
+                        ) )
     suite.description = 'urlgrabber tests'
     runner = TextTestRunner(stream=sys.stdout,
                             descriptions=descriptions,
--- a/test/test_grabber.py
+++ b/test/test_grabber.py
@@ -616,7 +616,12 @@
 
 def suite():
     tl = TestLoader()
-    return tl.loadTestsFromModule(sys.modules[__name__])
+    suite = TestSuite()
+    suite.addTest(tl.loadTestsFromTestCase(CommonRegetTests))
+    suite.addTest(tl.loadTestsFromTestCase(FailureTestCase))
+    suite.addTest(tl.loadTestsFromTestCase(URLGrabberModuleTestCase))
+    suite.addTest(tl.loadTestsFromTestCase(URLGrabberTestCase))
+    return suite
 
 if __name__ == '__main__':
     grabber.DEBUG = 0
