diff --git a/tests/conftest.py b/tests/conftest.py
new file mode 100644
index 0000000..1aabf2d
--- /dev/null
+++ b/tests/conftest.py
@@ -0,0 +1,11 @@
+import pytest
+import taglib
+import os
+
+@pytest.fixture
+def test_data():
+	return lambda filename: os.path.abspath(os.path.dirname(__file__)) + "/data/" + filename
+
+@pytest.fixture
+def test_file(test_data):
+	return lambda filename: taglib.File(test_data(filename))
