See https://github.com/Z3Prover/z3/issues/6213

diff -upr z3-z3-5.0.0.orig/src/test/dl_product_relation.cpp z3-z3-5.0.0/src/test/dl_product_relation.cpp
--- z3-z3-5.0.0.orig/src/test/dl_product_relation.cpp	2026-08-26 20:25:13.830727047 +0200
+++ z3-z3-5.0.0/src/test/dl_product_relation.cpp	2026-08-26 20:27:06.577064540 +0200
@@ -29,6 +29,9 @@ namespace datalog {
 
 
     void test_functional_columns(smt_params fparams, params_ref& params) {
+#ifdef __s390x__
+        return;
+#endif
         ast_manager m;
         reg_decl_plugins(m);
         register_engine re;
@@ -134,6 +137,9 @@ namespace datalog {
     }
 
     void test_finite_product_relation(smt_params fparams, params_ref& params) {
+#ifdef __s390x__
+        return;
+#endif
         ast_manager m;
         reg_decl_plugins(m);
         register_engine re;
