From eefb032e171ed2f6f20b93cdcc09b04d0e45e78d Mon Sep 17 00:00:00 2001
From: JailBird <jailbird@fdf.net>
Date: Wed, 3 Dec 2025 21:42:59 -0700
Subject: [PATCH] Add sockaddr cast for MUSL

---
 src/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/daemon.c b/src/daemon.c
index 399a92ca..5acbc9df 100644
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -76,7 +76,7 @@ static void sysd_notify_ready(void)
     if (fd < 0)
         log_fatal("Cannot create AF_UNIX socket");
 
-    const ssize_t strv = sendto(fd, msg, strlen(msg), 0, &sun, sun_len);
+    const ssize_t strv = sendto(fd, msg, strlen(msg), 0, (const struct sockaddr *)&sun, sun_len);
     if (strv < 0)
         log_fatal("sendto() systemd NOTIFY_SOCKET failed: %s", logf_errno());
 
