diff --git a/src/ftpd.c b/src/ftpd.c
index 60b95b6..6aee9bc 100644
--- a/src/ftpd.c
+++ b/src/ftpd.c
@@ -1080,8 +1080,6 @@ void dobanner(const int type)
 
 #endif
 
-#ifndef MINIMAL
-
 int modernformat(const char *file, char *target, size_t target_size,
                  const char * const prefix)
 {
@@ -1228,8 +1226,6 @@ void doallo(const off_t size)
     }
 }
 
-#endif
-
 void dositetime(void)
 {
     char tmp[64];

