--- micropython-1.12-orig/ports/unix/mpconfigport.mk
+++ micropython-1.12/ports/unix/mpconfigport.mk
@@ -24,10 +24,10 @@
 MICROPY_PY_FFI = 1
 
 # ussl module requires one of the TLS libraries below
-MICROPY_PY_USSL = 1
+MICROPY_PY_USSL = 0
 # axTLS has minimal size but implements only a subset of modern TLS
 # functionality, so may have problems with some servers.
-MICROPY_SSL_AXTLS = 1
+MICROPY_SSL_AXTLS = 0
 # mbedTLS is more up to date and complete implementation, but also
 # more bloated.
 MICROPY_SSL_MBEDTLS = 0
