diff --git a/lib/memzip/make-memzip.py b/lib/memzip/make-memzip.py
index 9730f5e..3a1f7a5 100755
--- a/lib/memzip/make-memzip.py
+++ b/lib/memzip/make-memzip.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Takes a directory of files and zips them up (as uncompressed files).
 # This then gets converted into a C data structure which can be read
diff --git a/ports/teensy/make-pins.py b/ports/teensy/make-pins.py
index 0f6c5f2..5d21d8f 100755
--- a/ports/teensy/make-pins.py
+++ b/ports/teensy/make-pins.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Creates the pin file for the Teensy."""
 
 from __future__ import print_function
diff --git a/tools/make-frozen.py b/tools/make-frozen.py
index 1051b52..06ee7a3 100755
--- a/tools/make-frozen.py
+++ b/tools/make-frozen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Create frozen modules structure for MicroPython.
 #
diff --git a/tools/pyboard.py b/tools/pyboard.py
index 16ee41f..17b7f98 100755
--- a/tools/pyboard.py
+++ b/tools/pyboard.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is part of the MicroPython project, http://micropython.org/
 #
diff --git a/tools/pydfu.py b/tools/pydfu.py
index 8c0220d..55ea952 100755
--- a/tools/pydfu.py
+++ b/tools/pydfu.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # This file is part of the OpenMV project.
 # Copyright (c) 2013/2014 Ibrahim Abdelkader <i.abdalkader@gmail.com>
 # This work is licensed under the MIT license, see the file LICENSE for
