diff --git a/src/porting.h b/src/porting.h
index 492c9d3..39499ef 100644
--- a/src/porting.h
+++ b/src/porting.h
@@ -59,6 +59,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #else
 	#include <unistd.h>
 	#include <stdint.h> //for uintptr_t
+	#include <sys/time.h>
+	#include <time.h>
 
 	#if (defined(linux) || defined(__linux) || defined(__GNU__)) && !defined(_GNU_SOURCE)
 		#define _GNU_SOURCE
@@ -190,8 +192,6 @@ void initIrrlicht(irr::IrrlichtDevice * );
 	}
 
 #else // Posix
-#include <sys/time.h>
-#include <time.h>
 #if defined(__MACH__) && defined(__APPLE__)
 #include <mach/clock.h>
 #include <mach/mach.h>
