Skip git commit info in the "version" command.

--- a/commands.go
+++ b/commands.go
@@ -643,9 +643,8 @@ func CommandUndo(conf Config, args []string, ctx, query Query) error {
 // CommandVersion prints version information for the dstask binary.
 func CommandVersion() {
 	fmt.Printf(
-		"Version: %s\nGit commit: %s\nBuild date: %s\n",
+		"Version: %s\nBuild date: %s\n",
 		VERSION,
-		GIT_COMMIT,
 		BUILD_DATE,
 	)
 }
