diff -urw src/spl-master/include/sys/sysmacros.h spl-master/include/sys/sysmacros.h
--- src/spl-master/include/sys/sysmacros.h	2014-06-06 17:04:01.416666671 +0000
+++ spl-master/include/sys/sysmacros.h	2014-06-06 17:02:23.586666828 +0000
@@ -104,7 +104,9 @@
 
 /* Missing macros
  */
+#ifndef PAGESIZE
 #define PAGESIZE			PAGE_SIZE
+#endif
 
 /* from Solaris sys/byteorder.h */
 #define BSWAP_8(x)	((x) & 0xff)
