diff --git a/3party/CMakeLists.txt b/3party/CMakeLists.txt
index dd851d7db21f..8f360a9ba780 100644
--- a/3party/CMakeLists.txt
+++ b/3party/CMakeLists.txt
@@ -57,7 +57,6 @@ endif()
 
 add_subdirectory(agg)
 add_subdirectory(bsdiff-courgette)
-add_subdirectory(glaze)
 add_subdirectory(minizip)
 add_subdirectory(open-location-code)
 add_subdirectory(opening_hours)
@@ -77,10 +76,13 @@ if (PLATFORM_DESKTOP)
   set(GLFW_BUILD_WAYLAND OFF CACHE BOOL "")
   # Disable ARC for glfw and re-enable after it because it's globally set in the root CMakeLists.txt
   set(CMAKE_OBJC_FLAGS "")
-  add_subdirectory(glfw)
-  set_target_properties(glfw PROPERTIES UNITY_BUILD OFF)
-  set_target_properties(glfw PROPERTIES XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC NO)
-  set(CMAKE_OBJC_FLAGS -fobjc-arc)
 
-  add_subdirectory(imgui)
+  if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
+    add_subdirectory(glfw)
+    set_target_properties(glfw PROPERTIES UNITY_BUILD OFF)
+    set_target_properties(glfw PROPERTIES XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC NO)
+    set(CMAKE_OBJC_FLAGS -fobjc-arc)
+  
+    add_subdirectory(imgui)
+  endif()
 endif()
diff --git a/libs/platform/platform_tests/CMakeLists.txt b/libs/platform/platform_tests/CMakeLists.txt
index 2cea6978df79..454c2e55ba3d 100644
--- a/libs/platform/platform_tests/CMakeLists.txt
+++ b/libs/platform/platform_tests/CMakeLists.txt
@@ -7,7 +7,6 @@ set(SRC
   downloader_tests/downloader_test.cpp
   downloader_utils_tests.cpp
   get_text_by_id_tests.cpp
-  glaze_test.cpp
   jansson_test.cpp
   language_test.cpp
   local_country_file_tests.cpp
@@ -25,5 +24,4 @@ target_link_libraries(${PROJECT_NAME}
   platform_tests_support
   platform
   cppjansson
-  glaze::glaze
 )

