diff --git a/storage/maria/unittest/CMakeLists.txt b/storage/maria/unittest/CMakeLists.txt
index a2da150..fd04ef4 100644
--- a/storage/maria/unittest/CMakeLists.txt
+++ b/storage/maria/unittest/CMakeLists.txt
@@ -60,10 +60,10 @@ ADD_EXECUTABLE(ma_test_loghandler_readonly-t
         ma_test_loghandler_multigroup-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c  sequence_storage.c)
 MY_ADD_TEST(ma_test_loghandler_readonly)
 
-SET_TARGET_PROPERTIES(ma_test_loghandler_readonly-t  PROPERTIES COMPILE_FLAGS "-DREADONLY_TEST")
-ADD_EXECUTABLE(ma_test_loghandler_nologs-t
-        ma_test_loghandler_nologs-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c)
-MY_ADD_TEST(ma_test_loghandler_nologs)
+#SET_TARGET_PROPERTIES(ma_test_loghandler_readonly-t  PROPERTIES COMPILE_FLAGS "-DREADONLY_TEST")
+#ADD_EXECUTABLE(ma_test_loghandler_nologs-t
+#        ma_test_loghandler_nologs-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c)
+#MY_ADD_TEST(ma_test_loghandler_nologs)
 
 SET(ma_pagecache_single_src     ma_pagecache_single.c test_file.c test_file.h)
 SET(ma_pagecache_consist_src ma_pagecache_consist.c test_file.c test_file.h)
--- a/unittest/mysys/CMakeLists.txt	2020-11-10 14:15:06.000000000 +0100
+++ b/unittest/mysys/CMakeLists.txt	2020-11-24 21:08:46.721183004 +0100
@@ -14,8 +14,8 @@
 # Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1335 USA
 
 MY_ADD_TESTS(bitmap base64 my_atomic my_rdtsc lf my_malloc my_getopt dynstring
-             byte_order my_tzinfo
-             queues stacktrace crc32 LINK_LIBRARIES mysys)
+             byte_order
+             queues crc32 LINK_LIBRARIES mysys)
 MY_ADD_TESTS(my_vsnprintf LINK_LIBRARIES strings mysys)
 MY_ADD_TESTS(aes LINK_LIBRARIES  mysys mysys_ssl)
 ADD_DEFINITIONS(${SSL_DEFINES})
