diff --git a/storage/maria/unittest/CMakeLists.txt b/storage/maria/unittest/CMakeLists.txt
index a2da150..fd04ef4 100644
--- a/storage/maria/unittest/CMakeLists.txt
+++ b/storage/maria/unittest/CMakeLists.txt
@@ -60,10 +60,10 @@ ADD_EXECUTABLE(ma_test_loghandler_readonly-t
         ma_test_loghandler_multigroup-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c  sequence_storage.c)
 MY_ADD_TEST(ma_test_loghandler_readonly)
 
-SET_TARGET_PROPERTIES(ma_test_loghandler_readonly-t  PROPERTIES COMPILE_FLAGS "-DREADONLY_TEST")
-ADD_EXECUTABLE(ma_test_loghandler_nologs-t
-        ma_test_loghandler_nologs-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c)
-MY_ADD_TEST(ma_test_loghandler_nologs)
+#SET_TARGET_PROPERTIES(ma_test_loghandler_readonly-t  PROPERTIES COMPILE_FLAGS "-DREADONLY_TEST")
+#ADD_EXECUTABLE(ma_test_loghandler_nologs-t
+#        ma_test_loghandler_nologs-t.c ma_maria_log_cleanup.c ma_loghandler_examples.c)
+#MY_ADD_TEST(ma_test_loghandler_nologs)
 
 SET(ma_pagecache_single_src     ma_pagecache_single.c test_file.c test_file.h)
 SET(ma_pagecache_consist_src ma_pagecache_consist.c test_file.c test_file.h)
