--- a/scripts/CMakeLists.txt
+++ b/scripts/CMakeLists.txt
@@ -168,23 +168,13 @@
 # i.e. makes access relative the current directory. This matches
 # the documentation, so better not change this.
 
-IF(INSTALL_LAYOUT MATCHES "STANDALONE")
-  SET(prefix ".")
-  SET(bindir ${prefix}/${INSTALL_BINDIR})
-  SET(sbindir ${prefix}/${INSTALL_SBINDIR})
-  SET(scriptdir ${prefix}/${INSTALL_BINDIR})
-  SET(libexecdir ${prefix}/${INSTALL_SBINDIR})
-  SET(pkgdatadir ${prefix}/${INSTALL_MYSQLSHAREDIR})
-  SET(localstatedir ${prefix}/data)
-ELSE()
-  SET(prefix "${CMAKE_INSTALL_PREFIX}")
-  SET(bindir ${INSTALL_BINDIRABS})
-  SET(sbindir ${INSTALL_SBINDIRABS})
-  SET(scriptdir ${INSTALL_BINDIRABS})
-  SET(libexecdir ${INSTALL_SBINDIRABS})
-  SET(pkgdatadir ${INSTALL_MYSQLSHAREDIRABS})
-  SET(localstatedir ${MYSQL_DATADIR})
-ENDIF()
+SET(prefix "${CMAKE_INSTALL_PREFIX}")
+SET(bindir ${prefix}/${INSTALL_BINDIR})
+SET(sbindir ${prefix}/${INSTALL_SBINDIR})
+SET(scriptdir ${prefix}/${INSTALL_BINDIR})
+SET(libexecdir ${prefix}/${INSTALL_SBINDIR})
+SET(pkgdatadir ${prefix}/${INSTALL_MYSQLSHAREDIR})
+SET(localstatedir ${MYSQL_DATADIR})
 
 SET(resolveip_locations "$basedir/${INSTALL_BINDIR} $basedir/bin")
 SET(mysqld_locations "$basedir/${INSTALL_SBINDIR} $basedir/libexec $basedir/sbin $basedir/bin")
