diff -Nru mysql.old/cmake/ssl.cmake mysql/cmake/ssl.cmake
--- mysql.old/cmake/ssl.cmake	2018-10-31 22:54:02.000000000 +0100
+++ mysql/cmake/ssl.cmake	2019-01-10 13:13:34.884030256 +0100
@@ -182,7 +182,6 @@
                         HAVE_SHA512_DIGEST_LENGTH)
     SET(CMAKE_REQUIRED_INCLUDES)
     IF(OPENSSL_INCLUDE_DIR AND OPENSSL_LIBRARIES AND
-       OPENSSL_MAJOR_VERSION STRLESS "101" AND
        CRYPTO_LIBRARY AND HAVE_SHA512_DIGEST_LENGTH)
 
       SET(SSL_SOURCES "")
