diff --git a/include/wsrep/provider_options.hpp b/include/wsrep/provider_options.hpp
index 022e159..7009a20 100644
--- a/wsrep-lib/include/wsrep/provider_options.hpp
+++ b/wsrep-lib/include/wsrep/provider_options.hpp
@@ -31,6 +31,7 @@
 #include <map>
 #include <memory>
 #include <string>
+#include <cstdint>
 
 namespace wsrep
 {
diff --git a/storage/rocksdb/rocksdb/db/blob/blob_file_meta.h b/storage/rocksdb/rocksdb/db/blob/blob_file_meta.h
index d7c8a124..d688825c 100644
--- a/storage/rocksdb/rocksdb/db/blob/blob_file_meta.h
+++ b/storage/rocksdb/rocksdb/db/blob/blob_file_meta.h
@@ -10,6 +10,7 @@
 #include <memory>
 #include <string>
 #include <unordered_set>
+#include <cstdint>
 
 #include "rocksdb/rocksdb_namespace.h"
 
diff --git a/storage/rocksdb/rocksdb/include/rocksdb/trace_record.h b/storage/rocksdb/rocksdb/include/rocksdb/trace_record.h
index 3d0c0acb..1b4a2841 100644
--- a/storage/rocksdb/rocksdb/include/rocksdb/trace_record.h
+++ b/storage/rocksdb/rocksdb/include/rocksdb/trace_record.h
@@ -8,6 +8,7 @@
 #include <memory>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 #include "rocksdb/rocksdb_namespace.h"
 #include "rocksdb/slice.h"
diff --git a/storage/rocksdb/rocksdb/include/rocksdb/utilities/transaction_db_mutex.h b/storage/rocksdb/rocksdb/include/rocksdb/utilities/transaction_db_mutex.h
index 13d899c3..5665aba0 100644
--- a/storage/rocksdb/rocksdb/include/rocksdb/utilities/transaction_db_mutex.h
+++ b/storage/rocksdb/rocksdb/include/rocksdb/utilities/transaction_db_mutex.h
@@ -7,6 +7,7 @@
 #ifndef ROCKSDB_LITE
 
 #include <memory>
+#include <cstdint>
 
 #include "rocksdb/status.h"
 
