This would introduce a circular dependency.

--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -44,7 +44,6 @@ buildreq="
         help2man   1.29        https://www.gnu.org/s/help2man
         make       3.81        https://www.gnu.org/s/make
         makeinfo   4.8         https://www.gnu.org/s/texinfo
-        xz         4.999.8beta https://tukaani.org/xz
 "
 
 # Instructions on how to install packages in $buildreq.
