From bf3cb42c6db1f4479467d66a111b08a98afef31b Mon Sep 17 00:00:00 2001
From: Lars Wendler <polynomial-c@gentoo.org>
Date: Tue, 1 Mar 2022 16:08:06 +0100
Subject: [PATCH 4/5] Add have_x11=no if --without-x was given to make later
 checks happy

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index 4420c6d3..8816a950 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1129,6 +1129,7 @@ else if test "x$with_x" != xno; then
              have_x11=no
             ])
 else
+    have_x11=no
     AS_IF([test "x$enable_x11_autolaunch" = "xyes"], [
       AC_MSG_ERROR([--enable-x11-autolaunch and --without-x are not compatible])
     ])
-- 
2.36.0

