NoMethodError: undefined method `escape' for URI:Module

--- a/test/uri/escape_test.rb
+++ b/test/uri/escape_test.rb
@@ -2,13 +2,6 @@
 require 'uri'
 
 class UriEscapeTest < Minitest::Test
-  def test_uri_stdlib_compatibility
-    (0..127).each do |i|
-      c = i.chr
-      assert_equal URI.escape(c), EscapeUtils.escape_uri(c)
-    end
-  end
-
   def test_uri_containing_tags
     assert_equal "fo%3Co%3Ebar", EscapeUtils.escape_uri("fo<o>bar")
   end
