--- a/test_all.py
+++ b/test_all.py
@@ -10,6 +10,7 @@
 import platform
 import errno
 
+unionfs_srcdir = os.path.abspath('src')
 
 def call(cmd):
 	return subprocess.check_output(cmd, shell=True, stderr=subprocess.STDOUT)
@@ -37,8 +38,9 @@
 
 class Common:
 	def setUp(self):
-		self.unionfs_path = os.path.abspath('src/unionfs')
-		self.unionfsctl_path = os.path.abspath('src/unionfsctl')
+		global unionfs_srcdir
+		self.unionfs_path = os.path.join(unionfs_srcdir, 'unionfs')
+		self.unionfsctl_path = os.path.join(unionfs_srcdir, 'unionfsctl')
 
 		self.tmpdir = tempfile.mkdtemp()
 		self.original_cwd = os.getcwd()
--- a/Makefile
+++ b/Makefile
@@ -12,7 +12,7 @@
 clean: clean_coverage
 	$(MAKE) -C src/ clean
 
-test: clean build
+test: build
 	python3 -m pytest
 
 test_coverage: clean build_coverage coverage
