From fb115fc533b8e1072d1b9c5b06e5f926ec99dac1 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Wed, 21 Feb 2018 15:54:02 +0100
Subject: [PATCH] Keep screen on by default on emulator

---
 modules/display.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/display.h b/modules/display.h
index 8bd4513..43a0ae5 100644
--- a/modules/display.h
+++ b/modules/display.h
@@ -434,7 +434,7 @@ typedef enum {
  * logic without modifying the timeout values.
  */
 # define MCE_SETTING_BLANKING_INHIBIT_MODE               MCE_SETTING_DISPLAY_PATH "/inhibit_blank_mode"
-# define MCE_DEFAULT_BLANKING_INHIBIT_MODE               0 // = INHIBIT_OFF
+# define MCE_DEFAULT_BLANKING_INHIBIT_MODE               1 // = INHIBIT_ON
 
 /** Kbd slide inhibit type */
 typedef enum {
