commit bd7cddda33a911b0450fa79f5ab9a380777318b4
Author: tcely <tcely@users.noreply.github.com>
Date:   Wed Mar 6 01:14:10 2019 -0500

    do not include from contrib/lmdb directly

diff --git a/src/knot/journal/knot_lmdb.h b/src/knot/journal/knot_lmdb.h
index 35a8884..b1d09cb 100644
--- a/src/knot/journal/knot_lmdb.h
+++ b/src/knot/journal/knot_lmdb.h
@@ -16,7 +16,7 @@
 
 #pragma once
 
-#include "contrib/lmdb/lmdb.h"
+#include <lmdb.h>
 
 #include <stdbool.h>
 #include <stdlib.h>
