--- snownews-1.5.12/configure
+++ snownews-1.5.12/configure.patched
@@ -13,7 +13,7 @@

 my $prefix = "/usr/local";
 my $cflags = "-Wall -Wno-format-y2k -O2 -DLOCALEPATH=\"\\\"\$(LOCALEPATH)\\\"\" -DOS=\\\"$os\\\" $xmlcflags \$(EXTRA_CFLAGS) ";
-my $ldflags = "-lncurses -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";
+my $ldflags = "-lintl -lncursesw -lcrypto $xmlldflags \$(EXTRA_LDFLAGS) ";

 my $use_nls = 1;
 parse_cmdl_line();
