diff -upr zmap-2.1.1.orig/src/CMakeLists.txt zmap-2.1.1/src/CMakeLists.txt
--- zmap-2.1.1.orig/src/CMakeLists.txt	2015-10-10 06:59:56.963317056 +0200
+++ zmap-2.1.1/src/CMakeLists.txt	2015-10-10 07:01:05.206393054 +0200
@@ -122,7 +122,7 @@ add_custom_command(OUTPUT lexer.c
 )
 
 add_custom_command(OUTPUT parser.c
-    COMMAND byacc -d -o "${CMAKE_CURRENT_BINARY_DIR}/parser.c" "${CMAKE_CURRENT_SOURCE_DIR}/parser.y"
+    COMMAND yacc -d -o "${CMAKE_CURRENT_BINARY_DIR}/parser.c" "${CMAKE_CURRENT_SOURCE_DIR}/parser.y"
     DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/parser.y"
 )
 
