diff --git a/src/anyrtpproxy/Makefile b/src/anyrtpproxy/Makefile
index ef15ff3..190cb9d 100644
--- a/src/anyrtpproxy/Makefile
+++ b/src/anyrtpproxy/Makefile
@@ -47,7 +47,7 @@ CFLAGS = -g -Wall
 CXX = g++
 CXXFLAGS = -g -Wall
 LD = g++
-LDFLAGS = -g -Wall -O2 -lboost_thread -lboost_serialization -lboost_system
+LDFLAGS = -g -Wall -O2 -lboost_thread-mt -lboost_serialization -lboost_system
 
 OBJS = anyrtpproxy.o \
        ../signalController.o \
diff --git a/src/configure b/src/configure
index f070150..eb4e131 100755
--- a/src/configure
+++ b/src/configure
@@ -215,7 +215,7 @@ case $TARGET in
     ln -sf posix/posixDaemon.h daemonService.h
     ln -sf posix/posixDaemon.cpp daemonService.cpp
     echo "loading Linux specific TUN Device"
-    LDFLAGS=$LDFLAGS' -lboost_thread -lboost_serialization -lboost_system -lboost_date_time -lpthread'
+    LDFLAGS=$LDFLAGS' -lboost_thread-mt -lboost_serialization -lboost_system -lboost_date_time -lpthread'
     LOG_TARGETS='-DLOG_SYSLOG -DLOG_FILE -DLOG_STDOUT'
   ;;
   OpenBSD|FreeBSD|NetBSD|GNU/kFreeBSD)
