diff --git a/src/configure b/src/configure
index 65a0c74..357628e 100755
--- a/src/configure
+++ b/src/configure
@@ -217,8 +217,8 @@ if [ -n "$ERRORS" ] && [ $EBUILD_COMPAT -ne 1 ]; then
 fi
 
 if [ $USE_CLANG -eq 0 ]; then
-  CXXFLAGS='-g -Wall -Werror -Wno-error=unused-variable -O2'
-  LDFLAGS='-g -Wall -Werror -O2'
+  CXXFLAGS='-g -Wall -Wno-error=unused-variable -O2 -DBOOST_SYSTEM_ENABLE_DEPRECATED'
+  LDFLAGS='-g -Wall -O2'
   COMPILER='g++'
 else
   CXXFLAGS='-g -O2'
