diff --git a/websocket/tests/echo-server.py b/websocket/tests/echo-server.py
deleted file mode 100644
index 8736def..0000000
--- a/websocket/tests/echo-server.py
+++ /dev/null
@@ -1,18 +0,0 @@
-#!/usr/bin/env python
-
-# From https://github.com/aaugustin/websockets/blob/main/example/echo.py
-
-import asyncio
-import websockets
-
-
-async def echo(websocket, path):
-    async for message in websocket:
-        await websocket.send(message)
-
-
-async def main():
-    async with websockets.serve(echo, "localhost", 8765):
-        await asyncio.Future()  # run forever
-
-asyncio.run(main())
