Index: src/via_vt162x.c
===================================================================
--- src/via_vt162x.c	(revision 787)
+++ src/via_vt162x.c	(revision 786)
@@ -684,30 +650,11 @@ VT1622ModeI2C(ScrnInfoPtr pScrn, DisplayModePtr mo
             xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x2B, Table.RGB[4]);
         if (Table.RGB[5])
             xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x2C, Table.RGB[5]);
-        if (pBIOSInfo->TVEncoder == VIA_VT1625) {
-            if (pBIOSInfo->TVType < TVTYPE_480P) {
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x02, 0x12);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x23, 0x7E);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4A, 0x85);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4B, 0x0A);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4E, 0x00);
-            } else {
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x02, 0x12);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4A, 0x85);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4B, 0x0A);
-            }
-        }
     } else if (pBIOSInfo->TVOutput == TVOUTPUT_YCBCR) {
         xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x02, 0x03);
         xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x65, Table.YCbCr[0]);
         xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x66, Table.YCbCr[1]);
         xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x67, Table.YCbCr[2]);
-        if (pBIOSInfo->TVEncoder == VIA_VT1625) {
-            if (pBIOSInfo->TVType < TVTYPE_480P) {
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x23, 0x7E);
-                xf86I2CWriteByte(pBIOSInfo->TVI2CDev, 0x4E, 0x00);
-            }
-        }
     }
 
     /* Configure flicker filter. */
