diff -rupN a/setup.py b/setup.py
--- a/setup.py	2013-03-23 12:42:15.000000000 +0000
+++ b/setup.py	2026-04-01 00:10:10.030000000 +0000
@@ -8,7 +8,9 @@ pkgname = pkg.__name__
 setuptools.setup(
     name=pkgname,
     version=pkg.__version__,
-    packages=setuptools.find_packages(),
+    packages=[
+        'sphinxcontrib'
+        ],
     install_requires=[
         'sphinx'
         ],
@@ -17,7 +19,6 @@ setuptools.setup(
     url='https://github.com/shomah4a/sphinx-gist-embed',
     description='''embedding gist to sphinx''',
     long_description=pkg.__doc__,
-    namespace_packages=['sphinxcontrib'],
     classifiers='''
 Programming Language :: Python
 Development Status :: 4 - Beta
diff -rupN a/sphinxcontrib/__init__.py b/sphinxcontrib/__init__.py
--- a/sphinxcontrib/__init__.py	2013-03-23 12:15:05.000000000 +0000
+++ b/sphinxcontrib/__init__.py	2026-04-01 00:07:24.570000000 +0000
@@ -1 +0,0 @@
-__import__('pkg_resources').declare_namespace(__name__)
