diff --git a/cmd/yaegi/yaegi_test.go b/cmd/yaegi/yaegi_test.go
index bd51b60..37afa11 100644
--- a/cmd/yaegi/yaegi_test.go
+++ b/cmd/yaegi/yaegi_test.go
@@ -6,7 +6,6 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
-	"runtime"
 	"strconv"
 	"strings"
 	"testing"
@@ -51,9 +50,6 @@ func TestYaegiCmdCancel(t *testing.T) {
 	yaegi := filepath.Join(tmp, "yaegi")
 
 	args := []string{"build"}
-	if raceDetectorSupported(runtime.GOOS, runtime.GOARCH) {
-		args = append(args, "-race")
-	}
 	args = append(args, "-o", yaegi, ".")
 
 	build := exec.Command("go", args...)
@@ -123,16 +119,3 @@ func TestYaegiCmdCancel(t *testing.T) {
 		}
 	}
 }
-
-func raceDetectorSupported(goos, goarch string) bool {
-	switch goos {
-	case "linux":
-		return goarch == "amd64" || goarch == "ppc64le" || goarch == "arm64"
-	case "darwin":
-		return goarch == "amd64" || goarch == "arm64"
-	case "freebsd", "netbsd", "openbsd", "windows":
-		return goarch == "amd64"
-	default:
-		return false
-	}
-}
