diff --git a/src/gsad.c b/src/gsad.c
index 7536183..4722a8d 100644
--- a/src/gsad.c
+++ b/src/gsad.c
@@ -5296,7 +5296,7 @@ main (int argc, char **argv)
       if (termination_signal)
         {
           g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Received %s signal.\n",
-                 sys_siglist[termination_signal]);
+                 strsignal(termination_signal));
           gsad_cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           signal (termination_signal, SIG_DFL);
