--- a/greylist.conf
+++ b/greylist.conf
@@ -5,9 +5,9 @@
 # $Id: greylist.conf,v 1.49 2013/01/27 02:10:19 manu Exp $
 #
 
-pidfile "/var/run/milter-greylist.pid"
-socket "/var/milter-greylist/milter-greylist.sock"
-dumpfile "/var/milter-greylist/greylist.db" 600
+pidfile "/var/run/milter-greylist/milter-greylist.pid"
+socket "/var/run/milter-greylist/milter-greylist.sock" 660
+dumpfile "/var/lib/milter-greylist/greylist.db" 600
 dumpfreq 1
 user "smmsp"
 
