diff --git a/src/connection-editor/nm-connection-editor.c b/src/connection-editor/nm-connection-editor.c
index ca74d12..be211f2 100644
--- a/src/connection-editor/nm-connection-editor.c
+++ b/src/connection-editor/nm-connection-editor.c
@@ -31,7 +31,7 @@
 #include <errno.h>
 #include <gdk/gdkx.h>
 
-#ifdef WITH_SELINUX
+#if WITH_SELINUX
 #include <selinux/selinux.h>
 #endif
 
@@ -183,7 +183,7 @@ update_sensitivity (NMConnectionEditor *editor)
 	}
 }
 
-#ifdef WITH_SELINUX
+#if WITH_SELINUX
 /* This is what the files in ~/.cert would get. */
 static const char certcon[] = "unconfined_u:object_r:home_cert_t:s0";
 
