--- a/vendored-quanta/src/clocks/monotonic/unix.rs
+++ b/vendored-quanta/src/clocks/monotonic/unix.rs
@@ -6,10 +6,7 @@ pub struct Monotonic {
 impl Monotonic {
     #[allow(clippy::cast_sign_loss)]
     pub fn now(self) -> u64 {
-        let mut ts = libc::timespec {
-            tv_sec: 0,
-            tv_nsec: 0,
-        };
+        let mut ts = libc::timespec::default();
         unsafe {
             libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
         }
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -7,3 +7,6 @@ version = "0.24.2"
 
 [profile.release]
 debug = true
+
+[patch.crates-io]
+quanta = { path = "vendored-quanta" }
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3081,8 +3081,6 @@ dependencies = [
 [[package]]
 name = "quanta"
 version = "0.12.6"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "f3ab5a9d756f0d97bdc89019bd2e4ea098cf9cde50ee7564dde6b81ccc8f06c7"
 dependencies = [
  "crossbeam-utils",
  "libc",
