diff -ru policyd-spf-fs_23.orig/Makefile policyd-spf-fs_23/Makefile
--- policyd-spf-fs_23.orig/Makefile	2016-11-02 14:44:48.660570292 +0200
+++ policyd-spf-fs_23/Makefile	2016-11-02 14:46:25.591847406 +0200
@@ -20,7 +20,7 @@
 all: policyd-spf-fs
 
 policyd-spf-fs: policyd-spf-fs.o Makefile
-	$(CC) $(CFLAGS) $(LIBS) policyd-spf-fs.o -o policyd-spf-fs
+	$(CC) $(CFLAGS) policyd-spf-fs.o $(LIBS) -o policyd-spf-fs
 
 install: policyd-spf-fs
 	strip policyd-spf-fs
Only in policyd-spf-fs_23: policyd-spf-fs
