diff --git a/policyd-spf-fs.c b/policyd-spf-fs.c
index 198ba98..fd03019 100644
--- a/policyd-spf-fs.c
+++ b/policyd-spf-fs.c
@@ -359,7 +359,7 @@ static void pf_response(SPF_client_options_t    *opts, SPF_response_t *spf_respo
                 case SPF_RESULT_PASS:
                         strcpy(result, POSTFIX_DUNNO);
                         printf("action=PREPEND X-%s\n",SPF_response_get_received_spf(spf_response));
-                        snprintf(spf_comment, RESULTSIZE, SPF_response_get_received_spf(spf_response));
+                        snprintf(spf_comment, RESULTSIZE, "%s", SPF_response_get_received_spf(spf_response));
                         break;
                 case SPF_RESULT_FAIL:
                 	strcpy(result, POSTFIX_REJECT);
@@ -386,7 +386,7 @@ static void pf_response(SPF_client_options_t    *opts, SPF_response_t *spf_respo
                 default:
                         strcpy(result, POSTFIX_DUNNO);
                         printf("action=PREPEND X-%s\n",SPF_response_get_received_spf(spf_response));
-                        snprintf(spf_comment, RESULTSIZE, SPF_response_get_received_spf(spf_response));
+                        snprintf(spf_comment, RESULTSIZE, "%s", SPF_response_get_received_spf(spf_response));
                         break;
         }
         
