--- policyd-spf-fs_23/policyd-spf-fs.c	Tue Sep 11 06:57:55 2007
+++ policyd-spf-fs-23.patched/policyd-spf-fs.c	Mon Sep 21 12:09:33 2009
@@ -624,6 +624,7 @@
 		FREE_RESPONSE(spf_response);
 
 		spf_request = SPF_request_new(spf_server);
+		spf_request->use_local_policy = 1;
 
 		if (SPF_request_set_ipv4_str(spf_request, req->ip) && SPF_request_set_ipv6_str(spf_request, req->ip)) {
 			syslog(LOG_WARNING, "Invalid IP address.\n" );
