--- ./configure.orig	2011-03-26 17:09:34.000000000 +0100
+++ ./configure	2011-03-29 14:23:54.647074988 +0200
@@ -36343,13 +36343,13 @@
 # Check for the iptc_init symbol in the library.
 if test "x$with_libiptc" = "xyes" && test "x$with_own_libiptc" = "xno"
 then
-	{ echo "$as_me:$LINENO: checking for iptc_init in -liptc" >&5
-echo $ECHO_N "checking for iptc_init in -liptc... $ECHO_C" >&6; }
-if test "${ac_cv_lib_iptc_iptc_init+set}" = set; then
+	{ echo "$as_me:$LINENO: checking for iptc_init in -lip4tc" >&5
+echo $ECHO_N "checking for iptc_init in -lip4tc... $ECHO_C" >&6; }
+if test "${ac_cv_lib_ip4tc_iptc_init+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-liptc  $LIBS"
+LIBS="-liptc -lip6tc -lip4tc  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
