time_t is 64bit always, so %lld is appropriate
diff --git a/boreas/alivedetection.c b/boreas/alivedetection.c
index 1067b05..aa1abde 100644
--- a/boreas/alivedetection.c
+++ b/boreas/alivedetection.c
@@ -290,7 +290,7 @@ scan (alive_test_t alive_test)
 
   gettimeofday (&end_time, NULL);
 
-  g_message ("Alive scan %s finished in %ld seconds: %d alive hosts of %d.",
+  g_message ("Alive scan %s finished in %lld seconds: %d alive hosts of %d.",
              scan_id, end_time.tv_sec - start_time.tv_sec,
              g_hash_table_size (scanner.hosts_data->alivehosts),
              number_of_targets);
diff --git a/boreas/cli.c b/boreas/cli.c
index 5a67e44..6b62d5f 100644
--- a/boreas/cli.c
+++ b/boreas/cli.c
@@ -162,7 +162,7 @@ run_cli_scan (scanner_t *scanner, alive_test_t alive_test)
                                            scanner->hosts_data->alivehosts);
   gettimeofday (&end_time, NULL);
   if (scanner->print_results == 1)
-    printf ("Alive scan finished in %ld seconds: %d alive hosts of %d.\n",
+    printf ("Alive scan finished in %lld seconds: %d alive hosts of %d.\n",
             end_time.tv_sec - start_time.tv_sec,
             number_of_targets - number_of_dead_hosts, number_of_targets);
 
