diff --git a/util/nvticache.c b/util/nvticache.c
index 162e5851..9a2b7287 100644
--- a/util/nvticache.c
+++ b/util/nvticache.c
@@ -189,7 +189,7 @@ nvticache_save ()
     }
   old_version = nvticache_feed_version ();
   feed_version = nvt_feed_version ();
-  if (g_strcmp0 (old_version, feed_version))
+  if (feed_version && g_strcmp0 (old_version, feed_version))
     {
       kb_item_set_str (cache_kb, NVTICACHE_STR, feed_version, 0);
       g_message ("Updated NVT cache from version %s to %s", old_version,
