--- a/src/common/eventfd.cpp
+++ b/src/common/eventfd.cpp
@@ -15,10 +15,10 @@
 
 lttng::eventfd::eventfd(bool use_semaphore_semantics, std::uint64_t initial_value) :
 	file_descriptor([use_semaphore_semantics, initial_value]() {
-		int flags = ::EFD_CLOEXEC;
+		int flags = EFD_CLOEXEC;
 
 		if (use_semaphore_semantics) {
-			flags |= ::EFD_SEMAPHORE;
+			flags |= EFD_SEMAPHORE;
 		}
 
 		const auto raw_fd = ::eventfd(initial_value, flags);
