--- a/deps/sqlite3.gyp
+++ b/deps/sqlite3.gyp
@@ -13,23 +13,18 @@
       'type': 'none',
       'hard_dependency': 1,
       'actions': [{
-        'action_name': 'extract_sqlite3',
-        'inputs': ['sqlcipher.tar.gz'],
+        'action_name': 'copy_builtin_sqlite3',
+        'inputs': [
+          'sqlite3/sqlite3.c',
+          'sqlite3/sqlite3.h',
+          'sqlite3/sqlite3ext.h',
+        ],
         'outputs': [
           '<(SHARED_INTERMEDIATE_DIR)/sqlite3/sqlite3.c',
           '<(SHARED_INTERMEDIATE_DIR)/sqlite3/sqlite3.h',
           '<(SHARED_INTERMEDIATE_DIR)/sqlite3/sqlite3ext.h',
         ],
-        'conditions': [
-          ['OS == "win"', {
-            'outputs': [
-              '<(SHARED_INTERMEDIATE_DIR)/sqlite3/>(openssl_root)/libssl.lib',
-              '<(SHARED_INTERMEDIATE_DIR)/sqlite3/>(openssl_root)/libcrypto.lib',
-              '<(SHARED_INTERMEDIATE_DIR)/sqlite3/>(openssl_root)/ossl_static.pdb',
-            ],
-          }],
-        ],
-        'action': ['node', 'extract.js', '<(SHARED_INTERMEDIATE_DIR)/sqlite3'],
+        'action': ['cp', './sqlite3/sqlite3.c', './sqlite3/sqlite3.h', './sqlite3/sqlite3ext.h', '<(SHARED_INTERMEDIATE_DIR)/sqlite3'],
       }],
     },
     {
@@ -98,8 +93,7 @@
         { # Linux
           'link_settings': {
             'libraries': [
-              # This statically links libcrypto, whereas -lcrypto would dynamically link it
-              '<(SHARED_INTERMEDIATE_DIR)/sqlite3/OpenSSL-linux-<(target_arch)/libcrypto.a'
+              '-lcrypto'
             ]
           }
         }],
