--- ./node/build_node_bridge.py.orig
+++ ./node/build_node_bridge.py
@@ -67,7 +67,7 @@
     if 'npm_config_libsignal_debug_level_logs' not in os.environ:
         features.append('log/release_max_level_info')
 
-    cmdline = ['cargo', 'build', '--target', cargo_target, '-p', 'libsignal-node', '--features', ','.join(features)]
+    cmdline = ['cargo', 'auditable', 'build', '--target', cargo_target, '-p', 'libsignal-node', '--features', ','.join(features)]
     if configuration_name == 'Release':
         cmdline.append('--release')
     print("Running '%s'" % (' '.join(cmdline)))
