--- ./src/rust/build.rs.orig
+++ ./src/rust/build.rs
@@ -41,15 +41,15 @@
     if cfg!(feature = "native") {
         if let Ok(out_dir) = out_dir {
             println!(
-                "cargo:rustc-link-search=native={}/{}/obj/",
+                "cargo:rustc-link-search=native={}/{}/",
                 out_dir, build_type,
             );
-            println!("cargo:rerun-if-changed={}/{}/obj/", out_dir, build_type,);
+            println!("cargo:rerun-if-changed={}/{}/", out_dir, build_type,);
         } else {
             println!("cargo:warning=No WebRTC output directory (OUTPUT_DIR) defined!");
         }
 
-        println!("cargo:rustc-link-lib=webrtc");
+        println!("cargo:rustc-link-lib=dylib=signaldeswebrtc");
 
         if cfg!(target_os = "macos") {
             println!("cargo:rustc-link-lib=dylib=c++");
