--- a/ts/components/DialogExpiredBuild.tsx
+++ b/ts/components/DialogExpiredBuild.tsx
@@ -29,9 +29,9 @@
       containerWidthBreakpoint={containerWidthBreakpoint}
       type="error"
       onClick={() => {
-        openLinkInWebBrowser('https://signal.org/download/');
+        openLinkInWebBrowser('https://pkgs.alpinelinux.org/packages?name=signal-desktop');
       }}
-      clickLabel={i18n('upgrade')}
+      clickLabel={<code>apk upgrade signal-desktop</code>}
       hasAction
     >
       {i18n('expiredWarning')}{' '}
--- a/ts/components/DialogUpdate.tsx
+++ b/ts/components/DialogUpdate.tsx
@@ -27,7 +27,7 @@
   currentVersion: string;
 };
 
-const PRODUCTION_DOWNLOAD_URL = 'https://signal.org/download/';
+const PRODUCTION_DOWNLOAD_URL = 'https://pkgs.alpinelinux.org/packages?name=signal-desktop';
 const BETA_DOWNLOAD_URL = 'https://support.signal.org/beta';
 
 export const DialogUpdate = ({
