--- a/ts/components/DialogExpiredBuild.tsx
+++ b/ts/components/DialogExpiredBuild.tsx
@@ -29,9 +29,9 @@
       containerWidthBreakpoint={containerWidthBreakpoint}
       type="error"
       onClick={() => {
-        openLinkInWebBrowser('https://signal.org/download/');
+        openLinkInWebBrowser('https://pkgs.alpinelinux.org/packages?name=signal-desktop');
       }}
-      clickLabel={i18n('upgrade')}
+      clickLabel={<code>apk upgrade signal-desktop</code>}
       hasAction
     >
       {i18n('expiredWarning')}{' '}
--- ./ts/types/support.ts.orig
+++ ./ts/types/support.ts
@@ -1,7 +1,7 @@
 // Copyright 2023 Signal Messenger, LLC
 // SPDX-License-Identifier: AGPL-3.0-only
 
-export const PRODUCTION_DOWNLOAD_URL = 'https://signal.org/download/';
+export const PRODUCTION_DOWNLOAD_URL = 'https://pkgs.alpinelinux.org/packages?name=signal-desktop';
 export const BETA_DOWNLOAD_URL = 'https://support.signal.org/beta';
 export const UNSUPPORTED_OS_URL =
   'https://support.signal.org/hc/articles/5109141421850';
