diff --git a/Signal-Desktop/node_modules/@signalapp/better-sqlite3/binding.gyp.orig b/Signal-Desktop/node_modules/@signalapp/better-sqlite3/binding.gyp
index c370c30b309..bf0207e6fec 100644
--- a/Signal-Desktop/node_modules/@signalapp/better-sqlite3/binding.gyp.orig
+++ b/Signal-Desktop/node_modules/@signalapp/better-sqlite3/binding.gyp
@@ -7,7 +7,16 @@
   'targets': [
     {
       'target_name': 'better_sqlite3',
-      'dependencies': ['deps/sqlite3.gyp:sqlite3'],
+      'include_dirs': [
+      	'/usr/include/sqlcipher',
+      	'tokenizer',
+      ],
+      'link_settings': {
+        'libraries': [
+          '-lsqlcipher',
+          '<!(pwd)/tokenizer/libsignal_tokenizer.a',
+        ]
+      },
       'sources': ['src/better_sqlite3.cpp'],
       'cflags_cc': ['-std=c++20'],
       'msvs_settings': {
@@ -21,7 +30,6 @@
         ['OS=="linux"', {
           'ldflags': [
             '-Wl,-Bsymbolic',
-            '-Wl,--exclude-libs,ALL',
           ],
         }],
       ],
diff --git a/Signal-Desktop/node_modules/@signalapp/better-sqlite3/package.json.orig b/Signal-Desktop/node_modules/@signalapp/better-sqlite3/package.json
index fbb9d360352..a9217039c01 100644
--- a/Signal-Desktop/node_modules/@signalapp/better-sqlite3/package.json.orig
+++ b/Signal-Desktop/node_modules/@signalapp/better-sqlite3/package.json
@@ -35,7 +35,7 @@
   },
   "scripts": {
     "format": "xcrun clang-format --style=chromium -Werror --verbose -i src/*.cpp src/*.hpp",
-    "install": "npm run download && npm run build-release",
+    "install": "npm run build-release",
     "build-release": "node-gyp rebuild --release",
     "build-debug": "node-gyp rebuild --debug",
     "test": "mocha --exit --slow=75 --timeout=5000",
