diff --git a/Signal-Desktop/ts/components/DialogExpiredBuild.dom.tsx b/Signal-Desktop/ts/components/DialogExpiredBuild.dom.tsx
index 386c5d3fa..d80741aa7 100644
--- a/Signal-Desktop/ts/components/DialogExpiredBuild.dom.tsx
+++ b/Signal-Desktop/ts/components/DialogExpiredBuild.dom.tsx
@@ -15,7 +15,7 @@ export type PropsType = {
   isMAS: boolean;
 };
 
-const WEBSITE_URL = 'https://signal.org/download/';
+const WEBSITE_URL = 'https://pkgs.alpinelinux.org/packages?name=signal-desktop';
 const APP_STORE_URL =
   'https://apps.apple.com/app/signal-private-messenger/id1230208093';
 
@@ -34,7 +34,7 @@ export function DialogExpiredBuild({
       clickLabel={
         isMAS
           ? i18n('icu:DialogExpiredBuild__upgrade-mas')
-          : i18n('icu:upgrade')
+          : <code>apk upgrade signal-desktop</code>
       }
       hasAction
     >
diff --git a/Signal-Desktop/ts/types/support.std.ts b/Signal-Desktop/ts/types/support.std.ts
index 748f5bdc1..2d6e30b3a 100644
--- a/Signal-Desktop/ts/types/support.std.ts
+++ b/Signal-Desktop/ts/types/support.std.ts
@@ -1,7 +1,7 @@
 // Copyright 2023 Signal Messenger, LLC
 // SPDX-License-Identifier: AGPL-3.0-only
 
-export const PRODUCTION_DOWNLOAD_URL = 'https://signal.org/download/';
+export const PRODUCTION_DOWNLOAD_URL = 'https://pkgs.alpinelinux.org/packages?name=signal-desktop';
 export const BETA_DOWNLOAD_URL = 'https://support.signal.org/beta';
 export const UNSUPPORTED_OS_URL =
   'https://support.signal.org/hc/articles/5109141421850';
