--- a/webrtc/build/config/compiler/BUILD.gn.orig
+++ b/webrtc/build/config/compiler/BUILD.gn
@@ -637,8 +637,8 @@
     # other problems.
     if (default_toolchain != "//build/toolchain/cros:target") {
       cflags += [
-        "-mllvm",
-        "-split-threshold-for-reg-with-hint=0",
+        #"-mllvm",
+        #"-split-threshold-for-reg-with-hint=0",
       ]
       if (use_thin_lto && is_a_target_toolchain) {
         if (is_win) {
@@ -835,7 +835,7 @@
         import_instr_limit = 5
       }
 
-      ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
+      #ldflags += [ "-Wl,-mllvm,-import-instr-limit=$import_instr_limit" ]
 
       if (is_apple) {
         ldflags += [ "-Wcrl,object_path_lto" ]
