--- ./BUILD.gn.orig
+++ ./BUILD.gn
@@ -38,7 +38,7 @@
   # 'ninja default' and then 'ninja all', the second build should do no work.
   group("default") {
     testonly = true
-    deps = [ ":webrtc" ]
+    deps = [ ":signaldeswebrtc" ]
     if (rtc_build_examples) {
       deps += [ "examples" ]
     }
@@ -464,7 +464,7 @@
 
 if (!build_with_chromium) {
   # Target to build all the WebRTC production code.
-  rtc_static_library("webrtc") {
+  rtc_shared_library("signaldeswebrtc") {
     # Only the root target and the test should depend on this.
     visibility = [
       "//:default",
@@ -472,7 +472,6 @@
     ]
 
     sources = []
-    complete_static_lib = true
     suppressed_configs += [ "//build/config/compiler:thin_archive" ]
     defines = []
 
diff --git a/third_party/googletest/BUILD.gn.orig b/third_party/googletest/BUILD.gn
index 14089f0..b7dc621 100644
--- a/third_party/googletest/BUILD.gn.orig
+++ b/third_party/googletest/BUILD.gn
@@ -48,7 +48,6 @@ config("gtest_config") {
 
   configs = [
     "//third_party/abseil-cpp:absl_include_config",
-    "//third_party/re2:re2_config",
   ]
 }
 
diff --git a/third_party/googletest/BUILD.gn.orig b/third_party/googletest/BUILD.gn
index b7dc621..367f929 100644
--- a/third_party/googletest/BUILD.gn.orig
+++ b/third_party/googletest/BUILD.gn
@@ -133,7 +133,6 @@ source_set("gtest") {
 
   # googletest only needs `absl`, but this makes gn check happier.
   deps = [ "//third_party/abseil-cpp:absl_full" ]
-  public_deps = [ "//third_party/re2" ]
   if (is_nacl || !build_with_chromium) {
     defines += [ "GTEST_DISABLE_PRINT_STACK_TRACE" ]
     sources -= [
diff --git a/third_party/fuzztest/BUILD.gn.orig b/third_party/fuzztest/BUILD.gn
index 57ee790..ba1d297 100644
--- a/third_party/fuzztest/BUILD.gn.orig
+++ b/third_party/fuzztest/BUILD.gn
@@ -309,7 +309,6 @@ source_set("fuzztest_internal") {
     # For RE2 mutators. It's questionable whether we want to pull this library
     # into every fuzztest target, but this is the approach used in other
     # fuzztest contexts so we'll do the same
-    "//third_party/re2",
 
     # For protobuf mutators
     "$protobuf_target_prefix:protobuf_lite",
