--- ./webrtc/rtc_base/system/file_wrapper.h.orig
+++ ./webrtc/rtc_base/system/file_wrapper.h
@@ -13,6 +13,7 @@
 
 #include <stddef.h>
 #include <stdio.h>
+#include <stdint.h>
 
 #include <string>
 
--- ./webrtc/modules/audio_coding/neteq/reorder_optimizer.cc.orig
+++ ./webrtc/modules/audio_coding/neteq/reorder_optimizer.cc
@@ -11,6 +11,7 @@
 #include "modules/audio_coding/neteq/reorder_optimizer.h"
 
 #include <algorithm>
+#include <cstdint>
 #include <limits>
 #include <vector>

--- ./webrtc/rtc_base/ssl_stream_adapter.h.orig
+++ ./webrtc/rtc_base/ssl_stream_adapter.h
@@ -24,6 +24,7 @@
 #include "rtc_base/ssl_certificate.h"
 #include "rtc_base/ssl_identity.h"
 #include "rtc_base/stream.h"
+#include <optional>
 
 namespace rtc {
 

