diff --git a/p2p/base/port_interface.h b/p2p/base/port_interface.h
index 6ff4c4c04a..0d63ed5029 100644
--- a/webrtc/p2p/base/port_interface.h
+++ b/webrtc/p2p/base/port_interface.h
@@ -52,7 +52,7 @@ class PortInterface {
   virtual ~PortInterface();
 
   virtual IceCandidateType Type() const = 0;
-  virtual const Network* Network() const = 0;
+  virtual const ::webrtc::Network* Network() const = 0;
 
   // Methods to set/get ICE role and tiebreaker values.
   virtual void SetIceRole(IceRole role) = 0;
