--- ./BUILD.gn.orig
+++ ./BUILD.gn
@@ -38,7 +38,7 @@
   # 'ninja default' and then 'ninja all', the second build should do no work.
   group("default") {
     testonly = true
-    deps = [ ":webrtc" ]
+    deps = [ ":signaldeswebrtc" ]
     if (rtc_build_examples) {
       deps += [ "examples" ]
     }
@@ -464,7 +464,7 @@
 
 if (!build_with_chromium) {
   # Target to build all the WebRTC production code.
-  rtc_static_library("webrtc") {
+  rtc_shared_library("signaldeswebrtc") {
     # Only the root target and the test should depend on this.
     visibility = [
       "//:default",
@@ -472,7 +472,6 @@
     ]
 
     sources = []
-    complete_static_lib = true
     suppressed_configs += [ "//build/config/compiler:thin_archive" ]
     defines = []
 
