diff --git a/webrtc/modules/desktop_capture/linux/wayland/egl_dmabuf.h b/webrtc/modules/desktop_capture/linux/wayland/egl_dmabuf.h
index 54068f92ec..9e5caea3ba 100644
--- a/webrtc/modules/desktop_capture/linux/wayland/egl_dmabuf.h
+++ b/webrtc/modules/desktop_capture/linux/wayland/egl_dmabuf.h
@@ -15,6 +15,7 @@
 #include <EGL/eglplatform.h>
 #include <GL/gl.h>
 #include <gbm.h>
+#include <sys/types.h>
 
 #include <cstdint>
 #include <map>
