this breaks the clang include path
--
--- a/build/config/compiler/BUILD.gn
+++ a/build/config/compiler/BUILD.gn
@@ -1310,7 +1310,7 @@
   # paths to the tools it invokes. We don't want this because we don't
   # really need it and it can mess up the goma cache entries.
   if (is_clang && (!is_nacl || is_nacl_saigo)) {
-    cflags += [ "-no-canonical-prefixes" ]
+    #cflags += [ "-no-canonical-prefixes" ]
 
     # Same for links: Let the compiler driver invoke the linker
     # with a relative path and pass relative paths to built-in
@@ -1319,7 +1319,7 @@
     # We don't link on goma, so this change is just for cleaner
     # internal linker invocations, for people who work on the build.
     if (!is_win) {
-      ldflags += [ "-no-canonical-prefixes" ]
+      #ldflags += [ "-no-canonical-prefixes" ]
     }
   }
 }
