diff --git a/webrtc/p2p/base/port_interface.h b/webrtc/p2p/base/port_interface.h
index 2335fc2..1484a14 100644
--- a/webrtc/p2p/base/port_interface.h
+++ b/webrtc/p2p/base/port_interface.h
@@ -55,7 +55,7 @@
   virtual ~PortInterface();
 
   virtual IceCandidateType Type() const = 0;
-  virtual const Network* Network() const = 0;
+  virtual const ::webrtc::Network* Network() const = 0;
 
   // Methods to set/get ICE role and tiebreaker values.
   virtual void SetIceRole(IceRole role) = 0;
diff --git a/webrtc/pc/codec_vendor.h b/webrtc/pc/codec_vendor.h
index aaddc4f..719a082 100644
--- a/webrtc/pc/codec_vendor.h
+++ b/webrtc/pc/codec_vendor.h
@@ -119,7 +119,7 @@
 class CodecLookupHelper {
  public:
   virtual ~CodecLookupHelper() = default;
-  virtual PayloadTypeSuggester* PayloadTypeSuggester() = 0;
+  virtual ::webrtc::PayloadTypeSuggester* PayloadTypeSuggester() = 0;
   // Look up the codec vendor to use, depending on context.
   // This call may get additional arguments in the future, to aid
   // in selection of the correct context.
diff --git a/webrtc/pc/jsep_transport.h b/webrtc/pc/jsep_transport.h
index 017aad7..50dc991 100644
--- a/webrtc/pc/jsep_transport.h
+++ b/webrtc/pc/jsep_transport.h
@@ -115,14 +115,12 @@
   }
 
   RTCError SetLocalJsepTransportDescription(
-      const JsepTransportDescription& jsep_description,
-      SdpType type);
+      const JsepTransportDescription& jsep_description, SdpType type);
 
   // Set the remote TransportDescription to be used by DTLS and ICE channels
   // that are part of this Transport.
   RTCError SetRemoteJsepTransportDescription(
-      const JsepTransportDescription& jsep_description,
-      SdpType type);
+      const JsepTransportDescription& jsep_description, SdpType type);
   RTCError AddRemoteCandidates(const Candidates& candidates);
 
   // Set the "needs-ice-restart" flag as described in JSEP. After the flag is
@@ -204,7 +202,9 @@
     return rtp_dtls_transport_;
   }
 
-  scoped_refptr<SctpTransport> SctpTransport() const { return sctp_transport_; }
+  scoped_refptr<::webrtc::SctpTransport> SctpTransport() const {
+    return sctp_transport_;
+  }
 
   // TODO(bugs.webrtc.org/9719): Delete method, update callers to use
   // SctpTransport() instead.
