--- a/util/imb_api.h
+++ b/util/imb_api.h
@@ -49,8 +49,8 @@
 /* DOS defines wchar_t in stdlib.h */
 #else 
  // defined(LINUX) | defined(SOLARIS)
-#ifndef _WCHAR_T
-#define _WCHAR_T
+#ifndef __DEFINED_wchar_t
+#define __DEFINED_wchar_t
 typedef long    wchar_t;
 #endif
 #endif
--- a/util/mem_if.c
+++ b/util/mem_if.c
@@ -65,6 +65,7 @@
 #include <sys/param.h>
 #include <sys/mman.h>
 #include <sys/ioctl.h>
+#include <linux/param.h>
 #endif
 #if defined(SOLARIS) || defined(BSD)
 #define EXEC_PAGESIZE     4096
--- a/util/ipmimv.c
+++ b/util/ipmimv.c
@@ -62,6 +62,7 @@
 #include <time.h>
 #include <errno.h>
 #include <sys/ioctl.h>
+#include <sys/select.h>
 #if defined(MACOS)
 #include <sys/time.h>
 #else
