--- ipmiutil-3.0.7/lib/Makefile.am.old	2017-09-20 10:08:20.000000000 -0500
+++ ipmiutil-3.0.7/lib/Makefile.am	2018-01-01 18:03:45.080832425 -0600
@@ -33,10 +33,11 @@
 	    else cp -f libipmiapi64.a.redhat  libipmiapi.a; fi \
           else cp -f libipmiapi32.a libipmiapi.a; fi \
 	fi
-	if [ "$(PLUSFLAGS)" = "-DHAVE_LANPLUS" ]; then \
-	  cd lanplus; make clean; make  ; \
-	  pwd; cp -f libipmi_lanplus.a .. ; \
-	fi
+	case "$(PLUSFLAGS)" in \
+	  "-DHAVE_LANPLUS"*) \
+	    cd lanplus; make clean; make  ; \
+	    pwd; cp -f libipmi_lanplus.a .. ;; \
+	esac
 
 clean:
 	rm -f libipmiapi.a lib*_lanplus.a lib*_lanplus.so
@@ -51,10 +52,11 @@
 	cd lanplus; make distclean
 	
 install:
-	if [ "$(PLUSFLAGS)" = "-DHAVE_LANPLUS" ]; then \
-	   $(MKDIR) ${datato} ; \
-	   cd lanplus; make install ; \
-	fi
+	case "$(PLUSFLAGS)" in \
+	  "-DHAVE_LANPLUS"*) \
+	     $(MKDIR) ${datato} ; \
+	     cd lanplus; make install ;; \
+	esac
 
 check:
 
--- ipmiutil-3.0.7/lib/Makefile.in.old	2017-09-20 10:08:28.000000000 -0500
+++ ipmiutil-3.0.7/lib/Makefile.in	2018-01-01 18:02:55.187678005 -0600
@@ -392,10 +392,11 @@
 	    else cp -f libipmiapi64.a.redhat  libipmiapi.a; fi \
           else cp -f libipmiapi32.a libipmiapi.a; fi \
 	fi
-	if [ "$(PLUSFLAGS)" = "-DHAVE_LANPLUS" ]; then \
-	  cd lanplus; make clean; make  ; \
-	  pwd; cp -f libipmi_lanplus.a .. ; \
-	fi
+	case "$(PLUSFLAGS)" in \
+	   "-DHAVE_LANPLUS"*) \
+	    cd lanplus; make clean; make  ; \
+	    pwd; cp -f libipmi_lanplus.a .. ;; \
+	esac
 
 clean:
 	rm -f libipmiapi.a lib*_lanplus.a lib*_lanplus.so
@@ -410,10 +411,11 @@
 	cd lanplus; make distclean
 
 install:
-	if [ "$(PLUSFLAGS)" = "-DHAVE_LANPLUS" ]; then \
-	   $(MKDIR) ${datato} ; \
-	   cd lanplus; make install ; \
-	fi
+	case "$(PLUSFLAGS)" in \
+	    "-DHAVE_LANPLUS"*) \
+	       $(MKDIR) ${datato} ; \
+	       cd lanplus; make install ;; \
+	esac
 
 check:
 
