From c6fe28de4a05e33c317019261711f2babe98b95d Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 3 Jan 2011 22:16:24 +0300
Subject: [PATCH 10/11] Move CMake defines

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 CMakeLists.txt      |    3 +++
 core/CMakeLists.txt |    6 ------
 2 files changed, 3 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 78a6dfb..fdc88f4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -80,6 +80,7 @@ SET(CMAKE_VERBOSE_MAKEFILE ON)
 #   (which means that current debian and gentoo packages don't work)
 IF(SEMS_USE_SPANDSP)
 	FIND_PACKAGE(Spandsp REQUIRED)
+	ADD_DEFINITIONS(-DUSE_SPANDSP -D__STDC_LIMIT_MACROS)
 	MESSAGE(STATUS "Using spandsp: YES")
 ELSE(SEMS_USE_SPANDSP)
 	MESSAGE(STATUS "Using spandsp: NO (default)")
@@ -89,6 +90,7 @@ ENDIF(SEMS_USE_SPANDSP)
 # (see http://www.mega-nerd.com/SRC/)
 IF(SEMS_USE_LIBSAMPLERATE)
 	FIND_PACKAGE(Libsamplerate REQUIRED)
+	ADD_DEFINITIONS(-DUSE_LIBSAMPLERATE)
 	MESSAGE(STATUS "Using libsamplerate: YES")
 ELSE(SEMS_USE_LIBSAMPLERATE)
 	MESSAGE(STATUS "Using libsamplerate: NO (default)")
@@ -128,6 +130,7 @@ ELSE(SEMS_USE_MONITORING)
 ENDIF(SEMS_USE_MONITORING)
 
 IF(SEMS_USE_IPV6)
+	ADD_DEFINITIONS(-DSUPPORT_IPV6)
 	MESSAGE(STATUS "Enable IPv6 support: YES")
 ELSE(SEMS_USE_IPV6)
 	MESSAGE(STATUS "Enable IPv6 support: NO (default)")
diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 7bd1341..5f0fb3f 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -18,19 +18,13 @@ ENDIF(NOT MAX_RTP_SESSIONS)
 ADD_DEFINITIONS(-DMAX_RTP_SESSIONS=${MAX_RTP_SESSIONS})
 
 IF(SPANDSP_FOUND)
-	ADD_DEFINITIONS(-DUSE_SPANDSP -D__STDC_LIMIT_MACROS)
 	TARGET_LINK_LIBRARIES(sems ${CMAKE_DL_LIBS} spandsp)
 ENDIF(SPANDSP_FOUND)
 
 IF(LIBSAMPLERATE_FOUND)
-	ADD_DEFINITIONS(-DUSE_LIBSAMPLERATE)
 	TARGET_LINK_LIBRARIES(sems ${CMAKE_DL_LIBS} samplerate)
 ENDIF(LIBSAMPLERATE_FOUND)
 
-IF(SEMS_USE_IPV6)
-	ADD_DEFINITIONS(-DSUPPORT_IPV6)
-ENDIF(SEMS_USE_IPV6)
-
 ADD_SUBDIRECTORY(plug-in)
 
 # Create config-file from template
-- 
1.7.3.4

