From b22778270ad05fed21797d5a5d4b782d6e273a59 Mon Sep 17 00:00:00 2001
From: Jean-Louis Dupond <jean-louis@dupond.be>
Date: Sat, 5 May 2012 17:15:29 +0200
Subject: [PATCH] Fix memory leak

---
 remmina-plugins/rdp/rdp_cliprdr.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/remmina-plugins/rdp/rdp_cliprdr.c b/remmina-plugins/rdp/rdp_cliprdr.c
index 772709c..ea8baa0 100644
--- a/remmina-plugins/rdp/rdp_cliprdr.c
+++ b/remmina-plugins/rdp/rdp_cliprdr.c
@@ -74,6 +74,7 @@ void remmina_rdp_cliprdr_get_target_types(uint32** dst_formats, uint16* size, Gd
 			image = TRUE;
 			matches++;
 		}
+		g_free(name);
 	}
 	//Only add text formats if we don't have image formats
 	if (!image)
-- 
1.7.10

