--- ./remmina/CMakeLists.txt.orig
+++ ./remmina/CMakeLists.txt
@@ -138,7 +138,7 @@
 endif()
 
 if(GTK3_FOUND)
-	set(_VTE_VERSION_NUM 2.90)
+	set(_VTE_VERSION_NUM 2.91)
 else()
 	set(_VTE_VERSION_NUM)
 endif()
--- ./cmake/FindVTE.cmake.orig
+++ ./cmake/FindVTE.cmake
@@ -21,11 +21,9 @@
 	set(_VTE_LIB_NAME vte)
 	set(_VTE_VERSION vte)
 else()
-	set(_VTE_LIB_NAME vte${_VTE_VERSION_NUM})
+	set(_VTE_LIB_NAME vte-${_VTE_VERSION_NUM})
 	set(_VTE_VERSION vte-${_VTE_VERSION_NUM})
 endif()
-
-string(REPLACE . _ _VTE_LIB_NAME ${_VTE_LIB_NAME})
 
 pkg_check_modules(PC_VTE ${_VTE_VERSION})
 
--- ./remmina/src/remmina_ssh_plugin.c.orig
+++ ./remmina/src/remmina_ssh_plugin.c
@@ -136,7 +136,7 @@
 	THREADS_ENTER
 	if (charset && charset[0] != '\0')
 	{
-		vte_terminal_set_encoding (VTE_TERMINAL (gpdata->vte), charset);
+		vte_terminal_set_encoding (VTE_TERMINAL (gpdata->vte), charset, NULL);
 	}
 	vte_terminal_set_pty (VTE_TERMINAL (gpdata->vte), shell->slave);
 	THREADS_LEAVE
@@ -183,7 +183,7 @@
 	gpdata = (RemminaPluginSshData*) g_object_get_data (G_OBJECT(gp), "plugin-data");
 	if (remmina_pref.vte_font && remmina_pref.vte_font[0])
 	{
-		vte_terminal_set_font_from_string (VTE_TERMINAL (gpdata->vte), remmina_pref.vte_font);
+		vte_terminal_set_font (VTE_TERMINAL (gpdata->vte), pango_font_description_from_string (remmina_pref.vte_font));
 	}
 	vte_terminal_set_allow_bold (VTE_TERMINAL (gpdata->vte), remmina_pref.vte_allow_bold_text);
 	if (remmina_pref.vte_lines > 0)
@@ -224,7 +224,7 @@
 	remmina_plugin_service->protocol_plugin_register_hostkey (gp, vte);
 
 #if GTK_VERSION == 3
-	vscrollbar = gtk_scrollbar_new (GTK_ORIENTATION_VERTICAL, vte_terminal_get_adjustment (VTE_TERMINAL (vte)));
+	vscrollbar = gtk_scrollbar_new (GTK_ORIENTATION_VERTICAL, gtk_scrollable_get_vadjustment (GTK_SCROLLABLE (vte)));
 #elif GTK_VERSION == 2
 	vscrollbar = gtk_vscrollbar_new (vte_terminal_get_adjustment (VTE_TERMINAL (vte)));
 #endif
