From 5c4ce980285aca894c71c1bdbfce35f05d0b0261 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 20 Apr 2024 23:43:07 +0200
Subject: [PATCH] zig 0.12: adapt build.zig

---
 build.zig | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/build.zig b/build.zig
index c15e65d..a61082e 100644
--- a/build.zig
+++ b/build.zig
@@ -1,8 +1,6 @@
 const std = @import("std");
-const Builder = std.build.Builder;
-const LibExeObjStep = std.build.LibExeObjStep;
 
-fn setDependencies(step: *std.build.LibExeObjStep) void {
+fn setDependencies(step: *std.Build.Step.Compile) void {
     step.linkSystemLibrary("c");
     step.linkSystemLibrary("SDL2");
     step.linkSystemLibrary("SDL2_gfx");
@@ -11,7 +9,7 @@ fn setDependencies(step: *std.build.LibExeObjStep) void {
     step.linkSystemLibrary("curl");
 }
 
-pub fn build(b: *Builder) void {
+pub fn build(b: *std.Build) void {
     b.installDirectory(.{
         .source_dir = .{ .path = "scripts" },
         .install_dir = .{ .bin = {} },
