From 78cc65d4d5b8e5613e62751b0ce0e82992f1365d Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sun, 21 Apr 2024 00:13:09 +0200
Subject: [PATCH] zig 0.12: std.mem.copy -> std.mem.copyForwards

---
 src/Mepo.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Mepo.zig b/src/Mepo.zig
index eb7f8cf..9541a2b 100644
--- a/src/Mepo.zig
+++ b/src/Mepo.zig
@@ -362,7 +362,7 @@ pub fn mepolang_execute(mepo: *@This(), mepolang_text: []const u8) !void {
         if (FnTable.get(statement[0])) |fn_spec| {
             const args: []types.MepoArg = try utilmepolang.argize(arena.allocator(), statement[1..statement.len]);
             var caller_args: [types.MepoFnNargs]types.MepoArg = undefined;
-            std.mem.copy(types.MepoArg, caller_args[0..], args);
+            std.mem.copyForwards(types.MepoArg, caller_args[0..], args);
 
             utildbg.log("Running API function for: {s} requested by mepolang input statement: {s}\n", .{ statement[0], statement });
             _ = mepolang_execute_validate_args(fn_spec, args) catch |e| {
