From 004a6d00e6cd0602d297d60ca270589684ed2f90 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sun, 21 Apr 2024 00:18:16 +0200
Subject: [PATCH] zig 0.12: std.os.exit -> std.posix.exit

---
 src/Mepo.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Mepo.zig b/src/Mepo.zig
index 9541a2b..998f7d9 100644
--- a/src/Mepo.zig
+++ b/src/Mepo.zig
@@ -618,7 +618,7 @@ pub fn graceful_terminate_sdl() void {
     sdl.IMG_Quit();
     sdl.SDL_VideoQuit();
     sdl.SDL_Quit();
-    std.os.exit(0);
+    std.posix.exit(0);
 }
 
 pub fn sighandle_terminate(arg: c_int) callconv(.C) void {
