From 22fa7887383187260db8a8506c3b7f4c8b21e5b2 Mon Sep 17 00:00:00 2001
From: dmo <marillat@users.noreply.github.com>
Date: Mon, 12 Feb 2024 09:40:31 +0100
Subject: [PATCH 1/2] Fix SyntaxWarning in #213

---
 extruct/rdfa.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extruct/rdfa.py b/extruct/rdfa.py
index 273b0b5..58a0dd9 100644
--- a/extruct/rdfa.py
+++ b/extruct/rdfa.py
@@ -85,7 +85,7 @@ def _replaceNS(self, prop, html_element, head_element):
 
         match = None
         if head_element.get("prefix"):
-            match = re.search(prefix + ": [^\s]+", head_element.get("prefix"))
+            match = re.search(prefix + ": [^\\s]+", head_element.get("prefix"))
 
         # if namespace taken from prefix attribute in head tag
         if match:

From dc8ef20b99d02ce720284e731e842e1cd7fd6da1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Adri=C3=A1n=20Chaves?= <adrian@chaves.io>
Date: Fri, 12 Apr 2024 13:31:48 +0200
Subject: [PATCH 2/2] Update extruct/rdfa.py

Co-authored-by: Andrey Rakhmatullin <wrar@wrar.name>
---
 extruct/rdfa.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extruct/rdfa.py b/extruct/rdfa.py
index 58a0dd9..c313e2b 100644
--- a/extruct/rdfa.py
+++ b/extruct/rdfa.py
@@ -85,7 +85,7 @@ def _replaceNS(self, prop, html_element, head_element):
 
         match = None
         if head_element.get("prefix"):
-            match = re.search(prefix + ": [^\\s]+", head_element.get("prefix"))
+            match = re.search(prefix + r": [^\s]+", head_element.get("prefix"))
 
         # if namespace taken from prefix attribute in head tag
         if match:
