diff --git a/common/dynamips_common.h b/common/dynamips_common.h
index 9c20f38..0c8ca2c 100644
--- a/common/dynamips_common.h
+++ b/common/dynamips_common.h
@@ -50,7 +50,7 @@
 #define ARCH_BIG_ENDIAN     0x4321
 #define ARCH_LITTLE_ENDIAN  0x1234
 
-#if defined(PPC) || defined(__powerpc__) || defined(__ppc__)
+#if defined(PPC) || defined(__powerpc__) || defined(__ppc__) || defined(__s390x__)
 #define ARCH_BYTE_ORDER ARCH_BIG_ENDIAN
 #elif defined(__sparc) || defined(__sparc__)
 #define ARCH_BYTE_ORDER ARCH_BIG_ENDIAN
