--- a/libgsystemservice/meson.build
+++ b/libgsystemservice/meson.build
@@ -23,7 +23,6 @@ libgsystemservice_public_deps = [
   dependency('gobject-2.0', version: '>= 2.44'),
 ]
 libgsystemservice_private_deps = [
-  dependency('libsystemd'),
   dependency('polkit-gobject-1'),
 ]
 
--- a/libgsystemservice/service.c
+++ b/libgsystemservice/service.c
@@ -34,7 +34,6 @@
 #include <polkit/polkit.h>
 #include <stdio.h>
 #include <sysexits.h>
-#include <systemd/sd-daemon.h>
 
 
 /**
@@ -565,9 +564,6 @@ name_acquired_cb (GDBusConnection *connection,
 {
   GssService *self = GSS_SERVICE (user_data);
 
-  /* Notify systemd we’re ready. */
-  sd_notify (0, "READY=1");
-
   /* Potentially start a timeout to exiting due to inactivity. */
   gss_service_release (self);
 }
@@ -975,9 +971,6 @@ gss_service_run (GssService  *self,
 
   gss_service_hold (self);
 
-  /* Notify systemd we’re shutting down. */
-  sd_notify (0, "STOPPING=1");
-
   /* Debug. */
   g_debug ("Shutting down: cancellable: %s, run_error: %s, run_exited: %s, "
            "run_exit_signal: %d",
