diff --git a/plugins/Makefile b/plugins/Makefile
index a93142e..304df7b 100644
--- a/plugins/Makefile
+++ b/plugins/Makefile
@@ -385,7 +385,7 @@ $(COVERAGE_LIB): $(COVERAGE_OBJS)
 INSTALL = install
 INSTALL_DIRS = $(INSTALL) -d
 
-INSTALL_DBUS_CONF_DIR = $(DESTDIR)/etc/dbus-1/system.d
+INSTALL_DBUS_CONF_DIR = $(DESTDIR)/usr/share/dbus-1/system.d
 
 install: $(INSTALL_DBUS_CONF_DIR)
 	$(INSTALL) -m 644 */*.conf $(INSTALL_DBUS_CONF_DIR)

