no setuptools_scm in main
--
diff --git a/setup.py b/setup.py
index e82a832..da01efd 100755
--- a/setup.py
+++ b/setup.py
@@ -49,11 +49,8 @@ class build_py(_build_py):
 
 setup(
     name='libfdt',
-    use_scm_version={
-        "root": srcdir,
-    },
     cmdclass = {'build_py' : build_py},
-    setup_requires = ['setuptools_scm'],
+    version="@VERSION@",
     author='Simon Glass',
     author_email='sjg@chromium.org',
     description='Python binding for libfdt',
