diff -upr simh-3.11-1.orig/PDP11/pdp11_defs.h simh-3.11-1/PDP11/pdp11_defs.h
--- simh-3.11-1.orig/PDP11/pdp11_defs.h	2021-07-24 09:40:30.894068560 +0200
+++ simh-3.11-1/PDP11/pdp11_defs.h	2021-07-24 09:41:15.954102912 +0200
@@ -900,13 +900,13 @@ void cpu_set_boot (int32 pc);
 #define WrMemW(pa,d)    uc15_WrMemW (pa, d)
 #define WrMemB(pa, d)   uc15_WrMemB (pa, d)
 
-uint32 uc15_memsize;
-int32 uc15_RdMemW (int32 pa);
-int32 uc15_RdMemB (int32 pa);
-void uc15_WrMemW (int32 pa, int32 d);
-void uc15_WrMemB (int32 pa, int32 d);
-int32 Map_Read18 (uint32 ba, int32 bc, uint32 *buf);
-int32 Map_Write18 (uint32 ba, int32 bc, uint32 *buf);
+extern uint32 uc15_memsize;
+extern int32 uc15_RdMemW (int32 pa);
+extern int32 uc15_RdMemB (int32 pa);
+extern void uc15_WrMemW (int32 pa, int32 d);
+extern void uc15_WrMemB (int32 pa, int32 d);
+extern int32 Map_Read18 (uint32 ba, int32 bc, uint32 *buf);
+extern int32 Map_Write18 (uint32 ba, int32 bc, uint32 *buf);
 
 #else                                                   /* PDP-11 */
 
