Index: ocaml-4.02.2/tools/Makefile.shared
===================================================================
--- ocaml-4.02.2.orig/tools/Makefile.shared
+++ ocaml-4.02.2/tools/Makefile.shared
@@ -286,7 +286,7 @@ CCOUT = -o $(EMPTY)
 endif
 
 objinfo_helper$(EXE): objinfo_helper.c ../config/s.h
-	$(BYTECC) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
+	$(MKEXE) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
           objinfo_helper.c $(LIBBFD_LINK)
 
 OBJINFO=../compilerlibs/ocamlcommon.cma \
