Index: ocaml-3.12.0/tools/Makefile.shared
===================================================================
--- ocaml-3.12.0.orig/tools/Makefile.shared
+++ ocaml-3.12.0/tools/Makefile.shared
@@ -229,7 +229,7 @@ beforedepend:: opnames.ml
 # Display info on compiled files
 
 objinfo_helper$(EXE): objinfo_helper.c ../config/s.h
-	$(BYTECC) -o objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
+	$(MKEXE) -o objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
           objinfo_helper.c $(LIBBFD_LINK)
 
 OBJINFO=../utils/misc.cmo ../utils/config.cmo ../bytecomp/bytesections.cmo \
