diff --git a/server/etcdmain/etcd.go b/server/etcdmain/etcd.go
index 9921b89..888b792 100644
--- a/server/etcdmain/etcd.go
+++ b/server/etcdmain/etcd.go
@@ -472,7 +472,8 @@ func checkSupportArch() {
 	if runtime.GOARCH == "amd64" ||
 		runtime.GOARCH == "arm64" ||
 		runtime.GOARCH == "ppc64le" ||
-		runtime.GOARCH == "s390x" {
+		runtime.GOARCH == "s390x" ||
+                runtime.GOARCH == "loong64" {
 		return
 	}
 	// unsupported arch only configured via environment variable

