diff --git a/setup.py b/setup.py
index 4f0debf..0b07b24 100644
--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'docutils',
         ],
         testing=[
-            'nose',
+            'pytest',
             'pep8 >=1.3',
             'reportlab',
             'docutils',
@@ -67,7 +67,6 @@ setup(
             'flake8-isort',
         ],
     ),
-    test_suite='nose.collector',
     entry_points="""
        [console_scripts]
        actdiag = actdiag.command:main
diff --git a/src/actdiag/tests/test_generate_diagram.py b/src/actdiag/tests/test_generate_diagram.py
index c5ee3d5..4b0d71b 100644
--- a/src/actdiag/tests/test_generate_diagram.py
+++ b/src/actdiag/tests/test_generate_diagram.py
@@ -14,18 +14,23 @@
 #  limitations under the License.
 
 import os
+import re
+import pytest
 
 from blockdiag.tests.test_generate_diagram import (get_diagram_files,
-                                                   testcase_generator)
+                                                   generate)
 
 import actdiag.command
 
+base_path = os.path.dirname(__file__)
+files = get_diagram_files(base_path)
+generate_testdata = []
+for file_source in files:
+    # FIXME generate_testdata.append((file_source, 'svg', []))
+    generate_testdata.append((file_source, 'png', []))
+    generate_testdata.append((file_source, 'png', ['--antialias']))
+    generate_testdata.append((file_source, 'pdf', []))
 
-def test_generate():
-    mainfunc = actdiag.command.main
-    basepath = os.path.dirname(__file__)
-    files = get_diagram_files(basepath)
-    options = []
-
-    for testcase in testcase_generator(basepath, mainfunc, files, options):
-        yield testcase
+@pytest.mark.parametrize("source,file_type,options", generate_testdata)
+def test_generate(source, file_type, options):
+    generate(source, file_type, options)
