diff --git a/tornado/Common.h b/tornado/Common.h
index 0373316..4f7bd7c 100644
--- a/tornado/Common.h
+++ b/tornado/Common.h
@@ -15,7 +15,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
-#if defined(_M_X64) || defined(_M_AMD64) || defined(__x86_64__) || defined(_ARCH_PPC64) || defined(__arm64__) || defined(__aarch64__)
+#if defined(_M_X64) || defined(_M_AMD64) || defined(__x86_64__) || defined(_ARCH_PPC64) || defined(__arm64__) || defined(__aarch64__) || defined(__loongarch64)
 #define FREEARC_64BIT
 #endif
 
