From f8c3aff70bbea57527ee301c16b94dcc8adc4cc9 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Tue, 1 Mar 2016 16:23:12 +0100
Subject: [PATCH 5/8] Set _DATE_FMT to '%+' if not defined in
 libspl/timestamp.c

Signed-off-by: Carlo Landmeter <clandmeter@gmail.com>
Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>
Closes #4458
---
 lib/libspl/timestamp.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/libspl/timestamp.c b/lib/libspl/timestamp.c
index a4f4cf4..e2838da 100644
--- a/lib/libspl/timestamp.c
+++ b/lib/libspl/timestamp.c
@@ -28,6 +28,10 @@
 #include <langinfo.h>
 #include "statcommon.h"
 
+#ifndef _DATE_FMT
+#define	_DATE_FMT "%+"
+#endif
+
 /*
  * Print timestamp as decimal reprentation of time_t value (-T u was specified)
  * or in date(1) format (-T d was specified).
-- 
2.7.4

