--- a/configure
+++ b/configure
@@ -103580,7 +103580,7 @@ printf "%s\n" "no" >&6; }
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking default bash completion directory" >&5
 printf %s "checking default bash completion directory... " >&6; }
 	case "$VENDOR" in
-		arch|artix|debian|gentoo|ubuntu)
+		alpine|arch|artix|debian|gentoo|ubuntu)
 			bashcompletiondir=/usr/share/bash-completion/completions
 			;;
 		freebsd)
