--- a/configure
+++ b/configure
@@ -139180,6 +139180,7 @@ printf "%s\n" "no" >&6; }
 	{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking default bash completion directory" >&5
 printf %s "checking default bash completion directory... " >&6; }
 	case "$VENDOR" in
+		alpine)     bashcompletiondir=/usr/share/bash-completion/completions   ;;
 		ubuntu)     bashcompletiondir=/usr/share/bash-completion/completions   ;;
 		debian)     bashcompletiondir=/usr/share/bash-completion/completions   ;;
 		freebsd)    bashcompletiondir=$sysconfdir/bash_completion.d;;
