diff -rupN a/opfunu/cec_based/cec.py b/opfunu/cec_based/cec.py
--- a/opfunu/cec_based/cec.py	2024-06-04 16:42:29.000000000 +0000
+++ b/opfunu/cec_based/cec.py	2026-03-17 02:33:27.570000000 +0000
@@ -5,7 +5,7 @@
 # --------------------------------------------------%
 
 import numpy as np
-import pkg_resources
+from importlib.resources import as_file, files
 from abc import ABC
 from opfunu.benchmark import Benchmark
 
@@ -78,7 +78,7 @@ class CecBenchmark(Benchmark, ABC):
         self.verbose = False
 
     def make_support_data_path(self, data_name):
-        self.support_path = pkg_resources.resource_filename("opfunu", f"cec_based/{data_name}")
+        self.support_path = files("opfunu") / "cec_based" / data_name
 
     def check_shift_data(self, f_shift):
         if type(f_shift) is str:
@@ -135,30 +135,34 @@ class CecBenchmark(Benchmark, ABC):
             raise ValueError(f"m_group is positive integer!")
 
     def load_shift_data(self, filename=None):
-        data = np.genfromtxt(f"{self.support_path}/{filename}.txt", dtype=float)
-        return data.reshape((-1))
+        with as_file(self.support_path) as path:
+            data = np.genfromtxt(f"{path}/{filename}.txt", dtype=float)
+            return data.reshape((-1))
 
     def load_matrix_data(self, filename=None):
         try:
-            data = np.genfromtxt(f"{self.support_path}/{filename}.txt", dtype=float)
-            return data
+            with as_file(self.support_path) as path:
+                data = np.genfromtxt(f"{path}/{filename}.txt", dtype=float)
+                return data
         except FileNotFoundError:
             print(f'The file named: {filename}.txt is not found.')
             print(f"{self.__class__.__name__} problem is only supported ndim in {self.dim_supported}!")
             exit(1)
 
     def load_shift_and_matrix_data(self, filename=None):
-        data = np.genfromtxt(f"{self.support_path}/{filename}.txt", dtype=float)
-        shift_data = data[:1, :].ravel()
-        matrix_data = data[1:, :]
-        return shift_data, matrix_data
+        with as_file(self.support_path) as path:
+            data = np.genfromtxt(f"{path}/{filename}.txt", dtype=float)
+            shift_data = data[:1, :].ravel()
+            matrix_data = data[1:, :]
+            return shift_data, matrix_data
 
     def load_two_matrix_and_shift_data(self, filename=None):
-        data = np.genfromtxt(f"{self.support_path}/{filename}.txt", dtype=float)
-        a_matrix = data[:100, :]
-        b_matrix = data[100:200, :]
-        shift_data = data[200:, :].ravel()
-        return shift_data, a_matrix, b_matrix
+        with as_file(self.support_path) as path:
+            data = np.genfromtxt(f"{path}/{filename}.txt", dtype=float)
+            a_matrix = data[:100, :]
+            b_matrix = data[100:200, :]
+            shift_data = data[200:, :].ravel()
+            return shift_data, a_matrix, b_matrix
 
     def check_solution(self, x, dim_max=None, dim_support=None):
         """
