diff --git a/extras/Makefile.in b/extras/Makefile.in
index a29780f..3e19a79 100644
--- a/extras/Makefile.in
+++ b/extras/Makefile.in
@@ -69,7 +69,6 @@ install:	all
 	rm -f $(BINDIR)/makeweb $(BINDIR)/htpasswd $(BINDIR)/syslogtocern
 	cp makeweb $(BINDIR)/makeweb
 	chgrp $(WEBGROUP) $(BINDIR)/makeweb
-	chmod 2755 $(BINDIR)/makeweb
 	cp htpasswd $(BINDIR)/htpasswd
 	cp syslogtocern $(BINDIR)/syslogtocern
 	rm -f $(MANDIR)/man1/makeweb.1
