diff --git a/src/insets/InsetListings.h b/src/insets/InsetListings.h
index 4be28f8..d4736a5 100644
--- a/src/insets/InsetListings.h
+++ b/src/insets/InsetListings.h
@@ -14,6 +14,7 @@
 
 #include "InsetListingsParams.h"
 
+#include <cstring>
 
 namespace lyx {
 
diff --git a/src/lyxfind.cpp b/src/lyxfind.cpp
index a3a0b8b..7feb0eb 100644
--- a/src/lyxfind.cpp
+++ b/src/lyxfind.cpp
@@ -33,6 +33,8 @@
 #include "TexRow.h"
 #include "Text.h"
 
+#include <iterator>
+
 #include "frontends/Application.h"
 #include "frontends/alert.h"
 
