diff --git a/tests/Makefile.in b/tests/Makefile.in
index 4d77c1f..b6fa2c3 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -1586,7 +1586,6 @@ inout-3.tst	\
 options-1.tst	\
 pages-1.tst	\
 pages-2.tst	\
-printers.tst	\
 prolog-1.tst	\
 prolog-2.tst	\
 strip.tst	\
@@ -1899,13 +1898,6 @@ pages-2.tst.log: pages-2.tst
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-printers.tst.log: printers.tst
-	@p='printers.tst'; \
-	b='printers.tst'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 prolog-1.tst.log: prolog-1.tst
 	@p='prolog-1.tst'; \
 	b='prolog-1.tst'; \
