--- a2ps-4.14.orig/lib/options.c
+++ a2ps-4.14/lib/options.c
@@ -57,7 +57,6 @@
 
 #define MAN_LINES               66	/* no lines for a man */
 extern char *program_name;
-extern const char *program_invocation_name;
 
 /*
  * Hooks used
--- a2ps-4.14.orig/src/main.c
+++ a2ps-4.14/src/main.c
@@ -104,7 +104,6 @@
    defined twice, see lib/confg.gperf, handling of `Options:'. */
 
 char *program_name;
-const char *program_invocation_name;
 
 /* Stores the data of liba2ps.  */
 
