diff -upr booster-0.7.orig/generator/console.go booster-0.7/generator/console.go
--- booster-0.7.orig/generator/console.go	2022-04-02 18:59:39.005144892 +0200
+++ booster-0.7/generator/console.go	2022-04-02 19:00:09.075241832 +0200
@@ -104,14 +104,14 @@ func loadKeymap(keymap, keymapToggle str
 }
 
 func readFontFile(font string) (blob []byte, err error) {
-	entries, err := os.ReadDir("/usr/share/kbd/consolefonts/")
+	entries, err := os.ReadDir("/usr/share/consolefonts/")
 	if err != nil {
 		return nil, err
 	}
 	for _, d := range entries {
 		name := d.Name()
 		if strings.HasPrefix(name, font+".") {
-			fileName := "/usr/share/kbd/consolefonts/" + name
+			fileName := "/usr/share/consolefonts/" + name
 			debug("font %s matched to file %s", font, fileName)
 			blob, err := os.ReadFile(fileName)
 			if err != nil {
