From 39d1b1a4cea4f5f562d571bd6b090e72c057becb Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Mon, 4 Mar 2019 22:03:25 +0100
Subject: [PATCH] [PATCH scdoc] use TZ=UTC in test case

---
 test/preamble | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/test/preamble b/test/preamble
index 969d674..71f1372 100755
--- a/test/preamble
+++ b/test/preamble
@@ -70,10 +70,11 @@ test-manual(8) "" "Header"
 EOF
 end 0
 
+export TZ=UTC
 export SOURCE_DATE_EPOCH=$(date --date="2017-12-09 23:18:57" +'%s')
 
 begin "Supports \$SOURCE_DATE_EPOCH"
-scdoc <<EOF | grep '^\.TH "reproducible-manual" "8" "2017-12-10"' >/dev/null
+scdoc <<EOF | grep '^\.TH "reproducible-manual" "8" "2017-12-09"' >/dev/null
 reproducible-manual(8)
 EOF
 end 0
-- 
2.21.0

