diff -rupN a/src/XspfReader.cpp b/src/XspfReader.cpp
--- a/src/XspfReader.cpp	2021-05-29 11:04:32.000000000 +0000
+++ b/src/XspfReader.cpp	2026-04-23 16:06:33.950000000 +0000
@@ -61,7 +61,8 @@
 #include <cstring> // size_t
 #include <cstddef> // int
 
-#if (URI_VER_MINOR < 7) || ((URI_VER_MINOR == 7) && (URI_VER_RELEASE < 2))
+#if ((URI_VER_MAJOR == 0) && (URI_VER_MINOR < 7)) || \
+((URI_VER_MAJOR == 0) && (URI_VER_MINOR == 7) && (URI_VER_RELEASE < 2))
 # error uriparser 0.7.2 or later is required
 #endif
 
