From 65d71ea5f80b6d56907bd67825981042eaf98d6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 10 Jan 2017 09:51:32 +0200
Subject: [PATCH 3/3] fix aarch64 compile error due to undefined variable

fixes #36
---
 procinterrupts.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/procinterrupts.c b/procinterrupts.c
index 6b37a88..c5c034c 100644
--- a/procinterrupts.c
+++ b/procinterrupts.c
@@ -148,6 +148,9 @@ GList* collect_full_irq_list()
 	char *line = NULL;
 	size_t size = 0;
 	char *irq_name, *irq_mod, *savedptr, *last_token, *p;
+#ifdef AARCH64
+	char *tmp;
+#endif
 
 	file = fopen("/proc/interrupts", "r");
 	if (!file)
-- 
2.11.0

