diff --git a/src/process_handling/mock_event_source.rs b/src/process_handling/mock_event_source.rs
index 034518d..04906f5 100644
--- a/src/process_handling/mock_event_source.rs
+++ b/src/process_handling/mock_event_source.rs
@@ -100,7 +100,7 @@ impl MockBuilder {
     pub fn breakpoint(mut self, addr: u64) -> Self {
         self.push_step(ThreadStep::Breakpoint(addr));
         // pre-populate the image so read_address returns something non-zero
-        self.image.entry(addr & !0x7).or_insert(0xDEAD_BEEF); // ends with INT3
+        self.image.entry(addr & !0x7).or_insert(0xDEAD_BEEFu32 as c_long); // ends with INT3
         self
     }
 
