--- a/hledger-lib/hledger-lib.cabal
+++ b/hledger-lib/hledger-lib.cabal
@@ -113,7 +113,7 @@
     , aeson-pretty
     , ansi-terminal >=0.9
     , array
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , base-compat
     , blaze-markup >=0.5.1
     , bytestring
@@ -171,7 +171,7 @@
     , aeson-pretty
     , ansi-terminal >=0.9
     , array
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , base-compat
     , blaze-markup >=0.5.1
     , bytestring
@@ -232,7 +232,7 @@
     , aeson-pretty
     , ansi-terminal >=0.9
     , array
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , base-compat
     , blaze-markup >=0.5.1
     , bytestring
--- a/hledger-ui/hledger-ui.cabal
+++ b/hledger-ui/hledger-ui.cabal
@@ -72,7 +72,7 @@
   build-depends:
       ansi-terminal >=0.9
     , async
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , brick >=2.1.1 && <2.4
     , cmdargs >=0.8
     , containers >=0.5.9
@@ -115,7 +115,7 @@
   ghc-options: -Wall -Wno-incomplete-uni-patterns -Wno-missing-signatures -Wno-orphans -Wno-type-defaults -Wno-unused-do-bind
   cpp-options: -DVERSION="1.32.3"
   build-depends:
-      base >=4.14 && <4.19
+      base >=4.14 && <4.20
     , hledger-ui
   default-language: Haskell2010
   if flag(threaded)
--- a/hledger-web/hledger-web.cabal
+++ b/hledger-web/hledger-web.cabal
@@ -155,7 +155,7 @@
   build-depends:
       Decimal >=0.5.1
     , aeson >=1 && <2.3
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , base64
     , blaze-html
     , blaze-markup
@@ -215,7 +215,7 @@
   ghc-options: -Wall -Wredundant-constraints -Wincomplete-record-updates -Wincomplete-uni-patterns
   cpp-options: -DVERSION="1.32.3"
   build-depends:
-      base >=4.14 && <4.19
+      base >=4.14 && <4.20
     , hledger-web
   default-language: Haskell2010
   if (flag(dev)) || (flag(library-only))
@@ -235,7 +235,7 @@
   ghc-options: -Wall -Wredundant-constraints -Wincomplete-record-updates -Wincomplete-uni-patterns
   cpp-options: -DVERSION="1.32.3"
   build-depends:
-      base >=4.14 && <4.19
+      base >=4.14 && <4.20
     , hledger-web
   default-language: Haskell2010
   if (flag(dev)) || (flag(library-only))
--- a/hledger/hledger.cabal
+++ b/hledger/hledger.cabal
@@ -142,7 +142,7 @@
     , Diff >=0.2
     , aeson >=1 && <2.3
     , ansi-terminal >=0.9
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , bytestring
     , cmdargs >=0.10
     , containers >=0.5.9
@@ -193,7 +193,7 @@
       Decimal >=0.5.1
     , aeson >=1 && <2.3
     , ansi-terminal >=0.9
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , bytestring
     , cmdargs >=0.10
     , containers >=0.5.9
@@ -244,7 +244,7 @@
       Decimal >=0.5.1
     , aeson >=1 && <2.3
     , ansi-terminal >=0.9
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , bytestring
     , cmdargs >=0.10
     , containers >=0.5.9
@@ -292,7 +292,7 @@
       Decimal >=0.5.1
     , aeson >=1 && <2.3
     , ansi-terminal >=0.9
-    , base >=4.14 && <4.19
+    , base >=4.14 && <4.20
     , bytestring
     , cmdargs >=0.10
     , containers >=0.5.9
