--- a/src/httrack.c
+++ b/src/httrack.c
@@ -69,7 +69,8 @@
 #include <unistd.h>
 #endif
 #include <ctype.h>
-#ifdef __linux
+//#ifdef __linux
+#ifdef __GLIBC__
 #include <execinfo.h>
 #endif
 /* END specific definitions */
@@ -878,7 +879,8 @@
 #define FD_ERR 2
 
 static void print_backtrace(void) {
-#ifdef __linux
+//#ifdef __linux
+#ifdef __GLIBC__
   void *stack[256];
   const int size = backtrace(stack, sizeof(stack)/sizeof(stack[0]));
   if (size != 0) {
