--- a/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py.orig
+++ b/src/3rdparty/chromium/tools/gn/bootstrap/bootstrap.py
@@ -168,7 +168,7 @@
   mkdir_p(root_gen_dir)
 
   write_buildflag_header_manually(root_gen_dir, 'base/allocator/features.h',
-      {'USE_EXPERIMENTAL_ALLOCATOR_SHIM': 'true' if is_linux else 'false'})
+      {'USE_EXPERIMENTAL_ALLOCATOR_SHIM': 'false' if is_linux else 'false'})
 
   write_buildflag_header_manually(root_gen_dir, 'base/debug/debugging_flags.h',
       {'ENABLE_PROFILING': 'false'})
