--- a/src/3rdparty/chromium/third_party/yasm/BUILD.gn.orig
+++ b/src/3rdparty/chromium/third_party/yasm/BUILD.gn
@@ -46,6 +46,7 @@
     if (is_posix) {
       cflags = [ "-std=gnu99" ]
     }
+    libs = [ "intl" ]
   }
 
   executable("genmacro") {
--- a/src/3rdparty/chromium/third_party/skia/gyp/yasm.gyp.orig
+++ b/src/3rdparty/chromium/third_party/skia/gyp/yasm.gyp
@@ -199,6 +199,9 @@
         '-mthumb',
         '-march=armv7-a',
       ],
+      'link_settings': {
+        'libraries': [ '-lintl' ],
+      }
       'xcode_settings': {
         'WARNING_CFLAGS': [
           '-w',
