--- a/src/3rdparty/chromium/base/third_party/libevent/BUILD.gn.orig
+++ b/src/3rdparty/chromium/base/third_party/libevent/BUILD.gn
@@ -33,7 +33,7 @@
     include_dirs = [ "mac" ]
   } else if (is_linux) {
     sources += [ "epoll.c" ]
-    include_dirs = [ "linux" ]
+    include_dirs = [ "linux", "compat" ]
   } else if (is_android) {
     sources += [ "epoll.c" ]
     include_dirs = [ "android" ]
