--- a/src/3rdparty/chromium/net/dns/host_resolver_impl.cc.orig
+++ b/src/3rdparty/chromium/net/dns/host_resolver_impl.cc
@@ -2074,8 +2074,7 @@
   NetworkChangeNotifier::AddIPAddressObserver(this);
   NetworkChangeNotifier::AddConnectionTypeObserver(this);
   NetworkChangeNotifier::AddDNSObserver(this);
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
   EnsureDnsReloaderInit();
 #endif
 
--- a/src/3rdparty/chromium/net/dns/dns_reloader.cc.orig
+++ b/src/3rdparty/chromium/net/dns/dns_reloader.cc
@@ -4,8 +4,7 @@
 
 #include "net/dns/dns_reloader.h"
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
 
 #include <resolv.h>
 
--- a/src/3rdparty/chromium/net/dns/host_resolver_proc.cc.orig
+++ b/src/3rdparty/chromium/net/dns/host_resolver_proc.cc
@@ -193,8 +193,7 @@
   // Restrict result set to only this socket type to avoid duplicates.
   hints.ai_socktype = SOCK_STREAM;
 
-#if defined(OS_POSIX) && !defined(OS_MACOSX) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
   DnsReloaderMaybeReload();
 #endif
   int err = getaddrinfo(host.c_str(), NULL, &hints, &ai);
