--- a/src/3rdparty/chromium/third_party/swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h.orig
+++ b/src/3rdparty/chromium/third_party/swiftshader/third_party/LLVM/include-linux/llvm/Config/config.h
@@ -125,7 +125,7 @@
 #define HAVE_ERROR_T 1
 
 /* Define to 1 if you have the <execinfo.h> header file. */
-#define HAVE_EXECINFO_H 1
+/* #undef HAVE_EXECINFO_H */
 
 /* Define to 1 if you have the <fcntl.h> header file. */
 #define HAVE_FCNTL_H 1
@@ -240,10 +240,10 @@
 /* #undef HAVE_MACH_O_DYLD_H */
 
 /* Define if mallinfo() is available on this platform. */
-#define HAVE_MALLINFO 1
+/* #undef HAVE_MALLINFO */
 
 /* Define to 1 if you have the <malloc.h> header file. */
-#define HAVE_MALLOC_H 1
+/* #undef HAVE_MALLOC_H */
 
 /* Define to 1 if you have the <malloc/malloc.h> header file. */
 /* #undef HAVE_MALLOC_MALLOC_H */
--- a/src/3rdparty/chromium/third_party/swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp.orig
+++ b/src/3rdparty/chromium/third_party/swiftshader/third_party/LLVM/lib/Support/DynamicLibrary.cpp
@@ -160,7 +160,7 @@
 // On linux we have a weird situation. The stderr/out/in symbols are both
 // macros and global variables because of standards requirements. So, we
 // boldly use the EXPLICIT_SYMBOL macro without checking for a #define first.
-#if defined(__linux__) && !defined(__ANDROID__)
+#if defined(__GLIBC__)
   {
     EXPLICIT_SYMBOL(stderr);
     EXPLICIT_SYMBOL(stdout);
--- a/src/3rdparty/chromium/third_party/swiftshader/third_party/llvm-subzero/include/llvm/Config/config.h.orig
+++ b/src/3rdparty/chromium/third_party/swiftshader/third_party/llvm-subzero/include/llvm/Config/config.h
@@ -172,7 +172,7 @@
 /* #undef HAVE_MALLINFO */
 
 /* Define to 1 if you have the <malloc.h> header file. */
-#define HAVE_MALLOC_H 1
+/* #define HAVE_MALLOC_H 1 */
 
 /* Define to 1 if you have the <malloc/malloc.h> header file. */
 /* #undef HAVE_MALLOC_MALLOC_H */
--- a/src/3rdparty/chromium/third_party/swiftshader/src/Common/Socket.cpp.orig
+++ b/src/3rdparty/chromium/third_party/swiftshader/src/Common/Socket.cpp
@@ -17,6 +17,7 @@
 #if defined(_WIN32)
 	#include <ws2tcpip.h>
 #else
+	#include <sys/select.h>
 	#include <unistd.h>
 	#include <netdb.h>
 	#include <netinet/in.h>
