Author: Patrycja Rosa <alpine@ptrcnull.me>

	Replace systemctl-hwdb with udevadm

diff --git a/test/test_udev_rules.py b/test/test_udev_rules.py
index 0e8e99c..cc3759e 100644
--- a/test/test_udev_rules.py
+++ b/test/test_udev_rules.py
@@ -20,11 +20,10 @@ import sys
 
 
 @pytest.fixture(scope='session', autouse=True)
-def systemd_reload():
+def udevadm_reload():
     '''Make sure our hwdb and udev rules are up-to-date'''
     import subprocess
-    subprocess.run(['systemd-hwdb', 'update'])
-    subprocess.run(['systemctl', 'daemon-reload'])
+    subprocess.run(['udevadm', 'hwdb', '-u'])
 
 
 def pytest_generate_tests(metafunc):
diff --git a/tools/libwacom-update-db.py b/tools/libwacom-update-db.py
index 54298cf..167aad6 100755
--- a/tools/libwacom-update-db.py
+++ b/tools/libwacom-update-db.py
@@ -241,7 +241,7 @@ if __name__ == "__main__":
             print(f"New hwdb file: {hwdbfile}")
 
             if not ns.skip_systemd_hwdb_update:
-                subprocess.run(["systemd-hwdb", "update"],
+                subprocess.run(["udevadm", "hwdb", "-u"],
                         capture_output=True, check=True, text=True)
             print("Finished, please unplug and replug your device")
         except PermissionError as e:
