Author: Patrycja Rosa <alpine@ptrcnull.me>

	Replace systemctl-hwdb with udevadm

diff --git a/test/test_udev_rules.py b/test/test_udev_rules.py
index b0d023a..610ca41 100644
--- a/test/test_udev_rules.py
+++ b/test/test_udev_rules.py
@@ -19,7 +19,7 @@ import shutil
 
 
 @pytest.fixture(scope="session", autouse=True)
-def systemd_reload():
+def udevadm_reload():
     """Make sure our hwdb and udev rules are up-to-date"""
 
     try:
@@ -33,20 +33,15 @@ def systemd_reload():
 
             warnings.warn("LIBWACOM_HWDB_FILE is not set, using already installed hwdb")
 
-        subprocess.run(["systemd-hwdb", "update"], check=True)
+        subprocess.run(["udevadm", "hwdb", "-u"], check=True)
 
         yield
 
         if hwdb:
             os.unlink(target)
 
-        subprocess.run(["systemd-hwdb", "update"], check=True)
+        subprocess.run(["udevadm", "hwdb", "-u"], check=True)
 
-    except (IOError, FileNotFoundError, subprocess.CalledProcessError) as e:
-        # If any of the commands above are not found (most likely the system
-        # simply does not use systemd), just skip.
-        logging.critical(f"{e}")
-        pytest.skip(f"Skipping test: {e}")
     except Exception as e:
         logging.critical(f"{e}")
         pytest.fail(f"Aborting test: {e}")
@@ -122,7 +117,7 @@ def test_hwdb_files(tablet):
     logging.debug(query)
 
     r = subprocess.run(
-        ["systemd-hwdb", "query", query], check=True, capture_output=True
+        ["udevadm", "hwdb", "-t", query], check=True, capture_output=True
     )
     stdout = r.stdout.decode("utf-8").strip()
     assert stdout, f"No output recorded for query {query}"
