Author: Patrycja Rosa <alpine@ptrcnull.me>

	Replace systemctl-hwdb with udevadm

diff --git a/test/test_udev_rules.py b/test/test_udev_rules.py
index 0e8e99c..cc3759e 100644
--- a/test/test_udev_rules.py
+++ b/test/test_udev_rules.py
@@ -20,11 +20,10 @@ import sys
 
 
 @pytest.fixture(scope='session', autouse=True)
-def systemd_reload():
+def udevadm_reload():
     '''Make sure our hwdb and udev rules are up-to-date'''
     import subprocess
-    subprocess.run(['systemd-hwdb', 'update'])
-    subprocess.run(['systemctl', 'daemon-reload'])
+    subprocess.run(['udevadm', 'hwdb', '-u'])
 
 
 def pytest_generate_tests(metafunc):
