diff --git a/tests/test_find_python_executable.py b/tests/test_find_python_executable.py
index 161dc7a..c6f0ebc 100644
--- a/tests/test_find_python_executable.py
+++ b/tests/test_find_python_executable.py
@@ -18,12 +18,12 @@ def test_self():
 
 
 def test_abs():
-    abs_path = "C:\\PythonXY\\python.exe" if os.name == 'nt' else '/usr/bin/python'
+    abs_path = "C:\\PythonXY\\python.exe" if os.name == 'nt' else '/usr/bin/python3'
     assert find_python_executable(abs_path) == abs_path
 
 
 def test_find_in_path():
-    assert isabs(find_python_executable("python"))
+    assert isabs(find_python_executable("python3"))
 
 
 def test_env(tmp_path):
