waf 2.1.0 switched from optparse to argparse

--- a/wscript
+++ b/wscript
@@ -109,7 +109,7 @@
             help_str = 'build examples (true)',
             help_disable_str = 'do not build examples')
 
-    ctx.add_option('--with-target-platform', type='string',
+    ctx.add_option('--with-target-platform', type=str,
             help='set target platform for cross-compilation',
             dest='target_platform')
 
