--- a/utils/launch.sh
+++ b/utils/launch.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env ash
 
 # Copyright 2016 Joel Martin
 # Copyright 2016 Solly Ross
@@ -141,7 +141,7 @@
 ${WEBSOCKIFY} --web ${WEB} ${CERT:+--cert ${CERT}} ${PORT} ${VNC_DEST} &
 proxy_pid="$!"
 sleep 1
-if ! ps -p ${proxy_pid} >/dev/null; then
+if ! ps | grep ${proxy_pid} >/dev/null; then
     proxy_pid=
     echo "Failed to start WebSockets proxy"
     exit 1
