--- wbar-2.3.4/src/config/Functions.cc	2012-09-19 02:54:53.000000000 +0000
+++ ../../Functions.modified	2015-12-15 15:35:05.920314007 +0000
@@ -120,15 +120,15 @@
     GtkCellRenderer *renderer;
 
     renderer = gtk_cell_renderer_pixbuf_new ();
-    column = gtk_tree_view_column_new_with_attributes (_("Icon"), renderer, "pixbuf", TREE_COLUMN_ICON, NULL);
+    column = gtk_tree_view_column_new_with_attributes (_("Icon"), renderer, "pixbuf", TREE_COLUMN_ICON, (void *)NULL);
     gtk_tree_view_append_column (GTK_TREE_VIEW (treeview), column);
 
     renderer = gtk_cell_renderer_text_new ();
-    column = gtk_tree_view_column_new_with_attributes (_("Title"), renderer, "markup", TREE_COLUMN_TITLE, NULL);
+    column = gtk_tree_view_column_new_with_attributes (_("Title"), renderer, "markup", TREE_COLUMN_TITLE, (void *)NULL);
     gtk_tree_view_append_column (GTK_TREE_VIEW (treeview), column);
 
     renderer = gtk_cell_renderer_text_new ();
-    column = gtk_tree_view_column_new_with_attributes (_("Command"), renderer, "markup", TREE_COLUMN_COMMAND, NULL);
+    column = gtk_tree_view_column_new_with_attributes (_("Command"), renderer, "markup", TREE_COLUMN_COMMAND, (void *)NULL);
     gtk_tree_view_append_column (GTK_TREE_VIEW (treeview), column);
 }
 
@@ -145,7 +145,7 @@
 
     cell = gtk_cell_renderer_text_new();
     gtk_cell_layout_pack_start(GTK_CELL_LAYOUT(combo), cell, TRUE);
-    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(combo), cell, "text", 0, NULL); 
+    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(combo), cell, "text", 0, (void *)NULL);
 
     gtk_list_store_append(list_store, &iter);
     gtk_list_store_set (list_store, &iter, COMBO_COLUMN_TEXT, _("top"), COMBO_COLUMN_ID, POS_TOP, -1);
@@ -182,7 +182,7 @@
 
     cell = gtk_cell_renderer_text_new();
     gtk_cell_layout_pack_start(GTK_CELL_LAYOUT(combo), cell, TRUE);
-    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(combo), cell, "text", 0, NULL); 
+    gtk_cell_layout_set_attributes(GTK_CELL_LAYOUT(combo), cell, "text", 0, (void *)NULL);
 
     gtk_list_store_append(list_store, &iter);
     gtk_list_store_set (list_store, &iter, COMBO_COLUMN_TEXT, _("none"), COMBO_COLUMN_ID, FILTER_NONE, -1);
@@ -1261,7 +1261,7 @@
      				      GTK_FILE_CHOOSER_ACTION_OPEN,
      				      GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
      				      GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
-     				      NULL);
+     				      (void *)NULL);
 
     GtkWidget * command = glade_xml_get_widget (xml, "entry_command");
     std::string filename = gtk_entry_get_text (GTK_ENTRY(command));
@@ -1285,7 +1285,7 @@
      				      GTK_FILE_CHOOSER_ACTION_OPEN,
      				      GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
      				      GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
-     				      NULL);
+     				      (void *)NULL);
 
     GtkWidget * icon = glade_xml_get_widget (xml, "entry_icon");
     GtkFileFilter *filter = gtk_file_filter_new();
@@ -1315,7 +1315,7 @@
      				      GTK_FILE_CHOOSER_ACTION_OPEN,
      				      GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
      				      GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
-     				      NULL);
+     				      (void *)NULL);
 
     GtkWidget * font = glade_xml_get_widget (xml, "entry_font_name");
     GtkFileFilter *filter = gtk_file_filter_new();
@@ -1347,7 +1347,7 @@
      				      GTK_FILE_CHOOSER_ACTION_OPEN,
      				      GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL,
      				      GTK_STOCK_OPEN, GTK_RESPONSE_ACCEPT,
-     				      NULL);
+     				      (void *)NULL);
 
     GtkWidget * icon = glade_xml_get_widget (xml, "entry_image_name");
     GtkFileFilter *filter = gtk_file_filter_new();
@@ -1400,6 +1400,6 @@
         "documenters", documenters,
         "translator-credits", translater,
         "logo", pixbuf,
-        NULL);
+        (void *)NULL);
 }
 
