--- wbar-2.3.4/src/core/Main.cc	2012-09-18 02:26:41.000000000 +0000
+++ ../../Main.modified	2015-12-15 15:30:14.362095758 +0000
@@ -326,7 +326,7 @@
                 	{
                     	    if (fork()==0)
                     	    {
-                        	if (execlp("sh", "sh", "-c", barra->iconCommand(inum).c_str(), NULL) != 0)
+                        	if (execlp("sh", "sh", "-c", barra->iconCommand(inum).c_str(), (void *)NULL) != 0)
                         	{
                             	    std::cout << _("Error run program: ") << barra->iconCommand(inum) << std::endl;
                         	}
