From bd36f7e1710fd429c80ef1814beb81db909c177d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 21 Nov 2014 12:02:46 +0200
Subject: [PATCH] fix missing symbols from lua module

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 8814b15..f8acd32 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -7,7 +7,7 @@ common-objs		+= filterdb.o authdb.o blob.o addr.o config.o reporting.o
 squark-filter-objs	+= squark-filter.o $(common-objs)
 squark-auth-snmp-objs	+= squark-auth-snmp.o $(common-objs)
 squark-auth-ip-objs	+= squark-auth-ip.o $(common-objs)
-squarkdb.so-objs	+= lua-squarkdb.o filterdb.o blob.o
+squarkdb.so-objs	+= lua-squarkdb.o filterdb.o blob.o reporting.o
 
 CFLAGS_config.o		+= -DSQUARK_VERSION=\"$(FULL_VERSION)\" \
 			   -DCONFDIR=\"$(CONFDIR)\" \
-- 
2.2.0

