--- a/autogen.sh
+++ b/autogen.sh
@@ -1,20 +1,6 @@
 #! /bin/sh
 
-if ! GRAPHVIZ_GIT_DATE=$( git log -n 1 --format=%ci ) ; then
-    GRAPHVIZ_VERSION_DATE="0"
-    echo "Warning: build not started in a Git clone, or Git is not installed: setting version date to 0." >&2
-else
-    if ! GRAPHVIZ_VERSION_DATE=$( date -u +%Y%m%d.%H%M -d "$GRAPHVIZ_GIT_DATE" 2>/dev/null ) ; then
-        # try date with FreeBSD syntax
-        if ! GRAPHVIZ_VERSION_DATE=$( date -u -j -f "%Y-%m-%d %H:%M:%S %z" "$GRAPHVIZ_GIT_DATE" +%Y%m%d.%H%M ); then
-            echo "Warning: we do not know how to invoke date correctly." >&2
-        fi    
-    fi
-    echo "Graphviz: version date is based on time of last commit: $GRAPHVIZ_VERSION_DATE"
-
-    GRAPHVIZ_VERSION_COMMIT=$( git log -n 1 --format=%h )
-    echo "Graphviz: abbreviated hash of last commit: $GRAPHVIZ_VERSION_COMMIT"
-fi
+GRAPHVIZ_VERSION_DATE=0
 
 # initialize version for a "stable" build
 cat >./version.m4 <<EOF
