From 55104fc641dc955ce0574c7b3ee67fc5ed249a94 Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Sun, 26 Nov 2023 06:16:36 -0500
Subject: [PATCH] point qmlplugindump to qmlplugindump-qt5

---
 modules/U1db/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules/U1db/CMakeLists.txt b/modules/U1db/CMakeLists.txt
index 05cc7c4..db4e97b 100644
--- a/modules/U1db/CMakeLists.txt
+++ b/modules/U1db/CMakeLists.txt
@@ -50,7 +50,7 @@ install(FILES qmldir
 add_custom_command(
     TARGET U1DBPlugin
     POST_BUILD
-    COMMAND "qmlplugindump" "U1db" "1.0" "${CMAKE_BINARY_DIR}/modules" ">" "plugins.qmltypes"
+    COMMAND "qmlplugindump-qt5" "U1db" "1.0" "${CMAKE_BINARY_DIR}/modules" ">" "plugins.qmltypes"
     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
     )
 
-- 
2.42.0

