replaces a space with U+202f
diff --git a/tests/timeTest.js b/tests/timeTest.js
index 96803f2..7f07081 100644
--- a/tests/timeTest.js
+++ b/tests/timeTest.js
@@ -35,7 +35,7 @@ function formatTimeWithTZOffsetTest() {
     // mock to always use 12 hour format
     Time._setIs12HourFunction(() => { return true; });
 
-    JsUnit.assertEquals('10:54 PM',
+    JsUnit.assertEquals('10:54 PM',
                         Time.formatTimeWithTZOffset(1607982864000, 3600000));
 }
 
@@ -50,9 +50,9 @@ function formatTimeFromHoursAndMinsTest() {
     // mock to always use 12 hour format
     Time._setIs12HourFunction(() => { return true; });
 
-    JsUnit.assertEquals('12:34 PM', Time.formatTimeFromHoursAndMins(12, 34));
-    JsUnit.assertEquals('12:00 AM', Time.formatTimeFromHoursAndMins(24, 0));
-    JsUnit.assertEquals('12:01 PM', Time.formatTimeFromHoursAndMins(12, 1));
+    JsUnit.assertEquals('12:34 PM', Time.formatTimeFromHoursAndMins(12, 34));
+    JsUnit.assertEquals('12:00 AM', Time.formatTimeFromHoursAndMins(24, 0));
+    JsUnit.assertEquals('12:01 PM', Time.formatTimeFromHoursAndMins(12, 1));
 }
 
 formatTimeWithTZOffsetTest();
