diff --git a/Makefile b/Makefile
index ccbf433..bd0026c 100644
--- a/Makefile
+++ b/Makefile
@@ -94,12 +94,12 @@ install-guake:
 	@sed -i -e 's|{{ LOGIN_DESTOP_PATH }}|"$(LOGIN_DESTOP_PATH)"|g' guake/paths.py
 	@sed -i -e 's|{{ AUTOSTART_FOLDER }}|"$(AUTOSTART_FOLDER)"|g' guake/paths.py
 
-	@$(PYTHON_INTERPRETER) -m pip install . --root "$(DESTDIR)" --prefix="/usr" || echo -e "\033[31;1msetup.py install failed, you may need to run \"sudo git config --global --add safe.directory '*'\"\033[0m"
+	@gpep517 build-wheel --wheel-dir dist --output-fd 3 3>&1 >&2
+	@python3 -m installer -d "$(DESTDIR)" dist/*.whl
 
 	@rm -f guake/paths.py
 	@if [ -f guake/paths.py.dev ]; then mv guake/paths.py.dev guake/paths.py; fi
 
-	@update-desktop-database || echo "Could not run update-desktop-database, are you root?"
 	@rm -rfv build *.egg-info
 
 install-locale:
