diff --git a/Makefile b/Makefile
index 41873fa..dab56e1 100644
--- a/Makefile
+++ b/Makefile
@@ -63,10 +63,10 @@ install: all
 	install -d $(DESTDIR)/etc
 	install -d $(DESTDIR)/lib/systemd/system
 	install -d $(DESTDIR)/usr/share/doc/mbpfan
-	install -d $(DESTDIR)/lib/modules-load.d
+	install -d $(DESTDIR)/usr/lib/modules-load.d
 	install $(BIN) $(DESTDIR)/usr/sbin
 	install -m644 $(CONF) $(DESTDIR)/etc
-	install -m644 $(DEPEND_MODULE) $(DESTDIR)/lib/modules-load.d
+	install -m644 $(DEPEND_MODULE) $(DESTDIR)/usr/lib/modules-load.d
 	install -m644 $(DOC) $(DESTDIR)/usr/share/doc/mbpfan
 	install -d $(DESTDIR)/usr/share/man/man8
 	install -m644 $(MAN) $(DESTDIR)/usr/share/man/man8
