--- a/terminal/terminal-dbus.c.orig	2007-01-23 22:25:12.000000000 +0000
+++ b/terminal/terminal-dbus.c	2007-01-23 22:25:50.000000000 +0000
@@ -191,6 +191,7 @@ terminal_dbus_register_service (Terminal
 
   /* register DBus connection with GLib main loop */
   dbus_connection_setup_with_g_main (connection, NULL);
+  dbus_connection_set_exit_on_disconnect (connection, FALSE);
 
   if (dbus_bus_request_name (connection, TERMINAL_DBUS_SERVICE, 0, &derror) < 0)
     {
