diff --git a/test/test_rospkg_os_detect.py b/test/test_rospkg_os_detect.py
index 5c2ba3f..ae930c1 100644
--- a/test/test_rospkg_os_detect.py
+++ b/test/test_rospkg_os_detect.py
@@ -398,7 +398,7 @@ def test_OsDetector():
 def test_tripwire_uname_get_machine():
     from rospkg.os_detect import uname_get_machine
     retval = uname_get_machine()
-    assert retval in [None, 'aarch64', 'arm64', 'armv7l', 'armv8l', 'i386', 'i686', 'ppc', 'ppc64', 'ppc64le', 'riscv32', 'riscv64', 's390', 's390x', 'x86_64']
+    assert retval in [None, 'aarch64', 'arm64', 'armv7l', 'armv8l', 'i386', 'i686', 'loongarch64', 'ppc', 'ppc64', 'ppc64le', 'riscv32', 'riscv64', 's390', 's390x', 'x86_64']
 
 
 def test_tripwire_rhel():
