--- old/src/mod_h264_streaming.c
+++ new/src/mod_h264_streaming.c
@@ -168,7 +168,7 @@
 
     p->config_storage[i] = s;
 
-    if (0 != config_insert_values_global(srv, ((data_config *)srv->config_context->data[i])->value, cv)) {
+    if (0 != config_insert_values_global(srv, ((data_config *)srv->config_context->data[i])->value, cv, i == 0 ? T_CONFIG_SCOPE_SERVER : T_CONFIG_SCOPE_CONNECTION)) {
       return HANDLER_ERROR;
     }
   }
