diff -Nurp a/src/dmclock/support/src/indirect_intrusive_heap.h b/src/dmclock/support/src/indirect_intrusive_heap.h
--- a/src/dmclock/support/src/indirect_intrusive_heap.h	2020-11-21 21:07:48.760968790 +0000
+++ b/src/dmclock/support/src/indirect_intrusive_heap.h	2020-11-21 21:49:24.492529708 +0000
@@ -25,6 +25,9 @@
 
 #include "assert.h"
 
+#ifndef uint
+  typedef unsigned uint;
+#endif
 
 namespace crimson {
   using IndIntruHeapData = size_t;
