diff -uNr ceph-15.2.4/src/librbd/image/OpenRequest.cc ceph-15.2.4-fix_librbd_duplicate/src/librbd/image/OpenRequest.cc
--- ceph-15.2.4/src/librbd/image/OpenRequest.cc	2020-06-30 15:40:51.000000000 +0000
+++ ceph-15.2.4-fix_librbd_duplicate/src/librbd/image/OpenRequest.cc	2020-09-30 07:58:38.979496776 +0000
@@ -9,7 +9,7 @@
 #include "librbd/Utils.h"
 #include "librbd/cache/ObjectCacherObjectDispatch.h"
 #include "librbd/cache/WriteAroundObjectDispatch.h"
-#include "librbd/cache/ParentCacheObjectDispatch.cc"
+#include "librbd/cache/ParentCacheObjectDispatch.h"
 #include "librbd/image/CloseRequest.h"
 #include "librbd/image/RefreshRequest.h"
 #include "librbd/image/SetSnapRequest.h"
