--- a/yabasic.h
+++ b/yabasic.h
@@ -570,7 +570,7 @@
 void dot (struct command *);
 void create_line (int);
 void line (struct command *);
-void moveorigin ();
+void moveorigin (char *);
 int check_alignement (char *);
 void circle (struct command *);
 void triangle (struct command *);
@@ -612,7 +612,7 @@
 void create_exception (int);
 void exception (struct command *);
 void create_poke (char);
-void poke ();
+void poke (struct command *);
 void pokefile (struct command *);
 void create_dblrelop (char);
 void dblrelop (struct command *);
