--- a/PLUGINS/src/softhddevice/audio.c
+++ b/PLUGINS/src/softhddevice/audio.c
@@ -48,6 +48,7 @@
 #include <inttypes.h>
 #include <string.h>
 #include <math.h>
+#include <sched.h>
 
 #include <libintl.h>
 #define _(str) gettext(str)		///< gettext shortcut
@@ -1584,7 +1585,7 @@
 	if (err < 0) {			// underrun error
 	    return -1;
 	}
-	pthread_yield();
+	sched_yield();
 	usleep(OssFragmentTime * 1000);	// let fill/empty the buffers
 	return 0;
     }
