--- a/cmake/pcl_utils.cmake
+++ b/cmake/pcl_utils.cmake
@@ -107,7 +107,7 @@
     if(WIN32 AND NOT MINGW)
         set(PCLCONFIG_INSTALL_DIR "cmake")
       else()
-        set(PCLCONFIG_INSTALL_DIR "share/${PROJECT_NAME_LOWER}-${PCL_VERSION_MAJOR}.${PCL_VERSION_MINOR}")
+        set(PCLCONFIG_INSTALL_DIR "lib/cmake/${PROJECT_NAME_LOWER}-${PCL_VERSION_MAJOR}.${PCL_VERSION_MINOR}")
       endif()
 endmacro()
 
