--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,12 @@
 .PHONY: all clean install uninstall debug man compose test-harness
 VERSION=2.5.0
-COMMIT=$(shell git describe --no-match --always --abbrev=7 --dirty)
 VKBD=uinput
 PREFIX?=/usr/local
 
 CONFIG_DIR?=/etc/keyd
 SOCKET_PATH=/var/run/keyd.socket
 
-CFLAGS:=-DVERSION=\"v$(VERSION)\ \($(COMMIT)\)\" \
-	-I/usr/local/include \
-	-L/usr/local/lib \
+CFLAGS:=-DVERSION=\"v$(VERSION)\" \
 	-Wall \
 	-Wextra \
 	-Wno-unused \
@@ -36,7 +33,7 @@
 	cp scripts/keyd-application-mapper bin/
 	sed -e 's#@PREFIX@#$(PREFIX)#' keyd.service.in > keyd.service
 	sed -e 's#@PREFIX@#$(PREFIX)#' src/vkbd/usb-gadget.service.in > src/vkbd/usb-gadget.service
-	$(CC) $(CFLAGS) -O3 $(COMPAT_FILES) src/*.c src/vkbd/$(VKBD).c -lpthread -o bin/keyd $(LDFLAGS)
+	$(CC) $(CFLAGS) $(COMPAT_FILES) src/*.c src/vkbd/$(VKBD).c -lpthread -o bin/keyd $(LDFLAGS)
 debug:
 	CFLAGS="-g -fsanitize=address -Wunused" $(MAKE)
 compose:
