--- a/lib/Util.ml
+++ b/lib/Util.ml
@@ -200,7 +200,7 @@ let run_command (command: string): command_output =
     }
 
 let compile_c_code (source_path: string) (output_path: string): command_output =
-  let cmd = "cc " ^ source_path ^ " -fwrapv -lm -o " ^ output_path in
+  let cmd = "cc " ^ source_path ^ " -Wno-incompatible-pointer-types -fwrapv -lm -o " ^ output_path in
   let o = run_command cmd in
   let (CommandOutput { command; code; stdout; stderr }) = o in
   if code <> 0 then
