From 88ba23f91929dc0d797394515fb1f8f6b3dc0ad6 Mon Sep 17 00:00:00 2001
From: Peter Pettersson <boretrk@hotmail.com>
Date: Thu, 14 Jul 2022 18:28:58 +0200
Subject: [PATCH] clar: call ftruncate from util_tests instead of libgit2_tests

---
 tests/libgit2/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/libgit2/CMakeLists.txt b/tests/libgit2/CMakeLists.txt
index 27f421ad629..00445f4e3e7 100644
--- a/tests/libgit2/CMakeLists.txt
+++ b/tests/libgit2/CMakeLists.txt
@@ -64,8 +64,9 @@ if(MSVC_IDE)
 endif()
 
 include(AddClarTest)
+add_clar_test(util_tests    util                -v -sftruncate)
 add_clar_test(libgit2_tests offline             -v -xonline)
-add_clar_test(libgit2_tests invasive            -v -score::ftruncate -sfilter::stream::bigfile -sodb::largefiles -siterator::workdir::filesystem_gunk -srepo::init -srepo::init::at_filesystem_root)
+add_clar_test(libgit2_tests invasive            -v -sfilter::stream::bigfile -sodb::largefiles -siterator::workdir::filesystem_gunk -srepo::init -srepo::init::at_filesystem_root)
 add_clar_test(libgit2_tests online              -v -sonline -xonline::customcert)
 add_clar_test(libgit2_tests online_customcert   -v -sonline::customcert)
 add_clar_test(libgit2_tests gitdaemon           -v -sonline::push)
