From 2fa13adf09c8dd1f334f57023390043ea3f71cb2 Mon Sep 17 00:00:00 2001
From: Florian Pircher <florian@formkunft.com>
Date: Thu, 2 Jan 2025 17:19:58 +0100
Subject: [PATCH] include: Fix code comment termination

---
 include/git2/stdint.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/git2/stdint.h b/include/git2/stdint.h
index 4b235c73f87..4f532e13c85 100644
--- a/include/git2/stdint.h
+++ b/include/git2/stdint.h
@@ -221,7 +221,7 @@ typedef uint64_t  uintmax_t;
 #endif /* __STDC_LIMIT_MACROS ] */
 
 
-/* 7.18.4 Limits of other integer types
+/* 7.18.4 Limits of other integer types */
 
 #if !defined(__cplusplus) || defined(__STDC_CONSTANT_MACROS) /* [    See footnote 224 at page 260 */
 
