diff --git a/tests/libgit2/CMakeLists.txt b/tests/libgit2/CMakeLists.txt
index 00445f4..7be8f0e 100644
--- a/tests/libgit2/CMakeLists.txt
+++ b/tests/libgit2/CMakeLists.txt
@@ -66,11 +66,11 @@ endif()
 include(AddClarTest)
 add_clar_test(util_tests    util                -v -sftruncate)
 add_clar_test(libgit2_tests offline             -v -xonline)
-add_clar_test(libgit2_tests invasive            -v -sfilter::stream::bigfile -sodb::largefiles -siterator::workdir::filesystem_gunk -srepo::init -srepo::init::at_filesystem_root)
-add_clar_test(libgit2_tests online              -v -sonline -xonline::customcert)
-add_clar_test(libgit2_tests online_customcert   -v -sonline::customcert)
-add_clar_test(libgit2_tests gitdaemon           -v -sonline::push)
-add_clar_test(libgit2_tests ssh                 -v -sonline::push -sonline::clone::ssh_cert -sonline::clone::ssh_with_paths -sonline::clone::path_whitespace_ssh)
-add_clar_test(libgit2_tests proxy               -v -sonline::clone::proxy)
-add_clar_test(libgit2_tests auth_clone          -v -sonline::clone::cred)
-add_clar_test(libgit2_tests auth_clone_and_push -v -sonline::clone::push -sonline::push)
+# add_clar_test(libgit2_tests invasive            -v -sfilter::stream::bigfile -sodb::largefiles -siterator::workdir::filesystem_gunk -srepo::init -srepo::init::at_filesystem_root)
+# add_clar_test(libgit2_tests online              -v -sonline -xonline::customcert)
+# add_clar_test(libgit2_tests online_customcert   -v -sonline::customcert)
+# add_clar_test(libgit2_tests gitdaemon           -v -sonline::push)
+# add_clar_test(libgit2_tests ssh                 -v -sonline::push -sonline::clone::ssh_cert -sonline::clone::ssh_with_paths -sonline::clone::path_whitespace_ssh)
+# add_clar_test(libgit2_tests proxy               -v -sonline::clone::proxy)
+# add_clar_test(libgit2_tests auth_clone          -v -sonline::clone::cred)
+# add_clar_test(libgit2_tests auth_clone_and_push -v -sonline::clone::push -sonline::push)
