From 91571cce03adc58a10b5d76307629efc7c994c3e Mon Sep 17 00:00:00 2001
From: Rob Clark <robdclark@gmail.com>
Date: Tue, 26 Aug 2014 17:31:26 -0400
Subject: modesetting: Don't pretend to support rotation

It only ends in tears

Signed-off-by: Rob Clark <robdclark@gmail.com>
Reviewed-by: Alex Deucher <alexander.deucher@amd.com>

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index c533324..12aa987 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -568,17 +568,6 @@ drmmode_set_scanout_pixmap(xf86CrtcPtr crtc, PixmapPtr ppix)
 }
 #endif
 
-static void *drmmode_shadow_allocate(xf86CrtcPtr crtc, int width, int height)
-{
-	return NULL;
-}
-
-static PixmapPtr drmmode_shadow_create(xf86CrtcPtr crtc, void *data, int width,
-				       int height)
-{
-	return NULL;
-}
-
 static const xf86CrtcFuncsRec drmmode_crtc_funcs = {
     .dpms = drmmode_crtc_dpms,
     .set_mode_major = drmmode_set_mode_major,
@@ -593,8 +582,6 @@ static const xf86CrtcFuncsRec drmmode_crtc_funcs = {
 #ifdef MODESETTING_OUTPUT_SLAVE_SUPPORT
     .set_scanout_pixmap = drmmode_set_scanout_pixmap,
 #endif
-    .shadow_allocate = drmmode_shadow_allocate,
-    .shadow_create = drmmode_shadow_create,
 };
 
 static void
-- 
cgit v0.10.2

