--- a/drbdmanage/server.py
+++ b/drbdmanage/server.py
@@ -165,7 +165,7 @@
         KEY_MIN_PORT_NR    : str(DEFAULT_MIN_PORT_NR),
         KEY_MAX_PORT_NR    : str(DEFAULT_MAX_PORT_NR),
         KEY_MAX_FAIL_COUNT : str(DEFAULT_MAX_FAIL_COUNT),
-        KEY_DRBDADM_PATH   : "/usr/sbin",
+        KEY_DRBDADM_PATH   : "/sbin",
         KEY_EXTEND_PATH    : "/sbin:/usr/sbin:/bin:/usr/bin",
         KEY_DRBD_CONFPATH  : "/var/lib/drbd.d",
         KEY_DRBDCTRL_VG    : DEFAULT_VG,
