From 0293978345d9b9188da36e48aa5fa419f81a57ac Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Mon, 18 Apr 2022 10:35:52 +0200
Subject: [PATCH] fix x86

---
 src/lib/Iex/IexMathFpu.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lib/Iex/IexMathFpu.cpp b/src/lib/Iex/IexMathFpu.cpp
index b73e98f..2482c36 100644
--- a/src/lib/Iex/IexMathFpu.cpp
+++ b/src/lib/Iex/IexMathFpu.cpp
@@ -251,14 +251,14 @@ restoreControlRegs (const ucontext_t & ucon, bool clearExceptions)
 inline void
 restoreControlRegs (const ucontext_t & ucon, bool clearExceptions)
 {
-#        if defined(__GLIBC__) && defined(__i386__)
+#        if defined(__i386__)
     setCw ((ucon.uc_mcontext.fpregs->cw & cwRestoreMask) | cwRestoreVal);
 #else
     setCw ((ucon.uc_mcontext.fpregs->cwd & cwRestoreMask) | cwRestoreVal);
 #        endif
 
     _fpstate* kfp = reinterpret_cast<_fpstate*> (ucon.uc_mcontext.fpregs);
-#        if defined(__GLIBC__) && defined(__i386__)
+#        if defined(__i386__)
     setMxcsr (kfp->magic == 0 ? kfp->mxcsr : 0, clearExceptions);
 #else
     setMxcsr (kfp->mxcsr, clearExceptions);
-- 
2.35.2

