diff --git a/libpolyml/polyffi.cpp b/libpolyml/polyffi.cpp
index 5424dd8..a11987f 100644
--- a/libpolyml/polyffi.cpp
+++ b/libpolyml/polyffi.cpp
@@ -110,9 +110,10 @@ static struct _abiTable { const char *abiName; ffi_abi abiCode; } abiTable[] =
     {"ms_cdecl", FFI_MS_CDECL},
 #elif defined(X86_WIN64)
     {"win64", FFI_WIN64},
+#elif defined(X86_64)
+    {"unix64", FFI_UNIX64},
 #elif defined(X86_ANY)
     {"sysv", FFI_SYSV},
-    {"unix64", FFI_UNIX64},
 #endif
     { "default", FFI_DEFAULT_ABI}
 };
