From 4cfcd2d4ddaf603d2e7fee3b61af4f69cf671965 Mon Sep 17 00:00:00 2001
From: Willow Barraco <contact@willowbarraco.fr>
Date: Tue, 23 Jun 2026 14:21:51 +0200
Subject: [PATCH] tests: skip dbus tests which fails with missing machine-id

Signed-off-by: Willow Barraco <contact@willowbarraco.fr>
---
 testsuite/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/testsuite/meson.build b/testsuite/meson.build
index 2fc2aeb..4b1b9f1 100644
--- a/testsuite/meson.build
+++ b/testsuite/meson.build
@@ -28,10 +28,12 @@ if have_gdbus_codegen
   )
 endif
 
+have_machine_id = import('fs').is_file('/etc/machine-id')
+
 testsuite = {
   'test-async-result': {},
   'test-channel': {},
-  'test-dbus': {'extra-sources': dbus_foo, 'disable': not have_gdbus_codegen},
+  'test-dbus': {'extra-sources': dbus_foo, 'disable': not have_gdbus_codegen or not have_machine_id},
   'test-object': {},
   'test-fiber': {},
   'test-future': {},
-- 
2.54.0

