diff --git a/desmume/src/commandline.cpp b/desmume/src/commandline.cpp
index cb6c23a..dfaf564 100644
--- a/desmume/src/commandline.cpp
+++ b/desmume/src/commandline.cpp
@@ -397,7 +397,7 @@ bool CommandLine::parse(int argc,char **argv)
 
 	if(opt_help)
 	{
-		printf(help_string);
+		printf("%s", help_string);
 		exit(1);
 	}
 
@@ -571,7 +571,7 @@ bool CommandLine::validate()
 
 void CommandLine::errorHelp(const char* binName)
 {
-	printerror(help_string);
+	printerror("%s", help_string);
 }
 
 void CommandLine::process_movieCommands()
