diff --git a/x2goserver-xsession/etc/Xsession b/x2goserver-xsession/etc/Xsession
index ebee0c5..a63cdd0 100755
--- a/x2goserver-xsession/etc/Xsession
+++ b/x2goserver-xsession/etc/Xsession
@@ -145,7 +145,7 @@ if [ -f /etc/debian_version ] || [ -f /etc/devuan_version ]; then
 		done
 		set -e
 	fi
-elif [ -f /etc/redhat-release ] || [ -f /etc/gentoo-release ] || [ -f /etc/SUSE-brand ] || [ -f /etc/SuSE-release ] || [ -f /etc/os-rt-release ]; then
+elif [ -f /etc/alpine-release ] || [ -f /etc/redhat-release ] || [ -f /etc/gentoo-release ] || [ -f /etc/SUSE-brand ] || [ -f /etc/SuSE-release ] || [ -f /etc/os-rt-release ]; then
 
 	# define a fallback... (should never be needed). The XSESSION_EXEC var gets set in
 	# X2Go's x2goruncommand script and can be used with obsolete switchdesk or with
