diff --git a/jsonpickle/pickler.py b/jsonpickle/pickler.py
index ce64592..62ba575 100644
--- a/jsonpickle/pickler.py
+++ b/jsonpickle/pickler.py
@@ -462,6 +462,9 @@ class Pickler(object):
             if has_getinitargs:
                 data[tags.INITARGS] = self._flatten(obj.__getinitargs__())
 
+        if type(obj).__name__ == 'TextIOWrapper':
+            return None
+
         if has_getstate:
             try:
                 state = obj.__getstate__()

