We don't need systemd files
diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-03-06 20:12:27.339495351 +0100
+++ b/CMakeLists.txt	2024-03-06 20:16:05.869420274 +0100
@@ -41,7 +41,5 @@
 install(FILES data/api.conf DESTINATION ${NGINXDIR})
 
 install(FILES data/autocomplete.sh DESTINATION ${CMAKE_INSTALL_FULL_DATADIR}/bash-completion/completions RENAME grommunio-admin)
-install(FILES data/grommunio-admin-api.service data/grommunio-admin-api.socket DESTINATION ${UNITDIR})
 install(FILES data/portrait.jpg DESTINATION ${COMMONDIR})
-install(FILES data/tmpfiles-grommunio-admin-api.conf DESTINATION ${TMPFILESDIR} RENAME grommunio-admin-api.conf)
 install(FILES data/grommunio-sudo DESTINATION ${CMAKE_INSTALL_FULL_SYSCONFDIR}/sudoers.d)
