# openapi_core 0.19 has the required fixes for python 3.12
--- a/api/core.py
+++ b/api/core.py
@@ -22,9 +22,9 @@
             from openapi_core.contrib.flask import FlaskOpenAPIRequest, FlaskOpenAPIResponse
         if self.version < [0, 15, 0]:
             from openapi_core import create_spec
-        elif self.version < [0, 18, 0]:
+        elif self.version < [0, 19, 0]:
             from openapi_core.spec.shortcuts import create_spec
-        if self.version < [0, 18, 0]:
+        if self.version < [0, 19, 0]:
             self.spec = create_spec(apiSpec)
         else:
             self.spec = openapi_core.Spec.from_dict(apiSpec)
