Add alternative command name "exa" for backward compatibility.

--- a/completions/bash/eza
+++ b/completions/bash/eza
@@ -58,4 +58,4 @@
             ;;
     esac
 } &&
-complete -o filenames -o bashdefault -F _eza eza
+complete -o filenames -o bashdefault -F _eza eza exa
--- a/completions/zsh/_eza
+++ b/completions/zsh/_eza
@@ -1,4 +1,4 @@
-#compdef eza
+#compdef eza exa
 
 # Save this file as _eza in /usr/local/share/zsh/site-functions or in any
 # other folder in $fpath.  E.g. save it in a folder called ~/.zfunc and add a
