--- a/setup.py
+++ b/setup.py
@@ -32,5 +32,4 @@ setup(
     classifiers=classifiers,
     packages=["shortuuid"],
     test_suite='shortuuid.tests',
-    tests_require=['pep8'],
 )
Index: shortuuid-0.5.0/shortuuid/tests.py
===================================================================
--- a/shortuuid/tests.py
+++ b/shortuuid/tests.py
@@ -2,7 +2,6 @@ import os
 import string
 import sys
 import unittest
-import pep8
 from collections import defaultdict
 
 from uuid import UUID, uuid4
@@ -125,16 +124,6 @@ class ClassShortUUIDTest(unittest.TestCa
         su4 = ShortUUID()
         self.assertEqual(su4.encoded_length(num_bytes=8), 11)
 
-    def test_pep8(self):
-        pep8style = pep8.StyleGuide([['statistics', True],
-                                     ['show-sources', True],
-                                     ['repeat', True],
-                                     ['paths', [os.path.dirname(
-                                         os.path.abspath(__file__))]]],
-                                    parse_argv=False)
-        report = pep8style.check_files()
-        assert report.total_errors == 0
-
 
 class ShortUUIDPaddingTest(unittest.TestCase):
     def test_padding(self):

