Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: run tests non-interactively
----

--- a/gl2psTest.c
+++ b/gl2psTest.c
@@ -60,8 +60,8 @@

 static const char *format_string = "Postscript";
 static float rotation = -58.;
-static GLsizei window_w = 0;
-static GLsizei window_h = 0;
+static GLsizei window_w = 800;
+static GLsizei window_h = 800;
 static GLboolean display_multi = GL_TRUE;
 static GLboolean blend = GL_FALSE;
 static GLboolean teapot = GL_FALSE;
@@ -629,6 +629,12 @@
   glutReshapeFunc(reshape);
   glutKeyboardFunc(keyboard);
   glutMotionFunc(motion);
-  glutMainLoop();
+  keyboard('s', 0, 0);
+  keyboard('p', 0, 0);
+  keyboard('s', 0, 0);
+  keyboard('p', 0, 0);
+  keyboard('s', 0, 0);
+  keyboard('p', 0, 0);
+  keyboard('s', 0, 0);
   return 0;
 }

--- sa/gl2psTestSimple.c
+++ b/gl2psTestSimple.c
@@ -202,6 +202,6 @@
 
   glutDisplayFunc(display);
   glutKeyboardFunc(keyboard);
-  glutMainLoop();
+  keyboard('s', 0, 0);
   return 0;
 }
