--- a/configure.ac
+++ b/configure.ac
@@ -67,24 +67,24 @@
 AC_FUNC_STRTOD
 AC_CHECK_FUNCS([__argz_count __argz_next __argz_stringify dcgettext getcwd getwd mempcpy memset munmap nl_langinfo setlocale stpcpy strcasecmp strchr strcspn strdup strstr strtoul])

-# check for libncursesw
+# check for libncurses
 case $host_os in
   *bsd* | darwin*)
     AC_DEFINE([OS_BSD], [1], [OS BSD's curses includes ncursesw])
     # check for courses.h
     AC_CHECK_HEADER(curses.h, [], [AC_MSG_ERROR(
       [curses.h header is missing: See INSTALL file for further information])])
-    # check for libncursesw
+    # check for libncurses
     AC_CHECK_LIB(ncursesw, add_wch, [], [AC_MSG_ERROR(
       [ncursesw library is missing: See INSTALL file for further information])])
     ;;
   *)
-    AC_CHECK_HEADER(ncursesw/ncurses.h, HAVE_NCURSESW_H=1)
-    AC_CHECK_LIB(ncursesw, add_wch, HAVE_LIBNCURSESW=1)
+    AC_CHECK_HEADER(ncurses.h, HAVE_NCURSESW_H=1)
+    AC_CHECK_LIB(ncurses, add_wch, HAVE_LIBNCURSESW=1)
     if test -n "$HAVE_NCURSESW_H" -a -n "$HAVE_LIBNCURSESW";  then
-       LIBS="-lncursesw $LIBS"
+       LIBS="-lncurses $LIBS"
     else
-       echo -e "Error:  both library and header files for the ncursesw library\n"\
+       echo -e "Error:  both library and header files for the ncurses library\n"\
            "are required to build this package.  See INSTALL file for"\
            "further information. On Debian/Ubuntu you need to install libncursesw5-dev."
        exit 1;
@@ -93,7 +93,7 @@
 esac

 AC_SEARCH_LIBS(cbreak, tinfo, [],
-  [AC_MSG_ERROR([Can't find cbreak() in -lncursesw or -ltinfo])])
+  [AC_MSG_ERROR([Can't find cbreak() in -lncurses or -ltinfo])])


 # iconv
diff -wbBur gtypist-2.9.4/src/cursmenu.c gtypist-2.9.4.my/src/cursmenu.c
--- gtypist-2.9.4/src/cursmenu.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/cursmenu.c	2014-02-03 13:09:04.669316878 +0400
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include "error.h"
diff -wbBur gtypist-2.9.4/src/error.c gtypist-2.9.4.my/src/error.c
--- gtypist-2.9.4/src/error.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/error.c	2014-02-03 13:09:04.669316878 +0400
@@ -25,7 +25,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include <stdlib.h>
diff -wbBur gtypist-2.9.4/src/gtypist.c gtypist-2.9.4.my/src/gtypist.c
--- gtypist-2.9.4/src/gtypist.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/gtypist.c	2014-02-03 13:09:04.669316878 +0400
@@ -31,7 +31,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include <time.h>
diff -wbBur gtypist-2.9.4/src/script.c gtypist-2.9.4.my/src/script.c
--- gtypist-2.9.4/src/script.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/script.c	2014-02-03 13:09:04.669316878 +0400
@@ -24,7 +24,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include "error.h"
diff -wbBur gtypist-2.9.4/src/utf8.c gtypist-2.9.4.my/src/utf8.c
--- gtypist-2.9.4/src/utf8.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/utf8.c	2014-02-03 13:09:04.669316878 +0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include <stdlib.h>
diff -wbBur gtypist-2.9.4/src/infoview.c gtypist-2.9.4.my/src/infoview.c
--- gtypist-2.9.4/src/infoview.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/infoview.c	2014-02-03 13:09:04.669316878 +0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include "infoview.h"
diff -wbBur gtypist-2.9.4/src/speedbox.c gtypist-2.9.4.my/src/speedbox.c
--- gtypist-2.9.4/src/speedbox.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/speedbox.c	2014-02-03 13:09:04.669316878 +0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include "speedbox.h"
diff -wbBur gtypist-2.9.4/src/banner.c gtypist-2.9.4.my/src/banner.c
--- gtypist-2.9.4/src/banner.c	2013-08-18 18:36:14.000000000 +0400
+++ gtypist-2.9.4.my/src/banner.c	2014-02-03 13:09:04.669316878 +0400
@@ -23,7 +23,7 @@
 #ifdef HAVE_PDCURSES
 #include <curses.h>
 #else
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #endif

 #include "banner.h"

