diff --git a/src/mount_image.c b/src/mount_image.c
index 3852351..202c562 100644
--- a/src/mount_image.c
+++ b/src/mount_image.c
@@ -1866,6 +1866,11 @@ wimfs_removexattr(const char *path, const char *name)
 	return 0;
 }
 
+
+/* taken from glibc */
+# define RENAME_NOREPLACE (1 << 0)
+# define RENAME_EXCHANGE (1 << 1)
+
 static int
 wimfs_rename(const char *from, const char *to, unsigned int flags)
 {
