make this opt-in
--
diff --git a/lib/config/defaults.h b/lib/config/defaults.h
index 1ff3637..9c2711b 100644
--- a/lib/config/defaults.h
+++ b/lib/config/defaults.h
@@ -199,7 +199,7 @@
 #define DEFAULT_LABELSECTOR UINT64_C(1)
 #define DEFAULT_READ_AHEAD "auto"
 #define DEFAULT_UDEV_RULES 1
-#define DEFAULT_UDEV_SYNC 1
+#define DEFAULT_UDEV_SYNC 0
 #define DEFAULT_NOTIFY_DBUS 1
 #define DEFAULT_VERIFY_UDEV_OPERATIONS 0
 #define DEFAULT_RETRY_DEACTIVATION 1
