See https://gitlab.alpinelinux.org/alpine/aports/-/issues/16103

diff -upr LVM2.2.03.23.orig/lib/device/device_id.c LVM2.2.03.23/lib/device/device_id.c
--- LVM2.2.03.23.orig/lib/device/device_id.c	2024-05-12 12:05:45.824137483 +0200
+++ LVM2.2.03.23/lib/device/device_id.c	2024-05-12 12:06:00.537497883 +0200
@@ -30,6 +30,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <time.h>
+#include <libgen.h>
 #include <sys/types.h>
 #include <sys/file.h>
 #include <sys/sysmacros.h>
