From eac43c6bbfff6875d41c46415fb7ada7b33880cb Mon Sep 17 00:00:00 2001
From: "Alex Xu (Hello71)" <alex_y_xu@yahoo.ca>
Date: Fri, 20 Aug 2021 09:04:04 -0400
Subject: [PATCH] add missing includes

---
 lib/device/dev-cache.c | 1 +
 tools/lvmdevices.c     | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/lib/device/dev-cache.c b/lib/device/dev-cache.c
index d7fa93fd6..d54a63fbe 100644
--- a/lib/device/dev-cache.c
+++ b/lib/device/dev-cache.c
@@ -29,6 +29,7 @@
 #include <unistd.h>
 #include <dirent.h>
 #include <locale.h>
+#include <sys/file.h>
 
 struct dev_iter {
 	struct btree_iter *current;
diff --git a/tools/lvmdevices.c b/tools/lvmdevices.c
index 3448bdd14..aa793f2e4 100644
--- a/tools/lvmdevices.c
+++ b/tools/lvmdevices.c
@@ -16,6 +16,8 @@
 #include "lib/cache/lvmcache.h"
 #include "lib/device/device_id.h"
 
+#include <sys/file.h>
+
 static void _search_devs_for_pvids(struct cmd_context *cmd, struct dm_list *search_pvids, struct dm_list *found_devs)
 {
 	struct dev_iter *iter;
-- 
2.33.0

