diff -ur old/Makefile dev/Makefile
--- old/Makefile	Sat Oct 14 17:49:55 2006
+++ dev/Makefile	Sat Oct 14 17:50:18 2006
@@ -1,4 +1,3 @@
-MAKE    = make
 SUBDIRS = bin include src
 
 # User defines:
@@ -43,10 +42,10 @@
 #Menu configuration system
 
 extra/config/conf:
-	make -C extra/config conf
+	$(MAKE) -C extra/config conf
 
 extra/config/mconf:
-	make -C extra/config ncurses mconf
+	$(MAKE) -C extra/config ncurses mconf
 
 menuconfig: extra/config/mconf
 	@./extra/config/mconf extra/Configs/Config.in
@@ -71,7 +70,7 @@
 
 include/system_configuration.h: .config
 	@if [ ! -x ./extra/config/conf ] ; then \
-		make -C extra/config conf; \
+		$(MAKE) -C extra/config conf; \
 	fi;
 	@./extra/config/conf -o extra/Configs/Config.in
 
