--- ./xnbd_server.c.orig
+++ ./xnbd_server.c
@@ -181,7 +181,7 @@
 
 static void signal_handler(int signum)
 {
-	dbg("sig: signal catched, code %d (%s)", signum, sys_siglist[signum]);
+	dbg("sig: signal catched, code %d (%s)", signum, strsignal(signum));
 
 	if (signum == SIGCHLD)
 		got_sigchld = 1;
@@ -495,7 +495,7 @@
 					info("   with exit status=%d", WEXITSTATUS(status));
 
 				if (WIFSIGNALED(status))
-					info("   killed by signal=%d(%s)", WTERMSIG(status), sys_siglist[WTERMSIG(status)]);
+					info("   killed by signal=%d(%s)", WTERMSIG(status), strsignal(WTERMSIG(status)));
 			}
 
 			const bool single_client_at_most = (connect_fd != -1);
--- ./xnbd_watchdog.c.orig
+++ ./xnbd_watchdog.c
@@ -26,10 +26,10 @@
 
 static void nbddev_watchdog_sigalarm_handler(int signum)
 {
-	info("sig: signal catched, code %d (%s)", signum, sys_siglist[signum]);
+	info("sig: signal catched, code %d (%s)", signum, strsignal(signum));
 
 	if (signum != SIGALRM)
-		warn("unexpected signal, code %d (%s)", signum, sys_siglist[signum]);
+		warn("unexpected signal, code %d (%s)", signum, strsignal(signum));
 }
 
 /* string is dummy */
