diff --git a/setup.py b/setup.py
index d6e0461..01eda33 100644
--- a/setup.py
+++ b/setup.py
@@ -55,7 +55,7 @@ with codecs.open(os.path.join(os.path.abspath(os.path.dirname(
 
 
 install_requires = ['chardet', 'multidict>=2.1.4',
-                    'async_timeout>=1.1.0', 'yarl>=0.9.8,<0.10']
+                    'async_timeout>=1.1.0', 'yarl>=0.9.8']
 
 if sys.version_info < (3, 4, 2):
     raise RuntimeError("aiohttp requires Python 3.4.2+")
