--- a/src/libs/xpcom18a4/nsprpub/pr/src/misc/prnetdb.c
+++ b/src/libs/xpcom18a4/nsprpub/pr/src/misc/prnetdb.c
@@ -93,9 +93,7 @@
 #define _PR_HAVE_GETPROTO_R_POINTER
 #endif
 
-#if defined(SOLARIS) || (defined(BSDI) && defined(_REENTRANT)) \
-	|| (defined(LINUX) && defined(_REENTRANT) \
-        && !(defined(__GLIBC__) && __GLIBC__ >= 2))
+#if defined(SOLARIS) || (defined(BSDI) && defined(_REENTRANT))
 #define _PR_HAVE_GETPROTO_R
 #define _PR_HAVE_GETPROTO_R_POINTER
 #endif
