--- a/src/VBox/Main/Makefile.kmk
+++ b/src/VBox/Main/Makefile.kmk
@@ -1292,10 +1292,13 @@
  VBoxMain-hardening-inst_INST = $(INST_BIN)components/
  VBoxMain-hardening-inst_SYMLINKS = \
 	VBoxDDU.so=>../VBoxDDU.so \
-	VBoxREM.so=>../VBoxREM.so \
 	VBoxRT.so=>../VBoxRT.so \
 	VBoxVMM.so=>../VBoxVMM.so \
 	VBoxXPCOM.so=>../VBoxXPCOM.so
+	ifdef VBOX_WITH_REM
+	VBoxMain-hardening-inst_SYMLINKS += \
+		VBoxREM.so=>../VBoxREM.so
+	endif
 endif
 
 