--- a/configure
+++ b/configure
@@ -755,15 +755,15 @@
 {
   test_header libIDL
 
-  if which_wrapper libIDL-config-2 > /dev/null; then
-    libidl_ver=`libIDL-config-2 --version`
+  if which_wrapper libidl/bin/libIDL-config-2 > /dev/null; then
+    libidl_ver=`libidl/bin/libIDL-config-2 --version`
     if [ $? -ne 0 ]; then
       log_failure "libIDL-config-2 not working"
       fail
     else
       log_success "found version $libidl_ver"
       cnf_append "VBOX_LIBIDL_CONFIG" \
-        "PKG_CONFIG_PATH=`libIDL-config-2 --prefix`/$LIB/pkgconfig `which_wrapper libIDL-config-2`"
+        "PKG_CONFIG_PATH=`libidl/bin/libIDL-config-2 --prefix`/$LIB/pkgconfig `which_wrapper libidl/bin/libIDL-config-2`"
     fi
   elif check_avail "libIDL-config" libIDL-config; then
     libidl_ver=`libIDL-config --version`

--- a/src/VBox/Installer/linux/VBox.sh
+++ b/src/VBox/Installer/linux/VBox.sh
@@ -19,7 +19,7 @@
 
 # The below is GNU-specific.  See slightly further down for a version which
 # works on Solaris and OS X.
-TARGET=`readlink -e -- "${0}"` || exit 1
+TARGET=`readlink -- "${0}"` || exit 1
 MY_DIR="${TARGET%/[!/]*}"
 
 # (
@@ -86,7 +86,7 @@
 EOF
 fi
 
-SERVER_PID=`ps -U \`whoami\` | grep VBoxSVC | awk '{ print $1 }'`
+SERVER_PID=`ps \`whoami\` | grep VBoxSVC | awk '{ print $1 }'`
 if [ -z "$SERVER_PID" ]; then
     # Server not running yet/anymore, cleanup socket path.
     # See IPC_GetDefaultSocketPath()!
