--- a/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceLibCWrappers.h
+++ b/src/VBox/ExtPacks/VBoxDTrace/include/VBoxDTraceLibCWrappers.h
@@ -31,11 +31,6 @@
 #else
 # include <sys/types.h>
 # include <limits.h>        /* Workaround for syslimit.h bug in gcc 4.8.3 on gentoo. */
-# ifdef RT_OS_DARWIN
-#  include <sys/syslimits.h> /* PATH_MAX */
-# elif !defined(RT_OS_SOLARIS) && !defined(RT_OS_FREEBSD)
-#  include <syslimits.h>    /* PATH_MAX */
-# endif
 # include <libgen.h>        /* basename */
 # include <unistd.h>
 # include <strings.h>       /* bzero & bcopy.*/
--- a/src/VBox/HostDrivers/adpctl/VBoxNetAdpCtl.cpp
+++ b/src/VBox/HostDrivers/adpctl/VBoxNetAdpCtl.cpp
@@ -37,9 +37,9 @@
 # include <linux/types.h>
 /* Older versions of ethtool.h rely on these: */
 typedef unsigned long long u64;
-typedef __uint32_t u32;
-typedef __uint16_t u16;
-typedef __uint8_t u8;
+typedef uint32_t u32;
+typedef uint16_t u16;
+typedef uint8_t u8;
 # include <limits.h> /* for INT_MAX */
 # include <linux/ethtool.h>
 # include <linux/sockios.h>
@@ -571,7 +571,7 @@
     snprintf(IfReq.ifr_name, sizeof(IfReq.ifr_name), "%s", pszName);
 
     EthToolVal.cmd = ETHTOOL_GLINK;
-    IfReq.ifr_data = (caddr_t)&EthToolVal;
+    IfReq.ifr_data = (char*)&EthToolVal;
     int rc = ioctl(fd, SIOCETHTOOL, &IfReq);
     if (rc == 0)
     {
@@ -580,7 +580,7 @@
             memset(&IfReq, 0, sizeof(IfReq));
             snprintf(IfReq.ifr_name, sizeof(IfReq.ifr_name), "%s", pszName);
             EthToolReq.cmd = ETHTOOL_GSET;
-            IfReq.ifr_data = (caddr_t)&EthToolReq;
+            IfReq.ifr_data = (char*)&EthToolReq;
             rc = ioctl(fd, SIOCETHTOOL, &IfReq);
             if (rc == 0)
             {
--- a/src/VBox/Main/src-server/linux/HostDnsServiceLinux.cpp
+++ b/src/VBox/Main/src-server/linux/HostDnsServiceLinux.cpp
@@ -84,7 +84,6 @@
 struct InotifyEventWithName
 {
     struct inotify_event e;
-    char name[NAME_MAX];
 };
 
 HostDnsServiceLinux::~HostDnsServiceLinux()
--- a/src/VBox/Runtime/include/internal/fs.h
+++ b/src/VBox/Runtime/include/internal/fs.h
@@ -62,8 +62,6 @@
 #ifdef RT_OS_LINUX
 # ifdef __USE_MISC
 #  define HAVE_STAT_TIMESPEC_BRIEF
-# else
-#  define HAVE_STAT_NSEC
 # endif
 #endif