--- a/src/VBox/RDP/client-1.8.4/Makefile.kmk
+++ b/src/VBox/RDP/client-1.8.4/Makefile.kmk
@@ -52,7 +52,7 @@ rdesktop-vrdp_DEFS.linux = \
 	ICONV_CONST= RDPSND_OSS=1 WITH_RDPUSB=1 HAVE_SYS_STATFS_H=1
 # Solaris 10 doesn't have dirfd(), and as thus we go for the low tech solution.
 rdesktop-vrdp_DEFS.linux += \
-	HAVE_DIRFD=1 HAVE_DECL_DIRFD=1 VBOX_USB_WITH_SYSFS=1
+	HAVE_DIRFD=1 HAVE_DECL_DIRFD=1 VBOX_USB_WITH_SYSFS=1 _BSD_SOURCE
 ifdef VBOX_WITH_LINUX_COMPILER_H
  rdesktop-vrdp_DEFS.linux += VBOX_WITH_LINUX_COMPILER_H
 endif
