--- a/src/VBox/HostDrivers/adpctl/VBoxNetAdpCtl.cpp
+++ b/src/VBox/HostDrivers/adpctl/VBoxNetAdpCtl.cpp
@@ -34,12 +34,13 @@
 #ifdef RT_OS_LINUX
 # include <arpa/inet.h>
 # include <net/if.h>
+# include <sys/types.h>
 # include <linux/types.h>
 /* Older versions of ethtool.h rely on these: */
 typedef unsigned long long u64;
-typedef __uint32_t u32;
-typedef __uint16_t u16;
-typedef __uint8_t u8;
+typedef uint32_t u32;
+typedef uint16_t u16;
+typedef uint8_t u8;
 # include <limits.h> /* for INT_MAX */
 # include <linux/ethtool.h>
 # include <linux/sockios.h>
