--- a/src/VBox/NetworkServices/NAT/pxping.c
+++ b/src/VBox/NetworkServices/NAT/pxping.c
@@ -53,7 +53,7 @@
 #include "lwip/icmp.h"
 
 #if defined(RT_OS_LINUX) && !defined(__USE_GNU)
-#if __GLIBC_PREREQ(2, 8)
+#if __MUSL_PREREQ
 /*
  * XXX: This is gross.  in6_pktinfo is now hidden behind _GNU_SOURCE
  * https://sourceware.org/bugzilla/show_bug.cgi?id=6775
--- a/src/VBox/RDP/client-1.8.4/rdpsnd_dsp.c
+++ b/src/VBox/RDP/client-1.8.4/rdpsnd_dsp.c
@@ -306,7 +306,7 @@
 
 	outsize = outnum * samplewidth;
 	*out = (unsigned char *) xmalloc(outsize);
-	bzero(*out, outsize);
+	//bzero(*out, outsize);
 
 	for (i = 0; i < outsize / (resample_to_channels * samplewidth); i++)
 	{
--- a/src/VBox/Runtime/r3/linux/semevent-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/semevent-linux.cpp
@@ -25,7 +25,7 @@
  */
 
 #include <features.h>
-#if __GLIBC_PREREQ(2,6) && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
+#if __MUSL_PREREQ && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
 
 /*
  * glibc 2.6 fixed a serious bug in the mutex implementation. We wrote this
--- a/src/VBox/Runtime/r3/linux/semeventmulti-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/semeventmulti-linux.cpp
@@ -26,7 +26,7 @@
 
 
 #include <features.h>
-#if __GLIBC_PREREQ(2,6) && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
+#if __MUSL_PREREQ && !defined(IPRT_WITH_FUTEX_BASED_SEMS)
 
 /*
  * glibc 2.6 fixed a serious bug in the mutex implementation. We wrote this
--- a/src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp
+++ b/src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp
@@ -32,7 +32,7 @@
 # define _GNU_SOURCE
 #endif
 #include <features.h>
-#if __GLIBC_PREREQ(2,4)
+#if __MUSL_PREREQ
 
 #include <sched.h>
 #include <unistd.h>