diff --git a/.plzconfig.alpine b/.plzconfig.alpine
index a3f43db..1072a9a 100644
--- a/.plzconfig.alpine
+++ b/.plzconfig.alpine
@@ -1,6 +1,7 @@
 [Plugin "go"]
 DefaultStatic = true
 GoTool = go
+Stdlib = //third_party/go:std
 
 [buildconfig]
 static-sandbox = true
diff --git a/third_party/go/BUILD b/third_party/go/BUILD
index 4542720..ba17755 100644
--- a/third_party/go/BUILD
+++ b/third_party/go/BUILD
@@ -14,6 +14,8 @@ go_toolchain(
     version = "1.19",
 )
 
+go_stdlib(name = "std")
+
 go_mod_download(
     name = "build_tools_dl",
     module = "github.com/peterebden/buildtools",
