diff --git a/services/sddm-autologin.pam b/services/sddm-autologin.pam
index b42991e..25c43d1 100644
--- a/services/sddm-autologin.pam
+++ b/services/sddm-autologin.pam
@@ -6,8 +6,8 @@ auth        required    pam_nologin.so
 auth        required    pam_permit.so
 -auth       optional    pam_gnome_keyring.so
 -auth       optional    pam_kwallet5.so
-account     include     system-local-login
-password    include     system-local-login
-session     include     system-local-login
+account     include     base-account
+password    include     base-password
+session     include     base-session
 -session    optional    pam_gnome_keyring.so auto_start
 -session    optional    pam_kwallet5.so auto_start
diff --git a/services/sddm.pam b/services/sddm.pam
index df11003..5c2d8a6 100644
--- a/services/sddm.pam
+++ b/services/sddm.pam
@@ -1,15 +1,15 @@
 #%PAM-1.0
 
-auth        include     system-login
+auth        include     base-auth
 -auth       optional    pam_gnome_keyring.so
 -auth       optional    pam_kwallet5.so
 
-account     include     system-login
+account     include     base-account
 
-password    include     system-login
+password    include     base-password
 -password   optional    pam_gnome_keyring.so    use_authtok
 
 session     optional    pam_keyinit.so          force revoke
-session     include     system-login
+session     include     base-session
 -session    optional    pam_gnome_keyring.so    auto_start
 -session    optional    pam_kwallet5.so         auto_start
