Reason: PAM 1.4 removed the pam_tally module
Upstream: No
diff --git a/services/sddm-autologin.pam b/services/sddm-autologin.pam
index 4c82048..4006ebb 100755
--- a/services/sddm-autologin.pam
+++ b/services/sddm-autologin.pam
@@ -1,9 +1,11 @@
 #%PAM-1.0
+auth        required    pam_faillock.so preauth
 auth        required    pam_env.so
-auth        required    pam_tally.so file=/var/log/faillog onerr=succeed
 auth        required    pam_shells.so
 auth        required    pam_nologin.so
 auth        required    pam_permit.so
+auth        required    pam_faillock.so authfail
+auth        required    pam_faillock.so authsucc
 -auth       optional    pam_gnome_keyring.so
 -auth       optional    pam_kwallet5.so
 account     include     system-local-login
