diff --git a/data/man/CMakeLists.txt b/data/man/CMakeLists.txt
index 1d84ec2..7fee115 100644
--- a/data/man/CMakeLists.txt
+++ b/data/man/CMakeLists.txt
@@ -1,4 +1,4 @@
-find_program(RST2MAN_EXECUTABLE NAMES rst2man rst2man.py rst2man2 rst2man2.py)
+find_program(RST2MAN_EXECUTABLE NAMES rst2man rst2man.py rst2man2 rst2man2.py rst2man-3)
 if(NOT RST2MAN_EXECUTABLE)
     message(FATAL "rst2man (python-docutils) is required to build man pages")
 endif()
