--- ./src/dblib/Makefile.am.orig
+++ ./src/dblib/Makefile.am
@@ -20,5 +20,5 @@
 SYMBOLS		=	-export-symbols-regex '^(db|bcp_|tdsdump_open|tdsdbopen|.*_xact|close_commit|open_commit|.?asprintf).*'
 endif
 libsybdb_la_LDFLAGS=	-version-info 5:0:0 $(SYMBOLS) $(FREETDS_SYMBOLIC)
-libsybdb_la_LIBADD=	../tds/libtds.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LIBICONV) $(FREETDS_LIBGCC)
+libsybdb_la_LIBADD=	../tds/libtds.la ../replacements/libreplacements.la $(NETWORK_LIBS) $(LTLIBICONV) $(FREETDS_LIBGCC)
 
--- ./src/tds/Makefile.am.orig
+++ ./src/tds/Makefile.am
@@ -19,7 +19,7 @@
 	bulk.c win_mutex.c \
 	$(AUTH_FILES)
 libtds_la_LDFLAGS=
-libtds_la_LIBADD=
+libtds_la_LIBADD= $(LTLIBICONV)
 
 noinst_HEADERS		= tds_willconvert.h encodings.h num_limits.h tds_types.h
 EXTRA_DIST		= tds_willconvert.h encodings.h num_limits.h tds_types.h \
