--- a/makefile
+++ b/makefile
@@ -23,7 +23,7 @@
 endif
 
 PREFIX = $(DESTDIR)/usr
-BIN_DIR = $(PREFIX)/games/
+BIN_DIR = $(PREFIX)/bin/
 DOC_DIR = $(PREFIX)/share/doc/$(PROG)/
 ICON_DIR = $(PREFIX)/share/icons/hicolor/
 DESKTOP_DIR = $(PREFIX)/share/applications/
@@ -35,7 +35,7 @@
 ifeq ($(UNIX),0)
 DATA_DIR =
 else
-DATA_DIR = $(PREFIX)/share/games/edgar/
+DATA_DIR = $(PREFIX)/share/edgar/
 endif
 
 CFLAGS += -Wall -pedantic
