--- a/phoronix-test-suite
+++ b/phoronix-test-suite
@@ -67,9 +67,9 @@
 if [ ! "X$PHP_BIN" = "X" ] && [ -x $PHP_BIN ]
 then
 	export PHP_BIN=$PHP_BIN
-elif [ -x /usr/bin/php8 ] || [ -x /usr/local/bin/php8 ] || [ -x /usr/pkg/bin/php8 ]
+elif [ -x /usr/bin/php82 ] || [ -x /usr/local/bin/php8 ] || [ -x /usr/pkg/bin/php8 ]
 then
-	export PHP_BIN="php8"
+	export PHP_BIN="php82"
 elif [ -x /usr/bin/php7 ] || [ -x /usr/local/bin/php7 ] || [ -x /usr/pkg/bin/php7 ]
 then
 	export PHP_BIN="php7"
