--- a/utils/hp2ps/ghc.mk	2017-02-25 22:37:12.000000000 -0600
+++ b/utils/hp2ps/ghc.mk	2017-02-25 23:21:33.000000000 -0600
@@ -17,10 +17,21 @@
                                    Utilities.c
 utils/hp2ps_dist_EXTRA_LIBRARIES = m
 utils/hp2ps_dist_PROGNAME        = $(CrossCompilePrefix)hp2ps
-utils/hp2ps_dist_INSTALL         = YES
+utils/hp2ps_dist_INSTALL         = NO
 utils/hp2ps_dist_INSTALL_INPLACE = YES
 
 utils/hp2ps_CC_OPTS += $(addprefix -I,$(GHC_INCLUDE_DIRS))
 
 $(eval $(call build-prog,utils/hp2ps,dist,0))
 
+utils/hp2ps_dist-install_C_SRCS  = AreaBelow.c Curves.c Error.c Main.c \
+                                   Reorder.c TopTwenty.c AuxFile.c Deviation.c \
+                                   HpFile.c Marks.c Scale.c TraceElement.c \
+                                   Axes.c Dimensions.c Key.c PsFile.c Shade.c \
+                                   Utilities.c
+utils/hp2ps_dist-install_PROGNAME = hp2ps
+utils/hp2ps_dist-install_TOPDIR  = YES
+utils/hp2ps_dist-install_INSTALL = YES
+utils/hp2ps_dist-install_INSTALL_INPLACE = NO
+
+$(eval $(call build-prog,utils/hp2ps,dist-install,1))
