--- openvswitch-2.0.0/tests/test-atomic.c
+++ openvswitch-2.0.0.alpine/tests/test-atomic.c
@@ -83,22 +83,26 @@
     TEST_ATOMIC_TYPE(atomic_uint, unsigned int);
     TEST_ATOMIC_TYPE(atomic_long, long int);
     TEST_ATOMIC_TYPE(atomic_ulong, unsigned long int);
+#if 0
     TEST_ATOMIC_TYPE(atomic_llong, long long int);
     TEST_ATOMIC_TYPE(atomic_ullong, unsigned long long int);
+#endif
     TEST_ATOMIC_TYPE(atomic_size_t, size_t);
     TEST_ATOMIC_TYPE(atomic_ptrdiff_t, ptrdiff_t);
+#if 0
     TEST_ATOMIC_TYPE(atomic_intmax_t, intmax_t);
     TEST_ATOMIC_TYPE(atomic_uintmax_t, uintmax_t);
+#endif
     TEST_ATOMIC_TYPE(atomic_intptr_t, intptr_t);
     TEST_ATOMIC_TYPE(atomic_uintptr_t, uintptr_t);
-    TEST_ATOMIC_TYPE(atomic_uint8_t, uint8_t);
-    TEST_ATOMIC_TYPE(atomic_int8_t, int8_t);
     TEST_ATOMIC_TYPE(atomic_uint16_t, uint16_t);
     TEST_ATOMIC_TYPE(atomic_int16_t, int16_t);
     TEST_ATOMIC_TYPE(atomic_uint32_t, uint32_t);
     TEST_ATOMIC_TYPE(atomic_int32_t, int32_t);
+#if 0
     TEST_ATOMIC_TYPE(atomic_uint64_t, uint64_t);
     TEST_ATOMIC_TYPE(atomic_int64_t, int64_t);
+#endif

     test_atomic_flag();

