not just aarch64

--- a/bindgen-tests/tests/tests.rs
+++ b/bindgen-tests/tests/tests.rs
@@ -781,7 +781,7 @@ fn test_wrap_static_fns() {
     // aarch64-linux has a bug, remove custom source when it is solved:
     // https://github.com/rust-lang/rust-bindgen/issues/3234
     let wrap_static_fns_c_name =
-        if cfg!(all(target_arch = "aarch64", target_os = "linux")) {
+        if cfg!(all(any(target_arch = "aarch64", target_arch = "arm"), target_os = "linux")) {
             "wrap_static_fns_aarch64_linux"
         } else {
             "wrap_static_fns"
@@ -804,7 +804,7 @@ fn test_wrap_static_fns() {
 
     // aarch64-linux has a bug, remove when it is solved:
     // https://github.com/rust-lang/rust-bindgen/issues/3234
-    #[cfg(all(target_arch = "aarch64", target_os = "linux"))]
+    #[cfg(all(any(target_arch = "aarch64", target_arch = "arm"), target_os = "linux"))]
     {
         builder = builder.clang_arg("-DDISABLE_VA");
     }
