--- a/io/mmap.c
+++ b/io/mmap.c
@@ -201,7 +201,11 @@
 			prot |= PROT_EXEC;
 			break;
 		case 'S':
+#if defined(MAP_SYNC) 
 			flags = MAP_SYNC | MAP_SHARED_VALIDATE;
+#else
+			flags = 0;
+#endif
 
 			/*
 			 * If MAP_SYNC and MAP_SHARED_VALIDATE aren't defined
@@ -281,7 +285,11 @@
 	mapping->offset = offset;
 	mapping->name = filename;
 	mapping->prot = prot;
+#if defined(MAP_SYNC)
 	mapping->map_sync = (flags == (MAP_SYNC | MAP_SHARED_VALIDATE));
+#else
+	mapping->map_sync = 0;
+#endif
 	return 0;
 }
 
