From ad88cf33e0b4c4aa5487f5f873cb2b1bb094dd61 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Tue, 10 May 2022 18:48:22 -0700
Subject: [PATCH 1/2] auto.master: change paths in etc

---
 samples/auto.master | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/samples/auto.master b/samples/auto.master
index 5483e3c..2bb64d9 100644
--- a/samples/auto.master
+++ b/samples/auto.master
@@ -4,7 +4,7 @@
 # mount-point [map-type[,format]:]map [options]
 # For details of the format look at auto.master(5).
 #
-/misc	/etc/auto.misc
+/misc	/etc/autofs/auto.misc
 #
 # NOTE: mounts done from a hosts map will be mounted with the
 #	"nosuid" and "nodev" options unless the "suid" and "dev"
@@ -12,14 +12,14 @@
 #
 /net	-hosts
 #
-# Include /etc/auto.master.d/*.autofs
+# Include /etc/autofs/auto.master.d/*.autofs
 # To add an extra map using this mechanism you will need to add
 # two configuration items - one /etc/auto.master.d/extra.autofs file
 # (using the same line format as the auto.master file)
 # and a separate mount map (e.g. /etc/auto.extra or an auto.extra NIS map)
 # that is referred to by the extra.autofs file.
 #
-+dir:/etc/auto.master.d
++dir:/etc/autofs/auto.master.d
 #
 # If you have fedfs set up and the related binaries, either
 # built as part of autofs or installed from another package,
-- 
2.36.1

