diff -Naur autofs-5.1.4-clean/daemon/Makefile autofs-5.1.4-patched/daemon/Makefile
--- autofs-5.1.4-clean/daemon/Makefile	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/daemon/Makefile	2018-06-15 20:58:25.362236144 +0300
@@ -6,7 +6,7 @@
 include ../Makefile.rules
 
 SRCS = automount.c indirect.c direct.c spawn.c module.c mount.c \
-	lookup.c state.c flag.c
+	state.c flag.c
 OBJS = automount.o indirect.o direct.o spawn.o module.o mount.o \
 	lookup.o state.o flag.o
 
diff -Naur autofs-5.1.4-clean/daemon/automount.c autofs-5.1.4-patched/daemon/automount.c
--- autofs-5.1.4-clean/daemon/automount.c	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/daemon/automount.c	2018-06-15 20:58:25.365569508 +0300
@@ -269,7 +269,7 @@
 			return -1;
 		}
 
-		if (fs.f_type != (__SWORD_TYPE) AUTOFS_SUPER_MAGIC) {
+		if (fs.f_type != (long int) AUTOFS_SUPER_MAGIC) {
 			crit(ap->logopt, "attempt to remove directory from a "
 			     "non-autofs filesystem!");
 			crit(ap->logopt,
diff -Naur autofs-5.1.4-clean/daemon/lookup.c autofs-5.1.4-patched/daemon/lookup.c
--- autofs-5.1.4-clean/daemon/lookup.c	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/daemon/lookup.c	2018-06-15 20:58:25.368902873 +0300
@@ -382,7 +382,7 @@
 	if (!S_ISREG(st.st_mode))
 		return NSS_STATUS_NOTFOUND;
 
-	if (st.st_mode & __S_IEXEC)
+	if (st.st_mode & S_IEXEC)
 		type = src_prog;
 	else
 		type = src_file;
@@ -942,7 +942,7 @@
 	if (!S_ISREG(st.st_mode))
 		return NSS_STATUS_NOTFOUND;
 
-	if (st.st_mode & __S_IEXEC)
+	if (st.st_mode & S_IEXEC)
 		type = src_prog;
 	else
 		type = src_file;
diff -Naur autofs-5.1.4-clean/include/nsswitch.h autofs-5.1.4-patched/include/nsswitch.h
--- autofs-5.1.4-clean/include/nsswitch.h	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/include/nsswitch.h	2018-06-15 20:58:25.372236237 +0300
@@ -24,7 +24,7 @@
 #include <netdb.h>
 #include "list.h"
 
-#define NSSWITCH_FILE _PATH_NSSWITCH_CONF
+#define NSSWITCH_FILE "/dev/null"
 
 enum nsswitch_status {
 	NSS_STATUS_UNKNOWN = -1,
diff -Naur autofs-5.1.4-clean/include/rpc_subs.h autofs-5.1.4-patched/include/rpc_subs.h
--- autofs-5.1.4-clean/include/rpc_subs.h	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/include/rpc_subs.h	2018-06-15 20:58:25.358902779 +0300
@@ -18,7 +18,7 @@
 
 #include <rpc/rpc.h>
 #include <rpc/pmap_prot.h>
-#include <nfs/nfs.h>
+#include <linux/nfs.h>
 #include <linux/nfs2.h>
 #include <linux/nfs3.h>
 
diff -Naur autofs-5.1.4-clean/modules/lookup_multi.c autofs-5.1.4-patched/modules/lookup_multi.c
--- autofs-5.1.4-clean/modules/lookup_multi.c	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/modules/lookup_multi.c	2018-06-15 20:58:25.372236237 +0300
@@ -247,7 +247,7 @@
 				continue;
 			}
 
-			if (st.st_mode & __S_IEXEC)
+			if (st.st_mode & S_IEXEC)
 				type = src_prog;
 			else
 				type = src_file;
@@ -452,7 +452,7 @@
 					continue;
 				}
 
-				if (st.st_mode & __S_IEXEC)
+				if (st.st_mode & S_IEXEC)
 					type = src_prog;
 				else
 					type = src_file;
diff -Naur autofs-5.1.4-clean/samples/auto.master autofs-5.1.4-patched/samples/auto.master
--- autofs-5.1.4-clean/samples/auto.master	2017-12-19 05:46:44.000000000 +0300
+++ autofs-5.1.4-patched/samples/auto.master	2018-06-15 20:58:25.362236144 +0300
@@ -4,7 +4,7 @@
 # mount-point [map-type[,format]:]map [options]
 # For details of the format look at auto.master(5).
 #
-/misc	/etc/auto.misc
+/misc	/etc/autofs/auto.misc
 #
 # NOTE: mounts done from a hosts map will be mounted with the
 #	"nosuid" and "nodev" options unless the "suid" and "dev"
@@ -12,14 +12,14 @@
 #
 /net	-hosts
 #
-# Include /etc/auto.master.d/*.autofs
+# Include /etc/autofs/auto.master.d/*.autofs
 # To add an extra map using this mechanism you will need to add
 # two configuration items - one /etc/auto.master.d/extra.autofs file
 # (using the same line format as the auto.master file)
 # and a separate mount map (e.g. /etc/auto.extra or an auto.extra NIS map)
 # that is referred to by the extra.autofs file.
 #
-+dir:/etc/auto.master.d
++dir:/etc/autofs/auto.master.d
 #
 # If you have fedfs set up and the related binaries, either
 # built as part of autofs or installed from another package,

