diff --git a/third_party/xla/graphcycles.h b/third_party/xla/graphcycles.h
index df73735..8b59a64 100755
--- a/third_party/xla/graphcycles.h
+++ b/third_party/xla/graphcycles.h
@@ -21,6 +21,7 @@ limitations under the License.
 #ifndef XLA_SERVICE_GRAPHCYCLES_GRAPHCYCLES_H_
 #define XLA_SERVICE_GRAPHCYCLES_GRAPHCYCLES_H_
 
+#include <cstdint>
 #include <vector>
 
 // GraphCycles detects the introduction of a cycle into a directed
