diff --git a/CMakeLists.txt b/CMakeLists.txt
index a67905f..6c80698 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -128,8 +128,8 @@ endif()
 target_compile_definitions(vvave
   PRIVATE $<$<OR:$<CONFIG:Debug>,$<CONFIG:RelWithDebInfo>>:QT_QML_DEBUG>)
 
-target_include_directories(vvave PRIVATE ${TAGLIB_INCLUDE_DIRS})
-target_link_libraries(vvave MauiKit Qt5::Network Qt5::Sql Qt5::Qml Qt5::Xml Qt5::Multimedia Qt5::QuickControls2 KF5::CoreAddons KF5::I18n ${TAGLIB_LIBRARIES})
+target_include_directories(vvave PRIVATE Taglib:Taglib)
+target_link_libraries(vvave MauiKit Qt5::Network Qt5::Sql Qt5::Qml Qt5::Xml Qt5::Multimedia Qt5::QuickControls2 KF5::CoreAddons KF5::I18n Taglib::Taglib)
 
 if(UNIX AND NOT APPLE AND NOT ANDROID)
     install(TARGETS vvave ${KDE_INSTALL_TARGETS_DEFAULT_ARGS})
