use alpine pebble instead of downloading

--- a/test/integration_tests/dnsrobocert_test.py
+++ b/test/integration_tests/dnsrobocert_test.py
@@ -87,7 +87,9 @@
     workspace = tmp_path / "workspace"
     os.mkdir(str(workspace))
 
-    pebble_path, challtestsrv_path, pebble_config_path = _fetch(str(workspace))
+    pebble_path = "/usr/bin/pebble"
+    challtestsrv_path = "/usr/bin/pebble-challtestsrv"
+    pebble_config_path = _build_pebble_config(str(workspace))
 
     environ = os.environ.copy()
     environ["PEBBLE_VA_NOSLEEP"] = "1"
