diff -ru a/C/CpuArch.c b/C/CpuArch.c
--- a/C/CpuArch.c	2021-07-13 05:10:00.000000000 -0400
+++ b/C/CpuArch.c	2022-03-05 11:08:23.363390656 -0500
@@ -421,8 +421,6 @@
 
 #ifdef USE_HWCAP
 
-#include <asm/hwcap.h>
-
   #define MY_HWCAP_CHECK_FUNC_2(name1, name2) \
   BoolInt CPU_IsSupported_ ## name1() { return (getauxval(AT_HWCAP)  & (HWCAP_  ## name2)) ? 1 : 0; }
 
diff -ru a/C/Threads.c b/C/Threads.c
--- a/C/Threads.c	2021-12-21 08:00:00.000000000 -0500
+++ b/C/Threads.c	2022-03-05 11:07:47.203504536 -0500
@@ -257,7 +257,7 @@
       */
 
       // ret2 =
-      pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
+      //pthread_attr_setaffinity_np(&attr, sizeof(*cpuSet), cpuSet);
       // if (ret2) ret = ret2;
       #endif
     }
@@ -267,14 +267,12 @@
     if (!ret)
     {
       p->_created = 1;
-      /*
       if (cpuSet)
       {
         // ret2 =
         pthread_setaffinity_np(p->_tid, sizeof(*cpuSet), cpuSet);
         // if (ret2) ret = ret2;
       }
-      */
     }
   }
   // ret2 =
diff -ru a/C/Threads.h b/C/Threads.h
--- a/C/Threads.h	2021-12-21 08:00:00.000000000 -0500
+++ b/C/Threads.h	2022-03-05 11:07:47.203504536 -0500
@@ -19,6 +19,7 @@
 #endif
 
 #include <pthread.h>
+#include <sched.h>
 
 #endif
 
diff -ru a/CPP/Windows/SystemInfo.cpp b/CPP/Windows/SystemInfo.cpp
--- a/CPP/Windows/SystemInfo.cpp	2021-11-06 04:19:39.000000000 -0400
+++ b/CPP/Windows/SystemInfo.cpp	2022-03-05 11:26:27.178812388 -0500
@@ -36,9 +36,6 @@
 #endif
 */
 
-#ifdef MY_CPU_ARM_OR_ARM64
-#include <asm/hwcap.h>
-#endif
 #endif
 
 #ifdef __linux__
