From cdc4476e00d0b9924d0b535c222addb565a1aa8f Mon Sep 17 00:00:00 2001
From: Michael van Tellingen <michael@mvantellingen.nl>
Date: Mon, 26 Jun 2017 10:03:05 +0200
Subject: [PATCH 8/8] Allow latest django-haystack release (for Django 1.11)

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 587f10e..41e106b 100755
--- a/setup.py
+++ b/setup.py
@@ -26,7 +26,7 @@ install_requires = [
     # https://github.com/AndrewIngram/django-extra-views/issues/114
     'django-extra-views>=0.2,<0.6.5',
     # Search support
-    'django-haystack>=2.5.0,<2.6.0',
+    'django-haystack>=2.5.0,<2.7.0',
     # Treebeard is used for categories
     'django-treebeard>=4.1.0',
     # Sorl is used as the default thumbnailer
-- 
2.9.4

