commit c042121
Author: Pablo Correa Gómez <pabloyoyoista@postmarketos.org>
Date:   Fri Jul 25 11:45:00 2025 +0200

    merge-usr: fix return value not being a variable

diff --git a/merge-usr b/merge-usr
index 34dc646..e4866ee 100755
--- a/merge-usr
+++ b/merge-usr
@@ -474,7 +474,7 @@ copy_tree() {
 				copy_tree "$src" "$dst"
 				errors=$((errors + $?))
 				if [ "$DRYRUN" = "false" ]; then
-					return errors
+					return $errors
 				fi
 			else
 				log_error "Conflict for directory '$src'"
