diff --git a/configure.ac b/configure.ac
index 14aa8d3..95e14fc 100755
--- a/configure.ac
+++ b/configure.ac
@@ -243,7 +243,7 @@ AC_CONFIG_FILES([shadowsocks-libev.pc
 
 AM_COND_IF([USE_SYSTEM_SHARED_LIB],
 		   [AC_DEFINE([USE_SYSTEM_SHARED_LIB], [1], [Define if use system shared lib.])],
-		   [AC_CONFIG_FILES([libbloom/Makefile libcork/Makefile libipset/Makefile])])
+		   [])
 
 AC_ARG_ENABLE(connmarktos,
 [AS_HELP_STRING(--enable-connmarktos, Enable saved connmark to IP TOS QoS feature)],

