From 161c306ffb9d8f96233e1c0588e013f4f01de7c0 Mon Sep 17 00:00:00 2001
From: Newbyte <newbie13xd@gmail.com>
Date: Tue, 14 Mar 2023 19:44:30 +0100
Subject: [PATCH 2/2] DOWNSTREAM: remove systemd dependencies we can't satisfy

---
 meson.build | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/meson.build b/meson.build
index 835712192..e190ab677 100644
--- a/meson.build
+++ b/meson.build
@@ -100,15 +100,8 @@ if xdmcp_dep.found() and get_option('tcp-wrappers')
   libwrap_dep = cc.find_library('wrap')
 endif
 # systemd
-systemd_dep = dependency('systemd')
 libsystemd_dep = dependency('libsystemd')
-systemd_multiseat_x = find_program('systemd-multi-seat-x',
-  required: false,
-  dirs: [
-    systemd_dep.get_pkgconfig_variable('systemdutildir'),
-    '/lib/systemd',
-    '/usr/lib/systemd',
-  ])
+systemd_multiseat_x = find_program('systemd-multi-seat-x', required: false)
 systemd_x_server = systemd_multiseat_x.found()? systemd_multiseat_x.path() : '/lib/systemd/systemd-multi-seat-x'
 # Plymouth
 plymouth_dep = dependency('ply-boot-client', required: get_option('plymouth'))
-- 
2.39.2

