we use base-* named configs
--
diff --git a/data/pam-openembedded/gdm-autologin.pam b/data/pam-openembedded/gdm-autologin.pam
index a9f4354..3e0708b 100644
--- a/data/pam-openembedded/gdm-autologin.pam
+++ b/data/pam-openembedded/gdm-autologin.pam
@@ -1,5 +1,5 @@
 #%PAM-1.0
-auth       required    pam_permit.so
-account    include     common-account
-password   include     common-auth
-session    include     common-session
+auth       required    pam_rootok.so
+account    include     base-account
+password   include     base-auth
+session    include     base-session
diff --git a/data/pam-openembedded/gdm-launch-environment.pam b/data/pam-openembedded/gdm-launch-environment.pam
index a9f4354..3e0708b 100644
--- a/data/pam-openembedded/gdm-launch-environment.pam
+++ b/data/pam-openembedded/gdm-launch-environment.pam
@@ -1,5 +1,5 @@
 #%PAM-1.0
-auth       required    pam_permit.so
-account    include     common-account
-password   include     common-auth
-session    include     common-session
+auth       required    pam_rootok.so
+account    include     base-account
+password   include     base-auth
+session    include     base-session
diff --git a/data/pam-openembedded/gdm-password.pam b/data/pam-openembedded/gdm-password.pam
index 758464d..f4643d1 100644
--- a/data/pam-openembedded/gdm-password.pam
+++ b/data/pam-openembedded/gdm-password.pam
@@ -1,5 +1,5 @@
 #%PAM-1.0
-auth       include     common-auth
-account    include     common-account
-password   include     common-password
-session    include     common-session
+auth       include     base-auth
+account    include     base-account
+password   include     base-password
+session    include     base-session
