Use system-provided libsass.

--- a/lib/sassc/native.rb
+++ b/lib/sassc/native.rb
@@ -6,12 +6,7 @@
   module Native
     extend FFI::Library
 
-    dl_ext = RbConfig::MAKEFILE_CONFIG['DLEXT']
-    begin
-      ffi_lib File.expand_path("libsass.#{dl_ext}", __dir__)
-    rescue LoadError # Some non-rvm environments don't copy a shared object over to lib/sassc
-      ffi_lib File.expand_path("libsass.#{dl_ext}", "#{__dir__}/../../ext")
-    end
+    ffi_lib "/usr/lib/libsass.so.1"
 
     require_relative "native/sass_value"
 
