From a72fed555a3d2698b3aebc7a429c9f038139308d Mon Sep 17 00:00:00 2001
From: Tycho Andersen <tycho@docker.com>
Date: Thu, 16 Mar 2017 11:48:08 -0700
Subject: [PATCH] drop some binaries from policycoreutils

Signed-off-by: Tycho Andersen <tycho@docker.com>
---
 policycoreutils/Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 962ac12..817fe36 100644
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,7 @@
-SUBDIRS = sepolicy setfiles semanage load_policy newrole run_init sandbox secon audit2allow sestatus semodule_package semodule semodule_link semodule_expand semodule_deps sepolgen-ifgen setsebool scripts po man gui hll
+SUBDIRS = sepolicy setfiles semanage load_policy newrole run_init sandbox secon audit2allow sestatus semodule_package semodule semodule_link semodule_expand semodule_deps sepolgen-ifgen setsebool scripts po man gui hll
 
 INOTIFYH = $(shell ls /usr/include/sys/inotify.h 2>/dev/null)
 
-ifeq (${INOTIFYH}, /usr/include/sys/inotify.h)
-	SUBDIRS += restorecond
-endif
-
 all install relabel clean indent:
 	@for subdir in $(SUBDIRS); do \
 		(cd $$subdir && $(MAKE) $@) || exit 1; \
-- 
2.11.1

