diff --git a/newrole/newrole.pamd b/newrole/newrole.pamd
index 683c544..dc947b7 100644
--- a/newrole/newrole.pamd
+++ b/newrole/newrole.pamd
@@ -1,8 +1,8 @@
 #%PAM-1.0
 # Uncomment the next line if you do not want to enter your passwd every time
 # auth       sufficient   pam_rootok.so
-auth       include	system-auth
-account    include	system-auth
-password   include	system-auth
-session    include	system-auth
+auth       include	base-auth
+account    include	base-account
+password   include	base-password
+session    include	base-session
 session    optional	pam_xauth.so
diff --git a/run_init/run_init.pamd b/run_init/run_init.pamd
index ef46013..74cb988 100644
--- a/run_init/run_init.pamd
+++ b/run_init/run_init.pamd
@@ -1,8 +1,8 @@
 #%PAM-1.0
 # Uncomment the next line if you do not want to enter your passwd every time
 #auth       sufficient   pam_rootok.so
-auth       include	system-auth
-account    include	system-auth
-password   include	system-auth
-session    include	system-auth
+auth       include	base-auth
+account    include	base-account
+password   include	base-password
+session    include	base-session
 session    optional	pam_xauth.so
