--- a/policycoreutisestatus/Makefile
+++ b/sestatus/Makefile
@@ -28,10 +28,6 @@ install: all
 	[ -d $(DESTDIR)$(MANDIR)/man5 ] || mkdir -p $(DESTDIR)$(MANDIR)/man5
 	-mkdir -p $(DESTDIR)$(BINDIR)
 	-mkdir -p $(DESTDIR)$(SBINDIR)
-	# Some tools hard code /usr/sbin/sestatus ; add a compatibility symlink
-	# install will overwrite a symlink, so create the symlink before calling
-	# install to allow distributions with BINDIR == SBINDIR
-	ln -sf --relative $(DESTDIR)$(BINDIR)/sestatus $(DESTDIR)$(SBINDIR)
 	install -m 755 sestatus $(DESTDIR)$(BINDIR)
 	install -m 644 sestatus.8 $(DESTDIR)$(MANDIR)/man8
 	install -m 644 sestatus.conf.5 $(DESTDIR)$(MANDIR)/man5
