--- a/Makefile
+++ b/Makefile
@@ -28,8 +28,6 @@ install:
 	strip $(destbin)/tkey-ssh-agent
 	install -Dm644 system/tkey-ssh-agent.1 $(destman1)/tkey-ssh-agent.1
 	gzip -n9f $(destman1)/tkey-ssh-agent.1
-	install -Dm644 system/tkey-ssh-agent.service.tmpl $(destunit)/tkey-ssh-agent.service
-	sed -i -e "s,##BINDIR##,$(PREFIX)/bin," $(destunit)/tkey-ssh-agent.service
 	install -Dm644 system/60-tkey.rules $(destrules)/60-tkey.rules
 .PHONY: uninstall
 uninstall:
