--- namecoin-nc*/src/Makefile
+++ namecoin-nc*/src/Makefile
@@ -9,24 +9,20 @@
 DEFS += -DBOOST_THREAD_USE_LIB
 
 INCLUDEPATHS= \
-	-I../libs/openssl-1.0.1i/include \
-	-I../libs/db-4.7.25.NC/build_unix \
-	-I../libs/boost_1_50_0
+	-I/usr/include
 
 LIBPATHS= \
-	-L../libs/openssl-1.0.1i \
-	-L../libs/db-4.7.25.NC/build_unix \
-	-L../libs/boost_1_50_0/stage/lib
+	-L/usr/lib
 
 LIBBOOST_SUFFIX=
 
 LIBS= \
- -Wl,-Bstatic \
+ -Wl,-Bdynamic \
    -l boost_system$(LIBBOOST_SUFFIX) \
    -l boost_filesystem$(LIBBOOST_SUFFIX) \
-   -l boost_program_options$(LIBBOOST_SUFFIX) \
-   -l boost_thread$(LIBBOOST_SUFFIX) \
-   -l boost_chrono$(LIBBOOST_SUFFIX) \
+   -l boost_program_options-mt$(LIBBOOST_SUFFIX) \
+   -l boost_thread-mt$(LIBBOOST_SUFFIX) \
+   -l boost_chrono-mt$(LIBBOOST_SUFFIX) \
    -l db_cxx \
    -l ssl \
    -l crypto
