From 6f00b80368b4c1f24ab47a9ee216ae39d17a5e45 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Fri, 10 Jul 2026 23:11:56 +0200
Subject: [PATCH 6/7] tools-headers.patch

Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 tools/arch/x86/include/asm/insn.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/arch/x86/include/asm/insn.h b/tools/arch/x86/include/asm/insn.h
index 8f10f2943370..9537b0078e5a 100644
--- a/tools/arch/x86/include/asm/insn.h
+++ b/tools/arch/x86/include/asm/insn.h
@@ -7,6 +7,7 @@
  * Copyright (C) IBM Corporation, 2009
  */
 
+#include <linux/compiler.h>
 #include <asm/byteorder.h>
 /* insn_attr_t is defined in inat.h */
 #include "inat.h" /* __ignore_sync_check__ */
-- 
2.55.0

