diff --git a/include/tscore/Allocator.h b/include/tscore/Allocator.h
index fd5200b..8b5ba64 100644
--- a/include/tscore/Allocator.h
+++ b/include/tscore/Allocator.h
@@ -45,7 +45,6 @@
 #include "tscore/ink_queue.h"
 #include "tscore/ink_defs.h"
 #include "tscore/ink_resource.h"
-#include <execinfo.h>
 
 #define RND16(_x) (((_x) + 15) & ~15)
 
@@ -200,14 +199,9 @@ public:
   C *
   alloc()
   {
-    void *callstack[3];
-    int frames = backtrace(callstack, 3);
     C *ptr     = ClassAllocator<C, Destruct_on_free>::alloc();
 
     const void *symbol = nullptr;
-    if (frames == 3 && callstack[2] != nullptr) {
-      symbol = callstack[2];
-    }
 
     tracker.increment(symbol, (int64_t)sizeof(C), this->fl->name);
     ink_mutex_acquire(&trackerLock);
diff --git a/src/tscore/ink_resource.cc b/src/tscore/ink_resource.cc
index 14a1904..d8160d3 100644
--- a/src/tscore/ink_resource.cc
+++ b/src/tscore/ink_resource.cc
@@ -24,7 +24,6 @@
 #include "tscore/ink_assert.h"
 #include "tscore/ink_atomic.h"
 #include "tscore/ink_resource.h"
-#include <execinfo.h>
 
 int res_track_memory          = 0; // Disabled by default
 uint64_t ssl_memory_allocated = 0;
--- a/iocore/eventsystem/P_UnixEThread.h
+++ b/iocore/eventsystem/P_UnixEThread.h
@@ -32,7 +32,6 @@
 
 #include "I_EThread.h"
 #include "I_EventProcessor.h"
-#include <execinfo.h>
 
 const ink_hrtime DELAY_FOR_RETRY = HRTIME_MSECONDS(10);
 
