--- ./configure
+++ ./configure
@@ -18517,6 +18517,20 @@
 	{ echo "$as_me:$LINENO: result: $ac_cv_ps_command" >&5
 echo "${ECHO_T}$ac_cv_ps_command" >&6; }
 
+##########
+# 8/12/10 R. Boisvert added to work with Busybox ps
+elif ps 2>/dev/null | \
+	egrep -i "^PID +USER +TIME +COMMAND" >/dev/null
+then
+	ac_cv_ps_command="$PATH_TO_PS"
+	ac_cv_ps_varlist="&procpid,&procuid,&procpcpu,procprog"
+	ac_cv_ps_format="%d %s %s %s"
+	ac_cv_ps_cols=4
+	{ echo "$as_me:$LINENO: result: $ac_cv_ps_command" >&5
+echo "${ECHO_T}$ac_cv_ps_command" >&6; }
+# 8/12/10 R. Boisvert added to work with Busybox ps
+##########
+
 elif ps -Al 2>/dev/null | \
 	egrep -i "^ *F +S +UID +PID +PPID +CLS +PRI +NI +C +ADDR +SZ +WCHAN +TTY +TIME +COMD" >/dev/null
 then
@@ -18750,6 +18764,18 @@
 	{ echo "$as_me:$LINENO: result: $with_ping_command" >&5
 echo "${ECHO_T}$with_ping_command" >&6; }
 
+##########
+# 8/12/10 R. Boisvert added to work with Busybox ping
+elif $PATH_TO_PING -c 1 127.0.0.1 2>/dev/null | \
+	egrep -i "^round-trip|^rtt" >/dev/null
+then
+	with_ping_command="$PATH_TO_PING -c %d %s"
+	ac_cv_ping_packets_first=yes
+	{ echo "$as_me:$LINENO: result: $with_ping_command" >&5
+echo "${ECHO_T}$with_ping_command" >&6; }
+# 8/12/10 R. Boisvert added to work with Busybox ping
+##########
+
 else
 	{ echo "$as_me:$LINENO: WARNING: unable to find usable ping syntax" >&5
 echo "$as_me: WARNING: unable to find usable ping syntax" >&2;}
@@ -19037,7 +19063,12 @@
 	then
 		{ echo "$as_me:$LINENO: checking for nslookup syntax" >&5
 echo $ECHO_N "checking for nslookup syntax... $ECHO_C" >&6; }
-		if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep "Invalid option: sil" >/dev/null
+##########
+# 8/12/10 R. Boisvert added to work with Busybox nslookup
+# original	if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep "Invalid option: sil" >/dev/null
+		if $PATH_TO_NSLOOKUP -sil 127.0.0.1 2>&1 | grep "Usage" >/dev/null
+# 8/12/10 R. Boisvert added to work with Busybox nslookup
+##########
 		then
 			ac_cv_nslookup_command="$PATH_TO_NSLOOKUP"
 			{ echo "$as_me:$LINENO: result: $ac_cv_nslookup_command" >&5
