--- lyx-2.4.4/src/support/gzstream.cpp~
+++ lyx-2.4.4/src/support/gzstream.cpp
@@ -93,7 +93,7 @@
     int n_putback = gptr() - eback();
     if ( n_putback > 4)
         n_putback = 4;
-    memcpy( buffer + (4 - n_putback), gptr() - n_putback, n_putback);
+    memmove( buffer + (4 - n_putback), gptr() - n_putback, n_putback);
 
     int num = gzread( file, buffer+4, bufferSize-4);
     if (num <= 0) // ERROR or EOF
