diff --git a/internal/testutil/dockertestutil.go b/internal/testutil/dockertestutil.go
index 9f5fa2f6..b5570f85 100644
--- a/internal/testutil/dockertestutil.go
+++ b/internal/testutil/dockertestutil.go
@@ -12,6 +12,8 @@
 // RunDockerAndGetOutputOrSkip runs Docker and returns the output as a string.
 func RunDockerAndGetOutputOrSkip(tb testing.TB, args ...string) string {
 	tb.Helper()
+	tb.Skip("skipping test that runs docker...")
+
 	tb.Logf("running docker %v", args)
 
 	c := exec.Command("docker", args...)
