--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,9 +51,8 @@
 
 find_package(ZLIB REQUIRED)
 
-# find_package(Intl REQUIRED)
-find_package(QUAZIP REQUIRED)
-#find_package(QuaZip-Qt5 REQUIRED)
+find_package(Intl REQUIRED)
+pkg_check_modules(QUAZIP REQUIRED IMPORTED_TARGET quazip1-qt5)
 
 if(CLICK_MODE)
     # Find out the architecture for package building
@@ -113,9 +113,6 @@
     ui/resources.qrc
     ${QML_COMPILED_RES})
 
-#target_include_directories(sturmreader SYSTEM PRIVATE QuaZip::QuaZip)
-target_include_directories(sturmreader SYSTEM PRIVATE ${QUAZIP_INCLUDE_DIR})
-
 add_library(qtrarobjslib STATIC IMPORTED)
 set_property(TARGET qtrarobjslib PROPERTY IMPORTED_LOCATION ${CMAKE_CURRENT_BINARY_DIR}/libqtrarobjs.a)
 
@@ -125,8 +122,7 @@
 target_link_libraries(sturmreader
     poppler-qt5
     ${Intl_LIBRARIES}
-    #QuaZip::QuaZip
-    ${QUAZIP_LIBRARIES}
+    PkgConfig::QUAZIP
     qtrarlib qtrarobjslib
     pthread
     ZLIB::ZLIB
