diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4b8fe0d..5b93a59 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -76,9 +76,8 @@ else()
 endif()
 
 find_package(ZLIB REQUIRED)
-# find_package(Intl REQUIRED)
-find_package(QUAZIP REQUIRED)
-# find_package(QuaZip-Qt5 REQUIRED)
+find_package(Intl REQUIRED)
+pkg_check_modules(QUAZIP REQUIRED IMPORTED_TARGET quazip1-qt6)
 if(${QT_MAJOR} EQUAL 5)
 	pkg_check_modules(POPPLER_QT5 REQUIRED poppler-qt5)
 elseif(${QT_MAJOR} EQUAL 6)
@@ -158,17 +157,13 @@ add_executable(sturmreader
     ui/resources.qrc
     ${QML_COMPILED_RES})
 
-#target_include_directories(sturmreader SYSTEM PRIVATE QuaZip::QuaZip)
-target_include_directories(sturmreader SYSTEM PRIVATE ${QUAZIP_INCLUDE_DIR})
-
 
 add_dependencies(sturmreader qtrar qtrarobjs)
 if(${QT_MAJOR} EQUAL 5)
 	target_link_libraries(sturmreader
 		poppler-qt5
 		${Intl_LIBRARIES}
-		#QuaZip::QuaZip
-		${QUAZIP_LIBRARIES}
+		PkgConfig::QUAZIP
 		qtrar qtrarobjs
 		#qtrarlib qtrarobjslib
 		pthread
@@ -178,7 +173,7 @@ elseif(${QT_MAJOR} EQUAL 6)
 	target_link_libraries(sturmreader
 		poppler-qt6
 		${Intl_LIBRARIES}
-		${QUAZIP_LIBRARIES}
+		PkgConfig::QUAZIP
 		qtrar qtrarobjs
 		# qtrarlib qtrarobjslib
 		pthread
