diff --git a/data/meson.build b/data/meson.build
index f35d456..15c54ae 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -33,7 +33,7 @@ install_data('com.gitlab.lgtrombetta.Compass.gschema.xml',
   install_dir: join_paths(get_option('datadir'), 'glib-2.0/schemas')
 )
 
-install_data('90-magn.rules', install_dir: '/lib/udev/rules.d')
+install_data('90-magn.rules', install_dir: '/usr/lib/udev/rules.d/')
 
 compile_schemas = find_program('glib-compile-schemas', required: false)
 if compile_schemas.found()
