diff --git a/include/wx/matrix.h b/include/wx/matrix.h
index d18a0d2..f24b583 100644
--- a/include/wx/matrix.h
+++ b/include/wx/matrix.h
@@ -37,11 +37,10 @@
 //  at a certain coordinate and angle within another parent picture.
 //  At all times m_isIdentity is set if the matrix itself is an Identity matrix.
 //  It is used where possible to optimize calculations.
-class
 #ifndef WXBUILDING
 wxDEPRECATED_MSG("use wxAffineMatrix2D instead")
 #endif
-WXDLLIMPEXP_CORE wxTransformMatrix: public wxObject
+WXDLLIMPEXP_CORE class wxTransformMatrix: public wxObject
 {
 public:
     wxTransformMatrix();
