--- desktop-2.5.2/src/CMakeLists.txt	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/CMakeLists.txt	2019-04-22 04:01:31.260028906 +0000
@@ -4,7 +4,7 @@
 
 set(synclib_NAME ${APPLICATION_EXECUTABLE}sync)
 
-find_package(Qt5 5.6 COMPONENTS Core Network Xml Concurrent WebEngineWidgets WebEngine REQUIRED)
+find_package(Qt5 5.6 COMPONENTS Core Network Xml Concurrent REQUIRED)
 if (Qt5Core_VERSION VERSION_LESS 5.9.0)
 message(STATUS "For HTTP/2 support, compile with Qt 5.9 or higher.")
 endif()
--- desktop-2.5.2/src/gui/CMakeLists.txt	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/gui/CMakeLists.txt	2019-04-22 04:15:53.230034980 +0000
@@ -40,7 +40,6 @@
     wizard/owncloudoauthcredspage.ui
     wizard/owncloudsetupnocredspage.ui
     wizard/owncloudwizardresultpage.ui
-    wizard/webview.ui
 )
 
 set(client_SRCS
@@ -103,8 +102,6 @@
     creds/credentialsfactory.cpp
     creds/httpcredentialsgui.cpp
     creds/oauth.cpp
-    creds/webflowcredentials.cpp
-    creds/webflowcredentialsdialog.cpp
     wizard/postfixlineedit.cpp
     wizard/abstractcredswizardpage.cpp
     wizard/owncloudadvancedsetuppage.cpp
@@ -115,8 +112,6 @@
     wizard/owncloudwizardcommon.cpp
     wizard/owncloudwizard.cpp
     wizard/owncloudwizardresultpage.cpp
-    wizard/webviewpage.cpp
-    wizard/webview.cpp
     wizard/slideshow.cpp
 )
 
@@ -295,7 +290,7 @@
 endif()
 
 add_library(updater STATIC ${updater_SRCS})
-target_link_libraries(updater ${synclib_NAME} Qt5::Widgets Qt5::Network Qt5::Xml Qt5::WebEngineWidgets)
+target_link_libraries(updater ${synclib_NAME} Qt5::Widgets Qt5::Network Qt5::Xml)
 target_include_directories(updater PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 
 set_target_properties( ${APPLICATION_EXECUTABLE} PROPERTIES
--- desktop-2.5.2/src/gui/accountmanager.cpp	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/gui/accountmanager.cpp	2019-04-22 04:17:04.440035482 +0000
@@ -253,20 +253,6 @@
         acc->setUrl(urlConfig.toUrl());
     }
 
-    // Migrate to webflow
-    if (authType == QLatin1String("http")) {
-        authType = "webflow";
-        settings.setValue(QLatin1String(authTypeC), authType);
-
-        foreach(QString key, settings.childKeys()) {
-            if (!key.startsWith("http_"))
-                continue;
-            auto newkey = QString::fromLatin1("webflow_").append(key.mid(5));
-            settings.setValue(newkey, settings.value((key)));
-            settings.remove(key);
-        }
-    }
-
     qCInfo(lcAccountManager) << "Account for" << acc->url() << "using auth type" << authType;
 
     acc->_serverVersion = settings.value(QLatin1String(serverVersionC)).toString();
--- desktop-2.5.2/src/gui/creds/credentialsfactory.cpp	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/gui/creds/credentialsfactory.cpp	2019-04-22 04:23:19.590038126 +0000
@@ -21,7 +21,6 @@
 #ifndef NO_SHIBBOLETH
 #include "creds/shibbolethcredentials.h"
 #endif
-#include "creds/webflowcredentials.h"
 
 namespace OCC {
 
@@ -40,8 +39,6 @@
         } else if (type == "shibboleth") {
             return new ShibbolethCredentials;
 #endif
-        } else if (type == "webflow") {
-            return new WebFlowCredentials;
         } else {
             qCWarning(lcGuiCredentials, "Unknown credentials type: %s", qPrintable(type));
             return new DummyCredentials;
--- desktop-2.5.2/src/gui/wizard/owncloudwizard.cpp	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/gui/wizard/owncloudwizard.cpp	2019-04-22 04:24:59.360038829 +0000
@@ -26,7 +26,6 @@
 #endif
 #include "wizard/owncloudadvancedsetuppage.h"
 #include "wizard/owncloudwizardresultpage.h"
-#include "wizard/webviewpage.h"
 
 #include "QProgressIndicator.h"
 
@@ -51,7 +50,6 @@
     , _advancedSetupPage(new OwncloudAdvancedSetupPage)
     , _resultPage(new OwncloudWizardResultPage)
     , _credentialsPage(nullptr)
-    , _webViewPage(new WebViewPage(this))
     , _setupLog()
     , _registration(false)
 {
@@ -64,7 +62,6 @@
 #endif
     setPage(WizardCommon::Page_AdvancedSetup, _advancedSetupPage);
     setPage(WizardCommon::Page_Result, _resultPage);
-    setPage(WizardCommon::Page_WebView, _webViewPage);
 
     connect(this, &QDialog::finished, this, &OwncloudWizard::basicSetupFinished);
 
@@ -79,7 +76,6 @@
 #ifndef NO_SHIBBOLETH
     connect(_shibbolethCredsPage, &OwncloudShibbolethCredsPage::connectToOCUrl, this, &OwncloudWizard::connectToOCUrl);
 #endif
-    connect(_webViewPage, &WebViewPage::connectToOCUrl, this, &OwncloudWizard::connectToOCUrl);
     connect(_advancedSetupPage, &OwncloudAdvancedSetupPage::createLocalAndRemoteFolders,
         this, &OwncloudWizard::createLocalAndRemoteFolders);
     connect(this, &QWizard::customButtonClicked, this, &OwncloudWizard::skipFolderConfiguration);
@@ -168,10 +164,6 @@
         break;
 #endif
 
-    case WizardCommon::Page_WebView:
-        _webViewPage->setConnected();
-        break;
-
     case WizardCommon::Page_AdvancedSetup:
         _advancedSetupPage->directoriesCreated();
         break;
@@ -195,8 +187,6 @@
 #endif
         if (type == DetermineAuthTypeJob::OAuth) {
         _credentialsPage = _browserCredsPage;
-    } else if (type == DetermineAuthTypeJob::WebViewFlow) {
-        _credentialsPage = _webViewPage;
     } else { // try Basic auth even for "Unknown"
         _credentialsPage = _httpCredsPage;
     }
--- desktop-2.5.2/src/libsync/networkjobs.h	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/libsync/networkjobs.h	2019-04-22 05:21:11.950062594 +0000
@@ -411,8 +411,7 @@
     enum AuthType {
         Basic, // also the catch-all fallback for backwards compatibility reasons
         OAuth,
-        Shibboleth,
-        WebViewFlow
+        Shibboleth
     };
 
     explicit DetermineAuthTypeJob(AccountPtr account, QObject *parent = nullptr);
--- desktop-2.5.2/src/libsync/networkjobs.cpp	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/libsync/networkjobs.cpp	2019-04-22 05:33:28.590067785 +0000
@@ -922,7 +922,7 @@
 
     // WebViewFlow > OAuth > Shib > Basic
     if (_account->serverVersionInt() >= Account::makeServerVersion(12, 0, 0)) {
-        result = WebViewFlow;
+        result = Basic;
     }
 
     qCInfo(lcDetermineAuthTypeJob) << "Auth type for" << _account->davUrl() << "is" << result;
--- desktop-2.5.2/src/gui/wizard/owncloudsetuppage.cpp	2019-03-19 01:52:24.000000000 +0000
+++ desktop-2.5.2/src/gui/wizard/owncloudsetuppage.cpp	2019-04-22 05:32:01.750067173 +0000
@@ -143,7 +143,7 @@
 void OwncloudSetupPage::slotGotoProviderList()
 {
     _ocWizard->setRegistration(true);
-    _ocWizard->setAuthType(DetermineAuthTypeJob::AuthType::WebViewFlow);
+    _ocWizard->setAuthType(DetermineAuthTypeJob::AuthType::Basic);
     _authTypeKnown = true;
     _checking = false;
     emit completeChanged();
@@ -260,8 +260,6 @@
         return WizardCommon::Page_OAuthCreds;
     case DetermineAuthTypeJob::Shibboleth:
         return WizardCommon::Page_ShibbolethCreds;
-    case DetermineAuthTypeJob::WebViewFlow:
-        return WizardCommon::Page_WebView;
     }
     return WizardCommon::Page_HttpCreds;
 }
--- desktop-2.5.2/src/gui/wizard/owncloudwizard.h	2019-03-18 21:52:24.000000000 -0400
+++ desktop-2.5.2/src/gui/wizard/owncloudwizard.h	2019-04-22 02:07:03.600081984 -0400
@@ -39,7 +39,6 @@
 class OwncloudWizardResultPage;
 class AbstractCredentials;
 class AbstractCredentialsWizardPage;
-class WebViewPage;
 
 /**
  * @brief The OwncloudWizard class
@@ -106,7 +105,6 @@
     OwncloudAdvancedSetupPage *_advancedSetupPage;
     OwncloudWizardResultPage *_resultPage;
     AbstractCredentialsWizardPage *_credentialsPage;
-    WebViewPage *_webViewPage;
 
     QStringList _setupLog;
 
