'-mindirect-branch-register' is not supported on AR.

--- a/tests/unit/wasm-c-api/CMakeLists.txt
+++ b/tests/unit/wasm-c-api/CMakeLists.txt
@@ -23,7 +23,11 @@
 
 # compiling and linking flags
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--gc-sections -pie -fPIE")
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wformat -Wformat-security -mindirect-branch-register")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wformat -Wformat-security")
+
+if (WAMR_BUILD_TARGET MATCHES "X86_.*" OR WAMR_BUILD_TARGET STREQUAL "AMD_64")
+  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mindirect-branch-register")
+endif()
 
 # build out vmlib
 # hard code path here
