Adjust error message for musl-based systems.

--- a/tests/test_tcp.py
+++ b/tests/test_tcp.py
@@ -221,8 +221,8 @@ class _TestTCP:
             addr = sock.getsockname()
 
             with self.assertRaisesRegex(OSError,
-                                        r"error while attempting.*\('127.*: "
-                                        r"address( already)? in use"):
+                                        r"\[Errno 98\] error while attempting.*\('127.*: "
+                                        r"(\[errno 98\] )?address in use"):
 
                 self.loop.run_until_complete(
                     self.loop.create_server(object, *addr))
