--- a/platform/linux-alpine.c
+++ b/platform/linux-alpine.c
@@ -18,19 +18,7 @@ void platform_init(void) {
 }
 
 void platform_stdinc_paths(StringArray *paths) {
-  // Replace this block with absolute path if you intend to
-  // execute the compiler outside of source directory.
-  // If you are thinking of just removing the error while keeping
-  // the relative search path, please read:
-  // https://github.com/rui314/chibicc/issues/162
-  {
-    char *hdr_dir = format("%s/slimcc_headers", dirname(strdup(argv0)));
-    if (!file_exists(hdr_dir))
-      error("can't find built-in headers");
-
-    add_include_path(paths, format("%s/include", hdr_dir));
-  }
-
+  add_include_path(paths, "/usr/lib/slimcc/include");
   add_include_path(paths, "/usr/include");
 }
 
