Apparently, musl adds a "+" in front of years >9999

--- a/test
+++ b/test
@@ -100,7 +100,7 @@ timemax='encode: STR:     ffffffff-ffff-7fff-bfff-ffffffffffff
         SIV:     340282366920937858995853114098753470463
 decode: variant: DCE 1.1, ISO/IEC 11578:1996
         version: 7 (UNIX time + random data)
-        content: time:   10889-08-02 05:31:50.655 UTC
+        content: time:   +10889-08-02 05:31:50.655 UTC
                  random: 0F:FF:3F:FF:FF:FF:FF:FF:FF:FF'
 [ "$($uuid -d ffffffff-ffff-7fff-bfff-ffffffffffff)" = "$timemax" ] || {
 	echo "uuid: -d ffffffff-ffff-7fff-bfff-ffffffffffff wrong" >&2
--- a/uuid.c
+++ b/uuid.c
@@ -556,7 +556,7 @@ static uuid_rc_t uuid_export_txt(const uuid_t *uuid, void *_data_ptr, size_t *da
     char tmp_siv[UUID_LEN_SIV+1];
     void *tmp_ptr;
     size_t tmp_len;
-    char t_buf[sizeof("10889-08-02 07:31:50")]; // The maximal representable UUID time is 0x0FFFFFFFFFFFFFFF (5236-03-31 21:21:00);
+    char t_buf[sizeof("+10889-08-02 07:31:50")]; // The maximal representable UUID time is 0x0FFFFFFFFFFFFFFF (5236-03-31 21:21:00);
                                                 // the maximal UUIDv7 UNIX time is 0xFFFFFFFFFFFF, q.v.
     struct tm *tm;
     char out_buf[512], *cur = out_buf;
