diff -u -r1.48 isakmp_inf.c
--- a/src/racoon/isakmp_inf.c	29 Aug 2012 12:01:30 -0000	1.48
+++ b/src/racoon/isakmp_inf.c	24 Jan 2013 06:46:45 -0000
@@ -492,7 +492,7 @@
 		"delete payload for protocol %s\n",
 		s_ipsecdoi_proto(delete->proto_id));
 
-	if(!iph1->rmconf->weak_phase1_check && !encrypted) {
+	if((iph1 == NULL || !iph1->rmconf->weak_phase1_check) && !encrypted) {
 		plog(LLV_WARNING, LOCATION, iph1->remote,
 			"Ignoring unencrypted delete payload "
 			"(check the weak_phase1_check option)\n");
