--- ipsec-tools-0.8.2.orig/src/racoon/cftoken.l	2014-04-15 14:47:11.000000000 -0300
+++ ipsec-tools-0.8.2/src/racoon/cftoken.l	2014-04-15 18:49:14.589422339 -0300
@@ -110,6 +110,8 @@
 static int yy_first_time = 1;
 %}
 
+%option noyywrap
+
 /* common seciton */
 nl		\n
 ws		[ \t]+
diff -ru ipsec-tools-0.8.2.orig/src/setkey/token.l ipsec-tools-0.8.2/src/setkey/token.l
--- ipsec-tools-0.8.2.orig/src/setkey/token.l	2012-08-23 14:10:46.000000000 -0300
+++ ipsec-tools-0.8.2/src/setkey/token.l	2014-04-15 18:48:11.023100041 -0300
@@ -111,7 +111,7 @@
 
 %s S_PL S_AUTHALG S_ENCALG
 
-%option noinput nounput
+%option noinput nounput noyywrap
 %%
 
 
