From 7d5caaa5c9e48ca83e1b99340ffc845b8e6a88af Mon Sep 17 00:00:00 2001
From: Brandon Boese <brandonboese@protonmail.com>
Date: Tue, 28 Nov 2023 21:01:24 +0100
Subject: [PATCH] point qmlscene to qmlscene-qt5

---
 CMakeLists.txt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7c092b80..2ae948c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,10 +86,10 @@ if(${CLICK_MODE})
   install(FILES ${CMAKE_CURRENT_BINARY_DIR}/manifest.json DESTINATION ${CMAKE_INSTALL_PREFIX})
   install(FILES "${APP_HARDCODE}.json" DESTINATION ${CMAKE_INSTALL_PREFIX})
   install(PROGRAMS lomiri-clock-app-migrate.py DESTINATION ${CMAKE_INSTALL_PREFIX})
-  set(EXEC "lomiri-clock-app-migrate.py qmlscene -I ${MODULE_PATH} %U ${LOMIRI-CLOCK_APP_DIR}/${MAIN_QML}")
+  set(EXEC "lomiri-clock-app-migrate.py qmlscene-qt5 -I ${MODULE_PATH} %U ${LOMIRI-CLOCK_APP_DIR}/${MAIN_QML}")
 else(CLICK_MODE)
   install(PROGRAMS lomiri-clock-app-migrate.py DESTINATION ${CMAKE_INSTALL_PREFIX}/${LOMIRI-CLOCK_APP_DIR})
-  set(EXEC "qmlscene $@ -I ${MODULE_PATH}  ${CMAKE_INSTALL_PREFIX}/${LOMIRI-CLOCK_APP_DIR}/${MAIN_QML}")
+  set(EXEC "qmlscene-qt5 $@ -I ${MODULE_PATH}  ${CMAKE_INSTALL_PREFIX}/${LOMIRI-CLOCK_APP_DIR}/${MAIN_QML}")
 endif()
 
 install(FILES ${URLS_FILE} DESTINATION "${CMAKE_INSTALL_DATADIR}/lomiri-url-dispatcher/urls")
@@ -124,6 +124,6 @@ add_custom_target("autopilot" chmod +x ${CMAKE_SOURCE_DIR}/tests/autopilot/run
                     DEPENDS timezone timezone-qmldir alarmsettings alarmsettings-qmldir datetime datetime-qmldir geolocation geolocation-qmldir
                     WORKING_DIRECTORY ./app)
 
-add_custom_target("run" /usr/bin/qmlscene -I ${CMAKE_BINARY_DIR}/backend  ${CMAKE_SOURCE_DIR}/app/lomiri-clock-app.qml
+add_custom_target("run" /usr/bin/qmlscene-qt5 -I ${CMAKE_BINARY_DIR}/backend  ${CMAKE_SOURCE_DIR}/app/lomiri-clock-app.qml
                     DEPENDS timezone timezone-qmldir alarmsettings alarmsettings-qmldir datetime datetime-qmldir geolocation geolocation-qmldir
                     WORKING_DIRECTORY ./app)
-- 
2.43.0

