diff --git a/surf.c b/surf.c
index 1f3d534..c4c82ce 100644
--- a/surf.c
+++ b/surf.c
@@ -273,12 +273,12 @@ externalpipe_execute(char* buffer, Arg *arg) {
 }
 
 static void
-externalpipe_resource_done(WebKitWebResource *r, GAsyncResult *s, Arg *arg)
+externalpipe_resource_done(GObject *r, GAsyncResult *s, gpointer arg)
 {
 	GError *gerr = NULL;
-	guchar *buffer = webkit_web_resource_get_data_finish(r, s, NULL, &gerr);
+	guchar *buffer = webkit_web_resource_get_data_finish(WEBKIT_WEB_RESOURCE(r), s, NULL, &gerr);
 	if (gerr == NULL) {
-		externalpipe_execute((char *) buffer, arg);
+		externalpipe_execute((char *) buffer, (Arg *)arg);
 	} else {
 		g_error_free(gerr);
 	}
@@ -286,17 +286,17 @@ externalpipe_resource_done(WebKitWebResource *r, GAsyncResult *s, Arg *arg)
 }
 
 static void
-externalpipe_js_done(WebKitWebView *wv, GAsyncResult *s, Arg *arg)
+externalpipe_js_done(GObject *wv, GAsyncResult *s, gpointer arg)
 {
 	WebKitJavascriptResult *j = webkit_web_view_run_javascript_finish(
-		wv, s, NULL);
+		WEBKIT_WEB_VIEW(wv), s, NULL);
 	if (!j) {
 		return;
 	}
 	JSCValue *v = webkit_javascript_result_get_js_value(j);
 	if (jsc_value_is_string(v)) {
 		char *buffer = jsc_value_to_string(v);
-		externalpipe_execute(buffer, arg);
+		externalpipe_execute(buffer, (Arg *)arg);
 		g_free(buffer);
 	}
 	webkit_javascript_result_unref(j);
