diff -ru libreswan-3.27.orig/programs/pluto/connections.c libreswan-3.27/programs/pluto/connections.c
--- libreswan-3.27.orig/programs/pluto/connections.c	2018-10-08 05:52:09.000000000 +0300
+++ libreswan-3.27/programs/pluto/connections.c	2018-11-12 09:38:40.643062668 +0200
@@ -41,6 +41,7 @@
 #include <arpa/inet.h>
 #include <resolv.h>
 #include <errno.h>
+#include <limits.h>
 
 #include <libreswan.h>
 #include "libreswan/pfkeyv2.h"
diff -ru libreswan-3.27.orig/programs/pluto/rcv_whack.c libreswan-3.27/programs/pluto/rcv_whack.c
--- libreswan-3.27.orig/programs/pluto/rcv_whack.c	2018-10-08 05:52:09.000000000 +0300
+++ libreswan-3.27/programs/pluto/rcv_whack.c	2018-11-12 11:00:40.133725270 +0200
@@ -36,6 +36,7 @@
 #include <arpa/inet.h>
 #include <resolv.h>
 #include <fcntl.h>
+#include <limits.h>
 #include <unistd.h>		/* for gethostname() */
 
 #include <libreswan.h>
