diff -ru libreswan-3.21.orig/programs/pluto/ikev2_ipseckey.c libreswan-3.21/programs/pluto/ikev2_ipseckey.c
--- libreswan-3.21.orig/programs/pluto/ikev2_ipseckey.c	2017-08-09 20:47:34.000000000 +0000
+++ libreswan-3.21/programs/pluto/ikev2_ipseckey.c	2017-08-14 12:00:26.380840562 +0000
@@ -23,6 +23,7 @@
 #include <stddef.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <limits.h>
 #include <errno.h>
 #include <arpa/inet.h> /* for inet_ntop */
 #include <arpa/nameser.h>
