--- rt-4.2.2/configure.ac.orig	2014-05-06 19:01:59.784199966 -0300
+++ rt-4.2.2/configure.ac	2014-05-06 19:02:21.004085165 -0300
@@ -7,7 +7,7 @@
 
 dnl Setup autoconf
 AC_PREREQ([2.53])
-AC_INIT(RT, m4_esyscmd([( git describe --tags || cat ./.tag 2> /dev/null || echo "rt-3.9.EXPORTED" )| tr -d "\n"]), [rt-bugs@bestpractical.com])
+AC_INIT(RT, m4_esyscmd([( cat ./.tag 2> /dev/null || echo "rt-3.9.EXPORTED" )| tr -d "\n"]), [rt-bugs@bestpractical.com])
 AC_CONFIG_SRCDIR([lib/RT.pm])
 
 dnl Save our incant early since $@ gets overwritten by some macros.
