--- a/configure.ac
+++ b/configure.ac
@@ -379,7 +379,7 @@
 AC_SUBST([CONFIG_FILE_PATH],		${exp_sysconfdir})
 AC_SUBST([RT_BIN_PATH],			${exp_bindir})
 AC_SUBST([RT_SBIN_PATH],		${exp_sbindir})
-AC_SUBST([RT_VAR_PATH],			${exp_localstatedir})
+AC_SUBST([RT_VAR_PATH],			${exp_datadir})
 AC_SUBST([RT_MAN_PATH],			${exp_mandir})
 AC_SUBST([RT_FONT_PATH],			${exp_fontdir})
 AC_SUBST([RT_PLUGIN_PATH],			${exp_plugindir})
@@ -431,7 +431,7 @@
 AC_SUBST([CONFIG_FILE_PATH_R],		${exp_sysconfdir})
 AC_SUBST([RT_BIN_PATH_R],			${exp_bindir})
 AC_SUBST([RT_SBIN_PATH_R],		${exp_sbindir})
-AC_SUBST([RT_VAR_PATH_R],			${exp_localstatedir})
+AC_SUBST([RT_VAR_PATH_R],			${exp_datadir})
 AC_SUBST([RT_MAN_PATH_R],			${exp_mandir})
 AC_SUBST([RT_FONT_PATH_R],			${exp_fontdir})
 AC_SUBST([MASON_DATA_PATH_R],		${exp_masonstatedir})
