diff --git a/samba/librpc/ndr/libndr_proto.h b/samba/librpc/ndr/libndr_proto.h
index f60d93d..b97df1d 100644
--- a/samba/librpc/ndr/libndr_proto.h
+++ b/samba/librpc/ndr/libndr_proto.h
@@ -9,6 +9,7 @@
 #define _PUBLIC_
 #endif
 
+typedef int (*comparison_fn_t)(const void *, const void *);
 
 /* The following definitions come from librpc/ndr/ndr.c  */
 
diff --git a/samba/lib/ldb/common/ldb_msg.c b/samba/lib/ldb/common/ldb_msg.c
index 0f7a214..7ff1583 100644
--- a/samba/lib/ldb/common/ldb_msg.c
+++ b/samba/lib/ldb/common/ldb_msg.c
@@ -35,6 +35,8 @@
 #include "includes.h"
 #include "ldb/include/includes.h"
 
+typedef int (*comparison_fn_t)(const void *, const void *);
+
 /*
   create a new ldb_message in a given memory context (NULL for top level)
 */
diff --git a/samba/lib/ldb/include/includes.h b/samba/lib/ldb/include/includes.h
index ce0d40e..c07f117 100644
--- a/samba/lib/ldb/include/includes.h
+++ b/samba/lib/ldb/include/includes.h
@@ -16,7 +16,7 @@
 #define dyn_MODULESDIR dyn_LIBDIR
 #endif
 
-
+typedef int (*comparison_fn_t)(const void *, const void *);
 
 #define discard_const(ptr) ((void *)((intptr_t)(ptr)))
 #define discard_const_p(type, ptr) ((type *)discard_const(ptr))
