See https://gitlab.alpinelinux.org/alpine/aports/-/issues/16106

diff -upr bluez-5.77.orig/tools/hex2hcd.c bluez-5.77/tools/hex2hcd.c
--- bluez-5.77.orig/tools/hex2hcd.c	2024-09-04 20:10:39.417142409 +0200
+++ bluez-5.77/tools/hex2hcd.c	2024-09-04 20:11:13.323811772 +0200
@@ -25,6 +25,10 @@
 #include <stdbool.h>
 #include <sys/stat.h>
 
+/* GNU basename version */
+#define basename(path) \
+	(strrchr((path),'/') ? strrchr((path),'/')+1 : (path))
+
 static ssize_t process_record(int fd, const char *line, uint16_t *upper_addr)
 {
 	const char *ptr = line + 1;
