diff --git a/obexd/plugins/phonebook-ebook.c b/obexd/plugins/phonebook-ebook.c
index d772edca0445..62c88f02623b 100644
--- a/obexd/plugins/phonebook-ebook.c
+++ b/obexd/plugins/phonebook-ebook.c
@@ -109,13 +109,13 @@ static char *evcard_to_string(EVCard *evcard, unsigned int format,
 	char *vcard;
 
 	if (!filter)
-		return e_vcard_to_string(evcard, format);
+		return e_vcard_convert_to_string(evcard, format);
 
 	/*
 	 * Mandatory attributes for vCard 2.1 are VERSION ,N and TEL.
 	 * Mandatory attributes for vCard 3.0 are VERSION, N, FN and TEL
 	 */
-	filter = format == EVC_FORMAT_VCARD_30 ? filter | 0x87: filter | 0x85;
+	filter = format == E_VCARD_VERSION_30 ? filter | 0x87: filter | 0x85;
 
 	l = e_vcard_get_attributes(evcard);
 	evcard2 = e_vcard_new();
@@ -140,7 +140,7 @@ static char *evcard_to_string(EVCard *evcard, unsigned int format,
 		}
 	}
 
-	vcard = e_vcard_to_string(evcard2, format);
+	vcard = e_vcard_convert_to_string(evcard2, format);
 	g_object_unref(evcard2);
 
 	return vcard;
@@ -312,10 +312,10 @@ static void phonebook_pull_read_ready(GObject *source_object,
 		char *vcard;
 
 		if (data->params->format == PB_FORMAT_VCARD30)
-			vcard = evcard_to_string(evcard, EVC_FORMAT_VCARD_30,
+			vcard = evcard_to_string(evcard, E_VCARD_VERSION_30,
 							 data->params->filter);
 		else if (data->params->format == PB_FORMAT_VCARD21)
-			vcard = evcard_to_string(evcard, EVC_FORMAT_VCARD_21,
+			vcard = evcard_to_string(evcard, E_VCARD_VERSION_21,
 							 data->params->filter);
 		else
 			error("unknown format: %d", data->params->format);
@@ -393,10 +393,10 @@ static void phonebook_get_entry_ready(GObject *source_object,
 	evcard = E_VCARD(contact);
 
 	if (data->params->format == PB_FORMAT_VCARD30)
-		vcard = evcard_to_string(evcard, EVC_FORMAT_VCARD_30,
+		vcard = evcard_to_string(evcard, E_VCARD_VERSION_30,
 							 data->params->filter);
 	else if (data->params->format == PB_FORMAT_VCARD21)
-		vcard = evcard_to_string(evcard, EVC_FORMAT_VCARD_21,
+		vcard = evcard_to_string(evcard, E_VCARD_VERSION_21,
 							 data->params->filter);
 	else
 		error("Unknown vCard format: %d", data->params->format);

