From f7701809e24ea67411fceb25d5d1f0036c59872d Mon Sep 17 00:00:00 2001
From: Marvin Preuss <marvin@xsteadfastx.org>
Date: Mon, 16 Aug 2021 12:45:19 +0200
Subject: [PATCH] moves uint64 on struct top

---
 attacker/resolver.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/attacker/resolver.go b/attacker/resolver.go
index 8281492..4993ea8 100644
--- a/attacker/resolver.go
+++ b/attacker/resolver.go
@@ -8,8 +8,8 @@ import (
 )
 
 type resolver struct {
-	addrs []string
 	idx   uint64
+	addrs []string
 }
 
 func NewResolver(addrs []string) *net.Resolver {
-- 
2.32.0

