diff -urN captagent-6.3.1.orig/configure.ac captagent-6.3.1/configure.ac
--- captagent-6.3.1.orig/configure.ac	2021-05-23 08:50:19.526922187 -0600
+++ captagent-6.3.1/configure.ac	2021-05-23 08:50:41.406989317 -0600
@@ -170,7 +170,7 @@
 AC_PROG_MAKE_SET
 AC_LIBTOOL_DLOPEN
 AC_PROG_LIBTOOL
-AC_PROG_LEX
+AC_PROG_LEX([noyywrap])
 
 if test "x$LEXLIB" = "x"; then
        AC_MSG_ERROR([captagent requires but cannot find libfl])
