diff --git a/config/redis-openvas.conf b/config/redis-openvas.conf
index eb9e930..6502f04 100644
--- a/config/redis-openvas.conf
+++ b/config/redis-openvas.conf
@@ -260,7 +260,7 @@ dbfilename dump.rdb
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/redis

 ################################# REPLICATION #################################
