--- a/configure.in	2004-06-30 22:11:08.000000000 -0700
+++ b/configure.in	2004-08-22 16:45:36.279473008 -0700
@@ -463,16 +463,16 @@
   if test ! -d $vpopmaildir
   then
     AC_MSG_WARN( creating $vpopmaildir )
-    mkdir -p $vpopmaildir
-    chown $vpopmailuid $vpopmaildir
-    chgrp $vpopmailgid $vpopmaildir
-    chmod 755 $vpopmaildir
+    mkdir -p "$pkgdir"/$vpopmaildir
+    chown $vpopmailuid "$pkgdir"/$vpopmaildir
+    chgrp $vpopmailgid "$pkgdir"/$vpopmaildir
+    chmod 755 "$pkgdir"/$vpopmaildir
   fi
   AC_MSG_WARN( making a vpopmail etc directory )
-  mkdir $vpopmaildir/etc
-  chown $vpopmailuid "$vpopmaildir"/etc
-  chgrp $vpopmailgid "$vpopmaildir"/etc
-  chmod 755 "$vpopmaildir"/etc
+  mkdir "$pkgdir"/$vpopmaildir/etc
+  chown $vpopmailuid "$pkgdir"/$vpopmaildir/etc
+  chgrp $vpopmailgid "$pkgdir"/$vpopmaildir/etc
+  chmod 755 "$pkgdir"/$vpopmaildir/etc
 fi
 
 #----------------------------------------------------------------------
