--- ./configure.ac.orig
+++ ./configure.ac
@@ -723,7 +723,7 @@
 else
   dnl preload c-client default locations/options
   case $host in
-    *-linux-gnu*|*-k*bsd*-gnu*|*-gnu*)
+    *-linux-gnu*|*-k*bsd*-gnu*|*-gnu*|*-linux-uclibc|*-linux-musl*)
       if test -f /etc/fedora-release -o -f /etc/redhat-release -o -f /etc/redhat_version ; then  
         alpine_SSLTYPE="nopwd"
 	if test -d /etc/pki/tls ; then  
@@ -1059,7 +1059,7 @@
 dnl build from c-client.  Most of this will go away when c-client
 dnl adopts configure
 case "$host" in
-  *-linux-gnu*|*-k*bsd*-gnu*|*-gnu*)
+  *-linux-gnu*|*-k*bsd*-gnu*|*-gnu*|*-linux-uclibc|*-linux-musl*)
     alpine_path_delim="/"
     alpine_mode_readonly="(0600)"
     if test -f /etc/fedora-release ; then  
