diff --git a/basic.cfg b/basic.cfg
index d8e4336..7cd0050 100644
--- a/basic.cfg
+++ b/basic.cfg
@@ -2,20 +2,17 @@
 # sensible values for "standard" setup.
 
 verbose: 0;
-foreground: false;
-inetd: false;
 numeric: false;
 transparent: false;
 timeout: 2;
 user: "nobody";
-pidfile: "/var/run/sslh.pid";
 chroot: "/var/empty";
 
 
 # Change hostname with your external address name.
 listen:
 (
-    { host: "thelonious"; port: "443"; }
+	{ host: "0.0.0.0"; port: "443"; }
 );
 
 protocols:
