diff --git a/cmd/alps/main.go b/cmd/alps/main.go
index 5a00957..f537177 100644
--- a/cmd/alps/main.go
+++ b/cmd/alps/main.go
@@ -23,7 +23,7 @@ import (
 	_ "git.sr.ht/~migadu/alps/plugins/viewtext"
 )
 
-var themesPath = "./themes"
+var themesPath = "/usr/lib/alps/themes"
 
 func main() {
 	var (
diff --git a/plugin.go b/plugin.go
index e634f02..43b8164 100644
--- a/plugin.go
+++ b/plugin.go
@@ -7,7 +7,7 @@ import (
 )
 
 // PluginDir is the path to the plugins directory.
-const PluginDir = "plugins"
+const PluginDir = "/usr/lib/alps/plugins"
 
 // Plugin extends alps with additional functionality.
 type Plugin interface {
