diff --unified --recursive --text imageio-ffmpeg-0.4.3.orig/setup.py imageio-ffmpeg-0.4.3.new/setup.py
--- a/setup.py	2021-01-04 17:18:12.000000000 +0100
+++ b/setup.py	2021-02-13 09:51:58.744209490 +0100
@@ -57,7 +57,7 @@
     platforms="any",
     provides=["imageio_ffmpeg"],
     python_requires=">=3.4",
-    setup_requires=["pip>19"],
+    setup_requires=[],
     install_requires=[],
     packages=["imageio_ffmpeg"],
     package_dir={"imageio_ffmpeg": "imageio_ffmpeg"},
