diff --unified --recursive --text imageio-ffmpeg-0.4.3.orig/setup.py imageio-ffmpeg-0.4.3.new/setup.py
diff --git a/setup.py b/setup.py
index cfd5aac..beea984 100644
--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
     platforms="any",
     provides=["imageio_ffmpeg"],
     python_requires=">=3.5",
-    setup_requires=["pip>19"],
+    setup_requires=[],
     install_requires=[],
     packages=["imageio_ffmpeg"],
     package_dir={"imageio_ffmpeg": "imageio_ffmpeg"},
