From 624c264934d14048009113d14c3b4116e687b224 Mon Sep 17 00:00:00 2001
From: techknowlogick <techknowlogick@gitea.io>
Date: Mon, 7 Jun 2021 16:40:44 -0400
Subject: [PATCH] skip pipe tests in e2e

Signed-off-by: techknowlogick <techknowlogick@gitea.io>
---
 test/e2e/e2e_test.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/e2e/e2e_test.go b/test/e2e/e2e_test.go
index ef4222b..c9a6d58 100644
--- a/test/e2e/e2e_test.go
+++ b/test/e2e/e2e_test.go
@@ -152,6 +152,7 @@ func TestJsonOutput(t *testing.T) {
 }

 func TestPipes(t *testing.T) {
+	t.Skip("Skip these tests")
 	if runtime.GOOS == "windows" {
 		t.Skip("Skipping test as it applies to linux based OS")
 	}
--
2.31.1
