diff --git a/speedtest.py b/speedtest.py
index 92a2be0b93..23d8ddbe4e 100755
--- a/speedtest.py
+++ b/speedtest.py
@@ -1171,7 +1171,7 @@ class Speedtest(object):
             client = get_attributes_by_tag_name(root, 'client')
 
         ignore_servers = list(
-            map(int, server_config['ignoreids'].split(','))
+            map(int, server_config['ignoreids'].split(',') if len(server_config['ignoreids']) else []) 
         )
 
         ratio = int(upload['ratio'])
