diff --git a/configure b/configure
index 6e62da8..54c72cd 100755
--- a/configure
+++ b/configure
@@ -24506,11 +24506,11 @@ if test -n "$BABL_CFLAGS"; then
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
     { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"babl >= 0.1.78\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "babl >= 0.1.78") 2>&5
+  ($PKG_CONFIG --exists --print-errors "babl-0.1 >= 0.1.78") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_BABL_CFLAGS=`$PKG_CONFIG --cflags "babl >= 0.1.78" 2>/dev/null`
+  pkg_cv_BABL_CFLAGS=`$PKG_CONFIG --cflags "babl-0.1 >= 0.1.78" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -24522,12 +24522,12 @@ if test -n "$BABL_LIBS"; then
     pkg_cv_BABL_LIBS="$BABL_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"babl >= 0.1.78\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "babl >= 0.1.78") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"babl-0.1 >= 0.1.78\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "babl-0.1 >= 0.1.78") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_BABL_LIBS=`$PKG_CONFIG --libs "babl >= 0.1.78" 2>/dev/null`
+  pkg_cv_BABL_LIBS=`$PKG_CONFIG --libs "babl-0.1 >= 0.1.78" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -24548,9 +24548,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        BABL_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "babl >= 0.1.78" 2>&1`
+	        BABL_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "babl-0.1 >= 0.1.78" 2>&1`
         else
-	        BABL_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "babl >= 0.1.78" 2>&1`
+	        BABL_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "babl-0.1 >= 0.1.78" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$BABL_PKG_ERRORS" >&5
