--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -13,7 +13,8 @@
 chilli.h options.h util.c garden.c garden.h ssl.h ssl.c \
 dns.c dns.h session.c session.h limits.h pkt.h pkt.c \
 chksum.c net.h net.c ms_chap.c options.c statusfile.c \
-conn.h conn.c 
+conn.h conn.c
+libchilli_la_LIBADD = $(LIBOPENSSL)
 
 AM_CFLAGS = -D_GNU_SOURCE -Wall -fno-builtin -fno-strict-aliasing \
   -O2 -fomit-frame-pointer -funroll-loops -pipe \
@@ -33,7 +34,7 @@
 #test_radius_SOURCES = test-radius.c 
 #test_dhcp_SOURCES = test-dhcp.c 
 
-LDADD = libchilli.la $(top_builddir)/bstring/libbstring.la ${LIBRT}
+LDADD = libchilli.la $(top_builddir)/bstring/libbstring.la $(LIBOPENSSL)
 
 if WITH_OPENSSL
 LDADD += ${LIBSSL}
