--- ./src/system.h.orig	2014-05-01 13:22:35.122472244 -0200
+++ ./src/system.h	2014-05-01 13:23:19.896392972 -0200
@@ -121,18 +121,6 @@
 #include <signal.h>
 #endif
 
-#if defined(__linux__)
-#include <asm/types.h>
-#include <linux/if.h>
-#include <linux/if_packet.h>
-#include <linux/if_ether.h>
-#include <linux/if_tun.h>
-#include <linux/netlink.h>
-#include <linux/rtnetlink.h>
-#ifndef HAVE_SYS_UN_H
-#include <linux/un.h>
-#endif
-
 #elif defined (__FreeBSD__)  || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__) 
 #include <net/if.h>
 #include <net/bpf.h>
@@ -256,6 +244,18 @@
 #include <unistd.h>
 #include <errno.h>
 
+#if defined(__linux__)
+#include <asm/types.h>
+#include <linux/if.h>
+#include <linux/if_packet.h>
+#include <linux/if_ether.h>
+#include <linux/if_tun.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#ifndef HAVE_SYS_UN_H
+#include <linux/un.h>
+#endif
+
 #define safe_snprintf portable_snprintf
 char *safe_strncpy(char *dst, const char *src, size_t size);
 
