diff --git a/examples/cppunittest/OrthodoxTest.h b/examples/cppunittest/OrthodoxTest.h
index 8fc2a08..a2bc678 100644
--- a/examples/cppunittest/OrthodoxTest.h
+++ b/examples/cppunittest/OrthodoxTest.h
@@ -38,6 +38,8 @@ private:
   public:
     Value( int value =0 ) : m_value( value ) {}
 
+    Value( const Value & ) = default;
+
     Value& operator= ( const Value& v )
     {
       m_value = v.m_value;
@@ -143,6 +145,8 @@ private:
     {
       return ValueBadCall( -1 - m_value );
     }
+
+    ValueBadCall &operator =( const ValueBadCall & ) = default;
   };
 
 
@@ -161,6 +165,8 @@ private:
       ++m_value;
       return *this;
     }
+
+    ValueBadAssignment( const ValueBadAssignment & ) = default;
   };
 
