diff --git a/Common/Data/Format/IniFile.h b/Common/Data/Format/IniFile.h
index 6499562..a0b6ef2 100644
--- a/Common/Data/Format/IniFile.h
+++ b/Common/Data/Format/IniFile.h
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <istream>
 #include <map>
 #include <string>
diff --git a/Common/GPU/OpenGL/GLFeatures.h b/Common/GPU/OpenGL/GLFeatures.h
index 6a8f15b..63ff9c0 100644
--- a/Common/GPU/OpenGL/GLFeatures.h
+++ b/Common/GPU/OpenGL/GLFeatures.h
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 
 // TODO: Replace with thin3d's vendor enum.
diff --git a/Common/Net/NetBuffer.h b/Common/Net/NetBuffer.h
index 6247aca..19f24b3 100644
--- a/Common/Net/NetBuffer.h
+++ b/Common/Net/NetBuffer.h
@@ -1,5 +1,6 @@
 ﻿#pragma once
 
+#include <cstdint>
 #include "Common/Buffer.h"
 
 namespace net {
diff --git a/Common/StringUtils.h b/Common/StringUtils.h
index 89127be..de8c360 100644
--- a/Common/StringUtils.h
+++ b/Common/StringUtils.h
@@ -19,6 +19,7 @@
 
 #include "Common.h"
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/Core/Reporting.h b/Core/Reporting.h
index a1b04ab..77e5847 100644
--- a/Core/Reporting.h
+++ b/Core/Reporting.h
@@ -17,6 +17,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/GPU/Common/IndexGenerator.cpp b/GPU/Common/IndexGenerator.cpp
index 446d7a6..f3fb34c 100644
--- a/GPU/Common/IndexGenerator.cpp
+++ b/GPU/Common/IndexGenerator.cpp
@@ -17,6 +17,7 @@
 
 #include <cstring>
 
+#include <cstdint>
 #include "ppsspp_config.h"
 #include "CPUDetect.h"
 #include "Common.h"
