diff --git a/sopel/modules/translate.py b/sopel/modules/translate.py
index 4084827093..1e69709d4c 100644
--- a/sopel/modules/translate.py
+++ b/sopel/modules/translate.py
@@ -105,7 +105,7 @@ def tr(bot, trigger):
 
 @commands('translate', 'tr')
 @example('.tr :en :fr my dog', '"mon chien" (en to fr, translate.google.com)')
-@example('.tr היי', '"Hey" (iw to en, translate.google.com)')
+@example('.tr היי', '"Hi" (iw to en, translate.google.com)')
 @example('.tr mon chien', '"my dog" (fr to en, translate.google.com)')
 def tr2(bot, trigger):
     """Translates a phrase, with an optional language hint."""
diff --git a/sopel/modules/search.py b/sopel/modules/search.py
index 783cc7278f..833304d65b 100644
--- a/sopel/modules/search.py
+++ b/sopel/modules/search.py
@@ -94,7 +94,7 @@ def duck_api(query):
 # test for bad Unicode handling in py2
 @example('.duck grandorder.wiki chulainn alter', 'https://grandorder.wiki/Cú_Chulainn_(Alter)')
 # the last example is what .help displays
-@example('.duck sopel irc bot', r'https?:\/\/sopel\.chat\/?', re=True)
+@example('.duck sopel irc bot', r'https?:\/\/(sopel\.chat|github\.com\/sopel-irc\/sopel)\/?', re=True)
 def duck(bot, trigger):
     """Queries Duck Duck Go for the specified input."""
     query = trigger.group(2)
