diff --git a/sopel/modules/help.py b/sopel/modules/help.py
index 5c3811ff74..341b6af8a1 100644
--- a/sopel/modules/help.py
+++ b/sopel/modules/help.py
@@ -84,13 +84,13 @@ def create_list(bot, msg):
     msg = 'Command listing for {}@{}\n\n'.format(bot.nick, bot.config.core.host) + msg

     try:
-        result = requests.post('https://clbin.com/', data={'clbin': msg})
+        result = requests.post('https://tpaste.us/', data={'tpaste': msg})
     except requests.RequestException:
         bot.say("Sorry! Something went wrong.")
         logger.exception("Error posting commands")
         return
     result = result.text
-    if "https://clbin.com/" in result:
+    if "://tpaste.us/" in result:
         return result
     else:
         bot.say("Sorry! Something went wrong.")

