diff --git a/lib/libimhex/source/helpers/stacktrace.cpp b/lib/libimhex/source/helpers/stacktrace.cpp
index b6ee42a..bfc5826 100644
--- a/lib/libimhex/source/helpers/stacktrace.cpp
+++ b/lib/libimhex/source/helpers/stacktrace.cpp
@@ -94,7 +94,8 @@
 
     }
 
-#elif defined(HEX_HAS_EXECINFO) && __has_include(BACKTRACE_HEADER)
+#elif defined(HEX_HAS_EXECINFO)
+    #if __has_include(BACKTRACE_HEADER)
 
     #include BACKTRACE_HEADER
     #include <llvm/Demangle/Demangle.h>
@@ -121,8 +122,10 @@
         }
 
     }
+    #endif
 
-#elif defined(HEX_HAS_BACKTRACE) && __has_include(BACKTRACE_HEADER)
+#elif defined(HEX_HAS_BACKTRACE) 
+    #if __has_include(BACKTRACE_HEADER)
 
     #include BACKTRACE_HEADER
     #include <llvm/Demangle/Demangle.h>
@@ -163,6 +166,7 @@
         }
 
     }
+    #endif
 
 #else
 
