--- flawfinder-2.0.7/makefile
+++ flawfinder-2.0.7-use-python3/makefile
@@ -12,8 +12,7 @@
 ARCH=noarch
 
 SAMPLE_DIR=/usr/src/linux-2.2.16
-PYTHON=python
-PYTHON2=python2
+PYTHON=python3
 PYTHON3=python3
 
 # Flawfinder has traditionally used INSTALL_DIR, INSTALL_DIR_BIN, and
@@ -190,9 +189,6 @@
 
 # Usual check routine. Run all tests using *both* python2 and python3.
 check:
-	@echo "Testing with $(PYTHON2)"
-	@PYTHON="$(PYTHON2)" $(MAKE) test
-	@echo
 	@echo "Testing with $(PYTHON3)"
 	@PYTHON="$(PYTHON3)" $(MAKE) test
 
