diff --git a/src/modules/engines/software_x11/Makefile.am b/src/modules/engines/software_x11/Makefile.am
index 8c0961d..edab474 100644
--- a/src/modules/engines/software_x11/Makefile.am
+++ b/src/modules/engines/software_x11/Makefile.am
@@ -40,6 +40,7 @@ AM_CPPFLAGS = \
 @FREETYPE_CFLAGS@ \
 @PIXMAN_CFLAGS@ \
 @EVAS_GENERAL_CFLAGS@ \
+@FRIBIDI_CFLAGS@ \
 @evas_engine_software_xcb_cflags@
 
 SOFTWARE_X11_SOURCES += \
