commit 6d05d3d02fac411080363665e1c4fc7a2c84f7aa
Author: Isaac Dunham <ibid.ag@gmail.com>
Date:   Tue Jul 15 17:11:04 2014 -0700

    Add sys/select.h to headers

diff --git a/src/daemon/old_main.c b/src/daemon/old_main.c
index 9cd4e56..0ea0b84 100644
--- a/src/daemon/old_main.c
+++ b/src/daemon/old_main.c
@@ -26,6 +26,7 @@
 #include <errno.h>                  /* guess again       */
 #include <unistd.h>                 /* unlink            */
 #include <sys/stat.h>               /* chmod             */
+#include <sys/select.h>             /* fd_set, FD_ZERO   */
 
 #include <linux/kd.h>               /* linux hd*         */
 
diff --git a/src/daemon/processmouse.c b/src/daemon/processmouse.c
index b3f3994..af3400a 100644
--- a/src/daemon/processmouse.c
+++ b/src/daemon/processmouse.c
@@ -30,6 +30,7 @@
 #include <unistd.h>                 /* close             */
 #include <time.h>                   /* time              */
 #include <sys/time.h>               /* gettimeofday      */
+#include <sys/select.h>             /* FD_ZERO           */
 
 #include "headers/message.h"        /* messaging in gpm */
 #include "headers/daemon.h"         /* daemon internals */
diff --git a/src/lib/libcurses.c b/src/lib/libcurses.c
index e7ebf84..e6fa877 100644
--- a/src/lib/libcurses.c
+++ b/src/lib/libcurses.c
@@ -26,6 +26,7 @@
 #include <unistd.h>        /* select(); */
 #include <sys/time.h>      /* timeval */
 #include <sys/types.h>     /* socket() */
+#include <sys/select.h>    /* FD_ZERO  */
 
 #include "headers/gpmInt.h"
 
diff --git a/src/lib/liblow.c b/src/lib/liblow.c
index e789d09..e449321 100644
--- a/src/lib/liblow.c
+++ b/src/lib/liblow.c
@@ -35,6 +35,7 @@
 #include <sys/un.h>        /* struct sockaddr_un */
 #include <sys/fcntl.h>     /* O_RDONLY */
 #include <sys/stat.h>      /* stat() */
+#include <sys/select.h>    /* FD_ZERO */
 
 #ifdef  SIGTSTP         /* true if BSD system */
 #include <sys/file.h>
diff --git a/src/lib/libxtra.c b/src/lib/libxtra.c
index e6dd4aa..b7ec631 100644
--- a/src/lib/libxtra.c
+++ b/src/lib/libxtra.c
@@ -25,6 +25,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/time.h>
+#include <sys/select.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <string.h>
diff --git a/src/mice.c b/src/mice.c
index 6cbd491..6989f92 100644
--- a/src/mice.c
+++ b/src/mice.c
@@ -55,6 +55,7 @@
 #include <sys/types.h>
 #include <sys/stat.h> /* stat() */
 #include <sys/time.h> /* select() */
+#include <sys/select.h> /* select() */
 
 #include <linux/kdev_t.h> /* MAJOR */
 #include <linux/keyboard.h>
diff --git a/src/prog/display-buttons.c b/src/prog/display-buttons.c
index de8e5b2..2385bf0 100644
--- a/src/prog/display-buttons.c
+++ b/src/prog/display-buttons.c
@@ -37,6 +37,7 @@
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
 #include <gpm.h>              /* gpm information      */
+#include <sys/select.h>       /* FD_ZERO              */
 
 /* display resulting data */
 int display_data(Gpm_Event *event, void *data)
diff --git a/src/prog/display-coords.c b/src/prog/display-coords.c
index ed15c8a..5aa7228 100644
--- a/src/prog/display-coords.c
+++ b/src/prog/display-coords.c
@@ -37,6 +37,7 @@
 #include <stdio.h>            /* printf()             */
 #include <time.h>             /* time()               */
 #include <errno.h>            /* errno                */
+#include <sys/select.h>       /* FD_ZERO()            */
 #include <gpm.h>              /* gpm information      */
 
 /* display resulting data */
diff --git a/src/prog/mev.c b/src/prog/mev.c
index 97622d4..22cfb05 100644
--- a/src/prog/mev.c
+++ b/src/prog/mev.c
@@ -36,6 +36,7 @@
 #include <signal.h>
 #include <sys/types.h>
 #include <sys/time.h>
+#include <sys/select.h>
 #include <unistd.h>
 #include <termios.h>
 #include <errno.h>
diff --git a/src/prog/mouse-test.c b/src/prog/mouse-test.c
index 0bb1982..9a1b624 100644
--- a/src/prog/mouse-test.c
+++ b/src/prog/mouse-test.c
@@ -37,6 +37,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <sys/select.h>
 #include <fcntl.h>
 #include <time.h>
 #include <signal.h>
diff --git a/src/synaptics.c b/src/synaptics.c
index 2418c75..ea40a53 100644
--- a/src/synaptics.c
+++ b/src/synaptics.c
@@ -216,6 +216,7 @@
 #include <string.h>
 #include <sys/time.h>
 #include <sys/ioctl.h>
+#include <sys/select.h>
 
 #include "headers/gpm.h"
 #include "headers/daemon.h"       /* which_mouse */
