On 32-bit systems ReadXRefSubsection errors earlier with a different error

--- a/test/unit/ParserTest.cpp
+++ b/test/unit/ParserTest.cpp
@@ -809,7 +809,7 @@ void PdfParserTest::testReadXRefSubsection()
     }
     catch (PdfError& error)
     {
-        REQUIRE(error.GetCode() == PdfErrorCode::InvalidXRef);
+        REQUIRE((error.GetCode() == PdfErrorCode::InvalidXRef || error.GetCode() == PdfErrorCode::ValueOutOfRange));
     }
     catch (exception&)
     {
@@ -954,7 +954,7 @@ void PdfParserTest::testReadXRefSubsection()
     }
     catch (PdfError& error)
     {
-        REQUIRE(error.GetCode() == PdfErrorCode::InvalidXRef);
+        REQUIRE((error.GetCode() == PdfErrorCode::InvalidXRef || error.GetCode() == PdfErrorCode::ValueOutOfRange));
     }
     catch (exception&)
     {
