--- ./lib/POSIX/strftime/Compiler.pm.orig
+++ ./lib/POSIX/strftime/Compiler.pm
@@ -238,6 +238,8 @@
     'c' => [q!$weekday_abbr[$_[WDAY]] . ' ' . $month_abbr[$_[MONTH]] . ' ' . substr(' '.$_[DAY],-2) . ' %H:%M:%S %Y'!,1],
     'C' => [q!substr('0'.int(($_[YEAR]+1900)/100), -2)!],  #century
     'h' => [q!$month_abbr[$_[MONTH]]!],
+    'k' => [q!substr(' '.$_[HOUR],-2)!],
+    'l' => [q!substr(' '.($_[HOUR]%12 || 1),-2)!],
     'N' => [q!substr(sprintf('%.9f', $_[SEC] - int $_[SEC]), 2)!],
     'n' => [q!"\n"!],
     'p' => [q!$_[HOUR] > 0 && $_[HOUR] < 13 ? "AM" : "PM"!],
@@ -261,8 +263,6 @@
         'V' => [q!substr('0'.isoweeknum(@_),-2)!,1],
         'e' => [q!substr(' '.$_[DAY],-2)!],
         'g' => [q!substr('0'.isoyearnum(@_)%100,-2)!,1],
-        'k' => [q!substr(' '.$_[HOUR],-2)!],
-        'l' => [q!substr(' '.($_[HOUR]%12 || 1),-2)!],
         's' => [q!int(Time::Local::timegm(@_))!,1],
         'u' => [q!$_[WDAY] || 7!,1],
         'z' => [q!tzoffset(@_)!,1],
