From 92dd35b9a15b5feae533e48310fdc1136b5a46c5 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Sat, 18 Dec 2021 12:26:59 +0100
Subject: [PATCH 2/2] fix strtol

---
 src/utils/NumberUtils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/utils/NumberUtils.h b/src/utils/NumberUtils.h
index d0a7471..e3967b2 100644
--- a/src/utils/NumberUtils.h
+++ b/src/utils/NumberUtils.h
@@ -140,7 +140,7 @@ really_inline from_chars_result from_chars(const char *first, const char *last,
 #ifdef _WIN32
     tempval = _strtol_l(first, &endptr, 0, loc.local);
 #else
-    tempval = ::strtol_l(first, &endptr, 0, loc.local);
+    tempval = ::strtol(first, &endptr, 0);
 #endif
 
     if (errno != 0)
-- 
2.34.1

