--- a/share/cmake/modules/Findyaml-cpp.cmake
+++ b/share/cmake/modules/Findyaml-cpp.cmake
@@ -39,11 +39,11 @@
 
     if(NOT DEFINED yaml-cpp_ROOT)
         # Search for yaml-cpp-config.cmake
-        find_package(yaml-cpp ${yaml-cpp_FIND_VERSION} CONFIG QUIET)
+        find_package(yaml-cpp::yaml-cpp ${yaml-cpp_FIND_VERSION} CONFIG QUIET)
     endif()
 
     if(yaml-cpp_FOUND)
-        get_target_property(yaml-cpp_LIBRARY yaml-cpp LOCATION)
+        get_target_property(yaml-cpp_LIBRARY yaml-cpp::yaml-cpp LOCATION)
     else()
 
         # As yaml-cpp-config.cmake search fails, search an installed library
--- a/src/cmake/Config.cmake.in
+++ b/src/cmake/Config.cmake.in
@@ -46,7 +46,7 @@
         find_dependency(pystring @pystring_VERSION@)
     endif()
 
-    if (NOT yaml-cpp)
+    if (NOT yaml-cpp::yaml-cpp)
         find_dependency(yaml-cpp @yaml-cpp_VERSION@)
     endif()
 
