From 23e10c9755265b7c3df926155a1549a750d37de8 Mon Sep 17 00:00:00 2001
From: "Sicelo A. Mhlongo" <absicsz@gmail.com>
Date: Wed, 29 Nov 2023 08:54:17 +0200
Subject: [PATCH 2/2] isimodem: fix typo

---
 drivers/isimodem/radio-settings.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/isimodem/radio-settings.c b/drivers/isimodem/radio-settings.c
index 4e51b980..0875d715 100644
--- a/drivers/isimodem/radio-settings.c
+++ b/drivers/isimodem/radio-settings.c
@@ -371,4 +371,4 @@ static const struct ofono_radio_settings_driver driver = {
 	.set_fast_dormancy	= isi_set_fast_dormancy,
 };
 
-OFONO_ATOM_DRIVER_BUILTIN(radio_settings, isidriver, &driver)
+OFONO_ATOM_DRIVER_BUILTIN(radio_settings, isimodem, &driver)
-- 
2.43.0

