diff --git a/autotests/directoryIteratorTest.cpp b/autotests/directoryIteratorTest.cpp
index 4f10371..b0b2847 100644
--- a/autotests/directoryIteratorTest.cpp
+++ b/autotests/directoryIteratorTest.cpp
@@ -1,6 +1,10 @@
 // SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 // SPDX-FileCopyrightText: 2022 Harald Sitter <sitter@kde.org>
 
+#ifndef S_BLKSIZE
+#define S_BLKSIZE 512
+#endif
+
 #include <QDebug>
 #include <QFileInfo>
 #include <QTest>
diff --git a/src/posixWalker.cpp b/src/posixWalker.cpp
index 246a04a..5adfe25 100644
--- a/src/posixWalker.cpp
+++ b/src/posixWalker.cpp
@@ -1,6 +1,10 @@
 // SPDX-License-Identifier: GPL-2.0-only OR GPL-3.0-only OR LicenseRef-KDE-Accepted-GPL
 // SPDX-FileCopyrightText: 2022 Harald Sitter <sitter@kde.org>
 
+#ifndef S_BLKSIZE
+#define S_BLKSIZE 512
+#endif
+
 #include "posixWalker.h"
 
 static void outputError(const QByteArray &path)
