--- ./src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp.orig
+++ ./src/VBox/Runtime/r3/linux/thread-affinity-linux.cpp
@@ -32,6 +32,8 @@
 # define _GNU_SOURCE
 #endif
 #include <features.h>
+
+#if defined(__GLIBC__)
 #if __GLIBC_PREREQ(2,4)
 
 #include <sched.h>
@@ -87,6 +89,11 @@
 
     return VINF_SUCCESS;
 }
+
+#else
+# include "../../generic/RTThreadGetAffinity-stub-generic.cpp"
+# include "../../generic/RTThreadSetAffinity-stub-generic.cpp"
+#endif
 
 #else
 # include "../../generic/RTThreadGetAffinity-stub-generic.cpp"
