diff --git a/sublime_music/ui/state.py b/sublime_music/ui/state.py
index 90c601a..ba3962f 100644
--- a/sublime_music/ui/state.py
+++ b/sublime_music/ui/state.py
@@ -38,6 +38,10 @@ class RepeatType(Enum):
         }[loop_status]
 
 
+class _DefaultGenre(Genre):
+    def __init__(self):
+        self.name = "Rock"
+
 @dataclass
 class UIState:
     """Represents the UI state of the application."""
@@ -83,15 +87,11 @@ class UIState:
     loading_play_queue: bool = False
 
     # State for Album sort.
-    class _DefaultGenre(Genre):
-        def __init__(self):
-            self.name = "Rock"
-
-    current_album_search_query: AlbumSearchQuery = AlbumSearchQuery(
+    current_album_search_query: AlbumSearchQuery = field(default_factory=lambda: AlbumSearchQuery(
         AlbumSearchQuery.Type.RANDOM,
         genre=_DefaultGenre(),
         year_range=this_decade(),
-    )
+    ))
 
     active_playlist_id: Optional[str] = None
 
