--- a/repmgr.conf.sample
+++ b/repmgr.conf.sample
@@ -78,7 +78,7 @@
 
 # stderr can be redirected to an arbitrary file:
 #
-#logfile='/var/log/repmgr/repmgr.log'
+logfile='/var/log/postgresql/repmgr.log'
 
 # event notifications can be passed to an arbitrary external program
 # together with the following parameters:
@@ -121,19 +121,17 @@
 # If you use sudo, the user repmgr runs as (usually 'postgres')  must have
 # passwordless sudo access to execute the command
 #
-# For example, to use systemd, you may use the following configuration:
+# For example, to use OpenRC, you may use the following configuration:
 #
 #    # this is required when running sudo over ssh without -t:
 #    Defaults:postgres !requiretty
-#    postgres ALL = NOPASSWD: /usr/bin/systemctl stop postgresql-9.5, \
-#       /usr/bin/systemctl start postgresql-9.5, \
-#       /usr/bin/systemctl restart postgresql-9.5
+#    postgres ALL = NOPASSWD: /etc/init.d/postgresql
 #
-# service_start_command = systemctl start postgresql-9.5
-# service_stop_command = systemctl stop postgresql-9.5
-# service_restart_command = systemctl restart postgresql-9.5
-# service_reload_command = pg_ctlcluster 9.5 main reload
-# service_promote_command = pg_ctlcluster 9.5 main promote
+service_start_command = /etc/init.d/postgresql start
+service_stop_command = /etc/init.d/postgresql stop
+service_restart_command = /etc/init.d/postgresql restart
+# service_reload_command = pg_ctl reload
+# service_promote_command = pg_ctl promote
 
 # external command options
 
