diff --git a/Makefile b/Makefile
index d14359d..3e00d71 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-OFLAGS=-O2 -march=native
+OFLAGS=-O2
 WFLAGS=-Wall -Wpedantic -Wextra -Wconversion -Wstrict-prototypes -Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=int-conversion
 CFLAGS=`pkg-config --cflags libcurl yajl mpv`
 LFLAGS=`pkg-config --libs libcurl yajl mpv` -pthread

