"listen on localhost" doesn't work on Linux

diff --git a/usr.sbin/smtpd/smtpd.conf b/usr.sbin/smtpd/smtpd.conf
index 72d985a..7281fff 100644
--- a/usr.sbin/smtpd/smtpd.conf
+++ b/usr.sbin/smtpd/smtpd.conf
@@ -1,16 +1,14 @@
-#	$OpenBSD: smtpd.conf,v 1.10 2018/05/24 11:40:17 gilles Exp $
-
 # This is the smtpd server system-wide configuration file.
 # See smtpd.conf(5) for more information.
 
-table aliases file:/etc/mail/aliases
+table aliases file:/etc/smtpd/aliases
 
 # To also accept external mail over IPv4 or IPv6,
 # respectively replace "listen on localhost" with:
 #
 # listen on 0.0.0.0
 # listen on ::
-listen on localhost
+listen on lo
 
 action "local" maildir alias <aliases>
 action "relay" relay
