--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = include testsuite man
+SUBDIRS = include testsuite
 if BUILD_DOCS
 ## This hack is needed because it doesn't seem possible to make a
 ## conditional info_TEXINFOS in Automake.  At least Automake 1.14
@@ -12,6 +12,7 @@
 ## straightforward way -- or simply unconditionally tries to build the
 ## info file.
 SUBDIRS += doc
+SUBDIRS += man
 endif
 
 EXTRA_DIST = LICENSE ChangeLog.old					\
