--- sitaramc-gitolite-b552f41.orig/src/triggers/post-compile/update-gitweb-access-list
+++ sitaramc-gitolite-b552f41/src/triggers/post-compile/update-gitweb-access-list
@@ -23,7 +23,7 @@
 # since mktemp does not honor umask, we just use it to generate a temp
 # filename (note: 'mktemp -u' on some systems, this gets close enough)
 tmpfile=`mktemp $plf.tmp_XXXXXXXX`
-unlink $tmpfile;
+rm "$tmpfile" 2>/dev/null;
 
 (
     gitolite list-phy-repos | gitolite access % gitweb R any | grep -v DENIED
