From 10ebbc6862f16077e03a9415e750c63ef6dbd2fc Mon Sep 17 00:00:00 2001
From: Jonathan Esk-Riddell <jr@jriddell.org>
Date: Wed, 9 Feb 2022 14:21:47 +0000
Subject: [PATCH] fix typo

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ad4a3a1..a21aedc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -35,7 +35,7 @@ find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS Core Gui Network Qml Quic
 find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS Kirigami2 I18n People Notifications Config CoreAddons DBusAddons ModemManagerQt)
 find_package(PhoneNumber COMPONENTS PhoneNumber REQUIRED)
 
-find_package(QCore5 COMPONENTS Core DBus QUIET)
+find_package(QCoro5 COMPONENTS Core DBus QUIET)
 if(NOT QCoro5_FOUND)
     find_package(QCoro REQUIRED)
 endif()
-- 
GitLab

