diff --git a/src/database/search_handler.h b/src/database/search_handler.h
index dd63bbb..5912f96 100644
--- a/src/database/search_handler.h
+++ b/src/database/search_handler.h
@@ -32,7 +32,7 @@
 #include "util/tools.h"
 
 #include <algorithm>
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <map>
 #include <memory>
 #include <tuple>
diff --git a/src/exceptions.h b/src/exceptions.h
index 9a437b4..c9967c5 100644
--- a/src/exceptions.h
+++ b/src/exceptions.h
@@ -34,7 +34,7 @@
 #ifndef __EXCEPTIONS_H__
 #define __EXCEPTIONS_H__
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 #include <stdexcept>
 #include <string>
 
diff --git a/src/web/config_load.cc b/src/web/config_load.cc
--- a/src/web/config_load.cc
+++ b/src/web/config_load.cc
@@ -150,7 +150,7 @@
 template <typename T>
 void Web::ConfigLoad::setValue(Json::Value& item, const T& value)
 {
-    static_assert(fmt::has_formatter<T, fmt::format_context>::value, "T must be formattable");
+    static_assert(fmt::is_formattable<T>::value, "T must be formattable");
     item[CONFIG_LOAD_VALUE] = fmt::to_string(value);
 }
 
diff --git a/test/scripting/test_internal_m3u8_playlist.cc b/test/scripting/test_internal_m3u8_playlist.cc
--- a/test/scripting/test_internal_m3u8_playlist.cc
+++ b/test/scripting/test_internal_m3u8_playlist.cc
@@ -28,6 +28,7 @@
 
 #include <duktape.h>
 #include <fmt/core.h>
+#include <fmt/format.h>
 #include <memory>
 
 // Extends ScriptTestFixture to allow
