i have no idea

diff --git a/src/web/config_load.cc b/src/web/config_load.cc
index 4dd9688..3cd1b6b 100644
--- a/src/web/config_load.cc
+++ b/src/web/config_load.cc
@@ -529,7 +529,7 @@ void Web::ConfigLoad::writeTranscoding(pugi::xml_node& values)
 
         item = values.append_child(CONFIG_LOAD_ITEM);
         createItem(item, cs->getItemPath(pr, { ConfigVal::A_TRANSCODING_MIMETYPE_FILTER, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_NOTRANSCODING }), cs->option, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_NOTRANSCODING, cs);
-        setValue(item, fmt::join(filter->getNoTranscodingMimeTypes(), ","));
+        setValue(item, fmt::to_string(fmt::join(filter->getNoTranscodingMimeTypes(), ",")));
 
         item = values.append_child(CONFIG_LOAD_ITEM);
         createItem(item, cs->getItemPath(pr, { ConfigVal::A_TRANSCODING_MIMETYPE_FILTER, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_CLIENTFLAGS }), cs->option, ConfigVal::A_TRANSCODING_PROFILES_PROFLE_CLIENTFLAGS, cs);
@@ -654,7 +654,7 @@ void Web::ConfigLoad::writeAutoscan(pugi::xml_node& values)
 
             item = values.append_child(CONFIG_LOAD_ITEM);
             createItem(item, ascs->getItemPath(i, { ConfigVal::A_AUTOSCAN_DIRECTORY_MODE }), ascs->option, ConfigVal::A_AUTOSCAN_DIRECTORY_MODE);
-            setValue(item, AutoscanDirectory::mapScanmode(adir->getScanMode()));
+            setValue(item, std::string(AutoscanDirectory::mapScanmode(adir->getScanMode())));
 
             item = values.append_child(CONFIG_LOAD_ITEM);
             createItem(item, ascs->getItemPath(i, { ConfigVal::A_AUTOSCAN_DIRECTORY_INTERVAL }), ascs->option, ConfigVal::A_AUTOSCAN_DIRECTORY_INTERVAL);
