Fixes for BusyBox's truncate
--- a/tests/t0290-gpt-name.sh
+++ b/tests/t0290-gpt-name_.sh
@@ -21,7 +21,7 @@
 
 dev=loop-file
 
-truncate --size 50MiB "$dev" || fail=1
+truncate -s 50M "$dev" || fail=1
 
 # create partition with empty name
 parted --script "$dev" mklabel gpt mkpart '""' ext4 1MiB 49MiB > out 2>&1 || fail=1

--- a/tests/t0215-gpt-attrs.sh
+++ b/tests/t0215-gpt-attrs.sh
@@ -23,7 +23,7 @@
 dev=loop-file
 
 # create device
-truncate --size 50MiB "$dev" || fail=1
+truncate -s 50M "$dev" || fail=1
 
 # create gpt label and one partitions
 parted --script "$dev" mklabel gpt > out 2>&1 || fail=1

--- a/tests/t0801-json-msdos.sh
+++ b/tests/t0801-json-msdos.sh
@@ -23,7 +23,7 @@
 dev=loop-file
 
 # create device
-truncate --size 50MiB "$dev" || fail=1
+truncate -s 50M "$dev" || fail=1
 
 # create msdos label and some partitions
 parted --script "$dev" mklabel msdos > out 2>&1 || fail=1

--- a/tests/t0800-json-gpt.sh
+++ b/tests/t0800-json-gpt.sh
@@ -23,7 +23,7 @@
 dev=loop-file
 
 # create device
-truncate --size 50MiB "$dev" || fail=1
+truncate -s 50M "$dev" || fail=1
 
 # create gpt label and some partitions
 parted --script "$dev" mklabel gpt > out 2>&1 || fail=1
