diff --git a/tests/Makefile.in b/tests/Makefile.in
index 11baba0..ccc8a74 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -1644,11 +1644,8 @@ TESTS = \
   t0213-mkpart-start-negative.sh \
   t0220-gpt-msftres.sh \
   t0250-gpt.sh \
-  t0251-gpt-unicode.sh \
   t0280-gpt-corrupt.sh \
   t0281-gpt-grow.sh \
-  t0282-gpt-move-backup.sh \
-  t0283-overlap-partitions.sh \
   t0300-dos-on-gpt.sh \
   t0301-overwrite-gpt-pmbr.sh \
   t0350-mac-PT-increases-sector-size.sh \
@@ -1673,8 +1670,6 @@ TESTS = \
   t3200-resize-partition.sh \
   t3200-type-change.sh \
   t3300-palo-prep.sh \
-  t3310-flags.sh \
-  t3400-whole-disk-FAT-partition.sh \
   t4000-sun-raid-type.sh \
   t4001-sun-vtoc.sh \
   t4100-msdos-partition-limits.sh \
