From 0bb6c9dd19989dcc54d0a2c93bb67b5db9abff0f Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Sun, 31 Dec 2023 13:37:18 -0500
Subject: [PATCH] include paths.h

---
 src/plugin/placesmodel/qmtabparser.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/plugin/placesmodel/qmtabparser.cpp b/src/plugin/placesmodel/qmtabparser.cpp
index 0ce1e73b..45549b42 100644
--- a/src/plugin/placesmodel/qmtabparser.cpp
+++ b/src/plugin/placesmodel/qmtabparser.cpp
@@ -24,6 +24,8 @@
 #include <QFileInfo>
 #include <QStringList>
 
+#include <paths.h>
+
 class MtabFileGuard
 {
     FILE *mtabFile;
-- 
2.43.0

