From fef0cf28275e471e069f570e460746b89bb6d7a6 Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Sun, 31 Dec 2023 13:40:37 -0500
Subject: [PATCH] link libintl

---
 src/app/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index dc22cc94..fb978121 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -12,7 +12,7 @@ set(filemanager_SRCS
 add_executable(lomiri-filemanager-app ${filemanager_SRCS})
 
 qt5_use_modules(lomiri-filemanager-app Gui Qml Quick)
-target_link_libraries(lomiri-filemanager-app stdc++)
+target_link_libraries(lomiri-filemanager-app stdc++ -lintl)
 
 if(NOT "${CMAKE_CURRENT_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_BINARY_DIR}")
 add_custom_target(lomiri-filemanager-app-qmlfiles ALL
-- 
2.43.0

