--- a/bin/BackupPC
+++ b/bin/BackupPC
@@ -1885,7 +1885,7 @@
             print(LOG $bpc->timeStamp, "unix socket() failed: $!\n");
             exit(1);
         }
-        my $sockFile = "$LogDir/BackupPC.sock";
+        my $sockFile = "/run/backuppc/BackupPC.sock";
         unlink($sockFile);
         if ( !bind(SERVER_UNIX, sockaddr_un($sockFile)) ) {
             print(LOG $bpc->timeStamp, "unix bind() failed: $!\n");
--- a/lib/BackupPC/Lib.pm
+++ b/lib/BackupPC/Lib.pm
@@ -686,7 +686,7 @@
     #
     # First try the unix-domain socket
     #
-    my $sockFile = "$bpc->{LogDir}/BackupPC.sock";
+    my $sockFile = "/run/backuppc/BackupPC.sock";
     socket(*FH, PF_UNIX, SOCK_STREAM, 0)     || return "unix socket: $!";
     if ( !connect(*FH, sockaddr_un($sockFile)) ) {
         my $err = "unix connect: $!";
