diff -up ftgl-2.1.3~rc5/demo/FTGLDemo.cpp.ttf_font ftgl-2.1.3~rc5/demo/FTGLDemo.cpp
--- ftgl-2.1.3~rc5/demo/FTGLDemo.cpp.ttf_font	2008-06-08 17:49:10.000000000 +0200
+++ ftgl-2.1.3~rc5/demo/FTGLDemo.cpp	2009-05-21 02:18:42.000000000 +0200
@@ -48,7 +48,7 @@
 #       define FONT_FILE "C:\\WINNT\\Fonts\\arial.ttf"
 #   else
         // Put your font file here if configure did not find it.
-#       define FONT_FILE 0
+#       define FONT_FILE "/usr/share/fonts/dejavu/DejaVuSans.ttf"
 #   endif
 #endif
 
