From eefc1e7e3dd8fb422baf0f13aec1df9880541b83 Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <bug@jasper.la>
Date: Thu, 10 Aug 2017 08:44:01 +0100
Subject: [PATCH] Missing have_cpuinfo check.

---
 virt-what.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/virt-what.in b/virt-what.in
index 8c27b11..9050035 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -308,7 +308,7 @@ if ! "$skip_qemu_kvm"; then
         # option, since /proc/cpuinfo will not contain the QEMU
         # string. QEMU 2.10 added a new CPUID leaf, so this
         # problem only triggered for older QEMU
-        if grep -q 'QEMU' "${root}/proc/cpuinfo"; then
+        if have_cpuinfo && grep -q 'QEMU' "${root}/proc/cpuinfo"; then
             echo qemu
         fi
     fi
-- 
2.15.0

