diff -ru sslscan-1.8.2/Makefile sslscan-1.8.2.fc12/Makefile
--- sslscan-1.8.2/Makefile	2010-01-16 20:33:16.000000000 +0100
+++ sslscan-1.8.2.fc12/Makefile	2010-01-16 20:41:10.000000000 +0100
@@ -1,17 +1,17 @@
 SRCS = sslscan.c
-BINPATH = /usr/bin/
-MANPATH = /usr/share/man/
+BINPATH ?= /usr/bin/
+MANPATH ?= /usr/share/man/
 
 all:
-	gcc -g -Wall -lssl -o sslscan $(SRCS) $(LDFLAGS) $(CFLAGS)
+	gcc -g -Wall -lssl -lcrypto -o sslscan $(SRCS) $(LDFLAGS) $(CFLAGS)
 
 install:
-	cp sslscan $(BINPATH)
-	cp sslscan.1 $(MANPATH)man1
+	install -D -m 755 sslscan $(DESTDIR)$(BINPATH)sslscan
+	install -D -m 644 sslscan.1 $(DESTDIR)$(MANPATH)man1/sslscan.1
 
 uninstall:
-	rm -f $(BINPATH)sslscan
-	rm -f $(MANPATH)man1/sslscan.1
+	rm -f $(DESTDIR)$(BINPATH)sslscan
+	rm -f $(DESTDIR)$(MANPATH)man1/sslscan.1
 
 clean:
 	rm -f sslscan
