diff -rupN a/mygpoclient/http_test.py b/mygpoclient/http_test.py
--- a/mygpoclient/http_test.py	2024-11-23 16:29:23.000000000 +0000
+++ b/mygpoclient/http_test.py	2026-04-14 02:02:58.460000000 +0000
@@ -141,12 +141,12 @@ class Test_HttpClient(unittest.TestCase)
                                                       args=(self.PORT, self.USERNAME, self.PASSWORD, self.RESPONSE))
         self.server_process.start()
         import time
-        time.sleep(.1)
+        time.sleep(10)
 
     def tearDown(self):
         self.server_process.terminate()
         import time
-        time.sleep(.1)
+        time.sleep(10)
 
     def test_UnknownResponse(self):
         client = HttpClient()
