diff --git a/Makefile.in b/Makefile.in
index 1a828b7..b4fcf35 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -8563,7 +8563,8 @@ AM_TESTS_ENVIRONMENT = PATH="$(PATH)"
 @FAPI_TRUE@@UNIT_TRUE@                            -Wl,--wrap=write \
 @FAPI_TRUE@@UNIT_TRUE@                            -Wl,--wrap=fileno \
 @FAPI_TRUE@@UNIT_TRUE@                            -Wl,--wrap=fclose \
-@FAPI_TRUE@@UNIT_TRUE@                            -Wl,--wrap=stat
+@FAPI_TRUE@@UNIT_TRUE@                            -Wl,--wrap=stat \
+                                                  -Wl,--wrap=__stat_time64
 
 @FAPI_TRUE@@UNIT_TRUE@test_unit_fapi_io_SOURCES = test/unit/fapi-io.c \
 @FAPI_TRUE@@UNIT_TRUE@                            src/tss2-fapi/ifapi_json_deserialize.c \
--- a/test/unit/fapi-io.c
+++ b/test/unit/fapi-io.c
@@ -57,6 +57,12 @@ int
     return 0;
 }
 
+int
+ __wrap___stat_time64(const char *pathname, struct stat *statbuf, ...)
+{
+    return __wrap_stat(pathname, statbuf);
+}
+
 FILE *
 __real_fopen(const char *pathname, const char* mode, ...);
 FILE *
