diff --git a/Makefile-test.am b/Makefile-test.am
index c4fb7803..b7844672 100644
--- a/Makefile-test.am
+++ b/Makefile-test.am
@@ -447,7 +447,8 @@ if ENABLE_TCTI_PCAP
 test_unit_tcti_pcap_CFLAGS  = $(CMOCKA_CFLAGS) $(TESTS_CFLAGS)
 test_unit_tcti_pcap_LDADD   = $(CMOCKA_LIBS) $(libtss2_mu) $(libutil)
 test_unit_tcti_pcap_LDFLAGS = -Wl,--wrap=getenv -Wl,--wrap=rand -Wl,--wrap=clock_gettime \
-        -Wl,--wrap=open -Wl,--wrap=read -Wl,--wrap=write -Wl,--wrap=close
+        -Wl,--wrap=open -Wl,--wrap=read -Wl,--wrap=write -Wl,--wrap=close \
+	-Wl,--wrap=__clock_gettime64
 test_unit_tcti_pcap_SOURCES = test/unit/tcti-pcap.c \
     src/tss2-tcti/tcti-common.c \
     src/tss2-tcti/tcti-pcap.c src/tss2-tcti/tcti-pcap.h \
diff --git a/test/unit/tcti-pcap.c b/test/unit/tcti-pcap.c
index c540de77..6fdc353e 100644
--- a/test/unit/tcti-pcap.c
+++ b/test/unit/tcti-pcap.c
@@ -288,6 +288,11 @@ __wrap_clock_gettime (clockid_t clk_id, struct timespec *tp)
     return EXIT_FAILURE;
 }
 
+int
+__wrap___clock_gettime64 (clockid_t clk_id, struct timespec *tp)
+{
+    return __wrap_clock_gettime(clk_id, tp);
+}
 
 int
 __real_open (const char *pathname, int flags, mode_t mode);
