From 440278d4971c82b366077d61f682105e197f9680 Mon Sep 17 00:00:00 2001
From: WANG Rui <wangrui@loongson.cn>
Date: Fri, 1 Mar 2024 11:43:27 +0800
Subject: [PATCH 02/12] tests: Add loongarch64-unknown-linux-musl target

---
 tests/assembly/targets/targets-elf.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tests/assembly/targets/targets-elf.rs b/tests/assembly/targets/targets-elf.rs
index 41f5df0fb..bf89790a5 100644
--- a/tests/assembly/targets/targets-elf.rs
+++ b/tests/assembly/targets/targets-elf.rs
@@ -231,6 +231,9 @@
 // revisions: loongarch64_unknown_linux_gnu
 // [loongarch64_unknown_linux_gnu] compile-flags: --target loongarch64-unknown-linux-gnu
 // [loongarch64_unknown_linux_gnu] needs-llvm-components: loongarch
+// revisions: loongarch64_unknown_linux_musl
+// [loongarch64_unknown_linux_musl] compile-flags: --target loongarch64-unknown-linux-musl
+// [loongarch64_unknown_linux_musl] needs-llvm-components: loongarch
 // revisions: loongarch64_unknown_none
 // [loongarch64_unknown_none] compile-flags: --target loongarch64-unknown-none
 // [loongarch64_unknown_none] needs-llvm-components: loongarch
-- 
2.44.0

