diff -urp optipng-0.7.6-orig/configure optipng-0.7.6/configure
--- optipng-0.7.6-orig/configure	2014-12-01 07:25:00.000000000 +0200
+++ optipng-0.7.6/configure	2016-05-17 20:56:08.950224977 +0300
@@ -234,6 +234,10 @@ then
         fi
         if test "$CPPFLAGS"
         then
+            if test "$with_system_zlib" -eq 0
+            then
+                CPPFLAGS="-I../zlib ${CPPFLAGS}"
+            fi
             sed_preconfig_libpng="
                 $sed_preconfig_libpng
                 s|^CPPFLAGS *=.*|CPPFLAGS = $CPPFLAGS|
