# longaarch64 only - https://gitlab.freedesktop.org/upower/upower/-/issues/290
diff --git a/src/linux/integration-test.py b/src/linux/integration-test.py
index 4cdefa8..70fbad6 100755
--- a/src/linux/integration-test.py
+++ b/src/linux/integration-test.py
@@ -243,7 +243,7 @@ class Tests(dbusmock.DBusTestCase):
             except OSError:
                 pass
             try:
-                self.assertEqual(self.daemon.wait(timeout=5.0), 0)
+                self.assertEqual(self.daemon.wait(timeout=10.0), 0)
             except subprocess.TimeoutExpired:
                 try:
                     self.daemon.kill()
