Tests otherwise fail in armhf due to a timeout.

diff --git i/src/meson.build w/src/meson.build
index 406f9cb..f45df9f 100644
--- i/src/meson.build
+++ w/src/meson.build
@@ -135,7 +135,7 @@ if os_backend == 'linux' and gobject_introspection.found()
              args: ut_args,
              env: env,
              depends: [ libupower_glib_typelib, upowerd ],
-             timeout: 50,
+             timeout: 60,
             )
     endforeach
 endif
