Index: /trunk/oscam-conf.c
===================================================================
--- trunk/oscam-conf.c	(revision 11018)
+++ trunk/oscam-conf.c	(revision 11019)
@@ -525,5 +525,8 @@
 	strncat(tmp_file, ".tmp", sizeof(tmp_file) - strlen(tmp_file) - 1);
 	strncat(bak_file, ".bak", sizeof(bak_file) - strlen(bak_file) - 1);
-	fclose(f);
+	if(f)
+	{
+		fclose(f);
+	}
 	return safe_overwrite_with_bak(dst_file, tmp_file, bak_file, cfg.http_overwrite_bak_file);
 }
Index: /trunk/oscam-config.c
===================================================================
--- trunk/oscam-config.c	(revision 11018)
+++ trunk/oscam-config.c	(revision 11019)
@@ -436,5 +436,9 @@
 	{ 
 		fp = create_config_file("oscam.srvid2");
-		flush_config_file(fp, "oscam.srvid2");
+		if(fp)
+		{
+			flush_config_file(fp, "oscam.srvid2");
+		}
+		
 		return 0;
 	}
