--- a/setup.py
+++ b/setup.py
@@ -40,7 +40,7 @@
     )
 )  # noqa: E128

-SETUP_REQUIRES = ['setuptools >= 34.4', 'setuptools_scm >= 3.0']
+SETUP_REQUIRES = ['setuptools >= 34.4']

 INSTALL_REQUIRES = [
     'attrs >= 16.3.0',
@@ -91,8 +91,8 @@
 }

 setup(
+    version="%VERSION%",
     name="cmd2",
-    use_scm_version={'git_describe_command': 'git describe --dirty --tags --long --exclude plugin-*'},
     description=DESCRIPTION,
     long_description=LONG_DESCRIPTION,
     long_description_content_type='text/markdown',
