--- a/event_emitter.gemspec	2017-07-25 02:30:16.000000000 -0400
+++ b/event_emitter.gemspec	2024-10-03 14:01:24.233398437 -0400
@@ -11,12 +11,12 @@
   gem.summary       = gem.description
   gem.homepage      = "http://shokai.github.com/event_emitter"
 
-  gem.files         = `git ls-files`.split($/).reject{|i| i=="Gemfile.lock" }
+  gem.files         = Dir["lib/**/*"]
   gem.executables   = gem.files.grep(%r{^bin/}).map{ |f| File.basename(f) }
   gem.test_files    = gem.files.grep(%r{^(test|spec|features)/})
   gem.require_paths = ["lib"]
 
   gem.add_development_dependency "minitest"
   gem.add_development_dependency "rake"
-  gem.add_development_dependency "bundler", "~> 1.15"
+  gem.add_development_dependency "bundler"
 end
