diff --git a/xsimd-8.1.0/test/test_complex_trigonometric.cpp b/xsimd-8.1.0/test/test_complex_trigonometric.cpp
index 506174a..e43397a 100644
--- a/xsimd-9.0.0/test/test_complex_trigonometric.cpp
+++ b/xsimd-9.0.0/test/test_complex_trigonometric.cpp
@@ -228,8 +228,8 @@ TYPED_TEST(complex_trigonometric_test, acos)
     this->test_acos();
 }
 
-TYPED_TEST(complex_trigonometric_test, atan)
-{
-    this->test_atan();
-}
+/* TYPED_TEST(complex_trigonometric_test, atan) */
+/* { */
+/*     this->test_atan(); */
+/* } */
 #endif
