From: =?utf-8?q?Timo_R=C3=B6hling?= <roehling@debian.org>
Date: Fri, 19 Aug 2022 20:28:12 +0200
Subject: Set DRACO_LIBRARIES for backwards compatibility

---
 cmake/draco-config.cmake.template | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/draco-config.cmake.template b/cmake/draco-config.cmake.template
index ed86823..12d54b7 100644
--- a/cmake/draco-config.cmake.template
+++ b/cmake/draco-config.cmake.template
@@ -1,3 +1,5 @@
 @PACKAGE_INIT@
 
 include("${CMAKE_CURRENT_LIST_DIR}/draco-targets.cmake")
+
+set(DRACO_LIBRARIES "draco::draco")
