Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: use gtest package
----

--- a/cmake/draco_dependencies.cmake
+++ b/cmake/draco_dependencies.cmake
@@ -107,11 +107,10 @@
   endif()
 
   list(APPEND draco_test_include_paths ${draco_include_paths}
-              "${gtest_path}/include" "${gtest_path}/googlemock"
-              "${gtest_path}/googletest/include" "${gtest_path}/googletest")
+              "${gtest_path}/include/gmock/" "${gtest_path}/include/gtest/" "${gtest_path}/src/gtest/")
 
-  list(APPEND draco_gtest_all "${gtest_path}/googletest/src/gtest-all.cc")
-  list(APPEND draco_gtest_main "${gtest_path}/googletest/src/gtest_main.cc")
+  list(APPEND draco_gtest_all "${gtest_path}/src/gtest/src/gtest-all.cc")
+  list(APPEND draco_gtest_main "${gtest_path}/src/gtest/src/gtest_main.cc")
 endmacro()
 
 
