From: Sebastian Ramacher <sramacher@debian.org>
Date: Sun, 19 Jan 2020 11:24:59 +0100
Subject: Build without OpenSSL

---
 config.linux-with-shared-libraries | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/config.linux-with-shared-libraries b/config.linux-with-shared-libraries
index 2ae04ef..041da7f 100644
--- a/config.linux-with-shared-libraries
+++ b/config.linux-with-shared-libraries
@@ -24,7 +24,7 @@ libgroupsock_VERSION_AGE=2
 libgroupsock_LIB_SUFFIX=so.$(shell expr $(libgroupsock_VERSION_CURRENT) - $(libgroupsock_VERSION_AGE)).$(libgroupsock_VERSION_AGE).$(libgroupsock_VERSION_REVISION)
 #####
 
-COMPILE_OPTS =		$(INCLUDES) -I/usr/local/include -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -fPIC
+COMPILE_OPTS =		$(INCLUDES) -I/usr/include -I. -O2 -DSOCKLEN_T=socklen_t -D_LARGEFILE_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DNO_OPENSSL=1 -fPIC
 C =			c
 C_COMPILER =		$(CC)
 C_FLAGS =		$(COMPILE_OPTS) $(CPPFLAGS) $(CFLAGS)
@@ -39,7 +39,7 @@ LIBRARY_LINK =		$(CXX) -o
 SHORT_LIB_SUFFIX =	so.$(shell expr $($(NAME)_VERSION_CURRENT) - $($(NAME)_VERSION_AGE))
 LIB_SUFFIX =	 	$(SHORT_LIB_SUFFIX).$($(NAME)_VERSION_AGE).$($(NAME)_VERSION_REVISION)
 LIBRARY_LINK_OPTS =	-shared -Wl,-soname,$(NAME).$(SHORT_LIB_SUFFIX) $(CPPFLAGS) $(LDFLAGS)
-LIBS_FOR_CONSOLE_APPLICATION = -lssl -lcrypto
+LIBS_FOR_CONSOLE_APPLICATION =
 LIBS_FOR_GUI_APPLICATION =
 EXE =
 INSTALL2 =		install_shared_libraries
