From 52947f2457102189173861a1ebd829fb7f56d40a Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Mon, 14 Oct 2024 21:40:17 -0700
Subject: [PATCH] Makefile: install to /usr/sbin

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e003832..8050eae 100644
--- a/Makefile
+++ b/Makefile
@@ -149,7 +149,7 @@ c2sh c2tmac ver: CC = ${HOSTCC}
 .INTERMEDIATE: c2sh c2tmac ver
 
 install_sbin: File = auibusy aumvdown auplink mount.aufs umount.aufs
-install_sbin: Tgt = ${DESTDIR}/sbin
+install_sbin: Tgt = ${DESTDIR}/usr/sbin
 install_ubin: File = aubusy auchk aubrsync #auctl
 install_ubin: Tgt = ${DESTDIR}/usr/bin
 install_sbin install_ubin: ${File}
-- 
2.46.2

