diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -148,11 +148,13 @@ if(BUILD_SHARED_LIBS)
 	set_property(TARGET mojoshader PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
+find_package(VulkanHeaders CONFIG)
+
 # FNA3D folders as includes, for other targets to consume
 target_include_directories(FNA3D PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Headers/include>
+	Vulkan::Headers
 )
 if(BUILD_DXVK_NATIVE)
 	add_definitions(-DFNA3D_DXVK_NATIVE)
@@ -161,7 +163,7 @@ if(BUILD_DXVK_NATIVE)
 endif()
 target_include_directories(mojoshader PUBLIC
 	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/MojoShader>
-	$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/Vulkan-Headers/include>
+	Vulkan::Headers
 )
 
 # MinGW builds should statically link libgcc
