diff --git a/CMakeLists.txt b/CMakeLists.txt
index aebce40..f9d155b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,7 +27,7 @@ execute_process(COMMAND git describe --exact-match --tags OUTPUT_VARIABLE GIT_TA
 if(GIT_ON_TAG)
     set(IS_DEVEL_BUILD "false")
 else()
-    set(IS_DEVEL_BUILD "true")
+    set(IS_DEVEL_BUILD "false")
 endif()
 
 # In order to avoid for things to get out of sync, it is also possible to hand
