diff --git a/decoder_plugins/ffmpeg/ffmpeg.c b/decoder_plugins/ffmpeg/ffmpeg.c
index 206b2b1..6af6759 100644
--- a/decoder_plugins/ffmpeg/ffmpeg.c
+++ b/decoder_plugins/ffmpeg/ffmpeg.c
@@ -66,9 +66,6 @@ GCC_DIAG_OFF(deprecated-declarations)
 #include <libavformat/avformat.h>
 GCC_DIAG_ON(deprecated-declarations)
 #include <libavutil/mathematics.h>
-#ifdef HAVE_AV_GET_CHANNEL_LAYOUT_NB_CHANNELS
-#include <libavutil/audioconvert.h>
-#endif
 #else
 #include <ffmpeg/avformat.h>
 #endif
