diff --git a/setup.py b/setup.py
index 6bf1104..732809f 100644
--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@ sys.path.insert(0, os.path.join(dist_dir, 'src', 'quark_sphinx_theme'))
 from __version__ import __version__  # noqa
 del sys.path[0]
 
-tests_require = ['html5lib', 'tinycss']
+tests_require = ['html5lib', 'tinycss2']
 if sys.version_info[:2] < (3, 3):
     tests_require.append('mock')
 
