libtinfo is just a split part of ncurses
--
diff --git a/src/musikcube/CMakeLists.txt b/src/musikcube/CMakeLists.txt
index e16ec8b..7d10500 100644
--- a/src/musikcube/CMakeLists.txt
+++ b/src/musikcube/CMakeLists.txt
@@ -113,9 +113,6 @@ else()
         message(STATUS "[ncurses] detected OpenBSD, unsetting LIBTINFO")
         set(LIBTINFO "")
     else()
-        message(STATUS "[ncurses] not Darwin! will attempt to link against libtinfo")
-        find_library(LIBTINFO NAMES tinfo)
-        message(STATUS "[musikcube] using libtinfo at: " ${LIBTINFO})
     endif()
 endif()
 
