From 21668119c0f83f309b423b8a443dbef5206ab778 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 4 Jun 2010 18:02:39 +0300
Subject: [PATCH 4/4] crypto/engine: autoload padlock dynamic engine

---
 crypto/engine/eng_all.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/crypto/engine/eng_all.c b/crypto/engine/eng_all.c
index 6093376..210b2d7 100644
--- a/crypto/engine/eng_all.c
+++ b/crypto/engine/eng_all.c
@@ -120,6 +120,16 @@ void ENGINE_load_builtin_engines(void)
 	ENGINE_load_capi();
 #endif
 #endif
+#ifdef OPENSSL_NO_STATIC_ENGINE
+	{
+		ENGINE *e;
+		e = ENGINE_by_id("padlock");
+		if (e != NULL) {
+			ENGINE_add(e);
+			ENGINE_free(e);
+		}
+	}
+#endif
 	ENGINE_register_all_complete();
 	}
 
-- 
1.7.11.3

