From b96276c8f11e656e4296955bf1a8d0ac2b6094fe Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 4 Jun 2010 18:02:39 +0300
Subject: [PATCH 5/5] crypto/engine: autoload padlock dynamic engine

---
 crypto/engine/eng_all.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/crypto/engine/eng_all.c b/crypto/engine/eng_all.c
index 22c1204..827e447 100644
--- a/crypto/engine/eng_all.c
+++ b/crypto/engine/eng_all.c
@@ -112,6 +112,16 @@ void ENGINE_load_builtin_engines(void)
 	ENGINE_load_capi();
 #endif
 #endif
+#ifdef OPENSSL_NO_STATIC_ENGINE
+	{
+		ENGINE *e;
+		e = ENGINE_by_id("padlock");
+		if (e != NULL) {
+			ENGINE_add(e);
+			ENGINE_free(e);
+		}
+	}
+#endif
 	}
 
 #if defined(__OpenBSD__) || defined(__FreeBSD__) || defined(HAVE_CRYPTODEV)
-- 
1.7.0.4

