--- openssl-0.9.8i.orig/Makefile.org	Wed Nov  5 21:43:31 2008
+++ openssl-0.9.8i/Makefile.org	Wed Nov  5 22:26:40 2008
@@ -551,7 +551,7 @@
 		filecase=-i; \
 	fi; \
 	set -e; for i in doc/apps/*.pod; do \
-		fn=`basename $$i .pod`; \
+		fn=`basename $$i .pod || true`; \
 		sec=`$(PERL) util/extract-section.pl 1 < $$i`; \
 		echo "installing man$$sec/$$fn.$${sec}$(MANSUFFIX)"; \
 		(cd `$(PERL) util/dirname.pl $$i`; \
@@ -568,7 +568,7 @@
 			 done); \
 	done; \
 	set -e; for i in doc/crypto/*.pod doc/ssl/*.pod; do \
-		fn=`basename $$i .pod`; \
+		fn=`basename $$i .pod || true`; \
 		sec=`$(PERL) util/extract-section.pl 3 < $$i`; \
 		echo "installing man$$sec/$$fn.$${sec}$(MANSUFFIX)"; \
 		(cd `$(PERL) util/dirname.pl $$i`; \
