diff -up openssl-1.0.1/crypto/cversion.c.version openssl-1.0.1/crypto/cversion.c
--- openssl-1.0.1/crypto/cversion.c.version	2004-04-19 20:09:22.000000000 +0200
+++ openssl-1.0.1/crypto/cversion.c	2012-03-14 20:58:20.630352536 +0100
@@ -110,8 +110,15 @@ const char *SSLeay_version(int t)
 	return("not available");
 	}
 
-unsigned long SSLeay(void)
+unsigned long _original_SSLeay(void)
+	{
+	return(0x10000003);
+	}
+
+unsigned long _current_SSLeay(void)
 	{
 	return(SSLEAY_VERSION_NUMBER);
 	}
 
+__asm__(".symver _original_SSLeay,SSLeay@");
+__asm__(".symver _current_SSLeay,SSLeay@@OPENSSL_1.0.1");
diff -up openssl-1.0.1/crypto/opensslv.h.version openssl-1.0.1/crypto/opensslv.h
--- openssl-1.0.1/crypto/opensslv.h.version	2012-03-14 20:58:19.914337879 +0100
+++ openssl-1.0.1/crypto/opensslv.h	2012-03-14 20:58:20.630352536 +0100
@@ -83,7 +83,7 @@
  * should only keep the versions that are binary compatible with the current.
  */
 #define SHLIB_VERSION_HISTORY ""
-#define SHLIB_VERSION_NUMBER "1.0.0"
+#define SHLIB_VERSION_NUMBER "1.0.1d"
 
 
 #endif /* HEADER_OPENSSLV_H */
diff -up openssl-1.0.1/Makefile.shared.version openssl-1.0.1/Makefile.shared
--- openssl-1.0.1/Makefile.shared.version	2012-03-14 20:58:20.553350959 +0100
+++ openssl-1.0.1/Makefile.shared	2012-03-14 20:58:20.631352556 +0100
@@ -151,7 +151,7 @@ DO_GNU_SO=$(CALC_VERSIONS); \
 	SHLIB_SUFFIX=; \
 	ALLSYMSFLAGS='-Wl,--whole-archive'; \
 	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,--default-symver,--version-script=version.map -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
 
 DO_GNU_APP=LDFLAGS="$(CFLAGS)"
 
diff -up openssl-1.0.1/version.map.version openssl-1.0.1/version.map
--- openssl-1.0.1/version.map.version	2012-03-14 20:58:20.631352556 +0100
+++ openssl-1.0.1/version.map	2012-03-14 20:58:20.631352556 +0100
@@ -0,0 +1,7 @@
+OPENSSL_1.0.1 {
+    global:
+	    SSLeay;
+    local:
+	    _original*;
+	    _current*;
+};
