--- a/Makefile	2023-09-18 11:54:09.000000000 +0200
+++ b/Makefile	2023-09-18 20:19:59.347761343 +0200
@@ -81,7 +81,7 @@ ifneq (,$(DOCKER_BUILD))
 LD_OPTS_VARS += -X '$(GO_MODULE_NAME)/pkg/cwversion.System=docker'
 endif
 
-GO_TAGS := netgo,osusergo,sqlite_omit_load_extension
+GO_TAGS := netgo,osusergo,sqlite_omit_load_extension,libsqlite3
 
 # this will be used by Go in the make target, some distributions require it
 export PKG_CONFIG_PATH:=/usr/local/lib/pkgconfig:$(PKG_CONFIG_PATH)
