diff --git a/Makefile b/Makefile
index 334e738..149bee5 100644
--- a/Makefile
+++ b/Makefile
@@ -38,8 +38,8 @@ ifdef BUILD_STATIC
 $(warning WARNING: The BUILD_STATIC variable is deprecated and has no effect. Builds are static by default since v1.5.0.)
 endif
 
-export LD_OPTS=-ldflags "-s -w -extldflags '-static' $(LD_OPTS_VARS)" \
-	-trimpath -tags netgo,osusergo,sqlite_omit_load_extension
+export LD_OPTS=-ldflags "$(LD_OPTS_VARS)" \
+	-trimpath -tags netgo,osusergo,sqlite_omit_load_extension,libsqlite3
 
 ifneq (,$(TEST_COVERAGE))
 LD_OPTS += -cover
