diff -rNU3 fortify-headers-1.1.old/README fortify-headers-1.1/README
--- fortify-headers-1.1.old/README	2019-04-14 10:26:11.000000000 +0200
+++ fortify-headers-1.1/README	2021-02-16 18:10:51.305259894 +0100
@@ -67,7 +67,6 @@
 mempcpy
 memset
 poll
-ppoll
 pread
 read
 readlink
diff -rNU3 fortify-headers-1.1.old/include/poll.h fortify-headers-1.1/include/poll.h
--- fortify-headers-1.1.old/include/poll.h	2019-04-14 10:26:11.000000000 +0200
+++ fortify-headers-1.1/include/poll.h	2021-02-16 18:11:25.415259531 +0100
@@ -39,19 +39,6 @@
 	return __orig_poll(__f, __n, __s);
 }
 
-#ifdef _GNU_SOURCE
-#undef ppoll
-_FORTIFY_FN(ppoll) int ppoll(struct pollfd *__f, nfds_t __n, const struct timespec *__s,
-                             const sigset_t *__m)
-{
-	__typeof__(sizeof 0) __b = __builtin_object_size(__f, 0);
-
-	if (__n > __b / sizeof(struct pollfd))
-		__builtin_trap();
-	return __orig_ppoll(__f, __n, __s, __m);
-}
-#endif
-
 #ifdef __cplusplus
 }
 #endif
