diff --git a/main.c b/main.c
index 8ecf99c..7bfda07 100644
--- a/main.c
+++ b/main.c
@@ -427,7 +427,7 @@ int parent_isalive()
 	return 1;
 }
 
-void child_sigchld()
+void child_sigchld(int signo)
 {
 	if ( getppid() != 1 )
 		return;
@@ -437,7 +437,7 @@ void child_sigchld()
 	return;
 }
 
-int sethandler_sigchld ( void ( *handler ) () )
+int sethandler_sigchld ( void ( *handler ) ( int ) )
 {
 	struct sigaction sa;
 	sa.sa_handler = handler;
