--- gdnsd-1.6.9/configure.ac
+++ gdnsd-1.6.9/configure.ac
@@ -1,8 +1,8 @@
 AC_INIT([gdnsd],[1.6.9],[blblack@gmail.com],[gdnsd],[https://github.com/blblack/gdnsd])
-AC_PREREQ([2.69])
+AC_PREREQ([2.68])
 AC_CONFIG_AUX_DIR([acaux])
 AC_CONFIG_MACRO_DIR([m4])
-AM_INIT_AUTOMAKE([1.12 dist-xz foreign tar-ustar -Wall])
+AM_INIT_AUTOMAKE([1.11 dist-xz foreign tar-ustar -Wall])
 AM_SILENT_RULES([yes])
 
 dnl These lines pretty much *have* to be in this order, and before
