--- a/src/partitionmanager.cpp
+++ b/src/partitionmanager.cpp
@@ -285,8 +285,8 @@
                     && quota.dqb_bsoftlimit != 0)
                 quotaAvailable = std::max((qint64)dbtob(quota.dqb_bsoftlimit) - (qint64)quota.dqb_curspace, 0LL);
 
-            struct statvfs64 stat;
-            if (::statvfs64(partition->mountPath.toUtf8().constData(), &stat) == 0) {
+            struct statvfs stat;
+            if (::statvfs(partition->mountPath.toUtf8().constData(), &stat) == 0) {
                 partition->bytesTotal = stat.f_blocks * stat.f_frsize;
                 qint64 bytesFree = stat.f_bfree * stat.f_frsize;
                 qint64 bytesAvailable = std::min((qint64)(stat.f_bavail * stat.f_frsize), quotaAvailable);
