--- a/build/ldflags.sh
+++ b/build/ldflags.sh
@@ -5,6 +5,6 @@
 cd "$(dirname "${0}")"/..
 
 # The build.utcTime format must remain in sync with TimeFormat in info.go.
-echo '-X "github.com/cockroachdb/cockroach/pkg/build.tag='$(git describe --dirty --tags)'"' \
+echo '-X "github.com/cockroachdb/cockroach/pkg/build.tag='$PKGVERSION'"' \
      '-X "github.com/cockroachdb/cockroach/pkg/build.utcTime='$(date -u '+%Y/%m/%d %H:%M:%S')'"' \
      '-X "github.com/cockroachdb/cockroach/pkg/build.deps='$(build/depvers.sh)'"'
