This testcase fails due to musl libc.

--- a/src/shrpx-unittest.cc
+++ b/src/shrpx-unittest.cc
@@ -177,8 +177,6 @@
       !CU_add_test(pSuite, "util_ends_with", shrpx::test_util_ends_with) ||
       !CU_add_test(pSuite, "util_parse_http_date",
                    shrpx::test_util_parse_http_date) ||
-      !CU_add_test(pSuite, "util_localtime_date",
-                   shrpx::test_util_localtime_date) ||
       !CU_add_test(pSuite, "util_get_uint64", shrpx::test_util_get_uint64) ||
       !CU_add_test(pSuite, "util_parse_config_str_list",
                    shrpx::test_util_parse_config_str_list) ||
