diff -ruN a/include/iprt/types.h b/include/iprt/types.h
--- a/include/iprt/types.h	2024-10-10 18:19:39.000000000 +0000
+++ b/include/iprt/types.h	2025-11-08 23:10:42.120000000 +0000
@@ -249,55 +249,9 @@
 #endif
 
 /*
- * C doesn't have bool, nor does VisualAge for C++ v3.08.
+ * Starting with c23, bool, true and false are keywords.
  */
-#if !defined(__cplusplus) || (defined(__IBMCPP__) && defined(RT_OS_OS2))
-# if defined(__GNUC__)
-#  if defined(RT_OS_LINUX) && __GNUC__ < 3
-typedef uint8_t bool;
-#  elif defined(RT_OS_FREEBSD)
-#   ifndef __bool_true_false_are_defined
-typedef _Bool bool;
-#   endif
-#  elif defined(RT_OS_NETBSD)
-#   if !defined(_KERNEL)
-     /*
-      * For the kernel code <stdbool.h> is not available, but bool is
-      * provided by <sys/types.h> included above.
-      */
-#    include <stdbool.h>
-
-     /*
-      * ... but the story doesn't end here.  The C standard says that
-      * <stdbool.h> defines preprocessor macro "bool" that expands to
-      * "_Bool", but adds that a program may undefine/redefine it
-      * (this is 7.16 in C99 and 7.18 in C11).  We have to play this
-      * game here because X11 code uses "bool" as a struct member name
-      * - so undefine "bool" and provide it as a typedef instead.  We
-      * still keep #include <stdbool.h> so that any code that might
-      * include it later doesn't mess things up.
-      */
-#    undef bool
-     typedef _Bool bool;
-#   endif
-#  else
-#   undef bool /* see above netbsd explanation */
-typedef _Bool bool;
-#  endif
-# else
-#  if RT_MSC_PREREQ(RT_MSC_VER_VC120)
-#   include <stdbool.h>
-#  else
-typedef unsigned char bool;
-#  endif
-# endif
-# ifndef true
-#  define true  (1)
-# endif
-# ifndef false
-#  define false (0)
-# endif
-#endif
+#include <stdbool.h>
 
 
 /**
