--- VirtualBox-7.0.14.orig/src/VBox/Additions/common/VBoxService/Makefile.kmk
+++ VirtualBox-7.0.14/src/VBox/Additions/common/VBoxService/Makefile.kmk
@@ -215,6 +215,10 @@
 VBoxService_VBOX_IMPORT_CHECKER.win.x86 = nt31
 VBoxService_VBOX_IMPORT_CHECKER.win.amd64 = xp64
 
+# utmps libraries
+VBoxService_CFLAGS += -I/usr/include/utmps
+VBoxService_LIBS.linux += utmps skarnet
+
 $(call VBOX_SET_VER_INFO_EXE,VBoxService,VirtualBox Guest Additions Service,$(VBOX_WINDOWS_ICON_FILE)) # Version info / description.
 
 include $(FILE_KBUILD_SUB_FOOTER)
