diff -rupN VirtualBox-6.0.4-bak/src/VBox/Additions/x11/VBoxClient/Makefile.kmk VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/Makefile.kmk
--- VirtualBox-6.0.4-bak/src/VBox/Additions/x11/VBoxClient/Makefile.kmk	2019-03-28 16:29:24.277508391 +0100
+++ VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/Makefile.kmk	2019-03-28 16:30:24.893944894 +0100
@@ -72,14 +72,6 @@ VBoxClient_LIBS += \
 	supc++ \
 	gcc_eh
 
-# This forces the memcpy references in the static libraries to go to
-# __wrap_memcpy, which we can wrap around memcpy@GLIBC_2.2.5.  I do not know
-# how else to do that without recompiling or implementing our own memcpy.
-ifeq ($(KBUILD_TARGET),linux)
-VBoxClient_LDFLAGS.amd64 += \
-	-Wl,--wrap=memcpy
-endif
-
 ifdef VBOX_WITH_GUEST_PROPS
 VBoxClient_DEFS += VBOX_WITH_GUEST_PROPS
 endif
diff -rupN VirtualBox-6.0.4-bak/src/VBox/Additions/x11/VBoxClient/chk_stubs.c VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/chk_stubs.c
--- VirtualBox-6.0.4-bak/src/VBox/Additions/x11/VBoxClient/chk_stubs.c	2019-03-28 16:29:24.277508391 +0100
+++ VirtualBox-6.0.4/src/VBox/Additions/x11/VBoxClient/chk_stubs.c	2019-03-28 16:29:52.860733197 +0100
@@ -43,17 +43,3 @@ void __stack_chk_fail(void)
     fprintf(stderr, "Stack check failed!\n");
     _exit(1);
 }
-
-#ifdef __x86_64
-/* Furthermore, wrap references to memcpy to force them to go to the right
- * version.  We are forced to do it this way because the shared libraries
- * supc++ and gcc_eh contain references which we cannot change. */
-
-extern void *__wrap_memcpy(void *dest, const void *src, size_t n);
-
-asm (".symver memcpy, memcpy@GLIBC_2.2.5");
-void *__wrap_memcpy(void *dest, const void *src, size_t n)
-{
-    return memcpy(dest, src, n);
-}
-#endif
