--- ./src/VBox/Additions/common/VBoxGuestLib/Makefile.kmk.orig
+++ ./src/VBox/Additions/common/VBoxGuestLib/Makefile.kmk
@@ -38,16 +38,6 @@
 LIBRARIES += \
 	VBoxGuestR3Lib \
 	VBoxGuestR3LibShared
-ifndef VBOX_ONLY_VALIDATIONKIT
- if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd)
-  LIBRARIES += \
-  	VBoxGuestR3LibXFree86
- endif
- if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd solaris)
-  LIBRARIES += \
-  	VBoxGuestR3LibXOrg
- endif
-endif
 LIBRARIES.win.amd64 += VBoxGuestR3Lib-x86 VBoxGuestR3LibShared-x86
 
 
