--- ./setup.py.orig
+++ ./setup.py
@@ -29,7 +29,7 @@
                 'No templates needed. Serves pure pythonic library with no dependencies.',
     long_description=long_description,
     long_description_content_type='text/markdown',
-    packages=find_packages(),
+    packages=find_packages(exclude=["tests"]),
     requires=[],
     install_requires=[],
     keywords='natural html generator compiler template-less',
