diff --git a/tests/test_frontend.py b/tests/test_frontend.py
index 46b4860..56cdc9b 100644
--- a/tests/test_frontend.py
+++ b/tests/test_frontend.py
@@ -309,6 +309,7 @@ def test_backend_build_editable_bad(tmp_path: Path, demo_pkg_inline: Path, monke
     assert exc.exc_msg == "'build_editable' on 'build' returned 1 but expected type <class 'str'>"
 
 
+@pytest.mark.skip(reason="who·uses·python·2?")
 def test_can_build_on_python_2(demo_pkg_inline: Path, tmp_path: Path) -> None:
     fronted = SubprocessFrontend(*SubprocessFrontend.create_args_from_folder(demo_pkg_inline)[:-1])
     env = session_via_cli(["-p", "2.7", str(tmp_path / "venv")])
