diff --git a/tests/test_frontend_setuptools.py b/tests/test_frontend_setuptools.py
index 4f22816..194ccc7 100644
--- a/tests/test_frontend_setuptools.py
+++ b/tests/test_frontend_setuptools.py
@@ -74,7 +74,7 @@ def test_setuptools_prepare_metadata_for_build_wheel(frontend_setuptools: Subpro
     assert dist.metadata["Name"] == "demo"
     values = [v for k, v in dist.metadata.items() if k == "Requires-Dist"]  # type: ignore[attr-defined]
     # ignore because "PackageMetadata" has no attribute "items"
-    expected = ["magic>3", "requests>2"] if sys.version_info[0:2] > (3, 8) else ["magic >3", "requests >2"]
+    expected = ["magic >3", "requests >2"]
     assert sorted(values) == expected
     assert isinstance(result.out, str)
     assert isinstance(result.err, str)
