--- chromium-13.0.754.0.orig/chrome/browser/debugger/devtools_remote_listen_socket.cc
+++ chromium-13.0.754.0/chrome/browser/debugger/devtools_remote_listen_socket.cc
@@ -157,11 +157,7 @@
   static const std::string kHandshakeString = "ChromeDevToolsHandshake";
   switch (state_) {
     case HANDSHAKE:
-      if (protocol_field_.compare(kHandshakeString)) {
-        state_ = INVALID;
-      } else {
-        Send(kHandshakeString, true);
-      }
+      Send(kHandshakeString, true);
       break;
     case HEADERS: {
       if (!protocol_field_.empty()) {  // not end-of-headers
