--- chromium-17.0.942.0.orig/chrome/browser/idle_query_linux.cc
+++ chromium-17.0.942.0/chrome/browser/idle_query_linux.cc
@@ -4,30 +4,12 @@
 
 #include "chrome/browser/idle_query_linux.h"
 
-#include <X11/extensions/scrnsaver.h>
-#include "ui/base/x/x11_util.h"
-
 namespace browser {
 
 class IdleData {
  public:
-  IdleData() {
-    int event_base;
-    int error_base;
-    if (XScreenSaverQueryExtension(ui::GetXDisplay(), &event_base,
-                                   &error_base)) {
-      mit_info = XScreenSaverAllocInfo();
-    } else {
-      mit_info = NULL;
-    }
-  }
-
-  ~IdleData() {
-    if (mit_info)
-      XFree(mit_info);
-  }
-
-  XScreenSaverInfo *mit_info;
+  IdleData() {}
+  ~IdleData() {}
 };
 
 IdleQueryLinux::IdleQueryLinux() : idle_data_(new IdleData()) {}
@@ -35,16 +17,7 @@
 IdleQueryLinux::~IdleQueryLinux() {}
 
 int IdleQueryLinux::IdleTime() {
-  if (!idle_data_->mit_info)
-    return 0;
-
-  if (XScreenSaverQueryInfo(ui::GetXDisplay(),
-                            RootWindow(ui::GetXDisplay(), 0),
-                            idle_data_->mit_info)) {
-    return (idle_data_->mit_info->idle) / 1000;
-  } else {
-    return 0;
-  }
+  return 0;
 }
 
 }  // namespace browser
--- chromium-17.0.942.0.orig/chrome/chrome.gyp
+++ chromium-17.0.942.0/chrome/chrome.gyp
@@ -530,11 +530,6 @@
           'dependencies': [
             '../build/linux/system.gyp:gtk',
           ],
-          'link_settings': {
-            'libraries': [
-              '-lXss',
-            ],
-          },
         }],
         ['OS=="linux" and chromeos==1', {
           'include_dirs': [
--- chromium-17.0.942.0.orig/chrome/chrome_common.gypi
+++ chromium-17.0.942.0/chrome/chrome_common.gypi
@@ -257,7 +257,6 @@
             'libraries': [
               '-lX11',
               '-lXrender',
-              '-lXss',
               '-lXext',
             ],
           },
--- chromium-17.0.942.0.orig/chrome/chrome_browser.gypi
+++ chromium-17.0.942.0/chrome/chrome_browser.gypi
@@ -4260,11 +4260,6 @@
           ],
         }],
         ['os_posix == 1 and OS != "mac"', {
-          'link_settings': {
-            'libraries': [
-              '-lXss',
-            ],
-          },
           'conditions': [
             ['linux_breakpad==1', {
               'sources': [
