From fb41faec1a32ff11699eaa28e74411b644ed4ec7 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 20 Mar 2026 14:00:19 +0100
Subject: [PATCH 2/3] chore: prettify CMake usage

---
 CMakeLists.txt     | 12 ++++++++----
 src/CMakeLists.txt | 13 +++++++++----
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd204d4..4c6e983 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,12 +17,16 @@ include(AsteroidTranslations)
 
 add_subdirectory(src)
 
-configure_file(${CMAKE_CURRENT_SOURCE_DIR}/asteroid-calculator.in
+configure_file(
+	${CMAKE_CURRENT_SOURCE_DIR}/asteroid-calculator.in
 	${CMAKE_BINARY_DIR}/asteroid-calculator
-	@ONLY)
+	@ONLY
+)
 
-install(PROGRAMS ${CMAKE_BINARY_DIR}/asteroid-calculator
-	DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(
+	PROGRAMS ${CMAKE_BINARY_DIR}/asteroid-calculator
+	DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 
 build_translations(i18n)
 generate_desktop(${CMAKE_SOURCE_DIR} asteroid-calculator)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cb23ff5..df2154f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,8 +1,13 @@
 add_library(asteroid-calculator main.cpp resources.qrc)
 set_target_properties(asteroid-calculator PROPERTIES PREFIX "")
 
-target_link_libraries(asteroid-calculator PUBLIC
-	AsteroidApp)
+target_link_libraries(
+	asteroid-calculator
+	PUBLIC
+		AsteroidApp
+)
 
-install(TARGETS asteroid-calculator
-	DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(
+	TARGETS asteroid-calculator
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}
+)
-- 
2.53.0

