diff --git a/asteroid-calculator.pro b/asteroid-calculator.pro
index 1423169..6bb7d18 100644
--- a/asteroid-calculator.pro
+++ b/asteroid-calculator.pro
@@ -1,5 +1,6 @@
 TARGET = asteroid-calculator
-CONFIG += asteroidapp
+CONFIG += asteroidapp link_pkgconfig
+PKGCONFIG += asteroidapp
 
 SOURCES +=     main.cpp
 RESOURCES +=   resources.qrc
@@ -10,3 +11,6 @@ OTHER_FILES += calculator.js \
 
 lupdate_only{ SOURCES += i18n/asteroid-calculator.desktop.h }
 TRANSLATIONS = $$files(i18n/$$TARGET.*.ts)
+
+target.path = /usr/bin/
+INSTALLS += target
