--- a/libunwind/src/CMakeLists.txt
+++ b/libunwind/src/CMakeLists.txt
@@ -123,7 +123,7 @@
   if(COMMAND llvm_setup_rpath)
     llvm_setup_rpath(unwind_shared)
   endif()
-  target_link_libraries(unwind_shared PRIVATE ${LIBUNWIND_LIBRARIES})
+  target_link_libraries(unwind_shared PRIVATE "${LIBUNWIND_LIBRARIES} -lssp_nonshared")
   set_target_properties(unwind_shared PROPERTIES
     CXX_EXTENSIONS OFF
     CXX_STANDARD 11
@@ -148,7 +148,7 @@
   else()
     target_compile_options(unwind_static PRIVATE -fno-rtti)
   endif()
-  target_link_libraries(unwind_static PRIVATE ${LIBUNWIND_LIBRARIES})
+  target_link_libraries(unwind_static PRIVATE "${LIBUNWIND_LIBRARIES} -lssp_nonshared")
   set_target_properties(unwind_static PROPERTIES
     CXX_EXTENSIONS OFF
     CXX_STANDARD 11
