# Fix web application directory to match Alpine Linux standards
diff -ruN a/Makefile b/Makefile
--- a/Makefile	2024-03-11 14:57:43.678248621 +0100
+++ b/Makefile	2024-03-11 15:04:24.448068860 +0100
@@ -1,8 +1,8 @@
 prefix = /usr
-datadir = ${prefix}/share
+datadir = ${prefix}/share/webapps
 gepdir = ${datadir}/grommunio-error-pages
 cssdir = ${gepdir}/_errors
-nglocdir = ${datadir}/grommunio-common/nginx/locations.d
+nglocdir = ${prefix}/share/grommunio-common/nginx/locations.d
 INSTALL = install
 MKDIR_P = mkdir -p
 
