diff --git a/moby-26.1.0/oci/fixtures/default.json b/moby-26.1.0/oci/fixtures/default.json
index 8d4d211..b63bfc9 100644
--- a/moby-26.1.0/oci/fixtures/default.json
+++ b/moby-26.1.0/oci/fixtures/default.json
@@ -47,6 +47,10 @@
 			"subArchitectures": [
 				"SCMP_ARCH_S390"
 			]
+		},
+		{
+			"architecture": "SCMP_ARCH_LOONGARCH64",
+			"subArchitectures": null
 		}
 	],
 	"syscalls": [
@@ -810,4 +814,4 @@
 			"excludes": {}
 		}
 	]
-}
\ No newline at end of file
+}
diff --git a/moby-26.1.0/profiles/seccomp/default.json b/moby-26.1.0/profiles/seccomp/default.json
index c4d9110..3501693 100644
--- a/moby-26.1.0/profiles/seccomp/default.json
+++ b/moby-26.1.0/profiles/seccomp/default.json
@@ -52,6 +52,10 @@
 		{
 			"architecture": "SCMP_ARCH_RISCV64",
 			"subArchitectures": null
+		},
+		{
+			"architecture": "SCMP_ARCH_LOONGARCH64",
+			"subArchitectures": null
 		}
 	],
 	"syscalls": [
@@ -830,4 +834,4 @@
 			}
 		}
 	]
-}
\ No newline at end of file
+}
diff --git a/moby-26.1.0/profiles/seccomp/default_linux.go b/moby-26.1.0/profiles/seccomp/default_linux.go
index 09fb337..c0c5e31 100644
--- a/moby-26.1.0/profiles/seccomp/default_linux.go
+++ b/moby-26.1.0/profiles/seccomp/default_linux.go
@@ -39,6 +39,10 @@ func arches() []Architecture {
 			Arch:      specs.ArchRISCV64,
 			SubArches: nil,
 		},
+		{
+			Arch:      specs.ArchLOONGARCH64,
+			SubArches: nil,
+		},
 	}
 }
 
diff --git a/moby-26.1.0/profiles/seccomp/seccomp_linux.go b/moby-26.1.0/profiles/seccomp/seccomp_linux.go
index 4d8fed6..9eb0741 100644
--- a/moby-26.1.0/profiles/seccomp/seccomp_linux.go
+++ b/moby-26.1.0/profiles/seccomp/seccomp_linux.go
@@ -41,6 +41,7 @@ var nativeToSeccomp = map[string]specs.Arch{
 	"ppc64le":     specs.ArchPPC64LE,
 	"s390":        specs.ArchS390,
 	"s390x":       specs.ArchS390X,
+	"loong64":     specs.ArchLOONGARCH64,
 }
 
 // GOARCH => libseccomp string
@@ -59,6 +60,7 @@ var goToNative = map[string]string{
 	"ppc64le":     "ppc64le",
 	"s390":        "s390",
 	"s390x":       "s390x",
+	"loong64":     "loong64",
 }
 
 // inSlice tests whether a string is contained in a slice of strings or not.
diff --git a/moby-26.1.0/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go b/moby-26.1.0/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go
index 4e7717d..96e04af 100644
--- a/moby-26.1.0/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go
+++ b/moby-26.1.0/vendor/github.com/opencontainers/runtime-spec/specs-go/config.go
@@ -741,6 +741,7 @@ const (
 	ArchPARISC      Arch = "SCMP_ARCH_PARISC"
 	ArchPARISC64    Arch = "SCMP_ARCH_PARISC64"
 	ArchRISCV64     Arch = "SCMP_ARCH_RISCV64"
+	ArchLOONGARCH64 Arch = "SCMP_ARCH_LOONGARCH64"
 )
 
 // LinuxSeccompAction taken upon Seccomp rule match
