diff --git a/components/cli/scripts/build/dynbinary b/components/cli/scripts/build/dynbinary
index 4feb7e7..46d910b 100755
--- a/components/cli/scripts/build/dynbinary
+++ b/components/cli/scripts/build/dynbinary
@@ -9,6 +9,6 @@ source ./scripts/build/.variables
 
 echo "Building dynamically linked $TARGET"
 export CGO_ENABLED=1
-go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" -buildmode=pie "${SOURCE}"
+go build -o "${TARGET}" -tags pkcs11 --ldflags "${LDFLAGS}" -buildmode=${BUILDMODE:=pie} "${SOURCE}"
 
 ln -sf "$(basename "${TARGET}")" build/docker
