--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -837,8 +837,8 @@ IF(ENABLE_XLSX)
           MESSAGE(STATUS "Using system-provided QXlsx ${QXlsx_VERSION}")
      ELSE()
           CPMAddPackage(NAME QXlsxQt${QT_VERSION_MAJOR}
-               URL https://github.com/QtExcel/QXlsx/archive/refs/tags/v1.5.0.tar.gz
-               URL_HASH SHA256=4461bb0fc9b88b6a970520e8b32eb7d2fc391a1a5c326b139a0af836278cdb09
+               URL https://github.com/QtExcel/QXlsx/archive/90d762625750c6b2c73f6cd96b633e9158aed72e.tar.gz
+               URL_HASH SHA256=61aa4b574fdfc2fb4b8bdcdb9580c06b9139c6c97aaffae29304cfb6e11175c6
                EXCLUDE_FROM_ALL ON
                SOURCE_SUBDIR QXlsx)
      ENDIF()
