diff --git a/wifi/scan.py b/wifi/scan.py
index 29f61ad..43f0af7 100644
--- a/wifi/scan.py
+++ b/wifi/scan.py
@@ -35,7 +35,7 @@ class Cell(object):
         Returns a list of all cells extracted from the output of iwlist.
         """
         try:
-            iwlist_scan = subprocess.check_output(['/sbin/iwlist', interface, 'scan'],
+            iwlist_scan = subprocess.check_output(['/usr/sbin/iwlist', interface, 'scan'],
                                                   stderr=subprocess.STDOUT)
         except subprocess.CalledProcessError as e:
             raise InterfaceError(e.output.strip())

