--- ./src/notification.c
+++ ./src/notification.c
@@ -30,6 +30,9 @@
 
 #ifdef HAVE_LIBNOTIFY
 #include <libnotify/notify.h>
+#ifndef NOTIFY_CHECK_VERSION
+#define NOTIFY_CHECK_VERSION(x,y,z) 0
+#endif
 #endif
 
 #include "zenity.h"
@@ -184,10 +187,14 @@
             icon = freeme = g_filename_to_uri (icon_file, NULL, NULL);
           }
 
+#if NOTIFY_CHECK_VERSION (0, 7, 0)
+          /* notify_notification_new_with_status_icon was removed */
+#else
           notif = notify_notification_new_with_status_icon (
                               message[0] /* title */,
                               message[1] /* summary */,
                               icon, status_icon);
+#endif
 
           g_strfreev (message);
           g_free (freeme);
