--- ./configure.in.orig
+++ ./configure.in
@@ -43,7 +43,6 @@
 AC_ISC_POSIX
 AC_PROG_CPP
 AM_PROG_CC_STDC
-AM_C_PROTOTYPES
 AC_PROG_YACC
 fp_PROG_ECHO
 AC_PROG_GPERF
--- ./lib/Makefile.am.orig
+++ ./lib/Makefile.am
@@ -17,7 +17,6 @@
 
 ## Process this file with automake to produce Makefile.in.
 ## This seems to make problems with some makes
-AUTOMAKE_OPTIONS = $(top_builddir)/lib/ansi2knr
 
 #
 # Definition of the local target
--- ./src/Makefile.am.orig
+++ ./src/Makefile.am
@@ -23,9 +23,6 @@
 
 ## Process this file with automake to produce Makefile.in
 
-## Since this package is written in ansi, be ready to un-ansify
-AUTOMAKE_OPTIONS = $(top_builddir)/lib/ansi2knr
-
 bin_PROGRAMS = a2ps
 
 DEFS = @DEFS@ -DLOCALEDIR=\"$(datadir)/locale\"
--- ./contrib/sample/Makefile.am.orig
+++ ./contrib/sample/Makefile.am
@@ -25,8 +25,6 @@
 
 ## Process this file with automake to produce Makefile.in
 
-## Since this package is written in ansi, be ready to un-ansify
-AUTOMAKE_OPTIONS = $(top_builddir)/lib/ansi2knr
 localedir = $(datadir)/locale
 
 noinst_PROGRAMS = sample
