--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -2460,6 +2460,11 @@
         }, {
           'toolsets': ['target'],
         }],
+        [ 'host_arch=="mips" or host_arch=="mipsel"', {
+          'link_settings': {
+            'libraries': [ '-latomic' ],
+          },
+        }],
         ['component=="shared_library"', {
           'direct_dependent_settings': {
             'defines': [ 'USING_V8_PLATFORM_SHARED' ],
--- a/node.gyp
+++ b/node.gyp
@@ -315,6 +315,11 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ 'host_arch=="mips" or host_arch=="mipsel"', {
+          'link_settings': {
+            'libraries': [ '-latomic' ],
+          },
+        }],
         [ 'node_intermediate_lib_type=="static_library" and '
             'node_shared=="true" and OS=="aix"', {
           # For AIX, shared lib is linked by static lib and .exp. In the
