--- a/deps/v8/gypfiles/v8.gyp
+++ b/deps/v8/gypfiles/v8.gyp
@@ -2012,6 +2012,11 @@
         }],
       ],
       'conditions': [
+        [ 'host_arch=="mips" or host_arch=="mipsel"', {
+          'link_settings': {
+            'libraries': [ '-latomic' ],
+          },
+        }],
         ['component=="shared_library"', {
           'defines': [
             'BUILDING_V8_BASE_SHARED',
--- a/node.gyp
+++ b/node.gyp
@@ -479,6 +479,11 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ 'host_arch=="mips" or host_arch=="mipsel"', {
+          'link_settings': {
+            'libraries': [ '-latomic' ],
+          },
+        }],
         [ 'node_code_cache_path!=""', {
           'sources': [ '<(node_code_cache_path)' ]
         }, {
