diff --git a/common.gypi b/common.gypi
index fec31779ece..7466bd929c4 100644
--- a/common.gypi
+++ b/common.gypi
@@ -528,7 +528,7 @@
               [ 'host_arch=="ppc64" and OS not in "aix os400"', {
                 'conditions': [
                   [ 'clang==0', {
-                    'cflags': [ '-mminimal-toc' ],
+                    'cflags': [ '-mcmodel=large' ],
                   }],
                 ],
                 'cflags': [ '-m64' ],
@@ -553,7 +553,7 @@
               [ 'target_arch=="ppc64" and OS not in "aix os400"', {
                 'conditions': [
                   [ 'clang==0', {
-                    'cflags': [ '-mminimal-toc' ],
+                    'cflags': [ '-mcmodel=large' ],
                   }],
                 ],
                 'cflags': [ '-m64' ],
