From: Johannes Müller <alpinelinux@jogemu.at>
Date: Fri Jul 12 11:33:42 2026 +0000
Subject: Resolve undefined reference to FloatMinMaxHelper

--- a/deps/v8/src/maglev/riscv/maglev-ir-riscv.cc
+++ b/deps/v8/src/maglev/riscv/maglev-ir-riscv.cc
@@ -881,7 +881,7 @@
   DoubleRegister right = ToDoubleRegister(RightInput());
   DoubleRegister out = ToDoubleRegister(result());
 
-  __ FloatMinMaxHelper<double>(out, left, right, MaxMinKind::kMin);
+  __ Float64Min(out, left, right);
 }
 
 void Float64Max::SetValueLocationConstraints() {
@@ -896,7 +896,7 @@
   DoubleRegister right = ToDoubleRegister(RightInput());
   DoubleRegister out = ToDoubleRegister(result());
 
-  __ FloatMinMaxHelper<double>(out, left, right, MaxMinKind::kMax);
+  __ Float64Max(out, left, right);
 }
 
 int Float64Ieee754Unary::MaxCallStackArgs() const { return 0; }
