From: Johannes Müller <50362-jogemu@users.gitlab.alpinelinux.org>
Date: Fri Oct 31 07:00:11 2025 +0000
Subject: Fix x86 build because _mm_cvtsi128_si64 is x64 only

--- a/deps/v8/src/strings/string-hasher.cc
+++ b/deps/v8/src/strings/string-hasher.cc
@@ -23,7 +23,7 @@
     DCHECK_LE(p[5], 0xff);
     DCHECK_LE(p[6], 0xff);
     DCHECK_LE(p[7], 0xff);
-#ifdef __SSE2__
+#if defined(__SSE2__) && defined(__x86_64__)
     __m128i x = _mm_loadu_si128(reinterpret_cast<const __m128i*>(p));
     return _mm_cvtsi128_si64(_mm_packus_epi16(x, x));
 #elif defined(__ARM_NEON__)
@@ -44,7 +44,7 @@
     DCHECK_LE(p[1], 0xff);
     DCHECK_LE(p[2], 0xff);
     DCHECK_LE(p[3], 0xff);
-#ifdef __SSE2__
+#if defined(__SSE2__) && defined(__x86_64__)
     __m128i x = _mm_loadu_si64(reinterpret_cast<const __m128i*>(p));
     return _mm_cvtsi128_si64(_mm_packus_epi16(x, x));
 #elif defined(__ARM_NEON__)
