--- origin/src/drivers/main.cpp
+++ mednafen/src/drivers/main.cpp
@@ -696,7 +696,7 @@
 	if(argc <= 1)
 	{
 	 printf(_("No command-line arguments specified.\n\n"));
-	 printf(usage_string, argv[0]);
+	 printf(usage_string, "mednafen");
 	 printf(_("\tPlease refer to the documentation for option parameters and usage.\n\n"));
 	 return(0);
 	}
@@ -713,11 +713,11 @@
 
 	 if(ShowCLHelp)
 	 {
-          printf(usage_string, argv[0]);
+          printf(usage_string, "mednafen");
           ShowArgumentsHelp(MDFNArgs, false);
 	  printf("\n");
 	  printf(_("Each setting(listed in the documentation) can also be passed as an argument by prefixing the name with a hyphen,\nand specifying the value to change the setting to as the next argument.\n\n"));
-	  printf(_("For example:\n\t%s -pce.stretch aspect -pce.pixshader autoipsharper \"Hyper Bonk Soldier.pce\"\n\n"), argv[0]);
+	  printf(_("For example:\n\tmednafen -pce.stretch aspect -pce.pixshader autoipsharper \"Hyper Bonk Soldier.pce\"\n\n"));
 	  printf(_("Settings specified in this manner are automatically saved to the configuration file, hence they\ndo not need to be passed to future invocations of the Mednafen executable.\n"));
 	  printf("\n");
 	  return(0);
