diff -ru mednafen.orig/src/drivers/video.cpp mednafen/src/drivers/video.cpp
--- mednafen.orig/src/drivers/video.cpp	2015-02-23 21:06:16.000000000 +0000
+++ mednafen/src/drivers/video.cpp	2015-04-10 05:38:20.784481562 +0000
@@ -147,7 +147,7 @@
 
 static MDFNSetting GlobalVideoSettings[] =
 {
- { "video.driver", MDFNSF_NOFLAGS, gettext_noop("Video output method/driver."), NULL, MDFNST_ENUM, "opengl", NULL, NULL, NULL, NULL, VDriver_List },
+ { "video.driver", MDFNSF_NOFLAGS, gettext_noop("Video output method/driver."), NULL, MDFNST_ENUM, "sdl", NULL, NULL, NULL, NULL, VDriver_List },
 
  { "video.fs", MDFNSF_NOFLAGS, gettext_noop("Enable fullscreen mode."), NULL, MDFNST_BOOL, "0", },
  { "video.glvsync", MDFNSF_NOFLAGS, gettext_noop("Attempt to synchronize OpenGL page flips to vertical retrace period."), 
