From e795d3eeb099e77a9e8540e26bcfab5dd0260c79 Mon Sep 17 00:00:00 2001
From: Andy Postnikov <apostnikov@gmail.com>
Date: Fri, 2 Nov 2018 15:44:25 +0200
Subject: [PATCH] Make spx_ui_assets_dir respect INSTALL_ROOT

---
 Makefile.frag | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.frag b/Makefile.frag
index a042552..9f3ae0b 100644
--- a/Makefile.frag
+++ b/Makefile.frag
@@ -1,5 +1,5 @@
 
-spx_ui_assets_dir = $(prefix)/share/misc/php-spx/assets/web-ui
+spx_ui_assets_dir = $(INSTALL_ROOT)$(prefix)/share/misc/php-spx/assets/web-ui
 
 CFLAGS += -DSPX_HTTP_UI_ASSETS_DIR=\"$(spx_ui_assets_dir)\"
 
