Index: configure.ac
==================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -151,10 +151,11 @@
 	;;
 esac
 AC_PROG_OBJC($potential_compilers)
 AC_PROG_OBJCPP
 AC_PROG_EGREP
+AC_PROG_SED
 AC_PROG_LN_S
 
 BUILDSYS_CHECK_IOS
 
 AC_ARG_WITH(wii,
@@ -596,19 +597,19 @@
 			} @finally {
 				foo();
 			}
 		])
 	], [
-		AS_IF([tr -dc "[[:print:]]" <conftest.$ac_objext | \
+		AS_IF([$SED 's/[[^[:print:]]]//g' <conftest.$ac_objext | \
 		    $EGREP __gnu_objc_personality_v0 >/dev/null], [
 			exception_type="DWARF"
 		])
-		AS_IF([tr -dc "[[:print:]]" <conftest.$ac_objext | \
+		AS_IF([$SED 's/[[^[:print:]]]//g' <conftest.$ac_objext | \
 		    $EGREP __gnu_objc_personality_sj0 >/dev/null], [
 			exception_type="SjLj"
 		])
-		AS_IF([tr -dc "[[:print:]]" <conftest.$ac_objext | \
+		AS_IF([$SED 's/[[^[:print:]]]//g' <conftest.$ac_objext | \
 		    $EGREP __gnu_objc_personality_seh0 >/dev/null], [
 			exception_type="SEH"
 		])
 
 		case "$exception_type" in
@@ -816,18 +817,18 @@
 	AC_COMPILE_IFELSE([
 		AC_LANG_SOURCE([
 			double endianess = 2.993700760838795055656993580068609688772747263874402942272934826871811872228512759832626847251963763755836687759498519784550143745834860002945223766052808125982053455555265216112722718870586961456110693379343178124592311441022662940307099598578775368547768968914916965731708568179631324904813506101190853720749196062963892799499230635163056742330563321122389331703618066046034494287335316842529021563862331183541255013987734473643350285400060357711238514186776429325214739886098119655678483017894951556639821088508565036657794343031121375178126860889964700274558728491825977274341798997758923017217660272136611938897932105874133412726223468780517578125e-259;
 		])
 	], [
-		AS_IF([tr -dc "[[:print:]]" <conftest.$ac_objext | \
+		AS_IF([$SED 's/[[^[:print:]]]//g' <conftest.$ac_objext | \
 		    $EGREP BigEnd >/dev/null], [
 			AC_DEFINE(OF_FLOAT_BIG_ENDIAN, 1,
 				[Whether floats are big endian])
 			fp_endianess="big endian"
 		], [
-			AS_IF([tr -dc "[[:print:]]" <conftest.$ac_objext | \
-			    $EGREP dnEgiB >/dev/null], [
+			AS_IF([$SED 's/[[^[:print:]]]//g' \
+			    <conftest.$ac_objext | $EGREP dnEgiB >/dev/null], [
 				fp_endianess="little endian"
 			])
 		])
 	])
 ], [

