--- a/meson.build
+++ b/meson.build
@@ -21,6 +21,7 @@ cargo_home = meson.project_build_root() / 'cargo-home'
 cargo_manifest = meson.project_source_root() / 'Cargo.toml'
 
 cargo_bin = find_program('cargo')
+find_program('cargo-auditable')
 
 cargo_env = environment()
 cargo_env.set('CARGO_HOME', cargo_home)
@@ -34,6 +35,7 @@ cargo_build = custom_target(
   env: cargo_env,
   command: [
     cargo_bin,
+    'auditable',
     'build',
     '--target-dir', cargo_target_dir,
     '--profile', cargo_profile,
