diff --git a/Makefile b/Makefile
index 92522fb..fb2d3f5 100644
--- a/Makefile
+++ b/Makefile
@@ -3,11 +3,11 @@
 NAME       = pure-maps
 VERSION    = 1.21.1
 RELEASE    = $(NAME)-$(VERSION)
-DESTDIR    =
 PREFIX     = /usr
 EXEDIR     = $(DESTDIR)$(PREFIX)/bin
 EXE        = $(EXEDIR)/$(NAME)
 DATADIR    = $(DESTDIR)$(PREFIX)/share/$(NAME)
+LANGDIR    = $(DESTDIR)$(PREFIX)/share/locale
 DESKTOPDIR = $(DESTDIR)$(PREFIX)/share/applications
 ICONDIR    = $(DESTDIR)$(PREFIX)/share/icons/hicolor
 METADIR    = $(DESTDIR)$(PREFIX)/share/metainfo
@@ -19,8 +19,8 @@ LCONVERT   = $(or $(wildcard /usr/lib/qt5/bin/lconvert),\
 
 define install-translation =
     # GNU gettext translations for Python use.
-    mkdir -p $(DATADIR)/locale/$(1)/LC_MESSAGES
-    msgfmt po/$(1).po -o $(DATADIR)/locale/$(1)/LC_MESSAGES/pure-maps.mo
+    mkdir -p $(LANGDIR)/$(1)/LC_MESSAGES
+    msgfmt po/$(1).po -o $(LANGDIR)/$(1)/LC_MESSAGES/pure-maps.mo
     # Qt linguist translations for QML use.
     mkdir -p $(DATADIR)/translations
     $(LCONVERT) -o $(DATADIR)/translations/$(NAME)-$(1).qm po/$(1).po
