diff -r d72695b410e3 defs.sh
--- a/defs.sh   Sun Jul 26 15:38:58 2015 -0400
+++ b/defs.sh   Sun Jan 31 23:18:18 2016 +0000
@@ -221,7 +221,7 @@
 gccprereqs() {
     if [ ! -e gcc-$GCC_VERSION/gmp ]
     then
-        fetchextract http://gmplib.org/download/gmp/ gmp-$GMP_VERSION .tar.bz2
+        fetchextract http://gmplib.org/download/gmp/archive/ gmp-$GMP_VERSION .tar.bz2
         mv gmp-$GMP_VERSION gcc-$GCC_VERSION/gmp
     fi

--- a/extra/build-gcc-deps.sh   Sun Jul 26 15:38:58 2015 -0400
+++ b/extra/build-gcc-deps.sh   Sun Jan 31 23:18:18 2016 +0000
@@ -28,7 +28,7 @@
 PREFIX="$CC_PREFIX/$TRIPLE"

 # GMP
-fetchextract ftp://ftp.gmplib.org/pub/gmp-$GMP_VERSION/ gmp-$GMP_VERSION .tar.bz2
+fetchextract ftp://ftp.gmplib.org/pub/archive/gmp-$GMP_VERSION/ gmp-$GMP_VERSION .tar.bz2
 cp -f "$MUSL_CC_BASE/extra/config.sub" gmp-$GMP_VERSION/configfsf.sub
 buildinstall '' gmp-$GMP_VERSION --host="$TRIPLE" --enable-static --disable-shared
