--- src/core/abi.cc.orig	2010-03-15 21:26:20.000000000 +0100
+++ src/core/abi.cc	2013-08-06 21:05:08.693772583 +0200
@@ -11,6 +11,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <dirent.h>
+#include <cstdlib>
 
 __ID("@(#) $Id: mem.cc 1352 2006-05-27 23:54:13Z ezix $");
 
@@ -38,10 +39,10 @@
     for(i=0; i<n; i++)
     {
       system.addCapability(namelist[i]->d_name);
-      free(namelist[i]);
+      std::free(namelist[i]);
     }
     if(namelist)
-      free(namelist);
+      std::free(namelist);
   }
 
   popd();
