--- a/setup.py
+++ b/setup.py
@@ -15,7 +15,7 @@
     long_description_content_type="text/markdown",
     url="https://github.com/thieu1995/opfunu",
     download_url="https://github.com/thieu1995/opfunu/archive/v0.8.0.zip",
-    packages=find_packages(),
+    packages=find_packages(exclude=['examples*']),
     include_package_data=True,
     license="MIT",
     classifiers=[
@@ -39,4 +39,4 @@
     ],
     install_requires=["numpy"],
     python_requires='>=3.6',
-)
\ No newline at end of file
+)
