diff --git a/pptp_compat.c b/pptp_compat.c
index 1760a3b..6ebbea4 100644
--- a/pptp_compat.c
+++ b/pptp_compat.c
@@ -2,6 +2,7 @@
  *
  */
 
+#if defined (__SVR4) && defined (__sun) /* Solaris */
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
@@ -14,8 +15,6 @@
 #include <stdio.h>
 #include "util.h"
 
-
-#if defined (__SVR4) && defined (__sun) /* Solaris */
 /*
  * daemon implementation from uClibc
  */
