--- tools/rpcgen/rpc_main.c     Tue Apr  7 21:53:12 2009
+++ tools/rpcgen/rpc_main.c	Tue Apr  7 21:53:12 2009
@@ -550,7 +550,7 @@
 	  f_print(fout, "#include <sys/ttycom.h>/* TIOCNOTTY */\n");
 #else
 	if( !tirpcflag )
-	  f_print(fout, "#include <sys/ttycom.h>/* TIOCNOTTY */\n");
+	  f_print(fout, "#include <sys/ioctl.h>/* TIOCNOTTY */\n");
 #endif
 	if( Cflag && (inetdflag || pmflag ) ) {
 	  f_print(fout, "#ifdef __cplusplus\n");
