diff --git a/cmake/SearchForStuff.cmake b/cmake/SearchForStuff.cmake
index ea84fd6..3d5c32b 100644
--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -191,7 +191,6 @@ endif()
 
 if(USE_VULKAN)
 	add_subdirectory(3rdparty/glslang EXCLUDE_FROM_ALL)
-	add_subdirectory(3rdparty/vulkan-headers EXCLUDE_FROM_ALL)
 endif()
 
 if(CUBEB_API)
diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index 2c3c9a8..cc215c4 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -146,7 +146,7 @@ endif()
 
 if(USE_VULKAN)
 	target_link_libraries(common PUBLIC
-		Vulkan-Headers glslang
+		glslang
 	)
 	target_sources(common PRIVATE
 		Vulkan/ShaderCache.cpp
