diff --git a/pyproject.toml.orig b/pyproject.toml
index 87b1df348ab..66e61a69da1 100755
--- a/pyproject.toml.orig
+++ b/pyproject.toml
@@ -24,9 +24,6 @@ classifiers = [
     "Topic :: Software Development :: Libraries :: Python Modules",
     "Topic :: System :: Systems Administration :: Authentication/Directory",
 ]
-include = [
-  "example/*"
-]
 
 [tool.poetry.urls]
 repository = "https://github.com/pyradius/pyrad"
