--- espeak-1.40.02-source.orig/src/Makefile	2008-12-26 10:11:49 +0000
+++ espeak-1.40.02-source/src/Makefile	2008-12-26 10:27:48 +0000
@@ -60,15 +60,15 @@
 
 SRCS1=$(speak_SOURCES)
 OBJS1=$(patsubst %.cpp,%.o,$(SRCS1))
-LIBS1=-lstdc++ $(LIB_AUDIO) -lpthread $(EXTRA_LIBS)
+LIBS1=-lm $(LIB_AUDIO) -lpthread $(EXTRA_LIBS)
 
 SRCS2=$(libespeak_SOURCES)
 OBJS2=$(patsubst %.cpp,x_%.o,$(SRCS2))
-LIBS2=-lstdc++ $(LIB_AUDIO) -lpthread
+LIBS2=-lm $(LIB_AUDIO) -lpthread
 
 SRCS3 = espeak.cpp
 OBJS3=$(patsubst %.cpp,%.o,$(SRCS3))
-LIBS3=-lstdc++ -L . -lespeak
+LIBS3=-L . -lespeak -lpthread
 
 CXXFLAGS=-O2
 
