--- tdesktop-3.1.9-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp
+++ tdesktop-3.1.9-full-patched/Telegram/SourceFiles/platform/linux/specific_linux.cpp
@@ -47,11 +47,7 @@
 
 #include <private/qguiapplication_p.h>
 
-#ifdef Q_OS_FREEBSD
-#include <malloc_np.h>
-#else // Q_OS_FREEBSD
-#include <jemalloc/jemalloc.h>
-#endif // Q_OS_FREEBSD
+#include <malloc.h>
 
 #ifndef DESKTOP_APP_DISABLE_DBUS_INTEGRATION
 #include <glibmm.h>
--- tdesktop-3.1.9-full/cmake/external/CMakeLists.txt
+++ tdesktop-3.1.9-full-patched/cmake/external/CMakeLists.txt
@@ -37,9 +37,6 @@
 	add_checked_subdirectory(hunspell)
 endif()
 add_checked_subdirectory(iconv)
-if (LINUX)
-    add_checked_subdirectory(jemalloc)
-endif()
 add_checked_subdirectory(jpeg)
 if (LINUX AND NOT DESKTOP_APP_DISABLE_WAYLAND_INTEGRATION)
     add_checked_subdirectory(kwayland)
--- a/cmake/options_linux.cmake	2022-01-14 20:40:40.000000000 +0000
+++ b/options_linux.cmake	2022-01-17 10:12:14.971307520 +0000
@@ -63,12 +63,6 @@ if (DESKTOP_APP_SPECIAL_TARGET)
     target_link_options(common_options INTERFACE $<IF:$<CONFIG:Debug>,,-g -flto -fuse-linker-plugin>)
 endif()
 
-target_link_libraries(common_options
-INTERFACE
-    desktop-app::external_jemalloc
-    ${CMAKE_DL_LIBS}
-)
-
 if (DESKTOP_APP_USE_ALLOCATION_TRACER)
     target_link_options(common_options
     INTERFACE
--- tdesktop-3.1.9-full/Telegram/SourceFiles/platform/linux/specific_linux.cpp
+++ tdesktop-3.1.9-full-patched/Telegram/SourceFiles/platform/linux/specific_linux.cpp
@@ -657,9 +657,6 @@
 namespace Platform {
 
 void start() {
-	auto backgroundThread = true;
-	mallctl("background_thread", nullptr, nullptr, &backgroundThread, sizeof(bool));
-
 	// Prevent any later calls into setlocale() by Qt
 	QCoreApplicationPrivate::initLocale();
 
