disables forced linking against libdispatch

--- a/Telegram/lib_crl/CMakeLists.txt
+++ b/Telegram/lib_crl/CMakeLists.txt
@@ -70,12 +70,9 @@
 if (LINUX)
     target_link_libraries(lib_crl
     PUBLIC
-        desktop-app::external_dispatch
         pthread
     )
+elseif (build_macstore)
-endif()
-
-if (build_macstore OR LINUX)
     target_compile_definitions(lib_crl
     PUBLIC
         CRL_USE_COMMON_QUEUE
--- tdesktop-3.3.1-full/cmake/external/CMakeLists.txt
+++ tdesktop-3.3.1-full-patched/cmake/external/CMakeLists.txt
@@ -17,9 +17,6 @@
 if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
     add_checked_subdirectory(dbusmenu_qt)
 endif()
-if (LINUX)
-    add_checked_subdirectory(dispatch)
-endif()
 add_checked_subdirectory(expected)
 if (LINUX AND NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
     add_checked_subdirectory(fcitx_qt5)
