rebased paths -lnl

From 5ca91dbb811c84591780236abc31431e313faf39 Mon Sep 17 00:00:00 2001
From: Ilya Fedin <fedin-ilja2010@ya.ru>
Date: Sat, 23 Sep 2023 10:35:16 +0400
Subject: [PATCH] Add missing include for Qt 6.6

---
 base/algorithm.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Telegram/lib_base/base/algorithm.h b/Telegram/lib_base/base/algorithm.h
index 5c11285d..4d7be795 100644
--- a/Telegram/lib_base/base/algorithm.h
+++ b/Telegram/lib_base/base/algorithm.h
@@ -7,6 +7,7 @@
 #pragma once
 
 #include <QtCore/QLatin1String>
+#include <QtCore/QString>
 #include <memory>
 
 namespace base {
