From 213170ebcee0ad963c5671a012993df3204d992b Mon Sep 17 00:00:00 2001
From: Olivier Mauras <olivier@mauras.ch>
Date: Wed, 18 Mar 2015 08:29:43 +0100
Subject: [PATCH] Fix musl

---
 spectrwm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/spectrwm.c b/spectrwm.c
index b01ed60..cc16e87 100644
--- a/spectrwm.c
+++ b/spectrwm.c
@@ -8244,7 +8244,7 @@ setconfvalue(const char *selector, const char *value, int flags)
 			return (0);
 
 		bzero(s, sizeof s);
-		if (sscanf(value, "ws[%d]:%1023c", &ws_id, s) != 2)
+		if (sscanf(value, "ws[%d]:%1023s", &ws_id, s) != 2)
 			errx(1, "invalid entry, should be 'ws[<idx>]:name'");
 		ws_id--;
 		if (ws_id < 0 || ws_id >= workspace_limit)
@@ -8365,7 +8365,7 @@ setautorun(const char *selector, const char *value, int flags)
 		return (0);
 
 	bzero(s, sizeof s);
-	if (sscanf(value, "ws[%d]:%1023c", &ws_id, s) != 2)
+	if (sscanf(value, "ws[%d]:%1023s", &ws_id, s) != 2)
 		errx(1, "invalid autorun entry, should be 'ws[<idx>]:command'");
 	ws_id--;
 	if (ws_id < 0 || ws_id >= workspace_limit)
@@ -8435,7 +8435,7 @@ setlayout(const char *selector, const char *value, int flags)
 		return (0);
 
 	bzero(s, sizeof s);
-	if (sscanf(value, "ws[%d]:%d:%d:%d:%d:%1023c",
+	if (sscanf(value, "ws[%d]:%d:%d:%d:%d:%1023s",
 	    &ws_id, &mg, &ma, &si, &ar, s) != 6)
 		errx(1, "invalid layout entry, should be 'ws[<idx>]:"
 		    "<master_grow>:<master_add>:<stack_inc>:<always_raise>:"
-- 
2.2.1

