diff --git a/tests/test_encode.py b/tests/test_encode.py
index 429a9f0..b6ecf2a 100644
--- a/tests/test_encode.py
+++ b/tests/test_encode.py
@@ -22,3 +22,3 @@ class TestBasicInfo(TestCase):
         }
-        self.assertEquals(self.swagger['info'], expected)
+        self.assertEqual(self.swagger['info'], expected)
 
@@ -27,3 +27,3 @@ class TestBasicInfo(TestCase):
         expected = '2.0'
-        self.assertEquals(self.swagger['swagger'], expected)
+        self.assertEqual(self.swagger['swagger'], expected)
 
@@ -32,3 +32,3 @@ class TestBasicInfo(TestCase):
         expected = 'www.example.com'
-        self.assertEquals(self.swagger['host'], expected)
+        self.assertEqual(self.swagger['host'], expected)
 
@@ -37,3 +37,3 @@ class TestBasicInfo(TestCase):
         expected = ['https']
-        self.assertEquals(self.swagger['schemes'], expected)
+        self.assertEqual(self.swagger['schemes'], expected)
 
@@ -74,3 +74,3 @@ class TestPaths(TestCase):
         }
-        self.assertEquals(self.swagger['paths'][self.path]['get'], expected)
+        self.assertEqual(self.swagger['paths'][self.path]['get'], expected)
         expected = {
@@ -85,3 +85,3 @@ class TestPaths(TestCase):
         }
-        self.assertEquals(self.swagger['paths'][self.path]['post'], expected)
+        self.assertEqual(self.swagger['paths'][self.path]['post'], expected)
 
@@ -99,3 +99,3 @@ class TestParameters(TestCase):
     def test_expected_fields(self):
-        self.assertEquals(len(self.swagger), 1)
+        self.assertEqual(len(self.swagger), 1)
         expected = {
@@ -107,2 +107,2 @@ class TestParameters(TestCase):
         }
-        self.assertEquals(self.swagger[0], expected)
+        self.assertEqual(self.swagger[0], expected)
