--- a/terminal-table.gemspec
+++ b/terminal-table.gemspec
@@ -13,7 +13,7 @@
   spec.homepage      = "https://github.com/tj/terminal-table"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.files         = Dir["lib/**/*"]
   spec.require_paths = ["lib"]
 
   spec.add_development_dependency "bundler", "~> 2"
