Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: generate manpages
----

--- a/doc/Doxyfile
+++ b/doc/Doxyfile
@@ -1086,7 +1086,7 @@
 # need to set EXTENSION_MAPPING for the extension otherwise the files are not
 # properly processed by doxygen.

-FILTER_PATTERNS        = *.h=doxyfilter.sh
+FILTER_PATTERNS        = *.h=./doxyfilter.sh

 # If the FILTER_SOURCE_FILES tag is set to YES, the input filter (if set using
 # INPUT_FILTER) will also be used to filter the input files that are used for
@@ -2140,7 +2140,7 @@
 # classes and files.
 # The default value is: NO.

-GENERATE_MAN           = NO
+GENERATE_MAN           = YES

 # The MAN_OUTPUT tag is used to specify where the man pages will be put. If a
 # relative path is entered the value of OUTPUT_DIRECTORY will be put in front of
