diff --git a/setup.cfg b/setup.cfg
index f7abbc2..acbe7b3 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -42,9 +42,6 @@ install_requires =
 	webencodings
 tests_require = 
 	pytest-runner
-	pytest-cov
-	pytest-flake8
-	pytest-isort
 python_requires = >= 3.5
 
 [options.package_data]
@@ -57,9 +54,6 @@ doc =
 	sphinx_rtd_theme
 test = 
 	pytest-runner
-	pytest-cov
-	pytest-flake8
-	pytest-isort
 
 [bdist_wheel]
 python-tag = py3
@@ -72,7 +66,7 @@ build-dir = docs/_build
 test = pytest
 
 [tool:pytest]
-addopts = --flake8 --isort cssselect2
+addopts = cssselect2
 norecursedirs = build dist .cache .eggs .git
 
 [coverage:run]
