--- ./configure.ac.orig	2013-08-14 09:23:19.968703427 +0000
+++ ./configure.ac	2013-08-14 09:24:02.892913484 +0000
@@ -597,7 +597,7 @@
 else
   dnl preload c-client default locations/options
   case $host in
-    *-linux-gnu)
+    *-linux-gnu|*-linux-uclibc|*-linux-musl*)
       if test -f /etc/fedora-release -o -f /etc/redhat-release -o -f /etc/redhat_version ; then  
         alpine_SSLTYPE="nopwd"
 	if test -d /etc/pki/tls ; then  
@@ -1271,7 +1271,7 @@
 dnl build from c-client.  Most of this will go away when c-client
 dnl adopts configure
 case "$host" in
-  *-linux-gnu*|*-k*bsd*-gnu*)
+  *-linux-gnu*|*-k*bsd*-gnu*|*-linux-uclibc|*-linux-musl*)
     alpine_path_delim="/"
     alpine_mode_readonly="(0600)"
     if test -f /etc/fedora-release ; then  
