diff --git a/vendor/boringssl/crypto/fipsmodule/rand/getrandom_fillin.h b/vendor/boringssl/crypto/fipsmodule/rand/getrandom_fillin.h
index 0f290e9..3eac6d2 100644
--- a/vendor/boringssl/crypto/fipsmodule/rand/getrandom_fillin.h
+++ b/vendor/boringssl/crypto/fipsmodule/rand/getrandom_fillin.h
@@ -30,6 +30,8 @@
 #define EXPECTED_NR_getrandom 278
 #elif defined(OPENSSL_ARM)
 #define EXPECTED_NR_getrandom 384
+#elif defined(OPENSSL_LOONG64)
+#define EXPECTED_NR_getrandom 278
 #elif defined(OPENSSL_RISCV64)
 #define EXPECTED_NR_getrandom 278
 #endif
diff --git a/vendor/boringssl/include/openssl/target.h b/vendor/boringssl/include/openssl/target.h
index 2760f52..bc944d2 100644
--- a/vendor/boringssl/include/openssl/target.h
+++ b/vendor/boringssl/include/openssl/target.h
@@ -34,6 +34,9 @@
 #elif defined(__ARMEL__) || defined(_M_ARM)
 #define OPENSSL_32_BIT
 #define OPENSSL_ARM
+#elif defined(__loongarch64)
+#define OPENSSL_64_BIT
+#define OPENSSL_LOONG64
 #elif defined(__MIPSEL__) && !defined(__LP64__)
 #define OPENSSL_32_BIT
 #define OPENSSL_MIPS
