diff --git a/Makefile.am b/Makefile.am
index 7a28edc..15cc6d0 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -9,15 +9,11 @@ rdfind_SOURCES = rdfind.cc Checksum.cc  Dirlist.cc  Fileinfo.cc  Rdutil.cc \
 #these are the test scripts to execute - I do not know how to glob here,
 #feedback welcome.
 TESTS=testcases/largefilesupport.sh \
-      testcases/hardlink_fails.sh \
-      testcases/symlinking_action.sh \
       testcases/verify_filesize_option.sh \
-      testcases/verify_dryrun_option.sh \
       testcases/verify_ranking.sh \
       testcases/verify_deterministic_operation.sh \
       testcases/checksum_options.sh \
-      testcases/md5collisions.sh \
-      testcases/sha1collisions.sh
+      testcases/md5collisions.sh
 
 AUXFILES=testcases/common_funcs.sh \
          testcases/md5collisions/letter_of_rec.ps \
-
-
-
-
-
-
 set_more_warnings=no
 # Check whether --enable-warnings was given.
 if test "${enable_warnings+set}" = set; then :
diff --git a/configure.ac b/configure.ac
index c494cb4..e1f8d85 100644
--- a/configure.ac
+++ b/configure.ac
@@ -70,8 +70,6 @@ AC_ARG_ENABLE(cppstandardcheck,
 	[ ENABLECPPSTANDARDCHECK=yes] )
 if test "$ENABLECPPSTANDARDCHECK" = "yes"; then
    AC_MSG_NOTICE([checking for c++ 11, disable with --disable-cppstandardcheck])
-   dnl AX_CXX_COMPILE_STDCXX_11([noext], [mandatory])
-   AX_CXX_COMPILE_STDCXX([11],[noext],[mandatory])
 else
    AC_MSG_NOTICE([not checking for c++11 support.])
 fi
