--- a/src/target/legacy_py.rs.orig
+++ b/src/target/legacy_py.rs
@@ -60,7 +60,7 @@
 pub(super) static ANDROID_ARCHES: &[&str] = &["armeabi_v7a", "arm64_v8a", "x86", "x86_64"];
 
 pub(super) static MANYLINUX_ARCHES: &[&str] = &[
-    "x86_64", "i686", "aarch64", "armv7l", "ppc64le", "s390x", "ppc64", "riscv64",
+    "x86_64", "i686", "aarch64", "armv7l", "armv6l","ppc64le", "s390x", "ppc64", "riscv64",
 ];
 
 pub(super) static MUSLLINUX_ARCHES: &[&str] = &[
--- a/src/auditwheel/musllinux-policy.json.orig
+++ b/src/auditwheel/musllinux-policy.json
@@ -20,6 +20,8 @@
       },
       "s390x": {
       },
+      "armv6l": {
+      },
       "armv7l": {
       },
       "riscv64": {
