--- a/Makefile	2020-12-08 17:33:42.000000000 +0000
+++ b/Makefile	2021-04-23 18:17:00.499072686 +0000
@@ -51,6 +51,6 @@
 	@rm $(OUTPUT)
 
 install: $(OUTPUT)
-	install -Dm755 "cpufetch"   "/usr/bin/cpufetch"
-	install -Dm644 "LICENSE"    "/usr/share/licenses/cpufetch-git/LICENSE"
-	install -Dm644 "cpufetch.8" "/usr/share/man/man8/cpufetch.8.gz"
+	install -Dm755 "cpufetch"   "$(DESTDIR)/usr/bin/cpufetch"
+	install -Dm644 "LICENSE"    "$(DESTDIR)/usr/share/licenses/cpufetch-git/LICENSE"
+	install -Dm644 "cpufetch.8" "$(DESTDIR)/usr/share/man/man8/cpufetch.8.gz"
