--- a/Makefile
+++ b/Makefile
@@ -64,7 +64,7 @@
 .PHONY: tkey-ssh-agent
 tkey-ssh-agent: apps
 	cp -af apps/signer/app.bin cmd/tkey-ssh-agent/app.bin
-	CGO_ENABLED=0 go build -ldflags "-X main.version=$(TKEY_SSH_AGENT_VERSION) -X main.signerAppNoTouch=$(TKEY_SIGNER_APP_NO_TOUCH)" -trimpath ./cmd/tkey-ssh-agent
+	go build -ldflags "-X main.version=$(TKEY_SSH_AGENT_VERSION) -X main.signerAppNoTouch=$(TKEY_SIGNER_APP_NO_TOUCH)" -trimpath ./cmd/tkey-ssh-agent
 
 .PHONY: clean
 clean:
