--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -920,7 +920,7 @@
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/crypto-config.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/crypto/cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/crypto"
     COMPONENT Development)
 
 if (BUILD_SHARED_LIBS)
@@ -930,6 +930,6 @@
 endif()
 
 install(EXPORT crypto-targets
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/crypto/cmake/${TARGET_DIR}"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/crypto/${TARGET_DIR}"
     NAMESPACE AWS::
     COMPONENT Development)
--- a/ssl/CMakeLists.txt
+++ b/ssl/CMakeLists.txt
@@ -111,7 +111,7 @@
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/ssl-config.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/ssl/cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/ssl"
     COMPONENT Development)
 
 if (BUILD_SHARED_LIBS)
@@ -121,6 +121,6 @@
 endif()
 
 install(EXPORT ssl-targets
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/ssl/cmake/${TARGET_DIR}"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/ssl/${TARGET_DIR}"
     NAMESPACE AWS::
     COMPONENT Development)
