--- a/crypto/CMakeLists.txt
+++ b/crypto/CMakeLists.txt
@@ -933,7 +933,7 @@ configure_file("cmake/crypto-config.cmake"
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/crypto-config.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/crypto/cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/crypto"
     COMPONENT Development)
 
 if (BUILD_SHARED_LIBS)
@@ -943,7 +943,7 @@ else()
 endif()
 
 install(EXPORT crypto-targets
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/crypto/cmake/${TARGET_DIR}"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/crypto/${TARGET_DIR}"
     NAMESPACE AWS::
     COMPONENT Development)
 
--- a/ssl/CMakeLists.txt
+++ b/ssl/CMakeLists.txt
@@ -129,7 +129,7 @@ configure_file("cmake/ssl-config.cmake"
     @ONLY)
 
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/ssl-config.cmake"
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/ssl/cmake"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/ssl"
     COMPONENT Development)
 
 if (BUILD_SHARED_LIBS)
@@ -139,7 +139,7 @@ else()
 endif()
 
 install(EXPORT ssl-targets
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/ssl/cmake/${TARGET_DIR}"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/aws-lc/ssl/${TARGET_DIR}"
     NAMESPACE AWS::
     COMPONENT Development)
 
