diff --git a/setup.py b/setup.py
index bd0cd83..090122d 100755
--- a/setup.py
+++ b/setup.py
@@ -94,6 +94,6 @@ setup(
        'cython>=0.28',
        'setuptools_scm>=1.11.1',
     ] + pytest_runner,
-    tests_require=['unittest2', 'pytest'],
+    tests_require=['pytest'],
     cmdclass=cmdclass,
 )
diff --git a/tests/test_pyclipper.py b/tests/test_pyclipper.py
index 15f4f79..f0e8cc7 100644
--- a/tests/test_pyclipper.py
+++ b/tests/test_pyclipper.py
@@ -4,7 +4,7 @@ Tests for Pyclipper wrapper library.
 """
 
 from __future__ import print_function
-from unittest2 import TestCase, main
+from unittest import TestCase, main
 import sys
 
 if sys.version_info < (3,):
