diff -urN kubesplit-0.3.1.orig/setup.py kubesplit-0.3.1/setup.py
--- kubesplit-0.3.1.orig/setup.py	2021-07-26 23:14:23.440970320 -0600
+++ kubesplit-0.3.1/setup.py	2021-07-26 23:15:01.301173153 -0600
@@ -13,9 +13,9 @@
 
 requirements = ["ruamel.yaml>=0.16.10", "yamkix>=0.9.0"]
 
-setup_requirements = ["pytest-runner"]
+setup_requirements = []
 
-test_requirements = ["pytest"]
+test_requirements = []
 
 setup(
     author="Christophe Furmaniak",
