--- a/collects/openssl/libcrypto.rkt
+++ b/collects/openssl/libcrypto.rkt
@@ -37,7 +37,12 @@
     ;; Other specific known versions
     "1.0.1k" "1.0.1j" "1.0.1g" "1.0.1e"
     "1.0" "1.0.0" "1.0.0e" "1.0.0d" "1.0.0c" "1.0.0b" "1.0.0a"
-    "0.9.8e" "0.9.8b" "0.9.8" "0.9.7"))
+    "0.9.8e" "0.9.8b" "0.9.8" "0.9.7"
+
+    ;; LibreSSL
+    "42"  ;; 2.6.3+
+    "41"  ;; 2.5.5+
+))
 
 (define libcrypto-load-fail-reason #f)
 
--- a/collects/openssl/libssl.rkt
+++ b/collects/openssl/libssl.rkt
@@ -9,6 +9,24 @@
 (provide libssl
          libssl-load-fail-reason)
 
+(define libssl-versions
+  '(;; Versionless (eg from devel pkg)
+    ""
+
+    ;; Compatibility-based version / SONAME
+    "10"     ;; Fedora
+    "1.0.0"  ;; Debian, Ubuntu
+
+    ;; Other specific known versions
+    "1.0.1k" "1.0.1j" "1.0.1g" "1.0.1e"
+    "1.0" "1.0.0" "1.0.0e" "1.0.0d" "1.0.0c" "1.0.0b" "1.0.0a"
+    "0.9.8e" "0.9.8b" "0.9.8" "0.9.7"
+
+    ;; LibreSSL
+    "44"  ;; 2.6.3+
+    "43"  ;; 2.5.3+
+))
+
 (define libssl-load-fail-reason #f)
 
 ;; We need to declare because they might be distributed with PLT Scheme
@@ -28,4 +46,4 @@
                         (lambda (x)
                           (set! libssl-load-fail-reason (exn-message x))
                           #f)])
-         (ffi-lib libssl-so openssl-lib-versions))))
+         (ffi-lib libssl-so libssl-versions))))
