Patch-Source: https://github.com/php/php-src/pull/14801

--- a/scripts/gdb/debug_gdb_scripts_gen.php	2024-07-03 23:44:39.673411357 +0200
+++ b/scripts/gdb/debug_gdb_scripts_gen.php	2024-07-03 23:45:02.123796983 +0200
@@ -28,7 +28,7 @@
      * See https://sourceware.org/gdb/current/onlinedocs/gdb.html/dotdebug_005fgdb_005fscripts-section.html#dotdebug_005fgdb_005fscripts-section
      */
     asm(
-        ".pushsection \".debug_gdb_scripts\", \"MS\",@progbits,1\n"
+        ".pushsection \".debug_gdb_scripts\", \"MS\",%%progbits,1\n"
         ".byte 4 /* Python Text */\n"
         ".ascii \"gdb.inlined-script\\n\"\n"
         %s
--- a/main/debug_gdb_scripts.c
+++ b/main/debug_gdb_scripts.c
@@ -7,7 +7,7 @@
  * See https://sourceware.org/gdb/current/onlinedocs/gdb.html/dotdebug_005fgdb_005fscripts-section.html#dotdebug_005fgdb_005fscripts-section
  */
 asm(
-    ".pushsection \".debug_gdb_scripts\", \"MS\",@progbits,1\n"
+    ".pushsection \".debug_gdb_scripts\", \"MS\",%progbits,1\n"
     ".byte 4 /* Python Text */\n"
     ".ascii \"gdb.inlined-script\\n\"\n"
     ".ascii \"gdb.execute('''\\n\"\n"
