diff --git a/src/xo-shapes.c b/src/xo-shapes.c
index 33840fd..4ed17da 100644
--- a/src/xo-shapes.c
+++ b/src/xo-shapes.c
@@ -72,42 +72,42 @@ void calc_inertia(double *pt, int start, int end, struct Inertia *s)
 
 /* compute normalized quantities */
 
-inline double center_x(struct Inertia s) 
-{ 
+static inline double center_x(struct Inertia s)
+{
   return s.sx/s.mass;
 }
 
-inline double center_y(struct Inertia s) 
-{ 
+static inline double center_y(struct Inertia s)
+{
   return s.sy/s.mass;
 }
 
-inline double I_xx(struct Inertia s)
+static inline double I_xx(struct Inertia s)
 {
   if (s.mass <= 0.) return 0.;
   return (s.sxx - s.sx*s.sx/s.mass)/s.mass;
 }
 
-inline double I_xy(struct Inertia s)
+static inline double I_xy(struct Inertia s)
 {
   if (s.mass <= 0.) return 0.;
   return (s.sxy - s.sx*s.sy/s.mass)/s.mass;
 }
 
-inline double I_yy(struct Inertia s)
+static inline double I_yy(struct Inertia s)
 {
   if (s.mass <= 0.) return 0.;
   return (s.syy - s.sy*s.sy/s.mass)/s.mass;
 }
 
-inline double I_rad(struct Inertia s)
+static inline double I_rad(struct Inertia s)
 {
   double ixx = I_xx(s), iyy = I_yy(s);
   if (ixx+iyy <= 0.) return 0.;
   return sqrt(ixx+iyy);
 }
 
-inline double I_det(struct Inertia s)
+static inline double I_det(struct Inertia s)
 {
   double ixx = I_xx(s), iyy = I_yy(s), ixy = I_xy(s);
   if (s.mass <= 0.) return 0.;
