--- a/src/modules/hime-setup-chewing.c
+++ b/src/modules/hime-setup-chewing.c
@@ -160,17 +160,11 @@
     else
         gtk_box_pack_start (GTK_BOX (hbox_cancel_ok), button_cancel, TRUE, TRUE, 0);
     button_ok = gtk_button_new_from_stock (GTK_STOCK_OK);
-#if !GTK_CHECK_VERSION(2, 91, 2)
+
     if (button_order)
         gtk_box_pack_end (GTK_BOX (hbox_cancel_ok), button_ok, TRUE, TRUE, 5);
     else
         gtk_box_pack_start (GTK_BOX (hbox_cancel_ok), button_ok, TRUE, TRUE, 5);
-#else
-    if (button_order)
-        gtk_grid_attach_next_to (GTK_BOX (hbox_cancel_ok), button_ok, button_cancel, GTK_POS_LEFT, 1, 1);
-    else
-        gtk_grid_attach_next_to (GTK_BOX (hbox_cancel_ok), button_ok, button_cancel, GTK_POS_RIGHT, 1, 1);
-#endif
 
     g_signal_connect (G_OBJECT (button_cancel), "clicked",
                       G_CALLBACK (cb_close_window),
--- a/src/tray.c
+++ b/src/tray.c
@@ -89,7 +89,7 @@
 
     cairo_surface_t *cst = cairo_image_surface_create (CAIRO_FORMAT_ARGB32, iw, ih);
     cr = cairo_create (cst);
-    gdk_cairo_set_source_color (cr, &red_color_fg);
+    gdk_cairo_set_source_rgba (cr, &red_color_fg);
 
     if (pix) {
         gdk_cairo_set_source_pixbuf (cr, pix, 0, 0);
@@ -101,14 +101,14 @@
     }
 
     if (current_CS) {
-        gdk_cairo_set_source_color (cr, &red_color_fg);
+        gdk_cairo_set_source_rgba (cr, &red_color_fg);
         if (current_fullwidth_mode ()) {
             get_text_w_h (full, &w, &h);
             cairo_move_to (cr, iw - w, ih - h);
             pango_cairo_show_layout (cr, pango);
         }
         if (current_CS->b_im_enabled && !chinese_mode ()) {
-            gdk_cairo_set_source_color (cr, &blue_color_fg);
+            gdk_cairo_set_source_rgba (cr, &blue_color_fg);
             get_text_w_h (engst, &w, &h);
             cairo_move_to (cr, 0, 0);
             pango_cairo_show_layout (cr, pango);
@@ -116,7 +116,7 @@
     }
 
     if (gb_output) {
-        gdk_cairo_set_source_color (cr, &red_color_fg);
+        gdk_cairo_set_source_rgba (cr, &red_color_fg);
         get_text_w_h (sim, &w, &h);
         cairo_move_to (cr, 0, ih - h);
         pango_cairo_show_layout (cr, pango);
