diff --git a/compiler/errors/exception.hh b/compiler/errors/exception.hh
index 5aa80ac..c4a5bc3 100644
--- a/compiler/errors/exception.hh
+++ b/compiler/errors/exception.hh
@@ -29,7 +29,7 @@ Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 //#include <io.h>
 #endif
 
-#if !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
+#if 0 && !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
 #include <execinfo.h>
 #endif
 
@@ -54,7 +54,7 @@ class faustexception : public std::runtime_error {
 
 inline void stacktrace(std::stringstream& str, int val)
 {
-#if !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
+#if 0 && !defined(EMCC) && !defined(WIN32) && !defined(ANDROID) && !defined(ALPINE)
     void*  callstack[val];
     int    frames = backtrace(callstack, val);
     char** strs   = backtrace_symbols(callstack, frames);
