diff -ruN a/setup.py b/setup.py
--- a/setup.py	2017-08-29 21:04:01.704564192 +0200
+++ b/setup.py	2017-08-29 21:04:30.369416183 +0200
@@ -99,7 +99,9 @@
 if '--minimal' in sys.argv:
     minimal_build = True
     sys.argv.remove('--minimal') 
-    libraries = []
+    macros.append(('USE_PORTAUDIO', None))
+    ad_files.append("ad_portaudio.c")
+    libraries = ["portaudio"]
 else:
     minimal_build = False
     # portaudio
