diff --git a/cron/cron_test.go b/cron/cron_test.go
index 9256514..25f17dd 100644
--- a/cron/cron_test.go
+++ b/cron/cron_test.go
@@ -132,7 +132,7 @@ var runJobTestCases = []struct {
 		},
 	},
 	{
-		fmt.Sprintf("python -c 'print(\"a\" * %d * 3)'", READ_BUFFER_SIZE), true, &basicContext,
+		fmt.Sprintf("python3 -c 'print(\"a\" * %d * 3)'", READ_BUFFER_SIZE), true, &basicContext,
 		[]*logrus.Entry{
 			{Message: "starting", Level: logrus.InfoLevel, Data: noData},
 			{Message: strings.Repeat("a", READ_BUFFER_SIZE), Level: logrus.InfoLevel, Data: stdoutData},
