--- a/test/helper.rb
+++ b/test/helper.rb
@@ -11,19 +11,12 @@
 #   - "stress" - run tests with GC.stress set to true
 # - NOKOGIRI_GC: read more in test/test_memory_leak.rb
 #
-require 'simplecov'
-SimpleCov.start do
-  add_filter "/test/"
-end
 
 $VERBOSE = true
 
 require 'minitest/autorun'
-require 'minitest/reporters'
 NOKOGIRI_MINITEST_REPORTERS_OPTIONS = { color: true, slow_count: 10, detailed_skip: false }
 NOKOGIRI_MINITEST_REPORTERS_OPTIONS[:fast_fail] = true if ENV["NOKOGIRI_TEST_FAIL_FAST"]
-puts "Minitest::Reporters options: #{NOKOGIRI_MINITEST_REPORTERS_OPTIONS}"
-Minitest::Reporters.use!(Minitest::Reporters::DefaultReporter.new(NOKOGIRI_MINITEST_REPORTERS_OPTIONS))
 
 require 'fileutils'
 require 'tempfile'
