From 396eb59465c783b65baff0f1d8e7d4153236a8db Mon Sep 17 00:00:00 2001
From: Sympatron GmbH <35803463+Sympatron@users.noreply.github.com>
Date: Fri, 16 May 2025 14:18:45 +0200
Subject: [PATCH] Update traitobject to fix #154

---
 vendor/hyper-0.10.16/Cargo.toml | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/vendor/hyper-0.10.16/Cargo.toml b/vendor/hyper-0.10.16/Cargo.toml
index 01c2583..f5433ca 100644
--- a/vendor/hyper-0.10.16/Cargo.toml
+++ b/vendor/hyper-0.10.16/Cargo.toml
@@ -39,8 +39,7 @@ version = "1.0"
 version = "0.1"
 
 [dependencies.traitobject]
-git = "https://github.com/reem/rust-traitobject"
-rev = "b3471a15917b2caf5a8b27debb0b4b390fc6634f"
+version = "0.1.1"
 
 [dependencies.typeable]
 version = "0.1"
