--- a/t/SATest.pm
+++ b/t/SATest.pm
@@ -58,7 +58,7 @@
   };
 
   # Clean PATH so taint doesn't complain
-  $ENV{'PATH'} = '/bin:/usr/bin:/usr/local/bin';
+  $ENV{'PATH'} = '/bin:/usr/bin';
   # Remove tainted envs, at least ENV used in FreeBSD
   delete @ENV{qw(IFS CDPATH ENV BASH_ENV)};
 }
--- a/t/config_tree_recurse.t
+++ b/t/config_tree_recurse.t
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -T
+#!/usr/bin/perl
 #
 # Test that config_tree_recurse works ok in taint mode; bug 6019
 
--- a/t/spamd_utf8.t
+++ b/t/spamd_utf8.t
@@ -1,4 +1,4 @@
-#!/usr/bin/perl -T
+#!/usr/bin/perl
 
 use lib '.'; use lib 't';
 use SATest; sa_t_init("spamd_utf8");
