Point /etc/pam.d/polkit-1 to the right PAM files.

diff --git configure.ac configure.ac
index 36df239..53b5573 100644
--- a/configure.ac
+++ b/configure.ac
@@ -470,6 +470,11 @@ elif test x$with_os_type = xfreebsd -o x$with_os_type = xnetbsd; then
     PAM_FILE_INCLUDE_ACCOUNT=system
     PAM_FILE_INCLUDE_PASSWORD=system
     PAM_FILE_INCLUDE_SESSION=system
+elif test x$with_os_type = xalpine; then
+   PAM_FILE_INCLUDE_AUTH=base-auth
+   PAM_FILE_INCLUDE_ACCOUNT=base-account
+   PAM_FILE_INCLUDE_PASSWORD=base-password
+   PAM_FILE_INCLUDE_SESSION=base-session
 else
    PAM_FILE_INCLUDE_AUTH=system-auth
    PAM_FILE_INCLUDE_ACCOUNT=system-auth
