diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index 13b50e8..3fc1a33 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -991,7 +991,7 @@ void AudacityApp::OnTimer(wxTimerEvent& WXUNUSED(event))
 #define WL(lang,sublang)
 #endif
 
-#if wxCHECK_VERSION(3, 0, 1)
+#if 0
 wxLanguageInfo userLangs[] =
 {
    // Bosnian is defined in wxWidgets already
@@ -1317,7 +1317,7 @@ bool AudacityApp::OnInit()
    //
    // TODO:  The whole Language initialization really need to be reworked.
    //        It's all over the place.
-#if wxCHECK_VERSION(3, 0, 1)
+#if 0
    for (size_t i = 0, cnt = WXSIZEOF(userLangs); i < cnt; i++)
    {
       wxLocale::AddLanguage(userLangs[i]);
diff --git a/src/MixerBoard.cpp b/src/MixerBoard.cpp
index 0f3c67b..b5b9e6b 100644
--- a/src/MixerBoard.cpp
+++ b/src/MixerBoard.cpp
@@ -1314,7 +1314,7 @@ void MixerBoard::LoadMusicalInstruments()
    wxMemoryDC dc;
 
    for (const auto &data : table) {
-      auto bmp = std::make_unique<wxBitmap>(data.bitmap,24);
+      auto bmp = std::make_unique<wxBitmap>(data.bitmap);
       dc.SelectObject(*bmp);
       AColor::Bevel(dc, false, bev);
       mMusicalInstruments.push_back(std::make_unique<MusicalInstrument>(
