diff -urN ircservices-5.1.24.orig/modules/httpd/dbaccess.c ircservices-5.1.24/modules/httpd/dbaccess.c
--- ircservices-5.1.24.orig/modules/httpd/dbaccess.c	2020-11-17 04:41:44.669809306 -0700
+++ ircservices-5.1.24/modules/httpd/dbaccess.c	2020-11-17 04:43:54.833589199 -0700
@@ -59,10 +59,10 @@
 #define next_maskdata (*p_next_maskdata)
 
 /* Imported from NickServ: */
-typeof(get_nickinfo) *p_get_nickinfo;
-typeof(put_nickinfo) *p_put_nickinfo;
-typeof(first_nickinfo) *p_first_nickinfo;
-typeof(next_nickinfo) *p_next_nickinfo;
+static typeof(get_nickinfo) *p_get_nickinfo;
+static typeof(put_nickinfo) *p_put_nickinfo;
+static typeof(first_nickinfo) *p_first_nickinfo;
+static typeof(next_nickinfo) *p_next_nickinfo;
 static typeof(_get_ngi) *p__get_ngi;
 static typeof(_get_ngi_id) *p__get_ngi_id;
 typeof(put_nickgroupinfo) *p_put_nickgroupinfo;
@@ -76,10 +76,10 @@
 
 /* Imported from ChanServ: */
 static typeof(CSMaxReg) *p_CSMaxReg;
-typeof(get_channelinfo) *p_get_channelinfo;
-typeof(put_channelinfo) *p_put_channelinfo;
-typeof(first_channelinfo) *p_first_channelinfo;
-typeof(next_channelinfo) *p_next_channelinfo;
+static typeof(get_channelinfo) *p_get_channelinfo;
+static typeof(put_channelinfo) *p_put_channelinfo;
+static typeof(first_channelinfo) *p_first_channelinfo;
+static typeof(next_channelinfo) *p_next_channelinfo;
 #define CSMaxReg (*p_CSMaxReg)
 #define get_channelinfo (*p_get_channelinfo)
 #define put_channelinfo (*p_put_channelinfo)
@@ -87,10 +87,10 @@
 #define next_channelinfo (*p_next_channelinfo)
 
 /* Imported from StatServ: */
-typeof(get_serverstats) *p_get_serverstats;
-typeof(put_serverstats) *p_put_serverstats;
-typeof(first_serverstats) *p_first_serverstats;
-typeof(next_serverstats) *p_next_serverstats;
+static typeof(get_serverstats) *p_get_serverstats;
+static typeof(put_serverstats) *p_put_serverstats;
+static typeof(first_serverstats) *p_first_serverstats;
+static typeof(next_serverstats) *p_next_serverstats;
 #define get_serverstats (*p_get_serverstats)
 #define put_serverstats (*p_put_serverstats)
 #define first_serverstats (*p_first_serverstats)
diff -urN ircservices-5.1.24.orig/modules/httpd/redirect.c ircservices-5.1.24/modules/httpd/redirect.c
--- ircservices-5.1.24.orig/modules/httpd/redirect.c	2020-11-17 04:41:44.669809306 -0700
+++ ircservices-5.1.24/modules/httpd/redirect.c	2020-11-17 04:42:55.550052479 -0700
@@ -26,18 +26,18 @@
 
 
 /* Imported from NickServ: */
-typeof(get_nickinfo) *p_get_nickinfo;
-typeof(put_nickinfo) *p_put_nickinfo;
-typeof(_get_ngi) *p__get_ngi;
-typeof(put_nickgroupinfo) *p_put_nickgroupinfo;
+static typeof(get_nickinfo) *p_get_nickinfo;
+static typeof(put_nickinfo) *p_put_nickinfo;
+static typeof(_get_ngi) *p__get_ngi;
+static typeof(put_nickgroupinfo) *p_put_nickgroupinfo;
 #define get_nickinfo (*p_get_nickinfo)
 #define put_nickinfo (*p_put_nickinfo)
 #define _get_ngi (*p__get_ngi)
 #define put_nickgroupinfo (*p_put_nickgroupinfo)
 
 /* Imported from ChanServ: */
-typeof(get_channelinfo) *p_get_channelinfo;
-typeof(put_channelinfo) *p_put_channelinfo;
+static typeof(get_channelinfo) *p_get_channelinfo;
+static typeof(put_channelinfo) *p_put_channelinfo;
 #define get_channelinfo (*p_get_channelinfo)
 #define put_channelinfo (*p_put_channelinfo)
 
diff -urN ircservices-5.1.24.orig/modules/protocol/sjoin.c ircservices-5.1.24/modules/protocol/sjoin.c
--- ircservices-5.1.24.orig/modules/protocol/sjoin.c	2020-11-17 04:45:58.764017222 -0700
+++ ircservices-5.1.24/modules/protocol/sjoin.c	2020-11-17 04:47:24.534313498 -0700
@@ -23,7 +23,7 @@
 
 static Module *sjoin_module_chanserv;
 
-int CSSetChannelTime;
+static int CSSetChannelTime;
 
 /*************************************************************************/
 
