diff --git a/src/result_output.c b/src/result_output.c
index 48eeeca..9293e50 100644
--- a/src/result_output.c
+++ b/src/result_output.c
@@ -752,7 +752,7 @@ static void printPropertyResultControl(FILE * f, char *propname, char *subfmt, R
             s = (char *) emalloc(MAXWORDLEN + 1);
             n = strftime(s, (size_t) MAXWORDLEN, fmt, localtime(&(pv->value.v_date)));
             if (n && f)
-                fprintf(f, s);
+                fprintf(f, "%s", s);
             efree(s);
         }
         break;
