diff --git a/lib/getopt.h b/lib/getopt.h
index c4adc30..adcb131 100644
--- a/lib/getopt.h
+++ b/lib/getopt.h
@@ -105,7 +105,7 @@ struct option
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int argc, char *const *argv, const char *shortopts);
 #else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int argc, char *const *argv, const char *shortopts);
 #endif /* __GNU_LIBRARY__ */
 extern int getopt_long (int argc, char *const *argv, const char *shortopts,
 		        const struct option *longopts, int *longind);
diff --git a/lib/glob/glob.c b/lib/glob/glob.c
index f4f1bf0..a47f663 100644
--- a/lib/glob/glob.c
+++ b/lib/glob/glob.c
@@ -165,7 +165,7 @@ extern void abort (), exit ();
 
 /* joe: added this... __P was blowing up everywhere */
 #ifndef __P
-#define __P(a) ()
+#define __P(a) a
 #endif
 
 /* joe: added this */
