diff --git a/CMakeLists.txt b/CMakeLists.txt
index 33a55c7..1739565 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -67,7 +67,7 @@ set_package_properties(LibGpgError PROPERTIES TYPE REQUIRED)
 find_package(Gpgme ${GPGME_REQUIRED_VERSION})
 set_package_properties(Gpgme PROPERTIES TYPE REQUIRED)
 
-g10_get_full_version()
+option(G10_FULL_VERSION "Patched version to not have to use git" "")
 
 # create the VERSION file
 find_package(Git QUIET)
@@ -80,7 +80,7 @@ if (GIT_FOUND AND EXISTS "${CMAKE_SOURCE_DIR}/.git")
     string(STRIP "${commit_id}" commit_id) # strip trailing whitespace
     file(WRITE "${CMAKE_BINARY_DIR}/VERSION" "${G10_FULL_VERSION}\n${commit_id}\n")
 elseif(EXISTS "${CMAKE_SOURCE_DIR}/VERSION")
-    configure_file("${CMAKE_SOURCE_DIR}/VERSION" "VERSION" COPYONLY)
+    configure_file("${CMAKE_SOURCE_DIR}/VERSION" "G10_FULL_VERSION" COPYONLY)
 else()
     file(WRITE "${CMAKE_BINARY_DIR}/VERSION" "${G10_FULL_VERSION}\nunknown\n")
 endif()
