--- a/galerautils/src/gu_alloc.cpp
+++ b/galerautils/src/gu_alloc.cpp
@@ -27,10 +27,10 @@
     if (gu_likely(size <= left_))
     {
         /* to avoid too frequent allocation, make it (at least) 64K */
-        static page_size_type const PAGE_SIZE(gu_page_size_multiple(1 << 16));
+        static page_size_type const MEM_PAGE_SIZE(gu_page_size_multiple(1 << 16));

         page_size_type const page_size
-            (std::min(std::max(size, PAGE_SIZE), left_));
+            (std::min(std::max(size, MEM_PAGE_SIZE), left_));

         Page* ret = new HeapPage (page_size);
