diff --git a/hack/generate-bindata.sh b/hack/generate-bindata.sh
index 995f1d17..32d04188 100755
--- a/hack/generate-bindata.sh
+++ b/hack/generate-bindata.sh
@@ -30,7 +30,6 @@ fi
 
 # kube::golang::build_kube_toolchain installs the vendored go-bindata in
 # $GOPATH/bin, so make sure that's explicitly part of our $PATH.
-export PATH="${KUBE_OUTPUT_BINPATH}:${PATH}"
 
 if ! which go-bindata &>/dev/null ; then
        echo "Cannot find go-bindata."
diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
index f20a55b3..ca6ae34d 100755
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -778,11 +778,11 @@ kube::golang::build_binaries() {
     host_platform=$(kube::golang::host_platform)
 
     local goflags goldflags goasmflags gogcflags
-    # If GOLDFLAGS is unset, then set it to the a default of "-s -w".
+    # If GOLDFLAGS is unset, then set it to the a default of "-w".
     # Disable SC2153 for this, as it will throw a warning that the local
     # variable goldflags will exist, and it suggest changing it to this.
     # shellcheck disable=SC2153
-    goldflags="${GOLDFLAGS=-s -w} $(kube::version::ldflags)"
+    goldflags="${GOLDFLAGS=-w} $(kube::version::ldflags)"
     goasmflags="-trimpath=${KUBE_ROOT}"
     gogcflags="${GOGCFLAGS:-} -trimpath=${KUBE_ROOT}"

