diff --git a/build/root/Makefile b/build/root/Makefile
index ea5eec8..5a88385 100644
--- a/build/root/Makefile
+++ b/build/root/Makefile
@@ -39,7 +39,7 @@ MAKEFLAGS += --no-builtin-rules
 # Constants used throughout.
 .EXPORT_ALL_VARIABLES:
 OUT_DIR ?= _output
-BIN_DIR := $(OUT_DIR)/bin
+BIN_DIR := $(OUT_DIR)/local/go/bin
 PRJ_SRC_PATH := k8s.io/kubernetes
 GENERATED_FILE_PREFIX := zz_generated.
 
diff --git a/hack/lib/util.sh b/hack/lib/util.sh
index 50f6b55..c8a8b5e 100755
--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -168,7 +168,7 @@ kube::util::find-binary-for-platform() {
   local -r lookfor="$1"
   local -r platform="$2"
   local locations=(
-    "${KUBE_ROOT}/_output/bin/${lookfor}"
+    "${KUBE_ROOT}/_output/local/go/bin/${lookfor}"
     "${KUBE_ROOT}/_output/dockerized/bin/${platform}/${lookfor}"
     "${KUBE_ROOT}/_output/local/bin/${platform}/${lookfor}"
     "${KUBE_ROOT}/platforms/${platform}/${lookfor}"
