diff --git a/hack/lib/test.sh b/hack/lib/test.sh
index 575664a..7318cf9 100644
--- a/hack/lib/test.sh
+++ b/hack/lib/test.sh
@@ -73,10 +73,10 @@ kube::test::object_assert() {
   local object=$2
   local request=$3
   local expected=$4
-  local args=${5:-}
+  local get_args=${5:-}
 
   for j in $(seq 1 ${tries}); do
-    res=$(eval kubectl get "${kube_flags[@]}" ${args} $object -o go-template=\"$request\")
+	res=$(eval kubectl ${args} get "${kube_flags[@]}" ${get_args} $object -o go-template=\"$request\")
     if [[ "$res" =~ ^$expected$ ]]; then
         echo -n ${green}
         echo "$(kube::test::get_caller 3): Successful get $object $request: $res"
diff --git a/hack/make-rules/test-e2e-node.sh b/hack/make-rules/test-e2e-node.sh
index 2a68768..54a8233 100755
--- a/hack/make-rules/test-e2e-node.sh
+++ b/hack/make-rules/test-e2e-node.sh
@@ -168,6 +168,6 @@ else
     --container-runtime-endpoint=${container_runtime_endpoint} \
     --image-service-endpoint=${image_service_endpoint} \
     --alsologtostderr --v 4 --report-dir=${artifacts} --node-name $(hostname) \
-    $test_args --k8s-bin-dir /usr/bin" --build-dependencies=true 2>&1 | tee -i "${artifacts}/build-log.txt"
+    $test_args" --build-dependencies=true 2>&1 | tee -i "${artifacts}/build-log.txt"
   exit $?
 fi
