--- kubernetes-1.8.4/build/root/Makefile.generated_files.orig
+++ kubernetes-1.8.4/build/root/Makefile.generated_files
@@ -155,7 +155,7 @@
 # is what the .stamp file represents.
 $(foreach dir, $(ALL_GO_DIRS),  \
     $(META_DIR)/$(dir)/$(GOFILES_META)):
-	FILES=$$(ls $</*.go | grep --color=never -v $(GENERATED_FILE_PREFIX));  \
+	FILES=$$(ls $</*.go | grep -v $(GENERATED_FILE_PREFIX));  \
 	mkdir -p $(@D);                                           \
 	echo "gofiles__$< := $$(echo $${FILES})" >$@.tmp;         \
 	if ! cmp -s $@.tmp $@; then                               \
@@ -187,7 +187,7 @@
 endif
 ALL_K8S_TAG_FILES := $(shell                             \
     find $(ALL_GO_DIRS) -maxdepth 1 -type f -name \*.go  \
-        | xargs grep --color=never -l '^// *+k8s:'       \
+        | xargs grep -l '^// *+k8s:'       \
 )
 
 #
@@ -214,7 +214,7 @@
     $(warning ***** finding all +k8s:deepcopy-gen tags)
 endif
 DEEPCOPY_DIRS := $(shell                                             \
-    grep --color=never -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
+    grep -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
         | xargs -n1 dirname                                          \
         | LC_ALL=C sort -u                                           \
 )
@@ -289,7 +289,7 @@
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./vendor/k8s.io/code-generator/cmd/deepcopy-gen                  \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -352,7 +352,7 @@
     $(warning ***** finding all +k8s:defaulter-gen tags)
 endif
 DEFAULTER_DIRS := $(shell                                            \
-    grep --color=never -l '+k8s:defaulter-gen=' $(ALL_K8S_TAG_FILES) \
+    grep -l '+k8s:defaulter-gen=' $(ALL_K8S_TAG_FILES) \
         | xargs -n1 dirname                                          \
         | LC_ALL=C sort -u                                           \
 )
@@ -453,7 +453,7 @@
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./vendor/k8s.io/code-generator/cmd/defaulter-gen                 \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -506,7 +506,7 @@
     $(warning ***** finding all +k8s:openapi-gen tags)
 endif
 OPENAPI_DIRS := $(shell                                             \
-    grep --color=never -l '+k8s:openapi-gen=' $(ALL_K8S_TAG_FILES)  \
+    grep -l '+k8s:openapi-gen=' $(ALL_K8S_TAG_FILES)  \
         | xargs -n1 dirname                                         \
         | LC_ALL=C sort -u                                          \
 )
@@ -557,7 +557,7 @@
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./vendor/k8s.io/code-generator/cmd/openapi-gen                   \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -621,7 +621,7 @@
     $(warning ***** finding all +k8s:conversion-gen tags)
 endif
 CONVERSION_DIRS := $(shell                                              \
-    grep --color=never '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
+    grep '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
         | cut -f1 -d:                                                   \
         | xargs -n1 dirname                                             \
         | LC_ALL=C sort -u                                              \
@@ -683,7 +683,7 @@
 # is what the .stamp file represents.
 $(foreach dir, $(CONVERSION_DIRS),  \
     $(META_DIR)/$(dir)/$(CONVERSIONS_META)):
-	TAGS=$$(grep --color=never -h '^// *+k8s:conversion-gen=' $</*.go   \
+	TAGS=$$(grep -h '^// *+k8s:conversion-gen=' $</*.go   \
 	    | cut -f2- -d=                                                  \
 	    | sed 's|$(PRJ_SRC_PATH)/||'                                    \
 	    | sed 's|^k8s.io/|vendor/k8s.io/|');                            \
@@ -771,7 +771,7 @@
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./vendor/k8s.io/code-generator/cmd/conversion-gen                \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
