diff --git a/build/root/Makefile.generated_files b/build/root/Makefile.generated_files
index 45287dc..8617f49 100644
--- a/build/root/Makefile.generated_files
+++ b/build/root/Makefile.generated_files
@@ -155,7 +155,7 @@ $(foreach dir, $(ALL_GO_DIRS), $(eval           \
 # is what the .stamp file represents.
 $(foreach dir, $(ALL_GO_DIRS),  \
     $(META_DIR)/$(dir)/$(GOFILES_META)):
-	FILES=$$(ls $</*.go | grep --color=never -v $(GENERATED_FILE_PREFIX));  \
+	FILES=$$(ls $</*.go | grep -v $(GENERATED_FILE_PREFIX));  \
 	mkdir -p $(@D);                                           \
 	echo "gofiles__$< := $$(echo $${FILES})" >$@.tmp;         \
 	cmp -s $@.tmp $@ || touch $@.stamp;                       \
@@ -181,7 +181,7 @@ ifeq ($(DBG_MAKEFILE),1)
 endif
 ALL_K8S_TAG_FILES := $(shell                             \
     find $(ALL_GO_DIRS) -maxdepth 1 -type f -name \*.go  \
-        | xargs grep --color=never -l '^// *+k8s:'       \
+        | xargs grep -l '^// *+k8s:'       \
 )
 
 #
@@ -208,7 +208,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:deepcopy-gen tags)
 endif
 DEEPCOPY_DIRS := $(shell                                             \
-    grep --color=never -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
+    grep -l '+k8s:deepcopy-gen=' $(ALL_K8S_TAG_FILES)  \
         | xargs -n1 dirname                                          \
         | LC_ALL=C sort -u                                           \
 )
@@ -275,7 +275,7 @@ $(META_DIR)/$(DEEPCOPY_GEN).mk:
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./cmd/libs/go2idl/deepcopy-gen                                   \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -332,7 +332,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:defaulter-gen tags)
 endif
 DEFAULTER_DIRS := $(shell                                            \
-    grep --color=never -l '+k8s:defaulter-gen=' $(ALL_K8S_TAG_FILES) \
+    grep -l '+k8s:defaulter-gen=' $(ALL_K8S_TAG_FILES) \
         | xargs -n1 dirname                                          \
         | LC_ALL=C sort -u                                           \
 )
@@ -425,7 +425,7 @@ $(META_DIR)/$(DEFAULTER_GEN).mk:
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./cmd/libs/go2idl/defaulter-gen                                  \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -472,7 +472,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:openapi-gen tags)
 endif
 OPENAPI_DIRS := $(shell                                             \
-    grep --color=never -l '+k8s:openapi-gen=' $(ALL_K8S_TAG_FILES)  \
+    grep -l '+k8s:openapi-gen=' $(ALL_K8S_TAG_FILES)  \
         | xargs -n1 dirname                                         \
         | LC_ALL=C sort -u                                          \
 )
@@ -523,7 +523,7 @@ $(META_DIR)/$(OPENAPI_GEN).mk:
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./cmd/libs/go2idl/openapi-gen                                    \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
@@ -581,7 +581,7 @@ ifeq ($(DBG_MAKEFILE),1)
     $(warning ***** finding all +k8s:conversion-gen tags)
 endif
 CONVERSION_DIRS := $(shell                                              \
-    grep --color=never '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
+    grep '^// *+k8s:conversion-gen=' $(ALL_K8S_TAG_FILES) \
         | cut -f1 -d:                                                   \
         | xargs -n1 dirname                                             \
         | LC_ALL=C sort -u                                              \
@@ -639,7 +639,7 @@ $(foreach dir, $(CONVERSION_DIRS), $(eval           \
 # is what the .stamp file represents.
 $(foreach dir, $(CONVERSION_DIRS),  \
     $(META_DIR)/$(dir)/$(CONVERSIONS_META)):
-	TAGS=$$(grep --color=never -h '^// *+k8s:conversion-gen=' $</*.go   \
+	TAGS=$$(grep -h '^// *+k8s:conversion-gen=' $</*.go   \
 	    | cut -f2- -d=                                                  \
 	    | sed 's|$(PRJ_SRC_PATH)/||');                                  \
 	mkdir -p $(@D);                                                     \
@@ -717,7 +717,7 @@ $(META_DIR)/$(CONVERSION_GEN).mk:
 	 ./hack/run-in-gopath.sh go list                                      \
 	     -f '{{.ImportPath}}{{"\n"}}{{range .Deps}}{{.}}{{"\n"}}{{end}}'  \
 	     ./cmd/libs/go2idl/conversion-gen                                 \
-	     | grep --color=never "^$(PRJ_SRC_PATH)/"                         \
+	     | grep "^$(PRJ_SRC_PATH)/"                         \
 	     | xargs ./hack/run-in-gopath.sh go list                          \
 	         -f '{{$$d := .Dir}}{{$$d}}{{"\n"}}{{range .GoFiles}}{{$$d}}/{{.}}{{"\n"}}{{end}}'  \
 	     | paste -sd' ' -                                                 \
