diff --git a/cmd/hyperkube/kube-controller-manager.go b/cmd/hyperkube/kube-controller-manager.go
index 35061e9..482958d 100644
--- a/cmd/hyperkube/kube-controller-manager.go
+++ b/cmd/hyperkube/kube-controller-manager.go
@@ -29,7 +29,7 @@ func NewKubeControllerManager() *Server {
 	hks := Server{
 		name:            "controller-manager",
 		AlternativeName: "kube-controller-manager",
-		SimpleUsage:     "controller-manager",
+		SimpleUsage:     "kube-controller-manager",
 		Long:            "A server that runs a set of active components. This includes replication controllers, service endpoints and nodes.",
 		Run: func(_ *Server, args []string) error {
 			return app.Run(s)
diff --git a/cmd/hyperkube/kube-proxy.go b/cmd/hyperkube/kube-proxy.go
index 76bc2dc..1eba8a7 100644
--- a/cmd/hyperkube/kube-proxy.go
+++ b/cmd/hyperkube/kube-proxy.go
@@ -33,7 +33,7 @@ func NewKubeProxy() *Server {
 	hks := Server{
 		name:            "proxy",
 		AlternativeName: "kube-proxy",
-		SimpleUsage:     "proxy",
+		SimpleUsage:     "kube-proxy",
 		Long:            command.Long,
 	}
 
diff --git a/cmd/hyperkube/kube-scheduler.go b/cmd/hyperkube/kube-scheduler.go
index fd00103..587a9a1 100644
--- a/cmd/hyperkube/kube-scheduler.go
+++ b/cmd/hyperkube/kube-scheduler.go
@@ -29,7 +29,7 @@ func NewScheduler() *Server {
 	hks := Server{
 		name:            "scheduler",
 		AlternativeName: "kube-scheduler",
-		SimpleUsage:     "scheduler",
+		SimpleUsage:     "kube-scheduler",
 		Long:            "Implements a Kubernetes scheduler.  This will assign pods to kubelets based on capacity and constraints.",
 		Run: func(_ *Server, _ []string) error {
 			return app.Run(s)
diff --git a/cmd/hyperkube/main.go b/cmd/hyperkube/main.go
index e1e07b7..10ed8b1 100644
--- a/cmd/hyperkube/main.go
+++ b/cmd/hyperkube/main.go
@@ -33,7 +33,6 @@ func main() {
 	}
 
 	hk.AddServer(NewKubectlServer())
-	hk.AddServer(NewKubeAPIServer())
 	hk.AddServer(NewKubeControllerManager())
 	hk.AddServer(NewScheduler())
 	hk.AddServer(NewKubelet())
