diff --git a/utils.py b/utils.py
index 22430ce..2ec4845 100644
--- a/utils.py
+++ b/utils.py
@@ -265,9 +265,11 @@ def is_libvirtd_up():
     """
     Checks if libvirtd.service is up.
     """
-    cmd = ['systemctl', 'is-active', 'libvirtd.service']
+    cmd = ['rc-service', 'is-active', 'libvirtd.service']
+    cmd = ['rc-service', 'libvirtd', 'status']
     output, error, rc = run_command(cmd, silent=True)
-    return True if output == 'active\n' else False
+    return True if "started" in output else False
+
 
 
 def is_s390x():
