link to system md4c
--
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1f82fe7..77fa2a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ include_directories(libraries/sonnet/src/core)
 #some hunspell settings for windows
 include(libraries/sonnet/src/plugins/hunspell/hunspell/CMakeLists.txt)
 
-add_subdirectory(libraries/md4c)
+find_package(md4c)
 
 add_executable(QOwnNotes "")
 
