diff --git a/make.conf b/make.conf
index a29d8ac..8f4706b 100644
--- a/make.conf
+++ b/make.conf
@@ -31,7 +31,7 @@ EFI_CRT_OBJS = ${EFILIB}/crt0-efi-${ARCH}.o
 EFI_LDS = ${EFILIB}/elf_${ARCH}_efi.lds
 
 CFLAGS = \
-        -Wall -Werror \
+        -Wall \
         ${EFIINCS} -fno-stack-protector -fpic -fshort-wchar \
         -DGIT_VERSION=\"$(shell git rev-parse HEAD | cut -c1-12)\"
 ifeq (${ARCH},x86_64)
