See https://gitlab.gnome.org/GNOME/jsonrpc-glib/issues/2
diff --git a/tests/meson.build b/tests/meson.build
index 40b410b..3318c79 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -38,13 +38,6 @@ test_server = executable('test-server', 'test-server.c',
 )
 test('test-server', test_server, env: test_env)
 
-test_stress = executable('test-stress', 'test-stress.c',
-        c_args: test_cflags,
-     link_args: test_link_args,
-  dependencies: test_deps,
-)
-test('test-stress', test_stress, env: test_env)
-
 test_gauntlet = executable('test-gauntlet', 'test-gauntlet.c',
         c_args: test_cflags,
      link_args: test_link_args,
