From 2f9fda23cdb610fddd406790922c41e7ebfa4dfb Mon Sep 17 00:00:00 2001
From: Alan Messias Cordeiro <alancordeiro@gmail.com>
Date: Wed, 11 Dec 2013 15:29:53 +0000
Subject: [PATCH] dbd enabled by default

---
 servers/slapd/slapd.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/servers/slapd/slapd.conf b/servers/slapd/slapd.conf
index 4938b85..a2ccbd3 100644
--- a/servers/slapd/slapd.conf
+++ b/servers/slapd/slapd.conf
@@ -15,7 +15,7 @@ argsfile	%LOCALSTATEDIR%/run/slapd.args
 
 # Load dynamic backend modules:
 # modulepath	%MODULEDIR%
-# moduleload	back_bdb.la
+moduleload	back_bdb.la
 # moduleload	back_hdb.la
 # moduleload	back_ldap.la
 
-- 
1.8.4.3

