diff -upr openldap-2.4.44.orig/servers/slapd/slapd.ldif openldap-2.4.44/servers/slapd/slapd.ldif
--- openldap-2.4.44.orig/servers/slapd/slapd.ldif	2016-03-20 16:31:37.592683978 +0100
+++ openldap-2.4.44/servers/slapd/slapd.ldif	2016-03-20 16:33:13.022396171 +0100
@@ -30,7 +30,7 @@ olcPidFile: %LOCALSTATEDIR%/run/slapd.pi
 #objectClass: olcModuleList
 #cn: module
 #olcModulepath:	%MODULEDIR%
-#olcModuleload:	back_bdb.la
+olcModuleload:	back_bdb.la
 #olcModuleload:	back_hdb.la
 #olcModuleload:	back_ldap.la
 #olcModuleload:	back_passwd.la
