From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 31 Oct 2021 01:20:07 +0200
Subject: [PATCH] Fix lloadd tests - wrong ldapcompare usage

--- a/tests/scripts/lloadd/test004-monitor
+++ b/tests/scripts/lloadd/test004-monitor
@@ -218,7 +218,7 @@
 # wait for it to settle
 echo "Waiting until connections are established..."
 for i in 0 1 2 3 4 5; do
-    $LDAPCOMPARE "cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+    $LDAPCOMPARE -H $URI6 "cn=Load Balancer,cn=Backends,cn=monitor" \
         'olmOutgoingConnections:4' > /dev/null 2>&1
     RC=$?
     if test $RC = 6 ; then
@@ -269,7 +269,7 @@
 # wait for it to settle
 echo "Waiting until connections are established..."
 for i in 0 1 2 3 4 5; do
-    $LDAPCOMPARE "cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+    $LDAPCOMPARE -H $URI6 "cn=Load Balancer,cn=Backends,cn=monitor" \
         'olmOutgoingConnections:13' > /dev/null 2>&1
     RC=$?
     if test $RC = 6 ; then
@@ -306,7 +306,7 @@
 # wait for it to settle
 echo "Waiting until global counters are updated..."
 for i in 0 1 2 3 4 5; do
-    $LDAPCOMPARE "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+    $LDAPCOMPARE -H $URI6 "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" \
         'olmCompletedOps:2' > /dev/null 2>&1
     RC=$?
     if test $RC = 6 ; then
--- a/tests/scripts/lloadd/test007-coherence
+++ b/tests/scripts/lloadd/test007-coherence
@@ -239,7 +239,7 @@
 # wait for it to settle
 echo "Waiting until connections are established..."
 for i in 0 1 2 3 4 5; do
-	$LDAPCOMPARE "cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+	$LDAPCOMPARE -H $URI6 "cn=Load Balancer,cn=Backends,cn=monitor" \
 		'olmOutgoingConnections:4' > /dev/null 2>&1
 	RC=$?
 	if test $RC = 6 ; then
@@ -291,7 +291,7 @@
 # wait for it to settle
 echo "Waiting until connections are established..."
 for i in 0 1 2 3 4 5; do
-	$LDAPCOMPARE "cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+	$LDAPCOMPARE -H $URI6 "cn=Load Balancer,cn=Backends,cn=monitor" \
 		'olmOutgoingConnections:13' > /dev/null 2>&1
 	RC=$?
 	if test $RC = 6 ; then
@@ -338,7 +338,7 @@
 # wait for it to settle
 echo "Waiting until global counters are updated..."
 for i in 0 1 2 3 4 5; do
-	$LDAPCOMPARE "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+	$LDAPCOMPARE -H $URI6 "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" \
 		'olmCompletedOps:20' > /dev/null 2>&1
 	RC=$?
 	if test $RC = 6 ; then
@@ -443,7 +443,7 @@
 # wait for it to settle
 echo "Waiting until global counters are updated..."
 for i in 0 1 2 3 4 5; do
-	$LDAPCOMPARE "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" -H $URI6 \
+	$LDAPCOMPARE -H $URI6 "cn=Other,cn=Operations,cn=Load Balancer,cn=Backends,cn=monitor" \
 		'olmCompletedOps:28' > /dev/null 2>&1
 	RC=$?
 	if test $RC = 6 ; then
