diff --git a/Makefile.in b/Makefile.in
index 9a99f98..0e54305 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -43,9 +43,9 @@ host_os = @host_os@
 build = @build@
 # Whether to link curl-impersonate with libcurl-impersonate statically.
 static_build = @static_build@
-# Use the locally built zlib/zstd unless overridden.
-with_zlib ?= $(zlib_install_dir)
-with_zstd ?= $(zstd_install_dir)
+# Use system libraries for zlib/zstd
+with_zlib ?= /usr
+with_zstd ?= /usr
 # Path to be passed to curl's --with-ca-bundle configure option.
 with_ca_bundle = @with_ca_bundle@
 # Path to be passed to curl's --with-ca-path configure option.
@@ -83,10 +83,7 @@ libidn2_install_dir := $(abspath libidn2-$(LIBIDN2_VERSION)/installed)
 libidn2_static_libs := $(libidn2_install_dir)/lib/libidn2.a
 
 # Dependencies needed to compile the Chrome version
-chrome_libs := $(zlib_static_libs) $(zstd_static_libs) $(brotli_static_libs) $(boringssl_static_libs) $(nghttp2_static_libs) $(ngtcp2_static_libs) $(nghttp3_static_libs)
-ifneq ($(filter linux%,$(host_os)),)
-chrome_libs += $(libidn2_static_libs)
-endif
+chrome_libs := $(boringssl_static_libs) $(ngtcp2_static_libs) $(nghttp3_static_libs)
 
 # Auto-generate Makefile help.
 # Borrowed from https://marmelab.com/blog/2016/02/29/auto-documented-makefile.html
@@ -245,8 +242,6 @@ $(brotli_static_libs): brotli-$(BROTLI_VERSION).tar.gz brotli-$(BROTLI_VERSION)/
 
 
 boringssl-$(BORING_SSL_COMMIT).zip:
-	curl -L https://github.com/google/boringssl/archive/$(BORING_SSL_COMMIT).zip \
-		-o boringssl-$(BORING_SSL_COMMIT).zip
 
 # Patch boringssl and use a dummy '.patched' file to mark it patched
 boringssl-$(BORING_SSL_COMMIT)/.patched: $(srcdir)/patches/boringssl.patch
@@ -338,7 +333,6 @@ $(nghttp2_static_libs): nghttp2-$(NGHTTP2_VERSION).tar.bz2
 	$(MAKE) install MAKEFLAGS=
 
 ngtcp2-$(NGTCP2_VERSION).tar.bz2:
-	curl -L $(NGTCP2_URL) -o ngtcp2-$(NGTCP2_VERSION).tar.bz2
 
 # Patch ngtcp2 and use a dummy '.patched' file to mark it patched
 ngtcp2-$(NGTCP2_VERSION)/.patched: $(srcdir)/patches/ngtcp2.patch
@@ -373,7 +367,6 @@ $(ngtcp2_static_libs): ngtcp2-$(NGTCP2_VERSION).tar.bz2 ngtcp2-$(NGTCP2_VERSION)
 	$(MAKE) install MAKEFLAGS=
 
 nghttp3-$(NGHTTP3_VERSION).tar.bz2:
-	curl -L $(NGHTTP3_URL) -o nghttp3-$(NGHTTP3_VERSION).tar.bz2
 
 # Patch nghttp3 and use a dummy '.patched' file to mark it patched
 nghttp3-$(NGHTTP3_VERSION)/.patched: $(srcdir)/patches/nghttp3.patch
@@ -437,8 +430,6 @@ $(libidn2_static_libs): libidn2-$(LIBIDN2_VERSION).tar.gz $(libunistring_static_
 	$(MAKE) install MAKEFLAGS=
 
 $(CURL_VERSION).tar.gz:
-	curl -L "https://github.com/curl/curl/archive/$(CURL_VERSION).tar.gz" \
-		-o "$(CURL_VERSION).tar.gz"
 
 # Apply the "Chorme version" patches and mark using a dummy file
 $(CURL_VERSION)/.patched: $(srcdir)/patches/curl.patch
@@ -463,8 +454,8 @@ $(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.
 	# https://daniel.haxx.se/blog/2024/01/10/psl-in-curl/
 	{ \
 	  config_flags="--prefix=@prefix@"; \
-	  config_flags="$$config_flags --with-brotli=$(brotli_install_dir)"; \
-	  config_flags="$$config_flags --with-nghttp2=$(nghttp2_install_dir)"; \
+	  config_flags="$$config_flags --with-brotli=/usr"; \
+	  config_flags="$$config_flags --with-nghttp2=/usr"; \
 	  config_flags="$$config_flags --with-ngtcp2=$(ngtcp2_install_dir)"; \
 	  config_flags="$$config_flags --with-nghttp3=$(nghttp3_install_dir)"; \
 	  case "$(host_os)" in \
@@ -475,12 +466,12 @@ $(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.
 	      config_flags="$$config_flags --without-libidn2"; \
 	      ;; \
 	    *) \
-	      config_flags="$$config_flags --with-libidn2=$(libidn2_install_dir)"; \
+	      config_flags="$$config_flags --with-libidn2=/usr"; \
 	      ;; \
 	  esac; \
 	  config_flags="$$config_flags --with-openssl=$(boringssl_dir)"; \
-	  config_flags="$$config_flags --with-zlib=$(with_zlib)"; \
-	  config_flags="$$config_flags --with-zstd=$(with_zstd)"; \
+	  config_flags="$$config_flags --with-zlib=/usr"; \
+	  config_flags="$$config_flags --with-zstd=/usr"; \
 	  config_flags="$$config_flags --without-librtmp"; \
 	  config_flags="$$config_flags --disable-ldap --disable-ldaps"; \
 	  config_flags="$$config_flags --without-libpsl"; \
@@ -504,28 +495,25 @@ $(CURL_VERSION)/.chrome: $(chrome_libs)	$(CURL_VERSION).tar.gz $(CURL_VERSION)/.
 	  case "$(host_os)" in \
 	    darwin*|ios*|tvos*|watchos*) add_libs="-pthread -lc++" ;; \
 	    *android*) add_libs="-pthread -lc++" ;; \
-	    *) add_libs="-pthread -lstdc++ -L$(libunistring_install_dir)/lib -lunistring" ;; \
+	    *) add_libs="-pthread -lstdc++ -lunistring" ;; \
 	  esac; \
 	}
 
 	echo PKG_CONFIG_PATH="$(ngtcp2_install_dir)/lib/pkgconfig"
 	echo "Configuring curl with: $$config_flags"
 
-	{ \
-		lib_path="$(nghttp2_install_dir)/lib:$(ngtcp2_install_dir)/lib:$(nghttp3_install_dir)/lib:$(zstd_install_dir)/lib:$(zlib_install_dir)/lib:$(brotli_install_dir)/lib"; \
-		case "$(host_os)" in \
-		  darwin*|ios*|tvos*|watchos*) ;; \
-		  *android*) ;; \
-		  *) lib_path="$$lib_path:$(libidn2_install_dir)/lib:$(libunistring_install_dir)/lib" ;; \
-		esac; \
-	}
+	lib_path="$(ngtcp2_install_dir)/lib:$(nghttp3_install_dir)/lib"
 
 	DYLD_LIBRARY_PATH="$$lib_path" \
 	LD_LIBRARY_PATH="$$lib_path" \
 	PKG_CONFIG_LIBDIR= \
 	PKG_CONFIG_PATH="$(ngtcp2_install_dir)/lib/pkgconfig" \
+	CPPFLAGS="-isystem $(boringssl_dir)/include" \
+	LDFLAGS="-L$(boringssl_dir)/lib" \
 		./configure \
 			 $$config_flags \
+			 ac_cv_func_SSL_set1_ech_config_list=yes \
+			 ac_cv_lib_ngtcp2_crypto_boringssl_ngtcp2_crypto_recv_client_initial_cb=yes \
 			 LIBS="$$add_libs"
 
 	# Remove possible leftovers from a previous compilation
