--- ./src/makefile.unix.old	2013-12-14 13:14:50.445065372 +0000
+++ ./src/makefile.unix	2013-12-14 13:13:35.834253950 +0000
@@ -37,7 +37,7 @@
    -l boost_system$(BOOST_LIB_SUFFIX) \
    -l boost_filesystem$(BOOST_LIB_SUFFIX) \
    -l boost_program_options$(BOOST_LIB_SUFFIX) \
-   -l boost_thread$(BOOST_LIB_SUFFIX) \
+   -l boost_thread-mt$(BOOST_LIB_SUFFIX) \
    -l db_cxx$(BDB_LIB_SUFFIX) \
    -l ssl \
    -l crypto
