--- vlc-2.1.2.orig/modules/codec/omxil/omxil_core.c
+++ vlc-2.1.2/modules/codec/omxil/omxil_core.c
@@ -202,6 +202,13 @@
 {
 #ifdef RPI_OMX
     { "video_decoder.avc", "OMX.broadcom.video_decode" },
+    { "video_decoder.mpeg4", "OMX.broadcom.video_decode" },
+    { "video_decoder.mpeg2", "OMX.broadcom.video_decode" },
+    { "video_decoder.vp6", "OMX.broadcom.video_decode" },
+    { "video_decoder.vp8", "OMX.broadcom.video_decode" },
+    { "video_decoder.theora", "OMX.broadcom.video_decode" },
+    { "video_decoder.mjpg", "OMX.broadcom.video_decode" },
+    { "video_decoder.vc1", "OMX.broadcom.video_decode" },
     { "iv_renderer", "OMX.broadcom.video_render" },
 #endif
     { 0, 0 }
