uclibc don't have fmaxf()

--- ./modules/control/dbus/dbus_player.c.orig	2013-12-11 15:00:16.338111047 +0000
+++ ./modules/control/dbus/dbus_player.c	2013-12-11 15:01:48.649150243 +0000
@@ -179,7 +179,7 @@
     if( VLC_SUCCESS != DemarshalSetPropertyValue( p_from, &d_dbus_vol ) )
         return DBUS_HANDLER_RESULT_NOT_YET_HANDLED;
 
-    playlist_VolumeSet( PL, fmaxf( d_dbus_vol, 0.f ) );
+    playlist_VolumeSet( PL, d_dbus_vol > 0.f ? d_dbus_vol : 0.f );
 
     REPLY_SEND;
 }

