--- vlc-2.0.0-rc1.orig/modules/misc/Modules.am
+++ vlc-2.0.0-rc1/modules/misc/Modules.am
@@ -26,14 +26,9 @@
 libxdg_screensaver_plugin_la_CFLAGS = $(AM_CFLAGS)
 libxdg_screensaver_plugin_la_LIBADD = $(AM_LIBADD)
 libxdg_screensaver_plugin_la_DEPENDENCIES =
-libxscreensaver_plugin_la_SOURCES = inhibit/xscreensaver.c
-libxscreensaver_plugin_la_CFLAGS = $(AM_CFLAGS)
-libxscreensaver_plugin_la_LIBADD = $(AM_LIBADD)
-libxscreensaver_plugin_la_DEPENDENCIES =
 if HAVE_XCB
 libvlc_LTLIBRARIES += \
-	libxdg_screensaver_plugin.la \
-	libxscreensaver_plugin.la
+	libxdg_screensaver_plugin.la
 endif
 
 libmce_plugin_la_SOURCES = inhibit/mce.c
