diff -ru vlc-2.1.0.orig/include/vlc_network.h vlc-2.1.0/include/vlc_network.h
--- vlc-2.1.0.orig/include/vlc_network.h	2013-06-24 21:00:38.000000000 +0300
+++ vlc-2.1.0/include/vlc_network.h	2013-10-06 21:59:53.275530885 +0300
@@ -222,6 +222,9 @@
 #ifndef AI_NUMERICSERV
 # define AI_NUMERICSERV 0
 #endif
+#if defined(__UCLIBC__)
+#undef AI_IDN
+#endif
 #ifndef AI_IDN
 # define AI_IDN 0 /* GNU/libc extension */
 #endif
