--- a/modules/misc/inhibit/xdg.c.orig
+++ b/modules/misc/inhibit/xdg.c
@@ -26,7 +26,12 @@
 #include <vlc_plugin.h>
 #include <vlc_inhibit.h>
 #include <assert.h>
-#include <spawn.h>
+#if !defined(_POSIX_SPAWN)
+# define _POSIX_SPAWN -1
+#else
+# include <spawn.h>
+#endif
+
 #include <sys/wait.h>
 
 static int Open (vlc_object_t *);
@@ -137,7 +142,15 @@
         pid_t pid;
 
         vlc_mutex_unlock (&p_sys->lock);
+#if (_POSIX_SPAWN >= 0)
         if (!posix_spawnp (&pid, "xdg-screensaver", NULL, NULL, argv, environ))
+#else
+        pid = fork();
+        if (pid == 0) {
+            execvp("xdg-screensaver", argv);
+            exit(1);
+        } else if (pid > 0)
+#endif
         {
             int status;
 
