diff --git a/src/control/vlm.c b/src/control/vlm.c
index 05417e5..46fa957 100644
--- a/src/control/vlm.c
+++ b/src/control/vlm.c
@@ -25,6 +25,10 @@
 # include <config.h>
 #endif
 
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
 #include <vlc/libvlc.h>
 #include <vlc/libvlc_vlm.h>
 #include <vlc_es.h>
diff --git a/src/misc/linux_specific.c b/src/misc/linux_specific.c
index 1644ce6..1440e1c 100644
--- a/src/misc/linux_specific.c
+++ b/src/misc/linux_specific.c
@@ -83,14 +83,14 @@ static struct
     unsigned refs;
 } once = { VLC_STATIC_MUTEX, 0 };
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 # include <gnu/libc-version.h>
 # include <stdlib.h>
 #endif
 
 void system_Init (libvlc_int_t *libvlc, int *argc, const char *argv[])
 {
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
     const char *glcv = gnu_get_libc_version ();
 
     /* gettext in glibc 2.5-2.7 is not thread-safe. LibVLC keeps crashing,
