diff --git a/rngd_rtlsdr.c b/rngd_rtlsdr.c
index eca0ada..fed1f88 100644
--- a/rngd_rtlsdr.c
+++ b/rngd_rtlsdr.c
@@ -180,7 +180,7 @@ static size_t condition_buffer(unsigned char *in, unsigned char *out, size_t ins
 	* Finalise the encryption. Further ciphertext bytes may be written at
 	* this stage.
 	*/
-	if(1 != EVP_EncryptFinal_ex(ctx, out + len, &len))
+	if(1 != EVP_EncryptFinal_ex(ctx, out, &len))
 		return 0;
 	ciphertext_len += len;
 
