From 3bf88a0f1d34ee84e7c0c54034a4e2c2ccfc9bd8 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 23 Aug 2013 13:19:43 +0200
Subject: [PATCH] pingu_conf: fix segfault when declaring iface twice

---
 src/pingu_conf.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/pingu_conf.c b/src/pingu_conf.c
index 263a1f4..aa8395a 100644
--- a/pingu_conf.c
+++ b/pingu_conf.c
@@ -125,7 +125,8 @@ static int pingu_conf_read_iface(struct pingu_conf *conf, char *ifname)
 
 	iface = pingu_iface_get_by_name(ifname);
 	if (iface != NULL) {
-		log_error("Interface %s already declared (line %i)", conf->lineno);
+		log_error("Interface %s already declared (line %i)", ifname,
+			  conf->lineno);
 		return -1;
 	}
 
-- 
1.8.5.1

