--- a/config.m4
+++ b/config.m4
@@ -19,7 +19,7 @@
   if test "$PHP_XDEBUG_FOUND_VERNUM" -lt "70200"; then
     AC_MSG_ERROR([not supported. Need a PHP version >= 7.2.0 and < 8.2.0 (found $PHP_XDEBUG_FOUND_VERSION)])
   else
-    if test "$PHP_XDEBUG_FOUND_VERNUM" -ge "80200"; then
+    if test "$PHP_XDEBUG_FOUND_VERNUM" -ge "80201"; then
       AC_MSG_ERROR([not supported. Need a PHP version >= 7.2.0 and < 8.2.0 (found $PHP_XDEBUG_FOUND_VERSION)])
     else
       AC_MSG_RESULT([supported ($PHP_XDEBUG_FOUND_VERSION)])
