diff --git a/setup.cfg b/setup.cfg
index aef20c6..61d3ef4 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -69,10 +69,6 @@ addopts = -rxXs
   --strict-config
   --strict-markers
 xfail_strict=True
-filterwarnings=
-    # Turn warnings that aren't filtered into exceptions
-    error
-    ignore: \"watchgod\" is depreciated\, you should switch to watchfiles \(`pip install watchfiles`\)\.:DeprecationWarning
 
 [coverage:run]
 omit = venv/*
