diff -up soundtouch/include/STTypes.h~ soundtouch/include/STTypes.h
--- soundtouch/include/STTypes.h~	2009-01-25 17:32:53.000000000 +0100
+++ soundtouch/include/STTypes.h	2009-02-14 22:22:37.000000000 +0100
@@ -87,7 +87,7 @@ namespace soundtouch
  
  #endif
 
-    #if (WIN32 || __i386__ || __x86_64__)
+    #if (WIN32 || __i386__)
         /// Define this to allow X86-specific assembler/intrinsic optimizations. 
         /// Notice that library contains also usual C++ versions of each of these
         /// these routines, so if you're having difficulties getting the optimized 
