diff -upr RSound-1.1.orig/src/rsound-common.c RSound-1.1/src/rsound-common.c
--- RSound-1.1.orig/src/rsound-common.c	2015-11-22 03:25:55.595163404 +0100
+++ RSound-1.1/src/rsound-common.c	2015-11-22 03:26:56.161613187 +0100
@@ -49,7 +49,7 @@
 
 
 /* Not really portable, need to find something better */
-#define PIDFILE "/tmp/.rsound.pid"
+#define PIDFILE "/var/run/rsound.pid"
 
 /* Pulls in callback structs depending on compilation options. */
 
