From 2ab92ce00f420d98e10f2b04eee229ffcf540015 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 12 Feb 2014 14:00:50 +0000
Subject: [PATCH 5/7] exportfs: define _GNU_SOURCE for stat64

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 utils/exportfs/exportfs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index 8c86790..1feadaa 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -12,6 +12,10 @@
 #include <config.h>
 #endif
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/vfs.h>
-- 
1.8.5.3

