--- ./utils/osd_login/Makefile.in.orig
+++ ./utils/osd_login/Makefile.in
@@ -426,7 +426,7 @@
 all-local: $(OSD_LOGIN_FILES)
 
 install-data-hook:
-	$(INSTALL) --mode 755 osd_login $(DESTDIR)/sbin/osd_login
+	$(INSTALL) -D -m 755 osd_login $(DESTDIR)/sbin/osd_login
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
