From daa17a3e9c3e7ac5a97e7c7ab10a382abd7d7562 Mon Sep 17 00:00:00 2001
From: jvoisin <julien.voisin@dustri.org>
Date: Sun, 12 Mar 2023 13:18:39 +0100
Subject: [PATCH] Fix the CI on Archlinux

---
 tests/test_libmat2.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/tests/test_libmat2.py b/tests/test_libmat2.py
index b71f7c2..ee43fba 100644
--- a/tests/test_libmat2.py
+++ b/tests/test_libmat2.py
@@ -480,11 +480,12 @@ class TestCleaning(unittest.TestCase):
             'expected_meta': {
                 'AverageBitrate': 465641,
                 'BufferSize': 0,
-                'CompatibleBrands': ['isom', 'iso2', 'avc1', 'mp41'],
-                'ColorProfiles': 'nclx',
                 'ColorPrimaries': 'BT.709',
+                'ColorProfiles': 'nclx',
                 'ColorRepresentation': 'nclx 1 1 1',
+                'CompatibleBrands': ['isom', 'iso2', 'avc1', 'mp41'],
                 'CompressorID': 'avc1',
+                'CompressorName': 'JVT/AVC Coding',
                 'GraphicsMode': 'srcCopy',
                 'HandlerDescription': 'SoundHandler',
                 'HandlerType': 'Metadata',
@@ -495,6 +496,7 @@ class TestCleaning(unittest.TestCase):
                 'MediaDataOffset': 48,
                 'MediaDataSize': 379872,
                 'MediaHeaderVersion': 0,
+                'MediaLanguageCode': 'eng',
                 'MinorVersion': '0.2.0',
                 'MovieDataOffset': 48,
                 'MovieHeaderVersion': 0,
-- 
GitLab

