diff --git a/dtd.e b/dtd.e
index e075ed9..66f8935 100644
--- a/dtd.e
+++ b/dtd.e
@@ -6,5 +6,4 @@ typedef struct _ElementType {
       mixed, empty, cdata, stag, etag, pre, break_before, break_after;
   string parents[76];
 } ElementType;
-extern const ElementType * lookup_element(
-                                   );
+extern const ElementType * lookup_element(const char *str, size_t len);
diff --git a/dtd.hash b/dtd.hash
index 221d415..5266c0f 100644
--- a/dtd.hash
+++ b/dtd.hash
@@ -43,8 +43,7 @@ EXPORT typedef struct _ElementType {
 } ElementType;
 
 /* lookup_element -- look up the string in the hash table */
-EXPORT const ElementType * lookup_element(/* register const char *str,
-					  register unsigned int len */);
+EXPORT const ElementType * lookup_element(const char *str, size_t len);
 
 /* Different kinds of parent elements: */
 #define PHRASE "abbr", "acronym", "b", "bdi", "bdo", "big", "cite", "code", "dfn", "em", "i", "kbd", "q", "s", "samp", "small", "span", "strong", "sub", "sup", "time", "tt", "u", "var"
