diff -upr pcc-1.1.0.orig/cc/cc/cc.c pcc-1.1.0/cc/cc/cc.c
--- pcc-1.1.0.orig/cc/cc/cc.c	2016-09-16 20:38:14.932292071 +0200
+++ pcc-1.1.0/cc/cc/cc.c	2016-09-16 20:39:17.022094921 +0200
@@ -206,7 +206,7 @@ char *cppmdadd[] = CPPMDADD;
 #define PCCLIBDIR	NULL
 #endif
 #ifndef DEFLIBDIRS	/* default library search paths */
-#ifdef MULTIARCH_PATH
+#ifndef MULTIARCH_PATH
 #define DEFLIBDIRS	{ "/usr/lib/", 0 }
 #else
 #define DEFLIBDIRS	{ "/usr/lib/", "/usr/lib/" MULTIARCH_PATH "/", 0 }
