diff -upr pcc-1.1.0.orig/os/linux/ccconfig.h pcc-1.1.0/os/linux/ccconfig.h
--- pcc-1.1.0.orig/os/linux/ccconfig.h	2016-09-16 20:00:46.350281549 +0200
+++ pcc-1.1.0/os/linux/ccconfig.h	2016-09-16 20:04:28.006146652 +0200
@@ -39,14 +39,14 @@
 
 #if defined(mach_i386)
 #define CPPMDADD	{ "-D__i386__", NULL, }
-#define DYNLINKER	{ "-dynamic-linker", "/lib/ld-linux.so.2", NULL }
+#define DYNLINKER	{ "-dynamic-linker", "/lib/ld-musl-i386.so.1", NULL }
 #elif defined(mach_powerpc)
 #define CPPMDADD	{ "-D__ppc__", NULL, }
 #define DYNLINKER	{ "-dynamic-linker", "/lib/ld-linux.so.2", NULL }
 #elif defined(mach_amd64)
 #define CPPMDADD	{ "-D__x86_64__", "-D__x86_64", "-D__amd64__", \
 	"-D__amd64", "-D__LP64__", "-D_LP64", NULL, }
-#define	DYNLINKER { "-dynamic-linker", "/lib64/ld-linux-x86-64.so.2", NULL }
+#define	DYNLINKER { "-dynamic-linker", "/lib/ld-musl-x86_64.so.1", NULL }
 #ifndef MULTIARCH_PATH
 #define	DEFLIBDIRS	{ "/usr/lib64/", 0 }
 #else
