diff --git a/src/protocol3.c b/src/protocol3.c
index e0f3aa5..cb29ad2 100644
--- a/src/protocol3.c
+++ b/src/protocol3.c
@@ -1812,9 +1812,9 @@ static int get_metadata_chunks(njb,
      njb_t *njb;
      unsigned char *command_block;
      int command_block_size;
-     int (*create_metadata_post) ();
-     int (*add_to_metadata_post) ();
-     int (*terminate_metadata_post) ();
+  int (*create_metadata_post) (u_int32_t postid, unsigned char **target);
+  int (*add_to_metadata_post) (u_int16_t frameid, u_int16_t framelen, unsigned char *data, unsigned char **target);
+  int (*terminate_metadata_post) (njb_t *njb, unsigned char **target);
 {
   __dsub= "get_metadata_chunks";
 
@@ -2036,9 +2036,9 @@ static int parse_metadata_block(njb,
      njb_t *njb;
      unsigned char *data;
      u_int32_t size;
-     int (*create_metadata_post) ();
-     int (*add_to_metadata_post) ();
-     int (*terminate_metadata_post) ();
+  int (*create_metadata_post) (u_int32_t postid, unsigned char **target);
+  int (*add_to_metadata_post) (u_int16_t frameid, u_int16_t framelen, unsigned char *data, unsigned char **target);
+  int (*terminate_metadata_post) (njb_t *njb, unsigned char **target);
 {
   u_int32_t i = 0;
   unsigned char *target;
