diff --git a/envmanager/config.h b/envmanager/config.h
index 273ffc4c7..bc95f8314 100644
--- a/envmanager/config.h
+++ b/envmanager/config.h
@@ -36,7 +36,7 @@ const QMap<QString, QMap<QString, QVariant>> KDEGLOBALS_SETTINGS = {{"KDE", {{"L
 const QMap<QString, QMap<QString, QVariant>> KWINRC_DEFAULT_SETTINGS = {
     {"Wayland",
      {
-         {"InputMethod", "/usr/share/applications/com.github.maliit.keyboard.desktop"} // ensure maliit is our default vkbd
+         {"InputMethod", "/usr/share/applications/org.kde.plasma.keyboard.desktop"} // ensure plasma-keyboard is our default vkbd
      }}};
 
 // .config/plasma-mobile/kwinrc - immutable settings:
