--- a/src/3rdparty/chromium/third_party/libxml/src/encoding.c
+++ b/src/3rdparty/chromium/third_party/libxml/src/encoding.c
@@ -2028,7 +2028,7 @@
 #ifdef LIBXML_ICU_ENABLED
     else if (handler->uconv_out != NULL) {
         ret = xmlUconvWrapper(handler->uconv_out, 0, out, outlen, in, inlen,
-                              TRUE);
+                              true);
     }
 #endif /* LIBXML_ICU_ENABLED */
     else {
