diff --git a/3rdparty/perfparser.cmake b/3rdparty/perfparser.cmake
index fc401a7..9d39219 100644
--- a/3rdparty/perfparser.cmake
+++ b/3rdparty/perfparser.cmake
@@ -44,7 +44,7 @@ if (Zstd_FOUND)
     target_compile_definitions(libhotspot-perfparser PUBLIC HAVE_ZSTD=1)
 endif()
 
-if (HAVE_DWFL_GET_DEBUGINFOD_CLIENT)
+if (false)
     target_link_libraries(libhotspot-perfparser PRIVATE ${LIBDEBUGINFOD_LIBRARIES})
     target_compile_definitions(libhotspot-perfparser PRIVATE HAVE_DWFL_GET_DEBUGINFOD_CLIENT=1)
 endif()
