--- ./rtapd.c.orig
+++ ./rtapd.c
@@ -466,7 +466,7 @@
       funct = *(pblock + 5);
       memcpy(id, pblock + 6, 8);
       id[8] = '\0';
-      pagenum = strtoq(id, NULL, 16);
+      pagenum = strtoll(id, NULL, 16);
       sprintf(id, "!%qX", pagenum);
       pxend = strchr(id, '\0');
       pblock = pblock + 13;
@@ -527,7 +527,7 @@
       }
       memcpy(id, pblock + 7, usedlen);
       id[usedlen] = '\0';
-      pagenum = strtoq(id, NULL, 16);
+      pagenum = strtoll(id, NULL, 16);
       sprintf(id, "!%qX", pagenum);
       pxend = strchr(id, '\0');
       pblock = pblock + 6 + caplen + msnlen;
