upstream: https://github.com/voutilad/virtio_vmmci/pull/16

diff --git a/virtio_vmmci.c b/virtio_vmmci.c
index adfcaf5..2c3cca8 100644
--- a/virtio_vmmci.c
+++ b/virtio_vmmci.c
@@ -89,11 +89,6 @@ static struct ctl_table drift_table[] = {
 	{ },
 };
 
-static struct ctl_table vmmci_table = {
-	.procname	= "vmmci",
-	.child		= drift_table,
-};
-
 /* Define our basic commands and structs for our device including the
  * virtio feature tables.
  */
@@ -282,7 +277,9 @@ static int vmmci_probe(struct virtio_device *vdev)
 
 	INIT_WORK(&vmmci->sync_work, sync_work_func);
 
-	vmmci_table_header = register_sysctl_table(&vmmci_table);
+	vmmci_table_header = register_sysctl("dev/vmmci", drift_table);
+	if (!vmmci_table_header)
+		return -ENOMEM;
 
 	log("started VMM Control Interface driver\n");
 	return 0;
