Patch-Source: https://gitlab.alpinelinux.org/alpine/aports/-/issues/17458

--- pacman-v7.0.0/scripts/libmakepkg/source/git.sh.in	2024-07-14 16:11:18.000000000 +0700
+++ pacman-v7.0.0/scripts/libmakepkg/source/git.sh.in.new	2025-08-27 12:23:23.907096985 +0700
@@ -49,7 +49,7 @@
 
 	if [[ ! -d "$dir" ]] || dir_is_empty "$dir" ; then
 		msg2 "$(gettext "Cloning %s %s repo...")" "${repo}" "git"
-		if ! git clone --origin=origin ---mirror "$url" "$dir"; then
+		if ! git clone --origin=origin --mirror "$url" "$dir"; then
 			error "$(gettext "Failure while downloading %s %s repo")" "${repo}" "git"
 			plainerr "$(gettext "Aborting...")"
 			exit 1
