Force build system to use libintl coming with gettext.

--- a/meson.build
+++ b/meson.build
@@ -77,12 +77,10 @@
 
 libintl = dependency('', required: false)
 if get_option('i18n')
-  if not cc.has_function('ngettext')
     libintl = cc.find_library('intl', required : false, static: get_option('buildstatic'))
     if not libintl.found()
       error('ngettext not found but NLS support requested')
     endif
-  endif
   conf.set('ENABLE_NLS', 1)
 endif
 
@@ -341,7 +339,7 @@
   pacman_sources,
   include_directories : includes,
   link_with : [libalpm, libcommon],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
@@ -350,7 +348,7 @@
   pacman_conf_sources,
   include_directories : includes,
   link_with : [libalpm, libcommon],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
@@ -359,7 +357,7 @@
   testpkg_sources,
   include_directories : includes,
   link_with : [libalpm],
-  dependencies : [libarchive],
+  dependencies : [libarchive, libintl],
   install : true,
 )
 
