diff -ru extundelete-0.2.4.orig/src/insertionops.cc extundelete-0.2.4/src/insertionops.cc
--- extundelete-0.2.4.orig/src/insertionops.cc	2018-06-04 22:19:22.460051515 +0000
+++ extundelete-0.2.4/src/insertionops.cc	2018-06-04 22:19:49.256717673 +0000
@@ -33,7 +33,7 @@
   os << "File flags: " << inode.i_flags << std::endl;
   os << "File version (for NFS): " << inode.i_generation << std::endl;
   os << "File ACL: " << inode.i_file_acl << std::endl;
-  os << "Directory ACL: " << inode.i_dir_acl << std::endl;
+  os << "Directory ACL: " << inode.i_size_high << std::endl;
   os << "Fragment address: " << inode.i_faddr << std::endl;
   os << "Direct blocks: ";
   for (int n = 0; n < EXT2_NDIR_BLOCKS; n++)

