shellcheck is only available for x86_64.

--- a/src/shell.rs
+++ b/src/shell.rs
@@ -65,6 +65,7 @@
     }
 
     #[apply(opts)]
+    #[ignore]
     fn bash_shellcheck(cmd: Option<&str>, hook: InitHook, echo: bool, resolve_symlinks: bool) {
         let opts = Opts { cmd, hook, echo, resolve_symlinks };
         let source = Bash(&opts).render().unwrap();
@@ -186,6 +187,7 @@
     }
 
     #[apply(opts)]
+    #[ignore]
     fn posix_shellcheck_(cmd: Option<&str>, hook: InitHook, echo: bool, resolve_symlinks: bool) {
         let opts = Opts { cmd, hook, echo, resolve_symlinks };
         let source = Posix(&opts).render().unwrap();
@@ -277,6 +279,7 @@
     }
 
     #[apply(opts)]
+    #[ignore]
     fn zsh_shellcheck(cmd: Option<&str>, hook: InitHook, echo: bool, resolve_symlinks: bool) {
         let opts = Opts { cmd, hook, echo, resolve_symlinks };
         let source = Zsh(&opts).render().unwrap();
