shellcheck is only available for x86_64.

--- a/src/shell.rs
+++ b/src/shell.rs
@@ -107,6 +107,7 @@
                 }
 
                 #[test]
+                #[ignore]
                 fn bash_shellcheck_#i() {
                     let opts = dbg!(&opts()[i]);
                     let source = Bash(opts).render().unwrap();
@@ -246,6 +247,7 @@
                 }
 
                 #[test]
+                #[ignore] 
                 fn posix_shellcheck_#i() {
                     let opts = dbg!(&opts()[i]);
                     let source = Posix(opts).render().unwrap();
@@ -338,6 +340,7 @@
                 }
 
                 #[test]
+                #[ignore]
                 fn zsh_shellcheck_#i() {
                     let opts = dbg!(&opts()[i]);
                     let source = Zsh(opts).render().unwrap();
