shellcheck is only available for x86_64.

--- a/src/shell.rs
+++ b/src/shell.rs
@@ -60,6 +60,7 @@
     }
 
     #[rstest]
+    #[ignore]
     fn bash_shellcheck(
         #[values(None, Some("z"))] cmd: Option<&str>,
         #[values(InitHook::None, InitHook::Prompt, InitHook::Pwd)] hook: InitHook,
@@ -234,6 +235,7 @@
     }
 
     #[rstest]
+    #[ignore]
     fn posix_shellcheck_(
         #[values(None, Some("z"))] cmd: Option<&str>,
         #[values(InitHook::None, InitHook::Prompt, InitHook::Pwd)] hook: InitHook,
@@ -367,6 +369,7 @@
     }
 
     #[rstest]
+    #[ignore]
     fn zsh_shellcheck(
         #[values(None, Some("z"))] cmd: Option<&str>,
         #[values(InitHook::None, InitHook::Prompt, InitHook::Pwd)] hook: InitHook,
