--- a/bootstrap.sh
+++ b/bootstrap.sh
@@ -392,18 +392,10 @@
 
   if need_pkg ${PKG} ${VER_MATCH}
   then
-    echo
-    if [ -r "${PKG}-${VER}.tar.gz" ]
-    then
-        echo "Using local tarball for ${PKG}-${VER}."
-    else
-        echo "Downloading ${PKG}-${VER}..."
-        fetch_pkg ${PKG} ${VER}
-    fi
-    unpack_pkg ${PKG} ${VER}
-    cd "${PKG}-${VER}"
+    cd "../${PKG}-${VER}"
+    cp "../${PKG}.cabal" .
     install_pkg ${PKG} ${VER}
-    cd ..
+    cd ../cabal-install-*
   fi
 }
 
