Or it would install into /usr/local/bin

diff --git a/seq/Makefile b/seq/Makefile
index e6db62e..d1a4c1d 100644
--- a/seq/Makefile
+++ b/seq/Makefile
@@ -1,8 +1,6 @@
 SUBDIRS = sbiload
 TOP = ..
 
-all:
-	@for i in $(SUBDIRS); do cd $$i; echo $$i; ./gitcompile $(GITCOMPILE_ARGS) || exit 1; cd ..; $(MAKE) -C $$i; done
 
 configure:
 	@for i in $(SUBDIRS); do cd $$i; echo $$i; ./configure $(CONFIGURE_ARGS) || exit 1; cd ..; $(MAKE) -C $$i; done
