diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6e434bb..6acc4a8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -303,7 +303,7 @@ if (OPENDHT_STATIC)
         target_include_directories(opendht-static SYSTEM PRIVATE ${argon2_INCLUDE_DIRS})
         target_link_libraries(opendht-static
             PRIVATE PkgConfig::argon2
-            PUBLIC ${CMAKE_THREAD_LIBS_INIT} PkgConfig::GnuTLS PkgConfig::Nettle msgpackc-cxx
+            PUBLIC ${CMAKE_THREAD_LIBS_INIT} PkgConfig::GnuTLS PkgConfig::Nettle msgpack-cxx
                 ${FMT_LIBRARY} ${HTTP_PARSER_LIBRARY})
         if (Jsoncpp_FOUND)
             target_link_libraries(opendht-static PUBLIC PkgConfig::Jsoncpp)
@@ -374,7 +374,7 @@ if (OPENDHT_SHARED)
     set_target_properties (opendht PROPERTIES SOVERSION ${opendht_VERSION_MAJOR} VERSION ${opendht_VERSION})
     target_compile_definitions(opendht PRIVATE OPENDHT_BUILD)
     target_link_libraries(opendht
-        PUBLIC ${CMAKE_THREAD_LIBS_INIT} msgpackc-cxx
+        PUBLIC ${CMAKE_THREAD_LIBS_INIT} msgpack-cxx
         PRIVATE PkgConfig::GnuTLS PkgConfig::Nettle
                 ${FMT_LIBRARY} ${HTTP_PARSER_LIBRARY} PkgConfig::argon2)
     if (Jsoncpp_FOUND)
