Upstream: Not applicable
Reason: Use system loky and cloudpickle

diff --git a/joblib/__init__.py b/joblib/__init__.py
index e35eed5..4f48954 100644
--- a/joblib/__init__.py
+++ b/joblib/__init__.py
@@ -124,7 +124,7 @@ from .parallel import register_parallel_backend
 from .parallel import parallel_backend
 from .parallel import effective_n_jobs
 
-from .externals.loky import wrap_non_picklable_objects
+from loky import wrap_non_picklable_objects
 
 
 __all__ = ['Memory', 'MemorizedResult', 'PrintTime', 'Logger', 'hash', 'dump',
diff --git a/joblib/_memmapping_reducer.py b/joblib/_memmapping_reducer.py
index d583822..f109cfb 100644
--- a/joblib/_memmapping_reducer.py
+++ b/joblib/_memmapping_reducer.py
@@ -34,7 +34,7 @@ except ImportError:
 from .numpy_pickle import dump, load, load_temporary_memmap
 from .backports import make_memmap
 from .disk import delete_folder
-from .externals.loky.backend import resource_tracker
+from loky.backend import resource_tracker
 
 # Some system have a ramdisk mounted by default, we can use it instead of /tmp
 # as the default folder to dump big arrays to share with subprocesses.
@@ -58,7 +58,7 @@ JOBLIB_MMAPS = set()
 
 
 def _log_and_unlink(filename):
-    from .externals.loky.backend.resource_tracker import _resource_tracker
+    from loky.backend.resource_tracker import _resource_tracker
     util.debug(
         "[FINALIZER CALL] object mapping to {} about to be deleted,"
         " decrementing the refcount of the file (pid: {})".format(
diff --git a/joblib/_parallel_backends.py b/joblib/_parallel_backends.py
index 4264528..432eda4 100644
--- a/joblib/_parallel_backends.py
+++ b/joblib/_parallel_backends.py
@@ -21,7 +21,7 @@ if mp is not None:
     # Compat between concurrent.futures and multiprocessing TimeoutError
     from multiprocessing import TimeoutError
     from concurrent.futures._base import TimeoutError as CfTimeoutError
-    from .externals.loky import process_executor, cpu_count
+    from loky import process_executor, cpu_count
 
 
 class ParallelBackendBase(metaclass=ABCMeta):
diff --git a/joblib/executor.py b/joblib/executor.py
index 9273fed..962661f 100644
--- a/joblib/executor.py
+++ b/joblib/executor.py
@@ -10,7 +10,7 @@ copy between the parent and child processes.
 
 from ._memmapping_reducer import get_memmapping_reducers
 from ._memmapping_reducer import TemporaryResourcesManager
-from .externals.loky.reusable_executor import _ReusablePoolExecutor
+from loky.reusable_executor import _ReusablePoolExecutor
 
 
 _executor_args = None
diff --git a/joblib/parallel.py b/joblib/parallel.py
index db1b1b9..118151f 100644
--- a/joblib/parallel.py
+++ b/joblib/parallel.py
@@ -26,8 +26,8 @@ from .disk import memstr_to_bytes
 from ._parallel_backends import (FallbackToBackend, MultiprocessingBackend,
                                  ThreadingBackend, SequentialBackend,
                                  LokyBackend)
-from .externals.cloudpickle import dumps, loads
-from .externals import loky
+from cloudpickle import dumps, loads
+import loky
 
 # Make sure that those two classes are part of the public joblib.parallel API
 # so that 3rd party backend implementers can import them from here.
