diff -upr mod_perl-2.0.6.orig/lib/Apache2/Build.pm mod_perl-2.0.6/lib/Apache2/Build.pm
--- mod_perl-2.0.6.orig/lib/Apache2/Build.pm	2012-05-26 22:31:00.000000000 +0300
+++ mod_perl-2.0.6/lib/Apache2/Build.pm	2012-05-26 22:31:32.000000000 +0300
@@ -2207,7 +2207,8 @@ sub has_large_files_conflict {
     # with it is that we didn't have such a case yet, but may need to
     # deal with it later
 
-    return $perl_lfs64 ^ $apr_lfs64;
+    return 0;
+    # $perl_lfs64 ^ $apr_lfs64;
 }
 
 # if perl is built with uselargefiles, but apr not, the build won't
