From abbf307b9720140a3dd71e96f8ea0328e1bd7115 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 30 Jan 2015 09:25:53 +0200
Subject: [PATCH 3/3] Add checkgauge in lpostcmd

---
 examples/lpostcmd.lua | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/examples/lpostcmd.lua b/examples/lpostcmd.lua
index 8ad339e..e413112 100644
--- a/examples/lpostcmd.lua
+++ b/examples/lpostcmd.lua
@@ -11,6 +11,14 @@ local rsyncpostcmd = {
 	-- based on default rsync.
 	default.rsync,
 
+	checkgauge = {
+		default.rsync.checkgauge,
+		host = true,
+		targetdir = true,
+		target = true,
+		postcmd = true,
+	},
+
 	-- for this config it is important to keep maxProcesses at 1, so
 	-- the postcmds will only be spawned after the rsync completed
 	maxProcesses = 1,
-- 
2.2.2

