--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -127,6 +127,6 @@
 target_link_libraries( lsyncd ${LUA_LIBRARIES} )
 
 install( TARGETS lsyncd RUNTIME DESTINATION bin )
-install( FILES docs/manpage/lsyncd.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 COMPONENT man )
+install( FILES docs/manpage/lsyncd.1 DESTINATION share/man/man1 COMPONENT man )
-install( DIRECTORY examples DESTINATION doc )
+install( DIRECTORY examples DESTINATION share/doc )
 install( DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/html DESTINATION doc OPTIONAL)
