diff -ru lsyncd-2.1.5.orig/examples/lpostcmd.lua lsyncd-2.1.5/examples/lpostcmd.lua
--- lsyncd-2.1.5.orig/examples/lpostcmd.lua	2015-01-30 09:28:40.376456301 -0200
+++ lsyncd-2.1.5/examples/lpostcmd.lua	2015-01-30 09:28:12.746457070 -0200
@@ -6,7 +6,7 @@
 -- after every successfullycompleted rsync operation.
 -- for example to restart servlets on the target host or so.
 
+rsyncpostcmd = {
-local rsyncpostcmd = {
 
 	-- based on default rsync.
 	default.rsync,
@@ -83,12 +83,3 @@
 	end
 }
 
-
-sync {
-	rsyncpostcmd,
-	source = "src",
-	host = "beetle",
-	targetdir = "/path/to/trg",
-	postcmd = "/usr/local/bin/restart-servelt.sh",
-}
-
