--- a/examples/lpostcmd.lua
+++ b/examples/lpostcmd.lua
@@ -6,7 +6,7 @@
 -- after every successfullycompleted rsync operation.
 -- for example to restart servlets on the target host or so.
 
-local rsyncpostcmd = {
+rsyncpostcmd = {
 
 	-- based on default rsync.
 	default.rsync,
@@ -25,7 +25,7 @@
 			-- uses rawget to test if "isPostcmd" has been set without
 			-- triggering an error if not.
 			local isPostcmd = rawget(event, "isPostcmd")
-			if event.isPostcmd then
+			if isPostcmd then
 				spawn(event, "/usr/bin/ssh", 
 					config.host, config.postcmd)
         		return
@@ -58,7 +58,7 @@
 			return default.collect(agent,exitcode)
 		end
 		error("this should never be reached")
-	end
+	end,
 
 	-- called before anything else
 	-- builds the target from host and targetdir
@@ -74,14 +74,5 @@
 		end
 		return default.rsync.prepare(config)
 	end
-}
-
-
-sync {
-	rsyncpostcmd, 
-	source = "src",
-	host = "beetle",
-	targetdir = "/path/to/trg",
-	postcmd = "/usr/local/bin/restart-servelt.sh",
 }

