From 38388a8a93d94f61f0aa470969187d218f033cb8 Mon Sep 17 00:00:00 2001
From: john30 <ebusd@ebusd.eu>
Date: Mon, 9 May 2022 20:54:18 +0200
Subject: [PATCH] use glob to find mqtt cfg files (fixes #601)

---
 CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b89c0a..43743b7 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,5 +165,6 @@ if(EXISTS "${ROOT}/etc/debian_version")
   install(FILES ${CMAKE_SOURCE_DIR}/contrib/debian/systemd/ebusd.service DESTINATION /lib/systemd/system/)
 endif()
 if(HAVE_MQTT)
-  install(FILES ${CMAKE_SOURCE_DIR}/contrib/etc/ebusd/mqtt-*.cfg DESTINATION /etc/ebusd/)
-endif(HAVE_MQTT)
\ No newline at end of file
+  FILE(GLOB MQTT_CFG_FILES "${CMAKE_SOURCE_DIR}/contrib/etc/ebusd/mqtt-*.cfg")
+  install(FILES ${MQTT_CFG_FILES} DESTINATION /etc/ebusd/)
+endif(HAVE_MQTT)
-- 
2.34.2

