--- a/node.gyp
+++ b/node.gyp
@@ -247,6 +247,9 @@
       'msvs_disabled_warnings!': [4244],
 
       'conditions': [
+        [ '(target_arch=="mips" or target_arch=="mipsel") and mips_arch_variant!="r6"', {
+          'libraries': [ '-latomic' ],
+        }],
         [ 'node_intermediate_lib_type=="static_library" and '
             'node_shared=="true" and OS=="aix"', {
           # For AIX, shared lib is linked by static lib and .exp. In the
@@ -934,6 +937,9 @@
       ],
 
       'conditions': [
+        [ '(target_arch=="mips" or target_arch=="mipsel") and mips_arch_variant!="r6"', {
+          'libraries': [ '-latomic' ],
+        }],
         [ 'node_use_openssl=="true"', {
           'defines': [
             'HAVE_OPENSSL=1',
