diff --git a/deps/ada/ada.gyp b/deps/ada/ada.gyp
index 55cea00..b5a0865 100644
--- a/deps/ada/ada.gyp
+++ b/deps/ada/ada.gyp
@@ -6,12 +6,14 @@
   'targets': [
     {
       'target_name': 'ada',
-      'type': 'static_library',
-      'include_dirs': ['.'],
+      'type': 'shared_library',
+      'include_dirs': ['/usr/include/ada'],
       'direct_dependent_settings': {
-        'include_dirs': ['.'],
-      },
-      'sources': [ '<@(ada_sources)' ]
+        'include_dirs': ['/usr/include/ada'],
+        'linkflags': ['-lada'],
+        'ldflags': ['-lada'],
+        'libraries': ['-lada']
+      }
     },
   ]
 }
diff --git a/node.gyp b/node.gyp
index 67909bd..800c916 100644
--- a/node.gyp
+++ b/node.gyp
@@ -533,6 +533,7 @@
       'dependencies': [
         'deps/histogram/histogram.gyp:histogram',
         'deps/uvwasi/uvwasi.gyp:uvwasi',
+        'deps/ada/ada.gyp:ada',
       ],
 
       'msvs_settings': {
diff --git a/src/node_url.cc b/src/node_url.cc
index 94510aa..9722590 100644
--- a/src/node_url.cc
+++ b/src/node_url.cc
@@ -114,7 +114,7 @@ void BindingData::DomainToUnicode(const FunctionCallbackInfo<Value>& args) {
   if (!out->set_hostname(input)) {
     return args.GetReturnValue().Set(String::Empty(env->isolate()));
   }
-  std::string result = ada::unicode::to_unicode(out->get_hostname());
+  std::string result = ada::idna::to_unicode(out->get_hostname());
 
   args.GetReturnValue().Set(String::NewFromUtf8(env->isolate(),
                                                 result.c_str(),
@@ -491,7 +491,7 @@ std::optional<std::string> FileURLToPath(Environment* env,
     // about percent encoding because the URL parser will have
     // already taken care of that for us. Note that this only
     // causes IDNs with an appropriate `xn--` prefix to be decoded.
-    return "\\\\" + ada::unicode::to_unicode(hostname) + decoded_pathname;
+    return "\\\\" + ada::idna::to_unicode(hostname) + decoded_pathname;
   }
 
   char letter = decoded_pathname[1] | 0x20;
