From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 19 Nov 2023 22:45:01 +0100
Subject: [PATCH] Link with shared libbase64

The second part is our modified base64.gyp.

--- a/node.gyp
+++ b/node.gyp
@@ -512,7 +512,8 @@
       'dependencies': [
         'deps/histogram/histogram.gyp:histogram',
         'deps/uvwasi/uvwasi.gyp:uvwasi',
         'deps/ada/ada.gyp:ada',
+        'deps/base64/base64.gyp:base64'
       ],
 
       'msvs_settings': {
@@ -1117,6 +1119,7 @@
         'deps/histogram/histogram.gyp:histogram',
         'deps/uvwasi/uvwasi.gyp:uvwasi',
         'deps/ada/ada.gyp:ada',
+        'deps/base64/base64.gyp:base64',
       ],
 
       'includes': [
@@ -1228,6 +1231,7 @@
         'deps/histogram/histogram.gyp:histogram',
         'deps/uvwasi/uvwasi.gyp:uvwasi',
         'deps/ada/ada.gyp:ada',
+        'deps/base64/base64.gyp:base64',
       ],
 
       'includes': [
