diff --git a/ext/standard/tests/file/bug52820.phpt b/ext/standard/tests/file/bug52820.phpt
index 2d3cedad879..06aa3463c4e 100644
--- a/ext/standard/tests/file/bug52820.phpt
+++ b/ext/standard/tests/file/bug52820.phpt
@@ -45,22 +45,22 @@ function do_stuff($url) {
 --EXPECTREGEX--
 temp stream \(close after\):
 About to rewind!
-(\* processing: file:\/\/\/i_dont_exist\/\n)?\* Couldn't open file \/i_dont_exist\/
+(\* processing: file:\/\/\/i_dont_exist\/\n)?\* (Couldn't|Could not) open file \/i_dont_exist\/
 \* [Cc]losing connection( #?-?\d+)?
 
 memory stream \(close after\):
 About to rewind!
-(\* processing: file:\/\/\/i_dont_exist\/\n)?\* Couldn't open file \/i_dont_exist\/
+(\* processing: file:\/\/\/i_dont_exist\/\n)?\* (Couldn't|Could not) open file \/i_dont_exist\/
 \* [Cc]losing connection( #?-?\d+)?
 
 temp stream \(leak\):
 About to rewind!
-(\* processing: file:\/\/\/i_dont_exist\/\n)?\* Couldn't open file \/i_dont_exist\/
+(\* processing: file:\/\/\/i_dont_exist\/\n)?\* (Couldn't|Could not) open file \/i_dont_exist\/
 \* [Cc]losing connection( #?-?\d+)?
 
 memory stream \(leak\):
 About to rewind!
-(\* processing: file:\/\/\/i_dont_exist\/\n)?\* Couldn't open file \/i_dont_exist\/
+(\* processing: file:\/\/\/i_dont_exist\/\n)?\* (Couldn't|Could not) open file \/i_dont_exist\/
 \* [Cc]losing connection( #?-?\d+)?
 
 Done\.
