From ef8a6559641967bcadd4f1e08a2ed6dec8311542 Mon Sep 17 00:00:00 2001
From: isidentical <batuhanosmantaskaya@gmail.com>
Date: Mon, 2 Sep 2019 20:07:41 +0300
Subject: [PATCH] put 2 newlines between func defs, resolves #156

---
 tests/test_code_gen.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/test_code_gen.py b/tests/test_code_gen.py
index 3cb7e64..c4935ae 100644
--- a/tests/test_code_gen.py
+++ b/tests/test_code_gen.py
@@ -171,9 +171,11 @@ def test_positional_only_arguments(self):
         def test(a, b, /, c, *, d, **kwargs):
             pass
 
+
         def test(a=3, b=4, /, c=7):
             pass
 
+
         def test(a, b=4, /, c=8, d=9):
             pass
         """
