diff --git a/src/libcharon/processing/jobs/process_message_job.c b/src/libcharon/processing/jobs/process_message_job.c
index a6795e7..bb77b3d 100644
--- a/src/libcharon/processing/jobs/process_message_job.c
+++ b/src/libcharon/processing/jobs/process_message_job.c
@@ -90,14 +90,20 @@ METHOD(job_t, get_priority, job_priority_t,
 	switch (this->message->get_exchange_type(this->message))
 	{
 		case IKE_AUTH:
+		case ID_PROT:
+		case AUTH_ONLY:
+		case AGGRESSIVE:
 			/* IKE auth is rather expensive and often blocking, low priority */
 			return JOB_PRIO_LOW;
 		case INFORMATIONAL:
+		case INFORMATIONAL_V1:
+		case TRANSACTION:
 			/* INFORMATIONALs are inexpensive, for DPD we should have low
 			 * reaction times */
 			return JOB_PRIO_HIGH;
 		case IKE_SA_INIT:
 		case CREATE_CHILD_SA:
+		case QUICK_MODE:
 		default:
 			/* IKE_SA_INIT is expensive, but we will drop them in the receiver
 			 * if we are overloaded */
