From e1519e0a9feb11b92012f29736a7feb81ea066f5 Mon Sep 17 00:00:00 2001
From: Ben Boyter <ben@boyter.org>
Date: Thu, 30 Nov 2023 08:52:10 +1100
Subject: [PATCH] ensure 32 bit works

---
 cmd/badges/simplecache.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmd/badges/simplecache.go b/cmd/badges/simplecache.go
index 3d66f64..fb4b74f 100644
--- a/cmd/badges/simplecache.go
+++ b/cmd/badges/simplecache.go
@@ -87,7 +87,7 @@ func (cache *SimpleCache) expireItems() {
 	count := 10
 	if len(cache.items) >= cache.maxItems {
 		lfuKey := ""
-		lfuLowestCount := math.MaxInt64
+		lfuLowestCount := math.MaxInt

 		for k, v := range cache.items {
 			v.hits--
--
2.43.0

