--- ./src/framework/mlt_property.h.orig	2018-11-12 15:08:18.251882145 +0000
+++ ./src/framework/mlt_property.h	2018-11-12 15:08:43.245342335 +0000
@@ -30,7 +30,7 @@
 #include <sys/param.h>
 #endif
 
-#if (defined(__GLIBC__) && !defined(__APPLE__)) || HAVE_LOCALE_H
+#if (defined(__linux__) && !defined(__APPLE__)) || HAVE_LOCALE_H
 #  include <locale.h>
 #elif defined(__APPLE__) || (__FreeBSD_version >= 900506)
 #  include <xlocale.h>
