diff --git a/src/swig/python/build b/src/swig/python/build
index db31e16..4eb69e2 100755
--- a/src/swig/python/build
+++ b/src/swig/python/build
@@ -8,12 +8,12 @@ then
 	exit 0
 fi
 
-path=`which python 2> /dev/null`
+path=`which python3 2> /dev/null`
 
 if [ $? = 0 ]
 then
 	# Change this as needed
-	export PYTHON_INCLUDE=`python -c "import sys;print(\"{}/include/python{}.{}\".format(sys.prefix,*sys.version_info))"`
+	export PYTHON_INCLUDE=`python3 -c "import sys;print(\"{}/include/python{}.{}m\".format(sys.prefix,*sys.version_info))"`
 
 	[ ! -d "$PYTHON_INCLUDE" ] && echo python development missing && exit 1
 
