Fix -Wimplicit-function-declaration error with gcc 14.

--- dircproxy-RELEASE-1_2_0-BETA-origin/getopt/getopt.c
+++ dircproxy-RELEASE-1_2_0-BETA/getopt/getopt.c
@@ -75,6 +75,8 @@
 #define	__alloca	alloca
 #endif	/* GNU C library.  */
 
+#include <string.h>
+
 /* If GETOPT_COMPAT is defined, `+' as well as `--' can introduce a
    long-named option.  Because this is not POSIX.2 compliant, it is
    being phased out.  */
