From fe1d021c4aae19c5772e2869b68a9c6ccea976d5 Mon Sep 17 00:00:00 2001
From: "francois.harvey" <francois.harvey@francoisharvey.ca>
Date: Sun, 4 Jan 2009 23:27:19 +0000
Subject: [PATCH] Fix for #54

---
 src/irc_server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/irc_server.c b/src/irc_server.c
index 185107e..550a1ff 100644
--- a/src/irc_server.c
+++ b/src/irc_server.c
@@ -1155,7 +1155,7 @@ static int _ircserver_gotmsg(struct ircproxy *p, const char *str) {
       
         if (!strcmp(cmsg.cmd, "ACTION")) {
           irclog_log(p, IRC_LOG_ACTION, logdest, msg.src.orig,
-                     "%s", (cmsg.paramstarts != NULL) ?  cmsg.paramstarts[0]: "none");
+                     "%s", (cmsg.paramstarts != NULL) ?  cmsg.paramstarts[0]: "");
 
         } else if (!strcmp(cmsg.cmd, "DCC")
                    && p->conn_class->dcc_proxy_incoming) {
