--- a/em-websocket.gemspec
+++ b/em-websocket.gemspec
@@ -13,9 +13,7 @@
   s.description = %q{EventMachine based WebSocket server}
   s.license     = 'MIT'
 
-  s.files         = `git ls-files`.split("\n")
-  s.test_files    = `git ls-files -- {test,spec,features}/*`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
+  s.files         = Dir["lib/**/*"]
   s.require_paths = ["lib"]
 
   s.add_dependency("eventmachine", ">= 0.12.9")
