Use time_t in overflow_add() time value.

--- calcurse-4.8.1-orig/src/calcurse.h
+++ calcurse-4.8.1/src/calcurse.h
@@ -1272,7 +1272,7 @@
 int starts_with(const char *, const char *);
 int starts_with_ci(const char *, const char *);
 int hash_matches(const char *, const char *);
-long overflow_add(long, long, long *);
+long overflow_add(long, long, time_t *);
 long overflow_mul(long, long, long *);
 time_t next_wday(time_t, int);
 int wday_per_year(int, int);
--- calcurse-4.8.1-orig/src/utils.c
+++ calcurse-4.8.1/src/utils.c
@@ -1260,8 +1260,8 @@
 	dur += in;
 	if (start) {
 		/* wanted: end = start + dur * MININSEC */
-		time_t end;
-		long p, s;
+		time_t end, s;
+		long p;
 		if (overflow_mul(dur, MININSEC, &p))
 			return 0;
 		if (overflow_add(start, p, &s))
@@ -2043,7 +2043,7 @@
 /*
  * Overflow check for addition with positive second term.
  */
-long overflow_add(long x, long y, long *z)
+long overflow_add(long x, long y, time_t *z)
 {
 	if (!YEAR1902_2037)
 		goto exit;
