--- firetools-0.9.30/configure
+++ firetools-0.9.30/configure.new
@@ -3811,18 +3811,24 @@
 		as_fn_error $? "*** Please install Qt5 svg support - apt-get install libqt5svg5 ***" "$LINENO" 5
 	fi
 
-# Arch Linux 64bit
+# Arch Linux 64bit & Alpine Linux
 elif test -f /usr/lib/libQt5Core.so
 then
+	if test -d /usr/lib/qt5
+	then
+		qtdir='qt5'
+	else
+		qtdir='qt'
+	fi
         echo "Found qt5 library in /usr/lib directory"
         # check libqt5svg5 library
-        if test -f /usr/lib/qt/plugins/imageformats/libqsvg.so
+        if test -f /usr/lib/$qtdir/plugins/imageformats/libqsvg.so
         then
                 echo "Found qt5 svg library"
         else
                 as_fn_error $? "*** Please install Qt5 svg support - pacman -S qt5-svg ***" "$LINENO" 5
         fi
-        if test -f /usr/lib/qt/plugins/iconengines/libqsvgicon.so
+        if test -f /usr/lib/$qtdir/plugins/iconengines/libqsvgicon.so
         then
                 echo "Found qt5 svg icon library"
         else

--- firetools-0.9.30/configure
+++ firetools-0.9.30/configure.new
@@ -3939,6 +3939,11 @@
 		QMAKE=/usr/lib64/qt4/bin/qmake
 	fi
 
+        # fix for Alpine Linux
+        if test -f /usr/lib/qt5/bin/qmake; then
+                QMAKE=/usr/lib/qt5/bin/qmake
+        fi
+
 	if test -z "$QMAKE"
 	then
 		as_fn_error $? "qmake is missing, please install Qt4 or Qt5 development packages." "$LINENO" 5

