--- freetype-2.4.2.subpixel-patched/include/freetype/config/ftoption.h	2010-09-09 18:08:11.375855632 -0500
+++ freetype-2.4.2.subpixel-enabled/include/freetype/config/ftoption.h	2010-09-09 18:09:48.491864607 -0500
@@ -93,7 +93,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
@@ -510,7 +510,7 @@
   /*   This option requires TT_CONFIG_OPTION_BYTECODE_INTERPRETER to be    */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING
 
 
   /*************************************************************************/
@@ -527,7 +527,7 @@
   /*   This option requires TT_CONFIG_OPTION_SUBPIXEL_HINTING to be        */
   /*   defined.                                                            */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_SUBPIXEL_HINTING_ADDITIONAL_TWEAKS */
+#define TT_CONFIG_OPTION_SUBPIXEL_HINTING_ADDITIONAL_TWEAKS
 
 
   /*************************************************************************/
