--- a/modules.cfg	2013-11-12 13:02:26.000000000 +0100
+++ b/modules.cfg	2014-03-07 19:18:00.546655169 +0100
@@ -110,7 +110,7 @@ RASTER_MODULES += smooth
 AUX_MODULES += cache
 
 # TrueType GX/AAT table validation.  Needs ftgxval.c below.
-# AUX_MODULES += gxvalid
+AUX_MODULES += gxvalid
 
 # Support for streams compressed with gzip (files with suffix .gz).
 #
@@ -129,7 +129,7 @@ AUX_MODULES += bzip2
 
 # OpenType table validation.  Needs ftotval.c below.
 #
-# AUX_MODULES += otvalid
+AUX_MODULES += otvalid
 
 # Auxiliary PostScript driver component to share common code.
 #
