From 82f73fd89bbb37d703fc387f5249471f0ff7bd45 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?George=20Florea=20B=C4=83nu=C8=99?= <georgefb899@gmail.com>
Date: Thu, 30 Sep 2021 19:30:28 +0300
Subject: [PATCH] fix(mpris): only show filename if there's no title

BUG: 443157
---
 src/mpris2/mediaplayer2player.cpp | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/src/mpris2/mediaplayer2player.cpp b/src/mpris2/mediaplayer2player.cpp
index ac13726..ca34d7c 100644
--- a/src/mpris2/mediaplayer2player.cpp
+++ b/src/mpris2/mediaplayer2player.cpp
@@ -125,12 +125,15 @@ QVariantMap MediaPlayer2Player::Metadata()
                     m_mpv->getProperty("duration").toDouble() * 1000 * 1000);
     metadata.insert(QStringLiteral("mpris:trackid"),
                     QVariant::fromValue<QDBusObjectPath>(QDBusObjectPath("/org/kde/haruna")));
-    metadata.insert(QStringLiteral("xesam:title"),
-                    m_mpv->getProperty("filename").toString());
+
+    auto mpvMediaTitle = m_mpv->getProperty("media-title").toString();
+    auto mpvFilename = m_mpv->getProperty("filename").toString();
+    auto title = mpvMediaTitle.isEmpty() || mpvMediaTitle.isNull() ? mpvFilename : mpvMediaTitle;
+    metadata.insert(QStringLiteral("xesam:title"), title);
+
     QUrl url(m_mpv->getProperty("path").toString());
     url.setScheme("file");
-    metadata.insert(QStringLiteral("xesam:url"),
-                    url.toString());
+    metadata.insert(QStringLiteral("xesam:url"), url.toString());
 
     return metadata;
 }
-- 
GitLab

