Upstream: not yet, need proper alternative.
--- a/libs/pbd/wscript	2020-05-27 15:46:59.641983074 +0200
+++ b/libs/pbd/wscript	2020-05-27 15:47:08.208661970 +0200
@@ -181,8 +181,7 @@
     obj.install_path = bld.env['LIBDIR']
     obj.defines     += [ 'PACKAGE="' + I18N_PACKAGE + '"' ]
 
-    if sys.platform.startswith('netbsd'):
-        obj.linkflags = '-lexecinfo'
+    obj.ldflags = '-lexecinfo'
 
     if bld.env['build_target'] == 'x86_64':
         obj.defines += [ 'USE_X86_64_ASM' ]
