diff --git a/configure b/configure
index 99fc2de..5aad904 100755
--- a/configure
+++ b/configure
@@ -5392,19 +5392,19 @@ fi
 if test x$build_html = xyes; then
 
 pkg_failed=no
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for webkit2gtk-4.0" >&5
-printf %s "checking for webkit2gtk-4.0... " >&6; }
+{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for webkit2gtk-4.1" >&5
+printf %s "checking for webkit2gtk-4.1... " >&6; }
 
 if test -n "$HTML_CFLAGS"; then
     pkg_cv_HTML_CFLAGS="$HTML_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_HTML_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.0" 2>/dev/null`
+  pkg_cv_HTML_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.1" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -5416,12 +5416,12 @@ if test -n "$HTML_LIBS"; then
     pkg_cv_HTML_LIBS="$HTML_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0") 2>&5
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1\""; } >&5
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_HTML_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.0" 2>/dev/null`
+  pkg_cv_HTML_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.1" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -5442,9 +5442,9 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        HTML_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.0" 2>&1`
+	        HTML_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.1" 2>&1`
         else
-	        HTML_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.0" 2>&1`
+	        HTML_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.1" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$HTML_PKG_ERRORS" >&5
