--- a/src/gallium/winsys/radeon/drm/Makefile.am	2017-08-10 11:45:20.737377457 -0400
+++ b/src/gallium/winsys/radeon/drm/Makefile.am	2017-08-10 11:50:23.414534307 -0400
@@ -7,4 +7,6 @@
 
 noinst_LTLIBRARIES = libradeonwinsys.la
 
+libradeonwinsys_la_LIBADD = -ldrm_radeon
+
 libradeonwinsys_la_SOURCES = $(C_SOURCES)
--- a/src/gallium/winsys/radeon/drm/Makefile.in	2017-08-10 11:45:25.577298181 -0400
+++ b/src/gallium/winsys/radeon/drm/Makefile.in	2017-08-10 11:57:11.151189074 -0400
@@ -118,7 +118,7 @@
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-libradeonwinsys_la_LIBADD =
+libradeonwinsys_la_LIBADD = -ldrm_radeon
 am__objects_1 = radeon_drm_bo.lo radeon_drm_cs.lo \
 	radeon_drm_surface.lo radeon_drm_winsys.lo
 am_libradeonwinsys_la_OBJECTS = $(am__objects_1)
--- a/src/gallium/winsys/amdgpu/drm/Makefile.am	2017-08-10 11:58:51.446212948 -0400
+++ b/src/gallium/winsys/amdgpu/drm/Makefile.am	2017-08-10 11:57:54.447146570 -0400
@@ -10,5 +10,5 @@
 
 noinst_LTLIBRARIES = libamdgpuwinsys.la
 
-libamdgpuwinsys_la_LIBADD = $(top_builddir)/src/amd/addrlib/libamdgpu_addrlib.la
+libamdgpuwinsys_la_LIBADD = $(top_builddir)/src/amd/addrlib/libamdgpu_addrlib.la -ldrm_amdgpu
 libamdgpuwinsys_la_SOURCES = $(C_SOURCES)
--- a/src/gallium/winsys/amdgpu/drm/Makefile.in	2017-08-10 11:59:12.009209467 -0400
+++ b/src/gallium/winsys/amdgpu/drm/Makefile.in	2017-08-10 11:58:23.626668621 -0400
@@ -561,7 +561,7 @@
 
 AM_CXXFLAGS = $(AM_CFLAGS)
 noinst_LTLIBRARIES = libamdgpuwinsys.la
-libamdgpuwinsys_la_LIBADD = $(top_builddir)/src/amd/addrlib/libamdgpu_addrlib.la
+libamdgpuwinsys_la_LIBADD = $(top_builddir)/src/amd/addrlib/libamdgpu_addrlib.la -ldrm_amdgpu
 libamdgpuwinsys_la_SOURCES = $(C_SOURCES)
 all: all-am
 
