diff -ru Mesa-7.4.orig/src/mesa/drivers/dri/Makefile Mesa-7.4/src/mesa/drivers/dri/Makefile
--- Mesa-7.4.orig/src/mesa/drivers/dri/Makefile	2009-04-11 18:08:41.000000000 +0000
+++ Mesa-7.4/src/mesa/drivers/dri/Makefile	2009-04-11 18:10:09.000000000 +0000
@@ -18,7 +18,7 @@
 $(TOP)/$(LIB_DIR)/libdricore.so: $(TOP)/$(LIB_DIR) libdricore.so
 	$(INSTALL) libdricore.so $(TOP)/$(LIB_DIR) 
 
-subdirs:
+subdirs: $(TOP)/$(LIB_DIR)/libdricore.so
 	@for dir in $(DRI_DIRS) ; do \
 		if [ -d $$dir ] ; then \
 			(cd $$dir && $(MAKE)) || exit 1 ; \
