diff --git a/src/mesa/main/imports.c b/src/mesa/main/imports.c
index 46ffb92..6192fe6 100644
--- a/src/mesa/main/imports.c
+++ b/src/mesa/main/imports.c
@@ -915,7 +915,7 @@ _mesa_atoi(const char *s)
 double
 _mesa_strtod( const char *s, char **end )
 {
-#ifdef _GNU_SOURCE
+#if defined(_GNU_SOURCE) && ! defined(__UCLIBC__)
    static locale_t loc = NULL;
    if (!loc) {
       loc = newlocale(LC_CTYPE_MASK, "C", NULL);
