--- Mesa-7.10/src/mesa/main/imports.c
+++ Mesa-7.10.mod/src/mesa/main/imports.c
@@ -757,7 +757,7 @@
 float
 _mesa_strtof( const char *s, char **end )
 {
-#if defined(_GNU_SOURCE) && !defined(__CYGWIN__) && !defined(__FreeBSD__)
+#if defined(_GNU_SOURCE) && !defined(__CYGWIN__) && !defined(__FreeBSD__) && !defined(__UCLIBC__)
    static locale_t loc = NULL;
    if (!loc) {
       loc = newlocale(LC_CTYPE_MASK, "C", NULL);
Only in Mesa-7.10.mod: src/mesa/main/imports.c.orig
Only in Mesa-7.10.mod: src/mesa/main/imports.c.rej
