--- Mesa-7.10.orig/src/gallium/auxiliary/util/u_debug_symbol.c
+++ Mesa-7.10/src/gallium/auxiliary/util/u_debug_symbol.c
@@ -145,7 +145,7 @@
 }
 #endif
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
 #include <execinfo.h>
 
 /* This can only provide dynamic symbols, or binary offsets into a file.
@@ -171,7 +171,7 @@
       return;
 #endif
 
-#ifdef __GLIBC__
+#if defined(__GLIBC__) && !defined(__UCLIBC__)
    debug_symbol_name_glibc(addr, buf, size);
    if(buf[0])
       return;
