--- a/src/util/rand_xor.c
+++ b/src/util/rand_xor.c
@@ -24,6 +24,8 @@
 
 #if defined(__linux__)
 #include <sys/file.h>
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <fcntl.h>
 #else
--- a/src/gallium/winsys/svga/drm/vmw_screen.h	2018-04-03 20:27:45.000000000 +0300
+++ b/src/gallium/winsys/svga/drm/vmw_screen.h	2018-04-15 23:56:44.392333597 +0300
@@ -34,7 +34,7 @@
 #ifndef VMW_SCREEN_H_
 #define VMW_SCREEN_H_
 
-
+#include <sys/types.h>
 #include "pipe/p_compiler.h"
 #include "pipe/p_state.h"
 
