--- src/mesa/Makefile.orig
+++ ./src/mesa/Makefile
@@ -65,7 +65,10 @@
 
 # Default: build dependencies, then asm_subdirs, GLSL built-in lib,
 # then convenience libs (.a) and finally the device drivers:
-default: $(DEPENDS) asm_subdirs $(MESA_LIBS) $(DRICORE_LIBS) driver_subdirs
+default: $(DEPENDS)
+	@for i in asm_subdirs $(MESA_LIBS) $(DRICORE_LIBS) driver_subdirs; do \
+		$(MAKE) $$i || exit 1; \
+	done
 
 # include glapi_gen.mk for generating glapi headers for GLES
 GLAPI := $(TOP)/src/mapi/glapi/gen
