--- a/app/main/index.ts
+++ b/app/main/index.ts
@@ -22,7 +22,6 @@ import {bundlePath, bundleUrl, publicPath} from "../common/paths.js";
 import type {RendererMessage} from "../common/typed-ipc.js";
 import type {MenuProperties} from "../common/types.js";
 
-import {appUpdater, shouldQuitForUpdate} from "./autoupdater.js";
 import * as BadgeSettings from "./badge-settings.js";
 import handleExternalLink from "./handle-external-link.js";
 import * as AppMenu from "./menu.js";
@@ -106,7 +105,7 @@ function createMainWindow(): BrowserWindow {
       app.quit();
     }
 
-    if (!isQuitting && !shouldQuitForUpdate()) {
+    if (!isQuitting) {
       event.preventDefault();
 
       if (process.platform === "darwin") {
@@ -296,13 +295,6 @@ function createMainWindow(): BrowserWindow {
     _isOnline(url, ses),
   );
 
-  page.once("did-frame-finish-load", async () => {
-    // Initiate auto-updates on MacOS and Windows
-    if (ConfigUtil.getConfigItem("autoUpdate", true)) {
-      await appUpdater();
-    }
-  });
-
   app.on(
     "certificate-error",
     (

