diff --git a/CMakeLists.txt b/CMakeLists.txt
index fda60b7..cca69ff 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,8 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 add_definitions(-DBOOST_MPL_LIMIT_LIST_SIZE=30)
 add_definitions(-DBOOST_MPL_CFG_NO_PREPROCESSED_HEADERS)
+add_definitions(-DSPDLOG_COMPILED_LIB)
+add_definitions(-DSPDLOG_FMT_EXTERNAL)
 
 include(GNUInstallDirs)
 
@@ -395,6 +397,7 @@ set(COMMON_LIBS
     MatrixClient::MatrixClient
     ${Boost_LIBRARIES}
     cmark::cmark
+    spdlog::spdlog
     Qt5::Widgets
     Qt5::Svg
     Qt5::Concurrent

