--- openjdk.orig/hotspot/agent/src/os/solaris/dbx/Makefile
+++ openjdk/hotspot/agent/src/os/solaris/dbx/Makefile
@@ -50,7 +50,7 @@
 CFLAGS_32bit := -xarch=v8
 CFLAGS_64bit := -xarch=v9
 CFLAGS   := -PIC -xO3 $(INCLUDES)
-LIBS     := -lsocket -lnsl -lrtld_db
+LIBS     := -lsocket  -lrtld_db
 LDFLAGS  := -G
 
 ifneq "$(ARCH)" "i486"
--- openjdk.orig/jdk/make/java/hpi/hpi_common.gmk
+++ openjdk/jdk/make/java/hpi/hpi_common.gmk
@@ -86,5 +86,5 @@
 # Things that must be linked in.
 #
 ifneq ($(PLATFORM), windows)
-OTHER_LDLIBS += $(LIBSOCKET) -lnsl $(LIBM) -ldl
+OTHER_LDLIBS += $(LIBSOCKET)  $(LIBM) -ldl
 endif
--- openjdk.orig/jdk/make/java/java/Makefile
+++ openjdk/jdk/make/java/java/Makefile
@@ -205,7 +205,7 @@
 OTHER_LDLIBS += $(JVMLIB) -libpath:$(OBJDIR)/../../../fdlibm/$(OBJDIRNAME) fdlibm.lib \
 		       -libpath:$(OBJDIR)/../../../verify/$(OBJDIRNAME) verify.lib
 else
-OTHER_LDLIBS += $(JVMLIB) -lverify $(LIBSOCKET) -lnsl -ldl \
+OTHER_LDLIBS += $(JVMLIB) -lverify $(LIBSOCKET)  -ldl \
 		-L$(OBJDIR)/../../../fdlibm/$(OBJDIRNAME) -lfdlibm.$(ARCH)
 endif
 
--- openjdk.orig/jdk/make/java/java_hprof_demo/Makefile
+++ openjdk/jdk/make/java/java_hprof_demo/Makefile
@@ -83,7 +83,7 @@
 ifeq ($(PLATFORM), windows)
   OTHER_LDLIBS += wsock32.lib winmm.lib
 else
-  OTHER_LDLIBS += $(LIBSOCKET) -lnsl -ldl
+  OTHER_LDLIBS += $(LIBSOCKET)  -ldl
 endif
 
 #
--- openjdk.orig/jdk/make/java/net/Makefile
+++ openjdk/jdk/make/java/net/Makefile
@@ -97,7 +97,7 @@
   # Will not compile at warning level 3 if warnings are fatal
   COMPILER_WARNINGS_FATAL=false
 else
-  OTHER_LDLIBS = $(LIBSOCKET) -lnsl -ldl $(JVMLIB)
+  OTHER_LDLIBS = $(LIBSOCKET)  -ldl $(JVMLIB)
 endif
 ifeq ($(PLATFORM), linux)
   OTHER_LDLIBS += -lpthread
--- openjdk.orig/jdk/make/jpda/transport/socket/Makefile
+++ openjdk/jdk/make/jpda/transport/socket/Makefile
@@ -42,11 +42,11 @@
 endif
 
 ifeq ($(PLATFORM), linux)
-  OTHER_LDLIBS += -lnsl $(LIBSOCKET) -lpthread
+  OTHER_LDLIBS +=  $(LIBSOCKET) -lpthread
 endif
 
 ifeq ($(PLATFORM), solaris)
-  OTHER_LDLIBS += -lnsl $(LIBSOCKET)
+  OTHER_LDLIBS +=  $(LIBSOCKET)
 endif
 
 ifeq ($(PLATFORM), windows)
--- openjdk.orig/jdk/make/mkdemo/jvmti/hprof/Makefile
+++ openjdk/jdk/make/mkdemo/jvmti/hprof/Makefile
@@ -39,10 +39,10 @@
   EXTRA_LIBS += wsock32.lib winmm.lib
 endif
 ifeq ($(PLATFORM), solaris)
-  OTHER_LDLIBS += $(LIBSOCKET) -lnsl -ldl
+  OTHER_LDLIBS += $(LIBSOCKET)  -ldl
 endif
 ifeq ($(PLATFORM), linux)
-  OTHER_LDLIBS += $(LIBSOCKET) -lnsl -ldl -lpthread
+  OTHER_LDLIBS += $(LIBSOCKET)  -ldl -lpthread
 endif
 
 #
