--- ./openjdk/jdk/make/java/main/java/Makefile.orig	2013-04-26 08:10:58.853977182 +0000
+++ ./openjdk/jdk/make/java/main/java/Makefile	2013-04-26 08:15:20.819826501 +0000
@@ -58,6 +58,10 @@
 OTHER_CPPFLAGS += -DEXPAND_CLASSPATH_WILDCARDS
 OTHER_CPPFLAGS += -DLAUNCHER_NAME='"$(LAUNCHER_NAME)"'
 
+# We need pax mark java early
+$(ACTUAL_PROGRAM)::
+	paxctl -c -mr $@
+
 ifeq ($(PLATFORM), solaris)
 LDFLAGS += -R$(OPENWIN_LIB)
 endif
