diff --git a/tests/conftest.py.orig b/tests/conftest.py
index e8d616d..01cdb99 100644
--- a/tests/conftest.py.orig
+++ b/tests/conftest.py
@@ -11,7 +11,6 @@ import pymongo
 import pytest
 import redis
 import redis.sentinel
-import valkey
 
 
 def check_redis_cluster_ready(host, port):
diff --git a/pytest.ini.orig b/pytest.ini
index 993f08a..5b72c13 100644
--- a/pytest.ini.orig
+++ b/pytest.ini
@@ -19,7 +19,6 @@ addopts =
     -rfEsxX
     --cov=limits
     -m "not benchmark"
-    -K
 filterwarnings =
     error
     module::ResourceWarning
