--- gnash-0.8.9.orig/libmedia/ffmpeg/MediaParserFfmpeg.cpp
+++ gnash-0.8.9/libmedia/ffmpeg/MediaParserFfmpeg.cpp
@@ -52,8 +52,8 @@
 	return p->readPacket(buf, buf_size);
 }
 
-boost::int64_t
-MediaParserFfmpeg::seekMediaWrapper(void *opaque, boost::int64_t offset, int whence)
+::int64_t
+MediaParserFfmpeg::seekMediaWrapper(void *opaque, ::int64_t offset, int whence)
 {
 	MediaParserFfmpeg* p = static_cast<MediaParserFfmpeg*>(opaque);
 	return p->seekMedia(offset, whence);
--- gnash-0.8.9.orig/libmedia/ffmpeg/MediaParserFfmpeg.h
+++ gnash-0.8.9/libmedia/ffmpeg/MediaParserFfmpeg.h
@@ -126,7 +126,7 @@
 	boost::int64_t seekMedia(boost::int64_t offset, int whence);
 
 	/// ffmpeg callback function
-	static boost::int64_t seekMediaWrapper(void *opaque, boost::int64_t offset, int whence);
+	static int64_t seekMediaWrapper(void *opaque, int64_t offset, int whence);
 
 	/// Read some of the input to figure an AVInputFormat
 	AVInputFormat* probeStream();
