diff --git a/third_party/dbus_gatt/CMakeLists.txt b/third_party/dbus_gatt/CMakeLists.txt
index ef07d8e..1a5b988 100644
--- a/third_party/dbus_gatt/CMakeLists.txt
+++ b/third_party/dbus_gatt/CMakeLists.txt
@@ -27,7 +27,7 @@ endif ()
 include_directories(
     include
     ${GLIB_INCLUDE_DIRS}
-    ${DBUS_INCLUDE_DIRS}
+    ${DBus_INCLUDE_DIRS}
     ${GIO_INCLUDE_DIR}
     ${Boost_INCLUDE_DIRS}
 )
diff --git a/third_party/dbus_gatt/cmake/FindDBus.cmake b/third_party/dbus_gatt/cmake/FindDBus.cmake
index e677b77..475e5bc 100644
--- a/third_party/dbus_gatt/cmake/FindDBus.cmake
+++ b/third_party/dbus_gatt/cmake/FindDBus.cmake
@@ -29,27 +29,27 @@
 # ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 FIND_PACKAGE(PkgConfig)
-PKG_CHECK_MODULES(PC_DBUS QUIET dbus-1)
+PKG_CHECK_MODULES(PC_DBus QUIET dbus-1)
 
-FIND_LIBRARY(DBUS_LIBRARIES
+FIND_LIBRARY(DBus_LIBRARIES
     NAMES dbus-1
-    HINTS ${PC_DBUS_LIBDIR}
-    ${PC_DBUS_LIBRARY_DIRS}
+    HINTS ${PC_DBus_LIBDIR}
+    ${PC_DBus_LIBRARY_DIRS}
     )
 
-FIND_PATH(DBUS_INCLUDE_DIR
+FIND_PATH(DBus_INCLUDE_DIR
     NAMES dbus/dbus.h
-    HINTS ${PC_DBUS_INCLUDEDIR}
-    ${PC_DBUS_INCLUDE_DIRS}
+    HINTS ${PC_DBus_INCLUDEDIR}
+    ${PC_DBus_INCLUDE_DIRS}
     )
 
-GET_FILENAME_COMPONENT(_DBUS_LIBRARY_DIR ${DBUS_LIBRARIES} PATH)
-FIND_PATH(DBUS_ARCH_INCLUDE_DIR
+GET_FILENAME_COMPONENT(_DBus_LIBRARY_DIR ${DBus_LIBRARIES} PATH)
+FIND_PATH(DBus_ARCH_INCLUDE_DIR
     NAMES dbus/dbus-arch-deps.h
-    HINTS ${PC_DBUS_INCLUDEDIR}
-    ${PC_DBUS_INCLUDE_DIRS}
-    ${_DBUS_LIBRARY_DIR}
-    ${DBUS_INCLUDE_DIR}
+    HINTS ${PC_DBus_INCLUDEDIR}
+    ${PC_DBus_INCLUDE_DIRS}
+    ${_DBus_LIBRARY_DIR}
+    ${DBus_INCLUDE_DIR}
     PATH_SUFFIXES dbus-1.0/include
     )
 
@@ -58,10 +58,10 @@ FIND_PATH(DBUS_ARCH_INCLUDE_DIR
 #message("!!!!!!!!!!! ${DBUS_LIBRARIES}   | ${PC_DBUS_LIBDIR}     | ${PC_DBUS_LIBRARY_DIRS}")
 #message("!!!!!!!!!!! ${DBUS_INCLUDE_DIR} | ${PC_DBUS_INCLUDEDIR} | ${PC_DBUS_INCLUDE_DIRS}")
 
-SET(DBUS_INCLUDE_DIRS ${DBUS_INCLUDE_DIR} ${DBUS_ARCH_INCLUDE_DIR})
+SET(DBus_INCLUDE_DIRS ${DBus_INCLUDE_DIR} ${DBus_ARCH_INCLUDE_DIR})
 
 #message("!!!!!!!!!!! ${DBUS_ARCH_INCLUDE_DIR} |")
 #message("!!!!!!!!!!! ${DBUS_INCLUDE_DIRS} | ${PC_DBUS_INCLUDEDIR} | ${PC_DBUS_INCLUDE_DIRS}")
 
 INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(DBUS REQUIRED_VARS DBUS_INCLUDE_DIRS DBUS_LIBRARIES)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(DBus REQUIRED_VARS DBus_INCLUDE_DIRS DBus_LIBRARIES)
