--- a/third_party/dbus_gatt/src/impl/dbus_method.h
+++ b/third_party/dbus_gatt/src/impl/dbus_method.h
@@ -23,6 +23,7 @@
 #ifndef DBUS_GATT_DBUS_METHOD_H_
 #define DBUS_GATT_DBUS_METHOD_H_
 
+#include <cstdint>
 #include <functional>
 #include <iostream>
 #include <string>
--- a/third_party/dbus_gatt/include/dbus_gatt/dbus_gatt_types.h
+++ b/third_party/dbus_gatt/include/dbus_gatt/dbus_gatt_types.h
@@ -24,6 +24,7 @@
 #define DBUS_GATT_SRC_DBUS_GATT_TYPES_H_
 
 #include <cstddef>
+#include <cstdint>
 #include <functional>
 #include <variant>
 
