--- a/cmd/grype/cli/options/grype.go
+++ b/cmd/grype/cli/options/grype.go
@@ -68,7 +68,7 @@ func DefaultGrype(id clio.Identification) *Grype {
 		},
 		Match:                      defaultMatchConfig(),
 		ExternalSources:            defaultExternalSources(),
-		CheckForAppUpdate:          true,
+		CheckForAppUpdate:          false,
 		VexAdd:                     []string{},
 		MatchUpstreamKernelHeaders: false,
 		SortBy:                     defaultSortBy(),
