https://bugzilla.xfce.org/show_bug.cgi?id=11083#c9

diff -ru xfce4-power-manager-1.5.2.orig/src/xfpm-power.c xfce4-power-manager-1.5.2/src/xfpm-power.c
--- xfce4-power-manager-1.5.2.orig/src/xfpm-power.c	2015-06-11 18:34:58.000000000 +0300
+++ xfce4-power-manager-1.5.2/src/xfpm-power.c	2015-10-26 16:56:30.748006741 +0200
@@ -965,7 +965,7 @@
 
     XResetScreenSaver (dpy);
 
-#ifdef HAVE_LIBXTST
+#if 0 && defined HAVE_LIBXTST
     /* keycode of 255 does not map to any actual key,
      * this works for xscreensaver */
     XTestFakeKeyEvent (dpy, 255, TRUE, 0);
