--- deadbeef-0.4.4.orig/junklib.c
+++ deadbeef-0.4.4/junklib.c
@@ -155,7 +155,7 @@
 int
 junk_iconv (const char *in, int inlen, char *out, int outlen, const char *cs_in, const char *cs_out) {
 #if HAVE_ICONV
-    iconv_t cd = iconv_open (cs_out, cs_in);
+    iconv_t cd = libiconv_open (cs_out, cs_in);
     if (cd == (iconv_t)-1) {
         return -1;
     }
@@ -171,9 +171,9 @@
     char *pout = out;
     memset (out, 0, outbytesleft);
 
-    size_t res = iconv (cd, &pin, &inbytesleft, &pout, &outbytesleft);
+    size_t res = libiconv (cd, &pin, &inbytesleft, &pout, &outbytesleft);
     int err = errno;
-    iconv_close (cd);
+    libiconv_close (cd);
 
     //trace ("iconv -f %s -t %s '%s': returned %d, inbytes %d/%d, outbytes %d/%d, errno=%d\n", cs_in, cs_out, in, res, inlen, inbytesleft, outlen, outbytesleft, err);
     if (res == -1) {
