--- a/common.mak
+++ b/common.mak
@@ -51,7 +51,7 @@ inst_inf.o  : inst_inf.c term.h inst.h basic.h
 main.o  : main.c term.h inst.h basic.h
 	$(CCC) $(CFLAGS) main.c
 toam.o  : toam.c term.h inst.h basicd.h basic.h toam.h event.h frame.h emu_inst.h
-	$(CCC) $(CFLAGS) toam.c
+	$(CCC) -std=gnu17 $(CFLAGS) toam.c
 toamprofile.o  : toamprofile.c term.h inst.h basicd.h basic.h toam.h event.h frame.h
 	$(CCC) $(CFLAGS) toamprofile.c
 unify.o : unify.c term.h basic.h event.h bapi.h frame.h
@@ -67,13 +67,13 @@ float1.o	: float1.c term.h basic.h bapi.h
 arith.o	: arith.c term.h basic.h bapi.h
 	$(CCC) $(CFLAGS) arith.c
 bigint.o	: bigint.c term.h basic.h bapi.h
-	$(CCC) $(CFLAGS) bigint.c
+	$(CCC) -std=gnu17 $(CFLAGS) bigint.c
 token.o	: token.c term.h basic.h bapi.h
 	$(CCC) $(CFLAGS) token.c
 global.o	: global.c term.h basic.h bapi.h
 	$(CCC) $(CFLAGS) global.c
 builtins.o	: builtins.c term.h basic.h bapi.h
-	$(CCC) $(CFLAGS) builtins.c
+	$(CCC) -std=gnu17 $(CFLAGS) builtins.c
 mic.o	: mic.c term.h basic.h bapi.h
 	$(CCC) $(CFLAGS) mic.c
 numbervars.o	: numbervars.c term.h basic.h bapi.h
@@ -89,13 +89,15 @@ findall.o	: findall.c term.h basic.h bapi.h
 clause.o	: clause.c term.h basic.h bapi.h dynamic.h
 	$(CCC) $(CFLAGS) clause.c
 delay.o	: delay.c term.h basic.h bapi.h
-	$(CCC) $(CFLAGS) delay.c
+	$(CCC) -std=gnu17 $(CFLAGS) delay.c
 clpfd.o : clpfd.c term.h basic.h bapi.h event.h frame.h
 	$(CCC) $(CFLAGS) clpfd.c
 clpfd_libs.o : clpfd_libs.c term.h basic.h bapi.h event.h frame.h
 	$(CCC) $(CFLAGS) clpfd_libs.c
 kapi.o : kapi.c term.h basic.h bapi.h kapi.h
 	$(CCC) $(CFLAGS) kapi.c
+getline.o : getline.c
+	$(CCC) -std=gnu17 $(CFLAGS) -c -o $@ $<
 ki.o : kapi.c term.h basic.h bapi.h kapi.h
 	$(CCC) $(CFLAGS) kapi.c
 event.o : event.c term.h basic.h bapi.h kapi.h event.h
@@ -115,7 +117,7 @@ expand_bp.o : expand_bp.c gc.h term.h basic.h bapi.h frame.h
 espresso_bp.o : espresso_bp.c term.h basic.h bapi.h frame.h
 	$(CCC) $(CFLAGS) -Iespresso espresso_bp.c
 sapi.o : sapi.c sapi.h term.h basic.h bapi.h frame.h
-	$(CCC) $(CFLAGS) sapi.c
+	$(CCC) -std=gnu17 $(CFLAGS) sapi.c
 kissat_picat.o : kissat_picat.c term.h basic.h bapi.h frame.h
 	$(CCC) $(CFLAGS) kissat_picat.c
 black_white.o : espresso/black_white.c
@@ -123,7 +125,7 @@ black_white.o : espresso/black_white.c
 canonical.o : espresso/canonical.c
 	$(CCC) $(ESPRESSO_FLAGS) -o canonical.o espresso/canonical.c
 cofactor.o : espresso/cofactor.c
-	$(CCC) $(ESPRESSO_FLAGS) -o cofactor.o espresso/cofactor.c
+	$(CCC) -std=gnu17 $(ESPRESSO_FLAGS) -o cofactor.o espresso/cofactor.c
 cols.o : espresso/cols.c
 	$(CCC) $(ESPRESSO_FLAGS) -o cols.o espresso/cols.c
 compl.o : espresso/compl.c
@@ -149,7 +151,7 @@ equiv.o : espresso/equiv.c
 espresso.o : espresso/espresso.c
 	$(CCC) $(ESPRESSO_FLAGS) -o espresso.o espresso/espresso.c
 espresso_expand.o : espresso/espresso_expand.c
-	$(CCC) $(ESPRESSO_FLAGS) -o espresso_expand.o espresso/espresso_expand.c
+	$(CCC) -std=gnu17 $(ESPRESSO_FLAGS) -o espresso_expand.o espresso/espresso_expand.c
 espresso_main.o : espresso/espresso_main.c
 	$(CCC) $(ESPRESSO_FLAGS) -o espresso_main.o espresso/espresso_main.c
 essen.o : espresso/essen.c
@@ -199,7 +201,7 @@ sharp.o : espresso/sharp.c
 sigma.o : espresso/sigma.c
 	$(CCC) $(ESPRESSO_FLAGS) -o sigma.o espresso/sigma.c
 signature.o : espresso/signature.c
-	$(CCC) $(ESPRESSO_FLAGS) -o signature.o espresso/signature.c
+	$(CCC) -std=gnu17 $(ESPRESSO_FLAGS) -o signature.o espresso/signature.c
 signature_exact.o : espresso/signature_exact.c
 	$(CCC) $(ESPRESSO_FLAGS) -o signature_exact.o espresso/signature_exact.c
 sminterf.o : espresso/sminterf.c
