From bec760eddf7f0e3a7959a7278b275287878ac0ee Mon Sep 17 00:00:00 2001
From: Rob Peters <info@domoticz.com>
Date: Fri, 10 May 2019 07:41:06 +0200
Subject: [PATCH] Better way to check for boost version number (works also with
 boost 1.70)

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 28e41a2b3..c0225e897 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_minimum_required(VERSION 2.8.4)
 project(domoticz)
 
 ## required min. libBoost version
-SET(DOMO_MIN_LIBBOOST_VERSION 106000)
+SET(DOMO_MIN_LIBBOOST_VERSION 1.60.0)
 ##
 
 MACRO(Gitversion_GET_REVISION dir variable)
