diff --git a/tests/compat/test_partial_clone.py b/tests/compat/test_partial_clone.py
index e263abb..35e260e 100644
--- a/tests/compat/test_partial_clone.py
+++ b/tests/compat/test_partial_clone.py
@@ -335,6 +335,7 @@ class PartialCloneServerTestCase(CompatTestCase):
 
 
 @skipIf(sys.platform == "win32", "Broken on windows, with very long fail time.")
+@skipIf(True, "git daemon tests unreliable")
 class PartialCloneClientTestCase(CompatTestCase):
     """Tests for partial clone client compatibility with git server."""
 
