diff --git a/ssl/Makefile.am b/ssl/Makefile.am
index 99d138a..9d8f1f4 100644
--- a/ssl/Makefile.am
+++ b/ssl/Makefile.am
@@ -7,7 +7,7 @@ EXTRA_DIST += CMakeLists.txt
 EXTRA_DIST += ssl.sym
 
 libssl_la_LDFLAGS = -version-info @LIBSSL_VERSION@ -no-undefined -export-symbols $(top_srcdir)/ssl/ssl.sym
-libssl_la_LIBADD = $(abs_top_builddir)/crypto/libcrypto.la
+libssl_la_LIBADD = $(abs_top_builddir)/crypto/libcrypto.la $(abs_top_builddir)/crypto/libcompat.la
 
 libssl_la_SOURCES = bio_ssl.c
 libssl_la_SOURCES += bs_ber.c
