From: <hesso@pool.math.tu-berlin.de>
Subject: No description.

diff -Naur nvi-1.81.6.orig/vi/v_sentence.c nvi-1.81.6/vi/v_sentence.c
--- nvi-1.81.6.orig/vi/v_sentence.c	2007-11-18 17:41:42.000000000 +0100
+++ nvi-1.81.6/vi/v_sentence.c	2008-05-01 18:15:37.000000000 +0200
@@ -291,7 +291,7 @@
 			 * we can end up where we started.  Fix it.
 			 */
 			if (vp->m_start.lno != cs.cs_lno ||
-			    vp->m_start.cno != cs.cs_cno)
+			    vp->m_start.cno > cs.cs_cno)
 				goto okret;
 
 			/*
