--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -1,4 +1,4 @@
-add_subdirectory(gtest)
+find_package(GTest REQUIRED)
 add_subdirectory(libtransmission)
 if(ENABLE_UTILS)
     add_subdirectory(utils)
--- a/tests/libtransmission/CMakeLists.txt
+++ b/tests/libtransmission/CMakeLists.txt
@@ -73,7 +73,8 @@ target_compile_options(libtransmission-test
 target_link_libraries(libtransmission-test
     PRIVATE
         ${TR_NAME}
-        gtestall
+        GTest::gtest
+        GTest::gtest_main
         dht::dht
         fmt::fmt-header-only
         libevent::event
