diff -ruN a/config.php.dist b/config.php.dist
--- a/config.php.dist	2024-03-06 10:53:34.113527229 +0100
+++ b/config.php.dist	2024-03-06 10:55:15.483483870 +0100
@@ -49,7 +49,7 @@
 
 	// Defines the temp path (absolute). Here uploaded attachments will be saved.
 	// The web client doesn't work without this directory.
-	define("TMP_PATH", "/var/lib/grommunio-web/tmp");
+	define("TMP_PATH", "/var/tmp/grommunio-web");
 
 	// Define the path to the plugin directory (No slash at the end)
 	define("PATH_PLUGIN_DIR", "plugins");
@@ -67,7 +67,7 @@
 	define("APPDATA_DIR", "appdata");
 
 	// Path used for full text index of every user
-	define("SQLITE_INDEX_PATH", "/var/lib/grommunio-web/sqlite-index");
+	define("SQLITE_INDEX_PATH", "/var/cache/grommunio-web/sqlite-index");
 
 	// General grommunio Web theme. This will be loaded by default for every user
 	// (if the theme is installed as a plugin)
