--- a/src/base/net/stratum/DaemonClient.cpp	2021-08-09 04:09:15.000000000 -0500
+++ b/src/base/net/stratum/DaemonClient.cpp.new	2021-08-09 18:25:14.000000000 -0500
@@ -65,7 +65,7 @@
 
 static constexpr size_t kBlobReserveSize    = 8;
 
-static const char kZMQGreeting[64] = { -1, 0, 0, 0, 0, 0, 0, 0, 0, 127, 3, 0, 'N', 'U', 'L', 'L' };
+static const char kZMQGreeting[64] = { static_cast<char>(-1), 0, 0, 0, 0, 0, 0, 0, 0, 127, 3, 0, 'N', 'U', 'L', 'L' };
 static constexpr size_t kZMQGreetingSize1 = 11;
 
 static const char kZMQHandshake[] = "\4\x19\5READY\xbSocket-Type\0\0\0\3SUB";
