diff -rupN a/internal/lsp/server_test.go b/internal/lsp/server_test.go
--- a/internal/lsp/server_test.go	2025-09-10 13:28:32.000000000 +0000
+++ b/internal/lsp/server_test.go	2026-02-08 04:18:20.300000000 +0000
@@ -27,7 +27,7 @@ const (
 	// defaultTimeout is set based on the investigation done as part of
 	// https://github.com/open-policy-agent/regal/issues/931. 20 seconds is 10x the
 	// maximum time observed for an operation to complete.
-	defaultTimeout             = 20 * time.Second
+	defaultTimeout             = 600 * time.Second
 	defaultBufferedChannelSize = 5
 )
 
