diff -rupN a/tests/test_cli.py b/tests/test_cli.py
--- a/tests/test_cli.py	2023-09-14 12:13:18.000000000 +0000
+++ b/tests/test_cli.py	2026-03-24 00:39:04.000000000 +0000
@@ -17,14 +17,14 @@ def runner():
 
 def test_main_succeeds(runner):
     result = runner.invoke(cli)
-    got = result.exit_code
-    want = 0
-    assert got == want
+    # Help shown via no_args_is_help results in exit code 2 in click >=8.2.0
+    # https://github.com/pallets/click/pull/1489
+    assert result.exit_code == 2
 
 
 # check verbose mode
 def test_cli_print_help(runner):
-    result = runner.invoke(cli)
+    result = runner.invoke(cli, ["--help"])
     assert result.exit_code == 0
     assert "kallisto" in result.output
 
