--- a/solid/solid/backends/udisks2/udisksblock.h
+++ b/solid/solid/backends/udisks2/udisksblock.h
@@ -21,6 +21,7 @@
 #ifndef UDISKS2BLOCK_H
 #define UDISKS2BLOCK_H
 
+#include <sys/types.h>
 #include <solid/ifaces/block.h>
 #include "udisksdeviceinterface.h"
 
--- a/kdecore/io/kdirwatch_p.h
+++ b/kdecore/io/kdirwatch_p.h
@@ -64,7 +64,8 @@
 #endif
 
 #include <sys/time.h>
-#include <sys/param.h> // ino_t
+#include <sys/param.h> // ino_t , but not there !!
+#include <sys/types.h> // maybe, here ?
 #include <ctime>
 
 
