--- a/accountd/filter.c
+++ b/accountd/filter.c
@@ -266,7 +266,6 @@ filter_print_vacation(struct config *con
 
     fprintf(file, "# MSshell :: vacation\n");
     fprintf(file, "if personal\n");
-    fprintf(file, "   alias %s@cam.ac.uk\n", pwd->pw_name);
 
     if (!filter_print_aliases(config, file))
         return (NIL);
@@ -278,7 +277,7 @@ filter_print_vacation(struct config *con
     fprintf(file, ("This message is automatically generated "
                    "in response to your mail\\n\\" "\n"));
     fprintf(file, ("message (perhaps re-directed) to "
-                   "$local_part@hermes.cam.ac.uk.\\n\\n\"\n"));
+                   "$local_part@$local_domain.\\n\\n\"\n"));
     fprintf(file, "    file ${home}/vacation.message\n");
     fprintf(file, "    log  ${home}/vacation.log\n");
     fprintf(file, "    once ${home}/vacation.once\n");
@@ -307,10 +306,10 @@ filter_print_spam(struct config *config,
     threshold = atoi(filter->threshold);
 
     /* XXX (threshold == 0) okay? */
-    fprintf(file, "if $h_X-Cam-SpamScore contains \"");
+    fprintf(file, "if $h_X-Spam-Level contains \"");
 
     for (i=0 ; i < threshold; i++)
-        fputc('s', file);
+        fputc('*', file);
 
     fprintf(file, "\" then\n");
     fprintf(file, "    save mail/spam\n");
