diff --git a/common/Build.PL b/common/Build.PL
index ff0f4c0..7755044 100644
--- a/common/Build.PL
+++ b/common/Build.PL
@@ -1,3 +1,4 @@
+BEGIN { push @INC, '.'; }
 use Module::Build;
 
 my $build = Module::Build->new(
diff --git a/master/Build.PL b/master/Build.PL
index 6dc31ea..e7de302 100644
--- a/master/Build.PL
+++ b/master/Build.PL
@@ -1,3 +1,4 @@
+BEGIN { push @INC, '.'; }
 use MasterBuilder;
 
 use warnings;
diff --git a/node/Build.PL b/node/Build.PL
index c0e6420..def4774 100644
--- a/node/Build.PL
+++ b/node/Build.PL
@@ -1,3 +1,4 @@
+BEGIN { push @INC, '.'; }
 use NodeBuilder;
 
 use warnings;
diff --git a/plugins/Build.PL b/plugins/Build.PL
index cee0a97..8f5942f 100644
--- a/plugins/Build.PL
+++ b/plugins/Build.PL
@@ -1,3 +1,4 @@
+BEGIN { push @INC, '.'; }
 use PluginsBuilder;
 
 use warnings;
