diff -up xf86-video-cirrus-1.2.0/src/alp_driver.c.jx xf86-video-cirrus-1.2.0/src/alp_driver.c
--- xf86-video-cirrus-1.2.0/src/alp_driver.c.jx	2008-03-19 10:29:23.000000000 -0400
+++ xf86-video-cirrus-1.2.0/src/alp_driver.c	2009-02-27 10:59:27.000000000 -0500
@@ -812,6 +812,20 @@ AlpPreInit(ScrnInfoPtr pScrn, int flags)
      else
  	xf86SetDDCproperties(pScrn,xf86PrintEDID(
  	    xf86DoEDID_DDC2(pScrn->scrnIndex,pCir->I2CPtr1)));
+#ifdef XSERVER_LIBPCIACCESS
+     if (!pScrn->monitor->DDC &&
+	 ((pCir->PciInfo->subvendor_id & 0xffff) == 0x1af4)) {
+	xf86DrvMsg(pScrn->scrnIndex, X_INFO,
+		   "Defaulting to 1024x768 for QEMU\n");
+	pScrn->monitor->nHsync = 1;
+	pScrn->monitor->hsync[0].lo = 31.5;
+	pScrn->monitor->hsync[0].hi = 48.0;
+	pScrn->monitor->nVrefresh = 1;
+	pScrn->monitor->vrefresh[0].lo = 56.0;
+	pScrn->monitor->vrefresh[0].hi = 60.1;
+	pScrn->monitor->maxPixClock = 65000;
+     }
+#endif
  
      /* Probe the possible LCD display */
      AlpProbeLCD(pScrn);

