--- a/Cargo.toml
+++ b/Cargo.toml
@@ -40,11 +40,8 @@
 signal-hook = "0.3"
 tiny_http = { version = "0.9", optional = true }
 
-[dependencies.electrs-rocksdb]
-# Workaround the following issues:
-# - https://github.com/romanz/electrs/issues/403 (support building on ARM 32-bit)
-# - https://github.com/romanz/electrs/issues/469 (dynamic linking on Debian 11)
-version = "0.15.0-e2"
+[dependencies.rocksdb]
+version = "0.18.0"
 default-features = false
 # ZSTD is used for data compression
 # Snappy is only for checking old DB
