diff --git a/krusader/DiskUsage/filelightParts/fileTree.h b/krusader/DiskUsage/filelightParts/fileTree.h
index 80b40082..ac0ab302 100644
--- a/krusader/DiskUsage/filelightParts/fileTree.h
+++ b/krusader/DiskUsage/filelightParts/fileTree.h
@@ -28,6 +28,7 @@
 #include <time.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 // TODO these are pointlessly general purpose now, make them incredibly specific
 
diff --git a/krusader/FileSystem/krpermhandler.h b/krusader/FileSystem/krpermhandler.h
index 427171ab..316b1a8d 100644
--- a/krusader/FileSystem/krpermhandler.h
+++ b/krusader/FileSystem/krpermhandler.h
@@ -33,6 +33,7 @@
 #include <time.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 #define NO_PERM      0
 #define UNKNOWN_PERM 1
