diff -Nurp a/sdk/include/opentelemetry/sdk/metrics/aggregation/aggregation_config.h b/sdk/include/opentelemetry/sdk/metrics/aggregation/aggregation_config.h
--- a/sdk/include/opentelemetry/sdk/metrics/aggregation/aggregation_config.h	2025-07-11 22:11:41.000000000 +0000
+++ b/sdk/include/opentelemetry/sdk/metrics/aggregation/aggregation_config.h	2025-09-01 05:22:03.079342620 +0000
@@ -6,6 +6,7 @@
 #include <vector>
 
 #include "opentelemetry/version.h"
+#include <cstdint>
 
 OPENTELEMETRY_BEGIN_NAMESPACE
 namespace sdk
