diff --git a/src/keyfile.c b/src/keyfile.c
index f8d729e5c..ad3628a5f 100644
--- a/src/keyfile.c
+++ b/src/keyfile.c
@@ -91,7 +91,7 @@
 # define GEANY_DEFAULT_FONT_MSG_WINDOW	"Menlo Medium 12"
 # define GEANY_DEFAULT_FONT_EDITOR		"Menlo Medium 12"
 #else
-# define GEANY_DEFAULT_TOOLS_TERMINAL	"xterm -e \"/bin/sh %c\""
+# define GEANY_DEFAULT_TOOLS_TERMINAL	"Terminal -e \"/bin/sh %c\""
 # define GEANY_DEFAULT_USE_NATIVE_DLGS	FALSE
 # define GEANY_DEFAULT_FONT_SYMBOL_LIST	"Sans 9"
 # define GEANY_DEFAULT_FONT_MSG_WINDOW	"Monospace 9"
