diff --git a/tests/test_encodings.c b/tests/test_encodings.c
index a5bdf62be..7030b8d50 100644
--- a/tests/test_encodings.c
+++ b/tests/test_encodings.c
@@ -221,16 +221,9 @@ static void test_encodings_convert_utf_other_to_utf8_auto(void)
 	TEST(FALSE, UTF32_BE_BOM"\000\000\000W\000\000\000i\000\000\000t\000\000\000h\000\000\000 \000\000\000B\000\000\000O\000\000\000M\000\000\000\000\000\000\000a\000\000\000n\000\000\000d\000\000\000 \000\000\000N\000\000\000U\000\000\000L\000\000\000s", "With BOM\0and NULs", TRUE, NULL);
 	TEST(FALSE, UTF32_BE_BOM"\000\000\000W\000\000\000\355\000\000\000t\000\000\000h\000\000\000 \000\000\000B\000\000\000\330\000\000\000M\000\000\000\000\000\000\000a\000\000\000\361\000\000\000d\000\000\000 \000\000\000N\000\000\000\331\000\000\000L\000\000\000s", "Wíth BØM\0añd NÙLs", TRUE, NULL);
 
-	/* meh, UTF-7 */
-	TEST(TRUE, "No B+ANg-M", "No BØM", FALSE, "UTF-7");
-	TEST(TRUE, "+/v8-With B+ANg-M", "With BØM", TRUE, NULL);
-	TEST(FALSE, "No B+ANg-M+AAA-but NULs", "No BØM\0but NULs", FALSE, "UTF-7");
-	/* Fails to load as UTF-7 because of the NUL, and succeeds as UTF-8 but
-	 * obviously doesn't match expectations */
-	/*TEST(FALSE, "+/v8-With B+ANg-M+AAA-and NULs", "With BØM\0and NULs", TRUE, NULL);*/
+	/* UTF-7 tests removed - not supported by musl */
 
 	/* empty data with BOMs */
-	TEST_ENC(TRUE, "+/v8-", "", TRUE, NULL, "UTF-7"); /* UTF-7 */
 	TEST_ENC(TRUE, UTF16_BE_BOM, "", TRUE, NULL, "UTF-16BE");
 	TEST_ENC(TRUE, UTF16_LE_BOM, "", TRUE, NULL, "UTF-16LE");
 	TEST_ENC(TRUE, UTF32_BE_BOM, "", TRUE, NULL, "UTF-32BE");
