From 193dbf82809b5ed5db8f8c37c315592d529ee37a Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Mon, 27 Jan 2014 13:26:44 +0000
Subject: [PATCH] change public key location

---
 mtproto-client.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mtproto-client.c b/mtproto-client.c
index 06ddfdc..4279512 100644
--- a/mtproto-client.c
+++ b/mtproto-client.c
@@ -129,7 +129,7 @@ int Response_len;
  *
  */
 
-char *rsa_public_key_name = "tg.pub";
+char *rsa_public_key_name = "/usr/share/telegram/tg.pub";
 RSA *pubKey;
 long long pk_fingerprint;
 
-- 
1.8.5.3

