From 33d358d83dd882d1e4eebd34dc4e6cf622f0c854 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Mon, 27 Jan 2014 11:42:33 +0000
Subject: [PATCH] uclibc does not provide aio.h

---
 mtproto-client.c | 2 ++
 mtproto-common.c | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/mtproto-client.c b/mtproto-client.c
index 5f1a997..06ddfdc 100644
--- a/mtproto-client.c
+++ b/mtproto-client.c
@@ -35,7 +35,9 @@
 #include <sys/endian.h>
 #endif
 #include <sys/types.h>
+#ifndef __UCLIBC__
 #include <aio.h>
+#endif
 #include <netdb.h>
 #include <openssl/rand.h>
 #include <openssl/rsa.h>
diff --git a/mtproto-common.c b/mtproto-common.c
index ceeb5c5..1c34550 100644
--- a/mtproto-common.c
+++ b/mtproto-common.c
@@ -31,7 +31,9 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <sys/types.h>
+#ifndef __UCLIBC__
 #include <aio.h>
+#endif
 #include <netdb.h>
 #include <openssl/bn.h>
 #include <openssl/rand.h>
-- 
1.8.5.3

