--- a/generate.c.orig
+++ b/generate.c
@@ -25,7 +25,6 @@
 
 #include <stdio.h>
 #include <signal.h>
-#include <execinfo.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <sys/types.h>
--- a/tlc.c.orig
+++ b/tlc.c
@@ -35,7 +35,6 @@
 
 #include <signal.h>
 #include "config.h"
-#include <execinfo.h>
 #include <stdarg.h>
 
 int verbosity;
--- a/net.c.orig
+++ b/net.c
@@ -29,7 +29,7 @@
 #include <netdb.h>
 #include <netinet/in.h>
 #include <netinet/tcp.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <errno.h>
