--- ./error.h.orig	2014-04-02 14:09:23.163839471 +0000
+++ ./error.h	2014-04-02 14:14:07.630616871 +0000
@@ -22,6 +22,10 @@
 
 #include <stdarg.h>
 
+#if !defined(__ASSERT_FUNCTION)
+#define __ASSERT_FUNCTION __func__
+#endif
+
 #ifdef __cplusplus
 extern "C"
 {
--- ./io.c.orig	2014-04-02 14:14:29.104210187 +0000
+++ ./io.c	2014-04-02 14:14:59.264575341 +0000
@@ -31,6 +31,8 @@
 #include <termios.h>
 #include <unistd.h>
 
+#include <sys/time.h>
+
 /* Application header files */
 #include "io.h"
 
