diff --git a/ApfsLib/DeviceLinux.cpp b/ApfsLib/DeviceLinux.cpp
index 1ddd47d..80936b4 100644
--- a/ApfsLib/DeviceLinux.cpp
+++ b/ApfsLib/DeviceLinux.cpp
@@ -53,9 +53,9 @@ bool DeviceLinux::Open(const char* name)
 		return false;
 	}
 
-	struct stat64 st;
+	struct stat st;
 
-	fstat64(m_device, &st);
+	fstat(m_device, &st);
 
 	if (S_ISREG(st.st_mode))
 	{
@@ -85,7 +85,7 @@ bool DeviceLinux::Read(void* data, uint64_t offs, uint64_t len)
 {
 	size_t nread;
 
-	nread = pread64(m_device, data, len, offs);
+	nread = pread(m_device, data, len, offs);
 
 	// TODO: Better error handling ...
 	return nread == len;
