diff --git a/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp b/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
index bce9d38..0bab381 100644
--- a/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
+++ b/scipy/optimize/rectangular_lsap/rectangular_lsap.cpp
@@ -44,6 +44,8 @@ Author: PM Larsen
 #include <cmath>
 #include <vector>
 
+#include <stdint.h>
+
 static int
 augmenting_path(int nc, std::vector<double>& cost, std::vector<double>& u,
                 std::vector<double>& v, std::vector<int>& path,
