diff --git a/xdg.gemspec b/xdg.gemspec
index 389a676..2c2906e 100644
--- a/xdg.gemspec
+++ b/xdg.gemspec
@@ -19,9 +19,6 @@ Gem::Specification.new do |spec|
     "source_code_uri" => "https://github.com/bkuhlmann/xdg"
   }
 
-  spec.signing_key = Gem.default_key_path
-  spec.cert_chain = [Gem.default_cert_path]
-
   spec.required_ruby_version = ">= 3.3", "<= 3.4"
 
   spec.files = Dir["*.gemspec", "lib/**/*"]
