diff --git a/xdg.gemspec b/xdg.gemspec
index e5ad564..d49cb31 100644
--- a/xdg.gemspec
+++ b/xdg.gemspec
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
     "source_code_uri" => "https://github.com/bkuhlmann/xdg"
   }
 
-  spec.required_ruby_version = "~> 3.2"
+  spec.required_ruby_version = "~> 3"
 
   spec.files = Dir["*.gemspec", "lib/**/*"]
   spec.extra_rdoc_files = Dir["README*", "LICENSE*"]
