--- a/QCEDL.CLI/QCEDL.CLI.csproj
+++ b/QCEDL.CLI/QCEDL.CLI.csproj
@@ -11,21 +11,13 @@
         <ImplicitUsings>enable</ImplicitUsings>
         <Nullable>enable</Nullable>
         <RuntimeIdentifiers>win-x64;win-arm64;linux-x64;linux-arm64;osx-x64;osx-arm64</RuntimeIdentifiers>
-        <DebugType>None</DebugType>
-        <DebugSymbols>false</DebugSymbols>
         <AssemblyName>edl-ng</AssemblyName>
         <Product>edl-ng</Product>
         <IncludeSourceRevisionInInformationalVersion>false</IncludeSourceRevisionInInformationalVersion>
         <InvariantGlobalization>true</InvariantGlobalization>
         <IlcGenerateMstatFile>true</IlcGenerateMstatFile>
         <IlcGenerateDgmlFile>true</IlcGenerateDgmlFile>
-        <OptimizationPreference>Size</OptimizationPreference>
-        <StripSymbols>true</StripSymbols>
-        <PublishTrimmed>true</PublishTrimmed>
-        <PublishAot>true</PublishAot>
         <PublishSingleFile>true</PublishSingleFile>
-        <SelfContained>true</SelfContained>
-        <TrimmerRemoveSymbols>true</TrimmerRemoveSymbols>
         <HttpActivityPropagationSupport>false</HttpActivityPropagationSupport>
         <XmlResolverIsNetworkingEnabledByDefault>false</XmlResolverIsNetworkingEnabledByDefault>
         <OutputType>Exe</OutputType>
