--- a/QCEDL.Analyzer/QCEDL.Analyzer.csproj
+++ b/QCEDL.Analyzer/QCEDL.Analyzer.csproj
@@ -15,6 +15,6 @@
             <PrivateAssets>all</PrivateAssets>
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.14.0"/>
+        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.12.0"/>
     </ItemGroup>
 </Project>
