*** a/downloader.c	2026-06-18 11:57:20.232179189 -0500
--- b/downloader.c	2026-06-18 11:59:00.264777995 -0500
***************
*** 587,593 ****
  		goto out;
  	}
  
! 	assert(p->fd == -1 || p->obufsz < 0 || sess->opts->dry_run);
  	assert(tokres == TOKEN_EOF);
  
  	/*
--- 587,593 ----
  		goto out;
  	}
  
!   assert(p->fd < 0 || p->obufsz == 0 || sess->opts->dry_run);
  	assert(tokres == TOKEN_EOF);
  
  	/*
