See: https://foss.heptapod.net/python-libs/passlib/-/issues/190

--- a/passlib/handlers/bcrypt.py
+++ b/passlib/handlers/bcrypt.py
@@ -617,7 +617,7 @@
         except ImportError: # pragma: no cover
             return False
         try:
-            version = _bcrypt.__about__.__version__
+            version = _bcrypt.__version__
         except:
             log.warning("(trapped) error reading bcrypt version", exc_info=True)
             version = '<unknown>'
