commit 65bf418b28b7bc65982b4ef59dd528d4ffbc5d1b
Author: Zach DeCook <zachdecook@librem.one>
Date:   Sun Jun 7 21:18:30 2026 -0400

    CMake: All installing python library as well

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index fe08b3e2c..85d23a33b 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -269,6 +269,16 @@ elseif (BUILD_AS_PYTHON)
 
     add_library(verovio STATIC ${all_SRC})
 
+    string(REGEX REPLACE "^([0-9]+)\\.([0-9]+).*" "\\1.\\2" PYTHON_VERSION_MAJMIN "${Python_VERSION}")
+    install(
+        TARGETS verovio_module_${OUTPUT_VERSION_STRING}
+        DESTINATION lib/python${PYTHON_VERSION_MAJMIN}/site-packages/verovio
+    )
+    install(
+        FILES ${CMAKE_BINARY_DIR}/verovio.py ../bindings/python/__init__.py
+        DESTINATION lib/python${PYTHON_VERSION_MAJMIN}/site-packages/verovio
+    )
+
 #####################
 # Command-line tool #
 #####################
