--- a/config.nims
+++ b/config.nims
@@ -90,8 +90,7 @@ proc configWinSSL() =
 
 
 proc configUnixSSL() =
-    --dynlibOverride:ssl
-    --dynlibOverride:crypto
+    --define:"useOpenssl3"
 
 
 proc main() =
--- a/src/library/Net.nim
+++ b/src/library/Net.nim
@@ -24,7 +24,7 @@ when defined(ssl):
     when defined(windows): 
         {.passL: "-Bstatic -Lsrc/extras/openssl/deps/windows -lssl -lcrypto -lws2_32 -Bdynamic".}
     elif defined(linux):
-        {.passL: "-Bstatic -Lsrc/extras/openssl/deps/linux -lssl -lcrypto -Bdynamic".}
+        {.passL: "-lssl -lcrypto".}
     elif defined(macosx):
         # TODO(Net) Use OpenSSL 1.1 in all cases
         #  right now, every OS/architecture statically links OpenSSL 1.1.
