diff --git a/Cargo.lock b/Cargo.lock
index 9e47fc4..615a760 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1509,7 +1509,6 @@ version = "0.28.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "0c10584274047cb335c23d3e61bcef8e323adae7c5c8c760540f73610177fc3f"
 dependencies = [
- "cc",
  "pkg-config",
  "vcpkg",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 9452b3d..1a7ca78 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -111,7 +111,7 @@ tracing-subscriber = { version = "0.3.17", features = ["env-filter"] }
 
 lru-cache = "0.1.2"
 parking_lot = { version = "0.12.1", optional = true }
-rusqlite = { version = "0.31", optional = true, features = ["bundled"] }
+rusqlite = { version = "0.31", optional = true }
 
 # crossbeam = { version = "0.8.2", optional = true }
 num_cpus = "1.15.0"
