diff --git a/gnulib/tests/Makefile.am b/gnulib/tests/Makefile.am
index e1d3c9c..f8b3797 100644
--- a/gnulib/tests/Makefile.am
+++ b/gnulib/tests/Makefile.am
@@ -544,15 +544,6 @@ EXTRA_libtests_a_SOURCES += getopt.c getopt1.c
 
 ## end   gnulib module getopt-posix
 
-## begin gnulib module getopt-posix-tests
-
-TESTS += test-getopt-posix
-check_PROGRAMS += test-getopt-posix
-test_getopt_posix_LDADD = $(LDADD) $(LIBINTL)
-EXTRA_DIST += macros.h signature.h test-getopt-posix.c test-getopt-main.h test-getopt.h
-
-## end   gnulib module getopt-posix-tests
-
 ## begin gnulib module getpagesize
 
 
diff --git a/gnulib/tests/Makefile.in b/gnulib/tests/Makefile.in
index 19ff066..a319bb7 100644
--- a/gnulib/tests/Makefile.in
+++ b/gnulib/tests/Makefile.in
@@ -123,7 +123,7 @@ TESTS = test-accept$(EXEEXT) test-alloca-opt$(EXEEXT) \
 	test-fread$(EXEEXT) test-free$(EXEEXT) test-fstat$(EXEEXT) \
 	test-ftruncate.sh test-fwrite$(EXEEXT) \
 	test-getcwd-lgpl$(EXEEXT) test-getdtablesize$(EXEEXT) \
-	test-getopt-gnu$(EXEEXT) test-getopt-posix$(EXEEXT) \
+	test-getopt-gnu$(EXEEXT) \
 	test-getprogname$(EXEEXT) test-gettimeofday$(EXEEXT) \
 	test-iconv-h$(EXEEXT) test-iconv$(EXEEXT) \
 	test-ignore-value$(EXEEXT) test-inet_pton$(EXEEXT) \
@@ -180,7 +180,7 @@ check_PROGRAMS = test-accept$(EXEEXT) test-alloca-opt$(EXEEXT) \
 	test-fread$(EXEEXT) test-free$(EXEEXT) test-fstat$(EXEEXT) \
 	test-ftruncate$(EXEEXT) test-fwrite$(EXEEXT) \
 	test-getcwd-lgpl$(EXEEXT) test-getdtablesize$(EXEEXT) \
-	test-getopt-gnu$(EXEEXT) test-getopt-posix$(EXEEXT) \
+	test-getopt-gnu$(EXEEXT) \
 	test-getprogname$(EXEEXT) test-gettimeofday$(EXEEXT) \
 	test-iconv-h$(EXEEXT) test-iconv$(EXEEXT) \
 	test-ignore-value$(EXEEXT) test-inet_pton$(EXEEXT) \
@@ -525,10 +525,6 @@ test_getopt_gnu_SOURCES = test-getopt-gnu.c
 test_getopt_gnu_OBJECTS = test-getopt-gnu.$(OBJEXT)
 test_getopt_gnu_DEPENDENCIES = $(am__DEPENDENCIES_2) \
 	$(am__DEPENDENCIES_1)
-test_getopt_posix_SOURCES = test-getopt-posix.c
-test_getopt_posix_OBJECTS = test-getopt-posix.$(OBJEXT)
-test_getopt_posix_DEPENDENCIES = $(am__DEPENDENCIES_2) \
-	$(am__DEPENDENCIES_1)
 test_getprogname_SOURCES = test-getprogname.c
 test_getprogname_OBJECTS = test-getprogname.$(OBJEXT)
 test_getprogname_DEPENDENCIES = $(am__DEPENDENCIES_2)
@@ -1046,7 +1042,6 @@ am__depfiles_remade = ./$(DEPDIR)/accept.Po ./$(DEPDIR)/alloca.Po \
 	./$(DEPDIR)/test-fwrite.Po ./$(DEPDIR)/test-getcwd-lgpl.Po \
 	./$(DEPDIR)/test-getdtablesize.Po \
 	./$(DEPDIR)/test-getopt-gnu.Po \
-	./$(DEPDIR)/test-getopt-posix.Po \
 	./$(DEPDIR)/test-getprogname.Po \
 	./$(DEPDIR)/test-gettimeofday.Po ./$(DEPDIR)/test-iconv-h.Po \
 	./$(DEPDIR)/test-iconv.Po ./$(DEPDIR)/test-ignore-value.Po \
@@ -3142,10 +3137,6 @@ test-getopt-gnu$(EXEEXT): $(test_getopt_gnu_OBJECTS) $(test_getopt_gnu_DEPENDENC
 	@rm -f test-getopt-gnu$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_getopt_gnu_OBJECTS) $(test_getopt_gnu_LDADD) $(LIBS)
 
-test-getopt-posix$(EXEEXT): $(test_getopt_posix_OBJECTS) $(test_getopt_posix_DEPENDENCIES) $(EXTRA_test_getopt_posix_DEPENDENCIES) 
-	@rm -f test-getopt-posix$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_getopt_posix_OBJECTS) $(test_getopt_posix_LDADD) $(LIBS)
-
 test-getprogname$(EXEEXT): $(test_getprogname_OBJECTS) $(test_getprogname_DEPENDENCIES) $(EXTRA_test_getprogname_DEPENDENCIES) 
 	@rm -f test-getprogname$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_getprogname_OBJECTS) $(test_getprogname_LDADD) $(LIBS)
@@ -3612,7 +3603,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-getcwd-lgpl.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-getdtablesize.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-getopt-gnu.Po@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-getopt-posix.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-getprogname.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-gettimeofday.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-iconv-h.Po@am__quote@ # am--include-marker
@@ -4192,13 +4182,6 @@ test-getopt-gnu.log: test-getopt-gnu$(EXEEXT)
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-getopt-posix.log: test-getopt-posix$(EXEEXT)
-	@p='test-getopt-posix$(EXEEXT)'; \
-	b='test-getopt-posix'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 test-getprogname.log: test-getprogname$(EXEEXT)
 	@p='test-getprogname$(EXEEXT)'; \
 	b='test-getprogname'; \
@@ -5064,7 +5047,6 @@ distclean: distclean-recursive
 	-rm -f ./$(DEPDIR)/test-getcwd-lgpl.Po
 	-rm -f ./$(DEPDIR)/test-getdtablesize.Po
 	-rm -f ./$(DEPDIR)/test-getopt-gnu.Po
-	-rm -f ./$(DEPDIR)/test-getopt-posix.Po
 	-rm -f ./$(DEPDIR)/test-getprogname.Po
 	-rm -f ./$(DEPDIR)/test-gettimeofday.Po
 	-rm -f ./$(DEPDIR)/test-iconv-h.Po
@@ -5299,7 +5281,6 @@ maintainer-clean: maintainer-clean-recursive
 	-rm -f ./$(DEPDIR)/test-getcwd-lgpl.Po
 	-rm -f ./$(DEPDIR)/test-getdtablesize.Po
 	-rm -f ./$(DEPDIR)/test-getopt-gnu.Po
-	-rm -f ./$(DEPDIR)/test-getopt-posix.Po
 	-rm -f ./$(DEPDIR)/test-getprogname.Po
 	-rm -f ./$(DEPDIR)/test-gettimeofday.Po
 	-rm -f ./$(DEPDIR)/test-iconv-h.Po
