Ref: https://github.com/pallets/quart/issues/461
---
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -23,7 +23,7 @@
             (b"Referer", b"hypercorn"),
         ],
         "client": ("127.0.0.1", 80),
-        "server": None,
+        "server": ("127.0.0.1", 80),
         "state": {},  # type: ignore[typeddict-item]
         "extensions": {},
     }
@@ -46,7 +46,7 @@
             (b"Referer", b"hypercorn"),
         ],
         "client": ("127.0.0.1", 80),
-        "server": None,
+        "server": ("127.0.0.1", 80),
         "subprotocols": [],
         "state": {},  # type: ignore[typeddict-item]
         "extensions": {},
