diff --git a/lib/sigevent.c b/lib/sigevent.c
index c80a729..1221c25 100644
--- a/lib/sigevent.c
+++ b/lib/sigevent.c
@@ -244,6 +244,7 @@ static void
 trap_default_signals(void)
 {
   static const int core_signals[] = {
+#if 0
     SIGQUIT,
     SIGILL,
 #ifdef SIGEMT
@@ -261,6 +262,7 @@ trap_default_signals(void)
 #ifdef SIGXFSZ
     SIGXFSZ,
 #endif
+#endif
   };
   static const int exit_signals[] = {
     SIGHUP,
