--- a/java/settings.gradle
+++ b/java/settings.gradle
@@ -9,7 +9,7 @@
 include ':client', ':server'
 
 if (JavaVersion.current().isJava11Compatible()) {
-    include ':android', ':android:benchmarks'
+//    include ':android', ':android:benchmarks'
 } else if (gradle.startParameter.getTaskNames().any { name -> name.contains('android:') }) {
     throw new GradleException("building for Android requires JDK 11 or newer")
 } else if (gradle.startParameter.getTaskNames().any { name -> name.lastIndexOf(':') <= 0 }) {
