diff --git a/zotero/node_modules/mocha/mocha.js.orig b/zotero/node_modules/mocha/mocha.js
index c3e0c12..95fff4d 100644
--- a/zotero/node_modules/mocha/mocha.js.orig
+++ b/zotero/node_modules/mocha/mocha.js
@@ -11473,9 +11473,9 @@
 
   var browser$1 = {
   	info: 'ℹ️',
-  	success: '✅',
+  	success: '✔',
   	warning: '⚠️',
-  	error: '❌️'
+  	error: 'x'
   };
 
   var require$$0 = /*@__PURE__*/getAugmentedNamespace(_polyfillNode_events)
