From 2da4c6ff5146626b255e4858d619c1b6ac3f281d Mon Sep 17 00:00:00 2001
From: nibon7 <nibon7@163.com>
Date: Thu, 15 Jul 2021 09:28:06 +0800
Subject: [PATCH] remove dataframe feature from default

Signed-off-by: nibon7 <nibon7@163.com>
---
 Cargo.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Cargo.toml b/Cargo.toml
index afb73348..aa1f9050 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -84,7 +84,6 @@ default = [
     "post",
     "fetch",
     "zip-support",
-    "dataframe",
 ]
 
 stable = ["default"]
-- 
2.32.0

