diff --git a/libpod.conf b/libpod.conf
index 7e0228c1..c767b80d 100644
--- a/libpod.conf
+++ b/libpod.conf
@@ -24,10 +24,10 @@ conmon_env_vars = [
 ]
 
 # CGroup Manager - valid values are "systemd" and "cgroupfs"
-cgroup_manager = "systemd"
+cgroup_manager = "cgroupfs"
 
 # Container init binary
-#init_path = "/usr/libexec/podman/catatonit"
+#init_path = "/usr/bin/catatonit"
 
 # Directory for persistent libpod files (database, etc)
 # By default, this will be configured relative to where containers/storage
@@ -50,6 +50,7 @@ cni_config_dir = "/etc/cni/net.d/"
 
 # Directories where the CNI plugin binaries may be located
 cni_plugin_dir = [
+	       "/usr/share/cni-plugins/bin",
 	       "/usr/libexec/cni",
 	       "/usr/lib/cni",
 	       "/usr/local/lib/cni",

