--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,6 +17,7 @@
 debug = true # We want to be able to debug in the release build too 
 
 [patch.crates-io]
+pkg-config = { git = "https://github.com/selfisekai/pkg-config-rs", branch = "pkgc2-breaking" }
 # once a new piet (current v0.6.2) is released with the updated cairo, this can be removed
 piet = {git = "https://github.com/linebender/piet", rev = "1d8a6fd627d8d6492bc42a42c734205a981077f8"}
 piet-cairo = {git = "https://github.com/linebender/piet", rev = "1d8a6fd627d8d6492bc42a42c734205a981077f8"}
