fixes broken crate that doesn't like non-rustup triples
--
diff --git a/Cargo.lock b/Cargo.lock
index 2c66202..30940fb 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -4104,9 +4104,8 @@ dependencies = [
 
 [[package]]
 name = "system-deps"
-version = "6.0.3"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "2955b1fe31e1fa2fbd1976b71cc69a606d7d4da16f6de3333d0c92d51419aeff"
+version = "6.0.4"
+source = "git+https://github.com/nekopsykose/system-deps?tag=6.0.4#db1452ceff33c992cb947bfc089b7c6eb31f6273"
 dependencies = [
  "cfg-expr",
  "heck",
diff --git a/Cargo.toml b/Cargo.toml
index e9d01ca..7046726 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,4 +19,5 @@ codegen-units = 1
 [patch.crates-io]
 # once a new piet (current v0.6.2) is released with the updated cairo, this can be removed
 piet = {git = "https://github.com/linebender/piet", rev = "413397445e6772e0a651d60cf58a823b98f67c23"}
-piet-cairo = {git = "https://github.com/linebender/piet", rev = "413397445e6772e0a651d60cf58a823b98f67c23"}
\ No newline at end of file
+piet-cairo = {git = "https://github.com/linebender/piet", rev = "413397445e6772e0a651d60cf58a823b98f67c23"}
+system-deps = { git = "https://github.com/nekopsykose/system-deps", tag = "6.0.4" }
