--- a/Cargo.toml
+++ b/Cargo.toml
@@ -20,3 +20,6 @@
 # once a new piet (current v0.6.2) is released with the updated cairo, this can be removed
 piet = {git = "https://github.com/linebender/piet", rev = "1d8a6fd627d8d6492bc42a42c734205a981077f8"}
 piet-cairo = {git = "https://github.com/linebender/piet", rev = "1d8a6fd627d8d6492bc42a42c734205a981077f8"}
+
+[patch.'https://github.com/flxzt/ink-stroke-modeler-rs']
+ink-stroke-modeler-rs = { git = "https://github.com/comcloudway/ink-stroke-modeler-rs" }
