diff -up siege-2.69/doc/Makefile.in.good siege-2.69/doc/Makefile.in
--- siege-2.69/doc/Makefile.in.good	2009-07-04 11:12:24.411980518 -0300
+++ siege-2.69/doc/Makefile.in	2009-07-04 11:13:19.914733285 -0300
@@ -498,32 +498,32 @@ uninstall-man: uninstall-man1 uninstall-
 
 
 install-exec-hook: 
-	@if test -f $(SIEGERC); then \
-	  if cmp -s $(srcdir)/siegerc $(SIEGERC); then echo ""; \
-	    else \
-	    echo ' $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC).new'; \
-	    $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC).new; \
-	    echo "#####################################################"; \
-	    echo "WARNING: File $(SIEGERC) already exists."; \
-	    echo "         A new resource file has been installed as"; \
-	    echo "         $(SIEGERC).new. You may want to"; \
-	    echo "         consider using the newer version in order to"; \
-	    echo "         take advantage of any new features."; \
-	    echo "#####################################################"; \
-	  fi; \
-	else \
-	  $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC); \
-	fi 
-	@if test -f $(URLSTXT); then \
-	  if cmp -s $(srcdir)/siegerc $(URLSTXT); then echo ""; \
-	    else \
-	    echo "WARNING: File $(URLSTXT) already exists."; \
-	    echo "         It was NOT replaced with this installation."; \
-	  fi; \
-	else \
-	  $(mkinstalldirs) $(sysconfdir); \
-	  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
-	fi 
+	#@if test -f $(SIEGERC); then \
+	#  if cmp -s $(srcdir)/siegerc $(SIEGERC); then echo ""; \
+	#    else \
+	#    echo ' $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC).new'; \
+	#    $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC).new; \
+	#    echo "#####################################################"; \
+	#    echo "WARNING: File $(SIEGERC) already exists."; \
+	#    echo "         A new resource file has been installed as"; \
+	#    echo "         $(SIEGERC).new. You may want to"; \
+	#    echo "         consider using the newer version in order to"; \
+	#    echo "         take advantage of any new features."; \
+	#    echo "#####################################################"; \
+	#  fi; \
+	#else \
+	#  $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC); \
+	#fi 
+	#@if test -f $(URLSTXT); then \
+	#  if cmp -s $(srcdir)/siegerc $(URLSTXT); then echo ""; \
+	#    else \
+	#    echo "WARNING: File $(URLSTXT) already exists."; \
+	#    echo "         It was NOT replaced with this installation."; \
+	#  fi; \
+	#else \
+	#  $(mkinstalldirs) $(sysconfdir); \
+	#  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
+	#fi 
 
 uninstall:
 	rm -f $(SIEGERC)
