Remove bundled zlib.
--- a/Makefile.common
+++ b/Makefile.common
@@ -2,5 +2,4 @@
     $(wildcard $(CORE_DIR)/crocods-core/*.c) \
     $(wildcard $(CORE_DIR)/crocods-core/iniparser/*.c) \
     $(wildcard $(CORE_DIR)/crocods-core/resources/*.c) \
-    $(wildcard $(CORE_DIR)/deps/*.c) \
-    $(wildcard $(CORE_DIR)/deps/libz/*.c)
+    $(wildcard $(CORE_DIR)/deps/*.c)
--- a/Makefile
+++ b/Makefile
@@ -290,7 +290,7 @@
 
 CFLAGS += 
 LFLAGS := 
-LDFLAGS += $(LIBM)
+LDFLAGS += $(LIBM) -lz
 
 ifeq ($(platform), osx)
 ifndef ($(NOUNIVERSAL))
--- a/deps/ziptool.c
+++ b/deps/ziptool.c
@@ -4,7 +4,8 @@
 #include <string.h>
 #include <ctype.h>
 
-#include "libz/zlib.h"
+#include <zlib.h>
+
 #include "ziptool.h"
 
 typedef char bool;
