diff --git a/vendor/github.com/creack/goselect/fdset_64.go b/vendor/github.com/creack/goselect/fdset_64.go
index 0e70213..88136e5 100644
--- a/vendor/github.com/creack/goselect/fdset_64.go
+++ b/vendor/github.com/creack/goselect/fdset_64.go
@@ -1,5 +1,5 @@
 // +build !darwin,!netbsd,!openbsd
-// +build amd64 arm64 ppc64le mips64 mips64le s390x
+// +build amd64 arm64 ppc64le mips64 mips64le s390x loong64
 
 package goselect
 
