--- a/test/comments-ml.t/run.t
+++ b/test/comments-ml.t/run.t
@@ -1,9 +1,5 @@
 Format basic
-  $ refmt --print re ./input.re > ./formatted.re
-  refmt: FILENAMES… arguments: no './input.re' file
-  Usage: refmt [OPTION]… [FILENAMES]…
-  Try 'refmt --help' for more information.
-  [1]
+  $ refmt --print re ./input.ml > ./formatted.re
 
 Type-check basics
   $ ocamlc -c -pp 'refmt --print binary' -intf-suffix .rei -impl formatted.re
@@ -13,6 +9,3 @@
 
 Ensure idempotency: first format and second format are the same
   $ diff formatted.re formatted_back.re
-  0a1
-  > 
-  [1]
