gdbm.h has a conficting declaration of gdbm_version

diff --git a/makegdbm/options.h b/makegdbm/options.h
index 51597d1..9019af7 100644
--- a/makegdbm/options.h
+++ b/makegdbm/options.h
@@ -39,7 +39,6 @@
 #endif
 
 extern gdbm_error gdbm_errno;
-extern char *gdbm_version;
 
 #define COPYRIGHT \
   "(c) 1999 Horms <horms@verge.net.au>\nReleased under the GNU GPL\n"
diff --git a/perdition/db/gdbm/perditiondb_gdbm.h b/perdition/db/gdbm/perditiondb_gdbm.h
index 4047025..c23a8db 100644
--- a/perdition/db/gdbm/perditiondb_gdbm.h
+++ b/perdition/db/gdbm/perditiondb_gdbm.h
@@ -31,7 +31,6 @@
 #include <gdbm.h>
 
 extern gdbm_error gdbm_errno;
-extern char *gdbm_version;
 
 #ifndef PERDITIONDB_GDBM_SYSCONFDIR
 #define PERDITIONDB_GDBM_SYSCONFDIR "/usr/local/etc/perdition"
