diff --git a/nss/coreconf/config.gypi b/nss/coreconf/config.gypi
index 6900f79..9188d53 100644
--- a/nss/coreconf/config.gypi
+++ b/nss/coreconf/config.gypi
@@ -35,7 +35,7 @@
         ['OS=="mac"', {
           'use_system_sqlite%': 1,
         },{
-          'use_system_sqlite%': 0,
+          'use_system_sqlite%': 1,
         }],
         ['OS=="mac" or OS=="ios" or OS=="solaris" or OS=="win"', {
           'cc_use_gnu_ld%': 0,
@@ -102,8 +102,8 @@
     'disable_intel_hw_sha%': 0,
     'disable_tests%': 0,
     'disable_chachapoly%': 0,
-    'disable_deprecated_seed%': 0,
-    'disable_deprecated_rc2%': 0,
+    'disable_deprecated_seed%': 1,
+    'disable_deprecated_rc2%': 1,
     'disable_dbm%': 1,
     'disable_libpkix%': 1,
     'disable_werror%': 0,
