From cb890abcc236c6778eec3779fa80fc1a81175a6c Mon Sep 17 00:00:00 2001
From: Daniel Gray <dngray@polarbear.army>
Date: Mon, 14 Apr 2025 12:47:09 +0000
Subject: [PATCH 2/2] Separate init install script

---
 software/Makefile.linux | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/software/Makefile.linux b/software/Makefile.linux
index df8b3d2..adb51e7 100644
--- a/software/Makefile.linux
+++ b/software/Makefile.linux
@@ -52,9 +52,15 @@ install: infnoise
 	install -m 0755 infnoise $(PREFIX)/sbin/
 	install -d $(PREFIX)/lib/udev/rules.d/
 	install -m 0644 init_scripts/75-infnoise.rules $(PREFIX)/lib/udev/rules.d/
+
+install-systemd:
 	install -d $(PREFIX)/lib/systemd/system
 	install -m 0644 init_scripts/infnoise.service $(PREFIX)/lib/systemd/system
 
+install-openrc:
+	install -d $(DESTDIR)/etc/init.d/
+	install -m 0644 init_scripts/infnoise.gentoo.openrc $(DESTDIR)/etc/init.d/infnoise
+
 postinstall:
 	systemctl restart systemd-udevd
 	systemctl enable infnoise
-- 
2.49.0

