diff --git a/pre_commit/constants.py b/pre_commit/constants.py
index 3dcbbac..858fbb4 100644
--- a/pre_commit/constants.py
+++ b/pre_commit/constants.py
@@ -1,10 +1,5 @@
 import sys
 
-if sys.version_info < (3, 8):  # pragma: no cover (<PY38)
-    import importlib_metadata
-else:  # pragma: no cover (PY38+)
-    import importlib.metadata as importlib_metadata
-
 CONFIG_FILE = '.pre-commit-config.yaml'
 MANIFEST_FILE = '.pre-commit-hooks.yaml'
 
@@ -13,7 +8,7 @@ INSTALLED_STATE_VERSION = '1'
 # Bump when modifying `empty_template`
 LOCAL_REPO_VERSION = '1'
 
-VERSION = importlib_metadata.version('pre_commit')
+VERSION = %%PKGVER%%
 
 # `manual` is not invoked by any installed git hook.  See #719
 STAGES = (
