diff --git a/setup.py b/setup.py
index 43c4cab..d79ca61 100644
--- a/setup.py
+++ b/setup.py
@@ -68,8 +68,8 @@ setup(name='inquirer',
       include_package_data=True,
       zip_safe=False,
       install_requires=[
-          'blessings == 1.7',
-          'readchar == 2.0.1',
-          'python-editor==1.0.4'
+          'blessings >=1.7<2',
+          'readchar >=2<3',
+          'python-editor >=1<2'
       ],
       )
