--- ./egg/egg-asn1x.h.orig	2013-01-15 12:35:55.308701917 +0000
+++ ./egg/egg-asn1x.h	2013-01-15 12:41:45.224900393 +0000
@@ -25,6 +25,7 @@
 #define EGG_ASN1X_H_
 
 #include <glib.h>
+#include <libtasn1.h>
 
 #ifndef HAVE_EGG_ALLOCATOR
 typedef void* (*EggAllocator) (void* p, gsize);
@@ -35,13 +36,13 @@
 
 struct static_struct_asn;
 
-GNode*              egg_asn1x_create                 (const struct static_struct_asn *defs,
+GNode*              egg_asn1x_create                 (const ASN1_ARRAY_TYPE *defs,
                                                       const gchar *type);
 
-GNode*              egg_asn1x_create_quark           (const struct static_struct_asn *defs,
+GNode*              egg_asn1x_create_quark           (const ASN1_ARRAY_TYPE *defs,
                                                       GQuark type);
 
-GNode*              egg_asn1x_create_and_decode      (const struct static_struct_asn *defs,
+GNode*              egg_asn1x_create_and_decode      (const ASN1_ARRAY_TYPE *defs,
                                                       const gchar *type,
                                                       gconstpointer data,
                                                       gsize n_data);
