--- wine-1.7.8.orig/server/file.h
+++ wine-1.7.8/server/file.h
@@ -21,6 +21,7 @@
 #ifndef __WINE_SERVER_FILE_H
 #define __WINE_SERVER_FILE_H
 
+#include <fcntl.h>
 #include "object.h"
 
 struct fd;
--- ./dlls/ws2_32/socket.c.orig
+++ ./dlls/ws2_32/socket.c
@@ -100,7 +100,7 @@
 
 #ifdef HAVE_NETIPX_IPX_H
 # include <netipx/ipx.h>
-#elif defined(HAVE_LINUX_IPX_H)
+#elif defined(HAVE_LINUX_IPX_H) && defined(__GLIBC__)
 # ifdef HAVE_ASM_TYPES_H
 #  include <asm/types.h>
 # endif
