--- a/loader/Makefile.in
+++ b/loader/Makefile.in
@@ -25,7 +25,7 @@
 wine64_LDFLAGS = $(WINELOADER_LDFLAGS) $(LDEXECFLAGS) -lwine $(PTHREAD_LIBS)
 
 wine_preloader_OBJS = preloader.o preloader_mac.o
-wine_preloader_LDFLAGS = $(WINEPRELOADER_LDFLAGS)
+wine_preloader_LDFLAGS = $(WINEPRELOADER_LDFLAGS) -no-pie
 
 wine64_preloader_OBJS = preloader.o preloader_mac.o
-wine64_preloader_LDFLAGS = $(WINEPRELOADER_LDFLAGS)
+wine64_preloader_LDFLAGS = $(WINEPRELOADER_LDFLAGS) -no-pie
