From 4d940c7932c32ed6885c72b5a2e70101b018ad8c Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 6 Jul 2009 08:48:25 +0000
Subject: [PATCH] fetch: respect --simulate option

---
 src/fetch.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/fetch.c b/src/fetch.c
index 84f0b39..501dff1 100644
--- a/src/fetch.c
+++ b/src/fetch.c
@@ -82,6 +82,9 @@ static int fetch_package(struct fetch_ctx *fctx,
 		return -1;
 	}
 
+	if (apk_flags & APK_SIMULATE)
+		return 0;
+
 	snprintf(infile, sizeof(infile), "%s/%s-%s.apk",
 		 db->repos[i].url, pkg->name->name, pkg->version);
 
-- 
1.6.3.3

