diff --git a/Make.rules b/Make.rules
index c864064..81881a7 100644
--- a/Make.rules
+++ b/Make.rules
@@ -47,7 +47,7 @@ export srctree objtree
 ##
 # Consult SCM for better version string.
 
-GIT_REV := $(shell git describe || echo exported)
+GIT_REV := $(shell test -d .git && git describe || echo exported)
 ifneq ($(GIT_REV), exported)
 FULL_VERSION	:= $(patsubst $(PACKAGE)-%,%,$(GIT_REV))
 FULL_VERSION	:= $(patsubst v%,%,$(FULL_VERSION))
