--- a/make/system.mk.orig
+++ b/make/system.mk
@@ -66,6 +66,10 @@
   override ARCHITECTURE   = arm32
   ARCHITECTURE_FAMILY     = arm32
   ARCHITECTURE_CFLAGS    := -march=armv7-a -marm
+else ifeq ($(patsubst armv8l%,armv7,$(BUILD_ARCH)),armv7)
+  override ARCHITECTURE   = arm32
+  ARCHITECTURE_FAMILY     = arm32
+  ARCHITECTURE_CFLAGS    := -march=armv7-a -marm
 else ifeq ($(patsubst armv8%,armv8,$(BUILD_ARCH)),armv8)
   override ARCHITECTURE   = aarch64
   ARCHITECTURE_FAMILY     = aarch64
