From 2c81d1cba0ccf59593d2c154a099fa1e766838ac Mon Sep 17 00:00:00 2001
From: James Browning <jamesb.fe80@gmail.com>
Date: Wed, 14 Oct 2020 11:06:19 -0700
Subject: [PATCH 01/18] FFI: Fix the libntpc install path.

---
 libntp/wscript | 1 -
 1 file changed, 1 deletion(-)

diff --git a/libntp/wscript b/libntp/wscript
index 34d84f68b..6f7659997 100644
--- a/libntp/wscript
+++ b/libntp/wscript
@@ -53,7 +53,6 @@ def build(ctx):
         # Loadable FFI stub
         ctx(
             features="c cshlib",
-            install_path='${LIBDIR}/ntp',
             includes=[ctx.bldnode.parent.abspath(), "../include"],
             source=["ntp_c.c", "pymodule-mac.c"] + libntp_source_sharable,
             target="../pylib/ntpc",  # Put the output in the pylib directory
-- 
2.29.2

