diff --git a/include/sys/thread.h b/include/sys/thread.h
index 433a076..1ca9911 100644
--- a/include/sys/thread.h
+++ b/include/sys/thread.h
@@ -46,7 +46,7 @@
 typedef void (*thread_func_t)(void *);
 
 #define thread_create(stk, stksize, func, arg, len, pp, state, pri)      \
-	__thread_create(stk, stksize, (thread_func_t)func,               \
+	__thread_create(stk, stksize, func,                              \
 	                #func, arg, len, pp, state, pri)
 #define thread_exit()			__thread_exit()
 #define thread_join(t)			VERIFY(0)
