--- gnome-doc-utils-0.18.1/doc/xslt/Makefile.in.orig
+++ gnome-doc-utils-0.18.1/doc/xslt/Makefile.in
@@ -947,11 +947,11 @@
 all: $(xsldoc_docs) $(xsldoc_xmls)
 
 $(xsldoc_docs): $(xsldoc_xsls) xsldoc.awk xsldoc-fill.xsl
-	$(GDU_AWK) -f "$(srcdir)/xsldoc.awk" "$(filter %/$(basename $(notdir $@)).xsl,$(xsldoc_xsls))" \
-	| xsltproc -o "$@" \
+	$(GDU_AWK) -f "$(srcdir)/xsldoc.awk" "$(filter %/$(basename $(notdir $@)).xsl,$(xsldoc_xsls))" > $@.tmp
+	xsltproc -o "$@" \
 	--stringparam basename "$(basename $(notdir $@))" \
 	--stringparam xsl_file "$(filter %/$(basename $(notdir $@)).xsl,$(xsldoc_xsls))" \
-	"$(srcdir)/xsldoc-fill.xsl" -
+	"$(srcdir)/xsldoc-fill.xsl" $@.tmp && rm -f $@.tmp
 
 $(xsldoc_xmls): xsldoc-docbook.xsl
 $(xsldoc_xmls): C/%.xml : C/%.xsldoc
