diff --git a/pyproject.toml b/pyproject.toml
index a7d504f..3e764bb 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -125,10 +125,6 @@ addopts = "-v -rsxfE --durations=10 --color=yes --cov-config=pyproject.toml"
 filterwarnings = [
     # From Cython-1753
     "ignore:can't resolve:ImportWarning",
-    "error:::dask[.*]",
-    "error:::pandas[.*]",
-    "error:::numpy[.*]",
-    "error:::distributed[.*]",
     # From https://github.com/numpy/numpy/issues/20225
     'ignore:The distutils\.sysconfig module is deprecated, use sysconfig instead:DeprecationWarning:numpy',
     # With no network access, distributed raises a warning when detecting local IP address
