--- ./inc/nls.h.orig
+++ ./inc/nls.h
@@ -1,11 +1,10 @@
 #ifndef NLS_H
 #define NLS_H
 
-#include <libintl.h>
-
 #ifdef NO_LIBINTL_OR_GETTEXT
 #define _(String) (String)
 #else
+#include <libintl.h>
 #define _(String) gettext(String)
 #endif
 #define gettext_noop(String) (String)
--- ./Makefile.orig
+++ ./Makefile
@@ -80,10 +80,14 @@
 
 MAN_SRC = $(wildcard $(MAN)/*.*)
 
+ifeq ($(NLS),no)
+CFLAGS += -DNO_LIBINTL_OR_GETTEXT
+else
 PO_FILES = $(filter-out $(MESSAGES),$(wildcard $(PO)/*.po))
 MO_FILES = $(PO_FILES:$(PO)/%.po=$(MO)/%.mo)
 LANGUAGES ?= $(PO_FILES:$(PO)/%.po=%)
 NLS_FILES = $(LANGUAGES:%=$(DESTDIR)$(LOCALEDIR)/%/$(MESSDIR)/$(PACKAGE).mo)
+endif
 MAN_FILES = $(foreach FILE, $(MAN_SRC), \
               $(DESTDIR)$(MANDIR)/man$(subst .,,$(suffix $(FILE)))/$(FILE:$(MAN)/%=%))
 
