From b12b4ad0a6b594db981a6f38358004009c6f75ec Mon Sep 17 00:00:00 2001
From: Jonathan Dowland <jon@dow.land>
Date: Wed, 12 Jun 2024 16:32:01 +0100
Subject: [PATCH] =?UTF-8?q?Adjustments=20for=20po4a=20=E2=89=A5=200.7.0?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

---
 IkiWiki/Plugin/po.pm | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/IkiWiki/Plugin/po.pm b/IkiWiki/Plugin/po.pm
index 858f4e0f5..b4c76546c 100644
--- a/IkiWiki/Plugin/po.pm
+++ b/IkiWiki/Plugin/po.pm
@@ -904,16 +904,13 @@ sub refreshpot ($) {
 
 	my $doc=Locale::Po4a::Chooser::new(po4a_type($masterfile),
 					   po4a_options($masterfile));
-	$doc->read($masterfile);
+	$doc->read($masterfile,$masterfile,"UTF-8");
 	$doc->{TT}{utf_mode} = 1;
-	$doc->detected_charset('UTF-8');
-	$doc->{TT}{po_in}->set_charset('UTF-8');
 	# let's cheat a bit to force porefs option to be passed to
 	# Locale::Po4a::Po; this is undocument use of internal
 	# Locale::Po4a::TransTractor's data, compulsory since this module
 	# prevents us from using the porefs option.
 	$doc->{TT}{po_out}=Locale::Po4a::Po->new({ 'porefs' => 'none' });
-	$doc->{TT}{po_out}->set_charset('UTF-8');
 	# do the actual work
 	$doc->parse;
 	IkiWiki::prep_writefile(basename($potfile),dirname($potfile));
-- 
2.39.2

