diff --git a/unittests/test_background.py.orig b/unittests/test_background.py
index 3b8884788a6..602772e1269 100644
--- a/unittests/test_background.py.orig
+++ b/unittests/test_background.py
@@ -4,6 +4,9 @@ from colored.library import Library
 from colored.colored import back, back_rgb
 from colored.background import Back
 
+from colored import set_tty_aware
+set_tty_aware(False)
+
 
 class TestBackground(unittest.TestCase):
 
