--- ./configure.orig	2016-08-02 14:56:14.888475820 +0200
+++ ./configure	2016-08-02 14:56:26.633158063 +0200
@@ -17247,7 +17247,7 @@
 # The reason for -Wno-error=unused-result is a glibc/gcc interaction
 # idiocy where write is marked as warn_unused_result, causing build
 # failures.
-CFLAGS="${CFLAGS} -Wall -Werror"
+#CFLAGS="${CFLAGS} -Wall -Werror"
 OCFLAGS="${OCFLAGS} -Wall -Werror"
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts -Werror=unused-result" >&5
 $as_echo_n "checking whether C compiler accepts -Werror=unused-result... " >&6; }
