--- a/src/ci.c
+++ b/src/ci.c
@@ -175,7 +175,7 @@ main(int argc, char **argv)
         if( toupper(getchar()) == 'Y'){
           create_htmlHome=1;
           fprintf(stderr,"Redirection to /html/index.html is enabled.\n");
-          stdin = freopen(NULL,"r",stdin); //cleanup stdin to remove \n
+          freopen(NULL,"r",stdin); //cleanup stdin to remove \n
         }
         else {
           fprintf(stderr,"\nCanceled!\n");
