--- a/cmake/modules/FindGIOUnix.cmake
+++ b/cmake/modules/FindGIOUnix.cmake
@@ -12,20 +12,20 @@
 
 if(GIOUNIX_INCLUDE_DIR)
     # Already in cache, be silent
-    set(GIOUNIX_FIND_QUIETLY TRUE)
+    set(GIOUnix_FIND_QUIETLY TRUE)
 endif()
 
 include(UsePkgConfig)
 pkg_check_modules(PC_LibGIOUnix gio-unix-2.0)
 
-find_path(GIOUNIX_MAIN_INCLUDE_DIR
+find_path(GIOUnix_MAIN_INCLUDE_DIR
           NAMES gio/gunixconnection.h
           HINTS ${PC_LibGIOUnix_INCLUDEDIR}
           PATH_SUFFIXES gio-unix-2.0)
 
-set(GIOUNIX_INCLUDE_DIR "${GIOUNIX_MAIN_INCLUDE_DIR}")
+set(GIOUnix_INCLUDE_DIR "${GIOUnix_MAIN_INCLUDE_DIR}")
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(GIOUNIX DEFAULT_MSG GIOUNIX_MAIN_INCLUDE_DIR)
+find_package_handle_standard_args(GIOUnix DEFAULT_MSG GIOUnix_MAIN_INCLUDE_DIR)
 
-mark_as_advanced(GIOUNIX_INCLUDE_DIR)
+mark_as_advanced(GIOUnix_INCLUDE_DIR)
