--- ipfw3-2012.orig/kipfw/ipfw2_mod.c
+++ ipfw3-2012/kipfw/ipfw2_mod.c
@@ -466,7 +466,12 @@
  * so we have an #ifdef to set the proper argument type.
  */
 static unsigned int
-call_ipfw(unsigned int hooknum,
+call_ipfw(
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3,13,0)
+	unsigned int hooknum,
+#else
+	const struct nf_hook_ops *hooknum,
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23) // in 2.6.22 we have **
 	struct sk_buff  **skb,
 #else
