--- ipfw3-2012.orig/sys/netinet/ipfw/ip_dummynet.c
+++ ipfw3-2012/sys/netinet/ipfw/ip_dummynet.c
@@ -635,7 +635,7 @@
 		fs->sched->fp->free_fsk(fs);
 	fs->sched = NULL;
 	if (flags & DN_DELETE_FS) {
-		bzero(fs, sizeof(fs));	/* safety */
+		bzero(fs, sizeof(*fs));	/* safety */
 		free(fs, M_DUMMYNET);
 		dn_cfg.fsk_count--;
 	} else {
--- ipfw3-2012.orig/kipfw/ipfw2_mod.c
+++ ipfw3-2012/kipfw/ipfw2_mod.c
@@ -218,7 +218,7 @@
 	struct thread t;
 	int ret = EINVAL;
 
-	memset(s, 0, sizeof(s));
+	memset(s, 0, sizeof(*s));
 	s->sopt_name = cmd;
 	s->sopt_dir = dir;
 	s->sopt_valsize = len;
