--- python-debian-0.1.44/lib/debian/tests/test_deb822.py
+++ python-debian-0.1.44/lib/debian/tests/test_deb822.py
@@ -1334,6 +1334,7 @@
             changes = deb822.Changes(f)
         self.assertEqual('python-debian', changes['Source'])
 
+    @unittest.skip("r.date.strftime('%s') != '1388595834'")
     def test_removals(self):
         # type: () -> None
         with open(find_test_file('test_removals.822')) as f:
