diff --git a/gnome-music.in b/gnome-music.in
index 13f8aeb..043490f 100755
--- a/gnome-music.in
+++ b/gnome-music.in
@@ -105,8 +105,6 @@ def set_log_level():
 
 def set_internationalization():
     """Sets application internationalization."""
-    locale.bindtextdomain('@application_id@', LOCALE_DIR)
-    locale.textdomain('@application_id@')
     gettext.bindtextdomain('@application_id@', LOCALE_DIR)
     gettext.textdomain('@application_id@')
 
