--- a/ini/ini_validators_ut_check.c
+++ b/ini/ini_validators_ut_check.c
@@ -608,7 +608,7 @@ START_TEST(test_ini_allowed_options_wrong_regex)
     ret = strcmp(errmsg,
                  "[rule/options_for_foo]: Cannot compile regular expression "
                  "from option 'section_re'. "
-                 "Error: 'Unmatched [ or [^'");
+                 "Error: 'Missing ']''");
     if (ret != 0) {
         ret = strcmp(errmsg,
                      "[rule/options_for_foo]: Cannot compile regular "
@@ -1042,7 +1042,7 @@ START_TEST(test_ini_allowed_sections_wrong_regex)
     errmsg = ini_errobj_get_msg(errobj);
     ret = strcmp(errmsg,
                  "[rule/section_list]: Validator failed to use regex "
-                 "[^foo\\(*$]:[Unmatched ( or \\(]");
+                 "[^foo\\(*$]:[Missing ')']");
     if (ret !=0) {
         ret = strcmp(errmsg,
                      "[rule/section_list]: Validator failed to use regex "
--- a/path_utils/path_utils.c
+++ b/path_utils/path_utils.c
@@ -31,7 +31,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <dirent.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <sys/stat.h>
 
 #include <libgen.h>
