diff --git a/Makefile.am b/Makefile.am
index ef4fd47..53ed4a1 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -306,9 +306,6 @@ tests_mapper_CFLAGS = -O0
 test_alignment_SOURCES = tests/test-alignment.c tests/util.c
 test_alignment_LDADD = libyara.la
 test_alignment_LDFLAGS = -static
-test_arena_SOURCES = tests/test-arena.c tests/util.c
-test_arena_LDADD = libyara.la
-test_arena_LDFLAGS = -static
 test_atoms_SOURCES = tests/test-atoms.c tests/util.c
 test_atoms_LDADD = libyara.la
 test_atoms_LDFLAGS = -static
@@ -327,9 +324,6 @@ test_elf_LDFLAGS = -static
 test_version_SOURCES = tests/test-version.c tests/util.c
 test_version_LDADD = libyara.la
 test_version_LDFLAGS = -static
-test_api_SOURCES = tests/test-api.c tests/util.c
-test_api_LDADD = libyara.la
-test_api_LDFLAGS = -static
 test_bitmask_SOURCES = tests/test-bitmask.c tests/util.c
 test_bitmask_LDADD = libyara.la
 test_bitmask_LDFLAGS = -static
@@ -353,10 +347,8 @@ TESTS = $(check_PROGRAMS)
 TESTS_ENVIRONMENT = TOP_SRCDIR=$(top_srcdir) TOP_BUILDDIR=$(top_builddir)
 
 check_PROGRAMS = \
-  test-arena \
   test-alignment \
   test-atoms \
-  test-api \
   test-rules \
   test-pe \
   test-elf \
