diff --git a/Makefile.am b/Makefile.am
index 062b77f..5bde105 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -65,11 +65,6 @@
 test_arena_LDADD = libyara/.libs/libyara.a
 test_atoms_SOURCES = tests/test-atoms.c tests/util.c
 test_atoms_LDADD = libyara/.libs/libyara.a
-test_rules_SOURCES = tests/test-rules.c tests/util.c
-test_rules_LDADD = libyara/.libs/libyara.a
-if POSIX
-EXTRA_test_rules_DEPENDENCIES = tests/mapper$(EXEEXT)
-endif
 test_pe_SOURCES = tests/test-pe.c tests/util.c
 test_pe_LDADD = libyara/.libs/libyara.a
 test_elf_SOURCES = tests/test-elf.c tests/util.c
@@ -97,7 +92,6 @@
   test-alignment \
   test-atoms \
   test-api \
-  test-rules \
   test-pe \
   test-elf \
   test-version \
