diff --git a/sources/CMakeLists.txt b/sources/CMakeLists.txt
index 81f91a9..602d327 100644
--- a/sources/CMakeLists.txt
+++ b/sources/CMakeLists.txt
@@ -77,7 +77,7 @@ if(THREADS_HAVE_PTHREAD_ARG)
     set_property(TARGET ${od_binary} PROPERTY INTERFACE_COMPILE_OPTIONS "-pthread")
 endif()
 
-target_link_libraries(${od_binary} ${od_libraries} ${CMAKE_THREAD_LIBS_INIT} m)
+target_link_libraries(${od_binary} ${od_libraries} ${CMAKE_THREAD_LIBS_INIT} m argp)
 
 if (BUILD_COMPRESSION)
     target_link_libraries(${od_binary} ${compression_libraries})
