--- a/src/recall_list_manager.cpp
+++ b/src/recall_list_manager.cpp
@@ -57,7 +57,7 @@
 void recall_list_manager::erase_if_matches_id(const std::string &unit_id)
 {
 	recall_list_.erase(std::remove_if(recall_list_.begin(), recall_list_.end(),
-		[&unit_id](const unit_ptr & ptr) { return ptr->id() == unit_id; }),
+		[unit_id](const unit_ptr & ptr) { return ptr->id() == unit_id; }),
 	                       recall_list_.end());
 }
 
