diff --git a/src/lib/Coordinate.h b/src/lib/Coordinate.h
index bec3fe2..b87608d 100644
--- a/src/lib/Coordinate.h
+++ b/src/lib/Coordinate.h
@@ -10,6 +10,8 @@
 #ifndef INCLUDED_COORDINATE_H
 #define INCLUDED_COORDINATE_H
 
+#include <cstdint>
+
 #include "MSPUBConstants.h"
 
 namespace libmspub
diff --git a/src/lib/MSPUBMetaData.h b/src/lib/MSPUBMetaData.h
index 4456c2a..917336a 100644
--- a/src/lib/MSPUBMetaData.h
+++ b/src/lib/MSPUBMetaData.h
@@ -14,6 +14,7 @@
 #include <utility>
 #include <vector>
 #include <sys/types.h> /* for uintX_t types */
+#include <cstdint>
 
 #include <librevenge/librevenge.h>
 
