diff --git a/tests/httpbin.h.in b/tests/httpbin.h.in
index 0010068..6a96e66 100644
--- a/tests/httpbin.h.in
+++ b/tests/httpbin.h.in
@@ -45,7 +45,7 @@ struct Instance
               core::posix::exec("/usr/bin/python3", {"-c", "from httpbin import app; app.run()"}, {}, core::posix::StandardStream::stdout /*| core::posix::StandardStream::stderr*/)
           }
     {
-        std::this_thread::sleep_for(std::chrono::milliseconds{1000});
+        std::this_thread::sleep_for(std::chrono::milliseconds{10000});
     }
 
     ~Instance()
