diff -Nuard ocfs2-tools-1.4.3.orig/ocfs2_controld/Makefile ocfs2-tools-1.4.3/ocfs2_controld/Makefile
--- ocfs2-tools-1.4.3.orig/ocfs2_controld/Makefile	2009-04-14 00:10:40.000000000 +0200
+++ ocfs2-tools-1.4.3/ocfs2_controld/Makefile	2009-09-30 08:22:04.000000000 +0200
@@ -17,7 +17,7 @@
 PCMK_INCLUDES = -I/usr/include/pacemaker -I/usr/include/heartbeat/ $(GLIB_CFLAGS)
 endif
 
-INCLUDES = -I$(TOPDIR)/include -I. $(PCMK_INCLUDES)
+INCLUDES = -I/usr/include/libxml2 -I$(TOPDIR)/include -I. $(PCMK_INCLUDES)
 LIBO2CB_LIBS = -L$(TOPDIR)/libo2cb -lo2cb
 LIBO2CB_DEPS = $(TOPDIR)/libo2cb/libo2cb.a
 LIBOCFS2_LIBS = -L$(TOPDIR)/libocfs2 -locfs2
diff -Nuard ocfs2-tools-1.4.3.orig/ocfs2_controld/pacemaker.c ocfs2-tools-1.4.3/ocfs2_controld/pacemaker.c
--- ocfs2-tools-1.4.3.orig/ocfs2_controld/pacemaker.c	2009-05-06 03:36:57.000000000 +0200
+++ ocfs2-tools-1.4.3/ocfs2_controld/pacemaker.c	2009-09-30 08:22:22.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <bzlib.h>
 
+#include <corosync/corodefs.h>
 #include <pacemaker/crm_config.h>
 /* heartbeat support is irrelevant here */
 #undef SUPPORT_HEARTBEAT
@@ -155,7 +156,7 @@
 	crm_log_init("ocfs2_controld", LOG_INFO, FALSE, TRUE, 0, NULL);
 
 	if(init_ais_connection(NULL, NULL, NULL, &local_node_uname, &our_nodeid) == FALSE) {
-		log_error("Connection to our AIS plugin (%d) failed", CRM_SERVICE);
+		log_error("Connection to our AIS plugin (%d) failed", PCMK_SERVICE);
 		return -1;
 	}
 
