From: Jakub Jirutka <jakub@jirutka.cz>
Date: Wed, 23 May 2018 21:18:00 +0200
Subject: [PATCH] Replace include sys/raw.h with linux/raw.h

--- a/debugfs.ocfs2/include/main.h
+++ b/debugfs.ocfs2/include/main.h
@@ -45,7 +45,7 @@
 #include <arpa/inet.h>
 #include <netinet/in.h>
 #include <signal.h>
-#include <sys/raw.h>
+#include <linux/raw.h>
 #include <linux/kdev_t.h>
 #include <inttypes.h>
 #include <utime.h>
--- a/fswreck/include/main.h
+++ b/fswreck/include/main.h
@@ -43,7 +43,7 @@
 #include <arpa/inet.h>
 #include <netinet/in.h>
 #include <signal.h>
-#include <sys/raw.h>
+#include <linux/raw.h>
 #include <linux/kdev_t.h>
 #include <inttypes.h>
 
--- a/o2info/operations.c
+++ b/o2info/operations.c
@@ -22,7 +22,7 @@
 #define _GNU_SOURCE /* Because libc really doesn't want us using O_DIRECT? */
 
 #include <errno.h>
-#include <sys/raw.h>
+#include <linux/raw.h>
 #include <inttypes.h>
 
 #include "ocfs2/ocfs2.h"
