diff --git a/src/third-party/flatbuffers/stl_emulation.h b/src/third-party/flatbuffers/stl_emulation.h
index 975de21..5464029 100644
--- a/src/third-party/flatbuffers/stl_emulation.h
+++ b/src/third-party/flatbuffers/stl_emulation.h
@@ -547,7 +547,7 @@ class span FLATBUFFERS_FINAL_CLASS {
   FLATBUFFERS_CONSTEXPR_CPP14 span &operator=(const span &other)
       FLATBUFFERS_NOEXCEPT {
     data_ = other.data_;
-    count_ = other.count_;
+    //count_ = other.count_;
   }
 
   // Limited implementation of
diff --git a/src/third-party/flatbuffers/util.h b/src/third-party/flatbuffers/util.h
index f2659bd..3779c50 100644
--- a/src/third-party/flatbuffers/util.h
+++ b/src/third-party/flatbuffers/util.h
@@ -217,6 +217,7 @@ inline std::string IntToStringHex(int i, int xdigits) {
 
 // clang-format off
 // Use locale independent functions {strtod_l, strtof_l, strtoll_l, strtoull_l}.
+#define FLATBUFFERS_LOCALE_INDEPENDENT 0
 #if defined(FLATBUFFERS_LOCALE_INDEPENDENT) && (FLATBUFFERS_LOCALE_INDEPENDENT > 0)
   class ClassicLocale {
     #ifdef _WIN32
@@ -250,7 +251,7 @@ inline std::string IntToStringHex(int i, int xdigits) {
     #define __strtof_impl(s, pe) strtof_l(s, pe, ClassicLocale::Get())
   #endif
 #else
-  #include "flatbuffers/cata_error_locale_dependent.h"
+  // #include "flatbuffers/cata_error_locale_dependent.h"
   #define __strtod_impl(s, pe) strtod(s, pe)
   #define __strtof_impl(s, pe) static_cast<float>(strtod(s, pe))
   #ifdef _MSC_VER
