diff -up dosbox-0.74/src/gui/sdlmain.cpp.orig dosbox-0.74/src/gui/sdlmain.cpp
--- dosbox-0.74/src/gui/sdlmain.cpp.orig	2013-12-18 15:06:41.227839719 +0100
+++ dosbox-0.74/src/gui/sdlmain.cpp	2013-12-18 15:07:14.434769053 +0100
@@ -1529,7 +1529,7 @@ static void show_warning(char const * co
 	if ( !sdl.inited && SDL_Init(SDL_INIT_VIDEO|SDL_INIT_NOPARACHUTE) < 0 ) textonly = true;
 	sdl.inited = true;
 #endif
-	printf(message);
+	printf("%s", message);
 	if(textonly) return;
 	if(!sdl.surface) sdl.surface = SDL_SetVideoMode(640,400,0,0);
 	if(!sdl.surface) return;
