Index: hyperic-sigar-1.6.4+dfsg/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java
===================================================================
--- hyperic-sigar-1.6.4+dfsg.orig/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java	2014-11-03 13:25:30.620030459 +0800
+++ hyperic-sigar-1.6.4+dfsg/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java	2014-11-03 13:25:30.608030459 +0800
@@ -75,7 +75,8 @@
         if (ArchName.is64()) {
             getProject().setProperty("jni.arch64", "true");
             if (ArchLoader.IS_LINUX) {
-                if (!osArch.equals("ia64")) {
+                if (!osArch.equals("ia64") && !osArch.equals("mips64el")
+                      && !osArch.equals("mips64") && !osArch.equals("aarch64")) {
                     getProject().setProperty("jni.gccm", "-m64");
                 }
             }
