--- gettext-tiny-0.2.0/include/libintl.h.old	2017-08-02 06:14:20.000000000 -0500
+++ gettext-tiny-0.2.0/include/libintl.h	2018-06-15 02:31:14.241446687 -0500
@@ -34,14 +34,14 @@
 #undef gettext_noop
 #define gettext_noop(X) X
 
-#ifndef LIBINTL_NO_MACROS
+#ifdef LIBINTL_MACROS
 /* if these macros are defined, configure checks will detect libintl as
  * built into the libc because test programs will work without -lintl.
  * for example:
  * checking for ngettext in libc ... yes
  * the consequence is that -lintl will not be added to the LDFLAGS.
- * so if for some reason you want that libintl.a gets linked,
- * add -DLIBINTL_NO_MACROS=1 to your CPPFLAGS. */
+ * so if for some reason you dont want that libintl.a gets linked,
+ * add -DLIBINTL_MACROS=1 to your CPPFLAGS. */
 
 #define gettext(X) ((char*) (X))
 #define dgettext(dom, X) ((void)(dom), (char*) (X))
