--- a/hjson/tool.py
+++ b/hjson/tool.py
@@ -11,7 +11,8 @@
 from __future__ import with_statement
 import sys
 import hjson
-import pkg_resources  # part of setuptools
+
+VERSION='0.0.0-dev'
 
 HELP="""Hjson, a user interface for JSON
 
@@ -42,7 +43,7 @@
         elif arg == '-j': format = 'json'
         elif arg == '-c': format = 'compact'
         elif arg == '-V' or arg == '--version':
-            showerr('Hjson ' + pkg_resources.require("Hjson")[0].version)
+            showerr('Hjson ' + VERSION)
             return
 
         elif arg[0] == '-':
