Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: install *.cmake files to /usr/lib; set SOVERSION
----

--- a/CMakeLists.txt.orig
+++ b/CMakeLists.txt
@@ -42,6 +42,9 @@
 )

 add_library(OpenDrive ${SOURCES})
+set_target_properties(OpenDrive PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR})
 target_include_directories(OpenDrive
     PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>
@@ -63,4 +66,4 @@
     ${CMAKE_CURRENT_SOURCE_DIR}/thirdparty/pugixml/pugiconfig.hpp DESTINATION "include/pugixml/")

 install(TARGETS OpenDrive EXPORT OpenDriveConfig)
-install(EXPORT OpenDriveConfig NAMESPACE OpenDrive:: DESTINATION cmake)
+install(EXPORT OpenDriveConfig NAMESPACE OpenDrive:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
