--- lftp-4.4.15/lib/stdbool.in.h.orig
+++ lftp-4.4.15/lib/stdbool.in.h
@@ -111,7 +111,7 @@
       (see ISO C 99 6.7.2.2.(4)); however, '_Bool' must promote to 'int'
       (see ISO C 99 6.3.1.1.(2)).  So add a negative value to the
       enum; this ensures that '_Bool' promotes to 'int'.  */
-typedef enum { _Bool_must_promote_to_int = -1, false = 0, true = 1 } _Bool;
+typedef enum { _Bool_must_promote_to_int = -1, false = 0, true = 1 }; _Bool;
 #   endif
 #  endif
 # endif
