diff --git a/build.rs b/build.rs
index c9183bd..60489f4 100644
--- a/build.rs
+++ b/build.rs
@@ -4,7 +4,7 @@ use std::path::PathBuf;
 fn main() {
     println!("cargo:rerun-if-changed=proto/galaxy.protocols.communication_service.proto");
     println!("cargo:rerun-if-env-changed=PROTOC");
-    let vendored_protoc = protoc_bin_vendored::protoc_bin_path().unwrap();
+    let vendored_protoc = PathBuf::from("/usr/bin/protoc");
     let protoc_path = match std::env::var("PROTOC") {
         Ok(protoc) => PathBuf::from(&protoc),
         Err(_) => vendored_protoc,
