From f8e83c4c7866ae830abe9bec15cfe805dd825b6e Mon Sep 17 00:00:00 2001
From: xentec <xentec@aix0.eu>
Date: Tue, 12 Sep 2017 07:18:38 +0200
Subject: [PATCH] Disable dynamic lib tests for musl's dlclose() is noop

---
 unittests/Support/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/unittests/Support/CMakeLists.txt b/unittests/Support/CMakeLists.txt
index 641163e..4f508d9 100644
--- a/llvm/unittests/Support/CMakeLists.txt
+++ b/llvm/unittests/Support/CMakeLists.txt
@@ -124,5 +124,5 @@
 
 if(NOT LLVM_INTEGRATED_CRT_ALLOC)
   # The test doesn't pass when using a custom allocator, PR47881.
-  add_subdirectory(DynamicLibrary)
+  # add_subdirectory(DynamicLibrary)
 endif()


