diff --git a/config.mk b/config.mk
index 6e369c2..ecc4990 100644
--- a/config.mk
+++ b/config.mk
@@ -26,7 +26,7 @@ WITH_TLS:=yes
 # Comment out to disable TLS/PSK support in the broker and client. Requires
 # WITH_TLS=yes.
 # This must be disabled if using openssl < 1.0.
-WITH_TLS_PSK:=yes
+WITH_TLS_PSK:=no
 
 # Comment out to disable client client threading support.
 WITH_THREADING:=yes
@@ -46,7 +46,7 @@ WITH_PERSISTENCE:=yes
 # Comment out to remove memory tracking support from the broker. If disabled,
 # mosquitto won't track heap memory usage nor export '$SYS/broker/heap/current
 # size', but will use slightly less memory and CPU time.
-WITH_MEMORY_TRACKING:=yes
+WITH_MEMORY_TRACKING:=no
 
 # Compile with database upgrading support? If disabled, mosquitto won't
 # automatically upgrade old database versions.
@@ -65,7 +65,7 @@ WITH_SRV:=yes
 WITH_UUID:=yes
 
 # Build with websockets support on the broker.
-WITH_WEBSOCKETS:=no
+WITH_WEBSOCKETS:=yes
 
 # Use elliptic keys in broker
 WITH_EC:=yes
@@ -78,6 +78,7 @@ WITH_SOCKS:=yes
 
 # Build with async dns lookup support for bridges (temporary). Requires glibc.
 #WITH_ADNS:=yes
+WITH_ADNS=no
 
 # =============================================================================
 # End of user configuration
@@ -159,10 +160,6 @@ ifeq ($(UNAME),QNX)
 	LIB_LIBS:=$(LIB_LIBS) -lsocket
 endif
 
-ifeq ($(UNAME),Linux)
-	BROKER_LIBS:=$(BROKER_LIBS) -lanl
-endif
-
 ifeq ($(WITH_WRAP),yes)
 	BROKER_LIBS:=$(BROKER_LIBS) -lwrap
 	BROKER_CFLAGS:=$(BROKER_CFLAGS) -DWITH_WRAP
@@ -253,7 +250,7 @@ ifeq ($(WITH_DOCS),yes)
 endif
 
 INSTALL?=install
-prefix=/usr/local
+prefix=/usr
 mandir=${prefix}/share/man
 localedir=${prefix}/share/locale
 STRIP?=strip
