diff --git a/config.mk b/config.mk
index 46ffb6d..4e7b31d 100644
--- a/config.mk
+++ b/config.mk
@@ -26,7 +26,7 @@ WITH_TLS:=yes
 # Comment out to disable TLS/PSK support in the broker and client. Requires
 # WITH_TLS=yes.
 # This must be disabled if using openssl < 1.0.
-WITH_TLS_PSK:=yes
+WITH_TLS_PSK:=no
 
 # Comment out to disable client threading support.
 WITH_THREADING:=yes
@@ -46,7 +46,7 @@ WITH_PERSISTENCE:=yes
 # Comment out to remove memory tracking support from the broker. If disabled,
 # mosquitto won't track heap memory usage nor export '$SYS/broker/heap/current
 # size', but will use slightly less memory and CPU time.
-WITH_MEMORY_TRACKING:=yes
+WITH_MEMORY_TRACKING:=no
 
 # Compile with database upgrading support? If disabled, mosquitto won't
 # automatically upgrade old database versions.
@@ -69,7 +69,7 @@ WITH_SRV:=no
 WITH_UUID:=yes
 
 # Build with websockets support on the broker.
-WITH_WEBSOCKETS:=no
+WITH_WEBSOCKETS:=yes
 
 # Use elliptic keys in broker
 WITH_EC:=yes
@@ -88,6 +88,7 @@ WITH_STATIC_LIBRARIES:=no
 
 # Build with async dns lookup support for bridges (temporary). Requires glibc.
 #WITH_ADNS:=yes
+WITH_ADNS=no
 
 # Build with epoll support.
 WITH_EPOLL:=yes
