diff --git a/doc/Makefile b/doc/Makefile
index c10ced6..524b5a2 100644
--- a/doc/Makefile
+++ b/doc/Makefile
@@ -28,7 +28,7 @@ default: $(SUBDIRS) CHANGES.gz
 include $(BUILDRULES)
 
 CHANGES.gz:
-	$(ZIP) --best -c < CHANGES > $@
+	$(ZIP) -c < CHANGES > $@
 
 install: default
 	$(SUBDIRS_MAKERULE)
diff --git a/include/buildmacros b/include/buildmacros
index 9e81bb8..b33e148 100644
--- a/include/buildmacros
+++ b/include/buildmacros
@@ -139,7 +139,7 @@ endif
 MAN_MAKERULE = \
 	@for f in *.[12345678] ""; do \
 		if test ! -z "$$f"; then \
-			$(ZIP) --best -c < $$f > $$f.gz; \
+			$(ZIP) -c < $$f > $$f.gz; \
 		fi; \
 	done
 
