--- a/sal/osl/unx/backtrace.h
+++ b/sal/osl/unx/backtrace.h
@@ -20,7 +20,7 @@
 #ifndef INCLUDED_SAL_OSL_UNX_BACKTRACE_H
 #define INCLUDED_SAL_OSL_UNX_BACKTRACE_H
 
-#if defined (LINUX)
+#if defined (LINUX) && defined (__GLIBC__)
 
 #include <execinfo.h>
 
--- a/sal/osl/unx/signal.cxx
+++ b/sal/osl/unx/signal.cxx
@@ -36,7 +36,7 @@
 
 #endif /* MACOSX */
 
-#ifdef LINUX
+#if defined(LINUX) && defined(__GLIBC__)
 #include <execinfo.h>
 #include <link.h>
 #define INCLUDE_BACKTRACE

--- a/sal/osl/unx/backtrace.c
+++ b/sal/osl/unx/backtrace.c
@@ -279,7 +279,7 @@ void backtrace_symbols_fd( void **buffer, int size, int fd )
     }
 }
 
-#elif !defined LINUX
+#elif !defined(__GLIBC__)
 
 int backtrace( void **buffer, int max_frames )
 {
