--- libreoffice-5.3.0.3/configure.ac.orig	2017-01-26 22:22:46.000000000 +0000
+++ libreoffice-5.3.0.3/configure.ac	2017-02-08 13:14:16.777547599 +0000
@@ -652,7 +652,7 @@
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|linux-musl*|k*bsd*-gnu*)
     test_gtk=yes
     build_gstreamer_1_0=yes
     build_gstreamer_0_10=yes
@@ -4170,7 +4170,7 @@
     esac
     ;;
 
-linux-gnu*)
+linux-gnu*|linux-musl*)
     COM=GCC
     USING_X11=TRUE
     OS=LINUX
@@ -7197,7 +7197,7 @@
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -9748,21 +9748,6 @@
 fi
 AC_SUBST(SYSTEM_LIBEXTTEXTCAT_DATA)
 
-dnl ***************************************
-dnl testing libc version for Linux...
-dnl ***************************************
-if test "$_os" = "Linux"; then
-    AC_MSG_CHECKING([whether libc is >= 2.1.1])
-    exec 6>/dev/null # no output
-    AC_CHECK_LIB(c, gnu_get_libc_version, HAVE_LIBC=yes; export HAVE_LIBC)
-    exec 6>&1 # output on again
-    if test "$HAVE_LIBC"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_ERROR([no, upgrade libc])
-    fi
-fi
-
 dnl =========================================
 dnl Check for the Windows  SDK.
 dnl =========================================
@@ -12775,7 +12760,7 @@
 
     case "$host_os" in
 
-    aix*|dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    aix*|dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != ""; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
