--- a/sal/osl/unx/thread.cxx
+++ b/sal/osl/unx/thread.cxx
@@ -252,7 +252,7 @@
     short             nFlags)
 {
     Thread_Impl* pImpl;
-#if defined OPENBSD || defined MACOSX || (defined LINUX && !ENABLE_RUNTIME_OPTIMIZATIONS)
+#if 1
     pthread_attr_t attr;
     size_t stacksize;
 #endif
@@ -268,7 +268,7 @@
 
     pthread_mutex_lock (&(pImpl->m_Lock));
 
-#if defined OPENBSD || defined MACOSX || (defined LINUX && !ENABLE_RUNTIME_OPTIMIZATIONS)
+#if 1
     if (pthread_attr_init(&attr) != 0)
         return nullptr;
 
@@ -287,7 +287,7 @@
 
     if ((nRet = pthread_create (
         &(pImpl->m_hThread),
-#if defined OPENBSD || defined MACOSX || (defined LINUX && !ENABLE_RUNTIME_OPTIMIZATIONS)
+#if 1
         &attr,
 #else
         PTHREAD_ATTR_DEFAULT,
@@ -305,7 +305,7 @@
         return nullptr;
     }
 
-#if defined OPENBSD || defined MACOSX || (defined LINUX && !ENABLE_RUNTIME_OPTIMIZATIONS)
+#if 1
     pthread_attr_destroy(&attr);
 #endif
 
