--- libreoffice-7.2.2.2/external/skia/UnpackedTarball_skia.mk.orig	2021-08-10 19:27:48.000000000 +0200
+++ libreoffice-7.2.2.2/external/skia/UnpackedTarball_skia.mk	2021-08-12 18:09:22.813517161 +0200
@@ -41,6 +41,7 @@
     fix-warnings.patch.1 \
     disable-freetype-colrv1.1 \
     windows-libraries-system32.patch.1 \
+    skia-freetype2.11.patch \
 
 $(eval $(call gb_UnpackedTarball_set_patchlevel,skia,1))
 
