--- a/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/wrapper_gpl.cxx
@@ -216,9 +216,15 @@
                                                   : nullptr));
 #if POPPLER_CHECK_VERSION(22, 6, 0)
                     pDocUnique = std::unique_ptr<PDFDoc>(
+#if POPPLER_CHECK_VERSION(25, 10, 0)
+                        new PDFDoc(pFileName->copy(),
+                                   pOwnerPasswordStr ? std::optional<GooString>(pOwnerPasswordStr->toStr()) : std::nullopt,
+                                   pUserPasswordStr ? std::optional<GooString>(pUserPasswordStr->toStr()) : std::nullopt));
+#else
                         new PDFDoc(std::make_unique<GooString>(pFileName),
                                    std::optional<GooString>(pOwnerPasswordStr),
                                    std::optional<GooString>(pUserPasswordStr)));
+#endif
 #else
                     pDocUnique = std::unique_ptr<PDFDoc>(
                         new PDFDoc(pFileName, pOwnerPasswordStr, pUserPasswordStr));
