diff --git a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
index f12478cb2..fd5add558 100644
--- a/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
+++ b/sdext/source/pdfimport/xpdfwrapper/pdfioutdev_gpl.cxx
@@ -678,15 +678,16 @@ void PDFOutDev::updateLineDash(GfxState *state)
         return;
     assert(state);
 
-    double* dashArray; int arrayLen; double startOffset;
-    state->getLineDash(&dashArray, &arrayLen, &startOffset);
+    double startOffset;
+    const std::vector<double> &dashArray = state->getLineDash(&startOffset);
+    auto arrayLen = dashArray.size();
 
     printf( "updateLineDash" );
-    if( arrayLen && dashArray )
+    if( arrayLen > 0 )
     {
-        printf( " %f %d", normalize(startOffset), arrayLen );
-        for( int i=0; i<arrayLen; ++i )
-            printf( " %f", normalize(*dashArray++) );
+        printf( " %f %lu", normalize(startOffset), arrayLen );
+        for(auto &v : dashArray)
+            printf( " %f", normalize(v));
     }
     printf( "\n" );
 }
