diff --git a/tests/copy_cut_paste.cpp b/tests/copy_cut_paste.cpp
index ccd0158..60b3580 100644
--- a/tests/copy_cut_paste.cpp
+++ b/tests/copy_cut_paste.cpp
@@ -108,8 +108,8 @@ TEST_F(CopyCutPaste, given_source_has_offered_when_sink_gets_focus_it_sees_offer
 
     EXPECT_CALL(mdol, offer(_, StrEq(any_mime_type)));
     EXPECT_CALL(sink.listener, data_offer(_,_))
-        .WillOnce(Invoke([&](struct wl_data_device*, struct wl_data_offer* id)
-        { mdol.listen_to(id); }));
+        .WillOnce([&](struct wl_data_device*, struct wl_data_offer* id)
+        { mdol.listen_to(id); });
 
     sink.create_surface_with_focus();
 }
@@ -120,8 +120,8 @@ TEST_F(CopyCutPaste, given_sink_has_focus_when_source_makes_offer_sink_sees_offe
 
     EXPECT_CALL(mdol, offer(_, StrEq(any_mime_type)));
     EXPECT_CALL(sink.listener, data_offer(_,_))
-        .WillOnce(Invoke([&](struct wl_data_device*, struct wl_data_offer* id)
-        { mdol.listen_to(id); }));
+        .WillOnce([&](struct wl_data_device*, struct wl_data_offer* id)
+        { mdol.listen_to(id); });
 
     source.offer(any_mime_type);
 }
diff --git a/tests/gtk_primary_selection.cpp b/tests/gtk_primary_selection.cpp
index 9b12ab2..ccf2a35 100644
--- a/tests/gtk_primary_selection.cpp
+++ b/tests/gtk_primary_selection.cpp
@@ -163,7 +163,7 @@ TEST_F(GtkPrimarySelection, sink_can_listen)
 
     InSequence seq;
     EXPECT_CALL(device_listener, data_offer(_, _))
-        .WillOnce(Invoke([&](auto*, auto* id) { offer_listener.listen_to(id); }));
+        .WillOnce([&](auto*, auto* id) { offer_listener.listen_to(id); });
 
     EXPECT_CALL(offer_listener, offer(_, StrEq(any_mime_type)));
 
@@ -201,7 +201,7 @@ TEST_F(GtkPrimarySelection, source_sees_request)
 
     EXPECT_CALL(source_listener, send(_, _, _))
         .Times(1)
-        .WillRepeatedly(Invoke([&](auto*, auto*, int fd) { close(fd); }));
+        .WillRepeatedly([&](auto*, auto*, int fd) { close(fd); });
 
     Pipe pipe;
     gtk_primary_selection_offer_receive(device_listener.selected, any_mime_type, pipe.source);
diff --git a/tests/primary_selection.cpp b/tests/primary_selection.cpp
index 38ff88f..be40cc4 100644
--- a/tests/primary_selection.cpp
+++ b/tests/primary_selection.cpp
@@ -164,7 +164,7 @@ TEST_F(PrimarySelection, sink_can_listen)
 
     InSequence seq;
     EXPECT_CALL(device_listener, data_offer(_, _))
-        .WillOnce(Invoke([&](auto*, auto* id) { offer_listener.listen_to(id); }));
+        .WillOnce([&](auto*, auto* id) { offer_listener.listen_to(id); });
 
     EXPECT_CALL(offer_listener, offer(_, StrEq(any_mime_type)));
 
@@ -202,7 +202,7 @@ TEST_F(PrimarySelection, source_sees_request)
 
     EXPECT_CALL(source_listener, send(_, _, _))
         .Times(1)
-        .WillRepeatedly(Invoke([&](auto*, auto*, int fd) { close(fd); }));
+        .WillRepeatedly([&](auto*, auto*, int fd) { close(fd); });
 
     Pipe pipe;
     zwp_primary_selection_offer_v1_receive(device_listener.selected, any_mime_type, pipe.source);
diff --git a/tests/text_input_v2_with_input_method_v1.cpp b/tests/text_input_v2_with_input_method_v1.cpp
index 54168d1..881f450 100644
--- a/tests/text_input_v2_with_input_method_v1.cpp
+++ b/tests/text_input_v2_with_input_method_v1.cpp
@@ -140,7 +140,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, text_input_deactivates_context_on_disab
 
     bool is_deactivated = false;
     EXPECT_CALL(*this, deactivate(_))
-        .WillOnce(Invoke([&]{ is_deactivated = true; }));
+        .WillOnce([&]{ is_deactivated = true; });
     zwp_text_input_v2_disable(text_input, app_surface->wl_surface());
     zwp_text_input_v2_update_state(text_input, text_input.serial, 0);
     input_client_wait_for_app_client_roundtrip([&]() { return is_deactivated; });
@@ -155,7 +155,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, setting_surrounding_text_on_text_input_
     enable_text_input();
     bool is_triggered = false;
     EXPECT_CALL(*input_method_context, surrounding_text(text, cursor, anchor))
-        .WillOnce(Invoke([&]{ is_triggered = true; }));
+        .WillOnce([&]{ is_triggered = true; });
     zwp_text_input_v2_set_surrounding_text(text_input, text, cursor, anchor);
     zwp_text_input_v2_update_state(text_input, text_input.serial, 0);
 
@@ -169,7 +169,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_change_text)
     enable_text_input();
     bool has_been_committed = false;
     EXPECT_CALL(text_input, commit_string(text))
-        .WillOnce(Invoke([&]{ has_been_committed = true; }));
+        .WillOnce([&]{ has_been_committed = true; });
     zwp_input_method_context_v1_commit_string(
         *input_method_context, input_method_context->serial, text);
 
@@ -187,11 +187,11 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_delete_text)
 
     int callback_count = 0;
     EXPECT_CALL(text_input, commit_string(text))
-        .WillOnce(Invoke([&]{ callback_count++; }));
+        .WillOnce([&]{ callback_count++; });
     EXPECT_CALL(text_input, cursor_position(index, 0))
-        .WillOnce(Invoke([&]{ callback_count++; }));
+        .WillOnce([&]{ callback_count++; });
     EXPECT_CALL(text_input, delete_surrounding_text(0, length))
-        .WillOnce(Invoke([&]{ callback_count++; }));
+        .WillOnce([&]{ callback_count++; });
     input_client.roundtrip();
     zwp_input_method_context_v1_delete_surrounding_text(*input_method_context, index, length);
     zwp_input_method_context_v1_commit_string(*input_method_context, input_method_context->serial, text);
@@ -209,7 +209,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_send_keysym)
 
     bool has_received_keysym = false;
     EXPECT_CALL(text_input, keysym(time, sym, state, modifiers))
-        .WillOnce(Invoke([&]{ has_received_keysym = true; }));
+        .WillOnce([&]{ has_received_keysym = true; });
     zwp_input_method_context_v1_keysym(
         *input_method_context,
         input_method_context->serial,
@@ -226,7 +226,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_string)
 
     bool has_received_preedit_string = false;
     EXPECT_CALL(text_input, preedit_string(preedit_text, preedit_commit))
-        .WillOnce(Invoke([&]{ has_received_preedit_string = true; }));
+        .WillOnce([&]{ has_received_preedit_string = true; });
     zwp_input_method_context_v1_preedit_string(
         *input_method_context,
         input_method_context->serial,
@@ -247,7 +247,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_style)
 
     bool has_received_preedit_style = false;
     EXPECT_CALL(text_input, predit_styling(index, length, style))
-        .WillOnce(Invoke([&]{ has_received_preedit_style = true; }));
+        .WillOnce([&]{ has_received_preedit_style = true; });
     zwp_input_method_context_v1_preedit_styling(
         *input_method_context,
         index,
@@ -271,7 +271,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_preedit_cursor)
 
     bool has_received_preedit_cursor = false;
     EXPECT_CALL(text_input, preedit_cursor(index))
-        .WillOnce(Invoke([&]{ has_received_preedit_cursor = true; }));
+        .WillOnce([&]{ has_received_preedit_cursor = true; });
     zwp_input_method_context_v1_preedit_cursor(
         *input_method_context,
         index);
@@ -297,7 +297,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_modifiers_map)
 
     bool has_received_modifier_map = false;
     EXPECT_CALL(text_input, modifiers_map(_))
-        .WillOnce(Invoke([&]{ has_received_modifier_map = true; }));
+        .WillOnce([&]{ has_received_modifier_map = true; });
     zwp_input_method_context_v1_modifiers_map(
         *input_method_context,
         &map);
@@ -314,7 +314,7 @@ TEST_F(TextInputV2WithInputMethodV1Test, input_method_can_set_direction)
 
     bool has_received_text_direction = false;
     EXPECT_CALL(text_input, text_direction(direction))
-        .WillOnce(Invoke([&]{ has_received_text_direction = true; }));
+        .WillOnce([&]{ has_received_text_direction = true; });
     zwp_input_method_context_v1_text_direction(
         *input_method_context,
         input_method_context->serial,
diff --git a/tests/wlr_virtual_pointer_v1.cpp b/tests/wlr_virtual_pointer_v1.cpp
index a0d4c18..5f72046 100644
--- a/tests/wlr_virtual_pointer_v1.cpp
+++ b/tests/wlr_virtual_pointer_v1.cpp
@@ -126,7 +126,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_is_moved_client_sees_motion)
         wl_fixed_from_int(pointer_start_y + motion_y)));
 
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
 
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_motion(handle, 0, wl_fixed_from_int(motion_x), wl_fixed_from_int(motion_y));
@@ -146,7 +146,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_is_moved_multiple_times_client
 
     EXPECT_CALL(listener, motion(_, _, _)).Times(AnyNumber());
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillOnce([&]{ recieved_frame = true; });
     zwlr_virtual_pointer_v1_motion(handle, 0, wl_fixed_from_int(motion1_x), wl_fixed_from_int(motion1_y));
     zwlr_virtual_pointer_v1_frame(handle);
     send_client.roundtrip();
@@ -156,7 +156,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_is_moved_multiple_times_client
         wl_fixed_from_int(pointer_start_x + motion1_x + motion2_x),
         wl_fixed_from_int(pointer_start_y + motion1_y + motion2_y)));
     recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     zwlr_virtual_pointer_v1_motion(handle, 0, wl_fixed_from_int(motion2_x), wl_fixed_from_int(motion2_y));
     zwlr_virtual_pointer_v1_frame(handle);
     send_client.roundtrip();
@@ -167,7 +167,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_left_clicks_client_sees_button
 {
     EXPECT_CALL(listener, button(_, _, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED));
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -179,7 +179,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_left_releases_client_sees_butt
 {
     EXPECT_CALL(listener, button(_, _, _, _)).Times(AnyNumber());
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -189,7 +189,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_left_releases_client_sees_butt
 
     EXPECT_CALL(listener, button(_, _, BTN_LEFT, WL_POINTER_BUTTON_STATE_RELEASED));
     recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_RELEASED);
     zwlr_virtual_pointer_v1_frame(handle);
     send_client.roundtrip();
@@ -201,7 +201,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_given_multiple_button_presses_
     EXPECT_CALL(listener, button(_, _, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED));
     EXPECT_CALL(listener, button(_, _, BTN_MIDDLE, WL_POINTER_BUTTON_STATE_PRESSED));
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_MIDDLE, WL_POINTER_BUTTON_STATE_PRESSED);
@@ -214,7 +214,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_presses_and_releases_different
 {
     EXPECT_CALL(listener, button(_, _, _, _)).Times(AnyNumber());
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillRepeatedly(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AnyNumber()).WillRepeatedly([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_PRESSED);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -225,7 +225,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_presses_and_releases_different
     EXPECT_CALL(listener, button(_, _, BTN_LEFT, WL_POINTER_BUTTON_STATE_RELEASED));
     EXPECT_CALL(listener, button(_, _, BTN_RIGHT, WL_POINTER_BUTTON_STATE_PRESSED));
     recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_LEFT, WL_POINTER_BUTTON_STATE_RELEASED);
     zwlr_virtual_pointer_v1_button(handle, 0, BTN_RIGHT, WL_POINTER_BUTTON_STATE_PRESSED);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -242,7 +242,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_scrolls_client_sees_axis)
     EXPECT_CALL(listener, axis(_, WL_POINTER_AXIS_VERTICAL_SCROLL, wl_fixed_from_int(5)));
     EXPECT_CALL(listener, axis_source(_)).Times(AnyNumber());
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_axis(handle, 0, WL_POINTER_AXIS_VERTICAL_SCROLL, wl_fixed_from_int(5));
     zwlr_virtual_pointer_v1_frame(handle);
@@ -256,7 +256,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_scrolls_with_steps_client_sees
     EXPECT_CALL(listener, axis_discrete(WL_POINTER_AXIS_HORIZONTAL_SCROLL, 4));
     EXPECT_CALL(listener, axis_source(_)).Times(AnyNumber());
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_axis_discrete(handle, 0, WL_POINTER_AXIS_HORIZONTAL_SCROLL, wl_fixed_from_int(5), 4);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -269,7 +269,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_specifies_axis_source_client_s
     EXPECT_CALL(listener, axis(_, _, _)).Times(AnyNumber());
     EXPECT_CALL(listener, axis_source(WL_POINTER_AXIS_SOURCE_CONTINUOUS));
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_axis(handle, 0, WL_POINTER_AXIS_VERTICAL_SCROLL, wl_fixed_from_int(5));
     zwlr_virtual_pointer_v1_axis_source(handle, WL_POINTER_AXIS_SOURCE_CONTINUOUS);
@@ -277,7 +277,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_specifies_axis_source_client_s
     send_client.roundtrip();
     receive_client.dispatch_until([&] { return recieved_frame; });
     auto axis_source = false;
-    EXPECT_CALL(listener, axis_source(WL_POINTER_AXIS_SOURCE_WHEEL)).WillOnce(Invoke([&]{ axis_source = true; }));
+    EXPECT_CALL(listener, axis_source(WL_POINTER_AXIS_SOURCE_WHEEL)).WillOnce([&]{ axis_source = true; });
     zwlr_virtual_pointer_v1_axis(handle, 0, WL_POINTER_AXIS_VERTICAL_SCROLL, wl_fixed_from_int(5));
     zwlr_virtual_pointer_v1_axis_source(handle, WL_POINTER_AXIS_SOURCE_WHEEL);
     zwlr_virtual_pointer_v1_frame(handle);
@@ -322,7 +322,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_is_moved_with_absolute_coordin
         wl_fixed_from_int(move_to_x),
         wl_fixed_from_int(move_to_y)));
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
 
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_motion_absolute(handle, 0, move_to_x, move_to_y, output_size.first, output_size.second);
@@ -348,7 +348,7 @@ TEST_F(VirtualPointerV1Test, when_virtual_pointer_is_moved_with_absolute_coordin
         wl_fixed_from_int(move_to_x),
         wl_fixed_from_int(move_to_y)));
     auto recieved_frame = false;
-    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce(Invoke([&]{ recieved_frame = true; }));
+    EXPECT_CALL(listener, frame()).Times(AtLeast(1)).WillOnce([&]{ recieved_frame = true; });
 
     auto const handle = zwlr_virtual_pointer_manager_v1_create_virtual_pointer(manager, nullptr);
     zwlr_virtual_pointer_v1_motion_absolute(
