--- a/baiacu.h
+++ b/baiacu.h
@@ -211,7 +211,7 @@ invalid:
 }
 #endif /* OpenBSD < 3.6, FreeBSD < 6.1, NetBSD < 8, !MidnightBSD, !DragonFly */
 
-#ifdef __linux__
+#if 0
 static inline char *fgetln(FILE *restrict f, size_t *lenp) {
 	/* Fail if we can't get access to stdio. */
 	if (pledge("stdio", NULL) == -1)
@@ -290,9 +290,8 @@ static inline void errc(int ecode, int wcode, char fmt[], ...) {
 		exit(ecode);
 }
 #endif
-#if !defined(__linux__) || defined(__GLIBC__)
 #include <err.h>
-#else
+#if 0
 #define vwarnx(fmt, ap) __vwarncx(fmt, "\n", ap)
 #define vwarn(fmt, ap) vwarnc(errno, fmt, ap)
 static inline void __dead warnx(char fmt[], ...) {
