--- a/core/src/xmake/hash/xxhash.c
+++ b/core/src/xmake/hash/xxhash.c
@@ -32,7 +32,7 @@
 #define XXH_NAMESPACE XM_
 #define XXH_STATIC_LINKING_ONLY
 #define XXH_IMPLEMENTATION
-#include "xxhash/xxhash.h"
+#include "xxhash.h"
 
 /* //////////////////////////////////////////////////////////////////////////////////////
  * implementation
--- a/core/src/xmake/semver/prefix.h
+++ b/core/src/xmake/semver/prefix.h
@@ -25,7 +25,7 @@
  * includes
  */
 #include "../prefix.h"
-#include "semver.h"
+#include "sv/semver.h"
 
 /* //////////////////////////////////////////////////////////////////////////////////////
  * extern
