From 40759432575c5769170dbd939811a42f48756ba5 Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Thu, 3 Jul 2025 21:55:14 +0200
Subject: [PATCH] don't require debuginfod

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index cfcf36a..a6dec5c 100644
--- a/setup.py
+++ b/setup.py
@@ -218,7 +218,6 @@ BINARY_FORMAT = BINARY_FORMATS.get(sys.platform, "elf")
 library_flags = {"libraries": ["lz4"]}
 if IS_LINUX:
     library_flags["libraries"].append("unwind")
-    library_flags["libraries"].append("debuginfod")
 
 try:
     library_flags = pkgconfig.parse(
-- 
2.50.0

