--- a/ChezScheme/s/machine.zuo
+++ b/ChezScheme/s/machine.zuo
@@ -20,11 +20,7 @@
      ;; OpenBSD 5.5 and later:
      64]
     [else
-     (cond
-       [(string=? arch "i3") 32]
-       [(string=? arch "ppc32") 32]
-       [(string=? arch "arm32") 32]
-       [else 64])]))
+     64]))  ;; XXX-Patched: time_t is always 64 bits on musl
 
 ;; Get or synthesize a "machine.def" file and supporting files,
 ;; returns
