--- a/config
+++ b/config
@@ -1,15 +1,15 @@
 # Driver (leave uncommented ONLY the line with the name of the driver)
 #T= mysql
 #T= oci8
-#T= odbc
+T= odbc
 #T= postgres
 #T= sqlite
-T=sqlite3
+#T=sqlite3
 
 # Installation directories
 
 # Default prefix
-PREFIX = /usr/local
+PREFIX = /usr
 
 # System's libraries directory (where binary libraries are installed)
 LUA_LIBDIR= $(PREFIX)/lib/lua/5.1
@@ -45,11 +45,11 @@
 #DRIVER_LIBS= -lsqlite
 #DRIVER_INCS=
 ######## SQLite3 
-DRIVER_LIBS= -L/opt/local/lib -lsqlite3
-DRIVER_INCS= -I/opt/local/include
+#DRIVER_LIBS= -L/opt/local/lib -lsqlite3
+#DRIVER_INCS= -I/opt/local/include
 ######## ODBC
-#DRIVER_LIBS= -L/usr/local/lib -lodbc
-#DRIVER_INCS= -DUNIXODBC -I/usr/local/include
+DRIVER_LIBS= -L/usr/local/lib -lodbc
+DRIVER_INCS= -DUNIXODBC -I/usr/local/include
 
 WARN= -Wall -Wmissing-prototypes -Wmissing-declarations -ansi -pedantic
 INCS= -I$(LUA_INC)
