--- a/src/ls_odbc.c
+++ b/src/ls_odbc.c
@@ -150,6 +150,7 @@
         case SQL_TYPE_DATE: case SQL_TYPE_TIME: case SQL_TYPE_TIMESTAMP: 
         case SQL_DATE: case SQL_INTERVAL: case SQL_TIMESTAMP: 
         case SQL_LONGVARCHAR:
+        case SQL_WCHAR: case SQL_WVARCHAR: case SQL_WLONGVARCHAR:
             return "string";
         case SQL_BIGINT: case SQL_TINYINT: case SQL_NUMERIC: 
         case SQL_DECIMAL: case SQL_INTEGER: case SQL_SMALLINT: 
