--- a/src/base/env_utils.hpp
+++ b/src/base/env_utils.hpp
@@ -21,6 +21,7 @@
 #define BUILDCACHE_ENV_UTILS_HPP_
 
 #include <string>
+#include <cstdint>
 
 namespace bcache {
 /// @brief A helper class for reading and parsing environment variables.
--- a/src/base/hmac.cpp
+++ b/src/base/hmac.cpp
@@ -45,6 +45,7 @@
 #include <array>
 #include <stdexcept>
 #include <vector>
+#include <cstdint>
 
 namespace bcache {
 
