readDNSRecord is undefined. Use correct function name in tests.

--- a/mtasts/mtasts_test.go
+++ b/mtasts/mtasts_test.go
@@ -57,7 +57,7 @@ func TestReadDNSRecord(t *testing.T) {
 
 	for _, c := range cases {
 		t.Run(c.value, func(t *testing.T) {
-			id, err := readDNSRecord(c.value)
+			id, err := ReadDNSRecord(c.value)
 			if c.fail {
 				if err == nil {
 					t.Errorf("expected failure for %v, but got with id=%v", c.value, id)
