--- mozilla-2.0.orig/browser/base/content/browser.xul
+++ mozilla-2.0/browser/base/content/browser.xul
@@ -992,12 +992,18 @@
                   contentcontextmenu="contentAreaContextMenu"
                   autocompletepopup="PopupAutoComplete"
                   onclick="return contentAreaClick(event, false);"/>
-      <statuspanel id="statusbar-display" label=""/>
     </vbox>
     <vbox id="browser-border-end" hidden="true" layer="true"/>
   </hbox>
 
   <vbox id="browser-bottombox" layer="true">
+    <statusbar id="status-bar" ordinal="1000"
+#ifdef WINCE
+               hidden="true"
+#endif
+               class="chromeclass-status">
+      <statusbarpanel id="statusbar-display" label=""/>
+    </statusbar>
     <toolbar id="addon-bar"
              toolbarname="&addonBarCmd.label;" accesskey="&addonBarCmd.accesskey;"
              collapsed="true"
@@ -1005,13 +1011,12 @@
              context="toolbar-context-menu" toolboxid="navigator-toolbox"
              mode="icons" iconsize="small" defaulticonsize="small"
              lockiconsize="true"
-             defaultset="addonbar-closebutton,spring,status-bar"
+             defaultset="addonbar-closebutton,spring"
              customizable="true"
              key="key_toggleAddonBar">
       <toolbarbutton id="addonbar-closebutton"
                      tooltiptext="&addonBarCloseButton.tooltip;"
                      oncommand="setToolbarVisibility(this.parentNode, false);"/>
-      <statusbar id="status-bar" ordinal="1000"/>
     </toolbar>
   </vbox>
 
