diff --git a/libvalhalla.pc.in b/libvalhalla.pc.in
index 886b7fdf93ff..27fbf52d305d 100644
--- a/libvalhalla.pc.in
+++ b/libvalhalla.pc.in
@@ -1,12 +1,12 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
 Name: libvalhalla
 Description: valhalla c++ library
 Version: @VERSION@
-Libs: -L${libdir} -lvalhalla
+Libs: -L${libdir} -lvalhalla -lprotobuf-lite -pthread -lcurl -lz
 Libs.private: @LIBS_PRIVATE@
 Requires: @REQUIRES@
 Cflags: -I${includedir} @CFLAGS@
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c4238fb7ded6..f350978b7c4f 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -215,13 +215,12 @@ set_target_properties(valhalla PROPERTIES
 
 # pkg-config installation
 if(PKG_CONFIG_FOUND)
-  include(ValhallaPkgConfig)
-
-  configure_valhalla_pc()
-
-  install(FILES
-    ${CMAKE_BINARY_DIR}/libvalhalla.pc
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
+  configure_file(
+    "../libvalhalla.pc.in"
+    "${CMAKE_CURRENT_BINARY_DIR}/libvalhalla.pc"
+    @ONLY)
+  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libvalhalla.pc
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
 
 

