diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index c4238fb7ded6..3bc131ff9e7e 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -240,11 +240,6 @@ if(BUILD_SHARED_LIBS)
     DESTINATION "${CMAKE_INSTALL_LIBDIR}"
     COMPONENT development)
 
-  if(UNIX)
-    add_custom_target(${LIBVALHALLA_SO_LINK} ALL
-      COMMAND ${CMAKE_COMMAND} -E create_symlink "${LIBVALHALLA_SO_LINK}.${VALHALLA_VERSION_MAJOR}" ${LIBVALHALLA_SO_LINK})
-    set_target_properties(${LIBVALHALLA_SO_LINK} PROPERTIES FOLDER "Library")
-  endif()
 endif()
 
 install(FILES ${VALHALLA_SOURCE_DIR}/COPYING ${VALHALLA_SOURCE_DIR}/CHANGELOG.md ${VALHALLA_SOURCE_DIR}/README.md

