--- a/es-core/src/ThemeData.cpp
+++ b/es-core/src/ThemeData.cpp
@@ -538,7 +538,7 @@
 
 	static const size_t pathCount = 2;
 	fs::path paths[pathCount] = { 
-		"/etc/emulationstation/themes", 
+		"/usr/share/emulationstation/themes", 
 		getHomePath() + "/.emulationstation/themes" 
 	};
 
--- a/es-app/src/SystemData.cpp
+++ b/es-app/src/SystemData.cpp
@@ -397,7 +397,7 @@
 	if(forWrite || fs::exists(filePath))
 		return filePath.generic_string();
 
-	return "/etc/emulationstation/gamelists/" + mName + "/gamelist.xml";
+	return "/usr/share/emulationstation/gamelists/" + mName + "/gamelist.xml";
 }
 
 std::string SystemData::getThemePath() const
