diff --git a/os.h b/os.h
index 2a1c2ca6f..53bf09e50 100644
--- a/os.h
+++ b/os.h
@@ -250,17 +250,13 @@ extern int errno;
 #endif
 
 #if defined(UTMPOK) || defined(BUGGYGETLOGIN)
-# if defined(SVR4) && !defined(DGUX) && !defined(__hpux) && !defined(linux)
+# if defined(linux)
 #  include <utmpx.h>
 #  define UTMPFILE	UTMPX_FILE
 #  define utmp		utmpx
-#  define getutent	getutxent
 #  define getutid	getutxid
 #  define getutline	getutxline
 #  define pututline	pututxline
-#  define setutent	setutxent
-#  define endutent	endutxent
-#  define ut_time	ut_xtime
 # else /* SVR4 */
 #  include <utmp.h>
 # endif /* SVR4 */
diff --git a/utmp.c b/utmp.c
index 1fb360368..abf4e6d04 100644
--- a/utmp.c
+++ b/utmp.c
@@ -891,7 +891,7 @@ xpututline(u)
 struct utmp *u;
 {
   struct utmp *u2;
-  pututline(u);
+  pututxline(u);
   setutent();
   u2 = getutline(u);
   if (u2 == 0)
