--- screen-4.6.2/os.h.old	2018-09-27 23:11:11.655480028 -0700
+++ screen-4.6.2/os.h	2018-09-27 23:11:34.745808665 -0700
@@ -250,17 +250,13 @@
 #endif
 
 #if defined(UTMPOK) || defined(BUGGYGETLOGIN)
-# if defined(SVR4) && !defined(DGUX) && !defined(__hpux) && !defined(linux)
+# if defined(linux)
 #  include <utmpx.h>
 #  define UTMPFILE	UTMPX_FILE
 #  define utmp		utmpx
-#  define getutent	getutxent
 #  define getutid	getutxid
 #  define getutline	getutxline
 #  define pututline	pututxline
-#  define setutent	setutxent
-#  define endutent	endutxent
-#  define ut_time	ut_xtime
 # else /* SVR4 */
 #  include <utmp.h>
 # endif /* SVR4 */
