diff --git a/t/run_test_fstrm_io_file.sh b/t/run_test_fstrm_io_file.sh
index 554fbe8..ee00985 100755
--- a/t/run_test_fstrm_io_file.sh
+++ b/t/run_test_fstrm_io_file.sh
@@ -9,7 +9,7 @@ fi
 
 for QUEUE_MODEL in SPSC MPSC; do
     for NUM_THREADS in 1 4 16; do
-        for NUM_MESSAGES in 1 1000 100000; do
+        for NUM_MESSAGES in 1 1000 10000; do
             $DIRNAME/$TNAME "$FILENAME" $QUEUE_MODEL $NUM_THREADS $NUM_MESSAGES
         done
     done
diff --git a/t/run_test_fstrm_io_tcp.sh b/t/run_test_fstrm_io_tcp.sh
index f8a14ba..c59e997 100755
--- a/t/run_test_fstrm_io_tcp.sh
+++ b/t/run_test_fstrm_io_tcp.sh
@@ -9,7 +9,7 @@ fi
 
 for QUEUE_MODEL in SPSC MPSC; do
     for NUM_THREADS in 1 4 16; do
-        for NUM_MESSAGES in 1 1000 100000; do
+        for NUM_MESSAGES in 1 1000 10000; do
             $DIRNAME/$TNAME $SOCKADDR $QUEUE_MODEL $NUM_THREADS $NUM_MESSAGES
         done
     done
diff --git a/t/run_test_fstrm_io_unix.sh b/t/run_test_fstrm_io_unix.sh
index ac3080e..cf6ce82 100755
--- a/t/run_test_fstrm_io_unix.sh
+++ b/t/run_test_fstrm_io_unix.sh
@@ -9,7 +9,7 @@ fi
 
 for QUEUE_MODEL in SPSC MPSC; do
     for NUM_THREADS in 1 4 16; do
-        for NUM_MESSAGES in 1 1000 100000; do
+        for NUM_MESSAGES in 1 1000 10000; do
             $DIRNAME/$TNAME "$SOCKNAME" $QUEUE_MODEL $NUM_THREADS $NUM_MESSAGES
         done
     done
