--- a/test/tester.tcl
+++ b/test/tester.tcl
@@ -1585,6 +1585,10 @@
     set col ""
     catch { set col $color($opcode) }
 
+    # For VCheck, P4 is a pointer to a Table object, printing
+    # it as a string results in an unprintable character
+    if {$opcode == "VCheck"} { set p4 [binary encode hex $p4] }
+
     output2 [format {%-4d  %s%s%-12.12s%s  %-6d  %-6d  %-6d  % -17s %s  %s} \
       $addr $I $col $opcode $D $p1 $p2 $p3 $p4 $p5 $comment
     ]
