--- a/autoconf/tea/teaish.tcl
+++ b/autoconf/tea/teaish.tcl
@@ -20,9 +20,9 @@ array set sqlite__Config [proj-strip-hash-comments {
 apply {{dir} {
   # Figure out the version number...
   set version ""
-  if {[file exists $dir/../VERSION]} {
+  if {[file exists $dir/../../VERSION]} {
     # The canonical SQLite TEA(ish) build
-    set version [proj-file-content -trim $dir/../VERSION]
+    set version [proj-file-content -trim $dir/../../VERSION]
     set ::sqlite__Config(is-canonical) 1
     set distname sqlite-tcl
   } elseif {[file exists $dir/generic/tclsqlite3.c]} {
