--- a/python/mozbuild/mozbuild/controller/building.py	2019-07-03 15:25:28.000000000 +0000
+++ b/python/mozbuild/mozbuild/controller/building.py	2019-07-17 02:49:12.693079588 +0000
@@ -940,7 +940,7 @@ class BuildDriver(MozbuildObject):
         warnings_path = self._get_state_filename('warnings.json')
         monitor = self._spawn(BuildMonitor)
         monitor.init(warnings_path)
-        ccache_start = monitor.ccache_stats()
+        ccache_start = None # monitor.ccache_stats()
         footer = BuildProgressFooter(self.log_manager.terminal, monitor)
 
         # Disable indexing in objdir because it is not necessary and can slow
@@ -1158,7 +1158,7 @@ class BuildDriver(MozbuildObject):
         if high_finder:
             print(FINDER_SLOW_MESSAGE % finder_percent)
 
-        ccache_end = monitor.ccache_stats()
+        ccache_end = None # monitor.ccache_stats()
 
         ccache_diff = None
         if ccache_start and ccache_end:
