From a2a15c826a5e1743ba71288543b9e144603fba26 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 25 Nov 2011 08:40:53 +0000
Subject: [PATCH 4/4] xulrunner: malloc_usable_size

---
 memory/mozalloc/mozalloc.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/memory/mozalloc/mozalloc.cpp b/memory/mozalloc/mozalloc.cpp
index 58e5192..500c651 100644
--- a/memory/mozalloc/mozalloc.cpp
+++ b/memory/mozalloc/mozalloc.cpp
@@ -267,7 +267,7 @@ moz_malloc_usable_size(void *ptr)
 #if defined(XP_MACOSX)
     return malloc_size(ptr);
 #elif defined(MOZ_MEMORY)
-    return malloc_usable_size(ptr);
+    return 0;
 #elif defined(XP_WIN)
     return _msize(ptr);
 #else
-- 
1.7.7.4

