--- ./mozilla/js/src/ctypes/CTypes.cpp.orig
+++ ./mozilla/js/src/ctypes/CTypes.cpp
@@ -467,7 +467,7 @@
 #ifdef WIN32
   return _finite(f) != 0;
 #else
-  return finite(f);
+  return isfinite(f);
 #endif
 }
 
--- ./mozilla/xpcom/ds/nsMathUtils.h
+++ ./mozilla/xpcom/ds/nsMathUtils.h
@@ -131,7 +131,11 @@ inline NS_HIDDEN_(bool) NS_finite(double d)
     // NOTE: '!!' casts an int to bool without spamming MSVC warning C4800.
     return !!_finite(d);
 #else
-    return finite(d);
+#ifdef _GLIBCXX_CMATH
+    return std::isfinite(d);
+#else
+    return isfinite(d);
+#endif
 #endif
 }
 
