--- a/toolkit/moz.configure.orig	2020-04-07 18:04:13.886159258 +0200
+++ b/toolkit/moz.configure	2020-04-07 18:04:37.905834036 +0200
@@ -1422,11 +1422,6 @@
             "with system libvpx. Use --without-system-libvpx to build "
             "with in-tree libvpx."))
 
-        check_symbol('vpx_codec_dec_init_ver', flags=vpx.libs, onerror=lambda: die(
-            "--with-system-libvpx requested but symbol vpx_codec_dec_init_ver "
-            "not found"
-        ))
-
         set_config('MOZ_SYSTEM_LIBVPX', True)
 
