--- ./lib/elixir/test/elixir/system_test.exs.orig
+++ ./lib/elixir/test/elixir/system_test.exs
@@ -63,7 +63,7 @@ defmodule SystemTest do
     assert {"hello\n", 0} = System.cmd "echo", ["hello"]
 
     opts = [into: [], cd: System.cwd!, env: %{"foo" => "bar"},
-            arg0: "hecho", stderr_to_stdout: true, parallelism: true]
+            arg0: "echo", stderr_to_stdout: true, parallelism: true]
     assert {["hello\n"], 0} = System.cmd "echo", ["hello"], opts
 
     with_tmp_dir(fn dir ->
@@ -73,7 +73,7 @@ defmodule SystemTest do
 
       File.cd!(dir)
       assert :enoent = catch_error(System.cmd("echo2", ["hello"]))
-      assert {"hello\n", 0} = System.cmd(Path.join([System.cwd, "echo2"]), ["hello"])
+      assert {"hello\n", 0} = System.cmd(Path.join([System.cwd, "echo2"]), ["hello"], [{:arg0, "echo"}])
     end)
   end
 
