diff --git a/src/meson.build b/src/meson.build
index d99036d..0a937f4 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -22,7 +22,7 @@ cargo_options = [ '--manifest-path', meson.project_source_root() / 'Cargo.toml'
 cargo_options += [ '--target-dir', meson.project_build_root() / 'src' ]
 
 if get_option('profile') == 'default'
-  cargo_options += [ '--release' ]
+  cargo_options += [ '--release', '--frozen' ]
   rust_target = 'release'
   message('Building in release mode')
 else
@@ -30,8 +30,6 @@ else
   message('Building in debug mode')
 endif
 
-cargo_env = [ 'CARGO_HOME=' + meson.project_build_root() / 'cargo-home' ]
-
 cargo_build = custom_target(
   'cargo-build',
   build_by_default: true,
@@ -43,8 +41,8 @@ cargo_build = custom_target(
   depends: resources,
   command: [
     'env',
-    cargo_env,
-    cargo, 'build',
+    [],
+    cargo, 'auditable', 'build',
     cargo_options,
     '&&',
     'cp', 'src' / rust_target / meson.project_name(), '@OUTPUT@',
