diff --git a/src/meson.build b/src/meson.build
index c3bc2f4..2e25d5d 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -19,7 +19,7 @@ run_command(
 )
 
 if get_option('profile') == 'default'
-  cargo_release_options = [ '--release' ]
+  cargo_release_options = [ '--release', '--frozen' ]
   cargo_clippy_deny_warnings = []
   rust_target = 'release'
   message('Building in release mode')
@@ -49,7 +49,7 @@ cargo_build = custom_target(
   depends: resources,
   env: cargo_env,
   command: [
-    cargo, 'build',
+    cargo, 'auditable', 'build',
     cargo_build_options,
     '&&',
     'cp', cargo_target / rust_target / meson.project_name(), '@OUTPUT@',
  )
diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -53,7 +53,6 @@ endif
 cargo_home = meson.project_build_root() / 'cargo-home'
 cargo_target = meson.project_build_root() / 'cargo-target'
 cargo_env = environment()
-cargo_env.set('CARGO_HOME', cargo_home)
 cargo_env.set('CARGO_TARGET_DIR', cargo_target)
 
 meson.add_devenv(cargo_env)
