diff --git a/ipython_genutils/tests/test_path.py b/ipython_genutils/tests/test_path.py
index 04ca85d..4a9e75d 100644
--- a/ipython_genutils/tests/test_path.py
+++ b/ipython_genutils/tests/test_path.py
@@ -49,3 +49,3 @@ class TestLinkOrCopy(object):
     def assert_inode_not_equal(self, a, b):
-        nt.assert_not_equals(os.stat(a).st_ino, os.stat(b).st_ino,
+        nt.assert_not_equal(os.stat(a).st_ino, os.stat(b).st_ino,
                              "%r and %r do reference the same indoes" %(a, b))
@@ -53,3 +53,3 @@ class TestLinkOrCopy(object):
     def assert_inode_equal(self, a, b):
-        nt.assert_equals(os.stat(a).st_ino, os.stat(b).st_ino,
+        nt.assert_equal(os.stat(a).st_ino, os.stat(b).st_ino,
                          "%r and %r do not reference the same indoes" %(a, b))
@@ -59,3 +59,3 @@ class TestLinkOrCopy(object):
             with open(b) as b_f:
-                nt.assert_equals(a_f.read(), b_f.read())
+                nt.assert_equal(a_f.read(), b_f.read())
 
