diff --git a/contrib/tests/shpproj.sh b/contrib/tests/shpproj.sh
index 81ea143..90c447a 100755
--- a/contrib/tests/shpproj.sh
+++ b/contrib/tests/shpproj.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 testdir="$(dirname "$(readlink -f $0)")"
 
diff --git a/tests/test2.sh b/tests/test2.sh
index 266fe54..01dadd5 100755
--- a/tests/test2.sh
+++ b/tests/test2.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 testdir="$(dirname "$(readlink -f $0)")"
 
diff --git a/tests/test3.sh b/tests/test3.sh
index 900e218..9111924 100755
--- a/tests/test3.sh
+++ b/tests/test3.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 #
 #	Use example programs to create a very simple dataset that
@@ -26,7 +26,7 @@ cd "$top_builddir"
 ) > "$testdir/s3.out"
 
 result=$(diff "$testdir/s3.out" "$testdir/stream3.out")
-if [ "$result" == "" ]; then
+if [ "$result" = "" ]; then
 	echo "******* Stream 3 Succeeded *********"
 	rm "$testdir/s3.out"
 	exit 0
