diff --git a/tests/_data/string_files/do_format_code.toml b/tests/_data/string_files/do_format_code.toml
index 5d886f3..794431e 100644
--- a/tests/_data/string_files/do_format_code.toml
+++ b/tests/_data/string_files/do_format_code.toml
@@ -43,3 +43,3 @@ outstring='''
     """docstring for CONST."""
-'''
+    '''
 
@@ -60,3 +60,3 @@ outstring='''
         """
-'''
+    '''
 
@@ -86,3 +86,3 @@ outstring='''
         """This is a second class variable docstring."""
-'''
+    '''
 
