Patch-Source: https://github.com/longxinH/xhprof/pull/66

diff --git a/extension/tests/xhprof_005.phpt b/extension/tests/xhprof_005.phpt
index 5188257..9c2429a 100644
--- a/tests/xhprof_005.phpt
+++ b/tests/xhprof_005.phpt
@@ -49,7 +49,7 @@ $output = xhprof_disable();
 
 function verify($expected, $actual, $description) {
 
-  echo "Verifying ${description}...\n";
+  echo "Verifying {$description}...\n";
 
   // 25% tolerance
   $range_low = ($expected * 0.75);
@@ -57,10 +57,10 @@ function verify($expected, $actual, $description) {
 
   if (($actual < $range_low) ||
       ($actual > $range_high)) {
-     echo "Failed ${description}. Expected: ${expected} microsecs. ".
-          "Actual: ${actual} microsecs.\n";
+     echo "Failed {$description}. Expected: {$expected} microsecs. ".
+          "Actual: {$actual} microsecs.\n";
   } else {
-     echo "OK: ${description}\n";
+     echo "OK: {$description}\n";
   }
   echo "-------------\n";
 }
