--- a/gettext-tools/tests/msgconv-2.old	2016-06-09 22:56:00.000000000 +0000
+++ b/gettext-tools/tests/msgconv-2	2017-08-11 17:54:55.677206381 +0000
@@ -7,6 +7,8 @@
 # Note: This test fails on Linux with musl libc versions that don't support
 # the BIG5 encoding in 'iconv'.
 
+Exit 77
+
 cat <<\EOF > mco-test2.po
 # Chinese translation for GNU gettext messages.
 #
--- a/gettext-tools/tests/msgmerge-compendium-6.old	2016-06-09 22:56:00.000000000 +0000
+++ b/gettext-tools/tests/msgmerge-compendium-6	2017-08-11 17:55:41.440408716 +0000
@@ -10,6 +10,8 @@
 # Note: This test fails on Linux with musl libc versions that don't support
 # the EUC-KR encoding in 'iconv'.
 
+Exit 77
+
 : ${MSGCONV=msgconv}
 ${MSGCONV} --to-code=UTF-8 -o mm-ko.utf-8.pot "$wabs_srcdir"/mm-ko.ascii.pot
 
--- a/gettext-tools/tests/xgettext-python-3.old	2016-06-09 22:56:00.000000000 +0000
+++ b/gettext-tools/tests/xgettext-python-3	2017-08-11 17:54:26.533956530 +0000
@@ -1,6 +1,8 @@
 #!/bin/sh
 . "${srcdir=.}/init.sh"; path_prepend_ . ../src
 
+Exit 77
+
 # Test Python support: --from-code option and encoding recognition.
 
 # Note: This test fails on Linux with musl libc versions that don't support
