--- syslog-summary-1.14/Makefile
+++ syslog-summary-1.14/Makefile.new
@@ -6,8 +6,12 @@
 version = $(shell grep ^version syslog-summary | cut -d\" -f2)
 
 install:
+	install -d $(DESTDIR)/usr/bin
+	install -d $(DESTDIR)/etc/syslog-summary
+	install -d $(DESTDIR)/usr/share/man/man1
 	install -m 755 syslog-summary $(DESTDIR)/usr/bin/syslog-summary
 	install -m 644 ignore.rules $(DESTDIR)/etc/syslog-summary/ignore.rules
+	install -m 644 syslog-summary.1 $(DESTDIR)/usr/share/man/man1
 
 uninstall:
 	[ ! -f $(DESTDIR)/usr/bin/syslog-summary ] || rm -v $(DESTDIR)/usr/bin/syslog-summary
