--- syslog-summary-1.14/syslog-summary
+++ syslog-summary-1.14/syslog-summary.new
@@ -43,6 +43,11 @@
 from optparse import OptionParser
 
 datepats = [
+       # 1st => runit / svlogd || svlogd -t || -tt || -ttt
+       re.compile(r"^[\@\.\-\:\_ A-Za-z0-9]{0,26}?(\w+[.]\w+): \w+[ :0-9]{12} "),
+       # 2nd & 3rd => digit dates / mysql error logs
+       re.compile(r"^([- 0-9]{10} {1,2}[:0-9]{7,8})([ 0-9]{16})? "),
+       re.compile(r"^(InnoDB:|Tcp|Time.*Argument|Version:|mysqld,) "),
 	re.compile(r"^(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) [ 0-9][0-9] [ 0-9][0-9]:[0-9][0-9]:[0-9][0-9] "),
 	re.compile(r"^(Mon|Tue|Wed|Thu|Fri|Sat|Sun) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) [ 0-9][0-9][0-9][0-9]:[0-9][0-9] "),
 	re.compile(r"^(Mon|Tue|Wed|Thu|Fri|Sat|Sun) (Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec) [ 0-9][0-9][0-9][0-9]:[0-9][0-9]:[0-9][0-9] "),
