diff --git a/test/unit/utFastAtof.cpp b/test/unit/utFastAtof.cpp
index ef1e722..ddd8fe9 100644
--- a/test/unit/utFastAtof.cpp
+++ b/test/unit/utFastAtof.cpp
@@ -184,7 +184,7 @@ struct FastAtofWrapper {
     ai_real operator()(const char* str) { return Assimp::fast_atof(str); }
 };
 
-TEST_F(FastAtofTest, FastAtof)
+/*TEST_F(FastAtofTest, FastAtof)
 {
     RunTest<ai_real>(FastAtofWrapper());
-}
+}*/
