From 895b93933ef885df69ceda763fb3d1616e8e4370 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Wed, 30 May 2012 16:24:38 +0300
Subject: [PATCH 2/2] ntp_sources: decrease maximum re-resolving delay

From approximately an hour to 7.5 minutes.
---
 ntp_sources.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ntp_sources.c b/ntp_sources.c
index 43a5d91..9c26bda 100644
--- a/ntp_sources.c
+++ b/ntp_sources.c
@@ -250,7 +250,7 @@ resolve_sources(void *arg)
 
   if (unresolved_sources) {
     /* Try again later */
-    if (resolving_interval < 9)
+    if (resolving_interval < 6)
       resolving_interval++;
     resolving_id = SCH_AddTimeoutByDelay(7 * (1 << resolving_interval), resolve_sources, NULL);
   } else {
-- 
1.7.10.2

