--- a/lib/gs-appstream.c
+++ b/lib/gs-appstream.c
@@ -587,8 +587,6 @@ gs_appstream_refine_add_provides (GsApp *app, XbNode *component, GError **error)
 					kind = AS_PROVIDED_KIND_FIRMWARE_RUNTIME;
 				else if (g_strcmp0 (fw_type, "flashed") == 0)
 					kind = AS_PROVIDED_KIND_FIRMWARE_FLASHED;
-			} else if (g_strcmp0 (element_name, "python2") == 0) {
-				kind = AS_PROVIDED_KIND_PYTHON_2;
 			} else if (g_strcmp0 (element_name, "python3") == 0) {
 				kind = AS_PROVIDED_KIND_PYTHON;
 			} else if (g_strcmp0 (element_name, "dbus") == 0) {
