diff --git a/tests/config/run-format-test.js b/tests/config/run-format-test.js
index 9bac63e82..54b1a27df 100644
--- a/tests/config/run-format-test.js
+++ b/tests/config/run-format-test.js
@@ -321,9 +321,9 @@ function runFormatTest(fixtures, parsers, options) {
           (currentParser === "acorn" && acornDisabledTests.has(filename)) ||
           (currentParser === "espree" && espreeDisabledTests.has(filename)) ||
           (currentParser === "meriyah" && meriyahDisabledTests.has(filename)) ||
-          (currentParser === "oxc" && oxcDisabledTests.has(filename)) ||
-          (currentParser === "oxc-ts" && oxcTsDisabledTests.has(filename)) ||
-          (currentParser === "hermes" && hermesDisabledTests.has(filename)) ||
+          (currentParser === "oxc") ||
+          (currentParser === "oxc-ts") ||
+          (currentParser === "hermes") ||
           (currentParser === "flow" && flowDisabledTests.has(filename)) ||
           (currentParser === "babel-ts" &&
             babelTsDisabledTests.has(filename)) ||
