Disable creation of rpath
---
diff --git a/SetPaths.cmake b/SetPaths.cmake
index 310a4a6..61bfd45 100644
--- a/SetPaths.cmake
+++ b/SetPaths.cmake
@@ -257,17 +257,5 @@ endif (NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
 
 
 
-# setup default RPATH/install_name handling, may be overridden by KDE4_HANDLE_RPATH_FOR_[LIBRARY|EXECUTABLE]
-# default is to build with RPATH for the install dir, so it doesn't need to relink
-if (UNIX)
-   if (APPLE)
-      set(CMAKE_INSTALL_NAME_DIR ${LIB_INSTALL_DIR})
-   else (APPLE)
-      # add our LIB_INSTALL_DIR to the RPATH and use the RPATH figured out by cmake when compiling
-      set(CMAKE_INSTALL_RPATH ${LIB_INSTALL_DIR} )
-      set(CMAKE_SKIP_BUILD_RPATH TRUE)
-      set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
-      set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-   endif (APPLE)
-endif (UNIX)
+

-- 
2.45.0
