--- a/bin/psh
+++ b/bin/psh
@@ -1,4 +1,4 @@
-#!bin/picolisp lib.l
+#!/usr/bin/picolisp /usr/lib/picolisp/lib.l
 # 29sep20 Software Lab. Alexander Burger
 
 (load "@lib/net.l" "@lib/misc.l" "@lib/http.l")
--- a/bin/pty
+++ b/bin/pty
@@ -1,4 +1,4 @@
-#!/data/data/com.termux/files/home/pil21/pil
+#!/usr/bin/pil
 # 10jan22abu
 # Pseudo Terminal (PilBox)
 
--- a/bin/watchdog
+++ b/bin/watchdog
@@ -1,4 +1,4 @@
-#!bin/picolisp lib.l
+#!/usr/bin/picolisp /usr/lib/picolisp/lib.l
 # 10oct20 Software Lab. Alexander Burger
 # Use: bin/watchdog <host> <port> <from> <to1> <to2> ..
 
--- a/misc/bigtest
+++ b/misc/bigtest
@@ -1,4 +1,4 @@
-#!bin/picolisp lib.l
+#!/usr/bin/picolisp /usr/lib/picolisp/lib.l
 # 18may16abu
 # misc/bigtest <seed>
 
