--- ndisc6-1.0.2.orig/src/Makefile.am
+++ ndisc6-1.0.2/src/Makefile.am
@@ -12,9 +12,8 @@
 	-DLOCALEDIR=\"$(localedir)\"
 AM_LIBADD = @top_builddir@/compat/libcompat.a
 
-mysuidexec_PROGRAMS = ndisc6 rdisc6 rltraceroute6
+mysuidexec_PROGRAMS = ndisc6 rdisc6
 myexec_PROGRAMS = tcptraceroute6
-myexec_SCRIPTS = tracert6
 bin_PROGRAMS = tcpspray name2addr
 bin_SCRIPTS = tcpspray6 dnssort addr2name
 noinst_HEADERS = gettime.h
@@ -41,15 +40,9 @@
 rdisc6_LDADD = $(LIBRT) $(AM_LIBADD)
 
 # traceroute6
-rltraceroute6_SOURCES = traceroute.c traceroute.h \
-			trace-tcp.c trace-udp.c trace-icmp.c
-rltraceroute6_LDADD = $(LIBRT) $(AM_LIBADD)
 tcptraceroute6_SOURCES = tcptraceroute.c
 tcptraceroute6_CPPFLAGS = $(AM_CPPFLAGS) \
 	-DRLTRACEROUTE6=\"`echo rltraceroute6 | sed '$(transform)'`\"
-
-tracert6: Makefile.am gen-alias
-	$(alias_verbose)$(gen_alias) tracert6 rltraceroute6 -I
 
 # tcpspray
 tcpspray_SOURCES = tcpspray.c
