--- otp_src_17.5-orig/lib/os_mon/ebin/os_mon.app
+++ otp_src_17.5-fixed/lib/os_mon/ebin/os_mon.app
@@ -21,7 +21,7 @@
    [{description, "CPO  CXC 138 46"},
     {vsn, "2.3.1"},
     {modules, [os_mon, os_mon_mib, os_sup,
-               disksup, memsup, cpu_sup, os_mon_sysinfo, nteventlog]},
+               disksup, memsup, cpu_sup, os_mon_sysinfo]},
     {registered, [os_mon_sup, os_mon_sysinfo, disksup, memsup, cpu_sup, 
                   os_sup_server]},
     {applications, [kernel, stdlib, sasl]},
