diff --git a/server/tests/Makefile.am b/server/tests/Makefile.am
index d7f7af9..aa47f27 100644
--- a/server/tests/Makefile.am
+++ b/server/tests/Makefile.am
@@ -54,7 +54,6 @@ check_PROGRAMS =				\
 	test-stream				\
 	test-agent-msg-filter			\
 	test-loop				\
-	test-qxl-parsing			\
 	test-stat-file				\
 	test-leaks				\
 	test-vdagent				\
@@ -62,7 +61,6 @@ check_PROGRAMS =				\
 	test-empty-success			\
 	test-channel				\
 	test-stream-device			\
-	test-listen				\
 	test-record				\
 	$(NULL)
 
@@ -146,6 +144,3 @@ endif
 
 EXTRA_DIST += video-encoders
 
-if HAVE_SASL
-check_PROGRAMS += test-sasl
-endif
