--- a/server/tests/Makefile.am
+++ b/server/tests/Makefile.am
@@ -57,7 +57,6 @@
 	test-vdagent				\
 	test-fail-on-null-core-interface	\
 	test-empty-success			\
-	test-channel				\
 	$(NULL)
 
 noinst_PROGRAMS =				\
--- a/server/tests/Makefile.in
+++ b/server/tests/Makefile.in
@@ -95,7 +95,7 @@
 	test-qxl-parsing$(EXEEXT) test-stat-file$(EXEEXT) \
 	test-leaks$(EXEEXT) test-vdagent$(EXEEXT) \
 	test-fail-on-null-core-interface$(EXEEXT) \
-	test-empty-success$(EXEEXT) test-channel$(EXEEXT) \
+	test-empty-success$(EXEEXT) \
 	$(am__EXEEXT_1)
 noinst_PROGRAMS = test-display-no-ssl$(EXEEXT) \
 	test-display-streaming$(EXEEXT) \
@@ -181,14 +181,6 @@
 	$(top_builddir)/server/libserver.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
-test_channel_SOURCES = test-channel.c
-test_channel_OBJECTS = test-channel.$(OBJEXT)
-test_channel_LDADD = $(LDADD)
-test_channel_DEPENDENCIES = libtest.a \
-	$(top_builddir)/spice-common/common/libspice-common.la \
-	$(top_builddir)/server/libserver.la $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
-	$(am__DEPENDENCIES_1)
 test_codecs_parsing_SOURCES = test-codecs-parsing.c
 test_codecs_parsing_OBJECTS = test-codecs-parsing.$(OBJEXT)
 test_codecs_parsing_LDADD = $(LDADD)
@@ -383,7 +375,7 @@
 SOURCES = $(libtest_stat1_a_SOURCES) $(libtest_stat2_a_SOURCES) \
 	$(libtest_stat3_a_SOURCES) $(libtest_stat4_a_SOURCES) \
 	$(libtest_a_SOURCES) $(spice_server_replay_SOURCES) \
-	test-agent-msg-filter.c test-channel.c test-codecs-parsing.c \
+	test-agent-msg-filter.c test-codecs-parsing.c \
 	test-display-no-ssl.c test-display-resolution-changes.c \
 	test-display-streaming.c test-display-width-stride.c \
 	test-empty-success.c test-fail-on-null-core-interface.c \
@@ -394,7 +386,7 @@
 DIST_SOURCES = $(libtest_stat1_a_SOURCES) $(libtest_stat2_a_SOURCES) \
 	$(libtest_stat3_a_SOURCES) $(libtest_stat4_a_SOURCES) \
 	$(libtest_a_SOURCES) $(spice_server_replay_SOURCES) \
-	test-agent-msg-filter.c test-channel.c test-codecs-parsing.c \
+	test-agent-msg-filter.c test-codecs-parsing.c \
 	test-display-no-ssl.c test-display-resolution-changes.c \
 	test-display-streaming.c test-display-width-stride.c \
 	test-empty-success.c test-fail-on-null-core-interface.c \
@@ -982,10 +974,6 @@
 	@rm -f test-agent-msg-filter$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_agent_msg_filter_OBJECTS) $(test_agent_msg_filter_LDADD) $(LIBS)
 
-test-channel$(EXEEXT): $(test_channel_OBJECTS) $(test_channel_DEPENDENCIES) $(EXTRA_test_channel_DEPENDENCIES) 
-	@rm -f test-channel$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(test_channel_OBJECTS) $(test_channel_LDADD) $(LIBS)
-
 test-codecs-parsing$(EXEEXT): $(test_codecs_parsing_OBJECTS) $(test_codecs_parsing_DEPENDENCIES) $(EXTRA_test_codecs_parsing_DEPENDENCIES) 
 	@rm -f test-codecs-parsing$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(test_codecs_parsing_OBJECTS) $(test_codecs_parsing_LDADD) $(LIBS)
@@ -1077,7 +1065,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/libtest_stat4_a-stat-test.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/replay.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-agent-msg-filter.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-channel.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-codecs-parsing.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-display-base.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/test-display-no-ssl.Po@am__quote@
@@ -1473,13 +1460,6 @@
 test-empty-success.log: test-empty-success$(EXEEXT)
 	@p='test-empty-success$(EXEEXT)'; \
 	b='test-empty-success'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-test-channel.log: test-channel$(EXEEXT)
-	@p='test-channel$(EXEEXT)'; \
-	b='test-channel'; \
 	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
