diff --git a/server/tests/meson.build b/server/tests/meson.build
index 09ba0f2..9833746 100644
--- a/server/tests/meson.build
+++ b/server/tests/meson.build
@@ -50,7 +50,6 @@ tests = [
   ['test-empty-success', true],
   ['test-channel', true],
   ['test-stream-device', true],
-  ['test-listen', true],
   ['test-record', true],
   ['test-display-no-ssl', false],
   ['test-display-streaming', false],
@@ -60,10 +59,6 @@ tests = [
   ['test-display-width-stride', false],
 ]
 
-if spice_server_has_sasl
-  tests += [['test-sasl', true]]
-endif
-
 if spice_server_has_smartcard == true
   tests += [['test-smartcard', true]]
 endif
