--- a/sample-config/nrpe.cfg.in
+++ b/sample-config/nrpe.cfg.in
@@ -35,15 +35,6 @@
 
 
 
-# PID FILE
-# The name of the file in which the NRPE daemon should write it's process ID
-# number.  The file is only written if the NRPE daemon is started by the root
-# user and is running in standalone mode.
-
-pid_file=@piddir@/nrpe.pid
-
-
-
 # PORT NUMBER
 # Port number we should wait for connections on.
 # NOTE: This must be a non-privileged port (i.e. > 1024).
@@ -75,8 +66,10 @@
 # You can either supply a username or a UID.
 #
 # NOTE: This option is ignored if NRPE is running under either inetd or xinetd
+# NOTE: OpenRC starts NRPE already as @nrpe_user@ user. You can change the user
+#       in /etc/conf.d/nrpe.
 
-nrpe_user=@nrpe_user@
+#nrpe_user=@nrpe_user@
 
 
 
@@ -85,8 +78,10 @@
 # You can either supply a group name or a GID.
 #
 # NOTE: This option is ignored if NRPE is running under either inetd or xinetd
+# NOTE: OpenRC starts NRPE already as @nrpe_group@ group. You can change the
+#       group in /etc/conf.d/nrpe.
 
-nrpe_group=@nrpe_group@
+#nrpe_group=@nrpe_group@
 
 
 
