diff --git a/Makefile b/Makefile
index d804874..c661181 100644
--- a/Makefile
+++ b/Makefile
@@ -9,9 +9,9 @@ PYTHON ?= python
 
 
 # flags
-LUA_LIBS := $(shell $(PKG_CONFIG) --libs lua 2>/dev/null || echo "-llua")
+LUA_LIBS := $(shell $(PKG_CONFIG) --libs lua5.3 2>/dev/null || echo "-llua")
 LIBS += -lm -lpthread $(LUA_LIBS) -ldl
-LUA_CFLAGS := $(shell $(PKG_CONFIG) --cflags lua 2>/dev/null || echo "-I/usr/include/lua")
+LUA_CFLAGS := $(shell $(PKG_CONFIG) --cflags lua5.3 2>/dev/null || echo "-I/usr/include/lua")
 CFLAGS += $(LUA_CFLAGS)
 
 

