From 7eed4bf5938cf912348362f4dc08844af536f740 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Bart=C5=82omiej=20Piotrowski?= <b@bpiotrowski.pl>
Date: Tue, 8 Sep 2015 11:39:42 +0200
Subject: [PATCH] config.sh: check for ncursesw6

---
 config.sh | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/config.sh b/config.sh
index a9f0081..916908c 100755
--- a/config.sh
+++ b/config.sh
@@ -110,7 +110,9 @@ check_pkg "stfl" || fail "stfl"
 if [ `uname -s` = "Darwin" ]; then
 	check_custom "ncurses5.4" "ncurses5.4-config" || fail "ncurses5.4"
 elif [ `uname -s` != "OpenBSD" ]; then
-	check_custom "ncursesw5" "ncursesw5-config" ||  fail "ncursesw"
+	check_custom "ncursesw5" "ncursesw5-config" || \
+		check_custom "ncursesw6" "ncursesw6-config" \
+		||  fail "ncursesw"
 fi
 check_ssl_implementation
 all_aboard_the_fail_boat
-- 
2.5.1

