--- /dev/null
+++ git-branchless-0.10.0/.cargo/config.toml
@@ -0,0 +1,2 @@
+[patch.crates-io]
+esl01-indexedlog = { path = "./vendor/esl01-indexedlog" }
--- git-branchless-0.10.0.orig/vendor/esl01-indexedlog/src/lock.rs
+++ git-branchless-0.10.0/vendor/esl01-indexedlog/src/lock.rs
@@ -132,10 +132,10 @@
 
         // Lock
         match (opts.exclusive, opts.non_blocking) {
-            (true, false) => file.lock_exclusive(),
-            (true, true) => file.try_lock_exclusive(),
-            (false, false) => file.lock_shared(),
-            (false, true) => file.try_lock_shared(),
+            (true, false) => fs2::FileExt::lock_exclusive(&file),
+            (true, true) => fs2::FileExt::try_lock_exclusive(&file),
+            (false, false) => fs2::FileExt::lock_shared(&file),
+            (false, true) => fs2::FileExt::try_lock_shared(&file),
         }
         .context(&path, || {
             format!(
