Upstream: No
Reason: Musl compatibility

--- a/lib/util/siglist.in	2019-10-10 11:32:54.000000000 -0500
+++ b/lib/util/siglist.in	2019-10-14 16:42:46.259938722 -0500
@@ -17,11 +17,12 @@
     EMT     EMT trap
     FPE     Floating point exception
     KILL    Killed
+# before UNUSED (musl defines them as the same number)
+    SYS     Bad system call
 # before BUS (Older Linux doesn't really have a BUS, but defines it to UNUSED)
     UNUSED  Unused
     BUS     Bus error
     SEGV    Memory fault
-    SYS     Bad system call
     PIPE    Broken pipe
     ALRM    Alarm clock
     TERM    Terminated
