diff --git a/src/importexport/mnx/CMakeLists.txt.orig b/src/importexport/mnx/CMakeLists.txt
index b287d2a..4dfa87e 100644
--- a/src/importexport/mnx/CMakeLists.txt.orig
+++ b/src/importexport/mnx/CMakeLists.txt
@@ -18,12 +18,11 @@ if(MUE_COMPILE_USE_SYSTEM_MNXDOM)
 else()
     FetchContent_Declare(
         mnxdom
-        GIT_REPOSITORY https://github.com/rpatters1/mnxdom.git
-        GIT_TAG        e7c947bf768caccf315426dcae0dfac02caf738b
+        SOURCE_DIR ${MNXDOM_SOURCE_DIR}
     )
     FetchContent_MakeAvailable(mnxdom)
     set(MNXDOM_LIB mnxdom)
-    message(STATUS "Using fetched mnxdom")
+    message(STATUS "Using vendored mnxdom source: ${MNXDOM_SOURCE_DIR}")
 endif()
 
 target_sources(iex_mnx PRIVATE
