diff --git a/CMakeLists.txt b/CMakeLists.txt
index 43139ec..05b3c6c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -222,13 +222,13 @@ endif(NOT BUILD_SHARED_LIBS)
 
 set(KDDockWidgets_QTQUICK ON CACHE BOOL "Build for QtQuick instead of QtWidgets" FORCE)
 set(KDDockWidgets_EXAMPLES OFF CACHE BOOL "Build the examples" FORCE)
-add_subdirectory(thirdparty/KDDockWidgets)
+add_subdirectory(thirdparty/KDDockWidgets EXCLUDE_FROM_ALL)
 
 ###########################################
 # Setup Packaging
 ###########################################
 
 if (OS_IS_LIN)
     include(SetupAppImagePackaging)
 endif(OS_IS_LIN)
 
 if (OS_IS_WIN)
