diff --git a/src/framework/audio/CMakeLists.txt b/src/framework/audio/CMakeLists.txt
index 8be5fe9..03a28fd 100644
--- a/src/framework/audio/CMakeLists.txt
+++ b/src/framework/audio/CMakeLists.txt
@@ -20,6 +20,12 @@
 
 set(MODULE audio)
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(lame lame IMPORTED_TARGET GLOBAL REQUIRED)
+pkg_check_modules(libopusenc libopusenc IMPORTED_TARGET GLOBAL REQUIRED)
+pkg_check_modules(opus opus IMPORTED_TARGET GLOBAL REQUIRED)
+pkg_check_modules(flac flac++ IMPORTED_TARGET GLOBAL REQUIRED)
+
 include(GetPlatformInfo)
 
 if (OS_IS_WIN)
@@ -264,11 +270,8 @@ if (MUE_ENABLE_AUDIO_EXPORT)
         ${CMAKE_CURRENT_LIST_DIR}/internal/soundtracks/soundtrackwriter.h
         )
 
-    add_subdirectory(${PROJECT_SOURCE_DIR}/thirdparty/lame lame)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/thirdparty/opusenc opusenc)
-    add_subdirectory(${PROJECT_SOURCE_DIR}/thirdparty/flac flac)
-
-    set(MODULE_LINK ${MODULE_LINK} lame opusenc flac)
+    set(MODULE_INCLUDE ${MODULE_INCLUDE} ${lame_INCLUDE_DIRS} ${opus_INCLUDE_DIRS} ${flac_INCLUDE_DIRS} ${libopusenc_INCLUDE_DIRS})
+    set(MODULE_LINK ${MODULE_LINK} ${lame_LIBRARIES} ${opus_LIBRARIES} ${flac_LIBRARIES} ${libopusenc_LIBRARIES})
 endif()
 
 if (ARCH_IS_X86_64)
diff --git a/src/framework/draw/CMakeLists.txt b/src/framework/draw/CMakeLists.txt
index c00ffa2..4df2cd3 100644
--- a/src/framework/draw/CMakeLists.txt
+++ b/src/framework/draw/CMakeLists.txt
@@ -83,9 +83,10 @@ else()
         ${CMAKE_CURRENT_LIST_DIR}/internal/qimagepainterprovider.h
         )
 
-    add_subdirectory(${THIRDPARTY_DIR}/freetype freetype)
-    set(MODULE_INCLUDE ${THIRDPARTY_DIR}/freetype/include)
-    set(MODULE_LINK freetype)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(freetype freetype2 IMPORTED_TARGET GLOBAL REQUIRED)
+    set(MODULE_INCLUDE ${freetype_INCLUDE_DIRS})
+    set(MODULE_LINK ${freetype_LIBRARIES})
 
 endif()
 
