diff --git a/CMakeLists.txt b/CMakeLists.txt
index b56a8c0..0b4b806 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,9 +8,9 @@ IF(COMMAND CMAKE_POLICY)
 	CMAKE_POLICY(SET CMP0005 NEW)
 	CMAKE_POLICY(SET CMP0003 NEW)
 	IF (CMAKE_MAJOR_VERSION GREATER 2)
-		CMAKE_POLICY(SET CMP0026 OLD)
+		CMAKE_POLICY(SET CMP0026 NEW)
 		CMAKE_POLICY(SET CMP0045 NEW)
-		CMAKE_POLICY(SET CMP0050 OLD)
+		CMAKE_POLICY(SET CMP0050 NEW)
 	ENDIF()
 ENDIF(COMMAND CMAKE_POLICY)
 
@@ -539,7 +539,8 @@ IF(WIN32 OR WIN64)
 	ADD_CUSTOM_TARGET(bin2res COMMAND g++ "\"${BIN2RES_CPP}\"" -o "\"${BIN2RES}\"" DEPENDS "${BIN2RES_CPP}")
 ELSE(WIN32 OR WIN64)
 	ADD_EXECUTABLE(bin2res buildtools/bin2res.cpp)
-	GET_TARGET_PROPERTY(BIN2RES bin2res LOCATION)
+	# CMP0026: `NEW` behavior for this policy is to not to allow reading the `LOCATION` properties from build-targets.
+	SET(BIN2RES $<TARGET_FILE:bin2res>)
 ENDIF(WIN32 OR WIN64)
 
 # we somehow have to make LMMS-binary depend on MOC-files
