From ac17ff1e4400b80ce0be64911bf1c4f1ac548c34 Mon Sep 17 00:00:00 2001
From: Shiz <hi@shiz.me>
Date: Sat, 29 Jun 2019 02:21:47 +0200
Subject: [PATCH] Core: add missing X11 includes

---
 Source/Core/DolphinNoGUI/PlatformX11.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/Core/DolphinNoGUI/PlatformX11.cpp b/Source/Core/DolphinNoGUI/PlatformX11.cpp
index af8993a4a58..dfa1c0729d3 100644
--- a/Source/Core/DolphinNoGUI/PlatformX11.cpp
+++ b/Source/Core/DolphinNoGUI/PlatformX11.cpp
@@ -18,6 +18,7 @@
 
 #include <X11/Xatom.h>
 #include <X11/Xlib.h>
+#include <X11/Xutil.h>
 #include <X11/keysym.h>
 #include "UICommon/X11Utils.h"
 #include "VideoCommon/RenderBase.h"
