diff --git a/libeantic/srcxx/renf_elem_class.cpp b/libeantic/srcxx/renf_elem_class.cpp
index 7966f98..a0dc8ff 100644
--- a/libeantic/srcxx/renf_elem_class.cpp
+++ b/libeantic/srcxx/renf_elem_class.cpp
@@ -12,6 +12,7 @@
 
 #include <iostream>
 #include <cassert>
+#include <gmp.h>
 #include <flint/fmpq.h>
 #include <cstdlib>
 #include <functional>
diff --git a/libeantic/test/fmpq_poly_extra/t-evaluate.c b/libeantic/test/fmpq_poly_extra/t-evaluate.c
index dde92fc..df2b843 100644
--- a/libeantic/test/fmpq_poly_extra/t-evaluate.c
+++ b/libeantic/test/fmpq_poly_extra/t-evaluate.c
@@ -201,7 +201,7 @@ int main(void)
         arf_clear(d);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c b/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c
index cfd71b1..5faee37 100644
--- a/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c
+++ b/libeantic/test/fmpq_poly_extra/t-set_str_pretty.c
@@ -148,7 +148,7 @@ int main(void)
 
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
     }
 
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-has_real_root.c b/libeantic/test/fmpz_poly_extra/t-has_real_root.c
index b4755bc..6b295fc 100644
--- a/libeantic/test/fmpz_poly_extra/t-has_real_root.c
+++ b/libeantic/test/fmpz_poly_extra/t-has_real_root.c
@@ -70,7 +70,7 @@ int main(void)
     }
 
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c b/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c
index 51ee1f5..2580a14 100644
--- a/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c
+++ b/libeantic/test/fmpz_poly_extra/t-isolate_real_roots.c
@@ -392,7 +392,7 @@ int main(void)
             }
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
 
         printf("PASS\n");
     }
diff --git a/libeantic/test/fmpz_poly_extra/t-num_real_roots.c b/libeantic/test/fmpz_poly_extra/t-num_real_roots.c
index 2993562..32d3e39 100644
--- a/libeantic/test/fmpz_poly_extra/t-num_real_roots.c
+++ b/libeantic/test/fmpz_poly_extra/t-num_real_roots.c
@@ -47,7 +47,7 @@ int main(void)
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c b/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c
index 3665c0d..dbcb714 100644
--- a/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c
+++ b/libeantic/test/fmpz_poly_extra/t-num_real_roots_0_1.c
@@ -44,7 +44,7 @@ int main(void)
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c b/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c
index 832e24f..e907ee8 100644
--- a/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c
+++ b/libeantic/test/fmpz_poly_extra/t-num_real_roots_upper_bound.c
@@ -59,7 +59,7 @@ int main(void)
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c b/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c
index efec2da..ad164e0 100644
--- a/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c
+++ b/libeantic/test/fmpz_poly_extra/t-num_real_roots_vca.c
@@ -80,7 +80,7 @@ int main(void)
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c b/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c
index bb34bc6..575fb6d 100644
--- a/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c
+++ b/libeantic/test/fmpz_poly_extra/t-positive_root_upper_bound_2exp.c
@@ -103,7 +103,7 @@ int main(void)
         fmpz_poly_clear(q);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c b/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c
index 41b6149..f147aea 100644
--- a/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c
+++ b/libeantic/test/fmpz_poly_extra/t-randtest_irreducible.c
@@ -48,6 +48,6 @@ int main(void)
         fmpz_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c b/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c
index 646cc2f..fcc25b8 100644
--- a/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c
+++ b/libeantic/test/fmpz_poly_extra/t-scale_0_1_fmpq.c
@@ -89,7 +89,7 @@ int main(void)
         _fmpq_vec_clear(v2, n);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c b/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c
index 8ccdc2b..51daa6b 100644
--- a/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c
+++ b/libeantic/test/fmpz_poly_extra/t-set_str_pretty.c
@@ -146,7 +146,7 @@ int main(void)
             fmpz_poly_clear(b);
         }
 
-        FLINT_TEST_CLEANUP(state)
+        FLINT_TEST_CLEAR(state)
     }
 
     return 0;
diff --git a/libeantic/test/fmpz_poly_extra/t-squarefree_part.c b/libeantic/test/fmpz_poly_extra/t-squarefree_part.c
index 503a55f..2d27541 100644
--- a/libeantic/test/fmpz_poly_extra/t-squarefree_part.c
+++ b/libeantic/test/fmpz_poly_extra/t-squarefree_part.c
@@ -67,6 +67,6 @@ int main(void)
 
     printf("PASS\n");
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf/t-init.c b/libeantic/test/renf/t-init.c
index 2d8c319..1bcd2e1 100644
--- a/libeantic/test/renf/t-init.c
+++ b/libeantic/test/renf/t-init.c
@@ -127,7 +127,7 @@ int main(void)
     check_init(p, -0.48296797364451214901, 1e-15, 3, 100, state);
 
     fmpq_poly_clear(p);
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/renf/t-init_nth_root_fmpq.c b/libeantic/test/renf/t-init_nth_root_fmpq.c
index 8b565b0..d1d595c 100644
--- a/libeantic/test/renf/t-init_nth_root_fmpq.c
+++ b/libeantic/test/renf/t-init_nth_root_fmpq.c
@@ -52,7 +52,7 @@ int main(void)
         fmpq_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     printf("PASS\n");
     return 0;
diff --git a/libeantic/test/renf/t-randtest.c b/libeantic/test/renf/t-randtest.c
index 80b2c9c..34b79e0 100644
--- a/libeantic/test/renf/t-randtest.c
+++ b/libeantic/test/renf/t-randtest.c
@@ -51,6 +51,6 @@ int main(void)
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf/t-real_embeddings.c b/libeantic/test/renf/t-real_embeddings.c
index 7846529..35adc1c 100644
--- a/libeantic/test/renf/t-real_embeddings.c
+++ b/libeantic/test/renf/t-real_embeddings.c
@@ -37,7 +37,7 @@ int main(void)
         for (i = 0; i < n; i++) renf_clear(nfemb + i);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-ceil.c b/libeantic/test/renf_elem/t-ceil.c
index 45c660e..ca333ba 100644
--- a/libeantic/test/renf_elem/t-ceil.c
+++ b/libeantic/test/renf_elem/t-ceil.c
@@ -219,7 +219,7 @@ int main(void)
         arb_clear(e);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-cmp.c b/libeantic/test/renf_elem/t-cmp.c
index db47ca2..9e7ee5e 100644
--- a/libeantic/test/renf_elem/t-cmp.c
+++ b/libeantic/test/renf_elem/t-cmp.c
@@ -292,6 +292,6 @@ int main(void)
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-cmp_fmpq.c b/libeantic/test/renf_elem/t-cmp_fmpq.c
index 47fee17..c5c4546 100644
--- a/libeantic/test/renf_elem/t-cmp_fmpq.c
+++ b/libeantic/test/renf_elem/t-cmp_fmpq.c
@@ -94,6 +94,6 @@ int main(void)
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-equal_fmpq.c b/libeantic/test/renf_elem/t-equal_fmpq.c
index 1272a81..a1e9d83 100644
--- a/libeantic/test/renf_elem/t-equal_fmpq.c
+++ b/libeantic/test/renf_elem/t-equal_fmpq.c
@@ -49,6 +49,6 @@ int main(void)
     fmpq_clear(d);
     renf_clear(nf);
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-floor.c b/libeantic/test/renf_elem/t-floor.c
index a80e79f..96bf984 100644
--- a/libeantic/test/renf_elem/t-floor.c
+++ b/libeantic/test/renf_elem/t-floor.c
@@ -264,7 +264,7 @@ int main(void)
     }
 
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff --git a/libeantic/test/renf_elem/t-gen.c b/libeantic/test/renf_elem/t-gen.c
index 86f865d..e569edf 100644
--- a/libeantic/test/renf_elem/t-gen.c
+++ b/libeantic/test/renf_elem/t-gen.c
@@ -60,7 +60,7 @@ int main(void)
         fmpq_clear(d);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     printf("PASS\n");
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-get_arb.c b/libeantic/test/renf_elem/t-get_arb.c
index 197da06..00730b4 100644
--- a/libeantic/test/renf_elem/t-get_arb.c
+++ b/libeantic/test/renf_elem/t-get_arb.c
@@ -76,7 +76,7 @@ int main(void)
         renf_elem_clear(a, nf);
     }
     renf_clear(nf);
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
 
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-get_cfrac.c b/libeantic/test/renf_elem/t-get_cfrac.c
index 1770085..3ace935 100644
--- a/libeantic/test/renf_elem/t-get_cfrac.c
+++ b/libeantic/test/renf_elem/t-get_cfrac.c
@@ -116,6 +116,6 @@ int main(void)
 
     fmpq_clear(d);
     
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renf_elem/t-get_str_pretty.c b/libeantic/test/renf_elem/t-get_str_pretty.c
index 0da1062..b8a11c4 100644
--- a/libeantic/test/renf_elem/t-get_str_pretty.c
+++ b/libeantic/test/renf_elem/t-get_str_pretty.c
@@ -113,7 +113,7 @@ int main(void)
     renf_elem_clear(a, nf);
     renf_clear(nf);
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff --git a/libeantic/test/renf_elem/t-is_rational.c b/libeantic/test/renf_elem/t-is_rational.c
index b7eb14f..bab58c1 100644
--- a/libeantic/test/renf_elem/t-is_rational.c
+++ b/libeantic/test/renf_elem/t-is_rational.c
@@ -67,7 +67,7 @@ int main(void)
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff --git a/libeantic/test/renf_elem/t-pow.c b/libeantic/test/renf_elem/t-pow.c
index 2ee2d4b..f8d2b36 100644
--- a/libeantic/test/renf_elem/t-pow.c
+++ b/libeantic/test/renf_elem/t-pow.c
@@ -52,6 +52,6 @@ int main(void)
         renf_clear(nf);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
diff --git a/libeantic/test/renfxx/t-ceil.cpp b/libeantic/test/renfxx/t-ceil.cpp
index 12e9007..1d1c6ce 100644
--- a/libeantic/test/renfxx/t-ceil.cpp
+++ b/libeantic/test/renfxx/t-ceil.cpp
@@ -76,7 +76,7 @@ int main(void)
         fmpq_poly_clear(p);
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
diff --git a/libeantic/test/renfxx/t-get.cpp b/libeantic/test/renfxx/t-get.cpp
index cf3e1a7..2d89906 100644
--- a/libeantic/test/renfxx/t-get.cpp
+++ b/libeantic/test/renfxx/t-get.cpp
@@ -37,7 +37,7 @@ int main(void)
         a.get_renf_elem();
     }
 
-    FLINT_TEST_CLEANUP(state)
+    FLINT_TEST_CLEAR(state)
     return 0;
 }
 
