diff --git a/libeantic/test/Makefile.am b/libeantic/test/Makefile.am
index ce01fb7..759f7c1 100644
--- a/libeantic/test/Makefile.am
+++ b/libeantic/test/Makefile.am
@@ -25,7 +25,6 @@ check_PROGRAMS =                                     \
     fmpz_poly_extra/t-scale_0_1_fmpq                 \
     fmpz_poly_extra/t-set_str_pretty                 \
     fmpz_poly_extra/t-squarefree_part                \
-    renf/t-init                                      \
     renf/t-init_nth_root_fmpq                        \
     renf/t-randtest                                  \
     renf/t-real_embeddings                           \
@@ -83,7 +82,6 @@ fmpz_poly_extra_t_randtest_irreducible_SOURCES = fmpz_poly_extra/t-randtest_irre
 fmpz_poly_extra_t_scale_0_1_fmpq_SOURCES = fmpz_poly_extra/t-scale_0_1_fmpq.c
 fmpz_poly_extra_t_set_str_pretty_SOURCES = fmpz_poly_extra/t-set_str_pretty.c
 fmpz_poly_extra_t_squarefree_part_SOURCES = fmpz_poly_extra/t-squarefree_part.c
-renf_t_init_SOURCES = renf/t-init.c
 renf_t_init_nth_root_fmpq_SOURCES = renf/t-init_nth_root_fmpq.c
 renf_t_randtest_SOURCES = renf/t-randtest.c
 renf_t_real_embeddings_SOURCES = renf/t-real_embeddings.c
