diff --git a/Core/FileSystems/FileSystem.h b/Core/FileSystems/FileSystem.h
index 9e64d16f..e98fb363 100644
--- a/Core/FileSystems/FileSystem.h
+++ b/Core/FileSystems/FileSystem.h
@@ -20,6 +20,7 @@
 #include <vector>
 #include <string>
 #include <cstring>
+#include <cstdint>
 #include <ctime>
 
 #include "Common.h"
diff --git a/ext/armips/Util/Util.h b/ext/armips/Util/Util.h
index d4916764..af77ed00 100644
--- a/ext/armips/Util/Util.h
+++ b/ext/armips/Util/Util.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
