Author: Jordi Mallach <jordi@debian.org>
Description: Pass $(ARCHOPTS) to standalone m68000 makefile.
Forwarded: no

Index: mame/src/devices/cpu/m68000/makefile
===================================================================
--- mame.orig/src/devices/cpu/m68000/makefile
+++ mame/src/devices/cpu/m68000/makefile
@@ -25,7 +25,7 @@ clean:
 	-@rm -f m68kops.*
 
 m68kmake.o: m68kmake.cpp
-	$(SILENT) $(CC) -x c++ -std=c++11 -o "$@" -c "$<"
+	$(SILENT) $(CC) -x c++ -std=c++11 $(ARCHOPTS) -o "$@" -c "$<"
 
 m68kmake$(EXE) : m68kmake.o
 	@echo Linking $@...
