diff --git a/Makefile b/Makefile
index 6ef0f71..075b12a 100644
--- a/Makefile
+++ b/Makefile
@@ -64,14 +64,13 @@ install: multitail
 	$(INSTALL_DIR) $(DESTDIR)$(DOCDIR)
 	$(INSTALL_EXEC) multitail $(DESTDIR)$(BINDIR)
 	$(INSTALL_DATA) multitail.1 $(DESTDIR)$(MAN1DIR)/multitail.1
-	$(INSTALL_DATA) *.txt INSTALL manual*.html $(DESTDIR)$(DOCDIR)
+	$(INSTALL_DATA) INSTALL manual*.html $(DESTDIR)$(DOCDIR)
 	#
 	### COPIED multitail.conf.new, YOU NEED TO REPLACE THE multitail.conf
 	### YOURSELF WITH THE NEW FILE
 	#
 	$(INSTALL_DIR) $(DESTDIR)$(CONFIG_DIR)
 	$(INSTALL_DATA) multitail.conf $(DESTDIR)$(CONFIG_FILE).new
-	$(INSTALL_EXEC) conversion-scripts/* $(DESTDIR)$(CONFIG_DIR)
 #rm -f $(DESTDIR)$(PREFIX)/share/man/man1/multitail.1.gz
 #gzip -9 $(DESTDIR)$(PREFIX)/share/man/man1/multitail.1
 	#
