diff --git a/mt.h b/mt.h
index e3f0542..411217f 100644
--- a/mt.h
+++ b/mt.h
@@ -59,21 +59,8 @@ typedef enum { SCHEME_TYPE_EDIT = 0, SCHEME_TYPE_FILTER } filter_edit_scheme_t;
 #define __USE_BSD	/* manpage says _BSD_SOURCE, stdlib.h says __USE_BSD */
 #endif
 
-#if defined(UTF8_SUPPORT) && !defined(__APPLE__)
-	#include <ncursesw/panel.h>
-	#include <ncursesw/ncurses.h>
-#else
-	#if defined(__APPLE__)
-        #include <ncurses.h>
-        #include <panel.h>
-    #elif defined(sun) || defined(__sun) || defined(scoos) || defined(_HPUX_SOURCE) || defined(AIX) || defined(__CYGWIN__)
-		#include <ncurses/panel.h>
-		#include <ncurses/ncurses.h>
-	#else
-		#include <panel.h>
-		#include <ncurses.h>
-	#endif
-#endif
+#include <panel.h>
+#include <ncurses.h>
 
 /* it seems the default HP-UX c-compiler doesn't understand atoll and
  * strtoll while it does understand 'long long'

