--- castero-0.9.5-origin/tests/test_display.py
+++ castero-0.9.5/tests/test_display.py
@@ -58,7 +58,7 @@
     assert display._footer_window.clear.call_count == 1
     assert display._footer_window.addstr.call_count == 2
     display._footer_window.addstr.assert_any_call(1, 0, "prompt")
-    assert display._footer_window.called_with(1, len("prompt"))
+    display._footer_window.addstr.assert_called_with(1, 0, "prompt")
 
 
 def test_display_get_y_n(display):
@@ -66,7 +66,7 @@
     assert display._footer_window.clear.call_count == 1
     assert display._footer_window.addstr.call_count == 2
     display._footer_window.addstr.assert_any_call(1, 0, "prompt")
-    assert display._footer_window.called_with(1, len("prompt"))
+    display._footer_window.addstr.assert_called_with(1, 0, "prompt")
 
 
 def test_display_input_keys(display):
