diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index cd9fe7c..0aeda05 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -2,16 +2,9 @@ ksutil_add_test(pools)
 ksutil_add_test(realloc)
 ksutil_add_test(acl)
 ksutil_add_test(threadpools)
-ksutil_add_test(threadmutex)
-ksutil_add_test(time)
-ksutil_add_test(q)
 ksutil_add_test(hash)
-ksutil_add_test(sock)
 ksutil_add_test(sock2)
 ksutil_add_test(websock)
 ksutil_add_test(websock2)
-ksutil_add_test(http)
 ksutil_add_test(string)
 ksutil_add_test(log)
-ksutil_add_test(json)
-ksutil_add_test(handle)
