diff -upr bvi-1.4.0.orig/comm.c bvi-1.4.0/comm.c
--- bvi-1.4.0.orig/comm.c	2016-07-25 10:04:18.054650413 +0200
+++ bvi-1.4.0/comm.c	2016-07-25 10:04:29.471264368 +0200
@@ -87,7 +87,7 @@ extern	int		numfiles, curfile;
 extern	int		errno;
 
 static	char	oldbuf[CMDSZ];		/** for :!! command **/
-
+int save_chk(char *fname, char *start, char *end, int flags);
 
 /*
  * docmdline() - handle a colon command
