Index: src/PDFDocument.cxx
===================================================================
--- src/PDFDocument.cxx	(revision 349)
+++ src/PDFDocument.cxx	(revision 350)
@@ -683,7 +683,7 @@
     //for get text we must exchange y coordinate, don't ask me where logic here.
     PopplerRectangle textRect = { rect->getX1() / getZoom(),
                                   (pageHeight - rect->getY2() / getZoom()),
-                                  rect->getX2() * getZoom() / getZoom(),
+                                  rect->getX2() / getZoom(),
                                   (pageHeight - rect->getY1() / getZoom())};
     repairEmpty(textRect);
 
