Index: src/PDFDocument.cxx
===================================================================
--- src/PDFDocument.cxx	(revision 350)
+++ src/PDFDocument.cxx	(revision 351)
@@ -167,6 +167,7 @@
                                        (height - matchRect->y1));
             results = g_list_prepend (results, rect);
         }
+        g_object_unref (G_OBJECT (page));
     }
 
     return g_list_reverse (results);
Index: THANKS
===================================================================
--- THANKS	(revision 350)
+++ THANKS	(revision 351)
@@ -16,6 +16,7 @@
     - Fridrich Strba <fridrich.strba@bluewin.ch>, for his Windows patch.
     - Ihar Hrachyshka <ihar.hrachyshka@gmail.com>, for his vim keybindings patch.
     - Igor Vagulin <igor_vagulin@mail.ru>, for his Russian translation and patches.
+    - Jim Ramsay <i.am@jimramsay.com>, for his patches.
     - Lester Godwin <lgodwin@pushcorp.com>, for his patches.
     - LoneFox <>, for his patches.
     - Michael Opitz <opitz.michael@gmail.com>, for his patches.
