--- a/etc/ldap.conf
+++ b/etc/ldap.conf
@@ -1,11 +1,9 @@
-# /etc/ldap.conf
-#
-# This is the configuration file for OpenSSH LDAP Public Keys (ssh-ldap-pubkey).
+# Configuration file for OpenSSH LDAP Public Keys (ssh-ldap-pubkey).
 #
 # This file actually uses a subset of directives from configuration file of the
 # LDAP nameservice switch library and the LDAP PAM module, so the same file can
-# be used for all these services.
-#
+# be used for all these services. However, neither of them is needed for using
+# ssh-ldap-pubkey!
 
 # Specifies the URI of the LDAP server to connect to. The URI scheme may be
 # ldap, or ldaps, specifying LDAP over TCP or SSL respectively. A port number
