diff -urN linux-5.4.orig/arch/mips/kvm/emulate.c linux-5.4/arch/mips/kvm/emulate.c
--- linux-5.4.orig/arch/mips/kvm/emulate.c	2020-06-04 09:01:01.953454087 -0600
+++ linux-5.4/arch/mips/kvm/emulate.c	2020-06-04 09:01:50.763579884 -0600
@@ -1716,7 +1716,8 @@
 
 	case lwu_op:
 		vcpu->mmio_needed = 1;	/* unsigned */
-		/* fall through */
+		run->mmio.len = 4;
+		break;
 #endif
 	case lw_op:
 		run->mmio.len = 4;
