--- whois-5.2.15/mkpasswd.c
+++ whois-5.2.15/mkpasswd.c.new
@@ -104,11 +104,9 @@
 #if defined FreeBSD
     { "nt",		"$3$",  0,	0,	0, "NT-Hash" },
 #endif
-#if defined HAVE_SHA_CRYPT
     /* http://people.redhat.com/drepper/SHA-crypt.txt */
     { "sha-256",	"$5$",	8,	16,	1, "SHA-256" },
     { "sha-512",	"$6$",	8,	16,	1, "SHA-512" },
-#endif
     /* http://www.crypticide.com/dropsafe/article/1389 */
     /*
      * Actually the maximum salt length is arbitrary, but Solaris by default
