diff -upr whois-5.4.0.orig/mkpasswd.c whois-5.4.0/mkpasswd.c
--- whois-5.4.0.orig/mkpasswd.c	2018-11-03 11:37:41.579032463 +0100
+++ whois-5.4.0/mkpasswd.c	2018-11-03 11:37:47.892515510 +0100
@@ -108,14 +108,12 @@ static const struct crypt_method methods
     { "bcrypt",		"$2b$", 22,	22,	2, "bcrypt" },
     { "bcrypt-a",	"$2a$", 22,	22,	2, "bcrypt (obsolete $2a$ version)" },
 #endif
-#if defined HAVE_SHA_CRYPT
     /* http://people.redhat.com/drepper/SHA-crypt.txt */
     { "sha512crypt",	"$6$",	8,	16,	1, "SHA-512" },
     { "sha256crypt",	"$5$",	8,	16,	1, "SHA-256" },
     /* compatibility aliases for mkpasswd versions < 5.4.0 */
     { "sha-256",	"$5$",	8,	16,	1, NULL },
     { "sha-512",	"$6$",	8,	16,	1, NULL },
-#endif
 #if (defined __SVR4 && defined __sun) || defined XCRYPT_VERSION_NUM
     { "sunmd5",		"$md5$", 8,	8,	1, "SunMD5" },
 #endif
