diff --git a/configure.cmake b/configure.cmake
index 172831c6..76813841 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -403,11 +403,6 @@ add_check_for_support(
    CONDITION NOT LINUX OR PkgConfig_FOUND
    MESSAGE "A pkg-config support is required."
 )
-add_check_for_support(
-   MODULES QtWebEngine QtPdf
-   CONDITION NOT LINUX OR TEST_glibc
-   MESSAGE "A suitable version >= 2.17 of glibc is required."
-)
 add_check_for_support(
    MODULES QtWebEngine QtPdf
    CONDITION NOT LINUX OR TEST_khr
