--- a/lib/protocol/protocol.go
+++ b/lib/protocol/protocol.go
Author: Jakob Borg <jakob@kastelo.net>
Date:   Wed Jun 6 09:59:33 2018 +0200

    lib/protocol: Correct block size calculation on 32 bit archs (fixes #4990) (#4991)
@@ -54,7 +54,7 @@
 func BlockSize(fileSize int64) int {
 	var blockSize int
 	for _, blockSize = range BlockSizes {
-		if fileSize < int64(DesiredPerFileBlocks*blockSize) {
+		if fileSize < DesiredPerFileBlocks*int64(blockSize) {
 			break
 		}
 	}
