--- a/src/block/Cargo.toml
+++ b/src/block/Cargo.toml
@@ -28 +28 @@
-zstd = { version = "0.9", default-features = false }
+zstd = { version = "0.9", default-features = false, features = ["pkg-config"] }
--- a/src/model/Cargo.toml
+++ b/src/model/Cargo.toml
@@ -31 +31 @@
-zstd = { version = "0.9", default-features = false }
+zstd = { version = "0.9", default-features = false, features = ["pkg-config"] }
