--- ./lib/gettext.h.orig	2013-11-18 08:27:03.257572285 +0000
+++ ./lib/gettext.h	2013-11-18 08:28:02.271566431 +0000
@@ -50,9 +50,9 @@
 /* Many header files from the libstdc++ coming with g++ 3.3 or newer include
    <libintl.h>, which chokes if dcgettext is defined as a macro.  So include
    it now, to make later inclusions of <libintl.h> a NOP.  */
-#if defined(__cplusplus) && defined(__GNUG__) && (__GNUC__ >= 3)
+#if defined(__cplusplus) && defined(__GNUG__) && (__GNUC__ >= 3) && !defined (__UCLIBC__)
 # include <cstdlib>
-# if (__GLIBC__ >= 2 && !defined __UCLIBC__) || _GLIBCXX_HAVE_LIBINTL_H
+# if (__GLIBC__ >= 2) || _GLIBCXX_HAVE_LIBINTL_H
 #  include <libintl.h>
 # endif
 #endif
