From 610820ba6acfbe7d0845b557bda2d999d3e22cd1 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Wed, 7 Feb 2024 12:50:43 +0000
Subject: [PATCH] fix install

---
 Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Makefile b/Makefile
index 8496e0f..ba632ed 100644
--- a/Makefile
+++ b/Makefile
@@ -33,9 +33,7 @@ install-lib-all: install-lib install-dev
 
 .PHONY: install-dirs
 install-dirs:
-	mkdir -p ${DESTDIR}${PREFIX}
 	mkdir -p ${DESTDIR}${PREFIX}/bin
-	mkdir -p ${DESTDIR}${PREFIX}/lib
 	mkdir -p ${DESTDIR}${PREFIX}/lib/pkgconfig
 	mkdir -p ${DESTDIR}${PREFIX}/include
 
-- 
2.42.0

