--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,12 +24,6 @@
 
 option(ADA_BENCHMARKS "Build benchmarks" OFF)
 
-if(BUILD_TESTING OR ADA_BENCHMARKS)
-  include(${PROJECT_SOURCE_DIR}/cmake/import.cmake)
-  import_dependency(simdjson simdjson/simdjson 0a3a00c95665cc1fda760e121ba8d442945ede13)
-  add_dependency(simdjson)
-endif()
-
 if (BUILD_TESTING)
   message(STATUS "The tests are enabled.")
   add_subdirectory(tests)
