diff --git a/SoftLayer/CLI/sshkey/add.py b/SoftLayer/CLI/sshkey/add.py
index 93b523c..fc6ec69 100644
--- a/SoftLayer/CLI/sshkey/add.py
+++ b/SoftLayer/CLI/sshkey/add.py
@@ -33,7 +33,7 @@ def cli(env, label, in_file, key, note):
     if key:
         key_text = key
     else:
-        with open(path.expanduser(in_file), 'rU', encoding="utf-8") as key_file:
+        with open(path.expanduser(in_file), 'r', encoding="utf-8") as key_file:
             key_text = key_file.read().strip()
             key_file.close()
 
