diff -urw src/eigen-eigen/unsupported/test/FFTW.cpp eigen-eigen/unsupported/test/FFTW.cpp
--- src/eigen-eigen/unsupported/test/FFTW.cpp	2018-12-11 17:57:55.000000000 +0000
+++ eigen-eigen/unsupported/test/FFTW.cpp	2019-10-06 22:32:27.239613362 +0000
@@ -8,6 +8,7 @@
 // with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 #include "main.h"
+#include <stdint.h>
 #include <unsupported/Eigen/FFT>
 
 template <typename T> 
