Add support for detecting more 64 bit systems
--- a/include/config.h	2025-02-10 13:42:28.000000000 +0100
+++ b/include/config.h	2025-03-14 08:17:31.256953674 +0100
@@ -171,7 +171,7 @@
 #define EXEC_TM_ROUND 20U
 
 /* 64bit arch MACRO */
-#if (defined(__x86_64__) || defined(__arm64__) || defined(__aarch64__))
+#if (defined(__x86_64__) || defined(__arm64__) || defined(__aarch64__) || defined(__powerpc64le__) || defined(__s390x__) || (__riscv_xlen == 64) || defined(__loongarch64))
   #define WORD_SIZE_64 1
 #endif
 
