--- ./configure.ac.orig	2013-02-28 07:05:33.024962268 +0000
+++ ./configure.ac	2013-02-28 07:13:27.005783738 +0000
@@ -40,7 +40,11 @@
 AC_CHECK_FUNCS([printf sprintf snprintf vsnprintf gettimeofday strndup strlcpy strlcat])
 AC_FUNC_STAT
 
-PKG_CHECK_MODULES([MOWGLI], [libmowgli >= 0.9.50], [], [AC_MSG_ERROR([Install Mowgli from http://atheme.org/project/mowgli.])])
+PKG_CHECK_MODULES([MOWGLI], [libmowgli-2], [],
+    [PKG_CHECK_MODULES([MOWGLI], [libmowgli >= 0.9.50], [],
+        [AC_MSG_ERROR([Install Mowgli from http://atheme.org/project/mowgli.])
+    ])
+])
 
 # Check for optional features.
 EXAMPLES_BUILD=""
