Description: Improve tests
Origin: backport, https://github.com/tiran/defusedxml/commit/4e6cea5f5bb44bf06dd30a723cf13334693c3150
Last-Update: 2024-12-08

diff --git a/tests.py b/tests.py
index 32cdc57..139151e 100644
--- a/tests.py
+++ b/tests.py
@@ -7,6 +7,7 @@ import unittest
 import warnings

 from xml.etree import ElementTree as orig_elementtree
+from xml.etree.ElementTree import Element as ElementBeforeImport
 from xml.sax.saxutils import XMLGenerator
 from xml.sax import SAXParseException
 from pyexpat import ExpatError
@@ -22,6 +23,8 @@ from defusedxml import (
 from defusedxml.common import PY3


+from xml.etree.ElementTree import Element as ElementAfterImport
+
 if sys.version_info < (3, 7):
     warnings.filterwarnings("once", category=DeprecationWarning)

@@ -49,7 +52,7 @@ except ImportError:
     lxml_warnings = None


-warnings.filterwarnings("error", category=DeprecationWarning, module=r"defusedxml\..*")
+warnings.filterwarnings("always", category=DeprecationWarning, module=r"defusedxml\..*")


 HERE = os.path.dirname(os.path.abspath(__file__))
@@ -198,10 +201,12 @@ class TestDefusedElementTree(BaseTests):
         return list(self.module.iterparse(source, **kwargs))

     def test_html_arg(self):
-        with self.assertRaises(DeprecationWarning):
-            ElementTree.XMLParse(html=0)
+        with warnings.catch_warnings(record=True) as w:
+            self.module.XMLParse(html=0)
+        self.assertEqual(len(w), 1)
+        self.assertIs(w[0].category, DeprecationWarning)
         with self.assertRaises(TypeError):
-            ElementTree.XMLParse(html=1)
+            self.module.XMLParse(html=1)

     def test_aliases(self):
         parser = self.module.DefusedXMLParser
@@ -223,6 +228,15 @@ class TestDefusedElementTree(BaseTests):
             self.assertIsInstance(e, orig_elementtree.ParseError)
             self.assertIsInstance(e, self.module.ParseError)

+    def test_etree_element(self):
+        tree = tree = self.module.parse(self.xml_simple)
+        root = tree.getroot()
+        root.append(orig_elementtree.Element("module-import"))
+        root.append(ElementBeforeImport("before-import"))
+        root.append(ElementAfterImport("after-import"))
+        s = orig_elementtree.tostring(root)
+        self.assertEqual(s.count(b"import"), 3, s)
+

 class TestDefusedcElementTree(TestDefusedElementTree):
     module = cElementTree
@@ -289,22 +303,16 @@ class TestDefusedSax(BaseTests):
     dtd_external_ref = True

     def parse(self, xmlfile, **kwargs):
-        if PY3:
-            result = io.StringIO()
-        else:
-            result = io.BytesIO()
-        handler = XMLGenerator(result)
-        self.module.parse(xmlfile, handler, **kwargs)
-        return result.getvalue()
+        with  io.StringIO() as result:
+            handler = XMLGenerator(result)
+            self.module.parse(xmlfile, handler, **kwargs)
+            return result.getvalue()

     def parseString(self, xmlstring, **kwargs):
-        if PY3:
-            result = io.StringIO()
-        else:
-            result = io.BytesIO()
-        handler = XMLGenerator(result)
-        self.module.parseString(xmlstring, handler, **kwargs)
-        return result.getvalue()
+        with io.StringIO() as result:
+            handler = XMLGenerator(result)
+            self.module.parseString(xmlstring, handler, **kwargs)
+            return result.getvalue()

     def test_exceptions(self):
         with self.assertRaises(EntitiesForbidden) as ctx:
@@ -329,6 +337,7 @@ class TestDefusedSax(BaseTests):
         self.assertEqual(repr(ctx.exception), msg)


+@unittest.skipUnless(lxml is not None, "test requires lxml")
 class TestDefusedLxml(BaseTests):
     module = lxml

@@ -490,9 +499,11 @@ class TestXmlRpc(DefusedTestCase):
             xmlrpc.unmonkey_patch()


+@unittest.skipUnless(gzip is not None, "test requires gzip")
 class TestDefusedGzip(DefusedTestCase):
     def get_gzipped(self, length):
         f = io.BytesIO()
+        self.addCleanup(f.close)
         gzf = gzip.GzipFile(mode="wb", fileobj=f)
         gzf.write(b"d" * length)
         gzf.close()
@@ -544,17 +555,19 @@ class TestDefusedGzip(DefusedTestCase):

 def test_main():
     suite = unittest.TestSuite()
-    suite.addTests(unittest.makeSuite(TestDefusedcElementTree))
-    suite.addTests(unittest.makeSuite(TestDefusedElementTree))
-    suite.addTests(unittest.makeSuite(TestDefusedMinidom))
-    suite.addTests(unittest.makeSuite(TestDefusedMinidomWithParser))
-    suite.addTests(unittest.makeSuite(TestDefusedPulldom))
-    suite.addTests(unittest.makeSuite(TestDefusedSax))
-    suite.addTests(unittest.makeSuite(TestXmlRpc))
-    if lxml is not None:
-        suite.addTests(unittest.makeSuite(TestDefusedLxml))
-    if gzip is not None:
-        suite.addTests(unittest.makeSuite(TestDefusedGzip))
+    cls = [
+        TestDefusedElementTree,
+        TestDefusedcElementTree,
+        TestDefusedMinidom,
+        TestDefusedMinidomWithParser,
+        TestDefusedPulldom,
+        TestDefusedSax,
+        TestDefusedLxml,
+        TestXmlRpc,
+        TestDefusedGzip,
+    ]
+    for c in cls:
+        suite.addTest(unittest.defaultTestLoader.loadTestsFromTestCase(c))
     return suite

