diff --git a/demo/zrtptestMulti.cpp b/demo/zrtptestMulti.cpp
index b1515a0..a7c072a 100644
--- a/demo/zrtptestMulti.cpp
+++ b/demo/zrtptestMulti.cpp
@@ -295,13 +295,13 @@ public:
                 if (zrxcbMulti != NULL) {
                     str = session->getMultiStrParams(&zrtpMaster);
                     zrxcbMulti->setMultiStrParams(str, zrtpMaster);
-                    fprintf(stderr, "Master (test r): %p\n", zrtpMaster);
+                    fprintf(stderr, "Master (test r): %p\n", (void*)zrtpMaster);
                     zrxcbMulti->start();
                 }
                 if (ztxcbMulti != NULL) {
                     str = session->getMultiStrParams(&zrtpMaster);
                     ztxcbMulti->setMultiStrParams(str, zrtpMaster);
-                    fprintf(stderr, "Master (test t): %p\n", zrtpMaster);
+                    fprintf(stderr, "Master (test t): %p\n", (void*)zrtpMaster);
                     ztxcbMulti->start();
                 }
                 if (sender) {
