--- pcmciautils-018/src/read-cis.c.orig	2014-11-06 10:07:20.363207536 +0600
+++ pcmciautils-018/src/read-cis.c	2014-11-06 10:08:37.143611772 +0600
@@ -51,7 +51,7 @@
 		/* Get indirect link from the MFC tuple */
 		read_cis(tuple->Flags.link_space,
 			       tuple->LinkOffset, 5, link);
-		ofs = *(u_int *)(link+1);
+		ofs = *(unsigned int *)(link+1);
 		tuple->Flags.space = (link[0] == CISTPL_MFC_ATTR);
 		/* Move to the next indirect link */
 		tuple->LinkOffset += 5;
--- pcmciautils-018/src/yacc_config.y.orig	2014-11-06 10:08:55.857046934 +0600
+++ pcmciautils-018/src/yacc_config.y	2014-11-06 10:09:10.543788999 +0600
@@ -40,7 +40,7 @@
 
 %union {
     char *str;
-    u_long num;
+    unsigned long num;
     struct adjust_list_t *adjust;
 }
 
