diff --git a/pvclock.c b/pvclock.c
index e918b97..249f134 100644
--- a/pvclock.c
+++ b/pvclock.c
@@ -6,6 +6,7 @@
 #include <linux/sched.h>
 #include <linux/sched/clock.h>
 #include <asm/msr.h>
+#include <asm/tsc.h>
 
 #include "pvclock.h"
 
diff --git a/vmm_clock_main.c b/vmm_clock_main.c
index 67ac296..47c3a51 100644
--- a/vmm_clock_main.c
+++ b/vmm_clock_main.c
@@ -131,8 +131,7 @@ static uint32_t vmm_detect(void)
 		return 0;
 
 	if (boot_cpu_has(X86_FEATURE_HYPERVISOR))
-		return hypervisor_cpuid_base("OpenBSDVMM58", 0);
-
+		return cpuid_base_hypervisor("OpenBSDVMM58", 0);
 	return 0;
 }
 
