diff --git a/tools/build/test/test_all.py b/tools/build/test/test_all.py
index 9efddc8..abf3dc5 100644
--- a/tools/build/test/test_all.py
+++ b/tools/build/test/test_all.py
@@ -206,8 +206,8 @@ tests = ["absolute_sources",
          "core_update_now",
          "core_variables_in_actions",
          "custom_generator",
-         "debugger",
-         "debugger-mi",
+#         "debugger",
+#         "debugger-mi",
          "default_build",
          "default_features",
 # This test is known to be broken itself.
