diff --git a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
index cbd6419..0c29df6 100644
--- a/tools/build/src/tools/python.jam
+++ b/tools/build/src/tools/python.jam
@@ -907,14 +907,14 @@ local rule configure ( version ? : cmd-or-prefix ? : includes * : libraries ? :
     # for a particular target OS as the default. This makes it so that we can
     # select a python interpreter with only knowledge of the target OS. And hence
     # can configure different Pythons based on the target OS only.
-    local toolset-requirements = [ toolset.requirements ] ;
-    local toolset-target-os-requirements
-        = [ property.evaluate-conditionals-in-context
-            [ $(toolset-requirements).raw ] : <target-os>$(target-os) ] ;
-    if ! <python> in $(toolset-target-os-requirements:G)
-    {
-        toolset.add-requirements <target-os>$(target-os):<python>$(version:E=default) ;
-    }
+#    local toolset-requirements = [ toolset.requirements ] ;
+#    local toolset-target-os-requirements
+#        = [ property.evaluate-conditionals-in-context
+#            [ $(toolset-requirements).raw ] : <target-os>$(target-os) ] ;
+#    if ! <python> in $(toolset-target-os-requirements:G)
+#    {
+#        toolset.add-requirements <target-os>$(target-os):<python>$(version:E=default) ;
+#    }
 
     # Register the right suffix for extensions.
     register-extension-suffix $(extension-suffix) : $(target-requirements) ;
