http://www.openwall.com/lists/musl/2018/08/10/3

diff --git a/boost/process/detail/posix/is_running.hpp b/boost/process/detail/posix/is_running.hpp
index 0d431a9..342be7d 100644
--- a/boost/process/detail/posix/is_running.hpp
+++ b/boost/process/detail/posix/is_running.hpp
@@ -17,7 +17,7 @@ namespace boost { namespace process { namespace detail { namespace posix {
 // This bit arrangement of status codes is not guaranteed by POSIX, but (according to comments in
 // the glibc <bits/waitstatus.h> header) is the same across systems in practice.
 constexpr int still_active = 0x7F;
-static_assert(!WIFEXITED(still_active) && !WIFSIGNALED(still_active), "Internal Error");
+static_assert(!WIFEXITED(still_active), "Internal Error");
 
 inline bool is_running(int code)
 {
