--- seamonkey-2.39/mozilla/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc.orig
+++ seamonkey-2.39/mozilla/toolkit/crashreporter/google-breakpad/src/common/stabs_reader.cc
@@ -34,7 +34,15 @@
 #include "common/stabs_reader.h"
 
 #include <assert.h>
-#include <stab.h>
+//#include <stab.h>
+//#define N_UNDF 0x0
+#define N_FUN 0x24
+#define N_SLINE 0x44
+#define N_SO 0x64
+#define N_LSYM 0x80
+#define N_BINCL 0x82
+#define N_SOL 0x84
+
 #include <string.h>
 
 #include <string>
