--- seamonkey-2.46/mozilla/extensions/spellcheck/hunspell/glue/hunspell_alloc_hooks.h.orig
+++ seamonkey-2.46/mozilla/extensions/spellcheck/hunspell/glue/hunspell_alloc_hooks.h
@@ -50,9 +50,9 @@
 #include "mozilla/mozalloc.h"
 #include "mozHunspellAllocator.h"
 
-#define malloc(size) HunspellAllocator::CountingMalloc(size)
-#define calloc(count, size) HunspellAllocator::CountingCalloc(count, size)
-#define free(ptr) HunspellAllocator::CountingFree(ptr)
-#define realloc(ptr, size) HunspellAllocator::CountingRealloc(ptr, size)
+//#define malloc(size) HunspellAllocator::CountingMalloc(size)
+//#define calloc(count, size) HunspellAllocator::CountingCalloc(count, size)
+//#define free(ptr) HunspellAllocator::CountingFree(ptr)
+//#define realloc(ptr, size) HunspellAllocator::CountingRealloc(ptr, size)
 
 #endif
