--- a/shell-rs/meson.build
+++ b/shell-rs/meson.build
@@ -4,7 +4,7 @@ if gobject_introspection_dep.found()
   gir_dir = shell_rs_root / 'gir-files'
 
   # required to make sure the bindings have stable formatting
-  cargo_fmt = find_program('cargo-fmt', required: true)
+  cargo_fmt = find_program('cargo-fmt', required: false)
   run_target('update-rust-bindings',
     command: [
       shell_rs_root / 'update-bindings.sh',
