diff --git a/meson.build b/meson.build
index 5aed415f5265..6c6b2159d77b 100644
--- a/meson.build
+++ b/meson.build
@@ -131,7 +131,7 @@ top_inc = include_directories('.')
 
 glib_req_version = '>= 2.75.0'
 gtk_req_version = '>= 4.17.1'
-libaw_req_version = '>= 1.8.alpha'
+libaw_req_version = '>= 1.8'
 exempi_req_version = '>= 2.0'
 
 gdk_pixbuf_dep = dependency('gdk-pixbuf-2.0', version: '>= 2.40.0')

