From 2e9217bb88b76cebfcd2b65b76c8733f80da77ea Mon Sep 17 00:00:00 2001
From: Jeremy White <jwhite@codeweavers.com>
Date: Thu, 21 Mar 2013 10:59:13 -0500
Subject: Include xf86Modes.h to use functions from
 hw/xfree86/modes/xf86Modes.c.

Signed-off-by: Jeremy White <jwhite@codeweavers.com>
Reviewed-by: Robert Morell <rmorell@nvidia.com>

diff --git a/src/savage_driver.h b/src/savage_driver.h
index ec4b1ea..de042af 100644
--- a/src/savage_driver.h
+++ b/src/savage_driver.h
@@ -54,6 +54,7 @@
 #if GET_ABI_MAJOR(ABI_VIDEODRV_VERSION) < 6
 #include "xf86Resources.h"
 #endif
+#include "xf86Modes.h"
 #include "xf86Pci.h"
 #include "xf86_OSproc.h"
 #include "xf86Cursor.h"
-- 
cgit v0.10.2
From 16a672d1ba183601d513c4cdca32b47e926a6d1e Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Wed, 21 May 2014 14:11:46 -0400
Subject: Use own thunk function instead of vgaHW*Weak

I plan to remove the Weak functions from a future server.

Signed-off-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/savage_driver.c b/src/savage_driver.c
index bca2c60..e82794f 100644
--- a/src/savage_driver.c
+++ b/src/savage_driver.c
@@ -4576,6 +4576,12 @@ SavageDDC1Read(ScrnInfoPtr pScrn)
     return ((unsigned int) (tmp & 0x08));
 }
 
+static void
+SavageDDC1SetSpeed(ScrnInfoPtr pScrn, xf86ddcSpeed speed)
+{
+    vgaHWddc1SetSpeed(pScrn, speed);
+}
+
 static Bool
 SavageDDC1(ScrnInfoPtr pScrn)
 {
@@ -4589,7 +4595,8 @@ SavageDDC1(ScrnInfoPtr pScrn)
     InI2CREG(byte,psav->I2CPort);
     OutI2CREG(byte | 0x12,psav->I2CPort);
 
-    pMon = xf86DoEDID_DDC1(XF86_SCRN_ARG(pScrn),vgaHWddc1SetSpeedWeak(),SavageDDC1Read);
+    pMon = xf86DoEDID_DDC1(XF86_SCRN_ARG(pScrn), SavageDDC1SetSpeed,
+			   SavageDDC1Read);
     if (!pMon)
         return FALSE;
     
-- 
cgit v0.10.2
From d28cd83c7b0b4a943efbe5ddf257c8ee2646ea73 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 29 Jul 2014 09:15:26 -0400
Subject: dri: Stop (uselessly) initializing the ValidateTree hooks

Signed-off-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/savage_dri.c b/src/savage_dri.c
index bdb78e4..0842821 100644
--- a/src/savage_dri.c
+++ b/src/savage_dri.c
@@ -875,9 +875,6 @@ Bool SAVAGEDRIScreenInit( ScreenPtr pScreen )
    psav->coreWakeupHandler = pDRIInfo->wrap.WakeupHandler;
    pDRIInfo->wrap.WakeupHandler = SAVAGEWakeupHandler;
 
-   pDRIInfo->wrap.ValidateTree = NULL;
-   pDRIInfo->wrap.PostValidateTree = NULL;
-
    pDRIInfo->createDummyCtx = TRUE;
    pDRIInfo->createDummyCtxPriv = FALSE;
 
-- 
cgit v0.10.2

