diff -upr intervals-0.8.1.orig/tests/interval/test_initialization.py intervals-0.8.1/tests/interval/test_initialization.py
--- intervals-0.8.1.orig/tests/interval/test_initialization.py	2016-07-10 21:14:03.000000000 +0300
+++ intervals-0.8.1/tests/interval/test_initialization.py	2019-11-01 17:26:19.862962549 +0200
@@ -23,7 +23,7 @@ class TestIntervalInit(object):
             'First argument should be a list or tuple. If you wish to '
             'initialize an interval from string, use from_string factory '
             'method.'
-        ) in str(e)
+        ) in str(e.value)
 
     def test_invalid_argument(self):
         with raises(IllegalArgument) as e:
@@ -31,7 +31,7 @@ class TestIntervalInit(object):
         assert (
             'The bounds may be equal only if at least one of the bounds is '
             'closed.'
-        ) in str(e)
+        ) in str(e.value)
 
     def test_floats(self):
         interval = FloatInterval((0.2, 0.5))
