--- a/miracle-wm-config/include/miracle/keyboard.h
+++ b/miracle-wm-config/include/miracle/keyboard.h
@@ -22,6 +22,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <cstdlib>
 #include <mir_toolkit/events/enums.h>
 
+typedef unsigned int uint;
+
 namespace miracle
 {
 constexpr std::array<std::pair<const char*, uint>, mir_keyboard_actions - 1> mir_keyboard_actions_strings = {
--- a/miracle-wm-config/include/miracle/miracle-wm-config-c.h
+++ b/miracle-wm-config/include/miracle/miracle-wm-config-c.h
@@ -21,6 +21,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <mir_toolkit/mir_input_device_types.h>
 #include <stdlib.h>
 
+typedef unsigned int uint;
+
 #ifdef __cplusplus
 extern "C"
 {
--- a/miracle-wm-config/include/miracle/miracle-wm-config.h
+++ b/miracle-wm-config/include/miracle/miracle-wm-config.h
@@ -35,6 +35,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <string>
 #include <vector>
 
+typedef unsigned int uint;
+
 namespace miracle
 {
 struct MIRACLE_WM_CONFIG_API KeyCommand
--- a/miracle-wm-config/include/miracle/modifiers.h
+++ b/miracle-wm-config/include/miracle/modifiers.h
@@ -22,6 +22,8 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #include <cstdlib>
 #include <mir_toolkit/events/enums.h>
 
+typedef unsigned int uint;
+
 namespace miracle
 {
 constexpr uint miracle_input_event_modifier_default = 1 << 18;
