--- a/dist/admin/html.open/lib/CData.php.orig
+++ b/dist/admin/html.open/lib/CData.php
@@ -46,7 +46,7 @@
 
         error_log("Migrate plain conf to xml from $SERVER_ROOT\n");
 
-        if ($SERVER_ROOT{-1} != '/')
+        if ($SERVER_ROOT[-1] != '/')
             $SERVER_ROOT .= '/';
 
         define('SERVER_ROOT', $SERVER_ROOT);
@@ -77,7 +77,7 @@
             die("Require removexml as input param with value 1 or 0.");
 
         error_log("Migrate xml to plain conf under server root $SERVER_ROOT\n");
-        if ($SERVER_ROOT{-1} != '/')
+        if ($SERVER_ROOT[-1] != '/')
             $SERVER_ROOT .= '/';
 
         define('SERVER_ROOT', $SERVER_ROOT);
--- a/dist/admin/html.open/lib/PathTool.php.orig
+++ b/dist/admin/html.open/lib/PathTool.php
@@ -5,7 +5,7 @@
 
     public static function getAbsolutePath($root, $path)
     {
-        if ($path{-1} != '/')
+        if ($path[-1] != '/')
             $path .= '/';
 
         $newPath = $this->getAbsoluteFile($root, $path);
@@ -14,7 +14,7 @@
 
     public static function getAbsoluteFile($root, $path)
     {
-        if ($path{0} != '/')
+        if ($path[0] != '/')
             $path = $root . '/' . $path;
 
         $newPath = $this->clean($path);
--- a/dist/admin/html.open/lib/PlainConfParser.php.orig
+++ b/dist/admin/html.open/lib/PlainConfParser.php
@@ -20,7 +20,7 @@
         $parentid = $index - 1;
         $level = ($index > 0) ? $this->_list[$parentid][1] + 1 : 0;
         $fullpath = $filename;
-        if ($filename{0} != '/') {
+        if ($filename[0] != '/') {
             if ($parentid) {
                 $fullpath = $this->_list[$parentid][3] . '/' . $filename;
             } else {
