--- a/dist/admin/html.open/lib/DAttrBase.php.orig
+++ b/dist/admin/html.open/lib/DAttrBase.php
@@ -250,9 +250,9 @@
             $value = $this->extractCheckBoxOr();
         else {
             $value = UIBase::GrabInput("post", $this->_key);
-            if (get_magic_quotes_gpc()) {
+            /* if (get_magic_quotes_gpc()) {
                 $value = stripslashes($value);
-            }
+            } */
         }
         $value = str_replace("\r\n", "\n", $value);
 
