--- a/dist/conf/httpd_config.conf.in
+++ b/dist/conf/httpd_config.conf.in
@@ -20,7 +20,7 @@
 indexFiles                       index.html, index.php
 
 errorlog $SERVER_ROOT/logs/error.log {
-        logLevel             DEBUG
+        logLevel             INFO
         debugLevel           0
         rollingSize          10M
         enableStderrLog      1
@@ -115,9 +115,9 @@
 	deny
 }
  
-extProcessor lsphp5{
+extProcessor lsphp7{
     type                            lsapi 
-    address                         uds://tmp/lshttpd/lsphp.sock 
+    address                         uds://tmp/lshttpd/lsphp7.sock 
     maxConns                        35 
     env                             PHP_LSAPI_MAX_REQUESTS=500 
     env                             PHP_LSAPI_CHILDREN=35 
@@ -127,7 +127,7 @@
     pcKeepAliveTimeout
     respBuffer                      0 
     autoStart                       1 
-    path                            $SERVER_ROOT/fcgi-bin/lsphp5 
+    path                            $SERVER_ROOT/fcgi-bin/lsphp7 
     backlog                         100 
     instances                       1 
     priority                        0 
@@ -138,7 +138,7 @@
 }
 
 scriptHandler{
-    add lsapi:lsphp5  php
+    add lsapi:lsphp7  php
 }
 
 railsDefaults{
