--- a/src/main/httpserver.cpp.orig
+++ b/src/main/httpserver.cpp
@@ -138,7 +138,7 @@
 
 #if defined(__FreeBSD__ ) || defined(__NetBSD__) || defined(__OpenBSD__) \
     || defined(macintosh) || defined(__APPLE__) || defined(__APPLE_CC__)
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 #endif
 
 #define FILEMODE            0644
--- a/src/main/lshttpdmain.cpp.orig
+++ b/src/main/lshttpdmain.cpp
@@ -50,7 +50,7 @@
 #include <util/httpfetch.h>
 #include <socket/gsockaddr.h>
 
-#include <sys/sysctl.h>
+#include <linux/sysctl.h>
 
 #include <extensions/cgi/cgidworker.h>
 #include <extensions/registry/extappregistry.h>

--- a/src/sslpp/sslsesscache.h.orig
+++ b/src/sslpp/sslsesscache.h
@@ -24,6 +24,7 @@
 #include <util/tsingleton.h>
 #include <stdint.h>
 #include <unistd.h>
+#include <time.h>
 
 
 #define LS_SSLSESSCACHE_DEFAULTSIZE 40*1024
--- a/src/modules/lua/lsluadefs.h.orig
+++ b/src/modules/lua/lsluadefs.h
@@ -22,8 +22,8 @@
 extern "C" {
 #endif
 /* need lua_State * */
-#include <luajit.h>
-#include <lauxlib.h>
+#include <luajit-2.1/luajit.h>
+#include <luajit-2.1/lauxlib.h>
 #ifdef __cplusplus
 };
 #endif
--- a/configure.orig
+++ b/configure
@@ -16031,16 +16031,16 @@
 if test "$OPENLSWS_BROTLI" = "no" ; then
     echo "Brotli compression disabled!!!"
 else
-    LIBBROTLI=" $OPENLSWS_BROTLI/out/libbrotlidec-static.a $OPENLSWS_BROTLI/out/libbrotlienc-static.a $OPENLSWS_BROTLI/out/libbrotlicommon-static.a "
-    as_ac_File=`$as_echo "ac_cv_file_$OPENLSWS_BROTLI/out/libbrotlidec-static.a" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $OPENLSWS_BROTLI/out/libbrotlidec-static.a" >&5
-$as_echo_n "checking for $OPENLSWS_BROTLI/out/libbrotlidec-static.a... " >&6; }
+    LIBBROTLI=" /usr/lib/libbrotlidec-static.a /usr/lib/libbrotlienc-static.a /usr/lib/libbrotlicommon-static.a "
+    as_ac_File=`$as_echo "ac_cv_file_/usr/lib/libbrotlidec-static.a" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for /usr/lib/libbrotlidec-static.a" >&5
+$as_echo_n "checking for /usr/lib/libbrotlidec-static.a... " >&6; }
 if eval \${$as_ac_File+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   test "$cross_compiling" = yes &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "$OPENLSWS_BROTLI/out/libbrotlidec-static.a"; then
+if test -r "/usr/lib/libbrotlidec-static.a"; then
   eval "$as_ac_File=yes"
 else
   eval "$as_ac_File=no"
@@ -16052,21 +16052,21 @@
 if eval test \"x\$"$as_ac_File"\" = x"yes"; then :
 
 cat >>confdefs.h <<_ACEOF
-#define `$as_echo "HAVE_$OPENLSWS_BROTLI/out/libbrotlidec-static.a" | $as_tr_cpp` 1
+#define `$as_echo "HAVE_/usr/lib/libbrotlidec-static.a" | $as_tr_cpp` 1
 _ACEOF
 
 else
   LIBBROTLI=
 fi
-as_ac_File=`$as_echo "ac_cv_file_$OPENLSWS_BROTLI/c/include/brotli/decode.h" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $OPENLSWS_BROTLI/c/include/brotli/decode.h" >&5
-$as_echo_n "checking for $OPENLSWS_BROTLI/c/include/brotli/decode.h... " >&6; }
+as_ac_File=`$as_echo "ac_cv_file_/usr/include/brotli/decode.h" | $as_tr_sh`
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for /usr/include/brotli/decode.h" >&5
+$as_echo_n "checking for /usr/include/brotli/decode.h... " >&6; }
 if eval \${$as_ac_File+:} false; then :
   $as_echo_n "(cached) " >&6
 else
   test "$cross_compiling" = yes &&
   as_fn_error $? "cannot check for file existence when cross compiling" "$LINENO" 5
-if test -r "$OPENLSWS_BROTLI/c/include/brotli/decode.h"; then
+if test -r "/usr/include/brotli/decode.h"; then
   eval "$as_ac_File=yes"
 else
   eval "$as_ac_File=no"
@@ -16078,7 +16078,7 @@
 if eval test \"x\$"$as_ac_File"\" = x"yes"; then :
 
 cat >>confdefs.h <<_ACEOF
-#define `$as_echo "HAVE_$OPENLSWS_BROTLI/c/include/brotli/decode.h" | $as_tr_cpp` 1
+#define `$as_echo "HAVE_/sur/include/brotli/decode.h" | $as_tr_cpp` 1
 _ACEOF
 
 else
@@ -16089,8 +16089,8 @@
         echo "Brotli files check failed. If continue, brotli compression will be disabled."
     else
         echo "Brotli checked and enabled!!!"
-        CPPFLAGS="$CPPFLAGS -I$OPENLSWS_BROTLI/c/include "
-        LIBBROTLI=" $OPENLSWS_BROTLI/out/libbrotlidec-static.a $OPENLSWS_BROTLI/out/libbrotlienc-static.a $OPENLSWS_BROTLI/out/libbrotlicommon-static.a "
+        CPPFLAGS="$CPPFLAGS -I/usr/include "
+        LIBBROTLI=" /usr/lib/libbrotlidec-static.a /usr/lib/libbrotlienc-static.a /usr/lib/libbrotlicommon-static.a "
 
 cat >>confdefs.h <<_ACEOF
 #define USE_BROTLI 1

