--- a/configure
+++ b/configure
@@ -15863,7 +15863,7 @@
 
 
 if test "$OPENLSWS_BSSL" = no ; then
-    if test "$OSNAME" = Darwin ; then
+    if test "$OSNAME" = Linux ; then
         usedynossl=yes
         CPPFLAGS="$CPPFLAGS -I../../src -I../../include -I/usr/local/include -I/usr/local -I/usr -I/usr/pkg "
     else
@@ -16131,7 +16131,7 @@
 
 
 
-if test "$OSNAME" = Darwin ; then
+if test "$OSNAME" = Linux ; then
 
 
 # Check whether --with-openssl was given.
--- a/dist/install.sh
+++ b/dist/install.sh
@@ -49,11 +49,7 @@
     fi
         
     if [ "x$OS" = "xLinux" ] ; then
-        if [ "x$OSTYPE" != "x64" ] ; then
-            $DLCMD $LSWS_HOME/admin/fcgi-bin/admin_php http://www.litespeedtech.com/packages/lsphp5_bin/i386/lsphp5
-        else
-            $DLCMD $LSWS_HOME/admin/fcgi-bin/admin_php http://www.litespeedtech.com/packages/lsphp5_bin/x86_64/lsphp5
-        fi
+        ln -s /usr/bin/lsphp7 $LSWS_HOME/admin/fcgi-bin/admin_php
         
         if [ $? = 0 ] ; then 
             HASADMINPHP=y
--- a/dist/functions.sh
+++ b/dist/functions.sh
@@ -375,7 +375,7 @@
 
 	if [ $INST_USER = "root" ]; then
 		if [ $SUCC -eq "1" ]; then
-			chown -R "$DIR_OWN" "$LSWS_HOME"
+			: #chown -R "$DIR_OWN" "$LSWS_HOME"
 		fi
 	fi
 }
@@ -626,7 +626,7 @@
 	  if [ ! -d "$LSWS_HOME/$arg" ]; then
 		  mkdir "$LSWS_HOME/$arg"
 	  fi
-	  chown "$OWNER" "$LSWS_HOME/$arg"
+	  #chown "$OWNER" "$LSWS_HOME/$arg"
 	  chmod $PERM  "$LSWS_HOME/$arg"
 	done
 	
@@ -643,7 +643,7 @@
 	  do
 	  if [ -f "$LSINSTALL_DIR/$arg" ]; then 
 	    cp -f "$LSINSTALL_DIR/$arg" "$LSWS_HOME/$arg"
-	    chown "$OWNER" "$LSWS_HOME/$arg"
+	    #chown "$OWNER" "$LSWS_HOME/$arg"
 	    chmod $PERM  "$LSWS_HOME/$arg"
 	  fi
 	done
@@ -662,7 +662,7 @@
 	    cp "$LSINSTALL_DIR/$arg" "$LSWS_HOME/$arg"
 	  fi
 	  if [ -f "$LSWS_HOME/$arg" ]; then
-        chown "$OWNER" "$LSWS_HOME/$arg"
+        #chown "$OWNER" "$LSWS_HOME/$arg"
         chmod $PERM  "$LSWS_HOME/$arg"
       fi
 	done
@@ -678,7 +678,7 @@
 	for arg
 	  do
 	  cp -R "$LSINSTALL_DIR/$arg/"* "$LSWS_HOME/$arg/"
-	  chown -R "$OWNER" "$LSWS_HOME/$arg/"*
+	  #chown -R "$OWNER" "$LSWS_HOME/$arg/"*
       #chmod -R $PERM  $LSWS_HOME/$arg/*
 	done
 }
@@ -696,7 +696,7 @@
 	for arg
 	  do
 	  cp -R "$LSINSTALL_DIR/$arg/"* "$LSWS_HOME/$arg.$VERSION/"
-	  chown -R "$OWNER" "$LSWS_HOME/$arg.$VERSION"
+	  #chown -R "$OWNER" "$LSWS_HOME/$arg.$VERSION"
       $TEST_BIN -L "$LSWS_HOME/$arg"
       if [ $? -eq 0 ]; then
 		  rm -f "$LSWS_HOME/$arg"
@@ -718,7 +718,7 @@
 	for arg
 	  do
 	  cp -f "$LSINSTALL_DIR/$arg" "$LSWS_HOME/$arg.$VERSION"
-	  chown "$OWNER" "$LSWS_HOME/$arg.$VERSION"
+	  #chown "$OWNER" "$LSWS_HOME/$arg.$VERSION"
 	  chmod $PERM  "$LSWS_HOME/$arg.$VERSION"
       $TEST_BIN -L "$LSWS_HOME/$arg"
       if [ $? -eq 0 ]; then
@@ -735,11 +735,11 @@
     umask 022
 	if [ $INST_USER = "root" ]; then
 		SDIR_OWN="root:$ROOTGROUP"
-        chown $SDIR_OWN $LSWS_HOME
+        #chown $SDIR_OWN $LSWS_HOME
 	else
 		SDIR_OWN=$DIR_OWN
 	fi
-	sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc"
+	#sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc"
 
 	OWNER=$1
 	PERM=$2
@@ -750,7 +750,7 @@
 	  if [ ! -f "$LSWS_HOME/$arg" ]; then
 		  cp "$LSINSTALL_DIR/$arg" "$LSWS_HOME/$arg"
 	  fi
-	  chown "$OWNER" "$LSWS_HOME/$arg"
+	  #chown "$OWNER" "$LSWS_HOME/$arg"
 	  chmod $PERM  "$LSWS_HOME/$arg"
 	done
 }
@@ -765,7 +765,7 @@
 	for arg
 	  do
 	  cp -R "$LSINSTALL_DIR/$arg/"* "$LSWS_HOME/$arg/"
-	  chown -R "$OWNER" "$LSWS_HOME/$arg/"*
+	  #chown -R "$OWNER" "$LSWS_HOME/$arg/"*
       #chmod -R $PERM  $LSWS_HOME/$arg/*
 	done
 }
@@ -784,7 +784,7 @@
 	  do
       if [ -d "$LSINSTALL_DIR/$arg" ]; then
         cp -R "$LSINSTALL_DIR/$arg/"* "$LSWS_HOME/$arg.$VERSION/"
-        chown -R "$OWNER" "$LSWS_HOME/$arg.$VERSION"
+        #chown -R "$OWNER" "$LSWS_HOME/$arg.$VERSION"
         $TEST_BIN -L "$LSWS_HOME/$arg"
         if [ $? -eq 0 ]; then
 		  rm -f "$LSWS_HOME/$arg"
@@ -809,7 +809,7 @@
 	  if [ -f "$LSINSTALL_DIR/$arg" ]; then
 	  
         cp -f "$LSINSTALL_DIR/$arg" "$LSWS_HOME/$arg.$VERSION"
-        chown "$OWNER" "$LSWS_HOME/$arg.$VERSION"
+        #chown "$OWNER" "$LSWS_HOME/$arg.$VERSION"
         chmod $PERM  "$LSWS_HOME/$arg.$VERSION"
         $TEST_BIN -L "$LSWS_HOME/$arg"
         if [ $? -eq 0 ]; then
@@ -841,11 +841,11 @@
 
 create_lsadm()
 {
-    groupadd lsadm 
+    #groupadd lsadm 
     #1>/dev/null 2>&1
     lsadm_gid=`grep "^lsadm:" /etc/group | awk -F : '{ print $3; }'`
-    useradd -g $lsadm_gid -d / -r -s /sbin/nologin lsadm 
-    usermod -a -G $WS_GROUP lsadm
+    #useradd -g $lsadm_gid -d / -r -s /sbin/nologin lsadm 
+    #usermod -a -G $WS_GROUP lsadm
     #1>/dev/null 2>&1
     
 }
@@ -880,12 +880,12 @@
             CONF_OWN="lsadm:lsadm"
         fi
 		SDIR_OWN="root:$ROOTGROUP"
-        chown $SDIR_OWN $LSWS_HOME
+        #chown $SDIR_OWN $LSWS_HOME
 	else
 		SDIR_OWN=$DIR_OWN
 	fi
-	sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc"
-	sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.gentoo.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc.gentoo"
+	#sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc"
+	#sed "s:%LSWS_CTRL%:$LSWS_HOME/bin/lswsctrl:" "$LSINSTALL_DIR/admin/misc/lsws.rc.gentoo.in" > "$LSINSTALL_DIR/admin/misc/lsws.rc.gentoo"
     if [ -d "$LSWS_HOME/admin/html.$VERSION" ]; then
         rm -rf "$LSWS_HOME/admin/html.$VERSION"
     fi
@@ -894,10 +894,10 @@
     util_mkdir "$CONF_OWN" $SDIR_MOD conf conf/cert conf/templates conf/vhosts conf/vhosts/Example admin/conf admin/tmp phpbuild
     util_mkdir "$SDIR_OWN" $SDIR_MOD admin/cgid admin/cgid/secret
     util_mkdir "$DIR_OWN" $SDIR_MOD tmp/ocspcache
-    chgrp  $WS_GROUP $LSWS_HOME/admin/tmp $LSWS_HOME/admin/cgid
+    #chgrp  $WS_GROUP $LSWS_HOME/admin/tmp $LSWS_HOME/admin/cgid
     chmod  g+x $LSWS_HOME/admin/tmp $LSWS_HOME/admin/cgid 
-    chown  $CONF_OWN $LSWS_HOME/admin/tmp/sess_* 1>/dev/null 2>&1
-    chown  $DIR_OWN $LSWS_HOME/cachedata
+    #chown  $CONF_OWN $LSWS_HOME/admin/tmp/sess_* 1>/dev/null 2>&1
+    #chown  $DIR_OWN $LSWS_HOME/cachedata
     util_mkdir "$SDIR_OWN" $DIR_MOD Example 
 
     find "$LSWS_HOME/admin/tmp" -type s -atime +1 -delete 2>/dev/null
@@ -919,7 +919,7 @@
     
     util_ccpfile "$SDIR_OWN" $EXEC_MOD fcgi-bin/lsperld.fpl fcgi-bin/RackRunner.rb
     util_cpfile "$SDIR_OWN" $EXEC_MOD  fcgi-bin/RailsRunner.rb  fcgi-bin/RailsRunner.rb.2.3
-	util_cpfile "$SDIR_OWN" $EXEC_MOD admin/misc/rc-inst.sh admin/misc/admpass.sh admin/misc/rc-uninst.sh admin/misc/uninstall.sh admin/misc/lsws.rc admin/misc/lsws.rc.gentoo admin/misc/enable_phpa.sh admin/misc/mgr_ver.sh admin/misc/gzipStatic.sh admin/misc/fp_install.sh admin/misc/create_admin_keypair.sh admin/misc/awstats_install.sh admin/misc/update.sh admin/misc/cleancache.sh admin/misc/lsup.sh
+	#util_cpfile "$SDIR_OWN" $EXEC_MOD admin/misc/rc-inst.sh admin/misc/admpass.sh admin/misc/rc-uninst.sh admin/misc/uninstall.sh admin/misc/lsws.rc admin/misc/lsws.rc.gentoo admin/misc/enable_phpa.sh admin/misc/mgr_ver.sh admin/misc/gzipStatic.sh admin/misc/fp_install.sh admin/misc/create_admin_keypair.sh admin/misc/awstats_install.sh admin/misc/update.sh admin/misc/cleancache.sh admin/misc/lsup.sh
     util_cpfile "$SDIR_OWN" $EXEC_MOD admin/misc/ap_lsws.sh.in admin/misc/build_ap_wrapper.sh admin/misc/cpanel_restart_httpd.in admin/misc/build_admin_php.sh admin/misc/convertxml.sh
 	util_cpfile "$SDIR_OWN" $DOC_MOD admin/misc/gdb-bt admin/misc/htpasswd.php admin/misc/php.ini admin/misc/genjCryptionKeyPair.php admin/misc/purge_cache_byurl.php
     util_cpfile "$SDIR_OWN" $DOC_MOD admin/misc/convertxml.php
@@ -971,7 +971,7 @@
 		util_cpdir "$SDIR_OWN" $DOC_MOD Example/html Example/cgi-bin
 	fi
 
-    chown -R "$CONF_OWN" "$LSWS_HOME/conf/"
+    #chown -R "$CONF_OWN" "$LSWS_HOME/conf/"
     chmod -R 0755 "$LSWS_HOME/conf/"
     chmod 0600 "$LSWS_HOME/conf/httpd_config.conf"
     chmod 0600 "$LSWS_HOME/conf/vhosts/Example/vhconf.conf"
