--- a/include/lsr/ls_lock.h
+++ b/include/lsr/ls_lock.h
@@ -52,16 +52,9 @@
 extern "C" {
 #endif
 
-#if defined(linux) || defined(__linux) || defined(__linux__) || defined(__gnu_linux__) \
-    || defined(__FreeBSD__ ) || defined(__NetBSD__) || defined(__OpenBSD__)
-#define USE_F_MUTEX
-#define USE_MUTEX_ADAPTIVE
-
-#else
 #undef USE_F_MUTEX
 #define USE_MUTEX_LOCK
 
-#endif
 #define USE_ATOMIC_SPIN
 
 #define MAX_FUTEX_SPINCNT      10
--- a/src/lsr/ls_lock.c.orig
+++ b/src/lsr/ls_lock.c
@@ -118,11 +118,7 @@
 #else  /* defined(USE_MUTEX_ADAPTIVE) */
     /* pthread_mutexattr_settype(&myAttr, PTHREAD_MUTEX_NORMAL); */
     pthread_mutexattr_settype(&myAttr,
-#if defined(linux) || defined(__linux) || defined(__linux__) || defined(__gnu_linux__)
-                              PTHREAD_MUTEX_ERRORCHECK_NP
-#else  /* defined(linux) */
                               PTHREAD_MUTEX_ERRORCHECK
-#endif  /* defined(linux) */
                              );
 #endif  /* defined(USE_MUTEX_ADAPTIVE) */
     /* pthread_mutexattr_settype(&myAttr, PTHREAD_MUTEX_RECURSIVE); */

