--- ./electron-builder.ts.orig
+++ ./electron-builder.ts
@@ -1,11 +1,10 @@
 // Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
-const semver = require('semver');
 
 const pkg = require('./package.json');
 
 function getMacVersions() {
-    const stableVersion = semver.coerce(pkg.version)?.version || pkg.version;
+    const stableVersion = pkg.version;
     let buildVersion = stableVersion;
     if (process.env.GITHUB_RUN_ID) {
         buildVersion = `${process.env.GITHUB_RUN_ID}${process.env.GITHUB_RUN_ATTEMPT || ''}`;
