From 4c2c153f266df924620c98efd33d87769ec7cadb Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Fri, 17 Sep 2021 10:56:50 +0200
Subject: [PATCH] build type none

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b0840ac..2dcdfa9 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -65,6 +65,7 @@ endif()
 ###############################################################################
 # Define compilation mode i.e. debug or release
 
+if(false)
 if(NOT DEFINED CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL "")
     message(STATUS "Setting build type to 'Release' as none was specified.")
     set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
@@ -86,6 +87,7 @@ if(NOT "${CMAKE_BUILD_TYPE}" IN_LIST CMAKE_CONFIGURATION_TYPES)
     message(FATAL_ERROR 
             "CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} is unsupported. Supported values are: ${_CMAKE_CONFIGURATION_TYPES_STR}.")
 endif()
+endif()
 
 # Is that in debug mode?
 
-- 
2.33.0

