diff --git a/src/core/Config.cpp b/src/core/Config.cpp
index f5cb379..5ea178e 100644
--- a/src/core/Config.cpp
+++ b/src/core/Config.cpp
@@ -330,7 +330,7 @@ OCIO_NAMESPACE_ENTER
                 sanitytext_ = rhs.sanitytext_;
                 
                 cacheids_ = rhs.cacheids_;
-                cacheidnocontext_ = cacheidnocontext_;
+                cacheidnocontext_ = rhs.cacheidnocontext_;
             }
             return *this;
         }
diff --git a/src/pyglue/PyAllocationTransform.cpp b/src/pyglue/PyAllocationTransform.cpp
index 20bb50e..06b418a 100644
--- a/src/pyglue/PyAllocationTransform.cpp
+++ b/src/pyglue/PyAllocationTransform.cpp
@@ -53,7 +53,6 @@ OCIO_NAMESPACE_ENTER
         ///
         
         int PyOCIO_AllocationTransform_init(PyOCIO_Transform * self, PyObject * args, PyObject * kwds);
-        PyObject * PyOCIO_AllocationTransform_equals(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getAllocation(PyObject * self);
         PyObject * PyOCIO_AllocationTransform_setAllocation(PyObject * self,  PyObject * args);
         PyObject * PyOCIO_AllocationTransform_getNumVars(PyObject * self);

