Description: Add Debian menu structure to system.mwmrc-menu
Forwarded: not-needed
Author: Stefan Bauer <stefan.bauer@cubewerk.de>
Author: Graham Inggs <graham@nerve.org.za>
Last-Update: 2013-01-21
--- a/clients/mwm/system.mwmrc
+++ b/clients/mwm/system.mwmrc
@@ -49,6 +49,8 @@
 	"Pack Icons"		f.pack_icons
 !	"Toggle Behavior..."	f.set_behavior
 	 no-label		f.separator
+	"Debian"		f.menu "/Debian"
+	 no-label		f.separator
 	"Restart..."		f.restart
 	"Quit..."		f.quit_mwm
 
@@ -141,3 +143,7 @@
 !!
 !!  END OF mwm RESOURCE DESCRIPTION FILE
 !!
+
+! debian menu
+!include-menu-defs
+include-menu-defs
