diff -ruN gocryptfs_v1.7.1_src-deps/Makefile gocryptfs_v1.7.1_src-deps/Makefile
--- gocryptfs_v1.7.1_src-deps/Makefile
+++ gocryptfs_v1.7.1_src-deps/Makefile
@@ -1,7 +1,6 @@
 .phony: build
 build:
 	./build.bash
-	./Documentation/MANPAGE-render.bash
 
 .phony: test
 test:
@@ -20,6 +19,4 @@
 install:
 	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs
 	install -Dm755 -t "$(DESTDIR)/usr/bin/" gocryptfs-xray/gocryptfs-xray
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs.1
-	install -Dm644 -t "$(DESTDIR)/usr/share/man/man1/" Documentation/gocryptfs-xray.1
 	install -Dm644 -t "$(DESTDIR)/usr/share/licenses/gocryptfs" LICENSE
