diff -ruN a/expander/conversion.ml b/expander/conversion.ml
--- a/expander/conversion.ml	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/conversion.ml	2026-07-18 14:34:45.202106338 -0400
@@ -59,7 +59,7 @@
   let maybe_apply_generic ~loc ~types ~binds maybe_arg cases =
     let expr =
       match maybe_arg with
-      | None -> pexp_function ~loc cases
+      | None -> pexp_function_cases ~loc cases
       | Some arg -> pexp_match ~loc arg cases
     in
     with_types ~loc ~types (with_let ~loc ~binds expr)
diff -ruN a/expander/conversion.mli b/expander/conversion.mli
--- a/expander/conversion.mli	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/conversion.mli	2026-07-18 14:34:45.202204637 -0400
@@ -2,7 +2,7 @@
 open! Ppxlib
 
 (** Sexp conversion function, expressed as either a single expression or as a collection
-    of [match] cases. Expressing as cases rather than wrapping directly in [pexp_function]
+    of [match] cases. Expressing as cases rather than wrapping directly in [pexp_function_cases]
     allows us to simplify some expressions built on this. *)
 type t
 
diff -ruN a/expander/expand_of_sexp.ml b/expander/expand_of_sexp.ml
--- a/expander/expand_of_sexp.ml	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/expand_of_sexp.ml	2026-07-18 14:34:45.202389760 -0400
@@ -187,7 +187,8 @@
     | { ptyp_desc = Ptyp_class (_, _); _ }
     | { ptyp_desc = Ptyp_alias (_, _); _ }
     | { ptyp_desc = Ptyp_package _; _ }
-    | { ptyp_desc = Ptyp_extension _; _ } ->
+    | { ptyp_desc = Ptyp_extension _; _ }
+    | { ptyp_desc = Ptyp_open _; _ } ->
       Location.raise_errorf ~loc "Type unsupported for ppx [of_sexp] conversion"
 
   (* Conversion of tuples *)
@@ -702,7 +703,7 @@
     in
     let default_case = case ~lhs:(ppat_any ~loc) ~guard:None ~rhs:(eint ~loc (-1)) in
     let cases = List.concat [ field_cases; [ default_case ] ] in
-    pexp_function ~loc cases
+    pexp_function_cases ~loc cases
   ;;
 
   let create_arg_for_record_of_sexp td fields ~loc ~constructor =
diff -ruN a/expander/expand_sexp_of.ml b/expander/expand_sexp_of.ml
--- a/expander/expand_sexp_of.ml	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/expand_sexp_of.ml	2026-07-18 14:34:45.202495043 -0400
@@ -87,7 +87,8 @@
     | { ptyp_desc = Ptyp_class (_, _); _ }
     | { ptyp_desc = Ptyp_alias (_, _); _ }
     | { ptyp_desc = Ptyp_package _; _ }
-    | { ptyp_desc = Ptyp_extension _; _ } ->
+    | { ptyp_desc = Ptyp_extension _; _ }
+    | { ptyp_desc = Ptyp_open _; _ } ->
       Location.raise_errorf ~loc "Type unsupported for ppx [sexp_of] conversion"
 
   (* Conversion of tuples *)
diff -ruN a/expander/helpers.ml b/expander/helpers.ml
--- a/expander/helpers.ml	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/helpers.ml	2026-07-18 14:34:45.202577072 -0400
@@ -156,7 +156,7 @@
 let rec is_value_expression expr =
   match expr.pexp_desc with
   (* Syntactic values. *)
-  | Pexp_ident _ | Pexp_constant _ | Pexp_function _ | Pexp_fun _ | Pexp_lazy _ -> true
+  | Pexp_ident _ | Pexp_constant _ | Pexp_function _ | Pexp_lazy _ -> true
   (* Type-only wrappers; we check their contents. *)
   | Pexp_constraint (expr, (_ : core_type))
   | Pexp_coerce (expr, (_ : core_type option), (_ : core_type))
diff -ruN a/expander/ppx_sexp_conv_grammar.ml b/expander/ppx_sexp_conv_grammar.ml
--- a/expander/ppx_sexp_conv_grammar.ml	2023-04-25 09:12:25.000000000 -0400
+++ b/expander/ppx_sexp_conv_grammar.ml	2026-07-18 14:34:45.202664603 -0400
@@ -312,7 +312,8 @@
             grammar_of_polymorphic_variant ~loc ~rec_flag ~tags_of_doc_comments rows)
        | Ptyp_poly _ -> unsupported ~loc "explicitly polymorphic types"
        | Ptyp_package _ -> unsupported ~loc "first-class module types"
-       | Ptyp_extension _ -> unsupported ~loc "unexpanded ppx extensions")
+       | Ptyp_extension _ -> unsupported ~loc "unexpanded ppx extensions"
+       | Ptyp_open _ -> unsupported ~loc "local module open")
   in
   grammar_of_type_tags core_type grammar ~tags_of_doc_comments
 
