From 9075231b636d354cd56b456f9d4ada43f254e879 Mon Sep 17 00:00:00 2001
From: Ben Greiner <code@bnavigator.de>
Date: Sat, 12 Aug 2023 09:37:44 -0700
Subject: [PATCH] Bump Jedi upper pin to <0.20

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 7e62d24c..e0b6e07a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -13,7 +13,7 @@ readme = "README.md"
 license = {text = "MIT"}
 requires-python = ">=3.7"
 dependencies = [
-    "jedi>=0.17.2,<0.19.0",
+    "jedi>=0.17.2,<0.20.0",
     "python-lsp-jsonrpc>=1.0.0",
     "pluggy>=1.0.0",
     "docstring-to-markdown",
