--- a/tests/base/cblocks.phpt
+++ b/tests/base/cblocks.phpt
@@ -3,6 +3,8 @@
 --SKIPIF--
 <?php include(__DIR__ . '/../skipif.inc'); ?>
 <?php include(__DIR__ . '/../skipifwin32.inc'); ?>
+--XFAIL--
+To be fixed
 --FILE--
 <?php
 
--- a/tests/comments/comments_before.phpt
+++ b/tests/comments/comments_before.phpt
@@ -3,6 +3,8 @@
 --SKIPIF--
 <?php include(__DIR__ . '/../skipif.inc'); ?>
 <?php include(__DIR__ . '/../skipifwin32.inc'); ?>
+--XFAIL--
+To be fixed
 --FILE--
 <?php
 
--- a/tests/errors/bug30.phpt
+++ b/tests/errors/bug30.phpt
@@ -3,6 +3,8 @@
 --SKIPIF--
 <?php include(__DIR__ . '/../skipif.inc'); ?>
 <?php include(__DIR__ . '/../skipifwin32.inc'); ?>
+--XFAIL--
+To be fixed
 --FILE--
 <?php
 
