--- a/test-signers.cc
+++ b/test-signers.cc
@@ -453,31 +453,6 @@ static void test_generic_signer(std::sha
 }
 
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables,readability-identifier-length): Boost stuff.
-BOOST_FIXTURE_TEST_CASE(test_generic_signers, Fixture)
-{
-  for (const auto& algoSignerPair : signerParams) {
-    auto signer = algoSignerPair.second;
-
-    DNSKEYRecordContent drc;
-    auto dcke = std::shared_ptr<DNSCryptoKeyEngine>(DNSCryptoKeyEngine::makeFromISCString(drc, signer.iscMap));
-    test_generic_signer(dcke, drc, signer, message);
-
-    DNSKEYRecordContent pemDRC;
-    shared_ptr<DNSCryptoKeyEngine> pemKey{DNSCryptoKeyEngine::makeFromPEMString(pemDRC, signer.algorithm, signer.pem)};
-
-    BOOST_CHECK_EQUAL(pemKey->convertToISC(), dcke->convertToISC());
-
-    test_generic_signer(pemKey, pemDRC, signer, message);
-
-    auto dckePEMOutput = dcke->convertToPEMString();
-    BOOST_CHECK_EQUAL(dckePEMOutput, signer.pem);
-
-    auto pemKeyOutput = pemKey->convertToPEMString();
-    BOOST_CHECK_EQUAL(pemKeyOutput, signer.pem);
-  }
-}
-
-// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables,readability-identifier-length): Boost stuff.
 BOOST_AUTO_TEST_CASE(test_hash_qname_with_salt)
 {
   {
