diff --git a/pdns/dnsname.hh b/pdns/dnsname.hh
index 20d1994b2..b65ef8eed 100644
--- a/dnsname.hh
+++ b/dnsname.hh
@@ -35,6 +35,7 @@
 
 // it crashes on OSX and doesn't compile on OpenBSD
 #if BOOST_VERSION >= 105300 && ! defined( __APPLE__ ) && ! defined(__OpenBSD__)
+#define BOOST_CONTAINER_USE_STD_EXCEPTIONS
 #include <boost/container/string.hpp>
 #endif
 
diff --git a/pdns/dnswriter.cc b/pdns/dnswriter.cc
index 1cbd2b219..b3f2fdd16 100644
--- a/dnswriter.cc
+++ b/dnswriter.cc
@@ -24,6 +24,7 @@
 #endif
 #include <boost/version.hpp>
 #if BOOST_VERSION >= 105400
+#define BOOST_CONTAINER_USE_STD_EXCEPTIONS
 #include <boost/container/static_vector.hpp>
 #endif
 #include "dnswriter.hh"
diff --git a/pdns/pdns_recursor.cc b/pdns/pdns_recursor.cc
index 4a2e9e561..a686470f0 100644
--- a/pdns_recursor.cc
+++ b/pdns_recursor.cc
@@ -27,6 +27,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #ifdef HAVE_BOOST_CONTAINER_FLAT_SET_HPP
+#define BOOST_CONTAINER_USE_STD_EXCEPTIONS
 #include <boost/container/flat_set.hpp>
 #endif
 #include "ws-recursor.hh"
