diff -ru seabios-1.9.2/Makefile seabios-1.9.2.fix/Makefile
--- seabios-1.9.2/Makefile	2016-03-07 16:19:08.000000000 +0200
+++ seabios-1.9.2.fix/Makefile	2016-11-02 17:50:07.418748334 +0200
@@ -62,7 +62,7 @@
     -minline-all-stringops -fomit-frame-pointer \
     -freg-struct-return -ffreestanding -fno-delete-null-pointer-checks \
     -ffunction-sections -fdata-sections -fno-common -fno-merge-constants
-COMMONCFLAGS += $(call cc-option,$(CC),-nopie,)
+COMMONCFLAGS += $(call cc-option,$(CC),-no-pie -fno-pic,)
 COMMONCFLAGS += $(call cc-option,$(CC),-fno-stack-protector,)
 COMMONCFLAGS += $(call cc-option,$(CC),-fno-stack-protector-all,)
 COMMONCFLAGS += $(call cc-option,$(CC),-fstack-check=no,)
