diff --git a/configure.ac b/configure.ac
index 56d3dd0..026dbcd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -285,8 +285,8 @@ $saved_error
 dnl double check ImageMagick install (bug #2582232)
    CPPFLAGS_save="$CPPFLAGS"
    CPPFLAGS="$CPPFLAGS $MAGICK_CFLAGS"
-   AC_CHECK_HEADER([wand/MagickWand.h], [],
-     [AC_MSG_FAILURE([ImageMagick package found but wand/MagickWand.h not installed?!
+   AC_CHECK_HEADER([MagickWand/MagickWand.h], [],
+     [AC_MSG_FAILURE([ImageMagick package found but MagicWand/MagickWand.h not installed?!
 this is a problem with your ImageMagick install,
 please resolve the inconsistency and try again...])])
    CPPFLAGS="$CPPFLAGS_save"])
diff --git a/zbarimg/zbarimg.c b/zbarimg/zbarimg.c
index 5d2db34..a6bc2d1 100644
--- a/zbarimg/zbarimg.c
+++ b/zbarimg/zbarimg.c
@@ -38,7 +38,7 @@
 #include <assert.h>
 
 #include <zbar.h>
-#include <wand/MagickWand.h>
+#include <MagickWand/MagickWand.h>
 
 /* in 6.4.5.4 MagickGetImagePixels changed to MagickExportImagePixels.
  * (still not sure this check is quite right...
--- ./configure.orig
+++ ./configure
@@ -21741,8 +21741,8 @@
    CPPFLAGS_save="$CPPFLAGS"
    CPPFLAGS="$CPPFLAGS $MAGICK_CFLAGS"
    if test "${ac_cv_header_wand_MagickWand_h+set}" = set; then
-  { $as_echo "$as_me:$LINENO: checking for wand/MagickWand.h" >&5
-$as_echo_n "checking for wand/MagickWand.h... " >&6; }
+  { $as_echo "$as_me:$LINENO: checking for MagickWand/MagickWand.h" >&5
+$as_echo_n "checking for MagickWand/MagickWand.h... " >&6; }
 if test "${ac_cv_header_wand_MagickWand_h+set}" = set; then
   $as_echo_n "(cached) " >&6
 fi
@@ -21750,8 +21750,8 @@
 $as_echo "$ac_cv_header_wand_MagickWand_h" >&6; }
 else
   # Is the header compilable?
-{ $as_echo "$as_me:$LINENO: checking wand/MagickWand.h usability" >&5
-$as_echo_n "checking wand/MagickWand.h usability... " >&6; }
+{ $as_echo "$as_me:$LINENO: checking MagickWand/MagickWand.h usability" >&5
+$as_echo_n "checking MagickWand/MagickWand.h usability... " >&6; }
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
@@ -21759,7 +21759,7 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 $ac_includes_default
-#include <wand/MagickWand.h>
+#include <MagickWand/MagickWand.h>
 _ACEOF
 rm -f conftest.$ac_objext
 if { (ac_try="$ac_compile"
@@ -21792,15 +21792,15 @@
 $as_echo "$ac_header_compiler" >&6; }
 
 # Is the header present?
-{ $as_echo "$as_me:$LINENO: checking wand/MagickWand.h presence" >&5
-$as_echo_n "checking wand/MagickWand.h presence... " >&6; }
+{ $as_echo "$as_me:$LINENO: checking MagickWand/MagickWand.h presence" >&5
+$as_echo_n "checking MagickWand/MagickWand.h presence... " >&6; }
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <wand/MagickWand.h>
+#include <MagickWand/MagickWand.h>
 _ACEOF
 if { (ac_try="$ac_cpp conftest.$ac_ext"
 case "(($ac_try" in
@@ -21834,25 +21834,25 @@
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
   yes:no: )
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: accepted by the compiler, rejected by the preprocessor!" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: proceeding with the compiler's result" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: proceeding with the compiler's result" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: accepted by the compiler, rejected by the preprocessor!" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: proceeding with the compiler's result" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: proceeding with the compiler's result" >&2;}
     ac_header_preproc=yes
     ;;
   no:yes:* )
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: present but cannot be compiled" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: present but cannot be compiled" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h:     check for missing prerequisite headers?" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h:     check for missing prerequisite headers?" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: see the Autoconf documentation" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: see the Autoconf documentation" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h:     section \"Present But Cannot Be Compiled\"" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h:     section \"Present But Cannot Be Compiled\"" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: proceeding with the preprocessor's result" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: proceeding with the preprocessor's result" >&2;}
-    { $as_echo "$as_me:$LINENO: WARNING: wand/MagickWand.h: in the future, the compiler will take precedence" >&5
-$as_echo "$as_me: WARNING: wand/MagickWand.h: in the future, the compiler will take precedence" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: present but cannot be compiled" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: present but cannot be compiled" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h:     check for missing prerequisite headers?" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h:     check for missing prerequisite headers?" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: see the Autoconf documentation" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: see the Autoconf documentation" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h:     section \"Present But Cannot Be Compiled\"" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h:     section \"Present But Cannot Be Compiled\"" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: proceeding with the preprocessor's result" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: proceeding with the preprocessor's result" >&2;}
+    { $as_echo "$as_me:$LINENO: WARNING: MagickWand/MagickWand.h: in the future, the compiler will take precedence" >&5
+$as_echo "$as_me: WARNING: MagickWand/MagickWand.h: in the future, the compiler will take precedence" >&2;}
     ( cat <<\_ASBOX
 ## ------------------------------------------- ##
 ## Report this to spadix@users.sourceforge.net ##
@@ -21861,8 +21861,8 @@
      ) | sed "s/^/$as_me: WARNING:     /" >&2
     ;;
 esac
-{ $as_echo "$as_me:$LINENO: checking for wand/MagickWand.h" >&5
-$as_echo_n "checking for wand/MagickWand.h... " >&6; }
+{ $as_echo "$as_me:$LINENO: checking for MagickWand/MagickWand.h" >&5
+$as_echo_n "checking for MagickWand/MagickWand.h... " >&6; }
 if test "${ac_cv_header_wand_MagickWand_h+set}" = set; then
   $as_echo_n "(cached) " >&6
 else
@@ -21877,11 +21877,11 @@
 else
   { { $as_echo "$as_me:$LINENO: error: in \`$ac_pwd':" >&5
 $as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-{ { $as_echo "$as_me:$LINENO: error: ImageMagick package found but wand/MagickWand.h not installed?!
+{ { $as_echo "$as_me:$LINENO: error: ImageMagick package found but MagickWand/MagickWand.h not installed?!
 this is a problem with your ImageMagick install,
 please resolve the inconsistency and try again...
 See \`config.log' for more details." >&5
-$as_echo "$as_me: error: ImageMagick package found but wand/MagickWand.h not installed?!
+$as_echo "$as_me: error: ImageMagick package found but MagickWand/MagickWand.h not installed?!
 this is a problem with your ImageMagick install,
 please resolve the inconsistency and try again...
 See \`config.log' for more details." >&2;}
