Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: use pkgconf to find ZSTD package
----

--- a/cmake/zstd.cmake
+++ b/cmake/zstd.cmake
@@ -4,7 +4,8 @@
 option(WITH_ZSTD
     "Build support for compression/decompression with Zstd." TRUE)
 if (WITH_ZSTD)
-    find_package(zstd CONFIG QUIET)
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(ZSTD REQUIRED libzstd)
     set_package_properties(ZSTD PROPERTIES TYPE
         PURPOSE "General compression support")
     if (ZSTD_FOUND)


--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -276,7 +276,7 @@
 if (ZSTD_FOUND)
     target_link_libraries(${PDAL_LIB_NAME}
         PRIVATE
-        zstd::libzstd)
+        ${ZSTD_LIBRARIES})
 endif()
 
 if (ZLIB_FOUND)
