From 3354a01f768acc7efe8bef29d40c6756e39b9cff Mon Sep 17 00:00:00 2001
From: Marvin Preuss <marvin@xsteadfastx.org>
Date: Wed, 29 Sep 2021 12:12:02 +0200
Subject: [PATCH] Fix: log.Infof format test error

Signed-off-by: Marvin Preuss <marvin@xsteadfastx.org>
---
 speedtest/client.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/speedtest/client.go b/speedtest/client.go
index 1a0ef92..3dd181a 100644
--- a/speedtest/client.go
+++ b/speedtest/client.go
@@ -74,7 +74,7 @@ func NewClient(configURL string, serversURL string) (*Client, error) {
 	closestServers := stClient.GetClosestServers(allServers)
 	// log.Infof("Closest Servers: %s", closestServers)
 	testServer := stClient.GetFastestServer(closestServers)
-	log.Infof("Test server: %s", testServer)
+	log.Infof("Test server: %v", testServer)
 
 	return &Client{
 		Server:          testServer,
@@ -101,6 +101,6 @@ func (client *Client) NetworkMetrics() map[string]float64 {
 	result["download"] = downloadMbps
 	result["upload"] = uploadMbps
 	result["ping"] = ping
-	log.Infof("Speedtest results: %s", result)
+	log.Infof("Speedtest results: %v", result)
 	return result
 }
-- 
2.33.0

