diff -upr csync-0.50.0.orig/src/csync_log.h csync-0.50.0/src/csync_log.h
--- csync-0.50.0.orig/src/csync_log.h	2015-09-29 19:44:49.692758996 +0200
+++ csync-0.50.0/src/csync_log.h	2015-09-29 19:45:48.439175389 +0200
@@ -55,7 +55,7 @@ enum csync_log_priority_e {
 };
 
 #define CSYNC_LOG(priority, ...) \
-  csync_log(priority, __FUNCTION__, __VA_ARGS__)
+  csync_log(priority, __func__, __VA_ARGS__)
 
 void csync_log(int verbosity,
                const char *function,
