diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1cc90bc..f8309dd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -28,7 +28,7 @@ set(CSYNC_LINK_LIBRARIES
   ${SQLITE3_LIBRARIES}
 )
 
-if(HAVE_ICONV AND WITH_ICONV)
+if(WITH_ICONV)
     list(APPEND CSYNC_PRIVATE_INCLUDE_DIRS ${ICONV_INCLUDE_DIR})
     list(APPEND CSYNC_LINK_LIBRARIES ${ICONV_LIBRARIES})
 endif()
diff --git a/src/std/c_string.c b/src/std/c_string.c
index f175734..ccc0644 100644
--- a/src/std/c_string.c
+++ b/src/std/c_string.c
@@ -39,7 +39,7 @@
 #include <windows.h>
 #endif
 
-#if defined(HAVE_ICONV) && defined(WITH_ICONV)
+#if defined(WITH_ICONV)
 # ifdef HAVE_ICONV_H
 #  include <iconv.h>
 # endif
