Fix https://github.com/Alexays/Waybar/issues/698

--- a/resources/config
+++ b/resources/config
@@ -75,7 +75,8 @@
     "clock": {
         // "timezone": "America/New_York",
         "tooltip-format": "<big>{:%Y %B}</big>\n<tt><small>{calendar}</small></tt>",
-        "format-alt": "{:%Y-%m-%d}"
+        "format-alt": "{:%Y-%m-%d}",
+        "locale": "C"
     },
     "cpu": {
         "format": "{usage}% ",
