From b24c8359e3010bd98813ba611745993569ce55c2 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Tue, 8 Oct 2024 10:23:51 -0700
Subject: [PATCH] user/v84: makefile: use libdir when installing

---
 user/v84/Makefile.in | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/user/v84/Makefile.in b/user/v84/Makefile.in
index 0fcefc5..33bff0b 100644
--- a/user/v84/Makefile.in
+++ b/user/v84/Makefile.in
@@ -110,19 +110,19 @@ ifeq ($(WITH_84_SUPPORT),yes)
 	install -d $(DESTDIR)$(localstatedir)/lib/drbd
 	install -d $(DESTDIR)$(localstatedir)/run/drbd
 	install -d $(DESTDIR)$(localstatedir)/lock
-	install -d $(DESTDIR)/lib/drbd/
+	install -d $(DESTDIR)$(LIBDIR)
 	if getent group haclient > /dev/null 2> /dev/null ; then	\
-		install -g haclient -m 4750 drbdsetup-84 $(DESTDIR)/lib/drbd/ ;	\
-		install -m 755 drbdadm-84 $(DESTDIR)/lib/drbd/ ;		\
+		install -g haclient -m 4750 drbdsetup-84 $(DESTDIR)$(LIBDIR)/ ;	\
+		install -m 755 drbdadm-84 $(DESTDIR)$(LIBDIR)/ ;		\
 	else								\
-		install -m 755 drbdsetup-84 $(DESTDIR)/lib/drbd/ ;		\
-		install -m 755 drbdadm-84 $(DESTDIR)/lib/drbd/ ; 		\
+		install -m 755 drbdsetup-84 $(DESTDIR)$(LIBDIR)/ ;		\
+		install -m 755 drbdadm-84 $(DESTDIR)$(LIBDIR)/ ; 		\
 	fi
 endif
 
 uninstall:
-	rm -f $(DESTDIR)/lib/drbd/drbdsetup-84
-	rm -f $(DESTDIR)/lib/drbd/drbdadm-84
+	rm -f $(DESTDIR)$(LIBDIR)/drbdsetup-84
+	rm -f $(DESTDIR)$(LIBDIR)/drbdadm-84
 
 spell:
 	for f in drbdadm_adjust.c drbdadm_main.c drbdadm_parser.c drbdadm_usage_cnt.c drbdsetup.c drbdtool_common.c; do \
-- 
2.47.1

