diff -Naur cups-1.4.0/backend/Makefile cups-1.4.0.new/backend/Makefile
--- cups-1.4.0/backend/Makefile	2009-03-03 20:39:21.000000000 +0100
+++ cups-1.4.0.new/backend/Makefile	2009-09-01 14:00:05.592615359 +0200
@@ -91,6 +91,7 @@
 	done
 	$(RM) $(SERVERBIN)/backend/http
 	$(LN) ipp $(SERVERBIN)/backend/http
+	$(LN) ipp $(SERVERBIN)/backend/https
 	if test "x$(DNSSD_BACKEND)" != x; then \
 		$(RM) $(SERVERBIN)/backend/mdns; \
 		$(LN) $(DNSSD_BACKEND) $(SERVERBIN)/backend/mdns; \
