Author: Steve Kemp <skx@debian.org>
Description: mailsnarf does not parse mail correctly, Closes #149330.
--- a/mailsnarf.c	2011-06-19 17:14:23.839999384 -0500
+++ b/mailsnarf.c	2011-06-19 17:14:59.327999376 -0500
@@ -178,7 +178,7 @@
 	if (smtp->state != SMTP_DATA) {
 		while ((i = buf_index(&buf, "\r\n", 2)) >= 0) {
 			line = buf_tok(&buf, NULL, i + 2);
-			line->base[line->end] = '\0';
+			line->base[line->end-1] = '\0';
 			p = buf_ptr(line);
 			
 			if (strncasecmp(p, "RSET", 4) == 0) {
