Author: Luciano Bello <luciano@linux.org.ar>
Description: distinguish between pop versions
--- a/decode.c	2011-06-19 17:13:48.679999392 -0500
+++ b/decode.c	2011-06-19 17:15:29.771999369 -0500
@@ -63,7 +63,8 @@
 	{ "http",	decode_http },
 	{ "ospf",	decode_ospf },
 	{ "poppass",	decode_poppass },
-	{ "pop",	decode_pop },
+	{ "pop2",	decode_pop },
+	{ "pop3",	decode_pop },
 	{ "nntp",	decode_nntp },
 	{ "smb",	decode_smb },
 	{ "imap",	decode_imap },
--- a/dsniff.services	2011-06-19 17:13:48.883999392 -0500
+++ b/dsniff.services	2011-06-19 17:15:29.799999369 -0500
@@ -10,8 +10,8 @@
 ospf		89/ip
 http		98/tcp
 poppass		106/tcp
-pop		109/tcp
-pop		110/tcp
+pop2		109/tcp
+pop3		110/tcp
 portmap		111/tcp
 portmap		-111/tcp
 portmap		111/udp
