Author: <bdefreese@debian2.bddebian.com>
Description: Correctly 0 out the c struct.
--- a/msgsnarf.c	2011-06-19 17:15:01.091999376 -0500
+++ b/msgsnarf.c	2011-06-19 17:15:39.739999369 -0500
@@ -584,6 +584,7 @@
 		if (i == 0) {
 			if ((c = malloc(sizeof(*c))) == NULL)
 				nids_params.no_mem("sniff_msgs");
+			memset(c, 0, sizeof(*c));
 			c->ip = ts->addr.saddr;
 			c->nick = strdup("unknown");
 			SLIST_INSERT_HEAD(&client_list, c, next);
