--- dsniff-2.3.org/configure	2014-08-12 23:09:56.696284744 +0200
+++ dsniff-2.3/configure	2014-08-12 23:21:46.476285867 +0200
@@ -3035,11 +3035,7 @@
      ;;
   esac 
 else
-   if test -f ${prefix}/include/pcap.h; then
-     PCAPINC="-I${prefix}/include"
-     PCAPLIB="-L${prefix}/lib -lpcap"
-  elif test -f /usr/include/pcap/pcap.h; then
-     PCAPINC="-I/usr/include/pcap"
+  if true ; then
      PCAPLIB="-lpcap"
   elif test -f /usr/include/pcap.h; then
      PCAPLIB="-lpcap"
@@ -3083,12 +3079,7 @@
      ;;
   esac 
 else
-   if test -f ${prefix}/include/libnet.h; then
-     CFLAGS="$CFLAGS `${prefix}/bin/libnet-config --defines`"
-     LNETINC="-I${prefix}/include"
-     LNETLIB="-L${prefix}/lib -lnet"
-  elif test -f /usr/include/libnet.h; then
-     CFLAGS="$CFLAGS `libnet-config --defines`"
+  if true ; then
      LNETLIB="-lnet"
   else
      echo "$ac_t""no" 1>&6
@@ -3128,10 +3119,7 @@
      ;;
   esac 
 else
-   if test -f ${prefix}/include/nids.h; then
-     NIDSINC="-I${prefix}/include"
-     NIDSLIB="-L${prefix}/lib -lnids"
-  elif test -f /usr/include/nids.h; then
+  if true; then
      NIDSLIB="-lnids"
   else
      echo "$ac_t""no" 1>&6
@@ -3193,13 +3181,7 @@
      ;;
   esac 
 else
-   if test -f ${prefix}/include/openssl/ssl.h; then
-     SSLINC="-I${prefix}/include"
-     SSLLIB="-L${prefix}/lib -lssl -lcrypto"
-  elif test -f ${prefix}/ssl/include/openssl/ssl.h; then
-     SSLINC="-I${prefix}/ssl/include"
-     SSLLIB="-L${prefix}/ssl/lib -lssl -lcrypto"
-  elif test -f /usr/include/openssl/ssl.h; then
+  if true ; then
      SSLLIB="-lssl -lcrypto"
   else
      echo "$ac_t""no" 1>&6
