From 75cab07ea0451fc1d2ec319bf3a078fe34ff3fcb Mon Sep 17 00:00:00 2001
From: OPNA2608 <opna2608@protonmail.com>
Date: Fri, 18 Jul 2025 00:49:52 +0200
Subject: [PATCH] CMakeLists.txt: Raise C++ standard to C++17

GTest 1.17.0 demands it.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 13dfd39..dbc8196 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_MODULE_PATH}")
 find_package(PkgConfig REQUIRED)
 include(GNUInstallDirs)
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 ${COMMON_FLAGS} -fno-strict-aliasing -fvisibility-inlines-hidden -Wall -pedantic -Wextra")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 ${COMMON_FLAGS} -fno-strict-aliasing -fvisibility-inlines-hidden -Wall -pedantic -Wextra")
 
 option(Werror "Treat warnings as errors" ON)
 option(enable_tests "Build tests" ON)
-- 
2.53.0

