From 42d04e0d484b3715e7b9935e3ef3e2fa6c33b409 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Mon, 1 Sep 2025 12:43:05 +0200
Subject: [PATCH] CMakeLists.txt: Bump cmake_minimum_required() to version
 3.10.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbc8196..41b6fb4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,5 @@
 project(gmenu-harness C CXX)
-cmake_minimum_required(VERSION 2.8.9)
+cmake_minimum_required(VERSION 3.10)
 
 set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake" "${CMAKE_MODULE_PATH}")
 find_package(PkgConfig REQUIRED)
-- 
2.53.0

