--- a/spec/spec_helper.rb
+++ b/spec/spec_helper.rb
@@ -1,7 +1,3 @@
-require "codeclimate-test-reporter"
-CodeClimate::TestReporter.start
-
-require "pry"
 require "rspec"
 require "hamster/hash"
 require "hamster/set"
--- a/spec/lib/hamster/vector/insert_spec.rb
+++ b/spec/lib/hamster/vector/insert_spec.rb
@@ -1,6 +1,5 @@
 require "spec_helper"
 require "hamster/vector"
-require 'pry'
 
 describe Hamster::Vector do
   describe "#insert" do
@@ -66,4 +65,4 @@
       end
     end
   end
-end
\ No newline at end of file
+end
