From 27a4a167191efb5dbd5612b821da78bde3ce9fad Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=2E=20Neusch=C3=A4fer?= <j.ne@posteo.net>
Date: Thu, 31 Jul 2025 01:06:35 +0000
Subject: [PATCH] Fix build with GCC 15

---
 xl2tpd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xl2tpd.c b/xl2tpd.c
index 791d5a4..41d2f8b 100644
--- a/xl2tpd.c
+++ b/xl2tpd.c
@@ -74,7 +74,7 @@ static int control_handle_lac_hangup(FILE* resf, char* bufp);
 static int control_handle_lac_disconnect(FILE* resf, char* bufp);
 static int control_handle_lac_add_modify(FILE* resf, char* bufp);
 static int control_handle_lac_remove(FILE* resf, char* bufp);
-static int control_handle_lac_status();
+static int control_handle_lac_status(FILE* resf, char* bufp);
 static int control_handle_lns_remove(FILE* resf, char* bufp);
 
 static struct control_requests_handler control_handlers[] = {
@@ -1549,7 +1549,7 @@ static int control_handle_lac_remove(FILE* resf, char* bufp){
     return 1;
 }
 
-static int control_handle_lac_status(){
+static int control_handle_lac_status(FILE*, char*){
     show_status ();
     return 1;
 }
-- 
2.50.1

