--- Makefile.orig
+++ Makefile
@@ -184,7 +184,7 @@
 	*.gz) \
 	    SUF=`echo $(MANSUF)|sed 's/\.gz//'`; \
 	    $(INSTALL) -c -m 0644 nfswatch.8 $(DESTDIR)$(MANDIR)/nfswatch.$$SUF; \
-	    gzip --best $(DESTDIR)$(MANDIR)/nfswatch.$$SUF; \
+	    gzip $(DESTDIR)$(MANDIR)/nfswatch.$$SUF; \
 	    ;; \
 	*) \
 	    $(INSTALL) -c -m 0644 nfswatch.8 $(DESTDIR)$(MANDIR)/nfswatch.$(MANSUF); \
@@ -199,7 +199,7 @@
 	*.gz) \
 	    SUF=`echo $(MANSUF)|sed 's/\.gz//'`; \
 	    $(INSTALL) -c -m 0644 nfslogsum.8 $(DESTDIR)$(MANDIR)/nfslogsum.$$SUF; \
-	    gzip --best $(DESTDIR)$(MANDIR)/nfslogsum.$$SUF; \
+	    gzip $(DESTDIR)$(MANDIR)/nfslogsum.$$SUF; \
 	    ;; \
 	*) \
 	    $(INSTALL) -c -m 0644 nfslogsum.8 $(DESTDIR)$(MANDIR)/nfslogsum.$(MANSUF); \
