--- a/lisp/emacspeak-preamble.el
+++ b/lisp/emacspeak-preamble.el
@@ -53,7 +53,7 @@
 
 ;;;###autoload
 (defvar emacspeak-directory
-  (expand-file-name "../" (file-name-directory load-file-name))
+  (file-name-as-directory "/usr/share/emacs/site-lisp/emacspeak/")
   "emacspeak directory")
 
 ;;;###autoload
@@ -83,15 +83,20 @@
 
 ;;;###autoload
 (defvar emacspeak-info-directory
-  (expand-file-name  "info/" emacspeak-directory)
+  (file-name-as-directory "/usr/share/info/emacspeak/")
   "Info")
 
 ;;;###autoload
+(defvar emacspeak-doc-directory
+  (file-name-as-directory "/usr/share/doc/emacspeak/")
+  "Documentation")
+
+;;;###autoload
 (defvar emacspeak-user-directory (expand-file-name "~/.emacspeak/")
   "Resources.")
 
 (defvar emacspeak-readme-file
-  (expand-file-name "README" emacspeak-directory)
+  (expand-file-name "README.md" emacspeak-doc-directory)
   "README.")
 
 ;;;###autoload
--- a/lisp/emacspeak-wizards.el
+++ b/lisp/emacspeak-wizards.el
@@ -120,12 +120,12 @@
 (defun emacspeak-view-emacspeak-news ()
   "Display emacspeak News for a given version."
   (interactive)
-  (cl-declare (special emacspeak-etc-directory))
+  (cl-declare (special emacspeak-doc-directory))
   (find-file-read-only
    (expand-file-name
     (completing-read "News: "
-                     (directory-files emacspeak-etc-directory nil "NEWS*"))
-    emacspeak-etc-directory))
+                     (directory-files emacspeak-doc-directory nil "NEWS*"))
+    emacspeak-doc-directory))
   (emacspeak-auditory-icon 'news)
   (org-mode)
   (org-next-visible-heading 1)
@@ -134,11 +134,11 @@
 (defun emacspeak-view-emacspeak-tips ()
   "Browse  Emacspeak productivity tips."
   (interactive)
-  (cl-declare (special emacspeak-etc-directory))
+  (cl-declare (special emacspeak-doc-directory))
   (emacspeak-xslt-without-xsl
    (browse-url
     (format "file:///%stips.html"
-            emacspeak-etc-directory)))
+            emacspeak-doc-directory)))
   (emacspeak-auditory-icon 'help)
   (emacspeak-speak-mode-line))
 
--- a/lisp/emacspeak.el
+++ b/lisp/emacspeak.el
@@ -293,8 +293,8 @@
   (cl-declare (special emacspeak-packages-to-prepare))
   (setq-default line-move-visual nil)
   (setq use-dialog-box nil)
-  (when (boundp 'Info-directory-list)
-    (push emacspeak-info-directory Info-directory-list))
+  (when (boundp 'Info-default-directory-list)
+    (push emacspeak-info-directory Info-default-directory-list))
   (mapc
    #'(lambda (pair)
        (emacspeak-do-package-setup  (cl-first pair) (cl-second pair)))
