--- a/servers/espeak
+++ b/servers/espeak
@@ -509,9 +509,7 @@
 if {[info exists server_p]} {
     set tts(input) sock0
 }
-set servers [file normalize [file dirname $argv0]]
-set tclTTS $servers/native-espeak
-load $tclTTS/tclespeak[info sharedlibextension]
+load /usr/lib/emacspeak/tclespeak.so
 
 # Get the list of voice variants available so you can call for en-us:whisper
 # and not en-us:en-us+whisper
