--- a/autoconf/config.guess	2016-08-04 01:13:53.941899947 +0200
+++ b/autoconf/config.guess	2016-08-04 01:16:30.084553757 +0200
@@ -142,6 +142,10 @@
 
 # Note: order is significant - the case branches are not exclusive.
 
+if [ -n "$CBUILD" ]; then
+	echo "$CBUILD"; exit
+fi
+
 case "${UNAME_MACHINE}:${UNAME_SYSTEM}:${UNAME_RELEASE}:${UNAME_VERSION}" in
     *:NetBSD:*:*)
 	# NetBSD (nbsd) targets should (where applicable) match one or
