commit ce5835fb59d8d52354875d73f103527d704b93cd
Author: Jakub Panek <me@panekj.dev>
Date:   Sun Jun 26 16:27:19 2022 +0200
Author: Jakub Jirutka <jakub@jirutka.cz>
Date:   Mon Jul 25 01:56:21 2022 +0200

    fix: link against system libraries

diff --git a/codec/Cargo.toml b/codec/Cargo.toml
index 3cce0dd9a..f4a7f6a7c 100644
--- a/codec/Cargo.toml
+++ b/codec/Cargo.toml
@@ -21,7 +21,7 @@ termwiz = { path = "../termwiz" }
 thiserror = "1.0"
 varbincode = "0.1"
 wezterm-term = { path = "../term", features=["use_serde"] }
-zstd = "0.6"
+zstd = { version = "0.6", features = ["pkg-config"] }
 
 [dev-dependencies]
 base91 = { path = "../base91" }
--- a/config/Cargo.toml
+++ b/config/Cargo.toml
@@ -28 +28 @@
-mlua = {version="0.8.0-beta.4", features=["vendored", "lua54", "async", "send"]}
+mlua = {version="0.8.0-beta.4", features=["lua54", "async", "send"]}
diff --git a/deps/freetype/build.rs b/deps/freetype/build.rs
index d701272cc..af0274603 100644
--- a/deps/freetype/build.rs
+++ b/deps/freetype/build.rs
@@ -233,9 +233,9 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    zlib();
-    libpng();
-    freetype();
+    println!("cargo:rustc-link-lib=z");
+    println!("cargo:rustc-link-lib=png");
+    println!("cargo:rustc-link-lib=freetype");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
diff --git a/deps/harfbuzz/build.rs b/deps/harfbuzz/build.rs
index ba3e8f0fa..1b8528660 100644
--- a/deps/harfbuzz/build.rs
+++ b/deps/harfbuzz/build.rs
@@ -119,7 +119,7 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    harfbuzz();
+    println!("cargo:rustc-link-lib=harfbuzz");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
--- a/window/Cargo.toml
+++ b/window/Cargo.toml
@@ -80 +80 @@
-xcb-imdkit = { version="0.2", git="https://github.com/wez/xcb-imdkit-rs.git", rev="ede7c71b85fe2537efef6cf999a45690316211cf"}
+xcb-imdkit = { version="0.2", git="https://github.com/wez/xcb-imdkit-rs.git", rev="ede7c71b85fe2537efef6cf999a45690316211cf", features=["use-system-lib"]}
--- a/wezterm-ssh/Cargo.toml
+++ b/wezterm-ssh/Cargo.toml
@@ -33 +33 @@
-libssh-rs = {version="0.1.4", features=["vendored"], git="https://github.com/wez/libssh-rs.git", optional = true}
+libssh-rs = {version="0.1.4", git="https://github.com/wez/libssh-rs.git", optional = true}
