commit ce5835fb59d8d52354875d73f103527d704b93cd
Author: Jakub Panek <me@panekj.dev>
Date:   Sun Jun 26 16:27:19 2022 +0200
Author: Jakub Jirutka <jakub@jirutka.cz>
Date:   Mon Jul 25 01:56:21 2022 +0200

    fix: link against system libraries

diff --git a/Cargo.lock b/Cargo.lock
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1964,7 +1964,6 @@
 source = "git+https://github.com/wez/libssh-rs.git#d36e5648274063c187968fd5ce0b1a7aa350a025"
 dependencies = [
  "cc",
- "libz-sys",
  "openssl-sys",
  "pkg-config",
 ]
@@ -5309,6 +5309,7 @@ checksum = "ebd5b733d7cf2d9447e2c3e76a5589b4f5e5ae065c22a2bc0b023cbc331b6c8e"
 dependencies = [
  "cc",
  "libc",
+ "pkg-config",
 ]
 
 [[package]]
diff --git a/codec/Cargo.toml b/codec/Cargo.toml
index 3cce0dd9a..f4a7f6a7c 100644
--- a/codec/Cargo.toml
+++ b/codec/Cargo.toml
@@ -21,7 +21,7 @@ termwiz = { path = "../termwiz" }
 thiserror = "1.0"
 varbincode = "0.1"
 wezterm-term = { path = "../term", features=["use_serde"] }
-zstd = "0.6"
+zstd = { version = "0.6", features = ["pkg-config"] }
 
 [dev-dependencies]
 base91 = { path = "../base91" }
diff --git a/deps/freetype/build.rs b/deps/freetype/build.rs
index d701272cc..af0274603 100644
--- a/deps/freetype/build.rs
+++ b/deps/freetype/build.rs
@@ -233,9 +233,9 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    zlib();
-    libpng();
-    freetype();
+    println!("cargo:rustc-link-lib=z");
+    println!("cargo:rustc-link-lib=png");
+    println!("cargo:rustc-link-lib=freetype");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
diff --git a/deps/harfbuzz/build.rs b/deps/harfbuzz/build.rs
index ba3e8f0fa..1b8528660 100644
--- a/deps/harfbuzz/build.rs
+++ b/deps/harfbuzz/build.rs
@@ -119,7 +119,7 @@ fn git_submodule_update() {
 }
 
 fn main() {
-    harfbuzz();
+    println!("cargo:rustc-link-lib=harfbuzz");
     let out_dir = env::var("OUT_DIR").unwrap();
     println!("cargo:outdir={}", out_dir);
     println!("cargo:rustc-env=MACOSX_DEPLOYMENT_TARGET=10.9");
--- a/wezterm-ssh/Cargo.toml
+++ b/wezterm-ssh/Cargo.toml
@@ -33 +33 @@
-libssh-rs = {version="0.1.4", features=["vendored"], git="https://github.com/wez/libssh-rs.git", optional = true}
+libssh-rs = {version="0.1.4", git="https://github.com/wez/libssh-rs.git", optional = true}
