From 80f81670323a09025ffb7d030be56e47aec5492c Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 6 Jan 2010 12:34:15 +0000
Subject: [PATCH] Support for parallell builds

---
 src/Makefile.am |   12 +++++++-----
 1 files changed, 7 insertions(+), 5 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 6e8a774..49d6b16 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -134,7 +134,7 @@ libmysql_drv_la_LIBADD = $(libmysql_drv_libs) -L. -ldspam
 libmysql_drv_la_CPPFLAGS = $(libmysql_drv_cppflags)
 libmysql_drv_la_LDFLAGS = $(libmysql_drv_ldflags)
 libmysql_drv_la_LDFLAGS += -rpath '$(pkglibdir)' -version-info $(libversion) 
-libmysql_drv_la_DEPENDENCIES =
+libmysql_drv_la_DEPENDENCIES = libdspam.la
 
 libpgsql_drv_la_SOURCES = \
 	pgsql_drv.c pgsql_drv.h 
@@ -142,7 +142,7 @@ libpgsql_drv_la_LIBADD = $(libpgsql_drv_libs) -L. -ldspam
 libpgsql_drv_la_CPPFLAGS = $(libpgsql_drv_cppflags)
 libpgsql_drv_la_LDFLAGS = $(libpgsql_drv_ldflags)
 libpgsql_drv_la_LDFLAGS += -rpath '$(pkglibdir)' -version-info $(libversion)
-libpgsql_drv_la_DEPENDENCIES =
+libpgsql_drv_la_DEPENDENCIES = libdspam.la
 
 libsqlite_drv_la_SOURCES = \
 	sqlite_drv.c sqlite_drv.h 
@@ -150,7 +150,7 @@ libsqlite_drv_la_LIBADD = $(libsqlite_drv_libs) -L. -ldspam
 libsqlite_drv_la_CPPFLAGS = $(libsqlite_drv_cppflags)
 libsqlite_drv_la_LDFLAGS = $(libsqlite_drv_ldflags)
 libsqlite_drv_la_LDFLAGS += -rpath '$(pkglibdir)' -version-info $(libversion)
-libsqlite_drv_la_DEPENDENCIES =
+libsqlite_drv_la_DEPENDENCIES = libdspam.la
 
 libsqlite3_drv_la_SOURCES = \
 	sqlite3_drv.c sqlite3_drv.h 
@@ -158,7 +158,7 @@ libsqlite3_drv_la_LIBADD = $(libsqlite3_drv_libs) -L. -ldspam
 libsqlite3_drv_la_CPPFLAGS = $(libsqlite3_drv_cppflags)
 libsqlite3_drv_la_LDFLAGS = $(libsqlite3_drv_ldflags)
 libsqlite3_drv_la_LDFLAGS += -rpath '$(pkglibdir)' -version-info $(libversion)
-libsqlite3_drv_la_DEPENDENCIES =
+libsqlite3_drv_la_DEPENDENCIES = libdspam.la
 
 libhash_drv_la_SOURCES = \
 	hash_drv.c hash_drv.h
@@ -166,7 +166,7 @@ libhash_drv_la_LIBADD = $(libhash_drv_libs) -L. -ldspam
 libhash_drv_la_CPPFLAGS = $(libhash_drv_cppflags)
 libhash_drv_la_LDFLAGS = $(libhash_drv_ldflags)
 libhash_drv_la_LDFLAGS += -rpath '$(pkglibdir)' -version-info $(libversion)
-libhash_drv_la_DEPENDENCIES =
+libhash_drv_la_DEPENDENCIES = libdspam.la
 
 endif
 
@@ -181,6 +181,7 @@ dspam_SOURCES = \
 	agent_shared.c agent_shared.h
 
 dspam_LDADD = -lm pref.o libdspam.la
+dspam_DEPENDENCIES = pref.o libdspam.la
 
 dspamc_SOURCES = \
 	config.h language.h \
@@ -192,6 +193,7 @@ dspamc_SOURCES = \
 
 dspamc_LDADD = -lm buffer.o config_shared.o error.o read_config.o pref.o nodetree.o util.o $(NETLIBS)
 dspamc_LDFLAGS = 
+dspamc_DEPENDENCIES = buffer.o config_shared.o error.o read_config.o pref.o nodetree.o util.o 
 
 # Failing of chgrp is expected and normal case when someone do
 #   make install DESTDIR=...
-- 
1.6.6

