--- a/src/setuptools_scm/_file_finders/__init__.py
+++ b/src/setuptools_scm/_file_finders/__init__.py
@@ -10,7 +10,7 @@
 from .._entrypoints import iter_entry_points
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeGuard
+    from typing import TypeGuard
 
 
 log = _log.log.getChild("file_finder")
--- a/src/setuptools_scm/_integration/toml.py
+++ b/src/setuptools_scm/_integration/toml.py
@@ -15,7 +15,7 @@
     from tomli import loads as load_toml
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeAlias
+    from typing import TypeAlias
 
 from .. import _log
 
--- a/src/setuptools_scm/_types.py
+++ b/src/setuptools_scm/_types.py
@@ -10,7 +10,7 @@
 
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeAlias
+    from typing import TypeAlias
     from . import version
 
 PathT: TypeAlias = Union["os.PathLike[str]", str]
--- a/src/setuptools_scm/version.py
+++ b/src/setuptools_scm/version.py
@@ -17,8 +17,8 @@
 from . import _modify_version
 
 if TYPE_CHECKING:
-    from typing_extensions import Concatenate
-    from typing_extensions import ParamSpec
+    from typing import Concatenate
+    from typing import ParamSpec
 
     _P = ParamSpec("_P")
 
--- a/testing/conftest.py
+++ b/testing/conftest.py
@@ -8,7 +8,7 @@
 from typing import Iterator
 
 import pytest
-from typing_extensions import Self
+from typing import Self
 
 from .wd_wrapper import WorkDir
 from setuptools_scm._run_cmd import run
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,7 +42,6 @@
   "packaging>=20",
   "setuptools",
   'tomli>=1; python_version < "3.11"',
-  "typing-extensions",
 ]
 [project.optional-dependencies]
 docs = [
