diff --git a/pyproject.toml b/pyproject.toml
index 1d11e4c..b2ee286 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -2,6 +2,5 @@
 requires = [
     "setuptools>=45",
     "packaging>=20.0",
-    "typing_extensions",
 ]
 build-backend = "setuptools.build_meta"
diff --git a/src/setuptools_scm/_entrypoints.py b/src/setuptools_scm/_entrypoints.py
index 2efb9f8..3c1ff5a 100644
--- a/src/setuptools_scm/_entrypoints.py
+++ b/src/setuptools_scm/_entrypoints.py
@@ -12,7 +12,7 @@ from .version import ScmVersion
 
 if TYPE_CHECKING:
     from .config import Configuration
-    from typing_extensions import Protocol
+    from typing import Protocol
     from . import _types as _t
 else:
     Configuration = Any
diff --git a/src/setuptools_scm/_integration/pyproject_reading.py b/src/setuptools_scm/_integration/pyproject_reading.py
index 04801b5..1e102c9 100644
--- a/src/setuptools_scm/_integration/pyproject_reading.py
+++ b/src/setuptools_scm/_integration/pyproject_reading.py
@@ -11,7 +11,7 @@ from typing import TYPE_CHECKING
 from .setuptools import read_dist_name_from_setup_cfg
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeAlias
+    from typing import TypeAlias
 
 _ROOT = "root"
 TOML_RESULT: TypeAlias = Dict[str, Any]
diff --git a/src/setuptools_scm/_types.py b/src/setuptools_scm/_types.py
index 6c6bdf8..6653da5 100644
--- a/src/setuptools_scm/_types.py
+++ b/src/setuptools_scm/_types.py
@@ -12,7 +12,7 @@ if TYPE_CHECKING:
     from setuptools_scm import version
     import os
 
-from typing_extensions import ParamSpec, TypeAlias, Protocol
+from typing import ParamSpec, TypeAlias, Protocol
 
 PathT = Union["os.PathLike[str]", str]
 
diff --git a/src/setuptools_scm/file_finder.py b/src/setuptools_scm/file_finder.py
index f14a946..9f44092 100644
--- a/src/setuptools_scm/file_finder.py
+++ b/src/setuptools_scm/file_finder.py
@@ -4,7 +4,7 @@ import os
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from typing_extensions import TypeGuard
+    from typing import TypeGuard
     from . import _types as _t
 
 from .utils import trace
diff --git a/src/setuptools_scm/version.py b/src/setuptools_scm/version.py
index a53f50b..724a706 100644
--- a/src/setuptools_scm/version.py
+++ b/src/setuptools_scm/version.py
@@ -17,7 +17,7 @@ from typing import Tuple
 from typing import TYPE_CHECKING
 
 if TYPE_CHECKING:
-    from typing_extensions import Concatenate
+    from typing import Concatenate
 
     from . import _types as _t
 
