diff --git a/IGC/CMakeLists.txt b/IGC/CMakeLists.txt
index c0f871718..5939a94b7 100644
--- a/IGC/CMakeLists.txt
+++ b/IGC/CMakeLists.txt
@@ -1825,7 +1825,7 @@ else()
 endif()
 set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES OUTPUT_NAME "${IGC_LIBRARY_NAME}${_cpuSuffix}")
 set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES
-                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}+${IGC_BUILD_METADATA}"
+                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
                       SOVERSION "${IGC_API_MAJOR_VERSION}")
 unset(_cpuSuffix)
 
