diff --git a/IGC/CMakeLists.txt b/IGC/CMakeLists.txt
index 16b315a60..6a3d3a66a 100644
--- a/IGC/CMakeLists.txt
+++ b/IGC/CMakeLists.txt
@@ -1831,7 +1831,7 @@ else()
 endif()
 set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES OUTPUT_NAME "${IGC_LIBRARY_NAME}${_cpuSuffix}")
 set_target_properties("${IGC_BUILD__PROJ__igc_dll}" PROPERTIES
-                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}+${IGC_BUILD_METADATA}"
+                      VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
                       SOVERSION "${IGC_API_MAJOR_VERSION}")
 unset(_cpuSuffix)
 
@@ -1844,7 +1844,7 @@ if (FCL_ENABLED)
   endif()
   set_target_properties("${IGC_BUILD__PROJ__fcl_dll}" PROPERTIES OUTPUT_NAME "${FCL_LIBRARY_NAME}${_cpuSuffix}")
   set_target_properties("${IGC_BUILD__PROJ__fcl_dll}" PROPERTIES
-                        VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}+${IGC_BUILD_METADATA}"
+                        VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
                         SOVERSION "${IGC_API_MAJOR_VERSION}")
   unset(_cpuSuffix)
 endif()
diff --git a/visa/iga/IGALibrary/CMakeLists.txt b/visa/iga/IGALibrary/CMakeLists.txt
index 53f1fe0b9..a80a3a0d8 100644
--- a/visa/iga/IGALibrary/CMakeLists.txt
+++ b/visa/iga/IGALibrary/CMakeLists.txt
@@ -384,7 +384,7 @@ endif(ANDROID AND MEDIA_IGA)
 
   if(IGC_BUILD)
     set_target_properties(IGA_DLL PROPERTIES
-                          VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}+${IGC_BUILD_METADATA}"
+                          VERSION "${IGC_API_MAJOR_VERSION}.${IGC_API_MINOR_VERSION}.${IGC_API_PATCH_VERSION}"
                           SOVERSION "${IGC_API_MAJOR_VERSION}")
   endif()
 
