RTLD_DEEPBIND is a glibc thing
---
--- a/IGC/VectorCompiler/lib/GenXCodeGen/GenXWrapper.cpp
+++ b/IGC/VectorCompiler/lib/GenXCodeGen/GenXWrapper.cpp
@@ -101,7 +101,7 @@
 
 #if defined(__linux__)
   // Hack to workaround cmoc crashes during loading of SPIRV library
-  static auto DeepBindHack = dlopen(SpirvLibName, RTLD_NOW | RTLD_DEEPBIND);
+  static auto DeepBindHack = dlopen(SpirvLibName, RTLD_NOW);
 #endif // __linux__
 
   using DL = sys::DynamicLibrary;
