diff --git a/src/tools/Makefile.in b/src/tools/Makefile.in
index bebf92e..8f3dc1d 100644
--- a/src/tools/Makefile.in
+++ b/src/tools/Makefile.in
@@ -709,14 +709,7 @@ uninstall-am: uninstall-dist_binSCRIPTS
 .PRECIOUS: Makefile
 
 
-pgpool_setup:
-	ln -s ../test/pgpool_setup .
-
-watchdog_setup:
-	ln -s ../test/watchdog_setup .
-
 clean-local:
-	-rm -f $(dist_bin_SCRIPTS)
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
