diff --git a/src/debug/CrashReporter.cpp b/src/debug/CrashReporter.cpp
index 5d80345..46d1913 100644
--- a/src/debug/CrashReporter.cpp
+++ b/src/debug/CrashReporter.cpp
@@ -1,7 +1,6 @@
 #include "CrashReporter.hpp"
 #include <random>
 #include <sys/utsname.h>
-#include <execinfo.h>
 #include <fstream>
 #include <signal.h>
 
@@ -78,13 +77,6 @@ void CrashReporter::createAndSaveCrash(int sig) {
 
     finalCrashReport += "Backtrace:\n";
 
-    void*  bt[1024];
-    size_t btSize;
-    char** btSymbols;
-
-    btSize    = backtrace(bt, 1024);
-    btSymbols = backtrace_symbols(bt, btSize);
-
 #if defined(KERN_PROC_PATHNAME)
     int mib[] = {
         CTL_KERN,
@@ -110,20 +102,6 @@ void CrashReporter::createAndSaveCrash(int sig) {
     const auto FPATH = std::filesystem::canonical("/proc/self/exe");
 #endif
 
-    for (size_t i = 0; i < btSize; ++i) {
-        finalCrashReport += getFormat("\t#%lu | %s\n", i, btSymbols[i]);
-
-#ifdef __clang__
-        const auto CMD = getFormat("llvm-addr2line -e %s -f 0x%lx", FPATH.c_str(), (uint64_t)bt[i]);
-#else
-        const auto CMD = getFormat("addr2line -e %s -f 0x%lx", FPATH.c_str(), (uint64_t)bt[i]);
-#endif
-        const auto ADDR2LINE = replaceInString(execAndGet(CMD.c_str()), "\n", "\n\t\t");
-        finalCrashReport += "\t\t" + ADDR2LINE.substr(0, ADDR2LINE.length() - 2);
-    }
-
-    free(btSymbols);
-
     finalCrashReport += "\n\nLog tail:\n";
 
     finalCrashReport += execAndGet(("cat \"" + Debug::logFile + "\" | tail -n 50").c_str());
