commit ec092bd601d9d351ff6ca34bd97f12055b2a4dd9
Author: Vaxry <vaxry@vaxry.net>
Date:   Tue May 7 14:26:26 2024 +0100

    core: chase hyprwayland-scanner

diff --git a/flake.lock b/flake.lock
index 976553d3..4dc7da82 100644
--- a/flake.lock
+++ b/flake.lock
@@ -84,11 +84,11 @@
         ]
       },
       "locked": {
-        "lastModified": 1714755542,
-        "narHash": "sha256-D0pg+ZRwrt4lavZ97Ca8clsgbPA3duLj8iEM7riaIFY=",
+        "lastModified": 1715088365,
+        "narHash": "sha256-cVH43+fiiCXqr6F2vUA8KmNI9ytWIQoZGnVz0IpBbZw=",
         "owner": "hyprwm",
         "repo": "hyprwayland-scanner",
-        "rev": "1270ebaa539e56d61b708c24b072b09cbbd3a828",
+        "rev": "6119dc2a965fd30ffa45c50a8398d5da3150df4c",
         "type": "github"
       },
       "original": {
diff --git a/src/protocols/OutputManagement.cpp b/src/protocols/OutputManagement.cpp
index 9ebcba07..e7779726 100644
--- a/src/protocols/OutputManagement.cpp
+++ b/src/protocols/OutputManagement.cpp
@@ -166,7 +166,7 @@ void COutputHead::sendAllData() {
                     LOGM(LOG, "  | sending current mode for {}: {}x{}@{}", pMonitor->szName, m->mode->width, m->mode->height, m->mode->refresh);
                 else
                     LOGM(LOG, "  | sending current mode for {}: null (fake)", pMonitor->szName);
-                resource->sendCurrentMode(m->resource->resource());
+                resource->sendCurrentMode(m->resource.get());
                 break;
             }
         }
@@ -197,7 +197,7 @@ void COutputHead::updateMode() {
                     LOGM(LOG, "  | sending current mode for {}: {}x{}@{}", pMonitor->szName, m->mode->width, m->mode->height, m->mode->refresh);
                 else
                     LOGM(LOG, "  | sending current mode for {}: null (fake)", pMonitor->szName);
-                resource->sendCurrentMode(m->resource->resource());
+                resource->sendCurrentMode(m->resource.get());
                 break;
             }
         }
