diff --git a/hugolib/resource_chain_babel_test.go b/hugolib/resource_chain_babel_test.go
index c9ddb21..25072a6 100644
--- a/hugolib/resource_chain_babel_test.go
+++ b/hugolib/resource_chain_babel_test.go
@@ -32,6 +32,7 @@ import (
 )
 
 func TestResourceChainBabel(t *testing.T) {
+	t.Skip("nodejs lol")
 	if !isCI() {
 		t.Skip("skip (relative) long running modules test when running locally")
 	}
diff --git a/hugolib/resource_chain_test.go b/hugolib/resource_chain_test.go
index 67641bd..53f8b09 100644
--- a/hugolib/resource_chain_test.go
+++ b/hugolib/resource_chain_test.go
@@ -841,6 +841,7 @@ Hello2: Bonjour
 }
 
 func TestResourceChainPostCSS(t *testing.T) {
+	t.Skip("nodejs lol")
 	if !isCI() {
 		t.Skip("skip (relative) long running modules test when running locally")
 	}
