From 3a11d22da365434ee47df818371d4f32a397b73d Mon Sep 17 00:00:00 2001
From: Hannes Braun <hannes@hannesbraun.net>
Date: Wed, 26 Feb 2025 21:17:06 +0100
Subject: [PATCH] resources/image: Mark loong64 as FMA-using architecture

---
 resources/images/imagetesting/testing.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/resources/images/imagetesting/testing.go b/resources/images/imagetesting/testing.go
index 25f2ab087..f7a6af7ea 100644
--- a/resources/images/imagetesting/testing.go
+++ b/resources/images/imagetesting/testing.go
@@ -231,4 +231,5 @@ var UsesFMA = runtime.GOARCH == "s390x" ||
 	runtime.GOARCH == "ppc64" ||
 	runtime.GOARCH == "ppc64le" ||
 	runtime.GOARCH == "arm64" ||
-	runtime.GOARCH == "riscv64"
+	runtime.GOARCH == "riscv64" ||
+	runtime.GOARCH == "loong64"
-- 
2.49.0

