diff --git a/resources/resource_transformers/js/js_integration_test.go b/resources/resource_transformers/js/js_integration_test.go
index a11a6fbe9..5e2074ba8 100644
--- a/resources/resource_transformers/js/js_integration_test.go
+++ b/resources/resource_transformers/js/js_integration_test.go
@@ -79,6 +79,7 @@ JS Content:{{ $js.Content }}:End:
 }
 
 func TestBuildWithModAndNpm(t *testing.T) {
+	t.Skip()
 	if !htesting.IsCI() {
 		t.Skip("skip (relative) long running modules test when running locally")
 	}
