diff --git a/Makefile.am b/Makefile.am
index 718041634..820378495 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -501,6 +501,7 @@ AM_CPPFLAGS = \
     -I$(srcdir)/src/sss_client \
     -I$(srcdir)/src \
     -I. \
+    -I/usr/include/nspr \
     $(POPT_CFLAGS) \
     $(TALLOC_CFLAGS) \
     $(TDB_CFLAGS) \
@@ -905,6 +906,7 @@ libsss_child_la_LIBADD = \
     $(TEVENT_LIBS) \
     $(DHASH_LIBS) \
     libsss_debug.la \
+    -lintl \
     $(NULL)
 libsss_child_la_LDFLAGS = -avoid-version
 
@@ -1754,6 +1756,7 @@ sss_signal_SOURCES = \
     $(NULL)
 sss_signal_LDADD = \
     libsss_debug.la \
+    -lintl \
     $(NULL)
 
 sss_override_SOURCES = \
@@ -1912,6 +1915,7 @@ libsss_test_common_la_LIBADD = \
     $(TEVENT_LIBS) \
     $(LDB_LIBS) \
     $(SSSD_INTERNAL_LTLIBS) \
+    -lintl \
     $(NULL)
 
 if HAVE_CHECK
@@ -2659,6 +2663,7 @@ test_authtok_LDADD = \
     $(POPT_LIBS) \
     libsss_test_common.la \
     libsss_debug.la \
+    -lintl \
     $(NULL)
 
 test_prompt_config_SOURCES = \
@@ -4356,7 +4361,8 @@ ldap_child_LDADD = \
     $(TALLOC_LIBS) \
     $(POPT_LIBS) \
     $(DHASH_LIBS) \
-    $(KRB5_LIBS)
+    $(KRB5_LIBS) \
+    -lintl
 
 if BUILD_SEMANAGE
 selinux_child_SOURCES = \
@@ -4447,6 +4453,7 @@ p11_child_LDADD = \
     $(DHASH_LIBS) \
     $(POPT_LIBS) \
     libsss_crypt.la \
+    -lintl \
     $(NULL)
 if HAVE_NSS
 p11_child_LDADD += \
-- 
2.27.0

