From 0c7d5f32de56ee254c4b1abc63e7c3709afd1cb2 Mon Sep 17 00:00:00 2001
From: Vsevolod Stakhov <vsevolod@highsecure.ru>
Date: Sat, 10 Dec 2016 13:21:24 +0000
Subject: [PATCH] [Minor] Try to fix build on non x86 architectures

---
 src/libcryptobox/CMakeLists.txt | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/libcryptobox/CMakeLists.txt b/src/libcryptobox/CMakeLists.txt
index 637bfde..0ca8a2f 100644
--- a/src/libcryptobox/CMakeLists.txt
+++ b/src/libcryptobox/CMakeLists.txt
@@ -32,12 +32,12 @@ SET(ASM_CODE "
 	")
 ASM_OP(HAVE_DOLLARMACRO "dollar macro convention")
 
-IF(NOT HAVE_SLASHMACRO AND NOT HAVE_DOLLARMACRO)
-	MESSAGE(FATAL_ERROR "Your assembler cannot compile macros, please check your CMakeFiles/CMakeError.log")
-ENDIF()
-
 # For now we support only x86_64 architecture with optimizations
 IF("${ARCH}" STREQUAL "x86_64")
+	IF(NOT HAVE_SLASHMACRO AND NOT HAVE_DOLLARMACRO)
+		MESSAGE(FATAL_ERROR "Your assembler cannot compile macros, please check your CMakeFiles/CMakeError.log")
+	ENDIF()
+
 	SET(ASM_CODE "vpaddq %ymm0, %ymm0, %ymm0")
 	ASM_OP(HAVE_AVX2 "avx2")
 	SET(ASM_CODE "vpaddq %xmm0, %xmm0, %xmm0")
From 4e5fae43635e824e1b48f10af0bea7dccee0412d Mon Sep 17 00:00:00 2001
From: Vsevolod Stakhov <vsevolod@highsecure.ru>
Date: Sat, 10 Dec 2016 14:29:22 +0000
Subject: [PATCH] [Minor] Leftover for i386

---
 src/libcryptobox/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libcryptobox/CMakeLists.txt b/src/libcryptobox/CMakeLists.txt
index 0ca8a2f..aef9c9d 100644
--- a/src/libcryptobox/CMakeLists.txt
+++ b/src/libcryptobox/CMakeLists.txt
@@ -61,6 +61,9 @@ IF("${ARCH}" STREQUAL "x86_64")
 
 	SET(BLAKE2SRC ${BLAKE2SRC} ${CMAKE_CURRENT_SOURCE_DIR}/blake2/x86-64.S)
 ELSEIF("${ARCH}" STREQUAL "i386")
+	IF(NOT HAVE_SLASHMACRO AND NOT HAVE_DOLLARMACRO)
+		MESSAGE(FATAL_ERROR "Your assembler cannot compile macros, please check your CMakeFiles/CMakeError.log")
+	ENDIF()
 	SET(POLYSRC ${POLYSRC} ${CMAKE_CURRENT_SOURCE_DIR}/poly1305/ref-32.c)
 	SET(CURVESRC ${CURVESRC} ${CMAKE_CURRENT_SOURCE_DIR}/curve25519/curve25519-donna.c)
 	SET(BLAKE2SRC ${BLAKE2SRC} ${CMAKE_CURRENT_SOURCE_DIR}/blake2/x86-32.S)
