Alpine uses versioned Qt binaries

diff --git a/cmake/modules/QmlTest.cmake b/cmake/modules/QmlTest.cmake
index 8241703..d209b55 100644
--- a/cmake/modules/QmlTest.cmake
+++ b/cmake/modules/QmlTest.cmake
@@ -15,7 +15,7 @@
 # qmltest_DEFAULT_IMPORT_PATHS
 # qmltest_DEFAULT_PROPERTIES
 
-find_program(qmltestrunner_exe qmltestrunner)
+find_program(qmltestrunner_exe qmltestrunner-qt5)
 
 if(NOT qmltestrunner_exe)
   message(FATAL_ERROR "Could not locate qmltestrunner.")
