Alpine uses versioned Qt binaries

diff --git a/cmake/modules/QmlTest.cmake b/cmake/modules/QmlTest.cmake
index 72394f2..1cd52fb 100644
--- a/cmake/modules/QmlTest.cmake
+++ b/cmake/modules/QmlTest.cmake
@@ -20,7 +20,7 @@ if (QT_VERSION_MAJOR GREATER_EQUAL 6)
   set(qmltestrunner_exe Qt${QT_VERSION_MAJOR}::qmltestrunner)
 else()
   # FIXME: how to make sure we get the right version?
-  find_program(qmltestrunner_exe qmltestrunner)
+  find_program(qmltestrunner_exe qmltestrunner-qt5)
 
   if(NOT qmltestrunner_exe)
     message(FATAL_ERROR "Could not locate qmltestrunner.")

