diff -ru a/config-scripts/cups-compiler.m4 b/config-scripts/cups-compiler.m4
--- a/config-scripts/cups-compiler.m4	2020-04-27 14:04:29.000000000 -0400
+++ b/config-scripts/cups-compiler.m4	2021-08-06 20:53:29.316010169 -0400
@@ -11,9 +11,6 @@
 
 dnl Clear the debugging and non-shared library options unless the user asks
 dnl for them...
-INSTALL_STRIP=""
-AC_SUBST(INSTALL_STRIP)
-
 AC_ARG_WITH([optim], AS_HELP_STRING([--with-optim], [set optimization flags]), [
     OPTIM="$withval"
 ], [
@@ -31,7 +28,6 @@
 AS_IF([test x$enable_debug = xyes -a "x$OPTIM" = x], [
     OPTIM="-g"
 ], [
-    INSTALL_STRIP="-s"
 ])
 
 dnl Debug printfs can slow things down, so provide a separate option for that
diff -ru a/configure b/configure
--- a/configure	2020-04-27 14:04:29.000000000 -0400
+++ b/configure	2021-08-06 20:54:01.585892266 -0400
@@ -726,7 +726,6 @@
 ARCHFLAGS
 UNITTESTS
 OPTIM
-INSTALL_STRIP
 LIBTOOL_INSTALL
 LIBTOOL_CXX
 LIBTOOL_CC
@@ -7902,7 +7901,6 @@
 
 
 
-INSTALL_STRIP=""
 
 
 
@@ -7956,10 +7954,6 @@
 
     OPTIM="-g"
 
-else $as_nop
-
-    INSTALL_STRIP="-s"
-
 fi
 
 if test x$enable_debug_printfs = xyes
diff -ru a/Makedefs.in b/Makedefs.in
--- a/Makedefs.in	2020-04-27 14:04:29.000000000 -0400
+++ b/Makedefs.in	2021-08-06 20:53:42.565961754 -0400
@@ -46,12 +46,12 @@
 # Installation programs...
 #
 
-INSTALL_BIN	=	@LIBTOOL_INSTALL@ $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
+INSTALL_BIN	=	@LIBTOOL_INSTALL@ $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@
 INSTALL_COMPDATA =	$(INSTALL) -c -m 444 @INSTALL_GZIP@
 INSTALL_CONFIG	=	$(INSTALL) -c -m @CUPS_CONFIG_FILE_PERM@
 INSTALL_DATA	=	$(INSTALL) -c -m 444
 INSTALL_DIR	=	$(INSTALL) -d
-INSTALL_LIB	=	@LIBTOOL_INSTALL@ $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@ @INSTALL_STRIP@
+INSTALL_LIB	=	@LIBTOOL_INSTALL@ $(INSTALL) -c -m @CUPS_EXE_FILE_PERM@
 INSTALL_MAN	=	$(INSTALL) -c -m 444
 INSTALL_SCRIPT	=	$(INSTALL) -c -m @CUPS_EXE_FILE_PERM@
