diff -up cups-1.5b1/config-scripts/cups-ssl.m4.no-export-ssllibs cups-1.5b1/config-scripts/cups-tls.m4
--- cups-1.6.2/config-scripts/cups-tls.m4.no-export-ssllibs	2011-05-11 02:52:08.000000000 +0200
+++ cups-1.6.2/config-scripts/cups-tls.m4	2011-05-23 17:47:27.000000000 +0200
@@ -107,5 +107,5 @@
 AC_SUBST([TLSFLAGS])
 AC_SUBST([TLSLIBS])
 
-EXPORT_TLSLIBS="$TLSLIBS"
+EXPORT_TLSLIBS=""
 AC_SUBST([EXPORT_TLSLIBS])
