diff --git a/backend/lpd.c b/backend/lpd.c
index 48d1a5f..0a1bd1e 100644
--- a/backend/lpd.c
+++ b/backend/lpd.c
@@ -1269,7 +1269,7 @@ rresvport_af(int *port,			/* IO - Port number to bind to */
     * Try binding the port to the socket; return if all is OK...
     */
 
-    if (!bind(fd, (struct sockaddr *)&addr, httpAddrSize(&addr)))
+    if (!bind(fd, (struct sockaddr *)&addr, httpAddrLength(&addr)))
       return (fd);
 
    /*
