--- ./configure.ac.orig
+++ ./configure.ac
@@ -188,16 +188,6 @@
 AC_MSG_CHECKING([for platform OS])
 AC_MSG_RESULT($platform)
 
-AS_IF([test $framer = "linux"],
-  [ MORELIBS="${MORELIBS} -lm"
-    AC_MSG_CHECKING([for procfs support])
-    if test -r /proc/net/dev; then
-      AC_MSG_RESULT(okay)
-    else
-      AC_MSG_ERROR([You must have ProcFS enabled in your kernel!]);
-    fi],
-  [])
-
 AS_IF([test $framer = "generic"],
   MORELIBS="${MORELIBS} -lwpcap -lpcap -ldnet",
   [])
