diff --git a/src/mod/applications/mod_avmd/avmd_desa2_tweaked.c b/src/mod/applications/mod_avmd/avmd_desa2_tweaked.c
index 6bf0b41..7ef31f7 100644
--- a/src/mod/applications/mod_avmd/avmd_desa2_tweaked.c
+++ b/src/mod/applications/mod_avmd/avmd_desa2_tweaked.c
@@ -16,9 +16,9 @@
     #define ISNAN(x) (!!(_isnan(x)))
     #define ISINF(x) (isinf(x))
 #else
-    int __isnan(double);
-    #define ISNAN(x) (__isnan(x))
-    #define ISINF(x) (__isinf(x))
+    #include <math.h>
+    #define ISNAN(x) (isnan(x))
+    #define ISINF(x) (isinf(x))
 #endif
 
 #include "avmd_buffer.h"
