diff --git a/configure.ac b/configure.ac
index 127ee01..1b2c041 100644
--- a/configure.ac
+++ b/configure.ac
@@ -593,14 +593,14 @@ AC_SUBST(SYS_XMLRPC_CFLAGS)
 AC_SUBST(SYS_XMLRPC_LDFLAGS)
 AM_CONDITIONAL([SYSTEM_XMLRPCC],[test "${enable_xmlrpcc}" = "yes"])
 
-for luaversion in luajit lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
+for luaversion in lua5.3 lua-5.3 lua53 lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
   PKG_CHECK_MODULES([LUA],[${luaversion}],[have_lua=yes],[have_lua=no])
   if test ${have_lua} = yes; then
     break
   fi
 done
 if test x"${LUA_LIBS}" = x"" ; then
-   LUA_LIBS="-llua"
+   LUA_LIBS="-llua5.3"
 fi
 
 AC_ARG_ENABLE(srtp,
