--- a/configure.ac
+++ b/configure.ac
@@ -597,7 +597,7 @@
 AC_SUBST(SYS_XMLRPC_LDFLAGS)
 AM_CONDITIONAL([SYSTEM_XMLRPCC],[test "${enable_xmlrpcc}" = "yes"])
 
-for luaversion in lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
+for luaversion in lua5.3 lua-5.3 lua53 lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
   PKG_CHECK_MODULES([LUA],[${luaversion}],[have_lua=yes],[have_lua=no])
   if test ${have_lua} = yes; then
     break
--- a/configure
+++ b/configure
@@ -20080,7 +20080,7 @@
 fi
 
 
-for luaversion in lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
+for luaversion in lua5.3 lua-5.3 lua53 lua5.2 lua-5.2 lua52 lua5.1 lua-5.1 lua; do
 
   succeeded=no
 
