diff --git a/configure.ac b/configure.ac
index 3eb0a47..2670bbd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -414,13 +414,11 @@ if test "x${ax_cv_c_compiler_vendor}"  =  "xsun" ; then
 elif test "x${ax_cv_c_compiler_vendor}" = "xclang" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC -ffast-math)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC -ffast-math)
-    APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
 elif test "x${ax_cv_c_compiler_vendor}" = "xgnu" ; then
     APR_ADDTO(SWITCH_AM_CFLAGS, -fPIC -ffast-math)
     APR_ADDTO(SWITCH_AM_CXXFLAGS, -fPIC -ffast-math)
     AC_SUBST([AM_MOD_AVMD_CXXFLAGS], [-std=gnu99])      # FS-8809, needed for MAP_POPULATE
     if test "$ac_cv_gcc_supports_w_no_unused_result" = yes; then
-      APR_ADDTO(SWITCH_AM_CFLAGS, -Werror)
       APR_ADDTO(SWITCH_AM_CFLAGS, -Wno-unused-result)
     fi
     if test "$ac_cv_gcc_supports_w_no_misleading_indentation" = yes; then
