diff -ru freeswitch-1.4.26.orig/src/mod/applications/mod_conference/mod_conference.c freeswitch-1.4.26/src/mod/applications/mod_conference/mod_conference.c
--- freeswitch-1.4.26.orig/src/mod/applications/mod_conference/mod_conference.c	2015-11-19 20:59:46.000000000 +0000
+++ freeswitch-1.4.26/src/mod/applications/mod_conference/mod_conference.c	2016-10-25 06:16:53.115953333 +0000
@@ -5051,6 +5051,7 @@
 		/* Wait for the input thread to end */
 		if (member->input_thread) {
 			switch_thread_join(&st, member->input_thread);
+			member->input_thread = NULL;
 		}
 	}
 

