diff --git a/libs/libscgi/src/include/scgi.h b/libs/libscgi/src/include/scgi.h
index cc6b3c0..332a071 100644
--- a/libs/libscgi/src/include/scgi.h
+++ b/libs/libscgi/src/include/scgi.h
@@ -116,7 +116,7 @@ typedef int scgi_filehandle_t;
 #include <sys/types.h>
 #include <sys/select.h>
 #include <netinet/tcp.h>
-#include <sys/signal.h>
+#include <signal.h>
 #include <unistd.h>
 #include <ctype.h>
 #endif
diff --git a/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_os_nw.h b/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_os_nw.h
index ed6cf38..75b73b0 100644
--- a/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_os_nw.h
+++ b/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_os_nw.h
@@ -58,7 +58,7 @@
 #endif
 
 #if SU_HAVE_POLL
-#include <sys/poll.h>
+#include <poll.h>
 #endif
 
 SOFIA_BEGIN_DECLS
diff --git a/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_wait.h b/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_wait.h
index ad67723..7beaae6 100644
--- a/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_wait.h
+++ b/libs/sofia-sip/libsofia-sip-ua/su/sofia-sip/su_wait.h
@@ -47,7 +47,7 @@
 #endif
 
 #if SU_HAVE_POLL
-#include <sys/poll.h>
+#include <poll.h>
 #endif
 
 SOFIA_BEGIN_DECLS
diff --git a/libs/spandsp/src/spandsp/private/timezone.h b/libs/spandsp/src/spandsp/private/timezone.h
index 87c6797..b2397cf 100644
--- a/libs/spandsp/src/spandsp/private/timezone.h
+++ b/libs/spandsp/src/spandsp/private/timezone.h
@@ -30,7 +30,7 @@
 
 #define TZ_MAX_LEAPS            50      /* Maximum number of leap second corrections */
 
-#define TZNAME_MAX              255
+#define SPANDSP_TZNAME_MAX              255
 
 /* The TZ_MAX_TIMES value below is enough to handle a bit more than a
  * year's worth of solar time (corrected daily to the nearest second) or
@@ -74,14 +74,14 @@ struct tz_state_s
     time_t ats[TZ_MAX_TIMES];
     uint8_t types[TZ_MAX_TIMES];
     struct tz_ttinfo_s ttis[TZ_MAX_TYPES];
-    char chars[TZ_BIGGEST(TZ_MAX_CHARS + 1, (2*(TZNAME_MAX + 1)))];
+    char chars[TZ_BIGGEST(TZ_MAX_CHARS + 1, (2*(SPANDSP_TZNAME_MAX + 1)))];
     struct tz_lsinfo_s lsis[TZ_MAX_LEAPS];
 };
 
 struct tz_s
 {
     struct tz_state_s state;
-    char lcl_tzname[TZNAME_MAX + 1];
+    char lcl_tzname[SPANDSP_TZNAME_MAX + 1];
     int lcl_is_set;
     const char *tzname[2];
 };
diff --git a/src/mod/applications/mod_spandsp/mod_spandsp_modem.h b/src/mod/applications/mod_spandsp/mod_spandsp_modem.h
index 8d41cd2..02dc074 100644
--- a/src/mod/applications/mod_spandsp/mod_spandsp_modem.h
+++ b/src/mod/applications/mod_spandsp/mod_spandsp_modem.h
@@ -89,7 +89,7 @@
 	(uint64_t)bswap_32((uint32_t)((value) >> 32)))
 #endif
 #include <sys/time.h>
-#include <sys/signal.h>
+#include <signal.h>
 #endif
 #include <sys/types.h>
 #include <sys/stat.h>
