--- ./src/mod/formats/mod_shout/Makefile.in.orig
+++ ./src/mod/formats/mod_shout/Makefile.in
@@ -111,7 +111,7 @@
 am__installdirs = "$(DESTDIR)$(moddir)"
 LTLIBRARIES = $(mod_LTLIBRARIES)
 mod_shout_la_DEPENDENCIES = $(switch_builddir)/libfreeswitch.la \
-	$(LAMELA) $(SHOUTLA) $(MPG123LA)
+	$(SHOUTLA) $(MPG123LA)
 am_mod_shout_la_OBJECTS = mod_shout_la-mod_shout.lo
 mod_shout_la_OBJECTS = $(am_mod_shout_la_OBJECTS)
 AM_V_lt = $(am__v_lt_@AM_V@)
@@ -434,10 +434,10 @@
 mod_LTLIBRARIES = mod_shout.la
 mod_shout_la_SOURCES = mod_shout.c
 mod_shout_la_CFLAGS = $(AM_CFLAGS)
-mod_shout_la_CPPFLAGS = $(CURL_CFLAGS) $(AM_CPPFLAGS) -I$(SHOUT_DIR)/include -I$(LAME_DIR)/include -I$(MPG123_DIR)/src  -I$(MPG123_BUILDDIR)/src -I$(SHOUT_BUILDDIR)/include
-mod_shout_la_LIBADD = $(switch_builddir)/libfreeswitch.la $(LAMELA) $(SHOUTLA) $(MPG123LA)
-mod_shout_la_LDFLAGS = $(CURL_LIBS) -avoid-version -module -no-undefined -shared `cat $(SHOUT_BUILDDIR)/Makefile | grep SPEEX_LIBS | sed -e 's|^SPEEX_LIBS =||'`  `cat $(SHOUTLA) | grep dependency_libs  | sed -e 's|^dependency_libs=||' | sed -e "s|'||g"`
-BUILT_SOURCES = $(LAMELA) $(SHOUTLA) $(MPG123LA)
+mod_shout_la_CPPFLAGS = $(CURL_CFLAGS) $(AM_CPPFLAGS) -I$(SHOUT_DIR)/include -I/usr/include/lame -I$(MPG123_DIR)/src  -I$(MPG123_BUILDDIR)/src -I$(SHOUT_BUILDDIR)/include
+mod_shout_la_LIBADD = $(switch_builddir)/libfreeswitch.la $(SHOUTLA) $(MPG123LA)
+mod_shout_la_LDFLAGS = $(CURL_LIBS) -lmp3lame -avoid-version -module -no-undefined -shared `cat $(SHOUT_BUILDDIR)/Makefile | grep SPEEX_LIBS | sed -e 's|^SPEEX_LIBS =||'`  `cat $(SHOUTLA) | grep dependency_libs  | sed -e 's|^dependency_libs=||' | sed -e "s|'||g"`
+BUILT_SOURCES = $(SHOUTLA) $(MPG123LA)
 all: $(BUILT_SOURCES)
 	$(MAKE) $(AM_MAKEFLAGS) all-am
 
