--- ./libtransmission/crypto-utils-openssl.c.orig
+++ ./libtransmission/crypto-utils-openssl.c
@@ -147,7 +147,7 @@
 ****
 ***/
 
-#if OPENSSL_VERSION_NUMBER < 0x0090802fL
+#if OPENSSL_VERSION_NUMBER < 0x0090802fL || defined(LIBRESSL_VERSION_NUMBER)
 
 static EVP_CIPHER_CTX *
 openssl_evp_cipher_context_new (void)
@@ -230,7 +230,7 @@
 ****
 ***/
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(LIBRESSL_VERSION_NUMBER)
 static inline int
 DH_set0_pqg (DH     * dh,
              BIGNUM * p,
