--- ./libtransmission/platform-quota.c.orig	2013-07-22 13:15:39.734489641 +0000
+++ ./libtransmission/platform-quota.c	2013-07-22 13:16:24.671640558 +0000
@@ -239,8 +239,6 @@
       spaceused = (int64_t) dq.dqb_curblocks >> 1;
 #elif defined(SYS_DARWIN)
       spaceused = (int64_t) dq.dqb_curbytes;
-#elif defined(__UCLIBC__)
-      spaceused = (int64_t) btodb(dq.dqb_curblocks);
 #elif defined(__sun) || (_LINUX_QUOTA_VERSION < 2)
       spaceused = (int64_t) dq.dqb_curblocks >> 1;
 #else

