diff --git a/src/go/plugins/system/uname/uname_uint8.go b/src/go/plugins/system/uname/uname_uint8.go
index 56868f9..6418627 100644
--- a/src/go/plugins/system/uname/uname_uint8.go
+++ b/src/go/plugins/system/uname/uname_uint8.go
@@ -1,5 +1,5 @@
-//go:build (linux && arm) || (linux && ppc64le) || (linux && s390x)
-// +build linux,arm linux,ppc64le linux,s390x
+//go:build (linux && arm) || (linux && ppc64le) || (linux && s390x) || (linux && riscv64)
+// +build linux,arm linux,ppc64le linux,s390x linux,riscv64
 
 /*
 ** Zabbix
