diff --git a/src/go/plugins/system/uname/uname_int8.go b/src/go/plugins/system/uname/uname_int8.go
index 2f09660..f1db573 100644
--- a/src/go/plugins/system/uname/uname_int8.go
+++ b/src/go/plugins/system/uname/uname_int8.go
@@ -1,5 +1,5 @@
-//go:build (linux && 386) || (linux && amd64) || (linux && arm64)
-// +build linux,386 linux,amd64 linux,arm64
+//go:build (linux && 386) || (linux && amd64) || (linux && arm64) || (linux && riscv64)
+// +build linux,386 linux,amd64 linux,arm64 linux,riscv64
 
 /*
 ** Zabbix
