diff --git a/src/go/conf/zabbix_agent2.conf b/src/go/conf/zabbix_agent2.conf
index c635f2069d..d75b925d72 100644
--- a/src/go/conf/zabbix_agent2.conf
+++ b/src/go/conf/zabbix_agent2.conf
@@ -27,7 +27,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agent2.log
+LogFile=/var/log/zabbix/zabbix_agent2.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -124,7 +124,7 @@ ServerActive=127.0.0.1
 # Default:
 # Hostname=
 
-Hostname=Zabbix server
+# Hostname=Zabbix server
 
 ### Option: HostnameItem
 #	Item used for generating Hostname if it is undefined. Ignored if Hostname is defined.
@@ -132,7 +132,7 @@ Hostname=Zabbix server
 #
 # Mandatory: no
 # Default:
-# HostnameItem=system.hostname
+HostnameItem=system.hostname
 
 ### Option: HostMetadata
 #	Optional parameter that defines host metadata.
@@ -247,7 +247,7 @@ Hostname=Zabbix server
 # Default:
 # ControlSocket=
 
-ControlSocket=/tmp/agent.sock
+ControlSocket=/run/zabbix/zabbix_agent2.sock
 
 ####### TLS-RELATED PARAMETERS #######
 
