--- a/Cargo.lock
+++ b/Cargo.lock
@@ -2587,8 +2587,7 @@
 [[package]]
 name = "pkg-config"
 version = "0.3.27"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "26072860ba924cbfa98ea39c8c19b4dd6a4a25423dbdf219c1eca91aa0cf6964"
+source = "git+https://github.com/selfisekai/pkg-config-rs?branch=pkgc2-breaking#4ea97bced7657de9f93b8ba151ac5029e6346ecc"
 
 [[package]]
 name = "png"
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,6 +56,7 @@
 
 [patch.crates-io]
 "curve25519-dalek" = { git = 'https://github.com/Schmiddiii/curve25519-dalek', rev = '1b9f81352bb659999a46af961f069d635ca432d3' }
+pkg-config = { git = "https://github.com/selfisekai/pkg-config-rs", branch = "pkgc2-breaking" }
 
 [patch."https://github.com/signalapp/curve25519-dalek"]
 "curve25519-dalek" = { git = 'https://github.com/Schmiddiii/curve25519-dalek', rev = '1b9f81352bb659999a46af961f069d635ca432d3' }
