diff --git a/src/meson.build b/src/meson.build
index f82dd01..b37dbed 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -23,7 +23,7 @@ cargo_options = [ '--manifest-path', meson.project_source_root() / 'Cargo.toml'
 cargo_options += [ '--target-dir', meson.project_build_root() / 'target' ]
 
 if get_option('profile') == 'default'
-  cargo_options += [ '--release' ]
+  cargo_options += [ '--release', '--frozen' ]
   rust_target = 'release'
   message('Building in release mode')
 elif get_option('profile') == 'screenshot'
@@ -35,8 +35,6 @@ else
   message('Building in debug mode')
 endif
 
-cargo_env = [ 'CARGO_HOME=' + meson.project_build_root() / 'cargo-home' ]
-
 cargo_build = custom_target(
   'cargo-build',
   build_by_default: true,
@@ -47,8 +45,6 @@ cargo_build = custom_target(
   install_dir: bindir,
   depends: resources,
   command: [
-    'env',
-    cargo_env,
     cargo, 'build',
     cargo_options,
     '&&',
