--- a/Makefile
+++ b/Makefile
@@ -71,7 +71,7 @@
 DEFS += -DVERSION='"$(VERSION)"'
 DEFS += -DTOP_DATADIR='"$(PKGDATADIR)"'
 DEFS += -DHISCOREDIR='"$(STATEDIR)"'
-DEFS += -DENABLE_NLS=1
+DEFS += -DENABLE_NLS=0
 DEFS += -DLOCALEDIR='"$(LOCALEDIR)"'
 
 FILES_H := $(wildcard src/*.h)
--- a/Makefile
+++ b/Makefile
@@ -518,7 +518,7 @@
 # TODO dist and windist..

 .PHONY: install
-install: toppler $(TRANSLATIONFILES_INST)
+install: toppler
	$(INSTALL) -m755 -d $(DESTDIR)$(PKGDATADIR)
	$(INSTALL) -m755 -d $(DESTDIR)$(BINDIR)
	$(INSTALL) -m755 -d $(DESTDIR)$(MANDIR)/man6
