--- a/Makefile	2025-10-24 00:00:00.000000000 +0000
+++ b/Makefile	2025-10-24 00:00:00.000000000 +0000
@@ -48,7 +48,10 @@
 .DELETE_ON_ERROR:
 
 .PHONY: default
-default: all
+default: toppler-only
+
+.PHONY: toppler-only
+toppler-only: $(CROSS)toppler$(EXEEXT)
 
 VERSION := $(shell cat src/version)
 ALL_SOURCES := $(wildcard src/*)
--- a/Makefile	2025-10-24 00:00:00.000000000 +0000
+++ b/Makefile	2025-10-24 00:00:00.000000000 +0000
@@ -510,9 +510,6 @@
 _build/toppler.pot: $(FILES_CPP) $(MISSIONNAMESFILES)
 	xgettext --c++ -o - -k_ -kN_ $(FILES_CPP) $(MISSIONNAMESFILES) > _build/toppler.pot
 
-src/po/%.po: _build/toppler.pot
-	msgmerge -U $@ _build/toppler.pot
-
 
 #---------------------------------------------------------------------------------------#
 # packaging and installation stuff                                                      #
--- a/Makefile
+++ b/Makefile
@@ -518,7 +518,7 @@
 # TODO dist and windist..
 
 .PHONY: install
-install: toppler.dat toppler $(TRANSLATIONFILES_INST)
+install: toppler $(TRANSLATIONFILES_INST)
 	$(INSTALL) -m755 -d $(DESTDIR)$(PKGDATADIR)
 	$(INSTALL) -m755 -d $(DESTDIR)$(BINDIR)
 	$(INSTALL) -m755 -d $(DESTDIR)$(MANDIR)/man6
