--- a/modules/lsp-runtime-lib/src/main/ipc/Library.cpp	2026-06-14 05:20:21.883388051 +0200
+++ b/modules/lsp-runtime-lib/src/main/ipc/Library.cpp	2026-06-14 05:19:31.129851205 +0200
@@ -73,13 +73,13 @@
                 if (str == NULL)
                     return STATUS_NO_MEM;
 
-            #ifdef _GNU_SOURCE
+            #if defined(_GNU_SOURCE) && defined(__GLIBC__)
                 void * const handle    = (new_namespace) ?
                     ::dlmopen(LM_ID_NEWLM, str, RTLD_NOW) :
                     ::dlopen(str, RTLD_NOW);
             #else
                 void * const handle    = ::dlopen(str, RTLD_NOW);
-            #endif /* _GNU_SOURCE */
+            #endif /* defined(_GNU_SOURCE) && defined(__GLIBC__) */
                 if (handle == NULL)
                 {
                     lsp_warn("Error loading module %s: %s", path->get_native(), ::dlerror());
