diff --git a/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/common/libpath.h b/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/common/libpath.h
index 9571a2e..2c484df 100644
--- a/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/common/libpath.h
+++ b/modules/lsp-plugin-fw/include/lsp-plug.in/plug-fw/wrap/common/libpath.h
@@ -241,7 +241,7 @@ namespace lsp
         // Open file for reading
         FILE *fd = fopen("/proc/self/maps", "r");
         if (fd == NULL)
-            return NULL;
+            return false;
 
         char *line      = NULL;
         ssize_t len     = 0;
