We don't need systemd files
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,5 +39,5 @@
 endif()
 
 install(TARGETS grommunio-index RUNTIME)
-install(FILES grommunio-index.service grommunio-index.timer DESTINATION ${UNITDIR})
-install(FILES system-user-groindex.conf DESTINATION ${SYSUSERDIR})
+#install(FILES grommunio-index.service grommunio-index.timer DESTINATION ${UNITDIR})
+#install(FILES system-user-groindex.conf DESTINATION ${SYSUSERDIR})
