--- a/mini_mime.gemspec
+++ b/mini_mime.gemspec
@@ -15,9 +15,7 @@
   spec.homepage      = "https://github.com/discourse/mini_mime"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
-    f.match(%r{^(test|spec|features)/})
-  end
+  spec.files         = Dir["lib/**/*"]
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
