diff --git a/Makefile b/Makefile
index 82221e99156d6042e22d2ecbae6558c41c1253c7..145555a4d8bfac4d052bef1074b092ab39ef168a 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ lint-lua: $(patsubst %.lua.in,%.lua,$(wildcard */*/*.lua.in))
 .PHONY: all install check clean doc info lint
 
 # Dependencies
-KNOT_MINVER := 2.7.2
+KNOT_MINVER := 2.8.0
 $(eval $(call find_lib,libknot,$(KNOT_MINVER),yes))
 $(eval $(call find_lib,libdnssec,$(KNOT_MINVER),yes))
 $(eval $(call find_lib,libzscanner,$(KNOT_MINVER),yes))
diff --git a/daemon/lua/kres-gen.lua b/daemon/lua/kres-gen.lua
index 5e40a610283f60456763859cac45139cfd29f74f..3d5345f7b68870d34297f7729d8a26a9264277d1 100644
--- a/daemon/lua/kres-gen.lua
+++ b/daemon/lua/kres-gen.lua
@@ -395,6 +395,7 @@ struct zs_scanner {
 		_Bool automatic;
 		void (*record)(zs_scanner_t *);
 		void (*error)(zs_scanner_t *);
+		void (*comment)(zs_scanner_t *);
 		void *data;
 	} process;
 	struct {
