new re2 requires c++14 or greater, but 17 for string_view
--
diff --git a/src/modules/extra/m_regex_re2.cpp b/src/modules/extra/m_regex_re2.cpp
index 19a826f..b9c4328 100644
--- a/src/modules/extra/m_regex_re2.cpp
+++ b/src/modules/extra/m_regex_re2.cpp
@@ -17,7 +17,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-/// $CompilerFlags: -std=c++11 find_compiler_flags("re2" "")
+/// $CompilerFlags: -std=c++17 find_compiler_flags("re2" "")
 /// $LinkerFlags: find_linker_flags("re2" "-lre2")
 
 /// $PackageInfo: require_system("arch") pkgconf re2
