new re2 requires c++14 or greater, but 17 for string_view
--
diff --git a/src/modules/extra/m_regex_re2.cpp b/src/modules/extra/m_regex_re2.cpp
index a554605..4ecd3d3 100644
--- a/src/modules/extra/m_regex_re2.cpp
+++ b/src/modules/extra/m_regex_re2.cpp
@@ -17,7 +17,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-/// $CompilerFlags: -std=c++11 find_compiler_flags("re2" "")
+/// $CompilerFlags: -std=c++17 find_compiler_flags("re2" "")
 /// $CompilerFlags: require_version("re2" "11") -std=c++17
 
 /// $LinkerFlags: find_linker_flags("re2" "-lre2")
