From 34c454562fb6ee91fa880aaf1f6f780fa405c81b Mon Sep 17 00:00:00 2001
From: firefly-cpp <iztok@iztok-jr-fister.eu>
Date: Tue, 20 Sep 2022 09:42:47 +0200
Subject: [PATCH] Remove pytest-runner

---
 setup.py | 1 -
 1 file changed, 1 deletion(-)

diff --git a/setup.py b/setup.py
index a94a030..996872d 100644
--- a/setup.py
+++ b/setup.py
@@ -21,6 +21,5 @@
     package_data={
         # "": ["README"],
     },
-    setup_requires=["pytest-runner"],
     tests_require=["pytest"],
 )
