These tests are failing on build-3-20-x86_64

--- a/Makefile.in
+++ b/Makefile.in
@@ -2547,7 +2547,6 @@
 	tests/datamash-pair-tests.pl \
 	tests/datamash-check-tabular.pl \
 	tests/datamash-output-format.pl \
-	tests/datamash-i18n-de.pl \
 	tests/datamash-sort-header.pl \
 	tests/datamash-sort-header-deprecated.pl \
 	tests/datamash-sort-errors.sh \
@@ -2556,7 +2555,6 @@
 	tests/datamash-strbin.sh \
 	tests/datamash-valgrind.sh \
 	tests/decorate-tests.pl \
-	tests/decorate-errors.pl \
 	tests/decorate-sort-tests.pl
 
 
