diff --git a/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp b/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp
index 23c5083..54c7ecf 100644
--- a/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp
+++ b/src/3rdParty/salomesmesh/src/DriverSTL/Basics_Utils.cpp
@@ -29,7 +29,6 @@
 #ifndef WIN32
 #include <unistd.h>
 #include <sys/stat.h>
-#include <execinfo.h>
 #endif
 
 
@@ -109,20 +108,7 @@ namespace Kernel_Utils
 #ifndef WIN32
   void print_traceback()
   {
-    void *array[50];
-    size_t size;
-    char **strings;
-    size_t i;
-
-    size = backtrace (array, 40);
-    strings = backtrace_symbols (array, size);
-
-    for (i = 0; i < size; i++)
-      {
-        std::cerr << strings[i] << std::endl;
-      }
-
-    free (strings);
+    std::cerr << "there is no backtrace." << std::endl;
   }
 #else
   #if (_MSC_VER >= 1400) // Visual Studio 2005
diff --git a/src/App/Application.cpp b/src/App/Application.cpp
index 3081623..ba3525f 100644
--- a/src/App/Application.cpp
+++ b/src/App/Application.cpp
@@ -1732,7 +1732,6 @@ static void freecadNewHandler ()
 #endif
 
 #if defined(FC_OS_LINUX)
-#include <execinfo.h>
 #include <dlfcn.h>
 #include <cxxabi.h>
 
