--- ./cMake/FreeCAD_Helpers/ConfigureCMakeVariables.cmake
+++ ./cMake/FreeCAD_Helpers/ConfigureCMakeVariables.cmake
@@ -23,7 +23,7 @@
         "Path to the directory containing PyCXX's cxxextensions.c source file")
 
     # used as compiler defines
-    set(RESOURCEDIR "${CMAKE_INSTALL_DATADIR}")
+    set(RESOURCEDIR "../../share/freecad")
     set(DOCDIR "${CMAKE_INSTALL_DOCDIR}")
 
     message(STATUS "prefix: ${CMAKE_INSTALL_PREFIX}")
