diff -urN lua-evdev-evdev-2.2.1.orig/evdev/core.c lua-evdev-evdev-2.2.1/evdev/core.c
--- lua-evdev-evdev-2.2.1.orig/evdev/core.c	2020-07-22 01:58:24.039086727 -0600
+++ lua-evdev-evdev-2.2.1/evdev/core.c	2020-07-22 01:58:48.622532514 -0600
@@ -99,7 +99,7 @@
 	}
 
 	/* return: timestamp, event type, event code, event value */
-	lua_pushnumber(L, evt.time.tv_sec + evt.time.tv_usec/1000000.0);
+	lua_pushnumber(L, evt.input_event_sec + evt.input_event_usec/1000000.0);
 	lua_pushinteger(L, evt.type);
 	lua_pushinteger(L, evt.code);
 	lua_pushinteger(L, evt.value);
