diff --git a/wk-we-adblock/build.rs b/wk-we-adblock/build.rs
index f945e71..7153e04 100644
--- a/wk-we-adblock/build.rs
+++ b/wk-we-adblock/build.rs
@@ -5,7 +5,7 @@ use std::env;
 use std::path::PathBuf;
 
 fn main() {
-    let webkit2gtk = pkg_config::probe_library("webkit2gtk-4.0").unwrap();
+    let webkit2gtk = pkg_config::probe_library("webkit2gtk-4.1").unwrap();
     let gtk = pkg_config::probe_library("gtk+-3.0").unwrap();
 
     println!("cargo:rerun-if-changed=wrapper.h");
