From 0126e85796d645820a7883a5f133b52c1408d53c Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Mon, 21 Sep 2015 12:16:16 +0300
Subject: [PATCH 01/14] httpd.conf: ServerRoot

---
 docs/conf/httpd.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/conf/httpd.conf.in b/docs/conf/httpd.conf.in
index 37d7c0b..3e21599 100644
--- a/docs/conf/httpd.conf.in
+++ b/docs/conf/httpd.conf.in
@@ -28,7 +28,7 @@
 # same ServerRoot for multiple httpd daemons, you will need to change at
 # least PidFile.
 #
-ServerRoot "@@ServerRoot@@"
+ServerRoot /var/www
 
 #
 # Mutex: Allows you to set the mutex mechanism and mutex file directory
-- 
2.9.4

