From 537f190142ce17f01db06d0f9cdd4c8b163eee9c Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Fri, 11 Sep 2015 11:27:24 +0300
Subject: [PATCH 11/14] httpd.conf: IncludeOptional

---
 docs/conf/httpd.conf.in | 46 +++-------------------------------------------
 1 file changed, 3 insertions(+), 43 deletions(-)

diff --git a/docs/conf/httpd.conf.in b/docs/conf/httpd.conf.in
index 47fe513..545e663 100644
--- a/docs/conf/httpd.conf.in
+++ b/docs/conf/httpd.conf.in
@@ -373,50 +373,10 @@ LogLevel warn
 #EnableMMAP off
 #EnableSendfile on
 
-# Supplemental configuration
 #
-# The configuration files in the @rel_sysconfdir@/extra/ directory can be 
-# included to add extra features or to modify the default configuration of 
-# the server, or you may simply copy their contents here and change as 
-# necessary.
-
-# Server-pool management (MPM specific)
-#Include @rel_sysconfdir@/extra/httpd-mpm.conf
-
-# Multi-language error messages
-#Include @rel_sysconfdir@/extra/httpd-multilang-errordoc.conf
-
-# Fancy directory listings
-#Include @rel_sysconfdir@/extra/httpd-autoindex.conf
-
-# Language settings
-#Include @rel_sysconfdir@/extra/httpd-languages.conf
-
-# User home directories
-#Include @rel_sysconfdir@/extra/httpd-userdir.conf
-
-# Real-time info on requests and configuration
-#Include @rel_sysconfdir@/extra/httpd-info.conf
-
-# Virtual hosts
-#Include @rel_sysconfdir@/extra/httpd-vhosts.conf
-
-# Local access to the Apache HTTP Server Manual
-#Include @rel_sysconfdir@/extra/httpd-manual.conf
-
-# Distributed authoring and versioning (WebDAV)
-#Include @rel_sysconfdir@/extra/httpd-dav.conf
-
-# Various default settings
-#Include @rel_sysconfdir@/extra/httpd-default.conf
-
-# Configure mod_proxy_html to understand HTML4/XHTML1
-<IfModule proxy_html_module>
-Include @rel_sysconfdir@/extra/proxy-html.conf
-</IfModule>
-
-# Secure (SSL/TLS) connections
-#Include @rel_sysconfdir@/extra/httpd-ssl.conf
+# Load config files from the config directory "/etc/apache2/conf.d".
+#
+IncludeOptional /etc/apache2/conf.d/*.conf
 
 #
 # uncomment out the below to deal with user agents that deliberately
-- 
2.5.0

