From 867d4c6caac66cb458316b97cd24761f339861ca Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Fri, 11 Sep 2015 14:59:32 +0300
Subject: [PATCH 12/14] httpd.conf: MIMEMagicFile

---
 docs/conf/httpd.conf.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/docs/conf/httpd.conf.in b/docs/conf/httpd.conf.in
index 66d20fe..4266f87 100644
--- a/docs/conf/httpd.conf.in
+++ b/docs/conf/httpd.conf.in
@@ -341,7 +341,9 @@ LogLevel warn
 # contents of the file itself to determine its type.  The MIMEMagicFile
 # directive tells the module where the hint definitions are located.
 #
-#MIMEMagicFile @rel_sysconfdir@/magic
+<IfModule mime_magic_module>
+    MIMEMagicFile @rel_sysconfdir@/magic
+</IfModule>
 
 #
 # Customizable error responses come in three flavors:
-- 
2.5.0

