diff --git a/communicate.c b/communicate.c
index 56eba7a..646836d 100644
--- a/communicate.c
+++ b/communicate.c
@@ -576,7 +576,7 @@ void send_get_fakem(struct fake_msg *buf)
                &fm,
                sizeof(fm)-sizeof(fm.mtype),0,0);
 
-      ptr = &fm;
+      ptr = (uint8_t*)&fm;
       for (k=0; k<16; k++) {
         magic_candidate = *(uint32_t*)&ptr[k];
         if (magic_candidate == FAKEROOT_MAGIC_LE || magic_candidate == FAKEROOT_MAGIC_BE) {
diff --git a/faked.c b/faked.c
index e0d060a..9e05c27 100644
--- a/faked.c
+++ b/faked.c
@@ -1090,7 +1090,7 @@ void get_msg()
   do {
     r=msgrcv(msg_get,&fm,sizeof(struct fake_msg_buf),0,0);
 
-    ptr = &fm;
+    ptr = (uint8_t*)&fm;
     for (k=0; k<16; k++) {
       magic_candidate = *(uint32_t*)&ptr[k];
       if (magic_candidate == FAKEROOT_MAGIC_LE || magic_candidate == FAKEROOT_MAGIC_BE) {
