--- fakeroot-1.18.4.orig/configure.ac	2012-06-02 18:23:41.000000000 +0000
+++ fakeroot-1.18.4configure.ac	2015-09-24 23:15:56.917470073 +0000
@@ -301,7 +301,7 @@
   FUNC=`echo $SEARCH|sed -e 's/.*%//'`
   PRE=`echo $SEARCH|sed -e 's/%.*//'`
   FOUND=
-  for WRAPPED in __${PRE}x${FUNC} _${PRE}x${FUNC} __${PRE}${FUNC}13 ${PRE}${FUNC}; do
+  for WRAPPED in ${PRE}${FUNC}; do
     AC_CHECK_FUNCS($WRAPPED,FOUND=$WRAPPED)
 dnl
 dnl to unconditionally define only the _* functions, comment out the 2 lines above,
