--- fakeroot-1.18.4.orig/configure	2012-06-02 18:28:46.000000000 +0000
+++ fakeroot-1.18.4/configure	2015-09-24 21:46:28.804464060 +0000
@@ -12577,7 +12577,7 @@
   FUNC=`echo $SEARCH|sed -e 's/.*%//'`
   PRE=`echo $SEARCH|sed -e 's/%.*//'`
   FOUND=
-  for WRAPPED in __${PRE}x${FUNC} _${PRE}x${FUNC} __${PRE}${FUNC}13 ${PRE}${FUNC}; do
+  for WRAPPED in ${PRE}${FUNC}; do
     for ac_func in $WRAPPED
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
--- fakeroot-1.18.4.orig/configure.ac	2012-06-02 18:23:41.000000000 +0000
+++ fakeroot-1.18.4configure.ac	2015-09-24 23:15:56.917470073 +0000
@@ -301,7 +301,7 @@
   FUNC=`echo $SEARCH|sed -e 's/.*%//'`
   PRE=`echo $SEARCH|sed -e 's/%.*//'`
   FOUND=
-  for WRAPPED in __${PRE}x${FUNC} _${PRE}x${FUNC} __${PRE}${FUNC}13 ${PRE}${FUNC}; do
+  for WRAPPED in ${PRE}${FUNC}; do
     AC_CHECK_FUNCS($WRAPPED,FOUND=$WRAPPED)
 dnl
 dnl to unconditionally define only the _* functions, comment out the 2 lines above,
