diff --git a/pueue/src/client/query/filters.rs b/pueue/src/client/query/filters.rs
index 76c1218..5714187 100644
--- a/pueue/src/client/query/filters.rs
+++ b/pueue/src/client/query/filters.rs
@@ -1,3 +1,4 @@
+#![allow(bindings_with_variant_name)]
 use anyhow::{bail, Context, Result};
 use chrono::{DateTime, Duration, Local, NaiveDate, NaiveDateTime, NaiveTime};
 use pest::iterators::Pair;
diff --git a/pueue/src/client/query/order_by.rs b/pueue/src/client/query/order_by.rs
index 2280afa..982c234 100644
--- a/pueue/src/client/query/order_by.rs
+++ b/pueue/src/client/query/order_by.rs
@@ -1,3 +1,4 @@
+#![allow(bindings_with_variant_name)]
 use anyhow::Result;
 use pest::iterators::Pair;
 
