diff --git a/src/tscore/unit_tests/test_BufferWriterFormat.cc b/src/tscore/unit_tests/test_BufferWriterFormat.cc
index 99a0ee3..c7494de 100644
--- a/src/tscore/unit_tests/test_BufferWriterFormat.cc
+++ b/src/tscore/unit_tests/test_BufferWriterFormat.cc
@@ -512,7 +512,7 @@ TEST_CASE("bwstring std formats", "[libts][bwprint]")
   w.print("{}", ts::bwf::Errno(13));
   REQUIRE(w.view() == "EACCES: Permission denied [13]"sv);
   w.reset().print("{}", ts::bwf::Errno(134));
-  REQUIRE(w.view().substr(0, 22) == "Unknown: Unknown error"sv);
+  REQUIRE(w.view().substr(0, 29) == "Unknown: No error information"sv);
 
   time_t t = 1528484137;
   // default is GMT
