diff --git a/include/tscore/ink_hrtime.h b/include/tscore/ink_hrtime.h
index a1405a9..9f113da 100644
--- a/include/tscore/ink_hrtime.h
+++ b/include/tscore/ink_hrtime.h
@@ -35,6 +35,7 @@
 #include <ctime>
 #include <sys/time.h>
 #include <cstdlib>
+#include <cstdint>
 typedef int64_t ink_hrtime;
 
 int squid_timestamp_to_buf(char *buf, unsigned int buf_size, long timestamp_sec, long timestamp_usec);
