API changed in miniupnpc 2.2.8

--- a/core/network/miniupnp.cpp
+++ b/core/network/miniupnp.cpp
@@ -38,7 +38,7 @@
 		INFO_LOG(MODEM, "UPnP discover failed: error %d", error);
 		return false;
 	}
-	error = UPNP_GetValidIGD(devlist, &urls, &data, lanAddress, sizeof(lanAddress));
+	error = UPNP_GetValidIGD(devlist, &urls, &data, lanAddress, sizeof(lanAddress), nullptr, 0);
 	freeUPNPDevlist(devlist);
 	if (error != 1)
 	{
