--- flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/hw/maple/maple_devs.cpp.orig
+++ flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/hw/maple/maple_devs.cpp
@@ -8,7 +8,7 @@
 #include <math.h>
 #include <time.h>
 
-#include "deps/zlib/zlib.h"
+#include <zlib.h>
 #include "deps/xxhash/xxhash.h"
 
 #define LOGJVS(...) DEBUG_LOG(JVS, __VA_ARGS__)
--- flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/archive/ZipArchive.h.orig
+++ flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/archive/ZipArchive.h
@@ -22,7 +22,7 @@
 #ifndef CORE_ARCHIVE_ZIPARCHIVE_H_
 #define CORE_ARCHIVE_ZIPARCHIVE_H_
 #include "archive.h"
-#include "deps/libzip/zip.h"
+#include <zip.h>
 
 class ZipArchive : public Archive
 {
--- flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/network/miniupnp.cpp.orig
+++ flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/core/network/miniupnp.cpp
@@ -16,8 +16,8 @@
     You should have received a copy of the GNU General Public License
     along with Flycast.  If not, see <https://www.gnu.org/licenses/>.
 */
-#include <miniupnpc.h>
-#include <upnpcommands.h>
+#include <miniupnpc/miniupnpc.h>
+#include <miniupnpc/upnpcommands.h>
 #include "types.h"
 #include "miniupnp.h"
 
--- flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/Makefile.common.orig
+++ flycast-6f27e7121d2b9f2c084d18219a9e439c403a1b89/Makefile.common
@@ -9,6 +9,7 @@
 					-I$(LIBRETRO_COMM_DIR)/include \
 					-I$(DEPS_DIR)/vixl \
 					-I$(DEPS_DIR)/stb
+LIBS += -lminiupnpc
 
 SOURCES_C   ?=
 SOURCES_CXX := \
@@ -169,62 +170,6 @@
 					$(DEPS_DIR)/libchdr/src/libchdr_flac.c \
 					$(DEPS_DIR)/libchdr/src/libchdr_huffman.c \
 
-LIBZIP_SOURCES_C = \
-					$(DEPS_DIR)/libzip/mkstemp.c \
-					$(DEPS_DIR)/libzip/zip_add.c \
-					$(DEPS_DIR)/libzip/zip_add_dir.c \
-					$(DEPS_DIR)/libzip/zip_close.c \
-					$(DEPS_DIR)/libzip/zip_delete.c \
-					$(DEPS_DIR)/libzip/zip_dirent.c \
-					$(DEPS_DIR)/libzip/zip_entry_free.c \
-					$(DEPS_DIR)/libzip/zip_entry_new.c \
-					$(DEPS_DIR)/libzip/zip_error.c \
-					$(DEPS_DIR)/libzip/zip_error_clear.c \
-					$(DEPS_DIR)/libzip/zip_error_get.c \
-					$(DEPS_DIR)/libzip/zip_error_get_sys_type.c \
-					$(DEPS_DIR)/libzip/zip_error_strerror.c \
-					$(DEPS_DIR)/libzip/zip_error_to_str.c \
-					$(DEPS_DIR)/libzip/zip_err_str.c \
-					$(DEPS_DIR)/libzip/zip_fclose.c \
-					$(DEPS_DIR)/libzip/zip_file_error_clear.c \
-					$(DEPS_DIR)/libzip/zip_file_error_get.c \
-					$(DEPS_DIR)/libzip/zip_file_get_offset.c \
-					$(DEPS_DIR)/libzip/zip_filerange_crc.c \
-					$(DEPS_DIR)/libzip/zip_file_strerror.c \
-					$(DEPS_DIR)/libzip/zip_fopen.c \
-					$(DEPS_DIR)/libzip/zip_fopen_index.c \
-					$(DEPS_DIR)/libzip/zip_fread.c \
-					$(DEPS_DIR)/libzip/zip_free.c \
-					$(DEPS_DIR)/libzip/zip_get_archive_comment.c \
-					$(DEPS_DIR)/libzip/zip_get_archive_flag.c \
-					$(DEPS_DIR)/libzip/zip_get_file_comment.c \
-					$(DEPS_DIR)/libzip/zip_get_name.c \
-					$(DEPS_DIR)/libzip/zip_get_num_files.c \
-					$(DEPS_DIR)/libzip/zip_memdup.c \
-					$(DEPS_DIR)/libzip/zip_name_locate.c \
-					$(DEPS_DIR)/libzip/zip_new.c \
-					$(DEPS_DIR)/libzip/zip_open.c \
-					$(DEPS_DIR)/libzip/zip_rename.c \
-					$(DEPS_DIR)/libzip/zip_replace.c \
-					$(DEPS_DIR)/libzip/zip_set_archive_comment.c \
-					$(DEPS_DIR)/libzip/zip_set_archive_flag.c \
-					$(DEPS_DIR)/libzip/zip_set_file_comment.c \
-					$(DEPS_DIR)/libzip/zip_set_name.c \
-					$(DEPS_DIR)/libzip/zip_source_buffer.c \
-					$(DEPS_DIR)/libzip/zip_source_file.c \
-					$(DEPS_DIR)/libzip/zip_source_filep.c \
-					$(DEPS_DIR)/libzip/zip_source_free.c \
-					$(DEPS_DIR)/libzip/zip_source_function.c \
-					$(DEPS_DIR)/libzip/zip_source_zip.c \
-					$(DEPS_DIR)/libzip/zip_stat.c \
-					$(DEPS_DIR)/libzip/zip_stat_index.c \
-					$(DEPS_DIR)/libzip/zip_stat_init.c \
-					$(DEPS_DIR)/libzip/zip_strerror.c \
-					$(DEPS_DIR)/libzip/zip_unchange_all.c \
-					$(DEPS_DIR)/libzip/zip_unchange_archive.c \
-					$(DEPS_DIR)/libzip/zip_unchange.c \
-					$(DEPS_DIR)/libzip/zip_unchange_data.c
-
 PICOTCP_INCFLAGS = \
 					-I$(DEPS_DIR)/picotcp/include \
 					-I$(DEPS_DIR)/picotcp/modules
@@ -256,18 +201,6 @@
 
 XXHASH_SOURCES_C = $(DEPS_DIR)/xxhash/xxhash.c
 
-ZLIB_SOURCES_C = \
-					$(DEPS_DIR)/zlib/deflate.c \
-					$(DEPS_DIR)/zlib/uncompr.c \
-					$(DEPS_DIR)/zlib/zutil.c \
-					$(DEPS_DIR)/zlib/inffast.c \
-					$(DEPS_DIR)/zlib/crc32.c \
-					$(DEPS_DIR)/zlib/inflate.c \
-					$(DEPS_DIR)/zlib/infback.c \
-					$(DEPS_DIR)/zlib/inftrees.c \
-					$(DEPS_DIR)/zlib/trees.c \
-					$(DEPS_DIR)/zlib/adler32.c
-
 SOURCES_ASM :=
 
 ifeq ($(SYSTEM_LIBZIP), 1)
@@ -298,7 +231,6 @@
 endif
 
 ifeq ($(ENABLE_MODEM), 1)
-	INCFLAGS    += -I$(CORE_DIR)/core/deps/miniupnpc
 	SOURCES_CXX += $(CORE_DIR)/core/hw/modem/dns.cpp \
 						$(CORE_DIR)/core/hw/modem/modem.cpp \
 						$(CORE_DIR)/core/hw/bba/bba.cpp \
@@ -306,23 +238,7 @@
 						$(CORE_DIR)/core/network/picoppp.cpp \
 						$(CORE_DIR)/core/network/miniupnp.cpp
 
-	SOURCES_C += \
-					 $(CORE_DIR)/core/deps/miniupnpc/addr_is_reserved.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/igd_desc_parse.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/upnpreplyparse.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/upnpcommands.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/upnperrors.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/connecthostport.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/portlistingparse.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/receivedata.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/upnpdev.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/minissdpc.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/miniwget.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/miniupnpc.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/minixml.c \
-					 $(CORE_DIR)/core/deps/miniupnpc/minisoap.c
-
-	CORE_DEFINES += -DENABLE_MODEM -DMINIUPNP_STATICLIB
+	CORE_DEFINES += -DENABLE_MODEM
 endif
 
 ifeq ($(HAVE_GL), 1)
