Upstream: Should be
Reason: Fixes compilation with gcc-10

diff --git a/src/legacy/i810/i810.h b/src/legacy/i810/i810.h
index 347188c..19be049 100644
--- a/src/legacy/i810/i810.h
+++ b/src/legacy/i810/i810.h
@@ -322,6 +322,6 @@ extern void I810InitMC(ScreenPtr pScreen);
 extern const OptionInfoRec *I810AvailableOptions(int chipid, int busid);
 
 extern const int I810CopyROP[16];
-const int I810PatternROP[16];
+extern const int I810PatternROP[16];
 
 #endif /* _I810_H_ */
