diff --git a/blist/_blist.c b/blist/_blist.c
index f7fcb60..7728448 100644
--- a/blist/_blist.c
+++ b/blist/_blist.c
@@ -3989,10 +3989,8 @@ blist_richcompare_list(PyBList *v, PyListObject *w, int op)
                 /* Shortcut: if the lengths differ, the lists differ */
                 PyObject *res;
                 if (op == Py_EQ) {
-                false:
                         res = Py_False;
                 } else {
-                true:
                         res = Py_True;
                 }
                 Py_INCREF(res);
@@ -4015,8 +4013,8 @@ blist_richcompare_list(PyBList *v, PyListObject *w, int op)
                         ITER_CLEANUP();
                         return _ob(NULL);
                 } else if (!cmp) {
-                        if (op == Py_EQ) { ITER_CLEANUP(); goto false; }
-                        if (op == Py_NE) { ITER_CLEANUP(); goto true; }
+                        if (op == Py_EQ) { ITER_CLEANUP(); Py_INCREF(Py_False); return _ob(Py_False); }
+                        if (op == Py_NE) { ITER_CLEANUP(); Py_INCREF(Py_True); return _ob(Py_True); }
 
                         /* Last RichComparebool may have modified the list */
                         if (i >= PyList_GET_SIZE(w)) {
@@ -4053,8 +4051,8 @@ blist_richcompare_list(PyBList *v, PyListObject *w, int op)
                 return _ob(NULL);
         }
 
-        if (cmp) goto true;
-        else goto false;
+        if (cmp) { Py_INCREF(Py_True); return _ob(Py_True); }
+        else { Py_INCREF(Py_False); return _ob(Py_False); }
 }
 #endif
 
