diff --git a/src/nginx/wok.conf b/src/nginx/wok.conf
index 7fadf38..560d956 100644
--- a/src/nginx/wok.conf
+++ b/src/nginx/wok.conf
@@ -3,7 +3,8 @@
 # If you need to change it for any reason, please, validate /etc/wok/wok.conf
 #
 
-client_max_body_size 4194304k;
+#Check nginx.conf which already have this directive
+#client_max_body_size ${max_body_size}k;
 
 # Set timeout, based on configuration values, to avoid the 504 Gateway Timeout
 # when Wok is processing a request.
