--- a/src/wok/pushserver.py
+++ b/src/wok/pushserver.py
@@ -67,7 +67,7 @@
     def set_socket_file(self):
         if not os.path.isdir(BASE_DIRECTORY):
             try:
-                os.mkdir(BASE_DIRECTORY)
+                os.makedirs(BASE_DIRECTORY)
             except OSError:
                 raise RuntimeError('PushServer base UNIX socket dir %s '
                                    'not found.' % BASE_DIRECTORY)
