diff --git a/setup.py b/setup.py
index aef5c4e..adc4065 100755
--- a/setup.py
+++ b/setup.py
@@ -72,9 +72,6 @@ setup(name='cpplint',
       description='Automated checker to ensure C++ files follow Google\'s style guide',
       long_description=open('README.rst').read(),
       license='BSD-3-Clause',
-      setup_requires=[
-          "pytest-runner==5.2"
-      ],
       tests_require=test_required,
       # extras_require allow pip install .[dev]
       extras_require={
