diff -upr b/src/game/CPlayerManager.h a/src/game/CPlayerManager.h
--- b/src/game/CPlayerManager.h	2021-07-21 17:34:18.436826042 +0200
+++ a/src/game/CPlayerManager.h	2021-07-21 17:35:06.237594868 +0200
@@ -15,6 +15,7 @@
 #include <SDL2/SDL.h>
 #include <map>
 #include <deque>
+#include <string>
 
 enum {
     kLNotConnected,
diff -upr b/vendor/nanogui/nanogui/common.h a/vendor/nanogui/nanogui/common.h
--- b/vendor/nanogui/nanogui/common.h	2021-07-21 17:34:18.550159448 +0200
+++ a/vendor/nanogui/nanogui/common.h	2021-07-21 17:34:35.390170327 +0200
@@ -28,6 +28,8 @@
 #include <stdint.h>
 #include <array>
 #include <vector>
+#include <string>
+#include <stdexcept>
 #include <SDL2/SDL.h>
 
 #include <glm/vec3.hpp> // glm::vec3
