From d88e32f1b05ad287a8d612e6f61ed7432fc72957 Mon Sep 17 00:00:00 2001
From: Karthikeyan Singaravelan <tir.karthi@gmail.com>
Date: Mon, 10 Aug 2020 10:06:14 +0530
Subject: [PATCH] Remove encoding parameter from json.load for Python 3.9
 compatibility. (#151)

---
 tests/test_features.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_features.py b/tests/test_features.py
index 529de70..e2b56e9 100644
--- a/tests/test_features.py
+++ b/tests/test_features.py
@@ -82,7 +82,7 @@ def test_feature_class(self):
                 ' "properties": {"summary": "The first feature",'
                 ' "link": "http://example.org/features/1",'
                 ' "title": "Feature 1"}}')
-        feature = geojson.loads(json, object_hook=factory, encoding="utf-8")
+        feature = geojson.loads(json, object_hook=factory)
         self.assertEqual(repr(type(feature)),
                          "<class 'geojson.examples.SimpleWebFeature'>")
         self.assertEqual(feature.id, '1')
