diff --git a/configure.ac b/configure.ac
index cdaa5da..b11d9cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -221,8 +221,8 @@ AC_CHECK_FUNCS([MD5_Init EVP_get_digestbyname])
 ################################################################
 ## Includes
 
-m4_include([src/dfxml/src/dfxml_configure.m4])
-m4_include([src/be13_api/be13_configure.m4])
+#m4_include([src/dfxml/src/dfxml_configure.m4])
+#m4_include([src/be13_api/be13_configure.m4])
 
 ################################################################
 # PTHREAD support
