diff --git a/requests/certs.py b/requests/certs.py
index d1a378d7..4e0bffd4 100644
--- a/requests/certs.py
+++ b/requests/certs.py
@@ -12,7 +12,8 @@ If you are packaging Requests, e.g., for a Linux distribution or a managed
 environment, you can change the definition of where() to return a separately
 packaged CA bundle.
 """
-from certifi import where
+def where():
+    return "/etc/ssl/certs/ca-certificates.crt"
 
 if __name__ == '__main__':
     print(where())
