diff -auN a/libcrystalhd_int_if.cpp b/libcrystalhd_int_if.cpp
--- a/libcrystalhd_int_if.cpp	2012-11-05 19:25:34.000000000 +0000
+++ b/libcrystalhd_int_if.cpp	2020-04-17 11:34:32.000000000 +0000
@@ -33,7 +33,11 @@
 #include "libcrystalhd_int_if.h"
 #include "libcrystalhd_fwcmds.h"
 
-#include <emmintrin.h>
+#if defined(__aarch64__)
+       #include "sse2neon.h"
+#else
+       #include <emmintrin.h>
+#endif
 
 #define SV_MAX_LINE_SZ 128
 #define PCI_GLOBAL_CONTROL MISC2_GLOBAL_CTRL
