--- a/contrib/src/dhtnet/rules.mak
+++ b/contrib/src/dhtnet/rules.mak
@@ -19,7 +19,8 @@
 DHTNET_CONF = -DBUILD_SHARED_LIBS=Off \
 	-DBUILD_BENCHMARKS=Off \
 	-DBUILD_TOOLS=Off \
-	-DBUILD_TESTING=Off
+	-DBUILD_TESTING=Off \
+	-DBUILD_DEPENDENCIES=Off
 
 $(TARBALLS)/dhtnet-$(DHTNET_VERSION).tar.gz:
 	$(call download,$(DHTNET_URL))
@@ -30,8 +31,7 @@
 
 dhtnet: dhtnet-$(DHTNET_VERSION).tar.gz
 	mkdir -p $(UNPACK_DIR)
-	$(UNPACK) -C $(UNPACK_DIR)
-	$(APPLY) $(SRC)/dhtnet/libjami-testable.patch
+	$(UNPACK) -C $(UNPACK_DIR)/../
 	$(MOVE)
 
 .dhtnet: dhtnet toolchain.cmake .sum-dhtnet
