diff --git a/src/archiver.h b/src/archiver.h
index c71267f..b511808 100644
--- a/src/archiver.h
+++ b/src/archiver.h
@@ -22,6 +22,7 @@
 
 #include "noncopyable.h"
 
+#include <cstdint>
 #include <string>
 #include <vector>
 #include <map>
diff --git a/src/jami/trace-tools.h b/src/jami/trace-tools.h
index 2420a76..7e3a9e7 100644
--- a/src/jami/trace-tools.h
+++ b/src/jami/trace-tools.h
@@ -34,6 +34,7 @@
 
 #ifdef HAVE_CXXABI_H
 #include <cxxabi.h>
+#include <string>
 
 template<typename T>
 std::string demangle()
diff --git a/src/scheduled_executor.h b/src/scheduled_executor.h
index 7316214..badb897 100644
--- a/src/scheduled_executor.h
+++ b/src/scheduled_executor.h
@@ -29,6 +29,7 @@
 #include <mutex>
 #include <condition_variable>
 #include <ciso646>
+#include <string>
 
 #include "noncopyable.h"
 
diff --git a/src/string_utils.h b/src/string_utils.h
index 0197f55..bb0ad3c 100644
--- a/src/string_utils.h
+++ b/src/string_utils.h
@@ -21,6 +21,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <vector>
 #include <set>
