Author: Milan P. Stanić <mps@arvanta.net>
Date: 2025-06-30 20:52:32.000000000 +0000

fix build DtmfDecoderTest.cpp on Alpine Linux

--- a/src/svxlink/trx/DtmfDecoderTest.cpp	2025-06-30 20:52:32.000000000 +0000
+++ b/src/svxlink/trx/DtmfDecoderTest.cpp	2025-08-02 08:11:27.286535268 +0000
@@ -12,6 +12,7 @@
 
 #include "DtmfDecoder.h"
 #include "DtmfEncoder.h"
+#include <cstdint>
 
 using namespace std;
 using namespace Async;
