diff --git a/bindings/python/time64.i b/bindings/python/time64.i
index 93d9f26..1d68c05 100644
--- a/bindings/python/time64.i
+++ b/bindings/python/time64.i
@@ -149,8 +149,8 @@
                                                  t.tm_mday, t.tm_hour, t.tm_min,
                                                  t.tm_sec, 0);
 
-      $result = SWIG_Python_AppendOutput($result, tp);
-  } else $result = SWIG_Python_AppendOutput($result, Py_None);
+      $result = SWIG_AppendOutput($result, tp);
+  } else $result = SWIG_AppendOutput($result, Py_None);
 }
 
 %apply time64 *date { time64 *last_date };
