diff -ru a/dlib/interposition.c b/dlib/interposition.c
--- a/dlib/interposition.c	2020-11-11 11:11:56.000000000 -0500
+++ b/dlib/interposition.c	2021-08-07 12:47:57.546368583 -0400
@@ -214,7 +214,11 @@
 #if SOCKS_CLIENT
 { SYMBOL_ACCEPT,               LIBRARY_ACCEPT,         NULL,   NULL, NULL },
 { SYMBOL_BIND,                 LIBRARY_BIND,           NULL,   NULL, NULL },
+
+#if HAVE_BINDRESVPORT
 { SYMBOL_BINDRESVPORT,         LIBRARY_BINDRESVPORT,   NULL,   NULL, NULL },
+#endif
+
 { SYMBOL_CONNECT,              LIBRARY_CONNECT,        NULL,   NULL, NULL },
 { SYMBOL_GETPEERNAME,          LIBRARY_GETPEERNAME,    NULL,   NULL, NULL },
 { SYMBOL_GETSOCKNAME,          LIBRARY_GETSOCKNAME,    NULL,   NULL, NULL },
