--- a/configure
+++ b/configure
@@ -22093,7 +22093,7 @@
 {
 
 #ifdef EAI_ADDRFAMILY
-gaierrval: EAI_ADDRFAMILY
+"gaierrval: EAI_ADDRFAMILY"
 #else
 #error "gai error value EAI_ADDRFAMILY not defined"
 #endif
@@ -22127,7 +22127,7 @@
 {
 
 #ifdef EAI_AGAIN
-gaierrval: EAI_AGAIN
+"gaierrval: EAI_AGAIN"
 #else
 #error "gai error value EAI_AGAIN not defined"
 #endif
@@ -22161,7 +22161,7 @@
 {
 
 #ifdef EAI_BADFLAGS
-gaierrval: EAI_BADFLAGS
+"gaierrval: EAI_BADFLAGS"
 #else
 #error "gai error value EAI_BADFLAGS not defined"
 #endif
@@ -22195,7 +22195,7 @@
 {
 
 #ifdef EAI_BADHINTS
-gaierrval: EAI_BADHINTS
+"gaierrval: EAI_BADHINTS"
 #else
 #error "gai error value EAI_BADHINTS not defined"
 #endif
@@ -22229,7 +22229,7 @@
 {
 
 #ifdef EAI_FAIL
-gaierrval: EAI_FAIL
+"gaierrval: EAI_FAIL"
 #else
 #error "gai error value EAI_FAIL not defined"
 #endif
@@ -22263,7 +22263,7 @@
 {
 
 #ifdef EAI_FAMILY
-gaierrval: EAI_FAMILY
+"gaierrval: EAI_FAMILY"
 #else
 #error "gai error value EAI_FAMILY not defined"
 #endif
@@ -22297,7 +22297,7 @@
 {
 
 #ifdef EAI_MEMORY
-gaierrval: EAI_MEMORY
+"gaierrval: EAI_MEMORY"
 #else
 #error "gai error value EAI_MEMORY not defined"
 #endif
@@ -22331,7 +22331,7 @@
 {
 
 #ifdef EAI_NODATA
-gaierrval: EAI_NODATA
+"gaierrval: EAI_NODATA"
 #else
 #error "gai error value EAI_NODATA not defined"
 #endif
@@ -22365,7 +22365,7 @@
 {
 
 #ifdef EAI_NONAME
-gaierrval: EAI_NONAME
+"gaierrval: EAI_NONAME"
 #else
 #error "gai error value EAI_NONAME not defined"
 #endif
@@ -22399,7 +22399,7 @@
 {
 
 #ifdef EAI_OVERFLOW
-gaierrval: EAI_OVERFLOW
+"gaierrval: EAI_OVERFLOW"
 #else
 #error "gai error value EAI_OVERFLOW not defined"
 #endif
@@ -22433,7 +22433,7 @@
 {
 
 #ifdef EAI_PROTOCOL
-gaierrval: EAI_PROTOCOL
+"gaierrval: EAI_PROTOCOL"
 #else
 #error "gai error value EAI_PROTOCOL not defined"
 #endif
@@ -22467,7 +22467,7 @@
 {
 
 #ifdef EAI_SERVICE
-gaierrval: EAI_SERVICE
+"gaierrval: EAI_SERVICE"
 #else
 #error "gai error value EAI_SERVICE not defined"
 #endif
@@ -22501,7 +22501,7 @@
 {
 
 #ifdef EAI_SOCKTYPE
-gaierrval: EAI_SOCKTYPE
+"gaierrval: EAI_SOCKTYPE"
 #else
 #error "gai error value EAI_SOCKTYPE not defined"
 #endif
@@ -22535,7 +22535,7 @@
 {
 
 #ifdef EAI_SYSTEM
-gaierrval: EAI_SYSTEM
+"gaierrval: EAI_SYSTEM"
 #else
 #error "gai error value EAI_SYSTEM not defined"
 #endif
@@ -22570,7 +22570,7 @@
 {
 
 #ifdef EAI_ALLDONE
-gaierrval: EAI_ALLDONE
+"gaierrval: EAI_ALLDONE"
 #else
 #error "gai error value EAI_ALLDONE not defined"
 #endif
@@ -22604,7 +22604,7 @@
 {
 
 #ifdef EAI_CANCELED
-gaierrval: EAI_CANCELED
+"gaierrval: EAI_CANCELED"
 #else
 #error "gai error value EAI_CANCELED not defined"
 #endif
@@ -22638,7 +22638,7 @@
 {
 
 #ifdef EAI_IDN_ENCODE
-gaierrval: EAI_IDN_ENCODE
+"gaierrval: EAI_IDN_ENCODE"
 #else
 #error "gai error value EAI_IDN_ENCODE not defined"
 #endif
@@ -22672,7 +22672,7 @@
 {
 
 #ifdef EAI_INPROGRESS
-gaierrval: EAI_INPROGRESS
+"gaierrval: EAI_INPROGRESS"
 #else
 #error "gai error value EAI_INPROGRESS not defined"
 #endif
@@ -22706,7 +22706,7 @@
 {
 
 #ifdef EAI_INTR
-gaierrval: EAI_INTR
+"gaierrval: EAI_INTR"
 #else
 #error "gai error value EAI_INTR not defined"
 #endif
@@ -22740,7 +22740,7 @@
 {
 
 #ifdef EAI_NOTCANCELED
-gaierrval: EAI_NOTCANCELED
+"gaierrval: EAI_NOTCANCELED"
 #else
 #error "gai error value EAI_NOTCANCELED not defined"
 #endif
@@ -22774,7 +22774,7 @@
 {
 
 #ifdef EAI_BADEXTFLAGS
-gaierrval: EAI_BADEXTFLAGS
+"gaierrval: EAI_BADEXTFLAGS"
 #else
 #error "gai error value EAI_BADEXTFLAGS not defined"
 #endif
