diff --git a/doc/Makefile.am b/doc/Makefile.am
index a864d46..1c15fc0 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-docdir = $(prefix)/doc/@PACKAGE@
+docdir = $(datarootdir)/doc/@PACKAGE@
 
 doc_DATA = \
 	CUnit_doc.css \
@@ -13,4 +13,4 @@ doc_DATA = \
 	test_registry.html \
 	writing_tests.html
 
-SUBDIRS = headers
\ No newline at end of file
+SUBDIRS = headers
diff --git a/doc/headers/Makefile.am b/doc/headers/Makefile.am
index 9926e8b..3c5f1ba 100644
--- a/doc/headers/Makefile.am
+++ b/doc/headers/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-dochdrdir = $(prefix)/doc/@PACKAGE@/headers
+dochdrdir = $(includedir)/@PACKAGE@/headers
 
 INCLUDE_FILES = \
 	Automated.h \
