diff --git a/tests/ofproto-dpif.at b/tests/ofproto-dpif.at
index 729beced2..2b5cade8d 100644
--- a/tests/ofproto-dpif.at
+++ b/tests/ofproto-dpif.at
@@ -11828,6 +11828,7 @@ AT_CHECK([ovs-ofctl add-flows br0 flows.txt])
 dnl Start a new connection from port 1.
 AT_CHECK([ovs-appctl netdev-dummy/receive p1 'in_port(1),eth(src=50:54:00:00:00:09,dst=50:54:00:00:00:0a),eth_type(0x0800),ipv4(src=10.1.1.1,dst=10.1.1.2,proto=17,tos=0,ttl=64,frag=no),udp(src=1,dst=2)'])
 
+sleep 1
 AT_CHECK([cat ovs-vswitchd.log | strip_ufid | filter_flow_install], [0], [dnl
 recirc_id(0),in_port(1),packet_type(ns=0,id=0),eth_type(0x0800),ipv4(proto=17,frag=no), actions:ct(commit)
 ])
diff --git a/tests/ovsdb-server.at b/tests/ovsdb-server.at
index ce6d32aee..ac0e1d01f 100644
--- a/tests/ovsdb-server.at
+++ b/tests/ovsdb-server.at
@@ -2804,8 +2804,8 @@ AT_CHECK([uuidfilt monitor.stdout | sed '/^$/d'], [0], [dnl
 <8> delete 4 four
 <9> insert 4 four
 ])
-OVSDB_SERVER_SHUTDOWN(["/Address already in use/d"])
-OVSDB_SERVER_SHUTDOWN2(["/Address already in use/d"])
+OVSDB_SERVER_SHUTDOWN(["/Address in use/d"])
+OVSDB_SERVER_SHUTDOWN2(["/Address in use/d"])
 
 dnl Starting a replay.
 AT_CHECK([ovsdb-server --replay=./replay_dir                      dnl
