Remove bundled zlib.
--- a/Makefile
+++ b/Makefile
@@ -7,7 +7,7 @@
 
 SOURCES_C   :=
 SOURCES_CXX :=
-LIBS    :=
+LIBS    := -lz
 
 ifeq ($(platform),)
 	platform = unix
--- a/Makefile.common
+++ b/Makefile.common
@@ -10,8 +10,7 @@
    -I$(LIBRETRO_COMM_DIR)/include \
 	-I$(GUI) \
 	-I$(GUI)/nuklear \
-	-I$(GUI)/retro \
-   -I$(DEPS)/zlib
+	-I$(GUI)/retro
 
 SOURCES_C := \
 				 $(CORE_DIR)/libretro/libretro-core.c \
@@ -34,20 +33,6 @@
 				 $(CORE_DIR)/libretro/retro_disk_control.c \
 				 $(CORE_DIR)/libretro/retro_events.c \
 				 $(CORE_DIR)/libretro/retro_snd.c
-
-SOURCES_C += \
-      $(DEPS)/zlib/adler32.c \
-      $(DEPS)/zlib/inflate.c \
-      $(DEPS)/zlib/inffast.c \
-      $(DEPS)/zlib/inftrees.c \
-      $(DEPS)/zlib/zutil.c \
-      $(DEPS)/zlib/gzread.c \
-      $(DEPS)/zlib/gzwrite.c \
-      $(DEPS)/zlib/gzclose.c \
-      $(DEPS)/zlib/gzlib.c \
-      $(DEPS)/zlib/trees.c \
-      $(DEPS)/zlib/crc32.c \
-      $(DEPS)/zlib/deflate.c
 
 ifneq ($(STATIC_LINKING), 1)
 SOURCES_C += \
