diff -uNr dahdi-linux-2.2.0.2.ORIG/drivers/dahdi/xpp/xpp_dahdi.c dahdi-linux-2.2.0.2/drivers/dahdi/xpp/xpp_dahdi.c
--- dahdi-linux-2.2.0.2.ORIG/drivers/dahdi/xpp/xpp_dahdi.c	2009-10-22 18:52:23.000000000 +0100
+++ dahdi-linux-2.2.0.2/drivers/dahdi/xpp/xpp_dahdi.c	2009-10-22 18:52:55.000000000 +0100
@@ -865,9 +865,11 @@
 #endif
 	spin_lock_irqsave(&xbus->lock, flags);
 	atomic_inc(&xpd->open_counter);
+/*
 	LINE_DBG(DEVICES, xpd, pos, "%s[%d]: open_counter=%d\n",
 		current->comm, current->pid,
 		atomic_read(&xpd->open_counter));
+*/
 	spin_unlock_irqrestore(&xbus->lock, flags);
 	if(xpd->xops->card_open)
 		xpd->xops->card_open(xpd, pos);
@@ -885,9 +887,11 @@
 	spin_unlock_irqrestore(&xbus->lock, flags);
 	if(xpd->xops->card_close)
 		xpd->xops->card_close(xpd, pos);
+/*
 	LINE_DBG(DEVICES, xpd, pos, "%s[%d]: open_counter=%d\n",
 		current->comm, current->pid,
 		atomic_read(&xpd->open_counter));
+*/
 	atomic_dec(&xpd->open_counter);		/* from xpp_open() */
 	put_xpd(__FUNCTION__, xpd);					/* from xpp_open() */
 	return 0;
