diff --git a/tests/conftest.py b/tests/conftest.py
index de76c5f3..41ba55c7 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -21,6 +21,8 @@ import pytest
 from filelock import BaseFileLock, FileLock
 from packaging.requirements import InvalidRequirement, Requirement
 
+import cx_Freeze
+
 if sys.version_info < (3, 11):
     import tomli as tomllib
 else:
@@ -82,7 +84,7 @@ class TempPackage:
         self.system_prefix: Path = prefix
         self.relative_bin: str = sysexe.parent.relative_to(prefix).as_posix()
         self.relative_site: str = (
-            Path(pytest.__file__).parent.parent.relative_to(prefix).as_posix()
+            Path(cx_Freeze.__file__).parent.parent.relative_to(prefix).as_posix()
         )
 
         # make a temporary directory and set it as current
