diff --git a/src/apkd/meson.build b/src/apkd/meson.build
index 95b8d09..3a84667 100644
--- a/src/apkd/meson.build
+++ b/src/apkd/meson.build
@@ -12,6 +12,7 @@ apkd_lib = library(
     d_module_versions: d_mod_version,
     include_directories: src_inc,
     dependencies: [apk_toolsd_dep, libapkd_common_dep, apk_tools_dep],
+    version: meson.project_version(),
 )
 
 apkd_dep = declare_dependency(
diff --git a/src/apkd_common/meson.build b/src/apkd_common/meson.build
index ce7747b..0f9b391 100644
--- a/src/apkd_common/meson.build
+++ b/src/apkd_common/meson.build
@@ -17,6 +17,7 @@ libapkd_common = library(
     common_src,
     install: true,
     include_directories: src_inc,
+    version: meson.project_version(),
 )
 
 libapkd_common_dep = declare_dependency(
diff --git a/src/apkd_dbus_server/meson.build b/src/apkd_dbus_server/meson.build
index 924e0ca..1244519 100644
--- a/src/apkd_dbus_server/meson.build
+++ b/src/apkd_dbus_server/meson.build
@@ -36,6 +36,7 @@ apkd_dbus_server_lib_dep = declare_dependency(
         libapkd_common_dep.partial_dependency(compile_args: true),
     ],
     compile_args: '-J' + join_paths(meson.current_source_dir(), '..', '..', 'data'),
+    version: meson.project_version(),
 )
 
 apkd_dbus_exe = executable(
@@ -57,6 +58,7 @@ apkd_dbus_client_lib = library(
     apkd_dbus_client_lib_src,
     install: true,
     dependencies: [apkd_dbus_client_deps],
+    version: meson.project_version(),
 )
 
 apkd_dbus_client_lib_dep = declare_dependency(
