Author: psykose <alice@ayaya.dev>
Summary: fix werror=format-security
Upstream: No
----

--- a/contrib/gdal/gdalbridge.c
+++ b/contrib/gdal/gdalbridge.c
@@ -125,7 +125,7 @@
 
         if( pfnTest == NULL )
         {
-            sprintf( szPath, papszSOFilenames[iSOFile] );
+            sprintf( szPath, "%s", papszSOFilenames[iSOFile] );
             pfnTest = GBGetSymbol( szPath, "GDALOpen" );
         }
     }
