diff --git a/contrib/gdal/gdalbridge.c b/contrib/gdal/gdalbridge.c
index d4b4ec8..a91233c 100644
--- a/contrib/gdal/gdalbridge.c
+++ b/contrib/gdal/gdalbridge.c
@@ -124,7 +124,7 @@ int GDALBridgeInitialize( const char * pszTargetDir )
 
         if( pfnTest == NULL )
         {
-            sprintf( szPath, papszSOFilenames[iSOFile] );
+            sprintf( szPath, "%s", papszSOFilenames[iSOFile] );
             pfnTest = GBGetSymbol( szPath, "GDALOpen" );
         }
     }
