diff --git a/ogdi/gltpd/asyncsvr.c b/ogdi/gltpd/asyncsvr.c
index a0cbc4b..a423742 100644
--- a/ogdi/gltpd/asyncsvr.c
+++ b/ogdi/gltpd/asyncsvr.c
@@ -80,7 +80,7 @@ ECS_CVSID("$Id$");
 
 #define COMTIMEOUT 900
 
-long timecount;
+time_t timecount;
 
 static void dispatchno_1();
 extern void ecsprog_1();
@@ -533,7 +533,7 @@ void gltpd_svc_run()
   struct timeval timeout;
   xdrproc_t xdr_argument;
   /*xdrproc_t xdr_result;*/
-  long currenttime;
+  time_t currenttime;
   
   timeout.tv_sec = COMTIMEOUT;
   timeout.tv_usec = 0;
