--- gcc-4.0.2/gcc/gcc.c.nondef~	2006-01-09 00:09:50 +0100
+++ gcc-4.0.2/gcc/gcc.c	2006-01-09 00:15:31 +0100
@@ -764,6 +764,6 @@
 
 static const char *asm_debug;
 static const char *cpp_spec = CPP_SPEC;
-static const char *cc1_spec = CC1_SPEC;
+static const char *cc1_spec = CC1_SPEC CC1_HARDENED_SPEC;
 static const char *cc1plus_spec = CC1PLUS_SPEC;
 static const char *link_gcc_c_sequence_spec = LINK_GCC_C_SEQUENCE_SPEC;
