--- gcc-4.8.1/gcc/configure.ac.orig	2013-09-25 04:08:25.046595893 +0000
+++ gcc-4.8.1/gcc/configure.ac	2013-09-25 04:09:06.023263205 +0000
@@ -4822,7 +4822,7 @@
       gcc_cv_target_dl_iterate_phdr=no
     fi
     ;;
-  *-linux-musl*)
+  *-linux-musl* | *-linux-uclibc* | *-linux-gnu*)
     gcc_cv_target_dl_iterate_phdr=yes
     ;;
 esac
--- gcc-4.8.1/gcc/configure.orig	2013-09-25 04:08:17.826596392 +0000
+++ gcc-4.8.1/gcc/configure	2013-09-25 04:09:24.393263387 +0000
@@ -26923,7 +26923,7 @@
       gcc_cv_target_dl_iterate_phdr=no
     fi
     ;;
-  *-linux-musl*)
+  *-linux-musl* | *-linux-uclibc* | *-linux-gnu*)
     gcc_cv_target_dl_iterate_phdr=yes
     ;;
 esac
