--- a/gcc/config/arm/linux-eabi.h
+++ b/gcc/config/arm/linux-eabi.h
@@ -64,7 +64,10 @@
 
 /* musl has no "classic" (i.e. broken) mode */
 #undef  MUSL_DYNAMIC_LINKER
-#define MUSL_DYNAMIC_LINKER "/lib/ld-musl-arm.so.1"
+#define MUSL_DYNAMIC_LINKER \
+	"%{mfloat-abi=hard:/lib/ld-musl-armhf.so.1}" \
+	"%{mfloat-abi=soft*:/lib/ld-musl-arm.so.1}" \
+	"%{!mfloat-abi=*:/lib/ld-musl-armhf.so.1}"
 
 /* At this point, bpabi.h will have clobbered LINK_SPEC.  We want to
    use the GNU/Linux version, not the generic BPABI version.  */
