--- a/libstdc++-v3/configure	2009-08-04 15:41:02.000000000 +0000
+++ b/libstdc++-v3/configure	2009-08-04 15:46:20.000000000 +0000
@@ -113523,14 +113523,14 @@
   # Based on system-libunwind and target, do we have ipinfo?
   if  test x$with_system_libunwind = xyes; then
     case ${target} in
-      ia64-*-*) have_unwind_getipinfo=no ;;
+      ia64-*-*|*-uclibc*) have_unwind_getipinfo=no ;;
       *) have_unwind_getipinfo=yes ;;
     esac
   else
     # Darwin before version 9 does not have _Unwind_GetIPInfo.
 
     case ${target} in
-      *-*-darwin[3-8]|*-*-darwin[3-8].*) have_unwind_getipinfo=no ;;
+      *-*-darwin[3-8]|*-*-darwin[3-8].*|*-uclibc*) have_unwind_getipinfo=no ;;
       *) have_unwind_getipinfo=yes ;;
     esac
 
