--- icedtea-2.4.7/acinclude.m4.orig	2014-04-17 13:41:57.548895009 +0000
+++ icedtea-2.4.7/acinclude.m4	2014-04-17 13:45:29.752728702 +0000
@@ -2376,10 +2376,10 @@
     AC_MSG_CHECKING([which options to pass to ${PAX_COMMAND}])
     case "${host_cpu}" in
       i?86)
-        PAX_COMMAND_ARGS="-msp"
+        PAX_COMMAND_ARGS="-c -msp"
         ;;
       *)
-        PAX_COMMAND_ARGS="-m"
+        PAX_COMMAND_ARGS="-c -m"
         ;;
     esac
     AC_MSG_RESULT(${PAX_COMMAND_ARGS})
