diff --git a/tests/compute_engine/test__metadata.py b/tests/compute_engine/test__metadata.py
index bf48882..59872ca 100644
--- a/tests/compute_engine/test__metadata.py
+++ b/tests/compute_engine/test__metadata.py
@@ -16,7 +16,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 from six.moves import reload_module
diff --git a/tests/compute_engine/test_credentials.py b/tests/compute_engine/test_credentials.py
index ee415db..22352b8 100644
--- a/tests/compute_engine/test_credentials.py
+++ b/tests/compute_engine/test_credentials.py
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/conftest.py b/tests/conftest.py
index c9e3f84..02896fe 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -14,7 +14,7 @@
 
 import sys
 
-import mock
+from unittest import mock
 import pytest
 
 
diff --git a/tests/crypt/test__python_rsa.py b/tests/crypt/test__python_rsa.py
index d13105f..564b85d 100644
--- a/tests/crypt/test__python_rsa.py
+++ b/tests/crypt/test__python_rsa.py
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 from pyasn1_modules import pem
 import pytest
 import rsa
diff --git a/tests/oauth2/test__client.py b/tests/oauth2/test__client.py
index 3ec7fc6..96c5330 100644
--- a/tests/oauth2/test__client.py
+++ b/tests/oauth2/test__client.py
@@ -16,7 +16,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 import six
 from six.moves import http_client
diff --git a/tests/oauth2/test_credentials.py b/tests/oauth2/test_credentials.py
index 922c3bb..cced0ed 100644
--- a/tests/oauth2/test_credentials.py
+++ b/tests/oauth2/test_credentials.py
@@ -16,7 +16,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/oauth2/test_id_token.py b/tests/oauth2/test_id_token.py
index 360f92f..61f8499 100644
--- a/tests/oauth2/test_id_token.py
+++ b/tests/oauth2/test_id_token.py
@@ -14,7 +14,7 @@
 
 import json
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import exceptions
diff --git a/tests/oauth2/test_service_account.py b/tests/oauth2/test_service_account.py
index 54ac0f5..e8fc47a 100644
--- a/tests/oauth2/test_service_account.py
+++ b/tests/oauth2/test_service_account.py
@@ -16,7 +16,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 
 from google.auth import _helpers
 from google.auth import crypt
diff --git a/tests/test__cloud_sdk.py b/tests/test__cloud_sdk.py
index 58c7270..39ad93f 100644
--- a/tests/test__cloud_sdk.py
+++ b/tests/test__cloud_sdk.py
@@ -17,7 +17,7 @@ import json
 import os
 import subprocess
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _cloud_sdk
diff --git a/tests/test__default.py b/tests/test__default.py
index 3fb0fa1..c2db09d 100644
--- a/tests/test__default.py
+++ b/tests/test__default.py
@@ -15,7 +15,7 @@
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _default
diff --git a/tests/test__oauth2client.py b/tests/test__oauth2client.py
index 832b24f..151574b 100644
--- a/tests/test__oauth2client.py
+++ b/tests/test__oauth2client.py
@@ -16,7 +16,7 @@ import datetime
 import os
 import sys
 
-import mock
+from unittest import mock
 import oauth2client.client
 import oauth2client.contrib.gce
 import oauth2client.service_account
diff --git a/tests/test_app_engine.py b/tests/test_app_engine.py
index 70fa5ca..177b2c4 100644
--- a/tests/test_app_engine.py
+++ b/tests/test_app_engine.py
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import app_engine
diff --git a/tests/test_iam.py b/tests/test_iam.py
index cc09085..cef1cb6 100644
--- a/tests/test_iam.py
+++ b/tests/test_iam.py
@@ -16,7 +16,7 @@ import base64
 import datetime
 import json
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 
diff --git a/tests/test_impersonated_credentials.py b/tests/test_impersonated_credentials.py
index 68a2af8..fc2b8d3 100644
--- a/tests/test_impersonated_credentials.py
+++ b/tests/test_impersonated_credentials.py
@@ -16,7 +16,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 from six.moves import http_client
 
diff --git a/tests/test_jwt.py b/tests/test_jwt.py
index 4a64717..fd5a565 100644
--- a/tests/test_jwt.py
+++ b/tests/test_jwt.py
@@ -17,7 +17,7 @@ import datetime
 import json
 import os
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/transport/test_grpc.py b/tests/transport/test_grpc.py
index c98dcff..6360dd7 100644
--- a/tests/transport/test_grpc.py
+++ b/tests/transport/test_grpc.py
@@ -14,7 +14,7 @@
 
 import datetime
 
-import mock
+from unittest import mock
 import pytest
 
 from google.auth import _helpers
diff --git a/tests/transport/test_requests.py b/tests/transport/test_requests.py
index 311992a..9396fcc 100644
--- a/tests/transport/test_requests.py
+++ b/tests/transport/test_requests.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mock
+from unittest import mock
 import requests
 import requests.adapters
 from six.moves import http_client
diff --git a/tests/transport/test_urllib3.py b/tests/transport/test_urllib3.py
index a119b74..a3e9afb 100644
--- a/tests/transport/test_urllib3.py
+++ b/tests/transport/test_urllib3.py
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import mock
+from unittest import mock
 from six.moves import http_client
 import urllib3
 
