diff --git a/resources/ews/test/isolatedtestbase.cpp b/resources/ews/test/isolatedtestbase.cpp
index 5f0440c..82ade7d 100644
--- a/resources/ews/test/isolatedtestbase.cpp
+++ b/resources/ews/test/isolatedtestbase.cpp
@@ -12,7 +12,7 @@
 #include <Akonadi/AgentManager>
 #include <Akonadi/Control>
 
-#include "ewsresourceinterface.h"
+#include "ewsresource.h"
 #include "ewssettings.h"
 #include "ewswallet.h"
 #include "fakeewsserverthread.h"
