diff --git a/src/c/ffi.d b/src/c/ffi.d
index 8174977..b42335c 100644
--- a/src/c/ffi.d
+++ b/src/c/ffi.d
@@ -132,10 +132,13 @@ static struct {
   {@':stdcall', FFI_STDCALL},
 #elif defined(X86_WIN64)
   {@':win64', FFI_WIN64},
-#elif defined(X86_ANY) || defined(X86) || defined(X86_64)
-  {@':cdecl', FFI_SYSV},
-  {@':sysv', FFI_SYSV},
+#elif defined (X86_64)
+  {@':sysv', FFI_UNIX64},
+  {@':cdecl', FFI_UNIX64},
   {@':unix64', FFI_UNIX64},
+#elif defined(X86_ANY)
+  {@':sysv', FFI_SYSV},
+  {@':cdecl', FFI_SYSV},
 #endif
 };

