--- steghide-0.5.1/po/Makefile.in.in.org	2003-09-28 17:57:14.000000000 +0200
+++ steghide-0.5.1/po/Makefile.in.in	2006-07-12 17:17:38.000000000 +0200
@@ -27,7 +27,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(MKINSTALLDIRS)
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
--- steghide-0.5.1/src/Makefile.am.org	2003-09-28 18:33:21.000000000 +0200
+++ steghide-0.5.1/src/Makefile.am	2006-07-12 17:13:26.000000000 +0200
@@ -33,5 +33,4 @@
 WavPCMSampleValue.cc error.cc main.cc msg.cc SMDConstructionHeuristic.cc
 LIBS = @LIBINTL@ @LIBS@
 localedir = $(datadir)/locale
-LIBTOOL = $(SHELL) libtool
 MAINTAINERCLEANFILES = Makefile.in
