--- llvm-7.0.1.src/unittests/Support/DynamicLibrary/DynamicLibraryTest.cpp
+++ b/unittests/Support/DynamicLibrary/DynamicLibraryTest.cpp
@@ -109,6 +109,7 @@
   EXPECT_TRUE(DynamicLibrary::SearchOrder == DynamicLibrary::SO_Linker);
 }
 
+/**
 TEST(DynamicLibrary, Shutdown) {
   std::string A("PipSqueak"), B, C("SecondLib");
   std::vector<std::string> Order;
@@ -164,6 +165,7 @@
   EXPECT_EQ(Order.front(), "SecondLib");
   EXPECT_EQ(Order.back(), "PipSqueak");
 }
+**/
 
 #else
 
