diff --git a/src/capmt.c b/src/descrambler/capmt.c
index 5d86c3d..0a1072c 100644
--- a/src/capmt.c
+++ b/src/capmt.c
@@ -778,7 +778,8 @@ capmt_table_input(struct th_descrambler *td, struct service *s,
           pos += ecd.cad_length + 2;
 
           LIST_FOREACH(cce2, &ct->ct_caid_ecm, cce_link) {
-            capmt_descriptor_t cad = { 
+            if (cce2->cce_caid == 0x1817) continue;
+	    capmt_descriptor_t cad = { 
               .cad_type = 0x09, 
               .cad_length = 0x04,
               .cad_data = { 

