required against new fcitx libs
diff --git a/gui/CMakeLists.txt b/gui/CMakeLists.txt
index 2327132..8a6f0d8 100644
--- a/gui/CMakeLists.txt
+++ b/gui/CMakeLists.txt
@@ -17,7 +17,7 @@ qt5_wrap_ui(KKC_CONFIG_SRCS dictwidget.ui adddictdialog.ui shortcutwidget.ui add
 add_library(fcitx5-kkc-config
   MODULE ${KKC_CONFIG_SRCS})
 
-add_definitions(-DQT_NO_KEYWORDS)
+add_definitions(-DQT_NO_KEYWORDS -std=c++17)
 
 target_include_directories(fcitx5-kkc-config PRIVATE ${PROJECT_BINARY_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
