--- icedtea-2.6.8.orig/acinclude.m4
+++ icedtea-2.6.8/acinclude.m4
@@ -2759,7 +2759,7 @@
     PKG_CHECK_MODULES(NSS_JAVA, nss-java, [NSS_JAVA_FOUND=yes], [NSS_JAVA_FOUND=no])
     if test "x${NSS_SOFTOKN_FOUND}" = "xyes"; then
       SUNEC_CFLAGS=$NSS_SOFTOKN_CFLAGS;
-      SUNEC_LIBS="$NSS_LIBS -lfreebl";
+      SUNEC_LIBS="$NSS_LIBS -lfreebl3";
     elif test "x${NSS_JAVA_FOUND}" = "xyes"; then
       SUNEC_CFLAGS="$NSS_JAVA_CFLAGS -DLEGACY_NSS";
       SUNEC_LIBS=$NSS_JAVA_LIBS;
