diff --git a/crates/cli/Cargo.toml b/crates/cli/Cargo.toml
index 0892fde..0c02242 100644
--- a/crates/cli/Cargo.toml
+++ b/crates/cli/Cargo.toml
@@ -47,14 +47,17 @@ harness = false
 [[test]]
 name = "ui_with_python"
 harness = false
+test = false
 
 [[test]]
 name = "ui_with_jinja"
 harness = false
+test = false
 
 [[test]]
 name = "ui_with_dbt"
 harness = false
+test = false
 
 [[test]]
 name = "ui_github"
diff --git a/crates/lib/Cargo.toml b/crates/lib/Cargo.toml
index ac0649e..ea93ed1 100644
--- a/crates/lib/Cargo.toml
+++ b/crates/lib/Cargo.toml
@@ -24,6 +24,7 @@ harness = false
 [[test]]
 name = "templaters"
 harness = false
+test = false
 
 [[bench]]
 name = "parsing"
