--- a/src/device.c
+++ b/src/device.c
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/time.h>
--- a/src/mount-monitor.c
+++ b/src/mount-monitor.c
@@ -28,6 +28,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <mntent.h>
 
--- a/src/daemon.c
+++ b/src/daemon.c
@@ -39,6 +39,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <sys/resource.h>
--- a/tools/udisks.c
+++ b/tools/udisks.c
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <fcntl.h>
--- a/tools/umount-udisks.c
+++ b/tools/umount-udisks.c
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/sysmacros.h>
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <pwd.h>
