--- a/contrib/forgejo-runner.service
+++ b/contrib/forgejo-runner.service
@@ -4,12 +4,10 @@ Documentation=https://forgejo.org/docs/latest/admin/actions/
 After=docker.service
 
 [Service]
-ExecStart=/usr/local/bin/forgejo-runner daemon
+ExecStart=/usr/bin/forgejo-runner daemon --config /etc/forgejo-runner/config.yaml
 ExecReload=/bin/kill -s HUP $MAINPID
-
-# This user and working directory must already exist
-User=runner 
-WorkingDirectory=/home/runner
+User=forgejo-runner
+WorkingDirectory=/var/lib/forgejo-runner
 Restart=on-failure
 TimeoutSec=0
 RestartSec=10
