#Correct syntax error
--- a/PyPlucker/helper/gettext.py
+++ b/PyPlucker/helper/gettext.py
@@ -94,7 +94,7 @@
         """Returns the position in the code where the function was called.
         The function uses some knowledge about python stack frames."""
         import sys
-import atexit
+        import atexit
         # get access to the stack frame by generating an exception.
         try:
                 raise RuntimeError
