--- taskd-1.0.0/src/Path.cpp
+++ taskd-1.0.0/src/Path.cpp.changed
@@ -37,6 +37,15 @@
 #include <Directory.h>
 #include <Path.h>
 
+/* fixes build with musl libc */
+#ifndef GLOB_TILDE
+#define GLOB_TILDE 0
+#endif
+
+#ifndef GLOB_BRACE
+#define GLOB_BRACE 0
+#endif
+
 ////////////////////////////////////////////////////////////////////////////////
 std::ostream& operator<< (std::ostream& out, const Path& path)
 {
