From b1b1f4bed9507d3c547d174f67be9e3a9c2e5d99 Mon Sep 17 00:00:00 2001
From: Radovan Sroka <rsroka@redhat.com>
Date: Sun, 11 Oct 2020 18:04:56 +0200
Subject: [PATCH] Remove reading of nsswitch.conf.

---
 src/Daemon/NSHandler.cpp | 42 ++--------------------------------------
 1 file changed, 2 insertions(+), 40 deletions(-)

diff --git a/src/Daemon/NSHandler.cpp b/src/Daemon/NSHandler.cpp
index 9cd0845f..536e3f05 100644
--- a/src/Daemon/NSHandler.cpp
+++ b/src/Daemon/NSHandler.cpp
@@ -150,46 +150,8 @@ namespace usbguard
 
   void NSHandler::parseNSSwitch()
   {
-    USBGUARD_LOG(Info) << "Loading nsswitch from " << _nsswitch_path;
-    std::ifstream nss(_nsswitch_path);
-
-    if (!nss.is_open()) {
-      throw ErrnoException("NSSwitch parsing", _nsswitch_path, errno);
-    }
-
-    _parser.parseStream(nss);
-    _parsedOptions = _parser.getMap();
-    nss.close();
-    USBGUARD_LOG(Debug) << "Map contains:";
-
-    for (auto x: _parsedOptions) {
-      USBGUARD_LOG(Debug) << "--> " << x.first << " -> " << x.second << " <--";
-    }
-
-    std::locale loc;
-
-    for (unsigned i = 0 ; i < _parsedOptions["USBGUARD"].length() ; i++) {
-      _parsedOptions["USBGUARD"][i] = std::toupper(_parsedOptions["USBGUARD"][i], loc);
-    }
-
-    USBGUARD_LOG(Info) << "Fetched value is -> " << _parsedOptions["USBGUARD"] << " <-";
-
-    if (_parsedOptions["USBGUARD"] == "FILES") {
-      _source = SourceType::LOCAL;
-    }
-    else if (_parsedOptions["USBGUARD"] == "LDAP") {
-#ifdef HAVE_LDAP
-      _source = SourceType::LDAP;
-#else
-      USBGUARD_LOG(Info) << "USBGuard has been compiled without LDAP support";
-      USBGUARD_LOG(Info) << "Using default value FILES";
-      _source = SourceType::LOCAL;
-#endif
-    }
-    else {
-      USBGUARD_LOG(Info) << "Value is not valid or not set, using default FILES";
-      _source = SourceType::LOCAL;
-    }
+    USBGUARD_LOG(Info) << "Using default value FILES";
+    _source = SourceType::LOCAL;
   }
 } /* namespace usbguard */
 
