diff --git a/webui/module/Application/view/layout/login.phtml.in b/webui/module/Application/view/layout/login.phtml.in
index 5c7dbf5..8a9d801 100644
--- a/webui/module/Application/view/layout/login.phtml.in
+++ b/webui/module/Application/view/layout/login.phtml.in
@@ -32,6 +32,17 @@ echo $this->doctype();

    <head>

+   <script>
+     function showhide() {
+       var x = document.getElementById("alpine-patches");
+       if (x.style.display === "none") {
+         x.style.display = "block";
+       } else {
+         x.style.display = "none";
+       }
+     }
+   </script>
+
       <meta charset="utf-8">

       <?php echo $this->headTitle('Bareos','PREPEND')->setSeparator(' - ')->setAutoEscape(false) ?>
@@ -90,6 +101,19 @@ echo $this->doctype();
       <br /><br /><br />
       <?php echo $this->content; ?>
       <br />
+        <div id="alpine">
+          In order to be compliant with AGPL 3.0, you should be noticed that Alpine package has patched BareOS source with the following patches:<br />
+          <button onclick="showhide()">Alpine Linux Patches</button>
+            <div id="alpine-patches" style="display:none">
+              <a href="https://git.alpinelinux.org/aports/tree/community/bareos/path-mounted.patch">Link to Alpine Patch</a><br />
+              <code>
+                This patch has been part of Alpine BareOS package since<br>
+                Date:   Wed May 18 16:28:14 2016 +0000<br>
+                See: commit <a href="https://git.alpinelinux.org/aports/commit/?id=3f7f468646b60bd49dc0d33533086d15a2348692">3f7f468646b60bd49dc0d33533086d15a2348692</a>
+              </code>
+              <br />
+            </div>
+        </div>
       <p>
          <a href="@BareosComSubscriptionLink@" title="@BareosComSubscriptionLink@" target="_blank">@BareosGetOfficialBinariesAndSupportOnBareosComLoginPage@</a><br/>
          Version <?php echo $bareos_full_version; ?> &copy; 2013 - <?php echo date('Y') ?> <a href="http://www.bareos.com/" target="_blank">Bareos GmbH  &amp; Co. KG</a>,<br />
