diff --git a/webui/install/apache/bareos-webui.conf.in b/webui/install/apache/bareos-webui.conf.in
index d860f40..4bb4985 100644
--- a/webui/install/apache/bareos-webui.conf.in
+++ b/webui/install/apache/bareos-webui.conf.in
@@ -2,6 +2,8 @@
 # Bareos WebUI Apache configuration file
 #
 
+LoadModule rewrite_module modules/mod_rewrite.so
+
 # Environment Variable for Application Debugging
 # Set to "development" to turn on debugging mode or
 # "production" to turn off debugging mode.
