diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 62cb8de..f26ddaf 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -384,7 +384,7 @@ if(MSVC)
       /D_CRT_SECURE_NO_DEPRECATE # deprecation warning for stdlib functions
   )
 else()
-  set(BAREOS_WARNING_FLAGS -Werror -Wall -Wextra -Wshadow)
+  set(BAREOS_WARNING_FLAGS -Wall -Wextra -Wshadow)
 endif()
 list(JOIN BAREOS_WARNING_FLAGS " " BAREOS_WARNING_FLAGS_STRING)
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${BAREOS_WARNING_FLAGS_STRING}")
