diff --git a/core/CMakeLists.txt b/core/CMakeLists.txt
index 7e76a14..24ccd7d 100644
--- a/core/CMakeLists.txt
+++ b/core/CMakeLists.txt
@@ -369,8 +369,8 @@ endif()
 include(FindIntl)

 if(NOT MSVC)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror -Wall -Wextra -Wshadow")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror -Wall -Wextra -Wshadow")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wextra -Wshadow")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wshadow")
 else()
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /W4")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W4")
