diff --git a/systemtests/CMakeLists.txt b/systemtests/CMakeLists.txt
index 6ddf39b..dcd6d66 100644
--- a/systemtests/CMakeLists.txt
+++ b/systemtests/CMakeLists.txt
@@ -192,12 +192,6 @@ execute_process(
   RESULT_VARIABLE RESULT_GENERATE_MINIO_CERTS
   OUTPUT_QUIET ERROR_QUIET
 )
-if(NOT "${RESULT_GENERATE_MINIO_CERTS}" STREQUAL "0")
-  message(
-    FATAL_ERROR
-      "Creation of certificates failed: ${RESULT_GENERATE_MINIO_CERTS} ${CMAKE_BINARY_DIR}"
-  )
-endif()

 configure_file(
   "CTestCustom.cmake.in" "${CMAKE_BINARY_DIR}/CTestCustom.cmake" @ONLY
