diff --git a/core/src/lib/edit.h b/core/src/lib/edit.h
index b49f788..7cd745a 100644
--- a/core/src/lib/edit.h
+++ b/core/src/lib/edit.h
@@ -23,6 +23,7 @@
 
 #include <vector>
 #include <string>
+#include <pthread.h>
 #include "include/bc_types.h"
 
 namespace edit {
diff --git a/core/src/stored/backends/dedupable/device_options.cc b/core/src/stored/backends/dedupable/device_options.cc
index 6b1f9d3..8473dfd 100644
--- a/core/src/stored/backends/dedupable/device_options.cc
+++ b/core/src/stored/backends/dedupable/device_options.cc
@@ -22,6 +22,7 @@
 
 #include "device_options.h"
 #include "stored/backends/util.h"
+#include <pthread.h>
 #include <cstdint>
 #include "lib/edit.h"
 
