diff --git a/core/cmake/distname.sh b/core/cmake/distname.sh
index 2dfc47f..5026e11 100755
--- a/core/cmake/distname.sh
+++ b/core/cmake/distname.sh
@@ -184,6 +184,10 @@ else
       elif test "$CYGWIN" = yes
       then
          PLATFORM=cygwin
+      elif test -f /etc/alpine-release
+      then
+         PLATFORM=alpine
+         DISTVER=`cat /etc/alpine-release`
       else
          PLATFORM=unknown
          DISTVER=unknown
