--- qemu-1.7.0.org/hw/virtio/vhost.c
+++ qemu-1.7.0/hw/virtio/vhost.c
@@ -22,6 +22,15 @@
 #include "exec/address-spaces.h"
 #include "hw/virtio/virtio-bus.h"
 
+#ifndef HAVE_FFSLL
+static int ffsll(long long i) {
+    unsigned long long x = i & -i;
+    if (x <= 0xffffffff) return ffs (i);
+    else return 32 + ffs (i >> 32);
+}
+#endif
+
+
 static void vhost_dev_sync_region(struct vhost_dev *dev,
                                   MemoryRegionSection *section,
                                   uint64_t mfirst, uint64_t mlast,
