--- ./linux-user/syscall_defs.h.orig
+++ ./linux-user/syscall_defs.h
@@ -2552,12 +2552,14 @@
     abi_ulong ptr;
 };
 
+#define SIGEV_PAD_ARRAY_SIZE ARRAY_SIZE(((struct sigevent *)0)->__pad)
+
 struct target_sigevent {
     target_sigval_t sigev_value;
     int32_t sigev_signo;
     int32_t sigev_notify;
     union {
-        int32_t _pad[ARRAY_SIZE(((struct sigevent *)0)->_sigev_un._pad)];
+        int32_t _pad[SIGEV_PAD_ARRAY_SIZE];
         int32_t _tid;
 
         struct {
