diff -ruN a/sbc/sbc_primitives_armv6.c b/sbc/sbc_primitives_armv6.c
--- a/sbc/sbc_primitives_armv6.c	2025-03-21 08:18:31.000000000 +0000
+++ b/sbc/sbc_primitives_armv6.c	2025-10-19 17:54:25.460000000 +0000
@@ -22,7 +22,8 @@
 
 #ifdef SBC_BUILD_WITH_ARMV6_SUPPORT
 
-static void __attribute__((naked)) sbc_analyze_four_armv6()
+static void __attribute__((naked)) sbc_analyze_four_armv6(int16_t *in,
+    int32_t *out, const int16_t *consts)
 {
 	/* r0 = in, r1 = out, r2 = consts */
 	__asm__ volatile (
@@ -93,7 +94,8 @@
 	((void (*)(int16_t *, int32_t *, const FIXED_T*)) \
 		sbc_analyze_four_armv6)((in), (out), (consts))
 
-static void __attribute__((naked)) sbc_analyze_eight_armv6()
+static void __attribute__((naked)) sbc_analyze_eight_armv6(int16_t *in,
+    int32_t *out, const int16_t *consts)
 {
 	/* r0 = in, r1 = out, r2 = consts */
 	__asm__ volatile (
