diff -Naur a/scripts/test_runner.sh b/scripts/test_runner.sh
--- a/scripts/test_runner.sh	2024-11-01 11:12:18.000000000 -0400
+++ b/scripts/test_runner.sh	2024-11-09 22:33:26.593335473 -0500
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-REPO_DIR=$(git rev-parse --show-toplevel)
+REPO_DIR=$(realpath .)
 
 nvim_t() {
 	nvim -u "$REPO_DIR/tests/minimal_init.lua" -c "set runtimepath+=$REPO_DIR" "$@"
