From 571e31f8bfeac732daa68b27727bdfff442d3de5 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 8 May 2023 22:45:58 +0200
Subject: [PATCH 1006/1006] Install alpine provider

---
 Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 79f75aa..b0aaef2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 PREFIX?=/
 
-SUBPACKAGES = core network openrc aws azure gcp oci nocloud
+SUBPACKAGES = core network openrc aws azure gcp oci nocloud alpine
 
 .PHONY: check install $(SUBPACKAGES)
 
@@ -56,6 +56,11 @@ nocloud:
 	install -Dm644 -t $(PREFIX)/lib/tiny-cloud/cloud/nocloud \
 		lib/tiny-cloud/cloud/nocloud/*
 
+alpine:
+	install -Dm644 -t $(PREFIX)/lib/tiny-cloud/cloud/alpine \
+		lib/tiny-cloud/cloud/alpine/init
+	ln -s ../nocloud/imds $(PREFIX)/lib/tiny-cloud/cloud/alpine/imds
+
 check: tests/Kyuafile Kyuafile
 	kyua test || (kyua report --verbose && exit 1)
 
-- 
2.40.1

