diff -ru pjproject-2.2.1.orig/Makefile pjproject-2.2.1/Makefile
--- pjproject-2.2.1.orig/Makefile	2014-04-15 08:14:35.075716775 -0300
+++ pjproject-2.2.1/Makefile	2014-04-15 08:21:39.683417350 -0300
@@ -119,7 +119,7 @@
 install:
 	mkdir -p $(DESTDIR)$(libdir)/
 #	cp -af $(APP_LIB_FILES) $(DESTDIR)$(libdir)/
-	cp -af $(APP_LIBXX_FILES) $(DESTDIR)$(libdir)/
+	cp -af $(sort $(APP_LIBXX_FILES)) $(DESTDIR)$(libdir)/
 	mkdir -p $(DESTDIR)$(includedir)/
 	for d in pjlib pjlib-util pjnath pjmedia pjsip; do \
 		cp -RLf $$d/include/* $(DESTDIR)$(includedir)/; \
