From 4fafe94319a6b714ebb4086fe0be69c3ca9ff5ab Mon Sep 17 00:00:00 2001
From: Hubert LOBIT <hubert.lobit@allocloud.com>
Date: Thu, 18 Mar 2021 19:00:54 +0100
Subject: [PATCH] Fix installation of libpjsua2 shared libraries

---
 Makefile | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 31a6d39d4..74e246a44 100644
--- a/Makefile
+++ b/Makefile
@@ -121,9 +121,10 @@ cmp_wav:
 
 install:
 	mkdir -p $(DESTDIR)$(libdir)/
-	cp -af $(APP_LIB_FILES) $(DESTDIR)$(libdir)/
 	if [ "$(PJ_EXCLUDE_PJSUA2)x" = "x" ] ; then \
-	    cp -af $(PJ_DIR)/pjsip/lib/libpjsua2-$(LIB_SUFFIX) $(DESTDIR)$(libdir)/; \
+	    cp -af $(APP_LIBXX_FILES) $(DESTDIR)$(libdir)/; \
+	else \
+	    cp -af $(APP_LIB_FILES) $(DESTDIR)$(libdir)/; \
 	fi
 	mkdir -p $(DESTDIR)$(includedir)/
 	for d in pjlib pjlib-util pjnath pjmedia pjsip; do \
