diff --git a/gui/main.cpp b/gui/main.cpp
index fc68b7f..a682178 100644
--- a/gui/main.cpp
+++ b/gui/main.cpp
@@ -57,6 +57,9 @@ int main(int argc, char *argv[])
     QSettings* settings = new QSettings("Cppcheck", "Cppcheck-GUI", &app);
 
     // Set data dir..
+    if ((settings->value("DATADIR", QString()).toString()).isEmpty())
+        settings->setValue("DATADIR", "/usr/share/cppcheck");
+
     const QStringList args = QApplication::arguments();
     auto it = std::find_if(args.cbegin(), args.cend(), [](const QString& arg) {
         return arg.startsWith("--data-dir=");
@@ -66,6 +69,9 @@ int main(int argc, char *argv[])
         return 0;
     }
 
+    if ((settings->value(SETTINGS_LANGUAGE, QString()).toString()).isEmpty())
+        settings->setValue(SETTINGS_LANGUAGE, "en");
+
     TranslationHandler* th = new TranslationHandler(&app);
     th->setLanguage(settings->value(SETTINGS_LANGUAGE, th->suggestLanguage()).toString());
 
