--- ./src/Makefile.orig	2013-08-15 11:20:54.575932175 +0000
+++ ./src/Makefile	2013-08-15 11:21:37.813472296 +0000
@@ -24,6 +24,8 @@
         KERNEL_TREE := $(RHEL5_TREE)
 endif
 
+DEPMOD ?= depmod
+
 obj-m += flashcache.o
 flashcache-objs := flashcache_conf.o flashcache_main.o flashcache_subr.o flashcache_ioctl.o flashcache_procfs.o
 
@@ -42,7 +44,7 @@
 modules_install: modules
 	install -o root -g root -m 0755 -d $(DESTDIR)/lib/modules/$(KERNEL_SOURCE_VERSION)/extra/flashcache/
 	install -o root -g root -m 0755 flashcache.ko $(DESTDIR)/lib/modules/$(KERNEL_SOURCE_VERSION)/extra/flashcache/
-	depmod -a $(KERNEL_SOURCE_VERSION)
+	$(DEPMOD) -a $(KERNEL_SOURCE_VERSION)
 
 .PHONY: utils_install
 utils_install: utils
