diff --git a/vendor/CMakeLists.txt b/vendor/CMakeLists.txt
index f351d65..9f7db0e 100644
--- a/vendor/CMakeLists.txt
+++ b/vendor/CMakeLists.txt
@@ -22,12 +22,15 @@ if(ANDROID_BUILD_TOOLS_PATCH_VENDOR AND EXISTS "${ANDROID_PATCH_DIR}/")
         file(GLOB patches ${ANDROID_PATCH_DIR}/${v}/*.patch)
         if(patches)
             message(STATUS "Applying patches for: ${v}")
-            execute_process(COMMAND git -C
-                ${CMAKE_CURRENT_SOURCE_DIR}/${v} am ${patches}
-                RESULT_VARIABLE ret)
-            if(NOT "${ret}" STREQUAL "0")
-                message(FATAL_ERROR "Couldn't apply patches for ${v}")
-            endif()
+            foreach (patch ${patches})
+                message(STATUS "Patches: ${patch}")
+                execute_process(COMMAND patch -d
+                    ${CMAKE_CURRENT_SOURCE_DIR}/${v} -p1 -i ${patch}
+                    RESULT_VARIABLE ret)
+                if(NOT "${ret}" STREQUAL "0")
+                    message(FATAL_ERROR "Couldn't apply patches for ${v}")
+                endif()
+            endforeach(patch)
         endif(patches)
     endforeach(v)
 endif()
