From 25f06615a1819d9678114869e48d900d5e918f73 Mon Sep 17 00:00:00 2001
From: Julien Portalier <julien@portalier.com>
Date: Tue, 30 Apr 2019 10:51:56 +0200
Subject: [PATCH] Fix: ... is now valid crystal

---
 test/integration/build_test.cr | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/integration/build_test.cr b/test/integration/build_test.cr
index 861c064..0863c13 100644
--- a/test/integration/build_test.cr
+++ b/test/integration/build_test.cr
@@ -60,7 +60,7 @@ class BuildCommandTest < Minitest::Test
   end
 
   def test_reports_error_when_target_failed_to_compile
-    File.write File.join(application_path, "src", "cli.cr"), "a = ..."
+    File.write File.join(application_path, "src", "cli.cr"), "a = ......"
 
     Dir.cd(application_path) do
       ex = assert_raises(FailedCommand) do
