--- homer-api-5.0.6/api/Authentication/LDAP.php
+++ homer-api-5.0.6.php7/api/Authentication/LDAP.php
@@ -118,7 +118,7 @@
                         
 
                         // Assigne Admin Privs, should be read from the LDAP Directory in the future
-                        $ADMIN_USER = split(",", LDAP_ADMIN_USER);
+                        $ADMIN_USER = explode(",", LDAP_ADMIN_USER);
                         foreach($ADMIN_USER as &$value) {
 
                             if ($value == $param['username']) {
