diff --git a/tools/updaterevision/UpdateRevision.cmake b/tools/updaterevision/UpdateRevision.cmake
index 619a868..170cf53 100755
--- a/tools/updaterevision/UpdateRevision.cmake
+++ b/tools/updaterevision/UpdateRevision.cmake
@@ -74,7 +74,7 @@ function(main)
 	if(NOT Hash)
 		message("Failed to get commit info: ${Error}")
 		set(Hash "0")
-		set(Tag "<unknown version>")
+		set(Tag "${ALPINE_PKGVER}")
 		set(Timestamp "")
 	endif()
 
