--- a/src/crash.c.orig	2009-09-05 22:12:33.000000000 +0000
+++ b/src/crash.c	2009-09-05 22:14:34.000000000 +0000
@@ -42,7 +42,7 @@
 #	include <sys/utsname.h>
 #endif
 
-#if defined(__GNU_LIBRARY__)
+#if defined(__GNU_LIBRARY__) && !defined(__UCLIBC__)
 #	include <gnu/libc-version.h>
 #endif
 
@@ -446,7 +446,9 @@
  */
 static const gchar *get_lib_version(void)
 {
-#if defined(__GNU_LIBRARY__)
+#if defined(__UCLIBC__)
+	return g_strdup_printf("uClibc");
+#elif defined(__GNU_LIBRARY__)
 	return g_strdup_printf("GNU libc %s", gnu_get_libc_version());
 #else
 	return g_strdup(_("Unknown"));
