--- a/command.c
+++ b/command.c
@@ -1897,7 +1897,7 @@ void comment_at_eol()
   }
 
   lnend();
-  insertspacetab(FALSE);
+  insertspacetab();
   for(int c = 0; curbp->b_mode->slc[c] != '\0'; c++)
     input[c] = curbp->b_mode->slc[c];
   input[strlen(curbp->b_mode->slc)] = ' ';
--- a/header.h
+++ b/header.h
@@ -2,7 +2,9 @@
    Derived from: Atto January 2017
    Derived from: Anthony's Editor January 93
 */
+#ifndef _XOPEN_SOURCE
 #define _XOPEN_SOURCE
+#endif
 #define __USE_XOPEN
 #include <stdlib.h>
 #include <stdarg.h>
--- a/main.c
+++ b/main.c
@@ -67,13 +67,13 @@ int LINES;
 int COLS;
 int MSGLINE;			/*  */
 
-static void graceful_exit()
+static void graceful_exit(int)
 {
   tb_shutdown();
   exit(1);
 }
 
-static void cont()
+static void cont(int)
 {
   int ol = LINES, oc = COLS;
   tb_init();
