Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: remove rpath entry from library
----

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -203,7 +203,6 @@
   set(CMAKE_INSTALL_LIBDIR "${_LIBDIR_DEFAULT}" CACHE PATH "object code libraries (${_LIBDIR_DEFAULT})")
 endif()
 
-SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
 SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
 #SET(CMAKE_MACOSX_RPATH ON)
 
