diff --git a/Makefile b/Makefile
index 1910db6..31b480f 100644
--- a/Makefile
+++ b/Makefile
@@ -97,12 +97,10 @@ $(PROGRAM_NAME): $(OBJS)
 #	$(CC) $(CFLAGS) $(LDFLAGS) -o $(PROGRAM_NAME) $(OBJS)
 
 installdirs:
-	@[ "$(ON_WINDOWS)" = "1" ] && echo "Do not use install rules on Windows" && exit 1
 	test -e $(DESTDIR)$(LIB_DIR) || $(MKDIR) $(DESTDIR)$(LIB_DIR)
 	test -e $(DESTDIR)$(MAN7_DIR) || $(MKDIR) $(DESTDIR)$(MAN7_DIR)
 
 install: sharedlib staticlib installdirs
-	@[ "$(ON_WINDOWS)" = "1" ] && echo "Do not use install rules on Windows" && exit 1
 	$(INSTALL_DATA)	$(PROGRAM_NAME).so $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME).so.$(VERSION)
 	$(LN)		$(PROGRAM_NAME).so.$(VERSION) $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME).so.$(VERSION_MAJOR)
 	$(LN)		$(PROGRAM_NAME).so.$(VERSION) $(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME).so
@@ -111,13 +109,11 @@ install: sharedlib staticlib installdirs
 	$(INSTALL_DATA)	$(PROGRAM_NAME).7 $(DESTDIR)$(MAN7_DIR)/$(PROGRAM_NAME).7
 
 uninstalldirs:
-	@[ "$(ON_WINDOWS)" = "1" ] && echo "Do not use install rules on Windows" && exit 1
 	-test -e $(DESTDIR)$(LIB_DIR) && $(RMDIR) $(DESTDIR)$(LIB_DIR)
 	-test -e $(DESTDIR)$(INC_DIR) && $(RMDIR) $(DESTDIR)$(INC_DIR)
 	-test -e $(DESTDIR)$(MAN7_DIR) && $(RMDIR) $(DESTDIR)$(MAN7_DIR)
 
 uninstall: uninstalldirs
-	@[ "$(ON_WINDOWS)" = "1" ] && echo "Do not use install rules on Windows" && exit 1
 	$(RM)	$(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME).so.$(VERSION)
 	$(RM)	$(DESTDIR)$(LIB_DIR)/$(PROGRAM_NAME).so.$(VERSION_MAJOR)
 	$(RM)	$(DESTDIR)$(INC_DIR)/$(PROGRAM_NAME).a
