diff --git a/etc/openwsman.conf b/etc/openwsman.conf
index 84326f2..0d701c3 100644
--- a/etc/openwsman.conf
+++ b/etc/openwsman.conf
@@ -44,7 +44,7 @@ ssl_disabled_protocols = SSLv2 SSLv3
 #ssl_cipher_list = 
 
 # set these to enable digest authentication against a local datbase
-#digest_password_file = /etc/openwsman/digest_auth.passwd
+digest_password_file = /etc/openwsman/digest_auth.passwd
 
 # set these to enable basic authentication against a local datbase
 #basic_password_file = /etc/openwsman/simple_auth.passwd
@@ -59,7 +59,8 @@ max_connections_per_thread = 20
 # Authentication backend for BASIC authentication. Default is to read a configuration file defined with 'basic_password_file'
 #
 
-basic_authenticator = libwsman_pam_auth.so
+#By default, Alpine does not ship pam
+#basic_authenticator = libwsman_pam_auth.so
 basic_authenticator_arg = openwsman
 
 #
@@ -76,7 +77,7 @@ basic_authenticator_arg = openwsman
 # Location of plugins
 #  defaults to /usr/lib(64)/openwsman/plugins
 #
-# plugin_dir = /usr/lib/openwsman/plugins
+plugin_dir = /usr/lib/openwsman/plugins
 
 
 ##################################
