diff --git a/tox.ini b/tox.ini
index 4ffce74..5763603 100644
--- a/tox.ini
+++ b/tox.ini
@@ -49,12 +49,6 @@ commands =
     mypy src test
 
 [pytest]
-addopts =
-    --cov=versioningit
-    # Explicitly setting the path to the coverage config file is necessary due
-    # to some tests spawning subprocesses with changed working directories.
-    --cov-config=tox.ini
-    --no-cov-on-fail
 filterwarnings = error
 markers =
     describe_exclude: Tests that use `git describe --exclude` (Added in Git 2.13.0)
