From bb2f641258d4b22bef389b91e2b2698ca065097e Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Sat, 30 Aug 2025 14:22:04 +0200
Subject: [PATCH] Fix tests on aarch64

---
 test_prctl.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/test_prctl.py b/test_prctl.py
index 4852703..5b9aa1c 100644
--- a/test_prctl.py
+++ b/test_prctl.py
@@ -247,8 +247,8 @@ class PrctlTest(unittest.TestCase):
 
     @require('pac_reset_keys')
     def test_pac_reset_keys(self):
-        if self.arch == 'arm64':
-            # FIXME untested
+        if self.arch in ('arm64', 'aarch64'):
+            # FIXME untested on arm64
             self.assertEqual(prctl.pac_reset_keys(prctl.PAC_APIAKEY), None)
             self.assertRaises(ValueError, prctl.pac_reset_keys, 0xff)
         else:
@@ -385,7 +385,7 @@ class PrctlTest(unittest.TestCase):
     @require('set_tsc')
     def test_tsc(self):
         """Test manipulation of the timestamp counter flag"""
-        if re.match('i.86|x86_64', self.arch):
+        if re.match('i.86|x86_64', self.arch) or self.arch == 'aarch64':
             prctl.set_tsc(prctl.TSC_SIGSEGV)
             self.assertEqual(prctl.get_tsc(), prctl.TSC_SIGSEGV)
             prctl.set_tsc(prctl.TSC_ENABLE)
-- 
2.51.0

