--- virglrenderer-0.6.0/vtest/util.c
+++ virglrenderer-0.6.0/vtest/util.c.new
@@ -26,6 +26,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
+#include <sys/select.h>
 
 int vtest_wait_for_fd_read(int fd)
 {
--- virglrenderer-0.6.0/vtest/vtest_server.c
+++ virglrenderer-0.6.0/vtest/vtest_server.c.new
@@ -31,6 +31,7 @@
 #include <netinet/in.h>
 #include <sys/un.h>
 #include <fcntl.h>
+#include <string.h>
 
 #include "util.h"
 #include "vtest.h"
