From a7702c888571e8c13a02227c3fb680f7f63fa577 Mon Sep 17 00:00:00 2001
From: 6543 <6543@obermui.de>
Date: Tue, 28 Feb 2023 21:41:11 +0100
Subject: [PATCH] relax-version-dep-frozendict

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 4dc228e..f0ef910 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -156,12 +156,12 @@ jsonschema = ">=3.0.0"
 # frozendict 2.1.2 is broken on Debian 10: https://github.com/Marco-Sulla/python-frozendict/issues/41
 # We cannot test our wheels against the 2.3.5 release in CI. Putting in an upper bound for this
 # because frozendict has been more trouble than it's worth; we would like to move to immutabledict.
-frozendict = ">=1,!=2.1.2,<2.3.5"
+frozendict = ">2.1.2"
 # We require 2.1.0 or higher for type hints. Previous guard was >= 1.1.0
 unpaddedbase64 = ">=2.1.0"
 # We require 1.5.0 to work around an issue when running against the C implementation of
 # frozendict: https://github.com/matrix-org/python-canonicaljson/issues/36
-canonicaljson = "^1.5.0"
+canonicaljson = ">=0"
 # we use the type definitions added in signedjson 1.1.
 signedjson = "^1.1.0"
 # validating SSL certs for IP addresses requires service_identity 18.1.
