diff --git a/gns3server/web/web_server.py b/gns3server/web/web_server.py
index 8352134..53a8252 100644
--- a/gns3server/web/web_server.py
+++ b/gns3server/web/web_server.py
@@ -42,8 +42,8 @@ import gns3server.handlers
 import logging
 log = logging.getLogger(__name__)
 
-if not aiohttp.__version__.startswith("2.2"):
-    raise RuntimeError("aiohttp 2.2 is required to run the GNS3 server")
+if not aiohttp.__version__.startswith("2.3"):
+    raise RuntimeError("aiohttp 2.3 is required to run the GNS3 server")
 
 
 class WebServer:
diff --git a/gns3server/web/web_server.py b/gns3server/web/web_server.py
index 8352134..e999c11 100644
--- a/gns3server/web/web_server.py
+++ b/gns3server/web/web_server.py
@@ -101,7 +101,7 @@ class WebServer:
         if self._app:
             yield from self._app.shutdown()
         if self._handler:
-            yield from self._handler.finish_connections(2)  # Parameter is timeout
+            yield from self._handler.shutdown(2.0)  # Parameter is timeout
         if self._app:
             yield from self._app.cleanup()
 
