diff --git a/src/lha.h b/src/lha.h
index 07b2e3d..006be3b 100644
--- a/src/lha.h
+++ b/src/lha.h
@@ -215,7 +215,7 @@ int fnmatch(const char *pattern, const char *string, int flags);
 
 struct encode_option {
 #if defined(__STDC__) || defined(AIX)
-    void            (*output) ();
+    void            (*output) (int, int);
     void            (*encode_start) ();
     void            (*encode_end) ();
 #else
diff --git a/src/lhadd.c b/src/lhadd.c
index 5d0eca8..5144d77 100644
--- a/src/lhadd.c
+++ b/src/lhadd.c
@@ -9,7 +9,7 @@
 /* ------------------------------------------------------------------------ */
 #include "lha.h"
 /* ------------------------------------------------------------------------ */
-static void     remove_files();
+static void     remove_files(int filec, char **filev);
 
 static char     new_archive_name_buffer[FILENAME_LENGTH];
 static char    *new_archive_name;
diff --git a/src/slide.c b/src/slide.c
index 4b94130..136319d 100644
--- a/src/slide.c
+++ b/src/slide.c
@@ -37,11 +37,11 @@ static unsigned int *prev;      /* previous posiion associated with hash */
 static struct encode_option encode_define[2] = {
 #if defined(__STDC__) || defined(AIX)
     /* lh1 */
-    {(void (*) ()) output_dyn,
+    {(void (*) (int, int)) output_dyn,
      (void (*) ()) encode_start_fix,
      (void (*) ()) encode_end_dyn},
     /* lh4, 5, 6, 7 */
-    {(void (*) ()) output_st1,
+    {(void (*) (int, int)) output_st1,
      (void (*) ()) encode_start_st1,
      (void (*) ()) encode_end_st1}
 #else
