diff --git a/Makefile b/Makefile
index d4d7687..cdc7f04 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,7 @@
 # Set to where you'd like grepcidr installed
 INSTALLDIR=/usr/local
 INSTALLDIR_BIN=${INSTALLDIR}/bin
-INSTALLDIR_MAN=${INSTALLDIR}/man/man1
-#INSTALLDIR_MAN=${INSTALLDIR}/share/man/man1
+INSTALLDIR_MAN=${INSTALLDIR}/share/man/man1
 
 # Set to your favorite C compiler and flags
 # with GCC, -O3 makes a lot of difference
@@ -25,8 +24,8 @@ grepcidr:	grepcidr.c
 	$(CC) $(CFLAGS) -o grepcidr grepcidr.c
 
 install:	all  grepcidr.1
-	install grepcidr $(INSTALLDIR_BIN)
-	install -m 0644 grepcidr.1 $(INSTALLDIR_MAN)
+	install grepcidr $(INSTALLDIR_BIN)
+	install -m 0644 grepcidr.1 $(INSTALLDIR_MAN)
 
 clean:
 	rm -f grepcidr
