diff --git a/guile/modules/gnutls.in b/guile/modules/gnutls.in
index 451f384..bf38bc5 100644
--- a/guile/modules/gnutls.in
+++ b/guile/modules/gnutls.in
@@ -436,6 +436,7 @@
            pk-algorithm/gost-12-512
            pk-algorithm/ecdh-x448
            pk-algorithm/eddsa-ed448
+           pk-algorithm/rsa-oaep
            sign-algorithm/unknown
            sign-algorithm/rsa-sha1
            sign-algorithm/dsa-sha1
diff --git a/guile/modules/gnutls/build/enums.scm b/guile/modules/gnutls/build/enums.scm
index 658f4c7..752d172 100644
--- a/guile/modules/gnutls/build/enums.scm
+++ b/guile/modules/gnutls/build/enums.scm
@@ -637,7 +637,8 @@
                     gost-12-256
                     gost-12-512
                     ecdh-x448
-                    eddsa-ed448)
+                    eddsa-ed448
+                    rsa-oaep)
                   "gnutls_pk_algorithm_get_name"
                   "GNUTLS_PK_"))
 
