Author: Milan P. Stanić <mps@arvanta.net>
Date:   Thu Jun 3 09:10:21 2021 +0000

Fix missing PATH_MAX on ppc64le

--- a/alsactl/utils.c	2021-05-27 17:18:36.000000000 +0000
+++ b/alsactl/utils.c	2021-06-03 12:05:43.343126321 +0000
@@ -31,6 +31,7 @@
 #include <sys/stat.h>
 #include <sys/mman.h>
 #include "alsactl.h"
+#include <limits.h>
 
 int file_map(const char *filename, char **buf, size_t *bufsize)
 {
