From 94742e085ac96d366401df883df96d35d0eafd5a Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 1 Feb 2017 04:04:52 +0000
Subject: [PATCH 1/7] call /sbin/mkmntdirs in localmount OpenRC service

---
 init.d/localmount.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/init.d/localmount.in b/init.d/localmount.in
index cae80c34..3dbb375b 100644
--- a/init.d/localmount.in
+++ b/init.d/localmount.in
@@ -21,6 +21,8 @@ depend()
 
 start()
 {
+	[ -x /sbin/mkmntdirs ] && mkmntdirs
+
 	# Mount local filesystems in /etc/fstab.
 	# The types variable must start with no, and must be a type
 	local critical= types="noproc" x= no_netdev= rc=
-- 
2.11.0

