From 4d86d346354543f27058923587dd25f8c460363c Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Fri, 26 Mar 2010 22:10:03 +0000
Subject: [PATCH] Fix -s and -S.

---
 src/rc/rc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/rc/rc.c b/src/rc/rc.c
index 6d04b3d..ccafbe3 100644
--- a/src/rc/rc.c
+++ b/src/rc/rc.c
@@ -773,7 +773,7 @@ handle_bad_signal(int sig)
 #endif
 
 #include "_usage.h"
-#define getoptstring "o:" getoptstring_COMMON
+#define getoptstring "o:s:S" getoptstring_COMMON
 static const struct option longopts[] = {
 	{ "override", 1, NULL, 'o' },
 	{ "service",  1, NULL, 's' },
-- 
1.7.3.4

