From eb1d69ae84f6d2cb7f081dfca3251b2446a039f0 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 28 Nov 2017 13:35:10 +0100
Subject: [PATCH] fsck: don't add -C0 to busybox fsck

---
 init.d/fsck.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/init.d/fsck.in b/init.d/fsck.in
index 7052d808..99a9ae4f 100644
--- a/init.d/fsck.in
+++ b/init.d/fsck.in
@@ -82,7 +82,7 @@ start()
 		local skiptypes
 		skiptypes=$(printf 'no%s,' ${net_fs_list} ${extra_net_fs_list})
 		[ "${skiptypes}" = "no," ] && skiptypes=""
-		fsck_opts="$fsck_opts -C0 -T -t ${skiptypes}noopts=_netdev"
+		fsck_opts="$fsck_opts -T -t ${skiptypes}noopts=_netdev"
 		if [ -z "$fsck_passno" -a -z "$fsck_mnt" ]; then
 			fsck_args=${fsck_args:--A -p}
 			if echo 2>/dev/null >/.test.$$; then
@@ -90,6 +90,9 @@ start()
 				fsck_opts="$fsck_opts -R"
 			fi
 		fi
+		if [ "$(readlink -f $(which fsck))" != "/bin/busybox" ]; then
+			fsck_opts="$fsck_opts -C0"
+		fi
 	fi
 
 	trap : INT QUIT
-- 
2.15.0

