From 97f5ab74b2a159ebd01f9ed92a86e0ba5747edf4 Mon Sep 17 00:00:00 2001
From: "Anna (navi) Figueiredo Gomes" <navi@vlhl.dev>
Date: Fri, 25 Apr 2025 18:19:56 +0200
Subject: [PATCH] shared/misc: handle EOF when waiting for notify.

Bug: https://bugs.gentoo.org/952108
---
 src/shared/misc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/shared/misc.c b/src/shared/misc.c
index e57907e9..3305963b 100644
--- a/src/shared/misc.c
+++ b/src/shared/misc.c
@@ -585,6 +585,9 @@ bool notify_wait(const char *applet, struct notify notify)
 	for (;;) {
 		char buf[BUFSIZ];
 		ssize_t bytes = read(notify.fd, buf, BUFSIZ);
+
+		if (bytes == 0)
+			return false;
 		if (bytes == -1) {
 			if (errno != EINTR) {
 				eerror("%s: read failed '%s'\n", applet, strerror(errno));
