From 412a4881a1e1d8a4e0141d1c315cd261e1bb4cba Mon Sep 17 00:00:00 2001
From: "Anna (navi) Figueiredo Gomes" <navi@vlhl.dev>
Date: Sun, 27 Apr 2025 05:14:24 +0200
Subject: [PATCH] src/librc: loop readdir properly in rm_dir()

---
 src/librc/librc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/librc/librc.c b/src/librc/librc.c
index acb448a0..879e0171 100644
--- a/src/librc/librc.c
+++ b/src/librc/librc.c
@@ -138,7 +138,7 @@ rm_dir(int targetfd, const char *pathname, bool top)
 	if (!(dp = do_opendirat(targetfd, pathname)))
 		return false;
 
-	while (!(d = readdir(dp))) {
+	while ((d = readdir(dp))) {
 		if (strcmp(d->d_name, ".") == 0 || strcmp(d->d_name, "..") == 0)
 			continue;
 		if (fstatat(dirfd(dp), pathname, &s, 0) != 0) {
