From 3b5923663cd43a19563575362ceed66e13fa06f7 Mon Sep 17 00:00:00 2001
From: Ariadne Conill <ariadne@dereferenced.org>
Date: Tue, 7 Sep 2021 20:31:06 -0600
Subject: [PATCH] fix trailing whitespace in tools/meson_runlevels.sh

---
 tools/meson_runlevels.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/meson_runlevels.sh b/tools/meson_runlevels.sh
index 7cf068a9..4c38ef90 100755
--- a/tools/meson_runlevels.sh
+++ b/tools/meson_runlevels.sh
@@ -52,7 +52,7 @@ nonetwork="local"
 shutdown="savecache"
 case "${os}" in
 	Linux)
-		shutdown="${shutdown} killprocs mount-ro" 
+		shutdown="${shutdown} killprocs mount-ro"
 		;;
 esac
 
-- 
2.33.0

