Index: src/librc/librc-depend.c
===================================================================
--- a/src/librc/librc-depend.c (revision b4104957b198b38aba609ed3889e7fa605012d19)
+++ b/src/librc/librc-depend.c (revision 7fbb6ad42742ebc5e6ec244eedc85e9ac0a6236d)
@@ -802,4 +802,9 @@
 				continue;
 			}
+
+			/* Don't provide ourself */
+			if (strcmp(type, "iprovide") == 0 &&
+			    strcmp(depend, service) == 0)
+				continue;
 
 			/* .sh files are not init scripts */
