diff -ru openrc-0.5.0.orig/init.d/localmount.in openrc-0.5.0/init.d/localmount.in
--- openrc-0.5.0.orig/init.d/localmount.in	2009-10-09 09:08:43.000000000 +0000
+++ openrc-0.5.0/init.d/localmount.in	2009-10-09 09:09:08.000000000 +0000
@@ -31,7 +31,7 @@
 {
 	# We never unmount / or /dev or $RC_SVCDIR
 	local x= no_umounts_r="/|/dev|/dev/.*|${RC_SVCDIR}"
-	no_umounts_r="${no_umounts_r}|/bin|/sbin|/lib|/libexec"
+	no_umounts_r="${no_umounts_r}|/bin|/sbin|/lib|/libexec|/.modloop"
 	# RC_NO_UMOUNTS is an env var that can be set by plugins
 	OIFS=$IFS SIFS=${IFS-y}
 	IFS=$IFS:
