diff --git a/init.d/sysfs.in b/init.d/sysfs.in
index 4f214f6..759f246 100644
--- a/init.d/sysfs.in
+++ b/init.d/sysfs.in
@@ -96,7 +96,7 @@ mount_misc()
 		modprobe -q efivarfs
 		if grep -qs efivarfs /proc/filesystems; then
 			ebegin "Mounting efivarfs filesystem"
-			mount -n -t efivarfs -o ${sysfs_opts} \
+			mount -n -t efivarfs -o ro,${sysfs_opts} \
 				efivarfs /sys/firmware/efi/efivars
 			eend $?
 		fi
