diff --git a/init.d/swap.in b/init.d/swap.in
index fef8ab3..c540bc2 100644
--- a/init.d/swap.in
+++ b/init.d/swap.in
@@ -24,8 +24,10 @@ stop()
 
 	# Try to unmount all tmpfs filesystems not in use, else a deadlock may
 	# occure. As $RC_SVCDIR may also be tmpfs we cd to it to lock it
-	cd "$RC_SVCDIR"
-	umount -a -t tmpfs 2>/dev/null
+	if [ "$RC_RUNLEVEL" = "shutdown" ]; then
+		cd "$RC_SVCDIR"
+		umount -a -t tmpfs 2>/dev/null
+	fi
 
 	case "$RC_UNAME" in
 		NetBSD|OpenBSD)	swapctl -U -t noblk >/dev/null;;
