diff --git a/init.d/hwclock.in b/init.d/hwclock.in
index 024bff0..758f1fa 100644
--- a/init.d/hwclock.in
+++ b/init.d/hwclock.in
@@ -41,7 +41,7 @@ setupopts()
 {
 	case "$(uname -m)" in
 		s390*)
-			utc="s390"
+			utc="UTC"
 			;;
 		*)
 			if [ -e /proc/devices ] && \
@@ -98,7 +98,7 @@ start()
 	fi
 
 	ebegin "Setting system clock using the hardware clock [$utc]"
-	if [ -e /proc/modules ]; then
+	if [ -e /proc/modules ] && [ "$(arch)" != "s390x" ]; then
 		if ! rtc_exists; then
 			for x in rtc-cmos rtc genrtc; do
 				modprobe -q $x && rtc_exists && modname="$x" && break
@@ -118,7 +118,10 @@ start()
 		: $(( retval += $? ))
 	fi
 
-	if yesno ${clock_hctosys:-YES}; then
+	if [ "$(arch)" = "s390x" ]; then
+		date -s "@$(/bin/tod_s390x)"
+		: $(( retval += $? ))
+	elif yesno ${clock_hctosys:-YES}; then
 		_hwclock --hctosys $utc_cmd $(get_noadjfile) $clock_args
 		: $(( retval += $? ))
 	fi
@@ -132,6 +135,7 @@ stop()
 {
 	# Don't tweak the hardware clock on LiveCD halt.
 	[ -n "$CDBOOT" ] && return 0
+	[ "$(arch)" = "s390x" ] && return 0
 	yesno ${clock_systohc:-YES} || return 0
 
 	local retval=0 errstr=""
