diff --git a/dynagen b/dynagen
index 81ebdec..5980bcc 100755
--- a/dynagen
+++ b/dynagen
@@ -43,7 +43,7 @@ from optparse import OptionParser
 VERSION = '0.11.0'
 CONFIGSPECPATH = ['/usr/share/dynagen', '/usr/local/share']
 CONFIGSPEC = 'configspec'
-INIPATH = ['/etc', '/usr/local/etc']
+INIPATH = ['/etc/dynagen', '/etc']
 INIFILE = 'dynagen.ini'
 MODELTUPLE = (  # A tuple of known model objects
     C1700,
diff --git a/dynagen.ini b/dynagen.ini
index 885090b..7bec1ed 100644
--- a/dynagen.ini
+++ b/dynagen.ini
@@ -3,7 +3,7 @@
 # Specify the file that stores the idle-pc database
 # Defaults to the user's home directory
 # Usage:
-# idledb = /usr/local/etc/dynagenidledb.ini
+# idledb = /etc/dynagen/dynagenidledb.ini
 
 # -----------------------------------------------------------------------------
 # udp option
