diff --git a/libfwupdplugin/fu-common.c b/libfwupdplugin/fu-common.c
index 7ad33f8..f3423a7 100644
--- a/libfwupdplugin/fu-common.c
+++ b/libfwupdplugin/fu-common.c
@@ -501,7 +501,6 @@ fu_common_firmware_builder (GBytes *bytes,
 	fu_common_add_argv (argv, "--die-with-parent");
 	fu_common_add_argv (argv, "--ro-bind /usr /usr");
 	fu_common_add_argv (argv, "--ro-bind /lib /lib");
-	fu_common_add_argv (argv, "--ro-bind /lib64 /lib64");
 	fu_common_add_argv (argv, "--ro-bind /bin /bin");
 	fu_common_add_argv (argv, "--ro-bind /sbin /sbin");
 	fu_common_add_argv (argv, "--dir /tmp");
