diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3180626..102b8c8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -324,7 +324,7 @@ if( TOMAHAWK_QT5 )
 else()
     macro_optional_find_package(QtKeychain 0.1.0)
 endif()
-macro_log_feature(QTKEYCHAIN_FOUND "QtKeychain" "Provides support for secure credentials storage" "https://github.com/frankosterfeld/qtkeychain" TRUE "" "")
+macro_log_feature(QtKeychain_FOUND "QtKeychain" "Provides support for secure credentials storage" "https://github.com/frankosterfeld/qtkeychain" TRUE "" "")
 endif()
 
 if( UNIX AND NOT APPLE )
