Ref https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/199

diff --git a/meson.build b/meson.build
--- a/meson.build
+++ b/meson.build
@@ -183,11 +183,13 @@ python3_test_modules = {
 }
 python3_available_modules = []
 
+if get_option('tests')
 foreach module, required : python3_test_modules
     if required and run_command(python3, '-c', 'import @0@'.format(module)).returncode() != 0
         warning('Python3 module \'' + module + '\' required by test suite not found')
     endif
 endforeach
+endif
 
 cdata = configuration_data()
 cdata.set_quoted('GETTEXT_PACKAGE', meson.project_name())
@@ -202,6 +204,8 @@ config_h = configure_file(
   configuration: cdata
 )
 
+address_sanitizer = get_option('b_sanitize') == 'address'
+
 subdir('src')
 subdir('data')
 subdir('utils')
@@ -211,7 +215,9 @@ endif
 if get_option('gtk_doc')
     subdir('doc')
 endif
-subdir('tests')
+if get_option('tests')
+    subdir('tests')
+endif
 subdir('po')
 
 output = []
diff --git a/tests/meson.build b/tests/meson.build
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -16,8 +16,6 @@ python_tests = [
     # }
 ]
 
-address_sanitizer = get_option('b_sanitize') == 'address'
-
 tests = [
     'fprintd',
     'test_fprintd_utils',
diff --git a/meson_options.txt b/meson_options.txt
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -23,3 +23,7 @@
     type: 'boolean',
     value: false,
     description: 'Use gtk-doc to build documentation')
+option('tests',
+    type: 'boolean',
+    value: true,
+    description: 'Enable tests')
