see https://gitlab.alpinelinux.org/alpine/aports/-/issues/13235
--- a/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in
+++ b/taskcluster/docker/firefox-flatpak/org.mozilla.firefox.appdata.xml.in
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <application>
-  <id type="desktop">org.mozilla.firefox</id>
+  <id type="desktop">org.mozilla.firefoxesr</id>
   <launchable type="desktop-id">org.mozilla.firefox.desktop</launchable>
-  <name>Firefox</name>
-  <summary>Mozilla Firefox Web Browser</summary>
+  <name>Firefox ESR</name>
+  <summary>Mozilla Firefox Web Browser Extended Support Release</summary>
   <metadata_license>CC0-1.0</metadata_license>
   <project_license>MPL-2.0</project_license>
   <description>
