diff -upr /tmp/firefox-60.5.0.orig/tools/profiler/core/platform-linux-android.cpp firefox-60.5.0/tools/profiler/core/platform-linux-android.cpp
--- /tmp/firefox-60.5.0.orig/tools/profiler/core/platform-linux-android.cpp	2019-02-11 18:55:48.543925300 +0100
+++ firefox-60.5.0/tools/profiler/core/platform-linux-android.cpp	2019-02-12 10:00:02.735569929 +0100
@@ -497,8 +497,10 @@ static void PlatformInit(PSLockRef aLock
 ucontext_t sSyncUContext;
 
 void Registers::SyncPopulate() {
+#if defined(__GLIBC__)
   if (!getcontext(&sSyncUContext)) {
     PopulateRegsFromContext(*this, &sSyncUContext);
   }
+#endif
 }
 #endif
diff -upr /tmp/firefox-60.5.0.orig/tools/profiler/core/platform.h firefox-60.5.0/tools/profiler/core/platform.h
--- /tmp/firefox-60.5.0.orig/tools/profiler/core/platform.h	2019-02-11 18:55:48.540591965 +0100
+++ firefox-60.5.0/tools/profiler/core/platform.h	2019-02-12 10:00:02.735569929 +0100
@@ -29,6 +29,8 @@
 #ifndef TOOLS_PLATFORM_H_
 #define TOOLS_PLATFORM_H_
 
+#include <sys/types.h>
+
 #include <stdint.h>
 #include <math.h>
 #include "MainThreadUtils.h"
diff -upr /tmp/firefox-60.5.0.orig/tools/profiler/lul/LulElf.cpp firefox-60.5.0/tools/profiler/lul/LulElf.cpp
--- /tmp/firefox-60.5.0.orig/tools/profiler/lul/LulElf.cpp	2019-02-11 18:55:48.547258635 +0100
+++ firefox-60.5.0/tools/profiler/lul/LulElf.cpp	2019-02-12 10:00:59.802296448 +0100
@@ -459,10 +459,10 @@ string FormatIdentifier(unsigned char id
 // Return the non-directory portion of FILENAME: the portion after the
 // last slash, or the whole filename if there are no slashes.
 string BaseFileName(const string& filename) {
-  // Lots of copies!  basename's behavior is less than ideal.
-  char* c_filename = strdup(filename.c_str());
-  string base = basename(c_filename);
-  free(c_filename);
+  // basename's behavior is less than ideal so avoid it
+  const char *c_filename = filename.c_str();
+  const char *p = strrchr(c_filename, '/');
+  string base = p ? p+1 : c_filename;
   return base;
 }
 
