diff --git a/media/libaom/moz.build b/media/libaom/moz.build
index fd8c7f2..db3f9ca 100644
--- a/media/libaom/moz.build
+++ b/media/libaom/moz.build
@@ -55,7 +55,7 @@ elif CONFIG['CPU_ARCH'] == 'arm':
 
     for f in SOURCES:
         if f.endswith('neon.c'):
-            SOURCES[f].flags += CONFIG['VPX_ASFLAGS']
+            SOURCES[f].flags += CONFIG['NEON_FLAGS']
 
     if CONFIG['OS_TARGET'] == 'Android':
         # For cpu-features.h
