--- a/gfx/2d/Rect.h
+++ b/gfx/2d/Rect.h
@@ -14,6 +14,7 @@
 #include "Tools.h"
 #include "mozilla/Maybe.h"
 
+#include <cstdint>
 #include <cmath>
 
 namespace mozilla {
--- a/dom/media/webrtc/sdp/RsdparsaSdpGlue.cpp
+++ b/dom/media/webrtc/sdp/RsdparsaSdpGlue.cpp
@@ -3,6 +3,7 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <cstdint>
 #include <string>
 
 #include "sdp/RsdparsaSdpInc.h"
--- a/toolkit/components/telemetry/pingsender/pingsender.cpp
+++ b/toolkit/components/telemetry/pingsender/pingsender.cpp
@@ -3,6 +3,7 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+#include <cstdint>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
