diff --git a/scripts/cargo-doc.sh b/scripts/cargo-doc.sh
index 06df3ac..b88cb99 100755
--- a/scripts/cargo-doc.sh
+++ b/scripts/cargo-doc.sh
@@ -2,7 +2,6 @@
 
 export OUTPUT="$2"
 export CARGO_TARGET_DIR="$3"/target
-export CARGO_HOME="$CARGO_TARGET_DIR"/cargo-home
 export FRACTAL_PROFILE="$4"
 
 echo "GENERATING DOCUMENTATION"
diff --git a/scripts/cargo.sh b/scripts/cargo.sh
index a8c7763..bebdc48 100755
--- a/scripts/cargo.sh
+++ b/scripts/cargo.sh
@@ -2,7 +2,6 @@
 
 export OUTPUT="$2"
 export CARGO_TARGET_DIR="$3"/target
-export CARGO_HOME="$CARGO_TARGET_DIR"/cargo-home
 export FRACTAL_PROFILE="$4"
 
 if [ "$FRACTAL_PROFILE" = "Devel" ]
diff --git a/scripts/test.sh b/scripts/test.sh
index 35ed241..4ba4e35 100755
--- a/scripts/test.sh
+++ b/scripts/test.sh
@@ -4,6 +4,5 @@ set -o errexit
 set -o pipefail
 
 export CARGO_TARGET_DIR="$1/target/"
-export CARGO_HOME="$CARGO_TARGET_DIR/cargo-home"
 
 cargo test -j 1 -- --test-threads=1 --nocapture
