From 1b64651badf2cb91d1582241adf06b64ee39d3f3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?K=C3=A9vin=20Commaille?= <zecakeh@tedomum.fr>
Date: Mon, 15 Dec 2025 10:35:18 +0100
Subject: [PATCH] utils: Remove unused import

---
 src/utils/matrix/mod.rs | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/utils/matrix/mod.rs b/src/utils/matrix/mod.rs
index 4f491c70a..4991f78c1 100644
--- a/src/utils/matrix/mod.rs
+++ b/src/utils/matrix/mod.rs
@@ -27,7 +27,6 @@ use ruma::{
     serde::Raw,
 };
 use thiserror::Error;
-use tracing::error;
 
 pub(crate) mod ext_traits;
 mod media_message;
-- 
GitLab

