diff -Naur a/common/utils.c b/common/utils.c
--- a/common/utils.c	2025-09-29 23:20:32.358886395 -0400
+++ b/common/utils.c	2025-10-01 22:28:21.606790266 -0400
@@ -1,5 +1,9 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include "utils.h"
-#include "string.h"
+#include <string.h>
 #include <dlfcn.h>
 #include <sys/stat.h>
 #include <errno.h>
diff -Naur a/protocol/hp_ipp.c b/protocol/hp_ipp.c
--- a/protocol/hp_ipp.c	2025-09-29 23:20:32.404885807 -0400
+++ b/protocol/hp_ipp.c	2025-10-01 22:28:16.173861434 -0400
@@ -18,6 +18,10 @@
 
 \******************************************************************************/
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
diff -Naur a/scan/sane/bb_ledm.c b/scan/sane/bb_ledm.c
--- a/scan/sane/bb_ledm.c	2025-09-29 23:20:32.408885756 -0400
+++ b/scan/sane/bb_ledm.c	2025-10-01 22:29:23.498979537 -0400
@@ -10,6 +10,7 @@
 # define _GNU_SOURCE
 # endif
 
+# include <unistd.h>
 # include <stdarg.h>
 # include <syslog.h>
 # include <stdio.h>
diff -Naur a/scan/sane/hpaio.c b/scan/sane/hpaio.c
--- a/scan/sane/hpaio.c	2025-09-29 23:20:32.408885756 -0400
+++ b/scan/sane/hpaio.c	2025-10-01 22:44:50.752840311 -0400
@@ -50,6 +50,7 @@
 #include "sclpml.h"
 #include "escl.h"
 #include "io.h"
+#include "orblite.h"
 #include "orblitei.h"
 
 
diff -Naur a/scan/sane/http.c b/scan/sane/http.c
--- a/scan/sane/http.c	2025-09-29 23:20:32.409885743 -0400
+++ b/scan/sane/http.c	2025-10-01 22:29:59.694505411 -0400
@@ -31,6 +31,7 @@
 #define _GNU_SOURCE
 #endif
 
+#include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -Naur a/scan/sane/orblite.c b/scan/sane/orblite.c
--- a/scan/sane/orblite.c	2025-09-29 23:20:32.409885743 -0400
+++ b/scan/sane/orblite.c	2025-10-01 22:30:46.119897284 -0400
@@ -14,6 +14,7 @@
  extern "C" {
 #endif
 
+#include <dlfcn.h>
 #include "sane.h"           /* SANE API */
 #define DEBUG_NOT_STATIC
 #undef NDEBUG
diff -Naur a/scan/sane/sclpml.c b/scan/sane/sclpml.c
--- a/scan/sane/sclpml.c	2025-09-29 23:20:32.409885743 -0400
+++ b/scan/sane/sclpml.c	2025-10-01 22:31:47.766089783 -0400
@@ -25,11 +25,16 @@
 
 \************************************************************************************/
 
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <string.h>
+#include <sys/time.h>
 #include "sane.h"
 #include "saneopts.h"
 #include "common.h"
