diff -Naur a/protocol/hp_ipp.h b/protocol/hp_ipp.h
--- a/protocol/hp_ipp.h	2025-09-29 23:20:32.404885807 -0400
+++ b/protocol/hp_ipp.h	2025-10-01 22:41:38.899349462 -0400
@@ -149,6 +149,7 @@
 
 
 http_t*    acquireCupsInstance();
+void _releaseCupsInstance(void);
 const char *getCupsErrorString(int status);
 void freePrinterList(printer_t *list);
  
@@ -162,6 +163,10 @@
 ipp_t *usbDoFileRequest(ipp_t *request, int iFileHandle,char *device_uri, const char *resource);
 ipp_t * networkDoRequest(ipp_t *request, char* device_uri,const char *resource);
 ipp_t * getDeviceStatusAttributes(char* device_uri,char* printer_name, int *count);
+int addCupsPrinter(char *name, char *device_uri, char *location, char *ppd_file, char *model, char *info);
+int delCupsPrinter(char *pr_name);
+int setDefaultCupsPrinter(char *pr_name);
+int controlCupsPrinter(char *pr_name, int op);
 int     getCupsPrinters(printer_t **printer_list);
 
 HPIPP_RESULT parseResponseHeader(char* header, int *content_length, int *chunked, int* header_size);
