diff -ru hplip-3.16.10.orig/protocol/discovery/mdns.c hplip-3.16.10/protocol/discovery/mdns.c
--- hplip-3.16.10.orig/protocol/discovery/mdns.c	2016-10-21 13:48:34.000000000 +0300
+++ hplip-3.16.10/protocol/discovery/mdns.c	2016-11-03 14:41:38.828298853 +0200
@@ -24,10 +24,15 @@
  Author: Sanjay Kumar
  \*****************************************************************************/
 
-//#include <stdio.h>
+#include <stdio.h>
+#include <ctype.h>
+#include <fcntl.h>
+#include <unistd.h>
 #include <string.h>
 #include <syslog.h>
+#include <malloc.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 #include <netinet/in.h>
 #include <netdb.h>
 #include <arpa/inet.h>
