--- a/createPPD.sh
+++ b/createPPD.sh
@@ -81,6 +81,7 @@
 install -d ppd/hpcups
 ppdc -d ppd/hpcups prnt/drv/hpcups.drv
 for i in ppd/hpcups/*.ppd; do
+      [ -f "$i" ] || continue
       gzip -c "$i" > "$i.gz"
 done
 
@@ -88,6 +89,7 @@
 ppdc -d ppd/hpijs prnt/drv/hpijs.drv
 echo "gzip ppd files..."
 for i in ppd/hpijs/*.ppd; do
+      [ -f "$i" ] || continue
       gzip -c "$i" > "$i.gz"
 done
 echo "building foomatic_drv.inc..."
