diff --git a/configure.in b/configure.in
index 470486d..d6acb9d 100755
--- a/configure.in
+++ b/configure.in
@@ -611,10 +611,6 @@ if test "$class_driver" = "no" && test "$hpijs_only_build" = "no" && test "$lite
    PYTHONINCLUDEDIR=`$PYTHON -c "from distutils.sysconfig import get_python_inc; print (get_python_inc());"`
    AC_MSG_RESULT("using $PYTHONINCLUDEDIR ....  python${PYTHON_VERSION}/Python.h")
    AC_ARG_VAR(PYTHONINCLUDEDIR, [path to Python.h C header file])
-   AC_CHECK_HEADERS([python${PYTHON_VERSION}/Python.h python${PYTHON_VERSION}mu/Python.h python${PYTHON_VERSION}m/Python.h ],
-          [FOUND_HEADER=yes; break;])
-   AS_IF([test "x$FOUND_HEADER" != "xyes"],
-          [AC_MSG_ERROR([cannot find python-devel support], 6)])
 fi
 
 if test "$hpijs_only_build" = "no" && test "$scan_build" = "yes" && test "$hpcups_only_build" = "no"; then
