diff -Naur a/prnt/hpps/pserror.c b/prnt/hpps/pserror.c
--- a/prnt/hpps/pserror.c	2025-09-29 23:20:32.397885897 -0400
+++ b/prnt/hpps/pserror.c	2025-10-01 22:35:09.565446410 -0400
@@ -24,7 +24,7 @@
 void message(int flags, char *format, ...)
 {
   va_list args ;
-  static column = 0 ;		/* current screen column for message wrap */
+  static int column = 0 ;		/* current screen column for message wrap */
   char msgbuf[MAX_MESSAGE] ;	/* buffer in which to put the message */
   char *bufptr = msgbuf ;	/* message buffer pointer */
 
