--- a/libuuu/buffer.cpp
+++ b/libuuu/buffer.cpp
@@ -54,7 +54,7 @@
 #define stat_os stat
 #include "dirent.h"
 #else
-#define stat_os stat64
+#define stat_os stat
 #include "dirent.h"
 #endif
 
@@ -1350,7 +1350,7 @@
 			return -1;
 		}
 
-		m_pDatabuffer = (uint8_t *)mmap64(0, sz, PROT_READ, MAP_SHARED, fd, 0);
+		m_pDatabuffer = (uint8_t *)mmap(0, sz, PROT_READ, MAP_SHARED, fd, 0);
 		if (m_pDatabuffer == MAP_FAILED) {
 			m_pDatabuffer = nullptr;
 			set_last_err_string("mmap failure\n");
