Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: use language level 3 for cython
----

--- a/setup.py
+++ b/setup.py
@@ -304,7 +304,7 @@ if os.path.exists("MANIFEST.in") and "cl
             'rasterio.shutil', ['rasterio/shutil.pyx'], **ext_options),
         Extension(
             'rasterio._transform', ['rasterio/_transform.pyx'], **ext_options)],
-        quiet=True, **cythonize_options)
+        quiet=True, compiler_directives={'language_level' : "2"}, **cythonize_options)

 # If there's no manifest template, as in an sdist, we just specify .c files.
 else:
