Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: GDAL does not read HDF5
----

--- a/tests/test_read.py
+++ b/tests/test_read.py
@@ -259,14 +259,6 @@ class ReaderContextTest(unittest.TestCas
             self.assertEqual(a.shape, (1, 2, 2))
             self.assertTrue(hasattr(a, 'mask'))

-    @pytest.mark.skipif(not has_hdf, reason="HDF driver not available")
-    def test_read_no_band(self):
-        with rasterio.open('tests/data/no_band.h5') as s:
-            self.assertEqual(s.count, 0)
-            self.assertEqual(s.meta['dtype'], 'float_')
-            self.assertIsNone(s.meta['nodata'])
-            self.assertRaises(ValueError, s.read)
-

 @pytest.mark.parametrize("shape,indexes", [
     ((72, 80), 1),          # Single band
