diff --git a/utils/test/test-aa-notify.py b/utils/test/test-aa-notify.py
index 2484c7f..9013167 100644
--- a/utils/test/test-aa-notify.py
+++ b/utils/test/test-aa-notify.py
@@ -154,7 +154,7 @@ Feb  4 13:40:38 XPS-13-9370 kernel: [128552.880347] audit: type=1400 audit({epoc
 
 Display AppArmor notifications or messages for DENIED entries.
 
-optional arguments:
+options:
   -h, --help            show this help message and exit
   -p, --poll            poll AppArmor logs and display notifications
   --display DISPLAY     set the DISPLAY environment variable (might be needed if
