--- a/packages/process/_test.pony
+++ b/packages/process/_test.pony
@@ -63,7 +63,7 @@
 
   fun ref apply(h: TestHelper) =>
     let notifier: ProcessNotify iso = _ProcessClient(0,
-      "cat: file_does_not_exist: No such file or directory\n", 1, h)
+      "cat: can't open 'file_does_not_exist': No such file or directory\n", 1, h)
     try
       let path = FilePath(h.env.root as AmbientAuth, "/bin/cat")?
       let args: Array[String] iso = recover Array[String](2) end
