--- a/Makefile
+++ b/Makefile
@@ -40,7 +40,6 @@
 # Default settings (silent release build).
 config ?= release
 arch ?= native
-tune ?= generic
 bits ?= $(shell getconf LONG_BIT)
 
 ifndef verbose
@@ -90,9 +89,9 @@
 destdir ?= $(prefix)/lib/pony/$(tag)
 
 LIB_EXT ?= a
-BUILD_FLAGS = -march=$(arch) -mtune=$(tune) -Werror -Wconversion \
+BUILD_FLAGS = -march=$(arch) -Werror -Wconversion \
   -Wno-sign-conversion -Wextra -Wall
-LINKER_FLAGS = -march=$(arch) -mtune=$(tune)
+LINKER_FLAGS = -march=$(arch)
 AR_FLAGS ?= rcs
 ALL_CFLAGS = -std=gnu11 -fexceptions \
   -DPONY_VERSION=\"$(tag)\" -DLLVM_VERSION=\"$(llvm_version)\" \
