Don't hassle with -march and -mtune, use gcc defaults.

--- a/Makefile
+++ b/Makefile
@@ -90,9 +90,9 @@
 destdir ?= $(prefix)/lib/pony/$(tag)
 
 LIB_EXT ?= a
-BUILD_FLAGS = -march=$(arch) -mtune=$(tune) -Werror -Wconversion \
+BUILD_FLAGS = -Werror -Wconversion \
   -Wno-sign-conversion -Wextra -Wall
-LINKER_FLAGS = -march=$(arch) -mtune=$(tune)
+LINKER_FLAGS =
 AR_FLAGS ?= rcs
 ALL_CFLAGS = -std=gnu11 -fexceptions \
   -DPONY_VERSION=\"$(tag)\" -DLLVM_VERSION=\"$(llvm_version)\" \
@@ -102,7 +102,7 @@
   -DPONY_VERSION_STR=\"$(version_str)\" \
   -D_FILE_OFFSET_BITS=64
 ALL_CXXFLAGS = -std=gnu++11 -fno-rtti
-LL_FLAGS = -mcpu=$(cpu)
+LL_FLAGS =
 
 # Determine pointer size in bits.
 BITS := $(bits)
