From b44f99b7ab158f3e0cf3a2344fcc58d90982378c Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 27 May 2010 09:31:03 +0000
Subject: [PATCH] create install dir

---
 konference/Makefile |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/konference/Makefile b/konference/Makefile
index 9cc2201..e0901df 100644
--- a/konference/Makefile
+++ b/konference/Makefile
@@ -158,6 +158,7 @@ vad_test: vad_test.o libspeex/preprocess.o libspeex/misc.o libspeex/smallft.o
 	$(CC) $(PROFILE) -o $@ $^ -lm
 
 install:
+	$(INSTALL) -d $(INSTALL_MODULES_DIR)
 	$(INSTALL) -m 755 $(TARGET) $(INSTALL_MODULES_DIR)
 
 
-- 
1.7.1

