diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index fe4f2c3f1..729d4d3e9 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -544,6 +544,10 @@ foreach(arch ${archs})
                     )
                 endif()
             endif()
+            if(cpu MATCHES "arm.*")
+                # https://bugreports.qt.io/browse/QTBUG-129985
+                list(APPEND gnArgArg build_webnn_with_xnnpack=false)
+            endif()
             unset(cpu)
         endif()
