diff --git a/configure.cmake b/configure.cmake
index 213eab5..2ab62c5 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -321,15 +321,6 @@ qt_webengine_configure_check("static-build"
     MESSAGE "Static build is not supported."
 )
 
-qt_webengine_configure_check("nodejs"
-    MODULES QtWebEngine
-    CONDITION TARGET Nodejs::Nodejs AND
-        NOT (Nodejs_ARCH STREQUAL "ia32") AND
-        NOT (Nodejs_ARCH STREQUAL "x86") AND
-        NOT (Nodejs_ARCH STREQUAL "arm")
-    MESSAGE "64-bit Node.js ${QT_CONFIGURE_CHECK_nodejs_version} version or later is required."
-    DOCUMENTATION "64-bit Nodejs ${QT_CONFIGURE_CHECK_nodejs_version} version or later."
-)
 qt_webengine_configure_check("python3"
     MODULES QtWebEngine QtPdf
     CONDITION Python3_FOUND
