diff --git a/src/3rdparty/chromium/BUILD.gn b/src/3rdparty/chromium/BUILD.gn
index cbf745544ac..9fc90085bd1 100644
--- a/src/3rdparty/chromium/BUILD.gn
+++ b/src/3rdparty/chromium/BUILD.gn
@@ -1628,8 +1628,7 @@ if (use_blink && !is_cronet_build) {
 
 # TODO(cassew): Add more OS's that don't support x86.
 is_valid_x86_target =
-    target_os != "ios" && target_os != "mac" &&
-    (target_os != "linux" || use_libfuzzer || !build_with_chromium)
+    target_os != "ios" && target_os != "mac"
 
 # Note: v8_target_cpu == arm allows using the V8 arm simulator on x86 for fuzzing.
 assert(
diff --git a/src/3rdparty/chromium/third_party/protobuf/src/google/protobuf/arenastring.cc b/src/3rdparty/chromium/third_party/protobuf/src/google/protobuf/arenastring.cc
index d886ddad66a..5ada1b5af4d 100644
--- a/src/3rdparty/chromium/third_party/protobuf/src/google/protobuf/arenastring.cc
+++ b/src/3rdparty/chromium/third_party/protobuf/src/google/protobuf/arenastring.cc
@@ -64,8 +64,6 @@ constexpr size_t kNewAlign = alignof(std::max_align_t);
 #endif
 constexpr size_t kStringAlign = alignof(std::string);
 
-static_assert((kStringAlign > kNewAlign ? kStringAlign : kNewAlign) >= 8, "");
-static_assert(alignof(ExplicitlyConstructedArenaString) >= 8, "");
 
 }  // namespace
 
diff --git a/src/3rdparty/chromium/v8/BUILD.gn b/src/3rdparty/chromium/v8/BUILD.gn
index b83b22ea0d5..d5b9332bdee 100644
--- a/src/3rdparty/chromium/v8/BUILD.gn
+++ b/src/3rdparty/chromium/v8/BUILD.gn
@@ -752,7 +752,7 @@ config("internal_config") {
     defines += [ "BUILDING_V8_SHARED" ]
   }
 
-  if (v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32") {
+  if (v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32" || v8_current_cpu == "x86") {
     if (!is_clang) {
       libs = [ "atomic" ]
     }
@@ -5878,6 +5878,7 @@ v8_source_set("v8_base_without_compiler") {
   if (v8_current_cpu == "mips64" || v8_current_cpu == "mips64el" ||
       v8_current_cpu == "ppc" || v8_current_cpu == "ppc64" ||
       v8_current_cpu == "s390" || v8_current_cpu == "s390x" ||
+      v8_current_cpu == "x86" ||
       v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32") {
     if (!is_clang) {
       libs += [ "atomic" ]
@@ -6272,7 +6273,7 @@ v8_component("v8_libbase") {
     sources += [ "src/base/ubsan.cc" ]
   }
 
-  if (v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32") {
+  if (v8_current_cpu == "riscv64" || v8_current_cpu == "riscv32" || v8_current_cpu == "x86") {
     if (!is_clang) {
       libs += [ "atomic" ]
     }
diff --git a/configure.cmake b/configure.cmake
index 7ccb67a..4af3f4f 100644
--- a/configure.cmake
+++ b/configure.cmake
@@ -492,11 +492,4 @@
    MESSAGE "node.js version 14 or later is required."
 )
-add_check_for_support(
-    MODULES QtWebEngine
-    CONDITION NOT (Nodejs_ARCH STREQUAL "ia32") AND
-              NOT (Nodejs_ARCH STREQUAL "x86") AND
-              NOT (Nodejs_ARCH STREQUAL "arm")
-    MESSAGE "32bit version of Nodejs is not supported."
-)
 add_check_for_support(
    MODULES QtWebEngine QtPdf
