diff --git a/src/3rdparty/chromium/components/variations/seed_response.h b/src/3rdparty/chromium/components/variations/seed_response.h
index d17d74c447..4e8b358338 100644
--- a/src/3rdparty/chromium/components/variations/seed_response.h
+++ b/src/3rdparty/chromium/components/variations/seed_response.h
@@ -5,6 +5,7 @@
 #ifndef COMPONENTS_VARIATIONS_SEED_RESPONSE_H_
 #define COMPONENTS_VARIATIONS_SEED_RESPONSE_H_
 
+#include <cstdint>
 #include <string>
 
 #include "base/component_export.h"
