diff --git a/src/3rdparty/chromium/third_party/libyuv/BUILD.gn b/src/3rdparty/chromium/third_party/libyuv/BUILD.gn
index d3f8dcf72..cfa9aec05 100644
--- a/src/3rdparty/chromium/third_party/libyuv/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/libyuv/BUILD.gn
@@ -183,6 +183,12 @@ static_library("libyuv_internal") {
       "-ffp-contract=fast",  # Enable fma vectorization for NEON.
     ]
   }
+
+  if (target_cpu == "loong64") {
+    cflags += [
+      "-mno-lsx",  # Disable 128 bit vector.
+    ]
+  }
 }
 
 if (libyuv_use_neon) {
