--- a/src/3rdparty/chromium/third_party/libvpx/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/libvpx/BUILD.gn
@@ -47,6 +47,8 @@ if (current_cpu == "x86") {
   }
 } else if (current_cpu == "riscv64") {
   cpu_arch_full = "generic"
+} else if (current_cpu == "ppc64") {
+  cpu_arch_full = "generic"
 } else if (current_cpu == "loong64") {
   cpu_arch_full = "loongarch"
 } else {
@@ -516,10 +518,6 @@ source_set("libvpx_loongarch_headers") {
   sources = libvpx_srcs_loongarch_headers
 }
 
-source_set("libvpx_ppc64_headers") {
-  sources = libvpx_srcs_ppc64_headers
-}
-
 source_set("libvpx_nacl_headers") {
   sources = libvpx_srcs_nacl_headers
 }
@@ -574,8 +572,8 @@ static_library("bundled_libvpx") {
       public_deps = [ ":libvpx_arm64_headers" ]
     }
   } else if (current_cpu == "ppc64") {
-    sources = libvpx_srcs_ppc64
-    public_deps = [ ":libvpx_ppc64_headers" ]
+    sources = libvpx_srcs_generic
+    public_deps = [ ":libvpx_generic_headers" ]
   } else if (current_cpu == "riscv64") {
     sources = libvpx_srcs_generic
     public_deps = [ ":libvpx_generic_headers" ]
