diff --git a/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.mjs b/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.mjs
index b85411dc71..ac54145c3b 100644
--- a/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.mjs
+++ b/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.mjs
@@ -19,7 +19,7 @@ export default commandLineArgs => ({
     sourcemap: Boolean(commandLineArgs.configSourcemaps),
   }],
   plugins: [
-    terser(),
+    // terser(),
     {
       name: 'devtools-plugin',
       resolveId(source, importer) {
