diff --git a/cmake/modules/FindClang.cmake b/cmake/modules/FindClang.cmake
index 34a8c38198..9ebb9c4a58 100644
--- a/cmake/modules/FindClang.cmake
+++ b/cmake/modules/FindClang.cmake
@@ -106,9 +106,9 @@ if(CLANG_FOUND)
             PATHS "${CLANG_LIBRARY_DIRS}"
                   "${CLANG_INCLUDE_DIRS}"
             PATH_SUFFIXES "clang/${CLANG_VERSION}/include"
-                          "../../../clang/${CLANG_VERSION}/include"
+                          "../../clang/${CLANG_VERSION}/include"
                           "clang/${CLANG_VERSION_CLEAN}/include"
-                          "../../../clang/${CLANG_VERSION_CLEAN}/include"
+                          "../../clang/${CLANG_VERSION_CLEAN}/include"
             NO_DEFAULT_PATH
   )
 
