--- a/repology.d
+++ b/repology.d
@@ -53,16 +53,7 @@
     string uri = "https://repology.org/api/v1/project";
     string[string] queryParts = null;
 
-    version (OpenBSD) options.repo = "openbsd";
-    else version (FreeBSD) options.repo = "freebsd";
-    else version (NetBSD) options.repo = "pkgsrc_current";
-    else version (OSX) options.repo = "homebrew";
-    else version (Windows) options.repo = "chocolatey";
-    else
-    {
-        stderr.writeln("repology: specify your repo with the --repo flag");
-        return 1;
-    }
+    options.repo = "alpine_edge";
 
     auto opts = getopt(
         args,
