diff -ru fwupd-1.2.9.orig/src/fu-common.c fwupd-1.2.9/src/fu-common.c
--- fwupd-1.2.9.orig/src/fu-common.c	2019-05-20 13:17:42.000000000 +0300
+++ fwupd-1.2.9/src/fu-common.c	2019-06-09 13:48:00.861388043 +0300
@@ -460,7 +460,6 @@
 	fu_common_add_argv (argv, "--die-with-parent");
 	fu_common_add_argv (argv, "--ro-bind /usr /usr");
 	fu_common_add_argv (argv, "--ro-bind /lib /lib");
-	fu_common_add_argv (argv, "--ro-bind /lib64 /lib64");
 	fu_common_add_argv (argv, "--ro-bind /bin /bin");
 	fu_common_add_argv (argv, "--ro-bind /sbin /sbin");
 	fu_common_add_argv (argv, "--dir /tmp");
diff -ru fwupd-1.2.9.orig/data/tests/builder/startup.sh fwupd-1.2.9/data/tests/builder/startup.sh
--- fwupd-1.2.9.orig/data/tests/builder/startup.sh	2019-05-20 13:17:42.000000000 +0300
+++ fwupd-1.2.9/data/tests/builder/startup.sh	2019-06-07 21:43:43.543150907 +0300
@@ -1,2 +1,2 @@
-#/bin/sh
+#!/bin/sh
 cat source.bin | rev > firmware.bin
diff -ru fwupd-1.2.9.orig/src/fu-engine.c fwupd-1.2.9/src/fu-engine.c
--- fwupd-1.2.9.orig/src/fu-engine.c	2019-05-20 13:17:42.000000000 +0300
+++ fwupd-1.2.9/src/fu-engine.c	2019-06-09 16:26:11.573561932 +0300
@@ -4517,7 +4517,6 @@
 	self->usb_ctx = g_usb_context_new (error);
 	if (self->usb_ctx == NULL) {
 		g_prefix_error (error, "Failed to get USB context: ");
-		return FALSE;
 	}
 
 	/* delete old data files */
@@ -4562,13 +4561,15 @@
 	fu_engine_plugins_coldplug (self, FALSE);
 
 	/* coldplug USB devices */
-	g_signal_connect (self->usb_ctx, "device-added",
-			  G_CALLBACK (fu_engine_usb_device_added_cb),
-			  self);
-	g_signal_connect (self->usb_ctx, "device-removed",
-			  G_CALLBACK (fu_engine_usb_device_removed_cb),
-			  self);
-	g_usb_context_enumerate (self->usb_ctx);
+	if (self->usb_ctx != NULL) {
+		g_signal_connect (self->usb_ctx, "device-added",
+				  G_CALLBACK (fu_engine_usb_device_added_cb),
+				  self);
+		g_signal_connect (self->usb_ctx, "device-removed",
+				  G_CALLBACK (fu_engine_usb_device_removed_cb),
+				  self);
+		g_usb_context_enumerate (self->usb_ctx);
+	}
 
 	/* coldplug udev devices */
 	fu_engine_enumerate_udev (self);
