diff -ru fwupd-1.2.9.orig/src/fu-common.c fwupd-1.2.9/src/fu-common.c
--- fwupd-1.2.9.orig/src/fu-common.c	2019-05-20 13:17:42.000000000 +0300
+++ fwupd-1.2.9/src/fu-common.c	2019-06-09 13:48:00.861388043 +0300
@@ -460,7 +460,6 @@
 	fu_common_add_argv (argv, "--die-with-parent");
 	fu_common_add_argv (argv, "--ro-bind /usr /usr");
 	fu_common_add_argv (argv, "--ro-bind /lib /lib");
-	fu_common_add_argv (argv, "--ro-bind /lib64 /lib64");
 	fu_common_add_argv (argv, "--ro-bind /bin /bin");
 	fu_common_add_argv (argv, "--ro-bind /sbin /sbin");
 	fu_common_add_argv (argv, "--dir /tmp");
diff -ru fwupd-1.2.9.orig/data/tests/builder/startup.sh fwupd-1.2.9/data/tests/builder/startup.sh
--- fwupd-1.2.9.orig/data/tests/builder/startup.sh	2019-05-20 13:17:42.000000000 +0300
+++ fwupd-1.2.9/data/tests/builder/startup.sh	2019-06-07 21:43:43.543150907 +0300
@@ -1,2 +1,2 @@
-#/bin/sh
+#!/bin/sh
 cat source.bin | rev > firmware.bin
