diff --git a/Cargo.lock b/Cargo.lock
index e489448..19abe15 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -3029,4 +3029,5 @@ checksum = "9fd07cbbc53846d9145dbffdf6dd09a7a0aa52be46741825f5c97bdd4f73f12b"
 dependencies = [
  "cc",
  "libc",
+ "pkg-config",
 ]
diff --git a/Cargo.toml b/Cargo.toml
index 09774ae..00b8555 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -65,7 +65,7 @@ toml = "0.5"
 url = "2.1"
 wait-timeout = "0.2"
 xz2 = "0.1.3"
-zstd = "0.11"
+zstd = { version = "0.11", features = ["pkg-config"] }
 
 [dependencies.retry]
 default-features = false
