--- a/tests/class_property/test_class_property.py
+++ b/tests/class_property/test_class_property.py
@@ -44,7 +44,7 @@
 
         with self.subTest("Docstring"):
             self.assertEqual(
-                "Object identifier" if has_docstring else None,
+                "Object identifier" if has_docstring else "The type of the None singleton.",
                 cls.__dict__.get("identifier").__doc__,
             )
 
@@ -140,7 +140,7 @@
                         fget=get_identifier if has_getter else None,
                         fset=set_identifier if has_setter else None,
                         fdel=del_identifier if has_deleter else None,
-                        doc="Object identifier" if has_docstring else None,
+                        doc="Object identifier" if has_docstring else "The type of the None singleton.",
                     )
 
                 self.check_class(
--- a/tests/class_property/test_static_property.py
+++ b/tests/class_property/test_static_property.py
@@ -44,7 +44,7 @@
 
         with self.subTest("Docstring"):
             self.assertEqual(
-                "Object identifier" if has_docstring else None,
+                "Object identifier" if has_docstring else "The type of the None singleton.",
                 cls.__dict__.get("var").__doc__,
             )
 
@@ -137,7 +137,7 @@
                         fget=get_var if has_getter else None,
                         fset=set_var if has_setter else None,
                         fdel=del_var if has_deleter else None,
-                        doc="Object identifier" if has_docstring else None,
+                        doc="Object identifier" if has_docstring else "The type of the None singleton.",
                     )
 
                 self.check_class(
--- a/tests/test_property.py
+++ b/tests/test_property.py
@@ -48,7 +48,7 @@
 
         with self.subTest("Docstring"):
             self.assertEqual(
-                "1 based index" if has_docstring else None, cls.__dict__["i1"].__doc__
+                "1 based index" if has_docstring else "The type of the None singleton.", cls.__dict__["i1"].__doc__
             )
 
     def test_property_basic(self):
@@ -129,7 +129,7 @@
                         fget=get_i1 if has_getter else None,
                         fset=set_i1 if has_setter else None,
                         fdel=del_i1 if has_deleter else None,
-                        doc="1 based index" if has_docstring else None,
+                        doc="1 based index" if has_docstring else "The type of the None singleton.",
                     )
 
                 self.check_class(
