--- a/Makefile.orig	2009-05-21 08:22:51.000000000 +0000
+++ b/Makefile	2009-05-21 08:23:06.000000000 +0000
@@ -568,7 +568,7 @@
 	lib/string_list-c.cpp lib/find_speller.cpp lib/speller-c.cpp \
 	lib/string_pair_enumeration-c.cpp lib/new_checker.cpp \
 	modules/filter/url.cpp $(am__append_3)
-libaspell_la_LIBADD = $(LTLIBINTL) $(PTHREAD_LIB)
+libaspell_la_LIBADD = $(LTLIBINTL) $(PTHREAD_LIB) -lm
 libaspell_la_LDFLAGS = -version-info 16:4:1 -no-undefined
 #libaspell_la_LDFLAGS = -version-info 16:4:0 -no-undefined
 libpspell_la_SOURCES = lib/dummy.cpp
