diff --git a/src/Mercury.options b/src/Mercury.options
index 947dea7..539a0fc 100644
--- a/src/Mercury.options
+++ b/src/Mercury.options
@@ -9,7 +9,7 @@ CFLAGS += -D_FILE_OFFSET_BITS=64
 MLLIBS-bower += -lgpgme
 
 # Statically link to Mercury libraries.
-MCFLAGS-bower += --mercury-linkage static
+MCFLAGS-bower += --mercury-linkage shared
 
 # General optimisation flags.
 MCFLAGS += --optimise-constructor-last-call
