diff --git a/flask_limiter/_typing.py.orig b/flask_limiter/_typing.py
index 965427f..9a84e26 100644
--- a/flask_limiter/_typing.py.orig
+++ b/flask_limiter/_typing.py
@@ -7,7 +7,7 @@ from typing import (
     cast,
 )
 
-from typing_extensions import Self
+from typing import Self
 
 R = TypeVar("R")
 P = ParamSpec("P")
diff --git a/flask_limiter/commands.py.orig b/flask_limiter/commands.py
index 2c0dc9e..55f95bc 100644
--- a/flask_limiter/commands.py.orig
+++ b/flask_limiter/commands.py
@@ -30,7 +30,7 @@ try:
     CLI_DEPS_AVAILABLE = True
 except ImportError:  # noqa
     CLI_DEPS_AVAILABLE = False
-from typing_extensions import TypedDict
+from typing import TypedDict
 from werkzeug.routing import Rule
 
 from ._extension import Limiter
