diff --git a/test/python/gexiv2.py b/test/python/gexiv2.py
index fa80f9f..4f2506a 100644
--- a/test/python/gexiv2.py
+++ b/test/python/gexiv2.py
@@ -48,9 +48,9 @@ class TestGexiv2(unittest.TestCase):
         md.open_path(self.get_sample_path(sample))
 
         (lo, la, alt) = md.get_gps_info()
-        self.assertEqual(lo, -1.508425)
-        self.assertEqual(la, 48.631806166666664)
-        self.assertEqual(alt, -0.926000)
+        self.assertAlmostEqual(lo, -1.508425)
+        self.assertAlmostEqual(la, 48.631806166666664)
+        self.assertAlmostEqual(alt, -0.926000)
 
     def test_xmp_packet_formatting(self):
         sample = 'CaorVN.jpeg'
