--- a/Makefile
+++ b/Makefile
@@ -15,7 +15,7 @@
 install: all
 	$(INSTALL) -d $(SHARE_DIR)
 	$(INSTALL) -d $(DOC_DIR)
-	cp .version zsh-syntax-highlighting.zsh $(SHARE_DIR)
+	cp .version zsh-syntax-highlighting.zsh zsh-syntax-highlighting.plugin.zsh $(SHARE_DIR)
 	cp COPYING.md README.md changelog.md $(DOC_DIR)
 	sed -e '1s/ .*//' -e '/^\[build-status-[a-z]*\]: /d' < README.md > $(DOC_DIR)/README.md
 	cp .revision-hash $(SHARE_DIR)
