Author: Luke Yelavich <themuso@ubuntu.com>
Description: Support for checking virtio devices for dmraid metadata.
--- dmraid/lib/device/scan.c
+++ dmraid/lib/device/scan.c
@@ -174,7 +174,7 @@ interested(struct lib_context *lc, char 
 	 * Whole IDE and SCSI disks only.
 	 */
 	return (!isdigit(name[strlen(name) - 1]) &&
-		(*(name + 1) == 'd' && (*name == 'h' || *name == 's')))
+		(*(name + 1) == 'd' && (*name == 'h' || *name == 's' || *name == 'v')))
 #ifdef	DMRAID_TEST
 		/*
 		 * Include dm devices for testing.
