diff --git a/lib/activate/activate.c b/lib/activate/activate.c
index 7bd5ab5..2c1a7aa 100644
--- a/lib/activate/activate.c
+++ b/lib/activate/activate.c
@@ -861,7 +861,7 @@ dm_unregister_for_event(char *dev_name, char *lib_name)
 
 #define LIB_NAME_LENGTH 255
 static int
-do_device(struct lib_context *lc, struct raid_set *rs, int (*f) ())
+do_device(struct lib_context *lc, struct raid_set *rs, int (*f)(char *, char *))
 {
 	int ret = 0;
 	char lib_name[LIB_NAME_LENGTH];
diff --git a/lib/misc/file.c b/lib/misc/file.c
index 7c663cb..2e0a6d7 100644
--- a/lib/misc/file.c
+++ b/lib/misc/file.c
@@ -58,11 +58,11 @@ rw_file(struct lib_context *lc, const char *who, int flags,
 	int fd, ret = 0;
 	loff_t o;
 	struct {
-		ssize_t(*func) ();
+		ssize_t(*func)(int, void *, size_t);
 		const char *what;
 	} rw_spec[] = {
 		{ read, "read"},
-		{ write, "writ"},
+		{ (ssize_t(*)(int, void *, size_t))write, "writ"},
 	}, *rw = rw_spec + ((flags & O_WRONLY) ? 1 : 0);
 
 	if ((fd = open(path, flags, lc->mode)) == -1)
