diff --git a/Makefile b/Makefile
index 54b71ee..ce463c1 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@ MANS=sponge.1 vidir.1 vipe.1 isutf8.1 ts.1 combine.1 ifdata.1 ifne.1 pee.1 zrun.
 CFLAGS?=-O2 -g -Wall
 INSTALL_BIN?=install -s
 PREFIX?=/usr
-DOCBOOKXSL?=/usr/share/xml/docbook/stylesheet/docbook-xsl
+DOCBOOKXSL?=/usr/share/xml/docbook/xsl-stylesheets-1.79.1
 
 DOCBOOK2XMAN=xsltproc --param man.authors.section.enabled 0 $(DOCBOOKXSL)/manpages/docbook.xsl
 
