--- a/Makefile	2017-02-19 20:44:09.469193976 +0100
+++ b/Makefile	2017-02-19 20:44:32.302484346 +0100
@@ -8,7 +8,7 @@
 ifneq (,$(findstring CYGWIN,$(shell uname)))
 	DOCBOOKXSL?=/usr/share/sgml/docbook/xsl-stylesheets
 else
-	DOCBOOKXSL?=/usr/share/xml/docbook/stylesheet/docbook-xsl
+	DOCBOOKXSL?=/usr/share/xml/docbook/xsl-stylesheets-1.79.1
 endif
 
 DOCBOOK2XMAN=xsltproc --param man.authors.section.enabled 0 $(DOCBOOKXSL)/manpages/docbook.xsl
