From 4cbba11d65910f6daa8c739c8942baf22ec5ee91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Thomas=20B=C3=B6hler?= <witcher@wiredspace.de>
Date: Sat, 26 Apr 2025 14:02:43 +0200
Subject: [PATCH 1/3] don't statically link libz-sys
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Thomas Böhler <witcher@wiredspace.de>
---
 meli/Cargo.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meli/Cargo.toml b/meli/Cargo.toml
index 6f086bd2..00fbf41f 100644
--- a/meli/Cargo.toml
+++ b/meli/Cargo.toml
@@ -32,7 +32,7 @@ futures = { version = "0.3.30", default-features = false, features = ["async-awa
 indexmap = { version = "^2.3", default-features = false, features = ["serde", "std"] }
 itoa = { version = "1.0.11", default-features = false }
 libc = { version = "0.2.125", default-features = false, features = ["extra_traits"] }
-libz-sys = { version = "1.1", features = ["static"], optional = true }
+libz-sys = { version = "1.1", optional = true }
 linkify = { version = "^0.10", default-features = false }
 melib = { path = "../melib", version = "0.8.12", features = [] }
 nix = { version = "0.29", default-features = false, features = ["signal", "poll", "term", "ioctl", "process"] }
-- 
2.49.0

