From 4b85a3444812c1a2fb0b2bcd78a80d09746000d7 Mon Sep 17 00:00:00 2001
From: Martin Kletzander <mkletzan@redhat.com>
Date: Tue, 25 Mar 2014 11:49:24 +0100
Subject: [PATCH] generator: Add virConnectDomainQemuMonitorEventCallback to
 skipped_types

Signed-off-by: Martin Kletzander <mkletzan@redhat.com>
---
 generator.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/generator.py b/generator.py
index 0e9600f..e409921 100755
--- a/generator.py
+++ b/generator.py
@@ -273,6 +273,7 @@ skipped_types = {
      'virConnectDomainEventWatchdogCallback': "No function types in python",
      'virConnectDomainEventIOErrorCallback': "No function types in python",
      'virConnectDomainEventGraphicsCallback': "No function types in python",
+     'virConnectDomainQemuMonitorEventCallback': "No function types in python",
      'virStreamEventCallback': "No function types in python",
      'virEventHandleCallback': "No function types in python",
      'virEventTimeoutCallback': "No function types in python",
-- 
1.9.1

