--- a/v/1.12.1/app/libraries/RainLoop/Actions.php
+++ b/v/1.12.1/app/libraries/RainLoop/Actions.php
@@ -297,7 +297,7 @@
 					if ('sqlite' === $sDsnType)
 					{
 						$mResult = new \RainLoop\Providers\AddressBook\PdoAddressBook(
-							'sqlite:'.APP_PRIVATE_DATA.'AddressBook.sqlite', '', '', 'sqlite');
+							'sqlite:/etc/rainloop/AddressBook.sqlite', '', '', 'sqlite');
 					}
 					else
 					{
