diff --git a/v/1.13.0/app/libraries/RainLoop/Actions.php b/v/1.13.0/app/libraries/RainLoop/Actions.php
index 1e1263e..fc3932f 100644
--- a/v/1.13.0/app/libraries/RainLoop/Actions.php
+++ b/v/1.13.0/app/libraries/RainLoop/Actions.php
@@ -297,7 +297,7 @@ class Actions
 					if ('sqlite' === $sDsnType)
 					{
 						$mResult = new \RainLoop\Providers\AddressBook\PdoAddressBook(
-							'sqlite:'.APP_PRIVATE_DATA.'AddressBook.sqlite', '', '', 'sqlite');
+							'sqlite:/etc/rainloop/AddressBook.sqlite', '', '', 'sqlite');
 					}
 					else
 					{
