diff --git a/Makefile b/Makefile
index 1cd84fc..0b241bd 100644
--- a/Makefile
+++ b/Makefile
@@ -55,7 +55,7 @@ MANIFEST_NAME := manifest.txt
 MANIFEST := ${TMP_DIR}/${MANIFEST_NAME}
 
 build:
-	cargo build --release
+	cargo auditable build --frozen --release
 
 ${TMP_DIR}:
 	mkdir -p "${@}"
@@ -146,7 +146,7 @@ docs/publish: docs/setup docs/build
 
 std-docs/build:
 	rm -rf std/build
-	cargo build
+	cargo auditable build --frozen --release
 	cd std && idoc --compiler ../target/debug/inko
 
 std-docs/publish: std-docs/build
