diff --git a/compiler-rt/lib/scudo/standalone/CMakeLists.txt b/compiler-rt/lib/scudo/standalone/CMakeLists.txt
index d75b7fd..bcb63ff 100644
--- a/compiler-rt/lib/scudo/standalone/CMakeLists.txt
+++ b/compiler-rt/lib/scudo/standalone/CMakeLists.txt
@@ -27,7 +27,7 @@ append_list_if(COMPILER_RT_HAS_FNO_LTO_FLAG -fno-lto SCUDO_CFLAGS)
 if(COMPILER_RT_DEBUG)
   list(APPEND SCUDO_CFLAGS -O0 -DSCUDO_DEBUG=1)
 else()
-  list(APPEND SCUDO_CFLAGS -O3)
+  list(APPEND SCUDO_CFLAGS -O3 -DSCUDO_CAN_USE_PRIMARY64=0)
 endif()
 
 set(SCUDO_LINK_FLAGS)
