Author: Rejah Rehim <rejah@beaglesecurity.com>
Summary: Fixes dependency compilation error

--- a/setup.py
+++ b/setup.py
@@ -17,20 +17,6 @@
 from pyvows import version
 
 
-_test_requires = [
-    'argparse>=1.4.0',
-    'colorama>=0.3.7',
-    'coverage>=4.1.1'
-
-]
-_install_requires = [
-    'gevent>=1.2.2',
-    'preggy>=1.3.0',
-]
-if sys.version_info < (2, 7):
-    _install_requires.append('argparse >= 1.1')
-
-
 setup(
     ### OVERVIEW
     name='pyVows',
@@ -52,10 +38,6 @@
 
     ### TECHNICAL INFO
     version=version.to_str(),
-    install_requires=_install_requires,
-    extras_require={
-        'tests': _test_requires,
-    },
     packages=find_packages(exclude=['tests', 'tests.*']),
     package_dir={'pyvows': 'pyvows'},
     entry_points={
