diff --git a/pytest.ini b/pytest.ini
index 4a00d9a..38e1959 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -8,3 +8,4 @@ filterwarnings =
     # python3 raises this when importing setuptools
     ignore:the imp module is deprecated in favour of importlib.*:PendingDeprecationWarning
     ignore:the imp module is deprecated in favour of importlib.*:DeprecationWarning
+    ignore:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning
