diff --git a/src/location_service/com/lomiri/location/satellite_based_positioning_state.cpp b/src/location_service/com/lomiri/location/satellite_based_positioning_state.cpp
index 022b595..ceadd6a 100644
--- a/src/location_service/com/lomiri/location/satellite_based_positioning_state.cpp
+++ b/src/location_service/com/lomiri/location/satellite_based_positioning_state.cpp
@@ -20,6 +20,7 @@
 
 #include <iostream>
 #include <unordered_map>
+#include <cstdint>
 
 namespace location = com::lomiri::location;
 
diff --git a/src/location_service/com/lomiri/location/wifi_and_cell_reporting_state.cpp b/src/location_service/com/lomiri/location/wifi_and_cell_reporting_state.cpp
index 4ef18a2..f22b0f6 100644
--- a/src/location_service/com/lomiri/location/wifi_and_cell_reporting_state.cpp
+++ b/src/location_service/com/lomiri/location/wifi_and_cell_reporting_state.cpp
@@ -20,6 +20,7 @@
 
 #include <iostream>
 #include <unordered_map>
+#include <cstdint>
 
 namespace location = com::lomiri::location;
 
