--- a/linux/CMakeLists.txt
+++ b/linux/CMakeLists.txt
@@ -123,9 +123,6 @@
     COMPONENT Runtime)
 endif()
 
-# Copy the Helper
-install(DIRECTORY "../build/linux/helper" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
-
 # Copy Linux support files
 install(DIRECTORY "../resources/linux/linux_support" DESTINATION "${BUILD_BUNDLE_DIR}" USE_SOURCE_PERMISSIONS)
 install(FILES "../resources/linux/README.adoc" DESTINATION "${BUILD_BUNDLE_DIR}")
