diff --git a/lib/desktop/init.dart b/lib/desktop/init.dart
index 43a2ddfe..6a978f31 100755
--- a/lib/desktop/init.dart
+++ b/lib/desktop/init.dart
@@ -203,6 +203,7 @@ Future<Widget> initialize(List<String> argv) async {
     Uri.file(Platform.resolvedExecutable).resolve('features.json').toFilePath(),
   );
 
+  exe = 'authenticator-helper';
   final rpcFuture = _initHelper(exe!);
   _initLicenses();
 
