diff -urbB cups-filters-1.0.29/filter/pdftoopvp/oprs/OPRS.cxx cups-filters-1.0-20130119/filter/pdftoopvp/oprs/OPRS.cxx
--- cups-filters-1.0.29/filter/pdftoopvp/oprs/OPRS.cxx	2012-09-10 14:52:00.000000000 -0500
+++ cups-filters-1.0-20130119/filter/pdftoopvp/oprs/OPRS.cxx	2013-01-18 18:40:36.000000000 -0600
@@ -5,6 +5,9 @@
 //========================================================================
 
 #include <config.h>
+#ifdef HAVE_CPP_POPPLER_VERSION_H
+#include "cpp/poppler-version.h"
+#endif
 
 #ifdef USE_GCC_PRAGMAS
 #pragma implementation
@@ -235,7 +238,11 @@
 			      SplashColorMode srcMode, GBool srcAlpha,
 			      int w, int h, SplashCoord *mat) {
     if (rasterMode) {
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
 	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
+#else
+	return splash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat,gFalse);
+#endif
     } else {
 	return opvpSplash->drawImage(src,srcData,srcMode,srcAlpha,w,h,mat);
     }
diff -urbB cups-filters-1.0.29/filter/pdftoopvp/OPVPOutputDev.cxx cups-filters-1.0-20130119/filter/pdftoopvp/OPVPOutputDev.cxx
--- cups-filters-1.0.29/filter/pdftoopvp/OPVPOutputDev.cxx	2012-07-20 03:22:23.000000000 -0500
+++ cups-filters-1.0-20130119/filter/pdftoopvp/OPVPOutputDev.cxx	2013-01-18 18:40:36.000000000 -0600
@@ -1804,8 +1804,14 @@
   maskSplash = new Splash(maskBitmap, gFalse);
   maskColor[0] = 0;
   maskSplash->clear(maskColor);
+#if POPPLER_VERSION_MAJOR <= 0 && (POPPLER_VERSION_MINOR <= 20 || (POPPLER_VERSION_MINOR == 21 && POPPLER_VERSION_MICRO <= 2))
   maskSplash->drawImage(&imageSrc, &imgMaskData,
 			splashModeMono8, gFalse, maskWidth, maskHeight, mat);
+#else
+  maskSplash->drawImage(&imageSrc, &imgMaskData,
+			splashModeMono8, gFalse, maskWidth, maskHeight,
+                        mat,gFalse);
+#endif
   delete imgMaskData.imgStr;
   maskStr->close();
   gfree(imgMaskData.lookup);
diff -urbB cups-filters-1.0.29/NEWS cups-filters-1.0-20130119/NEWS
--- cups-filters-1.0.29/NEWS	2013-01-02 12:23:13.000000000 -0600
+++ cups-filters-1.0-20130119/NEWS	2013-01-18 18:40:36.000000000 -0600
@@ -1,6 +1,11 @@
 NEWS - OpenPrinting CUPS Filters v1.0.29 - 2013-01-02
 -----------------------------------------------------
 
+CHANGES IN V1.0.30
+
+	- pdftoopvp: Let it build with Poppler 0.22.x. Thanks to Koji Otani
+	  from BBR Inc. (Bug #1089).
+
 CHANGES IN V1.0.29
 
 	- Fixed ./configure option "--with-rcdir=no". Thanks to Jiri
