From 8c782e54ba244981141d91e7d44922952563677c Mon Sep 17 00:00:00 2001
From: Ghabry <gabriel+github@mastergk.de>
Date: Thu, 4 May 2023 16:30:41 +0200
Subject: [PATCH] Require C++17

---
 CMakeLists.txt | 4 ++--
 Makefile.am    | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f4fbbd07..afe71465 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,8 +20,8 @@ set(CMAKE_DEBUG_POSTFIX "d" CACHE STRING "Override CMAKE_DEBUG_POSTFIX.")
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/builds/cmake/Modules)
 include(ConfigureWindows)
 
-# C++14 is required
-set(CMAKE_CXX_STANDARD 14)
+# C++17 is required
+set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS ON)
 
diff --git a/Makefile.am b/Makefile.am
index 33662a74..85033b16 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -31,7 +31,7 @@ liblcf_la_CPPFLAGS = \
 	-I$(srcdir)/src \
 	-I$(srcdir)/src/generated
 liblcf_la_CXXFLAGS = \
-	-std=gnu++14 \
+	-std=gnu++17 \
 	-fno-math-errno \
 	$(AM_CXXFLAGS) \
 	$(EXPAT_CFLAGS) \
@@ -341,7 +341,7 @@ test_runner_CPPFLAGS = \
 	-I$(srcdir)/src \
 	-I$(srcdir)/src/generated
 test_runner_CXXFLAGS = \
-	-std=gnu++14 \
+	-std=gnu++17 \
 	$(EXPAT_CXXFLAGS) \
 	$(ICU_CXXFLAGS)
 test_runner_LDADD = \
