--- a/rake-compiler.gemspec
+++ b/rake-compiler.gemspec
@@ -22,14 +22,8 @@
   s.add_development_dependency 'cucumber', '~> 1.1.4'
 
   # components, files and paths
-  s.files = Dir.glob("features/**/*.{feature,rb}")
   s.files += ["bin/rake-compiler"]
   s.files += Dir.glob("lib/**/*.rb")
-  s.files += ["spec/spec.opts"]
-  s.files += Dir.glob("spec/**/*.rb")
-  s.files += Dir.glob("tasks/**/*.rake")
-  s.files += ["Rakefile", "Gemfile"]
-  s.files += Dir.glob("*.{rdoc,txt,yml}")
 
   s.bindir      = 'bin'
   s.executables = ['rake-compiler']
@@ -38,8 +32,6 @@
 
   # documentation
   s.rdoc_options << '--main'  << 'README.rdoc' << '--title' << 'rake-compiler -- Documentation'
-
-  s.extra_rdoc_files = %w(README.rdoc LICENSE.txt History.txt)
 
   # project information
   s.homepage          = 'https://github.com/rake-compiler/rake-compiler'
