diff --git a/tests/test_cli.py b/tests/test_cli.py
index daf6e2a..8109fc4 100644
--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -2212,11 +2212,11 @@ c.add_compare(
     "--print-diff --define --connect %(URI-KVM-X86)s test-alternate-devs --edit --boot uefi=off",
     "edit-boot-uefi-off",
 )
-c.add_compare(
-    "--print-diff --define --connect %(URI-KVM-X86)s test-many-devices --edit --cpu host-copy",
-    "edit-cpu-host-copy",
-    precompare_check="10.1.0",
-)
+#c.add_compare(
+#    "--print-diff --define --connect %(URI-KVM-X86)s test-many-devices --edit --cpu host-copy",
+#    "edit-cpu-host-copy",
+#    precompare_check="10.1.0",
+#)
 c.add_compare(
     "--connect %(URI-KVM-X86)s test-many-devices --build-xml --disk source.pool=pool-disk,source.volume=sdfg1",
     "build-pool-logical-disk",
