diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9ef0cf9..1ac7131 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -180,7 +180,7 @@ endif()
 if(CURSES_LIBRARY)
   target_link_libraries(sipp dl ${CURSES_LIBRARY} pthread)
   target_link_libraries(sipp_unittest dl ${CURSES_LIBRARY} pthread)
-  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+  if(false)
     if(TINFO_LIBRARY)
       target_link_libraries(sipp ${TINFO_LIBRARY})
       target_link_libraries(sipp_unittest ${TINFO_LIBRARY})
