diff --git a/src/OVAL/probes/unix/linux/rpminfo_probe.c b/src/OVAL/probes/unix/linux/rpminfo_probe.c
index 18aaabd..a26f8f3 100644
--- a/src/OVAL/probes/unix/linux/rpminfo_probe.c
+++ b/src/OVAL/probes/unix/linux/rpminfo_probe.c
@@ -60,6 +60,7 @@
 #include <regex.h>
 
 /* RPM headers */
+#include <rpm/rpmcrypto.h>
 #include "rpm-helper.h"
 
 /* SEAP */
diff --git a/src/OVAL/probes/unix/linux/rpmverify_probe.c b/src/OVAL/probes/unix/linux/rpmverify_probe.c
index 1d9f2ac..372fce4 100644
--- a/src/OVAL/probes/unix/linux/rpmverify_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverify_probe.c
@@ -44,6 +44,7 @@
 #include "rpm-helper.h"
 
 /* Individual RPM headers */
+#include <rpm/rpmcrypto.h>
 #include <rpm/rpmfi.h>
 #include <rpm/rpmcli.h>
 
diff --git a/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c b/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c
index d93e805..9d882f5 100644
--- a/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverifyfile_probe.c
@@ -47,6 +47,7 @@
 #include "oscap_helpers.h"
 
 /* Individual RPM headers */
+#include <rpm/rpmcrypto.h>
 #include <rpm/rpmfi.h>
 #include <rpm/rpmcli.h>
 
diff --git a/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c b/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c
index 8bce0a8..60457b0 100644
--- a/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c
+++ b/src/OVAL/probes/unix/linux/rpmverifypackage_probe.c
@@ -47,6 +47,7 @@
 #include "probe-chroot.h"
 
 /* Individual RPM headers */
+#include <rpm/rpmcrypto.h>
 #include <rpm/rpmfi.h>
 #include <rpm/rpmcli.h>
 #include <popt.h>
