diff --git a/src/apm.h b/src/apm.h
index bc66714..86c7663 100644
--- a/src/apm.h
+++ b/src/apm.h
@@ -109,7 +109,7 @@ typedef struct {
     char		*MemMap;
     pointer		BltMap;
     Bool		UnlockCalled;
-    IOADDRESS		iobase, xport, xbase;
+    unsigned long	iobase, xport, xbase;
     unsigned char	savedSR10;
     CARD8		MiscOut;
     CARD8		c9, d9, db, Rush;
diff --git a/src/apm_driver.c b/src/apm_driver.c
index f172bab..f77bb2b 100644
--- a/src/apm_driver.c
+++ b/src/apm_driver.c
@@ -418,8 +418,14 @@ ApmPreInit(ScrnInfoPtr pScrn, int flags)
 
     hwp = VGAHWPTR(pScrn);
     vgaHWGetIOBase(hwp);
-    pApm->iobase = hwp->PIOOffset;
-    pApm->xport = hwp->PIOOffset + 0x3C4;
+#if ABI_VIDEODRV_VERSION < 12
+#define PIOOFFSET hwp->PIOOffset
+#else
+/* FIXME reintroduce domain support */
+#define PIOOFFSET 0
+#endif
+    pApm->iobase = PIOOFFSET;
+    pApm->xport = PIOOFFSET + 0x3C4;
 
     /* Set pScrn->monitor */
     pScrn->monitor = pScrn->confScreen->monitor;
