diff --git a/tests/check/meson.build b/tests/check/meson.build
index b44f1a0..2c91214 100644
--- a/tests/check/meson.build
+++ b/tests/check/meson.build
@@ -40,7 +40,6 @@ good_tests = [
   [ 'elements/deinterlace', get_option('deinterlace').disabled()],
   [ 'elements/dtmf', get_option('dtmf').disabled()],
   [ 'elements/flvdemux', get_option('flv').disabled()],
-  [ 'elements/flvmux', get_option('flv').disabled()],
   [ 'elements/mulawdec', get_option('law').disabled()],
   [ 'elements/mulawenc', get_option('law').disabled()],
   [ 'elements/icydemux', get_option('icydemux').disabled()],
