--- a/test/offline/utf82locale.c
+++ b/test/offline/utf82locale.c
@@ -15,9 +15,11 @@
         old_locale = setlocale(LC_ALL, "cs_CZ.UTF-8");
         if (old_locale != NULL) return 0;
     } else if ('2' == type) {
+        return -1;  // XXX-Patched
         old_locale = setlocale(LC_ALL, "cs_CZ.ISO8859-2");
         if (old_locale != NULL) return 0;
     } else {
+        return -1;  // XXX-Patched
         old_locale = setlocale(LC_ALL, "C");
         if (old_locale != NULL) return 0;
     }
