--- a/scripts/hak/target.ts
+++ b/scripts/hak/target.ts
@@ -28,8 +28,10 @@
     'i686-unknown-linux-musl' |
     'i686-unknown-linux-gnu' |
     'x86_64-unknown-linux-musl' |
+    'x86_64-alpine-linux-musl' |
     'x86_64-unknown-linux-gnu' |
     'aarch64-unknown-linux-musl' |
+    'aarch64-alpine-linux-musl' |
     'aarch64-unknown-linux-gnu' |
     'powerpc64le-unknown-linux-musl' |
     'powerpc64le-unknown-linux-gnu';
@@ -118,6 +120,13 @@
     libC: 'musl',
 };
 
+const x8664AlpineLinuxMusl: LinuxTarget = {
+    id: 'x86_64-alpine-linux-musl',
+    platform: 'linux',
+    arch: 'x64',
+    libC: 'musl',
+};
+
 const i686UnknownLinuxGnu: LinuxTarget = {
     id: 'i686-unknown-linux-gnu',
     platform: 'linux',
@@ -146,6 +155,13 @@
     libC: 'musl',
 };
 
+const aarch64AlpineLinuxMusl: LinuxTarget = {
+    id: 'aarch64-alpine-linux-musl',
+    platform: 'linux',
+    arch: 'arm64',
+    libC: 'musl',
+};
+
 const powerpc64leUnknownLinuxGnu: LinuxTarget = {
     id: 'powerpc64le-unknown-linux-gnu',
     platform: 'linux',
@@ -172,8 +188,10 @@
     'i686-unknown-linux-musl': i686UnknownLinuxMusl,
     'i686-unknown-linux-gnu': i686UnknownLinuxGnu,
     'x86_64-unknown-linux-musl': x8664UnknownLinuxMusl,
+    'x86_64-alpine-linux-musl': x8664AlpineLinuxMusl,
     'x86_64-unknown-linux-gnu': x8664UnknownLinuxGnu,
     'aarch64-unknown-linux-musl': aarch64UnknownLinuxMusl,
+    'aarch64-alpine-linux-musl': aarch64AlpineLinuxMusl,
     'aarch64-unknown-linux-gnu': aarch64UnknownLinuxGnu,
     'powerpc64le-unknown-linux-musl': powerpc64leUnknownLinuxMusl,
     'powerpc64le-unknown-linux-gnu': powerpc64leUnknownLinuxGnu,
