diff --git a/src/lib66/exec/ssexec_scandir_create.c b/src/lib66/exec/ssexec_scandir_create.c
index 40b49e95..c4cc1349 100644
--- a/src/lib66/exec/ssexec_scandir_create.c
+++ b/src/lib66/exec/ssexec_scandir_create.c
@@ -174,7 +174,7 @@ static void inline log_perm(char const *str,uid_t *uid,gid_t *gid)
     if (!yourgid(gid,*uid)) log_dieusys(LOG_EXIT_SYS,"set gid of: ",str) ;
 }
 
-void inline shebang(buffer *b, char const *opts)
+void static inline shebang(buffer *b, char const *opts)
 {
     log_flow() ;
 
