From e369537daefd5f6aa8b7590a54d2ca454067e4aa Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Mon, 11 Mar 2024 16:01:49 -0400
Subject: [PATCH 2/3] point qmlplugindump to qmlplugindump-qt5

---
 src/qml/MediaScanner.0.1/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qml/MediaScanner.0.1/CMakeLists.txt b/src/qml/MediaScanner.0.1/CMakeLists.txt
index 6e39cdf..c70bd13 100644
--- a/src/qml/MediaScanner.0.1/CMakeLists.txt
+++ b/src/qml/MediaScanner.0.1/CMakeLists.txt
@@ -94,7 +94,7 @@ else()
     )
 
     if(NOT CMAKE_CROSSCOMPILING)
-      find_program(qmlplugindump_exe qmlplugindump)
+      find_program(qmlplugindump_exe qmlplugindump-qt5)
       if(NOT qmlplugindump_exe)
         message(FATAL_ERROR "Could not locate qmlplugindump.")
       endif()
-- 
2.53.0

