From 1af9d8f6209534c54c573b021e4f04e678bb090b Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Mon, 11 Mar 2024 16:01:49 -0400
Subject: [PATCH 2/3] point qmlplugindump to qmlplugindump-qt5

---
 src/qml/MediaScanner.0.1/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/qml/MediaScanner.0.1/CMakeLists.txt b/src/qml/MediaScanner.0.1/CMakeLists.txt
index 73b5146..489988a 100644
--- a/src/qml/MediaScanner.0.1/CMakeLists.txt
+++ b/src/qml/MediaScanner.0.1/CMakeLists.txt
@@ -59,7 +59,7 @@ install(
 )
 
 if(NOT CMAKE_CROSSCOMPILING)
-  find_program(qmlplugindump_exe qmlplugindump)
+  find_program(qmlplugindump_exe qmlplugindump-qt5)
   if(NOT qmlplugindump_exe)
     message(FATAL_ERROR "Could not locate qmlplugindump.")
   endif()
-- 
2.43.2

