diff --git a/src/extractor/TaglibExtractor.cc b/src/extractor/TaglibExtractor.cc
index 9812d58..c30156f 100644
--- a/src/extractor/TaglibExtractor.cc
+++ b/src/extractor/TaglibExtractor.cc
@@ -205,7 +205,7 @@ void parse_id3v2(const TagLib::ID3v2::Tag *tag, MediaFileBuilder &builder) {
 }
 
 void parse_mp4(const TagLib::MP4::Tag *tag, MediaFileBuilder &builder) {
-    const auto& items = const_cast<TagLib::MP4::Tag*>(tag)->itemListMap();
+    const auto& items = const_cast<TagLib::MP4::Tag*>(tag)->itemMap();
 
     if (items.contains("\251day")) {
         builder.setDate(check_date(items["\251day"].toStringList().front().to8Bit(true)));
