Ref: https://github.com/swig/swig/issues/3260#issuecomment-3272521479
---
diff -ruN a/src/python/s2_common.i b/src/python/s2_common.i
--- a/src/python/s2_common.i	2024-02-03 08:01:27.000000000 +0000
+++ b/src/python/s2_common.i	2025-11-11 04:33:58.580000000 +0000
@@ -169,7 +169,7 @@
     if (SWIG_IsOK(res)) {
       loops.push_back(loop->Clone());
     } else {
-      SWIG_Python_TypeError(SWIG_TypePrettyName($descriptor(S2Loop *)), element);
+      SWIG_Error(SWIG_TypeError, "'$descriptor(S2Loop *)' is expected");
       SWIG_Python_ArgFail(i);
       Py_DECREF(element);
       Py_DECREF(iterator);
