diff --git a/tools/fiptool/fiptool.c.orig b/tools/fiptool/fiptool.c
index 27119a1aaa4..7d3ef0c739a 100644
--- a/tools/fiptool/fiptool.c.orig
+++ b/tools/fiptool/fiptool.c
@@ -315,7 +316,7 @@ static int parse_fip(const char *filename, fip_toc_header_t *toc_header_out)
 
 #ifdef BLKGETSIZE64
 	if ((st.st_mode & S_IFBLK) != 0)
-		if (ioctl(fileno(fp), BLKGETSIZE64, &st_size) == -1)
+		if (ioctl(fileno(fp), (int)BLKGETSIZE64, &st_size) == -1)
 			log_err("ioctl %s", filename);
 #endif
 
