diff --git a/include/lib/el3_runtime/cpu_data.h b/include/lib/el3_runtime/cpu_data.h
index 6445e281f..ac382a3b2 100644
--- a/include/lib/el3_runtime/cpu_data.h
+++ b/include/lib/el3_runtime/cpu_data.h
@@ -31,8 +31,8 @@
 #define CPU_DATA_WARMBOOT_EP_INFO_ALIGN	CPU_WORD_SIZE
 #define CPU_DATA_CPU_OPS_PTR_SIZE	CPU_WORD_SIZE
 #define CPU_DATA_CPU_OPS_PTR_ALIGN	CPU_WORD_SIZE
-#define CPU_DATA_PSCI_SVC_CPU_DATA_SIZE	CACHE_WRITEBACK_GRANULE
-#define CPU_DATA_PSCI_SVC_CPU_DATA_ALIGN CACHE_WRITEBACK_GRANULE
+#define CPU_DATA_PSCI_SVC_CPU_DATA_SIZE 12
+#define CPU_DATA_PSCI_SVC_CPU_DATA_ALIGN CPU_WORD_SIZE
 #if ENABLE_PAUTH
 /* uint64_t apiakey[2] */
 #define CPU_DATA_APIAKEY_SIZE		16
diff --git a/include/lib/psci/psci.h b/include/lib/psci/psci.h
index 18e16c7ea..3db4219ac 100644
--- a/include/lib/psci/psci.h
+++ b/include/lib/psci/psci.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2026, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2013-2025, Arm Limited and Contributors. All rights reserved.
  * Copyright (c) 2023, NVIDIA Corporation. All rights reserved.
  *
  * SPDX-License-Identifier: BSD-3-Clause
@@ -296,7 +296,7 @@ typedef struct psci_power_state {
  * It is populated in the per-cpu data array. In return we get a guarantee that
  * this information will not reside on a cache line shared with another cpu.
  ******************************************************************************/
-struct __packed __aligned(CACHE_WRITEBACK_GRANULE) psci_cpu_data {
+typedef struct psci_cpu_data {
 	/* State as seen by PSCI Affinity Info API */
 	aff_info_state_t aff_info_state;
 
@@ -308,17 +308,14 @@ struct __packed __aligned(CACHE_WRITEBACK_GRANULE) psci_cpu_data {
 
 	/* The local power state of this CPU */
 	plat_local_state_t local_state;
-	/* pad up to a cache line */
-	uint8_t _padding[CACHE_WRITEBACK_GRANULE -
-			 (sizeof(aff_info_state_t) + sizeof(unsigned int) +
-			  sizeof(plat_local_state_t))];
-};
+/* This should not straddle cache lines so align to closest power of 2. */
+} psci_cpu_data_t __aligned(16);
 
 /*
- * Sanity check that the struct will fit alone within a cache line. Otherwise
- * CMOs might affect other memory and have unexpected consequences.
+ * Sanity check that the struct will fit within a cache line. Otherwise CMOs
+ * might affect other memory and have unexpected consequences.
  */
-CASSERT(sizeof(struct psci_cpu_data) == CACHE_WRITEBACK_GRANULE,
+CASSERT(sizeof(psci_cpu_data_t) <= CACHE_WRITEBACK_GRANULE,
 	psci_cpu_data_bigger_than_cache_line);
 
 /*******************************************************************************
diff --git a/lib/psci/psci_setup.c b/lib/psci/psci_setup.c
index 7994b5d0d..c090c2bd2 100644
--- a/lib/psci/psci_setup.c
+++ b/lib/psci/psci_setup.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2013-2026, Arm Limited and Contributors. All rights reserved.
+ * Copyright (c) 2013-2025, Arm Limited and Contributors. All rights reserved.
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -55,7 +55,7 @@ static void __init psci_init_pwr_domain_node(uint16_t node_idx,
 		psci_non_cpu_pd_nodes[node_idx].local_state =
 							 PLAT_MAX_OFF_STATE;
 	} else {
-		struct psci_cpu_data *svc_cpu_data;
+		psci_cpu_data_t *svc_cpu_data;
 
 		assert(node_idx < PLATFORM_CORE_COUNT);
 
