--- a/plat/rockchip/rk3399/drivers/m0/Makefile
+++ b/plat/rockchip/rk3399/drivers/m0/Makefile
@@ -29,7 +29,7 @@ C_SOURCES		:= src/dram.c	\
 C_SOURCES_PMU		:= src/suspend.c
 
 # Flags definition
-COMMON_FLAGS		:= -g -mcpu=$(ARCH) -mthumb -Wall -O3 -nostdlib -mfloat-abi=soft
+COMMON_FLAGS		:= -g -mcpu=$(ARCH) --target=arm-none-eabi -mthumb -Wall -O3 -nostdlib -mfloat-abi=soft
 CFLAGS			:= -ffunction-sections -fdata-sections -fomit-frame-pointer -fno-common
 ASFLAGS			:= -Wa,--gdwarf-2
 LDFLAGS			:= -Wl,--gc-sections -Wl,--build-id=none
--- a/make_helpers/toolchains/rk3399-m0.mk
+++ b/make_helpers/toolchains/rk3399-m0.mk
@@ -6,26 +6,26 @@
 
 rk3399-m0-name := RK3399 M0
 
-rk3399-m0-cc-default-id := gnu-gcc
-rk3399-m0-cc-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)gcc
+rk3399-m0-cc-default-id := llvm-clang
+rk3399-m0-cc-default := clang
 
-rk3399-m0-cpp-default-id := gnu-gcc
-rk3399-m0-cpp-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)gcc
+rk3399-m0-cpp-default-id := llvm-clang
+rk3399-m0-cpp-default := clang
 
-rk3399-m0-as-default-id := gnu-gcc
-rk3399-m0-as-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)gcc
+rk3399-m0-as-default-id := llvm-clang
+rk3399-m0-as-default := clang
 
-rk3399-m0-ld-default-id := gnu-gcc
-rk3399-m0-ld-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)gcc
+rk3399-m0-ld-default-id := llvm-clang
+rk3399-m0-ld-default := clang
 
-rk3399-m0-oc-default-id := gnu-objcopy
-rk3399-m0-oc-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)objcopy
+rk3399-m0-oc-default-id := llvm-objcopy
+rk3399-m0-oc-default := llvm-objcopy
 
-rk3399-m0-od-default-id := gnu-objdump
-rk3399-m0-od-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)objdump
+rk3399-m0-od-default-id := llvm-objdump
+rk3399-m0-od-default := llvm-objdump
 
-rk3399-m0-ar-default-id := gnu-ar
-rk3399-m0-ar-default := $(or $(M0_CROSS_COMPILE),arm-none-eabi-)gcc-ar
+rk3399-m0-ar-default-id := llvm-ar
+rk3399-m0-ar-default := llvm-ar
 
 rk3399-m0-dtc-default-id := generic-dtc
 rk3399-m0-dtc-default := dtc


Needed to stop linker errors with ldd:
ld.lld: error: section .ARM.exidx virtual address range overlaps with .m0_bin
489>>> .ARM.exidx range is [0x0, 0xF]
490>>> .m0_bin range is [0x0, 0x267]
491ld.lld: error: section .ARM.exidx load address range overlaps with .m0_bin
492>>> .ARM.exidx range is [0x0, 0xF]
493>>> .m0_bin range is [0x0, 0x267]
494ld.lld: error: section .ARM.exidx virtual address range overlaps with .m0_bin
495>>> .ARM.exidx range is [0x0, 0xF]
496>>> .m0_bin range is [0x0, 0x347]
497ld.lld: error: section .ARM.exidx load address range overlaps with .m0_bin
498>>> .ARM.exidx range is [0x0, 0xF]
499>>> .m0_bin range is [0x0, 0x347]
 
diff --git a/plat/rockchip/rk3399/drivers/m0/src/rk3399m0.ld.S b/plat/rockchip/rk3399/drivers/m0/src/rk3399m0.ld.S
index bfe054e..fedb012 100644
--- a/plat/rockchip/rk3399/drivers/m0/src/rk3399m0.ld.S
+++ b/plat/rockchip/rk3399/drivers/m0/src/rk3399m0.ld.S
@@ -22,5 +22,5 @@ SECTIONS {
 		*(.co_stack*)
 	}
 
-	/DISCARD/ : { *(.comment) *(.note*) }
+	/DISCARD/ : { *(.comment) *(.note*) *(.ARM.exidx) }
 }
