From 30751070371ee32a3b58f8f416c07ddc1d3b5680 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Tue, 1 Feb 2022 09:26:02 +0000
Subject: [PATCH] disable gcc ubsan

---
 GNUmakefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index a27b453..fbc3605 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -169,7 +169,7 @@ $(eval $(call CHECK_CFLAGS,CFLAGS,-fanalyzer))
 $(eval $(call CHECK_CFLAGS,CFLAGS,-fstack-protector-strong))
 $(eval $(call CHECK_CFLAGS,CFLAGS,-fstack-clash-protection))
 $(eval $(call CHECK_CFLAGS,CFLAGS,-fcf-protection))
-$(eval $(call CHECK_CFLAGS,CFLAGS,-fsanitize=undefined))
+$(eval $(call CHECK_CFLAGS,CFLAGS))
 
 
 CFLAGS += -save-temps=obj
-- 
2.35.1

