From 648ebb687339269bfd056703f0aa5a62a9ba4f06 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Tue, 1 Feb 2022 09:31:33 +0000
Subject: [PATCH 2/2] remove O2 optimization

---
 GNUmakefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/GNUmakefile b/GNUmakefile
index 9747798..d192bab 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -147,7 +147,7 @@ common_LIBS      =
 
 common_CPPFLAGS += -DPACKAGE_TARNAME='"$(PACKAGE_TARNAME)"'
 common_CPPFLAGS += -DPACKAGE_VERSION='"$(PACKAGE_VERSION)"'
-common_CFLAGS   += -O2 -g
+common_CFLAGS   += -g
 common_CFLAGS   += -std=gnu99
 common_CFLAGS   += $(if $(filter %.o,$@),-Wa$(comma)-adhlns=$(@:.o=.lst))
 common_CFLAGS   += -pedantic
-- 
2.35.1

