diff -upr doxygen-1.8.17.orig/doc/CMakeLists.txt doxygen-1.8.17/doc/CMakeLists.txt
--- doxygen-1.8.17.orig/doc/CMakeLists.txt	2020-01-05 16:37:34.415645891 +0100
+++ doxygen-1.8.17/doc/CMakeLists.txt	2020-01-05 16:38:10.395666071 +0100
@@ -209,22 +209,22 @@ install(FILES
         "${PROJECT_BINARY_DIR}/man/doxywizard.1"
         "${PROJECT_BINARY_DIR}/man/doxysearch.1"
         "${PROJECT_BINARY_DIR}/man/doxyindexer.1"
-        DESTINATION ${CMAKE_INSTALL_MANDIR}/man1
+        DESTINATION ${CMAKE_INSTALL_MANDIR}
 )
 
 install(FILES
         "${PROJECT_BINARY_DIR}/latex/doxygen_manual.pdf"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
 
 if (build_doc_chm)
 install(FILES
         "${PROJECT_BINARY_DIR}/chm/doxygen_manual.chm"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
 endif ()
 
 install(DIRECTORY
         "${PROJECT_BINARY_DIR}/html"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
