--- doxygen-1.8.14/doc/CMakeLists.txt.config	2017-12-31 15:28:32.351163123 +0100
+++ doxygen-1.8.14/doc/CMakeLists.txt	2017-12-31 15:29:11.490119182 +0100
@@ -168,15 +168,15 @@ install(FILES
         "${PROJECT_BINARY_DIR}/man/doxywizard.1"
         "${PROJECT_BINARY_DIR}/man/doxysearch.1"
         "${PROJECT_BINARY_DIR}/man/doxyindexer.1"
-        DESTINATION ${CMAKE_INSTALL_MANDIR}/man1
+        DESTINATION "${MAN_INSTALL_DIR}"
 )
 
 install(FILES
         "${PROJECT_BINARY_DIR}/latex/doxygen_manual.pdf"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
 
 install(DIRECTORY
         "${PROJECT_BINARY_DIR}/html"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
