diff -up doxygen-1.8.10/doc/CMakeLists.txt.than doxygen-1.8.10/doc/CMakeLists.txt
--- doxygen-1.8.10/doc/CMakeLists.txt.than	2015-08-28 11:46:15.675460840 +0200
+++ doxygen-1.8.10/doc/CMakeLists.txt	2015-08-28 11:51:27.894592457 +0200
@@ -88,17 +88,17 @@ install(FILES
         "${PROJECT_BINARY_DIR}/man/doxywizard.1"
         "${PROJECT_BINARY_DIR}/man/doxysearch.1"
         "${PROJECT_BINARY_DIR}/man/doxyindexer.1"
-        DESTINATION man/man1
+        DESTINATION "${MAN_INSTALL_DIR}"
 )
 
 install(FILES
         "${PROJECT_BINARY_DIR}/latex/doxygen_manual.pdf"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
 
 install(DIRECTORY
         "${PROJECT_BINARY_DIR}/html"
-        DESTINATION "${CMAKE_INSTALL_PREFIX}/${DOC_INSTALL_DIR}"
+        DESTINATION "${DOC_INSTALL_DIR}"
 )
 
 endif()
