--- ./Make.defaults.orig
+++ ./Make.defaults
@@ -140,11 +140,11 @@
            -I$(TOPDIR)/inc/protocol
 
 ifeq (FreeBSD, $(findstring FreeBSD, $(OS)))
-CFLAGS  += $(ARCH3264) -g -O2 -fpic -Wall -Wextra -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -fpic -Wall -Wextra \
            -fshort-wchar -fno-strict-aliasing \
            -ffreestanding -fno-stack-protector
 else
-CFLAGS  += $(ARCH3264) -g -O2 -fpic -Wall -Wextra -Werror \
+CFLAGS  += $(ARCH3264) -g -O2 -fpic -Wall -Wextra \
            -fshort-wchar -fno-strict-aliasing \
            -fno-merge-constants -ffreestanding -fno-stack-protector \
            -fno-stack-check
