diff -ru xbmc-12.3-Frodo.orig/configure.in xbmc-12.3-Frodo/configure.in
--- xbmc-12.3-Frodo.orig/configure.in	2014-04-15 05:52:42.214313750 +0000
+++ xbmc-12.3-Frodo/configure.in	2014-04-15 05:49:51.921324165 +0000
@@ -556,11 +556,11 @@
      use_cpu="i686"
      ffmpeg_target_os=linux
      ;;
-  i*86*-linux-gnu*)
+  i*86*-linux-gnu*|i*86*-linux-uclibc*|i*86*-linux-musl*)
      ARCH="i486-linux"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
      ;;
-  x86_64-*-linux-gnu*)
+  x86_64-*-linux-gnu*|x86_64-*-linux-uclibc*|x86_64-*-linux-musl*)
      ARCH="x86_64-linux"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX")
      ;;
@@ -619,7 +619,7 @@
      ARCH="powerpc64-linux"
      AC_SUBST(ARCH_DEFINES, "-DTARGET_POSIX -DTARGET_LINUX -D_LINUX -D_POWERPC64")
      ;;
-  arm*-*-linux-gnu*)
+  arm*-*-linux-gnu*|arm*-*-linux-muslgnu*)
      use_texturepacker=no
      ARCH="arm"
      use_arch="arm"
diff -ru xbmc-12.3-Frodo.orig/lib/SlingboxLib/SlingboxLib.cpp xbmc-12.3-Frodo/lib/SlingboxLib/SlingboxLib.cpp
--- xbmc-12.3-Frodo.orig/lib/SlingboxLib/SlingboxLib.cpp	2013-12-12 21:47:49.000000000 +0000
+++ xbmc-12.3-Frodo/lib/SlingboxLib/SlingboxLib.cpp	2014-04-15 06:07:15.746789122 +0000
@@ -29,7 +29,9 @@
 #include <time.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
+#include <sys/time.h>
 #include <sys/socket.h>
+#include <sys/select.h>
 #define INVALID_SOCKET (SOCKET)(~0)
 #define SOCKET_ERROR (-1)
 #endif
diff -ru xbmc-12.3-Frodo.orig/lib/libXDAAP/thread.h xbmc-12.3-Frodo/lib/libXDAAP/thread.h
--- xbmc-12.3-Frodo.orig/lib/libXDAAP/thread.h	2013-12-12 21:47:49.000000000 +0000
+++ xbmc-12.3-Frodo/lib/libXDAAP/thread.h	2014-04-15 06:11:09.940947997 +0000
@@ -37,7 +37,7 @@
 #define ts_mutex      pthread_mutex_t
 #define ts_condition  pthread_cond_t
 
-#if defined(__APPLE__) || defined(__FreeBSD__)
+#if !defined(PTHREAD_MUTEX_RECURSIVE_NP)
 	#define PTHREAD_MUTEX_RECURSIVE_NP PTHREAD_MUTEX_RECURSIVE 
 #endif 
 	
diff -ru xbmc-12.3-Frodo.orig/xbmc/visualizations/XBMCProjectM/libprojectM/BeatDetect.cpp xbmc-12.3-Frodo/xbmc/visualizations/XBMCProjectM/libprojectM/BeatDetect.cpp
--- xbmc-12.3-Frodo.orig/xbmc/visualizations/XBMCProjectM/libprojectM/BeatDetect.cpp	2014-04-15 05:52:42.214313750 +0000
+++ xbmc-12.3-Frodo/xbmc/visualizations/XBMCProjectM/libprojectM/BeatDetect.cpp	2014-04-15 05:49:51.921324165 +0000
@@ -33,7 +33,7 @@
 
 #include "Common.hpp"
 #include "PCM.hpp"
-#include <cmath>
+#include "math.h"
 #include "BeatDetect.hpp"
 
 #ifdef _WIN32
diff -ru xbmc-12.3-Frodo.orig/tools/TexturePacker/XBTFWriter.cpp xbmc-12.3-Frodo/tools/TexturePacker/XBTFWriter.cpp
--- xbmc-12.3-Frodo.orig/tools/TexturePacker/XBTFWriter.cpp	2013-12-12 21:47:49.000000000 +0000
+++ xbmc-12.3-Frodo/tools/TexturePacker/XBTFWriter.cpp	2014-04-15 09:58:29.222353358 +0000
@@ -24,11 +24,7 @@
 #include <inttypes.h>
 #include "guilib/XBTF.h"
 #include "utils/EndianSwap.h"
-#if defined(__FreeBSD__)
 #include <stdlib.h>
-#elif !defined(__APPLE__)
-#include <malloc.h>
-#endif
 #include <memory.h>
 
 #define WRITE_STR(str, size, file) fwrite(str, size, 1, file)
diff -ru xbmc-12.3-Frodo.orig/xbmc/cores/DllLoader/ldt_keeper.c xbmc-12.3-Frodo/xbmc/cores/DllLoader/ldt_keeper.c
--- xbmc-12.3-Frodo.orig/xbmc/cores/DllLoader/ldt_keeper.c	2013-12-12 21:47:49.000000000 +0000
+++ xbmc-12.3-Frodo/xbmc/cores/DllLoader/ldt_keeper.c	2014-04-15 10:10:17.091888453 +0000
@@ -48,11 +48,7 @@
 #if defined(__GLIBC__) &&  (__GLIBC__ < 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ == 0))
 _syscall3( int, modify_ldt, int, func, void *, ptr, unsigned long, bytecount );
 #else
-#if defined(__ANDROID__) && defined(__i386__) && !defined(modify_ldt)
 #define modify_ldt(a,b,c) syscall( __NR_modify_ldt,  a, b, c);
-#else
-int modify_ldt(int func, void *ptr, unsigned long bytecount);
-#endif
 #endif
 #ifdef  __cplusplus
 }
