--- a/jj_tui.opam
+++ b/jj_tui.opam
@@ -23,7 +23,16 @@ depends: [
   "picos_std" {= "0.5.0"}
   "picos_io" {= "0.5.0"}
   "uutf"
+  "logs"
+  "logs-ppx"
+  "ppx_deriving"
+  "re"
+  "signal"
+  "spawn"
+  "yaml"
   "yojson"
+  "ppx_deriving_yaml"
+  "ppx_deriving_yojson"
   "ppx_record_updater"
   "odoc" {with-doc}
 ]
@@ -49,5 +58,6 @@ pin-depends:[
 ["eio-process.dev" "git+https://github.com/mbarbin/eio-process#ac1e965eb33f19fcedc7ce203111bca0c3fc8278"]
 ["nottui.dev" "git+https://github.com/faldor20/nottui#085cbd489600fa7c0279352d0a773dfa133ab39a"]
 ["lwd.dev" "git+https://github.com/faldor20/lwd#c19bc2fd55c2de977cdd283458ce06402b08febe"]
+["ppx_record_updater.dev" "git+https://github.com/faldor20/ppx_record_updater#15a6ac0fa1a98e21e2b4b68b2eaee088186d5515"]
 ]
 
