--- ./data/sql/lib.model.schema.sql.orig
+++ ./data/sql/lib.model.schema.sql
@@ -21,7 +21,7 @@
 	`account` VARCHAR(40),
 	PRIMARY KEY (`id`),
 	UNIQUE KEY `name_index` (`name`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 #-----------------------------------------------------------------------------
 #-- records
@@ -47,7 +47,7 @@
 	CONSTRAINT `records_FK_1`
 		FOREIGN KEY (`domain_id`)
 		REFERENCES `domains` (`id`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 #-----------------------------------------------------------------------------
 #-- supermasters
@@ -63,7 +63,7 @@
 	`account` VARCHAR(40),
 	`id` INTEGER  NOT NULL AUTO_INCREMENT,
 	PRIMARY KEY (`id`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 #-----------------------------------------------------------------------------
 #-- template
@@ -78,7 +78,7 @@
 	`name` VARCHAR(255),
 	`type` VARCHAR(45),
 	PRIMARY KEY (`id`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 #-----------------------------------------------------------------------------
 #-- template_record
@@ -101,7 +101,7 @@
 	CONSTRAINT `template_record_FK_1`
 		FOREIGN KEY (`template_id`)
 		REFERENCES `template` (`id`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 #-----------------------------------------------------------------------------
 #-- setting
@@ -115,7 +115,7 @@
 	`name` VARCHAR(255)  NOT NULL,
 	`value` TEXT,
 	PRIMARY KEY (`name`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 # This restores the fkey checks, after having unset them earlier
 SET FOREIGN_KEY_CHECKS = 1;
--- ./data/sql/plugins.sfPropelAuditPlugin.lib.model.schema.sql.orig
+++ ./data/sql/plugins.sfPropelAuditPlugin.lib.model.schema.sql
@@ -22,7 +22,7 @@
 	`type` VARCHAR(255),
 	`created_at` DATETIME,
 	PRIMARY KEY (`id`)
-)Type=MyISAM;
+) Engine=MyISAM;
 
 # This restores the fkey checks, after having unset them earlier
 SET FOREIGN_KEY_CHECKS = 1;
