diff --git a/build/install b/build/install
index a0bc399..514bce7 100755
--- a/build/install
+++ b/build/install
@@ -61,4 +61,7 @@ if [ -f Makefile ]; then
 fi
 
 #Perform the compilation
-phpize && ./configure --enable-phalcon && make && make install && echo -e "\nThanks for compiling Phalcon!\nBuild succeed: Please restart your web server to complete the installation"
+# phpize && ./configure --enable-phalcon && make && make install && echo -e "\nThanks for compiling Phalcon!\nBuild succeed: Please restart your web server to complete the installation"
+export PHALCONARCH="$DIR"
+phpize && ./configure --enable-phalcon && make
+
