diff --git a/driver/others/memory.c b/driver/others/memory.c
index 38d0637..5c5d8ed 100644
--- a/driver/others/memory.c
+++ b/driver/others/memory.c
@@ -187,25 +187,24 @@ int i,n;
      
 #if !defined(__GLIBC_PREREQ)
    return nums;
-#endif   
-#if !__GLIBC_PREREQ(2, 3)
+#else
+ #if !__GLIBC_PREREQ(2, 3)
    return nums;
-#endif   
+ #endif
 
-#if !__GLIBC_PREREQ(2, 7)
+ #if !__GLIBC_PREREQ(2, 7)
   ret = sched_getaffinity(0,sizeof(cpu_set_t), cpusetp);
   if (ret!=0) return nums;
   n=0;
-#if !__GLIBC_PREREQ(2, 6)  
+  #if !__GLIBC_PREREQ(2, 6)
   for (i=0;i<nums;i++)
      if (CPU_ISSET(i,cpusetp)) n++;
   nums=n;   
-#else
+  #else
   nums = CPU_COUNT(sizeof(cpu_set_t),cpusetp);
-#endif
+  #endif
   return nums;
-#endif
-
+ #else
   cpusetp = CPU_ALLOC(nums);
   if (cpusetp == NULL) return nums;
   size = CPU_ALLOC_SIZE(nums);
@@ -214,6 +213,8 @@ int i,n;
   nums = CPU_COUNT_S(size,cpusetp);
   CPU_FREE(cpusetp);
   return nums;
+ #endif
+#endif
 }
 #endif
 #endif
