--- ./arch/x86/xen/mmu.c.orig
+++ ./arch/x86/xen/mmu.c
@@ -1941,6 +1941,7 @@
 		set_page_prot(level2_kernel_pgt, PAGE_KERNEL_RO);
 		set_page_prot(level2_fixmap_pgt, PAGE_KERNEL_RO);
 		set_page_prot(level1_fixmap_pgt, PAGE_KERNEL_RO);
+		set_page_prot(level1_vsyscall_pgt, PAGE_KERNEL_RO);
 
 		/* Pin down new L4 */
 		pin_pagetable_pfn(MMUEXT_PIN_L4_TABLE,
--- ./arch/x86/include/asm/pgtable_64.h.orig
+++ ./arch/x86/include/asm/pgtable_64.h
@@ -24,6 +24,7 @@
 extern pmd_t level2_fixmap_pgt[512];
 extern pmd_t level2_ident_pgt[512*2];
 extern pte_t level1_fixmap_pgt[512];
+extern pte_t level1_vsyscall_pgt[512];
 extern pgd_t init_level4_pgt[512];
 
 #define swapper_pg_dir init_level4_pgt
