diff -upr xterm-323.orig/Tekproc.c xterm-323/Tekproc.c
--- xterm-323.orig/Tekproc.c	2016-03-09 15:33:04.900639959 +0100
+++ xterm-323/Tekproc.c	2016-03-09 15:33:14.412994110 +0100
@@ -253,9 +253,9 @@ static Dimension defOne = 1;
 static XtResource resources[] =
 {
     {XtNwidth, XtCWidth, XtRDimension, sizeof(Dimension),
-     XtOffsetOf(CoreRec, core.width), XtRDimension, (caddr_t) & defOne},
+     XtOffsetOf(CoreRec, core.width), XtRDimension, (void*) & defOne},
     {XtNheight, XtCHeight, XtRDimension, sizeof(Dimension),
-     XtOffsetOf(CoreRec, core.height), XtRDimension, (caddr_t) & defOne},
+     XtOffsetOf(CoreRec, core.height), XtRDimension, (void*) & defOne},
     Fres("fontLarge", XtCFont, tek.Tfont[TEK_FONT_LARGE], "9x15"),
     Fres("font2", XtCFont, tek.Tfont[TEK_FONT_2], "6x13"),
     Fres("font3", XtCFont, tek.Tfont[TEK_FONT_3], "8x13"),
