diff --git a/test/test.py b/test/test.py
index be1b10d..5caab39 100755
--- a/test/test.py
+++ b/test/test.py
@@ -1350,6 +1349,7 @@ class TestBashGlobal(TestBash):
             self.sh.run_command("cd " + os.getcwd())
             self.assertEqual(self.sh.run_command("python3 -m package.prog basic f\t"), "foo\r\n")
 
+    @unittest.skip("no pip")
     def _test_console_script(self, package=False, wheel=False):
         with TempDir(prefix="test_dir_py", dir="."):
             self.sh.run_command("cd " + os.getcwd())
