diff --git a/src/core/richtext/richcss.h b/src/core/richtext/richcss.h
index d0e2ee4..bf2cc3d 100644
--- a/src/core/richtext/richcss.h
+++ b/src/core/richtext/richcss.h
@@ -57,7 +57,11 @@ public:
 signals:
 	void changed();
 
+#ifdef SUBTITLECOMPOSER_TESTS
+public:
+#else
 private:
+#endif
 	static Selector parseCssSelector(const QChar **stylesheet);
 	static RuleList parseCssRules(const QChar **stylesheet);
 	static QString parseCssKey(const QChar **stylesheet);
diff --git a/src/tests/richcsstest.cpp b/src/tests/richcsstest.cpp
index 07f7133..0e999e1 100644
--- a/src/tests/richcsstest.cpp
+++ b/src/tests/richcsstest.cpp
@@ -4,9 +4,8 @@
     SPDX-License-Identifier: GPL-2.0-or-later
 */
 
-#define private public // TODO: FIXME: compare selectors with final stylesheet
+#define SUBTITLECOMPOSER_TESTS
 #include "richcsstest.h"
-#undef private
 
 #include "helpers/common.h"
 
