diff -rupN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-09-14 20:00:57.000000000 +0000
+++ b/CMakeLists.txt	2025-11-23 18:45:42.110000000 +0000
@@ -34,6 +34,10 @@ include(FeatureSummary)
 find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS
 	Core Widgets Gui Test Qml ${QT_EXTRA_COMPONENTS})
 
+if (Qt6_VERSION VERSION_GREATER_EQUAL "6.10.0")
+  find_package(Qt6WidgetsPrivate ${QT_MINIMUM_VERSION} REQUIRED)
+endif()
+
 find_package(KF${KF_MAJOR_VERSION} ${KF_MIN_VERSION} REQUIRED COMPONENTS
 	Config ConfigWidgets CoreAddons I18n KIO XmlGui
 	Sonnet Codecs TextWidgets WidgetsAddons)
