From 6de152a1687ba5c2a5c522280942cd1f7647fc5c Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Mon, 12 Nov 2018 21:29:10 +0100
Subject: [PATCH] allow passing $(DESTDIR)

---
 Makefile | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 271fd1d..af17196 100644
--- a/Makefile
+++ b/Makefile
@@ -20,13 +20,15 @@ clean:
 
 .PHONY: install
 install:
-	cp $(NAME) $(PREFIX)/bin/$(NAME)
-	cp man/$(MANPAGE) $(PREFIX)/share/man/man$(MAN_SECTION)/$(MANPAGE)
+	install -d $(DESTDIR)/$(PREFIX)/bin
+	install -d $(DESTDIR)/$(PREFIX)/share/man/man$(MAN_SECTION)
+	install -m 0755 $(NAME) $(DESTDIR)/$(PREFIX)/bin/$(NAME)
+	install -m 0644 man/$(MANPAGE) $(DESTDIR)/$(PREFIX)/share/man/man$(MAN_SECTION)/$(MANPAGE)
 
 .PHONY: uninstall
 uninstall:
-	rm -f $(PREFIX)/bin/$(NAME)
-	rm -f $(PREFIX)/share/man/man$(MAN_SECTION)/$(MANPAGE)
+	rm -f $(DESTDIR)/$(PREFIX)/bin/$(NAME)
+	rm -f $(DESTDIR)/$(PREFIX)/share/man/man$(MAN_SECTION)/$(MANPAGE)
 
 .PHONY: check
 check:
-- 
2.19.1

