diff --git a/src/output.c b/src/output.c
index e9b1442..c2a6b6e 100644
--- a/src/output.c
+++ b/src/output.c
@@ -223,8 +223,8 @@ int print_header(int option) {
 		        fprintf(tmp_out_file,"<title>bwm-ng stats</title>\n</head>\n<body>\n");
 			}
 	        fprintf(tmp_out_file,"<div class=\"bwm-ng-header\">bwm-ng bwm-ng v" VERSION " (refresh %is); input: ",html_refresh);
-            fprintf(tmp_out_file,input2str());
-            fprintf(tmp_out_file,show_all_if2str());
+            fprintf(tmp_out_file,"%s",input2str());
+            fprintf(tmp_out_file,"%s",show_all_if2str());
 	        fprintf(tmp_out_file,"</div><table class=\"bwm-ng-output\">");
 			fprintf(tmp_out_file,"<tr class=\"bwm-ng-head\"><td class=\"bwm-ng-name\">Interface</td><td>Rx</td><td>Tx</td><td>Total</td></tr>");
 			break;
@@ -234,7 +234,7 @@ int print_header(int option) {
 			if (output_method==PLAIN_OUT && ansi_output) printf("\033[1;2H");
 	        printf("bwm-ng v" VERSION " (delay %2.3fs); ",(float)delay/1000);
 			if (output_method==PLAIN_OUT) printf("press 'ctrl-c' to end this%s",(ansi_output ? "\033[2;2H" : "")); else printf("input: ");
-            printf(input2str());
+            printf("%s",input2str());
             printf("%s\n",show_all_if2str());
 			if (output_method==PLAIN_OUT) {
 				if (ansi_output)
