diff --git a/src/output.c b/src/output.c
index f5f01cd..b451832 100644
--- a/src/output.c
+++ b/src/output.c
@@ -151,7 +151,7 @@ int print_header(int option) {
 	        erase();
 		    mvwprintw(stdscr,1,2,"bwm-ng v" VERSION" (probing every %2.3fs), press 'h' for help",(float)delay/1000);
             mvwprintw(stdscr,2,2,"input: %s type: %s",input2str(),output_type2str());
-            wprintw(stdscr,show_all_if2str());
+            wprintw(stdscr, "%s", show_all_if2str());
 	        mvwprintw(stdscr,3,2,"%c         iface                   Rx                   Tx                Total",(char)IDLE_CHARS[option]);
 	        /* go to next char for next run */
 	        option++;
