commit d467b855dfe250a259281e61b73959684895904e
Author: Jan Radon <j.radon@tallence.com>
Date:   Thu Feb 13 00:01:17 2020 +0100

    librados nautilus latests

diff --git a/src/librmb/rados-storage-impl.cpp b/src/librmb/rados-storage-impl.cpp
index fb60e41..375ca4f 100644
--- a/src/librmb/rados-storage-impl.cpp
+++ b/src/librmb/rados-storage-impl.cpp
@@ -278,7 +278,7 @@ int RadosStorageImpl::move(std::string &src_oid, const char *src_ns, std::string
     src_io_ctx.dup(dest_io_ctx);
     src_io_ctx.set_namespace(src_ns);
     dest_io_ctx.set_namespace(dest_ns);
-    write_op.copy_from(src_oid, src_io_ctx, 0);
+    write_op.copy_from(src_oid, src_io_ctx, 0,0);
 
   } else {
     src_io_ctx = dest_io_ctx;
@@ -335,7 +335,7 @@ int RadosStorageImpl::copy(std::string &src_oid, const char *src_ns, std::string
   } else {
     src_io_ctx = dest_io_ctx;
   }
-  write_op.copy_from(src_oid, src_io_ctx, 0);
+  write_op.copy_from(src_oid, src_io_ctx, 0, 0);
 
   // because we create a copy, save date needs to be updated
   // as an alternative we could use &ctx->data.save_date here if we save it to xattribute in write_metadata
diff --git a/src/tests/sync-rbox/it_test_sync_rbox_2.cpp b/src/tests/sync-rbox/it_test_sync_rbox_2.cpp
index a91dbba..719783f 100644
--- a/src/tests/sync-rbox/it_test_sync_rbox_2.cpp
+++ b/src/tests/sync-rbox/it_test_sync_rbox_2.cpp
@@ -73,7 +73,7 @@ static void copy_object(struct mail_namespace *_ns, struct mailbox *box) {
   i_debug("Last Version = %lu for obj: %s , stat =%d, %ld", r_storage->s->get_io_ctx().get_last_version(), oid.c_str(),
           stat_ret, size);
 
-  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version());
+  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version(),0);
   int ret = r_storage->s->get_io_ctx().operate(test_oid, &write_op);
 
   i_debug("copy operate: %d for %s", ret, test_oid.c_str());
diff --git a/src/tests/sync-rbox/it_test_sync_rbox_alt.cpp b/src/tests/sync-rbox/it_test_sync_rbox_alt.cpp
index 2904c90..dee5978 100644
--- a/src/tests/sync-rbox/it_test_sync_rbox_alt.cpp
+++ b/src/tests/sync-rbox/it_test_sync_rbox_alt.cpp
@@ -73,7 +73,7 @@ static void copy_object(struct mail_namespace *_ns, struct mailbox *box) {
   i_debug("Last Version = %lu for obj: %s , stat =%d, %ld", r_storage->s->get_io_ctx().get_last_version(), oid.c_str(),
           stat_ret, size);
 
-  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version());
+  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version(),0);
   int ret = r_storage->s->get_io_ctx().operate(test_oid, &write_op);
 
   i_debug("copy operate: %d for %s", ret, test_oid.c_str());
diff --git a/src/tests/sync-rbox/it_test_sync_rbox_duplicate_uid.cpp b/src/tests/sync-rbox/it_test_sync_rbox_duplicate_uid.cpp
index b7253c7..e41f184 100644
--- a/src/tests/sync-rbox/it_test_sync_rbox_duplicate_uid.cpp
+++ b/src/tests/sync-rbox/it_test_sync_rbox_duplicate_uid.cpp
@@ -73,7 +73,7 @@ static void copy_object(struct mail_namespace *_ns, struct mailbox *box) {
   i_debug("Last Version = %lu for obj: %s , stat =%d, %ld", r_storage->s->get_io_ctx().get_last_version(), oid.c_str(),
           stat_ret, size);
 
-  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version());
+  write_op.copy_from(oid, r_storage->s->get_io_ctx(), r_storage->s->get_io_ctx().get_last_version(),0);
   int ret = r_storage->s->get_io_ctx().operate(test_oid, &write_op);
 
   i_debug("copy operate: %d for %s", ret, test_oid.c_str());
