syntax::test::test_pretty_print assertion failed

--- a/helix-core/src/syntax.rs  2022-12-13 21:41:42
+++ b/helix-core/src/syntax.rs  2022-12-14 11:01:48
@@ -2507,8 +2507,9 @@ mod test {
                 "(singleton_method\n",
                 "  object: (self)\n",
                 "  name: (identifier)\n",
-                "  body: (body_statement\n",
-                "    (true)))"
+                //"  body: (body_statement\n",
+                //"    (true)))"
+                "  (true))"
             ),
             0,
             source.len(),
