diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe21597..1dd3266 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@ ADD_DEFINITIONS(-DQT_DISABLE_DEPRECATED_BEFORE=0x060000 -DUSE_WIDGETS)
 find_package(Qt6 COMPONENTS BuildInternals Widgets LinguistTools REQUIRED)
 
 #get_target_property(QT_QTPATHS_EXECUTABLE Qt6::qtpaths IMPORTED_LOCATION)
-get_target_property(QT_LRELEASE_EXECUTABLE Qt6::lrelease IMPORTED_LOCATION)
+set(QT_LRELEASE_EXECUTABLE /usr/lib/qt6/bin/lrelease)
 
 #if(QT_QTPATHS_EXECUTABLE)
 #    message(STATUS "Found qtpaths executable: " ${QT_QTPATHS_EXECUTABLE})
