diff --git a/aclocal.m4.original b/aclocal.m4
index 2062b0d..9fb016f 100644
--- a/aclocal.m4.original
+++ b/aclocal.m4
@@ -227,7 +227,7 @@ AC_DEFUN([FPTOOLS_SET_HASKELL_PLATFORM_VARS],
 
     checkVendor() {
         case [$]1 in
-        dec|none|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld)
+        dec|none|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld|alpine)
             ;;
         *)
             echo "Unknown vendor [$]1"
