diff --git a/libraries/base/tests/IO/all.T b/libraries/base/tests/IO/all.T
index 9475183..ba662de 100644
--- a/libraries/base/tests/IO/all.T
+++ b/libraries/base/tests/IO/all.T
@@ -119,7 +119,9 @@ test('encoding001', [], compile_and_run, [''])
 
 test('encoding002', normal, compile_and_run, [''])
 test('encoding003', normal, compile_and_run, [''])
-test('encoding004', extra_files(['encoded-data/']), compile_and_run, [''])
+test('encoding004',
+     [when(platform('x86_64-alpine-linux'), skip), extra_files(['encoded-data/'])],
+     compile_and_run, [''])
 test('encoding005', normal, compile_and_run, [''])
 
 test('environment001', [], makefile_test, ['environment001-test'])
diff --git a/testsuite/tests/ghci/linking/all.T b/testsuite/tests/ghci/linking/all.T
index 29a3142..0fd3be9 100644
--- a/testsuite/tests/ghci/linking/all.T
+++ b/testsuite/tests/ghci/linking/all.T
@@ -7,6 +7,7 @@ test('ghcilink001',
 
 test('ghcilink002', [extra_files(['TestLink.hs', 'f.c']),
                      when(unregisterised(), fragile(16085)),
+                     when(platform('x86_64-alpine-linux'), skip),
                      unless(doing_ghci, skip)],
      makefile_test, ['ghcilink002'])
 
diff --git a/testsuite/tests/ghci/linking/dyn/all.T b/testsuite/tests/ghci/linking/dyn/all.T
index 127c970..62b053c 100644
--- a/testsuite/tests/ghci/linking/dyn/all.T
+++ b/testsuite/tests/ghci/linking/dyn/all.T
@@ -25,6 +25,7 @@ test('T10955dyn', [extra_files(['A.c', 'B.c'])], makefile_test, ['compile_libAB_
 test('T10458',
      [extra_files(['A.c']),
       unless(doing_ghci, skip),
+      when(platform('x86_64-alpine-linux'), skip),
       pre_cmd('$MAKE -s --no-print-directory compile_libT10458'),
       extra_hc_opts('-L"$PWD/T10458dir" -lAS')],
      ghci_script, ['T10458.script'])
