Iterable has been moved to collections.abc
diff --git a/testsuite/driver/testlib.py b/testsuite/driver/testlib.py
index e899d61b..2f967666 100644
--- a/testsuite/driver/testlib.py
+++ b/testsuite/driver/testlib.py
@@ -15,7 +15,7 @@ import glob
 import sys
 from math import ceil, trunc
 from pathlib import Path, PurePath
-import collections
+import collections.abc
 import subprocess
 
 from testglobals import config, ghc_env, default_testopts, brokens, t, \
@@ -809,7 +809,7 @@ def join_normalisers(*a):
         Taken from http://stackoverflow.com/a/2158532/946226
         """
         for el in l:
-            if (isinstance(el, collections.Iterable)
+            if (isinstance(el, collections.abc.Iterable)
                 and not isinstance(el, (bytes, str))):
                 for sub in flatten(el):
                     yield sub
