diff --git a/numpy/_core/include/numpy/npy_endian.h b/numpy/_core/include/numpy/npy_endian.h
index 5e58a7f..19e5e64 100644
--- a/numpy/_core/include/numpy/npy_endian.h
+++ b/numpy/_core/include/numpy/npy_endian.h
@@ -12,7 +12,7 @@
     #if defined(NPY_HAVE_ENDIAN_H)
     #include <endian.h>
     #elif defined(NPY_HAVE_SYS_ENDIAN_H)
-    #include <sys/endian.h>
+    #include <bsd/sys/endian.h>
     #endif
 
     #if defined(BYTE_ORDER) && defined(BIG_ENDIAN) && defined(LITTLE_ENDIAN)
diff --git a/numpy/_core/meson.build b/numpy/_core/meson.build
index dbf1a14..2abc06b 100644
--- a/numpy/_core/meson.build
+++ b/numpy/_core/meson.build
@@ -494,7 +494,7 @@ cdata.set10('HAVE_LDOUBLE_' + longdouble_format, true)
 if cc.has_header('endian.h')
   cdata.set10('NPY_HAVE_ENDIAN_H', true)
 endif
-if cc.has_header('sys/endian.h')
+if cc.has_header('bsd/sys/endian.h')
   cdata.set10('NPY_HAVE_SYS_ENDIAN_H', true)
 endif
 # Build-time option to disable threading is stored and exposed in numpyconfig.h
