diff --git a/numpy/distutils/command/config.py b/numpy/distutils/command/config.py
index 1f4037b..bf1d1c5 100644
--- a/numpy/distutils/command/config.py
+++ b/numpy/distutils/command/config.py
@@ -16,7 +16,6 @@ from distutils.file_util import copy_file
 from distutils.ccompiler import CompileError, LinkError
 import distutils
 from numpy.distutils.exec_command import filepath_from_subprocess_output
-from numpy.distutils.mingw32ccompiler import generate_manifest
 from numpy.distutils.command.autodist import (check_gcc_function_attribute,
                                               check_gcc_function_attribute_with_intrinsics,
                                               check_gcc_variable_attribute,
@@ -158,8 +157,8 @@ class config(old_config):
                 if fileexists: continue
                 log.warn('could not find library %r in directories %s' \
                          % (libname, library_dirs))
-        elif self.compiler.compiler_type == 'mingw32':
-            generate_manifest(self)
+        # elif self.compiler.compiler_type == 'mingw32':
+        #     generate_manifest(self)
         return self._wrap_method(old_config._link, lang,
                                  (body, headers, include_dirs,
                                   libraries, library_dirs, lang))
