--- a/cxxtest/cxxtestgen.py
+++ b/cxxtest/cxxtestgen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 '''Usage: %s [OPTIONS] <input file(s)>
 Generate test source file for CxxTest.
 
--- a/share/extensions/addnodes.py
+++ b/share/extensions/addnodes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 This extension either adds nodes to a path so that
     a) no segment is longer than a maximum value 
--- a/share/extensions/bezmisc.py
+++ b/share/extensions/bezmisc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2010 Nick Drobchenko, nick@cnc-club.ru
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
--- a/share/extensions/chardataeffect.py
+++ b/share/extensions/chardataeffect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Jos Hirth, kaioa.com
 Copyright (C) 2007 bulia byak
--- a/share/extensions/color_blackandwhite.py
+++ b/share/extensions/color_blackandwhite.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect,sys
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_brighter.py
+++ b/share/extensions/color_brighter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_custom.py
+++ b/share/extensions/color_custom.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_darker.py
+++ b/share/extensions/color_darker.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_desaturate.py
+++ b/share/extensions/color_desaturate.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/coloreffect.py
+++ b/share/extensions/coloreffect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Jos Hirth, kaioa.com
 Copyright (C) 2007 Aaron C. Spike
--- a/share/extensions/color_grayscale.py
+++ b/share/extensions/color_grayscale.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_HSL_adjust.py
+++ b/share/extensions/color_HSL_adjust.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # standard library
 import random
--- a/share/extensions/color_lesshue.py
+++ b/share/extensions/color_lesshue.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_lesslight.py
+++ b/share/extensions/color_lesslight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_lesssaturation.py
+++ b/share/extensions/color_lesssaturation.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_morehue.py
+++ b/share/extensions/color_morehue.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_morelight.py
+++ b/share/extensions/color_morelight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_moresaturation.py
+++ b/share/extensions/color_moresaturation.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect, inkex
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_negative.py
+++ b/share/extensions/color_negative.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_randomize.py
+++ b/share/extensions/color_randomize.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import random
 
--- a/share/extensions/color_removeblue.py
+++ b/share/extensions/color_removeblue.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_removegreen.py
+++ b/share/extensions/color_removegreen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_removered.py
+++ b/share/extensions/color_removered.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/color_replace.py
+++ b/share/extensions/color_replace.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 import inkex
--- a/share/extensions/color_rgbbarrel.py
+++ b/share/extensions/color_rgbbarrel.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import coloreffect
 
 class C(coloreffect.ColorEffect):
--- a/share/extensions/convert2dashes.py
+++ b/share/extensions/convert2dashes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 This extension converts a path into a dashed line using 'stroke-dasharray'
 It is a modification of the file addnodes.py
--- a/share/extensions/cspsubdiv.py
+++ b/share/extensions/cspsubdiv.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 from bezmisc import *
 from ffgeom import *
 
--- a/share/extensions/cubicsuperpath.py
+++ b/share/extensions/cubicsuperpath.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 cubicsuperpath.py
 
--- a/share/extensions/dimension.py
+++ b/share/extensions/dimension.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 dimension.py
 An Inkscape effect for adding CAD style dimensions to selected objects
--- a/share/extensions/dm2svg.py
+++ b/share/extensions/dm2svg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 dm2svg.py - import a DHW file from ACECAD DigiMemo
 
--- a/share/extensions/dots.py
+++ b/share/extensions/dots.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/dpiswitcher.py
+++ b/share/extensions/dpiswitcher.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 This extension scales a document to fit different SVG DPI -90/96-
 
--- a/share/extensions/draw_from_triangle.py
+++ b/share/extensions/draw_from_triangle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Beard john.j.beard@gmail.com
 
--- a/share/extensions/dxf_input.py
+++ b/share/extensions/dxf_input.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 dxf_input.py - input a DXF file >= (AutoCAD Release 13 == AC1012)
 
--- a/share/extensions/dxf_outlines.py
+++ b/share/extensions/dxf_outlines.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005,2007,2008 Aaron Spike, aaron@ekips.org
 Copyright (C) 2008,2010 Alvin Penner, penner@vaxxine.com
--- a/share/extensions/dxf_templates.py
+++ b/share/extensions/dxf_templates.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 r14_header = '''  0
 SECTION
   2
--- a/share/extensions/edge3d.py
+++ b/share/extensions/edge3d.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 Terry Brown, terry_n_brown@yahoo.com
 
--- a/share/extensions/embedimage.py
+++ b/share/extensions/embedimage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005,2007 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/empty_business_card.py
+++ b/share/extensions/empty_business_card.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/empty_desktop.py
+++ b/share/extensions/empty_desktop.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/empty_dvd_cover.py
+++ b/share/extensions/empty_dvd_cover.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/empty_generic.py
+++ b/share/extensions/empty_generic.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/empty_icon.py
+++ b/share/extensions/empty_icon.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/empty_page.py
+++ b/share/extensions/empty_page.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Rewritten by Tavmjong Bah to add correct viewBox, inkscape:cx, etc. attributes
 
--- a/share/extensions/empty_video.py
+++ b/share/extensions/empty_video.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Tavmjong Bah
 
--- a/share/extensions/eqtexsvg.py
+++ b/share/extensions/eqtexsvg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: cp1252 -*-
 """
 eqtexsvg.py
--- a/share/extensions/export_gimp_palette.py
+++ b/share/extensions/export_gimp_palette.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Author: Jos Hirth, kaioa.com
 License: GNU General Public License - http://www.gnu.org/licenses/gpl.html
--- a/share/extensions/extractimage.py
+++ b/share/extensions/extractimage.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/extrude.py
+++ b/share/extensions/extrude.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007
 
--- a/share/extensions/ffgeom.py
+++ b/share/extensions/ffgeom.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
     ffgeom.py
     Copyright (C) 2005 Aaron Cyril Spike, aaron@ekips.org
--- a/share/extensions/fig2dev-ext.py
+++ b/share/extensions/fig2dev-ext.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 fig2dev-ext.py
--- a/share/extensions/flatten.py
+++ b/share/extensions/flatten.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/foldablebox.py
+++ b/share/extensions/foldablebox.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 '''
 Copyright (C) 2009 Aurelio A. Heckert <aurium (a) gmail dot com>
 
--- a/share/extensions/fractalize.py
+++ b/share/extensions/fractalize.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Carsten Goetze c.goetze@tu-bs.de
 
--- a/share/extensions/funcplot.py
+++ b/share/extensions/funcplot.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 Tavmjong Bah, tavmjong@free.fr
 Copyright (C) 2006 Georg Wiora, xorx@quarkbox.de
--- a/share/extensions/gcodetools.py
+++ b/share/extensions/gcodetools.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 """
 Comments starting "#LT" or "#CLT" are by Chris Lusby Taylor who rewrote the engraving function in 2011.
 History of CLT changes to engraving and other functions it uses:
--- a/share/extensions/generate_voronoi.py
+++ b/share/extensions/generate_voronoi.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2010 Alvin Penner, penner@vaxxine.com
 
--- a/share/extensions/gimp_xcf.py
+++ b/share/extensions/gimp_xcf.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Aaron Spike, aaron@ekips.org
 Copyright (C) 2010-2012 Nicolas Dufour, nicoduf@yahoo.fr
--- a/share/extensions/grid_cartesian.py
+++ b/share/extensions/grid_cartesian.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Beard john.j.beard@gmail.com
 
--- a/share/extensions/grid_isometric.py
+++ b/share/extensions/grid_isometric.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 
 #Copyright (C) 2010 Jean-Luc JOULIN "JeanJouX" jean-luc.joulin@laposte.net
 
--- a/share/extensions/grid_polar.py
+++ b/share/extensions/grid_polar.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Beard john.j.beard@gmail.com
 
--- a/share/extensions/guides_creator.py
+++ b/share/extensions/guides_creator.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Guides Creator v2.31 (05/07/2009)
 http://code.google.com/p/inkscape-guides-creator/
--- a/share/extensions/guillotine.py
+++ b/share/extensions/guillotine.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 guillotine.py
 
--- a/share/extensions/handles.py
+++ b/share/extensions/handles.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/hershey.py
+++ b/share/extensions/hershey.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 Hershey Text - renders a line of text using "Hershey" fonts for plotters
--- a/share/extensions/hpgl_input.py
+++ b/share/extensions/hpgl_input.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # coding=utf-8
 '''
 Copyright (C) 2013 Sebastian Wüst, sebi@timewaster.de
--- a/share/extensions/hpgl_output.py
+++ b/share/extensions/hpgl_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # coding=utf-8
 '''
 Copyright (C) 2013 Sebastian Wüst, sebi@timewaster.de
--- a/share/extensions/image_attributes.py
+++ b/share/extensions/image_attributes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 image_attributes.py - adjust image attributes which don't have global
 GUI options yet
--- a/share/extensions/ink2canvas.py
+++ b/share/extensions/ink2canvas.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2011 Karlisson Bezerra <contact@hacktoon.com>
 
--- a/share/extensions/inkex.py
+++ b/share/extensions/inkex.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 inkex.py
--- a/share/extensions/inkscape_follow_link.py
+++ b/share/extensions/inkscape_follow_link.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import threading
 import webbrowser
--- a/share/extensions/inkwebeffect.py
+++ b/share/extensions/inkwebeffect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2009 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/interp_att_g.py
+++ b/share/extensions/interp_att_g.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2009 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/interp.py
+++ b/share/extensions/interp.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/jessyInk_autoTexts.py
+++ b/share/extensions/jessyInk_autoTexts.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_effects.py
+++ b/share/extensions/jessyInk_effects.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_export.py
+++ b/share/extensions/jessyInk_export.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_install.py
+++ b/share/extensions/jessyInk_install.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_keyBindings.py
+++ b/share/extensions/jessyInk_keyBindings.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_masterSlide.py
+++ b/share/extensions/jessyInk_masterSlide.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_mouseHandler.py
+++ b/share/extensions/jessyInk_mouseHandler.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_summary.py
+++ b/share/extensions/jessyInk_summary.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_transitions.py
+++ b/share/extensions/jessyInk_transitions.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_uninstall.py
+++ b/share/extensions/jessyInk_uninstall.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_video.py
+++ b/share/extensions/jessyInk_video.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jessyInk_view.py
+++ b/share/extensions/jessyInk_view.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2008, 2009 Hannes Hochreiner
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
--- a/share/extensions/jitternodes.py
+++ b/share/extensions/jitternodes.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2012 Juan Pablo Carbajal ajuanpi-dev@gmail.com
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
--- a/share/extensions/launch_webbrowser.py
+++ b/share/extensions/launch_webbrowser.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # standard library
 import webbrowser
 import threading
--- a/share/extensions/layers2svgfont.py
+++ b/share/extensions/layers2svgfont.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches <juca@members.fsf.org>
 
--- a/share/extensions/layout_nup.py
+++ b/share/extensions/layout_nup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 Terry Brown, terry_n_brown@yahoo.com
 
--- a/share/extensions/lindenmayer.py
+++ b/share/extensions/lindenmayer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/lorem_ipsum.py
+++ b/share/extensions/lorem_ipsum.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Jos Hirth, kaioa.com
 
--- a/share/extensions/markers_strokepaint.py
+++ b/share/extensions/markers_strokepaint.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2006 Aaron Spike, aaron@ekips.org
 Copyright (C) 2010 Nicolas Dufour, nicoduf@yahoo.fr (color options)
--- a/share/extensions/measure.py
+++ b/share/extensions/measure.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 This extension module can measure arbitrary path and object length
 It adds text to the selected path containing the length in a given unit.
--- a/share/extensions/merge_styles.py
+++ b/share/extensions/merge_styles.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright (C) 2014 Martin Owens
 #
--- a/share/extensions/motion.py
+++ b/share/extensions/motion.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/new_glyph_layer.py
+++ b/share/extensions/new_glyph_layer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches
 
--- a/share/extensions/next_glyph_layer.py
+++ b/share/extensions/next_glyph_layer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches
 
--- a/share/extensions/nicechart.py
+++ b/share/extensions/nicechart.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #  nicechart.py
 #
--- a/share/extensions/param_curves.py
+++ b/share/extensions/param_curves.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2009 Michel Chatelain.
 Copyright (C) 2007 Tavmjong Bah, tavmjong@free.fr
--- a/share/extensions/pathalongpath.py
+++ b/share/extensions/pathalongpath.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2006 Jean-Francois Barraud, barraud@math.univ-lille1.fr
 
--- a/share/extensions/pathmodifier.py
+++ b/share/extensions/pathmodifier.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2006 Jean-Francois Barraud, barraud@math.univ-lille1.fr
 
--- a/share/extensions/pathscatter.py
+++ b/share/extensions/pathscatter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2006 Jean-Francois Barraud, barraud@math.univ-lille1.fr
 
--- a/share/extensions/perfectboundcover.py
+++ b/share/extensions/perfectboundcover.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Bintz, jcoswell@cosellproductions.org
 
--- a/share/extensions/perspective.py
+++ b/share/extensions/perspective.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/pixelsnap.py
+++ b/share/extensions/pixelsnap.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 TODO: This only snaps selected elements, and if those elements are part of a
--- a/share/extensions/plotter.py
+++ b/share/extensions/plotter.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # coding=utf-8
 '''
 Copyright (C) 2013 Sebastian Wüst, sebi@timewaster.de
--- a/share/extensions/plt_output.py
+++ b/share/extensions/plt_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 plt_output.py
--- a/share/extensions/polyhedron_3d.py
+++ b/share/extensions/polyhedron_3d.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Beard john.j.beard@gmail.com
 
--- a/share/extensions/previous_glyph_layer.py
+++ b/share/extensions/previous_glyph_layer.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches
 
--- a/share/extensions/printing_marks.py
+++ b/share/extensions/printing_marks.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 This extension allows you to draw crop, registration and other
 printing marks in Inkscape.
--- a/share/extensions/print_win32_vector.py
+++ b/share/extensions/print_win32_vector.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 print_win32_vector.py
 This extension will generate vector graphics printout, specifically for Windows GDI32.
--- a/share/extensions/ps2pdf-ext.py
+++ b/share/extensions/ps2pdf-ext.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 ps2pdf-ext.py
--- a/share/extensions/pturtle.py
+++ b/share/extensions/pturtle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/render_alphabetsoup_config.py
+++ b/share/extensions/render_alphabetsoup_config.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Syntax format: (raise your hand if you know lisp :-)
 #
 # 'state0':  ("file.svg", ( ( ('state1', dx, dy, T-B, L|R),),
--- a/share/extensions/render_alphabetsoup.py
+++ b/share/extensions/render_alphabetsoup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2001-2002 Matt Chisholm matt@theory.org
 Copyright (C) 2008 Joel Holdsworth joel@airwebreathe.org.uk
--- a/share/extensions/render_barcode_datamatrix.py
+++ b/share/extensions/render_barcode_datamatrix.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 # -*- coding: UTF-8 -*-
 '''
 Copyright (C) 2009 John Beard john.j.beard@gmail.com
--- a/share/extensions/render_barcode.py
+++ b/share/extensions/render_barcode.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright (C) 2007 Martin Owens
 #
--- a/share/extensions/render_barcode_qrcode.py
+++ b/share/extensions/render_barcode_qrcode.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import math, sys
 import inkex
--- a/share/extensions/render_gear_rack.py
+++ b/share/extensions/render_gear_rack.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2013 Brett Graham (hahahaha @ hahaha.org)
 
--- a/share/extensions/render_gears.py
+++ b/share/extensions/render_gears.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2007 Aaron Spike  (aaron @ ekips.org)
 Copyright (C) 2007 Tavmjong Bah (tavmjong @ free.fr)
--- a/share/extensions/replace_font.py
+++ b/share/extensions/replace_font.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 replace_font.py
 
--- a/share/extensions/restack.py
+++ b/share/extensions/restack.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2007-2011 Rob Antonishen; rob.antonishen@gmail.com
 
--- a/share/extensions/rtree.py
+++ b/share/extensions/rtree.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 Copyright (C) 2015 su_v, suv-sf@users.sf.net
--- a/share/extensions/rubberstretch.py
+++ b/share/extensions/rubberstretch.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2006 Jean-Francois Barraud, barraud@math.univ-lille1.fr
 
--- a/share/extensions/run_command.py
+++ b/share/extensions/run_command.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import os
 import sys
 import tempfile
--- a/share/extensions/scour.inkscape.py
+++ b/share/extensions/scour.inkscape.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 import sys, platform, inkex
 
--- a/share/extensions/seamless_pattern.py
+++ b/share/extensions/seamless_pattern.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Written by Jabiertxof
 # V.06
--- a/share/extensions/setup_typography_canvas.py
+++ b/share/extensions/setup_typography_canvas.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches
 
--- a/share/extensions/sk1_output.py
+++ b/share/extensions/sk1_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 sk1_output.py
--- a/share/extensions/spirograph.py
+++ b/share/extensions/spirograph.py
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/env python3
 '''
 Copyright (C) 2007 Joel Holdsworth joel@airwebreathe.org.uk
 
--- a/share/extensions/split.py
+++ b/share/extensions/split.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2009 Karlisson Bezerra, contato@nerdson.com
 
--- a/share/extensions/straightseg.py
+++ b/share/extensions/straightseg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/summersnight.py
+++ b/share/extensions/summersnight.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/svg_and_media_zip_output.py
+++ b/share/extensions/svg_and_media_zip_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 svg_and_media_zip_output.py
 An extention which collects all images to the documents directory and
--- a/share/extensions/svgcalendar.py
+++ b/share/extensions/svgcalendar.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 '''
 calendar.py
--- a/share/extensions/svgfont2layers.py
+++ b/share/extensions/svgfont2layers.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2011 Felipe Correa da Silva Sanches <juca@members.fsf.org>
 
--- a/share/extensions/synfig_fileformat.py
+++ b/share/extensions/synfig_fileformat.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 synfig_fileformat.py
 Synfig file format utilities
--- a/share/extensions/synfig_output.py
+++ b/share/extensions/synfig_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 synfig_output.py
 An Inkscape extension for exporting Synfig files (.sif)
--- a/share/extensions/synfig_prepare.py
+++ b/share/extensions/synfig_prepare.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 synfig_prepare.py
 Simplifies SVG files in preparation for sif export.
--- a/share/extensions/tar_layers.py
+++ b/share/extensions/tar_layers.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright (C) 2014 Martin Owens, email@doctormo.org
 #
--- a/share/extensions/test/addnodes.test.py
+++ b/share/extensions/test/addnodes.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../addnodes.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/chardataeffect.test.py
+++ b/share/extensions/test/chardataeffect.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../chardataeffect.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/coloreffect.test.py
+++ b/share/extensions/test/coloreffect.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../coloreffect.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/color_randomize.test.py
+++ b/share/extensions/test/color_randomize.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../color_randomize.py
 --
--- a/share/extensions/test/dots.test.py
+++ b/share/extensions/test/dots.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../dots.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/draw_from_triangle.test.py
+++ b/share/extensions/test/draw_from_triangle.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../draw_from_triangle.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/dxf_outlines.test.py
+++ b/share/extensions/test/dxf_outlines.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../dxf_outlines.py
 Revision history:
--- a/share/extensions/test/edge3d.test.py
+++ b/share/extensions/test/edge3d.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../edge3d.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/embedimage.test.py
+++ b/share/extensions/test/embedimage.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../embedimage.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/eqtexsvg.test.py
+++ b/share/extensions/test/eqtexsvg.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../eqtexsvg.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/extractimage.test.py
+++ b/share/extensions/test/extractimage.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../extractimage.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/extrude.test.py
+++ b/share/extensions/test/extrude.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../extrude.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/flatten.test.py
+++ b/share/extensions/test/flatten.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../flatten.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/foldablebox.test.py
+++ b/share/extensions/test/foldablebox.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import sys
 sys.path.append('..') # this line allows to import the extension code
--- a/share/extensions/test/fractalize.test.py
+++ b/share/extensions/test/fractalize.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../fractalize.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/funcplot.test.py
+++ b/share/extensions/test/funcplot.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../funcplot.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/gimp_xcf.test.py
+++ b/share/extensions/test/gimp_xcf.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../gimp_xcf.py
 Revision history:
--- a/share/extensions/test/grid_cartesian.test.py
+++ b/share/extensions/test/grid_cartesian.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../grid_cartesian.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/grid_polar.test.py
+++ b/share/extensions/test/grid_polar.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../grid_polar.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/guides_creator.test.py
+++ b/share/extensions/test/guides_creator.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../guides_creator.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/handles.test.py
+++ b/share/extensions/test/handles.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../handles.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/hpgl_output.test.py
+++ b/share/extensions/test/hpgl_output.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../hpgl_output.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/inkex.test.py
+++ b/share/extensions/test/inkex.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 '''
 Unit test file for ../inkex.py
--- a/share/extensions/test/inkwebeffect.test.py
+++ b/share/extensions/test/inkwebeffect.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../inkwebeffect.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/interp_att_g.test.py
+++ b/share/extensions/test/interp_att_g.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../interp-att-g.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/interp.test.py
+++ b/share/extensions/test/interp.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../interp.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/jitternodes.test.py
+++ b/share/extensions/test/jitternodes.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../jitternodes.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/lindenmayer.test.py
+++ b/share/extensions/test/lindenmayer.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../lindenmayer.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/lorem_ipsum.test.py
+++ b/share/extensions/test/lorem_ipsum.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../lorem_ipsum.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/markers_strokepaint.test.py
+++ b/share/extensions/test/markers_strokepaint.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../markers_strokepaint.py
 Revision history:
--- a/share/extensions/test/measure.test.py
+++ b/share/extensions/test/measure.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../measure.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/motion.test.py
+++ b/share/extensions/test/motion.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../motion.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/pathmodifier.test.py
+++ b/share/extensions/test/pathmodifier.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../pathmodifier.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/perfectboundcover.test.py
+++ b/share/extensions/test/perfectboundcover.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../perfectboundcover.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/perspective.test.py
+++ b/share/extensions/test/perspective.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../perspective.py
 Revision history:
--- a/share/extensions/test/polyhedron_3d.test.py
+++ b/share/extensions/test/polyhedron_3d.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../polyhedron_3d.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/printing_marks.test.py
+++ b/share/extensions/test/printing_marks.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../printing-marks.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/render_alphabetsoup.test.py
+++ b/share/extensions/test/render_alphabetsoup.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../render_alphabetsoup.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/render_barcode.test.py
+++ b/share/extensions/test/render_barcode.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright (C) 2010 Martin Owens
 #
--- a/share/extensions/test/render_gears.test.py
+++ b/share/extensions/test/render_gears.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../gears.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/restack.test.py
+++ b/share/extensions/test/restack.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../restack.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/rtree.test.py
+++ b/share/extensions/test/rtree.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../rtree.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/simplestyle.test.py
+++ b/share/extensions/test/simplestyle.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import unittest, sys
 sys.path.append('..') # this line allows to import the extension code
--- a/share/extensions/test/spirograph.test.py
+++ b/share/extensions/test/spirograph.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../spirograph.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/straightseg.test.py
+++ b/share/extensions/test/straightseg.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../straightseg.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/summersnight.test.py
+++ b/share/extensions/test/summersnight.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Unit test file for ../summersnight.py
 Revision history:
--- a/share/extensions/test/svg_and_media_zip_output.test.py
+++ b/share/extensions/test/svg_and_media_zip_output.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../svg_and_media_zip_output.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/svgcalendar.test.py
+++ b/share/extensions/test/svgcalendar.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import sys
 sys.path.append('..') # this line allows to import the extension code
--- a/share/extensions/test/triangle.test.py
+++ b/share/extensions/test/triangle.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../triangle.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/test/whirl.test.py
+++ b/share/extensions/test/whirl.test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This is only the automatic generated test file for ../whirl.py
 # This must be filled with real tests and this commentary
--- a/share/extensions/text_braille.py
+++ b/share/extensions/text_braille.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 import chardataeffect, inkex, string
--- a/share/extensions/text_extract.py
+++ b/share/extensions/text_extract.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2011 Nicolas Dufour (jazzynico)
 Direction code from the Restack extension, by Rob Antonishen
--- a/share/extensions/text_flipcase.py
+++ b/share/extensions/text_flipcase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 class C(chardataeffect.CharDataEffect):
--- a/share/extensions/text_lowercase.py
+++ b/share/extensions/text_lowercase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 class C(chardataeffect.CharDataEffect):
--- a/share/extensions/text_merge.py
+++ b/share/extensions/text_merge.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 Copyright (C) 2013 Nicolas Dufour (jazzynico)
 Direction code from the Restack extension, by Rob Antonishen
--- a/share/extensions/text_randomcase.py
+++ b/share/extensions/text_randomcase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 import random
--- a/share/extensions/text_sentencecase.py
+++ b/share/extensions/text_sentencecase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 class C(chardataeffect.CharDataEffect):
--- a/share/extensions/text_titlecase.py
+++ b/share/extensions/text_titlecase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 class C(chardataeffect.CharDataEffect):
--- a/share/extensions/text_uppercase.py
+++ b/share/extensions/text_uppercase.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import chardataeffect, inkex, string
 
 class C(chardataeffect.CharDataEffect):
--- a/share/extensions/triangle.py
+++ b/share/extensions/triangle.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2007 John Beard john.j.beard@gmail.com
 
--- a/share/extensions/ungroup_deep.py
+++ b/share/extensions/ungroup_deep.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 see #inkscape on Freenode and
--- a/share/extensions/uniconv-ext.py
+++ b/share/extensions/uniconv-ext.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 uniconv-ext.py
@@ -58,7 +58,7 @@
                      'https://sk1project.net/modules.php?name=Products&product=uniconvertor&op=download\n'+\
                      'and install into your Inkscape\'s Python location\n'))
         sys.exit(1)
-    cmd = 'python -c "import uniconvertor; uniconvertor.uniconv_run()"'
+    cmd = 'python3 -c "import uniconvertor; uniconvertor.uniconv_run()"'
 
 run((cmd+' "%s" "%%s"') % sys.argv[1].replace("%","%%"), "UniConvertor")
 
--- a/share/extensions/uniconv_output.py
+++ b/share/extensions/uniconv_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 uniconv_output.py
@@ -125,7 +125,7 @@
                          'https://sk1project.net/modules.php?name=Products&product=uniconvertor&op=download\n'+\
                          'and install into your Inkscape\'s Python location\n'))
             sys.exit(1)
-        cmd = 'python -c "import uniconvertor; uniconvertor.uniconv_run();"'
+        cmd = 'python3 -c "import uniconvertor; uniconvertor.uniconv_run();"'
 
     return cmd
 
--- a/share/extensions/voronoi2svg.py
+++ b/share/extensions/voronoi2svg.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 
 voronoi2svg.py
--- a/share/extensions/voronoi.py
+++ b/share/extensions/voronoi.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #############################################################################
 #
 # Voronoi diagram calculator/ Delaunay triangulator
--- a/share/extensions/web-set-att.py
+++ b/share/extensions/web-set-att.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2009 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/webslicer_create_group.py
+++ b/share/extensions/webslicer_create_group.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2010 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/webslicer_create_rect.py
+++ b/share/extensions/webslicer_create_rect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2010 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/webslicer_effect.py
+++ b/share/extensions/webslicer_effect.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2010 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/webslicer_export.py
+++ b/share/extensions/webslicer_export.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2010 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/web-transmit-att.py
+++ b/share/extensions/web-transmit-att.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 Copyright (C) 2009 Aurelio A. Heckert, aurium (a) gmail dot com
 
--- a/share/extensions/whirl.py
+++ b/share/extensions/whirl.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 '''
 Copyright (C) 2005 Aaron Spike, aaron@ekips.org
 
--- a/share/extensions/wireframe_sphere.py
+++ b/share/extensions/wireframe_sphere.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python 
+#!/usr/bin/env python3 
 # -*- coding: UTF-8 -*-
 '''
 Copyright (C) 2009 John Beard john.j.beard@gmail.com
--- a/share/extensions/wmf_output.py
+++ b/share/extensions/wmf_output.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 """
 wmf_output.py
--- a/share/filters/i18n.py
+++ b/share/filters/i18n.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from xml.dom import minidom
 import sys
--- a/share/palettes/i18n.py
+++ b/share/palettes/i18n.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 import sys
 import glob
--- a/share/patterns/i18n.py
+++ b/share/patterns/i18n.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from xml.dom import minidom
 import sys
--- a/share/symbols/i18n.py
+++ b/share/symbols/i18n.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from xml.dom import minidom
 import sys
--- a/share/templates/i18n.py
+++ b/share/templates/i18n.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 from xml.dom import minidom
 import sys
--- a/src/extension/implementation/script.cpp
+++ b/src/extension/implementation/script.cpp
@@ -87,7 +87,7 @@
 #ifdef WIN32
         {"python", "python-interpreter", "pythonw" },
 #else
-        {"python", "python-interpreter", "python" },
+        {"python", "python-interpreter", "python3" },
 #endif
         {"ruby",   "ruby-interpreter",   "ruby"   },
         {"shell",  "shell-interpreter",  "sh"     },
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -569,11 +569,11 @@
     gchar const *path = g_getenv("PATH");
     gchar const *pythonpath = g_getenv("PYTHONPATH");
 
-    gchar *python = g_build_filename(exe, "python", NULL);
-    gchar *scripts = g_build_filename(exe, "python", "Scripts", NULL);
-    gchar *perl = g_build_filename(exe, "python", NULL);
-    gchar *pythonlib = g_build_filename(exe, "python", "Lib", NULL);
-    gchar *pythondll = g_build_filename(exe, "python", "DLLs", NULL);
+    gchar *python = g_build_filename(exe, "python3", NULL);
+    gchar *scripts = g_build_filename(exe, "python3", "Scripts", NULL);
+    gchar *perl = g_build_filename(exe, "python3", NULL);
+    gchar *pythonlib = g_build_filename(exe, "python3", "Lib", NULL);
+    gchar *pythondll = g_build_filename(exe, "python3", "DLLs", NULL);
 
     // Python 2.x needs short paths in PYTHONPATH.
     // Otherwise it doesn't work when Inkscape is installed in Unicode directories.
@@ -605,7 +605,7 @@
     printf("PATH = %s\n\n", g_getenv("PATH"));
     printf("PYTHONPATH = %s\n\n", g_getenv("PYTHONPATH"));
 
-    gchar *p = g_find_program_in_path("python");
+    gchar *p = g_find_program_in_path("python3");
     if (p) {
         printf("python in %s\n\n", p);
         g_free(p);
