diff --git a/src/extension/internal/pdfinput/pdf-parser.cpp b/src/extension/internal/pdfinput/pdf-parser.cpp
index d6e2ede..131c20a 100644
--- a/src/extension/internal/pdfinput/pdf-parser.cpp
+++ b/src/extension/internal/pdfinput/pdf-parser.cpp
@@ -685,19 +685,19 @@ void PdfParser::opConcat(Object args[], int /*numArgs*/)
 // TODO not good that numArgs is ignored but args[] is used:
 void PdfParser::opSetDash(Object args[], int /*numArgs*/)
 {
-  double *dash = nullptr;
+  std::vector<double> dash;
 
   Array *a = args[0].getArray();
   int length = a->getLength();
   if (length != 0) {
-    dash = (double *)gmallocn(length, sizeof(double));
+    dash.reserve(length);
     for (int i = 0; i < length; ++i) {
       Object obj;
       dash[i] = _POPPLER_CALL_ARGS_DEREF(obj, a->get, i).getNum();
       _POPPLER_FREE(obj);
     }
   }
-  state->setLineDash(dash, length, args[1].getNum());
+  state->setLineDash(std::move(dash), args[1].getNum());
   builder->updateStyle(state);
 }
 
diff --git a/src/extension/internal/pdfinput/svg-builder.cpp b/src/extension/internal/pdfinput/svg-builder.cpp
index 901cc8c..1ba8ae5 100644
--- a/src/extension/internal/pdfinput/svg-builder.cpp
+++ b/src/extension/internal/pdfinput/svg-builder.cpp
@@ -388,10 +388,9 @@ void SvgBuilder::_setStrokeStyle(SPCSSAttr *css, GfxState *state) {
     sp_repr_css_set_property(css, "stroke-miterlimit", os_ml.str().c_str());
 
     // Line dash
-    double *dash_pattern;
-    int dash_length;
     double dash_start;
-    state->getLineDash(&dash_pattern, &dash_length, &dash_start);
+    const std::vector<double> &dash_pattern = state->getLineDash(&dash_start);
+    auto dash_length = dash_pattern.size();
     if ( dash_length > 0 ) {
         Inkscape::CSSOStringStream os_array;
         for ( int i = 0 ; i < dash_length ; i++ ) {
