From c0e11c43d28d63eae768c57d58b11800d15a84d6 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 17 Apr 2026 14:12:30 +0200
Subject: [PATCH] fix(cmake): add the -qt6 suffix to the lib name in the
 pkgconfig file

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 47c8aee..38b1b30 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -44,7 +44,7 @@ install(FILES ${PUBLIC_HEADER_FILES}
 
 include(ECMGeneratePkgConfigFile)
 ecm_generate_pkgconfig_file(
-    BASE_NAME qofonoext
+    BASE_NAME qofonoext${QTVERSION_SUFFIX}
     INCLUDE_INSTALL_DIR ${CMAKE_INSTALL_INCLUDEDIR}/qofonoext${QTVERSION_SUFFIX}
     URL https://github.com/sailfishos/libqofonoext
     DESCRIPTION "Qt bindings for ofono extensions"
-- 
2.53.0

