diff --color -Naur NetHack-3.6.7/include-orig/system.h NetHack-3.6.7/include/system.h
--- NetHack-3.6.7/include-orig/system.h	2025-01-22 20:05:17.954360996 -0600
+++ NetHack-3.6.7/include/system.h	2025-01-22 20:05:52.440085109 -0600
@@ -60,7 +60,7 @@
 #endif
 #ifndef SIG_RET_TYPE
 #if defined(NHSTDC) || defined(POSIX_TYPES) || defined(OS2) || defined(__DECC)
-#define SIG_RET_TYPE void (*)()
+#define SIG_RET_TYPE void (*)(int)
 #endif
 #endif
 #ifndef SIG_RET_TYPE
@@ -96,7 +96,7 @@
 E void FDECL(srand48, (long));
 #else
 E long lrand48();
-E void srand48();
+E void srand48(long);
 #endif /* MACOSX */
 #endif /* BSD || ULTRIX || RANDOM */
 
@@ -355,7 +355,7 @@
 E void sleep();
 #endif
 #if defined(ULTRIX) || defined(SYSV)
-E unsigned sleep();
+E unsigned sleep(unsigned int);
 #endif
 #if defined(HPUX)
 E unsigned int FDECL(sleep, (unsigned int));
@@ -519,7 +519,8 @@
 #else
 #if !(defined(HPUX) && defined(_POSIX_SOURCE))
 E int FDECL(tgetent, (char *, const char *));
-E void FDECL(tputs, (const char *, int, int (*)()));
+//E void FDECL(tputs, (const char *, int, int (*)()));
+E void FDECL(tputs, (const char *, int, int (*)(int)));
 #endif
 E int FDECL(tgetnum, (const char *));
 E int FDECL(tgetflag, (const char *));
diff --color -Naur NetHack-3.6.7/include-orig/winX.h NetHack-3.6.7/include/winX.h
--- NetHack-3.6.7/include-orig/winX.h	2025-01-22 20:05:17.956361038 -0600
+++ NetHack-3.6.7/include/winX.h	2025-01-22 20:05:52.442085151 -0600
@@ -279,7 +279,8 @@
 } AppResources;
 
 E AppResources appResources;
-E void (*input_func)();
+//E void (*input_func)();
+E void (*input_func)(Widget, XEvent *, String *, Cardinal *);
 
 extern struct window_procs X11_procs;
 
diff --color -Naur NetHack-3.6.7/include-orig/xwindow.h NetHack-3.6.7/include/xwindow.h
--- NetHack-3.6.7/include-orig/xwindow.h	2025-01-22 20:05:17.957361059 -0600
+++ NetHack-3.6.7/include/xwindow.h	2025-01-22 20:05:52.442085151 -0600
@@ -76,8 +76,8 @@
 #define XtNexposeCallback "exposeCallback"
 #define XtNresizeCallback "resizeCallback"
 
-extern XFontStruct *WindowFontStruct(/* Widget */);
-extern Font WindowFont(/* Widget */);
+extern XFontStruct *WindowFontStruct(Widget w);
+extern Font WindowFont(Widget w);
 
 #define XtCWindowResource "WindowResource"
 #define XtCRows "Rows"
