diff --git a/src/dsp_conv.c b/src/dsp_conv.c
index 6c6878c..e141b99 100644
--- a/src/dsp_conv.c
+++ b/src/dsp_conv.c
@@ -660,11 +660,6 @@ CS_RETCODE dsp_datetime_conv( ctx, dt_fmt, dt, buf, len, type )
      * type of date that we are processing and replace the ms
      * field if it exists.
      */
-#if defined(CS_BIGDATETIME_TYPE) && defined(CS_BIGTIME_TYPE)
-    if (dt_fmt->datatype == CS_BIGDATETIME_TYPE || dt_fmt->datatype == CS_BIGTIME_TYPE)
-        fmt = dsp_datetime_strip( dt_fmt->datatype, conv_fmt, (int) dr.datesecfrac );
-    else
-#endif
         fmt = dsp_datetime_strip( dt_fmt->datatype, conv_fmt, (int) dr.datemsecond );
 
     /*
