We need -D_XOPEN_SOURCE_EXTENDED for widechar in ncurses

diff --git a/stfl_internals.h b/stfl_internals.h
index 3f9f45b..1559626 100644
--- a/stfl_internals.h
+++ b/stfl_internals.h
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #include "stfl.h"
-#include <ncursesw/ncurses.h>
+#include <ncurses.h>
 #include <pthread.h>
 
 struct stfl_widget_type;
--- ./Makefile.orig
+++ ./Makefile
@@ -21,7 +21,7 @@
 include Makefile.cfg
 
 export CC = gcc -pthread
-export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -fPIC
+export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -D_XOPEN_SOURCE_EXTENDED -fPIC
 export LDLIBS += -lncursesw
 
 SONAME  := libstfl.so.0
