diff --git a/src/cc/libbpf.c b/src/cc/libbpf.c
index 974dc272..c74d9472 100644
--- a/src/cc/libbpf.c
+++ b/src/cc/libbpf.c
@@ -26,6 +26,7 @@
 #include <linux/bpf_common.h>
 #include <linux/if_packet.h>
 #include <linux/types.h>
+#include <linux/stddef.h>
 #include <linux/perf_event.h>
 #include <linux/pkt_cls.h>
 #include <linux/rtnetlink.h>
diff --git a/src/cc/libbpf/src/libbpf.c b/src/cc/libbpf/src/libbpf.c
index 514b1a5..30c2d3e 100644
--- a/src/cc/libbpf/src/libbpf.c
+++ b/src/cc/libbpf/src/libbpf.c
@@ -26,6 +26,7 @@
 #include <errno.h>
 #include <asm/unistd.h>
 #include <linux/err.h>
+#include <linux/stddef.h>
 #include <linux/kernel.h>
 #include <linux/bpf.h>
 #include <linux/btf.h>
diff --git a/src/cc/perf_reader.c b/src/cc/perf_reader.c
index dedb11d2..5d8a7e3a 100644
--- a/src/cc/perf_reader.c
+++ b/src/cc/perf_reader.c
@@ -26,6 +26,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <linux/types.h>
+#include <linux/stddef.h>
 #include <linux/perf_event.h>

 #include "libbpf.h"
diff --git a/src/cc/api/BPF.cc b/src/cc/api/BPF.cc
index a3a14fee..1eff49ab 100644
--- a/src/cc/api/BPF.cc
+++ b/src/cc/api/BPF.cc
@@ -15,6 +15,7 @@
  */

 #include <linux/bpf.h>
+#include <linux/stddef.h>
 #include <linux/perf_event.h>
 #include <unistd.h>
 #include <cstdio>
diff --git a/src/cc/api/BPFTable.cc b/src/cc/api/BPFTable.cc
index db62de74..0a2b4d04 100644
--- a/src/cc/api/BPFTable.cc
+++ b/src/cc/api/BPFTable.cc
@@ -16,6 +16,7 @@

 #include <fcntl.h>
 #include <linux/elf.h>
+#include <linux/stddef.h>
 #include <linux/perf_event.h>
 #include <sys/epoll.h>
 #include <unistd.h>
