diff --git a/src/cc/CMakeLists.txt b/src/cc/CMakeLists.txt
index c4675dee..2bb97395 100644
--- a/src/cc/CMakeLists.txt
+++ b/src/cc/CMakeLists.txt
@@ -145,6 +145,7 @@ if(LIBBPF_FOUND)
 endif()
 
 install(TARGETS bcc-shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS bcc-static LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
 install(FILES ${bcc_table_headers} DESTINATION include/bcc)
 install(FILES ${bcc_api_headers} DESTINATION include/bcc)
 install(DIRECTORY ${libbpf_uapi} DESTINATION include/bcc/compat/linux FILES_MATCHING PATTERN "*.h")
@@ -152,3 +153,5 @@ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libbcc.pc DESTINATION ${CMAKE_INSTALL_
 endif(ENABLE_CLANG_JIT)
 install(FILES ${bcc_common_headers} DESTINATION include/bcc)
 install(TARGETS bccbpf-shared LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS bccbpf-static LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS bcc-loader-static LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR})
