--- userland-d4aa617de3b196399bb8e2ce32e181768cb52179.orig/host_applications/linux/apps/raspicam/RaspiVidYUV.c
+++ userland-d4aa617de3b196399bb8e2ce32e181768cb52179/host_applications/linux/apps/raspicam/RaspiVidYUV.c
@@ -106,8 +106,6 @@
 /// Run/record forever
 #define WAIT_METHOD_FOREVER        4
 
-extern FILE *stderr, *stdout;
-
 int mmal_status_to_int(MMAL_STATUS_T status);
 static void signal_handler(int signal_number);
 
--- userland-d4aa617de3b196399bb8e2ce32e181768cb52179.orig/host_applications/linux/libs/debug_sym/debug_sym.c
+++ userland-d4aa617de3b196399bb8e2ce32e181768cb52179/host_applications/linux/libs/debug_sym/debug_sym.c
@@ -67,6 +67,8 @@
 # else
 #  define PAGE_SIZE   4096
 # endif
+#endif
+#ifndef PAGE_MASK
 #define PAGE_MASK   (~(PAGE_SIZE - 1))
 #endif
 
