--- a/libmailutils/mailer/progmailer.c
+++ b/libmailutils/mailer/progmailer.c
@@ -45,7 +45,7 @@ struct _mu_progmailer
 {
   int fd;
   pid_t pid;
-  void (*sighandler)();
+  void (*sighandler)(int);
   char *command;
 };
 
--- a/mh/scan.c
+++ b/mh/scan.c
@@ -200,7 +200,7 @@ clear_screen (void)
 	      char *clr = tgetstr ("cl", &buffer);
 	      if (clr)
 		{
-		  tputs(clr, 1, (int (*)())putstdout);
+		  tputs(clr, 1, (int (*)(int))putstdout);
 		  return;
 		}
 	    }
