diff --git a/configure b/configure
index 9bff5d5..cc5d2ce 100755
--- a/configure
+++ b/configure
@@ -266,7 +266,7 @@ fi
 #
 # Check for some compiler warning flags.
 #
-for f in all extra error shadow "format=2" missing-prototypes \
+for f in all extra shadow "format=2" missing-prototypes \
 	missing-declarations nested-externs vla no-overlength-strings \
 	unsafe-loop-optimizations undef sign-compare \
 	missing-include-dirs old-style-definition \
