diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05bb06b..bbaff7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,8 @@
-project(Applauncherd)
+cmake_minimum_required(VERSION 3.6.0)
+cmake_policy(VERSION 3.6.0)
 
-cmake_minimum_required(VERSION 2.6)
-cmake_policy(VERSION 2.6)
+project(Applauncherd
+	VERSION 4.1.40)
 
 include(FeatureSummary)
 include(GNUInstallDirs)
diff --git a/src/launcherlib/CMakeLists.txt b/src/launcherlib/CMakeLists.txt
index a704668..0fb3665 100644
--- a/src/launcherlib/CMakeLists.txt
+++ b/src/launcherlib/CMakeLists.txt
@@ -1,5 +1,3 @@
-include(GNUInstallDirs)
-
 set(COMMON ${CMAKE_HOME_DIRECTORY}/src/common)
 
 # Find systemd
@@ -21,12 +19,23 @@ set(HEADERS appdata.h booster.h connection.h daemon.h logger.h launcherlib.h
 
 # Set executable
 add_library(applauncherd MODULE ${SRC} ${MOC_SRC})
-set_target_properties(applauncherd PROPERTIES VERSION 0.1 SOVERSION 0)
-
 target_link_libraries(applauncherd ${SYSTEMD_LIBRARIES})
 
+set_target_properties(applauncherd PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR})
+
+# Configure and install the pkgconfig file
+configure_file(applauncherd.pc.in
+	${CMAKE_CURRENT_BINARY_DIR}/applauncherd.pc
+	@ONLY)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/applauncherd.pc
+	DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/pkgconfig)
 
 # Add install rule
-install(TARGETS applauncherd DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
-install(FILES ${HEADERS} DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/applauncherd
-  PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
+install(TARGETS applauncherd
+    LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR})
+install(FILES ${HEADERS}
+    DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}/applauncherd
+    COMPONENT Devel
+    PERMISSIONS OWNER_READ GROUP_READ WORLD_READ)
diff --git a/src/launcherlib/applauncherd.pc.in b/src/launcherlib/applauncherd.pc.in
new file mode 100644
index 0000000..2a8edad
--- /dev/null
+++ b/src/launcherlib/applauncherd.pc.in
@@ -0,0 +1,9 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@/applauncherd
+
+Name: applauncherd
+Version: @PROJECT_VERSION@
+Libs: -lapplauncherd
+Cflags: -I${includedir}
