--- a/oletools/oletimes.py
+++ b/oletools/oletimes.py
@@ -78,7 +78,7 @@
 
 import olefile
 from oletools.thirdparty import xglob
-from oletools.thirdparty.prettytable import prettytable
+from prettytable import prettytable
 
 
 # === FUNCTIONS ==============================================================
--- a/oletools/pyxswf.py
+++ b/oletools/pyxswf.py
@@ -78,7 +78,7 @@
 import optparse, sys, os
 from . import rtfobj
 from io import BytesIO
-from .thirdparty.xxxswf import xxxswf
+from xxxswf import xxxswf
 import olefile
 
 
--- a/setup.py
+++ b/setup.py
@@ -94,8 +94,6 @@
     "oletools.common",
     "oletools.common.log_helper",
     'oletools.thirdparty',
-    'oletools.thirdparty.xxxswf',
-    'oletools.thirdparty.prettytable',
     'oletools.thirdparty.xglob',
     'oletools.thirdparty.tablestream',
     'oletools.thirdparty.oledump',
@@ -176,12 +174,6 @@
 
     'oletools.thirdparty.xglob': [
         'LICENSE.txt',
-        ],
-    'oletools.thirdparty.xxxswf': [
-        'LICENSE.txt',
-        ],
-    'oletools.thirdparty.prettytable': [
-        'CHANGELOG', 'COPYING', 'README'
         ],
     'oletools.thirdparty.DridexUrlDecoder': [
         'LICENSE.txt',
