--- screen-4.0.3-orig/acls.h	Mon Sep  8 14:23:38 2003
+++ screen-4.0.3/acls.h	Fri Feb 27 18:32:07 2009
@@ -20,6 +20,8 @@
  *
  ****************************************************************
  */
+#ifndef _SCREEN_ACLS_H_
+#define _SCREEN_ACLS_H_
 
 #ifdef MULTIUSER
 
@@ -91,3 +93,4 @@
 
 extern int DefaultEsc, DefaultMetaEsc;
 
+#endif /* define _SCREEN_ACLS_H_ */
--- screen-4.0.3-orig/ansi.h	Mon Feb 24 16:55:08 2003
+++ screen-4.0.3/ansi.h	Fri Feb 27 18:32:49 2009
@@ -21,6 +21,8 @@
  ****************************************************************
  * $Id: ansi.h,v 1.6 1994/05/31 12:31:28 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_ANSI_H_
+#define _SCREEN_ANSI_H_
 
 #define NATTR		6
 
@@ -168,3 +170,5 @@
 # define dw_left(ml, x, enc) 0
 # define dw_right(ml, x, enc) 0
 #endif
+
+#endif /* define _SCREEN_ANSI_H_ */
--- screen-4.0.3-orig/braille.h	Sat Apr 28 13:26:42 2001
+++ screen-4.0.3/braille.h	Fri Feb 27 18:33:51 2009
@@ -24,6 +24,8 @@
  ****************************************************************
  * $Id: braille.h,v 1.1 1995/09/06 15:51:18 jnweiger Exp jnweiger $ FAU
  */
+#ifndef _SCREEN_BRAILLE_H_
+#define _SCREEN_BRAILLE_H_
 
 #ifdef HAVE_BRAILLE
 
@@ -81,3 +83,5 @@
 #define BD_FORE bd.bd_dpy->d_fore
 
 #endif
+
+#endif /* define _SCREEN_BRAILLE_H_ */
--- screen-4.0.3-orig/display.h	Tue Jul  1 14:01:42 2003
+++ screen-4.0.3/display.h	Fri Feb 27 18:34:30 2009
@@ -21,6 +21,8 @@
  ****************************************************************
  * $Id: display.h,v 1.9 1994/05/31 12:31:54 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_DISPLAY_H_
+#define _SCREEN_DISPLAY_H_
 
 #ifdef MAPKEYS
 
@@ -343,3 +345,5 @@
 #define HSTATUS_MESSAGE		2
 #define HSTATUS_HS		3
 #define HSTATUS_ALWAYS		(1<<2)
+
+#endif /* define _SCREEN_DISPLAY_H_ */
--- screen-4.0.3-orig/extern.h	Fri Aug 22 12:27:57 2003
+++ screen-4.0.3/extern.h	Fri Feb 27 18:35:17 2009
@@ -21,6 +21,8 @@
  ****************************************************************
  * $Id: extern.h,v 1.18 1994/05/31 12:31:57 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_EXTERN_H_
+#define _SCREEN_EXTERN_H_
 
 #if !defined(__GNUC__) || __GNUC__ < 2
 #undef __attribute__
@@ -490,3 +492,5 @@
 # endif
 #endif
 extern int   EncodeChar __P((char *, int, int, int *));
+
+#endif /* define _SCREEN_EXTERN_H_ */
--- screen-4.0.3-orig/image.h	Fri Dec  5 13:45:41 2003
+++ screen-4.0.3/image.h	Fri Feb 27 18:35:53 2009
@@ -21,8 +21,9 @@
  ****************************************************************
  * $Id: image.h,v 1.9 1994/05/31 12:31:54 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_IMAGE_H_
+#define _SCREEN_IMAGE_H_
 
-
 #undef IFFONT
 #undef IFCOLOR
 
@@ -161,3 +162,5 @@
 #  define cole2i(c) ((c) ^ 9)
 # endif
 #endif
+
+#endif /* define _SCREEN_IMAGE_H_ */
--- screen-4.0.3-orig/layer.h	Tue Jan  8 15:42:25 2002
+++ screen-4.0.3/layer.h	Fri Feb 27 18:36:50 2009
@@ -22,6 +22,9 @@
  * $Id: overlay.h,v 1.3 1994/05/31 12:32:31 mlschroe Exp $ FAU
  */
 
+#ifndef _SCREEN_LAYER_H_
+#define _SCREEN_LAYER_H_
+
 /*
  * This is the overlay structure. It is used to create a seperate
  * layer over the current windows.
@@ -103,3 +106,4 @@
 	  display = olddisplay;				\
 	} while(0)
 
+#endif /* define _SCREEN_LAYER_H_ */
--- screen-4.0.3-orig/logfile.h	Tue Jan  8 15:42:27 2002
+++ screen-4.0.3/logfile.h	Fri Feb 27 18:37:29 2009
@@ -22,6 +22,9 @@
  * $Id: logfile.h,v 1.11 1994/05/31 12:33:27 jnweiger Exp $ FAU
  */
 
+#ifndef _SCREEN_LOGFILE_H_
+#define _SCREEN_LOGFILE_H_
+
 struct logfile
 {
   struct logfile *next;
@@ -80,3 +83,5 @@
  * It closes fd and opens wantfd to access whatever fd accessed.
  */
 int lf_move_fd __P((int fd, int wantfd));
+
+#endif /* define _SCREEN_LOGFILE_H_ */
--- screen-4.0.3-orig/mark.h	Tue Jan  8 15:42:30 2002
+++ screen-4.0.3/mark.h	Fri Feb 27 18:38:02 2009
@@ -21,6 +21,8 @@
  ****************************************************************
  * $Id: mark.h,v 1.1.1.1 1993/06/16 23:51:13 jnweiger Exp $ FAU
  */
+#ifndef _SCREEN_MARK_H_
+#define _SCREEN_MARK_H_
 
 struct markdata
 {
@@ -47,3 +49,4 @@
 #define W2D(y) ((y) - markdata->hist_offset)
 #define D2W(y) ((y) + markdata->hist_offset)
 
+#endif /* define _SCREEN_MARK_H_ */
--- screen-4.0.3-orig/os.h	Tue Jan  8 15:42:33 2002
+++ screen-4.0.3/os.h	Fri Feb 27 18:39:09 2009
@@ -21,6 +21,8 @@
  ****************************************************************
  * $Id: os.h,v 1.10 1994/05/31 12:32:22 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_OS_H_
+#define _SCREEN_OS_H_
 
 #include <stdio.h>
 #include <errno.h>
@@ -531,3 +533,4 @@
  */
 #define IOSIZE		4096
 
+#endif /* define _SCREEN_OS_H_ */
--- screen-4.0.3-orig/osdef.h.in	Sat Apr 28 13:26:43 2001
+++ screen-4.0.3/osdef.h.in	Fri Feb 27 18:38:36 2009
@@ -27,6 +27,8 @@
  * mangled the screen source through 'gcc -Wall'.
  ****************************************************************
  */
+#ifndef _SCREEN_OSDEF_H_
+#define _SCREEN_OSDEF_H_
 
 extern int   printf __P((char *, ...));
 extern int   fprintf __P((FILE *, char *, ...));
@@ -199,3 +201,4 @@
 extern int getloadavg(double *, int);
 #endif
 
+#endif /* define _SCREEN_OSDEF_H_ */
--- screen-4.0.3-orig/patchlevel.h	Mon Oct 23 13:04:11 2006
+++ screen-4.0.3/patchlevel.h	Fri Feb 27 18:40:53 2009
@@ -523,6 +523,8 @@
  *                     (cstone & Rich Felker).
  *                     -- DISTRIBUTED
  */
+#ifndef _SCREEN_PATCHLEVEL_H_
+#define _SCREEN_PATCHLEVEL_H_
 
 #define ORIGIN "FAU"
 #define REV 4
@@ -530,3 +532,5 @@
 #define PATCHLEVEL 3
 #define DATE "23-Oct-06"
 #define STATE ""
+
+#endif /* define _SCREEN_PATCHLEVEL_H_ */
--- screen-4.0.3-orig/pty.c	Mon Sep  8 14:26:18 2003
+++ screen-4.0.3/pty.c	Fri Feb 27 19:31:53 2009
@@ -33,11 +33,6 @@
 # include <sys/ioctl.h>
 #endif
 
-/* for solaris 2.1, Unixware (SVR4.2) and possibly others */
-#ifdef HAVE_SVR4_PTYS
-# include <sys/stropts.h>
-#endif
-
 #if defined(sun) && defined(LOCKPTY) && !defined(TIOCEXCL)
 # include <sys/ttold.h>
 #endif
--- screen-4.0.3-orig/sched.h	Tue Jan  8 15:42:43 2002
+++ screen-4.0.3/sched.h	Fri Feb 27 19:39:49 2009
@@ -21,6 +21,9 @@
  ****************************************************************
  * $Id: sched.h,v 1.1.1.1 1993/06/16 23:51:13 jnweiger Exp $ FAU
  */
+#ifndef _SCREEN_SCHED_H_
+#define _SCREEN_SCHED_H_
+#include <sys/types.h>
 
 struct event
 {
@@ -41,3 +44,5 @@
 #define EV_READ		1
 #define EV_WRITE	2
 #define EV_ALWAYS	3
+
+#endif /* define _SCREEN_SCHED_H_ */
--- screen-4.0.3-orig/screen.h	Fri Aug 22 12:28:43 2003
+++ screen-4.0.3/screen.h	Fri Feb 27 18:42:11 2009
@@ -22,6 +22,9 @@
  * $Id: screen.h,v 1.12 1994/05/31 12:32:54 mlschroe Exp $ FAU
  */
 
+#ifndef _SCREEN_SCREEN_H_
+#define _SCREEN_SCREEN_H_
+
 #include "os.h"
 
 #if defined(__STDC__)
@@ -293,3 +296,5 @@
  */
 #define WLIST_NUM 0
 #define WLIST_MRU 1
+
+#endif /* define _SCREEN_SCREEN_H_ */
--- screen-4.0.3-orig/window.h	Thu Aug 21 14:57:30 2003
+++ screen-4.0.3/window.h	Fri Feb 27 18:42:45 2009
@@ -21,8 +21,9 @@
  ****************************************************************
  * $Id: window.h,v 1.11 1994/05/31 12:33:27 mlschroe Exp $ FAU
  */
+#ifndef _SCREEN_WINDOW_H_
+#define _SCREEN_WINDOW_H_
 
-
 /* keep this in sync with the initialisations in window.c */
 struct NewWindow
 {
@@ -315,3 +316,5 @@
     : &fore->w_mlines[y - fore->w_histheight])
 
 #define Layer2Window(l) ((struct win *)(l)->l_bottom->l_data)
+
+#endif /* define _SCREEN_WINDOW_H_ */
