diff -upr ktoblzcheck-1.55.orig/src/python/test_ktoblzcheck.py ktoblzcheck-1.55/src/python/test_ktoblzcheck.py
--- ktoblzcheck-1.55.orig/src/python/test_ktoblzcheck.py	2024-01-03 21:14:37.299488606 +0100
+++ ktoblzcheck-1.55/src/python/test_ktoblzcheck.py	2024-01-03 21:14:47.899508730 +0100
@@ -16,7 +16,7 @@ class TestAccountNumberCheck(unittest.Te
         self.assertIsInstance(res, ktoblzcheck.Record)
         if isinstance(res, ktoblzcheck.Record):
             self.assertEqual(res.bankId.decode('ascii'), '20010020')
-            self.assertEqual(res.bankName.decode('ascii'), 'Postbank Ndl der Deutsche Bank')
+            self.assertEqual(res.bankName.decode('ascii'), 'Postbank Ndl der DB Privat- und Firmenkundenbank')
             self.assertEqual(res.location.rstrip().decode('ascii'), 'Hamburg')
 
     def test_findBank_fail(self):
Only in ktoblzcheck-1.55/src/python: test_ktoblzcheck.py.orig
