--- /dev/null
+++ ./pyproject.toml.in
@@ -0,0 +1,17 @@
+[build-system]
+requires = ["sip >=6, <7"]
+build-backend = "sipbuild.api"
+
+[tool.sip.metadata]
+name = "pyArcus"
+
+[tool.sip.project]
+builder-factory = "CMakeBuilder"
+sip-files-dir = "${CMAKE_CURRENT_SOURCE_DIR}/python/"
+sip-include-dirs = ["CMAKE_CURRENT_SOURCE_DIR/python/"]
+build-dir = "${CMAKE_CURRENT_BINARY_DIR}/pyArcus/"
+
+[tool.sip.bindings.pyArcus]
+exceptions = true
+release-gil = true
+concatenate = 8
