--- a/mcabber/screen.c
+++ b/mcabber/screen.c
@@ -90,7 +90,7 @@ static void scr_glog_print(const gchar *log_domain, GLogLevelFlags log_level,
                            const gchar *message, gpointer user_data);
 
 #ifdef XEP0085
-static gboolean scr_chatstates_timeout();
+static gboolean scr_chatstates_timeout(void *);
 #endif
 
 #if defined(WITH_ENCHANT) || defined(WITH_ASPELL)
@@ -2485,7 +2485,7 @@ static void set_chatstate(int state)
 }
 
 #ifdef XEP0085
-static gboolean scr_chatstates_timeout(void)
+static gboolean scr_chatstates_timeout(void *)
 {
   time_t now;
   time(&now);
--- a/mcabber/xmpp.c
+++ b/mcabber/xmpp.c
@@ -50,7 +50,7 @@ LmConnection* lconnection = NULL;
 static guint AutoConnection;
 
 inline void update_last_use(void);
-inline gboolean xmpp_reconnect();
+inline gboolean xmpp_reconnect(void *);
 
 enum imstatus mystatus = offline;
 static enum imstatus mywantedstatus = available;
@@ -781,7 +781,7 @@ static void connection_auth_cb(LmConnection *connection, gboolean success,
     scr_LogPrint(LPRINT_LOGNORM, "Authentication failed");
 }
 
-gboolean xmpp_reconnect()
+gboolean xmpp_reconnect(void *)
 {
   if (!lconnection)
     xmpp_connect();
