--- a/META.json
+++ b/META.json
@@ -49,12 +49,11 @@
             "MIME::QuotedPrint" : "0"
          },
          "requires" : {
-            "Filter::signatures" : "0.13",
             "Future" : "0.31",
             "HTTP::Headers" : "6.07",
             "HTTP::Tiny" : "0",
             "Moo" : "2",
-            "perl" : "5.006"
+            "perl" : "5.022"
          }
       },
       "test" : {
--- a/META.yml
+++ b/META.yml
@@ -28,12 +28,11 @@
   MIME::Base64: '0'
   MIME::QuotedPrint: '0'
 requires:
-  Filter::signatures: '0.13'
   Future: '0.31'
   HTTP::Headers: '6.07'
   HTTP::Tiny: '0'
   Moo: '2'
-  perl: '5.006'
+  perl: '5.022'
 resources:
   license: https://dev.perl.org/licenses/
   repository: git://github.com/Corion/Future-HTTP.git
--- a/Makefile.PL
+++ b/Makefile.PL
@@ -69,7 +69,6 @@
     },
 
     PREREQ_PM => {
-        'Filter::signatures'         => '0.13', # only needed for Perl <5.22
         'Moo'                        => 2, # yeah, not all that great, but ...
         'Future'                     => '0.31', # Future 0.25 is incompatible as it doesn't allow Future->done
         'HTTP::Tiny'                 => 0, # our fallback, even distributed with Perl 5.14+
--- a/lib/Future/HTTP.pm
+++ b/lib/Future/HTTP.pm
@@ -1,6 +1,5 @@
 package Future::HTTP;
 use strict;
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/API/HTTPTiny.pm
+++ b/lib/Future/HTTP/API/HTTPTiny.pm
@@ -1,7 +1,6 @@
 package # hide from indexer as it's not really ready
     Future::HTTP::API::HTTPTiny;
 use strict;
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 require HTTP::Tiny;
--- a/lib/Future/HTTP/AnyEvent.pm
+++ b/lib/Future/HTTP/AnyEvent.pm
@@ -4,7 +4,6 @@
 use AnyEvent::HTTP ();
 use AnyEvent::Future;
 use Moo 2; # or Moo::Lax if you can't have Moo v2
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/Handler.pm
+++ b/lib/Future/HTTP/Handler.pm
@@ -1,6 +1,5 @@
 package Future::HTTP::Handler;
 use Moo::Role;
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/Mojo.pm
+++ b/lib/Future/HTTP/Mojo.pm
@@ -3,7 +3,6 @@
 use Future::Mojo;
 use Mojo::UserAgent;
 use Moo 2; # or Moo::Lax if you can't have Moo v2
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/NetAsync.pm
+++ b/lib/Future/HTTP/NetAsync.pm
@@ -2,7 +2,6 @@
 use strict;
 use Net::Async::HTTP;
 use Moo 2; # or Moo::Lax if you can't have Moo v2
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/Tiny.pm
+++ b/lib/Future/HTTP/Tiny.pm
@@ -3,7 +3,6 @@
 use Future;
 use HTTP::Tiny;
 use Moo 2; # or Moo::Lax if you can't have Moo v2
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
--- a/lib/Future/HTTP/Tiny/Paranoid.pm
+++ b/lib/Future/HTTP/Tiny/Paranoid.pm
@@ -3,7 +3,6 @@
 use Future;
 use HTTP::Tiny::Paranoid 0.07; # 0.04 had spurious CPAN tester failures
 use Moo 2; # or Moo::Lax if you can't have Moo v2
-use Filter::signatures;
 no warnings 'experimental::signatures';
 use feature 'signatures';
 
