Upstream: No
Reason: We use BUILD_TYPE=None

diff --git a/ngraph/CMakeLists.txt b/ngraph/CMakeLists.txt
index 356ce18..1c4984f 100644
--- a/ngraph/CMakeLists.txt
+++ b/ngraph/CMakeLists.txt
@@ -23,14 +23,6 @@ if("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
     message(FATAL_ERROR "In-source builds are not allowed.")
 endif()
 
-if (CMAKE_BUILD_TYPE)
-    set(RELEASE_TYPES Debug Release RelWithDebInfo MinSizeRel)
-    list(FIND RELEASE_TYPES ${CMAKE_BUILD_TYPE} INDEX_FOUND)
-    if (${INDEX_FOUND} EQUAL -1)
-        message(FATAL_ERROR "CMAKE_BUILD_TYPE must be one of Debug, Release, RelWithDebInfo, or MinSizeRel")
-    endif()
-endif()
-
 include(git_tags)
 include(var_functions)
 
