From 24e932ce16d99d384be19d43626bca03a1f21459 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Thu, 1 Jul 2021 15:34:07 +0200
Subject: [PATCH] tinfo

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef88317..e733978 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -525,7 +525,7 @@ else()
         else()
             find_package(Curses REQUIRED)
             find_package(ZLIB REQUIRED)
-            find_library(NCURSES_TINFO_LIBRARY tinfo)
+            find_library(NCURSES_TINFO_LIBRARY ncursesw)
             target_link_libraries(${PROJECT_NAME} pthread ${ZLIB_LIBRARIES} ${NCURSES_TINFO_LIBRARY} ${CURSES_LIBRARIES})
         endif()
     endif()
-- 
2.32.0

