Fix tests for logs by adding pkg namespace

--- a/pkg/logs/aws_test.go
+++ b/pkg/logs/aws_test.go
@@ -23 +23 @@ func TestNewAWSWrapper(t *testing.T) {
-	assert.Equal(t, "<nil> DBG aws_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/logs/aws_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/datadog_test.go
+++ b/pkg/logs/datadog_test.go
@@ -23 +23 @@ func TestNewDatadogLogger(t *testing.T) {
-	assert.Equal(t, "<nil> DBG datadog_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/logs/datadog_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/elastic_test.go
+++ b/pkg/logs/elastic_test.go
@@ -23 +23 @@ func TestNewElasticLogger(t *testing.T) {
-	assert.Equal(t, "<nil> ERR elastic_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> ERR github.com/traefik/traefik/v3/pkg/logs/elastic_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/gokit_test.go
+++ b/pkg/logs/gokit_test.go
@@ -23 +23 @@ func TestNewGoKitWrapper(t *testing.T) {
-	assert.Equal(t, "<nil> DBG gokit_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/logs/gokit_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/hclog_test.go
+++ b/pkg/logs/hclog_test.go
@@ -23 +23 @@ func TestNewRetryableHTTPLogger(t *testing.T) {
-	assert.Equal(t, "<nil> INF hclog_test.go:21 > Foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/logs/hclog_test.go:21 > Foo\n", buf.String())
--- a/pkg/logs/instana_test.go
+++ b/pkg/logs/instana_test.go
@@ -23 +23 @@ func TestNewInstanaLogger(t *testing.T) {
-	assert.Equal(t, "<nil> INF instana_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/logs/instana_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/log_test.go
+++ b/pkg/logs/log_test.go
@@ -23 +23 @@ func TestNoLevel(t *testing.T) {
-	assert.Equal(t, "<nil> INF log_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/logs/log_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/logrus_test.go
+++ b/pkg/logs/logrus_test.go
@@ -23 +23 @@ func TestNewLogrusStdWrapper(t *testing.T) {
-	assert.Equal(t, "<nil> DBG logrus_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/logs/logrus_test.go:21 > foo\n", buf.String())
--- a/pkg/logs/oxy_test.go
+++ b/pkg/logs/oxy_test.go
@@ -23 +23 @@ func TestNewOxyWrapper(t *testing.T) {
-	assert.Equal(t, "<nil> INF oxy_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/logs/oxy_test.go:21 > foo\n", buf.String())
