diff --git a/pkg/observability/logs/aws_test.go b/pkg/observability/logs/aws_test.go
index 36a970d..b315f8b 100644
--- a/pkg/observability/logs/aws_test.go
+++ b/pkg/observability/logs/aws_test.go
@@ -21,5 +21,5 @@ func TestNewAWSWrapper(t *testing.T) {
 
 	logger.Logf(logging.Debug, "%s", "foo")
 
-	assert.Equal(t, "<nil> DBG aws_test.go:22 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/observability/logs/aws_test.go:22 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/datadog_test.go b/pkg/observability/logs/datadog_test.go
index 41c6df4..47eadad 100644
--- a/pkg/observability/logs/datadog_test.go
+++ b/pkg/observability/logs/datadog_test.go
@@ -20,5 +20,5 @@ func TestNewDatadogLogger(t *testing.T) {
 
 	logger.Log("foo")
 
-	assert.Equal(t, "<nil> DBG datadog_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/observability/logs/datadog_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/elastic_test.go b/pkg/observability/logs/elastic_test.go
index e00604d..16e1304 100644
--- a/pkg/observability/logs/elastic_test.go
+++ b/pkg/observability/logs/elastic_test.go
@@ -20,5 +20,5 @@ func TestNewElasticLogger(t *testing.T) {
 
 	logger.Errorf("foo")
 
-	assert.Equal(t, "<nil> ERR elastic_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> ERR github.com/traefik/traefik/v3/pkg/observability/logs/elastic_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/gokit_test.go b/pkg/observability/logs/gokit_test.go
index 791157b..7531651 100644
--- a/pkg/observability/logs/gokit_test.go
+++ b/pkg/observability/logs/gokit_test.go
@@ -20,5 +20,5 @@ func TestNewGoKitWrapper(t *testing.T) {
 
 	_ = logger.Log("foo")
 
-	assert.Equal(t, "<nil> DBG gokit_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/observability/logs/gokit_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/hclog_test.go b/pkg/observability/logs/hclog_test.go
index 02681d7..50fe5e2 100644
--- a/pkg/observability/logs/hclog_test.go
+++ b/pkg/observability/logs/hclog_test.go
@@ -20,5 +20,5 @@ func TestNewRetryableHTTPLogger(t *testing.T) {
 
 	logger.Info("foo")
 
-	assert.Equal(t, "<nil> INF hclog_test.go:21 > Foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/observability/logs/hclog_test.go:21 > Foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/instana_test.go b/pkg/observability/logs/instana_test.go
index 38b53f2..dc758f0 100644
--- a/pkg/observability/logs/instana_test.go
+++ b/pkg/observability/logs/instana_test.go
@@ -20,5 +20,5 @@ func TestNewInstanaLogger(t *testing.T) {
 
 	logger.Info("foo")
 
-	assert.Equal(t, "<nil> INF instana_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/observability/logs/instana_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/log_test.go b/pkg/observability/logs/log_test.go
index 5c80e76..26fd44a 100644
--- a/pkg/observability/logs/log_test.go
+++ b/pkg/observability/logs/log_test.go
@@ -20,5 +20,5 @@ func TestNoLevel(t *testing.T) {
 
 	logger.Info().Msg("foo")
 
-	assert.Equal(t, "<nil> INF log_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/observability/logs/log_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/logrus_test.go b/pkg/observability/logs/logrus_test.go
index af3c085..d30889a 100644
--- a/pkg/observability/logs/logrus_test.go
+++ b/pkg/observability/logs/logrus_test.go
@@ -20,5 +20,5 @@ func TestNewLogrusStdWrapper(t *testing.T) {
 
 	logger.Println("foo")
 
-	assert.Equal(t, "<nil> DBG logrus_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> DBG github.com/traefik/traefik/v3/pkg/observability/logs/logrus_test.go:21 > foo\n", buf.String())
 }
diff --git a/pkg/observability/logs/oxy_test.go b/pkg/observability/logs/oxy_test.go
index 5645986..26f915c 100644
--- a/pkg/observability/logs/oxy_test.go
+++ b/pkg/observability/logs/oxy_test.go
@@ -20,5 +20,5 @@ func TestNewOxyWrapper(t *testing.T) {
 
 	logger.Info("foo")
 
-	assert.Equal(t, "<nil> INF oxy_test.go:21 > foo\n", buf.String())
+	assert.Equal(t, "<nil> INF github.com/traefik/traefik/v3/pkg/observability/logs/oxy_test.go:21 > foo\n", buf.String())
 }
