From a09298e2638282e8529c554704a2493006af379f Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Fri, 22 Oct 2010 16:54:29 +0000
Subject: Fix previous commit

---
diff --git a/giscanner/dumper.py b/giscanner/dumper.py
index 7462aa3..e44f4f4 100644
--- a/giscanner/dumper.py
+++ b/giscanner/dumper.py
@@ -200,7 +200,7 @@ class DumpCompiler(object):
             args.append('--tag=CC')
             args.append('--silent')
 
-        args = self._linker_cmd.split()
+        args.extend(self._linker_cmd.split())
         args.extend(['-o', output])
         if libtool:
             args.append('-export-dynamic')
--
cgit v0.8.3.1
