--- a/dj_database_url/__init__.py
+++ b/dj_database_url/__init__.py
@@ -3,7 +3,7 @@
 import urllib.parse as urlparse
 from typing import Any, Dict, Optional, Union
 
-from typing_extensions import TypedDict
+from typing import TypedDict
 
 DEFAULT_ENV = "DATABASE_URL"
 
--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@
     long_description=readme,
     long_description_content_type="text/x-rst",
     packages=["dj_database_url"],
-    install_requires=["Django>=4.2", "typing_extensions>=4.0.0"],
+    install_requires=["Django>=4.2"],
     include_package_data=True,
     package_data={
         "dj_database_url": ["py.typed"],
