diff -upr st-0.6.orig/config.mk st-0.6/config.mk
--- st-0.6.orig/config.mk	2016-07-11 00:41:11.428451510 +0200
+++ st-0.6/config.mk	2016-07-11 00:41:49.134970777 +0200
@@ -20,8 +20,8 @@ LIBS = -L/usr/lib -lc -L${X11LIB} -lm -l
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -D_XOPEN_SOURCE=600
-CFLAGS += -g -std=c99 -pedantic -Wall -Wvariadic-macros -Os ${INCS} ${CPPFLAGS}
-LDFLAGS += -g ${LIBS}
+CFLAGS += -std=c99 -pedantic -Wall -Wvariadic-macros ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 
 # compiler and linker
 # CC = cc
