--- a/tests/journey/gix.sh
+++ b/tests/journey/gix.sh
@@ -205,7 +205,7 @@
             )
           )
         )
-        (on_ci
+        (CI= on_ci
           if test "$kind" = "max"; then
           (with "https:// protocol"
             (with "version 1"
@@ -584,7 +584,7 @@
           }
         )
         fi
-        (on_ci
+        (CI= on_ci
           if [[ "$kind" = "max" ]]; then
           (with "https:// protocol"
             (with "version 1"
