See https://github.com/restic/restic/issues/3687

diff -upr restic-0.13.0.orig/internal/fs/setflags_linux_test.go restic-0.13.0/internal/fs/setflags_linux_test.go
--- restic-0.13.0.orig/internal/fs/setflags_linux_test.go	2022-03-29 21:31:21.121497170 +0200
+++ restic-0.13.0/internal/fs/setflags_linux_test.go	2022-03-29 21:31:35.918229947 +0200
@@ -59,7 +59,7 @@ func supportsNoatime(t *testing.T, f *os
 	err := unix.Fstatfs(int(f.Fd()), &fsinfo)
 	rtest.OK(t, err)
 
-	return fsinfo.Type == unix.BTRFS_SUPER_MAGIC ||
+	return int64(fsinfo.Type) == int64(unix.BTRFS_SUPER_MAGIC) ||
 		fsinfo.Type == unix.EXT2_SUPER_MAGIC ||
 		fsinfo.Type == unix.EXT3_SUPER_MAGIC ||
 		fsinfo.Type == unix.EXT4_SUPER_MAGIC ||
