diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90998c7..ea3060b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,13 +24,6 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/scripts/cmake)
 
 option(ADA_BENCHMARKS "Build benchmarks" OFF)
 
-# We use Google Benchmark, but it does not build under several 32-bit systems.
-if((BUILD_TESTING OR ADA_BENCHMARKS) AND (CMAKE_SIZEOF_VOID_P EQUAL 8))
-  include(${PROJECT_SOURCE_DIR}/cmake/import.cmake)
-  import_dependency(simdjson simdjson/simdjson 0a3a00c95665cc1fda760e121ba8d442945ede13)
-  add_dependency(simdjson)
-endif()
-
 if (BUILD_TESTING)
   message(STATUS "The tests are enabled.")
   add_subdirectory(tests)
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 8495977..443a79d 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -2,6 +2,5 @@ add_executable(adaparse adaparse.cpp)
 target_link_libraries(adaparse PRIVATE ada)
 target_include_directories(adaparse PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
 
-import_dependency(cxxopts jarro2783/cxxopts eb78730)
-add_dependency(cxxopts)
+find_package(cxxopts)
 target_link_libraries(adaparse PRIVATE cxxopts::cxxopts)
