diff --git a/CMakeLists.txt b/CMakeLists.txt
index 32f92a4..9124315 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,14 +24,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/scripts/cmake)
 
 option(ADA_BENCHMARKS "Build benchmarks" OFF)
 
-# We use Google Benchmark, but it does not build under several 32-bit systems.
-if((BUILD_TESTING OR ADA_BENCHMARKS) AND (CMAKE_SIZEOF_VOID_P EQUAL 8))
-  include(${PROJECT_SOURCE_DIR}/cmake/import.cmake)
-  import_dependency(simdjson simdjson/simdjson 14d927128ba14e13913fc0e7c2cf538790bd1622)
-  add_dependency(simdjson)
-  import_dependency(gtest google/googletest 797b0ad2a3a45608ecf5c67e6e289d377a3521ca)
-  add_dependency(gtest)
-endif()
+find_package(GTest)
 
 if (BUILD_TESTING)
   message(STATUS "The tests are enabled.")
diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index 4b99579..443a79d 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -2,7 +2,5 @@ add_executable(adaparse adaparse.cpp)
 target_link_libraries(adaparse PRIVATE ada)
 target_include_directories(adaparse PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
 
-include(${PROJECT_SOURCE_DIR}/cmake/import.cmake)
-import_dependency(cxxopts jarro2783/cxxopts eb78730)
-add_dependency(cxxopts)
+find_package(cxxopts)
 target_link_libraries(adaparse PRIVATE cxxopts::cxxopts)
