--- busybox-1.13.2/modutils/depmod.c.orig	2009-02-25 11:54:02.000000000 +0000
+++ busybox-1.13.2/modutils/depmod.c	2009-02-25 11:57:33.000000000 +0000
@@ -36,7 +36,8 @@
 	ARG_b = (1<<2), /* base directory when modules are in staging area */
 	ARG_e = (1<<3), /* with -F, print unresolved symbols */
 	ARG_F = (1<<4), /* System.map that contains the symbols */
-	ARG_n = (1<<5)  /* dry-run, print to stdout only */
+	ARG_n = (1<<5), /* dry-run, print to stdout only */
+	ARG_r = (1<<6)  /* Compat dummy. Linux Makefile uses it */
 };
 
 static int FAST_FUNC parse_module(const char *fname, struct stat *sb,
@@ -138,7 +139,7 @@
 	struct utsname uts;
 	int tmp;
 
-	getopt32(argv, "aAb:eF:n", &moddir_base, NULL);
+	getopt32(argv, "aAb:eF:nr", &moddir_base, NULL);
 	argv += optind;
 
 	/* goto modules location */
