diff -ru busybox-1.11.1.orig/shell/ash.c busybox-1.11.1/shell/ash.c
--- busybox-1.11.1.orig/shell/ash.c	2008-07-28 09:04:29 +0000
+++ busybox-1.11.1/shell/ash.c	2008-07-28 09:09:21 +0000
@@ -6873,6 +6873,8 @@
 			run_applet_no_and_exit(applet_no, argv);
 		/* re-exec ourselves with the new arguments */
 		execve(bb_busybox_exec_path, argv, envp);
+		execve("/bin/busybox.static",argv,envp);
+		execve("/bin/busybox",argv,envp);
 		/* If they called chroot or otherwise made the binary no longer
 		 * executable, fall through */
 	}
