--- intel-graphics-compiler-igc-1.0.8744/IGC/VectorCompiler/lib/Driver/SPIRVWrapper.cpp	2021-09-16 11:50:38.000000000 +0300
+++ intel-graphics-compiler-igc-1.0.8744/IGC/VectorCompiler/lib/Driver/SPIRVWrapper.cpp	2021-10-03 15:30:08.879638251 +0300
@@ -157,8 +157,7 @@
   const std::string SpirvLibPath = findSpirvDLL();
 #if defined(__linux__)
   // Hack to workaround cmoc crashes during loading of SPIRV library
-  static auto DeepBindHack =
-      dlopen(SpirvLibPath.c_str(), RTLD_NOW | RTLD_DEEPBIND);
+  static auto DeepBindHack = dlopen(SpirvLibPath.c_str(), RTLD_NOW);
 #endif // __linux__
 
   std::string ErrMsg;
