From 38cc56d08d243c886526d16bad3ee0b6571d6914 Mon Sep 17 00:00:00 2001
From: Andrea Florio <andrea@opensuse.org>
Date: Sun, 18 Sep 2011 08:49:36 +0200
Subject: [PATCH] fixed link with latest ld

---
 src/Makefile.am |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 4a750c5..0e2f058 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -27,7 +27,7 @@ gpicview_SOURCES = \
 
 # the library search path.
 # use -nodefaultlibs to prevent linking libstdc++
-gpicview_LDADD = $(X11_LIBS) $(GTK_LIBS)
+gpicview_LDADD = $(X11_LIBS) $(GTK_LIBS) -lm
 
 noinst_HEADERS = \
 	glib-mem.h \
-- 
1.7.4.1

