Patch-Source: https://marc.info/?l=mandoc-discuss&m=175200050325256&w=2

--- a/man_validate.c
+++ b/man_validate.c
@@ -481,7 +481,7 @@ post_TH(CHKARGS)
 	/* ->TITLE<- MSEC DATE OS VOL */
 
 	n = n->child;
-	if (n != NULL && n->string != NULL) {
+	if (n != NULL && n->string != NULL && *n->string != '\0') {
 		for (p = n->string; *p != '\0'; p++) {
 			/* Only warn about this once... */
 			if (isalpha((unsigned char)*p) &&
@@ -494,8 +494,8 @@ post_TH(CHKARGS)
 		}
 		man->meta.title = mandoc_strdup(n->string);
 	} else {
-		man->meta.title = mandoc_strdup("");
-		mandoc_msg(MANDOCERR_TH_NOTITLE, nb->line, nb->pos, "TH");
+		man->meta.title = mandoc_strdup("UNTITLED");
+		mandoc_msg(MANDOCERR_DT_NOTITLE, nb->line, nb->pos, "TH");
 	}
 
 	/* TITLE ->MSEC<- DATE OS VOL */
--- a/regress/man/TH/noarg.out_ascii
+++ b/regress/man/TH/noarg.out_ascii
@@ -1,4 +1,4 @@
-()                                                                          ()
+UNTITLED()                                                          UNTITLED()
 
 NNAAMMEE
        TH-noarg - no arguments to the TH macro
@@ -6,4 +6,4 @@ NNAAMMEE
 DDEESSCCRRIIPPTTIIOONN
        some text
 
-OpenBSD                                                                     ()
+OpenBSD                                                             UNTITLED()
--- a/regress/man/TH/noarg.out_lint
+++ b/regress/man/TH/noarg.out_lint
@@ -1,3 +1,3 @@
-mandoc: noarg.in:2:2: WARNING: missing manual title, using "": TH
-mandoc: noarg.in:2:2: WARNING: missing manual section, using "": TH 
+mandoc: noarg.in:2:2: WARNING: missing manual title, using UNTITLED: TH
+mandoc: noarg.in:2:2: WARNING: missing manual section, using "": TH UNTITLED
 mandoc: noarg.in:2:2: WARNING: missing date, using "": TH

