--- a/sudo
+++ b/sudo
@@ -13,6 +13,9 @@
 	This is not the original sudo, but the doas shim for sudo. It supports only
 	a subset of the sudo options that have an equivalent in doas, plus option -i
 	(--login). Long options are supported. Refer to sudo(1) for more information.
+
+	If you need some sudo features that are not supported in doas, replace
+	package 'doas-sudo-shim' with 'sudo': apk add sudo !doas-sudo-shim.
 	EOF
 }
 
