diff -rupN hidapi-hidapi-0.8.0-rc1-old/libusb/hid.c hidapi-hidapi-0.8.0-rc1/libusb/hid.c
--- hidapi-hidapi-0.8.0-rc1-old/libusb/hid.c	2019-01-23 10:46:10.388569561 +0100
+++ hidapi-hidapi-0.8.0-rc1/libusb/hid.c	2019-01-23 10:47:09.798295989 +0100
@@ -44,7 +44,7 @@
 #include <wchar.h>
 
 /* GNU / LibUSB */
-#include "libusb.h"
+#include "libusb-1.0/libusb.h"
 #include "iconv.h"
 
 #include "hidapi.h"
