https://github.com/microsoft/onnxruntime/issues/25815

diff -Nurp a/cmake/external/abseil-cpp.cmake b/cmake/external/abseil-cpp.cmake
--- a/cmake/external/abseil-cpp.cmake	2025-08-12 16:38:01.000000000 +0000
+++ b/cmake/external/abseil-cpp.cmake	2025-09-14 18:53:16.688779651 +0000
@@ -114,7 +114,6 @@ absl::absl_check
 absl::hash_function_defaults
 absl::function_ref
 absl::city
-absl::low_level_hash
 absl::fixed_array
 absl::variant
 absl::meta
