--- a/cmake/modules/LimitJobs.cmake
+++ b/cmake/modules/LimitJobs.cmake
@@ -11,8 +11,8 @@
 if(_num_cores LESS _avg_compile_jobs)
   set(_avg_compile_jobs ${_num_cores})
 endif()
-set(NINJA_MAX_COMPILE_JOBS "${_avg_compile_jobs}" CACHE STRING
-  "The maximum number of concurrent compilation jobs, for Ninja build system." FORCE)
+set(NINJA_MAX_COMPILE_JOBS "1" CACHE STRING
+  "The maximum number of concurrent compilation jobs, for Ninja build system.")
 mark_as_advanced(NINJA_MAX_COMPILE_JOBS)
 if(NINJA_MAX_COMPILE_JOBS)
   math(EXPR _heavy_compile_jobs "${_avg_compile_jobs} / 2")
@@ -21,7 +21,7 @@
   endif()
   set_property(GLOBAL APPEND PROPERTY JOB_POOLS
     avg_compile_job_pool=${NINJA_MAX_COMPILE_JOBS}
-    heavy_compile_job_pool=${_heavy_compile_jobs})
+    heavy_compile_job_pool=${NINJA_MAX_COMPILE_JOBS})
   set(CMAKE_JOB_POOL_COMPILE avg_compile_job_pool)
 endif()
 
