diff --git a/tools/jdk/BUILD.tools b/tools/jdk/BUILD.tools
index 0c37719bbc..83e1a5039b 100644
--- a/tools/jdk/BUILD.tools
+++ b/tools/jdk/BUILD.tools
@@ -193,7 +193,6 @@ alias(
 alias(
     name = "ijar_prebuilt_binary_or_cc_binary",
     actual = select({
-        "//src/conditions:linux_x86_64": ":ijar_prebuilt_binary_linux",
         "//src/conditions:darwin_x86_64": ":ijar_prebuilt_binary_darwin_x86_64",
         "//src/conditions:darwin_arm64": ":ijar_prebuilt_binary_darwin_arm64",
         "//src/conditions:windows": ":ijar_prebuilt_binary_windows",
@@ -225,7 +224,6 @@ alias(
 alias(
     name = "singlejar_prebuilt_or_cc_binary",
     actual = select({
-        "//src/conditions:linux_x86_64": ":prebuilt_singlejar_linux",
         "//src/conditions:darwin_x86_64": ":prebuilt_singlejar_darwin_x86_64",
         "//src/conditions:darwin_arm64": ":prebuilt_singlejar_darwin_arm64",
         "//src/conditions:windows": ":prebuilt_singlejar_windows",
