diff --git a/src/main/cpp/util/numbers.h b/src/main/cpp/util/numbers.h
index f1ef6f7991..62afeb65c2 100644
--- a/src/main/cpp/util/numbers.h
+++ b/src/main/cpp/util/numbers.h
@@ -15,6 +15,7 @@
 #define BAZEL_SRC_MAIN_CPP_UTIL_NUMBERS_H_
 
 #include <string>
+#include <cstdint>
 
 namespace blaze_util {
 
diff --git a/src/tools/singlejar/port.h b/src/tools/singlejar/port.h
index 1bf71fc16a..07e9028ead 100644
--- a/src/tools/singlejar/port.h
+++ b/src/tools/singlejar/port.h
@@ -32,7 +32,7 @@
 typedef off_t off64_t;
 #elif defined(_WIN32)
 typedef __int64 off64_t;
-#elif defined(__OpenBSD__)
+#else
 typedef int64_t off64_t;
 #endif
 static_assert(sizeof(off64_t) == 8, "File offset type must be 64-bit");
diff --git a/src/main/native/unix_jni_linux.cc b/src/main/native/unix_jni_linux.cc
index b32c1d33a1..1ee8c7e9d8 100644
--- a/src/main/native/unix_jni_linux.cc
+++ b/src/main/native/unix_jni_linux.cc
@@ -17,8 +17,9 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/xattr.h>
-
 #include <string>
+#define stat64 stat
+#define fstatat64 fstatat
 
 #include "src/main/native/unix_jni.h"

diff --git a/src/main/native/unix_jni.cc b/src/main/native/unix_jni.cc
index 1af7983b74..c29b385fdc 100644
--- a/src/main/native/unix_jni.cc
+++ b/src/main/native/unix_jni.cc
@@ -12,6 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#define lstat64 lstat
+#define stat64 stat
+#define fstatat64 fstatat
 #include "src/main/native/unix_jni.h"
 
 #include <dirent.h>

diff --git a/examples/py_native/test.py b/examples/py_native/test.py
index f9543aa727..113b11e098 100644
--- a/examples/py_native/test.py
+++ b/examples/py_native/test.py
@@ -8,10 +8,10 @@ from fib import Fib
 class TestGetNumber(unittest.TestCase):
 
   def test_ok(self):
-    self.assertEquals(GetNumber(), 42)
+    self.assertEqual(GetNumber(), 42)
 
   def test_fib(self):
-    self.assertEquals(Fib(5), 8)
+    self.assertEqual(Fib(5), 8)
 
 if __name__ == '__main__':
   unittest.main()
