From d1ea37c3c75599dff8c138756c6d38fe85394622 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Thu, 21 Aug 2025 01:40:39 +0200
Subject: [PATCH] Zig 0.15: Use .root_module in Build

---
 build.zig | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/build.zig b/build.zig
index 923eca5..50da559 100644
--- a/build.zig
+++ b/build.zig
@@ -24,9 +24,11 @@ pub fn build(b: *std.Build) void {
     const optimize = b.standardOptimizeOption(.{});
     const exe = b.addExecutable(.{
         .name = "mepo",
-        .root_source_file = b.path("src/main.zig"),
-        .target = target,
-        .optimize = optimize,
+        .root_module = b.createModule(.{
+            .root_source_file = b.path("src/main.zig"),
+            .target = target,
+            .optimize = optimize,
+        }),
     });
     setDependencies(exe);
     b.installArtifact(exe);
@@ -42,9 +44,11 @@ pub fn build(b: *std.Build) void {
 
     // Setup test
     const tests = b.addTest(.{
-        .root_source_file = b.path("./src/test.zig"),
-        .target = target,
-        .optimize = optimize,
+        .root_module = b.createModule(.{
+            .root_source_file = b.path("./src/test.zig"),
+            .target = target,
+            .optimize = optimize,
+        }),
     });
     setDependencies(tests);
 
