From 4d8e2345590bf1bba3ad16687d365f943158b79e Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sun, 9 Mar 2025 21:23:52 +0100
Subject: [PATCH] Zig 0.14: std.posix.sigaction does not error

---
 src/api/bind_signal.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/api/bind_signal.zig b/src/api/bind_signal.zig
index 9f70c00..34cfd2b 100644
--- a/src/api/bind_signal.zig
+++ b/src/api/bind_signal.zig
@@ -45,7 +45,7 @@ fn bind_signal(mepo: *Mepo, signo_str: [:0]const u8, expression: []const u8) !vo
         .flags = 0,
     };
 
-    std.posix.sigaction(signal_name, &signal_action, null) catch return error.FailedToSetupSighandler;
+    std.posix.sigaction(signal_name, &signal_action, null);
 
     if (mepo.table_signals.get(signal_name)) |heap_str| {
         mepo.allocator.free(heap_str);
