From b5c3f73141d8182b85a669ccf1b5b106d7645b87 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sun, 9 Mar 2025 21:33:03 +0100
Subject: [PATCH] Zig 0.14: std.mem.split -> std.mem.splitSequence

---
 src/api/clippaste.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/api/clippaste.zig b/src/api/clippaste.zig
index 70acf16..2ce3d63 100644
--- a/src/api/clippaste.zig
+++ b/src/api/clippaste.zig
@@ -23,7 +23,7 @@ fn execute(mepo: *Mepo, _: [types.MepoFnNargs]types.MepoArg) !void {
     // trim commas, and see if can parse as floats; if so assume lat/lon
     // if not - break out and run whatever is in clipboard as jsonapi
     parse_lat_lon: {
-        var it = std.mem.split(u8, text, " ");
+        var it = std.mem.splitSequence(u8, text, " ");
         var lat: ?f64 = null;
         var lon: ?f64 = null;
 
