From aa2159c285771788fb98d14c14f481b5bc6252cd Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Thu, 21 Aug 2025 03:51:13 +0200
Subject: [PATCH] Zig 0.15: Remove std.posix.empty_sigset

---
 src/api/bind_signal.zig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/api/bind_signal.zig b/src/api/bind_signal.zig
index 67c1dbc..defbcb4 100644
--- a/src/api/bind_signal.zig
+++ b/src/api/bind_signal.zig
@@ -39,9 +39,10 @@ fn bind_signal(mepo: *Mepo, signo_str: [:0]const u8, expression: []const u8) !vo
     }
 
     // Register generic signal handler
+    const sigset_t = @typeInfo(std.posix.sigset_t).array;
     const signal_action = std.posix.Sigaction{
         .handler = .{ .handler = utilsdl.sdl_push_event_signal },
-        .mask = std.posix.empty_sigset,
+        .mask = [_]sigset_t.child{0} ** sigset_t.len,
         .flags = 0,
     };
 
