From 6b7cff4e269e7c7ddba5d14dd13926e3c5870e07 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Thu, 21 Aug 2025 03:52:26 +0200
Subject: [PATCH] Zig 0.15: std.time.sleep -> std.Thread.sleep

---
 src/Downloader.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Downloader.zig b/src/Downloader.zig
index 327a294..d61fcd7 100644
--- a/src/Downloader.zig
+++ b/src/Downloader.zig
@@ -46,7 +46,7 @@ pub fn cache_dlbbox(tile_cache: *TileCache, download_request_str: []const u8) !v
             delay_seconds,
         },
     );
-    std.time.sleep(std.time.ns_per_s * delay_seconds);
+    std.Thread.sleep(std.time.ns_per_s * delay_seconds);
     _ = tile_cache.download_loop(false) catch unreachable;
 }
 
