diff --git a/src/kmk/kmkbuiltin/ftsfake.h b/src/kmk/kmkbuiltin/ftsfake.h
index 2518d30..5410256 100644
--- a/src/kmk/kmkbuiltin/ftsfake.h
+++ b/src/kmk/kmkbuiltin/ftsfake.h
@@ -38,6 +38,9 @@
 # include "kmkbuiltin/mscfakes.h"
 #endif
 
+/* for u_int */
+#include <sys/types.h>
+
 typedef struct {
 	struct _ftsent *fts_cur;	/* current node */
 	struct _ftsent *fts_child;	/* linked list of children */
