--- ./src/lib/k/kDefs.h.orig
+++ ./src/lib/k/kDefs.h
@@ -82,7 +82,7 @@
 #  define K_OS      K_OS_FREEBSD
 # elif defined(__HAIKU__)
 #  define K_OS      K_OS_HAIKU
-# elif defined(__gnu_linux__)
+# elif defined(__linux__)
 #  define K_OS      K_OS_LINUX
 # elif defined(__NetBSD__) /*??*/
 #  define K_OS      K_OS_NETBSD
--- ./src/kmk/kmkbuiltin/fts.c.orig
+++ ./src/kmk/kmkbuiltin/fts.c
@@ -75,7 +75,7 @@
 #endif
 
 #if ! HAVE_NBTOOL_CONFIG_H
-# if !defined(__sun__) && !defined(__gnu_linux__) && !defined(__HAIKU__)
+# if !defined(__sun__) && !defined(__linux__) && !defined(__HAIKU__)
 #  define HAVE_STRUCT_DIRENT_D_NAMLEN 1
 # endif
 #endif
