Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: set soversion of library
----

--- a/QB3lib/CMakeLists.txt
+++ b/QB3lib/CMakeLists.txt
@@ -32,6 +32,8 @@
     PUBLIC_HEADER "QB3.h;${CMAKE_CURRENT_BINARY_DIR}/libqb3_export.h"
     DEBUG_POSTFIX "d"
     PREFIX ""
+    VERSION ${PROJECT_VERSION} 
+    SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
 )
 
 include(GenerateExportHeader)
