Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: set destination of cmake files
----

--- a/QB3lib/CMakeLists.txt
+++ b/QB3lib/CMakeLists.txt
@@ -53,7 +53,7 @@
 install(EXPORT "${PROJECT_NAME}Targets"
     FILE "${PROJECT_NAME}Targets.cmake"
     NAMESPACE ${namespace}::
-    DESTINATION cmake
+    DESTINATION lib/cmake
 )
 
 # generate the version file for the config file
@@ -65,13 +65,13 @@
 # create config file
 configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
-    INSTALL_DESTINATION cmake
+    INSTALL_DESTINATION lib/cmake
 )
 # install config files
 install(FILES
     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
     "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
-    DESTINATION cmake
+    DESTINATION lib/cmake
 )
 # generate the export targets for the build tree
 export(EXPORT "${PROJECT_NAME}Targets"
