diff --git a/apps/components_tests/esm/testrefid.cpp b/apps/components_tests/esm/testrefid.cpp
index 1911cd1a5a..b03c3832c1 100644
--- a/apps/components_tests/esm/testrefid.cpp
+++ b/apps/components_tests/esm/testrefid.cpp
@@ -255,7 +255,7 @@ namespace ESM
         const std::vector<std::pair<RefId, std::string>> toStringParams = {
             { RefId(), std::string() },
             { RefId::stringRefId("foo"), "foo" },
-            { RefId::stringRefId(std::string({ 'a', 0, -1, '\n', '\t' })), { 'a', 0, -1, '\n', '\t' } },
+            { RefId::stringRefId(std::string({ 'a', 0, static_cast<char>(-1), '\n', '\t' })), { 'a', 0, static_cast<char>(-1), '\n', '\t' } },
             { RefId::formIdRefId({ .mIndex = 42, .mContentFile = 0 }), "0x2a" },
             { RefId::formIdRefId({ .mIndex = 0xffffff, .mContentFile = std::numeric_limits<std::int32_t>::max() }),
                 "0x7fffffffffffff" },
@@ -295,7 +295,7 @@ namespace ESM
             { RefId(), "Empty{}" },
             { RefId::stringRefId("foo"), "\"foo\"" },
             { RefId::stringRefId("BAR"), "\"BAR\"" },
-            { RefId::stringRefId(std::string({ 'a', 0, -1, '\n', '\t' })), "\"a\\x0\\xff\\xa\\x9\"" },
+            { RefId::stringRefId(std::string({ 'a', 0, static_cast<char>(-1), '\n', '\t' })), "\"a\\x0\\xff\\xa\\x9\"" },
             { RefId::stringRefId("Логово дракона"), "\"Логово дракона\"" },
             { RefId::stringRefId("\xd0\x9b"), "\"Л\"" },
             { RefId::stringRefId("\xff\x9b"), "\"\\xff\\x9b\"" },
@@ -334,7 +334,7 @@ namespace ESM
             { RefId(), "" },
             { RefId::stringRefId("foo"), "foo" },
             { RefId::stringRefId("BAR"), "bar" },
-            { RefId::stringRefId(std::string({ 'a', 0, -1, '\n', '\t' })), { 'a', 0, -1, '\n', '\t' } },
+            { RefId::stringRefId(std::string({ 'a', 0, static_cast<char>(-1), '\n', '\t' })), { 'a', 0, static_cast<char>(-1), '\n', '\t' } },
             { RefId::formIdRefId({ .mIndex = 1, .mContentFile = 0 }), "FormId:0x1" },
             { RefId::formIdRefId({ .mIndex = 0x1f, .mContentFile = 0 }), "FormId:0x1f" },
             { RefId::formIdRefId({ .mIndex = 0x1f, .mContentFile = 2 }), "FormId:0x200001f" },
diff --git a/components/esm3/variant.hpp b/components/esm3/variant.hpp
index ed72b1dc05..fb049675aa 100644
--- a/components/esm3/variant.hpp
+++ b/components/esm3/variant.hpp
@@ -5,6 +5,7 @@
 #include <string>
 #include <tuple>
 #include <variant>
+#include <cstdint>
 
 namespace ESM
 {
