diff --git a/apps/openmw/mwgui/controllers.hpp b/apps/openmw/mwgui/controllers.hpp
index 416f104..1485a25 100644
--- a/apps/openmw/mwgui/controllers.hpp
+++ b/apps/openmw/mwgui/controllers.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_CONTROLLERS_H
 #define MWGUI_CONTROLLERS_H
 
+#include <cstdint>
 #include <string>
 #include <MyGUI_ControllerItem.h>
 
diff --git a/apps/openmw/mwgui/itemselection.hpp b/apps/openmw/mwgui/itemselection.hpp
index 6132bac..3ca680e 100644
--- a/apps/openmw/mwgui/itemselection.hpp
+++ b/apps/openmw/mwgui/itemselection.hpp
@@ -1,6 +1,7 @@
 #ifndef OPENMW_GAME_MWGUI_ITEMSELECTION_H
 #define OPENMW_GAME_MWGUI_ITEMSELECTION_H
 
+#include <cstdint>
 #include <MyGUI_Delegate.h>
 
 #include "windowbase.hpp"
diff --git a/apps/openmw/mwgui/timeadvancer.hpp b/apps/openmw/mwgui/timeadvancer.hpp
index b8456f3..06d6f6d 100644
--- a/apps/openmw/mwgui/timeadvancer.hpp
+++ b/apps/openmw/mwgui/timeadvancer.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_TIMEADVANCER_H
 #define MWGUI_TIMEADVANCER_H
 
+#include <cstdint>
 #include <MyGUI_Widget.h>
 
 namespace MWGui
diff --git a/apps/openmw/mwgui/widgets.hpp b/apps/openmw/mwgui/widgets.hpp
index 3c55287..4632347 100644
--- a/apps/openmw/mwgui/widgets.hpp
+++ b/apps/openmw/mwgui/widgets.hpp
@@ -1,6 +1,7 @@
 #ifndef MWGUI_WIDGETS_H
 #define MWGUI_WIDGETS_H
 
+#include <cstdint>
 #include "../mwmechanics/stat.hpp"
 
 #include <components/esm/effectlist.hpp>
diff --git a/apps/openmw/mwinput/controlswitch.hpp b/apps/openmw/mwinput/controlswitch.hpp
index 38d0106..1e68180 100644
--- a/apps/openmw/mwinput/controlswitch.hpp
+++ b/apps/openmw/mwinput/controlswitch.hpp
@@ -1,6 +1,7 @@
 #ifndef MWINPUT_CONTROLSWITCH_H
 #define MWINPUT_CONTROLSWITCH_H
 
+#include <cstdint>
 #include <map>
 #include <string>
 
diff --git a/components/misc/utf8stream.hpp b/components/misc/utf8stream.hpp
index e499d15..666ee56 100644
--- a/components/misc/utf8stream.hpp
+++ b/components/misc/utf8stream.hpp
@@ -2,6 +2,7 @@
 #define MISC_UTF8ITER_HPP
 
 #include <cstring>
+#include <cstdint>
 #include <tuple>
 
 class Utf8Stream
diff --git a/components/myguiplatform/myguidatamanager.cpp b/components/myguiplatform/myguidatamanager.cpp
index 0310e99..f9c73fa 100644
--- a/components/myguiplatform/myguidatamanager.cpp
+++ b/components/myguiplatform/myguidatamanager.cpp
@@ -2,6 +2,7 @@
 
 #include <MyGUI_DataFileStream.h>
 
+#include <memory>
 #include <boost/filesystem.hpp>
 #include <boost/filesystem/fstream.hpp>
 
