From 1166f9d65632a6cb3f0b103a009ead91732d6009 Mon Sep 17 00:00:00 2001
From: ptrcnull <git@ptrcnull.me>
Date: Thu, 3 Mar 2022 19:10:39 +0100
Subject: [PATCH] Remove positional arguments from i18n.merge_file

---
 data/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index a6138165..8d16368a 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -2,7 +2,6 @@ install_man('gnote.1')
 
 desktop_file = 'org.gnome.Gnote.desktop'
 gnote_desktop_file = i18n.merge_file(
-  desktop_file,
   type: 'desktop',
   input: desktop_file + '.in',
   output: desktop_file,
@@ -39,7 +38,6 @@ configure_file(
 
 gnote_appdata = 'org.gnome.Gnote.appdata.xml'
 appdata_file = i18n.merge_file(
-  gnote_appdata,
   input: gnote_appdata + '.in',
   output: gnote_appdata,
   po_dir: '../po',
-- 
2.35.1

