diff -aurN a/mk/module/platform.sh b/mk/module/platform.sh
--- a/mk/module/platform.sh	2014-06-04 03:45:33.000000000 +0000
+++ b/mk/module/platform.sh	2020-04-17 13:43:14.000000000 +0000
@@ -629,6 +629,9 @@
                 ppc)
                     _default_MK_BUILD_ARCH="powerpc"
                     ;;
+		aarch64)
+                    _default_MK_BUILD_ARCH="aarch64"
+                    ;;
                 *)
                     mk_fail "unknown architecture: `uname -m`"
                     ;;
