diff --git a/daemon/Makefile.in b/daemon/Makefile.in
index ce88e8d..d82b3e7 100644
--- a/daemon/Makefile.in
+++ b/daemon/Makefile.in
@@ -337,7 +337,6 @@ INCLUDES = \
 	-D_LARGEFILE64_SOURCE                   \
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
 	-DGNOME_VFS_LIBDIR=\"$(libdir)\"	\
diff --git a/libgnomevfs/Makefile.in b/libgnomevfs/Makefile.in
index 5f96b43..0908007 100644
--- a/libgnomevfs/Makefile.in
+++ b/libgnomevfs/Makefile.in
@@ -430,7 +430,6 @@ INCLUDES = \
 	-D_LARGEFILE64_SOURCE                   \
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_BINDIR=\"$(bindir)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
diff --git a/modules/Makefile.in b/modules/Makefile.in
index 5482f1d..fcf80de 100644
--- a/modules/Makefile.in
+++ b/modules/Makefile.in
@@ -468,7 +468,6 @@ INCLUDES = \
 	-D_POSIX_PTHREAD_SEMANTICS		\
 	-D_REENTRANT				\
 	-DSSH_PROGRAM=\"$(SSH_PROGRAM)\"	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGNOME_VFS_PREFIX=\"$(prefix)\"	\
 	-DGNOME_VFS_DATADIR=\"$(datadir)\"	\
 	-DGNOME_VFS_LIBDIR=\"$(libdir)\"	\
diff --git a/programs/Makefile.in b/programs/Makefile.in
index a7a84f1..be5f238 100644
--- a/programs/Makefile.in
+++ b/programs/Makefile.in
@@ -345,7 +345,6 @@ INCLUDES = \
 	-I$(top_builddir)			\
 	$(TEST_CFLAGS)				\
 	$(VFS_CFLAGS)				\
-	-DG_DISABLE_DEPRECATED
 
 libraries = \
 	$(top_builddir)/libgnomevfs/libgnomevfs-2.la	\
diff --git a/test/Makefile.in b/test/Makefile.in
index 9a3b4b0..6d8e337 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -471,7 +471,6 @@ INCLUDES = \
 	-I$(top_builddir)/libgnomevfs           \
 	$(TEST_CFLAGS)				\
 	$(VFS_CFLAGS)				\
-	-DG_DISABLE_DEPRECATED			\
 	-DMODULES_PATH=\"$(libdir)/vfs/modules\"
 
 @OS_WIN32_FALSE@platform_only_programs = test-dns-sd test-symlinks test-parse-ls-lga
