--- gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-result.c.orig
+++ gnome-vfs-2.24.4/libgnomevfs/gnome-vfs-result.c
@@ -199,7 +199,7 @@
 {
 	switch (h_errno_code) {
 	case HOST_NOT_FOUND:	return GNOME_VFS_ERROR_HOST_NOT_FOUND;
-	case NO_ADDRESS:	return GNOME_VFS_ERROR_HOST_HAS_NO_ADDRESS;
+	case NO_DATA:		return GNOME_VFS_ERROR_HOST_HAS_NO_ADDRESS;
 	case TRY_AGAIN:		return GNOME_VFS_ERROR_NAMESERVER;
 	case NO_RECOVERY:	return GNOME_VFS_ERROR_NAMESERVER;
 	default:
