needed for musl-1.2.3
diff --git a/source/mupen64plus-video-glide64mk2/src/Glide64/Main.cpp b/source/mupen64plus-video-glide64mk2/src/Glide64/Main.cpp
index 58cf44d..7997659 100644
--- a/source/mupen64plus-video-glide64mk2/src/Glide64/Main.cpp
+++ b/source/mupen64plus-video-glide64mk2/src/Glide64/Main.cpp
@@ -1184,7 +1184,7 @@ int InitGfx ()
     char strSstWinOpenExt[] ="grSstWinOpenExt";
     GRWINOPENEXT grSstWinOpenExt = (GRWINOPENEXT)grGetProcAddress(strSstWinOpenExt);
     if (grSstWinOpenExt)
-      gfx_context = grSstWinOpenExt ((uintptr_t)NULL,
+      gfx_context = grSstWinOpenExt ((uintptr_t)0,
       res_data,
       GR_REFRESH_60Hz,
       GR_COLORFORMAT_RGBA,
@@ -1194,7 +1194,7 @@ int InitGfx ()
       1);   // 1 auxillary buffer
   }
   if (!gfx_context)
-    gfx_context = grSstWinOpen ((uintptr_t)NULL,
+    gfx_context = grSstWinOpen ((uintptr_t)0,
     res_data,
     GR_REFRESH_60Hz,
     GR_COLORFORMAT_RGBA,
