Author: Milan P. Stanić <mps@arvanta.net>
Date:   Tue Oct 27 19:42:261 2020 +0000

fix install options for busybox install applet

--- a/Makefile	2020-10-13 01:14:04.000000000 +0000
+++ b/Makefile	2020-10-27 19:52:24.111001813 +0000
@@ -83,7 +83,7 @@
 	rm -f bpfmon bpfmon.o psort psort.o
 
 install: bpfmon
-	$(INSTALL) -TD -m 0755 $< $(DESTDIR)$(PREFIX)/sbin/$<
+	$(INSTALL) -D -m 0755 $< $(DESTDIR)$(PREFIX)/sbin/$<
 	$(STRIP) $<
 
 mkotar:
