--- ./linux/rust.cmake.orig
+++ ./linux/rust.cmake
@@ -2,19 +2,9 @@
 # many dependencies we would need to install Corrosion on the system.
 # See instructions on https://github.com/AndrewGaspar/corrosion#cmake-install
 # Once done, uncomment this line:
-# find_package(Corrosion REQUIRED)
+find_package(Corrosion REQUIRED)
 
-include(FetchContent)
-
-FetchContent_Declare(
-    Corrosion
-    GIT_REPOSITORY https://github.com/AndrewGaspar/corrosion.git
-    GIT_TAG origin/master # Optionally specify a version tag or branch here
-)
-
-FetchContent_MakeAvailable(Corrosion)
-
-corrosion_import_crate(MANIFEST_PATH ../intiface-engine-flutter-bridge/Cargo.toml)
+corrosion_import_crate(MANIFEST_PATH ../intiface-engine-flutter-bridge/Cargo.toml FROZEN)
 
 # Flutter-specific
 
