From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 10 Jan 2022 02:13:54 +0100
Subject: [PATCH] Fix invalid rnpk option in test_rnpkeys_lists

--- a/src/tests/cli_tests.py
+++ b/src/tests/cli_tests.py
@@ -1616,7 +1616,7 @@
 
         _, out, _ = run_proc(RNPK, ['--homedir', data_path('keyrings/1'), '--list-keys'])
         compare_file_any(allow_y2k38_on_32bit(path + 'keyring_1_list_keys'), out, 'keyring 1 key listing failed')
-        _, out, _ = run_proc(RNPK, ['--hom', data_path('keyrings/1'), '-l', '--with-sigs'])
+        _, out, _ = run_proc(RNPK, ['--home', data_path('keyrings/1'), '-l', '--with-sigs'])
         compare_file_any(allow_y2k38_on_32bit(path + 'keyring_1_list_sigs'), out, 'keyring 1 sig listing failed')
         _, out, _ = run_proc(RNPK, ['--home', data_path('keyrings/1'), '--list-keys', '--secret'])
         compare_file_any(allow_y2k38_on_32bit(path + 'keyring_1_list_keys_sec'), out, 'keyring 1 sec key listing failed')
