diff --git a/modules/codec/gstreamer/gstvlcvideopool.h b/modules/codec/gstreamer/gstvlcvideopool.h
index 373a5665e4..40ea01553e 100644
--- a/modules/codec/gstreamer/gstvlcvideopool.h
+++ b/modules/codec/gstreamer/gstvlcvideopool.h
@@ -27,8 +27,8 @@
 #ifndef VLC_GST_VIDEO_POOL_H
 #define VLC_GST_VIDEO_POOL_H
 
+#include <gst/video/video.h>
 #include <gst/gstbufferpool.h>
-#include <gst/video/gstvideopool.h>
 
 #include "gstvlcpictureplaneallocator.h"
 
