--- vlc-2.2.0/modules/codec/omxil/omxil_core.c.orig	2015-02-28 08:37:54.044936036 -0200
+++ vlc-2.2.0/modules/codec/omxil/omxil_core.c	2015-02-28 08:38:38.738271654 -0200
@@ -204,6 +204,12 @@
 #ifdef RPI_OMX
     { "video_decoder.avc", "OMX.broadcom.video_decode" },
     { "video_decoder.mpeg2", "OMX.broadcom.video_decode" },
+    { "video_decoder.mpeg4", "OMX.broadcom.video_decode" },
+    { "video_decoder.vp6", "OMX.broadcom.video_decode" },
+    { "video_decoder.vp8", "OMX.broadcom.video_decode" },
+    { "video_decoder.theora", "OMX.broadcom.video_decode" },
+    { "video_decoder.mjpg", "OMX.broadcom.video_decode" },
+    { "video_decoder.vc1", "OMX.broadcom.video_decode" },
     { "iv_renderer", "OMX.broadcom.video_render" },
 #endif
     { 0, 0 }
