diff -upr drawterm-20170119.orig/Make.unix drawterm-20170119/Make.unix
--- drawterm-20170119.orig/Make.unix	2017-01-23 22:41:27.402630694 +0100
+++ drawterm-20170119/Make.unix	2017-01-23 22:41:43.879309573 +0100
@@ -5,12 +5,15 @@ AR=ar
 AS=as
 RANLIB=ranlib
 X11=/usr/X11R6
-CC=gcc
-CFLAGS=-Wall -Wno-missing-braces -ggdb -I$(ROOT) -I$(ROOT)/include -I$(ROOT)/kern -c -I$(X11)/include -D_THREAD_SAFE $(PTHREAD) -O2
+X11INC=$(X11)/include
+X11LIB=$(X11)/lib
+CC?=gcc
+CFLAGS?=-O2
+CFLAGS+=-Wall -Wno-missing-braces -ggdb -I$(ROOT) -I$(ROOT)/include -I$(ROOT)/kern -c -I$(X11INC) -D_THREAD_SAFE $(PTHREAD)
 O=o
 OS=posix
 GUI=x11
-LDADD=-L$(X11)/lib64 -L$(X11)/lib -lX11 -ggdb
+LDADD=-L$(X11LIB) -lX11 -ggdb
 LDFLAGS=$(PTHREAD)
 TARG=drawterm
 # AUDIO=none
