From 49da91c786855228158cb8937c972cd741bd4c79 Mon Sep 17 00:00:00 2001
From: Martin Dagarin <SloCompTech@users.noreply.github.com>
Date: Mon, 4 Sep 2017 21:10:15 +0200
Subject: [PATCH] Fixed compile bug

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index f2a6800f..f193fec5 100644
--- a/meson.build
+++ b/meson.build
@@ -18,6 +18,7 @@ jsoncpp_cdata.set('JSONCPP_VERSION', meson.project_version())
 jsoncpp_cdata.set('JSONCPP_VERSION_MAJOR', jsoncpp_major_version)
 jsoncpp_cdata.set('JSONCPP_VERSION_MINOR', jsoncpp_minor_version)
 jsoncpp_cdata.set('JSONCPP_VERSION_PATCH', jsoncpp_patch_version)
+jsoncpp_cdata.set('JSONCPP_USE_SECURE_MEMORY',0)
 
 jsoncpp_gen_sources = configure_file(
   input : 'src/lib_json/version.h.in',
