diff --git a/libweston/backend-drm/meson.build b/libweston/backend-drm/meson.build
index 002669e6..37fcb1ed 100644
--- a/libweston/backend-drm/meson.build
+++ b/libweston/backend-drm/meson.build
@@ -4,7 +4,7 @@ endif
 
 dep_libdisplay_info = dependency(
 	'libdisplay-info',
-	version: ['>= 0.1.1', '< 0.3.0'],
+	version: ['>= 0.1.1'],
 	fallback: ['display-info', 'di_dep'],
 	default_options: [
 		'werror=false',
