Upstream: Yes
Reason: Fixes compilation on musl

diff --git a/tests/weston-test-fixture-compositor.c b/tests/weston-test-fixture-compositor.c
index 0c9855f..e0e32c9 100644
--- a/tests/weston-test-fixture-compositor.c
+++ b/tests/weston-test-fixture-compositor.c
@@ -31,6 +31,7 @@
 #include <unistd.h>
 #include <sys/file.h>
 #include <errno.h>
+#include <fcntl.h>
 
 #include "shared/helpers.h"
 #include "weston-test-fixture-compositor.h"
