diff --git a/blockchain/blockexplorer.py b/blockchain/blockexplorer.py
index 35b2b87..8a99d61 100644
--- a/blockchain/blockexplorer.py
+++ b/blockchain/blockexplorer.py
@@ -6,7 +6,8 @@ at https://blockchain.info/api/blockchain_api
 from . import util
 import json
 from enum import Enum
-from past.builtins import basestring
+from builtins import str
+
 
 def get_block(block_id, api_code=None):
     """Get a single block based on a block hash.
@@ -124,7 +125,7 @@ def get_multi_address(addresses, filter=None, limit=None, offset=None, api_code=
     :return: an instance of :class:`MultiAddress` class
     """
 
-    if isinstance(addresses, basestring):
+    if isinstance(addresses, str):
         resource = 'multiaddr?active=' + addresses
     else:
         resource = 'multiaddr?active=' + '|'.join(addresses)
@@ -153,7 +154,7 @@ def get_balance(addresses, filter=None, api_code=None):
     :return: a dictionary of str, :class:`Balance`
     """
 
-    if isinstance(addresses, basestring):
+    if isinstance(addresses, str):
         resource = 'balance?active=' + addresses
     else:
         resource = 'balance?active=' + '|'.join(addresses)
@@ -180,7 +181,7 @@ def get_unspent_outputs(addresses, confirmations=None, limit=None, api_code=None
     :return: an array of :class:`UnspentOutput` objects
     """
 
-    if isinstance(addresses, basestring):
+    if isinstance(addresses, str):
         resource = 'unspent?active=' + addresses
     else:
         resource = 'unspent?active=' + '|'.join(addresses)
