diff -Naur ldns-1.6.17.orig/examples/ldns-signzone.c ldns-1.6.17/examples/ldns-signzone.c
--- ldns-1.6.17.orig/examples/ldns-signzone.c	2014-01-11 06:04:41.000000000 +0900
+++ ldns-1.6.17/examples/ldns-signzone.c	2015-09-25 01:54:54.364663640 +0900
@@ -470,6 +470,7 @@
 		case 'A':
 			signflags |= LDNS_SIGN_DNSKEY_WITH_ZSK;
 			break;
+#ifdef HAVE_ENGINE_LOAD_CRYPTODEV
 		case 'E':
 			ENGINE_load_builtin_engines();
 			ENGINE_load_dynamic();
@@ -494,6 +495,7 @@
 				ENGINE_set_default(engine, 0);
 			}
 			break;
+#endif
 		case 'k':
 			eng_key_l = strchr(optarg, ',');
 			if (eng_key_l && strlen(eng_key_l) > 1) {
