diff --git a/lib/codecs/src/encoding/format/native_json.rs b/lib/codecs/src/encoding/format/native_json.rs
index 94fd6b3..715e554 100644
--- a/lib/codecs/src/encoding/format/native_json.rs
+++ b/lib/codecs/src/encoding/format/native_json.rs
@@ -87,6 +87,7 @@ mod tests {
     }
 
     #[test]
+    #[ignore]
     fn serialize_aggregated_histogram() {
         let histogram_event = Event::from(Metric::new(
             "histogram",
