diff --git a/Cargo.lock b/Cargo.lock
index bd72b08a7..c62c70251 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -4128,6 +4128,7 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "843fba2746e448b37e26a819579957415c8cef339bf08564fe8b7ddbd959573c"
 dependencies = [
  "crc32fast",
+ "libz-ng-sys",
  "miniz_oxide",
  "zlib-rs",
 ]
@@ -6330,6 +6331,16 @@ dependencies = [
  "vcpkg",
 ]
 
+[[package]]
+name = "libz-ng-sys"
+version = "1.1.25"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "89b6bd0c289b22f4973012fa91dfed9d5f4b633567bb26f9454fbf437817b499"
+dependencies = [
+ "cmake",
+ "libc",
+]
+
 [[package]]
 name = "libz-sys"
 version = "1.1.22"
diff --git a/Cargo.toml b/Cargo.toml
index c6cbdbd2c..f256674d8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -161,7 +161,7 @@ darling = { version = "0.20.11", default-features = false, features = ["suggesti
 dashmap = { version = "6.1.0", default-features = false }
 derivative = { version = "2.2.0", default-features = false }
 exitcode = { version = "1.1.2", default-features = false }
-flate2 = { version = "1.1.2", default-features = false, features = ["zlib-rs"] }
+flate2 = { version = "1.1.2", default-features = false, features = ["zlib-ng"] }
 futures = { version = "0.3.31", default-features = false, features = ["compat", "io-compat", "std"], package = "futures" }
 futures-util = { version = "0.3.29", default-features = false }
 glob = { version = "0.3.3", default-features = false }
@@ -440,7 +440,7 @@ thread_local = { version = "1.1.9", default-features = false, optional = true }
 typetag = { version = "0.2.20", default-features = false }
 url.workspace = true
 warp = { version = "0.3.7", default-features = false }
-zstd = { version = "0.13.0", default-features = false }
+zstd = { version = "0.13.0", default-features = false, features = ["pkg-config"] }
 arr_macro = { version = "0.2.1" }
 
 # depending on fork for bumped nix dependency
@@ -515,7 +515,7 @@ enable-api-client = ["base", "api-client"]
 enable-unix = ["enable-api-client", "sources-dnstap", "unix"]
 
 default-msvc = ["enable-api-client", "rdkafka?/cmake_build"]
-default-musl = ["enable-unix", "rdkafka?/gssapi-vendored", "rdkafka?/cmake_build"]
+default-musl = ["enable-unix", "rdkafka?/gssapi", "rdkafka?/dynamic-linking"]
 default-no-api-client = ["base", "sources-dnstap", "unix", "rdkafka?/gssapi-vendored"]
 
 tokio-console = ["dep:console-subscriber", "tokio/tracing"]
