--- a/Cargo.toml
+++ b/Cargo.toml
@@ -325,7 +325,7 @@
 url = { version = "2.4.0", default-features = false, features = ["serde"] }
 uuid = { version = "1", default-features = false, features = ["serde", "v4"] }
 warp = { version = "0.3.5", default-features = false }
-zstd = { version = "0.12.4", default-features = false }
+zstd = { version = "0.12.4", features = ["pkg-config"] }
 arr_macro = { version = "0.2.1" }
 
 # depending on fork for bumped nix dependency
