--- ./configure.in.orig	2012-12-31 14:40:30.297439517 +0000
+++ ./configure.in	2012-12-31 14:40:51.587629835 +0000
@@ -51,7 +51,7 @@
 cflags_set=${CFLAGS+set}
 
 AM_INIT_AUTOMAKE([no-define -Wno-portability dist-bzip2])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 
 # Support silent build rules, requires at least automake-1.11. Enable
 # by either passing --enable-silent-rules to configure or passing V=0
@@ -300,7 +300,6 @@
 
 # Checks for programs.
 AC_ISC_POSIX
-AM_PROG_CC_STDC
 AM_PROG_CC_C_O
 AC_PROG_INSTALL
 AC_PROG_MAKE_SET
