--- gtk+-2.24.0.orig/gdk/gdkwindow.h
+++ gtk+-2.24.0/gdk/gdkwindow.h
@@ -728,7 +728,6 @@
 
 #define GDK_ROOT_PARENT()             (gdk_get_default_root_window ())
 #define gdk_window_get_size            gdk_drawable_get_size
-#define gdk_window_get_type            gdk_window_get_window_type
 #define gdk_window_get_colormap        gdk_drawable_get_colormap
 #define gdk_window_set_colormap        gdk_drawable_set_colormap
 #define gdk_window_ref                 g_object_ref
