--- criu-2.6/test/zdtm/static/socket6_udp.c
+++ criu-2.6/test/zdtm.new/static/socket6_udp.c
@@ -13,7 +13,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>  /* for sockaddr_in and inet_ntoa() */
-#include <wait.h>
+#include <sys/wait.h>
 
 static int port = 8880;
 static char buf[64];
--- criu-2.6/test/zdtm/static/socket_aio.c
+++ criu-2.6/test/zdtm.new/static/socket_aio.c
@@ -16,7 +16,7 @@
 #include <errno.h>
 #include <stdlib.h>
 #include <aio.h>
-#include <wait.h>
+#include <sys/wait.h>
 #include <netinet/tcp.h>
 
 static int port = 8880;
--- criu-2.6/test/zdtm/static/socket_listen.c
+++ criu-2.6/test/zdtm.new/static/socket_listen.c
@@ -19,7 +19,7 @@
 #include <unistd.h>
 #include <errno.h>
 #include <stdlib.h>
-#include <wait.h>
+#include <sys/wait.h>
 #include <netinet/tcp.h>
 
 static int port = 8880;
--- criu-2.6/test/zdtm/static/socket_udp.c
+++ criu-2.6/test/zdtm.new/static/socket_udp.c
@@ -18,7 +18,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>  /* for sockaddr_in and inet_ntoa() */
-#include <wait.h>
+#include <sys/wait.h>
 
 static int port = 8880;
 static char buf[8];
--- criu-2.6/test/zdtm/static/socket_udplite.c
+++ criu-2.6/test/zdtm.new/static/socket_udplite.c
@@ -18,7 +18,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <arpa/inet.h>  /* for sockaddr_in and inet_ntoa() */
-#include <wait.h>
+#include <sys/wait.h>
 
 static int port = 8890;
 static char buf[8];
