--- a/test/zdtm/static/unlink_multiple_largefiles.c
+++ b/test/zdtm/static/unlink_multiple_largefiles.c
@@ -124,7 +124,7 @@
 	check_extent_map(fiemap);
 	free(fiemap);
 
-	lseek64(fd, FSIZE, SEEK_SET);
+	lseek(fd, FSIZE, SEEK_SET);
 
 	create_check_pattern(ebuf, BUFSIZE, seed);
 
@@ -174,7 +174,7 @@
 	}
 	test_msg("Created file: %s, fd %d\n", fnm, fd);
 
-	if (lseek64(fd, FSIZE, SEEK_SET) < 0) {
+	if (lseek(fd, FSIZE, SEEK_SET) < 0) {
 		pr_perror("Cannot seek to offset %llx", FSIZE);
 		goto failed;
 	}
