--- ./testing/test_integration.py
+++ ./testing/test_integration.py
@@ -119,7 +119,7 @@
     monkeypatch.setenv(PRETEND_KEY, "dummy")
     wd.write("setup.py", SETUP_PY_PLAIN)
     assert wd.get_version(write_to="test.py") == "dummy"
-    assert wd("python setup.py --version") == "0.0.0"
+    assert wd("python3 setup.py --version") == "0.0.0"
 
 
 def test_own_setup_fails_on_old_python(monkeypatch):
