--- rpm-4.10.2.orig/misc/fts.c
+++ rpm-4.10.2/misc/fts.c
@@ -42,6 +42,11 @@
 #include <string.h>
 #include <unistd.h>
 #else
+#include "system.h"
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include "misc/fts.h"
 #if defined(hpux)
 # define        _INCLUDE_POSIX_SOURCE
 #   define __errno_location() 	(&errno)
@@ -61,11 +66,10 @@
 #   define _STAT_VER		0
 #   define __fxstat64(_stat_ver, _fd, _sbp) fstat64((_fd), (_sbp))
 #endif
-#include "system.h"
-#include <stdlib.h>
-#include <string.h>
-#include <errno.h>
-#include "misc/fts.h"
+#if defined(__UCLIBC__)
+#   define _STAT_VER		0
+#   define __fxstat64(_stat_ver, _fd, _sbp) fstat64((_fd), (_sbp))
+#endif
 #   define __set_errno(val) (*__errno_location ()) = (val)
 #   define __open	open
 #   define __close	close
