diff --git a/advance/menu/emulator.cc b/advance/menu/emulator.cc
index cfa4300..3c3d5d6 100644
--- a/advance/menu/emulator.cc
+++ b/advance/menu/emulator.cc
@@ -2203,7 +2203,7 @@ bool advmess::run(const game& g, const game* bios, unsigned orientation, bool se
 						}
 					}
 					error += "\n";
-					target_err(error.c_str());
+					target_err("%s", error.c_str());
 				}
 
 				for (int i = 0; i < argc; ++i)
