From 9980b30b89206cd7b289f77ac5d0f1fc3fc0fa64 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 9 Jan 2017 19:27:03 +0000
Subject: [PATCH] make: install nvme and zfs files

---
 Makefile | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index e6bd473..4329072 100644
--- a/Makefile
+++ b/Makefile
@@ -29,10 +29,13 @@ CONF_FILES	:= mkinitfs.conf \
 		features.d/lvm.files \
 		features.d/lvm.modules \
 		features.d/mmc.modules \
+		features.d/nbd.files \
+		features.d/nbd.modules \
 		features.d/network.files \
 		features.d/network.modules \
+		features.d/nvme.modules \
 		features.d/ocfs2.modules \
-		features.d/raid.files\
+		features.d/raid.files \
 		features.d/raid.modules \
 		features.d/reiserfs.modules \
 		features.d/scsi.modules \
@@ -41,8 +44,8 @@ CONF_FILES	:= mkinitfs.conf \
 		features.d/usb.modules \
 		features.d/virtio.modules \
 		features.d/xfs.modules \
-		features.d/nbd.files \
-		features.d/nbd.modules
+		features.d/zfs.files \
+		features.d/zfs.modules
 
 SCRIPTS		:= mkinitfs bootchartd initramfs-init
 IN_FILES	:= $(addsuffix .in,$(SCRIPTS))
-- 
2.11.0

