Common subdirectories: a/features.d and b/features.d
diff -u a/initramfs-init.in b/initramfs-init.in
--- a/initramfs-init.in	2016-06-15 12:57:27.000000000 +0200
+++ b/initramfs-init.in	2016-08-16 00:30:42.936992624 +0200
@@ -248,7 +248,7 @@
 myopts="alpine_dev autodetect autoraid chart cryptroot cryptdm debug_init
	dma init_args keep_apk_new modules ovl_dev pkgs quiet root_size root
	usbdelay ip alpine_repo apkovl alpine_start splash blacklist
-	overlaytmpfs rootfstype rootflags"
+	overlaytmpfs rootfstype rootflags resume"

 for opt; do
	case "$opt" in
@@ -351,16 +351,25 @@
	fi

	ebegin "Mounting root"
+	nlplug-findfs $cryptopts -p /sbin/mdev $KOPT_root
+
+	if [ -n "$KOPT_resume" ]; then
+		echo "Resume from disk"
+		if [ -e /sys/power/resume ]; then
+			printf "%d:%d" $(stat -Lc "0x%t 0x%T" "$KOPT_resume") >/sys/power/resume
+		else
+			echo "resume: no hibernation support found"
+		fi
+	fi
+
	if [ "$KOPT_overlaytmpfs" = "yes" ]; then
		mkdir -p /media/root-ro /media/root-rw $sysroot/media/root-ro \
			$sysroot/media/root-rw
-		nlplug-findfs $cryptopts -p /sbin/mdev $KOPT_root \
-			&& mount -o ro $KOPT_root /media/root-ro
+		mount -o ro $KOPT_root /media/root-ro
		mount -t tmpfs root-tmpfs /media/root-rw
		mkdir -p /media/root-rw/work /media/root-rw/root
		mount -t overlay -o lowerdir=/media/root-ro,upperdir=/media/root-rw/root,workdir=/media/root-rw/work overlayfs $sysroot
	else
-		nlplug-findfs $cryptopts -p /sbin/mdev $KOPT_root
		mount ${KOPT_rootfstype:+-t} ${KOPT_rootfstype} \
			-o ${KOPT_rootflags:-ro} \
			$KOPT_root $sysroot
