From 2124cf8473a0d1ebbeb6f9d639a8dd254135351d Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 30 Jul 2015 15:11:47 +0200
Subject: [PATCH] mkinitfs: copy modules.order and modules.builtin if those
 exist

---
 mkinitfs.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mkinitfs.in b/mkinitfs.in
index 04dc99c..c724e68 100755
--- a/mkinitfs.in
+++ b/mkinitfs.in
@@ -115,6 +115,11 @@ initfs_kmods() {
 	for file in $(find_kmods); do
 		echo "${file#/}"
 	done | sort -u | cpio -pdm "$tmpdir" || return 1
+	for file in modules.order modules.builtin; do
+		if [ -f "$kerneldir"/$file ]; then
+			cp "$kerneldir"/$file "$tmpdir"/lib/modules/$kernel/
+		fi
+	done
 	depmod $kernel -b "$tmpdir"
 	cd "$oldpwd"
 }
-- 
2.5.0

