From bf94c9b9aa7884fc50d3110d69e2d28e413159ed Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 24 Nov 2009 12:23:38 +0000
Subject: [PATCH] init: never overwrite existing files

ref #197
---
 initramfs-init.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index 04798de..bd97ace 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -372,7 +372,7 @@ if [ -n "$KOPT_chart" ]; then
 fi
 apkflags="--initdb --quiet --progress --force --no-network"
 if [ -z "$KOPT_keep_apk_new" ]; then
-	apkflags="$apkflags --clean-protected"
+	apkflags="$apkflags --clean-protected --never-overwrite"
 fi
 apk add --root $sysroot $repo_opt $apkflags $pkgs >/dev/null
 eend $?
-- 
1.6.5.3

