From d50371d52ddad76f6393ceadd74ed5f4433b9666 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 7 Apr 2011 19:46:29 +0000
Subject: [PATCH] init: use new location for world

---
 initramfs-init.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index 1040288..abc888f 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -391,7 +391,7 @@ if [ -f "$ovl" ]; then
 	umount /media/$i 2>/dev/null &
 	pkgs="$pkgs $(sed 's/\#.*//' $sysroot/etc/lbu/packages.list 2>/dev/null)"
 	rm -f "$sysroot"/etc/lbu/packages.list
-	pkgs="$pkgs $(cat $sysroot/var/lib/apk/world 2>/dev/null)"
+	pkgs="$pkgs $(cat $sysroot/var/lib/apk/world $sysroot/etc/apk/world 2>/dev/null)"
 else
 	# add some boot services by default
 	rc_add devfs sysinit
-- 
1.7.4.4

