From fd347152fd873803441570c29fbcee90326722d9 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 25 Jan 2019 16:05:28 +0000
Subject: [PATCH] features: add crc32 modules to f2fs

---
 features.d/f2fs.modules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/features.d/f2fs.modules b/features.d/f2fs.modules
index b16edae..920b83c 100644
--- a/features.d/f2fs.modules
+++ b/features.d/f2fs.modules
@@ -1 +1,4 @@
+kernel/arch/*/crypto/crc32*
+kernel/crypto/crc32*
+kernel/fs/crypto/fscrypto
 kernel/fs/f2fs
-- 
2.20.1

