From ace78b7225286039bf89aaae86a02b653a8ef64d Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@alpinelinux.org>
Date: Mon, 25 Jun 2018 15:25:02 +0000
Subject: [PATCH] features: add vc4 to kms for rpi

---
 features.d/kms.modules | 1 +
 1 file changed, 1 insertion(+)

diff --git a/features.d/kms.modules b/features.d/kms.modules
index 8806aab..b62e75b 100644
--- a/features.d/kms.modules
+++ b/features.d/kms.modules
@@ -3,3 +3,4 @@ kernel/drivers/gpu
 kernel/drivers/i2c
 kernel/drivers/video
 kernel/arch/x86/video/fbdev.ko
+kernel/drivers/gpu/drm/vc4/vc4.ko
-- 
2.18.0

