From 56d4208345b8da08ada6e6176a0a46cf94f89516 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@alpinelinux.org>
Date: Sun, 29 Apr 2018 21:14:36 +0000
Subject: [PATCH] features: virtio_net depends on virtio_pci

---
 features.d/network.modules | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/features.d/network.modules b/features.d/network.modules
index caa37a8..e4f084c 100644
--- a/features.d/network.modules
+++ b/features.d/network.modules
@@ -2,4 +2,5 @@ kernel/drivers/net/ethernet
 kernel/net/packet/af_packet.ko
 kernel/drivers/net/hyperv
 kernel/drivers/net/vmxnet3
-kernel/drivers/net/virtio_net*
+kernel/drivers/virtio/virtio_pci.ko
+kernel/drivers/net/virtio_net.ko
-- 
2.17.0

