From 3d83e2968ac40a80162821c358316cfea12a33e2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 2 Dec 2013 13:10:54 +0000
Subject: [PATCH] cyptsetup: add another missing dep, libpthread

ref #2445
---
 files.d/cryptsetup | 1 +
 1 file changed, 1 insertion(+)

diff --git a/files.d/cryptsetup b/files.d/cryptsetup
index 6b170d2..e9810ef 100644
--- a/files.d/cryptsetup
+++ b/files.d/cryptsetup
@@ -7,3 +7,4 @@
 /lib/libcrypto.so.*
 /lib/libz.so.*
 /lib/libdl.so.*
+/lib/libpthread*.so*
-- 
1.8.4.3

