From 92801060dccf632c827867eb361396646e04ae99 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 21 Dec 2016 23:03:01 +0000
Subject: [PATCH] ext3 functionality is provided by ext4 module

ref #6167
---
 features.d/ext3.modules | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/features.d/ext3.modules b/features.d/ext3.modules
index eeadf78..a5bb1c3 100644
--- a/features.d/ext3.modules
+++ b/features.d/ext3.modules
@@ -1 +1 @@
-kernel/fs/ext3
+kernel/fs/ext4
-- 
2.11.0

