From f6e59ac9e500f9a32f96c39201f4323c058c9e2e Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 18 Nov 2013 13:27:53 +0000
Subject: [PATCH] cryptsetup: add missing libdl dependency

---
 files.d/cryptsetup | 1 +
 1 file changed, 1 insertion(+)

diff --git a/files.d/cryptsetup b/files.d/cryptsetup
index 5c7198d..6b170d2 100644
--- a/files.d/cryptsetup
+++ b/files.d/cryptsetup
@@ -6,3 +6,4 @@
 /lib/libssl.so.*
 /lib/libcrypto.so.*
 /lib/libz.so.*
+/lib/libdl.so.*
-- 
1.8.4.3

