From e2aa461b2e6385ff5a38278425132dbb850bb3dd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 10 Aug 2016 15:16:06 +0200
Subject: [PATCH] group: sync with alpine-baselayout

---
 group | 14 ++++----------
 1 file changed, 4 insertions(+), 10 deletions(-)

diff --git a/group b/group
index 5a6cbcc..9559744 100644
--- a/group
+++ b/group
@@ -21,34 +21,28 @@ cdrom:x:19:
 dialout:x:20:root
 ftp:x:21:
 sshd:x:22:
+input:x:23:
 at:x:25:at
 tape:x:26:root
 video:x:27:root
 netdev:x:28:
 readproc:x:30:
 squid:x:31:squid
-gdm:x:32:gdm
 xfs:x:33:xfs
 kvm:x:34:kvm
 games:x:35:
-named:x:40:named
-mysql:x:60:
+shadow:x:42:
 postgres:x:70:
 cdrw:x:80:
-apache:x:81:
-nut:x:84:
 usb:x:85:
-avahi:x:86:
 vpopmail:x:89:
 users:x:100:games
 ntp:x:123:
 nofiles:x:200:
-postfix:x:207:
-postdrop:x:208:
 smmsp:x:209:smmsp
-distcc:x:240:
-slocate:x:245:
+locate:x:245:
 abuild:x:300:
 utmp:x:406:
+ping:x:999:
 nogroup:x:65533:
 nobody:x:65534:
-- 
2.9.1

