From 841a76b3a96955c7840edc29f8397078514a55a9 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 25 Jan 2017 08:57:24 +0000
Subject: [PATCH] init: pull in libressl instead of openssl for encrypted
 apkovl

ref #6689
---
 initramfs-init.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index f4cd274..adcc7be 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -55,7 +55,7 @@ unpack_apkovl() {
 	fi
 
 	# we need openssl. let apk handle deps
-	apk add --quiet --initdb --repositories-file $repofile openssl || return 1
+	apk add --quiet --initdb --repositories-file $repofile libressl || return 1
 
 	if ! openssl list-cipher-commands | grep "^$suffix$" > /dev/null; then
 		errstr="Cipher $suffix is not supported"
-- 
2.11.0

