From 2a6bc56f7be2c2356e7b33dd2fc3af709ace0aa3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?=
 <pabloyoyoista@postmarketos.org>
Date: Fri, 4 Jul 2025 15:10:42 +0200
Subject: [PATCH 1/3] mkinitfs: make sure /usr symlinks exist in initramfs

---
 mkinitfs.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mkinitfs.in b/mkinitfs.in
index 7295dec..964ff92 100755
--- a/mkinitfs.in
+++ b/mkinitfs.in
@@ -37,11 +37,12 @@ feature_files() {
 
 initfs_base() {
 	local i= dirs= glob= file=
-	for i in dev proc sys sbin bin run .modloop lib/modules media/cdrom \
-	    etc/apk media/floppy media/usb newroot; do
+	for i in dev proc sys usr/sbin usr/bin run .modloop usr/lib/modules \
+	    media/cdrom etc/apk media/floppy media/usb newroot; do
 		dirs="$dirs $tmpdir/$i"
 	done
 	mkdir -p $dirs
+	ln -s usr/bin usr/sbin usr/lib "$tmpdir"/
 
 	local oldpwd="$PWD"
 	cd "${basedir}"
-- 
GitLab


From 8db00b3a661af1a18c37ec3976b117a4624544e1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?=
 <pabloyoyoista@postmarketos.org>
Date: Fri, 4 Jul 2025 15:22:18 +0200
Subject: [PATCH 2/3] ci: fix moved location of oksh

---
 .gitlab-ci.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.gitlab-ci.yml b/.gitlab-ci.yml
index 524681b..5b44f80 100644
--- a/.gitlab-ci.yml
+++ b/.gitlab-ci.yml
@@ -18,7 +18,7 @@ test-oksh:
   extends: test-default
   before_script:
     - apk add oksh
-    - ln -sf /bin/oksh /bin/sh
+    - ln -sf /usr/bin/oksh /bin/sh
 
 #test-yash:
 #  extends: test-default
-- 
GitLab


From 6cd600490e41c1f2e3b8528489547b0c670fbf10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?=
 <pabloyoyoista@postmarketos.org>
Date: Wed, 27 Aug 2025 18:10:16 +0200
Subject: [PATCH 3/3] initramfs-init: create tmpfs sysroot /usr-merged

And make sure that tests follow the convention
---
 initramfs-init.in         | 2 ++
 tests/initramfs-init.test | 8 ++++++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/initramfs-init.in b/initramfs-init.in
index ece38bd..16048e1 100755
--- a/initramfs-init.in
+++ b/initramfs-init.in
@@ -855,6 +855,8 @@ if [ -n "$KOPT_rootflags" ]; then
 fi
 
 $MOCK mount -t tmpfs -o $rootflags tmpfs $sysroot
+$MOCK mkdir -p "$sysroot"/usr/lib "$sysroot"/usr/bin "$sysroot"/usr/sbin
+$MOCK ln -s usr/bin usr/sbin usr/lib "$sysroot"/
 
 if [ -z "$KOPT_apkovl" ]; then
 	# Not manually set, use the apkovl found by nlplug
diff --git a/tests/initramfs-init.test b/tests/initramfs-init.test
index 49c06af..c170bbf 100755
--- a/tests/initramfs-init.test
+++ b/tests/initramfs-init.test
@@ -41,13 +41,15 @@ fake_cmdline() {
 }
 
 fake_bin() {
-	mkdir -p bin
+	mkdir -p usr/bin
+	ln -s usr/bin .
 	cat > bin/"$1"
 	chmod +x bin/"$1"
 }
 
 fake_sysroot_init() {
-	mkdir -p sysroot/sbin
+	mkdir -p sysroot/usr/sbin
+	ln -s usr/sbin sysroot/sbin
 	touch sysroot/sbin/init
 	chmod +x sysroot/sbin/init
 }
@@ -119,6 +121,8 @@ initramfs_init_tmpfs_root_body() {
 	atf_check \
 		-o match:"nlplug-findfs" \
 		-o match:"mount.*tmpfs .*/sysroot" \
+		-o match:"mkdir -p .*/sysroot/usr/lib .*/sysroot/usr/bin .*/sysroot/usr/sbin" \
+		-o match:"ln -s usr/bin usr/sbin usr/lib .*/sysroot" \
 		-o match:"switch_root OK" \
 		initramfs-init
 }
-- 
GitLab

