diff --git a/external/partio_zip/zip_manager.hpp b/external/partio_zip/zip_manager.hpp
index 666884d..15683d2 100644
--- a/external/partio_zip/zip_manager.hpp
+++ b/external/partio_zip/zip_manager.hpp
@@ -44,6 +44,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGES.
 #include <fstream>
 #include <iostream>
 #include <map>
+#include <memory>
 #include <string>
 #include <stdexcept>
 #include <vector>
