diff --git a/src/supertux/game_session.cpp b/src/supertux/game_session.cpp
index 41de52e..a8ebfc9 100644
--- a/src/supertux/game_session.cpp
+++ b/src/supertux/game_session.cpp
@@ -14,6 +14,8 @@
 //  You should have received a copy of the GNU General Public License
 //  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+#include <cfloat>
+
 #include "supertux/game_session.hpp"
 
 #include "audio/sound_manager.hpp"
