--- a/Makefile.am
+++ b/Makefile.am
@@ -45,9 +45,6 @@
 if CPU_X86_64
 include aes/Makefile.am
 endif
-if CPU_X86_32
-include aes/Makefile.am
-endif
 if CPU_AARCH64
 include aes/Makefile.am
 endif
@@ -66,7 +63,7 @@
 endif
 
 if CPU_X86_32
-libisal_crypto_la_SOURCES += ${lsrc_x86_32}
+libisal_crypto_la_SOURCES += ${lsrc_base_aliases}
 endif
 
 if CPU_AARCH64
--- a/fips/Makefile.am
+++ b/fips/Makefile.am
@@ -30,9 +30,8 @@
 src_include     += -I $(srcdir)/fips
 extern_hdrs     += include/isal_crypto_api.h include/aes_xts.h include/aes_keyexp.h include/sha1_mb.h include/sha256_mb.h
 
-lsrc_x86_64     += fips/self_tests.c
-lsrc_aarch64    += fips/self_tests_generic.c
-lsrc            += fips/aes_self_tests.c
+lsrc_x86_64     += fips/self_tests.c fips/aes_self_tests.c
+lsrc_aarch64    += fips/self_tests_generic.c fips/aes_self_tests.c
 lsrc            += fips/sha_self_tests.c
 
 lsrc_x86_64     += fips/asm_self_tests.asm
--- a/mh_sha1/mh_sha1.c
+++ b/mh_sha1/mh_sha1.c
@@ -137,7 +137,7 @@ mh_sha1_finalize_base(struct isal_mh_sha1_ctx *ctx, void *mh_sha1_digest)
 }
 
 #if (!defined(NOARCH)) &&                                                                          \
-        (defined(__i386__) || defined(__x86_64__) || defined(_M_X64) || defined(_M_IX86))
+        (defined(__x86_64__) || defined(_M_X64))
 
 /***************mh_sha1_update***********/
 // mh_sha1_update_sse.c
--- a/mh_sha1_murmur3_x64_128/mh_sha1_murmur3_x64_128.c
+++ b/mh_sha1_murmur3_x64_128/mh_sha1_murmur3_x64_128.c
@@ -165,7 +165,7 @@ mh_sha1_murmur3_x64_128_finalize_base(struct isal_mh_sha1_murmur3_x64_128_ctx *c
 }
 
 #if (!defined(NOARCH)) &&                                                                          \
-        (defined(__i386__) || defined(__x86_64__) || defined(_M_X64) || defined(_M_IX86))
+        (defined(__x86_64__) || defined(_M_X64))
 /***************mh_sha1_murmur3_x64_128_update***********/
 // mh_sha1_murmur3_x64_128_update_sse.c
 #define UPDATE_FUNCTION _mh_sha1_murmur3_x64_128_update_sse
--- a/mh_sha256/mh_sha256.c
+++ b/mh_sha256/mh_sha256.c
@@ -140,7 +140,7 @@ mh_sha256_finalize_base(struct isal_mh_sha256_ctx *ctx, void *mh_sha256_digest)
 }
 
 #if (!defined(NOARCH)) &&                                                                          \
-        (defined(__i386__) || defined(__x86_64__) || defined(_M_X64) || defined(_M_IX86))
+        (defined(__x86_64__) || defined(_M_X64))
 /***************mh_sha256_update***********/
 // mh_sha256_update_sse.c
 #define MH_SHA256_UPDATE_FUNCTION _mh_sha256_update_sse
