--- a/src/3rdparty/ninja/configure.py
+++ b/src/3rdparty/ninja/configure.py
@@ -23,11 +23,12 @@ from __future__ import print_function

 from optparse import OptionParser
 import os
-import pipes
 import string
 import subprocess
 import sys

+from shlex import quote
+
 sourcedir = os.path.dirname(os.path.realpath(__file__))
 sys.path.insert(0, os.path.join(sourcedir, 'misc'))
 import ninja_syntax
@@ -259,7 +260,7 @@ n.variable('configure_args', ' '.join(configure_args))
 env_keys = set(['CXX', 'AR', 'CFLAGS', 'LDFLAGS'])
 configure_env = dict((k, os.environ[k]) for k in os.environ if k in env_keys)
 if configure_env:
-    config_str = ' '.join([k + '=' + pipes.quote(configure_env[k])
+    config_str = ' '.join([k + '=' + quote(configure_env[k])
                            for k in configure_env])
     n.variable('configure_env', config_str + '$ ')
 n.newline()
