--- a/build.sh
+++ b/build.sh
@@ -33,7 +33,7 @@
 	echo '##########'
 	echo ""
 	cd $BASEDIR
-	python $BASEDIR/builder_script/make.py $BASEDIR/svgs/$F -n $THEMENAME -o $BASEDIR/built_themes/$THEMENAME --fps 60 -f
+	python3 $BASEDIR/builder_script/make.py $BASEDIR/svgs/$F -n $THEMENAME -o $BASEDIR/built_themes/$THEMENAME --fps 60 -f
 
 	# Create archives if enabled and add optional install instructions
 	if [ "$ARCHIVES" = "1" ];
