--- a/lib/xpp/CMakeLists.txt
+++ b/lib/xpp/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
 project(xpp)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${PROJECT_SOURCE_DIR}/cmake)
--- a/lib/i3ipcpp/CMakeLists.txt
+++ b/lib/i3ipcpp/CMakeLists.txt
@@ -1,6 +1,6 @@
 # Project setup
 
-cmake_minimum_required(VERSION 3.1 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.5.0 FATAL_ERROR)
 project(i3ipc++ CXX)
 
 option(WITH_LOGGING "Build with log support" OFF)
