--- a/src/net.cpp
+++ b/src/net.cpp
@@ -1087,7 +1087,7 @@

 #ifndef __WXMSW__
         r = UPNP_AddPortMapping(urls.controlURL, data.first.servicetype,
-	                        port, port, lanaddr, 0, "TCP", 0);
+	                        port, port, lanaddr, 0, "TCP", 0, "0");
 #else
         r = UPNP_AddPortMapping(urls.controlURL, data.first.servicetype,
	                        port, port, lanaddr, 0, "TCP", 0, "0");
