diff --git a/src/init.cpp b/src/init.cpp
index bc7e714..295e20f 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -6,7 +6,6 @@
 #include "bitcoinrpc.h"
 #include "net.h"
 #include "init.h"
-#include "strlcpy.h"
 #include <boost/filesystem.hpp>
 #include <boost/filesystem/fstream.hpp>
 #include <boost/interprocess/sync/file_lock.hpp>
diff --git a/src/irc.cpp b/src/irc.cpp
index fea0aa2..fd5d8c4 100644
--- a/src/irc.cpp
+++ b/src/irc.cpp
@@ -5,7 +5,6 @@
 #include "headers.h"
 #include "irc.h"
 #include "net.h"
-#include "strlcpy.h"
 
 using namespace std;
 using namespace boost;
diff --git a/src/net.cpp b/src/net.cpp
index 4d2f651..7df2b77 100644
--- a/src/net.cpp
+++ b/src/net.cpp
@@ -7,7 +7,6 @@
 #include "db.h"
 #include "net.h"
 #include "init.h"
-#include "strlcpy.h"
 
 #ifdef __WXMSW__
 #include <string.h>
diff --git a/src/qt/bitcoin.cpp b/src/qt/bitcoin.cpp
index 30dcca6..21a11af 100644
--- a/src/qt/bitcoin.cpp
+++ b/src/qt/bitcoin.cpp
@@ -15,7 +15,6 @@
 #include "ui_interface.h"
 #include "paymentserver.h"
 #include "../allocators.h"
-#include "../strlcpy.h"
 
 #include <boost/filesystem.hpp>
 
diff --git a/src/util.cpp b/src/util.cpp
index fc51717..50ce19f 100644
--- a/src/util.cpp
+++ b/src/util.cpp
@@ -2,7 +2,6 @@
 // Distributed under the MIT/X11 software license, see the accompanying
 // file license.txt or http://www.opensource.org/licenses/mit-license.php.
 #include "headers.h"
-#include "strlcpy.h"
 #include <boost/program_options/detail/config_file.hpp>
 #include <boost/program_options/parsers.hpp>
 #include <boost/filesystem.hpp>
