--- a/loudmouth/lm-ssl-gnutls.c	2008-10-29 14:45:10.000000000 +0100
+++ b/loudmouth/lm-ssl-gnutls.c.new	2015-04-17 22:35:06.703672306 +0200
@@ -196,17 +196,9 @@
 {
 	int ret;
 	gboolean auth_ok = TRUE;
-	const int cert_type_priority[] =
-		{ GNUTLS_CRT_X509, GNUTLS_CRT_OPENPGP, 0 };
-	const int compression_priority[] =
-		{ GNUTLS_COMP_DEFLATE, GNUTLS_COMP_NULL, 0 };
 
 	gnutls_init (&ssl->gnutls_session, GNUTLS_CLIENT);
 	gnutls_set_default_priority (ssl->gnutls_session);
-	gnutls_certificate_type_set_priority (ssl->gnutls_session,
-					      cert_type_priority);
-	gnutls_compression_set_priority (ssl->gnutls_session,
-					 compression_priority);
 	gnutls_credentials_set (ssl->gnutls_session,
 				GNUTLS_CRD_CERTIFICATE,
 				ssl->gnutls_xcred);
