diff -up rsnapshot-1.3.1/rsnapshot-program.pl.pod2man rsnapshot-1.3.1/rsnapshot-program.pl
--- rsnapshot-1.3.1/rsnapshot-program.pl.pod2man	2013-07-06 00:39:44.500520483 -0700
+++ rsnapshot-1.3.1/rsnapshot-program.pl	2013-07-06 00:43:17.696767540 -0700
@@ -6666,6 +6666,8 @@ additional disk space will be taken up.
 
 =back
 
+=back
+
 Remember that tabs must separate all elements, and that
 there must be a trailing slash on the end of every directory.
 
