From f73df57a1aabae050c55fd66ae03eb0f2e74bf0a Mon Sep 17 00:00:00 2001
From: David Demelier <markand@malikania.fr>
Date: Mon, 3 Feb 2025 15:21:07 +0100
Subject: [PATCH] Fix non-integer cast to pthread_t (#17512)

---
 input/common/linux_common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/input/common/linux_common.c b/input/common/linux_common.c
index 32f7c45dded..dcee89e129c 100644
--- a/input/common/linux_common.c
+++ b/input/common/linux_common.c
@@ -266,7 +266,7 @@ void linux_close_illuminance_sensor(linux_illuminance_sensor_t *sensor)
 
    if (sensor->thread)
    {
-      pthread_t thread = sthread_get_thread_id(sensor->thread);
+      pthread_t thread = (pthread_t)sthread_get_thread_id(sensor->thread);
       sensor->done = true;
 
       if (pthread_cancel(thread) != 0)
