--- a/setup.cfg
+++ b/setup.cfg
@@ -5,7 +5,7 @@
 universal = 1
 
 [tool:pytest]
-addopts = --cov=tinycss2 --flake8 --isort tinycss2/test.py
+addopts = --cov=tinycss2 tinycss2/test.py
 norecursedirs = dist .cache .git build *.egg-info .eggs venv
 
 [egg_info]
--- a/setup.py
+++ b/setup.py
@@ -49,7 +49,7 @@
     setup_requires=pytest_runner,
     test_suite='tinycss2.test',
     tests_require=[
-        'pytest-runner', 'pytest-cov', 'pytest-flake8', 'pytest-isort'],
+        'pytest-runner', 'pytest-cov'],
     extras_require={'test': [
-        'pytest-runner', 'pytest-cov', 'pytest-flake8', 'pytest-isort']},
+        'pytest-runner', 'pytest-cov']},
 )
--- a/tinycss2.egg-info/requires.txt
+++ b/tinycss2.egg-info/requires.txt
@@ -3,5 +3,3 @@
 [test]
 pytest-runner
 pytest-cov
-pytest-flake8
-pytest-isort
