From 5bfaa9d6e5343805a3bcb2387aa96458d580c1b0 Mon Sep 17 00:00:00 2001
From: Mikael Olenfalk <mikael@olenfalk.se>
Date: Thu, 14 Jan 2021 09:09:37 +0000
Subject: [PATCH] Add dummy udev_device_get_usec_since_initialized() impl

---
 udev.h        | 1 +
 udev_device.c | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/udev.h b/udev.h
index 51ba317..bc61ce3 100644
--- a/udev.h
+++ b/udev.h
@@ -22,6 +22,7 @@ struct udev *udev_new(void);
 struct udev *udev_ref(struct udev *udev);
 struct udev *udev_unref(struct udev *udev);
 
+unsigned long long udev_device_get_usec_since_initialized(struct udev_device *udev_device);
 const char *udev_device_get_syspath(struct udev_device *udev_device);
 const char *udev_device_get_sysname(struct udev_device *udev_device);
 const char *udev_device_get_sysnum(struct udev_device *udev_device);
diff --git a/udev_device.c b/udev_device.c
index e189659..87631a5 100644
--- a/udev_device.c
+++ b/udev_device.c
@@ -19,6 +19,11 @@ struct udev_device {
     int refcount;
 };
 
+unsigned long long udev_device_get_usec_since_initialized(struct udev_device *udev_device)
+{
+    return 0ull;
+}
+
 const char *udev_device_get_syspath(struct udev_device *udev_device)
 {
     return udev_device_get_property_value(udev_device, "SYSPATH");
