Only in b: config.log
--- a/configure.in
+++ b/configure.in
@@ -60,7 +60,7 @@
 AC_SUBST(TOTEM_PL_PARSER_VERSION_MINOR)
 AC_SUBST(TOTEM_PL_PARSER_VERSION_MICRO)
 
-pkg_modules="glib-2.0 >= $GLIB_REQS libxml-2.0 gthread-2.0 gio-2.0 >= $GIO_REQS libsoup-gnome-2.4"
+pkg_modules="glib-2.0 >= $GLIB_REQS libxml-2.0 gthread-2.0 gio-2.0 >= $GIO_REQS libsoup-2.4"
 
 ##################################
 # Checking libgmime dependency
--- a/plparse/totem-pl-parser-podcast.c
+++ b/plparse/totem-pl-parser-podcast.c
@@ -25,7 +25,7 @@
 #include <glib.h>
 
 #ifndef TOTEM_PL_PARSER_MINI
-#include <libsoup/soup-gnome.h>
+#include <libsoup/soup.h>
 #include "xmlparser.h"
 #include "totem-pl-parser.h"
 #include "totemplparser-marshal.h"
@@ -579,7 +579,6 @@
 	GByteArray *data;
 
 	session = soup_session_sync_new_with_options (
-	    SOUP_SESSION_ADD_FEATURE_BY_TYPE, SOUP_TYPE_GNOME_FEATURES_2_26,
 	    SOUP_SESSION_ADD_FEATURE_BY_TYPE, SOUP_TYPE_CONTENT_DECODER,
 	    SOUP_SESSION_USER_AGENT, "iTunes/7.4.1",
 	    SOUP_SESSION_ACCEPT_LANGUAGE_AUTO, TRUE,
