--- a/src/arg_int.c
+++ b/src/arg_int.c
@@ -29,6 +29,7 @@ USA.
 /* #endif */
 
 #include "argtable2.h"
+#include <ctype.h>
 #include <limits.h>
 
 /* local error codes */
--- a/tests/fntests.c
+++ b/tests/fntests.c
@@ -1,5 +1,6 @@
 #include "../src/argtable2.h"
 #include <assert.h>
+#include <string.h>
 
 /* for memory leak debugging */
 #ifdef DMALLOC
--- a/tests/test_file.c
+++ b/tests/test_file.c
@@ -21,6 +21,7 @@ USA.
 
 #include "../src/argtable2.h"
 #include <stdlib.h>
+#include <string.h>
 
 /* for memory leak debugging */
 #ifdef DMALLOC
