From fe8d2cdedff30ce3173c6f5ee518cc4164b4d2c8 Mon Sep 17 00:00:00 2001
From: Chris Talbot <chris@talbothome.com>
Date: Tue, 29 Nov 2022 16:56:03 -0500
Subject: [PATCH] window: Cast timeout string

---
 src/aspam-window.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/aspam-window.c b/src/aspam-window.c
index 5b7ebe6..14efdab 100644
--- a/src/aspam-window.c
+++ b/src/aspam-window.c
@@ -76,7 +76,7 @@ aspam_window_window_populate (ASpamWindow *self)
 
   gtk_entry_buffer_set_text (GTK_ENTRY_BUFFER (self->new_whitelist_text_buffer), "", -1);
   callback_timeout = aspam_settings_get_callback_timeout (settings);
-  callback_timeout_string = g_strdup_printf ("%li", callback_timeout);
+  callback_timeout_string = g_strdup_printf ("%lu", (long unsigned) callback_timeout);
 
   gtk_entry_buffer_set_text (GTK_ENTRY_BUFFER (self->callback_timeout_text_buffer),
                              callback_timeout_string, -1);
-- 
GitLab

