Update numpy ptp function for numpy 2.0 compatibility.

--- chaospy-4.3.17-origin/chaospy/distributions/kernel/baseclass.py
+++ chaospy-4.3.17/chaospy/distributions/kernel/baseclass.py
@@ -48,7 +48,7 @@
         if h_mat is None:
 
             if estimator_rule in ("scott", "silverman"):
-                qrange = numpy.quantile(samples, [0.25, 0.75], axis=1).ptp(axis=0)
+                qrange = numpy.ptp(numpy.quantile(samples, [0.25, 0.75], axis=1), axis=0)
                 scale = numpy.min([numpy.std(samples, axis=1), qrange / 1.34], axis=0)
                 factor = samples.shape[1]
                 if estimator_rule == "silverman":
