--- a/auto/modules/php
+++ b/auto/modules/php
@@ -102,11 +102,7 @@
                          `${NXT_PHP_CONFIG} --libs`"
 
     else
-        if [ $NXT_PHP_MAJOR_VERSION -ge 8 ]; then
-            NXT_PHP_LIB="-lphp"
-        else
-            NXT_PHP_LIB="-lphp${NXT_PHP_VERSION%%.*}"
-        fi
+        NXT_PHP_LIB="-lphp${NXT_PHP_MAJOR_VERSION}${NXT_PHP_MINOR_VERSION}"
 
         if [ "$NXT_PHP_LIB_PATH" != "" ]; then
             # "php-config --ldflags" does not contain path to libphp, but
