diff --git a/src/ansiblelint/app.py b/src/ansiblelint/app.py
index 1fc085d..c0e80b8 100644
--- a/src/ansiblelint/app.py
+++ b/src/ansiblelint/app.py
@@ -338,7 +338,7 @@ class App:
 
         # on offline mode and when run under pre-commit we do not want to
         # check for updates.
-        if not self.options.offline and os.environ.get("PRE_COMMIT", "0") != "1":
+        if False and not self.options.offline and os.environ.get("PRE_COMMIT", "0") != "1":
             version_warning = get_version_warning()
             if version_warning:
                 msg += f"\n{version_warning}"
