diff -upr notmuch-0.24.1.orig/test/T030-config.sh notmuch-0.24.1/test/T030-config.sh
--- notmuch-0.24.1.orig/test/T030-config.sh	2017-08-17 19:19:19.245017863 +0200
+++ notmuch-0.24.1/test/T030-config.sh	2017-08-17 19:19:35.725123923 +0200
@@ -45,8 +45,8 @@ test_expect_equal "$(notmuch config get
 test_begin_subtest "List all items"
 notmuch config list 2>&1 | notmuch_config_sanitize > OUTPUT
 cat <<EOF > EXPECTED
-Error opening database at MAIL_DIR/.notmuch: No such file or directory
 database.path=MAIL_DIR
+Error opening database at MAIL_DIR/.notmuch: No such file or directory
 user.name=Notmuch Test Suite
 user.primary_email=test_suite@notmuchmail.org
 user.other_email=test_suite_other@notmuchmail.org;test_suite@otherdomain.org
diff -upr notmuch-0.24.1.orig/test/T650-regexp-query.sh notmuch-0.24.1/test/T650-regexp-query.sh
--- notmuch-0.24.1.orig/test/T650-regexp-query.sh	2017-08-17 20:27:35.577369368 +0200
+++ notmuch-0.24.1/test/T650-regexp-query.sh	2017-08-17 20:28:06.570896484 +0200
@@ -99,7 +99,7 @@ test_begin_subtest "regexp error reporti
 notmuch search 'from:/unbalanced[/' 1>OUTPUT 2>&1
 cat <<EOF > EXPECTED
 notmuch search: A Xapian exception occurred
-A Xapian exception occurred parsing query: Invalid regular expression
+A Xapian exception occurred parsing query: Missing ']'
 Query string was: from:/unbalanced[/
 EOF
 test_expect_equal_file EXPECTED OUTPUT
