diff --git a/build-aux/checks.sh b/build-aux/checks.sh
index c277f77..bb2f269 100755
--- a/build-aux/checks.sh
+++ b/build-aux/checks.sh
@@ -204,10 +204,10 @@ check_potfiles() {
         fi
     fi
 
-    ui_files=(`/usr/bin/grep -lIrE '[C]?_\(' src/gtk/*.blp`)
+    ui_files=(`grep -lIrE '[C]?_\(' src/gtk/*.blp`)
 
     # Get Rust files with regex 'i18n[!]?\('.
-    rs_files=(`/usr/bin/grep -lIrE 'i18n[!]?\(' --exclude=src/i18n.rs src/*`)
+    rs_files=(`grep -lIrE 'i18n[!]?\(' --exclude=src/i18n.rs src/*`)
 
     # Remove common files
     to_diff1=("${ui_potfiles[@]}")
