--- a/builder/MyBuilder.pm
+++ b/builder/MyBuilder.pm
@@ -14,8 +14,8 @@
         generate_ppport_h    => 'src/ppport.h',
         c_source             => 'src',
         xs_files             => { 'src/EV__Hiredis.xs' => 'lib/EV/Hiredis.xs' },
-        include_dirs         => ['src', 'deps/hiredis', "${installsitearch}/EV", $installsitearch],
-        extra_linker_flags   => ["deps/hiredis/libhiredis$Config{lib_ext}"],
+        include_dirs         => ['src', '/usr/include/hiredis', "${installsitearch}/EV", $installsitearch],
+        extra_linker_flags   => ["-lhiredis"],
     );
 
     my $make;
@@ -30,7 +30,6 @@
         $make = $Config{make};
     }
 
-    $self->do_system($make, '-C', 'deps/hiredis', 'static');
     return $self;
 }
 
