From 5c1bc43771fbee6a483865169cd1ee3ce31e6de1 Mon Sep 17 00:00:00 2001
From: Thomas Liske <thomas@fiasko-nw.net>
Date: Sun, 20 Apr 2025 11:41:27 +0200
Subject: [PATCH] fix exception when WGPEER_A_LAST_HANDSHAKE_TIME is None

---
 wgnlpy/wireguardpeer.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wgnlpy/wireguardpeer.py b/wgnlpy/wireguardpeer.py
index ddf24a4..7f4de2b 100644
--- a/wgnlpy/wireguardpeer.py
+++ b/wgnlpy/wireguardpeer.py
@@ -27,7 +27,7 @@ class WireGuardPeer(object):
         self.endpoint = peer.get_attr('WGPEER_A_ENDPOINT')
         self.persistent_keepalive_interval = peer.get_attr('WGPEER_A_PERSISTENT_KEEPALIVE_INTERVAL')
         self.last_handshake_time = peer.get_attr('WGPEER_A_LAST_HANDSHAKE_TIME')
-        if not self.last_handshake_time > 0:
+        if self.last_handshake_time is not None and not self.last_handshake_time > 0:
             self.last_handshake_time = None
         self.rx_bytes = peer.get_attr('WGPEER_A_RX_BYTES')
         self.tx_bytes = peer.get_attr('WGPEER_A_TX_BYTES')
-- 
2.39.5

