diff -urN notcurses-2.2.3.orig/src/lib/direct.cpp notcurses-2.2.3/src/lib/direct.cpp
--- notcurses-2.2.3.orig/src/lib/direct.cpp	2021-03-14 23:07:33.248882041 -0600
+++ notcurses-2.2.3/src/lib/direct.cpp	2021-03-14 23:08:35.776019992 -0600
@@ -708,6 +708,7 @@
   int r = -1;
   if(stylebits == 0 && n->tcache.sgr0){
     r = term_emit(n->tcache.sgr0, n->ttyfp, false);
+    fprintf(stderr, "term_emit/1: %d; n->tcache.sgr0 = %s, ttyfp = %p", r, n->tcache.sgr0, n->ttyfp);
   }else if(n->tcache.sgr){
     r = term_emit(tiparm(n->tcache.sgr, stylebits & NCSTYLE_STANDOUT,
                          stylebits & NCSTYLE_UNDERLINE,
@@ -717,15 +718,18 @@
                          stylebits & NCSTYLE_BOLD,
                          stylebits & NCSTYLE_INVIS,
                          stylebits & NCSTYLE_PROTECT, 0), out, false);
+    fprintf(stderr, "term_emit/2: %d; ttyfp = %p", r, n->ttyfp);
   }
   // sgr resets colors, so set them back up if not defaults
   if(r == 0){
     // FIXME need to handle palette-indexed colors
     if(!ncdirect_fg_default_p(n)){
       r |= ncdirect_set_fg_rgb(n, channels_fg_rgb(n->channels));
+      fprintf(stderr, "ncdirect_set_fg_rgb/3: %d; ttyfp = %p", r, n->ttyfp);
     }
     if(!ncdirect_bg_default_p(n)){
       r |= ncdirect_set_bg_rgb(n, channels_bg_rgb(n->channels));
+      fprintf(stderr, "ncdirect_set_bg_rgb/4: %d; ttyfp = %p", r, n->ttyfp);
     }
   }
   return r;
