diff -urN notcurses-2.2.3.orig/src/lib/direct.cpp notcurses-2.2.3/src/lib/direct.cpp
--- notcurses-2.2.3.orig/src/lib/direct.cpp	2021-03-14 23:07:33.248882041 -0600
+++ notcurses-2.2.3/src/lib/direct.cpp	2021-03-14 23:30:28.547072308 -0600
@@ -706,6 +706,10 @@
 static inline int
 ncdirect_style_emit(ncdirect* n, unsigned stylebits, FILE* out){
   int r = -1;
+
+  fprintf(stderr, "ncdirect_style_emit/1: stylebits=%x, out=%p, sgr=%p, sgr0=%p\n", stylebits, out,
+          n->tcache.sgr, n->tcache.sgr0);
+
   if(stylebits == 0 && n->tcache.sgr0){
     r = term_emit(n->tcache.sgr0, n->ttyfp, false);
   }else if(n->tcache.sgr){
@@ -718,6 +722,9 @@
                          stylebits & NCSTYLE_INVIS,
                          stylebits & NCSTYLE_PROTECT, 0), out, false);
   }
+
+  fprintf(stderr, "ncdirect_style_emit/2: r=%d\n", r);
+
   // sgr resets colors, so set them back up if not defaults
   if(r == 0){
     // FIXME need to handle palette-indexed colors
@@ -728,6 +735,8 @@
       r |= ncdirect_set_bg_rgb(n, channels_bg_rgb(n->channels));
     }
   }
+
+  fprintf(stderr, "ncdirect_style_emit/3: r=%d\n", r);
   return r;
 }
 
@@ -783,18 +792,22 @@
 int ncdirect_set_styles(ncdirect* n, unsigned stylebits){
   uint32_t stylemask = stylebits;
   if(ncdirect_style_emit(n, stylemask, n->ttyfp) == 0){
+    fprintf(stderr, "ncdirect_set_styles/1\n");
     n->stylemask &= !(NCSTYLE_ITALIC | NCSTYLE_STRUCK); // sgr clears both
     if(term_setstyle(n->ttyfp, n->stylemask, stylemask, NCSTYLE_ITALIC,
                      n->tcache.italics, n->tcache.italoff)){
+      fprintf(stderr, "ncdirect_set_styles/2\n");
       return -1;
     }
     if(term_setstyle(n->ttyfp, n->stylemask, stylemask, NCSTYLE_STRUCK,
                      n->tcache.struck, n->tcache.struckoff)){
+      fprintf(stderr, "ncdirect_set_styles/3\n");
       return -1;
     }
     n->stylemask = stylemask;
     return 0;
   }
+  fprintf(stderr, "ncdirect_set_styles/4\n");
   return -1;
 }
 
