https://github.com/microsoft/onnxruntime/issues/25815

diff -Nurp a/cmake/external/abseil-cpp.cmake b/cmake/external/abseil-cpp.cmake
--- a/cmake/external/abseil-cpp.cmake	2025-08-12 16:38:01.000000000 +0000
+++ b/cmake/external/abseil-cpp.cmake	2025-09-14 18:53:16.688779651 +0000
@@ -114,7 +114,6 @@ absl::absl_check
 absl::hash_function_defaults
 absl::function_ref
 absl::city
-absl::low_level_hash
 absl::fixed_array
 absl::variant
 absl::meta
diff -Nurp a/cmake/onnxruntime_webassembly.cmake b/cmake/onnxruntime_webassembly.cmake
--- a/cmake/onnxruntime_webassembly.cmake	2025-08-12 16:38:01.000000000 +0000
+++ b/cmake/onnxruntime_webassembly.cmake	2025-09-14 18:54:51.529028737 +0000
@@ -278,7 +278,7 @@ else()
     target_compile_options(absl_log_globals PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
     target_compile_options(absl_hash PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
     target_compile_options(absl_city PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
-    target_compile_options(absl_low_level_hash PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
+    target_compile_options(absl_hash PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
     target_compile_options(absl_bad_variant_access PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
     target_compile_options(absl_vlog_config_internal PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
     target_compile_options(absl_synchronization PRIVATE ${SMEMORY_FLAG} -Wno-experimental)
