This is downstream patch only.

--- a/src/compiler/crystal/command/deps.cr
+++ b/src/compiler/crystal/command/deps.cr
@@ -5,7 +5,7 @@
   private def deps
     path_to_shards = `which shards`.chomp
     if path_to_shards.empty?
-      error "`shards` executable is missing. Please install shards: https://github.com/crystal-lang/shards"
+      error "`shards` executable is missing. Please install shards: apk add shards"
     end
 
     status = Process.run(path_to_shards, args: options, output: Process::Redirect::Inherit, error: Process::Redirect::Inherit)
