Fix symlink targets containing DESTDIR. Assuming LN_S is 'ln -s'

--- a/doc/user/Makefile.in
+++ b/doc/user/Makefile.in
@@ -763,7 +763,7 @@
 	$(MKDIR_P) "$(DESTDIR)$(infodir)"
 	for i in "$(DESTDIR)$(htmldir)/"*.png; do	\
 	  (cd "$(DESTDIR)$(infodir)";			\
-	   rm -f "`basename $$i`"; $(LN_S) "$$i");	\
+	   rm -f "`basename $$i`"; $(LN_S) "$(htmldir)/`basename $$i`");	\
 	done
 
 uninstall-local:
