--- a/rake.gemspec
+++ b/rake.gemspec
@@ -30,8 +30,7 @@
     "source_code_uri"   => "https://github.com/ruby/rake/tree/v#{s.version}",
   }
 
-  s.files = %x[git ls-files -z].split("\x0").reject { |f| f.match(%r{^(test|spec|features|\.github)/}) } -
-            %w[.rubocop.yml .gitignore .travis.yml appveyor.yml]
+  s.files = Dir["exe/*", "lib/**/*.rb"]
   s.bindir = "exe"
   s.executables = s.files.grep(%r{^exe/}) { |f| File.basename(f) }
   s.require_paths = ["lib".freeze]
