diff --git a/httpbin/filters.py b/httpbin/filters.py
index 4deeaaa..c6268b6 100644
--- a/httpbin/filters.py
+++ b/httpbin/filters.py
@@ -10,7 +10,7 @@ This module provides response filter decorators.
 import gzip as gzip2
 import zlib
 
-import brotlicffi as _brotli
+import brotli as _brotli
 
 from six import BytesIO
 from decimal import Decimal
diff --git a/pyproject.toml b/pyproject.toml
index 192fdab..5238117 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,7 +31,7 @@ classifiers = [
     "Programming Language :: Python :: 3.12",
 ]
 dependencies = [
-    "brotlicffi",
+    "brotli",
     "decorator",
     "flask >= 2.2.4",
     'greenlet < 3.0; python_version<"3.12"',
