diff --git a/src/procattr.c b/src/procattr.c
index 48dd8af..8fa31d8 100644
--- a/src/procattr.c
+++ b/src/procattr.c
@@ -22,14 +22,6 @@ static pthread_key_t destructor_key;
 static int destructor_key_initialized = 0;
 static __thread char destructor_initialized;
 
-#ifndef __BIONIC__
-/* Bionic declares this in unistd.h and has a definition for it */
-static pid_t gettid(void)
-{
-	return syscall(__NR_gettid);
-}
-#endif
-
 static void procattr_thread_destructor(void __attribute__((unused)) *unused)
 {
 	if (prev_current != UNSET)
