Author: Vlad Glagolev <scm@vaygr.net>
URL: N/A
Summary: fixes build with groff
--- a/Makefile.am.orig
+++ b/Makefile.am
@@ -118,7 +118,7 @@
 
 .8.8.html:
 	@rm -f $@
-	($(TBL) $< | $(NROFF) -mdoc2html > $@) || (rm -f $@; exit 1)
+	($(TBL) $< | $(NROFF) -mhtml > $@) || (rm -f $@; exit 1)
 
 .5.5.ps:
 	@rm -f $@
@@ -126,7 +126,7 @@
 
 .5.5.html:
 	@rm -f $@
-	($(TBL) $< | $(NROFF) -mdoc2html > $@) || (rm -f $@; exit 1)
+	($(TBL) $< | $(NROFF) -mhtml > $@) || (rm -f $@; exit 1)
 
 .cat5.cat5-dist:
 	@rm -f $@
