diff --git a/postfix/makedefs b/postfix/makedefs
index 64b42f44..aea15d6f 100644
--- a/makedefs
+++ b/makedefs
@@ -228,19 +228,6 @@ case $# in
  *) echo usage: $0 [system release] 1>&2; exit 1;;
 esac
 
-case "$SYSTEM" in
- Linux)
-    case "`PATH=/bin:/usr/bin ldd /bin/sh`" in
-     *-musl-*)
-	case "$CCARGS" in
-	 *-DNO_DNSSEC*) ;;
-	 *) echo Warning: libc-musl breaks DANE/TLSA security. 1>&2
-	    echo This build will not support DANE/TLSA. 1>&2
-	    CCARGS="$CCARGS -DNO_DNSSEC";;
-	esac;;
-    esac;;
-esac
-
 case "$SYSTEM.$RELEASE" in
    SCO_SV.3.2)	SYSTYPE=SCO5
 		# Use the native compiler by default
diff --git a/postfix/src/dns/dns_lookup.c b/postfix/src/dns/dns_lookup.c
index 11c92813..b3bd537a 100644
--- a/src/dns/dns_lookup.c
+++ b/src/dns/dns_lookup.c
@@ -350,7 +350,9 @@ static int dns_res_query(const char *name, int class, int type,
 	if (msg_verbose)
 	    msg_info("res_mkquery() failed");
 	return (len);
-    } else if ((len = res_send(msg_buf, len, answer, anslen)) < 0) {
+    }
+    msg_buf[3] |= 32; // AD flag
+    if ((len = res_send(msg_buf, len, answer, anslen)) < 0) {
 	SET_H_ERRNO(TRY_AGAIN);
 	if (msg_verbose)
 	    msg_info("res_send() failed");
@@ -491,11 +493,12 @@ static int dns_query(const char *name, int type, unsigned flags,
     for (;;) {
 	_res.options &= ~saved_options;
 	_res.options |= flags;
-	if (keep_notfound && var_dns_ncache_ttl_fix) {
+	if (1) {
 #ifdef HAVE_RES_SEND
 	    len = dns_res_query((char *) name, C_IN, type, reply->buf,
 				reply->buf_len);
 #else
+#error HAVE_RES_SEND not defined
 	    var_dns_ncache_ttl_fix = 0;
 	    msg_warn("system library does not support %s=yes"
 		     " -- ignoring this setting", VAR_DNS_NCACHE_TTL_FIX);
