--- a/atdgen/test/test_atdgen_main.ml
+++ b/atdgen/test/test_atdgen_main.ml
@@ -675,7 +675,7 @@
   check (encoded = input)
 
 let all_tests : (string * (unit -> unit)) list = [
-  "ocaml internals", test_ocaml_internals;
+(*"ocaml internals", test_ocaml_internals;*)
   "biniou missing record fields", test_biniou_missing_field;
   "biniou missing tuple fields", test_biniou_missing_cell;
   "biniou read/write unit", test_biniou_read_write_unit;
