diff --git a/lib/nfs_v4.c b/lib/nfs_v4.c
index f6f0149..a0554f6 100644
--- a/lib/nfs_v4.c
+++ b/lib/nfs_v4.c
@@ -523,7 +523,7 @@ nfs_parse_attributes(struct nfs_context *nfs, struct nfs4_cb_data *data,
         len -= 8;
         /* FSID */
         CHECK_GETATTR_BUF_SPACE(len, 16);
-        st->nfs_dev = ((uint64_t *)buf)[0] ^ ((uint64_t *)buf)[1];
+        st->nfs_dev = ((uint64_t *)(void *)buf)[0] ^ ((uint64_t *)(void *)buf)[1];
         buf += 16;
         len -= 16;
         /* Inode */
@@ -600,7 +600,7 @@ nfs_parse_attributes(struct nfs_context *nfs, struct nfs4_cb_data *data,
         len -= pad;
         /* raw device */
         CHECK_GETATTR_BUF_SPACE(len, 8);
-        st->nfs_rdev = specdata4_to_rdev((uint32_t *)buf);
+        st->nfs_rdev = specdata4_to_rdev((uint32_t *)(void *)buf);
         buf += 8;
         len -= 8;
         /* Space Used */
diff --git a/lib/socket.c b/lib/socket.c
index 1baf10d..d46d811 100644
--- a/lib/socket.c
+++ b/lib/socket.c
@@ -718,12 +718,12 @@ rpc_read_from_socket(struct rpc_context *rpc)
                         case IPPROTO_IP:
                                 sin = (struct sockaddr_in *)&rpc->udp_dst;
                                 sin->sin_family = AF_INET;
-                                sin->sin_addr.s_addr = ((struct in_pktinfo *)CMSG_DATA(cmsg))->ipi_addr.s_addr;
+                                sin->sin_addr.s_addr = ((struct in_pktinfo *)(void *)CMSG_DATA(cmsg))->ipi_addr.s_addr;
                                 break;
                         case IPPROTO_IPV6:
                                 sin6 = (struct sockaddr_in6 *)&rpc->udp_dst;
                                 sin6->sin6_family = AF_INET6;
-                                memcpy(&sin6->sin6_addr.s6_addr[0], &((struct in6_pktinfo *)CMSG_DATA(cmsg))->ipi6_addr.s6_addr[0], 16);
+                                memcpy(&sin6->sin6_addr.s6_addr[0], &((struct in6_pktinfo *)(void *)CMSG_DATA(cmsg))->ipi6_addr.s6_addr[0], 16);
                                 break;
                         }
                         break;
