diff --git a/retro-gtk/retro-core.c b/retro-gtk/retro-core.c
index 6d5c6e2..800e677 100644
--- a/retro-gtk/retro-core.c
+++ b/retro-gtk/retro-core.c
@@ -1467,7 +1467,7 @@ retro_core_get_memory_size (RetroCore       *self,
                             RetroMemoryType  memory_type)
 {
   g_autoptr(GError) error = NULL;
-  gsize size;
+  guint64 size;
   IpcRunner *proxy;
 
   g_return_val_if_fail (RETRO_IS_CORE (self), 0UL);
