--- a/src/preferences.c.orig	2013-11-11 14:39:32.000000000 +0400
+++ b/src/preferences.c	2013-11-11 14:40:03.970776522 +0400
@@ -381,7 +381,7 @@
 	}	
 	/**now go through and make sure we have no duplicates */
 	for (i=0;NULL != keylist[i].name; ++i){
-		if(0 !=  keylist[i].keyval[0]){
+		if(NULL != keylist[i].keyval && 0 != keylist[i].keyval[0]){
 			/**see if it exists elsewhere  */
 			for (l=0;NULL != keylist[l].name; ++l){
 				if(l!=i && 0 != keylist[l].keyval[0]){
