diff -rupN a/meson.build b/meson.build
--- a/meson.build	2026-09-18 21:39:03.000000000 +0000
+++ b/meson.build	2026-09-19 00:19:07.399153902 +0000
@@ -519,7 +519,6 @@ if tests.allowed()
 
       c_args: qjs_c_args,
       dependencies: [qjs_dep, qjs_libc_dep],
-      build_by_default: false,
     ),
   )
 
@@ -533,7 +532,6 @@ if tests.allowed()
       'libunicode.c',
 
       c_args: qjs_c_args,
-      build_by_default: false,
     ),
   )
 
@@ -546,7 +544,6 @@ if tests.allowed()
 
       c_args: qjs_c_args,
       dependencies: [qjs_dep, qjs_libc_dep],
-      build_by_default: false,
     ),
   )
 endif
