--- a/Makefile	2025-02-16 17:48:32.000000000 -0500
+++ b/Makefile	2025-02-18 21:55:33.836669921 -0500
@@ -101,8 +101,8 @@
 $(LUAJIT_INCLUDE_DIR): ; git submodule update --init
 
 # Native binary for whatever platform you're currently on
-fennel-bin: src/launcher.fnl $(BIN_LUA_DIR)/src/lua $(NATIVE_LUA_LIB) fennel
-	$(COMPILE_ARGS) $(BIN_LUA_DIR)/src/lua fennel \
+fennel-bin: src/launcher.fnl $(BIN_LUA_DIR) $(NATIVE_LUA_LIB) fennel
+	$(COMPILE_ARGS) $(BIN_LUA_DIR) fennel \
 		--no-compiler-sandbox --compile-binary \
 		$< $@ $(NATIVE_LUA_LIB) $(LUA_INCLUDE_DIR)
 
@@ -111,8 +111,8 @@
 		--no-compiler-sandbox --compile-binary \
 		$< $@ $(NATIVE_LUAJIT_LIB) $(LUAJIT_INCLUDE_DIR)
 
-$(BIN_LUA_DIR)/src/lua: $(LUA_INCLUDE_DIR) ; make -C $(BIN_LUA_DIR)
-$(NATIVE_LUA_LIB): $(LUA_INCLUDE_DIR) ; $(MAKE) -C $(BIN_LUA_DIR)/src liblua.a
+$(BIN_LUA_DIR): $(LUA_INCLUDE_DIR)
+$(NATIVE_LUA_LIB): $(LUA_INCLUDE_DIR)
 $(NATIVE_LUAJIT_LIB): $(LUAJIT_INCLUDE_DIR)
 	$(MAKE) -C $(BIN_LUAJIT_DIR) BUILDMODE=static
 
