--- ./src/modules/engines/software_x11/evas_xcb_outbuf.c.orig
+++ ./src/modules/engines/software_x11/evas_xcb_outbuf.c
@@ -871,7 +871,7 @@
           }
         else
           {
-             int pixelb = evas_software_xcb_output_buffer_depth(obr->xob) / 8;
+             int pixelb = evas_software_xcb_output_buffer_depth(obr->xcbob) / 8;
              int run;
              int dstjump;
              
