diff --git a/CMakeLists.txt b/CMakeLists.txt
index 241155d..09559b7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -274,7 +274,7 @@ FILE( GLOB app_menu "${CMAKE_CURRENT_SOURCE_DIR}/resources/menu_data/aqemu.deskt
 INSTALL( FILES ${app_menu} DESTINATION share/applications )
 
 FILE( GLOB app_menu "${CMAKE_CURRENT_SOURCE_DIR}/resources/menu_data/aqemu.appdata.xml" )
-INSTALL( FILES ${app_menu} DESTINATION share/appdata )
+INSTALL( FILES ${app_menu} DESTINATION share/metainfo )
 
 # Create docs .bz2 files
 MACRO( create_docs _sources )
