--- lsof_4.87_src/dialects/linux/dlsof.h
+++ lsof_4.87_src/dialects/linux/dlsof.h
@@ -49,12 +49,7 @@
 #include <setjmp.h>
 #include <unistd.h>
 #include <netinet/in.h>
-
-# if	defined(GLIBCV) || defined(__UCLIBC__)
 #include <netinet/tcp.h>
-# else	/* !defined(GLIBCV) && !defined(__UCLIBC__) */
-#include <linux/tcp.h>
-# endif	/* defined(GLIBCV) || defined(__UCLIBC__) */
 
 # if	!defined(HASNORPC_H)
 #include <rpc/rpc.h>
