--- old/buildSrc/linux.gradle
+++ new/buildSrc/linux.gradle
@@ -45,10 +45,6 @@
         "-fstack-protector",
         "-Wextra", "-Wall", "-Wformat-security", "-Wno-unused", "-Wno-parentheses", "-Werror=implicit-function-declaration", "-Werror=trampolines"] // warning flags
 
-if (!IS_64) {
-    commonFlags += "-m32"
-}
-
 // Specify the compilation parameters and link parameters
 def ccFlags = [
         commonFlags, "-I$JDK_HOME/include", "-I$JDK_HOME/include/linux", "-c",
--- old/modules/javafx.media/src/main/native/gstreamer/projects/linux/avplugin/Makefile
+++ new/modules/javafx.media/src/main/native/gstreamer/projects/linux/avplugin/Makefile
@@ -64,11 +64,6 @@
 LDFLAGS  += $(shell pkg-config --libs libavcodec libavformat)
 endif
 
-ifeq ($(ARCH), x32)
-    CFLAGS += -m32
-    LDFLAGS += -m32
-endif
-
 SOURCES = av/fxavcodecplugin.c  \
           av/avelement.c        \
           av/decoder.c          \
--- old/modules/javafx.media/src/main/native/gstreamer/projects/linux/fxplugins/Makefile
+++ new/modules/javafx.media/src/main/native/gstreamer/projects/linux/fxplugins/Makefile
@@ -55,11 +55,6 @@
           -z relro \
           -static-libgcc -static-libstdc++ -Wl,--gc-sections
 
-ifeq ($(ARCH), x32)
-    CFLAGS += -m32
-    LDFLAGS += -m32
-endif
-
 SOURCES = fxplugins.c                        \
           progressbuffer/progressbuffer.c    \
           progressbuffer/hlsprogressbuffer.c \
--- old/modules/javafx.media/src/main/native/gstreamer/projects/linux/gstreamer-lite/Makefile
+++ new/modules/javafx.media/src/main/native/gstreamer/projects/linux/gstreamer-lite/Makefile
@@ -80,11 +80,6 @@
           -z relro \
           -static-libgcc -static-libstdc++ -Wl,--gc-sections
 
-ifeq ($(ARCH), x32)
-    CFLAGS += -m32
-    LDFLAGS += -m32
-endif
-
 SOURCES = \
 gstreamer/gst/gst.c \
             gstreamer/gst/gstallocator.c \
--- old/modules/javafx.media/src/main/native/jfxmedia/projects/linux/Makefile
+++ new/modules/javafx.media/src/main/native/jfxmedia/projects/linux/Makefile
@@ -75,11 +75,6 @@
 	CFLAGS += -g -Wall -D_DEBUG
 endif
 
-ifeq ($(ARCH), x32)
-    CFLAGS += -m32
-    LDFLAGS += -m32
-endif
-
 CPP_SOURCES = \
         jni/com_sun_media_jfxmedia_logging_Logger.cpp           \
         jni/JavaBandsHolder.cpp 				\
