diff -ru openvpn-2.0.9~/init.c openvpn-2.0.9/init.c
--- openvpn-2.0.9~/init.c	2008-05-13 23:21:30.000000000 +0200
+++ openvpn-2.0.9/init.c	2008-05-13 23:22:51.000000000 +0200
@@ -347,7 +347,8 @@
 	msg (M_FATAL|M_OPTERR,
 	     "options --mktun or --rmtun should only be used together with --dev");
       tuncfg (options->dev, options->dev_type, options->dev_node,
-	      options->tun_ipv6, options->persist_mode);
+	      options->tun_ipv6, options->persist_mode,
+	      &options->tuntap_options);
       return true;
     }
 #endif
diff -ru openvpn-2.0.9~/tun.c openvpn-2.0.9/tun.c
--- openvpn-2.0.9~/tun.c	2008-05-13 23:21:30.000000000 +0200
+++ openvpn-2.0.9/tun.c	2008-05-13 23:22:51.000000000 +0200
@@ -1095,13 +1095,14 @@
 #ifdef TUNSETPERSIST
 
 void
-tuncfg (const char *dev, const char *dev_type, const char *dev_node, bool ipv6, int persist_mode)
+tuncfg (const char *dev, const char *dev_type, const char *dev_node, bool ipv6, int persist_mode, const struct tuntap_options *options)
 {
   struct tuntap *tt;
 
   ALLOC_OBJ (tt, struct tuntap);
   clear_tuntap (tt);
   tt->type = dev_type_enum (dev, dev_type);
+  tt->options = *options;
   open_tun (dev, dev_type, dev_node, ipv6, tt);
   if (ioctl (tt->fd, TUNSETPERSIST, persist_mode) < 0)
     msg (M_ERR, "Cannot ioctl TUNSETPERSIST(%d) %s", persist_mode, dev);
diff -ru openvpn-2.0.9~/tun.h openvpn-2.0.9/tun.h
--- openvpn-2.0.9~/tun.h	2008-05-13 23:21:30.000000000 +0200
+++ openvpn-2.0.9/tun.h	2008-05-13 23:22:51.000000000 +0200
@@ -194,7 +194,7 @@
 int read_tun (struct tuntap* tt, uint8_t *buf, int len);
 
 void tuncfg (const char *dev, const char *dev_type, const char *dev_node,
-	     bool ipv6, int persist_mode);
+	     bool ipv6, int persist_mode, const struct tuntap_options *options);
 
 const char *guess_tuntap_dev (const char *dev,
 			      const char *dev_type,
