From 91a777349f232df7601f9e82680b23f8b6c46acd Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Mon, 23 Feb 2026 17:25:09 -0800
Subject: [PATCH] sysuser.conf: change plasma-setup user shell to /bin/sh

---
 files/plasma-setup-sysuser.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/files/plasma-setup-sysuser.conf b/files/plasma-setup-sysuser.conf
index a3deefc..258129c 100644
--- a/files/plasma-setup-sysuser.conf
+++ b/files/plasma-setup-sysuser.conf
@@ -8,4 +8,4 @@
 #
 # /usr/lib/sysusers.d/plasma-setup.conf
 
-u plasma-setup - "Plasma Setup" /run/plasma-setup /bin/bash
+u plasma-setup - "Plasma Setup" /run/plasma-setup /bin/sh
-- 
2.53.0

