diff --git a/scripts/eturnalctl b/scripts/eturnalctl
index 77cb16b..064ab96 100755
--- a/scripts/eturnalctl
+++ b/scripts/eturnalctl
@@ -163,7 +163,7 @@ else
             install -d -m 755 -o "$user" -g "$user" "$RUNTIME_DIRECTORY"
         fi
         arg="$(printf '%s' "$*" | sed 's/[^[:alnum:][:space:]]/\\&/g')"
-        exec su "$user" -c "$cmd $arg"
+        exec su -s '/bin/sh' "$user" -c "$cmd $arg"
     else
         exec "$cmd" "$@"
     fi
