diff --git a/tests/largetransfer.lua b/tests/largetransfer.lua
index 9ada332..5b9727c 100644
--- a/tests/largetransfer.lua
+++ b/tests/largetransfer.lua
@@ -41,7 +41,7 @@ local function runtest()
       copas.sleep(0)
       local skt = socket.tcp()
       skt = copas.wrap(skt, cparams)
-      skt:connect("localhost", 49500)
+      skt:connect("127.0.0.1", 49500)
       skt:send(body)
       print("Writing... 49500... Done!", socket.gettime()-start, err, #body)
       skt = nil
@@ -54,7 +54,7 @@ local function runtest()
       copas.sleep(0)
       local skt = socket.tcp()
       skt = copas.wrap(skt, cparams)
-      skt:connect("localhost", 49501)
+      skt:connect("127.0.0.1", 49501)
       skt:send(body)
       print("Writing... 49501... Done!", socket.gettime()-start, err, #body)
       skt = nil
