--- clutter-1.6.0/clutter/glx/clutter-backend-glx.c
+++ clutter-1.6.0.mod/clutter/glx/clutter-backend-glx.c
@@ -634,6 +634,7 @@
   /* Try and create a GLXWindow to use with extensions dependent on
    * GLX versions >= 1.3 that don't accept regular X Windows as GLX
    * drawables. */
+#if 0
   if (glXQueryVersion (backend_x11->xdpy, &major, &minor) &&
       major == 1 && minor >= 3)
     {
@@ -642,6 +643,7 @@
                                                    backend_glx->dummy_xwin,
                                                    NULL);
     }
+#endif
 
   if (backend_glx->dummy_glxwin)
     dummy_drawable = backend_glx->dummy_glxwin;
--- clutter-1.6.0/clutter/glx/clutter-stage-glx.c
+++ clutter-1.6.0.mod/clutter/glx/clutter-stage-glx.c
@@ -124,6 +124,7 @@
        * GLX versions >= 1.3 that don't accept regular X Windows as GLX
        * drawables.
        */
+#if 0
       if (glXQueryVersion (backend_x11->xdpy, &major, &minor) &&
           major == 1 && minor >= 3 &&
           _clutter_backend_glx_get_fbconfig (backend_glx, &config))
@@ -133,6 +134,7 @@
                                                stage_x11->xwin,
                                                NULL);
         }
+#endif
     }
 
 #ifdef GLX_INTEL_swap_event
