diff --git a/src/cups/backend_common.h b/src/cups/backend_common.h
index 2decab2..b756e8c 100644
--- a/src/cups/backend_common.h
+++ b/src/cups/backend_common.h
@@ -58,25 +58,25 @@
 #define le16_to_cpu(__x) __x
 #define be16_to_cpu(__x) ntohs(__x)
 #define be32_to_cpu(__x) ntohl(__x)
-#define be64_to_cpu(__x) ((__uint64_t)(                         \
-        (((__uint64_t)(__x) & (__uint64_t)0x00000000000000ffULL) << 56) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x000000000000ff00ULL) << 40) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x0000000000ff0000ULL) << 24) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x00000000ff000000ULL) <<  8) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x000000ff00000000ULL) >>  8) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x0000ff0000000000ULL) >> 24) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x00ff000000000000ULL) >> 40) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0xff00000000000000ULL) >> 56)))
+#define be64_to_cpu(__x) ((uint64_t)(                         \
+        (((uint64_t)(__x) & (uint64_t)0x00000000000000ffULL) << 56) |   \
+        (((uint64_t)(__x) & (uint64_t)0x000000000000ff00ULL) << 40) |   \
+        (((uint64_t)(__x) & (uint64_t)0x0000000000ff0000ULL) << 24) |   \
+        (((uint64_t)(__x) & (uint64_t)0x00000000ff000000ULL) <<  8) |   \
+        (((uint64_t)(__x) & (uint64_t)0x000000ff00000000ULL) >>  8) |   \
+        (((uint64_t)(__x) & (uint64_t)0x0000ff0000000000ULL) >> 24) |   \
+        (((uint64_t)(__x) & (uint64_t)0x00ff000000000000ULL) >> 40) |   \
+        (((uint64_t)(__x) & (uint64_t)0xff00000000000000ULL) >> 56)))
 #else
-#define le64_to_cpu(__x) ((__uint64_t)(                         \
-        (((__uint64_t)(__x) & (__uint64_t)0x00000000000000ffULL) << 56) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x000000000000ff00ULL) << 40) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x0000000000ff0000ULL) << 24) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x00000000ff000000ULL) <<  8) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x000000ff00000000ULL) >>  8) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x0000ff0000000000ULL) >> 24) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0x00ff000000000000ULL) >> 40) |   \
-        (((__uint64_t)(__x) & (__uint64_t)0xff00000000000000ULL) >> 56)))
+#define le64_to_cpu(__x) ((uint64_t)(                         \
+        (((uint64_t)(__x) & (uint64_t)0x00000000000000ffULL) << 56) |   \
+        (((uint64_t)(__x) & (uint64_t)0x000000000000ff00ULL) << 40) |   \
+        (((uint64_t)(__x) & (uint64_t)0x0000000000ff0000ULL) << 24) |   \
+        (((uint64_t)(__x) & (uint64_t)0x00000000ff000000ULL) <<  8) |   \
+        (((uint64_t)(__x) & (uint64_t)0x000000ff00000000ULL) >>  8) |   \
+        (((uint64_t)(__x) & (uint64_t)0x0000ff0000000000ULL) >> 24) |   \
+        (((uint64_t)(__x) & (uint64_t)0x00ff000000000000ULL) >> 40) |   \
+        (((uint64_t)(__x) & (uint64_t)0xff00000000000000ULL) >> 56)))
 #define le32_to_cpu(x)							\
 	({								\
 		uint32_t __x = (x);					\
diff --git a/src/testpattern/testpatterny.h b/src/testpattern/testpatterny.h
index 259ea29..fce45eb 100644
--- a/src/testpattern/testpatterny.h
+++ b/src/testpattern/testpatterny.h
@@ -43,6 +43,7 @@ extern int yydebug;
 /* Token type.  */
 #ifndef YYTOKENTYPE
 # define YYTOKENTYPE
+#undef PAGESIZE
   enum yytokentype
   {
     tINT = 258,
