From eceebc4fbf064ca04d0f0a639c8a7c600190170f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 13 Jan 2019 19:07:16 +0100
Subject: [PATCH] adduser: default to /sbin/nologin as shell for system
 accounts

---
 loginutils/adduser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loginutils/adduser.c b/loginutils/adduser.c
index b2b5be5b3..9326a9795 100644
--- a/loginutils/adduser.c
+++ b/loginutils/adduser.c
@@ -235,7 +235,7 @@ int adduser_main(int argc UNUSED_PARAM, char **argv)
 			usegroup = "nogroup";
 		}
 		if (!(opts & OPT_SHELL)) {
-			pw.pw_shell = (char *) "/bin/false";
+			pw.pw_shell = (char *) "/sbin/nologin";
 		}
 	}
 	pw.pw_gid = usegroup ? xgroup2gid(usegroup) : -1; /* exits on failure */
