From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 06 Sep 2021 23:13:05 +0200
Subject: [PATCH] Avoid redefined warnings when building with utmps

--- a/include/libbb.h
+++ b/include/libbb.h
@@ -107,6 +107,9 @@
 #  define _PATH_UTMPX _PATH_UTMP
 # else
 #  if !defined(__FreeBSD__)
+/* _PATH_UTMP and _PATH_WTMP are defined both in paths.h and utmps/utmp.h. */
+#   undef _PATH_UTMP
+#   undef _PATH_WTMP
 #   include <utmp.h>
 #  else
 #   define _PATH_UTMPX "/var/run/utx.active"
