From 4d277601bdd3717ff304b2eef9823cc4c17134e2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 11 Mar 2010 17:06:59 +0000
Subject: [PATCH] beep: the -d option takes milliseconds not microseconds

---
 miscutils/beep.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/miscutils/beep.c b/miscutils/beep.c
index c17cbfd..10768e7 100644
--- a/miscutils/beep.c
+++ b/miscutils/beep.c
@@ -83,7 +83,7 @@ int beep_main(int argc, char **argv)
 			usleep(1000 * length);
 			ioctl(speaker, KIOCSOUND, (void*)0);
 			if (--rep)
-				usleep(delay);
+				usleep(1000 * delay);
 		}
 	}
 
-- 
1.7.0.2

