From a3016293eb42eb3679400148ed4520005ff6b8a7 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 27 Dec 2016 20:46:59 +0100
Subject: [PATCH 02/12] app location for vi and lspci

---
 editors/vi.c       | 2 +-
 util-linux/lspci.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/editors/vi.c b/editors/vi.c
index 38a4692fd..dd3efda75 100644
--- a/editors/vi.c
+++ b/editors/vi.c
@@ -166,7 +166,7 @@
 //config:	  Unless you want more (or less) frequent "undo points" while typing,
 //config:	  you should probably leave this unchanged.
 
-//applet:IF_VI(APPLET(vi, BB_DIR_BIN, BB_SUID_DROP))
+//applet:IF_VI(APPLET(vi, BB_DIR_USR_BIN, BB_SUID_DROP))
 
 //kbuild:lib-$(CONFIG_VI) += vi.o
 
diff --git a/util-linux/lspci.c b/util-linux/lspci.c
index 8b38a2366..f918f0972 100644
--- a/util-linux/lspci.c
+++ b/util-linux/lspci.c
@@ -16,7 +16,7 @@
 //config:
 //config:	  This version uses sysfs (/sys/bus/pci/devices) only.
 
-//applet:IF_LSPCI(APPLET(lspci, BB_DIR_USR_BIN, BB_SUID_DROP))
+//applet:IF_LSPCI(APPLET(lspci, BB_DIR_USR_SBIN, BB_SUID_DROP))
 
 //kbuild:lib-$(CONFIG_LSPCI) += lspci.o
 
-- 
2.11.0

