From 9fc807db6b3b48da354e4c9427f18baa48f0a574 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 28 Sep 2024 22:06:22 +0200
Subject: [PATCH] od: Skip `od -B` on big-endian
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fails on s390x.

Signed-off-by: Sören Tempel <soeren+git@soeren-tempel.net>
---
 testsuite/od.tests | 1 +
 1 file changed, 1 insertion(+)

diff --git a/testsuite/od.tests b/testsuite/od.tests
index 4f245a7e8..be987a42d 100755
--- a/testsuite/od.tests
+++ b/testsuite/od.tests
@@ -61,6 +61,7 @@ testing "od -a (DESKTOP)" \
 "\xf0\xf1\xf2\xf3\xf4\xf5\xf6\xf7\xf8\xf9\xfa\xfb\xfc\xfd\xfe\xff"
 SKIP=
 
+$little_endian || SKIP=1
 testing "od -B" \
         "od -B" \
 "\
