From 383201e7258c3c1646f6d5cab4e813b5c3b3ecd0 Mon Sep 17 00:00:00 2001
From: Denys Vlasenko <vda.linux@googlemail.com>
Date: Wed, 10 Feb 2016 07:06:31 +0100
Subject: [PATCH 4002/4007] ntpd: shorter message

Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
 networking/ntpd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/networking/ntpd.c b/networking/ntpd.c
index 32590a1..1651670 100644
--- a/networking/ntpd.c
+++ b/networking/ntpd.c
@@ -1947,7 +1947,7 @@ recv_and_process_peer_pkt(peer_t *p)
 			adjust_poll(MINPOLL);
 		} else {
 			VERB3 if (rc > 0)
-				bb_error_msg("want smaller poll interval: offset/jitter ratio > %u",
+				bb_error_msg("want smaller poll interval: offset/jitter > %u",
 					POLLADJ_GATE);
 			adjust_poll(-G.poll_exp * 2);
 		}
-- 
2.8.3

