diff -Naur busybox-1.19.3-old/include/platform.h busybox-1.19.3/include/platform.h
--- busybox-1.19.3-old/include/platform.h	2011-10-29 05:43:01.000000000 -0600
+++ busybox-1.19.3/include/platform.h	2011-12-18 15:53:49.817278374 -0700
@@ -443,6 +443,13 @@
 # undef HAVE_NET_ETHERNET_H
 #endif
 
+#if defined(__musl__)
+# undef HAVE_SETBIT
+# include <stddef.h>
+# include <termios.h>
+# include <sys/ioctl.h>
+#endif
+
 /*
  * Now, define prototypes for all the functions defined in platform.c
  * These must come after all the HAVE_* macros are defined (or not)
diff -Naur busybox-1.19.3-old/miscutils/man.c busybox-1.19.3/miscutils/man.c
--- busybox-1.19.3-old/miscutils/man.c	2011-09-05 20:35:17.000000000 -0600
+++ busybox-1.19.3/miscutils/man.c	2011-12-18 15:53:49.817278374 -0700
@@ -116,7 +116,7 @@
 	/* "2>&1" is added so that nroff errors are shown in pager too.
 	 * Otherwise it may show just empty screen */
 	cmd = xasprintf(
-		man ? "gtbl | nroff -Tlatin1 -mandoc 2>&1 | %s"
+		man ? "nroff -Tutf -man 2>&1 | %s"
 		    : "%s",
 		pager);
 	system(cmd);
diff -Naur busybox-1.19.3-old/networking/ifconfig.c busybox-1.19.3/networking/ifconfig.c
--- busybox-1.19.3-old/networking/ifconfig.c	2011-09-05 20:35:17.000000000 -0600
+++ busybox-1.19.3/networking/ifconfig.c	2011-12-18 15:53:49.817278374 -0700
@@ -56,7 +56,7 @@
 #endif
 
 #if ENABLE_FEATURE_IFCONFIG_SLIP
-# include <net/if_slip.h>
+# include <linux/if_slip.h>
 #endif
 
 /* I don't know if this is needed for busybox or not.  Anyone? */
diff -Naur busybox-1.19.3-old/networking/libiproute/iplink.c busybox-1.19.3/networking/libiproute/iplink.c
--- busybox-1.19.3-old/networking/libiproute/iplink.c	2011-09-05 20:35:17.000000000 -0600
+++ busybox-1.19.3/networking/libiproute/iplink.c	2011-12-18 15:53:49.821278291 -0700
@@ -5,7 +5,6 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 #include <net/if.h>
-#include <net/if_packet.h>
 #include <netpacket/packet.h>
 #include <netinet/if_ether.h>
 
diff -Naur busybox-1.19.3-old/util-linux/fdisk.c busybox-1.19.3/util-linux/fdisk.c
--- busybox-1.19.3-old/util-linux/fdisk.c	2011-09-05 20:35:17.000000000 -0600
+++ busybox-1.19.3/util-linux/fdisk.c	2011-12-18 15:53:49.821278291 -0700
@@ -550,7 +550,7 @@
 {
 #if ENABLE_FDISK_SUPPORT_LARGE_DISKS
 	off64_t off = (off64_t)secno * sector_size;
-	if (lseek64(dev_fd, off, SEEK_SET) == (off64_t) -1)
+	if (lseek(dev_fd, off, SEEK_SET) == (off64_t) -1)
 		fdisk_fatal(unable_to_seek);
 #else
 	uint64_t off = (uint64_t)secno * sector_size;
