--- busybox-1.17.1-updated/networking/ifupdown.c
+++ busybox-1.17.1/networking/ifupdown.c
@@ -353,6 +353,11 @@
 #endif
 }
 
+static int FAST_FUNC manual_up_down6(struct interface_defn_t *ifd UNUSED_PARAM, execfn *exec UNUSED_PARAM)
+{
+	return 1;
+}
+
 static int FAST_FUNC static_up6(struct interface_defn_t *ifd, execfn *exec)
 {
 	int result;
@@ -398,10 +403,11 @@
 
 static const struct method_t methods6[] = {
 #if ENABLE_FEATURE_IFUPDOWN_IP
-	{ "v4tunnel", v4tunnel_up, v4tunnel_down, },
+	{ "v4tunnel" , v4tunnel_up     , v4tunnel_down   , },
 #endif
-	{ "static", static_up6, static_down6, },
-	{ "loopback", loopback_up6, loopback_down6, },
+	{ "static"   , static_up6      , static_down6    , },
+	{ "manual"   , manual_up_down6 , manual_up_down6 , },
+	{ "loopback" , loopback_up6    , loopback_down6  , },
 };
 
 static const struct address_family_t addr_inet6 = {
