Some slower nics needs more attempts to get a lease

--- ./networking/udhcp/dhcpc.c.orig
+++ ./networking/udhcp/dhcpc.c
@@ -1238,7 +1238,7 @@
 	llist_t *list_x = NULL;
 	int tryagain_timeout = 20;
 	int discover_timeout = 3;
-	int discover_retries = 3;
+	int discover_retries = 5;
 	uint32_t server_addr = server_addr; /* for compiler */
 	uint32_t requested_ip = 0;
 	uint32_t xid = xid; /* for compiler */
