From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 05 Jun 2016 14:11:00 +0200
Last-Updated: Thu, 19 Oct 2017 01:04:00 +0200
Subject: [PATCH] Disable PaX memory protection for julia binary
--- a/Makefile
+++ b/Makefile
@@ -111,8 +111,14 @@
 julia-sysimg-debug : julia-inference julia-ui-debug
 	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT) $(build_private_libdir)/sys-debug$(CPUID_TAG).$(SHLIB_EXT) JULIA_BUILD_MODE=debug
 
-julia-debug julia-release : julia-% : julia-ui-% julia-sysimg-% julia-symlink julia-libccalltest
+julia-pax-release : julia-ui-release
+	paxmark m $(JULIA_EXECUTABLE_release)
 
+julia-pax-debug : julia-ui-debug
+	paxmark m $(JULIA_EXECUTABLE_debug)
+
+julia-debug julia-release : julia-% : julia-ui-% julia-pax-% julia-sysimg-% julia-symlink julia-libccalltest
+
 debug release : % : julia-%
 
 docs: julia-sysimg-$(JULIA_BUILD_MODE)
@@ -551,7 +557,7 @@
 	test testall testall1 test clean distcleanall cleanall clean-* \
 	run-julia run-julia-debug run-julia-release run \
 	install binary-dist light-source-dist.tmp light-source-dist \
-	dist full-source-dist source-dist examples
+	dist full-source-dist source-dist examples julia-pax-release julia-pax-debug
 
 test: check-whitespace $(JULIA_BUILD_MODE)
 	@$(MAKE) $(QUIET_MAKE) -C $(BUILDROOT)/test default JULIA_BUILD_MODE=$(JULIA_BUILD_MODE)
