Description: Ensure that build is verbose
Author: Sébastien Villemot <sebastien@debian.org>
Author: Shiz <hi@shiz.me>
Author: Jakub Jirutka <jakub@jirutka.cz>
Forwarded: not-needed
Last-Updated: 2017-10-19
---
--- a/deps/libuv.mk
+++ b/deps/libuv.mk
@@ -30,7 +30,7 @@
 	touch -c $(SRCDIR)/srccache/$(LIBUV_SRC_DIR)/configure
 	mkdir -p $(dir $@)
 	cd $(dir $@) && \
-	$(dir $<)/configure --with-pic $(CONFIGURE_COMMON) $(UV_FLAGS)
+	$(dir $<)/configure --with-pic $(CONFIGURE_COMMON) $(UV_FLAGS) --disable-silent-rules
 	echo 1 > $@
 
 $(BUILDDIR)/$(LIBUV_SRC_DIR)/build-compiled: $(BUILDDIR)/$(LIBUV_SRC_DIR)/build-configured
