--- a/src/main.rs
+++ b/src/main.rs
@@ -46,9 +46,8 @@
 }
 
 fn setup_logger(level: LevelFilter, log_file: Option<impl AsRef<Path>>) {
-    use log::Log;
 
-    let dispatcher = fern::Dispatch::new()
+    let mut dispatcher = fern::Dispatch::new()
         .format(|out, message, record| {
             out.finish(format_args!(
                 "{}[{}][{}] {}",
@@ -61,13 +60,8 @@
         .level(level)
         .chain(std::io::stdout());
 
-    let dispatcher = if let Some(log_file) = log_file {
-        dispatcher.chain(fern::log_file(log_file).unwrap())
-    } else {
-        const VERSION: &str = env!("CARGO_PKG_VERSION");
-        let systemd_logger =
-            systemd_journal_logger::JournalLog::with_extra_fields(vec![("VERSION", VERSION)]);
-        dispatcher.chain(fern::Output::call(move |record| systemd_logger.log(record)))
+    if let Some(log_file) = log_file {
+        dispatcher = dispatcher.chain(fern::log_file(log_file).unwrap());
     };
 
     dispatcher.apply().unwrap();
