diff --git a/setup.py b/setup.py.new
index 0fd974742f..4a42241d37 100644
--- a/setup.py
+++ b/setup.py.new
@@ -33,7 +33,6 @@ setup(
     author="Phillip J. Eby",
     author_email="peak@eby-sarna.com",
     license="ZPL 2.1",
-    test_suite = 'simplegeneric.test_suite',
     py_modules = ['simplegeneric'],
     classifiers = [
         line.strip() for line in open('classifiers.txt') if line.strip()
