diff --git a/build-aux/cargo.sh b/build-aux/cargo.sh
index ad9573c..527d530 100755
--- a/build-aux/cargo.sh
+++ b/build-aux/cargo.sh
@@ -20,7 +20,7 @@ then
         cp "$CARGO_TARGET_DIR"/debug/shortwave $APP_OUTPUT
 else
     echo "** RELEASE MODE **"
-    cargo build --manifest-path \
+    cargo build --verbose --manifest-path \
         "$MESON_SOURCE_ROOT"/Cargo.toml --message-format=short --release && \
         cp "$CARGO_TARGET_DIR"/release/shortwave $APP_OUTPUT
 fi
