--- a/net-imap.gemspec
+++ b/net-imap.gemspec
@@ -23,12 +23,7 @@
   spec.metadata["source_code_uri"] = spec.homepage
   spec.metadata["changelog_uri"] = spec.homepage + "/releases"
 
-  # Specify which files should be added to the gem when it is released.
-  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z 2>/dev/null`.split("\x0")
-      .reject {|f| f.match(%r{^(bin|test|spec|benchmarks|features|rfcs)/}) }
-  end
+  spec.files         = Dir["lib/**/*.rb"]
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
