From 9cd01b922db81e0d97b71c698d155437cc98876a Mon Sep 17 00:00:00 2001
From: Tobias Fella <fella@posteo.de>
Date: Sun, 27 Dec 2020 19:46:27 +0100
Subject: [PATCH] Fix login for servers without well-known

---
 src/controller.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/controller.cpp b/src/controller.cpp
index 752b9ec..0090a51 100644
--- a/src/controller.cpp
+++ b/src/controller.cpp
@@ -110,7 +110,7 @@ void Controller::loginWithCredentials(const QString &serverAddr, const QString &
     // we are using a fake mixd since resolveServer just set the homeserver url :sigh:
     conn->resolveServer("@username:" + serverUrl.host() + ":" + QString::number(serverUrl.port(443)));
 
-    connect(conn, &Connection::homeserverChanged, this, [this, user, conn, pass, deviceName]() {
+    connect(conn, &Connection::loginFlowsChanged, this, [this, user, conn, pass, deviceName]() {
         conn->loginWithPassword(user, pass, deviceName, "");
         connect(conn, &Connection::connected, this, [this, conn, deviceName] {
             AccountSettings account(conn->userId());
-- 
2.29.2

