From b712f929f8096f679f5dfdd148579cc88203dc45 Mon Sep 17 00:00:00 2001
From: longxinhui <longxinhui.e@gmail.com>
Date: Thu, 9 Aug 2018 00:02:17 +0800
Subject: [PATCH] fix hp_inline_hash(s390x)

#15
---
 extension/xhprof.c | 43 ++++++++++++++++++++++++++++---------------
 1 file changed, 28 insertions(+), 15 deletions(-)

diff --git a/extension/xhprof.c b/extension/xhprof.c
index 5828176..a6ccc32 100644
--- a/extension/xhprof.c
+++ b/extension/xhprof.c
@@ -347,22 +347,35 @@ static void hp_register_constants(INIT_FUNC_ARGS)
  *
  * @author cjiang
  */
-static inline uint8 hp_inline_hash(char * str)
+static inline uint8 hp_inline_hash(char *str)
 {
-    ulong h = 5381;
-    uint i = 0;
-    uint8 res = 0;
-
-    while (*str) {
-        h += (h << 5);
-        h ^= (ulong) *str++;
-    }
-
-    for (i = 0; i < sizeof(ulong); i++) {
-        res += ((uint8 *)&h)[i];
-    }
-
-    return res;
+    size_t len = strlen(str);
+    register uint8 hash = 0;
+
+    /* variant with the hash unrolled eight times */
+    for (; len >= 8; len -= 8) {
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+        hash = ((hash << 5) + hash) + *str++;
+    }
+    switch (len) {
+        case 7: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 6: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 5: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 4: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 3: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 2: hash = ((hash << 5) + hash) + *str++; /* fallthrough... */
+        case 1: hash = ((hash << 5) + hash) + *str++; break;
+        case 0: break;
+EMPTY_SWITCH_DEFAULT_CASE()
+    }
+
+    return hash;
 }
 
 /**
