diff -upr cifs-utils-7.0.orig/cifscreds.c cifs-utils-7.0/cifscreds.c
--- cifs-utils-7.0.orig/cifscreds.c	2024-05-12 11:56:19.023002421 +0200
+++ cifs-utils-7.0/cifscreds.c	2024-05-12 11:57:09.486444286 +0200
@@ -29,6 +29,7 @@
 #include <keyutils.h>
 #include <getopt.h>
 #include <errno.h>
+#include <libgen.h>
 #include "cifskey.h"
 #include "mount.h"
 #include "resolve_host.h"
diff -upr cifs-utils-7.0.orig/getcifsacl.c cifs-utils-7.0/getcifsacl.c
--- cifs-utils-7.0.orig/getcifsacl.c	2024-05-12 11:56:19.019669081 +0200
+++ cifs-utils-7.0/getcifsacl.c	2024-05-12 11:56:58.229753405 +0200
@@ -38,6 +38,7 @@
 #include "cifsacl.h"
 #include "idmap_plugin.h"
 #include <ftw.h>
+#include <libgen.h>
 
 static void *plugin_handle;
 static bool plugin_loaded;
diff -upr cifs-utils-7.0.orig/mount.cifs.c cifs-utils-7.0/mount.cifs.c
--- cifs-utils-7.0.orig/mount.cifs.c	2024-05-12 11:56:19.039669124 +0200
+++ cifs-utils-7.0/mount.cifs.c	2024-05-12 11:56:42.343052575 +0200
@@ -38,6 +38,7 @@
 #include <errno.h>
 #include <netdb.h>
 #include <string.h>
+#include <libgen.h>
 #include <mntent.h>
 #include <fcntl.h>
 #include <limits.h>
diff -upr cifs-utils-7.0.orig/setcifsacl.c cifs-utils-7.0/setcifsacl.c
--- cifs-utils-7.0.orig/setcifsacl.c	2024-05-12 11:56:19.026335762 +0200
+++ cifs-utils-7.0/setcifsacl.c	2024-05-12 11:57:20.573134789 +0200
@@ -49,6 +49,7 @@
 #include <limits.h>
 #include <ctype.h>
 #include <sys/xattr.h>
+#include <libgen.h>
 
 #include "cifsacl.h"
 #include "idmap_plugin.h"
