diff --git a/Makefile.am b/Makefile.am
index fe9cd34..6ab4b01 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -119,10 +119,10 @@ endif
 SUBDIRS = contrib
 
 install-exec-hook:
-	(cd $(ROOTSBINDIR) && ln -sf mount.cifs mount.smb3)
+	(cd $(DESTDIR)$(ROOTSBINDIR) && ln -sf mount.cifs mount.smb3)
 
 install-data-hook:
-	(cd $(man8dir) && ln -sf mount.cifs.8 mount.smb3.8)
+	(cd $(DESTDIR)$(man8dir) && ln -sf mount.cifs.8 mount.smb3.8)
 
 uninstall-hook:
 	(cd $(ROOTSBINDIR) && rm -f $(ROOTSBINDIR)/mount.smb3)
