diff --git a/include/librsvg/meson.build b/include/librsvg/meson.build
index 2d254f38e839..2774f7c205c5 100644
--- a/include/librsvg/meson.build
+++ b/include/librsvg/meson.build
@@ -1,9 +1,12 @@
 includes = files(
     'rsvg.h',
     'rsvg-cairo.h',
-    'rsvg-pixbuf.h'
 )
 
+if pixbuf_dep.found()
+  files+='rsvg-pixbuf.h'
+endif
+
 features_data = configuration_data()
 features_data.set10('LIBRSVG_HAVE_PIXBUF', pixbuf_dep.found())
 

