diff --git a/getopt.h b/getopt.h
index 634cba1..9b86f7e 100644
--- a/getopt.h
+++ b/getopt.h
@@ -133,7 +133,7 @@ struct option
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int __argc, char *const *__argv, const char *__shortopts);
 # else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int __argc, char *const *__argv, const char *__shortopts);
 # endif /* __GNU_LIBRARY__ */
 
 # ifndef __need_getopt
