diff --git a/src/lxc/confile.c b/src/lxc/confile.c
index 5d2c99f8a..279e52357 100644
--- a/src/lxc/confile.c
+++ b/src/lxc/confile.c
@@ -3214,6 +3214,7 @@ int lxc_config_parse_arch(const char *arch, signed long *persona)
 		{ "ppc64el",   PER_LINUX   },
 		{ "ppc64le",   PER_LINUX   },
 		{ "powerpc64", PER_LINUX   },
+		{ "riscv64",   PER_LINUX   },
 		{ "s390x",     PER_LINUX   },
 		{ "x86_64",    PER_LINUX   },
 	};
