--- ./src/lxc/lxc-shutdown.in.orig	2013-02-08 13:28:28.489406380 +0000
+++ ./src/lxc/lxc-shutdown.in	2013-02-08 13:30:31.910466926 +0000
@@ -113,11 +113,21 @@
     exit 1
 fi
 
+signal_reboot=INT
+signal_poweroff=PWR
+init_exe=$(readlink -f /proc/$pid/exe)
+case ${init_exe} in
+    */busybox)
+        signal_reboot=TERM
+        signal_poweroff=USR2
+       ;;
+esac
+
 if [ $reboot -eq 1 ]; then
-    kill -INT $pid
+    kill -s $signal_reboot  $pid
     exit 0
 else
-    kill -PWR $pid
+    kill -s $signale_poweroff $pid
 fi
 
 if [ $dowait -eq 0 ]; then
