diff -ru lxc-0.9.0.orig/src/lua-lxc/core.c lxc-0.9.0/src/lua-lxc/core.c
--- lxc-0.9.0.orig/src/lua-lxc/core.c	2013-10-10 09:22:58.841387605 +0300
+++ lxc-0.9.0/src/lua-lxc/core.c	2013-10-10 09:23:13.847971218 +0300
@@ -21,7 +21,6 @@
  */
 
 #define LUA_LIB
-#define _GNU_SOURCE
 #include <lua.h>
 #include <lauxlib.h>
 #include <string.h>
diff -ru lxc-0.9.0.orig/src/lxc/attach.c lxc-0.9.0/src/lxc/attach.c
--- lxc-0.9.0.orig/src/lxc/attach.c	2013-10-10 09:22:58.841387605 +0300
+++ lxc-0.9.0/src/lxc/attach.c	2013-10-10 09:23:13.847971218 +0300
@@ -21,13 +21,14 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
+#include "config.h"
 #include <unistd.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
 #include <fcntl.h>
+#include <sched.h>
 #include <sys/param.h>
 #include <sys/prctl.h>
 #include <sys/mount.h>
@@ -44,7 +45,6 @@
 #include "log.h"
 #include "attach.h"
 #include "caps.h"
-#include "config.h"
 #include "apparmor.h"
 
 lxc_log_define(lxc_attach, lxc);
diff -ru lxc-0.9.0.orig/src/lxc/caps.c lxc-0.9.0/src/lxc/caps.c
--- lxc-0.9.0.orig/src/lxc/caps.c	2013-10-10 09:22:58.841387605 +0300
+++ lxc-0.9.0/src/lxc/caps.c	2013-10-10 09:23:13.847971218 +0300
@@ -21,7 +21,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
+#include "config.h"
 #include <unistd.h>
 #include <fcntl.h>
 #include <stdlib.h>
@@ -29,7 +29,6 @@
 #include <sys/prctl.h>
 #include <errno.h>
 
-#include "config.h"
 #include "log.h"
 
 lxc_log_define(lxc_caps, lxc);
diff -ru lxc-0.9.0.orig/src/lxc/cgroup.c lxc-0.9.0/src/lxc/cgroup.c
--- lxc-0.9.0.orig/src/lxc/cgroup.c	2013-10-10 09:22:58.841387605 +0300
+++ lxc-0.9.0/src/lxc/cgroup.c	2013-10-10 09:28:10.249698806 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
@@ -37,7 +36,6 @@
 #include <net/if.h>
 
 #include "error.h"
-#include "config.h"
 #include "commands.h"
 
 #include <lxc/log.h>
diff -ru lxc-0.9.0.orig/src/lxc/commands.c lxc-0.9.0/src/lxc/commands.c
--- lxc-0.9.0.orig/src/lxc/commands.c	2013-10-10 09:22:58.841387605 +0300
+++ lxc-0.9.0/src/lxc/commands.c	2013-10-10 09:23:13.851304534 +0300
@@ -28,7 +28,7 @@
 #include <fcntl.h>
 #include <sys/socket.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/param.h>
 #include <malloc.h>
 #include <stdlib.h>
diff -ru lxc-0.9.0.orig/src/lxc/conf.c lxc-0.9.0/src/lxc/conf.c
--- lxc-0.9.0.orig/src/lxc/conf.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/conf.c	2013-10-10 09:28:30.296256871 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <stdarg.h>
 #include <errno.h>
@@ -58,7 +57,6 @@
 #include "network.h"
 #include "error.h"
 #include "parse.h"
-#include "config.h"
 #include "utils.h"
 #include "conf.h"
 #include "log.h"
diff -ru lxc-0.9.0.orig/src/lxc/confile.c lxc-0.9.0/src/lxc/confile.c
--- lxc-0.9.0.orig/src/lxc/confile.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/confile.c	2013-10-10 09:28:45.472841319 +0300
@@ -20,7 +20,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -38,7 +38,6 @@
 #include <net/if.h>
 
 #include "parse.h"
-#include "config.h"
 #include "confile.h"
 #include "utils.h"
 
diff -ru lxc-0.9.0.orig/src/lxc/freezer.c lxc-0.9.0/src/lxc/freezer.c
--- lxc-0.9.0.orig/src/lxc/freezer.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/freezer.c	2013-10-10 09:29:01.486087901 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <unistd.h>
diff -ru lxc-0.9.0.orig/src/lxc/lxc_attach.c lxc-0.9.0/src/lxc/lxc_attach.c
--- lxc-0.9.0.orig/src/lxc/lxc_attach.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/lxc_attach.c	2013-10-10 09:29:19.079325924 +0300
@@ -21,7 +21,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
+#include "config.h"
 #include <unistd.h>
 #include <errno.h>
 #include <pwd.h>
@@ -36,7 +36,6 @@
 #include "arguments.h"
 #include "caps.h"
 #include "cgroup.h"
-#include "config.h"
 #include "confile.h"
 #include "start.h"
 #include "sync.h"
diff -ru lxc-0.9.0.orig/src/lxc/lxc_checkpoint.c lxc-0.9.0/src/lxc/lxc_checkpoint.c
--- lxc-0.9.0.orig/src/lxc/lxc_checkpoint.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/lxc_checkpoint.c	2013-10-10 09:25:41.587170847 +0300
@@ -20,7 +20,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <assert.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -35,7 +35,6 @@
 #include <lxc/utils.h>
 
 #include "arguments.h"
-#include "config.h"
 #include "caps.h"
 
 lxc_log_define(lxc_checkpoint_ui, lxc_checkpoint);
diff -ru lxc-0.9.0.orig/src/lxc/lxc_console.c lxc-0.9.0/src/lxc/lxc_console.c
--- lxc-0.9.0.orig/src/lxc/lxc_console.c	2013-10-10 09:22:58.844720784 +0300
+++ lxc-0.9.0/src/lxc/lxc_console.c	2013-10-10 09:23:13.854637849 +0300
@@ -21,9 +21,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
@@ -35,7 +33,7 @@
 #include <sys/param.h>
 #include <sys/types.h>
 #include <sys/stat.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/ioctl.h>
 
 #include "error.h"
diff -ru lxc-0.9.0.orig/src/lxc/lxc_execute.c lxc-0.9.0/src/lxc/lxc_execute.c
--- lxc-0.9.0.orig/src/lxc/lxc_execute.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxc_execute.c	2013-10-10 09:25:21.023948914 +0300
@@ -20,7 +20,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
 #include <errno.h>
@@ -37,7 +37,6 @@
 #include "conf.h"
 #include "confile.h"
 #include "arguments.h"
-#include "config.h"
 #include "start.h"
 #include "utils.h"
 
diff -ru lxc-0.9.0.orig/src/lxc/lxc_init.c lxc-0.9.0/src/lxc/lxc_init.c
--- lxc-0.9.0.orig/src/lxc/lxc_init.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxc_init.c	2013-10-10 09:23:13.854637849 +0300
@@ -30,7 +30,6 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#define _GNU_SOURCE
 #include <getopt.h>
 
 #include "log.h"
diff -ru lxc-0.9.0.orig/src/lxc/lxc_restart.c lxc-0.9.0/src/lxc/lxc_restart.c
--- lxc-0.9.0.orig/src/lxc/lxc_restart.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxc_restart.c	2013-10-10 09:25:59.123742508 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
@@ -33,7 +32,6 @@
 #include "lxc.h"
 #include "caps.h"
 #include "conf.h"
-#include "config.h"
 #include "confile.h"
 #include "arguments.h"
 #include "utils.h"
diff -ru lxc-0.9.0.orig/src/lxc/lxc_start.c lxc-0.9.0/src/lxc/lxc_start.c
--- lxc-0.9.0.orig/src/lxc/lxc_start.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxc_start.c	2013-10-10 09:24:30.684221628 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <libgen.h>
 #include <stdlib.h>
 #include <unistd.h>
@@ -46,7 +45,6 @@
 #include "conf.h"
 #include "cgroup.h"
 #include "utils.h"
-#include "config.h"
 #include "confile.h"
 #include "arguments.h"
 
diff -ru lxc-0.9.0.orig/src/lxc/lxc_unshare.c lxc-0.9.0/src/lxc/lxc_unshare.c
--- lxc-0.9.0.orig/src/lxc/lxc_unshare.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxc_unshare.c	2013-10-10 09:23:13.854637849 +0300
@@ -20,9 +20,7 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <stdlib.h>
 #include <unistd.h>
 #include <libgen.h>
diff -ru lxc-0.9.0.orig/src/lxc/lxccontainer.c lxc-0.9.0/src/lxc/lxccontainer.c
--- lxc-0.9.0.orig/src/lxc/lxccontainer.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxccontainer.c	2013-10-10 09:23:13.854637849 +0300
@@ -800,7 +800,7 @@
 	if (pid < 0)
 		return false;
 	if (pid == 0) { // child
-		ret = execlp("lxc-destroy", "lxc-destroy", "-n", c->name, "-P", c->config_path, NULL);
+		ret = execlp("lxc-destroy", "lxc-destroy", "-n", c->name, "-P", c->config_path, (void*)0);
 		perror("execl");
 		exit(1);
 	}
diff -ru lxc-0.9.0.orig/src/lxc/lxclock.c lxc-0.9.0/src/lxc/lxclock.c
--- lxc-0.9.0.orig/src/lxc/lxclock.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxclock.c	2013-10-10 09:23:13.854637849 +0300
@@ -18,8 +18,8 @@
  */
 
 #include "lxclock.h"
-#include <malloc.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #define OFLAG (O_CREAT | O_RDWR)
 #define SEMMODE 0660
diff -ru lxc-0.9.0.orig/src/lxc/lxcutmp.c lxc-0.9.0/src/lxc/lxcutmp.c
--- lxc-0.9.0.orig/src/lxc/lxcutmp.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/lxcutmp.c	2013-10-10 09:23:13.854637849 +0300
@@ -63,6 +63,8 @@
 #include "lxc.h"
 #include "log.h"
 
+#include <utmp.h>
+
 #ifndef __USE_GNU
 #define __USE_GNU
 #endif
diff -ru lxc-0.9.0.orig/src/lxc/network.c lxc-0.9.0/src/lxc/network.c
--- lxc-0.9.0.orig/src/lxc/network.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/network.c	2013-10-10 09:30:26.518960572 +0300
@@ -20,9 +20,8 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCe
 #include <stdlib.h>
 #include <unistd.h>
 #include <fcntl.h>
diff -ru lxc-0.9.0.orig/src/lxc/parse.c lxc-0.9.0/src/lxc/parse.c
--- lxc-0.9.0.orig/src/lxc/parse.c	2013-10-10 09:22:58.848053921 +0300
+++ lxc-0.9.0/src/lxc/parse.c	2013-10-10 09:23:30.724546457 +0300
@@ -20,16 +20,14 @@
  * License along with this library; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
-#define _GNU_SOURCE
+#include "config.h"
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
 #include <dirent.h>
 
 #include "parse.h"
-#include "config.h"
 #include <lxc/log.h>
 
 /* Define getline() if missing from the C library */
diff -ru lxc-0.9.0.orig/src/lxc/restart.c lxc-0.9.0/src/lxc/restart.c
--- lxc-0.9.0.orig/src/lxc/restart.c	2013-10-10 09:22:58.851387051 +0300
+++ lxc-0.9.0/src/lxc/restart.c	2013-10-10 09:23:13.857971164 +0300
@@ -24,7 +24,6 @@
 #include "config.h"
 
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <errno.h>
diff -ru lxc-0.9.0.orig/src/lxc/seccomp.c lxc-0.9.0/src/lxc/seccomp.c
--- lxc-0.9.0.orig/src/lxc/seccomp.c	2013-10-10 09:22:58.851387051 +0300
+++ lxc-0.9.0/src/lxc/seccomp.c	2013-10-10 09:23:13.857971164 +0300
@@ -21,7 +21,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
 #include <stdio.h>
 #include <stdlib.h>
 #include <seccomp.h>
diff -ru lxc-0.9.0.orig/src/lxc/start.c lxc-0.9.0/src/lxc/start.c
--- lxc-0.9.0.orig/src/lxc/start.c	2013-10-10 09:22:58.851387051 +0300
+++ lxc-0.9.0/src/lxc/start.c	2013-10-10 09:23:13.857971164 +0300
@@ -24,7 +24,6 @@
 #include "config.h"
 
 #include <stdio.h>
-#undef _GNU_SOURCE
 #include <string.h>
 #include <stdlib.h>
 #include <dirent.h>
@@ -43,7 +42,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/un.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/syscall.h>
 
 #if HAVE_SYS_CAPABILITY_H
diff -ru lxc-0.9.0.orig/src/lxc/utils.c lxc-0.9.0/src/lxc/utils.c
--- lxc-0.9.0.orig/src/lxc/utils.c	2013-10-10 09:22:58.851387051 +0300
+++ lxc-0.9.0/src/lxc/utils.c	2013-10-10 09:23:13.857971164 +0300
@@ -21,7 +21,6 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
  */
 
-#define _GNU_SOURCE
 #include <errno.h>
 #include <unistd.h>
 #include <stdlib.h>
diff -ru lxc-0.9.0.orig/src/lxc/utils.h lxc-0.9.0/src/lxc/utils.h
--- lxc-0.9.0.orig/src/lxc/utils.h	2013-10-10 09:22:58.851387051 +0300
+++ lxc-0.9.0/src/lxc/utils.h	2013-10-10 09:23:13.857971164 +0300
@@ -23,6 +23,8 @@
 #ifndef _utils_h
 #define _utils_h
 
+#include <sys/types.h>
+
 extern int lxc_copy_file(const char *src, const char *dst);
 extern int lxc_setup_fs(void);
 extern int get_u16(unsigned short *val, const char *arg, int base);
