--- lxc-lxc-2.1.1/config/templates/alpine.common.conf.in
+++ lxc-lxc-2.1.1.setpcap/config/templates/alpine.common.conf.in
@@ -8,7 +8,6 @@
 lxc.cap.drop = audit_write
 lxc.cap.drop = ipc_owner
 lxc.cap.drop = mknod
-lxc.cap.drop = setpcap
 lxc.cap.drop = sys_nice
 lxc.cap.drop = sys_pacct
 lxc.cap.drop = sys_ptrace
Only in lxc-lxc-2.1.1.setpcap: config/templates/alpine.common.conf.in.orig
--- lxc-lxc-2.1.1/templates/lxc-alpine.in
+++ lxc-lxc-2.1.1.setpcap/templates/lxc-alpine.in
@@ -398,6 +398,9 @@
 		# hostname(1).
 		lxc.cap.drop = sys_admin
 
+		# Comment this out if required by your applications.
+		lxc.cap.drop = setpcap
+
 		# Include common configuration.
 		lxc.include = $LXC_TEMPLATE_CONFIG/alpine.common.conf
 	EOF
Only in lxc-lxc-2.1.1.setpcap: templates/lxc-alpine.in.orig
