diff -urp dwm-6.1.orig/drw.c dwm-6.1/drw.c
--- dwm-6.1.orig/drw.c	2015-11-08 14:39:37.000000000 -0800
+++ dwm-6.1/drw.c	2016-12-05 17:55:37.082900924 -0800
@@ -21,7 +21,7 @@ utf8decodebyte(const char c, size_t *i)
 {
 	for (*i = 0; *i < (UTF_SIZ + 1); ++(*i))
 		if (((unsigned char)c & utfmask[*i]) == utfbyte[*i])
-			return (unsigned char)c & ~utfmask[*i];
+			return (unsigned char)c;
 	return 0;
 }
 

