--- a/setup.py
+++ b/setup.py
@@ -31,7 +31,7 @@
     long_description=long_description,
     long_description_content_type='text/markdown',
     url='https://github.com/karakatic/EvoPreprocess',
-    packages=setuptools.find_packages(),
+    packages=setuptools.find_packages(exclude=("*tests.*", "*tests")),
     classifiers=[
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
