diff --git a/Cargo.toml b/Cargo.toml
index f1ed52e..dc2f5f9 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,15 +9,6 @@ readme = "README.md"
 homepage = "https://github.com/ArniDagur/python-adblock"
 repository = "https://github.com/ArniDagur/python-adblock"
 
-[package.metadata.maturin]
-classifier = [
-    "Programming Language :: Python",
-    "Programming Language :: Rust",
-    "License :: OSI Approved :: MIT License",
-    "License :: OSI Approved :: Apache Software License",
-]
-requires-python = ">=3.7"
-
 [profile.release]
 debug = true
 
diff --git a/pyproject.toml b/pyproject.toml
index c3656a4..f80b2f2 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -3,6 +3,12 @@ name = "adblock"
 version = "0.0.0"
 description = "Brave's adblocking in Python"
 authors = ["Árni Dagur <arni@dagur.eu>"]
+classifiers = [
+    "Programming Language :: Python",
+    "Programming Language :: Rust",
+    "License :: OSI Approved :: MIT License",
+    "License :: OSI Approved :: Apache Software License",
+]
 
 [tool.poetry.dependencies]
 python = "^3.6"
