--- ltrace-0.7.3.orig/sysdeps/linux-gnu/mips/plt.c
+++ ltrace-0.7.3/sysdeps/linux-gnu/mips/plt.c
@@ -22,7 +22,7 @@
  */
 
 #include <string.h>
-#include <error.h>
+#include <err.h>
 #include <errno.h>
 #include <gelf.h>
 #include <sys/ptrace.h>
@@ -166,8 +166,7 @@
 	sym_index += lte->arch.mips_gotsym;
 
 	if (gelf_getsym(lte->dynsym, sym_index, sym) == NULL){
-		error(EXIT_FAILURE, 0,
-			"Couldn't get relocation from \"%s\"", filename);
+		errx(EXIT_FAILURE, "Couldn't get relocation from \"%s\"", filename);
 	}
 
 	name = lte->dynstr + sym->st_name;
@@ -212,7 +211,7 @@
 	if (elf_get_section_type(lte, SHT_DYNAMIC, &scn, &shdr) < 0
 	    || scn == NULL) {
 	fail:
-		error(0, 0, "Couldn't get SHT_DYNAMIC: %s",
+		err(0, "Couldn't get SHT_DYNAMIC: %s",
 		      elf_errmsg(-1));
 		return -1;
 	}
