--- a/tests/test_pathlib2.py
+++ b/tests/test_pathlib2.py
@@ -42,7 +42,8 @@
     )
 
 try:
-    from test import support  # type: ignore
+    from test.support import os_helper as support  # type: ignore
+    from test.support.os_helper import TESTFN, can_symlink, skip_unless_symlink
 except ImportError:
     from test import test_support as support  # type: ignore
 
@@ -2297,7 +2298,7 @@
                          'pwd module does not expose getpwall()')
     def test_expanduser(self):
         P = self.cls
-        support.import_module('pwd')
+        #support.import_module('pwd')
         import pwd
         pwdent = pwd.getpwuid(os.getuid())
         username = pwdent.pw_name
