diff --git a/pyproject.toml b/pyproject.toml
index 9c141b6..c63442e 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -60,10 +60,6 @@ addopts = "-raXs --durations 10 --color=yes --doctest-modules"
 testpaths = [
     "tests/",
 ]
-filterwarnings = [
-    "error",
-    "ignore:Using or importing the ABCs from 'collections':DeprecationWarning:jsonschema"
-]
 
 [tool.mypy]
 check_untyped_defs = true
