From a17c04554ff647a91bd4804a9a1caf14d7063271 Mon Sep 17 00:00:00 2001
From: RemoveWingman <git@rw8.addy.io>
Date: Thu, 12 Mar 2026 21:18:07 +0100
Subject: [PATCH] fix poor text rendering https://github.com/lite-xl/lite-xl/issues/2209

---
 src/renderer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/renderer.c b/src/renderer.c
index 990c1a91..6db6ecd8 100644
--- a/src/renderer.c
+++ b/src/renderer.c
@@ -170,7 +170,7 @@ static int font_set_render_options(RenFont* font) {
     switch (font->hinting) {
       case FONT_HINTING_NONE: FT_Library_SetLcdFilter(library, FT_LCD_FILTER_NONE); break;
       case FONT_HINTING_SLIGHT:
-      case FONT_HINTING_FULL: FT_Library_SetLcdFilterWeights(library, weights); break;
+      case FONT_HINTING_FULL:
     }
     return FT_RENDER_MODE_LCD;
   } else {
-- 
2.53.0

