--- wine-1.7.8.orig/dlls/dnsapi/query.c
+++ wine-1.7.8/dlls/dnsapi/query.c
@@ -146,7 +146,9 @@
     case HOST_NOT_FOUND: return DNS_ERROR_RCODE_NAME_ERROR;
     case TRY_AGAIN:      return DNS_ERROR_RCODE_SERVER_FAILURE;
     case NO_RECOVERY:    return DNS_ERROR_RCODE_REFUSED;
+#ifdef NETDB_INTERNAL
     case NETDB_INTERNAL: return DNS_ERROR_RCODE;
+#endif
     default:
         FIXME( "unmapped error code: %d\n", error );
         return DNS_ERROR_RCODE_NOT_IMPLEMENTED;
--- wine-1.7.8.orig/dlls/ntdll/directory.c
+++ wine-1.7.8/dlls/ntdll/directory.c
@@ -119,6 +119,7 @@
     char           d_name[256];
 } KERNEL_DIRENT64;
 
+#undef getdents64
 static inline int getdents64( int fd, char *de, unsigned int size )
 {
     return syscall( __NR_getdents64, fd, de, size );
--- wine-1.7.8.orig/dlls/ws2_32/socket.c
+++ wine-1.7.8/dlls/ws2_32/socket.c
@@ -101,7 +101,7 @@
 #ifdef HAVE_NETIPX_IPX_H
 # include <netipx/ipx.h>
 # define HAS_IPX
-#elif defined(HAVE_LINUX_IPX_H)
+#elif defined(HAVE_LINUX_IPX_H) && defined(__GLIBC__)
 # ifdef HAVE_ASM_TYPES_H
 #  include <asm/types.h>
 # endif
--- wine-1.7.8.orig/server/file.h
+++ wine-1.7.8/server/file.h
@@ -21,6 +21,7 @@
 #ifndef __WINE_SERVER_FILE_H
 #define __WINE_SERVER_FILE_H
 
+#include <fcntl.h>
 #include "object.h"
 
 struct fd;
