--- a/makefile
+++ b/makefile
@@ -26,7 +26,6 @@
 
 CXXFLAGS += `sdl2-config --cflags` -DVERSION=$(VERSION) -DREVISION=$(REVISION) -DDATA_DIR=\"$(DATA_DIR)\"
-CXXFLAGS += -Wall -Wempty-body -Werror -Wstrict-prototypes -Werror=maybe-uninitialized -Warray-bounds
+CXXFLAGS += -Wall -Wempty-body -Wstrict-prototypes -Werror=maybe-uninitialized -Warray-bounds
-CXXFLAGS += -g -lefence
 
 LDFLAGS += `sdl2-config --libs` -lSDL2_mixer -lSDL2_image -lSDL2_ttf -lm
 
