--- a/src/log.hpp
+++ b/src/log.hpp
@@ -53,6 +53,7 @@
 
 #include <iostream> // needed else all files including log.hpp need to do it.
 #include <sstream> // as above. iostream (actually, iosfwd) declares stringstream as an incomplete type, but does not define it
+#include <cstdint>
 #include <string>
 #include <utility>
 #include <ctime>
--- a/src/deprecation.hpp
+++ b/src/deprecation.hpp
@@ -15,6 +15,7 @@
 #pragma once
 
 #include <string>
+#include <cstdint>
 
 /** See https://wiki.wesnoth.org/CompatibilityStandards for more info. */
 enum class DEP_LEVEL : uint8_t { INDEFINITE = 1, PREEMPTIVE, FOR_REMOVAL, REMOVED };
--- a/src/serialization/base64.hpp
+++ b/src/serialization.base64.hpp
@@ -15,6 +15,7 @@
 
 #pragma once
 
+#include <cstdint>
 #include <string_view>
 #include <vector>
 
