diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7d9d00b..55e1e12 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,9 +90,6 @@ else()
 endif()
 message("Installing DBus interfaces to ${DBUSIFACEDIR}")
 
-pkg_check_modules(SYSTEMD REQUIRED systemd)
-pkg_get_variable(SYSTEMD_USER_UNIT_DIR systemd systemduserunitdir)
-
 include_directories(${CMAKE_CURRENT_SOURCE_DIR})
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 79f7547..e7e0a76 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -27,45 +27,6 @@ com.lomiri.URLDispatcher.xml
 DESTINATION ${DBUSIFACEDIR}
 )
 
-###########################
-# systemd
-###########################
-
-configure_file(
-lomiri-url-dispatcher.service.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
-)
-
-configure_file(
-lomiri-url-dispatcher-update-system-dir.path.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-system-dir.path
-)
-
-configure_file(
-lomiri-url-dispatcher-update-system-dir.service.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-system-dir.service
-)
-
-configure_file(
-lomiri-url-dispatcher-update-user-dir.path.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-user-dir.path
-)
-
-configure_file(
-lomiri-url-dispatcher-update-user-dir.service.in
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-user-dir.service
-)
-
-install(
-FILES
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher.service
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-system-dir.path
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-system-dir.service
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-user-dir.path
-${CMAKE_CURRENT_BINARY_DIR}/lomiri-url-dispatcher-update-user-dir.service
-DESTINATION ${SYSTEMD_USER_UNIT_DIR}
-)
-
 ###########################
 # Click Hook
 ###########################
