--- a/sqlint.gemspec
+++ b/sqlint.gemspec
@@ -15,21 +15,9 @@
   EOF
 
   s.email = 'steve@sanityinc.com'
-  s.files = `git ls-files`.split($RS).reject do |file|
-    file =~ %r{^(?:
-    spec/.*
-    |Gemfile
-    |Rakefile
-    |\.rspec
-    |\.ruby-version
-    |\.rbenv-gemsets
-    |\.gitignore
-    |\.travis.yml
-    )$}x
-  end
+  s.files = Dir['bin/*', 'lib/**/*']
   s.test_files = []
   s.executables = s.files.grep(%r{^bin/}) { |f| File.basename(f) }
-  s.extra_rdoc_files = ['LICENSE.txt', 'README.md']
   s.homepage = 'https://github.com/purcell/sqlint'
   s.licenses = ['MIT']
   s.require_paths = ['lib']
