Patch from https://github.com/ranger/ranger/pull/2935
Author: Jonathan Chen <force@xe2.io>
Date:   Thu Jan 11 04:22:56 2024

--- a/ranger/gui/ui.py
+++ b/ranger/gui/ui.py
@@ -147,7 +147,11 @@ class UI(  # pylint: disable=too-many-instance-attributes,too-many-public-method
             pass
         if self.settings.mouse_enabled:
             _setup_mouse(dict(value=False))
-        curses.endwin()
+        try:
+            curses.endwin()
+        except curses.error:
+            pass
+
         self.is_on = False
 
     def set_load_mode(self, boolean):
