Updated: 23 Feb 2022 by Nicolas Lorin <androw95220@gmail.com>
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 23 Aug 2017 22:57:00 +0200
Subject: [PATCH] Fix default nimblepath in nim.cfg

--- a/config/nim.cfg
+++ b/config/nim.cfg.new
@@ -44,8 +44,8 @@
 path="$lib/pure"
 
 @if not windows:
-  nimblepath="/opt/nimble/pkgs2/"
-  nimblepath="/opt/nimble/pkgs/"
+  nimblepath="/usr/share/nimble/pkgs2/"
+  nimblepath="/usr/share/nimble/pkgs/"
 @else:
   # TODO:
 @end
