--- a/Makefile	2017-10-23 18:59:44.000000000 +0200
+++ b/Makefile	2019-11-22 18:54:48.089235572 +0100
@@ -134,7 +134,7 @@
 #
 # Flag ordering: arch, then -f, then -m, then -W
 DEBUG_FLAGS := $(if ${DEBUG},-g -O0,-Os)
-WERROR := -Werror
+WERROR := 
 COMMON_FLAGS := -nostdinc -pipe \
 	-ffreestanding -fno-builtin -fno-stack-protector \
 	${WERROR} -Wall -Wstrict-prototypes ${DEBUG_FLAGS}
@@ -648,7 +648,7 @@
 # The unified firmware utility will eventually replace all the others
 FUTIL_BIN = ${BUILD}/futility/futility
 # But we still need both static (tiny) and dynamic (with openssl) versions.
-FUTIL_STATIC_BIN = ${FUTIL_BIN}_s
+#FUTIL_STATIC_BIN = ${FUTIL_BIN}_s
 
 # These are the executables that are now built in to futility. We'll create
 # symlinks for these so the old names will still work.
@@ -1004,7 +1004,7 @@
 # Link tests for external repos
 ${BUILD}/host/linktest/extern: ${HOSTLIB}
 ${BUILD}/host/linktest/extern: LIBS = ${HOSTLIB}
-${BUILD}/host/linktest/extern: LDLIBS += -static
+#${BUILD}/host/linktest/extern: LDLIBS += -static
 TEST_OBJS += ${BUILD}/host/linktest/extern.o
 
 .PHONY: hostlib
@@ -1096,7 +1096,7 @@
 ${UTIL_BINS} ${UTIL_BINS_STATIC}: LIBS = ${UTILLIB}
 
 # Utilities for auto-update toolkits must be statically linked.
-${UTIL_BINS_STATIC}: LDFLAGS += -static
+#${UTIL_BINS_STATIC}: LDFLAGS += -static
 
 
 .PHONY: utils
