diff -ruN a/lib/asn1/typedefs.h b/lib/asn1/typedefs.h
--- a/lib/asn1/typedefs.h	2019-01-08 22:13:13.000000000 +0000
+++ b/lib/asn1/typedefs.h	2025-08-01 19:16:40.960000000 +0000
@@ -44,6 +44,10 @@
 #include <dirent.h>
 #include <time.h>
 
+#if HAVE_STDBOOL_H
+#  include <stdbool.h>
+#endif
+
 #define STR char
 #define STRA "s"
 
@@ -52,7 +56,9 @@
 
 #endif
 
+#if __STDC_VERSION__ < 202311L
 typedef unsigned char bool;
+#endif
 typedef unsigned int u_int;
 
 #define DBG1(s...)
