From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Thu, 26 Aug 2021 00:58 +0100
Subject: [PATCH] cloud-init: Adjust Alpine apk upgrade call

Add "--available" to Alpine "apk upgrade" call.

----

diff -aur a/cloudinit/distros/alpine.py b/cloudinit/distros/alpine.py
--- a/cloudinit/distros/alpine.py
+++ b/cloudinit/distros/alpine.py
@@ -120,6 +120,9 @@
         # Redirect output
         cmd.append("--quiet")
 
+        if command == 'upgrade':
+          cmd.append("--available")
+
         if args and isinstance(args, str):
             cmd.append(args)
         elif args and isinstance(args, list):
