diff --git a/otpush/push.py b/otpush/push.py
index cd660fd..8622e05 100755
--- a/otpush/push.py
+++ b/otpush/push.py
@@ -133,8 +133,9 @@ class RepoServer:
         address it is bound to. If the address is not received by the
         instance's timeout value, an exception is raised.
         """
-        addr_queue = multiprocessing.Queue()
-        self.proc = multiprocessing.Process(
+        ctx = multiprocessing.get_context('fork')
+        addr_queue = ctx.Queue()
+        self.proc = ctx.Process(
             target=self._run_server,
             args=(self.path, addr_queue)
         )
