--- a/etc/manticore.conf.in
+++ b/etc/manticore.conf.in
@@ -267,14 +267,14 @@
         listen                  = 127.0.0.1:9312
         listen                  = 127.0.0.1:9306:mysql
         listen                  = 127.0.0.1:9307:mysql_readonly
-        log                     = /var/piler/manticore/manticore.log
+        log                     = /var/log/piler/manticore.log
         binlog_max_log_size     = 256M
         binlog_path             = /var/piler/manticore
         binlog_flush            = 2
         # Uncomment the following line to enable the query_log
-        #query_log               = /var/piler/manticore/query.log
+        #query_log               = /var/log/piler/query.log
         network_timeout         = 5
-        pid_file                = /var/piler/run/searchd.pid
+        pid_file                = /run/piler/searchd.pid
         seamless_rotate         = 1
         preopen_tables          = 1
         unlink_old              = 1
