musl iconv does not support //IGNORE
diff -ruN a/webui/system/helper/mime.php b/webui/system/helper/mime.php
--- a/webui/system/helper/mime.php	2026-07-24 20:04:37.897749321 +0200
+++ b/webui/system/helper/mime.php	2026-07-24 20:04:55.685404883 +0200
@@ -330,7 +330,7 @@
             $headers['content-type']['charset'] = 'GBK';
          }
 
-         $body = iconv($headers['content-type']['charset'], 'utf-8' . '//IGNORE', $body);
+         $body = iconv($headers['content-type']['charset'], 'utf-8', $body);
       }
 
       if(strtolower($headers['content-type']['type']) == 'text/plain') {
