--- a/js/src/build/moz.build
+++ b/js/src/build/moz.build
@@ -83,15 +83,3 @@
 NO_EXPAND_LIBS = True
 
 DIST_INSTALL = True
-
-# Run SpiderMonkey style checker after linking the static library. This avoids
-# running the script for no-op builds.
-GENERATED_FILES += ['spidermonkey_checks']
-run_checks = GENERATED_FILES['spidermonkey_checks']
-run_checks.script = '/config/run_spidermonkey_checks.py'
-run_checks.inputs += [
-    '!%sjs_static.%s' % (CONFIG['LIB_PREFIX'], CONFIG['LIB_SUFFIX']),
-    '/config/check_spidermonkey_style.py',
-    '/config/check_macroassembler_style.py',
-    '/config/check_js_opcode.py'
-]
