--- a/tests/meson.build
+++ b/tests/meson.build
@@ -21,7 +21,6 @@
 tests = [
   [ 'test-coverage' ],
   [ 'testboundaries' ],
-  [ 'testboundaries_ucd' ],
   [ 'testcolor' ],
   [ 'testscript' ],
   [ 'testlanguage' ],
@@ -70,7 +69,6 @@
 
     if host_system != 'darwin'
       tests += [
-        [ 'test-layout', [ 'test-layout.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] + common_deps ],
         [ 'test-fonts', [ 'test-fonts.c', 'test-common.c' ], [ libpangocairo_dep, libpangoft2_dep ] + common_deps ],
         [ 'test-no-fonts', [ 'test-no-fonts.c' ], [ libpangocairo_dep, libpangoft2_dep ] ],
       ]
