diff --git a/pango/pango-coverage.h b/pango/pango-coverage.h
index 9189af2..e46f34d 100644
--- a/pango/pango-coverage.h
+++ b/pango/pango-coverage.h
@@ -25,7 +25,7 @@
 #include <glib.h>
 
 #include <pango/pango-version-macros.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
diff --git a/pango/pango-font.h b/pango/pango-font.h
index 13730fa..41d9dff 100644
--- a/pango/pango-font.h
+++ b/pango/pango-font.h
@@ -26,7 +26,7 @@
 #include <pango/pango-types.h>
 
 #include <glib-object.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
diff --git a/pango/pangofc-fontmap.h b/pango/pangofc-fontmap.h
index 495adc5..7835146 100644
--- a/pango/pangofc-fontmap.h
+++ b/pango/pangofc-fontmap.h
@@ -26,7 +26,7 @@
 #include <fontconfig/fontconfig.h>
 #include <pango/pangofc-decoder.h>
 #include <pango/pangofc-font.h>
-#include <hb.h>
+#include <harfbuzz/hb.h>
 
 G_BEGIN_DECLS
 
