--- a/src/anbox/cmds/system_info.cpp
+++ b/src/anbox/cmds/system_info.cpp
@@ -31,7 +31,9 @@
 #include "OpenGLESDispatch/EGLDispatch.h"
 
 #include "cpu_features_macros.h"
+#if defined(CPU_FEATURES_ARCH_X86)
 #include "cpuinfo_x86.h"
+#endif
 
 namespace fs = boost::filesystem;
 
--- a/src/anbox/cmds/check_features.cpp
+++ b/src/anbox/cmds/check_features.cpp
@@ -19,7 +19,9 @@
 #include "anbox/utils.h"
 
 #include "cpu_features_macros.h"
+#if defined(CPU_FEATURES_ARCH_X86)
 #include "cpuinfo_x86.h"
+#endif
 
 namespace {
 std::vector<std::string> cpu_whitelist = {
