diff --git a/external/process-cpp-minimal/include/core/posix/standard_stream.h b/external/process-cpp-minimal/include/core/posix/standard_stream.h
index f50a9b7..4a67a80 100644
--- a/external/process-cpp-minimal/include/core/posix/standard_stream.h
+++ b/external/process-cpp-minimal/include/core/posix/standard_stream.h
@@ -23,6 +23,10 @@

 #include <cstdint>

+#undef stdin
+#undef stdout
+#undef stderr
+
 namespace core
 {
 namespace posix
diff --git a/external/process-cpp-minimal/src/CMakeLists.txt b/external/process-cpp-minimal/src/CMakeLists.txt
index dec4659..20c9053 100644
--- a/external/process-cpp-minimal/src/CMakeLists.txt
+++ b/external/process-cpp-minimal/src/CMakeLists.txt
@@ -46,4 +46,5 @@ target_link_libraries(
 
   ${Boost_LIBRARIES}
   ${CMAKE_THREAD_LIBS_INIT}
+  execinfo
 )

