--- a/lib/screenshooter-capture.c	
+++ a/lib/screenshooter-capture.c	
@@ -381,6 +381,18 @@ static GdkPixbuf
                   rec_height += rectangle.y;
                 }
 
+              if (rec_x < 0)
+                {
+                  rec_width = rec_width + rec_x;
+                  rec_x = 0;
+                }
+
+              if (rec_y < 0)
+                {
+                  rec_height = rec_height + rec_y;
+                  rec_y = 0;
+                }
+
               if (x_orig + rec_x + rec_width > gdk_screen_width ())
                 rec_width = gdk_screen_width () - x_orig - rec_x;
 
