--- mhddfs-0.1.39/Makefile.orig
+++ mhddfs-0.1.39/Makefile
@@ -9,7 +9,7 @@
 #  This program is distributed in the hope that it will be useful,
 #  but WITHOUT ANY WARRANTY; without even the implied warranty of
 #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#  GNU General Public License for more details.
+	#  GNU General Public License for more details.
 
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <http://www.gnu.org/licenses/>.
@@ -24,6 +24,9 @@
 
 CFLAGS	=	-Wall $(shell pkg-config fuse --cflags) \
 			-DFUSE_USE_VERSION=26 -MMD
+
+CFLAGS	+=	-I${srcdir}/uthash-1.9.9/src
+
 ifdef WITHOUT_XATTR
 CFLAGS	+=	-DWITHOUT_XATTR
 endif
