--- mupdf-1.10-source/source/fitz/load-jpx.c.orig
+++ mupdf-1.10-source/source/fitz/load-jpx.c
@@ -481,14 +481,6 @@
 
 #else /* HAVE_LURATECH */
 
-/* Without the definition of OPJ_STATIC, compilation fails on windows
- * due to the use of __stdcall. We believe it is required on some
- * linux toolchains too. */
-#define OPJ_STATIC
-#ifndef _MSC_VER
-#define OPJ_HAVE_STDINT_H
-#endif
-
 #include <openjpeg.h>
 
 /* OpenJPEG does not provide a safe mechanism to intercept
