Apply fix https://github.com/metrics-rs/quanta/pull/121

--- a/src/clocks/monotonic/unix.rs
+++ b/src/clocks/monotonic/unix.rs
@@ -6,10 +6,7 @@
 impl Monotonic {
     #[allow(clippy::cast_sign_loss)]
     pub fn now(self) -> u64 {
-        let mut ts = libc::timespec {
-            tv_sec: 0,
-            tv_nsec: 0,
-        };
+        let mut ts = <libc::timespec as std::default::Default>::default();
         unsafe {
             libc::clock_gettime(libc::CLOCK_MONOTONIC, &mut ts);
         }
