From: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Date: Sat, 28 Dec 2019 15:17:53 +0100
Subject: Disable Qt Quick

---
 cmake/Libraries.cmake | 6 ------
 src/main.cpp          | 4 ----
 2 files changed, 10 deletions(-)

diff --git a/cmake/Libraries.cmake b/cmake/Libraries.cmake
index fed98dc..1d92f6e 100644
--- a/cmake/Libraries.cmake
+++ b/cmake/Libraries.cmake
@@ -13,12 +13,6 @@ FIND_PACKAGE(Qt5Svg ${MIN_QT_VERSION} REQUIRED)
 FIND_PACKAGE(Qt5LinguistTools ${MIN_QT_VERSION} REQUIRED)
 FIND_PACKAGE(Qt5WebSockets ${MIN_QT_VERSION} REQUIRED)
 
-IF(NOT ANDROID_BUILD_AAR)
-	FIND_PACKAGE(Qt5Qml ${MIN_QT_VERSION} REQUIRED)
-	FIND_PACKAGE(Qt5Quick ${MIN_QT_VERSION} REQUIRED)
-	FIND_PACKAGE(Qt5QuickControls2 ${MIN_QT_VERSION} REQUIRED)
-ENDIF()
-
 IF(DESKTOP)
 	FIND_PACKAGE(Qt5Widgets ${MIN_QT_VERSION} REQUIRED)
 
diff --git a/src/main.cpp b/src/main.cpp
index 54c9c78..35da465 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -97,10 +97,6 @@ Q_IMPORT_PLUGIN(UIPlugInWidgets)
 Q_IMPORT_PLUGIN(UIPlugInAidl)
 #endif
 
-#ifndef ANDROID_BUILD_AAR
-Q_IMPORT_PLUGIN(UIPlugInQml)
-#endif
-
 Q_IMPORT_PLUGIN(UIPlugInJson)
 Q_IMPORT_PLUGIN(UIPlugInWebSocket)
 
