From: John Paul Adrian Glaubitz <glaubitz@physik.fu-berlin.de>
Date: Sat, 28 Dec 2019 15:30:19 +0100
Subject: Fix path for resource file

---
 src/global/ResourceLoader.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/global/ResourceLoader.cpp b/src/global/ResourceLoader.cpp
index 76b71c2..a41f813 100644
--- a/src/global/ResourceLoader.cpp
+++ b/src/global/ResourceLoader.cpp
@@ -19,7 +19,7 @@ defineSingleton(ResourceLoader)
 ResourceLoader::ResourceLoader()
 	: mFilenames(
 		{
-			QStringLiteral("AusweisApp2.rcc")
+			QStringLiteral("/usr/share/AusweisApp2/AusweisApp2.rcc")
 		})
 	, mLoadedResources()
 {
@@ -42,7 +42,7 @@ void ResourceLoader::init()
 {
 	for (const auto& file : mFilenames)
 	{
-		QString path = FileDestination::getPath(file);
+		QString path = file;
 
 #ifndef QT_NO_DEBUG
 		const auto& filePathFromTestDir = QCoreApplication::applicationDirPath() + QStringLiteral("/../../src/") + file;
