--- a/kpatch-build/create-diff-object.c	2022-08-03 21:51:37.842435672 +0200
+++ b/kpatch-build/create-diff-object.c	2022-08-03 21:54:36.105879793 +0200
@@ -40,7 +40,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <error.h>
+#include <err.h>
 #include <gelf.h>
 #include <argp.h>
 #include <libgen.h>
@@ -57,7 +57,7 @@
 #define DIFF_FATAL(format, ...) \
 ({ \
 	fprintf(stderr, "ERROR: %s: " format "\n", childobj, ##__VA_ARGS__); \
-	error(EXIT_STATUS_DIFF_FATAL, 0, "unreconcilable difference"); \
+	err(EXIT_STATUS_DIFF_FATAL, "unreconcilable difference"); \
 })
 
 char *childobj;
--- a/kpatch-build/log.h	2022-08-03 21:58:58.461357398 +0200
+++ b/kpatch-build/log.h	2022-08-03 21:59:28.205088528 +0200
@@ -1,7 +1,7 @@
 #ifndef _LOG_H_
 #define _LOG_H_
 
-#include <error.h>
+#include <err.h>
 #include "kpatch.h"
 
 /* Files that include log.h must define loglevel and childobj */
@@ -9,7 +9,7 @@ extern enum loglevel loglevel;
 extern char *childobj;
 
 #define ERROR(format, ...) \
-	error(EXIT_STATUS_ERROR, 0, "ERROR: %s: %s: %d: " format, childobj, __FUNCTION__, __LINE__, ##__VA_ARGS__)
+	err(EXIT_STATUS_ERROR, "ERROR: %s: %s: %d: " format, childobj, __FUNCTION__, __LINE__, ##__VA_ARGS__)
 
 #define log_debug(format, ...) log(DEBUG, format, ##__VA_ARGS__)
 #define log_normal(format, ...) log(NORMAL, "%s: " format, childobj, ##__VA_ARGS__)
--- a/kpatch-build/kpatch-elf.c	2022-08-03 21:58:28.346617157 +0200
+++ b/kpatch-build/kpatch-elf.c	2022-08-03 22:00:38.050796676 +0200
@@ -27,7 +27,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <unistd.h>
-#include <error.h>
+#include <err.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 
--- a/kpatch-build/lookup.c	2022-08-03 22:06:59.318970951 +0200
+++ b/kpatch-build/lookup.c	2022-04-12 21:15:17.000000000 +0200
@@ -31,7 +31,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
-#include <error.h>
+#include <err.h>
 #include <gelf.h>
 #include <unistd.h>
 #include <libgen.h>
