diff --git a/node_build/make.js b/node_build/make.js
index 18397b1..a7f9da9 100644
--- a/node_build/make.js
+++ b/node_build/make.js
@@ -26,7 +26,7 @@ var LDFLAGS = process.env['LDFLAGS'];
 // march=native really only makes a lot of sense on x86/amd64 where the available features
 // are a hodgepodge per-CPU. On arm (32) you may or may not have NEON available but in any
 // case clang doesn't reliably support march except on x86/amd64.
-var NO_MARCH_FLAG = ['arm', 'ppc', 'ppc64', 'arm64'];
+var NO_MARCH_FLAG = ['arm', 'arm64', 'ppc', 'ppc64', 'arm64'];
 
 Builder.configure({
     buildDir: process.env['OUT_DIR'], // set by cargo
