diff --git a/brctl/brctl.c b/brctl/brctl.c
index 454b8dd..8a7f095 100644
--- a/brctl/brctl.c
+++ b/brctl/brctl.c
@@ -19,7 +19,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <sys/errno.h>
+#include <errno.h>
 #include <getopt.h>
 
 #include "libbridge.h"
diff --git a/libbridge/libbridge.h b/libbridge/libbridge.h
index 39964f2..73de4fb 100644
--- a/libbridge/libbridge.h
+++ b/libbridge/libbridge.h
@@ -23,6 +23,9 @@
 #include <linux/if.h>
 #include <linux/if_bridge.h>
 
+#include <sys/types.h>
+#include <sys/time.h>
+
 /* defined in net/if.h but that conflicts with linux/if.h... */
 extern unsigned int if_nametoindex (const char *__ifname);
 extern char *if_indextoname (unsigned int __ifindex, char *__ifname);
diff --git a/libbridge/libbridge_devif.c b/libbridge/libbridge_devif.c
index aa8bc36..1da3549 100644
--- a/libbridge/libbridge_devif.c
+++ b/libbridge/libbridge_devif.c
@@ -23,7 +23,7 @@
 #include <errno.h>
 #include <string.h>
 #include <dirent.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 
 #include "libbridge.h"
 #include "libbridge_private.h"
diff --git a/libbridge/libbridge_if.c b/libbridge/libbridge_if.c
index 77d3f8a..9cf4bac 100644
--- a/libbridge/libbridge_if.c
+++ b/libbridge/libbridge_if.c
@@ -20,7 +20,7 @@
 #include <stdlib.h>
 #include <errno.h>
 #include <string.h>
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/ioctl.h>
 
 #include "libbridge.h"
