--- a/asyncgit/Cargo.toml
+++ b/asyncgit/Cargo.toml
@@ -16,8 +16,6 @@ scopetime = { path = "../scopetime", ver
 git2 = "0.13"
 # git2 = { path = "../../extern/git2-rs", features = ["vendored-openssl"]}
 # git2 = { git="https://github.com/extrawurst/git2-rs.git", rev="fc13dcc", features = ["vendored-openssl"]}
-# pinning to vendored openssl, using the git2 feature this gets lost with new resolver
-openssl-sys = { version = '0.9', features= ["vendored"] }
 rayon-core = "1.9"
 crossbeam-channel = "0.5"
 log = "0.4"
