--- a/asyncgit/Cargo.toml
+++ b/asyncgit/Cargo.toml
@@ -13,7 +13,7 @@
 
 [dependencies]
 scopetime = { path = "../scopetime", version = "0.1" }
-git2 = { version = "0.13", features = ["vendored-openssl"] }
+git2 = { version = "0.13" }
 # git2 = { path = "../../github/git2-rs", features = ["vendored-openssl"]}
 # git2 = { git="https://github.com/extrawurst/git2-rs.git", rev="513a8c9", features = ["vendored-openssl"]}
 rayon-core = "1.9"
