diff --git a/tests/test_mocker.py b/tests/test_mocker.py
index 7a4061c..c2e8881 100644
--- a/tests/test_mocker.py
+++ b/tests/test_mocker.py
@@ -219,7 +219,7 @@ class MockerTests(base.TestCase):
         @requests_mock.Mocker()
         def inner(arg1, m):
             self._do_test(m)
-            self.assertEquals(
+            self.assertEqual(
                 len(m.request_history), 1,
                 "Failed to provide clean mock on subsequent calls"
             )
