From 63fb21fb6257be7e1e5ff7a96019ae5c5f3a6609 Mon Sep 17 00:00:00 2001
From: Integral <integral@member.fsf.org>
Date: Thu, 26 Jun 2025 18:14:00 +0800
Subject: [PATCH] fix: resolve Wayland icon issue by correcting desktop
 filename

---
 src/kiwixapp.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/kiwixapp.cpp b/src/kiwixapp.cpp
index b3d6339d3..aeeb12a61 100644
--- a/src/kiwixapp.cpp
+++ b/src/kiwixapp.cpp
@@ -84,7 +84,7 @@ void KiwixApp::init()
     setWindowIcon(icon);
 
     setApplicationName("Kiwix");
-    setDesktopFileName("kiwix.desktop");
+    setDesktopFileName("org.kiwix.desktop.desktop");
     setStyleSheet(parseStyleFromFile(":/css/style.css"));
 
     createActions();
