From 238bd534a4f99ee331823e8ec2766c21999098c4 Mon Sep 17 00:00:00 2001
From: Matthieu Gautier <mgautier@kymeria.fr>
Date: Tue, 19 Sep 2023 16:38:58 +0200
Subject: [PATCH] SearchRenderer now take the NameMapper on the getHtml method.

---
 src/urlschemehandler.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/urlschemehandler.cpp b/src/urlschemehandler.cpp
index 8a6dd3d3..956c8840 100644
--- a/src/urlschemehandler.cpp
+++ b/src/urlschemehandler.cpp
@@ -133,7 +133,6 @@ UrlSchemeHandler::handleSearchRequest(QWebEngineUrlRequestJob* request)
     }
     kiwix::SearchRenderer renderer(
         search->getResults(start, pageLength),
-        std::make_shared<IdNameMapper>(),
         search->getEstimatedMatches(),
         start);
     renderer.setSearchPattern(searchQuery);
@@ -141,7 +140,8 @@ UrlSchemeHandler::handleSearchRequest(QWebEngineUrlRequestJob* request)
     renderer.setProtocolPrefix("zim://");
     renderer.setSearchProtocolPrefix("zim://" + host.toStdString() + "/");
     renderer.setPageLength(pageLength);
-    auto content = renderer.getHtml();
+    IdNameMapper mapper;
+    auto content = renderer.getHtml(mapper, nullptr);
     QBuffer *buffer = new QBuffer;
     buffer->setData(content.data(), content.size());
     connect(request, &QObject::destroyed, buffer, &QObject::deleteLater);
