diff --git a/vendor/opencc-rust/build.rs b/vendor/opencc-rust/build.rs
index 856bc53..1f937ca 100644
--- a/vendor/opencc-rust/build.rs
+++ b/vendor/opencc-rust/build.rs
@@ -4,7 +4,7 @@ use std::path::PathBuf;
 use std::process::Command;
 
 const MIN_VERSION: &str = "1.1.2";
-const MAX_VERSION: &str = "1.2.0";
+const MAX_VERSION: &str = "2.0.0";
 
 fn main() {
     if cfg!(target_os = "freebsd") {
