From 4aad215befceabc5145e7f543763b210a3811d4e Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Tue, 18 Aug 2026 16:26:35 +0200
Subject: [PATCH] npm-ci

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index f725714..e0180f5 100644
--- a/Makefile
+++ b/Makefile
@@ -42,7 +42,7 @@ build-ext:  ## Build package extensions in-place
 	$(PYTHON) -m pip install --no-build-isolation --config-settings=editable.rebuild=true -ve .
 
 $(reporters_path)/templates/assets/%.js: $(reporters_path)/assets/%.js
-	$(NPM) install
+	$(NPM) ci --ignore-scripts
 	$(NPM) run-script build
 	touch $(generated_js_files)
 
@@ -50,7 +50,7 @@ $(reporters_path)/templates/assets/%.js: $(reporters_path)/assets/%.js
 build-js: $(generated_js_files)  ## Build package assets in-place
 
 $(vendor_assets): package.json webpack.config.js
-	$(NPM) install
+	$(NPM) ci --ignore-scripts
 	$(NPM) run-script build
 	touch $(vendor_assets)
 
-- 
2.55.0

