--- a/spec/spec_helper.rb.orig
+++ b/spec/spec_helper.rb
@@ -1,4 +1,4 @@
-require "bundler/setup"
+#require "bundler/setup"  # XXX-Patched
 require "zstd-ruby"
 
 RSpec.configure do |config|
--- a/spec/zstd-ruby-stream_reader_spec.rb
+++ b/spec/zstd-ruby-stream_reader_spec.rb
@@ -1,6 +1,6 @@
 require "spec_helper"
 require 'zstd-ruby'
-require 'pry'
+#require 'pry'  # XXX-Patched
 
 RSpec.describe Zstd::StreamReader do
   describe 'read' do
