Stop messing with CFLAGS!

--- a/configure
+++ b/configure
@@ -2139,8 +2139,8 @@
 MYPCFILES="kyotocabinet.pc"
 
 # Building flags
-MYCFLAGS="-Wall -ansi -pedantic -fPIC -fsigned-char -g0 -O2"
-MYCXXFLAGS="-Wall -fPIC -fsigned-char -g0 -O2"
+MYCFLAGS="-Wall -ansi -pedantic -fPIC -fsigned-char"
+MYCXXFLAGS="-Wall -fPIC -fsigned-char"
 MYCPPFLAGS="-I. -I\$(INCLUDEDIR) -I/usr/local/include"
 MYCPPFLAGS="$MYCPPFLAGS -DNDEBUG -D_GNU_SOURCE=1"
 MYCPPFLAGS="$MYCPPFLAGS -D_FILE_OFFSET_BITS=64 -D_REENTRANT -D__EXTENSIONS__"
@@ -4006,27 +4004,6 @@
 
 
 # System-depending optimization
-printf 'checking for 64-bit availability... '
-if printf 'int main(void) {return 0;}' | $CC -xc -m64 -o config.tmp - >config.tmp 2>&1
-then
-  MYCFLAGS="-m64 $MYCFLAGS"
-  MYCXXFLAGS="-m64 $MYCXXFLAGS"
-  printf 'yes\n'
-else
-  printf 'no\n'
-fi
-if test "$enable_opt" != "no"
-then
-  printf 'checking for CPU optimization availability... '
-  if printf 'int main(void) {return 0;}' | $CC -xc -march=native -o config.tmp - >config.tmp 2>&1
-  then
-    MYCFLAGS="-march=native $MYCFLAGS"
-    MYCXXFLAGS="-march=native $MYCXXFLAGS"
-    printf 'yes\n'
-  else
-    printf 'no\n'
-  fi
-fi
 printf 'checking for useless warnings... '
 if printf 'int main(void) {return 0;}' | $CC -xc \
   -Wno-unused-but-set-variable -Wno-unused-but-set-parameter -o config.tmp - >config.tmp 2>&1
