https://github.com/php/php-src/issues/12635

diff --git a/ext/intl/tests/bug69398.phpt b/ext/intl/tests/bug69398.phpt
index 02c4b7daef..d7633d54bb 100644
--- a/ext/intl/tests/bug69398.phpt
+++ b/ext/intl/tests/bug69398.phpt
@@ -13,5 +13,5 @@
 echo IntlDateFormatter::formatObject ($date, array(IntlDateFormatter::SHORT, IntlDateFormatter::NONE), 'ko_KR'), "\n";
 ?>
 --EXPECT--
-04/04/2015
+4/4/15
 15. 4. 4.
