diff --git a/ext/openssl/tests/openssl_error_string_basic.phpt b/ext/openssl/tests/openssl_error_string_basic.phpt
index e4ea264b3b..fd1f7ef9d3 100644
--- a/ext/openssl/tests/openssl_error_string_basic.phpt
+++ b/ext/openssl/tests/openssl_error_string_basic.phpt
@@ -6,6 +6,8 @@ openssl
 <?php
 if (OPENSSL_VERSION_NUMBER >= 0x30000000) die('skip For OpenSSL < 3.0');
 ?>
+--XFAIL--
+Somehow unstable
 --FILE--
 <?php
 // helper function to check openssl errors
