--- a/resolv.gemspec
+++ b/resolv.gemspec
@@ -20,9 +20,7 @@
   spec.metadata["homepage_uri"] = spec.homepage
   spec.metadata["source_code_uri"] = spec.homepage
 
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
+  spec.files         = Dir["lib/**/*"]
   spec.bindir        = "exe"
   spec.executables   = []
   spec.require_paths = ["lib"]
