--- a/resolv.gemspec	2025-12-10 00:11:45.000000000 -0500
+++ b/resolv.gemspec	2025-12-14 12:14:34.664637505 -0500
@@ -22,7 +22,7 @@
   spec.metadata["source_code_uri"] = spec.homepage
 
   excludes = %W[/.git* /bin /test /*file /#{File.basename(__FILE__)}]
-  spec.files = IO.popen(%W[git -C #{__dir__} ls-files -z --] + excludes.map {|e| ":^#{e}"}, &:read).split("\x0")
+  spec.files = Dir["lib/**/*"]
   spec.bindir        = "exe"
   spec.executables   = []
   spec.require_paths = ["lib"]
