--- resolv-0.6.0-origin/Rakefile
+++ resolv-0.6.0/Rakefile
@@ -1,7 +1,8 @@
-require "bundler/gem_tasks"
+require "mkmf"
 require "rake/testtask"
 
-if RUBY_ENGINE == "ruby"
+if RUBY_ENGINE == "ruby" and have_library("iphlpapi")
+  require "bundler/gem_tasks"
   require "ruby-core/extensiontask"
   helper = Bundler::GemHelper.instance
   extask = RubyCore::ExtensionTask.new(helper.gemspec)
@@ -9,7 +10,9 @@
 end
 
 Rake::TestTask.new(:test) do |t|
-  t.libs.unshift(*extask.libs) if extask
+  if extask
+    t.libs.unshift(*extask.libs)
+  end
   t.libs << "test/lib"
   t.ruby_opts << "-rhelper"
   t.test_files = FileList["test/**/test_*.rb"]
