Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: read PLUGIN_INSTALLATION_MESSAGE from environment
----

--- a/cmake/helpers/GdalDriverHelper.cmake
+++ b/cmake/helpers/GdalDriverHelper.cmake
@@ -80,8 +80,9 @@
     else()
         set(_var_PLUGIN_INSTALLATION_MESSAGE "OGR_DRIVER_${_KEY}_PLUGIN_INSTALLATION_MESSAGE")
     endif()
-    if (DEFINED ${_var_PLUGIN_INSTALLATION_MESSAGE})
-        target_compile_definitions(${_DRIVER_TARGET}_core PRIVATE "-DPLUGIN_INSTALLATION_MESSAGE=\"${${_var_PLUGIN_INSTALLATION_MESSAGE}}\"")
+    if (DEFINED ENV{${_var_PLUGIN_INSTALLATION_MESSAGE}})
+        message(STATUS "**** setting -DPLUGIN_INSTALLATION_MESSAGE=\"$ENV{${_var_PLUGIN_INSTALLATION_MESSAGE}}\"")
+        target_compile_definitions(${_DRIVER_TARGET}_core PRIVATE "-DPLUGIN_INSTALLATION_MESSAGE=\"$ENV{${_var_PLUGIN_INSTALLATION_MESSAGE}}\"")
     endif()
     gdal_standard_includes(${_DRIVER_TARGET}_core)
     add_dependencies(${_DRIVER_TARGET}_core generate_gdal_version_h)
~