--- old/lib/src/main/java/org/asamk/signal/manager/config/ServiceConfig.java
+++ new/lib/src/main/java/org/asamk/signal/manager/config/ServiceConfig.java
@@ -48,14 +48,13 @@
 
     public static boolean isSignalClientAvailable() {
         try {
-            try {
-                org.signal.libsignal.internal.Native.UuidCiphertext_CheckValidContents(new byte[0]);
-            } catch (IllegalArgumentException ignored) {
-            }
+            org.signal.libsignal.internal.Native.UuidCiphertext_CheckValidContents(new byte[0]);
             return true;
         } catch (UnsatisfiedLinkError e) {
             logger.warn("Failed to call libsignal-client: {}", e.getMessage());
             return false;
+        } catch (Exception e) {
+            return true;
         }
     }
 
