--- a/pTk/mTk/additions/ClientWin.c
+++ b/pTk/mTk/additions/ClientWin.c
@@ -30,7 +30,7 @@ in this Software without prior written authorization from the X Consortium.
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 
-static Window TryChildren();
+static Window TryChildren (Display *dpy, Window win, Atom WM_STATE);
 
 /* Find a window with WM_STATE, else return win itself, as per ICCCM */
 
--- a/pTk/mTk/unix/tkUnixEvent.c
+++ b/pTk/mTk/unix/tkUnixEvent.c
@@ -416,8 +416,8 @@
 	 * message.
 	 */
 
-	void (*oldHandler)();
+	void (*oldHandler)(int);
 
-	oldHandler = (void (*)()) signal(SIGPIPE, SIG_IGN);
+	oldHandler = (void (*)(int)) signal(SIGPIPE, SIG_IGN);
 	XNoOp(display);
 	XFlush(display);
--- a/pTk/mTk/unix/tkUnixSelect.c
+++ b/pTk/mTk/unix/tkUnixSelect.c
@@ -367,7 +367,7 @@
 	    ((char *) buffer)[numItems] = 0;
 
 	    errorHandler = Tk_CreateErrorHandler(eventPtr->xproperty.display,
-		    -1, -1, -1, (int (*)()) NULL, (ClientData) NULL);
+		    -1, -1, -1, NULL, (ClientData) NULL);
 	    /*
 	     * Encode the data using the proper format for each type.
 	     */
@@ -872,7 +872,7 @@
 	    Tcl_GetThreadData(&dataKey, sizeof(ThreadSpecificData));
 
     errorHandler = Tk_CreateErrorHandler(eventPtr->display, -1, -1,-1,
-	    (int (*)()) NULL, (ClientData) NULL);
+	    NULL, (ClientData) NULL);
 
     /*
      * Initialize the reply event.
@@ -1149,7 +1149,7 @@
 	Tcl_DeleteTimerHandler(incr.timeout);
 	/* winPtr may have been destroyed now */
 	errorHandler = Tk_CreateErrorHandler(reply.display,
-		-1, -1,-1, (int (*)()) NULL, (ClientData) NULL);
+		-1, -1,-1, NULL, (ClientData) NULL);
 	XSelectInput(reply.display, reply.requestor, 0L);
 	Tk_DeleteErrorHandler(errorHandler);
 	if (tsdPtr->pendingIncrs == &incr) {
