--- a/tests/test-data/test_configs/example.toml
+++ b/tests/test-data/test_configs/example.toml
@@ -38,3 +38,7 @@
 ## directory: path on the host filesystem to where zone files are stored.
 # directory = "/var/named"
+# the zone files for this example config were copied to $_zonedir.
+# when setting up your zones, it is recommended you copy the files you
+# need to /var/named and use that directory instead.
+directory = "$_zonedir"
 
--- a/bin/tests/integration/config_tests.rs
+++ b/bin/tests/integration/config_tests.rs
@@ -47,7 +47,7 @@ fn test_read_config() {
     assert_eq!(config.listen_addrs_ipv6(), Ok(Vec::<Ipv6Addr>::new()));
     assert_eq!(config.tcp_request_timeout(), Duration::from_secs(5));
     assert_eq!(config.log_level(), tracing::Level::INFO);
-    assert_eq!(config.directory(), Path::new("/var/named"));
+    assert!(config.directory().is_absolute());

     assert_eq!(config.zones()[0].zone, "localhost");
     assert_eq!(config.zones()[0].zone_type(), ZoneType::Primary);
