--- a/lld/CMakeLists.txt
+++ b/lld/CMakeLists.txt
@@ -168,6 +168,9 @@ if (LLD_USE_VTUNE)
   endif()
 endif()
 
+option(LLD_BUILD_SHARED_LIBS
+  "Build lld libraries as shared libraries instead of static" OFF)
+
 option(LLD_BUILD_TOOLS
   "Build the lld tools. If OFF, just generate build targets." ON)
 
--- a/lld/cmake/modules/AddLLD.cmake
+++ b/lld/cmake/modules/AddLLD.cmake
@@ -7,7 +7,7 @@ macro(add_lld_library name)
     ""
     ""
     ${ARGN})
-  if(ARG_SHARED)
+  if( LLD_BUILD_SHARED_LIBS OR ARG_SHARED )
     set(ARG_ENABLE_SHARED SHARED)
   endif()
   llvm_add_library(${name} ${ARG_ENABLE_SHARED} ${ARG_UNPARSED_ARGUMENTS})
