diff --git a/etc/systemd/ibft-rule-generator b/etc/systemd/ibft-rule-generator
index a464fd6..f0fcd32 100644
--- a/etc/systemd/ibft-rule-generator
+++ b/etc/systemd/ibft-rule-generator
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Systemd rule generator for ibft interfaces
 #
diff --git a/iscsiuio/src/unix/build_date.sh b/iscsiuio/src/unix/build_date.sh
index 65888fe..be999c8 100755
--- a/iscsiuio/src/unix/build_date.sh
+++ b/iscsiuio/src/unix/build_date.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # build the build_date.c and build_date.h files
 #
diff --git a/libopeniscsiusr/docs/list-man-pages.sh b/libopeniscsiusr/docs/list-man-pages.sh
index dda8277..044eb71 100755
--- a/libopeniscsiusr/docs/list-man-pages.sh
+++ b/libopeniscsiusr/docs/list-man-pages.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # list man pages found given one or more ??? passed in
 #
diff --git a/utils/iscsi-gen-initiatorname.sh.template b/utils/iscsi-gen-initiatorname.sh.template
index 5c2bfdf..86148f0 100644
--- a/utils/iscsi-gen-initiatorname.sh.template
+++ b/utils/iscsi-gen-initiatorname.sh.template
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # iscsi-gen-initiatorname
 #
diff --git a/utils/iscsi_discovery.sh b/utils/iscsi_discovery.sh
index be2f390..fbe066b 100755
--- a/utils/iscsi_discovery.sh
+++ b/utils/iscsi_discovery.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Copyright (C) Voltaire Ltd. 2006.  ALL RIGHTS RESERVED.
 #
diff --git a/utils/iscsi_fw_login.sh.template b/utils/iscsi_fw_login.sh.template
index aae9e4c..6758da2 100644
--- a/utils/iscsi_fw_login.sh.template
+++ b/utils/iscsi_fw_login.sh.template
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # iscsi_fw_login -- login to iscsi firmware targets, if any
 #
diff --git a/utils/iscsi_offload.sh b/utils/iscsi_offload.sh
index 1869fe1..5de465a 100755
--- a/utils/iscsi_offload.sh
+++ b/utils/iscsi_offload.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # iscsi_offload
 #
@@ -239,7 +239,7 @@ if iscsiadm -m iface -I $ioe_iface > /dev/null 2>&1 ; then
     ioe_mac=$(iscsiadm -m iface -I $ioe_iface 2> /dev/null| sed -n "s/iface\.hwaddress = \(.*\)/\1/p")
     ioe_mod=$(iscsiadm -m iface -I $ioe_iface 2> /dev/null| sed -n "s/iface\.transport_name = \(.*\)/\1/p")
     ipaddr=$(iscsiadm -m iface -I $ioe_iface 2> /dev/null| sed -n "s/iface\.ipaddress = \(.*\)/\1/p")
-    if [ "$ipaddr" == "<empty>" ] ; then
+    if [ "$ipaddr" = "<empty>" ] ; then
 	ipaddr=
     fi
 elif [ "$mod" = "be2iscsi" ] ; then
