diff --git a/usr/Makefile b/usr/Makefile
index f9445ad..6d8ff62 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -60,14 +60,14 @@ all: $(PROGRAMS)
 
 iscsid: $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(DISCOVERY_SRCS) \
 	iscsid.o session_mgmt.o discoveryd.o mntcheck.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@  -lisns -lcrypto -lrt -lmount $(ISCSI_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@  -lisns -lcrypto -lrt -lmount $(ISCSI_LIB) $(LDFLAGS)
 
 iscsiadm: $(ISCSI_LIB_SRCS) $(DISCOVERY_SRCS) iscsiadm.o session_mgmt.o mntcheck.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -lisns -lcrypto -lmount $(ISCSI_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -lisns -lcrypto -lmount $(ISCSI_LIB) $(LDFLAGS)
 
 iscsistart: $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(FW_BOOT_SRCS) \
 		iscsistart.o statics.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -lrt $(ISCSI_LIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ -o $@ -lrt $(ISCSI_LIB) $(LDFLAGS)
 clean:
 	rm -f *.o $(PROGRAMS) .depend $(LIBSYS)
 
diff --git a/utils/Makefile b/utils/Makefile
index f65f1e7..472a1b4 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -7,7 +7,7 @@ PROGRAMS = iscsi-iname
 all: $(PROGRAMS)
 
 iscsi-iname: md5.o iscsi-iname.o
-	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(DBM_LIB) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $^ $(DBM_LIB) -o $@ $(LDFLAGS)
 
 clean:
 	rm -f *.o $(PROGRAMS) .depend
