diff -Nuar open-iscsi-2.0-871.orig/usr/Makefile open-iscsi-2.0-871/usr/Makefile
--- open-iscsi-2.0-871.orig/usr/Makefile	2009-07-10 20:55:58.000000000 -0700
+++ open-iscsi-2.0-871/usr/Makefile	2009-11-12 00:58:16.467960344 -0800
@@ -48,14 +48,14 @@
 all: $(PROGRAMS)
 
 iscsid: $(COMMON_SRCS) $(IPC_OBJ) $(INITIATOR_SRCS) iscsid.o
-	$(CC) $(CFLAGS) $^ -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
 iscsiadm: $(COMMON_SRCS) $(FW_BOOT_SRCS) strings.o discovery.o iscsiadm.o
-	$(CC) $(CFLAGS) $^ -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 
 iscsistart: $(IPC_OBJ) $(ISCSI_LIB_SRCS) $(INITIATOR_SRCS) $(FW_BOOT_SRCS) \
 		iscsistart.o statics.o
-	$(CC) $(CFLAGS) -static $^ -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^
 clean:
 	rm -f *.o $(PROGRAMS) .depend $(LIBSYS)
 
diff -Nuar open-iscsi-2.0-871.orig/utils/Makefile open-iscsi-2.0-871/utils/Makefile
--- open-iscsi-2.0-871.orig/utils/Makefile	2009-07-10 20:55:58.000000000 -0700
+++ open-iscsi-2.0-871/utils/Makefile	2009-11-12 00:58:56.404623435 -0800
@@ -6,7 +6,7 @@
 all: $(PROGRAMS)
 
 iscsi-iname: md5.o iscsi-iname.o
-	$(CC) $(CFLAGS) $^ $(DBM_LIB) -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(DBM_LIB) $^
 
 clean:
 	rm -f *.o $(PROGRAMS) .depend
