Index: asterisk-team-group-issue8824/main/xmldoc.c
===================================================================
--- asterisk-team-group-issue8824.orig/main/xmldoc.c	2008-12-11 14:44:01.000000000 +0200
+++ asterisk-team-group-issue8824/main/xmldoc.c	2008-12-11 14:44:35.000000000 +0200
@@ -1786,8 +1786,8 @@
 	ast_register_atexit(xmldoc_unload_documentation);
 
 	/* Get every *-LANG.xml file inside $(ASTDATADIR)/documentation */
-	ast_asprintf(&xmlpattern, "%s/documentation{/thirdparty/,/}*-{%s,%.2s_??,%s}.xml", ast_config_AST_DATA_DIR,
-			documentation_language, documentation_language, default_documentation_language);
+	ast_asprintf(&xmlpattern, "%s/documentation/*-%s.xml", ast_config_AST_DATA_DIR,
+			documentation_language);
 	globbuf.gl_offs = 0;    /* initialize it to silence gcc */
 	globret = glob(xmlpattern, MY_GLOB_FLAGS, NULL, &globbuf);
 	if (globret == GLOB_NOSPACE) {
