diff --git a/CMakeLists.txt b/CMakeLists.txt
index 193bbe0..ee1fd6a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -320,9 +320,6 @@ install(TARGETS postsrsd RUNTIME DESTINATION ${CMAKE_INSTALL_SBINDIR})
 install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.conf"
         DESTINATION "${CMAKE_INSTALL_DATADIR}/doc/${PROJECT_NAME}"
 )
-install(FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}.service"
-        DESTINATION "${SYSTEMD_UNITDIR}"
-)
 
 if(GENERATE_SRS_SECRET)
     find_program(DD dd DOC "path to dd executable")
