diff --git a/data/postsrsd.conf.in b/data/postsrsd.conf.in
index fd3ecba..f93d478 100644
--- a/data/postsrsd.conf.in
+++ b/data/postsrsd.conf.in
@@ -44,7 +44,7 @@ domains = {}
 #     socketmap = unix:/var/spool/postfix/srs
 #     socketmap = inet:localhost:10003
 #
-socketmap = unix:/var/spool/postfix/srs
+socketmap = inet:localhost:10003
 
 # Socketmap connection keep-alive timeout.
 # After PostSRSd has served a socketmap request, it will keep the connection
@@ -112,7 +112,7 @@ original-envelope = embedded
 # Example:
 #     secrets-file = "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@.secret"
 #
-secrets-file = "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@.secret"
+secrets-file = "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@/@PROJECT_NAME@.secret"
 
 # SRS tag separator
 # This is the character following the initial SRS0 or SRS1 tag of a generated
diff --git a/src/postsrsd_build_config.h.in b/src/postsrsd_build_config.h.in
index f2d89b7..15e9d2e 100644
--- a/src/postsrsd_build_config.h.in
+++ b/src/postsrsd_build_config.h.in
@@ -20,8 +20,8 @@
 /* clang-format off */
 #define POSTSRSD_VERSION          "@PROJECT_VERSION@"
 #define SIZEOF_UNSIGNED_LONG      @SIZEOF_UNSIGNED_LONG@
-#define DEFAULT_CONFIG_FILE       "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@.conf"
-#define DEFAULT_SECRETS_FILE      "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@.secret"
+#define DEFAULT_CONFIG_FILE       "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@/@PROJECT_NAME@.conf"
+#define DEFAULT_SECRETS_FILE      "@CMAKE_INSTALL_FULL_SYSCONFDIR@/@PROJECT_NAME@/@PROJECT_NAME@.secret"
 #define DEFAULT_CHROOT_DIR        "@POSTSRSD_CHROOTDIR@"
 #define DEFAULT_POSTSRSD_USER     "@POSTSRSD_USER@"
 /* clang-format on */
