--- ./src/rebar_src_test.config.in
+++ ./src/rebar_src_test.config.in
@@ -8,8 +8,6 @@
 {sub_dirs,
  ["lib/cloudi_core",
   "lib/cloudi_service_api_requests",
-  "lib/cloudi_service_db_mysql",
-  "lib/cloudi_service_db_pgsql",
   "lib/cloudi_service_filesystem",
   "lib/cloudi_service_http_client",
   "lib/cloudi_service_http_cowboy1",
@@ -18,9 +16,7 @@
   "lib/cloudi_service_map_reduce",
   "lib/cloudi_service_monitoring",
   "lib/cloudi_service_null",
-  "lib/cloudi_service_oauth1",
   "lib/cloudi_service_queue",
-  "lib/cloudi_service_quorum",
   "lib/cloudi_service_request_rate",
   "lib/cloudi_service_router",
   "lib/cloudi_service_tcp",
