diff --git a/src/libknot/internal/wire_ctx.h b/src/libknot/internal/wire_ctx.h
index ae15390..a96250f 100644
--- a/src/libknot/internal/wire_ctx.h
+++ b/src/libknot/internal/wire_ctx.h
@@ -18,11 +18,14 @@
 
 #include <assert.h>
 #include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
 #include <string.h>
+#include <sys/types.h>
 
-#include "libknot/rdata.h"
 #include "libknot/internal/endian.h"
 #include "libknot/internal/errcode.h"
+#include "libknot/rdata.h"
 
 /*!
  * \brief Struct to keep the wire context.
