diff -rupN a/tests/test_decoder.py b/tests/test_decoder.py
--- a/tests/test_decoder.py	2024-12-04 02:09:45.093099392 +0100
+++ b/tests/test_decoder.py	2024-12-04 02:11:21.889038455 +0100
@@ -492,12 +492,7 @@ def test_datetime_date_out_of_range(impl
     with pytest.raises(impl.CBORDecodeError) as excinfo:
         impl.loads(unhexlify("a6c11b00002401001b000000000000ff00"))
 
-    if sys.maxsize == 2147483647:
-        cause_exc_class = OverflowError
-    elif platform.system() == "Windows":
-        cause_exc_class = OSError
-    else:
-        cause_exc_class = ValueError
+    cause_exc_class = ValueError
 
     assert isinstance(excinfo.value.__cause__, cause_exc_class)
 
