diff --git a/Makefile.in b/Makefile.in
index 2804807..85636c5 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -193,7 +193,7 @@ gource_LDADD = $(LDADD)
 gource_LINK = $(CXXLD) $(gource_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
	$(LDFLAGS) -o $@
 am__gource_tests_SOURCES_DIST = src/test/main.cpp \
-	src/test/datetime_tests.cpp src/test/regex_tests.cpp \
+	src/test/regex_tests.cpp \
	src/action.cpp src/bloom.cpp src/caption.cpp \
	src/core/conffile.cpp src/core/display.cpp \
	src/core/frustum.cpp src/core/fxfont.cpp src/core/logger.cpp \
@@ -729,7 +729,6 @@ gource_tests_LDFLAGS = ${BOOST_LDFLAGS}
 gource_tests_LDADD = ${BOOST_UNIT_TEST_FRAMEWORK_LIB}
 gource_tests_SOURCES = \
	src/test/main.cpp \
-	src/test/datetime_tests.cpp \
	src/test/regex_tests.cpp \
	${sources}

@@ -3179,7 +3178,7 @@ distcleancheck: distclean
	       exit 1; } >&2
 check-am: all-am
	$(MAKE) $(AM_MAKEFLAGS) $(check_PROGRAMS)
-	$(MAKE) $(AM_MAKEFLAGS) check-TESTS
+	$(MAKE) $(AM_MAKEFLAGS)
 check: check-am
 all-am: Makefile $(PROGRAMS) $(DATA)
 installdirs:
