--- newt-0.52.7/Makefile.in.orig	2008-01-15 16:57:12.000000000 +0100
+++ newt-0.52.7/Makefile.in	2008-01-15 16:58:38.000000000 +0100
@@ -15,7 +15,7 @@
 PYTHONVERS = @PYTHONVERS@
 WHIPTCLSO = @WHIPTCLSO@
 
-PROGS = test whiptail $(WHIPTCLSO) testgrid testtree showchars showkey
+PROGS = test whiptail testgrid testtree showchars showkey
 TESTOBJS = test.o testgrid.o testtree.o showchars.o showkey.o
 NDIALOGOBJS = whiptail.o dialogboxes.o
 WHIPTCLOBJS = shared/whiptcl.o shared/dialogboxes.o
@@ -78,9 +78,6 @@
 whiptail: $(NDIALOGOBJS) $(LIBNEWTSH)
 	$(CC) -g -o whiptail $(NDIALOGOBJS) -L . -lnewt $(LIBS) -lpopt
 
-whiptcl.so: $(WHIPTCLOBJS) $(LIBNEWTSH)
-	$(CC) -shared $(SHCFLAGS) -o whiptcl.so $(WHIPTCLOBJS) -L . -lnewt  $(LIBTCL) -lslang -lpopt -lm
-
 $(LIBNEWT): $(LIBOBJS)
 	ar rv $@ $^
 
@@ -120,12 +117,11 @@
 	install -m 644 whiptail.1 $(instroot)/$(man1dir)
 	make -C po datadir=$(instroot)/$(datadir) install
 
-install-sh: sharedlib $(WHIPTCLSO) _snackmodule.so
+install-sh: sharedlib _snackmodule.so
 	[ -d $(instroot)/$(libdir) ] || install -m 755 -d $(instroot)/$(libdir)
 	install -m 755 $(LIBNEWTSH) $(instroot)/$(libdir)
 	ln -sf $(LIBNEWTSONAME) $(instroot)/$(libdir)/libnewt.so
 	ln -sf $(LIBNEWTSH) $(instroot)/$(libdir)/$(LIBNEWTSONAME)
-	[ -n "$(WHIPTCLSO)" ] && install -m 755 whiptcl.so $(instroot)/$(libdir) || :
 	for ver in $(PYTHONVERS) ; do \
 	   [ -d $(instroot)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 755 $$ver/_snackmodule.so $(instroot)/$(libdir)/$$ver/site-packages ;\
