diff --git a/tests/Makefile.am b/tests/Makefile.am
index fa491ec..f170147 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -35,8 +35,7 @@ EXTRA_DIST = $(TESTS_THAT_PASS) $(DTDDECL_TEST)
 TESTS_ENVIRONMENT = top_srcdir=$(top_srcdir) \
 	PATH=.:../nsgmls:../sgmlnorm:../sx:../spent:../spcat:../spam:$$PATH \
 	ONSGMLS=`echo onsgmls|sed '$(transform)'`\
-	OSGMLNORM=`echo osgmlnorm|sed '$(transform)'`\
-	$(SHELL)
+	OSGMLNORM=`echo osgmlnorm|sed '$(transform)'`
 
 # Two auxiliary programs used by the tests.
 INCLUDES = -I.. -I$(top_srcdir)/lib -I$(top_srcdir)/intl
