--- a/include/InternalInputSource.h
+++ b/include/InternalInputSource.h
@@ -34,7 +34,7 @@
   void pushCharRef(Char ch, const NamedCharRef &);
   Boolean rewind(Messenger &);
   const StringC *contents();
-  InternalInputSource *InternalInputSource::asInternalInputSource();
+  InternalInputSource *asInternalInputSource();
   ~InternalInputSource();
 private:
   InternalInputSource(const InternalInputSource &); // undefined
