diff --git a/tests/Makefile.am b/tests/Makefile.am
index 2ea8cb1..fa491ec 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -16,20 +16,20 @@ af-18 af-19

 # These tests currently fail, but should not.  The problems need to be tracked
 # down and fixed, then the tests moved to TESTS_THAT_PASS.
-TESTS_THAT_FAIL = catalog-2 \
-af-7 af-8 af-9 af-10 af-11
+#TESTS_THAT_FAIL = catalog-2 \
+#af-7 af-8 af-9 af-10 af-11

-SHOWSTOPPERS = $(TESTS_THAT_FAIL)
+#SHOWSTOPPERS = $(TESTS_THAT_FAIL)

 DTDDECL_TEST = catalog-1

-TESTS = $(TESTS_THAT_PASS) $(SHOWSTOPPERS)
+TESTS = $(TESTS_THAT_PASS)

 if DTDDECL
 TESTS += $(DTDDECL_TEST)
 endif

-EXTRA_DIST = $(TESTS_THAT_PASS) $(TESTS_THAT_FAIL) $(DTDDECL_TEST)
+EXTRA_DIST = $(TESTS_THAT_PASS) $(DTDDECL_TEST)


 TESTS_ENVIRONMENT = top_srcdir=$(top_srcdir) \
diff --git a/tests/Makefile.in b/tests/Makefile.in
index 19b6648..500caec 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -220,13 +220,12 @@ af-18 af-19

 # These tests currently fail, but should not.  The problems need to be tracked
 # down and fixed, then the tests moved to TESTS_THAT_PASS.
-TESTS_THAT_FAIL = catalog-2 \
-af-7 af-8 af-9 af-10 af-11
+# TESTS_THAT_FAIL = catalog-2 \
+# af-7 af-8 af-9 af-10 af-11

-SHOWSTOPPERS = $(TESTS_THAT_FAIL)
 DTDDECL_TEST = catalog-1
-TESTS = $(TESTS_THAT_PASS) $(SHOWSTOPPERS) $(am__append_1)
-EXTRA_DIST = $(TESTS_THAT_PASS) $(TESTS_THAT_FAIL) $(DTDDECL_TEST)
+TESTS = $(TESTS_THAT_PASS) $(am__append_1)
+EXTRA_DIST = $(TESTS_THAT_PASS) $(DTDDECL_TEST)
 TESTS_ENVIRONMENT = top_srcdir=$(top_srcdir) \
	PATH=.:../nsgmls:../sgmlnorm:../sx:../spent:../spcat:../spam:$$PATH \
	ONSGMLS=`echo onsgmls|sed '$(transform)'`\
