diff -urw emacs-7eb2077f9dd14ae9d70a12c621151afd50ea6e96/lib-src/Makefile.in emacs-7eb2077f9dd14ae9d70a12c621151afd50ea6e96/lib-src/Makefile.in
--- orig/lib-src/Makefile.in	2015-02-06 20:37:24.297691017 +0000
+++ src/lib-src/Makefile.in	2015-02-06 20:38:05.006592590 +0000
@@ -414,7 +414,6 @@
 
 update-game-score${EXEEXT}: ${srcdir}/update-game-score.c $(NTLIB) $(config_h)
 	$(AM_V_CCLD)$(CC) ${ALL_CFLAGS} \
-	  -DHAVE_SHARED_GAME_DIR="\"$(gamedir)\"" \
 	  $< $(NTLIB) $(LOADLIBES) -o $@
 
 emacsclient.res: ../nt/emacsclient.rc $(NTINC)/../icons/emacs.ico
