From 7a3bb97369d9af74857937616239107c61d559a6 Mon Sep 17 00:00:00 2001
From: Hugo Osvaldo Barrera <hugo@whynothugo.nl>
Date: Sat, 30 Aug 2025 14:06:15 +0200
Subject: [PATCH] Fix tests for ppc64le

---
 test_prctl.py | 15 +++++++++++++--
 1 file changed, 13 insertions(+), 2 deletions(-)

diff --git a/test_prctl.py b/test_prctl.py
index 5dee220..4852703 100644
--- a/test_prctl.py
+++ b/test_prctl.py
@@ -122,6 +122,11 @@ class PrctlTest(unittest.TestCase):
             prctl.set_endian(prctl.ENDIAN_LITTLE)
             self.assertEqual(prctl.get_endian(), prctl.ENDIAN_LITTLE)
             self.assertRaises(ValueError, prctl.set_endian, 999)
+        if self.arch == 'ppc64le':
+            # Switching to BIG_ENDIAN crashes due to memory corruption.
+            prctl.set_endian(prctl.ENDIAN_LITTLE)
+            self.assertEqual(prctl.get_endian(), prctl.ENDIAN_LITTLE)
+            self.assertRaises(ValueError, prctl.set_endian, 999)
         else:
             self.assertRaises(OSError, prctl.get_endian)
             self.assertRaises(OSError, prctl.set_endian)
@@ -146,6 +151,12 @@ class PrctlTest(unittest.TestCase):
             prctl.set_fpexc(prctl.FP_EXC_SW_ENABLE)
             self.assertEqual(prctl.get_fpexc() & prctl.PR_FP_EXC_SW_ENABLE, prctl.PR_FP_EXC_SW_ENABLE)
             self.assertRaises(ValueError, prctl.set_fpexc, 999)
+        elif self.arch == 'ppc64le':
+            self.assertEqual(prctl.get_fpexc(), 0)
+            prctl.set_fpexc() # TODO: raises with any argument?
+            # TODO: should this really raise?
+            self.assertRaises(OSError, prctl.set_fpexc, prctl.FP_EXC_SW_ENABLE)
+            self.assertRaises(ValueError, prctl.set_fpexc, 999)
         else:
             self.assertRaises(OSError, prctl.get_fpexc)
             self.assertRaises(OSError, prctl.set_fpexc)
@@ -386,8 +397,8 @@ class PrctlTest(unittest.TestCase):
 
     def test_unalign(self):
         """Test manipulation of the unaligned access setting"""
-        if self.arch in ('ia64', 'parisc', 'powerpc', 'alpha'):
-            # FIXME untested
+        if self.arch in ('ia64', 'parisc', 'powerpc', 'alpha', 'ppc64le'):
+            # FIXME untested (tested only on ppc64le)
             prctl.set_unalign(prctl.UNALIGN_NOPRINT)
             self.assertEqual(prctl.get_unalign(), prctl.UNALIGN_NOPRINT)
             prctl.set_unalign(prctl.UNALIGN_SIGBUS)
-- 
2.51.0

