--- a/setup.py
+++ b/setup.py
@@ -102,7 +102,7 @@
     maintainer_email='erik.n.welch@gmail.com',
     license = 'BSD',
     packages=['cytoolz', 'cytoolz.curried'],
-    package_data={'cytoolz': ['*.pyx', '*.pxd', 'curried/*.pyx', 'tests/*.py']},
+    package_data={'cytoolz': ['*.pyx', '*.pxd', 'curried/*.pyx']},
     # include_package_data = True,
     keywords=('functional utility itertools functools iterator generator '
                 'curry memoize lazy streaming bigdata cython toolz cytoolz'),
