diff --git a/src/sanlock_sock.c b/src/sanlock_sock.c
index b83f9ae..305c16f 100644
--- a/src/sanlock_sock.c
+++ b/src/sanlock_sock.c
@@ -15,6 +15,8 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 
+#include <string.h>
+
 #include "sanlock.h"
 #include "sanlock_sock.h"
 
diff --git a/src/sanlock_sock.h b/src/sanlock_sock.h
index 0121b9c..144e8d4 100644
--- a/src/sanlock_sock.h
+++ b/src/sanlock_sock.h
@@ -10,6 +10,8 @@
 #ifndef __SANLOCK_SOCK_H__
 #define __SANLOCK_SOCK_H__
 
+#include <stdint.h>
+
 #define SANLK_SOCKET_NAME "sanlock.sock"
 
 #define SM_MAGIC 0x04282010
diff --git a/wdmd/wdmd.h b/wdmd/wdmd.h
index b32598f..fd8427b 100644
--- a/wdmd/wdmd.h
+++ b/wdmd/wdmd.h
@@ -12,6 +12,9 @@
 
 #define WDMD_NAME_SIZE 128
 
+#include <stdint.h>
+#include <string.h>
+
 int wdmd_connect(void);
 int wdmd_register(int con, char *name);
 int wdmd_refcount_set(int con);
