--- a/index.php
+++ b/index.php
@@ -24,7 +24,7 @@
 #==============================================================================
 # Includes
 #==============================================================================
-require_once("conf/config.inc.php");
+require_once("/etc/ltb-project/ssp/conf/config.inc.php");
 require_once("lib/vendor/defuse-crypto.phar");
 require_once("lib/functions.inc.php");
 if ($use_recaptcha) {
