--- vanessa_logger-0.0.8.orig/libvanessa_logger/vanessa_logger.c
+++ vanessa_logger-0.0.8/libvanessa_logger/vanessa_logger.c
@@ -677,8 +677,6 @@
 {
 	int i;
 
-	extern CODE facilitynames[];
-
 	if (facility_name == NULL) {
 		fprintf(stderr,
 			"__vanessa_logger_get_facility_byname: "
@@ -1102,8 +1100,7 @@
 		case HOST_NOT_FOUND:
 			str = "Unknown host";
 			break;
-		case NO_ADDRESS:
-		/* case NO_DATA: */
+		case NO_DATA:
 			str = "Host has no address";
 			break;
 		case NO_RECOVERY:
