diff --git a/avida-core/source/cpu/cTestCPUInterface.cc b/avida-core/source/cpu/cTestCPUInterface.cc
index 7e78a59..a827712 100644
--- a/avida-core/source/cpu/cTestCPUInterface.cc
+++ b/avida-core/source/cpu/cTestCPUInterface.cc
@@ -152,7 +152,7 @@ int cTestCPUInterface::GetStateGridID(cAvidaContext& ctx)
 Apto::Array<int> cTestCPUInterface::GetFormedGroupArray()
 {
   Apto::Array<int> null_array;
-  null_array.SetAll(NULL);
+  null_array.SetAll(0);
   return null_array;
 }
 
