diff --git a/config.default.php b/config.default.php
index 87297554..51e42ad7 100644
--- a/config.default.php
+++ b/config.default.php
@@ -194,7 +194,7 @@ return array(
 	'extensions' => [],
 
 	# Disable self-update,
-	'disable_update' => false,
+	'disable_update' => true,
 
 	# Trusted IPs (e.g. of last proxy) that are allowed to send unsafe HTTP headers.
 	# The connection IP used during FreshRSS setup is automatically added to this list.
diff --git a/constants.php b/constants.php
index 824a7a5f..a09fe6c2 100644
--- a/constants.php
+++ b/constants.php
@@ -45,7 +45,7 @@ $dataPath = getenv('DATA_PATH');
 if (is_string($dataPath) && $dataPath !== '') {
 	define('DATA_PATH', $dataPath);
 } else {
-	defined('DATA_PATH') or define('DATA_PATH', FRESHRSS_PATH . '/data');
+	defined('DATA_PATH') or define('DATA_PATH', '/var/lib/freshrss/data');
 }
 
 defined('UPDATE_FILENAME') or define('UPDATE_FILENAME', DATA_PATH . '/update.php');
