diff --git a/config.default.php b/config.default.php
index d0873d5..59758d0 100644
--- a/config.default.php
+++ b/config.default.php
@@ -194,7 +194,7 @@ return array(
 	],
 
 	# Disable self-update,
-	'disable_update' => false,
+	'disable_update' => true,
 
 	# Trusted IPs that are allowed to send unsafe headers
 	# Please read the documentation, before configuring this
diff --git a/constants.php b/constants.php
index 396e397..d006e90 100644
--- a/constants.php
+++ b/constants.php
@@ -39,7 +39,7 @@ defined('COPY_SYSLOG_TO_STDERR') or define('COPY_SYSLOG_TO_STDERR', filter_var(g
 defined('MAX_LOG_SIZE') or define('MAX_LOG_SIZE', 1048576);
 
 //This directory must be writable
-defined('DATA_PATH') or define('DATA_PATH', FRESHRSS_PATH . '/data');
+defined('DATA_PATH') or define('DATA_PATH', '/var/lib/freshrss/data');
 
 defined('UPDATE_FILENAME') or define('UPDATE_FILENAME', DATA_PATH . '/update.php');
 defined('USERS_PATH') or define('USERS_PATH', DATA_PATH . '/users');
