--- open-vm-tools-9.4.6-1770165.orig/configure.ac
+++ open-vm-tools-9.4.6-1770165/configure.ac
@@ -772,6 +772,7 @@
 ###
 
 AC_CHECK_HEADERS([crypt.h])
+AC_CHECK_HEADERS([fts.h])
 AC_CHECK_HEADERS([inttypes.h])
 AC_CHECK_HEADERS([stdint.h])
 AC_CHECK_HEADERS([stdlib.h])
--- open-vm-tools-9.4.6-1770165.orig/lib/file/filePosix.c
+++ open-vm-tools-9.4.6-1770165/lib/file/filePosix.c
@@ -99,7 +99,7 @@
  * happens the prosix version should be updated to use the generic code.
  */
 
-#if defined(__USE_FILE_OFFSET64) || defined(sun) || defined(__ANDROID__)
+#if defined(__USE_FILE_OFFSET64) || defined(sun) || defined(__ANDROID__) || !defined(HAVE_FTS_H)
 # define CAN_USE_FTS 0
 #else
 # define CAN_USE_FTS 1
