--- open-vm-tools-9.4.6-1770165.orig/lib/include/vm_basic_defs.h
+++ open-vm-tools-9.4.6-1770165/lib/include/vm_basic_defs.h
@@ -550,7 +550,7 @@
 #endif
 #endif
 
-#if defined __linux__ && !defined __KERNEL__ && !defined MODULE && \
+#if defined __GLIBC__ && !defined __KERNEL__ && !defined MODULE && \
                          !defined VMM && !defined FROBOS && !defined __ANDROID__
 #include <features.h>
 #if __GLIBC_PREREQ(2, 1) && !defined GLIBC_VERSION_21
