--- open-vm-tools-9.4.6-1770165.orig/configure.ac
+++ open-vm-tools-9.4.6-1770165/configure.ac
@@ -575,6 +575,7 @@
 
 AC_CHECK_FUNCS([ecvt])
 AC_CHECK_FUNCS([fcvt])
+AC_CHECK_FUNCS([setresuid setresgid])
 
 if test "$os" = "freebsd" -a "$osVersion" -ge 600000; then
    AC_CHECK_LIB(
--- open-vm-tools-9.4.6-1770165.orig/lib/procMgr/procMgrPosix.c
+++ open-vm-tools-9.4.6-1770165/lib/procMgr/procMgrPosix.c
@@ -137,7 +137,7 @@
 #define  BASH_PATH "/bin/bash"
 #endif
 
-#if defined(linux) && !defined(GLIBC_VERSION_23) && !defined(__UCLIBC__)
+#if defined(linux)
 /*
  * Implements the system calls (they are not wrapped by glibc til 2.3.2).
  *
@@ -146,7 +146,7 @@
  *
  * (In fact, newer Linux kernels don't even define _syscall macros anymore.)
  */
-
+#ifndef HAVE_SETRESUID
 static INLINE int
 setresuid(uid_t ruid,
           uid_t euid,
@@ -154,8 +154,9 @@
 {
    return syscall(__NR_setresuid, ruid, euid, suid);
 }
+#endif
 
-
+#ifndef HAVE_SETRESGID
 static INLINE int
 setresgid(gid_t ruid,
           gid_t euid,
@@ -163,6 +164,8 @@
 {
    return syscall(__NR_setresgid, ruid, euid, suid);
 }
+#endif
+
 #endif
 
 
