--- open-vm-tools-9.4.6-1770165.orig/configure.ac
+++ open-vm-tools-9.4.6-1770165/configure.ac
@@ -367,6 +367,20 @@
                   AC_MSG_WARN([Fuse is missing, vmblock-fuse will be disabled.])])
 
 #
+# Check for libtirpc
+#
+AC_VMW_CHECK_LIB([libtirpc],
+                 [LIBTIRPC],
+                 [libtirpc],
+                 [],
+                 [],
+                 [rpc/xdr.h],
+                 [xdr_void],
+                 [have_libtirpc=yes],
+                 [have_libtitirpc=no;
+                  AC_MSG_WARN([libtirpc is missing.])])
+
+#
 # Check for PAM.
 #
 AC_ARG_WITH([pam],
--- open-vm-tools-9.4.6-1770165.orig/lib/dynxdr/Makefile.am
+++ open-vm-tools-9.4.6-1770165/lib/dynxdr/Makefile.am
@@ -21,3 +21,9 @@
 libDynxdr_la_SOURCES += dynxdr.c
 libDynxdr_la_SOURCES += xdrutil.c
 
+libDynxdr_la_CPPFLAGS = 
+libDynxdr_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
+
+libDynxdr_la_LIBADD =
+libDynxdr_la_LIBADD += @LIBTIRPC_LIBS@
+
--- open-vm-tools-9.4.6-1770165.orig/lib/guestRpc/Makefile.am
+++ open-vm-tools-9.4.6-1770165/lib/guestRpc/Makefile.am
@@ -20,6 +20,12 @@
 libGuestRpc_la_SOURCES =
 libGuestRpc_la_SOURCES += nicinfo_xdr.c
 
+libGuestRpc_la_CPPFLAGS =
+libGuestRpc_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
+
+libGuestRpc_la_LIBADD =
+libGuestRpc_la_LIBADD += @LIBTIRPC_LIBS@
+
 # XXX: Autoreconf complains about this and recommends using AM_CFLAGS instead.
 # Problem is, $(CFLAGS) is appended to the compiler command line after AM_CFLAGS
 # and after libGuestRpc_la_CFLAGS, so "-Wall -Werror" will override this flag.
--- open-vm-tools-9.4.6-1770165.orig/lib/netUtil/Makefile.am
+++ open-vm-tools-9.4.6-1770165/lib/netUtil/Makefile.am
@@ -20,3 +20,7 @@
 libNetUtil_la_SOURCES =
 libNetUtil_la_SOURCES += netUtilLinux.c
 
+libNetUtil_la_CPPFLAGS =
+libNetUtil_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
+
+libNetUtil_la_LIBADD = @LIBTIRPC_LIBS@
--- open-vm-tools-9.4.6-1770165.orig/lib/rpcChannel/Makefile.am
+++ open-vm-tools-9.4.6-1770165/lib/rpcChannel/Makefile.am
@@ -23,3 +23,7 @@
 
 libRpcChannel_la_CPPFLAGS =
 libRpcChannel_la_CPPFLAGS += @VMTOOLS_CPPFLAGS@
+libRpcChannel_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
+
+libRpcChannel_la_LIBADD =
+libRpcChannel_la_LIBADD += @LIBTIRPC_LIBS@
--- open-vm-tools-9.4.6-1770165.orig/libguestlib/Makefile.am
+++ open-vm-tools-9.4.6-1770165/libguestlib/Makefile.am
@@ -22,11 +22,15 @@
 
 libguestlib_la_LIBADD =
 libguestlib_la_LIBADD += @VMTOOLS_LIBS@
+libguestlib_la_LIBADD += @LIBTIRPC_LIBS@
 
 libguestlib_la_SOURCES =
 libguestlib_la_SOURCES += guestlibV3_xdr.c
 libguestlib_la_SOURCES += guestlibIoctl_xdr.c
 libguestlib_la_SOURCES += vmGuestLib.c
+
+libguestlib_la_CPPFLAGS =
+libguestlib_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 
 libguestlib_la_LDFLAGS =
 # We require GCC, so we're fine passing compiler-specific flags.
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/dndcp/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/dndcp/Makefile.am
@@ -24,6 +24,7 @@
 libdndcp_la_CPPFLAGS += @GTK_CPPFLAGS@
 libdndcp_la_CPPFLAGS += @GTKMM_CPPFLAGS@
 libdndcp_la_CPPFLAGS += @PLUGIN_CPPFLAGS@
+libdndcp_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 libdndcp_la_CPPFLAGS += -I$(top_srcdir)/services/plugins/dndcp/dnd
 libdndcp_la_CPPFLAGS += -I$(top_srcdir)/services/plugins/dndcp/dndGuest
 libdndcp_la_CPPFLAGS += -I$(top_srcdir)/services/plugins/dndcp/stringxx
@@ -38,6 +39,7 @@
 libdndcp_la_LIBADD += @GTKMM_LIBS@
 libdndcp_la_LIBADD += @VMTOOLS_LIBS@
 libdndcp_la_LIBADD += @HGFS_LIBS@
+libdndcp_la_LIBADD += @LIBTIRPC_LIBS@
 
 libdndcp_la_SOURCES =
 
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/guestInfo/getlib/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/guestInfo/getlib/Makefile.am
@@ -25,7 +25,11 @@
 
 libGuestInfo_la_CPPFLAGS =
 libGuestInfo_la_CPPFLAGS += @GLIB2_CPPFLAGS@
+libGuestInfo_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 libGuestInfo_la_CPPFLAGS += -I$(srcdir)/..
+
+libGuestInfo_la_LIBADD =
+libGuestInfo_la_LIBADD += @LIBTIRPC_LIBS@
 
 AM_CFLAGS = $(DNET_CPPFLAGS)
 
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/guestInfo/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/guestInfo/Makefile.am
@@ -23,6 +23,7 @@
 
 libguestInfo_la_CPPFLAGS =
 libguestInfo_la_CPPFLAGS += @PLUGIN_CPPFLAGS@
+libguestInfo_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 
 libguestInfo_la_LDFLAGS =
 libguestInfo_la_LDFLAGS += @PLUGIN_LDFLAGS@
@@ -31,6 +32,7 @@
 libguestInfo_la_LIBADD += @VMTOOLS_LIBS@
 libguestInfo_la_LIBADD += @PROCPS_LIBS@
 libguestInfo_la_LIBADD += @XDR_LIBS@
+libguestInfo_la_LIBADD += @LIBTIRPC_LIBS@
 libguestInfo_la_LIBADD += getlib/libGuestInfo.la
 
 if HAVE_DNET
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/resolutionSet/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/resolutionSet/Makefile.am
@@ -21,6 +21,7 @@
 libresolutionSet_la_CPPFLAGS =
 libresolutionSet_la_CPPFLAGS += @GTK_CPPFLAGS@
 libresolutionSet_la_CPPFLAGS += @PLUGIN_CPPFLAGS@
+libresolutionSet_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 libresolutionSet_la_CPPFLAGS += -DRESOLUTION_X11
 
 libresolutionSet_la_LDFLAGS =
@@ -30,6 +31,7 @@
 libresolutionSet_la_LIBADD += @COMMON_XLIBS@
 libresolutionSet_la_LIBADD += @GTK_LIBS@
 libresolutionSet_la_LIBADD += @VMTOOLS_LIBS@
+libresolutionSet_la_LIBADD += @LIBTIRPC_LIBS@
 
 libresolutionSet_la_SOURCES =
 libresolutionSet_la_SOURCES += libvmwarectrl.c
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/vix/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/vix/Makefile.am
@@ -20,6 +20,7 @@
 
 libvix_la_CPPFLAGS =
 libvix_la_CPPFLAGS += @PLUGIN_CPPFLAGS@
+libvix_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 
 libvix_la_LDFLAGS =
 libvix_la_LDFLAGS += @PLUGIN_LDFLAGS@
@@ -28,6 +29,7 @@
 libvix_la_LIBADD += @VIX_LIBADD@
 libvix_la_LIBADD += @VMTOOLS_LIBS@
 libvix_la_LIBADD += @HGFS_LIBS@
+libvix_la_LIBADD += @LIBTIRPC_LIBS@
 libvix_la_LIBADD += $(top_builddir)/lib/auth/libAuth.la
 libvix_la_LIBADD += $(top_builddir)/lib/foundryMsg/libFoundryMsg.la
 libvix_la_LIBADD += $(top_builddir)/lib/impersonate/libImpersonate.la
--- open-vm-tools-9.4.6-1770165.orig/services/plugins/vmbackup/Makefile.am
+++ open-vm-tools-9.4.6-1770165/services/plugins/vmbackup/Makefile.am
@@ -20,6 +20,7 @@
 
 libvmbackup_la_CPPFLAGS =
 libvmbackup_la_CPPFLAGS += @PLUGIN_CPPFLAGS@
+libvmbackup_la_CPPFLAGS += @LIBTIRPC_CPPFLAGS@
 
 libvmbackup_la_LDFLAGS =
 libvmbackup_la_LDFLAGS += @PLUGIN_LDFLAGS@
@@ -27,6 +28,7 @@
 libvmbackup_la_LIBADD =
 libvmbackup_la_LIBADD += @GOBJECT_LIBS@
 libvmbackup_la_LIBADD += @VMTOOLS_LIBS@
+libvmbackup_la_LIBADD += @LIBTIRPC_LIBS@
 
 libvmbackup_la_SOURCES =
 libvmbackup_la_SOURCES += nullProvider.c
