Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: output format changed with gdal 3.13
----


--- a/tests/test_collection.py
+++ b/tests/test_collection.py
@@ -1018,7 +1018,7 @@
             dst.writerecords(ftr for ftr in src)
 
     with open(dstfile) as f:
-        assert '"coordinates": [ [ [ -111.74, 42.0 ], [ -111.66, 42.0 ]' in f.read(2000)
+        assert '"coordinates":[[[-111.74,42.0],[-111.66,42.0]' in f.read(2000)
 
 
 @pytest.mark.network


--- a/tests/test_subtypes.py
+++ b/tests/test_subtypes.py
@@ -50,8 +50,8 @@
     with open(os.fspath(path)) as f:
         data = f.read()
 
-    assert """"bool": true""" in data
-    assert """"not_bool": 1""" in data
+    assert """"bool":true""" in data
+    assert """"not_bool":1""" in data
 
 
 def test_write_int16_subtype(tmp_path):
