--- a/nDPI/src/include/ipq_api.h.orig
+++ b/nDPI/src/include/ipq_api.h
@@ -24,6 +24,7 @@
 #ifndef __IPOQUE_API_INCLUDE_FILE__
 #define __IPOQUE_API_INCLUDE_FILE__
 
+#include <sys/types.h>
 
 #if defined(HAVE_NTOP) && defined(WIN32)
 #include <winsock2.h>
--- a/nDPI/src/lib/ipq_main.c.orig
+++ b/nDPI/src/lib/ipq_main.c
@@ -21,6 +21,7 @@
  */
 
 
+#include <sys/types.h>
 #include "ipq_main.h"
 #include "ipq_protocols.h"
 #include "ipq_utils.h"
