--- a/c++/src/ColumnReader.cc
+++ b/c++/src/ColumnReader.cc
@@ -464,7 +464,7 @@
       uint64_t bufferNum = 0;
       if constexpr (isLittleEndian && std::is_same_v<ValueType, ItemType>) {
         bufferNum =
-            std::min(numValues, static_cast<size_t>(bufferEnd_ - bufferPointer_) / bytesPerValue_);
+            std::min(numValues, static_cast<uint64_t>(bufferEnd_ - bufferPointer_) / static_cast<uint64_t>(bytesPerValue_));
         uint64_t bufferBytes = bufferNum * bytesPerValue_;
         if (bufferBytes > 0) {
           memcpy(outArray, bufferPointer_, bufferBytes);
