Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: link against openblas instead of blas
----

--- a/configure
+++ b/configure
@@ -12790,11 +12790,11 @@
 
 
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for dnrm2_ in -lblas" >&5
-$as_echo_n "checking for dnrm2_ in -lblas... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for dnrm2_ in -lopenblas" >&5
+$as_echo_n "checking for dnrm2_ in -lopenblas... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-lblas $MATHLIB  $LIBS"
+LIBS="-lopenblas $MATHLIB  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -12824,14 +12824,14 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_blas_dnrm2_" >&5
 $as_echo "$ac_cv_lib_blas_dnrm2_" >&6; }
 if test "x$ac_cv_lib_blas_dnrm2_" = xyes; then :
-  BLASLIB="$BLASLIB -lblas "
+  BLASLIB="$BLASLIB -lopenblas "
 else
 
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for dnrm2_ in -lblas" >&5
-$as_echo_n "checking for dnrm2_ in -lblas... " >&6; }
+{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for dnrm2_ in -lopenblas" >&5
+$as_echo_n "checking for dnrm2_ in -lopenblas... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-lblas $MATHLIB -lg2c $LIBS"
+LIBS="-lopenblas $MATHLIB -lg2c $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -12861,7 +12861,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_blas_dnrm2_" >&5
 $as_echo "$ac_cv_lib_blas_dnrm2_" >&6; }
 if test "x$ac_cv_lib_blas_dnrm2_" = xyes; then :
-  BLASLIB="$BLASLIB -lblas -lg2c"
+  BLASLIB="$BLASLIB -lopenblas -lg2c"
 else
 
 LDFLAGS=${ac_save_ldflags}
@@ -13043,11 +13043,11 @@
 
 # BLAS in PhiPACK libraries? (requires generic BLAS, too)
 if test $blas_ok = no; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sgemm_ in -lblas" >&5
-$as_echo_n "checking for sgemm_ in -lblas... " >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sgemm_ in -lopenblas" >&5
+$as_echo_n "checking for sgemm_ in -lopenblas... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-lblas  $LIBS"
+LIBS="-lopenblas  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13081,7 +13081,7 @@
 $as_echo_n "checking for dgemm_ in -ldgemm... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-ldgemm -lblas $LIBS"
+LIBS="-ldgemm -lopenblas $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13115,7 +13115,7 @@
 $as_echo_n "checking for sgemm_ in -lsgemm... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-lsgemm -lblas $LIBS"
+LIBS="-lsgemm -lopenblas $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13145,7 +13145,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_sgemm_sgemm_" >&5
 $as_echo "$ac_cv_lib_sgemm_sgemm_" >&6; }
 if test "x$ac_cv_lib_sgemm_sgemm_" = xyes; then :
-  blas_ok=yes; BLASLIB="-lsgemm -ldgemm -lblas"
+  blas_ok=yes; BLASLIB="-lsgemm -ldgemm -lopenblas"
 fi
 
 fi
@@ -13237,11 +13237,11 @@
 
 # Generic BLAS library
 if test $blas_ok = no; then
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sgemm_ in -lblas" >&5
-$as_echo_n "checking for sgemm_ in -lblas... " >&6; }
+	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for sgemm_ in -lopenblas" >&5
+$as_echo_n "checking for sgemm_ in -lopenblas... " >&6; }
 
 ac_check_lib_save_LIBS=$LIBS
-LIBS="-lblas  $LIBS"
+LIBS="-lopenblas  $LIBS"
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
@@ -13271,7 +13271,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_blas_sgemm_" >&5
 $as_echo "$ac_cv_lib_blas_sgemm_" >&6; }
 if test "x$ac_cv_lib_blas_sgemm_" = xyes; then :
-  blas_ok=yes; BLASLIB="-lblas"
+  blas_ok=yes; BLASLIB="-lopenblas"
 fi
 
 fi
