diff --git a/src/scrapers/image/ImageProvider.h b/src/scrapers/image/ImageProvider.h
index 575bd63..a6c80f3 100644
--- a/src/scrapers/image/ImageProvider.h
+++ b/src/scrapers/image/ImageProvider.h
@@ -134,5 +134,5 @@ signals:
 } // namespace mediaelch
 
 
-Q_DECLARE_METATYPE(mediaelch::scraper::ImageProvider*)
 Q_DECLARE_OPAQUE_POINTER(mediaelch::scraper::ImageProvider*)
+Q_DECLARE_METATYPE(mediaelch::scraper::ImageProvider*)
