--- a/src/power.cpp
+++ b/src/power.cpp
@@ -38,7 +38,7 @@
 #include "mission.h"
 #include "intdisplay.h"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #define EXTRACT_POINTS      1
 #define MAX_POWER           1000000
--- a/src/frontend.cpp
+++ b/src/frontend.cpp
@@ -85,7 +85,7 @@
 #include "clparse.h" // for autorating
 #include "hci/groups.h"
 
-#include <fmt/core.h>
+#include <fmt/format.h>
 
 #if defined(__EMSCRIPTEN__)
 # include "emscripten_helpers.h"
