--- a/3rdparty/CMakeLists.txt
+++ b/3rdparty/CMakeLists.txt
@@ -116,7 +116,7 @@
 		message(FATAL_ERROR "Failed to create directory for basis_universal_host_build")
 	  endif()
 	  execute_process(
-		  COMMAND ${CMAKE_COMMAND} -DCMAKE_BUILD_TYPE=Release "${CMAKE_CURRENT_SOURCE_DIR}/basis_universal_host_build"
+		  COMMAND ${CMAKE_COMMAND} -G Ninja -DCMAKE_BUILD_TYPE=Release "${CMAKE_CURRENT_SOURCE_DIR}/basis_universal_host_build"
 		  WORKING_DIRECTORY "${BASISU_BINARY_DIR}"
 		  RESULT_VARIABLE _basis_result
 	  )
