diff --git a/src/tests/CMakeLists.txt b/src/tests/CMakeLists.txt
index 8238855..86788c5 100644
--- a/src/tests/CMakeLists.txt
+++ b/src/tests/CMakeLists.txt
@@ -80,7 +80,7 @@ if(Kwalify_FOUND)
       -DQUERY_PREFIX=filter=
       -P ${CMAKE_CURRENT_SOURCE_DIR}/kwalify_runtime_query.cmake
   )
-  set_tests_properties("kwalify:runtime:filters" PROPERTIES TIMEOUT 300)
+  set_tests_properties("kwalify:runtime:filters" PROPERTIES TIMEOUT 600)
 
   add_test(
     NAME "kwalify:runtime:transitions"
