--- a/Makefile
+++ b/Makefile
@@ -1,15 +1,15 @@
 # Linux (Debian Lenny)
-#CFLAGS=-g -O2 -fpic -I/usr/include/lua5.1
-#LDFLAGS=-O -shared -fpic -lsasl2
-#LUAPATH=/usr/share/lua/5.1
-#CPATH=/usr/lib/lua/5.1
+CFLAGS=-g -O2 -fpic -I/usr/include/lua5.1
+LDFLAGS=-O -shared -fpic -lsasl2
+LUAPATH=/usr/share/lua/5.1
+CPATH=/usr/lib/lua/5.1
 
 # MacOS
-CFLAGS=-g -Wall -O2
-LDFLAGS=-bundle -undefined dynamic_lookup -lsasl2
-MACOSX_VERSION=10.5
-LUAPATH=/usr/local/share/lua/5.1
-CPATH=/usr/local/lib/lua/5.1
+#CFLAGS=-g -Wall -O2
+#LDFLAGS=-bundle -undefined dynamic_lookup -lsasl2
+#MACOSX_VERSION=10.5
+#LUAPATH=/usr/local/share/lua/5.1
+#CPATH=/usr/local/lib/lua/5.1
 
 #########################################################
 #
