diff -upr xterm-391.orig/main.c xterm-391/main.c
--- xterm-391.orig/main.c
+++ xterm-391/main.c
@@ -3125,7 +3125,7 @@
 	close(opened_tty);
 	opened_tty = -1;
     }
-#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME) && defined(HAVE_GRANTPT_PTY_ISATTY)
+#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME)
     if ((*pty = posix_openpt(O_RDWR)) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {
