--- a/main.c
+++ b/main.c
@@ -2809,7 +2809,7 @@
 	close(opened_tty);
 	opened_tty = -1;
     }
-#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME) && defined(HAVE_GRANTPT_PTY_ISATTY)
+#elif defined(HAVE_POSIX_OPENPT) && defined(HAVE_PTSNAME)
     if ((*pty = posix_openpt(O_RDWR)) >= 0) {
 	char *name = ptsname(*pty);
 	if (name != 0) {
@@ -3909,7 +3909,7 @@
 	    /*
 	     * now in child process
 	     */
-#if defined(_POSIX_SOURCE) || defined(SVR4) || defined(__convex__) || defined(__SCO__) || defined(__QNX__)
+#if defined(_POSIX_VERSION) || defined(SVR4) || defined(__convex__) || defined(__SCO__) || defined(__QNX__)
 	    int pgrp = setsid();	/* variable may not be used... */
 #else
 	    int pgrp = getpid();
