diff -up gnumeric-1.12.33/doc/Makefile.am.destdir gnumeric-1.12.33/doc/Makefile.am
--- gnumeric-1.12.33/doc/Makefile.am.destdir	2017-01-30 21:47:16.000000000 +0100
+++ gnumeric-1.12.33/doc/Makefile.am	2017-01-31 08:19:26.170304443 +0100
@@ -527,7 +527,7 @@ HELP_MEDIA= \
 
 install-exec-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		install -d $$helpdir; \
 		$(RM) $$helpdir/index.docbook; \
 		$(LN_S) gnumeric.xml $$helpdir/index.docbook; \
@@ -535,7 +535,7 @@ install-exec-hook:
 
 uninstall-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		$(RM) $$helpdir/index.docbook; \
 	done
 
diff -up gnumeric-1.12.33/doc/Makefile.in.destdir gnumeric-1.12.33/doc/Makefile.in
--- gnumeric-1.12.33/doc/Makefile.in.destdir	2017-01-30 21:48:07.000000000 +0100
+++ gnumeric-1.12.33/doc/Makefile.in	2017-01-31 08:19:42.562456025 +0100
@@ -1347,7 +1347,7 @@ uninstall-am: uninstall-manpageDATA unin
 
 install-exec-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		install -d $$helpdir; \
 		$(RM) $$helpdir/index.docbook; \
 		$(LN_S) gnumeric.xml $$helpdir/index.docbook; \
@@ -1355,7 +1355,7 @@ install-exec-hook:
 
 uninstall-hook:
 	for d in C $(HELP_LINGUAS); do \
-		helpdir="$(HELP_DIR)/$$d/$(HELP_ID)"; \
+		helpdir="$(DESTDIR)$(HELP_DIR)/$$d/$(HELP_ID)"; \
 		$(RM) $$helpdir/index.docbook; \
 	done
 
