--- a/Makefile	2008-10-14 02:09:13.000000000 +0200
+++ b/Makefile	2008-10-14 18:31:38.000000000 +0200
@@ -26,5 +26,5 @@
 # Where we get installed
-BIN = /usr/local/bin
-MAN = /usr/share/man/man8
-CONF= /usr/local/etc
+BIN = $(DESTDIR)/usr/bin
+MAN = $(DESTDIR)/usr/share/man/man8
+CONF= $(DESTDIR)/etc
 # Setup for compiling and cross-compiling for Windows
diff -uNr foremost-1.4.orig/config.c foremost-1.4/config.c
--- a/config.c	2006-08-31 17:50:47.000000000 +0100
+++ b/config.c	2006-08-31 17:51:09.000000000 +0100
@@ -288,7 +288,7 @@
 #ifdef __WIN32
 		set_config_file(s, "/Program Files/foremost/foremost.conf");
 #else
-		set_config_file(s, "/usr/local/etc/foremost.conf");
+		set_config_file(s, "/etc/foremost.conf");
 #endif
 		if ((f = fopen(get_config_file(s), "r")) == NULL)
 			{
