--- a/meson.build
+++ b/meson.build
@@ -3551,7 +3551,7 @@
   libvduse = libvduse_proj.get_variable('libvduse_dep')
 endif
 
-tomlplusplus = dependency('tomlplusplus', fallback: ['tomlplusplus', 'tomlplusplus_dep'], default_options: ['default_library=static'])
+toml = dependency('tomlplusplus')
 
 xxhash = dependency('libxxhash', fallback: ['xxhash', 'xxhash_dep'], default_options: ['default_library=static'])
 
@@ -3926,7 +3926,7 @@
 
 specific_ss.add(files('xemu-xbe.c', 'xemu-version.c'))
 
-common_ss.add(tomlplusplus)
+common_ss.add(toml)
 common_ss.add(genconfig)
 
 common_ss.add(files('cpu-common.c'))
