diff --git a/extensions/raw_files/gth-metadata-provider-raw.c b/extensions/raw_files/gth-metadata-provider-raw.c
index 2d9f502..b03f489 100644
--- a/extensions/raw_files/gth-metadata-provider-raw.c
+++ b/extensions/raw_files/gth-metadata-provider-raw.c
@@ -62,7 +62,7 @@ gth_metadata_provider_raw_read (GthMetadataProvider *self,
 	if (!_g_mime_type_is_raw (gth_file_data_get_mime_type (file_data)))
 		return;
 
-	raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+	raw_data = libraw_init (LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
 	if (raw_data == NULL)
 		goto fatal_error;
 
diff --git a/extensions/raw_files/main.c b/extensions/raw_files/main.c
index 25829c1..a6ebdc1 100644
--- a/extensions/raw_files/main.c
+++ b/extensions/raw_files/main.c
@@ -213,7 +213,7 @@ _cairo_image_surface_create_from_raw (GInputStream  *istream,
 	size_t         size;
 	GthImage      *image = NULL;
 
-	raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+	raw_data = libraw_init (LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
 	if (raw_data == NULL) {
 		_libraw_set_gerror (error, errno);
 		goto fatal_error;
@@ -300,7 +300,7 @@ _cairo_image_surface_create_from_raw (GInputStream  *istream,
 		if ((original_width != NULL) && (original_height != NULL)) {
 			libraw_close (raw_data);
 
-			raw_data = libraw_init (LIBRAW_OPIONS_NO_MEMERR_CALLBACK | LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
+			raw_data = libraw_init (LIBRAW_OPIONS_NO_DATAERR_CALLBACK);
 			if (raw_data == NULL)
 				goto fatal_error;
 
