diff --git a/gthumb/main-migrate-catalogs.c b/gthumb/main-migrate-catalogs.c
index 75f5cb1..88e6451 100644
--- a/gthumb/main-migrate-catalogs.c
+++ b/gthumb/main-migrate-catalogs.c
@@ -171,7 +171,7 @@ migration_for_each_file (GFile     *file,
 				gboolean    b;
 				int         line_ofs;
 				/*gboolean    all_keywords;*/
-				long        date;
+				time_t      date;
 				int         date_scope;
 
 				extension = ".search";
