diff -Naur man-1.5p-orig/src/man.conf.in man-1.5p/src/man.conf.in
--- man-1.5p-orig/src/man.conf.in	2005-01-08 10:50:45.000000000 -0800
+++ man-1.5p/src/man.conf.in	2005-03-18 13:22:17.000000000 -0800
@@ -88,9 +88,9 @@
 # If you have a new troff (version 1.18.1?) and its colored output
 # causes problems, add the -c option to TROFF, NROFF, JNROFF.
 #
-TROFF		@troff@
-NROFF		@nroff@
-JNROFF		@jnroff@
+TROFF		@troff@ -c
+NROFF		@nroff@ -c
+JNROFF		@jnroff@ -c
 EQN		@eqn@
 NEQN		@neqn@
 JNEQN		@jneqn@
