diff --git a/loader/Makefile.in b/loader/Makefile.in
index 8190037..b4f2902 100644
--- a/loader/Makefile.in
+++ b/loader/Makefile.in
@@ -29,4 +29,4 @@ wine_preloader_OBJS = preloader.o
 wine_preloader_LDFLAGS = -static -nostartfiles -nodefaultlibs -Wl,-Ttext=0x7c400000
 
 wine64_preloader_OBJS = preloader.o
-wine64_preloader_LDFLAGS = -static -nostartfiles -nodefaultlibs -Wl,-Ttext=0x7c400000
+wine64_preloader_LDFLAGS = -static -nostartfiles -nodefaultlibs -Wl,-Ttext=0x7c400000 -no-pie
