diff --git a/dlls/ntdll/unix/serial.c b/dlls/ntdll/unix/serial.c
index bb8d52cc627..110cf765270 100644
--- a/dlls/ntdll/unix/serial.c
+++ b/dlls/ntdll/unix/serial.c
@@ -127,9 +127,9 @@ static NTSTATUS get_baud_rate(int fd, SERIAL_BAUD_RATE* sbr)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 
@@ -219,9 +219,9 @@ static NTSTATUS get_hand_flow(int fd, SERIAL_HANDFLOW* shf)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -281,9 +281,9 @@ static NTSTATUS get_line_control(int fd, SERIAL_LINE_CONTROL* slc)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -397,9 +397,9 @@ static NTSTATUS get_special_chars(int fd, SERIAL_CHARS* sc)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -512,9 +512,9 @@ static NTSTATUS set_baud_rate(int fd, const SERIAL_BAUD_RATE* sbr)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 
@@ -526,9 +526,9 @@ static NTSTATUS set_baud_rate(int fd, const SERIAL_BAUD_RATE* sbr)
     port.c_cflag |= BOTHER << IBSHIFT;
     port.c_ispeed = sbr->BaudRate;
 
-    if (ioctl(fd, TCSETS2, &port) == -1)
+    if (ioctl(fd, TCSETS, &port) == -1)
     {
-        ERR("ioctl TCSETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCSETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -674,9 +674,9 @@ static NTSTATUS set_handflow(int fd, const SERIAL_HANDFLOW* shf)
         return STATUS_NOT_SUPPORTED;
 
 #ifdef HAVE_ASM_TERMBITS_H
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -726,9 +726,9 @@ static NTSTATUS set_handflow(int fd, const SERIAL_HANDFLOW* shf)
         port.c_iflag &= ~IXON;
 
 #ifdef HAVE_ASM_TERMBITS_H
-    if (ioctl(fd, TCSETS2, &port) == -1)
+    if (ioctl(fd, TCSETS, &port) == -1)
     {
-        ERR("ioctl TCSETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCSETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -748,9 +748,9 @@ static NTSTATUS set_line_control(int fd, const SERIAL_LINE_CONTROL* slc)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -857,9 +857,9 @@ static NTSTATUS set_line_control(int fd, const SERIAL_LINE_CONTROL* slc)
     }
     /* otherwise it hangs with pending input*/
 #ifdef HAVE_ASM_TERMBITS_H
-    if (ioctl(fd, TCSETS2, &port) == -1)
+    if (ioctl(fd, TCSETS, &port) == -1)
     {
-        ERR("ioctl TCSETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCSETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -883,9 +883,9 @@ static NTSTATUS set_special_chars(int fd, const SERIAL_CHARS* sc)
 #ifdef HAVE_ASM_TERMBITS_H
     struct termios2 port;
 
-    if (ioctl(fd, TCGETS2, &port) == -1)
+    if (ioctl(fd, TCGETS, &port) == -1)
     {
-        ERR("ioctl TCGETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCGETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
@@ -906,9 +906,9 @@ static NTSTATUS set_special_chars(int fd, const SERIAL_CHARS* sc)
     port.c_cc[VSTOP ] = sc->XoffChar;
 
 #ifdef HAVE_ASM_TERMBITS_H
-    if (ioctl(fd, TCSETS2, &port) == -1)
+    if (ioctl(fd, TCSETS, &port) == -1)
     {
-        ERR("ioctl TCSETS2 error '%s'\n", strerror(errno));
+        ERR("ioctl TCSETS error '%s'\n", strerror(errno));
         return errno_to_status( errno );
     }
 #else
