From 200e8020b34d4a2881d56a6c17c7ed653f43eb8b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 3 May 2023 17:12:04 +0200
Subject: [PATCH] fix ebs alias

Add missing -b option
---
 persistent-storage | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/persistent-storage b/persistent-storage
index cc834ae..bb97264 100644
--- a/persistent-storage
+++ b/persistent-storage
@@ -118,7 +118,7 @@ fi
 if [ "${MDEV#nvme}" != "$MDEV" ] && [ "$model" = "Amazon_Elastic_Block_Store" ] && command -v nvme >/dev/null; then
 	n=30
 	while [ $n -gt 0 ]; do
-		ebs_alias=$(nvme id-ctrl /dev/$_check_dev \
+		ebs_alias=$(nvme id-ctrl -b /dev/$_check_dev \
 			| dd bs=32 skip=96 count=1 2>/dev/null \
 			| sed -nre '/^(\/dev\/)?(s|xv)d[a-z]{1,2} /p' \
 			| tr -d ' ')
-- 
2.40.1

