--- daq-2.0.1.orig/os-daq-modules/daq_ipfw.c
+++ daq-2.0.1/os-daq-modules/daq_ipfw.c
@@ -26,7 +26,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/unistd.h>
+#include <unistd.h>
 
 #include <netinet/in.h>
 #include <sys/socket.h>
--- daq-2.0.1.orig/os-daq-modules/daq_ipq.c
+++ daq-2.0.1/os-daq-modules/daq_ipq.c
@@ -27,7 +27,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/unistd.h>
+#include <unistd.h>
 
 #include <netinet/ip.h>
 
--- daq-2.0.1.orig/os-daq-modules/daq_nfq.c
+++ daq-2.0.1/os-daq-modules/daq_nfq.c
@@ -27,7 +27,7 @@
 
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/unistd.h>
+#include <unistd.h>
 
 #include <netinet/ip.h>
 
