From 986afad517c3456a0d6f9e598c31c8dfa9b42ac8 Mon Sep 17 00:00:00 2001
From: ptrcnull <git@ptrcnull.me>
Date: Tue, 13 Sep 2022 02:04:39 +0200
Subject: [PATCH] LibWeb: Fallback to 'unknown' on unknown architectures

---
 Userland/Libraries/LibWeb/Loader/ResourceLoader.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Userland/Libraries/LibWeb/Loader/ResourceLoader.h b/Userland/Libraries/LibWeb/Loader/ResourceLoader.h
index 40454db13cd8d..bc1ce9c5521ec 100644
--- a/Userland/Libraries/LibWeb/Loader/ResourceLoader.h
+++ b/Userland/Libraries/LibWeb/Loader/ResourceLoader.h
@@ -24,6 +24,8 @@ namespace Web {
 #    define CPU_STRING "x86_64"
 #elif ARCH(AARCH64)
 #    define CPU_STRING "AArch64"
+#else
+#    define CPU_STRING "unknown"
 #endif
 
 constexpr auto default_user_agent = "Mozilla/5.0 (SerenityOS; " CPU_STRING ") LibWeb+LibJS/1.0 Browser/1.0";
