diff --git a/Utilities.cpp b/Utilities.cpp
index 93bd83e..8966855 100644
--- a/Utilities.cpp
+++ b/Utilities.cpp
@@ -30,18 +30,6 @@ void platform_init()
     extern void android_platform_init();
     android_platform_init();
 #else
-    s_serenity_resource_root = [] {
-        auto const* source_dir = getenv("SERENITY_SOURCE_DIR");
-        if (source_dir) {
-            return String::formatted("{}/Base", source_dir);
-        }
-        auto* home = getenv("XDG_CONFIG_HOME") ?: getenv("HOME");
-        VERIFY(home);
-        auto home_lagom = String::formatted("{}/.lagom", home);
-        if (Core::File::is_directory(home_lagom))
-            return home_lagom;
-        auto app_dir = akstring_from_qstring(QCoreApplication::applicationDirPath());
-        return LexicalPath(app_dir).parent().append("share"sv).string();
-    }();
+    s_serenity_resource_root = String::formatted("/usr/share/ladybird");
 #endif
 }
diff --git a/WebContent/main.cpp b/WebContent/main.cpp
index 2623693..5a059a2 100644
--- a/WebContent/main.cpp
+++ b/WebContent/main.cpp
@@ -48,7 +48,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     Web::ResourceLoader::initialize(RequestManagerQt::create());
     Web::WebSockets::WebSocketClientManager::initialize(Ladybird::WebSocketClientManagerLadybird::create());
 
-    Web::FrameLoader::set_default_favicon_path(String::formatted("{}/res/icons/16x16/app-browser.png", s_serenity_resource_root));
+    Web::FrameLoader::set_default_favicon_path(String::formatted("/usr/share/icons/hicolor/16x16/apps/ladybird.png"));
 
     Web::Platform::FontPlugin::install(*new Ladybird::FontPluginQt);
 
