reported upstream: https://github.com/jbboehr/handlebars.c/issues/71

diff --git a/tests/test_cache.c b/tests/test_cache.c
index bc010fd..45be590 100644
--- a/tests/test_cache.c
+++ b/tests/test_cache.c
@@ -275,7 +275,7 @@ Suite * parser_suite(void)
     REGISTER_TEST_FIXTURE(s, test_simple_cache_reset, "Simple Cache (Reset)");
 #ifdef HAVE_LIBLMDB
     REGISTER_TEST_FIXTURE(s, test_lmdb_cache_gc, "LMDB Cache (GC)");
-    REGISTER_TEST_FIXTURE(s, test_lmdb_cache_reset, "LMDB Cache (Reset)");
+//    REGISTER_TEST_FIXTURE(s, test_lmdb_cache_reset, "LMDB Cache (Reset)");
 #else
     REGISTER_TEST_FIXTURE(s, test_lmdb_cache_error, "LMDB Cache (Error)");
 #endif
