From 09dfa87f3189c6b26d717a7cd470d7c79254f2f7 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Mon, 26 Feb 2024 09:55:57 +0100
Subject: [PATCH 1/3] CMakeLists.txt: Use SVG icon as desktop app icon.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ebcb355..a085c40 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,7 +21,7 @@ set(LIBREOFFICE_INCLUDE_DIR "" CACHE PATH "Custom LibreOfficeKit include directo
 
 set(APP_NAME            lomiri-docviewer-app)
 set(DESKTOP_FILE        "${PROJECT_NAME}.desktop")
-set(ICON_FILE           docviewer-app.png)
+set(ICON_FILE           docviewer-app.svg)
 set(SPLASH_FILE         docviewer-app-splash.svg)
 set(AUTOPILOT_DIR       lomiri_docviewer_app)
 
-- 
GitLab


From 3797d91f205ca72166e413bf2272a09164dd5c72 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Mon, 26 Feb 2024 09:56:06 +0100
Subject: [PATCH 2/3] data/CMakeLists.txt: In non-click mode, fix installation
 of desktop app icon into datadir.

---
 data/CMakeLists.txt | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 05b1f4e..ce552e1 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -11,9 +11,7 @@ else(CLICK_MODE)
   set(DESKTOP_DIR ${CMAKE_INSTALL_DATADIR}/applications)
   set(EXEC "${APP_NAME}")
 
-  install(DIRECTORY icons/hicolor DESTINATION ${CMAKE_INSTALL_DATADIR}/icons
-    FILES_MATCHING PATTERN *.png
-  )
+  install(FILES ${ICON_FILE} DESTINATION ${DATA_DIR})
 endif(CLICK_MODE)
 
 set(TRYEXEC "${APP_NAME}")
-- 
GitLab


From 4d09277013f3ae12772a3888d90083c4c7ceff1c Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Mon, 26 Feb 2024 09:56:31 +0100
Subject: [PATCH 3/3] data/CMakeLists.txt: Do install splash file in non-click
 mode.

---
 data/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index ce552e1..da34181 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -8,10 +8,12 @@ if(CLICK_MODE)
   install(FILES ${SPLASH_FILE} DESTINATION ${DATA_DIR})
 else(CLICK_MODE)
   set(ICON "${CMAKE_INSTALL_PREFIX}/${DATA_DIR}/${ICON_FILE}")
+  set(SPLASH "${CMAKE_INSTALL_PREFIX}/${DATA_DIR}/${SPLASH_FILE}")
   set(DESKTOP_DIR ${CMAKE_INSTALL_DATADIR}/applications)
   set(EXEC "${APP_NAME}")
 
   install(FILES ${ICON_FILE} DESTINATION ${DATA_DIR})
+  install(FILES ${SPLASH_FILE} DESTINATION ${DATA_DIR})
 endif(CLICK_MODE)
 
 set(TRYEXEC "${APP_NAME}")
-- 
GitLab

