diff --git a/ginkgo_t_dsl.go b/ginkgo_t_dsl.go
index 40d1e1a..7ad9ab8 100644
--- a/ginkgo_t_dsl.go
+++ b/ginkgo_t_dsl.go
@@ -72,6 +72,7 @@ type GinkgoTInterface interface {
 	TempDir() string
 	Attr(key, value string)
 	Output() io.Writer
+	ArtifactDir() string
 }
 
 /*
diff --git a/internal/output_interceptor_test.go b/internal/output_interceptor_test.go
diff --git a/internal/testingtproxy/testing_t_proxy.go b/internal/testingtproxy/testing_t_proxy.go
index 9806e31..e55bbbe 100644
--- a/internal/testingtproxy/testing_t_proxy.go
+++ b/internal/testingtproxy/testing_t_proxy.go
@@ -235,3 +235,6 @@ func (t *ginkgoTestingTProxy) Output() io.Writer {
 func (t *ginkgoTestingTProxy) Attr(key, value string) {
 	t.addReportEntry(key, value, internal.Offset(1), types.ReportEntryVisibilityFailureOrVerbose)
 }
+func (t *ginkgoTestingTProxy) ArtifactDir() string {
+	panic("Not implemented")
+}
