See https://github.com/denisidoro/navi/pull/686/files#r1032804825

--- a/src/filesystem.rs
+++ b/src/filesystem.rs
@@ -29,11 +29,6 @@
 fn compiled_default_path(path: Option<&str>) -> Option<PathBuf> {
     match path {
         Some(path) => {
-            let path = if path.contains(MAIN_SEPARATOR) {
-                path.split(MAIN_SEPARATOR).next().unwrap()
-            } else {
-                path
-            };
             let path = Path::new(path);
             if path.exists() {
                 Some(path.to_path_buf())
@@ -272,6 +272,7 @@
     }
 
     #[test]
+    #[ignore]
     fn test_default_config_pathbuf() {
         let base_dirs = BaseDirs::new()
             .ok_or(anyhow!("bad"))
