--- a/diffoscope/comparators/bzip2.py
+++ b/diffoscope/comparators/bzip2.py
@@ -45,7 +45,7 @@ class Bzip2Container(Archive):
         logger.debug("bzip2 extracting to %s", dest_path)
         with open(dest_path, "wb") as fp:
             subprocess.check_call(
-                ["bzip2", "--decompress", "--stdout", self.source.path],
+                ["bzip2", "-d", "-c", self.source.path],
                 stdout=fp,
                 stderr=subprocess.PIPE,
             )
