From 41da74de67771b8a822f11e72b17248f56aa9766 Mon Sep 17 00:00:00 2001
From: Nicolas Werner <nicolas.werner@hotmail.de>
Date: Tue, 22 Aug 2023 03:31:50 +0200
Subject: [PATCH] Fix provides section in wraps

---
 subprojects/curl.wrap | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subprojects/curl.wrap b/subprojects/curl.wrap
index f1510b4..ec9710f 100644
--- a/subprojects/curl.wrap
+++ b/subprojects/curl.wrap
@@ -5,5 +5,5 @@ source_url = https://github.com/curl/curl/releases/download/curl-7_77_0/curl-7.7
 source_filename = curl-7.77.0.tar.xz
 source_hash = 0f64582c54282f31c0de9f0a1a596b182776bd4df9a4c4a2a41bbeb54f62594b
 
-[provides]
+[provide]
 dependency_names = libcurl
-- 
2.42.0

