diff --git a/Makefile b/Makefile
index 47cf410..2d3f916 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 VERSION=3.3
 
 # To test strict C ANSI compliance
-CC = gcc -ansi -pedantic
+CC ?= gcc -ansi -pedantic
 LIB= -lm
 
 # This allow you to write commands like "make PURE=purify demo1"
@@ -33,7 +33,7 @@ CFLAGS += -DLITTLE_ENDIAN
 # GENERAL FLAGS FOR GCC
 
 # Optimized code
-CFLAGS += -O6
+CFLAGS ?= -O2
 
 # Debug mode with gdb
 #CFLAGS += -g 
