diff --git a/configure.self b/configure.self
index 514e9c6..393cd5d 100644
--- a/configure.self
+++ b/configure.self
@@ -63,17 +63,19 @@ function checks {
         fi
     fi
 
-    # Check that C++11 is supported
+    # Check that C++17 is supported (required by avro headers)
     if [[ $ENABLE_AVRO_CPP == y ]]; then
-        mkl_meta_set "cxx11" "name" "C++11 support"
-        if mkl_compile_check "cxx11" HAVE_CXX11 fail CXX "--std=c++11" \
+        mkl_meta_set "cxx17" "name" "C++17 support"
+        if mkl_compile_check "cxx17" HAVE_CXX17 fail CXX "--std=c++17" \
                              "
+#include <optional>
+#include <any>
 int foo () {
-     auto var = 15;
+     std::optional<int> var = 15;
 
-     return var;
+     return *var;
 }"; then
-            mkl_mkvar_append CXXFLAGS CXXFLAGS "--std=c++11"
+            mkl_mkvar_append CXXFLAGS CXXFLAGS "--std=c++17"
         fi
     fi
 
