diff --git a/test/python_magic_test.py b/test/python_magic_test.py
index 624a443..00472fa 100755
--- a/test/python_magic_test.py
+++ b/test/python_magic_test.py
@@ -134,7 +134,7 @@ class MagicTest(unittest.TestCase):
             self.assert_values(m, {
                 # some versions return '' for the extensions of a gz file,
                 # including w/ the command line.  Who knows...
-                'test.gz': ('gz/tgz/tpz/zabw/svgz', '', '???'),
+                'test.gz': ('gz/tgz/tpz/zabw/svgz', '', '???', 'gz/tgz/tpz/zabw/svgz/adz/kmy/xcfgz'),
                 'name_use.jpg': 'jpeg/jpg/jpe/jfif',
             })
         except NotImplementedError:
