diff -rupN a/test/python_magic_test.py b/test/python_magic_test.py
--- a/test/python_magic_test.py	2022-06-07 19:50:57.000000000 +0000
+++ b/test/python_magic_test.py	2026-03-28 03:53:45.400000000 +0000
@@ -93,7 +93,7 @@ class MagicTest(unittest.TestCase):
                 'magic._pyc_': ('application/octet-stream', 'text/x-bytecode.python', 'application/x-bytecode.python'),
                 'test.pdf': 'application/pdf',
                 'test.gz': ('application/gzip', 'application/x-gzip'),
-                'test.snappy.parquet': 'application/octet-stream',
+                'test.snappy.parquet': 'application/vnd.apache.parquet',
                 'text.txt': 'text/plain',
                 b'\xce\xbb'.decode('utf-8'): 'text/plain',
                 b'\xce\xbb': 'text/plain',
