diff --git a/examples/example_postprocess.py b/examples/example_postprocess.py
index 069fb6e..057ea30 100755
--- a/examples/example_postprocess.py
+++ b/examples/example_postprocess.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 import fileinput
 import random
 import sys
diff --git a/xkcdpass/xkcd_password.py b/xkcdpass/xkcd_password.py
index fe4d8a8..31274ef 100755
--- a/xkcdpass/xkcd_password.py
+++ b/xkcdpass/xkcd_password.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # encoding: utf-8
 
 from __future__ import print_function
