From 53065f4a99d94c6576067df4bc34f1f6fd32d59a Mon Sep 17 00:00:00 2001
From: Remi Collet <remi@remirepo.net>
Date: Wed, 9 Jun 2021 15:51:54 +0200
Subject: [PATCH] relax test, hide deprecation message

---
 tests/yaml_parse_file_002.phpt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/yaml_parse_file_002.phpt b/tests/yaml_parse_file_002.phpt
index 83c54c7..d626eab 100644
--- a/tests/yaml_parse_file_002.phpt
+++ b/tests/yaml_parse_file_002.phpt
@@ -5,6 +5,8 @@ yaml_parse_file - error cases
 --INI--
 yaml.decode_timestamp=1
 date.timezone=GMT
+; E_ALL - E_DEPRECATED to hide Deprecated: yaml_parse_file(): Passing null to parameter #1...
+error_reporting=24575
 --FILE--
 <?php
 try {
