Index: sh/gendepends.sh.in
===================================================================
--- a/sh/gendepends.sh.in	(revision 1511)
+++ b/sh/gendepends.sh.in	(working copy)
@@ -81,6 +81,7 @@
 		[ -e @SYSCONFDIR@/rc.conf ] && . @SYSCONFDIR@/rc.conf
 
 		if . "${_dir}/${RC_SVCNAME}"; then
+			_shell_var_RC_SVCNAME=$(shell_var "${RC_SVCNAME}")
 			echo "${RC_SVCNAME}" >&3
 			depend
 
@@ -91,9 +92,9 @@
 				IFS=:
 				set -- ${_deptype}
 				unset IFS
-				eval _depends=\$rc_$(shell_var "${RC_SVCNAME}")_$1
+				eval _depends=\$rc_${_shell_var_RC_SVCNAME}_$1
 				[ -z "${_depends}" ] && eval _depends=\$rc_$1
-				[ -z "${_depends}" ] && eval _depends=\$RC_$(shell_var "${RC_SVCNAME}")_$2
+				[ -z "${_depends}" ] && eval _depends=\$RC_${_shell_var_RC_SVCNAME}_$2
 				[ -z "${_depends}" ] && eval _depends=\$RC_$2
 
 				$1 ${_depends}
