diff --git a/tests/meson.build b/tests/meson.build
index 6c8077622d..9b77e84be9 100644
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -1,7 +1,6 @@
 tests_sources = [
   'util.c',
   'coroutine.c',
-  'session.c',
   'uri.c',
   'file-transfer.c',
 ]
@@ -10,9 +9,6 @@
   tests_sources += 'pipe.c'
 endif
 
-if spice_gtk_has_usbredir
-  tests_sources += 'cd-emu.c'
-endif
 
 if spice_gtk_has_polkit
   tests_sources += [
