--- a/build/moz.configure/rust.configure
+++ b/build/moz.configure/rust.configure
@@ -127,10 +127,10 @@
             ('x86_64', 'OpenBSD'): 'x86_64-unknown-openbsd',
             # Linux
             ('aarch64', 'Linux'): 'aarch64-unknown-linux-gnu',
-            ('arm', 'Linux'): 'armv7-unknown-linux-gnueabihf',
+            ('arm', 'Linux'): 'armv6-alpine-linux-musleabihf',
             ('sparc64', 'Linux'): 'sparc64-unknown-linux-gnu',
-            ('x86', 'Linux'): 'i686-unknown-linux-gnu',
-            ('x86_64', 'Linux'): 'x86_64-unknown-linux-gnu',
+            ('x86', 'Linux'): 'i686-alpine-linux-musl',
+            ('x86_64', 'Linux'): 'x86_64-alpine-linux-musl',
             # OS X
             ('x86', 'OSX'): 'i686-apple-darwin',
             ('x86_64', 'OSX'): 'x86_64-apple-darwin',
