--- a/build/moz.configure/rust.configure	2017-05-27 10:48:23.927528005 +0000
+++ b/build/moz.configure/rust.configure	2017-05-27 10:52:42.973790729 +0000
@@ -130,7 +130,7 @@
             ('arm', 'Linux'): 'armv7-unknown-linux-gnueabihf',
             ('sparc64', 'Linux'): 'sparc64-unknown-linux-gnu',
             ('x86', 'Linux'): 'i686-unknown-linux-gnu',
-            ('x86_64', 'Linux'): 'x86_64-unknown-linux-gnu',
+            ('x86_64', 'Linux'): 'x86_64-alpine-linux-musl',
             # OS X
             ('x86', 'OSX'): 'i686-apple-darwin',
             ('x86_64', 'OSX'): 'x86_64-apple-darwin',
