diff --git a/tools/profiler/core/platform.h b/tools/profiler/core/platform.h
index 5f0fb69..086a511 100644
--- a/tools/profiler/core/platform.h
+++ b/tools/profiler/core/platform.h
@@ -42,10 +42,11 @@
 #include "PlatformMacros.h"
 #include <vector>
 
-// We need a definition of gettid(), but glibc doesn't provide a
-// wrapper for it.
-#if defined(__GLIBC__)
+// We need a definition of gettid(), but Linux libc implementations don't
+// provide a wrapper for it (except for Bionic)
+#if defined(__linux__)
 #  include <unistd.h>
+#if !defined(__BIONIC__)
 #  include <sys/syscall.h>
 static inline pid_t gettid() { return (pid_t)syscall(SYS_gettid); }
 #elif defined(GP_OS_darwin)
@@ -61,6 +62,7 @@ static inline pid_t gettid() { return (pid_t)syscall(SYS_thread_selfid); }
 #    define getpid _getpid
 #  endif
 #endif
+#endif
 
 extern mozilla::LazyLogModule gProfilerLog;
 
