--- ./security/sandbox/linux/moz.build.orig
+++ ./security/sandbox/linux/moz.build
@@ -77,6 +77,9 @@
         'SandboxOpenedFiles.cpp',
     ]
 
+if CONFIG['MOZ_ALSA']:
+    DEFINES['MOZ_ALSA'] = True
+
 # This copy of SafeSPrintf doesn't need to avoid the Chromium logging
 # dependency like the one in libxul does, but this way the behavior is
 # consistent.  See also the comment in SandboxLogging.h.
