Hack to workaround buggy error handling code.

Upstream-status: Not submitted
--- libparistraceroute-0.93.orig/paris-traceroute/paris-traceroute.c
+++ libparistraceroute-0.93/paris-traceroute/paris-traceroute.c
@@ -453,6 +453,7 @@
         goto ERR_PT_LOOP;
     }
     exit_code = EXIT_SUCCESS;
+    errno = 0;
 
     // Leave the program
 ERR_PT_LOOP:
--- libparistraceroute-0.93.orig/ping/ping.c
+++ libparistraceroute-0.93/ping/ping.c
@@ -500,6 +500,7 @@
     }
 
     exit_code = EXIT_SUCCESS;
+    errno = 0;
 
     // Leave the program
 ERR_PT_LOOP:
