--- a/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-body-parts-custom.phpt
+++ b/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-body-parts-custom.phpt
@@ -35,7 +35,7 @@
 $tester->start();
 $tester->expectLogStartNotices();
 echo $tester
-    ->request(stdin: [
+    ->request('', [], null, null, null, null, false, null, [
         'parts' => [
             'count' => 30,
         ]
--- a/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-body-parts-default.phpt
+++ b/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-body-parts-default.phpt
@@ -34,7 +34,7 @@
 $tester->start();
 $tester->expectLogStartNotices();
 echo $tester
-    ->request(stdin: [
+    ->request('', [], null, null, null, null, false, null, [
         'parts' => [
             'count' => 30,
         ]
--- a/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-file-uploads.phpt
+++ b/sapi/fpm/tests/ghsa-54hq-v5wp-fqgv-max-file-uploads.phpt
@@ -33,7 +33,7 @@
 $tester->start();
 $tester->expectLogStartNotices();
 echo $tester
-    ->request(stdin: [
+    ->request('', [], null, null, null, null, false, null, [
         'parts' => [
             'count' => 10,
             'param' => 'filename'
