diff --git b/ext/intl/tests/locale_filter_matches3.phpt a/ext/intl/tests/locale_filter_matches3.phpt
index 2816ba5e12..8c39d189eb 100644
--- b/ext/intl/tests/locale_filter_matches3.phpt
+++ a/ext/intl/tests/locale_filter_matches3.phpt
@@ -1,8 +1,8 @@
 --TEST--
-locale_filter_matches.phpt() ICU >= 51.2
+locale_filter_matches.phpt() ICU >= 51.2 && ICU < 67.1
 --SKIPIF--
 <?php if( !extension_loaded( 'intl' ) ) print 'skip'; ?>
-<?php if (version_compare(INTL_ICU_VERSION, '51.2') < 0) die('skip for ICU >= 51.2'); ?>
+<?php if (version_compare(INTL_ICU_VERSION, '67.1') >= 0) die('skip for ICU < 67.1'); ?>
 --FILE--
 <?php
 
diff --git b/ext/intl/tests/locale_lookup_variant2.phpt a/ext/intl/tests/locale_lookup_variant2.phpt
index 75a5350abf..af104e1ba8 100644
--- b/ext/intl/tests/locale_lookup_variant2.phpt
+++ a/ext/intl/tests/locale_lookup_variant2.phpt
@@ -2,7 +2,7 @@
 locale_lookup.phpt()
 --SKIPIF--
 <?php if( !extension_loaded( 'intl' ) ) print 'skip'; ?>
-<?php if (version_compare(INTL_ICU_VERSION, '51.2') < 0) die('skip for ICU >= 51.2'); ?>
+<?php if (version_compare(INTL_ICU_VERSION, '67.1') >= 0) die('skip for ICU < 67.1'); ?>
 --FILE--
 <?php
 
