--- a/sapi/fpm/fpm/fpm_conf.c
+++ b/sapi/fpm/fpm/fpm_conf.c
@@ -1180,3 +1180,3 @@
 	if (!fpm_global_config.error_log) {
-		fpm_global_config.error_log = strdup("log/php-fpm.log");
+		fpm_global_config.error_log = strdup("log/php7/error.log");
 	}
@@ -1191,3 +1191,3 @@
 	if (!fpm_global_config.syslog_ident) {
-		fpm_global_config.syslog_ident = strdup("php-fpm");
+		fpm_global_config.syslog_ident = strdup("php-fpm7");
 	}
@@ -1683,3 +1683,3 @@
 		} else {
-			spprintf(&tmp, 0, "%s/etc/php-fpm.conf", fpm_globals.prefix);
+			spprintf(&tmp, 0, "%s/etc/php7/php-fpm.conf", fpm_globals.prefix);
 		}
--- a/sapi/fpm/php-fpm.conf.in
+++ b/sapi/fpm/php-fpm.conf.in
@@ -16,3 +16,3 @@
 ; Default Value: none
-;pid = run/php-fpm.pid
+;pid = run/php-fpm7.pid
 
@@ -22,4 +22,4 @@
 ; Note: the default prefix is @EXPANDED_LOCALSTATEDIR@
-; Default Value: log/php-fpm.log
-;error_log = log/php-fpm.log
+; Default Value: log/php7/error.log
+;error_log = log/php7/error.log
 
@@ -35,4 +35,4 @@
 ; which must suit common needs.
-; Default Value: php-fpm
-;syslog.ident = php-fpm
+; Default Value: php-fpm7
+;syslog.ident = php-fpm7
 
--- a/sapi/fpm/www.conf.in
+++ b/sapi/fpm/www.conf.in
@@ -252,3 +252,3 @@
 ; Default: not set
-;access.log = log/$pool.access.log
+;access.log = log/php7/$pool.access.log
 
@@ -316,3 +316,3 @@
 ; Note: slowlog is mandatory if request_slowlog_timeout is set
-;slowlog = log/$pool.log.slow
+;slowlog = log/php7/$pool.slow.log
 
@@ -410,3 +410,3 @@
 ;php_flag[display_errors] = off
-;php_admin_value[error_log] = /var/log/fpm-php.www.log
+;php_admin_value[error_log] = /var/log/php7/$pool.error.log
 ;php_admin_flag[log_errors] = on
