diff --git a/src/rtpp_log_stand.c b/src/rtpp_log_stand.c
index c1e4331..d02eeb4 100644
--- a/src/rtpp_log_stand.c
+++ b/src/rtpp_log_stand.c
@@ -273,7 +273,7 @@ _rtpp_log_write_va(struct rtpp_log_inst *rli, int level, const char *function,
     fprintf(stderr, rli->format_se[0], rtpp_time_buff, strlvl(level),
       call_id, function);
     vfprintf(stderr, format, ap);
-    fprintf(stderr, rli->format_se[1]);
+    fprintf(stderr, "%s", rli->format_se[1]);
     fflush(stderr);
     _rtpp_log_unlock();
 }
