--- a/public_suffix.gemspec
+++ b/public_suffix.gemspec
@@ -22,12 +22,7 @@
     "source_code_uri" => "https://github.com/weppos/publicsuffix-ruby/tree/v#{s.version}",
   }
 
-  s.files = Dir.chdir(__dir__) do
-    `git ls-files -z`.split("\x0").reject do |f|
-      (File.expand_path(f) == __FILE__) ||
-        f.start_with?(*%w[bin/ test/ .git .rubocop Gemfile Rakefile])
-    end
-  end
+  s.files         = Dir["data/**/*", "lib/**/*"]
   s.require_paths = ["lib"]
   s.extra_rdoc_files = %w( LICENSE.txt )
 end
