Patch-Source: https://github.com/bbangert/routes/pull/113

diff --git a/routes/mapper.py b/routes/mapper.py
index 1541f3d..d7acbe4 100644
--- a/routes/mapper.py
+++ b/routes/mapper.py
@@ -4,8 +4,6 @@ import itertools as it
 import re
 import threading
 
-import six
-
 from routes import request_config
 from routes.util import (
     controller_scan,
@@ -189,7 +187,7 @@ class SubMapper(SubMapperParent):
         newkargs = {}
         _routename = routename
         _path = path
-        for key, value in six.iteritems(self.kwargs):
+        for key, value in self.kwargs.items():
             if key == 'path_prefix':
                 if path is not None:
                     # if there's a name_prefix, add it to the route name
@@ -616,7 +614,7 @@ class Mapper(SubMapperParent):
             if 'controller' in route.hardcoded:
                 clist = [route.defaults['controller']]
             if 'action' in route.hardcoded:
-                alist = [six.text_type(route.defaults['action'])]
+                alist = [str(route.defaults['action'])]
             for controller in clist:
                 for action in alist:
                     actiondict = gendict.setdefault(controller, {})
@@ -646,7 +644,7 @@ class Mapper(SubMapperParent):
             else:
                 clist = self.controller_scan
 
-        for key, val in six.iteritems(self.maxkeys):
+        for key, val in self.maxkeys.items():
             for route in val:
                 route.makeregexp(clist)
 
@@ -822,11 +820,11 @@ class Mapper(SubMapperParent):
         # If the URL didn't depend on the SCRIPT_NAME, we'll cache it
         # keyed by just by kargs; otherwise we need to cache it with
         # both SCRIPT_NAME and kargs:
-        cache_key = six.text_type(args).encode('utf8') + \
-            six.text_type(kargs).encode('utf8')
+        cache_key = str(args).encode('utf8') + \
+            str(kargs).encode('utf8')
 
         if self.urlcache is not None:
-            if six.PY3:
+            if True:
                 cache_key_script_name = b':'.join((script_name.encode('utf-8'),
                                                    cache_key))
             else:
@@ -850,7 +848,7 @@ class Mapper(SubMapperParent):
 
         keys = frozenset(kargs.keys())
         cacheset = False
-        cachekey = six.text_type(keys)
+        cachekey = str(keys)
         cachelist = sortcache.get(cachekey)
         if args:
             keylist = args
@@ -1132,7 +1130,7 @@ class Mapper(SubMapperParent):
         def swap(dct, newdct):
             """Swap the keys and values in the dict, and uppercase the values
             from the dict during the swap."""
-            for key, val in six.iteritems(dct):
+            for key, val in dct.items():
                 newdct.setdefault(val.upper(), []).append(key)
             return newdct
         collection_methods = swap(collection, {})
@@ -1175,7 +1173,7 @@ class Mapper(SubMapperParent):
             return opts
 
         # Add the routes for handling collection methods
-        for method, lst in six.iteritems(collection_methods):
+        for method, lst in collection_methods.items():
             primary = (method != 'GET' and lst.pop(0)) or None
             route_options = requirements_for(method)
             for action in lst:
@@ -1199,7 +1197,7 @@ class Mapper(SubMapperParent):
                      action='index', conditions={'method': ['GET']}, **options)
 
         # Add the routes that deal with new resource methods
-        for method, lst in six.iteritems(new_methods):
+        for method, lst in new_methods.items():
             route_options = requirements_for(method)
             for action in lst:
                 name = "new_" + member_name
@@ -1218,7 +1216,7 @@ class Mapper(SubMapperParent):
         requirements_regexp = '[^\\/]+(?<!\\\\)'
 
         # Add the routes that deal with member methods of a resource
-        for method, lst in six.iteritems(member_methods):
+        for method, lst in member_methods.items():
             route_options = requirements_for(method)
             route_options['requirements'] = {'id': requirements_regexp}
             if method not in ['POST', 'GET', 'any']:
diff --git a/routes/route.py b/routes/route.py
index 719636a..a721126 100644
--- a/routes/route.py
+++ b/routes/route.py
@@ -1,8 +1,7 @@
 import re
 import sys
 
-import six
-from six.moves.urllib import parse as urlparse
+from urllib import parse as urlparse
 
 from routes.util import _url_quote as url_quote, _str_encode, as_unicode
 
@@ -97,7 +96,7 @@ class Route(object):
 
         # Build a req list with all the regexp requirements for our args
         self.req_regs = {}
-        for key, val in six.iteritems(self.reqs):
+        for key, val in self.reqs.items():
             self.req_regs[key] = re.compile('^' + val + '$')
         # Update our defaults and set new default keys if needed. defaults
         # needs to be saved
@@ -133,14 +132,14 @@ class Route(object):
 
     def make_unicode(self, s):
         """Transform the given argument into a unicode string."""
-        if isinstance(s, six.text_type):
+        if isinstance(s, str):
             return s
         elif isinstance(s, bytes):
             return s.decode(self.encoding)
         elif callable(s):
             return s
         else:
-            return six.text_type(s)
+            return str(s)
 
     def _pathkeys(self, routepath):
         """Utility function to walk the route, and pull out the valid
@@ -567,7 +566,7 @@ class Route(object):
         matchdict = match.groupdict()
         result = {}
         extras = self._default_keys - frozenset(matchdict.keys())
-        for key, val in six.iteritems(matchdict):
+        for key, val in matchdict.items():
             if key != 'path_info' and self.encoding:
                 # change back into python unicode objects from the URL
                 # representation
diff --git a/routes/util.py b/routes/util.py
index 7a7d8bf..8e0d310 100644
--- a/routes/util.py
+++ b/routes/util.py
@@ -8,8 +8,7 @@ framework.
 import os
 import re
 
-import six
-from six.moves import urllib
+import urllib
 
 from routes import request_config
 
@@ -34,8 +33,8 @@ def _screenargs(kargs, mapper, environ, force_explicit=False):
     """
     # Coerce any unicode args with the encoding
     encoding = mapper.encoding
-    for key, val in six.iteritems(kargs):
-        if isinstance(val, six.text_type):
+    for key, val in kargs.items():
+        if isinstance(val, str):
             kargs[key] = val.encode(encoding)
 
     if mapper.explicit and mapper.sub_domains and not force_explicit:
@@ -60,7 +59,7 @@ def _screenargs(kargs, mapper, environ, force_explicit=False):
         memory_kargs = {}
 
     # Remove keys from memory and kargs if kargs has them as None
-    empty_keys = [key for key, value in six.iteritems(kargs) if value is None]
+    empty_keys = [key for key, value in kargs.items() if value is None]
     for key in empty_keys:
         del kargs[key]
         memory_kargs.pop(key, None)
@@ -79,7 +78,7 @@ def _subdomain_check(kargs, mapper, environ):
     on the current subdomain or lack therof."""
     if mapper.sub_domains:
         subdomain = kargs.pop('sub_domain', None)
-        if isinstance(subdomain, six.text_type):
+        if isinstance(subdomain, str):
             subdomain = str(subdomain)
 
         fullhost = environ.get('HTTP_HOST') or environ.get('SERVER_NAME')
@@ -112,13 +111,13 @@ def _subdomain_check(kargs, mapper, environ):
 def _url_quote(string, encoding):
     """A Unicode handling version of urllib.quote."""
     if encoding:
-        if isinstance(string, six.text_type):
+        if isinstance(string, str):
             s = string.encode(encoding)
-        elif isinstance(string, six.text_type):
+        elif isinstance(string, str):
             # assume the encoding is already correct
             s = string
         else:
-            s = six.text_type(string).encode(encoding)
+            s = str(string).encode(encoding)
     else:
         s = str(string)
     return urllib.parse.quote(s, '/')
@@ -126,13 +125,13 @@ def _url_quote(string, encoding):
 
 def _str_encode(string, encoding):
     if encoding:
-        if isinstance(string, six.text_type):
+        if isinstance(string, str):
             s = string.encode(encoding)
-        elif isinstance(string, six.text_type):
+        elif isinstance(string, str):
             # assume the encoding is already correct
             s = string
         else:
-            s = six.text_type(string).encode(encoding)
+            s = str(string).encode(encoding)
     return s
 
 
@@ -216,16 +215,16 @@ def url_for(*args, **kargs):
             if kargs:
                 url += '?'
                 query_args = []
-                for key, val in six.iteritems(kargs):
+                for key, val in kargs.items():
                     if isinstance(val, (list, tuple)):
                         for value in val:
                             query_args.append("%s=%s" % (
-                                urllib.parse.quote(six.text_type(key).encode(encoding)),
-                                urllib.parse.quote(six.text_type(value).encode(encoding))))
+                                urllib.parse.quote(str(key).encode(encoding)),
+                                urllib.parse.quote(str(value).encode(encoding))))
                     else:
                         query_args.append("%s=%s" % (
-                            urllib.parse.quote(six.text_type(key).encode(encoding)),
-                            urllib.parse.quote(six.text_type(val).encode(encoding))))
+                            urllib.parse.quote(str(key).encode(encoding)),
+                            urllib.parse.quote(str(val).encode(encoding))))
                 url += '&'.join(query_args)
     environ = getattr(config, 'environ', {})
     if 'wsgiorg.routing_args' not in environ:
@@ -366,16 +365,16 @@ class URLGenerator(object):
                 if kargs:
                     url += '?'
                     query_args = []
-                    for key, val in six.iteritems(kargs):
+                    for key, val in kargs.items():
                         if isinstance(val, (list, tuple)):
                             for value in val:
                                 query_args.append("%s=%s" % (
-                                    urllib.parse.quote(six.text_type(key).encode(encoding)),
-                                    urllib.parse.quote(six.text_type(value).encode(encoding))))
+                                    urllib.parse.quote(str(key).encode(encoding)),
+                                    urllib.parse.quote(str(value).encode(encoding))))
                         else:
                             query_args.append("%s=%s" % (
-                                urllib.parse.quote(six.text_type(key).encode(encoding)),
-                                urllib.parse.quote(six.text_type(val).encode(encoding))))
+                                urllib.parse.quote(str(key).encode(encoding)),
+                                urllib.parse.quote(str(val).encode(encoding))))
                     url += '&'.join(query_args)
         if not static:
             route_args = []
diff --git a/setup.py b/setup.py
index 71732bd..ac9f17a 100644
--- a/setup.py
+++ b/setup.py
@@ -64,9 +64,6 @@ setup(name="Routes",
       license="MIT",
       include_package_data=True,
       zip_safe=False,
-      install_requires=[
-          "six"
-      ],
       extras_require=extras_require,
       **extra_options
 )
diff --git a/tests/test_functional/test_generation.py b/tests/test_functional/test_generation.py
index ff9effd..8b0ad96 100644
--- a/tests/test_functional/test_generation.py
+++ b/tests/test_functional/test_generation.py
@@ -1,6 +1,6 @@
 """test_generation"""
 import sys, time, unittest
-from six.moves import urllib
+import urllib
 
 from routes import *
 
diff --git a/tests/test_functional/test_nonminimization.py b/tests/test_functional/test_nonminimization.py
index 744a879..cfddcd4 100644
--- a/tests/test_functional/test_nonminimization.py
+++ b/tests/test_functional/test_nonminimization.py
@@ -1,5 +1,5 @@
 """Test non-minimization recognition"""
-from six.moves import urllib
+import urllib
 
 from routes import url_for
 from routes.mapper import Mapper
diff --git a/tests/test_functional/test_recognition.py b/tests/test_functional/test_recognition.py
index cf60554..23b8f5c 100644
--- a/tests/test_functional/test_recognition.py
+++ b/tests/test_functional/test_recognition.py
@@ -4,7 +4,7 @@ import sys
 import time
 import unittest
 import pytest
-from six.moves import urllib
+import urllib
 from routes import *
 from routes.util import RoutesException
 
