From 33981fda8c319c9feafc958f5f0b1670c48666ef Mon Sep 17 00:00:00 2001
From: matthiakl <t-m.42@mailpost.spdns.org>
Date: Sat, 14 Aug 2021 14:33:30 +0200
Subject: [PATCH] Added missing direct incude

---
 src/network/bufferedconnection.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/network/bufferedconnection.h b/src/network/bufferedconnection.h
index fe974b9ea54..7e1ecf87e4d 100644
--- a/src/network/bufferedconnection.h
+++ b/src/network/bufferedconnection.h
@@ -20,6 +20,7 @@
 #ifndef WL_NETWORK_BUFFEREDCONNECTION_H
 #define WL_NETWORK_BUFFEREDCONNECTION_H
 
+#include <map>
 #include <memory>
 #include <mutex>
 #include <thread>
