diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7371b02..350ae47 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -382,8 +382,8 @@ if(NOT MSVC)
     message(WARNING "This compiler is known to cause false-positive warnings.")
   else()
     # Turn some warnings into errors.
-    message(STATUS "Warnings will be treated as errors.")
-    wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror")
+    # message(STATUS "Warnings will be treated as errors.")
+    # wl_add_flag(WL_COMPILE_DIAGNOSTICS "-Werror")
     if(("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 10.0))
       # Silence some false positives on older g++
       message(STATUS "Disabling known false-positive warnings for this compiler.")
