diff --git a/xdg/org.widelands.Widelands.desktop b/xdg/org.widelands.Widelands.desktop
index 670eaf7..90331f9 100644
--- a/xdg/org.widelands.Widelands.desktop
+++ b/xdg/org.widelands.Widelands.desktop
@@ -83,7 +83,8 @@ Comment[zh_CN]=即时战略游戏
 Comment[zh_TW]=一款即時演變的策略遊戲
 Icon=org.widelands.Widelands
 TryExec=widelands
-Exec=widelands
+# force glx, https://github.com/widelands/widelands/issues/4958
+Exec=env SDL_VIDEODRIVER=x11 widelands
 Categories=Game;StrategyGame;
 Keywords=Game;RTS;Real-time;Strategy;
 Terminal=false
diff --git a/xdg/org.widelands.Widelands.desktop.stub b/xdg/org.widelands.Widelands.desktop.stub
index 95ef67b..1aff536 100644
--- a/xdg/org.widelands.Widelands.desktop.stub
+++ b/xdg/org.widelands.Widelands.desktop.stub
@@ -4,7 +4,8 @@ Type=Application
 GENERIC_NAME_COMMENT_HOOK
 Icon=org.widelands.Widelands
 TryExec=widelands
-Exec=widelands
+# force glx, https://github.com/widelands/widelands/issues/4958
+Exec=env SDL_VIDEODRIVER=x11 widelands
 Categories=Game;StrategyGame;
 Keywords=Game;RTS;Real-time;Strategy;
 Terminal=false
