From 9c161bbccc3964847ec21c915ec396b40c024c5c Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Wed, 6 May 2026 06:21:45 +0000
Subject: [PATCH] statically build incus-agent and incus-migrate

---
 Makefile | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/Makefile b/Makefile
index ceb15e6..5978834 100644
--- a/Makefile
+++ b/Makefile
@@ -37,8 +37,8 @@ ifeq "$(TAG_SQLITE3)" ""
 endif
 
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" $(GO) install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
-	CGO_ENABLED=0 $(GO) install -v -tags netgo ./cmd/incus-migrate
-	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo ./cmd/incus-agent
+	CGO_ENABLED=0 $(GO) install -v -tags netgo,static -buildmode default ./cmd/incus-migrate
+	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo,static -buildmode default ./cmd/incus-agent
 	@echo "Incus built successfully"
 
 .PHONY: client
@@ -50,13 +50,13 @@ client:
 .PHONY: incus-agent
 # doc: Build the Incus agent
 incus-agent:
-	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo ./cmd/incus-agent
+	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo,static -buildmode default ./cmd/incus-agent
 	@echo "Incus agent built successfully"
 
 .PHONY: incus-migrate
 # doc: Build the Incus migration tool
 incus-migrate:
-	CGO_ENABLED=0 $(GO) install -v -tags netgo ./cmd/incus-migrate
+	CGO_ENABLED=0 $(GO) install -v -tags netgo,static -buildmode default ./cmd/incus-migrate
 	@echo "Incus migration tool built successfully"
 
 .PHONY: debug
@@ -68,8 +68,8 @@ ifeq "$(TAG_SQLITE3)" ""
 endif
 
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" $(GO) install -v -tags "$(TAG_SQLITE3) logdebug" $(DEBUG) ./...
-	CGO_ENABLED=0 $(GO) install -v -tags "netgo,logdebug" ./cmd/incus-migrate
-	CGO_ENABLED=0 $(GO) install -v -tags "agent,netgo,logdebug" ./cmd/incus-agent
+	CGO_ENABLED=0 $(GO) install -v -tags "netgo,logdebug,static" -buildmode default ./cmd/incus-migrate
+	CGO_ENABLED=0 $(GO) install -v -tags "agent,netgo,logdebug,static" -buildmode default ./cmd/incus-agent
 	@echo "Incus built successfully"
 
 .PHONY: nocache
@@ -81,8 +81,8 @@ ifeq "$(TAG_SQLITE3)" ""
 endif
 
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" $(GO) install -a -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
-	CGO_ENABLED=0 $(GO) install -a -v -tags netgo ./cmd/incus-migrate
-	CGO_ENABLED=0 $(GO) install -a -v -tags agent,netgo ./cmd/incus-agent
+	CGO_ENABLED=0 $(GO) install -a -v -tags netgo,static -buildmode default ./cmd/incus-migrate
+	CGO_ENABLED=0 $(GO) install -a -v -tags agent,netgo,static -buildmode default ./cmd/incus-agent
 	@echo "Incus built successfully"
 
 .PHONY: race
@@ -94,8 +94,8 @@ ifeq "$(TAG_SQLITE3)" ""
 endif
 
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" $(GO) install -race -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
-	CGO_ENABLED=0 $(GO) install -v -tags netgo ./cmd/incus-migrate
-	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo ./cmd/incus-agent
+	CGO_ENABLED=0 $(GO) install -v -tags netgo,static -buildmode default ./cmd/incus-migrate
+	CGO_ENABLED=0 $(GO) install -v -tags agent,netgo,static -buildmode default ./cmd/incus-agent
 	@echo "Incus built successfully"
 
 # section(Dependencies): Manage Incus dependencies
-- 
2.52.0

