--- old/rsync.c
+++ new/rsync.c
@@ -48,7 +48,7 @@
  * @param bufsize is the size of the destination buffer.
  *
  * @return index of the terminating byte.
- **/
+ * disabled
 static size_t strlcpy(char *d, const char *s, size_t bufsize)
 {
         size_t len = strlen(s);
@@ -61,7 +61,7 @@
         }
         return ret;
 }
-
+ **/
 /* splits filepath at the last '/', if any, like so:
  *	dirname		basename	filepath
  *	"/"		""		"/"
