From 809c779513531debbb00369e09b882296abdbc73 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Fri, 5 May 2017 06:11:19 +0000
Subject: [PATCH] remove bash dependency

---
 autogen.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/autogen.sh b/autogen.sh
index 665de9c..e5b866d 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 autoreconf --force --install -I m4
 rm -Rf autom4te.cache;
-- 
2.12.2

