From 60a0c678e1c1fce9a979cc18d66f25853e965181 Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Fri, 5 May 2017 06:14:22 +0000
Subject: [PATCH] fix includes

---
 utils/ndptool.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/utils/ndptool.c b/utils/ndptool.c
index 1d96f4c..96479fa 100644
--- a/utils/ndptool.c
+++ b/utils/ndptool.c
@@ -28,6 +28,7 @@
 #include <arpa/inet.h>
 #include <errno.h>
 #include <ndp.h>
+#include <sys/select.h>
 
 enum verbosity_level {
 	VERB1,
-- 
2.12.2

