Skip specs that require gem unicode/emoji (optional dependency).

--- a/spec/display_width_spec.rb
+++ b/spec/display_width_spec.rb
@@ -145,15 +145,15 @@
   end
 
   describe '[emoji]' do
-    it 'does not count modifiers and zjw sequences for valid emoji' do
+    xit 'does not count modifiers and zjw sequences for valid emoji' do
       expect( "🤾🏽‍♀️".display_width(1, {}, emoji: true) ).to eq 2
     end
 
-    it 'respects different ambiguous values' do
+    xit 'respects different ambiguous values' do
       expect( "🤾🏽‍♀️".display_width(2, {}, emoji: true) ).to eq 2
     end
 
-    it 'works with flags' do
+    xit 'works with flags' do
       expect( "🇵🇹".display_width(1, {}, emoji: true) ).to eq 2
     end
   end
