--- a/ax25ipd/io.c	2019-03-20 16:48:59.000000000 +0000
+++ b/ax25ipd/io.c	2023-09-11 21:10:13.156930299 +0000
@@ -19,7 +19,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
-#include <termio.h>
+#include <termios.h>
 #include <time.h>
 #include <unistd.h>
 #include <arpa/inet.h>
@@ -29,10 +29,11 @@
 #include <sys/types.h>
 #include <sys/time.h>
 #include <sys/socket.h>
+#include <bits/ioctl.h>
 
 #include "ax25ipd.h"
 
-static struct termio nterm;
+static struct termios nterm;
 
 int ttyfd = -1;
 static int udpsock = -1;
