--- a/lightdm.service
+++ b/lightdm.service
@@ -8,10 +8,7 @@ StartLimitBurst=5
 OnFailure=lightdm-failure-handler.service
 
 [Service]
-# temporary safety check until all DMs are converted to correct
-# display-manager.service symlink handling
-ExecStartPre=/bin/sh -c '[ "$(basename $(cat /etc/X11/default-display-manager 2>/dev/null))" = "lightdm" ]'
-ExecStart=/usr/sbin/lightdm
+ExecStart=/usr/bin/lightdm
 Restart=always
 RestartSec=5s
 BusName=org.freedesktop.DisplayManager
