From 6f99ef55a8bd75d99882078b50e3c7665106c514 Mon Sep 17 00:00:00 2001
From: Leah Neukirchen <leah@vuxu.org>
Date: Mon, 21 Aug 2017 19:40:00 +0200
Subject: [PATCH 1/2] mgenmid: printb36: fix off-by-one

Closes #73.
---
 mgenmid.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mgenmid.c b/mgenmid.c
index 36aa8c8..655df95 100644
--- a/mgenmid.c
+++ b/mgenmid.c
@@ -21,9 +21,10 @@ printb36(uint64_t x)
 
 	char outbuf[16];
 	char *o = outbuf + sizeof outbuf;
-	*o = 0;
 
+	*--o = 0;
 	do { *--o = base36[x % 36]; } while (x /= 36);
+
 	fputs(o, stdout);
 }
 
-- 
2.14.1

