Author: Milan P. Stanić <mps@arvanta.net>
Date:   Tue Dec 29 10:04:22 2020 +0000

dirty hack to fix backtrace_symbols missing

--- a/scribus/util_debug.cpp	2020-11-14 22:37:18.000000000 +0000
+++ b/scribus/util_debug.cpp	2020-12-22 13:50:51.014947500 +0000
@@ -59,8 +59,9 @@
 	char **messages = ( char ** ) nullptr;
 	int i, trace_size = 0;
 
-	trace_size = backtrace ( trace, nFrames + 1 );
-	messages = backtrace_symbols ( trace, trace_size );
+	// trace_size = backtrace ( trace, nFrames + 1 );
+	// messages = backtrace_symbols ( trace, trace_size );
+	messages = 0;
 	if ( messages )
 	{
 		for ( i=1; i < trace_size; ++i )
