diff --git a/cranelift/Cargo.toml b/cranelift/Cargo.toml
index 8550082f6..1237a0c9f 100644
--- a/cranelift/Cargo.toml
+++ b/cranelift/Cargo.toml
@@ -65,4 +65,4 @@ disas = ["capstone"]
 souper-harvest = ["cranelift-codegen/souper-harvest", "rayon"]
 all-arch = ["cranelift-codegen/all-arch"]
 all-native-arch = ["cranelift-codegen/all-native-arch"]
-pulley = ['cranelift-codegen/pulley', 'dep:pulley-interpreter']
+pulley = ['cranelift-codegen/pulley', 'pulley-interpreter']
diff --git a/cranelift/codegen/Cargo.toml b/cranelift/codegen/Cargo.toml
index c06f4953c..36218a119 100644
--- a/cranelift/codegen/Cargo.toml
+++ b/cranelift/codegen/Cargo.toml
@@ -90,7 +90,7 @@ arm64 = []
 s390x = []
 riscv64 = []
 pulley = [
-    "dep:pulley-interpreter",
+    "pulley-interpreter",
     "pulley-interpreter/encode",
     "pulley-interpreter/disas",
     "cranelift-codegen-meta/pulley",
diff --git a/cranelift/codegen/meta/Cargo.toml b/cranelift/codegen/meta/Cargo.toml
index 0cc134a94..bb273c829 100644
--- a/cranelift/codegen/meta/Cargo.toml
+++ b/cranelift/codegen/meta/Cargo.toml
@@ -23,4 +23,4 @@ pulley-interpreter = { workspace = true, optional = true }
 heck = "0.5.0"
 
 [features]
-pulley = ['dep:pulley-interpreter']
+pulley = ['pulley-interpreter']
diff --git a/crates/cranelift/Cargo.toml b/crates/cranelift/Cargo.toml
index c3c1a95ef..c939a995d 100644
--- a/crates/cranelift/Cargo.toml
+++ b/crates/cranelift/Cargo.toml
@@ -39,7 +39,7 @@ wasmtime-unwinder = { workspace = true, features = ["cranelift"] }
 [features]
 all-arch = ["cranelift-codegen/all-arch"]
 host-arch = ["cranelift-codegen/host-arch"]
-pulley = ["cranelift-codegen/pulley", "dep:pulley-interpreter"]
+pulley = ["cranelift-codegen/pulley", "pulley-interpreter"]
 trace-log = ["cranelift-codegen/trace-log"]
 component-model = ["wasmtime-environ/component-model"]
 incremental-cache = ["cranelift-codegen/incremental-cache"]
