diff --git a/cranelift/Cargo.toml b/cranelift/Cargo.toml
index 8550082f6..1237a0c9f 100644
--- a/cranelift/Cargo.toml
+++ b/cranelift/Cargo.toml
@@ -65,4 +65,4 @@ disas = ["capstone"]
 souper-harvest = ["cranelift-codegen/souper-harvest", "rayon"]
 all-arch = ["cranelift-codegen/all-arch"]
 all-native-arch = ["cranelift-codegen/all-native-arch"]
-pulley = ['cranelift-codegen/pulley', 'dep:pulley-interpreter']
+pulley = ['cranelift-codegen/pulley', 'pulley-interpreter']
diff --git a/cranelift/codegen/Cargo.toml b/cranelift/codegen/Cargo.toml
index 6941b9a11..fbb3e563c 100644
--- a/cranelift/codegen/Cargo.toml
+++ b/cranelift/codegen/Cargo.toml
@@ -88,7 +88,7 @@ arm64 = []
 s390x = []
 riscv64 = []
 pulley = [
-    "dep:pulley-interpreter",
+    "pulley-interpreter",
     "pulley-interpreter/encode",
     "pulley-interpreter/disas",
     "cranelift-codegen-meta/pulley",
diff --git a/cranelift/codegen/meta/Cargo.toml b/cranelift/codegen/meta/Cargo.toml
index 669c8ef28..b709eef6b 100644
--- a/cranelift/codegen/meta/Cargo.toml
+++ b/cranelift/codegen/meta/Cargo.toml
@@ -22,4 +22,4 @@ cranelift-codegen-shared = { path = "../shared", version = "0.121.0" }
 pulley-interpreter = { workspace = true, optional = true }
 
 [features]
-pulley = ['dep:pulley-interpreter']
+pulley = ['pulley-interpreter']
diff --git a/crates/cranelift/Cargo.toml b/crates/cranelift/Cargo.toml
index c4ef5cbd2..113e1e7ae 100644
--- a/crates/cranelift/Cargo.toml
+++ b/crates/cranelift/Cargo.toml
@@ -37,7 +37,7 @@ pulley-interpreter = { workspace = true, optional = true }
 [features]
 all-arch = ["cranelift-codegen/all-arch"]
 host-arch = ["cranelift-codegen/host-arch"]
-pulley = ["cranelift-codegen/pulley", "dep:pulley-interpreter"]
+pulley = ["cranelift-codegen/pulley", "pulley-interpreter"]
 trace-log = ["cranelift-codegen/trace-log"]
 component-model = ["wasmtime-environ/component-model"]
 incremental-cache = ["cranelift-codegen/incremental-cache"]
