diff --git a/crates/cache/Cargo.toml b/crates/cache/Cargo.toml
index bd6b571..442ac5c 100644
--- a/crates/cache/Cargo.toml
+++ b/crates/cache/Cargo.toml
@@ -21,7 +21,7 @@ serde = { workspace = true }
 serde_derive = { workspace = true }
 sha2 = "0.10.2"
 toml = { workspace = true }
-zstd = { version = "0.13.0", default-features = false }
+zstd = { version = "0.13.0", default-features = false, features = ["pkg-config"] }
 
 [target.'cfg(target_os = "windows")'.dependencies.windows-sys]
 workspace = true
