--- a/rspec-expectations.gemspec
+++ b/rspec-expectations.gemspec
@@ -13,8 +13,7 @@
   s.summary     = "rspec-expectations-#{RSpec::Expectations::Version::STRING}"
   s.description = "rspec-expectations provides a simple, readable API to express expected outcomes of a code example."
 
-  s.files            = `git ls-files -- lib/*`.split("\n")
-  s.files           += %w[README.md LICENSE.md Changelog.md .yardopts .document]
+  s.files            = Dir["lib/**/*"]
   s.test_files       = []
   s.rdoc_options     = ["--charset=UTF-8"]
   s.require_path     = "lib"
