diff -ru alsa-lib-1.1.0.orig/aserver/aserver.c alsa-lib-1.1.0/aserver/aserver.c
--- alsa-lib-1.1.0.orig/aserver/aserver.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/aserver/aserver.c	2016-02-11 14:31:58.800479098 +0200
@@ -20,7 +20,7 @@
 
 #include <sys/shm.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/un.h>
 #include <sys/uio.h>
 #include <stdio.h>
diff -ru alsa-lib-1.1.0.orig/include/asoundlib-head.h alsa-lib-1.1.0/include/asoundlib-head.h
--- alsa-lib-1.1.0.orig/include/asoundlib-head.h	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/include/asoundlib-head.h	2016-02-11 14:31:58.800479098 +0200
@@ -35,6 +35,6 @@
 #include <string.h>
 #include <fcntl.h>
 #include <assert.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <errno.h>
 #include <stdarg.h>
diff -ru alsa-lib-1.1.0.orig/include/asoundlib.h alsa-lib-1.1.0/include/asoundlib.h
--- alsa-lib-1.1.0.orig/include/asoundlib.h	2015-11-09 09:40:19.000000000 +0200
+++ alsa-lib-1.1.0/include/asoundlib.h	2016-02-11 14:31:58.800479098 +0200
@@ -35,7 +35,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <assert.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <errno.h>
 #include <stdarg.h>
 #include <endian.h>
diff -ru alsa-lib-1.1.0.orig/include/local.h alsa-lib-1.1.0/include/local.h
--- alsa-lib-1.1.0.orig/include/local.h	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/include/local.h	2016-02-11 14:31:58.797145726 +0200
@@ -47,7 +47,7 @@
 #error Header defining endianness not defined
 #endif
 #include <stdarg.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/types.h>
 #include <errno.h>
 #include <linux/types.h>
diff -ru alsa-lib-1.1.0.orig/src/control/control.c alsa-lib-1.1.0/src/control/control.c
--- alsa-lib-1.1.0.orig/src/control/control.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/control/control.c	2016-02-11 14:31:58.807145843 +0200
@@ -48,7 +48,7 @@
 #include <string.h>
 #include <fcntl.h>
 #include <signal.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include "control_local.h"
 
 /**
diff -ru alsa-lib-1.1.0.orig/src/control/control_shm.c alsa-lib-1.1.0/src/control/control_shm.c
--- alsa-lib-1.1.0.orig/src/control/control_shm.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/control/control_shm.c	2016-02-11 14:31:58.800479098 +0200
@@ -27,7 +27,7 @@
 #include <fcntl.h>
 #include <sys/shm.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/un.h>
 #include <sys/uio.h>
 #include <sys/mman.h>
diff -ru alsa-lib-1.1.0.orig/src/pcm/pcm.c alsa-lib-1.1.0/src/pcm/pcm.c
--- alsa-lib-1.1.0.orig/src/pcm/pcm.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/pcm/pcm.c	2016-02-11 14:31:58.833812820 +0200
@@ -634,7 +634,7 @@
 #include <stdarg.h>
 #include <signal.h>
 #include <ctype.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/mman.h>
 #include <limits.h>
 #include "pcm_local.h"
diff -ru alsa-lib-1.1.0.orig/src/pcm/pcm_direct.c alsa-lib-1.1.0/src/pcm/pcm_direct.c
--- alsa-lib-1.1.0.orig/src/pcm/pcm_direct.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/pcm/pcm_direct.c	2016-02-11 14:31:58.837146193 +0200
@@ -30,7 +30,7 @@
 #include <grp.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/shm.h>
 #include <sys/sem.h>
 #include <sys/wait.h>
diff -ru alsa-lib-1.1.0.orig/src/pcm/pcm_mmap.c alsa-lib-1.1.0/src/pcm/pcm_mmap.c
--- alsa-lib-1.1.0.orig/src/pcm/pcm_mmap.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/pcm/pcm_mmap.c	2016-02-11 14:31:58.817145959 +0200
@@ -22,7 +22,7 @@
 #include <stdio.h>
 #include <malloc.h>
 #include <string.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/mman.h>
 #ifdef HAVE_SYS_SHM_H
 #include <sys/shm.h>
diff -ru alsa-lib-1.1.0.orig/src/pcm/pcm_share.c alsa-lib-1.1.0/src/pcm/pcm_share.c
--- alsa-lib-1.1.0.orig/src/pcm/pcm_share.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/pcm/pcm_share.c	2016-02-11 14:31:58.820479332 +0200
@@ -34,7 +34,7 @@
 #include <signal.h>
 #include <math.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <pthread.h>
 #include "pcm_local.h"
 
diff -ru alsa-lib-1.1.0.orig/src/pcm/pcm_shm.c alsa-lib-1.1.0/src/pcm/pcm_shm.c
--- alsa-lib-1.1.0.orig/src/pcm/pcm_shm.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/pcm/pcm_shm.c	2016-02-11 14:31:58.820479332 +0200
@@ -36,7 +36,7 @@
 #include <sys/ioctl.h>
 #include <sys/shm.h>
 #include <sys/socket.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/un.h>
 #include <sys/mman.h>
 #include <netinet/in.h>
diff -ru alsa-lib-1.1.0.orig/src/seq/seq.c alsa-lib-1.1.0/src/seq/seq.c
--- alsa-lib-1.1.0.orig/src/seq/seq.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/seq/seq.c	2016-02-11 14:31:58.813812587 +0200
@@ -777,7 +777,7 @@
 
 */
 
-#include <sys/poll.h>
+#include <poll.h>
 #include "seq_local.h"
 
 /****************************************************************************
diff -ru alsa-lib-1.1.0.orig/src/shmarea.c alsa-lib-1.1.0/src/shmarea.c
--- alsa-lib-1.1.0.orig/src/shmarea.c	2015-11-09 09:39:18.000000000 +0200
+++ alsa-lib-1.1.0/src/shmarea.c	2016-02-11 14:31:58.807145843 +0200
@@ -27,7 +27,7 @@
 #include <malloc.h>
 #include <string.h>
 #include <errno.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/mman.h>
 #include <sys/shm.h>
 #include "list.h"
