--- a/test/Makefile.am	2024-11-12 10:36:52.000000000 +0100
+++ b/test/Makefile.am	2024-12-26 11:10:26.182941258 +0100
@@ -2,7 +2,7 @@
 
 check_PROGRAMS=control pcm pcm_min latency seq seq-ump-example \
 	       playmidi1 timer rawmidi midiloop umpinfo \
-	       oldapi queue_timer namehint client_event_filter \
+	       queue_timer namehint client_event_filter \
 	       chmap audio_time user-ctl-element-set pcm-multi-thread
 
 control_LDADD=../src/libasound.la
@@ -18,7 +18,6 @@
 rawmidi_LDADD=../src/libasound.la
 midiloop_LDADD=../src/libasound.la
 umpinfo_LDADD=../src/libasound.la
-oldapi_LDADD=../src/libasound.la
 queue_timer_LDADD=../src/libasound.la
 namehint_LDADD=../src/libasound.la
 client_event_filter_LDADD=../src/libasound.la
--- a/test/Makefile.in	2024-11-12 10:46:14.000000000 +0100
+++ b/test/Makefile.in	2024-12-26 11:08:38.629326010 +0100
@@ -90,7 +90,7 @@
 check_PROGRAMS = control$(EXEEXT) pcm$(EXEEXT) pcm_min$(EXEEXT) \
 	latency$(EXEEXT) seq$(EXEEXT) seq-ump-example$(EXEEXT) \
 	playmidi1$(EXEEXT) timer$(EXEEXT) rawmidi$(EXEEXT) \
-	midiloop$(EXEEXT) umpinfo$(EXEEXT) oldapi$(EXEEXT) \
+	midiloop$(EXEEXT) umpinfo$(EXEEXT) \
 	queue_timer$(EXEEXT) namehint$(EXEEXT) \
 	client_event_filter$(EXEEXT) chmap$(EXEEXT) \
 	audio_time$(EXEEXT) user-ctl-element-set$(EXEEXT) \
@@ -137,9 +137,6 @@
 namehint_SOURCES = namehint.c
 namehint_OBJECTS = namehint.$(OBJEXT)
 namehint_DEPENDENCIES = ../src/libasound.la
-oldapi_SOURCES = oldapi.c
-oldapi_OBJECTS = oldapi.$(OBJEXT)
-oldapi_DEPENDENCIES = ../src/libasound.la
 pcm_SOURCES = pcm.c
 pcm_OBJECTS = pcm.$(OBJEXT)
 pcm_DEPENDENCIES = ../src/libasound.la
@@ -203,7 +200,7 @@
 am__depfiles_remade = ./$(DEPDIR)/audio_time.Po ./$(DEPDIR)/chmap.Po \
 	./$(DEPDIR)/client_event_filter.Po ./$(DEPDIR)/control.Po \
 	./$(DEPDIR)/latency.Po ./$(DEPDIR)/midiloop.Po \
-	./$(DEPDIR)/namehint.Po ./$(DEPDIR)/oldapi.Po \
+	./$(DEPDIR)/namehint.Po \
 	./$(DEPDIR)/pcm-multi-thread.Po ./$(DEPDIR)/pcm.Po \
 	./$(DEPDIR)/pcm_min.Po ./$(DEPDIR)/playmidi1.Po \
 	./$(DEPDIR)/queue_timer.Po ./$(DEPDIR)/rawmidi.Po \
@@ -230,12 +227,12 @@
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
 SOURCES = audio_time.c chmap.c client_event_filter.c control.c \
-	latency.c midiloop.c namehint.c oldapi.c pcm.c \
+	latency.c midiloop.c namehint.c pcm.c \
 	pcm-multi-thread.c pcm_min.c playmidi1.c queue_timer.c \
 	rawmidi.c seq.c seq-ump-example.c timer.c umpinfo.c \
 	user-ctl-element-set.c
 DIST_SOURCES = audio_time.c chmap.c client_event_filter.c control.c \
-	latency.c midiloop.c namehint.c oldapi.c pcm.c \
+	latency.c midiloop.c namehint.c pcm.c \
 	pcm-multi-thread.c pcm_min.c playmidi1.c queue_timer.c \
 	rawmidi.c seq.c seq-ump-example.c timer.c umpinfo.c \
 	user-ctl-element-set.c
@@ -452,7 +449,6 @@
 rawmidi_LDADD = ../src/libasound.la
 midiloop_LDADD = ../src/libasound.la
 umpinfo_LDADD = ../src/libasound.la
-oldapi_LDADD = ../src/libasound.la
 queue_timer_LDADD = ../src/libasound.la
 namehint_LDADD = ../src/libasound.la
 client_event_filter_LDADD = ../src/libasound.la
@@ -537,10 +533,6 @@
 	@rm -f namehint$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(namehint_OBJECTS) $(namehint_LDADD) $(LIBS)
 
-oldapi$(EXEEXT): $(oldapi_OBJECTS) $(oldapi_DEPENDENCIES) $(EXTRA_oldapi_DEPENDENCIES) 
-	@rm -f oldapi$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(oldapi_OBJECTS) $(oldapi_LDADD) $(LIBS)
-
 pcm$(EXEEXT): $(pcm_OBJECTS) $(pcm_DEPENDENCIES) $(EXTRA_pcm_DEPENDENCIES) 
 	@rm -f pcm$(EXEEXT)
 	$(AM_V_CCLD)$(pcm_LINK) $(pcm_OBJECTS) $(pcm_LDADD) $(LIBS)
@@ -598,7 +590,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/latency.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/midiloop.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/namehint.Po@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/oldapi.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pcm-multi-thread.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pcm.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/pcm_min.Po@am__quote@ # am--include-marker
@@ -864,7 +855,6 @@
 	-rm -f ./$(DEPDIR)/latency.Po
 	-rm -f ./$(DEPDIR)/midiloop.Po
 	-rm -f ./$(DEPDIR)/namehint.Po
-	-rm -f ./$(DEPDIR)/oldapi.Po
 	-rm -f ./$(DEPDIR)/pcm-multi-thread.Po
 	-rm -f ./$(DEPDIR)/pcm.Po
 	-rm -f ./$(DEPDIR)/pcm_min.Po
@@ -928,7 +918,6 @@
 	-rm -f ./$(DEPDIR)/latency.Po
 	-rm -f ./$(DEPDIR)/midiloop.Po
 	-rm -f ./$(DEPDIR)/namehint.Po
-	-rm -f ./$(DEPDIR)/oldapi.Po
 	-rm -f ./$(DEPDIR)/pcm-multi-thread.Po
 	-rm -f ./$(DEPDIR)/pcm.Po
 	-rm -f ./$(DEPDIR)/pcm_min.Po
