diff --git a/pyfirmata/pyfirmata.py b/pyfirmata/pyfirmata.py
index 045efc4..06b0c9d 100755
--- a/pyfirmata/pyfirmata.py
+++ b/pyfirmata/pyfirmata.py
@@ -182,7 +182,7 @@ class Board(object):
 
     def add_cmd_handler(self, cmd, func):
         """Adds a command handler for a command."""
-        len_args = len(inspect.getargspec(func)[0])
+        len_args = len(inspect.getfullargspec(func)[0])
 
         def add_meta(f):
             def decorator(*args, **kwargs):
