diff --git a/modules/gpgsqlbackend/4.1.10_to_4.1.11.schema.pgsql.sql b/modules/gpgsqlbackend/4.1.10_to_4.1.11.schema.pgsql.sql
new file mode 100644
index 0000000..b0c2ee1
--- /dev/null
+++ b/modules/gpgsqlbackend/4.1.10_to_4.1.11.schema.pgsql.sql
@@ -0,0 +1 @@
+ALTER TABLE domains ALTER notified_serial TYPE bigint USING CASE WHEN notified_serial >= 0 THEN notified_serial::bigint END;
diff --git a/modules/gpgsqlbackend/Makefile.am b/modules/gpgsqlbackend/Makefile.am
index 8a820d5..9e2f271 100644
--- a/modules/gpgsqlbackend/Makefile.am
+++ b/modules/gpgsqlbackend/Makefile.am
@@ -12,6 +12,7 @@ dist_doc_DATA = \
 	schema.pgsql.sql \
 	nodnssec-3.x_to_3.4.0_schema.pgsql.sql \
 	dnssec-3.x_to_3.4.0_schema.pgsql.sql \
+	4.1.10_to_4.1.11.schema.pgsql.sql \
 	3.4.0_to_4.1.0_schema.pgsql.sql
 
 libgpgsqlbackend_la_SOURCES = \
diff --git a/modules/gpgsqlbackend/Makefile.in b/modules/gpgsqlbackend/Makefile.in
index 4c1f978..9793c9d 100644
--- a/modules/gpgsqlbackend/Makefile.in
+++ b/modules/gpgsqlbackend/Makefile.in
@@ -479,6 +479,7 @@ dist_doc_DATA = \
 	schema.pgsql.sql \
 	nodnssec-3.x_to_3.4.0_schema.pgsql.sql \
 	dnssec-3.x_to_3.4.0_schema.pgsql.sql \
+	4.1.10_to_4.1.11.schema.pgsql.sql \
 	3.4.0_to_4.1.0_schema.pgsql.sql
 
 libgpgsqlbackend_la_SOURCES = \
