diff --git a/pdns/opensslsigners.cc b/pdns/opensslsigners.cc
index 8e3bb54..459939a 100644
--- a/pdns/opensslsigners.cc
+++ b/pdns/opensslsigners.cc
@@ -33,7 +33,7 @@
 #include "opensslsigners.hh"
 #include "dnssecinfra.hh"
 
-#if (OPENSSL_VERSION_NUMBER < 0x1010000fL || defined LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER < 0x1010000fL)
 /* OpenSSL < 1.1.0 needs support for threading/locking in the calling application. */
 static pthread_mutex_t *openssllocks;
 
