Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: Patch configure script for python3
----

--- a/configure
+++ b/configure
@@ -19209,8 +19209,9 @@ $as_echo "$python_path" >&6; }
         # Check for Python library path
         { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Python library path" >&5
 $as_echo_n "checking for Python library path... " >&6; }
+        PYTHON_MAJOR=`echo $PYTHON_VERSION | cut -d. -f1`
         for i in "$base_python_path/lib/python$PYTHON_VERSION/config/" "$base_python_path/lib/python$PYTHON_VERSION/" "$base_python_path/lib/python/config/" "$base_python_path/lib/python/" "$base_python_path/" "$base_python_path/libs/" ; do
-                python_path=`find $i -name libpython$PYTHON_VERSION.* -print 2> /dev/null | sed "1q"`
+                python_path=`find $i -name libpython$PYTHON_MAJOR.* -print 2> /dev/null | sed "1q"`
                 if test -n "$python_path" ; then
                         break
                 fi
@@ -19221,7 +19222,7 @@ $as_echo "$python_path" >&6; }
         if test -z "$python_path" ; then
                 as_fn_error $? "cannot find Python library path" "$LINENO" 5
         fi
-        PYTHON_LDFLAGS="-L$python_path -lpython$PYTHON_VERSION"
+        PYTHON_LDFLAGS="-L$python_path -lpython$PYTHON_MAJOR"

         #
         python_site=`echo $base_python_path | sed "s/config/site-packages/"`
@@ -19234,7 +19235,7 @@ $as_echo "$python_path" >&6; }
 $as_echo_n "checking python extra libraries... " >&6; }
         PYTHON_EXTRA_LIBS=`$PYTHON -c "import distutils.sysconfig; \
                 conf = distutils.sysconfig.get_config_var; \
-                print (conf('LOCALMODLIBS') or '')+' '+(conf('LIBS') or '')"
+                print ((conf('LOCALMODLIBS') or '')+' '+(conf('LIBS') or ''))"
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: $PYTHON_EXTRA_LIBS" >&5
 $as_echo "$PYTHON_EXTRA_LIBS" >&6; }`

