diff --git a/tests/for_pytest_plugin/test_two.py b/tests/for_pytest_plugin/test_two.py
index 4f24a43..acd1668 100644
--- a/tests/for_pytest_plugin/test_two.py
+++ b/tests/for_pytest_plugin/test_two.py
@@ -27,6 +27,7 @@ describe "one":
         def collector(self):
             return 3
 
+        @pytest.mark.asyncio
         async it "seven", register, collector:
             assert register == 2
             assert collector == 3
@@ -37,6 +38,7 @@ describe "one":
         def collector(self):
             return 4
 
+        @pytest.mark.asyncio
         async it "nine", register, collector:
             assert register == 2
             assert collector == 4
