From 4b1c4f932a38557fbb0adeab9f8d28ec672216ca Mon Sep 17 00:00:00 2001
From: ptrcnull <git@ptrcnull.me>
Date: Tue, 15 Mar 2022 00:14:13 +0100
Subject: [PATCH] tls/acme: set default hostname

---
 internal/tls/acme/acme.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/internal/tls/acme/acme.go b/internal/tls/acme/acme.go
index 556e6372..65f9e7c3 100644
--- a/internal/tls/acme/acme.go
+++ b/internal/tls/acme/acme.go
@@ -90,6 +90,7 @@ func (l *Loader) Init(cfg *config.Map) error {
 	l.cfg = certmagic.New(l.cache, certmagic.Config{
 		Storage: l.store, // not sure if it is necessary to set these twice
 		Logger:  cmLog,
+		DefaultServerName: hostname,
 	})
 	mngr := certmagic.NewACMEManager(l.cfg, certmagic.ACMEManager{
 		Logger: cmLog,
