--- a/RegEdit.c
+++ b/RegEdit.c
@@ -34,19 +34,19 @@
 
 
 /* static forward. move from global in the original Editres code */
-static void _XEditResCheckMessages();
-static void _XEditResPutString8();
-static void _XEditResPut8();
-static void _XEditResPut16();
-static void _XEditResPut32();
-static void _XEditResPutWidgetInfo();
-static void _XEditResResetStream();
-static Boolean _XEditResGet8();
-static Boolean _XEditResGet16();
-static Boolean _XEditResGetSigned16();
-static Boolean _XEditResGet32();
-static Boolean _XEditResGetString8(); 
-static Boolean _XEditResGetWidgetInfo();
+static void _XEditResCheckMessages(Widget w, XtPointer data, XEvent *event, Boolean *cont);
+static void _XEditResPutString8(ProtocolStream * stream,  char * val);
+static void _XEditResPut8(ProtocolStream * stream, unsigned int val);
+static void _XEditResPut16(ProtocolStream * stream, unsigned int val);
+static void _XEditResPut32(ProtocolStream * stream, unsigned int val);
+static void _XEditResPutWidgetInfo(ProtocolStream * stream, WidgetInfo * info);
+static void _XEditResResetStream(ProtocolStream*);
+static Boolean _XEditResGet8(ProtocolStream * stream, unsigned char * val);
+static Boolean _XEditResGet16(ProtocolStream * stream, unsigned short * val);
+static Boolean _XEditResGetSigned16(ProtocolStream * stream, short * val);
+static Boolean _XEditResGet32(ProtocolStream * stream, unsigned long * val);
+static Boolean _XEditResGetString8(ProtocolStream * stream,  char ** val); 
+static Boolean _XEditResGetWidgetInfo(ProtocolStream * stream, WidgetInfo * info);
 
 /* the only entry point here */
 void 
@@ -140,16 +140,24 @@
 
 static Globals globals;
 
-static void SendFailure(), SendCommand(), InsertWidget(), ExecuteCommand();
-static void FreeEvent(), ExecuteSetValues(), ExecuteGetGeometry();
-static void ExecuteGetResources();
+static void SendFailure(Widget w, Atom sel, ResIdent ident, char * str);
+static void SendCommand(Widget w, Atom sel, ResIdent ident, EditresCommand command, ProtocolStream * stream);
+static void InsertWidget(ProtocolStream * stream, Widget w);
+static void ExecuteCommand(Widget w, Atom sel, ResIdent ident, EditresEvent * event);
+static void FreeEvent(EditresEvent * event);
+static void ExecuteSetValues(Widget w, SetValuesEvent * sv_event, WidgetInfo * entry, ProtocolStream * stream, unsigned short * count);
+static void ExecuteGetGeometry(Widget w, ProtocolStream * stream);
+static void ExecuteGetResources(Widget w, ProtocolStream * stream);
 
-static void GetCommand();
-static void LoadResources();
-static Boolean IsChild();
-static void DumpChildren();
-static char *DumpWidgets(), *DoSetValues(), *DoFindChild();
-static char *DoGetGeometry(), *DoGetResources();
+static void GetCommand(Widget w, XtPointer data, Atom *selection, Atom *type, XtPointer value, unsigned long *length, int * format);
+static void LoadResources(Widget w);
+static Boolean IsChild(Widget top, Widget parent, Widget child);
+static void DumpChildren(Widget w, ProtocolStream * stream, unsigned short *count);
+static char *DumpWidgets(Widget w, EditresEvent * event, ProtocolStream * stream);
+static char *DoSetValues(Widget w, EditresEvent * event, ProtocolStream * stream);
+static char *DoFindChild(Widget w, EditresEvent * event, ProtocolStream * stream);
+static char *DoGetGeometry(Widget w, EditresEvent * event, ProtocolStream * stream);
+static char *DoGetResources(Widget w, EditresEvent * event, ProtocolStream * stream);
 
 /************************************************************
  *
@@ -439,7 +447,7 @@
 ResIdent ident;
 EditresEvent * event;
 {
-    char * (*func)();
+    char * (*func)(Widget w, EditresEvent * event, ProtocolStream * stream);
     char * str;
 
     if (globals.block == BlockAll) {
