diff --git a/GNUmakefile b/GNUmakefile
index c17ece6..beec01c 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -15,7 +15,7 @@ PKG_CONFIG = pkg-config
 PKGS_IDA := libexif libpng libtiff-4 pixman-1
 PKGS_FBI := freetype2 fontconfig libdrm libexif libpng libtiff-4 pixman-1
 PKGS_FBPDF := libdrm poppler-glib gbm egl epoxy pixman-1
-HAVE_DEPS := $(shell $(PKG_CONFIG) $(PKGS_FBI) $(PKGS_FBPDF) && echo yes)
+HAVE_DEPS := $(shell $(PKG_CONFIG) $(PKGS_FBI) && echo yes)
 
 # map pkg-config names to debian packages using apt-file
 APT_REGEX = /($(shell echo $(PKGS_FBI) $(PKGS_FBPDF) | sed -e 's/ /|/g')).pc
@@ -46,7 +46,7 @@ all: build
 # what to build
 TARGETS := exiftran thumbnail.cgi
 ifeq ($(HAVE_LINUX_FB_H),yes)
-  TARGETS += fbi fbpdf kbdtest
+  TARGETS += fbi kbdtest
 endif
 ifeq ($(HAVE_MOTIF),yes)
   TARGETS += ida
@@ -250,7 +250,6 @@ install: build
 ifeq ($(HAVE_LINUX_FB_H),yes)
 	$(INSTALL_BINARY) fbi $(bindir)
 	$(INSTALL_SCRIPT) fbgs $(bindir)
-	$(INSTALL_SCRIPT) fbpdf $(bindir)
 	$(INSTALL_DATA) $(srcdir)/man/fbi.1 $(mandir)/man1
 	$(INSTALL_DATA) $(srcdir)/man/fbgs.1 $(mandir)/man1
 endif

