--- a/GNUmakefile	2016-04-05 12:59:50.000000000 +0300
+++ b/GNUmakefile	2016-06-10 10:31:08.242069624 +0300
@@ -15,7 +15,7 @@
 PKGS_IDA := libexif
 PKGS_FBI := freetype2 fontconfig libdrm libexif
 PKGS_FBPDF := libdrm poppler-glib gbm epoxy cairo-gl
-HAVE_DEPS := $(shell $(PKG_CONFIG) $(PKGS_FBI) $(PKGS_FBPDF) && echo yes)
+HAVE_DEPS := $(shell $(PKG_CONFIG) $(PKGS_FBI) && echo yes)
 
 ifeq ($(HAVE_LINUX_FB_H),yes)
 ifneq ($(HAVE_DEPS),yes)
@@ -39,7 +39,7 @@
 # what to build
 TARGETS := exiftran thumbnail.cgi
 ifeq ($(HAVE_LINUX_FB_H),yes)
-  TARGETS += fbi fbpdf kbdtest
+  TARGETS += fbi kbdtest
 endif
 ifeq ($(HAVE_MOTIF),yes)
   TARGETS += ida
@@ -250,7 +250,7 @@
 ifeq ($(HAVE_LINUX_FB_H),yes)
 	$(INSTALL_BINARY) fbi $(bindir)
 	$(INSTALL_SCRIPT) fbgs $(bindir)
-	$(INSTALL_SCRIPT) fbpdf $(bindir)
+	#$(INSTALL_SCRIPT) fbpdf $(bindir)
 	$(INSTALL_DATA) $(srcdir)/fbi.man $(mandir)/man1/fbi.1
 	$(INSTALL_DATA) $(srcdir)/fbgs.man $(mandir)/man1/fbgs.1
 endif
