Add suport for building libwebsockets-4.1

diff --git a/PLUGINS/src/osd2web/websock.c b/PLUGINS/src/osd2web/websock.c
index 5df6975..8e5a7c3 100644
--- a/PLUGINS/src/osd2web/websock.c
+++ b/PLUGINS/src/osd2web/websock.c
@@ -79,7 +79,9 @@ int cWebSock::init(int aPort, int aTimeout)
    info.protocols = protocols;
    info.ssl_cert_filepath = certPath;
    info.ssl_private_key_filepath = keyPath;
+#if LWS_LIBRARY_VERSION_NUMBER >= 2001000 && LWS_LIBRARY_VERSION_NUMBER < 4000000
    info.ws_ping_pong_interval = timeout;
+#endif
    info.gid = -1;
    info.uid = -1;
    info.options = 0;
