--- a/PLUGINS/src/streamdev/Makefile
+++ b/PLUGINS/src/streamdev/Makefile
@@ -37,8 +37,9 @@
 -include $(PLGCFG)
 
 ### export all vars for sub-makes, using absolute paths
-LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
-LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
+# paths don't exist and absolute path isn't needed
+#LIBDIR := $(shell cd $(LIBDIR) >/dev/null 2>&1 && pwd)
+#LOCDIR := $(shell cd $(LOCDIR) >/dev/null 2>&1 && pwd)
 export
 unexport PLUGIN
 
