diff --color -rupN a/okonomiyaki/platforms/_arch.py b/okonomiyaki/platforms/_arch.py
--- a/okonomiyaki/platforms/_arch.py	2023-06-29 19:20:58.000000000 +0200
+++ b/okonomiyaki/platforms/_arch.py	2024-07-06 14:04:45.057503494 +0200
@@ -15,11 +15,23 @@ from okonomiyaki.errors import Okonomiya
 class ArchitectureKind(enum.Enum):
     x86 = 0
     x86_64 = 1
+    arm = 2
+    aarch64 = 3
+    riscv64 = 4
+    ppc64le = 5
+    s390x = 6
+    loongarch64 = 7
 
 
 _KIND_TO_BITWIDTHS = {
     ArchitectureKind.x86: 32,
     ArchitectureKind.x86_64: 64,
+    ArchitectureKind.arm: 32,
+    ArchitectureKind.aarch64: 64,
+    ArchitectureKind.riscv64: 64,
+    ArchitectureKind.ppc64le: 64,
+    ArchitectureKind.s390x: 64,
+    ArchitectureKind.loongarch64: 64,
 }
 for k in ArchitectureKind.__members__:
     assert ArchitectureKind[k] in _KIND_TO_BITWIDTHS
@@ -32,6 +44,13 @@ _NORMALIZED_NAMES = {
     "amd64": ArchitectureKind.x86_64,
     "AMD64": ArchitectureKind.x86_64,
     "x86_64": ArchitectureKind.x86_64,
+
+    "armv8l": ArchitectureKind.arm,
+    "aarch64": ArchitectureKind.aarch64,
+    "riscv64": ArchitectureKind.riscv64,
+    "ppc64le": ArchitectureKind.ppc64le,
+    "s390x": ArchitectureKind.s390x,
+    "loongarch64": ArchitectureKind.loongarch64,
 }
 
 
