From 9168bf77f3e0404369d438e074e814ce5222d9cf Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Sun, 19 Feb 2023 10:48:33 +0100
Subject: [PATCH] force-remove werror

---
 setup.py | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/setup.py b/setup.py
index 79558d0..1c79430 100644
--- a/setup.py
+++ b/setup.py
@@ -467,10 +467,2 @@ for e in extensions:
 
-    if (
-            "CI" in os.environ
-            and not e.name.startswith("_sdl2")
-            and e.name not in ("pypm", "_sprite", "gfxdraw")
-    ):
-        # Do -Werror only on CI, and exclude -Werror on Cython C files and gfxdraw
-        e.extra_compile_args.append("/WX" if sys.platform == "win32" else "-Werror")
-
 # if not building font, try replacing with ftfont
