From 279509baf4872e58103768a53ca87918254767fc Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Fri, 25 Feb 2011 14:52:58 +0000
Subject: Plug memory leak

---
diff --git a/libnautilus-private/nautilus-icon-info.c b/libnautilus-private/nautilus-icon-info.c
index 97e88cb..97f42fd 100644
--- a/libnautilus-private/nautilus-icon-info.c
+++ b/libnautilus-private/nautilus-icon-info.c
@@ -393,6 +393,7 @@ nautilus_icon_info_lookup (GIcon *icon,
 
 		filename = gtk_icon_info_get_filename (gtkicon_info);
 		if (filename == NULL) {
+			gtk_icon_info_free (gtkicon_info);
 			return nautilus_icon_info_new_for_pixbuf (NULL);
 		}
 
--
cgit v0.9
