--- ./src/Makefile.in.orig
+++ ./src/Makefile.in
@@ -469,7 +469,7 @@
 LDADD = \
 	$(top_builddir)/src/file-manager/libnautilus-file-manager.la \
 	$(top_builddir)/libnautilus-private/libnautilus-private.la  \
-	$(CORE_LIBS) \
+	$(CORE_LIBS) -lgmodule-2.0 \
 	$(EXIF_LIBS) \
 	$(EXEMPI_LIBS) \
 	$(POPT_LIBS) \
--- ./test/Makefile.in.orig
+++ ./test/Makefile.in
@@ -383,7 +383,7 @@
 
 LDADD = \
 	$(top_builddir)/libnautilus-private/libnautilus-private.la \
-	$(CORE_LIBS) \
+	$(CORE_LIBS) -lgmodule-2.0 \
 	$(NULL)
 
 test_nautilus_copy_SOURCES = test-copy.c test.c
