From eda551a4d77d3dfe6eaef12f6fc71a0d93542467 Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Mon, 16 Aug 2021 15:50:07 +0200
Subject: [PATCH] man: fix scdoc syntax error

---
 tlstunnel.1.scd | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tlstunnel.1.scd b/tlstunnel.1.scd
index a6a70cb..d995250 100644
--- a/tlstunnel.1.scd
+++ b/tlstunnel.1.scd
@@ -44,7 +44,7 @@ The following directives are supported:
 
 	Each address is in the form _<name>:<port>_. The name may be omitted to
 	match all hosts. The name can contain a wildcard, but only to represent the
-	first label of the hostname (e.g. _*.example.org_ works, matches
+	first label of the hostname (e.g. _\*.example.org_ works, matches
 	_foo.example.org_ but doesn't match _foo.bar.example.org_).
 
 	The frontend directive supports the following sub-directives:
-- 
2.32.0

