Tcl 8.6.14: source $filePath
Tcl 8.6.15: source -encoding utf-8 $filePath

--- a/library/convlib.tcl
+++ b/library/convlib.tcl
@@ -57,7 +57,7 @@
                 set allOK 0
                 continue
             }
-            set filePath [lindex $auto_index($procName) 1]
+            set filePath [lindex $auto_index($procName) end]
             set fileName [file tail $filePath] 
             if {[lsearch $ignore $fileName] >= 0} continue
             
