diff --git a/include/core/posix/linux/proc/process/stat.h b/include/core/posix/linux/proc/process/stat.h
index bfbe440..8b815bc 100644
--- a/include/core/posix/linux/proc/process/stat.h
+++ b/include/core/posix/linux/proc/process/stat.h
@@ -23,6 +23,7 @@
 #include <core/posix/visibility.h>
 
 #include <string>
+#include <sys/types.h>
 
 namespace core
 {
diff --git a/tests/posix_process_test.cpp b/tests/posix_process_test.cpp
index 3481a4b..cfaaac6 100644
--- a/tests/posix_process_test.cpp
+++ b/tests/posix_process_test.cpp
@@ -24,6 +24,7 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include <array>
 #include <chrono>
 #include <map>
 #include <thread>
