diff --git a/Makefile b/Makefile
index c44ad29..e0a9654 100644
--- a/Makefile
+++ b/Makefile
@@ -55,14 +55,7 @@ DYN_DEBUG_GO_FLAGS := -gcflags "all=-N -l" -ldflags "$(GO_LDFLAGS)" -tags netgo
 # Docker mount flag, ignored on native docker host. see (https://docs.docker.com/docker-for-mac/osxfs-caching/#delegated)
 MOUNT_FLAGS := :delegated
 
-NETGO_CHECK = @strings $@ | grep cgo_stub\\\.go >/dev/null || { \
-       rm $@; \
-       echo "\nYour go standard library was built without the 'netgo' build tag."; \
-       echo "To fix that, run"; \
-       echo "    sudo go clean -i net"; \
-       echo "    sudo go install -tags netgo std"; \
-       false; \
-}
+NETGO_CHECK =
 
 # Protobuf files
 PROTO_DEFS := $(shell find . $(DONT_FIND) -type f -name '*.proto' -print)
@@ -134,7 +127,7 @@ logcli-image:
 	$(SUDO) docker build -t $(IMAGE_PREFIX)/logcli:$(IMAGE_TAG) -f cmd/logcli/Dockerfile .
 
 cmd/logcli/logcli:
-	CGO_ENABLED=0 go build $(GO_FLAGS) -o $@ ./cmd/logcli
+	CGO_ENABLED=0 go build $(GO_FLAGS) $(GOFLAGS) -o $@ ./cmd/logcli
 	$(NETGO_CHECK)
 
 ########
@@ -145,7 +138,7 @@ loki: cmd/loki/loki
 loki-debug: cmd/loki/loki-debug
 
 cmd/loki/loki:
-	CGO_ENABLED=0 go build $(GO_FLAGS) -o $@ ./$(@D)
+	CGO_ENABLED=0 go build $(GO_FLAGS) $(GOFLAGS) -o $@ ./$(@D)
 	$(NETGO_CHECK)
 
 cmd/loki/loki-debug:
@@ -203,7 +196,7 @@ $(PROMTAIL_GENERATED_FILE): $(PROMTAIL_UI_FILES)
 	GOOS=$(shell go env GOHOSTOS) go generate -x -v ./clients/pkg/promtail/server/ui
 
 clients/cmd/promtail/promtail:
-	CGO_ENABLED=$(PROMTAIL_CGO) go build $(PROMTAIL_GO_FLAGS) -o $@ ./$(@D)
+	CGO_ENABLED=$(PROMTAIL_CGO) go build $(PROMTAIL_GO_FLAGS) $(GOFLAGS) -o $@ ./$(@D)
 	$(NETGO_CHECK)
 
 clients/cmd/promtail/promtail-debug:
