diff --git a/url.rs b/url.rs
index 9d4ce36..95d765e 100644
--- a/crates/dprint/src/utils/url.rs
+++ b/crates/dprint/src/utils/url.rs
@@ -535,7 +535,7 @@ kGUMOx8j0U5fU8eSLECGi0FxBA==
     match result {
       Ok(child) => Some(ChildDrop { child }),
       Err(err) => {
-        if err.to_string().contains("Not found") {
+        if err.to_string().contains("No") {
           return None;
         } else {
           panic!("Failed running Deno: {:#}", err);
