remove hardcoded version to use the setuptools_scm one

diff --git a/setup.py b/setup.py
index 6478bb7..6bd964d 100755
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,6 @@ if sys.argv[-1] in ('sdist', 'bdist_wheel'):
 
 setup(
     name="ydcv",
-    version="0.0.1",
     description="YouDao Console Version, a simple wrapper for Youdao API",
     long_description_markdown_filename='README.md',
     author="Felix Yan",
