diff --git a/pkg/analyze/storage.go b/pkg/analyze/storage.go

Default size is 2GB, which cannot be mmapped on 32-bits systems. Causes test
failure.

Ref: https://discuss.dgraph.io/t/lower-memory-usage/12813
diff --git a/pkg/analyze/badger_32bits.go b/pkg/analyze/badger_32bits.go
new file mode 100644
index 0000000..8eb8f1d
--- /dev/null
+++ b/pkg/analyze/badger_32bits.go
@@ -0,0 +1,5 @@
+//go:build 386 || arm
+
+package analyze
+
+const maxValueLogFileSize = 1 << 20
diff --git a/pkg/analyze/badger_64bits.go b/pkg/analyze/badger_64bits.go
new file mode 100644
index 0000000..a21ca2e
--- /dev/null
+++ b/pkg/analyze/badger_64bits.go
@@ -0,0 +1,5 @@
+//go:build !(386 || arm)
+
+package analyze
+
+const maxValueLogFileSize = 1 << 30
diff --git a/pkg/analyze/storage.go b/pkg/analyze/storage.go
index b09e879..14d4269 100644
--- a/pkg/analyze/storage.go
+++ b/pkg/analyze/storage.go
@@ -56,6 +56,7 @@ func (s *Storage) IsOpen() bool {
 // Open opens badger DB
 func (s *Storage) Open() func() {
 	options := badger.DefaultOptions(s.storagePath)
+	options.ValueLogFileSize = maxValueLogFileSize
 	options.Logger = nil
 	db, err := badger.Open(options)
 	if err != nil {
