--- a/qt5-im/Makefile	2017-03-01 00:27:31.389349736 +0800
+++ b/qt5-im/Makefile	2017-03-01 00:27:39.142665623 +0800
@@ -10,6 +10,10 @@
 INCS+=-I/usr/include/qt/QtGui/$(MODVERSION)/QtGui -I/usr/include/i386-linux-gnu/qt5/QtGui/$(MODVERSION)/QtGui -I/usr/include/x86_64-linux-gnu/qt5/QtGui/$(MODVERSION)/QtGui
 INCS+=-I/usr/include/x86_64-linux-gnu/qt5/QtGui/$(MODVERSION)/QtGui
 
+# dirty fix for Alpine's QT5 path
+INCS+=-I/usr/include/QtGui/$(MODVERSION) -I/usr/include/QtCore/$(MODVERSION)
+INCS+=-I/usr/include/QtGui/$(MODVERSION)/QtGui -I/usr/include/QtGui
+
 CXXFLAGS=$(OPTFLAGS) $(INCS) -Wall -D_REENTRANT -DUNIX=1 -fPIC  -DQT5 -DQT_SHARED -DQT_IMMODULE -DPIC -DDEBUG="0$(GCIN_DEBUG)"
 OBJS= moc_gcin-qt5.o gcin-qt5.o gcin-imcontext-qt5.o moc_qgcinplatforminputcontext.o
 .SUFFIXES:	.c .cpp .a .so .E .h
