From 3bdba8c2bc3c3625d9751501ab42b047ec01addf Mon Sep 17 00:00:00 2001
From: Simon Ser <contact@emersion.fr>
Date: Wed, 6 Mar 2024 19:22:45 +0100
Subject: [PATCH] build: set library version and soversion

---
 meson.build | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/meson.build b/meson.build
index f426e0c..8877259 100644
--- a/meson.build
+++ b/meson.build
@@ -11,6 +11,11 @@ project(
 	],
 )
 
+version = meson.project_version()
+version_major = version.split('.')[0]
+version_minor = version.split('.')[1]
+assert(version_major == '0')
+
 cc = meson.get_compiler('c')
 
 add_project_arguments(cc.get_supported_arguments([
@@ -28,6 +33,8 @@ scfg_lib = library(
 	files('scfg.c'),
 	include_directories: scfg_inc,
 	install: true,
+	version: version,
+	soversion: version_minor,
 )
 
 scfg = declare_dependency(
-- 
2.45.0

