From b72d92a3bebaaa6d7382f582aef149bb604f3e11 Mon Sep 17 00:00:00 2001
From: Javier Jardón <jjardon@gnome.org>
Date: Thu, 30 Sep 2010 16:42:44 +0000
Subject: Replacing GtkNotebookPage with GtkWidget according to changes in GTK+

Signed-off-by: Chris Kühl <chrisk@openismus.com>
---
diff --git a/src/callbacks.cpp b/src/callbacks.cpp
index be2c16a..b0ac269 100644
--- a/src/callbacks.cpp
+++ b/src/callbacks.cpp
@@ -323,7 +323,7 @@ cb_tree_popup_menu (GtkWidget *widget, gpointer data)
 
 
 void
-cb_switch_page (GtkNotebook *nb, GtkNotebookPage *page,
+cb_switch_page (GtkNotebook *nb, GtkWidget *page,
 		gint num, gpointer data)
 {
 	cb_change_current_page (nb, num, data);
diff --git a/src/callbacks.h b/src/callbacks.h
index 4c84793..6ce3e18 100644
--- a/src/callbacks.h
+++ b/src/callbacks.h
@@ -60,7 +60,7 @@ gboolean	cb_tree_button_pressed (GtkWidget *widget, GdkEventButton *event,
 
 void		cb_change_current_page (GtkNotebook *nb,
 					gint num, gpointer data);
-void		cb_switch_page (GtkNotebook *nb, GtkNotebookPage *page,
+void		cb_switch_page (GtkNotebook *nb, GtkWidget *page,
 				gint num, gpointer data);
 
 gint		cb_update_disks (gpointer data);
--
cgit v0.9
