From e302c23c43112fe336bf9d0481bdd94fc5af7b21 Mon Sep 17 00:00:00 2001
From: Chris Kühl <chrisk@openismus.com>
Date: Wed, 26 Jan 2011 13:12:09 +0000
Subject: Replaced occurances of gtk_dialog_set_has_separator with g_object_set.

---
diff --git a/src/lsof.cpp b/src/lsof.cpp
index 24f14a5..7773eff 100644
--- a/src/lsof.cpp
+++ b/src/lsof.cpp
@@ -312,7 +312,6 @@ void procman_lsof(ProcData *procdata)
   //		   G_CALLBACK(close_dialog), NULL);
   gtk_window_set_resizable(GTK_WINDOW(dialog), TRUE);
   gtk_window_set_default_size(GTK_WINDOW(dialog), 575, 400);
-  // gtk_dialog_set_has_separator(GTK_DIALOG(dialog), FALSE);
   gtk_container_set_border_width(GTK_CONTAINER(dialog), 12);
   GtkWidget *mainbox = gtk_vbox_new(FALSE, 12);
   gtk_container_add(GTK_CONTAINER(dialog), mainbox);
diff --git a/src/memmaps.cpp b/src/memmaps.cpp
index 9a6fcae..3205721 100644
--- a/src/memmaps.cpp
+++ b/src/memmaps.cpp
@@ -608,7 +608,6 @@ create_single_memmaps_dialog (GtkTreeModel *model, GtkTreePath *path,
 	gtk_window_set_title(GTK_WINDOW(memmapsdialog), _("Memory Maps"));
 	gtk_window_set_resizable(GTK_WINDOW(memmapsdialog), TRUE);
 	gtk_window_set_default_size(GTK_WINDOW(memmapsdialog), 575, 400);
-	// gtk_dialog_set_has_separator(GTK_DIALOG(dialog), FALSE);
 	gtk_container_set_border_width(GTK_CONTAINER(memmapsdialog), 12);
 
 	GtkWidget *mainbox = gtk_vbox_new(FALSE, 12);
diff --git a/src/openfiles.cpp b/src/openfiles.cpp
index f4a088a..c4ba2c8 100644
--- a/src/openfiles.cpp
+++ b/src/openfiles.cpp
@@ -335,7 +335,7 @@ create_single_openfiles_dialog (GtkTreeModel *model, GtkTreePath *path,
 						       NULL);
 	gtk_window_set_resizable (GTK_WINDOW (openfilesdialog), TRUE);
 	gtk_window_set_default_size (GTK_WINDOW (openfilesdialog), 575, 400);
-	gtk_dialog_set_has_separator (GTK_DIALOG (openfilesdialog), FALSE);
+	g_object_set(G_OBJECT(openfilesdialog), "has-separator", FALSE, NULL);
 	gtk_container_set_border_width (GTK_CONTAINER (openfilesdialog), 5);
 
 	vbox = gtk_dialog_get_content_area (GTK_DIALOG (openfilesdialog));
diff --git a/src/procdialogs.cpp b/src/procdialogs.cpp
index 4c814a0..398e68f 100644
--- a/src/procdialogs.cpp
+++ b/src/procdialogs.cpp
@@ -178,7 +178,7 @@ procdialog_create_renice_dialog (ProcData *procdata)
 				              NULL);
   	renice_dialog = dialog;
 	gtk_window_set_resizable (GTK_WINDOW (renice_dialog), FALSE);
-	gtk_dialog_set_has_separator (GTK_DIALOG (renice_dialog), FALSE);
+	g_object_set (G_OBJECT (renice_dialog), "has-separator", FALSE, NULL);
 	gtk_container_set_border_width (GTK_CONTAINER (renice_dialog), 5);
   	
 	button = gtk_button_new ();
@@ -510,7 +510,7 @@ procdialog_create_preferences_dialog (ProcData *procdata)
 	/* any function to set list height by contents/items inside it. */
 	gtk_window_set_default_size (GTK_WINDOW (dialog), 400, 420);
 	gtk_container_set_border_width (GTK_CONTAINER (dialog), 5);
-	gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
+	g_object_set (G_OBJECT (dialog), "has-separator", FALSE, NULL);
 	prefs_dialog = dialog;
 	
 	main_vbox = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
--
cgit v0.9
