--- lttng-tools-2.9.0.orig/tests/regression/ust/ust-dl/prog.c
+++ lttng-tools-2.9.0/tests/regression/ust/ust-dl/prog.c
@@ -21,7 +21,7 @@
 	if (!h0) {
 		goto get_error;
 	}
-	h1 = dlmopen(LM_ID_BASE, "libfoo.so", RTLD_LAZY);
+	h1 = dlopen("libfoo.so", RTLD_LAZY);
 	if (!h1) {
 		goto get_error;
 	}
--- lttng-tools-2.9.0.orig/tests/regression/ust/ust-dl/test_ust-dl.py
+++ lttng-tools-2.9.0/tests/regression/ust/ust-dl/test_ust-dl.py
@@ -31,7 +31,7 @@
 from test_utils import *
 
 
-NR_TESTS = 14
+NR_TESTS = 13
 current_test = 1
 print("1..{0}".format(NR_TESTS))
 
@@ -65,7 +65,6 @@
     bail("Could not open babeltrace. Please make sure it is installed.", session_info)
 
 dlopen_event_found = 0
-dlmopen_event_found = 0
 build_id_event_found = 0
 debug_link_event_found = 0
 dlclose_event_found = 0
@@ -82,8 +81,6 @@
     event_line = event_line.decode('utf-8').replace("\n", "")
     if re.search(r".*lttng_ust_dl:dlopen.*", event_line) is not None:
         dlopen_event_found += 1
-    elif re.search(r".*lttng_ust_dl:dlmopen.*", event_line) is not None:
-        dlmopen_event_found += 1
     elif re.search(r".*lttng_ust_dl:build_id.*", event_line) is not None:
         build_id_event_found += 1
     elif re.search(r".*lttng_ust_dl:debug_link.*", event_line) is not None:
@@ -111,9 +108,6 @@
 current_test += 1
 
 print_test_result(dlopen_event_found > 0, current_test, "lttng_ust_dl:dlopen event found in resulting trace")
-current_test += 1
-
-print_test_result(dlmopen_event_found > 0, current_test, "lttng_ust_dl:dlmopen event found in resulting trace")
 current_test += 1
 
 print_test_result(build_id_event_found > 0, current_test, "lttng_ust_dl:build_id event found in resulting trace")
