diff --git i/host_applications/linux/apps/raspicam/RaspiCommonSettings.c w/host_applications/linux/apps/raspicam/RaspiCommonSettings.c
index 38b07ab..20ad385 100644
--- i/host_applications/linux/apps/raspicam/RaspiCommonSettings.c
+++ w/host_applications/linux/apps/raspicam/RaspiCommonSettings.c
@@ -41,6 +41,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <unistd.h>
 #include <errno.h>
 #include <sysexits.h>
+#include <libgen.h>
 
 #include "interface/vcos/vcos.h"
 #include "interface/mmal/mmal.h"
diff --git i/host_applications/linux/apps/raspicam/RaspiHelpers.c w/host_applications/linux/apps/raspicam/RaspiHelpers.c
index 9421148..da4afae 100644
--- i/host_applications/linux/apps/raspicam/RaspiHelpers.c
+++ w/host_applications/linux/apps/raspicam/RaspiHelpers.c
@@ -32,6 +32,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <memory.h>
 #include <unistd.h>
 #include <stdint.h>
+#include <libgen.h>
 
 #include <fcntl.h>
 #include <sys/ioctl.h>
diff --git i/host_applications/linux/apps/raspicam/RaspiStill.c w/host_applications/linux/apps/raspicam/RaspiStill.c
index a88bcc8..417a87e 100644
--- i/host_applications/linux/apps/raspicam/RaspiStill.c
+++ w/host_applications/linux/apps/raspicam/RaspiStill.c
@@ -57,6 +57,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <unistd.h>
 #include <errno.h>
 #include <sysexits.h>
+#include <libgen.h>
 
 #include "bcm_host.h"
 #include "interface/vcos/vcos.h"
diff --git i/host_applications/linux/apps/raspicam/RaspiStillYUV.c w/host_applications/linux/apps/raspicam/RaspiStillYUV.c
index 4acd1ab..dbe0f76 100644
--- i/host_applications/linux/apps/raspicam/RaspiStillYUV.c
+++ w/host_applications/linux/apps/raspicam/RaspiStillYUV.c
@@ -56,6 +56,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <sysexits.h>
 #include <unistd.h>
 #include <errno.h>
+#include <libgen.h>
 
 #include "bcm_host.h"
 #include "interface/vcos/vcos.h"
diff --git i/host_applications/linux/apps/raspicam/RaspiVid.c w/host_applications/linux/apps/raspicam/RaspiVid.c
index 93114bf..01f7fe5 100644
--- i/host_applications/linux/apps/raspicam/RaspiVid.c
+++ w/host_applications/linux/apps/raspicam/RaspiVid.c
@@ -60,6 +60,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <ctype.h>
 #include <memory.h>
 #include <sysexits.h>
+#include <libgen.h>
 
 #include <sys/types.h>
 #include <sys/socket.h>
diff --git i/host_applications/linux/apps/raspicam/RaspiVidYUV.c w/host_applications/linux/apps/raspicam/RaspiVidYUV.c
index 87ab502..37ac095 100644
--- i/host_applications/linux/apps/raspicam/RaspiVidYUV.c
+++ w/host_applications/linux/apps/raspicam/RaspiVidYUV.c
@@ -58,6 +58,7 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <string.h>
 #include <memory.h>
 #include <sysexits.h>
+#include <libgen.h>
 
 #include <sys/types.h>
 #include <sys/socket.h>
