diff --git a/host_applications/linux/CMakeLists.txt b/host_applications/linux/CMakeLists.txt
index 928b637..4caaac8 100644
--- a/host_applications/linux/CMakeLists.txt
+++ b/host_applications/linux/CMakeLists.txt
@@ -11,7 +11,6 @@ if(NOT ARM64)
 endif()
 add_subdirectory(libs/debug_sym)
 add_subdirectory(apps/dtoverlay)
-add_subdirectory(apps/dtmerge)
 
 if(ALL_APPS)
  add_subdirectory(apps/vcdbg)
