--- config.mk.orig	2017-09-19 11:39:26.939320274 +0200
+++ config.mk	2017-09-19 11:39:32.689379959 +0200
@@ -21,8 +21,8 @@ LIBS = -L/usr/lib -lc -lm -L${X11LIB} -l
 
 # flags
 CPPFLAGS = -DVERSION=\"${VERSION}\" -D_XOPEN_SOURCE=600
-CFLAGS += -g -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
-LDFLAGS += -g ${LIBS}
+CFLAGS += -std=c99 -pedantic -Wall ${INCS} ${CPPFLAGS}
+LDFLAGS += ${LIBS}
 #CFLAGS += -std=c99 -pedantic -Wall -Os ${INCS} ${CPPFLAGS}
 #LDFLAGS += ${LIBS}
 
