From bfa1e407c45ae7de7e9f738745eb92109c1657dc Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Thu, 14 Mar 2024 13:50:00 +0000
Subject: [PATCH] do not build man pages

---
 lib/common.Makefile | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/lib/common.Makefile b/lib/common.Makefile
index b62530d..2651154 100644
--- a/lib/common.Makefile
+++ b/lib/common.Makefile
@@ -35,11 +35,7 @@ $(DAEMONSRCS) $(HASHSRCS):	$(patsubst %,../daemon/%,$(DAEMONSRCS)) $(patsubst %,
 		cat ../daemon/"$@" ) > "$@"
 
 %.8: ../docs/%.md
-	cat "$<" | sed '/^# /d; s/^##/#/' | \
-		pandoc -s -t man \
-			-M "footer:$(RTPENGINE_VERSION)" \
-			-M "date:$(BUILD_DATE)" \
-			-o "$@"
+		echo 'building man pages disabled due to pandoc unavailable on all arches'
 
 resample.c codeclib.strhash.c mix.c packet.c:	fix_frame_channel_layout.h
 
-- 
2.43.1

