--- a/lib/poller.c
+++ b/lib/poller.c
@@ -53,11 +53,10 @@
 }
 
 static void poller_map_add(struct poller_map *map) {
-	pthread_t tid = -1;
+	pthread_t tid = pthread_self();
 	struct poller *p;
 	if (!map)
 		return;
-	tid = pthread_self();
 
 	LOCK(&map->lock);
 	p = poller_new();
--- a/daemon/control_tcp.c
+++ b/daemon/control_tcp.c
@@ -101,7 +101,7 @@
 		free(output);
 	}
 
-	pcre2_substring_list_free((PCRE2_SPTR *) out);
+	pcre2_substring_list_free((void *) out);
 	pcre2_match_data_free(md);
 	log_info_pop();
 	return 1;
--- a/daemon/control_udp.c
+++ b/daemon/control_udp.c
@@ -60,7 +60,7 @@
 
 		socket_sendiov(udp_buf->listener, iov, iovlen, &udp_buf->sin, &udp_buf->local_addr);
 
-		pcre2_substring_list_free((PCRE2_SPTR *) out);
+		pcre2_substring_list_free((void *) out);
 		pcre2_match_data_free(md);
 
 		return;
@@ -129,7 +129,7 @@
 		cookie_cache_remove(&u->cookie_cache, &cookie);
 
 out:
-	pcre2_substring_list_free((PCRE2_SPTR *) out);
+	pcre2_substring_list_free((void *) out);
 	pcre2_match_data_free(md);
 	log_info_pop();
 }
