diff --git a/pyproject.toml.orig b/pyproject.toml
index 92de76a4a32..14382a40121 100644
--- a/pyproject.toml.orig
+++ b/pyproject.toml
@@ -24,7 +24,6 @@ classifiers = [
     "Framework :: Django",
     "Framework :: Django :: 4.2",
     "Framework :: Django :: 5.0",
-    "Framework :: Django :: 5.1",
     "Intended Audience :: Developers",
     "License :: OSI Approved :: Apache Software License",
     "Operating System :: OS Independent",
