--- xe-guest-utilities-5.5.0.orig/uclibc-sources/Config.mk
+++ xe-guest-utilities-5.5.0/uclibc-sources/Config.mk
@@ -70,7 +70,6 @@
       export XEN_ROOT
   endef
 
-CFLAGS += -g
 
 CFLAGS += -fno-strict-aliasing
 
--- xe-guest-utilities-5.5.0.orig/uclibc-sources/config/StdGNU.mk
+++ xe-guest-utilities-5.5.0/uclibc-sources/config/StdGNU.mk
@@ -44,7 +44,6 @@
 SHLIB_CFLAGS = -shared
 
 ifneq ($(debug),y)
-CFLAGS += -O2 -fomit-frame-pointer
 else
 # Less than -O1 produces bad code and large stack frames
 CFLAGS += -O1 -fno-omit-frame-pointer -fno-optimize-sibling-calls
--- xe-guest-utilities-5.5.0.orig/uclibc-sources/config/x86_32.mk
+++ xe-guest-utilities-5.5.0/uclibc-sources/config/x86_32.mk
@@ -7,7 +7,6 @@
 CONFIG_XCUTILS := y
 CONFIG_IOEMU := n
 
-CFLAGS += -m32 -march=i686
 
 # Use only if calling $(LD) directly.
 LDFLAGS_DIRECT_OpenBSD = _obsd
--- xe-guest-utilities-5.5.0.orig/uclibc-sources/tools/Rules.mk
+++ xe-guest-utilities-5.5.0/uclibc-sources/tools/Rules.mk
@@ -27,9 +27,7 @@
 CFLAGS += -D__XEN_TOOLS__
 
 # Enable implicit LFS support *and* explicit LFS names.
-CFLAGS  += $(shell getconf LFS_CFLAGS)
-CFLAGS  += -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
-LDFLAGS += $(shell getconf LFS_LDFLAGS)
+CFLAGS  += -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 
 # 32-bit x86 does not perform well with -ve segment accesses on Xen.
 CFLAGS-$(CONFIG_X86_32) += $(call cc-option,$(CC),-mno-tls-direct-seg-refs)
