--- xe-guest-utilities-5.5.0.orig/uclibc-sources/tools/xenstore/Makefile
+++ xe-guest-utilities-5.5.0/uclibc-sources/tools/xenstore/Makefile
@@ -97,7 +97,7 @@
 	$(INSTALL_DIR) $(DESTDIR)/usr/bin
 	$(INSTALL_PROG) xenstore $(DESTDIR)/usr/bin
 	set -e ; for c in $(CLIENTS) ; do \
-		ln -f $(DESTDIR)/usr/bin/xenstore $(DESTDIR)/usr/bin/$${c} ; \
+		ln -sf $(DESTDIR)/usr/bin/xenstore $(DESTDIR)/usr/bin/$${c} ; \
 	done
 
 .PHONY: install
