Subject: build: force -no-pie

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 47a96c3..6a6c07c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1522,7 +1522,7 @@ if test x"$efiemu_excuse" = x ; then
   if test x"$grub_cv_target_cc_efiemu64_link_format" = xunknown; then
     efiemu_excuse="no suitable link format for efiemu64 found"
   else
-    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format"
+    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format -no-pie"
   fi
 fi
 if test x"$enable_efiemu" = xyes && test x"$efiemu_excuse" != x ; then
