--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -144,9 +144,13 @@
   if test -n "${initrd}" ; then
     # TRANSLATORS: ramdisk isn't identifier. Should be translated.
     message="$(gettext_printf "Loading initial ramdisk ...")"
+    initrd_line="${rel_dirname}/${initrd}"
+    if [ -f "/boot/intel-ucode.img" ]; then
+	initrd_line="${rel_dirname}/intel-ucode.img $initrd_line"
+    fi
     sed "s/^/$submenu_indentation/" << EOF
 	echo	'$(echo "$message" | grub_quote)'
-	initrd	${rel_dirname}/${initrd}
+	initrd	${initrd_line}
 EOF
   fi
   sed "s/^/$submenu_indentation/" << EOF
