--- ./configure.ac.orig
+++ ./configure.ac
@@ -1100,7 +1100,7 @@
   if test x"$grub_cv_target_cc_efiemu64_link_format" = xunknown; then
     efiemu_excuse="no suitable link format for efiemu64 found"
   else
-    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format"
+    EFIEMU64_LINK_FORMAT="-Wl,$grub_cv_target_cc_efiemu64_link_format -no-pie"
   fi
 fi
 if test x"$enable_efiemu" = xyes && test x"$efiemu_excuse" != x ; then
