diff --git a/util/grub.d/20_linux_xen.in b/util/grub.d/20_linux_xen.in
index 2f78bcf..5ea1bd3 100644
--- a/util/grub.d/20_linux_xen.in
+++ b/util/grub.d/20_linux_xen.in
@@ -187,7 +187,7 @@ for i in /boot/vmlinu[xz]-* /vmlinu[xz]-* /boot/kernel-*; do
 	version=$(echo $basename | sed -e "s,^[^-]*-,,g")
 	dirname=$(dirname $i)
 	config=
-	for j in "${dirname}/config-${version}" "${dirname}/config-${alt_version}" "/etc/kernels/kernel-config-${version}" ; do
+	for j in "${dirname}/config-${version}" "${dirname}/config-${alt_version}" "/etc/kernels/kernel-config-${version}" ${dirname}/config-[0-9]*-${version}; do
 	    if test -e "${j}" ; then
 		config="${j}"
 		break
