--- ./autogen.sh.orig
+++ ./autogen.sh
@@ -105,7 +105,7 @@
 
 if [ $lt_major_version -ge 2 ]; then
     for file in config.guess config.sub; do
-        configfile=${LIBTOOL_CONFIG-`cd $ltpath/../share/libtool/config ; pwd`/$file}
+        configfile=${LIBTOOL_CONFIG-`cd $ltpath/../share/libtool/config ; pwd`}/$file
 
         if [ ! -f $configfile ]; then
             echo "$configfile not found (try setting the LIBTOOL_CONFIG environment variable)"
