--- subversion-1.7.2/subversion/bindings/swig/perl/native/Makefile.PL.in.orig
+++ subversion-1.7.2/subversion/bindings/swig/perl/native/Makefile.PL.in
@@ -49,11 +49,7 @@
 my $cflags = '@CFLAGS@';
 my $apr_cflags = '@SVN_APR_INCLUDES@';
 my $apu_cflags = '@SVN_APRUTIL_INCLUDES@';
-
-# According to the log of r7937, the flags guarded by the conditional break
-# the build on FreeBSD if not conditionalized.
-my $apr_ldflags = '@SVN_APR_LIBS@'
-   if $^O eq 'darwin' or $^O eq 'cygwin';
+my $apr_ldflags = '@SVN_APR_LIBS@';
 
 chomp $apr_shlib_path_var;
 
--- subversion-1.7.2/configure.ac.orig
+++ subversion-1.7.2/configure.ac
@@ -328,6 +328,6 @@
 AC_MSG_CHECKING([whether to avoid circular linkage at all costs])
 case $host in
-  *-*-cygwin*)
+  *linux*)
     AC_MSG_RESULT([yes])
     AC_DEFINE([SVN_AVOID_CIRCULAR_LINKAGE_AT_ALL_COSTS_HACK], 1,
               [Define if circular linkage is not possible on this platform.])
