diff --git a/build/ac-macros/aprutil.m4 b/build/ac-macros/aprutil.m4
index 54f5e32..40c2ed3 100644
--- a/build/ac-macros/aprutil.m4
+++ b/build/ac-macros/aprutil.m4
@@ -103,9 +103,9 @@ AC_DEFUN(SVN_LIB_APRUTIL,
       AC_MSG_ERROR([apu-config --link-ld --libs failed])
     fi
   else
-    SVN_APRUTIL_LIBS="`$apu_config --link-ld`"
+    SVN_APRUTIL_LIBS="`$apu_config --link-ld --libs`"
     if test $? -ne 0; then
-      AC_MSG_ERROR([apu-config --link-ld failed])
+      AC_MSG_ERROR([apu-config --link-ld --libs failed])
     fi
   fi
   SVN_APRUTIL_LIBS="`SVN_REMOVE_STANDARD_LIB_DIRS($SVN_APRUTIL_LIBS)`"
