--- a/apps/irssi/configure
+++ b/apps/irssi/configure
@@ -15109,8 +15109,8 @@
 		    ;;
 		esac
 
-				echo "main(){perl_alloc(); return 0;}" > conftest.c
-		$CC $CFLAGS conftest.c -o conftest $LDFLAGS $PERL_LDFLAGS 2> perl.error.tmp > /dev/null
+				printf '#include <EXTERN.h>\n#include <perl.h>\nint main(){perl_alloc(); return 0;}' > conftest.c
+		$CC $CFLAGS $PERL_CFLAGS conftest.c -o conftest $LDFLAGS $PERL_LDFLAGS 2> perl.error.tmp > /dev/null
 		if test ! -s conftest -a "x$ignore_perl_errors" = "x"; then
 			perl_check_error="Error linking with perl libraries: $PERL_LDFLAGS: `cat perl.error.tmp`"
 			{ $as_echo "$as_me:${as_lineno-$LINENO}: result: error linking with perl libraries, building without Perl" >&5
