diff -rupN a/apps/irssi/src/fe-text/terminfo-core.c b/apps/irssi/src/fe-text/terminfo-core.c
--- a/apps/irssi/src/fe-text/terminfo-core.c	2014-05-12 17:53:02.000000000 +0000
+++ b/apps/irssi/src/fe-text/terminfo-core.c	2026-08-06 20:33:53.659473295 +0000
@@ -12,28 +12,24 @@ inline static int term_putchar(int c)
         return fputc(c, current_term->out);
 }
 
+#ifdef HAVE_TERM_H
+#include <curses.h>
+#include <term.h>
+#else
 /* Don't bother including curses.h because of these -
    they might not even be defined there */
-char *tparm();
-int tputs();
+char *tparm(char *str, ...);
+int tputs(const char *str, int count, int (*putc)(int));
+
+int setupterm(char *term, int fd, int *err);
+char *tigetstr(char *ti_name);
+int tigetnum(char *ti_name);
+int tigetflag(char *ti_name);
+#endif
 
-#ifdef HAVE_TERMINFO
-int setupterm();
-char *tigetstr();
-int tigetnum();
-int tigetflag();
 #define term_getstr(x, buffer) tigetstr(x.ti_name)
 #define term_getnum(x) tigetnum(x.ti_name);
 #define term_getflag(x) tigetflag(x.ti_name);
-#else
-int tgetent();
-char *tgetstr();
-int tgetnum();
-int tgetflag();
-#define term_getstr(x, buffer) tgetstr(x.tc_name, &buffer)
-#define term_getnum(x) tgetnum(x.tc_name)
-#define term_getflag(x) tgetflag(x.tc_name)
-#endif
 
 #define CAP_TYPE_FLAG	0
 #define CAP_TYPE_INT	1
