Index: ocaml-4.03.0+beta1/tools/Makefile.shared
===================================================================
--- ocaml-4.03.0+beta1.orig/tools/Makefile.shared
+++ ocaml-4.03.0+beta1/tools/Makefile.shared
@@ -273,7 +273,7 @@ CCOUT = -o $(EMPTY)
 endif
 
 objinfo_helper$(EXE): objinfo_helper.c ../config/s.h
-	$(BYTECC) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
+	$(MKEXE) $(CCOUT)objinfo_helper$(EXE) $(BYTECCCOMPOPTS) \
           $(DEF_SYMBOL_PREFIX) $(LIBBFD_INCLUDE) objinfo_helper.c $(LIBBFD_LINK)
 
 OBJINFO=../compilerlibs/ocamlcommon.cma \
