--- ./configure.in.orig	2012-12-31 21:13:54.969154614 +0000
+++ ./configure.in	2012-12-31 21:14:11.542636354 +0000
@@ -56,7 +56,7 @@
 dnl> generate the config header
 dnl>
 umask 002
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 
 HAS_ERROR=
 HAS_WARNING=
@@ -324,7 +324,6 @@
 echo
 echo "Checking C compiler program and features..."
 echo
-AM_PROG_CC_STDC
 if test ".${ac_cv_prog_cc_stdc}" = ".no"; then
     echo
     echo "*******************************************************************"
