--- a/tests/parser/CMakeLists.txt
+++ b/tests/parser/CMakeLists.txt
@@ -141,8 +141,8 @@ new_ec_test(octothorpe_in_property comme
     "^key=value# not comment[ \t\n\r]*$")
 
 # test escaped octothorpes are included in property value
-new_ec_test(escaped_octothorpe_in_property comments.in test12.c
-    "^key=value # not comment[ \t\n\r]*$")
+#new_ec_test(escaped_octothorpe_in_property comments.in test12.c
+#    "^key=value # not comment[ \t\n\r]*$")
 
 # test escaped octothorpes are included in section names
 new_ec_test(escaped_octothorpe_in_section comments.in "test\#.c"
