From 3a61e3d30e7291c11d9bfa3dfcab499e31c9138b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 8 Nov 2011 15:42:46 +0000
Subject: [PATCH] blacklist viafb and e_powersaver

---
 blacklist.conf |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/blacklist.conf b/blacklist.conf
index 167fc31..9e0cfd6 100644
--- a/blacklist.conf
+++ b/blacklist.conf
@@ -55,6 +55,7 @@ blacklist sstfb
 blacklist neofb
 blacklist tridentfb
 blacklist tdfxfb
+blacklist viafb
 blacklist virgefb
 blacklist vga16fb
 blacklist matroxfb_base
@@ -69,3 +70,5 @@ blacklist hfcmulti
 blacklist hfcpci
 blacklist hfcsusb
 
+# blacklist C7 cpu freq. use acpi-cpufreq instead
+blacklist e_powersaver
-- 
1.7.7.2

