--- a/rr.gemspec
+++ b/rr.gemspec
@@ -2,3 +2,2 @@
 
-require 'rake'
 require File.expand_path('../lib/rr/version', __FILE__)
@@ -16,14 +15,3 @@
-  gem.files = FileList[
+  gem.files = Dir[
-    'Appraisals',
-    'CHANGES.md',
-    'CREDITS.md',
-    'Gemfile',
-    'LICENSE',
-    'README.md',
-    'Rakefile',
-    'doc/*.md',
-    'gemfiles/**/*',
     'lib/**/*.rb',
-    'rr.gemspec',
-    'spec/**/*'
-  ].to_a
+  ]
