From fdc640e20c4100bc960c591d62334e1c43322a45 Mon Sep 17 00:00:00 2001
From: Robert Swiecki <robert@swiecki.net>
Date: Tue, 25 Oct 2022 08:33:23 +0200
Subject: [PATCH 2/2] subproc.h: make cloneProc declaration match the
 definition

---
 subproc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/subproc.h b/subproc.h
index 269b9d0..a63416e 100644
--- a/subproc.h
+++ b/subproc.h
@@ -41,7 +41,7 @@ void killAndReapAll(nsjconf_t* nsjconf, int signal);
 /* Returns the exit code of the first failing subprocess, or 0 if none fail */
 int reapProc(nsjconf_t* nsjconf);
 int systemExe(const std::vector<std::string>& args, char** env);
-pid_t cloneProc(uintptr_t flags, int exit_signal);
+pid_t cloneProc(uint64_t flags, int exit_signal);
 
 }  // namespace subproc
 
-- 
2.38.1

