Reverts https://github.com/pypy/pypy/pull/5189

--- a/rpython/rlib/clibffi.py
+++ b/rpython/rlib/clibffi.py
@@ -74,14 +74,8 @@ if not _WIN32:
     else:
         pre_include_bits = []
 
-    library_dirs = platform.library_dirs_for_libffi()
     libraries = ['ffi']
     link_files = []
-    for libdir in library_dirs:
-        candidate = os.path.join(libdir, "libffi.a")
-        if os.path.exists(candidate):
-            link_files.append(candidate)
-            libraries = []
 
     eci = ExternalCompilationInfo(
         pre_include_bits = pre_include_bits,
@@ -90,7 +84,7 @@ if not _WIN32:
         separate_module_sources = separate_module_sources,
         post_include_bits = post_include_bits,
         include_dirs = platform.include_dirs_for_libffi(),
-        library_dirs = library_dirs,
+        library_dirs = platform.library_dirs_for_libffi(),
         link_files = link_files,
         testonly_libraries = ['ffi'],
     )
