diff --git a/parser/missingdefs.h b/parser/missingdefs.h
new file mode 100644
index 0000000..9b2057e
--- /dev/null
+++ b/parser/missingdefs.h
@@ -0,0 +1,9 @@
+#ifndef PARSER_MISSINGDEFS_H
+#define PARSER_MISSINGDEFS_H
+
+typedef int (*__compar_fn_t) (const void *, const void *);
+typedef __compar_fn_t comparison_fn_t;
+typedef void (*__free_fn_t) (void *__nodep);
+
+#endif
+
diff --git a/parser/parser_alias.c b/parser/parser_alias.c
index f5b6da4..d50a72b 100644
--- a/parser/parser_alias.c
+++ b/parser/parser_alias.c
@@ -24,6 +24,7 @@
 #include "immunix.h"
 #include "parser.h"
 #include "profile.h"
+#include "missingdefs.h"
 
 struct alias_rule {
 	char *from;
diff --git a/parser/parser_symtab.c b/parser/parser_symtab.c
index 3e667d8..d5a8270 100644
--- a/parser/parser_symtab.c
+++ b/parser/parser_symtab.c
@@ -24,6 +24,7 @@
 
 #include "immunix.h"
 #include "parser.h"
+#include "missingdefs.h"
 
 enum var_type {
 	sd_boolean,
-- 
2.11.2

