diff --git a/parser/parser_misc.c b/parser/parser_misc.c
index f7772e6..90e8b9c 100644
--- a/parser/parser_misc.c
+++ b/parser/parser_misc.c
@@ -124,6 +124,9 @@ static struct keyword_table rlimit_table[] = {
 	{"core",		RLIMIT_CORE},
 	{"rss",			RLIMIT_RSS},
 	{"nofile",		RLIMIT_NOFILE},
+#ifndef RLIMIT_OFILE
+#define RLIMIT_OFILE RLIMIT_NOFILE
+#endif
 	{"ofile",		RLIMIT_OFILE},
 	{"as",			RLIMIT_AS},
 	{"nproc",		RLIMIT_NPROC},
-- 
2.11.2

