diff --git a/utils/Makefile b/utils/Makefile
index 67caa0d..98fb8d7 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -32,7 +32,6 @@ MANPAGES = ${TOOLS:=.8} logprof.conf.5
 
 all: docs
 	$(MAKE) -C po all
-	$(MAKE) -C vim all
 
 .PHONY: docs
 docs: ${MANPAGES} ${HTMLMANPAGES}
@@ -58,7 +57,6 @@ install: ${MANPAGES} ${HTMLMANPAGES}
 	install -m 755 $(filter-out aa-easyprof, ${TOOLS}) ${BINDIR}
 	$(MAKE) -C po install DESTDIR=${DESTDIR} NAME=${NAME}
 	$(MAKE) install_manpages DESTDIR=${DESTDIR}
-	$(MAKE) -C vim install DESTDIR=${DESTDIR}
 	ln -sf aa-status.8 ${DESTDIR}/${MANDIR}/man8/apparmor_status.8
 	${PYTHON} ${PYSETUP} install --prefix=${PYPREFIX} --root=${DESTDIR} --version=${VERSION}
 
@@ -69,7 +67,6 @@ endif
 clean: pod_clean
 	rm -f core core.* *.o *.s *.a *~
 	$(MAKE) -C po clean
-	$(MAKE) -C vim clean
 	$(MAKE) -C test clean
 	rm -rf staging/ build/
 	rm -f apparmor/*.pyc apparmor/rule/*.pyc
@@ -100,4 +97,3 @@ check: check_severity_db check_pod_files
 		$(PYFLAKES) $$i || exit 1; \
 	done
 	$(MAKE) -C test check
-	$(MAKE) -C vim check
-- 
2.11.2

