diff --git a/utils/test/test-aa.py b/utils/test/test-aa.py
index 94d29f91..5c5645fa 100644
--- a/utils/test/test-aa.py
+++ b/utils/test/test-aa.py
@@ -154,12 +154,12 @@ class AaTest_get_interpreter_and_abstraction(AATest):
     tests = [
         ('#!/bin/bash',             ('/bin/bash',           'abstractions/bash')),
         ('#!/bin/dash',             ('/bin/dash',           'abstractions/bash')),
-        ('#!/bin/sh',               ('/bin/sh',             'abstractions/bash')),
-        ('#!  /bin/sh  ',           ('/bin/sh',             'abstractions/bash')),
-        ('#!  /bin/sh  -x ',        ('/bin/sh',             'abstractions/bash')),  # '-x' is not part of the interpreter path
+        ('#!/bin/sh',               ('/bin/sh',             None)),
+        ('#!  /bin/sh  ',           ('/bin/sh',             None)),
+        ('#!  /bin/sh  -x ',        ('/bin/sh',             None)),  # '-x' is not part of the interpreter path
         ('#!/usr/bin/perl',         ('/usr/bin/perl',       'abstractions/perl')),
         ('#!/usr/bin/perl -w',      ('/usr/bin/perl',       'abstractions/perl')),  # '-w' is not part of the interpreter path
-        ('#!/usr/bin/python',       ('/usr/bin/python',     'abstractions/python')),
+        ('#!/usr/bin/python',       ('/usr/bin/python2.7',  'abstractions/python')),
         ('#!/usr/bin/python2',      ('/usr/bin/python2',    'abstractions/python')),
         ('#!/usr/bin/python2.7',    ('/usr/bin/python2.7',  'abstractions/python')),
         ('#!/usr/bin/python3',      ('/usr/bin/python3',    'abstractions/python')),
-- 
2.16.1

