diff --color -Nur calibre-6.17.0.orig/src/calibre/gui2/update.py calibre-6.17.0/src/calibre/gui2/update.py
--- calibre-6.17.0.orig/src/calibre/gui2/update.py	2023-05-06 11:36:35.678461036 -0700
+++ calibre-6.17.0/src/calibre/gui2/update.py	2023-05-06 11:39:10.365134930 -0700
@@ -82,20 +82,6 @@
         while not self.shutdown_event.is_set():
             calibre_update_version = NO_CALIBRE_UPDATE
             plugins_update_found = 0
-            try:
-                version = get_newest_version()
-                if version[:2] > numeric_version[:2]:
-                    calibre_update_version = version
-            except Exception as e:
-                prints('Failed to check for calibre update:', as_unicode(e))
-            try:
-                update_plugins = get_plugin_updates_available(raise_error=True)
-                if update_plugins is not None:
-                    plugins_update_found = len(update_plugins)
-            except Exception as e:
-                prints('Failed to check for plugin update:', as_unicode(e))
-            if calibre_update_version != NO_CALIBRE_UPDATE or plugins_update_found > 0:
-                self.signal.update_found.emit(calibre_update_version, plugins_update_found)
             self.shutdown_event.wait(self.INTERVAL)
 
     def shutdown(self):

