diff -Nur calibre-8.5.0.orig/setup/build.py calibre-8.5.0/setup/build.py
--- calibre-8.5.0.orig/setup/build.py  20:54:09.140654340 -0600
+++ setup/build.py	2025-06-24 20:54:25.318654245 -0600
@@ -646,7 +646,7 @@
         bdir = self.j(self.build_dir, 'headless')
         if os.path.exists(bdir):
             shutil.rmtree(bdir)
-        cmd = [CMAKE]
+        cmd = [CMAKE, '-GUnix Makefiles']
         if is_macos_universal_build:
             cmd += ['-DCMAKE_OSX_ARCHITECTURES=x86_64;arm64']
         if sw and os.path.exists(os.path.join(sw, 'qt')):
