diff --git a/CMakeLists.txt b/CMakeLists.txt
index 98dc9d9..861a4f8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,7 +346,7 @@ if(LINPHONE_BUILDER_GROUP_EXTERNAL_SOURCE_PATH_BUILDERS)
 	set(BcToolbox_FIND_COMPONENTS tester)
 	include("${EP_bctoolbox_CONFIG_DIR}/BcToolboxConfig.cmake")
 else()
-	find_package(BcToolbox 0.0.3 REQUIRED OPTIONAL_COMPONENTS tester)
+	find_package(BcToolbox 4.3.0 REQUIRED OPTIONAL_COMPONENTS tester)
 endif()
 
 include_directories(
