diff --git a/conf/example.gitolite.rc b/conf/example.gitolite.rc
index d800357..95c5a24 100644
--- a/conf/example.gitolite.rc
+++ b/conf/example.gitolite.rc
@@ -19,8 +19,8 @@ $GL_CONF_COMPILED="$GL_ADMINDIR/conf/gitolite.conf-compiled.pm";
 # DO NOT CHANGE THE NEXT FOUR LINES UNLESS YOU REALLY KNOW WHAT YOU'RE DOING.
 # These variables are set automatically by the install method you choose.
 #            (PACKAGE MAINTAINERS: PLEASE READ doc/packaging.mkd)
-# $GL_PACKAGE_CONF = "";
-# $GL_PACKAGE_HOOKS = "";
+$GL_PACKAGE_CONF = "/etc/gitolite";
+$GL_PACKAGE_HOOKS = "/usr/share/gitolite/hooks";
 
 # ------------------------------------------------------------------------------
 # most often used/changed variables
diff --git a/src/gl-setup b/src/gl-setup
index fd8357b..59de183 100755
--- a/src/gl-setup
+++ b/src/gl-setup
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-GL_PACKAGE_CONF=/tmp/share/gitolite/conf
+GL_PACKAGE_CONF=/etc/gitolite
 # must be the same as the value for the same variable in
 # $GL_PACKAGE_CONF/example.gitolite.rc.  Sorry about the catch-22 :)
 
