diff --git a/cpan/Time-Piece/t/02core_dst.t b/cpan/Time-Piece/t/02core_dst.t
index 3f54fff..4b1faa8 100644
--- a/cpan/Time-Piece/t/02core_dst.t
+++ b/cpan/Time-Piece/t/02core_dst.t
@@ -119,7 +119,7 @@ cmp_ok($t->month_last_day, '==', 31); # test more
 
 
 SKIP: {
-	skip "Extra tests for Linux, BSD only.", 8 unless $is_linux or $is_mac or $is_bsd;
+	skip "Extra tests for Linux, BSD only.", 8 unless $is_mac or $is_bsd;
 
     local $ENV{TZ} = "EST5EDT4";
     Time::Piece::_tzset();
