diff -Naur a/gio/glib-compile-schemas.c b/gio/glib-compile-schemas.c
--- a/gio/glib-compile-schemas.c	2018-09-21 15:23:52.000000000 +0100
+++ b/gio/glib-compile-schemas.c	2019-02-10 14:37:30.034879344 +0000
@@ -1233,19 +1233,6 @@
       return;
     }
 
-  if (path && (g_str_has_prefix (path, "/apps/") ||
-               g_str_has_prefix (path, "/desktop/") ||
-               g_str_has_prefix (path, "/system/")))
-    {
-      gchar *message = NULL;
-      message = g_strdup_printf (_("Warning: Schema “%s” has path “%s”.  "
-                                   "Paths starting with "
-                                   "“/apps/”, “/desktop/” or “/system/” are deprecated."),
-                                 id, path);
-      g_printerr ("%s\n", message);
-      g_free (message);
-    }
-
   state->schema_state = schema_state_new (path, gettext_domain,
                                           extends, extends_name, list_of);
 
