--- a/config.m4	2006-11-07 22:31:19.000000000 +0100
+++ b/config.m4	2009-04-19 20:23:37.923512186 +0200
@@ -47,8 +47,8 @@
     -L$FILEINFO_DIR/lib
   ])
 
-  MAGIC_MIME_DIRS="/usr/local/share/file /usr/share/file /usr/share/misc/file /etc /usr/share/misc"
-  MAGIC_MIME_FILENAMES="magic magic.mime"
+  MAGIC_MIME_DIRS="/usr/share/misc/file"
+  MAGIC_MIME_FILENAMES="magic.mgc"
   
   for i in $MAGIC_MIME_DIRS; do
     for j in $MAGIC_MIME_FILENAMES; do
