--- dpkg-1.18.4.orig/ostable
+++ dpkg-1.18.4/ostable
@@ -15,6 +15,7 @@
 uclibceabi-linux	linux-uclibceabi	linux[^-]*-uclibceabi
 uclibc-linux		linux-uclibc		linux[^-]*-uclibc
 musleabihf-linux	linux-musleabihf	linux[^-]*-musleabihf
+musleabihf-linux	linux-musleabihf	linux[^-]*-muslgnueabihf
 musl-linux		linux-musl		linux[^-]*-musl
 gnueabihf-linux		linux-gnueabihf		linux[^-]*-gnueabihf
 gnueabi-linux		linux-gnueabi		linux[^-]*-gnueabi
