--- a/lib/dpkg/t/t-file.c
+++ b/lib/dpkg/t/t-file.c
@@ -67,7 +67,7 @@ test_file_realpath(void)
 	free(path);
 
 	path = file_realpath("//./..///../././..///..");
-	test_str(path, ==, "/");
+	test_str(path, ==, "//");
 	free(path);
 }
 
@@ -86,7 +87,7 @@ test_file_canonicalize(void)
 	free(path);
 
 	path = file_canonicalize("//./..///../././..///..");
-	test_str(path, ==, "/");
+	test_str(path, ==, "//");
 	free(path);
 
 	/* Canonicalize filenames that do not exist. */
