diff --git a/test/unit/utVersion.cpp b/test/unit/utVersion.cpp
index 5826b28..30301f7 100644
--- a/test/unit/utVersion.cpp
+++ b/test/unit/utVersion.cpp
@@ -61,17 +61,13 @@ TEST_F( utVersion, aiGetVersionMajorTest ) {
 }
 
 TEST_F( utVersion, aiGetVersionPatchTest ) {
     EXPECT_EQ(aiGetVersionPatch(), 0U );
 }
 
 TEST_F( utVersion, aiGetCompileFlagsTest ) {
     EXPECT_NE( aiGetCompileFlags(), 0U );
 }
 
-TEST_F( utVersion, aiGetVersionRevisionTest ) {
-    EXPECT_NE( aiGetVersionRevision(), 0U );
-}
-
 TEST_F( utVersion, aiGetBranchNameTest ) {
     EXPECT_NE( nullptr, aiGetBranchName() );
 }
