diff --git a/CMakeLists.txt b/CMakeLists.txt
index 458a328..62bbd0a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -56,7 +56,7 @@ IF(ASSIMP_HUNTER_ENABLED)
   add_definitions(-DASSIMP_USE_HUNTER)
 ENDIF()
 
-PROJECT(Assimp VERSION 5.2.4)
+PROJECT(Assimp VERSION 5.2.5)
 
 # All supported options ###############################################
 
diff --git a/test/unit/utVersion.cpp b/test/unit/utVersion.cpp
index 5826b28..30301f7 100644
--- a/test/unit/utVersion.cpp
+++ b/test/unit/utVersion.cpp
@@ -61,17 +61,13 @@ TEST_F( utVersion, aiGetVersionMajorTest ) {
 }
 
 TEST_F( utVersion, aiGetVersionPatchTest ) {
-    EXPECT_EQ(aiGetVersionPatch(), 4U );
+    EXPECT_EQ(aiGetVersionPatch(), 5U );
 }
 
 TEST_F( utVersion, aiGetCompileFlagsTest ) {
     EXPECT_NE( aiGetCompileFlags(), 0U );
 }
 
-TEST_F( utVersion, aiGetVersionRevisionTest ) {
-    EXPECT_NE( aiGetVersionRevision(), 0U );
-}
-
 TEST_F( utVersion, aiGetBranchNameTest ) {
     EXPECT_NE( nullptr, aiGetBranchName() );
 }
