--- a/python-ethtool/etherinfo.c
+++ b/python-ethtool/etherinfo.c
@@ -18,7 +18,6 @@
 #include <Python.h>
 #include "include/py3c/compat.h"
 #include <bytesobject.h>
-#include <bits/sockaddr.h>
 #include <stdio.h>
 #include <string.h>
 #include <sys/types.h>
--- a/python-ethtool/ethtool.c
+++ b/python-ethtool/ethtool.c
@@ -45,9 +45,9 @@
 #endif
 
 typedef unsigned long long u64;
-typedef __uint32_t u32;
-typedef __uint16_t u16;
-typedef __uint8_t u8;
+typedef uint32_t u32;
+typedef uint16_t u16;
+typedef uint8_t u8;
 
 #include "ethtool-copy.h"
 #include <linux/sockios.h>  /* for SIOCETHTOOL */
