--- a/src/lib/utils.c
+++ b/src/lib/utils.c
@@ -107,7 +107,7 @@
 	return defval;
 }
 
-uint64_t get_uint64(const char *path, uint64_t defval, const char *name)
+uint64_t get_uint64(const char *path, unsigned long defval, const char *name)
 {
 	char buf[BUF_SZ_NUM];
 	char *p = get_text_to_dest(path, name, buf, sizeof(buf));
--- a/src/lib/utils.h
+++ b/src/lib/utils.h
@@ -138,7 +138,7 @@
  * @return Value read from a file if successful, otherwise a value stored in
  *         defval argument.
  */
-uint64_t get_uint64(const char *path, uint64_t defval, const char *name);
+uint64_t get_uint64(const char *path, unsigned long defval, const char *name);
 
 /**
  * @brief Reads a content of a text file.
