Fix `configure syntax error: unexpected word (expecting ")")`.

--- ledmon-1.0.0-origin/configure.ac
+++ ledmon-1.0.0/configure.ac
@@ -69,10 +69,6 @@
 
 # Add linker flags.
 AC_DEFUN([AX_AM_LDFLAGS_ADD],[AX_CHECK_LINK_FLAG($1, AM_LDFLAGS="$AM_LDFLAGS $1")])
-AX_AM_LDFLAGS_ADD([-pie])
-AX_AM_LDFLAGS_ADD([[-Wl,-z,relro]])
-AX_AM_LDFLAGS_ADD([[-Wl,-z,now]])
-AX_AM_LDFLAGS_ADD([[-Wl,-z,noexecstack]])
 
 # Add library compiler flags.
 AM_LIB_CFLAGS=''
