Relax some constraints to support GHC 9.14.1.

diff --git a/cabal-install-solver/cabal-install-solver.cabal b/cabal-install-solver/cabal-install-solver.cabal
index eaa060b47..18de666f8 100644
--- a/cabal-install-solver/cabal-install-solver.cabal
+++ b/cabal-install-solver/cabal-install-solver.cabal
@@ -99,7 +99,7 @@ library
 
   build-depends:
     , array         >=0.4      && <0.6
-    , base          >=4.13     && <4.22
+    , base          >=4.13     && <5
     , bytestring    >=0.10.6.0 && <0.13
     , Cabal         ^>=3.16.1.0
     , Cabal-syntax  ^>=3.16.1.0
diff --git a/cabal-install/cabal-install.cabal b/cabal-install/cabal-install.cabal
index e27feba83..2b91b3869 100644
--- a/cabal-install/cabal-install.cabal
+++ b/cabal-install/cabal-install.cabal
@@ -58,7 +58,7 @@ common warnings
 
 common base-dep
     build-depends:
-      , base >=4.13 && <4.22
+      , base >=4.13 && <5
 
 common cabal-dep
     build-depends:
@@ -243,7 +243,7 @@ library
       , edit-distance >= 0.2.2 && < 0.3
       , exceptions >= 0.10.4   && < 0.11
       , filepath   >= 1.4.0.0  && < 1.6
-      , HTTP       >= 4000.1.5 && < 4000.5
+      , HTTP       >= 4000.1.5 && < 4000.6
       , mtl        >= 2.0      && < 2.4
       , network-uri >= 2.6.2.0 && < 2.7
       , pretty     >= 1.1      && < 1.2
