--- a/src/deliver.c	2018-04-15 02:18:10.000000000 +0300
+++ b/src/deliver.c	2018-06-15 14:01:58.196926447 +0300
@@ -7291,7 +7291,7 @@
 	"MIME-Version: 1.0\n\n"
 
 	"--%s\n"
-	"Content-type: text/plain; charset=us-ascii\n\n"
+	"Content-type: text/plain; charset=utf-8\n\n"
 
 	"This message was created automatically by mail delivery software.\n"
 	" ----- The following addresses had successful delivery notifications -----\n",
@@ -7560,7 +7560,7 @@
 
       /* output human readable part as text/plain section */
       fprintf(f, "--%s\n"
-	  "Content-type: text/plain; charset=us-ascii\n\n",
+	  "Content-type: text/plain; charset=utf-8\n\n",
 	bound);
 
       if ((emf_text = next_emf(emf, US"intro")))
@@ -8163,7 +8163,7 @@
 
         /* output human readable part as text/plain section */
         fprintf(f, "--%s\n"
-	    "Content-type: text/plain; charset=us-ascii\n\n",
+	    "Content-type: text/plain; charset=utf-8\n\n",
 	  bound);
 
         if ((wmf_text = next_emf(wmf, US"intro")))
