diff --git a/src/devicestate/devicestate.cpp b/src/devicestate/devicestate.cpp
index 8535c304..864ddafc 100644
--- a/src/devicestate/devicestate.cpp
+++ b/src/devicestate/devicestate.cpp
@@ -34,7 +34,7 @@
 
 #include <dsme/thermalmanager_dbus_if.h>
 #ifdef HAVE_SAILFISHUSERMANAGER
-#include <sailfishusermanagerinterface.h>
+#include <sailfishusermanager/sailfishusermanagerinterface.h>
 #endif
 
 #include <QDBusConnection>
