diff --git a/src/capture_eep.c b/src/capture_eep.c
index 964a9a8..a420d50 100644
--- a/src/capture_eep.c
+++ b/src/capture_eep.c
@@ -719,14 +719,14 @@ capture_eep_receive_v3(const u_char *pkt, uint32_t size)
 {
 
     struct hep_generic hg;
+    char *password = NULL;
+    int password_len = 0;
     hep_chunk_ip4_t src_ip4, dst_ip4;
 #ifdef USE_IPV6
     hep_chunk_ip6_t src_ip6, dst_ip6;
 #endif
     hep_chunk_t payload_chunk;
     hep_chunk_t authkey_chunk;
-    char *password;
-    int password_len;
     unsigned char *payload = 0;
     uint32_t total_len, pos;
     char buffer[MAX_CAPTURE_LEN] ;
@@ -843,7 +843,12 @@ capture_eep_receive_v3(const u_char *pkt, uint32_t size)
             case CAPTURE_EEP_CHUNK_AUTH_KEY:
                 memcpy(&authkey_chunk, (void*) buffer + pos, sizeof(authkey_chunk));
                 password_len = ntohs(authkey_chunk.length) - sizeof(authkey_chunk);
+                if (password_len <= 0) {
+                    pos += chunk_len;
+                    continue;
+                }
                 password = sng_malloc(password_len);
+                password[password_len - 1] = '\0';
                 memcpy(password, (void*) buffer + pos + sizeof(hep_chunk_t), password_len);
                 break;
             case CAPTURE_EEP_CHUNK_PAYLOAD:
@@ -865,16 +870,21 @@ capture_eep_receive_v3(const u_char *pkt, uint32_t size)
         pos += chunk_len;
     }
 
-    // Validate password
+    // Validate password SAFELY
     if (eep_cfg.capt_srv_password != NULL) {
         // No password in packet
-        if (strlen(password) == 0)
+        if (!password || strlen(password) == 0)
             return NULL;
         // Check password matches configured
         if (strcmp(password, eep_cfg.capt_srv_password) != 0)
             return NULL;
     }
 
+    if (!payload) {
+        sng_free(password);
+        return NULL;
+    }
+
     // Build a custom frame pcap header
     frame_pcap_header = capture_eep_build_frame_data(header, payload,header.caplen, src, dst, &frame_payload);
     if (!frame_payload) {
@@ -892,8 +902,11 @@ capture_eep_receive_v3(const u_char *pkt, uint32_t size)
     sng_free(frame_payload);
 
     /* FREE */
-    sng_free(payload);
-    sng_free(password);
+    if (payload)
+        sng_free(payload);
+    
+    if (password)
+        sng_free(password);
 
     return pkt_new;
 }
