From ea2ef8fe05fa0ff60bd11ff6db0d98b7cbb78299 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Wed, 7 Jun 2023 12:03:31 +0200
Subject: [PATCH] docs: use qdoc-qt5 directly instead of qtswitcher

---
 docs/qml/CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/docs/qml/CMakeLists.txt b/docs/qml/CMakeLists.txt
index 8fd39d22..0f6c29a6 100644
--- a/docs/qml/CMakeLists.txt
+++ b/docs/qml/CMakeLists.txt
@@ -1,10 +1,9 @@
 # add a target to generate API documentation with qdoc 5.x
 SET(QDOC_WORKS FALSE)
-SET(ENV{QT_SELECT} qt5)
-FIND_PROGRAM(QDOC_EXECUTABLE qdoc)
+FIND_PROGRAM(QDOC_EXECUTABLE qdoc-qt5)
 if(QDOC_EXECUTABLE)
 EXECUTE_PROCESS(COMMAND ${QDOC_EXECUTABLE} --version OUTPUT_VARIABLE QDOC_OUTPUT ERROR_QUIET)
-string(REGEX REPLACE "qdoc ([0-9]+(\\.[0-9]+)+).*" "\\1" QDOC_VERSION ${QDOC_OUTPUT})
+string(REGEX REPLACE "qdoc-qt5 ([0-9]+(\\.[0-9]+)+).*" "\\1" QDOC_VERSION ${QDOC_OUTPUT})
 if(QDOC_VERSION MATCHES "^5\\.")
 set(QDOC_WORKS TRUE)
 endif()
-- 
2.43.0

