Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: define SOVERSION
----

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -153,7 +153,7 @@
 
 set(BASISU_LIB_TARGET "basisu_lib")
 add_library(${BASISU_LIB_TARGET} ${BASISU_SRC_LIST})
-set_target_properties(${BASISU_LIB_TARGET} PROPERTIES OUTPUT_NAME "basisu")
+set_target_properties(${BASISU_LIB_TARGET} PROPERTIES OUTPUT_NAME "basisu" VERSION "${PROJECT_VERSION}" SOVERSION "${PROJECT_VERSION}")
 target_include_directories(${BASISU_LIB_TARGET} PUBLIC
                            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/transcoder>
                            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/encoder>
