passes -G Ninja to make build go more faster
diff --git a/setup.py b/setup.py
index 68c45398..c9fc98ff 100644
--- a/setup.py
+++ b/setup.py
@@ -175,6 +175,7 @@ class awscrt_build_ext(setuptools.command.build_ext.build_ext):
         cmake_args = [cmake]
         cmake_args.append(f'-H{source_dir}')
         cmake_args.append(f'-B{build_dir}')
+        cmake_args.append('-G Ninja')
         cmake_args.extend(determine_generator_args())
         cmake_args.extend(determine_cross_compile_args())
         cmake_args.extend([
