diff --git a/crt/aws-lc/CMakeLists.txt b/crt/aws-lc/CMakeLists.txt
index 389de069..90ee5d4b 100644
--- a/crt/aws-lc/CMakeLists.txt
+++ b/crt/aws-lc/CMakeLists.txt
@@ -92,6 +92,9 @@ if (NOT PERL_EXECUTABLE)
   message(STATUS "Perl not found. Disabling some code generation and using pre-generated code in generated-src/")
 endif()
 
+set(GO_EXECUTABLE go)
+set(PERL_EXECUTABLE perl)
+
 if(NOT PERL_EXECUTABLE OR NOT GO_EXECUTABLE)
   set(GENERATE_CODE_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/generated-src")
 endif()
