--- a/systems/fbdev/fbdev.c
+++ b/systems/fbdev/fbdev.c
@@ -2327,14 +2327,18 @@ dfb_fbdev_set_gamma_ramp( DFBSurfacePixelFormat format )
           for (i = 0; i < blue_size; i++)
                cmap->blue[i] |= cmap->blue[i] << 8;
      }
-
+     /*
+      * Commenting out section setting gamma ramp
+      * per: http://directfb-users.directfb.narkive.com/fUkXSRmm/hello
+      */
+     /*
      if (FBDEV_IOCTL( FBIOPUTCMAP, cmap ) < 0) {
           D_PERROR( "DirectFB/FBDev: "
                      "Could not set gamma ramp" );
 
           return errno2result(errno);
      }
-
+     */
      return DFB_OK;
 }
 
