diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 8bd8298..620a44a 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2672,6 +2672,7 @@ Tool *Solaris::buildLinker() const {
 /// Distribution (very bare-bones at the moment).

 enum Distro {
+  AlpineLinux,
   ArchLinux,
   DebianLenny,
   DebianSqueeze,
@@ -2783,6 +2784,9 @@ static Distro DetectDistro(llvm::Triple::ArchType Arch) {
   if (llvm::sys::fs::exists("/etc/arch-release"))
     return ArchLinux;

+  if (llvm::sys::fs::exists("/etc/alpine-release"))
+    return AlpineLinux;
+
   return UnknownDistro;
 }

--
2.1.4
