diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 620a44a..006915c 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2939,7 +2939,7 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)

   Distro Distro = DetectDistro(Arch);

-  if (IsOpenSUSE(Distro) || IsUbuntu(Distro)) {
+  if (IsOpenSUSE(Distro) || IsUbuntu(Distro) || Distro == AlpineLinux) {
     ExtraOpts.push_back("-z");
     ExtraOpts.push_back("relro");
   }
--
2.1.4
