diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index f789fd5..6d75b4b 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -2667,12 +2673,13 @@ Tool *Solaris::buildLinker() const {
   return new tools::solaris::Link(*this);
 }

 /// Distribution (very bare-bones at the moment).

 enum Distro {
+  AlpineLinux,
   ArchLinux,
   DebianLenny,
   DebianSqueeze,
   DebianWheezy,
   DebianJessie,
   Exherbo,
@@ -2778,12 +2785,15 @@ static Distro DetectDistro(llvm::Triple::ArchType Arch) {
   if (llvm::sys::fs::exists("/etc/exherbo-release"))
     return Exherbo;

   if (llvm::sys::fs::exists("/etc/arch-release"))
     return ArchLinux;

+  if (llvm::sys::fs::exists("/etc/alpine-release"))
+    return AlpineLinux;
+
   return UnknownDistro;
 }

 /// \brief Get our best guess at the multiarch triple for a target.
 ///
 /// Debian-based systems are starting to use a multiarch setup where they use
