--- a/Makefile
+++ b/Makefile
@@ -495,7 +493,6 @@
 
 clean: | $(CLEAN_TARGETS)
 	@$(MAKE) -C base clean
-	@$(MAKE) -C doc clean
 	@$(MAKE) -C src clean
 	@$(MAKE) -C ui clean
 	@$(MAKE) -C test clean
@@ -517,7 +514,6 @@
 
 distcleanall: cleanall
 	@$(MAKE) -C deps distcleanall
-	@$(MAKE) -C doc cleanall
 	rm -fr $(build_prefix) $(build_staging)
 
 .PHONY: default debug release check-whitespace release-candidate \
