diff -up CEGUI-0.6.2/ImageCodecModules/CoronaImageCodec/Makefile.am.patch1 CEGUI-0.6.2/ImageCodecModules/CoronaImageCodec/Makefile.am
--- CEGUI-0.6.2/ImageCodecModules/CoronaImageCodec/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/ImageCodecModules/CoronaImageCodec/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -4,6 +4,8 @@ pkgincludedir=$(prefix)/include/CEGUI/Im
 if MINGW32_BUILD
 libCEGUICoronaImageCodec_la_CPPFLAGS = -DCEGUICORONAIMAGECODEC_EXPORTS
 libCEGUICoronaImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUICoronaImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUICoronaImageCodec_la_LIBADD = $(Corona_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.am.patch1 CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.am
--- CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -5,6 +5,8 @@ pkgincludedir=$(prefix)/include/CEGUI/Im
 if MINGW32_BUILD
 libCEGUIDevILImageCodec_la_CPPFLAGS = -DCEGUIDEVILIMAGECODEC_EXPORTS
 libCEGUIDevILImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUIDevILImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUIDevILImageCodec_la_LIBADD = $(DevIL_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.in.patch1 CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.in
--- CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.in.patch1	2008-11-30 21:11:37.000000000 +0100
+++ CEGUI-0.6.2/ImageCodecModules/DevILImageCodec/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -246,6 +246,7 @@ INCLUDES = -I$(top_srcdir)/include $(Dev
 
 @MINGW32_BUILD_TRUE@libCEGUIDevILImageCodec_la_CPPFLAGS = -DCEGUIDEVILIMAGECODEC_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUIDevILImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUIDevILImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUIDevILImageCodec_la_LIBADD = $(DevIL_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.am.patch1 CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.am
--- CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -4,6 +4,8 @@ pkgincludedir=$(prefix)/include/CEGUI/Im
 if MINGW32_BUILD
 libCEGUIFreeImageImageCodec_la_CPPFLAGS = -DCEGUIFREEIMAGEIMAGECODEC_EXPORTS
 libCEGUIFreeImageImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUIFreeImageImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUIFreeImageImageCodec_la_LIBADD = $(FreeImage_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.in.patch1 CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.in
--- CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.in.patch1	2008-11-30 21:11:37.000000000 +0100
+++ CEGUI-0.6.2/ImageCodecModules/FreeImageImageCodec/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -245,6 +245,7 @@ INCLUDES = -I$(top_srcdir)/include $(Fre
 
 @MINGW32_BUILD_TRUE@libCEGUIFreeImageImageCodec_la_CPPFLAGS = -DCEGUIFREEIMAGEIMAGECODEC_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUIFreeImageImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUIFreeImageImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUIFreeImageImageCodec_la_LIBADD = $(FreeImage_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.am.patch1 CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.am
--- CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -4,6 +4,8 @@ pkgincludedir=$(prefix)/include/CEGUI/Im
 if MINGW32_BUILD
 libCEGUISILLYImageCodec_la_CPPFLAGS = -DCEGUISILLYIMAGECODEC_EXPORTS
 libCEGUISILLYImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUISILLYImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUISILLYImageCodec_la_LIBADD = $(SILLY_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.in.patch1 CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.in
--- CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.in.patch1	2008-11-30 21:11:37.000000000 +0100
+++ CEGUI-0.6.2/ImageCodecModules/SILLYImageCodec/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -245,6 +245,7 @@ INCLUDES = -I$(top_srcdir)/include $(SIL
 
 @MINGW32_BUILD_TRUE@libCEGUISILLYImageCodec_la_CPPFLAGS = -DCEGUISILLYIMAGECODEC_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUISILLYImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUISILLYImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUISILLYImageCodec_la_LIBADD = $(SILLY_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.am.patch1 CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.am
--- CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -4,6 +4,8 @@ pkgincludedir=$(prefix)/include/CEGUI/Im
 if MINGW32_BUILD
 libCEGUITGAImageCodec_la_CPPFLAGS = -DCEGUITGAIMAGECODEC_EXPORTS
 libCEGUITGAImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUITGAImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 lib_LTLIBRARIES=libCEGUITGAImageCodec.la
diff -up CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.in.patch1 CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.in
--- CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.in.patch1	2008-11-30 21:11:37.000000000 +0100
+++ CEGUI-0.6.2/ImageCodecModules/TGAImageCodec/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -245,6 +245,7 @@ INCLUDES = -I$(top_srcdir)/include 
 
 @MINGW32_BUILD_TRUE@libCEGUITGAImageCodec_la_CPPFLAGS = -DCEGUITGAIMAGECODEC_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUITGAImageCodec_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUITGAImageCodec_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 lib_LTLIBRARIES = libCEGUITGAImageCodec.la
 libCEGUITGAImageCodec_la_SOURCES = CEGUITGAImageCodec.cpp CEGUITGAImageCodecModule.cpp
diff -up CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.am.patch1 CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.am
--- CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.am.patch1	2008-07-21 12:45:32.000000000 +0200
+++ CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -23,7 +23,7 @@ endif
 
 libCEGUIOpenGLRenderer_la_LIBADD = $(OpenGL_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
-libCEGUIOpenGLRenderer_la_LDFLAGS += -version-info @CEGUI_OPENGL_VERSION_INFO@
+libCEGUIOpenGLRenderer_la_LDFLAGS += -release @PACKAGE_VERSION@
 
 lib_LTLIBRARIES=libCEGUIOpenGLRenderer.la
 libCEGUIOpenGLRenderer_la_SOURCES = openglrenderer.cpp opengltexture.cpp
diff -up CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.in.patch1 CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.in
--- CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.in.patch1	2008-11-30 21:11:37.000000000 +0100
+++ CEGUI-0.6.2/RendererModules/OpenGLGUIRenderer/Makefile.in	2008-12-03 12:15:36.000000000 +0100
@@ -250,7 +250,7 @@ xerces_LIBS = @xerces_LIBS@
 INCLUDES = $(am__append_1) -I$(top_srcdir)/include -I$(top_srcdir) $(OpenGL_CFLAGS)
 pkginclude_HEADERS = openglrenderer.h opengltexture.h
 
-libCEGUIOpenGLRenderer_la_LDFLAGS = $(am__append_2) -version-info @CEGUI_OPENGL_VERSION_INFO@
+libCEGUIOpenGLRenderer_la_LDFLAGS = $(am__append_2) -release @PACKAGE_VERSION@
 
 @MINGW32_BUILD_TRUE@libCEGUIOpenGLRenderer_la_CPPFLAGS = -DOPENGL_GUIRENDERER_EXPORTS $(am__append_3)
 
diff -up CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am.patch1 CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am
--- CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am.patch1	2008-05-26 20:56:07.000000000 +0200
+++ CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -9,7 +9,7 @@ libCEGUILuaScriptModule_la_CPPFLAGS += -
 libCEGUILuaScriptModule_la_LDFLAGS += -no-undefined -avoid-version
 endif
 
-libCEGUILuaScriptModule_la_LDFLAGS += -version-info @CEGUI_VERSION_INFO@
+libCEGUILuaScriptModule_la_LDFLAGS += -release @PACKAGE_VERSION@
 
 
 lib_LTLIBRARIES = libCEGUILuaScriptModule.la
diff -up CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.in.patch1 CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.in
--- CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.in.patch1	2008-11-30 21:11:40.000000000 +0100
+++ CEGUI-0.6.2/ScriptingModules/CEGUILua/LuaScriptModule/src/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -251,7 +251,7 @@ INCLUDES = -I$(top_srcdir)/include \
            -I$(top_srcdir)/ScriptingModules/CEGUILua/LuaScriptModule/include\
 $(am__append_3) $(am__append_5)
 
-libCEGUILuaScriptModule_la_LDFLAGS = $(am__append_2) -version-info @CEGUI_VERSION_INFO@
+libCEGUILuaScriptModule_la_LDFLAGS = $(am__append_2) -release @PACKAGE_VERSION@
 libCEGUILuaScriptModule_la_CPPFLAGS = $(Lua_CFLAGS) $(am__append_1)
 
 lib_LTLIBRARIES = libCEGUILuaScriptModule.la
diff -up CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.am.patch1 CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.am
--- CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.am.patch1	2008-03-15 17:51:39.000000000 +0100
+++ CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -7,7 +7,7 @@ libCEGUItoluapp_la_CPPFLAGS = -DTOLUA_EX
 libCEGUItoluapp_la_LDFLAGS += -no-undefined
 endif
 
-libCEGUItoluapp_la_LDFLAGS += -version-info @CEGUI_VERSION_INFO@
+libCEGUItoluapp_la_LDFLAGS += -release @PACKAGE_VERSION@
 
 lib_LTLIBRARIES = libCEGUItoluapp.la
 
diff -up CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.in.patch1 CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.in
--- CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.in.patch1	2008-11-30 21:11:40.000000000 +0100
+++ CEGUI-0.6.2/ScriptingModules/CEGUILua/tolua++/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -244,7 +244,7 @@ xerces_CFLAGS = @xerces_CFLAGS@
 xerces_LIBS = @xerces_LIBS@
 INCLUDES = $(Lua_CFLAGS) -I$(top_srcdir)/ScriptingModules/CEGUILua/tolua++
 
-libCEGUItoluapp_la_LDFLAGS = $(am__append_1) -version-info @CEGUI_VERSION_INFO@
+libCEGUItoluapp_la_LDFLAGS = $(am__append_1) -release @PACKAGE_VERSION@
 
 @MINGW32_BUILD_TRUE@libCEGUItoluapp_la_CPPFLAGS = -DTOLUA_EXPORTS
 
diff -up CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.am.patch1 CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.am
--- CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -8,7 +8,7 @@ libCEGUIFalagardWRBase_la_CPPFLAGS = -DF
 libCEGUIFalagardWRBase_la_LDFLAGS += -no-undefined -avoid-version
 endif
 
-libCEGUIFalagardWRBase_la_LDFLAGS += -version-info @CEGUI_VERSION_INFO@
+libCEGUIFalagardWRBase_la_LDFLAGS += -release @PACKAGE_VERSION@
 
 libCEGUIFalagardWRBase_la_LIBADD = $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.in.patch1 CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.in
--- CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.in.patch1	2008-11-30 21:11:41.000000000 +0100
+++ CEGUI-0.6.2/WindowRendererSets/Falagard/src/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -246,7 +246,7 @@ INCLUDES = -I$(top_srcdir)/include \
            -I$(top_srcdir)/WindowRendererSets/Falagard/include
 
 
-libCEGUIFalagardWRBase_la_LDFLAGS = $(am__append_1) -version-info @CEGUI_VERSION_INFO@
+libCEGUIFalagardWRBase_la_LDFLAGS = $(am__append_1) -release @PACKAGE_VERSION@
 
 @MINGW32_BUILD_TRUE@libCEGUIFalagardWRBase_la_CPPFLAGS = -DFALAGARDWRBASE_EXPORTS -DCEGUIWRMODULE_EXPORTS
 
diff -up CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.am.patch1 CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.am
--- CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.am.patch1	2008-05-26 20:56:07.000000000 +0200
+++ CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -3,6 +3,8 @@ INCLUDES = -I$(top_srcdir)/include
 if MINGW32_BUILD
 libCEGUITinyXMLParser_la_CPPFLAGS = -DCEGUITINYXMLPARSER_EXPORTS
 libCEGUITinyXMLParser_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUITinyXMLParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 lib_LTLIBRARIES=libCEGUITinyXMLParser.la
diff -up CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.in.patch1 CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.in
--- CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.in.patch1	2008-11-30 21:11:41.000000000 +0100
+++ CEGUI-0.6.2/XMLParserModules/TinyXMLParser/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -253,6 +253,7 @@ INCLUDES = -I$(top_srcdir)/include $(am_
 
 @MINGW32_BUILD_TRUE@libCEGUITinyXMLParser_la_CPPFLAGS = -DCEGUITINYXMLPARSER_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUITinyXMLParser_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUITinyXMLParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 lib_LTLIBRARIES = libCEGUITinyXMLParser.la
 libCEGUITinyXMLParser_la_LIBADD = $(top_builddir)/src/libCEGUIBase.la $(am__append_3)
diff -up CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.am.patch1 CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.am
--- CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -3,6 +3,8 @@ INCLUDES = -I$(top_srcdir)/include $(xer
 if MINGW32_BUILD
 libCEGUIXercesParser_la_CPPFLAGS = -DCEGUIXERCESPARSER_EXPORTS
 libCEGUIXercesParser_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUIXercesParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUIXercesParser_la_LIBADD = $(xerces_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.in.patch1 CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.in
--- CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.in.patch1	2008-11-30 21:11:41.000000000 +0100
+++ CEGUI-0.6.2/XMLParserModules/XercesParser/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -245,6 +245,7 @@ INCLUDES = -I$(top_srcdir)/include $(xer
 
 @MINGW32_BUILD_TRUE@libCEGUIXercesParser_la_CPPFLAGS = -DCEGUIXERCESPARSER_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUIXercesParser_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUIXercesParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUIXercesParser_la_LIBADD = $(xerces_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.am.patch1 CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.am
--- CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -3,6 +3,8 @@ INCLUDES = -I$(top_srcdir)/include $(exp
 if MINGW32_BUILD
 libCEGUIExpatParser_la_CPPFLAGS = -DCEGUIEXPATPARSER_EXPORTS
 libCEGUIExpatParser_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUIExpatParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUIExpatParser_la_LIBADD = $(expat_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.in.patch1 CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.in
--- CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.in.patch1	2008-11-30 21:11:41.000000000 +0100
+++ CEGUI-0.6.2/XMLParserModules/expatParser/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -245,6 +245,7 @@ INCLUDES = -I$(top_srcdir)/include $(exp
 
 @MINGW32_BUILD_TRUE@libCEGUIExpatParser_la_CPPFLAGS = -DCEGUIEXPATPARSER_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUIExpatParser_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUIExpatParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUIExpatParser_la_LIBADD = $(expat_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.am.patch1 CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.am
--- CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.am.patch1	2008-05-25 10:25:55.000000000 +0200
+++ CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -3,6 +3,8 @@ INCLUDES = -I$(top_srcdir)/include $(lib
 if MINGW32_BUILD
 libCEGUILibxmlParser_la_CPPFLAGS = -DCEGUILIBXMLPARSER_EXPORTS
 libCEGUILibxmlParser_la_LDFLAGS = -no-undefined -avoid-version
+else
+libCEGUILibxmlParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 endif
 
 libCEGUILibxmlParser_la_LIBADD = $(libxml_LIBS) $(top_builddir)/src/libCEGUIBase.la
diff -up CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.in.patch1 CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.in
--- CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.in.patch1	2008-11-30 21:11:41.000000000 +0100
+++ CEGUI-0.6.2/XMLParserModules/libxmlParser/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -246,6 +246,7 @@ INCLUDES = -I$(top_srcdir)/include $(lib
 
 @MINGW32_BUILD_TRUE@libCEGUILibxmlParser_la_CPPFLAGS = -DCEGUILIBXMLPARSER_EXPORTS
 @MINGW32_BUILD_TRUE@libCEGUILibxmlParser_la_LDFLAGS = -no-undefined -avoid-version
+@MINGW32_BUILD_FALSE@libCEGUILibxmlParser_la_LDFLAGS = -release @PACKAGE_VERSION@
 
 libCEGUILibxmlParser_la_LIBADD = $(libxml_LIBS) $(top_builddir)/src/libCEGUIBase.la
 
diff -up CEGUI-0.6.2/src/Makefile.am.patch1 CEGUI-0.6.2/src/Makefile.am
--- CEGUI-0.6.2/src/Makefile.am.patch1	2008-06-24 21:01:53.000000000 +0200
+++ CEGUI-0.6.2/src/Makefile.am	2008-12-03 12:14:26.000000000 +0100
@@ -9,7 +9,7 @@ libCEGUIBase_la_LDFLAGS += -shared -no-u
 libCEGUIBase_la_LIBADD += -lwinmm
 endif
 
-libCEGUIBase_la_LDFLAGS += -version-info @CEGUI_VERSION_INFO@
+libCEGUIBase_la_LDFLAGS += -release @PACKAGE_VERSION@
 libCEGUIBase_la_LIBADD += $(freetype2_LIBS) $(pcre_LIBS)
 
 lib_LTLIBRARIES=libCEGUIBase.la
diff -up CEGUI-0.6.2/src/Makefile.in.patch1 CEGUI-0.6.2/src/Makefile.in
--- CEGUI-0.6.2/src/Makefile.in.patch1	2008-11-30 21:11:42.000000000 +0100
+++ CEGUI-0.6.2/src/Makefile.in	2008-12-03 12:14:26.000000000 +0100
@@ -247,7 +247,7 @@ xerces_CFLAGS = @xerces_CFLAGS@
 xerces_LIBS = @xerces_LIBS@
 INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir) $(freetype2_CFLAGS) $(pcre_CFLAGS) $(am__append_3)
 
-libCEGUIBase_la_LDFLAGS = $(am__append_1) -version-info @CEGUI_VERSION_INFO@
+libCEGUIBase_la_LDFLAGS = $(am__append_1) -release @PACKAGE_VERSION@
 libCEGUIBase_la_LIBADD = $(am__append_2) $(freetype2_LIBS) $(pcre_LIBS)
 
 @MINGW32_BUILD_TRUE@libCEGUIBase_la_CPPFLAGS = -DCEGUIBASE_EXPORTS
