--- a/pg.gemspec
+++ b/pg.gemspec
@@ -21,13 +21,9 @@
   spec.metadata["documentation_uri"] = "http://deveiate.org/code/pg"
 
   # Specify which files should be added to the gem when it is released.
-  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files = Dir.chdir(File.expand_path(__dir__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{\A(?:test|spec|features)/}) }
-  end
+  spec.files         = Dir["ext/**/*", "lib/**/*.rb"]
   spec.extensions    = ["ext/extconf.rb"]
   spec.require_paths = ["lib"]
-  spec.cert_chain    = ["certs/ged.pem"]
   spec.rdoc_options  = ["--main", "README.md",
                         "--title", "PG: The Ruby PostgreSQL Driver"]
   spec.extra_rdoc_files = `git ls-files -z *.rdoc *.md lib/*.rb lib/*/*.rb ext/*.c ext/*.h`.split("\x0")
