From 58a9262c480b524ad5489bcf393047c006bef225 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 12 Sep 2022 17:50:35 +0200
Subject: [PATCH] make: increase test timeout

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4f36624..2d1ebe7 100644
--- a/Makefile
+++ b/Makefile
@@ -106,7 +106,7 @@ tests/Kyuafile: $(wildcard tests/*_test)
 	echo "syntax(2)" > $@
 	echo "test_suite('abuild')" >> $@
 	for i in $(notdir $(wildcard tests/*_test)); do \
-		echo "atf_test_program{name='$$i',timeout=5}" >> $@ ; \
+		echo "atf_test_program{name='$$i',timeout=45}" >> $@ ; \
 	done
 
 Kyuafile: tests/Kyuafile
-- 
2.37.3

