diff --git a/abuild.in b/abuild.in
index 38d2598b8f..14955357af 100644
--- a/abuild.in
+++ b/abuild.in
@@ -812,7 +812,7 @@ prepare_metafiles() {
 	cd "$dir"
 	mkdir -p "$controldir"
 	local builddate=$(date -u "+%s")
-	local size=$(sync;du -sk | awk '{print $1 * 1024}')
+	local size=$(du -sk | awk '{print $1 * 1024}')
 
 	if [ "$arch" != "$apkbuild_arch" ]; then
 		local msg="Split function set arch=\"$arch\" for $name, use subpackages=pkg:split:arch format instead"
