From a7f9bff0f73dba6a82af6ccc60fdd2fab73a6566 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Tue, 14 Feb 2017 05:52:54 +0000
Subject: [PATCH] functions: fix cross-compile triplet for armhf architecture

---
 functions.sh.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/functions.sh.in b/functions.sh.in
index 58567d6..d04c776 100644
--- a/functions.sh.in
+++ b/functions.sh.in
@@ -6,7 +6,7 @@ program=${0##*/}
 arch_to_hostspec() {
 	case "$1" in
 	aarch64)	echo "aarch64-alpine-linux-musl" ;;
-	armhf)		echo "armhf-alpine-linux-muslgnueabihf" ;;
+	armhf)		echo "armv6-alpine-linux-muslgnueabihf" ;;
 	armv7)		echo "armv7-alpine-linux-musleabihf" ;;
 	ppc)		echo "powerpc-alpine-linux-musl" ;;
 	ppc64)		echo "powerpc64-alpine-linux-musl" ;;
-- 
2.11.1

