From 5adf47c168aff63878872383e5fd2c0e7757818d Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 31 Jan 2020 09:47:44 +0000
Subject: [PATCH] functions.sh: use apk --print-arch for CARCH if gcc is
 missing

---
 functions.sh.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/functions.sh.in b/functions.sh.in
index 6d5181a..82ac114 100644
--- a/functions.sh.in
+++ b/functions.sh.in
@@ -125,6 +125,13 @@ readconfig() {
 	[ "$(arch_to_hostspec $CTARGET)" != "unknown" ] && CTARGET="$(arch_to_hostspec $CTARGET)"
 
 	[ -z "$CARCH" ] && CARCH="$(hostspec_to_arch $CHOST)"
+
+	# use apk --print-arch for CARCH if gcc is missing
+	if [ "$CARCH" = "unknown" ]; then
+		local apk_arch="$(${APK:-apk} --print-arch 2>/dev/null)"
+		CARCH=${apk_arch:-unknown}
+	fi
+
 	[ -z "$CLIBC" ] && CLIBC="$(hostspec_to_libc $CHOST)"
 	[ -z "$CBUILD_ARCH" ] && CBUILD_ARCH="$(hostspec_to_arch $CBUILD)"
 	[ -z "$CTARGET_ARCH" ] && CTARGET_ARCH="$(hostspec_to_arch $CTARGET)"
-- 
2.27.0

