diff --git a/abuild.in b/abuild.in
index 35b7995572..2d1d9e644d 100644
--- a/abuild.in
+++ b/abuild.in
@@ -812,6 +812,13 @@ prepare_metafiles() {
 	cd "$dir"
 	mkdir -p "$controldir"
 	local builddate=$(date -u "+%s")
+
+	# Fix package size on several filesystems
+	case "$(df -PT . | awk 'END {print $2}')" in
+		btrfs|ecryptfs|zfs)
+		sync;;
+	esac
+
 	local size=$(du -sk | awk '{print $1 * 1024}')
 
 	if [ "$arch" != "$apkbuild_arch" ]; then
