From f640500e4cac6d126b8c109ffd277c2775243bdd Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 11 Oct 2012 18:47:29 +0000
Subject: [PATCH] abuild: include /usr/share/ri in -doc packages

---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 0fde169..62a860f 100755
--- a/abuild.in
+++ b/abuild.in
@@ -1132,7 +1132,7 @@ default_doc() {
 	arch=${arch_doc:-"noarch"}
 
 	local i
-	for i in doc man info html sgml licenses gtk-doc; do
+	for i in doc man info html sgml licenses gtk-doc ri; do
 		if [ -d "$pkgdir/usr/share/$i" ]; then
 			mkdir -p "$subpkgdir/usr/share"
 			mv "$pkgdir/usr/share/$i" "$subpkgdir/usr/share/"
-- 
1.7.12.3

