From 7b47e30b8725fe6bf6c929f55b1108489b0887f2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 4 Oct 2013 14:05:12 +0000
Subject: [PATCH] abuild: support qt5 mkspecs

---
 abuild.in | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/abuild.in b/abuild.in
index b12df7a..d004ffb 100755
--- a/abuild.in
+++ b/abuild.in
@@ -1301,9 +1301,11 @@ default_dev() {
 	for i in usr/include usr/lib/pkgconfig usr/share/aclocal\
 			usr/share/gettext usr/bin/*-config	\
 			usr/share/vala/vapi usr/share/gir-[0-9]*\
-			usr/share/qt/mkspecs			\
+			usr/share/qt*/mkspecs			\
+			usr/lib/qt*/mkspecs			\
 			$(find -name include -type d) 		\
-			$(find lib/ usr/ -name '*.[acho]' 2>/dev/null); do
+			$(find lib/ usr/ -name '*.[acho]' \
+				-o -name '*.prl' 2>/dev/null); do
 		if [ -e "$pkgdir/$i" ] || [ -L "$pkgdir/$i" ]; then
 			d="$subpkgdir/${i%/*}"	# dirname $i
 			mkdir -p "$d"
-- 
1.8.4

