From a5f67dcf25fae7c0200298623cccb61d9275fd28 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 23 May 2012 09:42:27 +0000
Subject: [PATCH] abuild: add support for uncompressed tar archives

---
 abuild.in |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/abuild.in b/abuild.in
index 2780d2e..32dd153 100755
--- a/abuild.in
+++ b/abuild.in
@@ -404,6 +404,9 @@ default_unpack() {
 	for u in $source; do
 		local s="$SRCDEST/${u##*/}"	# $(basename $s)
 		case "$s" in
+			*.tar)
+				msg "Unpacking $s..."
+				tar -C "$srcdir" -xf "$s" || return 1;;
 			*.tar.gz|*.tgz)
 				msg "Unpacking $s..."
 				tar -C "$srcdir" -zxf "$s" || return 1;;
-- 
1.7.10.2

