From eeca95d195652ef3506a154e9d18fd98f0dbf5b2 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 19 Apr 2024 23:41:45 +0200
Subject: [PATCH] abuild: pass APORTS_BOOTSTRAP to bubblewrap

---
 abuild.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/abuild.in b/abuild.in
index 3e41902..ac0b6ff 100644
--- a/abuild.in
+++ b/abuild.in
@@ -2627,6 +2627,7 @@ rootbld() {
 		--setenv REPODEST "$REPODEST" \
 		--setenv ABUILD_USERDIR "$ABUILD_USERDIR" \
 		${ABUILD_BOOTSTRAP:+--setenv ABUILD_BOOTSTRAP "$ABUILD_BOOTSTRAP"} \
+		${APORTS_BOOTSTRAP:+--setenv APORTS_BOOTSTRAP "$APORTS_BOOTSTRAP"} \
 		--setenv SOURCE_DATE_EPOCH "$SOURCE_DATE_EPOCH" \
 		--setenv ABUILD_LAST_COMMIT "$ABUILD_LAST_COMMIT" \
 		--setenv PATH ${USE_CCACHE:+/usr/lib/ccache/bin:}/bin:/usr/bin:/sbin:/usr/sbin \
-- 
2.44.0

