From 4dfc57946e7f1c97da41da64ec44d31d55fbfe3e Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Fri, 20 Oct 2017 14:53:34 +0000
Subject: [PATCH] abuild: log command: providers

---
 abuild.in | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/abuild.in b/abuild.in
index 7f9818b..41b465d 100644
--- a/abuild.in
+++ b/abuild.in
@@ -1004,6 +1004,19 @@ prepare_pkgconfig_provides() {
 	done
 }
 
+prepare_command_provides() {
+	local dir="${subpkgdir:-$pkgdir}"
+	options_has "!tracedeps" && return 0
+	cd "$dir" || return 1
+	for i in bin/* sbin/* usr/bin/* usr/sbin/*; do
+		if ! [ -x "$i" ]; then
+			continue
+		fi
+		local f=${i##*/}
+		echo $f >> "$controldir"/.provides-command
+	done
+}
+
 # check if dir has arch specific binaries
 dir_has_arch_binaries() {
 	local dir="$1"
@@ -1047,6 +1060,7 @@ prepare_package() {
 		&& prepare_trace_rpaths \
 		&& prepare_symlinks \
 		&& prepare_pkgconfig_provides \
+		&& prepare_command_provides \
 		|| return 1
 	archcheck
 }
@@ -1184,6 +1198,10 @@ trace_apk_deps() {
 		sed 's/^/provides = pc:/' "$dir"/.provides-pc | sort -u \
 			>> "$dir"/.PKGINFO
 	fi
+	if [ -f "$dir"/.provides-command ]; then
+		sed 's/^/provides = command:/' "$dir"/.provides-command | sort -u \
+			>> "$dir"/.PKGINFO
+	fi
 	[ -z "$autodeps" ] && return 0
 	for i in $autodeps; do
 		echo "depend = $i"
-- 
2.14.2

