From f24e997104849f4a7cf9e2f73a3cc66a83624fcb Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 31 Aug 2009 09:12:04 +0000
Subject: [PATCH] abuild: strip versions from dependencies when building recursively

---
 abuild.in |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/abuild.in b/abuild.in
index b718a67..23c9f96 100755
--- a/abuild.in
+++ b/abuild.in
@@ -712,7 +712,11 @@ depparse_aports() {
 }
 
 deptrace() {
-	local deps="$@"
+	local deps= i=
+	# strip versions from deps
+	for i in "$@"; do
+		deps="$deps ${i%[<>=]*}"
+	done
 	[ -z "$deps" ] && return 0
 	( 	depparse_aports 
 		if [ -z "$upgrade" ]; then
-- 
1.6.4

