From 18beb69377c4e4809c572cb287e8879d51bebedc Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 11 Apr 2011 11:51:39 +0000
Subject: [PATCH 2/2] buildrepo: fix previous commit

---
 buildrepo.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/buildrepo.in b/buildrepo.in
index 405e500..79954da 100755
--- a/buildrepo.in
+++ b/buildrepo.in
@@ -61,8 +61,8 @@ build() {
 		pkgs=$(abuild listpkg)
 		if all_exist $pkgs; then
 			echo ">>> Copying " $pkgs
-			cp -p -l $pkgs "$repodir/$repo/$CARCH"/ 2>/dev/null \
-				|| cp -p $pkgs "$repodir/$repo/$CARCH"/ \
+			cp -p -l $pkgs "$repodir/$repo/$arch"/ 2>/dev/null \
+				|| cp -p $pkgs "$repodir/$repo/$arch"/ \
 				|| needbuild="$needbuild $i"
 		else
 			needbuild="$needbuild $i"
@@ -81,7 +81,7 @@ build() {
 	if [ -n "$dopurge" ]; then
 		local tmp=$(mktemp /tmp/$program-XXXXXX)
 		local purgefiles
-		cd "$repodir/$repo/$CARCH" || return 1
+		cd "$repodir/$repo/$arch" || return 1
 		trap 'rm -f "$tmp"; exit 1' INT
 		( listpackages "$1") >$tmp
 		purge=$(ls *.apk 2>/dev/null | grep -v -w -f $tmp)
-- 
1.7.4.4

