From 1e3d4338a576a8cae81a1738ff635ec7ee4498c4 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Wed, 8 Jul 2015 12:20:46 +0300
Subject: [PATCH 2/2] apkbuild-gem-resolver: fix syntax

---
 apkbuild-gem-resolver.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/apkbuild-gem-resolver.in b/apkbuild-gem-resolver.in
index 5d3951d..69f4049 100644
--- a/apkbuild-gem-resolver.in
+++ b/apkbuild-gem-resolver.in
@@ -35,8 +35,8 @@ class Aport
       'ruby-io-console' => [ 'io-console', '0.4.2'] # its actually 0.4.3 but 
       						    # that version is not
 						    # published on network
-    }
-    '2.2.2' = > {
+    },
+    '2.2.2' => {
       'ruby-json' => ['json', '1.8.1'],
       'ruby-minitest' => ['minitest', '5.4.3'],
       'ruby-rake' => ['rake', '10.4.2'],
@@ -44,6 +44,7 @@ class Aport
       'ruby-io-console' => [ 'io-console', '0.4.2'] # its actually 0.4.3 but
 						    # that version is not
 						    # published on network
+    }
   }
 
   @@aports = {}
-- 
2.1.0

