commit c30c346362223658980275d54d5b7911a3a7486c
Author: Natanael Copa <ncopa@alpinelinux.org>
Date:   Wed Jun 30 14:59:07 2010 +0000

    abuild: detect and report conflicting dependencies properly

diff --git a/abuild.in b/abuild.in
index 0c94653..4034a86 100755
--- a/abuild.in
+++ b/abuild.in
@@ -841,15 +841,16 @@ builddeps() {
 	# find which deps are missing
 	for i in $deps; do
 		if [ "${i#\!}" != "$i" ]; then
-			list_has ${i#\!} $installed_deps \
+			$APK info -q -e "${i#\!}" \
 				&& conflicts="$conflicts ${i#\!}"
 		elif ! deplist_has $i $installed_deps || [ -n "$upgrade" ]; then
 			missing="$missing $i"
 		fi
 	done
-
+	
 	if [ -n "$conflicts" ]; then
-		die "Conflicting package(s) installed:$conflics"
+		error "Conflicting package(s) installed:$conflicts"
+		return 1
 	fi
 	
 	if [ -z "$install_deps" ] && [ -z "$recursive" ]; then
