Ref https://gitlab.alpinelinux.org/alpine/abuild/-/merge_requests/386

--- a/default.conf
+++ b/default.conf
@@ -13,6 +13,8 @@ export CARGO_BUILD_JOBS=$JOBS
 export CMAKE_BUILD_PARALLEL_LEVEL=$JOBS
 export CTEST_PARALLEL_LEVEL=$JOBS
 
+export CMAKE_GENERATOR=Ninja
+
 export CARGO_PROFILE_RELEASE_OPT_LEVEL="s"
 export CARGO_PROFILE_RELEASE_PANIC="abort"
 export CARGO_PROFILE_RELEASE_CODEGEN_UNITS=1
--- a/newapkbuild.in
+++ b/newapkbuild.in
@@ -67,7 +67,7 @@ build_cmake() {
 	if [ "\$CBUILD" != "\$CHOST" ]; then
 		local crossopts="-DCMAKE_SYSTEM_NAME=Linux -DCMAKE_HOST_SYSTEM_NAME=Linux"
 	fi
-	cmake -B build -G Ninja \\
+	cmake -B build \\
 		-DCMAKE_INSTALL_PREFIX=/usr \\
 		-DCMAKE_INSTALL_LIBDIR=lib \\
 		-DBUILD_SHARED_LIBS=ON \\
