From 01ce16195f4177c6a0f1b24db1bbf80cd82d1053 Mon Sep 17 00:00:00 2001
From: Sertonix <sertonix@posteo.net>
Date: Sat, 28 Dec 2024 15:47:39 +0100
Subject: [PATCH] abuild: move usr/share/cmake in default_dev

---
 abuild.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/abuild.in b/abuild.in
index 4cd09bf..6180ba0 100644
--- a/abuild.in
+++ b/abuild.in
@@ -2064,7 +2064,7 @@ default_dev() {
 			usr/share/aclocal usr/share/gettext \
 			usr/bin/*-config usr/share/vala/vapi \
 			usr/share/gir-[0-9]* usr/share/qt*/mkspecs \
-			usr/lib/qt*/mkspecs usr/lib/cmake \
+			usr/lib/qt*/mkspecs usr/lib/cmake usr/share/cmake \
 			usr/lib/glade/modules usr/share/glade/catalogs \
 			$(find . -name include -type d) \
 			$(subpackage_types_has static || find $libdirs \
-- 
2.47.1

