From 00fd7b07f12a18615da1ad32b915787f72074dfa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 17 Jan 2012 15:59:01 +0200
Subject: [PATCH 1/2] lua: fix compile error

---
 src/lua-apk.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/lua-apk.c b/src/lua-apk.c
index 6fbd1cf..45488c2 100644
--- a/src/lua-apk.c
+++ b/src/lua-apk.c
@@ -208,7 +208,7 @@ static int Papk_exists(lua_State *L)
 	apk_blob_t blob = APK_BLOB_STR(depstr);
 	apk_blob_pull_dep(&blob, db, &dep);
 
-	if (APK_BLOB_IS_NULL(blob) || b.len > 0)
+	if (APK_BLOB_IS_NULL(blob) || blob.len > 0)
 		goto ret_nil;
 
 	name = dep.name;
-- 
1.7.7.3

