From a6c4103df550ec7c1d99af6a5573c2e3f946b408 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Wed, 2 Oct 2013 16:06:10 +0300
Subject: [PATCH 1/3] db: allow using cached packages with --no-network

---
 src/database.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/database.c b/src/database.c
index afe0a8c..9eadb44 100644
--- a/src/database.c
+++ b/src/database.c
@@ -2151,7 +2151,7 @@ int apk_db_add_repository(apk_database_t _db, apk_blob_t _repository)
 		db->available_repos &= ~BIT(repo_num);
 		r = 0;
 	} else {
-		db->repo_tags[tag_id].allowed_repos |= BIT(repo_num) & db->available_repos;
+		db->repo_tags[tag_id].allowed_repos |= BIT(repo_num);
 	}
 
 	return 0;
-- 
1.8.5.1

