From fc4c60f233ebe9fe85b7b03d0cf5deec4ab1b32d Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 16 Jun 2010 07:37:17 +0000
Subject: [PATCH] Makefile: do not require lua pkgconfig unless you intend build lua module

---
 src/Makefile |   11 ++++++-----
 1 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 50f9664..bea288e 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,12 +1,9 @@
-PKGDEPS := openssl zlib lua
-
-ifeq ($(shell pkg-config --print-errors --exists $(PKGDEPS) || echo fail),fail)
-$(error Build dependencies are not met)
-endif
+PKGDEPS := openssl zlib
 
 # lua module
 ifneq ($(LUAAPK),)
 LIBAPK			:= YesPlease
+PKGDEPS			+= lua
 shlibs-y		+= apk.so
 apk.so-objs		:= lua-apk.o
 CFLAGS_lua-apk.o	:= -DAPK_VERSION=\"$(FULL_VERSION)\"
@@ -16,6 +13,10 @@ install-LUA_LIB-y	:= $(INSTALLDIR) $(DESTDIR)$(LUA_LIBDIR) \
 			   $(INSTALL) $(LUA_LIB-y) $(DESTDIR)$(LUA_LIBDIR)
 endif
 
+ifeq ($(shell pkg-config --print-errors --exists $(PKGDEPS) || echo fail),fail)
+$(error Build dependencies are not met)
+endif
+
 progs-y			+= apk
 apk-objs		:= apk.o add.o del.o fix.o update.o info.o \
 			   search.o upgrade.o cache.o ver.o index.o fetch.o \
-- 
1.7.1

