From ca368916e0333bf24cdcbdbe42130ec6a92c3f6e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Thu, 5 Jan 2017 16:14:44 +0200
Subject: [PATCH] tar: return correct error for short read of tar archive

---
 src/archive.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/archive.c b/src/archive.c
index e86a53a..07b33e2 100644
--- a/src/archive.c
+++ b/src/archive.c
@@ -105,12 +105,17 @@ static ssize_t tar_entry_read(void *stream, void *ptr, size_t size)
 
 	if (size > teis->bytes_left)
 		size = teis->bytes_left;
-        if (size == 0)
-                return 0;
+	if (size == 0)
+		return 0;
 
 	r = teis->tar_is->read(teis->tar_is, ptr, size);
-	if (r < 0)
+	if (r <= 0) {
+		/* If inner stream returned zero (end-of-stream), we
+		 * are getting short read, because tar header indicated
+		 * more was to be expected. */
+		if (r == 0) return -ECONNABORTED;
 		return r;
+	}
 
 	teis->bytes_left -= r;
 	if (teis->csum == NULL)
-- 
2.11.0

