From a57db1bba098808ed3545579833ce4372ee4ba20 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Sat, 31 Mar 2012 23:27:15 +0300
Subject: [PATCH 2/2] solver: fix installation of non-repository packages

---
 src/solver.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/solver.c b/src/solver.c
index 820f840..c26fd7b 100644
--- a/src/solver.c
+++ b/src/solver.c
@@ -1116,7 +1116,8 @@ static int reconsider_name(struct apk_solver_state *ss, struct apk_name *name)
 
 		if (ps0 == NULL || ps0->locked || ps0->conflicts ||
 		    ss->topology_position < pkg0->topology_hard ||
-		    (pkg0->ipkg == NULL && (!ps0->allowed || !pkg_available(ss->db, pkg0))))
+		    (pkg0->ipkg == NULL && pkg0->filename == NULL &&
+		     (!ps0->allowed || !pkg_available(ss->db, pkg0))))
 			continue;
 
 		for (j = 0; j < pkg0->provides->num; j++) {
-- 
1.7.9.4

