From a2a5cd9c696d0d6f2c14b8831e62caf24d16fa50 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 16 Feb 2016 15:59:55 +0200
Subject: [PATCH] fetch: allow enabling --simulate

---
 src/apk.c   | 3 +++
 src/fetch.c | 5 +++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/apk.c b/src/apk.c
index 91673d8..25c5260 100644
--- a/src/apk.c
+++ b/src/apk.c
@@ -106,6 +106,9 @@ static int option_parse_global(void *ctx, struct apk_db_options *dbopts, int opt
 	case 0x101:
 		apk_flags |= APK_PROGRESS;
 		break;
+	case 0x104:
+		apk_flags |= APK_SIMULATE;
+		break;
 	case 0x110:
 		apk_flags &= ~APK_PROGRESS;
 		break;
diff --git a/src/fetch.c b/src/fetch.c
index 14e0790..bc63983 100644
--- a/src/fetch.c
+++ b/src/fetch.c
@@ -93,8 +93,9 @@ static int option_parse_applet(void *ctx, struct apk_db_options *dbopts, int opt
 static const struct apk_option options_applet[] = {
 	{ 'L', "link",		"Create hard links if possible" },
 	{ 'R', "recursive",	"Fetch the PACKAGE and all its dependencies" },
-	{ 's', "stdout",
-	  "Dump the .apk to stdout (incompatible with -o, -R, --progress)" },
+	{ 0x104, "simulate",	"Show what would be done without actually doing it" },
+	{ 's', "stdout",	"Dump the .apk to stdout (incompatible "
+				"with -o, -R, --progress)" },
 	{ 'o', "output",	"Directory to place the PACKAGEs to",
 	  required_argument, "DIR" },
 };
-- 
2.7.1

