From 7419771b02e84a91a71b12869a40208fd2e8b773 Mon Sep 17 00:00:00 2001
From: Timo Teras <timo.teras@iki.fi>
Date: Thu, 6 Aug 2009 08:57:50 +0300
Subject: [PATCH 2/7] audit: fix --backup

---
 src/audit.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/src/audit.c b/src/audit.c
index d61b321..e47f182 100644
--- a/src/audit.c
+++ b/src/audit.c
@@ -51,6 +51,9 @@ static int audit_directory(apk_hash_item item, void *ctx)
 	char tmp[PATH_MAX], reason;
 	DIR *dir;
 
+	if (!(dbd->flags & APK_DBDIRF_PROTECTED))
+		return 0;
+
 	dir = fdopendir(openat(db->root_fd, dbd->name, O_RDONLY));
 	if (dir == NULL)
 		return 0;
@@ -97,7 +100,7 @@ static int audit_directory(apk_hash_item item, void *ctx)
 
 static int audit_backup(struct apk_database *db)
 {
-	return apk_hash_foreach(&db->installed.dirs, audit_directory, &db);
+	return apk_hash_foreach(&db->installed.dirs, audit_directory, db);
 }
 
 static int audit_system(struct apk_database *db)
-- 
1.6.4

