From 4b7b5d4668865b1026c7050c8872f256bc773605 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 28 Sep 2012 10:24:27 +0300
Subject: [PATCH] info: --provides alias -p is reserved for --root, use -P

fixes #1396
---
 src/info.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/info.c b/src/info.c
index e9f1b59..9b37f38 100644
--- a/src/info.c
+++ b/src/info.c
@@ -427,7 +427,7 @@ static int info_parse(void *ctx, struct apk_db_options *dbopts,
 	case 'R':
 		ictx->subaction_mask |= APK_INFO_DEPENDS;
 		break;
-	case 'p':
+	case 'P':
 		ictx->subaction_mask |= APK_INFO_PROVIDES;
 		break;
 	case 'r':
@@ -489,7 +489,7 @@ static struct apk_option info_options[] = {
 	{ 'e', "installed",	"Check if PACKAGE is installed" },
 	{ 'W', "who-owns",	"Print the package owning the specified file" },
 	{ 'R', "depends",	"List packages that the PACKAGE depends on" },
-	{ 'p', "provides",	"List virtual packages provided by PACKAGE" },
+	{ 'P', "provides",	"List virtual packages provided by PACKAGE" },
 	{ 'r', "rdepends",	"List all packages depending on PACKAGE" },
 	{ 0x10000, "replaces",	"List packages whom files PACKAGE might replace" },
 	{ 'i', "install-if",	"List the PACKAGE's install-if rule" },
-- 
1.7.12.2

