From 00958d92070c583e4fa9d01f5010bce12bcd1b40 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 3 Oct 2012 09:04:35 +0200
Subject: [PATCH] solver: fix compile with -Werror on 64bit

---
 src/solver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/solver.c b/src/solver.c
index ec44958..f9a2117 100644
--- a/src/solver.c
+++ b/src/solver.c
@@ -1664,7 +1664,7 @@ static void update_progress(struct progress *prog, size_t percent)
 		draw_progress(percent);
 	if (prog->progress_fd != 0) {
 		char buf[8];
-		size_t n = snprintf(buf, sizeof(buf), "%d\n", percent);
+		size_t n = snprintf(buf, sizeof(buf), "%zu\n", percent);
 		write(prog->progress_fd, buf, n);
 	}
 }
-- 
1.7.12.2

