From 0b2052ed064708767358290a73d600ed55fee5b2 Mon Sep 17 00:00:00 2001
From: Timo Teras <timo.teras@iki.fi>
Date: Thu, 6 Aug 2009 11:47:44 +0300
Subject: [PATCH 4/7] state: ignore missing world dependencies with --force and --quiet

---
 src/state.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/state.c b/src/state.c
index 50c1ce3..d886754 100644
--- a/src/state.c
+++ b/src/state.c
@@ -148,7 +148,7 @@ struct apk_state *apk_state_new(struct apk_database *db)
 	 * choices */
 	for (i = 0; db->world != NULL && i < db->world->num; i++) {
 		r = apk_state_prune_dependency(state, &db->world->item[i]);
-		if (r < 0) {
+		if (r < 0 && apk_verbosity && !(apk_flags & APK_FORCE)) {
 			apk_error("Top level dependencies for %s are "
 				  "conflicting or unsatisfiable.",
 				  db->world->item[i].name->name);
-- 
1.6.4

