From a6b28beef055c9c454fba49d02995765e6f3aeab Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 7 May 2012 10:53:45 +0200
Subject: [PATCH 5/5] ver: show all packages with -v

ref #1122
---
 src/ver.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ver.c b/src/ver.c
index 05ffce9..c20547a 100644
--- a/src/ver.c
+++ b/src/ver.c
@@ -161,7 +161,7 @@ static int ver_main(void *ctx, struct apk_database *db, int argc, char **argv)
 	if (ictx->limchars) {
 		if (strlen(ictx->limchars) == 0)
 			ictx->limchars = NULL;
-	} else if (argc == 0) {
+	} else if (argc == 0 && apk_verbosity == 1) {
 		ictx->limchars = "<";
 	}
 
-- 
1.7.10

