From 06ae5fdfdccd0c8e6d5501d93666bd915d2604d1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 8 Jul 2016 11:09:00 +0300
Subject: [PATCH] detect aarch64 architecture and assign default name for it

---
 src/apk_defines.h | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/apk_defines.h b/src/apk_defines.h
index 2aca364..9b2e904 100644
--- a/src/apk_defines.h
+++ b/src/apk_defines.h
@@ -93,9 +93,10 @@ extern char **apk_argv;
 #define APK_DEFAULT_ARCH	"armhf"
 #elif defined(__arm__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
 #define APK_DEFAULT_ARCH	"armel"
+#elif defined(__aarch64__) && __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define APK_DEFAULT_ARCH	"aarch64"
 #else
-#warning APK_DEFAULT_ARCH is not set for this architecture
-#define APK_DEFAULT_ARCH        "noarch"
+#error APK_DEFAULT_ARCH not detected for this architecture
 #endif
 
 #define APK_MAX_REPOS		32	/* see struct apk_package */
-- 
2.9.0

