From 3ae196ecc95d956adca4342bbcc7ac1dca409d6c Mon Sep 17 00:00:00 2001
From: Jiri Kastner <cz172638@gmail.com>
Date: Wed, 29 Jan 2025 16:59:01 +0100
Subject: [PATCH] libfetch/http: reuse fetch_netrc_auth for http(s)

Signed-off-by: Jiri Kastner <cz172638@gmail.com>
---
 libfetch/http.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libfetch/http.c b/libfetch/http.c
index 881318c..cb4ce88 100644
--- a/libfetch/http.c
+++ b/libfetch/http.c
@@ -908,6 +908,8 @@ http_request(struct url *URL, const char *op, struct url_stat *us,
 		/* proxy authorization */
 		http_proxy_authorize(conn, purl);
 
+		if (need_auth && !*url->pwd && !*url->user)
+			fetch_netrc_auth(url);
 		/* server authorization */
 		if (need_auth || *url->user || *url->pwd) {
 			if (*url->user || *url->pwd)
-- 
2.50.1

