diff --git a/src/package.c b/src/package.c
index 2aeb21c..e317026 100644
--- a/src/package.c
+++ b/src/package.c
@@ -1117,7 +1117,7 @@ static int write_depends(struct apk_ostream *os, const char *field,
 int apk_pkg_write_index_entry(struct apk_package *info,
 			      struct apk_ostream *os)
 {
-	char buf[512];
+	char buf[1024];
 	apk_blob_t bbuf = APK_BLOB_BUF(buf);
 
 	apk_blob_push_blob(&bbuf, APK_BLOB_STR("C:"));
