diff --git a/src/database.c b/src/database.c
index 93ce2ed..66fe7f1 100644
--- a/src/database.c
+++ b/src/database.c
@@ -1766,8 +1766,8 @@ static int unshare_mount_namespace(bool usermode)
 		if (write_file("/proc/self/setgroups", "deny") != 0) return -1;
 		if (write_file("/proc/self/gid_map", "0 %d 1", gid) != 0) return -1;
 	} else {
-		// if unshare fails as root, we continue with chroot
-		if (unshare(CLONE_NEWNS) != 0) return 0;
+		// no unshare as root for now - breaks certain nested namespace setups
+		return 0;
 	}
 	if (mount("none", "/", NULL, MS_REC|MS_PRIVATE, NULL) != 0) return -1;
 	// Create /proc and /dev in the chroot
