Alpine's cni-plugins are installs at /usr/libexec/cni/

--- a/docs/crio.conf.5.md
+++ b/docs/crio.conf.5.md
@@ -498,7 +498,7 @@
 **network_dir**="/etc/cni/net.d/"
 Path to the directory where CNI configuration files are located.
 
-**plugin_dirs**=["/opt/cni/bin/",]
+**plugin_dirs**=["/usr/libexec/bin/",]
 List of paths to directories where CNI plugin binaries are located.
 
 ## CRIO.METRICS TABLE
--- a/pkg/config/config_unix.go
+++ b/pkg/config/config_unix.go
@@ -5,7 +5,7 @@
 // Defaults for linux/unix if none are specified
 const (
 	cniConfigDir             = "/etc/cni/net.d/"
-	cniBinDir                = "/opt/cni/bin/"
+	cniBinDir                = "/usr/libexec/cni/"
 	containerExitsDir        = "/var/run/crio/exits"
 	ContainerAttachSocketDir = "/var/run/crio"
