--- a/test/helpers.bash
+++ b/test/helpers.bash
@@ -68,7 +68,7 @@
 # The default log directory where all logs will go unless directly specified by the kubelet
 DEFAULT_LOG_PATH=${DEFAULT_LOG_PATH:-/var/log/crio/pods}
 # Cgroup manager to be used
-CONTAINER_CGROUP_MANAGER=${CONTAINER_CGROUP_MANAGER:-systemd}
+CONTAINER_CGROUP_MANAGER=${CONTAINER_CGROUP_MANAGER:-cgroupfs}
 # Image volumes handling
 CONTAINER_IMAGE_VOLUMES=${CONTAINER_IMAGE_VOLUMES:-mkdir}
 # Container pids limit
@@ -166,7 +166,7 @@
     # Copy all the CNI dependencies around to ensure encapsulated tests
     CRIO_CNI_PLUGIN="$TESTDIR/cni-bin"
     mkdir "$CRIO_CNI_PLUGIN"
-    cp /opt/cni/bin/* "$CRIO_CNI_PLUGIN"
+    cp /usr/libexec/cni/* "$CRIO_CNI_PLUGIN"
     cp "$INTEGRATION_ROOT"/cni_plugin_helper.bash "$CRIO_CNI_PLUGIN"
     sed -i "s;%TEST_DIR%;$TESTDIR;" "$CRIO_CNI_PLUGIN"/cni_plugin_helper.bash
 
--- a/test/cni_plugin_helper.bash
+++ b/test/cni_plugin_helper.bash
@@ -43,7 +43,7 @@
 . "$TEST_DIR"/cni_plugin_helper_input.env
 rm -f "$TEST_DIR"/cni_plugin_helper_input.env
 
-result=$(/opt/cni/bin/bridge "$@") || exit $?
+result=$(/usr/libexec/cni/bridge "$@") || exit $?
 
 if [[ "${DEBUG_ARGS}" == "malformed-result" ]]; then
     cat <<-EOF
--- a/test/cgroups.bats
+++ b/test/cgroups.bats
@@ -26,6 +26,10 @@
 }
 
 @test "conmon custom cgroup" {
+	if [[ "$CONTAINER_CGROUP_MANAGER" != "systemd" ]]; then
+		skip "need systemd cgroup manager"
+	fi
+
 	CONTAINER_CGROUP_MANAGER="systemd" CONTAINER_DROP_INFRA_CTR=false CONTAINER_MANAGE_NS_LIFECYCLE=false CONTAINER_CONMON_CGROUP="customcrioconmon.slice" start_crio
 
 	jq '	  .linux.cgroup_parent = "Burstablecriotest123.slice"' \
