diff --git a/Makefile b/Makefile
index 19f8052..135385c 100644
--- a/Makefile
+++ b/Makefile
@@ -403,7 +403,7 @@ get-script:
 	sed -i '/# INCLUDE/q' scripts/get
 	cat contrib/bundle/install-paths contrib/bundle/install >> scripts/get
 
-install: .gopathok install.bin install.man install.completions install.systemd install.config
+install: .gopathok install.bin install.man install.completions install.config
 
 install.bin-nobuild:
 	install ${SELINUXOPT} -D -m 755 bin/crio $(BINDIR)/crio
