From a977c911cc4edab55434a23a9cedaa4db9367760 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Thu, 3 Apr 2025 11:34:38 -0700
Subject: [PATCH 1/2] justfile: use prefix in dconf profile path

---
 Justfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Justfile b/Justfile
index c9d73d6..d348ec8 100644
--- a/Justfile
+++ b/Justfile
@@ -11,7 +11,7 @@ debug_args := if debug == '1' { '' } else { '--release' }
 cargo_args := vendor_args + ' ' + debug_args
 xdp_cosmic := '/usr/libexec/xdg-desktop-portal-cosmic'
 orca := '/usr/bin/orca'
-cosmic_dconf_profile := '/usr/share/dconf/profile/cosmic'
+cosmic_dconf_profile := prefix + '/share/dconf/profile/cosmic'
 
 bindir := prefix + '/bin'
 systemddir := prefix + '/lib/systemd/user'
-- 
2.49.0

