--- a/src/CodeBlocks-unix.cbp
+++ b/src/CodeBlocks-unix.cbp
@@ -97,20 +97,6 @@
 					<Add directory="sdk/wxscintilla/include" />
 				</ResourceCompiler>
 			</Target>
-			<Target title="wxpropgrid">
-				<Option output="devel/libwxpropgrid" prefix_auto="0" extension_auto="1" />
-				<Option working_dir="devel" />
-				<Option type="3" />
-				<Option compiler="gcc" />
-				<Option createStaticLib="1" />
-				<Compiler>
-					<Add option="-fvisibility=hidden" />
-					<Add option="-DEXPORT_LIB" />
-					<Add option="-DwxPG_SUPPORT_TOOLTIPS" />
-					<Add directory="sdk/wxpropgrid/include" />
-					<Add directory="include" />
-				</Compiler>
-			</Target>
 			<Target title="sdk">
 				<Option output="devel/libcodeblocks" prefix_auto="0" extension_auto="1" />
 				<Option working_dir="devel" />
@@ -135,7 +121,6 @@
 					<Add directory="include/mozilla_chardet/xpcom" />
 					<Add directory="include/mozilla_chardet/xpcom/base" />
 					<Add directory="include/mozilla_chardet/xpcom/glue" />
-					<Add directory="sdk/wxpropgrid/include" />
 				</Compiler>
 				<Linker>
 					<Add option="-lwxscintilla_cb" />
@@ -174,7 +159,6 @@
 				<Linker>
 					<Add option="`pkg-config gtk+-2.0 --libs`" />
 					<Add library="codeblocks" />
-					<Add library="wxpropgrid" />
 				</Linker>
 			</Target>
 			<Target title="Abbreviations">
@@ -266,7 +250,6 @@
 					<Add option="-Wl,--no-undefined" />
 					<Add library="codeblocks" />
 					<Add library="depslib" />
-					<Add library="wxpropgrid" />
 				</Linker>
 			</Target>
 			<Target title="Debugger">
@@ -462,9 +445,9 @@
 			</Environment>
 		</Build>
 		<VirtualTargets>
-			<Add alias="All" targets="tinyXML;AutoRevision;ConsoleRunner;Squirrel;Squirrel std lib;SqPlus;scintilla;wxpropgrid;sdk;src;Abbreviations;AStyle;Autosave;Compiler depslib;Compiler;Debugger;Code-completion;Class wizard;Default MIME handler;Occurrences Highlighting;Open files list;Projects-workspaces importer;Scripted wizard;To-do;" />
+			<Add alias="All" targets="tinyXML;AutoRevision;ConsoleRunner;Squirrel;Squirrel std lib;SqPlus;scintilla;sdk;src;Abbreviations;AStyle;Autosave;Compiler depslib;Compiler;Debugger;Code-completion;Class wizard;Default MIME handler;Occurrences Highlighting;Open files list;Projects-workspaces importer;Scripted wizard;To-do;" />
 			<Add alias="Core app &amp; plugins" targets="sdk;src;Abbreviations;AStyle;Autosave;Compiler depslib;Compiler;Debugger;Code-completion;Class wizard;Default MIME handler;Occurrences Highlighting;Open files list;Projects-workspaces importer;Scripted wizard;To-do;" />
-			<Add alias="Third-party libs" targets="tinyXML;AutoRevision;ConsoleRunner;Squirrel;Squirrel std lib;SqPlus;scintilla;wxpropgrid;Compiler depslib;" />
+			<Add alias="Third-party libs" targets="tinyXML;AutoRevision;ConsoleRunner;Squirrel;Squirrel std lib;SqPlus;scintilla;Compiler depslib;" />
 		</VirtualTargets>
 		<Compiler>
 			<Add option="-Wall" />
@@ -479,7 +462,6 @@
 			<Add option="-DcbDEBUG" />
 			<Add option="-DCB_PRECOMP" />
 			<Add directory="sdk/wxscintilla/include" />
-			<Add directory="sdk/wxpropgrid/include" />
 			<Add directory="include/tinyxml" />
 		</Compiler>
 		<Linker>
@@ -3125,57 +3107,6 @@
 		</Unit>
 		<Unit filename="sdk/workspaceloader.cpp">
 			<Option target="sdk" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/advprops.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/editors.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/extras.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/manager.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/odcombo.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/propdev.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/propgrid.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/props.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/include/wx/propgrid/xh_propgrid.h">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/advprops.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/editors.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/extras.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/manager.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/odcombo.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/propgrid.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/props.cpp">
-			<Option target="wxpropgrid" />
-		</Unit>
-		<Unit filename="sdk/wxpropgrid/src/xh_propgrid.cpp">
-			<Option target="wxpropgrid" />
 		</Unit>
 		<Unit filename="sdk/wxscintilla/include/wx/wxscintilla.h">
 			<Option target="scintilla" />
--- a/configure.ac
+++ b/configure.ac
@@ -318,16 +318,10 @@
     AC_MSG_RESULT(yes (version $WX_VERSION))
     AC_MSG_NOTICE([(Re)setting libs for wxWidgets $WX_VERSION])
     WX_LIBS=`$WX_CONFIG_WITH_ARGS --libs std,aui,propgrid,richtext`
-    WXPROPGRID_DIR=
-    WXPROPGRID_LIB=
-    WXPROPGRID_INCLUDE=
     SCINTILLA_GTK_CFLAGS="$WX_GDK_CFLAGS"
     SCINTILLA_GTK_LIBS="$WX_GDK_LIBS"
 else
     AC_MSG_RESULT(no)
-    WXPROPGRID_DIR=wxpropgrid
-    WXPROPGRID_LIB=wxpropgrid/libwxpropgrid.la
-    WXPROPGRID_INCLUDE=-I\$\(top_srcdir\)/src/sdk/wxpropgrid/include
 
     if test "x$HAVE_GCC48_TRUE" = "x" ; then
 	gcc48_flags="-Wno-unused-local-typedefs"
@@ -372,10 +366,6 @@
 AC_SUBST(CB_FONTCONFIG_LIBS)
 AC_SUBST(WX_X11_CFLAGS)
 AC_SUBST(WX_X11_LIBS)
-AC_SUBST(WXPROPGRID_DIR)
-AC_SUBST(WXPROPGRID_LIB)
-AC_SUBST(WXPROPGRID_INCLUDE)
-AC_SUBST(WXPROPGRID_HEADERS)
 
 AC_SUBST(POSTLINK_COMMAND)
 AC_SUBST(MACSETFILE)
@@ -434,7 +424,6 @@
 	src/include/scripting/squirrel/Makefile
 	src/include/tinyxml/Makefile
 	src/sdk/Makefile
-	src/sdk/wxpropgrid/Makefile
 	src/sdk/scripting/Makefile
 	src/sdk/scripting/bindings/Makefile
 	src/sdk/scripting/squirrel/Makefile
--- a/src/sdk/Makefile.am
+++ b/src/sdk/Makefile.am
@@ -1,12 +1,11 @@
-SUBDIRS = scripting wxscintilla $(WXPROPGRID_DIR) resources
-DIST_SUBDIRS = scripting wxscintilla  wxpropgrid resources
+SUBDIRS = scripting wxscintilla resources
+DIST_SUBDIRS = scripting wxscintilla resources
 
 AM_CPPFLAGS = $(WX_CXXFLAGS) \
 			$(CB_GLIB2_CFLAGS) \
 			$(CB_SQUIRREL_CFLAGS) \
 			-I$(top_srcdir)/src/include \
 			-I$(top_srcdir)/src/sdk/wxscintilla/include \
-			 $(WXPROPGRID_INCLUDE) \
 			-I$(top_srcdir)/src/include/scripting/bindings \
  			-I$(top_srcdir)/src/include/scripting/sqplus \
  			-I$(top_srcdir)/src/include/mozilla_chardet \
@@ -26,7 +25,6 @@
 			$(WX_LIBS) \
 			$(SCINTILLA_GTK_LIBS) \
 			$(CB_GLIB2_LIBS) \
-			$(WXPROPGRID_LIB) \
 			$(CB_TINYXML_LIBS) \
 			$(CB_SQUIRREL_LIBS)
 
