--- a/t/06-telemetry/01-basic.t
+++ b/t/06-telemetry/01-basic.t
@@ -88,7 +88,7 @@
 ok @report[0].starts-with('Telemetry Report of Process'), 'line 1 of report';
 is @report[1], 'Number of Snapshots: 0', 'line 2 of report';
 
-{ # https://github.com/rakudo/rakudo/issues/1714
+if %*ENV<CARCH> ne any <x86 armv7 armhf> { # https://github.com/rakudo/rakudo/issues/1714
     (temp %*ENV)<RAKUDO_REPORT_COLUMNS> = 'blahblah';
     is-run ｢
       use snapper;
@@ -98,6 +98,9 @@
       print 'pass'
     ｣, :err{.contains: 'Unknown Telemetry column `blahblah`'}, :out<pass>,
     'giving unknown column in RAKUDO_REPORT_COLUMNS warns instead of crashing'
+}
+else {
+    skip 'Memory allocation failures on 32-bit architectures';
 }
 
 # vim: expandtab shiftwidth=4
