--- ghc-8.0.2/aclocal.m4.orig	2017-02-17 12:00:45.179765870 +0000
+++ ghc-8.0.2/aclocal.m4	2017-02-17 12:00:56.033299769 +0000
@@ -227,7 +227,7 @@
 
     checkVendor() {
         case [$]1 in
-        dec|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld)
+        dec|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld|alpine)
             ;;
         *)
             echo "Unknown vendor [$]1"
