diff -urw orig/src/Makefile patched/src/Makefile
--- orig/src/Makefile	2016-11-13 21:33:58.078368577 +0000
+++ patched/src/Makefile	2016-11-13 21:34:21.574907952 +0000
@@ -19,8 +19,8 @@
 #INC+=-I/usr/local/libressl/include
 #LIBS+=-L/usr/local/libressl/lib64
 #LIBS+=-Wl,--rpath=/usr/local/libressl/lib64
-#DEFS+=-DHAVE_BN_GENCB_NEW=0
-#DEFS+=-DHAVE_LIBRESSL
+DEFS+=-DHAVE_BN_GENCB_NEW=0
+DEFS+=-DHAVE_LIBRESSL
 
 
 # BoringSSL. Be warned, BoringSSL creates incompatible
