diff -Nurp a/src/crypto/isa-l/CMakeLists.txt b/src/crypto/isa-l/CMakeLists.txt
--- a/src/crypto/isa-l/CMakeLists.txt	2025-02-08 08:28:40.954763698 +0000
+++ b/src/crypto/isa-l/CMakeLists.txt	2025-02-08 08:33:46.500668452 +0000
@@ -1,17 +1,19 @@
 # build isa-l_crypto from its makefile and expose as target ISAL::Crypto
-include(BuildISALCrypto)
-build_isal_crypto()
+#include(BuildISALCrypto)
+#build_isal_crypto()
 
 set(isal_crypto_plugin_srcs
   isal_crypto_accel.cc 
   isal_crypto_plugin.cc)
 
 if(HAVE_NASM_X64)
+find_package(PkgConfig)
+pkg_check_modules(ISALCRYPTO REQUIRED IMPORTED_TARGET libisal_crypto)
 add_dependencies(crypto_plugins ceph_crypto_isal)
 endif(HAVE_NASM_X64)
 
 add_library(ceph_crypto_isal SHARED ${isal_crypto_plugin_srcs})
-target_link_libraries(ceph_crypto_isal PRIVATE ISAL::Crypto Boost::context)
+target_link_libraries(ceph_crypto_isal PkgConfig::ISALCRYPTO Boost::context)
 
 set_target_properties(ceph_crypto_isal PROPERTIES
   VERSION 1.0.0
