diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cbc5f55edf1..37027866a90 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -419,7 +419,6 @@ set(libcommon_files
   ${CMAKE_BINARY_DIR}/src/include/ceph_ver.h
   ceph_ver.c
   global/global_context.cc
-  xxHash/xxhash.c
   common/error_code.cc
   common/tracer.cc
   log/Log.cc
@@ -454,6 +453,8 @@ add_library(common-objs OBJECT ${libcommon_files})
 target_compile_definitions(common-objs PRIVATE
   $<TARGET_PROPERTY:${FMT_LIB},INTERFACE_COMPILE_DEFINITIONS>)
 add_dependencies(common-objs legacy-option-headers)
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
+target_link_libraries(common-objs PkgConfig::XXHASH)
 
 if(WITH_JAEGER)
   add_dependencies(common-objs jaeger_base)
diff --git a/src/common/Checksummer.h b/src/common/Checksummer.h
index a42f5b682a1..8faee143e01 100644
--- a/src/common/Checksummer.h
+++ b/src/common/Checksummer.h
@@ -8,7 +8,7 @@
 #include "include/byteorder.h"
 #include "include/ceph_assert.h"
 
-#include "xxHash/xxhash.h"
+#include "xxhash.h"
 
 class Checksummer {
 public:
diff --git a/src/crimson/CMakeLists.txt b/src/crimson/CMakeLists.txt
index 6bbd7b49ec7..1d10e8fd07a 100644
--- a/src/crimson/CMakeLists.txt
+++ b/src/crimson/CMakeLists.txt
@@ -122,7 +122,6 @@ add_library(crimson-common STATIC
   ${PROJECT_SOURCE_DIR}/src/osd/OSDMap.cc
   ${PROJECT_SOURCE_DIR}/src/osd/PGPeeringEvent.cc
   ${PROJECT_SOURCE_DIR}/src/common/scrub_types.cc
-  ${PROJECT_SOURCE_DIR}/src/xxHash/xxhash.c
   ${crimson_common_srcs}
   $<TARGET_OBJECTS:common_mountcephfs_objs>
   $<TARGET_OBJECTS:common-options-objs>)
@@ -146,10 +145,12 @@ if(NOT WITH_SYSTEM_BOOST)
   list(APPEND crimson_common_deps ${ZLIB_LIBRARIES})
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(crimson-common
   PUBLIC
     ${crimson_common_public_deps}
   PRIVATE
+    PkgConfig::XXHASH
     crc32 arch
     ${crimson_common_deps}
     OpenSSL::Crypto)
diff --git a/src/os/CMakeLists.txt b/src/os/CMakeLists.txt
index 5ceda652075..3581b317b67 100644
--- a/src/os/CMakeLists.txt
+++ b/src/os/CMakeLists.txt
@@ -47,9 +47,11 @@ if(HAVE_LIBZFS)
   list(APPEND libos_srcs $<TARGET_OBJECTS:os_zfs_objs>)
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 add_library(os STATIC ${libos_srcs})
 target_link_libraries(os
   blk
+  PkgConfig::XXHASH
   ${FMT_LIB})
 
 target_link_libraries(os heap_profiler kv)
-- 
2.49.0

