From d60ed24fcf1da7756292d176f5f9a3de09d8e156 Mon Sep 17 00:00:00 2001
From: "V.Krishn" <vkrishn4@gmail.com>
Date: Sun, 8 Mar 2015 09:35:27 +0000
Subject: [PATCH] Avoid-use-of-internal-glibc-sys-cdefs.h-header

---
 lib/private-libwebsockets.h | 1 -
 lib/sha-1.c                 | 1 -
 2 files changed, 2 deletions(-)

diff --git a/lib/private-libwebsockets.h b/lib/private-libwebsockets.h
index 6fae889..34b8224 100755
--- a/lib/private-libwebsockets.h
+++ b/lib/private-libwebsockets.h
@@ -194,7 +194,6 @@ typedef unsigned __int64 u_int64_t;
 #else
 
 #include <sys/stat.h>
-#include <sys/cdefs.h>
 #include <sys/time.h>
 
 #if defined(__APPLE__)
diff --git a/lib/sha-1.c b/lib/sha-1.c
index 98b208b..bcad845 100644
--- a/lib/sha-1.c
+++ b/lib/sha-1.c
@@ -99,7 +99,6 @@ static const unsigned int _K[] =
 		sha1_step(ctxt);		\
 	}
 
-static void sha1_step __P((struct sha1_ctxt *));
 
 static void
 sha1_step(struct sha1_ctxt *ctxt)
-- 
2.2.2

