--- ./set_aliases.orig
+++ ./set_aliases
@@ -144,7 +144,7 @@
 default_make ()
 {
 save_path
-$INSTALL_DIR/studio/spec/$ISE_PLATFORM/bin/finish_freezing -library >> $COMPILE_LOG
+$INSTALL_DIR/studio/spec/$ISE_PLATFORM/bin/finish_freezing -library
 cd $TMP_PATH
 }
 all_makes ()
@@ -193,7 +193,7 @@
 	CANCEL
 fi
 cd F_code
-$INSTALL_DIR/studio/spec/$ISE_PLATFORM/bin/finish_freezing -silent >> $COMPILE_LOG
+$INSTALL_DIR/studio/spec/$ISE_PLATFORM/bin/finish_freezing
 if [ ! -f $1 ]; then
 	remtrace Couldnt generate $1
 	# Do not cancel if requested by providing a 2nd argument.
