--- a/Makefile.in
+++ b/Makefile.in
@@ -467,7 +467,7 @@
 
 install_doc: install_doc_html install_doc_man
 
-install: install_bin install_include install_lib install_doc
+install: install_include install_lib install_doc
 
 tests_unit: $(TESTS_UNIT:$(srcroot)%.c=$(objroot)%$(EXE))
 tests_integration: $(TESTS_INTEGRATION:$(srcroot)%.c=$(objroot)%$(EXE)) $(TESTS_INTEGRATION_CPP:$(srcroot)%.cpp=$(objroot)%$(EXE))
