--- orig/wiredtiger-2.6.0/dist/s_typedef
+++ wiredtiger-2.6.0/dist/s_typedef
@@ -21,7 +21,7 @@
 	l=`ls ../src/include/*.[hi] ../src/include/*.in |
 	    sed -e '/wiredtiger.*/d' -e '/queue.h/d'`
 	egrep -h \
-	    '^[	 ]*(((struct|union)[	 ].*__wt_.*{)|WT_PACKED_STRUCT_BEGIN)' \
+	    '^[	 ]*(((struct|union)[	 ].*__wt_.*\{)|WT_PACKED_STRUCT_BEGIN)' \
 	    $l |
 	    sed -e 's/WT_PACKED_STRUCT_BEGIN(\(.*\))/struct \1 {/' \
 	        -e 's/WT_COMPILER_TYPE_ALIGN(.*)[ ]*//' \
