diff -rupN jack-1.9.10.org/tests/iodelay.cpp jack-1.9.10/tests/iodelay.cpp
--- jack-1.9.10.org/tests/iodelay.cpp	2014-07-18 08:28:13.000000000 -0400
+++ jack-1.9.10/tests/iodelay.cpp	2016-02-13 18:38:33.280737975 -0500
@@ -167,8 +167,8 @@ static jack_client_t  *jack_handle;
 static jack_port_t    *jack_capt;
 static jack_port_t    *jack_play;
 
-jack_latency_range_t   capture_latency = {-1, -1};
-jack_latency_range_t   playback_latency = {-1, -1};
+jack_latency_range_t   capture_latency = {(jack_nframes_t)-1, (jack_nframes_t)-1};
+jack_latency_range_t   playback_latency = {(jack_nframes_t)-1, (jack_nframes_t)-1};
 
 void
 latency_cb (jack_latency_callback_mode_t mode, void *arg)
@@ -266,4 +266,4 @@ int main (int ac, char *av [])
     return 0;
 }
 
-// --------------------------------------------------------------------------------
\ No newline at end of file
+// --------------------------------------------------------------------------------
