diff --git a/common/wscript b/common/wscript
index afc251a2..70503788 100644
--- a/common/wscript
+++ b/common/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 import re
diff --git a/compat/alloca/wscript b/compat/alloca/wscript
index 36798973..9e740f1c 100644
--- a/compat/alloca/wscript
+++ b/compat/alloca/wscript
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # encoding: utf-8
 #
 # Copyright (C) 2018 Karl Linden <karl.j.linden@gmail.com>
diff --git a/compat/wscript b/compat/wscript
index e3ec8d33..25eb2a3f 100644
--- a/compat/wscript
+++ b/compat/wscript
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # encoding: utf-8
 #
 # Copyright (C) 2018 Karl Linden <karl.j.linden@gmail.com>
diff --git a/dbus/wscript b/dbus/wscript
index 535d4121..fa40b4f1 100644
--- a/dbus/wscript
+++ b/dbus/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 import os.path
diff --git a/example-clients/jack_control b/example-clients/jack_control
index 61ac40c1..54277543 100755
--- a/example-clients/jack_control
+++ b/example-clients/jack_control
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 from __future__ import print_function
 name_base = 'org.jackaudio'
diff --git a/example-clients/wscript b/example-clients/wscript
index 6163d222..a8857aa7 100644
--- a/example-clients/wscript
+++ b/example-clients/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 example_programs = {
diff --git a/man/wscript b/man/wscript
index 9e4d3043..def57078 100644
--- a/man/wscript
+++ b/man/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 import re
diff --git a/systemd/wscript b/systemd/wscript
index fc8b8f15..d10bdf96 100644
--- a/systemd/wscript
+++ b/systemd/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 
diff --git a/tests/wscript b/tests/wscript
index 756fd3bb..96a63bc9 100644
--- a/tests/wscript
+++ b/tests/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 
 test_programs = {
diff --git a/waf b/waf
index a2efc852..845fba5e 100755
--- a/waf
+++ b/waf
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # encoding: latin-1
 # Thomas Nagy, 2005-2018
 #
diff --git a/wscript b/wscript
index 5e8485e6..7e48f750 100644
--- a/wscript
+++ b/wscript
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#! /usr/bin/python3
 # encoding: utf-8
 from __future__ import print_function
 

