--- programs/Makefile	2014-07-31 12:20:17.748459143 -0400
+++ programs/Makefile	2014-07-31 12:21:35.388454356 -0400
@@ -43,7 +43,7 @@
 	test/selftest$(EXEXT)		test/ssl_test$(EXEXT)		\
 	util/pem2der$(EXEXT)		util/strerror$(EXEXT)		\
 	x509/cert_app$(EXEXT)		x509/crl_app$(EXEXT)		\
-	x509/cert_req$(EXEXT)
+	x509/cert_req$(EXEXT)		x509/cert_write$(EXEXT)
 
 ifdef OPENSSL
 APPS +=	test/o_p_test
@@ -241,6 +241,10 @@
 	echo   "  CC    x509/cert_req.c"
 	$(CC) $(CFLAGS) $(OFLAGS) x509/cert_req.c    $(LDFLAGS) -o $@
 
+x509/cert_write$(EXEXT): x509/cert_write.c ../library/libmbedtls.a
+	echo   "  CC    x509/cert_write.c"
+	$(CC) $(LOCAL_CFLAGS) $(CFLAGS) x509/cert_write.c    $(LOCAL_LDFLAGS) $(LDFLAGS) -o $@
+
 clean:
 ifndef WINDOWS
 	rm -f $(APPS)
