--- a/clntlib/restart.c
+++ b/clntlib/restart.c
@@ -217,19 +217,7 @@
 	if (!f) {
 		dcc_error_msg("fopen(%s): %s", pidpath->c, ERROR_STR());
 	} else {
-#ifdef linux
-		/* Linux threads are broken.  Signals given the
-		 * original process are delivered to only the
-		 * thread that happens to have that PID.  The
-		 * sendmail libmilter thread that needs to hear
-		 * SIGINT and other signals is known only to the milter.
-		 * Unless you put its PID into the file, it will not hear
-		 * the signals.  That breaks scripts that need to stop dccm.
-		 * However, signaling the process group works. */
-		fprintf(f, "-%d\n", (u_int)getpgrp());
-#else
 		fprintf(f, "%d\n", (u_int)getpid());
-#endif
 		fclose(f);
 	}
 }
