--- a/libqalculate.pc.in	2021-10-09 05:03:47.712401038 +0200
+++ b/libqalculate.pc.in	2021-10-09 05:06:54.039307183 +0200
@@ -8,5 +8,6 @@
 Version: @VERSION@
 Libs.private: @LIBCURL_LIBS@ @ICU_LIBS@ @LTLIBINTL@ @LTLIBICONV@ -lmpfr -lgmp -lpthread
 Required.private: libxml-2.0 >= 2.3.8
+Requires.private: gmp mpfr
 Libs: -L${libdir} -lqalculate
 Cflags: @LIBCURL_CFLAGS@ @ICU_CFLAGS@ -I${includedir}
