diff --git a/test/gstreamer/gstreamer_multi_stream_test.cpp b/test/gstreamer/gstreamer_multi_stream_test.cpp
index cd669308..d15b5f53 100644
--- a/test/gstreamer/gstreamer_multi_stream_test.cpp
+++ b/test/gstreamer/gstreamer_multi_stream_test.cpp
@@ -40,10 +40,10 @@ protected:
 			return status_;
 
 		const gchar *streamDescription = "queue ! fakesink";
-		g_autoptr(GError) error = NULL;
+		g_autoptr(GError) error = nullptr;
 
 		stream0_ = gst_parse_bin_from_description_full(streamDescription, TRUE,
-							       NULL,
+							       nullptr,
 							       GST_PARSE_FLAG_FATAL_ERRORS,
 							       &error);
 		if (!stream0_) {
@@ -53,7 +53,7 @@ protected:
 		g_object_ref_sink(stream0_);
 
 		stream1_ = gst_parse_bin_from_description_full(streamDescription, TRUE,
-							       NULL,
+							       nullptr,
 							       GST_PARSE_FLAG_FATAL_ERRORS,
 							       &error);
 		if (!stream1_) {
@@ -72,7 +72,7 @@ protected:
 	{
 		/* Build the pipeline */
 		gst_bin_add_many(GST_BIN(pipeline_), libcameraSrc_,
-				 stream0_, stream1_, NULL);
+				 stream0_, stream1_, nullptr);
 
 		g_autoptr(GstPad) src_pad = gst_element_get_static_pad(libcameraSrc_, "src");
 		g_autoptr(GstPad) request_pad = gst_element_request_pad_simple(libcameraSrc_, "src_%u");
