diff --git a/test/gstreamer/gstreamer_multi_stream_test.cpp b/test/gstreamer/gstreamer_multi_stream_test.cpp
index d3d7a9d3..dad01bb0 100644
--- a/test/gstreamer/gstreamer_multi_stream_test.cpp
+++ b/test/gstreamer/gstreamer_multi_stream_test.cpp
@@ -80,11 +80,11 @@ protected:
 
 	int run() override
 	{
-		g_object_set(libcameraSrc_, "camera-name", cameraName_.c_str(), NULL);
+		g_object_set(libcameraSrc_, "camera-name", cameraName_.c_str(), nullptr);
 
 		/* Build the pipeline */
 		gst_bin_add_many(GST_BIN(pipeline_), libcameraSrc_,
-				 stream0_, stream1_, NULL);
+				 stream0_, stream1_, nullptr);
 
 		g_autoptr(GstPad) src_pad = gst_element_get_static_pad(libcameraSrc_, "src");
 		g_autoptr(GstPad) request_pad = gst_element_get_request_pad(libcameraSrc_, "src_%u");
diff --git a/test/gstreamer/gstreamer_single_stream_test.cpp b/test/gstreamer/gstreamer_single_stream_test.cpp
index a0dd12cf..66a265bf 100644
--- a/test/gstreamer/gstreamer_single_stream_test.cpp
+++ b/test/gstreamer/gstreamer_single_stream_test.cpp
@@ -51,7 +51,7 @@ protected:
 	int run() override
 	{
 		/* Build the pipeline */
-		gst_bin_add_many(GST_BIN(pipeline_), libcameraSrc_, stream0_, NULL);
+		gst_bin_add_many(GST_BIN(pipeline_), libcameraSrc_, stream0_, nullptr);
 		if (gst_element_link(libcameraSrc_, stream0_) != TRUE) {
 			g_printerr("Elements could not be linked.\n");
 			return TestFail;
