diff --git a/src/libcamera/base/meson.build b/src/libcamera/base/meson.build
index 05fed7ac..aab6d1bc 100644
--- a/src/libcamera/base/meson.build
+++ b/src/libcamera/base/meson.build
@@ -22,9 +22,7 @@ libcamera_base_sources = files([
 libdw = cc.find_library('libdw', required : false)
 libunwind = cc.find_library('libunwind', required : false)
 
-if cc.has_header_symbol('execinfo.h', 'backtrace')
-    config_h.set('HAVE_BACKTRACE', 1)
-endif
+config_h.set('HAVE_BACKTRACE', 0)
 
 if libdw.found()
     config_h.set('HAVE_DW', 1)
