diff --git a/src/lc-compliance/meson.build b/src/lc-compliance/meson.build
index 130ddbb5..1a4a4f4e 100644
--- a/src/lc-compliance/meson.build
+++ b/src/lc-compliance/meson.build
@@ -16,8 +16,7 @@ if get_option('android_platform') == 'cros'
     endif
 
 else
-    libgtest_sp = subproject('gtest')
-    libgtest = libgtest_sp.get_variable('gtest_dep')
+    libgtest = dependency('gtest')
 endif
 
 lc_compliance_enabled = true
