diff --git a/Documentation/conf.py b/Documentation/conf.py
index 78602c85..222e05a7 100644
--- a/Documentation/conf.py
+++ b/Documentation/conf.py
@@ -56,7 +56,7 @@ master_doc = 'index'
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+language = "en"
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
