--- a/tests/test_surface.py
+++ b/tests/test_surface.py
@@ -262,18 +262,6 @@
     with pytest.raises(TypeError):
         cairo.SVGSurface.version_to_string(object())
 
-
-@pytest.mark.skipif(not hasattr(cairo.SVGSurface, "get_document_unit"),
-                    reason="too old cairo")
-def test_svg_surface_get_document_unit():
-    with cairo.SVGSurface(None, 10, 10) as surface:
-        assert surface.get_document_unit() == cairo.SVGUnit.PT
-        surface.set_document_unit(cairo.SVGUnit.PX)
-        assert surface.get_document_unit() == cairo.SVGUnit.PX
-    with pytest.raises(cairo.Error):
-        surface.set_document_unit(cairo.SVGUnit.PX)
-
-
 def test_svg_surface_restrict_to_version():
     surface = cairo.SVGSurface(None, 10, 10)
     surface.restrict_to_version(cairo.SVG_VERSION_1_1)

--- a/tests/test_typing.py
+++ b/tests/test_typing.py
@@ -16,7 +16,7 @@
         names = set()
         for key, value in vars(t).items():
             if key in ["XlibSurface", "XCBSurface", "Win32PrintingSurface",
-                       "Win32Surface"]:
+                       "Win32Surface", "TeeSurface"]:
                 continue
             if key.startswith("_"):
                 continue
