diff --git a/vendor/github.com/bugsnag/panicwrap/dup2.go b/vendor/github.com/bugsnag/panicwrap/dup2.go
index de523c8..639bf8e 100644
--- a/vendor/github.com/bugsnag/panicwrap/dup2.go
+++ b/vendor/github.com/bugsnag/panicwrap/dup2.go
@@ -1,4 +1,4 @@
-// +build darwin dragonfly freebsd linux,!arm64 netbsd openbsd
+//go:build !(linux && (arm64 || loong64 || riscv64))
 
 package panicwrap
 
diff --git a/vendor/github.com/bugsnag/panicwrap/dup3.go b/vendor/github.com/bugsnag/panicwrap/dup3.go
index 9721b36..00701cf 100644
--- a/vendor/github.com/bugsnag/panicwrap/dup3.go
+++ b/vendor/github.com/bugsnag/panicwrap/dup3.go
@@ -1,4 +1,4 @@
-// +build linux,arm64
+//go:build linux && (arm64 || loong64 || riscv64)
 
 package panicwrap
 
