diff --git a/test-suite/test/sysdep/ffi_include_std.lua b/test-suite/test/sysdep/ffi_include_std.lua
index 301c073..d29d554 100644
--- a/test-suite/test/sysdep/ffi_include_std.lua
+++ b/test-suite/test/sysdep/ffi_include_std.lua
@@ -28,12 +28,7 @@ do
 ]]
   fp:close()
 
-  local flags
-  if jit.arch == "arm64" then
-    flags = ""
-  else
-    flags = ffi.abi("32bit") and "-m32" or "-m64"
-  end
+  local flags = ""
   fp = assert(io.popen("cc -E -P -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE /tmp/__tmp.c "..flags))
   local s = fp:read("*a")
   fp:close()
