commit 52260c8dbc45c4493aea6e458f486a18f7ff8b96
Author: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date:   Tue Jan 21 16:10:54 2025 +0900

    second

diff --git a/src/main-window.c b/src/main-window.c
index 574a75f..45b8ecd 100644
--- a/src/main-window.c
+++ b/src/main-window.c
@@ -6250,7 +6250,7 @@ GtkWidget* main_task_view_new( FMMainWindow* main_window )
 
 // ============== socket commands
 
-gboolean bool( const char* value )
+gboolean truthy( const char* value )
 {
     return ( !( value && value[0] ) || !strcmp( value, "1") || 
                     !strcmp( value, "true") || 
@@ -6441,14 +6441,14 @@ _missing_arg:
         }
         else if ( !strcmp( argv[i], "window_maximized" ) )
         {
-            if ( bool( argv[i+1] ) )
+            if ( truthy( argv[i+1] ) )
                 gtk_window_maximize( GTK_WINDOW( main_window ) );
             else
                 gtk_window_unmaximize( GTK_WINDOW( main_window ) );
         }
         else if ( !strcmp( argv[i], "window_fullscreen" ) )
         {
-            xset_set_b( "main_full", bool( argv[i+1] ) );
+            xset_set_b( "main_full", truthy( argv[i+1] ) );
             on_fullscreen_activate( NULL, main_window );
         }
         else if ( !strcmp( argv[i], "screen_size" ) )
@@ -6591,7 +6591,7 @@ _missing_arg:
                                                                     argv[i] );
                     return 2;
                 }
-                xset_set_b_panel( j, "show", bool( argv[i+1] ) );
+                xset_set_b_panel( j, "show", truthy( argv[i+1] ) );
                 show_panels_all_windows( NULL, main_window );
                 return 0;
             }
@@ -6602,9 +6602,9 @@ _missing_arg:
             if ( use_mode )
                 xset_set_b_panel_mode( panel, str,
                                         main_window->panel_context[panel-1],
-                                        bool( argv[i+1] ) );
+                                        truthy( argv[i+1] ) );
             else
-                xset_set_b_panel( panel, str, bool( argv[i+1] ) );
+                xset_set_b_panel( panel, str, truthy( argv[i+1] ) );
             update_views_all_windows( NULL, file_browser );
         }
         else if ( !strcmp( argv[i], "panel_hslider_top" ) ||
@@ -6709,23 +6709,23 @@ _missing_arg:
         {
             if ( !strcmp( argv[i] + 5, "ascend" ) )
             {
-                ptk_file_browser_set_sort_type( file_browser, bool( argv[i+1] ) ?
+                ptk_file_browser_set_sort_type( file_browser, truthy( argv[i+1] ) ?
                                         GTK_SORT_ASCENDING : GTK_SORT_DESCENDING );
                 return 0;
             }
             else if ( !strcmp( argv[i] + 5, "natural" ) )
             {
                 str = "sortx_natural";
-                xset_set_b( str, bool( argv[i+1] ) );
+                xset_set_b( str, truthy( argv[i+1] ) );
             }
             else if ( !strcmp( argv[i] + 5, "case" ) )
             {
                 str = "sortx_case";
-                xset_set_b( str, bool( argv[i+1] ) );
+                xset_set_b( str, truthy( argv[i+1] ) );
             }
             else if ( !strcmp( argv[i] + 5, "hidden_first" ) )
             {
-                str = bool( argv[i+1] ) ? "sortx_hidfirst" : "sortx_hidlast";
+                str = truthy( argv[i+1] ) ? "sortx_hidfirst" : "sortx_hidlast";
                 xset_set_b( str, TRUE );
             }
             else if ( !strcmp( argv[i] + 5, "first" ) )
@@ -6748,7 +6748,7 @@ _missing_arg:
         }
         else if ( !strcmp( argv[i], "show_thumbnails" ) )
         {
-            if ( app_settings.show_thumbnail != bool( argv[i+1] ) )
+            if ( app_settings.show_thumbnail != truthy( argv[i+1] ) )
                 main_window_toggle_thumbnails_all_windows();
         }
         else if ( !strcmp( argv[i], "large_icons" ) )
@@ -6757,7 +6757,7 @@ _missing_arg:
             {
                 xset_set_b_panel_mode( panel, "list_large",
                                         main_window->panel_context[panel-1],
-                                        bool( argv[i+1] ) );
+                                        truthy( argv[i+1] ) );
                 update_views_all_windows( NULL, file_browser );
             }
         }
diff --git a/src/settings.c b/src/settings.c
index e405af8..07e0fa4 100644
--- a/src/settings.c
+++ b/src/settings.c
@@ -2731,9 +2731,9 @@ gboolean xset_get_bool( const char* name, const char* var )
 gboolean xset_get_bool_panel( int panel, const char* name, const char* var )
 {
     char* fullname = g_strdup_printf( "panel%d_%s", panel, name );
-    gboolean bool = xset_get_bool( fullname, var );
+    gboolean truthy = xset_get_bool( fullname, var );
     g_free( fullname );
-    return bool;
+    return truthy;
 }
 
 int xset_get_int_set( XSet* set, const char* var )
