This test requires ConsoleKit to be running.

--- polkit-0.105/test/polkitbackend/Makefile.am.old	2012-04-24 11:05:34.000000000 -0500
+++ polkit-0.105/test/polkitbackend/Makefile.am	2017-09-27 20:48:42.479959296 -0500
@@ -36,8 +36,8 @@
 TEST_PROGS += polkitbackendlocalauthorizationstoretest
 polkitbackendlocalauthorizationstoretest_SOURCES = polkitbackendlocalauthorizationstoretest.c
 
-TEST_PROGS += polkitbackendlocalauthoritytest
-polkitbackendlocalauthoritytest_SOURCES = polkitbackendlocalauthoritytest.c
+#TEST_PROGS += polkitbackendlocalauthoritytest
+#polkitbackendlocalauthoritytest_SOURCES = polkitbackendlocalauthoritytest.c
 
 # ----------------------------------------------------------------------------------------------------
 
