--- polkit-0.105/test/mocklibc/src/pwd.c.old	2012-04-24 11:05:34.000000000 -0500
+++ polkit-0.105/test/mocklibc/src/pwd.c	2017-09-27 19:40:57.883227673 -0500
@@ -16,6 +16,7 @@
  * Author: Nikki VonHollen <vonhollen@gmail.com>
  */
 
+#define _GNU_SOURCE
 #include <pwd.h>
 
 #include <stdio.h>
--- polkit-0.105/test/mocklibc/src/grp.c.old	2012-04-24 11:05:34.000000000 -0500
+++ polkit-0.105/test/mocklibc/src/grp.c	2017-09-27 19:44:57.759238450 -0500
@@ -16,6 +16,7 @@
  * Author: Nikki VonHollen <vonhollen@gmail.com>
  */
 
+#define _GNU_SOURCE
 #include <grp.h>
 
 #include <stdio.h>
