Only in b/devtools/Site: site.config.m4
diff -ru a/libmilter/libmilter.h b/libmilter/libmilter.h
--- a/libmilter/libmilter.h	2014-06-12 19:30:47.000000000 +0200
+++ b/libmilter/libmilter.h	2016-11-06 22:58:10.106901590 +0100
@@ -127,10 +127,10 @@
 # define MI_SOCK_READ(s, b, l)	read(s, b, l)
 # define MI_SOCK_READ_FAIL(x)	((x) < 0)
 # define MI_SOCK_WRITE(s, b, l)	write(s, b, l)
-
-# define thread_create(ptid,wr,arg) pthread_create(ptid, NULL, wr, arg)
 # define sthread_get_id()	pthread_self()

+extern int thread_create(pthread_t *ptid, void *(*wr) (void *), void *arg);
+
 typedef pthread_mutex_t smutex_t;
 # define smutex_init(mp)	(pthread_mutex_init(mp, NULL) == 0)
 # define smutex_destroy(mp)	(pthread_mutex_destroy(mp) == 0)
diff -ru a/libmilter/main.c b/libmilter/main.c
--- a/libmilter/main.c	2014-06-12 19:30:47.000000000 +0200
+++ b/libmilter/main.c	2016-11-06 22:59:27.266904577 +0100
@@ -16,6 +16,12 @@
 #include <fcntl.h>
 #include <sys/stat.h>

+int thread_create(pthread_t *ptid, void *(*wr) (void *), void *arg) {
+	pthread_attr_t attr;
+	pthread_attr_init(&attr);
+	pthread_attr_setstacksize(&attr,8*1024*1024);
+	return pthread_create(ptid, &attr, wr, arg);
+}

 static smfiDesc_ptr smfi = NULL;

Only in b: obj.Linux.4.4.17-2-t430.x86_64
