diff -ruN a/data/com.belmoussaoui.Authenticator.metainfo.xml.in.in b/data/com.belmoussaoui.Authenticator.metainfo.xml.in.in
--- a/data/com.belmoussaoui.Authenticator.metainfo.xml.in.in	2025-03-25 18:14:42.000000000 +0000
+++ b/data/com.belmoussaoui.Authenticator.metainfo.xml.in.in	2025-05-16 05:49:29.688217674 +0000
@@ -122,13 +122,19 @@
       </description>
     </release>
     <release version="4.1.5" date="2022-06-06">
-      <description>Fix a crash when loading an account token</description>
+      <description>
+        <p>Fix a crash when loading an account token</p>
+      </description>
     </release>
     <release version="4.1.4" date="2022-06-04">
-      <description>Fix fallback to host keyring and shell search provider results</description>
+      <description>
+        <p>Fix fallback to host keyring and shell search provider results</p>
+      </description>
     </release>
     <release version="4.1.3" date="2022-05-31">
-      <description>Fallback to host keyring in case the portal is not available</description>
+      <description>
+        <p>Fallback to host keyring in case the portal is not available</p>
+      </description>
     </release>
     <release version="4.1.2" date="2022-05-29">
       <description>
@@ -332,8 +338,6 @@
   <provides>
     <id>com.github.bilelmoussaoui.Authenticator</id>
   </provides>
-  <!-- developer_name tag deprecated with Appstream 1.0 -->
-  <developer_name translatable="no">Bilal Elmoussaoui</developer_name>
   <developer id="com.belmoussaoui">
     <name translatable="no">Bilal Elmoussaoui</name>
   </developer>
