--- ./setup.cfg.old
+++ ./setup.cfg
@@ -8,6 +8,7 @@
     # for python 3.10+: mypy still relies on distutils on windows. We hence ignore those warnings
     ignore:The distutils package is deprecated and slated for removal in Python 3.12:DeprecationWarning
     ignore:The distutils.sysconfig module is deprecated, use sysconfig instead:DeprecationWarning
+    ignore:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning
 
 [flake8]
 max-line-length = 120
