diff --git a/setup.cfg b/setup.cfg
index 505479b..907ea17 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -11,6 +11,7 @@ filterwarnings =
     # for Python 3.11
     ignore:path is deprecated.*:DeprecationWarning:certifi
     ignore:module 'sre_constants' is deprecated:DeprecationWarning:pkg_resources
+    ignore:Creating a LegacyVersion has been deprecated and will be removed in the next major release:DeprecationWarning
 
 [flake8]
 max-line-length = 120
