diff --git a/src/client/netclient.cpp b/src/client/netclient.cpp
index 30b8d5c..c761473 100644
--- a/src/client/netclient.cpp
+++ b/src/client/netclient.cpp
@@ -43,7 +43,7 @@ CNetClient::CNetClient(bool bMustLogin, bool bUseSSL):CNet()
     {
       showDebug(3, "initializing client ssl\n");
       SSLeay_add_ssl_algorithms();
-      meth = SSLv2_client_method();
+      meth = TLS_client_method();
       SSL_load_error_strings();
       ctx = SSL_CTX_new(meth);
       if (!ctx)
diff --git a/src/server/netserver.cpp b/src/server/netserver.cpp
index b3ba1c7..6bd9944 100644
--- a/src/server/netserver.cpp
+++ b/src/server/netserver.cpp
@@ -39,7 +39,7 @@ CNetServer::CNetServer(unsigned short int port):CNet()
     {
       SSL_load_error_strings();
       SSLeay_add_ssl_algorithms();
-      meth = SSLv23_server_method();
+      meth = TLS_server_method();
       ctx = SSL_CTX_new(meth);
       if (!ctx)
         {
