diff --git a/tests/Makefile.am b/tests/Makefile.am
index 5d77f9c..9adeede 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -1,4 +1,4 @@
-TESTS = proxy proxy-continuous threaded oauth oauth-async oauth2 flickr lastfm xml custom-serialize
+TESTS = proxy proxy-continuous threaded oauth2 flickr lastfm xml custom-serialize
 # TODO: fix this test case
 XFAIL_TESTS = xml
 
@@ -11,8 +11,6 @@ check_PROGRAMS = $(TESTS)
 proxy_SOURCES = proxy.c
 proxy_continuous_SOURCES = proxy-continuous.c
 threaded_SOURCES = threaded.c
-oauth_SOURCES = oauth.c
-oauth_async_SOURCES = oauth-async.c
 oauth2_SOURCES = oauth2.c
 flickr_SOURCES = flickr.c
 lastfm_SOURCES = lastfm.c
diff --git a/tests/Makefile.in b/tests/Makefile.in
index 35709e5..02dd069 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -88,7 +88,7 @@ POST_UNINSTALL = :
 build_triplet = @build@
 host_triplet = @host@
 TESTS = proxy$(EXEEXT) proxy-continuous$(EXEEXT) threaded$(EXEEXT) \
-	oauth$(EXEEXT) oauth-async$(EXEEXT) oauth2$(EXEEXT) \
+	oauth2$(EXEEXT) \
 	flickr$(EXEEXT) lastfm$(EXEEXT) xml$(EXEEXT) \
 	custom-serialize$(EXEEXT)
 XFAIL_TESTS = xml$(EXEEXT)
@@ -110,7 +110,7 @@ CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 am__EXEEXT_1 = proxy$(EXEEXT) proxy-continuous$(EXEEXT) \
-	threaded$(EXEEXT) oauth$(EXEEXT) oauth-async$(EXEEXT) \
+	threaded$(EXEEXT) \
 	oauth2$(EXEEXT) flickr$(EXEEXT) lastfm$(EXEEXT) xml$(EXEEXT) \
 	custom-serialize$(EXEEXT)
 am_custom_serialize_OBJECTS = custom-serialize.$(OBJEXT)
@@ -126,12 +126,6 @@ flickr_LDADD = $(LDADD)
 am_lastfm_OBJECTS = lastfm.$(OBJEXT)
 lastfm_OBJECTS = $(am_lastfm_OBJECTS)
 lastfm_LDADD = $(LDADD)
-am_oauth_OBJECTS = oauth.$(OBJEXT)
-oauth_OBJECTS = $(am_oauth_OBJECTS)
-oauth_LDADD = $(LDADD)
-am_oauth_async_OBJECTS = oauth-async.$(OBJEXT)
-oauth_async_OBJECTS = $(am_oauth_async_OBJECTS)
-oauth_async_LDADD = $(LDADD)
 am_oauth2_OBJECTS = oauth2.$(OBJEXT)
 oauth2_OBJECTS = $(am_oauth2_OBJECTS)
 oauth2_LDADD = $(LDADD)
@@ -182,11 +176,11 @@ am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
 SOURCES = $(custom_serialize_SOURCES) $(flickr_SOURCES) \
-	$(lastfm_SOURCES) $(oauth_SOURCES) $(oauth_async_SOURCES) \
+	$(lastfm_SOURCES) \
 	$(oauth2_SOURCES) $(proxy_SOURCES) $(proxy_continuous_SOURCES) \
 	$(threaded_SOURCES) $(xml_SOURCES)
 DIST_SOURCES = $(custom_serialize_SOURCES) $(flickr_SOURCES) \
-	$(lastfm_SOURCES) $(oauth_SOURCES) $(oauth_async_SOURCES) \
+	$(lastfm_SOURCES) \
 	$(oauth2_SOURCES) $(proxy_SOURCES) $(proxy_continuous_SOURCES) \
 	$(threaded_SOURCES) $(xml_SOURCES)
 am__can_run_installinfo = \
@@ -576,8 +570,6 @@ AM_LDFLAGS = $(SOUP_LIBS) $(GCOV_LDFLAGS) \
 proxy_SOURCES = proxy.c
 proxy_continuous_SOURCES = proxy-continuous.c
 threaded_SOURCES = threaded.c
-oauth_SOURCES = oauth.c
-oauth_async_SOURCES = oauth-async.c
 oauth2_SOURCES = oauth2.c
 flickr_SOURCES = flickr.c
 lastfm_SOURCES = lastfm.c
@@ -638,14 +630,6 @@ lastfm$(EXEEXT): $(lastfm_OBJECTS) $(lastfm_DEPENDENCIES) $(EXTRA_lastfm_DEPENDE
 	@rm -f lastfm$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(lastfm_OBJECTS) $(lastfm_LDADD) $(LIBS)
 
-oauth$(EXEEXT): $(oauth_OBJECTS) $(oauth_DEPENDENCIES) $(EXTRA_oauth_DEPENDENCIES) 
-	@rm -f oauth$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(oauth_OBJECTS) $(oauth_LDADD) $(LIBS)
-
-oauth-async$(EXEEXT): $(oauth_async_OBJECTS) $(oauth_async_DEPENDENCIES) $(EXTRA_oauth_async_DEPENDENCIES) 
-	@rm -f oauth-async$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(oauth_async_OBJECTS) $(oauth_async_LDADD) $(LIBS)
-
 oauth2$(EXEEXT): $(oauth2_OBJECTS) $(oauth2_DEPENDENCIES) $(EXTRA_oauth2_DEPENDENCIES) 
 	@rm -f oauth2$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(oauth2_OBJECTS) $(oauth2_LDADD) $(LIBS)
@@ -675,8 +659,6 @@ distclean-compile:
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/custom-serialize.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/flickr.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/lastfm.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/oauth-async.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/oauth.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/oauth2.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/proxy-continuous.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/proxy.Po@am__quote@
@@ -924,20 +906,6 @@ threaded.log: threaded$(EXEEXT)
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-oauth.log: oauth$(EXEEXT)
-	@p='oauth$(EXEEXT)'; \
-	b='oauth'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
-oauth-async.log: oauth-async$(EXEEXT)
-	@p='oauth-async$(EXEEXT)'; \
-	b='oauth-async'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 oauth2.log: oauth2$(EXEEXT)
 	@p='oauth2$(EXEEXT)'; \
 	b='oauth2'; \
