diff --git a/Makefile b/Makefile
index 471f5fc..d41ba40 100644
--- a/Makefile
+++ b/Makefile
@@ -9,7 +9,7 @@ LUA_INCLUDE_DIR ?= /usr/include/lua5.3
 DESTDIR=/usr/bin
 
 compile-lua:
-	fennel --compile src/toAPK.fnl > src/toAPK.lua
+	fennel5.3 --compile src/toAPK.fnl > src/toAPK.lua
 	sed -i '1 i\-- Author: Will Sinatra <wpsinatra@gmail.com> | License: GPLv3' src/toAPK.lua
 	sed -i '1 i\#!/usr/bin/$(LUA)' src/toAPK.lua
 
@@ -17,7 +17,7 @@ install-lua:
 	install ./src/toAPK.lua -D $(DESTDIR)/toAPK
 
 compile-bin:
-	cd ./src/ && fennel --compile-binary toAPK.fnl toAPK-bin $(STATIC_LUA_LIB) $(LUA_INCLUDE_DIR)
+	cd ./src/ && fennel5.3 --compile-binary toAPK.fnl toAPK-bin $(STATIC_LUA_LIB) $(LUA_INCLUDE_DIR)
 
 install-bin:
 	install ./src/toAPK-bin -D $(DESTDIR)/toAPK
