diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index c15b389..9ac12bc 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -362,7 +362,7 @@ endif()  # SANITIZER != "none"
 set_target_properties(libjxl_test PROPERTIES PREFIX "tests/")
 target_link_libraries(libjxl_test jxl)
 if (NOT MSVC)
-target_compile_options(libjxl_test PRIVATE -Wall -Wextra -Werror)
+target_compile_options(libjxl_test PRIVATE -Wall -Wextra)
 if(NOT WIN32)
   target_compile_options(libjxl_test PRIVATE -pedantic)
 endif()  # NOT WIN32
