diff -upr valgrind-3.14.0.orig/configure.ac valgrind-3.14.0/configure.ac
--- valgrind-3.14.0.orig/configure.ac	2019-06-13 21:56:14.023469250 +0200
+++ valgrind-3.14.0/configure.ac	2019-06-13 21:56:35.130452552 +0200
@@ -1103,7 +1103,7 @@ case "${GLIBC_VERSION}" in
      musl)
 	AC_MSG_RESULT(Musl)
 	AC_DEFINE([MUSL_LIBC], 1, [Define to 1 if you're using Musl libc])
-	# no DEFAULT_SUPP file yet for musl libc.
+	DEFAULT_SUPP="musl.supp ${DEFAULT_SUPP}"
 	;;
      2.0|2.1|*)
 	AC_MSG_RESULT([unsupported version ${GLIBC_VERSION}])
