diff --git a/none/tests/s390x/ex_sig.c b/none/tests/s390x/ex_sig.c
index 9ff33fc..b5aaa7a 100644
--- a/none/tests/s390x/ex_sig.c
+++ b/none/tests/s390x/ex_sig.c
@@ -1,5 +1,7 @@
 #include <features.h>
+#ifdef __GLIBC__
 #include <fpu_control.h>
+#endif
 #include <signal.h>
 #include <sys/types.h>
 #include <signal.h>
diff --git a/none/tests/s390x/test_sig.c b/none/tests/s390x/test_sig.c
index b24dbbe..b93088b 100644
--- a/none/tests/s390x/test_sig.c
+++ b/none/tests/s390x/test_sig.c
@@ -1,5 +1,7 @@
 #include <features.h>
+#ifdef __GLIBC__
 #include <fpu_control.h>
+#endif
 #include <signal.h>
 #include <sys/types.h>
 #include <signal.h>
diff --git a/none/tests/s390x/traps.c b/none/tests/s390x/traps.c
index 86874c3..ce1fc74 100644
--- a/none/tests/s390x/traps.c
+++ b/none/tests/s390x/traps.c
@@ -1,5 +1,7 @@
 #include <features.h>
+#ifdef __GLIBC__
 #include <fpu_control.h>
+#endif
 #include <signal.h>
 #include <sys/types.h>
 #include <signal.h>
diff --git a/tests/s390x_features.c b/tests/s390x_features.c
index 507f3ab..40774de 100644
--- a/tests/s390x_features.c
+++ b/tests/s390x_features.c
@@ -48,12 +48,16 @@ jmp_buf env;
 // of the CPU facility list.  To read the HWCAP, use 'getauxval' if available --
 // which should be the case with glibc versions >= 2.16.  A system with an older
 // glibc is unlikely to support any of these features anyhow.
+#ifdef __GLIBC__
 #if __GLIBC_PREREQ(2, 16)
 #include <sys/auxv.h>
 #define GET_HWCAP() getauxval(AT_HWCAP)
 #else
 #define GET_HWCAP() 0UL
 #endif
+#else
+#define GET_HWCAP() 0UL
+#endif
 
 /* Number of double words needed to store all facility bits. */
 #define S390_NUM_FACILITY_DW 3
