diff --git a/memcheck/tests/linux/timerfd-syscall.c b/memcheck/tests/linux/timerfd-syscall.c
index 61d75b5..1721249 100644
--- a/memcheck/tests/linux/timerfd-syscall.c
+++ b/memcheck/tests/linux/timerfd-syscall.c
@@ -78,6 +78,9 @@
 #elif defined(__s390x__)
 #define __NR_timerfd_settime 320
 #define __NR_timerfd_gettime 321
+#elif defined(__arm__) && defined(__NR_timerfd_settime64) && defined(__NR_timerfd_gettime64)
+#define __NR_timerfd_settime __NR_timerfd_settime64
+#define __NR_timerfd_gettime __NR_timerfd_gettime64
 #else
 #error Cannot detect your architecture!
 #endif
