diff --git a/src/bindings/python/bt2/bt2/native_bt.i b/src/bindings/python/bt2/bt2/native_bt.i
index 2300cee..c43b178 100644
--- a/src/bindings/python/bt2/bt2/native_bt.i
+++ b/src/bindings/python/bt2/bt2/native_bt.i
@@ -83,12 +83,12 @@ typedef uint64_t bt_listener_id;
 
 %typemap(argout) (const char **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result, SWIG_Python_str_FromChar(*$1));
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result, SWIG_Python_str_FromChar(*$1));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
@@ -99,14 +99,14 @@ typedef uint64_t bt_listener_id;
 
 %typemap(argout) (bt_value **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_value, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
@@ -116,7 +116,7 @@ typedef uint64_t bt_listener_id;
 }
 
 %typemap(argout) uint64_t * {
-	$result = SWIG_Python_AppendOutput(resultobj,
+	$result = SWIG_AppendOutput(resultobj,
 			SWIG_From_unsigned_SS_long_SS_long((*$1)));
 }
 
@@ -126,7 +126,7 @@ typedef uint64_t bt_listener_id;
 }
 
 %typemap(argout) (int64_t *) {
-	$result = SWIG_Python_AppendOutput(resultobj, SWIG_From_long_SS_long((*$1)));
+	$result = SWIG_AppendOutput(resultobj, SWIG_From_long_SS_long((*$1)));
 }
 
 /* Output argument typemap for initialized unsigned int output parameter (always appends) */
@@ -135,7 +135,7 @@ typedef uint64_t bt_listener_id;
 }
 
 %typemap(argout) (unsigned int *) {
-	$result = SWIG_Python_AppendOutput(resultobj,
+	$result = SWIG_AppendOutput(resultobj,
 			SWIG_From_unsigned_SS_long_SS_long((uint64_t) (*$1)));
 }
 
@@ -145,7 +145,7 @@ typedef uint64_t bt_listener_id;
 }
 
 %typemap(argout) bt_bool * {
-	$result = SWIG_Python_AppendOutput(resultobj,
+	$result = SWIG_AppendOutput(resultobj,
 			SWIG_From_bool(*$1));
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_component.i b/src/bindings/python/bt2/bt2/native_bt_component.i
index aa3f725..a545dcc 100644
--- a/src/bindings/python/bt2/bt2/native_bt_component.i
+++ b/src/bindings/python/bt2/bt2/native_bt_component.i
@@ -31,14 +31,14 @@
 
 %typemap(argout) bt_self_component_port_input ** {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_self_component_port_input, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
@@ -51,14 +51,14 @@
 
 %typemap(argout) (bt_self_component_port_output **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_self_component_port_output, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_event_class.i b/src/bindings/python/bt2/bt2/native_bt_event_class.i
index 14f5162..6eaba67 100644
--- a/src/bindings/python/bt2/bt2/native_bt_event_class.i
+++ b/src/bindings/python/bt2/bt2/native_bt_event_class.i
@@ -32,8 +32,8 @@
 }
 
 %typemap(argout) bt_event_class_log_level * {
-	/* SWIG_Python_AppendOutput() steals the created object */
-	$result = SWIG_Python_AppendOutput($result, SWIG_From_int(*$1));
+	/* SWIG_AppendOutput() steals the created object */
+	$result = SWIG_AppendOutput($result, SWIG_From_int(*$1));
 }
 
 %include <babeltrace2/trace-ir/event-class.h>
diff --git a/src/bindings/python/bt2/bt2/native_bt_field_class.i b/src/bindings/python/bt2/bt2/native_bt_field_class.i
index 48f48c0..85af573 100644
--- a/src/bindings/python/bt2/bt2/native_bt_field_class.i
+++ b/src/bindings/python/bt2/bt2/native_bt_field_class.i
@@ -40,10 +40,10 @@
 			PyList_SET_ITEM(py_label_list, i, PyUnicode_FromString((*$1)[i]));
 		}
 
-		$result = SWIG_Python_AppendOutput($result, py_label_list);
+		$result = SWIG_AppendOutput($result, py_label_list);
 	} else {
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_graph.i b/src/bindings/python/bt2/bt2/native_bt_graph.i
index 0bda11f..bbd6906 100644
--- a/src/bindings/python/bt2/bt2/native_bt_graph.i
+++ b/src/bindings/python/bt2/bt2/native_bt_graph.i
@@ -32,14 +32,14 @@
 %typemap(argout)
 	(const bt_connection **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_connection, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
@@ -64,40 +64,40 @@
 
 %typemap(argout) (const bt_component_source **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_component_source, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
 %typemap(argout) (const bt_component_filter **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_component_filter, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
 %typemap(argout) (const bt_component_sink **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_component_sink, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_message.i b/src/bindings/python/bt2/bt2/native_bt_message.i
index 8cc9055..33a79a1 100644
--- a/src/bindings/python/bt2/bt2/native_bt_message.i
+++ b/src/bindings/python/bt2/bt2/native_bt_message.i
@@ -33,14 +33,14 @@
 %typemap(argout)
 	(const bt_clock_snapshot **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_clock_snapshot, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_message_iterator.i b/src/bindings/python/bt2/bt2/native_bt_message_iterator.i
index 23b6979..902ea52 100644
--- a/src/bindings/python/bt2/bt2/native_bt_message_iterator.i
+++ b/src/bindings/python/bt2/bt2/native_bt_message_iterator.i
@@ -32,14 +32,14 @@
 %typemap(argout)
 	(bt_message_iterator **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_message_iterator, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
diff --git a/src/bindings/python/bt2/bt2/native_bt_plugin.i b/src/bindings/python/bt2/bt2/native_bt_plugin.i
index edd91e7..074a0b4 100644
--- a/src/bindings/python/bt2/bt2/native_bt_plugin.i
+++ b/src/bindings/python/bt2/bt2/native_bt_plugin.i
@@ -32,14 +32,14 @@
 %typemap(argout)
 	(const bt_plugin **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_plugin, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
@@ -53,14 +53,14 @@
 %typemap(argout)
 	(const bt_plugin_set **) {
 	if (*$1) {
-		/* SWIG_Python_AppendOutput() steals the created object */
-		$result = SWIG_Python_AppendOutput($result,
+		/* SWIG_AppendOutput() steals the created object */
+		$result = SWIG_AppendOutput($result,
 				SWIG_NewPointerObj(SWIG_as_voidptr(*$1),
 					SWIGTYPE_p_bt_plugin_set, 0));
 	} else {
-		/* SWIG_Python_AppendOutput() steals Py_None */
+		/* SWIG_AppendOutput() steals Py_None */
 		Py_INCREF(Py_None);
-		$result = SWIG_Python_AppendOutput($result, Py_None);
+		$result = SWIG_AppendOutput($result, Py_None);
 	}
 }
 
