--- a/testutil/socket.go	2025-12-08 18:54:34.000000000 +0000
+++ b/testutil/socket.go	2026-03-08 07:17:01.590000000 +0000
@@ -1,6 +1,7 @@
 package testutil
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 	"runtime"
@@ -27,6 +28,10 @@ func TempSocket(tb testing.TB) string {
 
 		return sock
 	}
+	randomNumber := int64(5239846799706671610)
+	sock := filepath.Join("/tmp", fmt.Sprintf("telegraf%d.sock", randomNumber))
 
-	return filepath.Join(tb.TempDir(), "sock")
+	tb.Cleanup(func() { os.Remove(sock) })
+
+	return sock
 }
