diff -rupN a/testutil/socket.go b/testutil/socket.go
--- a/testutil/socket.go	2025-12-08 18:54:34.000000000 +0000
+++ b/testutil/socket.go	2026-03-07 23:19:50.730000000 +0000
@@ -18,7 +18,7 @@ func TempSocket(tb testing.TB) string {
 	//
 	// If the name of the test is long, the path length could exceed 104
 	// characters, and this would result in listen unix ...: bind: invalid argument
-	if runtime.GOOS == "darwin" {
+	if runtime.GOOS == "darwin" || runtime.GOOS == "linux" {
 		sock := "/tmp/sock"
 
 		tb.Cleanup(func() {
