diff --git a/gpscat b/gpscat
index 26c9acf..6cf4765 100755
--- a/gpscat
+++ b/gpscat
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 "Display GPS output.  Hexify it if necessary."
 
diff --git a/gpsfake b/gpsfake
index e02d64a..ec10974 100755
--- a/gpsfake
+++ b/gpsfake
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 '''
 gpsfake -- test harness for gpsd
diff --git a/ubxtool b/ubxtool
index 6bdb0ba..82f5b2e 100755
--- a/ubxtool
+++ b/ubxtool
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: UTF-8
 '''
 ubxtool -- u-blox configurator and packet decoder
diff --git a/zerk b/zerk
index 745c765..2e1b73a 100755
--- a/zerk
+++ b/zerk
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: UTF-8
 '''
 zerk -- GREIS configurator and packet decoder
 diff --git a/contrib/ntpshmviz b/contrib/ntpshmviz
index 5aa0016..371fd70 100755
--- a/contrib/ntpshmviz
+++ b/contrib/ntpshmviz
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 '''
 ntpshmviz - graph the drift of NTP servers
 Written by Keane Wolter <daemoneye2@gmail.com>
diff --git a/contrib/skyview2svg b/contrib/skyview2svg
index 7fdc327..3ed04c3 100644
--- a/contrib/skyview2svg
+++ b/contrib/skyview2svg
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 """
 skyview2svg -- Create an SVG image of GPS satellites sky view.
diff --git a/contrib/webgps.py b/contrib/webgps.py
index 99a70c6..a1bcbd2 100755
--- a/contrib/webgps.py
+++ b/contrib/webgps.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # encoding: utf-8
 
 """webgps.py
diff --git a/devtools/ais.py b/devtools/ais.py
index f269433..531a4c7 100755
--- a/devtools/ais.py
+++ b/devtools/ais.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # A Python AIVDM/AIVDO decoder
 #
diff --git a/devtools/aivdmtable b/devtools/aivdmtable
index c572070..89f9e0d 100755
--- a/devtools/aivdmtable
+++ b/devtools/aivdmtable
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Generate an asciidoc table of the six-bit encoding used in AIVDM packets.
 #
diff --git a/devtools/cycle_analyzer b/devtools/cycle_analyzer
index 53ddd51..6992c10 100755
--- a/devtools/cycle_analyzer
+++ b/devtools/cycle_analyzer
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is Copyright (c) 2010 by the GPSD project
 # BSD terms apply: see the file COPYING in the distribution root for details.
diff --git a/devtools/flocktest b/devtools/flocktest
index d301ee7..8cf2d9e 100755
--- a/devtools/flocktest
+++ b/devtools/flocktest
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is Copyright (c) 2010 by the GPSD project
 # BSD terms apply: see the file COPYING in the distribution root for details.
diff --git a/devtools/identify_failing_build_options.py b/devtools/identify_failing_build_options.py
index ec10a84..480a3fb 100755
--- a/devtools/identify_failing_build_options.py
+++ b/devtools/identify_failing_build_options.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This code runs compatibly under Python 2 and 3.x for x >= 2.
 # Preserve this property!
diff --git a/devtools/regress-builder b/devtools/regress-builder
index 67df358..1ea75fd 100755
--- a/devtools/regress-builder
+++ b/devtools/regress-builder
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is Copyright (c) 2010 by the GPSD project
 # BSD terms apply: see the file COPYING in the distribution root for details.
diff --git a/devtools/regressdiff b/devtools/regressdiff
index fa658c1..0f85a72 100755
--- a/devtools/regressdiff
+++ b/devtools/regressdiff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Walk through a pair of textfiles looking for where they begin to differ.
 # May be useful for comparing logs when regression tests break.
diff --git a/devtools/sizes b/devtools/sizes
index b8afa6e..11e39ab 100755
--- a/devtools/sizes
+++ b/devtools/sizes
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # sizes -- explore the sizes of static gpsd binaries
 #
diff --git a/devtools/striplog b/devtools/striplog
index 5c63dda..70d41ad 100755
--- a/devtools/striplog
+++ b/devtools/striplog
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # striplog -- strip leading lines from logs
 #
diff --git a/devtools/tablegen.py b/devtools/tablegen.py
index b8a3ed3..f42a59e 100755
--- a/devtools/tablegen.py
+++ b/devtools/tablegen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This tool is intended to automate away the drudgery in bring up support
 # for a new AIS message type.  It parses the tabular description of a message
diff --git a/devtools/test_json_validity.py b/devtools/test_json_validity.py
index 959bfe8..cac10da 100755
--- a/devtools/test_json_validity.py
+++ b/devtools/test_json_validity.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Christian Gagneraud - 2012
 # Simple python script that will parse json dictionaries on its input,
diff --git a/devtools/uninstall_cleanup.py b/devtools/uninstall_cleanup.py
index e78d412..916fbab 100755
--- a/devtools/uninstall_cleanup.py
+++ b/devtools/uninstall_cleanup.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # This code runs compatibly under Python 2 and 3.x for x >= 2.
 # Preserve this property!
diff --git a/gegps b/gegps
index 51af633..945e730 100755
--- a/gegps
+++ b/gegps
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 '''
 usage: gegps [-i] [-d kmldir]
diff --git a/gps/gps.py b/gps/gps.py
index 7835c5e..f0ed6f4 100644
--- a/gps/gps.py
+++ b/gps/gps.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 '''gps.py -- Python interface to GPSD.
 
diff --git a/gpsprof b/gpsprof
index 2f0c60e..a8715b3 100755
--- a/gpsprof
+++ b/gpsprof
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 '''
 Collect and plot latency-profiling data from a running gpsd.
diff --git a/jsongen.py b/jsongen.py
index db4b871..ed0fde7 100755
--- a/jsongen.py
+++ b/jsongen.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is Copyright (c) 2010 by the GPSD project
 # SPDX-License-Identifier: BSD-2-clause
diff --git a/leapsecond.py b/leapsecond.py
index aa5a80c..5fc657e 100755
--- a/leapsecond.py
+++ b/leapsecond.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 
 Usage: leapsecond.py [-v] { [-h] | [-f filename] | [-g filename]
diff --git a/maskaudit.py b/maskaudit.py
index d5cdff9..bcdaeed 100755
--- a/maskaudit.py
+++ b/maskaudit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This file is Copyright (c) 2010 by the GPSD project
 # SPDX-License-Identifier: BSD-2-clause
diff --git a/test_maidenhead.py b/test_maidenhead.py
index 665fd49..54804c0 100755
--- a/test_maidenhead.py
+++ b/test_maidenhead.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Test grid locator conversion.
 #
diff --git a/test_misc.py b/test_misc.py
index 111d8e0..d630771 100755
--- a/test_misc.py
+++ b/test_misc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Test gps/misc.py
 #
diff --git a/test_xgps_deps.py b/test_xgps_deps.py
index cb9448e..e1f5301 100755
--- a/test_xgps_deps.py
+++ b/test_xgps_deps.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """Test imports needed by X11-based tools."""
 
 from __future__ import print_function
diff --git a/valgrind-audit.py b/valgrind-audit.py
index 749b2fa..30a1817 100755
--- a/valgrind-audit.py
+++ b/valgrind-audit.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # This is a valgrind torture test for the gpsd daemon.
 # It's not really expected to spot anything as long as we aren't using
diff --git a/xgps b/xgps
index af36225..d231ee5 100755
--- a/xgps
+++ b/xgps
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: UTF-8
 '''
 xgps -- test client for gpsd
diff --git a/xgpsspeed b/xgpsspeed
index baada44..a4b0ad4 100755
--- a/xgpsspeed
+++ b/xgpsspeed
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # by
 # Robin Wittler <real@the-real.org> (speedometer mode)

