diff --git a/gpscat b/gpscat
index 26c9acf..6cf4765 100755
--- a/gpscat
+++ b/gpscat
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 "Display GPS output.  Hexify it if necessary."
 
diff --git a/gpsfake b/gpsfake
index e02d64a..ec10974 100755
--- a/gpsfake
+++ b/gpsfake
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 '''
 gpsfake -- test harness for gpsd
diff --git a/ubxtool b/ubxtool
index 6bdb0ba..82f5b2e 100755
--- a/ubxtool
+++ b/ubxtool
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: UTF-8
 '''
 ubxtool -- u-blox configurator and packet decoder
diff --git a/zerk b/zerk
index 745c765..2e1b73a 100755
--- a/zerk
+++ b/zerk
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: UTF-8
 '''
 zerk -- GREIS configurator and packet decoder
