diff --git a/ppsthread.c b/ppsthread.c
index 6819c6f..45002cb 100644
--- a/ppsthread.c
+++ b/ppsthread.c
@@ -75,12 +75,11 @@
  */
 #define FAKE_PPS_NAME "ktimer"
 
-#if defined(HAVE_SYS_TIMEPPS_H)
+// We supply our own timepps.h for Linux
 // include unistd.h here as it is missing on older pps-tools releases.
 // 'close' is not defined otherwise.
 #include <unistd.h>
-#include <sys/timepps.h>
-#endif
+#include "sys/timepps.h"
 
 #include "timespec.h"
 #include "ppsthread.h"
