diff --git a/src/gui.c b/src/gui.c
index b24d4f3..91b050e 100644
--- a/src/gui.c
+++ b/src/gui.c
@@ -13,9 +13,9 @@ static const char *card_values[13] = {"A", "2", "3",  "4", "5", "6", "7",
                                       "8", "9", "10", "J", "Q", "K"};
 
 static void draw_value(struct card *card) {
-  mvwprintw(card->frame->window, 0, 0, card_values[card->value]);
+  mvwprintw(card->frame->window, 0, 0, "%s", card_values[card->value]);
   mvwprintw(card->frame->window, 4, 7 - strlen(card_values[card->value]),
-            card_values[card->value]);
+            "%s", card_values[card->value]);
 }
 
 static void draw_suit(struct card *card) {
@@ -43,9 +43,9 @@ static void draw_suit(struct card *card) {
     }
   }
   mvwprintw(card->frame->window, 0, strlen(card_values[card->value]),
-            card_suits[card->suit]);
+            "%s", card_suits[card->suit]);
   mvwprintw(card->frame->window, 4, 6 - strlen(card_values[card->value]),
-            card_suits[card->suit]);
+            "%s", card_suits[card->suit]);
   if (card->suit % 2 == 0) {
     wattroff(card->frame->window, COLOR_PAIR(RED_ON_WHITE));
   } else {
