diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 13dd0d8..4db55f9 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -115,7 +115,6 @@ function(add_dbusmock_test_by_name name)
   target_link_libraries (${TEST_NAME} indicatordatetimeservice ${SERVICE_DEPS_LIBRARIES} ${DBUSTEST_LIBRARIES} ${GTEST_LIBRARIES} ${GMOCK_LIBRARIES})
 endfunction()
 add_dbusmock_test_by_name(test-timezone-geoclue)
-add_dbusmock_test_by_name(test-timezones)
 
 set(
   COVERAGE_TEST_TARGETS
