--- a/ext/dtls/gstdtlsagent.c
+++ b/ext/dtls/gstdtlsagent.c
@@ -176,11 +176,11 @@
 
   ERR_clear_error ();
 
-#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
-  priv->ssl_context = SSL_CTX_new (DTLS_method ());
-#else
+//#if OPENSSL_VERSION_NUMBER >= 0x1000200fL
+ // priv->ssl_context = SSL_CTX_new (DTLS_method ());
+//#else
   priv->ssl_context = SSL_CTX_new (DTLSv1_method ());
-#endif
+//#endif
   if (ERR_peek_error () || !priv->ssl_context) {
     char buf[512];
 
