--- Linux-PAM-1.1.6.orig/modules/pam_lastlog/pam_lastlog.c
+++ Linux-PAM-1.1.6/modules/pam_lastlog/pam_lastlog.c
@@ -10,6 +10,7 @@
 
 #include "config.h"
 
+#include <paths.h>
 #include <fcntl.h>
 #include <time.h>
 #include <errno.h>
@@ -48,6 +49,10 @@
 
 #ifndef _PATH_BTMP
 # define _PATH_BTMP "/var/log/btmp"
+#endif
+
+#ifndef __GLIBC__
+#define logwtmp(args...)
 #endif
 
 /* XXX - time before ignoring lock. Is 1 sec enough? */
