diff --git a/examples/show-gadgets.c b/examples/show-gadgets.c
index 707d448..a2a21c8 100644
--- a/examples/show-gadgets.c
+++ b/examples/show-gadgets.c
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
+#include <sys/sysmacros.h>
 #include <netinet/ether.h>
 #include <usbg/usbg.h>
 #include <usbg/function/ms.h>
diff --git a/src/usbg_common.c b/src/usbg_common.c
index f8822fc..6f12a38 100644
--- a/src/usbg_common.c
+++ b/src/usbg_common.c
@@ -15,6 +15,7 @@
 
 #include <errno.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <malloc.h>
 #include <stdio.h>
diff --git a/src/usbg_common_libconfig.c b/src/usbg_common_libconfig.c
index d2576d7..321b2db 100644
--- a/src/usbg_common_libconfig.c
+++ b/src/usbg_common_libconfig.c
@@ -13,6 +13,8 @@
 #include "usbg/usbg_internal.h"
 #include "usbg/usbg_internal_libconfig.h"
 
+#include <sys/sysmacros.h>
+
 int usbg_get_config_node_int(config_setting_t *root,
 			     const char *node_name, void *val)
 {
