these definitions conflict with directfb which also typedefs `u64` `s64`
diff --git a/lib/irrlicht/include/irrTypes.h b/lib/irrlicht/include/irrTypes.h
index be79677..a63c7a4 100644
--- a/lib/irrlicht/include/irrTypes.h
+++ b/lib/irrlicht/include/irrTypes.h
@@ -5,6 +5,7 @@
 #ifndef __IRR_TYPES_H_INCLUDED__
 #define __IRR_TYPES_H_INCLUDED__
 
+#include <stdint.h>
 #include "IrrCompileConfig.h"
 
 namespace irr
@@ -76,7 +77,7 @@ typedef unsigned __int64			u64;
 #if __WORDSIZE == 64
 typedef unsigned long int 			u64;
 #else
-__extension__ typedef unsigned long long	u64;
+typedef uint64_t u64;
 #endif
 #else
 typedef unsigned long long			u64;
@@ -90,7 +91,7 @@ typedef __int64					s64;
 #if __WORDSIZE == 64
 typedef long int 				s64;
 #else
-__extension__ typedef long long			s64;
+typedef int64_t s64;
 #endif
 #else
 typedef long long				s64;
