diff --git a/host/lib/include/uhdlib/transport/link_if.hpp b/host/lib/include/uhdlib/transport/link_if.hpp
index 6aec5e820..eb1fd5ab0 100644
--- a/host/lib/include/uhdlib/transport/link_if.hpp
+++ b/host/lib/include/uhdlib/transport/link_if.hpp
@@ -8,6 +8,7 @@
 
 #include <uhd/transport/adapter_id.hpp>
 #include <uhd/transport/frame_buff.hpp>
+#include <cstdint>
 #include <memory>
 
 namespace uhd { namespace transport {
