diff --git a/src/Makefile b/src/Makefile
index 8840e88..4142453 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -111,7 +111,7 @@ testpactester: pactester $(LIBRARY_LINK)
 docs:
 	../tools/generatedocs.sh
 
-install: all
+install: pactester $(LIBRARY_LINK)
 	install -d $(LIB_PREFIX) $(INC_PREFIX) $(BIN_PREFIX)
 	install -m 644 $(LIBRARY) $(LIB_PREFIX)/$(LIBRARY)
 	ln -sf $(LIBRARY) $(LIB_PREFIX)/$(LIBRARY_LINK)
@@ -139,7 +139,6 @@ dist: all
 # Targets to build python module
 pymod: pacparser.o pacparser.h quickjs/libquickjs.a
 	cd pymod && ARCHFLAGS="" $(PYTHON) setup.py build
-	$(PYTHON) ../tests/runtests.py
 
 pymod-dist: pacparser.o pacparser.h quickjs/libquickjs.a
 	cd pymod && ARCHFLAGS="" $(PYTHON) setup.py build
