diff --git a/src/Makefile b/src/Makefile
index 87f5c1b..cd653a7 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -39,7 +39,7 @@ LIBRARY = $(LIBRARY_NAME).$(SO_SUFFIX).$(LIB_VER)
 MKSHLIB = $(CC) -shared
 LIB_OPTS = -Wl,-soname=$(LIBRARY) -Wl,-exclude-libs=libjs.a
 SHFLAGS = -fPIC
-SMCFLAGS = -DHAVE_VA_COPY -DVA_COPY=__va_copy
+SMCFLAGS = -DHAVE_VA_COPY -DVA_COPY=va_copy
 
 ifeq ($(OS_ARCH),Darwin)
   PREFIX ?= /usr/local
