From 8c803432699a815349c73630e620d5eaa4a16727 Mon Sep 17 00:00:00 2001
From: Felix Janda <felix.janda@posteo.de>
Date: Mon, 19 Jun 2017 21:14:24 -0400
Subject: [PATCH 4/4] audisp/audispd.c: Include <limits.h> for PATH_MAX

---
 audisp/audispd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/audisp/audispd.c b/audisp/audispd.c
index 9831cf3..329b629 100644
--- a/audisp/audispd.c
+++ b/audisp/audispd.c
@@ -34,6 +34,7 @@
 #include <sys/poll.h>
 #include <netdb.h>
 #include <arpa/inet.h>
+#include <limits.h>
 
 #include "audispd-config.h"
 #include "audispd-pconfig.h"
-- 
2.13.1

