--- a/auparse/auparse.h
+++ b/auparse/auparse.h
@@ -51,7 +51,7 @@
 void auparse_destroy(auparse_state_t *au);
 void auparse_destroy_ext(auparse_state_t *au, auparse_destroy_what_t what);
 auparse_state_t *auparse_init(ausource_t source, const void *b)
-	__attribute_malloc__ __attr_dealloc (auparse_destroy, 1);
+	__attribute__((__malloc__)) __attr_dealloc (auparse_destroy, 1);
 int auparse_new_buffer(auparse_state_t *au, const char *data, size_t data_len)
 	__attr_access ((__read_only__, 2, 3));
 int auparse_feed(auparse_state_t *au, const char *data, size_t data_len)
