From: Dermot Bradley <dermot_bradley@yahoo.com>
Date: Fri, 29 May 2020 19:55:23 +0100
Updated: Wed, 23 Jul 2025 04:50:16 +0000

correct paths in plugins

--- a/audisp/audispd-pconfig.c
+++ b/audisp/audispd-pconfig.c
@@ -401,7 +401,7 @@
 	return 1;
 }
 
-static const char *BUILTIN_PATH="/sbin/audisp-af_unix";
+static const char *BUILTIN_PATH="/usr/sbin/audisp-af_unix";
 static int path_parser(struct nv_pair *nv, int line,
 	plugin_conf_t *config)
 {
--- a/audisp/plugins/remote/au-remote.conf
+++ b/audisp/plugins/remote/au-remote.conf
@@ -5,7 +5,7 @@
 
 active = no
 direction = out
-path = /sbin/audisp-remote
+path = /usr/sbin/audisp-remote
 type = always
 #args =
 format = string
--- a/audisp/plugins/syslog/syslog.conf
+++ b/audisp/plugins/syslog/syslog.conf
@@ -8,7 +8,7 @@
 
 active = no
 direction = out
-path = /sbin/audisp-syslog
+path = /usr/sbin/audisp-syslog
 type = always 
 args = LOG_INFO
 format = string
--- a/init.d/augenrules
+++ b/init.d/augenrules
@@ -40,7 +40,7 @@
 
 try_load() {
 	if [ $LoadRules -eq 1 ] ; then
-		/sbin/auditctl -R ${DestinationFile}
+		/usr/sbin/auditctl -R ${DestinationFile}
 		RETVAL=$?
 	fi
 }
