diff --git a/metadata.c b/metadata.c
index 39a65ef..74d5488 100644
--- a/metadata.c
+++ b/metadata.c
@@ -30,9 +30,9 @@
 #include "image_utils.h"
 #include <jpeglib.h>
 #include <setjmp.h>
-#include <avutil.h>
-#include <avcodec.h>
-#include <avformat.h>
+#include <libavutil/avutil.h>
+#include <libavfilter/avcodec.h>
+#include <libavformat/avformat.h>
 #include "tagutils/tagutils.h"
 
 #include "upnpglobalvars.h"
