--- ./minissdp.c.orig	2014-08-23 00:40:46.218655070 +0500
+++ ./minissdp.c	2014-08-23 00:41:17.972176864 +0500
@@ -41,6 +41,7 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "minidlnapath.h"
 #include "upnphttp.h"
--- ./clients.h.orig	2014-08-23 00:39:47.001636613 +0500
+++ ./clients.h	2014-08-23 00:39:56.545026648 +0500
@@ -2,6 +2,7 @@
 #define __CLIENTS_H__
 #include <stdint.h>
 #include <netinet/in.h>
+#include <time.h>
 
 #define CLIENT_CACHE_SLOTS 20
 
