--- a/coderay.gemspec	2020-05-30 03:20:17.000000000 -0400
+++ b/coderay.gemspec	2024-09-17 14:30:39.750244140 -0400
@@ -5,11 +5,7 @@
 Gem::Specification.new do |s|
   s.name = 'coderay'
   
-  if ENV['RELEASE']
-    s.version = CodeRay::VERSION
-  else
-    s.version = "#{CodeRay::VERSION}.rc#{ENV['RC'] || 1}"
-  end
+  s.version = CodeRay::VERSION
   
   s.authors     = ['Kornelius Kalnbach']
   s.email       = ['murphy@rubychan.de']
@@ -24,8 +20,8 @@
   
   readme_file = 'README_INDEX.rdoc'
   
-  s.files         = `git ls-files -- lib/* #{readme_file} MIT-LICENSE`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
+  s.files         = Dir['lib/**/*']
+  s.executables   = s.files.grep(%r{^exe/}) { |f| File.basename(f) }
   s.require_paths = ['lib']
   
   s.rdoc_options      = '-SNw2', "-m#{readme_file}", '-t CodeRay Documentation'
