From 3c5e4dbeaea744fd52515c3daba0e28741cb2091 Mon Sep 17 00:00:00 2001
From: "Jens L." <jens@beryju.org>
Date: Mon, 5 Aug 2024 13:50:13 +0200
Subject: [PATCH] Use raw python string for regex

---
 opencontainers/distribution/reggie/defaults.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/opencontainers/distribution/reggie/defaults.py b/opencontainers/distribution/reggie/defaults.py
index d7aa151..6942e4e 100644
--- a/opencontainers/distribution/reggie/defaults.py
+++ b/opencontainers/distribution/reggie/defaults.py
@@ -14,6 +14,6 @@
     __version__
 )
 URL_REGEX = (
-    "http[s]?:\/\/(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
+    r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+"
 )
 VALID_METHODS = ["HEAD", "GET", "POST", "PATCH", "PUT", "DELETE", "OPTIONS"]
