--- a/nx-X11/lib/font/fontfile/encparse.c
+++ b/nx-X11/lib/font/fontfile/encparse.c
@@ -785,13 +785,13 @@
     return 0;
 }
 
-char*
+const char*
 FontEncDirectory()
 {
-    static char* dir = NULL;
+    static const char* dir = NULL;
 
     if(dir == NULL) {
-        char *c = getenv("FONT_ENCODINGS_DIRECTORY");
+        const char *c = getenv("FONT_ENCODINGS_DIRECTORY");
         if(c) {
             dir = malloc(strlen(c) + 1);
             if(!dir)
@@ -899,7 +899,7 @@
 {
     FontEncPtr encoding;
     char dir[MAXFONTFILENAMELEN], dirname[MAXFONTFILENAMELEN];
-    char *d;
+    const char *d;
 
     if(fontFileName) {
         parseFontFileName(fontFileName, dirname, dir);
