diff --git a/nx-X11/programs/Xserver/os/access.c b/nx-X11/programs/Xserver/os/access.c
index bea4b96..282307e 100644
--- a/nx-X11/programs/Xserver/os/access.c
+++ b/nx-X11/programs/Xserver/os/access.c
@@ -1941,7 +1941,7 @@ siHostnameAddrMatch(int family, void * addr, int len,
 	char hostname[SI_HOSTNAME_MAXLEN];
 	int f, hostaddrlen;
 	void * hostaddr;
-	const char **addrlist;
+	char **addrlist;
 
 	if (siAddrLen >= sizeof(hostname)) 
 	    return FALSE;
