--- nx-libs-3.5.99.27.orig/nx-X11/programs/Xserver/dix/dixfonts.c
+++ nx-libs-3.5.99.27/nx-X11/programs/Xserver/dix/dixfonts.c
@@ -2482,7 +2482,7 @@
     .get_client_resolutions = get_client_resolutions,
     .get_default_point_size = get_default_point_size,
     .get_new_font_client_id = get_new_font_client_id,
-    .get_time_in_millis = GetTimeInMillis,
+    .get_time_in_millis = (uint32_t(*)(void)) GetTimeInMillis,
     .init_fs_handlers = _init_fs_handlers,
     .register_fpe_funcs = register_fpe_funcs,
     .remove_fs_handlers = _remove_fs_handlers,
