diff -up fatresize-1.0.2/configure.ac.orig fatresize-1.0.2/configure.ac
--- fatresize-1.0.2/configure.ac.orig	2005-09-20 10:30:35.000000000 +0200
+++ fatresize-1.0.2/configure.ac	2012-12-17 23:33:06.238862845 +0100
@@ -29,8 +29,8 @@ CFLAGS="$CFLAGS -D_FILE_OFFSET_BITS=${ac
 dnl libparted
 # hack
 LIBS="$LIBS -ldl"
-PARTED_CHECK_LIBPARTED(1.6.0, ,
-	[AC_MSG_ERROR([*** libparted >= 1.6.0 not installed - please install first ***])])
+AC_CHECK_LIB(parted, ped_malloc,  [], AC_MSG_ERROR([*** libparted >= 3.1.0 not installed - please install first ***]))
+AC_CHECK_LIB(parted-fs-resize, ped_file_system_open,  [], AC_MSG_ERROR([*** libparted-fs-resize >= 3.1.0 not installed - please install first ***]))
 
 AC_SUBST(PARTED_LIBS)
 
