diff -up fatresize-1.0.2/fatresize.c.orig fatresize-1.0.2/fatresize.c
--- fatresize-1.0.2/fatresize.c.orig	2005-09-20 10:29:08.000000000 +0200
+++ fatresize-1.0.2/fatresize.c	2012-12-18 00:07:00.000000000 +0100
@@ -217,7 +217,7 @@ fatresize_handler(PedException *ex)
 static int
 snap(PedSector* sector, PedSector new_sector, PedGeometry* range)
 {
-    PED_ASSERT(ped_geometry_test_sector_inside (range, *sector), return 0);
+    PED_ASSERT(ped_geometry_test_sector_inside (range, *sector));
     if (!ped_geometry_test_sector_inside(range, new_sector))
 	return 0;
 
@@ -281,7 +281,7 @@ snap_to_boundaries (PedGeometry* new_geo
 			  end_part->geom.end, end_part->geom.start - 1, -1);
 	}
 
-	PED_ASSERT (start <= end, return);
+	PED_ASSERT (start <= end);
 	ped_geometry_set (new_geom, start, end - start + 1);
 }

