diff --git a/kubernetes/config/kube_config_test.py b/kubernetes/config/kube_config_test.py
index b415492..1c8bbbe 100644
--- a/kubernetes/config/kube_config_test.py
+++ b/kubernetes/config/kube_config_test.py
@@ -1617,3 +1617,3 @@ class TestKubeConfigLoader(BaseTestCase):
         self.assertTrue(callable(actual._config_persister))
-        self.assertEquals(actual._config_persister.__name__, "save_changes")
+        self.assertEqual(actual._config_persister.__name__, "save_changes")
 
