Patch from: Jeff Bilyk <jbilyk@alpinelinux.org>

--- smokeping-2.6.1/lib/Makefile.in
+++ smokeping-2.6.1/lib/Makefile.in.new
@@ -216,7 +216,7 @@
 PROBE := $(wildcard Smokeping/probes/*.pm)
 MATCH := $(wildcard Smokeping/matchers/*.pm)
 EXTRA_DIST = $(PM) $(SP) $(SORT) $(PROBE) $(MATCH)
-perllibdir = $(DESTDIR)$(prefix)/lib
+perllibdir = $(prefix)/lib
 nobase_perllib_DATA = $(EXTRA_DIST)
 all: all-am
 
