diff --git a/etc/config.dist.in b/etc/config.dist.in
index 8b9fdf3..b6f6180 100644
--- a/etc/config.dist.in
+++ b/etc/config.dist.in
@@ -1,19 +1,19 @@
 *** General ***
 
-owner    = Peter Random
-contact  = some@address.nowhere
+owner    = Super User
+contact  = root@localhost
 mailhost = my.mail.host
-sendmail = @SENDMAIL@
+sendmail = /usr/sbin/sendmail
 # NOTE: do not put the Image Cache below cgi-bin
 # since all files under cgi-bin will be executed ... this is not
 # good for images.
-imgcache = @prefix@/cache
-imgurl   = cache
-datadir  = @prefix@/data
-piddir  = @prefix@/var
-cgiurl   = http://some.url/smokeping.cgi
-smokemail = @prefix@/etc/smokemail.dist
-tmail = @prefix@/etc/tmail.dist
+imgcache = /var/lib/smokeping/images
+imgurl   = /smokeping/images
+datadir  = /var/lib/smokeping
+piddir   = /run/smokeping
+cgiurl   = http://localhost/smokeping/sm.cgi
+smokemail = /etc/smokeping/smokemail
+tmail     = /etc/smokeping/tmail
 # specify this to get syslog logging
 syslogfacility = local0
 # each probe is now run in its own process
@@ -21,8 +21,8 @@ syslogfacility = local0
 # concurrentprobes = no
 
 *** Alerts ***
-to = alertee@address.somewhere
-from = smokealert@company.xy
+to = root@localhost
+from = root@localhost
 
 +someloss
 type = loss
@@ -47,7 +47,7 @@ AVERAGE  0.5 144   720
 
 *** Presentation ***
 
-template = @prefix@/etc/basepage.html.dist
+template = /etc/smokeping/basepage.html
 htmltitle = yes
 graphborders = no
 
@@ -110,7 +110,7 @@ unison_tolerance = 2
 binary = /usr/sbin/fping
 
 *** Slaves ***
-secrets=@prefix@/etc/smokeping_secrets.dist
+secrets=/etc/smokeping/smokeping_secrets
 +boomer
 display_name=boomer
 color=0000ff
@@ -121,28 +121,57 @@ color=00ff00
 
 *** Targets ***
 
+#menuextra = <a target='_blank' href='/smokeping/tr.html{HOST}' class='{CLASS}' \
+#               onclick="window.open(this.href,this.target, \
+#               'width=800,height=500,toolbar=no,location=no,status=no,scrollbars=no'); \
+#                return false;">*</a>
+#
 probe = FPing
 
 menu = Top
 title = Network Latency Grapher
-remark = Welcome to the SmokePing website of xxx Company. \
+remark = Welcome to the SmokePing website of <b>Insert Company Name Here</b>. \
          Here you will learn all about the latency of our network.
 
-+ Test
-menu= Targets
-#parents = owner:/Test/James location:/
++ Ping
 
-++ James
+menu = Alpine
+title = Alpine Pings
 
-menu = James
-title =James
-alerts = someloss
-slaves = boomer slave2
-host = james.address
+++ AlpinelinuxOrg
 
-++ MultiHost
+menu = alpinelinux.org
+title = Alpine Linux Webserver
+host = alpinelinux.org
 
-menu = Multihost
-title = James and James as seen from Boomer
-host = /Test/James /Test/James~boomer
+++ DocsAlpinelinuxOrg
+
+menu = docs.alpinelinux.org
+title = Alpine Linux Docs Webserver
+host = docs.alpinelinux.org
+
+++ PkgsAlpinelinuxOrg
+
+menu = pkgs.alpinelinux.org
+title = Alpine Linux Packages
+host = pkgs.alpinelinux.org
+
+#
+# Lots of more options are available, please have a look in the included documentation.
+#
+
+
+#++ James
+
+#menu = James
+#title =James
+#alerts = someloss
+#slaves = boomer slave2
+#host = james.address
+
+#++ MultiHost
+
+#menu = Multihost
+#title = James and James as seen from Boomer
+#host = /Test/James /Test/James~boomer
 
