From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 16 Dec 2021 01:20:00 +0100
Subject: [PATCH] Change paths to follow FHS

--- a/src/constants.hpp
+++ b/src/constants.hpp
@@ -8,44 +8,44 @@
 {
   inline const QString& picon_fp()
   {
-    static const QString picon_fp = normalize_path("../assets/icons/popup/");
+    static const QString picon_fp = normalize_path("../share/nvui/assets/icons/popup/");
   	return picon_fp;
 	}
 
   inline const QString& appicon()
 	{
-		static const QString appicon = normalize_path("../assets/appicon.png");
+		static const QString appicon = normalize_path("../share/nvui/assets/appicon.png");
 		return appicon;
 	}
 
   inline const QString& maxicon()
 	{
-		static const QString maxicon = normalize_path("../assets/max-windows.svg");
+		static const QString maxicon = normalize_path("../share/nvui/assets/max-windows.svg");
 		return maxicon;
 	}
 
   inline const QString& maxicon_second()
 	{
-		static const QString maxicon_second = normalize_path("../assets/max-windows-second.svg");
+		static const QString maxicon_second = normalize_path("../share/nvui/assets/max-windows-second.svg");
 		return maxicon_second;
 	}
 
   inline const QString& minicon()
   {
-		static const QString minicon = normalize_path("../assets/min-windows.svg");
+		static const QString minicon = normalize_path("../share/nvui/assets/min-windows.svg");
 		return minicon;
 	}
 
   inline const QString& closeicon()
 	{
-		static const QString closeicon = normalize_path("../assets/close-windows.svg");
+		static const QString closeicon = normalize_path("../share/nvui/assets/close-windows.svg");
 		return closeicon;
 	}
 	
 	/// Directory for all the Vim files (plugin, doc)
 	inline const QString& script_dir()
 	{
-		static const QString dir = normalize_path("../vim");
+		static const QString dir = normalize_path("../share/nvui/vim");
 		return dir;
 	}
 
