--- libusbx-1.0.17.orig/libusb/io.c
+++ libusbx-1.0.17/libusb/io.c
@@ -38,6 +38,14 @@
 #include "libusbi.h"
 #include "hotplug.h"
 
+#ifndef TIMESPEC_TO_TIMEVAL
+#define TIMESPEC_TO_TIMEVAL(tv, ts)                                     \
+        do {                                                            \
+                (tv)->tv_sec = (ts)->tv_sec;                            \
+                (tv)->tv_usec = (ts)->tv_nsec / 1000;                   \
+        } while (0)
+#endif
+
 /**
  * \page io Synchronous and asynchronous device I/O
  *
