diff --git a/tests/test_oauth.py b/tests/test_oauth.py
index 5885456..1ab3cdb 100644
--- a/tests/test_oauth.py
+++ b/tests/test_oauth.py
@@ -451,4 +451,4 @@ class TestRequest(unittest.TestCase, ReallyEqualMixin):
 
-        self.assertEquals(exp.netloc, res.netloc)
-        self.assertEquals(exp.path, res.path)
+        self.assertEqual(exp.netloc, res.netloc)
+        self.assertEqual(exp.path, res.path)
 
@@ -456,3 +456,3 @@ class TestRequest(unittest.TestCase, ReallyEqualMixin):
         b = parse_qs(res.query)
-        self.assertEquals(a, b)
+        self.assertEqual(a, b)
 
@@ -491,4 +491,4 @@ class TestRequest(unittest.TestCase, ReallyEqualMixin):
 
-        self.assertEquals("http://example.com", host)
-        self.assertEquals(expected_params, params)
+        self.assertEqual("http://example.com", host)
+        self.assertEqual(expected_params, params)
 
