diff --git a/IGC/OCLFE/igd_fcl_mcl/source/LoadBuffer.cpp b/IGC/OCLFE/igd_fcl_mcl/source/LoadBuffer.cpp
index c74bb981e..7104ad8c9 100644
--- a/IGC/OCLFE/igd_fcl_mcl/source/LoadBuffer.cpp
+++ b/IGC/OCLFE/igd_fcl_mcl/source/LoadBuffer.cpp
@@ -19,7 +19,7 @@ using namespace llvm;
 #include <inttypes.h>
 #include <libgen.h>
 #include <link.h>
-#include <linux/limits.h>
+#include <limits.h>
 #include <stddef.h>
 #include <stdint.h>
 #include <unistd.h>
diff --git a/IGC/ZEBinWriter/zebin/source/autogen/ZEInfo.hpp b/IGC/ZEBinWriter/zebin/source/autogen/ZEInfo.hpp
index 051f8daed..948929672 100644
--- a/IGC/ZEBinWriter/zebin/source/autogen/ZEInfo.hpp
+++ b/IGC/ZEBinWriter/zebin/source/autogen/ZEInfo.hpp
@@ -21,6 +21,7 @@ SPDX-License-Identifier: MIT
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace zebin {
 
