From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 03 Jul 2022 01:52:19 +0200
Subject: Skip test case in E02xtrace failing on musl

--- a/Test/E02xtrace.ztst
+++ b/Test/E02xtrace.ztst
@@ -150,7 +150,6 @@
   test_cases=(
       f            # baseline
       foo-bar      # Dash
-      ヌ           # Meta (the UTF-8 representation of this character has an 0x83 byte)
       \$\'ba\\0z\' # Nul, escaped as though by ${(qqqq)}
   )
   for 1 in "$test_cases[@]"; do
@@ -172,10 +171,6 @@
 >	# traced
 >	echo inner
 >}
->$'\M-c\M-\C-C\M-\C-L' () {
->	# traced
->	echo inner
->}
 >$'ba\C-@z' () {
 >	# traced
 >	echo inner
