Fix tests on a system with a user called "user"

--- a/Test/Y01completion.ztst
+++ b/Test/Y01completion.ztst
@@ -73,16 +73,16 @@
 >NO:{file2}
 
   comptesteval $'zstyle -d \'*\' glob'
-  comptesteval '_users () { compadd user1 user2 }'
+  comptesteval '_users () { compadd zsh-test-user1 zsh-test-user2 }'
   comptest $': ~\t\t\t\t\t'
 0:tilde
->line: {: ~user}{}
->line: {: ~user}{}
->NO:{user1}
->NO:{user2}
->line: {: ~user1}{}
->line: {: ~user2}{}
->line: {: ~user1}{}
+>line: {: ~zsh-test-user}{}
+>line: {: ~zsh-test-user}{}
+>NO:{zsh-test-user1}
+>NO:{zsh-test-user2}
+>line: {: ~zsh-test-user1}{}
+>line: {: ~zsh-test-user2}{}
+>line: {: ~zsh-test-user1}{}
 
   comptest $'echo ;:\C-b\C-b\t'
 0:directories and files before separator
