From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 13 Mar 2022 22:44:00 +0100
Subject: [PATCH] Recommend alpine-zsh-config in zsh-newuser-install

--- a/Functions/Newuser/zsh-newuser-install
+++ b/Functions/Newuser/zsh-newuser-install
@@ -959,6 +959,10 @@
      the file by hand, if so desired).
 "
   fi
+  print -r "Or you can install Alpine's system-wide Z Shell configuration, which provides
+sensible defaults for a great user experience right out of the box.
+To do this, press (q) and run this command as root: apk add alpine-zsh-config.
+"
 
   read -k key$longprompt
   print
