From b87125e3eeb3e3446f80566083e5d66e7a667567 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Sat, 4 Feb 2012 21:22:13 +0000
Subject: [PATCH 1/3] proper build

---
 src/makefile |   50 +++++++++++++++++++++++++++++---------------------
 1 files changed, 29 insertions(+), 21 deletions(-)

diff --git a/src/makefile b/src/makefile
index 886237f..10ce82d 100644
--- a/src/makefile
+++ b/src/makefile
@@ -13,10 +13,12 @@ INSTALL_BINARY_DIR	=	/usr/bin/
 # GENERAL
 #
 
+CFLAGS		?=	-Wall -Werror -O3
+CXXFLAGS	?=	-Wall -Werror -O3
+
 INCLUDES	=	-I.
-LIBRARIES	=	-ldl -lm
-CFLAGS		=	$(INCLUDES) -Wall -Werror -O3 -fPIC
-CXXFLAGS	=	$(CFLAGS)
+DYNAMIC_LD_LIBS =	-ldl
+LIBRARIES	=	$(DYNAMIC_LD_LIBS) -lm
 PLUGINS		=	../plugins/amp.so				\
 			../plugins/delay.so				\
 			../plugins/filter.so				\
@@ -25,21 +27,27 @@ PLUGINS		=	../plugins/amp.so				\
 PROGRAMS	=	../bin/analyseplugin				\
 			../bin/applyplugin 				\
 			../bin/listplugins
-CC		=	cc
-CPP		=	c++
+CC		?=	cc
+CXX		?=	c++
+MKDIR_P		?=	mkdir -p
 
 ###############################################################################
 #
 # RULES TO BUILD PLUGINS FROM C OR C++ CODE
 #
 
+LDFLAGS_filter	= -lm
+
 ../plugins/%.so:	plugins/%.c ladspa.h
-	$(CC) $(CFLAGS) -o plugins/$*.o -c plugins/$*.c
-	$(LD) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CC) $(CFLAGS) $(INCLUDES) -fPIC -o plugins/$*.o -c plugins/$*.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ../plugins/$*.so plugins/$*.o -shared $(LDFLAGS_$*)
 
 ../plugins/%.so:	plugins/%.cpp ladspa.h
-	$(CPP) $(CXXFLAGS) -o plugins/$*.o -c plugins/$*.cpp
-	$(CPP) -o ../plugins/$*.so plugins/$*.o -shared
+	$(CXX) $(CXXFLAGS) $(INCLUDES) -fPIC -o plugins/$*.o -c plugins/$*.cpp
+	$(CXX) $(LDFLAGS) -o ../plugins/$*.so plugins/$*.o -shared $(LDFLAGS_$*)
+
+%.o: %.c
+	$(CC) $(CFLAGS) -DPLUGINDIR=\"$(INSTALL_PLUGINS_DIR)\" -c $<
 
 ###############################################################################
 #
@@ -59,12 +67,12 @@ test:	/tmp/test.wav ../snd/noise.wav always
 	@echo Test complete.
 
 install:	targets
-	-mkdirhier $(INSTALL_PLUGINS_DIR)
-	-mkdirhier $(INSTALL_INCLUDE_DIR)
-	-mkdirhier $(INSTALL_BINARY_DIR)
-	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
-	cp ladspa.h $(INSTALL_INCLUDE_DIR)
-	cp ../bin/* $(INSTALL_BINARY_DIR)
+	-$(MKDIR_P) $(DESTDIR)$(INSTALL_PLUGINS_DIR)
+	-$(MKDIR_P) $(DESTDIR)$(INSTALL_INCLUDE_DIR)
+	-$(MKDIR_P) $(DESTDIR)$(INSTALL_BINARY_DIR)
+	cp ../plugins/* $(DESTDIR)$(INSTALL_PLUGINS_DIR)
+	cp ladspa.h $(DESTDIR)$(INSTALL_INCLUDE_DIR)
+	cp ../bin/* $(DESTDIR)$(INSTALL_BINARY_DIR)
 
 /tmp/test.wav:	targets ../snd/noise.wav
 	../bin/listplugins
@@ -90,19 +98,19 @@ targets:	$(PLUGINS) $(PROGRAMS)
 #
 
 ../bin/applyplugin:	applyplugin.o load.o default.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS)				\
 		-o ../bin/applyplugin					\
-		applyplugin.o load.o default.o
+		applyplugin.o load.o default.o $(LIBRARIES)
 
 ../bin/analyseplugin:	analyseplugin.o load.o default.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS)				\
 		-o ../bin/analyseplugin 				\
-		analyseplugin.o load.o default.o
+		analyseplugin.o load.o default.o $(LIBRARIES)
 
 ../bin/listplugins:	listplugins.o search.o
-	$(CC) $(CFLAGS) $(LIBRARIES)					\
+	$(CC) $(CFLAGS) $(INCLUDES) $(LDFLAGS)				\
 		-o ../bin/listplugins	 				\
-		listplugins.o search.o
+		listplugins.o search.o $(LIBRARIES)
 
 ###############################################################################
 #
-- 
1.7.9

