--- ./src/transports/janus_websockets.c.orig	2022-05-30 11:58:21.181331704 +0300
+++ ./src/transports/janus_websockets.c	2022-05-30 11:58:51.031137205 +0300
@@ -412,7 +412,7 @@
 	if(item && item->value) {
 		ip = (char *)item->value;
 		struct in_addr addr;
-		if(inet_net_pton(AF_INET, ip, &addr, sizeof(addr)) > 0)
+		if(inet_pton(AF_INET, ip, &addr) > 0)
 			ipv4_only = 1;
 		char *iface = janus_websockets_get_interface_name(ip);
 		if(iface == NULL) {

