diff --git a/src/daemon/pulseaudio.desktop b/src/daemon/pulseaudio.desktop
new file mode 100644
index 0000000..06cde27
--- /dev/null
+++ b/src/daemon/pulseaudio.desktop
@@ -0,0 +1,11 @@
+[Desktop Entry]
+Version=1.0
+_Name=PulseAudio Sound System
+_Comment=Start the PulseAudio Sound System
+Exec=start-pulseaudio-x11
+Terminal=false
+Type=Application
+Categories=
+GenericName=
+X-GNOME-Autostart-Phase=Initialization
+X-KDE-autostart-phase=1
