--- a/src/tests/meson.build
+++ b/src/tests/meson.build
@@ -111,13 +111,6 @@
     ]
   endif
 
-  if host_machine.system() != 'darwin'
-    default_tests += [
-      [ 'once-test', 'once-test.c',
-        [ check_dep, thread_dep, libpulse_dep, libpulsecommon_dep, libpulsecore_dep, libatomic_ops_dep ] ],
-    ]
-  endif
-
   if alsa_dep.found()
     default_tests += [
       [ 'alsa-mixer-path-test', 'alsa-mixer-path-test.c',
