From cda8f7af382f5c5f1e9a395eb03e2b819770d499 Mon Sep 17 00:00:00 2001
From: Yuji Saeki <44311901+YujiSaeki@users.noreply.github.com>
Date: Mon, 1 Jun 2020 22:32:16 +0200
Subject: [PATCH 1/1] fix wrong math

Signed-off-by: Christian Hesse <mail@eworm.de>
---
 xcur2png.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/xcur2png.c b/xcur2png.c
index 8723a10..f7dd95d 100644
--- a/xcur2png.c
+++ b/xcur2png.c
@@ -586,9 +586,9 @@ int writePngFileFromXcur (const XcursorDim width, const XcursorDim height,
     unsigned int red = (pixels[i]>>16) & 0xff;
     unsigned int green = (pixels[i]>>8) & 0xff;
     unsigned int blue = pixels[i] & 0xff;
-    red = (div (red * 256, alpha).quot) & 0xff;
-    green = (div (green * 256,  alpha).quot) & 0xff;
-    blue = (div (blue * 256, alpha).quot) & 0xff;
+    red = (div (red * 255, alpha).quot) & 0xff;
+    green = (div (green * 255,  alpha).quot) & 0xff;
+    blue = (div (blue * 255, alpha).quot) & 0xff;
     pix[i] = (alpha << 24) + (red << 16) + (green << 8) + blue;
   }
 
