From a392a940bd0c83393fd76eabe08b1743b396b095 Mon Sep 17 00:00:00 2001
From: Gilles Chehade <gilles@poolp.org>
Date: Sat, 13 Feb 2016 14:03:02 +0100
Subject: [PATCH 3/3] no need for openssl in libasr...

---
 openbsd-compat/includes.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/openbsd-compat/includes.h b/openbsd-compat/includes.h
index bc3d46b..5b680d3 100644
--- a/openbsd-compat/includes.h
+++ b/openbsd-compat/includes.h
@@ -70,8 +70,6 @@
 #endif
 /* end of chl*/
 
-#include <openssl/opensslv.h> /* For OPENSSL_VERSION_NUMBER */
-
 #include "defines.h"
 
 #include "openbsd-compat.h"
-- 
2.11.1

