diff -ruN a/src/proto.h b/src/proto.h
--- a/src/proto.h	2019-11-19 15:08:24.000000000 +0000
+++ b/src/proto.h	2025-07-30 19:58:13.700000000 +0000
@@ -45,7 +45,7 @@
 int is_empty_archive(void);
 void symlink_latest(void);
 struct emailsubdir *msg_subdir(int, time_t);
-char *msg_href(struct emailinfo *, struct emailinfo *, int);
+char *msg_href(struct emailinfo *, struct emailinfo *, bool);
 char *msg_relpath(struct emailinfo *, struct emailinfo *);
 char *articlehtmlfilename(struct emailinfo *);
 char *htmlfilename(const char *, struct emailinfo *, const char *);
diff -ruN a/src/setup.h b/src/setup.h
--- a/src/setup.h	2019-11-19 15:08:24.000000000 +0000
+++ b/src/setup.h	2025-07-30 19:49:51.140000000 +0000
@@ -1,7 +1,9 @@
 #ifndef __SETUP_H
 #define __SETUP_H
 
-typedef int bool;
+#if __STDC_VERSION__ < 202311L
+  typedef int bool;
+#endif
 
 struct Config {
     char *label;		/* the real-name of what to set */
