diff --git a/turbo.lua b/turbo.lua
index 543cce4..e03f8ce 100644
--- a/turbo.lua
+++ b/turbo.lua
@@ -88,6 +88,9 @@ assert(pcall(require, "bit") or pcall(require, "bit32"),
     "No bit or compatible library available")
 turbo.platform =        require "turbo.platform"
 turbo.log =             require "turbo.log"
+if _G.__TURBO_USE_LUASOCKET__ == nil then
+    _G.__TURBO_USE_LUASOCKET__ = true
+end
 if not turbo.platform.__LINUX__ then
     if not pcall(require, "socket") then
         turbo.log.error("Could not load LuaSocket. Aborting.")
