--- a/xmake.lua
+++ b/xmake.lua
@@ -2,7 +2,7 @@
 ---@diagnostic disable: undefined-global
 add_rules("mode.debug", "mode.release")
 add_requires("rime")
-add_requires("glib")
+add_requires("glib-2.0")
 
 target("tmux-rime")
 do
@@ -18,5 +18,5 @@ do
     add_files("src/tmux-rime.c")
     add_installfiles("src/*.h", { prefixdir = "include" })
     add_packages("rime")
-    add_packages("glib")
+    add_packages("glib-2.0")
 end
