--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@
 	mkdir -p $$dir && cd $$dir && \
 	$(FETCH) $(MPACK_URL) | $(UNTGZ)
 
-release: mpack-src
+release:
 	rm -f libmpack-lua-$(LMPACK_VERSION).tar.gz
 	tar cvfz libmpack-lua-$(LMPACK_VERSION).tar.gz \
 		--transform 's,^,libmpack-lua-$(LMPACK_VERSION)/,' \
@@ -95,10 +95,10 @@
 		$(DEPS_PREFIX)/lib/luarocks/rocks/busted/2.0.rc12-1/bin/busted test.lua
 
 ifeq ($(USE_SYSTEM_LUA),no)
-$(MPACK): $(LUAROCKS) mpack-src lmpack.c
+$(MPACK): $(LUAROCKS) lmpack.c
 	$(LUAROCKS) make CFLAGS='$(CFLAGS)'
 else
-$(MPACK): mpack-src lmpack.c
+$(MPACK): lmpack.c
 	$(CC) -shared $(CFLAGS) $(INCLUDES) $(LDFLAGS) $^ -o $@ $(LIBS)
 endif
