diff --git a/cli/Cargo.toml b/cli/Cargo.toml
index 9d52bebf..78dc60b1 100644
--- a/cli/Cargo.toml
+++ b/cli/Cargo.toml
@@ -18,7 +18,7 @@ bench = false
 [features]
 default = ["repl", "doc", "format"]
 repl = ["nickel-lang-core/repl"]
-doc = ["nickel-lang-core/doc", "comrak"]
+doc = ["nickel-lang-core/doc", "dep:comrak"]
 format = ["nickel-lang-core/format", "dep:tempfile"]
 metrics = ["dep:metrics", "dep:metrics-util", "nickel-lang-core/metrics"]
 
