diff --git a/src/network_inspectors/packet_capture/packet_capture.h b/src/network_inspectors/packet_capture/packet_capture.h
index 3417f91..78aab9c 100644
--- a/src/network_inspectors/packet_capture/packet_capture.h
+++ b/src/network_inspectors/packet_capture/packet_capture.h
@@ -20,6 +20,7 @@
 #ifndef PACKET_CAPTURE_H
 #define PACKET_CAPTURE_H
 
+#include <cstdint>
 #include <string>
 
 void packet_capture_enable(const std::string&, const int16_t g = -1);
