diff --git a/src/js_norm/js_identifier_ctx.h b/src/js_norm/js_identifier_ctx.h
index b8d5121..6ef1ce2 100644
--- a/src/js_norm/js_identifier_ctx.h
+++ b/src/js_norm/js_identifier_ctx.h
@@ -20,6 +20,7 @@
 #ifndef JS_IDENTIFIER_CTX
 #define JS_IDENTIFIER_CTX
 
+#include <cstdint>
 #include <list>
 #include <string>
 #include <unordered_map>
diff --git a/src/network_inspectors/packet_capture/packet_capture.h b/src/network_inspectors/packet_capture/packet_capture.h
index 3417f91..78aab9c 100644
--- a/src/network_inspectors/packet_capture/packet_capture.h
+++ b/src/network_inspectors/packet_capture/packet_capture.h
@@ -20,6 +20,7 @@
 #ifndef PACKET_CAPTURE_H
 #define PACKET_CAPTURE_H
 
+#include <cstdint>
 #include <string>
 
 void packet_capture_enable(const std::string&, const int16_t g = -1);
diff --git a/src/packet_io/sfdaq_config.h b/src/packet_io/sfdaq_config.h
index 990c457..f4c1042 100644
--- a/src/packet_io/sfdaq_config.h
+++ b/src/packet_io/sfdaq_config.h
@@ -21,6 +21,7 @@
 #ifndef SFDAQ_CONFIG_H
 #define SFDAQ_CONFIG_H
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
diff --git a/src/service_inspectors/wizard/magic.h b/src/service_inspectors/wizard/magic.h
index fa24d0b..a427108 100644
--- a/src/service_inspectors/wizard/magic.h
+++ b/src/service_inspectors/wizard/magic.h
@@ -20,6 +20,7 @@
 #ifndef MAGIC_H
 #define MAGIC_H
 
+#include <cstdint>
 #include <cassert>
 #include <string>
 #include <vector>
