diff --git a/hwy/targets.h b/hwy/targets.h
index 5dba12a..7398f2c 100644
--- a/hwy/targets.h
+++ b/hwy/targets.h
@@ -29,9 +29,7 @@
 #include "hwy/detect_targets.h"
 #include "hwy/highway_export.h"
 
-#if !HWY_ARCH_RVV && !defined(HWY_NO_LIBCXX)
 #include <atomic>
-#endif
 
 namespace hwy {
 
@@ -304,18 +302,10 @@ struct ChosenTarget {
   }
 
  private:
-  // TODO(janwas): remove RVV once <atomic> is available
-#if HWY_ARCH_RVV || defined(HWY_NO_LIBCXX)
-  int64_t LoadMask() const { return mask_; }
-  void StoreMask(int64_t mask) { mask_ = mask; }
-
-  int64_t mask_{1};  // Initialized to 1 so GetIndex() returns 0.
-#else
   int64_t LoadMask() const { return mask_.load(); }
   void StoreMask(int64_t mask) { mask_.store(mask); }
 
   std::atomic<int64_t> mask_{1};  // Initialized to 1 so GetIndex() returns 0.
-#endif  // HWY_ARCH_RVV
 };
 
 // For internal use (e.g. by FunctionCache and DisableTargets).
