--- a/test/test_helper.rb
+++ b/test/test_helper.rb
@@ -130,7 +130,8 @@
     end
 
     TEST_DIR = File.dirname(File.expand_path(__FILE__))
-    LIBGIT2_FIXTURE_DIR = File.expand_path("../../vendor/libgit2/tests/resources", __FILE__)
+    LIBGIT2_FIXTURE_DIR = ENV['LIBGIT2_FIXTURE_DIR'] \
+                          || File.expand_path("../../vendor/libgit2/tests/resources", __FILE__)
   end
 
   class OnlineTestCase < TestCase
