diff --git a/src/test/njs_unit_test.c b/src/test/njs_unit_test.c
index 67c57d1..f1d3634 100644
--- a/src/test/njs_unit_test.c
+++ b/src/test/njs_unit_test.c
@@ -11812,7 +11812,7 @@ static njs_unit_test_t  njs_test[] =
 
     { njs_str("/+/.test('')"),
       njs_str("SyntaxError: "
-              njs_pcre_var("pcre_compile2(\"+\") failed: quantifier does not follow a repeatable item at \"+\" in 1",
+              njs_pcre_var("pcre_compile2(\"+\") failed: quantifier does not follow a repeatable item at \"\" in 1",
                            "pcre_compile(\"+\") failed: nothing to repeat at \"+\" in 1")) },
 
     { njs_str("/^$/.test('')"),
