From 51b70498e2b26fa82161d1ac8007a448af0d5f20 Mon Sep 17 00:00:00 2001
From: Cormac Stephenson <c7s@kasku.net>
Date: Mon, 23 Aug 2021 23:40:33 +0100
Subject: [PATCH 1/2] Update gettext version

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 114a7c0..87ba47f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -51,7 +51,7 @@ AC_ISC_POSIX
 AC_HEADER_STDC
 AM_PROG_LIBTOOL
 
-AM_GNU_GETTEXT_VERSION([0.19.8])
+AM_GNU_GETTEXT_VERSION(0.20)
 AM_GNU_GETTEXT([external])
 
 # check ibus
-- 
2.32.0

