From a1e9c3b022ae893d679ae3102f5b5c63afa31379 Mon Sep 17 00:00:00 2001
From: Cormac Stephenson <c7s@kasku.net>
Date: Sun, 29 Aug 2021 16:08:52 +0100
Subject: [PATCH 2/2] Use gettext module

---
 setup/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup/main.py b/setup/main.py
index cb82f05..7915e27 100644
--- a/setup/main.py
+++ b/setup/main.py
@@ -294,7 +294,7 @@ class Setup ():
         self.__settings.set_value(key, v)
 
 if __name__ == "__main__":
-    locale.bindtextdomain(config.gettext_package, config.localedir)
+    gettext.bindtextdomain(config.gettext_package, config.localedir)
 
     GLib.set_prgname("ibus-setup-hangul")
     GLib.set_application_name(_("IBusHangul Setup"))
-- 
2.32.0

