From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 09 May 2023 02:12:02 +0200
Subject: [PATCH] Improve default .yashrc

Make it easier to follow.

--- a/share/initialization/default
+++ b/share/initialization/default
@@ -12,10 +12,12 @@
 You are seeing this message because a user-specific initialization script is
 not found at ~/.yashrc. The current session has been loaded with common
 settings that should be useful for many users. To suppress this message, copy
-a sample initialization script from ${yashdatadir}/initialization/sample
-to ~/.yashrc and make any customization in it if you want. For bare default
-settings, put an empty file at ~/.yashrc. Without ~/.yashrc, you will see this
-message again next time you start yash.
+a sample init script to ~/.yashrc and make any customization if you want:
+
+  cp ${yashdatadir}/initialization/sample ~/.yashrc
+
+For bare default settings, put an empty file at ~/.yashrc. Without ~/.yashrc,
+you will see this message again next time you start yash.
 
 For more instructions, see the manual at: https://magicant.github.io/yash/doc/
 
