From 3bfe5519b33af389367439756e79bb8fa7380a10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=2E=20Neusch=C3=A4fer?= <j.ne@posteo.net>
Date: Tue, 29 Jul 2025 23:31:55 +0000
Subject: [PATCH] fix build with gcc15

---
 cfgfile.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cfgfile.h b/cfgfile.h
index 38532ad..c86bc9b 100644
--- a/cfgfile.h
+++ b/cfgfile.h
@@ -13,7 +13,7 @@ typedef struct {
     int value;
 } config_enumeration_type;
 
-int read_config();
+int read_config(char *file, int whinge_on_error);
 
 char *config_get_string(const char *directive);
 int config_get_bool(const char *directive);
-- 
2.50.1

