diff -urN u-boot-2019.10.orig/include/config_distro_bootcmd.h u-boot-2019.10/include/config_distro_bootcmd.h
--- u-boot-2019.10.orig/include/config_distro_bootcmd.h	2020-01-08 00:18:39.088623768 +0000
+++ u-boot-2019.10/include/config_distro_bootcmd.h	2020-01-08 00:19:27.963822184 +0000
@@ -328,7 +328,8 @@
 #define BOOTENV_EFI_PXE_ARCH "0x1b"
 #define BOOTENV_EFI_PXE_VCI "PXEClient:Arch:00027:UNDI:003000"
 #elif defined(CONFIG_SANDBOX)
-# error "sandbox EFI support is only supported on ARM and x86"
+#define BOOTENV_EFI_PXE_ARCH "0x0"
+#define BOOTENV_EFI_PXE_VCI "PXEClient:Arch:00000:UNDI:003000"
 #else
 #error Please specify an EFI client identifier
 #endif
