--- a/libzpaq.cpp
+++ b/libzpaq.cpp
@@ -27,6 +27,7 @@
 #include <string>
 #include <vector>
 #include <stdio.h>
+#include <memory>
 
 #ifdef unix
 #ifndef NOJIT
@@ -2923,11 +2924,11 @@
   assert(state==SEG2);
 
   const int BUFSIZE=1<<14;
-  char buf[BUFSIZE];  // input buffer
+  std::unique_ptr<char[]> buf(new char[BUFSIZE]);
   while (n) {
     int nbuf=BUFSIZE;  // bytes read into buf
     if (n>=0 && n<nbuf) nbuf=n;
-    int nr=in->read(buf, nbuf);
+    int nr=in->read(buf.get(), nbuf);
     if (nr<0 || nr>BUFSIZE || nr>nbuf) error("invalid read size");
     if (nr<=0) return false;
     if (n>=0) n-=nr;
