--- a/src/telepathy-logger.c
+++ b/src/telepathy-logger.c
@@ -67,7 +67,8 @@
       struct tm tm;
 
       g_get_current_time (&now);
-      localtime_r (&(now.tv_sec), &tm);
+      time_t tmp_t = now.tv_sec;
+      localtime_r (&tmp_t, &tm);
       strftime (now_str, 32, "%Y-%m-%d %H:%M:%S", &tm);
       tmp = g_strdup_printf ("%s.%06ld: %s",
         now_str, now.tv_usec, message);
--- a/tests/dbus/test-log-manager.c
+++ b/tests/dbus/test-log-manager.c
@@ -70,7 +70,8 @@
       struct tm tm;
 
       g_get_current_time (&now);
-      localtime_r (&(now.tv_sec), &tm);
+      time_t tmp_t = now.tv_sec;
+      localtime_r (&tmp_t, &tm);
       strftime (now_str, 32, "%Y-%m-%d %H:%M:%S", &tm);
       tmp = g_strdup_printf ("%s.%06ld: %s",
           now_str, now.tv_usec, message);
--- a/tests/dbus/test-tpl-log-store-pidgin.c
+++ b/tests/dbus/test-tpl-log-store-pidgin.c
@@ -76,7 +76,8 @@
       struct tm tm;
 
       g_get_current_time (&now);
-      localtime_r (&(now.tv_sec), &tm);
+      time_t tmp_t = now.tv_sec;
+      localtime_r (&tmp_t, &tm);
       strftime (now_str, 32, "%Y-%m-%d %H:%M:%S", &tm);
       tmp = g_strdup_printf ("%s.%06ld: %s",
         now_str, now.tv_usec, message);
