--- homer-api-5.0.6/scripts/homer_partremove
+++ homer-api-5.0.6.alpine/scripts/homer_partremove
@@ -5,7 +5,7 @@
 
 bin_dir=`dirname $0`
 programm="$bin_dir/homer_mysql_remove_partitions.pl"
-config="$bin_dir/rotation.ini"
+config=/etc/homer/rotation.ini
 
 $programm $config
 
--- homer-api-5.0.6/scripts/homer_rotate
+++ homer-api-5.0.6.alpine/scripts/homer_rotate
@@ -5,7 +5,7 @@
 
 bin_dir=`dirname $0`
 programm="$bin_dir/homer_mysql_rotate.pl"
-config="$bin_dir/rotation.ini"
+config=/etc/homer/rotation.ini
 
 if [ $# -ne 0 ]
   then
