none of these are required/imported
--
diff --git a/setup.py b/setup.py
index d1c0022..5d33d6f 100644
--- a/setup.py
+++ b/setup.py
@@ -5,13 +5,8 @@ from setuptools import find_packages
 version = "1.0.1"
 
 install_requires = [
-    'acme>=0.31.0',
     'certbot>=0.31.0',
     'dns-lexicon>=3.4.1',
-    'dnspython',
-    'mock',
-    'setuptools',
-    'requests'
 ]
 
 # read the contents of the README file
