--- a/binding.gyp
+++ b/binding.gyp
@@ -11,13 +11,13 @@
       'cflags!': [ '-fno-exceptions' ],
       'cflags_cc!': [ '-fno-exceptions' ],
       'include_dirs': [
-        "libpg_query/include",
+        "/usr/include/node",
         "<!@(node -p \"require('node-addon-api').include\")"
       ],
       'dependencies': ["<!(node -p \"require('node-addon-api').gyp\")"],
       'conditions': [
         ['OS=="linux"', {
-          "libraries": [ "-L<!(pwd)/libpg_query/linux", "-lpg_query" ]
+          "libraries": [ "-lpg_query" ]
         }],
         ['OS=="mac"', {
           "libraries": [ "-L<!(pwd)/libpg_query/osx", "-lpg_query" ],
