--- a/configure
+++ b/configure
@@ -6,18 +6,8 @@
 OPT=yes
 DBG=yes
 X11=yes
-VER=`git describe --tags 2>/dev/null`
+VER="@PKGVERSION@"

-if [ -z "$VER" ]; then
-	VER=`git rev-parse --short HEAD 2>/dev/null`
-	if [ -z "$VER" ]; then
-		VER=v`pwd | grep 'libspnav-[0-9]\+\.' | sed 's/.*libspnav-\(\([0-9]\+\.\)\+[0-9]\+\).*$/\1/'`
-		if [ $VER = v ]; then
-			VER='<unknown version>'
-		fi
-	fi
-fi
-
 echo "configuring libspnav - $VER"

 srcdir="`dirname "$0"`"
