--- a/mysys/posix_timers.c
+++ b/mysys/posix_timers.c
@@ -188,10 +188,20 @@
           On error, -1 is returned, and errno is set to indicate the error.
 */

+struct linux_sigevent {
+    union sigval sigev_value;
+    int sigev_signo;
+    int sigev_notify;
+    union {
+       int _pad[64-sizeof(int) * 2 + sizeof(union sigval)];
+       int _tid;
+    } _sigev_un;
+};
+
 int
 my_timer_create(my_timer_t *timer)
 {
-  struct sigevent sigev;
+  struct linux_sigevent sigev;

   memset(&sigev, 0, sizeof(sigev));

@@ -200,7 +210,7 @@
   sigev.sigev_notify= SIGEV_SIGNAL | SIGEV_THREAD_ID;
   sigev.sigev_notify_thread_id= timer_notify_thread_id;

-  return timer_create(CLOCK_MONOTONIC, &sigev, &timer->id);
+  return timer_create(CLOCK_MONOTONIC, (struct sigevent *)&sigev, &timer->id);
 }
 #elif defined(HAVE_SIGEV_PORT)
 /**
@@ -311,7 +321,7 @@
 int
 my_timer_create(my_timer_t *timer)
 {
-  struct sigevent sigev;
+  struct linux_sigevent sigev;
   port_notify_t port_notify;

   port_notify.portnfy_port= port_id;
@@ -321,7 +331,7 @@
   sigev.sigev_value.sival_ptr= &port_notify;
   sigev.sigev_notify= SIGEV_PORT;

-  return timer_create(CLOCK_REALTIME, &sigev, &timer->id);
+  return timer_create(CLOCK_MONOTONIC, (struct sigevent *)&sigev, &timer->id);
 }
 #endif
