diff --git a/utilities/installmgr.cpp b/utilities/installmgr.cpp
index b705c25..132314a 100644
--- a/utilities/installmgr.cpp
+++ b/utilities/installmgr.cpp
@@ -73,6 +73,7 @@ virtual bool isUserDisclaimerConfirmed() const {
 		cout << "If you understand this and are willing to enable remote source features\n";
 		cout << "then type yes at the prompt\n\n";
 		cout << "enable? [no] ";
+		fflush(stdout);
 
 		char prompt[10];
 		fgets(prompt, 9, stdin);
