--- a/src/include/defaults.mk
+++ b/src/include/defaults.mk
@@ -72,15 +72,6 @@ override SOFLAGS = $(_SOFLAGS) \
 		   $(call family,SOFLAGS)
 
 HOST_ARCH=$(shell uname -m)
-ifneq ($(HOST_ARCH),ia64)
-ifneq ($(HOST_ARCH),riscv64)
-	HOST_MARCH=-march=native
-else
-	HOST_MARCH=
-endif
-else
-	HOST_MARCH=
-endif
 HOST_CPPFLAGS ?= $(CPPFLAGS)
 override _HOST_CPPFLAGS := $(HOST_CPPFLAGS)
 override HOST_CPPFLAGS = $(_HOST_CPPFLAGS) \
