diff --git a/src/include/defaults.mk b/src/include/defaults.mk
index 4da0cde..ccfadf2 100644
--- a/src/include/defaults.mk
+++ b/src/include/defaults.mk
@@ -73,6 +73,7 @@ override SOFLAGS = $(_SOFLAGS) \
 
 HOST_ARCH=$(shell uname -m)
 ifneq ($(HOST_ARCH),ia64)
+ifneq ($(HOST_ARCH),ppc64le)
 ifneq ($(HOST_ARCH),riscv64)
 	HOST_MARCH=-march=native
 else
@@ -81,6 +82,9 @@ endif
 else
 	HOST_MARCH=
 endif
+else
+	HOST_MARCH=
+endif
 HOST_CPPFLAGS ?= $(CPPFLAGS)
 override _HOST_CPPFLAGS := $(HOST_CPPFLAGS)
 override HOST_CPPFLAGS = $(_HOST_CPPFLAGS) \
