--- a/config.toml
+++ b/config.toml
@@ -1,7 +1,9 @@
-[als.iio]
-path = "/sys/bus/iio/devices"
-thresholds = { 0 = "night", 20 = "dark", 80 = "dim", 250 = "normal", 500 = "bright", 800 = "outdoors" }
+## Ambient Light Sensor
 
+# [als.iio]
+# path = "/sys/bus/iio/devices"
+# thresholds = { 0 = "night", 20 = "dark", 80 = "dim", 250 = "normal", 500 = "bright", 800 = "outdoors" }
+
 # [als.webcam]
 # video = 0
 # thresholds = { 0 = "night", 15 = "dark", 30 = "dim", 45 = "normal", 60 = "bright", 75 = "outdoors" }
@@ -9,20 +11,34 @@
 # [als.time]
 # thresholds = { 0 = "night", 7 = "dark", 9 = "dim", 11 = "normal", 13 = "bright", 16 = "normal", 18 = "dark", 20 = "night" }
 
-# [als.none]
+[als.none]
 
+
+## Display Backlight
+
 [[output.backlight]]
 name = "eDP-1"
-path = "/sys/class/backlight/intel_backlight"
-capturer = "wlroots"
+path = "/sys/class/backlight/intel_backlight"  # Intel GPU
+#path = "/sys/class/backlight/amdgpu_bl0"  # AMD GPU
+# Change to "wlroots" if you use wlroots-based compositor and want to adjust
+# brightness based on the screen contents.
+capturer = "none"
 
+# [[output.backlight]]
+# name = "eDP-1"
+# path = "/sys/class/backlight/amdgpu_bl0"  # AMD GPU
+# capturer = "wlroots"
+
 # [[output.ddcutil]]
 # name = "Dell Inc. DELL P2415Q"
 # capturer = "none"
 
-[[keyboard]]
-name = "keyboard-dell"
-path = "/sys/bus/platform/devices/dell-laptop/leds/dell::kbd_backlight"
+
+## Keyboard Backlight
+
+# [[keyboard]]
+# name = "keyboard-dell"
+# path = "/sys/bus/platform/devices/dell-laptop/leds/dell::kbd_backlight"
 
 # [[keyboard]]
 # name = "keyboard-thinkpad"
