diff --git a/src/mongocxx/CMakeLists.txt b/src/mongocxx/CMakeLists.txt
index 7106040..e3456ce 100644
--- a/src/mongocxx/CMakeLists.txt
+++ b/src/mongocxx/CMakeLists.txt
@@ -208,7 +208,7 @@ else()
       mongocxx_add_library(mongocxx_mocked "mongocxx-mocked" STATIC)
     endif()
 endif()
-if(ENABLE_TESTS)
+if(ENABLE_TESTING)
   target_link_libraries(mongocxx_mocked PUBLIC bsoncxx_testing)
   target_compile_definitions(mongocxx_mocked PUBLIC MONGOCXX_TESTING)
   if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
@@ -249,7 +249,7 @@ if(MONGOCXX_BUILD_STATIC)
 endif()
 mongocxx_install("${mongocxx_target_list}" "${mongocxx_pkg_dep}")
 
-if(ENABLE_TESTS)
+if(false)
   add_subdirectory(test)
 endif()
 
