diff --git a/reactivex/internal/concurrency.py b/reactivex/internal/concurrency.py
index 12db952..ef4ee33 100644
--- a/reactivex/internal/concurrency.py
+++ b/reactivex/internal/concurrency.py
@@ -1,7 +1,5 @@
 from threading import RLock, Thread
-from typing import Any, Callable, TypeVar
-
-from typing_extensions import ParamSpec
+from typing import Any, Callable, ParamSpec, TypeVar
 
 from reactivex.typing import StartableTarget
 
diff --git a/reactivex/internal/utils.py b/reactivex/internal/utils.py
index 109717d..f22761c 100644
--- a/reactivex/internal/utils.py
+++ b/reactivex/internal/utils.py
@@ -1,8 +1,6 @@
 from functools import update_wrapper
 from types import FunctionType
-from typing import TYPE_CHECKING, Any, Callable, Iterable, Optional, TypeVar, cast
-
-from typing_extensions import ParamSpec
+from typing import TYPE_CHECKING, Any, Callable, Iterable, Optional, ParamSpec, TypeVar, cast
 
 from reactivex import abc
 from reactivex.disposable import CompositeDisposable
