--- cpufreqd-2.4.2-origin/src/sock_utils.c
+++ cpufreqd-2.4.2/src/sock_utils.c
@@ -108,7 +108,7 @@
 	if ((fd = socket(PF_UNIX, SOCK_STREAM, 0)) == -1) {
 		clog(LOG_ERR, "socket(): %s.\n", strerror(errno));
 
-	} else if (bind(fd, &sa, sizeof(sa)) == -1) {
+	} else if (bind(fd, (struct sockaddr *)&sa, sizeof(sa)) == -1) {
 		clog(LOG_ERR, "bind(): %s.\n", strerror(errno));
 		close(fd);
 		fd = -1;
--- cpufreqd-2.4.2-origin/utils/getspeed.c
+++ cpufreqd-2.4.2/utils/getspeed.c
@@ -4,6 +4,7 @@
 #include <poll.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
--- cpufreqd-2.4.2-origin/utils/setspeed.c
+++ cpufreqd-2.4.2/utils/setspeed.c
@@ -3,6 +3,7 @@
 #include <fcntl.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/stat.h>
