diff -Nru squid-3.1.9.orig/helpers/basic_auth/MSNT/confload.c squid-3.1.9/helpers/basic_auth/MSNT/confload.c
--- squid-3.1.9.orig/helpers/basic_auth/MSNT/confload.c	2010-10-25 13:34:23.000000000 +0200
+++ squid-3.1.9/helpers/basic_auth/MSNT/confload.c	2010-11-27 13:43:30.000000000 +0100
@@ -27,7 +27,7 @@
 
 /* Path to configuration file */
 #ifndef SYSCONFDIR
-#define SYSCONFDIR "/usr/local/squid/etc"
+#define SYSCONFDIR "/etc/squid"
 #endif
 #define CONFIGFILE   SYSCONFDIR "/msntauth.conf"
 
diff -Nru squid-3.1.9.orig/helpers/basic_auth/MSNT/msntauth.conf.default squid-3.1.9/helpers/basic_auth/MSNT/msntauth.conf.default
--- squid-3.1.9.orig/helpers/basic_auth/MSNT/msntauth.conf.default	2010-10-25 13:34:24.000000000 +0200
+++ squid-3.1.9/helpers/basic_auth/MSNT/msntauth.conf.default	2010-11-27 13:43:30.000000000 +0100
@@ -8,6 +8,6 @@
 server other_PDC	other_BDC	otherdomain
 
 # Denied and allowed users. Comment these if not needed.
-#denyusers	/usr/local/squid/etc/msntauth.denyusers
-#allowusers	/usr/local/squid/etc/msntauth.allowusers
+#denyusers	/etc/squid/msntauth.denyusers
+#allowusers	/etc/squid/msntauth.allowusers
 
diff -Nru squid-3.1.9.orig/helpers/basic_auth/SMB/smb_auth.sh squid-3.1.9/helpers/basic_auth/SMB/smb_auth.sh
--- squid-3.1.9.orig/helpers/basic_auth/SMB/smb_auth.sh	2010-10-25 13:34:25.000000000 +0200
+++ squid-3.1.9/helpers/basic_auth/SMB/smb_auth.sh	2010-11-27 13:43:30.000000000 +0100
@@ -24,7 +24,7 @@
 read AUTHSHARE
 read AUTHFILE
 read SMBUSER
-read SMBPASS
+read -r SMBPASS
 
 # Find domain controller
 echo "Domain name: $DOMAINNAME"
@@ -47,7 +47,7 @@
   addropt=""
 fi
 echo "Query address options: $addropt"
-dcip=`nmblookup $addropt "$PASSTHROUGH#1c" | awk '/^[0-9.]+ / { print $1 ; exit }'`
+dcip=`nmblookup $addropt "$PASSTHROUGH#1c" | awk '/^[0-9.]+\..+ / { print $1 ; exit }'`
 echo "Domain controller IP address: $dcip"
 [ -n "$dcip" ] || exit 1

diff -Nru squid-3.1.9.orig/helpers/external_acl/session/squid_session.8 squid-3.1.9/helpers/external_acl/session/squid_session.8
--- squid-3.1.9.orig/helpers/external_acl/session/squid_session.8	2010-10-25 13:34:23.000000000 +0200
+++ squid-3.1.9/helpers/external_acl/session/squid_session.8	2010-11-27 13:43:30.000000000 +0100
@@ -35,7 +35,7 @@
 .P
 Configuration example using the default automatic mode
 .IP
-external_acl_type session ttl=300 negative_ttl=0 children=1 concurrency=200 %LOGIN /usr/local/squid/libexec/squid_session
+external_acl_type session ttl=300 negative_ttl=0 children=1 concurrency=200 %LOGIN /usr/lib/squid/squid_session
 .IP
 acl session external session
 .IP
diff -Nru squid-3.1.9.orig/helpers/external_acl/unix_group/squid_unix_group.8 squid-3.1.9/helpers/external_acl/unix_group/squid_unix_group.8
--- squid-3.1.9.orig/helpers/external_acl/unix_group/squid_unix_group.8	2010-10-25 13:34:25.000000000 +0200
+++ squid-3.1.9/helpers/external_acl/unix_group/squid_unix_group.8	2010-11-27 13:43:30.000000000 +0100
@@ -27,7 +27,7 @@
 This squid.conf example defines two Squid acls. usergroup1 matches users in group1, and usergroup2
 matches users in group2 or group3
 .IP
-external_acl_type unix_group %LOGIN /usr/local/squid/libexec/squid_unix_group -p
+external_acl_type unix_group %LOGIN /usr/lib/squid/squid_unix_group -p
 .IP
 acl usergroup1 external unix_group group1
 .IP
diff -Nru squid-3.1.9.orig/src/cf.data.pre squid-3.1.9/src/cf.data.pre
--- squid-3.1.9.orig/src/cf.data.pre	2010-10-25 13:34:25.000000000 +0200
+++ squid-3.1.9/src/cf.data.pre	2010-11-27 13:43:30.000000000 +0100
@@ -716,6 +716,7 @@
 acl Safe_ports port 488		# gss-http
 acl Safe_ports port 591		# filemaker
 acl Safe_ports port 777		# multiling http
+acl Safe_ports port 901		# SWAT
 acl CONNECT method CONNECT
 NOCOMMENT_END
 DOC_END
@@ -862,6 +863,9 @@
 http_access allow localnet
 http_access allow localhost
 
+# Allow the localhost to have access by default
+http_access allow localhost
+
 # And finally deny all other access to this proxy
 http_access deny all
 NOCOMMENT_END
@@ -4027,11 +4031,11 @@
 
 NAME: cache_mgr
 TYPE: string
-DEFAULT: webmaster
+DEFAULT: root
 LOC: Config.adminEmail
 DOC_START
 	Email-address of local cache manager who will receive
-	mail if the cache dies.  The default is "webmaster."
+	mail if the cache dies.  The default is "root".
 DOC_END
 
 NAME: mail_from
@@ -6345,7 +6349,7 @@
 NAME: forwarded_for
 COMMENT: on|off|transparent|truncate|delete
 TYPE: string
-DEFAULT: on
+DEFAULT: delete
 LOC: opt_forwarded_for
 DOC_START
 	If set to "on", Squid will append your client's IP address
diff -Nru squid-3.1.9.orig/src/debug.cc squid-3.1.9/src/debug.cc
--- squid-3.1.9.orig/src/debug.cc	2010-10-25 13:34:24.000000000 +0200
+++ squid-3.1.9/src/debug.cc	2010-11-27 13:43:30.000000000 +0100
@@ -452,7 +452,7 @@
 #if HAVE_SYSLOG && defined(LOG_LOCAL4)
 
     if (Debug::log_syslog)
-        openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY | LOG_CONS, syslog_facility);
+        openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY, syslog_facility);
 
 #endif /* HAVE_SYSLOG */
 
diff -Nru squid-3.1.9.orig/src/main.cc squid-3.1.9/src/main.cc
--- squid-3.1.9.orig/src/main.cc	2010-10-25 13:34:25.000000000 +0200
+++ squid-3.1.9/src/main.cc	2010-11-27 13:43:30.000000000 +0100
@@ -1556,7 +1556,7 @@
     if (*(argv[0]) == '(')
         return;
 
-    openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY | LOG_CONS, LOG_LOCAL4);
+    openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY, LOG_DAEMON);
 
     if ((pid = fork()) < 0)
         syslog(LOG_ALERT, "fork failed: %s", xstrerror());
@@ -1600,7 +1600,7 @@
 
         if ((pid = fork()) == 0) {
             /* child */
-            openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY | LOG_CONS, LOG_LOCAL4);
+            openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY, LOG_DAEMON);
             prog = xstrdup(argv[0]);
             argv[0] = xstrdup("(squid)");
             execvp(prog, argv);
@@ -1608,7 +1608,7 @@
         }
 
         /* parent */
-        openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY | LOG_CONS, LOG_LOCAL4);
+        openlog(APP_SHORTNAME, LOG_PID | LOG_NDELAY, LOG_DAEMON);
 
         syslog(LOG_NOTICE, "Squid Parent: child process %d started", pid);
 
