diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2026-02-03 16:31:07.792847693 -0500
+++ b/configure.ac	2026-02-03 16:34:18.327557160 -0500
@@ -468,7 +468,7 @@
 if test "x$enable_gc" != "xno" ; then
   if test "x$with_vcpkg" == "xno"; then
     AC_DEFINE(USEGC,1,[GC Enabled])
-    GCLIB="\$(GC)/.libs/libgc.a"
+    GCLIB="-lgc"
     INCL=$INCL" -I\$(GC)/include"
     AC_MSG_NOTICE([$GCNAME is enabled])
   fi
