From 32e0d045b74a20b0e0a5ad0081413b55c565a3ba Mon Sep 17 00:00:00 2001
From: Michael Richardson <mcr@sandelman.ca>
Date: Fri, 18 Jan 2019 20:59:39 -0500
Subject: [PATCH 4/9] wo#8100 . removed unneeded asm/types

---
 programs/pluto/server.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/programs/pluto/server.c b/programs/pluto/server.c
index b43fefd5c..f1e88ba45 100644
--- a/programs/pluto/server.c
+++ b/programs/pluto/server.c
@@ -47,7 +47,6 @@
 #include <resolv.h>
 
 #if defined(IP_RECVERR) && defined(MSG_ERRQUEUE)
-#  include <asm/types.h>	/* for __u8, __u32 */
 #  include <linux/errqueue.h>
 #  include <sys/uio.h>	/* struct iovec */
 #endif
-- 
2.11.0

