diff --git a/ext/armips/Util/Util.h b/ext/armips/Util/Util.h
index d4916764..af77ed00 100644
--- a/ext/armips/Util/Util.h
+++ b/ext/armips/Util/Util.h
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
--- a/Core/HLE/sceKernelTime.h
+++ b/Core/HLE/sceKernelTime.h
@@ -17,6 +17,8 @@
 
 #pragma once
 
+#include <cstdint>
+
 u32 sceKernelLibcGettimeofday(u32 timeAddr, u32 tzAddr);
 u32 sceKernelLibcTime(u32 outPtr);
 int sceKernelUSec2SysClock(u32 microsec, u32 clockPtr);
--- a/Core/HLE/sceRtc.h
+++ b/Core/HLE/sceRtc.h
@@ -19,6 +19,7 @@
 
 #include "Common/CommonTypes.h"
 #include "Common/Swap.h"
+#include <cstdint>
 
 class PointerWrap;
 
