--- ./configure.ac.orig
+++ ./configure.ac
@@ -419,7 +419,12 @@
 if test x"$EXTRA" = "xyes"
 then
 	AC_CHECK_FUNC( iconv, CHARSET=1 )
-	AC_CHECK_LIB( iconv, iconv_open, LIBICONV=-liconv ; CHARSET=1 )
+	dnl On some systems iconv is in a separate library, and may actually
+	dnl be named libiconv.
+	AC_CHECK_LIB( iconv, libiconv_open, LIBICONV=-liconv ; CHARSET=1 )
+	if test $ac_cv_lib_iconv_libiconv = no; then
+		AC_CHECK_LIB( iconv, iconv_open, LIBICONV=-liconv ; CHARSET=1 )
+	fi
 	if test "x$CHARSET" = "x1"
 	then
 		AC_MSG_CHECKING([whether iconv expects a const second argument])
