seems to be broken with new openssl disabling 1.0
diff --git a/src/testcurl/https/test_tls_options.c b/src/testcurl/https/test_tls_options.c
index d5aa831..658f258 100644
--- a/src/testcurl/https/test_tls_options.c
+++ b/src/testcurl/https/test_tls_options.c
@@ -136,20 +136,6 @@ main (int argc, char *const *argv)
     aes256_sha = "rsa_aes_256_sha";
   }
 
-  if (0 !=
-      test_wrap ("TLS1.0-AES-SHA1",
-                 &test_https_transfer, NULL, port, daemon_flags,
-                 aes128_sha,
-                 CURL_SSLVERSION_TLSv1,
-                 MHD_OPTION_HTTPS_MEM_KEY, srv_key_pem,
-                 MHD_OPTION_HTTPS_MEM_CERT, srv_self_signed_cert_pem,
-                 MHD_OPTION_HTTPS_PRIORITIES,
-                 "NONE:+VERS-TLS1.0:+AES-128-CBC:+SHA1:+RSA:+COMP-NULL",
-                 MHD_OPTION_END))
-  {
-    fprintf (stderr, "TLS1.0-AES-SHA1 test failed\n");
-    errorCount++;
-  }
   fprintf (stderr,
            "The following handshake should fail (and print an error message)...\n");
   if (0 !=
