diff --git a/pyproject.toml b/pyproject.toml
index fb8fae3..6f7fb12 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -75,42 +75,6 @@ raw-options = { local_scheme = "no-local-version" }
 [tool.hatch.build.targets.wheel]
 packages = ["src/attr", "src/attrs"]
 
-[tool.hatch.metadata.hooks.fancy-pypi-readme]
-content-type = "text/markdown"
-
-# PyPI doesn't support the <picture> tag.
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-text = """<p align="center">
-  <a href="https://www.attrs.org/">
-    <img src="https://raw.githubusercontent.com/python-attrs/attrs/main/docs/_static/attrs_logo.svg" width="35%" alt="attrs" />
-  </a>
-</p>
-"""
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "README.md"
-start-after = "<!-- teaser-begin -->"
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-text = """
-
-## Release Information
-
-"""
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-path = "CHANGELOG.md"
-pattern = "\n(###.+?\n)## "
-
-[[tool.hatch.metadata.hooks.fancy-pypi-readme.fragments]]
-text = """
-
----
-
-[Full changelog](https://www.attrs.org/en/stable/changelog.html)
-"""
-
-
 # Make coverage play nicely with pytest-xdist.
 [tool.hatch.build.targets.wheel.hooks.autorun]
 dependencies = ["hatch-autorun"]
