--- a/auto/modules/php
+++ b/auto/modules/php
@@ -102,10 +102,10 @@
                          `${NXT_PHP_CONFIG} --libs`"
 
     else
-        if [ $NXT_PHP_MAJOR_VERSION -ge 8 ]; then
+        if [ $NXT_PHP_MAJOR_VERSION -ge 8 ] && [ $NXT_PHP_MINOR_VERSION -ge 4 ]; then
             NXT_PHP_LIB="-lphp"
         else
-            NXT_PHP_LIB="-lphp${NXT_PHP_VERSION%%.*}"
+            NXT_PHP_LIB="-lphp${NXT_PHP_MAJOR_VERSION}${NXT_PHP_MINOR_VERSION}"
         fi
 
         if [ "$NXT_PHP_LIB_PATH" != "" ]; then
