From 1886d3d0573388aa5c13af99f714687ba6a66ec6 Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Wed, 01 Jun 2011 05:22:14 +0000
Subject: Replace deprecated g_atomic_int_exchange_and_add()

---
(limited to 'addressbook/libedata-book/e-data-book.c')

diff --git a/addressbook/libedata-book/e-data-book.c b/addressbook/libedata-book/e-data-book.c
index 9e17703..1a29e4b 100644
--- a/addressbook/libedata-book/e-data-book.c
+++ b/addressbook/libedata-book/e-data-book.c
@@ -117,7 +117,7 @@ construct_bookview_path (void)
 
 	return g_strdup_printf ("/org/gnome/evolution/dataserver/AddressBookView/%d/%d",
 				getpid (),
-				g_atomic_int_exchange_and_add ((int*)&counter, 1));
+				g_atomic_int_add ((int*)&counter, 1));
 }
 
 static void
--
cgit v0.9.0.2
--- ./addressbook/libedata-book/e-data-book-factory.c.orig
+++ ./addressbook/libedata-book/e-data-book-factory.c
@@ -229,7 +229,7 @@
 
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/addressbook/%d/%u",
-		getpid (), g_atomic_int_exchange_and_add (&counter, 1));
+		getpid (), g_atomic_int_add (&counter, 1));
 }
 
 static gboolean
--- ./calendar/libedata-cal/e-data-cal-factory.c.orig
+++ ./calendar/libedata-cal/e-data-cal-factory.c
@@ -245,7 +245,7 @@
 
 	return g_strdup_printf (
 		"/org/gnome/evolution/dataserver/calendar/%d/%u",
-		getpid (), g_atomic_int_exchange_and_add (&counter, 1));
+		getpid (), g_atomic_int_add (&counter, 1));
 }
 
 static gboolean
