From f1c9292caa3b26c890d628e99677dd05d62eba99 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Tue, 06 Sep 2011 14:43:04 +0000
Subject: Bug #655837 - Avoid use of deprecated g_unicode_canonical_decomposition

---
(limited to 'libedataserver/e-data-server-util.c')

diff --git a/libedataserver/e-data-server-util.c b/libedataserver/e-data-server-util.c
index 4c0310a..3f2ff02 100644
--- a/libedataserver/e-data-server-util.c
+++ b/libedataserver/e-data-server-util.c
@@ -251,7 +251,12 @@ e_util_utf8_strstrcase (const gchar *haystack,
 static gunichar
 stripped_char (gunichar ch)
 {
-	gunichar *decomp, retval;
+#if GLIB_CHECK_VERSION(2,29,12)
+	gunichar decomp[4];
+#else
+	gunichar *decomp;
+#endif
+	gunichar retval;
 	GUnicodeType utype;
 	gsize dlen;
 
@@ -268,11 +273,18 @@ stripped_char (gunichar ch)
 		/* Convert to lowercase, fall through */
 		ch = g_unichar_tolower (ch);
 	case G_UNICODE_LOWERCASE_LETTER:
+#if GLIB_CHECK_VERSION(2,29,12)
+		if ((dlen = g_unichar_fully_decompose (ch, FALSE, decomp, 4))) {
+			retval = decomp[0];
+			return retval;
+		}
+#else
 		if ((decomp = g_unicode_canonical_decomposition (ch, &dlen))) {
 			retval = decomp[0];
 			g_free (decomp);
 			return retval;
 		}
+#endif
 		break;
 	}
 
--
cgit v0.9.0.2
--- ./libedataserver/e-data-server-util.c.orig
+++ ./libedataserver/e-data-server-util.c
@@ -260,7 +260,6 @@
 	case G_UNICODE_CONTROL:
 	case G_UNICODE_FORMAT:
 	case G_UNICODE_UNASSIGNED:
-	case G_UNICODE_COMBINING_MARK:
 		/* Ignore those */
 		return 0;
 	default:
