diff --git a/hardware/OpenZWave.cpp b/hardware/OpenZWave.cpp
index 6bb88d1..9a77005 100644
--- a/hardware/OpenZWave.cpp
+++ b/hardware/OpenZWave.cpp
@@ -955,13 +955,15 @@ bool COpenZWave::OpenSerialConnector()
 	std::string UserPath = ConfigPath;
 	if (szStartupFolder != szUserDataFolder)
 	{
-		UserPath = szUserDataFolder;
+		ConfigPath = szUserDataFolder + "Config/";
+		UserPath = ConfigPath;
 	}
 	// Create the OpenZWave Manager.
 	// The first argument is the path to the config files (where the manufacturer_specific.xml file is located
 	// The second argument is the path for saved Z-Wave network state and the log file.  If you leave it NULL 
 	// the log file will appear in the program's working directory.
 	_log.Log(LOG_STATUS, "OpenZWave: using config in: %s", ConfigPath.c_str());
+	_log.Log(LOG_STATUS, "OpenZWave: writing config in: %s", UserPath.c_str());
 	OpenZWave::Options::Create(ConfigPath, UserPath, "--SaveConfiguration=true ");
 	EnableDisableDebug();
 	OpenZWave::Options::Get()->AddOptionInt("PollInterval", 60000); //enable polling each 60 seconds
