diff -urw src/capstone-4.0.1/bindings/python/capstone/__init__.py capstone-4.0.1/bindings/python/capstone/__init__.py
--- src/capstone-4.0.1/bindings/python/capstone/__init__.py	2019-01-10 13:45:56.000000000 +0000
+++ capstone-4.0.1/bindings/python/capstone/__init__.py	2019-10-18 20:56:27.509240524 +0000
@@ -301,6 +301,7 @@
 # - last-gasp attempt at some hardcoded paths on darwin and linux
 
 _path_list = [os.getenv('LIBCAPSTONE_PATH', None),
+              '/usr/lib/',
               pkg_resources.resource_filename(__name__, 'lib'),
               join(split(__file__)[0], 'lib'),
               '',
