--- a/contrib/traceroute.cgi
+++ b/contrib/traceroute.cgi
@@ -28,11 +28,10 @@
 # Global Settings
 #----------------
 $| = 1;
-my($nagios)  = "/usr/local/nagios";
 my($urlbase) = "/nagios";
 my($refresh) = 30;
 my($self) = basename($0);
-my($traceroute) = "/usr/sbin/traceroute -m 20 -q 1";
+my($traceroute) = "/usr/bin/traceroute -m 20 -q 1";
 
 # Generate HTTP header
 #---------------------
@@ -86,7 +85,7 @@
 my($bla);
 my($host);
 my(@hostlist);
-open(HOSTS, "$nagios/etc/hosts.cfg");
+open(HOSTS, "/etc/nagios/traceroute.cfg");
     @cfg = grep {!/#/ && /host_name/} <HOSTS>;
 close(HOSTS);
 
