diff -up ldc-1.42.0-src/runtime/CMakeLists.txt.orig ldc-1.42.0-src/runtime/CMakeLists.txt
--- ldc-1.42.0-src/runtime/CMakeLists.txt.orig	2026-03-01 19:43:34.000000000 +0100
+++ ldc-1.42.0-src/runtime/CMakeLists.txt	2026-03-01 23:40:03.742502776 +0100
@@ -276,7 +276,7 @@ set(install_libdir "${CMAKE_INSTALL_LIBD
 # Default -rpath linker option when linking against shared libraries.
 if(SHARED_LIBS_SUPPORTED)
     set(build_rpath   "${build_libdir}")
-    set(install_rpath "${install_libdir}")
+    set(install_rpath "")
 endif()
 
 # This is e.g. set from `ldc-build-runtime --installWithSuffix`.
