Include Alpine linux release file in version detection routine
--- a/mk/xe-linux-distribution
+++ b/mk/xe-linux-distribution
@@ -504,6 +504,24 @@ identify_boot2docker()
 
 }
 
+identify_alpine()
+{
+    # only tested with Alpine 3.7+
+    alpine_release="$1"
+    local major
+    local minor
+
+    if [ ! -f "${alpine_release}" ]; then
+        return 1
+    fi
+
+    major=$(awk -F. '{printf("%s", $1)}' ${alpine_release})
+    minor=$(awk -F. '{printf("%s.%s", $2, $3)}' ${alpine_release})
+
+    write_to_output "alpine" "${major}" "${minor}" "Alpine Linux $(head -n 1 ${alpine_release})"
+}
+
+
 if [ $# -eq 1 ] ; then
     exec 1>"$1"
 fi
@@ -530,6 +548,7 @@ if [ -z "${TEST}" ] ; then
     identify_lsb lsb_release         && exit 0
     identify_debian /etc/debian_version && exit 0
     identify_boot2docker /etc/boot2docker && exit 0
+    identify_alpine /etc/alpine-release && exit 0
 
 
     if [ $# -eq 1 ] ; then
