diff --git a/src/src.pro b/src/src.pro
index 4bbcad9..e677815 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -4,6 +4,8 @@ TEMPLATE = lib
 CONFIG += qt hide_symbols
 CONFIG += create_pc create_prl no_install_prl
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 # 'contacts' is too generic for the target name - use 'contactcache'
 TARGET = $${PACKAGENAME}
 target.path = $$[QT_INSTALL_LIBS]
diff --git a/tests/tests.pro b/tests/tests.pro
index d59e013..421d1bd 100644
--- a/tests/tests.pro
+++ b/tests/tests.pro
@@ -4,6 +4,8 @@ TEMPLATE = subdirs
 SUBDIRS = tst_synchronizelists tst_seasideimport tst_resolve
 OTHER_FILES += tests.xml.in
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 tests_xml.target = tests.xml
 tests_xml.depends = $$PWD/tests.xml.in
 tests_xml.commands = sed -e "s:@PACKAGENAME@:$${PACKAGENAME}:g" $< > $@
diff --git a/tests/tst_resolve/tst_resolve.pro b/tests/tst_resolve/tst_resolve.pro
index 4da24d3..d1e46ea 100644
--- a/tests/tst_resolve/tst_resolve.pro
+++ b/tests/tst_resolve/tst_resolve.pro
@@ -3,6 +3,8 @@ include(../../config.pri)
 TARGET = tst_resolve
 QT += contacts-private dbus
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 PKGCONFIG += mlocale5
 LIBS += -lphonenumber
 
diff --git a/tests/tst_seasideimport/tst_seasideimport.pro b/tests/tst_seasideimport/tst_seasideimport.pro
index 723a1bd..d301043 100644
--- a/tests/tst_seasideimport/tst_seasideimport.pro
+++ b/tests/tst_seasideimport/tst_seasideimport.pro
@@ -4,4 +4,6 @@ TARGET = tst_seasideimport
 
 SOURCES += tst_seasideimport.cpp
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 LIBS += ../../src/libcontactcache-qt5.so
diff --git a/tests/tst_synchronizelists/tst_synchronizelists.pro b/tests/tst_synchronizelists/tst_synchronizelists.pro
index 7c8f44e..b9101ca 100644
--- a/tests/tst_synchronizelists/tst_synchronizelists.pro
+++ b/tests/tst_synchronizelists/tst_synchronizelists.pro
@@ -1,4 +1,6 @@
 include(../common.pri)
 TARGET = tst_synchronizelists
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 SOURCES += tst_synchronizelists.cpp
diff --git a/translations/translations.pro b/translations/translations.pro
index 6c030ee..df7f6c6 100644
--- a/translations/translations.pro
+++ b/translations/translations.pro
@@ -3,6 +3,8 @@ TEMPLATE = aux
 TS_FILE = $$OUT_PWD/libcontacts.ts
 EE_QM = $$OUT_PWD/libcontacts_eng_en.qm
 
+INCLUDEPATH += /usr/include/QtContacts/5.0.0/ /usr/include/QtContacts/5.0.0/QtContacts
+
 ts.commands += lupdate $$PWD/../src -ts $$TS_FILE
 ts.output = $$TS_FILE
 ts.input = .

