diff --git a/test/gie/builtins.gie b/test/gie/builtins.gie
index 2501600..28c3706 100644
--- a/test/gie/builtins.gie
+++ b/test/gie/builtins.gie
@@ -6880,7 +6880,7 @@ expect  failure errno coord_transfm_outside_projection_domain
 -------------------------------------------------------------------------------
 operation +proj=vandg +a=6400000 +over
 -------------------------------------------------------------------------------
-tolerance 0.25 mm
+tolerance 0.50 mm
 
 accept  2 1
 expect  223395.249543407 111704.596633675
