--- ./toolkit/components/osfile/osfile_unix_allthreads.jsm.orig	2013-04-16 19:26:16.544461383 +0000
+++ ./toolkit/components/osfile/osfile_unix_allthreads.jsm	2013-04-16 19:31:21.717350121 +0000
@@ -40,6 +40,7 @@
   // Open libc
   let libc;
   let libc_candidates =  [ "libSystem.B.dylib",
+                           "libc.so.0.9.32",
                            "libc.so.6",
                            "libc.so" ];
   for (let i = 0; i < libc_candidates.length; ++i) {
