--- mozilla-release/netwerk.orig/sctp/src/netinet/sctp_bsd_addr.c
+++ mozilla-release/netwerk/sctp/src/netinet/sctp_bsd_addr.c
@@ -52,8 +52,6 @@
 #if defined(ANDROID)
 #include <unistd.h>
 #include <ifaddrs-android-ext.h>
-#else
-#include <sys/unistd.h>
 #endif
 #endif
 
--- mozilla-release/netwerk.orig/sctp/src/netinet/sctp_os_userspace.h
+++ mozilla-release/netwerk/sctp/src/netinet/sctp_os_userspace.h
@@ -381,11 +381,8 @@
 };
 
 #else /* !defined(Userspace_os_Windows) */
-#include <sys/cdefs.h> /* needed? added from old __FreeBSD__ */
 #include <sys/socket.h>
-#if defined(__Userspace_os_FreeBSD) || defined(__Userspace_os_OpenBSD) || defined(ANDROID)
 #include <pthread.h>
-#endif
 typedef pthread_mutex_t userland_mutex_t;
 typedef pthread_cond_t userland_cond_t;
 typedef pthread_t userland_thread_t;
--- mozilla-release/netwerk.orig/sctp/src/netinet/sctp_pcb.c
+++ mozilla-release/netwerk/sctp/src/netinet/sctp_pcb.c
@@ -30,6 +30,8 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+#define _BSD_SOURCE /* for IPPORT_RESERVED */
+
 #ifdef __FreeBSD__
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD: head/sys/netinet/sctp_pcb.c 246687 2013-02-11 21:02:49Z tuexen $");
@@ -69,11 +71,10 @@
 #endif
 #if defined(__Userspace__)
 #if !defined(__Userspace_os_Windows)
-#if defined(ANDROID)
+#include <netdb.h>
 #include <unistd.h>
+#if defined(ANDROID)
 #include <ifaddrs-android-ext.h>
-#else
-#include <sys/unistd.h>
 #endif
 #endif
 #include <user_socketvar.h>
--- mozilla-release/netwerk.orig/sctp/src/user_queue.h
+++ mozilla-release/netwerk/sctp/src/user_queue.h
@@ -31,9 +31,6 @@
 #ifndef _USER_QUEUE_H_
 #define	_USER_QUEUE_H_
 
-#if !defined (__Userspace_os_Windows)
-#include <sys/cdefs.h>
-#endif
 /*
  * This file defines four types of data structures: singly-linked lists,
  * singly-linked tail queues, lists and tail queues.
