--- ./media/webrtc/trunk/webrtc/system_wrappers/source/cpu_info.cc.orig	2013-10-25 22:27:35.000000000 +0000
+++ ./media/webrtc/trunk/webrtc/system_wrappers/source/cpu_info.cc	2013-10-29 10:10:33.578900357 +0000
@@ -17,7 +17,7 @@
 #include <sys/sysctl.h>
 #elif defined(WEBRTC_ANDROID)
 // Not implemented yet, might be possible to use Linux implementation
-#elif defined(WEBRTC_LINUX)
+#elif defined(WEBRTC_LINUX) && !defined(__UCLIBC__)
 #include <unistd.h>  // required for get_nprocs() with uClibc
 #include <sys/sysinfo.h>
 #else // defined(_SC_NPROCESSORS_ONLN)
@@ -39,7 +39,7 @@
     WEBRTC_TRACE(kTraceStateInfo, kTraceUtility, -1,
                  "Available number of cores:%d", number_of_cores_);
 
-#elif defined(WEBRTC_LINUX) && !defined(WEBRTC_ANDROID) && !defined(WEBRTC_GONK)
+#elif defined(WEBRTC_LINUX) && !defined(WEBRTC_ANDROID) && !defined(WEBRTC_GONK) && defined(__GLIBC__) && !defined(__UCLIBC__)
     number_of_cores_ = get_nprocs();
     WEBRTC_TRACE(kTraceStateInfo, kTraceUtility, -1,
                  "Available number of cores:%d", number_of_cores_);
