--- mozilla-2.0.orig/ipc/chromium/src/base/debug_util_posix.cc
+++ mozilla-2.0/ipc/chromium/src/base/debug_util_posix.cc
@@ -12,7 +12,6 @@
 #include <sys/types.h>
 #include <unistd.h>
 #ifndef ANDROID
-#include <execinfo.h>
 #include <sys/sysctl.h>
 #endif
 
@@ -116,6 +115,7 @@
 }
 
 StackTrace::StackTrace() {
+#if 0
   const int kMaxCallers = 256;
 
   void* callers[kMaxCallers];
@@ -134,12 +134,15 @@
   } else {
     trace_.resize(0);
   }
+#endif
 }
 
 void StackTrace::PrintBacktrace() {
+#if 0
   fflush(stderr);
 #ifndef ANDROID
   backtrace_symbols_fd(&trace_[0], trace_.size(), STDERR_FILENO);
+#endif
 #endif
 }
 
