diff -up mozilla-release/gfx/ycbcr/moz.build.arm mozilla-release/gfx/ycbcr/moz.build
--- mozilla-release/gfx/ycbcr/moz.build.arm	2013-12-05 17:07:32.000000000 +0100
+++ mozilla-release/gfx/ycbcr/moz.build	2013-12-10 14:47:43.235182479 +0100
@@ -61,7 +61,7 @@ else:
         'yuv_row_other.cpp',
     ]
 
-if CONFIG['CPU_ARCH'] == 'arm' and CONFIG['HAVE_ARM_NEON']:
+if CONFIG['HAVE_ARM_NEON']:
     ASFILES += [
         'yuv_row_arm.s',
     ]
diff -up mozilla-release/media/libtheora/lib/moz.build.arm mozilla-release/media/libtheora/lib/moz.build
--- mozilla-release/media/libtheora/lib/moz.build.arm	2013-12-05 17:07:48.000000000 +0100
+++ mozilla-release/media/libtheora/lib/moz.build	2013-12-12 11:45:41.882119059 +0100
@@ -8,7 +8,7 @@ MODULE = 'theora'
 
 LIBRARY_NAME = 'theora'
 
-if CONFIG['OS_TEST'] == 'arm' and CONFIG['GNU_AS']:
+if CONFIG['HAVE_ARM_NEON'] and CONFIG['GNU_AS']:
     asm_suffix = CONFIG['ASM_SUFFIX']
     ASFILES += [
         'armbits-gnu.%s' % asm_suffix,

