--- mozilla-2.0.orig/js/src/ctypes/CTypes.cpp
+++ mozilla-2.0/js/src/ctypes/CTypes.cpp
@@ -464,7 +464,7 @@
 #ifdef WIN32
   return _finite(f) != 0;
 #else
-  return finite(f);
+  return isfinite(f);
 #endif
 }
 
--- ./content/base/public/nsContentUtils.h.orig
+++ ./content/base/public/nsContentUtils.h
@@ -1994,7 +1994,11 @@
 #ifdef WIN32
   return _finite(f);
 #else
-  return finite(f);
+#ifdef _GLIBCXX_CMATH
+  return std::isfinite(f);
+#else
+  return isfinite(f);
+#endif
 #endif
 }
 
