--- ./xpcom/io/nsMultiplexInputStream.cpp.orig	2013-04-16 09:33:36.246386974 +0000
+++ ./xpcom/io/nsMultiplexInputStream.cpp	2013-04-16 09:34:21.240133225 +0000
@@ -529,7 +529,7 @@
                     rv = stream->Tell(&avail);
                     NS_ENSURE_SUCCESS(rv, rv);
 
-                    int64_t newPos = streamPos + XPCOM_MIN(avail, std::abs(remaining));
+                    int64_t newPos = streamPos + XPCOM_MIN(avail, (int64_t)std::abs(remaining));
 
                     rv = stream->Seek(NS_SEEK_END, -newPos);
                     NS_ENSURE_SUCCESS(rv, rv);
