--- mozilla-2.0.orig/xpcom/threads/nsThread.cpp
+++ mozilla-2.0/xpcom/threads/nsThread.cpp
@@ -54,7 +54,9 @@
   && defined(_GNU_SOURCE)
 # define MOZ_CANARY
 # include <unistd.h>
-# include <execinfo.h>
+# if !defined(__UCLIBC__)
+#  include <execinfo.h>
+# endif
 # include <signal.h>
 # include <fcntl.h>
 # include "nsXULAppAPI.h"
@@ -562,7 +564,9 @@
   const char msg[29] = "event took too long to run:\n";
   // use write to be safe in the signal handler
   write(Canary::sOutputFD, msg, sizeof(msg)); 
+#if !defined(__UCLIBC__)
   backtrace_symbols_fd(array, backtrace(array, 30), Canary::sOutputFD);
+#endif
 }
 
 #endif
