builders do not have sysfs mounted, so test fails

diff --git a/nodeup/pkg/model/networking/cilium_test.go b/nodeup/pkg/model/networking/cilium_test.go
index 1ea7bec..3088466 100644
--- a/nodeup/pkg/model/networking/cilium_test.go
+++ b/nodeup/pkg/model/networking/cilium_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package networking
 
 import (
+	"os"
 	"runtime"
 	"testing"
 
@@ -30,6 +31,11 @@ func TestCiliumBuilder(t *testing.T) {
 	if runtime.GOOS != "linux" {
 		t.Skipf("cilium nodeup test will only work on linux")
 	}
+
+	if _, err := os.Stat("/sys/fs/bpf"); os.IsNotExist(err) {
+		t.Skipf("cilium nodeup test requires sysfs mounted")
+	}
+
 	context := &model.NodeupModelContext{
 		Cluster: &kops.Cluster{
 			Spec: kops.ClusterSpec{
