Disable proxy test case as it requires network access.

diff -upr guile-git-v0.5.2.orig/Makefile.am guile-git-v0.5.2/Makefile.am
--- guile-git-v0.5.2.orig/Makefile.am	2023-07-29 17:06:49.767677526 +0200
+++ guile-git-v0.5.2/Makefile.am	2023-07-29 17:06:57.504310572 +0200
@@ -88,7 +88,6 @@ TESTS =
   tests/describe.scm                            \
   tests/diff.scm                                \
   tests/oid.scm                                 \
-  tests/proxy.scm                               \
   tests/reference.scm                           \
   tests/repository.scm                          \
   tests/reset.scm                               \
