diff --git a/meson.build b/meson.build
index 01510c4..36d67bb 100644
--- a/meson.build
+++ b/meson.build
@@ -447,7 +447,7 @@ configure_file(
 # updated. If git is not available, don't do anything if git-version.h
 # already exists because then we are probably working with a tarball
 # in which case the git-version.h we ship is correct.
-if git_bin.found() and run_command(
+if false and git_bin.found() and run_command(
     git_bin,
     'rev-parse',
     '--is-inside-work-tree',
