diff -U3 -r a/src/config.c b/src/config.c
--- a/src/config.c	2022-04-25 20:41:04.627290032 +0200
+++ b/src/config.c	2022-04-25 20:44:19.075583833 +0200
@@ -56,13 +56,13 @@
 static void setclearkey(dis_context_t dis_ctx, char* optarg)
 {
 	(void) optarg;
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_CLEAR_KEY, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_CLEAR_KEY, &flag);
 }
 static void setbekfile(dis_context_t dis_ctx, char* optarg)
 {
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_BEK_FILE, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_BEK_FILE, &flag);
 	dis_setopt(dis_ctx, DIS_OPT_SET_BEK_FILE_PATH, optarg);
 }
 static void setforceblock(dis_context_t dis_ctx, char* optarg)
@@ -76,14 +76,14 @@
 }
 static void setfvek(dis_context_t dis_ctx, char* optarg)
 {
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_FVEK_FILE, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_FVEK_FILE, &flag);
 	dis_setopt(dis_ctx, DIS_OPT_SET_FVEK_FILE_PATH, optarg);
 }
 static void setvmk(dis_context_t dis_ctx, char* optarg)
 {
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_VMK_FILE, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_VMK_FILE, &flag);
 	dis_setopt(dis_ctx, DIS_OPT_SET_VMK_FILE_PATH, optarg);
 }
 static void setlogfile(dis_context_t dis_ctx, char* optarg)
@@ -97,8 +97,8 @@
 }
 static void setrecoverypwd(dis_context_t dis_ctx, char* optarg)
 {
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_RECOVERY_PASSWORD, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_RECOVERY_PASSWORD, &flag);
 	dis_setopt(dis_ctx, DIS_OPT_SET_RECOVERY_PASSWORD, optarg);
 	hide_opt(optarg);
 }
@@ -111,19 +111,19 @@
 static void setro(dis_context_t dis_ctx, char* optarg)
 {
 	(void) optarg;
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_READ_ONLY, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_READ_ONLY, &flag);
 }
 static void setstateok(dis_context_t dis_ctx, char* optarg)
 {
 	(void) optarg;
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_DONT_CHECK_VOLUME_STATE, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_DONT_CHECK_VOLUME_STATE, &flag);
 }
 static void setuserpassword(dis_context_t dis_ctx, char* optarg)
 {
-	int true = TRUE;
-	dis_setopt(dis_ctx, DIS_OPT_USE_USER_PASSWORD, &true);
+	int flag = TRUE;
+	dis_setopt(dis_ctx, DIS_OPT_USE_USER_PASSWORD, &flag);
 	dis_setopt(dis_ctx, DIS_OPT_SET_USER_PASSWORD, optarg);
 	hide_opt(optarg);
 }
@@ -266,7 +266,7 @@
 		return -1;
 
 	dis_config_t* cfg = &dis_ctx->cfg;
-	int true = TRUE;
+	int flag = TRUE;
 
 
 	long_opts = malloc(nb_options * sizeof(struct option));
@@ -285,12 +285,12 @@
 		{
 			case 'c':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_CLEAR_KEY, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_CLEAR_KEY, &flag);
 				break;
 			}
 			case 'f':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_BEK_FILE, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_BEK_FILE, &flag);
 				dis_setopt(dis_ctx, DIS_OPT_SET_BEK_FILE_PATH, optarg);
 				break;
 			}
@@ -312,13 +312,13 @@
 			}
 			case 'k':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_FVEK_FILE, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_FVEK_FILE, &flag);
 				dis_setopt(dis_ctx, DIS_OPT_SET_FVEK_FILE_PATH, optarg);
 				break;
 			}
 			case 'K':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_VMK_FILE, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_VMK_FILE, &flag);
 				dis_setopt(dis_ctx, DIS_OPT_SET_VMK_FILE_PATH, optarg);
 				break;
 			}
@@ -340,7 +340,7 @@
 			}
 			case 'p':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_RECOVERY_PASSWORD, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_RECOVERY_PASSWORD, &flag);
 				dis_setopt(dis_ctx, DIS_OPT_SET_RECOVERY_PASSWORD, optarg);
 				hide_opt(optarg);
 				break;
@@ -353,17 +353,17 @@
 			}
 			case 'r':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_READ_ONLY, &true);
+				dis_setopt(dis_ctx, DIS_OPT_READ_ONLY, &flag);
 				break;
 			}
 			case 's':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_DONT_CHECK_VOLUME_STATE, &true);
+				dis_setopt(dis_ctx, DIS_OPT_DONT_CHECK_VOLUME_STATE, &flag);
 				break;
 			}
 			case 'u':
 			{
-				dis_setopt(dis_ctx, DIS_OPT_USE_USER_PASSWORD, &true);
+				dis_setopt(dis_ctx, DIS_OPT_USE_USER_PASSWORD, &flag);
 				dis_setopt(dis_ctx, DIS_OPT_SET_USER_PASSWORD, optarg);
 				hide_opt(optarg);
 				break;
