--- a/include/dislocker/ssl_bindings.h.in
+++ b/include/dislocker/ssl_bindings.h.in
@@ -26,18 +26,14 @@
 /*
  * Here stand the bindings for polarssl SHA256/SHA2/SHA-2 function for dislocker
  */
-#include "@POLARSSL_INC_FOLDER@/config.h"
+#include "@POLARSSL_INC_FOLDER@/mbedtls_config.h"
 #include "@POLARSSL_INC_FOLDER@/version.h"
 #include "@POLARSSL_INC_FOLDER@/aes.h"
 
 // Function's name changed
 #if defined(MBEDTLS_SHA256_C)
 #  include "mbedtls/sha256.h"
-#  if MBEDTLS_VERSION_NUMBER >= 0x02070000
-#    define SHA256(input, len, output)         mbedtls_sha256_ret(input, len, output, 0)
-#  else
-#    define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
-#  endif /* POLARSSL_VERSION_NUMBER >= 0x02070000 */
+#  define SHA256(input, len, output)         mbedtls_sha256(input, len, output, 0)
 #else /* defined(MBEDTLS_SHA256_C) */
 
 #  if defined(POLARSSL_SHA256_C)
