diff --git a/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn b/third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
index 744b0eb..32ff5a8 100644
--- ./third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
+++ ./third_party/swiftshader/third_party/llvm-7.0/BUILD.gn
@@ -55,7 +55,10 @@ config("swiftshader_llvm_private_config") {
   }
 
   if (is_clang) {
-    cflags += [ "-Wno-header-hygiene" ]
+    cflags += [
+        "-Wno-header-hygiene",
+        "-Wno-error=header-hygiene",
+    ]
     if (target_cpu == "x86" || target_cpu == "x64") {
       cflags += [ "-msse2" ]
     }
@@ -70,7 +73,6 @@ config("swiftshader_llvm_private_config") {
     "-Wno-unused-private-field",
     "-Wno-unused-result",
     "-Wno-unused-variable",
-    "-Wno-error=header-hygiene",
   ]
 
   defines = [
diff --git a/third_party/swiftshader/third_party/llvm-7.0/configs/linux/include/llvm/Config/config.h b/third_party/swiftshader/third_party/llvm-7.0/configs/linux/include/llvm/Config/config.h
index ff20611..ec22982 100644
--- ./third_party/swiftshader/third_party/llvm-7.0/configs/linux/include/llvm/Config/config.h
+++ ./third_party/swiftshader/third_party/llvm-7.0/configs/linux/include/llvm/Config/config.h
@@ -122,7 +122,7 @@
 /* #undef HAVE_MALLCTL */
 
 /* Define to 1 if you have the `mallinfo' function. */
-#define HAVE_MALLINFO 1
+// #define HAVE_MALLINFO 1
 
 /* Define to 1 if you have the <malloc.h> header file. */
 #define HAVE_MALLOC_H 1
