--- cc/metrics/compositor_frame_reporter.cc.orig
+++ cc/metrics/compositor_frame_reporter.cc
@@ -671,7 +671,7 @@
 
 void CompositorFrameReporter::ReportEventLatencyHistograms() const {
   for (const auto& event_metrics : events_metrics_) {
-    DCHECK_NE(event_metrics, nullptr);
+    DCHECK_EQ(event_metrics != nullptr, true);
     const std::string histogram_base_name =
         GetEventLatencyHistogramBaseName(*event_metrics);
     const int event_type_index = static_cast<int>(event_metrics->type());
