diff -up chromium-61.0.3163.79/chrome/browser/devtools/devtools_file_system_indexer.cc.gcc5fix chromium-61.0.3163.79/chrome/browser/devtools/devtools_file_system_indexer.cc
--- chrome/browser/devtools/devtools_file_system_indexer.cc.gcc5fix	2017-09-06 16:32:34.496782451 -0400
+++ chrome/browser/devtools/devtools_file_system_indexer.cc	2017-09-06 16:33:22.053865242 -0400
@@ -34,7 +34,6 @@ using base::TimeDelta;
 using base::TimeTicks;
 using content::BrowserThread;
 using std::map;
-using std::set;
 using std::string;
 using std::vector;
 
@@ -191,7 +190,7 @@ vector<FilePath> Index::Search(const str
     if (trigram != kUndefinedTrigram)
       trigrams.push_back(trigram);
   }
-  set<FileId> file_ids;
+  std::set<FileId> file_ids;
   bool first = true;
   vector<Trigram>::const_iterator it = trigrams.begin();
   for (; it != trigrams.end(); ++it) {
@@ -203,7 +202,7 @@ vector<FilePath> Index::Search(const str
       first = false;
       continue;
     }
-    set<FileId> intersection = base::STLSetIntersection<set<FileId> >(
+    std::set<FileId> intersection = base::STLSetIntersection<std::set<FileId> >(
         file_ids, index_[trigram]);
     file_ids.swap(intersection);
   }
diff -up chromium-61.0.3163.79/third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h.gcc5fix chromium-61.0.3163.79/third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h
--- third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h.gcc5fix	2017-09-06 16:33:35.213611900 -0400
+++ third_party/WebKit/Source/platform/wtf/typed_arrays/ArrayBufferContents.h	2017-09-06 16:34:34.471469606 -0400
@@ -63,7 +63,7 @@ class WTF_EXPORT ArrayBufferContents {
           allocation_length_(0),
           data_(data),
           data_length_(0),
-          kind_(AllocationKind::kNormal),
+          kind_(WTF::ArrayBufferContents::AllocationKind::kNormal),
           deleter_(deleter) {}
     DataHandle(void* allocation_base,
                size_t allocation_length,
@@ -94,11 +94,11 @@ class WTF_EXPORT ArrayBufferContents {
              reinterpret_cast<uintptr_t>(allocation_base_) +
                  allocation_length_);
       switch (kind_) {
-        case AllocationKind::kNormal:
+        case WTF::ArrayBufferContents::AllocationKind::kNormal:
           DCHECK(deleter_);
           deleter_(data_);
           return;
-        case AllocationKind::kReservation:
+        case WTF::ArrayBufferContents::AllocationKind::kReservation:
           ReleaseReservedMemory(allocation_base_, allocation_length_);
           return;
       }
diff -up chromium-61.0.3163.79/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc.gcc5fix chromium-61.0.3163.79/third_party/webrtc/modules/audio_processing/aec3/aec_state.cc
--- third_party/webrtc/modules/audio_processing/aec3/aec_state.cc.gcc5fix	2017-09-06 16:34:51.959131969 -0400
+++ third_party/webrtc/modules/audio_processing/aec3/aec_state.cc	2017-09-06 16:35:03.027919336 -0400
@@ -10,7 +10,7 @@
 
 #include "webrtc/modules/audio_processing/aec3/aec_state.h"
 
-#include <math.h>
+#include <cmath>
 #include <numeric>
 #include <vector>
 
