--- chrome/browser/ui/views/accessibility/invert_bubble_view.cc.orig
+++ chrome/browser/ui/views/accessibility/invert_bubble_view.cc
@@ -47,7 +47,7 @@
 // Tag value used to uniquely identify the "learn more" (?) button.
 constexpr int kLearnMoreButton = 100;
 
-std::unique_ptr<views::View> CreateExtraView(views::ButtonListener* listener) {
+std::unique_ptr<views::View> CreateExtraViewInternal(views::ButtonListener* listener) {
   auto learn_more = views::CreateVectorImageButtonWithNativeTheme(
       listener, vector_icons::kHelpOutlineIcon);
   learn_more->SetTooltipText(l10n_util::GetStringUTF16(IDS_LEARN_MORE));
@@ -85,7 +85,7 @@
       browser_(browser) {
   SetButtons(ui::DIALOG_BUTTON_OK);
   SetButtonLabel(ui::DIALOG_BUTTON_OK, l10n_util::GetStringUTF16(IDS_DONE));
-  SetExtraView(::CreateExtraView(this));
+  SetExtraView(::CreateExtraViewInternal(this));
   set_margins(gfx::Insets());
   chrome::RecordDialogCreation(chrome::DialogIdentifier::INVERT);
 }
