--- third_party/libaom/BUILD.gn.orig
+++ third_party/libaom/BUILD.gn
@@ -29,13 +29,7 @@
       cpu_arch_full = "x64"
     }
   } else if (current_cpu == "arm") {
-    if (arm_use_neon) {
-      cpu_arch_full = "arm-neon"
-    } else if (is_android) {
-      cpu_arch_full = "arm-neon-cpu-detect"
-    } else {
-      cpu_arch_full = "arm"
-    }
+    cpu_arch_full = "arm"
   } else {
     cpu_arch_full = current_cpu
   }
