--- third_party/pdfium/core/fxcodec/png/png_decoder.h.orig
+++ third_party/pdfium/core/fxcodec/png/png_decoder.h
@@ -9,6 +9,7 @@
 
 #include <memory>
 
+#include "core/fxcodec/fx_codec_def.h"
 #include "core/fxcodec/progressive_decoder_iface.h"
 
 #ifndef PDF_ENABLE_XFA_PNG
--- ./components/translate/core/language_detection/ngram_hash_ops_utils.cc.orig
+++ ./components/translate/core/language_detection/ngram_hash_ops_utils.cc
@@ -4,6 +4,7 @@
 
 #include "components/translate/core/language_detection/ngram_hash_ops_utils.h"
 
+#include <cstring>
 #include <vector>
 
 #include "third_party/utf/src/include/utf.h"
--- ./components/translate/core/language_detection/quantization_utils.h.orig
+++ ./components/translate/core/language_detection/quantization_utils.h
@@ -8,4 +8,5 @@
 #include <algorithm>
 #include <cmath>
+#include <cstdint>
 
 namespace translate {
