--- third_party/pdfium/core/fxcodec/png/png_decoder.h.orig
+++ third_party/pdfium/core/fxcodec/png/png_decoder.h
@@ -9,6 +9,7 @@
 
 #include <memory>
 
+#include "core/fxcodec/fx_codec_def.h"
 #include "core/fxcodec/progressive_decoder_iface.h"
 
 #ifndef PDF_ENABLE_XFA_PNG
--- ./ui/base/class_property.h.orig
+++ ./ui/base/class_property.h
@@ -10,6 +10,7 @@
 #include <map>
 #include <set>
 #include <type_traits>
+#include <memory>
 
 #include "base/component_export.h"
 #include "base/time/time.h"
--- ./device/bluetooth/bluetooth_low_energy_scan_filter.h.orig
+++ ./device/bluetooth/bluetooth_low_energy_scan_filter.h
@@ -8,6 +8,7 @@
 #include <stddef.h>
 #include <stdint.h>
 #include <vector>
+#include <memory>
 
 #include "base/time/time.h"
 #include "device/bluetooth/bluetooth_export.h"
--- ./components/password_manager/core/browser/hash_password_manager.h.orig
+++ ./components/password_manager/core/browser/hash_password_manager.h
@@ -6,6 +6,7 @@
 #define COMPONENTS_PASSWORD_MANAGER_CORE_BROWSER_HASH_PASSWORD_MANAGER_H_
 
 #include <string>
+#include <vector>
 
 #include "base/callback.h"
 #include "base/callback_list.h"
--- ./extensions/browser/extension_dialog_auto_confirm.cc.orig
+++ ./extensions/browser/extension_dialog_auto_confirm.cc
@@ -5,6 +5,7 @@
 #include "extensions/browser/extension_dialog_auto_confirm.h"
 
 #include <utility>
+#include <cstring>
 
 #include "base/check.h"
 
--- ./chrome/browser/devtools/devtools_embedder_message_dispatcher.h.orig
+++ ./chrome/browser/devtools/devtools_embedder_message_dispatcher.h
@@ -8,6 +8,7 @@
 #include <map>
 #include <memory>
 #include <string>
+#include <vector>
 
 #include "base/callback.h"
 #include "ui/gfx/geometry/insets.h"
