--- ./base/files/file_util_linux.cc.orig
+++ ./base/files/file_util_linux.cc
@@ -30,7 +30,7 @@
     case EXT2_SUPER_MAGIC:  // Also ext3 and ext4
     case MSDOS_SUPER_MAGIC:
     case REISERFS_SUPER_MAGIC:
-    case static_cast<int>(BTRFS_SUPER_MAGIC):
+    case static_cast<unsigned int>(BTRFS_SUPER_MAGIC):
     case 0x5346544E:  // NTFS
     case 0x58465342:  // XFS
     case 0x3153464A:  // JFS
@@ -40,14 +40,14 @@
       *type = FILE_SYSTEM_NFS;
       break;
     case SMB_SUPER_MAGIC:
-    case static_cast<int>(0xFF534D42):  // CIFS
+    case static_cast<unsigned int>(0xFF534D42):  // CIFS
       *type = FILE_SYSTEM_SMB;
       break;
     case CODA_SUPER_MAGIC:
       *type = FILE_SYSTEM_CODA;
       break;
-    case static_cast<int>(HUGETLBFS_MAGIC):
-    case static_cast<int>(RAMFS_MAGIC):
+    case static_cast<unsigned int>(HUGETLBFS_MAGIC):
+    case static_cast<unsigned int>(RAMFS_MAGIC):
     case TMPFS_MAGIC:
       *type = FILE_SYSTEM_MEMORY;
       break;
--- ./base/system/sys_info_posix.cc.orig
+++ ./base/system/sys_info_posix.cc
@@ -102,8 +102,8 @@
 
   switch (stats.f_type) {
     case TMPFS_MAGIC:
-    case static_cast<int>(HUGETLBFS_MAGIC):
-    case static_cast<int>(RAMFS_MAGIC):
+    case static_cast<unsigned int>(HUGETLBFS_MAGIC):
+    case static_cast<unsigned int>(RAMFS_MAGIC):
       return true;
   }
   return false;
