--- ./third_party/zlib/BUILD.gn.orig
+++ ./third_party/zlib/BUILD.gn
@@ -67,7 +67,7 @@
     #      implement the CRC32* instructions. These are optional in ARMv8.0."
     #  - Fuchsia just added a syscall for feature detection.
     # TODO(cavalcantii): crbug.com/810125.
-    if (!is_ios && !is_fuchsia) {
+    if (is_clang && !is_ios && !is_fuchsia) {
       defines = [ "CRC32_ARMV8_CRC32" ]
       if (is_android) {
         defines += [ "ARMV8_OS_ANDROID" ]
