diff --git a/build/config/linux/BUILD.gn b/build/config/linux/BUILD.gn
index 45fee9a..0769c41 100644
--- build/config/linux/BUILD.gn
+++ build/config/linux/BUILD.gn
@@ -13,6 +13,7 @@ group("linux") {
 # is applied to all targets. It is here to separate out the logic that is
 # Linux-only. This is not applied to Android, but is applied to ChromeOS.
 config("compiler") {
+  cflags_cc = [ "-fno-delete-null-pointer-checks" ]
 }
 
 # This is included by reference in the //build/config/compiler:runtime_library
diff --git a/third_party/angle/src/compiler/translator/Types.h b/third_party/angle/src/compiler/translator/Types.h
index 77db3ac..201ed4e 100644
--- third_party/angle/src/compiler/translator/Types.h
+++ third_party/angle/src/compiler/translator/Types.h
@@ -236,7 +236,7 @@ class TType
     {
     }
 
-    TBasicType getBasicType() const { return type; }
+    constexpr TBasicType getBasicType() const { return type; }
     void setBasicType(TBasicType t);
 
     TPrecision getPrecision() const { return precision; }
