--- ./third_party/blink/renderer/core/xml/xslt_processor.h.orig
+++ ./third_party/blink/renderer/core/xml/xslt_processor.h
@@ -77,7 +77,7 @@
 
   void reset();
 
-  static void ParseErrorFunc(void* user_data, const xmlError*);
+  static void ParseErrorFunc(void* user_data, xmlError*);
   static void GenericErrorFunc(void* user_data, const char* msg, ...);
 
   // Only for libXSLT callbacks
--- ./third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc.orig
+++ ./third_party/blink/renderer/core/xml/xslt_processor_libxslt.cc
@@ -66,7 +66,7 @@
   // It would be nice to do something with this error message.
 }
 
-void XSLTProcessor::ParseErrorFunc(void* user_data, const xmlError* error) {
+void XSLTProcessor::ParseErrorFunc(void* user_data, xmlError* error) {
   FrameConsole* console = static_cast<FrameConsole*>(user_data);
   if (!console)
     return;
