--- ui/accessibility/platform/inspect/tree_formatter.h.orig
+++ ui/accessibility/platform/inspect/tree_formatter.h
@@ -5,6 +5,9 @@
 #ifndef UI_ACCESSIBILITY_PLATFORM_INSPECT_TREE_FORMATTER_H_
 #define UI_ACCESSIBILITY_PLATFORM_INSPECT_TREE_FORMATTER_H_
 
+#include <vector>
+#include <memory>
+
 #include "ui/accessibility/platform/inspect/inspect.h"
 
 #include "ui/gfx/native_widget_types.h"
--- third_party/openscreen/src/util/crypto/random_bytes.h.orig
+++ third_party/openscreen/src/util/crypto/random_bytes.h
@@ -5,6 +5,7 @@
 #ifndef UTIL_CRYPTO_RANDOM_BYTES_H_
 #define UTIL_CRYPTO_RANDOM_BYTES_H_
 
+#include <cstdint>
 #include <array>
 
 namespace openscreen {
--- third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h.orig	2020-08-23 12:43:09.289768089 +0200
+++ third_party/webrtc/modules/audio_processing/aec3/clockdrift_detector.h	2020-08-23 12:43:39.386217911 +0200
@@ -12,5 +12,6 @@
 #define MODULES_AUDIO_PROCESSING_AEC3_CLOCKDRIFT_DETECTOR_H_
 
+#include <cstddef>
 #include <array>
 
 namespace webrtc {
--- chrome/browser/search/background/ntp_backgrounds.h.orig	2020-08-23 14:35:40.788447650 +0200
+++ chrome/browser/search/background/ntp_backgrounds.h	2020-08-23 14:36:16.414865836 +0200
@@ -6,5 +6,6 @@
 #define CHROME_BROWSER_SEARCH_BACKGROUND_NTP_BACKGROUNDS_H_
 
+#include <cstddef>
 #include <array>
 
 class GURL;
