diff --git a/third_party/crashpad/crashpad/snapshot/linux/cpu_context_linux.cc b/third_party/crashpad/crashpad/snapshot/linux/cpu_context_linux.cc
index 6ba52a8..0c7a9f9 100644
--- ./third_party/crashpad/crashpad/snapshot/linux/cpu_context_linux.cc
+++ ./third_party/crashpad/crashpad/snapshot/linux/cpu_context_linux.cc
@@ -14,6 +14,7 @@
 
 #include "snapshot/linux/cpu_context_linux.h"
 
+#include <limits>
 #include <stddef.h>
 #include <string.h>
 
