--- a/base/allocator/partition_allocator/partition_root.cc
+++ b/base/allocator/partition_allocator/partition_root.cc
@@ -248,9 +248,9 @@
   // However, no perfect solution really exists to make threads + fork()
   // cooperate, but deadlocks are real (and fork() is used in DEATH_TEST()s),
   // and other malloc() implementations use the same techniques.
-  int err =
-      pthread_atfork(BeforeForkInParent, AfterForkInParent, AfterForkInChild);
-  PA_CHECK(err == 0);
+  //int err =
+  //    pthread_atfork(BeforeForkInParent, AfterForkInParent, AfterForkInChild);
+  //PA_CHECK(err == 0);
 #endif  // BUILDFLAG(IS_LINUX) || BUILDFLAG(IS_CHROMEOS)
 }
 
