--- a/third_party/perfetto/src/trace_processor/perfetto_sql/engine/created_function.cc
+++ b/third_party/perfetto/src/trace_processor/perfetto_sql/engine/created_function.cc
@@ -107,7 +107,7 @@
   // the destructors run correctly for non-trivial members of the
   // union.
   using Data =
-      std::variant<int64_t, double, OwnedString, OwnedBytes, nullptr_t>;
+      std::variant<int64_t, double, OwnedString, OwnedBytes, std::nullptr_t>;
 
   StoredSqlValue(SqlValue value) {
     switch (value.type) {
@@ -132,7 +132,7 @@
   }
 
   SqlValue AsSqlValue() {
-    if (std::holds_alternative<nullptr_t>(data)) {
+    if (std::holds_alternative<std::nullptr_t>(data)) {
       return SqlValue();
     } else if (std::holds_alternative<int64_t>(data)) {
       return SqlValue::Long(std::get<int64_t>(data));
--- a/third_party/perfetto/src/trace_processor/sqlite/sql_source.h
+++ b/third_party/perfetto/src/trace_processor/sqlite/sql_source.h
@@ -17,6 +17,7 @@
 #ifndef SRC_TRACE_PROCESSOR_SQLITE_SQL_SOURCE_H_
 #define SRC_TRACE_PROCESSOR_SQLITE_SQL_SOURCE_H_
 
+#include <cstdint>
 #include <optional>
 #include <string>
 #include <string_view>
