--- ./net/third_party/quiche/src/quic/core/quic_one_block_arena.h
+++ ./net/third_party/quiche/src/quic/core/quic_one_block_arena.h
@@ -69,7 +69,7 @@
 
 // QuicConnections currently use around 1KB of polymorphic types which would
 // ordinarily be on the heap. Instead, store them inline in an arena.
-using QuicConnectionArena = QuicOneBlockArena<1056>;
+using QuicConnectionArena = QuicOneBlockArena<1408>;
 
 }  // namespace quic
 
