--- base/test/BUILD.gn
+++ base/test/BUILD.gn
@@ -225,9 +225,6 @@
     "//base/third_party/dynamic_annotations",
     "//testing/gmock",
     "//testing/gtest",
-    "//third_party/icu:icuuc",
-    "//third_party/libxml:libxml_utils",
-    "//third_party/libxml:xml_reader",
   ]
 
   if (is_posix || is_fuchsia) {
@@ -300,9 +297,6 @@
     ]
     public_deps -= [ "//base:i18n" ]
     deps -= [
-      "//third_party/icu:icuuc",
-      "//third_party/libxml:libxml_utils",
-      "//third_party/libxml:xml_reader",
     ]
   }
 }
--- components/policy/core/common/BUILD.gn
+++ components/policy/core/common/BUILD.gn
@@ -173,8 +173,6 @@
     "//net",
     "//services/identity/public/cpp:cpp_types",
     "//services/network/public/cpp",
-    "//third_party/libxml:xml_writer",
-    "//third_party/re2",
     "//url",
   ]
 
@@ -422,7 +420,6 @@
     "//services/network:test_support",
     "//testing/gmock",
     "//testing/gtest",
-    "//third_party/libxml:xml_reader",
   ]
 }
 
--- components/wifi/BUILD.gn
+++ components/wifi/BUILD.gn
@@ -26,8 +26,6 @@
   if (is_win) {
     libs = [ "iphlpapi.lib" ]
     deps += [
-      "//third_party/libxml:xml_reader",  # Safe, data come from system.
-      "//third_party/libxml:xml_writer",
     ]
   }
 
--- services/data_decoder/BUILD.gn
+++ services/data_decoder/BUILD.gn
@@ -33,7 +33,6 @@
     "//mojo/public/cpp/bindings",
     "//net",
     "//skia",
-    "//third_party/libxml:xml_reader",
     "//ui/gfx",
     "//ui/gfx/geometry",
   ]
@@ -117,8 +116,6 @@
   deps = [
     ":lib",
     "//base",
-    "//third_party/libxml:libxml_utils",
-    "//third_party/libxml:xml_reader",
   ]
   dict = "//third_party/libxml/fuzz/xml.dict"
   seed_corpus = "//third_party/libxml/fuzz/seed_corpus"
--- tools/traffic_annotation/auditor/BUILD.gn
+++ tools/traffic_annotation/auditor/BUILD.gn
@@ -26,7 +26,6 @@
 
   deps = [
     "//components/policy:full_runtime_code_generate",
-    "//third_party/protobuf:protobuf_full",
   ]
 }
 
@@ -70,10 +69,6 @@
     ":traffic_annotation",
     "//base",
     "//net/traffic_annotation",
-    "//third_party/libxml:xml_reader",  # Safe, consumes Chromium-produced file.
-    "//third_party/libxml:xml_writer",
-    "//third_party/protobuf:protobuf_full",
-    "//third_party/re2",
   ]
 }
 
--- chrome/browser/BUILD.gn
+++ chrome/browser/BUILD.gn
@@ -2221,10 +2221,8 @@
     "//third_party/icu",
     "//third_party/leveldatabase",
     "//third_party/libaddressinput",
-    "//third_party/libxml:xml_writer",
     "//third_party/libyuv",
     "//third_party/metrics_proto",
-    "//third_party/re2",
     "//third_party/smhasher:cityhash",
     "//third_party/webrtc_overrides:webrtc_component",
     "//third_party/widevine/cdm:buildflags",
