not needed with clang 15
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1540,14 +1540,14 @@
         cflags += [ "-Wno-unqualified-std-cast-call" ]
       }
 
-      if (!is_nacl && !(is_chromeos ||
-                        default_toolchain == "//build/toolchain/cros:target")) {
-        # TODO(https://crbug.com/1322823): Remove flags once potential miscompile is investigated.
-        cflags += [
-          "-Xclang",
-          "-no-opaque-pointers",
-        ]
-      }
+      # if (!is_nacl && !(is_chromeos ||
+      #                   default_toolchain == "//build/toolchain/cros:target")) {
+      #   # TODO(https://crbug.com/1322823): Remove flags once potential miscompile is investigated.
+      #   cflags += [
+      #     "-Xclang",
+      #     "-no-opaque-pointers",
+      #   ]
+      # }
 
       if (is_fuchsia) {
         # TODO(https://bugs.chromium.org/p/fuchsia/issues/detail?id=77383)
