--- ./third_party/skia/src/ports/SkFontHost_FreeType_common.cpp.orig
+++ ./third_party/skia/src/ports/SkFontHost_FreeType_common.cpp
@@ -712,7 +712,7 @@
             canvas->drawPaint(colrPaint);
             break;
         }
-        case FT_COLR_PAINTFORMAT_TRANSFORMED:
+        case FT_COLR_PAINTFORMAT_TRANSFORM:
         case FT_COLR_PAINTFORMAT_TRANSLATE:
         case FT_COLR_PAINTFORMAT_ROTATE:
         case FT_COLR_PAINTFORMAT_SKEW:
@@ -759,8 +759,8 @@
     SkMatrix transform;
 
     switch (colrv1_paint.format) {
-        case FT_COLR_PAINTFORMAT_TRANSFORMED: {
-            transform = ToSkMatrix(colrv1_paint.u.transformed.affine);
+        case FT_COLR_PAINTFORMAT_TRANSFORM: {
+            transform = ToSkMatrix(colrv1_paint.u.transform.affine);
             break;
         }
         case FT_COLR_PAINTFORMAT_TRANSLATE: {
@@ -880,10 +880,10 @@
             traverse_result = colrv1_start_glyph(canvas, palette, face, paint.u.colr_glyph.glyphID,
                                                  FT_COLOR_NO_ROOT_TRANSFORM);
             break;
-        case FT_COLR_PAINTFORMAT_TRANSFORMED:
+        case FT_COLR_PAINTFORMAT_TRANSFORM:
             colrv1_transform(canvas, face, paint);
             traverse_result = colrv1_traverse_paint(canvas, palette, face,
-                                                    paint.u.transformed.paint, visited_set);
+                                                    paint.u.transform.paint, visited_set);
             break;
         case FT_COLR_PAINTFORMAT_TRANSLATE:
             colrv1_transform(canvas, face, paint);
