--- ./net/dns/dns_reloader.cc
+++ ./net/dns/dns_reloader.cc
@@ -4,8 +4,7 @@
 
 #include "net/dns/dns_reloader.h"
 
-#if defined(OS_POSIX) && !defined(OS_APPLE) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
 
 #include <resolv.h>
 
--- ./net/dns/host_resolver_manager.cc
+++ ./net/dns/host_resolver_manager.cc
@@ -2779,8 +2779,7 @@
   NetworkChangeNotifier::AddConnectionTypeObserver(this);
   if (system_dns_config_notifier_)
     system_dns_config_notifier_->AddObserver(this);
-#if defined(OS_POSIX) && !defined(OS_APPLE) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID)
+#if defined(__GLIBC__)
   EnsureDnsReloaderInit();
 #endif
 
--- ./net/dns/host_resolver_proc.cc
+++ ./net/dns/host_resolver_proc.cc
@@ -159,8 +159,7 @@
   base::ScopedBlockingCall scoped_blocking_call(FROM_HERE,
                                                 base::BlockingType::WILL_BLOCK);
 
-#if defined(OS_POSIX) && !defined(OS_APPLE) && !defined(OS_OPENBSD) && \
-    !defined(OS_ANDROID) && !defined(OS_FUCHSIA)
+#if defined(__GLIBC__)
   DnsReloaderMaybeReload();
 #endif
   base::Optional<AddressInfo> ai;
--- ./net/dns/dns_config_service_linux.cc.orig
+++ ./net/dns/dns_config_service_linux.cc
@@ -486,20 +480,11 @@
 
 std::unique_ptr<struct __res_state>
 DnsConfigServiceLinux::ResolvReader::GetResState() {
-  auto res = std::make_unique<struct __res_state>();
-  memset(res.get(), 0, sizeof(struct __res_state));
-
-  if (res_ninit(res.get()) != 0) {
-    CloseResState(res.get());
-    return nullptr;
-  }
-
-  return res;
+  return nullptr;
 }
 
 void DnsConfigServiceLinux::ResolvReader::CloseResState(
     struct __res_state* res) {
-  res_nclose(res);
 }
 
 DnsConfigServiceLinux::DnsConfigServiceLinux()
