--- ./third_party/angle/third_party/vulkan-validation-layers/src/loader/loader.c.orig
+++ ./third_party/angle/third_party/vulkan-validation-layers/src/loader/loader.c
@@ -37,6 +37,9 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <sys/param.h>
 #endif
+#if defined(HAVE_GETAUXVAL)
+#include <sys/auxv.h>
+#endif
 #include <sys/types.h>
 #if defined(_WIN32)
 #include "dirent_on_windows.h"
@@ -240,6 +243,8 @@
 #elif defined(HAVE___SECURE_GETENV)
     (void)inst;
     return __secure_getenv(name);
+#elif defined(HAVE_GETAUXVAL)
+    return getauxval(AT_SECURE) ? NULL : getenv(name);
 #else
 #pragma message(                                                                       \
     "Warning:  Falling back to non-secure getenv for environmental lookups!  Consider" \
--- ./third_party/angle/third_party/vulkan-validation-layers/BUILD.gn.orig
+++ ./third_party/angle/third_party/vulkan-validation-layers/BUILD.gn
@@ -264,8 +264,7 @@
     cflags = [ "/wd4201" ]
   }
   if (is_linux) {
-    # assume secure_getenv() is available
-    defines += [ "HAVE_SECURE_GETENV" ]
+    defines += [ "HAVE_GETAUXVAL" ]
   }
 }
 
