--- third_party/blink/renderer/modules/webcodecs/BUILD.gn.orig
+++ third_party/blink/renderer/modules/webcodecs/BUILD.gn
@@ -63,6 +63,7 @@
     "//media/mojo/clients",
     "//media/mojo/mojom",
     "//third_party/libyuv:libyuv",
+    "//third_party/opus",
   ]
   if (media_use_openh264) {
     deps += [ "//third_party/openh264:encoder" ]
--- third_party/blink/renderer/modules/webcodecs/DEPS.orig
+++ third_party/blink/renderer/modules/webcodecs/DEPS
@@ -19,6 +19,7 @@
 
     "+third_party/libyuv",
     "+third_party/openh264",
+    "+third_party/opus",
 
     "+ui/gfx/color_space.h",
     "+ui/gfx/geometry/rect.h",
