--- a/malcontent-timer-extension-agent/meson.build
+++ b/malcontent-timer-extension-agent/meson.build
@@ -50,14 +50,14 @@ configure_file(
 configure_file(
   input: 'malcontent-timer-extension-agent.conf.in',
   output: 'malcontent-timer-extension-agent.conf',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'sysusersdir'),
+  install_dir: '/usr/lib/sysusers.d',
   install_tag: 'runtime',
   configuration: config,
 )
 configure_file(
   input: 'malcontent-timer-extension-agent.service.in',
   output: 'malcontent-timer-extension-agent.service',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'systemdsystemunitdir'),
+  install_dir: '/usr/lib/systemd/system',
   install_tag: 'runtime',
   configuration: config,
 )
--- a/malcontent-timerd/meson.build
+++ b/malcontent-timerd/meson.build
@@ -44,14 +44,14 @@ configure_file(
 configure_file(
   input: 'malcontent-timerd.conf.in',
   output: 'malcontent-timerd.conf',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'sysusersdir'),
+  install_dir: '/usr/lib/sysusers.d',
   install_tag: 'runtime',
   configuration: config,
 )
 configure_file(
   input: 'malcontent-timerd.service.in',
   output: 'malcontent-timerd.service',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'systemdsystemunitdir'),
+  install_dir: '/usr/lib/systemd/system',
   install_tag: 'runtime',
   configuration: config,
 )
--- a/malcontent-webd-update/meson.build
+++ b/malcontent-webd-update/meson.build
@@ -28,14 +28,14 @@ config.set('libexecdir', join_paths(get_option('prefix'), get_option('libexecdir
 configure_file(
   input: 'malcontent-webd-update.service.in',
   output: 'malcontent-webd-update.service',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'systemdsystemunitdir'),
+  install_dir: '/usr/lib/systemd/system',
   install_tag: 'runtime',
   configuration: config,
 )
 install_data(
   'malcontent-webd-update.timer',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'systemdsystemunitdir'),
+  install_dir: '/usr/lib/systemd/system',
   install_tag: 'runtime',
 )
 
-subdir('tests')
\ No newline at end of file
+subdir('tests')
--- a/malcontent-webd/meson.build
+++ b/malcontent-webd/meson.build
@@ -44,14 +44,14 @@ configure_file(
 configure_file(
   input: 'malcontent-webd.conf.in',
   output: 'malcontent-webd.conf',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'sysusersdir'),
+  install_dir: '/usr/lib/sysusers.d',
   install_tag: 'runtime',
   configuration: config,
 )
 configure_file(
   input: 'malcontent-webd.service.in',
   output: 'malcontent-webd.service',
-  install_dir: dependency('systemd').get_variable(pkgconfig: 'systemdsystemunitdir'),
+  install_dir: '/usr/lib/systemd/system',
   install_tag: 'runtime',
   configuration: config,
 )
