mktime is the posix function
--
diff --git a/src/convenience/convenience.c b/src/convenience/convenience.c
index f2e6567..51a4479 100644
--- a/src/convenience/convenience.c
+++ b/src/convenience/convenience.c
@@ -247,7 +247,7 @@ time_t localtimestr_to_time(const char * str, double * fraction) {
 	if (!p)
 		return 0;
 #ifndef _WIN32
-	return timelocal(p);
+	return mktime(p);
 #else
 	return mktime(p);
 #endif
