diff --git a/srsly/cloudpickle/cloudpickle.py b/srsly/cloudpickle/cloudpickle.py
index 317be69..7011fa9 100644
--- a/srsly/cloudpickle/cloudpickle.py
+++ b/srsly/cloudpickle/cloudpickle.py
@@ -56,7 +56,7 @@ import warnings
 from .compat import pickle
 from collections import OrderedDict
 from typing import ClassVar, Generic, Union, Tuple, Callable
-from pickle import _getattribute
+from pickle import _getattribute as _pickle_getattribute
 from importlib._bootstrap import _find_spec
 
 try:  # pragma: no branch
@@ -192,6 +192,14 @@ def _is_registered_pickle_by_value(module):
     return False
 
 
+if sys.version_info >= (3, 14):
+    def _getattribute(obj, name):
+        return _pickle_getattribute(obj, name.split('.'))
+else:
+    def _getattribute(obj, name):
+        return _pickle_getattribute(obj, name)[0]
+
+
 def _whichmodule(obj, name):
     """Find the module an object belongs to.
 
@@ -231,7 +239,7 @@ def _whichmodule(obj, name):
         ):
             continue
         try:
-            if _getattribute(module, name)[0] is obj:
+            if _getattribute(module, name) is obj:
                 return module_name
         except Exception:
             pass
@@ -306,7 +314,7 @@ def _lookup_module_and_qualname(obj, name=None):
         return None
 
     try:
-        obj2, parent = _getattribute(module, name)
+        obj2 = _getattribute(module, name)
     except AttributeError:
         # obj was not found inside the module it points to
         return None
@@ -383,7 +391,7 @@ def _find_imported_submodules(code, top_level_dependencies):
                 # sys.modules.
                 if name is not None and name.startswith(prefix):
                     # check whether the function can address the sub-module
-                    tokens = set(name[len(prefix):].split('.'))
+                    tokens = set(name[len(prefix) :].split('.'))
                     if not tokens - set(code.co_names):
                         subimports.append(sys.modules[name])
     return subimports
@@ -842,6 +850,10 @@ def _rehydrate_skeleton_class(skeleton_class, class_dict):
             registry = attr
         else:
             setattr(skeleton_class, attrname, attr)
+
+    if sys.version_info >= (3, 13) and "__firstlineno__" in class_dict:
+            skeleton_class.__firstlineno__ = class_dict["__firstlineno__"]
+
     if registry is not None:
         for subclass in registry:
             skeleton_class.register(subclass)
diff --git a/srsly/tests/cloudpickle/cloudpickle_test.py b/srsly/tests/cloudpickle/cloudpickle_test.py
index e4dba00..045bf68 100644
--- a/srsly/tests/cloudpickle/cloudpickle_test.py
+++ b/srsly/tests/cloudpickle/cloudpickle_test.py
@@ -112,7 +112,10 @@ def test_extract_class_dict():
             return "c"
 
     clsdict = _extract_class_dict(C)
-    assert sorted(clsdict.keys()) == ["C_CONSTANT", "__doc__", "method_c"]
+    expected_keys = ["C_CONSTANT", "__doc__", "method_c"]
+    if sys.version_info >= (3, 13):
+        expected_keys.insert(2, "__firstlineno__")
+    assert sorted(clsdict.keys()) == expected_keys
     assert clsdict["C_CONSTANT"] == 43
     assert clsdict["__doc__"] is None
     assert clsdict["method_c"](C()) == C().method_c()
@@ -2044,11 +2047,9 @@ class CloudPickleTest(unittest.TestCase):
         self.assertEqual(f2.__doc__, f.__doc__)
 
     def test_wraps_preserves_function_annotations(self):
-        def f(x):
+        def f(x: int) -> float:
             pass
 
-        f.__annotations__ = {'x': 1, 'return': float}
-
         @wraps(f)
         def g(x):
             f(x)
@@ -2260,6 +2261,12 @@ class CloudPickleTest(unittest.TestCase):
         inner_func = depickled_factory()
         assert inner_func() == _TEST_GLOBAL_VARIABLE
 
+    @pytest.mark.skipif(
+        sys.version_info > (3, 14),
+        reason="Can cause CPython 3.14 interpreter to crash",
+        # This interpreter crash is reported upstream in
+        # https://github.com/python/cpython/issues/131543
+    )
     def test_recursion_during_pickling(self):
         class A:
             def __getattribute__(self, name):
@@ -2403,7 +2410,7 @@ class CloudPickleTest(unittest.TestCase):
                 MyClass.__annotations__ = {'attribute': type_}
 
                 def check_annotations(obj, expected_type, expected_type_str):
-                    assert obj.__annotations__["attribute"] == expected_type
+                    assert type(obj).__annotations__["attribute"] == expected_type
                     assert (
                         obj.method.__annotations__["arg"] == expected_type
                     )
