--- a/src/php_memcache.h.a
+++ b/src/php_memcache.h
@@ -67,7 +67,7 @@
 PHP_FUNCTION(memcache_flush);
 PHP_FUNCTION(memcache_set_sasl_auth_data);
 
-#define PHP_MEMCACHE_VERSION "4.0.5.2"
+#define PHP_MEMCACHE_VERSION "8.0"
 
 #define MMC_DEFAULT_TIMEOUT 1				/* seconds */
 #define MMC_DEFAULT_RETRY 15 				/* retry failed server after x seconds */
--- a/src/memcache.c.a
+++ b/src/memcache.c
@@ -780,7 +780,7 @@
 	php_info_print_table_start();
 	php_info_print_table_header(2, "memcache support", "enabled");
 	php_info_print_table_row(2, "Version", PHP_MEMCACHE_VERSION);
-	php_info_print_table_row(2, "Revision", "$Revision$");
+	php_info_print_table_row(2, "Revision", "b:NON_BLOCKING_IO_php8 v:8.0 c:36d7181 d:2020-12-06");
 	php_info_print_table_end();
 
 	DISPLAY_INI_ENTRIES();
