diff -upr i3-4.11.orig/i3bar/src/main.c i3-4.11/i3bar/src/main.c
--- i3-4.11.orig/i3bar/src/main.c	2015-10-06 22:45:01.791199587 +0200
+++ i3-4.11/i3bar/src/main.c	2015-10-06 22:45:52.921029327 +0200
@@ -45,14 +45,7 @@ void debuglog(char *fmt, ...) {
  *
  */
 char *expand_path(char *path) {
-    static glob_t globbuf;
-    if (glob(path, GLOB_NOCHECK | GLOB_TILDE, NULL, &globbuf) < 0) {
-        ELOG("glob() failed\n");
-        exit(EXIT_FAILURE);
-    }
-    char *result = sstrdup(globbuf.gl_pathc > 0 ? globbuf.gl_pathv[0] : path);
-    globfree(&globbuf);
-    return result;
+    return sstrdup(path);
 }
 
 void print_usage(char *elf_name) {
diff -upr i3-4.11.orig/libi3/resolve_tilde.c i3-4.11/libi3/resolve_tilde.c
--- i3-4.11.orig/libi3/resolve_tilde.c	2015-10-06 22:45:01.801199553 +0200
+++ i3-4.11/libi3/resolve_tilde.c	2015-10-06 22:46:32.874229612 +0200
@@ -19,27 +19,5 @@
  *
  */
 char *resolve_tilde(const char *path) {
-    static glob_t globbuf;
-    char *head, *tail, *result;
-
-    tail = strchr(path, '/');
-    head = sstrndup(path, tail ? (size_t)(tail - path) : strlen(path));
-
-    int res = glob(head, GLOB_TILDE, NULL, &globbuf);
-    free(head);
-    /* no match, or many wildcard matches are bad */
-    if (res == GLOB_NOMATCH || globbuf.gl_pathc != 1)
-        result = sstrdup(path);
-    else if (res != 0) {
-        err(EXIT_FAILURE, "glob() failed");
-    } else {
-        head = globbuf.gl_pathv[0];
-        result = scalloc(strlen(head) + (tail ? strlen(tail) : 0) + 1, 1);
-        strncpy(result, head, strlen(head));
-        if (tail)
-            strncat(result, tail, strlen(tail));
-    }
-    globfree(&globbuf);
-
-    return result;
+    return sstrdup(path);
 }
