--- a/examples/chrony.conf.example.orig	2003-06-16 11:59:01.000000000 -0400
+++ b/examples/chrony.conf.example	2003-06-16 12:00:13.000000000 -0400
@@ -3,5 +3,5 @@
 #
 # This is an example chrony configuration file.  You should copy it to
-# /etc/chrony.conf after uncommenting and editing the options that you
+# /etc/chrony/chrony.conf after uncommenting and editing the options that you
 # want to enable.  I have not included the more obscure options.  Refer
 # to the documentation for these.
@@ -91,5 +91,5 @@
 # generally want this, so it is uncommented.
 
-driftfile /etc/chrony.drift
+driftfile /etc/chrony/chrony.drift
 
 # If you want to use the program called chronyc to configure aspects of
@@ -100,5 +100,5 @@
 # assumed by default.
 
-keyfile /etc/chrony.keys
+keyfile /etc/chrony/chrony.keys
 
 # Tell chronyd which numbered key in the file is used as the password
@@ -158,6 +158,6 @@
 ! log measurements statistics tracking
 
-If you have real time clock support enabled (see below), you might want
-this line instead:
+# If you have real time clock support enabled (see below), you might want
+# this line instead:
 
 ! log measurements statistics tracking rtc
@@ -269,5 +269,5 @@
 # kernel.  (Note, these options apply only to Linux.)
 
-! rtcfile /etc/chrony.rtc
+! rtcfile /etc/chrony/chrony.rtc
 
 # Your RTC can be set to keep Universal Coordinated Time (UTC) or local
@@ -285,5 +285,5 @@
 # using devfs), uncomment and edit the following line.
 
-! rtcdevice /dev/misc/rtc
+rtcdevice /dev/misc/rtc
 
 #######################################################################
