--- libreoffice-4.4.0.3/configure.ac.orig	2015-01-22 22:05:28.000000000 -0200
+++ libreoffice-4.4.0.3/configure.ac	2015-02-02 20:11:20.639411153 -0200
@@ -569,7 +569,7 @@
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|linux-musl*|k*bsd*-gnu*)
     test_gtk=yes
     build_gstreamer_1_0=yes
     build_gstreamer_0_10=yes
@@ -4058,7 +4058,7 @@
     esac
     ;;
 
-linux-gnu*)
+linux-gnu*|linux-musl*)
     COM=GCC
     GUIBASE=unx
     OS=LINUX
@@ -6967,7 +6967,7 @@
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
         ;;
 
-    linux-gnu*)
+    linux-gnu*|linux-musl*)
         JAVAINC="-I$JAVA_HOME/include"
         JAVAINC="$JAVAINC -I$JAVA_HOME/include/linux"
         test -d "$JAVA_HOME/include/native_thread" && JAVAINC="$JAVAINC -I$JAVA_HOME/include/native_thread"
@@ -9429,21 +9429,6 @@
 fi
 AC_SUBST(SYSTEM_LIBEXTTEXTCAT_DATA)
 
-dnl ***************************************
-dnl testing libc version for Linux...
-dnl ***************************************
-if test "$_os" = "Linux"; then
-    AC_MSG_CHECKING([whether libc is >= 2.1.1])
-    exec 6>/dev/null # no output
-    AC_CHECK_LIB(c, gnu_get_libc_version, HAVE_LIBC=yes; export HAVE_LIBC)
-    exec 6>&1 # output on again
-    if test "$HAVE_LIBC"; then
-        AC_MSG_RESULT([yes])
-    else
-        AC_MSG_ERROR([no, upgrade libc])
-    fi
-fi
-
 dnl =========================================
 dnl Check for the Windows  SDK.
 dnl =========================================
@@ -12741,7 +12726,7 @@
 
     case "$host_os" in
 
-    aix*|dragonfly*|freebsd*|linux-gnu*|*netbsd*|openbsd*)
+    aix*|dragonfly*|freebsd*|linux-gnu*|linux-musl*|*netbsd*|openbsd*)
         if test "$ENABLE_JAVA" != "" -a "$JDK"!="gcj"; then
             pathmunge "$JAVA_HOME/bin" "after"
         fi
@@ -12789,7 +12774,6 @@
     esac
 
     pathmunge "$SRC_ROOT/solenv/bin" "before"
-    pathmunge "." "before"
 fi
 
 AC_SUBST(LO_PATH)
