--- a/makefile	2021-01-28 23:45:28.755650703 +0100
+++ b/makefile	2021-01-29 00:06:45.740697386 +0100
@@ -44,9 +44,6 @@
 endif
 
 DEFINES = -DVERSION=$(VERSION) -DRELEASE=$(RELEASE) -DDEV=$(DEV) -DINSTALL_PATH=\"$(DATA_DIR)\" -DLOCALE_DIR=\"$(LOCALE_DIR)\" -DUNIX=$(UNIX)
-ifndef NO_PAK
-DEFINES += -DPAK_FILE=\"$(PAK_FILE)\"
-endif
 
 LDFLAGS += `sdl2-config --libs` -lSDL2_mixer -lSDL2_image -lSDL2_ttf -lz -lm -lpng
 
@@ -137,14 +134,8 @@
 clean:
 	$(RM) $(PROG) $(ED_PROG) $(PAK_PROG) $(PO_PROG) $(TILE_PROG) $(PAK_FILE) $(LOCALE_MO) $(TILE_PROG) *.o makefile.dep
 
-buildpak: $(PAK_PROG)
-ifndef NO_PAK
-	./$(PAK_PROG) data gfx music sound font $(PAK_FILE)
-	./$(PAK_PROG) -test $(PAK_FILE)
-endif
-
 # install
-install: all buildpak
+install: all
 ifeq ($(DEV),1)
 	echo Cannot install if DEV is set to 1!
 else
@@ -160,11 +151,7 @@
 	mkdir -p $(MAN_DIR)
 
 	cp $(PROG) $(BIN_DIR)$(PROG)
-ifndef NO_PAK
-	cp $(PAK_FILE) $(DATA_DIR)$(PAK_FILE)
-else
 	cp -a data gfx music sound font $(DATA_DIR)
-endif
 	cp $(DOCS) $(DOC_DIR)
 	cp $(ICONS)16x16.png $(ICON_DIR)16x16/apps/$(PROG).png
 	cp $(ICONS)32x32.png $(ICON_DIR)32x32/apps/$(PROG).png
