diff --git a/tests/test_srtfile.py b/tests/test_srtfile.py
index 819aa4d..b24e8ac 100644
--- a/tests/test_srtfile.py
+++ b/tests/test_srtfile.py
@@ -119,4 +119,4 @@ class TestSlice(unittest.TestCase):
     def test_at(self):
-        self.assertEquals(len(self.file.at((0, 0, 31, 0))), 1)
-        self.assertEquals(len(self.file.at(seconds=31)), 1)
+        self.assertEqual(len(self.file.at((0, 0, 31, 0))), 1)
+        self.assertEqual(len(self.file.at(seconds=31)), 1)
 
@@ -139,3 +139,3 @@ class TestText(unittest.TestCase):
         ])
-        self.assertEquals(srt_file.text, 'Hello')
+        self.assertEqual(srt_file.text, 'Hello')
 
@@ -146,3 +146,3 @@ class TestText(unittest.TestCase):
         ])
-        self.assertEquals(srt_file.text, 'Hello\nWorld !')
+        self.assertEqual(srt_file.text, 'Hello\nWorld !')
 
@@ -253,3 +253,3 @@ class TestIntegration(unittest.TestCase):
         items = pysrt.open(os.path.join(self.base_path, 'no-indexes.srt'))
-        self.assertEquals(len(items), 7)
+        self.assertEqual(len(items), 7)
 
diff --git a/tests/test_srtitem.py b/tests/test_srtitem.py
index f4a6f2f..9d29538 100644
--- a/tests/test_srtitem.py
+++ b/tests/test_srtitem.py
@@ -199,4 +199,4 @@ class TestSerialAndParsing(unittest.TestCase):
         item = SubRipItem.from_string(self.string_index)
-        self.assertEquals(item.index, 'foo')
-        self.assertEquals(item.text, 'Hello !')
+        self.assertEqual(item.index, 'foo')
+        self.assertEqual(item.text, 'Hello !')
 
@@ -204,4 +204,4 @@ class TestSerialAndParsing(unittest.TestCase):
         item = SubRipItem.from_string(self.no_index)
-        self.assertEquals(item.index, None)
-        self.assertEquals(item.text, 'Hello world !')
+        self.assertEqual(item.index, None)
+        self.assertEqual(item.text, 'Hello world !')
 
@@ -209,3 +209,3 @@ class TestSerialAndParsing(unittest.TestCase):
         item = SubRipItem.from_string(self.junk_after_timestamp)
-        self.assertEquals(item, self.item)
+        self.assertEqual(item, self.item)
 
