diff --git a/version.sh b/version.sh
index 956f135..4dae9b9 100755
--- a/version.sh
+++ b/version.sh
@@ -1,17 +1,2 @@
-#!/usr/bin/env bash
-cd `dirname "$0"`
-if [ -d ".git" ] ; then
-    v=`git describe`
-    echo "$v" | grep -qv '^beta-' && v=`git describe --match '[0-9]*'`
-    [ -z "$v" ] && v=`git describe --match 'beta-[0-9]*'`
-    v=${v/-/.}
-    v="${v/-/\~}"
-    printf $v | tee version
-    if echo $v | grep -qv '~' ; then
-        printf $v > release_version
-    fi
-elif [ -f version ] ; then
-    cat version
-else
-    cat release_version
-fi
+#!/bin/sh
+cat release_version
