diff --git a/src/Makefile b/src/Makefile
index e9021fe..1d86ecd 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -54,11 +54,7 @@ INSTALL = install
 
 DESTDIR ?=
 
-ifeq ($(shell uname -m),x86_64)
-	LIBSUBDIR := lib64
-else
-	LIBSUBDIR := lib
-endif
+LIBSUBDIR := lib
 
 PREFIX ?= /usr
 LIBDIR ?= $(PREFIX)/$(LIBSUBDIR)
@@ -77,9 +73,9 @@ $(OBJDIR)/libbpf.so.$(VERSION): $(OBJDIR)/libbpf.so.$(LIBBPF_VERSION)
 	ln -sf $(^F) $@
 
 $(OBJDIR)/libbpf.so.$(LIBBPF_VERSION): $(OBJS)
-	$(CC) -shared $(ALL_LDFLAGS) -Wl,--version-script=$(VERSION_SCRIPT) \
+	$(CC) -shared -Wl,--version-script=$(VERSION_SCRIPT) \
 				     -Wl,-soname,libbpf.so.$(VERSION) \
-				     $^ -o $@
+				     $^ -o $@ $(ALL_LDFLAGS) 
 
 $(OBJDIR)/libbpf.pc:
 	sed -e "s|@PREFIX@|$(PREFIX)|" \
