=== modified file 'protocols/twitter/twitter_lib.c'
--- protocols/twitter/twitter_lib.c	2011-12-26 18:17:51 +0000
+++ protocols/twitter/twitter_lib.c	2012-01-10 23:36:29 +0000
@@ -886,10 +886,6 @@
 		td->http_fails = 0;
 		if (!(ic->flags & OPT_LOGGED_IN))
 			imcb_connected(ic);
-	} else if (req->status_code == 401) {
-		imcb_error(ic, "Authentication failure");
-		imc_logout(ic, FALSE);
-		goto end;
 	} else {
 		// It didn't go well, output the error and return.
 		if (++td->http_fails >= 5)
@@ -938,10 +934,6 @@
 		td->http_fails = 0;
 		if (!(ic->flags & OPT_LOGGED_IN))
 			imcb_connected(ic);
-	} else if (req->status_code == 401) {
-		imcb_error(ic, "Authentication failure");
-		imc_logout(ic, FALSE);
-		goto end;
 	} else {
 		// It didn't go well, output the error and return.
 		if (++td->http_fails >= 5)

