--- a/test/test_bom_detection.rb
+++ b/test/test_bom_detection.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class BOMDetectionTest < MiniTest::Test
+class BOMDetectionTest < Minitest::Test
   include DetencHelper
 
   SAMPLE = "Pâté: €3.20"
--- a/test/test_encoding_detection.rb
+++ b/test/test_encoding_detection.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class EncodingDetectionTest < MiniTest::Test
+class EncodingDetectionTest < Minitest::Test
   include DetencHelper
 
   SAMPLE = "Pâté: €3.20"
--- a/test/test_iso_8859_15.rb
+++ b/test/test_iso_8859_15.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class ISO_8859_15_Test < MiniTest::Test
+class ISO_8859_15_Test < Minitest::Test
   include DetencHelper
 
   INVALID_BYTES = (0x00..0x08).to_a + (0x0E..0x1F).to_a + (0x7F..0x9F).to_a
--- a/test/test_unknown_encoding.rb
+++ b/test/test_unknown_encoding.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class UnknownEncodingTest < MiniTest::Test
+class UnknownEncodingTest < Minitest::Test
   include DetencHelper
 
   def test_should_be_unknown_if_it_contains_all_bytes
--- a/test/test_us_ascii.rb
+++ b/test/test_us_ascii.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class US_ASCII_Test < MiniTest::Test
+class US_ASCII_Test < Minitest::Test
   include DetencHelper
 
   def test_should_be_us_ascii_if_it_contains_all_printable_bytes_under_0x7F
--- a/test/test_utf_8.rb
+++ b/test/test_utf_8.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class UTF8StressTest < MiniTest::Test
+class UTF8StressTest < Minitest::Test
   include DetencHelper
 
   # From Markus Kuhn's stress tests
--- a/test/test_windows_1252.rb
+++ b/test/test_windows_1252.rb
@@ -1,6 +1,6 @@
 require_relative 'test_helper'
 
-class Windows1252Test < MiniTest::Test
+class Windows1252Test < Minitest::Test
   include DetencHelper
 
   INVALID_BYTES = (0x00..0x08).to_a + (0x0E..0x1F).to_a + [0x7F, 0x81, 0x8D, 0x8F, 0x90, 0x9D]
