diff -urp a/src/Logger.hpp b/src/Logger.hpp
--- a/src/Logger.hpp	2024-02-19 17:02:33.774879053 +0100
+++ b/src/Logger.hpp	2024-02-19 17:04:47.334832260 +0100
@@ -22,6 +22,11 @@
 #include "String.hpp"
 #include "UdpSocket.hpp"
 
+// Musl uses "#define stdin (stdin)", so the enum declaration fails with a syntax error
+#undef stdin
+#undef stdout
+#undef stderr
+
 // only C++14 : using namespace std::string_literals;
 
 class FileRec {
