diff -urp a/src/ListContainer.cpp b/src/ListContainer.cpp
--- a/src/ListContainer.cpp	2024-02-19 17:17:59.304562545 +0100
+++ b/src/ListContainer.cpp	2024-02-19 17:18:15.114557160 +0100
@@ -833,7 +833,7 @@ bool ListContainer::findEndsWith(const c
             }
         }
     }
-    return NULL;
+    return false;
 }
 
 // For phrase lists - grab the text, score and type of a given phrase, based on item number within list
