diff --git a/testsuite/libffi.bhaible/bhaible.exp b/testsuite/libffi.bhaible/bhaible.exp
index 44aebc5..245f18c 100644
--- a/testsuite/libffi.bhaible/bhaible.exp
+++ b/testsuite/libffi.bhaible/bhaible.exp
@@ -24,7 +24,7 @@ global compiler_vendor
 # was done in a pretty lazy fashion, and requires the use of compiler
 # flags to disable warnings for now.
 if { [string match $compiler_vendor "gnu"] } {
-    set warning_options "-Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized";
+    set warning_options "-Wno-unused-variable -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-uninitialized -Wno-stringop-overflow";
 }
 if { [string match $compiler_vendor "microsoft"] } {
     # -wd4996  suggest use of vsprintf_s instead of vsprintf
