--- a/rspec-its.gemspec
+++ b/rspec-its.gemspec
@@ -21,7 +21,7 @@
     'source_code_uri'   => 'https://github.com/rspec/rspec-its',
   }
 
-  spec.files         = `git ls-files`.split($/) - %w[cucumber.yml]
+  spec.files         = Dir["lib/**/*"]
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^(test|spec|features)/})
   spec.require_paths = ["lib"]
