--- ./include/private/gcconfig.h.orig
+++ ./include/private/gcconfig.h
@@ -2483,6 +2483,7 @@
 #if ((defined(UNIX_LIKE) && (defined(DARWIN) || defined(HURD) \
                              || defined(OPENBSD) || defined(ARM32) \
                              || defined(MIPS) || defined(AVR32))) \
+			     || defined(__UCLIBC__) \
      || (defined(LINUX) && (defined(SPARC) || defined(M68K))) \
      || (defined(RTEMS) && defined(I386))) && !defined(NO_GETCONTEXT)
 # define NO_GETCONTEXT
