--- a/src/util/locale_utils.cc
+++ b/src/util/locale_utils.cc
@@ -86,10 +86,10 @@ const char *locale_charset( void )

 bool is_utf8_locale( void ) {
   /* Verify locale calls for UTF-8 */
-  if ( strcmp( locale_charset(), "UTF-8" ) != 0 &&
-       strcmp( locale_charset(), "utf-8" ) != 0 ) {
-    return 0;
-  }
+  //if ( strcmp( locale_charset(), "UTF-8" ) != 0 &&
+  //    strcmp( locale_charset(), "utf-8" ) != 0 ) {
+  // return 0;
+  //}
   return 1;
 }
