--- a/scripts/Makefile.am	2022-10-27 01:18:11.000000000 +0000
+++ b/scripts/Makefile.am	2022-10-28 20:22:44.569619128 +0000
@@ -5,6 +5,6 @@
 CLEANFILES = $(bin_SCRIPTS)
 
 mosh:	mosh.pl ../VERSION Makefile
-	perl -Mdiagnostics -c $(srcdir)/mosh.pl
+	perl -c $(srcdir)/mosh.pl
 	@sed -e "s/\@VERSION\@/`cat ../VERSION`/" -e "s/\@PACKAGE_STRING\@/@PACKAGE_STRING@/" $(srcdir)/mosh.pl > mosh
 	@chmod a+x mosh
