--- a/configure
+++ b/configure
@@ -224,7 +224,7 @@
   cc="${toolprefix}gcc"
   clinker="${toolprefix}gcc"
   cprepro="${toolprefix}gcc"
-  cprepro_options="-std=c99 -U__GNUC__ '-D__REDIRECT(name,proto,alias)=name proto' '-D__REDIRECT_NTH(name,proto,alias)=name proto' -E"
+  cprepro_options="-std=c99 -U__GNUC__ '-D__builtin_constant_p(x)=0' '-D__REDIRECT(name,proto,alias)=name proto' '-D__REDIRECT_NTH(name,proto,alias)=name proto' -E"
   libmath="-lm"
   system="linux"
 fi
@@ -274,7 +274,7 @@
         cc="${toolprefix}gcc"
         clinker="${toolprefix}gcc"
         cprepro="${toolprefix}gcc"
-        cprepro_options="-std=c99 -U__GNUC__ -E"
+        cprepro_options="-std=c99 -U__GNUC__ '-D__builtin_constant_p(x)=0' -U__SIZEOF_INT128__ -E"
         libmath="-lm"
         system="linux"
         ;;
@@ -320,7 +320,7 @@
         clinker="${toolprefix}gcc"
         clinker_options="-m32"
         cprepro="${toolprefix}gcc"
-        cprepro_options="-std=c99 -m32 -U__GNUC__ -E"
+        cprepro_options="-std=c99 -m32 -U__GNUC__ '-D__builtin_constant_p(x)=0' -E"
         libmath="-lm"
         system="linux"
         ;;
@@ -380,7 +380,7 @@
         clinker="${toolprefix}gcc"
         clinker_options="-m64"
         cprepro="${toolprefix}gcc"
-        cprepro_options="-std=c99 -m64 -U__GNUC__ -E"
+        cprepro_options="-std=c99 -m64 -U__GNUC__ '-D__builtin_constant_p(x)=0' -U__SIZEOF_INT128__ -E"
         libmath="-lm"
         system="linux"
         ;;
@@ -423,7 +423,7 @@
   clinker="${toolprefix}gcc"
   clinker_options="$model_options"
   cprepro="${toolprefix}gcc"
-  cprepro_options="$model_options -std=c99 -U__GNUC__ -E"
+  cprepro_options="$model_options -std=c99 -U__GNUC__ '-D__builtin_constant_p(x)=0' -U__SIZEOF_INT128__ -E"
   libmath="-lm"
   system="linux"
 fi
