diff --git a/pppd/plugins/radius/pathnames.h b/pppd/plugins/radius/pathnames.h
index 3a00d2c..c606f33 100644
--- a/pppd/plugins/radius/pathnames.h
+++ b/pppd/plugins/radius/pathnames.h
@@ -22,7 +22,7 @@
 
 /* normally defined in the Makefile */
 #ifndef PPP_PATH_ETC_RADIUSCLIENT_CONF
-#define PPP_PATH_ETC_RADIUSCLIENT_CONF       SYSCONFDIR "/radiusclient.conf"
+#define PPP_PATH_ETC_RADIUSCLIENT_CONF       SYSCONFDIR "/radiusclient/radiusclient.conf"
 #endif
 
 #endif /* PATHNAMES_H */
