--- a/pppd/plugins/rp-pppoe/pppoe.h	2020-02-06 09:59:38.086103782 +0100
+++ b/pppd/plugins/rp-pppoe/pppoe.h	2020-02-06 20:50:20.159705281 +0100
@@ -84,18 +84,6 @@
 #include <linux/if_ether.h>
 #endif
 
-#ifdef HAVE_NETINET_IF_ETHER_H
-#include <sys/types.h>
-
-#ifdef HAVE_SYS_SOCKET_H
-#include <sys/socket.h>
-#endif
-#ifndef HAVE_SYS_DLPI_H
-#include <netinet/if_ether.h>
-#endif
-#endif
-
-
 /* Ethernet frame types according to RFC 2516 */
 #define ETH_PPPOE_DISCOVERY 0x8863
 #define ETH_PPPOE_SESSION   0x8864
