--- linux-3.7/drivers/media/dvb-core/dvb-usb-ids.h.orig
+++ linux-3.7/drivers/media/dvb-core/dvb-usb-ids.h
@@ -151,6 +151,7 @@
 #define USB_PID_KWORLD_395U_2				0xe39b
 #define USB_PID_KWORLD_395U_3				0xe395
 #define USB_PID_KWORLD_395U_4				0xe39a
+#define USB_PID_KWORLD_D393				0xd393
 #define USB_PID_KWORLD_MC810				0xc810
 #define USB_PID_KWORLD_PC160_2T				0xc160
 #define USB_PID_KWORLD_PC160_T				0xc161
--- linux-3.7/drivers/media/usb/dvb-usb-v2/rtl28xxu.c.orig
+++ linux-3.7/drivers/media/usb/dvb-usb-v2/rtl28xxu.c
@@ -1350,6 +1350,8 @@
 		&rtl2832u_props, "Dexatek DK mini DVB-T Dongle", NULL) },
 	{ DVB_USB_DEVICE(USB_VID_TERRATEC, 0x00d7,
 		&rtl2832u_props, "TerraTec Cinergy T Stick+", NULL) },
+	{ DVB_USB_DEVICE(USB_VID_KWORLD_2, 0xd393,
+		&rtl2832u_props, "DVB-T TV Stick D393", NULL) },
 	{ }
 };
 MODULE_DEVICE_TABLE(usb, rtl28xxu_id_table);
