--- a/Makefile
+++ b/Makefile
@@ -127,16 +127,16 @@ realclean:
 distclean: realclean
 
 install:
-	$(MKPATH) $(PREFIX)/bin
-	$(INSTALL) $(exe) $(PREFIX)/bin
-	$(MKPATH) $(PREFIX)/share/man/man1/
-	$(INSTALLD) $(man) $(PREFIX)/share/man/man1/
+	$(MKPATH) $(DESTDIR)$(PREFIX)/bin
+	$(INSTALL) $(exe) $(DESTDIR)$(PREFIX)/bin
+	$(MKPATH) $(DESTDIR)$(PREFIX)/share/man/man1/
+	$(INSTALLD) $(man) $(DESTDIR)$(PREFIX)/share/man/man1/
 
 libinstall:
-	$(MKPATH) $(LIBROOT)/lib
-	$(CPR) lib/. $(LIBROOT)/lib/
-	find $(LIBROOT) -type f -exec chmod 644 {} +
-	find $(LIBROOT) -type d -exec chmod 755 {} +
+	$(MKPATH) $(DESTDIR)$(LIBROOT)/lib
+	$(CPR) lib/. $(DESTDIR)$(LIBROOT)/lib/
+	find $(DESTDIR)$(LIBROOT) -type f -exec chmod 644 {} +
+	find $(DESTDIR)$(LIBROOT) -type d -exec chmod 755 {} +
 
 uninstall:
 	$(UNINSTALL) $(PREFIX)/bin/$(exe)
