--- a/icmake/installer
+++ b/icmake/installer
@@ -5,13 +5,13 @@
     exit 0
 fi
 
-for src in `find -mindepth 1 -type d`           # create missing target dirs
+for src in `find . -mindepth 1 -type d`           # create missing target dirs
 do
     [ ! -e $1$src ] && mkdir -p $1$src
 done
 
-for file in `find -type f -or -type l`
+for file in `find . -type f -or -type l`
 do
-    cp -d --preserve=timestamps $file $1$file
+    cp -H $file $1$file
 done
 
--- a/icmake/findall
+++ b/icmake/findall
@@ -15,7 +15,7 @@
     if (pattern != "")
         pattern = "-name '" + pattern + "'";
 
-    entries = backtick(cmd + " " + pattern + " -printf \"%f\\n\"");
+    entries = backtick(cmd + " " + pattern + " -exec basename {} \\;");
 
     for (idx = listlen(entries); idx--; )
         ret += (list)cutEoln(entries[idx]);
