--- a/diffiehellman/diffiehellman.ih
+++ b/diffiehellman/diffiehellman.ih
@@ -15,7 +15,7 @@
 using namespace std;
 using namespace FBB;
 
-#if  OPENSSL_VERSION_NUMBER < 0x10100000L
+#if  OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 
 inline BN_GENCB *BN_GENCB_new()
 {
--- a/digestbuf/digestbuf.ih
+++ b/digestbuf/digestbuf.ih
@@ -9,7 +9,7 @@
 
 #include "../ohexstreambuf/ohexstreambuf"
 
-#if  OPENSSL_VERSION_NUMBER < 0x10100000L
+#if  OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
     inline EVP_MD_CTX *EVP_MD_CTX_new()
     {
         EVP_MD_CTX *ret = new EVP_MD_CTX;
--- a/hmacbuf/hmacbuf.ih
+++ b/hmacbuf/hmacbuf.ih
@@ -7,7 +7,7 @@
 #include <bobcat/exception>
 #include "../ohexstreambuf/ohexstreambuf"
 
-#if  OPENSSL_VERSION_NUMBER < 0x10100000L
+#if  OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
 
 inline HMAC_CTX *HMAC_CTX_new()
 {
