diff --git a/src/oidentd_util.h.orig b/src/oidentd_util.h
index a136c47..639d49f 100644
--- a/src/oidentd_util.h.orig
+++ b/src/oidentd_util.h
@@ -58,7 +58,7 @@ int find_user(const char *temp_user, uid_t *uid);
 int find_group(const char *temp_group, gid_t *gid);

 int random_seed(void);
-inline int randval(int i);
+extern int randval(int i);

 #ifndef HAVE_SNPRINTF
 	int snprintf(char *str, size_t n, char const *fmt, ...);
