--- a/Makefile
+++ b/Makefile
@@ -156,23 +156,23 @@
 	go build -ldflags "$(LD_FLAGS)" -o out/cf .
 else
 out/cf: $(GOSRC)
-	CGO_ENABLED=0 go build \
+	go build \
 		$(REQUIRED_FOR_STATIC_BINARY) \
 		-ldflags "$(LD_FLAGS_LINUX)" -o out/cf .
 endif
 
 out/cf-cli_linux_i686: $(GOSRC)
-	CGO_ENABLED=0 GOARCH=386 GOOS=linux go build \
+	GOARCH=386 GOOS=linux go build \
 							$(REQUIRED_FOR_STATIC_BINARY) \
 							-ldflags "$(LD_FLAGS_LINUX)" -o out/cf-cli_linux_i686 .
 
 out/cf-cli_linux_x86-64: $(GOSRC)
-	CGO_ENABLED=0 GOARCH=amd64 GOOS=linux go build \
+	GOARCH=amd64 GOOS=linux go build \
 							$(REQUIRED_FOR_STATIC_BINARY) \
 							-ldflags "$(LD_FLAGS_LINUX)" -o out/cf-cli_linux_x86-64 .
 							
 out/cf-cli_linux_arm64: $(GOSRC)
-	CGO_ENABLED=0 GOARCH=arm64 GOOS=linux go build \
+	GOARCH=arm64 GOOS=linux go build \
 							$(REQUIRED_FOR_STATIC_BINARY) \
 							-ldflags "$(LD_FLAGS_LINUX)" -o out/cf-cli_linux_arm64 .
 
