--- a/vhs.go
+++ b/vhs.go
@@ -130,7 +130,7 @@
 
 	path, _ := launcher.LookPath()
 	enableNoSandbox := os.Getenv("VHS_NO_SANDBOX") != ""
-	u, err := launcher.New().Leakless(false).Bin(path).NoSandbox(enableNoSandbox).Launch()
+	u, err := launcher.New().Leakless(false).Bin(path).Set("disable-gpu").NoSandbox(enableNoSandbox).Launch()
 	if err != nil {
 		return fmt.Errorf("could not launch browser: %w", err)
 	}
