--- a/motion-dist.conf.in
+++ b/motion-dist.conf.in
@@ -633,70 +633,6 @@
 # NOTE: If motion doesn't properly detect a lost camera, it also won't know it found one.
 ; on_camera_found value
 
-#####################################################################
-# Common Options for database features.
-# Options require database options to be active also.
-#####################################################################
-
-# Log to the database when creating motion triggered picture file  (default: on)
-; sql_log_picture on
-
-# Log to the database when creating a snapshot image file (default: on)
-; sql_log_snapshot on
-
-# Log to the database when creating motion triggered movie file (default: off)
-; sql_log_movie off
-
-# Log to the database when creating timelapse movies file (default: off)
-; sql_log_timelapse off
-
-# SQL query string that is sent to the database
-# Use same conversion specifiers has for text features
-# Additional special conversion specifiers are
-# %n = the number representing the file_type
-# %f = filename with full path
-# Default value:
-# Create tables :
-##
-# Mysql
-# CREATE TABLE security (camera int, filename char(80) not null, frame int, file_type int, time_stamp timestamp(14), event_time_stamp timestamp(14));
-#
-# Postgresql
-# CREATE TABLE security (camera int, filename char(80) not null, frame int, file_type int, time_stamp timestamp without time zone, event_time_stamp timestamp without time zone);
-#
-# insert into security(camera, filename, frame, file_type, time_stamp, text_event) values('%t', '%f', '%q', '%n', '%Y-%m-%d %T', '%C')
-; sql_query insert into security(camera, filename, frame, file_type, time_stamp, event_time_stamp) values('%t', '%f', '%q', '%n', '%Y-%m-%d %T', '%C')
-
-
-############################################################
-# Database Options
-############################################################
-
-# database type : mysql, postgresql, sqlite3 (default : not defined)
-; database_type value
-
-# database to log to (default: not defined)
-# for sqlite3, the full path and name for the database.
-; database_dbname value
-
-# The host on which the database is located (default: localhost)
-; database_host value
-
-# User account name for database (default: not defined)
-; database_user value
-
-# User password for database (default: not defined)
-; database_password value
-
-# Port on which the database is located
-#  mysql 3306 , postgresql 5432 (default: not defined)
-; database_port value
-
-# Database wait time in milliseconds for locked database to
-# be unlocked before returning database locked error (default 0)
-; database_busy_timeout 0
-
-
 
 ############################################################
 # Video Loopback Device (vloopback project)
