From bd7c67faad24770c7378464a640b9baa72257ede Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Adamski?= <michal@ert.pl>
Date: Mon, 16 Nov 2020 15:11:30 +0100
Subject: [PATCH] relax dependencies

---
 requirements.txt | 12 ++++++------
 setup.py         |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/requirements.txt b/requirements.txt
index 7dfdf28..467e9f2 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
-lxml==4.3.5
-pycryptodomex==3.9.4
-construct==2.10.54
-argon2-cffi==19.2.0
-python-dateutil==2.8.0
-future==0.17.1
+lxml==4.3.*
+pycryptodomex==3.9.*
+construct==2.10.*
+argon2-cffi==19.2.*
+python-dateutil==2.8.*
+future==0.17.*
diff --git a/setup.py b/setup.py
index 55c17bc..0609302 100644
--- a/setup.py
+++ b/setup.py
@@ -18,7 +18,7 @@ setup(
     install_requires=[
         "python-dateutil",
         # FIXME python2 - last version to support python2
-        "construct==2.10.54",
+        "construct==2.10.*",
         "argon2_cffi",
         "pycryptodomex>=3.6.2",
         "lxml",
-- 
2.25.1

