diff --git a/pykeepass/kdbx_parsing/common.py b/pykeepass/kdbx_parsing/common.py
index 188552e..b04d36d 100644
--- a/pykeepass/kdbx_parsing/common.py
+++ b/pykeepass/kdbx_parsing/common.py
@@ -1,6 +1,6 @@
-from Crypto.Cipher import AES, ChaCha20, Salsa20
+from Cryptodome.Cipher import AES, ChaCha20, Salsa20
 from .twofish import Twofish
-from Crypto.Util import Padding as CryptoPadding
+from Cryptodome.Util import Padding as CryptoPadding
 import hashlib
 from construct import (
     Adapter, BitStruct, BitsSwapped, Container, Flag, Padding, RepeatUntil,
diff --git a/pykeepass/kdbx_parsing/twofish.py b/pykeepass/kdbx_parsing/twofish.py
index 05779fb..c4dc7a7 100644
--- a/pykeepass/kdbx_parsing/twofish.py
+++ b/pykeepass/kdbx_parsing/twofish.py
@@ -25,8 +25,8 @@
 __all__ = ['Twofish']
 
 from . import pytwofish
-from Crypto.Util.strxor import strxor
-from Crypto.Util.Padding import pad
+from Cryptodome.Util.strxor import strxor
+from Cryptodome.Util.Padding import pad
 
 MODE_ECB = 1
 MODE_CBC = 2
