--- a/yeahconsole.c
+++ b/yeahconsole.c
@@ -26,7 +26,7 @@
 /*************************************************************************/
 
 #include <X11/Xlib.h>
-#include <X11/keysym.h>
+#include <X11/XKBlib.h>
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 #include <unistd.h>
@@ -130,7 +130,7 @@
 			       CurrentTime);
 	    break;
 	case KeyPress:
-	    key = XKeycodeToKeysym(dpy, event.xkey.keycode, 0);
+		key = XkbKeycodeToKeysym(dpy, event.xkey.keycode, 0, 0);
 	    if (key == opt_key) {
 		if (!hidden) {
 		    XGetInputFocus(dpy, &current_focused, &revert_to);
@@ -351,7 +351,7 @@
     XMapWindow(dpy, win);
 }
 
-void init_xterm(move)
+void init_xterm(int move)
 {
     XEvent ev;
     long dummy;
