--- a/Makefile	2019-11-09 13:53:03.712153728 +0100
+++ b/Makefile	2019-11-09 13:54:27.943704512 +0100
@@ -151,8 +151,8 @@
 		if [ ! -d $(DESTDIR)$(MAN8PATH) ]; 		\
 		then	mkdir -p $(DESTDIR)$(MAN8PATH);	fi
 		#
-		touch       		$(DESTDIR)$(DEFPATH)/atop
-		chmod 644      		$(DESTDIR)$(DEFPATH)/atop
+		#touch       		$(DESTDIR)$(DEFPATH)/atop
+		#chmod 644      		$(DESTDIR)$(DEFPATH)/atop
 		#
 		cp atop   		$(DESTDIR)$(BINPATH)/atop
 		chown root		$(DESTDIR)$(BINPATH)/atop
