commit 62a62017514b7c56a88a0b3d9671b5778ab693e3
Author: Andy Postnikov <apostnikov@gmail.com>
Date:   Mon Feb 16 03:18:45 2026 +0100

    fix clang 21 warning

diff --git a/extension/meminfo.c b/extension/meminfo.c
index fb5ff4e..9beb8d1 100644
--- a/meminfo.c
+++ b/meminfo.c
@@ -196,7 +196,7 @@ void meminfo_browse_zvals_from_symbol_table(php_stream *stream, char* frame_labe
     HashPosition pos;
 
     zend_string *key;
-    zend_long index;
+    zend_ulong index;
 
     zend_hash_internal_pointer_reset_ex(p_symbol_table, &pos);
 
@@ -299,6 +299,8 @@ void meminfo_hash_dump(php_stream *stream, HashTable *ht, zend_bool is_object, H
             case HASH_KEY_IS_LONG:
                 php_stream_printf(stream, "            \"%ld\":\"%s\"", num_key, zval_id);
                 break;
+            default:
+                break;
         }
 
         zend_hash_move_forward_ex(ht, &pos);
commit 86b7699f2681ddd71ad75a1ba260edf5d3623cf2
Author: Andy Postnikov <apostnikov@gmail.com>
Date:   Mon Feb 16 03:27:58 2026 +0100

    use ZEND_ULONG_FMT for compatibility with 32 and 64 bits

diff --git a/extension/meminfo.c b/extension/meminfo.c
index 9beb8d1..e6273af 100644
--- a/meminfo.c
+++ b/meminfo.c
@@ -297,7 +297,7 @@ void meminfo_hash_dump(php_stream *stream, HashTable *ht, zend_bool is_object, H
 
                 break;
             case HASH_KEY_IS_LONG:
-                php_stream_printf(stream, "            \"%ld\":\"%s\"", num_key, zval_id);
+                php_stream_printf(stream, "            \"" ZEND_ULONG_FMT "\":\"%s\"", num_key, zval_id);
                 break;
             default:
                 break;
@@ -348,7 +348,7 @@ void meminfo_zval_dump(php_stream * stream, char * frame_label, zend_string * sy
 
     php_stream_printf(stream, "    \"%s\" : {\n", zval_identifier);
     php_stream_printf(stream, "        \"type\" : \"%s\",\n", zend_get_type_by_const(Z_TYPE_P(zv)));
-    php_stream_printf(stream, "        \"size\" : \"%ld\",\n", meminfo_get_element_size(zv));
+    php_stream_printf(stream, "        \"size\" : \"" ZEND_ULONG_FMT "\",\n", meminfo_get_element_size(zv));
 
     if (frame_label) {
         zend_string * escaped_frame_label;
