--- a/local-projects/pgloader-3.6.9/src/main.lisp
+++ b/local-projects/pgloader-3.6.9/src/main.lisp
@@ -81,9 +81,6 @@
     (("after") :type string :optional t
      :documentation "SQL script to run after loading the data")
 
-    ("self-upgrade" :type string :optional t
-                    :documentation "Path to pgloader newer sources")
-
     ("regress" :type boolean :optional t
                :documentation "Drive regression testing")))
 
@@ -130,33 +127,6 @@
   (command-line-arguments:show-option-help *opt-spec*)
   (when quit (uiop:quit +os-code-error-usage+)))
 
-(defvar *self-upgraded-already* nil
-  "Keep track if we did reload our own source code already.")
-
-(defun self-upgrade (namestring &optional debug)
-  "Load pgloader sources at PATH-TO-PGLOADER-SOURCES."
-  (let ((pgloader-pathname (uiop:directory-exists-p
-                            (uiop:parse-unix-namestring namestring))))
-    (unless pgloader-pathname
-      (format t "No such directory: ~s~%" namestring)
-      (uiop:quit +os-code-error+))
-
-    ;; now the real thing
-    (handler-case
-        (handler-bind ((warning #'muffle-warning))
-          (let ((asdf:*central-registry* (list* pgloader-pathname
-                                                asdf:*central-registry*)))
-            (format t "Self-upgrading from sources at ~s~%"
-                    (uiop:native-namestring pgloader-pathname))
-            (with-output-to-string (*standard-output*)
-              (asdf:load-system :pgloader
-                                :verbose nil
-                                :force-not *self-upgrade-immutable-systems*))))
-      (condition (c)
-        (format t "Fatal: ~a~%" c)
-        (format t "~a~%" *self-upgrade-immutable-systems*)
-        (when debug (invoke-debugger c))))))
-
 (defun parse-summary-filename (summary debug)
   "Return the pathname where to write the summary output."
   (when summary
@@ -199,7 +169,7 @@
                                 dry-run on-error-stop context
                                 ((:load-lisp-file load))
 				client-min-messages log-min-messages summary
-				root-dir self-upgrade
+				root-dir
                                 with set field cast type encoding before after
                                 no-ssl-cert-verification
                                 regress)
@@ -217,13 +187,6 @@
               verbose (member *client-min-messages* '(:info :debug :data))
               debug   (member *client-min-messages* '(:debug :data))
               quiet   (and (not verbose) (not debug)))
-
-        ;; First thing: Self Upgrade?
-        (when self-upgrade
-          (unless *self-upgraded-already*
-            (self-upgrade self-upgrade debug)
-            (let ((*self-upgraded-already* t))
-              (main argv))))
 
         ;; --list-encodings, -E
 	(when list-encodings
