--- a/Makefile
+++ b/Makefile
@@ -70,6 +70,9 @@
 $(LIBRARY): $(OBJECTS)
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) $(LDFLAGS) -o $@ $^
 
+libtermkey.a: $(OBJECTS)
+	$(LIBTOOL) --mode=link $(CC) -o $@ $^
+
 demo: $(LIBRARY) demo.lo
 	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
 
@@ -112,9 +115,10 @@
 	install -d $(DESTDIR)$(LIBDIR)/pkgconfig
 	sed "s,@LIBDIR@,$(LIBDIR),;s,@INCDIR@,$(INCDIR)," <termkey.pc.in >$(DESTDIR)$(LIBDIR)/pkgconfig/termkey.pc
 
-install-lib: $(LIBRARY)
+install-lib: $(LIBRARY) libtermkey.a
 	install -d $(DESTDIR)$(LIBDIR)
 	$(LIBTOOL) --mode=install install libtermkey.la $(DESTDIR)$(LIBDIR)/libtermkey.la
+	$(LIBTOOL) --mode=install install libtermkey.a $(DESTDIR)$(LIBDIR)/libtermkey.a
 
 install-man:
 	install -d $(DESTDIR)$(MAN3DIR)
