diff --git a/Makefile.am b/Makefile.am
index fb601c7..8b68ad6 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -128,11 +128,11 @@ libraft_la_SOURCES = \
 libdqlite_la_SOURCES += $(libraft_la_SOURCES)
 endif # BUILD_RAFT_ENABLED
 
-check_PROGRAMS = unit-test integration-test
+check_PROGRAMS = unit-test
 
 check_LTLIBRARIES = libtest.la
 
-libtest_la_CFLAGS = $(AM_CFLAGS) -DMUNIT_TEST_NAME_LEN=60 -Wno-unknown-warning-option -Wno-unused-result -Wno-conversion -Wno-uninitialized -Wno-maybe-uninitialized -Wno-strict-prototypes -Wno-old-style-definition
+libtest_la_CFLAGS = $(AM_CFLAGS) -DMUNIT_TEST_NAME_LEN=60 -Wno-unused-result -Wno-conversion -Wno-uninitialized -Wno-maybe-uninitialized -Wno-strict-prototypes -Wno-old-style-definition
 libtest_la_SOURCES = \
   test/lib/endpoint.c \
   test/lib/fault.c \
@@ -170,7 +170,7 @@ unit_test_SOURCES += \
   test/unit/test_vfs.c \
   test/unit/test_vfs2.c \
   test/unit/main.c
-unit_test_CFLAGS = $(AM_CFLAGS) -Wno-unknown-warning-option -Wno-uninitialized -Wno-maybe-uninitialized -Wno-float-equal -Wno-conversion
+unit_test_CFLAGS = $(AM_CFLAGS) -Wno-uninitialized -Wno-maybe-uninitialized -Wno-float-equal -Wno-conversion
 unit_test_LDFLAGS = $(AM_LDFLAGS)
 unit_test_LDADD = libtest.la
 
@@ -192,7 +192,6 @@
   raft-core-unit-test \
   raft-core-integration-test \
   raft-uv-unit-test \
-  raft-uv-integration-test \
   raft-core-fuzzy-test
 
 libtest_la_SOURCES += \
