--- mozilla-1.9.2.orig/ipc/chromium/src/base/debug_util_posix.cc
+++ mozilla-1.9.2/ipc/chromium/src/base/debug_util_posix.cc
@@ -6,7 +6,6 @@
 #include "base/debug_util.h"
 
 #include <errno.h>
-#include <execinfo.h>
 #include <fcntl.h>
 #include <stdio.h>
 #include <sys/stat.h>
@@ -114,6 +113,7 @@
 }
 
 StackTrace::StackTrace() {
+#if 0
   const int kMaxCallers = 256;
 
   void* callers[kMaxCallers];
@@ -128,11 +128,14 @@
   } else {
     trace_.resize(0);
   }
+#endif
 }
 
 void StackTrace::PrintBacktrace() {
+#if 0
   fflush(stderr);
   backtrace_symbols_fd(&trace_[0], trace_.size(), STDERR_FILENO);
+#endif
 }
 
 void StackTrace::OutputToStream(std::ostream* os) {
