From 761eba5b9b9c87a872a18b7ba4bf3d77a96e2157 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horv=C3=A1th=20Bal=C3=A1zs?= <q@qroa.ch>
Date: Tue, 15 Oct 2013 01:17:52 +0200
Subject: [PATCH] Fix build with automake-1.12.

---
 configure.ac | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configure.ac b/configure.ac
index f54bb92..9f36932 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4,6 +4,7 @@ AC_PROG_CC
 AC_PROG_INSTALL
 AC_PROG_RANLIB
 AM_PROG_LEX
+m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 AC_PROG_YACC
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_FILES([Makefile doc/Makefile src/Makefile src/lib/Makefile src/tools/Makefile extras/Makefile])
-- 
1.8.4

