Author: Milan P. Stanić <mps@arvanta.net>
Date:   Thu Oct 10 20:30:00 2021 +0000

mupdf 1.19.0 doesn't build libmupdf-third

--- a/meson.build	2021-07-14 16:53:51.000000000 +0000
+++ b/meson.build	2021-10-10 18:16:59.745621832 +0000
@@ -20,7 +20,6 @@ girara = dependency('girara-gtk3')
 glib = dependency('glib-2.0')
 cairo = dependency('cairo')
 mupdf = cc.find_library('mupdf')
-mupdfthird = cc.find_library('mupdf-third')
 
 build_dependencies = [
   zathura,
@@ -28,7 +27,6 @@ build_dependencies = [
   glib,
   cairo,
   mupdf,
-  mupdfthird
 ]
 
 if get_option('link-external')
