--- multipath-tools-0.4.9.orig/libmultipath/prioritizers/datacore.c	2010-05-22 05:01:58.000000000 -0700
+++ multipath-tools-0.4.9/libmultipath/prioritizers/datacore.c	2010-11-28 12:22:12.037228583 -0800
@@ -94,7 +94,7 @@
 	if ((io_hdr.info & SG_INFO_OK_MASK) != SG_INFO_OK)
 		return 0;
 
-	snprintf(vendor, 9, "%.8s\n", inqBuffp + 8);
+	snprintf(vendor, 8, "%.8s\n", inqBuffp + 8);
 	snprintf(product, 17, "%.16s", inqBuffp + 16);
 	snprintf(luname, 21, "%.19s", inqBuffp + 36);
 	snprintf(wwpn, 17, "%.16s", inqBuffp + 96);
--- multipath-tools-0.4.9.orig/libmultipath/prioritizers/hds.c	2010-05-22 05:01:58.000000000 -0700
+++ multipath-tools-0.4.9/libmultipath/prioritizers/hds.c	2010-11-28 12:22:44.199483997 -0800
@@ -123,7 +123,7 @@
 		return -1;
 	}
 
-	snprintf (vendor, 9, "%.8s", inqBuffp + 8);
+	snprintf (vendor, 8, "%.8s", inqBuffp + 8);
 	snprintf (product, 17, "%.16s", inqBuffp + 16);
 	snprintf (serial, 5, "%.4s", inqBuffp + 40);
 	snprintf (ldev, 5, "%.4s", inqBuffp + 44);
