From cf02559aea6a24ef0de2519f78e65420b483e7ac Mon Sep 17 00:00:00 2001
From: Clayton Craft <craftyguy@postmarketos.org>
Date: Wed, 4 Mar 2026 14:03:37 -0800
Subject: [PATCH] parse_config: follow symlinks when checking permissions

fixes #9

Signed-off-by: Clayton Craft <craftyguy@postmarketos.org>
---
 src/main.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.rs b/src/main.rs
index 28b7147..9a94713 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -373,7 +373,7 @@ fn parse_configs(
                     continue;
                 }
 
-                match entry.metadata() {
+                match fs::metadata(entry.path()) {
                     Ok(md) => {
                         if !check_perms(&md) {
                             return Err(
-- 
2.53.0

