From 6309e06441751c82e60785f4021e1ed487c15580 Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Thu, 20 Nov 2025 15:24:14 +0100
Subject: [PATCH] Add systemd system service

---
 hkdm.service | 12 ++++++++++++
 1 file changed, 12 insertions(+)
 create mode 100644 hkdm.service

diff --git a/hkdm.service b/hkdm.service
new file mode 100644
index 0000000..ff0c247
--- /dev/null
+++ b/hkdm.service
@@ -0,0 +1,12 @@
+[Unit]
+Description=Hotkey Daemon for Mobile
+Documentation=https://gitlab.postmarketos.org/postmarketOS/hkdm
+
+[Service]
+Type=simple
+ExecStart=/usr/sbin/hkdm
+Restart=always
+RestartSec=5s
+
+[Install]
+WantedBy=default.target
-- 
GitLab

