--- ./configure.ac.orig	2013-02-28 07:22:41.577340017 +0000
+++ ./configure.ac	2013-02-28 07:24:44.075085706 +0000
@@ -40,7 +40,10 @@
 AC_CHECK_FUNCS([memset setlocale strcasecmp strchr strdup strerror strtol strtod])
 AC_FUNC_STAT
 
-PKG_CHECK_MODULES([MOWGLI], [libmowgli >= 0.4.0], [], [AC_MSG_ERROR([libmowgli 0.4.0 or newer required])])
+PKG_CHECK_MODULES([MOWGLI], [libmowgli-2], [],
+    [PKG_CHECK_MODULES([MOWGLI], [libmowgli >= 0.4.0], [],
+        [AC_MSG_ERROR([libmowgli 0.4.0 or newer required])])
+    ])
 
 dnl gconf
 dnl =====
