diff -upr gpm-1.20.7.orig/src/headers/daemon.h gpm-1.20.7/src/headers/daemon.h
--- gpm-1.20.7.orig/src/headers/daemon.h	2021-07-21 13:08:21.833110930 +0200
+++ gpm-1.20.7/src/headers/daemon.h	2021-07-21 13:08:44.273115400 +0200
@@ -180,7 +180,7 @@ extern struct mouse_features  mouse_tabl
 extern Gpm_Type         mice[];
 extern Gpm_Type         *repeated_type;
 
-time_t                  last_selection_time;
+extern time_t           last_selection_time;
 
 
 
