--- a/src/daemon/open_console.c
+++ b/src/daemon/open_console.c
@@ -22,6 +22,7 @@
 #include <fcntl.h>                  /* open and co.      */
 #include <sys/stat.h>               /* stat()            */
 #include <sys/types.h>              /* major()           */
+#include <sys/sysmacros.h>
 #include <sys/ioctl.h>              /* ioctl             */
 
 /* Linux specific (to be outsourced in gpm2 */
