diff --git a/mono/mini/Makefile.am b/mono/mini/Makefile.am
index 6c840f88b..402a8c8f6 100644
--- a/mono/mini/Makefile.am
+++ b/mono/mini/Makefile.am
@@ -742,7 +742,7 @@ MemoryIntrinsics.dll: MemoryIntrinsics.il
 
 GENMDESC_OPTS=
 
-GENMDESC_PRG=python $(srcdir)/genmdesc.py $(target_define) $(srcdir)
+GENMDESC_PRG=python3 $(srcdir)/genmdesc.py $(target_define) $(srcdir)
 
 cpu-wasm.h: mini-ops.h cpu-wasm.md
 	$(GENMDESC_PRG) cpu-wasm.h wasm_desc $(srcdir)/cpu-wasm.md
diff --git a/mono/mini/Makefile.am.in b/mono/mini/Makefile.am.in
index 6c840f88b..402a8c8f6 100755
--- a/mono/mini/Makefile.am.in
+++ b/mono/mini/Makefile.am.in
@@ -742,7 +742,7 @@ MemoryIntrinsics.dll: MemoryIntrinsics.il
 
 GENMDESC_OPTS=
 
-GENMDESC_PRG=python $(srcdir)/genmdesc.py $(target_define) $(srcdir)
+GENMDESC_PRG=python3 $(srcdir)/genmdesc.py $(target_define) $(srcdir)
 
 cpu-wasm.h: mini-ops.h cpu-wasm.md
 	$(GENMDESC_PRG) cpu-wasm.h wasm_desc $(srcdir)/cpu-wasm.md
diff --git a/mono/mini/Makefile.in b/mono/mini/Makefile.in
index d3555b8c7..8a7e85ad4 100644
--- a/mono/mini/Makefile.in
+++ b/mono/mini/Makefile.in
@@ -1212,7 +1212,7 @@ libmonoincludedir = $(includedir)/mono-$(API_VER)/mono/jit
 libmonoinclude_HEADERS = jit.h
 CSFLAGS = -unsafe -nowarn:0219,0169,0414,0649,0618
 GENMDESC_OPTS = 
-GENMDESC_PRG = python $(srcdir)/genmdesc.py $(target_define) $(srcdir)
+GENMDESC_PRG = python3 $(srcdir)/genmdesc.py $(target_define) $(srcdir)
 LLVM_AOT_RUNTIME_OPTS = $(if $(LLVM),--llvm,)
 GSHAREDVT_RUNTIME_OPTS = $(if $(GSHAREDVT),-O=gsharedvt,)
 fullaot_regtests = $(regtests)
diff --git a/mono/mini/genmdesc.py b/mono/mini/genmdesc.py
index 987332bdf..c3593ba96 100755
--- a/mono/mini/genmdesc.py
+++ b/mono/mini/genmdesc.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 #
 # This tool is used to generate the cpu-<ARCH>.h files used by the JIT. The input is the
diff --git a/scripts/mono-heapviz b/scripts/mono-heapviz
index 27b6f7f60..d19ca33d9 100755
--- a/scripts/mono-heapviz
+++ b/scripts/mono-heapviz
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 
 # Generate a heap visualization for SGen from the heap dump written by
 # mono if the MONO_GC_DEBUG is set to something like

