From 80e45e4d931a6e90d999846b86471cac00d2a6d5 Mon Sep 17 00:00:00 2001
From: Andreas Schwab <schwab@suse.de>
Date: Wed, 21 Feb 2024 22:05:02 +0100
Subject: [PATCH] debug+riscv: use correct comment syntax

Signed-off-by: Andreas Schwab <schwab@suse.de>
---
 debug/+riscv64/getfp.s | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/debug/+riscv64/getfp.s b/debug/+riscv64/getfp.s
index 9c3698af..b822036a 100644
--- a/debug/+riscv64/getfp.s
+++ b/debug/+riscv64/getfp.s
@@ -1,5 +1,5 @@
-// SPDX-License-Identifier: MPL-2.0
-// (c) Hare authors <https://harelang.org>
+# SPDX-License-Identifier: MPL-2.0
+# (c) Hare authors <https://harelang.org>
 
 .section ".text.debug.getfp","ax"
 .global debug.getfp
-- 
2.43.0

