--- src/xvkbd.c.orig
+++ src/xvkbd.c
@@ -3642,10 +3642,7 @@
   icon_pixmap = XCreateBitmapFromData(XtDisplay(w), XtWindow(w),
 				      (char *)xvkbd_icon_bits,
 				      xvkbd_icon_width, xvkbd_icon_height);;
-  icon_mask = XCreateBitmapFromData(XtDisplay(w), XtWindow(w),
-				    (char *)xvkbd_iconmask_bits,
-				    xvkbd_iconmask_width, xvkbd_iconmask_height);
-  XtVaSetValues(w, XtNiconPixmap, icon_pixmap, XtNiconMask, icon_mask, NULL);
+  XtVaSetValues(w, XtNiconPixmap, icon_pixmap, NULL);
 }
 
 /*
