--- ./configure.in.orig	2012-12-31 11:43:05.395168229 +0000
+++ ./configure.in	2012-12-31 11:43:20.171958905 +0000
@@ -3,7 +3,7 @@
 AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADER(config.h)
 
 AC_ARG_WITH( mp4v2,  [  --with-mp4v2            compile libmp4v2],WITHMP4V2=$withval, WITHMP4V2=yes)
 AC_ARG_ENABLE( drm,  [  --enable-drm            Digital Radio Mondiale support], enable_drm=$enableval, enable_drm=no)
