From e035d532bf93b597216fcd11c99f2b8b6ba6eac1 Mon Sep 17 00:00:00 2001
From: Alex Crichton <alex@alexcrichton.com>
Date: Fri, 4 Nov 2016 15:44:46 -0700
Subject: [PATCH] Update makefile to 0.14.0

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 98d5e91c8053..08dc61626ee1 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -1,4 +1,4 @@
-CFG_RELEASE_NUM=0.13.0
+CFG_RELEASE_NUM=0.14.0
 CFG_RELEASE_LABEL=
 
 OPENSSL_VERS=1.0.2j
-- 
1.8.3.1

