--- a/spec/util_spec.moon
+++ b/spec/util_spec.moon
@@ -160,7 +160,7 @@
       util.to_json {
         items: setmetatable {}, json.array_mt
       }
-    '{"items":[]}'
+    '{"items":{}}'
   }
 
   { -- encoding non-empty array with array_mt
@@ -174,7 +174,7 @@
   { -- encoding top-level empty array with array_mt
     ->
       util.to_json setmetatable({}, json.array_mt)
-    '[]'
+    '{}'
   }
 
   {
