Use symlinks instead of hardlinks

--- a/uip/Makefile.in
+++ b/uip/Makefile.in
@@ -35,7 +35,7 @@
 
 COMPILE = $(CC) -c $(DEFS) -DVERSION="\"$(VERSION)\"" $(INCLUDES) $(CFLAGS)
 LINK    = $(CC) $(LDFLAGS) -o $@
-LN = ln
+LN = ln -s
 
 INSTALL         = @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
@@ -250,14 +250,14 @@
 	rm -f $(DESTDIR)$(bindir)/prev
 	rm -f $(DESTDIR)$(bindir)/next
 	rm -f $(DESTDIR)$(bindir)/scan
-	$(LN) $(DESTDIR)$(bindir)/flist  $(DESTDIR)$(bindir)/flists
-	$(LN) $(DESTDIR)$(bindir)/folder $(DESTDIR)$(bindir)/folders
-	$(LN) $(DESTDIR)$(bindir)/new    $(DESTDIR)$(bindir)/fnext
-	$(LN) $(DESTDIR)$(bindir)/new    $(DESTDIR)$(bindir)/fprev
-	$(LN) $(DESTDIR)$(bindir)/new    $(DESTDIR)$(bindir)/unseen
-	$(LN) $(DESTDIR)$(bindir)/show   $(DESTDIR)$(bindir)/prev
-	$(LN) $(DESTDIR)$(bindir)/show   $(DESTDIR)$(bindir)/next
-	$(LN) $(DESTDIR)$(bindir)/pick   $(DESTDIR)$(bindir)/scan
+	$(LN) flist  $(DESTDIR)$(bindir)/flists
+	$(LN) folder $(DESTDIR)$(bindir)/folders
+	$(LN) new    $(DESTDIR)$(bindir)/fnext
+	$(LN) new    $(DESTDIR)$(bindir)/fprev
+	$(LN) new    $(DESTDIR)$(bindir)/unseen
+	$(LN) show   $(DESTDIR)$(bindir)/prev
+	$(LN) show   $(DESTDIR)$(bindir)/next
+	$(LN) pick   $(DESTDIR)$(bindir)/scan
 
 # install misc support binaries
 install-misc:
