--- a/lib/default_theme.json
+++ b/lib/default_theme.json
@@ -179,7 +179,7 @@
     fg_color_selected: "red",
     fill: true,
     focusable: true,
-    font_desc: { family:"Bitstream Vera Sans Mono", size:14 },
+    font_desc: { family:"DejaVu Sans Mono", size:14 },
     cursor_width: 2,
     cursor_color: "pink",
     active_cursor_color: "0xff0000cc",
--- a/lib/font.ml
+++ b/lib/font.ml
@@ -74,7 +74,7 @@
 let font_exts = ref [".ttf"]
 let font_dirs = ref
   [ Filename.current_dir_name, false ;
-    "/usr/share/fonts/truetype", true ;
+    "/usr/share/fonts", true ;
   ]
 
 let fonts = ref SMap.empty
--- a/lib/font.mli
+++ b/lib/font.mli
@@ -85,7 +85,7 @@
   {!App.init} will call {!val-load_fonts} which uses [font_dirs] to look
   for available fonts,
   so additional directories must be set before initializing application.
-  Default value is [[ Filename.current_dir_name, false ; "/usr/share/fonts/truetype", true ]].
+  Default value is [[ Filename.current_dir_name, false ; "/usr/share/fonts", true ]].
 *)
 val font_dirs : (string * bool) list ref
 
