diff --git a/vendor/src/github.com/spacemonkeygo/openssl/ctx.go b/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
index 22d6dd1..894fecd 100644
--- a/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
+++ b/vendor/src/github.com/spacemonkeygo/openssl/ctx.go
@@ -140,8 +140,8 @@ const (
 func NewCtxWithVersion(version SSLVersion) (*Ctx, error) {
 	var method *C.SSL_METHOD
 	switch version {
-	case SSLv3:
-		method = C.SSLv3_method()
+//	case SSLv3:
+//		method = C.SSLv3_method()
 	case TLSv1:
 		method = C.TLSv1_method()
 	case TLSv1_1:
diff --git a/vendor/src/github.com/spacemonkeygo/openssl/hostname.c b/vendor/src/github.com/spacemonkeygo/openssl/hostname.c
index 7ebea17..41c5951 100644
--- a/vendor/src/github.com/spacemonkeygo/openssl/hostname.c
+++ b/vendor/src/github.com/spacemonkeygo/openssl/hostname.c
@@ -346,7 +346,7 @@ static int do_x509_check(X509 *x, const unsigned char *chk, size_t chklen,
 	return 0;
 	}
 
-#if OPENSSL_VERSION_NUMBER < 0x1000200fL
+#if OPENSSL_VERSION_NUMBER < 0x1000200fL || defined(LIBRESSL_VERSION_NUMBER)
 
 int X509_check_host(X509 *x, const unsigned char *chk, size_t chklen,
 					unsigned int flags)
