--- a/Makefile.in
+++ b/Makefile.in
@@ -80,7 +80,7 @@ LDFLAGS.libfossil.external = \
 LDFLAGS.libfossil = \
   $(LDFLAGS.core) $(LDFLAGS.rpath) \
   $(LDFLAGS.iconv) $(LDFLAGS.libfossil.external) $(LDFLAGS.configure)
-LDFLAGS.libfossil.soname = # TODO: via configure script, a-la sqlite's handling
+LDFLAGS.libfossil.soname = -Wl,-soname,$(libfossil.dll).$(libfossil.dll.version)
 LDFLAGS.libfossil.os-specific = \
   @LDFLAGS_OUT_IMPLIB@
 
--- a/auto.def
+++ b/auto.def
@@ -43,7 +43,7 @@ options {}
 #    prefix:=[get-env HOME /usr/local] -> "Installation prefix."
 
 define FSL_PACKAGE_NAME "libfossil"
-define FSL_LIBRARY_VERSION 0.5-ish
+define FSL_LIBRARY_VERSION 0.5.0
 
 ########################################################################
 # See if we can get the fossil schema version from the current
