--- a/auto.def
+++ b/auto.def
@@ -149,13 +149,13 @@ if {[opt-bool amal]} {
       HAVE_CLOCK_GETTIME 1 0
     }] {
       if {$valUnix eq $valWin || $notOnWindows eq $valWin} {
-        puts $out -nonewline [subst -nocommands \
+        puts -nonewline $out [subst -nocommands \
 {#if !defined($def)
 #  define $def $valUnix
 #endif
 }]
       } else {
-        puts $out -nonewline [subst -nocommands \
+        puts -nonewline $out [subst -nocommands \
 {#if defined(_WIN32)
 #  if !defined($def)
 #    define $def $valWin
