diff --git a/src/sc.c b/src/sc.c
index 680fa91..867f430 100644
--- a/src/sc.c
+++ b/src/sc.c
@@ -512,6 +512,7 @@ static int fsl_sc_popen_xRead(fsl_sc * const ch, fsl_buffer * tgt, fsl_int_t how
       assert( fp ? (fsl_sc_popen_e_filebuf == st->response.mode)
               : (fsl_sc_popen_e_filebuf != st->response.mode) );
       assert( fp ? fdIn<0 : fdIn>0 );
+      buf[0] = 0;
 
       switch( howMuch ){
 
