From bfda0d8509d3ce9e3e5e2b2744b9aa12fcd01f40 Mon Sep 17 00:00:00 2001
From: 6543 <6543@obermui.de>
Date: Sun, 21 Jan 2024 23:37:16 +0100
Subject: [PATCH] we use the prebuild webui

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0b0f700f5..3d41d0193 100644
--- a/Makefile
+++ b/Makefile
@@ -187,7 +187,7 @@ test: test-agent test-server test-server-datastore test-cli test-lib test-ui ##
 build-ui: ## Build UI
 	(cd web/; pnpm install --frozen-lockfile; pnpm build)
 
-build-server: build-ui generate-swagger ## Build server
+build-server: generate-swagger ## Build server
 	CGO_ENABLED=${CGO_ENABLED} GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -ldflags '${LDFLAGS}' -o dist/woodpecker-server${BIN_SUFFIX} go.woodpecker-ci.org/woodpecker/v2/cmd/server
 
 build-agent: ## Build agent
-- 
2.43.0

