# Set cgo_enabled during swagger generation
--- a/Makefile
+++ b/Makefile
@@ -109,11 +109,11 @@
 
 .PHONY: generate
 generate: install-tools generate-swagger ## Run all code generations
-	CGO_ENABLED=0 go generate ./...
+	CGO_ENABLED=1 go generate ./...
 
 generate-swagger: install-tools ## Run swagger code generation
 	swag init -g server/api/ -g cmd/server/swagger.go --outputTypes go -output cmd/server/docs
-	CGO_ENABLED=0 go generate cmd/server/swagger.go
+	CGO_ENABLED=1 go generate cmd/server/swagger.go
 
 generate-license-header: install-tools
 	addlicense -c "Woodpecker Authors" -ignore "vendor/**" **/*.go
