diff --git a/Makefile b/Makefile
index a20af51..da4ca9f 100644
--- a/Makefile
+++ b/Makefile
@@ -174,7 +174,7 @@ test: test-agent test-server test-server-datastore test-cli test-lib test-ui ##
 ##@ Build
 
 build-ui: ## Build UI
-	(cd web/; pnpm install --frozen-lockfile; pnpm build)
+	(cd web/; pnpm install --frozen-lockfile; printf "const { parse, xxhashBase64Url } = require('@rollup/wasm-node/dist/wasm-node/bindings_wasm.js');\nexports.parse = parse;\nexports.parseAsync = async (code, allowReturnOutsideFunction, _signal) =>\nparse(code, allowReturnOutsideFunction);\nexports.xxhashBase64Url = xxhashBase64Url;\n" > node_modules/.pnpm/rollup@4.5.0/node_modules/rollup/dist/native.js; pnpm build)
 
 build-server: build-ui generate-swagger ## Build server
 	CGO_ENABLED=${CGO_ENABLED} GOOS=${TARGETOS} GOARCH=${TARGETARCH} go build -ldflags '${LDFLAGS}' -o dist/woodpecker-server${BIN_SUFFIX} go.woodpecker-ci.org/woodpecker/cmd/server
diff --git a/web/package.json b/web/package.json
index cfbcdee..ab83fc8 100644
--- a/web/package.json
+++ b/web/package.json
@@ -19,6 +19,7 @@
   "dependencies": {
     "@intlify/unplugin-vue-i18n": "^1.0.0",
     "@kyvg/vue3-notification": "^3.0.0",
+    "@rollup/wasm-node": "4.5.1",
     "@vueuse/core": "^10.5.0",
     "ansi_up": "^6.0.0",
     "dayjs": "^1.11.9",
diff --git a/web/pnpm-lock.yaml b/web/pnpm-lock.yaml
index ce8bef2..dd36ab8 100644
--- a/web/pnpm-lock.yaml
+++ b/web/pnpm-lock.yaml
@@ -14,6 +14,9 @@ dependencies:
   '@kyvg/vue3-notification':
     specifier: ^3.0.0
     version: 3.0.2(vue@3.3.8)
+  '@rollup/wasm-node':
+    specifier: 4.5.1
+    version: 4.5.1
   '@vueuse/core':
     specifier: ^10.5.0
     version: 10.6.1(vue@3.3.8)
@@ -902,6 +905,14 @@ packages:
     dev: true
     optional: true
 
+  /@rollup/wasm-node@4.5.1:
+    resolution: {integrity: sha512-bPkGcyeyz/iuvtzfg5iMy3lszRRkSSmFRru03moBGGKmKLSDWXMdutVAuYOLb4yN4TOpn1ewpqgS4o3iUCPi2Q==}
+    engines: {node: '>=18.0.0', npm: '>=8.0.0'}
+    hasBin: true
+    optionalDependencies:
+      fsevents: 2.3.3
+    dev: false
+
   /@sindresorhus/is@6.1.0:
     resolution: {integrity: sha512-BuvU07zq3tQ/2SIgBsEuxKYDyDjC0n7Zir52bpHy2xnBbW81+po43aLFPLbeV3HRAheFbGud1qgcqSYfhtHMAg==}
     engines: {node: '>=16'}
