Ref: https://github.com/pdm-project/pdm-backend/issues/303
---
diff -rupN a/tests/test_api.py b/tests/test_api.py
--- a/tests/test_api.py	2025-04-07 00:47:51.000000000 +0000
+++ b/tests/test_api.py	2026-04-22 03:07:20.780000000 +0000
@@ -299,7 +299,7 @@ def test_build_editable(dist: Path, fixt
     assert api.get_requires_for_build_editable() == []
     with zipfile.ZipFile(dist / wheel_name) as zf:
         namelist = zf.namelist()
-        assert "demo_package.pth" in namelist
+        assert "_editable_impl_demo_package.pth" in namelist
         assert "_editable_impl_demo_package.py" in namelist
         assert "demo_package-0.1.0.dist-info/licenses/LICENSE" in namelist
 
@@ -308,7 +308,7 @@ def test_build_editable(dist: Path, fixt
         )
         assert "editables" in metadata.get_all("Requires-Dist", [])
 
-        pth_content = zf.read("demo_package.pth").decode("utf-8").strip()
+        pth_content = zf.read("_editable_impl_demo_package.pth").decode("utf-8").strip()
         assert pth_content == "import _editable_impl_demo_package"
 
         proxy_module = zf.read("_editable_impl_demo_package.py").decode("utf-8").strip()
@@ -327,14 +327,14 @@ def test_build_editable_src(dist: Path,
 
     with zipfile.ZipFile(dist / wheel_name) as zf:
         namelist = zf.namelist()
-        assert "demo_package.pth" in namelist
+        assert "_editable_impl_demo_package.pth" in namelist
         assert "_editable_impl_demo_package.py" in namelist
         assert "my_package/data.json" not in namelist, (
             "data files in proxy modules are excluded"
         )
         assert "data_out.json" in namelist
 
-        pth_content = zf.read("demo_package.pth").decode("utf-8").strip()
+        pth_content = zf.read("_editable_impl_demo_package.pth").decode("utf-8").strip()
         assert pth_content == "import _editable_impl_demo_package"
 
         proxy_module = zf.read("_editable_impl_demo_package.py").decode("utf-8").strip()
@@ -357,7 +357,7 @@ def test_build_editable_pep420(dist: Pat
 
     with zipfile.ZipFile(dist / wheel_name) as zf:
         namelist = zf.namelist()
-        assert "demo_package.pth" in namelist
+        assert "_editable_impl_demo_package.pth" in namelist
         assert "__editables_demo_package.py" not in namelist
 
         metadata = email.message_from_bytes(
@@ -365,7 +365,7 @@ def test_build_editable_pep420(dist: Pat
         )
         assert "editables" not in metadata.get_all("Requires-Dist", [])
 
-        pth_content = zf.read("demo_package.pth").decode("utf-8").strip()
+        pth_content = zf.read("_editable_impl_demo_package.pth").decode("utf-8").strip()
         assert pth_content == str(fixture_project.resolve())
 
 
