--- a/runtimes/CMakeLists.txt
+++ b/runtimes/CMakeLists.txt
@@ -171,8 +171,6 @@
 find_package(Python3 REQUIRED COMPONENTS Interpreter)
 
 # Host triple is used by tests to check if they are running natively.
-include(GetHostTriple)
-get_host_triple(LLVM_HOST_TRIPLE)
 message(STATUS "LLVM host triple: ${LLVM_HOST_TRIPLE}")
 
 # TODO: We shouldn't be using LLVM_DEFAULT_TARGET_TRIPLE for runtimes since we
