diff -rupN a/internal/github/client.go b/internal/github/client.go
--- a/internal/github/client.go	2024-09-14 20:39:16.000000000 +0000
+++ b/internal/github/client.go	2026-03-09 14:34:54.120000000 +0000
@@ -264,7 +264,7 @@ func (g *Client) GetRepoTarball(ctx cont
 		gopts.Ref = *opts.Ref
 	}
 
-	link, _, err := g.client.Repositories.GetArchiveLink(ctx, owner, name, github.Tarball, &gopts, true)
+	link, _, err := g.client.Repositories.GetArchiveLink(ctx, owner, name, github.Tarball + "/", &gopts, true)
 	if err != nil {
 		return nil, "", err
 	}
