--- fortune-0.1-origin/fortune/fortune.c
+++ fortune-0.1/fortune/fortune.c
@@ -33,6 +33,10 @@
  * SUCH DAMAGE.
  */
 
+/* Enable ntohl() and arc4random_uniform */
+#include <arpa/inet.h>
+#include <bsd/stdlib.h>
+
 #include <sys/stat.h>
 #include <sys/types.h>
 
--- fortune-0.1-origin/strfile/strfile.c
+++ fortune-0.1/strfile/strfile.c
@@ -32,6 +32,10 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
+/* Enable htonl() and arc4random_uniform() */
+#include <arpa/inet.h>
+#include <bsd/stdlib.h>
+
 #include	<sys/types.h>
 
 #include	<ctype.h>
