From ea94e97a58538090f65fae3b94395e5c08d4b8ee Mon Sep 17 00:00:00 2001
From: Chris Allan <chrisallan@pm.me>
Date: Sat, 17 Apr 2021 20:15:58 +0000
Subject: [PATCH] Fix incompatible types

---
 src/gulkan-texture.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gulkan-texture.c b/src/gulkan-texture.c
index 3fc5bd6..b06fcf3 100644
--- a/src/gulkan-texture.c
+++ b/src/gulkan-texture.c
@@ -47,7 +47,7 @@ gulkan_texture_init (GulkanTexture *self)
   self->image = VK_NULL_HANDLE;
   self->image_memory = VK_NULL_HANDLE;
   self->image_view = VK_NULL_HANDLE;
-  self->format = VK_NULL_HANDLE;
+  self->format = VK_FORMAT_UNDEFINED;
   self->mip_levels = 1;
 }
 
-- 
GitLab

