diff --git a/src/gulkan-descriptor-set.c b/src/gulkan-descriptor-set.c
index c2051db..6d8ffb0 100644
--- a/src/gulkan-descriptor-set.c
+++ b/src/gulkan-descriptor-set.c
@@ -27,7 +27,7 @@ G_DEFINE_TYPE (GulkanDescriptorSet, gulkan_descriptor_set, G_TYPE_OBJECT)
 static void
 gulkan_descriptor_set_init (GulkanDescriptorSet *self)
 {
-  self->pool = NULL;
+  self->pool = 0;
   self->handle = VK_NULL_HANDLE;
   self->descriptors = NULL;
   self->context = NULL;
diff --git a/src/gulkan-frame-buffer.c b/src/gulkan-frame-buffer.c
index f613285..7e601a3 100644
--- a/src/gulkan-frame-buffer.c
+++ b/src/gulkan-frame-buffer.c
@@ -326,7 +326,7 @@ _init_frame_buffer (GulkanFrameBuffer *self, GulkanRenderPass *render_pass)
     .attachmentCount = self->use_depth ? 2 : 1,
     .pAttachments = (VkImageView[]) {
       self->color_image_view,
-      self->use_depth ? self->depth_stencil_image_view : NULL,
+      self->use_depth ? self->depth_stencil_image_view : 0,
     },
     .width = self->extent.width,
     .height = self->extent.height,
