diff --git a/bite/io.py b/bite/io.py
index 63ebf2c..baf8a60 100644
--- a/bite/io.py
+++ b/bite/io.py
@@ -1,7 +1,7 @@
 from asyncio import IncompleteReadError, StreamReader
 from typing import Union
 
-from typing_extensions import Protocol
+from typing import Protocol
 
 
 def _copy_doc(source):
diff --git a/bite/parsers.py b/bite/parsers.py
index 98f7bbc..86ce2e3 100644
--- a/bite/parsers.py
+++ b/bite/parsers.py
@@ -12,7 +12,7 @@ from typing import (
     Union,
 )
 
-from typing_extensions import Protocol
+from typing import Protocol
 
 from bite.io import ParserBuffer
 
