diff --git a/cmd/plakar/subcommands/create/create_test.go b/cmd/plakar/subcommands/create/create_test.go
index 289f76c..bf0e92e 100644
--- a/cmd/plakar/subcommands/create/create_test.go
+++ b/cmd/plakar/subcommands/create/create_test.go
@@ -117,9 +117,9 @@ func TestExecuteCmdCreateDefaultWeakPassword(t *testing.T) {
 	// Duplicate the tty file descriptor to syscall.Stdin (fd 0)
 	originalStdin, err := syscall.Dup(syscall.Stdin)
 	require.NoError(t, err)
-	defer syscall.Dup2(originalStdin, syscall.Stdin)
+	defer syscall.Dup3(originalStdin, syscall.Stdin,0)
 
-	err = syscall.Dup2(int(tty.Fd()), syscall.Stdin)
+	err = syscall.Dup3(int(tty.Fd()), syscall.Stdin,0)
 	require.NoError(t, err)
 
 	ctx := appcontext.NewAppContext()
