From 1d841c8e5ab36b932d9e51a1e871539451786632 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Sat, 1 May 2021 16:21:51 +0200
Subject: [PATCH] remove rpath

---
 CMakeLists.txt | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 359114e..eebf833 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,19 +140,6 @@ elseif(${CMAKE_CXX_COMPILER_ID} MATCHES "Intel")
     set(CMAKE_COMPILER_IS_ICC 1)
 endif()
 
-
-if (NOT CMAKE_COMPILER_IS_ICC)
-    # Currently icc has a bug that asserts when linking rpaths containing long
-    # sequences of ':' that this command causes. The consequence is that examples
-    # built and installed using icc will not have an rpath pointing to the built
-    # OSD library which they depend on and will have to set LD_LIBRARY_PATH instead.
-    list(APPEND CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-endif()
-
-# add the automatically determined parts of the RPATH
-# which point to directories outside the build tree to the install RPATH
-SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-
 # ensure that ARC is shown as enabled in the Xcode UI
 if(CMAKE_GENERATOR STREQUAL "Xcode")
     set (CMAKE_XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC "YES")
-- 
2.31.1

