Author: Andreas Beckmann <anbe@debian.org>
Description: fix building with GCC 15

--- a/Makefile.am
+++ b/Makefile.am
@@ -91,7 +91,7 @@
        export CK_FORK;
 TESTS_ENVIRONMENT              ?= ${AM_TESTS_ENVIRONMENT}

-tests/check_%: CFLAGS += ${CHECK_CFLAGS} ${GNUTLS_CFLAGS} ${WOLFSSL_CFLAGS}
+tests/check_%: CFLAGS += ${CHECK_CFLAGS} ${GNUTLS_CFLAGS} ${WOLFSSL_CFLAGS} -std=gnu17
 tests/check_%: LDADD += ${CHECK_LIBS} ${DL_LIBS}
 tests/check_%: LDFLAGS += -no-install
 
