Patch-Related: https://github.com/php/php-src/issues/19925
--- a/ext/curl/tests/curl_setopt_ssl.phpt
+++ b/ext/curl/tests/curl_setopt_ssl.phpt
@@ -101,7 +101,7 @@
     echo "\n";
     echo "case 3: client cert and empty key from string\n";
     $ch = curl_init("https://127.0.0.1:$port/");
-    var_dump(curl_setopt($ch, CURLOPT_SSLCERT_BLOB, $clientCert));
+    var_dump(curl_setopt($ch, CURLOPT_SSLCERT_BLOB, ''));
     var_dump(curl_setopt($ch, CURLOPT_SSLKEY_BLOB, ''));
     var_dump(curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false));
     var_dump(curl_setopt($ch, CURLOPT_SSL_VERIFYHOST, false));
