diff --git a/utils/Makefile b/utils/Makefile
index 11509d4..c32ff10 100644
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -108,9 +108,9 @@ install: $(INSTALL_DEP)
 ifndef DISABLE_JOYSTICK
 	install -d $(DESTDIR)$(PREFIX)/share/joystick
 	install extract filter ident $(DESTDIR)$(PREFIX)/share/joystick
-	install -d $(DESTDIR)/lib/udev/rules.d
-	install js-set-enum-leds $(DESTDIR)/lib/udev
-	install -m 644 80-stelladaptor-joystick.rules $(DESTDIR)/lib/udev/rules.d
+	install -d $(DESTDIR)$(PREFIX)/lib/udev/rules.d
+	install js-set-enum-leds $(DESTDIR)$(PREFIX)/lib/udev
+	install -m 644 80-stelladaptor-joystick.rules $(DESTDIR)$(PREFIX)/lib/udev/rules.d
 endif
 
 .PHONY: compile clean distclean install
