--- a/crates/zed/resources/zed.desktop.in
+++ b/crates/zed/resources/zed.desktop.in
@@ -1,13 +1,13 @@
 [Desktop Entry]
 Version=1.0
 Type=Application
-Name=$APP_NAME
+Name=Zed
 GenericName=Text Editor
 Comment=A high-performance, multiplayer code editor.
 TryExec=zed
-StartupNotify=$DO_STARTUP_NOTIFY
-Exec=zed $APP_ARGS
-Icon=$APP_ICON
+StartupNotify=true
+Exec=/usr/bin/zed-editor
+Icon=zed
 Categories=Utility;TextEditor;Development;IDE;
 Keywords=zed;
 MimeType=text/plain;inode/directory;
