diff --git a/src/getopt.h b/src/getopt.h
index 305a5ad..96db3ce 100644
--- a/src/getopt.h
+++ b/src/getopt.h
@@ -144,7 +144,7 @@ struct option
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int __argc, char *const *__argv, const char *__shortopts);
 # else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+/* Use <unistd.h> for system declaration */
 # endif /* __GNU_LIBRARY__ */
 
 # ifndef __need_getopt
