From 47cf8b2caea0d9b42c1839eee9e09c92d1ce5f31 Mon Sep 17 00:00:00 2001
From: Jim Newsome <jnewsome@torproject.org>
Date: Sun, 7 Aug 2022 15:46:32 -0500
Subject: [PATCH] configure.ac: update version to 2.4.0

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 14c10b6..db11118 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,7 +3,7 @@
 ##############################################################################
 
 # Process this file with autoconf to produce a configure script.
-AC_INIT([torsocks], [2.3.0],[dgoulet@torproject.org],[],[https://torproject.org])
+AC_INIT([torsocks], [2.4.0],[dgoulet@torproject.org],[],[https://torproject.org])
 AC_CONFIG_AUX_DIR([config])
 AC_CANONICAL_TARGET
 # Get hostname and other information.
-- 
2.46.0

