--- a/rbs.gemspec
+++ b/rbs.gemspec
@@ -27,9 +27,7 @@
 
   # Specify which files should be added to the gem when it is released.
   # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features|bin)/}) }
-  end
+  spec.files         = Dir["core/**/*.rbs", "exe/*", "lib/**/*.rb", "schema/**/*.json", "sig/**/*.rbs", "stdlib/**/*.rbs"]
   spec.files << "lib/rbs/parser.rb"
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
