--- a/rbs.gemspec
+++ b/rbs.gemspec
@@ -26,15 +26,7 @@
   end
 
   # Specify which files should be added to the gem when it is released.
-  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject do |f|
-      [
-        %r{^(test|spec|features|bin|steep|benchmark)/},
-        /Gemfile/
-      ].any? {|r| f.match(r) }
-    end
-  end
+  spec.files         = Dir["core/**/*", "exe/*", "ext/**/*", "include/**/*.h", "lib/**/*.rb", "schema/**/*.json", "sig/**/*", "src/**/*", "stdlib/**/*"]
   spec.extensions = %w{ext/rbs_extension/extconf.rb}
 
   if false
