--- a/rbs.gemspec
+++ b/rbs.gemspec
@@ -29,4 +29 @@
-  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features|bin)/}) }
-  end
+  spec.files         = Dir["core/**/*.rbs", "exe/*", "ext/**/*", "lib/**/*.rb", "schema/**/*.json", "sig/**/*.rbs", "stdlib/**/*.rbs"]
