diff --git a/meson.build b/meson.build
index 448e262..a717fca 100644
--- a/meson.build
+++ b/meson.build
@@ -100,7 +100,6 @@ sources_dir = include_directories('source/')
 gtkd_dep = dependency('gtkd-3', version: '>=3.8.5')
 vted_dep = dependency('vted-3', version: '>=3.8.5')
 xlib_dep = dependency('x11')
-libunwind_dep = dependency('libunwind')
 libsecret_dep = dependency('libsecret-1', required: false)
 
 subdir('po')
@@ -113,7 +112,6 @@ executable('tilix',
     dependencies : [gtkd_dep,
                     vted_dep,
                     xlib_dep,
-                    libunwind_dep,
                     libsecret_dep],
     d_args: d_extra_args,
     d_module_versions: ['StdLoggerDisableTrace'],
