--- a/data/meson.build
+++ b/data/meson.build
@@ -98,7 +98,7 @@ conf.set('bindir', join_paths(get_option('prefix'), get_option('bindir')))
 
 configure_file(
   input: 'dbus/@0@.service.in'.format(meson.project_name()),
-  output: 'dbus/@0@.service'.format(meson.project_name()),
+  output: '@0@.service'.format(meson.project_name()),
   configuration: conf,
   install_dir: join_paths(get_option('datadir'), 'dbus-1/services')
 )
