From 9f85113234ef3a054234474d955f95bc7c949edb Mon Sep 17 00:00:00 2001
From: Phillip Lougher <phillip@squashfs.org.uk>
Date: Tue, 14 Sep 2021 05:41:27 +0100
Subject: [PATCH 5/5] Unsquashfs: Add makefile entry for unsquash-12.o

Signed-off-by: Phillip Lougher <phillip@squashfs.org.uk>
---
 squashfs-tools/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/squashfs-tools/Makefile b/squashfs-tools/Makefile
index 417f2ba..95926c0 100755
--- a/squashfs-tools/Makefile
+++ b/squashfs-tools/Makefile
@@ -393,7 +393,7 @@ unsquash-34.o: unsquashfs.h unsquash-34.c unsquashfs_error.h
 
 unsquash-1234.o: unsquash-1234.c unsquashfs_error.h
 
-unsquash-1234.o: unsquash-12.c
+unsquash-12.o: unsquash-12.c unsquashfs.h
 
 unsquashfs_xattr.o: unsquashfs_xattr.c unsquashfs.h squashfs_fs.h xattr.h unsquashfs_error.h
 
-- 
2.33.1

