--- a/squashfs-tools/mksquashfs.c	Wed Dec  2 10:30:31 2009
+++ b/squashfs-tools/mksquashfs.c	Wed Dec  2 10:31:34 2009
@@ -3705,7 +3705,7 @@
 			processors = 1;
 		}
 #else
-		processors = get_nprocs();
+		processors = sysconf(_SC_NPROCESSORS_ONLN);
 #endif
 	}
 
--- a/squashfs-tools/unsquashfs.c	Wed Dec  2 10:30:31 2009
+++ b/squashfs-tools/unsquashfs.c	Wed Dec  2 10:31:34 2009
@@ -1811,7 +1811,7 @@
 			processors = 1;
 		}
 #else
-		processors = get_nprocs();
+		processors = sysconf(_SC_NPROCESSORS_ONLN);
 #endif
 	}
 
