Patch-Source: https://github.com/KuabeM/piow/pull/2
--
From 6dccb4a4120c5bb17f9c53e6427cb1c8c26e7f91 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 12 Feb 2022 17:34:22 +0100
Subject: [PATCH] Add support for system-level config

diff --git a/src/config.rs b/src/config.rs
index e8e3702..ce0787c 100644
--- a/src/config.rs
+++ b/src/config.rs
@@ -36,17 +36,25 @@ impl Default for Config {
 
 impl Config {
     /// Load the configuration file from either `path` or the default config dir
-    /// `${XDG_CONFIG_HOME}`.
+    /// `${XDG_CONFIG_HOME}` or `/etc/xdg`.
     ///
     /// Error: File does not exist or can't be opened, syntax errors and other parsing errors.
     pub fn load(path: Option<PathBuf>) -> Result<Self, Error> {
         let cfg_path: PathBuf = if let Some(p) = path {
             p
         } else {
-            dirs::config_dir()
+            let user_path = dirs::config_dir()
                 .ok_or_else(|| format_err!("Can't access default config dir."))?
                 .join(env!("CARGO_PKG_NAME"))
-                .join("config.toml")
+                .join("config.toml");
+            let sys_path =
+                PathBuf::from(format!("/etc/xdg/{}/config.toml", env!("CARGO_PKG_NAME")));
+
+            if !user_path.exists() && sys_path.exists() {
+                sys_path
+            } else {
+                user_path
+            }
         };
         debug!("Loading config file '{}'", &cfg_path.to_string_lossy());
         let mut f = std::fs::File::open(cfg_path)?;
diff --git a/src/main.rs b/src/main.rs
index 4dec97a..70b6a41 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -22,7 +22,8 @@ Usage:
 Options:
   -h --help            Show this screen.
   --version            Print the version and exit.
-  --config=<cfg>       Path to config file. Defaults to $XDG_CONFIG_HOME or $HOME/.config.
+  --config=<cfg>       Path to config file. Defaults to piow/config.toml under
+                        $XDG_CONFIG_HOME, $HOME/.config, or /etc/xdg.
   --syslog             Send log messages to syslog instead of stdout.
 "#;
 
