From 72510f27b10889e119b42804835e8beaf70bc18d Mon Sep 17 00:00:00 2001
From: Andrej Kacian <ticho@claws-mail.org>
Date: Thu, 8 Jan 2015 22:51:04 +0100
Subject: [PATCH] RSSyl: Day-of-week name in RFC822 timestamps is optional.

---
 src/plugins/rssyl/libfeed/date.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/plugins/rssyl/libfeed/date.c b/src/plugins/rssyl/libfeed/date.c
index c37756d..92f5a86 100644
--- a/src/plugins/rssyl/libfeed/date.c
+++ b/src/plugins/rssyl/libfeed/date.c
@@ -132,8 +132,11 @@ time_t parseRFC822Date(gchar *date)
 	if (c != NULL)
 		setlocale(LC_TIME, "C");
 
-	if (!strptime(date, "%a, %d %b %Y %H:%M:%S %Z", &t) &&
-		!strptime(date, "%a, %d %b %Y %H:%M %Z", &t)) {
+	if (date != NULL &&
+			!strptime(date, "%a, %d %b %Y %H:%M:%S %Z", &t) &&
+			!strptime(date, "%a, %d %b %Y %H:%M %Z", &t) &&
+			!strptime(date, "%d %b %Y %H:%M:%S %Z", &t) &&
+			!strptime(date, "%d %b %Y %H:%M %Z", &t)) {
 		g_warning("Invalid RFC822 date!\n");
 		if (c != NULL)
 			setlocale(LC_TIME, c);
-- 
1.7.10.4

