--- omxplayer-6ee9a0a5488119786990b6f91990d59d6e35427c/Makefile.orig
+++ omxplayer-6ee9a0a5488119786990b6f91990d59d6e35427c/Makefile
@@ -48,7 +48,7 @@
 omxplayer.o: help.h keys.h
 
 version:
-	bash gen_version.sh > version.h 
+	sh gen_version.sh > version.h 
 
 omxplayer.bin: version $(OBJS)
 	$(CXX) $(LDFLAGS) -o omxplayer.bin $(OBJS) -lvchiq_arm -lvcos -ldbus-1 -lrt -lpthread -lavutil -lavcodec -lavformat -lswscale -lswresample -lpcre
@@ -79,14 +79,12 @@
 	make -f Makefile.ffmpeg
 	make -f Makefile.ffmpeg install
 
-dist: omxplayer.bin omxplayer.1
+install: omxplayer.bin omxplayer.1
 	mkdir -p $(DIST)/usr/lib/omxplayer
 	mkdir -p $(DIST)/usr/bin
 	mkdir -p $(DIST)/usr/share/doc/omxplayer
 	mkdir -p $(DIST)/usr/share/man/man1
-	cp omxplayer omxplayer.bin $(DIST)/usr/bin
+	cp omxplayer.bin $(DIST)/usr/bin/omxplayer
 	cp COPYING $(DIST)/usr/share/doc/omxplayer
 	cp README.md $(DIST)/usr/share/doc/omxplayer/README
 	cp omxplayer.1 $(DIST)/usr/share/man/man1
-	cp -a ffmpeg_compiled/usr/local/lib/*.so* $(DIST)/usr/lib/omxplayer/
-	cd $(DIST); tar -czf ../$(DIST).tgz *
