diff --git a/OMXControl.cpp b/OMXControl.cpp
index 0ef8e5f..834461f 100644
--- a/OMXControl.cpp
+++ b/OMXControl.cpp
@@ -88,7 +88,7 @@ int OMXControl::init(OMXClock *m_av_clock, OMXPlayerAudio *m_player_audio, OMXPl
 void OMXControl::dispatch()
 {
   if (bus)
-    dbus_connection_read_write_dispatch(bus, 0);
+    dbus_connection_read_write(bus, 0);
 }
 
 int OMXControl::dbus_connect(std::string& dbus_name)
