diff --git a/OMXAudio.cpp b/OMXAudio.cpp
index 41f282d..2fd12f1 100644
--- a/OMXAudio.cpp
+++ b/OMXAudio.cpp
@@ -899,7 +899,8 @@ unsigned int COMXAudio::AddPackets(const void* data, unsigned int len, double dt
         {
           //CLog::Log(LOGDEBUG, "%s::%s copy(%d,%d,%d) (s:%d f:%d sin:%d c:%d)", CLASSNAME, __func__, dst-(uint8_t *)omx_buffer->pBuffer, src-demuxer
           memcpy(dst, src, out_remaining * sample_pitch);
-          src += plane_size;
+          ///*JEHUTTING 20160124 issue-411*/ src += plane_size;
+          /*JEHUTTING 20160124 fix*/src += out_remaining * sample_pitch;
           dst += out_plane_size;
         }
         sample += out_remaining;

