--- ./src/lightdm.c.orig
+++ ./src/lightdm.c
@@ -1067,7 +1067,7 @@
     if (!config_has_key (config_get_instance (), "SeatDefaults", "start-session"))
         config_set_boolean (config_get_instance (), "SeatDefaults", "start-session", TRUE);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "allow-guest"))
-        config_set_boolean (config_get_instance (), "SeatDefaults", "allow-guest", TRUE);
+        config_set_boolean (config_get_instance (), "SeatDefaults", "allow-guest", FALSE);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "greeter-allow-guest"))
         config_set_boolean (config_get_instance (), "SeatDefaults", "greeter-allow-guest", TRUE);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "greeter-show-remote-login"))
--- ./data/lightdm.conf.orig
+++ ./data/lightdm.conf
@@ -91,7 +91,7 @@
 #greeter-show-manual-login=false
 #greeter-show-remote-login=true
 #user-session=default
-#allow-guest=true
+#allow-guest=false
 #guest-session=
 #session-wrapper=lightdm-session
 #greeter-wrapper=
