--- ./data/lightdm.conf.orig
+++ ./data/lightdm.conf
@@ -63,7 +63,7 @@
 #greeter-session=example-gtk-gnome
 #greeter-hide-users=false
 #user-session=default
-#allow-guest=true
+#allow-guest=false
 #guest-session=UNIMPLEMENTED
 #session-wrapper=lightdm-session
 #display-setup-script=
--- ./src/lightdm.c.orig
+++ ./src/lightdm.c
@@ -957,7 +957,7 @@
     if (!config_has_key (config_get_instance (), "SeatDefaults", "start-session"))
         config_set_boolean (config_get_instance (), "SeatDefaults", "start-session", TRUE);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "allow-guest"))
-        config_set_boolean (config_get_instance (), "SeatDefaults", "allow-guest", TRUE);
+        config_set_boolean (config_get_instance (), "SeatDefaults", "allow-guest", FALSE);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "greeter-session"))
         config_set_string (config_get_instance (), "SeatDefaults", "greeter-session", GREETER_SESSION);
     if (!config_has_key (config_get_instance (), "SeatDefaults", "user-session"))
