From 649ae8c5cf651c0a09226ed4712b2479d232e750 Mon Sep 17 00:00:00 2001
From: ptrcnull <git@ptrcnull.me>
Date: Thu, 3 Mar 2022 19:06:42 +0100
Subject: [PATCH] Remove positional arguments from i18n.merge_file

---
 data/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index c6aeffb4..0bc531a2 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -19,7 +19,6 @@ appdata_conf = configuration_data()
 appdata_conf.set('application_id', APPLICATION_ID)
 appdata = APPLICATION_ID + '.appdata.xml'
 appdata_file = I18N.merge_file(
-  appdata,
   input: configure_file(
     input: 'org.gnome.Devhelp.appdata.xml.in.in',
     output: APPLICATION_ID + '.appdata.xml.in',
@@ -45,7 +44,6 @@ desktop_conf = configuration_data()
 desktop_conf.set('application_id', APPLICATION_ID)
 desktop = APPLICATION_ID + '.desktop'
 desktop_file = I18N.merge_file(
-  desktop,
   type: 'desktop',
   input: configure_file(
     input: 'org.gnome.Devhelp.desktop.in.in',
-- 
2.35.1

