https://github.com/milkytracker/MilkyTracker/pull/319
--
diff --git a/src/midi/posix/MidiReceiver_pthread.cpp b/src/midi/posix/MidiReceiver_pthread.cpp
index 3585282..3997c01 100644
--- a/src/midi/posix/MidiReceiver_pthread.cpp
+++ b/src/midi/posix/MidiReceiver_pthread.cpp
@@ -145,4 +145,5 @@ unsigned int MidiReceiver::countPorts()
 	}
 	delete midiin;
 	midiin = NULL;
+	return nPorts;
 }
\ No newline at end of file
diff --git a/src/milkyplay/PlayerBase.h b/src/milkyplay/PlayerBase.h
index 18bcc74..6dc123c 100644
--- a/src/milkyplay/PlayerBase.h
+++ b/src/milkyplay/PlayerBase.h
@@ -115,7 +115,7 @@ private:
 							 (row << BITPOS_ROW) +
 							 (tempo << BITPOS_TEMPO) +
 							 (speed << BITPOS_SPEED)),
-			mainVolumeTicker((mainVol << BITPOS_MAINVOL) +
+			mainVolumeTicker(mainVol +
 							 (ticker << BITPOS_TICKER))
 		{
 		}
@@ -172,7 +172,7 @@ protected:
 	mp_sint32		tickSpeed;				// our tickspeed
 	mp_sint32		baseBpm;				// Support digibooster REAL BPM value
 	mp_sint32		bpm;					// BPM speed
-	mp_sint32		ticker;					// runs from 0 to tickspeed-1
+	mp_sint32		ticker = 0;					// runs from 0 to tickspeed-1
 
 	mp_sint32		rowcnt;					// counts through each row in a pattern
 	mp_sint32		poscnt;					// counts through each index the pattern index table
