Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: fix paths in daemon.sh
----

--- a/bin/daemon.sh
+++ b/bin/daemon.sh
@@ -88,7 +88,7 @@
 test ".$MAX_FD" = . && MAX_FD="maximum"
 # Setup parameters for running the jsvc
 #
-test ".$TOMCAT_USER" = . && TOMCAT_USER=tomcat
+test ".$TOMCAT_USER" = . && TOMCAT_USER=tomcat9
 # Set JAVA_HOME to working JDK or JRE
 # If not set we'll try to guess the JAVA_HOME
 # from java binary if on the PATH
@@ -111,16 +111,10 @@
 fi
 
 # Only set CATALINA_HOME if not already set
-test ".$CATALINA_HOME" = . && CATALINA_HOME=`cd "$DIRNAME/.." >/dev/null; pwd`
+test ".$CATALINA_HOME" = . && CATALINA_HOME=/usr/share/tomcat9
 test ".$CATALINA_BASE" = . && CATALINA_BASE="$CATALINA_HOME"
 test ".$CATALINA_MAIN" = . && CATALINA_MAIN=org.apache.catalina.startup.Bootstrap
-# If not explicitly set, look for jsvc in CATALINA_BASE first then CATALINA_HOME
-if [ -z "$JSVC" ]; then
-    JSVC="$CATALINA_BASE/bin/jsvc"
-    if [ ! -x "$JSVC" ]; then
-        JSVC="$CATALINA_HOME/bin/jsvc"
-    fi
-fi
+JSVC=/usr/bin/jsvc
 # Set the default service-start wait time if necessary
 test ".$SERVICE_START_WAIT_TIME" = . && SERVICE_START_WAIT_TIME=10
 
