diff -Naru arpwatch-2.1a15.orig/bihourly.sh arpwatch-2.1a15/bihourly.sh
--- arpwatch-2.1a15.orig/bihourly.sh	2006-07-28 22:19:45.000000000 +0400
+++ arpwatch-2.1a15/bihourly.sh	2006-09-22 21:29:38.000000000 +0400
@@ -10,8 +10,8 @@
 #
 list="`cat list`"
 cname="`cat cname`"
-temp1=/tmp/bihourly.1.$$
-temp2=/tmp/bihourly.2.$$
+temp1=$(mktemp)
+temp2=$(mktemp)
 d=/tmp/errs
 
 # imperfect hack
diff -Naru arpwatch-2.1a15.orig/mkdep arpwatch-2.1a15/mkdep
--- arpwatch-2.1a15.orig/mkdep	1996-06-23 13:25:24.000000000 +0400
+++ arpwatch-2.1a15/mkdep	2006-09-22 21:30:04.000000000 +0400
@@ -51,7 +51,7 @@
 	exit 1
 fi
 
-TMP=/tmp/mkdep$$
+TMP=$(mktemp)
 
 trap 'rm -f $TMP ; exit 1' 1 2 3 13 15
 
