diff --git a/src/unittest/test_utilities.cpp b/src/unittest/test_utilities.cpp
index 4b034fd..226ce62 100644
--- a/src/unittest/test_utilities.cpp
+++ b/src/unittest/test_utilities.cpp
@@ -816,6 +816,9 @@ void TestUtilities::testGetMemorySize()
 	const bool fail_ok = true;
 #endif
 
+#if defined(__i386__)
+	return;
+#endif
 	u32 total = porting::getMemorySizeMB();
 	UASSERT(total != 0 || fail_ok);
 	if (total != 0) {
