diff --git a/src/system/main.c b/src/system/main.c
index f0ccba0..122b6e0 100644
--- a/src/system/main.c
+++ b/src/system/main.c
@@ -160,7 +160,7 @@ int hin_process_argv (basic_args_t * args, const char * name) {
   break;
   case CWORKDIR:
   case CLOGDIR:
-  case CTMPDIR:
+  case CTMPDIR: {
     const char * path = basic_args_get (args);
     if (path == NULL) {
       printf ("missing %s path\n", cmd->nlong);
@@ -179,7 +179,7 @@ int hin_process_argv (basic_args_t * args, const char * name) {
       hin_directory_path (path, &master.tmpdir_path);
     break;
     }
-  break;
+  break; }
   case CPIDFILE: {
     const char * path = basic_args_get (args);
     if (path == NULL) {
