diff --git a/src/netcode/uring.c b/src/netcode/uring.c
index 6d7a46d..76ba719 100644
--- a/src/netcode/uring.c
+++ b/src/netcode/uring.c
@@ -9,6 +9,8 @@
 #include <unistd.h>
 #include <fcntl.h>
 
+#include <linux/stat.h>
+
 #include "hin.h"
 #include "hin_internal.h"
 #include "conf.h"
@@ -330,16 +332,6 @@ int hin_event_process () {
   }
 
   err = io_uring_peek_batch_cqe (&ring, cqes, 64);
-  /*if (err == -1) {
-    if (err == -EINTR) continue;
-    if (err == -EAGAIN) {
-      //printf ("received eagain\n");
-      break;
-    }
-    hin_error ("io_uring_peek_cqe: %s", strerror (-err));
-    //io_uring_cqe_seen (&ring, cqe);
-    continue;
-  }*/
 
   for (int i=0; i<err; i++) {
     struct io_uring_cqe *cqe = cqes[i];
