diff --git a/src/ls_odbc.c b/src/ls_odbc.c
index 5d44105..57b39d7 100644
--- a/src/ls_odbc.c
+++ b/src/ls_odbc.c
@@ -36,7 +36,7 @@
 typedef struct {
 	SQLPOINTER buf;
 	SQLINTEGER len;
-	SQLINTEGER type;
+	SQLLEN type;
 } param_data;
 
 /* general form of the driver objects */
@@ -685,7 +685,7 @@ static int raw_execute(lua_State *L, int istmt)
 		return create_cursor(L, -1, stmt, numcols);
 	} else {
 		/* if action has no results (e.g., UPDATE) */
-		SQLINTEGER numrows;
+		SQLLEN numrows;
 		if(error(SQLRowCount(stmt->hstmt, &numrows))) {
 			return fail(L, hSTMT, stmt->hstmt);
 		}
@@ -697,7 +697,7 @@ static int raw_execute(lua_State *L, int istmt)
 
 static int set_param(lua_State *L, stmt_data *stmt, int i, param_data *data)
 {
-	static SQLINTEGER cbNull = SQL_NULL_DATA;
+	static SQLLEN cbNull = SQL_NULL_DATA;
 
 	switch(lua_type(L, -1)) {
 	case LUA_TNIL: {
