--- a/src/f2fs/f2fs_fs.h
+++ b/src/f2fs/f2fs_fs.h
@@ -149,7 +149,8 @@
 
 #if HAVE_BYTESWAP_H
 #include <byteswap.h>
-#else
+#endif
+#if !defined(HAVE_DECL_BSWAP_16)
 /**
  * bswap_16 - reverse bytes in a uint16_t value.
  * @val: value whose bytes to swap.
@@ -163,7 +164,8 @@
 	return ((val & (uint16_t)0x00ffU) << 8)
 		| ((val & (uint16_t)0xff00U) >> 8);
 }
-
+#endif
+#if !defined(HAVE_DECL_BSWAP_32)
 /**
  * bswap_32 - reverse bytes in a uint32_t value.
  * @val: value whose bytes to swap.
@@ -181,7 +183,7 @@
 }
 #endif /* !HAVE_BYTESWAP_H */
 
-#if defined HAVE_DECL_BSWAP_64 && !HAVE_DECL_BSWAP_64
+#if (defined HAVE_DECL_BSWAP_64 && !HAVE_DECL_BSWAP_64) || !defined(HAVE_DECL_BSWAP_64)
 /**
  * bswap_64 - reverse bytes in a uint64_t value.
  * @val: value whose bytes to swap.
