diff --git a/profiles/apparmor.d/usr.sbin.dnsmasq b/profiles/apparmor.d/usr.sbin.dnsmasq
index 7ae9a14..0005b14 100644
--- a/profiles/apparmor.d/usr.sbin.dnsmasq
+++ b/profiles/apparmor.d/usr.sbin.dnsmasq
@@ -66,6 +66,7 @@ profile dnsmasq /usr/{bin,sbin}/dnsmasq flags=(attach_disconnected) {
   /var/lib/misc/dnsmasq.leases rw, # Required only for DHCP server usage

   /{,usr/}bin/{ba,da,}sh ix, # Required to execute --dhcp-script argument
+  /bin/busybox ix, # /bin/sh is a symlink of busybox on Alpine

   # access to iface mtu needed for Router Advertisement messages in IPv6
   # Neighbor Discovery protocol (RFC 2461)
