diff -ruN a/tests/test_libzim_creator.py b/tests/test_libzim_creator.py
--- a/tests/test_libzim_creator.py	2025-04-18 09:36:57.000000000 +0000
+++ b/tests/test_libzim_creator.py	2025-11-03 04:35:55.760000000 +0000
@@ -373,7 +373,7 @@
             c.add_item(None)  # pyright: ignore [reportCallIssue, reportArgumentType]
         with pytest.raises(RuntimeError):
             c.add_item("hello")  # pyright: ignore [reportCallIssue, reportArgumentType]
-        with pytest.raises(TypeError, match="takes exactly 1 positional argument"):
+        with pytest.raises(TypeError, match="got an unexpected keyword argument"):
             c.add_item(mimetype="text/html")  # pyright: ignore [reportCallIssue]
 
 
