diff --git a/src/confdb/confdb.h b/src/confdb/confdb.h
index 8af625f01..25a959e2b 100644
--- a/src/confdb/confdb.h
+++ b/src/confdb/confdb.h
@@ -22,6 +22,7 @@
 #ifndef _CONF_DB_H
 #define _CONF_DB_H
 
+#include <sys/types.h>
 #include <stdbool.h>
 #include <talloc.h>
 #include <tevent.h>
diff --git a/src/util/util.h b/src/util/util.h
index bc89ecbc2..d3f466578 100644
--- a/src/util/util.h
+++ b/src/util/util.h
@@ -31,6 +31,7 @@
 #include <pcre.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/param.h>
 #include <netinet/in.h>
 
 #include <talloc.h>
@@ -46,6 +47,10 @@
 #include "util/sss_format.h"
 #include "util/debug.h"
 
+#ifndef ALLPERMS
+#define ALLPERMS (S_ISUID | S_ISGID | S_IRGRP | S_IRWXU | S_IRWXG | S_IRWXO)
+#endif
+
 /* name of the monitor server instance */
 #define SSSD_PIDFILE PID_PATH"/sssd.pid"
 #define MAX_PID_LENGTH 10
