--- a/compiler/build.gradle
+++ b/compiler/build.gradle
@@ -55,8 +55,8 @@
                 linker.executable = 'powerpc64le-linux-gnu-g++'
             }
             target("aarch_64") {
-                cppCompiler.executable = 'aarch64-linux-gnu-g++'
-                linker.executable = 'aarch64-linux-gnu-g++'
+                cppCompiler.executable = 'aarch64-alpine-linux-musl-c++'
+                linker.executable = 'aarch64-alpine-linux-musl-c++'
             }
             target("s390_64")
             target("loongarch_64")
