--- startlxde.in.orig	2011-08-03 21:59:05.000000000 +0200
+++ startlxde.in	2012-07-02 19:40:19.982368074 +0200
@@ -8,7 +8,7 @@ fi
 PCMANFM_CONF_DIR="$XDG_CONFIG_HOME/pcmanfm/LXDE"
 if [ ! -f "$PCMANFM_CONF_DIR/pcmanfm.conf" ]; then
 	mkdir -p "$PCMANFM_CONF_DIR"
-	cp @prefix@/share/lxde/pcmanfm/pcmanfm.conf "$PCMANFM_CONF_DIR/pcmanfm.conf"
+	cp @sysconfdir@/xdg/pcmanfm/LXDE/pcmanfm.conf "$PCMANFM_CONF_DIR/pcmanfm.conf"
 fi
 
 # Ensure the existance of openbox config file
