diff -urNp corkscrew-2.0.orig/README corkscrew-2.0/README
--- corkscrew-2.0.orig/README	2007-05-19 10:56:12.000000000 +0530
+++ corkscrew-2.0/README	2007-05-19 10:56:45.000000000 +0530
@@ -83,7 +83,7 @@ if you could include the following infor
 
 *NOTE: I have had problems using the auth features with Mircosoft Proxy
  server.  The problems are sporadic, and I believe that they are related
- to the round-robin setup that I was testing it again.  Your mileage may
+ to the round-robin setup that I was testing it against.  Your mileage may
  vary.
 
 
