--- a/README
+++ b/README
@@ -66,7 +66,7 @@
 following line to your ~/.ssh/config file will usually do the trick
 (replace proxy.example.com and 8080 with correct values):
 
-    ProxyCommand /usr/local/bin/corkscrew proxy.example.com 8080 %h %p
+    ProxyCommand /usr/bin/corkscrew proxy.example.com 8080 %h %p
 
 NOTE: Command line syntax has changed since version 1.5. Please notice
 that the proxy port is NOT optional anymore and is required in the
@@ -91,7 +91,7 @@
 Now you will have to change the ProxyCommand line in your ~/.ssh/config
 file. Here's an example:
 
-    ProxyCommand /usr/local/bin/corkscrew proxy.work.com 80 %h %p ~/.ssh/myauth
+    ProxyCommand /usr/bin/corkscrew proxy.work.com 80 %h %p ~/.ssh/myauth
 
 The proxy authentication feature is very new and has not been tested
 extensively so your mileage may vary. If you encounter any problems when
