diff -urNp corkscrew-2.0.orig/corkscrew.c corkscrew-2.0/corkscrew.c
--- corkscrew-2.0.orig/corkscrew.c	2007-05-19 09:56:05.000000000 +0530
+++ corkscrew-2.0/corkscrew.c	2007-05-19 10:07:00.000000000 +0530
@@ -176,6 +176,7 @@ char *argv[];
 #endif
 	char *host = NULL, *desthost = NULL, *destport = NULL;
 	char *up = NULL;
+	char *tmp = NULL;
 	int port, sent, setup, code, csock;
 	fd_set rfd, sfd;
 	struct timeval tv;
@@ -262,6 +263,11 @@ char *argv[];
 						fprintf(stderr, "Proxy could not open connnection to %s: %s\n", desthost, descr);
 						exit(-1);
 					}
+					if (tmp = strstr(buffer,"\r\n\r\n"))
+					{
+					    tmp += 4;
+					    write(1, tmp, strlen(tmp));
+					}
 				}
 			}
 			if (FD_ISSET(csock, &sfd) && (sent == 0)) {
diff -urNp corkscrew-2.0.orig/README corkscrew-2.0/README
--- corkscrew-2.0.orig/README	2007-05-19 09:56:05.000000000 +0530
+++ corkscrew-2.0/README	2007-05-19 09:56:48.000000000 +0530
@@ -47,7 +47,7 @@ Setting up Corkscrew with SSH/OpenSSH is
 the following line to your ~/.ssh/config file will usually do
 the trick (replace proxy.example.com and 8080 with correct values):
 
-ProxyCommand /usr/local/bin/corkscrew proxy.example.com 8080 %h %p
+ProxyCommand /usr/bin/corkscrew proxy.example.com 8080 %h %p
 
 NOTE: Command line syntax has changed since version 1.5.  Please
 notice that the proxy port is NOT optional anymore and is required
@@ -70,7 +70,7 @@ chmod 600 myauth
 Now you will have to change the ProxyCommand line in your ~/.ssh/config
 file.  Here's an example :
 
-ProxyCommand /usr/local/bin/corkscrew proxy.work.com 80 %h %p ~/.ssh/myauth
+ProxyCommand /usr/bin/corkscrew proxy.work.com 80 %h %p ~/.ssh/myauth
 
 The proxy authentication feature is very new and has not been tested
 extensively so your mileage may vary.  If you encounter any problems
