diff --git a/Makefile b/Makefile
index 3b9055d..841a999 100644
--- a/Makefile
+++ b/Makefile
@@ -15,8 +15,8 @@ CFLAGS += -DVERSION=\"$(VERSION)$(GIT_VERSION)\"
 CFLAGS += -DCONFDIR=\"$(CONFDIR)\"
 CFLAGS += $(shell pkg-config --cflags glib-2.0)
 
-LIBS   ?= -lcurl
-LIBS   += $(shell pkg-config --libs   glib-2.0)
+LIBS   ?= $(shell pkg-config --libs libcurl)
+LIBS   += $(shell pkg-config --libs glib-2.0)
 
 CC     ?= gcc
 
@@ -33,7 +33,7 @@ all: $(TARGETS)
 #
 
 pastebinc: 
-	$(CC) -fPIC $(CFLAGS) $(LIBS) -o $(PROGNAME) pastebinc.c
+	$(CC) -fPIC $(CFLAGS) -o $(PROGNAME) pastebinc.c $(LIBS)
 
 clean:
 	rm -f *.o *.out $(PROGNAME)
