From 233894be12fa2c55f7f5ad838e66c6051da43d7b Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 21 Oct 2022 07:43:34 +0200
Subject: [PATCH] handle user auth failure

---
 mqtt/publish.lua | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/mqtt/publish.lua b/mqtt/publish.lua
index ee405a1..d021384 100644
--- a/mqtt/publish.lua
+++ b/mqtt/publish.lua
@@ -23,8 +23,14 @@ end
 
 function publish.multiple(msgs, hostname, port, client_id, keepalive)
 	local client = mqtt.new(client_id)
+	local retry = true
+	local all_sent = false
 
-	client.ON_CONNECT = function()
+	client.ON_CONNECT = function(success, rc, msg)
+		if not success then
+			retry = false
+			return
+		end
 		for i=1,#msgs do
 			client:publish(msgs[i].topic, msgs[i].payload,
 				       msgs[i].qos, msgs[i].retain)
@@ -36,10 +42,15 @@ function publish.multiple(msgs, hostname, port, client_id, keepalive)
 			msgs = table.remove(msgs, 1)
 		end
 		if msgs == nil or #msgs == 0 then
+			all_sent = true
 			client:disconnect()
 		end
 	end
 
+	client.ON_DISCONNECT = function()
+		retry = false
+	end
+
 	if publish.tls then
 		client:tls_set(publish.tls.cafile, publish.tls.capath,
 			publish.tls.certfile, publish.tls.keyfile)
@@ -49,8 +60,11 @@ function publish.multiple(msgs, hostname, port, client_id, keepalive)
 	end
 	client:connect(hostname or publish.hostname, port or publish.port,
 		       keepalive or publish.keepalive)
-	client:loop_forever()
+	while retry do
+		client:loop()
+	end
 	client:destroy()
+	return all_sent
 end
 
 function publish.single(topic, payload, qos, retain, hostname, port, client_id,
-- 
2.38.0

