diff -urp cairo-1.14.6.orig/src/cairo-ft-font.c cairo-1.14.6/src/cairo-ft-font.c
--- cairo-1.14.6.orig/src/cairo-ft-font.c	2016-12-05 15:58:48.292450747 -0800
+++ cairo-1.14.6/src/cairo-ft-font.c	2016-12-05 16:00:16.972824582 -0800
@@ -2260,7 +2260,7 @@ _cairo_ft_scaled_glyph_init (void			*abs
      * Moreover, none of our backends and compositors currently support
      * color glyphs.  As such, this is currently disabled.
      */
-    /* load_flags |= FT_LOAD_COLOR; */
+     load_flags |= FT_LOAD_COLOR;
 #endif
 
     error = FT_Load_Glyph (face,

