diff --git a/src/lib/base/String.h b/src/lib/base/String.h
index 9c5a53b..faf3596 100644
--- a/src/lib/base/String.h
+++ b/src/lib/base/String.h
@@ -23,6 +23,7 @@
 
 #include <stdarg.h>
 #include <vector>
+#include <cstdint>
 
 // use standard C++ string class for our string class
 typedef std::string String;
diff --git a/src/lib/net/FingerprintData.h b/src/lib/net/FingerprintData.h
index 938a695..8ab1631 100644
--- a/src/lib/net/FingerprintData.h
+++ b/src/lib/net/FingerprintData.h
@@ -20,6 +20,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace barrier {
 
