diff --git b/ext/phar/tests/bug81726.phpt a/ext/phar/tests/bug81726.phpt
index 4f7467c23a..a86c52a649 100644
--- b/ext/phar/tests/bug81726.phpt
+++ a/ext/phar/tests/bug81726.phpt
@@ -8,5 +8,5 @@
 var_dump(fopen("phar://" . __DIR__ . "/bug81726.gz", "r"));
 ?>
 --EXPECTF--
-Warning: fopen(phar://%s): failed to open stream: unable to decompress gzipped phar archive "%s" in %s on line %d
+Warning: fopen(phar://%s): Failed to open stream: unable to decompress gzipped phar archive "%s" in %s on line %d
 bool(false)
