From cc4bf7ac97f1fdcc156ebe1f3547e678e8bd7311 Mon Sep 17 00:00:00 2001
From: Ted Trask <ttrask01@yahoo.com>
Date: Wed, 2 May 2012 15:18:19 +0000
Subject: [PATCH] Shorewall package now renamed to shorewall-shell

---
 shorewall-model.lua |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shorewall-model.lua b/shorewall-model.lua
index 1d8fda0..7326094 100644
--- a/shorewall-model.lua
+++ b/shorewall-model.lua
@@ -8,7 +8,7 @@ format = require("acf.format")
 -- Set variables
 local configfile = "/etc/shorewall/shorewall.conf"
 local processname = "shorewall"
-local packagename = "shorewall"
+local packagename = "shorewall-shell"
 local baseurl = "/etc/shorewall/"
 --[[
 local config = {}
-- 
1.7.10.3

