diff --git a/src/third-party/rapidyaml/ryml_all.hpp b/src/third-party/rapidyaml/ryml_all.hpp
index 03734a1..6084389 100644
--- a/src/third-party/rapidyaml/ryml_all.hpp
+++ b/src/third-party/rapidyaml/ryml_all.hpp
@@ -498,6 +498,10 @@ C4_FOR_EACH(PRN_STRUCT_OFFSETS, a, b, c);
    #endif
    #define C4_BYTE_ORDER _C4EL
 
+#elif defined(__loongarch64)
+   #define C4_CPU_LOONGARCH64
+   #define C4_WORDSIZE 8
+
 #elif defined(__EMSCRIPTEN__)
 #   define C4_BYTE_ORDER _C4EL
 #   define C4_WORDSIZE 4
