diff --git a/Makefile b/Makefile
index b14e2fa..9345c89 100644
--- a/Makefile
+++ b/Makefile
@@ -73,10 +73,6 @@ install:	minissdpd
	$(INSTALL) minissdpd $(DESTDIR)$(SBININSTALLDIR)
	$(INSTALL) -d $(DESTDIR)$(MANINSTALLDIR)/man1
	$(INSTALL) minissdpd.1 $(DESTDIR)$(MANINSTALLDIR)/man1/minissdpd.1
-ifeq (, $(findstring darwin, $(OS)))
-	$(INSTALL) -d $(DESTDIR)/etc/init.d
-	$(INSTALL) minissdpd.init.d.script $(DESTDIR)/etc/init.d/minissdpd
-endif

 check:	validateminissdpd validatecodelength
