--- a/config.def.h
+++ b/config.def.h
@@ -1,10 +1,10 @@
 /* See LICENSE file for copyright and license details. */
 
-#define ENV_SUPATH	"/bin"
-#define ENV_PATH	"/bin"
+#define ENV_SUPATH	"/usr/sbin:/usr/bin:/sbin:/bin"
+#define ENV_PATH	"/usr/bin:/bin"
 #define PW_CIPHER	"$6$"	/* SHA-512 */
 #undef UTMP_PATH
-#define UTMP_PATH	"/var/run/utmp"
+#define UTMP_PATH	"/run/utmp"
 #undef BTMP_PATH
 #define BTMP_PATH	"/var/log/btmp"
 #undef WTMP_PATH
