From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 5 Sep 2021 18:27:48 +0200
Subject: [PATCH] Fix missing include <sys/sysmacros.h>

--- a/libutil/tty.c
+++ b/libutil/tty.c
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 
 #include <fcntl.h>
 #include <dirent.h>
--- a/mknod.c
+++ b/mknod.c
@@ -1,5 +1,6 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 
 #include <fcntl.h>
--- a/mountpoint.c
+++ b/mountpoint.c
@@ -1,5 +1,6 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 
 #include <mntent.h>
--- a/stat.c
+++ b/stat.c
@@ -1,5 +1,6 @@
 /* See LICENSE file for copyright and license details. */
 #include <sys/stat.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 
 #include <inttypes.h>
