From 017ab80d86203d66dda1702fe30b0ee8b8a0aaa9 Mon Sep 17 00:00:00 2001
From: Brandon Boese <brandonboese@protonmail.com>
Date: Sun, 15 Oct 2023 16:52:02 -0400
Subject: [PATCH 2/3] remove systemd dependency

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f1f2a5f..f177f18 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,6 @@ find_package(PkgConfig REQUIRED)
 pkg_check_modules(TP_QT5 REQUIRED TelepathyQt5)
 pkg_check_modules(SQLITE3 REQUIRED sqlite3)
 pkg_check_modules(QTGLIB QtGLib-2.0)
-pkg_check_modules(SYSTEMD REQUIRED systemd)
 
 find_program(DBUS_RUNNER dbus-test-runner)
 
-- 
2.45.1

