diff --color -rupN a/rfc_bibtex/rfc_bibtex.py b/rfc_bibtex/rfc_bibtex.py
--- a/rfc_bibtex/rfc_bibtex.py	2019-03-16 11:41:21.000000000 +0100
+++ b/rfc_bibtex/rfc_bibtex.py	2026-04-10 08:52:51.275743697 +0200
@@ -75,7 +75,9 @@ class RFCBibtex(object):
     @property
     def bibtex_entries(self):
         # remove Nones (errors returned by urllib), so that they're not printed
-        return filter(None.__ne__, [self.get_bibtex_from_id(id_name) for id_name in self._id_names])
+        return (
+            i for i in [self.get_bibtex_from_id(id_name) for id_name in self._id_names] if i is not None
+        )
 
     def _read_ids_from_plain_file(self, filename):
         with open(filename, 'r') as f:
