diff --git a/test/meson.build b/test/meson.build
index a33b16d..afff49e 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -122,19 +122,6 @@ test('TestIcu', executable(
   ],
 ))
 
-if libavahi_client_dep.found()
-  executable(
-    'run_avahi',
-    'run_avahi.cxx',
-    'ShutdownHandler.cxx',
-    include_directories: inc,
-    dependencies: [
-      zeroconf_dep,
-      util_dep,
-    ],
-  )
-endif
-
 if enable_inotify
   executable(
     'run_inotify',
