--- a/test/meson.build
+++ b/test/meson.build
@@ -123,21 +123,6 @@
   ],
 ))
 
-if libavahi_client_dep.found()
-  executable(
-    'run_avahi',
-    'run_avahi.cxx',
-    'ShutdownHandler.cxx',
-    '../src/Log.cxx',
-    '../src/LogBackend.cxx',
-    include_directories: inc,
-    dependencies: [
-      zeroconf_dep,
-      util_dep,
-    ],
-  )
-endif
-
 if enable_inotify
   executable(
     'run_inotify',
