From 8c85c46d2db93fcbb7ff1e0bbeaeb3e4dd0e71a1 Mon Sep 17 00:00:00 2001
From: Kenny Levinsen <kl@kl.wtf>
Date: Sun, 12 Sep 2021 12:04:38 +0200
Subject: [PATCH 4/4] man/seatd-launch: Make mssage about root clearer

---
 man/seatd-launch.1.scd | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/man/seatd-launch.1.scd b/man/seatd-launch.1.scd
index b1fb685..d882ba5 100644
--- a/man/seatd-launch.1.scd
+++ b/man/seatd-launch.1.scd
@@ -27,11 +27,11 @@ appropriately.  Once the specified command terminates, the seatd instance is
 also terminated.
 
 seatd requires root privileges to perform its tasks. This can be achieved
-through SUID of seatd-launch, *sudo(8)*/*doas(1)*, or by running seatd-launch
-as root. seatd-launch will drop privileges from the effective user to the real
-user before running the specified command. If the real user is root, this is
-simply a noop - only run directly as root if you intend for the specified
-command to run as root as well.
+through SUID of seatd-launch or by running seatd-launch as root. seatd-launch
+will drop privileges from the effective user to the real user before running
+the specified command. If the real user is root, this is simply a noop. You
+should only run seatd-launch as root if you intend for the specified command to
+run as root as well.
 
 seatd-launch serves a similar purpose to the libseat "builtin" backend, but is
 superior to it for two reasons:
-- 
2.33.0

