diff --git a/setup.py b/setup.py
index c1bf914..2f4c9d4 100644
--- a/setup.py
+++ b/setup.py
@@ -1,5 +1,5 @@
 import codecs
-from distutils.version import StrictVersion
+from distutils.version import LooseVersion
 import os
 import re
 import sys
@@ -62,7 +62,7 @@
 # So this dependency is not added for old Linux distributions with old setuptools,
 # in order to allow these systems to build certbot from sources.
 pywin32_req = 'pywin32>=227'  # do not forget to edit pywin32 dependency accordingly in windows-installer/construct.py
-if StrictVersion(setuptools_version) >= StrictVersion('36.2'):
+if LooseVersion(setuptools_version) >= LooseVersion('36.2'):
     install_requires.append(pywin32_req + " ; sys_platform == 'win32'")
 elif 'bdist_wheel' in sys.argv[1:]:
     raise RuntimeError('Error, you are trying to build certbot wheels using an old version '
