diff -rupN a/src/frontend/qt_sdl/CMakeLists.txt b/src/frontend/qt_sdl/CMakeLists.txt
--- a/src/frontend/qt_sdl/CMakeLists.txt	2022-11-03 19:35:21.000000000 +0000
+++ b/src/frontend/qt_sdl/CMakeLists.txt	2025-11-21 03:22:20.490000000 +0000
@@ -68,6 +68,9 @@ endif()
 if (USE_QT6)
     find_package(Qt6 COMPONENTS Core Gui Widgets Network Multimedia OpenGL OpenGLWidgets REQUIRED)
     set(QT_LINK_LIBS Qt6::Core Qt6::Gui Qt6::Widgets Qt6::Network Qt6::Multimedia Qt6::OpenGL Qt6::OpenGLWidgets)
+    if (LINUX AND Qt6_VERSION VERSION_GREATER_EQUAL "6.10.0")
+        find_package(Qt6GuiPrivate ${QT_MINIMUM_VERSION} REQUIRED)
+    endif()
 else()
     find_package(Qt5 COMPONENTS Core Gui Widgets Network Multimedia REQUIRED)
     set(QT_LINK_LIBS Qt5::Core Qt5::Gui Qt5::Widgets Qt5::Network Qt5::Multimedia)
@@ -134,6 +137,9 @@ else()
     target_include_directories(melonDS PRIVATE "${X11_INCLUDE_DIR}")
 endif()
 
+if (LINUX AND Qt6_VERSION VERSION_GREATER_EQUAL "6.10.0")
+    target_link_libraries(melonDS PRIVATE Qt6::GuiPrivate)
+endif()
 
 if (BUILD_STATIC)
     qt_import_plugins(melonDS INCLUDE Qt::QSvgPlugin)
