diff -up Unicode-Normalize-1.25/Makefile.PL.orig Unicode-Normalize-1.25/Makefile.PL
--- Unicode-Normalize-1.25/Makefile.PL.orig	2017-05-12 10:24:13.296305863 +0200
+++ Unicode-Normalize-1.25/Makefile.PL	2017-05-12 10:25:28.275822594 +0200
@@ -8,7 +8,7 @@ my $mm_ver = ExtUtils::MakeMaker->VERSIO
 if (-f "Normalize.xs") {
     print STDERR "Making header files for XS...\n";
 
-    do 'mkheader' or die $@ || "mkheader: $!";
+    do './mkheader' or die $@ || "mkheader: $!";
 
     $clean = { FILES => 'unfcan.h unfcmb.h unfcmp.h unfcpt.h unfexc.h' };
 }
