--- jailkit-2.15/Makefile.in
+++ jailkit-2.15/Makefile.in-patched
@@ -67,14 +67,6 @@
 	@cd src/ && $(MAKE) install
 	@cd py/ && $(MAKE) install
 	@cd man/ && $(MAKE) install
-	# test if the jk_chrootsh is already in /etc/shells
-	# this previously had @echo but that fails on FreeBSD
-	if test -w /etc/shells; then \
-		if ! grep ${prefix}/sbin/jk_chrootsh /etc/shells ; then \
-			echo "appending ${prefix}/sbin/jk_chroots to /etc/shells";\
-			echo ${prefix}/sbin/jk_chrootsh >> /etc/shells ;\
-		fi \
-	fi
 
 
 uninstall:
@@ -83,5 +75,3 @@
 	@cd man/ && $(MAKE) uninstall
 	@cd src/ && $(MAKE) uninstall
 	-rmdir --ignore-fail-on-non-empty ${DESTDIR}${iniprefix}
-	@echo "You must manually remove jk_chrootsh from /etc/shells"
-# remove jk_chrootsh from /etc/shells
