--- a/lib/private/config.php
+++ b/lib/private/config.php
@@ -229,7 +229,8 @@
 		$filePointer = fopen($this->configFilePath, 'r+');
 
 		// Prevent others not to read the config
-		chmod($this->configFilePath, 0640);
+		// Alpine Linux Development - don't do package's job
+		// chmod($this->configFilePath, 0640);
 
 		// File does not exist, this can happen when doing a fresh install
 		if(!is_resource ($filePointer)) {
