diff --git a/src/manage_sql.c b/src/manage_sql.c
index e316374..83d00e9 100644
--- a/src/manage_sql.c
+++ b/src/manage_sql.c
@@ -12709,7 +12709,7 @@ void
 manage_cleanup_process_error (int signal)
 {
   g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Received %s signal.\n",
-         sys_siglist[signal]);
+         strsignal(signal));
   if (sql_is_open ())
     {
       if (current_scanner_task)
diff --git a/src/openvasmd.c b/src/openvasmd.c
index 808a8f1..c9981ad 100644
--- a/src/openvasmd.c
+++ b/src/openvasmd.c
@@ -661,7 +661,7 @@ setup_signal_handler (int signal, void (*handler) (int), int block)
   if (sigaction (signal, &action, NULL) == -1)
     {
       g_critical ("%s: failed to register %s handler\n",
-                  __FUNCTION__, sys_siglist[signal]);
+                  __FUNCTION__, strsignal(signal));
       exit (EXIT_FAILURE);
     }
 }
@@ -1025,7 +1025,7 @@ serve_and_schedule ()
       if (termination_signal)
         {
           g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Received %s signal.\n",
-                 sys_siglist[termination_signal]);
+                 strsignal(termination_signal));
           cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           setup_signal_handler (termination_signal, SIG_DFL, 0);
@@ -1036,7 +1036,7 @@ serve_and_schedule ()
       if (sighup_update_nvt_cache)
         {
           g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Received %s signal.\n",
-                 sys_siglist[SIGHUP]);
+                 strsignal(SIGHUP));
           sighup_update_nvt_cache = 0;
           fork_update_nvt_cache ();
         }
@@ -1095,7 +1095,7 @@ serve_and_schedule ()
       if (termination_signal)
         {
           g_log (G_LOG_DOMAIN, G_LOG_LEVEL_DEBUG, "Received %s signal.\n",
-                 sys_siglist[termination_signal]);
+                 strsignal(termination_signal));
           cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           setup_signal_handler (termination_signal, SIG_DFL, 0);
