--- openvas-manager-7.0/src/openvasmd.c	2017-08-30 21:03:46 UTC (rev 29475)
+++ openvas-manager-7.0/src/openvasmd.c	2017-08-30 21:25:35 UTC (rev 29476)
@@ -661,8 +661,8 @@
           {
             if (openvas_server_new (GNUTLS_CLIENT,
                                     CACERT,
-                                    SCANNERCERT,
-                                    SCANNERKEY,
+                                    CLIENTCERT,
+                                    CLIENTKEY,
                                     &client_connection->session,
                                     &client_connection->credentials))
               exit (EXIT_FAILURE);
