diff --git a/meson.build b/meson.build
index 4602326..9dd9f8f 100644
--- a/meson.build
+++ b/meson.build
@@ -2,6 +2,7 @@ project('inih',
     ['c','cpp'],
     default_options : ['default_library=static'],
     license : 'BSD-3-Clause',
+    version : '@PKGVER@',
 )
 
 #### options ####
@@ -73,6 +74,7 @@ if distro_install
     pkg.generate(lib_inih,
         name : 'inih',
         description : 'simple .INI file parser',
+        version : meson.project_version(),
     )
 endif
 
@@ -99,6 +101,7 @@ if get_option('with_INIReader')
         pkg.generate(lib_INIReader,
             name : 'INIReader',
             description : 'simple .INI file parser for C++',
+            version : meson.project_version(),
         )
     endif
 
