--- hdparm-9.43/hdparm.h.orig
+++ hdparm-9.43/hdparm.h
@@ -1,7 +1,7 @@
 /* Some prototypes for extern functions. */
 
 //#undef __KERNEL_STRICT_NAMES
-#include <linux/types.h>
+#include <sys/types.h>
 
 #if !defined(__GNUC__) && !defined(__attribute__)
 #define __attribute__(x)
--- hdparm-9.43/sysfs.c.orig
+++ hdparm-9.43/sysfs.c
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <errno.h>
+#include <limits.h>
 #include <dirent.h>
 #include <sys/stat.h>
 #include <linux/types.h>
