diff --git a/asteroid-timer.pro b/asteroid-timer.pro
index f865860..d4d2968 100644
--- a/asteroid-timer.pro
+++ b/asteroid-timer.pro
@@ -1,5 +1,6 @@
 TARGET = asteroid-timer
-CONFIG += asteroidapp
+CONFIG += asteroidapp link_pkgconfig
+PKGCONFIG += asteroidapp
 
 SOURCES +=     main.cpp
 RESOURCES +=   resources.qrc
@@ -7,3 +8,6 @@ OTHER_FILES += main.qml
 
 lupdate_only{ SOURCES += i18n/asteroid-timer.desktop.h }
 TRANSLATIONS = $$files(i18n/$$TARGET.*.ts)
+
+target.path = /usr/bin/
+INSTALLS += target
