diff --git a/src/runtime/src/coreclr/vm/method.cpp.orig b/src/runtime/src/coreclr/vm/method.cpp
index 8b7239896a0..9ef542773c0 100644
--- a/src/runtime/src/coreclr/vm/method.cpp.orig
+++ b/src/runtime/src/coreclr/vm/method.cpp
@@ -1079,7 +1079,7 @@ PCODE MethodDesc::GetNativeCode()
         PCODE pCode = *ppCode;
 
 #ifdef TARGET_ARM
-        if (pCode != NULL)
+        if (pCode != (PCODE)NULL)
             pCode |= THUMB_CODE;
 #endif
         return pCode;
@@ -3194,7 +3194,7 @@ BOOL MethodDesc::SetNativeCodeInterlocked(PCODE addr, PCODE pExpected /*=NULL*/)
         _ASSERTE(IsThumbCode(addr) || (addr==NULL));
         addr &= ~THUMB_CODE;
 
-        if (pExpected != NULL)
+        if (pExpected != (PCODE)NULL)
         {
             _ASSERTE(IsThumbCode(pExpected));
             pExpected &= ~THUMB_CODE;
