--- conntrack-tools-1.4.2.orig/include/bitops.h
+++ conntrack-tools-1.4.2/include/bitops.h
@@ -1,7 +1,7 @@
 #ifndef _BITOPS_H_
 #define _BITOPS_H_
 
-#include <stdlib.h>
+#include <sys/types.h>
 
 static inline void set_bit_u32(int nr, u_int32_t *addr)
 {
--- conntrack-tools-1.4.2.orig/include/conntrackd.h
+++ conntrack-tools-1.4.2/include/conntrackd.h
@@ -12,6 +12,7 @@
 #include <stdio.h>
 #include <libnetfilter_conntrack/libnetfilter_conntrack.h>
 #include <syslog.h>
+#include <sys/select.h>
 
 /* UNIX facilities */
 #define CT_FLUSH_MASTER		0	/* flush kernel conntrack table */
--- conntrack-tools-1.4.2.orig/include/mcast.h
+++ conntrack-tools-1.4.2/include/mcast.h
@@ -2,6 +2,7 @@
 #define _MCAST_H_
 
 #include <stdint.h>
+#include <sys/select.h>
 #include <netinet/in.h>
 #include <net/if.h>
 
--- conntrack-tools-1.4.2.orig/include/tcp.h
+++ conntrack-tools-1.4.2/include/tcp.h
@@ -2,6 +2,7 @@
 #define _TCP_H_
 
 #include <stdint.h>
+#include <sys/select.h>
 #include <netinet/in.h>
 
 struct tcp_conf {
--- conntrack-tools-1.4.2.orig/include/udp.h
+++ conntrack-tools-1.4.2/include/udp.h
@@ -2,6 +2,7 @@
 #define _UDP_H_
 
 #include <stdint.h>
+#include <sys/select.h>
 #include <netinet/in.h>
 
 struct udp_conf {
