--- a/setup.py
+++ b/setup.py
@@ -35,9 +35,9 @@
 try:
     from setuptools_scm import get_version
 
-    __version__ = get_version()
+    __version__ = '%VERSION%'
 except ImportError:
-    __version__ = '0.0.1'
+    __version__ = '%VERSION%'
 
 
 ext_modules = [
@@ -68,11 +68,7 @@
     if ParallelCompile:
         ParallelCompile().install()
     setup(
-        setup_requires=[  # can be removed whenever we can drop pip 9 support
-            'setuptools_scm',  # so that version will work
-            'setuptools_scm_git_archive',  # enable version from github tarballs
-        ],
         ext_modules=ext_modules,
-        use_scm_version=True,
+        version='%VERSION%',
         cmdclass={"build_ext": build_ext},
     )
