--- a/config/defaults.inc.php
+++ b/config/defaults.inc.php
@@ -382,11 +382,11 @@
 // use this folder to store log files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
 // This is used by the 'file' log driver.
-$config['log_dir'] = RCUBE_INSTALL_PATH . 'logs/';
+$config['log_dir'] = '/var/log/roundcube';
 
 // use this folder to store temp files
 // must be writeable for the user who runs PHP process (Apache user if mod_php is being used)
-$config['temp_dir'] = RCUBE_INSTALL_PATH . 'temp/';
+$config['temp_dir'] = '/tmp';
 
 // expire files in temp_dir after 48 hours
 // possible units: s, m, h, d, w
--- a/program/include/iniset.php
+++ b/program/include/iniset.php
@@ -29,7 +29,7 @@
 }
 
 if (!defined('RCMAIL_CONFIG_DIR')) {
-    define('RCMAIL_CONFIG_DIR', INSTALL_PATH . 'config');
+    define('RCMAIL_CONFIG_DIR', '/etc/roundcube');
 }
 
 if (!defined('RCUBE_LOCALIZATION_DIR')) {
--- a/program/lib/Roundcube/bootstrap.php
+++ b/program/lib/Roundcube/bootstrap.php
@@ -65,7 +65,7 @@
 }
 
 if (!defined('RCUBE_CONFIG_DIR')) {
-    define('RCUBE_CONFIG_DIR', RCUBE_INSTALL_PATH . 'config/');
+    define('RCUBE_CONFIG_DIR', '/etc/roundcube');
 }
 
 if (!defined('RCUBE_PLUGINS_DIR')) {
