--- a/src/meson.build
+++ b/src/meson.build
@@ -19,6 +19,7 @@ run_command(
 cargo_options = [
   '--manifest-path', meson.project_source_root() / 'Cargo.toml',
   '--target-dir', meson.project_build_root() / 'src',
+  '--frozen',
 ]
 
 if profile == 'debug'
@@ -31,7 +32,6 @@ else
 endif
 
 cargo_env = [
-  'CARGO_HOME=' + meson.project_build_root() / 'cargo-home',
   'RUSTFLAGS=-L ' + (meson.project_build_root() / 'subprojects' / 'libgnome-games-support' / 'games')
         + ' -lgnome-games-support-3'
 ]
@@ -49,7 +49,7 @@ custom_target(
   command: [
     'env',
     cargo_env,
-    cargo, 'build', cargo_options,
+    cargo, 'auditable', 'build', cargo_options,
     '&&',
     'cp', 'src' / rust_target / meson.project_name(), '@OUTPUT@',
   ]
