diff --git a/tests/test_formatters/test_sh.py b/tests/test_formatters/test_sh.py
index 3b332c8..d1471ea 100644
--- a/tests/test_formatters/test_sh.py
+++ b/tests/test_formatters/test_sh.py
@@ -8,7 +8,7 @@ import pytest
 
 def run_shell_script(shell, sh_path, *args):
     return subprocess.check_output([
-            os.path.join("/bin", shell),
+            shell,
             sh_path,
         ] + list(args),
         shell=False
