TODO package py3-xattr

diff --git a/tests/test_options/test_cache.py b/tests/test_options/test_cache.py
index e8ae175..7782429 100644
--- a/tests/test_options/test_cache.py
+++ b/tests/test_options/test_cache.py
@@ -71,73 +71,6 @@ def test_xattr_basic(usual_setup_usual_teardown):
 
         head, *data, footer = run_rmlint('-D -S pa --xattr-clear')
 
-
-@pytest.mark.parametrize("extra_opts", ["", "-D"])
-def test_xattr_detail(usual_setup_usual_teardown, extra_opts):
-    if not runs_as_root():
-        # This tests need a ext4 fs which is created during the test.
-        # The mount step sadly needs root privileges.
-        return
-
-    with create_special_fs("this-is-not-tmpfs") as ext4_path:
-        # Keep the checksum fixed, if we change the default we don't want to
-        # break this test (although I'm sure some tests will break)
-        base_options = extra_opts + " -T df -S pa -a blake2b "
-
-        path_1 = os.path.join(ext4_path, "1")
-        path_2 = os.path.join(ext4_path, "2")
-        path_3 = os.path.join(ext4_path, "3")
-        path_4 = os.path.join(ext4_path, "4")
-
-        create_file("abc", path_1)
-        create_file("abc", path_2)
-        create_file("def", path_3)
-        create_file("longer", path_4)
-
-        head, *data, footer = run_rmlint(base_options + ' --xattr-write')
-        assert len(data) == 2
-
-        xattr_1 = must_read_xattr(path_1)
-        xattr_2 = must_read_xattr(path_2)
-        xattr_3 = must_read_xattr(path_3)
-        assert xattr_1["user.rmlint.blake2b.cksum"] == \
-                b'ba80a53f981c4d0d6a2797b69f12f6e94c212f14685ac4b74b12bb6fdbffa2d17d87c5392aab792dc252d5de4533cc9518d38aa8dbf1925ab92386edd4009923\x00'
-        assert xattr_1 == xattr_2
-
-        # no --write-unfinished given.
-        assert xattr_3 == {}
-
-        # Repeating the caching option should have no effect on the output.
-        for _ in range(10):
-            head, *data, footer = run_rmlint(base_options + ' --xattr')
-            # one more due to the unique_file
-            assert len(data) == 3
-
-            xattr_1 = must_read_xattr(path_1)
-            xattr_2 = must_read_xattr(path_2)
-            xattr_3 = must_read_xattr(path_3)
-            assert xattr_1["user.rmlint.blake2b.cksum"] == \
-                    b'ba80a53f981c4d0d6a2797b69f12f6e94c212f14685ac4b74b12bb6fdbffa2d17d87c5392aab792dc252d5de4533cc9518d38aa8dbf1925ab92386edd4009923\x00'
-            assert xattr_1 == xattr_2
-
-            # --write-unfinished will also write the unfinished one.
-            xattr_3 = must_read_xattr(path_3)
-            assert xattr_3["user.rmlint.blake2b.cksum"] == \
-                    b'36badf2227521b798b78d1bd43c62520a35b9b541547ff223f35f74b1168da2cd3c8d102aaee1a0cc217b601258d80151067cdee3a6352517b8fc7f7106902d3\x00'
-
-            # unique file which was not hashed -> does not need to be touched.
-            xattr_4 = must_read_xattr(path_4)
-            assert xattr_4 == {}
-
-        # Try clearing the attributes:
-        head, *data, footer = run_rmlint(base_options + '--xattr-clear')
-        assert len(data) == 2
-        assert must_read_xattr(path_1) == {}
-        assert must_read_xattr(path_2) == {}
-        assert must_read_xattr(path_3) == {}
-        assert must_read_xattr(path_4) == {}
-
-
 # regression test for GitHub issue #475
 # NB: this test is only effective if RM_TS_DIR is on an xattr-capable filesystem
 def test_treemerge_xattr_hardlink(usual_setup_usual_teardown):
diff --git a/tests/utils.py b/tests/utils.py
index 65cbcfd..01fb985 100644
--- a/tests/utils.py
+++ b/tests/utils.py
@@ -16,7 +16,6 @@ import shlex
 import struct
 import subprocess
 import contextlib
-import xattr
 import pytest
 
 
@@ -422,16 +421,6 @@ def create_special_fs(name, fs_type='ext4'):
         subprocess.run(unmount_command, shell=True, check=True)
 
 
-def must_read_xattr(path):
-    """
-    Read all extended attributes of a »path«.
-
-    NOTE: This will only work on non-tmpfs mounts.
-          See create_special_fs for a workaround.
-    """
-    return dict(xattr.xattr(os.path.join(TESTDIR_NAME, path)).items())
-
-
 @contextlib.contextmanager
 def assert_exit_code(status_code):
     """
