--- a/lib/utilities.h
+++ b/lib/utilities.h
@@ -129,14 +127,8 @@
 
 static inline gint64 rm_sys_preadv(int fd, const struct iovec *iov, int iovcnt,
                                    RmOff offset) {
-#if RM_IS_APPLE || RM_IS_CYGWIN
+#if RM_IS_APPLE || RM_IS_CYGWIN || RM_PLATFORM_32
     if(lseek(fd, offset, SEEK_SET) == -1) {
-        rm_log_perror("seek in emulated preadv failed");
-        return 0;
-    }
-    return readv(fd, iov, iovcnt);
-#elif RM_PLATFORM_32
-    if(lseek64(fd, offset, SEEK_SET) == -1) {
         rm_log_perror("seek in emulated preadv failed");
         return 0;
     }
