diff --git a/Makefile b/Makefile
index 852a417f3d..77f79634ec 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ include stdlib/StdlibModules
 
 CAMLC = $(BOOT_OCAMLC) $(BOOT_STDLIBFLAGS) -use-prims runtime/primitives
 CAMLOPT=$(OCAMLRUN) ./ocamlopt$(EXE) $(STDLIBFLAGS) -I otherlibs/dynlink
-ARCHES=amd64 arm64 power s390x riscv
+ARCHES=amd64 arm64 loongarch64 power s390x riscv
 VPATH = utils parsing typing bytecomp file_formats lambda middle_end \
   middle_end/closure middle_end/flambda middle_end/flambda/base_types \
   asmcomp driver toplevel tools runtime \
diff --git a/configure.ac b/configure.ac
index 167629fbbf..b5b22c775f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1610,6 +1610,7 @@ AS_IF([test x"$supports_shared_libraries" = 'xtrue'],
     [aarch64-*-freebsd*], [natdynlink=true],
     [aarch64-*-openbsd*], [natdynlink=true],
     [aarch64-*-netbsd*], [natdynlink=true],
+    [loongarch*-*-linux*], [natdynlink=true],
     [riscv*-*-linux*], [natdynlink=true],
     [[i[3456]86-*-gnu*]], [natdynlink=true],
     [x86_64-*-gnu*], [natdynlink=true])])
@@ -1731,6 +1732,8 @@ AS_CASE([$target],
     [has_native_backend=yes; arch=arm64; system=netbsd],
   [x86_64-*-cygwin],
     [has_native_backend=yes; arch=amd64; system=cygwin],
+  [loongarch64-*-linux*],
+    [has_native_backend=yes; arch=loongarch64; system=linux],
   [riscv64-*-linux*],
     [has_native_backend=yes; arch=riscv; model=riscv64; system=linux],
   [x86_64-*-gnu*],
@@ -1871,7 +1874,7 @@ default_as=''
 AS_CASE([$as_target,$ocaml_cc_vendor],
   [*-*-linux*,gcc-*],
     [AS_CASE([$as_cpu],
-      [x86_64|arm*|aarch64*|i[[3-6]]86|riscv*],
+      [x86_64|arm*|aarch64*|i[[3-6]]86|riscv*|loongarch*],
         [default_as="${toolpref}as"
         as_is_cc=false])],
   [*-*-cygwin,gcc-*],
diff --git a/runtime/caml/config.h b/runtime/caml/config.h
index 731cf8d6a8..4b5761b4c7 100644
--- a/runtime/caml/config.h
+++ b/runtime/caml/config.h
@@ -254,7 +254,7 @@ typedef uintptr_t uintnat;
    #define Cache_line_bsize 256
 #elif defined(TARGET_arm64) || defined(TARGET_power)
    #define Cache_line_bsize 128
-#elif defined(TARGET_amd64) || defined(TARGET_riscv)
+#elif defined(TARGET_amd64) || defined(TARGET_riscv) || defined(TARGET_loongarch64)
    #define Cache_line_bsize 64
 #elif (!defined(NATIVE_CODE))
    #define Cache_line_bsize 64
diff --git a/runtime/caml/stack.h b/runtime/caml/stack.h
index 169acc1e36..d3ad2fdb36 100644
--- a/runtime/caml/stack.h
+++ b/runtime/caml/stack.h
@@ -97,6 +97,16 @@
 #define CODE_POINTER_MARK_BIT 0
 #endif
 
+#ifdef TARGET_loongarch64
+/* Size of the gc_regs structure, in words.
+   See loongarch64.S and loongarch64/proc.ml for the indices */
+#define Wosize_gc_regs (2 + 23 /* int regs */ + 24 /* float regs */)
+#define Saved_return_address_raw(sp) *((intnat *)((sp) - 8))
+#define First_frame(sp) ((sp) + 16)
+#define Saved_gc_regs(sp) (*(value **)((sp) + 24))
+#define Stack_header_size 32
+#endif
+
 #ifdef CODE_POINTER_MARK_BIT
 #define CODE_POINTER_MARK_MASK ((uintnat) 1 << CODE_POINTER_MARK_BIT)
 #define Already_scanned(sp, retaddr) ((retaddr) & CODE_POINTER_MARK_MASK)
