diff --git a/stress-ramfs.c b/stress-ramfs.c
index 09d89a9..f02ce28 100644
--- a/stress-ramfs.c
+++ b/stress-ramfs.c
@@ -191,7 +191,8 @@ static int stress_ramfs_child(const args_t *args)
 #if defined(__NR_fsopen) &&	\
     defined(__NR_fsmount) &&	\
     defined(__NR_fsconfig) &&	\
-    defined(__NR_move_mount)
+    defined(__NR_move_mount) && \
+    defined(FSCONFIG_SET_STRING)
 		/*
 		 *  Use the new Linux 5.2 mount system calls
 		 */
