From b5c88a6ffefbbfde1bd212c16dd261800643c54f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=2E=20Neusch=C3=A4fer?= <j.ne@posteo.net>
Date: Tue, 29 Jul 2025 23:11:57 +0000
Subject: [PATCH] fix build with gcc15

---
 usb_modeswitch.c | 2 +-
 usb_modeswitch.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/usb_modeswitch.c b/usb_modeswitch.c
index 2d4c0aa..6c38f0e 100644
--- a/usb_modeswitch.c
+++ b/usb_modeswitch.c
@@ -1855,7 +1855,7 @@ int find_first_bulk_endpoint(int direction)
 }
 
 
-int get_current_config_value()
+int get_current_config_value(libusb_device *dev)
 {
 	SHOW_PROGRESS(output,"Get the current device configuration ...\n");
 	if (active_config != NULL) {
diff --git a/usb_modeswitch.h b/usb_modeswitch.h
index f1fd40d..9685662 100644
--- a/usb_modeswitch.h
+++ b/usb_modeswitch.h
@@ -52,7 +52,7 @@ void release_usb_device(int dummy);
 struct libusb_device* search_devices( int *numFound, int vendor, char* productList,
 		int targetClass, int configuration, int mode);
 int find_first_bulk_endpoint(int direction);
-int get_current_config_value();
+int get_current_config_value(libusb_device *dev);
 int get_interface_class();
 char* ReadParseParam(const char* FileName, char *VariableName);
 int hex2num(char c);
-- 
2.50.1

