From 022d42550282e6c10e821f52b45ad5ff9eeb59be Mon Sep 17 00:00:00 2001
From: Josh Holbrook <josh.holbrook@gmail.com>
Date: Mon, 26 May 2025 11:18:43 -0800
Subject: [PATCH] Remove fixture from trio test

---
 tests/test_trio.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tests/test_trio.py b/tests/test_trio.py
index 4ac814c..3edc54f 100644
--- a/tests/test_trio.py
+++ b/tests/test_trio.py
@@ -12,7 +12,7 @@ class PyeeTestError(Exception):
 
 
 @pytest.mark.trio
-async def test_trio_emit():
+async def test_trio_emit() -> None:
     """Test that the trio event emitter can handle wrapping
     coroutines
     """
@@ -35,7 +35,7 @@ async def event_handler():
 
 
 @pytest.mark.trio
-async def test_trio_once_emit():
+async def test_trio_once_emit() -> None:
     """Test that trio event emitters also wrap coroutines when
     using once
     """
@@ -58,7 +58,7 @@ async def event_handler():
 
 
 @pytest.mark.trio
-async def test_trio_error():
+async def test_trio_error() -> None:
     """Test that trio event emitters can handle errors when
     wrapping coroutines
     """
@@ -86,7 +86,7 @@ async def handle_error(exc):
 
 
 @pytest.mark.trio
-async def test_sync_error(event_loop):
+async def test_sync_error() -> None:
     """Test that regular functions have the same error handling as coroutines"""
 
     async with TrioEventEmitter() as ee:
