--- gitlab-runner/go.mod
+++ gitlab-runner/go.mod
@@ -81,6 +81,8 @@
 	sigs.k8s.io/yaml v1.1.0 // indirect
 )
 
-replace github.com/docker/docker v1.4.2-0.20190822180741-9552f2b2fdde => github.com/docker/engine v1.4.2-0.20190822180741-9552f2b2fdde
+replace github.com/docker/docker v1.4.2-0.20190822180741-9552f2b2fdde => github.com/docker/engine v19.03.9
 
 replace github.com/minio/go-homedir v0.0.0-20190425115525-017018655514 => gitlab.com/steveazz/go-homedir v0.0.0-20190425115525-017018655514
+
+replace github.com/vishvananda/netns => github.com/vishvananda/netns v0.0.0-20200218060606-52d707b772fe
