--- a/nspr/pr/src/misc/prnetdb.c
+++ b/nspr/pr/src/misc/prnetdb.c
@@ -58,9 +58,7 @@
  * an int, and glibc's flavor takes five arguments.
  */
 
-#if defined(SOLARIS) || (defined(BSDI) && defined(_REENTRANT)) \
-    || (defined(LINUX) && defined(_REENTRANT) \
-        && defined(__GLIBC__) && __GLIBC__ < 2)
+#if defined(SOLARIS) || (defined(BSDI) && defined(_REENTRANT))
 #define _PR_HAVE_GETPROTO_R
 #define _PR_HAVE_GETPROTO_R_POINTER
 #endif
