diff --git a/mozilla/nsprpub/configure.in b/mozilla/nsprpub/configure.in
index 25dcca1..b508762 100644
--- a/mozilla/nsprpub/configure.in
+++ b/mozilla/nsprpub/configure.in
@@ -511,7 +511,7 @@ if test "$target" != "$host"; then
 
     AC_MSG_CHECKING([whether the $host compiler ($HOST_CC $HOST_CFLAGS $HOST_LDFLAGS) works])
     AC_TRY_COMPILE([], [return(0);], 
-        [ac_cv_prog_host_cc_works=1 AC_MSG_RESULT([yes])],
+        [ac_cv_prog_host_cc_works=1; AC_MSG_RESULT([yes])],
         AC_MSG_ERROR([installation or configuration problem: $host compiler $HOST_CC cannot create executables.]) )
 
     CC=$_SAVE_CC
