diff -urN a/mozilla/nsprpub/config/config.mk b/mozilla/nsprpub/config/config.mk
--- a/mozilla/nsprpub/config/config.mk	2011-10-13 09:00:06.000000000 -0500
+++ b/mozilla/nsprpub/config/config.mk	2012-02-20 17:52:11.364823693 -0600
@@ -183,6 +183,7 @@
 RELEASE_INCLUDE_DIR = $(RELEASE_DIR)/$(BUILD_NUMBER)/$(OBJDIR_NAME)/include
 RELEASE_BIN_DIR = $(RELEASE_DIR)/$(BUILD_NUMBER)/$(OBJDIR_NAME)/bin
 RELEASE_LIB_DIR = $(RELEASE_DIR)/$(BUILD_NUMBER)/$(OBJDIR_NAME)/lib
+RELEASE_PC_DIR = $(RELEASE_LIB_DIR)/pkgconfig
 
 # autoconf.mk sets OBJ_SUFFIX to an error to avoid use before including
 # this file
diff -urN a/mozilla/nsprpub/config/Makefile.in b/mozilla/nsprpub/config/Makefile.in
--- a/mozilla/nsprpub/config/Makefile.in	2009-03-03 16:04:23.000000000 -0600
+++ b/mozilla/nsprpub/config/Makefile.in	2012-02-20 17:53:06.490819923 -0600
@@ -52,9 +52,10 @@
 
 # autoconf.mk must be deleted last (from the top-level directory)
 # because it is included by every makefile.
-DIST_GARBAGE	= nsprincl.mk nsprincl.sh nspr-config
+DIST_GARBAGE	= nsprincl.mk nsprincl.sh nspr-config nspr.pc
 
 RELEASE_BINS	= nspr-config
+RELEASE_PC	= nspr.pc
 
 include $(topsrcdir)/config/config.mk
 
diff -urN a/mozilla/nsprpub/config/nspr-config.in b/mozilla/nsprpub/config/nspr-config.in
--- a/mozilla/nsprpub/config/nspr-config.in	2005-05-10 19:53:41.000000000 -0500
+++ b/mozilla/nsprpub/config/nspr-config.in	2012-02-20 17:55:10.947811413 -0600
@@ -92,13 +92,13 @@
 
 # Set variables that may be dependent upon other variables
 if test -z "$exec_prefix"; then
-    exec_prefix=@exec_prefix@
+    exec_prefix=`pkg-config --variable=exec_prefix nspr`
 fi
 if test -z "$includedir"; then
-    includedir=@includedir@
+    includedir=`pkg-config --variable=includedir nspr`
 fi
 if test -z "$libdir"; then
-    libdir=@libdir@
+    libdir=`pkg-config --variable=libdir nspr`
 fi
 
 if test "$echo_prefix" = "yes"; then
diff -urN a/mozilla/nsprpub/config/nspr.pc.in b/mozilla/nsprpub/config/nspr.pc.in
--- a/mozilla/nsprpub/config/nspr.pc.in	1969-12-31 18:00:00.000000000 -0600
+++ b/mozilla/nsprpub/config/nspr.pc.in	2012-02-20 18:00:14.609790649 -0600
@@ -0,0 +1,10 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+libdir=@libdir@
+includedir=@includedir@
+
+Name: NSPR
+Description: The Netscape Portable Runtime
+Version: @MOD_MAJOR_VERSION@.@MOD_MINOR_VERSION@.@MOD_PATCH_VERSION@
+Libs: -L@libdir@ -lplds@MOD_MAJOR_VERSION@ -lplc@MOD_MAJOR_VERSION@ -lnspr@MOD_MAJOR_VERSION@ @OS_LIBS@
+Cflags: -I@includedir@
diff -urN a/mozilla/nsprpub/config/rules.mk b/mozilla/nsprpub/config/rules.mk
--- a/mozilla/nsprpub/config/rules.mk	2011-08-29 12:02:51.000000000 -0500
+++ b/mozilla/nsprpub/config/rules.mk	2012-02-20 18:11:56.112742680 -0600
@@ -207,10 +207,14 @@
 	rm -rf $(wildcard *.OBJ *.OBJD) dist $(ALL_TRASH) $(DIST_GARBAGE)
 	+$(LOOP_OVER_DIRS)
 
-install:: $(RELEASE_BINS) $(RELEASE_HEADERS) $(RELEASE_LIBS)
+install:: $(RELEASE_BINS) $(RELEASE_HEADERS) $(RELEASE_LIBS) $(RELEASE_PC)
 ifdef RELEASE_BINS
 	$(NSINSTALL) -t -m 0755 $(RELEASE_BINS) $(DESTDIR)$(bindir)
 endif
+ifdef RELEASE_PC
+	$(NSINSTALL) -D $(DESTDIR)$(libdir)/pkgconfig	
+	$(NSINSTALL) -t -m 0644 $(RELEASE_PC) $(DESTDIR)$(libdir)/pkgconfig/
+endif
 ifdef RELEASE_HEADERS
 	$(NSINSTALL) -t -m 0644 $(RELEASE_HEADERS) $(DESTDIR)$(includedir)/$(include_subdir)
 endif
diff -urN a/mozilla/nsprpub/configure.in b/mozilla/nsprpub/configure.in
--- a/mozilla/nsprpub/configure.in	2012-02-14 13:06:00.000000000 -0600
+++ b/mozilla/nsprpub/configure.in	2012-02-20 18:09:55.776750908 -0600
@@ -3357,6 +3357,7 @@
     config/nsprincl.mk
     config/nsprincl.sh
     config/nspr-config
+    config/nspr.pc
     lib/Makefile
     lib/ds/Makefile
     lib/libc/Makefile
