Fix tests on devices with exactly 10 cores

--- a/testcases/022-cpu-usage-tenth-cpu/expected_output.pl
+++ b/testcases/022-cpu-usage-tenth-cpu/expected_output.pl
@@ -5,7 +5,7 @@ use strict;
 use warnings;
 
 chomp(my $cpu_count = `grep -c -P '^processor\\s+:' /proc/cpuinfo`);
-if ($cpu_count < 10) {
+if ($cpu_count <= 10) {
     print "all: 00% CPU_0: 00% CPU_10: \n";
 } else {
     print "all: 00% CPU_0: 00% CPU_10: 00%\n";
