--- a/setup.py
+++ b/setup.py
@@ -5,7 +5,7 @@
 
 try:
     from setuptools import setup, Extension, find_packages
-    modules = find_packages(exclude=['examples'])
+    modules = find_packages(exclude=['examples', 'deap.tests'])
 except ImportError:
     warnings.append("warning: using distutils.core.setup, cannot use \"develop\" option")
     from distutils.core import setup, Extension
@@ -68,7 +68,7 @@
           author='deap Development Team',
           author_email='deap-users@googlegroups.com',
           url='https://www.github.com/deap',
-          packages=find_packages(exclude=['examples']),
+          packages=find_packages(exclude=['examples', 'deap.tests']),
         #   packages=['deap', 'deap.tools', 'deap.tools._hypervolume', 'deap.benchmarks', 'deap.tests'],
           platforms=['any'],
           keywords=['evolutionary algorithms', 'genetic algorithms', 'genetic programming', 'cma-es', 'ga', 'gp', 'es', 'pso'],
