--- ./Source/autotools/FindDependencies.m4.orig	2013-06-10 12:31:46.700819390 +0000
+++ ./Source/autotools/FindDependencies.m4	2013-06-10 12:32:15.101112741 +0000
@@ -373,6 +373,13 @@
     PKG_CHECK_MODULES([FREETYPE],
         [cairo-ft fontconfig >= fontconfig_required_version freetype2 >= freetype2_required_version harfbuzz >= harfbuzz_required_version])
 fi
+# HarfBuzz 0.9.18 splits harbuzz-icu into a separate library.
+# Since we support earlier HarfBuzz versions we keep this conditional by now.
+if $PKG_CONFIG --atleast-version 0.9.18 harfbuzz ; then
+    PKG_CHECK_MODULES(HARFBUZZ_ICU, harfbuzz-icu >= $harfbuzz_required_version)
+    FREETYPE_CFLAGS+=" $HARFBUZZ_ICU_CFLAGS"
+    FREETYPE_LIBS+=" $HARFBUZZ_ICU_LIBS"
+fi
 AC_SUBST([FREETYPE_CFLAGS])
 AC_SUBST([FREETYPE_LIBS])
 
