Description: Add forward declarations for byteswap8 & byteswap4.
Author: Dennis Heimbigner
Bug: https://github.com/Unidata/netcdf-c/issues/1687

diff --git a/libdispatch/dfilter.c b/libdispatch/dfilter.c
index 8938464..8efe591 100644
--- a/libdispatch/dfilter.c
+++ b/libdispatch/dfilter.c
@@ -20,6 +20,9 @@
 #include "hdf5internal.h"
 #endif
 
+static void byteswap8(unsigned char*);
+static void byteswap4(unsigned char*);
+
 /*
 Unified filter related code
 */
