diff --git a/libhdf5/hdf5debug.c b/libhdf5/hdf5debug.c
index fc8e206..cf9e1e3 100644
--- a/libhdf5/hdf5debug.c
+++ b/libhdf5/hdf5debug.c
@@ -5,7 +5,7 @@
 #include "config.h"
 #include <stdarg.h>
 #include <stdio.h>
-#if !defined _WIN32 && !defined __CYGWIN__
+#if !defined _WIN32 && !defined __CYGWIN__ && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -15,7 +15,7 @@
 
 #define STSIZE 1000
 
-#if !defined _WIN32 && !defined __CYGWIN__
+#if !defined _WIN32 && !defined __CYGWIN__ && defined(__GLIBC__)
 static void* stacktrace[STSIZE];
 #endif
 
