--- ./lib/nettle/Makefile.in.orig	2013-06-10 17:39:26.019517541 +0000
+++ ./lib/nettle/Makefile.in	2013-06-10 17:39:38.629643690 +0000
@@ -214,7 +214,7 @@
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 LTLIBRARIES = $(noinst_LTLIBRARIES)
-libcrypto_la_LIBADD =
+libcrypto_la_LIBADD = -lgmp
 am_libcrypto_la_OBJECTS = pk.lo mpi.lo mac.lo cipher.lo rnd.lo init.lo \
 	egd.lo
 libcrypto_la_OBJECTS = $(am_libcrypto_la_OBJECTS)
