Failing test: long-dns-crq

--- a/tests/cert-tests/Makefile.in
+++ b/tests/cert-tests/Makefile.in
@@ -311,7 +311,7 @@
 	key-invalid.sh pkcs8-eddsa.sh certtool-subca.sh \
 	certtool-verify-profiles.sh x509-duplicate-ext.sh \
 	x25519-and-x448.sh reject-negative-serial.sh certtool-utf8.sh \
-	crq.sh cert-non-digits-time.sh reject-invalid-time.sh \
+	cert-non-digits-time.sh reject-invalid-time.sh \
 	tolerate-invalid-time.sh provable-dh-default.sh \
 	template-test.sh pem-decoding.sh othername-test.sh \
 	krb5-test.sh sha3-test.sh md5-test.sh tlsfeature-test.sh \
@@ -2432,7 +2432,7 @@
 	certtool-verify-profiles.sh x509-duplicate-ext.sh \
 	x25519-and-x448.sh reject-negative-serial.sh key-id.sh \
 	ecdsa.sh pkcs8-invalid.sh key-invalid.sh pkcs8-decode.sh \
-	pkcs8.sh pkcs8-eddsa.sh certtool-utf8.sh crq.sh \
+	pkcs8.sh pkcs8-eddsa.sh certtool-utf8.sh \
 	$(am__append_1) $(am__append_2) $(am__append_3) \
 	$(am__append_4) $(am__append_5) $(am__append_6) \
 	$(am__append_7) certtool-rsa-pss.sh certtool-eddsa.sh
