Busybox date does not support %N, this is GNU extension.
--- a/tests/scripts/common.sh
+++ b/tests/scripts/common.sh
@@ -61,7 +61,7 @@
 # Find a port number not currently in use.
 GETPORT='rc=0; unset myrandom
     if test -n "$RANDOM"; then myrandom=$(($RANDOM + $RANDOM)); fi
-    if test -z "$myrandom"; then myrandom=$(date +%N | sed s/^0*//); fi
+    if test -z "$myrandom"; then myrandom=$(date +%s | sed s/^0*//); fi
     if test -z "$myrandom"; then myrandom=0; fi
     while test $rc = 0;do
 	PORT="$(((($$<<15)|$myrandom) % 63001 + 2000))"
