diff --git a/lib/posix/posix.nim b/lib/posix/posix.nim
index e206447cc57a..302263f724e8 100644
--- a/lib/posix/posix.nim
+++ b/lib/posix/posix.nim
@@ -846,7 +846,7 @@ var
   FE_UPWARD* {.importc, header: "<fenv.h>".}: cint
   FE_DFL_ENV* {.importc, header: "<fenv.h>".}: cint
 
-when not defined(haiku):
+when false:
   var
     MM_HARD* {.importc, header: "<fmtmsg.h>".}: cint
       ## Source of the condition is hardware.
diff --git a/lib/pure/osproc.nim b/lib/pure/osproc.nim
index 6e250f9d547a..bc21f963e73f 100644
--- a/lib/pure/osproc.nim
+++ b/lib/pure/osproc.nim
@@ -722,8 +722,7 @@ elif not defined(useNimRtl):
       chck posix_spawnattr_setsigmask(attr, mask)
       chck posix_spawnattr_setpgroup(attr, 0'i32)
 
-      chck posix_spawnattr_setflags(attr, POSIX_SPAWN_USEVFORK or
-                                          POSIX_SPAWN_SETSIGMASK or
+      chck posix_spawnattr_setflags(attr, POSIX_SPAWN_SETSIGMASK or
                                           POSIX_SPAWN_SETPGROUP)
 
       if not data.optionPoParentStreams:
