diff --git a/lib/posix/posix.nim b/lib/posix/posix.nim
index e206447cc57a..1af768d82a5c 100644
--- a/lib/posix/posix.nim
+++ b/lib/posix/posix.nim
@@ -1737,7 +1737,7 @@ when hasSpawnH:
     POSIX_SPAWN_SETSIGDEF* {.importc, header: "<spawn.h>".}: cint
     POSIX_SPAWN_SETSIGMASK* {.importc, header: "<spawn.h>".}: cint
 
-  when defined(linux):
+  when not defined(linux):
     # better be safe than sorry; Linux has this flag, macosx doesn't, don't
     # know about the other OSes
     var POSIX_SPAWN_USEVFORK* {.importc, header: "<spawn.h>".}: cint
