From 86891bf33b9aa2a484f546aee335eb4aae39c2b7 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Tue, 24 May 2022 11:03:33 +0200
Subject: [PATCH] FirstRun: add missing Alpine packages for KDE Pim

---
 modules/ksb/FirstRun.pm | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/modules/ksb/FirstRun.pm b/modules/ksb/FirstRun.pm
index a9aa922..80f18fd 100644
--- a/modules/ksb/FirstRun.pm
+++ b/modules/ksb/FirstRun.pm
@@ -739,7 +739,9 @@ autoconf
 automake
 bison
 boost-dev
+clang-extra-tools
 cmake
+cyrus-sasl-dev
 doxygen
 eudev-dev
 flex
@@ -749,6 +751,7 @@ gpgme-dev
 grantlee-dev
 gstreamer-dev
 gst-plugins-base-dev
+libaccounts-qt-dev
 libcanberra-dev
 libdbusmenu-qt-dev
 libdmtx-dev
@@ -769,6 +772,7 @@ lm-sensors-dev
 lmdb-dev
 networkmanager-dev
 openjpeg-dev
+openldap-dev
 perl
 perl-io-socket-ssl
 perl-uri
@@ -776,6 +780,7 @@ perl-yaml-libyaml
 polkit-elogind-dev
 pulseaudio-dev
 py3-sphinx
+qca-dev
 qt5-qtbase-dev
 qt5-qtdeclarative-dev
 qt5-qtquickcontrols2-dev
@@ -786,6 +791,7 @@ qt5-qtsvg-dev
 qt5-qttools-dev
 qt5-qtwayland-dev
 qt5-qtx11extras-dev
+signond-dev
 texinfo
 wayland-protocols
 xapian-core-dev
-- 
GitLab

