only x86_64 has prebuilt musl node all others are glibc

--- a/tests/nodeenv_test.py
+++ b/tests/nodeenv_test.py
@@ -21,7 +21,7 @@
     subprocess.check_call([
         # Enable coverage
         'coverage', 'run', '-p',
-        '-m', 'nodeenv', '--prebuilt', nenv_path,
+        '-m', 'nodeenv', '--node=system', nenv_path,
     ])
     assert os.path.exists(nenv_path)
     activate = pipes.quote(os.path.join(nenv_path, 'bin', 'activate'))
