See https://gitlab.alpinelinux.org/alpine/aports/-/issues/16106

diff -upr pciutils-3.12.0.orig/lib/sysfs.c pciutils-3.12.0/lib/sysfs.c
--- pciutils-3.12.0.orig/lib/sysfs.c	2024-05-12 12:49:17.273816524 +0200
+++ pciutils-3.12.0/lib/sysfs.c	2024-05-12 12:49:31.743853491 +0200
@@ -14,6 +14,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <libgen.h>
 #include <stdarg.h>
 #include <unistd.h>
 #include <errno.h>
