--- ./lib/pread.h.orig
+++ ./lib/pread.h
@@ -12,8 +12,8 @@
  *  don't define it.
  */
 
-#if defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0
-/* glibc 2.1 or newer -> pread/pwrite supported automatically */
+#if (defined(__GLIBC__) && __GLIBC__ == 2 && __GLIBC_MINOR__ > 0) || (_POSIX_VERSION >= 200809L)
+ /* glibc 2.1 or newer or posix 2008 -> pread/pwrite supported automatically */
 
 #elif defined(i386) && defined(__GLIBC__)
 /* glibc 2.0 on i386 -> call syscalls directly */
