--- webkitgtk-2.10.3/Source/WebCore/platform/linux/MemoryPressureHandlerLinux.cpp.orig
+++ webkitgtk-2.10.3/Source/WebCore/platform/linux/MemoryPressureHandlerLinux.cpp
@@ -202,7 +202,9 @@
 void MemoryPressureHandler::platformReleaseMemory(Critical)
 {
     ReliefLogger log("Run malloc_trim");
+#if defined(__GLIBC__)
     malloc_trim(0);
+#endif
 }
 
 void MemoryPressureHandler::ReliefLogger::platformLog()
