--- webkitgtk-2.10.3/Source/JavaScriptCore/runtime/Options.cpp.orig
+++ webkitgtk-2.10.3/Source/JavaScriptCore/runtime/Options.cpp
@@ -29,7 +29,6 @@
 #include "HeapStatistics.h"
 #include <algorithm>
 #include <limits>
-#include <math.h>
 #include <mutex>
 #include <stdlib.h>
 #include <string.h>
@@ -610,7 +609,7 @@
     case Options::Type::unsignedType:
         return m_entry.unsignedVal == other.m_entry.unsignedVal;
     case Options::Type::doubleType:
-        return (m_entry.doubleVal == other.m_entry.doubleVal) || (isnan(m_entry.doubleVal) && isnan(other.m_entry.doubleVal));
+        return (m_entry.doubleVal == other.m_entry.doubleVal) || (std::isnan(m_entry.doubleVal) && std::isnan(other.m_entry.doubleVal));
     case Options::Type::int32Type:
         return m_entry.int32Val == other.m_entry.int32Val;
     case Options::Type::optionRangeType:
