--- a/Source/JavaScriptCore/assembler/LinkBuffer.cpp.orig	2020-09-02 11:04:43.517875330 +0200
+++ b/Source/JavaScriptCore/assembler/LinkBuffer.cpp	2020-09-02 11:04:25.930873479 +0200
@@ -377,7 +377,7 @@
     if (useFastJITPermissions()) {
         ASSERT(codeOutData == outData);
         if (UNLIKELY(Options::dumpJITMemoryPath()))
-            dumpJITMemory(outData, outData, m_size);
+            JSC::dumpJITMemory(outData, outData, m_size);
     } else {
         ASSERT(codeOutData != outData);
         performJITMemcpy(codeOutData, outData, m_size);
