--- a/utils/kamctl/kamdbctl.base	Tue Dec 22 07:01:57 2009
+++ b/utils/kamctl/kamdbctl.base	Tue Dec 22 07:02:26 2009
@@ -32,9 +32,9 @@
 
 # Used by dbtext and db_berkeley to define tables to be created, used by
 # postgres to do the grants
-STANDARD_TABLES=${STANDARD_TABLES:-version acc dbaliases lcr domain grp
-		uri speed_dial gw pdt subscriber location re_grp trusted
-		address missed_calls usr_preferences aliases silo dialog
+STANDARD_TABLES=${STANDARD_TABLES:-version acc dbaliases lcr domain grp \
+		uri speed_dial gw pdt subscriber location re_grp trusted \
+		address missed_calls usr_preferences aliases silo dialog \
 		dispatcher dialplan}
 EXTRA_TABLES=${EXTRA_TABLES:-imc_members imc_rooms cpl sip_trace domainpolicy carrierroute
 		carrier_name domain_name carrierfailureroute userblacklist globalblacklist htable purplemap}
