diff --git a/asteroid-calendar.pro b/asteroid-calendar.pro
index 8b30ba2..dd5a9d8 100644
--- a/asteroid-calendar.pro
+++ b/asteroid-calendar.pro
@@ -1,5 +1,6 @@
 TARGET = asteroid-calendar
-CONFIG += asteroidapp
+CONFIG += asteroidapp link_pkgconfig
+PKGCONFIG += asteroidapp
 
 SOURCES +=     main.cpp
 RESOURCES +=   resources.qrc
@@ -7,3 +8,6 @@ OTHER_FILES += main.qml EventDialog.qml MonthSelector.qml
 
 lupdate_only{ SOURCES += i18n/asteroid-calendar.desktop.h }
 TRANSLATIONS = $$files(i18n/$$TARGET.*.ts)
+
+target.path = /usr/bin/
+INSTALLS += target
