From 419e10eadde5048a99e40f23a37f4d9718496709 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Wed, 23 Sep 2015 07:49:23 +0200
Subject: [PATCH] fastscan: fix Canal Digitaal HD on astra3

---
 data/conf/fastscan | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/data/conf/fastscan b/data/conf/fastscan
index 56aad0f..ca0d361 100644
--- a/data/conf/fastscan
+++ b/data/conf/fastscan
@@ -83,19 +83,19 @@
   {
     "name": "Canal Digitaal HD",
     "position": 235,
-    "frequency": 12187000,
+    "frequency": 12070000,
     "symbolrate" : 27500000,
     "polarisation" : "H",
-    "delsys" : "dvbs2",
+    "delsys" : "dvbs",
     "pid": 901
   },
   {
     "name": "TV Vlaanderen HD",
     "position": 235,
-    "frequency": 12187000,
+    "frequency": 12070000,
     "symbolrate" : 27500000,
     "polarisation" : "H",
-    "delsys" : "dvbs2",
+    "delsys" : "dvbs",
     "pid": 911
   },
     {
