--- ./drivers/dahdi/voicebus/voicebus.h.orig
+++ ./drivers/dahdi/voicebus/voicebus.h
@@ -89,7 +89,7 @@
 	void (*handle_receive)(struct voicebus *vb, struct list_head *buffers);
 	void (*handle_transmit)(struct voicebus *vb, struct list_head *buffers);
 	void (*handle_error)(struct voicebus *vb);
-};
+} __no_const;
 
 /**
  * struct voicebus_descriptor_list - A single descriptor list.
--- ./drivers/dahdi/wctc4xxp/base.c.orig
+++ ./drivers/dahdi/wctc4xxp/base.c
@@ -3193,7 +3193,7 @@
 	return ret;
 }
 
-static void wctc4xxp_setup_file_operations(struct file_operations *fops)
+static void wctc4xxp_setup_file_operations(file_operations_no_const *fops)
 {
 	fops->owner = THIS_MODULE;
 	fops->read =  wctc4xxp_read;
