--- dahdi-linux-2.6.0/drivers/dahdi/zaphfc.c.orig	2012-01-25 14:08:58.000000000 +0200
+++ dahdi-linux-2.6.0/drivers/dahdi/zaphfc.c	2012-01-25 14:29:26.000000000 +0200
@@ -122,7 +122,10 @@
     }
     spin_unlock_irqrestore(&hfctmp->lock,flags);
     if (hfctmp->ztdev != NULL) {
-	dahdi_unregister(&hfctmp->ztdev->span);
+	if (hfctmp->ztdev->ddev) {
+	    dahdi_unregister_device(hfctmp->ztdev->ddev);
+	    dahdi_free_device(hfctmp->ztdev->ddev);
+	}
 	vfree(hfctmp->ztdev);
 	printk(KERN_INFO "unregistered from DAHDI.\n");
     }
@@ -680,9 +683,15 @@
     struct hfc_card *hfctmp = zthfc->card;
     int i;
 
+    zthfc->ddev = dahdi_create_device();
+    if (!zthfc->ddev)
+	return -ENOMEM;
+
+    zthfc->ddev->manufacturer = "HFC";
+
     memset(&zthfc->span, 0x0, sizeof(struct dahdi_span)); // you never can tell...
 
-    sprintf(zthfc->span.name, "ZTHFC%d", hfc_dev_count + 1);
+    sprintf(zthfc->span.name, "ZTHFC/%d", hfc_dev_count + 1);
     if (hfctmp->regs.nt_mode == 1) {
 	sprintf(zthfc->span.desc, "HFC-S PCI A ISDN card %d [NT]", hfc_dev_count + 1);
     } else {
@@ -702,13 +711,14 @@
 
     for (i = 0; i < zthfc->span.channels; i++) {
 	memset(&(zthfc->chans[i]), 0x0, sizeof(struct dahdi_chan));
-	sprintf(zthfc->chans[i].name, "ZTHFC%d/%d/%d", hfc_dev_count + 1,0,i + 1);
+	sprintf(zthfc->chans[i].name, "ZTHFC/%d/%d/%d", hfc_dev_count + 1,0,i + 1);
 	zthfc->chans[i].pvt = zthfc;
 	zthfc->chans[i].sigcap =  DAHDI_SIG_EM | DAHDI_SIG_CLEAR | DAHDI_SIG_FXSLS | DAHDI_SIG_FXSGS | DAHDI_SIG_FXSKS | DAHDI_SIG_FXOLS | DAHDI_SIG_FXOGS | DAHDI_SIG_FXOKS | DAHDI_SIG_CAS | DAHDI_SIG_SF;
 	zthfc->chans[i].chanpos = i + 1; 
     }
 
-    if (dahdi_register(&zthfc->span,0)) {
+    list_add_tail(&zthfc->span.device_node, &zthfc->ddev->spans);
+    if (dahdi_register_device(zthfc->ddev, &zthfc->card->pcidev->dev)) {
 	printk(KERN_CRIT "unable to register DAHDI device!\n");
 	return -1;
     }
--- dahdi-linux-2.6.0/drivers/dahdi/zaphfc.h.orig	2012-01-25 14:23:37.000000000 +0200
+++ dahdi-linux-2.6.0/drivers/dahdi/zaphfc.h	2012-01-25 14:24:33.000000000 +0200
@@ -339,6 +339,7 @@
     struct dahdi_span span;
     struct dahdi_chan chans[3];
     struct dahdi_chan *_chans[3];
+    struct dahdi_device *ddev;
     struct hfc_card *card;
 } dahdi_hfc;
 
