--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -169,9 +169,9 @@
   SET (ARCH "i386")
 ENDIF (NOT WIN32)
 
-#IF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
-#  SET (ARCH "armhf")
-#ENDIF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
+IF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
+  SET (ARCH "aarch64")
+ENDIF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
 
 MESSAGE (STATUS "*** Build Architecture is ${ARCH}")
 SET(PACKAGE_RELEASE "1" CACHE STRING "Package release number")
