https://github.com/OpenCPN/OpenCPN/issues/676

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -534,6 +534,9 @@
             IF(CMAKE_BUILD_TYPE MATCHES "Rel*")
 		    MESSAGE (STATUS "Using Linux crash reporting")
                 ADD_DEFINITIONS(-DLINUX_CRASHRPT)
+                IF(EXISTS /etc/alpine-release)
+                    SET(EXTRA_LIBS ${EXTRA_LIBS} "execinfo")
+                ENDIF(EXISTS alpine-release)
             ENDIF(CMAKE_BUILD_TYPE MATCHES "Rel*")
         ENDIF(NOT APPLE AND NOT QT_ANDROID AND NOT MINGW)
     ENDIF(MSVC)
