index 967e558..27bad69 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -489,6 +489,9 @@ IF(OCPN_USE_CRASHREPORT)
             IF(CMAKE_BUILD_TYPE MATCHES "Rel*")
                    MESSAGE (STATUS "Using Linux crash reporting")
                 ADD_DEFINITIONS(-DLINUX_CRASHRPT)
+                IF(EXISTS /etc/alpine-release)
+                    SET(EXTRA_LIBS ${EXTRA_LIBS} "execinfo")
+                ENDIF(EXISTS alpine-release)
             ENDIF(CMAKE_BUILD_TYPE MATCHES "Rel*")
         ENDIF(NOT APPLE AND NOT QT_ANDROID AND NOT MINGW)
     ENDIF(MSVC
