From 9ad3d30db042f0a89673195de92d0c81a50affd3 Mon Sep 17 00:00:00 2001
From: didier <didier@users.sourceforge.net>
Date: Mon, 17 Dec 2018 15:12:38 +0100
Subject: [PATCH] compile without wxUSE_XLOCALE

---
 include/OCPNPlatform.h | 2 +-
 src/OCPNPlatform.cpp   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/OCPNPlatform.h b/include/OCPNPlatform.h
index d6bc4915d1..751ae8590c 100644
--- a/include/OCPNPlatform.h
+++ b/include/OCPNPlatform.h
@@ -160,9 +160,9 @@ class OCPNPlatform
 
     void SetLocaleSearchPrefixes( void );
     wxString GetDefaultSystemLocale();
-    wxString GetAdjustedAppLocale();
     
 #if wxUSE_XLOCALE    
+    wxString GetAdjustedAppLocale();
     wxString ChangeLocale(wxString &newLocaleID, wxLocale *presentLocale, wxLocale** newLocale);
 #endif
     
diff --git a/src/OCPNPlatform.cpp b/src/OCPNPlatform.cpp
index 785ff183b5..70ac73306d 100644
--- a/src/OCPNPlatform.cpp
+++ b/src/OCPNPlatform.cpp
@@ -818,6 +818,7 @@ wxString OCPNPlatform::GetDefaultSystemLocale()
 }
 
 
+#if wxUSE_XLOCALE || !wxCHECK_VERSION(3,0,0)
 wxString OCPNPlatform::GetAdjustedAppLocale()
 {
     wxString adjLocale = g_locale;
@@ -851,7 +852,6 @@ wxString OCPNPlatform::GetAdjustedAppLocale()
 }
 
 
-#if wxUSE_XLOCALE || !wxCHECK_VERSION(3,0,0)
 
 wxString OCPNPlatform::ChangeLocale(wxString &newLocaleID, wxLocale *presentLocale, wxLocale** newLocale)
 {
