diff --git a/include/zeroconf-detail.hpp b/include/zeroconf-detail.hpp
index 60ae125..3e7607a 100644
--- a/include/zeroconf-detail.hpp
+++ b/include/zeroconf-detail.hpp
@@ -11,6 +11,9 @@
 #include <vector>
 #include <memory>
 #include <chrono>
+#include <ctime>
+
+#include <sys/select.h>
 
 #if (defined(_WIN32) || defined(__WIN32__)) && !defined(WIN32)
 #define WIN32
@@ -385,4 +388,4 @@ inline bool Resolve(const std::string& serviceName, time_t scanTime,
 }  // namespace Detail
 }  // namespace Zeroconf
 
-#endif  // ZEROCONF_DETAIL_HPP
\ No newline at end of file
+#endif  // ZEROCONF_DETAIL_HPP
