--- a/include/ocpn_plugin.h
+++ b/include/ocpn_plugin.h
@@ -55,6 +55,7 @@
 #include <wx/bitmap.h>
 #endif  // ocpnUSE_SVG
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 #include <unordered_map>
--- a/libs/sound/include/SoundFileLoader.h
+++ b/libs/sound/include/SoundFileLoader.h
@@ -25,6 +25,7 @@
 #ifndef SOUND_FILE_LOADER_H
 #define SOUND_FILE_LOADER_H
 
+#include <cstdint>
 #include <functional>
 #include <memory>
 #include <string>
