diff --git a/src/crashprint.cpp b/src/crashprint.cpp
index 2ab4e0b..3ef82c4 100644
--- a/src/crashprint.cpp
+++ b/src/crashprint.cpp
@@ -32,7 +32,6 @@
 
 //! standard header
 #if defined(__linux__)
-#include <execinfo.h>    // Needed for backtrace
 #include <cxxabi.h>      // Needed for __cxa_demangle
 #include <unistd.h>
 #endif
@@ -79,15 +78,10 @@ void wxCrashPrint::Report () {
     wxString appname = wxTheApp->GetAppName();
 
     // get the backtrace with symbols
-    int btCount;
-    btCount = backtrace (m_btBuffer, maxBtCount);
+    int btCount = -1;
     if (btCount < 0) {
         wxPrintf (_T("\n%s: Backtrace could not be created\n"), appname.c_str());
     }
-    m_btStrings = backtrace_symbols (m_btBuffer, btCount);
-    if (!m_btStrings) {
-        wxPrintf (_T("\n%s: Backtrace could not get symbols\n"), appname.c_str());
-    }
 
     // print backtrace announcement
     wxPrintf (_T("\n*** %s (%s) crashed ***, see backtrace!\n"), appname.c_str(), wxVERSION_STRING);
