--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -87,7 +87,7 @@
 #Slamd64:   /etc/slamd64-version
 #Gentoo: /etc/gentoo-release
 #Ubuntu: /etc/debian_version
-
+#Alpine: /etc/alpine-release
 # Based on code from nohal
 IF (NOT WIN32)
   # default
@@ -164,6 +164,13 @@
 #      ENDIF (CMAKE_OSX_64)
     ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
   ENDIF()
+  IF (EXISTS /etc/alpine-release)
+    IF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
+      IF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+        SET (ARCH "arm64")
+      ENDIF (CMAKE_SIZEOF_VOID_P MATCHES "8")
+    ENDIF (CMAKE_SYSTEM_PROCESSOR MATCHES "arm*")
+  ENDIF (EXISTS /etc/alpine-release)
 ELSE (NOT WIN32)
   # On WIN32 probably CMAKE_SIZEOF_VOID_P EQUAL 8, but we don't use it at all now...
   SET (ARCH "i386")
