diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b31994..a4f8bf5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1276,8 +1276,7 @@ target_link_libraries(${PACKAGE_NAME} PRIVATE pico_sha2)
 add_subdirectory(libs/mongoose)
 target_link_libraries(${PACKAGE_NAME} PUBLIC ocpn::mongoose)
 
-add_subdirectory(libs/rapidjson)
-target_link_libraries(${PACKAGE_NAME} PUBLIC ocpn::rapidjson)
+find_package(RapidJSON)
 
 add_subdirectory(libs/mDNS)
 target_link_libraries(${PACKAGE_NAME} PUBLIC ocpn::mDNS)
@@ -3152,7 +3151,6 @@ if (NOT QT_ANDROID)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::N2KParser)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::nmea0183)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::pugixml)
-  target_link_libraries(opencpn-cmd PRIVATE ocpn::rapidjson)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::s52plib)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::serial)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::sound)
