--- restool-2.4.0.orig/Makefile
+++ restool-2.4.0/Makefile
@@ -62,7 +62,6 @@
 	install -D -m 755 scripts/ls-debug $(DESTDIR)$(bindir)/ls-debug
 	$(foreach symlink, $(RESTOOL_SCRIPT_SYMLINKS), sh -c "cd $(DESTDIR)$(bindir) && ln -sf ls-main $(symlink)" ;)
 	install -D -m 755 scripts/restool_completion.sh $(DESTDIR)$(bindir_completion)/restool
-	install -m 0644 -D $(MANPAGE) $(call get_manpage_destination,$(MANPAGE))
 
 clean:
 	rm -f $(OBJ) $(MANPAGE) \
