--- old/Makefile
+++ new/Makefile
@@ -106,7 +106,7 @@
 	@mkdir -p $(@D)
 	$(CXX) $(SNAPPY_CXX_OPTS) $(CXXFLAGS) -c $< -o $@
 
-$(SNAPPY_OUT)/BitShuffleNative.o: $(SRC)/org/xerial/snappy/BitShuffleNative.cpp $(SRC)/org/xerial/snappy/BitShuffleNative.h
+$(SNAPPY_OUT)/BitShuffleNative.o: $(SRC)/org/xerial/snappy/BitShuffleNative.cpp $(SRC)/org/xerial/snappy/BitShuffleNative.h $(BITSHUFFLE_UNPACKED)
 	@mkdir -p $(@D)
 	$(CXX) $(CXXFLAGS) -c $< -o $@
 
