 cmake/GetxxHash.cmake | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/GetxxHash.cmake b/cmake/GetxxHash.cmake
index 52e525107..5eac0f051 100644
--- a/cmake/GetxxHash.cmake
+++ b/cmake/GetxxHash.cmake
@@ -1,8 +1,8 @@
 cmake_minimum_required ( VERSION 3.17 FATAL_ERROR )
 
 set ( XXH_REPO "https://github.com/manticoresoftware/xxHash" )
-set ( XXH_REV "71c6c93" )
-set ( XXH_SRC_MD5 "705f9786460daf8712670eff92d94778" )
+set ( XXH_REV "72997b0" )
+set ( XXH_SRC_MD5 "90a5d68bf9d3adbb6d6155bbf12805de" )
 
 set ( XXH_GITHUB "${XXH_REPO}/archive/${XXH_REV}.zip" )
 set ( XXH_BUNDLE "${LIBS_BUNDLE}/xxHash-${XXH_REV}.zip" )
