From a1043543bffb78c8bd95e7ef1709ed4eac9a9a2a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rapha=C3=ABl=20Pinson?= <raphael.pinson@camptocamp.com>
Date: Mon, 13 Jul 2015 10:04:38 +0200
Subject: [PATCH 02/19] Shellvars: Add test for [[]]

---
 lenses/tests/test_shellvars.aug | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lenses/tests/test_shellvars.aug b/lenses/tests/test_shellvars.aug
index 778a8d2..a2d4f28 100644
--- a/lenses/tests/test_shellvars.aug
+++ b/lenses/tests/test_shellvars.aug
@@ -541,6 +541,10 @@ fi\n" =
     { "@condition" = "-f $FILENAME"
       { "type" = "[" } }
 
+  test Shellvars.lns get "[[ -f $FILENAME ]]\n" =
+    { "@condition" = "-f $FILENAME"
+      { "type" = "[[" } }
+
 (* Local Variables: *)
 (* mode: caml       *)
 (* End:             *)
-- 
2.5.0

