From 34d33eda74c9171aadf25686307f73fb307c7626 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Mon, 8 Dec 2014 12:43:51 +0200
Subject: [PATCH 05/19] Shellvars: guard against comments

---
 lenses/shellvars.aug | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lenses/shellvars.aug b/lenses/shellvars.aug
index c767067..78b1d16 100644
--- a/lenses/shellvars.aug
+++ b/lenses/shellvars.aug
@@ -34,7 +34,7 @@ module Shellvars =
   let xchgs   = Build.xchgs
   let semicol = del /;?/ ""
 
-  let char  = /[^`;() '"\t\n\\]|\\\\(.|\n)/
+  let char  = /[^`;()'"\n\\# \t]#*|\\\\(.|\n)/
   let dquot =
        let char = /[^"\\]|\\\\./ | Rx.cl
     in "\"" . char* . "\""                    (* " Emacs, relax *)
-- 
2.5.0

