From 1eb1db77c4c447ab45d2fff7eece4e87da001909 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Mon, 13 Jul 2015 14:48:18 +0300
Subject: [PATCH 12/19] Shellvars: use sto_to_semicol in condition lens

---
 lenses/shellvars.aug | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/lenses/shellvars.aug b/lenses/shellvars.aug
index ed9cc9c..0384a6f 100644
--- a/lenses/shellvars.aug
+++ b/lenses/shellvars.aug
@@ -107,9 +107,8 @@ module Shellvars =
     . ( Util.del_ws_spc . store Rx.integer )?
 
   let condition =
-       let sto_cond = store /[^[#; \t\n][^#;\n]+[^]#; \t\n]|[^[]#; \t\n]+/
-    in let cond (start:string) (end:string) = [ label "type" . store start ]
-                                            . Util.del_ws_spc . sto_cond
+       let cond (start:string) (end:string) = [ label "type" . store start ]
+                                            . Util.del_ws_spc . sto_to_semicol
                                             . Util.del_ws_spc . Util.del_str end
     in Util.indent . label "@condition" . (cond "[" "]" | cond "[[" "]]")
 
-- 
2.5.0

