diff --git a/CMakeLists.txt b/CMakeLists.txt
index f7a0dfe..d63b93e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -347,7 +347,7 @@ set (UNINSTALL_PROG_DIR "${CMAKE_INSTALL_FULL_DATADIR}/mongo-c-driver")
 #
 # This needs to be last (after all other add_subdirectory calls) to ensure that
 # the generated uninstall program is complete and correct
-add_subdirectory (generate_uninstall)
+# add_subdirectory (generate_uninstall)
 
 add_custom_target (uninstall
    COMMAND "${UNINSTALL_PROG_DIR}/${UNINSTALL_PROG}"
