See https://github.com/wbond/oscrypto/issues/58

diff -upr oscrypto-1.3.0.orig/tests/test_asymmetric.py oscrypto-1.3.0/tests/test_asymmetric.py
--- oscrypto-1.3.0.orig/tests/test_asymmetric.py	2022-03-18 13:13:34.580640689 +0100
+++ oscrypto-1.3.0/tests/test_asymmetric.py	2022-03-18 13:16:22.767789864 +0100
@@ -137,11 +137,6 @@ class AsymmetricTests(unittest.TestCase)
         self.assertIsInstance(private_reloaded, asymmetric.PrivateKey)
         self.assertEqual('rsa', private_reloaded.algorithm)
 
-    def test_load_rsa_pss_cert(self):
-        cert = asymmetric.load_certificate(os.path.join(fixtures_dir, 'keys/test-pss.crt'))
-        self.assertEqual('rsassa_pss', cert.algorithm)
-        self.assertEqual(2048, cert.bit_size)
-
     def test_dh_generate(self):
         dh_parameters = asymmetric.generate_dh_parameters(512)
         self.assertIsInstance(dh_parameters, algos.DHParameters)
