diff --git a/src/Makefile.in b/src/Makefile.in
index 76ec142..1911ac2 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -85,10 +85,10 @@ globule.o:
 	$(CC) $(CFLAGS) globule.c -c 
 
 install: cleanbin
-	if [ ! -d $(CONFDIR) ]; then mkdir -p $(CONFDIR); fi
-	if [ ! -e $(CONFDIR)/reaver.db ]; then cp reaver.db $(CONFDIR)/reaver.db && chmod -R a+rw $(CONFDIR); fi
-	if [ -e wash ]; then cp wash @bindir@/wash; fi
-	if [ -e reaver ]; then cp reaver @bindir@/reaver; fi
+	if [ ! -d $(CONFDIR) ]; then mkdir -p $(DESTDIR)$(CONFDIR); fi
+	if [ ! -e $(CONFDIR)/reaver.db ]; then cp reaver.db $(DESTDIR)$(CONFDIR)/reaver.db && chmod -R a+rw $(DESTDIR)$(CONFDIR); fi
+	if [ -e wash ]; then cp wash $(DESTDIR)@bindir@/wash; fi
+	if [ -e reaver ]; then cp reaver $(DESTDIR)@bindir@/reaver; fi
 
 clean:
 	rm -f *~ *.o reaver wash
