--- a/cmd/promu.go
+++ b/cmd/promu.go
@@ -73,11 +73,11 @@
 	config := &Config{}
 	config.Build.Binaries = []Binary{{Name: projInfo.Name, Path: "."}}
 	config.Build.Prefix = "."
-	config.Build.Static = true
+	config.Build.Static = false
 	config.Crossbuild.Platforms = defaultPlatforms
 	config.Tarball.Prefix = "."
 	config.Go.Version = "1.12"
-	config.Go.CGo = false
+	config.Go.CGo = true
 	config.Repository.Path = projInfo.Repo
 
 	return config
