diff -crB dart-sass-embedded-1.57.1/tool/grind.dart dart-sass-embedded-1.57.1-patched/tool/grind.dart
*** dart-sass-embedded-1.57.1/tool/grind.dart	Mon Dec 19 20:53:43 2022
--- dart-sass-embedded-1.57.1-patched/tool/grind.dart	Fri Dec 23 13:46:34 2022
***************
*** 55,61 ****
  ''');
    } else {
      File('build/protoc-gen-dart')
!         .writeAsStringSync('dart pub run protoc_plugin "\$@"');
      run('chmod', arguments: ['a+x', 'build/protoc-gen-dart']);
    }
  
--- 55,62 ----
  ''');
    } else {
      File('build/protoc-gen-dart')
!         .writeAsStringSync('''#!/bin/sh
! dart pub run protoc_plugin "\$@" < /dev/stdin''');
      run('chmod', arguments: ['a+x', 'build/protoc-gen-dart']);
    }
  
