Fix tests

--- a/test/parse.c
+++ b/test/parse.c
@@ -192,7 +192,7 @@
 			uint16_t buf16[256];
 			int res = utf8_to_ucs2(inin, strlen(inin), buf16, 256);
 			res = gsm7_encode(buf16, res, buf16);
-			int packedsize = gsm7_pack(buf16, res, (char*)pdu, res * 2, j);
+			int packedsize = gsm7_pack(buf16, res, (char*)pdu, 256, j);
 			hexify(pdu, (packedsize + 1) / 2, (char*)pdu);
 			res = unhex((const char*)pdu, pdu);
 			res = gsm7_unpack_decode((const char*)pdu, packedsize, buf16, res, j, 0, 0);
