diff --git a/src/meson.build b/src/meson.build
index 8a1af5a..b463ade 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -46,8 +46,6 @@ if libspelling.found ()
   message('Building with libspelling support')
 endif
 
-cargo_env = [ 'CARGO_HOME=' + meson.project_build_root() / 'cargo-home' ]
-
 cargo_build = custom_target(
   'cargo-build',
   build_by_default: true,
@@ -58,8 +56,6 @@ cargo_build = custom_target(
   install_dir: bindir,
   depends: resources,
   command: [
-    'env',
-    cargo_env,
     cargo, 'build',
     cargo_options,
     '&&',
