diff --git a/src/meson.build b/src/meson.build
index b463ade..24d70d2 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -25,7 +25,7 @@ cargo_options = [ '--manifest-path', meson.project_source_root() / 'Cargo.toml'
 cargo_options += [ '--target-dir', meson.project_build_root() / 'target' ]
 
 if get_option('profile') == 'default'
-  cargo_options += [ '--release' ]
+  cargo_options += [ '--release', '--frozen' ]
   rust_target = 'release'
   message('Building in release mode')
 elif get_option('profile') == 'screenshot'
@@ -56,7 +56,7 @@ cargo_build = custom_target(
   install_dir: bindir,
   depends: resources,
   command: [
-    cargo, 'build',
+    cargo, 'auditable', 'build',
     cargo_options,
     '&&',
     'cp', 'target' / rust_target / meson.project_name(), '@OUTPUT@',
