TODO: Remove when switching _default_ver to true.

--- a/contrib/postgres_fdw/expected/postgres_fdw.out
+++ b/contrib/postgres_fdw/expected/postgres_fdw.out
@@ -171,8 +171,8 @@
 	sslcrl 'value',
 	--requirepeer 'value',
 	krbsrvname 'value',
-	gsslib 'value',
-	gssdelegation 'value'
+	gsslib 'value'
+	--gssdelegation 'value'
 	--replication 'value'
 );
 -- Error, invalid list syntax
--- a/contrib/postgres_fdw/sql/postgres_fdw.sql
+++ b/contrib/postgres_fdw/sql/postgres_fdw.sql
@@ -185,8 +185,8 @@
 	sslcrl 'value',
 	--requirepeer 'value',
 	krbsrvname 'value',
-	gsslib 'value',
-	gssdelegation 'value'
+	gsslib 'value'
+	--gssdelegation 'value'
 	--replication 'value'
 );
 
