diff --git a/bubblejail/services.py b/bubblejail/services.py
index abf9bf7..fd2533f 100644
--- a/bubblejail/services.py
+++ b/bubblejail/services.py
@@ -346,9 +346,6 @@ class BubblejailDefaults(BubblejailService):
         yield ReadOnlyBind('/etc/login.defs')  # ???: is this file needed
         # ldconfig: linker cache
         # particularly needed for steam runtime to work
-        yield ReadOnlyBind('/etc/ld.so.cache')
-        yield ReadOnlyBind('/etc/ld.so.conf')
-        yield ReadOnlyBind('/etc/ld.so.conf.d')
 
         # Temporary directories
         yield DirCreate('/tmp')
