--- a/libclamunrar_iface/unrar_iface.cpp	2019-08-28 18:09:01.840609079 +0000
+++ b/libclamunrar_iface/unrar_iface.cpp	2019-08-28 18:08:49.567175407 +0000
@@ -341,7 +341,7 @@
 
     unrar_dbgmsg("unrar_peek_file_header:   Name:          %s\n", headerData.FileName);
     unrar_dbgmsg("unrar_peek_file_header:   Directory?:    %u\n", file_metadata->is_dir);
-    unrar_dbgmsg("unrar_peek_file_header:   Target Dir:    %ls\n", headerData.DirTarget);
+    unrar_dbgmsg("unrar_peek_file_header:   Target Dir:    %u\n", headerData.DirTarget);
     unrar_dbgmsg("unrar_peek_file_header:   RAR Version:   %u\n", headerData.UnpVer);
     unrar_dbgmsg("unrar_peek_file_header:   Packed Size:   %lld\n", file_metadata->pack_size);
     unrar_dbgmsg("unrar_peek_file_header:   Unpacked Size: %lld\n", file_metadata->unpack_size);
