diff --git a/shared/source/os_interface/linux/sys_calls_linux.cpp b/shared/source/os_interface/linux/sys_calls_linux.cpp
index 06ac6e9d62..0a8199c847 100644
--- a/shared/source/os_interface/linux/sys_calls_linux.cpp
+++ b/shared/source/os_interface/linux/sys_calls_linux.cpp
@@ -87,7 +87,7 @@ int fsync(int fd) {
 }
 
 int ioctl(int fileDescriptor, unsigned long int request, void *arg) {
-    return ::ioctl(fileDescriptor, request, arg);
+    return ::ioctl(fileDescriptor, (int)request, arg);
 }
 
 void *dlopen(const char *filename, int flag) {
