diff --git a/shared/source/compiler_interface/os_compiler_cache_helper.h b/shared/source/compiler_interface/os_compiler_cache_helper.h
index 7fe15210cf..0ebfbeadbe 100644
--- a/shared/source/compiler_interface/os_compiler_cache_helper.h
+++ b/shared/source/compiler_interface/os_compiler_cache_helper.h
@@ -7,6 +7,7 @@
 
 #pragma once
 #include <string>
+#include <ctime>
 
 namespace NEO {
 class EnvironmentVariableReader;
