diff --git a/level_zero/sysman/source/shared/linux/sysman_fs_access_interface.cpp b/level_zero/sysman/source/shared/linux/sysman_fs_access_interface.cpp
index 0ebe4bf476..018adfe951 100644
--- a/level_zero/sysman/source/shared/linux/sysman_fs_access_interface.cpp
+++ b/level_zero/sysman/source/shared/linux/sysman_fs_access_interface.cpp
@@ -15,6 +15,9 @@
 #include <fcntl.h>
 #include <sstream>
 #include <unistd.h>
+#ifndef __GLIBC__
+#include <limits.h> // PATH_MAX
+#endif
 
 namespace L0 {
 namespace Sysman {
diff --git a/level_zero/sysman/source/shared/linux/udev/udev_lib.h b/level_zero/sysman/source/shared/linux/udev/udev_lib.h
index 98acb2d2cb..2188842bba 100644
--- a/level_zero/sysman/source/shared/linux/udev/udev_lib.h
+++ b/level_zero/sysman/source/shared/linux/udev/udev_lib.h
@@ -9,6 +9,9 @@
 
 #include <string>
 #include <vector>
+#ifndef __dev_t_defined
+#include <sys/types.h>
+#endif /* __dev_t_defined */
 
 namespace L0 {
 namespace Sysman {
diff --git a/level_zero/tools/source/sysman/linux/udev/udev_lib.h b/level_zero/tools/source/sysman/linux/udev/udev_lib.h
index bc031ff5eb..a7c35f2a55 100644
--- a/level_zero/tools/source/sysman/linux/udev/udev_lib.h
+++ b/level_zero/tools/source/sysman/linux/udev/udev_lib.h
@@ -9,6 +9,9 @@
 
 #include <string>
 #include <vector>
+#ifndef __dev_t_defined
+#include <sys/types.h>
+#endif /* __dev_t_defined */
 
 namespace L0 {
 class UdevLib {
diff --git a/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h b/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
index dce5832692..8cf117f58c 100644
--- a/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
+++ b/shared/offline_compiler/source/utilities/linux/safety_guard_linux.h
@@ -13,7 +13,9 @@
 #include <csignal>
 #include <cstdio>
 #include <cstdlib>
+#ifdef __GLIBC__
 #include <execinfo.h>
+#endif
 
 static jmp_buf jmpbuf;
 
@@ -38,6 +40,7 @@ class SafetyGuardLinux : NEO::NonCopyableAndNonMovableClass {
     }
 
     static void sigAction(int sigNum, siginfo_t *info, void *ucontext) {
+#ifdef __GLIBC__
         const int callstackDepth = 30;
         void *addresses[callstackDepth];
         char **callstack;
@@ -51,6 +54,9 @@ class SafetyGuardLinux : NEO::NonCopyableAndNonMovableClass {
         }
 
         free(callstack);
+#else
+        printf("ERROR: MUSL HAS NO BACKTRACE() or execinfo!\n");
+#endif
         longjmp(jmpbuf, 1);
     }
 
diff --git a/shared/source/os_interface/linux/os_library_helper.cpp b/shared/source/os_interface/linux/os_library_helper.cpp
index 2dc56e9a4e..52825d8a9b 100644
--- a/shared/source/os_interface/linux/os_library_helper.cpp
+++ b/shared/source/os_interface/linux/os_library_helper.cpp
@@ -14,7 +14,11 @@ namespace NEO {
 namespace Linux {
 void adjustLibraryFlags(int &dlopenFlag) {
     if (debugManager.flags.DisableDeepBind.get()) {
+#ifdef __GLIBC__
         dlopenFlag &= ~RTLD_DEEPBIND;
+#else
+        dlopenFlag &= ~RTLD_GLOBAL;
+#endif
     }
 }
 } // namespace Linux
diff --git a/shared/source/os_interface/linux/os_library_linux.cpp b/shared/source/os_interface/linux/os_library_linux.cpp
index edf791fb16..82826064be 100644
--- a/shared/source/os_interface/linux/os_library_linux.cpp
+++ b/shared/source/os_interface/linux/os_library_linux.cpp
@@ -43,7 +43,11 @@ OsLibrary::OsLibrary(const OsLibraryCreateProperties &properties) {
     if (properties.libraryName.empty() || properties.performSelfLoad) {
         this->handle = SysCalls::dlopen(0, RTLD_LAZY);
     } else {
+#ifdef __GLIBC__
         auto dlopenFlag = RTLD_LAZY;
+#else
+        auto dlopenFlag = RTLD_LAZY | RTLD_GLOBAL;
+#endif
         dlopenFlag = properties.customLoadFlags ? *properties.customLoadFlags : dlopenFlag;
         adjustLibraryFlags(dlopenFlag);
         this->handle = SysCalls::dlopen(properties.libraryName.c_str(), dlopenFlag);
