diff --git a/setup.py b/setup.py
index 1289558..5fa5605 100644
--- a/setup.py
+++ b/setup.py
@@ -392,6 +392,6 @@ setup(
         'update': UpdateCommand},
     ext_modules=[Extension(
         name='yara',
-        include_dirs=['yara/libyara/include', 'yara/libyara/', '.'],
+        libraries = ['yara'],
         define_macros=[('BUCKETS_128', 1), ('CHECKSUM_1B', 1)],
         sources=['yara-python.c'])])
