diff --git scripts/halberd scripts/halberd
index cf01b1e..2b5c218 100755
--- scripts/halberd
+++ scripts/halberd
@@ -20,8 +20,9 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA

-
-import sys
+import sys, os
+if os.path.dirname(os.path.abspath(sys.argv[0])) == os.path.abspath("scripts"):
+    sys.path.insert(0, "./")

 import Halberd.shell
 import Halberd.logger
