--- bluez-4.101.orig/audio/ipc.c
+++ bluez-4.101/audio/ipc.c
@@ -22,6 +22,7 @@
  */
 
 #include "ipc.h"
+#include <string.h>
 
 #define ARRAY_SIZE(arr) (sizeof(arr) / sizeof((arr)[0]))
 
--- bluez-4.101.orig/compat/bnep.c
+++ bluez-4.101/compat/bnep.c
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <sys/socket.h>
 #include <sys/ioctl.h>
+#include <sys/time.h>
 
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/bnep.h>
--- bluez-4.101.orig/compat/dund.c
+++ bluez-4.101/compat/dund.c
@@ -67,7 +67,7 @@
 static int  persist;
 static int  use_sdp = 1;
 static int  auth;
-static int  encrypt;
+static int  do_encrypt;
 static int  secure;
 static int  master;
 static int  type = LANACCESS;
@@ -146,7 +146,7 @@
 		lm |= RFCOMM_LM_MASTER;
 	if (auth)
 		lm |= RFCOMM_LM_AUTH;
-	if (encrypt)
+	if (do_encrypt)
 		lm |= RFCOMM_LM_ENCRYPT;
 	if (secure)
 		lm |= RFCOMM_LM_SECURE;
@@ -492,7 +492,7 @@
 			break;
 
 		case 'E':
-			encrypt = 1;
+			do_encrypt = 1;
 			break;
 
 		case 'S':
--- bluez-4.101.orig/network/common.c
+++ bluez-4.101/network/common.c
@@ -29,6 +29,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <sys/types.h>
 #include <sys/param.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
--- bluez-4.101.orig/network/connection.c
+++ bluez-4.101/network/connection.c
@@ -28,6 +28,7 @@
 #include <stdio.h>
 #include <errno.h>
 #include <unistd.h>
+#include <sys/time.h>
 #include <netinet/in.h>
 
 #include <bluetooth/bluetooth.h>
--- bluez-4.101.orig/src/textfile.h
+++ bluez-4.101/src/textfile.h
@@ -24,6 +24,8 @@
 #ifndef __TEXTFILE_H
 #define __TEXTFILE_H
 
+#include <sys/types.h>
+
 int create_dirs(const char *filename, const mode_t mode);
 int create_file(const char *filename, const mode_t mode);
 int create_name(char *buf, size_t size, const char *path,
