diff --git a/bindings/python/unicorn/unicorn.py b/bindings/python/unicorn/unicorn.py
index c782356..b7d0be2 100644
--- a/bindings/python/unicorn/unicorn.py
+++ b/bindings/python/unicorn/unicorn.py
@@ -86,7 +86,7 @@ _path_list = [os.getenv('LIBUNICORN_PATH', None),
               os.path.join(os.path.split(__file__)[0], 'lib'),
               '',
               distutils.sysconfig.get_python_lib(),
-              "/usr/local/lib/" if sys.platform == 'darwin' else '/usr/lib64',
+              "/usr/local/lib/" if sys.platform == 'darwin' else '/usr/lib',
               os.getenv('PATH', '')]
 
 #print(_path_list)
