diff --git a/src/gui/MainWindow.cpp b/src/gui/MainWindow.cpp
index 5829fe38..5a3739d9 100644
--- a/src/gui/MainWindow.cpp
+++ b/src/gui/MainWindow.cpp
@@ -1228,12 +1228,12 @@ void MainWindow::openBugReportUrl()
 
 void MainWindow::openGettingStartedGuide()
 {
-    customOpenUrl(QString("file:///%1").arg(resources()->dataPath("docs/KeePassXC_GettingStarted.html")));
+    customOpenUrl("file:///usr/share/doc/keepassxc/KeePassXC_GettingStarted.html");
 }
 
 void MainWindow::openUserGuide()
 {
-    customOpenUrl(QString("file:///%1").arg(resources()->dataPath("docs/KeePassXC_UserGuide.html")));
+    customOpenUrl("file:///usr/share/doc/keepassxc/KeePassXC_UserGuide.html");
 }
 
 void MainWindow::openOnlineHelp()
@@ -1243,7 +1243,7 @@ void MainWindow::openOnlineHelp()
 
 void MainWindow::openKeyboardShortcuts()
 {
-    customOpenUrl(QString("file:///%1").arg(resources()->dataPath("docs/KeePassXC_KeyboardShortcuts.html")));
+    customOpenUrl("file:///usr/share/doc/keepassxc/KeePassXC_KeyboardShortcuts.html");
 }
 
 void MainWindow::switchToDatabases()
