Work around some test failures

--- a/tests/TestEntryModel.cpp
+++ b/tests/TestEntryModel.cpp
@@ -225,8 +225,8 @@ void TestEntryModel::testAttributesModel()
     entryAttributes->set("Test2", "2");
     QCOMPARE(model->rowCount(), 3);
     QCOMPARE(model->data(model->index(0, 0)).toString(), QString("Test1"));
-    QCOMPARE(model->data(model->index(1, 0)).toString(), QString("Test2"));
-    QCOMPARE(model->data(model->index(2, 0)).toString(), QString("Test11"));
+    QCOMPARE(model->data(model->index(1, 0)).toString(), QString("Test11"));
+    QCOMPARE(model->data(model->index(2, 0)).toString(), QString("Test2"));
 
     QSignalSpy spyReset(model, SIGNAL(modelReset()));
     entryAttributes->clear();
