From 182e2e4e03567a779c9ba35a5d6c4293362bd520 Mon Sep 17 00:00:00 2001
From: Damon To <9162521+damonto@users.noreply.github.com>
Date: Thu, 6 Mar 2025 10:17:50 +0800
Subject: [PATCH] fix(driver): set correct open flags for MBIM device proxy
 connection (#206)

---
 driver/apdu/mbim.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/apdu/mbim.c b/driver/apdu/mbim.c
index b8e40e54..75fe84e0 100644
--- a/driver/apdu/mbim.c
+++ b/driver/apdu/mbim.c
@@ -131,7 +131,7 @@ static int apdu_interface_connect(struct euicc_ctx *ctx)
 
     MbimDeviceOpenFlags open_flags = MBIM_DEVICE_OPEN_FLAGS_NONE;
     if (mbim_priv->use_proxy)
-        open_flags |= MBIM_DEVICE_OPEN_FLAGS_NONE;
+        open_flags |= MBIM_DEVICE_OPEN_FLAGS_PROXY;
 
     mbim_device_open_sync(mbim_priv->device, open_flags, mbim_priv->context, &error);
     if (error) {
