Upstream: https://github.com/libretro/parallel-n64/issues/671
--- parallel-n64-0a67445ce63513584d92e5c57ea87efe0da9b3bd/mupen64plus-core/src/r4300/new_dynarec/arm64/assem_arm64.h.orig
+++ parallel-n64-0a67445ce63513584d92e5c57ea87efe0da9b3bd/mupen64plus-core/src/r4300/new_dynarec/arm64/assem_arm64.h
@@ -1,6 +1,8 @@
 #ifndef M64P_R4300_ASSEM_ARM64_H
 #define M64P_R4300_ASSEM_ARM64_H
 
+#include <sys/types.h>
+
 #define HOST_REGS 29
 #define HOST_CCREG 20 /* callee-save */
 #define HOST_BTREG 19 /* callee-save */
