From ae702996a635214ac672ed91b2fbbe6052958d56 Mon Sep 17 00:00:00 2001
From: "Alex Xu (Hello71)" <alex_y_xu@yahoo.ca>
Date: Fri, 20 Aug 2021 09:59:06 -0400
Subject: [PATCH] [build] use autoconf standard directories

---
 Makefile.in | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 32e1f30..1407d27 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -204,10 +204,9 @@ LDFLAGS+=-static
 endif
 
 INSTALL:=@INSTALL@
-PREFIX:=@prefix@
-BINDIR:=$(DESTDIR)$(PREFIX)/sbin
-DATADIR:=$(DESTDIR)$(PREFIX)/share
-MANPATH:=$(DATADIR)/man
+BINDIR:=$(DESTDIR)@sbindir@
+DATADIR:=$(DESTDIR)@datarootdir@
+MANPATH:=$(DESTDIR)@mandir@
 
 vpath %.cc $(TOP_DIR)
 
-- 
2.33.0

