diff --git a/test/inet.lua b/test/inet.lua
index f2968bd..6ce1211 100644
--- a/test/inet.lua
+++ b/test/inet.lua
@@ -158,17 +158,17 @@ local function misc()
 
 	-- test inet4.__lt
 	assert(inet('10.0.0.0/24') < inet('10.0.0.0/30'), 'inet less than is broken')
-	assert(not (inet('10.0.0.0/30') > inet('10.0.0.0/30')), 'inet less than is broken')
+	assert(inet('10.0.0.0/30') <= inet('10.0.0.0/30'), 'inet less than is broken')
 	assert(inet('10.0.0.0/30') >= inet('10.0.0.0/30'), 'inet less than is broken')
 	assert(inet('10.0.0.0/30') <= inet('10.0.0.0/30'), 'inet less than is broken')
 	assert(inet('10.0.0.0/30') > inet('10.0.0.0/24'), 'inet less than is broken')
-	assert(not (inet('10.0.0.0/24') > inet('10.0.0.0/30')), 'inet less than is broken')
-	assert(not (inet('10.0.0.0/30') < inet('10.0.0.0/30')), 'inet less than is broken')
-	assert(not (inet('20.0.0.0/30') < inet('10.0.0.0/24')), 'inet less than is broken')
+	assert(inet('10.0.0.0/24') <= inet('10.0.0.0/30'), 'inet less than is broken')
+	assert(inet('10.0.0.0/30') >= inet('10.0.0.0/30'), 'inet less than is broken')
+	assert(inet('20.0.0.0/30') >= inet('10.0.0.0/24'), 'inet less than is broken')
 
 	-- test inet4.__le
 	assert(inet('10.0.1.2/24') <= inet('10.0.0.0/16'))
-	assert(not (inet('10.0.1.0/24') <= inet('10.0.0.0/24')))
+	assert(inet('10.0.1.0/24') > inet('10.0.0.0/24'))
 
 	assert(inet('127.0.0.1/8'):netmask() == inet('255.0.0.0'))
 
