From a8214fd58bc46d23b64b3a55db023c7f5a5ea6af Mon Sep 17 00:00:00 2001
From: Francois Marier <francois@debian.org>
Date: Fri, 24 Jul 2020 21:22:47 -0700
Subject: [PATCH] Fix compilation with GCC's -fno-common flag (fixes #305)

---
 client/log_msg.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/client/log_msg.h b/client/log_msg.h
index cc17716b..3dda1614 100644
--- a/client/log_msg.h
+++ b/client/log_msg.h
@@ -38,7 +38,7 @@ enum
     LOG_VERBOSITY_INFO,         /*!< Constant to define a INFO message */
     LOG_VERBOSITY_DEBUG,        /*!< Constant to define a DEBUG message */
     LOG_LAST_VERBOSITY
-} log_level_t;
+};
 
 #define LOG_DEFAULT_VERBOSITY   LOG_VERBOSITY_NORMAL    /*!< Default verbosity to use */
 
