diff --git a/cme/cmeserver.py b/cme/cmeserver.py
index 4549095..5bc336b 100644
--- a/cme/cmeserver.py
+++ b/cme/cmeserver.py
@@ -1,17 +1,9 @@
 import BaseHTTPServer
 import threading
 import ssl
-<<<<<<< HEAD:cme/cmeserver.py
-<<<<<<< 23d8a6517fb1a36bac9b09e3923201ee291d00ac:cme/cmeserver.py
 import os
 import sys
 from getpass import getuser
-=======
-import sys
->>>>>>> Added error handling for CME server in case the port is already in use.:core/cmeserver.py
-=======
-import sys
->>>>>>> da4e55380a1d1ad46d3176c513b829bd4a7099f4:core/cmeserver.py
 from BaseHTTPServer import BaseHTTPRequestHandler
 from logging import getLogger
 from gevent import sleep
