From d69f9402552d1ae7b0c7bb1d895c7b3a43f0f889 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Mon, 29 Mar 2021 20:12:13 -0700
Subject: [PATCH] remove backtrace_symbols

---
 webrtc/base/checks.cc | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/webrtc/base/checks.cc b/webrtc/base/checks.cc
index 49a31f2..4f440b2 100644
--- a/webrtc/base/checks.cc
+++ b/webrtc/base/checks.cc
@@ -55,33 +55,6 @@ void PrintError(const char* format, ...) {
 // to get usable symbols on Linux. This is copied from V8. Chromium has a more
 // advanced stace trace system; also more difficult to copy.
 void DumpBacktrace() {
-#if defined(__GLIBCXX__) && !defined(__UCLIBC__)
-  void* trace[100];
-  int size = backtrace(trace, sizeof(trace) / sizeof(*trace));
-  char** symbols = backtrace_symbols(trace, size);
-  PrintError("\n==== C stack trace ===============================\n\n");
-  if (size == 0) {
-    PrintError("(empty)\n");
-  } else if (symbols == NULL) {
-    PrintError("(no symbols)\n");
-  } else {
-    for (int i = 1; i < size; ++i) {
-      char mangled[201];
-      if (sscanf(symbols[i], "%*[^(]%*[(]%200[^)+]", mangled) == 1) {  // NOLINT
-        PrintError("%2d: ", i);
-        int status;
-        size_t length;
-        char* demangled = abi::__cxa_demangle(mangled, NULL, &length, &status);
-        PrintError("%s\n", demangled != NULL ? demangled : mangled);
-        free(demangled);
-      } else {
-        // If parsing failed, at least print the unparsed symbol.
-        PrintError("%s\n", symbols[i]);
-      }
-    }
-  }
-  free(symbols);
-#endif
 }
 
 FatalMessage::FatalMessage(const char* file, int line) {
-- 
2.31.1

