diff --git a/cgit.mk b/cgit.mk
index 3b8b79a..2e2992f 100644
--- a/cgit.mk
+++ b/cgit.mk
@@ -29,30 +29,18 @@ ifdef NO_LUA
 	LUA_MESSAGE := linking without specified Lua support
 	CGIT_CFLAGS += -DNO_LUA
 else
-LUAJIT_CFLAGS := $(shell pkg-config --cflags luajit 2>/dev/null)
-LUAJIT_LIBS := $(shell pkg-config --libs luajit 2>/dev/null)
-LUA_LIBS := $(shell pkg-config --libs lua 2>/dev/null)
-LUA_CFLAGS := $(shell pkg-config --cflags lua 2>/dev/null)
-ifeq (JIT,$(LUA_IMPLEMENTATION))
-	ifeq ($(strip $(LUAJIT_LIBS)),)
- 		$(error LuaJIT specified via LUA_IMPLEMENTATION=JIT, but library could not be found.)
-	endif
-	LUA_MESSAGE := linking with selected LuaJIT
-	CGIT_LIBS += $(LUAJIT_LIBS)
-	CGIT_CFLAGS += $(LUAJIT_CFLAGS)
-else ifeq (VANILLA,$(LUA_IMPLEMENTATION))
-	ifeq ($(strip $(LUA_LIBS)),)
- 		$(error Lua specified via LUA_IMPLEMENTATION=VANILLA, but library could not be found.)
-	endif
-	LUA_MESSAGE := linking with selected Lua
-	CGIT_LIBS += $(LUA_LIBS)
-	CGIT_LIBS += $(LUA_CFLAGS)
-else ifneq ($(strip $(LUAJIT_LIBS)),)
-	LUA_MESSAGE := linking with autodetected LuaJIT
-	CGIT_LIBS += $(LUAJIT_LIBS)
-	CGIT_CFLAGS += $(LUAJIT_CFLAGS)
-else ifneq ($(strip $(LUA_LIBS)),)
-	LUA_MESSAGE := linking with autodetected Lua
+ifeq ($(LUA_PKGCONFIG),)
+	LUA_PKGCONFIG := $(shell for pc in luajit lua lua5.2 lua5.1; do \
+			pkg-config --exists $$pc && echo $$pc && break; \
+			done)
+	LUA_MODE := autodetected
+else
+	LUA_MODE := specified
+endif
+ifneq ($(LUA_PKGCONFIG),)
+	LUA_MESSAGE := linking with $(LUA_MODE) $(LUA_PKGCONFIG)
+	LUA_LIBS := $(shell pkg-config --libs $(LUA_PKGCONFIG) 2>/dev/null)
+	LUA_CFLAGS := $(shell pkg-config --cflags $(LUA_PKGCONFIG) 2>/dev/null)
 	CGIT_LIBS += $(LUA_LIBS)
 	CGIT_CFLAGS += $(LUA_CFLAGS)
 else

