--- orig/mongodb-src-r3.0.2/src/mongo/util/net/socket_poll.h
+++ mongodb-src-r3.0.2/src/mongo/util/net/socket_poll.h
@@ -29,7 +29,7 @@
 #pragma once
 
 #ifndef _WIN32
-# include <sys/poll.h>
+# include <poll.h>
 #else
 # if defined(NTDDI_VERSION) && ( !defined(NTDDI_VISTA) || ( NTDDI_VERSION < NTDDI_VISTA ) )
     // These are only defined in winsock2.h on newer windows but we need them everywhere.
