--- orig/mongodb-src-r3.0.2/src/mongo/platform/backtrace.cpp
+++ mongodb-src-r3.0.2/src/mongo/platform/backtrace.cpp
@@ -28,6 +28,12 @@
 #if !defined(_WIN32)
 #if defined(__sunos__) || !defined(MONGO_HAVE_EXECINFO_BACKTRACE)
 
+// dlfcn.h requires _GNU_SOURCE to be defined in order to export
+// Dl_* structures. Issue gonna be addressed in 3.1.3, see SERVER-17199
+#ifndef _GNU_SOURCE
+# define _GNU_SOURCE
+#endif
+
 #include "mongo/platform/backtrace.h"
 
 #include <boost/smart_ptr/scoped_array.hpp>
@@ -42,6 +48,8 @@
 
 using std::string;
 using std::vector;
+
+typedef Dl_info         Dl_info_t;
 
 namespace mongo {
 namespace pal {
