diff --git a/SConstruct b/SConstruct
index 64c8387..3db1f3e 100644
--- a/SConstruct
+++ b/SConstruct
@@ -2292,7 +2292,8 @@ def doConfigure(myenv):
     if (conf.CheckCXXHeader( "execinfo.h" ) and
         conf.CheckDeclaration('backtrace', includes='#include <execinfo.h>') and
         conf.CheckDeclaration('backtrace_symbols', includes='#include <execinfo.h>') and
-        conf.CheckDeclaration('backtrace_symbols_fd', includes='#include <execinfo.h>')):
+        conf.CheckDeclaration('backtrace_symbols_fd', includes='#include <execinfo.h>') and
+        conf.CheckFunc('backtrace')):
 
         conf.env.SetConfigHeaderDefine("MONGO_CONFIG_HAVE_EXECINFO_BACKTRACE")
 
