--- a/src/mongo/util/processinfo_linux.cpp
+++ b/src/mongo/util/processinfo_linux.cpp
@@ -40,11 +40,11 @@
 #include <sys/mman.h>
 #include <sys/utsname.h>
 #include <unistd.h>
-#ifdef __UCLIBC__
+//#ifdef __UCLIBC__
 #include <features.h>
-#else
-#include <gnu/libc-version.h>
-#endif
+//#else
+//#include <gnu/libc-version.h>
+//#endif
 
 #include <boost/filesystem.hpp>
 #include <boost/none.hpp>
@@ -487,13 +487,9 @@
 
     BSONObjBuilder bExtra;
     bExtra.append("versionString", LinuxSysHelper::readLineFromFile("/proc/version"));
-#ifdef __UCLIBC__
     stringstream ss;
-    ss << "uClibc-" << __UCLIBC_MAJOR__ << "." << __UCLIBC_MINOR__ << "." << __UCLIBC_SUBLEVEL__;
+    ss << "musl-1.1.16";
     bExtra.append("libcVersion", ss.str());
-#else
-    bExtra.append("libcVersion", gnu_get_libc_version());
-#endif
     if (!verSig.empty())
         // optional
         bExtra.append("versionSignature", verSig);
