--- ./src/mongo/util/processinfo_linux2.cpp.orig	2013-06-10 19:46:43.082371900 +0000
+++ ./src/mongo/util/processinfo_linux2.cpp	2013-06-10 20:01:45.581386405 +0000
@@ -20,7 +20,11 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <sys/mman.h>
+#if defined(__UCLIBC__)
+#define gnu_get_libc_version(x) "uClibc"
+#else
 #include <gnu/libc-version.h>
+#endif
 #include <sys/utsname.h>
 
 #include "processinfo.h"
