--- cv.orig/cv.c
+++ cv/cv.c
@@ -44,7 +44,7 @@
 #include "sizes.h"
 #include "hlist.h"
 
-char *proc_names[] = {"cp", "mv", "dd", "tar", "gzip", "gunzip", "cat", "grep", "fgrep", "egrep", "cut", "sort", NULL};
+char *proc_names[] = {"cp", "mv", "dd", "tar", "gzip", "gunzip", "cat", "grep", "fgrep", "egrep", "cut", "sort", "cpio", "xz", "bzip2", "bunzip2", NULL};
 char *proc_specifiq = NULL;
 WINDOW *mainwin;
 signed char flag_quiet = 0;
