diff --git a/src/fe-gtk/fe-gtk.c b/src/fe-gtk/fe-gtk.c
index 251a7d7..f427ad4 100644
--- a/src/fe-gtk/fe-gtk.c
+++ b/src/fe-gtk/fe-gtk.c
@@ -958,6 +958,10 @@ fe_open_url_inner (const char *url)
 	/* everything failed, what now? just try firefox */
 	if (try_browser ("firefox", NULL, url))
 		return;
+	
+	/* try midori before giving up */
+	if (try_browser ("midori", NULL, url))
+		return;
 
 	/* fresh out of ideas... */
 	try_browser ("mozilla", NULL, url);
