From fbb4979c96f4d5b7df8fbd59591fb6ec3d489e05 Mon Sep 17 00:00:00 2001
From: Laurent Montel <montel@kde.org>
Date: Mon, 23 Aug 2021 06:56:25 +0200
Subject: [PATCH] Build against last kf5

---
 src/widgets/configuredialog/configurefontwidget.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/widgets/configuredialog/configurefontwidget.cpp b/src/widgets/configuredialog/configurefontwidget.cpp
index 35f36d487..3a858202c 100644
--- a/src/widgets/configuredialog/configurefontwidget.cpp
+++ b/src/widgets/configuredialog/configurefontwidget.cpp
@@ -22,6 +22,7 @@
 #include "ruqolaglobalconfig.h"
 #include <KFontChooser>
 #include <KLocalizedString>
+#include <kwidgetsaddons_version.h>
 
 #include <QCheckBox>
 #include <QVBoxLayout>
@@ -29,8 +30,15 @@
 ConfigureFontWidget::ConfigureFontWidget(QWidget *parent)
     : QWidget(parent)
     , mCustomFontCheck(new QCheckBox(i18n("&Use custom fonts"), this))
+#if KWIDGETSADDONS_VERSION >= QT_VERSION_CHECK(5, 86, 0)
+    , mFontChooser(new KFontChooser(KFontChooser::DisplayFrame, this))
+#else
     , mFontChooser(new KFontChooser(this, KFontChooser::DisplayFrame, QStringList(), 4))
+#endif
 {
+#if KWIDGETSADDONS_VERSION >= QT_VERSION_CHECK(5, 86, 0)
+    mFontChooser->setMinVisibleItems(4);
+#endif
     auto mainLayout = new QVBoxLayout(this);
     mainLayout->setObjectName(QStringLiteral("mainLayout"));
     mainLayout->setContentsMargins({});
-- 
GitLab

