diff -ruN a/setup.py b/setup.py
diff --git a/setup.py b/setup.py
index a62a191..92c0141 100644
--- a/setup.py
+++ b/setup.py
@@ -131,6 +131,9 @@ libraries = []
 # Special flag to build without portaudio, portmidi and liblo deps.
 if "--minimal" in sys.argv:
     sys.argv.remove("--minimal")
+    macros.append(('USE_PORTAUDIO', None))
+    ad_files.append("ad_portaudio.c")
+    libraries.append("portaudio")
 else:
     # portaudio
     macros.append(("USE_PORTAUDIO", None))
