diff --git a/debian/ifupdown.sh b/debian/ifupdown.sh
index 952d1b0..7aed7e0 100755
--- a/debian/ifupdown.sh
+++ b/debian/ifupdown.sh
@@ -35,8 +35,10 @@ if [ -f $SERVICE_UNIT ] && [ -x /bin/systemctl ]; then
         systemctl start openvswitch-switch.service
     fi
 else
-    if service openvswitch-switch status > /dev/null 2>&1; then
-        service openvswitch-switch start
+    if ! /etc/init.d/ovs-vswitchd status &>/dev/null; then
+        /etc/init.d/ovs-modules start
+        /etc/init.d/ovsdb-server start
+        /etc/init.d/ovs-vswitchd start
     fi
 fi
 
