--- gnome-bluetooth-2.32.0.orig/configure
+++ gnome-bluetooth-2.32.0/configure
@@ -638,8 +638,6 @@
 MOBLIN_CFLAGS
 PLUGINS_LIBS
 PLUGINS_CFLAGS
-GEOCLUE_LIBS
-GEOCLUE_CFLAGS
 COMMON_LIBS
 COMMON_CFLAGS
 LIBGNOMEBT_LIBS
@@ -899,8 +897,6 @@
 LIBGNOMEBT_LIBS
 COMMON_CFLAGS
 COMMON_LIBS
-GEOCLUE_CFLAGS
-GEOCLUE_LIBS
 PLUGINS_CFLAGS
 PLUGINS_LIBS
 MOBLIN_CFLAGS
@@ -13662,106 +13658,6 @@
 
 fi
 
-
-pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for GEOCLUE" >&5
-$as_echo_n "checking for GEOCLUE... " >&6; }
-
-if test -n "$GEOCLUE_CFLAGS"; then
-    pkg_cv_GEOCLUE_CFLAGS="$GEOCLUE_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gconf-2.0
-		  dbus-glib-1\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0
-		  dbus-glib-1") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_GEOCLUE_CFLAGS=`$PKG_CONFIG --cflags "gconf-2.0
-		  dbus-glib-1" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$GEOCLUE_LIBS"; then
-    pkg_cv_GEOCLUE_LIBS="$GEOCLUE_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"gconf-2.0
-		  dbus-glib-1\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "gconf-2.0
-		  dbus-glib-1") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_GEOCLUE_LIBS=`$PKG_CONFIG --libs "gconf-2.0
-		  dbus-glib-1" 2>/dev/null`
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        GEOCLUE_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors "gconf-2.0
-		  dbus-glib-1" 2>&1`
-        else
-	        GEOCLUE_PKG_ERRORS=`$PKG_CONFIG --print-errors "gconf-2.0
-		  dbus-glib-1" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$GEOCLUE_PKG_ERRORS" >&5
-
-	as_fn_error $? "Package requirements (gconf-2.0
-		  dbus-glib-1) were not met:
-
-$GEOCLUE_PKG_ERRORS
-
-Consider adjusting the PKG_CONFIG_PATH environment variable if you
-installed software in a non-standard prefix.
-
-Alternatively, you may set the environment variables GEOCLUE_CFLAGS
-and GEOCLUE_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details." "$LINENO" 5
-
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	{ { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "The pkg-config script could not be found or is too old.  Make sure it
-is in your PATH or set the PKG_CONFIG environment variable to the full
-path to pkg-config.
-
-Alternatively, you may set the environment variables GEOCLUE_CFLAGS
-and GEOCLUE_LIBS to avoid the need to call pkg-config.
-See the pkg-config man page for more details.
-
-To get pkg-config, see <http://pkg-config.freedesktop.org/>.
-See \`config.log' for more details" "$LINENO" 5; }
-
-else
-	GEOCLUE_CFLAGS=$pkg_cv_GEOCLUE_CFLAGS
-	GEOCLUE_LIBS=$pkg_cv_GEOCLUE_LIBS
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-
-fi
 
 
 pkg_failed=no
--- gnome-bluetooth-2.32.0.orig/lib/Makefile.in
+++ gnome-bluetooth-2.32.0/lib/Makefile.in
@@ -383,7 +383,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = plugins
+SUBDIRS = 
 EXTRA_DIST = $(am__append_2) marshal.list bluetooth-client.xml \
 	bluetooth-agent.xml obex-agent.xml gnome-bluetooth.symbols
 BUILT_GIRSOURCES = $(am__append_1)
