--- a/remote.ml	2016-05-23 18:40:05.000000000 +0200
+++ b/remote.ml	2017-03-23 11:33:24.784228112 +0100
@@ -450,12 +450,7 @@
      let s = Bytearray.marshal data [Marshal.No_sharing] in
      let l = Bytearray.length s in
      ((s, 0, l) :: rem, l)),
-  (fun buf pos ->
-      try Bytearray.unmarshal buf pos
-      with Failure s -> raise (Util.Fatal (Printf.sprintf 
-"Fatal error during unmarshaling (%s),
-possibly because client and server have been compiled with different\
-versions of the OCaml compiler." s)))
+  (fun buf pos -> Bytearray.unmarshal buf pos)
 
 let makeMarshalingFunctions payloadMarshalingFunctions string =
   let (marshalPayload, unmarshalPayload) = payloadMarshalingFunctions in
