--- ./tools/blktap2/include/blktaplib.h.orig
+++ ./tools/blktap2/include/blktaplib.h
@@ -198,7 +198,9 @@
 } msg_lock_t;
 
 #define READ 0
+#ifndef WRITE
 #define WRITE 1
+#endif
 
 /*Control Messages between manager and tapdev*/
 #define CTLMSG_PARAMS          1
--- ./tools/blktap/lib/blktaplib.h.orig
+++ ./tools/blktap/lib/blktaplib.h
@@ -196,7 +196,9 @@
 } msg_pid_t;
 
 #define READ 0
+#ifndef WRITE
 #define WRITE 1
+#endif
 
 /*Control Messages between manager and tapdev*/
 #define CTLMSG_PARAMS      1
