--- ./xen/arch/x86/boot/Makefile.orig
+++ ./xen/arch/x86/boot/Makefile
@@ -2,7 +2,7 @@
 
 head.o: reloc.S
 
-BOOT_TRAMPOLINE := $(shell sed -n 's,^\#define[[:space:]]\{1\,\}BOOT_TRAMPOLINE[[:space:]]\{1\,\},,p' $(BASEDIR)/include/asm-x86/config.h)
+BOOT_TRAMPOLINE := $(shell sed -n 's,^\#define[[:space:]]\+BOOT_TRAMPOLINE[[:space:]]\+,,p' $(BASEDIR)/include/asm-x86/config.h)
 %.S: %.c
 	RELOC=$(BOOT_TRAMPOLINE) $(MAKE) -f build32.mk $@
 
