--- xen-4.3.1.orig/tools/xenstore/Makefile
+++ xen-4.3.1/tools/xenstore/Makefile
@@ -118,6 +118,7 @@
 	$(INSTALL_PROG) xenstored $(DESTDIR)$(SBINDIR)
 	$(INSTALL_PROG) xenstore-control $(DESTDIR)$(BINDIR)
 	$(INSTALL_PROG) xenstore $(DESTDIR)$(BINDIR)
+	$(INSTALL_PROG) init-xenstore-domain $(DESTDIR)$(BINDIR)
 	set -e ; for c in $(CLIENTS) ; do \
 		ln -f $(DESTDIR)$(BINDIR)/xenstore $(DESTDIR)$(BINDIR)/$${c} ; \
 	done
