diff -aur a/src/drivers/net/via-velocity.c b/src/drivers/net/via-velocity.c
--- a/src/drivers/net/via-velocity.c	2016-05-12 19:14:33.231788641 +1000
+++ b/src/drivers/net/via-velocity.c	2016-05-12 19:17:08.235494746 +1000
@@ -69,7 +69,7 @@
 
 /* NIC specific static variables go here */
 #define VELOCITY_PARAM(N,D) \
-        static const int N[MAX_UNITS]=OPTION_DEFAULT;
+        static __attribute__ ((unused)) const int N[MAX_UNITS]=OPTION_DEFAULT;
 /*        MODULE_PARM(N, "1-" __MODULE_STRING(MAX_UNITS) "i");\
         MODULE_PARM_DESC(N, D); */
 
