diff -aru a/src/drivers/net/via-rhine.c b/src/drivers/net/via-rhine.c
--- a/src/drivers/net/via-rhine.c	2016-05-12 19:24:14.047825550 +1000
+++ b/src/drivers/net/via-rhine.c	2016-05-12 19:33:18.061858418 +1000
@@ -945,13 +945,15 @@
             /* added comment by guard */
             /* For supporting VT6107, please use revision id to recognize different chips in driver */
             // if (tp->chip_id == 0x3065)
-            if( tp->chip_revision < 0x80 && tp->chip_revision >=0x40 )
+            if( tp->chip_revision < 0x80 && tp->chip_revision >=0x40 ) {
                 intr_status |= inb(nic->ioaddr + IntrStatus2) << 16;
                 intr_status = (intr_status & ~DEFAULT_INTR);
-                if ( action == ENABLE ) 
+                if ( action == ENABLE ) {
                     intr_status = intr_status | DEFAULT_INTR;
                     outw(intr_status, nic->ioaddr + IntrEnable);
+                }
                 break;
+            }
         case FORCE :
             outw(0x0010, nic->ioaddr + 0x84);
            break;
