diff --git a/tools/Makefile b/tools/Makefile
index 0edeb62..d5d02e8 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -166,6 +166,9 @@ subdir-all-qemu-xen-dir: qemu-xen-dir-find
 		-L$(XEN_ROOT)/tools/xenstore" \
 		--bindir=$(LIBEXEC) \
 		--datadir=$(SHAREDIR)/qemu-xen \
+		--mandir=$(PREFIX)/share/man \
+		--docdir=$(PREFIX)/share/doc \
+		--sysconfdir=/etc/qemu \
 		--disable-kvm \
 		--python=$(PYTHON) \
 		$(IOEMU_CONFIGURE_CROSS); \

