diff -urN xen-4.13.1.orig/tools/qemu-xen/contrib/vhost-user-input/main.c xen-4.13.1/tools/qemu-xen/contrib/vhost-user-input/main.c
--- xen-4.13.1.orig/tools/qemu-xen/contrib/vhost-user-input/main.c	2020-07-22 02:06:07.554540365 -0600
+++ xen-4.13.1/tools/qemu-xen/contrib/vhost-user-input/main.c	2020-07-22 02:36:31.253058465 -0600
@@ -115,13 +115,16 @@
 static void vi_handle_status(VuInput *vi, virtio_input_event *event)
 {
     struct input_event evdev;
+    struct timeval tv;
     int rc;
 
-    if (gettimeofday(&evdev.time, NULL)) {
+    if (gettimeofday(&tv, NULL)) {
         perror("vi_handle_status: gettimeofday");
         return;
     }
 
+    evdev.input_event_sec = tv.tv_sec;
+    evdev.input_event_usec = tv.tv_usec;
     evdev.type = le16toh(event->type);
     evdev.code = le16toh(event->code);
     evdev.value = le32toh(event->value);
diff -urN xen-4.13.1.orig/tools/qemu-xen/hw/input/virtio-input-host.c xen-4.13.1/tools/qemu-xen/hw/input/virtio-input-host.c
--- xen-4.13.1.orig/tools/qemu-xen/hw/input/virtio-input-host.c	2020-07-22 02:06:07.561207062 -0600
+++ xen-4.13.1/tools/qemu-xen/hw/input/virtio-input-host.c	2020-07-22 02:13:05.923156710 -0600
@@ -192,14 +192,17 @@
                                             virtio_input_event *event)
 {
     VirtIOInputHost *vih = VIRTIO_INPUT_HOST(vinput);
+    struct timeval tv;
     struct input_event evdev;
     int rc;
 
-    if (gettimeofday(&evdev.time, NULL)) {
+    if (gettimeofday(&tv, NULL)) {
         perror("virtio_input_host_handle_status: gettimeofday");
         return;
     }
 
+    evdev.input_event_sec = tv.tv_sec;
+    evdev.input_event_usec = tv.tv_usec;
     evdev.type = le16_to_cpu(event->type);
     evdev.code = le16_to_cpu(event->code);
     evdev.value = le32_to_cpu(event->value);
