--- ./tools/firmware/rombios/32bit/tcgbios/Makefile.orig
+++ ./tools/firmware/rombios/32bit/tcgbios/Makefile
@@ -3,7 +3,7 @@
 
 TARGET  = tcgbiosext.o
 
-CFLAGS += $(CFLAGS_xeninclude) -I.. -I../.. -I../../../../libacpi
+CFLAGS += $(CFLAGS_xeninclude) -I.. -I../.. -I../../../../libacpi -fno-pie
 
 .PHONY: all
 all: $(TARGET)
--- ./tools/firmware/rombios/32bit/Makefile.orig
+++ ./tools/firmware/rombios/32bit/Makefile
@@ -3,7 +3,7 @@
 
 TARGET = 32bitbios_flat.h
 
-CFLAGS += $(CFLAGS_xeninclude) -I.. -I../../../libacpi
+CFLAGS += $(CFLAGS_xeninclude) -I.. -I../../../libacpi -fno-pie
 
 SUBDIRS = tcgbios
 
