--- xen-4.0.1/xen/arch/x86/boot/Makefile
+++ xen-4.0.1.mod/xen/arch/x86/boot/Makefile
@@ -2,7 +2,7 @@
 
 head.o: reloc.S
 
-BOOT_TRAMPOLINE := $(shell sed -n 's,^\#define[[:space:]]\{1\,\}BOOT_TRAMPOLINE[[:space:]]\{1\,\},,p' $(BASEDIR)/include/asm-x86/config.h)
+BOOT_TRAMPOLINE := 0x7c000
 %.S: %.c
 	RELOC=$(BOOT_TRAMPOLINE) XEN_BITSPERLONG=$(patsubst x86_%,%,$(TARGET_SUBARCH)) $(MAKE) -f build32.mk $@
 
