From 8f406849849a8b0a5623757543e4613a12393f88 Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Sun, 6 Apr 2025 15:23:42 -0400
Subject: [PATCH 02/10] Update tox.ini

---
 tox.ini | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/tox.ini b/tox.ini
index ebf4415..0803cbc 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,13 +1,13 @@
 [tox]
 envlist =
-    py{38,39}-django{42}
-    py{310,311,312,py3}-django{42,50,51}
+    py{38,39,310,311,312}-django{42}
+    py{310,311,312,313,py3}-django{51,52}
 [testenv]
 deps =
+    # LTS April 2025 - April 2028
+    django52: django<5.3
     # August 2024 - December 2025
     django51: django<5.2
-    # January 2024 - April 2025
-    django50: django<5.1
     # LTS April 2023 - April 2026
     django42: django<4.3
     -rtest/requirements.txt

From d3cd8a42052132ef28fb49127316c734fd7b127c Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Sun, 6 Apr 2025 16:20:05 -0400
Subject: [PATCH 03/10] Update pyproject.toml

---
 pyproject.toml | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 57ab455..f60452c 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -15,7 +15,6 @@ classifiers = [
     'Environment :: Web Environment',
     'Framework :: Django',
     'Intended Audience :: Developers',
-    'License :: OSI Approved :: MIT License',
     'Operating System :: OS Independent',
     'Programming Language :: Python',
     'Programming Language :: Python :: Implementation :: CPython',

From 06a8ca94843049aa50022f74a9c5c27eda3524ff Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Sun, 6 Apr 2025 16:22:34 -0400
Subject: [PATCH 04/10] py38 eol

---
 tox.ini | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tox.ini b/tox.ini
index 0803cbc..8f29766 100644
--- a/tox.ini
+++ b/tox.ini
@@ -1,6 +1,6 @@
 [tox]
 envlist =
-    py{38,39,310,311,312}-django{42}
+    py{39,310,311,312}-django{42}
     py{310,311,312,313,py3}-django{51,52}
 [testenv]
 deps =

From 4b81b755365c77ad943b4abf0c9b77be093c0f5c Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Sun, 6 Apr 2025 16:31:55 -0400
Subject: [PATCH 06/10] Update for new license format

---
 pyproject.toml | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pyproject.toml b/pyproject.toml
index f60452c..5ab0bda 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -10,6 +10,8 @@ authors = [
 description = "Deletes old files."
 readme = "README.rst"
 keywords = ["django"]
+license = "MIT"
+license-files = [ "LICENSE" ]
 classifiers = [
     'Development Status :: 5 - Production/Stable',
     'Environment :: Web Environment',

From dea575cc770a24a9cec72c54c25c42448e9cd6e6 Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Sun, 6 Apr 2025 16:46:05 -0400
Subject: [PATCH 08/10] py3.8 eon

---
 pyproject.toml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 5ab0bda..3b6b1b1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ classifiers = [
     'Programming Language :: Python :: Implementation :: PyPy',
     'Topic :: Utilities'
 ]
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 dynamic = ["version"]
 
 [project.urls]

From 24baebba7bdaa2d35394dfd518479e6421ffe80b Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Wed, 9 Apr 2025 20:30:01 -0400
Subject: [PATCH 09/10] Update test_all.py

---
 test/test_all.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/test_all.py b/test/test_all.py
index e5d64d0..2758c0b 100644
--- a/test/test_all.py
+++ b/test/test_all.py
@@ -2,6 +2,7 @@
 import os
 import pickle
 import re
+import sys
 import tempfile
 
 from django.conf import settings as django_settings
@@ -23,7 +24,8 @@
 
 LINE = re.compile(r'line \d{1,3}')
 
-TB = '''Traceback (most recent call last):
+if sys.version_info < (3, 13):
+    TB = '''Traceback (most recent call last):
   File "{handlers}", line xxx, in run_on_commit
     file_.delete(save=False)
   File "{files}", line xxx, in delete
@@ -31,6 +33,8 @@
   File "{storage}", line xxx, in delete
     os.remove(name)
 {error}: [Errno 2] No such file or directory: '{{picture}}\''''
+else:
+    TB = ''''''
 
 
 

From 26862a2b870866d4ac75e9499ea1802e064e1344 Mon Sep 17 00:00:00 2001
From: Mario Rosa <vinnyrose@users.noreply.github.com>
Date: Wed, 9 Apr 2025 20:38:17 -0400
Subject: [PATCH 10/10] Update test_all.py

---
 test/test_all.py | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/test/test_all.py b/test/test_all.py
index 2758c0b..a2ce89c 100644
--- a/test/test_all.py
+++ b/test/test_all.py
@@ -24,31 +24,31 @@
 
 LINE = re.compile(r'line \d{1,3}')
 
-if sys.version_info < (3, 13):
-    TB = '''Traceback (most recent call last):
-  File "{handlers}", line xxx, in run_on_commit
-    file_.delete(save=False)
-  File "{files}", line xxx, in delete
-    self.storage.delete(self.name)
-  File "{storage}", line xxx, in delete
-    os.remove(name)
-{error}: [Errno 2] No such file or directory: '{{picture}}\''''
-else:
-    TB = ''''''
-
 
 
 def get_traceback(picture):
     fileabspath = os.path.abspath
     error = 'FileNotFoundError'
-
-    return f'''Traceback (most recent call last):
+    if sys.version_info < (3, 13):
+        return f'''Traceback (most recent call last):
+  File "{fileabspath(handlers.__file__)}", line xxx, in run_on_commit
+    file_.delete(save=False)
+  File "{fileabspath(files.__file__)}", line xxx, in delete
+    self.storage.delete(self.name)
+  File "{fileabspath(storage.__file__)}", line xxx, in delete
+    os.remove(name)
+{error}: [Errno 2] No such file or directory: '{picture}\''''
+    else:
+        return f'''Traceback (most recent call last):
   File "{fileabspath(handlers.__file__)}", line xxx, in run_on_commit
     file_.delete(save=False)
+    ~~~~~~~~~~~~^^^^^^^^^^^^
   File "{fileabspath(files.__file__)}", line xxx, in delete
     self.storage.delete(self.name)
+    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^
   File "{fileabspath(storage.__file__)}", line xxx, in delete
     os.remove(name)
+    ~~~~~~~~~^^^^^^
 {error}: [Errno 2] No such file or directory: '{picture}\''''
 
 
