From bb6b1ef03c05dfd7f8209e923c2e8ec5365ccaac Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Mon, 08 Nov 2010 20:53:56 +0000
Subject: Fix crash in always group mode (bug #6792).

---
diff --git a/plugins/tasklist/tasklist-widget.c b/plugins/tasklist/tasklist-widget.c
index cdf696d..72283ba 100644
--- a/plugins/tasklist/tasklist-widget.c
+++ b/plugins/tasklist/tasklist-widget.c
@@ -1586,6 +1586,7 @@ xfce_tasklist_update_icon_geometries (gpointer data)
 
   toplevel = gtk_widget_get_toplevel (GTK_WIDGET (tasklist));
   gtk_window_get_position (GTK_WINDOW (toplevel), &root_x, &root_y);
+  panel_return_val_if_fail (XFCE_IS_TASKLIST (tasklist), FALSE);
 
   for (li = tasklist->windows; li != NULL; li = li->next)
     {
@@ -1603,7 +1604,7 @@ xfce_tasklist_update_icon_geometries (gpointer data)
 
         case CHILD_TYPE_GROUP:
           alloc = &child->button->allocation;
-          for (lp = child->windows; li != NULL; li = li->next)
+          for (lp = child->windows; lp != NULL; lp = lp->next)
             {
               child2 = lp->data;
               panel_return_val_if_fail (WNCK_IS_WINDOW (child2->window), FALSE);
@@ -2382,7 +2383,6 @@ xfce_tasklist_button_activate (XfceTasklistChild *child,
   WnckWorkspace *workspace;
 
   panel_return_if_fail (XFCE_IS_TASKLIST (child->tasklist));
-  panel_return_if_fail (child->type == CHILD_TYPE_WINDOW);
   panel_return_if_fail (WNCK_IS_WINDOW (child->window));
 
   if (wnck_window_is_active (child->window))
--
cgit v0.8.3.4
