From de2ebcb59c77eeb30918d40271cb8ab0393d6fc0 Mon Sep 17 00:00:00 2001
From: Nick Schermer <nick@xfce.org>
Date: Sun, 07 Nov 2010 12:07:19 +0000
Subject: Don't destroy the about dialog.

---
diff --git a/panel/panel-dialogs.c b/panel/panel-dialogs.c
index 14596e9..f2f3b15 100644
--- a/panel/panel-dialogs.c
+++ b/panel/panel-dialogs.c
@@ -44,9 +44,6 @@ panel_dialogs_show_about_email_hook (GtkAboutDialog *dialog,
 {
   if (g_strcmp0 ("tictactoe@xfce.org", uri) == 0)
     {
-      /* gtk needs to finish some stuff before the dialog can be closed */
-      exo_gtk_object_destroy_later (GTK_OBJECT (dialog));
-
       /* open tic-tac-toe */
       panel_tic_tac_toe_show ();
     }
--
cgit v0.8.3.4
