--- seamonkey-2.40/mozilla/xpcom/base/nsMemoryReporterManager.cpp.orig
+++ seamonkey-2.40/mozilla/xpcom/base/nsMemoryReporterManager.cpp
@@ -176,6 +176,7 @@
 };
 NS_IMPL_ISUPPORTS(ResidentUniqueReporter, nsIMemoryReporter)
 
+#ifdef __GLIBC__
 #define HAVE_SYSTEM_HEAP_REPORTER 1
 nsresult
 SystemHeapSize(int64_t* aSizeOut)
@@ -195,6 +196,7 @@
     *aSizeOut = size_t(info.hblkhd) + size_t(info.uordblks);
     return NS_OK;
 }
+#endif
 
 #elif defined(__DragonFly__) || defined(__FreeBSD__) \
     || defined(__NetBSD__) || defined(__OpenBSD__) \
