https://kannel.org/issues/587

--- gateway-1.5.0/gwlib/log.c.orig	2016-04-05 15:56:21.757679376 +0300
+++ gateway-1.5.0/gwlib/log.c	2016-04-05 15:59:43.253393207 +0300
@@ -431,13 +431,8 @@
     int translog;
     
     if (level >= sysloglevel && dosyslog) {
-	if (args == NULL) {
-	    strncpy(buf, format, sizeof(buf));
-	    buf[sizeof(buf) - 1] = '\0';
-	} else {
-	    vsnprintf(buf, sizeof(buf), format, args);
-	    /* XXX vsnprint not 100% portable */
-	}
+	vsnprintf(buf, sizeof(buf), format, args);
+	/* XXX vsnprint not 100% portable */
 
 	switch(level) {
 	case GW_DEBUG:
