diff -upr gateway-1.5.0.orig/gw/wap_push_ota.c gateway-1.5.0/gw/wap_push_ota.c
--- gateway-1.5.0.orig/gw/wap_push_ota.c	2021-07-23 22:35:34.588435112 +0200
+++ gateway-1.5.0/gw/wap_push_ota.c	2021-07-23 22:36:18.868435897 +0200
@@ -113,8 +113,8 @@ static BearerboxAddress *bearerbox = NUL
 
 static List *ota_queue = NULL;
 
-wap_dispatch_func_t *dispatch_to_wsp;
-wap_dispatch_func_t *dispatch_to_wsp_unit;
+static wap_dispatch_func_t *dispatch_to_wsp;
+static wap_dispatch_func_t *dispatch_to_wsp_unit;
 
 /**************************************************************************
  *
diff -upr gateway-1.5.0.orig/wap/wtp_resp.c gateway-1.5.0/wap/wtp_resp.c
--- gateway-1.5.0.orig/wap/wtp_resp.c	2021-07-23 22:35:34.551768355 +0200
+++ gateway-1.5.0/wap/wtp_resp.c	2021-07-23 22:35:50.038440800 +0200
@@ -96,9 +96,9 @@ static Counter *resp_machine_id_counter
 static enum { limbo, running, terminating } resp_run_status = limbo;
 
 
-wap_dispatch_func_t *dispatch_to_wdp;
-wap_dispatch_func_t *dispatch_to_wsp;
-wap_dispatch_func_t *dispatch_to_push;
+static wap_dispatch_func_t *dispatch_to_wdp;
+static wap_dispatch_func_t *dispatch_to_wsp;
+static wap_dispatch_func_t *dispatch_to_push;
 
 /*
  * Queue of events to be handled by WTP responder.
