Use standard integer types instead.

Upstream: proposed.
--- a/basics.h	2020-06-04 14:05:29.657008378 +0200
+++ b/basics.h	2020-06-04 14:05:44.430412859 +0200
@@ -38,6 +38,7 @@
 #define _ISOC99_SOURCE 1
 #define _ISOC9X_SOURCE 1
 
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
@@ -49,14 +50,14 @@
 
 #include "ladspa.h"
 
-typedef __int8_t			int8;
-typedef __uint8_t			uint8;
-typedef __int16_t			int16;
-typedef __uint16_t		uint16;
-typedef __int32_t			int32;
-typedef __uint32_t		uint32;
-typedef __int64_t			int64;
-typedef __uint64_t		uint64;
+typedef int8_t			int8;
+typedef uint8_t			uint8;
+typedef int16_t			int16;
+typedef uint16_t		uint16;
+typedef int32_t			int32;
+typedef uint32_t		uint32;
+typedef int64_t			int64;
+typedef uint64_t		uint64;
 
 #define MIN_GAIN 1e-6 /* -120 dB */
 /* smallest non-denormal 32 bit IEEE float is 1.18e-38 */
