diff --git a/Makefile b/Makefile
index e0c20f3..2f25107 100644
--- a/Makefile
+++ b/Makefile
@@ -45,8 +45,6 @@ install:
 	sed -i "s|^package_locale_dir = '_not_set_'|package_locale_dir = '$(PACKAGE_LOCALE_DIR)'|" $(DESTDIR)/$(PREFIX)/bin/gtkman
 	install -D -m 644 src/gtkman.glade $(DESTDIR)/$(PREFIX)/share/gtkman/gtkman.glade
 	install -D -m 644 gtkman.desktop $(DESTDIR)/$(PREFIX)/share/applications/gtkman.desktop
-	[ -f man/gtkman.man ] && \
-		install -D -m 644 man/gtkman.man $(DESTDIR)/$(PREFIX)/share/man/man1/gtkman.1
 	install -d -m 755 $(DESTDIR)/$(PREFIX)/share/icons/hicolor/scalable/apps/
 	install -m 644 icons/gtkman.svg $(DESTDIR)/$(PREFIX)/share/icons/hicolor/scalable/apps/
 	for i in 32 24 22 16; do \
