--- a/spec/rmagick/image/liquid_rescale_spec.rb
+++ b/spec/rmagick/image/liquid_rescale_spec.rb
@@ -4,9 +4,8 @@
 
     begin
       image.liquid_rescale(15, 15)
-    rescue NotImplementedError
-      puts 'liquid_rescale not implemented.'
-      return
+    rescue NotImplementedError, Magick::ImageMagickError
+      skip 'liquid_rescale not implemented.'
     end
 
     result = image.liquid_rescale(15, 15)

