Description: Ensure that build is verbose
Author: Sébastien Villemot <sebastien@debian.org>
Author: Shiz <hi@shiz.me>
Forwarded: not-needed
Last-Update: 2017-05-19
---
--- a/deps/libuv.mk
+++ b/deps/libuv.mk
@@ -33,7 +33,7 @@
 	touch -c $(SRCDIR)/srccache/$(LIBUV_SRC_DIR)/configure
 	mkdir -p $(dir $@)
 	cd $(dir $@) && \
-	$< --with-pic $(CONFIGURE_COMMON) $(UV_FLAGS)
+	$< --with-pic $(CONFIGURE_COMMON) $(UV_FLAGS) --disable-silent-rules
 	touch -c $@
 $(UV_SRC_TARGET): $(BUILDDIR)/$(LIBUV_SRC_DIR)/config.status
 	$(MAKE) -C $(dir $<) $(UV_MFLAGS)
