Index: libx11/nls/locale.dir.pre
===================================================================
--- libx11.orig/nls/locale.dir.pre	2006-08-02 22:04:01.000000000 +0000
+++ libx11/nls/locale.dir.pre	2006-08-02 22:04:04.000000000 +0000
@@ -436,7 +436,7 @@
 pt_BR.UTF-8/XLC_LOCALE:			pt_BR.UTF-8
 en_US.UTF-8/XLC_LOCALE:			pt_PT.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ro_RO.UTF-8
-en_US.UTF-8/XLC_LOCALE:			ru_RU.UTF-8
+ru_RU.UTF-8/XLC_LOCALE:			ru_RU.UTF-8
 en_US.UTF-8/XLC_LOCALE:			ru_UA.UTF-8
 en_US.UTF-8/XLC_LOCALE:         rw_RW.UTF-8
 en_US.UTF-8/XLC_LOCALE:			se_NO.UTF-8
Index: libx11/nls/ru_RU.UTF-8/XLC_LOCALE.pre
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libx11/nls/ru_RU.UTF-8/XLC_LOCALE.pre	2006-08-02 22:04:04.000000000 +0000
@@ -0,0 +1,169 @@
+XCOMM
+XCOMM	XLocale Database Sample for ru_RU.UTF-8
+XCOMM	Derived from en_US.UTF-8
+XCOMM
+
+XCOMM
+XCOMM 	XLC_FONTSET category
+XCOMM
+XLC_FONTSET
+
+on_demand_loading	True
+
+object_name		generic
+
+XCOMM 	fs0 class (7 bit ASCII)
+fs0	{
+	charset	{
+		name	ISO8859-1:GL
+	}
+	font	{
+		primary	ISO8859-1:GL
+		vertical_rotate	all
+	}
+}
+XCOMM	fs1 class (ISO8859 families)
+fs1	{
+	charset	{
+		name	ISO8859-1:GR
+	}
+	font	{
+		primary	ISO8859-1:GR
+	}
+}
+XCOMM	fs2 class (KOI8-R)
+fs2	{
+	charset	{
+		name	KOI8-R:GR
+	}
+	font	{
+		primary	KOI8-R:GR
+	}
+}
+XCOMM	fs3 class (MICROSOFT-CP1251)
+fs3	{
+	charset	{
+		name	MICROSOFT-CP1251:GR
+	}
+	font	{
+		primary	MICROSOFT-CP1251:GR
+	}
+}
+XCOMM	fs4 class (ISO8859-5)
+fs4	{
+	charset	{
+		name	ISO8859-5:GR
+	}
+	font	{
+		primary	ISO8859-5:GR
+	}
+}
+XCOMM 	fs5 class (Kanji)
+fs5	{
+	charset	{
+		name	JISX0208.1983-0:GL
+	}
+	font	{
+		primary	JISX0208.1983-0:GL
+	}
+}
+XCOMM	fs6 class (Korean Character)
+fs6	{
+	charset	{
+		name	KSC5601.1987-0:GL
+	}
+	font	{
+		primary	KSC5601.1987-0:GL
+	}
+}
+XCOMM	fs7 class (Chinese Han Character)
+fs7	{
+	charset	{
+		name	GB2312.1980-0:GL
+	}
+	font	{
+		primary	GB2312.1980-0:GL
+	}
+}
+XCOMM	fs8 class (Half Kana)
+fs8	{
+	charset	{
+		name	JISX0201.1976-0:GR
+	}
+	font	{
+		primary		JISX0201.1976-0:GR
+		vertical_rotate	all
+	}
+}
+XCOMM	ISO10646 should come last so the fonts above will actually be used
+XCOMM	fs9 class
+fs9	{
+	charset	{
+		name	ISO10646-1
+	}
+	font	{
+		primary	ISO10646-1
+	}
+}
+END XLC_FONTSET
+
+XCOMM
+XCOMM 	XLC_XLOCALE category
+XCOMM
+XLC_XLOCALE
+
+encoding_name		UTF-8
+mb_cur_max		6
+state_depend_encoding	False
+
+XCOMM	cs0 class
+cs0	{
+	side		GL:Default
+	length		1
+	ct_encoding	ISO8859-1:GL
+}
+
+XCOMM	cs1 class
+cs1     {
+        side            GR:Default
+        length          1
+        ct_encoding     ISO8859-1:GR
+}
+
+XCOMM	cs2 class
+cs2	{
+	side		GR
+	length		2
+	ct_encoding	JISX0208.1983-0:GL; JISX0208.1983-0:GR;			JISX0208.1983-1:GL; JISX0208.1983-1:GR
+
+}
+
+XCOMM	cs3 class
+cs3     {
+        side            GL
+        length          2
+        ct_encoding     KSC5601.1987-0:GL; KSC5601.1987-0:GR;                        KSC5601.1987-1:GL; KSC5601.1987-1:GR
+
+}
+
+XCOMM	cs4 class
+cs4     {
+        side            GR
+        length          2
+        ct_encoding     GB2312.1980-0:GL; GB2312.1980-0:GR
+}
+
+XCOMM	cs5 class
+cs5	{
+	side		GR
+	length		1
+	ct_encoding	JISX0201.1976-0:GR
+}
+
+XCOMM	cs6 class
+cs6	{
+	side		none
+	ct_encoding	ISO10646-1
+}
+
+END XLC_XLOCALE
Index: libx11/nls/ru_RU.UTF-8/XI18N_OBJS
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libx11/nls/ru_RU.UTF-8/XI18N_OBJS	2006-08-02 22:04:04.000000000 +0000
@@ -0,0 +1,7 @@
+# CATEGORY(XLC|XIM|OM)	SHARED_LIBRARY_NAME	FUNCTION_NAME
+#
+#	XI18N objects table for en_US.UTF-8 locale
+#
+XLC	common/xlcUTF8Load	_XlcUtf8Loader	# XLC_open
+XIM	common/ximcp	_XimOpenIM _XimRegisterIMInstantiateCallback  _XimUnRegisterIMInstantiateCallback # XIM_open	XIM_register XIM_unregister
+XOM	common/xomGeneric 	_XomGenericOpenOM	# XOM_open
Index: libx11/configure.ac
===================================================================
--- libx11.orig/configure.ac	2006-08-02 20:31:39.000000000 +0000
+++ libx11/configure.ac	2006-08-02 22:04:04.000000000 +0000
@@ -400,6 +400,7 @@
 	   nls/mulelao-1/Makefile
 	   nls/nokhchi-1/Makefile
 	   nls/pt_BR.UTF-8/Makefile
+	   nls/ru_RU.UTF-8/Makefile
 	   nls/tatar-cyr/Makefile
 	   nls/th_TH/Makefile
 	   nls/th_TH.UTF-8/Makefile
Index: libx11/nls/Makefile.am
===================================================================
--- libx11.orig/nls/Makefile.am	2006-08-02 20:31:30.000000000 +0000
+++ libx11/nls/Makefile.am	2006-08-02 22:04:04.000000000 +0000
@@ -52,6 +52,7 @@
 	mulelao-1		\
 	nokhchi-1		\
 	pt_BR.UTF-8		\
+	ru_RU.UTF-8		\
 	tatar-cyr		\
 	th_TH			\
 	th_TH.UTF-8		\
Index: libx11/nls/ru_RU.UTF-8/Compose.pre
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libx11/nls/ru_RU.UTF-8/Compose.pre	2006-08-02 22:04:04.000000000 +0000
@@ -0,0 +1 @@
+include "/usr/share/X11/locale/en_US.UTF-8/Compose"
Index: libx11/nls/ru_RU.UTF-8/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ libx11/nls/ru_RU.UTF-8/Makefile.am	2006-08-02 22:04:04.000000000 +0000
@@ -0,0 +1,3 @@
+x11thislocaledir = $(X11_LOCALEDATADIR)/ru_RU.UTF-8
+
+include $(top_srcdir)/nls/localerules.in
