Upstream: No
Reason: Hardcode log file path
--- a/sentinel.conf
+++ b/sentinel.conf
@@ -33,7 +33,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile /var/log/redis/sentinel.log
 
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
