diff --git a/src/abstractfile.h b/src/abstractfile.h
index 3efe7e0..c8a8f7a 100644
--- a/src/abstractfile.h
+++ b/src/abstractfile.h
@@ -1,6 +1,7 @@
 #ifndef ABSTRACTFILE_H
 #define ABSTRACTFILE_H 1
 
+#include <cstdint>
 #include <string>
 
 class AbstractFile
diff --git a/src/archive.h b/src/archive.h
index de82316..52e403e 100644
--- a/src/archive.h
+++ b/src/archive.h
@@ -1,6 +1,7 @@
 #ifndef ARCHIVE_H
 #define ARCHIVE_H
 
+#include <cstdint>
 #include <string>
 #include <vector>
 
