diff --git a/src/gsad.c b/src/gsad.c
index 73bbff2..65a0d22 100644
--- a/src/gsad.c
+++ b/src/gsad.c
@@ -6249,7 +6249,7 @@ main (int argc, char **argv)
     {
       if (termination_signal)
         {
-          g_debug ("Received %s signal.\n", sys_siglist[termination_signal]);
+          g_debug ("Received %s signal.\n", strsignal(termination_signal));
           gsad_cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           signal (termination_signal, SIG_DFL);
