diff -ru greenbone-security-assistant-7.0.2.orig/src/gsad.c greenbone-security-assistant-7.0.2/src/gsad.c
--- greenbone-security-assistant-7.0.2.orig/src/gsad.c	2017-03-09 15:59:19.000000000 +0200
+++ greenbone-security-assistant-7.0.2/src/gsad.c	2017-03-09 16:17:28.506470054 +0200
@@ -5488,6 +5488,7 @@
            NULL, NULL, handler, NULL, MHD_OPTION_NOTIFY_COMPLETED,
            free_resources, NULL, MHD_OPTION_LISTEN_SOCKET, unix_socket,
            MHD_OPTION_PER_IP_CONNECTION_LIMIT, 30,
+           MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
            MHD_OPTION_EXTERNAL_LOGGER, mhd_logger, NULL, MHD_OPTION_END);
 }
 
@@ -5513,6 +5514,7 @@
           (MHD_USE_THREAD_PER_CONNECTION | MHD_USE_DEBUG | ipv6_flag, port,
            NULL, NULL, handler, NULL, MHD_OPTION_NOTIFY_COMPLETED,
            free_resources, NULL, MHD_OPTION_SOCK_ADDR, address,
+           MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
            MHD_OPTION_PER_IP_CONNECTION_LIMIT, 30,
            MHD_OPTION_EXTERNAL_LOGGER, mhd_logger, NULL, MHD_OPTION_END);
 }
@@ -5540,6 +5542,7 @@
            MHD_OPTION_HTTPS_MEM_CERT, cert,
            MHD_OPTION_NOTIFY_COMPLETED, free_resources, NULL,
            MHD_OPTION_SOCK_ADDR, address,
+           MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
            MHD_OPTION_PER_IP_CONNECTION_LIMIT, 30,
            MHD_OPTION_HTTPS_PRIORITIES, priorities,
            MHD_OPTION_EXTERNAL_LOGGER, mhd_logger, NULL,
