diff --git a/gsad/src/CMakeLists.txt b/gsad/src/CMakeLists.txt
index 27dbb22..b7f6d6f 100644
--- a/gsad/src/CMakeLists.txt
+++ b/gsad/src/CMakeLists.txt
@@ -177,7 +177,7 @@ add_definitions (-DPREFIX="${CMAKE_INSTALL_PREFIX}")
 ## Install
 
 install (TARGETS gsad
-         RUNTIME DESTINATION ${SBINDIR}
+         RUNTIME DESTINATION ${BINDIR}
          LIBRARY DESTINATION ${LIBDIR}
          ARCHIVE DESTINATION ${LIBDIR}/static)
 
