diff --git a/gsad/src/gsad.c b/gsad/src/gsad.c
index f176b62..37a9d66 100644
--- a/gsad/src/gsad.c
+++ b/gsad/src/gsad.c
@@ -2556,6 +2556,7 @@ start_unix_http_daemon (const char *unix_socket_path,
     0, NULL, NULL, handler, http_handlers, MHD_OPTION_NOTIFY_COMPLETED,
     free_resources, NULL, MHD_OPTION_LISTEN_SOCKET, unix_socket,
     MHD_OPTION_PER_IP_CONNECTION_LIMIT, get_per_ip_connection_limit (),
+	MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
     MHD_OPTION_EXTERNAL_LOGGER, mhd_logger, NULL, MHD_OPTION_END);
 }
 
@@ -2590,6 +2591,7 @@ start_http_daemon (int port,
     flags, port, NULL, NULL, handler, http_handlers,
     MHD_OPTION_NOTIFY_COMPLETED, free_resources, NULL, MHD_OPTION_SOCK_ADDR,
     address, MHD_OPTION_PER_IP_CONNECTION_LIMIT, get_per_ip_connection_limit (),
+	MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
     MHD_OPTION_EXTERNAL_LOGGER, mhd_logger, NULL, MHD_OPTION_END);
 }
 
@@ -2623,6 +2625,7 @@ start_https_daemon (int port, const char *key, const char *cert,
     MHD_OPTION_HTTPS_MEM_KEY, key, MHD_OPTION_HTTPS_MEM_CERT, cert,
     MHD_OPTION_NOTIFY_COMPLETED, free_resources, NULL, MHD_OPTION_SOCK_ADDR,
     address, MHD_OPTION_PER_IP_CONNECTION_LIMIT, get_per_ip_connection_limit (),
+	MHD_OPTION_THREAD_STACK_SIZE, (size_t) 4*1024*1024,
     MHD_OPTION_HTTPS_PRIORITIES, priorities, MHD_OPTION_EXTERNAL_LOGGER,
     mhd_logger, NULL,
 /* LibmicroHTTPD 0.9.35 and higher. */
