From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 07 May 2022 21:11:03 +0200
Subject: [PATCH] Customize exclude_list to Alpine

--- a/libtree.c
+++ b/libtree.c
@@ -84,19 +84,9 @@
 
 // Libraries we do not show by default -- this reduces the verbosity quite a
 // bit.
-char const *exclude_list[] = {"ld-linux-aarch64.so",
-                              "ld-linux-armhf.so",
-                              "ld-linux-x86-64.so",
-                              "ld-linux.so",
-                              "ld64.so",
-                              "libc.musl-aarch64.so",
-                              "libc.musl-armhf.so",
-                              "libc.musl-i386.so",
-                              "libc.musl-x86_64.so",
+char const *exclude_list[] = {"libc.musl-@@CARCH@@.so",
                               "libc.so",
-                              "libdl.so",
                               "libgcc_s.so",
-                              "libm.so",
                               "libstdc++.so"};
 
 struct header_64_t {
