diff --git a/platform/linuxbsd/crash_handler_linuxbsd.cpp b/platform/linuxbsd/crash_handler_linuxbsd.cpp
index 3a24546..ae7430f 100644
--- a/platform/linuxbsd/crash_handler_linuxbsd.cpp
+++ b/platform/linuxbsd/crash_handler_linuxbsd.cpp
@@ -37,7 +37,6 @@
 #include "main/main.h"
 
 #ifdef DEBUG_ENABLED
-#define CRASH_HANDLER_ENABLED 1
 #endif
 
 #ifdef CRASH_HANDLER_ENABLED
diff --git a/platform/linuxbsd/detect.py b/platform/linuxbsd/detect.py
index 3f713d2..e27e1eb 100644
--- a/platform/linuxbsd/detect.py
+++ b/platform/linuxbsd/detect.py
@@ -429,8 +429,8 @@ def configure(env: "Environment"):
     if not env["execinfo"] and platform.libc_ver()[0] != "glibc":
         # The default crash handler depends on glibc, so if the host uses
         # a different libc (BSD libc, musl), fall back to libexecinfo.
-        print("Note: Using `execinfo=yes` for the crash handler as required on platforms where glibc is missing.")
-        env["execinfo"] = True
+        # print("Note: Using `execinfo=yes` for the crash handler as required on platforms where glibc is missing.")
+        env["execinfo"] = False
 
     if env["execinfo"]:
         env.Append(LIBS=["execinfo"])
