From 3b5c9c282b1e25ecc364f2b33b2b9bcb558a6db5 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Fri, 8 Jan 2021 13:18:07 +0100
Subject: [PATCH] Add apk (Alpine Linux) support

---
 pako/package_manager_data.py | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/pako/package_manager_data.py b/pako/package_manager_data.py
index 384a5f3..1d70f75 100644
--- a/pako/package_manager_data.py
+++ b/pako/package_manager_data.py
@@ -24,7 +24,7 @@ from collections import OrderedDict
 from pako.config_loader import recursive_merge, load_package_managers_overrides
 
 __package_managers = {
-    '__order__': ['eopkg', 'apt-get'],
+    '__order__': ['eopkg', 'apt-get', 'apk'],
     'eopkg': {
         'sudo': True,
         'update': 'ur',
@@ -46,6 +46,17 @@ __package_managers = {
             'lib-dev': ['lib{}-dev', '{}-dev'],
             'lib-debug': ['lib{}-dbg', '{}-dbg']
         }
+    },
+    'apk': {
+        'sudo': True,
+        'update': 'upgrade -a',
+        'install': 'add',
+        'formats': {
+            'exe': ['{}', '{}-utils', '{}-progs', '{}-tools'],
+            'lib': ['{}', 'lib{}', '{}-libs'],
+            'lib-dev': ['{}-dev'],
+            'lib-debug': ['{}-dbg'],
+        }
     }
 }

