--- a/src/external_acl.cc
+++ b/src/external_acl.cc
@@ -1243,7 +1243,7 @@
     if (def->cache_size <= 0 || entry->result == ACCESS_DUNNO)
         return 1;
 
-    if (entry->date + (entry->result == ACCESS_ALLOWED ? def->ttl : def->negative_ttl) < squid_curtime)
+    if (entry->date + (entry->result == ACCESS_ALLOWED ? def->ttl : def->negative_ttl) <= squid_curtime)
         return 1;
     else
         return 0;
