http://bugs.squid-cache.org/show_bug.cgi?id=3679

diff --git a/src/external_acl.cc b/src/external_acl.cc
index b3821c5..d6c4338 100644
--- a/src/external_acl.cc
+++ b/src/external_acl.cc
@@ -1152,7 +1152,7 @@ external_acl_entry_expired(external_acl * def, external_acl_entry * entry)
     if (def->cache_size <= 0)
         return 1;
 
-    if (entry->date + (entry->result == 1 ? def->ttl : def->negative_ttl) < squid_curtime)
+    if (entry->date + (entry->result == 1 ? def->ttl : def->negative_ttl) <= squid_curtime)
         return 1;
     else
         return 0;
