--- ./src/Makefile.in.orig	2013-05-20 11:50:04.000000000 +0000
+++ ./src/Makefile.in	2013-05-23 07:37:34.364978371 +0000
@@ -2362,7 +2362,7 @@
 noinst_LTLIBRARIES = libsquid.la
 cf_gen_SOURCES = cf_gen.cc
 nodist_cf_gen_HEADER = cf_gen_defines.cci
-cf_gen_LDADD = 
+cf_gen_LDADD =  -lpthread
 ACL_REGISTRATION_SOURCES = AclRegs.cc AuthReg.cc
 DISKIO_SOURCE = \
 	DiskIO/DiskIOModule.cc \
@@ -7295,7 +7295,7 @@
 
 # cf_gen builds the configuration files.
 cf_gen$(EXEEXT): $(cf_gen_SOURCES) $(cf_gen_DEPENDENCIES) cf_gen_defines.cci
-	$(HOSTCXX) -o $@ $(srcdir)/cf_gen.cc -I$(srcdir) -I$(top_builddir)/include/ -I$(top_builddir)/src
+	$(HOSTCXX) $(cf_gen_LDADD) -o $@ $(srcdir)/cf_gen.cc -I$(srcdir) -I$(top_builddir)/include/ -I$(top_builddir)/src
 
 # squid.conf.default is built by cf_gen when making cf_parser.cci
 squid.conf.default squid.conf.documented: cf_parser.cci
