Index: squid-2.7.STABLE9/src/access_log.c
===================================================================
--- squid-2.7.STABLE9.orig/src/access_log.c	2010-11-18 15:02:26.000000000 +0200
+++ squid-2.7.STABLE9/src/access_log.c	2010-11-18 15:07:34.000000000 +0200
@@ -301,6 +301,7 @@
     LFT_REQUEST_URLPATH,
 /*LFT_REQUEST_QUERY, * // * this is not needed. see strip_query_terms */
     LFT_REQUEST_VERSION,
+    LFT_REQUEST_URLGROUP,
 
     LFT_REQUEST_SIZE_TOTAL,
 /*LFT_REQUEST_SIZE_LINE, */
@@ -411,6 +412,7 @@
 /* { "rq", LFT_REQUEST_QUERY }, * /     / * the query-string, INCLUDING the leading ? */
     {">v", LFT_REQUEST_VERSION},
     {"rv", LFT_REQUEST_VERSION},
+    {"rG", LFT_REQUEST_URLGROUP},
 
     {">st", LFT_REQUEST_SIZE_TOTAL},
 /*{ ">sl", LFT_REQUEST_SIZE_LINE }, * / / * the request line "GET ... " */
@@ -544,6 +546,12 @@
 	    quote = 1;
 	    break;
 
+        case LFT_REQUEST_URLGROUP:
+            if (al->request)
+                out = al->request->urlgroup;
+            quote = 1;
+            break;
+
 	case LFT_REPLY_HEADER:
 	    if (al->reply)
 		sb = httpHeaderGetByName(&al->reply->header, fmt->data.header.header);
