From bc5f17010a35406ed78ac2226e461b71340743f0 Mon Sep 17 00:00:00 2001
From: 6543 <6543@obermui.de>
Date: Sun, 9 May 2021 13:28:17 +0200
Subject: [PATCH] dont test against non existing git repo

---
 modules/git/repo_test.go | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/git/repo_test.go b/modules/git/repo_test.go
index 0b6986764..e9b30201d 100644
--- a/modules/git/repo_test.go
+++ b/modules/git/repo_test.go
@@ -13,7 +13,8 @@ import (
 )
 
 func TestGetLatestCommitTime(t *testing.T) {
-	lct, err := GetLatestCommitTime(".")
+	bareRepo1Path := filepath.Join(testReposDir, "repo1_bare")
+	lct, err := GetLatestCommitTime(bareRepo1Path)
 	assert.NoError(t, err)
 	// Time is in the past
 	now := time.Now()
-- 
2.31.1

