diff -ruN a/modules/git/url/url_test.go b/modules/git/url/url_test.go
--- a/modules/git/url/url_test.go	2025-09-11 03:47:39.000000000 +0000
+++ b/modules/git/url/url_test.go	2025-10-09 02:31:16.580000000 +0000
@@ -137,11 +137,11 @@
 			},
 		},
 		{
-			kase: "https://[fe80:14fc:cec5:c174:d88%2510]:20/go-gitea/gitea.git",
+			kase: "https://[fe80::14fc:cec5:c174:d88%2510]:20/go-gitea/gitea.git",
 			expected: &GitURL{
 				URL: &url.URL{
 					Scheme: "https",
-					Host:   "[fe80:14fc:cec5:c174:d88%10]:20",
+					Host:   "[fe80::14fc:cec5:c174:d88%10]:20",
 					Path:   "/go-gitea/gitea.git",
 				},
 				extraMark: 0,
