--- icedtea-3.3.0/pax-mark-vm.in.orig	2017-02-09 09:42:32.652307034 +0000
+++ icedtea-3.3.0/pax-mark-vm.in	2017-02-09 09:43:46.043633914 +0000
@@ -9,17 +9,15 @@
 }
 
 if test "x@PAX_COMMAND@" != "x"; then
-    if @PAX_COMMAND@ -m "${JDK}"/bin/java; then
 	for paxable in `list_paxables "${JDK}"/bin/* "${JDK}"/jre/bin/*`; do
 		echo "PaX mark @PAX_COMMAND_ARGS@ ${paxable}"
-		@PAX_COMMAND@ @PAX_COMMAND_ARGS@ "${paxable}"
+		if ! @PAX_COMMAND@ @PAX_COMMAND_ARGS@ "${paxable}"; then
+			if test "x${FAIL_ON_ERROR}" = "xtrue"; then
+			    echo "ERROR: Could not apply PaX markings to files in ${JDK}";
+			    exit 1;
+			else
+			    echo "WARNING: Could not apply PaX markings to files in ${JDK}";
+			fi
+		fi
 	done
-    else
-	if test "x${FAIL_ON_ERROR}" = "xtrue"; then
-	    echo "ERROR: Could not apply PaX markings to files in ${JDK}";
-	    exit 1;
-	else
-	    echo "WARNING: Could not apply PaX markings to files in ${JDK}";
-	fi
-    fi
 fi
