diff --git a/configure.ac b/configure.ac
index bddace0..b4bb928 100644
--- a/configure.ac
+++ b/configure.ac
@@ -121,6 +121,9 @@ AC_MSG_RESULT($HOST)
 #check for -lm
 AC_CHECK_LIB(m, pow, [ haslm="yes" ] , [ AC_MSG_ERROR([libm not found !]) ] )
 
+# check for dbus-glib
+PKG_CHECK_MODULES(DBUS_GLIB, dbus-glib-1)
+
 #################################################
 # Initialize the gnome integration
 #################################################
diff --git a/src/Makefile.am b/src/Makefile.am
index fce2976..6e80355 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -48,7 +48,7 @@ mp3splt_gtk_SOURCES = \
 
 INCLUDES = @GTK_CFLAGS@ @MP3SPLT_CFLAGS@
 
-mp3splt_gtk_LDADD = @GTK_LIBS@ -lgthread-2.0 @MP3SPLT_LIBS@ -lm
+mp3splt_gtk_LDADD = @GTK_LIBS@ -lgthread-2.0 @MP3SPLT_LIBS@ @DBUS_GLIB_LIBS@ -lm
 
 # Define a C macro LOCALEDIR indicating where catalogs will be installed.
 localedir = $(datadir)/locale
