diff --git a/client/scripts/linux b/client/scripts/linux
index 0c42969..2e7274b 100755
--- a/client/scripts/linux
+++ b/client/scripts/linux
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # dhclient-script for Linux. Dan Halbert, March, 1997.
 # Updated for Linux 2.[12] by Brian J. Murrell, January 1999.
 # No guarantees about this. I'm a novice at the details of Linux
@@ -428,7 +428,7 @@ case "$reason" in
             # Check if any IPv6 address on this interface is marked as
             # tentative.
             ${ip} addr show ${interface} | grep inet6 | grep tentative \
-                &> /dev/null
+                > /dev/null 2>&1
             if [ $? -eq 0 ]; then
                 # Wait for duplicate address detection to complete or for
                 # the timeout specified as --dad-wait-time.
@@ -437,7 +437,7 @@ case "$reason" in
                     # We're going to poll for the tentative flag every second.
                     sleep 1
                     ${ip} addr show ${interface} | grep inet6 | grep tentative \
-                        &> /dev/null
+                        > /dev/null 2>&1
                     if [ $? -ne 0 ]; then
                         break;
                     fi
