Description: Use /usr/bin/perl as Perl interpreter
Author: Ondřej Nový <onovy@debian.org>

--- a/src/vifm-convert-dircolors
+++ b/src/vifm-convert-dircolors
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 
 # vifm
 # Copyright (C) 2015 xaizek.
--- a/src/helpztags
+++ b/src/helpztags
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/usr/bin/perl
 #
 # helpztags generates tags for Vim helpfiles, for both .txt and .txt.gz files
 # Author: Jakub Turski <yacoob@chruptak.plukwa.net>
