diff -ruN a/tests/js/rtorrent.spec.js b/tests/js/rtorrent.spec.js
--- a/tests/js/rtorrent.spec.js
+++ b/tests/js/rtorrent.spec.js
@@ -17,6 +17,7 @@
 
 for (const src of [
   "../lang/en.js",
+  "../js/browser.js",
   "../js/common.js",
   "../js/content.js",
   "../js/rtorrent.js",
diff -ruN a/tests/plugins/rss/init.spec.js b/tests/plugins/rss/init.spec.js
--- a/tests/plugins/rss/init.spec.js
+++ b/tests/plugins/rss/init.spec.js
@@ -39,6 +39,7 @@
   "../js/custom-elements.js",
   "../js/category-list-elements.js",
   "../lang/en.js",
+  "../js/browser.js",
   "../js/common.js",
   "../js/content.js",
   "../js/rtorrent.js",
