--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,6 @@
 enable_testing()
 find_package(PkgConfig REQUIRED)
 find_package(Threads REQUIRED)
-find_package(Backtrace REQUIRED)
 find_package(Filesystem REQUIRED)
 if(WITH_GETTEXT)
 	find_package(Gettext)
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -255,7 +255,6 @@
 target_link_libraries(${APP_EXEC}
 	${Intl_LIBRARIES}
 	${CMAKE_THREAD_LIBS_INIT}
-	${Backtrace_LIBRARIES}
 	std::filesystem
 
 	${GTK3_LIBRARIES}
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -22,7 +22,6 @@
 */
 
 #include <unistd.h>
-#include <execinfo.h>
 #include <getopt.h>
 #include <cstdlib>
 #include <cstring>
@@ -370,10 +369,6 @@
 	void *bt[16];
 	std::string::size_type addr_start, addr_end;
 	std::string line, address, line_number;
-
-	/* Get the backtrace */
-	bt_size = backtrace(bt, 16);
-	bt_syms = backtrace_symbols(bt, bt_size);
 
 	/* Print the backtrace */
 	Logger::set_verbosity(LOG_ERROR); // disable DEBUG mode for popen_to_str()
