Author: Chris Hofstaedtler <zeha@debian.org>
Bug: https://github.com/lgi-devs/lgi/issues/336
Bug-Debian: https://bugs.debian.org/1101238
Forwarded: https://github.com/lgi-devs/lgi/pull/342
Description: Adapt to new Pango 1.56.2 bindings

--- a/tests/pango.lua
+++ b/tests/pango.lua
@@ -35,8 +35,7 @@
       local offset = items[i].offset
       local length = items[i].length
       local analysis = items[i].analysis
-      local pgs = Pango.GlyphString()
-      Pango.shape(string.sub(s,1+offset), length, analysis, pgs)
+      pgs = Pango.shape(string.sub(s,1+offset), length, analysis)
       -- Pull out individual glyphs with pgs.glyphs
       local glyphs = pgs.glyphs
       check(type(glyphs) == 'table')
