diff -urN nagios-4.4.6.orig/cgi/getcgi.c nagios-4.4.6/cgi/getcgi.c
--- nagios-4.4.6.orig/cgi/getcgi.c	2021-03-12 18:00:28.712911163 -0700
+++ nagios-4.4.6/cgi/getcgi.c	2021-03-12 18:02:09.746892595 -0700
@@ -245,7 +245,9 @@
 		formid = strstr(cookies, "NagFormId=");
 		if (formid) {
 			if(!(paircount % 256)) {
-				pairlist = (char **)realloc(pairlist, (paircount + 1) * sizeof(char *));
+				/* if no query parameters were provided, paircount can begin as zero, resulting in */
+				/* truncation of the pairlist array if we do not reserve at least two elements. */
+				pairlist = (char **)realloc(pairlist, (paircount + 2) * sizeof(char *));
 				if(pairlist == NULL) {
 					printf("getcgivars(): Could not re-allocate memory for name-value pairlist.\n");
 					exit(1);
