--- a/cmd/tools/vbuild-tools.v
+++ b/cmd/tools/vbuild-tools.v
@@ -16,7 +16,7 @@
 // non_packaged_tools are tools that should not be packaged with
 // prebuild versions of V, to keep the size smaller.
 // They are mainly usefull for the V project itself, not to end users.
-const non_packaged_tools = ['gen1m', 'gen_vc', 'fast', 'wyhash']
+const non_packaged_tools = ['gen1m', 'gen_vc', 'fast', 'wyhash', 'v']
 
 fn main() {
 	util.ensure_modules_for_all_tools_are_installed('-v' in os.args)
