crude hack to fix random display freeze

--- a/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c	2021-01-26 11:15:26.757702834 +0000
+++ b/drivers/gpu/drm/bridge/analogix/analogix_dp_core.c	2021-01-26 11:15:26.757702834 +0000
@@ -979,12 +979,6 @@
 	if (ret)
 		return ret;
 
-	if (analogix_dp_detect_sink_psr(dp)) {
-		ret = analogix_dp_enable_sink_psr(dp);
-		if (ret)
-			return ret;
-	}
-
 	return ret;
 }
 
