From 080d086b0bae0b779f14619b319ce8e60af2558b Mon Sep 17 00:00:00 2001
From: firefly-cpp <iztok@iztok-jr-fister.eu>
Date: Fri, 28 Apr 2023 21:27:44 +0200
Subject: [PATCH] Do not package/install the tests

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index ce5a243..947c2f2 100644
--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@
 setup(
     name='leap_ec',
     version=__version__,
-    packages=find_packages(),
+    packages=find_packages(exclude=["*tests*"]),
     license='Academic',
     author='Mark Coletti, Eric Scott, Jeff Bassett',
     author_email='mcoletti@gmail.com',
