diff --git a/Makefile b/Makefile
index 917bd5c..fa1e2e9 100644
--- a/Makefile
+++ b/Makefile
@@ -700,7 +700,7 @@ endif
 ifeq ($(optimize),yes)
 ifeq ($(debug), no)
 	ifeq ($(comp),clang)
-		CXXFLAGS += -flto=full
+		#CXXFLAGS += -flto=full
 		ifeq ($(target_windows),yes)
 			CXXFLAGS += -fuse-ld=lld
 		endif
@@ -710,11 +710,11 @@ ifeq ($(debug), no)
 # GCC on some systems.
 	else ifeq ($(comp),gcc)
 	ifeq ($(gccisclang),)
-		CXXFLAGS += -flto -flto-partition=one
-		LDFLAGS += $(CXXFLAGS) -flto=jobserver
+		#CXXFLAGS += -flto -flto-partition=one
+		#LDFLAGS += $(CXXFLAGS) -flto=jobserver
 	else
-		CXXFLAGS += -flto=full
-		LDFLAGS += $(CXXFLAGS)
+		#CXXFLAGS += -flto=full
+		#LDFLAGS += $(CXXFLAGS)
 	endif
 
 # To use LTO and static linking on Windows,
