--- a/pocaml_c.c
+++ b/pocaml_c.c
@@ -8,12 +8,12 @@
 #define CAML_NAME_SPACE 1
 
 /* Fix me (probably needs some setting the Makefile) */ 
-#include </usr/local/lib/ocaml/caml/alloc.h>
-#include </usr/local/lib/ocaml/caml/callback.h>
-#include </usr/local/lib/ocaml/caml/custom.h>
-#include </usr/local/lib/ocaml/caml/fail.h>
-#include </usr/local/lib/ocaml/caml/memory.h>
-#include </usr/local/lib/ocaml/caml/mlvalues.h>
+#include </usr/lib/ocaml/caml/alloc.h>
+#include </usr/lib/ocaml/caml/callback.h>
+#include </usr/lib/ocaml/caml/custom.h>
+#include </usr/lib/ocaml/caml/fail.h>
+#include </usr/lib/ocaml/caml/memory.h>
+#include </usr/lib/ocaml/caml/mlvalues.h>
 
 #include <EXTERN.h>
 #include <perl.h>
