--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -352,7 +352,7 @@
 AM_CPPFLAGS = $(tls_CFLAGS) $(libgsasl_CFLAGS) $(libidn2_CFLAGS) $(libsecret_CFLAGS)
 msmtp_LDADD = $(tls_LIBS) $(libgsasl_LIBS) $(libidn2_LIBS) $(libsecret_LIBS) $(LIBINTL)
 @BUILD_MSMTPD_TRUE@msmtpd_SOURCES = msmtpd.c base64.c base64.h password.c password.h tools.c tools.h xalloc.c xalloc.h netrc.c netrc.h
-@BUILD_MSMTPD_TRUE@msmtpd_LDADD = $(libsecret_LIBS)
+@BUILD_MSMTPD_TRUE@msmtpd_LDADD = $(libsecret_LIBS) $(LIBINTL)
 all: all-am
 
 .SUFFIXES:
