From 7457e26d3a78c7cd923242d87d04febadddea086 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Thu, 5 Feb 2015 10:06:31 +0200
Subject: [PATCH] maintain abi compat with no-freelist and regular build

---
 ssl/ssl.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ssl/ssl.h b/ssl/ssl.h
index 2b0f662..636cb5d 100644
--- a/ssl/ssl.h
+++ b/ssl/ssl.h
@@ -1113,6 +1113,10 @@ struct ssl_ctx_st {
     unsigned int freelist_max_len;
     struct ssl3_buf_freelist_st *wbuf_freelist;
     struct ssl3_buf_freelist_st *rbuf_freelist;
+#  else
+    unsigned int freelist_dummy0;
+    void *freelist_dummy1;
+    void *freelist_dummy2;
 #  endif
 #  ifndef OPENSSL_NO_SRP
     SRP_CTX srp_ctx;            /* ctx for SRP authentication */
-- 
2.2.2

