This test doesn't work without locale support
diff --git a/libgweather/test_libgweather.c b/libgweather/test_libgweather.c
index 8a21911..0716c41 100644
--- a/libgweather/test_libgweather.c
+++ b/libgweather/test_libgweather.c
@@ -653,19 +653,6 @@ test_location_names (void)
     g_assert_cmpstr (gweather_location_get_sort_name (brussels), ==, "brussels");
     g_assert_cmpstr (gweather_location_get_english_name (brussels), ==, "Brussels");
 
-    setlocale (LC_ALL, "fr_FR.UTF-8");
-    _gweather_location_reset_world ();
-
-    world = gweather_location_get_world ();
-    g_assert (world);
-
-    brussels = gweather_location_find_nearest_city (world, 50.833333, 4.333333);
-    g_assert (brussels);
-    g_assert_cmpstr (gweather_location_get_name (brussels), ==, "Bruxelles");
-    g_assert_cmpstr (gweather_location_get_sort_name (brussels), ==, "bruxelles");
-    g_assert_cmpstr (gweather_location_get_english_name (brussels), ==, "Brussels");
-
-    setlocale (LC_ALL, "");
     _gweather_location_reset_world ();
 }
 
