--- a/requirements.txt
+++ b/requirements.txt
@@ -1,3 +1,3 @@
-PyYAML ~= 3.11
+PyYAML
 ansicolor ~= 0.2.4
 chardet >= 2.3.0
--- a/vint/linting/config/config_file_source.py
+++ b/vint/linting/config/config_file_source.py
@@ -10,7 +10,7 @@
         config_file_path = self.get_file_path(env)
 
         with config_file_path.open() as file_obj:
-            self._config_dict = self.convert_config_dict(yaml.load(file_obj))
+            self._config_dict = self.convert_config_dict(yaml.full_load(file_obj))
 
 
     def convert_config_dict(self, yaml_dict):
