--- ./limits/util.py.orig
+++ ./limits/util.py
@@ -8,7 +8,7 @@
 from types import ModuleType
 from typing import TYPE_CHECKING, cast
 
-import importlib_resources
+from importlib import resources as importlib_resources
 from packaging.version import Version
 
 from limits.typing import Dict, List, NamedTuple, Optional, Tuple, Type, Union
--- ./limits/typing.py.orig
+++ ./limits/typing.py
@@ -13,7 +13,7 @@
     Union,
 )
 
-from typing_extensions import ClassVar, Counter, ParamSpec, Protocol, TypeAlias
+from typing import ClassVar, Counter, ParamSpec, Protocol, TypeAlias
 
 Serializable = Union[int, str, float]
 
