diff --git a/tests/conftest.py.orig b/tests/conftest.py
index 2aeb758dda6..a9b2b8b2bd1 100644
--- a/tests/conftest.py.orig
+++ b/tests/conftest.py
@@ -3,7 +3,6 @@ import platform
 import socket
 import time
 
-import etcd3
 import pymemcache
 import pymemcache.client
 import pymongo
diff --git a/pytest.ini.orig b/pytest.ini
index 38c40a713d0..8c6659e21c2 100644
--- a/pytest.ini.orig
+++ b/pytest.ini
@@ -17,7 +17,6 @@ addopts =
     -rfEsxX
     --cov=limits
     -m "not benchmark"
-    -K
 filterwarnings =
     error
     module::ResourceWarning
