diff --git a/config b/config
index a2e36ec..6243fa2 100644
--- a/config
+++ b/config
@@ -16,7 +16,7 @@
 # IPv6 syntax: [address]:port
 # For example: 0.0.0.0:9999, [::]:9999
 # IPv6 adresses are configured to only allow IPv6 connections
-#hosts = 0.0.0.0:5232
+hosts = 127.0.0.1:5232
 
 # Daemon flag
 #daemon = False
@@ -28,10 +28,10 @@
 #ssl = False
 
 # SSL certificate path
-#certificate = /etc/apache2/ssl/server.crt
+certificate = /etc/ssl/radicale/server.crt
 
 # SSL private key
-#key = /etc/apache2/ssl/server.key
+key = /etc/ssl/radicale/server.key
 
 # SSL Protocol used. See python's ssl module for available values
 #protocol = PROTOCOL_SSLv23
@@ -146,7 +146,7 @@
 #custom_handler =
 
 # File for rights management from_file
-#file = ~/.config/radicale/rights
+file = /var/lib/radicale/rights
 
 
 [storage]
@@ -163,7 +163,7 @@
 #custom_handler =
 
 # Folder for storing local collections, created if not present
-#filesystem_folder = ~/.config/radicale/collections
+filesystem_folder = /var/lib/radicale/collections
 
 # Database URL for SQLAlchemy
 # dialect+driver://user:password@host/dbname[?key=value..]
diff --git a/logging b/logging
index f4efb88..b10f88c 100644
--- a/logging
+++ b/logging
@@ -45,7 +45,7 @@ formatter = simple
 [handler_file]
 # File handler
 class = FileHandler
-args = ('/var/log/radicale',)
+args = ('/var/log/radicale/radicale.log',)
 formatter = full
 
 
