diff --git a/src/core/exif.cpp b/src/core/exif.cpp
index 6bc5c52..266abec 100644
--- a/src/core/exif.cpp
+++ b/src/core/exif.cpp
@@ -108,7 +108,7 @@ int siril_get_thumbnail_exiv(const char *path, uint8_t **buffer, size_t *size, c
 		//std::cerr << "[exiv2] "<< path << ": found thumbnail "<< preview.width() << "x" << preview.height() << std::endl;
 		memcpy(*buffer, tmp, _size);
 		return 0;
-	} catch (Exiv2::AnyError &e) {
+	} catch (Exiv2::Error &e) {
 		std::string s(e.what());
 		std::cerr << "[exiv2]: " << s << std::endl;
 		return 1;
