diff --git a/lib/parser-smi.h b/lib/parser-smi.h
index 90d7ec2..32dc2a8 100644
--- a/lib/parser-smi.h
+++ b/lib/parser-smi.h
@@ -24,7 +24,7 @@
 
 
 
-extern int smiparse();
+extern int smiparse(struct Parser *parser);
 
 
 
diff --git a/lib/parser-sming.h b/lib/parser-sming.h
index b88e847..8e58760 100644
--- a/lib/parser-sming.h
+++ b/lib/parser-sming.h
@@ -24,7 +24,7 @@
 
 
 
-extern int smingparse();
+extern int smingparse(struct Parser *parser);
 
 
 
diff --git a/lib/parser-yang.h b/lib/parser-yang.h
index dbfec46..e059474 100644
--- a/lib/parser-yang.h
+++ b/lib/parser-yang.h
@@ -13,7 +13,7 @@
 
 
 
-extern int yangparse();
+extern int yangparse(struct Parser *parser);
 
 
 
diff --git a/lib/smi-data.c b/lib/smi-data.c
index c9d95a4..6a3de02 100644
--- a/lib/smi-data.c
+++ b/lib/smi-data.c
@@ -39,12 +39,12 @@
 
 #ifdef BACKEND_SMI
 #include "scanner-smi.h"
-extern int smiparse();
+extern int smiparse(struct Parser *parser);
 #endif
 
 #ifdef BACKEND_SMING
 #include "scanner-sming.h"
-extern int smingparse();
+extern int smingparse(struct Parser *parser);
 #endif
 
 #define stringKind(kind) ( \
@@ -4655,7 +4655,7 @@ Module *loadModule(const char *modulename, Parser *parserPtr)
 	}
 	smiDepth++;
 	parser.line			= 1;
-	smiparse((void *)&parser);
+	smiparse(&parser);
 #if 0
 	/*
 	 * These nodes seem to be referenced in certain situations and
@@ -4710,7 +4710,7 @@ Module *loadModule(const char *modulename, Parser *parserPtr)
 	}
 	smiDepth++;
 	parser.line			= 1;
-	smingparse((void *)&parser);
+	smingparse(&parser);
 #if 0
 	/*
 	 * These nodes seem to be referenced in certain situations and
