diff --git a/weasyprint/document.py b/weasyprint/document.py
index 4401f86d..c45d777b 100644
--- a/weasyprint/document.py
+++ b/weasyprint/document.py
@@ -234,12 +234,18 @@ class Document:
 
     @classmethod
     def _render(cls, html, font_config, counter_style, options):
+        import gc
+        import sys
         if font_config is None:
             font_config = FontConfiguration()
 
         if counter_style is None:
             counter_style = CounterStyle()
 
+        gc_was_enabled = gc.isenabled()
+        if sys.maxsize <= 2**32 and gc_was_enabled:
+            gc.disable()
+        
         context = cls._build_layout_context(
             html, font_config, counter_style, options)
 
@@ -254,6 +260,11 @@ class Document:
             DocumentMetadata(**get_html_metadata(html)),
             html.url_fetcher, font_config)
         rendering._html = html
+
+        if sys.maxsize <= 2**32 and gc_was_enabled:
+            gc.enable()
+            gc.collect()
+
         return rendering
 
     def __init__(self, pages, metadata, url_fetcher, font_config):
diff --git a/weasyprint/layout/block.py b/weasyprint/layout/block.py
index 38e53999..7aa86d38 100644
--- a/weasyprint/layout/block.py
+++ b/weasyprint/layout/block.py
@@ -907,8 +907,8 @@ def collapse_margin(adjoining_margins):
     """Get the amount of collapsed margin for a list of adjoining margins."""
     margins = [0]  # add 0 to make sure that max/min don’t get an empty list
     margins.extend(adjoining_margins)
-    positives = (m for m in margins if m >= 0)
-    negatives = (m for m in margins if m <= 0)
+    positives = [m for m in margins if m >= 0]
+    negatives = [m for m in margins if m <= 0]
     return max(positives) + min(negatives)
 
 
