diff --git a/settings/meson.build b/settings/meson.build
index 7356790..e2383a9 100644
--- a/settings/meson.build
+++ b/settings/meson.build
@@ -27,11 +27,6 @@ config = configuration_data()
 config.set('pkgdatadir', pkgdatadir)
 
 GIT_TAG = ''
-git = find_program('git', required : false)
-if git.found()
-  GIT_TAG = run_command('git', 'rev-parse', '--short', 'HEAD', check: true).stdout().strip()
-endif
-
 config.set('version', '@0@.@1@'.format(meson.project_version(), GIT_TAG))
 
 configure_file(
