diff --git a/src/linux/mei.c b/src/linux/mei.c
index 3a500ce..a43246d 100644
--- a/src/linux/mei.c
+++ b/src/linux/mei.c
@@ -193,7 +193,7 @@ static inline int __mei_connect(struct mei *me, struct mei_connect_client_data *
 	int rc;
 
 	errno = 0;
-	rc = ioctl(me->fd, IOCTL_MEI_CONNECT_CLIENT, d);
+	rc = ioctl(me->fd, (int)IOCTL_MEI_CONNECT_CLIENT, d);
 	me->last_err = errno;
 	return rc == -1 ? -me->last_err : 0;
 }
@@ -204,7 +204,7 @@ static inline int __mei_connect_vtag(struct mei *me,
 	int rc;
 
 	errno = 0;
-	rc = ioctl(me->fd, IOCTL_MEI_CONNECT_CLIENT_VTAG, d);
+	rc = ioctl(me->fd, (int)IOCTL_MEI_CONNECT_CLIENT_VTAG, d);
 	me->last_err = errno;
 	return rc == -1 ? -me->last_err : 0;
 }
@@ -214,7 +214,7 @@ static inline int __mei_notify_set(struct mei *me, uint32_t *enable)
 	int rc;
 
 	errno = 0;
-	rc = ioctl(me->fd, IOCTL_MEI_NOTIFY_SET, enable);
+	rc = ioctl(me->fd, (int)IOCTL_MEI_NOTIFY_SET, enable);
 	me->last_err = errno;
 	return rc == -1 ? -me->last_err : 0;
 }
@@ -225,7 +225,7 @@ static inline int __mei_notify_get(struct mei *me)
 	int rc;
 
 	errno = 0;
-	rc = ioctl(me->fd, IOCTL_MEI_NOTIFY_GET, &notification);
+	rc = ioctl(me->fd, (int)IOCTL_MEI_NOTIFY_GET, &notification);
 	me->last_err = errno;
 	return rc == -1 ? -me->last_err : 0;
 }
