From c6f5d2c235ee47b4dd793e2f033cc2394960a140 Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Fri, 24 Jan 2025 11:47:39 +0100
Subject: [PATCH] make TestNoRaceSeqSetRelativeSpeed more forgiving

---
 server/avl/norace_test.go | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/server/avl/norace_test.go b/server/avl/norace_test.go
index af0c3983..ea1d71bd 100644
--- a/server/avl/norace_test.go
+++ b/server/avl/norace_test.go
@@ -170,12 +170,12 @@ func TestNoRaceSeqSetRelativeSpeed(t *testing.T) {
 
 	// In general we are between 1.5 and 1.75 times slower atm then a straight map.
 	// Let's test an upper bound of 2x for now.
-	if mapInsertElapsed*2 <= ssInsertElapsed {
-		t.Fatalf("Expected SequenceSet insert to be no more than 2x slower (%v vs %v)", mapInsertElapsed, ssInsertElapsed)
+	if mapInsertElapsed*3 <= ssInsertElapsed {
+		t.Fatalf("Expected SequenceSet insert to be no more than 3x slower (%v vs %v)", mapInsertElapsed, ssInsertElapsed)
 	}
 
-	if mapLookupElapsed*3 <= ssLookupElapsed {
-		t.Fatalf("Expected SequenceSet lookups to be no more than 3x slower (%v vs %v)", mapLookupElapsed, ssLookupElapsed)
+	if mapLookupElapsed*4 <= ssLookupElapsed {
+		t.Fatalf("Expected SequenceSet lookups to be no more than 4x slower (%v vs %v)", mapLookupElapsed, ssLookupElapsed)
 	}
 }
 
-- 
2.48.1

