--- a/src/include/liburing.h	2020-04-13 18:50:21.000000000 +0200
+++ b/src/include/liburing.h	2020-04-23 21:43:15.923487287 +0200
@@ -193,8 +193,8 @@
 }
 
 static inline void io_uring_prep_splice(struct io_uring_sqe *sqe,
-					int fd_in, loff_t off_in,
-					int fd_out, loff_t off_out,
+					int fd_in, off_t off_in,
+					int fd_out, off_t off_out,
 					unsigned int nbytes,
 					unsigned int splice_flags)
 {
