diff --git a/setup.py b/setup.py.new
index 2cc9975725..1f2e5dffba 100644
--- a/setup.py
+++ b/setup.py.new
@@ -36,10 +36,10 @@ setup(name=PACKAGE_NAME,
       description=('PRAW, an acronym for `Python Reddit API Wrapper`, is a '
                    'python package that allows for simple access to '
                    'reddit\'s API.'),
-      install_requires=['decorator >=4.0.9, <4.1',
-                        'prawcore >=0.3.0, <0.4',
+      install_requires=['decorator >=4.0.9, <4.4',
+                        'prawcore >=0.3.0',
                         'requests >=2.3.0',
-                        'six ==1.10',
+                        'six >=1.10',
                         'update_checker >=0.12'],
       keywords='reddit api wrapper',
       license='Simplified BSD License',
