diff --git a/completion/bash/task.bash b/completion/bash/task.bash
index 4ad8837..14896b3 100644
--- a/completion/bash/task.bash
+++ b/completion/bash/task.bash
@@ -6,7 +6,7 @@ _task_completion()
   # Remove colon from word breaks
   COMP_WORDBREAKS=${COMP_WORDBREAKS//:}
 
-  scripts=$(task -l | sed '1d' | awk '{ print $2 }' | sed 's/:$//');
+  scripts=$(go-task -l | sed '1d' | awk '{ print $2 }' | sed 's/:$//');
 
   curr_arg="${COMP_WORDS[COMP_CWORD]:-"."}"
 
@@ -18,4 +18,4 @@ _task_completion()
   COMPREPLY=($(compgen -c | echo "$scripts" | grep $curr_arg));
 }
 
-complete -F _task_completion task
+complete -F _task_completion go-task
diff --git a/completion/fish/task.fish b/completion/fish/task.fish
index 147d498..6e215f8 100644
--- a/completion/fish/task.fish
+++ b/completion/fish/task.fish
@@ -1,24 +1,24 @@
 function __task_get_tasks --description "Prints all available tasks with their description"
-	task -l | sed '1d' | awk '{ $1=""; print $0 }' | tr ': ', '\t' | string trim
+	go-task -l | sed '1d' | awk '{ $1=""; print $0 }' | tr ': ', '\t' | string trim
 end
 
-complete -c task -d 'Runs the specified task(s). Falls back to the "default" task if no task name was specified, or lists all tasks if an unknown task name was 
+complete -c go-task -d 'Runs the specified task(s). Falls back to the "default" task if no task name was specified, or lists all tasks if an unknown task name was 
 specified.' -xa "(__task_get_tasks)"
 
 
-complete -c task -s c -l color     -d 'colored output (default true)'
-complete -c task -s d -l dir       -d 'sets directory of execution'
-complete -c task      -l dry       -d 'compiles and prints tasks in the order that they would be run, without executing them'
-complete -c task -s f -l force     -d 'forces execution even when the task is up-to-date'
-complete -c task -s h -l help      -d 'shows Task usage'
-complete -c task -s i -l init      -d 'creates a new Taskfile.yml in the current folder'
-complete -c task -s l -l list      -d 'lists tasks with description of current Taskfile'
-complete -c task -s o -l output    -d 'sets output style: [interleaved|group|prefixed]' -xa "interleaved group prefixed"
-complete -c task -s p -l parallel  -d 'executes tasks provided on command line in parallel'
-complete -c task -s s -l silent    -d 'disables echoing'
-complete -c task      -l status    -d 'exits with non-zero exit code if any of the given tasks is not up-to-date'
-complete -c task      -l summary   -d 'show summary about a task'
-complete -c task -s t -l taskfile  -d 'choose which Taskfile to run. Defaults to "Taskfile.yml"'
-complete -c task -s v -l verbose   -d 'enables verbose mode'
-complete -c task      -l version   -d 'show Task version'
-complete -c task -s w -l watch     -d 'enables watch of the given task'
+complete -c go-task -s c -l color     -d 'colored output (default true)'
+complete -c go-task -s d -l dir       -d 'sets directory of execution'
+complete -c go-task      -l dry       -d 'compiles and prints tasks in the order that they would be run, without executing them'
+complete -c go-task -s f -l force     -d 'forces execution even when the task is up-to-date'
+complete -c go-task -s h -l help      -d 'shows Task usage'
+complete -c go-task -s i -l init      -d 'creates a new Taskfile.yml in the current folder'
+complete -c go-task -s l -l list      -d 'lists tasks with description of current Taskfile'
+complete -c go-task -s o -l output    -d 'sets output style: [interleaved|group|prefixed]' -xa "interleaved group prefixed"
+complete -c go-task -s p -l parallel  -d 'executes tasks provided on command line in parallel'
+complete -c go-task -s s -l silent    -d 'disables echoing'
+complete -c go-task      -l status    -d 'exits with non-zero exit code if any of the given tasks is not up-to-date'
+complete -c go-task      -l summary   -d 'show summary about a task'
+complete -c go-task -s t -l taskfile  -d 'choose which Taskfile to run. Defaults to "Taskfile.yml"'
+complete -c go-task -s v -l verbose   -d 'enables verbose mode'
+complete -c go-task      -l version   -d 'show Task version'
+complete -c go-task -s w -l watch     -d 'enables watch of the given task'
diff --git a/completion/zsh/_task b/completion/zsh/_task
index 48777e4..c2458d2 100755
--- a/completion/zsh/_task
+++ b/completion/zsh/_task
@@ -1,11 +1,11 @@
-#compdef task
+#compdef go-task
 
 # Listing commands from Taskfile.yml
 function __list() {
     local -a scripts
 
     if [ -f Taskfile.yml ]; then
-        scripts=($(task -l | sed '1d' | sed 's/^\* //' | awk '{ print $1 }' | sed 's/:$//' | sed 's/:/\\:/g'))
+        scripts=($(go-task -l | sed '1d' | sed 's/^\* //' | awk '{ print $1 }' | sed 's/:$//' | sed 's/:/\\:/g'))
         _describe 'script' scripts
     fi
 }
