--- a/run_tests.py.orig
+++ b/run_tests.py
@@ -7,7 +7,7 @@
 
 DIR_OF_THIS_SCRIPT = p.dirname( p.abspath( __file__ ) )
 DIR_OF_THIRD_PARTY = p.join( DIR_OF_THIS_SCRIPT, 'third_party' )
-DIR_OF_YCMD_THIRD_PARTY = p.join( DIR_OF_THIRD_PARTY, 'ycmd', 'third_party' )
+DIR_OF_YCMD_THIRD_PARTY = p.join( os.sep, 'usr', 'lib', 'python3.6', 'site-packages', 'ycmd', 'third_party' )
 
 python_path = []
 for folder in os.listdir( DIR_OF_THIRD_PARTY ):
