diff --git a/completion/bash/task.bash b/completion/bash/task.bash
index de93e4c..2151df9 100644
--- a/completion/bash/task.bash
+++ b/completion/bash/task.bash
@@ -52,4 +52,4 @@ function _task()
   __ltrim_colon_completions "$cur"
 }
 
-complete -F _task task
+complete -F _task go-task
diff --git a/completion/fish/task.fish b/completion/fish/task.fish
index ee2d0a5..303de4a 100644
--- a/completion/fish/task.fish
+++ b/completion/fish/task.fish
@@ -1,4 +1,4 @@
-set GO_TASK_PROGNAME task
+set GO_TASK_PROGNAME go-task
 
 function __task_get_tasks --description "Prints all available tasks with their description"
   # Read the list of tasks (and potential errors)
diff --git a/completion/zsh/_task b/completion/zsh/_task
index 56f2736..442a581 100755
--- a/completion/zsh/_task
+++ b/completion/zsh/_task
@@ -11,7 +11,7 @@ function __task_list() {
     local -i enabled=0
     local taskfile item task desc
 
-    cmd=(task)
+    cmd=(go-task)
     taskfile=${(Qv)opt_args[(i)-t|--taskfile]}
     taskfile=${taskfile//\~/$HOME}
 
