From 3c00fe8ce7eb6e7e0fbe79457e0f424da0dd01dd Mon Sep 17 00:00:00 2001
From: Aster Boese <asterboese@mailbox.org>
Date: Thu, 26 Oct 2023 02:45:23 -0400
Subject: [PATCH] remove docs/html from makefile

---
 CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0f0d04f..7e8b0e8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -126,5 +126,3 @@ add_doxygen(
         *::Priv
     ALL
 )
-install(DIRECTORY ${CMAKE_BINARY_DIR}/doc/html
-        DESTINATION share/doc/${LIBNAME})
-- 
2.42.0

