diff -u gcc/config/i386/linux.h gcc/config/i386/linux.h
--- a/gcc/config/i386/linux.h	2 Jul 2005 08:52:20 -0000	1.60
+++ b/gcc/config/i386/linux.h	2 Jul 2005 08:52:20 -0000	1.60
@@ -186,7 +186,7 @@
 /* This macro may be overridden in i386/k*bsd-gnu.h.  */
 #define REG_NAME(reg) reg
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* i386 glibc provides __stack_chk_guard in %gs:0x14.  */
 #define TARGET_THREAD_SSP_OFFSET	0x14
 #endif
diff -u gcc/config/i386/linux64.h gcc/config/i386/linux64.h
--- a/gcc/config/i386/linux64.h	2 Jul 2005 08:52:20 -0000	1.33
+++ b/gcc/config/i386/linux64.h	2 Jul 2005 08:52:20 -0000	1.33
@@ -74,7 +74,7 @@
 /* This macro may be overridden in i386/k*bsd-gnu.h.  */
 #define REG_NAME(reg) reg
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* i386 glibc provides __stack_chk_guard in %gs:0x14,
    x86_64 glibc provides it in %fs:0x28.  */
 #define TARGET_THREAD_SSP_OFFSET	(TARGET_64BIT ? 0x28 : 0x14)
diff -u gcc/config/rs6000/linux.h gcc/config/rs6000/linux.h
--- a/gcc/config/rs6000/linux.h	2 Jul 2005 08:52:11 -0000	1.53
+++ b/gcc/config/rs6000/linux.h	2 Jul 2005 08:52:11 -0000	1.53
@@ -114,7 +114,7 @@
 
 #define MD_UNWIND_SUPPORT "config/rs6000/linux-unwind.h"
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* ppc32 glibc provides __stack_chk_guard in -0x7008(2).  */
 #define TARGET_THREAD_SSP_OFFSET	-0x7008
 #endif
diff -u gcc/config/rs6000/linux64.h gcc/config/rs6000/linux64.h
--- a/gcc/config/rs6000/linux64.h	2 Jul 2005 08:52:15 -0000	1.81
+++ b/gcc/config/rs6000/linux64.h	2 Jul 2005 08:52:15 -0000	1.81
@@ -548,7 +548,7 @@
 
 #define MD_UNWIND_SUPPORT "config/rs6000/linux-unwind.h"
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* ppc32 glibc provides __stack_chk_guard in -0x7008(2),
    ppc64 glibc provides it at -0x7010(13).  */
 #define TARGET_THREAD_SSP_OFFSET	(TARGET_64BIT ? -0x7010 : -0x7008)
diff -u gcc/config/s390/linux.h gcc/config/s390/linux.h
--- a/gcc/config/s390/linux.h	2005-08-22 15:53:01.000000000 +0200	1.40
+++ b/gcc/config/s390/linux.h	2005-08-22 15:53:01.000000000 +0200	1.40
@@ -94,7 +94,7 @@
 
 #define MD_UNWIND_SUPPORT "config/s390/linux-unwind.h"
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* s390 glibc provides __stack_chk_guard in 0x14(tp),
    s390x glibc provides it at 0x28(tp).  */
 #define TARGET_THREAD_SSP_OFFSET        (TARGET_64BIT ? 0x28 : 0x14)
diff -u gcc/config/sparc/linux.h gcc/config/sparc/linux.h
--- a/gcc/config/sparc/linux.h	6 Jul 2005 07:48:57 -0000	1.72
+++ b/gcc/config/sparc/linux.h	6 Jul 2005 07:48:57 -0000	1.72
@@ -230,7 +230,7 @@
 #undef NEED_INDICATE_EXEC_STACK
 #define NEED_INDICATE_EXEC_STACK 1
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* sparc glibc provides __stack_chk_guard in [%g7 + 0x14].  */
 #define TARGET_THREAD_SSP_OFFSET	0x14
 #endif
diff -u gcc/config/sparc/linux64.h gcc/config/sparc/linux64.h
--- a/gcc/config/sparc/linux64.h	6 Jul 2005 07:48:57 -0000	1.95
+++ b/gcc/config/sparc/linux64.h	6 Jul 2005 07:48:57 -0000	1.95
@@ -364,7 +364,7 @@
 #undef NEED_INDICATE_EXEC_STACK
 #define NEED_INDICATE_EXEC_STACK 1
 
-#ifdef TARGET_LIBC_PROVIDES_SSP
+#if defined TARGET_LIBC_PROVIDES_SSP && !defined __UCLIBC__
 /* sparc glibc provides __stack_chk_guard in [%g7 + 0x14],
    sparc64 glibc provides it at [%g7 + 0x28].  */
 #define TARGET_THREAD_SSP_OFFSET	(TARGET_ARCH64 ? 0x28 : 0x14)
