diff -ruN a/tests/test-tty.lua b/tests/test-tty.lua
--- a/tests/test-tty.lua	2025-05-17 03:46:28.000000000 +0000
+++ b/tests/test-tty.lua	2025-05-17 20:36:03.053353909 +0000
@@ -13,6 +13,11 @@
 return require('lib/tap')(function (test)
 
   test("tty normal", function (print, p, expect, uv)
+    if true then
+      print("Skip, no tty on the builders")
+      return
+    end
+
     local stdin = uv.new_tty(0, true)
     local stdout = uv.new_tty(1, false)
 
