From 19538be6d68f463ef3711054da4c68b32325e63b Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Wed, 22 Apr 2015 14:30:41 +0200
Subject: [PATCH] use webapps dir structure

---
 lib/apache2/patchwork.wsgi.conf | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/lib/apache2/patchwork.wsgi.conf b/lib/apache2/patchwork.wsgi.conf
index 3756e5a..0be4174 100644
--- a/lib/apache2/patchwork.wsgi.conf
+++ b/lib/apache2/patchwork.wsgi.conf
@@ -1,20 +1,20 @@
 <IfModule mod_alias.c>
-        Alias /images/ "/srv/patchwork/htdocs/images/"
-        Alias /css/ "/srv/patchwork/htdocs/css/"
-        Alias /js/ "/srv/patchwork/htdocs/js/"
-        Alias /robots.txt "/srv/patchwork/htdocs/robots.txt"
-        <Directory "/srv/patchwork/htdocs">
+        Alias /images/ "/usr/share/webapps/patchwork/htdocs/images/"
+        Alias /css/ "/usr/share/webapps/patchwork/htdocs/css/"
+        Alias /js/ "/usr/share/webapps/patchwork/htdocs/js/"
+        Alias /robots.txt "/usr/share/webapps/patchwork/htdocs/robots.txt"
+        <Directory "/usr/share/webapps/patchwork/htdocs">
                 Order allow,deny
                 Allow from all
         </Directory>
 
-        Alias /media/ "/usr/share/python-support/python-django/django/contrib/admin/media/"
+        Alias /static/admin/ "/usr/lib/python2.7/site-packages/django/contrib/admin/static/admin/"
 
-        <Directory "/usr/share/python-support/python-django/django/contrib/admin/media/">
+        <Directory "/usr/lib/python2.7/site-packages/django/contrib/admin/static/admin/">
                 Order allow,deny
                 Allow from all
         </Directory>
 </IfModule>
 
-WSGIScriptAlias / "/srv/patchwork/lib/apache2/patchwork.wsgi"
+WSGIScriptAlias / "/usr/share/webapps/patchwork/lib/apache2/patchwork.wsgi"
 WSGIPassAuthorization On
-- 
2.3.5

