From 61f2c01a0d93b97e1c9eaad85b7e5137fc4f88cf Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Fri, 22 Mar 2024 10:52:13 +0100
Subject: [PATCH] fftw use pkgconfig

---
 cmake/modules/FindFFTW3.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmake/modules/FindFFTW3.cmake b/cmake/modules/FindFFTW3.cmake
index afb20c4..61c6c48 100644
--- a/cmake/modules/FindFFTW3.cmake
+++ b/cmake/modules/FindFFTW3.cmake
@@ -74,9 +74,9 @@ foreach(_dir ${CMAKE_PREFIX_PATH})
 endforeach()
 mark_as_advanced(PKG_FFTW_CONFIG_DIR)
 
-find_package(FFTW3 QUIET NO_MODULE
-    HINTS ${PKG_FFTW_CONFIG_DIR} /usr/lib/cmake/fftw3 /usr/local/lib/cmake/fftw3
-)
+# find_package(FFTW3 QUIET NO_MODULE
+#     HINTS ${PKG_FFTW_CONFIG_DIR} /usr/lib/cmake/fftw3 /usr/local/lib/cmake/fftw3
+# )
 mark_as_advanced(FFTW3_DIR)
 
 # if we found the FFTW3 CMake package then we can start
-- 
2.44.0

