diff -urN krita-4.2.9.orig/libs/ui/opengl/kis_opengl_canvas2.cpp krita-4.2.9/libs/ui/opengl/kis_opengl_canvas2.cpp
--- krita-4.2.9.orig/libs/ui/opengl/kis_opengl_canvas2.cpp	2020-05-22 01:52:07.430282697 -0600
+++ krita-4.2.9/libs/ui/opengl/kis_opengl_canvas2.cpp	2020-05-22 01:53:24.310617820 -0600
@@ -46,10 +46,6 @@
 #include "KisOpenGLModeProber.h"
 #include <KoColorModelStandardIds.h>
 
-#ifndef Q_OS_MACOS
-#include <QOpenGLFunctions_2_1>
-#endif
-
 #define NEAR_VAL -1000.0
 #define FAR_VAL 1000.0
 
@@ -104,10 +100,6 @@
     QVector3D vertices[6];
     QVector2D texCoords[6];
 
-#ifndef Q_OS_MACOS
-    QOpenGLFunctions_2_1 *glFn201;
-#endif
-
     qreal pixelGridDrawingThreshold;
     bool pixelGridEnabled;
     QColor gridColor;
@@ -271,16 +263,6 @@
 {
     KisOpenGL::initializeContext(context());
     initializeOpenGLFunctions();
-#ifndef Q_OS_MACOS
-    if (!KisOpenGL::hasOpenGLES()) {
-        d->glFn201 = context()->versionFunctions<QOpenGLFunctions_2_1>();
-        if (!d->glFn201) {
-            warnUI << "Cannot obtain QOpenGLFunctions_2_1, glLogicOp cannot be used";
-        }
-    } else {
-        d->glFn201 = nullptr;
-    }
-#endif
 
     KisConfig cfg(true);
     d->openGLImageTextures->setProofingConfig(canvas()->proofingConfiguration());
@@ -446,13 +428,7 @@
         glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
 
         glEnable(GL_COLOR_LOGIC_OP);
-#ifndef Q_OS_MACOS
-        if (d->glFn201) {
-            d->glFn201->glLogicOp(GL_XOR);
-        }
-#else
         glLogicOp(GL_XOR);
-#endif
     } else {
         glEnable(GL_BLEND);
         glBlendFuncSeparate(GL_ONE_MINUS_DST_COLOR, GL_ZERO, GL_ONE, GL_ONE);
