diff -ur pyenchant2-2.0.0/enchant/_enchant.py pyenchant2-2.0.0_fix/enchant/_enchant.py
--- pyenchant2-2.0.0/enchant/_enchant.py 2018-01-24 21:13:42.680056632 -0700
+++ pyenchant2-2.0.0_fix/enchant/_enchant.py 2018-01-24 21:14:20.620017925 -0700
@@ -72,6 +72,7 @@
         yield "libenchant.so"
     # See if ctypes can find the library for us, under various names.
     yield find_library("enchant")
+    yield find_library("enchant-2")
     yield find_library("libenchant")
     yield find_library("libenchant-1")
     # Special-case handling for enchant installed by macports.
