diff --git a/mozglue/interposers/pthread_atfork_interposer.cpp b/mozglue/interposers/pthread_atfork_interposer.cpp
--- a/mozglue/interposers/pthread_atfork_interposer.cpp
+++ b/mozglue/interposers/pthread_atfork_interposer.cpp
@@ -127,10 +127,13 @@
 
   rv = mozPthreadHandlers.add(aPrefork, aPostForkParent, aPostForkChild
 #  if defined(LIBC_GLIBC)
                               ,
                               dso_handle
+#  else
+                              ,
+                              (void*)(1)
 #  endif
                               )
            ? 0
            : 1;
   MOZ_ASSERT(rv == 0,

