--- a/browser/installer/linux/app/flatpak/files/share/appdata/org.mozilla.firefox.appdata.xml.in
+++ b/browser/installer/linux/app/flatpak/files/share/appdata/org.mozilla.firefox.appdata.xml.in
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop-application">
-  <id>org.mozilla.firefox</id>
-  <launchable type="desktop-id">org.mozilla.firefox.desktop</launchable>
-  <name>Firefox</name>
+  <id>org.mozilla.firefox-developer-edition</id>
+  <launchable type="desktop-id">org.mozilla.firefox-developer-edition.desktop</launchable>
+  <name>Firefox Developer Edition</name>
   <developer_name>Mozilla</developer_name>
   <summary>Fast, Private &amp; Safe Web Browser</summary>
   <metadata_license>CC0-1.0</metadata_license>
