diff -ur xfce4-wavelan-plugin-0.5.6.orig/configure.ac xfce4-wavelan-plugin-0.5.6/configure.ac
--- xfce4-wavelan-plugin-0.5.6.orig/configure.ac	2010-12-04 01:51:47.000000000 +0100
+++ xfce4-wavelan-plugin-0.5.6/configure.ac	2011-09-21 08:55:53.549999963 +0200
@@ -41,6 +41,7 @@
                   sys/socket.h sys/ioctl.h linux/types.h linux/if.h \
                   linux/wireless.h string.h \
                   ctype.h], [], [], [#include <sys/socket.h>])
+AC_SEARCH_LIBS(sin, m)
 
 dnl Check for i18n support
 XDT_I18N([ar ast be ca cs da de en_GB es eu fr gl hu id it ja kk ko lv nb pl pt_BR pt ru sq sv tr ug uk ur_PK ur vi zh_CN ])
