From bf5ed98a037d40bcec0e4fdc4550f7e41eb7e828 Mon Sep 17 00:00:00 2001
From: Axel Beckert <abe@deuxchevaux.org>
Date: Tue, 31 Jul 2012 13:31:32 +0200
Subject: [PATCH] Honor common compiler and linker flags

---
 CHANGELOG |    2 ++
 Makefile  |    2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 3472806..b4ef123 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,5 @@
+- Honor common compiler and linker flags
+
 Version 0.4.6:
 - Added Feature for not creating a header e.g. for already existing HTML-files (thanks to Nicolas Zagulajew)
 
diff --git a/Makefile b/Makefile
index b880ba0..f113108 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
 all: aha
 
 aha: aha.c
-	gcc -std=c99 aha.c -o aha
+	gcc -std=c99 $(CFLAGS) $(LDFLAGS) $(CPPFLAGS) aha.c -o aha
-- 
1.7.10.4

