diff --git a/Makefile b/Makefile
index e5de7d3..9c86fea 100644
--- a/Makefile
+++ b/Makefile
@@ -32,7 +32,7 @@ TOOFAR_OK=
 #CFLAGS = -O2
 #CFLAGS = -O2 -fomit-frame-pointer -Wall
 #CFLAGS = -Os -fomit-frame-pointer -Wall
-CFLAGS = -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
+CFLAGS ?= -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
 # [note that -Wall is a gcc-specific compilation flag ("all warnings on")]
 
 LDFLAGS =
@@ -41,16 +41,16 @@ E =
 
 PNGCRUSH  = pngcrush
 
-LIBS = -lm
+LIBS = -lm -lz
 # uncomment one of these 2 lines only if you are using an external zlib:
 #LIBS = -L${ZLIB} -lz -lm
 #LIBS = ${ZLIB}/libz.a -lm
 
 # uncomment these 4 lines only if you are NOT using an external copy of zlib:
-ZHDR = zlib.h
-ZOBJS  = adler32$(O) compress$(O) crc32$(O) deflate$(O) \
-	 infback$(O) inffast$(O) inflate$(O) inftrees$(O) \
-	 trees$(O) uncompr$(O) zutil$(O)
+#ZHDR = zlib.h
+#ZOBJS  = adler32$(O) compress$(O) crc32$(O) deflate$(O) \
+#	 infback$(O) inffast$(O) inflate$(O) inftrees$(O) \
+#	 trees$(O) uncompr$(O) zutil$(O)
 
 OBJS  = pngcrush$(O) png$(O) pngerror$(O) pngget$(O) pngmem$(O) \
 	pngpread$(O) pngread$(O) pngrio$(O) pngrtran$(O) pngrutil$(O) \
