--- pngcrush-1.7.25/Makefile
+++ pngcrush-1.7.25/Makefile
@@ -20,8 +20,8 @@
 #CFLAGS = -I. -O -Wall
 #CFLAGS = -I. -O2
 #CFLAGS = -I. -O3 -fomit-frame-pointer -Wall
-#CFLAGS = -I. -Os -fomit-frame-pointer -Wall
-CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
+CFLAGS = -I. -Os -fomit-frame-pointer -Wall
+#CFLAGS = -I. -O3 -funroll-loops -fomit-frame-pointer -Wall -Wshadow
 #CFLAGS = -I${ZINC} -I. -O3 -fomit-frame-pointer -Wall
 # [note that -Wall is a gcc-specific compilation flag ("all warnings on")]
 
