Author: Milan P. Stanić <mps@arvanta.net>
Date: 2026-08-08 18:56:09 +0200

musl libc doesn't have dlvsym

--- a/lgi_closure_guard.c	2026-08-08 18:56:09.000000000 +0000
+++ b/lgi_closure_guard.c	2026-08-10 10:44:30.722505598 +0000
@@ -243,7 +243,7 @@
 static void *
 guard_resolve(const char *sym)
 {
-	void *fn = dlvsym(RTLD_NEXT, sym, "LIBFFI_CLOSURE_8.0");
+	void *fn = dlsym(RTLD_NEXT, sym);
 
 	if (!fn)
 		fn = dlsym(RTLD_NEXT, sym);
