Author: Milan P. Stanić <mps@arvanta.net>
Date: 2026-04-17 22:03:46 +0200

musl does noy have dlvsym so removed it

--- a/lgi_closure_guard.c	2026-04-09 20:17:43.000000000 +0200
+++ b/lgi_closure_guard.c	2026-04-16 21:38:03.763942589 +0200
@@ -190,10 +190,6 @@
 	void (*fun)(ffi_cif *, void *, void **, void *),
 	void *user_data, void *codeloc)
 {
-	if (!real_ffi_prep) {
-		real_ffi_prep = dlvsym(RTLD_NEXT, "ffi_prep_closure_loc",
-			"LIBFFI_CLOSURE_8.0");
-	}
 	if (!real_ffi_prep)
 		real_ffi_prep = dlsym(RTLD_NEXT, "ffi_prep_closure_loc");
 
