From 7f851f5fb8fbcd7b3a93aaedff90b27124e17a7e Mon Sep 17 00:00:00 2001
From: Bruno Inec <contact@sweenu.xyz>
Date: Wed, 20 Mar 2024 10:36:56 +0100
Subject: [PATCH] Compatibility with Linux 6.8

---
 ddcci/ddcci.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ddcci/ddcci.c b/ddcci/ddcci.c
index 87b0818..6073c75 100644
--- a/ddcci/ddcci.c
+++ b/ddcci/ddcci.c
@@ -1666,7 +1666,7 @@ static int ddcci_detect(struct i2c_client *client, struct i2c_board_info *info)
 	pr_debug("detected %d:%02x\n", client->adapter->nr, outer_addr);
 
 	/* set device type */
-	strlcpy(info->type, (outer_addr == DDCCI_DEFAULT_DEVICE_ADDR) ? "ddcci" : "ddcci-dependent", I2C_NAME_SIZE);
+	strscpy(info->type, (outer_addr == DDCCI_DEFAULT_DEVICE_ADDR) ? "ddcci" : "ddcci-dependent", I2C_NAME_SIZE);
 
 	return 0;
 }
@@ -1824,7 +1824,7 @@ static struct i2c_driver ddcci_driver = {
 	#else
 	.remove		= ddcci_remove,
 	#endif
-	.class		= I2C_CLASS_DDC,
+	.class		= I2C_CLASS_SPD,
 	.detect		= ddcci_detect,
 	.address_list	= I2C_ADDRS(
 		DDCCI_DEFAULT_DEVICE_ADDR>>1
-- 
GitLab

