diff --git a/util/Makefile.LINUXdso b/util/Makefile.LINUXdso
index c39a146..d3fd0b7 100644
--- a/util/Makefile.LINUXdso
+++ b/util/Makefile.LINUXdso
@@ -30,7 +30,7 @@
 # DSO-specific rules for the util directory.
 #
 
-libfaxutil.${DSO}: ${OBJECTS}
+${LIBUTIL} libfaxutil.${DSO}: ${OBJECTS}
 	${C++} ${DSOOPTS} -o libfaxutil.${DSO} ${OBJECTS} \
 	    ${DSODELAY} ${LIBTIFF} ${DSODELAY} ${LIBZ} ${DSODELAY} ${LIBREGEX}
 
