diff --git a/include/modules/clock.hpp b/include/modules/clock.hpp
index b812b70..4a0c1e5 100644
--- a/include/modules/clock.hpp
+++ b/include/modules/clock.hpp
@@ -1,8 +1,8 @@
 #pragma once
 
+#include <fmt/chrono.h>
 #include <fmt/format.h>
 #include "ALabel.hpp"
-#include "fmt/time.h"
 #include "util/sleeper_thread.hpp"
 
 namespace waybar::modules {
diff --git a/src/bar.cpp b/src/bar.cpp
index 8a45869..8e64349 100644
--- a/src/bar.cpp
+++ b/src/bar.cpp
@@ -1,7 +1,7 @@
 #include "bar.hpp"
-#include <spdlog/spdlog.h>
 #include "client.hpp"
 #include "factory.hpp"
+#include <spdlog/spdlog.h>
 
 waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
     : output(w_output),
diff --git a/src/modules/sni/host.cpp b/src/modules/sni/host.cpp
index 015f756..c815c65 100644
--- a/src/modules/sni/host.cpp
+++ b/src/modules/sni/host.cpp
@@ -1,6 +1,6 @@
 #include "modules/sni/host.hpp"
-#include <spdlog/spdlog.h>
 #include <fmt/ostream.h>
+#include <spdlog/spdlog.h>
 
 namespace waybar::modules::SNI {
 

