--- a/src/develop/stack.c
+++ b/src/develop/stack.c
@@ -18,7 +18,11 @@
 #include "main/php_ini.h"
 
 #include "ext/standard/html.h"
+#if PHP_VERSION_ID < 70200
 #include "ext/standard/php_smart_string.h"
+#else
+#include "Zend/zend_smart_string.h"
+#endif
 #include "zend_exceptions.h"
 #include "zend_generators.h"
 
--- a/src/lib/var.c
+++ b/src/lib/var.c
@@ -20,7 +20,11 @@
 #include "zend.h"
 #include "zend_exceptions.h"
 #include "zend_extensions.h"
+#if PHP_VERSION_ID < 70200
 #include "ext/standard/php_smart_string.h"
+#else
+#include "Zend/zend_smart_string.h"
+#endif
 #include "zend_smart_str.h"
 #include "zend_closures.h"
 
