From 51aaf75afb3e75cf386502a9ee70da31096f0ef0 Mon Sep 17 00:00:00 2001
From: Ludovic Fernandez <ldez@users.noreply.github.com>
Date: Sat, 15 Mar 2025 13:08:01 +0100
Subject: [PATCH] tests: change ns (#2482)

---
 challenge/dns01/nameserver_test.go | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/challenge/dns01/nameserver_test.go b/challenge/dns01/nameserver_test.go
index 15b19beba0..0e66c2fd17 100644
--- a/challenge/dns01/nameserver_test.go
+++ b/challenge/dns01/nameserver_test.go
@@ -24,8 +24,8 @@ func TestLookupNameserversOK(t *testing.T) {
 			nss:  []string{"ns1.google.com.", "ns2.google.com.", "ns3.google.com.", "ns4.google.com."},
 		},
 		{
-			fqdn: "physics.georgetown.edu.",
-			nss:  []string{"ns4.georgetown.edu.", "ns5.georgetown.edu.", "ns6.georgetown.edu."},
+			fqdn: "mail.proton.me.",
+			nss:  []string{"ns1.proton.me.", "ns2.proton.me.", "ns3.proton.me."},
 		},
 	}
 
