diff -ur mstflint-4.25.0/mtcr_ul/mtcr_ul_com.c mstflint-4.25.0/mtcr_ul/mtcr_ul_com.c
--- mstflint-4.35.0/mtcr_ul/mtcr_ul_com.c       2022-04-28 11:50:04.000000000 +0000
+++ mstflint-4.35.0/mtcr_ul/mtcr_ul_com.c       2022-05-19 14:06:49.479797274 +0000
@@ -83,7 +83,13 @@
 #include <endian.h>
 #include <byteswap.h>
 #include <errno.h>
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
 #include <sys/types.h>
+#undef _GNU_SOURCE
+#else
+#include <sys/types.h>
+#endif
 #include <sys/stat.h>
 #include <fcntl.h>
 #include <string.h>
diff -ur mstflint-4.35.0/include/mtcr_ul/fwctrl_ioctl.h mstflint-4.35.0/include/mtcr_ul/fwctrl_ioctl.h
--- mstflint-4.35.0/include/mtcr_ul/fwctrl_ioctl.h
+++ mstflint-4.35.0/include/mtcr_ul/fwctrl_ioctl.h
@@ -153,9 +153,9 @@
 struct mlx5_umem_buff {
 	void *buff;
 	size_t size;
-	__uint32_t umem_id;
-	__uint32_t umem_mkey;
-	__uint32_t rsc_id;
+	u_int32_t umem_id;
+	u_int32_t umem_mkey;
+	u_int32_t rsc_id;
 };
 
 struct fwctl_rsc_umem_reg {
