--- Makefile.in
+++ src/nut-2.7.1/drivers/Makefile.in
@@ -135,7 +135,7 @@
 am_libnuthalmain_a_OBJECTS = main-hal.$(OBJEXT) dstate-hal.$(OBJEXT) \
 	usb-common.$(OBJEXT)
 libnuthalmain_a_OBJECTS = $(am_libnuthalmain_a_OBJECTS)
-am__EXEEXT_1 = al175$(EXEEXT) bcmxcp$(EXEEXT) belkin$(EXEEXT) \
+am__EXEEXT_1 = al175$(EXEEXT) belkin$(EXEEXT) \
 	belkinunv$(EXEEXT) bestfcom$(EXEEXT) bestfortress$(EXEEXT) \
 	bestuferrups$(EXEEXT) bestups$(EXEEXT) dummy-ups$(EXEEXT) \
 	etapro$(EXEEXT) everups$(EXEEXT) gamatronic$(EXEEXT) \
@@ -151,7 +151,7 @@
 	apcsmart-old$(EXEEXT) apcupsd-ups$(EXEEXT) riello_ser$(EXEEXT) \
 	nutdrv_qx$(EXEEXT)
 am__EXEEXT_2 = snmp-ups$(EXEEXT)
-am__EXEEXT_3 = usbhid-ups$(EXEEXT) bcmxcp_usb$(EXEEXT) \
+am__EXEEXT_3 = usbhid-ups$(EXEEXT) \
 	tripplite_usb$(EXEEXT) blazer_usb$(EXEEXT) \
 	richcomm_usb$(EXEEXT) riello_usb$(EXEEXT) nutdrv_qx$(EXEEXT)
 am__EXEEXT_4 = $(am__EXEEXT_3)
@@ -168,7 +168,6 @@
 am__installdirs = "$(DESTDIR)$(driverexecdir)" \
 	"$(DESTDIR)$(halexecdir)"
 am__EXEEXT_14 = hald-addon-usbhid-ups$(EXEEXT) \
-	hald-addon-bcmxcp_usb$(EXEEXT) \
 	hald-addon-tripplite_usb$(EXEEXT) \
 	hald-addon-blazer_usb$(EXEEXT) hald-addon-riello_usb$(EXEEXT) \
 	hald-addon-nutdrv_qx$(EXEEXT)
@@ -193,14 +192,7 @@
 apcupsd_ups_LINK = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(apcupsd_ups_CFLAGS) \
 	$(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
-am_bcmxcp_OBJECTS = bcmxcp.$(OBJEXT) bcmxcp_ser.$(OBJEXT)
-bcmxcp_OBJECTS = $(am_bcmxcp_OBJECTS)
 am__DEPENDENCIES_3 = $(am__DEPENDENCIES_2)
-bcmxcp_DEPENDENCIES = $(am__DEPENDENCIES_3)
-am_bcmxcp_usb_OBJECTS = bcmxcp_usb.$(OBJEXT) bcmxcp.$(OBJEXT) \
-	usb-common.$(OBJEXT)
-bcmxcp_usb_OBJECTS = $(am_bcmxcp_usb_OBJECTS)
-bcmxcp_usb_DEPENDENCIES = $(LDADD_DRIVERS) $(am__DEPENDENCIES_1)
 am_belkin_OBJECTS = belkin.$(OBJEXT)
 belkin_OBJECTS = $(am_belkin_OBJECTS)
 belkin_LDADD = $(LDADD)
@@ -264,11 +256,6 @@
 genericups_OBJECTS = $(am_genericups_OBJECTS)
 genericups_LDADD = $(LDADD)
 genericups_DEPENDENCIES = $(am__DEPENDENCIES_2)
-am_hald_addon_bcmxcp_usb_OBJECTS = bcmxcp_usb.$(OBJEXT) \
-	bcmxcp.$(OBJEXT)
-hald_addon_bcmxcp_usb_OBJECTS = $(am_hald_addon_bcmxcp_usb_OBJECTS)
-hald_addon_bcmxcp_usb_DEPENDENCIES = $(LDADD_HAL_DRIVERS) \
-	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 am_hald_addon_blazer_usb_OBJECTS = blazer.$(OBJEXT) \
 	blazer_usb.$(OBJEXT) libusb.$(OBJEXT)
 hald_addon_blazer_usb_OBJECTS = $(am_hald_addon_blazer_usb_OBJECTS)
@@ -486,13 +473,11 @@
 	$(LDFLAGS) -o $@
 SOURCES = $(libdummy_a_SOURCES) $(libnuthalmain_a_SOURCES) \
 	$(al175_SOURCES) $(apcsmart_SOURCES) $(apcsmart_old_SOURCES) \
-	$(apcupsd_ups_SOURCES) $(bcmxcp_SOURCES) $(bcmxcp_usb_SOURCES) \
 	$(belkin_SOURCES) $(belkinunv_SOURCES) $(bestfcom_SOURCES) \
 	bestfortress.c $(bestuferrups_SOURCES) $(bestups_SOURCES) \
 	$(blazer_ser_SOURCES) $(blazer_usb_SOURCES) $(clone_SOURCES) \
 	$(clone_outlet_SOURCES) $(dummy_ups_SOURCES) $(etapro_SOURCES) \
 	$(everups_SOURCES) $(gamatronic_SOURCES) $(genericups_SOURCES) \
-	$(hald_addon_bcmxcp_usb_SOURCES) \
 	$(hald_addon_blazer_usb_SOURCES) \
 	$(hald_addon_nutdrv_qx_SOURCES) \
 	$(hald_addon_riello_usb_SOURCES) \
@@ -514,13 +499,12 @@
 	$(victronups_SOURCES)
 DIST_SOURCES = $(libdummy_a_SOURCES) $(libnuthalmain_a_SOURCES) \
 	$(al175_SOURCES) $(apcsmart_SOURCES) $(apcsmart_old_SOURCES) \
-	$(apcupsd_ups_SOURCES) $(bcmxcp_SOURCES) $(bcmxcp_usb_SOURCES) \
+	$(apcupsd_ups_SOURCES) \
 	$(belkin_SOURCES) $(belkinunv_SOURCES) $(bestfcom_SOURCES) \
 	bestfortress.c $(bestuferrups_SOURCES) $(bestups_SOURCES) \
 	$(blazer_ser_SOURCES) $(blazer_usb_SOURCES) $(clone_SOURCES) \
 	$(clone_outlet_SOURCES) $(dummy_ups_SOURCES) $(etapro_SOURCES) \
 	$(everups_SOURCES) $(gamatronic_SOURCES) $(genericups_SOURCES) \
-	$(hald_addon_bcmxcp_usb_SOURCES) \
 	$(hald_addon_blazer_usb_SOURCES) \
 	$(hald_addon_nutdrv_qx_SOURCES) \
 	$(hald_addon_riello_usb_SOURCES) \
@@ -751,7 +735,7 @@
 AM_CFLAGS = -I$(top_srcdir)/include $(am__append_1) $(am__append_2) \
 	$(am__append_3) $(am__append_4) $(am__append_5) \
 	$(am__append_6)
-SERIAL_DRIVERLIST = al175 bcmxcp belkin belkinunv bestfcom	\
+SERIAL_DRIVERLIST = al175 belkin belkinunv bestfcom	\
  bestfortress bestuferrups bestups dummy-ups etapro everups 	\
  gamatronic genericups isbmex liebert liebert-esp2 masterguard metasys	\
  oldmge-shut mge-utalk microdowell mge-shut oneac optiups powercom rhino 	\
@@ -760,11 +744,11 @@
  nutdrv_qx
 
 SNMP_DRIVERLIST = snmp-ups
-USB_LIBUSB_DRIVERLIST = usbhid-ups bcmxcp_usb tripplite_usb \
+USB_LIBUSB_DRIVERLIST = usbhid-ups tripplite_usb \
  blazer_usb richcomm_usb riello_usb nutdrv_qx
 
 USB_DRIVERLIST = $(USB_LIBUSB_DRIVERLIST)
-HAL_DRIVERLIST = hald-addon-usbhid-ups hald-addon-bcmxcp_usb \
+HAL_DRIVERLIST = hald-addon-usbhid-ups \
  hald-addon-tripplite_usb hald-addon-blazer_usb hald-addon-riello_usb	\
  hald-addon-nutdrv_qx
 
@@ -783,8 +767,6 @@
 al175_SOURCES = al175.c
 apcsmart_SOURCES = apcsmart.c apcsmart_tabs.c
 apcsmart_old_SOURCES = apcsmart-old.c
-bcmxcp_SOURCES = bcmxcp.c bcmxcp_ser.c
-bcmxcp_LDADD = $(LDADD) -lm
 belkin_SOURCES = belkin.c
 belkinunv_SOURCES = belkinunv.c
 bestfcom_SOURCES = bestfcom.c
@@ -858,8 +840,6 @@
 usbhid_ups_LDADD = $(LDADD_DRIVERS) $(LIBUSB_LIBS)
 tripplite_usb_SOURCES = tripplite_usb.c libusb.c usb-common.c
 tripplite_usb_LDADD = $(LDADD_DRIVERS) $(LIBUSB_LIBS) -lm
-bcmxcp_usb_SOURCES = bcmxcp_usb.c bcmxcp.c usb-common.c
-bcmxcp_usb_LDADD = $(LDADD_DRIVERS) $(LIBUSB_LIBS)
 blazer_usb_SOURCES = blazer.c blazer_usb.c libusb.c usb-common.c
 blazer_usb_LDADD = $(LDADD_DRIVERS) $(LIBUSB_LIBS) -lm
 richcomm_usb_SOURCES = richcomm_usb.c usb-common.c
@@ -887,8 +867,6 @@
 hald_addon_usbhid_ups_LDADD = $(LDADD_HAL_DRIVERS) $(LIBUSB_LIBS) $(LIBHAL_LIBS)
 hald_addon_tripplite_usb_SOURCES = tripplite_usb.c libusb.c
 hald_addon_tripplite_usb_LDADD = $(LDADD_HAL_DRIVERS) $(LIBUSB_LIBS) $(LIBHAL_LIBS) -lm
-hald_addon_bcmxcp_usb_SOURCES = bcmxcp_usb.c bcmxcp.c
-hald_addon_bcmxcp_usb_LDADD = $(LDADD_HAL_DRIVERS) $(LIBUSB_LIBS) $(LIBHAL_LIBS)
 hald_addon_blazer_usb_SOURCES = blazer.c blazer_usb.c libusb.c
 hald_addon_blazer_usb_LDADD = $(LDADD_HAL_DRIVERS) $(LIBUSB_LIBS) $(LIBHAL_LIBS) -lm
 hald_addon_riello_usb_SOURCES = riello.c riello_usb.c libusb.c
@@ -929,8 +907,8 @@
 # List of header files. The purpose of this list is not dependency
 # tracking (which is automatic), but to ensure these files are
 # distributed by "make dist".
-dist_noinst_HEADERS = apc-mib.h apc-hid.h baytech-mib.h bcmxcp.h	\
- bcmxcp_io.h belkin.h belkin-hid.h bestpower-mib.h blazer.h cps-hid.h dstate.h \
+dist_noinst_HEADERS = apc-mib.h apc-hid.h baytech-mib.h \
+ belkin.h belkin-hid.h bestpower-mib.h blazer.h cps-hid.h dstate.h \
  dstate-hal.h dummy-ups.h eaton-mib.h explore-hid.h gamatronic.h genericups.h	\
  hidparser.h hidtypes.h ietf-mib.h libhid.h libshut.h libusb.h liebert-hid.h	\
  main.h main-hal.h mge-hid.h mge-mib.h mge-shut.h mge-utalk.h		\
@@ -1102,12 +1080,6 @@
 apcupsd-ups$(EXEEXT): $(apcupsd_ups_OBJECTS) $(apcupsd_ups_DEPENDENCIES) $(EXTRA_apcupsd_ups_DEPENDENCIES) 
 	@rm -f apcupsd-ups$(EXEEXT)
 	$(apcupsd_ups_LINK) $(apcupsd_ups_OBJECTS) $(apcupsd_ups_LDADD) $(LIBS)
-bcmxcp$(EXEEXT): $(bcmxcp_OBJECTS) $(bcmxcp_DEPENDENCIES) $(EXTRA_bcmxcp_DEPENDENCIES) 
-	@rm -f bcmxcp$(EXEEXT)
-	$(LINK) $(bcmxcp_OBJECTS) $(bcmxcp_LDADD) $(LIBS)
-bcmxcp_usb$(EXEEXT): $(bcmxcp_usb_OBJECTS) $(bcmxcp_usb_DEPENDENCIES) $(EXTRA_bcmxcp_usb_DEPENDENCIES) 
-	@rm -f bcmxcp_usb$(EXEEXT)
-	$(LINK) $(bcmxcp_usb_OBJECTS) $(bcmxcp_usb_LDADD) $(LIBS)
 belkin$(EXEEXT): $(belkin_OBJECTS) $(belkin_DEPENDENCIES) $(EXTRA_belkin_DEPENDENCIES) 
 	@rm -f belkin$(EXEEXT)
 	$(LINK) $(belkin_OBJECTS) $(belkin_LDADD) $(LIBS)
@@ -1153,9 +1125,6 @@
 genericups$(EXEEXT): $(genericups_OBJECTS) $(genericups_DEPENDENCIES) $(EXTRA_genericups_DEPENDENCIES) 
 	@rm -f genericups$(EXEEXT)
 	$(LINK) $(genericups_OBJECTS) $(genericups_LDADD) $(LIBS)
-hald-addon-bcmxcp_usb$(EXEEXT): $(hald_addon_bcmxcp_usb_OBJECTS) $(hald_addon_bcmxcp_usb_DEPENDENCIES) $(EXTRA_hald_addon_bcmxcp_usb_DEPENDENCIES) 
-	@rm -f hald-addon-bcmxcp_usb$(EXEEXT)
-	$(LINK) $(hald_addon_bcmxcp_usb_OBJECTS) $(hald_addon_bcmxcp_usb_LDADD) $(LIBS)
 hald-addon-blazer_usb$(EXEEXT): $(hald_addon_blazer_usb_OBJECTS) $(hald_addon_blazer_usb_DEPENDENCIES) $(EXTRA_hald_addon_blazer_usb_DEPENDENCIES) 
 	@rm -f hald-addon-blazer_usb$(EXEEXT)
 	$(LINK) $(hald_addon_blazer_usb_OBJECTS) $(hald_addon_blazer_usb_LDADD) $(LIBS)
@@ -1288,9 +1257,6 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/apcsmart_tabs.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/apcupsd_ups-apcupsd-ups.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/baytech-mib.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/bcmxcp.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/bcmxcp_ser.Po@am__quote@
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/bcmxcp_usb.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/belkin-hid.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/belkin.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/belkinunv.Po@am__quote@
