diff --git a/src/no_new_privs.c b/src/no_new_privs.c
index 386fde8..47e1868 100644
--- a/src/no_new_privs.c
+++ b/src/no_new_privs.c
@@ -24,13 +24,15 @@
 #include <string.h>
 #include <errno.h>
 
-#if HAVE_LINUX_PRCTL_H
-#include <linux/prctl.h>  /* Definition of PR_* constants */
-#endif
+
 #if HAVE_SYS_PRCTL_H
 #include <sys/prctl.h>
 #endif
 
+#if HAVE_LINUX_PRCTL_H && !HAVE_SYS_PRCTL_H
+#include <linux/prctl.h>
+#endif
+
 #include "arping.h"
 
 void
