From 9faf6026bae069302e8744284d870388659d80cd Mon Sep 17 00:00:00 2001
From: Marius Gripsgard <marius@ubports.com>
Date: Thu, 12 Jan 2023 01:08:08 +0100
Subject: [PATCH 2/6] Disable broken tests due to dropped mocks from mir 2.9

---
 tests/mocks/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tests/mocks/CMakeLists.txt b/tests/mocks/CMakeLists.txt
index 19d197d58..e0bebee5b 100644
--- a/tests/mocks/CMakeLists.txt
+++ b/tests/mocks/CMakeLists.txt
@@ -41,7 +41,8 @@ add_subdirectory(QMenuModel.1)
 add_subdirectory(QOfono)
 add_subdirectory(SessionBroadcast)
 add_subdirectory(Lomiri)
-add_subdirectory(QtMir/Application)
+# Broken due to dropped mocks from mir 2.9
+#add_subdirectory(QtMir/Application)
 add_subdirectory(QtMultimedia)
 add_subdirectory(WindowManager)
 add_subdirectory(Wizard)
-- 
2.46.0

