From b70c6a3f27ac0a79f3de9a7e729b080e47687a18 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Wed, 2 Apr 2025 19:23:59 +0200
Subject: [PATCH 5/6] Use Mir 2.x

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9728e5666..eb84878e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,7 +106,7 @@ pkg_check_modules(LOMIRISCHEMA REQUIRED lomiri-schemas>=0.1.4)
 option(WITH_MIR2 "Build against Mir 2" OFF)
 
 if (WITH_MIR2)
-    pkg_check_modules(QTMIRSERVER REQUIRED qt5mir2server>=0.6.0)
+    pkg_check_modules(QTMIRSERVER REQUIRED qtmirserver>=0.6.0)
     pkg_check_modules(MIRAL REQUIRED miral>=3.0)
     pkg_check_modules(MIROIL REQUIRED miroil)
 
-- 
2.49.0

