--- chromium-13.0.754.0.orig/third_party/tcmalloc/chromium/src/malloc_hook.cc
+++ chromium-13.0.754.0/third_party/tcmalloc/chromium/src/malloc_hook.cc
@@ -491,16 +491,6 @@
   return result;
 }
 
-// libc's version:
-extern "C" void* __sbrk(ptrdiff_t increment);
-
-extern "C" void* sbrk(ptrdiff_t increment) __THROW {
-  MallocHook::InvokePreSbrkHook(increment);
-  void *result = __sbrk(increment);
-  MallocHook::InvokeSbrkHook(result, increment);
-  return result;
-}
-
 /*static*/void* MallocHook::UnhookedMMap(void *start, size_t length, int prot,
                                          int flags, int fd, off_t offset) {
   return do_mmap64(start, length, prot, flags, fd, offset);
