--- chromium-17.0.942.0.orig/base/process_util_linux.cc
+++ chromium-17.0.942.0/base/process_util_linux.cc
@@ -640,7 +640,7 @@
 
 extern "C" {
 #if !defined(USE_TCMALLOC) && !defined(ADDRESS_SANITIZER) && \
-    !defined(OS_ANDROID)
+    !defined(OS_ANDROID) && !defined(__UCLIBC__)
 
 extern "C" {
 void* __libc_malloc(size_t size);
@@ -700,6 +700,7 @@
       OnNoMemorySize(size); \
     return ret; \
   }
+
 
 DIE_ON_OOM_1(malloc)
 DIE_ON_OOM_1(valloc)
