diff --git a/setup.py b/setup.py
index ae19f5d..636c5d5 100755
--- a/setup.py
+++ b/setup.py
@@ -24,8 +24,7 @@ setup(
     include_package_data=True,
     install_requires=["twisted[tls]>=22.4", "autobahn>=22.4.2", "asgiref>=3.5.2,<4"],
     python_requires=">=3.7",
-    setup_requires=["pytest-runner"],
-    extras_require={"tests": ["hypothesis", "pytest", "pytest-asyncio", "django"]},
+    extras_require={"tests": ["pytest-runner","hypothesis", "pytest", "pytest-asyncio", "django"]},
     entry_points={
         "console_scripts": ["daphne = daphne.cli:CommandLineInterface.entrypoint"]
     },
