From 78992219065dc4762987883c39e75dbb2921e187 Mon Sep 17 00:00:00 2001
From: Henrik Riomar <henrik.riomar@gmail.com>
Date: Thu, 6 Oct 2022 12:11:49 +0200
Subject: [PATCH] fix quoting style in assert

---
 tests/test_console.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/test_console.py b/tests/test_console.py
index 782fb83..62d16ae 100644
--- a/tests/test_console.py
+++ b/tests/test_console.py
@@ -204,7 +204,7 @@ def test_cli(monkeypatch):
 
     runner = CliRunner()
     result = runner.invoke(cli, ['notexists'])
-    assert 'Error: No such command "notexists"' in result.output
+    assert 'Error: No such command \'notexists\'' in result.output
 
 
 def test_choice_default(monkeypatch):
-- 
2.38.0

