diff -wbBur rsyslog-8.9.0.org/runtime/nsd_gtls.c rsyslog-8.9.0/runtime/nsd_gtls.c
--- rsyslog-8.9.0.org/runtime/nsd_gtls.c	2015-03-29 16:00:04.000000000 +0300
+++ rsyslog-8.9.0/runtime/nsd_gtls.c	2015-04-16 15:42:44.701275194 +0300
@@ -231,13 +231,13 @@
 gtlsClientCertCallback(gnutls_session_t session,
               __attribute__((unused)) const gnutls_datum_t* req_ca_rdn, int __attribute__((unused)) nreqs,
               __attribute__((unused)) const gnutls_pk_algorithm_t* sign_algos, int __attribute__((unused)) sign_algos_length,
-              gnutls_retr_st *st)
+              gnutls_retr2_st *st)
 {
 	nsd_gtls_t *pThis;
 
 	pThis = (nsd_gtls_t*) gnutls_session_get_ptr(session);
 
-	st->type = GNUTLS_CRT_X509;
+	st->cert_type = GNUTLS_CRT_X509;
 	st->ncerts = 1;
 	st->cert.x509 = &pThis->ourCert;
 	st->key.x509 = pThis->ourKey;
@@ -1673,14 +1673,14 @@
 	gnutls_session_set_ptr(pThis->sess, (void*)pThis);
 	iRet = gtlsLoadOurCertKey(pThis); /* first load .pem files */
 	if(iRet == RS_RET_OK) {
-		gnutls_certificate_client_set_retrieve_function(xcred, gtlsClientCertCallback);
+		gnutls_certificate_set_retrieve_function(xcred, gtlsClientCertCallback);
 	} else if(iRet != RS_RET_CERTLESS) {
 		FINALIZE; /* we have an error case! */
 	}
 
 	/* Use default priorities */
 	CHKgnutls(gnutls_set_default_priority(pThis->sess));
-	CHKgnutls(gnutls_certificate_type_set_priority(pThis->sess, cert_type_priority));
+	CHKgnutls(gnutls_set_default_priority(pThis->sess));
 
 	/* put the x509 credentials to the current session */
 	CHKgnutls(gnutls_credentials_set(pThis->sess, GNUTLS_CRD_CERTIFICATE, xcred));
