--- rsyslog-6.4.2.orig/tools/omfile.c
+++ rsyslog-6.4.2/tools/omfile.c
@@ -37,6 +37,7 @@
  */
 #include "config.h"
 #include "rsyslog.h"
+#include <fcntl.h>
 #include <stdio.h>
 #include <stdarg.h>
 #include <stdlib.h>
--- rsyslog-6.4.2.orig/tools/pidfile.c
+++ rsyslog-6.4.2/tools/pidfile.c
@@ -29,6 +29,7 @@
  *	First version (v0.2) released
  */
 
+#include <fcntl.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <sys/stat.h>
--- rsyslog-6.4.2.orig/tools/syslogd.c
+++ rsyslog-6.4.2/tools/syslogd.c
@@ -46,13 +46,14 @@
 #define DEFUPRI		(LOG_USER|LOG_NOTICE)
 #define TIMERINTVL	30		/* interval for checking flush, mark */
 
+#define GNU_SOURCE
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <fcntl.h>
 #include <stddef.h>
 #include <ctype.h>
 #include <limits.h>
-#define GNU_SOURCE
 #include <string.h>
 #include <stdarg.h>
 #include <time.h>
