diff --git a/tests/test_filters.py b/tests/test_filters.py
index e697c26..7c7f536 100644
--- a/tests/test_filters.py
+++ b/tests/test_filters.py
@@ -1520,33 +1520,3 @@ class TestAutoprefixer6Filter(TempEnvironmentHelper):
         assert 'webkit' in out
 
 
-class TestBabel(TempEnvironmentHelper):
-    default_files = {
-        'test.es6': """var x = (p) => { return false; };"""
-    }
-
-    def test_es2015(self):
-        es2015 = get_filter('babel', presets='es2015')
-        try:
-            self.mkbundle('test.es6', filters=es2015, output='output.js').build()
-        except FilterError as e:
-            # babel is not installed, that's ok.
-            if 'Program file not found' in e.message:
-                raise SkipTest()
-            else:
-                raise
-        assert "var x = function x" in self.get('output.js')
-
-    def test_extra_args(self):
-        self.env.config['BABEL_EXTRA_ARGS'] = ['--minified']
-        self.mkbundle('test.es6', filters='babel', output='output.js').build()
-        assert (self.get('output.js').strip() ==
-                'var x=p=>{return false};')
-
-    def test_run_in_debug_mode(self):
-        """A setting can be used to make babel not run in debug."""
-        self.env.debug = True
-        self.env.config['babel_run_in_debug'] = False
-        self.mkbundle('test.es6', filters='babel', output='output.js').build()
-        assert self.get('output.js') == self.default_files['test.es6']
-

