--- a/test/run-test.sh
+++ b/test/run-test.sh
@@ -453,7 +453,7 @@
 if [ -n "${SOURCE_DATE_EPOCH:-}" ]; then
     # epoch 0 has special meaning. increase to avoid epoch 0
     old_epoch=${SOURCE_DATE_EPOCH}
-    SOURCE_DATE_EPOCH=$(("$SOURCE_DATE_EPOCH" + 1))
+    SOURCE_DATE_EPOCH=$((SOURCE_DATE_EPOCH + 1))
 fi
 if [ -n "${SOURCE_DATE_EPOCH:-}" ] && [ ${#SOURCE_DATE_EPOCH} -gt 0 ]; then
     touch -m -t "$(fdate ${SOURCE_DATE_EPOCH})" "$FONTDIR"
@@ -486,7 +486,7 @@
 s!@CACHEDIR@!$MYOWNCACHEDIR!" < "$TESTDIR"/fonts.conf.in > "$BUILDTESTDIR"/my-fonts.conf
 
 if [ -n "${SOURCE_DATE_EPOCH:-}" ]; then
-  SOURCE_DATE_EPOCH=$(("$SOURCE_DATE_EPOCH" + 1))
+  SOURCE_DATE_EPOCH=$((SOURCE_DATE_EPOCH + 1))
 fi
 FONTCONFIG_FILE="$BUILDTESTDIR"/my-fonts.conf $FCCACHE -f "$FONTDIR"
 if [ -n "${SOURCE_DATE_EPOCH:-}" ]; then

