--- a/make-linux.mk
+++ b/make-linux.mk
@@ -238,6 +238,10 @@
 	ZT_ARCHITECTURE=6
 	override DEFS+=-DZT_NO_TYPE_PUNNING
 endif
+ifeq ($(CC_MACH),powerpc64le)
+	ZT_ARCHITECTURE=7
+	override DEFS+=-DZT_NO_TYPE_PUNNING
+endif
 ifeq ($(CC_MACH),s390x)
 	ZT_ARCHITECTURE=16
 endif
@@ -278,12 +278,12 @@
 # ARM32 hell -- use conservative CFLAGS
 ifeq ($(ZT_ARCHITECTURE),3)
 	ifeq ($(shell if [ -e /usr/bin/dpkg ]; then dpkg --print-architecture; fi),armel)
-		override CFLAGS+=-march=armv5t -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
-		override CXXFLAGS+=-march=armv5t -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
+		override CFLAGS+=-march=armv6z -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
+		override CXXFLAGS+=-march=armv6z -mfloat-abi=soft -msoft-float -mno-unaligned-access -marm
 		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 	else
-		override CFLAGS+=-march=armv5t -mno-unaligned-access -marm -fexceptions
-		override CXXFLAGS+=-march=armv5t -mno-unaligned-access -marm -fexceptions
+		override CFLAGS+=-march=armv6z -mno-unaligned-access -marm -fexceptions
+		override CXXFLAGS+=-march=armv6z -mno-unaligned-access -marm -fexceptions
 		ZT_USE_ARM32_NEON_ASM_CRYPTO=0
 	endif
 endif
