diff --git a/tests/check_signon.c b/tests/check_signon.c
index 3e1e758..405582e 100644
--- a/tests/check_signon.c
+++ b/tests/check_signon.c
@@ -388,7 +388,7 @@ START_TEST(test_auth_session_creation)
     const gchar *method = NULL;
 
     g_debug("%s", G_STRFUNC);
-    SignonIdentity *idty = signon_identity_new(NULL, NULL);
+    SignonIdentity *idty = signon_identity_new();
     fail_unless (idty != NULL, "Cannot create Identity object");
 
     SignonAuthSession *auth_session = signon_identity_create_session(idty,
@@ -442,7 +442,7 @@ START_TEST(test_auth_session_process_async)
     gboolean ok;
 
     g_debug("%s", G_STRFUNC);
-    SignonIdentity *idty = signon_identity_new(NULL, NULL);
+    SignonIdentity *idty = signon_identity_new();
     fail_unless (idty != NULL, "Cannot create Identity object");
 
     SignonAuthSession *auth_session = signon_identity_create_session(idty,
@@ -751,7 +751,7 @@ new_identity()
     GList *acl = g_list_append (NULL, signon_security_context_new_from_values ("*", "*"));
     guint id = 0;
 
-    identity = signon_identity_new(NULL, NULL);
+    identity = signon_identity_new();
     fail_unless (SIGNON_IS_IDENTITY (identity));
 
     SignonIdentityInfo *info = signon_identity_info_new ();
@@ -864,7 +864,7 @@ static void store_credentials_identity_cb (GObject *source_object,
 START_TEST(test_store_credentials_identity)
 {
     g_debug("%s", G_STRFUNC);
-    SignonIdentity *idty = signon_identity_new(NULL, NULL);
+    SignonIdentity *idty = signon_identity_new();
     fail_unless (idty != NULL);
     fail_unless (SIGNON_IS_IDENTITY (idty),
                  "Failed to initialize the Identity.");
@@ -911,7 +911,7 @@ START_TEST(test_verify_secret_identity)
     GList *acl = g_list_append (NULL, signon_security_context_new_from_values ("*", "*"));
 
     g_debug("%s", G_STRFUNC);
-    SignonIdentity *idty = signon_identity_new(NULL, NULL);
+    SignonIdentity *idty = signon_identity_new();
     fail_unless (idty != NULL);
     fail_unless (SIGNON_IS_IDENTITY (idty),
                  "Failed to initialize the Identity.");
