--- libmpdclient-2.8.orig/src/socket.c
+++ libmpdclient-2.8/src/socket.c
@@ -43,6 +43,7 @@
 #else
 #  include <netinet/in.h>
 #  include <arpa/inet.h>
+#  include <sys/types.h>
 #  include <sys/socket.h>
 #  include <netdb.h>
 #  include <sys/un.h>
--- libmpdclient-2.8.orig/src/sync.c
+++ libmpdclient-2.8/src/sync.c
@@ -35,6 +35,7 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <unistd.h>
+#include <sys/types.h>
 
 static bool
 ignore_errno(int e)
