diff --git a/ua-parser-builtins/hatch_build.py.orig b/ua-parser-builtins/hatch_build.py
index 1eac77643e0..7b9ede6aef7 100644
--- a/ua-parser-builtins/hatch_build.py.orig
+++ b/ua-parser-builtins/hatch_build.py
@@ -10,23 +10,11 @@ from typing import Any, Callable, ClassVar, Iterator, cast
 import yaml
 from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 from hatchling.metadata.plugin.interface import MetadataHookInterface
-from versioningit import get_version
 
 
 class MetadataHook(MetadataHookInterface):
     def update(self, metadata: dict[str, Any]) -> None:
-        v = get_version(
-            os.path.join(self.root, "uap-core"),
-            config={
-                "format": {
-                    "distance": "{next_version}.dev{distance}",
-                }
-            },
-        )
-        if v in ("0.15.0", "0.16.0", "0.18.0"):
-            v = f"{v}.post1"
-
-        metadata["version"] = v
+        metadata["version"] = "@@UAP_VERSION@@"
 
 
 class CompilerHook(BuildHookInterface):
