diff --git a/ua-parser-builtins/hatch_build.py.orig b/ua-parser-builtins/hatch_build.py
index 2a86012..e8fa532 100644
--- a/ua-parser-builtins/hatch_build.py.orig
+++ b/ua-parser-builtins/hatch_build.py
@@ -11,28 +11,11 @@ from typing import Any, Callable, ClassVar, Iterator, cast
 import yaml
 from hatchling.builders.hooks.plugin.interface import BuildHookInterface
 from hatchling.metadata.plugin.interface import MetadataHookInterface
-from versioningit import errors, get_version
 
 
 class MetadataHook(MetadataHookInterface):
     def update(self, metadata: dict[str, Any]) -> None:
-        try:
-            v = get_version(
-                os.path.join(self.root, "uap-core"),
-                config={
-                    "format": {
-                        "distance": "{next_version}.dev{distance}",
-                    }
-                },
-            )
-        except errors.NotSdistError:
-            with open(os.path.join(self.root, "uap-core", "package.json")) as ufile:
-                ujson = json.load(ufile)
-                v = ujson["version"]
-        if v in ("0.15.0", "0.16.0", "0.18.0"):
-            v = f"{v}.post1"
-
-        metadata["version"] = v
+        metadata["version"] = "@@UAP_VERSION@@"
 
 
 class CompilerHook(BuildHookInterface):
