diff --git a/src/ripple/app/main/Application.cpp b/src/ripple/app/main/Application.cpp
index 27d5498..b19548f 100644
--- a/src/ripple/app/main/Application.cpp
+++ b/src/ripple/app/main/Application.cpp
@@ -1807,7 +1807,7 @@ bool ApplicationImp::loadOldLedger (
                 }
             }
         }
-        else if (ledgerID.empty () || boost::beast::detail::iequals(ledgerID, "latest"))
+        else if (ledgerID.empty () || boost::beast::iequals(ledgerID, "latest"))
         {
             loadLedger = getLastFullLedger ();
         }
diff --git a/src/ripple/core/impl/Config.cpp b/src/ripple/core/impl/Config.cpp
index adb3039..6017b09 100644
--- a/src/ripple/core/impl/Config.cpp
+++ b/src/ripple/core/impl/Config.cpp
@@ -318,15 +318,15 @@ void Config::loadFromString (std::string const& fileContents)
 
     if (getSingleSection (secConfig, SECTION_NODE_SIZE, strTemp, j_))
     {
-        if (boost::beast::detail::iequals(strTemp, "tiny"))
+        if (boost::beast::iequals(strTemp, "tiny"))
             NODE_SIZE = 0;
-        else if (boost::beast::detail::iequals(strTemp, "small"))
+        else if (boost::beast::iequals(strTemp, "small"))
             NODE_SIZE = 1;
-        else if (boost::beast::detail::iequals(strTemp, "medium"))
+        else if (boost::beast::iequals(strTemp, "medium"))
             NODE_SIZE = 2;
-        else if (boost::beast::detail::iequals(strTemp, "large"))
+        else if (boost::beast::iequals(strTemp, "large"))
             NODE_SIZE = 3;
-        else if (boost::beast::detail::iequals(strTemp, "huge"))
+        else if (boost::beast::iequals(strTemp, "huge"))
             NODE_SIZE = 4;
         else
         {
@@ -376,9 +376,9 @@ void Config::loadFromString (std::string const& fileContents)
 
     if (getSingleSection (secConfig, SECTION_LEDGER_HISTORY, strTemp, j_))
     {
-        if (boost::beast::detail::iequals(strTemp, "full"))
+        if (boost::beast::iequals(strTemp, "full"))
             LEDGER_HISTORY = 1000000000u;
-        else if (boost::beast::detail::iequals(strTemp, "none"))
+        else if (boost::beast::iequals(strTemp, "none"))
             LEDGER_HISTORY = 0;
         else
             LEDGER_HISTORY = beast::lexicalCastThrow <std::uint32_t> (strTemp);
@@ -386,9 +386,9 @@ void Config::loadFromString (std::string const& fileContents)
 
     if (getSingleSection (secConfig, SECTION_FETCH_DEPTH, strTemp, j_))
     {
-        if (boost::beast::detail::iequals(strTemp, "none"))
+        if (boost::beast::iequals(strTemp, "none"))
             FETCH_DEPTH = 0;
-        else if (boost::beast::detail::iequals(strTemp, "full"))
+        else if (boost::beast::iequals(strTemp, "full"))
             FETCH_DEPTH = 1000000000u;
         else
             FETCH_DEPTH = beast::lexicalCastThrow <std::uint32_t> (strTemp);
diff --git a/src/ripple/net/impl/RPCCall.cpp b/src/ripple/net/impl/RPCCall.cpp
index 5d1fdb3..15946b5 100644
--- a/src/ripple/net/impl/RPCCall.cpp
+++ b/src/ripple/net/impl/RPCCall.cpp
@@ -163,7 +163,7 @@ private:
         // If odd number of params then 'novalidate' may have been specified
         if (sz & 1)
         {
-            using namespace boost::beast::detail;
+            using namespace boost::beast;
             if (iequals(jvParams[0u].asString(), "novalidate"))
                 ++i;
             else if (!iequals(jvParams[--sz].asString(), "novalidate"))
@@ -471,9 +471,9 @@ private:
             // This may look reversed, but it's intentional: jss::vetoed
             // determines whether an amendment is vetoed - so "reject" means
             // that jss::vetoed is true.
-            if (boost::beast::detail::iequals(action, "reject"))
+            if (boost::beast::iequals(action, "reject"))
                 jvRequest[jss::vetoed] = Json::Value (true);
-            else if (boost::beast::detail::iequals(action, "accept"))
+            else if (boost::beast::iequals(action, "accept"))
                 jvRequest[jss::vetoed] = Json::Value (false);
             else
                 return rpcError (rpcINVALID_PARAMS);
diff --git a/src/ripple/nodestore/impl/DatabaseShardImp.cpp b/src/ripple/nodestore/impl/DatabaseShardImp.cpp
index dbdbe05..df776d2 100644
--- a/src/ripple/nodestore/impl/DatabaseShardImp.cpp
+++ b/src/ripple/nodestore/impl/DatabaseShardImp.cpp
@@ -76,7 +76,7 @@ bool
 DatabaseShardImp::init()
 {
     using namespace boost::filesystem;
-    using namespace boost::beast::detail;
+    using namespace boost::beast;
 
     std::lock_guard<std::mutex> lock(m_);
     if (init_)
diff --git a/src/ripple/nodestore/impl/ManagerImp.cpp b/src/ripple/nodestore/impl/ManagerImp.cpp
index dc38a0c..5b14c6c 100644
--- a/src/ripple/nodestore/impl/ManagerImp.cpp
+++ b/src/ripple/nodestore/impl/ManagerImp.cpp
@@ -102,7 +102,7 @@ ManagerImp::find (std::string const& name)
     auto const iter = std::find_if(list_.begin(), list_.end(),
         [&name](Factory* other)
         {
-            return boost::beast::detail::iequals(name, other->getName());
+            return boost::beast::iequals(name, other->getName());
         } );
     if (iter == list_.end())
         return nullptr;
diff --git a/src/ripple/overlay/impl/OverlayImpl.cpp b/src/ripple/overlay/impl/OverlayImpl.cpp
index d3136b9..6bddce3 100644
--- a/src/ripple/overlay/impl/OverlayImpl.cpp
+++ b/src/ripple/overlay/impl/OverlayImpl.cpp
@@ -226,7 +226,7 @@ OverlayImpl::onHandoff (std::unique_ptr <beast::asio::ssl_bundle>&& ssl_bundle,
         if (std::find_if(types.begin(), types.end(),
                 [](std::string const& s)
                 {
-                    return boost::beast::detail::iequals(s, "peer");
+                    return boost::beast::iequals(s, "peer");
                 }) == types.end())
         {
             handoff.moved = false;
diff --git a/src/ripple/overlay/impl/PeerImp.cpp b/src/ripple/overlay/impl/PeerImp.cpp
index 224001f..281ac57 100644
--- a/src/ripple/overlay/impl/PeerImp.cpp
+++ b/src/ripple/overlay/impl/PeerImp.cpp
@@ -42,6 +42,7 @@
 #include <boost/algorithm/clamp.hpp>
 #include <boost/algorithm/string/predicate.hpp>
 #include <boost/algorithm/string.hpp>
+#include <boost/beast/core/ostream.hpp>
 #include <algorithm>
 #include <memory>
 #include <sstream>
@@ -251,7 +252,7 @@ PeerImp::crawl() const
     auto const iter = headers_.find("Crawl");
     if (iter == headers_.end())
         return false;
-    return boost::beast::detail::iequals(iter->value(), "public");
+    return boost::beast::iequals(iter->value(), "public");
 }
 
 std::string
diff --git a/src/ripple/server/WSSession.h b/src/ripple/server/WSSession.h
index 9734f36..7da5018 100644
--- a/src/ripple/server/WSSession.h
+++ b/src/ripple/server/WSSession.h
@@ -24,6 +24,7 @@
 #include <ripple/server/Port.h>
 #include <ripple/server/Writer.h>
 #include <boost/beast/core/buffers_prefix.hpp>
+#include <boost/beast/websocket.hpp>
 #include <boost/asio/buffer.hpp>
 #include <boost/asio/ip/tcp.hpp>
 #include <boost/logic/tribool.hpp>
diff --git a/src/ripple/server/impl/Port.cpp b/src/ripple/server/impl/Port.cpp
index dee4240..6299026 100644
--- a/src/ripple/server/impl/Port.cpp
+++ b/src/ripple/server/impl/Port.cpp
@@ -189,7 +189,7 @@ parse_Port (ParsedPort& port, Section const& section, std::ostream& log)
     {
         auto const lim = get (section, "limit", "unlimited");
 
-        if (!boost::beast::detail::iequals (lim, "unlimited"))
+        if (!boost::beast::iequals (lim, "unlimited"))
         {
             try
             {
diff --git a/src/test/server/ServerStatus_test.cpp b/src/test/server/ServerStatus_test.cpp
index fee6532..04834ed 100644
--- a/src/test/server/ServerStatus_test.cpp
+++ b/src/test/server/ServerStatus_test.cpp
@@ -30,6 +30,7 @@
 #include <boost/beast/http.hpp>
 #include <beast/test/yield_to.hpp>
 #include <boost/beast/core/multi_buffer.hpp>
+#include <boost/beast/core/ostream.hpp>
 #include <boost/asio.hpp>
 #include <boost/asio/ssl.hpp>
 #include <boost/algorithm/string/predicate.hpp>
