commit 3f0a5d44cccbd257226876e91730f2e2f9d89741
Author: Henrik Riomar <henrik.riomar@gmail.com>
Date:   Wed Mar 27 21:51:55 2019 +0100

    makefile: fix font path

diff --git a/Makefile b/Makefile
index e7a6b41..1c612ce 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ PID=$(VAR)/run
 MAN=$(PREFIX)/usr/share/man
 WEB=$(PREFIX)/usr/share/entropybroker/web
 DOC=$(PREFIX)/usr/share/doc/entropybroker
-FONT=/usr/share/fonts/truetype/freefont/FreeMono.ttf
+FONT=/usr/share/fonts/TTF/FreeMono.ttf
 
 CXX=g++
 DEBUG= # -pg #-DHELGRIND #-DCRYPTO_DEBUG #-D_DEBUG #-fprofile-arcs -ftest-coverage # -pg
diff --git a/entropy_broker.conf b/entropy_broker.conf
index 4ec9347..f716b91 100644
--- a/entropy_broker.conf
+++ b/entropy_broker.conf
@@ -94,4 +94,4 @@ users = users.txt
 # set port to -1 to disable
 webserver_interface = 0.0.0.0
 webserver_port = 48923
-# graph_font = /usr/share/fonts/truetype/freefont/FreeMono.ttf
+# graph_font = /usr/share/fonts/TTF/FreeMono.ttf
