index 7e27ed5..7d1f206 100644
--- a/Makefile
+++ b/Makefile
@@ -2,15 +2,15 @@ include version
 
 -include makefile.inc
 
-PREFIX=/usr/local/entropybroker
-BIN=$(PREFIX)/bin
-ETC=$(PREFIX)/etc
+PREFIX=$(DESTDIR)
+BIN=$(PREFIX)/usr/bin
+ETC=$(PREFIX)/etc/entropybroker
 VAR=$(PREFIX)/var
 CACHE=$(VAR)/cache
 PID=$(VAR)/run
-MAN=$(PREFIX)/share/man
-WEB=$(PREFIX)/share/eb/web
-DOC=$(PREFIX)/doc
+MAN=$(PREFIX)/usr/share/man
+WEB=$(PREFIX)/usr/share/entropybroker/web
+DOC=$(PREFIX)/usr/share/doc/entropybroker
 FONT=/usr/share/fonts/truetype/freefont/FreeMono.ttf
 
 CXX=g++
