diff --git a/user-dirs-update-gtk.desktop.in b/user-dirs-update-gtk.desktop.in
index 78d5f7c..6539552 100644
--- a/user-dirs-update-gtk.desktop.in
+++ b/user-dirs-update-gtk.desktop.in
@@ -4,7 +4,7 @@ Exec=xdg-user-dirs-gtk-update
 _Name=User folders update
 _Comment=Update common folders names to match current locale
 Terminal=false
-OnlyShowIn=GNOME;LXDE;Unity;
+OnlyShowIn=GNOME;LXDE;Unity;MATE;XFCE;
 Type=Application
 StartupNotify=false
 X-KDE-autostart-after=panel
