diff --git a/lib/jsonschema.lua b/lib/jsonschema.lua
index a59558e..3416a8d 100644
--- a/lib/jsonschema.lua
+++ b/lib/jsonschema.lua
@@ -49,7 +49,7 @@ if ngx then
 else
   match_pattern = function (s, p)
     if not rex_find then
-      local ok, rex = pcall(require, "rex_pcre")
+      local ok, rex = pcall(require, "rex_pcre2")
       if not ok then
         error("depends on lrexlib-pcre, please install it first: " .. rex)
       end
