--- ./src/egl/opengl/Makefile.in.orig
+++ ./src/egl/opengl/Makefile.in
@@ -111,10 +111,10 @@
 peglgears_LDADD = $(LDADD)
 xeglgears_SOURCES = xeglgears.c
 xeglgears_OBJECTS = xeglgears.$(OBJEXT)
-xeglgears_LDADD = $(LDADD)
+xeglgears_LDADD = $(LDADD) $(X11_LIBS)
 xeglthreads_SOURCES = xeglthreads.c
 xeglthreads_OBJECTS = xeglthreads.$(OBJEXT)
-xeglthreads_LDADD = $(LDADD)
+xeglthreads_LDADD = $(LDADD) $(X11_LIBS) -lpthread
 DEFAULT_INCLUDES = -I.@am__isrc@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
