diff -ru opennhrp-0.9.2.orig/nhrp/nhrp_peer.c opennhrp-0.9.2/nhrp/nhrp_peer.c
--- opennhrp-0.9.2.orig/nhrp/nhrp_peer.c	2009-02-02 13:12:46 +0000
+++ opennhrp-0.9.2/nhrp/nhrp_peer.c	2009-02-02 13:36:51 +0000
@@ -1260,7 +1260,8 @@
 {
 	char tmp[NHRP_PEER_FORMAT_LEN];
 
-	NHRP_BUG_ON(peer->flags & NHRP_PEER_FLAG_REMOVED);
+	if (peer->flags & NHRP_PEER_FLAG_REMOVED)
+		return;
 
 	nhrp_debug("Removing %s %s",
 		   nhrp_peer_type[peer->type],
