diff -ur chntpw-110511.orig/reged.c chntpw-110511/reged.c
--- chntpw-110511.orig/reged.c	2011-05-11 12:33:56.000000000 -0700
+++ chntpw-110511/reged.c	2011-11-01 13:46:55.569625286 -0700
@@ -167,6 +167,11 @@
 
   if (edit) {  /* Call editor. Rest of arguments are considered hives to load */
     hivename = argv[optind+no_hives];
+    if (!hivename) {
+      fprintf(stderr,"with -e you must specify at least one hive file name\n");
+      usage();
+      exit(1);
+    }
     do {
       if (!(hive[no_hives] = openHive(hivename,
 				      HMODE_RW|mode))) {
