diff -ur chntpw-110511.orig/ntreg.c chntpw-110511/ntreg.c
--- chntpw-110511.orig/ntreg.c	2011-05-11 12:33:56.000000000 -0700
+++ chntpw-110511/ntreg.c	2011-11-01 13:45:02.550421729 -0700
@@ -3389,7 +3389,14 @@
 
     fprintf(file,"\r\n"); /* Must end file with an empty line, windows does that */
 
-    fclose(file);
+    if (ferror (file)) {
+	printf("failed to write file '%s'\n", filename);
+	fclose (file);
+	return;
+    }
+    if (fclose(file))
+      printf("failed to write file '%s': %s\n", filename,
+	     strerror(errno));
 }
 
 /* ================================================================ */
