--- a/gnats/mail.c
+++ b/gnats/mail.c
@@ -67,7 +67,7 @@
   if (res != NULL)
     {
       if (res->admFields[ResponsibleAdmAlias] == NULL ||
-	  res->admFields[ResponsibleAdmAlias] == '\0')
+	  *res->admFields[ResponsibleAdmAlias] == '\0')
 	{
 	  if (res->admFields[ResponsibleAdmAlias] != NULL)
 	    {
--- a/gnats/misc.c
+++ b/gnats/misc.c
@@ -108,6 +108,7 @@
 	      fprintf (gnats_logfile, "%s", buf);
 	      break;
 	    }
+	  /* fallthrough */
 	  /* No log file, log to stderr. */
 	case STDERR:
 	  fprintf (stderr, "%s", buf);
--- a/gnats/regex.c
+++ b/gnats/regex.c
@@ -2310,6 +2310,7 @@
 	  if ((syntax & RE_BK_PLUS_QM)
 	      || (syntax & RE_LIMITED_OPS))
 	    goto normal_char;
+	  /* fallthrough */
 	handle_plus:
 	case '*':
 	  /* If there is no previous pattern... */
