--- a/integration/kubernetes_clusters_create_test.go
+++ b/integration/kubernetes_clusters_create_test.go
@@ -233,7 +233,7 @@
   },
   "node_pools": [
     {
-      "size": "s-1vcpu-2gb",
+      "size": "s-1vcpu-2gb-intel",
       "count": 3,
       "name": "some-cluster-name-default-pool"
     }
@@ -260,7 +260,7 @@
       "count": 2,
       "auto_scale": true,
       "name": "default",
-      "size": "s-1vcpu-2gb"
+      "size": "s-1vcpu-2gb-intel"
     }
   ]
 }
