diff --git a/src/util/virfile.c b/src/util/virfile.c
index 9863fd0..7046614 100644
--- a/src/util/virfile.c
+++ b/src/util/virfile.c
@@ -2953,9 +2953,8 @@ virFileGetDefaultHugepageSize(unsigned long long *size)
         goto cleanup;
 
     if (!(c = strstr(meminfo, HUGEPAGESIZE_STR))) {
-        virReportError(VIR_ERR_INTERNAL_ERROR,
-                       _("Unable to parse %s"),
-                       PROC_MEMINFO);
+        virReportError(VIR_ERR_NO_SUPPORT,
+                       _("Hugepage support not enabled in kernel"));
         goto cleanup;
     }
     c += strlen(HUGEPAGESIZE_STR);
@@ -2993,13 +2992,13 @@ virFileFindHugeTLBFS(virHugeTLBFSPtr *ret_fs,
     unsigned long long default_hugepagesz;
 
     if (virFileGetDefaultHugepageSize(&default_hugepagesz) < 0)
-        goto cleanup;
+        return ret;
 
     if (!(f = setmntent(PROC_MOUNTS, "r"))) {
         virReportSystemError(errno,
                              _("Unable to open %s"),
                              PROC_MOUNTS);
-        goto cleanup;
+        return ret;
     }
 
     while (getmntent_r(f, &mb, mntbuf, sizeof(mntbuf))) {
