From ca789ed64bf71281b6a998da41032270748aa90e Mon Sep 17 00:00:00 2001
From: Alexander Batischev <eual.jp@gmail.com>
Date: Wed, 1 Nov 2017 21:06:18 +0300
Subject: [PATCH] Use wchar_t to hold current input character

Fixes #43.
---
 filter/Scanner.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/filter/Scanner.h b/filter/Scanner.h
index 8ce5fc52..e578b8e1 100644
--- a/filter/Scanner.h
+++ b/filter/Scanner.h
@@ -222,7 +222,7 @@ class Scanner {
 	Token *tokens;    // list of tokens already peeked (first token is a dummy)
 	Token *pt;        // current peek token
 
-	int ch;           // current input character
+	wchar_t ch;       // current input character
 
 	int pos;          // byte position of current character
 	int charPos;      // position by unicode characters starting with 0
