--- a/controller/seafile-controller.c	2014-01-16 19:56:24.183523103 +0100
+++ b/controller/seafile-controller.c	2014-01-16 19:55:57.123523652 +0100
@@ -281,7 +281,7 @@
     tmp = g_path_get_dirname (bin_dir);
     installpath = g_path_get_dirname (tmp);
 
-    topdir = g_path_get_dirname (installpath);
+    //   topdir = g_path_get_dirname (installpath);
 
     g_free (exectuble);
     g_free (tmp);
@@ -643,6 +643,7 @@
         return -1;
     }
 
+    topdir = g_path_get_dirname(config_dir);
     ctl->client = ccnet_client_new ();
     ctl->sync_client = ccnet_client_new ();
 
@@ -657,14 +658,14 @@
     }
 
     if (logdir == NULL) {
-        char *topdir = g_path_get_dirname(config_dir);
+        //char *topdir = g_path_get_dirname(config_dir);
         logdir = g_build_filename (topdir, "logs", NULL);
         if (checkdir_with_mkdir(logdir) < 0) {
             fprintf (stderr, "failed to create log folder \"%s\": %s\n",
                      logdir, strerror(errno));
             return -1;
         }
-        g_free (topdir);
+        //g_free (topdir);
     }
 
     ctl->config_dir = config_dir;

