diff --git a/tests/test_daemon.cpp b/tests/test_daemon.cpp
index c848fb9..3e49d8d 100644
--- a/tests/test_daemon.cpp
+++ b/tests/test_daemon.cpp
@@ -174,7 +174,7 @@ TEST_F(Daemon, invoking_test_command_succeeds)
                         core::posix::wait::Flags::untraced)));
 }
 
-TEST_F(Daemon, invoking_run_succeeds)
+TEST_F(Daemon, DISABLED_invoking_run_succeeds)
 {
     auto json = R"_(
         {
