--- Server-Starter-0.11.orig/inc/Test/TCP.pm
+++ Server-Starter-0.11/inc/Test/TCP.pm
@@ -32,7 +32,7 @@
         next if _check_port($port);
         my $sock = IO::Socket::INET->new(
             Listen    => 5,
-            LocalAddr => '127.0.0.1',
+            LocalAddr => 'localhost',
             LocalPort => $port,
             Proto     => 'tcp',
             (($^O eq 'MSWin32') ? () : (ReuseAddr => 1)),
@@ -90,7 +90,7 @@
 
     my $remote = IO::Socket::INET->new(
         Proto    => 'tcp',
-        PeerAddr => '127.0.0.1',
+        PeerAddr => 'localhost',
         PeerPort => $port,
     );
     if ($remote) {
--- Server-Starter-0.11.orig/t/01-starter.t
+++ Server-Starter-0.11/t/01-starter.t
@@ -31,7 +31,7 @@
             my $buf;
             #sleep 1;
             my $sock = IO::Socket::INET->new(
-                PeerAddr => "127.0.0.1:$port",
+                PeerAddr => "localhost:$port",
                 Proto    => 'tcp',
             );
             ok($sock, 'connect');
@@ -61,7 +61,7 @@
                 'signal sent on hup',
             );
             $sock = IO::Socket::INET->new(
-                PeerAddr => "127.0.0.1:$port",
+                PeerAddr => "localhost:$port",
                 Proto    => 'tcp',
             );
             ok($sock, 'reconnect');
--- Server-Starter-0.11.orig/t/02-startfail.t
+++ Server-Starter-0.11/t/02-startfail.t
@@ -20,7 +20,7 @@
         sleep 3;
         {
             my $sock = IO::Socket::INET->new(
-                PeerAddr => "127.0.0.1:$port",
+                PeerAddr => "localhost:$port",
                 Proto    => 'tcp',
             );
             ok($sock, 'connect');
@@ -33,7 +33,7 @@
         sleep 1;
         {
             my $sock = IO::Socket::INET->new(
-                PeerAddr => "127.0.0.1:$port",
+                PeerAddr => "localhost:$port",
                 Proto    => 'tcp',
             );
             ok($sock, 'connect');
@@ -47,7 +47,7 @@
         sleep 5;
         {
             my $sock = IO::Socket::INET->new(
-                PeerAddr => "127.0.0.1:$port",
+                PeerAddr => "localhost:$port",
                 Proto    => 'tcp',
             );
             ok($sock, 'connect');
