--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@
 
 #Ask Techno about this one
 compile-bin:
-	CC_OPTS="-L/usr/local/lib/lua/5.3/ -llsqlite3 -static" cd ./src/ && fennel --compile-binary tkts.fnl tkts-bin $(STATIC_LUA_LIB) $(LUA_INCLUDE_DIR) --native-module /usr/local/lib/lua/5.3/lsqlite3.so
+	CC_OPTS="-L/usr/lib/lua/5.3/ -llsqlite3 -static" cd ./src/ && fennel5.3 --compile-binary tkts.fnl tkts-bin $(STATIC_LUA_LIB) $(LUA_INCLUDE_DIR) --native-library /usr/lib/lua/5.3/lsqlite3.so
 
 install-bin:
 	install ./src/tkts-bin -D $(DESTDIR)/tkts
