# Replace redis with valkey
--- a/etc/redis/grommunio.conf
+++ b/etc/redis/grommunio.conf
@@ -5,10 +5,10 @@
 timeout 0
 tcp-keepalive 300
 daemonize no
-supervised systemd
-pidfile /var/run/redis/default.pid
+supervised no
+pidfile /var/run/valkey/default.pid
 loglevel notice
-logfile /var/log/redis/default.log
+logfile /var/log/valkey/default.log
 databases 16
 always-show-logo yes
 save 900 1
@@ -19,7 +19,7 @@
 rdbchecksum yes
 dbfilename dump.rdb
 rdb-del-sync-files no
-dir /var/lib/redis/default/
+dir /var/lib/valkey/default/
 replica-serve-stale-data yes
 replica-read-only yes
 repl-diskless-sync no
