# Fix the deprecated call of http2 as listen parameter
--- a/usr/share/grommunio-common/nginx.conf
+++ b/usr/share/grommunio-common/nginx.conf
@@ -51,8 +51,9 @@
 
 server {
 	server_name  _;
-	listen [::]:443 ssl http2;
-	listen 443 ssl http2;
+	listen [::]:443 ssl;
+	listen 443 ssl;
+	http2 on;
 	include /etc/grommunio-common/nginx/ssl_*.conf;
 	include /usr/share/grommunio-common/nginx/ssl_params.conf;
 	include /usr/share/grommunio-common/nginx/proxy_params.conf;
