--- hexinject/Makefile.orig	2017-08-29 07:29:31.430661450 +0000
+++ hexinject/Makefile	2017-08-29 07:29:53.634411237 +0000
@@ -3,9 +3,9 @@
 LDFLAGS = -lpcap
 
 all:
-	$(CC) $(CFLAGS) $(LDFLAGS) -o hexinject hexinject.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o prettypacket prettypacket.c
-	$(CC) $(CFLAGS) $(LDFLAGS) -o hex2raw hex2raw.c
+	$(CC) $(CFLAGS) -o hexinject hexinject.c $(LDFLAGS)
+	$(CC) $(CFLAGS) -o prettypacket prettypacket.c $(LDFLAGS)
+	$(CC) $(CFLAGS) -o hex2raw hex2raw.c $(LDFLAGS)
 
 clean:
 	rm -f hexinject prettypacket hex2raw *~
