diff --git a/src/compat/compat.h b/src/compat/compat.h
index fe2d07e..7f23283 100644
--- a/src/compat/compat.h
+++ b/src/compat/compat.h
@@ -1024,15 +1024,6 @@ out:
 #define COMPAT_CANNOT_USE_MAX_MTU
 #endif
 
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 6, 0)
-#include <linux/skbuff.h>
-static inline void skb_reset_redirect(struct sk_buff *skb)
-{
-#ifdef CONFIG_NET_SCHED
-	skb_reset_tc(skb);
-#endif
-}
-#endif
 
 #if defined(ISUBUNTU1604)
 #include <linux/siphash.h>
