diff --git a/src/shared/rt.h b/src/shared/rt.h
index 24deb81..8a96542 100644
--- a/src/shared/rt.h
+++ b/src/shared/rt.h
@@ -20,9 +20,7 @@
 #include <sys/time.h>
 #include <time.h>
 
-#if HAVE_LIBBSD
-# include <bsd/sys/time.h>
-#else
+// timespecadd & timespecsub definitions come from <bsd/sys/time.h>
 # define timespecadd(ts_a, ts_b, dest) do { \
 		(dest)->tv_sec = (ts_a)->tv_sec + (ts_b)->tv_sec; \
 		(dest)->tv_nsec = (ts_a)->tv_nsec + (ts_b)->tv_nsec; \
@@ -39,7 +37,6 @@
 			(dest)->tv_nsec += 1000000000L; \
 		} \
 	} while (0)
-#endif
 
 /**
  * Structure used for time synchronization.
-- 
2.30.2

