--- a/src/compiler/crystal/config.cr	2019-09-23 16:08:27.000000000 +0000
+++ b/src/compiler/crystal/config.cr	2019-09-26 14:50:47.795131257 +0000
@@ -17,7 +17,7 @@
     def self.description
       formatted_sha = "[#{build_commit}] " if build_commit
       <<-DOC
-        Crystal #{version} #{formatted_sha}(#{date})
+        Crystal #{version} (#{date})
 
         LLVM: #{llvm_version}
         Default target: #{self.default_target}
