--- a/spec/std/kernel_spec.cr	2019-02-05 14:51:38.000000000 +0000
+++ b/spec/std/kernel_spec.cr	2019-02-19 19:11:02.129045734 +0000
@@ -232,7 +232,7 @@
     error.should_not contain("Stack overflow")
   end
 
-  it "detects stack overflow on the main stack" do
+  pending "detects stack overflow on the main stack" do
     # This spec can take some time under FreeBSD where
     # the default stack size is 0.5G.  Setting a
     # smaller stack size with `ulimit -s 8192`
