--- a/spec/std/socket/tcp_server_spec.cr
+++ b/spec/std/socket/tcp_server_spec.cr
@@ -32,7 +32,7 @@
         server.local_address.port.should be > 0
       end
 
-      it "raises when port is negative" do
+      pending "raises when port is negative" do
         expect_raises(Socket::Error, linux? ? "getaddrinfo: Servname not supported for ai_socktype" : "No address found for #{address}:-12 over TCP") do
           TCPServer.new(address, -12)
         end
