--- a/spec/std/http/cookie_spec.cr	2019-12-18 15:49:14.000000000 +0100
+++ b/spec/std/http/cookie_spec.cr	2020-02-18 12:07:51.036417255 +0100
@@ -218,7 +218,7 @@
         parse_set_cookie("bla=1; max-age=1").expired?.should eq false
       end
 
-      it "not expired" do
+      pending "not expired" do
         parse_set_cookie("bla=1; expires=Thu, 01 Jan 2020 00:00:00 -0000").expired?.should eq false
       end
 
