From 42244ce1672431345625263a3dbf8f86ecf4c934 Mon Sep 17 00:00:00 2001
From: Ali Polatel <alip@hexsys.org>
Date: Tue, 2 Dec 2025 22:23:35 +0100
Subject: [PATCH] lookup: fix ELOOP not reported with MISS_LAST (thx omni!)

---
 src/lookup.rs | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/lookup.rs b/src/lookup.rs
index 90fa7afd9..19d798a4a 100644
--- a/src/lookup.rs
+++ b/src/lookup.rs
@@ -2862,7 +2862,7 @@ pub fn safe_canonicalize<'a>(
             }
             Err(errno) => match miss_mode {
                 MissingHandling::Existing => return Err(errno),
-                MissingHandling::Normal if !parts.is_empty() => return Err(errno),
+                _ if !parts.is_empty() => return Err(errno),
                 _ => {
                     // Invalidate file type.
                     file_type = None;
@@ -4598,7 +4598,9 @@ mod tests {
             FsFlags::NO_FOLLOW_LAST | FsFlags::MISS_LAST,
             Some(&sandbox),
         );
-        assert!(matches!(result, Err(Errno::ELOOP)), "{result:?}");
+        //SAFETY: Missing is handled by read_path*.
+        //assert_eq!(result, Err(Errno::EEXIST));
+        assert!(result.is_ok(), "{result:?}");
 
         let result = safe_canonicalize(
             Pid::this(),
-- 
2.52.0

