--- a/Makefile
+++ b/Makefile
@@ -181,25 +181,25 @@
 	$(MAKE) install-vim
 install-man: $(MANS)
 	for man in $(MANS1); do \
-		$(INSTALL) -pm 0644 $$man $(DESTDIR)$(PREFIX)/$(MANDIR)/man1/; \
+		$(INSTALL) -D -pm 0644 $$man -t $(DESTDIR)$(PREFIX)/$(MANDIR)/man1/; \
 	done
 	for man in $(MANS2); do \
-		$(INSTALL) -pm 0644 $$man $(DESTDIR)$(PREFIX)/$(MANDIR)/man2/; \
+		$(INSTALL) -D -pm 0644 $$man -t $(DESTDIR)$(PREFIX)/$(MANDIR)/man2/; \
 	done
 	for man in $(MANS5); do \
-		$(INSTALL) -pm 0644 $$man $(DESTDIR)$(PREFIX)/$(MANDIR)/man5/; \
+		$(INSTALL) -D -pm 0644 $$man -t $(DESTDIR)$(PREFIX)/$(MANDIR)/man5/; \
 	done
 	for man in $(MANS7); do \
-		$(INSTALL) -pm 0644 $$man $(DESTDIR)$(PREFIX)/$(MANDIR)/man7/; \
+		$(INSTALL) -D -pm 0644 $$man -t $(DESTDIR)$(PREFIX)/$(MANDIR)/man7/; \
 	done
 install-vim: $(VIMS)
 	$(INSTALL) -d $(DESTDIR)$(PREFIX)/$(VIMDIR)/ftdetect
 	for f in $(VIMS_FTD); do \
-		$(INSTALL) -pm 0644 $$f $(DESTDIR)$(PREFIX)/$(VIMDIR)/ftdetect/; \
+		$(INSTALL) -D -pm 0644 $$f -t $(DESTDIR)$(PREFIX)/$(VIMDIR)/ftdetect/; \
 	done
 	$(INSTALL) -d $(DESTDIR)$(PREFIX)/$(VIMDIR)/syntax
 	for f in $(VIMS_SYN); do \
-		$(INSTALL) -pm 0644 $$f $(DESTDIR)$(PREFIX)/$(VIMDIR)/syntax/; \
+		$(INSTALL) -D -pm 0644 $$f -t $(DESTDIR)$(PREFIX)/$(VIMDIR)/syntax/; \
 	done
 uninstall:
 	for program in $(PROGRAMS); do \
