diff --git a/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c b/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c
index 053439b..3306236 100644
--- a/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c
+++ b/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl-bio.c
@@ -39,7 +39,7 @@
 #define MONGOC_LOG_DOMAIN "stream-tls-openssl-bio"
 
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 
 /* Magic vtable to make our BIO shim */
 static BIO_METHOD gMongocStreamTlsOpenSslRawMethods = {
diff --git a/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c b/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c
index fd1800b..0ccc4a1 100644
--- a/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c
+++ b/src/libmongoc/src/libmongoc/src/mongoc/mongoc-stream-tls-openssl.c
@@ -45,7 +45,7 @@
 
 #define MONGOC_STREAM_TLS_OPENSSL_BUFFER_SIZE 4096
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 static void
 BIO_meth_free (BIO_METHOD *meth)
 {
