--- a/configure.ac.orig	2016-12-15 04:44:32.000000000 +0100
+++ b/configure.ac	2017-01-05 23:24:59.072247182 +0100
@@ -2998,12 +2998,7 @@ if test x"$enable_lame" != "xno" ; then
 
 fi
 
-AC_PATH_PROGS(FCCACHE, fc-cache, false)
-if test x"$FCCACHE" = "xfalse"; then
-  AM_CONDITIONAL(HAVE_FC_CACHE, false)
-else
-  AM_CONDITIONAL(HAVE_FC_CACHE, true)
-fi
+AM_CONDITIONAL(HAVE_FC_CACHE, false)
 
 if test x"$is_amigaos4" = "xyes"; then
   if test x"$enable_sdlui" != "xyes" -a x"$enable_sdlui2" != "xyes"; then
