diff -ru a/src/opencbm.h b/src/opencbm.h
--- a/src/opencbm.h	2016-11-14 00:23:10.000000000 +0100
+++ b/src/opencbm.h	2017-01-20 19:54:12.098939871 +0100
@@ -131,6 +131,9 @@
 
 #endif
 
+/* There is no __u_char on alpine linux */
+#define __u_char unsigned char /*!< __u_char as unsigned char */
+
 /* specifiers for the IEC bus lines */
 #define IEC_DATA   0x01 /*!< Specify the DATA line */
 #define IEC_CLOCK  0x02 /*!< Specify the CLOCK line */
