--- a/build/meson/tests/meson.build
+++ b/build/meson/tests/meson.build
@@ -137,7 +137,7 @@
 # Tests (Use "meson test --list" to list all tests)
 # =============================================================================
 
-if tests_supported_oses.contains(host_machine_os)
+if false
   valgrind_prog = find_program('valgrind', ['/usr/bin/valgrind'], required: true)
   valgrindTest_py = files('valgrindTest.py')
   test('valgrindTest',
