--- ./modules/linux/vmhgfs/inode.c.orig
+++ ./modules/linux/vmhgfs/inode.c
@@ -1900,7 +1900,7 @@
                            p,
 #endif
                            &inode->i_dentry,
-                           d_alias) {
+                           d_u.d_alias) {
          int dcount = compat_d_count(dentry);
          if (dcount) {
             LOG(4, ("Found %s %d \n", dentry->d_name.name, dcount));
