--- userland-ffcc4bd7c4875b71376c4240116e251652c9bec0.orig/containers/CMakeLists.txt
+++ userland-ffcc4bd7c4875b71376c4240116e251652c9bec0/containers/CMakeLists.txt
@@ -13,7 +13,7 @@
 
 SET( GCC_COMPILER_FLAGS -Wall -g -O2 -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wcast-qual -Wwrite-strings -Wundef )
 SET( GCC_COMPILER_FLAGS ${GCC_COMPILER_FLAGS} -Wextra )#-Wno-missing-field-initializers )
-SET( GCC_COMPILER_FLAGS ${GCC_COMPILER_FLAGS} -D__STDC_VERSION__=199901L )
+SET( GCC_COMPILER_FLAGS ${GCC_COMPILER_FLAGS} -std=c99 )
 SET( GCC_COMPILER_FLAGS ${GCC_COMPILER_FLAGS} -Wno-missing-field-initializers )
 SET( GCC_COMPILER_FLAGS ${GCC_COMPILER_FLAGS} -Wno-unused-value )
 
--- userland-ffcc4bd7c4875b71376c4240116e251652c9bec0.orig/interface/vcos/pthreads/vcos_platform.h
+++ userland-ffcc4bd7c4875b71376c4240116e251652c9bec0/interface/vcos/pthreads/vcos_platform.h
@@ -44,6 +44,10 @@
 extern "C" {
 #endif
 
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 199309L
+#endif
+
 #include <pthread.h>
 #include <semaphore.h>
 #include <unistd.h>
