diff --git a/src/installer/src/core-sdk-tasks/TarGzFileCreateFromDirectory.cs.orig b/src/installer/src/core-sdk-tasks/TarGzFileCreateFromDirectory.cs
index 06b9857..066d595 100644
--- a/src/installer/src/core-sdk-tasks/TarGzFileCreateFromDirectory.cs.orig
+++ b/src/installer/src/core-sdk-tasks/TarGzFileCreateFromDirectory.cs
@@ -131,7 +131,7 @@ private string GetSourceSpecification()
 
         private string GetDestinationArchive()
         {
-            return $"-czf {DestinationArchive}";
+            return $"--hard-dereference -czf {DestinationArchive}";
         }
 
         private string GetExcludes()
