diff -rupN a/spsdk/tp/adapters/scard_utils.py b/spsdk/tp/adapters/scard_utils.py
--- a/spsdk/tp/adapters/scard_utils.py	2024-06-14 08:48:37.000000000 +0000
+++ b/spsdk/tp/adapters/scard_utils.py	2025-11-13 03:35:40.940000000 +0000
@@ -15,7 +15,7 @@ from spsdk.utils.misc import get_hash
 from spsdk.utils.spsdk_enum import SpsdkEnum
 
 try:
-    from smartcard.CardConnectionDecorator import CardConnection
+    from smartcard.CardConnection import CardConnection
     from smartcard.CardConnectionEvent import CardConnectionEvent
     from smartcard.CardConnectionObserver import CardConnectionObserver
     from smartcard.CardRequest import CardRequest
diff -rupN a/spsdk/tp/adapters/tpdev_scard.py b/spsdk/tp/adapters/tpdev_scard.py
--- a/spsdk/tp/adapters/tpdev_scard.py	2024-06-14 08:48:37.000000000 +0000
+++ b/spsdk/tp/adapters/tpdev_scard.py	2025-11-13 03:36:52.380000000 +0000
@@ -26,7 +26,7 @@ from spsdk.utils.misc import (
 )
 
 try:
-    from smartcard.CardConnectionDecorator import CardConnection
+    from smartcard.CardConnection import CardConnection
 except ImportError as e:
     raise SPSDKError(
         "pyscard package is missing, please install it with pip install 'spsdk[tp]' in order to use TP"
