diff --git a/Makefile.orig b/Makefile
index e1ddb9d510a..85ae5d83b20 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -19,7 +19,7 @@ build: $(all)
 
 # install system-wide
 # Set PREFIX and DESTDIR to configure where it is installed
-install: install-bins install-docs install-completions install-desktop
+install: install-bins install-docs install-completions
 
 # installs into your home directory
 install-home:
@@ -90,7 +90,7 @@ install-docs: docs install-mans
 		rsync -a --delete html/ $(DESTDIR)$(PREFIX)/$(SHAREDIR)/doc/git-annex/html/; \
 	fi
 
-install-bins: build
+install-bins:
 	install -d $(DESTDIR)$(PREFIX)/bin
 	install git-annex $(DESTDIR)$(PREFIX)/bin
 	ln -sf git-annex $(DESTDIR)$(PREFIX)/bin/git-annex-shell
@@ -98,10 +98,10 @@ install-bins: build
 	ln -sf git-annex $(DESTDIR)$(PREFIX)/bin/git-remote-tor-annex
 	ln -sf git-annex $(DESTDIR)$(PREFIX)/bin/git-remote-p2p-annex
 
-install-desktop: build Build/InstallDesktopFile
+install-desktop: Build/InstallDesktopFile
 	./Build/InstallDesktopFile $(PREFIX)/bin/git-annex || true
 
-install-completions: build
+install-completions:
 	install -d $(DESTDIR)$(PREFIX)/$(SHAREDIR)/bash-completion/completions
 	install -m 0644 bash-completion.bash $(DESTDIR)$(PREFIX)/$(SHAREDIR)/bash-completion/completions/git-annex
 	install -d $(DESTDIR)$(ZSH_COMPLETIONS_PATH)
