diff --git a/Utility/Url.hs b/Utility/Url.hs
index 40fa4832a1..903491c0e0 100644
--- a/Utility/Url.hs
+++ b/Utility/Url.hs
@@ -670,15 +670,6 @@ parseRequestRelaxed u = case uriAuthority u of
                        u { uriAuthority = Just $ ua { uriPort = "" } }
        _ -> parseRequest (show u)
 
-hAcceptEncoding :: CI.CI B.ByteString
-hAcceptEncoding = "Accept-Encoding"
-
-hContentDisposition :: CI.CI B.ByteString
-hContentDisposition = "Content-Disposition"
-
-hContentRange :: CI.CI B.ByteString
-hContentRange = "Content-Range"
-
 resumeFromHeader :: FileSize -> Header
 resumeFromHeader sz = (hRange, renderByteRanges [ByteRangeFrom sz])
