diff --git a/cabal.project.freeze.orig b/cabal.project.freeze
index 5382c516996..e7c1734b2cf 100644
--- a/cabal.project.freeze.orig
+++ b/cabal.project.freeze
@@ -342,7 +342,7 @@ constraints: any.Cabal ==3.10.2.0,
              any.wai ==3.2.4,
              any.wai-app-static ==3.1.9,
              wai-app-static +crypton -print,
-             any.wai-extra ==3.1.17,
+             any.wai-extra ==3.1.16,
              wai-extra -build-example,
              any.wai-logger ==2.5.0,
              any.warp ==3.4.6,
 index-state: hackage.haskell.org 2024-11-30T19:00:43Z
diff --git a/git-annex.cabal.orig b/git-annex.cabal
index 83d6b02489d..ca96d5ce071 100644
--- a/git-annex.cabal.orig
+++ b/git-annex.cabal
@@ -350,7 +350,7 @@ Executable git-annex
       warp (>= 3.2.8),
       warp-tls (>= 3.2.2),
       wai,
-      wai-extra,
+      wai-extra (< 3.1.17),
       blaze-builder,
       clientsession,
       template-haskell,
