--- a/cmd/glab/main.go
+++ b/cmd/glab/main.go
@@ -110,8 +110,6 @@ func main() {
 	rootCmd := commands.NewCmdRoot(cmdFactory)
 	cmd, _, err := rootCmd.Traverse(expandedArgs)
 
-	setupTelemetryHook(cfg, cmdFactory, cmd)
-
 	if err != nil || cmd == rootCmd {
 		originalArgs := expandedArgs
 		isShell := false
@@ -233,12 +231,6 @@ func setupSurveyCore(io *iostreams.IOStr
 	}
 }
 
-func setupTelemetryHook(cfg config.Config, f cmdutils.Factory, cmd *cobra.Command) {
-	if isTelemetryEnabled(cfg) {
-		cobra.OnFinalize(addTelemetryHook(f, cmd))
-	}
-}
-
 func checkForUpdate(f cmdutils.Factory, rootCmd *cobra.Command, debug bool) {
 	if !isUpdateCheckEnabled(f) {
 		return
