--- a/t/08-performance/99-misc.t
+++ b/t/08-performance/99-misc.t
@@ -31,6 +31,6 @@
 { # https://github.com/rakudo/rakudo/issues/1740
     my $t-plain = { (^∞).grep(*.is-prime)[1000];       now - ENTER now }();
     my $t-hyper = { (^∞).hyper.grep(*.is-prime)[1000]; now - ENTER now }();
-    cmp-ok $t-hyper, '≤', $t-plain*2,
+    cmp-ok $t-hyper, '≤', $t-plain*4,
         'hypered .grep .is-prime is not hugely slower than plain grep';
 }
