From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 15 Nov 2025 16:10:44 +0100
Subject: [PATCH] Don't install /usr/share/kea/meson-info

Why would we...?

diff --git a/meson.build b/meson.build
index 8ed5b2d..4f99748 100644
--- a/meson.build
+++ b/meson.build
@@ -1134,18 +1134,3 @@ endif
 
 # Print the setup report.
 message(run_command(['cat', CONFIG_REPORT], check: true).stdout())
-
-# Copy the meson.info directory during installation.
-install_meson_info = configure_file(
-    input: 'install-meson-info.sh.in',
-    output: 'install-meson-info.sh',
-    configuration: configuration_data(
-        {
-            'INSTALL': INSTALL.full_path(),
-            'PREFIX': PREFIX,
-            'TOP_BUILD_DIR': TOP_BUILD_DIR,
-            'DATADIR': DATADIR,
-        },
-    ),
-)
-meson.add_install_script(install_meson_info)
