From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sun, 16 Nov 2025 23:28:28 +0100
Subject: [PATCH] Don't build and install HTML docs

diff --git a/doc/sphinx/make-doc.sh.in b/doc/sphinx/make-doc.sh.in
index 5ba8b95..1200303 100755
--- a/doc/sphinx/make-doc.sh.in
+++ b/doc/sphinx/make-doc.sh.in
@@ -16,10 +16,10 @@ rm -f '@srcdir@/arm/platforms.rst'
 cp '@TOP_SOURCE_DIR@/platforms.rst' '@srcdir@/arm/platforms.rst'
 
 # html
-@SPHINXBUILD@ -M html @srcdir@ @sphinxbuilddir@ @sphinxopts@
+#@SPHINXBUILD@ -M html @srcdir@ @sphinxbuilddir@ @sphinxopts@
 
 # singlehtml
-@SPHINXBUILD@ -M singlehtml @srcdir@ @sphinxbuilddir@ @sphinxopts@
+#@SPHINXBUILD@ -M singlehtml @srcdir@ @sphinxbuilddir@ @sphinxopts@
 
 # text
 @SPHINXBUILD@ -M text @srcdir@ @sphinxbuilddir@ @sphinxopts@
diff --git a/doc/sphinx/install-docs.sh.in b/doc/sphinx/install-docs.sh.in
index 5caa0c8..96d6177 100755
--- a/doc/sphinx/install-docs.sh.in
+++ b/doc/sphinx/install-docs.sh.in
@@ -22,8 +22,8 @@ fi
 docdir="${target_path}/@docdir@"
 mandir8="${target_path}/@mandir8@"
 
-mkdir -p "${docdir}/html"
-cp -r '@abs_sphinxbuilddir@/html'/* "${docdir}/html"
+#mkdir -p "${docdir}/html"
+#cp -r '@abs_sphinxbuilddir@/html'/* "${docdir}/html"
 if test '@HAVE_PDFLATEX@' != 'no'; then
     @INSTALL@ -c -m 644 @abs_sphinxbuilddir@/latex/kea-arm.pdf "${docdir}"
     @INSTALL@ -c -m 644 @abs_sphinxbuilddir@/latex/kea-messages.pdf "${docdir}"
