From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 26 Oct 2021 22:40:00 +0200
Subject: [PATCH] Move api files to /usr/share/doc/kea

--- a/src/share/api/Makefile.am
+++ b/src/share/api/Makefile.am
@@ -10,8 +10,8 @@
 EXTRA_DIST += $(api_files)
 
 install-data-local:
-	mkdir -p $(DESTDIR)$(datarootdir)/${PACKAGE_NAME}/api
-	cp $(api_files) $(DESTDIR)$(datarootdir)/${PACKAGE_NAME}/api
+	mkdir -p $(DESTDIR)$(datarootdir)/doc/${PACKAGE_NAME}/api
+	cp $(api_files) $(DESTDIR)$(datarootdir)/doc/${PACKAGE_NAME}/api
 
 uninstall-local:
-	rm -rf $(DESTDIR)$(datarootdir)/${PACKAGE_NAME}/api
+	rm -rf $(DESTDIR)$(datarootdir)/doc/${PACKAGE_NAME}/api
