diff -rupN a/src/blockdiag/imagedraw/__init__.py b/src/blockdiag/imagedraw/__init__.py
--- a/src/blockdiag/imagedraw/__init__.py	2021-12-06 16:43:47.000000000 +0000
+++ b/src/blockdiag/imagedraw/__init__.py	2026-04-07 23:27:25.410000000 +0000
@@ -13,7 +13,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-import pkg_resources
+from importlib.metadata import entry_points
 
 from blockdiag.utils.logging import warning
 
@@ -21,7 +21,7 @@ drawers = {}
 
 
 def init_imagedrawers(debug=False):
-    for drawer in pkg_resources.iter_entry_points('blockdiag_imagedrawers'):
+    for drawer in entry_points(group='blockdiag_imagedrawers'):
         try:
             module = drawer.load()
             if hasattr(module, 'setup'):
diff -rupN a/src/blockdiag/noderenderer/__init__.py b/src/blockdiag/noderenderer/__init__.py
--- a/src/blockdiag/noderenderer/__init__.py	2021-12-06 16:43:47.000000000 +0000
+++ b/src/blockdiag/noderenderer/__init__.py	2026-04-07 23:27:36.700000000 +0000
@@ -15,14 +15,14 @@
 
 from __future__ import division
 
-import pkg_resources
+from importlib.metadata import entry_points
 
 renderers = {}
 searchpath = []
 
 
 def init_renderers():
-    for plugin in pkg_resources.iter_entry_points('blockdiag_noderenderer'):
+    for plugin in entry_points(group='blockdiag_noderenderer'):
         module = plugin.load()
         if hasattr(module, 'setup'):
             module.setup(module)
diff -rupN a/src/blockdiag/plugins/__init__.py b/src/blockdiag/plugins/__init__.py
--- a/src/blockdiag/plugins/__init__.py	2021-12-06 16:43:47.000000000 +0000
+++ b/src/blockdiag/plugins/__init__.py	2026-04-07 23:25:56.840000000 +0000
@@ -13,7 +13,7 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-from pkg_resources import iter_entry_points
+from importlib.metadata import entry_points
 
 from blockdiag.utils.logging import warning
 
@@ -28,7 +28,7 @@ def load(plugins, diagram, **kwargs):
             warning('plugin "%s" is already loaded. ignored.', name)
             return
 
-        for ep in iter_entry_points('blockdiag_plugins', name):
+        for ep in entry_points(group='blockdiag_plugins', name=name):
             module = ep.load()
             loaded_plugins.append(name)
             if hasattr(module, 'setup'):
