diff -rupN a/src/blockdiag/tests/diagrams/node_icon.diag b/src/blockdiag/tests/diagrams/node_icon.diag
--- a/src/blockdiag/tests/diagrams/node_icon.diag	2021-12-06 16:43:47.000000000 +0000
+++ b/src/blockdiag/tests/diagrams/node_icon.diag	2026-04-08 00:41:32.480000000 +0000
@@ -1,6 +1,6 @@
 {
   A -> B;
 
-  A [label = "aaaaaaaaaaaaaaaaa", icon = "/usr/share/pixmaps/debian-logo.png"];
-  B [label = "aaaaaaaaaaaaaaaaa", icon = "http://blockdiag.com/favicon.ico"];
+  A [label = "aaaaaaaaaaaaaaaaa", icon = "src/blockdiag/tests/diagrams/debian-logo-256color-palettealpha.png"];
+  B [label = "aaaaaaaaaaaaaaaaa", icon = "https://github.com/favicon.ico"];
 }
