diff -rupN a/src/blockdiag/utils/bootstrap.py b/src/blockdiag/utils/bootstrap.py
--- a/src/blockdiag/utils/bootstrap.py	2021-12-06 16:43:47.000000000 +0000
+++ b/src/blockdiag/utils/bootstrap.py	2026-04-08 00:52:58.510000000 +0000
@@ -13,7 +13,6 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-import codecs
 import os
 import re
 import sys
@@ -82,7 +81,7 @@ class Application(object):
             if self.code.startswith('\ufeff'):  # strip BOM
                 self.code = self.code[1:]
         else:
-            fp = codecs.open(self.options.input, 'r', 'utf-8-sig')
+            fp = open(self.options.input, encoding='utf-8-sig')
             self.code = fp.read()
 
         return self.module.parser.parse_string(self.code)
