From a23f3054b96b487215b04812c680075c5117470a Mon Sep 17 00:00:00 2001
From: Felix Schwarz <felix.schwarz@oss.schwarz.eu>
Date: Thu, 16 Sep 2021 23:02:46 +0200
Subject: [PATCH] get test suite passing with Python 3.10

---
 genshi/compat.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/genshi/compat.py b/genshi/compat.py
index 4a77571..8b307e5 100644
--- a/genshi/compat.py
+++ b/genshi/compat.py
@@ -99,6 +99,13 @@ def get_code_params(code):
 
 
     def build_code_chunk(code, filename, name, lineno):
+        if hasattr(code, 'replace'):
+            # Python 3.8+
+            return code.replace(
+                co_filename=filename,
+                co_name=name,
+                co_firstlineno=lineno,
+            )
         params =  [0, code.co_nlocals, code.co_kwonlyargcount,
                   code.co_stacksize, code.co_flags | 0x0040,
                   code.co_code, code.co_consts, code.co_names,
