From 38035cf0b6f9a8af1938930d151973687d4ba561 Mon Sep 17 00:00:00 2001
From: Leon Marz <lmarz@cs.uni-frankfurt.de>
Date: Tue, 1 Sep 2020 09:11:18 +0200
Subject: [PATCH 2/3] fix linking issue

---
 intern/ghost/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/intern/ghost/CMakeLists.txt b/intern/ghost/CMakeLists.txt
index 1739659..d6b442b 100644
--- a/intern/ghost/CMakeLists.txt
+++ b/intern/ghost/CMakeLists.txt
@@ -484,5 +484,5 @@ if(WITH_XR_OPENXR)
 endif()
 
 add_definitions(${GL_DEFINITIONS})
-
+list(APPEND LIB "-lX11" "-lXi" "-lXxf86vm" "-lXfixes" "-lXrender")
 blender_add_lib(bf_intern_ghost "${SRC}" "${INC}" "${INC_SYS}" "${LIB}")
-- 
2.31.1

