--- ./include/pgm/types.h
+++ ./include/pgm/types.h
@@ -27,6 +27,7 @@
 
 #ifndef _MSC_VER
 #	include <sys/param.h>
+#	include <sys/types.h>
 #endif
 #include <pgm/macros.h>
 
--- ./include/impl/security.h
+++ ./include/impl/security.h
@@ -33,7 +33,6 @@
 #include <stdio.h>
 #include <stdarg.h>
 #include <sys/types.h>
-#include <sys/timeb.h>
 #include <impl/i18n.h>
 #include <impl/errno.h>
 #include <impl/string.h>
@@ -41,6 +40,8 @@
 PGM_BEGIN_DECLS
 
 #ifdef HAVE_FTIME
+#include <sys/timeb.h>
+
 static inline
 errno_t
 #	if   !defined( _WIN32 )

