From 25a36c5cb4268544caa0843bde968a9c75df6596 Mon Sep 17 00:00:00 2001
From: Brenden Matthews <brenden@brndn.io>
Date: Sat, 24 Dec 2022 14:17:46 -0500
Subject: [PATCH] Add missing unistd include.

---
 src/display-wayland.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/display-wayland.cc b/src/display-wayland.cc
index ae01c20e0..6353f7b48 100644
--- a/src/display-wayland.cc
+++ b/src/display-wayland.cc
@@ -38,6 +38,7 @@
 #include <sys/epoll.h>
 #include <sys/mman.h>
 #include <sys/timerfd.h>
+#include <unistd.h>
 
 #include <wayland-client-protocol.h>
 #include <wlr-layer-shell-client-protocol.h>

