diff --git a/incoming/tests/test_datatypes.py b/incoming/tests/test_datatypes.py
index c1e7586..6d8a6dc 100644
--- a/incoming/tests/test_datatypes.py
+++ b/incoming/tests/test_datatypes.py
@@ -136,3 +136,3 @@ class TestJSON(TestCase):
         validator = datatypes.JSON(CustomJSONValidator)
-        self.assertEquals(
+        self.assertEqual(
             validator.cls.inner.cls.__name__,
diff --git a/incoming/tests/test_incoming.py b/incoming/tests/test_incoming.py
index 7f0321b..1ac82e7 100644
--- a/incoming/tests/test_incoming.py
+++ b/incoming/tests/test_incoming.py
@@ -70,3 +70,3 @@ class TestPayloadValidator(TestCase):
         self.assertTrue(result[0])
-        self.assertEquals(result[1], None)
+        self.assertEqual(result[1], None)
 
@@ -205,5 +205,5 @@ class TestPayloadValidator(TestCase):
         self.assertItemsEqual(errors.keys(), ['missing1', 'missing2'])
-        self.assertEquals(errors['missing1'][0],
+        self.assertEqual(errors['missing1'][0],
                           AnotherDummyValidator.strict_error)
-        self.assertEquals(errors['missing2'][0],
+        self.assertEqual(errors['missing2'][0],
                           AnotherDummyValidator.strict_error)
@@ -222,3 +222,3 @@ class TestPayloadValidator(TestCase):
         self.assertTrue(result)
-        self.assertEquals(errors, None)
+        self.assertEqual(errors, None)
 
