--- a/rstcheck.py
+++ b/rstcheck.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 
 # Copyright (C) 2013-2017 Steven Myint
 #
--- a/test.bash
+++ b/test.bash
@@ -73,9 +73,9 @@
 # Ignore message on configuration file
 ./rstcheck.py examples/with_configuration/bad-2.rst
 
-if python -c 'import sys; sys.exit(0 if sys.version_info >= (3,) else 1)'
+if python3 -c 'import sys; sys.exit(0 if sys.version_info >= (3,) else 1)'
 then
-    python -m doctest -v README.rst rstcheck.py
+    python3 -m doctest -v README.rst rstcheck.py
     ./rstcheck.py README.rst
 fi
 
