From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 15 Nov 2024 01:15:00 +0100
Subject: Fix compatibility with musl libc

musl doesn't implement GLOB_BRACE.

--- a/src/plugins/input/fds/fds.cpp
+++ b/src/plugins/input/fds/fds.cpp
@@ -113,7 +113,7 @@
 #ifndef GLOB_TILDE_CHECK
 #define GLOB_TILDE_CHECK GLOB_TILDE
 #endif
-    int glob_flags = GLOB_MARK | GLOB_BRACE | GLOB_TILDE_CHECK;
+    int glob_flags = GLOB_MARK | GLOB_TILDE_CHECK;
     size_t file_cnt;
     int ret;
 

--- a/src/plugins/input/ipfix/ipfix.c
+++ b/src/plugins/input/ipfix/ipfix.c
@@ -135,7 +135,7 @@
 #ifndef GLOB_TILDE_CHECK
 #define GLOB_TILDE_CHECK GLOB_TILDE
 #endif
-    int glob_flags = GLOB_MARK | GLOB_BRACE | GLOB_TILDE_CHECK;
+    int glob_flags = GLOB_MARK | GLOB_TILDE_CHECK;
     int rc = glob(pattern, glob_flags, NULL, list);
 
     switch (rc) {
--- a/src/tools/fdsdump/src/common/filelist.cpp
+++ b/src/tools/fdsdump/src/common/filelist.cpp
@@ -54,7 +54,7 @@
 
     std::unique_ptr<glob_t, decltype(&globfree)> glob_ptr {nullptr, &globfree};
     glob_t globbuf = {};
-    const int flags = GLOB_MARK | GLOB_BRACE | GLOB_TILDE;
+    const int flags = GLOB_MARK | GLOB_TILDE;
     int ret;
 
     ret = glob(pattern.c_str(), flags, NULL, &globbuf);
