diff --git a/src/meson.build b/src/meson.build
index 09b774a..a05f74c 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -81,7 +81,7 @@ deps = [
   dependency('gio-unix-2.0'),
   dependency('gtk4', version: '>= 4.0'),
   dependency('gtk4-unix-print', version: '>= 4.0'),
-  dependency('gsettings-desktop-schemas', version: '>= 47.alpha'),
+  dependency('gsettings-desktop-schemas', version: '>= 47'),
   dependency('gnome-desktop-4'),
   dependency('gnome-bg-4'),
   libadwaita_dep,
