Fix types on 32-bit

--- a/tests/futimesat.c
+++ b/tests/futimesat.c
@@ -100,7 +100,7 @@ main(void)
 	tv[0].tv_sec = 0xdeadbeefU;
 	tv[0].tv_usec = 0xfacefeedU;
 	tv[1].tv_sec = (typeof(tv[1].tv_sec)) 0xcafef00ddeadbeefLL;
-	tv[1].tv_usec = (suseconds_t) 0xbadc0dedfacefeedLL;
+	tv[1].tv_usec = (typeof(tv[1].tv_usec)) 0xbadc0dedfacefeedLL;
 
 	k_futimesat(kfdcwd, kfname, (uintptr_t) tv);
 	printf("futimesat(AT_FDCWD, %s, [", qname);
--- a/tests/xettimeofday.c
+++ b/tests/xettimeofday.c
@@ -58,7 +58,7 @@ main(void)
 	       tz->tz_minuteswest, tz->tz_dsttime, sprintrc(-1));
 
 	tv->tv_sec = (typeof(tv->tv_sec)) 0xcafef00ddeadbeefLL;
-	tv->tv_usec = (suseconds_t) 0xbadc0dedfacefeedLL;
+	tv->tv_usec = (typeof(tv->tv_usec)) 0xbadc0dedfacefeedLL;
 	assert(syscall(__NR_settimeofday, tv, tz) == -1);
 	printf("settimeofday({tv_sec=%lld, tv_usec=%llu}"
 	       ", {tz_minuteswest=%d, tz_dsttime=%d}) = %s\n",
--- a/tests/xselect.c
+++ b/tests/xselect.c
@@ -214,7 +214,7 @@ main(void)
 	 */
 	*l_rs = (1UL << fds[0]) | (1UL << fds[1]);
 	tv_in.tv_sec = (typeof(tv_in.tv_sec)) 0xcafef00ddeadbeefLL;
-	tv_in.tv_usec = (suseconds_t) 0xbadc0dedfacefeedLL;
+	tv_in.tv_usec = (typeof(tv_in.tv_usec)) 0xbadc0dedfacefeedLL;
 	memcpy(tv, &tv_in, sizeof(tv_in));
 	rc = xselect(nfds, a_rs, a_rs, a_rs, a_tv);
 	if (rc < 0) {
--- a/tests/xutimes.c
+++ b/tests/xutimes.c
@@ -87,7 +87,7 @@ main(void)
 	tv[0].tv_sec = 0xdeadbeefU;
 	tv[0].tv_usec = 0xfacefeedU;
 	tv[1].tv_sec = (typeof(tv[1].tv_sec)) 0xcafef00ddeadbeefLL;
-	tv[1].tv_usec = (suseconds_t) 0xbadc0dedfacefeedLL;
+	tv[1].tv_usec = (typeof(tv[1].tv_usec)) 0xbadc0dedfacefeedLL;
 
 	k_utimes(kfname, (uintptr_t) tv);
 	printf("%s(%s, [", TEST_SYSCALL_STR, qname);
