From df67e0112c102a2227f6f515091a81eb24a02e69 Mon Sep 17 00:00:00 2001
From: Bobby Rong <rjl931189261@126.com>
Date: Fri, 20 Feb 2026 23:10:36 +0800
Subject: [PATCH] build: Fix building with eds 3.59 and -Ddocs=true

Completes: 21b9642e82d95a18e7f73655d0ed78e6c70c11f8.
---
 docs/meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/docs/meson.build b/docs/meson.build
index 67c66451..83a011f3 100644
--- a/docs/meson.build
+++ b/docs/meson.build
@@ -39,6 +39,10 @@ if eds_backend_enabled
     docs_common_valadoc_flags += ['-D', 'HAS_EDS_3_41']
   endif
 
+  if eds_dep.version().version_compare('>=3.59.1')
+    docs_common_valadoc_flags += ['-D', 'HAS_EDS_3_60']
+  endif
+
   valadoc_targets += {
     'name': 'folks-eds',
     'source_files': eds_backendlib_sources,
-- 
GitLab

