diff --git a/src/sat/bmc/bmcMaj8.c b/src/sat/bmc/bmcMaj8.c
index b0c5391a9..eb87ab9f0 100644
--- a/src/sat/bmc/bmcMaj8.c
+++ b/src/sat/bmc/bmcMaj8.c
@@ -46,6 +46,7 @@ ABC_NAMESPACE_IMPL_START
 #include <unistd.h>
 #endif
 #include <limits.h>
+#include <stdint.h>
 
 ////////////////////////////////////////////////////////////////////////
 ///                        DECLARATIONS                              ///
diff --git a/src/sat/bmc/bmcMaj9.c b/src/sat/bmc/bmcMaj9.c
index ea8ac9d17..9247d08c7 100644
--- a/src/sat/bmc/bmcMaj9.c
+++ b/src/sat/bmc/bmcMaj9.c
@@ -34,6 +34,7 @@ typedef __int64 int64_t;
 #include <unistd.h>
 #endif
 #include <limits.h>
+#include <stdint.h>
 
 #define KISSAT_UNSAT 20
 #define KISSAT_SAT   10
