--- a/src/DkGui/DkNoMacs.cpp
+++ b/src/DkGui/DkNoMacs.cpp
@@ -25,6 +25,7 @@
 
  *******************************************************************************************************/
 
+#include <iostream>
 #include "DkNoMacs.h"
 
 // my stuff
--- a/src/DkCore/DkMetaData.cpp
+++ b/src/DkCore/DkMetaData.cpp
@@ -25,6 +25,7 @@
 
  *******************************************************************************************************/
 
+#include <iostream>
 #include "DkMetaData.h"
 #include "DkUtils.h"
 #include "DkMath.h"
