diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1a31e58..a8664dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,6 +150,7 @@ include_directories (
 	${QUAZIP_ZLIB_INCLUDE_DIR}
 	${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/libqpsd	# needed for linux psd hack
 	${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/drif
+	QuaZip::QuaZip
 )
 
 if (APPLE) # todo: somehow add this to Mac.cmake or MacBuildTarget.cmake
diff --git a/cmake/Unix.cmake b/cmake/Unix.cmake
index 5567084..ef7cc03 100644
--- a/cmake/Unix.cmake
+++ b/cmake/Unix.cmake
@@ -104,12 +104,8 @@ if(ENABLE_QUAZIP)
   if(USE_SYSTEM_QUAZIP)
     SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-    find_package(QuaZip5 QUIET)
-    if(NOT QUAZIP_FOUND)
-	    message(FATAL_ERROR "QUAZIP not found. It's mandatory when used with ENABLE_QUAZIP enabled, you can also disable USE_SYSTEM_QUAZIP")
-    else()
-	    add_definitions(-DWITH_QUAZIP)
-    endif()
+    find_package(QuaZip-Qt5 REQUIRED)
+    add_definitions(-DWITH_QUAZIP)
   else()
     find_package(ZLIB REQUIRED)
     set(QUAZIP_INCLUDE_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/quazip-0.7/quazip ${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/quazip-0.7/)
diff --git a/cmake/UnixBuildTarget.cmake b/cmake/UnixBuildTarget.cmake
index af3c766..e71fe8b 100644
--- a/cmake/UnixBuildTarget.cmake
+++ b/cmake/UnixBuildTarget.cmake
@@ -48,6 +48,7 @@ target_link_libraries(
 	${OpenCV_LIBS}
 	${TIFF_LIBRARIES}
 	${QUAZIP_LIBRARIES}
+	QuaZip::QuaZip
 	)
 set_property(TARGET ${DLL_CORE_NAME} PROPERTY VERSION ${NOMACS_VERSION_MAJOR}.${NOMACS_VERSION_MINOR}.${NOMACS_VERSION_PATCH})
 set_property(TARGET ${DLL_CORE_NAME} PROPERTY SOVERSION ${NOMACS_VERSION_MAJOR})
diff --git a/src/DkCore/DkBasicLoader.cpp b/src/DkCore/DkBasicLoader.cpp
index d29ae33..0045cb2 100644
--- a/src/DkCore/DkBasicLoader.cpp
+++ b/src/DkCore/DkBasicLoader.cpp
@@ -54,7 +54,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip5/JlCompress.h>
+#include <JlCompress.h>
 #endif
 
 // opencv
diff --git a/src/DkCore/DkImageContainer.cpp b/src/DkCore/DkImageContainer.cpp
index 75fd1b2..d486d38 100644
--- a/src/DkCore/DkImageContainer.cpp
+++ b/src/DkCore/DkImageContainer.cpp
@@ -41,7 +41,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip5/JlCompress.h>
+#include <JlCompress.h>
 #endif
 #pragma warning(pop)		// no warnings from includes - end
 
diff --git a/src/DkCore/DkImageLoader.cpp b/src/DkCore/DkImageLoader.cpp
index 2c1c38e..6327615 100644
--- a/src/DkCore/DkImageLoader.cpp
+++ b/src/DkCore/DkImageLoader.cpp
@@ -79,7 +79,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip5/JlCompress.h>
+#include <JlCompress.h>
 #endif
 
 // opencv
@@ -2212,4 +2212,4 @@ QString DkImageLoader::fileName() const {
 	return mCurrentImage->fileName();
 }
 
-}
\ No newline at end of file
+}
diff --git a/src/DkGui/DkDialog.cpp b/src/DkGui/DkDialog.cpp
index 008c9a3..fc21ee5 100644
--- a/src/DkGui/DkDialog.cpp
+++ b/src/DkGui/DkDialog.cpp
@@ -97,7 +97,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip5/JlCompress.h>
+#include <JlCompress.h>
 #endif
 
 #pragma warning(pop)		// no warnings from includes - end
