--- sxmo-dwm-6.3.0.orig/dwm.c
+++ sxmo-dwm-6.3.0/dwm.c
@@ -36,6 +36,7 @@
 #include <X11/Xlib.h>
 #include <X11/Xproto.h>
 #include <X11/Xutil.h>
+#include <X11/XKBlib.h>
 #ifdef XINERAMA
 #include <X11/extensions/Xinerama.h>
 #endif /* XINERAMA */
@@ -961,6 +962,11 @@
 	return m;
 }
 
+int
+iswide() {
+	return sw > 1000 ? 1 : 0;
+}
+
 void
 drawbar(Monitor *m)
 {
@@ -1267,11 +1273,6 @@
 }
 #endif /* XINERAMA */
 
-int
-iswide() {
-	return sw > 1000 ? 1 : 0;
-}
-
 void
 markhold(KeySym keysym)
 {
@@ -1976,7 +1977,7 @@
 	/* init appearance */
 	scheme = ecalloc(LENGTH(colors), sizeof(Clr *));
 	for (i = 0; i < LENGTH(colors); i++)
-		scheme[i] = drw_scm_create(drw, colors[i], 3);
+		scheme[i] = drw_scm_create(drw, (const char **) colors[i], 3);
 	for (j = 0; j <= SchemeTabInactive; ++j) {
 		free(colors[j][ColFg]);
 		free(colors[j][ColBg]);
