i don't even know

--- a/comm/build/moz.configure/gecko_source.configure
+++ b/comm/build/moz.configure/gecko_source.configure
@@ -51,7 +51,7 @@
                 pass
 
             if len(lines) != 3:
-                log.warn("sourcestamp.txt is corrupt!")
+                log.warning("sourcestamp.txt is corrupt!")
                 return
 
             if lines and lines[line2read].startswith("http"):
