From c5412405c3a697d4bb8ecae96442a9d1c050983a Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@alpinelinux.org>
Date: Mon, 14 May 2018 09:01:32 +0000
Subject: [PATCH] Add svg support to default mimetypes

---
 darkhttpd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/darkhttpd.c b/darkhttpd.c
index 1c5eff5..6b81bf8 100644
--- a/darkhttpd.c
+++ b/darkhttpd.c
@@ -317,6 +317,7 @@ static const char *default_extension_map[] = {
     "image/gif"            " gif",
     "image/jpeg"           " jpeg jpe jpg",
     "image/png"            " png",
+    "image/svg+xml"        " svg svgz",
     "text/css"             " css",
     "text/html"            " html htm",
     "text/javascript"      " js",
-- 
2.17.0

