--- ./configure.ac.orig
+++ ./configure.ac
@@ -88,7 +88,6 @@
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_RANLIB
-gl_EARLY
 
 GETTEXT_PACKAGE=jwhois
 AC_SUBST(GETTEXT_PACKAGE)
@@ -209,14 +208,11 @@
 OLDCFLAGS=$CFLAGS
 AC_SUBST(OLDCFLAGS)
 
-gl_INIT
-
 AC_OUTPUT([ po/Makefile.in m4/Makefile 
 Makefile 
 doc/Makefile 
 doc/sv/Makefile 
 src/Makefile 
-gl/Makefile
 example/Makefile 
 jwhois.spec
 ])
--- ./Makefile.am.orig
+++ ./Makefile.am
@@ -19,9 +19,8 @@
 ##
 
 EXTRA_DIST = config.rpath  m4/ChangeLog config.rpath include/*.h example/jwhois.conf m4/*.m4 jwhois.spec.in jwhois.spec
-SUBDIRS = m4 doc gl src example po
+SUBDIRS = m4 doc src example po
 
-AM_CPPFLAGS = -I$(top_srcdir)/gl
 LDADD = lib/libgnu.a
 
-ACLOCAL_AMFLAGS = -I m4 -I gl/m4
+ACLOCAL_AMFLAGS = -I m4
--- ./src/Makefile.am.orig
+++ ./src/Makefile.am
@@ -30,9 +30,7 @@
          -DLOCALSTATEDIR="\"$(localstatedir)\"" \
          -DLOCALEDIR="\"$(localedir)\""
 
-jwhois_LDADD = $(top_srcdir)/gl/libgnu.a
-
 AM_CFLAGS = $(jwhois_CFLAGS)
 
 
-INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/intl -I$(top_srcdir)/gl
+INCLUDES = -I$(top_srcdir)/include -I$(top_srcdir)/intl
