--- minicom-2.6.2.orig/src/dial.c
+++ minicom-2.6.2/src/dial.c
@@ -39,11 +39,9 @@
 #include "intl.h"
 
 #ifdef VC_MUSIC
-#  if defined(__GLIBC__)
 #    include <sys/ioctl.h>
 #    include <sys/kd.h>
 #    include <sys/time.h>
-#  endif
 #endif
 
 enum { CURRENT_VERSION = 6 };
--- minicom-2.6.2.orig/src/getsdir.h
+++ minicom-2.6.2/src/getsdir.h
@@ -22,6 +22,7 @@
  *	and licensing conditions.  See the source, Luke.
  */
 
+#include <sys/param.h>
 #include <dirent.h>
 
 typedef struct dirEntry {		/* structure of data item */
