From f4185e0b10e2bd88e2e5208ca9059083f0694f1f Mon Sep 17 00:00:00 2001
From: Antoine Martin <dev@ayakael.net>
Date: Tue, 26 Jul 2022 22:11:45 -0400
Subject: [PATCH 1/1] package-only-necessary

---
 _scripts/build.js | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/_scripts/build.js b/_scripts/build.js
index 7052fca8..8628a866 100644
--- a/_scripts/build.js
+++ b/_scripts/build.js
@@ -24,12 +24,13 @@ if (platform == 'darwin') {
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020-2021 freetubeapp@protonmail.com',
+  electronDist: '/usr/lib/electron',
   // asar: false,
   // compression: 'store',
   productName,
@@ -67,7 +68,7 @@ const config = {
   linux: {
     category: 'Network',
     icon: '_icons/icon.svg',
-    target: ['deb', 'zip', 'apk', 'rpm', 'AppImage', 'pacman'],
+    target: ['dir'],
   },
   // See the following issues for more information
   // https://github.com/jordansissel/fpm/issues/1503
-- 
2.36.2

