diff --git a/_scripts/build.js.orig b/_scripts/build.js
index 035f986..75991d9 100644
--- a/_scripts/build.js.orig
+++ b/_scripts/build.js
@@ -36,12 +36,13 @@ if (platform === 'darwin') {
     arch = Arch.armv7l
   }
 
-  targets = Platform.LINUX.createTarget(['deb', 'zip', '7z', 'apk', 'rpm', 'AppImage', 'pacman'], arch)
+  targets = Platform.LINUX.createTarget(['dir'], arch)
 }
 
 const config = {
   appId: `io.freetubeapp.${name}`,
   copyright: 'Copyleft © 2020-2023 freetubeapp@protonmail.com',
+  electronDist: '/usr/lib/electron',
   // asar: false,
   // compression: 'store',
   productName,
@@ -85,7 +86,7 @@ const config = {
   linux: {
     category: 'Network',
     icon: '_icons/icon.svg',
-    target: ['deb', 'zip', '7z', 'apk', 'rpm', 'AppImage', 'pacman'],
+    target: ['dir'],
   },
   // See the following issues for more information
   // https://github.com/jordansissel/fpm/issues/1503
