diff --git a/gdk/gdkseatdefault.c b/gdk/gdkseatdefault.c
index ce4bdc82d2..9650f4b121 100644
--- a/gdk/gdkseatdefault.c
+++ b/gdk/gdkseatdefault.c
@@ -134,22 +134,9 @@ gdk_seat_default_grab (GdkSeat                *seat,
 
   if (capabilities & GDK_SEAT_CAPABILITY_ALL_POINTING)
     {
-      /* ALL_POINTING spans 3 capabilities; get the mask for the ones we have */
-      GdkEventMask pointer_evmask = 0;
-
-      /* We let tablet styli take over the pointer cursor */
-      if (capabilities & (GDK_SEAT_CAPABILITY_POINTER |
-                          GDK_SEAT_CAPABILITY_TABLET_STYLUS))
-        {
-          pointer_evmask |= POINTER_EVENTS;
-        }
-
-      if (capabilities & GDK_SEAT_CAPABILITY_TOUCH)
-        pointer_evmask |= TOUCH_EVENTS;
-
       status = gdk_device_grab (priv->master_pointer, window,
                                 GDK_OWNERSHIP_NONE, owner_events,
-                                pointer_evmask, cursor,
+                                POINTER_EVENTS, cursor,
                                 evtime);
     }
 
