From 01fb19ba4af6936559878beb1fac8de008fc2f68 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 29 Apr 2015 19:27:20 +0200
Subject: [PATCH] build fix for openrc-0.14

---
 rc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rc.c b/rc.c
index 4ab8036..fcb3d5c 100644
--- a/rc.c
+++ b/rc.c
@@ -1,4 +1,5 @@
 
+#include <sys/queue.h>
 #include <string.h>
 
 #include <rc.h>
-- 
2.3.6

