From 11d5af13ce8c55e48272fd0a45a3ee780d1b6dec Mon Sep 17 00:00:00 2001
From: Frank Crawford <frank@crawford.emu.id.au>
Date: Sat, 31 Oct 2020 20:33:53 +1100
Subject: [PATCH] Address issue with '::' as an argument to rdiff-backup. See
 rdiff-backup issue #480.

---
 safekeep | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/safekeep b/safekeep
index 4123f1f..7875b93 100755
--- a/safekeep
+++ b/safekeep
@@ -1555,8 +1555,8 @@ def do_server_rdiff(cfg, bdir, nice, ionice, force):
 
     userhost = ''
     if cfg['host']:
-        userhost = '%s@%s' % (cfg['user'], cfg['host'])
-    args.extend([userhost + '::' + bdir, cfg['dir']])
+        userhost = '%s@%s::' % (cfg['user'], cfg['host'])
+    args.extend([userhost + bdir, cfg['dir']])
     ret = spawn(args)
     if ret:
         raise Exception('Failed to run rdiff-backup')
-- 
2.29.1

