From 535bc86ae27f4c4c5c76dc8ebf0019cd50139b0e Mon Sep 17 00:00:00 2001
From: Chris Mattmann <chris.mattmann@gmail.com>
Date: Sun, 6 Apr 2025 14:55:11 -0700
Subject: [PATCH] Fix issues in pdf.py and make sure dependencies are correct.

---
 requirements.txt | 1 +
 tika/pdf.py      | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/tika/pdf.py b/tika/pdf.py
index 128aab6b..b5daec50 100644
--- a/tika/pdf.py
+++ b/tika/pdf.py
@@ -16,7 +16,7 @@
 # limitations under the License.
 #
 
-import .tika.parser
+from tika import parser
 from io import StringIO
 from bs4 import BeautifulSoup
 
