diff --git a/tika/tests/test_tika.py b/tika/tests/test_tika.py
index c61cb81..183f195 100644
--- a/tika/tests/test_tika.py
+++ b/tika/tests/test_tika.py
@@ -33,7 +33,7 @@ class CreateTest(unittest.TestCase):
 
     def test_remote_html(self):
         """parse remote HTML"""
-        self.assertTrue(tika.parser.from_file('http://neverssl.com/index.html'))
+        self.assertTrue(tika.parser.from_file('http://bom.gov.au/index.html'))
 
     def test_remote_mp3(self):
         """parse remote mp3"""
