diff --git a/django_q/tests/settings.py.orig b/django_q/tests/settings.py
index b6246449429..b8b2f780e8e 100644
--- a/django_q/tests/settings.py.orig
+++ b/django_q/tests/settings.py
@@ -116,7 +116,7 @@ CACHES = {
         "LOCATION": f"redis://{REDIS_HOST}:6379/0",
         "OPTIONS": {
             "CLIENT_CLASS": "django_redis.client.DefaultClient",
-            "PARSER_CLASS": "redis.connection.HiredisParser",
+            "PARSER_CLASS": "redis.connection._HiredisParser",
         },
     }
 }
