diff --git a/art/runtime/arch/arm64/fault_handler_arm64.cc b/art/runtime/arch/arm64/fault_handler_arm64.cc
index 14a3fd52..576c914f 100644
--- a/art/runtime/arch/arm64/fault_handler_arm64.cc
+++ b/art/runtime/arch/arm64/fault_handler_arm64.cc
@@ -28,7 +28,9 @@
 #include "runtime_globals.h"
 #include "thread-current-inl.h"
 
+#ifdef __GLIBC__
 #define ucontext ucontext_t
+#endif
 
 extern "C" void art_quick_throw_stack_overflow();
 extern "C" void art_quick_throw_null_pointer_exception_from_signal();
