From 65b32b30a092d12d5ab3991f319c6fcfc10b5e04 Mon Sep 17 00:00:00 2001
From: Julian Winkler <julian.winkler1@web.de>
Date: Mon, 14 Apr 2025 16:43:25 +0200
Subject: [PATCH 1/2] libunwind: don't build target version

---
 external/libunwind/Android.mk | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/external/libunwind/Android.mk b/external/libunwind/Android.mk
index f2d65b234..bccec5cf2 100644
--- a/external/libunwind/Android.mk
+++ b/external/libunwind/Android.mk
@@ -253,16 +253,12 @@ endif
 
 libunwind_module := libunwind
 libunwind_module_tag := optional
-libunwind_build_type := target
 libunwind_build_target := SHARED_LIBRARY
-include $(LOCAL_PATH)/Android.build.mk
 libunwind_build_type := host
 include $(LOCAL_PATH)/Android.build.mk
-libunwind_build_type := target
 # Make sure the static library functions do not interfere with shared version.
 libunwind_cflags += -DUNW_ADDITIONAL_PREFIX
 libunwind_build_target := STATIC_LIBRARY
-include $(LOCAL_PATH)/Android.build.mk
 libunwind_build_type := host
 include $(LOCAL_PATH)/Android.build.mk
 
@@ -296,10 +292,8 @@ libunwindbacktrace_cflags += \
 
 libunwind_module := libunwindbacktrace
 libunwind_module_tag := optional
-libunwind_build_type := target
 libunwind_build_target := STATIC_LIBRARY
 libunwindbacktrace_whole_static_libraries := libunwind
-include $(LOCAL_PATH)/Android.build.mk
 libunwind_build_type := host
 include $(LOCAL_PATH)/Android.build.mk
 
@@ -323,9 +317,7 @@ libunwind-unit-tests_shared_libraries := \
 libunwind-unit-tests_multilib := both
 libunwind_module := libunwind-unit-tests
 libunwind_module_tag := optional
-libunwind_build_type := target
 libunwind_build_target := NATIVE_TEST
-include $(LOCAL_PATH)/Android.build.mk
 libunwind_build_type := host
 include $(LOCAL_PATH)/Android.build.mk
 
-- 
GitLab


From 460c65431c84e08a55d5f2bd3ba549b6f375efd1 Mon Sep 17 00:00:00 2001
From: Julian Winkler <julian.winkler1@web.de>
Date: Mon, 14 Apr 2025 16:44:10 +0200
Subject: [PATCH 2/2] libunwind x86: fix invalid cast

The cast is correct. See
https://github.com/libunwind/libunwind/commit/7f1aebadbfcbb2e96ca885663d1f8c9213b23e94.
So we have to cast it to void * first to avoid the compiler error on
latest GCC.
---
 external/libunwind/src/x86/Gos-linux.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/external/libunwind/src/x86/Gos-linux.c b/external/libunwind/src/x86/Gos-linux.c
index d84f125c2..089919506 100644
--- a/external/libunwind/src/x86/Gos-linux.c
+++ b/external/libunwind/src/x86/Gos-linux.c
@@ -303,7 +303,7 @@ x86_local_resume (unw_addr_space_t as, unw_cursor_t *cursor, void *arg)
       Debug (8, "resuming at ip=%x via sigreturn(%p)\n", c->dwarf.ip, sc);
 
 #if !defined(__ANDROID__)
-      x86_sigreturn (sc);
+      x86_sigreturn ((void *)sc);
 #endif
     }
   else
-- 
GitLab

