diff --git a/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp b/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
index 1e3c926b..6df78b1a 100644
--- a/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
+++ b/qt/OpenRGBNanoleafSettingsPage/OpenRGBNanoleafScanningThread.cpp
@@ -10,6 +10,8 @@
 #include <ifaddrs.h>
 #endif
 
+#include <sys/select.h>
+
 #include "mdns.h"
 
 #include "OpenRGBNanoleafScanningThread.h"
@@ -464,4 +466,4 @@ int OpenRGBNanoleafScanningThread::SendMDNSQuery()
 void OpenRGBNanoleafScanningThread::run()
 {
     SendMDNSQuery();
-}
\ No newline at end of file
+}
