--- a/contrib/sb-posix/constants.lisp
+++ b/contrib/sb-posix/constants.lisp
@@ -434,8 +434,8 @@
               (tcflag-t cflag "tcflag_t" "c_cflag")
               (tcflag-t lflag "tcflag_t" "c_lflag")
               ((array cc-t) cc "cc_t" "c_cc")
-              (speed-t ispeed "speed_t" "c_ispeed")
-              (speed-t ospeed "speed_t" "c_ospeed")))
+              (speed-t ispeed "speed_t" "__c_ispeed")
+              (speed-t ospeed "speed_t" "__c_ospeed")))
 
  ;; utime(), utimes()
  #-win32
