--- a/src/runtime/ppc-linux-os.c
+++ b/src/runtime/ppc-linux-os.c
@@ -32,6 +32,7 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <sys/prctl.h>
+#include <asm/ptrace.h>
 
 #include "validate.h"
 
--- a/src/runtime/os-common.c
+++ b/src/runtime/os-common.c
@@ -33,6 +33,9 @@
 #if defined LISP_FEATURE_UNIX && defined LISP_FEATURE_SOFT_CARD_MARKS
 #include "gc.h" // for find_page_index
 #endif
+#if defined(__powerpc__)
+# include <asm/ptrace.h>
+#endif
 
 /*
  * historically, this used sysconf to select the runtime page size
--- a/src/runtime/ppc-arch.c
+++ b/src/runtime/ppc-arch.c
@@ -10,6 +10,7 @@
  */
 
 #include <stdio.h>
+#include <asm/ptrace.h>
 
 #include "genesis/sbcl.h"
 #include "arch.h"
