musl libc does not support %z in strptime

diff --git a/changelog-date b/changelog-date
index e20904a..c5349cb 100755
--- a/changelog-date
+++ b/changelog-date
@@ -10,5 +10,5 @@ fi
 date="$(grep -E -e ' -- ' debian/changelog | head -1)"
 test "$date"
 date="$(echo "$date" | sed -Ee 's/.* (\S+ \S+ \S+ \S+ \S+)$/\1/')"
-date="$(./rewrite-time "$date" '%d %b %Y %H:%M:%S %z' '%Y-%m-%d')"
+date="$(./rewrite-time "$date" '%d %b %Y %H:%M:%S' '%Y-%m-%d')"
 echo "$date"
