fix library's basic functionality on big endian
diff --git a/pythongettext/msgfmt.py b/pythongettext/msgfmt.py
index 6641b52..61c4615 100755
--- a/pythongettext/msgfmt.py
+++ b/pythongettext/msgfmt.py
@@ -150,7 +150,7 @@ class Msgfmt:
         # Even though we don't use a hashtable, we still set its offset to be
         # binary compatible with the gnu gettext format produced by:
         # msgfmt file.po --no-hash
-        output = struct.pack("Iiiiiii",
+        output = struct.pack("<Iiiiiii",
                              0x950412de,        # Magic
                              0,                 # Version
                              len(keys),         # # of entries
