diff --git a/test/diff/submodule-editor-test b/test/diff/submodule-editor-test
index 26b1ef8..89161c9 100755
--- a/test/diff/submodule-editor-test
+++ b/test/diff/submodule-editor-test
@@ -40,7 +40,7 @@ setup_submodule_project()
 	cd "$output_dir"
 	git_init
 	cd "$output_dir/$work_dir"
-	git submodule add "$output_dir/base" bench
+	git -c protocol.file.allow=always submodule add "$output_dir/base" bench
   git_commit -m "Adding benchmark"
 }
 
diff --git a/test/status/submodule-editor-test b/test/status/submodule-editor-test
index 1c32639..094c7e5 100755
--- a/test/status/submodule-editor-test
+++ b/test/status/submodule-editor-test
@@ -28,7 +28,7 @@ setup_submodule_project()
 	cd "$output_dir"
 	git_init
 	cd "$output_dir/$work_dir"
-	git submodule add "$output_dir/base" bench
+	git -c protocol.file.allow=always submodule add "$output_dir/base" bench
 	git_commit -m "Adding benchmark"
 	echo CHANGE >> bench/README.md
 }
@@ -59,4 +59,4 @@ EOF
 assert_equals 'editor.log' <<EOF
 +63 README.md
 CHANGE
-EOF
\ No newline at end of file
+EOF
diff --git a/test/tree/submodule-editor-test b/test/tree/submodule-editor-test
index 3aa3ffb..e9be719 100755
--- a/test/tree/submodule-editor-test
+++ b/test/tree/submodule-editor-test
@@ -43,7 +43,7 @@ setup_submodule_project()
 	cd "$output_dir"
 	git_init
 	cd "$output_dir/$work_dir"
-	git submodule add "$output_dir/base" bench
+	git -c protocol.file.allow=always submodule add "$output_dir/base" bench
 	git_commit -m "Adding benchmark"
 }
 
