--- a/libyara/proc/linux.c
+++ b/libyara/proc/linux.c
@@ -249,7 +249,7 @@
   // target process VM.
   if (fd == -1)
   {
-    if (pread64(
+    if (pread(
             proc_info->mem_fd,
             (void*) context->buffer,
             block->size,
@@ -265,7 +265,7 @@
     {
       goto _exit;
     }
-    if (pread64(
+    if (pread(
             proc_info->pagemap_fd,
             pagemap,
             sizeof(uint64_t) * block->size / page_size,
@@ -284,7 +284,7 @@
       // swap-backed and if it differs from our mapping.
       uint8_t buffer[page_size];
 
-      if (pread64(
+      if (pread(
               proc_info->mem_fd,
               buffer,
               page_size,
