diff --git a/src/luaconf.h b/src/luaconf.h
index df802c9..f8f64e0 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -101,13 +99,22 @@
 
 #define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
 #define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT2	"/usr/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
+#define LUA_LDIR2	LUA_ROOT2 "share/lua/" LUA_VDIR
+#define LUA_LDIR3	LUA_ROOT2 "share/lua/common/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
+#define LUA_CDIR2	LUA_ROOT2 "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
 		LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
-		LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" "./?.lua"
+		LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" \
+		LUA_LDIR2"?.lua;"  LUA_LDIR2"?/init.lua;" \
+		LUA_CDIR2"?.lua;"  LUA_CDIR2"?/init.lua;" \
+		LUA_LDIR3"?.lua;"  LUA_LDIR3"?/init.lua;" \
+		"./?.lua"
 #define LUA_CPATH_DEFAULT \
-		LUA_CDIR"?.so;" LUA_CDIR"loadall.so;" "./?.so"
+		LUA_CDIR"?.so;" LUA_CDIR"loadall.so;" \
+		LUA_CDIR2"?.so;" LUA_CDIR2"loadall.so;" "./?.so"
 #endif			/* } */
 
 
