From 9f6c2a8fe34db847df957e55e0375b5cf15a44b9 Mon Sep 17 00:00:00 2001
From: nmcveity <nmcveity@gmail.com>
Date: Mon, 4 Mar 2013 22:49:27 -0800
Subject: [PATCH] Removed Lua 5.2 functions lua_len and lua_tointegerx

This fixes issue #71.
---
 lposix.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/lposix.c b/lposix.c
index ffd3c57..bf5accb 100644
--- a/lposix.c
+++ b/lposix.c
@@ -901,11 +901,7 @@ static int runexec(lua_State *L, int use_shell)
 	const char *path = luaL_checkstring(L, 1);
 	int i,n=lua_gettop(L), table = 0;
 	if (n >= 1 && lua_type(L, 2) == LUA_TTABLE) {
-		int isint;
-		lua_len(L, 2);
-		n = lua_tointegerx(L, -1, &isint);
-		if (!isint)
-			luaL_error(L, "argument 2 is a table, but has non-numeric length");
+		n = lua_objlen(L, 2);
 		table = 1;
 	} else
 		n--;
-- 
1.8.1.5

