From 1e2eb721ed696412c7aafc9538b3e45ac5c96ef9 Mon Sep 17 00:00:00 2001
From: Carlo Landmeter <clandmeter@gmail.com>
Date: Tue, 15 Apr 2014 09:51:53 +0000
Subject: [PATCH] alpinelinux: set correct lxc_arch for x86

---
 templates/lxc-alpine.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/templates/lxc-alpine.in b/templates/lxc-alpine.in
index d8c6674..7a22d5e 100644
--- a/templates/lxc-alpine.in
+++ b/templates/lxc-alpine.in
@@ -362,6 +362,7 @@ apk_arch=$arch
 case "$arch" in
     i[3-6]86)
         apk_arch=x86
+        lxc_arch=x86
         ;;
     x86)
         lxc_arch=i686
-- 
1.9.2

