--- ./src/lxc/lxc-shutdown.in.orig	2013-04-15 07:43:04.709177850 +0000
+++ ./src/lxc/lxc-shutdown.in	2013-04-15 07:45:43.213958405 +0000
@@ -118,11 +118,21 @@
     exit 1
 fi
 
+signal_reboot=INT
+signal_poweroff=PWR
+init_exe=$(readlink -f /proc/$pid/exe)
+case ${init_exe} in
+    */busybox)
+        signal_reboot=TERM
+        signal_poweroff=USR2
+       ;;
+esac
+
 if [ $reboot -eq 1 ]; then
-    kill -s INT $pid
+    kill -s $signal_reboot $pid
     exit 0
 else
-    kill -s PWR $pid
+    kill -s $signal_poweroff $pid
 fi
 
 if [ $dowait -eq 0 ]; then
