diff --git a/pypi_upload/setup.py b/pypi_upload/setup.py
index cd56011..427bcc9 100644
--- a/pypi_upload/setup.py
+++ b/pypi_upload/setup.py
@@ -6,8 +6,8 @@ from pathlib import Path
 from setuptools import find_packages, setup
 from setuptools.command.install import install
 
-project_root = Path(__file__).parents[1]
-
+import pathlib
+project_root = pathlib.Path().resolve()
 
 class VerifyCommand(install):
     """Custom command to verify module integrity"""
