--- anki-2.1.49/repos.bzl
+++ anki-2.1.49_local_ftl_repos/repos.bzl
@@ -126,48 +126,12 @@
 
     # translations
     ################
-
-    core_i18n_repo = "anki-core-i18n"
-    core_i18n_commit = "56d46d658a8badb9f13f48d89ee074419b181f69"
-    core_i18n_zip_csum = "8a1e83cc350364ef71500e6f3db9bdd2494832e0b9b1e61b0e2d93487171ca46"
-
-    qtftl_i18n_repo = "anki-desktop-ftl"
-    qtftl_i18n_commit = "5359f84328c1484c5db69edbb6b851acc4160095"
-    qtftl_i18n_zip_csum = "aa7c9ea94d05d2996eb02c198f228cc2bb171ebcda6ac802d7793027191ac73d"
-
-    i18n_build_content = """
-filegroup(
-    name = "files",
-    srcs = glob(["**/*.ftl"]),
-    visibility = ["//visibility:public"],
-)
-exports_files(["l10n.toml"])
-"""
-
-    maybe(
-        http_archive,
+    native.local_repository(
         name = "rslib_ftl",
-        build_file_content = i18n_build_content,
-        strip_prefix = core_i18n_repo + "-" + core_i18n_commit,
-        urls = [
-            "https://github.com/ankitects/{}/archive/{}.zip".format(
-                core_i18n_repo,
-                core_i18n_commit,
-            ),
-        ],
-        sha256 = core_i18n_zip_csum,
+        path = "ftlrslib",
     )
 
-    maybe(
-        http_archive,
+    native.local_repository(
         name = "extra_ftl",
-        build_file_content = i18n_build_content,
-        strip_prefix = qtftl_i18n_repo + "-" + qtftl_i18n_commit,
-        urls = [
-            "https://github.com/ankitects/{}/archive/{}.zip".format(
-                qtftl_i18n_repo,
-                qtftl_i18n_commit,
-            ),
-        ],
-        sha256 = qtftl_i18n_zip_csum,
+        path = "ftlextra",
     )
