Use the local version of protoc.

--- /dev/null
+++ anki-2.1.49_local_protoc/local_protoc/BUILD.bazel
@@ -0,0 +1,1 @@
+exports_files(["bin/protoc"])
--- anki-2.1.49/proto/protobuf.bzl
+++ anki-2.1.49_local_protoc/proto/protobuf.bzl
@@ -14,8 +14,8 @@
     actual = select({
         "@ankidesktop//platforms:windows_x86_64": "@protoc_bin_windows//:bin/protoc.exe",
         "@ankidesktop//platforms:macos_x86_64": "@protoc_bin_macos//:bin/protoc",
-        "@ankidesktop//platforms:linux_x86_64": "@protoc_bin_linux_x86_64//:bin/protoc",
-        "@ankidesktop//platforms:linux_arm64": "@protoc_bin_linux_arm64//:bin/protoc"
+        "@ankidesktop//platforms:linux_x86_64": "@local_protoc//:bin/protoc",
+        "@ankidesktop//platforms:linux_arm64": "@local_protoc//:bin/protoc"
     }),
     visibility = ["//visibility:public"]
 )
@@ -68,6 +68,11 @@
         build_file_content = """exports_files(["bin/protoc.exe"])""",
     )
 
+    native.local_repository(
+        name = "local_protoc",
+        path = "local_protoc",
+    )
+
     if not native.existing_rule(name):
         _setup_protoc(
             name = name,
