diff --color -Nurp anki-2.1.49/pylib/anki/BUILD.bazel anki-2.1.49_musllinux/pylib/anki/BUILD.bazel
--- anki-2.1.49/pylib/anki/BUILD.bazel	2021-10-26 02:53:24.000000000 +0200
+++ anki-2.1.49_musllinux/pylib/anki/BUILD.bazel	2022-01-04 11:52:42.566145221 +0100
@@ -73,8 +73,8 @@ py_wheel(
     platform = select({
         "//platforms:windows_x86_64": "win_amd64",
         "//platforms:macos_x86_64": "macosx_10_7_x86_64",
-        "//platforms:linux_x86_64": "manylinux2014_x86_64",
-        "//platforms:linux_arm64": "manylinux2014_aarch64",
+        "//platforms:linux_x86_64": "musllinux_1_2_x86_64",
+        "//platforms:linux_arm64": "musllinux_1_2_aarch64",
     }),
     python_tag = "cp38",
     python_version = ">=3.8",
