diff --git a/LuaXML_lib.c b/LuaXML_lib.c
index 62a2dc6..70c432a 100644
--- a/LuaXML_lib.c
+++ b/LuaXML_lib.c
@@ -322,9 +322,17 @@ int Xml_eval(lua_State *L) {
 		while(((token = Tokenizer_next(tok))!=0)&&(token[0]!=CLS)&&(token[0]!=ESC)) { // parse tag header
 			size_t sepPos=find(token, "=", 0);
 			if(token[sepPos]) { // regular attribute
-				const char* aVal =token+sepPos+2;
+				const char* aVal =token+sepPos+1;
 				lua_pushlstring(L, token, sepPos);
-				size_t lenVal = strlen(aVal)-1;
+				size_t lenVal = strlen(aVal);
+
+				if (lenVal < 2 || ((aVal[0] != '"' && aVal[0] != '\'') || (aVal[lenVal-1] != '"' && aVal[lenVal-1] != '\'')))
+					luaL_error(L, "Malformed XML: attribute value not quoted in '%s'", token);
+
+				// strip quote chars
+				aVal++;
+				lenVal -= 2;
+
 				if(!lenVal) Xml_pushDecode(L, "", 0);
 				else Xml_pushDecode(L, aVal, lenVal);
 				lua_settable(L, -3);
