diff -upr dwm-6.1.orig/config.mk dwm-6.1/config.mk
--- dwm-6.1.orig/config.mk	2015-11-09 11:40:01.970911763 +0100
+++ dwm-6.1/config.mk	2015-11-09 11:40:26.330814681 +0100
@@ -17,8 +17,6 @@ XINERAMAFLAGS = -DXINERAMA
 # freetype
 FREETYPELIBS = -lfontconfig -lXft
 FREETYPEINC = /usr/include/freetype2
-# OpenBSD (uncomment)
-FREETYPEINC = ${X11INC}/freetype2
 
 # includes and libs
 INCS = -I${X11INC} -I${FREETYPEINC}
