diff --git a/test-suite/tests/filesys.test b/test-suite/tests/filesys.test
index 45e77c8..a887109 100644
--- a/test-suite/tests/filesys.test
+++ b/test-suite/tests/filesys.test
@@ -457,19 +457,6 @@
         (pass-if-equal "file name of symlink" (test-file)
           (readlink (test-symlink)))
 
-        (pass-if-equal "port representing a symlink" (test-file)
-          (let ()
-            (unless (and (provided? 'readlink-port)
-                         (defined? 'O_NOFOLLOW)
-                         (defined? 'O_PATH)
-                         (not (= 0 O_NOFOLLOW))
-                         (not (= 0 O_PATH)))
-              (throw 'unsupported))
-            (define port (open (test-symlink) (logior O_NOFOLLOW O_PATH)))
-            (define points-to (false-if-exception (readlink port)))
-            (close-port port)
-            points-to))
-
         (pass-if-exception "not a port or file name" exception:wrong-type-arg
           (readlink '(stuff)))))
 
