diff --git a/setup.py b/setup.py
index 08571c8..063cc28 100755
--- a/setup.py
+++ b/setup.py
@@ -48,6 +48,7 @@
     )
 )
 _plat = sys.platform.lower()
+_arch = platform.machine()
 is_macos = 'darwin' in _plat
 is_openbsd = 'openbsd' in _plat
 is_freebsd = 'freebsd' in _plat
@@ -55,6 +56,7 @@
 is_dragonflybsd = 'dragonfly' in _plat
 is_bsd = is_freebsd or is_netbsd or is_dragonflybsd or is_openbsd
 is_arm = platform.processor() == 'arm' or platform.machine() == 'arm64'
+is_ppc64le = 'ppc64le' in _arch
 Env = glfw.Env
 env = Env()
 PKGCONFIG = os.environ.get('PKGCONFIG_EXE', 'pkg-config')
@@ -343,6 +345,8 @@ def init_env(
         # see https://github.com/kovidgoyal/kitty/issues/3126
         # -march=native is not supported when targeting Apple Silicon
         native_optimizations = False
+    if native_optimizations and is_ppc64le:
+        native_optimizations = False
     cc, ccver = cc_version()
     print('CC:', cc, ccver)
     stack_protector = first_successful_compile(cc, '-fstack-protector-strong', '-fstack-protector')
