--- a/setup.py
+++ b/setup.py
@@ -1393,24 +1393,6 @@ def create_linux_bundle_gunk(ddir: str,
     base = Path(ddir)
     in_src_launcher = base / (f'{libdir_name}/kitty/kitty/launcher/kitty')
     launcher = base / 'bin/kitty'
-    skip_docs = False
-    if not os.path.exists('docs/_build/html'):
-        kitten_exe = os.path.join(os.path.dirname(str(launcher)), 'kitten')
-        if os.path.exists(kitten_exe):
-            os.environ['KITTEN_EXE_FOR_DOCS'] = kitten_exe
-            make = 'gmake' if is_freebsd else 'make'
-            run_tool([make, 'docs'])
-        else:
-            if args.skip_building_kitten:
-                skip_docs = True
-                print('WARNING: You have chosen to skip building kitten.'
-                      ' This means docs could not be generated and will not be included in the linux package.'
-                      ' You should build kitten and then re-run this build.', file=sys.stderr)
-            else:
-                raise SystemExit(f'kitten binary not found at: {kitten_exe}')
-    if not skip_docs:
-        copy_man_pages(ddir)
-        copy_html_docs(ddir)
     for (icdir, ext) in {'256x256': 'png', 'scalable': 'svg'}.items():
         icdir = os.path.join(ddir, 'share', 'icons', 'hicolor', icdir, 'apps')
         safe_makedirs(icdir)
