From afc7e4251ed224c992ccc06c951a15f2dc87b151 Mon Sep 17 00:00:00 2001
From: Gcec21 <161324219+Gcec21@users.noreply.github.com>
Date: Sat, 19 Oct 2024 03:15:38 +0200
Subject: [PATCH] Update meson.build

---
 src/panel/applets/trash/meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/panel/applets/trash/meson.build b/src/panel/applets/trash/meson.build
index a7b05bd8..34aea5d9 100644
--- a/src/panel/applets/trash/meson.build
+++ b/src/panel/applets/trash/meson.build
@@ -34,6 +34,7 @@ applet_trash_deps = [
     dep_gtk3,
     dep_notify,
     dep_peas,
+    link_libpanelenum,
     link_libpanelplugin,
 ]
 
-- 
2.49.0

