--- a/ycmd/tests/shutdown_test.py.orig
+++ b/ycmd/tests/shutdown_test.py
@@ -53,13 +53,15 @@
   def FromHandlerWithSubservers_test( self ):
     self.Start()
 
-    filetypes = [ 'cs',
+    filetypes = [
+                  'cs',
                   'go',
                   'java',
                   'javascript',
                   'python',
                   'typescript',
-                  'rust' ]
+                  'rust'
+                ]
     for filetype in filetypes:
       self.StartSubserverForFiletype( filetype )
     self.AssertServersAreRunning()
@@ -84,13 +86,15 @@
   def FromWatchdogWithSubservers_test( self ):
     self.Start( idle_suicide_seconds = 5, check_interval_seconds = 1 )
 
-    filetypes = [ 'cs',
+    filetypes = [
+                  'cs',
                   'go',
                   'java',
                   'javascript',
                   'python',
                   'typescript',
-                  'rust' ]
+                  'rust'
+                ]
     for filetype in filetypes:
       self.StartSubserverForFiletype( filetype )
     self.AssertServersAreRunning()
