See https://github.com/golang/go/issues/37331

diff -upr go.orig/src/runtime/gc_test.go go/src/runtime/gc_test.go
--- go.orig/src/runtime/gc_test.go	2020-06-01 00:59:57.179164980 +0200
+++ go/src/runtime/gc_test.go	2020-06-01 01:00:52.282339799 +0200
@@ -15,9 +15,11 @@ import (
 	"testing"
 	"time"
 	"unsafe"
+	"internal/testenv"
 )
 
 func TestGcSys(t *testing.T) {
+	testenv.SkipFlaky(t, 37331)
 	if os.Getenv("GOGC") == "off" {
 		t.Skip("skipping test; GOGC=off in environment")
 	}
