diff --git a/src/internal/cpu/cpu_riscv64_linux.go b/src/internal/cpu/cpu_riscv64_linux.go
index b67bdf5876..59cba824ce 100644
--- a/src/internal/cpu/cpu_riscv64_linux.go
+++ b/src/internal/cpu/cpu_riscv64_linux.go
@@ -83,7 +83,7 @@
 
 	if pairs[0].key != -1 {
 		v := uint(pairs[0].value)
-		RISCV64.HasV = isSet(v, riscv_HWPROBE_IMA_V)
+		RISCV64.HasV = false // isSet(v, riscv_HWPROBE_IMA_V)
 		RISCV64.HasZbb = isSet(v, riscv_HWPROBE_EXT_ZBB)
 	}
 	if pairs[1].key != -1 {
