See https://github.com/golang/go/issues/20072

diff --git a/src/sync/waitgroup_test.go b/src/sync/waitgroup_test.go
index 4ab438c..5f4fc80 100644
--- a/src/sync/waitgroup_test.go
+++ b/src/sync/waitgroup_test.go
@@ -10,6 +10,7 @@ import (
 	. "sync"
 	"sync/atomic"
 	"testing"
+	"internal/testenv"
 )
 
 func testWaitGroup(t *testing.T, wg1 *WaitGroup, wg2 *WaitGroup) {
@@ -84,6 +85,7 @@ func pollUntilEqual(v *uint32, target uint32) {
 }
 
 func TestWaitGroupMisuse2(t *testing.T) {
+	testenv.SkipFlaky(t, 20072)
 	knownRacy(t)
 	if runtime.NumCPU() <= 4 {
 		t.Skip("NumCPU<=4, skipping: this test requires parallelism")
