See https://github.com/golang/go/issues/51787#issuecomment-1072700785

--- go.orig/src/cmd/link/internal/ld/config.go
+++ go/src/cmd/link/internal/ld/config.go
@@ -212,7 +212,7 @@
 		// windows/arm64 internal linking is not implemented.
 		return true, buildcfg.GOOS + "/" + buildcfg.GOARCH + " does not support internal cgo"
 	}
-	if iscgo && ctxt.Arch == sys.ArchPPC64 {
+	if iscgo && (ctxt.Arch == sys.ArchPPC64 || ctxt.Arch == sys.ArchPPC64LE) {
 		// Big Endian PPC64 cgo internal linking is not implemented for aix or linux.
 		return true, buildcfg.GOOS + " does not support internal cgo"
 	}
