--- a/build/Makefile.global
+++ b/build/Makefile.global
@@ -147,7 +147,7 @@
 	@if test -e "$(top_srcdir)/build/gen_stub.php"; then \
 		if test ! -z "$(PHP_EXECUTABLE)" && test -x "$(PHP_EXECUTABLE)"; then \
 			echo Parse $< to generate $@;\
-			$(PHP_EXECUTABLE) $(top_srcdir)/build/gen_stub.php $<; \
+			$(PHP_EXECUTABLE) -dextension=$(top_srcdir)/modules/tokenizer.so $(top_srcdir)/build/gen_stub.php $<; \
 		elif type php >/dev/null 2>/dev/null; then \
 			if test `php -v | head -n1 | cut -d" " -f 2 | sed "s/$$/\n7.0.99/" | sort -rV | head -n1` != "7.0.99"; then \
 				echo Parse $< to generate $@;\
