fix for fmt 9

diff --git a/src/mon/LogMonitor.h b/src/mon/LogMonitor.h
index 04d63c5..64dc076 100644
--- a/src/mon/LogMonitor.h
+++ b/src/mon/LogMonitor.h
@@ -19,6 +19,7 @@
 #include <map>
 #include <set>
 
+#define FMT_DEPRECATED_OSTREAM
 #include <fmt/format.h>
 #include <fmt/ostream.h>
 
diff --git a/src/tools/neorados.cc b/src/tools/neorados.cc
index 516dfbc..9587acb 100644
--- a/src/tools/neorados.cc
+++ b/src/tools/neorados.cc
@@ -14,6 +14,7 @@
  */
 
 #define BOOST_COROUTINES_NO_DEPRECATION_WARNING
+#define FMT_DEPRECATED_OSTREAM
 
 #include <algorithm>
 #include <cassert>
