diff --git a/src/common/async/bind_like.h b/src/common/async/bind_like.h
index a7fdbb869..a647869d5 100644
--- a/src/common/async/bind_like.h
+++ b/src/common/async/bind_like.h
@@ -15,9 +15,9 @@
 
 #include <boost/asio/associated_allocator.hpp>
 #include <boost/asio/associated_executor.hpp>
+#include <boost/asio/bind_allocator.hpp>
 #include <boost/asio/bind_executor.hpp>
 
-#include "common/async/bind_allocator.h"
 
 namespace ceph::async {
 template<typename Executor, typename Allocator, typename Completion>
