--- a/src/ceph_fuse.cc
+++ b/src/ceph_fuse.cc
@@ -223,7 +223,7 @@
 
 	  char buf[5050];
 	  string mountpoint = cfuse->get_mount_point();
-	  snprintf(buf, sizeof(buf), "fusermount -u -z %s", mountpoint.c_str());
+	  snprintf(buf, sizeof(buf), "fusermount3 -u -z %s", mountpoint.c_str());
 	  int umount_r = system(buf);
 	  if (umount_r) {
 	    if (umount_r != -1) {
--- a/qa/tasks/cephfs/fuse_mount.py
+++ b/qa/tasks/cephfs/fuse_mount.py
@@ -312,9 +312,9 @@
             return
 
         try:
-            log.info('Running fusermount -u on {name}...'.format(name=self.client_remote.name))
+            log.info('Running fusermount3 -u on {name}...'.format(name=self.client_remote.name))
             stderr = StringIO()
-            self.client_remote.run(args=['sudo', 'fusermount', '-u',
+            self.client_remote.run(args=['sudo', 'fusermount3', '-u',
                                          self.hostfs_mntpt],
                                    stderr=stderr,
                                    timeout=(30*60), omit_sudo=False)
