--- sqm-scripts-1.0.6/src/defaults.sh.orig
+++ sqm-scripts-1.0.6/src/defaults.sh
@@ -27,7 +27,7 @@
 [ -z "$IP" ] && IP=ip_wrapper
 [ -z "$IP_BINARY" ] && IP_BINARY=$(which ip)
 # Try modprobe first, fall back to insmod
-[ -z "$INSMOD" ] && INSMOD=$(which modprobe) || INSMOD=$(which insmod)
+[ -n "$INSMOD" ] || INSMOD=$(which modprobe) || INSMOD=$(which insmod)
 [ -z "$TARGET" ] && TARGET="5ms"
 [ -z "$IPT_MASK" ] && IPT_MASK="0xff"
 [ -z "$IPT_MASK_STRING" ] && IPT_MASK_STRING="/${IPT_MASK}"	# for set-mark
