--- a/src/defaults.sh
+++ b/src/defaults.sh
@@ -27,7 +27,7 @@
 [ -z "$IP" ] && IP=ip_wrapper
 [ -z "$IP_BINARY" ] && IP_BINARY=$(which ip)
 # Try modprobe first, fall back to insmod
-[ -z "$INSMOD" ] && INSMOD=$(which modprobe) || INSMOD=$(which insmod)
+[ -n "$INSMOD" ] || INSMOD=$(which modprobe) || INSMOD=$(which insmod)
 [ -z "$TARGET" ] && TARGET="5ms"
 [ -z "$IPT_MASK" ] && IPT_MASK="0xff" # to disable: set mask to 0xffffffff
 #sm: we need the functions above before trying to set the ingress IFB device
