--- a/Makefile
+++ b/Makefile
@@ -20,13 +20,8 @@
 	install -m 0744 src/run-openwrt.sh $(DESTDIR)$(PREFIX)/lib/sqm/run.sh
 
 install-linux: install-lib
-	install -m 0755 -d $(DESTDIR)$(PREFIX)/lib/systemd/system \
-		$(DESTDIR)$(PREFIX)/lib/tmpfiles.d $(DESTDIR)$(PREFIX)/bin
+	install -m 0755 -d $(DESTDIR)$(PREFIX)/bin
 	install -m 0644  platform/linux/eth0.iface.conf.example $(DESTDIR)/etc/sqm
-	install -m 0644  platform/linux/sqm@.service \
-		$(DESTDIR)$(PREFIX)/lib/systemd/system
-	install -m 0644  platform/linux/sqm-tmpfiles.conf \
-		$(DESTDIR)$(PREFIX)/lib/tmpfiles.d/sqm.conf
 	install -m 0755 platform/linux/sqm-bin $(DESTDIR)$(PREFIX)/bin/sqm
 	test -d $(DESTDIR)/etc/network/if-up.d && install -m 0755 platform/linux/sqm-ifup \
 		$(DESTDIR)/etc/network/if-up.d/sqm || exit 0
