--- a/tests/php8/methods/__construct.phpt
+++ b/tests/php8/methods/__construct.phpt
@@ -136,6 +136,8 @@
 C Decimal\Decimal::__construct() expected parameter 1 to be a string, integer, or decimal, float given
 D Decimal\Decimal::__construct() expected parameter 1 to be a string, integer, or decimal, null given
 E Decimal\Decimal::__construct(): Argument #2 ($precision) must be of type int, string given
+
+Deprecated: Decimal\Decimal::__construct(): Passing null to parameter #2 ($precision) of type int is deprecated in %s on line %d
 F Decimal precision out of range
 G Decimal precision out of range
 H Decimal precision out of range
