--- xdp-tools-1.3.1/xdp-trafficgen/xdp-trafficgen.c	2023-02-23 18:13:33.000000000 +0100
+++ xdp-tools-1.3.1/xdp-trafficgen/xdp-trafficgen.c	2023-04-25 22:11:08.445697391 +0200
@@ -790,7 +790,7 @@
 	}
 
 	sockaddr_sz = sizeof(local_saddr);
-	err = getsockname(sock, &local_saddr, &sockaddr_sz);
+	err = getsockname(sock, (struct sockaddr *)&local_saddr, &sockaddr_sz);
 	if (err) {
 		err = -errno;
 		pr_warn("Couldn't get local address: %s\n", strerror(-err));
