diff --git a/pyproject.toml b/pyproject.toml
index 7cd5a04..2312545 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,7 +11,6 @@ license = "AGPL-3.0-or-later"
 readme = "README.md"
 homepage = "https://github.com/greenbone/ospd-openvas"
 repository = "https://github.com/greenbone/ospd-openvas"
-# Full list: https://pypi.org/pypi?%3Aaction=list_classifiers
 classifiers=[
   "Development Status :: 4 - Beta",
   "License :: OSI Approved :: GNU Affero General Public License v3 or later (AGPLv3+)",
