--- a/rebar.config
+++ b/rebar.config
@@ -21,8 +21,6 @@
 {deps, [{if_not_rebar3, %% Needed because modules are not fully migrated to new structure and mix
          {if_var_true, elixir,
           {rebar_elixir_plugin, ".*", {git, "https://github.com/processone/rebar_elixir_plugin", "0.1.0"}}}},
-        {if_var_true, elixir,
-         {elixir, ".*", {git, "https://github.com/elixir-lang/elixir", {tag, "v1.4.4"}}}},
         {base64url, ".*", {git, "https://github.com/dvv/base64url", {tag, "1.0.1"}}},
         {cache_tab, ".*", {git, "https://github.com/processone/cache_tab", {tag, "1.0.30"}}},
         {eimp, ".*", {git, "https://github.com/processone/eimp", {tag, "1.0.22"}}},
--- a/ejabberdctl.template
+++ b/ejabberdctl.template
@@ -84,7 +84,7 @@
 [ -n "$ERL_DIST_PORT" ] && ERLANG_OPTS="$ERLANG_OPTS -erl_epmd_port $ERL_DIST_PORT -start_epmd false"
 # if vm.args file exists in config directory, pass it to Erlang VM
 [ -f "$VMARGS" ] && ERLANG_OPTS="$ERLANG_OPTS -args_file $VMARGS"
-ERL_LIBS='{{libdir}}'
+ERL_LIBS='{{libdir}}:/usr/lib/elixir/lib'
 ERL_CRASH_DUMP="$LOGS_DIR"/erl_crash_$(date "+%Y%m%d-%H%M%S").dump
 ERL_INETRC="$CONFIG_DIR"/inetrc
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -321,6 +321,7 @@
 		-e "s*{{config_dir}}*${ETCDIR}*g" \
 		-e "s*{{logs_dir}}*${LOGDIR}*g" \
 		-e "s*{{spool_dir}}*${SPOOLDIR}*g" \
+		-e "s*{{bindir}}/iex*$(iexpath)*g" \
 		-e "s*{{bindir}}*@bindir@*g" \
 		-e "s*{{libdir}}*@libdir@*g" \
 		-e "s*{{erl}}*@ERL@*g" \
