--- cowdancer-0.63.orig/Makefile
+++ cowdancer-0.63/Makefile
@@ -1,5 +1,5 @@
 SHELL=/bin/bash
-BINARY=libcowdancer.so cow-shell cowbuilder qemubuilder cowdancer-ilistcreate \
+BINARY=libcowdancer.so cow-shell cowdancer-ilistcreate \
 	cowdancer-ilistdump
 INSTALL_DIR=install -d -o root -g root -m 755
 INSTALL_FILE=install -o root -g root -m 644
@@ -23,19 +23,11 @@
 	$(INSTALL_FILE)  cow-shell.1 $(DESTDIR)/usr/share/man/man1/cow-shell.1
 	$(INSTALL_FILE)  cowdancer-ilistcreate.1 $(DESTDIR)/usr/share/man/man1/cowdancer-ilistcreate.1
 	$(INSTALL_FILE)  cowdancer-ilistdump.1 $(DESTDIR)/usr/share/man/man1/cowdancer-ilistdump.1
-	$(INSTALL_FILE)  cowbuilder.8 $(DESTDIR)/usr/share/man/man8/cowbuilder.8
-	$(INSTALL_FILE)  qemubuilder.8 $(DESTDIR)/usr/share/man/man8/qemubuilder.8
 	$(INSTALL_FILE)  libcowdancer.so $(DESTDIR)${LIBDIR}/cowdancer/libcowdancer.so
 	$(INSTALL_PROGRAM) cow-shell $(DESTDIR)/usr/bin/cow-shell
-	$(INSTALL_PROGRAM) cowbuilder $(DESTDIR)/usr/sbin/cowbuilder
-	$(INSTALL_PROGRAM) qemubuilder $(DESTDIR)/usr/sbin/qemubuilder
 	$(INSTALL_PROGRAM) cowdancer-ilistcreate $(DESTDIR)/usr/bin/cowdancer-ilistcreate
 	$(INSTALL_PROGRAM) cowdancer-ilistdump $(DESTDIR)/usr/bin/cowdancer-ilistdump
 
-	$(INSTALL_DIR) $(DESTDIR)/etc/bash_completion.d
-	$(INSTALL_FILE) bash_completion.qemubuilder $(DESTDIR)/etc/bash_completion.d/qemubuilder
-	$(INSTALL_FILE) bash_completion.cowbuilder $(DESTDIR)/etc/bash_completion.d/cowbuilder
-
 libcowdancer.so: cowdancer.lo ilistcreate.lo
 	$(CC) $(CFLAGS) -ldl -shared -o $@ $^
 
@@ -43,12 +35,6 @@
 	$(CC) $(CFLAGS) -o $@ $^
 
 cowdancer-ilistcreate: cowdancer-ilistcreate.o ilistcreate.o
-	$(CC) $(CFLAGS) -o $@ $^
-
-cowbuilder: cowbuilder.o parameter.o forkexec.o ilistcreate.o main.o
-	$(CC) $(CFLAGS) -o $@ $^
-
-qemubuilder: qemubuilder.lfso parameter.lfso forkexec.lfso qemuipsanitize.lfso qemuarch.lfso file.lfso main.lfso
 	$(CC) $(CFLAGS) -o $@ $^
 
 %.lo: %.c 
