diff -ru /src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in /src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in
--- /src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:05:19.092517025 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in /src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in
--- /src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_2.0_to_3.0.sh.in	2017-08-01 13:05:27.885738379 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in /src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in
--- /src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_3.0_to_4.0.sh.in	2017-08-01 13:11:20.277913463 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in /src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in
--- /src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_4.0_to_4.1.sh.in	2017-08-01 13:11:31.137774925 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in /src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in
--- /src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_4.1_to_5.0.sh.in	2017-08-01 13:11:37.317696084 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in /src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in
--- /src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_5.0_to_5.1.sh.in	2017-08-01 13:11:43.007623495 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`

diff -ru /src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in /src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in
--- /src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_5.1_to_5.2.sh.in	2017-08-01 13:11:43.007623495 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`

diff -ru /src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in /src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in
--- /src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in	2017-08-01 13:03:52.823615380 +0200
+++ /src/share/database/scripts/mysql/upgrade_5.2_to_6.0.sh.in	2017-08-01 13:11:43.007623495 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`mysql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in /src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in
--- /src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:04:53.279512387 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in /src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in
--- /src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_2.0_to_3.0.sh.in	2017-08-01 13:11:52.627500763 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in /src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in
--- /src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_3.0_to_3.1.sh.in	2017-08-01 13:11:59.840742064 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in /src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in
--- /src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_3.1_to_3.2.sh.in	2017-08-01 13:11:52.627500763 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in /src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in
--- /src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_3.2_to_3.3.sh.in	2017-08-01 13:11:52.627500763 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`
 
diff -ru /src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in /src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in
--- /src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/pgsql/upgrade_3.3_to_4.0.sh.in	2017-08-01 13:11:52.627500763 +0200
@@ -1,12 +1,7 @@
 #!/bin/sh
 
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh
 
 VERSION=`pgsql_version "$@"`

diff -ru /src/share/database/scripts/cql/upgrade_1.0_to_2.0.sh.in /src/share/database/scripts/cql/upgrade_1.0_to_2.0.sh.in
--- /src/share/database/scripts/cql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:03:52.826948671 +0200
+++ /src/share/database/scripts/cql/upgrade_1.0_to_2.0.sh.in	2017-08-01 13:11:52.627500763 +0200
@@ -1,13 +1,9 @@
 #!/bin/sh

 prefix=@prefix@
-# Include utilities. Use installed version if available and
-# use build version if it isn't.
-if [ -e @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh ]; then
-    . @datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh
-else
-    . @abs_top_builddir@/src/bin/admin/admin-utils.sh
-fi
+
+# Include utilities.
+. /usr/share/kea/scripts/admin-utils.sh

 version=$(cql_version "$@")
