diff --git a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
index ff0b686..5ceb1ed 100644
--- a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
+++ b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
@@ -51,7 +51,6 @@
 #include <signal.h>
 #include <unistd.h>  // fork
 #include <algorithm>
-#include <execinfo.h> // backtrace
 
 const char           *MosUtilitiesSpecificNext::m_szUserFeatureFile     = USER_FEATURE_FILE;
 MOS_PUF_KEYLIST      MosUtilitiesSpecificNext::m_ufKeyList              = nullptr;
@@ -2498,7 +2497,7 @@ void MosUtilities::MosTraceEvent(
             // max 32-2=30 layers call stack in 64bit driver.
             uint32_t nLen = 4*sizeof(uint32_t);
             void **stack = (void **)(traceBuf + nLen);
-            int num = backtrace(stack, ((sizeof(traceBuf)-nLen)/sizeof(void *)));
+            int num = 0;
             if (num > 0)
             {
                 uint32_t *header = (uint32_t *)traceBuf;
