diff --git a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
index bd1c5d1..82c8d35 100644
--- a/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
+++ b/media_softlet/linux/common/os/osservice/mos_utilities_specific.cpp
@@ -34,7 +34,6 @@
 #include <signal.h>
 #include <unistd.h>  // fork
 #include <algorithm>
-#include <execinfo.h> // backtrace
 #include <sys/types.h>
 #include <sys/stat.h>  // fstat
 #include <sys/ipc.h>  // System V IPC
@@ -2475,7 +2474,7 @@ void MosUtilities::MosTraceEvent(
             // max 32-2=30 layers call stack in 64bit driver.
             uint32_t nLen = 4*sizeof(uint32_t);
             void **stack = (void **)(traceBuf + nLen);
-            int num = backtrace(stack, ((sizeof(traceBuf)-nLen)/sizeof(void *)));
+            int num = 0;
             if (num > 0)
             {
                 uint32_t *header = (uint32_t *)traceBuf;
