--- a/src/low_main.cpp
+++ b/src/low_main.cpp
@@ -78,9 +78,9 @@
     low->in_gc = false;
     low->disallow_native = false;
 
-    low->web_thread = NULL;
+    low->web_thread = 0;
     for(int i = 0; i < LOW_NUM_DATA_THREADS; i++)
-        low->data_thread[i] = NULL;
+        low->data_thread[i] = 0;
 
     low->destroying = false;
     low->duk_flag_stop = 0;
--- a/src/LowHTTPDirect.h
+++ b/src/LowHTTPDirect.h
@@ -8,6 +8,7 @@
 #include "LowLoopCallback.h"
 #include "LowSocketDirect.h"
 
+#include <ctype.h>
 #include <pthread.h>
 
 #include "low_config.h"
--- a/src/low_crypto.h
+++ b/src/low_crypto.h
@@ -5,6 +5,7 @@
 #ifndef __LOW_CRYPTO_H__
 #define __LOW_CRYPTO_H__
 
+#include <ctype.h>
 #include "duktape.h"
 
 duk_ret_t low_crypto_create_hash(duk_context *ctx);
--- a/src/low_opcua.h
+++ b/src/low_opcua.h
@@ -7,6 +7,7 @@
 #define __LOW_OPCUA_H__
 
 #include "duktape.h"
+#include <ctype.h>
 
 #include "low_main.h"
 #include "LowLoopCallback.h"

