--- a/src/LowHTTPDirect.h
+++ b/src/LowHTTPDirect.h
@@ -8,6 +8,7 @@
 #include "LowLoopCallback.h"
 #include "LowSocketDirect.h"
 
+#include <ctype.h>
 #include <pthread.h>
 
 #include "low_config.h"
@@ -116,4 +117,4 @@
     bool mReadError, mWriteError, mHTTPError;
 };
 
-#endif /* __LOWHTTPDIRECT_H__ */
\ No newline at end of file
+#endif /* __LOWHTTPDIRECT_H__ */
--- a/src/low_crypto.h
+++ b/src/low_crypto.h
@@ -5,6 +5,7 @@
 #ifndef __LOW_CRYPTO_H__
 #define __LOW_CRYPTO_H__
 
+#include <ctype.h>
 #include "duktape.h"
 
 duk_ret_t low_crypto_create_hash(duk_context *ctx);
@@ -15,4 +16,4 @@
 
 duk_ret_t low_crypto_random_bytes(duk_context *ctx);
 
-#endif /* __LOW_CRYPTO_H__ */
\ No newline at end of file
+#endif /* __LOW_CRYPTO_H__ */
--- a/src/low_opcua.h
+++ b/src/low_opcua.h
@@ -7,6 +7,7 @@
 #define __LOW_OPCUA_H__
 
 #include "duktape.h"
+#include <ctype.h>
 
 #include "low_main.h"
 #include "LowLoopCallback.h"
