--- a/src/Navaids/NavDataCache.cxx
+++ b/src/Navaids/NavDataCache.cxx
@@ -19,6 +19,7 @@
 #include <mutex>
 #include <utility>
 #include <vector>
+#include <fcntl.h>
 
 #ifdef SYSTEM_SQLITE
 // the standard sqlite3.h doesn't give a way to set SQLITE_UINT64_TYPE,

--- a/src/Main/MultipleInstanceLock.cxx	2026-05-22 11:44:57.034934856 +0200
+++ b/src/Main/MultipleInstanceLock.cxx	2026-05-22 11:46:18.143561600 +0200
@@ -6,6 +6,7 @@
 #include <cassert>
 #include <cstdlib>
 
+#include <fcntl.h>
 
 #include "MultipleInstanceLock.hxx"
 
