From: Peter Lemenkov <lemenkov@gmail.com>
Date: Sat, 19 Jun 2010 09:25:18 +0400
Subject: [PATCH] Do not install Java sources

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

--- otp_src_18.0-orig/lib/ic/java_src/com/ericsson/otp/ic/Makefile
+++ otp_src_18.0-fixed/lib/ic/java_src/com/ericsson/otp/ic/Makefile
@@ -113,8 +113,6 @@
 include $(ERL_TOP)/make/otp_release_targets.mk
 
 release_spec: opt
-	$(INSTALL_DIR) "$(RELSYSDIR)/java_src/com/ericsson/otp/ic"
-	$(INSTALL_DATA) $(JAVA_FILES) "$(RELSYSDIR)/java_src/com/ericsson/otp/ic"
 	$(INSTALL_DIR) "$(RELSYSDIR)/priv"
 	$(INSTALL_DATA) $(JAVA_DEST_ROOT)$(JARFILE) "$(RELSYSDIR)/priv"
 
--- otp_src_18.0-orig/lib/jinterface/java_src/com/ericsson/otp/erlang/Makefile
+++ otp_src_18.0-fixed/lib/jinterface/java_src/com/ericsson/otp/erlang/Makefile
@@ -120,8 +120,6 @@
 	$(V_at)$(MAKE) $(MFLAGS) RELEASE_PATH="$(RELEASE_PATH)" $(TARGET_MAKEFILE)  $@_spec
 
 release_spec: opt
-	$(V_at)$(INSTALL_DIR) "$(RELSYSDIR)/java_src/com/ericsson/otp/erlang"
-	$(V_at)$(INSTALL_DATA) $(JAVA_SRC) "$(RELSYSDIR)/java_src/com/ericsson/otp/erlang"
 	$(V_at)$(INSTALL_DIR) "$(RELSYSDIR)/priv"
 	$(V_at)$(INSTALL_DATA) $(JAVA_DEST_ROOT)$(JARFILE) "$(RELSYSDIR)/priv"
 	$(V_at)$(INSTALL_DIR) "$(RELSYSDIR)/ebin"
