diff -urN python-ssdeep-3.4.orig/setup.py python-ssdeep-3.4/setup.py
--- python-ssdeep-3.4.orig/setup.py	2021-07-26 23:50:49.492670491 -0600
+++ python-ssdeep-3.4/setup.py	2021-07-26 23:51:14.572803032 -0600
@@ -96,14 +96,7 @@
 with open(os.path.join(base_dir, "README.rst")) as f:
     long_description = f.read()
 
-# On some systems(e.g. Debian 8, CentOS 7) the setuptools package is very old.
-# We try to install an old version of pytest-runner without setuptools_scm dependency.
-# See: https://github.com/pytest-dev/pytest-runner/blob/master/CHANGES.rst
-if parse_version(setuptools.__version__) < parse_version("12"):
-    setup_requires = ["pytest-runner<2.4"]
-else:
-    setup_requires = ["pytest-runner"]
-
+setup_requires = []
 
 setup(
     name=about["__title__"],
