--- a/binding/python/team/capi.i.in
+++ b/binding/python/team/capi.i.in
@@ -40,7 +40,7 @@
     if ($1) free($1);
 }
 %typemap(argout) (char *addr, unsigned int addr_len) {
-    $result = SWIG_Python_AppendOutput($result, SWIG_FromCharPtrAndSize($1,$2));
+    $result = SWIG_AppendOutput($result, SWIG_FromCharPtrAndSize($1,$2));
 }
 %apply char *OUTPUT {char *addr};
 int team_hwaddr_get(struct team_handle *th, uint32_t ifindex, char *addr, unsigned int addr_len);
