From: Jakub Jirutka <jakub@jirutka.cz>
Date: Sat, 06 May 2023 19:59:29 +0200
Subject: [PATCH] Adjust default config for Alpine Linux

--- a/config.toml
+++ b/config.toml
@@ -1,27 +1,24 @@
-# This file shows the default configuration of Meilisearch.
 # All variables are defined here: https://docs.meilisearch.com/learn/configuration/instance_options.html#environment-variables
 
-db_path = "./data.ms"
+db_path = "/var/lib/meilisearch/data.ms"
 # Designates the location where database files will be created and retrieved.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#database-path
 
-env = "development"
+env = "production"
 # Configures the instance's environment. Value must be either `production` or `development`.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#environment
 
-http_addr = "localhost:7700"
+http_addr = "127.0.0.1:7700"
 # The address on which the HTTP server will listen.
 
 # master_key = "YOUR_MASTER_KEY_VALUE"
 # Sets the instance's master key, automatically protecting all routes except GET /health.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#master-key
+#
+# NOTE: If this option is not set, the init script writes a random key to file
+# /etc/meilisearch/master_key, if it doesn't already exist, and loads it into
+# env. variable MEILI_MASTER_KEY at each start.
 
-# no_analytics = true
-# Deactivates Meilisearch's built-in telemetry when provided.
-# Meilisearch automatically collects data from all instances that do not opt out using this flag.
-# All gathered data is used solely for the purpose of improving Meilisearch, and can be deleted at any time.
-# https://docs.meilisearch.com/learn/configuration/instance_options.html#disable-analytics
-
 http_payload_size_limit = "100 MB"
 # Sets the maximum size of accepted payloads.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#payload-limit-size
@@ -43,11 +40,11 @@
 ### DUMPS ###
 #############
 
-dump_dir = "dumps/"
+dump_dir = "/var/lib/meilisearch/dumps/"
 # Sets the directory where Meilisearch will create dump files.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#dump-directory
 
-# import_dump = "./path/to/my/file.dump"
+# import_dump = "/var/lib/meilisearch/dumps/file.dump"
 # Imports the dump file located at the specified path. Path must point to a .dump file.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#import-dump
 
@@ -70,11 +67,11 @@
 # between each snapshot, in seconds.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#schedule-snapshot-creation
 
-snapshot_dir = "snapshots/"
+snapshot_dir = "/var/lib/meilisearch/snapshots/"
 # Sets the directory where Meilisearch will store snapshots.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#snapshot-destination
 
-# import_snapshot = "./path/to/my/snapshot"
+# import_snapshot = "/var/lib/meilisearch/snapshots/my-snapshot"
 # Launches Meilisearch after importing a previously-generated snapshot at the given filepath.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#import-snapshot
 
@@ -91,19 +88,19 @@
 ### SSL ###
 ###########
 
-# ssl_auth_path = "./path/to/root"
+# ssl_auth_path = "/etc/meilisearch/client-ca.crt"
 # Enables client authentication in the specified path.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#ssl-authentication-path
 
-# ssl_cert_path = "./path/to/certfile"
+# ssl_cert_path = "/etc/meilisearch/server.crt"
 # Sets the server's SSL certificates.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#ssl-certificates-path
 
-# ssl_key_path = "./path/to/private-key"
+# ssl_key_path = "/etc/meilisearch/server.key"
 # Sets the server's SSL key files.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#ssl-key-path
 
-# ssl_ocsp_path = "./path/to/ocsp-file"
+# ssl_ocsp_path = "/etc/meilisearch/server.ocsp"
 # Sets the server's OCSP file.
 # https://docs.meilisearch.com/learn/configuration/instance_options.html#ssl-ocsp-path
 
