Don't bundle CA certificates, FFS!

--- a/meilisearch/Cargo.toml
+++ b/meilisearch/Cargo.toml
@@ -59 +59 @@
-reqwest = { version = "0.11.12", features = ["rustls-tls", "json"], default-features = false }
+reqwest = { version = "0.11.12", features = ["rustls-tls-native-roots", "json"], default-features = false }
--- a/vendor/actix-net/actix-tls/Cargo.toml
+++ b/vendor/actix-net/actix-tls/Cargo.toml
@@ -33,7 +33,7 @@
 openssl = ["tls-openssl", "tokio-openssl"]
 
 # use rustls impls
-rustls = ["tokio-rustls", "webpki-roots"]
+rustls = ["tokio-rustls"]
 
 # use native-tls impls
 native-tls = ["tokio-native-tls"]
