Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: install *.cmake files to /usr/lib; set SOVERSION
----

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@
   set(CMAKE_BUILD_TYPE Release)
 endif()

-project(libOpenDrive VERSION 0.6.0 DESCRIPTION ".xodr library")
+project(libOpenDrive VERSION 0.6.1 DESCRIPTION ".xodr library")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)

@@ -30,7 +30,7 @@
 FetchContent_MakeAvailable(pugixml)

 # Only include testing dependencies if BUILD_TESTING is ON and BUILD_SHARED_LIBS is OFF
-if(BUILD_TESTING AND NOT BUILD_SHARED_LIBS)
+if(BUILD_TESTING)
     include(CTest)
     FetchContent_Declare(
         Catch2
@@ -66,6 +66,9 @@
 )

 add_library(OpenDrive ${SOURCES})
+set_target_properties(OpenDrive PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR})
 target_include_directories(
     OpenDrive PUBLIC
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -74,7 +77,7 @@
 )

 # Only build tests if BUILD_TESTING is ON and BUILD_SHARED_LIBS is OFF
-if(BUILD_TESTING AND NOT BUILD_SHARED_LIBS)
+if(BUILD_TESTING)
     add_executable(tests tests/test.cpp)
     target_link_libraries(tests PRIVATE OpenDrive Catch2::Catch2WithMain)
     catch_discover_tests(tests WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests)
@@ -95,4 +98,4 @@
 )
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(TARGETS OpenDrive EXPORT OpenDriveConfig)
-install(EXPORT OpenDriveConfig NAMESPACE OpenDrive:: DESTINATION cmake)
\ No newline at end of file
+install(EXPORT OpenDriveConfig NAMESPACE OpenDrive:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake)
\ No newline at end of file
