diff -Nur libwmf-0.2.8.4.orig/src/extra/gd/gd_png.c libwmf-0.2.8.4/src/extra/gd/gd_png.c
--- libwmf-0.2.8.4.orig/src/extra/gd/gd_png.c	2005-07-27 23:35:06.000000000 +0300
+++ libwmf-0.2.8.4/src/extra/gd/gd_png.c	2010-01-17 01:33:58.000000000 +0200
@@ -136,7 +136,7 @@
   /* first do a quick check that the file really is a PNG image; could
    * have used slightly more general png_sig_cmp() function instead */
   gdGetBuf (sig, 8, infile);
-  if (!png_check_sig (sig, 8))
+  if (png_sig_cmp (sig, 0, 8))
     return NULL;		/* bad signature */
 
 #ifndef PNG_SETJMP_NOT_SUPPORTED
