From 7eabc78d6dc4fc8e5b96adeb80e6b2edc150b18e Mon Sep 17 00:00:00 2001
From: Joe Thornber <ejt@redhat.com>
Date: Tue, 1 Apr 2014 17:43:15 +0100
Subject: [PATCH] [era] qualify btiset

---
 era/writeset_tree.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/era/writeset_tree.cc b/era/writeset_tree.cc
index 763427e..2fe47fe 100644
--- a/era/writeset_tree.cc
+++ b/era/writeset_tree.cc
@@ -54,7 +54,7 @@
 
 		void visit(btree_path const &path, era_detail const &era) {
 			era_ = path[0];
-			bitset bs(tm_, era.writeset_root, era.nr_bits);
+			persistent_data::bitset bs(tm_, era.writeset_root, era.nr_bits);
 			writeset_v_.writeset_begin(era_, era.nr_bits);
 			bs.walk_bitset(*this);
 			writeset_v_.writeset_end();
-- 
1.9.1

