diff --git a/CMakeModules/FindSSE.cmake b/CMakeModules/FindSSE.cmake
index f6ad475..47f9583 100644
--- a/CMakeModules/FindSSE.cmake
+++ b/CMakeModules/FindSSE.cmake
@@ -82,7 +82,7 @@ ELSEIF(CMAKE_SYSTEM_NAME MATCHES "Windows")
    set(SSSE3_FOUND  false CACHE BOOL "SSSE3 available on host")
    set(SSE4_1_FOUND false CACHE BOOL "SSE4.1 available on host")
 ELSE(CMAKE_SYSTEM_NAME MATCHES "Linux")
-   set(SSE2_FOUND   true  CACHE BOOL "SSE2 available on host")
+   set(SSE2_FOUND   false CACHE BOOL "SSE2 available on host")
    set(SSE3_FOUND   false CACHE BOOL "SSE3 available on host")
    set(SSSE3_FOUND  false CACHE BOOL "SSSE3 available on host")
    set(SSE4_1_FOUND false CACHE BOOL "SSE4.1 available on host")
