diff --git a/Makefile b/Makefile
index 0940c49..7a47dc2 100644
--- a/Makefile
+++ b/Makefile
@@ -23,11 +23,11 @@ AR	= ar
 W	= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
 THREADS = -pthread
 OPT = -O2 -g
-CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS)
+CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS) -fPIC
 LDLIBS	=
 SOLIBS	=
 SOEXT	= .so
-prefix	= /usr/local
+prefix	= /usr
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 libdir = $(exec_prefix)/lib
