diff --git a/Makefile b/Makefile
index dbb5d69..149bdbf 100644
--- a/Makefile
+++ b/Makefile
@@ -23,11 +23,11 @@ AR	= ar
 W	= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
 THREADS = -pthread
 OPT = -O2 -g
-CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS)
+CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS) -fPIC
 LDLIBS	=
 SOLIBS	=
-prefix	= /usr/local
-mandir = $(prefix)/man
+prefix	= /usr
+mandir = $(prefix)/share/man
 
 ########################################################################
 
@@ -42,7 +42,7 @@ install: $(ILIBS) $(IPROGS) $(IHDRS)
 	mkdir -p $(DESTDIR)$(prefix)/bin
 	mkdir -p $(DESTDIR)$(prefix)/lib
 	mkdir -p $(DESTDIR)$(prefix)/include
-	mkdir -p $(DESTDIR)$(prefix)/man/man1
+	mkdir -p $(DESTDIR)$(mandir)/man1
 	for f in $(IPROGS); do cp $$f $(DESTDIR)$(prefix)/bin; done
 	for f in $(ILIBS); do cp $$f $(DESTDIR)$(prefix)/lib; done
 	for f in $(IHDRS); do cp $$f $(DESTDIR)$(prefix)/include; done
