diff --git a/src/plugin/plugin.cpp b/src/plugin/plugin.cpp
index aeb4600..12ad747 100644
--- a/src/plugin/plugin.cpp
+++ b/src/plugin/plugin.cpp
@@ -75,7 +75,7 @@ public:
         qDBusRegisterMetaType<NemoDeviceLock::Fingerprint>();
         qDBusRegisterMetaType<QVector<NemoDeviceLock::Fingerprint>>();
 
-        qmlRegisterType<NemoDeviceLock::FingerprintModel>();
+        qmlRegisterType<NemoDeviceLock::FingerprintModel>(uri, 1, 0, "FingerprintModel");
 
         qmlRegisterSingletonType<NemoDeviceLock::DeviceLock>(uri, 1, 0, "DeviceLock", createDeviceLock);
 
