diff -up bacula-5.0.2/src/lib/crypto.c.openssl bacula-5.0.2/bacula-5.0.2/src/lib/crypto.c
--- bacula-5.0.2/src/lib/crypto.c.openssl	2010-04-27 21:58:29.000000000 +0200
+++ bacula-5.0.2/src/lib/crypto.c	2010-06-01 17:51:52.846894465 +0200
@@ -51,7 +51,7 @@
  * For OpenSSL version 1.x, EVP_PKEY_encrypt no longer
  *  exists.  It was not an official API.
  */
-#ifdef HAVE_OPENSSLv1
+#if (OPENSSL_VERSION_NUMBER >= 0x10000000L)
 #define EVP_PKEY_encrypt EVP_PKEY_encrypt_old
 #define EVP_PKEY_decrypt EVP_PKEY_decrypt_old
 #endif
diff -up bacula-5.0.2/bacula-5.0.2/src/lib/tls.c.openssl bacula-5.0.2/bacula-5.0.2/src/lib/tls.c
--- bacula-5.0.2/src/lib/tls.c.openssl	2010-04-27 21:58:29.000000000 +0200
+++ bacula-5.0.2/src/lib/tls.c	2010-06-01 17:52:15.535769446 +0200
@@ -315,7 +315,7 @@ bool tls_postconnect_verify_host(JCR *jc
          extname = OBJ_nid2sn(OBJ_obj2nid(X509_EXTENSION_get_object(ext)));
 
          if (strcmp(extname, "subjectAltName") == 0) {
-#ifdef HAVE_OPENSSLv1
+#if (OPENSSL_VERSION_NUMBER >= 0x10000000L)
             const X509V3_EXT_METHOD *method;
 #else
             X509V3_EXT_METHOD *method;
