--- bacula-3.0.3-orig/autoconf/configure.in	Sun Oct 18 09:10:16 2009
+++ bacula-3.0.3/autoconf/configure.in	Wed Dec  2 14:13:44 2009
@@ -2576,6 +2576,20 @@
    fi
    TAPEDRIVE="/dev/nrmt0"
   ;;
+alpine)
+   dnl Make sure hostname is resolved
+   ping -c 1 $hostname 2>&1 1>/dev/null
+   if test ! $? = 0; then
+      hostname="localhost"
+   fi
+   if `test -f /etc/alpine-release && grep -q alpine /etc/alpine-release`; then
+      DISTNAME="alpine"
+   fi
+   DISTVER=`cat /etc/alpine-release`
+   TAPEDRIVE="/dev/nst0"
+   PSCMD="ps -e -o pid,comm"
+   largefile_support="yes"
+  ;;
 bsdi)
    DISTVER=`uname -a |awk '{print $3}'`
    TAPEDRIVE="/dev/nrmt0"
