From 6af21775b8c560dd9d65cb2465299a9feae34649 Mon Sep 17 00:00:00 2001
From: Paul J Stevens <paul@nfg.nl>
Date: Thu, 14 Apr 2011 14:01:32 +0000
Subject: cleanups

---
diff --git a/debian/rules b/debian/rules
index 7804a1b..f459149 100755
--- a/debian/rules
+++ b/debian/rules
@@ -12,7 +12,7 @@ export SHARED=--enable-static=no
 export MANPAGES=--enable-manpages=yes
 export REAL_CLEAN=
 
-CFLAGS = -Wall -O1
+CFLAGS = -Wall -std=gnu99 -O1
 LDFLAGS = -lcrypt
 
 # These are used for cross-compiling and for saving the configure script
@@ -27,10 +27,9 @@ ifneq (,$(findstring devel,$(DEB_BUILD_OPTIONS)))
 	export MANPAGES=
 	export SHARED=--enable-shared=no
 	export STATIC=--enable-static=yes
-	export REAL_CLEAN=mrproper
 	export WITH_LDAP=
 	export WITH_SIEVE=
-	CFLAGS = -g -std=gnu99
+	CFLAGS = -Wall -std=gnu99 -g
 endif
 ifneq (,$(findstring ldap,$(DEB_BUILD_OPTIONS)))
 	export WITH_LDAP=--with-ldap
diff --git a/src/dm_http.c b/src/dm_http.c
index 4b7d2a6..797e63b 100644
--- a/src/dm_http.c
+++ b/src/dm_http.c
@@ -193,7 +193,7 @@ void Http_getUsers(T R)
 		evbuffer_add_printf(buf, "\n}}\n");
 	} 
 
-	if (buf->totallen)
+	if (EVBUFFER_LENGTH(buf))
 		Request_send(R, HTTP_OK, "OK", buf);
 	else		
 		Request_error(R, HTTP_SERVUNAVAIL, "Server error");
@@ -279,7 +279,7 @@ void Http_getMailboxes(T R)
 		MailboxState_free(&b);
 	}
 
-	if (buf->totallen)
+	if (EVBUFFER_LENGTH(buf))
 		Request_send(R, HTTP_OK, "OK", buf);
 	else		
 		Request_error(R, HTTP_SERVUNAVAIL, "Server error");
@@ -375,7 +375,7 @@ void Http_getMessages(T R)
 		}
 	}
 
-	if (buf->totallen)
+	if (EVBUFFER_LENGTH(buf))
 		Request_send(R, HTTP_OK, "OK", buf);
 	else		
 		Request_error(R, HTTP_SERVUNAVAIL, "Server error");
--
cgit v0.8.3.4
