diff --git a/gmic-qt/gmic_qt.pro.orig b/gmic-qt/gmic_qt.pro
index c5db25c..d3995bf 100644
--- a/gmic-qt/gmic_qt.pro.orig
+++ b/gmic-qt/gmic_qt.pro
@@ -451,7 +451,7 @@ SOURCES += \
 
 equals(GMIC_DYNAMIC_LINKING, "on" )|equals(GMIC_DYNAMIC_LINKING, "ON" ) {
   message(Dynamic linking with libgmic)
-  LIBS += -Wl,-rpath,. $$GMIC_PATH/libgmic.so
+  LIBS += $$GMIC_PATH/libgmic.so
 }
 
 equals(GMIC_DYNAMIC_LINKING, "off" )|equals(GMIC_DYNAMIC_LINKING, "OFF" ) {
diff --git a/src/Makefile.orig b/src/Makefile
index 3e2c070..25f3bdf 100644
--- a/src/Makefile.orig
+++ b/src/Makefile
@@ -183,7 +183,7 @@ endif
 ifeq ($(BUILD_OS),Darwin)
 MANDATORY_LIBS += -L. -L$(PREFIX)/$(LIB)
 else ifneq ($(BUILD_OS),Windows)
-MANDATORY_LIBS += -L. -L$(PREFIX)/$(LIB) -Wl,-rpath,.
+MANDATORY_LIBS += -L. -L$(PREFIX)/$(LIB)
 else
 MANDATORY_LIBS += -L. -L$(PREFIX)/$(LIB)
 endif
 