--- a/cmake/Modules/FindLZ4_EP.cmake
+++ b/cmake/Modules/FindLZ4_EP.cmake
@@ -34,11 +34,6 @@
 # Include some common helper functions.
 include(TileDBCommon)
 
-if(TILEDB_VCPKG)
-  find_package(lz4 REQUIRED)
-  return()
-endif()
-
 # First check for a static version in the EP prefix.
 find_library(LZ4_LIBRARIES
   NAMES


--- a/cmake/Modules/FindZstd_EP.cmake
+++ b/cmake/Modules/FindZstd_EP.cmake
@@ -35,16 +35,6 @@
 # Include some common helper functions.
 include(TileDBCommon)
 
-if (TILEDB_VCPKG)
-  find_package(zstd CONFIG REQUIRED)
-  if (TARGET zstd::libzstd_static)
-    set(ZSTD_TARGET zstd::libzstd_static)
-  else()
-    set(ZSTD_TARGET zstd::libzstd_shared)
-  endif()
-  return()
-endif()
-
 # First check for a static version in the EP prefix.
 find_library(ZSTD_LIBRARIES
   NAMES
