diff --git a/cmake/Modules/FindZstd.cmake b/cmake/Modules/FindZstd.cmake
new file mode 100644
index 000000000..683c949fe
--- /dev/null
+++ b/cmake/Modules/FindZstd.cmake
@@ -0,0 +1,30 @@
+#
+# FindZstd.cmake
+#
+# This module defines:
+#   - ZSTD_INCLUDE_DIRS, the directory containing zstd.h
+#   - ZSTD_LIBRARIES, the zstd library path
+#   - zstd::libzstd, an imported target for the zstd library
+
+find_path(ZSTD_INCLUDE_DIRS
+  NAMES zstd.h
+  HINTS ${zstd_ROOT_DIR}/include)
+
+find_library(ZSTD_LIBRARIES
+  NAMES zstd
+  HINTS ${zstd_ROOT_DIR}/lib)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Zstd DEFAULT_MSG ZSTD_LIBRARIES ZSTD_INCLUDE_DIRS)
+
+mark_as_advanced(
+  ZSTD_LIBRARIES
+  ZSTD_INCLUDE_DIRS)
+
+if(Zstd_FOUND AND NOT TARGET zstd::libzstd)
+  add_library(zstd::libzstd UNKNOWN IMPORTED)
+  set_target_properties(zstd::libzstd PROPERTIES
+    IMPORTED_LOCATION "${ZSTD_LIBRARIES}"
+    INTERFACE_INCLUDE_DIRECTORIES "${ZSTD_INCLUDE_DIRS}"
+  )
+endif()
diff --git a/tiledb/CMakeLists.txt b/tiledb/CMakeLists.txt
index f72c71294..bf117c680 100644
--- a/tiledb/CMakeLists.txt
+++ b/tiledb/CMakeLists.txt
@@ -576,7 +576,7 @@ find_package(BZip2 REQUIRED)
 find_package(LZ4 MODULE REQUIRED)
 find_package(spdlog REQUIRED)
 find_package(ZLIB REQUIRED)
-find_package(zstd CONFIG REQUIRED)
+find_package(Zstd MODULE REQUIRED)
 target_link_libraries(TILEDB_CORE_OBJECTS_ILIB
   INTERFACE
     BZip2::BZip2
diff --git a/tiledb/sm/compressors/CMakeLists.txt b/tiledb/sm/compressors/CMakeLists.txt
index a7aa3f968..a53422f73 100644
--- a/tiledb/sm/compressors/CMakeLists.txt
+++ b/tiledb/sm/compressors/CMakeLists.txt
@@ -39,7 +39,7 @@ commence(object_library compressors)
     find_package(BZip2 REQUIRED)
     find_package(LZ4 MODULE REQUIRED)
     find_package(ZLIB REQUIRED)
-    find_package(zstd CONFIG REQUIRED)
+    find_package(Zstd MODULE REQUIRED)
     this_target_link_libraries(BZip2::BZip2 lz4::lz4 ZLIB::ZLIB zstd::libzstd)
 conclude(object_library)
 
