diff --git a/tiledb/sm/misc/test/unit_parse_argument.cc b/tiledb/sm/misc/test/unit_parse_argument.cc
index 867af4aaa..a0a883c11 100644
--- a/tiledb/sm/misc/test/unit_parse_argument.cc
+++ b/tiledb/sm/misc/test/unit_parse_argument.cc
@@ -30,7 +30,7 @@
  * Tests for useful (global) functions.
  */
 
-#include "catch.hpp"
+#include <test/support/tdb_catch.h>
 #include "tiledb/sm/enums/datatype.h"
 #include "tiledb/sm/misc/parse_argument.h"
 
