Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: relax version requirements for external libraries
----

--- a/cmake/Modules/FindCapnp_EP.cmake
+++ b/cmake/Modules/FindCapnp_EP.cmake
@@ -56,7 +56,7 @@
   endif()
 
   find_package(CapnProto
-    ${TILEDB_CAPNPROTO_VERSION} EXACT
+    ${TILEDB_CAPNPROTO_VERSION}
     PATHS ${TILEDB_EP_INSTALL_PREFIX}
     ${TILEDB_CAPNP_NO_DEFAULT_PATH}
     )
