Author: Milan P. Stanić <mps@arvanta.net>
Date: 2026-04-01 09:51:16.854293622 +0000

define AT_HANDLE_FID

--- a/sys-utils/nsenter.c	2026-04-01 09:51:16.854293622 +0000
+++ b/sys-utils/nsenter.c	2026-04-08 09:18:48.328152549 +0000
@@ -37,6 +37,11 @@
 # include <selinux/selinux.h>
 #endif
 
+
+#ifndef AT_HANDLE_FID
+# define AT_HANDLE_FID   0x200
+#endif
+
 #include "strutils.h"
 #include "nls.h"
 #include "c.h"
