From 9f037ddd4f78a6c8f409f1480b1816f0662f92b9 Mon Sep 17 00:00:00 2001
From: Dominik Schulz <mail@dschulz.eu>
Date: Tue, 27 Sep 2022 21:50:39 +0200
Subject: [PATCH] Update config integration tests

Signed-off-by: Dominik Schulz <dominik.schulz@gauner.org>
---
 tests/config_test.go | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/config_test.go b/tests/config_test.go
index 0fffabce..d71d769a 100644
--- a/tests/config_test.go
+++ b/tests/config_test.go
@@ -19,6 +19,7 @@ func TestBaseConfig(t *testing.T) { //nolint:paralleltest
 autoimport: true
 cliptimeout: 45
 exportkeys: false
+keychain: false
 nopager: false
 notifications: true
 parsing: true
@@ -72,6 +73,7 @@ func TestMountConfig(t *testing.T) { //nolint:paralleltest
 autoimport: true
 cliptimeout: 45
 exportkeys: false
+keychain: false
 nopager: false
 notifications: true
 parsing: true
-- 
2.37.3

