diff -rupN a/tests/test_prefseditor_keybindings.py b/tests/test_prefseditor_keybindings.py
--- a/tests/test_prefseditor_keybindings.py	2025-05-22 01:22:36.000000000 +0000
+++ b/tests/test_prefseditor_keybindings.py	2026-03-27 02:58:12.300000000 +0000
@@ -161,7 +161,8 @@ def test_duplicate_accels_not_possible_t
         if accel != ""  # ignore empty key bindings
     }
     # Check that a test accelerator is indeed a duplicate
-    assert (key, mods) in all_default_accelerators
+    assert Gtk.accelerator_parse(Gtk.accelerator_name(key, mods)) in \
+        all_default_accelerators
 
     default_accelerator = Gtk.accelerator_parse(
         config.DEFAULTS["keybindings"][binding]
