Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: define SOVERSION
----

--- a/brunsli.cmake
+++ b/brunsli.cmake
@@ -80,7 +80,9 @@
 add_library(brunslienc-c SHARED
   c/enc/encode.cc
 )
+set_target_properties(brunslidec-c PROPERTIES VERSION "${VERSION}" SOVERSION "${VERSION}")
 target_link_libraries(brunslienc-c PRIVATE brunslienc-static)
+set_target_properties(brunslienc-c PROPERTIES VERSION "${VERSION}" SOVERSION "${VERSION}")
 list(APPEND BRUNSLI_LIBRARIES brunslidec-c brunslienc-c)
 endif()  # BRUNSLI_EMSCRIPTEN
 
