diff -upr sbcl-1.3.1.orig/src/runtime/interrupt.h sbcl-1.3.1/src/runtime/interrupt.h
--- sbcl-1.3.1.orig/src/runtime/interrupt.h	2016-04-05 19:48:17.820002917 +0200
+++ sbcl-1.3.1/src/runtime/interrupt.h	2016-04-05 19:48:58.625144439 +0200
@@ -14,6 +14,7 @@
 
 #include "runtime.h"
 #include <string.h>
+#include <signal.h>
 
 /*
  * This is a workaround for some slightly silly Linux/GNU Libc
diff -upr sbcl-1.3.1.orig/src/runtime/linux-os.h sbcl-1.3.1/src/runtime/linux-os.h
--- sbcl-1.3.1.orig/src/runtime/linux-os.h	2016-04-05 19:48:17.823336398 +0200
+++ sbcl-1.3.1/src/runtime/linux-os.h	2016-04-05 19:48:47.117967857 +0200
@@ -24,7 +24,7 @@
 #include <linux/version.h>
 
 // Needs to be defined before including target-arch.h
-typedef caddr_t os_vm_address_t;
+typedef void *os_vm_address_t;
 typedef size_t os_vm_size_t;
 typedef off_t os_vm_offset_t;
 typedef int os_vm_prot_t;
