diff -upr sbcl-1.3.11.orig/make-target-2.sh sbcl-1.3.11/make-target-2.sh
--- sbcl-1.3.11.orig/make-target-2.sh	2016-11-02 10:44:23.226711691 +0100
+++ sbcl-1.3.11/make-target-2.sh	2016-11-02 10:45:19.043220337 +0100
@@ -39,6 +39,7 @@ fi
 # system with the :SB-SHOW feature enabled, it does it rather silently,
 # without trying to tell you about what it's doing. So unless it hangs
 # for much longer than that, don't worry, it's likely to be normal.
+paxmark.sh -mr ./src/runtime/sbcl
 if [ "$1" != --load ]; then
     echo //doing warm init - compilation phase
     echo '(load "loader.lisp") (load-sbcl-file "make-host-1.lisp")' | \
