diff --git a/backend/CMakeLists.txt b/backend/CMakeLists.txt
index 67499ec..6c6b478 100644
--- a/backend/CMakeLists.txt
+++ b/backend/CMakeLists.txt
@@ -90,7 +90,7 @@ if(NOT OPENSSL_FOUND)
 endif()
 
 # Provides internal iconv
-if(NOT ICONV_FOUND)
+if(false)
     download_project(PROJ iconv
         URL "${CMAKE_CURRENT_SOURCE_DIR}/libiconv-1.16.tar.gz"
         URL_HASH MD5=739c5e8f9e29fb95c0a1ad67becb80a7
@@ -110,6 +110,8 @@ if(NOT ICONV_FOUND)
 endif()
 
 # Provides internal marisa-trie
+find_package(PkgConfig)
+pkg_check_modules(MARISA REQUIRED marisa IMPORTED_TARGET GLOBAL)
 if(NOT MARISA_FOUND)
     download_project(PROJ marisa
         URL "${CMAKE_CURRENT_SOURCE_DIR}/marisa-0.2.5.tar.gz"
