--- a/backend/libosmscout/cmake/features.cmake
+++ b/backend/libosmscout/cmake/features.cmake
@@ -140,7 +140,8 @@
 
 # check libraries and tools
 if(NOT IOS)
-  find_package(Marisa)
+  find_package(PkgConfig)
+  pkg_check_modules(Marisa REQUIRED marisa IMPORTED_TARGET GLOBAL)
 endif()
 find_package(LibXml2)
 if (LIBXML2_FOUND AND NOT BUILD_SHARED_LIBS)
--- a/backend/CMakeLists.txt
+++ b/backend/CMakeLists.txt
@@ -113,6 +113,8 @@
 endif()
 
 # Provides dependency marisa-trie
+find_package(PkgConfig)
+pkg_check_modules(MARISA REQUIRED marisa IMPORTED_TARGET GLOBAL)
 if(NOT MARISA_FOUND)
     download_project(PROJ marisa
         URL "${CMAKE_CURRENT_SOURCE_DIR}/marisa-0.2.5.tar.gz"
