#lua > 5.1 expects table.unpack
--- a/lapis/flow.lua	2023-11-07 09:44:27.873535156 -0500
+++ b/lapis/flow.lua	2023-11-07 09:45:14.539062500 -0500
@@ -3,6 +3,7 @@
   local _obj_0 = _G
   type, getmetatable, setmetatable, rawset, rawget = _obj_0.type, _obj_0.getmetatable, _obj_0.setmetatable, _obj_0.rawset, _obj_0.rawget
 end
+local unpack = unpack or table.unpack
 local Flow
 local is_flow
 is_flow = function(cls)
--- a/lapis/flow.moon	2023-11-02 15:44:50.000000000 -0400
+++ b/lapis/flow.moon	2023-11-07 10:24:25.822998046 -0500
@@ -1,6 +1,6 @@
 
 import type, getmetatable, setmetatable, rawset, rawget from _G
-
+unpack = unpack or table.unpack
 local Flow
 
 is_flow = (cls) ->
