--- libreoffice-4.4.0.3/vcl/inc/headless/svpinst.hxx	2015-02-02 20:38:46.879497689 -0200
+++ libreoffice-4.4.0.3/vcl/inc/headless/svpinst.hxx	2015-02-02 20:38:55.482831474 -0200
@@ -32,6 +32,7 @@
 #include <list>
 
 #include <time.h>
+#include <sys/time.h>
 
 #define VIRTUAL_DESKTOP_WIDTH 1024
 #define VIRTUAL_DESKTOP_HEIGHT 768
--- libreoffice-4.4.0.3/vcl/unx/generic/gdi/cairotextrender.cxx	2015-02-02 21:19:22.822959064 -0200
+++ libreoffice-4.4.0.3/vcl/unx/generic/gdi/cairotextrender.cxx	2015-02-02 21:19:48.312960404 -0200
@@ -18,6 +18,7 @@
  */
 
 #include "cairotextrender.hxx"
+#include "generic/glyphcache.hxx"
 
 #include <basegfx/polygon/b2dpolypolygon.hxx>
 #include <vcl/settings.hxx>
--- libreoffice-4.4.0.3/sd/source/ui/remotecontrol/BluetoothServer.cxx	2015-02-03 08:42:36.590056362 -0200
+++ libreoffice-4.4.0.3/sd/source/ui/remotecontrol/BluetoothServer.cxx	2015-02-03 08:42:46.030056456 -0200
@@ -22,7 +22,7 @@
   #include <dbus/dbus.h>
   #include <errno.h>
   #include <fcntl.h>
-  #include <sys/unistd.h>
+  #include <unistd.h>
   #include <sys/socket.h>
   #include <bluetooth/bluetooth.h>
   #include <bluetooth/rfcomm.h>
