Upstream: https://bugzilla.xfce.org/show_bug.cgi?id=7985

--- xfce4-screenshooter-1.7.9.orig/Makefile.in	2010-02-07 14:45:15.000000000 +0100
+++ xfce4-screenshooter-1.7.9/Makefile.in	2010-02-16 23:57:31.000000000 +0100
@@ -282,7 +282,7 @@
 INTLTOOL_PERL = @INTLTOOL_PERL@
 INTLTOOL_UPDATE = @INTLTOOL_UPDATE@
 LD = @LD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = @LDFLAGS@ -lm -lX11
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
