--- a/doc/Makefile	2021-10-24 02:24:18.314331751 +0300
+++ b/doc/Makefile	2021-10-24 02:24:13.446288454 +0300
@@ -48,8 +48,6 @@
 ../source/help_topic.h ../source/help_data.h: .version help.etx setext
 	@ echo "Creating NEdit help code `cat .version`"
 	@ ./setext -m -v version="`cat .version`" help.etx
-	@ mv -f help_topic.h ../source
-	@ mv -f help_data.h ../source
 
 nedit.html: .version help.etx setext
 	@ echo "Creating NEdit HTML documentation..."
