diff -ru libraw1394-2.1.0.orig/src/fw-iso.c libraw1394-2.1.0/src/fw-iso.c
--- libraw1394-2.1.0.orig/src/fw-iso.c	2012-06-27 23:10:19.000000000 +0300
+++ libraw1394-2.1.0/src/fw-iso.c	2013-09-30 17:38:30.070506505 +0300
@@ -303,7 +303,7 @@
 }
 
 static int handle_iso_event(raw1394handle_t handle,
-			    struct epoll_closure *closure, __uint32_t events)
+			    struct epoll_closure *closure, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct fw_cdev_event_iso_interrupt *interrupt;
diff -ru libraw1394-2.1.0.orig/src/fw.c libraw1394-2.1.0/src/fw.c
--- libraw1394-2.1.0.orig/src/fw.c	2012-07-08 21:27:52.000000000 +0300
+++ libraw1394-2.1.0/src/fw.c	2013-09-30 17:38:30.070506505 +0300
@@ -187,7 +187,7 @@
 
 static int
 handle_echo_pipe(raw1394handle_t handle,
-		 struct epoll_closure *ec, __uint32_t events)
+		 struct epoll_closure *ec, uint32_t events)
 {
 	quadlet_t value;
 
@@ -258,7 +258,7 @@
 
 static int
 handle_device_event(raw1394handle_t handle,
-		    struct epoll_closure *ec, __uint32_t events)
+		    struct epoll_closure *ec, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	union fw_cdev_event *u;
@@ -421,7 +421,7 @@
 
 static int
 handle_inotify(raw1394handle_t handle, struct epoll_closure *ec,
-	       __uint32_t events)
+	       uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct inotify_event *event;
diff -ru libraw1394-2.1.0.orig/src/fw.h libraw1394-2.1.0/src/fw.h
--- libraw1394-2.1.0.orig/src/fw.h	2012-07-08 21:27:52.000000000 +0300
+++ libraw1394-2.1.0/src/fw.h	2013-09-30 17:38:42.330440087 +0300
@@ -12,6 +12,7 @@
 #ifndef LIBRAW1394_FW_H
 #define LIBRAW1394_FW_H
 
+#include <stdint.h>
 #include <stdlib.h>
 #include <byteswap.h>
 
@@ -48,7 +49,7 @@
 
 struct epoll_closure {
 	int (*func)(raw1394handle_t handle,
-		    struct epoll_closure *closure, __uint32_t events);
+		    struct epoll_closure *closure, uint32_t events);
 };
 
 struct port {
