--- libraw1394-2.1.0.orig/src/fw-iso.c
+++ libraw1394-2.1.0/src/fw-iso.c
@@ -303,7 +303,7 @@
 }
 
 static int handle_iso_event(raw1394handle_t handle,
-			    struct epoll_closure *closure, __uint32_t events)
+			    struct epoll_closure *closure, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct fw_cdev_event_iso_interrupt *interrupt;
--- libraw1394-2.1.0.orig/src/fw.c
+++ libraw1394-2.1.0/src/fw.c
@@ -187,7 +187,7 @@
 
 static int
 handle_echo_pipe(raw1394handle_t handle,
-		 struct epoll_closure *ec, __uint32_t events)
+		 struct epoll_closure *ec, uint32_t events)
 {
 	quadlet_t value;
 
@@ -258,7 +258,7 @@
 
 static int
 handle_device_event(raw1394handle_t handle,
-		    struct epoll_closure *ec, __uint32_t events)
+		    struct epoll_closure *ec, uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	union fw_cdev_event *u;
@@ -421,7 +421,7 @@
 
 static int
 handle_inotify(raw1394handle_t handle, struct epoll_closure *ec,
-	       __uint32_t events)
+	       uint32_t events)
 {
 	fw_handle_t fwhandle = handle->mode.fw;
 	struct inotify_event *event;
--- libraw1394-2.1.0.orig/src/fw.h
+++ libraw1394-2.1.0/src/fw.h
@@ -48,7 +48,7 @@
 
 struct epoll_closure {
 	int (*func)(raw1394handle_t handle,
-		    struct epoll_closure *closure, __uint32_t events);
+		    struct epoll_closure *closure, uint32_t events);
 };
 
 struct port {
