we have no stacktraces in wxwidgets
diff --git a/sip/cpp/sip_corewxAppTraits.cpp b/sip/cpp/sip_corewxAppTraits.cpp
index e3de2bf..d38f8fe 100644
--- a/sip/cpp/sip_corewxAppTraits.cpp
+++ b/sip/cpp/sip_corewxAppTraits.cpp
@@ -471,7 +471,7 @@ static PyObject *meth_wxAppTraits_GetAssertStackTrace(PyObject *sipSelf, PyObjec
             PyErr_Clear();
 
             Py_BEGIN_ALLOW_THREADS
-            sipRes = new ::wxString((sipSelfWasArg ? sipCpp->::wxAppTraits::GetAssertStackTrace() : sipCpp->GetAssertStackTrace()));
+            sipRes = new ::wxString("");
             Py_END_ALLOW_THREADS
 
             if (PyErr_Occurred())
