--- django-oscar-promotions-1.0.0b1/oscar_promotions/apps.py
+++ django-oscar-promotions-1.0.0b1.dj4/oscar_promotions/apps.py
@@ -1,4 +1,4 @@
-from django.conf.urls import url
+from django.urls import re_path as url
 from django.utils.translation import gettext_lazy as _
 from oscar.core.application import OscarConfig
 from oscar.core.loading import get_class, get_model
--- django-oscar-promotions-1.0.0b1/oscar_promotions/dashboard/apps.py
+++ django-oscar-promotions-1.0.0b1.dj4/oscar_promotions/dashboard/apps.py
@@ -1,4 +1,4 @@
-from django.conf.urls import url
+from django.urls import re_path as url
 from django.utils.translation import gettext_lazy as _
 from oscar.core.application import OscarDashboardConfig
 from oscar.core.loading import get_class
