--- a/src/ibusenginesimple.c
+++ b/src/ibusenginesimple.c
@@ -501,7 +501,7 @@
 }
 
 static IBusEngineDict *
-load_emoji_dict ()
+load_emoji_dict (IBusEngineSimple *)
 {
     IBusEngineDict *emoji_dict;
     GList *keys;
--- a/util/IMdkit/Xi18n.h
+++ b/util/IMdkit/Xi18n.h
@@ -71,11 +71,13 @@
 #define I18N_SET	1
 #define I18N_GET	2
 
-typedef struct
+typedef struct _Xi18nCore *Xi18n;
+
+typedef struct _TransportSW
 {
     char        *transportname;
     int         namelen;
-    Bool        (*checkAddr) ();
+    Bool        (*checkAddr) (Xi18n, struct _TransportSW *, char *);
 } TransportSW;
 
 typedef struct _XIMPending
@@ -169,8 +171,6 @@
     void *trans_rec;		/* contains transport specific data  */
     struct _Xi18nClient *next;
 } Xi18nClient;
-
-typedef struct _Xi18nCore *Xi18n;
 
 /*
  * Callback Struct for XIM Protocol
--- a/util/IMdkit/i18nMethod.c
+++ b/util/IMdkit/i18nMethod.c
@@ -89,7 +89,7 @@
 #ifdef DNETCONN
     {"decnet",          6, _Xi18nCheckTransAddress},
 #endif
-    {(char *) NULL,     0, (Bool (*) ()) NULL}
+    {(char *) NULL,     0, (Bool (*) (Xi18n, TransportSW *, char *)) NULL}
 };
 
 static Bool GetInputStyles (Xi18n i18n_core, XIMStyles **p_style)
