skips installing gtest
diff --git a/CMakeLists.txt b/CMakeLists.txt
index c3062e2..75f1265 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -313,7 +313,7 @@ if(SNAPPY_BUILD_TESTS)
   set(build_gmock ON)
 
   # This project is tested using GoogleTest.
-  add_subdirectory("third_party/googletest")
+  add_subdirectory("third_party/googletest" EXCLUDE_FROM_ALL)
 
   # GoogleTest triggers a missing field initializers warning.
   if(SNAPPY_HAVE_NO_MISSING_FIELD_INITIALIZERS)
