diff --git a/Po4aBuilder.pm b/Po4aBuilder.pm
index e9d755b..ccf00c5 100755
--- a/Po4aBuilder.pm
+++ b/Po4aBuilder.pm
@@ -251,7 +251,7 @@ sub ACTION_man {
             system("xsltproc -o $outdir/$outfile.$section --nonet /usr/share/xml/docbook/stylesheet/docbook-xsl/manpages/docbook.xsl $file") and die;
             } else { # Not found locally, use the XSL file online
             print "Convert $outdir/$outfile.$section (online docbook.xsl file). ";
-            system("xsltproc -o $outdir/$outfile.$section --nonet http://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $file") and die;
+            system("xsltproc -o $outdir/$outfile.$section --nonet https://docbook.sourceforge.net/release/xsl/current/manpages/docbook.xsl $file") and die;
             }
                 system ("gzip -9 -f $outdir/$outfile.$section") and die;
             }
