--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -34,7 +34,7 @@
     pdf = io.BytesIO()
     document.write(pdf)
     command = [
-        'gs', '-q', '-dNOPAUSE', '-dSAFER', '-sDEVICE=png16m',
+        'gs', '-q', '-dNOPAUSE', '-dNEWPDF=false', '-dSAFER', '-sDEVICE=png16m',
         '-r576', '-dDownScaleFactor=8', '-sOutputFile=-', '-']
     png = run(command, input=pdf.getvalue(), stdout=PIPE).stdout
     image = Image.open(io.BytesIO(png))
