diff --git a/config.h.in b/config.h.in
index c31e5a9..a79e273 100644
--- a/config.h.in
+++ b/config.h.in
@@ -50,7 +50,7 @@
 /* Define if we have gh_vector_ref */
 #undef HAVE_GH_VECTOR_REF
 
-/* Define to 1 if you have the <guile/gh.h> header file. */
+/* Define to 1 if you have the <libguile.h> header file. */
 #undef HAVE_GUILE_GH_H
 
 /* Define to 1 if you have the <inttypes.h> header file. */
diff --git a/configure b/configure
index c51270d..48d783a 100755
--- a/configure
+++ b/configure
@@ -12878,7 +12878,7 @@ rm -f core conftest.err conftest.$ac_objext \
 fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
-for ac_header in libguile.h guile/gh.h
+for ac_header in libguile.h libguile.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
@@ -12929,7 +12929,7 @@ $as_echo_n "checking for gh_enter... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -12956,7 +12956,7 @@ $as_echo_n "checking for gh_eval_str... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -12983,7 +12983,7 @@ $as_echo_n "checking for gh_load... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -13010,7 +13010,7 @@ $as_echo_n "checking for gh_bool2scm... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -13037,7 +13037,7 @@ $as_echo_n "checking for gh_vector_ref... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -13064,7 +13064,7 @@ $as_echo_n "checking for gh_list_ref... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -13091,7 +13091,7 @@ $as_echo_n "checking for gh_length... " >&6; }
 ok=yes
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
-#include <guile/gh.h>
+#include <libguile.h>
 
 int
 main ()
@@ -13121,7 +13121,7 @@ cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 #if defined(HAVE_LIBGUILE_H)
 #  include <libguile.h>
 #else
-#  include <guile/gh.h>
+#  include <libguile.h>
 #endif
 
 int
@@ -13165,7 +13165,7 @@ cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 #if defined(HAVE_LIBGUILE_H)
 #  include <libguile.h>
 #else
-#  include <guile/gh.h>
+#  include <libguile.h>
 #endif
 
 int
@@ -13198,7 +13198,7 @@ else
 /* end confdefs.h.  */
 
 #include <stdlib.h>
-#include <guile/gh.h>
+#include <libguile.h>
 void main_entry(int argc, char *argv[])
 {
      gh_eval_str("(define foo 3.14159)");
diff --git a/configure.ac b/configure.ac
index 98e38c5..56532c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -82,7 +82,7 @@ AC_TRY_LINK_FUNC(gh_enter, AC_MSG_RESULT(yes),
    [AC_TRY_LINK_FUNC(scm_boot_guile, AC_MSG_RESULT(yes),
        [AC_MSG_RESULT(no)
         AC_MSG_ERROR(Guile could not be found)])])
-AC_CHECK_HEADERS([libguile.h guile/gh.h])
+AC_CHECK_HEADERS([libguile.h libguile.h])
 
 ###########################################################################
 
@@ -97,43 +97,43 @@ AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_enter])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_enter(0,0,0);], [AC_DEFINE(HAVE_GH_ENTER, 1, [Define if we have gh_enter.])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_eval_str])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_eval_str(0);], [AC_DEFINE(HAVE_GH_EVAL_STR, 1, [Define if we have gh_eval_str.])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_load])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_load(0);], [AC_DEFINE(HAVE_GH_LOAD, 1, [Define if we have gh_load.])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_bool2scm])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_bool2scm(0);], [AC_DEFINE(HAVE_GH_BOOL2SCM, 1, [Define if we have gh_bool2scm])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_vector_ref])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_vector_ref(0,0);], [AC_DEFINE(HAVE_GH_VECTOR_REF, 1, [Define if we have gh_vector_ref])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_list_ref])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_list_ref(0,0);], [AC_DEFINE(HAVE_GH_LIST_REF, 1, [Define if we have gh_list_ref])], ok=no)
 AC_MSG_RESULT($ok)
 
 AC_MSG_CHECKING([for gh_length])
 ok=yes
-AC_TRY_LINK([#include <guile/gh.h>
+AC_TRY_LINK([#include <libguile.h>
 ], [gh_length(0);], [AC_DEFINE(HAVE_GH_LENGTH, 1, [Define if we have gh_length])], ok=no)
 AC_MSG_RESULT($ok)
 
@@ -142,7 +142,7 @@ ok=yes
 AC_TRY_LINK([#if defined(HAVE_LIBGUILE_H)
 #  include <libguile.h>
 #else
-#  include <guile/gh.h>
+#  include <libguile.h>
 #endif
 ], [scm_flush_all_ports();], [AC_DEFINE(HAVE_SCM_FLUSH_ALL_PORTS, 1, [Define if we have scm_flush_all_ports])], ok=no)
 AC_MSG_RESULT($ok)
@@ -154,7 +154,7 @@ ok=yes
 AC_TRY_LINK([#if defined(HAVE_LIBGUILE_H)
 #  include <libguile.h> 
 #else 
-#  include <guile/gh.h>
+#  include <libguile.h>
 #endif
 ], [SCM x; SCM_COMPLEXP(x);], [AC_DEFINE(HAVE_SCM_COMPLEXP, 1, [Define if we have SCM_COMPLEXP])], ok=no)
 AC_MSG_RESULT($ok)
@@ -163,7 +163,7 @@ AC_MSG_CHECKING([whether gh_lookup works properly])
 ok=yes
 AC_TRY_RUN([
 #include <stdlib.h>
-#include <guile/gh.h>
+#include <libguile.h>
 void main_entry(int argc, char *argv[])
 {
      gh_eval_str("(define foo 3.14159)");
diff --git a/src/ctl.h.in b/src/ctl.h.in
index 1ecee3f..b0c0096 100644
--- a/src/ctl.h.in
+++ b/src/ctl.h.in
@@ -26,7 +26,7 @@
 #ifdef HAVE_NO_GH
 #  include <libguile.h>
 #else
-#  include <guile/gh.h>
+#  include <libguile.h>
 #endif
 
 #ifdef __cplusplus
