--- ./compiler.orig
+++ ./compiler
@@ -3991,12 +3991,12 @@
 __EOF__
 	done
 
-	save_command "echo 1 > /proc/sys/net/ipv4/conf/all/rp_filter"
-
 	if [ "$ROUTE_FILTER" = yes ]; then
 	    save_command "echo 1 > /proc/sys/net/ipv4/conf/default/rp_filter"
+	    save_command "echo 1 > /proc/sys/net/ipv4/conf/all/rp_filter"
 	elif [ "$ROUTE_FILTER" = no ]; then
 	    save_command "echo 0 > /proc/sys/net/ipv4/conf/default/rp_filter"
+	    save_command "echo 0 > /proc/sys/net/ipv4/conf/all/rp_filter"
 	fi
 
 	save_command "[ -n \"\$NOROUTES\" ] || ip route flush cache"
