diff -Nurp a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2025-09-18 19:36:02.325155787 +0000
+++ b/src/CMakeLists.txt	2025-09-18 19:36:02.339155974 +0000
@@ -432,7 +432,6 @@ set(libcommon_files
   ${CMAKE_BINARY_DIR}/src/include/ceph_ver.h
   ceph_ver.c
   global/global_context.cc
-  xxHash/xxhash.c
   common/error_code.cc
   common/tracer.cc
   log/Log.cc
@@ -566,6 +565,8 @@ add_library(common STATIC ${ceph_common_
 target_link_libraries(common
   ${ceph_common_deps}
   legacy-option-headers)
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
+target_link_libraries(common-objs PkgConfig::XXHASH)
 if(WITH_JAEGER)
 add_dependencies(common jaeger_base)
 endif()
diff -Nurp a/src/common/Checksummer.h b/src/common/Checksummer.h
--- a/src/common/Checksummer.h	2025-09-18 19:36:02.326155800 +0000
+++ b/src/common/Checksummer.h	2025-09-18 19:36:02.340155988 +0000
@@ -8,7 +8,7 @@
 #include "include/byteorder.h"
 #include "include/ceph_assert.h"
 
-#include "xxHash/xxhash.h"
+#include "xxhash.h"
 
 class Checksummer {
 public:
diff -Nurp a/src/crimson/CMakeLists.txt b/src/crimson/CMakeLists.txt
--- a/src/crimson/CMakeLists.txt	2025-09-18 19:36:02.328155827 +0000
+++ b/src/crimson/CMakeLists.txt	2025-09-18 19:36:02.340155988 +0000
@@ -117,7 +117,6 @@ add_library(crimson-common STATIC
   ${PROJECT_SOURCE_DIR}/src/osd/OSDMap.cc
   ${PROJECT_SOURCE_DIR}/src/osd/PGPeeringEvent.cc
   ${PROJECT_SOURCE_DIR}/src/common/scrub_types.cc
-  ${PROJECT_SOURCE_DIR}/src/xxHash/xxhash.c
   ${crimson_common_srcs}
   $<TARGET_OBJECTS:common_mountcephfs_objs>
   $<TARGET_OBJECTS:common-options-objs>)
@@ -141,10 +140,12 @@ if(NOT WITH_SYSTEM_BOOST)
   list(APPEND crimson_common_deps ${ZLIB_LIBRARIES})
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(crimson-common
   PUBLIC
     ${crimson_common_public_deps}
   PRIVATE
+    PkgConfig::XXHASH
     crc32 arch
     ${crimson_common_deps}
     OpenSSL::Crypto)
diff -Nurp a/src/os/CMakeLists.txt b/src/os/CMakeLists.txt
--- a/src/os/CMakeLists.txt	2025-09-18 19:36:02.330155854 +0000
+++ b/src/os/CMakeLists.txt	2025-09-18 19:36:02.341156001 +0000
@@ -42,10 +42,12 @@ if(HAVE_LIBXFS)
     fs/XFS.cc)
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 add_library(os STATIC ${libos_srcs})
 target_link_libraries(os
   legacy-option-headers
   blk
+  PkgConfig::XXHASH
   ${FMT_LIB})
 
 target_compile_definitions(os PRIVATE -DWITH_KSTORE)
diff -Nurp a/src/rgw/CMakeLists.txt b/src/rgw/CMakeLists.txt
--- a/src/rgw/CMakeLists.txt	2025-09-18 19:38:45.252336840 +0000
+++ b/src/rgw/CMakeLists.txt	2025-09-18 19:58:12.349978391 +0000
@@ -288,6 +288,7 @@ if(COMPILER_SUPPORTS_WARN_IMPLICIT_CONST
     HAVE_WARN_IMPLICIT_CONST_INT_FLOAT_CONVERSION)
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(rgw_common
   PRIVATE
     legacy-option-headers
@@ -315,6 +316,7 @@ target_link_libraries(rgw_common
     ${ARROW_FLIGHT_LIBRARIES}
     ${LMDB_LIBRARIES}
   PUBLIC
+    PkgConfig::XXHASH
     ${LUA_LIBRARIES}
     RapidJSON::RapidJSON
     Boost::context
diff -Nurp a/src/test/neorados/read_operations.cc b/src/test/neorados/read_operations.cc
--- a/src/test/neorados/read_operations.cc	2025-10-10 05:15:04.593294866 +0000
+++ b/src/test/neorados/read_operations.cc	2025-10-10 05:15:20.735502210 +0000
@@ -26,7 +26,7 @@
 #include <boost/system/error_code.hpp>
 #include <boost/system/errc.hpp>
 
-#include <xxHash/xxhash.h>
+#include <xxhash.h>
 
 #include "include/neorados/RADOS.hpp"
 #include "include/rbd/features.h" // for RBD_FEATURES_ALL
diff -Nurp a/src/test/librados/CMakeLists.txt b/src/test/librados/CMakeLists.txt
--- a/src/test/librados/CMakeLists.txt	2025-10-10 13:05:38.345172398 +0000
+++ b/src/test/librados/CMakeLists.txt	2025-10-10 13:20:16.324410133 +0000
@@ -96,16 +96,17 @@ add_executable(ceph_test_rados_api_cls
 target_link_libraries(ceph_test_rados_api_cls
   librados ${UNITTEST_LIBS} radostest-cxx)
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 add_executable(ceph_test_rados_api_misc
   misc.cc
   $<TARGET_OBJECTS:unit-main>)
 target_link_libraries(ceph_test_rados_api_misc
-  librados global ${UNITTEST_LIBS} radostest)
+  PkgConfig::XXHASH librados global ${UNITTEST_LIBS} radostest)
 add_executable(ceph_test_rados_api_misc_pp
   misc_cxx.cc
   $<TARGET_OBJECTS:unit-main>)
 target_link_libraries(ceph_test_rados_api_misc_pp
-  librados global ${UNITTEST_LIBS} radostest-cxx)
+  PkgConfig::XXHASH librados global ${UNITTEST_LIBS} radostest-cxx)
 
 add_executable(ceph_test_rados_api_lock
   lock.cc)
diff -Nurp a/src/test/neorados/CMakeLists.txt b/src/test/neorados/CMakeLists.txt
--- a/src/test/neorados/CMakeLists.txt	2025-10-10 12:35:51.950329587 +0000
+++ b/src/test/neorados/CMakeLists.txt	2025-10-10 13:03:33.795578646 +0000
@@ -77,7 +77,9 @@ install(TARGETS
 add_executable(ceph_test_neorados_read_operations
   read_operations.cc
   )
+  pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(ceph_test_neorados_read_operations
+  PkgConfig::XXHASH
   libneorados
   ${BLKID_LIBRARIES}
   ${CMAKE_DL_LIBS}
diff -Nurp a/src/test/CMakeLists.txt b/src/test/CMakeLists.txt
diff -Nurp a/src/test/CMakeLists.txt b/src/test/CMakeLists.txt
--- a/src/test/CMakeLists.txt	2025-10-10 14:50:43.419289886 +0000
+++ b/src/test/CMakeLists.txt	2025-10-10 15:05:24.036141003 +0000
@@ -316,6 +316,7 @@ add_executable(ceph_test_librgw_file_gp
   librgw_file_gp.cc
   )
 target_link_libraries(ceph_test_librgw_file_gp
+  PkgConfig::XXHASH
   rgw
   librados
   ceph-common
@@ -349,7 +350,9 @@ install(TARGETS ceph_test_librgw_file_nf
 add_executable(ceph_test_librgw_file_aw
   librgw_file_aw.cc
   )
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(ceph_test_librgw_file_aw
+  PkgConfig::XXHASH
   rgw
   librados
   ceph-common
