diff -Nurp a/src/common/tcp_info.cc b/src/common/tcp_info.cc
--- a/src/common/tcp_info.cc	2025-09-04 19:35:40.000000000 +0000
+++ b/src/common/tcp_info.cc	2025-09-18 14:22:33.474654670 +0000
@@ -104,12 +104,16 @@ bool dump_tcp_info(int fd, Formatter* f)
   if (info.tcpi_options & TCPI_OPT_ECN) {
     f->dump_string("option", "ecn");
   }
+#ifdef TCPI_OPT_ECN_SEEN
   if (info.tcpi_options & TCPI_OPT_ECN_SEEN) {
     f->dump_string("option", "ecn seen");
   }
+#endif
+#ifdef TCPI_OPT_SYN_DATA
   if (info.tcpi_options & TCPI_OPT_SYN_DATA) {
     f->dump_string("option", "syn data");
   }
+#endif
   f->close_section();
 
   f->close_section();
