--- a/util.c
+++ b/util.c
@@ -135,9 +133,9 @@
         && ioctl (fdesc, MTIOCGET, &s2) == 0
         && MT_SAME_POSITION (s1, s2))
     {
+        fprintf(stderr, "%s: warning: working around lseek kernel bug for file (%s)\n"
+  		"    of mt_type=0x%0lx -- see <sys/mtio.h> for the list of types\n",
+		program_name, filename, s2.mt_type);
-        error (0, 0, _("warning: working around lseek kernel bug for file (%s)\n\
-  of mt_type=0x%0lx -- see <sys/mtio.h> for the list of types"),
-               filename, s2.mt_type);
         errno = 0;
         new_position = -1;
     }
