diff -rupN a/tests/helpers.py b/tests/helpers.py
--- a/tests/helpers.py	2024-10-10 22:39:23.000000000 +0000
+++ b/tests/helpers.py	2026-04-02 23:43:30.320000000 +0000
@@ -15,7 +15,7 @@ def AsyncMock(*args, **kwargs):
 
 def _run(coro):
     """Run the given coroutine."""
-    return asyncio.get_event_loop().run_until_complete(coro)
+    return asyncio.new_event_loop().run_until_complete(coro)
 
 
 def make_sync(coro):
