--- ./include/sane/sanei_udp.h.orig
+++ ./include/sane/sanei_udp.h
@@ -19,6 +19,7 @@
 #define sanei_udp_h
 
 #include <sane/sane.h>
+#include <sys/types.h>
 
 #ifdef HAVE_WINSOCK2_H
 #include <winsock2.h>
--- ./backend/hp5400_sanei.c.orig
+++ ./backend/hp5400_sanei.c
@@ -57,6 +57,7 @@
 #include "hp5400_debug.h"
 #include <stdio.h>
 #include "../include/sane/sanei_usb.h"
+#include <sys/types.h>
 
 #define CMD_INITBULK1   0x0087	/* send 0x14 */
 #define CMD_INITBULK2   0x0083	/* send 0x24 */
--- ./backend/hp5590_low.c.orig
+++ ./backend/hp5590_low.c
@@ -50,6 +50,7 @@
 #ifdef HAVE_NETINET_IN_H
 # include <netinet/in.h>
 #endif /* HAVE_NETINET_IN_H */
+#include <sys/types.h>
 
 #include "byteorder.h"
 
--- ./backend/kvs20xx_cmd.h.orig
+++ ./backend/kvs20xx_cmd.h
@@ -9,6 +9,8 @@
    Panasonic KV-S20xx USB-SCSI scanners.
 */
 
+#include <sys/types.h>
+
 #define COMMAND_BLOCK	1
 #define DATA_BLOCK	2
 #define RESPONSE_BLOCK	3
--- ./backend/kvs40xx.h.orig
+++ ./backend/kvs40xx.h
@@ -10,6 +10,7 @@
 
 #include "../include/sane/config.h"
 #include <semaphore.h>
+#include <sys/types.h>
 
 #undef  BACKEND_NAME
 #define BACKEND_NAME kvs40xx
