diff --git a/include/mapnik/text/harfbuzz_shaper.hpp b/include/mapnik/text/harfbuzz_shaper.hpp
index 8b574b0..2d41efb 100644
--- a/include/mapnik/text/harfbuzz_shaper.hpp
+++ b/include/mapnik/text/harfbuzz_shaper.hpp
@@ -55,11 +55,7 @@ static inline hb_script_t _icu_script_to_script(UScriptCode script)
 static inline const uint16_t * uchar_to_utf16(const UChar* src)
 {
    static_assert(sizeof(UChar) == sizeof(uint16_t),"UChar is eq size to uint16_t");
-#if defined(_MSC_VER)
    return reinterpret_cast<const uint16_t *>(src);
-#else
-   return src;
-#endif
 }
 
 struct harfbuzz_shaper
diff --git a/include/mapnik/value_types.hpp b/include/mapnik/value_types.hpp
index 7682d52..9dba491 100644
--- a/include/mapnik/value_types.hpp
+++ b/include/mapnik/value_types.hpp
@@ -23,6 +23,7 @@
 #ifndef MAPNIK_VALUE_TYPES_HPP
 #define MAPNIK_VALUE_TYPES_HPP
 
+#include <unicode/unistr.h>
 // mapnik
 #include <mapnik/config.hpp>
 #include <mapnik/cxx11_support.hpp>
