--- mongo-tools-r3.4.3/build.sh.orig
+++ mongo-tools-r3.4.3/build.sh
@@ -7,8 +7,8 @@
 fi
 
 # make sure we're in the directory where the script lives
-SCRIPT_DIR="$(cd "$(dirname ${BASH_SOURCE[0]})" && pwd)"
-cd $SCRIPT_DIR
+#SCRIPT_DIR="$(cd "$(dirname ${BASH_SOURCE[0]})" && pwd)"
+#cd $SCRIPT_DIR
 
 sed -i.bak -e "s/built-without-version-string/$(git describe)/" \
            -e "s/built-without-git-spec/$(git rev-parse HEAD)/" \
