--- a/sphinxcontrib/adadomain.py
+++ b/sphinxcontrib/adadomain.py
@@ -17,14 +17,13 @@
 import string
 
 from docutils import nodes
-from docutils.parsers.rst import directives
+from docutils.parsers.rst import Directive, directives
 
 from sphinx import addnodes
 from sphinx.roles import XRefRole
 from sphinx.locale import l_, _
 from sphinx.directives import ObjectDescription
 from sphinx.domains import Domain, ObjType, Index
-from sphinx.util.compat import Directive
 from sphinx.util.nodes import make_refnode
 from sphinx.util.docfields import Field, TypedField
 
