diff -urN Ri-li-2.0.1-orig/src//audio.cc Ri-li-2.0.1/src//audio.cc
--- Ri-li-2.0.1-orig/src//audio.cc	2007-11-02 12:48:15.000000000 +0100
+++ Ri-li-2.0.1/src//audio.cc	2010-12-14 21:54:10.533334011 +0100
@@ -22,13 +22,16 @@
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 
 #include "audio.h"
 #include "utils.h"
 #include "preference.h"
 
+using std::cerr;
+using std::endl;
+
 /*** Variable globales ***/
 /*************************/
 extern sPreference Pref;
diff -urN Ri-li-2.0.1-orig/src//ecran.cc Ri-li-2.0.1/src//ecran.cc
--- Ri-li-2.0.1-orig/src//ecran.cc	2007-11-02 12:48:16.000000000 +0100
+++ Ri-li-2.0.1/src//ecran.cc	2010-12-14 21:54:50.346667343 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <math.h>
diff -urN Ri-li-2.0.1-orig/src//editeur.cc Ri-li-2.0.1/src//editeur.cc
--- Ri-li-2.0.1-orig/src//editeur.cc	2007-11-02 12:48:17.000000000 +0100
+++ Ri-li-2.0.1/src//editeur.cc	2010-12-14 21:55:25.770000677 +0100
@@ -25,7 +25,7 @@
 #include <windows.h>
 #endif
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -38,6 +38,9 @@
 #include "tableau.h"
 #include "mouse.h"
 
+using std::cerr;
+using std::endl;
+
 /*** Variables globales ***/
 /**************************/
 extern SDL_Surface *sdlVideo;
diff -urN Ri-li-2.0.1-orig/src//jeux.cc Ri-li-2.0.1/src//jeux.cc
--- Ri-li-2.0.1-orig/src//jeux.cc	2007-11-02 12:48:17.000000000 +0100
+++ Ri-li-2.0.1/src//jeux.cc	2010-12-14 22:06:21.783334023 +0100
@@ -25,7 +25,7 @@
 #include <windows.h>
 #endif
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -urN Ri-li-2.0.1-orig/src//loco.cc Ri-li-2.0.1/src//loco.cc
--- Ri-li-2.0.1-orig/src//loco.cc	2007-11-02 12:48:18.000000000 +0100
+++ Ri-li-2.0.1/src//loco.cc	2010-12-14 22:06:40.183334009 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
diff -urN Ri-li-2.0.1-orig/src//main.cc Ri-li-2.0.1/src//main.cc
--- Ri-li-2.0.1-orig/src//main.cc	2007-11-02 12:48:19.000000000 +0100
+++ Ri-li-2.0.1/src//main.cc	2010-12-14 22:07:15.993334009 +0100
@@ -23,7 +23,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
-#include <iostream.h>
+#include <iostream>
 #include <string.h>
 #include <SDL/SDL.h>
 #include <SDL/SDL_mixer.h>
@@ -39,6 +39,9 @@
 #include "editeur.h"
 #include "utils.h"
 
+using std::cerr;
+using std::endl;
+
 /*** Variables globales ***/
 /************************/
 SDL_Surface *sdlVideo; // Pointe sur l'écran video
diff -urN Ri-li-2.0.1-orig/src//menu.cc Ri-li-2.0.1/src//menu.cc
--- Ri-li-2.0.1-orig/src//menu.cc	2007-11-02 12:48:19.000000000 +0100
+++ Ri-li-2.0.1/src//menu.cc	2010-12-14 22:07:55.990000676 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -34,6 +34,9 @@
 
 #define PY 180
 
+using std::cerr;
+using std::endl;
+
 /*** Variables globales ***/
 /**************************/
 extern int Horloge;
diff -urN Ri-li-2.0.1-orig/src//mouse.cc Ri-li-2.0.1/src//mouse.cc
--- Ri-li-2.0.1-orig/src//mouse.cc	2007-11-02 12:48:20.000000000 +0100
+++ Ri-li-2.0.1/src//mouse.cc	2010-12-14 22:08:11.696667343 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include "mouse.h"
 #include "preference.h"
diff -urN Ri-li-2.0.1-orig/src//sprite.cc Ri-li-2.0.1/src//sprite.cc
--- Ri-li-2.0.1-orig/src//sprite.cc	2007-11-02 12:48:20.000000000 +0100
+++ Ri-li-2.0.1/src//sprite.cc	2010-12-14 22:09:17.650000676 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -30,6 +30,9 @@
 #include "preference.h"
 #include "utils.h"
 
+using std::cerr;
+using std::endl;
+
 /*** Variables Globales ***/
 /**************************/
 extern SDL_Surface *sdlVideo;
@@ -39,8 +42,8 @@
 extern int NSprites;
 extern sPreference Pref;
 
-static char* OrdreTexte="ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789.-!?*+<>%$()&;";
-static char* OrdreTexte2="abcdefghijklmnopqrstuvwxyz0123456789,_|?*+<>%$[]&;";
+static char* OrdreTexte=(char *)"ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789.-!?*+<>%$()&;";
+static char* OrdreTexte2=(char *)"abcdefghijklmnopqrstuvwxyz0123456789,_|?*+<>%$[]&;";
 static int TableTexte[256]; 
 
 char Langue[31][16]; // Mémorise les noms des langues
diff -urN Ri-li-2.0.1-orig/src//tableau.cc Ri-li-2.0.1/src//tableau.cc
--- Ri-li-2.0.1-orig/src//tableau.cc	2007-11-02 12:48:21.000000000 +0100
+++ Ri-li-2.0.1/src//tableau.cc	2010-12-14 22:09:32.603334046 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include "preference.h"
diff -urN Ri-li-2.0.1-orig/src//utils.cc Ri-li-2.0.1/src//utils.cc
--- Ri-li-2.0.1-orig/src//utils.cc	2007-11-02 12:48:22.000000000 +0100
+++ Ri-li-2.0.1/src//utils.cc	2010-12-14 22:09:58.343334046 +0100
@@ -21,7 +21,7 @@
 //    with this program; if not, write to the Free Software Foundation, Inc.,
 //    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-#include <iostream.h>
+#include <iostream>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
@@ -36,6 +36,9 @@
 #include "preference.h"
 #include "sprite.h"
 
+using std::cerr;
+using std::endl;
+
 /*** Variables globales ***/
 /**************************/
 extern sPreference Pref;
