diff --git a/setup.py b/setup.py
index a228345..59afd70 100755
--- a/setup.py
+++ b/setup.py
@@ -21,7 +21,7 @@ from DistUtilsExtra.auto import setup
 import glob
 
 # Create data files
-data = [ ('/usr/share/polkit-1/actions/', ['policykit/actions/com.ubuntu.pkexec.gufw.policy']),
+data = [ ('share/polkit-1/actions/', ['policykit/actions/com.ubuntu.pkexec.gufw.policy']),
          ('/etc/gufw/app_profiles',       glob.glob("data/app_profiles/*.*")) ]
 
 # Setup stage
