From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 13 Apr 2017 15:57:00 +0200
Subject: [PATCH] Use dynamic linking as default on musl

--- a/src/librustc_back/target/linux_musl_base.rs
+++ b/src/librustc_back/target/linux_musl_base.rs
@@ -19,9 +19,6 @@
     // argument is *not* necessary for normal builds, but it can't hurt!
     base.pre_link_args.push("-Wl,--eh-frame-hdr".to_string());
 
-    // Except for on MIPS, these targets statically link libc by default.
-    base.crt_static_default = true;
-
     // Static position-independent executables are supported.
     base.static_position_independent_executables = true;
 
