diff --git a/sibc/bsidh/main.py b/sibc/bsidh/main.py
index 4f2685b..f331ecf 100644
--- a/sibc/bsidh/main.py
+++ b/sibc/bsidh/main.py
@@ -1,4 +1,3 @@
-from pkg_resources import resource_filename
 import click
 
 from sibc.common import attrdict, geometric_serie, rounds
diff --git a/sibc/bsidh/precompute_parameters.py b/sibc/bsidh/precompute_parameters.py
index 67b7c87..3510373 100644
--- a/sibc/bsidh/precompute_parameters.py
+++ b/sibc/bsidh/precompute_parameters.py
@@ -1,6 +1,6 @@
 import sys
 import click
-from pkg_resources import resource_filename
+import importlib.resources
 
 from math import floor, sqrt, pi
 from sibc.common import attrdict
@@ -142,8 +142,8 @@ def bsidh_precompute_parameters(ctx):
 
     original_stdout = sys.stdout # Save a reference to the original standard output
     multievaluation = {True:'scaled', False:'unscaled'}[setting.multievaluation]
-    path = resource_filename('sibc', "data/ijk/" + algo.curve.model + '/bsidh/' + algo.curve.name + '-' + multievaluation)
-    with open(path, 'w') as f:
+    file_path = importlib.resources.files('sibc').joinpath("data/ijk", algo.curve.model, 'bsidh', algo.curve.name + '-' + multievaluation)
+    with file_path.open("w") as f:
         sys.stdout = f # Change the standard output to the file we created.
         parameters = dict()
         for idx in range(0, np, 1):
diff --git a/sibc/bsidh/precompute_strategy.py b/sibc/bsidh/precompute_strategy.py
index 379c3d7..5c9d93a 100644
--- a/sibc/bsidh/precompute_strategy.py
+++ b/sibc/bsidh/precompute_strategy.py
@@ -1,5 +1,5 @@
 import click
-from pkg_resources import resource_filename
+import importlib.resources
 
 from sibc.common import attrdict
 
@@ -30,16 +30,17 @@ def bsidh_precompute_strategy(ctx):
         + multievaluation
         + tuned
     )
-    file_path = resource_filename('sibc', file_path)
-    print("// Strategies to be computed")
-    # List of Small Isogeny Degree, Lp := [l_0, ..., l_{n-1}] [We need to
-    # include case l=2 and l=4]
-    Sp, Cp = algo.strategy.dynamic_programming_algorithm(SIDp[::-1], len(SIDp))
-    # List of Small Isogeny Degree, Lm := [l_0, ..., l_{n-1}]
-    Sm, Cm = algo.strategy.dynamic_programming_algorithm(SIDm[::-1], len(SIDm))
-    f = open(file_path, 'w')
-    f.writelines(' '.join([str(tmp) for tmp in Sp]) + '\n')
-    f.writelines(' '.join([str(tmp) for tmp in Sm]) + '\n')
-    f.close()
+    file_path = importlib.resources.files('sibc') / file_path
+    with importlib.resources.as_file(file_path) as path:
+        print("// Strategies to be computed")
+        # List of Small Isogeny Degree, Lp := [l_0, ..., l_{n-1}] [We need to
+        # include case l=2 and l=4]
+        Sp, Cp = algo.strategy.dynamic_programming_algorithm(SIDp[::-1], len(SIDp))
+        # List of Small Isogeny Degree, Lm := [l_0, ..., l_{n-1}]
+        Sm, Cm = algo.strategy.dynamic_programming_algorithm(SIDm[::-1], len(SIDm))
+        f = open(path, 'w')
+        f.writelines(' '.join([str(tmp) for tmp in Sp]) + '\n')
+        f.writelines(' '.join([str(tmp) for tmp in Sm]) + '\n')
+        f.close()
 
-    return attrdict(name='bsidh-precompute-strategy', **locals())
\ No newline at end of file
+    return attrdict(name='bsidh-precompute-strategy', **locals())
diff --git a/sibc/bsidh/strategy.py b/sibc/bsidh/strategy.py
index 49ad9d1..3f20282 100644
--- a/sibc/bsidh/strategy.py
+++ b/sibc/bsidh/strategy.py
@@ -1,5 +1,5 @@
 from random import SystemRandom
-from pkg_resources import resource_filename
+import importlib.resources
 
 from functools import reduce
 from math import floor, sqrt
@@ -37,23 +37,25 @@ class Strategy(object):
             self.S[i] = {}
 
         # Reading public generators points
-        f = open(resource_filename('sibc', 'data/gen/' + curve.model + '/bsidh/' + prime))
+        file_path = importlib.resources.files('sibc').joinpath("data/gen", curve.model, "bsidh", prime)
+        with importlib.resources.as_file(file_path) as path:
+            f = open(path)
+
+            # x(PA), x(QA) and x(PA - QA)
+            PQA = f.readline()
+            PQA = [int(x, 16) for x in PQA.split()]
+            self.PA = self.field(PQA[0:2])
+            self.QA = self.field(PQA[2:4])
+            self.PQA= self.field(PQA[4:6])
+
+            # x(PB), x(QB) and x(PB - QB)
+            PQB = f.readline()
+            PQB = [int(x, 16) for x in PQB.split()]
+            self.PB = self.field(PQB[0:2])
+            self.QB = self.field(PQB[2:4])
+            self.PQB= self.field(PQB[4:6])
 
-        # x(PA), x(QA) and x(PA - QA)
-        PQA = f.readline()
-        PQA = [int(x, 16) for x in PQA.split()]
-        self.PA = self.field(PQA[0:2])
-        self.QA = self.field(PQA[2:4])
-        self.PQA= self.field(PQA[4:6])
-
-        # x(PB), x(QB) and x(PB - QB)
-        PQB = f.readline()
-        PQB = [int(x, 16) for x in PQB.split()]
-        self.PB = self.field(PQB[0:2])
-        self.QB = self.field(PQB[2:4])
-        self.PQB= self.field(PQB[4:6])
-
-        f.close()
+            f.close()
 
         if not formula.uninitialized:
             file_path = (
@@ -69,19 +71,20 @@ class Strategy(object):
                 + formula.multievaluation_name
                 + formula.tuned_name
             )
-            file_path = resource_filename('sibc', file_path)
-            f = open(file_path)
-            # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
-            # l_{n-1}] [We need to include case l=2 and l=4]
-            tmp = f.readline()
-            tmp = [int(b) for b in tmp.split()]
-            self.Sp = list(tmp)
-            # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
-            # l_{n-1}]
-            tmp = f.readline()
-            tmp = [int(b) for b in tmp.split()]
-            self.Sm = list(tmp)
-            f.close()
+            file_path = importlib.resources.files('sibc').joinpath(file_path)
+            with importlib.resources.as_file(file_path) as path:
+                f = open(path)
+                # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
+                # l_{n-1}] [We need to include case l=2 and l=4]
+                tmp = f.readline()
+                tmp = [int(b) for b in tmp.split()]
+                self.Sp = list(tmp)
+                # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
+                # l_{n-1}]
+                tmp = f.readline()
+                tmp = [int(b) for b in tmp.split()]
+                self.Sm = list(tmp)
+                f.close()
         ######################################################################################################################
 
     def dynamic_programming_algorithm(self, L, n):
diff --git a/sibc/constants.py b/sibc/constants.py
index 5455caf..675f0d8 100644
--- a/sibc/constants.py
+++ b/sibc/constants.py
@@ -1,7 +1,7 @@
 import os
 from functools import reduce
 from .math import bitlength, is_prime
-from pkg_resources import resource_string, resource_filename
+import importlib.resources
 
 # When including a new csidh-prime, you need to update the next list
 # The csidh format in the following dictionary is
@@ -80,7 +80,8 @@ sidh_primes = (
 def csidh_get_sop_from_disk(prime):
     assert prime in csidh_primes.keys(), "unsupported prime for csidh"
     # List of Small odd primes, L := [l_0, ..., l_{n-1}]
-    L = resource_string(__name__, "data/sop/csidh/" + prime)
+    file_path = importlib.resources.files('sibc').joinpath("data/sop/csidh", prime)
+    L = file_path.read_bytes()
     L = [int(l) for l in L.split()]
     cofactor = L[0]
     L = list(L[1:])  #   Small Odd Primes l_i's
@@ -103,37 +104,40 @@ def csidh_get_sop_from_disk(prime):
 
 def csidh_get_exp_from_disk(exponent, style, security, attack, n):
 
-    m = resource_string(__name__, "data/exponents/" + attack + '/' + security + '/' + style + '/e' + exponent)
+    file_path = importlib.resources.files('sibc').joinpath("data/exponents", attack, security, style, "e" + exponent)
+    m = file_path.read_bytes()
     m  = [int(mi) for mi in m.split()]
     assert len(m) <= n, 'Not enough number of small odd prime numbers in the factorization of p + 1'
     return m + [0]*(n - len(m))
 
 def bsidh_get_sop_from_disk(prime):
     assert prime in bsidh_primes, "unsupported prime for bsidh"
-    f = open(resource_filename('sibc', 'data/sop/bsidh/'+ prime))
-
-    # The prime to be used
-    p = f.readline()
-    p = int(p, 16)
-
-    # List corresponding (p + 1)
-    Lp = f.readline()
-    Lp = [int(lp) for lp in Lp.split()]
-    # exponent_of_twop = Lp[0]
-    # Lp = Lp[1:]
-    Ep = f.readline()
-    Ep = [int(ep) for ep in Ep.split()]
-    assert len(Ep) == len(Lp)
-    np = len(Lp)
-
-    # List corresponding (p - 1)
-    Lm = f.readline()
-    Lm = [int(lm) for lm in Lm.split()]
-    Em = f.readline()
-    Em = [int(em) for em in Em.split()]
-    assert len(Em) == len(Lm)
-    nm = len(Lm)
-    f.close()
+    file_path = importlib.resources.files('sibc').joinpath('data/sop/bsidh', prime)
+    with importlib.resources.as_file(file_path) as path:
+        f = open(path)
+
+        # The prime to be used
+        p = f.readline()
+        p = int(p, 16)
+
+        # List corresponding (p + 1)
+        Lp = f.readline()
+        Lp = [int(lp) for lp in Lp.split()]
+        # exponent_of_twop = Lp[0]
+        # Lp = Lp[1:]
+        Ep = f.readline()
+        Ep = [int(ep) for ep in Ep.split()]
+        assert len(Ep) == len(Lp)
+        np = len(Lp)
+
+        # List corresponding (p - 1)
+        Lm = f.readline()
+        Lm = [int(lm) for lm in Lm.split()]
+        Em = f.readline()
+        Em = [int(em) for em in Em.split()]
+        assert len(Em) == len(Lm)
+        nm = len(Lm)
+        f.close()
     # pp = (2**exponent_of_twop) * reduce(lambda x,y : (x*y), [ Lp[i]**Ep[i] for i in range(0, np, 1)  ])
     pp = reduce(
         lambda x, y: (x * y), [Lp[i] ** Ep[i] for i in range(0, np, 1)]
@@ -167,27 +171,29 @@ def bsidh_get_sop_from_disk(prime):
 
 def sidh_get_sop_from_disk(prime):
     assert prime in sidh_primes, "unsupported prime for sidh/sike"
-    f = open(resource_filename('sibc', 'data/sop/sidh/'+ prime))
-
-    # p + 1 = 2^two x 3^three
-    Exponents = f.readline()
-    two, three = [int(e) for e in Exponents.split()]
-    p = (2 ** two) * (3 ** three) - 1
-
-    if p % 4 == 1:
-        print("// Case p = 1 mod 4 is not implemented yet!")
-        exit(-1)
-
-    p_minus_one_halves = (p - 1) // 2
-    p_minus_3_quarters = (p - 3) // 4
-    return dict(
-        two=two,
-        three=three,
-        p=p,
-        p_minus_one_halves=p_minus_one_halves,
-        p_bits=bitlength(p),
-        p_minus_3_quarters=p_minus_3_quarters,
-    )
+    file_path = importlib.resources.files('sibc').joinpath('data/sop/sidh', prime)
+    with importlib.resources.as_file(file_path) as path:
+        f = open(path)
+
+        # p + 1 = 2^two x 3^three
+        Exponents = f.readline()
+        two, three = [int(e) for e in Exponents.split()]
+        p = (2 ** two) * (3 ** three) - 1
+
+        if p % 4 == 1:
+            print("// Case p = 1 mod 4 is not implemented yet!")
+            exit(-1)
+
+        p_minus_one_halves = (p - 1) // 2
+        p_minus_3_quarters = (p - 3) // 4
+        return dict(
+            two=two,
+            three=three,
+            p=p,
+            p_minus_one_halves=p_minus_one_halves,
+            p_bits=bitlength(p),
+            p_minus_3_quarters=p_minus_3_quarters,
+        )
 
 parameters = dict(
     csidh=dict({
@@ -205,4 +211,4 @@ parameters = dict(
     sidh=dict({
         PRIME:dict(**sidh_get_sop_from_disk(PRIME)) for PRIME in sidh_primes
     })
-)
\ No newline at end of file
+)
diff --git a/sibc/csidh/bench.py b/sibc/csidh/bench.py
index 020dc94..5502923 100644
--- a/sibc/csidh/bench.py
+++ b/sibc/csidh/bench.py
@@ -1,4 +1,3 @@
-from pkg_resources import resource_filename
 import click
 
 from progress.bar import Bar
diff --git a/sibc/csidh/header.py b/sibc/csidh/header.py
index 1a63220..02186a0 100644
--- a/sibc/csidh/header.py
+++ b/sibc/csidh/header.py
@@ -2,7 +2,7 @@ import click
 import numpy
 
 from sibc.common import attrdict, geometric_serie, rounds, printl
-from pkg_resources import resource_filename
+import importlib.resources
 
 @click.command()
 @click.pass_context
@@ -57,16 +57,17 @@ def csidh_header(ctx):
         + algo.formula.multievaluation_name
         + algo.formula.tuned_name
     )
-    file_path = resource_filename('sibc', file_path)
-    f = open(file_path)
-    S_out = []
-    for i in range(0, len(r_out), 1):
+    file_path = importlib.resources.fies('sibc').joinpath(file_path)
+    with importlib.resources.as_file(file_path) as path:
+        f = open(path)
+        S_out = []
+        for i in range(0, len(r_out), 1):
 
-        tmp = f.readline()
-        tmp = [int(b) for b in tmp.split()]
-        S_out.append(tmp)
+            tmp = f.readline()
+            tmp = [int(b) for b in tmp.split()]
+            S_out.append(tmp)
 
-    f.close()
+        f.close()
 
     if (len(set(m)) == 1) or ((len(set(m)) == 2) and (0 in set(m))):
         L_out = list([L_out[0]])
diff --git a/sibc/csidh/main.py b/sibc/csidh/main.py
index 7ed3067..e27c6b8 100644
--- a/sibc/csidh/main.py
+++ b/sibc/csidh/main.py
@@ -1,4 +1,3 @@
-from pkg_resources import resource_filename
 import click
 
 from sibc.common import attrdict, geometric_serie, rounds
diff --git a/sibc/csidh/precompute_parameters.py b/sibc/csidh/precompute_parameters.py
index 758f12f..7b7cec1 100644
--- a/sibc/csidh/precompute_parameters.py
+++ b/sibc/csidh/precompute_parameters.py
@@ -1,6 +1,6 @@
 import sys
 import click
-from pkg_resources import resource_filename
+import importlib.resources
 import numpy
 
 from sibc.common import attrdict, printl
@@ -43,8 +43,10 @@ def csidh_precompute_parameters(ctx):
 
     original_stdout = sys.stdout # Save a reference to the original standard output
     multievaluation = {True:'scaled', False:'unscaled'}[setting.multievaluation]
-    path = resource_filename('sibc', "data/ijk/" + algo.curve.model + '/csidh/' + algo.curve.name + '-' + multievaluation)
-    with open(path, 'w') as f:
+    file_path = importlib.resources.files('sibc').joinpath("data/ijk", algo.curve.model, 'csidh', algo.curve.name + '-' + multievaluation)
+    with importlib.resources.as_file(file_path) as path:
+        f = open(path, "w")
+
         sys.stdout = f # Change the standard output to the file we created.
         parameters = dict()
         for idx in range(0, n, 1):
diff --git a/sibc/csidh/precompute_strategy.py b/sibc/csidh/precompute_strategy.py
index 058a994..f7241f6 100644
--- a/sibc/csidh/precompute_strategy.py
+++ b/sibc/csidh/precompute_strategy.py
@@ -1,4 +1,4 @@
-from pkg_resources import resource_filename
+import importlib.resources
 import click
 
 from sibc.common import attrdict, geometric_serie, rounds
@@ -65,16 +65,17 @@ def csidh_precompute_strategy(ctx):
         + multievaluation
         + tuned
     )
-    file_path = resource_filename('sibc', file_path)
-    print("// Strategies to be computed")
-    C_out, L_out, R_out, S_out, r_out = strategy_block_cost(
-        L[::-1], m[::-1]
-    )
-    f = open(file_path, 'w')
-    for i in range(0, len(r_out)):
+    file_path = importlib.resources.files('sibc').joinpath(file_path)
+    with importlib.resources.as_file(file_path) as path:
+        print("// Strategies to be computed")
+        C_out, L_out, R_out, S_out, r_out = strategy_block_cost(
+            L[::-1], m[::-1]
+        )
+        f = open(path, 'w')
+        for i in range(0, len(r_out)):
 
-        f.writelines(' '.join([str(tmp) for tmp in S_out[i]]) + '\n')
+            f.writelines(' '.join([str(tmp) for tmp in S_out[i]]) + '\n')
 
-    f.close()
+        f.close()
 
-    return attrdict(name='csidh-precompute-strategy', **locals())
\ No newline at end of file
+    return attrdict(name='csidh-precompute-strategy', **locals())
diff --git a/sibc/montgomery/curve.py b/sibc/montgomery/curve.py
index ab02734..a14b6cb 100644
--- a/sibc/montgomery/curve.py
+++ b/sibc/montgomery/curve.py
@@ -1,5 +1,5 @@
 from random import SystemRandom
-from pkg_resources import resource_filename
+import importlib.resources
 
 import numpy
 
@@ -41,7 +41,7 @@ def MontgomeryCurve(prime):
         p_minus_one_halves = parameters['csidh'][prime]['p_minus_one_halves']
         validation_stop = sum([bitlength(l_i) for l_i in L]) / 2.0 + 2
         field = PrimeField(p)
-        path = resource_filename('sibc', "data/sdacs/csidh/" + prime)
+        file_path = importlib.resources.files('sibc').joinpath("data/sdacs/csidh", prime)
 
     elif prime in parameters['bsidh'].keys():
         # B-SIDH only requires the factorization of p + 1 and p - 1
@@ -61,7 +61,7 @@ def MontgomeryCurve(prime):
         p_minus_one_halves = parameters['bsidh'][prime]['p_minus_one_halves']
         p_minus_3_quarters = parameters['bsidh'][prime]['p_minus_3_quarters']
         field = QuadraticField(p)
-        path = resource_filename('sibc', "data/sdacs/bsidh/" + prime)
+        file_path = importlib.resources.files('sibc').joinpath("data/sdacs/bsidh", prime)
 
     elif prime in parameters['sidh'].keys():
         # SIDH only requires the exponents two and three such that p + 1 = 2^two x 3^three
@@ -76,25 +76,26 @@ def MontgomeryCurve(prime):
         L = [3,4]
         n = 2
         montgomery_n = n
-        path = resource_filename('sibc', "data/sdacs/sidh")
+        file_path = importlib.resources.files('sibc').joinpath("data/sdacs/sidh")
 
     else:
         assert False, "only csidh, bsidh, and sidh are currently implemented"
 
     # Shortest Differential Addition Chains (SDACs) for each l_i
-    SDACS = filename_to_list_of_lists_of_ints(path)
-    assert len(SDACS) > 0, f'Not precomputed sdacs for {prime} prime'
-    SDACS_LENGTH = list(map(len, SDACS))
-    SDACS_REVERSED = list(map(lambda x:x[::-1], SDACS))
-
-    cmul = lambda l: numpy.array(
-        [
-            4.0 * (SDACS_LENGTH[L.index(l)] + 2),
-            2.0 * (SDACS_LENGTH[L.index(l)] + 2),
-            6.0 * (SDACS_LENGTH[L.index(l)] + 2) - 2.0,
-        ]
-    )
-    c_xmul = list(map(cmul, L))  # list of the costs of each [l]P
+    with importlib.resources.as_file(file_path) as path:
+        SDACS = filename_to_list_of_lists_of_ints(path)
+        assert len(SDACS) > 0, f'Not precomputed sdacs for {prime} prime'
+        SDACS_LENGTH = list(map(len, SDACS))
+        SDACS_REVERSED = list(map(lambda x:x[::-1], SDACS))
+
+        cmul = lambda l: numpy.array(
+            [
+                4.0 * (SDACS_LENGTH[L.index(l)] + 2),
+                2.0 * (SDACS_LENGTH[L.index(l)] + 2),
+                6.0 * (SDACS_LENGTH[L.index(l)] + 2) - 2.0,
+            ]
+        )
+        c_xmul = list(map(cmul, L))  # list of the costs of each [l]P
 
     SQR = 1.00
     ADD = 0.00
diff --git a/sibc/montgomery/isogeny.py b/sibc/montgomery/isogeny.py
index 075d4d8..2b4fba8 100644
--- a/sibc/montgomery/isogeny.py
+++ b/sibc/montgomery/isogeny.py
@@ -1,5 +1,5 @@
 from random import SystemRandom
-from pkg_resources import resource_filename
+import importlib.resources
 
 # Related polynomial multiplication procedures
 from sibc.polymul import PolyMul
@@ -44,23 +44,22 @@ def MontgomeryIsogeny(name : str, uninitialized = False):
                 self.sJ_list = []
 
                 if curve.name in parameters['csidh'].keys():
-                    algorithm = '/csidh'
+                    algorithm = 'csidh'
                 elif curve.name in parameters['bsidh'].keys():
-                    algorithm = '/bsidh'
+                    algorithm = 'bsidh'
                 else:
                     assert False, "only CSIDH and B-SIDH are currently implemented"
 
-                path = resource_filename('sibc', "data/ijk/" + curve.model + algorithm + '/' + curve.name + '-' + self.multievaluation_name)
-                f = open(path)
+                file_path=importlib.resources.files('sibc') .joinpath("data/ijk/", curve.model, algorithm, curve.name + '-' + self.multievaluation_name)
+                with importlib.resources.as_file(file_path) as path:
+                    f = open(path)
 
-                for i in range(0, curve.n, 1):
+                    for i in range(0, curve.n, 1):
 
-                    bc = f.readline()
-                    bc = [int(bci) for bci in bc.split()]
-                    self.sJ_list.append(bc[0])
-                    self.sI_list.append(bc[1])
-
-                f.close()
+                        bc = f.readline()
+                        bc = [int(bci) for bci in bc.split()]
+                        self.sJ_list.append(bc[0])
+                        self.sI_list.append(bc[1])
 
             else:
                 self.sI_list = None
@@ -1158,23 +1157,27 @@ def MontgomeryIsogeny(name : str, uninitialized = False):
                 random = SystemRandom()
 
                 # Reading public generators points
-                f = open(resource_filename('sibc', 'data/gen/' + self.curve.model + '/bsidh/' + self.curve.name))
-
-                # x(PA), x(QA) and x(PA - QA)
-                PQA = f.readline()
-                PQA = [int(x, 16) for x in PQA.split()]
-                PA = [self.field(PQA[0:2]), self.field(1)]
-                QA = [self.field(PQA[2:4]), self.field(1)]
-                PQA= [self.field(PQA[4:6]), self.field(1)]
-
-                # x(PB), x(QB) and x(PB - QB)
-                PQB = f.readline()
-                PQB = [int(x, 16) for x in PQB.split()]
-                PB = [self.field(PQB[0:2]), self.field(1)]
-                QB = [self.field(PQB[2:4]), self.field(1)]
-                PQB= [self.field(PQB[4:6]), self.field(1)]
-
-                f.close()
+                file_path = importlib.resources.files('sibc').joinpath('data/gen', self.curve.model, 'bsidh', self.curve.name)
+
+                with importlib.resources.as_file(file_path) as path:
+
+                    f = open(path)
+
+                    # x(PA), x(QA) and x(PA - QA)
+                    PQA = f.readline()
+                    PQA = [int(x, 16) for x in PQA.split()]
+                    PA = [self.field(PQA[0:2]), self.field(1)]
+                    QA = [self.field(PQA[2:4]), self.field(1)]
+                    PQA= [self.field(PQA[4:6]), self.field(1)]
+
+                    # x(PB), x(QB) and x(PB - QB)
+                    PQB = f.readline()
+                    PQB = [int(x, 16) for x in PQB.split()]
+                    PB = [self.field(PQB[0:2]), self.field(1)]
+                    QB = [self.field(PQB[2:4]), self.field(1)]
+                    PQB= [self.field(PQB[4:6]), self.field(1)]
+
+                    f.close()
 
                 # Three point ladder: case (p + 1)
                 A = [ self.field(8), self.field(4)]
diff --git a/sibc/plot_strategy.py b/sibc/plot_strategy.py
index 2266db1..e982f65 100644
--- a/sibc/plot_strategy.py
+++ b/sibc/plot_strategy.py
@@ -9,7 +9,7 @@ import matplotlib.pyplot as plt
 import networkx as nx
 
 from sibc.common import attrdict, strategy_evaluation, geometric_serie, rounds
-from pkg_resources import resource_filename
+import importlib.resources
 
 @click.command()
 @click.pass_context
@@ -58,16 +58,17 @@ def plot_strategy(ctx):
             + algo.formula.multievaluation_name
             + algo.formula.tuned_name
         )
-        file_path = resource_filename('sibc', file_path)
-        f = open(file_path)
-        S_out = []
-        for i in range(0, len(r_out), 1):
+        file_path = importlib.resources.files('sibc').joinpath(filepath)
+        with importlib.resources.as_file(file_path) as path:
+            f = open(path)
+            S_out = []
+            for i in range(0, len(r_out), 1):
 
-            tmp = f.readline()
-            tmp = [int(b) for b in tmp.split()]
-            S_out.append(tmp)
+                tmp = f.readline()
+                tmp = [int(b) for b in tmp.split()]
+                S_out.append(tmp)
 
-        f.close()
+            f.close()
 
         file_path = (
             'csidh'
@@ -101,20 +102,21 @@ def plot_strategy(ctx):
             + algo.formula.multievaluation_name
             + algo.formula.tuned_name
         )
-        file_path = resource_filename('sibc', file_path)
-        f = open(file_path)
-        S_out = []
-        # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
-        # l_{n-1}]
-        tmp = f.readline()
-        tmp = [int(b) for b in tmp.split()]
-        S_out.append(list(tmp))
-        # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
-        # l_{n-1}]
-        tmp = f.readline()
-        tmp = [int(b) for b in tmp.split()]
-        S_out.append(list(tmp))
-        f.close()
+        file_path = importlib.resources.files('sibc').joinpath(file_path)
+        with importlib.resources.as_file(file_path) as path:
+            f = open(path)
+            S_out = []
+            # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
+            # l_{n-1}]
+            tmp = f.readline()
+            tmp = [int(b) for b in tmp.split()]
+            S_out.append(list(tmp))
+            # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
+            # l_{n-1}]
+            tmp = f.readline()
+            tmp = [int(b) for b in tmp.split()]
+            S_out.append(list(tmp))
+            f.close()
 
         file_path = (
             'bsidh'
@@ -140,18 +142,19 @@ def plot_strategy(ctx):
                 + '-'
                 + setting.prime
         )
-        file_path = resource_filename('sibc', file_path)
-        f = open(file_path)
-        S_out = []
-        # Corresponding to 2-isogenies
-        tmp = f.readline()
-        tmp = [int(b) for b in tmp.split()]
-        S_out.append(list(tmp))
-        # Corresponding to 3-isogenies
-        tmp = f.readline()
-        tmp = [int(b) for b in tmp.split()]
-        S_out.append(list(tmp))
-        f.close()
+        file_path = importlib.resources.files('sibc').joinpath(file_path)
+        with importlib.resources.as_file(file_path) as path:
+            f = open(path)
+            S_out = []
+            # Corresponding to 2-isogenies
+            tmp = f.readline()
+            tmp = [int(b) for b in tmp.split()]
+            S_out.append(list(tmp))
+            # Corresponding to 3-isogenies
+            tmp = f.readline()
+            tmp = [int(b) for b in tmp.split()]
+            S_out.append(list(tmp))
+            f.close()
 
         file_path = (
                 'sidh'
diff --git a/sibc/precompute_sdacs.py b/sibc/precompute_sdacs.py
index 931dd5b..26365b4 100644
--- a/sibc/precompute_sdacs.py
+++ b/sibc/precompute_sdacs.py
@@ -1,7 +1,7 @@
 import math
 import click
 from click.exceptions import Exit
-from pkg_resources import resource_string, resource_filename
+import importlib.resources
 
 def write_list_of_lists_of_ints_to_file(path, data):
     with open(path, 'w') as fh:
@@ -28,46 +28,47 @@ def write_list_of_lists_of_ints_to_file(path, data):
 def main(prime, algorithm):
     assert prime != None, 'argument --prime is required'
     """ Computing and storing sdacs """
-    f = open(resource_filename('sibc', 'data/sop/' + algorithm + '/' + prime))
-    if algorithm == 'csidh':
-        # CSIDH only requires the factorization of p + 1
-        L = f.readline()
-        # The first value in L corresponds with the cofactor h of (p+1), which is not required here
-        L = [int(l) for l in L.split()][1:]
-        n = len(L)
-
-    elif algorithm == 'bsidh':
-        # B-SIDH only requires the factorization of p + 1 and p - 1
-        # The prime to be used
-        p = f.readline()
-        p = int(p, 16)
-
-        # List corresponding (p + 1)
-        Lp = f.readline()
-        Lp = [int(lp) for lp in Lp.split()]
-        # exponent_of_twop = Lp[0]
-        # Lp = Lp[1:]
-        Ep = f.readline()
-        Ep = [int(ep) for ep in Ep.split()]
-        assert len(Ep) == len(Lp)
-        np = len(Lp)
-
-        # List corresponding (p - 1)
-        Lm = f.readline()
-        Lm = [int(lm) for lm in Lm.split()]
-        Em = f.readline()
-        Em = [int(em) for em in Em.split()]
-        assert len(Em) == len(Lm)
-        nm = len(Lm)
-
-        L = list(Lp + Lm)
-        n = len(L)
-
-    else:
-        click.echo("only csidh and bsidh are currently implemented")
-        raise Exit(1)
-
-    f.close()
+    file_path = importlib.resources.files('sibc').joinpath('data/sop', algorithm, prime)
+    with importlib.resources.as_file(file_path) as path:
+        f = open(path)
+        if algorithm == 'csidh':
+            # CSIDH only requires the factorization of p + 1
+            L = f.readline()
+            # The first value in L corresponds with the cofactor h of (p+1), which is not required here
+            L = [int(l) for l in L.split()][1:]
+            n = len(L)
+
+        elif algorithm == 'bsidh':
+            # B-SIDH only requires the factorization of p + 1 and p - 1
+            # The prime to be used
+            p = f.readline()
+            p = int(p, 16)
+
+            # List corresponding (p + 1)
+            Lp = f.readline()
+            Lp = [int(lp) for lp in Lp.split()]
+            # exponent_of_twop = Lp[0]
+            # Lp = Lp[1:]
+            Ep = f.readline()
+            Ep = [int(ep) for ep in Ep.split()]
+            assert len(Ep) == len(Lp)
+            np = len(Lp)
+
+            # List corresponding (p - 1)
+            Lm = f.readline()
+            Lm = [int(lm) for lm in Lm.split()]
+            Em = f.readline()
+            Em = [int(em) for em in Em.split()]
+            assert len(Em) == len(Lm)
+            nm = len(Lm)
+
+            L = list(Lp + Lm)
+            n = len(L)
+
+        else:
+            click.echo("only csidh and bsidh are currently implemented")
+            raise Exit(1)
+        f.close()
 
     def dacs(l, r0, r1, r2, chain):
         """
@@ -106,11 +107,12 @@ def main(prime, algorithm):
         )
 
     # Shortest Differential Addition Chains (SDACs) for each l_i
-    path = resource_filename('sibc', "data/sdacs/" + algorithm + '/' + prime)
-    print("// Computing sdacs")
-    SDACS = generate_sdacs(L)
-    print("// Storing sdacs into a file")
-    write_list_of_lists_of_ints_to_file(path, SDACS)
+    file_path = importlib.resources.files('sibc').joinpath("data/sdacs", algorithm, prime)
+    with importlib.resources.as_file(file_path) as path:
+        print("// Computing sdacs")
+        SDACS = generate_sdacs(L)
+        print("// Storing sdacs into a file")
+        write_list_of_lists_of_ints_to_file(path, SDACS)
 
 if __name__ == "__main__":
-    main()
\ No newline at end of file
+    main()
diff --git a/sibc/sidh/precompute_strategy.py b/sibc/sidh/precompute_strategy.py
index 4d9035d..871ec6b 100644
--- a/sibc/sidh/precompute_strategy.py
+++ b/sibc/sidh/precompute_strategy.py
@@ -1,5 +1,5 @@
 import click
-from pkg_resources import resource_filename
+import importlib.resources
 
 from sibc.common import attrdict
 
@@ -23,16 +23,17 @@ def sidh_precompute_strategy(ctx):
         + '-'
         + setting.prime
     )
-    file_path = resource_filename('sibc', file_path)
-    print("// Strategies to be computed")
-    # List of Small Isogeny Degree, Lp := [l_0, ..., l_{n-1}] [We need to
-    # include case l=2 and l=4]
-    S2, C2 = algo.strategy.dynamic_programming_algorithm(2, e2)
-    # List of Small Isogeny Degree, Lm := [l_0, ..., l_{n-1}]
-    S3, C3 = algo.strategy.dynamic_programming_algorithm(3, e3)
-    f = open(file_path, 'w')
-    f.writelines(' '.join([str(tmp) for tmp in S2]) + '\n')
-    f.writelines(' '.join([str(tmp) for tmp in S3]) + '\n')
-    f.close()
+    file_path = importlib.resources.files('sibc').joinpath(file_path)
+    with importlib.resources.as_file(file_path) as path:
+        print("// Strategies to be computed")
+        # List of Small Isogeny Degree, Lp := [l_0, ..., l_{n-1}] [We need to
+        # include case l=2 and l=4]
+        S2, C2 = algo.strategy.dynamic_programming_algorithm(2, e2)
+        # List of Small Isogeny Degree, Lm := [l_0, ..., l_{n-1}]
+        S3, C3 = algo.strategy.dynamic_programming_algorithm(3, e3)
+        f = open(path, 'w')
+        f.writelines(' '.join([str(tmp) for tmp in S2]) + '\n')
+        f.writelines(' '.join([str(tmp) for tmp in S3]) + '\n')
+        f.close()
 
-    return attrdict(name='sidh-precompute-strategy', **locals())
\ No newline at end of file
+    return attrdict(name='sidh-precompute-strategy', **locals())
diff --git a/sibc/sidh/strategy.py b/sibc/sidh/strategy.py
index 0cbade8..86a7b7c 100644
--- a/sibc/sidh/strategy.py
+++ b/sibc/sidh/strategy.py
@@ -1,5 +1,5 @@
 from random import SystemRandom
-from pkg_resources import resource_filename
+import importlib.resources
 
 from functools import reduce
 from math import floor, sqrt
@@ -21,23 +21,25 @@ class Strategy(object):
         self.three = self.curve.three
 
         # Reading public generators points
-        f = open(resource_filename('sibc', 'data/gen/' + curve.model + '/sidh/' + prime))
+        file_path = importlib.resources.files('sibc').joinpath("data/gen", curve.model, "sidh", prime)
+        with importlib.resources.as_file(file_path) as path:
+            f = open(path)
 
-        # x(PA), x(QA) and x(PA - QA)
-        PQA = f.readline()
-        PQA = [int(x, 16) for x in PQA.split()]
-        self.PA = self.field(PQA[0:2])
-        self.QA = self.field(PQA[2:4])
-        self.PQA= self.field(PQA[4:6])
+            # x(PA), x(QA) and x(PA - QA)
+            PQA = f.readline()
+            PQA = [int(x, 16) for x in PQA.split()]
+            self.PA = self.field(PQA[0:2])
+            self.QA = self.field(PQA[2:4])
+            self.PQA= self.field(PQA[4:6])
 
-        # x(PB), x(QB) and x(PB - QB)
-        PQB = f.readline()
-        PQB = [int(x, 16) for x in PQB.split()]
-        self.PB = self.field(PQB[0:2])
-        self.QB = self.field(PQB[2:4])
-        self.PQB= self.field(PQB[4:6])
+            # x(PB), x(QB) and x(PB - QB)
+            PQB = f.readline()
+            PQB = [int(x, 16) for x in PQB.split()]
+            self.PB = self.field(PQB[0:2])
+            self.QB = self.field(PQB[2:4])
+            self.PQB= self.field(PQB[4:6])
 
-        f.close()
+            f.close()
         # To check the stored strategies format/name-file
         if not formula.uninitialized:
             file_path = (
@@ -48,19 +50,20 @@ class Strategy(object):
                 + '-'
                 + prime
             )
-            file_path = resource_filename('sibc', file_path)
-            f = open(file_path)
-            # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
-            # l_{n-1}] [We need to include case l=2 and l=4]
-            tmp = f.readline()
-            tmp = [int(b) for b in tmp.split()]
-            self.S2 = list(tmp)
-            # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
-            # l_{n-1}]
-            tmp = f.readline()
-            tmp = [int(b) for b in tmp.split()]
-            self.S3 = list(tmp)
-            f.close()
+            file_path = importlib.resources.files('sibc').joinpath(file_path)
+            with importlib.resources.as_file(file_path) as path:
+                f = open(file_path)
+                # Corresponding to the list of Small Isogeny Degree, Lp := [l_0, ...,
+                # l_{n-1}] [We need to include case l=2 and l=4]
+                tmp = f.readline()
+                tmp = [int(b) for b in tmp.split()]
+                self.S2 = list(tmp)
+                # Corresponding to the list of Small Isogeny Degree, Lm := [l_0, ...,
+                # l_{n-1}]
+                tmp = f.readline()
+                tmp = [int(b) for b in tmp.split()]
+                self.S3 = list(tmp)
+                f.close()
         ######################################################################################################################
 
     def dynamic_programming_algorithm(self, ell, e):
