diff --git a/tests/unit/test_library.py b/tests/unit/test_library.py
index fb92f92..483c758 100644
--- a/tests/unit/test_library.py
+++ b/tests/unit/test_library.py
@@ -987,5 +987,5 @@ class TestLibrary(unittest.TestCase):
         # JLinkarmDlInfo has not been instantiated.
-        self.assertEquals(0, mock_dlinfo_ctr.call_count)
+        self.assertEqual(0, mock_dlinfo_ctr.call_count)
         # Fallback to "search by file name" has succeeded.
-        self.assertEquals(1, mock_load_library.call_count)
+        self.assertEqual(1, mock_load_library.call_count)
         self.assertEqual(directories[0], lib._path)
@@ -1031,5 +1031,5 @@ class TestLibrary(unittest.TestCase):
         mock_find_library.assert_any_call('dl')
-        self.assertEquals(2, mock_find_library.call_count)
+        self.assertEqual(2, mock_find_library.call_count)
         # Called once in JLinkarmDlInfo and once in Library.
-        self.assertEquals(2, mock_load_library.call_count)
+        self.assertEqual(2, mock_load_library.call_count)
         # The dlinfo() dance silently failed, but will answer None resolved path.
@@ -1072,4 +1072,4 @@ class TestLibrary(unittest.TestCase):
         mock_find_library.assert_any_call('dl')
-        self.assertEquals(2, mock_find_library.call_count)
-        self.assertEquals(2, mock_load_library.call_count)
+        self.assertEqual(2, mock_find_library.call_count)
+        self.assertEqual(2, mock_load_library.call_count)
 
