diff --git a/pkg/ide/vscode/open.go b/pkg/ide/vscode/open.go
index 18aae60d..927c1cce 100644
--- a/pkg/ide/vscode/open.go
+++ b/pkg/ide/vscode/open.go
@@ -122,6 +122,8 @@ func findCLI(flavor Flavor) string {
 	if flavor == FlavorStable {
 		if command.Exists("code") {
 			return "code"
+		} else if command.Exists("code-oss") {
+			return "code-oss"
 		} else if runtime.GOOS == "darwin" && command.Exists("/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code") {
 			return "/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code"
 		}
