From 6898b9e9b479f58e44832b60867beab9ee909e51 Mon Sep 17 00:00:00 2001
From: Leon Marz <lmarz@cs.uni-frankfurt.de>
Date: Wed, 9 Dec 2020 21:14:50 +0100
Subject: [PATCH] add missing header

---
 openvdb/openvdb/TypeList.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/openvdb/openvdb/TypeList.h b/openvdb/openvdb/TypeList.h
index d2c5764..fc552ee 100644
--- a/openvdb/openvdb/TypeList.h
+++ b/openvdb/openvdb/TypeList.h
@@ -15,6 +15,7 @@
 #include "version.h"
 
 #include <tuple>
+#include <cstddef>
 #include <type_traits>
 
 namespace openvdb {
-- 
2.29.2

