Index: src/UI/Widgets/VideoPlayer.cpp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/UI/Widgets/VideoPlayer.cpp b/src/UI/Widgets/VideoPlayer.cpp
--- a/src/UI/Widgets/VideoPlayer.cpp	(revision 0fd18b756a3fde8686ef562122b82ceabd959189)
+++ b/src/UI/Widgets/VideoPlayer.cpp	(date 1650584978197)
@@ -68,7 +68,7 @@
         seek_backwards_button->add_css_class("video-control");
         seek_backwards_button->signal_clicked().connect([this]()
         {
-            m_media_file->seek(std::max(m_media_file->get_timestamp() - (G_USEC_PER_SEC * 10), 0l));
+            m_media_file->seek(std::max(m_media_file->get_timestamp() - (G_USEC_PER_SEC * 10), (gint64)0));
             set_controls_visible(true, m_media_file->get_playing());
         });
         buttons_box->append(*seek_backwards_button);
