From 618fc59553bf8d15254e6d695908c982fbd21dae Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Sun, 23 Aug 2020 14:09:54 +0200
Subject: [PATCH 1/2] Remove debian-isms from test runner

---
 tests/unit/runtest.sh | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/tests/unit/runtest.sh b/tests/unit/runtest.sh
index c19682df0..c426fe4d7 100755
--- a/tests/unit/runtest.sh
+++ b/tests/unit/runtest.sh
@@ -88,13 +88,7 @@ function create_test_cmd {
   EXE=$(abspath $_TARGETPATH)
   _CMD="-n $_TESTFILE -m 500"
 
-  DEB_HOST_ARCH=$(dpkg-architecture -qDEB_HOST_ARCH)
-  if [[ ${DEB_HOST_ARCH} =~ 'arm' ]]; then
-    _CMD="dbus-test-runner --task $EXE $_CMD"
-  else
-    _CMD="dbus-test-runner --task gdb -p --quiet $_CMD"
-    _CMD="$_CMD -p --batch -p -ex -p 'set print thread-events off' -p -ex -p run -p -ex -p bt -p --return-child-result -p --args -p $EXE"
-  fi
+  _CMD="dbus-test-runner --task $EXE $_CMD"
 
   if [[ 'minimal custom' == *$_MINIMAL* ]]; then
       _CMD="$_CMD -p -platform -p $_MINIMAL"
-- 
2.31.1

