From 481eb1c21cd60455737fa288a62941bd0aa12350 Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Tue, 15 Sep 2020 17:48:47 +0200
Subject: [PATCH 2/2] Don't build app-launch-profiler

---
 lomiri-sdk.pro | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lomiri-sdk.pro b/lomiri-sdk.pro
index a180b037f..e987f2b30 100644
--- a/lomiri-sdk.pro
+++ b/lomiri-sdk.pro
@@ -11,7 +11,7 @@ load(qt_parts)
 src_uitk_launcher.subdir = lomiri-ui-toolkit-launcher
 src_uitk_launcher.depends = sub-src
 
-SUBDIRS += po app-launch-profiler src_uitk_launcher apicheck
+SUBDIRS += po src_uitk_launcher apicheck
 !CONFIG(no_docs) {
     SUBDIRS += documentation
 }
-- 
2.31.1

