--- a/src/makefile
+++ b/src/makefile
@@ -1,8 +1,7 @@
 #  compiler and flags
 CC ?=		cc
 PERL ?= perl
-QUICKJS_INCLUDE ?= ../../quickjs
-QUICKJS_LIB ?= ../../quickjs
+QUICKJS_INCLUDE ?= /usr/include/quickjs
 CFLAGS +=	-Wall -Wno-unused -D_FILE_OFFSET_BITS=64
 ifeq ($(shell uname),Linux)
 	PLATFORM_CFLAGS = -DEDBROWSE_ON_LINUX
@@ -35,7 +34,7 @@
 LDFLAGS = $(STRIP) $(LINKER_LIBS) -lpthread -lm -lssl -lcrypto
 
 # LDFLAGS for quickjs loading.
-QUICKJS_LDFLAGS = $(QUICKJS_LIB)/libquickjs.a -ldl
+QUICKJS_LDFLAGS = -lquickjs
 ifeq ($(shell uname),Linux)
 	QUICKJS_LDFLAGS += -latomic
 endif
