--- hdf-4.2.15.orig/hdf/test/buffer.c
+++ hdf-4.2.15/hdf/test/buffer.c
@@ -46,6 +46,7 @@
 
 #define TESTMASTER
 
+#include <time.h>
 #include "hdf.h"
 #ifdef H4_HAVE_SYS_TIME_H
 #include <sys/time.h>
--- hdf-4.2.15.orig/hdf/util/he_file.c
+++ hdf-4.2.15/hdf/util/he_file.c
@@ -17,9 +17,7 @@
 #include "he.h"
 
 /* get the prototype for the wait() func. */
-#if defined SUN | defined HP9000 | defined IRIX | defined UNIX386
 #include <sys/wait.h>
-#endif /* SUN | HP9000 */
 #ifdef H4_HAVE_UNISTD_H
 #include <unistd.h>
 #endif
@@ -125,7 +123,7 @@
     if (fork() == 0)
       {
           /* this is the child */
-          if (execl(editor, editor, file, 0) == -1)
+          if (execl(editor, editor, file, NULL) == -1)
               fprintf(stderr, "Error while editing label.\n");
 
           /* return control to the parent if exec fails
@@ -134,7 +132,7 @@
       }
 
     /* the parent waits for the child to die */
-    wait(0);
+    wait(NULL);
 #elif defined(H4_HAVE_SYSTEM)
     {   char    cmd[256];
         if (HDstrlen(editor) > 100) {
--- hdf-4.2.15.orig/mfhdf/libsrc/file.c
+++ hdf-4.2.15/mfhdf/libsrc/file.c
@@ -20,6 +20,7 @@
 #include <assert.h>
 #endif /* DEBUG */
 
+#include <unistd.h>
 #include    <string.h>
 #include    <errno.h>
 #include    "local_nc.h"
--- hdf-4.2.15.orig/mfhdf/test/hdftest.c
+++ hdf-4.2.15/mfhdf/test/hdftest.c
@@ -59,7 +59,7 @@
 extern int test_coordvar();
 extern int test_chunk();
 extern int test_compression();
-extern int test_dimension();
+extern int test_dimensions();
 extern int test_attributes();
 extern int test_datasizes();
 extern int test_datainfo();
--- hdf-4.2.15.orig/mfhdf/test/tncvargetfill.c
+++ hdf-4.2.15/mfhdf/test/tncvargetfill.c
@@ -33,6 +33,8 @@
 #include "mfhdf.h"
 #include "hdftest.h"
 
+extern int verify_info_data(int32 sds_id, int32 expected_dimsize, int16 *result);
+
 #define DIM0 20	/* this value is used to declare larger buffer for when
 		   reading past the maximum number of records in the file */
 
