--- epson-inkjet-printer-escpr-1.8.5-origin/lib/epson-escpr-api.h
+++ epson-inkjet-printer-escpr-1.8.5/lib/epson-escpr-api.h
@@ -82,7 +82,12 @@
 extern EPS_ERR_CODE epsSetAdditionalData (EPS_UINT32, const void*                       );
 extern EPS_ERR_CODE epsRemAdditionalData (EPS_UINT32                                    );
 
-    /*** Find Printer Function                                                          */
+/* Expose additional functions */
+extern EPS_ERR_CODE SetupJobAttrib          (const EPS_JOB_ATTRIB*                      );
+extern EPS_ERR_CODE SendStartJob            (EPS_BOOL                                   );
+extern EPS_ERR_CODE PrintBand               (const EPS_UINT8*, EPS_UINT32, EPS_UINT32*  );
+
+
     /*** -------------------------------------------------------------------------------*/
 extern EPS_ERR_CODE epsFindPrinter       (EPS_INT32, EPS_UINT32                         );
 extern EPS_ERR_CODE epsProbePrinter      (const EPS_PROBE*                              );
--- epson-inkjet-printer-escpr-1.8.5-origin/src/filter.c
+++ epson-inkjet-printer-escpr-1.8.5/src/filter.c
@@ -33,6 +33,7 @@
 #include "epson-protocol.h"
 #include "epson-escpr-api.h"
 #include "epson-escpr-mem.h"
+#include "epson-escpr-services.h"
 
 #include "err.h"
 #include "mem.h"
--- epson-inkjet-printer-escpr-1.8.5-origin/src/mem.c
+++ epson-inkjet-printer-escpr-1.8.5/src/mem.c
@@ -22,6 +22,7 @@
 #endif
 
 #include <stdlib.h>
+#include "err.h"
 #include "mem.h"
 
 void *
