From daa10c8af6031f10168639b7fd3c181a5d788ee1 Mon Sep 17 00:00:00 2001
From: Stephen Hemminger <stephen.hemminger@vyatta.com>
Date: Mon, 23 Aug 2010 08:14:38 -0700
Subject: [PATCH 3/3] Snapshot for 2.6.35.1

---
 include/SNAPSHOT.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/include/SNAPSHOT.h b/include/SNAPSHOT.h
index de41b1b..8f79884 100644
--- a/include/SNAPSHOT.h
+++ b/include/SNAPSHOT.h
@@ -1 +1 @@
-static const char SNAPSHOT[] = "100804";
+static const char SNAPSHOT[] = "100823";
-- 
1.7.1

