--- src/packet.c.orig	2008-03-07 17:56:48.000000000 +0200
+++ src/packet.c	2008-03-07 18:05:41.000000000 +0200
@@ -101,8 +101,8 @@
             result = LINK_ETHERNET;
         else if (strncmp(ifname, "tun", 3) == 0)
             result = LINK_ETHERNET;
-        else if (strncmp(ifname, "vlan", 3) == 0)
-            result = LINK_VLAN;
+        else if (strncmp(ifname, "vlan", 4) == 0)
+            result = LINK_ETHERNET;
         else if (strncmp(ifname, "brg", 3) == 0)
             result = LINK_ETHERNET;
         else if (strncmp(ifname, "tap", 3) == 0)
@@ -231,14 +231,6 @@
     case LINK_IPIP:
         *packet = tpacket;
         break;
-    case LINK_VLAN:
-        *packet = tpacket + VLAN_ETH_HLEN;
-        readlen -= VLAN_ETH_HLEN;
-        /*
-         * Move IP datagram into an aligned buffer.
-         */
-        memmove(aligned_buf, *packet, min(SNAPSHOT_LEN, *readlen));
-        *packet = aligned_buf;
     default:
         *packet = (char *) NULL;        /* return a NULL packet to signal */
         break;                  /* an unrecognized link protocol */
