--- uClibc-0.9.33.2.orig/utils/ldd.c
+++ uClibc-0.9.33.2/utils/ldd.c
@@ -827,7 +827,7 @@
 			for (cur = lib_list; cur; cur = cur->next) {
 				if (cur->resolved == 0 && cur->path) {
 					got_em_all = 1;
-					printf("checking sub-depends for '%s'\n", cur->path);
+					fprintf(stderr, "checking sub-depends for '%s'\n", cur->path);
 					find_dependencies(cur->path);
 					cur->resolved = 1;
 				}
