diff --git a/libc/inet/getproto.c b/libc/inet/getproto.c
index bcf507b..9858900 100644
--- a/libc/inet/getproto.c
+++ b/libc/inet/getproto.c
@@ -28,7 +28,7 @@ aliases: case sensitive optional space or tab separated list of other names
 __UCLIBC_MUTEX_STATIC(mylock, PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP);
 
 #define	MAXALIASES	35
-#define BUFSZ		(80) /* one line */
+#define BUFSZ		(256) /* one line */
 #define SBUFSIZE	(BUFSZ + 1 + (sizeof(char *) * MAXALIASES))
 
 static parser_t *protop = NULL;
diff --git a/libc/inet/getservice.c b/libc/inet/getservice.c
index c38ff80..dbbc19c 100644
--- a/libc/inet/getservice.c
+++ b/libc/inet/getservice.c
@@ -29,7 +29,7 @@ aliases: case sensitive optional space or tab separated list of other names
 __UCLIBC_MUTEX_STATIC(mylock, PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP);
 
 #define	MAXALIASES	35
-#define BUFSZ		(80) /* one line */
+#define BUFSZ		(256) /* one line */
 #define SBUFSIZE	(BUFSZ + 1 + (sizeof(char *) * MAXALIASES))
 
 static parser_t *servp = NULL;
