From 0c71b7626655c1023c5e10ebb3b1a294f8e41353 Mon Sep 17 00:00:00 2001
From: William Pitcock <nenolod@dereferenced.org>
Date: Tue, 1 Nov 2011 18:55:25 -0500
Subject: [PATCH 2/2] libm: add cabsf() and cabsl() functions.

Signed-off-by: William Pitcock <nenolod@dereferenced.org>
---
 libm/w_cabs.c |   16 ++++++++++++++++
 1 files changed, 16 insertions(+), 0 deletions(-)

diff --git a/libm/w_cabs.c b/libm/w_cabs.c
index 4044f4b..b259248 100644
--- a/libm/w_cabs.c
+++ b/libm/w_cabs.c
@@ -13,3 +13,19 @@ double cabs(double _Complex z)
 	return hypot(__real__ z, __imag__ z);
 }
 libm_hidden_def(cabs)
+
+libm_hidden_proto(cabsf)
+float cabsf(float _Complex z)
+{
+	return (float) hypot(__real__ z, __imag__ z);
+}
+libm_hidden_def(cabsf)
+
+#if defined __UCLIBC_HAS_LONG_DOUBLE_MATH__ && !defined __NO_LONG_DOUBLE_MATH
+libm_hidden_proto(cabsl)
+long double cabsl(long double _Complex z)
+{
+	return hypotl(__real__ z, __imag__ z);
+}
+libm_hidden_def(cabsl)
+#endif
-- 
1.7.7.1

