--- uClibc-0.9.32-rc2/libc/sysdeps/linux/common/ssp.c
+++ uClibc-0.9.32-rc2.mod/libc/sysdeps/linux/common/ssp.c
@@ -86,6 +86,7 @@
 }
 
 void __stack_chk_fail(void) attribute_noreturn __cold;
+libc_hidden_proto(__stack_chk_fail)
 void __stack_chk_fail(void)
 {
 	static const char msg1[] = "stack smashing detected: ";
@@ -99,8 +100,10 @@
 	while(1)
 		terminate();
 }
+libc_hidden_def(__stack_chk_fail)
 
 void __chk_fail(void) attribute_noreturn;
+libc_hidden_proto(__chk_fail)
 void __chk_fail(void)
 {
 	static const char msg1[] = "buffer overflow detected: ";
@@ -114,4 +117,5 @@
 	while(1)
 		terminate();
 }
+libc_hidden_def(__chk_fail)
 
