diff --git a/libc/sysdeps/linux/i386/Makefile.arch b/libc/sysdeps/linux/i386/Makefile.arch
index b47f5a0..8b1edc2 100644
--- a/libc/sysdeps/linux/i386/Makefile.arch
+++ b/libc/sysdeps/linux/i386/Makefile.arch
@@ -18,4 +18,7 @@ endif
 
 ifeq ($(UCLIBC_HAS_CONTEXT_FUNCS),y)
 SSRC += makecontext.S setcontext.S getcontext.S swapcontext.S
+ASFLAGS-setcontext.S := -D_LIBC_REENTRANT
+ASFLAGS-getcontext.S := -D_LIBC_REENTRANT
+ASFLAGS-swapcontext.S := -D_LIBC_REENTRANT
 endif
