--- a/tests/conninfo.tcl
+++ b/tests/conninfo.tcl
@@ -4,8 +4,8 @@
 array set conninfo {
     host    localhost
     port    5432
-    dbname  ####
-    user    ####
+    dbname  pgtcl
+    user    @USER@
 }
 
 # Or copy and fill out the above datastructure into ~/.conninfo
--- a/tests/pgtcl.test
+++ b/tests/pgtcl.test
@@ -1045,7 +1045,7 @@
     lappend res [string equal $val $val2]
     ::pg::disconnect $conn
 
-    lappend res [regexp {(7|8|9|10|11|12)0([0-9])[012]([0-9])} $val]
+    lappend res [regexp {(7|8|9|10|11|12|13|14)0([0-9])[012]([0-9])} $val]
 
 } -result [list 1 1]
 
@@ -1079,9 +1079,9 @@
     lappend res [string equal $val $val2]
     ::pg::disconnect $conn
 
-    set match [regexp {(7|8|9|10|11|12)\.([0-9])\.([0-9])} $val]
+    set match [regexp {(7|8|9|10|11|12|13|14)\.([0-9])\.([0-9])} $val]
     if {!$match} {
-	set match [regexp {^(10|11|12)\.[1-9][0-9]* \(.*} $val]
+	set match [regexp {^(10|11|12|13|14)\.[1-9][0-9]*} $val]
     }
     lappend res $match
 
--- a/tests/import_sampledata.tcl
+++ b/tests/import_sampledata.tcl
@@ -4,6 +4,8 @@
 # $Id$
 #
 
+lappend auto_path ..
+
 package require Pgtcl
 
 if {[file exists conninfo.tcl]} {
--- a/tests/import_sampledata_connobj.tcl
+++ b/tests/import_sampledata_connobj.tcl
@@ -5,6 +5,8 @@
 # $Id$
 #
 
+lappend auto_path ..
+
 package require Pgtcl
 
 if {[file exists conninfo.tcl]} {
--- a/tests/import_sampledata_onetransaction.tcl
+++ b/tests/import_sampledata_onetransaction.tcl
@@ -5,6 +5,8 @@
 # $Id$
 #
 
+lappend auto_path ..
+
 package require Pgtcl
 
 if {[file exists conninfo.tcl]} {
--- a/tests/import_sampledata_prepared.tcl
+++ b/tests/import_sampledata_prepared.tcl
@@ -5,6 +5,8 @@
 # $Id$
 #
 
+lappend auto_path ..
+
 package require Pgtcl
 
 if {[file exists conninfo.tcl]} {
--- a/tests/import_sampledata_prepared_onetransaction.tcl
+++ b/tests/import_sampledata_prepared_onetransaction.tcl
@@ -5,6 +5,8 @@
 # $Id$
 #
 
+lappend auto_path ..
+
 package require Pgtcl
 
 if {[file exists conninfo.tcl]} {
