diff --git a/src/findlib/fstype.c b/src/findlib/fstype.c
index ab3b206..21f5754 100644
--- a/src/findlib/fstype.c
+++ b/src/findlib/fstype.c
@@ -46,6 +46,10 @@
 
 #define is_rootfs(x) bstrcmp("rootfs", x)
 
+#ifndef _PATH_MOUNTED
+#define _PATH_MOUNTED "/proc/mounts"
+#endif /* _PATH_MOUNTED */
+
 #if defined(HAVE_GETMNTINFO) || defined(HAVE_GETMNTENT)
 static pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
 #endif
