--- bacula-5.0.1/src/qt-console/bat.pro.in~	2010-02-24 09:33:48.000000000 -0600
+++ bacula-5.0.1/src/qt-console/bat.pro.in	2010-03-11 17:12:44.645296544 -0600
@@ -20,7 +20,7 @@
 TARGET       = bat
 DEPENDPATH  += .
 INCLUDEPATH += .. . ./console ./restore ./select
-LIBS        += -L../lib -lbaccfg -lbac @OPENSSL_LIBS@
+LIBS        += -L../lib -lbaccfg -lbac -ldl @OPENSSL_LIBS@
 LIBTOOL_LINK = @QMAKE_LIBTOOL@ --silent --tag=CXX --mode=link
 LIBTOOL_INSTALL = @QMAKE_LIBTOOL@ --silent --mode=install
 QMAKE_LINK   = $${LIBTOOL_LINK} $(CXX)
