--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -91,19 +91,22 @@
 
 target_link_libraries(ngs m pthread gc ffi dl json-c pcre)
 
-add_custom_target(man ALL WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doc COMMAND make man DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/doc/*.1.md)
+# XXX-Patched: pandoc is available for x86_64 only
+#add_custom_target(man ALL WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doc COMMAND make man DEPENDS ${CMAKE_CURRENT_SOURCE_DIR}/doc/*.1.md)
 
 
 install(FILES "${PROJECT_BINARY_DIR}/ngs" DESTINATION bin)
 install(PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/ngs DESTINATION bin)
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/lib/ DESTINATION lib/ngs)
-install(
-	DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bin/
-	DESTINATION bin FILES_MATCHING PATTERN "*.ngs"
-	PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_WRITE WORLD_EXECUTE
-)
-install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE DESTINATION doc/ngs)
-install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doc/ DESTINATION man/man1 FILES_MATCHING PATTERN "*.1")
+# XXX-Patched: these scripts are quite doubtful, looks like helper scripts for the project or experimental stuff
+#install(
+#	DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/bin/
+#	DESTINATION bin FILES_MATCHING PATTERN "*.ngs"
+#	PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_WRITE WORLD_EXECUTE
+#)
+#install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/LICENSE DESTINATION share/licenses/ngs)
+# XXX-Patched: pandoc is available for x86_64 only
+#install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/doc/ DESTINATION share/man/man1 FILES_MATCHING PATTERN "*.1")
 
 enable_testing()
 add_test(all bash -c "${CMAKE_CURRENT_BINARY_DIR}/ngs ${CMAKE_CURRENT_SOURCE_DIR}/test.ngs")
