diff --git a/tests/cloudpickle_test.py b/tests/cloudpickle_test.py
index a456b63..7ea48af 100644
--- a/tests/cloudpickle_test.py
+++ b/tests/cloudpickle_test.py
@@ -2107,22 +2107,6 @@ class CloudPickleTest(unittest.TestCase):
         with pytest.raises(pickle.PicklingError, match='recursion'):
             cloudpickle.dumps(a)
 
-    def test_out_of_band_buffers(self):
-        if self.protocol < 5:
-            pytest.skip("Need Pickle Protocol 5 or later")
-        np = pytest.importorskip("numpy")
-
-        class LocallyDefinedClass:
-            data = np.zeros(10)
-
-        data_instance = LocallyDefinedClass()
-        buffers = []
-        pickle_bytes = cloudpickle.dumps(data_instance, protocol=self.protocol,
-                                         buffer_callback=buffers.append)
-        assert len(buffers) == 1
-        reconstructed = pickle.loads(pickle_bytes, buffers=buffers)
-        np.testing.assert_allclose(reconstructed.data, data_instance.data)
-
     def test_pickle_dynamic_typevar(self):
         T = typing.TypeVar('T')
         depickled_T = pickle_depickle(T, protocol=self.protocol)
