--- startup-notification-0.12.orig/test/Makefile.am
+++ startup-notification-0.12/test/Makefile.am
@@ -5,16 +5,12 @@
 XLIB_TEST=					\
 	test-launcher				\
 	test-launchee				\
-	test-monitor				\
-	test-send-xmessage			\
-	test-watch-xmessages
+	test-monitor
 
 XCB_TEST=					\
-	test-send-xmessage-xcb			\
 	test-monitor-xcb			\
 	test-launchee-xcb			\
-	test-launcher-xcb			\
-	test-watch-xmessages-xcb
+	test-launcher-xcb
 
 check_PROGRAMS=$(XLIB_TEST) $(XCB_TEST)
 
