diff -upr dmenu-4.6.orig/config.mk dmenu-4.6/config.mk
--- dmenu-4.6.orig/config.mk	2015-11-09 11:55:46.363880396 +0100
+++ dmenu-4.6/config.mk	2015-11-09 11:55:57.450509604 +0100
@@ -15,8 +15,6 @@ XINERAMAFLAGS = -DXINERAMA
 # freetype
 FREETYPELIBS = -lfontconfig -lXft
 FREETYPEINC = /usr/include/freetype2
-# OpenBSD (uncomment)
-FREETYPEINC = ${X11INC}/freetype2
 
 # includes and libs
 INCS = -I${X11INC} -I${FREETYPEINC}
