diff -Naur a/atk-adaptor/bridge.c b/atk-adaptor/bridge.c
--- a/atk-adaptor/bridge.c	2018-09-03 22:29:05.000000000 +0100
+++ b/atk-adaptor/bridge.c	2019-02-24 20:10:53.941760839 +0000
@@ -916,7 +916,7 @@
 /*
  * Checks the status of the environment variables
  *
- * At this moment it only checks NO_AT_BRIDGE
+ * At this moment it only checks AT_BRIDGE
  *
  * Returns TRUE if there isn't anything on the environment preventing
  * you to load the bridge, FALSE otherwise
@@ -926,12 +926,12 @@
 {
   const gchar *envvar;
 
-  envvar = g_getenv ("NO_AT_BRIDGE");
+  envvar = g_getenv ("AT_BRIDGE");
 
   if (envvar && atoi (envvar) == 1)
-    return FALSE;
-  else
     return TRUE;
+  else
+    return FALSE;
 }
 
 void
