--- lirc-0.10.2.orig/plugins/mplay.c
+++ lirc-0.10.2/plugins/mplay.c
@@ -197,7 +197,6 @@
 	.latest_button			= MPLAY_CODE_ERROR,
 	.fd				= -1,
 	.pipefd				= { -1,	     -1	  },
-	.tid				= -1
 };
 
 /**
@@ -782,14 +781,11 @@
 int mplayfamily_deinit(void)
 {
 	log_trace("Entering mplayfamily_deinit()");
-	if (mplayfamily_local_data.tid != -1) {
-		if (pthread_cancel(mplayfamily_local_data.tid) < 0) {
-			log_perror_err("mplay could not cancel listener");
-			return 0;
-		}
-		pthread_join(mplayfamily_local_data.tid, NULL);
-		mplayfamily_local_data.tid = -1;
+	if (pthread_cancel(mplayfamily_local_data.tid) < 0) {
+		log_perror_err("mplay could not cancel listener");
+		return 0;
 	}
+	pthread_join(mplayfamily_local_data.tid, NULL);
 	if (mplayfamily_local_data.pipefd[0] != -1) {
 		close(mplayfamily_local_data.pipefd[0]);
 		mplayfamily_local_data.pipefd[0] = -1;
