--- a/src/core/ext/transport/chttp2/transport/chttp2_transport.cc
+++ b/src/core/ext/transport/chttp2/transport/chttp2_transport.cc
@@ -1051,6 +1051,7 @@ static void write_action_begin_locked(
   } else {
     r = grpc_chttp2_begin_write(t.get());
   }
+  #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
   if (r.writing) {
     set_write_state(t.get(),
                     r.partial ? GRPC_CHTTP2_WRITE_STATE_WRITING_WITH_MORE
