diff --git a/src/Makefile b/src/Makefile
index 45ed952..5e7aed3 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -37,10 +37,9 @@
 INSTALL = /usr/bin/install -c
 
 # Installation directories, good for Debian Packaging
-DESTDIR ?= /usr/local
-prefix = ${DESTDIR}
-mandir = ${prefix}/share/man/man1
-bindir = ${prefix}/bin
+PREFIX ?= /usr/local
+mandir = ${PREFIX}/share/man/man1
+bindir = ${PREFIX}/bin
 
 RM ?= rm -f
 
@@ -203,10 +202,10 @@ check: gsl$(EXE)
 	./gsl$(EXE) teststr
 
 install:
-	$(INSTALL) -m 755 -d "$(bindir)"
-	$(INSTALL) -m 755 gsl "$(bindir)"
-	$(INSTALL) -m 755 -d "$(mandir)"
-	$(INSTALL) -m 644 gsl.1 "$(mandir)"
+	$(INSTALL) -m 755 -d $(DESTDIR)/"$(bindir)"
+	$(INSTALL) -m 755 gsl $(DESTDIR)/"$(bindir)"
+	$(INSTALL) -m 755 -d $(DESTDIR)/"$(mandir)"
+	$(INSTALL) -m 644 gsl.1 $(DESTDIR)/"$(mandir)"
 
 uninstall:
 	$(RM) "$(bindir)/gsl"
