--- pgcluster-1.9.0rc5.orig/src/pgcluster/pgrp/main.c
+++ pgcluster-1.9.0rc5/src/pgcluster/pgrp/main.c
@@ -166,7 +166,7 @@
 static void startup_replication_server(void);
 static int replicate_loop(int fd);
 static void replicate_main(void);
-static void quick_exit(SIGNAL_ARGS);
+static void signal_exit(SIGNAL_ARGS);
 static void daemonize(void);
 static void write_pid_file(void);
 static void stop_pgreplicate(void);
@@ -231,10 +231,10 @@
 
 		close(fd);
 
-		PGRsignal(SIGHUP, quick_exit);	
-		PGRsignal(SIGINT, quick_exit);	
-		PGRsignal(SIGQUIT, quick_exit);	
-		PGRsignal(SIGTERM, quick_exit);	
+		PGRsignal(SIGHUP, signal_exit);	
+		PGRsignal(SIGINT, signal_exit);	
+		PGRsignal(SIGQUIT, signal_exit);	
+		PGRsignal(SIGTERM, signal_exit);	
 		PGRsignal(SIGALRM, SIG_IGN); 
 		PGRsignal(SIGPIPE, SIG_IGN); 
 		setpgid(0,pgid);
@@ -524,7 +524,7 @@
 
 /*--------------------------------------------------------------------
  * SYMBOL
- *    quick_exit()
+ *    signal_exit()
  * NOTES
  *    Exit child process
  * ARGS
@@ -534,10 +534,10 @@
  *--------------------------------------------------------------------
  */
 static void
-quick_exit(SIGNAL_ARGS)
+signal_exit(SIGNAL_ARGS)
 {
 #ifdef PRINT_DEBUG
-	show_debug("quick_exit:signo = %d", postgres_signal_arg);
+	show_debug("signal_exit:signo = %d", postgres_signal_arg);
 #endif
 	exit(0);
 }
--- pgcluster-1.9.0rc5.orig/src/pgcluster/pgrp/recovery.c
+++ pgcluster-1.9.0rc5/src/pgcluster/pgrp/recovery.c
@@ -398,7 +398,7 @@
 #ifdef PRINT_DEBUG
 		show_debug("%s:already recovery job runing",func);
 #endif			
-		memset(packet,0,sizeof(packet));
+		memset(packet,0,sizeof(*packet));
 		PGRset_recovery_packet_no(packet, RECOVERY_ERROR_OCCUPIED) ;
 		status = send_packet(&Target,packet);
 		loop_end = true;
