--- a/lib/private/Config.php
+++ b/lib/private/Config.php
@@ -229,9 +229,6 @@
 		touch ($this->configFilePath);
 		$filePointer = fopen($this->configFilePath, 'r+');
 
-		// Prevent others not to read the config
-		chmod($this->configFilePath, 0640);
-
 		// File does not exist, this can happen when doing a fresh install
 		if(!is_resource ($filePointer)) {
 			// TODO fix this via DI once it is very clear that this doesn't cause side effects due to initialization order
