--- a/lib/private/Template/CSSResourceLocator.php
+++ b/lib/private/Template/CSSResourceLocator.php
@@ -117,7 +117,7 @@
 			parent::append($root, $file, $webRoot, $throw);
 		} else {
 			if (!$webRoot) {
-				$tmpRoot = realpath($root);
+				$tmpRoot = $root;
 				/*
 				 * traverse the potential web roots upwards in the path
 				 *
--- a/lib/private/Template/ResourceLocator.php
+++ b/lib/private/Template/ResourceLocator.php
@@ -125,7 +125,7 @@
 		}
 
 		if (!$webRoot) {
-			$tmpRoot = realpath($root);
+			$tmpRoot = $root;
 			/*
 			 * traverse the potential web roots upwards in the path
 			 *
