diff --git a/libntech/configure b/libntech/configure
index 97f7e7c..817bde2 100644
--- a/libntech/configure
+++ b/libntech/configure
@@ -161,11 +161,11 @@ AC_ARG_ENABLE([fhs],

 AS_IF([test x"$enable_fhs" = xyes], [
   projlibdir='${pkglibdir}'
-  WORKDIR='${localstatedir}/lib/${PACKAGE}'
+  WORKDIR='${localstatedir}/lib/cfengine'
   MASTERDIR='default'
   INPUTDIR='default'
-  LOGDIR='${localstatedir}/log/${PACKAGE}'
-  PIDDIR='${runstatedir:-${localstatedir}/run}/${PACKAGE}'
+  LOGDIR='${localstatedir}/log/cfengine'
+  PIDDIR='${runstatedir:-${localstatedir}/run}/cfengine'
   STATEDIR='default'
 ], [
   if test x"$prefix" = xNONE || test x"$prefix" = x/var/cfengine; then
