From 7360793bcd73f7b1238cbefe51636899741170c3 Mon Sep 17 00:00:00 2001
From: Simon Mutch <smutch@unimelb.edu.au>
Date: Wed, 20 May 2020 17:52:57 +1000
Subject: [PATCH] Use platform dependent test for unpacked_struct_with_arrays

---
 tests/buffers/buffmt.pyx | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/buffers/buffmt.pyx b/tests/buffers/buffmt.pyx
index 5d015e4bff..0aa0cb3259 100644
--- a/tests/buffers/buffmt.pyx
+++ b/tests/buffers/buffmt.pyx
@@ -1,4 +1,5 @@
 from __future__ import unicode_literals
+import struct
 
 # Tests buffer format string parsing.
 
@@ -442,7 +443,10 @@ def packed_struct_with_arrays(fmt):
 @testcase
 def unpacked_struct_with_arrays(fmt):
     """
-    >>> unpacked_struct_with_arrays("T{i:a:(8)f:b:f:c:Q:d:(5)i:e:i:f:i:g:xxxx(4)d:h:i:i:}")
+    >>> if struct.calcsize('P') == 8:  # 64 bit
+    ...     unpacked_struct_with_arrays("T{i:a:(8)f:b:f:c:Q:d:(5)i:e:i:f:i:g:xxxx(4)d:h:i:i:}")
+    ... elif struct.calcsize('P') == 4:  # 32 bit
+    ...     unpacked_struct_with_arrays("T{i:a:(8)f:b:f:c:Q:d:(5)i:e:i:f:i:g:(4)d:h:i:i:}")
     """
 
     cdef object[UnpackedStructWithArrays] buf = MockBuffer(
