--- a/Makefile
+++ b/Makefile
@@ -128,7 +128,7 @@
 # It uses BUILD_MAIN as defined above to specify the entrypoint for building Git
 # LFS.
 BUILD = GOOS=$(1) GOARCH=$(2) \
-	$(GO) build \
+	$(GO) build $(EXTRA_GO_FLAGS) \
 	-ldflags="$(LD_FLAGS)" \
 	-gcflags="$(GC_FLAGS)" \
 	-o ./bin/git-lfs$(3) $(BUILD_MAIN)
--- a/t/Makefile
+++ b/t/Makefile
@@ -37,4 +37,4 @@
 	$(RM) $(TEST_CMDS)
 
 ../bin/%$X : cmd/%.go
-	go build -o $@ $^
+	go build $(EXTRA_GO_FLAGS) -o $@ $^
