--- ./gegl/buffer/gegl-buffer.c.orig
+++ ./gegl/buffer/gegl-buffer.c
@@ -80,7 +80,7 @@
 /* #define GEGL_BUFFER_DEBUG_ALLOCATIONS to print allocation stack
  * traces for leaked GeglBuffers using GNU C libs backtrace_symbols()
  */
-#ifndef G_OS_WIN32
+#if !defined(G_OS_WIN32) && !defined(__UCLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -876,8 +876,8 @@
 gegl_buffer_get_alloc_stack (void)
 {
   char  *result         = NULL;
-#ifdef G_OS_WIN32
-  result = g_strdup ("backtrack not available on win32\n");
+#if defined(G_OS_WIN32) || defined(__UCLIBC__)
+  result = g_strdup ("backtrack not available\n");
 #else
   void  *functions[MAX_N_FUNCTIONS];
   int    n_functions    = 0;
