diff --git a/operations/external/raw-load.c b/operations/external/raw-load.c
index 13eb661..0a3cc47 100644
--- a/operations/external/raw-load.c
+++ b/operations/external/raw-load.c
@@ -114,7 +114,7 @@ prepare (GeglOperation *operation)
         g_warning ("raw-load: Error Initializing raw library");
       else
         {
-          p->LibRaw->params.shot_select = o->image_num;
+          p->LibRaw->rawparams.shot_select = o->image_num;
     
           p->LibRaw->params.aber[0] = 1.0;
           p->LibRaw->params.aber[2] = 1.0;
