diff -ru at-spi2-core-2.16.0.orig/bus/at-spi-bus-launcher.c at-spi2-core-2.16.0/bus/at-spi-bus-launcher.c
--- at-spi2-core-2.16.0.orig/bus/at-spi-bus-launcher.c	2015-03-05 06:04:20.000000000 +0200
+++ at-spi2-core-2.16.0/bus/at-spi-bus-launcher.c	2015-10-25 16:00:32.158127066 +0200
@@ -104,7 +104,7 @@
 {
   ssize_t bytes_read;
 
-  while (max_bytes > 1 && (bytes_read = read (fd, buf, MAX (4096, max_bytes - 1))))
+  while (max_bytes > 1 && (bytes_read = read (fd, buf, max_bytes - 1)))
     {
       if (bytes_read < 0)
         return FALSE;
