diff --git a/Makefile b/Makefile
index 67485b0..6e057de 100644
--- a/Makefile
+++ b/Makefile
@@ -8,15 +8,11 @@ ARCHIVE=lxd-$(VERSION).tar
 HASH := \#
 TAG_SQLITE3=$(shell printf "$(HASH)include <dqlite.h>\nvoid main(){dqlite_node_id n = 1;}" | $(CC) ${CGO_CFLAGS} -o /dev/null -xc - >/dev/null 2>&1 && echo "libsqlite3")
 GOPATH ?= $(HOME)/go
-CGO_LDFLAGS_ALLOW ?= (-Wl,-wrap,pthread_create)|(-Wl,-z,now)
+CGO_LDFLAGS_ALLOW ?= (-Wl,-wrap,pthread_create)
 export GO111MODULE=off
 
 .PHONY: default
-default: get build
-
-.PHONY: get
-get:
-	go get -t -v -d ./...
+default: build
 
 .PHONY: build
 build:
@@ -32,7 +28,6 @@ endif
 
 .PHONY: client
 client:
-	go get -t -v -d ./...
 	go install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./lxc
 	@echo "LXD client built successfully"
 
@@ -89,7 +84,6 @@ deps:
 
 .PHONY: update
 update:
-	go get -t -v -d -u ./...
 	@echo "Dependencies updated"
 
 .PHONY: update-protobuf
@@ -107,7 +101,6 @@ update-schema:
 
 .PHONY: update-api
 update-api:
-	GO111MODULE=on go get -v -x github.com/go-swagger/go-swagger/cmd/swagger
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m
 
 .PHONY: debug
@@ -117,7 +110,6 @@ ifeq ($(TAG_SQLITE3),)
 	exit 1
 endif
 
-	go get -t -v -d ./...
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go install -v -tags "$(TAG_SQLITE3) logdebug" $(DEBUG) ./...
 	CGO_ENABLED=0 go install -v -tags "netgo,logdebug" ./lxd-p2c
 	CGO_ENABLED=0 go install -v -tags "agent,netgo,logdebug" ./lxd-agent
@@ -130,7 +122,6 @@ ifeq ($(TAG_SQLITE3),)
 	exit 1
 endif
 
-	go get -t -v -d ./...
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go install -a -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	CGO_ENABLED=0 go install -a -v -tags netgo ./lxd-p2c
 	CGO_ENABLED=0 go install -a -v -tags agent,netgo ./lxd-agent
@@ -142,7 +133,6 @@ ifeq ($(TAG_SQLITE3),)
 	exit 1
 endif
 
-	go get -t -v -d ./...
 	CC="$(CC)" CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go install -race -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	CGO_ENABLED=0 go install -v -tags netgo ./lxd-p2c
 	CGO_ENABLED=0 go install -v -tags agent,netgo ./lxd-agent
@@ -150,9 +140,6 @@ endif
 
 .PHONY: check
 check: default
-	go get -v -x github.com/rogpeppe/godeps
-	go get -v -x github.com/tsenart/deadcode
-	go get -v -x golang.org/x/lint/golint
 	CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go test -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	cd test && ./main.sh
 
@@ -167,9 +154,6 @@ dist:
 	mkdir -p $(TMP)/_dist/src/github.com/lxc
 	ln -s ../../../../lxd-$(VERSION) $(TMP)/_dist/src/github.com/lxc/lxd
 
-	# Download dependencies
-	cd $(TMP)/lxd-$(VERSION) && GOPATH=$(TMP)/_dist go get -t -v -d ./...
-
 	# Download the cluster-enabled sqlite/dqlite
 	mkdir $(TMP)/_dist/deps/
 	git clone --depth=1 https://github.com/canonical/dqlite $(TMP)/_dist/deps/dqlite
@@ -203,7 +187,6 @@ update-po:
 	done
 
 update-pot:
-	go get -v -x github.com/snapcore/snapd/i18n/xgettext-go/
 	xgettext-go -o po/$(DOMAIN).pot --add-comments-tag=TRANSLATORS: --sort-output --package-name=$(DOMAIN) --msgid-bugs-address=lxc-devel@lists.linuxcontainers.org --keyword=i18n.G --keyword-plural=i18n.NG lxc/*.go lxc/*/*.go
 
 build-mo: $(MOFILES)
