diff --git a/Makefile b/Makefile
index 9e17cd4..ed4465a 100644
--- a/Makefile
+++ b/Makefile
@@ -15,13 +15,11 @@ ifeq ($(TAG_SQLITE3),)
 	exit 1
 endif
 
-	go get -t -v -d ./...
 	CC=$(CC) go install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	@echo "LXD built successfully"
 
 .PHONY: client
 client:
-	go get -t -v -d ./...
 	go install -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./lxc
 	@echo "LXD client built successfully"
 
@@ -100,7 +98,6 @@ deps:
 
 .PHONY: update
 update:
-	go get -t -v -d -u ./...
 	@echo "Dependencies updated"
 
 .PHONY: update-protobuf
@@ -121,15 +118,11 @@ ifeq ($(TAG_SQLITE3),)
 	exit 1
 endif
 
-	go get -t -v -d ./...
 	CC=$(CC) go install -v -tags "$(TAG_SQLITE3) logdebug" $(DEBUG) ./...
 	@echo "LXD built successfully"
 
 .PHONY: check
 check: default
-	go get -v -x github.com/rogpeppe/godeps
-	go get -v -x github.com/remyoudompheng/go-misc/deadcode
-	go get -v -x github.com/golang/lint/golint
 	go test -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	cd test && ./main.sh
 
@@ -145,7 +138,6 @@ dist:
 	ln -s ../../../../lxd-$(VERSION) $(TMP)/_dist/src/github.com/lxc/lxd
 
 	# Download dependencies
-	cd $(TMP)/lxd-$(VERSION) && GOPATH=$(TMP)/_dist go get -t -v -d ./...
 
 	# Download the cluster-enabled sqlite/dqlite
 	mkdir $(TMP)/_dist/deps/
@@ -184,7 +176,6 @@ update-po:
 	done
 
 update-pot:
-	go get -v -x github.com/snapcore/snapd/i18n/xgettext-go/
 	xgettext-go -o po/$(DOMAIN).pot --add-comments-tag=TRANSLATORS: --sort-output --package-name=$(DOMAIN) --msgid-bugs-address=lxc-devel@lists.linuxcontainers.org --keyword=i18n.G --keyword-plural=i18n.NG lxc/*.go lxc/*/*.go
 
 build-mo: $(MOFILES)
