From 6e36e7730caf07b6cd0bfa265cdf9b5e31e7acad Mon Sep 17 00:00:00 2001
From: NiLuJe <ninuje@gmail.com>
Date: Wed, 14 Sep 2022 21:36:02 +0200
Subject: [PATCH] Warn on unknown dicts

---
 src/sdcv.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/sdcv.cpp b/src/sdcv.cpp
index 46db492..3a10e3c 100644
--- a/src/sdcv.cpp
+++ b/src/sdcv.cpp
@@ -190,6 +190,8 @@ try {
             auto it = bookname_to_ifo.find(*p);
             if (it != bookname_to_ifo.end()) {
                 order_list.push_back(it->second);
+            } else {
+                fprintf(stderr, _("Unknown dictionary: %s\n"), *p);
             }
         }
     } else {
@@ -205,6 +207,8 @@ try {
                 auto it = bookname_to_ifo.find(line);
                 if (it != bookname_to_ifo.end()) {
                     order_list.push_back(it->second);
+                } else {
+                    fprintf(stderr, _("Unknown dictionary: %s\n"), line.c_str());
                 }
             }
             fclose(ordering_file);
