diff --git a/etc/profile-templates/all/all/nssdatabases b/etc/profile-templates/all/all/nssdatabases
index 42fa310..815ddce 100644
--- a/etc/profile-templates/all/all/nssdatabases
+++ b/etc/profile-templates/all/all/nssdatabases
@@ -4,4 +4,3 @@
 passwd
 shadow
 group
-gshadow
diff --git a/etc/profile-templates/buildd/all/nssdatabases b/etc/profile-templates/buildd/all/nssdatabases
index 42fa310..815ddce 100644
--- a/etc/profile-templates/buildd/all/nssdatabases
+++ b/etc/profile-templates/buildd/all/nssdatabases
@@ -4,4 +4,3 @@
 passwd
 shadow
 group
-gshadow
diff --git a/etc/setup.d/20copyfiles b/etc/setup.d/20copyfiles
index 2c01af9..e427d35 100755
--- a/etc/setup.d/20copyfiles
+++ b/etc/setup.d/20copyfiles
@@ -39,9 +39,9 @@ copy_file()
         if [ -e "$2" ]; then
 
             # Device and inode
-            da=$(/usr/bin/stat --format="%d %i" "$1")
+            da=$(/bin/stat -c "%d %i" "$1")
             # This one can fail since it might not exist yet
-            db=$(/usr/bin/stat --format="%d %i" "$2" 2>/dev/null || :)
+            db=$(/bin/stat -c "%d %i" "$2" 2>/dev/null || :)
 
             if [ "$da" = "$db" ]; then
                 COPY="false"
diff --git a/etc/setup.d/20nssdatabases b/etc/setup.d/20nssdatabases
index 6ea83c0..3f4019c 100755
--- a/etc/setup.d/20nssdatabases
+++ b/etc/setup.d/20nssdatabases
@@ -42,9 +42,9 @@ if [ $STAGE = "setup-start" ] || [ $STAGE = "setup-recover" ]; then
                 fi
 
                 # Device and inode
-                dr=$(/usr/bin/stat --format="%d %i" "/etc/$db")
+                dr=$(/bin/stat -c "%d %i" "/etc/$db")
                 # This one can fail since it might not exist yet
-                dc=$(/usr/bin/stat --format="%d %i" "${CHROOT_PATH}/etc/$db" 2>/dev/null || :)
+                dc=$(/bin/stat -c "%d %i" "${CHROOT_PATH}/etc/$db" 2>/dev/null || :)
 
                 # If the database inside and outside the chroot is the
                 # same, it's very likely that dup_nss would blank the
