From 73936a423227aa78b7682bdd3edc20643763807b Mon Sep 17 00:00:00 2001
From: Andrea Brancaleoni <abc@pompel.me>
Date: Wed, 19 Sep 2018 11:22:47 +0200
Subject: [PATCH] musl

---
 sbuild/sbuild-feature.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git sbuild/sbuild-feature.cc sbuild/sbuild-feature.cc
index 5ac4725..9902db3 100644
--- a/sbuild/sbuild-feature.cc
+++ b/sbuild/sbuild-feature.cc
@@ -20,6 +20,8 @@
 
 #include <iostream>
 
+#include <libintl.h>
+
 #include "sbuild-feature.h"
 
 using namespace sbuild;
-- 
2.19.0

diff --git a/sbuild/Makefile.am b/sbuild/Makefile.am
index 03b5c7b..afa7638 100644
--- a/sbuild/Makefile.am
+++ b/sbuild/Makefile.am
@@ -222,7 +222,7 @@ pkgconfigdata_DATA = sbuild.pc
 # Uncomment to build a versioned library
 libsbuild_version = -version-info $(SBUILD_CURRENT_INTERFACE):$(SBUILD_INTERFACE_AGE):$(SBUILD_BINARY_AGE)
 libsbuild_la_LDFLAGS = $(libsbuild_version) \
-        -rpath $(libdir) -no-undefined
+        -rpath $(libdir) -no-undefined -lintl
 
 EXTRA_DIST =			\
 	CMakeLists.txt		\
diff --git a/test/Makefile.in b/test/Makefile.in
index ce78448..efbc287 100644
--- a/test/Makefile.in
+++ b/test/Makefile.in
@@ -334,7 +334,7 @@ am__v_CC_1 =
 CCLD = $(CC)
 LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(AM_LDFLAGS) $(LDFLAGS) -lcppunit -o $@
 AM_V_CCLD = $(am__v_CCLD_@AM_V@)
 am__v_CCLD_ = $(am__v_CCLD_@AM_DEFAULT_V@)
 am__v_CCLD_0 = @echo "  CCLD    " $@;
