Author: Simon Frankenberger <simon-alpine@fraho.eu>
Subject: Patch to compile for java 8 as java 6 is no longer supported
    in modern compilers

--- old/libraries/javacheck/CMakeLists.txt
+++ new/libraries/javacheck/CMakeLists.txt
@@ -4,7 +4,7 @@
 
 include(UseJava)
 set(CMAKE_JAVA_JAR_ENTRY_POINT JavaCheck)
-set(CMAKE_JAVA_COMPILE_FLAGS -target 1.6 -source 1.6 -Xlint:deprecation -Xlint:unchecked)
+set(CMAKE_JAVA_COMPILE_FLAGS -target 8 -source 8 -Xlint:deprecation -Xlint:unchecked)
 
 set(SRC
     JavaCheck.java
--- old/libraries/launcher/CMakeLists.txt
+++ new/libraries/launcher/CMakeLists.txt
@@ -4,7 +4,7 @@
 
 include(UseJava)
 set(CMAKE_JAVA_JAR_ENTRY_POINT org.multimc.EntryPoint)
-set(CMAKE_JAVA_COMPILE_FLAGS -target 1.6 -source 1.6 -Xlint:deprecation -Xlint:unchecked)
+set(CMAKE_JAVA_COMPILE_FLAGS -target 8 -source 8 -Xlint:deprecation -Xlint:unchecked)
 
 set(SRC
     org/multimc/EntryPoint.java
