--- ./src/octoprint/vendor/with_attrs_docs.py.orig
+++ ./src/octoprint/vendor/with_attrs_docs.py
@@ -4,7 +4,6 @@
 
 from typing import Type
 
-from class_doc import extract_docs_from_cls_obj
 from pydantic import BaseModel
 
 
@@ -17,16 +16,7 @@
     :param model: any pydantic model
     :param override_existing: override existing descriptions
     """
-    docs = extract_docs_from_cls_obj(model)
-
-    for field in model.__fields__.values():
-        if field.field_info.description and not override_existing:
-            continue
-
-        try:
-            field.field_info.description = '\n'.join(docs[field.name])
-        except KeyError:
-            pass
+    pass
 
 
 def with_attrs_docs(
