Fix crypto/digest tests to be compatible with LibreSSL.

--- a/test/app/crypto.result
+++ b/test/app/crypto.result
@@ -111,13 +111,13 @@
 ...
 ciph.decrypt(enc, bad_pass, iv)
 ---
-- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06065064:digital envelope
-    routines:EVP_DecryptFinal_ex:bad decrypt'
+- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06FFF064:digital envelope
+    routines:CRYPTO_internal:bad decrypt'
 ...
 ciph.decrypt(enc, pass, bad_iv)
 ---
-- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06065064:digital envelope
-    routines:EVP_DecryptFinal_ex:bad decrypt'
+- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06FFF064:digital envelope
+    routines:CRYPTO_internal:bad decrypt'
 ...
 test_run:cmd("clear filter")
 ---
--- a/test/app/digest.result
+++ b/test/app/digest.result
@@ -445,8 +445,8 @@
 ...
 digest.aes256cbc.decrypt(digest.aes256cbc.encrypt('test123', 'passpasspasspasspasspasspasspass', 'iv12tras8712cvbh'), 'nosspasspasspasspasspasspasspass', 'iv12tras8712cvbh')
 ---
-- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06065064:digital envelope
-    routines:EVP_DecryptFinal_ex:bad decrypt'
+- error: 'builtin/crypto.lua:<line>"]: Can''t finalize cipher:error:06FFF064:digital envelope
+    routines:CRYPTO_internal:bad decrypt'
 ...
 digest = nil
 ---
