diff --git a/CMakeLists.txt b/CMakeLists.txt
index a41d060c..0c49ddea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -61,7 +61,7 @@ pkg_check_modules(DBUS dbus-1 REQUIRED IMPORTED_TARGET)
 pkg_check_modules(DSME_DBUS dsme_dbus_if REQUIRED IMPORTED_TARGET)
 pkg_check_modules(GLIB2 glib-2.0 REQUIRED IMPORTED_TARGET)
 pkg_check_modules(GIO_UNIX gio-unix-2.0 REQUIRED IMPORTED_TARGET)
-pkg_check_modules(KEEPALIVE keepalive-qt6 REQUIRED IMPORTED_TARGET)
+pkg_check_modules(KEEPALIVE keepalive REQUIRED IMPORTED_TARGET)
 pkg_check_modules(LIBRESOURCE libresourceqt6 REQUIRED IMPORTED_TARGET)
 pkg_check_modules(LIBSYSTEMD libsystemd REQUIRED IMPORTED_TARGET)
 pkg_check_modules(MLITE mlite6 REQUIRED IMPORTED_TARGET)
@@ -71,7 +71,7 @@ pkg_check_modules(NEMODEVICELOCK nemodevicelock REQUIRED IMPORTED_TARGET)
 pkg_check_modules(NEMOCONNECTIVITY nemoconnectivity REQUIRED IMPORTED_TARGET)
 pkg_check_modules(NGF_QT6 ngf-qt6 REQUIRED IMPORTED_TARGET)
 pkg_check_modules(THERMALMANAGER_DBUS thermalmanager_dbus_if REQUIRED IMPORTED_TARGET)
-pkg_check_modules(SYSTEMSETTINGS systemsettings-qt6 REQUIRED IMPORTED_TARGET)
+pkg_check_modules(SYSTEMSETTINGS systemsettings REQUIRED IMPORTED_TARGET)
 pkg_check_modules(TIMED_QT timed-qt6 REQUIRED IMPORTED_TARGET)
 pkg_check_modules(WAYLAND_SERVER wayland-server REQUIRED IMPORTED_TARGET)
 pkg_check_modules(USB_MODED usb_moded REQUIRED IMPORTED_TARGET)
