From bccb03b5eace9df3539599de7bc0abd6753da412 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Wed, 15 Apr 2026 08:30:17 +0200
Subject: [PATCH 1/3] fix(cmake): use the correct project version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 38d8084..23abea7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.6.0)
 
 project(asteroid-launcher
-	VERSION 0.0.1
+	VERSION 2.1
 	DESCRIPTION "AsteroidOS launcher and Wayland compositor based on Qt5, QML and QtWayland via Lipstick")
 
 find_package(ECM REQUIRED NO_MODULE)
-- 
2.53.0

