From 20b371a7e90edcc813b393272a642e10ea99e770 Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Wed, 15 Apr 2026 08:31:15 +0200
Subject: [PATCH 2/3] chore: prettify CMake usage

---
 CMakeLists.txt     |  3 ++-
 src/CMakeLists.txt | 47 +++++++++++++++++++++++++++++-----------------
 2 files changed, 32 insertions(+), 18 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 23abea7..fedbe38 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,8 @@ cmake_minimum_required(VERSION 3.6.0)
 
 project(asteroid-launcher
 	VERSION 2.1
-	DESCRIPTION "AsteroidOS launcher and Wayland compositor based on Qt5, QML and QtWayland via Lipstick")
+	DESCRIPTION "AsteroidOS launcher and Wayland compositor based on Qt5, QML and QtWayland via Lipstick"
+)
 
 find_package(ECM REQUIRED NO_MODULE)
 find_package(AsteroidApp REQUIRED)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2d70079..fcb9461 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,22 +16,35 @@ set(HEADERS
 
 add_executable(asteroid-launcher ${SRC} ${HEADERS} resources-qml.qrc)
 
-target_link_libraries(asteroid-launcher PRIVATE
-	Qt5::Quick
-	Qt5::WaylandCompositor
-	Qt5::DBus
-	Mlite5::Mlite5
-	Lipstick::Lipstick
-	Timed::Timed)
+target_link_libraries(
+	asteroid-launcher
+	PRIVATE
+		Qt5::Quick
+		Qt5::WaylandCompositor
+		Qt5::DBus
+		Mlite5::Mlite5
+		Lipstick::Lipstick
+		Timed::Timed
+)
 
-install(TARGETS asteroid-launcher
-	DESTINATION ${CMAKE_INSTALL_BINDIR})
+install(
+	TARGETS asteroid-launcher
+	DESTINATION ${CMAKE_INSTALL_BINDIR}
+)
 
-install(DIRECTORY watchfaces
-	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher)
-install(DIRECTORY watchfaces-img
-	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher)
-install(DIRECTORY watchfaces-preview
-	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher)
-install(DIRECTORY applauncher
-	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher)
+install(
+	DIRECTORY watchfaces
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher
+)
+install(
+	DIRECTORY watchfaces-img
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher
+)
+install(
+	DIRECTORY watchfaces-preview
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher
+)
+install(
+	DIRECTORY applauncher
+	DESTINATION ${CMAKE_INSTALL_DATADIR}/asteroid-launcher
+)
-- 
2.53.0

