diff --git a/src/server/Makefile b/src/server/Makefile
index ed10378..7502d8e 100644
--- a/src/server/Makefile
+++ b/src/server/Makefile
@@ -71,10 +71,10 @@ else
 	CONFDIR := /etc
 endif
 
-SED_REPLACE := -e 's:@BIN@:$(DESTDIR)$(PREFIX)/bin/$(TARGET_BIN)$(OUTPUT):g' \
-	-e 's:@CONFIG@:$(DESTDIR)$(CONFDIR)/nymphcast/nymphcast_config.ini:g' \
-	-e 's:@APPS@:$(DESTDIR)$(PREFIX)/share/nymphcast/apps/:g' \
-	-e 's:@WALLPAPERS@:$(DESTDIR)$(PREFIX)/share/nymphcast/wallpapers/:g'
+SED_REPLACE := -e 's:@BIN@:$(PREFIX)/bin/$(OUTPUT):g' \
+	-e 's:@CONFIG@:$(CONFDIR)/nymphcast/nymphcast_config.ini:g' \
+	-e 's:@APPS@:$(PREFIX)/share/nymphcast/apps/:g' \
+	-e 's:@WALLPAPERS@:$(PREFIX)/share/nymphcast/wallpapers/:g'
 
 all: makedir bin/$(TARGET_BIN)$(OUTPUT)
 
@@ -131,6 +131,7 @@ install-systemd:
 install-openrc:
 	install -d $(DESTDIR)$(CONFDIR)/init.d/	
 	sed ${SED_REPLACE} openrc/nymphcast > $(DESTDIR)$(CONFDIR)/init.d/nymphcast
+	chmod +x $(DESTDIR)$(CONFDIR)/init.d/nymphcast
 
 
 # Construct an archive for Raspbian: audio-only, or full.
