From e45c1e13a344be106b52d5871a50ac1a481671f7 Mon Sep 17 00:00:00 2001
From: Maya Posch <maya@nyanko.ws>
Date: Sun, 1 Aug 2021 12:59:03 +0200
Subject: [PATCH] Add check for state of repeat queue checkbox.

---
 player/NymphCastPlayer/mainwindow.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/player/NymphCastPlayer/mainwindow.cpp b/player/NymphCastPlayer/mainwindow.cpp
index 4c8665b..00d352e 100644
--- a/player/NymphCastPlayer/mainwindow.cpp
+++ b/player/NymphCastPlayer/mainwindow.cpp
@@ -312,6 +312,12 @@ void MainWindow::setPlaying(uint32_t /*handle*/, NymphPlaybackStatus status) {
                 // Next track.
                 int crow = ui->mediaListWidget->currentRow();
                 if (++crow == ui->mediaListWidget->count()) {
+					if (ui->repeatQueueCheckBox->isChecked() == false) {
+						// Don't repeat playlist. End playback.
+						std::cout << "Repeat Queue checkbox is false. End playback." << std::endl;
+						return;
+					}
+					
                     // Restart at top.
                     crow = 0;
                 }
