--- a/src/server/gui/core/utils/FileSystemUtil.cpp
+++ b/src/server/gui/core/utils/FileSystemUtil.cpp
@@ -594,10 +594,10 @@
 				CloseHandle(hFile);
 			}
 #else // _WIN32
-			struct stat64 info;
+			struct stat info;
 
 			// check if lstat64 succeeded
-			if(lstat64(path.c_str(), &info) == 0)
+			if(lstat(path.c_str(), &info) == 0)
 			{
 				resolved.resize(info.st_size);
 				if(readlink(path.c_str(), (char*)resolved.data(), resolved.size()) > 0)
@@ -664,9 +664,9 @@
 			if (mPathExistsIndex.find(_path) == mPathExistsIndex.cend())
 			{
 				const std::string path = getGenericPath(_path);
-				struct stat64 info;
+				struct stat info;
 				// check if stat64 succeeded
-				mPathExistsIndex[_path] = (stat64(path.c_str(), &info) == 0);
+				mPathExistsIndex[_path] = (stat(path.c_str(), &info) == 0);
 			}
 
 			return mPathExistsIndex.at(_path);
@@ -692,10 +692,10 @@
 		bool isRegularFile(const std::string& _path)
 		{
 			const std::string path = getGenericPath(_path);
-			struct stat64     info;
+			struct stat     info;
 
 			// check if stat64 succeeded
-			if(stat64(path.c_str(), &info) != 0)
+			if(stat(path.c_str(), &info) != 0)
 				return false;
 
 			// check for S_IFREG attribute
@@ -708,10 +708,10 @@
 		bool isDirectory(const std::string& _path)
 		{
 			const std::string path = getGenericPath(_path);
-			struct stat64     info;
+			struct stat     info;
 
 			// check if stat64 succeeded
-			if(stat64(path.c_str(), &info) != 0)
+			if(stat(path.c_str(), &info) != 0)
 				return false;
 
 			// check for S_IFDIR attribute
@@ -731,10 +731,10 @@
 			if((Attributes != INVALID_FILE_ATTRIBUTES) && (Attributes & FILE_ATTRIBUTE_REPARSE_POINT))
 				return true;
 #else // _WIN32
-			struct stat64 info;
+			struct stat info;
 
 			// check if lstat64 succeeded
-			if(lstat64(path.c_str(), &info) != 0)
+			if(lstat(path.c_str(), &info) != 0)
 				return false;
 
 			// check for S_IFLNK attribute
@@ -777,10 +777,10 @@
 
 			// regular files and executables but not setuid, setgid, shared text (mode 0755)
 			const mode_t  mask = S_IFREG | S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH;
-			struct stat64 info;
+			struct stat info;
 
 			// check if stat64 succeeded
-			if(stat64(path.c_str(), &info) != 0)
+			if(stat(path.c_str(), &info) != 0)
 				return false;
 
 			// check for mask attributes only
