--- orig/tests/CodeAllocator.test.cpp
+++ luau-0.547/tests/CodeAllocator.test.cpp
@@ -127,6 +127,7 @@
     CHECK(info.destroyCalled);
 }
 
+#if !defined(__s390x__)
 TEST_CASE("WindowsUnwindCodesX64")
 {
     UnwindBuilderWin unwind;
@@ -187,6 +188,7 @@
     REQUIRE(data.size() == expected.size());
     CHECK(memcmp(data.data(), expected.data(), expected.size()) == 0);
 }
+#endif
 
 #if defined(__x86_64__) || defined(_M_X64)
 
