diff --git a/tools/patchwork-update-commits b/tools/patchwork-update-commits
index c84a6305e3..2d34f4f9a9 100755
--- a/tools/patchwork-update-commits
+++ b/tools/patchwork-update-commits
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 #
 # Patchwork - automated patch tracking system
 # Copyright (C) 2010 Jeremy Kerr <jk@ozlabs.org>
diff --git a/tools/post-receive.hook b/tools/post-receive.hook
index 8098ce825a..b8a1fd2f2a 100755
--- a/tools/post-receive.hook
+++ b/tools/post-receive.hook
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # Patchwork - automated patch tracking system
 # Copyright (C) 2010 martin f. krafft <madduck@madduck.net>
 #
@@ -21,7 +21,7 @@
 # Git post-receive hook to update Patchwork patches after Git pushes
 set -eu
 
-PW_DIR=/opt/patchwork/patchwork
+PW_DIR=/usr/share/webapps/patchwork/patchwork
 
 #TODO: the state map should really live in the repo's git-config
 STATE_MAP="refs/heads/master:Accepted"
