x86_64-v2 is the minimum
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -191,8 +191,11 @@
         message(STATUS "gcc will tune for ${GNUCC_ARCH}, ${TUNE_FLAG}")
     endif()
 elseif (CMAKE_COMPILER_IS_CLANG AND NOT CROSS_COMPILE)
-    if (ARCH_IA32 OR ARCH_X86_64)
-        set(GNUCC_ARCH native)
+    if (ARCH_IA32)
+        set(GNUCC_ARCH i586)
+        set(TUNE_FLAG generic)
+    elseif(ARCH_X86_64)
+        set(GNUCC_ARCH x86-64-v2)
         set(TUNE_FLAG generic)
     elseif(ARCH_AARCH64)
        set(GNUCC_ARCH armv8)
