--- a/lib/App/cpanminus/fatscript.pm
+++ b/lib/App/cpanminus/fatscript.pm
@@ -469,7 +469,7 @@
           mirrors => [],
           mirror_only => undef,
           mirror_index => undef,
-          cpanmetadb => "http://cpanmetadb.plackperl.org/v1.0/",
+          cpanmetadb => "https://cpanmetadb.plackperl.org/v1.0/",
           perl => $^X,
           argv => [],
           local_lib => undef,
@@ -926,7 +926,7 @@
   sub search_metacpan {
       my($self, $module, $version, $dev_release) = @_;
   
-      my $metacpan_uri = 'http://fastapi.metacpan.org/v1/download_url/';
+      my $metacpan_uri = 'https://fastapi.metacpan.org/v1/download_url/';
   
       my $url = $metacpan_uri . $module;
   
@@ -943,7 +943,7 @@
       if ($dist_meta && $dist_meta->{download_url}) {
           (my $distfile = $dist_meta->{download_url}) =~ s!.+/authors/id/!!;
           local $self->{mirrors} = $self->{mirrors};
-          $self->{mirrors} = [ 'http://cpan.metacpan.org' ];
+          $self->{mirrors} = [ 'https://cpan.metacpan.org' ];
           return $self->cpan_module($module, $distfile, $dist_meta->{version});
       }
   
@@ -1019,7 +1019,7 @@
       for my $try (sort { $b->{version_obj} cmp $a->{version_obj} } @found) {
           if ($self->satisfy_version($module, $try->{version_obj}, $version)) {
               local $self->{mirrors} = $self->{mirrors};
-              unshift @{$self->{mirrors}}, 'http://backpan.perl.org'
+              unshift @{$self->{mirrors}}, 'https://backpan.perl.org'
                 unless $try->{latest};
               return $self->cpan_module($module, $try->{distfile}, $try->{version});
           }
@@ -1671,7 +1671,7 @@
   sub configure_mirrors {
       my $self = shift;
       unless (@{$self->{mirrors}}) {
-          $self->{mirrors} = [ 'http://www.cpan.org' ];
+          $self->{mirrors} = [ 'https://www.cpan.org' ];
       }
       for (@{$self->{mirrors}}) {
           s!^/!file:///!;
