diff -rupN a/mopidy_spotify/__init__.py b/mopidy_spotify/__init__.py
--- a/mopidy_spotify/__init__.py	2024-10-06 23:45:32.000000000 +0000
+++ b/mopidy_spotify/__init__.py	2026-03-31 02:21:49.620000000 +0000
@@ -1,9 +1,9 @@
 import pathlib
 
-import pkg_resources
+from importlib.metadata import version
 from mopidy import config, ext
 
-__version__ = pkg_resources.get_distribution("Mopidy-Spotify").version
+__version__ = version("Mopidy-Spotify")
 
 
 class Extension(ext.Extension):
