--- ./lib/src/widgets/desktop_single_page.dart.orig	2024-06-22 13:25:32.373408447 +0200
+++ ./lib/src/widgets/desktop_single_page.dart	2024-06-22 13:25:44.330368140 +0200
@@ -14,7 +14,7 @@
     return LayoutBuilder(
       key: _layoutKey,
       builder: (context, constraints) {
-        final backgroundColor = Theme.of(context).colorScheme.surfaceBright;
+        final backgroundColor = Theme.of(context).colorScheme.surface;
         Widget child;
         if (constraints.maxWidth > kHugeDeviceBreakpoint) {
           child = Material(
