From c6563b85e29aa17967ceda5b7a300b31960dbe75 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Fri, 27 Dec 2024 18:47:46 +0100
Subject: [PATCH] ReleaseSafe is not debug

---
 src/apprt/gtk/Window.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/apprt/gtk/Window.zig b/src/apprt/gtk/Window.zig
index 48e88e49..918b7284 100644
--- a/src/apprt/gtk/Window.zig
+++ b/src/apprt/gtk/Window.zig
@@ -290,7 +290,7 @@
 
     // In debug we show a warning and apply the 'devel' class to the window.
     // This is a really common issue where people build from source in debug and performance is really bad.
-    if (comptime std.debug.runtime_safety) {
+    if (builtin.mode == .Debug) {
         const warning_box = gtk.Box.new(.vertical, 0);
         const warning_text = i18n._("⚠️ You're running a debug build of Ghostty! Performance will be degraded.");
         if (adw_version.supportsBanner()) {
