diff --git a/setup.cfg b/setup.cfg
index ee0a6c4..67f4a7d 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -46,8 +46,6 @@ install_requires =
 	mccabe>=0.6,<0.7
 	toml>=0.7.1
 	colorama;sys_platform=="win32"
-setup_requires = 
-	setuptools_scm
 python_requires = ~=3.6
 
 [options.packages.find]
diff --git a/setup.py b/setup.py
index d5d43d7..6068493 100644
--- a/setup.py
+++ b/setup.py
@@ -1,3 +1,3 @@
 from setuptools import setup
 
-setup(use_scm_version=True)
+setup()
