--- a/src/kmk/glob/fnmatch.c
+++ b/src/kmk/glob/fnmatch.c
@@ -121,7 +121,7 @@
    whose names are inconsistent.  */
 
 # if !defined _LIBC && !defined getenv && !defined _MSC_VER
-extern char *getenv ();
+extern char *getenv (const char*);
 # endif
 
 # ifndef errno
--- a/src/kmk/getopt.h
+++ b/src/kmk/getopt.h
@@ -103,7 +103,7 @@
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int argc, char *const *argv, const char *shortopts);
 #else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int argc, char *const *argv, const char *shortopts);
 #endif /* __GNU_LIBRARY__ */
 extern int getopt_long (int argc, char *const *argv, const char *shortopts,
 		        const struct option *longopts, int *longind);
--- a/src/oldsed/lib/getopt.c
+++ b/src/oldsed/lib/getopt.c
@@ -215,10 +215,10 @@
    whose names are inconsistent.  */
 
 #ifndef getenv
-extern char *getenv ();
+extern char *getenv (const char*);
 #endif
 #ifndef strncmp
-extern int strncmp ();
+extern int strncmp (const char*, const char*, size_t);
 #endif
 
 static char *
--- a/src/oldsed/lib/getopt.h
+++ b/src/oldsed/lib/getopt.h
@@ -105,7 +105,7 @@
    errors, only prototype getopt for the GNU C library.  */
 extern int getopt (int argc, char *const *argv, const char *shortopts);
 #else /* not __GNU_LIBRARY__ */
-extern int getopt ();
+extern int getopt (int argc, char *const *argv, const char *shortopts);
 #endif /* __GNU_LIBRARY__ */
 extern int getopt_long (int argc, char *const *argv, const char *shortopts,
 		        const struct option *longopts, int *longind);
@@ -119,11 +119,17 @@
 		             const struct option *longopts, int *longind,
 			     int long_only);
 #else /* not __STDC__ */
-extern int getopt ();
-extern int getopt_long ();
-extern int getopt_long_only ();
+extern int getopt (int argc, char *const *argv, const char *shortopts);
+extern int getopt_long (int argc, char *const *argv, const char *shortopts,
+		        const struct option *longopts, int *longind);
+extern int getopt_long_only (int argc, char *const *argv,
+			     const char *shortopts,
+		             const struct option *longopts, int *longind);
 
-extern int _getopt_internal ();
+extern int _getopt_internal (int argc, char *const *argv,
+			     const char *shortopts,
+		             const struct option *longopts, int *longind,
+			     int long_only);
 #endif /* __STDC__ */
 
 #ifdef	__cplusplus
