--- ./src/sed/Makefile.kmk.orig
+++ ./src/sed/Makefile.kmk
@@ -57,37 +57,30 @@
 	sed/mbcs.c \
 	lib/getopt1.c \
 	lib/getopt.c \
+	lib/obstack.c \
 	lib/utils.c
 kmk_sed_SOURCES.darwin = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.dragonfly = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.freebsd = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.haiku = \
-	lib/strverscmp.c \
-	lib/obstack.c
+	lib/strverscmp.c 
 kmk_sed_SOURCES.netbsd = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.openbsd = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.solaris = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/getline.c
 kmk_sed_SOURCES.win = \
 	lib/strverscmp.c \
-	lib/obstack.c \
 	lib/mkstemp.c \
 	lib/getline.c \
 	../lib/startuphacks-win.c
