diff --git a/src/oldsed/Makefile.kmk b/src/oldsed/Makefile.kmk
index f34b3e8..1734c59 100644
--- a/src/oldsed/Makefile.kmk
+++ b/src/oldsed/Makefile.kmk
@@ -57,6 +57,7 @@ kmk_sed_SOURCES = \
 	sed/mbcs.c \
 	lib/getopt1.c \
 	lib/getopt.c \
+	lib/obstack.c \
 	lib/utils.c
 kmk_sed_SOURCES.darwin = \
 	lib/strverscmp.c \
diff --git a/src/oldsed/lib/getopt.c b/src/oldsed/lib/getopt.c
index d341fe0..e6ad2c6 100644
--- a/src/oldsed/lib/getopt.c
+++ b/src/oldsed/lib/getopt.c
@@ -43,6 +43,7 @@
 #endif
 
 #include <stdio.h>
+#include <string.h>
 
 /* Comment out all this code if we are using the GNU C Library, and are not
    actually compiling the library itself.  This code is part of the GNU C
