diff --git a/src/ash/jobs.c b/src/ash/jobs.c
index 146326a..cbf0c3e 100644
--- ./src/kash/jobs.c.orig
+++ ./src/kash/jobs.c
@@ -396,8 +396,8 @@
 #endif
 				st = WTERMSIG(ps->status);
 			st &= 0x7f;
-			if (st < NSIG && sys_siglist[st])
-				scopyn(sys_siglist[st], s + col, 32);
+			if (st < NSIG && strsignal(st))
+				scopyn(strsignal(st), s + col, 32);
 			else
 				fmtstr(s + col, 16, "Signal %d", st);
 			if (WCOREDUMP(ps->status)) {
