diff --git a/kernel/yosys.cc b/kernel/yosys.cc
index de5baaee8..075e5f0dd 100644
--- a/kernel/yosys.cc
+++ b/kernel/yosys.cc
@@ -227,7 +227,8 @@ PYBIND11_MODULE(pyosys, m) {
 //
 // This should not affect using wheels as the dylib has to actually be called
 // libyosys_dummy.so for this function to be interacted with at all.
-PYBIND11_MODULE(libyosys_dummy, _) {
+PYBIND11_MODULE(libyosys_dummy, m) {
+	(void)m;
 	throw py::import_error("Change your import from 'import libyosys' to 'from pyosys import libyosys'.");
 }
 #endif
diff --git a/pyosys/hashlib.h b/pyosys/hashlib.h
index 386f1c0d8..5af520512 100644
--- a/pyosys/hashlib.h
+++ b/pyosys/hashlib.h
@@ -481,10 +481,12 @@ void bind_idict(module &m, const char *name_cstr) {
 		.def("keys", [](const C &s){
 			return make_iterator(s.begin(), s.end());
 		})
-		.def("values", [](args _){
+		.def("values", [](args a){
+			(void)a;
 			throw type_error("idicts do not support iteration on the integers");
 		})
-		.def("items", [](args _){
+		.def("items", [](args a){
+			(void)a;
 			throw type_error("idicts do not support pairwise iteration");
 		})
 		.def("update", [](C &s, iterable other) {
@@ -520,7 +522,8 @@ void bind_idict(module &m, const char *name_cstr) {
 		});
 
 	for (const char *mutator: {"__setitem__", "__delitem__", "pop", "popitem", "setdefault"}) {
-		cls.def(mutator, [](args _) {
+		cls.def(mutator, [](args a) {
+			(void)a;
 			throw type_error("idicts do not support arbitrary element mutation");
 		});
 	}
