From dc4b7b1e2036e79d0a263e1b902983f31a529833 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Thu, 16 Apr 2026 18:11:02 +0200
Subject: [PATCH] Add forward compatibility for KMime from KDE Frameworks

---
 CMakeLists.txt           | 6 +++++-
 autotests/CMakeLists.txt | 4 ++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01ad4011..8ee6492c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -112,7 +112,11 @@ ecm_check_outbound_license(LICENSES GPL-2.0-only FILES ${ALL_SOURCE_FILES})
 include(ECMGenerateQDoc)
 
 set(KPIM_MIME_VERSION "6.7.40")
-find_package(KPim6Mime ${KPIM_MIME_VERSION} CONFIG REQUIRED)
+find_package(KF6Mime ${KF_MIN_VERSION} CONFIG)
+if(NOT TARGET KF6::Mime)
+    find_package(KPim6Mime ${KPIM_MIME_VERSION} CONFIG REQUIRED)
+    add_library(KF6::Mime ALIAS KPim6::Mime)
+endif()
 set(AKONADICALENDAR_LIB_VERSION ${PIM_VERSION})
 set(AKONADICONTACT_LIB_VERSION "6.7.40")
 set(AKONADIMIME_LIB_VERSION "6.7.40")
diff --git a/autotests/CMakeLists.txt b/autotests/CMakeLists.txt
index 24cfad81..2e2125e1 100644
--- a/autotests/CMakeLists.txt
+++ b/autotests/CMakeLists.txt
@@ -22,7 +22,7 @@ set(common_sources
     ${Akonadi-Calendar_SOURCE_DIR}/src/utils_p.cpp
 )
 set(common_libs
-    "KPim6::AkonadiCalendar;KF6::CalendarCore;KPim6::Mime;KPim6::IdentityManagementCore;KPim6::AkonadiWidgets;KF6::Codecs;KF6::ConfigCore"
+    "KPim6::AkonadiCalendar;KF6::CalendarCore;KF6::Mime;KPim6::IdentityManagementCore;KPim6::AkonadiWidgets;KF6::Codecs;KF6::ConfigCore"
 )
 
 # the tests need the ical resource, which we might not have at this point (e.g. on the CI)
@@ -45,7 +45,7 @@ if(AKONADI_ICAL_RESOURCE)
             Qt::DBus
             Qt::Test
             KPim6::AkonadiCalendar
-            KPim6::Mime
+            KF6::Mime
             KF6::Contacts
             KPim6::AkonadiMime
             KPim6::IdentityManagementCore
-- 
GitLab

