From 12f20a383f0d0a102977259925beb410b238ab8e Mon Sep 17 00:00:00 2001
From: Leonardo Arena <rnalrd@alpinelinux.org>
Date: Fri, 31 Dec 2021 08:31:51 +0000
Subject: [PATCH] lxc-dont-go-get

---
 Makefile | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/Makefile b/Makefile
index 9a9aa3b..8b51ed1 100644
--- a/Makefile
+++ b/Makefile
@@ -88,7 +88,6 @@ ifneq "$(LXD_OFFLINE)" ""
 	@echo "The update target cannot be run in offline mode."
 	exit 1
 endif
-	go get -t -v -d -u ./...
 	go mod tidy
 	@echo "Dependencies updated"
 
@@ -106,9 +105,6 @@ update-schema:
 
 .PHONY: update-api
 update-api:
-ifeq "$(LXD_OFFLINE)" ""
-	(cd / ; GO111MODULE=on go get -v -x github.com/go-swagger/go-swagger/cmd/swagger)
-endif
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m
 
 .PHONY: debug
@@ -148,11 +144,6 @@ endif
 
 .PHONY: check
 check: default
-ifeq "$(LXD_OFFLINE)" ""
-	(cd / ; go get -v -x github.com/rogpeppe/godeps)
-	(cd / ; go get -v -x github.com/tsenart/deadcode)
-	(cd / ; go get -v -x golang.org/x/lint/golint)
-endif
 	CGO_LDFLAGS_ALLOW="$(CGO_LDFLAGS_ALLOW)" go test -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	cd test && ./main.sh
 
@@ -200,9 +191,6 @@ update-po:
 
 .PHONY: update-pot
 update-pot:
-ifeq "$(LXD_OFFLINE)" ""
-	(cd / ; go get -v -x github.com/snapcore/snapd/i18n/xgettext-go)
-endif
 	xgettext-go -o po/$(DOMAIN).pot --add-comments-tag=TRANSLATORS: --sort-output --package-name=$(DOMAIN) --msgid-bugs-address=lxc-devel@lists.linuxcontainers.org --keyword=i18n.G --keyword-plural=i18n.NG lxc/*.go lxc/*/*.go
 
 .PHONY: build-mo
-- 
2.34.1

