--- dhcpcd-6.1.0.orig/common.c
+++ dhcpcd-6.1.0/common.c
@@ -30,8 +30,6 @@
 #  define _GNU_SOURCE
 #endif
 
-#include <sys/cdefs.h>
-
 #ifdef __APPLE__
 #  include <mach/mach_time.h>
 #  include <mach/kern_return.h>
--- dhcpcd-6.1.0.orig/dhcp6.c
+++ dhcpcd-6.1.0/dhcp6.c
@@ -31,10 +31,6 @@
 #include <sys/utsname.h>
 
 #include <netinet/in.h>
-#ifdef __linux__
-#  define _LINUX_IN6_H
-#  include <linux/ipv6.h>
-#endif
 
 #include <ctype.h>
 #include <errno.h>
--- dhcpcd-6.1.0.orig/dhcpcd.c
+++ dhcpcd-6.1.0/dhcpcd.c
@@ -48,6 +48,7 @@
 #include <string.h>
 #include <syslog.h>
 #include <unistd.h>
+#include <fcntl.h>
 #include <time.h>
 
 #include "arp.h"
--- dhcpcd-6.1.0.orig/ipv6nd.c
+++ dhcpcd-6.1.0/ipv6nd.c
@@ -33,11 +33,6 @@
 #include <netinet/ip6.h>
 #include <netinet/icmp6.h>
 
-#ifdef __linux__
-#  define _LINUX_IN6_H
-#  include <linux/ipv6.h>
-#endif
-
 #include <errno.h>
 #include <stddef.h>
 #include <stdlib.h>
