--- a/Makefile.in
+++ b/Makefile.in
@@ -359,7 +359,7 @@
 tests_xargs_test_sigusr_SOURCES = tests/xargs/test-sigusr.c
 am__dirstamp = $(am__leading_dot)dirstamp
 tests_xargs_test_sigusr_OBJECTS = tests/xargs/test-sigusr.$(OBJEXT)
-tests_xargs_test_sigusr_LDADD = $(LDADD)
+tests_xargs_test_sigusr_LDADD = $(LDADD) gl/lib/libgnulib.a
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
 am__v_P_0 = false
@@ -2783,6 +2783,7 @@
 tests/xargs/$(DEPDIR)/$(am__dirstamp):
 	@$(MKDIR_P) tests/xargs/$(DEPDIR)
 	@: > tests/xargs/$(DEPDIR)/$(am__dirstamp)
+tests/xargs/test-sigusr.$(OBJEXT): CFLAGS=-Igl/lib
 tests/xargs/test-sigusr.$(OBJEXT): tests/xargs/$(am__dirstamp) \
 	tests/xargs/$(DEPDIR)/$(am__dirstamp)
 
