diff --git a/Makefile.in b/Makefile.in
index 5421f6c..1b69efe 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -384,7 +384,7 @@ am__EXEEXT_1 = tests/xargs/test-sigusr$(EXEEXT)
 tests_xargs_test_sigusr_SOURCES = tests/xargs/test-sigusr.c
 am__dirstamp = $(am__leading_dot)dirstamp
 tests_xargs_test_sigusr_OBJECTS = tests/xargs/test-sigusr.$(OBJEXT)
-tests_xargs_test_sigusr_LDADD = $(LDADD)
+tests_xargs_test_sigusr_LDADD = $(LDADD) gl/lib/libgnulib.a
 tests_xargs_test_sigusr_DEPENDENCIES =  \
 	$(top_builddir)/gl/lib/libgnulib.a
 AM_V_P = $(am__v_P_@AM_V@)
@@ -2984,6 +2984,7 @@ tests/xargs/$(am__dirstamp):
 tests/xargs/$(DEPDIR)/$(am__dirstamp):
 	@$(MKDIR_P) tests/xargs/$(DEPDIR)
 	@: >>tests/xargs/$(DEPDIR)/$(am__dirstamp)
+tests/xargs/test-sigusr.$(OBJEXT): CFLAGS=-Igl/lib
 tests/xargs/test-sigusr.$(OBJEXT): tests/xargs/$(am__dirstamp) \
 	tests/xargs/$(DEPDIR)/$(am__dirstamp)
 
