--- a/src/slave/kprop.c
+++ b/src/slave/kprop.c
@@ -91,7 +91,7 @@ main(argc, argv)
     int     argc;
     char    **argv;
 {
-    int     fd, database_fd, database_size;
+    int     fd = -1, database_fd, database_size;
     krb5_error_code retval;
     krb5_context context;
     krb5_creds *my_creds;
--- a/src/kadmin/ktutil/ktutil_funcs.c
+++ b/src/kadmin/ktutil/ktutil_funcs.c
@@ -64,7 +64,7 @@
     krb5_kt_list *list;
     int idx;
 {
-    krb5_kt_list lp, prev;
+    krb5_kt_list lp, prev = NULL;
     int i;
 
     for (lp = *list, i = 1; lp; prev = lp, lp = lp->next, i++) {
--- a/src/lib/kadm5/alt_prof.c
+++ b/src/lib/kadm5/alt_prof.c
@@ -164,7 +164,7 @@
     char **values;
     char *valp;
     int idx;
-    krb5_boolean val;
+    krb5_boolean val = 0;
 
     kret = krb5_aprof_getvals (acontext, hierarchy, &values);
     if (kret)
--- a/src/lib/krb5/unicode/ucstr.c
+++ b/src/lib/krb5/unicode/ucstr.c
@@ -109,7 +109,7 @@
             krb5_data ** newdataptr,
             unsigned flags)
 {
-    int i, j, len, clen, outpos, ucsoutlen, outsize;
+    int i, j, len, clen, outpos = 0, ucsoutlen, outsize;
     char *out = NULL, *outtmp, *s;
     krb5_ucs4 *ucs = NULL, *p, *ucsout = NULL;
     krb5_data *newdata;
diff --git a/src/util/profile/prof_init.c b/src/util/profile/prof_init.c
index 7dc5b47..cd90db8 100644
--- a/src/util/profile/prof_init.c
+++ b/src/util/profile/prof_init.c
@@ -255,7 +255,7 @@ copy_vtable_profile(profile_t profile, profile_t *ret_new_profile)
 {
     errcode_t err;
     void *cbdata;
-    profile_t new_profile;
+    profile_t new_profile = NULL;
 
     *ret_new_profile = NULL;
 
--- a/src/lib/krb5/krb/preauth2.c  2012-12-24 12:39:18.432678497 +0100
+++ b/src/lib/krb5/krb/preauth2.c   2012-12-24 12:50:49.444099126 +0100
@@ -956,7 +956,7 @@
     size_t i, h;
     int out_pa_list_size = 0;
     krb5_pa_data **out_pa_list = NULL;
-    krb5_error_code ret, module_ret;
+    krb5_error_code ret, module_ret = 0;
     krb5_responder_fn responder = opte->opt_private->responder;
     static const int paorder[] = { PA_INFO, PA_REAL };

--- a/src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c.orig	2013-02-15 14:38:43.742293824 +0000
+++ b/src/plugins/kdb/ldap/libkdb_ldap/ldap_principal2.c	2013-02-15 14:41:06.806870075 +0000
@@ -1359,8 +1359,8 @@
         goto cleanup;
 
     for (i=0; bvalues[i] != NULL; ++i) {
-        krb5_int16 n_kd;
-        krb5_key_data *kd;
+        krb5_int16 n_kd = 0;
+        krb5_key_data *kd = NULL;
         krb5_data in;
 
         if (bvalues[i]->bv_len == 0)
