--- ./scripts/linux/resume-vm-default.orig
+++ ./scripts/linux/resume-vm-default
@@ -25,37 +25,6 @@
 echo `date` ": Executing '$0'"
 echo
 
-find_networking_script() {
-    local script="error"
-    for dir in "/etc/init.d" "/sbin/init.d" "/etc" "/etc/rc.d" ; do
-        if [ -d "$dir/rc0.d" ] &&
-	    [ -d "$dir/rc1.d" ] &&
-	    [ -d "$dir/rc2.d" ] &&
-	    [ -d "$dir/rc3.d" ] &&
-	    [ -d "$dir/rc4.d" ] &&
-	    [ -d "$dir/rc5.d" ] &&
-	    [ -d "$dir/rc6.d" ]; then
-
-	    # Now find the appropriate networking script.
-	    if [ -d "$dir/init.d" ]; then
-		if [ -x "$dir/init.d/network" ]; then
-		    script="$dir/init.d/network"
-		elif [ -x "$dir/init.d/networking" ]; then
-		    script="$dir/init.d/networking"
-		fi
-	    else
-		if [ -x "$dir/network" ]; then
-		    script="$dir/network"
-		elif [ -x "$dir/networking" ]; then
-		    script="$dir/networking"
-		fi
-	    fi
-        fi
-    done
-
-    echo "$script"
-}
-
 rescue_NIC() {
    niclist="/var/run/vmware-active-nics"
 
@@ -121,7 +90,7 @@
 
 wakeNetworkManager
 
-network=`find_networking_script`
+network=/etc/init.d/networking
 if [ "$network" != "error" ]; then
    "$network" restart
    # Continue even if the networking init script wasn't successful.
