From 46e1404ddad66422bd2106ff6e6954741191826e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Adamski?= <michal@ert.pl>
Date: Wed, 14 Oct 2020 11:06:33 +0200
Subject: [PATCH] `apk add` instead of `apt-get install`


diff --git a/electrum/gui/qt/__init__.py b/electrum/gui/qt/__init__.py
index 1e1d2831f..b7fe2c42d 100644
--- a/electrum/gui/qt/__init__.py
+++ b/electrum/gui/qt/__init__.py
@@ -34,7 +34,7 @@ from typing import Optional, TYPE_CHECKING, List
     from electrum import GuiImportError
     raise GuiImportError(
         "Error: Could not import PyQt5. On Linux systems, "
-        "you may try 'sudo apt-get install python3-pyqt5'") from e
+        "you may try 'doas apk add py3-qt5'") from e
 
 from PyQt5.QtGui import QGuiApplication
 from PyQt5.QtWidgets import QApplication, QSystemTrayIcon, QWidget, QMenu, QMessageBox
-- 
2.25.1

