Author: Piotr Engelking <inkerman42@gmail.com>
Description: Disable the filtering packets with incorrect checksum (Closes: #372536).
--- a/urlsnarf.c	2011-06-19 17:15:10.559999375 -0500
+++ b/urlsnarf.c	2011-06-19 17:15:18.023999373 -0500
@@ -200,6 +200,7 @@
 	extern char *optarg;
 	extern int optind;
 	int c;
+	struct nids_chksum_ctl chksum_ctl;
 	
 	while ((c = getopt(argc, argv, "i:p:nvh?V")) != -1) {
 		switch (c) {
@@ -260,6 +261,12 @@
                 }
         }
 
+        chksum_ctl.netaddr = 0;
+        chksum_ctl.mask = 0;
+        chksum_ctl.action = NIDS_DONT_CHKSUM;
+
+        nids_register_chksum_ctl(&chksum_ctl, 1);
+
 	nids_run();
 	
 	/* NOTREACHED */
