Author: Luciano Bello <luciano@linux.org.ar>
Description: Aviod the "implicit declaration of function 'str*'" warning
--- a/arp.c	2011-06-19 17:15:04.587999376 -0500
+++ b/arp.c	2011-06-19 17:15:23.487999372 -0500
@@ -34,6 +34,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <string.h>
 
 #include "arp.h"
 
--- a/buf.c	2011-06-19 17:13:56.463999389 -0500
+++ b/buf.c	2011-06-19 17:15:23.487999372 -0500
@@ -17,6 +17,7 @@
 #include <unistd.h>
 #include <ctype.h>
 #include <err.h>
+#include <string.h>
 
 #include "buf.h"
 
--- a/decode_nntp.c	2011-06-19 17:13:53.631999390 -0500
+++ b/decode_nntp.c	2011-06-19 17:15:23.491999372 -0500
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "decode.h"
--- a/decode_pop.c	2011-06-19 17:13:54.911999390 -0500
+++ b/decode_pop.c	2011-06-19 17:15:23.495999372 -0500
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
--- a/decode_rlogin.c	2011-06-19 17:13:54.431999390 -0500
+++ b/decode_rlogin.c	2011-06-19 17:15:23.495999372 -0500
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
+#include <strlcat.h>
 
 #include "options.h"
 #include "decode.h"
--- a/decode_smb.c	2011-06-19 17:13:55.103999390 -0500
+++ b/decode_smb.c	2011-06-19 17:15:23.499999372 -0500
@@ -15,6 +15,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_smtp.c	2011-06-19 17:13:54.003999391 -0500
+++ b/decode_smtp.c	2011-06-19 17:15:23.503999372 -0500
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "base64.h"
 #include "options.h"
--- a/decode_sniffer.c	2011-06-19 17:13:56.075999390 -0500
+++ b/decode_sniffer.c	2011-06-19 17:15:23.503999372 -0500
@@ -15,6 +15,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "base64.h"
 #include "decode.h"
--- a/decode_socks.c	2011-06-19 17:13:55.271999391 -0500
+++ b/decode_socks.c	2011-06-19 17:15:23.507999372 -0500
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_tds.c	2011-06-19 17:13:55.899999390 -0500
+++ b/decode_tds.c	2011-06-19 17:15:23.511999372 -0500
@@ -18,6 +18,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
 
 #include "decode.h"
 
--- a/decode_telnet.c	2011-06-19 17:13:55.491999391 -0500
+++ b/decode_telnet.c	2011-06-19 17:15:23.515999372 -0500
@@ -14,6 +14,7 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcpy.h>
 
 #include "options.h"
 #include "decode.h"
--- a/decode_x11.c	2011-06-19 17:13:53.823999390 -0500
+++ b/decode_x11.c	2011-06-19 17:15:23.515999372 -0500
@@ -14,6 +14,8 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <strlcat.h>
+#include <strlcpy.h>
 
 #include "decode.h"
 
--- a/dnsspoof.c	2011-06-19 17:15:10.515999375 -0500
+++ b/dnsspoof.c	2011-06-19 17:15:23.519999372 -0500
@@ -20,6 +20,7 @@
 #include <stdlib.h>
 #include <signal.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <resolv.h>
 #include <err.h>
 #include <libnet.h>
--- a/magic.c	2011-06-19 17:13:55.703999390 -0500
+++ b/magic.c	2011-06-19 17:15:23.523999372 -0500
@@ -36,6 +36,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strlcpy.h>
 #include <ctype.h>
 #include <time.h>
 #include <err.h>
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ b/missing/strlcat.h	2011-06-19 17:15:23.527999372 -0500
@@ -0,0 +1 @@
+size_t strlcat(char *dst, const char *src, size_t siz);
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ b/missing/strlcpy.h	2011-06-19 17:15:23.531999372 -0500
@@ -0,0 +1 @@
+size_t strlcpy(char *dst, const char *src, size_t siz);
--- a/sshmitm.c	2011-06-19 17:15:10.527999375 -0500
+++ b/sshmitm.c	2011-06-19 17:15:23.531999372 -0500
@@ -24,6 +24,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
+#include <strlcat.h>
 
 #include "buf.h"
 #include "record.h"
