From 2425e14706120233ea8a3caa34660b714808d882 Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sun, 22 Dec 2024 13:58:05 +0100
Subject: [PATCH] Apply missing version bump to 1.3.0

---
 conda/meta.yaml | 2 +-
 minimock.py     | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/conda/meta.yaml b/conda/meta.yaml
index 4ceea26..d4eeeff 100644
--- a/conda/meta.yaml
+++ b/conda/meta.yaml
@@ -1,5 +1,5 @@
 {% set name = "MiniMock" %}
-{% set version = "1.2.10.dev0" %}
+{% set version = "1.3.0" %}
 
 package:
   name: {{ name|lower }}
diff --git a/minimock.py b/minimock.py
index 16b891f..7054f7a 100644
--- a/minimock.py
+++ b/minimock.py
@@ -40,7 +40,7 @@ implementation is simple because most of the work is done by doctest.
 
 __all__ = ("mock", "restore", "Mock", "TraceTracker", "assert_same_trace")
 
-__version__ = '1.2.10.dev0'  # note: also bump version in conda/meta.yaml
+__version__ = '1.3.0'  # note: also bump version in conda/meta.yaml
 
 import sys
 import inspect
-- 
2.47.1

