diff --git a/src/Makefile.in b/src/Makefile.in
index bdfadb1cc909..8d3ce3b6278b 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -347,7 +347,7 @@ powertop_SOURCES = parameters/persistent.cpp parameters/learn.cpp parameters/par
 powertop_CXXFLAGS = -fno-omit-frame-pointer -fstack-protector -Wall -Wshadow -Wformat $(NCURSES_CFLAGS) $(PCIUTILS_CFLAGS) $(LIBNL_CFLAGS) $(GLIB2_CFLAGS)
 powertop_CPPFLAGS = -D_FORTIFY_SOURCE=2 $(NCURSES_CFLAGS) $(PCIUTILS_CFLAGS) $(LIBNL_CFLAGS) $(GLIB2_CFLAGS) $(LIBZ_CFLAGS) -DLOCALEDIR=\"$(localedir)\"
 powertop_LDADD = ../traceevent/libtraceevnet.la
-AM_LDFLAGS = $(LIBS) $(NCURSES_LIBS) $(PCIUTILS_LIBS) $(LIBNL_LIBS) $(LIBZ_LIBS) $(PTHREAD_LIBS)  $(RESOLV_LIBS)
+AM_LDFLAGS = $(LIBS) $(NCURSES_LIBS) $(PCIUTILS_LIBS) $(LIBNL_LIBS) $(LIBZ_LIBS) $(PTHREAD_LIBS)  $(RESOLV_LIBS) $(LTLIBINTL)
 BUILT_SOURCES = css.h
 CLEANFILES = css.h
 all: $(BUILT_SOURCES)
