Fix fiber test to be compatible with musl libc.

--- a/test/unit/fiber.result
+++ b/test/unit/fiber.result
@@ -1,6 +1,6 @@
-#gh-1238: log uncaught errors
-SystemError Failed to allocate 42 bytes in allocator for exception: Cannot allocate memory
 	*** fiber_name_test ***
+#gh-1238: log uncaught errors
+SystemError Failed to allocate 42 bytes in allocator for exception: Out of memory
 # name of a new fiber: main.
 
 # set new fiber name: Horace.
