--- ettercap.orig/configure.in
+++ ettercap/configure.in
@@ -20,7 +20,7 @@
 dnl    $Id: ettercap-0.7.4-autotools.patch,v 1.2 2011/12/09 20:14:11 radhermit Exp $
 dnl
 
-AC_INIT
+AC_INIT(ettercap, 0.7.4)
 AC_CONFIG_SRCDIR([include/ec.h])
 
 dnl Force autoconf 2.57 
@@ -32,16 +32,8 @@
 dnl Do not recompile configure Makefile et similia
 AM_MAINTAINER_MODE
 
-VERSION=`grep EC_VERSION\  ./include/ec_version.h | cut -f2 -d"\""`
-SB=`./shtool echo -n -e %B`
-EB=`./shtool echo -n -e %b`
-AC_SUBST(SB)
-AC_SUBST(EB)
-AC_SUBST(VERSION)
-EC_MESSAGE(Configuring ettercap $VERSION)
-
 AC_CONFIG_SRCDIR(src)
-AM_INIT_AUTOMAKE(ettercap, $VERSION)
+AM_INIT_AUTOMAKE([subdir-objects])
 
 AC_CONFIG_HEADERS([include/config.h])
 dnl AC_CONFIG_HEADERS(include/config.h)
@@ -285,29 +277,18 @@
 
 AC_CONFIG_LIBOBJ_DIR([src/missing])
 
-AM_CONDITIONAL(MISSING_STRLCPY, false)
-AM_CONDITIONAL(MISSING_STRLCAT, false)
-AM_CONDITIONAL(MISSING_STRSEP, false)
-AM_CONDITIONAL(MISSING_MEMMEM, false)
-AM_CONDITIONAL(MISSING_MEMCMP, false)
-AM_CONDITIONAL(MISSING_BASENAME, false)
-AM_CONDITIONAL(MISSING_GETOPT_LONG, false)
-AM_CONDITIONAL(MISSING_STRCASESTR, false)
-AM_CONDITIONAL(MISSING_INET_ATON, false)
-AM_CONDITIONAL(MISSING_SCANDIR, false)
-
 dnl AC_REPLACE_FUNCS(strlcpy)
 
-AC_CHECK_FUNCS([strlcpy], , [AC_LIBOBJ(strlcpy) AM_CONDITIONAL(MISSING_STRLCPY, true)])
-AC_CHECK_FUNCS([strlcat], , [AC_LIBOBJ(strlcat) AM_CONDITIONAL(MISSING_STRLCAT, true)])
-AC_CHECK_FUNCS([strsep], , [AC_LIBOBJ(strsep) AM_CONDITIONAL(MISSING_STRSEP, true)])
-AC_CHECK_FUNCS([memmem], , [AC_LIBOBJ(memmem) AM_CONDITIONAL(MISSING_MEMMEM, true)])
-AC_CHECK_FUNCS([memcmp], , [AC_LIBOBJ(memcmp) AM_CONDITIONAL(MISSING_MEMCMP, true)])
-AC_CHECK_FUNCS([basename], , [AC_LIBOBJ(basename) AM_CONDITIONAL(MISSING_BASENAME, true)])
-AC_CHECK_FUNCS([getopt_long], , [AC_LIBOBJ(getopt) AM_CONDITIONAL(MISSING_GETOPT_LONG, true)])
-AC_CHECK_FUNCS([strcasestr], , [AC_LIBOBJ(strcasestr) AM_CONDITIONAL(MISSING_STRCASESTR, true)])
-AC_CHECK_FUNCS([scandir], , [AC_LIBOBJ(scandir) AM_CONDITIONAL(MISSING_SCANDIR, true)])
-AC_CHECK_FUNCS([inet_aton], , [AC_LIBOBJ(inet_aton) AM_CONDITIONAL(MISSING_INET_ATON, true)])
+AC_CHECK_FUNCS([strlcpy], , [AC_LIBOBJ(strlcpy)])
+AC_CHECK_FUNCS([strlcat], , [AC_LIBOBJ(strlcat)])
+AC_CHECK_FUNCS([strsep], , [AC_LIBOBJ(strsep)])
+AC_CHECK_FUNCS([memmem], , [AC_LIBOBJ(memmem)])
+AC_CHECK_FUNCS([memcmp], , [AC_LIBOBJ(memcmp)])
+AC_CHECK_FUNCS([basename], , [AC_LIBOBJ(basename)])
+AC_CHECK_FUNCS([getopt_long], , [AC_LIBOBJ(getopt)])
+AC_CHECK_FUNCS([strcasestr], , [AC_LIBOBJ(strcasestr)])
+AC_CHECK_FUNCS([scandir], , [AC_LIBOBJ(scandir)])
+AC_CHECK_FUNCS([inet_aton], , [AC_LIBOBJ(inet_aton)])
 
 AC_CHECK_LIB(resolv, inet_aton,
    AM_CONDITIONAL(INET_ATON_WANTS_RESOLV, true), 
@@ -561,7 +542,7 @@
 dnl ---------
 
 AH_TEMPLATE(HAVE_PCRE, [perl compatible regex])
-
+AM_CONDITIONAL(HAVE_PCRE, false)
 ac_ec_pcre=no
 
 AC_MSG_CHECKING(for libpcre)
--- ettercap.orig/utils/etterfilter/Makefile.am
+++ ettercap/utils/etterfilter/Makefile.am
@@ -8,7 +8,8 @@
 
 bin_PROGRAMS = etterfilter
 
-AM_YFLAGS = --defines
+BUILT_SOURCES = ef_grammar.h ef_syntax.c
+AM_YFLAGS = -d
 
 etterfilter_SOURCES = ef_compiler.c \
                       ef_main.c \
@@ -26,22 +27,6 @@
                       $(top_srcdir)/src/ec_strings.c
 
 noinst_HEADERS = ef_grammar.h
-                   
-if MISSING_STRSEP
-   etterfilter_SOURCES += $(top_srcdir)/src/missing/strsep.c
-endif
-if MISSING_STRLCPY
-   etterfilter_SOURCES += $(top_srcdir)/src/missing/strlcpy.c
-endif
-if MISSING_MEMMEM
-   etterfilter_SOURCES += $(top_srcdir)/src/missing/memmem.c
-endif
-if MISSING_INET_ATON
-   etterfilter_SOURCES += $(top_srcdir)/src/missing/inet_aton.c
-endif
-if MISSING_GETOPT_LONG
-   etterfilter_SOURCES += $(top_srcdir)/src/missing/getopt.c
-endif
 
 etterfilter_CFLAGS = @EC_CFLAGS@
 
@@ -49,7 +34,7 @@
    etterfilter_CFLAGS += -pthread
 endif
    
-etterfilter_LDADD = 
+etterfilter_LDADD = @LIBOBJS@
 
 if INET_ATON_WANTS_RESOLV
    etterfilter_LDADD += -lresolv
--- ettercap.orig/utils/etterlog/Makefile.am
+++ ettercap/utils/etterlog/Makefile.am
@@ -30,32 +30,10 @@
                    $(top_srcdir)/src/ec_passive.c \
                    $(top_srcdir)/src/ec_strings.c \
                    $(top_srcdir)/src/ec_format.c
-                   
-if MISSING_STRSEP
-   etterlog_SOURCES += $(top_srcdir)/src/missing/strsep.c
-endif
-if MISSING_STRLCAT
-   etterlog_SOURCES += $(top_srcdir)/src/missing/strlcat.c
-endif
-if MISSING_STRCASESTR
-   etterlog_SOURCES += $(top_srcdir)/src/missing/strcasestr.c
-endif
-if MISSING_MEMMEM
-   etterlog_SOURCES += $(top_srcdir)/src/missing/memmem.c
-endif
-if MISSING_BASENAME
-   etterlog_SOURCES += $(top_srcdir)/src/missing/basename.c
-endif
-if MISSING_INET_ATON
-   etterlog_SOURCES += $(top_srcdir)/src/missing/inet_aton.c
-endif
-if MISSING_GETOPT_LONG
-   etterlog_SOURCES += $(top_srcdir)/src/missing/getopt.c
-endif
 
 etterlog_CFLAGS = @EC_CFLAGS@
 
-etterlog_LDADD = -lz @ICONVLIB@
+etterlog_LDADD = -lz @ICONVLIB@ @LIBOBJS@
 
 if INET_ATON_WANTS_RESOLV
    etterlog_LDADD += -lresolv
--- ettercap.orig/Makefile.am
+++ ettercap/Makefile.am
@@ -43,6 +43,6 @@
 	./shtool fixperm ./
 
 install-exec-hook:
-	rm -f $(libdir)/ettercap/*.la
+	rm -f $(DESTDIR)$(libdir)/ettercap/*.la
 # vim:ts=4:noexpandtab
 
