diff --git a/src/common/rendering/vulkan/thirdparty/vk_mem_alloc/vk_mem_alloc.h b/src/common/rendering/vulkan/thirdparty/vk_mem_alloc/vk_mem_alloc.h
index fd44722..8ec07c6 100644
--- a/src/common/rendering/vulkan/thirdparty/vk_mem_alloc/vk_mem_alloc.h
+++ b/src/common/rendering/vulkan/thirdparty/vk_mem_alloc/vk_mem_alloc.h
@@ -2256,6 +2256,7 @@ Library has its own container implementation.
 Following headers are used in this CONFIGURATION section only, so feel free to
 remove them if not needed.
 */
+#include <cstdio>
 #include <cassert> // for assert
 #include <algorithm> // for min, max
 #include <mutex>
