diff --git a/champlain/champlain-network-tile-source.c b/champlain/champlain-network-tile-source.c
index 7cadfa5..ad24fd4 100644
--- a/champlain/champlain-network-tile-source.c
+++ b/champlain/champlain-network-tile-source.c
@@ -994,8 +994,8 @@ get_modified_time_string (ChamplainTile *tile)
 
   if (time == NULL)
     return NULL;
-
-  struct tm *other_time = gmtime (&time->tv_sec);
+  time_t sec = time->tv_sec;
+  struct tm *other_time = gmtime (&sec);
   char value[100];
 
 #ifdef G_OS_WIN32
