diff --git a/meson.build b/meson.build
index 5fd81ac..140a8ce 100644
--- a/meson.build
+++ b/meson.build
@@ -3,8 +3,6 @@ project('GlibD', 'd',
     version: '2.1.0'
 )
 
-project_soversion = 0
-
 pkg_conf = import('pkgconfig')
 
 source_root = meson.current_source_dir()
@@ -46,7 +44,6 @@ glibd = library('glibd-2.0',
     include_directories: [gir_bind_dir],
     dependencies: [glib_dep, gmodule_dep, gobject_dep, gio_dep],
     install: true,
-    soversion: project_soversion,
     version: meson.project_version())
 
 install_subdir(join_paths(build_root, gen_dir, 'glib'),    install_dir: 'include/d/glibd-2/')
