diff -ruN a/imgproc/clir_linux.c b/imgproc/clir_linux.c
--- a/imgproc/clir_linux.c	2023-05-14 05:40:17.000000000 +0000
+++ b/imgproc/clir_linux.c	2025-08-08 00:08:54.480000000 +0000
@@ -399,12 +399,12 @@
 
   if (verbose)
     printf("imgproc: loading clGetPlatformIDs\n");
-  clGetPlatformIDs = (cl_int(*)())dlsym(libOpenCL, "clGetPlatformIDs");
+  clGetPlatformIDs = (cl_int(*)(cl_uint,  void **, cl_uint *))dlsym(libOpenCL, "clGetPlatformIDs");
   if (clGetPlatformIDs == NULL)
     return -1;
   if (verbose)
     printf("imgproc: loading clGetDeviceIDs\n");
-  clGetDeviceIDs = (cl_int(*)())dlsym(libOpenCL, "clGetDeviceIDs");
+  clGetDeviceIDs = (cl_int(*)(void *, cl_device_type, cl_uint,  void **, cl_uint *))dlsym(libOpenCL, "clGetDeviceIDs");
   if (clGetDeviceIDs == NULL)
     return -1;
   if (verbose)
@@ -415,22 +415,22 @@
     return -1;
   if (verbose)
     printf("imgproc: loading clCreateContext\n");
-  clCreateContext = (void *(*)())dlsym(libOpenCL, "clCreateContext");
+  clCreateContext = (void *(*)(cl_context_properties *, cl_uint,  void **, void *, void *, cl_int *))dlsym(libOpenCL, "clCreateContext");
   if (clCreateContext == NULL)
     return -1;
   if (verbose)
     printf("imgproc: loading clCreateProgramWithSource\n");
   clCreateProgramWithSource =
-      (void *(*)())dlsym(libOpenCL, "clCreateProgramWithSource");
+      (void *(*)(void *, cl_uint, const char **, size_t *, cl_int *))dlsym(libOpenCL, "clCreateProgramWithSource");
   if (clCreateProgramWithSource == NULL)
     return -1;
-  clCreateImage = (void *(*)())dlsym(libOpenCL, "clCreateImage");
+  clCreateImage = (void *(*)(void *, cl_mem_flags, cl_image_format *, cl_image_desc *, void *, cl_int *))dlsym(libOpenCL, "clCreateImage");
   if (clCreateImage == NULL)
     return -1;
   if (verbose)
     printf("imgproc: loading clCreateCommandQueueWithProperties\n");
   clCreateCommandQueueWithProperties =
-      (void *(*)())dlsym(libOpenCL, "clCreateCommandQueueWithProperties");
+      (void *(*)(void *, void *, void *, cl_int *))dlsym(libOpenCL, "clCreateCommandQueueWithProperties");
   if (clCreateCommandQueueWithProperties == NULL)
     return -1;
   if (verbose)
