diff --git a/plugins/itemfakevim/fakevim/fakevimhandler.cpp b/plugins/itemfakevim/fakevim/fakevimhandler.cpp
index ec00168ef..a29996539 100644
--- a/plugins/itemfakevim/fakevim/fakevimhandler.cpp
+++ b/plugins/itemfakevim/fakevim/fakevimhandler.cpp
@@ -1057,14 +1057,6 @@ inline QString msgMarkNotSet(const QString &text)
     return Tr::tr("Mark \"%1\" not set.").arg(text);
 }
 
-static void initSingleShotTimer(QTimer *timer, int interval, FakeVimHandler::Private *receiver,
-                                void (FakeVimHandler::Private::*slot)())
-{
-    timer->setSingleShot(true);
-    timer->setInterval(interval);
-    QObject::connect(timer, &QTimer::timeout, receiver, slot);
-}
-
 class Input
 {
 public:
@@ -2424,6 +2416,16 @@ class FakeVimHandler::Private : public QObject
     FakeVimSettings &s = *fakeVimSettings();
 };
 
+static void initSingleShotTimer(QTimer *timer,
+                                int interval,
+                                FakeVimHandler::Private *receiver,
+                                void (FakeVimHandler::Private::*slot)())
+{
+    timer->setSingleShot(true);
+    timer->setInterval(interval);
+    QObject::connect(timer, &QTimer::timeout, receiver, slot);
+}
+
 FakeVimHandler::Private::GlobalData FakeVimHandler::Private::g;
 
 FakeVimHandler::Private::Private(FakeVimHandler *parent, QWidget *widget)
