diff --git a/setup.py b/setup.py
index 24dc9fb..dd33b78 100755
--- a/setup.py
+++ b/setup.py
@@ -7,3 +7,4 @@ PROJ_METADATA = '%s.json' % PROJ_NAME
 
-import os, json, imp
+import os, json
+from importlib.machinery import SourceFileLoader
 here = os.path.abspath(os.path.dirname(__file__))
@@ -15,3 +16,3 @@ except:
 CHANGELOG = open(os.path.join(here, 'CHANGELOG.rst'), encoding='utf-8').read()
-VERSION = imp.load_source('version', os.path.join(here, 'src/%s/version.py' % PACKAGE_NAME)).__version__
+VERSION = SourceFileLoader('version', os.path.join(here, 'src/%s/version.py' % PACKAGE_NAME)).load_module().__version__
 
