--- eudev-3.2.1/rules/80-drivers.rules.orig	2017-01-24 11:00:53.174192908 +0200
+++ eudev-3.2.1/rules/80-drivers.rules	2017-01-24 11:04:01.886714118 +0200
@@ -8,6 +8,7 @@
 SUBSYSTEM=="memstick", RUN{builtin}+="kmod load ms_block mspro_block"
 SUBSYSTEM=="i2o", RUN{builtin}+="kmod load i2o_block"
 SUBSYSTEM=="module", KERNEL=="parport_pc", RUN{builtin}+="kmod load ppdev"
+SUBSYSTEM=="graphics", RUN{builtin}+="kmod load fbcon"
 KERNEL=="mtd*ro", ENV{MTD_FTL}=="smartmedia", RUN{builtin}+="kmod load sm_ftl"
 
 LABEL="drivers_end"
