diff --git a/test/Makefile.am b/test/Makefile.am
index 3605dea..1e676bd 100644
--- a/test/Makefile.am
+++ b/test/Makefile.am
@@ -72,7 +72,7 @@ TESTS = test-hex_string_to_array.sh \
 		test-airdecap-ng-0004.sh
 
 if HAVE_SQLITE3
-TESTS += test-airolib-ng-0001.sh
+# TESTS += test-airolib-ng-0001.sh
 endif
 
 EXTRA_DIST = test-hex_string_to_array.sh \
@@ -104,7 +104,7 @@ EXTRA_DIST = test-hex_string_to_array.sh \
 			 test-airolib-sqlite.sh
 
 if HAVE_SQLITE3
-EXTRA_DIST += test-airolib-ng-0001.sh
+# EXTRA_DIST += test-airolib-ng-0001.sh
 endif
 
 
