diff --git a/src/sighand.c b/src/sighand.c
index c82e7e3..4d7001e 100644
--- a/src/sighand.c
+++ b/src/sighand.c
@@ -22,8 +22,9 @@
  * @file sighand.c
  * @brief Provides signal handling functions.
  */
-
+#ifdef HAVE_EXECINFO_H
 #include <execinfo.h> /* for backtrace() */
+#endif
 #include <glib.h>     /* for G_LOG_DOMAIN, for g_critical() */
 #include <signal.h>   /* for kill() */
 #include <sys/wait.h> /* for waitpid() */
@@ -106,7 +107,7 @@ sighand_chld (pid_t pid)
 
   waitpid (pid, &status, WNOHANG);
 }
-
+#ifdef HAVE_EXECINFO_H
 static void
 print_trace ()
 {
@@ -125,12 +126,15 @@ print_trace ()
 
   g_free (strings);
 }
+#endif
 
 void
 sighand_segv (int given_signal)
 {
   signal (SIGSEGV, _exit);
+#ifdef HAVE_EXECINFO_H
   print_trace ();
+#endif
   make_em_die (SIGTERM);
   /* Raise signal again, to exit with the correct return value,
    * and to enable core dumping. */
