--- emelfm2-0.9.1-origin/src/filesystem/e2_fs_mount.c
+++ emelfm2-0.9.1/src/filesystem/e2_fs_mount.c
@@ -46,7 +46,7 @@
 #endif
 //other exclusions go here ...
 
-# if defined(__linux__)
+# if defined(__GLIBC__)
 #  include <mntent.h>
 #  include <fstab.h>
 
@@ -62,7 +62,9 @@
 
 # else
 #  include <mntent.h>
+#  if defined(__GLIBC__)
 #  include <fstab.h>
+#  endif
 
 /*Some systems use statfs() to provide information about mounted
 file systems, other systems use statvfs(). The header files used with
@@ -1020,7 +1022,7 @@
 {
 	GList *fstab_list = NULL;
 	//CHECKME lock data file while accessing it ??
-#if defined(__E2BSD__) || defined(__linux__)
+#if defined(__E2BSD__) || defined(__GLIBC__)
     struct fstab *fs;
 
 	if (!setfsent ())
