diff --git a/tmuxinator.gemspec b/tmuxinator.gemspec
index 1552b49..53de3fd 100644
--- a/tmuxinator.gemspec
+++ b/tmuxinator.gemspec
@@ -41,7 +41,7 @@ Gem::Specification.new do |s|
 
   s.add_dependency "erubi", "~> 1.10"
   s.add_dependency "thor", "~> 1.0"
-  s.add_dependency "xdg", "~> 2.2", ">= 2.2.5"
+  s.add_dependency "xdg", "~> 5.3"
 
   s.add_development_dependency "activesupport", "< 5.0.0" # Please see issue #432
   s.add_development_dependency "awesome_print", "~> 1.2"
diff --git a/lib/tmuxinator/config.rb b/lib/tmuxinator/config.rb
index 9b64b75..79024c7 100644
--- a/lib/tmuxinator/config.rb
+++ b/lib/tmuxinator/config.rb
@@ -29,7 +29,8 @@ module Tmuxinator
       # a custom value. (e.g. if $XDG_CONFIG_HOME is set to ~/my-config, the
       # return value will be ~/my-config/tmuxinator)
       def xdg
-        XDG["CONFIG"].to_s + "/tmuxinator"
+        config = XDG::Config.new
+        (config.home + 'tmuxinator').to_s
       end
 
       def xdg?
