--- a/Tests/kstars_ui/CMakeLists.txt
+++ b/Tests/kstars_ui/CMakeLists.txt
@@ -129,7 +129,7 @@
 ADD_TEST(NAME TestEkosScheduler COMMAND test_ekos_scheduler)
 SET_TESTS_PROPERTIES( TestEkosScheduler PROPERTIES LABELS "stable;ui" TIMEOUT 120 )
 
-ADD_EXECUTABLE(test_ekos_scheduler_ops ${KSTARS_UI_EKOS_SRC} ${KSTARS_UI_MOCK_SRC} test_ekos_scheduler_helper.cpp test_ekos_scheduler_ops.cpp)
+ADD_EXECUTABLE(test_ekos_scheduler_ops ${KSTARS_UI_EKOS_SRC} ${KSTARS_UI_MOCK_SRC} test_ekos_helper.cpp test_ekos_scheduler_helper.cpp test_ekos_scheduler_ops.cpp)
 TARGET_LINK_LIBRARIES(test_ekos_scheduler_ops ${KSTARS_UI_EKOS_LIBS})
 ADD_CUSTOM_COMMAND( TARGET test_ekos_scheduler_ops POST_BUILD
     COMMAND ${CMAKE_COMMAND} -E copy
