--- ecryptfs-utils-111.orig/src/utils/mount.ecryptfs_private.c	2016-01-22 18:04:52.000000000 +0200
+++ ecryptfs-utils-111/src/utils/mount.ecryptfs_private.c	2017-01-20 08:39:31.877975729 +0200
@@ -235,7 +235,7 @@
 	 * deceive other programs with a crafted /proc/self/*. See
 	 * https://launchpad.net/bugs/1530566 for more details.
 	 */
-	__SWORD_TYPE f_type_whitelist[] = {
+	static const uint32_t f_type_whitelist[] = {
 		0x61756673 /* AUFS_SUPER_MAGIC */,
 		0x9123683E /* BTRFS_SUPER_MAGIC */,
 		0x00C36400 /* CEPH_SUPER_MAGIC */,
