diff --git a/lib/arm/cpu_features.h b/lib/arm/cpu_features.h
index edcf735..fcfe884 100644
--- a/lib/arm/cpu_features.h
+++ b/lib/arm/cpu_features.h
@@ -161,6 +161,7 @@ static inline u32 get_arm_cpu_features(void) { return 0; }
  */
 #if HAVE_CRC32_NATIVE || \
 	(HAVE_DYNAMIC_ARM_CPU_FEATURES && \
+	 (_ARM_ARCH >= 7) &&
 	 ((__has_builtin(__builtin_arm_crc32b) && !GCC_PREREQ(1, 0)) || \
 	  GCC_PREREQ(11, 3) || \
 	  (GCC_PREREQ(10, 4) && !GCC_PREREQ(11, 0)) || \
