--- a/lib/arm/cpu_features.h
+++ b/lib/arm/cpu_features.h
@@ -133,11 +133,12 @@
  * way to detect the binutils version directly from a C source file.
  */
 #define HAVE_CRC32_INTRIN \
+	(__ARM_ARCH >= 7 && \
 	(HAVE_CRC32_NATIVE || (HAVE_CRC32_TARGET && \
 			       (!GCC_PREREQ(1, 0) || \
 				GCC_PREREQ(11, 3) || \
 				(GCC_PREREQ(10, 4) && !GCC_PREREQ(11, 0)) || \
-				(GCC_PREREQ(9, 5) && !GCC_PREREQ(10, 0)))))
+				(GCC_PREREQ(9, 5) && !GCC_PREREQ(10, 0))))))
 
 /* SHA3 (needed for the eor3 instruction) */
 #ifdef __aarch64__
