diff --git a/booster-qtcomponents.pro b/booster-qtcomponents.pro
index 459a355..d67f7b3 100644
--- a/booster-qtcomponents.pro
+++ b/booster-qtcomponents.pro
@@ -14,8 +14,8 @@ service.files = data/$${TARGET}.service
 
 INSTALLS += target qml service
 
-LIBS += -lapplauncherd
-INCLUDEPATH += /usr/include/applauncherd/
+LIBS += -lapplauncherd -lmdeclarativecache5
+INCLUDEPATH += /usr/include/applauncherd/ /usr/include/mdeclarativecache5/
 
 SOURCES += src/qmlbooster.cpp src/eventhandler.cpp
 HEADERS += src/qmlbooster.h src/eventhandler.h
