diff --git a/src/dmabuf.rs b/src/dmabuf.rs
index f9ceaba..f2fa7c4 100644
--- a/src/dmabuf.rs
+++ b/src/dmabuf.rs
@@ -469,7 +469,7 @@ unsafe fn ioctl_loop(
 ) -> Result<(), String> {
     let req = code as libc::c_ulong;
     loop {
-        let ret = libc::ioctl(drm_fd.as_raw_fd(), req, arg);
+        let ret = libc::ioctl(drm_fd.as_raw_fd(), req.try_into().unwrap(), arg);
         let errno = errno::Errno::last_raw();
         if ret == 0 {
             return Ok(());
