--- a/src/dmabuf.rs
+++ b/src/dmabuf.rs
@@ -601,7 +601,7 @@
 ) -> Result<(), String> {
     let req = code as libc::c_ulong;
     loop {
-        let ret = libc::ioctl(drm_fd.as_raw_fd(), req, arg);
+        let ret = libc::ioctl(drm_fd.as_raw_fd(), req.try_into().unwrap(), arg);
         let errno = errno::Errno::last_raw();
         if ret == 0 {
             return Ok(());
@@ -697,7 +697,7 @@
         let req = code as libc::c_ulong;
         let arg: *mut c_void = &mut x as *mut _ as *mut c_void;
         loop {
-            let ret = libc::ioctl(dmabuf_fd.as_raw_fd(), req, arg);
+            let ret = libc::ioctl(dmabuf_fd.as_raw_fd(), req.try_into().unwrap(), arg);
             let errno = errno::Errno::last_raw();
             if ret == 0 {
                 break;
