--- a/lastversion/FeedRepoSession.py
+++ b/lastversion/FeedRepoSession.py
@@ -19,7 +19,7 @@
 
     # https://alex.miller.im/posts/python-3-feedfinder-rss-detection-from-url/
     def find_feed(self, site):
-        from six.moves.urllib.parse import urlparse
+        from urllib.parse import urlparse
         # noinspection PyPep8Naming
         from bs4 import BeautifulSoup as bs4
         raw = self.get(site).text
--- a/lastversion/GitHubRepoSession.py
+++ b/lastversion/GitHubRepoSession.py
@@ -8,7 +8,7 @@
 
 import feedparser
 from dateutil import parser
-from six.moves.urllib.parse import unquote
+from urllib.parse import unquote
 
 from .ProjectHolder import ProjectHolder
 from .utils import ApiCredentialsError, BadProjectError
--- a/lastversion/HelmChartRepoSession.py
+++ b/lastversion/HelmChartRepoSession.py
@@ -1,7 +1,7 @@
 import logging
 
 import yaml
-from six.moves.urllib.parse import urlparse
+from urllib.parse import urlparse
 
 from .ProjectHolder import ProjectHolder
 
--- a/lastversion/lastversion.py
+++ b/lastversion/lastversion.py
@@ -26,7 +26,7 @@
     from pathlib2 import Path  # python 2 backport
 import yaml
 from packaging.version import InvalidVersion
-from six.moves.urllib.parse import urlparse
+from urllib.parse import urlparse
 
 from .GitHubRepoSession import TOKEN_PRO_TIP
 from .HolderFactory import HolderFactory
--- a/setup.py
+++ b/setup.py
@@ -32,7 +32,6 @@
     'python-dateutil',
     'PyYAML',
     'tqdm',
-    'six',
     'beautifulsoup4',
     'distro',
     # pin due to https://github.com/ionrock/cachecontrol/issues/292
