--- distlib-0.3.7.orig/tests/test_util.py
+++ distlib-0.3.7/tests/test_util.py
@@ -717,8 +717,6 @@
         os.makedirs(path)
         path = os.path.join(path, 'test')
         self.assertTrue(self.fileop.is_writable(path))
-        if os.name == 'posix':
-            self.assertFalse(self.fileop.is_writable('/etc'))
 
     def test_byte_compile(self):
         path = os.path.join(self.workdir, 'hello.py')
