From 3a3ee1c45e54a808947cc26ca11f0a5299571053 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 25 Sep 2012 12:54:50 +0000
Subject: Remove mibstore.h

Signed-off-by: Adam Jackson <ajax@redhat.com>
---
diff --git a/src/rendition.c b/src/rendition.c
index 7c74f76..888bb4a 100644
--- a/src/rendition.c
+++ b/src/rendition.c
@@ -1173,7 +1173,6 @@ renditionScreenInit(SCREEN_INIT_ARGS_DECL)
     fbPictureInit (pScreen, 0, 0);
 
     xf86SetBlackWhitePixels(pScreen);
-    miInitializeBackingStore(pScreen);
    
     /*********************************************************/
     /* The actual setup of the driver-specific code          */
diff --git a/src/rendition.h b/src/rendition.h
index f7f13d4..297117e 100644
--- a/src/rendition.h
+++ b/src/rendition.h
@@ -28,9 +28,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 /* All drivers using the mi colormap manipulation need this */
 #include "micmap.h"
 
--
cgit v0.9.0.2-2-gbebe
From 30a4b73644ee22904824d4847e44ca69369fad02 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 30 Sep 2014 10:21:58 -0400
Subject: Fix build against xserver 1.17

Also port from pci_device_map_region to pci_device_map_range.  Somewhat
alarming that the previous code was using map_region but unmap_range...

Signed-off-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/rendition.c b/src/rendition.c
index 888bb4a..9677850 100644
--- a/src/rendition.c
+++ b/src/rendition.c
@@ -1371,6 +1371,18 @@ renditionMapMem(ScrnInfoPtr pScreenInfo)
        /* Override on users request */
     WriteCombine
 	= xf86ReturnOptValBool(pRendition->Options, OPTION_FBWC, WriteCombine);
+#ifdef XSERVER_LIBPCIACCESS
+    mapOption = PCI_DEV_MAP_FLAG_WRITABLE;
+    if (WriteCombine)
+	mapOption |= PCI_DEV_MAP_FLAG_WRITE_COMBINE;
+
+    err = pci_device_map_range(pRendition->PciInfo,
+			       pRendition->PciInfo->regions[0].base_addr,
+			       pRendition->PciInfo->regions[0].size,
+			       mapOption, (void *)&pRendition->board.vmem_base);
+
+    return (err == 0);
+#else
     if (WriteCombine) {
 	xf86DrvMsg(pScreenInfo->scrnIndex, X_CONFIG,
 		   ("Requesting Write-Combined memory access\n"));
@@ -1381,12 +1393,6 @@ renditionMapMem(ScrnInfoPtr pScreenInfo)
 	mapOption = VIDMEM_MMIO;
     }
 
-#ifdef XSERVER_LIBPCIACCESS
-    err = pci_device_map_region(pRendition->PciInfo, 0, TRUE);
-    pRendition->board.vmem_base = pRendition->PciInfo->regions[0].memory;
-
-    return (err == 0);
-#else
     pRendition->board.vmem_base=
         xf86MapPciMem(pScreenInfo->scrnIndex, mapOption,
 		      pRendition->pcitag,
@@ -1415,7 +1421,7 @@ renditionUnmapMem(ScrnInfoPtr pScreenInfo)
 #else
     pci_device_unmap_range(pRendition->PciInfo, 
 			   pRendition->board.vmem_base,
-			   pScreenInfo->videoRam * 1024);
+			   pRendition->PciInfo->regions[0].size);
 #endif
     return TRUE;
 #ifdef DEBUG0
-- 
cgit v0.10.2
From df5646745bf563bb36a0db1796c318fe64bf8efe Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Tue, 30 Sep 2014 10:24:18 -0400
Subject: Remove dependency on xf86PciInfo.h

Signed-off-by: Adam Jackson <ajax@redhat.com>

diff --git a/src/rendition.h b/src/rendition.h
index 297117e..4ef03a7 100644
--- a/src/rendition.h
+++ b/src/rendition.h
@@ -14,9 +14,6 @@
 /* Everything using inb/outb, etc needs "compiler.h" */
 #include "compiler.h"
 
-/* Drivers for PCI hardware need this */
-#include "xf86PciInfo.h"
-
 /* Drivers that need to access the PCI config space directly need this */
 #include "xf86Pci.h"
 
@@ -66,4 +63,8 @@
 
 #include "compat-api.h"
 /* end of __RENDITION_H__ */
+
+#define PCI_CHIP_V1000 0x0001
+#define PCI_CHIP_V2x00 0x2000
+
 #endif
-- 
cgit v0.10.2

