diff --git a/layers/external/vma/vk_mem_alloc.h b/layers/external/vma/vk_mem_alloc.h
index b47bd73..42d46d0 100644
--- a/layers/external/vma/vk_mem_alloc.h
+++ b/layers/external/vma/vk_mem_alloc.h
@@ -2682,6 +2682,8 @@ remove them if not needed.
     #include <cassert> // for assert
     #include <algorithm> // for min, max
     #include <mutex>
+    #include <cstdio>
+    #include <cstdint>
 #else
     #include VMA_CONFIGURATION_USER_INCLUDES_H
 #endif
