From 3235869b6842b7ea5f51e3bad5b32e9bd9cf98ed Mon Sep 17 00:00:00 2001
From: Joseph Turner <joseph@breatheoutbreathe.in>
Date: Tue, 7 Oct 2025 07:12:49 +0800
Subject: [PATCH] Fix hash table test to handle Emacs 30 simplified hash
 serialization

---
 test/persist-tests.el | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/test/persist-tests.el b/test/persist-tests.el
index 59ae319..ccaef28 100644
--- a/test/persist-tests.el
+++ b/test/persist-tests.el
@@ -67,7 +67,9 @@
          (default (copy-hash-table hash)))
     (persist-test-persist-save hash default
                                (puthash 'foo "bar" (symbol-value sym))
-                               "#s(hash-table size 65 test eql rehash-size 1.5 rehash-threshold 0.8125 data (foo \"bar\"))")))
+                               (if (<= 30 emacs-major-version)
+                                   "#s(hash-table data (foo \"bar\"))\n"
+                                 "#s(hash-table size 65 test eql rehash-size 1.5 rehash-threshold 0.8125 data (foo \"bar\"))"))))
 
 (ert-deftest test-persist-save-record ()
   "Test saving record."
