diff --git a/src/develop/openmp_maths.h b/src/develop/openmp_maths.h
index a3d466b..0428431 100644
--- a/src/develop/openmp_maths.h
+++ b/src/develop/openmp_maths.h
@@ -38,7 +38,7 @@ extern float fminf(const float x, const float y);
 DT_OMP_DECLARE_SIMD()
 extern float fabsf(const float x);
 
-DT_OMP_DECLARE_SIMD()
+//DT_OMP_DECLARE_SIMD()
 extern float powf(const float x, const float y);
 
 DT_OMP_DECLARE_SIMD()
@@ -47,16 +47,16 @@ extern float sqrtf(const float x);
 DT_OMP_DECLARE_SIMD()
 extern float cbrtf(const float x);
 
-DT_OMP_DECLARE_SIMD()
+//DT_OMP_DECLARE_SIMD()
 extern float log2f(const float x);
 
-DT_OMP_DECLARE_SIMD()
+//DT_OMP_DECLARE_SIMD()
 extern float exp2f(const float x);
 
 DT_OMP_DECLARE_SIMD()
 extern float log10f(const float x);
 
-DT_OMP_DECLARE_SIMD()
+//DT_OMP_DECLARE_SIMD()
 extern float expf(const float x);
 
 DT_OMP_DECLARE_SIMD()
