diff --git a/src/develop/openmp_maths.h b/src/develop/openmp_maths.h
index a3d466b8..562936fd 100644
--- a/src/develop/openmp_maths.h
+++ b/src/develop/openmp_maths.h
@@ -38,8 +38,8 @@ extern float fminf(const float x, const float y);
 DT_OMP_DECLARE_SIMD()
 extern float fabsf(const float x);
 
-DT_OMP_DECLARE_SIMD()
-extern float powf(const float x, const float y);
+//DT_OMP_DECLARE_SIMD()
+//extern float powf(const float x, const float y);
 
 DT_OMP_DECLARE_SIMD()
 extern float sqrtf(const float x);
