diff --git a/src/external/rawspeed/cmake/build-type.cmake b/src/external/rawspeed/cmake/build-type.cmake
index 4c15761..f752d1b 100644
--- a/src/external/rawspeed/cmake/build-type.cmake
+++ b/src/external/rawspeed/cmake/build-type.cmake
@@ -17,7 +17,7 @@ if(NOT CMAKE_BUILD_TYPE)
   SET(CMAKE_BUILD_TYPE ${default_build_type} CACHE STRING "" FORCE)
 endif(NOT CMAKE_BUILD_TYPE)
 
-set(RAWSPEED_STANDARD_BUILD_TYPES Debug RelWithDebInfo Release)
+set(RAWSPEED_STANDARD_BUILD_TYPES Debug RelWithDebInfo Release None)
 set(RAWSPEED_SPECIAL_BUILD_TYPES Coverage Sanitize TSan Fuzz)
 set(CMAKE_CONFIGURATION_TYPES ${RAWSPEED_STANDARD_BUILD_TYPES} ${RAWSPEED_SPECIAL_BUILD_TYPES} CACHE STRING "All the available build types" FORCE)
 
