diff --git a/src/external/rawspeed/cmake/build-type.cmake b/src/external/rawspeed/cmake/build-type.cmake
index c04263c..c72f7cb 100644
--- a/src/external/rawspeed/cmake/build-type.cmake
+++ b/src/external/rawspeed/cmake/build-type.cmake
@@ -19,7 +19,7 @@ endif(NOT CMAKE_BUILD_TYPE)
 
 # yes, these build types need to be specified here in upper-case.
 set(SPECIAL_BUILD_TYPES COVERAGE SANITIZE TSAN FUZZ)
-set(CMAKE_CONFIGURATION_TYPES DEBUG RELWITHDEBINFO RELEASE ${SPECIAL_BUILD_TYPES})
+set(CMAKE_CONFIGURATION_TYPES DEBUG RELWITHDEBINFO RELEASE NONE ${SPECIAL_BUILD_TYPES})
 set(CMAKE_CONFIGURATION_TYPES "${CMAKE_CONFIGURATION_TYPES}" CACHE STRING "All the available build types" FORCE)
 
 string(TOUPPER "${CMAKE_BUILD_TYPE}" CMAKE_BUILD_TYPE)
