diff --git a/driver/class/ColumnDefinition.h b/driver/class/ColumnDefinition.h
index 4919ee15..a05d4b5f 100644
--- a/driver/class/ColumnDefinition.h
+++ b/driver/class/ColumnDefinition.h
@@ -23,6 +23,7 @@
 
 #include <memory>
 #include <map>
+#include <cstdint>
 
 #include "mysql.h"
 
diff --git a/driver/interface/CmdInformation.h b/driver/interface/CmdInformation.h
index 228714b4..efe08e79 100644
--- a/driver/interface/CmdInformation.h
+++ b/driver/interface/CmdInformation.h
@@ -23,6 +23,7 @@
 
 #include <vector>
 #include <memory>
+#include <cstdint>
 
 namespace mariadb
 {
diff --git a/driver/interface/Row.h b/driver/interface/Row.h
index 3fa1cc54..872efe09 100644
--- a/driver/interface/Row.h
+++ b/driver/interface/Row.h
@@ -24,6 +24,7 @@
 #include <sstream>
 #include <vector>
 #include <memory>
+#include <cstdint>
 
 #include "mysql.h"
 
--- a/driver/interface/Exception.h
+++ b/driver/interface/Exception.h
@@ -22,6 +22,7 @@
 #define _EXCEPTION_H_
 
 #include <stdexcept>
+#include <cstdint> // for std::uint32_t and std::uint64_t
 #include "class/SQLString.h"
 
 namespace mariadb
--- a/driver/template/CArray.h
+++ b/driver/template/CArray.h
@@ -24,6 +24,7 @@
 #include <initializer_list>
 #include <vector>
 #include <stdexcept>
+#include <cstdint> // for std::uint32_t and std::uint64_t
 #include <cstring>
 #include <string>
 
