diff --git a/example/homer7_config/heplify-server.toml b/example/homer7_config/heplify-server.toml
index a7fa6ea..25a269b 100644
--- a/example/homer7_config/heplify-server.toml
+++ b/example/homer7_config/heplify-server.toml
@@ -15,8 +15,8 @@ PromTargetName        = ""
 DBShema               = "homer7"
 DBDriver              = "postgres"
 DBAddr                = "localhost:5432"
-DBUser                = "postgres"
-DBPass                = ""
+DBUser                = "homer_user"
+DBPass                = "homer_password"
 DBDataTable           = "homer_data"
 DBConfTable           = "homer_config"
 DBBulk                = 200
@@ -40,8 +40,8 @@ SIPHeader             = []
 LogDbg                = ""
 LogLvl                = "info"
 LogStd                = false
-LogSys                = false
-Config                = "./heplify-server.toml"
+LogSys                = true
+Config                = "/etc/homer/heplify-server.toml"
 ConfigHTTPAddr        = ""
 
 # Examples:
@@ -63,4 +63,4 @@ ConfigHTTPAddr        = ""
 # ConfigHTTPAddr  = "0.0.0.0:9876"
 # -------------------------------------
 # To hot reload PromTargetIP and PromTargetName run:
-# killall -HUP heplify-server
\ No newline at end of file
+# killall -HUP heplify-server
