diff --git a/pika-backup/src/meson.build b/pika-backup/src/meson.build
index 44bc0e9b..823e0cec 100644
--- a/pika-backup/src/meson.build
+++ b/pika-backup/src/meson.build
@@ -12,13 +12,13 @@ else
 endif
 
 cargo_options = [
+  '--frozen',
   '--target-dir', target_dir,
   '--profile', profile,
   '--manifest-path', manifest,
 ]
 
 cargo_env = {
-  'CARGO_HOME': meson.project_build_root() / 'cargo',
   'LOCALEDIR': get_option('prefix') / get_option('localedir'),
   'APPLICATION_ID_SUFFIX': get_option('app_id_suffix'),
   'GIT_DESCRIBE': run_command(
@@ -39,6 +39,7 @@ cargo_build = custom_target(
   env: cargo_env,
   command: [
     cargo_bin,
+    'auditable',
     'build',
     cargo_options,
   ],
@@ -99,4 +100,4 @@ test(
   ],
   is_parallel: false,
   timeout: 1800,
-)
\ No newline at end of file
+)
