--- a/strfry.conf
+++ b/strfry.conf
@@ -3,7 +3,7 @@
 ##
 
 # Directory that contains the strfry LMDB database (restart required)
-db = "./strfry-db/"
+db = "/var/lib/strfry/"
 
 dbParams {
     # Maximum number of threads/processes that can simultaneously have LMDB transactions open (restart required)
@@ -47,7 +47,7 @@
     port = 7777
 
     # Set OS-limit on maximum number of open files/sockets (if 0, don't attempt to set) (restart required)
-    nofiles = 1000000
+    nofiles = 0
 
     # HTTP header that contains the client's real IP, before reverse proxying (ie x-real-ip) (MUST be all lower-case)
     realIpHeader = ""
