--- a/src/rakudo-star-modules/OpenSSL/META6.json
+++ b/src/rakudo-star-modules/OpenSSL/META6.json
@@ -35,11 +35,7 @@
     "OpenSSL::X509": "lib/OpenSSL/X509.rakumod"
   },
   "raku": "6.*",
-  "resources": [
-    "ssleay32.dll",
-    "libeay32.dll",
-    "libraries.json"
-  ],
+  "resources": [ "libraries.json" ],
   "source-url": "https://github.com/raku-community-modules/OpenSSL.git",
   "tags": [
   ],
--- a/src/rakudo-star-modules/OpenSSL/lib/OpenSSL/NativeLib.rakumod
+++ b/src/rakudo-star-modules/OpenSSL/lib/OpenSSL/NativeLib.rakumod
@@ -1,23 +1,17 @@
 unit module OpenSSL::NativeLib;
 
-BEGIN my %libraries = Rakudo::Internals::JSON.from-json: %?RESOURCES<libraries.json>.slurp(:close);
+my $version = v3;
 
 sub ssl-lib is export {
-    state $lib = $*DISTRO.is-win
-        ?? dll-resource('ssleay32.dll')
-        !! $*VM.platform-library-name(%libraries<ssl>.IO).Str;
+    state $lib = $*VM.platform-library-name('ssl'.IO, :$version).Str;
 }
 
 sub gen-lib is export {
-    state $lib = $*DISTRO.is-win
-        ?? dll-resource('libeay32.dll')
-        !! $*VM.platform-library-name(%libraries<ssl>.IO).Str;
+    state $lib = $*VM.platform-library-name('ssl'.IO, :$version).Str;
 }
 
 sub crypto-lib is export {
-    state $lib = $*DISTRO.is-win
-        ?? dll-resource('libeay32.dll')
-        !! $*VM.platform-library-name(%libraries<crypto>.IO).Str;
+    state $lib = $*VM.platform-library-name('crypto'.IO, :$version).Str;
 }
 
 # Windows only
