--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,7 +190,9 @@ if(NOT_SUBPROJECT)
       install_script
        "set(install_pkgconfdir \"${PKGCONFIG_INSTALL_DIR}\")"
        "set(impl_pc_file \"${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2.pc.in\")"
+       "set(impl_pc_file_tmp \"${CMAKE_CURRENT_BINARY_DIR}/catch2.pc\")"
        "set(main_pc_file \"${CMAKE_CURRENT_SOURCE_DIR}/CMake/catch2-with-main.pc.in\")"
+       "set(main_pc_file_tmp \"${CMAKE_CURRENT_BINARY_DIR}/catch2-with-main.pc\")"
        "set(Catch2_VERSION ${Catch2_VERSION})"
        "set(include_dir \"${CMAKE_INSTALL_INCLUDEDIR}\")"
        "set(lib_dir \"${CMAKE_INSTALL_LIBDIR}\")"
@@ -198,19 +200,26 @@ if(NOT_SUBPROJECT)
          set(lib_name "$<TARGET_FILE_BASE_NAME:Catch2>")
          configure_file(
            "${impl_pc_file}"
-           "${CMAKE_INSTALL_PREFIX}/${install_pkgconfdir}/catch2.pc"
+           "${impl_pc_file_tmp}"
            @ONLY
          )
 
          set(lib_name "$<TARGET_FILE_BASE_NAME:Catch2WithMain>")
          configure_file(
            "${main_pc_file}"
-           "${CMAKE_INSTALL_PREFIX}/${install_pkgconfdir}/catch2-with-main.pc"
+           "${main_pc_file_tmp}"
            @ONLY
          )
        ]]
   )
   install(CODE "${install_script}")
+  install(
+    FILES
+      "${CMAKE_CURRENT_BINARY_DIR}/catch2.pc"
+      "${CMAKE_CURRENT_BINARY_DIR}/catch2-with-main.pc"
+    DESTINATION
+      ${PKGCONFIG_INSTALL_DIR}
+  )
 
   set(CPACK_PACKAGE_CONTACT "https://github.com/catchorg/Catch2/")
 
