--- src/crunch-3.4/Makefile.orig
+++ src/crunch-3.4/Makefile
@@ -58,16 +58,16 @@
 # Install generic target 
 geninstall: build
 	@echo "Creating directories..."
-	$(INSTALL) -d -m 755 $(INSTALL_OPTIONS) $(BINDIR)
-	$(INSTALL) -d -m 755 $(INSTALL_OPTIONS) $(MANDIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(BINDIR)
+	$(INSTALL) -d -m 755 $(DESTDIR)$(MANDIR)
 	@echo "Copying binary..."
-	$(INSTALL) crunch -m 755 $(INSTALL_OPTIONS) $(BINDIR)
+	$(INSTALL) crunch -m 755 $(DESTDIR)$(BINDIR)
 	@echo "Copying charset.lst..."
-	$(INSTALL) charset.lst -m 644 $(INSTALL_OPTIONS) $(BINDIR)
+	$(INSTALL) charset.lst -m 644 $(DESTDIR)$(BINDIR)
 	@echo "Copying GPL.TXT..."
-	$(INSTALL) GPL.TXT -m 644 $(INSTALL_OPTIONS) $(BINDIR)
+	$(INSTALL) GPL.TXT -m 644 $(DESTDIR)$(BINDIR)
 	@echo "Installing man page..."
-	$(INSTALL) crunch.1 -m 644 $(INSTALL_OPTIONS) $(MANDIR)
+	$(INSTALL) crunch.1 -m 644 $(DESTDIR)$(MANDIR)
 	@echo ""
 
 # Install BT specific target 
