--- a/tests/test_httpclient2.rb
+++ b/tests/test_httpclient2.rb
@@ -50,7 +50,7 @@
   def test_get
     content = nil
     EM.run {
-      setup_timeout(1)
+      setup_timeout(5)
       http = silent { EM::P::HttpClient2.connect :host => "google.com", :port => 80, :version => '1.0' }
       d = http.get "/"
       d.callback {
@@ -67,7 +67,7 @@
   def _test_get_multiple
     content = nil
     EM.run {
-      setup_timeout(1)
+      setup_timeout(5)
       http = silent { EM::P::HttpClient2.connect "google.com", :version => '1.0' }
       d = http.get "/"
       d.callback {
@@ -84,7 +84,7 @@
   def test_get_pipeline
     headers, headers2 = nil, nil
     EM.run {
-      setup_timeout(1)
+      setup_timeout(5)
       http = silent { EM::P::HttpClient2.connect "google.com", 80 }
       d = http.get("/")
       d.callback {
@@ -116,7 +116,7 @@
     omit_unless(EM.ssl?)
     d = nil
     EM.run {
-      setup_timeout(1)
+      setup_timeout(5)
       http = silent { EM::P::HttpClient2.connect :host => 'www.google.com', :port => 443, :ssl => true, :version => '1.0' }
       d = http.get "/"
       d.callback {EM.stop}
