diff -aur old/dict/stardict-plugins/stardict-wordnet-plugin/utils.h new/dict/stardict-plugins/stardict-wordnet-plugin/utils.h
--- old/dict/stardict-plugins/stardict-wordnet-plugin/utils.h	2011-07-18 06:58:58.519404744 -0700
+++ new/dict/stardict-plugins/stardict-wordnet-plugin/utils.h	2011-07-18 07:01:10.739734313 -0700
@@ -26,6 +26,7 @@
 #ifndef __TYPES_H__
 #define __TYPES_H__
 
+#include <cstddef>
 #include <math.h>
 
 typedef float single;
