diff -Naur old/dict/configure.ac new/dict/configure.ac
--- old/dict/configure.ac	2011-07-03 16:58:40.000000000 +1000
+++ new/dict/configure.ac	2012-07-12 22:31:52.875176842 +1000
@@ -266,6 +266,9 @@
 fi
 AC_SUBST(WIKI_PARSEDATA_PLUGIN_DIR)
 
+GMODULE2_LIBS=`pkg-config --libs gmodule-2.0`
+AC_SUBST([GMODULE2_LIBS])
+
 dnl ================================================================
 dnl main program checks.
 dnl ================================================================
diff -Naur old/dict/src/Makefile.am new/dict/src/Makefile.am
--- old/dict/src/Makefile.am	2011-07-03 16:58:40.000000000 +1000
+++ new/dict/src/Makefile.am	2012-07-12 22:29:53.714870692 +1000
@@ -33,7 +33,7 @@
 stardict_DEPENDENCIES = lib/libstardict.la $(LOCAL_SIGCPP_LIBFILE)
 stardict_LDFLAGS = 
 ## place libstardict.la before any system library, otherwise build with --as-needed linker option may fail
-stardict_LDADD = lib/libstardict.la $(STARDICT_LIBS) $(LOCAL_SIGCPP_LIBFILE)
+stardict_LDADD = lib/libstardict.la $(STARDICT_LIBS) $(LOCAL_SIGCPP_LIBFILE) $(GMODULE2_LIBS)
 
 if !GNOME_SUPPORT
 if MAEMO_SUPPORT
diff -Naur old/dict/src/tomboykeybinder.h new/dict/src/tomboykeybinder.h
--- old/dict/src/tomboykeybinder.h	2011-07-03 16:58:40.000000000 +1000
+++ new/dict/src/tomboykeybinder.h	2012-07-12 22:28:04.337920391 +1000
@@ -21,7 +21,6 @@
 #ifndef __TOMBOY_KEY_BINDER_H__
 #define __TOMBOY_KEY_BINDER_H__
 
-#include <glib.h>
 
 G_BEGIN_DECLS
 
