reported upstream:
https://github.com/p11-glue/p11-kit/issues/147

--- ./common/message.c.orig
+++ ./common/message.c
@@ -118,7 +118,7 @@
 	buffer[length] = 0;
 
 	snprintf (strerr, sizeof (strerr), "Unknown error %d", errnum);
-#ifdef HAVE_STRERROR_L
+#if defined(HAVE_STRERROR_L) && defined(__GLIBC__)
 	loc = uselocale ((locale_t) 0);
 	if (loc != NULL)
 		strncpy (strerr, strerror_l (errnum, loc), sizeof (strerr));
