--- a/sockd/sockd_io.c.old	2017-05-10 19:57:20.672000000 +0000
+++ b/sockd/sockd_io.c	2017-05-10 20:00:34.548000000 +0000
@@ -2912,7 +2912,7 @@ siginfo(sig, si, sc)
          int src_so_rcvbuf, dst_so_rcvbuf;
 #endif /* HAVE_RECVBUF_IOCTL */
 
-#if HAVE_SENDBUF_IOCTL
+#ifdef HAVE_SENDBUF_IOCTL
          int src_so_sndbuf, dst_so_sndbuf;
 #endif /* HAVE_SENDBUF_IOCTL */
 
@@ -2960,7 +2960,7 @@ siginfo(sig, si, sc)
          *src_bufferinfo = NUL;
          *dst_bufferinfo = NUL;
 
-#if HAVE_RECVBUF_IOCTL || HAVE_SENDBUF_IOCTL
+#if defined(HAVE_RECVBUF_IOCTL) || defined(HAVE_SENDBUF_IOCTL)
 
          havesocketinfo = 1;
 
@@ -2987,7 +2987,7 @@ siginfo(sig, si, sc)
          }
 #endif /* HAVE_RECVBUF_IOCTL */
 
-#if HAVE_SENDBUF_IOCTL
+#ifdef HAVE_SENDBUF_IOCTL
          if (havesocketinfo) {
             if (ioctl(src->s, SENDBUF_IOCTLVAL, &src_so_sndbuf) != 0) {
                swarn("%s: sendbuf size ioctl() on src-fd %d failed",
@@ -3005,7 +3005,7 @@ siginfo(sig, si, sc)
          }
 #endif /* HAVE_SENDBUF_IOCTL */
 
-#if HAVE_SENDBUF_IOCTL && HAVE_RECVBUF_IOCTL
+#ifdef HAVE_SENDBUF_IOCTL && HAVE_RECVBUF_IOCTL
          if (havesocketinfo) {
             snprintf(src_bufferinfo, sizeof(src_bufferinfo),
                      "%lu buffered (%lu + %lu + %lu)",
