diff --git a/pytest.ini b/pytest.ini
index c7a738e..2a8baff 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -21,16 +21,6 @@ addopts =
   # dump the test results in junit format:
   --junitxml=.test-results/pytest/results.xml
 
-  # Pre-load the `pytest-cov` plugin early:
-  -p pytest_cov
-
-  # `pytest-cov`:
-  --cov
-  --cov-report=xml:.test-results/pytest/cov.xml
-  --cov-context=test
-  --cov-config=.coveragerc
-  --no-cov-on-fail
-
   # Fail on config parsing warnings:
   # --strict-config
 
diff --git a/tox.ini b/tox.ini
index 3f66882..50a7dc9 100644
--- a/tox.ini
+++ b/tox.ini
@@ -13,11 +13,7 @@ allowlist_externals =
 isolated_build = true
 usedevelop = false
 deps =
-  # NOTE: `coverage` is a transitive dep but we need a minimum version boundary
-  # NOTE: to get a version supporting `source_pkgs`.
-  coverage >= 5.3
   pytest
-  pytest-cov
   pytest-forked
   pytest-xdist
 passenv =
@@ -27,7 +23,6 @@ commands =
   {envpython} -m \
     pytest \
     {tty:--color=yes} \
-    --no-cov \
     {posargs:}
 
 
diff --git a/pytest.ini b/pytest.ini
index 2a8baff..541272f 100644
--- a/pytest.ini
+++ b/pytest.ini
@@ -56,6 +56,7 @@ junit_suite_name = ansible_pylibssh_test_suite
 # A mapping of markers to their descriptions allowed in strict mode:
 markers =
   smoke: Quick post-build self-check smoke tests
+  forked: mark test to run in a forked subprocess
 
 minversion = 4.6.9
 
