--- a/Gem.gemspec
+++ b/Gem.gemspec
@@ -8,7 +8,7 @@
 Gem::Specification.new do |spec|
   spec.authors = ["Jordon Bedwell"]
   spec.version = Pathutil::VERSION
-  spec.files = %W(Rakefile Gemfile LICENSE) + Dir["{lib,bin}/**/*"]
+  spec.files = Dir["{lib,bin}/**/*"]
   spec.description = "Like Pathname but a little less insane."
   spec.summary = "Almost like Pathname but just a little less insane."
   spec.homepage = "http://github.com/envygeeks/pathutil"
