--- otp_src_18.0/lib/os_mon/src/disksup.erl
+++ otp_src_18.0-fixed/lib/os_mon/src/disksup.erl
@@ -87,7 +87,7 @@
 
 param_default(disk_space_check_interval) -> 30;
 param_default(disk_almost_full_threshold) -> 0.80;
-param_default(disksup_posix_only) -> false.
+param_default(disksup_posix_only) -> true.
 
 %%----------------------------------------------------------------------
 %% gen_server callbacks
--- otp_src_18.0/lib/os_mon/test/disksup_SUITE.erl
+++ otp_src_18.0-fixed/lib/os_mon/test/disksup_SUITE.erl
@@ -337,7 +337,7 @@
     [];
 restart(Config) when is_list(Config) ->
     ok = application:set_env(os_mon, start_disksup, true),
-    ok = application:set_env(os_mon, disksup_posix_only, false),
+    ok = application:set_env(os_mon, disksup_posix_only, true),
     {ok, _Pid} = supervisor:restart_child(os_mon_sup, disksup),
     ok.
 
