From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 27 Jun 2017 02:07:00 +0200
Subject: [PATCH] Show link to external docs if local is not installed

--- a/apps/settings/templates/help.php
+++ b/apps/settings/templates/help.php
@@ -42,6 +42,17 @@
 </div>

 <div id="app-content" class="help-includes">
+<?php if ($_['localDocs']) { ?>
 	<iframe src="<?php print_unescaped($_['url']); ?>" class="help-iframe">
 	</iframe>
+<?php } else { ?>
+	<div class="section">
+		<h2>Local documentation is not installed</h2>
+		<p>Please use
+			<a href="<?php print_unescaped($_['url']); ?>" target="_blank" rel="noreferrer">
+				<?php p($l->t('online documentation')); ?> ↗
+			</a>
+		</p>
+	</div>
+<?php } ?>
 </div>
