--- a/configure	2012-09-22 06:13:53.000000000 +0400
+++ b/configure	2017-02-24 00:28:09.645357065 +0300
@@ -6666,7 +6666,6 @@
   gl_LIBOBJS="$gl_LIBOBJS canonicalize-lgpl.$ac_objext"
 
 
-$as_echo "#define realpath rpl_realpath" >>confdefs.h
 
 
 
--- a/gnulib/m4/canonicalize-lgpl.m4	2010-03-26 22:38:17.000000000 +0300
+++ b/gnulib/m4/canonicalize-lgpl.m4	2017-02-24 00:28:38.515419315 +0300
@@ -12,8 +12,6 @@
   AC_CHECK_FUNCS_ONCE(canonicalize_file_name)
   if test $ac_cv_func_canonicalize_file_name = no; then
     AC_LIBOBJ(canonicalize-lgpl)
-    AC_DEFINE([realpath], [rpl_realpath],
-      [Define to a replacement function name for realpath().])
     gl_PREREQ_CANONICALIZE_LGPL
   fi
 ])
