diff --git a/src/installer/eng/Versions.props b/src/installer/eng/Versions.props
index b2330f533..8010b5f82 100644
--- a/src/installer/eng/Versions.props
+++ b/src/installer/eng/Versions.props
@@ -204,9 +204,9 @@
     <XamarinMacCatalystWorkloadManifestVersion>15.4.1006-rc.1</XamarinMacCatalystWorkloadManifestVersion>
     <XamarinMacOSWorkloadManifestVersion>12.3.1006-rc.1</XamarinMacOSWorkloadManifestVersion>
     <XamarinTvOSWorkloadManifestVersion>15.4.1006-rc.1</XamarinTvOSWorkloadManifestVersion>
-    <MonoWorkloadManifestVersion>$(MicrosoftNETCoreAppRefPackageVersion)</MonoWorkloadManifestVersion>
-    <MicrosoftNETWorkloadEmscriptennet6Manifest70100Version>7.0.0</MicrosoftNETWorkloadEmscriptennet6Manifest70100Version>
-    <MicrosoftNETWorkloadEmscriptennet7Manifest70100Version>7.0.0</MicrosoftNETWorkloadEmscriptennet7Manifest70100Version>
+    <MonoWorkloadManifestVersion>7.0.0-rtm.22504.4</MonoWorkloadManifestVersion>
+    <MicrosoftNETWorkloadEmscriptennet6Manifest70100Version>7.0.0-rtm.22504.3</MicrosoftNETWorkloadEmscriptennet6Manifest70100Version>
+    <MicrosoftNETWorkloadEmscriptennet7Manifest70100Version>7.0.0-rtm.22504.3</MicrosoftNETWorkloadEmscriptennet7Manifest70100Version>
     <EmscriptenWorkloadManifestVersion>$(MicrosoftNETWorkloadEmscriptennet7Manifest70100Version)</EmscriptenWorkloadManifestVersion>
   </PropertyGroup>
   <PropertyGroup
