--- OSPosix.cpp
+++ OSPosix.cpp
@@ -46,22 +46,22 @@
 			setvbuf(f, 0, _IOLBF, BUFSIZ);
 	}
 
-	void EnableFPE()
-	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
-		// clear any outstanding exceptions before enabling, otherwise they'll
-		// trip immediately
-		feclearexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
-		feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
+void EnableFPE()
+{
+#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !defined(__APPLE__)
+	// clear any outstanding exceptions before enabling, otherwise they'll
+	// trip immediately
+	feclearexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
+	feenableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
 #endif
-	}
+}
 
-	void DisableFPE()
-	{
-#if defined(_GNU_SOURCE) && !defined(__APPLE__)
-		fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
+void DisableFPE()
+{
+#if defined(_GNU_SOURCE) && defined(__GLIBC__) && !defined(__APPLE__)
+	fedisableexcept(FE_DIVBYZERO | FE_INVALID | FE_OVERFLOW);
 #endif
-	}
+}
 
 	Uint64 HFTimerFreq()
 	{
