diff --git a/docs/zzipdoc/htm2dbk.py b/docs/zzipdoc/htm2dbk.py
index 021a22a..d0f17cc 100644
--- a/docs/zzipdoc/htm2dbk.py
+++ b/docs/zzipdoc/htm2dbk.py
@@ -146,13 +146,8 @@ def htm2dbk_files(args):
             doc.filename = filename
             doc.add(f.read())
             f.close()
-<<<<<<< HEAD
-        except IOError as e:
-            print >> sys.stderr, "can not open "+filename
-=======
         except IOError:
             print("can not open "+filename, file=sys.stderr)
->>>>>>> develop
     return doc.value()
 
 def html2docbook(text):
