--- a/src/opt.c
+++ b/src/opt.c
@@ -222,6 +222,7 @@
                 /* Print the version command option, cleanup and exit. */
                 std_version();
                 exit_cleanup(true);
+		break;
 
             case 'h':
                 MSG_DEBUG("-h or --help command option");
@@ -229,6 +230,7 @@
                 /* Print the help screen command option, cleanup and exit. */
                 std_help();
                 exit_cleanup(true);
+		break;
 
             case '?':
             case ':':
@@ -285,7 +287,7 @@
     }
 
     /* Re-initialize the interface name command option to no interface? */
-    if (opt->interface != '\0')
+    if (*opt->interface != '\0')
         memset(opt->interface, '\0', (size_t)IF_NAMESIZE);
 
     /* Set the value of the interface name command option. */
--- a/src/sig.c
+++ b/src/sig.c
@@ -204,6 +204,7 @@
             case SIGINT:
                 /* Remove ^C symbol from the output terminal. */
                 printf("\r");
+		/* fallthrough */
 
             case SIGTERM:
                 MSG_DEBUG("Caught %s (%d)..", SIG_ITOA(sig), sig);
