--- mathomatic-15.7.1.orig/makefile
+++ mathomatic-15.7.1/makefile
@@ -33,7 +33,7 @@
 CC		?= gcc # C compiler to use; this statement doesn't work usually, instead using cc.
 M4		?= m4 # Change this to gm4 in Unix or a non-GNU system.
 INSTALL		?= install # Installer utility to use; change to ginstall under Unix.
-INSTALL_PROGRAM	?= $(INSTALL) -Cv # Command to install executable program files; compares to see if any differences, first.
+INSTALL_PROGRAM	?= $(INSTALL) -c # Command to install executable program files; compares to see if any differences, first.
 INSTALL_DATA	?= $(INSTALL) -m 0644 # command to install data files
 
 CC_OPTIMIZE	?= -O3 # Default C compiler optimization flags that are safe.
