diff --git a/Makefile b/Makefile
index d481e68..f7452b5 100644
--- a/Makefile
+++ b/Makefile
@@ -22,7 +22,7 @@ include Makefile.cfg
 
 export CC = gcc -pthread
 export CFLAGS += -I. -Wall -Os -ggdb -D_GNU_SOURCE -fPIC
-export LDLIBS += -lncursesw
+export LDLIBS += -lncursesw -liconv
 
 SONAME  := libstfl.so.0
 VERSION := 0.22
@@ -39,7 +39,7 @@ libstfl.a: public.o base.o parser.o dump.o style.o binding.o iconv.o \
 
 libstfl.so.$(VERSION): public.o base.o parser.o dump.o style.o binding.o iconv.o \
                        $(patsubst %.c,%.o,$(wildcard widgets/*.c))
-	$(CC) -shared -Wl,-soname,$(SONAME) -o $@ $(LDLIBS) $^
+	$(CC) -shared $(LDLIBS) -Wl,-soname,$(SONAME) -o $@ $(LDLIBS) $^
 
 clean:
 	rm -f libstfl.a example core core.* *.o Makefile.deps
@@ -65,6 +65,7 @@ install: all stfl.pc
 	install -m 644 stfl.pc $(DESTDIR)$(prefix)/$(libdir)/pkgconfig/
 	install -m 644 libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)
 	ln -fs libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)/libstfl.so
+	ln -fs libstfl.so.$(VERSION) $(DESTDIR)$(prefix)/$(libdir)/$(SONAME)
 
 stfl.pc: stfl.pc.in
 	sed 's,@VERSION@,$(VERSION),g' < $< | sed 's,@PREFIX@,$(prefix),g' > $@
diff --git a/Makefile.cfg b/Makefile.cfg
index af3e102..346682e 100644
--- a/Makefile.cfg
+++ b/Makefile.cfg
@@ -19,7 +19,7 @@
 #
 
 export libdir ?= lib
-export prefix ?= /usr/local
+export prefix ?= /usr
 export DESTDIR ?= /
 
 ifneq ($(shell spl-config --cflags 2>/dev/null),)
diff --git a/python/Makefile.snippet b/python/Makefile.snippet
index 8fd4052..af95b01 100644
--- a/python/Makefile.snippet
+++ b/python/Makefile.snippet
@@ -33,8 +33,7 @@ python/_stfl.so python/stfl.py python/stfl.pyc: libstfl.a stfl.h python/stfl.i s
 
 install_python: python/_stfl.so python/stfl.py python/stfl.pyc
 	mkdir -p $(DESTDIR)$(PYTHON_SITEARCH)/
-	mkdir -p $(DESTDIR)$(PYTHON_SITEARCH)/lib-dynload/
-	cp python/_stfl.so $(DESTDIR)$(PYTHON_SITEARCH)/lib-dynload/
+	cp python/_stfl.so $(DESTDIR)$(PYTHON_SITEARCH)//
 	cp python/stfl.pyc $(DESTDIR)$(PYTHON_SITEARCH)/
 	cp python/stfl.py $(DESTDIR)$(PYTHON_SITEARCH)/
 
