diff -rupN a/syncer.py b/syncer.py
--- a/syncer.py	2026-03-24 17:57:18.830000000 +0000
+++ b/syncer.py	2026-03-25 01:56:05.120000000 +0000
@@ -10,26 +10,34 @@ from typing import Any, Callable, Genera
 
 @singledispatch
 def sync(co: Any):
-    raise TypeError('Called with unsupported argument: {}'.format(co))
+    # Allow passing in a Future object
+    if asyncio.isfuture(co):
+        if co.done():
+            return co.result()
+        else:
+            raise asyncio.InvalidStateError(
+                'Called without setting result: {}'.format(co))
+
+    if not inspect.iscoroutine(co) and not asyncio.isfuture(co):
+        raise TypeError('Called with unsupported argument: {}'.format(co))
 
 
-@sync.register(asyncio.Future)
 @sync.register(types.GeneratorType)
 def sync_co(co: Generator[Any, None, Any]) -> Any:
     if not inspect.isawaitable(co):
         raise TypeError('Called with unsupported argument: {}'.format(co))
-    return asyncio.get_event_loop().run_until_complete(co)
+    return asyncio.new_event_loop().run_until_complete(co)
 
 
 @sync.register(types.FunctionType)
 @sync.register(types.MethodType)
 def sync_fu(f: Callable[..., Any]) -> Callable[..., Any]:
-    if not asyncio.iscoroutinefunction(f):
+    if not inspect.iscoroutinefunction(f):
         raise TypeError('Called with unsupported argument: {}'.format(f))
 
     @wraps(f)
     def run(*args, **kwargs):
-        return asyncio.get_event_loop().run_until_complete(f(*args, **kwargs))
+        return asyncio.new_event_loop().run_until_complete(f(*args, **kwargs))
     return run
 
 
diff -rupN a/tests/test_syncer.py b/tests/test_syncer.py
--- a/tests/test_syncer.py	2023-05-08 07:50:01.000000000 +0000
+++ b/tests/test_syncer.py	2026-03-25 01:45:47.230000000 +0000
@@ -94,7 +94,14 @@ class TestSyncerPY35(unittest.TestCase):
         self.assertEqual(a.a(1, 2, 3), 3)
 
     def test_future(self):
-        f = asyncio.Future()
+        async def a():
+            f = asyncio.new_event_loop().create_future()
+            f.set_result(1)
+            return await f
+        self.assertEqual(sync(a()), 1)
+
+        # As of python 3.14, loop must be specified
+        f = asyncio.Future(loop=asyncio.new_event_loop())
         f.set_result(1)
         self.assertEqual(sync(f), 1)
 
