--- libdv-1.0.0/configure.ac.orig	2007-01-19 15:48:43.000000000 -0500
+++ libdv-1.0.0/configure.ac	2007-01-19 15:50:08.000000000 -0500
@@ -162,7 +162,7 @@
 dnl Checks for libraries.
 have_gtk="false"
 if $use_gtk; then
-	REQUIRES='glib >= 1.2.4 gtk+ >= 1.2.4'
+	REQUIRES='glib-2.0 >= 2.1.0 gtk+-x11-2.0 >= 2.1.0'
 	PKG_CHECK_MODULES(GTK,$REQUIRES,have_gtk="true",have_gtk="false")
 	AC_DEFINE(HAVE_GTK)
 fi
