Increase timeout to avoid test failure

--- a/tests/test-recursive-monitor.c
+++ b/tests/test-recursive-monitor.c
@@ -204,7 +204,7 @@ test_basic (void)
   g_signal_connect (state.monitor, "changed", G_CALLBACK (monitor_changed_cb), &state);
 
   /* Add a timeout to avoid infinite running */
-  g_timeout_add_seconds (3, failed_timeout, &state);
+  g_timeout_add_seconds (10, failed_timeout, &state);
 
   dzl_recursive_file_monitor_start_async (state.monitor, NULL, started_cb, &state);
 
