diff --git a/tinydns-data.c b/tinydns-data.c
index 104ad2f..df716a8 100644
--- a/tinydns-data.c
+++ b/tinydns-data.c
@@ -254,19 +254,19 @@ int load(const char *fname)
 	if (!dns_domain_fromdot(&d1,f[0].s,f[0].len)) nomem();
 
 	if (!stralloc_0(&f[3])) nomem();
-	if (!scan_ulong(f[3].s,&u)) uint32_unpack_big(defaultsoa,&u);
+	if (!scan_ulong(f[3].s,&u)) uint32_unpack_big(defaultsoa,(uint32 *)&u);
 	uint32_pack_big(soa,u);
 	if (!stralloc_0(&f[4])) nomem();
-	if (!scan_ulong(f[4].s,&u)) uint32_unpack_big(defaultsoa + 4,&u);
+	if (!scan_ulong(f[4].s,&u)) uint32_unpack_big(defaultsoa + 4,(uint32 *)&u);
 	uint32_pack_big(soa + 4,u);
 	if (!stralloc_0(&f[5])) nomem();
-	if (!scan_ulong(f[5].s,&u)) uint32_unpack_big(defaultsoa + 8,&u);
+	if (!scan_ulong(f[5].s,&u)) uint32_unpack_big(defaultsoa + 8,(uint32 *)&u);
 	uint32_pack_big(soa + 8,u);
 	if (!stralloc_0(&f[6])) nomem();
-	if (!scan_ulong(f[6].s,&u)) uint32_unpack_big(defaultsoa + 12,&u);
+	if (!scan_ulong(f[6].s,&u)) uint32_unpack_big(defaultsoa + 12,(uint32 *)&u);
 	uint32_pack_big(soa + 12,u);
 	if (!stralloc_0(&f[7])) nomem();
-	if (!scan_ulong(f[7].s,&u)) uint32_unpack_big(defaultsoa + 16,&u);
+	if (!scan_ulong(f[7].s,&u)) uint32_unpack_big(defaultsoa + 16,(uint32 *)&u);
 	uint32_pack_big(soa + 16,u);
 
 	if (!stralloc_0(&f[8])) nomem();

diff --git a/chkshsgr.c b/chkshsgr.c
index 8cfd11e..03ecb61 100644
--- a/chkshsgr.c
+++ b/chkshsgr.c
@@ -4,7 +4,7 @@
 
 int main()
 {
-  short x[4];
+  int x[4]; /* #define __GID_T_TYPE __U32_TYPE */
 
   x[0] = x[1] = 0;
   if (getgroups(1,x) == 0) if (setgroups(1,x) == -1) _exit(1);

diff --git a/hier.c b/hier.c
index 4aef75b..3bc87cf 100644
--- a/hier.c
+++ b/hier.c
@@ -1,5 +1,9 @@
 #include "auto_home.h"
 
+extern void c(char *home,char *subdir,char *file,int uid,int gid,int mode);
+extern void d(char *home,char *subdir,int uid,int gid,int mode);
+extern void h(char *home,int uid,int gid,int mode);
+
 void hier()
 {
   c("/","etc","dnsroots.global",-1,-1,0644);

diff --git a/utime.c b/utime.c
index be6c936..5d099ec 100644
--- a/utime.c
+++ b/utime.c
@@ -20,6 +20,6 @@ int main(int argc,char **argv)
   scan_ulong(ustr,&u);

   ut[0] = ut[1] = u;
-  if (utime(fn,ut) == -1) _exit(111);
+  if (utime(fn,(struct utimbuf *)ut) == -1) _exit(111);
   _exit(0);
 }

