diff --git a/build/config/Linux b/build/config/Linux
index 959db5e..74497de 100644
--- a/build/config/Linux
+++ b/build/config/Linux
@@ -41,7 +41,7 @@ SHAREDLIBLINKEXT = .so
 CFLAGS          = $(SANITIZEFLAGS) -std=c11
 CFLAGS32        =
 CFLAGS64        =
-CXXFLAGS        = $(SANITIZEFLAGS) -std=c++14 -Wall -Wno-sign-compare
+CXXFLAGS        = $(SANITIZEFLAGS) -std=c++14 -Wall -Wno-sign-compare -fomit-frame-pointer
 CXXFLAGS32      =
 CXXFLAGS64      =
 LINKFLAGS       = $(SANITIZEFLAGS)
@@ -56,9 +56,9 @@ SHAREDOPT_LINK  = -Wl,-rpath,$(LIBPATH)
 DEBUGOPT_CC     = -g -D_DEBUG
 DEBUGOPT_CXX    = -g -D_DEBUG
 DEBUGOPT_LINK   = -g
-RELEASEOPT_CC   = -O2 -DNDEBUG
-RELEASEOPT_CXX  = -O2 -DNDEBUG
-RELEASEOPT_LINK = -O2
+RELEASEOPT_CC   = -Os -DNDEBUG
+RELEASEOPT_CXX  = -Os -DNDEBUG
+RELEASEOPT_LINK = -Os
 
 #
 # System Specific Flags
