--- ./src/makefile.orig
+++ ./src/makefile
@@ -12,7 +12,7 @@
 
 # See src/gui-qt/highlight.pro for the Qt GUI compilation options
 
-CXX=c++
+CXX?=c++
 
 # Added -qt4 because of Arch Linux Qt5 migration
 QMAKE=qmake-qt4
@@ -37,10 +37,11 @@
 # If Lua 5.2 is not default on your system yet you have to
 # use lua5.1 in the pkg-config calls
 
-LUA_CFLAGS=$(shell pkg-config --cflags lua)
+LUAPC?=lua
+LUA_CFLAGS=$(shell pkg-config --cflags $(LUAPC))
 
 # default lua lib
-LUA_LIBS=$(shell pkg-config --libs lua)
+LUA_LIBS=$(shell pkg-config --libs $(LUAPC))
 
 # luajit lib
 # LUA_LIBS=$(shell pkg-config --libs luajit)
