From 57f92cc97ecff604cb83ccb3d0c52575be957718 Mon Sep 17 00:00:00 2001
From: "W.C.A. Wijngaards" <wouter@nlnetlabs.nl>
Date: Thu, 28 May 2026 09:34:04 +0200
Subject: [PATCH] - Fix #1457: race condition causes segfault when starting  
 threads.

---
 daemon/daemon.c | 5 +++--
 doc/Changelog   | 4 ++++
 2 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/daemon/daemon.c b/daemon/daemon.c
index 9a692c1cbd..1f87d853e9 100644
--- a/daemon/daemon.c
+++ b/daemon/daemon.c
@@ -919,13 +919,14 @@ thread_start(void* arg)
 {
 	struct worker* worker = (struct worker*)arg;
 	int port_num = 0;
-	log_assert(worker->thr_id);
 	set_log_thread_id(worker, worker->daemon->cfg);
 	{
 		char name[16]; /* seems to be the safest size between
 				  different OSes */
 		snprintf(name, sizeof(name), "unbound/%u", worker->thread_num);
-		ub_thread_setname(worker->thr_id, name);
+		/* worker->thr_id can be written to after the thread was made
+		 * by the creating thread, so this uses pthread_self. */
+		ub_thread_setname(ub_thread_self(), name);
 	}
 	ub_thread_blocksigs();
 #ifdef THREADS_DISABLED
From 6aa5cfc903c40666eed5d2af04068fac0b12989b Mon Sep 17 00:00:00 2001
From: "W.C.A. Wijngaards" <wouter@nlnetlabs.nl>
Date: Fri, 19 Jun 2026 08:37:23 +0200
Subject: [PATCH] - Fix for #1457: fix thread setname for thread start of  
 dnstap, and fast_reload.

---
 daemon/remote.c   | 2 +-
 dnstap/dtstream.c | 2 +-
 doc/Changelog     | 4 ++++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/daemon/remote.c b/daemon/remote.c
index df12201085..720f0daab4 100644
--- a/daemon/remote.c
+++ b/daemon/remote.c
@@ -6909,7 +6909,7 @@ static void* fast_reload_thread_main(void* arg)
 #endif
 		log_thread_set(&fast_reload_thread->threadnum);
 
-	ub_thread_setname(fast_reload_thread->tid, name);
+	ub_thread_setname(ub_thread_self(), name);
 	(void)name; /* When setname is not defined, ignore the name variable. */
 
 	verbose(VERB_ALGO, "start fast reload thread");
diff --git a/dnstap/dtstream.c b/dnstap/dtstream.c
index 09bdfd6627..eee768ee22 100644
--- a/dnstap/dtstream.c
+++ b/dnstap/dtstream.c
@@ -2144,7 +2144,7 @@ static void* dnstap_io(void* arg)
 #endif
 		log_thread_set(&dtio->threadnum);
 
-	ub_thread_setname(dtio->tid, name);
+	ub_thread_setname(ub_thread_self(), name);
 
 	/* setup */
 	verbose(VERB_ALGO, "start dnstap io thread");
