--- ./doc/example.conf.in.orig
+++ ./doc/example.conf.in
@@ -226,6 +226,7 @@
 	# file to read root hints from.
 	# get one from ftp://FTP.INTERNIC.NET/domain/named.cache
 	# root-hints: ""
+	root-hints: /etc/unbound/root.hints
 
 	# enable to not answer id.server and hostname.bind queries.
 	# hide-identity: no
@@ -338,7 +339,7 @@
 	# with several entries, one file per entry.
 	# Zone file format, with DS and DNSKEY entries.
 	# Note this gets out of date, use auto-trust-anchor-file please.
-	# trust-anchor-file: ""
+	# trust-anchor-file: "@UNBOUND_ROOTKEY_FILE@"
 	
 	# Trusted key for validation. DS or DNSKEY. specify the RR on a
 	# single line, surrounded by "". TTL is ignored. class is IN default.
