From 964d4a05896bcb6dfbeec34dccb77e4f381363f1 Mon Sep 17 00:00:00 2001
From: Landry Breuil <landry@xfce.org>
Date: Wed, 22 Aug 2012 07:34:45 +0000
Subject: Don't leak a graduent pattern at each draw (bug #8521)

---
diff --git a/gtk-2.0/gradient_draw.c b/gtk-2.0/gradient_draw.c
index 02deb88..37af559 100644
--- a/gtk-2.0/gradient_draw.c
+++ b/gtk-2.0/gradient_draw.c
@@ -278,4 +278,6 @@ void gradient_draw_shaded(cairo_t *cr, gint x, gint y, gint width, gint height,
     cairo_fill (cr);
 
     cairo_restore(cr);
+
+    cairo_pattern_destroy(gradient);
 }
--
cgit v0.9.0.3
