diff -up tcp_wrappers_7.6/Makefile.patch2 tcp_wrappers_7.6/Makefile
--- tcp_wrappers_7.6/Makefile.patch2	2008-08-29 09:45:12.000000000 +0200
+++ tcp_wrappers_7.6/Makefile	2008-08-29 09:45:12.000000000 +0200
@@ -143,7 +143,7 @@ freebsd:
 
 linux:
 	@make REAL_DAEMON_DIR=$(REAL_DAEMON_DIR) STYLE=$(STYLE) \
-	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ=setenv.o \
+	LIBS= RANLIB=ranlib ARFLAGS=rv AUX_OBJ= \
 	NETGROUP= TLI= EXTRA_CFLAGS="$(RPM_OPT_FLAGS) -DUSE_STRERROR -DSYS_ERRLIST_DEFINED -DBROKEN_SO_LINGER" all
 
 # This is good for many SYSV+BSD hybrids with NIS, probably also for HP-UX 7.x.
