diff -up tcp_wrappers_7.6/tcpdmatch.8.inetdconf tcp_wrappers_7.6/tcpdmatch.8
--- tcp_wrappers_7.6/tcpdmatch.8.inetdconf	2010-06-16 14:39:32.000000000 +0200
+++ tcp_wrappers_7.6/tcpdmatch.8	2010-06-16 14:42:25.000000000 +0200
@@ -2,9 +2,9 @@
 .SH NAME
 tcpdmatch \- tcp wrapper oracle
 .SH SYNOPSYS
-tcpdmatch [-d] [-i inet_conf] daemon client
+tcpdmatch [-d] daemon client
 .sp
-tcpdmatch [-d] [-i inet_conf] daemon[@server] [user@]client
+tcpdmatch [-d] daemon[@server] [user@]client
 .SH DESCRIPTION
 .PP
 \fItcpdmatch\fR predicts how the tcp wrapper would handle a specific
@@ -48,10 +48,6 @@ The default user name is `unknown'.
 .IP -d
 Examine \fIhosts.allow\fR and \fIhosts.deny\fR files in the current
 directory instead of the default ones.
-.IP "-i inet_conf"
-Specify this option when \fItcpdmatch\fR is unable to find your
-\fIinetd.conf\fR or \fItlid.conf\fR network configuration file, or when
-you suspect that the program uses the wrong one.
 .SH EXAMPLES
 To predict how \fItcpd\fR would handle a telnet request from the local
 system:
@@ -82,11 +78,8 @@ The default locations of the \fItcpd\fR 
 .SH SEE ALSO
 .na
 .nf
-tcpdchk(8), tcpd configuration checker
 hosts_access(5), format of the tcpd access control tables.
 hosts_options(5), format of the language extensions.
-inetd.conf(5), format of the inetd control file.
-tlid.conf(5), format of the tlid control file.
 .SH AUTHORS
 .na
 .nf
diff -up tcp_wrappers_7.6/tcpdmatch.c.inetdconf tcp_wrappers_7.6/tcpdmatch.c
--- tcp_wrappers_7.6/tcpdmatch.c.inetdconf	2010-06-16 14:37:09.000000000 +0200
+++ tcp_wrappers_7.6/tcpdmatch.c	2010-06-16 14:39:18.000000000 +0200
@@ -140,25 +140,6 @@ char  **argv;
     }
 
     /*
-     * Analyze the inetd (or tlid) configuration file, so that we can warn
-     * the user about services that may not be wrapped, services that are not
-     * configured, or services that are wrapped in an incorrect manner. Allow
-     * for services that are not run from inetd, or that have tcpd access
-     * control built into them.
-     */
-    inetcf = inet_cfg(inetcf);
-    inet_set("portmap", WR_NOT);
-    inet_set("rpcbind", WR_NOT);
-    switch (inet_get(daemon)) {
-    case WR_UNKNOWN:
-	tcpd_warn("%s: no such process name in %s", daemon, inetcf);
-	break;
-    case WR_NOT:
-	tcpd_warn("%s: service possibly not wrapped", daemon);
-	break;
-    }
-
-    /*
      * Check accessibility of access control files.
      */
     (void) check_path(hosts_allow_table, &st);
@@ -319,10 +300,9 @@ char  **argv;
 static void usage(myname)
 char   *myname;
 {
-    fprintf(stderr, "usage: %s [-d] [-i inet_conf] daemon[@host] [user@]host\n",
+    fprintf(stderr, "usage: %s [-d] daemon[@host] [user@]host\n",
 	    myname);
     fprintf(stderr, "	-d: use allow/deny files in current directory\n");
-    fprintf(stderr, "	-i: location of inetd.conf file\n");
     exit(1);
 }
 
