diff -up tcp_wrappers_7.6-ipv6.4/hosts_access.c.ucharpatch tcp_wrappers_7.6-ipv6.4/hosts_access.c
--- tcp_wrappers_7.6-ipv6.4/hosts_access.c.ucharpatch	2013-01-28 10:19:20.424857730 +0100
+++ tcp_wrappers_7.6-ipv6.4/hosts_access.c	2013-01-28 10:19:44.719991745 +0100
@@ -514,7 +514,7 @@ static void ipv6_mask(in6p, maskbits)
 struct in6_addr *in6p;
 int maskbits;
 {
-    uchar_t *p = (uchar_t*) in6p;
+    unsigned char *p = (unsigned char*) in6p;
 
     if (maskbits < 0 || maskbits >= IPV6_ABITS)
 	return;
@@ -525,7 +525,7 @@ int maskbits;
     if (maskbits != 0)
 	*p++ &= 0xff << (8 - maskbits);
 
-    while (p < (((uchar_t*) in6p)) + sizeof(*in6p))
+    while (p < (((unsigned char*) in6p)) + sizeof(*in6p))
 	*p++ = 0;
 }
 #endif
