--- tcp_wrappers_7.6-ipv6.4/tcpdchk.c
+++ tcp_wrappers_7.6-ipv6.4/tcpdchk.c.new
@@ -410,15 +410,6 @@
 	/* SCO has no *netgrent() support */
 #else
 #ifdef NETGROUP
-	char   *machinep;
-	char   *userp;
-	char   *domainp;
-
-	setnetgrent(pat + 1);
-	if (getnetgrent(&machinep, &userp, &domainp) == 0)
-	    tcpd_warn("%s: unknown or empty netgroup", pat + 1);
-	endnetgrent();
-#else
 	tcpd_warn("netgroup support disabled");
 #endif
 #endif
