diff -uNr a/plugins/pty/pty.c b/plugins/pty/pty.c
--- a/plugins/pty/pty.c	2024-10-26 11:59:26.000000000 +0200
+++ b/plugins/pty/pty.c	2025-01-24 20:04:55.952597269 +0100
@@ -259,7 +259,7 @@
 
 }
 
-static void uwsgi_pty_winch() {
+static void uwsgi_pty_winch(int sig) {
 	// 2 uwsgi packets
 	char uwsgi_pkt[8];
 #ifdef TIOCGWINSZ
@@ -309,7 +309,7 @@
 	if (upty.uremote) {
 		signal(SIGWINCH, uwsgi_pty_winch);
 		// send current terminal size
-		uwsgi_pty_winch();
+		uwsgi_pty_winch(SIGWINCH);
 	}
 
 	upty.queue = event_queue_init();
diff -uNr a/plugins/tuntap/tuntap.c b/plugins/tuntap/tuntap.c
--- a/plugins/tuntap/tuntap.c	2024-10-26 11:59:26.000000000 +0200
+++ b/plugins/tuntap/tuntap.c	2025-01-24 20:33:49.513449322 +0100
@@ -409,7 +409,7 @@
 	}
 }
 
-static void uwsgi_tuntap_router() {
+static void uwsgi_tuntap_router(int (*start)(void *), char **argv) {
 
 	if (!utt.routers) return;
 
