--- ./CMakeLists.txt.orig
+++ ./CMakeLists.txt
@@ -63,21 +63,6 @@
 endif()
 
 
-if (NOT LIB_SUFFIX)
-	message(STATUS "")
-	message(STATUS "LIB_SUFFIX variable is not defined. It will be autodetected now.")
-	message(STATUS "You can set it manually with -DLIB_SUFFIX=<value> (e.g. 64).")
-	if (CMAKE_SIZEOF_VOID_P EQUAL 8)
-		message(STATUS "\nSetting LIB_SUFFIX=64\n")
-		set (LIB_SUFFIX "64")
-	else()
-		message(STATUS "\nSetting LIB_SUFFIX=\n")
-		set (LIB_SUFFIX "")
-	endif()
-endif()
-
-
-
 add_executable(tcplay-bin main.c ${SRCS_COMMON} ${SRCS_LINUX} ${SRCS_PBKDF})
 set_target_properties(tcplay-bin PROPERTIES OUTPUT_NAME tcplay)
 set_target_properties(tcplay-bin PROPERTIES COMPILE_FLAGS "${CFLAGS_COMMON} ${DEVMAPPER_CFLAGS} ${UUID_CFLAGS} ${GCRYPT_CFLAGS}")
