diff --git a/es.c b/es.c
index a89786c..fc3e0cd 100644
--- a/es.c
+++ b/es.c
@@ -567,7 +567,7 @@ void parse(CLIENT *client, char *buf)
   }
   else if ((!strcmp (buf, "r") || !strcmp(buf, "tts_set_speech_rate")) && token[0])
   {
-    if (text_buffered) es_synthesize(client);
+    if (text_buffered) es_synthesize();
     /* In the libspeech api, a rate of 1000 is defined as "normal."  Thus, fs.c
 	defines a rate of 1000 as equivalent to FLite's normal rate (about 175
 	wpm).  The following conversion is accurate for it but may not be
@@ -577,7 +577,7 @@ void parse(CLIENT *client, char *buf)
   }
   else if ((!strcmp (buf, "v") || !strcmp(buf, "tts_set_speech_volume")) && token[0])
   {
-    if (text_buffered) es_synthesize(client);
+    if (text_buffered) es_synthesize();
     client->param[S_VOLUME] = atoi(token[0]);
   }
   else if (!strcmp (buf, "reset"))
