From 3a387670ed13041db069068292b1f41229e79583 Mon Sep 17 00:00:00 2001
From: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>
Date: Tue, 16 Apr 2024 16:17:05 +0200
Subject: [PATCH] CMakeLists.txt: pkgconf module name changed: history-service
 -> lomiri-history-service.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c2bd3359..a0d94591 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +70,7 @@ pkg_check_modules(TPFS REQUIRED telepathy-farstream)
 pkg_check_modules(GST REQUIRED gstreamer-1.0)
 pkg_check_modules(FS REQUIRED farstream-0.2)
 pkg_check_modules(PULSEAUDIO libpulse)
-pkg_check_modules(HISTORY REQUIRED history-service)
+pkg_check_modules(HISTORY REQUIRED lomiri-history-service)
 
 if (WANT_UI_SERVICES)
     pkg_check_modules(MESSAGING_MENU REQUIRED messaging-menu)
-- 
2.47.0

