diff --git a/src/utils/cst_file_stdio.c b/src/utils/cst_file_stdio.c
index eb4f968..5c7c977 100644
--- a/src/utils/cst_file_stdio.c
+++ b/src/utils/cst_file_stdio.c
@@ -37,6 +37,7 @@
 /*                                                                       */
 /*************************************************************************/
 
+#include <limits.h>
 #include <stdio.h>
 #include <string.h>
 #include <stdarg.h>
@@ -151,7 +152,7 @@ int cst_sprintf(char *s, const char *fmt, ...)
     int rv;
 
     va_start(args, fmt);
-    rv = vsprintf(s, fmt, args);
+    rv = vsnprintf(s, INT_MAX, fmt, args);
     va_end(args);
 
     return rv;
