diff --git a/engine.cpp b/engine.cpp
index f6c5792..843d53a 100644
--- a/engine.cpp
+++ b/engine.cpp
@@ -234,7 +234,7 @@ move_s engine::gen_killer1(std::vector<move_s> &moves, move_s &tt_move,
     if (k1 == tt_move || !is_pseudo_legal(k1))
         return not_a_move;
     k1.priority = 253;
-    moves.insert(moves.begin() + move_num, k1);
+    moves.insert(moves.begin() + static_cast<std::vector<board_state::move_s>::iterator::difference_type>(move_num), k1);
     return k1;
 }
 
@@ -248,7 +248,7 @@ move_s engine::gen_killer2(std::vector<move_s> &moves, move_s &tt_move,
     if (k2 == tt_move || !is_pseudo_legal(k2))
         return not_a_move;
     k2.priority = 252;
-    moves.insert(moves.begin() + move_num, k2);
+    moves.insert(moves.begin() + static_cast<std::vector<board_state::move_s>::iterator::difference_type>(move_num), k2);
     return k2;
 }
 
