diff --git a/src/netcode/uring.c b/src/netcode/uring.c
index 417ec78..c30deeb 100644
--- a/src/netcode/uring.c
+++ b/src/netcode/uring.c
@@ -15,6 +15,9 @@
 #include "hin_internal.h"
 #include "conf.h"
 
+int statx(int dirfd, const char *restrict pathname, int flags,
+                 unsigned int mask, struct statx *restrict statxbuf);
+
 #if HIN_URING_REDUCE_SYSCALLS
 #define io_uring_submit1(x)
 #else
