From a563762682a784a24e8b8705995b0e4cafad469f Mon Sep 17 00:00:00 2001
From: firefly-cpp <iztok@iztok-jr-fister.eu>
Date: Mon, 6 Dec 2021 00:11:24 +0100
Subject: [PATCH] Do not install text files in site-packages folder (#63)

---
 pyproject.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 3e94f3e..ca30dcf 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -54,9 +54,9 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Physics",
 ]
 include = [
-    "CHANGELOG.rst",
-    "CONTRIBUTORS.rst",
-    "LICENSE.txt"
+    { path="CHANGELOG.rst", format="sdist" },
+    { path="CONTRIBUTORS.rst", format="sdist" },
+    { path="LICENSE.txt", format="sdist" }
 ]

 [tool.poetry.dependencies]
