--- src/prosody-0.4.2/Makefile.org	Thu Jun  4 13:44:20 2009
+++ src/prosody-0.4.2/Makefile	Thu Jun  4 13:44:11 2009
@@ -19,8 +19,8 @@
 	install -d $(BIN) $(CONFIG) $(MODULES) $(SOURCE) $(DATA)
 	install -d $(CONFIG)/certs
 	install -d $(SOURCE)/core $(SOURCE)/net $(SOURCE)/util
-	install ./prosody.install $(BIN)/prosody
-	install ./prosodyctl.install $(BIN)/prosodyctl
+	install -m755 ./prosody.install $(BIN)/prosody
+	install -m755 ./prosodyctl.install $(BIN)/prosodyctl
 	install -m644 core/* $(SOURCE)/core
 	install -m644 net/* $(SOURCE)/net
 	install -m644 util/* $(SOURCE)/util
