--- prosody.cfg.lua.org
+++ prosody.cfg.lua
@@ -22,6 +22,9 @@
 -- Example: admins = { "user1@example.com", "user2@example.net" }
 admins = { }
 
+daemonize = true;
+pidfile = "/var/run/prosody/prosody.pid";
+
 -- Enable use of libevent for better performance under high load
 -- For more information see: http://prosody.im/doc/libevent
 --use_libevent = true;
@@ -59,7 +62,7 @@
 		--"admin_telnet"; -- Opens telnet console interface on localhost port 5582
 
 	-- Other specific functionality
-		--"posix"; -- POSIX functionality, sends server to background, enables syslog, etc.
+		"posix"; -- POSIX functionality, sends server to background, enables syslog, etc.
 		--"bosh"; -- Enable BOSH clients, aka "Jabber over HTTP"
 		--"httpserver"; -- Serve static files from a directory over HTTP
 		--"groups"; -- Shared roster support
@@ -123,8 +126,8 @@
 -- Logging configuration
 -- For advanced logging see http://prosody.im/doc/logging
 log = {
-	info = "prosody.log"; -- Change 'info' to 'debug' for verbose logging
-	error = "prosody.err";
+	info = "/var/log/prosody.log"; -- Change 'info' to 'debug' for verbose logging
+	error = "/var/log/prosody.err";
 	-- "*syslog"; -- Uncomment this for logging to syslog
 	-- "*console"; -- Log to the console, useful for debugging with daemonize=false
 }
