--- a/Rakefile
+++ b/Rakefile
@@ -116,15 +116,9 @@
   end
 
   PKG_FILES = FileList.new([
-                             'AUTHORS',
                              'bin/*',
-                             'CONTRIBUTERS',
-                             'COPYING',
                              'data/**/*',
                              'lib/**/*.rb',
-                             'man/man1/kramdown.1',
-                             'README.md',
-                             'test/**/*',
                              'VERSION',
                            ])
 
@@ -188,7 +182,7 @@
       s.homepage = "http://kramdown.gettalong.org"
     end
 
-    task gemspec: ['CONTRIBUTERS', 'VERSION', 'man/man1/kramdown.1'] do
+    task gemspec: ['VERSION', 'man/man1/kramdown.1'] do
       print "Generating Gemspec\n"
       contents = spec.to_ruby
       File.write("kramdown.gemspec", contents)
