--- a/setup.py
+++ b/setup.py
@@ -39,9 +39,5 @@
     install_requires=[
         'python-stdnum',
         ],
-    extras_require={
-        'suds': ['suds'],
-        },
     test_suite="vatnumber.tests",
-    use_2to3=True,
     )
--- a/vatnumber/__init__.py
+++ b/vatnumber/__init__.py
@@ -80,7 +80,7 @@
         return False
     base = [5, 4, 3, 2, 7, 6, 5, 4, 3, 2]
     aux = 0
-    for i in xrange(10):
+    for i in range(10):
         aux += int(vat[i]) * base[i]
     aux = 11 - (aux - (int(aux / 11) * 11))
     if aux == 11:
--- a/vatnumber/tests.py
+++ b/vatnumber/tests.py
@@ -199,7 +199,7 @@
         Test vies
         '''
         for vat in VIES_NUMBERS:
-            self.assert_(vatnumber.check_vies(vat))
+            self.assertTrue(vatnumber.check_vies(vat))
 
     def test_countries(self):
         '''
