Author: Milan P. Stanić <mps@arvanta.net>
Date:   Tue Mar 32 20:06:31 2021 +0000

add needed includes for musl in tmon.h

--- a/tools/thermal/tmon/tmon.h	2020-12-13 22:41:30.000000000 +0000
+++ b/tools/thermal/tmon/tmon.h	2021-03-03 17:45:43.570040159 +0000
@@ -10,6 +10,9 @@
 #ifndef TMON_H
 #define TMON_H
 
+#include <pthread.h>
+#include <sys/time.h>
+
 #define MAX_DISP_TEMP 125
 #define MAX_CTRL_TEMP 105
 #define MIN_CTRL_TEMP 40
