diff --git a/gkraken/__main__.py b/gkraken/__main__.py
index 791d425..812628f 100755
--- a/gkraken/__main__.py
+++ b/gkraken/__main__.py
@@ -44,10 +44,6 @@ set_log_level(logging.INFO)
 
 LOG = logging.getLogger(__name__)
 
-# POSIX locale settings
-locale.setlocale(locale.LC_ALL, locale.getlocale())
-locale.bindtextdomain(APP_PACKAGE_NAME, LOCALE_DIR)
-
 gettext.bindtextdomain(APP_PACKAGE_NAME, LOCALE_DIR)
 gettext.textdomain(APP_PACKAGE_NAME)
 
