diff --git a/config.h.in b/config.h.in
index 132745f..c9e1b00 100644
--- a/config.h.in
+++ b/config.h.in
@@ -223,6 +223,18 @@
 #ifndef _POSIX_SOURCE
 # undef _POSIX_SOURCE
 #endif
+
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
+
 /* Define to 2 if needed for POSIX-compatible behavior.  */
 #ifndef _POSIX_1_SOURCE
 # undef _POSIX_1_SOURCE
diff --git a/libglnx/glnx-dirfd.h b/libglnx/glnx-dirfd.h
index 1960820..a390e6f 100644
--- a/libglnx/glnx-dirfd.h
+++ b/libglnx/glnx-dirfd.h
@@ -29,6 +29,16 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
 G_BEGIN_DECLS
  
 /**
diff --git a/libglnx/glnx-fdio.h b/libglnx/glnx-fdio.h
index af53479..36ade62 100644
--- a/libglnx/glnx-fdio.h
+++ b/libglnx/glnx-fdio.h
@@ -36,6 +36,17 @@
 #include <glnx-macros.h>
 #include <glnx-errors.h>
 
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
+
 G_BEGIN_DECLS
 
 /* Irritatingly, g_basename() which is what we want
diff --git a/libglnx/glnx-xattrs.c b/libglnx/glnx-xattrs.c
index d0e6ab3..dda8a3a 100644
--- a/libglnx/glnx-xattrs.c
+++ b/libglnx/glnx-xattrs.c
@@ -29,6 +29,16 @@
 #include <glnx-errors.h>
 #include <glnx-local-alloc.h>
 
+/* taken from glibc unistd.h and fixes musl */
+#ifndef TEMP_FAILURE_RETRY
+#define TEMP_FAILURE_RETRY(expression) \
+  (__extension__                                                              \
+    ({ long int __result;                                                     \
+       do __result = (long int) (expression);                                 \
+       while (__result == -1L && errno == EINTR);                             \
+       __result; }))
+#endif
+
 static GVariant *
 variant_new_ay_bytes (GBytes *bytes)
 {
diff --git a/src/libostree/ostree-repo-checkout.c b/src/libostree/ostree-repo-checkout.c
index 7c7d0cc..8d489a9 100644
--- a/src/libostree/ostree-repo-checkout.c
+++ b/src/libostree/ostree-repo-checkout.c
@@ -37,6 +37,8 @@
 
 #define OVERLAYFS_WHITEOUT_PREFIX ".ostree-wh."
 
+#define ALLPERMS (S_ISUID|S_ISGID|S_ISVTX|S_IRWXU|S_IRWXG|S_IRWXO)
+
 /* Per-checkout call state/caching */
 typedef struct {
   GString *path_buf; /* buffer for real path if filtering enabled */
