diff --git a/winexe/CMakeLists.txt b/winexe/CMakeLists.txt
index a0e8bb5..2cb6040 100644
--- a/winexe/CMakeLists.txt
+++ b/winexe/CMakeLists.txt
@@ -54,7 +54,7 @@ include_directories(${SMB_INCLUDES})
 include_directories(${POPT_INCLUDE_DIRS})
 include_directories(${GLIB_INCLUDE_DIRS})
 
-find_program(MINGW_GCC NAMES i686-w64-mingw32-gcc i586-mingw32msvc-gcc)
+find_program(MINGW_GCC NAMES i686-w64-mingw32-gcc i586-mingw32msvc-gcc x86_64-w64-mingw32-gcc x86-w64-mingw32-gcc)
 
 if (NOT MINGW_GCC)
   message(SEND_ERROR "i586-mingw32msvc-gcc not found in PATH.\nIt is needed to build the Windows Service for winexe.")
