From 007a3bd53811db3b01464c5968a42dc4d80f34e8 Mon Sep 17 00:00:00 2001
From: Rinigus <rinigus.git@gmail.com>
Date: Sun, 7 Jun 2020 23:33:32 +0300
Subject: [PATCH] add long int to toString

---
 include/mbgl/util/string.hpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/mbgl/util/string.hpp b/include/mbgl/util/string.hpp
index 27c3943c5f0..d235c980f54 100644
--- a/include/mbgl/util/string.hpp
+++ b/include/mbgl/util/string.hpp
@@ -47,6 +47,11 @@ inline std::string toString(uint8_t t) {
     return toString(static_cast<uint32_t>(t));
 }
 
+template <typename = std::enable_if<!std::is_same<int64_t, long int>::value>>
+inline std::string toString(long int t) {
+    return toString(static_cast<int64_t>(t));
+}
+
 template <typename = std::enable_if<!std::is_same<uint64_t, unsigned long>::value>>
 inline std::string toString(unsigned long t) {
     return toString(static_cast<uint64_t>(t));
