--- postgresql-bdr-9.4.14_bdr1.orig/config/c-compiler.m4
+++ postgresql-bdr-9.4.14_bdr1/config/c-compiler.m4
@@ -70,7 +70,7 @@
 ac_int64 a = 20000001;
 ac_int64 b = 40000005;
 
-int does_int64_work()
+int does_int64_work(void)
 {
   ac_int64 c,d;
 
@@ -84,8 +84,8 @@
     return 0;
   return 1;
 }
-main() {
-  exit(! does_int64_work());
+int main(void) {
+  return !does_int64_work();
 }],
 [Ac_cachevar=yes],
 [Ac_cachevar=no],
--- postgresql-bdr-9.4.14_bdr1.orig/configure.in
+++ postgresql-bdr-9.4.14_bdr1/configure.in
@@ -19,10 +19,6 @@
 
 AC_INIT([PostgreSQL], [9.4.14], [pgsql-bugs@postgresql.org])
 
-m4_if(m4_defn([m4_PACKAGE_VERSION]), [2.69], [], [m4_fatal([Autoconf version 2.69 is required.
-Untested combinations of 'autoconf' and PostgreSQL versions are not
-recommended.  You can remove the check from 'configure.in' but it is then
-your responsibility whether the result works or not.])])
 AC_COPYRIGHT([Copyright (c) 1996-2014, PostgreSQL Global Development Group])
 AC_CONFIG_SRCDIR([src/backend/access/common/heaptuple.c])
 AC_CONFIG_AUX_DIR(config)
