--- a/tests/test01.c	2021-01-26 14:54:03.000000000 +0000
+++ b/tests/test01.c	2021-02-01 10:44:05.788144855 +0000
@@ -136,7 +136,7 @@
 
 	CPU_ZERO(&cpuset);
 	CPU_SET(cpu, &cpuset);
-	status = pthread_attr_setaffinity_np(&attr, sizeof(cpuset), &cpuset);
+	status = pthread_setaffinity_np(&attr, sizeof(cpuset), &cpuset);
 	if (status != 0) {
 		error("failed to set blocker affinity to cpu %d: %s\n",
 			cpu, strerror(errno));
