diff --git a/tests/Makefile.in b/tests/Makefile.in
index 601eb30..2ddcfa1 100644
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -110,8 +110,6 @@ noinst_PROGRAMS = identify$(EXEEXT) $(am__EXEEXT_2)
 check_PROGRAMS = localhost$(EXEEXT) readutmp$(EXEEXT) \
 	waitdaemon$(EXEEXT) $(am__EXEEXT_1)
 @ENABLE_inetd_TRUE@am__append_1 = addrpeek tcpget
-@ENABLE_libls_TRUE@am__append_2 = ls
-@ENABLE_libls_TRUE@am__append_3 = libls.sh
 @ENABLE_ping_TRUE@am__append_4 = ping-localhost.sh
 @ENABLE_traceroute_TRUE@am__append_5 = traceroute-localhost.sh
 @ENABLE_inetd_TRUE@@ENABLE_tftp_TRUE@@ENABLE_tftpd_TRUE@am__append_6 = tftp.sh
@@ -241,7 +239,6 @@ CONFIG_HEADER = $(top_builddir)/config.h
 CONFIG_CLEAN_FILES =
 CONFIG_CLEAN_VPATH_FILES =
 @ENABLE_inetd_TRUE@am__EXEEXT_1 = addrpeek$(EXEEXT) tcpget$(EXEEXT)
-@ENABLE_libls_TRUE@am__EXEEXT_2 = ls$(EXEEXT)
 PROGRAMS = $(noinst_PROGRAMS)
 addrpeek_SOURCES = addrpeek.c
 addrpeek_OBJECTS = addrpeek.$(OBJEXT)
@@ -257,8 +254,6 @@ localhost_LDADD = $(LDADD)
 localhost_DEPENDENCIES = $(am__DEPENDENCIES_1)
 ls_SOURCES = ls.c
 ls_OBJECTS = ls.$(OBJEXT)
-@ENABLE_libls_TRUE@ls_DEPENDENCIES = $(am__DEPENDENCIES_1) \
-@ENABLE_libls_TRUE@	$(am__DEPENDENCIES_1)
 readutmp_SOURCES = readutmp.c
 readutmp_OBJECTS = readutmp.$(OBJEXT)
 readutmp_LDADD = $(LDADD)
@@ -271,7 +266,7 @@ waitdaemon_SOURCES = waitdaemon.c
 waitdaemon_OBJECTS = waitdaemon.$(OBJEXT)
 waitdaemon_LDADD = $(LDADD)
 waitdaemon_DEPENDENCIES = $(am__DEPENDENCIES_1)
-am__dist_check_SCRIPTS_DIST = utmp.sh libls.sh ping-localhost.sh \
+am__dist_check_SCRIPTS_DIST = utmp.sh ping-localhost.sh \
 	traceroute-localhost.sh tftp.sh syslogd.sh ftp-parser.sh \
 	ftp-localhost.sh inetd.sh telnet-localhost.sh hostname.sh \
 	dnsdomainname.sh ifconfig.sh
@@ -1147,7 +1142,6 @@ LIBGNU_LIBDEPS = @LIBGNU_LIBDEPS@
 LIBGNU_LTLIBDEPS = @LIBGNU_LTLIBDEPS@
 LIBIDN = @LIBIDN@
 LIBINTL = @LIBINTL@
-LIBLS = @LIBLS@
 LIBMULTITHREAD = @LIBMULTITHREAD@
 LIBNCURSES = @LIBNCURSES@
 LIBOBJS = @LIBOBJS@
@@ -1477,7 +1471,6 @@ enable_ftpd = @enable_ftpd@
 enable_hostname = @enable_hostname@
 enable_ifconfig = @enable_ifconfig@
 enable_inetd = @enable_inetd@
-enable_libls = @enable_libls@
 enable_logger = @enable_logger@
 enable_ping = @enable_ping@
 enable_ping6 = @enable_ping6@
@@ -1533,7 +1526,6 @@ iu_INCLUDES = @iu_INCLUDES@
 iu_LIBRARIES = @iu_LIBRARIES@
 libdir = @libdir@
 libexecdir = @libexecdir@
-libls_BUILD = @libls_BUILD@
 lispdir = @lispdir@
 localedir = @localedir@
 localstatedir = @localstatedir@
@@ -1620,7 +1612,6 @@ dist_check_SCRIPTS = utmp.sh $(am__append_3) $(am__append_4) \
 	$(am__append_5) $(am__append_6) $(am__append_7) \
 	$(am__append_8) $(am__append_9) $(am__append_10) \
 	$(am__append_11) $(am__append_12) $(am__append_13)
-@ENABLE_libls_TRUE@ls_LDADD = $(LIBLS) $(iu_LIBRARIES)
 TESTS_ENVIRONMENT = EXEEXT=$(EXEEXT)
 EXTRA_DIST = tools.sh.in ifconfig_modes.sh
 BUILT_SOURCES = tools.sh
@@ -1945,13 +1936,6 @@ utmp.sh.log: utmp.sh
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-libls.sh.log: libls.sh
-	@p='libls.sh'; \
-	b='libls.sh'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 ping-localhost.sh.log: ping-localhost.sh
 	@p='ping-localhost.sh'; \
 	b='ping-localhost.sh'; \

