diff --git a/src/gdu-gtk/gdu-size-widget.c b/src/gdu-gtk/gdu-size-widget.c
index 4dd6a8f..715ddad 100644
--- a/src/gdu-gtk/gdu-size-widget.c
+++ b/src/gdu-gtk/gdu-size-widget.c
@@ -564,7 +564,7 @@ update_stepping (GduSizeWidget *widget)
 
                 shown_extent = extent / unit_factor;
 
-                increment = (exp10 (floor (log10 (shown_extent))) / 10.0) * unit_factor;
+                increment = (exp (M_LN10 * floor (log10 (shown_extent))) / 10.0) * unit_factor;
 
                 gtk_range_set_increments (GTK_RANGE (widget->priv->hscale),
                                           increment,
