Ref https://github.com/rust-lang/cc-rs/issues/1581

--- a/third_party/rust/cc/src/target/llvm.rs
+++ b/third_party/rust/cc/src/target/llvm.rs
@@ -98,6 +98,7 @@
             "llvm" | "softfloat" | "uwp" | "vec-extabi" => "",
             "ilp32" => "_ilp32",
             "abi64" => "",
+            "elfv1" | "elfv2" => "",
             abi => abi,
         };
         Cow::Owned(match (vendor, env, abi) {
