diff --git a/src/defs.h b/src/defs.h
index 68cb80a..aae7905 100644
--- a/src/defs.h
+++ b/src/defs.h
@@ -156,6 +156,9 @@ struct session_data {
    int training_request;
    float spaminess;
    float Nham, Nspam;
+#ifdef __user
+#undef __user
+#endif
    float __acquire, __parsed, __av, __user, __policy, __training, __minefield, __as, __update, __store, __inject;
 #ifdef HAVE_MAILBUF
    char mailbuf[MAILBUFSIZE], discard_mailbuf;
diff --git a/webui/view/theme/default/templates/common/footer.tpl b/webui/view/theme/default/templates/common/footer.tpl
index e2a23fe..0f681bb 100644
--- a/webui/view/theme/default/templates/common/footer.tpl
+++ b/webui/view/theme/default/templates/common/footer.tpl
@@ -1,3 +1,3 @@
 
-   <p>clapf web UI, VERSION_STRING</p>
+   <p>clapf web UI, 0.4.7.4</p>
 
