diff --git a/tests.py b/tests.py
index 833fcc6..3c1c937 100644
--- a/tests.py
+++ b/tests.py
@@ -5,3 +5,3 @@ import os
 import sys
-import imp
+import importlib
 import shutil
@@ -310,6 +310,6 @@ def test_environment():
         # PySide is not available for Python > 3.4
-        imp.find_module("PySide")
-    imp.find_module("PySide2")
-    imp.find_module("PyQt4")
-    imp.find_module("PyQt5")
+        importlib.util.find_spec("PySide")
+    importlib.util.find_spec("PySide2")
+    importlib.util.find_spec("PyQt4")
+    importlib.util.find_spec("PyQt5")
 
