--- a/share/mk/dist/files.mk
+++ b/share/mk/dist/files.mk
@@ -49,7 +49,7 @@ $(_DISTVERSION): $(MAKEFILEDIR)/configure/version.mk $(MK) $(FORCE_DISTVERSION)
 	<$< \
 	$(SED) 's/^VERSION *:=.*/VERSION := $(VERSION)/' \
 	| $(SED) 's/^DISTDATE *:=.*/DISTDATE := $(DISTDATE)/' \
-	| $(INSTALL_DATA) -T /dev/stdin $@
+	| $(INSTALL_DATA) /dev/stdin $@
 
 $(_DISTOTHERS): $(_DISTDIR)/%: $(srcdir)/% $(MK) | $$(@D)/
 	$(info	$(INFO_)CP		$@)
--- a/share/mk/install/bin.mk
+++ b/share/mk/install/bin.mk
@@ -26,7 +26,7 @@ $(_bin): $(MK) | $$(@D)/
 
 $(_bin):
 	$(info	$(INFO_)INSTALL		$@)
-	$(INSTALL_PROGRAM) -T $< $@
+	$(INSTALL_PROGRAM) $< $@
 
 
 .PHONY: install-bin
--- a/share/mk/install/html.mk
+++ b/share/mk/install/html.mk
@@ -22,7 +22,7 @@ _htmlpages_rm := $(addsuffix -rm, $(wildcard $(_htmlpages)))
 
 $(_htmlpages): $(_htmldir)/%: $(_MANDIR)/% $(MK) | $$(@D)/
 	$(info	$(INFO_)INSTALL		$@)
-	$(INSTALL_DATA) -T $< $@
+	$(INSTALL_DATA) $< $@
 
 
 .PHONY: install-html
--- a/share/mk/install/man.mk
+++ b/share/mk/install/man.mk
@@ -51,7 +51,7 @@ $(foreach s, $(MANSECTIONS),                                                  \
 
 $(_manpages):
 	$(info	$(INFO_)INSTALL		$@)
-	$(INSTALL_DATA) -T $< $@
+	$(INSTALL_DATA) $< $@
 	$(SED) -i $(foreach s, $(MANSECTIONS), \
 		-e '/^\.so /s, man$(s)/\(.*\)\.$(s)$$, $(notdir $(man$(s)dir))/\1$(man$(s)ext),') \
 		$@
--- a/share/mk/install/pdf/book.mk
+++ b/share/mk/install/pdf/book.mk
@@ -20,7 +20,7 @@ _pdf_book_rm := $(addsuffix -rm, $(wildcard $(_pdf_book)))
 
 $(_pdf_book): $(_pdfdir)/%: $(_PDFDIR)/% $(MK) | $$(@D)/
 	$(info	$(INFO_)INSTALL		$@)
-	$(INSTALL_DATA) -T $< $@
+	$(INSTALL_DATA) $< $@
 
 
 .PHONY: install-pdf-book

