diff --git a/Makefile b/Makefile
index 50a4579..2496412 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,7 @@ install:
 
 # Check if groff reports warnings (may be words of sentances not displayed)
 # from http://lintian.debian.org/tags/manpage-has-errors-from-man.html 
-GROFF_LOG := $(shell mktemp /tmp/manpages-checksXXXX)
+GROFF_LOG := $(shell mktemp /tmp/manpages-checksXXXXXX)
 check-groff-warnings:
 	for i in man?/*.[1-9]; \
 	do \
