--- a/share/mk/dist.mk
+++ b/share/mk/dist.mk
@@ -49,7 +49,7 @@
 	<$< \
 	$(SED) "/^.TH/s/(date)/$$(git log --format=%cs -1 -- $< $(HIDE_ERR))/" \
 	| $(SED) '/^.TH/s/(unreleased)/$(DISTVERSION)/' \
-	| $(INSTALL_DATA) -T /dev/stdin $@
+	| $(INSTALL_DATA) /dev/stdin $@
 
 $(_DISTOTHERS): $(_DISTDIR)/%: $(srcdir)/% $(MK) | $$(@D)/
 	$(info CP	$@)
--- a/share/mk/install/html.mk
+++ b/share/mk/install/html.mk
@@ -23,7 +23,7 @@
 
 $(_htmlpages): $(_htmldir)/%: $(_HTMLDIR)/% $(MK) | $$(@D)/
 	$(info INSTALL	$@)
-	$(INSTALL_DATA) -T $< $@
+	$(INSTALL_DATA) $< $@
 
 
 .PHONY: install-html
--- a/share/mk/install/man.mk
+++ b/share/mk/install/man.mk
@@ -63,7 +63,7 @@
 	<$< \
 	$(SED) $(foreach s, $(MANSECTIONS), \
 		-e '/^\.so /s, man$(s)/\(.*\)\.$(s)$$, $(notdir $(man$(s)dir))/\1$(man$(s)ext)$(Z),') \
-	| $(INSTALL_DATA) -T /dev/stdin $@
+	| $(INSTALL_DATA) /dev/stdin $@
 ifeq ($(LINK_PAGES),symlink)
 	if $(GREP) '^\.so ' <$@ >/dev/null; then \
 		$(GREP) '^\.so ' <$@ \
