diff --git a/CMakeLists.txt b/CMakeLists.txt
index d23aa2f..edc9c09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,7 +37,7 @@ set_target_properties(piper_phonemize PROPERTIES
 # ---- onnxruntime ---
 
 # Look for onnxruntime files in <root>/lib
-if(NOT DEFINED ONNXRUNTIME_DIR)
+if(FALSE)
     if(NOT DEFINED ONNXRUNTIME_VERSION)
         set(ONNXRUNTIME_VERSION "1.14.1")
     endif()
@@ -99,6 +99,7 @@ endif()
 
 find_package(PkgConfig)
 pkg_check_modules(ESPEAK_NG REQUIRED espeak-ng<2)
+pkg_check_modules(ONNXRUNTIME REQUIRED libonnxruntime)
 
 if(FALSE)
     set(ESPEAK_NG_DIR "${CMAKE_CURRENT_BINARY_DIR}/ei")
@@ -128,12 +129,12 @@ endif()
 target_include_directories(
     piper_phonemize PUBLIC
     "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>"
-    ${ONNXRUNTIME_DIR}/include
+    ${ONNXRUNTIME_INCLUDE_DIRS}
 )
 
 target_link_directories(
     piper_phonemize PUBLIC
-    ${ONNXRUNTIME_DIR}/lib
+    ${ONNXRUNTIME_LIBRARY_DIRS}
 )
 
 target_link_libraries(
diff --git a/src/phonemize.cpp b/src/phonemize.cpp
index c920a95..cc752e0 100644
--- a/src/phonemize.cpp
+++ b/src/phonemize.cpp
@@ -3,7 +3,7 @@
 #include <vector>
 
 #include <espeak-ng/speak_lib.h>
-#include <onnxruntime_cxx_api.h>
+#include <onnxruntime/onnxruntime_cxx_api.h>
 
 #include "phonemize.hpp"
 #include "uni_algo.h"
diff --git a/src/tashkeel.cpp b/src/tashkeel.cpp
index 23683b7..48d12b2 100644
--- a/src/tashkeel.cpp
+++ b/src/tashkeel.cpp
@@ -6,7 +6,7 @@
 #include <string>
 #include <vector>
 
-#include <onnxruntime_cxx_api.h>
+#include <onnxruntime/onnxruntime_cxx_api.h>
 
 #include "tashkeel.hpp"
 #include "uni_algo.h"
diff --git a/src/tashkeel.hpp b/src/tashkeel.hpp
index 1611779..123418b 100644
--- a/src/tashkeel.hpp
+++ b/src/tashkeel.hpp
@@ -5,7 +5,7 @@
 #include <set>
 #include <string>
 
-#include <onnxruntime_cxx_api.h>
+#include <onnxruntime/onnxruntime_cxx_api.h>
 
 #include "shared.hpp"
 
