--- a/rpython/rtyper/lltypesystem/rffi.py	2021-10-12 13:51:19.679004256 +0200
+++ b/rpython/rtyper/lltypesystem/rffi.py	2021-10-12 13:51:39.411425448 +0200
@@ -693,7 +693,7 @@
     c_setter = "void %(setter_name)s (%(c_type)s v) { %(name)s = v; }" % locals()
 
     lines = ["#include <%s>" % i for i in eci.includes]
-    if declare_as_extern:
+    if declare_as_extern and name not in ['stdin', 'stdout', 'stderr']:
         lines.append('extern %s %s;' % (c_type, name))
     lines.append(c_getter)
     if not getter_only:
