diff -r b30914a448be pypy/module/sys/system.py
--- a/pypy/module/sys/system.py	Tue Jun 06 17:13:19 2017 +0100
+++ b/pypy/module/sys/system.py	Thu Jun 08 14:56:49 2017 +0300
@@ -117,8 +117,7 @@
         w_lock = space.newtext("semaphore" if rthread.USE_SEMAPHORES
                                else "mutex+cond")
         if rthread.CS_GNU_LIBPTHREAD_VERSION is not None:
-            w_version = space.newtext(
-                os.confstr(rthread.CS_GNU_LIBPTHREAD_VERSION))
+            w_version = space.newtext("NPTL")
         else:
             w_version = space.w_None
     else:
