--- ./Makefile.orig
+++ ./Makefile
@@ -64,11 +64,11 @@
 	install -m 0444 tbl-$(VERSION).md5 $(PREFIX)/snapshots/tbl.md5
 
 install:
-	mkdir -p $(BINDIR)
-	mkdir -p $(MANDIR)/man5
-	$(INSTALL_PROGRAM) tbl $(BINDIR)
-	$(INSTALL_MAN) tbl.5 $(MANDIR)/man5
-	$(INSTALL_MAN) tbl.1 $(MANDIR)/man1
+	mkdir -p $(DESTDIR)$(BINDIR)
+	mkdir -p $(DESTDIR)$(MANDIR)/man5
+	$(INSTALL_PROGRAM) tbl $(DESTDIR)$(BINDIR)
+	$(INSTALL_MAN) tbl.5 $(DESTDIR)$(MANDIR)/man5
+	$(INSTALL_MAN) tbl.1 $(DESTDIR)$(MANDIR)/man1
 
 main.o: main.c tbl.h
 
