From dcfd65fd5a2fb5d1cc74c961b26e46fcb35cb5f3 Mon Sep 17 00:00:00 2001
From: Devin Lin <espidev@gmail.com>
Date: Sat, 19 Jun 2021 11:58:12 -0400
Subject: [PATCH] Remove x1.5 mobile icon sizing

---
 style/kirigami/Units.qml | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/style/kirigami/Units.qml b/style/kirigami/Units.qml
index bcd944a..7359590 100644
--- a/style/kirigami/Units.qml
+++ b/style/kirigami/Units.qml
@@ -51,19 +51,12 @@ QtObject {
         property int tiny: fontMetrics.roundedIconSize(8 * devicePixelRatio)
         property int tinySmall: fontMetrics.roundedIconSize(12 * devicePixelRatio)
         // Breeze
-        property int small: fontMetrics.roundedIconSize(16 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1))
-        property int smallMedium: fontMetrics.roundedIconSize(22 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1))
-        property int medium: fontMetrics.roundedIconSize(32 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1))
-        property int large: fontMetrics.roundedIconSize(48 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1))
-        property int huge: fontMetrics.roundedIconSize(64 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1))
-        property int enormous: 128 * devicePixelRatio * (QtQuickControlsPrivate.Settings.isMobile ? 1.5 : 1)
-        // TODO: Using larger sizes on mobile because that's what Kirigami normally does. Maybe don't do this in the future.
-        //property int small: fontMetrics.roundedIconSize(16 * devicePixelRatio)
-        //property int smallMedium: fontMetrics.roundedIconSize(22 * devicePixelRatio)
-        //property int medium: fontMetrics.roundedIconSize(32 * devicePixelRatio)
-        //property int large: fontMetrics.roundedIconSize(48 * devicePixelRatio)
-        //property int huge: fontMetrics.roundedIconSize(64 * devicePixelRatio)
-        //property int enormous: 128 * devicePixelRatio
+        property int small: fontMetrics.roundedIconSize(16 * devicePixelRatio)
+        property int smallMedium: fontMetrics.roundedIconSize(22 * devicePixelRatio)
+        property int medium: fontMetrics.roundedIconSize(32 * devicePixelRatio)
+        property int large: fontMetrics.roundedIconSize(48 * devicePixelRatio)
+        property int huge: fontMetrics.roundedIconSize(64 * devicePixelRatio)
+        property int enormous: 128 * devicePixelRatio
     }
 
     //BEGIN Breeze Units
-- 
GitLab

