diff --git a/passwordsafe/keyfile_generator.py b/passwordsafe/keyfile_generator.py
index 346263e..fce1384 100644
--- a/passwordsafe/keyfile_generator.py
+++ b/passwordsafe/keyfile_generator.py
@@ -1,6 +1,6 @@
 from gi.repository import GLib
-from Crypto.Cipher import AES
-from Crypto.Random import get_random_bytes
+from Cryptodome.Cipher import AES
+from Cryptodome.Random import get_random_bytes
 import secrets
 
 
