From 2d6e3fe90ae2c24d29305037b452b4cb18bceaa0 Mon Sep 17 00:00:00 2001
From: Max Kellermann <max@musicpd.org>
Date: Mon, 12 Feb 2018 21:50:44 +0100
Subject: [PATCH] doc/meson.build: drop bogus "man" from manpage install path

Closes #12
---
 doc/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/meson.build b/doc/meson.build
index fd7e3d6..7793627 100644
--- a/doc/meson.build
+++ b/doc/meson.build
@@ -15,9 +15,9 @@ if sphinx.found()
     'Manpage documentation',
     output: 'man',
     input: ['index.rst', 'conf.py'],
-    command: [sphinx, '-q', '-b', 'man', '-d', '@OUTDIR@/doctrees', meson.current_source_dir(), '@OUTPUT@'],
+    command: [sphinx, '-q', '-b', 'man', '-d', '@OUTDIR@/doctrees', meson.current_source_dir(), '@OUTPUT@/man1'],
     build_by_default: true,
     install: true,
-    install_dir: join_paths(get_option('datadir'), 'man', 'man1'),
+    install_dir: get_option('datadir'),
   )
 endif
-- 
2.16.1

