diff --git a/meson.build b/meson.build
index d919bc3..3390fb5 100644
--- a/meson.build
+++ b/meson.build
@@ -126,7 +126,7 @@ endif
 
 with_documentation = get_option('documentation')
 if not with_documentation.disabled()
-  sphinx = find_program('sphinx-build', required: with_documentation)
+  sphinx = find_program('sphinx-build-3', required: with_documentation)
   if sphinx.found()
     subdir('doc')
   endif

