not required
diff --git a/pyproject.toml b/pyproject.toml
index ecd7b7e..7a8d82a 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -42,7 +42,6 @@ cryptography = ">=2"
 pyyaml = ">=3"
 requests = ">=2"
 beautifulsoup4 = ">=4"
-importlib-metadata = ">=4"
 # Optional dependencies required by some providers
 boto3 = { version = ">=1", optional = true }  # route53
 localzone = { version = ">=0.9.8", optional = true }  # localzone
