--- a/iodata-qt6.prf
+++ b/iodata-qt6.prf
@@ -5,11 +5,11 @@ QMAKE_EXTRA_COMPILERS += iodata_type_to_cxx iodata_type_to_h
 iodata_type_to_cxx.input = IODATA_TYPES
 iodata_type_to_cxx.output = iodata_${QMAKE_FILE_IN_BASE}.cpp
 iodata_type_to_cxx.variable_out = SOURCES
-iodata_type_to_cxx.commands = iodata-qt6-type-to-c++ ${QMAKE_FILE_IN} -o iodata_${QMAKE_FILE_IN_BASE}.cpp -d ${QMAKE_FILE_IN}.h
+iodata_type_to_cxx.commands = iodata-qt6-type-to-c++ -o iodata_${QMAKE_FILE_IN_BASE}.cpp -d ${QMAKE_FILE_IN}.h ${QMAKE_FILE_IN}
 
 iodata_type_to_h.input = IODATA_TYPES
 iodata_type_to_h.output = ${QMAKE_FILE_IN}.h
 iodata_type_to_h.variable_out = HEADERS
-iodata_type_to_h.commands = iodata-qt6-type-to-c++ ${QMAKE_FILE_IN} -o iodata_${QMAKE_FILE_IN_BASE}.cpp -d ${QMAKE_FILE_IN}.h
+iodata_type_to_h.commands = iodata-qt6-type-to-c++ -o iodata_${QMAKE_FILE_IN_BASE}.cpp -d ${QMAKE_FILE_IN}.h ${QMAKE_FILE_IN}
 
 # iodata-qt6-type-to-c++ ${QMAKE_FILE_IN} -d ${QMAKE_FILE_OUT}
