--- ./Lib/ctypes/util.py.orig	2013-05-27 14:59:31.253297476 +0000
+++ ./Lib/ctypes/util.py	2013-05-27 14:59:55.730215183 +0000
@@ -198,7 +198,7 @@
             abi_type = mach_map.get(machine, 'libc6')
 
             # XXX assuming GLIBC's ldconfig (with option -p)
-            expr = r'\s+(lib%s\.[^\s]+)\s+\(%s' % (re.escape(name), abi_type)
+            expr = r'\s+(lib%s\.[^\s]+)\s+\(' % (re.escape(name))
             f = os.popen('/sbin/ldconfig -p 2>/dev/null')
             try:
                 data = f.read()
