--- a/panels/background/cc-background-item.c
+++ b/panels/background/cc-background-item.c
@@ -172,7 +172,7 @@
                 gdk_color_parse (item->priv->secondary_color, &scolor);
         }
 
-        gnome_bg_set_color (item->priv->bg, item->priv->shading, &pcolor, &scolor);
+        gnome_bg_set_rgba (item->priv->bg, item->priv->shading, &pcolor, &scolor);
         gnome_bg_set_placement (item->priv->bg, item->priv->placement);
 }
 
