--- a/syslog.gemspec	2023-12-16 00:05:40.000000000 -0500
+++ b/syslog.gemspec	2024-10-07 16:24:00.450195312 -0400
@@ -18,9 +18,7 @@
   spec.metadata["homepage_uri"] = spec.homepage
   spec.metadata["source_code_uri"] = spec.homepage
 
-  spec.files         = Dir.chdir(File.expand_path('..', __FILE__)) do
-    `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
-  end
+  spec.files         = Dir["lib/**/*"]
   spec.extensions    = ["ext/syslog/extconf.rb"]
   spec.bindir        = "exe"
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
