From c87a4476240d4a2c3e4a9060464aaf9b266357e9 Mon Sep 17 00:00:00 2001
From: Wenlong Zhang <zhangwenlong@loongson.cn>
Date: Mon, 1 Jul 2024 12:42:41 +0000
Subject: [PATCH 1/2] fix build

---
 Makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 4a7f6d0..a5a6b81 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,7 @@
 VERSION=2.1
 
-DEBUG=-g
-LDFLAGS=-liconv_hook -lmrss $(DEBUG)
-CFLAGS=-O3 -Wall --std=gnu11 -DVERSION=\"$(VERSION)\" $(DEBUG)
+override LDFLAGS+=-lmrss
+override CFLAGS+=-Wall -DVERSION=\"$(VERSION)\" $(DEBUG)
 
 OBJS=r2t.o
 
-- 
2.45.2

