From 8a6097c31a770d9d4f8202f91aeb4a231a525102 Mon Sep 17 00:00:00 2001
From: Wenlong Zhang <zhangwenlong@loongson.cn>
Date: Mon, 1 Jul 2024 12:44:40 +0000
Subject: [PATCH 2/2] makefile suppor install destdir path

---
 Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index a5a6b81..4f36e2e 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@ override CFLAGS+=-Wall -DVERSION=\"$(VERSION)\" $(DEBUG)
 
 OBJS=r2t.o
 
-prefix ?= /usr/local
+prefix ?= /usr
 bindir ?= $(prefix)/bin
 mandir ?= $(prefix)/share/man/man1
 
@@ -15,10 +15,10 @@ rsstail: $(OBJS)
 	$(CC) -Wall -W $(OBJS) $(LDFLAGS) -o rsstail
 
 install: rsstail
-	mkdir -p $(bindir)
-	mkdir -p $(mandir)
-	install rsstail $(bindir)
-	install -m 644 rsstail.1 $(mandir)
+	mkdir -p $(DESTDIR)$(bindir)
+	mkdir -p $(DESTDIR)$(mandir)
+	install rsstail $(DESTDIR)$(bindir)
+	install -m 644 rsstail.1 $(DESTDIR)$(mandir)
 
 uninstall:
 	rm $(bindir)/rsstail $(mandir)/rsstail.1
-- 
2.45.2

