diff --git a/cpp/src/phonenumbers/geocoding/phonenumber_offline_geocoder.cc b/cpp/src/phonenumbers/geocoding/phonenumber_offline_geocoder.cc
index 1e5a5bacc..e15db99e2 100644
--- a/cpp/src/phonenumbers/geocoding/phonenumber_offline_geocoder.cc
+++ b/cpp/src/phonenumbers/geocoding/phonenumber_offline_geocoder.cc
@@ -77,7 +77,7 @@ void PhoneNumberOfflineGeocoder::Init(
 }
 
 PhoneNumberOfflineGeocoder::~PhoneNumberOfflineGeocoder() {
-  absl::MutexLock l(&mu_);
+  absl::MutexLock l(mu_);
   gtl::STLDeleteContainerPairSecondPointers(
       available_maps_.begin(), available_maps_.end());
 }
@@ -195,7 +195,7 @@ const char* PhoneNumberOfflineGeocoder::GetAreaDescription(
   const int country_calling_code = number.country_code();
   // NANPA area is not split in C++ code.
   const int phone_prefix = country_calling_code;
-  absl::MutexLock l(&mu_);
+  absl::MutexLock l(mu_);
   const AreaCodeMap* const descriptions = GetPhonePrefixDescriptions(
       phone_prefix, lang, script, region);
   const char* description = descriptions ? descriptions->Lookup(number) : NULL;
