--- ushare-1.1a/scripts/ushare.conf	2007-12-09 07:03:36.000000000 -0600
+++ b/scripts/ushare.conf	2010-11-15 12:09:45.000000000 -0600
@@ -31,15 +31,15 @@ USHARE_DIR=
 USHARE_OVERRIDE_ICONV_ERR=
 
 # Enable Web interface (yes/no)
-ENABLE_WEB=
+USHARE_ENABLE_WEB=
 
 # Enable Telnet control interface (yes/no)
-ENABLE_TELNET=
+USHARE_ENABLE_TELNET=
 
 # Use XboX 360 compatibility mode (yes/no)
-ENABLE_XBOX=
+USHARE_ENABLE_XBOX=
 
 # Use DLNA profile (yes/no)
 # This is needed for PlayStation3 to work (among other devices)
-ENABLE_DLNA=
+USHARE_ENABLE_DLNA=
 
