--- a/oslib/linux-blkzoned.c
+++ b/oslib/linux-blkzoned.c
@@ -26,6 +26,10 @@
 #define BLKFINISHZONE _IOW(0x12, 136, struct blk_zone_range)
 #endif

+#ifndef FALLOC_FL_ZERO_RANGE
+#define FALLOC_FL_ZERO_RANGE    0x10 /* zero range of bytes */
+#endif
+
 /*
  * If the uapi headers installed on the system lacks zone capacity support,
  * use our own locally defined macros. Since Linux 5.9 zone capacity is