--- a/oslib/linux-blkzoned.c
+++ b/oslib/linux-blkzoned.c
@@ -26,8 +26,12 @@
 #define BLKFINISHZONE _IOW(0x12, 136, struct blk_zone_range)
 #endif
 #include <linux/falloc.h>
 
+#ifndef FALLOC_FL_ZERO_RANGE
+#define FALLOC_FL_ZERO_RANGE    0x10 /* zero range of bytes */
+#endif
+
 /*
  * If the uapi headers installed on the system lacks zone capacity support,
  * use our local versions. If the installed headers are recent enough to
  * support zone capacity, do not redefine any structs.
