From b57aa0e889fe07d7b4663d0729a40f4a9ecf5578 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=2E=20David=20Ib=C3=A1=C3=B1ez?= <jdavid.ibp@gmail.com>
Date: Wed, 14 Oct 2020 08:36:08 +0200
Subject: [PATCH] Upgrade to libgit2 1.1

Fixes #1040

diff --git a/src/pygit2.c b/src/pygit2.c
index 4e144705..0693a790 100644
--- a/src/pygit2.c
+++ b/src/pygit2.c
@@ -590,6 +590,7 @@ PyInit__pygit2(void)
     ADD_CONSTANT_INT(m, GIT_BLAME_TRACK_COPIES_ANY_COMMIT_COPIES)
     ADD_CONSTANT_INT(m, GIT_BLAME_FIRST_PARENT)
     ADD_CONSTANT_INT(m, GIT_BLAME_USE_MAILMAP)
+    ADD_CONSTANT_INT(m, GIT_BLAME_IGNORE_WHITESPACE)
 
     /* Merge */
     ADD_CONSTANT_INT(m, GIT_MERGE_ANALYSIS_NONE)
diff --git a/src/types.h b/src/types.h
index 0f3c8cf9..8ea01d38 100644
--- a/src/types.h
+++ b/src/types.h
@@ -32,8 +32,8 @@
 #include <Python.h>
 #include <git2.h>
 
-#if !(LIBGIT2_VER_MAJOR == 1 && LIBGIT2_VER_MINOR == 0)
-#error You need a compatible libgit2 version (1.0.x)
+#if !(LIBGIT2_VER_MAJOR == 1 && LIBGIT2_VER_MINOR == 1)
+#error You need a compatible libgit2 version (1.1.x)
 #endif
 
 /*
