--- a/src/server/partimaged-client.cpp
+++ b/src/server/partimaged-client.cpp
@@ -62,7 +62,7 @@
 {
   pthread_mutex_lock(&mClients);
   Clients[client].Sock = sock;
-  Clients[client].Ssl = NULL;
+  Clients[client].Ssl = 0;
   
   pthread_mutex_unlock(&mClients);
 }
@@ -89,9 +89,9 @@
   pthread_mutex_lock(&mClients);
   showDebug(1, "%d released\n", client);
   shutdown(Clients[client].Sock, SHUT_RDWR);
-  Clients[client].Sock = NULL;
+  Clients[client].Sock = 0;
   Clients[client].Present = false;
-  Clients[client].MyPid = NULL;
+  Clients[client].MyPid = 0;
   pthread_mutex_unlock(&mClients);
 }
 
@@ -109,9 +109,9 @@
           found = true;
           showDebug(1, "client %d pid = %d released by pid\n", next, client_pid);
           shutdown(Clients[next].Sock, SHUT_RDWR);
-          Clients[next].Sock = NULL;
+          Clients[next].Sock = 0;
           Clients[next].Present = false;
-          Clients[next].MyPid = NULL;
+          Clients[next].MyPid = 0;
 	    }
       else
         next++;
