Description: Fix buffer overrun in libXm (FontS.c)
Origin: upstream commit dada5ff7f93cb48afaec2469f5dde24178d8d6da
Bug: http://bugs.motifzone.net/show_bug.cgi?id=1568
Author: Arthur Haas <ahaas@impactweather.com>
Author: Graham Inggs <graham@nerve.org.za>
Last-Update: 2013-02-19
--- a/lib/Xm/FontS.c
+++ b/lib/Xm/FontS.c
@@ -1879,7 +1879,7 @@
 		String params[1];
 		Cardinal num = 1;
 		
-		params[1] = new_font;
+		params[0] = new_font;
 		dbg(); _XmWarningMsg((Widget) fsw, XmNbadXlfdFont,
 			XmNbadXlfdFontMsg, params, num);
 	    }
