From 4ce6308fda83a15d8bf9c82293c76c5747aee199 Mon Sep 17 00:00:00 2001
From: Rinigus <rinigus.git@gmail.com>
Date: Tue, 23 Nov 2021 21:38:40 +0200
Subject: [PATCH] [cmake] Make tools/apikeys optional

---
 poor/CMakeLists.txt | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/poor/CMakeLists.txt b/poor/CMakeLists.txt
index c1bb24b2..e0a1c867 100644
--- a/poor/CMakeLists.txt
+++ b/poor/CMakeLists.txt
@@ -5,6 +5,9 @@ add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/paths.py"
 add_custom_target(sed_default_paths ALL
     DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/paths.py")
 
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/paths.py
+    DESTINATION ${DATADIR}/poor)
+
 add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/config.py"
     COMMAND sed -e \'s|_default_basemap *=.*|_default_basemap = \"${DEFAULT_BASEMAP}\"|g\' ${CMAKE_CURRENT_SOURCE_DIR}/config.py > ${CMAKE_CURRENT_BINARY_DIR}/config.py
     COMMAND sed -i -e \'s|_default_geocoder *=.*|_default_geocoder = \"${DEFAULT_GEOCODER}\"|g\' ${CMAKE_CURRENT_BINARY_DIR}/config.py
@@ -12,19 +15,23 @@ add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/config.py"
     COMMAND sed -i -e \'s|_default_router *=.*|_default_router = \"${DEFAULT_ROUTER}\"|g\' ${CMAKE_CURRENT_BINARY_DIR}/config.py
     COMMAND sed -i -e \'s|_default_profile *=.*|_default_profile = \"${DEFAULT_PROFILE}\"|g\' ${CMAKE_CURRENT_BINARY_DIR}/config.py)
 add_custom_target(sed_default_providers ALL
-    DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/config.py")
+  DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/config.py")
+
+install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/config.py
+    DESTINATION ${DATADIR}/poor)
 
-add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/apikeys.py"
+if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../tools/apikeys.py)
+  add_custom_command(OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/apikeys.py"
     COMMAND cp ${CMAKE_CURRENT_SOURCE_DIR}/../tools/apikeys.py ${CMAKE_CURRENT_BINARY_DIR})
-add_custom_target(inject_keys ALL
+  add_custom_target(inject_keys ALL
     DEPENDS "${CMAKE_CURRENT_BINARY_DIR}/apikeys.py")
 
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/paths.py
-    DESTINATION ${DATADIR}/poor)
-install(FILES
-    ${CMAKE_CURRENT_BINARY_DIR}/config.py
+  install(FILES
     ${CMAKE_CURRENT_BINARY_DIR}/apikeys.py
     DESTINATION ${DATADIR}/poor)
+endif()
+
 
 install(FILES ${CMAKE_CURRENT_SOURCE_DIR}/../thirdparty/astral/astral.py
     DESTINATION ${DATADIR}/poor/astral)
