From: Teemu Ikonen <tpikonen@mailbox.org>
Date: Mon, 22 Jul 2024 14:36:31 +0300
Subject: cmake: Add find_package() call for S2

---
 CMakeLists.txt     | 3 +++
 src/CMakeLists.txt | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 15dd9d2..f1b099f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,6 +115,9 @@ elseif(FLAVOR STREQUAL "silica")
     pkg_search_module(SAILFISH sailfishapp REQUIRED)
 endif()
 
+# s2
+find_package(s2 REQUIRED)
+
 # handle request for running from source dir
 if(RUN_FROM_SOURCE)
     set(DATADIR ${CMAKE_CURRENT_SOURCE_DIR})
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 0291031..35f1a1a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -70,7 +70,7 @@ target_link_libraries(${APP_NAME}
     Qt5::Quick
     Qt5::Positioning
     Qt5::DBus
-    s2)
+    s2::s2)
 
 set(LINKING_FLAVORS
     "kirigami"
