--- mongodb-src-r4.0.0/src/mongo/util/errno_util.cpp.orig
+++ mongodb-src-r4.0.0/src/mongo/util/errno_util.cpp
@@ -58,9 +58,7 @@
     char buf[kBuflen];
     char* msg{nullptr};
 
-#if defined(__GNUC__) && defined(_GNU_SOURCE) && (__ANDROID_API__ > 22)
-    msg = strerror_r(errNumber, buf, kBuflen);
-#elif defined(_WIN32)
+#if defined(_WIN32)
 
     LPWSTR errorText = nullptr;
     FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ALLOCATE_BUFFER |
