diff --git a/kalk/Cargo.toml b/kalk/Cargo.toml
index 99915ea..1d00a9c 100644
--- a/kalk/Cargo.toml
+++ b/kalk/Cargo.toml
@@ -19,4 +19,9 @@ 
 wasm-bindgen = "0.2.69"
+
+[dependencies.gmp-mpfr-sys]
+version = "1.6.2"
+default-features = false
+features = ["use-system-libs"]

 [dev-dependencies]
 wasm-bindgen-test = "0.3.19"
diff --git a/Cargo.lock b/Cargo.lock
index 99915ea..1d00a9c 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -165,6 +165,7 @@
 name = "kalk"
 version = "3.2.0"
 dependencies = [
+ "gmp-mpfr-sys",
  "lazy_static",
  "regex",
  "rug",
