From: Jakub Jirutka <jakub@jirutka.cz>
Date: Tue, 11 Nov 2025 10:53:59 +0100
Subject: [PATCH] Don't install compat53

It's provided by lua5.1-compat53 and lua5.2-compat53 dependency.

diff --git a/GNUmakefile b/GNUmakefile
index 46be0cc..dade806 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -119,16 +119,6 @@ install: all install-config
 	do \
 	   $(INSTALL_DATA) "$$f" '$(DESTDIR)$(luadir)'/`echo $$f | sed 's,^src/,,'`; \
 	done
-ifeq (,$(findstring $(LUA_VERSION),"5.3" "5.4"))
-	find src/compat53/ -type d | while read f; \
-	do \
-	   mkdir -p '$(DESTDIR)$(luadir)'/`echo $$f | sed 's,^src/,,'`; \
-	done
-	find src/compat53/ -type f -name '*.lua' | while read f; \
-	do \
-	   $(INSTALL_DATA) "$$f" '$(DESTDIR)$(luadir)'/`echo $$f | sed 's,^src/,,'`; \
-	done
-endif
 
 install-config:
 	mkdir -p '$(DESTDIR)$(luarocksconfdir)/'
