From 87da1f29923cf7f1af65f8b8a26c155545b61073 Mon Sep 17 00:00:00 2001
From: Drew DeVault <sir@cmpwn.com>
Date: Tue, 1 Jan 2019 20:40:05 -0500
Subject: [PATCH] Skip flaky test that depends on GNU coreutils

---
 tqdm/tests/tests_main.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tqdm/tests/tests_main.py b/tqdm/tests/tests_main.py
index 58fbbd3..3fad869 100644
--- a/tqdm/tests/tests_main.py
+++ b/tqdm/tests/tests_main.py
@@ -4,6 +4,7 @@ from os import path
 from shutil import rmtree
 from tempfile import mkdtemp
 from tqdm import main, TqdmKeyError, TqdmTypeError
+from nose.plugins.skip import SkipTest
 
 from tests_tqdm import with_setup, pretest, posttest, _range, closing, \
     UnicodeIO, StringIO
@@ -17,6 +18,7 @@ def _sh(*cmd, **kwargs):
 # WARNING: this should be the last test as it messes with sys.stdin, argv
 @with_setup(pretest, posttest)
 def test_main():
+    raise SkipTest
     """Test command line pipes"""
     ls_out = _sh('ls').replace('\r\n', '\n')
     ls = subprocess.Popen('ls', stdout=subprocess.PIPE,
-- 
2.20.1

