diff --git a/cmake/GitSubmoduleCheck.cmake b/cmake/GitSubmoduleCheck.cmake
index 49d79b2..b73a6e9 100644
--- a/cmake/GitSubmoduleCheck.cmake
+++ b/cmake/GitSubmoduleCheck.cmake
@@ -1,13 +1,6 @@
-find_package(Git REQUIRED)
-
 # https://gist.github.com/scivision/bb1d47a9529e153617414e91ff5390af
 
 function(git_submodule_check dir)
-    if(NOT EXISTS "${dir}/CMakeLists.txt")
-        execute_process(COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive -- ${dir}
-            WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-            COMMAND_ERROR_IS_FATAL ANY)
-    endif()
 #   Add a Git submodule directory to CMake, assuming the Git submodule directory is a CMake project.
 #   add_subdirectory(${dir} EXCLUDE_FROM_ALL)
 endfunction()
diff --git a/external/st_audiofile/CMakeLists.txt b/external/st_audiofile/CMakeLists.txt
index 0527eed..6361b71 100644
--- a/external/st_audiofile/CMakeLists.txt
+++ b/external/st_audiofile/CMakeLists.txt
@@ -35,8 +35,10 @@ if(APPLE AND PROJECT_SYSTEM_PROCESSOR MATCHES "(arm.*)")
     SET(WAVPACK_ENABLE_ASM OFF)
 endif()
 
-add_subdirectory("thirdparty/wavpack" EXCLUDE_FROM_ALL)
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(WAVPACK REQUIRED wavpack)
 target_link_libraries(st_audiofile_formats PUBLIC wavpack)
+target_include_directories(st_audiofile_formats PUBLIC /usr/include/wavpack)
 
 ###
 if(NOT ST_AUDIO_FILE_USE_SNDFILE)
diff --git a/src/sfizz/modulations/ModMatrix.cpp b/src/sfizz/modulations/ModMatrix.cpp
index 98a92c8..2581509 100644
--- a/src/sfizz/modulations/ModMatrix.cpp
+++ b/src/sfizz/modulations/ModMatrix.cpp
@@ -13,6 +13,7 @@
 #include "SIMDHelpers.h"
 #include "utility/Debug.h"
 #include <absl/container/flat_hash_map.h>
+#include "absl/strings/str_cat.h"
 #include <absl/strings/string_view.h>
 #include <vector>
 #include <algorithm>
diff --git a/src/sfizz/parser/Parser.h b/src/sfizz/parser/Parser.h
index e83fc1a..bbfd8de 100644
--- a/src/sfizz/parser/Parser.h
+++ b/src/sfizz/parser/Parser.h
@@ -10,6 +10,7 @@
 #include <absl/types/optional.h>
 #include <absl/container/flat_hash_map.h>
 #include <absl/container/flat_hash_set.h>
+#include <absl/strings/str_cat.h>
 #include <string>
 #include <memory>
 
