--- cvechecker-3.1/src/cvecheck.c
+++ cvechecker-3.1/src/cvecheck.c
@@ -1255,6 +1255,7 @@
 	}
 
 	cvelist = fopen(arg->cvedata, "r");
+	bzero (buffer, BUFFERSIZE);
 	while (fgets(buffer, sizeof(buffer), cvelist) != 0) {
 		int cvelength = 0;
 
@@ -1302,6 +1303,7 @@
 				mysql_dbimpl_store_cve_in_db_checkpoint(ws);
 		};
 		linenum++;
+		bzero (buffer, BUFFERSIZE);
 	};
 
 	if (ws->dbtype == sqlite)
