diff --git a/app/kuma-dp/pkg/config/validate_test.go b/app/kuma-dp/pkg/config/validate_test.go
index 1ba0b15ab8..9138791491 100644
--- a/app/kuma-dp/pkg/config/validate_test.go
+++ b/app/kuma-dp/pkg/config/validate_test.go
@@ -58,7 +58,7 @@ var _ = Describe("ValidateTokenPath", func() {
 			}),
 			Entry("can't parse token", testCase{
 				token:         "yJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJOYW1lIjoidGVzdCIsIk1lc2giOiJkZWZhdWx0IiwiVGFncyI6e30sIlR5cGUiOiIifQ.rdQ6l_6hzT93Kbk9kO-kZYY7BaexUH8QknvbdRy_f6s",
-				expectedError: "not valid JWT token. Can't parse it.: token is malformed: could not JSON decode header: invalid character 'È' looking for beginning of value",
+				expectedError: "not valid JWT token. Can't parse it.: token is malformed: could not JSON decode header: invalid character 'Ș' looking for beginning of value",
 			}),
 			Entry("need 3 segments", testCase{
 				token:         "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJOYW1lIjoidGVzdCIsIk1lc2giOiJkZWZhdWx0IiwiVGFncyI6e30sIlR5cGUiOiIifQ",
diff --git a/app/kumactl/cmd/apply/apply_test.go b/app/kumactl/cmd/apply/apply_test.go
index 6453c40e1a..aee8640944 100644
--- a/app/kumactl/cmd/apply/apply_test.go
+++ b/app/kumactl/cmd/apply/apply_test.go
@@ -416,7 +416,7 @@ mesh: default
 networking:
   inbound: 0 # should be a string
 `,
-			err: `resource[0]: failed to parse resource: invalid Dataplane object: "error unmarshaling JSON: while decoding JSON: json: cannot unmarshal number into Go value of type []json.RawMessage"`,
+			err: `resource[0]: failed to parse resource: invalid Dataplane object: "error unmarshaling JSON: while decoding JSON: json: cannot unmarshal number into Go value of type []jsontext.Value"`,
 		}),
 		Entry("no resource", testCase{
 			resource: ``,
