--- a/safe_yaml.gemspec
+++ b/safe_yaml.gemspec
@@ -10,7 +10,7 @@
   gem.summary       = %q{SameYAML provides an alternative implementation of YAML.load suitable for accepting user input in Ruby applications.}
   gem.homepage      = "https://github.com/dtao/safe_yaml"
   gem.license       = "MIT"
-  gem.files         = `git ls-files`.split($\)
+  gem.files         = Dir["lib/**/*"]
   gem.test_files    = gem.files.grep(%r{^spec/})
   gem.require_paths = ["lib"]
   gem.executables   = ["safe_yaml"]
