diff --git a/configure.ac b/configure.ac
index 6a7a1a5..c4cc84c 100755
--- a/configure.ac
+++ b/configure.ac
@@ -149,7 +149,7 @@ if test "$with_cpp" = "yes";  then
     AC_SUBST([BOOST_FILESYSTEM_LDADD], [$(echo "$BOOST_LIB_DIR/libboost_filesystem.a")])
     AC_SUBST([BOOST_SYSTEM_LDADD], [$(echo "$BOOST_LIB_DIR/libboost_system.a")])
     AC_SUBST([BOOST_TEST_LDADD], [$(echo "$BOOST_LIB_DIR/libboost_unit_test_framework.a")])
-    AC_SUBST([BOOST_THREAD_LDADD], [$(echo "$BOOST_LIB_DIR/libboost_thread.a")])
+    AC_SUBST([BOOST_THREAD_LDADD], [$(echo "$BOOST_LIB_DIR/libboost_thread-mt.a")])
     have_cpp="yes"
   fi
 
@@ -760,7 +760,7 @@ AC_ARG_ENABLE(boostthreads,
 if test "x[$]ENABLE_BOOSTTHREADS" = "x1"; then
   AC_MSG_WARN(enable boostthreads)
   AC_DEFINE([USE_BOOST_THREAD], [1], [experimental --enable-boostthreads that replaces POSIX pthread by boost::thread])
-  LIBS="-lboost_thread $LIBS"
+  LIBS="-lboost_thread-mt $LIBS"
 fi
 
 AM_CONDITIONAL([WITH_BOOSTTHREADS], [test "x[$]ENABLE_BOOSTTHREADS" = "x1"])

