diff --git a/lib/cpp/Makefile.am b/lib/cpp/Makefile.am
index 83ccd9b6..c6f2ff54 100755
--- a/lib/cpp/Makefile.am
+++ b/lib/cpp/Makefile.am
@@ -145,14 +145,18 @@ libthriftqt5_la_CPPFLAGS = $(AM_CPPFLAGS) $(QT5_CFLAGS)
 if QT5_REDUCE_RELOCATIONS
 libthriftqt5_la_CPPFLAGS += -fPIC
 endif
-libthriftnb_la_CXXFLAGS = $(AM_CXXFLAGS)
-libthriftz_la_CXXFLAGS  = $(AM_CXXFLAGS)
+libthriftnb_la_CXXFLAGS = $(AM_CXXFLAGS) -L/usr/lib -levent
+libthriftz_la_CXXFLAGS  = $(AM_CXXFLAGS) -L/lib -lz
 libthriftqt_la_CXXFLAGS  = $(AM_CXXFLAGS)
 libthriftqt5_la_CXXFLAGS  = $(AM_CXXFLAGS)
 libthriftnb_la_LDFLAGS  = -release $(VERSION) $(BOOST_LDFLAGS)
 libthriftz_la_LDFLAGS   = -release $(VERSION) $(BOOST_LDFLAGS) $(ZLIB_LIBS)
 libthriftqt_la_LDFLAGS   = -release $(VERSION) $(BOOST_LDFLAGS) $(QT_LIBS)
 libthriftqt5_la_LDFLAGS   = -release $(VERSION) $(BOOST_LDFLAGS) $(QT5_LIBS)
+libthriftnb_la_LIBADD  = libthrift.la
+libthriftz_la_LIBADD   = libthrift.la
+libthriftqt_la_LIBADD   = libthrift.la
+libthriftqt5_la_LIBADD   = libthrift.la
 
 include_thriftdir = $(includedir)/thrift
 include_thrift_HEADERS = \
