diff --git a/lib/cpp/test/Makefile.am b/lib/cpp/test/Makefile.am
index c298e26..a56889c 100755
--- a/lib/cpp/test/Makefile.am
+++ b/lib/cpp/test/Makefile.am
@@ -104,9 +104,7 @@ check_PROGRAMS += \
 endif
 
 TESTS_ENVIRONMENT= \
-	BOOST_TEST_LOG_SINK=tests.xml \
-	BOOST_TEST_LOG_LEVEL=test_suite \
-	BOOST_TEST_LOG_FORMAT=XML
+	BOOST_TEST_LOGGER=XML,test_suite,tests.xml
 
 TESTS = \
 	$(check_PROGRAMS)
