diff --git a/cpu/ext/ffmpeg-codecs/CMakeLists.txt b/cpu/ext/ffmpeg-codecs/CMakeLists.txt
index 7bde28b..ffd138d 100644
--- a/cpu/ext/ffmpeg-codecs/CMakeLists.txt
+++ b/cpu/ext/ffmpeg-codecs/CMakeLists.txt
@@ -28,19 +28,13 @@ set(TARGET ffmpeg-codecs)
 add_library(${TARGET} INTERFACE)
 
 # Set VPL dependencies directory
-if(NOT DEFINED ENV{VPL_BUILD_DEPENDENCIES})
-  set(VPL_DEP_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../../_deps)
-  message(STATUS "Using default destination located at ${VPL_DEP_DIR}")
-else()
-  set(VPL_DEP_DIR $ENV{VPL_BUILD_DEPENDENCIES})
-  message(STATUS "Using VPL_BUILD_DEPENDENCIES located at ${VPL_DEP_DIR}")
-endif()
+set(VPL_DEP_DIR "/usr")
 
 # Set basic FFmpeg and codec libs
 if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
-  set(SVTHEVCENC_LIB ${VPL_DEP_DIR}/lib/libSvtHevcEnc.a)
-  set(SVTAV1ENC_LIB ${VPL_DEP_DIR}/lib/libSvtAv1Enc.a)
-  set(DAV1D_LIB ${VPL_DEP_DIR}/lib/libdav1d.a)
+  set(SVTHEVCENC_LIB ${VPL_DEP_DIR}/lib/libSvtHevcEnc.so)
+  set(SVTAV1ENC_LIB ${VPL_DEP_DIR}/lib/libSvtAv1Enc.so)
+  set(DAV1D_LIB ${VPL_DEP_DIR}/lib/libdav1d.so)
 
   if(NOT EXISTS ${SVTHEVCENC_LIB}
      OR NOT EXISTS ${SVTAV1ENC_LIB}
@@ -49,10 +43,10 @@ if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "8")
   endif()
 endif()
 
-set(AVCODEC_LIB ${VPL_DEP_DIR}/lib/libavcodec.a)
-set(AVUTIL_LIB ${VPL_DEP_DIR}/lib/libavutil.a)
-set(AVFILTER_LIB ${VPL_DEP_DIR}/lib/libavfilter.a)
-set(SWSCALE_LIB ${VPL_DEP_DIR}/lib/libswscale.a)
+set(AVCODEC_LIB ${VPL_DEP_DIR}/lib/libavcodec.so)
+set(AVUTIL_LIB ${VPL_DEP_DIR}/lib/libavutil.so)
+set(AVFILTER_LIB ${VPL_DEP_DIR}/lib/libavfilter.so)
+set(SWSCALE_LIB ${VPL_DEP_DIR}/lib/libswscale.so)
 
 if(NOT EXISTS ${AVCODEC_LIB}
    OR NOT EXISTS ${AVUTIL_LIB}
@@ -135,7 +129,7 @@ endif()
 
 # Set AVC encoder lib name
 if(BUILD_GPL_X264)
-  set(H264_ENC_LIB ${VPL_DEP_DIR}/lib/libx264.a)
+  set(H264_ENC_LIB ${VPL_DEP_DIR}/lib/libx264.so)
   if(NOT EXISTS ${H264_ENC_LIB})
     message(FATAL_ERROR "Could not find x264 libraries")
   else()
