diff --git a/t/original.t b/t/original.t
index bba3fa0..2765eba 100755
--- a/t/original.t
+++ b/t/original.t
@@ -14,7 +14,8 @@ $mail{To}   = 'Sendmail Test <sendmail@alma.ch>';
 
 # if you want to get a copy of the test mail, you need to specify your
 # own server here, by name or IP address
-$server = 'mail.alma.ch';
+$server = '127.0.0.1';
+$port = 2525;
 #$server = 'my.usual.mail.server';
 
 BEGIN { $| = 1; print "1..2\n"; }
@@ -42,6 +43,7 @@ print "ok 1\n";
 
 if ($server) {
     $mail{Smtp} = $server;
+    $mail{Port} = $port;
     print "Server set to: $server\n";
 }
 
