diff --git a/meson.build b/meson.build
index d917839..44b210f 100644
--- a/meson.build
+++ b/meson.build
@@ -185,7 +185,7 @@ pkgconf.set('pkgincludedir', '${prefix}/@0@'.format(get_option('includedir')) +
 pkgconf.set('VMAJ', v_maj)
 pkgconf.set('PACKAGE_VERSION', meson.project_version())
 pkgconf.set('requirements_libdwarf_pc', 'zlib')
-pkgconf.set('requirements_libdwarf_pc', 'zstd')
+pkgconf.set('requirements_libdwarf_pc', 'libzstd')
 pkgconf.set('requirements_libdwarf_libs', '')
 
 pkg_install_dir = '@0@/pkgconfig'.format(get_option('libdir'))
diff --git a/src/lib/libdwarf/meson.build b/src/lib/libdwarf/meson.build
index 8eb103a..468fec2 100644
--- a/src/lib/libdwarf/meson.build
+++ b/src/lib/libdwarf/meson.build
@@ -64,15 +64,15 @@ libdwarf_src = [
 ]
 
 zlib_deps = dependency('zlib', method: 'pkg-config', required: false)
-zstd_deps = dependency('zstd', method: 'pkg-config', required: false)
+zstd_deps = dependency('libzstd', method: 'pkg-config', required: false)
 
 if zlib_deps.found() == true
-  config_h.set10('HAVE_ZLIB', true)
-  config_h.set10('HAVE_ZLIB_H', true)
+  config_h.set('HAVE_ZLIB', true)
+  config_h.set('HAVE_ZLIB_H', true)
 endif
 if zstd_deps.found() == true
-  config_h.set11('HAVE_ZSTD', true)
-  config_h.set11('HAVE_ZSTD_H', true)
+  config_h.set('HAVE_ZSTD', true)
+  config_h.set('HAVE_ZSTD_H', true)
 endif
 
 libdwarf_lib = library('dwarf', libdwarf_src,
