diff -Nurp a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2024-07-27 20:23:55.379291046 +0000
+++ b/src/CMakeLists.txt	2024-07-27 20:25:46.359819714 +0000
@@ -367,41 +367,6 @@ set_target_properties(Boost::MPL PROPERT
     "BOOST_MPL_CFG_NO_PREPROCESSED_HEADERS;BOOST_MPL_LIMIT_LIST_SIZE=30")
 
 if(WITH_SEASTAR)
-  find_package(c-ares 1.13.0 QUIET)
-  if(NOT c-ares_FOUND)
-    message(STATUS "Could not find c-ares, will build it")
-    include(Buildc-ares)
-    build_c_ares()
-  endif()
-  macro(find_package name)
-    if(NOT TARGET ${name})
-      _find_package(${ARGV})
-    endif()
-  endmacro ()
-  set(Seastar_API_LEVEL "6" CACHE STRING "" FORCE)
-  set(Seastar_HWLOC OFF CACHE BOOL "" FORCE)
-  set(Seastar_STD_OPTIONAL_VARIANT_STRINGVIEW ON CACHE BOOL "" FORCE)
-  if(Seastar_DPDK)
-    find_package(dpdk QUIET)
-    if(NOT DPDK_FOUND)
-      include(BuildDPDK)
-      build_dpdk(${CMAKE_BINARY_DIR}/src/dpdk)
-    endif()
-  endif()
-  list(APPEND Seastar_CXX_FLAGS
-    "-DSEASTAR_NO_EXCEPTION_HACK"
-    "-Wno-error"
-    "-Wno-sign-compare"
-    "-Wno-attributes"
-    "-Wno-pessimizing-move"
-    "-Wno-address-of-packed-member"
-    "-Wno-non-virtual-dtor")
-  set(Seastar_CXX_FLAGS "${Seastar_CXX_FLAGS}" CACHE STRING "" FORCE)
-  add_subdirectory(seastar)
-  # create the directory so cmake won't complain when looking at the imported
-  # target: Seastar exports this directory created at build-time
-  file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/seastar/gen/include")
-  file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/seastar/gen/src")
   add_subdirectory(crimson)
 endif()
 
diff -Nurp a/src/crimson/CMakeLists.txt b/src/crimson/CMakeLists.txt
--- a/src/crimson/CMakeLists.txt	2024-07-27 20:23:55.380291060 +0000
+++ b/src/crimson/CMakeLists.txt	2024-07-27 20:24:14.390552917 +0000
@@ -6,10 +6,11 @@ if((CMAKE_CXX_COMPILER_ID STREQUAL GNU A
  list(APPEND crimson_cflag_definitions
    "BOOST_ASIO_DISABLE_CONCEPTS")
 endif()
+pkg_check_modules(SEASTAR IMPORTED_TARGET seastar)
 set_target_properties(crimson::cflags PROPERTIES
   INTERFACE_COMPILE_DEFINITIONS "${crimson_cflag_definitions}"
   INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:-Wno-non-virtual-dtor>
-  INTERFACE_LINK_LIBRARIES Seastar::seastar)
+  INTERFACE_LINK_LIBRARIES PkgConfig::SEASTAR)
 
 set(crimson_common_srcs
   common/assert.cc
diff -Nurp a/src/crimson/os/alienstore/CMakeLists.txt b/src/crimson/os/alienstore/CMakeLists.txt
--- a/src/crimson/os/alienstore/CMakeLists.txt	2024-07-27 20:23:55.382291087 +0000
+++ b/src/crimson/os/alienstore/CMakeLists.txt	2024-07-27 20:24:14.390552917 +0000
@@ -2,8 +2,7 @@ include_directories(SYSTEM "${CMAKE_SOUR
 
 add_library(alien::cflags INTERFACE IMPORTED)
 set_target_properties(alien::cflags PROPERTIES
-  INTERFACE_COMPILE_DEFINITIONS "WITH_SEASTAR;WITH_ALIEN"
-  INTERFACE_INCLUDE_DIRECTORIES $<TARGET_PROPERTY:Seastar::seastar,INTERFACE_INCLUDE_DIRECTORIES>)
+  INTERFACE_COMPILE_DEFINITIONS "WITH_SEASTAR;WITH_ALIEN")
 
 set(crimson_alien_common_srcs
   ${PROJECT_SOURCE_DIR}/src/common/admin_socket.cc
