diff -Nurp a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2024-07-27 20:20:32.740499875 +0000
+++ b/src/CMakeLists.txt	2024-07-27 20:20:52.206768005 +0000
@@ -414,7 +414,6 @@ set(libcommon_files
   ${CMAKE_BINARY_DIR}/src/include/ceph_ver.h
   ceph_ver.c
   global/global_context.cc
-  xxHash/xxhash.c
   common/error_code.cc
   common/tracer.cc
   log/Log.cc
@@ -449,6 +448,8 @@ add_library(common-objs OBJECT ${libcomm
 target_compile_definitions(common-objs PRIVATE
   $<TARGET_PROPERTY:${FMT_LIB},INTERFACE_COMPILE_DEFINITIONS>)
 add_dependencies(common-objs legacy-option-headers)
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
+target_link_libraries(common-objs PkgConfig::XXHASH)
 
 if(WITH_JAEGER)
   add_dependencies(common-objs jaeger_base)
diff -Nurp a/src/common/Checksummer.h b/src/common/Checksummer.h
--- a/src/common/Checksummer.h	2024-07-27 20:20:32.742499903 +0000
+++ b/src/common/Checksummer.h	2024-07-27 20:20:52.206768005 +0000
@@ -8,7 +8,7 @@
 #include "include/byteorder.h"
 #include "include/ceph_assert.h"
 
-#include "xxHash/xxhash.h"
+#include "xxhash.h"
 
 class Checksummer {
 public:
diff -Nurp a/src/crimson/CMakeLists.txt b/src/crimson/CMakeLists.txt
--- a/src/crimson/CMakeLists.txt	2024-07-27 20:20:32.746499958 +0000
+++ b/src/crimson/CMakeLists.txt	2024-07-27 20:21:52.466598032 +0000
@@ -122,7 +122,6 @@ add_library(crimson-common STATIC
   ${PROJECT_SOURCE_DIR}/src/osd/OSDMap.cc
   ${PROJECT_SOURCE_DIR}/src/osd/PGPeeringEvent.cc
   ${PROJECT_SOURCE_DIR}/src/common/scrub_types.cc
-  ${PROJECT_SOURCE_DIR}/src/xxHash/xxhash.c
   ${crimson_common_srcs}
   $<TARGET_OBJECTS:common_mountcephfs_objs>
   $<TARGET_OBJECTS:common-options-objs>)
@@ -146,10 +145,12 @@ if(NOT WITH_SYSTEM_BOOST)
   list(APPEND crimson_common_deps ${ZLIB_LIBRARIES})
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 target_link_libraries(crimson-common
   PUBLIC
     ${crimson_common_public_deps}
   PRIVATE
+    PkgConfig::XXHASH
     crc32 arch
     ${crimson_common_deps}
     OpenSSL::Crypto)
diff -Nurp a/src/os/CMakeLists.txt b/src/os/CMakeLists.txt
--- a/src/os/CMakeLists.txt	2024-07-27 20:20:32.744499930 +0000
+++ b/src/os/CMakeLists.txt	2024-07-27 20:20:52.207768019 +0000
@@ -46,8 +46,9 @@ if(HAVE_LIBZFS)
   list(APPEND libos_srcs $<TARGET_OBJECTS:os_zfs_objs>)
 endif()
 
+pkg_check_modules(XXHASH IMPORTED_TARGET libxxhash)
 add_library(os STATIC ${libos_srcs})
-target_link_libraries(os blk)
+target_link_libraries(os blk PkgConfig::XXHASH)
 
 target_link_libraries(os heap_profiler kv)
 
