diff -Nurp a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-07-28 07:11:18.290388744 +0000
+++ b/CMakeLists.txt	2024-07-28 07:13:12.491899849 +0000
@@ -666,7 +666,7 @@ option(WITH_SYSTEM_BOOST "require and bu
 # Boost::thread depends on Boost::atomic, so list it explicitly.
 set(BOOST_COMPONENTS
   atomic chrono thread system regex random program_options date_time
-  iostreams context coroutine)
+  iostreams context coroutine url)
 set(BOOST_HEADER_COMPONENTS container)
 
 if(WITH_MGR)
diff -Nurp a/src/mds/BoostUrlImpl.cc b/src/mds/BoostUrlImpl.cc
--- a/src/mds/BoostUrlImpl.cc	2024-07-28 07:12:56.769692199 +0000
+++ b/src/mds/BoostUrlImpl.cc	1970-01-01 00:00:00.000000000 +0000
@@ -1,8 +0,0 @@
-/*
- * https://www.boost.org/doc/libs/1_82_0/libs/url/doc/html/url/overview.html#url.overview.requirements
- *
- * To use the library as header-only; that is, to eliminate the requirement 
- * to link a program to a static or dynamic Boost.URL library, 
- * simply place the following line in exactly one source file in your project.
- */
-#include <boost/url/src.hpp>
diff -Nurp a/src/mds/CMakeLists.txt b/src/mds/CMakeLists.txt
--- a/src/mds/CMakeLists.txt	2024-07-28 07:11:18.298388851 +0000
+++ b/src/mds/CMakeLists.txt	2024-07-28 07:14:59.747316408 +0000
@@ -45,12 +45,12 @@ set(mds_srcs
   QuiesceDbManager.cc
   QuiesceAgent.cc
   MDSRankQuiesce.cc
-  BoostUrlImpl.cc
   ${CMAKE_SOURCE_DIR}/src/common/TrackedOp.cc
   ${CMAKE_SOURCE_DIR}/src/common/MemoryModel.cc
   ${CMAKE_SOURCE_DIR}/src/osdc/Journaler.cc
   ${CMAKE_SOURCE_DIR}/src/mgr/MDSPerfMetricTypes.cc)
 add_library(mds STATIC ${mds_srcs})
 target_link_libraries(mds PRIVATE
-  heap_profiler cpu_profiler osdc ${LUA_LIBRARIES})
+  heap_profiler cpu_profiler osdc ${LUA_LIBRARIES}
+  Boost::url)
 target_include_directories(mds PRIVATE "${LUA_INCLUDE_DIR}")
diff -Nurp a/src/test/mds/CMakeLists.txt b/src/test/mds/CMakeLists.txt
--- a/src/test/mds/CMakeLists.txt	2024-07-28 07:11:18.303388917 +0000
+++ b/src/test/mds/CMakeLists.txt	2024-07-28 07:13:12.492899862 +0000
@@ -18,11 +18,10 @@ target_link_libraries(unittest_mds_sessi
 add_executable(unittest_mds_quiesce_db
   TestQuiesceDb.cc
   ../../../src/mds/QuiesceDbManager.cc
-  ../../../src/mds/BoostUrlImpl.cc
   $<TARGET_OBJECTS:unit-main>
 )
 add_ceph_unittest(unittest_mds_quiesce_db)
-target_link_libraries(unittest_mds_quiesce_db ceph-common global)
+target_link_libraries(unittest_mds_quiesce_db ceph-common global Boost::url)
 
 # unittest_mds_quiesce_agent
 add_executable(unittest_mds_quiesce_agent
