From 188612538ea06890d27321a5550c93a817ea9bc2 Mon Sep 17 00:00:00 2001
From: Guna K Kambalimath <Guna.Kambalimath@ibm.com>
Date: Tue, 20 Aug 2024 15:36:52 +0530
Subject: [PATCH] Fixes build failure on IBM Power

---
 src/common/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index c3fd51d26468f..8b9f3339e3885 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -225,6 +225,7 @@ elseif(HAVE_POWER8)
     list(APPEND crc32_srcs
       crc32c_ppc_asm.S
       crc32c_ppc_fast_zero_asm.S)
+    set_source_files_properties(crc32c_ppc_asm.S PROPERTIES COMPILE_FLAGS -D__ASSEMBLY__)
   endif(HAVE_PPC64LE)
 elseif(HAVE_ARMV8_CRC)
   list(APPEND crc32_srcs
From 83782c9ba033ce9008df287281f1c66dbf8ee28e Mon Sep 17 00:00:00 2001
From: Guna K Kambalimath <Guna.Kambalimath@ibm.com>
Date: Tue, 20 Aug 2024 17:27:44 +0530
Subject: [PATCH] Remove dynamic linking for power

---
 src/arch/CMakeLists.txt   | 6 +++++-
 src/common/CMakeLists.txt | 6 +++++-
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/arch/CMakeLists.txt b/src/arch/CMakeLists.txt
index e95d9bbb81fa8..d0e635dcd65e6 100644
--- a/src/arch/CMakeLists.txt
+++ b/src/arch/CMakeLists.txt
@@ -11,4 +11,8 @@ elseif(HAVE_S390X)
   list(APPEND arch_srcs s390x.c)
 endif()
 
-add_library(arch OBJECT ${arch_srcs})
+if(HAVE_POWER8)
+  add_library(arch STATIC ${arch_srcs})
+else()
+  add_library(arch OBJECT ${arch_srcs})
+endif()
\ No newline at end of file
diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 8b9f3339e3885..6cdbb169f2105 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -237,7 +237,11 @@ elseif(HAVE_S390X)
   )
 endif(HAVE_INTEL)
 
-add_library(crc32 OBJECT ${crc32_srcs})
+if(HAVE_POWER8)
+  add_library(crc32 STATIC ${crc32_srcs})
+else()
+  add_library(crc32 OBJECT ${crc32_srcs})
+endif()
 
 if(HAVE_ARMV8_CRC)
   set_target_properties(crc32 PROPERTIES
From 0efec175a0abc764a4aa883f27d2c13488f668c6 Mon Sep 17 00:00:00 2001
From: Guna K Kambalimath <Guna.Kambalimath@ibm.com>
Date: Tue, 20 Aug 2024 17:38:54 +0530
Subject: [PATCH] Removing set properties

---
 src/common/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/CMakeLists.txt b/src/common/CMakeLists.txt
index 6cdbb169f2105..b25b4bff570a3 100644
--- a/src/common/CMakeLists.txt
+++ b/src/common/CMakeLists.txt
@@ -225,7 +225,7 @@ elseif(HAVE_POWER8)
     list(APPEND crc32_srcs
       crc32c_ppc_asm.S
       crc32c_ppc_fast_zero_asm.S)
-    set_source_files_properties(crc32c_ppc_asm.S PROPERTIES COMPILE_FLAGS -D__ASSEMBLY__)
+    #set_source_files_properties(crc32c_ppc_asm.S PROPERTIES COMPILE_FLAGS -D__ASSEMBLY__)
   endif(HAVE_PPC64LE)
 elseif(HAVE_ARMV8_CRC)
   list(APPEND crc32_srcs
