From d731a9d7511f5cbdb9e2228dda992b396557bc61 Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 8 Jun 2024 02:27:53 +0200
Subject: [PATCH] zig 0.13: use std.Build.path

---
 build.zig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build.zig b/build.zig
index fe1f95e..38560b5 100644
--- a/build.zig
+++ b/build.zig
@@ -32,7 +32,7 @@ pub fn build(b: *std.Build) void {
 
     const exe = b.addExecutable(.{
         .name = "aercbook",
-        .root_source_file = .{ .path = "src/main.zig" },
+        .root_source_file = b.path("src/main.zig"),
         .target = target,
         .optimize = optimize,
     });
@@ -50,7 +50,7 @@ pub fn build(b: *std.Build) void {
     // Creates a step for unit testing. This only builds the test executable
     // but does not run it.
     const unit_tests = b.addTest(.{
-        .root_source_file = .{ .path = "src/email_iterator.zig" },
+        .root_source_file = b.path("src/email_iterator.zig"),
         .target = target,
         .optimize = optimize,
     });
