From fa4ed8b62c5a904da70ff0b43f95a7eb3e4588ad Mon Sep 17 00:00:00 2001
From: "lauren n. liberda" <lauren@selfisekai.rocks>
Date: Sat, 8 Jun 2024 02:29:05 +0200
Subject: [PATCH] zig 0.13: std.ChildProcess -> std.process.Child

---
 src/gitversiontag.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gitversiontag.zig b/src/gitversiontag.zig
index 08480a7..0b0fa7c 100644
--- a/src/gitversiontag.zig
+++ b/src/gitversiontag.zig
@@ -6,7 +6,7 @@ pub fn gitVersionTag(a: std.mem.Allocator) []const u8 {
         "--sort=-creatordate",
     };
 
-    if (std.ChildProcess.run(.{ .argv = args[0..], .allocator = a })) |ret| {
+    if (std.process.Child.run(.{ .argv = args[0..], .allocator = a })) |ret| {
         var it = std.mem.split(u8, ret.stdout, "\n");
         if (it.next()) |firstline| {
             return firstline;
