From c8a4e324581d790aa38a619bcb2fc0d773ae9d1c Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sun, 9 Mar 2025 20:40:35 +0100
Subject: [PATCH] Zig 0.14: Add .zig-cache to .gitignore

---
 .gitignore | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index 48af762..35972ea 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,4 +1,5 @@
 zig-out/
 zig-cache/
+.zig-cache/
 src/version.zig
 aerc-releases/
