From a93013f19da3d74317e000c4c694654bfe721d9a Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sun, 9 Mar 2025 20:50:24 +0100
Subject: [PATCH] Zig 0.14: std.mem.tokenize -> std.mem.tokenizeSequence

---
 src/main.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.zig b/src/main.zig
index b0394d8..2394207 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -248,7 +248,7 @@ fn parseAddresses(
     var it = emailIterator.init(buf);
     while (it.next()) |addr| {
         // split into parts separated by whitespace
-        var t_it = std.mem.tokenize(u8, addr, " \t\n\r");
+        var t_it = std.mem.tokenizeSequence(u8, addr, " \t\n\r");
         var parts = std.ArrayList([]const u8).init(a);
         while (t_it.next()) |part| {
             try parts.append(part);
