From 6db2f5c53f3d739cbc948f05d35fce68672f5ee7 Mon Sep 17 00:00:00 2001
From: Robert Mongold <rtmongold@gmail.com>
Date: Mon, 27 Apr 2026 08:21:00 -0400
Subject: [PATCH] Zig 0.15: fix stdin reader API

---
 src/main.zig | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/main.zig b/src/main.zig
index 0e7bf7a..f44aab3 100644
--- a/src/main.zig
+++ b/src/main.zig
@@ -275,12 +275,9 @@ fn parseMailFromStdin(alloc: std.mem.Allocator) !ParseMailResult {
     const stdin = std.fs.File.stdin();
 
-    var temp_buffer: [1024 * 1024]u8 = undefined;
-
     // read the 1st megabyte - we're interested in the header only
     var buffer: [1024 * 1024]u8 = undefined;
 
-    var reader = stdin.reader(&temp_buffer);
-    const howmany = try reader.read(&buffer);
+    const howmany = try stdin.read(&buffer);
     if (howmany <= 0) return error.ReadError;
 
     var ret = try ParseMailResult.init(alloc);
     