diff --git a/lib/cpp/test/TNonblockingSSLServerTest.cpp b/lib/cpp/test/TNonblockingSSLServerTest.cpp
index 3e9700f..b1ed162 100644
--- a/lib/cpp/test/TNonblockingSSLServerTest.cpp
+++ b/lib/cpp/test/TNonblockingSSLServerTest.cpp
@@ -31,6 +31,7 @@
 #include "gen-cpp/ParentService.h"
 
 #include <event.h>
+#include <signal.h>
 
 using namespace apache::thrift;
 using apache::thrift::concurrency::Guard;

