diff --git a/lib/lua/Makefile.am b/lib/lua/Makefile.am
index 3dfc27c8..b6aa0fa1 100644
--- a/lib/lua/Makefile.am
+++ b/lib/lua/Makefile.am
@@ -22,10 +22,10 @@ AUTOMAKE_OPTIONS = subdir-objects
 SUBDIRS = .
 
 lib_LTLIBRARIES = \
+			liblualongnumber.la \
 			libluasocket.la \
 			libluabpack.la \
-			libluabitwise.la \
-			liblualongnumber.la
+			libluabitwise.la
 
 libluasocket_la_SOURCES = \
 			src/luasocket.c \
