diff --git a/test/py/Makefile.am b/test/py/Makefile.am
index b3513dd..3b0d9ed 100644
--- a/test/py/Makefile.am
+++ b/test/py/Makefile.am
@@ -48,8 +48,8 @@ precross: $(thrift_gen)
 BUILT_SOURCES = $(thrift_gen)
 
 helper_scripts=                                 \
-        TestClient.py                           \
-        TestServer.py
+        TestClient.py
+#        TestServer.py
 
 check_SCRIPTS=                                  \
         $(thrift_gen) \
