diff --git a/test/py/RunClientServer.py b/test/py/RunClientServer.py
index 7c0f787..1ce55ea 100755
--- a/test/py/RunClientServer.py
+++ b/test/py/RunClientServer.py
@@ -66,9 +66,9 @@ def default_servers():
         'TNonblockingServer',
         'THttpServer',
     ]
-    if platform.system() != 'Windows':
-        servers.append('TProcessPoolServer')
-        servers.append('TForkingServer')
+#    if platform.system() != 'Windows':
+#        servers.append('TProcessPoolServer')
+#        servers.append('TForkingServer')
     return servers
 
 
