From 6204427f8b8e3d8c72b669717a3f129ffae401d9 Mon Sep 17 00:00:00 2001
From: Gabriele Musco <gabmus@disroot.org>
Date: Tue, 31 Aug 2021 09:07:14 +0200
Subject: [PATCH] updated gtk4 and libadwaita dependencies

---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1eb2a91..62efb39 100644
--- a/meson.build
+++ b/meson.build
@@ -75,7 +75,8 @@ if not py_installation.found()
 endif
 dependency('glib-2.0')
 dependency('gobject-introspection-1.0', version: '>=1.35.9')
-dependency('gtk+-3.0', version :'>=3.24')
+dependency('gtk4', version :'>=4.4.0')
+dependency('libadwaita-1', version :'>=1.0.0')
 
 gnome = import('gnome')
 prefix = get_option('prefix') # should be /usr
-- 
GitLab

