From 4ab5ed57a3b53e0271cf50f256cd7904c50cac55 Mon Sep 17 00:00:00 2001
From: Douglas Jerome <douglas@ttylinux.org>
Date: Wed, 15 May 2013 14:33:31 +0200
Subject: [PATCH] Makefile: respect ${DESTDIR} in tic invocation

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ff85deb..eb849f4 100644
--- a/Makefile
+++ b/Makefile
@@ -55,7 +55,7 @@ install: dvtm
 	@sed "s/VERSION/${VERSION}/g" < dvtm.1 > ${DESTDIR}${MANPREFIX}/man1/dvtm.1
 	@chmod 644 ${DESTDIR}${MANPREFIX}/man1/dvtm.1
 	@echo installing terminfo description
-	@tic -s dvtm.info
+	@tic -o ${DESTDIR}/usr/share/terminfo -s dvtm.info
 
 uninstall:
 	@echo removing executable file from ${DESTDIR}${PREFIX}/bin
-- 
1.8.3.450.gf3f2a46

