--- ./src/backup_phase2_server.c.orig
+++ ./src/backup_phase2_server.c
@@ -483,7 +483,11 @@
 	free(curpath);
 
 	blocklen=get_librsync_block_len(cb->endfile);
+#ifdef RS_DEFAULT_STRONG_LEN
 	if(!(p1b->sigjob=rs_sig_begin(blocklen, RS_DEFAULT_STRONG_LEN)))
+#else
+	if(!(p1b->sigjob=rs_sig_begin(blocklen, 8, RS_MD4_SIG_MAGIC)))
+#endif
 	{
 		logp("could not start signature job.\n");
 		return -1;
--- ./src/backup_phase4_server.c.orig
+++ ./src/backup_phase4_server.c
@@ -38,7 +38,7 @@
 	}
 	result=rs_sig_gzfile(dstfp, dstzp, sigp,
 		get_librsync_block_len(endfile),
-		RS_DEFAULT_STRONG_LEN, NULL, cntr);
+		8, NULL, cntr);
 	gzclose_fp(&dstzp);
 	close_fp(&dstfp);
 	if(close_fp(&sigp))
--- ./src/rs_buf.c.orig
+++ ./src/rs_buf.c
@@ -497,7 +497,7 @@
     rs_job_t        *job;
     rs_result       r;
 
-    job = rs_sig_begin(new_block_len, strong_len);
+    job = rs_sig_begin(new_block_len, strong_len, RS_MD4_SIG_MAGIC);
     r = rs_whole_gzrun(job, old_file, old_zfile, sig_file, NULL, cntr);
 /*
     if (stats)
