--- a/tests/integration/test_execute.py
+++ b/tests/integration/test_execute.py
@@ -80,7 +80,7 @@ def test_log_outputs_logs_multiline_error_output():
     flexmock(module).should_receive('command_for_process').and_return('grep')
 
     process = subprocess.Popen(
-        ['python', '-c', 'foopydoo'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT
+        ['python3', '-c', 'foopydoo'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT
     )
     flexmock(module).should_receive('output_buffer_for_process').and_return(process.stdout)
     flexmock(module.logger).should_call('log').at_least().times(3)
