--- gsm-1.0-pl12/src/code.c.warn	2006-04-26 21:16:50.000000000 +0200
+++ gsm-1.0-pl12/src/code.c	2007-05-11 00:26:52.000000000 +0200
@@ -9,8 +9,8 @@
 #include	"config.h"
 
 
-#ifdef	HAS_STDLIB_H
-#include	<stdlib.h>
+#ifdef	HAS_STRING_H
+#include	<string.h>
 #else
 #	include "proto.h"
 	extern char	* memcpy P((char *, char *, int));
--- gsm-1.0-pl12/src/toast.c.warn	2006-04-26 21:14:26.000000000 +0200
+++ gsm-1.0-pl12/src/toast.c	2007-05-11 00:26:52.000000000 +0200
@@ -6,6 +6,8 @@
 
 /* $Header: /home/fedora/jkeating/pkgs/rpms/gsm/devel/gsm-warnings.patch,v 1.1 2007/05/11 23:18:21 rathann Exp $ */
 
+#include	<stdio.h>
+#include	<getopt.h>
 #include	"toast.h"
 
 /*  toast -- lossy sound compression using the gsm library.
--- gsm-1.0-pl12/add-test/add_test.c.warn	2006-04-26 21:14:25.000000000 +0200
+++ gsm-1.0-pl12/add-test/add_test.c	2007-05-11 00:28:03.000000000 +0200
@@ -9,6 +9,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h>
 
 #include "gsm.h"
 
@@ -29,7 +30,7 @@
 longword M_gsm_L_mult P((word op1, word op2));
 longword M_gsm_L_add P((longword op1, longword op2));
 
-help()
+void help(void)
 {
 puts( "  add a b      sub a b     mult a b   div    a b" );
 puts( "L_add A B    L_sub A B   L_mult A B   mult_r a b" );
