Fixes build with ninja cmake backend

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -452,9 +452,9 @@
 			-cf ${docname} docs/man/*.[1-8] docs/html/
 	)
 
-	add_custom_target(${distname}
+	add_custom_target(${distname}-target
 		WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-		BYPRODUCTS ${distname} ${docname}
+		BYPRODUCTS ${distname}
 		VERBATIM
 		DEPENDS ChangeLog ${docname} po/rpm.pot
 		COMMAND git archive
@@ -475,7 +475,7 @@
 				${CMAKE_BINARY_DIR}/${docname}
 		COMMAND bzip2 -f ${CMAKE_BINARY_DIR}/${tarname}
 	)
-	add_custom_target(${targetname} DEPENDS ${distname})
+	add_custom_target(${targetname} DEPENDS ${distname}-target)
 endfunction()
 
 add_tarball(dist ${PROJECT_NAME}-${PROJECT_VERSION} HEAD)
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -53,7 +53,7 @@
 endforeach()
 
 foreach(cmd rpmverify rpmquery)
-	add_custom_target(${cmd} ALL COMMAND
+	add_custom_target(${cmd}-target ALL COMMAND
 			${CMAKE_COMMAND} -E create_symlink rpm ${cmd}
 			BYPRODUCTS ${cmd})
 	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${cmd} TYPE BIN)
