--- openmpi-2.1.1/opal/include/opal/sys/cma.h.orig
+++ openmpi-2.1.1/opal/include/opal/sys/cma.h
@@ -28,7 +28,7 @@
 #endif
 
 #ifdef HAVE_UNISTD_H
-#include <sys/unistd.h>
+#include <unistd.h>
 #endif
 
 #ifdef __linux__
--- openmpi-2.1.1/opal/mca/pmix/pmix112/pmix/include/pmix/pmix_common.h.orig
+++ openmpi-2.1.1/opal/mca/pmix/pmix112/pmix/include/pmix/pmix_common.h
@@ -51,6 +51,7 @@
 #include <stdint.h>
 #include <string.h>
 #include <sys/time.h> /* for struct timeval */
+#include <unistd.h>
 
 #include <pmix_version.h>
 
