diff --git a/src/Makefile.am b/src/Makefile.am
index 3a8300b63c65..e2fc759f3ce1 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -110,7 +110,7 @@ python-build:
 	$(PYTHON) setup.py build
 
 install-exec-hook:
-	$(PYTHON) -m pip install `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` \
+	$(PYTHON) -m pip install --no-build-isolation `test -n "$(DESTDIR)" && echo --root $(DESTDIR)` \
 	`test -n "$(PREFIX)" && echo --prefix $(PREFIX)` .
 
 uninstall-hook:

