Upstream-Issue: https://github.com/denoland/deno_core/pull/330
--
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 16 Nov 2023 21:56:42 +0100
Subject: [PATCH] Remove ICU_DATA from jsruntime

--- a/vendor/deno_core/runtime/jsruntime.rs
+++ b/vendor/deno_core/runtime/jsruntime.rs
@@ -341,11 +341,11 @@
   predictable: bool,
   expose_natives: bool,
 ) {
-  // Include 10MB ICU data file.
-  #[repr(C, align(16))]
-  struct IcuData([u8; 10631872]);
-  static ICU_DATA: IcuData = IcuData(*include_bytes!("icudtl.dat"));
-  v8::icu::set_common_data_73(&ICU_DATA.0).unwrap();
+  // Include 10MB ICU data file.  XXX-Patched
+  //#[repr(C, align(16))]
+  //struct IcuData([u8; 10631872]);
+  //static ICU_DATA: IcuData = IcuData(*include_bytes!("icudtl.dat"));
+  //v8::icu::set_common_data_73(&ICU_DATA.0).unwrap();
 
   let base_flags = concat!(
     " --wasm-test-streaming",
