From: Jakub Jirutka <jakub@jirutka.cz>
Date: Thu, 16 Nov 2023 21:56:42 +0100
Subject: [PATCH] Link with system-provided libraries

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -217 +217 @@
-rusqlite = { version = "0.34.0", features = ["unlock_notify", "bundled", "session", "modern_sqlite", "limits"] } # "modern_sqlite": need sqlite >= 3.49.0 for some db configs
+rusqlite = { version = "0.34.0", features = ["unlock_notify", "buildtime_bindgen", "session", "modern_sqlite", "limits"] }
@@ -310 +310 @@
-zstd = "=0.13.2"
+zstd = { version = "=0.13.2", features = ["pkg-config"] }
@@ -357 +357 @@
-libffi-sys = "=3.2.0"
+libffi-sys = { version = "=3.2.0", features = ["system"] }
--- a/ext/canvas/Cargo.toml
+++ b/ext/canvas/Cargo.toml
@@ -26 +26 @@
-lcms2 = { workspace = true, features = ["static"] }
+lcms2 = { workspace = true }
