diff --git a/Makefile b/Makefile
index 5f68100828..123bfe9974 100644
--- a/Makefile
+++ b/Makefile
@@ -102,7 +102,7 @@ vendor: $(GLIDE_YAML)
 	touch $@
 
 test: $(SOURCES) vendor
-	SHELL=/bin/sh GOOS= go test -v -tags "$(TAGS)" \
+	SHELL=/bin/sh GOOS= go test -v -tags "$(TAGS)" -vet=off \
 				github.com/junegunn/fzf/src \
 				github.com/junegunn/fzf/src/algo \
 				github.com/junegunn/fzf/src/tui \
