diff --git a/Makefile b/Makefile
index d5efac7..104be4e 100644
--- a/Makefile
+++ b/Makefile
@@ -100,7 +100,6 @@ dist_doc_DATA = README.md NEWS.md \
 DISTCLEANFILES = config.mak config.status lib/usual/config.h config.log
 
 DIST_SUBDIRS = doc test
-dist_man_MANS = doc/pgbouncer.1 doc/pgbouncer.5
 
 pgbouncer_LDFLAGS := $(TLS_LDFLAGS)
 pgbouncer_LDADD := $(CARES_LIBS) $(LIBEVENT_LIBS) $(TLS_LIBS) $(LIBS)
@@ -195,14 +194,6 @@ $(w32zip): pgbouncer.exe pgbevent.dll etc/pgbouncer.ini etc/userlist.txt README.
 tags:
 	ctags src/*.c include/*.h lib/usual/*.[ch] lib/usual/*/*.[ch]
 
-htmls:
-	for f in *.md doc/*.md; do \
-		mkdir -p html && $(PANDOC) $$f -o html/`basename $$f`.html; \
-	done
-
-doc/pgbouncer.1 doc/pgbouncer.5:
-	$(MAKE) -C doc $(@F)
-
 lint:
 	flake8
 
