--- a/main/source/crash_handlers.cpp
+++ b/main/source/crash_handlers.cpp
@@ -143,7 +143,7 @@
         try {
             std::rethrow_exception(std::current_exception());
         } catch (std::exception &ex) {
-            std::string exceptionStr = hex::format("{}()::what() -> {}", llvm::itaniumDemangle(typeid(ex).name(), nullptr, nullptr, nullptr), ex.what());
+            std::string exceptionStr = hex::format("{}()::what() -> {}", llvm::itaniumDemangle(typeid(ex).name()), ex.what());
             log::fatal("Program terminated with uncaught exception: {}", exceptionStr);
 
         }
