From f25205f4513c4fa72be6940081c62e613d1fddc6 Mon Sep 17 00:00:00 2001
From: spameier <40004508+spameier@users.noreply.github.com>
Date: Wed, 30 Apr 2025 16:18:53 +0200
Subject: [PATCH] fix(quickemu): correctly handle version 10.0.0 of QEMU

---
 quickemu | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/quickemu b/quickemu
index b188804f00..baee457459 100755
--- a/quickemu
+++ b/quickemu
@@ -1954,7 +1954,8 @@ if [ "${OS_KERNEL}" == "Darwin" ]; then
 fi
 
 QEMU_VER_LONG=$(${QEMU_IMG} --version | head -n 1 | awk '{print $3}')
-QEMU_VER_SHORT=$(echo "${QEMU_VER_LONG//./}" | cut -c1-2)
+# strip patch version and remove dots. 6.0.0 => 60 / 10.0.0 => 100
+QEMU_VER_SHORT=$(echo "${QEMU_VER_LONG%.*}" | sed 's/\.//g')
 if [ "${QEMU_VER_SHORT}" -lt 60 ]; then
     echo "ERROR! QEMU 6.0.0 or newer is required, detected ${QEMU_VER_LONG}."
     exit 1

