From 519e41833c1f7d353c09d8d51bfd6d13d0cfc16c Mon Sep 17 00:00:00 2001
From: Nicholas Marriott <nicholas.marriott@gmail.com>
Date: Mon, 18 Feb 2019 22:26:29 +0000
Subject: [PATCH] Do not add base64.c twice, from Rosen Penev.

---
 Makefile.am | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 611ae87..e9f2f64 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -123,6 +123,3 @@ endif
 if NO_STRTONUM
 nodist_fdm_SOURCES += compat/strtonum.c
 endif
-if NO_B64_NTOP
-nodist_fdm_SOURCES += compat/base64.c
-endif
From 3232e537ccaba4417b25d9d70264e4a5533042da Mon Sep 17 00:00:00 2001
From: Nicholas Marriott <nicholas.marriott@gmail.com>
Date: Mon, 18 Mar 2019 13:04:00 +0000
Subject: [PATCH] Fix bas64 declarations, from makepost at firemail dot cc.

---
 fdm.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/fdm.h b/fdm.h
index 41eaa37..511a7b1 100644
--- a/fdm.h
+++ b/fdm.h
@@ -20,7 +20,6 @@
 #define FDM_H
 
 #include <sys/param.h>
-#include <sys/cdefs.h>
 #include <sys/stat.h>
 
 #ifdef HAVE_QUEUE_H
@@ -725,8 +724,8 @@ size_t		 strlcat(char *, const char *, size_t);
 
 #ifndef HAVE_B64_NTOP
 /* base64.c */
-int b64_ntop(src, srclength, target, targsize);
-int b64_pton(src, target, targsize);
+int b64_ntop(u_char const *, size_t, char *, size_t);
+int b64_pton(char const *, u_char *, size_t);
 #endif
 
 /* shm.c */
