diff --git a/src/gps_functions.c b/src/gps_functions.c
index 04f88e0..e3df086 100644
--- a/src/gps_functions.c
+++ b/src/gps_functions.c
@@ -752,7 +752,7 @@ cb_gpsd_data(GIOChannel *src, GIOCondition condition, gpointer data)
 	{
 		gpsdata->satellites_used = libgps_gpsdata.satellites_used;
 		gpsdata->hdop = libgps_gpsdata.dop.hdop;
-		gpsdata->fix.time = libgps_gpsdata.fix.time;
+		gpsdata->fix.time = libgps_gpsdata.fix.time.tv_nsec;
 		if (isnan(gpsdata->fix.time))
 		{
 			gpsdata->fix.time = (time_t) 0;
