diff -u -r RDFLib-rdflib-2fb94c8/rdflib/plugins/sparql/evaluate.py RDFLib-rdflib-2fb94c8-py3.7/rdflib/plugins/sparql/evaluate.py
--- RDFLib-rdflib-2fb94c8/rdflib/plugins/sparql/evaluate.py	2017-01-29 17:51:48.000000000 +0100
+++ RDFLib-rdflib-2fb94c8-py3.7/rdflib/plugins/sparql/evaluate.py	2018-07-29 17:31:55.614217759 +0200
@@ -326,9 +326,12 @@
     # import pdb; pdb.set_trace()
     res = evalPart(ctx, slice.p)
     i = 0
-    while i < slice.start:
-        res.next()
-        i += 1
+    try:
+        while i < slice.start:
+            res.next()
+            i += 1
+    except StopIteration:
+        return
     i = 0
     for x in res:
         i += 1
