--- a/configure.ac
+++ b/configure.ac
@@ -34,6 +34,7 @@
 
 dnl Checks for programs.
 AC_PROG_CC
+AC_PROG_CPP
 AC_PROG_INSTALL
 AC_PROG_MKDIR_P
 AC_PROG_YACC
@@ -89,7 +90,6 @@
 
 dnl Checks for library functions.
 AC_TYPE_GETGROUPS
-AC_PROG_GCC_TRADITIONAL
 AC_FUNC_MMAP
 
 AC_FUNC_WAIT3
@@ -103,7 +111,9 @@
 char *foov[] = { "-a", "-bfoo", "bar" };
 int fooc = (sizeof(foov) / sizeof(foov[0]));
 
+_Noreturn void exit (int);
 int getenv() { return *foo; }
+int getopt(int, char * const [], const char *);
 int main() { while (-1 != getopt(fooc, foov, "ab:")); exit(0); }
 changequote([,])
 ]])],[es_cv_abused_getenv=no],[es_cv_abused_getenv=yes
