diff --git a/src/aspnetcore/Directory.Build.props.orig b/src/aspnetcore/Directory.Build.props
index 2754f04d06c..4fcdf7e0a61 100644
--- a/src/aspnetcore/Directory.Build.props.orig
+++ b/src/aspnetcore/Directory.Build.props
@@ -136,6 +136,9 @@
 
     <!-- // Temporary diagnostic code from https://github.com/dotnet/extensions/pull/4130 for metrics APIs used in test. -->
     <NoWarn>$(NoWarn);TBD</NoWarn>
+
+    <!-- don't warn about security issues -->
+    <NoWarn>$(NoWarn);NU1903;NU5104</NoWarn>
   </PropertyGroup>
 
   <!-- Source code settings -->
