diff --git a/apps/Utils.cpp b/apps/Utils.cpp
index 92d810b..9ea3dc7 100644
--- a/apps/Utils.cpp
+++ b/apps/Utils.cpp
@@ -44,7 +44,9 @@
 #include <cxxabi.h>
 #include <sys/ioctl.h>
 #include <sys/wait.h>  // WIFEXITED, WEXITSTATUS
+#ifdef __GLIBC__
 #include <execinfo.h> // backtrace
+#endif
 #include <dlfcn.h> // dladdr
 #endif
 
@@ -600,7 +602,7 @@ double Utils::normalizeLongitude(double longitude)
     return longitude;
 }
 
-
+#ifdef __GLIBC__
 std::vector<std::string> Utils::backtrace()
 {
     std::vector<std::string> lines;
@@ -659,7 +661,7 @@ std::vector<std::string> Utils::backtrace()
 #endif
     return lines;
 }
-
+#endif
 
 // Useful for debug on occasion.
 std::string Utils::hexDump(const char *buf, size_t count)
