--- lua-projects-0.4/stringy/Makefile
+++ lua-projects-0.4.libs/stringy/Makefile
@@ -1,6 +1,7 @@
 LUA ?= lua5.1
 LUA_PC ?= lua5.1
 LUA_CFLAGS = $(shell pkg-config $(LUA_PC) --cflags)
+LUA_LDFLAGS = $(shell pkg-config $(LUA_PC) --libs)
 
 CFLAGS ?= -O3 -Wall -Werror
 
@@ -12,7 +13,7 @@
 stringy.o: fastsearch.h
 
 stringy.so: stringy.o stringy_test.lua
-	$(CC) -shared stringy.o -o $@
+	$(CC) -shared $(LUA_LDFLAGS) stringy.o -o $@
 	$(LUA) stringy_test.lua
 
 clean:
