--- a/flowd.conf
+++ b/flowd.conf
@@ -1,7 +1,5 @@
-# $Id: flowd.conf,v 1.15 2005/05/14 06:04:48 djm Exp $
-
 # Specify where flowd should store its flow records
-logfile "/var/log/flowd"
+logfile "/var/lib/flowd/data"
 
 # Specify what addresses/ports flowd should listen on.
 # Multiple addresses may be specified
--- a/flowd.conf.5.in
+++ b/flowd.conf.5.in
@@ -156,7 +156,7 @@
 .Pp
 For example,
 .Bd -literal -offset indent
-logfile "/var/log/flowd.bin"
+logfile "/var/lib/flowd/data"
 .Ed
 .Pp
 There is no default value for this option and it it mandatory 
@@ -172,7 +172,7 @@
 .Pp
 For example,
 .Bd -literal -offset indent
-logsock "/var/log/flowd.sock"
+logsock "/var/lib/flowd/socket"
 .Ed
 .Pp
 This option accepts the modifier
@@ -184,7 +184,7 @@
 .Pp
 For example,
 .Bd -literal -offset indent
-logsock "/var/log/flowd.sock" bufsize 65536
+logsock "/var/lib/flowd/socket" bufsize 65536
 .Ed
 .Pp
 There is no default value for
