diff --git a/tests/all/lockfile.rs b/tests/all/lockfile.rs
index db29227..f2679b6 100644
--- a/tests/all/lockfile.rs
+++ b/tests/all/lockfile.rs
@@ -9,7 +9,7 @@ fn it_gets_wasm_bindgen_version() {
     fixture.cargo_check();
     let data = CrateData::new(&fixture.path, None).unwrap();
     let lock = Lockfile::new(&data).unwrap();
-    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.100"),);
+    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.108"),);
 }
 
 #[test]
@@ -48,7 +48,7 @@ fn it_gets_wasm_bindgen_version_in_crate_inside_workspace() {
                 crate-type = ["cdylib"]
 
                 [dependencies]
-                wasm-bindgen = "=0.2.100"
+                wasm-bindgen = "=0.2.108"
             "#,
         )
         .file(
@@ -64,7 +64,7 @@ fn it_gets_wasm_bindgen_version_in_crate_inside_workspace() {
     fixture.cargo_check();
     let data = CrateData::new(&fixture.path.join("blah"), None).unwrap();
     let lock = Lockfile::new(&data).unwrap();
-    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.100"),);
+    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.108"),);
 }
 
 #[test]
@@ -93,7 +93,7 @@ fn it_gets_wasm_bindgen_version_from_dependencies() {
                 crate-type = ["cdylib"]
 
                 [dependencies]
-                wasm-bindgen = "=0.2.100"
+                wasm-bindgen = "=0.2.108"
             "#,
         )
         .file(
@@ -132,5 +132,5 @@ fn it_gets_wasm_bindgen_version_from_dependencies() {
     fixture.cargo_check();
     let data = CrateData::new(&fixture.path.join("parent"), None).unwrap();
     let lock = Lockfile::new(&data).unwrap();
-    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.100"),);
+    assert_eq!(lock.wasm_bindgen_version(), Some("0.2.108"),);
 }
diff --git a/tests/all/readme.rs b/tests/all/readme.rs
index 61f6307..3b4bf40 100644
--- a/tests/all/readme.rs
+++ b/tests/all/readme.rs
@@ -60,7 +60,7 @@ fn it_copies_a_readme_provided_path() {
             # bindgen downloaded is what we expect, and if `=` is
             # removed then it will download whatever the newest version
             # of wasm-bindgen is which may not be what's listed here.
-            wasm-bindgen = "=0.2.100"
+            wasm-bindgen = "=0.2.108"
 
             [dev-dependencies]
             wasm-bindgen-test = "0.3"
@@ -120,7 +120,7 @@ fn it_ignores_a_disabled_readme() {
             # bindgen downloaded is what we expect, and if `=` is
             # removed then it will download whatever the newest version
             # of wasm-bindgen is which may not be what's listed here.
-            wasm-bindgen = "=0.2.100"
+            wasm-bindgen = "=0.2.108"
 
             [dev-dependencies]
             wasm-bindgen-test = "0.3"
diff --git a/tests/all/utils/fixture.rs b/tests/all/utils/fixture.rs
index 61eb7ed..ec3fc41 100644
--- a/tests/all/utils/fixture.rs
+++ b/tests/all/utils/fixture.rs
@@ -144,7 +144,7 @@ impl Fixture {
                     # bindgen downloaded is what we expect, and if `=` is
                     # removed then it will download whatever the newest version
                     # of wasm-bindgen is which may not be what's listed here.
-                    wasm-bindgen = "=0.2.100"
+                    wasm-bindgen = "=0.2.108"
 
                     [dev-dependencies]
                     wasm-bindgen-test = "0.3"
@@ -182,7 +182,7 @@ impl Fixture {
                     # bindgen downloaded is what we expect, and if `=` is
                     # removed then it will download whatever the newest version
                     # of wasm-bindgen is which may not be what's listed here.
-                    wasm-bindgen = "=0.2.100"
+                    wasm-bindgen = "=0.2.108"
 
                     [dev-dependencies]
                     wasm-bindgen-test = "0.3"
@@ -220,7 +220,7 @@ impl Fixture {
                     crate-type = ["cdylib"]
 
                     [dependencies]
-                    wasm-bindgen = "=0.2.100"
+                    wasm-bindgen = "=0.2.108"
 
                     [dev-dependencies]
                     wasm-bindgen-test = "=0.2.21"
@@ -265,7 +265,7 @@ impl Fixture {
 
         static INSTALL_WASM_BINDGEN: Once = Once::new();
         let cache = self.cache();
-        let version = "0.2.100";
+        let version = "0.2.108";
 
         let download = || {
             if let Ok(download) =
