diff --git a/run_tests.py b/run_tests.py
index f54e43e..daf370e 100644
--- a/run_tests.py
+++ b/run_tests.py
@@ -73,6 +73,7 @@ def load_tests(lines):
 def parse_options(checker, options):
     """Parse the CLI-style flags from `options` and expose to `checker`"""
     options_manager = OptionManager(
+        formatter_names=["asd"],
         version=flake8.__version__,
         plugin_versions=f"naming: {pep8ext_naming.__version__}",
         parents=[],
