diff --git a/xsdata/formats/dataclass/typing.py b/xsdata/formats/dataclass/typing.py
index 446c767..182557e 100644
--- a/xsdata/formats/dataclass/typing.py
+++ b/xsdata/formats/dataclass/typing.py
@@ -6,8 +6,8 @@ from typing import Type
 from typing import TypeVar
 from typing import Union
 
-from typing_extensions import get_args
-from typing_extensions import get_origin
+from typing import get_args
+from typing import get_origin
 
 NONE_TYPE = type(None)
 
