diff --git a/xsdata/formats/dataclass/serializers/mixins.py b/xsdata/formats/dataclass/serializers/mixins.py
index 4ab8509..31f4643 100644
--- a/xsdata/formats/dataclass/serializers/mixins.py
+++ b/xsdata/formats/dataclass/serializers/mixins.py
@@ -14,7 +14,7 @@ from typing import (
 from xml.etree.ElementTree import QName
 from xml.sax.handler import ContentHandler
 
-from typing_extensions import TypeAlias
+from typing import TypeAlias
 
 from xsdata.exceptions import XmlWriterError
 from xsdata.formats.converter import converter
diff --git a/xsdata/formats/dataclass/typing.py b/xsdata/formats/dataclass/typing.py
index a375346..defaf1b 100644
--- a/xsdata/formats/dataclass/typing.py
+++ b/xsdata/formats/dataclass/typing.py
@@ -1,7 +1,7 @@
 import sys
 from typing import Any, Iterator, Tuple, Type, TypeVar, Union
 
-from typing_extensions import get_args, get_origin
+from typing import get_args, get_origin
 
 NONE_TYPE = type(None)
 
