diff --git a/sic.c b/sic.c
--- a/sic.c
+++ b/sic.c
@@ -19,6 +19,7 @@
 static char bufin[4096];
 static char bufout[4096];
 static char channel[256];
+static char hidecmd[128] = {0};
 static time_t trespond;
 static FILE *srv;
 
@@ -99,6 +100,9 @@
        case 's':
            strlcpy(channel, p, sizeof channel);
            return;
+       case 'h':
+           strlcpy(hidecmd, p, sizeof hidecmd);
+           return;
        }
    }
    sout("%s", s);
@@ -129,7 +133,8 @@
    else if(!strcmp("PING", cmd))
        sout("PONG %s", txt);
    else {
-       pout(usr, ">< %s (%s): %s", cmd, par, txt);
+       if (!strcasestr(hidecmd, cmd))
+           pout(usr, ">< %s (%s): %s", cmd, par, txt);
        if(!strcmp("NICK", cmd) && !strcmp(usr, nick))
            strlcpy(nick, txt, sizeof nick);
    }
