diff -up cups-filters-1.0.41/Makefile.am.urftopdf cups-filters-1.0.41/Makefile.am
--- cups-filters-1.0.41/Makefile.am.urftopdf	2014-03-11 13:13:48.012222641 +0100
+++ cups-filters-1.0.41/Makefile.am	2014-03-11 13:13:48.020222538 +0100
@@ -410,8 +410,8 @@ pkgfilter_PROGRAMS += \
 	pdftoraster \
 	rastertoescpx \
 	rastertopclx \
-	texttopdf \
-	urftopdf
+	texttopdf
+
 if ENABLE_IMAGEFILTERS
 pkgfilter_PROGRAMS += \
 	imagetopdf \
diff -up cups-filters-1.0.41/Makefile.in.urftopdf cups-filters-1.0.41/Makefile.in
--- cups-filters-1.0.41/Makefile.in.urftopdf	2014-03-11 13:13:48.022222512 +0100
+++ cups-filters-1.0.41/Makefile.in	2014-03-11 13:14:45.729477255 +0100
@@ -94,7 +94,7 @@ pkgfilter_PROGRAMS = pdftoopvp$(EXEEXT)
 	commandtopclx$(EXEEXT) gstoraster$(EXEEXT) pdftoijs$(EXEEXT) \
 	pdftoippprinter$(EXEEXT) pdftops$(EXEEXT) pdftoraster$(EXEEXT) \
 	rastertoescpx$(EXEEXT) rastertopclx$(EXEEXT) \
-	texttopdf$(EXEEXT) urftopdf$(EXEEXT) $(am__EXEEXT_1)
+	texttopdf$(EXEEXT) $(am__EXEEXT_1)
 @ENABLE_IMAGEFILTERS_TRUE@am__append_1 = \
 @ENABLE_IMAGEFILTERS_TRUE@	imagetopdf \
 @ENABLE_IMAGEFILTERS_TRUE@	imagetoraster
diff -up cups-filters-1.0.41/mime/cupsfilters.convs.urftopdf cups-filters-1.0.41/mime/cupsfilters.convs
--- cups-filters-1.0.41/mime/cupsfilters.convs.urftopdf	2014-03-11 13:13:48.017222576 +0100
+++ cups-filters-1.0.41/mime/cupsfilters.convs	2014-03-11 13:13:48.023222499 +0100
@@ -68,7 +68,6 @@ image/x-xpixmap		application/vnd.cups-pd
 image/x-xwindowdump	application/vnd.cups-pdf		65	imagetopdf
 image/x-sun-raster	application/vnd.cups-pdf		65	imagetopdf
 application/vnd.cups-pdf-banner	application/pdf			32	bannertopdf
-image/urf		application/pdf				0	urftopdf
 
 ########################################################################
 #
