diff --git a/filter/foomatic-rip/foomaticrip.h b/filter/foomatic-rip/foomaticrip.h
index 875e6b8..b46abba 100644
--- a/filter/foomatic-rip/foomaticrip.h
+++ b/filter/foomatic-rip/foomaticrip.h
@@ -33,6 +33,7 @@
 #include <stddef.h>
 #include <stdio.h>
 #include <time.h>
+#include <limits.h>
 
 /* This is the location of the debug logfile (and also the copy of the
  * processed PostScript data) in case you have enabled debugging above.
@@ -75,7 +76,9 @@
 extern int spooler;
 
 
+#ifndef PATH_MAX
 #define PATH_MAX 65536
+#endif
 
 typedef struct {
     char printer[256];
