--- b/tests/common/idp/ninja/urls.py
+++ /dev/null
@@ -1,8 +0,0 @@
-from django.urls import path
-
-from .views import api
-
-
-urlpatterns = [
-    path("", api.urls),
-]
--- b/tests/common/idp/urls.py
+++ a/tests/common/idp/urls.py
@@ -3,5 +3,4 @@
 
 urlpatterns = [
     path("drf/", include("tests.common.idp.rest_framework.urls")),
-    path("ninja/", include("tests.common.idp.ninja.urls")),
 ]
