diff -upr cloc-1.74.orig/Unix/t/01_opts.t cloc-1.74/Unix/t/01_opts.t
--- cloc-1.74.orig/Unix/t/01_opts.t	2017-09-15 00:11:50.011466821 +0200
+++ cloc-1.74/Unix/t/01_opts.t	2017-09-15 00:12:12.404939113 +0200
@@ -65,60 +65,6 @@ my @Tests = (
                     'ref'  => '../tests/outputs/issues/114/T7.yaml',
                 },
                 {
-                    'name' => 'git submodule handling (github issue #131 T1)',
-                    'cd'   => '../tests/inputs',
-                    'args' => 'issues/131',
-                    'ref'  => '../tests/outputs/issues/131/T1.yaml',
-                },
-                {
-                    'name' => 'git submodule handling (github issue #131 T2)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs git issues/131',
-                    'ref'  => '../tests/outputs/issues/131/T2.yaml',
-                },
-                {
-                    'name' => 'all files (github issue #132 T1)',
-                    'cd'   => '../tests/inputs',
-                    'args' => 'issues/132',
-                    'ref'  => '../tests/outputs/issues/132/T1.yaml',
-                },
-                {
-                    'name' => '--vcs git issues/132 (github issue #132 T2)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs git issues/132',
-                    'ref'  => '../tests/outputs/issues/132/T2.yaml',
-                },
-                {
-                    'name' => '--vcs-git --exclude-dir ignore_dir (github issue #132 T3)',
-                    'cd'   => '../tests/inputs/issues/132',
-                    'args' => '--vcs git --exclude-dir ignore_dir .',
-                    'ref'  => '../tests/outputs/issues/132/T3.yaml',
-                },
-                {
-                    'name' => '--vcs git --fullpath --not-match-d issues/132/ignore_dir (github issue #132 T4)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs git --fullpath --not-match-d issues/132/ignore_dir issues/132',
-                    'ref'  => '../tests/outputs/issues/132/T4.yaml',
-                },
-                {
-                    'name' => '--vcs git --match-f C-Ansi (github issue #132 T5)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs git --match-f C-Ansi issues/132',
-                    'ref'  => '../tests/outputs/issues/132/T5.yaml',
-                },
-                {
-                    'name' => '--vcs git --match-f "\.c$" (github issue #132 T6)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs git --match-f "\.c$" issues/132',
-                    'ref'  => '../tests/outputs/issues/132/T6.yaml',
-                },
-                {
-                    'name' => '--vcs "find X" (github issue #147)',
-                    'cd'   => '../tests/inputs',
-                    'args' => '--vcs "find foo_bar"',
-                    'ref'  => '../tests/outputs/issues/147/T1.yaml',
-                },
-                {
                     'name' => '--read-lang-def w/remove_between_general (github issue #166)',
                     'cd'   => '../tests/inputs/issues/166',
                     'args' => '--read-lang-def X fake.thy',
