From c6b2ee7291fc9dee8c79f67fe8c9a69a52b9b17c Mon Sep 17 00:00:00 2001
From: Cameron Nemo <camerontnorman@gmail.com>
Date: Fri, 30 Nov 2018 19:58:58 -0800
Subject: [PATCH] swaylock: fix build with musl libc

Signed-off-by: Cameron Nemo <camerontnorman@gmail.com>
---
 swaylock/main.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/swaylock/main.c b/swaylock/main.c
index 50f145e3..9aeb4e64 100644
--- a/swaylock/main.c
+++ b/swaylock/main.c
@@ -4,6 +4,7 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <getopt.h>
+#include <poll.h>
 #include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -961,7 +962,7 @@ int main(int argc, char **argv) {
 	}
 
 	state.eventloop = loop_create();
-	loop_add_fd(state.eventloop, wl_display_get_fd(state.display), POLL_IN,
+	loop_add_fd(state.eventloop, wl_display_get_fd(state.display), POLLIN,
 			display_in, NULL);
 
 	state.run_display = true;
-- 
2.19.2

