Patch-Source: https://github.com/pallets/click/pull/2944

--- a/src/click/_termui_impl.py
+++ b/src/click/_termui_impl.py
@@ -427,8 +427,9 @@ def _pipepager(
     cmd_filepath = which(cmd)
     if not cmd_filepath:
         return False
-    # Resolves symlinks and produces a normalized absolute path string.
-    cmd_path = Path(cmd_filepath).resolve()
+
+    # Produces a normalized absolute path string.
+    cmd_path = Path(cmd_filepath).absolute()
     cmd_name = cmd_path.name
 
     import subprocess
