From ce210b9cb0be02dc51bf01de5e52ced27e10e24f Mon Sep 17 00:00:00 2001
From: Brandon Boese <brandonboese@protonmail.com>
Date: Tue, 17 Oct 2023 21:40:21 -0400
Subject: [PATCH] manually disable tests

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d8994c..e60d65f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,7 +176,6 @@ add_subdirectory(plugins/Lomiri/Thumbnailer.0.1)
 if (${ENABLE_UBUNTU_COMPAT})
     add_subdirectory(plugins/Ubuntu/Thumbnailer.0.1)
 endif()
-add_subdirectory(tests)
 add_subdirectory(include)
 add_subdirectory(man)
 #add_subdirectory(doc)
-- 
2.42.0

