Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: package name with capital C
----

--- a/tiledb/CMakeLists.txt
+++ a/tiledb/CMakeLists.txt
@@ -635,8 +642,8 @@
 )
 
 if(TILEDB_CRC32)
-  find_package(crc32c_EP REQUIRED)
-  find_package(crc32c CONFIG REQUIRED
+  find_package(Crc32c_EP REQUIRED)
+  find_package(Crc32c CONFIG REQUIRED
     HINTS
       ${TILEDB_EP_INSTALL_PREFIX}/lib/cmake
     )


--- a/test/external/src/crc32_link_test.cc
+++ b/test/external/src/crc32_link_test.cc
@@ -32,7 +32,7 @@
 
 #include <stdio.h>
 
-#include <include/crc32c/crc32c.h>
+#include <crc32c/crc32c.h>
 
 int main() {
   uint8_t buf[32];
