From 4d2ca97c00f3a59ab950585ad60b54dc96a63d2d Mon Sep 17 00:00:00 2001
From: Achill Gilgenast <achill@achill.org>
Date: Sat, 29 Nov 2025 00:21:38 +0100
Subject: [PATCH] build: allow build with wlroots-0.20

to test builds against wlroots master
---
 meson.build | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/meson.build b/meson.build
index 3563ac5..b5bb37e 100644
--- a/meson.build
+++ b/meson.build
@@ -22,10 +22,13 @@ wayland_egl    = dependency('wayland-egl')
 wayland_protos = dependency('wayland-protocols', version: '>=1.14')
 xkbcommon      = dependency('xkbcommon')
 xwayland       = dependency('xwayland', version: '>=24.1')
-wlroots        = dependency('wlroots-0.19', version: '>=0.19', required: false)
 
+wlroots = dependency('wlroots-0.20', version: '>=0.20', required: false)
 if not wlroots.found()
-  wlroots = dependency('wlroots-0.18', version: '>=0.18')
+  wlroots = dependency('wlroots-0.19', version: '>=0.19', required: false)
+  if not wlroots.found()
+    wlroots = dependency('wlroots-0.18', version: '>=0.18')
+  endif
 endif
 
 xwayland_executable_path = xwayland.get_variable(pkgconfig: 'xwayland')
-- 
GitLab

