--- a/modules/javafx.graphics/src/main/native-font/freetype.c
+++ b/modules/javafx.graphics/src/main/native-font/freetype.c
@@ -391,9 +391,11 @@
     (JNIEnv *env, jclass that, jlong arg0, jobject arg1, jlong arg2, jlong arg3)
 {
     FT_Vector *lpDelta = NULL;
+    FT_Vector tmp;
     if (arg2 || arg3) {
         FT_Vector vec = {arg2, arg3};
-        lpDelta = &vec;
+        tmp = vec;
+        lpDelta = &tmp;
     }
     if (arg1) {
         FT_Matrix matrix, *lpMatrix = NULL;
