diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 621b9a5..d5b76a9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -129,7 +129,7 @@ list(APPEND REMMINA_SRCS
 
 add_executable(remmina ${REMMINA_SRCS})
 include_directories(${GTK_INCLUDE_DIRS} ${gio_INCLUDE_DIRS} ${gio-unix_INCLUDE_DIRS})
-target_link_libraries(remmina ${GTK_LIBRARIES})
+target_link_libraries(remmina ${GTK_LIBRARIES} -lintl)
 
 if(WITH_MANPAGES)
 	install(FILES remmina.1 DESTINATION ${CMAKE_INSTALL_FULL_MANDIR}/man1)

