diff --git a/plugins/rdp/rdp_event.c b/plugins/rdp/rdp_event.c
index 15fef90..5aebae2 100644
--- a/plugins/rdp/rdp_event.c
+++ b/plugins/rdp/rdp_event.c
@@ -43,7 +43,9 @@
 #include <gdk/gdkkeysyms.h>
 #include <cairo/cairo-xlib.h>
 #include <freerdp/locale/keyboard.h>
+#ifdef HAVE_EXECINFO_H
 #include <execinfo.h>
+#endif
 
 static void remmina_rdp_event_on_focus_in(GtkWidget* widget, GdkEventKey* event, RemminaProtocolWidget* gp)
 {
