--- a/plugins/vnc/CMakeLists.txt	2012-06-04 13:03:46.542093149 -0600
+++ b/plugins/vnc/CMakeLists.txt	2012-06-04 13:03:34.112332497 -0600
@@ -28,7 +28,7 @@ 
 set_target_properties(remmina-plugin-vnc PROPERTIES PREFIX "")
 
 include_directories(${REMMINA_COMMON_INCLUDE_DIRS} ${LIBVNCSERVER_INCLUDE_DIRS})
-target_link_libraries(remmina-plugin-vnc ${REMMINA_COMMON_LIBRARIES} ${LIBVNCSERVER_LIBRARIES})
+target_link_libraries(remmina-plugin-vnc ${REMMINA_COMMON_LIBRARIES} ${LIBVNCSERVER_LIBRARIES} gnutls )
 
 install(TARGETS remmina-plugin-vnc DESTINATION ${REMMINA_PLUGINDIR})
 
