--- a/modules/juce_core/native/juce_SharedCode_posix.h
+++ b/modules/juce_core/native/juce_SharedCode_posix.h
@@ -167,8 +167,8 @@
 namespace
 {
    #if JUCE_LINUX || (JUCE_IOS && (! TARGET_OS_MACCATALYST) && (! __DARWIN_ONLY_64_BIT_INO_T)) // (this iOS stuff is to avoid a simulator bug)
-    using juce_statStruct = struct stat64;
-    #define JUCE_STAT  stat64
+    using juce_statStruct = struct stat;
+    #define JUCE_STAT  stat
    #else
     using juce_statStruct = struct stat;
     #define JUCE_STAT  stat
