--- a/modules/juce_core/native/juce_SystemStats_linux.cpp
+++ b/modules/juce_core/native/juce_SystemStats_linux.cpp
@@ -204,7 +204,11 @@ String SystemStats::getUserLanguage()
    #if JUCE_BSD
     return {};
    #else
+   #ifdef __GLIBC__
     return getLocaleValue (_NL_ADDRESS_LANG_AB);
+   #else
+    return String(); // musl doesn't support _NL_ADDRESS_LANG_AB
+   #endif
    #endif
 }

@@ -213,7 +217,11 @@ String SystemStats::getUserRegion()
    #if JUCE_BSD
     return {};
    #else
+   #ifdef __GLIBC__
     return getLocaleValue (_NL_ADDRESS_COUNTRY_AB2);
+   #else
+    return String(); // musl doesn't support _NL_ADDRESS_COUNTRY_AB2
+   #endif
    #endif
 }
