diff --git a/src/main/include/log4cxx/helpers/asyncbuffer.h b/src/main/include/log4cxx/helpers/asyncbuffer.h
index 5639f906..fd991f31 100644
--- a/src/main/include/log4cxx/helpers/asyncbuffer.h
+++ b/src/main/include/log4cxx/helpers/asyncbuffer.h
@@ -204,7 +204,7 @@ public: // Modifiers
 	{
 		auto store = FmtArgStore();
 		( store.push_back(std::forward<Args>(args)), ...);
-		initializeForFmt(std::move(fmt_str), std::move(store));
+		initializeForFmt(StringViewType(fmt_str), std::move(store));
 	}
 
 #if LOG4CXX_WCHAR_T_API || LOG4CXX_LOGCHAR_IS_WCHAR
@@ -215,7 +215,7 @@ public: // Modifiers
 	{
 		auto store = WideFmtArgStore();
 		( store.push_back(std::forward<Args>(args)), ...);
-		initializeForFmt(std::move(fmt_str), std::move(store));
+		initializeForFmt(WideStringViewType(fmt_str), std::move(store));
 	}
 #endif // LOG4CXX_WCHAR_T_API || LOG4CXX_LOGCHAR_IS_WCHAR
 #endif // LOG4CXX_ASYNC_BUFFER_SUPPORTS_FMT
