From c176b7de9c2017d0fc1877659cea3bb6c330aafa Mon Sep 17 00:00:00 2001
From: Hunter <wardlawhunter@gmail.com>
Date: Mon, 9 Aug 2021 23:51:49 -0400
Subject: [PATCH] Fixed libfreeaptx reporting wrong version number

---
 freeaptx.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/freeaptx.h b/freeaptx.h
index c022919..23440c8 100644
--- a/freeaptx.h
+++ b/freeaptx.h
@@ -22,8 +22,8 @@
 #define freeaptx_H
 
 #define freeaptx_MAJOR 0
-#define freeaptx_MINOR 2
-#define freeaptx_PATCH 0
+#define freeaptx_MINOR 1
+#define freeaptx_PATCH 1
 
 #include <stddef.h>
 
