--- a/hxcpp/git/src/hx/libs/std/Sys.cpp
+++ b/hxcpp/git/src/hx/libs/std/Sys.cpp
@@ -30,7 +30,7 @@
    #include <limits.h>
    #ifndef ANDROID
       #include <locale.h>
-      #if !defined(BLACKBERRY) && !defined(EPPC) && !defined(GCW0) && !defined(__GLIBC__)
+      #if !defined(BLACKBERRY) && !defined(EPPC) && !defined(GCW0) && !defined(NEKO_LINUX)
          #include <xlocale.h>
       #endif
    #endif
