diff -up rp-pppoe-3.11/src/configure.orig rp-pppoe-3.11/src/configure
--- rp-pppoe-3.11/src/configure.orig	2012-08-17 12:31:25.000000000 -0600
+++ rp-pppoe-3.11/src/configure	2013-02-11 14:23:05.000000000 -0700
@@ -3679,6 +3679,7 @@ do :
 #include<net/ethernet.h>
 #include<linux/if.h>
 #include<linux/in.h>
+#include<linux/in6.h>
 
 "
 if test "x$ac_cv_header_linux_if_pppox_h" = x""yes; then :
@@ -4525,6 +4526,7 @@ else
 #include <net/ethernet.h>
 #include <linux/if.h>
 #include <linux/in.h>
+#include <linux/in6.h>
 #include <linux/if_pppox.h>
 int main()
 {
