diff --git a/chrome/test/BUILD.gn.orig b/chrome/test/BUILD.gn
index deb1ac1..5cab7c6 100644
--- a/chrome/test/BUILD.gn.orig
+++ b/chrome/test/BUILD.gn
@@ -6417,7 +6417,6 @@ template("performance_test_suite_template_base") {
       data_deps += [
         "//chrome/test:load_library_perf_tests",
         "//components/tracing:tracing_perftests",
-        "//components/variations/test_data/cipd:single_group_per_study_prefer_existing_behavior_seed",
         "//testing:run_perf_test",
         "//ui/views:views_perftests",
       ]
