--- a/electron/default_app/default_app.ts
+++ b/electron/default_app/default_app.ts
@@ -61,7 +61,7 @@
   };
 
   if (process.platform === 'linux') {
-    options.icon = url.fileURLToPath(new URL('icon.png', import.meta.url));
+    options.icon = 'file:///usr/share/icons/hicolor/1024x1024/apps/electron.png';
   }
 
   mainWindow = new BrowserWindow(options);
--- a/electron/filenames.gni
+++ b/electron/filenames.gni
@@ -6,7 +6,6 @@
   ]
 
   default_app_static_sources = [
-    "default_app/icon.png",
     "default_app/index.html",
     "default_app/package.json",
     "default_app/styles.css",
