--- ./sandbox/linux/syscall_broker/broker_file_permission_unittest.cc.orig
+++ ./sandbox/linux/syscall_broker/broker_file_permission_unittest.cc
@@ -134,7 +134,7 @@
 #endif
 
   const int kNumberOfBitsInOAccMode = 2;
-  static_assert(O_ACCMODE == ((1 << kNumberOfBitsInOAccMode) - 1),
+  static_assert(O_ACCMODE == (((1 << kNumberOfBitsInOAccMode) - 1) | O_PATH),
                 "incorrect number of bits");
   // check every possible flag and act accordingly.
   // Skipping AccMode bits as they are present in every case.
