From 6f4685cff1ab8c68de98c0731bb8afaf8a05a723 Mon Sep 17 00:00:00 2001
From: knuxify <knuxify@gmail.com>
Date: Sat, 5 Apr 2025 14:10:37 +0200
Subject: [PATCH] Do not generate license file for headless-shell build

---
 headless/BUILD.gn | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/headless/BUILD.gn b/headless/BUILD.gn
index 798bb22..9d83f49 100644
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -934,10 +934,6 @@ executable("headless_shell") {
 
   deps = [ ":headless_shell_lib" ]
 
-  if (proprietary_codecs) {
-    deps += [ ":generate_headless_shell_license_file" ]
-  }
-
   if (!headless_use_embedded_resources) {
     data = [
       "$root_out_dir/headless_lib_data.pak",
-- 
2.49.0

