From 4b41417068045f11db9e7edead1447e93adb9073 Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Sat, 28 Jun 2025 18:13:59 +0200
Subject: [PATCH] Test net without vendored python

---
 net/test/python_utils.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/test/python_utils.cc b/net/test/python_utils.cc
index 2cdc07dad9948..0b2d42a5bf126 100644
--- a/net/test/python_utils.cc
+++ b/net/test/python_utils.cc
@@ -47,7 +47,7 @@ bool GetPython3Command(base::CommandLine* python_cmd) {
 #if BUILDFLAG(IS_WIN)
   python_cmd->SetProgram(base::FilePath(FILE_PATH_LITERAL("vpython3.bat")));
 #else
-  python_cmd->SetProgram(base::FilePath(FILE_PATH_LITERAL("vpython3")));
+  python_cmd->SetProgram(base::FilePath(FILE_PATH_LITERAL("python3")));
 #endif
 
 #if BUILDFLAG(IS_MAC)
