--- a/chrome/browser/process_singleton_posix.cc
+++ b/chrome/browser/process_singleton_posix.cc
@@ -607,7 +607,7 @@
   // |reader| is for sending back ACK message.
   void HandleMessage(const std::string& current_dir,
                      const std::vector<std::string>& argv,
-                     const std::vector<const uint8_t> additional_data,
+                     const std::vector<uint8_t> additional_data,
                      SocketReader* reader);
 
  private:
@@ -664,7 +664,7 @@
 void ProcessSingleton::LinuxWatcher::HandleMessage(
     const std::string& current_dir,
     const std::vector<std::string>& argv,
-    const std::vector<const uint8_t> additional_data,
+    const std::vector<uint8_t> additional_data,
     SocketReader* reader) {
   DCHECK(ui_task_runner_->BelongsToCurrentThread());
   DCHECK(reader);
@@ -754,7 +754,7 @@
   base::StringToSizeT(tokens[0], &num_args);
   std::vector<std::string> command_line(tokens.begin() + 1, tokens.begin() + 1 + num_args);
 
-  std::vector<const uint8_t> additional_data;
+  std::vector<uint8_t> additional_data;
   if (tokens.size() >= 3 + num_args) {
     size_t additional_data_size;
     base::StringToSizeT(tokens[1 + num_args], &additional_data_size);
@@ -763,7 +763,7 @@
         std::string(1, kTokenDelimiter));
     const uint8_t* additional_data_bits =
         reinterpret_cast<const uint8_t*>(remaining_args.c_str());
-    additional_data = std::vector<const uint8_t>(
+    additional_data = std::vector<uint8_t>(
         additional_data_bits, additional_data_bits + additional_data_size);
   }
 
--- a/chrome/browser/process_singleton.h
+++ b/chrome/browser/process_singleton.h
@@ -102,7 +102,7 @@
   using NotificationCallback =
        base::RepeatingCallback<bool(const base::CommandLine& command_line,
                                    const base::FilePath& current_directory,
-                                   const std::vector<const uint8_t> additional_data)>;
+                                   const std::vector<uint8_t> additional_data)>;
 
 #if BUILDFLAG(IS_WIN)
   ProcessSingleton(const std::string& program_name,
--- a/chrome/browser/process_singleton_win.cc
+++ b/chrome/browser/process_singleton_win.cc
@@ -81,7 +81,7 @@
 bool ParseCommandLine(const COPYDATASTRUCT* cds,
                       base::CommandLine* parsed_command_line,
                       base::FilePath* current_directory,
-                      std::vector<const uint8_t>* parsed_additional_data) {
+                      std::vector<uint8_t>* parsed_additional_data) {
   // We should have enough room for the shortest command (min_message_size)
   // and also be a multiple of wchar_t bytes. The shortest command
   // possible is L"START\0\0" (empty command line, current directory,
@@ -163,7 +163,7 @@
         msg.substr(fourth_null + 1, fifth_null - fourth_null);
     const uint8_t* additional_data_bytes =
         reinterpret_cast<const uint8_t*>(additional_data.c_str());
-    *parsed_additional_data = std::vector<const uint8_t>(additional_data_bytes,
+    *parsed_additional_data = std::vector<uint8_t>(additional_data_bytes,
         additional_data_bytes + additional_data_length);
 
     return true;
@@ -187,7 +187,7 @@
 
   base::CommandLine parsed_command_line(base::CommandLine::NO_PROGRAM);
   base::FilePath current_directory;
-  std::vector<const uint8_t> additional_data;
+  std::vector<uint8_t> additional_data;
   if (!ParseCommandLine(cds, &parsed_command_line, &current_directory, &additional_data)) {
     *result = TRUE;
     return true;
--- a/electron/shell/browser/api/electron_api_app.cc
+++ b/electron/shell/browser/api/electron_api_app.cc
@@ -519,10 +519,10 @@
     const base::RepeatingCallback<
         void(const base::CommandLine& command_line,
              const base::FilePath& current_directory,
-             const std::vector<const uint8_t> additional_data)>& callback,
+             const std::vector<uint8_t> additional_data)>& callback,
     const base::CommandLine& cmd,
     const base::FilePath& cwd,
-    const std::vector<const uint8_t> additional_data) {
+    const std::vector<uint8_t> additional_data) {
   // Make sure the callback is called after app gets ready.
   if (Browser::Get()->is_ready()) {
     callback.Run(cmd, cwd, std::move(additional_data));
@@ -1082,7 +1082,7 @@
 
 void App::OnSecondInstance(const base::CommandLine& cmd,
                            const base::FilePath& cwd,
-                           const std::vector<const uint8_t> additional_data) {
+                           const std::vector<uint8_t> additional_data) {
   v8::Isolate* isolate = JavascriptEnvironment::GetIsolate();
   v8::Locker locker(isolate);
   v8::HandleScope handle_scope(isolate);
--- a/electron/shell/browser/api/electron_api_app.h
+++ b/electron/shell/browser/api/electron_api_app.h
@@ -195,7 +195,7 @@
   std::string GetLocaleCountryCode();
   void OnSecondInstance(const base::CommandLine& cmd,
                         const base::FilePath& cwd,
-                        const std::vector<const uint8_t> additional_data);
+                        const std::vector<uint8_t> additional_data);
   bool HasSingleInstanceLock() const;
   bool RequestSingleInstanceLock(gin::Arguments* args);
   void ReleaseSingleInstanceLock();
