diff --git a/src/date@.sh b/src/date@.sh
index d5f40ae..9f9f5a9 100644
--- a/src/date@.sh
+++ b/src/date@.sh
@@ -7,4 +7,4 @@ else
 fi
 host=${1-0}
 port=13
-HOME/command/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec HOME/command/delcr <&6' | cat -v
+HOME/bin/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec HOME/bin/delcr <&6' | cat -v
diff --git a/src/finger@.sh b/src/finger@.sh
index 08853f5..a143445 100644
--- a/src/finger@.sh
+++ b/src/finger@.sh
@@ -8,7 +8,7 @@ fi
 host=${1-0}
 user=${2-}
 port=79
-echo "$user" | HOME/command/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c '
+echo "$user" | HOME/bin/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c '
   HOME/bin/addcr >&7
-  exec HOME/command/delcr <&6
+  exec HOME/bin/delcr <&6
 ' | cat -v
diff --git a/src/http@.sh b/src/http@.sh
index 28c67fa..5c0de0e 100644
--- a/src/http@.sh
+++ b/src/http@.sh
@@ -16,7 +16,7 @@ then
 fi
 echo "GET /$path HTTP/1.0
 Host: $host:$port
-" | HOME/command/tcpclient -RHl0 "$vers" $args -- "$host" "$port" sh -c '
-  HOME/command/addcr >&7
-  exec HOME/command/delcr <&6
+" | HOME/bin/tcpclient -RHl0 "$vers" $args -- "$host" "$port" sh -c '
+  HOME/bin/addcr >&7
+  exec HOME/bin/delcr <&6
 ' | awk '/^$/ { body=1; next } { if (body) print }'
diff --git a/src/mconnect.sh b/src/mconnect.sh
index 8842861..53d866a 100644
--- a/src/mconnect.sh
+++ b/src/mconnect.sh
@@ -7,4 +7,4 @@ else
 fi
 host=${1-0}
 port=${2-25}
-exec HOME/command/tcpclient -RHl0 "$vers" -- "$host" "$port" HOME/command/mconnect-io
+exec HOME/bin/tcpclient -RHl0 "$vers" -- "$host" "$port" HOME/bin/mconnect-io
diff --git a/src/tcpcat.sh b/src/tcpcat.sh
index c9d303f..27be71c 100644
--- a/src/tcpcat.sh
+++ b/src/tcpcat.sh
@@ -7,4 +7,4 @@ else
 fi
 host=${1-0}
 port=${2-17}
-exec HOME/command/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec cat <&6'
+exec HOME/bin/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec cat <&6'
diff --git a/src/who@.sh b/src/who@.sh
index 83f796d..c238186 100644
--- a/src/who@.sh
+++ b/src/who@.sh
@@ -7,4 +7,4 @@ else
 fi
 host=${1-0}
 port=11
-HOME/command/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec HOME/command/delcr <&6' | cat -v
+HOME/bin/tcpclient -RHl0 "$vers" -- "$host" "$port" sh -c 'exec HOME/bin/delcr <&6' | cat -v
