diff -ur a/com32/MCONFIG b/com32/MCONFIG
--- a/com32/MCONFIG	2008-09-26 01:46:02.000000000 +0200
+++ b/com32/MCONFIG	2008-10-28 16:10:16.107964907 +0100
@@ -19,6 +19,8 @@
 GCCOPT := $(call gcc_ok,-std=gnu99,) \
 	  $(call gcc_ok,-m32,) \
 	  $(call gcc_ok,-fno-stack-protector,) \
+	  $(call gcc_ok,-nopie,) \
+	  $(call gcc_ok,-fno-pie,) \
 	  -mregparm=3 -DREGPARM=3 -march=i386 -Os
 
 com32 = $(topdir)/com32
