diff -Nurp a/dmd/test/runnable/gdb4181.d b/dmd/test/runnable/gdb4181.d
deleted file mode 100644
--- a/dmd/test/runnable/gdb4181.d
+++ /dev/null
@@ -1,23 +0,0 @@
-/*
-REQUIRED_ARGS: -g
-PERMUTE_ARGS:
-GDB_SCRIPT:
----
-b 22
-r
-echo RESULT=
-p 'gdb.x' + 'gdb.STest.y'
----
-GDB_MATCH: RESULT=.*33
-*/
-module gdb;
-
-int x;
-struct STest { static int y; }
-
-void main()
-{
-    x = 11;
-    STest.y = 22;
-    // BP
-}
