diff --git a/joe/main.c b/joe/main.c
index 8df8174..df9792c 100644
--- a/joe/main.c
+++ b/joe/main.c
@@ -755,7 +755,7 @@ int main(int argc, char **real_argv, const char * const *envv)
 			/* Close stdin, so that if user kills the copying the program feeding stdin sees a SIGPIPE */
 			/* It won't if there are any extra file descriptors open */
 			fclose(stdin);
-			stdin = fopen("/dev/tty", "rb");
+			freopen("/dev/tty", "rb", stdin);
 		}
 	}
 
