From c6a6f06f3863bab89467ecb900dcd7591516da49 Mon Sep 17 00:00:00 2001
From: Robert Mader <robert.mader@collabora.com>
Date: Thu, 12 Mar 2026 10:55:33 +0100
Subject: [PATCH] cargo-auditable

---
 meson.build     | 1 +
 src/meson.build | 1 +
 2 files changed, 2 insertions(+)

diff --git a/meson.build b/meson.build
index f4cf80e..9f8b4ba 100644
--- a/meson.build
+++ b/meson.build
@@ -32,6 +32,7 @@ glib_compile_schemas = find_program('glib-compile-schemas', required: true)
 desktop_file_validate = find_program('desktop-file-validate', required: false)
 appstreamcli = find_program('appstreamcli', required: false)
 cargo = find_program('cargo', required: true)
+cargo_auditable = find_program('cargo-auditable', required: true)
 
 version = meson.project_version()
 
diff --git a/src/meson.build b/src/meson.build
index 35b98a1..c7fc059 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -44,6 +44,7 @@ cargo_build = custom_target(
   env: cargo_env,
   command: [
     cargo,
+    'auditable',
     'build',
     cargo_build_options,
     '&&',
-- 
2.53.0

