diff --git a/src/core/screen.c b/src/core/screen.c
index c5962cb..626bbc1 100644
--- a/src/core/screen.c
+++ b/src/core/screen.c
@@ -2664,8 +2664,8 @@ collect_timed_out_foreach (void *element,
   SnStartupSequence *sequence = element;
   double elapsed;
 
-  time_t tv_sec;
-  suseconds_t  tv_usec;
+  long tv_sec;
+  long tv_usec;
   gint64 tv;
 
   sn_startup_sequence_get_last_active_time (sequence, &tv_sec, &tv_usec);
