--- ./configure.in.orig
+++ ./configure.in
@@ -77,9 +77,6 @@
 	AC_MSG_RESULT(${GCC_version})
 fi
 
-dnl more automake stuff
-AM_C_PROTOTYPES
-
 AC_CHECK_HEADER(dmalloc.h)
 if test "${ac_cv_header_dmalloc_h}" = "yes"; then
 	AM_WITH_DMALLOC
--- ./doc/html/Makefile.am.orig
+++ ./doc/html/Makefile.am
@@ -1,7 +1,5 @@
 ## $Id: Makefile.am,v 1.7 2010/09/30 20:58:40 jaz001 Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
-
 docdir = $(datadir)/doc
 pkgdocdir = $(docdir)/$(PACKAGE)
 htmldir = $(docdir)/html
--- ./doc/man/Makefile.am.orig
+++ ./doc/man/Makefile.am
@@ -1,7 +1,5 @@
 ## $Id: Makefile.am,v 1.1 2000/10/22 11:39:44 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign ansi2knr
-
 man_MANS = lame.1
 EXTRA_DIST = ${man_MANS}
 
--- ./libmp3lame/i386/Makefile.am.orig
+++ ./libmp3lame/i386/Makefile.am
@@ -1,7 +1,5 @@
 ## $Id: Makefile.am,v 1.26 2011/04/04 09:42:34 aleidinger Exp $
 
-AUTOMAKE_OPTIONS = foreign $(top_srcdir)/ansi2knr
-
 DEFS = @DEFS@ @CONFIG_DEFS@
 
 ECHO ?= echo
