From ac3348c41541f04e4794ee2ed264e8a6f6543a51 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kunkku@alpinelinux.org>
Date: Sat, 6 May 2023 00:43:19 +0300
Subject: [PATCH 4/4] bgpd: fix parsing of MP EOR marker

---
 bgpd/bgp_packet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bgpd/bgp_packet.c b/bgpd/bgp_packet.c
index 5a8b18a3..36fe7f76 100644
--- a/bgpd/bgp_packet.c
+++ b/bgpd/bgp_packet.c
@@ -1972,7 +1972,7 @@ bgp_update_receive (struct peer *peer, bgp_size_t size)
       /* otherwise MP AFI/SAFI is an empty update, other than an empty
        * MP_UNREACH_NLRI attr (with an AFI/SAFI we recognise).
        */
-      else if (attr.flag == BGP_ATTR_MP_UNREACH_NLRI
+      else if (attr.flag & ATTR_FLAG_BIT(BGP_ATTR_MP_UNREACH_NLRI)
                && nlris[NLRI_MP_WITHDRAW].length == 0
                && bgp_afi_safi_valid_indices (nlris[NLRI_MP_WITHDRAW].afi,
                                               &nlris[NLRI_MP_WITHDRAW].safi))
-- 
2.40.1

