--- quagga-0.99.23.orig/configure.ac	2014-06-24 08:14:20.000000000 -0300
+++ quagga-0.99.23/configure.ac	2014-07-29 12:58:52.201839366 -0300
@@ -1202,18 +1202,8 @@
   fi
 
   if test "$zebra_cv_linux_ipv6" = "yes";then
-    AC_MSG_CHECKING(for GNU libc >= 2.1)
     AC_DEFINE(HAVE_IPV6,1,Linux IPv6)
-    AC_EGREP_CPP(yes, [
-#include <features.h>
-#if __GLIBC__ >= 2 && __GLIBC_MINOR__ >= 1
-  yes
-#endif], 
-      [glibc=yes
-       AC_DEFINE(LINUX_IPV6,1,Linux IPv6 stack)
-       AC_MSG_RESULT(yes)], 
-      AC_MSG_RESULT(no)
-    )
+    AC_DEFINE(LINUX_IPV6,1,Linux IPv6 stack)
     RIPNGD="ripngd"
     OSPF6D="ospf6d"
     if test "$glibc" != "yes"; then
