diff -ur quagga-0.99.11.orig/lib/Makefile.am quagga-0.99.11/lib/Makefile.am
--- quagga-0.99.11.orig/lib/Makefile.am	2008-09-24 15:22:43.000000000 +0000
+++ quagga-0.99.11/lib/Makefile.am	2008-10-09 20:29:17.000000000 +0000
@@ -18,7 +18,7 @@
 
 libzebra_la_DEPENDENCIES = @LIB_REGEX@
 
-libzebra_la_LIBADD = @LIB_REGEX@
+libzebra_la_LIBADD = @LIB_REGEX@ @LIBCAP@
 
 pkginclude_HEADERS = \
 	buffer.h checksum.h command.h filter.h getopt.h hash.h \
diff -ur quagga-0.99.11.orig/zebra/Makefile.am quagga-0.99.11/zebra/Makefile.am
--- quagga-0.99.11.orig/zebra/Makefile.am	2008-09-05 14:27:26.000000000 +0000
+++ quagga-0.99.11/zebra/Makefile.am	2008-10-09 20:29:17.000000000 +0000
@@ -39,7 +39,7 @@
 	connected.h ioctl.h rib.h rt.h zserv.h redistribute.h debug.h rtadv.h \
 	interface.h ipforward.h irdp.h router-id.h kernel_socket.h
 
-zebra_LDADD = $(otherobj) $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
+zebra_LDADD = $(otherobj) ../lib/libzebra.la $(LIBCAP) $(LIB_IPV6)
 
 testzebra_LDADD = $(LIBCAP) $(LIB_IPV6) ../lib/libzebra.la
 
