--- quagga-0.99.11.DIST/bgpd/bgp_fsm.c	2008-09-09 16:18:57.000000000 -0400
+++ quagga-0.99.11/bgpd/bgp_fsm.c	2009-04-13 13:09:25.000000000 -0400
@@ -1013,11 +1013,11 @@
   {
     /* Clearing, */
     {bgp_ignore,  Clearing},	/* BGP_Start                    */
-    {bgp_ignore,  Clearing},	/* BGP_Stop                     */
-    {bgp_ignore,  Clearing},	/* TCP_connection_open          */
-    {bgp_ignore,  Clearing},	/* TCP_connection_closed        */
-    {bgp_ignore,  Clearing},	/* TCP_connection_open_failed   */
-    {bgp_ignore,  Clearing},	/* TCP_fatal_error              */
+    {bgp_stop,    Clearing},	/* BGP_Stop                     */
+    {bgp_stop,    Clearing},	/* TCP_connection_open          */
+    {bgp_stop,    Clearing},	/* TCP_connection_closed        */
+    {bgp_stop,    Clearing},	/* TCP_connection_open_failed   */
+    {bgp_stop,    Clearing},	/* TCP_fatal_error              */
     {bgp_ignore,  Clearing},	/* ConnectRetry_timer_expired   */
     {bgp_ignore,  Clearing},	/* Hold_Timer_expired           */
     {bgp_ignore,  Clearing},	/* KeepAlive_timer_expired      */
