diff --git a/src/bs2bconvert.c b/src/bs2bconvert.c
index 24531b2..5e8c992 100644
--- a/src/bs2bconvert.c
+++ b/src/bs2bconvert.c
@@ -153,7 +153,7 @@ int main( int argc, char *argv[] )
 	if( ( infile = sf_open( infilename, SFM_READ, &sfinfo ) ) == NULL )
 	{
 		printf( "Not able to open input file %s.\n", infilename );
-		printf( sf_strerror( NULL ) );
+		printf( "%s", sf_strerror( NULL ) );
 		return 1;
 	}
 
