diff --git a/src/gtimelog/tests/test_timelog.py b/src/gtimelog/tests/test_timelog.py
index f91ae5a05f49..de9e39e080e2 100644
--- a/src/gtimelog/tests/test_timelog.py
+++ b/src/gtimelog/tests/test_timelog.py
@@ -483,7 +483,7 @@ def doctest_TimeWindow_last_entry():
         >>> stop
         datetime.datetime(2013, 12, 4, 9, 31)
         >>> duration
-        datetime.timedelta(0, 1860)
+        datetime.timedelta(seconds=1860)
         >>> entry
         'gtimelog: tests'
 
