diff --git a/lib/dnscore/include/dnscore/dnskey.h b/lib/dnscore/include/dnscore/dnskey.h
index 98d4347..af7705b 100644
--- a/lib/dnscore/include/dnscore/dnskey.h
+++ b/lib/dnscore/include/dnscore/dnskey.h
@@ -95,8 +95,6 @@
 //
 #if OPENSSL_VERSION_NUMBER < 0x10100000L
 #define SSL_API_LT_110 1
-#elif defined(LIBRESSL_VERSION_NUMBER)
-#define SSL_API_LT_110 1
 #else
 #define SSL_API_LT_110 0
 #endif
