diff --git a/lib/dnscore/src/signals.c b/lib/dnscore/src/signals.c
index b1dd816..b0480fb 100644
--- a/lib/dnscore/src/signals.c
+++ b/lib/dnscore/src/signals.c
@@ -57,7 +57,6 @@
 
 #if defined(__linux__) || defined(__gnu_hurd__)
 #define _GNU_SOURCE 1
-#include <execinfo.h>
 #include <sys/mman.h>
 #include <ucontext.h>
 #elif defined(__sun)
@@ -696,8 +695,8 @@ signal_handler(int signo, siginfo_t* info, void* context)
 
 #if defined(__GLIBC__) || defined(__gnu_hurd__)
                     void* buffer[MAXTRACE];
-                    char** strings;
-                    int n = backtrace(buffer, MAXTRACE);
+                    char** strings = NULL;
+                    int n = 0;
                     int i;
                     time_t now = time(NULL);
 
@@ -827,8 +826,6 @@ signal_handler(int signo, siginfo_t* info, void* context)
 #endif
                     
 #endif // linux
-                    strings = backtrace_symbols(buffer, n);
-
                     if(strings != NULL)
                     {
                         for(i = 0; i < n; i++)
