diff --git a/lib/dnscore/src/signals.c b/lib/dnscore/src/signals.c
index b1dd816..b0480fb 100644
--- a/lib/dnscore/src/signals.c
+++ b/lib/dnscore/src/signals.c
@@ -827,8 +826,6 @@ signal_handler(int signo, siginfo_t* info, void* context)
 #endif
                     
 #endif // linux
-                    strings = backtrace_symbols(buffer, n);
-
                     if(strings != NULL)
                     {
                         for(i = 0; i < n; i++)
