diff --git a/sbin/yadifad/poll-util.h b/sbin/yadifad/poll-util.h
index ef9a004..41d4d0c 100644
--- a/sbin/yadifad/poll-util.h
+++ b/sbin/yadifad/poll-util.h
@@ -38,11 +38,10 @@
  *
  ******************************************************************************************************************/
 
+#include <poll.h>
 #ifndef _POLL_H
 #define _POLL_H
 
-#include <poll.h>
-
 /**
  * This code maintains the limit on the tcp sockets
  * @todo Portability has to be tested.
