--- a/configure
+++ b/configure
@@ -27521,9 +27521,6 @@ ac_res=$ac_cv_search_backtrace
 if test "$ac_res" != no
 then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
-
-else $as_nop
-  exit 1
 fi
 
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for library containing backtrace_symbols" >&5
@@ -27582,9 +27579,6 @@ ac_res=$ac_cv_search_backtrace_symbols
 if test "$ac_res" != no
 then :
   test "$ac_res" = "none required" || LIBS="$ac_res $LIBS"
-
-else $as_nop
-  exit 1
 fi
 
 
