diff -U4 -r qmltermwidget-0.2.0--orig/lib/TerminalCharacterDecoder.cpp qmltermwidget-0.2.0--patched/lib/TerminalCharacterDecoder.cpp
--- qmltermwidget-0.2.0--orig/lib/TerminalCharacterDecoder.cpp	2018-11-24 22:06:16.000000000 +0100
+++ qmltermwidget-0.2.0--patched/lib/TerminalCharacterDecoder.cpp	2020-01-31 10:03:00.168118919 +0100
@@ -21,8 +21,11 @@
 
 // Own
 #include "TerminalCharacterDecoder.h"
 
+// stdlib
+#include <cwctype>
+
 // Qt
 #include <QTextStream>
 
 // KDE
