From 090efa5ea63003735f8c055c95e560d261003476 Mon Sep 17 00:00:00 2001
From: Florent Revest <revestflo@gmail.com>
Date: Wed, 20 Jan 2016 19:27:19 +0100
Subject: [PATCH] Fixes build and avoid md5sum mismatch with GMT timezones

---
 Makefile                  |  4 ++--
 scripts/find-aliases.perl | 15 ++++++++-------
 2 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/Makefile b/Makefile
index 8a7c0e2..a8a880a 100644
--- a/Makefile
+++ b/Makefile
 # Create time zones according to ISO 8601.
 iso8601zones: createdirs
@@ -52,7 +52,7 @@ find-aliases: process-zones list-zones-without-links
 	perl -w -s $(SCRIPTS_DIR)/find-aliases.perl \
 	-zones=$(BUILD_DIR)/zone.list -signatures=$(BUILD_DIR)/signatures \
 	-md5sum=$(BUILD_DIR)/md5sum -links=$(BUILD_DIR)/zone.link \
-	-zonetab=/usr/share/zoneinfo/zone.tab > $(BUILD_DIR)/zone.alias
+	-zonetab=$(SRC_DIR)/zone.tab > $(BUILD_DIR)/zone.alias
 
 prepare-timed-data: find-aliases
 	$(SCRIPTS_DIR)/prepare-timed-data.perl \
diff --git a/scripts/find-aliases.perl b/scripts/find-aliases.perl
index 3a1ab1d..22dcb1a 100755
--- a/scripts/find-aliases.perl
+++ b/scripts/find-aliases.perl
@@ -234,13 +234,14 @@ my $zulu = "Iso8601/+0000" ;
 add_alias_manually($zulu, $_) for qw/UTC GMT UCT Universal Greenwich Zulu GMT+0 GMT-0 GMT0/ ;
 add_alias_manually($zulu, "Etc/$_") for qw/UTC GMT UCT Universal Greenwich Zulu GMT+0 GMT-0 GMT0/ ;
 
-foreach my $h (-14 .. 12)
-{
-  next unless $h ;
-  my $iso = sprintf "Iso8601/%s%02d00", (-$h < 0 ? "-" : "+"), abs($h) ;
-  my $posix = sprintf "Etc/GMT%s%d", ($h < 0 ? "" : "+"), $h ;
-  add_alias_manually($iso, $posix) ;
-}
+# TODO: For some reasons those timezones don't match, we should try to fix that
+#foreach my $h (-14 .. 12)
+#{
+#  next unless $h ;
+#  my $iso = sprintf "Iso8601/%s%02d00", (-$h < 0 ? "-" : "+"), abs($h) ;
+#  my $posix = sprintf "Etc/GMT%s%d", ($h < 0 ? "" : "+"), $h ;
+#  add_alias_manually($iso, $posix) ;
+#}
 
 print STDERR "WARNING: unsupported zones ", join(", ", sort keys %$unsupported), "\n" ;
 
-- 
2.7.0.rc3

