
diff --git a/src/timer/src/timer.cpp b/src/timer/src/timer.cpp
index fb6a7ec..61221be 100644
--- a/src/timer/src/timer.cpp
+++ b/src/timer/src/timer.cpp
@@ -168,7 +168,7 @@ PrivateTimer::PrivateTimer () :
     mMaxTime (0),
     mMinDeadline (0),
     mMaxDeadline (0),
-    mCallBack (NULL)
+    mCallBack ([]() -> int { return 0; })
 {
 }
 
diff --git a/plugins/mousepoll/src/mousepoll.cpp b/plugins/mousepoll/src/mousepoll.cpp
index 7b7e717..147a591 100644
--- a/plugins/mousepoll/src/mousepoll.cpp
+++ b/plugins/mousepoll/src/mousepoll.cpp
@@ -201,7 +201,7 @@ MousePoller::getPosition ()
 MousePoller::MousePoller () :
     mActive (false),
     mPoint (0, 0),
-    mCallback (NULL)
+    mCallback ([](const CompPoint &) { })
 {
 }
 
