diff --git a/compizconfig/gsettings/mocks/CMakeLists.txt b/compizconfig/gsettings/mocks/CMakeLists.txt
index 17fe51e..d105bb1 100644
--- a/compizconfig/gsettings/mocks/CMakeLists.txt
+++ b/compizconfig/gsettings/mocks/CMakeLists.txt
@@ -8,7 +8,7 @@ if (GLIB_FOUND)
 			 ${CMAKE_CURRENT_SOURCE_DIR}/../gsettings_backend_shared
 			 ${GLIB_INCLUDE_DIRS})
 
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
     add_library (compizconfig_ccs_gsettings_wrapper_mock STATIC
 	         ${CMAKE_CURRENT_SOURCE_DIR}/ccs_gsettings_wrapper_mock.cpp
diff --git a/compizconfig/gsettings/tests/CMakeLists.txt b/compizconfig/gsettings/tests/CMakeLists.txt
index 9a81a5d..2e3bbb0 100644
--- a/compizconfig/gsettings/tests/CMakeLists.txt
+++ b/compizconfig/gsettings/tests/CMakeLists.txt
@@ -16,7 +16,7 @@ pkg_check_modules (COMPIZCONFIG_TEST_GSETTINGS gio-2.0 glib-2.0)
 
 if (COMPIZCONFIG_TEST_GSETTINGS_FOUND)
 
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
     link_directories (${CMAKE_CURRENT_BINARY_DIR}/../../libcompizconfig/tests)
     link_directories (${CMAKE_CURRENT_BINARY_DIR}/../testenvironments)
diff --git a/compizconfig/integration/gnome/gsettings/tests/CMakeLists.txt b/compizconfig/integration/gnome/gsettings/tests/CMakeLists.txt
index 683c83f..7778503 100644
--- a/compizconfig/integration/gnome/gsettings/tests/CMakeLists.txt
+++ b/compizconfig/integration/gnome/gsettings/tests/CMakeLists.txt
@@ -11,7 +11,7 @@ link_directories (${CMAKE_CURRENT_BINARY_DIR}/../../mocks/libcompizconfig)
 link_directories (${CMAKE_CURRENT_BINARY_DIR}/../../tests/)
 link_directories (${CMAKE_CURRENT_BINARY_DIR}/../src/)
 
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
 add_executable (compizconfig_test_ccs_gnome_gsettings_integrated_setting
 		${CMAKE_CURRENT_SOURCE_DIR}/compizconfig_test_ccs_gnome_gsettings_integrated_setting.cpp)
diff --git a/compizconfig/integration/gnome/tests/CMakeLists.txt b/compizconfig/integration/gnome/tests/CMakeLists.txt
index 30fd552..26fdff1 100644
--- a/compizconfig/integration/gnome/tests/CMakeLists.txt
+++ b/compizconfig/integration/gnome/tests/CMakeLists.txt
@@ -8,7 +8,7 @@ include_directories (${CMAKE_CURRENT_SOURCE_DIR}/../../../gsettings/mocks)
 include_directories (${CMAKE_SOURCE_DIR}/compizconfig/tests)
 link_directories (${CMAKE_INSTALL_PREFIX}/lib)
 
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
 add_library (compizconfig_ccs_mock_gnome_integrated_setting_composition
 	     ${CMAKE_CURRENT_SOURCE_DIR}/compizconfig_ccs_mock_gnome_integrated_setting_composition.c)
diff --git a/compizconfig/libcompizconfig/tests/CMakeLists.txt b/compizconfig/libcompizconfig/tests/CMakeLists.txt
index a3a42f9..78446f4 100644
--- a/compizconfig/libcompizconfig/tests/CMakeLists.txt
+++ b/compizconfig/libcompizconfig/tests/CMakeLists.txt
@@ -9,7 +9,7 @@ link_directories (${CMAKE_CURRENT_BINARY_DIR}/../../mocks/libcompizconfig)
 link_directories (${CMAKE_CURRENT_BINARY_DIR}/../../tests/)
 link_directories (${CMAKE_CURRENT_BINARY_DIR}/../src/)
 
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
 
 add_definitions (-DCONFIGDIR="${COMPIZCONFIG_CONFIG_DIR}")
 
diff --git a/cmake/CompizCommon.cmake b/cmake/CompizCommon.cmake
index b567cfc..d06e86b 100644
--- a/cmake/CompizCommon.cmake
+++ b/cmake/CompizCommon.cmake
@@ -54,7 +54,7 @@ set (
 
 # Almost everything is a shared library now, so almost everything needs -fPIC
 set (COMMON_FLAGS "-fPIC -Wall")
-set (COMMON_CXX_FLAGS "-std=c++11")
+set (COMMON_CXX_FLAGS "-std=c++17")
 
 option (COMPIZ_SUBOBJECT_LINKAGE_WARNINGS "Warn about subobject linkage" OFF)
 if (NOT COMPIZ_SUBOBJECT_LINKAGE_WARNINGS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index fcf974f..997d050 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -16,7 +16,6 @@ add_subdirectory( servergrab )
 IF (COMPIZ_BUILD_TESTING)
 add_subdirectory( privatescreen/tests )
 add_subdirectory( wrapsystem/tests )
-add_subdirectory( plugin )
 add_subdirectory( option/tests )
 add_subdirectory( tests )
 ENDIF (COMPIZ_BUILD_TESTING)
diff --git a/tests/acceptance-tests/autopilot/CMakeLists.txt b/tests/acceptance-tests/autopilot/CMakeLists.txt
index 9999379..d6e1f7c 100644
--- a/tests/acceptance-tests/autopilot/CMakeLists.txt
+++ b/tests/acceptance-tests/autopilot/CMakeLists.txt
@@ -7,6 +7,3 @@ set (COMPIZ_AUTOPILOT_ACCEPTANCE_TEST_LIBRARIES
 
 target_link_libraries (compiz_autopilot_acceptance_tests
                        ${COMPIZ_AUTOPILOT_ACCEPTANCE_TEST_LIBRARIES})
-
-install (TARGETS compiz_autopilot_acceptance_tests
-         RUNTIME DESTINATION ${exec_prefix})
