diff --git a/compizconfig/libcompizconfig/src/CMakeLists.txt b/compizconfig/libcompizconfig/src/CMakeLists.txt
index 53d92f3..b740d06 100644
--- a/compizconfig/libcompizconfig/src/CMakeLists.txt
+++ b/compizconfig/libcompizconfig/src/CMakeLists.txt
@@ -40,6 +40,7 @@ if (HAVE_PROTOBUF)
 	 ${LIBCOMPIZCONFIG_FILES}
 	 ${CMAKE_CURRENT_BINARY_DIR}/compizconfig.pb.cc
 	)
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17")
     set (LIBCOMPIZCONFIG_LIBRARIES
 	 ${LIBCOMPIZCONFIG_LIBRARIES}
 	 protobuf)
