--- src.orig/unetbootin.cpp
+++ src/unetbootin.cpp
@@ -572,7 +572,7 @@
                     {
 						if (!volidcommand.isEmpty())
 						{
-							if (QString(callexternapp(volidcommand, QString("-t %2").arg(usbfileinfoL.at(i).canonicalFilePath()))).contains(QRegExp("(vfat|ext2|ext3|ext4)")))
+							if (QString(callexternapp(volidcommand, QString("-t %2").arg(usbfileinfoL.at(i).canonicalFilePath()))).contains(QRegExp("(vfat|ntfs|ext2|ext3|ext4)")))
 								fulldrivelist.append(usbfileinfoL.at(i).canonicalFilePath());
 						}
 						else
@@ -580,7 +580,7 @@
 							QString tstrblk = QString(callexternapp(blkidcommand, QString("-s TYPE %2").arg(usbfileinfoL.at(i).canonicalFilePath())));
 							if (tstrblk.contains('='))
 							{
-								if (tstrblk.section('=', -1, -1).remove('"').contains(QRegExp("(vfat|ext2|ext3|ext4)")))
+								if (tstrblk.section('=', -1, -1).remove('"').contains(QRegExp("(vfat|ntfs|ext2|ext3|ext4)")))
 									fulldrivelist.append(usbfileinfoL.at(i).canonicalFilePath());
 							}
 						}
