diff --git a/examples/c++-lib/Makefile.am b/examples/c++-lib/Makefile.am
index 52c81e5..715de51 100644
--- a/examples/c++-lib/Makefile.am
+++ b/examples/c++-lib/Makefile.am
@@ -5,7 +5,7 @@
 AM_CPPFLAGS = -I$(top_srcdir)
 
 LDADD = ../../snapper/libsnapper.la
-AM_LDFLAGS = -lboost_system
+AM_LDFLAGS = -lboost_system-mt
 
 noinst_PROGRAMS = List ListAll Create CmpDirs CreateNumber CreateTimeline
 
diff --git a/server/Makefile.am b/server/Makefile.am
index 2a6a4c2..e007710 100644
--- a/server/Makefile.am
+++ b/server/Makefile.am
@@ -14,4 +14,4 @@ snapperd_SOURCES =					\
 	Types.cc		Types.h
 
 snapperd_LDADD = ../snapper/libsnapper.la ../dbus/libdbus.la -lrt
-snapperd_LDFLAGS = -lboost_system -lboost_thread
+snapperd_LDFLAGS = -lboost_system-mt -lboost_thread-mt
diff --git a/testsuite-cmp/Makefile.am b/testsuite-cmp/Makefile.am
index 81104bf..d88644e 100644
--- a/testsuite-cmp/Makefile.am
+++ b/testsuite-cmp/Makefile.am
@@ -3,7 +3,7 @@
 #
 
 AM_CPPFLAGS = -I$(top_srcdir)
-AM_LDFLAGS = -lboost_system
+AM_LDFLAGS = -lboost_system-mt
 
 LDADD = ../snapper/libsnapper.la
 
diff --git a/snapper/Makefile.am b/snapper/Makefile.am
index 36467da..cb05a3b 100644
--- a/snapper/Makefile.am
+++ b/snapper/Makefile.am
@@ -62,7 +62,7 @@ libsnapper_la_SOURCES +=				\
 endif
 
 libsnapper_la_LDFLAGS = -version-info @LIBVERSION_INFO@
-libsnapper_la_LIBADD = -lboost_thread -lboost_system -lxml2 -lacl -lz -lm
+libsnapper_la_LIBADD = -lboost_thread-mt -lboost_system-mt -lxml2 -lacl -lz -lm
 if ENABLE_ROLLBACK
 libsnapper_la_LIBADD += -lmount
 endif
