diff --git a/snapper/AppUtil.cc b/snapper/AppUtil.cc
index 052235d..ab81983 100644
--- a/snapper/AppUtil.cc
+++ b/snapper/AppUtil.cc
@@ -223,16 +223,10 @@ namespace snapper
     string
     stringerror(int errnum)
     {
-#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE
 	char buf1[100];
 	if (strerror_r(errno, buf1, sizeof(buf1)-1) == 0)
 	    return string(buf1);
 	return string("strerror failed");
-#else
-	char buf1[100];
-	const char* buf2 = strerror_r(errno, buf1, sizeof(buf1)-1);
-	return string(buf2);
-#endif
     }
 
 
