diff --git a/tests/CMakeLists.txt b/tests/CMakeLists.txt
index 1f0d117..8927b43 100644
--- a/tests/CMakeLists.txt
+++ b/tests/CMakeLists.txt
@@ -28,7 +28,7 @@ set_source_files_properties(${HEADERS} ${SOURCES} PROPERTIES COMPILE_FLAGS ${COM
 add_library("indicator-messages-service" STATIC ${HEADERS} ${HEADERS_GEN} ${SOURCES} ${SOURCES_GEN})
 target_include_directories("indicator-messages-service" PUBLIC ${PROJECT_DEPS_INCLUDE_DIRS})
 target_compile_definitions("indicator-messages-service" PUBLIC G_LOG_DOMAIN="Ayatana-Indicator-Messages")
-target_link_libraries("indicator-messages-service")
+add_dependencies("indicator-messages-service" "ayatana-indicator-messages-service")
 
 # test-gactionmuxer
 
