let abuild do strip, original commit 220968092461f20b1a26a159cc4927a1502029f5

--- a/py/mkenv.mk
+++ a/py/mkenv.mk
@@ -52,7 +52,6 @@ GDB = $(CROSS_COMPILE)gdb
 LD = $(CROSS_COMPILE)ld
 OBJCOPY = $(CROSS_COMPILE)objcopy
 SIZE = $(CROSS_COMPILE)size
-STRIP = $(CROSS_COMPILE)strip
 AR = $(CROSS_COMPILE)ar
 WINDRES = $(CROSS_COMPILE)windres
 
--- a/py/mkrules.mk
+++ a/py/mkrules.mk
@@ -233,11 +233,6 @@ $(BUILD)/$(PROG): $(OBJ)
 # Do not pass COPT here - it's *C* compiler optimizations. For example,
 # we may want to compile using Thumb, but link with non-Thumb libc.
 	$(Q)$(CC) -o $@ $^ $(LIB) $(LDFLAGS)
-ifndef DEBUG
-ifdef STRIP
-	$(Q)$(STRIP) $(STRIPFLAGS_EXTRA) $@
-endif
-endif
 	$(Q)$(SIZE) $$(find $(BUILD) -path "$(BUILD)/build/frozen*.o") $@
 
 clean: clean-prog
