breaks dynamic zlib on armv7? 🤨
also: no static
--
--- a/build/config/linux/BUILD.gn
+++ b/build/config/linux/BUILD.gn
@@ -15,10 +15,9 @@
 
   # Explicitly use static linking for libstdc++ and libgcc to minimize
   # dependencies.
-  ldflags += [
-    "-static-libgcc",
-    "-static-libstdc++",
-  ]
+
+  # look for libraries in /lib
+  ldflags += [ "-L/lib" ]
 
   if (sysroot != "") {
     cflags += [ "--sysroot=" + sysroot ]
