this breaks the default clang include dir
diff --git a/build/config/gcc/BUILD.gn b/build/config/gcc/BUILD.gn
index 6b941f0..e465e37 100644
--- a/build/config/gcc/BUILD.gn
+++ b/build/config/gcc/BUILD.gn
@@ -75,6 +75,6 @@ config("relative_paths") {
     # prefix matching absolute_path and hence be mapped to relative_path
     # in the debugging information, so this should actually be
     # superfluous for purposes of the debugging information.
-    "-no-canonical-prefixes",
+    #"-no-canonical-prefixes",
   ]
 }
