--- ./build/toolchain/linux/BUILD.gn.orig
+++ ./build/toolchain/linux/BUILD.gn
@@ -55,7 +55,7 @@
 }
 
 gcc_toolchain("arm") {
-  prefix = "arm-linux-gnueabihf-"
+  prefix = ""
   if (arm_toolchain_prefix != "") {
     prefix = arm_toolchain_prefix
   }
@@ -98,7 +98,7 @@
 }
 
 gcc_toolchain("arm64") {
-  prefix = "aarch64-linux-gnu-"
+  prefix = ""
   if (arm64_toolchain_prefix != "") {
     prefix = arm64_toolchain_prefix
   }
@@ -205,7 +205,7 @@
 }
 
 gcc_toolchain("x64") {
-  prefix = "x86_64-linux-gnu-"
+  prefix = ""
   if (x64_toolchain_prefix != "") {
     prefix = x64_toolchain_prefix
   }
@@ -268,7 +268,7 @@
 }
 
 gcc_toolchain("riscv64") {
-  prefix = "riscv64-linux-gnu-"
+  prefix = ""
   if (riscv64_toolchain_prefix != "") {
     prefix = riscv64_toolchain_prefix
   }
