diff --git a/src/mod/common/rfc7915/6to4.c b/src/mod/common/rfc7915/6to4.c
index fc55866..8af5deb 100644
--- a/src/mod/common/rfc7915/6to4.c
+++ b/src/mod/common/rfc7915/6to4.c
@@ -4,6 +4,7 @@
 #include <net/ip6_checksum.h>
 #include <net/udp.h>
 #include <net/tcp.h>
+#include <net/inet_dscp.h>
 
 #include "mod/common/ipv6_hdr_iterator.h"
 #include "mod/common/linux_version.h"
@@ -203,7 +204,8 @@ static verdict compute_flowix64(struct xlation *state)
 	hdr6 = pkt_ip6_hdr(&state->in);
 
 	flow4->flowi4_mark = state->in.skb->mark;
-	flow4->flowi4_tos = xlat_tos(&state->jool.globals, hdr6);
+	//flow4->flowi4_tos = xlat_tos(&state->jool.globals, hdr6);
+	flow4->flowi4_dscp = inet_dsfield_to_dscp(xlat_tos(&state->jool.globals, hdr6));
 	flow4->flowi4_scope = RT_SCOPE_UNIVERSE;
 	flow4->flowi4_proto = xlat_proto(hdr6);
 	/*
@@ -645,7 +647,8 @@ static verdict ttp64_ipv4_external(struct xlation *state)
 
 	hdr4->version = 4;
 	hdr4->ihl = 5;
-	hdr4->tos = flow4->flowi4_tos;
+//	hdr4->tos = flow4->flowi4_tos;
+	hdr4->tos = inet_dscp_to_dsfield(flow4->flowi4_dscp);
 	hdr4->tot_len = cpu_to_be16(state->out.skb->len);
 	generate_ipv4_id(state, hdr4, hdr_frag);
 	hdr4->frag_off = xlat_frag_off(hdr_frag, state);
