diff -upr a/nototools/unicode_data.py b/nototools/unicode_data.py
--- a/nototools/unicode_data.py	2016-07-18 23:42:00.000000000 +0200
+++ b/nototools/unicode_data.py	2016-12-16 01:26:52.293428181 +0100
@@ -33,7 +33,6 @@ from os import path
 import re
 import sys
 
-from fontTools.misc.py23 import unichr
 try:
   import unicodedata2 as unicodedata  # Unicode 8 compliant native lib
 except ImportError:
