diff --git a/deps/glad/CMakeLists.txt b/deps/glad/CMakeLists.txt
index 35a682e..84cd700 100644
--- a/deps/glad/CMakeLists.txt
+++ b/deps/glad/CMakeLists.txt
@@ -11,17 +11,17 @@ target_sources(
     src/glad.c
     $<$<PLATFORM_ID:Windows>:src/glad_wgl.c>
     $<$<TARGET_EXISTS:OpenGL::EGL>:src/glad_egl.c>
-    $<$<TARGET_EXISTS:OpenGL::EGL>:include/EGL/eglplatform.h>
+)
+
+target_include_directories(
+  obsglad
   PUBLIC
-    include/glad/glad.h
-    "$<$<PLATFORM_ID:Windows>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_wgl.h>"
-    "$<$<TARGET_EXISTS:OpenGL::EGL>:${CMAKE_CURRENT_SOURCE_DIR}/include/glad/glad_egl.h>"
+    "${CMAKE_CURRENT_SOURCE_DIR}/include"
+    $<$<TARGET_EXISTS:OpenGL::EGL>:/usr/include>  # Ensure EGL headers are found
 )

 target_compile_options(obsglad PRIVATE $<$<COMPILE_LANG_AND_ID:C,AppleClang,Clang>:-Wno-strict-prototypes>)

-target_include_directories(obsglad PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/include")
-
 target_link_libraries(
   obsglad
   PRIVATE $<$<NOT:$<PLATFORM_ID:Windows,Darwin>>:${CMAKE_DL_LIBS}>
