diff -urNp old/filters/latex/latex2img.py new/filters/latex/latex2img.py
--- old/filters/latex/latex2img.py	2018-06-05 15:27:44.153533130 +0200
+++ new/filters/latex/latex2img.py	2018-06-05 15:28:43.542828463 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 '''
 NAME
     latex2img - Converts LaTeX source to PNG or SVG file
diff -urNp old/filters/music/music2png.py new/filters/music/music2png.py
--- old/filters/music/music2png.py	2018-06-05 15:27:44.153533130 +0200
+++ new/filters/music/music2png.py	2018-06-05 15:28:56.502882780 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 '''
 NAME
     music2png - Converts textual music notation to classically notated PNG file
diff -urNp old/filters/unwraplatex.py new/filters/unwraplatex.py
--- old/filters/unwraplatex.py	2018-06-05 15:27:44.152533125 +0200
+++ new/filters/unwraplatex.py	2018-06-05 15:28:29.956767744 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python3
 '''
 NAME
     unwraplatex - Removes delimiters from LaTeX source text
diff -urNp old/Makefile.in new/Makefile.in
--- old/Makefile.in	2018-06-05 15:27:44.144533079 +0200
+++ new/Makefile.in	2018-06-05 15:27:56.328600132 +0200
@@ -107,7 +107,7 @@ $(DATATARGETS): % : %dir
 	$(INSTALL_DATA) $($@) $(DESTDIR)/$($<)/
 
 $(manp): %.1 : %.1.txt
-	python2 a2x.py -f manpage $<
+	python3 a2x.py -f manpage $<
 
 docs:
 	$(INSTALL) -d $(DESTDIR)/$(docdir)

