diff --git a/parso/python/errors.py b/parso/python/errors.py
index a6f3ae9..afc207d 100644
--- a/parso/python/errors.py
+++ b/parso/python/errors.py
@@ -977,8 +977,8 @@ class _CheckAssignmentRule(SyntaxRule):
         if error is not None:
             if is_namedexpr:
                 # c.f. CPython bpo-39176, should be changed in next release
-                # message = 'cannot use assignment expressions with %s' % error
-                message = 'cannot use named assignment with %s' % error
+                message = 'cannot use assignment expressions with %s' % error
+                #message = 'cannot use named assignment with %s' % error
             else:
                 cannot = "can't" if self._normalizer.version < (3, 8) else "cannot"
                 message = ' '.join([cannot, "delete" if is_deletion else "assign to", error])
