diff --git a/Makefile b/Makefile
index add307f..fc5ec1e 100644
--- a/Makefile
+++ b/Makefile
@@ -65,4 +65,4 @@ local: tar rpm
 git: gittar rpm
 
 install:
-	find . \( -iname '*.html' -or -iname '*.js' -or -iname '*.css' \) -exec install --mode=644 -D {} $(datadir)/spice-html5/{} \;
+	find . \( -iname '*.html' -or -iname '*.js' -or -iname '*.css' \) -exec install -m 644 -D {} $(datadir)/spice-html5/{} \;
