--- a/kqemu.c	2009-07-17 00:56:23.000000000 +0000
+++ b/kqemu.c	2009-08-22 13:20:00.000000000 +0000
@@ -93,9 +93,9 @@
 int qpi_io_memory;
 uint32_t kqemu_comm_base; /* physical address of the QPI communication page */
 
-#define cpuid(index, eax, ebx, ecx, edx) \
+#define cpuid(index, eax, ecx, edx) \
   asm volatile ("cpuid" \
-                : "=a" (eax), "=b" (ebx), "=c" (ecx), "=d" (edx) \
+                : "=a" (eax), "=c" (ecx), "=d" (edx) \
                 : "0" (index))
 
 #ifdef __x86_64__
@@ -125,7 +125,7 @@
 static void kqemu_update_cpuid(CPUState *env)
 {
     int critical_features_mask, features, ext_features, ext_features_mask;
-    uint32_t eax, ebx, ecx, edx;
+    uint32_t eax, ecx, edx;
 
     /* the following features are kept identical on the host and
        target cpus because they are important for user code. Strictly
@@ -140,7 +140,7 @@
         features = 0;
         ext_features = 0;
     } else {
-        cpuid(1, eax, ebx, ecx, edx);
+        cpuid(1, eax, ecx, edx);
         features = edx;
         ext_features = ecx;
     }
