See: https://gitlab.gnome.org/GNOME/epiphany/issues/829
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -150,14 +150,15 @@
        env: envs,
   )
 
-  web_app_utils_test = executable('test-ephy-web-app-utils',
-    'ephy-web-app-utils-test.c',
-    dependencies: ephymain_dep
-  )
-  test('Web app utils test',
-       web_app_utils_test,
-       env: envs
-  )
+  # FIXME: https://gitlab.gnome.org/GNOME/epiphany/issues/829
+  # web_app_utils_test = executable('test-ephy-web-app-utils',
+  #   'ephy-web-app-utils-test.c',
+  #   dependencies: ephymain_dep
+  # )
+  # test('Web app utils test',
+  #      web_app_utils_test,
+  #      env: envs
+  # )
 
   # FIXME: https://bugzilla.gnome.org/show_bug.cgi?id=780280
   # web_view_test = executable('test-ephy-web-view',
