diff --git a/Cargo.toml b/Cargo.toml
index abebbb46..968f4941 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -164,7 +164,8 @@ tracing-futures = { version = "0.2", features = ["futures-03"] }
 tracing-opentelemetry = "0.33"
 tracing-subscriber = {
     version = "0.3",
-    features = ["ansi", "fmt", "registry", "env-filter", "json"],
+    default-features = false,
+    features = ["fmt", "registry", "env-filter", "json", "smallvec", "std", "tracing-log"],
 }
 tracing-tree = "0.4"
 tree-sitter = "0.26.8"
