https://github.com/launchbadge/sqlx/issues/2198
--- a/vendor/sqlx/sqlx-core/src/sqlite/statement/unlock_notify.rs
+++ b/vendor/sqlx/sqlx-core/src/sqlite/statement/unlock_notify.rs
@@ -48,6 +48,8 @@
     }
 
     fn wait(&self) {
+        // We only want to wait until the lock is available again.
+        #[allow(let_underscore_lock)]
         let _ = self
             .condvar
             .wait_while(self.mutex.lock().unwrap(), |fired| !*fired)
