From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 13 May 2022 01:44:00 +0200
Subject: [PATCH] Build sqlx-core with system libsqlite3

--- a/vendor/sqlx/sqlx-core/Cargo.toml
+++ b/vendor/sqlx/sqlx-core/Cargo.toml
@@ -135,6 +135,6 @@
 libsqlite3-sys = { version = "0.24.1", optional = true, default-features = false, features = [
     "pkg-config",
     "vcpkg",
-    "bundled",
-    "unlock_notify"
+    "unlock_notify",
+    "buildtime_bindgen"
 ] }
