From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 13 May 2022 01:42:00 +0200
Subject: [PATCH] Build with system OpenSSL instead of bundled rusttls

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -33,12 +33,12 @@
 
 [workspace.dependencies.reqwest]
 version = "0.11"
-features = ["json", "rustls-tls-native-roots"]
+features = ["json", "default-tls"]
 default-features = false
 
 [workspace.dependencies.sqlx]
 version = "0.6"
-features = ["runtime-tokio-rustls", "chrono", "postgres"]
+features = ["runtime-tokio-native-tls", "chrono", "postgres"]
 
 [patch.crates-io]
 sqlx = { path = "./vendor/sqlx" }
