From 4f4e4cc5a52efa02b763440d5bf6f5923258eafc Mon Sep 17 00:00:00 2001
From: Cormac Stephenson <c7s@kasku.net>
Date: Thu, 13 Jan 2022 05:04:52 +0000
Subject: [PATCH] fix CXXFLAGS

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 63bf1833..abea58aa 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -16,7 +16,7 @@ ifeq ($(debug),yes)
     suffix := .debug
 else
     ifeq ($(debug),no)
-        CXXFLAGS += -O3
+        # CXXFLAGS += -O3
         suffix := .opt
     else
         $(error debug should be either yes or no)
-- 
2.34.1

