From 4f4e4cc5a52efa02b763440d5bf6f5923258eafc Mon Sep 17 00:00:00 2001
From: Cormac Stephenson <c7s@kasku.net>
Date: Thu, 13 Jan 2022 05:04:52 +0000
Subject: [PATCH] fix CXXFLAGS

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index 63bf1833..abea58aa 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ CPPFLAGS-debug-yes = -DKAK_DEBUG
 CXXFLAGS-debug-yes = -O0 -g3
 tag-debug-yes = .debug
 
-CXXFLAGS-debug-no = -O3 -g3
+# CXXFLAGS-debug-no = -O3 -g3
 tag-debug-no = .opt
 
 CXXFLAGS-sanitize-address = -fsanitize=address
