diff --git a/configure b/configure
index 9d6457d..f57df0d 100755
--- a/configure
+++ b/configure
@@ -7126,8 +7126,8 @@ check_optflags -fno-signed-zeros
 
 if enabled lto; then
     test "$cc_type" != "$ld_type" && die "LTO requires same compiler and linker"
-    check_cflags  -flto
-    check_ldflags -flto $cpuflags
+    check_cflags  -flto=auto
+    check_ldflags -flto=auto $cpuflags
     disable inline_asm_direct_symbol_refs
 fi
 
