--- ioq3-for-UrbanTerror-4-release-4.3.4/Makefile.orig	2020-04-06 15:39:29.112727739 +0200
+++ ioq3-for-UrbanTerror-4-release-4.3.4/Makefile	2020-04-06 15:39:31.009400225 +0200
@@ -154,7 +154,7 @@
 #############################################################################
 
 BD=$(BUILD_DIR)/debug-$(PLATFORM)-$(ARCH)
-BR=$(BUILD_DIR)/release-$(PLATFORM)-$(ARCH)
+BR=$(BUILD_DIR)/release-$(PLATFORM)
 CDIR=$(MOUNT_DIR)/client
 SDIR=$(MOUNT_DIR)/server
 RDIR=$(MOUNT_DIR)/renderer
@@ -854,11 +854,11 @@
 TARGETS =
 
 ifneq ($(BUILD_SERVER),0)
-  TARGETS += $(B)/Quake3-UrT-Ded.$(ARCH)$(BINEXT)
+  TARGETS += $(B)/Quake3-UrT-Ded$(BINEXT)
 endif
 
 ifneq ($(BUILD_CLIENT),0)
-  TARGETS += $(B)/Quake3-UrT.$(ARCH)$(BINEXT)
+  TARGETS += $(B)/Quake3-UrT$(BINEXT)
   ifneq ($(BUILD_CLIENT_SMP),0)
     TARGETS += $(B)/Quake3-UrT-smp.$(ARCH)$(BINEXT)
   endif
@@ -1296,7 +1296,7 @@
     $(B)/clientsmp/sdl_glimp.o
 endif
 
-$(B)/Quake3-UrT.$(ARCH)$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
+$(B)/Quake3-UrT$(BINEXT): $(Q3OBJ) $(Q3POBJ) $(LIBSDLMAIN)
 	$(echo_cmd) "LD $@"
 	$(Q)$(CC) -o $@ $(Q3OBJ) $(Q3POBJ) $(CLIENT_LDFLAGS) \
 		$(LDFLAGS) $(LIBSDLMAIN)
@@ -1437,7 +1437,7 @@
   endif
 endif
 
-$(B)/Quake3-UrT-Ded.$(ARCH)$(BINEXT): $(Q3DOBJ)
+$(B)/Quake3-UrT-Ded$(BINEXT): $(Q3DOBJ)
 	$(echo_cmd) "LD $@"
 	$(Q)$(CC) -o $@ $(Q3DOBJ) $(LDFLAGS)
 
