diff --git a/build/cmake/cpu.cmake b/build/cmake/cpu.cmake
index ef2d755..31e1118 100644
--- a/build/cmake/cpu.cmake
+++ b/build/cmake/cpu.cmake
@@ -13,13 +13,8 @@ if("${AOM_TARGET_CPU}" MATCHES "^arm")
   set(ARCH_ARM 1)
   set(RTCD_ARCH_ARM "yes")
 
-  if(ENABLE_NEON)
-    set(HAVE_NEON 1)
-    set(RTCD_HAVE_NEON "yes")
-  else()
-    set(HAVE_NEON 0)
-    set(AOM_RTCD_FLAGS ${AOM_RTCD_FLAGS} --disable-neon)
-  endif()
+  set(HAVE_NEON 0)
+  set(AOM_RTCD_FLAGS ${AOM_RTCD_FLAGS} --disable-neon)
 elseif("${AOM_TARGET_CPU}" MATCHES "^mips")
   set(ARCH_MIPS 1)
   set(RTCD_ARCH_MIPS "yes")
