diff -ur Mail-DomainKeys-1.0/META.yml Mail-DomainKeys-1.0-new/META.yml
--- Mail-DomainKeys-1.0/META.yml	2007-03-03 07:36:07.000000000 -0400
+++ Mail-DomainKeys-1.0-new/META.yml	2026-04-18 15:17:15.300682750 -0300
@@ -5,7 +5,7 @@
 version_from: lib/Mail/DomainKeys.pm
 installdirs:  site
 requires:
-    Crypt::OpenSSL::RSA:           0
+    Crypt::PK::RSA:                0
     Mail::Address:                 0
     MIME::Base64:                  0
     Net::DNS:                      0.34
diff -ur Mail-DomainKeys-1.0/Makefile.PL Mail-DomainKeys-1.0-new/Makefile.PL
--- Mail-DomainKeys-1.0/Makefile.PL	2004-12-07 23:34:29.000000000 -0400
+++ Mail-DomainKeys-1.0-new/Makefile.PL	2026-04-18 15:17:15.300682750 -0300
@@ -40,7 +40,7 @@
 	NAME          => "Mail::DomainKeys",
 	VERSION_FROM  => "lib/Mail/DomainKeys.pm",
 	PREREQ_PM     => {
-		"Crypt::OpenSSL::RSA"	=> 0,
+		"Crypt::PK::RSA"	=> 0,
 		"Mail::Address"		=> 0,
 		"MIME::Base64"		=> 0,
 		"Net::DNS"		=> 0.34,
diff -ur Mail-DomainKeys-1.0/README Mail-DomainKeys-1.0-new/README
--- Mail-DomainKeys-1.0/README	2007-03-03 07:34:02.000000000 -0400
+++ Mail-DomainKeys-1.0-new/README	2026-04-18 15:17:15.300682750 -0300
@@ -37,7 +37,7 @@
 if possible. :)  Please include the version numbers of the following
 modules:
 
-Crypt::OpenSSL::RSA
+Crypt::PK::RSA
 Mail::Address
 MIME::Base64
 Net::DNS
diff -ur Mail-DomainKeys-1.0/lib/Mail/DomainKeys/Key/Private.pm Mail-DomainKeys-1.0-new/lib/Mail/DomainKeys/Key/Private.pm
--- Mail-DomainKeys-1.0/lib/Mail/DomainKeys/Key/Private.pm	2006-10-12 21:43:20.000000000 -0300
+++ Mail-DomainKeys-1.0-new/lib/Mail/DomainKeys/Key/Private.pm	2026-04-18 15:17:15.300682750 -0300
@@ -39,7 +39,7 @@
 }
 
 sub convert {
-	use Crypt::OpenSSL::RSA;
+	use Crypt::PK::RSA;
 
 	my $self = shift;
 
@@ -61,7 +61,8 @@
 	my $cork;
 
 	eval {
-		$cork = new_private_key Crypt::OpenSSL::RSA($pkcs);
+		my $pk = Crypt::PK::RSA->new();
+		$cork = $pk->import_key(\$pkcs);
 	};
 
 	$@ and
@@ -87,7 +88,7 @@
 
 	my $rtrn;
 	
-	eval { $rtrn = $self->cork->sign($mail); };
+	eval { $rtrn = $self->cork->sign_message($mail, 'SHA1', 'v1.5'); };
 
 	$@ and
 		$self->errorstr($@),
diff -ur Mail-DomainKeys-1.0/lib/Mail/DomainKeys/Key/Public.pm Mail-DomainKeys-1.0-new/lib/Mail/DomainKeys/Key/Public.pm
--- Mail-DomainKeys-1.0/lib/Mail/DomainKeys/Key/Public.pm	2006-10-12 21:43:26.000000000 -0300
+++ Mail-DomainKeys-1.0-new/lib/Mail/DomainKeys/Key/Public.pm	2026-04-18 15:17:15.300682750 -0300
@@ -127,7 +127,7 @@
 }
 
 sub convert {
-	use Crypt::OpenSSL::RSA;
+	use Crypt::PK::RSA;
 
 	my $self = shift;
 
@@ -148,7 +148,8 @@
 	my $cork;
 	
 	eval {
-		$cork = new_public_key Crypt::OpenSSL::RSA($cert);
+		my $pk = Crypt::PK::RSA->new();
+		$cork = $pk->import_key(\$cert);
 	};
 
 	$@ and
@@ -173,7 +174,7 @@
 
 
 	my $rtrn = eval {
-		$self->cork->verify($prms{'Text'}, $prms{'Signature'});
+		$self->cork->verify_message($prms{'Signature'}, $prms{'Text'}, 'SHA1', 'v1.5');
 	}; 
 
 	$@ and
