--- a/vendor/github.com/containers/storage/pkg/unshare/unshare.c
+++ b/vendor/github.com/containers/storage/pkg/unshare/unshare.c
@@ -19,6 +19,7 @@
 #include <sys/vfs.h>
 #include <sys/mount.h>
 #include <linux/limits.h>
+#include <libgen.h>
 
 /* Open Source projects like conda-forge, want to package podman and are based
    off of centos:6, Conda-force has minimal libc requirements and is lacking
