https://github.com/uutils/coreutils/issues/11687

--- a/src/uu/stty/src/stty.rs
+++ b/src/uu/stty/src/stty.rs
@@ -630,36 +630,42 @@ fn print_terminal_size(
     // GNU linked against glibc 2.42 provides us baudrate 51 which panics cfgetospeed
     #[cfg(not(target_os = "linux"))]
     let speed = nix::sys::termios::cfgetospeed(termios);
-    #[cfg(target_os = "linux")]
-    #[cfg(all(not(target_arch = "powerpc"), not(target_arch = "powerpc64")))]
+    #[cfg(all(
+        target_os = "linux",
+        not(target_arch = "powerpc"),
+        not(target_arch = "powerpc64")
+    ))]
     ioctl_read_bad!(tcgets2, TCGETS2, termios2);
-    #[cfg(target_os = "linux")]
-    #[cfg(all(not(target_arch = "powerpc"), not(target_arch = "powerpc64")))]
+    #[cfg(all(
+        target_os = "linux",
+        not(target_arch = "powerpc"),
+        not(target_arch = "powerpc64")
+    ))]
     let speed = {
         let mut t2 = unsafe { std::mem::zeroed::<termios2>() };
         unsafe { tcgets2(opts.file.as_raw_fd(), &raw mut t2)? };
         t2.c_ospeed
     };
-    #[cfg(target_os = "linux")]
-    #[cfg(any(target_arch = "powerpc", target_arch = "powerpc64"))]
+    #[cfg(all(
+        target_os = "linux",
+        any(target_arch = "powerpc", target_arch = "powerpc64")
+    ))]
     let speed = nix::sys::termios::cfgetospeed(termios);
 
     let mut printer = WrappedPrinter::new(window_size);
 
-    // BSDs and Linux (not ppc/big-endian ppc64) use a u32 for the baud rate, so we can simply
-    // print it.
-    #[cfg(any(target_os = "linux", bsd))]
+    // BSDs and Linux (non-PowerPC) use a u32 for the baud rate, so we can simply print it.
     #[cfg(all(
+        any(target_os = "linux", bsd),
         not(target_arch = "powerpc"),
-        not(all(target_arch = "powerpc64", target_endian = "big"))
+        not(target_arch = "powerpc64")
     ))]
     printer.print(&translate!("stty-output-speed", "speed" => speed));
 
-    // Big-endian Linux PowerPC uses BaudRate enum, need to convert to display format
-    #[cfg(target_os = "linux")]
-    #[cfg(any(
-        target_arch = "powerpc",
-        all(target_arch = "powerpc64", target_endian = "big")
+    // PowerPC uses BaudRate enum, need to convert to display format
+    #[cfg(all(
+        target_os = "linux",
+        any(target_arch = "powerpc", target_arch = "powerpc64")
     ))]
     {
         // On PowerPC, find the corresponding baud rate string for display
-- 
2.53.0

