diff --git a/src/gvmd.c b/src/gvmd.c
index 419a585..3158bce 100644
--- a/src/gvmd.c
+++ b/src/gvmd.c
@@ -91,6 +91,7 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <glib/gmessages.h>
 
 #include <gvm/base/pidfile.h>
 #include <gvm/base/pwpolicy.h>
diff --git a/src/sql_pg.c b/src/sql_pg.c
index 06b2e1d..8803e83 100644
--- a/src/sql_pg.c
+++ b/src/sql_pg.c
@@ -33,7 +33,7 @@
 #include <glib.h>
 #include <inttypes.h>
 #include <netinet/in.h>
-#include <postgresql/libpq-fe.h>
+#include <libpq-fe.h>
 #include <stdlib.h>
 #include <string.h>
 
