diff --git a/tools/gvm-portnames-update.in b/tools/gvm-portnames-update.in
index ab49349..b2889ee 100644
--- a/tools/gvm-portnames-update.in
+++ b/tools/gvm-portnames-update.in
@@ -49,12 +49,12 @@ fi
 
 # Configure DB_DIR where our DB is located.
 if [ -z "$DB_DIR" ]; then
-  OPENVASSD=`which openvassd`
+  OPENVASSD=`which openvas`
   if [ -z "$OPENVASSD" ] ; then
     echo "[e] Error: openvassd is not in the path, could not determine the Manager directory."
     exit 1
   else
-    OV_DIR=`openvassd -s | awk -F" = " '/^plugins_folder/ { print $2 }' | sed -s 's/\(^.*\)\/plugins/\1/'`
+    OV_DIR=`openvas -s | awk -F" = " '/^plugins_folder/ { print $2 }' | sed -s 's/\(^.*\)\/plugins/\1/'`
   fi
   DB_DIR="$OV_DIR/gvmd"
 fi
