diff --git a/tools/greenbone-certdata-sync.in b/tools/greenbone-certdata-sync.in
index 97bccf9..a89a3ce 100644
--- a/tools/greenbone-certdata-sync.in
+++ b/tools/greenbone-certdata-sync.in
@@ -494,13 +494,11 @@ then
 fi
 (
   flock -n 9
-  date > $LOCK_FILE
   if [ $? -eq 1 ] ; then
     log_notice "Sync in progress, exiting."
     exit 1
   fi
   sync_certdata
-  echo -n > $LOCK_FILE
-) 9>$LOCK_FILE
+)
 
 exit 0
diff --git a/tools/greenbone-scapdata-sync.in b/tools/greenbone-scapdata-sync.in
index b54e251..7aec26d 100644
--- a/tools/greenbone-scapdata-sync.in
+++ b/tools/greenbone-scapdata-sync.in
@@ -517,13 +517,11 @@ then
 fi
 (
   flock -n 9
-  date > $LOCK_FILE
   if [ $? -eq 1 ] ; then
     log_notice "Sync in progress, exiting."
     exit 1
   fi
   sync_scapdata
-  echo -n > $LOCK_FILE
-) 9>$LOCK_FILE
+)
 
 exit 0
