diff --git a/src/gvmd.c b/src/gvmd.c
index 667ee96..37dbf32 100644
--- a/src/gvmd.c
+++ b/src/gvmd.c
@@ -1241,7 +1241,7 @@ serve_and_schedule ()
       if (termination_signal)
         {
           g_debug ("Received %s signal",
-                   sys_siglist[termination_signal]);
+                   strsignal(termination_signal));
           cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           setup_signal_handler (termination_signal, SIG_DFL, 0);
@@ -1330,7 +1330,7 @@ serve_and_schedule ()
       if (termination_signal)
         {
           g_debug ("Received %s signal",
-                   sys_siglist[termination_signal]);
+                   strsignal(termination_signal));
           cleanup ();
           /* Raise signal again, to exit with the correct return value. */
           setup_signal_handler (termination_signal, SIG_DFL, 0);
diff --git a/src/manage_sql.c b/src/manage_sql.c
index b41e5c6..288e0e3 100644
--- a/src/manage_sql.c
+++ b/src/manage_sql.c
@@ -16570,7 +16570,7 @@ cleanup_manage_process (gboolean cleanup)
 void
 manage_cleanup_process_error (int signal)
 {
-  g_debug ("Received %s signal", sys_siglist[signal]);
+  g_debug ("Received %s signal", strsignal(signal));
   if (sql_is_open ())
     {
       if (current_scanner_task)
diff --git a/src/utils.c b/src/utils.c
index 2a785b5..b9ebec6 100644
--- a/src/utils.c
+++ b/src/utils.c
@@ -796,7 +796,7 @@ setup_signal_handler (int signal, void (*handler) (int), int block)
   if (sigaction (signal, &action, NULL) == -1)
     {
       g_critical ("%s: failed to register %s handler",
-                  __func__, sys_siglist[signal]);
+                  __func__, strsignal(signal));
       exit (EXIT_FAILURE);
     }
 }
@@ -827,7 +827,7 @@ setup_signal_handler_info (int signal,
   if (sigaction (signal, &action, NULL) == -1)
     {
       g_critical ("%s: failed to register %s handler",
-                  __func__, sys_siglist[signal]);
+                  __func__, strsignal(signal));
       exit (EXIT_FAILURE);
     }
 }
