diff --git a/src/test_interface.py b/src/test_interface.py
index aa18622..d33c2a5 100644
--- a/src/test_interface.py
+++ b/src/test_interface.py
@@ -8,7 +8,7 @@ def import_error(msg):
   print()
 
 try:
-  from nose.tools import assert_raises, assert_almost_equals
+  from nose.tools import assert_raises, assert_almost_equal
 except ImportError:
   import_error("Please install nose to run tests.")
   raise
@@ -40,7 +40,7 @@ b = np.array([3.])
 dims = {'q': [], 'l': 2}
 
 def check_solution(solution, expected):
-  assert_almost_equals(solution, expected, places=5)
+  assert_almost_equal(solution, expected, places=5)
 
 def test_problems():
   myopts = {'feastol': 2e-8, 'reltol': 2e-8, 'abstol': 2e-8, 'verbose':True};
