diff --git a/tests/test_launcher.py.orig b/tests/test_launcher.py
index 182a153f10c..936ada75108 100644
--- a/tests/test_launcher.py.orig
+++ b/tests/test_launcher.py
@@ -113,6 +113,7 @@ class TestLauncher(unittest.TestCase):
         await browser.close()
         server.stop()
 
+    @unittest.skip('should fix hanging test unit.')
     @sync
     async def test_ignore_https_errors_interception(self):
         browser = await launch(DEFAULT_OPTIONS, ignoreHTTPSErrors=True)
