diff --git a/pyppeteer/chromium_downloader.py.orig b/pyppeteer/chromium_downloader.py
index 87c820dd5d4..4c61b28d5ff 100644
--- a/pyppeteer/chromium_downloader.py.orig
+++ b/pyppeteer/chromium_downloader.py
@@ -45,7 +45,7 @@ downloadURLs = {
 }
 
 chromiumExecutable = {
-    'linux': DOWNLOADS_FOLDER / REVISION / 'chrome-linux' / 'chrome',
+    'linux': Path('/usr/bin/chromium-headless-shell'),
     'mac': (DOWNLOADS_FOLDER / REVISION / 'chrome-mac' / 'Chromium.app' / 'Contents' / 'MacOS' / 'Chromium'),
     'win32': DOWNLOADS_FOLDER / REVISION / windowsArchive / 'chrome.exe',
     'win64': DOWNLOADS_FOLDER / REVISION / windowsArchive / 'chrome.exe',
@@ -135,7 +135,9 @@ def extract_zip(data: BytesIO, path: Path) -> None:
 
 def download_chromium() -> None:
     """Download and extract chromium."""
-    extract_zip(download_zip(get_url()), DOWNLOADS_FOLDER / REVISION)
+    exec_path = chromium_executable()
+    if not exec_path.exists():
+        logger.warning("Please install chromium via your package manager, eg 'sudo apk add chromium-headless-shell'")
 
 
 def chromium_executable() -> Path:
