Improve default config.
--- a/scripts/dkimproxy_out.conf.example
+++ b/scripts/dkimproxy_out.conf.example
@@ -12,10 +12,10 @@
 signature domainkeys(c=nofws)
 
 # specify location of the private key
-keyfile   /full/path/to/private.key
+keyfile   /etc/dkimproxy/private.key
 
 # specify the selector (i.e. the name of the key record put in DNS)
-selector  selector1
+selector  mail
 
 # control how many processes DKIMproxy uses
 #  - more information on these options (and others) can be found by
