--- a/src/plugins/clangformat/CMakeLists.txt
+++ b/src/plugins/clangformat/CMakeLists.txt
@@ -1,3 +1,4 @@
+find_package(LLVM CONFIG REQUIRED)
 add_qtc_plugin(ClangFormat
   CONDITION TARGET ${CLANG_FORMAT_LIB} AND LLVM_PACKAGE_VERSION VERSION_GREATER_EQUAL 10.0.0 AND (QTC_CLANG_BUILDMODE_MATCH OR CLANGTOOLING_LINK_CLANG_DYLIB)
   DEPENDS Utils Qt5::Widgets ${CLANG_FORMAT_LIB}
@@ -16,7 +17,7 @@
 
 if(TARGET ClangFormat)
   # "system" includes, so warnings are ignored
-  target_include_directories(ClangFormat SYSTEM PRIVATE "${CLANG_INCLUDE_DIRS}")
+  target_include_directories(ClangFormat SYSTEM PRIVATE "${CLANG_INCLUDE_DIRS}" PRIVATE "${LLVM_INCLUDE_DIRS}")
 endif()
 
 if(MSVC AND TARGET ClangFormat)
