From 4d2ada29872460c1d4718763967a0bf059094590 Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Sun, 13 Jul 2025 02:00:27 +0200
Subject: [PATCH 2/2] no-malloc_trim

Signed-off-by: fossdd <fossdd@pwned.life>
---
 src/plugins/coreplugin/icore.cpp | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/src/plugins/coreplugin/icore.cpp b/src/plugins/coreplugin/icore.cpp
index e0c75926f32..bfb147c98c3 100644
--- a/src/plugins/coreplugin/icore.cpp
+++ b/src/plugins/coreplugin/icore.cpp
@@ -1485,9 +1485,6 @@ void ICorePrivate::init()
         using namespace std::chrono_literals;
         m_trimTimer.setInterval(60s);
         // glibc may not actually free memory in free().
-#ifdef Q_OS_LINUX
-        connect(&m_trimTimer, &QTimer::timeout, this, [] { malloc_trim(0); });
-#endif
     }
 }
 
-- 
2.50.1

