fix for compiling with musl

diff -ru asterisk-12.4.0.orig/include/asterisk/lock.h asterisk-12.4.0/include/asterisk/lock.h
--- asterisk-12.4.0.orig/include/asterisk/lock.h	2013-11-02 06:05:24.000000000 -0200
+++ asterisk-12.4.0/include/asterisk/lock.h	2014-08-04 16:00:29.811967599 -0300
@@ -66,7 +66,7 @@
 #define AST_PTHREADT_NULL (pthread_t) -1
 #define AST_PTHREADT_STOP (pthread_t) -2
 
-#if (defined(SOLARIS) || defined(BSD))
+#if (defined(SOLARIS) || defined(BSD) || !defined(HAVE_PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP))
 #define AST_MUTEX_INIT_W_CONSTRUCTORS
 #endif /* SOLARIS || BSD */
 
