diff -uNr asterisk-1.6.2.0/Makefile asterisk-1.6.2.0-r1/Makefile
--- asterisk-1.6.2.0/Makefile	2010-01-16 18:55:43.738059027 +0100
+++ asterisk-1.6.2.0-r1/Makefile	2010-01-16 19:32:06.898555681 +0100
@@ -589,7 +589,7 @@
 	fi
 
 $(SUBDIRS_INSTALL):
-	@DESTDIR="$(DESTDIR)" ASTSBINDIR="$(ASTSBINDIR)" $(SUBMAKE) -C $(@:-install=) install
+	@+DESTDIR="$(DESTDIR)" ASTSBINDIR="$(ASTSBINDIR)" $(SUBMAKE) -C $(@:-install=) install
 
 NEWMODS:=$(foreach d,$(MOD_SUBDIRS),$(notdir $(wildcard $(d)/*.so)))
 OLDMODS=$(filter-out $(NEWMODS),$(notdir $(wildcard $(DESTDIR)$(MODULES_DIR)/*.so)))
@@ -848,7 +848,7 @@
 # last clean count we had
 
 cleantest:
-	@cmp -s .cleancount .lastclean || $(MAKE) clean
+	@+cmp -s .cleancount .lastclean || $(MAKE) clean
 
 $(SUBDIRS_UNINSTALL):
 	@$(SUBMAKE) -C $(@:-uninstall=) uninstall
@@ -897,7 +897,7 @@
 nmenuconfig: nmenuselect
 
 menuselect: menuselect/cmenuselect menuselect/nmenuselect menuselect/gmenuselect
-	@if [ -x menuselect/nmenuselect ]; then \
+	@+if [ -x menuselect/nmenuselect ]; then \
 		$(MAKE) nmenuselect; \
 	elif [ -x menuselect/cmenuselect ]; then \
 		$(MAKE) cmenuselect; \
diff -uNr asterisk-1.6.2.0/channels/Makefile asterisk-1.6.2.0-r1/channels/Makefile
--- asterisk-1.6.2.0/channels/Makefile	2010-01-16 18:55:43.733055737 +0100
+++ asterisk-1.6.2.0-r1/channels/Makefile	2010-01-16 18:57:51.589055800 +0100
@@ -100,7 +100,7 @@
 chan_usbradio.so: _ASTCFLAGS+=-DNDEBUG
 
 h323/Makefile.ast:
-	$(CMD_PREFIX) $(MAKE) -C h323 Makefile.ast
+	+$(CMD_PREFIX) $(MAKE) -C h323 Makefile.ast
 
 h323/libchanh323.a: h323/Makefile.ast
-	$(CMD_PREFIX) $(MAKE) -C h323 libchanh323.a
+	+$(CMD_PREFIX) $(MAKE) -C h323 libchanh323.a
diff -uNr asterisk-1.6.2.0/codecs/gsm/Makefile asterisk-1.6.2.0-r1/codecs/gsm/Makefile
--- asterisk-1.6.2.0/codecs/gsm/Makefile	2010-01-16 18:55:43.733055737 +0100
+++ asterisk-1.6.2.0-r1/codecs/gsm/Makefile	2010-01-16 18:58:31.557055848 +0100
@@ -358,12 +358,12 @@
 # Installation
 
 gsminstall:
-		-if [ x"$(GSM_INSTALL_ROOT)" != x ] ; then	\
+		-+if [ x"$(GSM_INSTALL_ROOT)" != x ] ; then	\
 			$(MAKE) $(GSM_INSTALL_TARGETS) ;	\
 		fi
 
 toastinstall:
-		-if [ x"$(TOAST_INSTALL_ROOT)" != x ]; then	\
+		-+if [ x"$(TOAST_INSTALL_ROOT)" != x ]; then	\
 			$(MAKE) $(TOAST_INSTALL_TARGETS);	\
 		fi
 
diff -uNr asterisk-1.6.2.0/main/Makefile asterisk-1.6.2.0-r1/main/Makefile
--- asterisk-1.6.2.0/main/Makefile	2010-01-16 18:55:43.738059027 +0100
+++ asterisk-1.6.2.0-r1/main/Makefile	2010-01-16 19:26:43.334555789 +0100
@@ -177,7 +177,7 @@
 clean::
 	rm -f asterisk
 	rm -f db1-ast/.*.d
-	@if [ -f editline/Makefile ]; then $(MAKE) -C editline distclean ; fi
+	@+if [ -f editline/Makefile ]; then $(MAKE) -C editline distclean ; fi
 	@$(MAKE) -C db1-ast clean
 	@$(MAKE) -C stdtime clean
 	rm -f libresample/src/*.o
diff -uNr asterisk-1.6.2.0/menuselect/Makefile asterisk-1.6.2.0-r1/menuselect/Makefile
--- asterisk-1.6.2.0/menuselect/Makefile	2010-01-16 18:55:43.738059027 +0100
+++ asterisk-1.6.2.0-r1/menuselect/Makefile	2010-01-16 19:03:34.381055693 +0100
@@ -96,7 +96,7 @@
 	$(CC) -o $@ $^ $(M_LIBS)
 
 mxml/libmxml.a:
-	@if test ! -f mxml/Makefile ; then cd mxml && ./configure ; fi
+	@+if test ! -f mxml/Makefile ; then cd mxml && ./configure ; fi
 	@$(MAKE) -C mxml libmxml.a
 
 test: menuselect
@@ -113,11 +113,11 @@
 
 clean:
 	rm -f menuselect cmenuselect gmenuselect nmenuselect $(OBJS) $(M_OBJS) $(C_OBJS) $(G_OBJS) $(N_OBJS)
-	@if test -f mxml/Makefile ; then $(MAKE) -C mxml clean ; fi
+	@+if test -f mxml/Makefile ; then $(MAKE) -C mxml clean ; fi
 
 dist-clean: distclean
 
 distclean: clean
-	@if test -f mxml/Makefile ; then $(MAKE) -C mxml distclean ; fi
+	@+if test -f mxml/Makefile ; then $(MAKE) -C mxml distclean ; fi
 	rm -f autoconfig.h config.status config.log makeopts
 	rm -rf autom4te.cache
