Use mariadb instead of mysql

diff --git a/addons/cdr_mysql.c b/addons/cdr_mysql.c
index 00c75dd..bc405bb 100644
--- a/addons/cdr_mysql.c
+++ b/addons/cdr_mysql.c
@@ -43,6 +43,7 @@
 #include "asterisk.h"
 
 #include <mysql/mysql.h>
+#include <mysql/mariadb_version.h>
 #include <mysql/errmsg.h>
 
 #include "asterisk/config.h"
@@ -648,7 +649,7 @@ static int my_load_module(int reload)
 	res |= my_load_config_string(cfg, "global", "ssl_cert", &ssl_cert, "");
 	res |= my_load_config_string(cfg, "global", "ssl_key", &ssl_key, "");
 
-	res |= my_load_config_number(cfg, "global", "port", &dbport, MYSQL_PORT);
+	res |= my_load_config_number(cfg, "global", "port", &dbport, MARIADB_PORT);
 	res |= my_load_config_number(cfg, "global", "timeout", &timeout, 0);
 	res |= my_load_config_string(cfg, "global", "compat", &compat, "no");
 	res |= my_load_config_string(cfg, "global", "cdrzone", &cdrzone, "");
diff --git a/addons/res_config_mysql.c b/addons/res_config_mysql.c
index ae43485..94d3b35 100644
--- a/addons/res_config_mysql.c
+++ b/addons/res_config_mysql.c
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 
 #include <mysql/mysql.h>
+#include <mysql/mariadb_version.h>
 #include <mysql/errmsg.h>
 
 #include "asterisk/channel.h"
