diff -ru asterisk-13.0.0.orig/include/asterisk/compat.h asterisk-13.0.0/include/asterisk/compat.h
--- asterisk-13.0.0.orig/include/asterisk/compat.h	2014-10-31 08:48:56.000000000 -0200
+++ asterisk-13.0.0/include/asterisk/compat.h	2014-10-31 08:49:10.163808393 -0200
@@ -68,7 +68,7 @@
 #endif
 
 #ifndef AST_POLL_COMPAT
-#include <sys/poll.h>
+#include <poll.h>
 #else
 #include "asterisk/poll-compat.h"
 #endif
diff -ru asterisk-13.0.0.orig/include/asterisk/poll-compat.h asterisk-13.0.0/include/asterisk/poll-compat.h
--- asterisk-13.0.0.orig/include/asterisk/poll-compat.h	2014-10-31 08:48:56.000000000 -0200
+++ asterisk-13.0.0/include/asterisk/poll-compat.h	2014-10-31 08:49:10.163808393 -0200
@@ -83,7 +83,7 @@
 
 #ifndef AST_POLL_COMPAT
 
-#include <sys/poll.h>
+#include <poll.h>
 
 #define ast_poll(a, b, c) poll(a, b, c)
 
diff -ru asterisk-13.0.0.orig/main/ast_expr2.c asterisk-13.0.0/main/ast_expr2.c
--- asterisk-13.0.0.orig/main/ast_expr2.c	2014-10-31 08:48:56.000000000 -0200
+++ asterisk-13.0.0/main/ast_expr2.c	2014-10-31 08:49:10.163808393 -0200
@@ -93,6 +93,7 @@
 
 #include "asterisk.h"
 
+#include <sys/cdefs.h>
 #include <sys/types.h>
 #include <stdio.h>
 
diff -ru asterisk-13.0.0.orig/main/ast_expr2.y asterisk-13.0.0/main/ast_expr2.y
--- asterisk-13.0.0.orig/main/ast_expr2.y	2014-10-31 08:48:56.000000000 -0200
+++ asterisk-13.0.0/main/ast_expr2.y	2014-10-31 08:49:10.163808393 -0200
@@ -14,6 +14,7 @@
 
 #include "asterisk.h"
 
+#include <sys/cdefs.h>
 #include <sys/types.h>
 #include <stdio.h>
 
