--- asterisk-11.7.0.orig/res/pjproject/pjlib/src/pj/os_core_unix.c
+++ asterisk-11.7.0/res/pjproject/pjlib/src/pj/os_core_unix.c
@@ -1091,8 +1091,7 @@
 	return PJ_RETURN_OS_ERROR(rc);
 
     if (type == PJ_MUTEX_SIMPLE) {
-#if (defined(PJ_LINUX) && PJ_LINUX!=0) || \
-    defined(PJ_HAS_PTHREAD_MUTEXATTR_SETTYPE)
+#if defined(PJ_LINUX) && PJ_LINUX!=0 && defined(__GLIBC__)
 	rc = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_FAST_NP);
 #elif (defined(PJ_RTEMS) && PJ_RTEMS!=0) || \
        defined(PJ_PTHREAD_MUTEXATTR_T_HAS_RECURSIVE)
@@ -1101,8 +1100,7 @@
 	rc = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_NORMAL);
 #endif
     } else {
-#if (defined(PJ_LINUX) && PJ_LINUX!=0) || \
-     defined(PJ_HAS_PTHREAD_MUTEXATTR_SETTYPE)
+#if (defined(PJ_LINUX) && PJ_LINUX!=0) && defined(__GLIBC__)
 	rc = pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE_NP);
 #elif (defined(PJ_RTEMS) && PJ_RTEMS!=0) || \
        defined(PJ_PTHREAD_MUTEXATTR_T_HAS_RECURSIVE)
