From 01e2188ba41cb1d7cbdc82a74f6691a3fb669863 Mon Sep 17 00:00:00 2001
From: Rodney Dawes <dobey.pwns@gmail.com>
Date: Fri, 8 Oct 2021 14:14:05 -0400
Subject: [PATCH] Use the correct kcm name for settings in Plasma Mobile

---
 src/plugin/inputmethod.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/plugin/inputmethod.cpp b/src/plugin/inputmethod.cpp
index c61d47d8..6397a098 100644
--- a/src/plugin/inputmethod.cpp
+++ b/src/plugin/inputmethod.cpp
@@ -763,7 +763,7 @@ void InputMethod::showSystemSettings()
     qunsetenv("QT_WAYLAND_SHELL_INTEGRATION");
 
     if (qgetenv("XDG_CURRENT_DESKTOP") == "KDE") {
-        QDesktopServices::openUrl(QUrl("systemsettings://kcm_virtualkeyboard"));
+        QDesktopServices::openUrl(QUrl("systemsettings://kcm_mobile_virtualkeyboard"));
     } else {
         QDesktopServices::openUrl(QUrl("settings://system/language"));
     }
