--- Regina-REXX-3.7.orig/gci/gci_convert.linux.i86
+++ Regina-REXX-3.7/gci/gci_convert.linux.i86
@@ -43,8 +43,8 @@
 #define GCI_I_4m      INT_MIN
 #define GCI_I_4M      INT_MAX
 #define GCI_I_8       signed long long
-#define GCI_I_8m      LONG_LONG_MIN
-#define GCI_I_8M      LONG_LONG_MAX
+#define GCI_I_8m      LLONG_MIN
+#define GCI_I_8M      LLONG_MAX
 
 #define GCI_Ur        strtoull
 #define GCI_Uw(s,v)   sprintf( s, "%Lu", v )
@@ -55,7 +55,7 @@
 #define GCI_U_4       unsigned
 #define GCI_U_4M      UINT_MAX
 #define GCI_U_8       unsigned long long
-#define GCI_U_8M      ULONG_LONG_MAX
+#define GCI_U_8M      ULLONG_MAX
 
 #define GCI_Fr        strtold
 #define GCI_Fw(s,v)   sprintf( s, "%.*LE", LDBL_MANT_DIG/3, v )
