From b6d871bf299c7d0f106c07ee4d8bd3b2337f53cc Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Thu, 9 Jul 2015 19:25:36 +1000
Subject: radeon: adopt for new X server dirty tracking APIs.

Signed-off-by: Dave Airlie <airlied@redhat.com>

diff --git a/src/radeon_kms.c b/src/radeon_kms.c
index ec8f51c..2dad0e6 100644
--- a/src/radeon_kms.c
+++ b/src/radeon_kms.c
@@ -287,7 +287,11 @@ redisplay_dirty(ScreenPtr screen, PixmapDirtyUpdatePtr dirty)
 
 	PixmapRegionInit(&pixregion, dirty->slave_dst);
 	DamageRegionAppend(&dirty->slave_dst->drawable, &pixregion);
+#ifdef HAS_DIRTYTRACKING_ROTATION
+	PixmapSyncDirtyHelper(dirty);
+#else
 	PixmapSyncDirtyHelper(dirty, &pixregion);
+#endif
 
 	radeon_cs_flush_indirect(pScrn);
 	DamageRegionProcessPending(&dirty->slave_dst->drawable);
-- 
cgit v0.10.2

--- ./src/drmmode_display.c.orig
+++ ./src/drmmode_display.c
@@ -794,7 +794,9 @@
 		screen->height = screenpix->drawable.height = max_height;
 	}
 	drmmode_crtc->scanout_pixmap_x = this_x;
-#ifdef HAS_DIRTYTRACKING2
+#ifdef HAS_DIRTYTRACKING_ROTATION
+	PixmapStartDirtyTracking(ppix, screenpix, 0, 0, this_x, 0, RR_Rotate_0);
+#elif defined(HAS_DIRTYTRACKING2)
 	PixmapStartDirtyTracking2(ppix, screenpix, 0, 0, this_x, 0);
 #else
 	PixmapStartDirtyTracking(ppix, screenpix, 0, 0);
