diff --git a/src/debug.h b/src/debug.h
index 7fdfda1..316b712 100644
--- a/src/debug.h
+++ b/src/debug.h
@@ -29,7 +29,7 @@
 # define ASSERT(left, operator, right) (void(0))
 #else
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__) && defined(__GLIBC__)
 #include <execinfo.h>
 #endif
 
@@ -41,7 +41,7 @@ void _on_assert_fail(const char* vara, const char* op, const char* varb,
       vara << "[" << a << "] " << op << " " << varb << "[" << b << "]";
   std::cerr << ss.str() << std::endl;
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__) && defined(__GNU_LIBRARY__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(__EMSCRIPTEN__) && defined(__GNU_LIBRARY__) && defined(__GLIBC__)
   void *callstack[64];
   size_t size;
   size = backtrace(callstack, 64);
