--- a/lib/message/tests/meson.build
+++ b/lib/message/tests/meson.build
@@ -22,7 +22,7 @@ test('test-contact',
      executable('test-contact',
                 '../mu-contact.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
@@ -30,7 +30,7 @@ test('test-document',
      executable('test-document',
                 '../mu-document.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
@@ -38,7 +38,7 @@ test('test-fields',
      executable('test-fields',
                 '../mu-fields.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
@@ -53,7 +53,7 @@ test('test-flags',
      executable('test-flags',
                 '../mu-flags.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
@@ -61,14 +61,14 @@ test('test-message',
      executable('test-message',
                 '../test-mu-message.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
 test('test-priority',
      executable('test-priority',
                 '../mu-priority.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gmime_dep, lib_mu_message_dep]))
 
@@ -76,7 +76,7 @@ test('test-message-file',
      executable('test-message-file',
                 '../mu-message-file.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_message_dep]))
 
@@ -84,6 +84,6 @@ test('test-message-part',
      executable('test-message-part',
                 '../mu-message-part.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_message_dep]))
--- a/lib/tests/meson.build
+++ b/lib/tests/meson.build
@@ -26,7 +26,7 @@ test('test-threads',
      executable('test-threads',
                 '../mu-query-threads.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -34,7 +34,7 @@ test('test-contacts-cache',
      executable('test-contacts-cache',
                 '../mu-contacts-cache.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -42,7 +42,7 @@ test('test-config',
      executable('test-config',
                 '../mu-config.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -50,7 +50,7 @@ test('test-query-macros',
      executable('test-query-macros',
                 '../mu-query-macros.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [lib_mu_dep]))
 
@@ -58,7 +58,7 @@ test('test-query-processor',
      executable('test-query-processor',
                 '../mu-query-processor.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [lib_mu_dep]))
 
@@ -66,7 +66,7 @@ test('test-query-parser',
      executable('test-query-parser',
                 '../mu-query-parser.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [lib_mu_dep]))
 
@@ -74,7 +74,7 @@ test('test-query-xapianizer',
      executable('test-query-xapianizer',
                 '../mu-query-xapianizer.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [lib_mu_dep]))
 
@@ -83,7 +83,7 @@ test('test-indexer',
      executable('test-indexer',
                 '../mu-indexer.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, config_h_dep,
                                lib_mu_dep]))
@@ -92,7 +92,7 @@ test('test-scanner',
      executable('test-scanner',
                 '../mu-scanner.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, config_h_dep,
                                lib_mu_utils_dep]))
@@ -101,7 +101,7 @@ test('test-xapian-db',
      executable('test-xapian-db',
                 '../mu-xapian-db.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [lib_mu_dep, config_h_dep]))
 
@@ -109,32 +109,32 @@ test('test-maildir',
      executable('test-maildir',
                 'test-mu-maildir.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, lib_mu_dep]))
 test('test-msg',
      executable('test-msg',
                 'test-mu-msg.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, lib_mu_dep]))
 test('test-store',
      executable('test-store',
                 'test-mu-store.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, lib_mu_dep]))
 test('test-query',
      executable('test-query',
                 'test-query.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, gmime_dep, lib_mu_dep]))
 
 test('test-store-query',
      executable('test-store-query',
                 'test-mu-store-query.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, gmime_dep, lib_mu_dep]))
 #
 # benchmarks
--- a/lib/utils/tests/meson.build
+++ b/lib/utils/tests/meson.build
@@ -25,63 +25,63 @@
 test('test-sexp',
      executable('test-sexp', '../mu-sexp.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep]))
 
 test('test-regex',
      executable('test-regex', '../mu-regex.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep]))
 
 test('test-command-handler',
      executable('test-command-handler', '../mu-command-handler.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep]))
 
 test('test-utils-file',
      executable('test-utils-file', '../mu-utils-file.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, gio_unix_dep,config_h_dep, lib_mu_utils_dep]))
 
 test('test-logger',
      executable('test-logger', '../mu-logger.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep,config_h_dep,thread_dep ]))
 
 test('test-option',
      executable('test-option', '../mu-option.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep ]))
 
 test('test-lang-detector',
      executable('test-lang-detector', '../mu-lang-detector.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [ config_h_dep, glib_dep, lib_mu_utils_dep ]))
 
 test('test-html-to-text',
      executable('test-html-to-text', '../mu-html-to-text.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep]))
 
 test('test-error',
      executable('test-error', '../mu-error.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_utils_dep]))
 
@@ -89,5 +89,5 @@ test('test-mu-utils',
      executable('test-mu-utils',
                 'test-utils.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, lib_mu_utils_dep]))
--- a/mu/tests/meson.build
+++ b/mu/tests/meson.build
@@ -23,7 +23,7 @@ test('test-cmd-add',
      executable('test-cmd-add',
                 '../mu-cmd-add.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -31,7 +31,7 @@ test('test-cmd-cfind',
      executable('test-cmd-cfind',
                 '../mu-cmd-cfind.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -39,7 +39,7 @@ test('test-cmd-extract',
      executable('test-cmd-extract',
                 '../mu-cmd-extract.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -47,7 +47,7 @@ test('test-cmd-find',
      executable('test-cmd-find',
                 '../mu-cmd-find.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -55,7 +55,7 @@ test('test-cmd-index',
      executable('test-cmd-index',
                 '../mu-cmd-index.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -63,14 +63,14 @@ test('test-cmd-init',
      executable('test-cmd-init',
                 '../mu-cmd-init.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 test('test-cmd-labels',
      executable('test-cmd-labels',
                 '../mu-cmd-labels.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -78,7 +78,7 @@ test('test-cmd-mkdir',
      executable('test-cmd-mkdir',
                 '../mu-cmd-mkdir.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -86,7 +86,7 @@ test('test-cmd-move',
      executable('test-cmd-move',
                 '../mu-cmd-move.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -94,7 +94,7 @@ test('test-cmd-remove',
      executable('test-cmd-remove',
                 '../mu-cmd-remove.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -102,7 +102,7 @@ test('test-cmd-verify',
      executable('test-cmd-verify',
                 '../mu-cmd-verify.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -110,7 +110,7 @@ test('test-cmd-view',
      executable('test-cmd-view',
                 '../mu-cmd-view.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 cpp_args: ['-DBUILD_TESTS'],
                 dependencies: [glib_dep, lib_mu_dep]))
 
@@ -118,7 +118,7 @@ test('test-cmd-query',
      executable('test-cmd-query',
                 'test-mu-query.cc',
                 install: false,
-                build_by_default: false,
+                build_by_default: get_option('tests').enabled(),
                 dependencies: [glib_dep, config_h_dep, lib_mu_dep]))
 
 gmime_test = executable(
