diff -ru gross-1.0.1.orig/doc/examples/grossd.conf gross-1.0.1/doc/examples/grossd.conf
--- gross-1.0.1.orig/doc/examples/grossd.conf	2008-06-03 09:27:37 +0000
+++ gross-1.0.1/doc/examples/grossd.conf	2008-06-03 09:31:33 +0000
@@ -71,11 +71,13 @@
 # 'statefile' is the full path of the file that the server will use to
 # store the state information. 
 # statefile = /var/db/grossd.state
+statefile = /var/db/gross/state
 
 # 'pidfile' is the full path of the file grossd writes its pid into.
 # You can set parameter 'check', if you want to keep grossd
 # from starting if pidfile already exists.
 # pidfile = /var/run/grossd.pid;check
+pidfile = /var/run/gross/grossd.pid
 
 # 'log_method' is used to list all the possible logging facilities.
 # currently only syslog is implemented
