From 9f468162a71b98916953a46e96b5a0729558cbaa Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Mon, 23 Mar 2026 00:03:58 +0100
Subject: [PATCH] Distro toolchain hacks

---
 crates/extension/src/extension_builder.rs | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/crates/extension/src/extension_builder.rs b/crates/extension/src/extension_builder.rs
index 742f793290..63ae9bc3ae 100644
--- a/crates/extension/src/extension_builder.rs
+++ b/crates/extension/src/extension_builder.rs
@@ -287,6 +287,7 @@ impl ExtensionBuilder {
                     "-shared",
                     "-Os",
                     &format!("--target={RUST_TARGET}"),
+                    "--sysroot=/usr/share/wasi-sysroot",
                 ])
                 .arg(format!("-Wl,--export=tree_sitter_{grammar_name}"))
                 .arg("-o")
@@ -429,6 +430,7 @@ impl ExtensionBuilder {
         if sysroot.join("lib/rustlib").join(RUST_TARGET).exists() {
             return Ok(());
         }
+        bail!("install rust-wasm");
 
         let output = util::command::new_smol_command("rustup")
             .args(["target", "add", RUST_TARGET])
@@ -463,6 +465,7 @@ impl ExtensionBuilder {
                 }
             }
         }
+        bail!("install clang and rust-wasm");
 
         let url = if let Some(asset_name) = WASI_SDK_ASSET_NAME {
             format!("{WASI_SDK_URL}{asset_name}")
@@ -536,6 +539,9 @@ impl ExtensionBuilder {
     async fn install_wasi_sysroot_if_needed(&self) -> Result<()> {
         let url = format!("{WASI_SDK_URL}{WASI_SDK_SYSROOT_NAME}");
         let wasi_sysroot_dir = self.cache_dir.join("wasi-sysroot");
+        fs::remove_dir_all(&wasi_sysroot_dir).ok();
+        std::os::unix::fs::symlink("/usr/share/wasi-sysroot", wasi_sysroot_dir)?;
+        return Ok(());
         let mut response = self.http.get(&url, AsyncBody::default(), true).await?;
         // Write the response to a temporary file
         let tar_gz_path = self.cache_dir.join(WASI_SDK_SYSROOT_NAME);
