Don't require dependencies for running tests and generating documentation.

--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -13,11 +13,7 @@
 # List of programs, minimum versions, and download urls required to
 # bootstrap, maintain and release this project.
 
-buildreq='
-    git        -       http://git-scm.com
-    ldoc       1.4.2   http://rocks.moonscript.org/manifests/steved/ldoc-1.4.2-1.rockspec
-    specl      14.1.0  http://rocks.moonscript.org/manifests/gvvaughan/specl-14.1.0-1.rockspec
-'
+buildreq=''
 
 # List of slingshot files to link into stdlib tree before autotooling.
 slingshot_files='
