--- a/repl_type_completor.gemspec
+++ b/repl_type_completor.gemspec
@@ -22,6 +22 @@
-  # The `git ls-files -z` loads the files in the RubyGem that have been added into git.
-  spec.files = Dir.chdir(__dir__) do
-    `git ls-files -z`.split("\x0").reject do |f|
-      (File.expand_path(f) == __FILE__) || f.start_with?(*%w[bin/ test/ spec/ features/ .git .circleci appveyor])
-    end
-  end
+  spec.files = Dir["lib/**/*.rb", "sig/**/*"]
