--- fakeroot-1.10.1.orig/scripts/fakeroot.in	Wed Nov  5 19:48:00 2008
+++ fakeroot-1.10.1/scripts/fakeroot.in	Wed Nov  5 19:49:12 2008
@@ -28,7 +28,7 @@
 
 libfound=no
 
-GETOPTEST=`getopt --version`
+GETOPTEST=`getopt --version 2>/dev/null`
 case $GETOPTEST in
 getopt*) # GNU getopt
     TEMP=`getopt -l lib: -l faked: -l unknown-is-real -l fd-base: -l version -l help -- +l:f:i:s:ub:vh "$@"`
@@ -133,9 +133,9 @@
 PID=`echo $KEY_PID|cut -d: -f2`
 
 if [ "$WAITINTRAP" -eq 0 ]; then
-  trap "kill -s @signal@ $PID" EXIT INT
+  trap "kill -@signal@ $PID" EXIT INT
 else
-  trap 'FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS"  LD_PRELOAD="$LIB" /bin/ls -l / >/dev/null 2>&1; while kill -s @signal@ $PID 2>/dev/null; do sleep 0.1; done' EXIT INT
+  trap 'FAKEROOTKEY=$FAKEROOTKEY LD_LIBRARY_PATH="$PATHS"  LD_PRELOAD="$LIB" /bin/ls -l / >/dev/null 2>&1; while kill -@signal@ $PID 2>/dev/null; do sleep 0.1; done' EXIT INT
 fi
 
 if test -z "$FAKEROOTKEY" || test -z "$PID"; then
