fixes paths to binaries
--
diff --git a/src/warpinator-launch.py b/src/warpinator-launch.py
index e5c8974..2f7c584 100644
--- a/src/warpinator-launch.py
+++ b/src/warpinator-launch.py
@@ -128,7 +128,7 @@ else:
 
     launch_args = []
 
-    launch_args += ["/bin/bwrap"]
+    launch_args += ["/usr/bin/bwrap"]
     launch_args += ["--ro-bind",         "/",                                               "/"]
     launch_args += ["--dev",             "/dev"]
     launch_args += ["--bind",            rundir + "/dconf",                                 rundir + "/dconf"]
@@ -149,7 +149,7 @@ else:
     launch_args += ["--die-with-parent"]
 
     # launch_args += ["/bin/sh"]
-    launch_args += ["/bin/python3",      warpinator_path]
+    launch_args += ["/usr/bin/python3",      warpinator_path]
 
     ret = 0
 
