From c4443cd85d708f1a6c3e6c249e4acda2029863f2 Mon Sep 17 00:00:00 2001
From: "Hoa V. DINH" <dinh.viet.hoa@gmail.com>
Date: Mon, 25 Nov 2013 20:40:48 -0800
Subject: [PATCH] Fixed #86: fixed build for GnuTLS

---
 src/data-types/mailstream_ssl.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/data-types/mailstream_ssl.c b/src/data-types/mailstream_ssl.c
index 6002421..b4c0c70 100644
--- a/src/data-types/mailstream_ssl.c
+++ b/src/data-types/mailstream_ssl.c
@@ -521,6 +521,7 @@ static struct mailstream_ssl_data * ssl_data_new(int fd, time_t timeout,
   gnutls_certificate_credentials_t xcred;
   int r;
   struct mailstream_ssl_context * ssl_context = NULL;
+  unsigned int timeout_value;
   
   mailstream_ssl_init();
   
@@ -552,7 +553,7 @@ static struct mailstream_ssl_data * ssl_data_new(int fd, time_t timeout,
   gnutls_dh_set_prime_bits(session, 512);
   
   if (timeout == 0) {
-		timeout_value = mailstream_network_delay.tv_sec * 1000 + timeout.tv_usec / 1000;
+		timeout_value = mailstream_network_delay.tv_sec * 1000 + mailstream_network_delay.tv_usec / 1000;
   }
   else {
 		timeout_value = timeout;
-- 
1.8.5.5

