From b375332c53dea84d32f42c785e96352aacb79513 Mon Sep 17 00:00:00 2001
From: "Hoa V. DINH" <dinh.viet.hoa@gmail.com>
Date: Wed, 20 Nov 2013 21:48:44 -0800
Subject: [PATCH] Fixed #86: Fixed build for GnuTLS

---
 src/data-types/mailstream_ssl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/data-types/mailstream_ssl.c b/src/data-types/mailstream_ssl.c
index d0ef259..6002421 100644
--- a/src/data-types/mailstream_ssl.c
+++ b/src/data-types/mailstream_ssl.c
@@ -154,6 +154,7 @@ struct mailstream_ssl_data {
 #endif
 
 // Used to make OpenSSL thread safe
+#ifndef USE_GNUTLS
 #if defined (HAVE_PTHREAD_H) && !defined (WIN32) && defined (USE_SSL) && defined (LIBETPAN_REENTRANT)
   struct CRYPTO_dynlock_value
   {
@@ -224,6 +225,7 @@ struct mailstream_ssl_data {
     CRYPTO_set_dynlock_destroy_callback(dyn_destroy_function);
   }
 #endif
+#endif
 
 void mailstream_ssl_init_lock(void)
 {
-- 
1.8.5.5

