--- a/intl/Makefile.in
+++ b/intl/Makefile.in
@@ -153,11 +153,11 @@
 	  : ; \
 	fi
 	if test '@USE_INCLUDED_LIBINTL@' = yes; then \
-	  test @GLIBC21@ != no || $(mkinstalldirs) $(DESTDIR)$(libdir); \
-	  temp=$(DESTDIR)$(libdir)/t-charset.alias; \
-	  dest=$(DESTDIR)$(libdir)/charset.alias; \
-	  if test -f $(DESTDIR)$(libdir)/charset.alias; then \
-	    orig=$(DESTDIR)$(libdir)/charset.alias; \
+	  test @GLIBC21@ != no || $(mkinstalldirs) $(DESTDIR)$(datadir); \
+	  temp=$(DESTDIR)$(datadir)/t-charset.alias; \
+	  dest=$(DESTDIR)$(datadir)/charset.alias; \
+	  if test -f $(DESTDIR)$(datadir)/charset.alias; then \
+	    orig=$(DESTDIR)$(datadir)/charset.alias; \
 	    sed -f ref-add.sed $$orig > $$temp; \
 	    $(INSTALL_DATA) $$temp $$dest; \
 	    rm -f $$temp; \
