diff --git a/configure.ac b/configure.ac
index 0835cc7..dae98e0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,7 +132,7 @@ if test "x$tasks" = "xyes"; then
 fi
 
 # Checks for libwebkit
-PKG_CHECK_MODULES(LIBWEBKIT, webkit2gtk-4.0 >= 2.8.0, have_libwebkit=true,
+PKG_CHECK_MODULES(LIBWEBKIT, webkit2gtk-4.1 >= 2.8.0, have_libwebkit=true,
     have_libwebkit=false)
 if test "x$have_libwebkit" = "xtrue"; then
     libwebkit="yes"
