--- libbsd-0.6.0.orig/src/flopen.c
+++ libbsd-0.6.0/src/flopen.c
@@ -32,6 +32,7 @@
 #include <sys/stat.h>
 
 #include <errno.h>
+#include <fcntl.h>
 #include <stdarg.h>
 #include <unistd.h>
 #include <libutil.h>
--- libbsd-0.6.0.orig/src/nlist.c
+++ libbsd-0.6.0/src/nlist.c
@@ -37,10 +37,11 @@
 #include <sys/stat.h>
 #include <sys/file.h>
 #include <arpa/inet.h>
+#include <linux/a.out.h>
 
 #include <errno.h>
-#include <a.out.h>
 #include <stdio.h>
+#include <fcntl.h>
 #include <string.h>
 #include <unistd.h>
 
