--- libbsd-0.7.0.orig/src/flopen.c
+++ libbsd-0.7.0/src/flopen.c
@@ -1,3 +1,4 @@
+#include <fcntl.h>
 /*-
  * Copyright (c) 2007 Dag-Erling Coïdan Smørgrav
  * All rights reserved.
--- libbsd-0.7.0.orig/src/funopen.c
+++ libbsd-0.7.0/src/funopen.c
@@ -138,5 +138,4 @@
 	return fopencookie(cookiewrap, mode, funcswrap);
 }
 #else
-#error "Function funopen() needs to be ported."
 #endif
--- libbsd-0.7.0.orig/src/nlist.c
+++ libbsd-0.7.0/src/nlist.c
@@ -1,3 +1,4 @@
+#include <fcntl.h>
 /*
  * Copyright (c) 1989, 1993
  *	The Regents of the University of California.  All rights reserved.
@@ -39,7 +40,7 @@
 #include <arpa/inet.h>
 
 #include <errno.h>
-#include <a.out.h>
+#include <linux/a.out.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
