--- old/make/Defs.gmk
+++ new/make/Defs.gmk
@@ -411,7 +411,7 @@
 SORT    = /usr/bin/sort
 TEST 	= /usr/bin/test
 TIDY 	= /usr/bin/tidy
-TOUCH 	= /usr/bin/touch
+TOUCH 	= /bin/touch
 UNZIP 	= /usr/bin/unzip
 WC 	= /usr/bin/wc
 ZIP 	= /usr/bin/zip
--- old/make/build-all.sh
+++ new/make/build-all.sh
@@ -60,35 +60,10 @@
     if [ $CYGWIN -eq 1 ]; then echo `cygpath -m $1`; else echo $1; fi
 }
 
-get_scm_type() {
-    if [ -d .hg ]; then
-        echo "HG"
-    elif [ -d .git ]; then
-        echo "GIT"
-    else 
-        echo "Error: unrecognized repository, it must be Git or Mercurial" >&2
-        exit 1
-    fi
-}
-
-SCM_TYPE=`get_scm_type`
-
 get_root() {
-   case $SCM_TYPE in
-       HG)  hg root ;;
-       GIT) git rev-parse --show-toplevel ;;
-       *) echo "Error: unknown SCM" >&2 ; exit 1 ;;
-   esac
+	echo $builddir
 }
 
-get_tag_info() {
-   case $SCM_TYPE in
-       HG)  hg tags | grep jtreg | head -1 ;;
-       GIT) git tag | grep jtreg | tail -1 ;;
-       *) echo "Error: unknown SCM" >&2 ; exit 1 ;;
-   esac
-}
-
 export JAVA_HOME=$1
 export PATH="$JAVA_HOME:$PATH"
 
@@ -147,7 +122,7 @@
 ANT_VERSION=${ANT_VERSION:-apache-ant-1.9.4}
 ANT_ZIP=${ANT_DIR}/${ANT_VERSION}.zip
 WGet ${APACHE_ANT_URL}/${ANT_VERSION}-bin.zip ${ANT_ZIP}
-echo "ec57a35eb869a307abdfef8712f3688fff70887f  ${ANT_ZIP}" | ${SHASUM} --check -
+echo "ec57a35eb869a307abdfef8712f3688fff70887f  ${ANT_ZIP}" | ${SHASUM} -c -
 ${UNZIP} ${UNZIP_OPTS} -d ${ANT_DIR} ${ANT_ZIP}
 
 ANT_JAR=${ANT_DIR}/${ANT_VERSION}/lib/ant.jar
@@ -210,17 +185,17 @@
 ASM_JAR=${JCOV_DEPS_DIR}/asm-6.2.jar
 ASM_JAR_CHECKSUM='1b6c4ff09ce03f3052429139c2a68e295cae6604'
 WGet ${MAVEN_REPO_URL}/org/ow2/asm/asm/6.2/asm-6.2.jar ${ASM_JAR}
-echo "${ASM_JAR_CHECKSUM}  ${ASM_JAR}" | ${SHASUM} --check -
+echo "${ASM_JAR_CHECKSUM}  ${ASM_JAR}" | ${SHASUM} -c -
 
 ASM_TREE_JAR=${JCOV_DEPS_DIR}/asm-tree-6.2.jar
 ASM_TREE_JAR_CHECKSUM='61570e046111559f38d4e0e580c005f75988c0a6'
 WGet ${MAVEN_REPO_URL}/org/ow2/asm/asm-tree/6.2/asm-tree-6.2.jar ${ASM_TREE_JAR}
-echo "${ASM_TREE_JAR_CHECKSUM}  ${ASM_TREE_JAR}" | ${SHASUM} --check -
+echo "${ASM_TREE_JAR_CHECKSUM}  ${ASM_TREE_JAR}" | ${SHASUM} -c -
 
 ASM_UTIL_JAR=${JCOV_DEPS_DIR}/asm-utils-6.2.jar
 ASM_UTIL_JAR_CHECKSUM='a9690730f92cc79eeadc20e400ebb41eccce10b1'
 WGet ${MAVEN_REPO_URL}/org/ow2/asm/asm-util/6.2/asm-util-6.2.jar ${ASM_UTIL_JAR}
-echo "${ASM_UTIL_JAR_CHECKSUM}  ${ASM_UTIL_JAR}" | ${SHASUM} --check -
+echo "${ASM_UTIL_JAR_CHECKSUM}  ${ASM_UTIL_JAR}" | ${SHASUM} -c -
 
 # Build jcov
 JCOV_SRC_ZIP=${JCOV_BUILD_DIR}/source.zip
@@ -263,7 +238,7 @@
 
 JUNIT_JAR=${JUNIT_DEPS_DIR}/junit-4.10.jar
 WGet ${MAVEN_REPO_URL}/junit/junit/4.10/junit-4.10.jar ${JUNIT_JAR}
-echo "e4f1766ce7404a08f45d859fb9c226fc9e41a861  ${JUNIT_JAR}" | ${SHASUM} --check -
+echo "e4f1766ce7404a08f45d859fb9c226fc9e41a861  ${JUNIT_JAR}" | ${SHASUM} -c -
 
 ${UNZIP} ${UNZIP_OPTS} ${JUNIT_JAR} LICENSE.txt -d ${JUNIT_DEPS_DIR}
 JUNIT_LICENSE=${JUNIT_DEPS_DIR}/LICENSE.txt
@@ -274,18 +249,17 @@
 
 TESTNG_JAR=${TESTNG_DEPS_DIR}/testng-6.9.5.jar
 WGet ${MAVEN_REPO_URL}/org/testng/testng/6.9.5/testng-6.9.5.jar ${TESTNG_JAR}
-echo "5d12ea207fc47c3f341a3f8ecc88a3eac396a777  ${TESTNG_JAR}" | ${SHASUM} --check -
+echo "5d12ea207fc47c3f341a3f8ecc88a3eac396a777  ${TESTNG_JAR}" | ${SHASUM} -c -
 
 TESTNG_LICENSE=${TESTNG_DEPS_DIR}/LICENSE.txt
 WGet https://raw.githubusercontent.com/cbeust/testng/testng-6.9.5/LICENSE.txt ${TESTNG_LICENSE}
 
 JCOMMANDER_JAR=${TESTNG_DEPS_DIR}/jcommander-1.72.jar
 WGet ${MAVEN_REPO_URL}/com/beust/jcommander/1.72/jcommander-1.72.jar ${JCOMMANDER_JAR}
-echo "6375e521c1e11d6563d4f25a07ce124ccf8cd171  ${JCOMMANDER_JAR}" | ${SHASUM} --check -
+echo "6375e521c1e11d6563d4f25a07ce124ccf8cd171  ${JCOMMANDER_JAR}" | ${SHASUM} -c -
 
 
 ## Set version and build numbers to the latest tagged version by default
-TAG_INFO=`get_tag_info`
 if [ -z ${BUILD_NUMBER:-} ]; then
     BUILD_NUMBER=`echo $TAG_INFO | sed 's/jtreg\([0-9]\.[0-9]\)-\(b[0-9]*\).*/\2/'`
 fi
