--- a/Makefile.cmdline
+++ b/Makefile.cmdline
@@ -32,8 +32,8 @@
 
 # If you want linenoise, you can enable these.  At the moment linenoise
 # will cause some harmless compilation warnings.
-#CCOPTS += -DDUK_CMDLINE_FANCY -I./linenoise
-#CMDLINE_SOURCES += linenoise/linenoise.c
+CCOPTS += -DDUK_CMDLINE_FANCY -D_BSD_SOURCE -I./linenoise
+CMDLINE_SOURCES += linenoise/linenoise.c
 #duk: linenoise
 
 # Use the tools/configure.py utility to modify Duktape default configuration:
