--- a/Makefile.cmdline
+++ b/Makefile.cmdline
@@ -40,8 +40,8 @@
 # http://duktape.org/guide.html#compiling
 # http://wiki.duktape.org/Configuring.html
 
-duk: $(DUKTAPE_SOURCES) $(CMDLINE_SOURCES)
-	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $@ $(DEFINES) $(CCOPTS) $(DUKTAPE_SOURCES) $(CMDLINE_SOURCES) $(CCLIBS)
+duk: $(CMDLINE_SOURCES)
+	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -o $@ $(DEFINES) $(CCOPTS) $(CMDLINE_SOURCES) $(CCLIBS) -L. -lduktape
 
 linenoise/linenoise.c: linenoise
 linenoise:
--- a/Makefile.sharedlibrary
+++ b/Makefile.sharedlibrary
@@ -72,6 +72,7 @@
 libduktape.$(SO_REALNAME_SUFFIX):
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -fPIC -Wall -Wextra -Os -Wl,$(LD_SONAME_ARG),libduktape.$(SO_SONAME_SUFFIX) \
 		-o $@ $(DUKTAPE_SRCDIR)/duktape.c
+	ln -s $@ libduktape.so
 
 libduktaped.$(SO_REALNAME_SUFFIX):
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -shared -fPIC -g -Wall -Wextra -Os -Wl,$(LD_SONAME_ARG),libduktaped.$(SO_SONAME_SUFFIX) \
