--- a/classes/io/drivers/driver_posix.cpp
+++ b/classes/io/drivers/driver_posix.cpp
@@ -142,7 +142,7 @@
 {
 	if (stream == -1) return -1;
 
-	return lseek64(stream, newPos, SEEK_SET);
+	return lseek(stream, newPos, SEEK_SET);
 }
 
 S::Bool S::IO::DriverPOSIX::Truncate(Int64 newSize)
@@ -174,9 +174,9 @@
 	if (stream == -1) return -1;
 
 	Int64	 oldPos = GetPos();
-	Int64	 size = lseek64(stream, 0, SEEK_END);
+	Int64	 size = lseek(stream, 0, SEEK_END);
 
-	lseek64(stream, oldPos, SEEK_SET);
+	lseek(stream, oldPos, SEEK_SET);
 
 	return size;
 }
@@ -185,5 +185,5 @@
 {
 	if (stream == -1) return -1;
 
-	return lseek64(stream, 0, SEEK_CUR);
+	return lseek(stream, 0, SEEK_CUR);
 }
