diff --git a/Makefile b/Makefile
index 8e37cc2..b29fef8 100644
--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@ install: Makefile.inc hwdata.pc
 		install -m 644 $$foo $(DESTDIR)$(datadir)/$(NAME) ;\
 	done
 	mkdir -p -m 755 $(DESTDIR)$(libdir)/modprobe.d
-	install -m 644 -T blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf
+	install -m 644 blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf
 	mkdir -p -m 755 $(DESTDIR)$(datadir)/pkgconfig
 	install -m 644 hwdata.pc $(DESTDIR)$(datadir)/pkgconfig/
 
