diff --git a/Makefile b/Makefile
index 3cc432d..3372aa0 100644
--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@ install: Makefile.inc hwdata.pc
 	done
 	@if [ "$(blacklist)" = true ]; then \
 		mkdir -p -m 755 $(DESTDIR)$(libdir)/modprobe.d ;\
-		install -m 644 -T blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf ;\
+		install -m 644 blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf ;\
 	fi;
 	mkdir -p -m 755 $(DESTDIR)$(datadir)/pkgconfig
 	install -m 644 hwdata.pc $(DESTDIR)$(datadir)/pkgconfig/
