--- ./Makefile.orig	2014-04-22 06:38:35.320410230 +0000
+++ ./Makefile	2014-04-22 06:39:15.260889885 +0000
@@ -33,7 +33,7 @@
 		install -m 644 $$foo $(DESTDIR)$(datadir)/$(NAME) ;\
 	done
 	mkdir -p -m 755 $(DESTDIR)$(libdir)/modprobe.d
-	install -m 644 -T blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf
+	install -m 644 blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf
 
 commit:
 	git commit -a ||:
