--- a/doc/man1/CMakeLists.txt
+++ b/doc/man1/CMakeLists.txt
@@ -16,6 +16,6 @@

 add_custom_target (man1 DEPENDS ${DOC_FILES})

-install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+install (DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
          DESTINATION ${TIMEW_MANDIR}
          FILES_MATCHING PATTERN "*.1")
--- a/doc/man7/CMakeLists.txt
+++ b/doc/man7/CMakeLists.txt
@@ -16,6 +16,6 @@

 add_custom_target (man7 DEPENDS ${DOC_FILES})

-install (DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+install (DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
          DESTINATION ${TIMEW_MANDIR}
          FILES_MATCHING PATTERN "*.7")
