Author: Holger Jaekel <holger.jaekel@gmx.de>
Summary: remove duplicate template definition
----

--- a/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h
+++ b/src/aws-cpp-sdk-core/include/aws/core/endpoint/EndpointProviderBase.h
@@ -70,7 +70,5 @@
             virtual ResolveEndpointOutcome ResolveEndpoint(const EndpointParameters& endpointParameters) const = 0;
         };
 
-        // Export symbol from the DLL:
-        template class AWS_CORE_API EndpointProviderBase<Aws::Client::GenericClientConfiguration</*HasEndpointDiscovery*/ true> >;
     } // namespace Endpoint
 } // namespace Aws


--- a/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h
+++ b/src/aws-cpp-sdk-core/include/aws/core/endpoint/DefaultEndpointProvider.h
@@ -110,7 +110,5 @@
             BuiltInParametersT m_builtInParameters;
         };
 
-        // Export symbol from the DLL:
-        template class AWS_CORE_API DefaultEndpointProvider<Aws::Client::GenericClientConfiguration</*HasEndpointDiscovery*/ true> >;
     } // namespace Endpoint
 } // namespace Aws

--- a/src/aws-cpp-sdk-core/source/endpoint/DefaultEndpointProvider.cpp
+++ b/src/aws-cpp-sdk-core/source/endpoint/DefaultEndpointProvider.cpp
@@ -12,13 +12,6 @@
 namespace Endpoint
 {
 
-/**
- * Export endpoint provider symbols from DLL
- */
-template class AWS_CORE_API DefaultEndpointProvider<Aws::Client::GenericClientConfiguration<false>,
-            Aws::Endpoint::BuiltInParameters,
-            Aws::Endpoint::ClientContextParameters>;
-
 char CharToDec(const char c)
 {
     if(c >= '0' && c <= '9')

--- a/src/aws-cpp-sdk-core/source/endpoint/EndpointProviderBase.cpp
+++ b/src/aws-cpp-sdk-core/source/endpoint/EndpointProviderBase.cpp
@@ -9,12 +9,6 @@
 {
 namespace Endpoint
 {
-/**
- * Export endpoint provider symbols from DLL
- */
-template class AWS_CORE_API EndpointProviderBase<Aws::Client::GenericClientConfiguration<false>,
-            Aws::Endpoint::BuiltInParameters,
-            Aws::Endpoint::ClientContextParameters>;
 
 } // namespace Endpoint
 } // namespace Aws

--- a/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3CrtEndpointProvider.h
+++ b/generated/src/aws-cpp-sdk-s3-crt/include/aws/s3-crt/S3CrtEndpointProvider.h
@@ -79,14 +79,6 @@
 
 namespace Endpoint
 {
-/**
- * Export endpoint provider symbols from DLL
- */
-template class AWS_S3CRT_API
-    Aws::Endpoint::EndpointProviderBase<S3Crt::Endpoint::S3CrtClientConfiguration, S3Crt::Endpoint::S3CrtBuiltInParameters, S3Crt::Endpoint::S3CrtClientContextParameters>;
-
-template class AWS_S3CRT_API
-    Aws::Endpoint::DefaultEndpointProvider<S3Crt::Endpoint::S3CrtClientConfiguration, S3Crt::Endpoint::S3CrtBuiltInParameters, S3Crt::Endpoint::S3CrtClientContextParameters>;
 } // namespace Endpoint
 
 namespace S3Crt

--- a/generated/src/aws-cpp-sdk-s3control/include/aws/s3control/S3ControlEndpointProvider.h
+++ b/generated/src/aws-cpp-sdk-s3control/include/aws/s3control/S3ControlEndpointProvider.h
@@ -61,14 +61,6 @@
 
 namespace Endpoint
 {
-/**
- * Export endpoint provider symbols from DLL
- */
-template class AWS_S3CONTROL_API
-    Aws::Endpoint::EndpointProviderBase<S3Control::Endpoint::S3ControlClientConfiguration, S3Control::Endpoint::S3ControlBuiltInParameters, S3Control::Endpoint::S3ControlClientContextParameters>;
-
-template class AWS_S3CONTROL_API
-    Aws::Endpoint::DefaultEndpointProvider<S3Control::Endpoint::S3ControlClientConfiguration, S3Control::Endpoint::S3ControlBuiltInParameters, S3Control::Endpoint::S3ControlClientContextParameters>;
 } // namespace Endpoint
 
 namespace S3Control

--- a/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3EndpointProvider.h
+++ b/generated/src/aws-cpp-sdk-s3/include/aws/s3/S3EndpointProvider.h
@@ -79,14 +79,6 @@
 
 namespace Endpoint
 {
-/**
- * Export endpoint provider symbols from DLL
- */
-template class AWS_S3_API
-    Aws::Endpoint::EndpointProviderBase<S3::Endpoint::S3ClientConfiguration, S3::Endpoint::S3BuiltInParameters, S3::Endpoint::S3ClientContextParameters>;
-
-template class AWS_S3_API
-    Aws::Endpoint::DefaultEndpointProvider<S3::Endpoint::S3ClientConfiguration, S3::Endpoint::S3BuiltInParameters, S3::Endpoint::S3ClientContextParameters>;
 } // namespace Endpoint
 
 namespace S3
