From 64676437e9aa46f3cf614e06e4f03e31fc461c94 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Wed, 23 Dec 2020 21:29:05 +0100
Subject: [PATCH] Fix double-free in charset sdb

---
 libr/core/core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libr/core/core.c b/libr/core/core.c
index 81c991ffc..f72e581d7 100644
--- a/libr/core/core.c
+++ b/libr/core/core.c
@@ -2714,6 +2714,9 @@ R_API bool r_core_init(RCore *core) {
 	core->lastcmd = NULL;
 	core->cmdlog = NULL;
 	core->print->charset->db = sdb_ns (core->sdb, "charset", 1);
+	core->print->charset->db->refs++; // increase reference counter to avoid double-free
+	// ideally sdb_ns_set should be used here, but it doesnt seems to work well. must fix
+	// sdb_ns_set (DB, "charset", core->print->charset->db);
 	core->stkcmd = NULL;
 	core->cmdqueue = NULL;
 	core->cmdrepeat = true;
