From 645835794b0013b52042d5762d98d03fcbf250f6 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Sergi=20=C3=80lvarez=20i=20Capilla?= <pancake@nowsecure.com>
Date: Mon, 15 Nov 2021 18:10:04 +0100
Subject: [PATCH] arm64's NAND isnt supported in capstone4

---
 libr/anal/p/anal_arm_cs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/libr/anal/p/anal_arm_cs.c b/libr/anal/p/anal_arm_cs.c
index dbdfcafd898..307b3997486 100644
--- a/libr/anal/p/anal_arm_cs.c
+++ b/libr/anal/p/anal_arm_cs.c
@@ -1361,13 +1361,13 @@ static int analop64_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int l
 	case ARM64_INS_AND:
 		OPCALL("&");
 		break;
-	case ARM64_INS_NAND:
-		OPCALL_NEG("&");
-		break;
 	case ARM64_INS_ORR:
 		OPCALL("|");
 		break;
 #if CS_API_MAJOR > 4	
+	case ARM64_INS_NAND:
+		OPCALL_NEG("&");
+		break;
 	case ARM64_INS_ADDS:
 	case ARM64_INS_ADCS:
 		OPCALL("+");
