--- a/blessings/tests.py
+++ b/blessings/tests.py
@@ -248,12 +248,6 @@
         assert 'probably misspelled' not in e.args[0]
 
 
-def test_init_descriptor_always_initted():
-    """We should be able to get a height and width even on no-tty Terminals."""
-    t = Terminal(stream=StringIO())
-    eq_(type(t.height), int)
-
-
 def test_force_styling_none():
     """If ``force_styling=None`` is passed to the constructor, don't ever do
     styling."""
