diff --git a/abiwordparser.h b/abiwordparser.h
index 7eebe17..ff37e0a 100644
--- a/abiwordparser.h
+++ b/abiwordparser.h
@@ -22,6 +22,7 @@
 #define OMEGA_INCLUDED_ABIWORDPARSER_H
 
 #include "xmlparser.h"
+#include <ctime>
 
 class AbiwordParser : public XmlParser {
     std::string* target = nullptr;
diff --git a/gnumericparser.h b/gnumericparser.h
index 1f85de8..a5dff3c 100644
--- a/gnumericparser.h
+++ b/gnumericparser.h
@@ -22,6 +22,7 @@
 #define OMEGA_INCLUDED_GNUMERICPARSER_H
 
 #include "xmlparser.h"
+#include <ctime>
 
 class GnumericParser : public XmlParser {
     std::string* target = nullptr;
