diff -ru uClibc-0.9.30.1.orig/libc/inet/resolv.c uClibc-0.9.30.1/libc/inet/resolv.c
--- uClibc-0.9.30.1.orig/libc/inet/resolv.c	2009-03-10 15:32:14.000000000 +0000
+++ uClibc-0.9.30.1/libc/inet/resolv.c	2009-03-10 15:39:20.000000000 +0000
@@ -1331,7 +1331,8 @@
 	i = __dns_lookup(dname, type, __nameserversXX, __nameserverXX, &packet, &a);
 
 	if (i < 0) {
-		h_errno = TRY_AGAIN;
+		if (!h_errno) /* TODO: can this ever happen? */
+			h_errno = TRY_AGAIN;
 		return -1;
 	}
 
