diff --git a/setup.py b/setup.py
index d5f3c7e..b90473d 100755
--- a/setup.py
+++ b/setup.py
@@ -28,12 +28,7 @@ def version_scheme(version):
 
 setup(
     name="sshuttle",
-    use_scm_version={
-        'write_to': "sshuttle/version.py",
-        'version_scheme': version_scheme,
-    },
-    setup_requires=['setuptools_scm'],
-    # version=version,
+    #version="",
     url='https://github.com/sshuttle/sshuttle',
     author='Brian May',
     author_email='brian@linuxpenguins.xyz',
