Patch-Source: https://github.com/softlayer/softlayer-zeep/pull/6
--- a/src/zeep/transports.py
+++ b/src/zeep/transports.py
@@ -184,12 +184,12 @@
         self.cache = cache
         self.wsdl_client = wsdl_client or httpx.Client(
             verify=verify_ssl,
-            proxies=proxy,
+            proxy=proxy,
             timeout=timeout,
         )
         self.client = client or httpx.AsyncClient(
             verify=verify_ssl,
-            proxies=proxy,
+            proxy=proxy,
             timeout=operation_timeout,
         )
         self.logger = logging.getLogger(__name__)
