From c0a0826a1d88b0268d5b29fd1613dfc572a7c138 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 2 Sep 2023 17:32:13 +0200
Subject: [PATCH] syscalls: add workaround for lack of stat64 symbol on musl

---
 guix/build/syscalls.scm | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/guix/build/syscalls.scm b/guix/build/syscalls.scm
index d947b010d3..f8a9011b6a 100644
--- a/guix/build/syscalls.scm
+++ b/guix/build/syscalls.scm
@@ -905,7 +905,11 @@ (define-c-struct %statfs                          ;<bits/statfs.h>
   (spare            (array fsword 4)))
 
 (define statfs
-  (let ((proc (syscall->procedure int "statfs64" '(* *))))
+  (let ((proc (syscall->procedure int (cond
+                                        (linux-gnu?  "statfs64")
+                                        (linux-musl? "statfs")
+                                        (else (error "unknown linux variant")))
+                                      '(* *))))
     (lambda (file)
       "Return a <file-system> data structure describing the file system
 mounted at FILE."
