diff --git a/visidata/shell.py b/visidata/shell.py
index a05fe27..d6d1f62 100644
--- a/visidata/shell.py
+++ b/visidata/shell.py
@@ -67,7 +67,7 @@ class ColumnShell(Column):
                 else:
                     args.append(arg)
 
-            p = subprocess.Popen([os.getenv('SHELL', 'bash'), '-c', ' '.join(args)],
+            p = subprocess.Popen([os.getenv('SHELL', 'sh'), '-c', ' '.join(args)],
                     stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             return p.communicate()
         except Exception as e:
