diff --git a/setup.py b/setup.py
index 23eed1d4..e84ce823 100755
--- a/setup.py
+++ b/setup.py
@@ -24,7 +24,6 @@ setup(name='visidata',
       install_requires=[
           'python-dateutil',
           'windows-curses; platform_system == "Windows"',
-          'importlib-metadata >= 3.6',
       ],
       packages=['visidata', 'visidata.loaders', 'visidata.vendor', 'visidata.tests', 'visidata.ddw', 'visidata.man'],
       data_files=[('share/man/man1', ['visidata/man/vd.1', 'visidata/man/visidata.1'])],
diff --git a/visidata/settings.py b/visidata/settings.py
index 1fc62df4..d9c44ad6 100644
--- a/visidata/settings.py
+++ b/visidata/settings.py
@@ -375,7 +375,7 @@ def loadConfigAndPlugins(vd, args=AttrDict()):
     # autoload installed plugins first
     args_plugins_autoload = args.plugins_autoload if 'plugins_autoload' in args else True
     if not args.nothing and args_plugins_autoload and vd.options.plugins_autoload:
-        from importlib_metadata import entry_points  # a backport which supports < 3.8 https://github.com/pypa/twine/pull/732
+        from importlib.metadata import entry_points
         try:
             eps = entry_points()
             eps_visidata = eps.select(group='visidata.plugins') if 'visidata.plugins' in eps.groups else []
