diff -ru nfdump-1.6.9.orig/bin/fts_compat.c nfdump-1.6.9/bin/fts_compat.c
--- nfdump-1.6.9.orig/bin/fts_compat.c	2014-11-18 11:55:29.794283835 -0200
+++ nfdump-1.6.9/bin/fts_compat.c	2011-12-29 17:11:27.000000000 -0200
@@ -949,7 +949,7 @@
 #endif
 
 #ifndef ALIGN
-#define ALIGN(p)   (((u_int)(p) + ALIGNBYTES) &~ ALIGNBYTES)
+#define ALIGN(p)   (((u_long)(p) + ALIGNBYTES) &~ ALIGNBYTES)
 #endif
 
 	size_t len;
