From fe3ce1d7fef26030b45ac4ab4e3a613735dc20d7 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kaarle.ritvanen@datakunkku.fi>
Date: Fri, 28 Mar 2014 10:56:00 +0200
Subject: make boolean variables work


diff --git a/awall/policy.lua b/awall/policy.lua
index 585b04c..4e5d72b 100644
--- a/awall/policy.lua
+++ b/awall/policy.lua
@@ -1,6 +1,6 @@
 --[[
 Policy file handling for Alpine Wall
-Copyright (C) 2012-2013 Kaarle Ritvanen
+Copyright (C) 2012-2014 Kaarle Ritvanen
 See LICENSE file for license details
 ]]--
 
@@ -43,7 +43,7 @@ function PolicyConfig:expand()
 	 table.insert(visited, name)
 	 
 	 local var = self.data.variable[name]
-	 if not var then raise('Invalid variable reference: '..name) end
+	 if var == nil then raise('Invalid variable reference: '..name) end
 	 
 	 if si == 1 and ei == string.len(value) then value = var
 	 elseif contains({'number', 'string'}, type(var)) then
-- 
cgit v0.10.1

