From 13bf8b6ab664bcd0b11c2a6dbf9eb3cd319062cd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Fri, 2 Aug 2013 16:10:08 +0300
Subject: [PATCH] fix ratelimiting to work with bursts properly

---
 awall/modules/filter.lua | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/awall/modules/filter.lua b/awall/modules/filter.lua
index d21b79e..bdf6ae6 100644
--- a/awall/modules/filter.lua
+++ b/awall/modules/filter.lua
@@ -212,7 +212,7 @@ function Filter:extraoptfrags()
       if count > RECENT_MAX_COUNT then
 	 ofrags = {
 	    {
-	       opts='-m hashlimit --hashlimit-upto '..count..'/second --hashlimit-mode srcip --hashlimit-name '..chain,
+	       opts='-m hashlimit --hashlimit-upto '..count..'/second --hashlimit-burst '..count..' --hashlimit-mode srcip --hashlimit-name '..chain,
 	       target=logchain(self.log, 'accept', 'ACCEPT')
 	    },
 	    {target='DROP'}
-- 
1.8.3.3

