diff --git a/setup.py b/setup.py
index b201a7b..03942ec 100644
--- a/setup.py
+++ b/setup.py
@@ -9,7 +9,7 @@ setup(
     name = "html",
     version = __version__,
     description = "simple, elegant HTML, XHTML and XML generation",
-    long_description = __doc__.decode('utf8'),
+    long_description = __doc__,
     author = "Richard Jones",
     author_email = "rjones@ekit-inc.com",
     py_modules = ['html'],

