diff --git a/configure b/configure
index 9e28a58..38ba929 100755
--- a/configure
+++ b/configure
@@ -24602,14 +24602,14 @@ if test -n "$EPUB_CFLAGS"; then
     pkg_cv_EPUB_CFLAGS="$EPUB_CFLAGS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0 >= \$WEBKIT_REQUIRED \\
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1 >= \$WEBKIT_REQUIRED \\
        libxml-2.0 >= \$LIBXML_REQUIRED zlib\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_EPUB_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  pkg_cv_EPUB_CFLAGS=`$PKG_CONFIG --cflags "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
@@ -24622,14 +24622,14 @@ if test -n "$EPUB_LIBS"; then
     pkg_cv_EPUB_LIBS="$EPUB_LIBS"
  elif test -n "$PKG_CONFIG"; then
     if test -n "$PKG_CONFIG" && \
-    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.0 >= \$WEBKIT_REQUIRED \\
+    { { printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"webkit2gtk-4.1 >= \$WEBKIT_REQUIRED \\
        libxml-2.0 >= \$LIBXML_REQUIRED zlib\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  ($PKG_CONFIG --exists --print-errors "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib") 2>&5
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_EPUB_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+  pkg_cv_EPUB_LIBS=`$PKG_CONFIG --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
@@ -24651,10 +24651,10 @@ else
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        EPUB_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+	        EPUB_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>&1`
         else
-	        EPUB_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.0 >= $WEBKIT_REQUIRED \
+	        EPUB_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "webkit2gtk-4.1 >= $WEBKIT_REQUIRED \
        libxml-2.0 >= $LIBXML_REQUIRED zlib" 2>&1`
         fi
 	# Put the nasty error message in config.log where it belongs
