--- tg_owt-575fb17d2853c43329e45f6693370f5e41668055/CMakeLists.txt
+++ tg_owt-575fb17d2853c43329e45f6693370f5e41668055-patched/CMakeLists.txt
@@ -69,7 +69,6 @@
 include(cmake/librnnoise.cmake)
 include(cmake/libsrtp.cmake)
 include(cmake/libusrsctp.cmake)
-include(cmake/libvpx.cmake)
 include(cmake/libyuv.cmake)
 if (NOT WIN32 AND NOT APPLE)
     include(cmake/libevent.cmake)
@@ -126,6 +125,7 @@
     PRIVATE
         tg_owt::tg_owt_avx2
         tg_owt::tg_owt_sse2
+        /usr/lib/libvpx.so
     )
 endif()
 
@@ -138,7 +138,6 @@
 
 target_sources(tg_owt
 PRIVATE
-    ${libvpx_yasm_objects}
     ${libopenh264_yasm_objects}
 )
 
@@ -2330,18 +2329,6 @@
 elseif (APPLE)
     set(platform_export
         libsdkmacos
-    )
-endif()
-
-set(vpx_export)
-if (is_x86 OR is_x64)
-    set(vpx_export
-        libvpx_mmx
-        libvpx_sse2
-        libvpx_ssse3
-        libvpx_sse4
-        libvpx_avx
-        libvpx_avx2
     )
 endif()
 
