diff --git a/import/Lomiri/Content/CMakeLists.txt b/import/Lomiri/Content/CMakeLists.txt
index d2382a3..055ea34 100644
--- a/import/Lomiri/Content/CMakeLists.txt
+++ b/import/Lomiri/Content/CMakeLists.txt
@@ -93,7 +93,7 @@ endif()
 
 if(NOT CMAKE_CROSSCOMPILING)
 add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/plugins.qmltypes
-    COMMAND qmlplugindump -qt=qt5 -noinstantiate -notrelocatable Lomiri.Content 1.1 ../../ > ${CMAKE_CURRENT_BINARY_DIR}/plugins.qmltypes
+    COMMAND qmlplugindump-qt5 -noinstantiate -notrelocatable Lomiri.Content 1.1 ../../ > ${CMAKE_CURRENT_BINARY_DIR}/plugins.qmltypes
     DEPENDS ${PLUGIN}
     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
 )
