--- a/test/test-list.h
+++ b/test/test-list.h
@@ -253,8 +253,10 @@
 TEST_DECLARE   (spawn_and_kill_with_std)
 TEST_DECLARE   (spawn_and_ping)
 TEST_DECLARE   (spawn_preserve_env)
+#if 0
 TEST_DECLARE   (spawn_setuid_fails)
 TEST_DECLARE   (spawn_setgid_fails)
+#endif
 TEST_DECLARE   (spawn_stdout_to_file)
 TEST_DECLARE   (spawn_stdout_and_stderr_to_file)
 TEST_DECLARE   (spawn_stdout_and_stderr_to_file2)
@@ -384,7 +386,9 @@
 TEST_DECLARE   (emfile)
 TEST_DECLARE   (close_fd)
 TEST_DECLARE   (spawn_fs_open)
+#if 0
 TEST_DECLARE   (spawn_setuid_setgid)
+#endif
 TEST_DECLARE   (we_get_signal)
 TEST_DECLARE   (we_get_signals)
 TEST_DECLARE   (we_get_signal_one_shot)
@@ -747,8 +751,10 @@
   TEST_ENTRY  (spawn_and_kill_with_std)
   TEST_ENTRY  (spawn_and_ping)
   TEST_ENTRY  (spawn_preserve_env)
+#if 0
   TEST_ENTRY  (spawn_setuid_fails)
   TEST_ENTRY  (spawn_setgid_fails)
+#endif
   TEST_ENTRY  (spawn_stdout_to_file)
   TEST_ENTRY  (spawn_stdout_and_stderr_to_file)
   TEST_ENTRY  (spawn_stdout_and_stderr_to_file2)
@@ -782,7 +788,9 @@
   TEST_ENTRY  (emfile)
   TEST_ENTRY  (close_fd)
   TEST_ENTRY  (spawn_fs_open)
+#if 0
   TEST_ENTRY  (spawn_setuid_setgid)
+#endif
   TEST_ENTRY  (we_get_signal)
   TEST_ENTRY  (we_get_signals)
   TEST_ENTRY  (we_get_signal_one_shot)
