diff -upr Idris2-0.7.0.orig/tests/idris2/reflection/reflection024/expected Idris2-0.7.0/tests/idris2/reflection/reflection024/expected
--- Idris2-0.7.0.orig/tests/idris2/reflection/reflection024/expected	2024-01-07 10:32:17.876048136 +0100
+++ Idris2-0.7.0/tests/idris2/reflection/reflection024/expected	2024-01-07 10:32:46.676121703 +0100
@@ -35,7 +35,7 @@ LOG elab:0: written to existentToWrite
 LOG elab:0: written to nonExistentToWrite
 existent to read
 second line
-cat: src/nonExistentToRead: No such file or directory
+cat: can't open 'src/nonExistentToRead': No such file or directory
 WRITTEN CONTENTS
 LA-LA-LA
 WRITTEN CONTENTS
