diff -upr luakit-2017.08.10.orig/config.mk luakit-2017.08.10/config.mk
--- luakit-2017.08.10.orig/config.mk	2017-09-16 13:29:59.436651544 +0200
+++ luakit-2017.08.10/config.mk	2017-09-16 13:30:42.353527111 +0200
@@ -2,7 +2,7 @@
 
 # Compile/link options.
 CC         ?= gcc
-CFLAGS     += -std=gnu99 -ggdb -W -Wall -Wextra -Werror=unused-result
+CFLAGS     += -std=gnu99 -W -Wall -Wextra -Werror=unused-result
 LDFLAGS    +=
 CPPFLAGS   +=
 
