--- a/src/engine/display/font_description.cpp
+++ b/src/engine/display/font_description.cpp
@@ -17,6 +17,7 @@
 #include "engine/display/font_description.hpp"
 
 #include <stdexcept>
+#include <cstdint>
 
 #include "util/file_reader.hpp"
 #include "util/raise_exception.hpp"
--- a/src/engine/display/font_description.hpp
+++ b/src/engine/display/font_description.hpp
@@ -18,6 +18,7 @@
 #define HEADER_PINGUS_ENGINE_DISPLAY_FONT_DESCRIPTION_HPP
 
 #include <vector>
+#include <cstdint>
 
 #include "math/rect.hpp"
 #include "util/pathname.hpp"
--- a/src/pingus/screens/demo_session.cpp
+++ b/src/pingus/screens/demo_session.cpp
@@ -18,6 +18,7 @@
 
 #include <algorithm>
 #include <iostream>
+#include <functional>
 
 #include "engine/gui/gui_manager.hpp"
 #include "engine/gui/surface_button.hpp"
--- a/src/util/pathname.hpp
+++ b/src/util/pathname.hpp
@@ -19,6 +19,7 @@
 
 #include <string>
 #include <vector>
+#include <cstdint>
 
 /** Simple class to allow a distinction of paths that refer to the
  *  filesystem and paths that refer to the datadir, it also hides
