--- a/galerautils/src/gu_thread.cpp
+++ b/galerautils/src/gu_thread.cpp
@@ -87,7 +87,8 @@
 #if defined(__sun__)
     struct sched_param spstr = { sp.prio(), { 0, } /* sched_pad array */};
 #else
-    struct sched_param spstr = { sp.prio() };
+    /* musl has some reserved fields in sched_param... */
+    struct sched_param spstr = { sp.prio(), 0, 0, 0, 0, 0, 0 };
 #endif
     int err;
     if ((err = pthread_setschedparam(thd, sp.policy(), &spstr)) != 0)
