Don't use systemd timers for healthchecks

diff --git a/libpod/healthcheck.go b/libpod/healthcheck.go
index 9b9d12b..ed399e0 100644
--- a/libpod/healthcheck.go
+++ b/libpod/healthcheck.go
@@ -257,11 +257,5 @@ func (c *Container) healthCheckStatus() (string, error) {
 }
 
 func (c *Container) disableHealthCheckSystemd() bool {
-	if os.Getenv("DISABLE_HC_SYSTEMD") == "true" {
-		return true
-	}
-	if c.config.HealthCheckConfig.Interval == 0 {
-		return true
-	}
-	return false
+	return true
 }
