--- a/lib/functions.php
+++ b/lib/functions.php
@@ -2147,7 +2147,7 @@
 			if ($_SESSION[APPCONFIG]->getValue('password', 'no_random_crypt_salt'))
 				$new_value = sprintf('{CRYPT}%s',crypt($password_clear,substr($password_clear,0,2)));
 			else
-				$new_value = sprintf('{CRYPT}%s',crypt($password_clear,random_salt(2)));
+				$new_value = sprintf('{CRYPT}%s',crypt($password_clear,'$6$'.random_salt(2)));
 
 			break;
 
