diff --git a/tests/test_backoff_async.py b/tests/test_backoff_async.py
index 226ef08..7d037a3 100644
--- a/tests/test_backoff_async.py
+++ b/tests/test_backoff_async.py
@@ -665,7 +665,7 @@ async def test_on_exception_callable_gen_kwargs():
 
 
 @pytest.mark.asyncio
-async def test_on_exception_coro_cancelling(event_loop):
+async def test_on_exception_coro_cancelling():
     sleep_started_event = asyncio.Event()
 
     @backoff.on_predicate(backoff.expo)
@@ -679,6 +679,11 @@ async def test_on_exception_coro_cancelling(event_loop):
 
         return False
 
+    try:
+        event_loop = asyncio.get_event_loop()
+    except RuntimeError:
+        event_loop = asyncio.new_event_loop()
+
     task = event_loop.create_task(coro())
 
     await sleep_started_event.wait()
