diff -urN libeatmydata-libeatmydata-105.orig/m4/pandora_canonical.m4 libeatmydata-libeatmydata-105/m4/pandora_canonical.m4
--- libeatmydata-libeatmydata-105.orig/m4/pandora_canonical.m4	2021-07-26 23:33:42.433846352 -0600
+++ libeatmydata-libeatmydata-105/m4/pandora_canonical.m4	2021-07-26 23:34:18.850708125 -0600
@@ -72,8 +72,6 @@
     ])
   ])
 
-  AC_CONFIG_MACRO_DIR([m4])
-
   m4_if(m4_substr(m4_esyscmd(test -d src && echo 0),0,1),0,[
     AC_CONFIG_HEADERS([src/config.h])
   ],[
