diff -rupN a/cmd/chasquid-util/test_dkim.cmy b/cmd/chasquid-util/test_dkim.cmy
--- a/cmd/chasquid-util/test_dkim.cmy	2026-06-07 11:25:56.000000000 +0000
+++ b/cmd/chasquid-util/test_dkim.cmy	2026-08-22 21:37:57.111399718 +0000
@@ -146,7 +146,7 @@ c wait 0
 c = ./chasquid-util -C=.config dkim-sign
 c -> Invalid message.
 c close
-c <- Error parsing message: malformed header line: Invalid message.
+c <- Error parsing message: malformed header line: "Invalid message."
 c wait 1
 
 c = ./chasquid-util -C=.config dkim-sign
