--- a/CMakeLists.txt	2021-09-08 07:43:30.622938936 +0200
+++ b/CMakeLists.txt	2021-09-08 07:29:53.172008240 +0200
@@ -20,17 +20,7 @@ include(mjpg_streamer_utils)
 #
 # Get the current git hash
 #
-execute_process(
-  COMMAND git rev-parse HEAD
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  RESULT_VARIABLE GIT_RESULT
-  OUTPUT_VARIABLE GIT_HASH
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-)
-
-if(GIT_RESULT EQUAL 0)
-  add_definitions("-DGIT_HASH=\"${GIT_HASH}\"")
-endif()
+add_definitions("-DGIT_HASH=\"${GIT_HASH}\"")
 
 #
 # Options
