--- ./generate_config	2004-07-23 05:58:48.000000000 +0000
+++ ./generate_config	2009-01-27 10:09:11.000000000 +0000
@@ -4,7 +4,7 @@
 # Figure out the system's mailname
 #
 
-syshostname=`hostname --fqdn`
+syshostname="localhost"
 if test -f /etc/mailname
 then
 	mailname="`head -1 /etc/mailname`"
@@ -15,23 +15,7 @@
 	mailname=$syshostname
 fi
 
-echo "Please enter the mail name of your system."
-echo "This is the hostname portion of the address to be shown"
-echo "on outgoing news and mail messages headers."
-echo "The default is $syshostname, your system's host name."
-echo
-echo -n "Mail name [$syshostname]: "
-read mailname
-echo
-
-echo -n "Please enter the SMTP port number [25]: "
-read smtpport
-if test -z "$smtpport"
-then
-	mailhub=$mailhub
-else
-	mailhub="$mailhub:$smtpport"
-fi
+mailhub="$mailhub:$smtpport"
 
 #
 # Generate configuration file
@@ -56,11 +40,5 @@
 # Where will the mail seem to come from?
 #rewriteDomain=`echo -n $mailname`
 # The full hostname
-hostname=`hostname --fqdn`
+#hostname="localhost"
 EOF
-
-echo
-echo
-echo "Please check the configuration file $1 for correctness."
-echo
-echo
