diff -up ssmtp.orig/ssmtp.c ssmtp/ssmtp.c
--- ssmtp.orig/ssmtp.c	2009-11-23 10:55:11.000000000 +0100
+++ ssmtp/ssmtp.c	2015-07-27 17:34:12.498140804 +0200
@@ -1157,7 +1157,7 @@ int smtp_open(char *host, int port)
 	snprintf(servname, sizeof(servname), "%d", port);
 
 	/* Check we can reach the host */
-	if (getaddrinfo(host, servname, &hints, &ai0)) {
+	if (getaddrinfo(host, servname, &hints, &ai0) != 0) {
 		log_event(LOG_ERR, "Unable to locate %s", host);
 		return(-1);
 	}
