diff -ur webkit-1.6.1/Source/JavaScriptCore/jit/JITStubs.cpp webkit-1.6.1.new/Source/JavaScriptCore/jit/JITStubs.cpp
--- webkit-1.6.1/Source/JavaScriptCore/jit/JITStubs.cpp	2011-10-21 21:18:46.000000000 +0400
+++ webkit-1.6.1.new/Source/JavaScriptCore/jit/JITStubs.cpp	2011-10-21 21:19:51.397529165 +0400
@@ -80,7 +80,7 @@
 #define THUMB_FUNC_PARAM(name)
 #endif
 
-#if (OS(LINUX) || OS(FREEBSD)) && CPU(X86_64)
+#if (OS(LINUX) || OS(FREEBSD)) && (CPU(X86) || CPU(X86_64))
 #define SYMBOL_STRING_RELOCATION(name) #name "@plt"
 #elif OS(DARWIN) || (CPU(X86_64) && COMPILER(MINGW) && !GCC_VERSION_AT_LEAST(4, 5, 0))
 #define SYMBOL_STRING_RELOCATION(name) "_" #name
diff -ur webkit-1.6.1/Source/JavaScriptCore/jit/ThunkGenerators.cpp webkit-1.6.1.new/Source/JavaScriptCore/jit/ThunkGenerators.cpp
--- webkit-1.6.1/Source/JavaScriptCore/jit/ThunkGenerators.cpp	2011-10-21 21:18:46.000000000 +0400
+++ webkit-1.6.1.new/Source/JavaScriptCore/jit/ThunkGenerators.cpp	2011-10-21 21:20:01.029529805 +0400
@@ -108,7 +108,7 @@
 #define SYMBOL_STRING(name) #name
 #endif
     
-#if (OS(LINUX) || OS(FREEBSD)) && CPU(X86_64)
+#if (OS(LINUX) || OS(FREEBSD)) && (CPU(X86) || CPU(X86_64))
 #define SYMBOL_STRING_RELOCATION(name) #name "@plt"
 #elif OS(DARWIN) || (CPU(X86_64) && COMPILER(MINGW) && !GCC_VERSION_AT_LEAST(4, 5, 0))
 #define SYMBOL_STRING_RELOCATION(name) "_" #name
