diff --git a/src/hdy-swipe-tracker.c b/src/hdy-swipe-tracker.c
index 03b417811764b1d1d608f4919fa94816d08a5cec..ba01bd057b49496fb3d4a8565cedcddacd600487 100644
--- a/src/hdy-swipe-tracker.c
+++ b/src/hdy-swipe-tracker.c
@@ -944,14 +944,14 @@ hdy_swipe_tracker_dispose (GObject *object)
 {
   HdySwipeTracker *self = HDY_SWIPE_TRACKER (object);
 
-  if (self->swipeable)
+  if (self->swipeable) {
     gtk_grab_remove (GTK_WIDGET (self->swipeable));
+    g_object_set_data (G_OBJECT (self->swipeable), "captured-event-handler", NULL);
+  }
 
   if (self->touch_gesture)
     g_signal_handlers_disconnect_by_data (self->touch_gesture, self);
 
-  g_object_set_data (G_OBJECT (self->swipeable), "captured-event-handler", NULL);
-
   g_clear_pointer (&self->event_history, g_array_unref);
   g_clear_object (&self->touch_gesture);
   set_swipeable (self, NULL);
