diff --git a/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java b/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java
index df3e0fa..e08d7b4 100644
--- a/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java
+++ b/bindings/java/hyperic_jni/src/org/hyperic/jni/ArchNameTask.java
@@ -75,7 +75,7 @@ public class ArchNameTask extends Task {
         if (ArchName.is64()) {
             getProject().setProperty("jni.arch64", "true");
             if (ArchLoader.IS_LINUX) {
-                if (!osArch.equals("ia64")) {
+                if (!osArch.equals("ia64") && !osArch.equals("loongarch64")) {
                     getProject().setProperty("jni.gccm", "-m64");
                 }
             }
diff --git a/src/os/linux/linux_sigar.c b/src/os/linux/linux_sigar.c
index 2c512b8..1b42311 100644
--- a/src/os/linux/linux_sigar.c
+++ b/src/os/linux/linux_sigar.c
@@ -24,6 +24,7 @@
 #include <sys/times.h>
 #include <sys/utsname.h>
 #include <linux/param.h>
+#include <sys/sysmacros.h>
 
 #include "sigar.h"
 #include "sigar_private.h"
