--- src/setup.py.orig	2019-02-27 08:48:47.000000000 +1100
+++ src/setup.py	2019-05-26 14:15:57.444497428 +1000
@@ -28,7 +28,7 @@ setup(
     author_email="software@alcyone.com",
     url="http://www.alcyone.com/software/empy",
     license="%LICENSE",
-    py_modules=["em", "emlib"],
+    py_modules=["em"],
     platforms=["unix", "linux", "win32"],
     description=DESCRIPTION,
     long_description=LONG_DESCRIPTION,
