diff --git a/CMakeLists.txt b/CMakeLists.txt
index bba6085150..9360c84784 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,6 +170,11 @@ if(ENABLE_NLS)
       check_function_exists(dgettext LIBC_HAS_DGETTEXT)
       if(LIBC_HAS_DGETTEXT)
         set(GETTEXT_FOUND TRUE)
+        find_package(Intl)
+        if(NOT Intl_LIBRARIES STREQUAL "")
+          message(STATUS "GetText Intl : ${Intl_LIBRARIES}")
+          set(GETTEXT_LIBRARIES ${Intl_LIBRARIES})
+        endif()
       else()
         find_library(LIBINTL_LIBRARY NAMES intl libintl)
         if(LIBINTL_LIBRARY)
