From b3d8127cccc934322b8d78ab17a0b3644052fee8 Mon Sep 17 00:00:00 2001
From: fossdd <fossdd@pwned.life>
Date: Tue, 22 Jul 2025 14:53:34 +0200
Subject: [PATCH 03/11] shared: include signal.h in async.h

Signed-off-by: fossdd <fossdd@pwned.life>
---
 src/shared/async.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/shared/async.h b/src/shared/async.h
index 1ca7ef8df..f10f57345 100644
--- a/src/shared/async.h
+++ b/src/shared/async.h
@@ -2,6 +2,7 @@
 #pragma once
 
 #include <sys/types.h>
+#include <signal.h>
 
 #include "macro.h"
 #include "rm-rf.h"
-- 
2.50.1

