diff --git a/src/basic/generate-cap-list.sh b/src/basic/generate-cap-list.sh
index 80027db..d52a19e 100755
--- a/src/basic/generate-cap-list.sh
+++ b/src/basic/generate-cap-list.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 # SPDX-License-Identifier: LGPL-2.1-or-later
 set -eu
 set -o pipefail
diff --git a/src/basic/generate-errno-list.sh b/src/basic/generate-errno-list.sh
index f756b2e..76a6391 100755
--- a/src/basic/generate-errno-list.sh
+++ b/src/basic/generate-errno-list.sh
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 # SPDX-License-Identifier: LGPL-2.1-or-later
 set -eu
 set -o pipefail
