From 44359b80a85cc2fa5bcb7c9c82d68a94fca3bd33 Mon Sep 17 00:00:00 2001
From: Jingyun Hua <huajingyun@loongson.cn>
Date: Mon, 8 Apr 2024 15:57:15 +0800
Subject: [PATCH] Remove the useless endina.h in getcontext.S for loongarch64

Fix issue #740
---
 src/loongarch64/getcontext.S | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/loongarch64/getcontext.S b/src/loongarch64/getcontext.S
index 3344fc2e..c568832c 100644
--- a/src/loongarch64/getcontext.S
+++ b/src/loongarch64/getcontext.S
@@ -25,7 +25,6 @@ OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.  */
 
 #include "offsets.h"
-#include <endian.h>
 	.text
 
 #define SREG(X) st.d $r##X, $r4, (LINUX_UC_MCONTEXT_GREGS + 8 * X)
-- 
2.27.0

