--- nginx-1.6.1/src/http/ngx_http_header_filter_module.c
+++ nginx-1.6.1/src/http/ngx_http_header_filter_module.c
@@ -46,8 +46,8 @@
 };
 
 
-static char ngx_http_server_string[] = "Server: nginx" CRLF;
-static char ngx_http_server_full_string[] = "Server: " NGINX_VER CRLF;
+static char ngx_http_server_string[] = "";
+static char ngx_http_server_full_string[] = "";
 
 
 static ngx_str_t ngx_http_status_lines[] = {
@@ -278,8 +278,8 @@
     clcf = ngx_http_get_module_loc_conf(r, ngx_http_core_module);
 
     if (r->headers_out.server == NULL) {
-        len += clcf->server_tokens ? sizeof(ngx_http_server_full_string) - 1:
-                                     sizeof(ngx_http_server_string) - 1;
+        len += clcf->server_tokens ? sizeof(ngx_http_server_full_string) - 0:
+                                     sizeof(ngx_http_server_string) - 0;
     }
 
     if (r->headers_out.date == NULL) {
--- nginx-1.6.1/src/http/ngx_http_special_response.c
+++ nginx-1.6.1/src/http/ngx_http_special_response.c
@@ -19,14 +19,14 @@
 
 
 static u_char ngx_http_error_full_tail[] =
-"<hr><center>" NGINX_VER "</center>" CRLF
+"<hr><center>127.0.0.1</center>" CRLF
 "</body>" CRLF
 "</html>" CRLF
 ;
 
 
 static u_char ngx_http_error_tail[] =
-"<hr><center>nginx</center>" CRLF
+"<hr><center>localhost</center>" CRLF
 "</body>" CRLF
 "</html>" CRLF
 ;
--- nginx-1.9.12/src/http/v2/ngx_http_v2_filter_module.c
+++ nginx-1.9.12/src/http/v2/ngx_http_v2_filter_module.c.new
@@ -229,9 +229,9 @@
 
     clcf = ngx_http_get_module_loc_conf(r, ngx_http_core_module);
 
-    if (r->headers_out.server == NULL) {
+/*  if (r->headers_out.server == NULL) {
         len += 1 + (clcf->server_tokens ? nginx_ver_len : sizeof(nginx));
-    }
+    } */
 
     if (r->headers_out.date == NULL) {
         len += 1 + ngx_http_v2_literal_size("Wed, 31 Dec 1986 18:00:00 GMT");
@@ -434,7 +434,7 @@
         pos = ngx_sprintf(pos, "%03ui", r->headers_out.status);
     }
 
-    if (r->headers_out.server == NULL) {
+/*  if (r->headers_out.server == NULL) {
         ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,
                        "http2 output header: \"server: %s\"",
                        clcf->server_tokens ? NGINX_VER : "nginx");
@@ -453,7 +453,7 @@
         } else {
             pos = ngx_cpymem(pos, nginx, sizeof(nginx));
         }
-    }
+    } */
 
     if (r->headers_out.date == NULL) {
         ngx_log_debug1(NGX_LOG_DEBUG_HTTP, fc->log, 0,

