Update numpy float cast for numpy 2.0 compatibility.

--- NiaPy-2.3.1-origin/niapy/algorithms/modified/shade.py
+++ NiaPy-2.3.1/niapy/algorithms/modified/shade.py
@@ -514,7 +514,7 @@
         max_nfe = task.max_evals
         nfe = task.evals
 
-        next_pop_size = np.int_(np.around((((4.0 - self.population_size) / np.float_(max_nfe)) * nfe) + self.population_size))
+        next_pop_size = np.int_(np.around((((4.0 - self.population_size) / np.float64(max_nfe)) * nfe) + self.population_size))
 
         if next_pop_size < 4:
             next_pop_size = 4
