From 95786eefc6861bfce3cafa6cf862f715cd4dc31b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Krystian=20Chachu=C5=82a?= <krystian@krystianch.com>
Date: Tue, 16 Nov 2021 17:08:02 +0100
Subject: [PATCH] setup-disk: enable password verification in luksFormat

---
 setup-disk.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup-disk.in b/setup-disk.in
index dc5c1ac..c060ae2 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -1185,7 +1185,8 @@ setup_crypt() {
 	echo "Preparing partition for encryption." >&2
 	echo "You will be prompted for your password at boot." >&2
 	echo "If you forget your password, your data will be lost." >&2
-	cryptsetup luksFormat -q --type luks2 "$dev" >&2 || return 1
+	cryptsetup luksFormat --batch-mode --verify-passphrase \
+		--type luks2 "$dev" >&2 || return 1
 	echo "Enter password again to unlock disk for installation." >&2
 	cryptsetup open "$dev" "$dmname" >&2 || return 1
 	echo "/dev/mapper/$dmname"
-- 
2.34.0

