From 78d392b36076dccf1035f0b90e5dbdb2000433a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 9 Aug 2016 13:17:25 +0300
Subject: [PATCH] update-kernel: don't use local in non-function scope

---
 update-kernel.in | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/update-kernel.in b/update-kernel.in
index ab7c528..fc0899d 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -205,7 +205,6 @@ extra_pkgs() {
 _apk add --initdb --update-cache
 
 if [ "$BUILDDIR" ]; then
-    local _install
     case "$ARCH" in
     arm*|aarch64*)	_install="zinstall dtbs_install" ;;
     *)			_install="install" ;;
@@ -264,8 +263,7 @@ fi
 mv $STAGING/* "$DESTDIR"
 
 if [ -d "$DTBDIR" ]; then
-    local _opwd=$PWD
-    local _dtb
+    _opwd=$PWD
     case "$FLAVOR" in
     rpi*) _dtb="$DESTDIR" ;;
     *) _dtb="$DESTDIR/dtbs" ;;
-- 
2.9.3

