From 631c361640d22e40d19b43d3b1e63b4da84ad18a Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 8 Nov 2011 09:22:53 +0000
Subject: [PATCH] setup-disk: fix booting from cciss devices

---
 setup-disk.in |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index a2b610f..6f49e6a 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -206,6 +206,14 @@ install_mounted_root() {
 			raid[456]) raidmod="$raidmod,raid456";;
 		esac
 	done
+
+	# check if we need hardware raid drivers
+	case $rootdev in
+		/dev/cciss/*)
+			initfs_features="$(echo $initfs_features | sed 's/raid//') raid"
+			;;
+	esac
+
 	if [ -n "$VERBOSE" ]; then
 		echo "Root device:     $rootdev"
 		echo "Root filesystem: $root_fs"
-- 
1.7.7.2

