From 8201bedeef8250452bdcdf5db0699c12b7ff7066 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 1 Mar 2013 10:54:12 +0000
Subject: [PATCH] setup-interfaces: fix vlan-raw-device

ref #1624
---
 setup-interfaces.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-interfaces.in b/setup-interfaces.in
index 263beec..54d861a 100755
--- a/setup-interfaces.in
+++ b/setup-interfaces.in
@@ -426,7 +426,7 @@ prompt_for_interfaces() {
 			echo -e "\tbond-slaves $bond_slaves" >> interfaces
 		fi
 		if [ -n "$raw_device" ]; then
-			echo -e "\traw-device $raw_device" >> interfaces
+			echo -e "\tvlan-raw-device $raw_device" >> interfaces
 		fi
 		case "$iface" in
 			*.[0-9]*|vlan[0-9]*) PKGS="$PKGS vlan";;
-- 
1.8.1.4

