From c15da3bf6ba6d7b05cac6c6182efdd4149554b82 Mon Sep 17 00:00:00 2001
From: Jakub Skrzypnik <j.skrzypnik@openmailbox.orxg>
Date: Fri, 6 Jan 2017 08:47:02 +0100
Subject: [PATCH] setup-disk: add xfs support as boot_fs

---
 setup-disk.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 023974e..c4103e4 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -141,7 +141,7 @@ find_mount_dev() {
 }
 
 supported_boot_fs() {
-	local supported="ext2 ext3 ext4 btrfs"
+	local supported="ext2 ext3 ext4 btrfs xfs"
 	local fs=
 	for fs in $supported; do
 		[ "$fs" = "$1" ] && return 0
@@ -941,7 +941,7 @@ usage() {
 		If BOOTFS, ROOTFS, VARFS are specified, then format a partition with specified
 		filesystem. If not specified, the default filesystem is ext4.
 		Supported filesystems for
-		  boot: ext2, ext3, ext4, btrfs
+		  boot: ext2, ext3, ext4, btrfs, xfs
 		  root: ext2, ext3, ext4, btrfs, xfs
 		   var: ext2, ext3, ext4, btrfs, xfs
 	__EOF__
-- 
2.13.0

