From fb134d265780f0688ef4ff4e3d747d0b7401c65d Mon Sep 17 00:00:00 2001
From: black <code@ares-macrotechnology.com>
Date: Fri, 18 Aug 2023 20:46:58 +0200
Subject: [PATCH 1/2] Handle release versions from /etc/apk/repositories
 correctly (#10554)

---
 setup-apkrepos.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/setup-apkrepos.in b/setup-apkrepos.in
index 2c5a67c..9eed9b3 100644
--- a/setup-apkrepos.in
+++ b/setup-apkrepos.in
@@ -113,6 +113,9 @@ get_alpine_release() {
 		[0-9]*.[0-9]*.[0-9]*)
 			# release in x.y.z format, cut last digit
 			release=v${version%.[0-9]*};;
+		v[0-9]*.[0-9]*)
+            # release in vx.y format, keep as is
+			release="${version}";;
 		*)	# fallback to edge
 			release="edge";;
 	esac
-- 
2.42.0

