From b709cfbbbf86ecfa74a674349ed0f9b5c04cbd46 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 13 Jun 2016 12:23:54 +0000
Subject: [PATCH] setup-alpine: fix openrc warning

---
 setup-alpine.in | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/setup-alpine.in b/setup-alpine.in
index 050016c..c120c99 100644
--- a/setup-alpine.in
+++ b/setup-alpine.in
@@ -48,7 +48,7 @@ while getopts "af:c:hq" opt ; do
 done
 shift `expr $OPTIND - 1`
 
-rc_sys=$(rc --sys)
+rc_sys=$(openrc --sys)
 # mount xenfs so we can detect xen dom0
 if [ "$rc_sys" = "XENU" ] && ! grep -q '^xenfs' /proc/mounts; then
 	modprobe xenfs
@@ -156,8 +156,8 @@ done
 /etc/init.d/hostname --quiet restart
 
 # start up the services
-rc boot
-rc default
+openrc boot
+openrc default
 
 # update /etc/hosts - after we have got dhcp address
 # Get default fully qualified domain name from *first* domain
-- 
2.8.4

