From 921a66c96c97634450c456b5ead039a20fa408e8 Mon Sep 17 00:00:00 2001
From: Richard Mortier <mort@cantab.net>
Date: Fri, 26 Apr 2019 10:39:39 +0100
Subject: [PATCH] lbu: fix invocation of `openssl list-cipher-commands` for
 v1.1.1b

---
 lbu.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lbu.in b/lbu.in
index 61be708..f439cf2 100644
--- a/lbu.in
+++ b/lbu.in
@@ -126,7 +126,7 @@ check_openssl() {
 	[ -z "$ENCRYPTION" ] && return 0
 	OPENSSL=$(which openssl 2>/dev/null) || die "openssl was not found"
 
-	$OPENSSL list-cipher-commands | grep "^$ENCRYPTION$" > /dev/null \
+	$OPENSSL enc -ciphers | grep "^$ENCRYPTION$" > /dev/null \
 		|| die "Cipher $ENCRYPTION is not supported"
 }
 
-- 
2.21.0

