From cf023ac1ffecfdcc1997ee7b8e57cacdde62870e Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 13 Jun 2016 13:34:02 +0000
Subject: [PATCH] setup-disk: only restart syslog if its already running

ref #5719
---
 setup-disk.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-disk.in b/setup-disk.in
index 0df6d96..3ab35cd 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -661,7 +661,7 @@ setup_var() {
 	mv /.var/* /var/
 	rmdir /.var
 
-	/etc/init.d/syslog --quiet restart
+	service syslog --quiet condrestart
 	setup_mdadm_conf
 }
 
-- 
2.8.4

