From 30687f8a92475a8b492d55ad51986e5cbf555bd0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sat, 19 Mar 2016 21:16:38 +0100
Subject: [PATCH] setup-alpine: cron service was renamed to crond

---
 setup-alpine.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-alpine.in b/setup-alpine.in
index cc10483..1a34704 100644
--- a/setup-alpine.in
+++ b/setup-alpine.in
@@ -147,7 +147,7 @@ fi
 rc-update --quiet add networking boot
 rc-update --quiet add urandom boot
 rc-update --quiet add acpid
-rc-update --quiet add cron
+rc-update --quiet add crond
 
 # enable new hostname
 /etc/init.d/hostname --quiet restart
-- 
2.7.4

