From f133bfa2ddf73b07e4ecf92d724ed7c416f4ff81 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 21 Dec 2016 20:44:08 +0000
Subject: [PATCH] setup-disk: dont use local in global scope

---
 setup-disk.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-disk.in b/setup-disk.in
index 588d37f..ab785fb 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -1031,7 +1031,7 @@ if [ -n "$diskdevs" ] && [ -z "$DISK_MODE" ]; then
 	while true; do
 		echo "The following $disk_is_or_disks_are selected${USE_LVM:+ (with LVM)}:"
 		show_disk_info $diskdevs
-		local _lvm=${USE_LVM:-", 'lvm'"}
+		_lvm=${USE_LVM:-", 'lvm'"}
 		echon "How would you like to use $it_them? ('sys', 'data'${_lvm#_lvm} or '?' for help) [?] "
 		default_read answer '?'
 		case "$answer" in
-- 
2.11.0

