From ad9996810186cbed04ec0c185bd3c998622a0a4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Milan=20P=2E=20Stani=C4=87?= <mps@arvanta.net>
Date: Thu, 16 Jan 2020 11:09:28 +0100
Subject: [PATCH] lbu: fix wrong cipher match in check_openssl for 'lbu commit
 -e'

fixes #10451
---
 lbu.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lbu.in b/lbu.in
index f439cf2..fb4fbe5 100644
--- a/lbu.in
+++ b/lbu.in
@@ -126,7 +126,7 @@ check_openssl() {
 	[ -z "$ENCRYPTION" ] && return 0
 	OPENSSL=$(which openssl 2>/dev/null) || die "openssl was not found"
 
-	$OPENSSL enc -ciphers | grep "^$ENCRYPTION$" > /dev/null \
+	$OPENSSL list -1 -cipher-commands | grep "^$ENCRYPTION$" > /dev/null \
 		|| die "Cipher $ENCRYPTION is not supported"
 }
 
-- 
2.25.0

