From ab68d58344a5e5421a52dccace155b5a42be4f14 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 21 Dec 2012 15:29:06 +0100
Subject: [PATCH] setup-interfaces: fix ip: can't find device 'eth1.3' message

ref #1404
---
 setup-interfaces.in | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/setup-interfaces.in b/setup-interfaces.in
index c4ffa47..263beec 100755
--- a/setup-interfaces.in
+++ b/setup-interfaces.in
@@ -73,7 +73,9 @@ get_default_mask() {
 }
 
 get_default_gateway() {
-	ip route show dev $1 | awk '/^default/ {print $3}'
+	if iface_exists $1; then
+		ip route show dev $1 | awk '/^default/ {print $3}'
+	fi
 }
 
 ipaddr_help() {
-- 
1.8.0.2

