From dc06856daff9ba68eea716956c2e2383d4097747 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 19 May 2014 15:50:21 +0200
Subject: [PATCH] setup-disk: fix partitioning of HP raid array (cciss)

ref #2893
---
 setup-disk.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-disk.in b/setup-disk.in
index e5cae87..e7a063c 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -950,7 +950,7 @@ if [ $# -gt 0 ]; then
 			echo "$i is not a suitable for partitioning"
 			exit 1
 		fi
-		diskdevs="$diskdevs /dev/$j"
+		diskdevs="$diskdevs /dev/${j//!//}"
 	done
 else
 	ask_disk "Which disk(s) would you like to use? (or '?' for help or 'none')" \
-- 
1.9.3

