From b7ac1f3f8b4ddb7cc4a585bc8b7ebdaa2dd8ae16 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 20 Aug 2009 07:31:21 +0000
Subject: [PATCH] setup-acf: fix password generation

---
 setup-acf.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/setup-acf.in b/setup-acf.in
index 8c386ee..92d769c 100644
--- a/setup-acf.in
+++ b/setup-acf.in
@@ -37,7 +37,7 @@ if [ "$create_passwd" != "no" ]; then
 	fi
 	# this will show password on process list but we assume user is alone 
 	# on the box at this stage
-	_md5passwd=$(echo -n "$_password" | md5sum)
+	_md5passwd=$(echo -n "$_password" | md5sum | cut -d' ' -f1)
 	echo "root:$_md5passwd:Admin account:ADMIN" >/etc/acf/passwd
 fi
 
-- 
1.6.4

