From bf182e3346ef7aa42b7db5c9d7fbc431e78269e3 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 1 Jul 2015 14:17:07 +0000
Subject: [PATCH] setup-disk: add raid to initfs if root is on lvm

ref #4421
---
 setup-disk.in | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 9444b7e..e2798f9 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -234,6 +234,14 @@ install_mounted_root() {
 	# check if our root is on raid so we can feed mkinitfs and
 	# update-exlinux.conf with the proper kernel module params
 	for dev in $rootdev $pvs; do
+
+		# check if we need hardware raid drivers
+		case $dev in
+		/dev/cciss/*)
+			initfs_features="${initfs_features% raid} raid"
+			;;
+		esac
+
 		[ -e "/sys/block/${dev#/dev/}/md" ] || continue
 
 		local md=${dev#/dev/}
@@ -245,12 +253,6 @@ install_mounted_root() {
 		esac
 	done
 
-	# check if we need hardware raid drivers
-	case $rootdev in
-		/dev/cciss/*)
-			initfs_features="${initfs_features% raid} raid"
-			;;
-	esac
 
 	if [ -n "$VERBOSE" ]; then
 		echo "Root device:     $rootdev"
-- 
2.4.5

