From 3ed9c1d7846d62d6a83b5f9d9f93cbfe2c515ef7 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 12 Jan 2012 11:55:21 +0000
Subject: [PATCH] setup-acf: use acfpasswd utility to set password

---
 setup-acf.in |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/setup-acf.in b/setup-acf.in
index 487db00..9e1489f 100644
--- a/setup-acf.in
+++ b/setup-acf.in
@@ -31,16 +31,13 @@ while [ $# -gt 0 ]; do
 done
 
 if [ "$create_passwd" != "no" ]; then
-	askpassword "root ACF"
 	mkdir -p /etc/acf
 	if [ -f /etc/acf/passwd ]; then
 		mv /etc/acf/passwd /etc/acf/passwd.backup
 	fi
-	# this will show password on process list but we assume user is alone 
-	# on the box at this stage
-	_md5passwd=$(echo -n "$_password" | md5sum | cut -d' ' -f1)
-	echo "root:$_md5passwd:Admin account:ADMIN" >/etc/acf/passwd
+	echo "root:x:Admin account:ADMIN" >/etc/acf/passwd
 	chmod 600 /etc/acf/passwd
+	acfpasswd -s root
 fi
 
 # install packages 
-- 
1.7.8.3

