From 6df463ada5052a39b32a443b276b489c431a115f Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 5 Nov 2013 14:15:44 +0100
Subject: [PATCH] setup-bootable: use UUID on new USBs

---
 setup-bootable.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup-bootable.in b/setup-bootable.in
index 18e41bf..2df03e6 100644
--- a/setup-bootable.in
+++ b/setup-bootable.in
@@ -256,7 +256,7 @@ for i in $tomove; do
 	mv "$destdir"/.new/$i "$destdir"/ || die "Failed to move $destdir/.new/ to $destdir"
 done
 
-if [ -n "$replace_alpine_dev" ] && [ -n "$UUID" ]; then
+if [ -n "$replace_alpine_dev" -o -z "$upgrade" ] && [ -n "$UUID" ]; then
 	sed -i -e "s/alpine_dev=[^ \t:]\+/alpine_dev=UUID=$UUID/" \
 		"$destdir"/syslinux.cfg
 fi
-- 
1.8.4.2

