From 16f383629c1e41b44a9c75a02f190ef6ec8d3ad7 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 7 Jan 2021 18:14:07 +0100
Subject: [PATCH] setup-alpine: fix qemu detection

---
 setup-alpine.in | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/setup-alpine.in b/setup-alpine.in
index 80e10ff..392290b 100644
--- a/setup-alpine.in
+++ b/setup-alpine.in
@@ -17,7 +17,8 @@ get_fqdn() {
 }
 
 is_qemu() {
-	grep -q "QEMU" /proc/cpuinfo
+	grep -q "QEMU" /proc/cpuinfo \
+		|| strings /sys/firmware/dmi/tables/DMI 2>/dev/null | grep -q QEMU
 }
 
 usage() {
-- 
2.30.0

