From b1febd985fe1b5cd183b19ee5e491fceda0e0a9a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Wed, 28 Feb 2018 17:04:27 +0100
Subject: [PATCH] setup-keymap: keymaps OpenRC service has been renamed to
 loadkmap

---
 setup-keymap.in | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/setup-keymap.in b/setup-keymap.in
index ebcd29c..21ba2ac 100644
--- a/setup-keymap.in
+++ b/setup-keymap.in
@@ -5,8 +5,8 @@ PREFIX=
 
 MAPDIR="$ROOT/usr/share/bkeymaps"
 
-if [ -f "$ROOT/etc/conf.d/keymaps" ]; then
-	. "$ROOT/etc/conf.d/keymaps"
+if [ -f "$ROOT/etc/conf.d/loadkmap" ]; then
+	. "$ROOT/etc/conf.d/loadkmap"
 	variant=$(basename "KEYMAP")
 	variant="${variant%%.*}"
 fi
@@ -52,15 +52,15 @@ select_layout() {
 
 setup_mapfile() {
 	local name=$(basename "$1")
-	local conf="$ROOT/etc/conf.d/keymaps"
+	local conf="$ROOT/etc/conf.d/loadkmap"
 	mkdir -p "$ROOT/etc/keymap"
 	mkdir -p "$ROOT/etc/conf.d/"
 	if cp "$1" "$ROOT/etc/keymap/$name" ; then
 		[ -f "$conf" ] && sed -i '/^KEYMAP=/d' "$conf"
 		echo "KEYMAP=/etc/keymap/$name" >> "$conf"
 		# we actually load the keymap now
-		rc-service keymaps restart
-		rc-update -q add keymaps boot
+		rc-service loadkmap restart
+		rc-update -q add loadkmap boot
 		goodbye 0
 	fi
 }
-- 
2.16.2

