From d7441947485aaa311d9d892d1341690ba9d99afb Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Tue, 24 Aug 2010 14:45:20 +0000
Subject: [PATCH] setup-disk: add cdrom, usb and floppy to fstab

---
 setup-disk.in |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 96e8e99..c6f56a1 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -202,6 +202,11 @@ EOF
 		mv "$mnt"/etc/fstab "$mnt"/etc/fstab.old
 	fi
 	enumerate_fstab "$mnt" >> "$mnt"/etc/fstab
+	cat >>"$mnt"/etc/fstab <<EOF
+/dev/cdrom	/media/cdrom	iso9660	noauto,ro 0 0
+/dev/fd0	/media/floppy	vfat	noauto	0 0
+/dev/usbdisk	/media/usb	vfat	noauto	0 0
+EOF
 
 	# install extlinux
 	apk add -q syslinux
-- 
1.7.2.2

