From 80370671291376ee040e5e25c0ea38a9c0780f2a Mon Sep 17 00:00:00 2001
From: black <code@ares-macrotechnology.com>
Date: Fri, 25 Aug 2023 20:02:34 +0200
Subject: [PATCH 2/2] Add regression test for #10554

---
 tests/setup_apkrepos_test | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/tests/setup_apkrepos_test b/tests/setup_apkrepos_test
index 2b87d30..719a5f3 100755
--- a/tests/setup_apkrepos_test
+++ b/tests/setup_apkrepos_test
@@ -11,6 +11,7 @@ init_tests \
 	setup_apkrepos_network_failure \
 	setup_apkrepos_interactive \
 	setup_apkrepos_alpine_stable \
+	setup_apkrepos_alpine_stable_preconfigured \
 	setup_apkrepos_alpine_edge
 
 setup_apkrepos_usage_body() {
@@ -97,6 +98,18 @@ setup_apkrepos_alpine_stable_body() {
 	atf_check -o match:"v3\.17" cat etc/apk/repositories
 }
 
+setup_apkrepos_alpine_stable_preconfigured_body() {
+	init_env
+	export WGETCONTENT="$(seq 0 9 | awk '{print "https://a" $0 ".example.com"}')"
+	mkdir -p etc/apk
+	echo "3.17.3" > etc/alpine-release
+	echo "http://dl-cdn.alpinelinux.org/alpine/v3.17/main" > etc/apk/repositories
+	atf_check -s exit:0 \
+		-o match:"Added mirror a0.example.com" \
+		setup-apkrepos -1
+	atf_check -o match:"a0\.example\.com/v3\.17/main" cat etc/apk/repositories
+}
+
 setup_apkrepos_alpine_edge_body() {
 	init_env
 	export WGETCONTENT="$(seq 0 9 | awk '{print "https://a" $0 ".example.com"}')"
-- 
2.42.0

