From 2e3bdc6d581b3efc5446d3b765272f7125bc0cba Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Thu, 6 Aug 2009 12:08:06 +0000
Subject: [PATCH 2/2] setup-disk: copy apk config to new root

---
 setup-disk.in |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index 6a90d3a..e1a2c6d 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -42,8 +42,12 @@ install_mounted_root() {
 
 	echon "Installing system on $rootdev: "
 	lbu package - | tar -C "$mnt" -zx
+	# apk reads config from target root so we need to copy the config
+	mkdir -p "$mnt"/etc/apk/keys/
+	cp /etc/apk/keys/* "$mnt"/etc/apk/keys/
+
 	apk add -q --progress --root "$mnt" $(cat "$mnt"/var/lib/apk/world) \
-		linux-grsec acct mkinitfs
+		acct linux-grsec alpine-base >/dev/null || return 1
 	echo ""
 	# make things bootable
 	kernel=$(ls "$mnt"/lib/modules)
@@ -61,7 +65,6 @@ install_mounted_root() {
 			rootdisk="$rootdisk /dev/${i}"
 		done
 	fi
-	chroot "$mnt" /sbin/mkinitfs -F "$features" $kernel
 
 	# create an extlinux.conf
 	sed '/append initrd/d' /media/*/syslinux.cfg > "$mnt"/boot/extlinux.conf
-- 
1.6.4

