From 37f3d5249c16a9166ff1c63ba031c6d4c9b86183 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Sat, 9 Jul 2016 11:44:16 +0300
Subject: [PATCH 2/6] update-kernel: fix "find: unpaired '('"

fixes #5883
---
 update-kernel.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update-kernel.in b/update-kernel.in
index 34a193d..5b8f6f3 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -240,7 +240,7 @@ if [ "$ARM" ]; then
     DTBDIR=$ROOT/usr/lib/linux-$KVER
     [ "$BUILDDIR" ] && DTBDIR=$BUILDDIR/source/arch/arm/boot/dts
     cp -a "$DTBDIR"/* $DTB_STAGING
-    find "$DTB_STAGING" -type f \! \( -name "*.dtb" -o -name "*.dtbo"\)  -delete
+    find "$DTB_STAGING" -type f \! \( -name "*.dtb" -o -name "*.dtbo" \)  -delete
 fi
 
 
-- 
2.9.3

