From a27d85e3be75350077361891ae2b8f2950da9bc2 Mon Sep 17 00:00:00 2001
From: "Alex Xu (Hello71)" <alex_y_xu@yahoo.ca>
Date: Thu, 7 Jan 2021 11:14:25 -0500
Subject: [PATCH 3/4] update-kernel: don't force /tmp

closes !11.
---
 update-kernel.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update-kernel.in b/update-kernel.in
index b1a8986..29110e9 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -218,7 +218,7 @@ fi
 
 [ "$ARCH" ] || ARCH=$(apk --print-arch)
 
-tmpdir=$(mktemp -d /tmp/$SCRIPT.XXXXXX)
+tmpdir=$(mktemp -dt $SCRIPT.XXXXXX)
 ROOT=$tmpdir/root
 BOOT=$ROOT/boot
 
-- 
2.33.1

