From b511518795b03520248d9a64ff488716e3f01c38 Mon Sep 17 00:00:00 2001
From: Kaarle Ritvanen <kunkku@alpinelinux.org>
Date: Wed, 17 Nov 2021 15:01:40 +0200
Subject: [PATCH 4/4] update-kernel: TMPDIR in help text

ref !18
---
 update-kernel.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/update-kernel.in b/update-kernel.in
index 29110e9..dda4589 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -6,7 +6,7 @@
 # with setup-bootable
 #
 # Copyright (c) 2014 Timo Teräs
-# Copyright (c) 2014-2015 Kaarle Ritvanen
+# Copyright (c) 2014-2021 Kaarle Ritvanen
 
 PREFIX=
 . $PREFIX/lib/libalpine.sh
@@ -63,6 +63,9 @@ usage() {
 		         -d|--keys-dir             Override directory of trusted keys for apk
 		         --repositories-file <f>   apk repositories file
 
+		On low-memory systems, you may want to point the TMPDIR environment variable to
+		a storage-backed directory.
+
 	__EOF__
 	exit $1
 }
-- 
2.33.1

