From db5429023efd449c3454286ae2cfdcfa7e37104d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?=
 <pabloyoyoista@postmarketos.org>
Date: Fri, 4 Jul 2025 15:03:33 +0200
Subject: [PATCH] setup-disk: bootstrap new installs /usr merged

---
 setup-disk.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/setup-disk.in b/setup-disk.in
index ca0a865..9c12e80 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -662,6 +662,8 @@ install_mounted_root() {
 	done
 
 	init_chroot_mounts "$mnt"
+	mkdir -p "$mnt"/usr/lib "$mnt"/usr/bin "$mnt"/usr/sbin
+	ln -s usr/bin usr/sbin usr/lib "$mnt"/
 	apk add --root "$mnt" $apkflags --overlay-from-stdin \
 		$repoflags $pkgs <$ovlfiles
 	local ret=$?
-- 
GitLab

