From 762c6558e48409159fba7b12ce8fb0baf4e6e2f5 Mon Sep 17 00:00:00 2001
From: Eivind Uggedal <eivind@uggedal.com>
Date: Mon, 12 Aug 2013 22:26:16 +0200
Subject: setup-timezone: fix bug in getopts parsing


diff --git a/setup-timezone.in b/setup-timezone.in
index 41d4f16..5034223 100755
--- a/setup-timezone.in
+++ b/setup-timezone.in
@@ -47,7 +47,7 @@ while getopts "hp:z:" opt; do
         case $opt in
                 h) usage;;
                 p) POSIXTZ="$OPTARG";;
-		t) ZONEINFOFOLDER="$OPTARG";;
+		z) ZONEINFOFOLDER="$OPTARG";;
         esac
 done
 
-- 
cgit v0.10.1

