From 0dd1f10889920ec8a63f6072601786c0acc9a249 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 20 May 2016 09:55:42 +0000
Subject: [PATCH] setup-disk: don't install syslinux for data-only

---
 setup-disk.in | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/setup-disk.in b/setup-disk.in
index ad0bdb7..0df6d96 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -458,7 +458,7 @@ init_progs() {
 		btrfs) fstools="$fstools btrfs-progs";;
 		esac
 	done
-	apk add --quiet sfdisk $lvmpkg $raidpkg syslinux $fstools $@
+	apk add --quiet sfdisk $lvmpkg $raidpkg $fstools $@
 }
 
 show_disk_info() {
@@ -505,7 +505,9 @@ setup_partitions() {
 	echo "w" | fdisk $diskdev >/dev/null
 
 	# fix the MBR while here so extlinux can boot
-	cat "$MBR" > $diskdev
+	if [ -f "$MBR" ]; then
+		cat "$MBR" > $diskdev
+	fi
 
 	local start=1M
 	local line=
@@ -771,7 +773,7 @@ native_disk_install_lvm() {
 	local lvm_size=
 	local root_dev=/dev/$vgname/lv_root
 
-	init_progs || return 1
+	init_progs syslinux || return 1
 	confirm_erase $@ || return 1
 
 	if [ -n "$USE_RAID" ]; then
@@ -802,7 +804,7 @@ native_disk_install() {
 	local root_size=
 	local root_dev= boot_dev= swap_dev=
 
-	init_progs || return 1
+	init_progs syslinux || return 1
 	confirm_erase $@ || return 1
 
 	if [ -n "$USE_RAID" ]; then
-- 
2.8.2

