diff --git a/setup-disk.in b/setup-disk.in
index 4ed0cbd..5eb8638 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -306,16 +306,14 @@ setup_grub() {
 		esac
 	fi
 
-	# setup GRUB config
-
-	# all_video is needed to remove the video error on boot
-	cat > "$mnt"/boot/grub/grub.cfg <<- EOF
-	set timeout=2
-	insmod all_video
-	menuentry "Alpine Linux" {
-	    linux   /boot/vmlinuz-$KERNEL_FLAVOR modules=$modules root=$root $kernel_opts
-	    initrd  /boot/initramfs-$KERNEL_FLAVOR
-	}
+	# setup GRUB config. trigger will generate final grub.cfg
+	install -d "$mnt"/etc/default/
+	cat > "$mnt"/etc/default/grub <<- EOF
+	GRUB_DISTRIBUTOR="Alpine"
+	GRUB_TIMEOUT=2
+	GRUB_DISABLE_SUBMENU=y
+	GRUB_DISABLE_RECOVERY=true
+	GRUB_CMDLINE_LINUX_DEFAULT="modules=$modules $kernel_opts"
 	EOF
 }
 
