From 1512b22ae7cbd99f5f77a027fbd2b6acb3ec1615 Mon Sep 17 00:00:00 2001
From: Jingyun Hua <huajingyun@loongson.cn>
Date: Mon, 18 Mar 2024 10:48:17 +0800
Subject: [PATCH] setup-disk: add loongarch64 support

---
 setup-disk.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup-disk.in b/setup-disk.in
index 56f5152..7f01bfb 100755
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -322,6 +322,7 @@ setup_grub() {
 			arm*)		target=arm-efi ; fwa=arm ;;
 			aarch64)	target=arm64-efi ; fwa=aa64 ;;
 			riscv64)	target=riscv64-efi ; fwa=riscv64 ;;
+			loongarch64)	target=loongarch64-efi ; fwa=loongarch64 ;;
 		esac
 
 		# currently disabling nvram so grub doesnt call efibootmgr
-- 
2.27.0

