commit e149170d4736017c76b09625fa9a4b6a90b5e3a3
Author: Natanael Copa <ncopa@alpinelinux.org>
Date:   Wed May 26 13:59:25 2010 +0000

    setup-disk: fix typo in the option handling

diff --git a/setup-disk.in b/setup-disk.in
index 7a485b8..aa4a92f 100644
--- a/setup-disk.in
+++ b/setup-disk.in
@@ -401,7 +401,7 @@ while getopts "hk:o:r" opt; do
 		*) usage;;
 	esac
 done
-shift $(( OPTIND - 1))
+shift $(( $OPTIND - 1))
 
 if [ -d "$1" ]; then
 	# install to given mounted root
