submitted as: https://github.com/facebook/rocksdb/pull/9800

--- a/tools/db_bench_tool.cc
+++ b/tools/db_bench_tool.cc
@@ -6251,9 +6251,9 @@
       }
 
       // Pick a Iterator to use
-      size_t db_idx_to_use =
+      uint64_t db_idx_to_use =
           (db_.db == nullptr)
-              ? (size_t{thread->rand.Next()} % multi_dbs_.size())
+              ? (uint64_t{thread->rand.Next()} % multi_dbs_.size())
               : 0;
       std::unique_ptr<Iterator> single_iter;
       Iterator* iter_to_use;
