diff -up recode-3.6/src/names.c.printf recode-3.6/src/names.c
--- recode-3.6/src/names.c.printf	2000-12-06 20:41:29.000000000 +0100
+++ recode-3.6/src/names.c	2013-12-18 12:48:42.578052395 +0100
@@ -892,7 +892,7 @@ list_concise_charset (RECODE_OUTER outer
 	    if (ucs2 >= 0)
 	      printf (format, code);
 	    else if (mnemonic || counter2 != 112)
-	      printf (blanks);
+	      printf ("%s", blanks);
 
 	    if (mnemonic)
 	      printf (counter2 == 112 ? " %s\n" : " %-3s", mnemonic);
