diff --git a/configure b/configure
index e49c08f..c6976fd 100755
--- a/configure
+++ b/configure
@@ -2882,7 +2882,7 @@ BuildPortDotH()
 	echo '#define HAS_FCHMOD 1'
 	Note "... configure use of fchmod"
 	CheckFuncDecl fchmod 'extern int fchmod(int, mode_t);' \
-	    unistd.h libc.h $OSFCNH sys/stat.h
+	    unistd.h $OSFCNH sys/stat.h libc.h
     }
     CheckFuncDecl mknod 'extern int mknod(const char*, mode_t, dev_t);' \
 	unistd.h sys/stat.h
