From http://bugs.hylafax.org/show_bug.cgi?id=971

--- a/libhylafax/FaxRecvInfo.c++
+++ b/libhylafax/FaxRecvInfo.c++
@@ -112,7 +112,7 @@
     if (cp == NULL || cp[1] != ',' || cp[2] != '"')
 	return (false);
     u_int i = 0;
-    while (cp+2 != '\0') {
+    while (*(cp+2) != '\0') {
 	callid[i] = cp+3;		// +1 for "/+1 for ,/+1 for "
 	if (*cp == '\"') break;
 	callid[i].resize(callid[i].next(0,'"'));
