diff -ru hylafax-6.0.6.orig/faxmail/faxmail.c++ hylafax-6.0.6/faxmail/faxmail.c++
--- hylafax-6.0.6.orig/faxmail/faxmail.c++	2012-06-06 03:58:38.000000000 +0300
+++ hylafax-6.0.6/faxmail/faxmail.c++	2018-01-09 09:11:28.918140061 +0200
@@ -475,8 +475,8 @@
     if (mime.parse(msg, emsg)) {
 	if (verbose)
 	    mime.trace(stderr);
-	// XXX anything but us-ascii is treated as ISO-8859-1
-	setISO8859(mime.getCharset() != CS_USASCII);
+	// XXX treat everything as ISO-8859-1
+	setISO8859(true);
 
 	/*
 	 * Check first for any external script/command to

