diff --git a/src/cad-pulse.c b/src/cad-pulse.c
index bfed98a..9db2f91 100644
--- a/src/cad-pulse.c
+++ b/src/cad-pulse.c
@@ -367,7 +367,7 @@ static void init_card_info(pa_context *ctx, const pa_card_info *info, int eol, v
     }

     prop = pa_proplist_gets(info->proplist, PA_PROP_DEVICE_BUS_PATH);
-    if (prop && !g_str_has_prefix(prop, CARD_BUS_PATH_PREFIX))
+    if (prop && (!g_str_has_prefix(prop, CARD_BUS_PATH_PREFIX) || g_str_has_suffix(prop, "_hdmi")))
         return;
     prop = pa_proplist_gets(info->proplist, PA_PROP_DEVICE_FORM_FACTOR);
     if (prop && strcmp(prop, CARD_FORM_FACTOR) != 0)
