diff --git a/etc/webapp_config.json b/etc/webapp_config.json
index 505a02f..93d576e 100644
--- a/etc/webapp_config.json
+++ b/etc/webapp_config.json
@@ -45,14 +45,14 @@
     "help": "Settings for the HOMER Webapp Server. If you have gzip_static = false, please be sure that your dist directory has uncompressed .js files",
     "host": "0.0.0.0",
     "port" : 9080,
-    "root" : "/usr/local/homer/dist",
+    "root" : "/usr/share/webapps/homer/dist",
     "gzip" : true,
     "gzip_static" : true,
     "debug" : false
   },
   "system_settings": {
     "help": "Settings for HOMER logs",
-    "logpath": "/usr/local/homer/log",
+    "logpath": "/var/log/homer",
     "logname" : "homer-app.log",
     "_loglevels": "can be: fatal, error, warn, info, debug, trace",
     "loglevel": "error",
@@ -86,4 +86,4 @@
     "bin": "/usr/bin/tshark",
     "protocols":  ["1_call","1_registration", "1_default"]
   }
-}
\ No newline at end of file
+}
diff --git a/main.go b/main.go
index ede7b44..5e7a3d5 100644
--- a/main.go
+++ b/main.go
@@ -169,7 +169,7 @@ func initFlags() {
 	appFlags.DatabaseHomerConfig = flag.String("database-homer-config", "homer_config", "database-homer-config")
 	appFlags.DatabaseHomerData = flag.String("database-homer-data", "homer_data", "database-homer-data")
 
-	appFlags.PathWebAppConfig = flag.String("webapp-config-path", "/usr/local/homer/etc", "the path to the webapp config file")
+	appFlags.PathWebAppConfig = flag.String("webapp-config-path", "/etc/homer", "the path to the webapp config file")
 	appFlags.LogName = flag.String("webapp-log-name", "", "the name prefix of the log file.")
 	appFlags.LogPathWebApp = flag.String("webapp-log-path", "", "the path for the log file.")
 
@@ -346,7 +346,7 @@ func configureAsHTTPServer() {
 	/* static */
 	rootPath := viper.GetString("http_settings.root")
 	if rootPath == "" {
-		rootPath = "/usr/local/homer/dist"
+		rootPath = "/usr/share/webapps/homer/dist"
 	}
 	/* static */
 	e.Use(middleware.Static(rootPath))
