diff -rupN a/src/settings/rfl/settings/definition.py b/src/settings/rfl/settings/definition.py
--- a/src/settings/rfl/settings/definition.py	2025-10-10 12:25:35.000000000 +0000
+++ b/src/settings/rfl/settings/definition.py	2026-03-21 16:44:12.960000000 +0000
@@ -5,7 +5,7 @@
 # SPDX-License-Identifier: LGPL-3.0-or-later
 
 from pathlib import Path
-import urllib
+from urllib.parse import ParseResult, urlparse
 import ipaddress
 
 import yaml
@@ -65,7 +65,7 @@ class SettingsParameterDefinition:
         "float": float,
         "path": Path,
         "bool": bool,
-        "uri": urllib.parse.ParseResult,
+        "uri": ParseResult,
         "ip": ipaddress._BaseAddress,
         "network": ipaddress._BaseNetwork,
         "list": list,
@@ -177,7 +177,7 @@ class SettingsParameterDefinition:
         if _type == "path":
             return Path(default)
         elif _type == "uri":
-            return urllib.parse.urlparse(default)
+            return urlparse(default)
         elif _type == "ip":
             try:
                 return ipaddress.ip_address(default)
