From 8032dd77047329a93edd4c4a8b686ffc36d2f7a7 Mon Sep 17 00:00:00 2001
From: Lionel Le Folgoc <lionel@lefolgoc.net>
Date: Sun, 09 Sep 2012 09:41:18 +0000
Subject: Appearance: Fix gtk3 theme detection in install script (bug #9272).

(cherry picked from commit e3c6db6f57e72cdbdf0e293bba7d6ca650d7521b)
---
diff --git a/dialogs/appearance-settings/appearance-install-theme b/dialogs/appearance-settings/appearance-install-theme
index 13f61ae..6dc1e17 100755
--- a/dialogs/appearance-settings/appearance-install-theme
+++ b/dialogs/appearance-settings/appearance-install-theme
@@ -140,7 +140,7 @@ if test "$retval" -eq 0; then
     detectthemes "gtk-2.0/gtkrc"
 
     # install gtk-3.0 themes
-    detectthemes "gtk-3.0/gtkrc"
+    detectthemes "gtk-3.0/gtk.css"
 
     # install xfwm4 themes
     detectthemes "xfwm4/themerc"
--
cgit v0.9.0.3
