diff -ur xfce4-settings-4.7.1.orig/configure.ac xfce4-settings-4.7.1/configure.ac
--- xfce4-settings-4.7.1.orig/configure.ac	2010-01-23 13:48:05.000000000 +0200
+++ xfce4-settings-4.7.1/configure.ac	2010-02-15 14:20:55.000000000 +0200
@@ -142,6 +142,8 @@
                            [libxklavier], [Keyboard layout selection])
 XDT_CHECK_OPTIONAL_PACKAGE([LIBXKLAVIER4], [libxklavier], [4.0],
                            [libxklavier], [Keyboard layout selection])
+XDT_CHECK_OPTIONAL_PACKAGE([LIBXKLAVIER5], [libxklavier], [5.0],
+                           [libxklavier], [Keyboard layout selection])
 
 dnl ************************************
 dnl *** Xsettings daemon is optional ***
diff -ur xfce4-settings-4.7.1.orig/dialogs/keyboard-settings/xfce-keyboard-settings.c xfce4-settings-4.7.1/dialogs/keyboard-settings/xfce-keyboard-settings.c
--- xfce4-settings-4.7.1.orig/dialogs/keyboard-settings/xfce-keyboard-settings.c	2010-01-23 13:48:05.000000000 +0200
+++ xfce4-settings-4.7.1/dialogs/keyboard-settings/xfce-keyboard-settings.c	2010-02-15 14:23:24.000000000 +0200
@@ -395,7 +395,11 @@
 
 #ifdef HAVE_LIBXKLAVIER
   /* Stop xklavier engine */
+#ifdef HAVE_LIBXKLAVIER5
+  xkl_engine_stop_listen (settings->priv->xkl_engine, XKLL_TRACK_KEYBOARD_STATE);
+#else
   xkl_engine_stop_listen (settings->priv->xkl_engine);
+#endif /* HAVE_LIBXKLAVIER5 */
 #endif /* HAVE_LIBXKLAVIER */
 
   g_object_unref (G_OBJECT (settings->priv->provider));
