int64_t is defined in cstdint

diff --git a/source/ConditionSet.h b/source/ConditionSet.h
index f300f6a..7b8efe1 100644
--- a/source/ConditionSet.h
+++ b/source/ConditionSet.h
@@ -21,6 +21,8 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <string>
 #include <vector>
 
+#include <cstdint>
+
 class ConditionsStore;
 class DataNode;
 class DataWriter;
diff --git a/source/ConditionsStore.h b/source/ConditionsStore.h
index e441d3c..ba70c2d 100644
--- a/source/ConditionsStore.h
+++ b/source/ConditionsStore.h
@@ -21,6 +21,8 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <map>
 #include <string>
 
+#include <cstdint>
+
 class DataNode;
 class DataWriter;
 
diff --git a/source/DataNode.h b/source/DataNode.h
index 9b9f93f..1142e9e 100644
--- a/source/DataNode.h
+++ b/source/DataNode.h
@@ -20,7 +20,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <string>
 #include <vector>
 
-
+#include <cstdint>
 
 // A DataNode is a single line of a DataFile. It consists of one or more tokens,
 // which can be interpreted either as strings or as floating point values, and
diff --git a/source/Mortgage.h b/source/Mortgage.h
index 542707b..2a26c48 100644
--- a/source/Mortgage.h
+++ b/source/Mortgage.h
@@ -18,6 +18,8 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 
 #include <string>
 
+#include <cstdint>
+
 class DataNode;
 class DataWriter;
 
diff --git a/source/text/Format.h b/source/text/Format.h
index 2aa62de..02842e4 100644
--- a/source/text/Format.h
+++ b/source/text/Format.h
@@ -21,6 +21,7 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #include <string>
 #include <vector>
 
+#include <cstdint>
 
 
 // Collection of functions for formatting strings for display.
