From 13de48f73b894dcb1a29d5cec72e584669a948e8 Mon Sep 17 00:00:00 2001
From: Samuel Kalbfleisch <contact+gnome@mallory.re>
Date: Tue, 7 Jan 2025 17:27:59 +0100
Subject: [PATCH] src/common/e-etesync-connection.c: Replace glong by time_t

---
 src/common/e-etesync-connection.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/common/e-etesync-connection.c b/src/common/e-etesync-connection.c
index 72c432a..b212235 100644
--- a/src/common/e-etesync-connection.c
+++ b/src/common/e-etesync-connection.c
@@ -745,7 +745,7 @@ e_etesync_connection_list_existing_sync (EEteSyncConnection *connection,
 								EtebaseItemMetadata *item_meta;
 								const gchar *summary;
 								gchar *ical_str;
-								glong now;
+								time_t now;
 
 								item_meta = etebase_item_get_meta (item);
 								summary = etebase_item_metadata_get_name (item_meta);
@@ -892,7 +892,7 @@ e_etesync_connection_get_changes_sync (EEteSyncConnection *connection,
 							EtebaseItemMetadata *item_meta;
 							const gchar *summary;
 							gchar *ical_str;
-							glong now;
+							time_t now;
 
 							item_meta = etebase_item_get_meta (item);
 							summary = etebase_item_metadata_get_name (item_meta);
