--- a/Makefile
+++ b/Makefile
@@ -144,11 +144,11 @@
 mandir ?= $(prefix)/share/man
 
 install: $(MUXPS_LIB) $(MUPDF_LIB) $(FITZ_LIB) $(APPS)
-	install -d $(bindir) $(libdir) $(incdir) $(mandir)/man1
-	install $(MUXPS_LIB) $(MUPDF_LIB) $(FITZ_LIB) $(libdir)
-	install fitz/fitz.h pdf/mupdf.h xps/muxps.h $(incdir)
-	install $(PDF_APPS) $(XPS_APPS) $(MUPDF) $(bindir)
-	install $(wildcard apps/man/*.1) $(mandir)/man1
+	install -d $(DESTDIR)/$(bindir) $(DESTDIR)/$(libdir) $(DESTDIR)/$(incdir) $(DESTDIR)/$(mandir)/man1
+	install $(MUXPS_LIB) $(MUPDF_LIB) $(FITZ_LIB) $(DESTDIR)/$(libdir)
+	install fitz/fitz.h pdf/mupdf.h xps/muxps.h $(DESTDIR)/$(incdir)
+	install $(PDF_APPS) $(XPS_APPS) $(MUPDF) $(DESTDIR)/$(bindir)
+	install $(wildcard apps/man/*.1) $(DESTDIR)/$(mandir)/man1
 
 # --- Clean and Default ---
 
