diff --git a/lib-src/FileDialog/generic/FileDialogPrivate.cpp b/lib-src/FileDialog/generic/FileDialogPrivate.cpp
index 315a4cb..baa2e60 100644
--- a/lib-src/FileDialog/generic/FileDialogPrivate.cpp
+++ b/lib-src/FileDialog/generic/FileDialogPrivate.cpp
@@ -392,7 +392,7 @@ void FileData::MakeItem( wxListItem &item )
    if (IsLink())
    {
       wxColour dg = wxTheColourDatabase->Find( wxT("MEDIUM GREY") );
-      if ( dg.Ok() )
+      if ( dg.IsOk() )
          item.SetTextColour(dg);
    }
    item.m_data = (long)this;
