Taken from https://github.com/pqwy/notty/pull/35

--- a/src-unix/notty_unix.ml
+++ b/src-unix/notty_unix.ml
@@ -156,7 +156,7 @@
             Buffer.reset t.buf; t.winched <- true; set_size t dim in f)
     } in
     winsize output |> iter (set_size t);
-    Lazy.force t.unwinch |> ignore;
+    (Lazy.force t.unwinch |> ignore) [@ocaml.warning "-5"];
     if dispose then at_exit (fun () -> release t);
     write t;
     t
--- a/src/dune
+++ b/src/dune
@@ -3,7 +3,7 @@
 (library
   (public_name notty)
   (synopsis "Declaring terminals")
-  (libraries uchar uutf)
+  (libraries uutf)
   (wrapped false)
   (modules notty notty_grapheme_cluster notty_uucp notty_uucp_data)
   (private_modules notty_grapheme_cluster notty_uucp notty_uucp_data))
@@ -14,6 +14,7 @@
   (name notty_top)
   (wrapped false)
   (modules notty_top)
+  (preprocess (action (run %{bin:cppo} -V OCAML:%{ocaml_version} %{input-file})))
   (libraries notty compiler-libs.toplevel))
 
 (install
--- a/src/notty_top.ml
+++ b/src/notty_top.ml
@@ -1,4 +1,4 @@
 (* Copyright (c) 2017 David Kaloper Meršinjak. All rights reserved.
    See LICENSE.md. *)
 
-let _ = Toploop.use_silently Format.err_formatter "notty_top_init.ml"
+let _ = Toploop.use_silently Format.err_formatter (Toploop.File "notty_top_init.ml")
