*** runtime/compiler/gcc.vim	2012-06-07 04:42:38.000000000 +0400
--- runtime/compiler/gcc.vim	2012-06-20 18:30:57.298165106 +0400
***************
*** 16,21 ****
--- 16,23 ----
        \%*[^\"]\"%f\"%*\\D%l:\ %m,
        \\"%f\"%*\\D%l:%c:\ %m,
        \\"%f\"%*\\D%l:\ %m,
+       \%-G%f:%l:%c:\ note:\ each\ undeclared\ identifier\ is\ reported\ only\ once
+             \\ for\ each\ function\ it\ appears\ in,
        \%-G%f:%l:\ %trror:\ (Each\ undeclared\ identifier\ is\ reported\ only\ once,
        \%-G%f:%l:\ %trror:\ for\ each\ function\ it\ appears\ in.),
        \%f:%l:%c:\ %trror:\ %m,
