--- mdocml-1.13.4/configure.orig
+++ mdocml-1.13.4/configure
@@ -139,10 +139,10 @@
 singletest() {
 	cat 1>&3 << __HEREDOC__
 ${1}: testing...
-${COMP} ${3} -o test-${1} test-${1}.c
+${COMP} -o test-${1} test-${1}.c ${3}
 __HEREDOC__
 
-	if ${COMP} ${3} -o "test-${1}" "test-${1}.c" 1>&3 2>&3; then
+	if ${COMP} -o "test-${1}" "test-${1}.c" ${3} 1>&3 2>&3; then
 		echo "${1}: ${CC} succeeded" 1>&3
 	else
 		echo "${1}: ${CC} failed with $?" 1>&3
