--- ./Makefile.orig
+++ ./Makefile
@@ -267,9 +267,15 @@
 WWW_OBJS	 = mdocml.tar.gz \
 		   mdocml.sha256
 
+#LIBMANDOC        = libmandoc.a
+#MANDICLIB        = $(LIBMANDOC)
+LIBMANDOC         = libmandoc.so
+MANDOCLIB         = -L. -lmandoc
+
 # === USER CONFIGURATION ===============================================
 
 include Makefile.local
+CFLAGS           += -fPIC
 
 # === DEPENDENCY HANDLING ==============================================
 
@@ -296,7 +302,7 @@
 	rm -f Makefile.local config.h config.h.old config.log config.log.old
 
 clean:
-	rm -f libmandoc.a $(LIBMANDOC_OBJS) $(COMPAT_OBJS)
+	rm -f $(LIBMANDOC) $(LIBMANDOC_OBJS) $(COMPAT_OBJS)
 	rm -f mandoc $(BASE_OBJS) $(DB_OBJS)
 	rm -f man.cgi $(CGI_OBJS)
 	rm -f manpage $(MANPAGE_OBJS)
@@ -314,7 +320,7 @@
 	mkdir -p $(DESTDIR)$(MANDIR)/man7
 	$(INSTALL_PROGRAM) mandoc demandoc $(DESTDIR)$(BINDIR)
 	ln -f $(DESTDIR)$(BINDIR)/mandoc $(DESTDIR)$(BINDIR)/$(BINM_MAN)
-	$(INSTALL_LIB) libmandoc.a $(DESTDIR)$(LIBDIR)
+	$(INSTALL_LIB) $(LIBMANDOC) $(DESTDIR)$(LIBDIR)
 	$(INSTALL_LIB) man.h mandoc.h mandoc_aux.h mdoc.h \
 		$(DESTDIR)$(INCLUDEDIR)
 	$(INSTALL_MAN) mandoc.1 demandoc.1 $(DESTDIR)$(MANDIR)/man1
@@ -366,17 +372,20 @@
 libmandoc.a: $(COMPAT_OBJS) $(LIBMANDOC_OBJS)
 	$(AR) rs $@ $(COMPAT_OBJS) $(LIBMANDOC_OBJS)
 
-mandoc: $(MAIN_OBJS) libmandoc.a
-	$(CC) $(LDFLAGS) -o $@ $(MAIN_OBJS) libmandoc.a $(DBLIB)
+libmandoc.so: $(COMPAT_OBJS) $(LIBMANDOC_OBJS)
+	$(CC) $(LDFLAGS) -shared -o $@ $(COMPAT_OBJS) $(LIBMANDOC_OBJS)
 
-manpage: $(MANPAGE_OBJS) libmandoc.a
-	$(CC) $(LDFLAGS) -o $@ $(MANPAGE_OBJS) libmandoc.a $(DBLIB)
+mandoc: $(MAIN_OBJS) $(LIBMANDOC)
+	$(CC) $(LDFLAGS) -o $@ $(MAIN_OBJS) $(MANDOCLIB) $(DBLIB)
 
-man.cgi: $(CGI_OBJS) libmandoc.a
-	$(CC) $(LDFLAGS) $(STATIC) -o $@ $(CGI_OBJS) libmandoc.a $(DBLIB)
+manpage: $(MANPAGE_OBJS) $(LIBMANDOC)
+	$(CC) $(LDFLAGS) -o $@ $(MANPAGE_OBJS) $(MANDOCLIB) $(DBLIB)
 
-demandoc: $(DEMANDOC_OBJS) libmandoc.a
-	$(CC) $(LDFLAGS) -o $@ $(DEMANDOC_OBJS) libmandoc.a
+man.cgi: $(CGI_OBJS) $(LIBMANDOC)
+	$(CC) $(LDFLAGS) $(STATIC) -o $@ $(CGI_OBJS) $(MANDOCLIB) $(DBLIB)
+
+demandoc: $(DEMANDOC_OBJS) $(LIBMANDOC)
+	$(CC) $(LDFLAGS) -o $@ $(DEMANDOC_OBJS) $(MANDOCLIB)
 
 # --- maintainer targets ---
 
