From 6deb9b4724afe97fd13ae5270353604bb9e93d4f Mon Sep 17 00:00:00 2001
From: Vladimir Stoiakin <VStoiakin@lavabit.com>
Date: Fri, 01 Jul 2022 21:03:09 +0300
Subject: [PATCH] build: add dependency on libavutil for GNU/Linux

Change-Id: I3d43fd9f5782f5ebf7db9b2cef6b5d294e7aa36c
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b88bf3..9c1e838 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -359,6 +359,7 @@
   endif()
 
   pkg_check_modules(XCB xcb)
+  pkg_check_modules(LIBAVUTIL libavutil>=55.75.100)
 
   include_directories(
     ${LIBCLIENT_SRC_DIR}
@@ -366,7 +367,8 @@
     ${LIBNOTIFY_INCLUDE_DIRS}
     ${LIBGDKPIXBUF_INCLUDE_DIRS}
     ${GLIB_INCLUDE_DIRS}
-    ${XCB_INCLUDE_DIRS})
+    ${XCB_INCLUDE_DIRS}
+    ${LIBAVUTIL_INCLUDE_DIRS})
 
   set(JAMI_DATA_PREFIX "${CMAKE_INSTALL_PREFIX}/share")
 
@@ -548,7 +550,8 @@
     ${LIBGDKPIXBUF_LIBRARIES}
     ${GLIB_LIBRARIES}
     ${GIO_LIBRARIES}
-    ${XCB_LIBRARIES})
+    ${XCB_LIBRARIES}
+    ${LIBAVUTIL_LIBRARIES})
 
   # Installation rules
   install(
