diff --git a/main_test.go b/main_test.go
index 3e4964f..337d8fa 100644
--- a/main_test.go
+++ b/main_test.go
@@ -260,7 +260,7 @@ func TestBuildDll(t *testing.T) {
 				exec: "dotnet",
 			}

-			binaryPath, err := host.buildDebuggingDLL(c.EntryPoint)
+			binaryPath, err := host.buildDebuggingDLL(e.RootPath, c.EntryPoint)

 			if c.ExpectedErrorContains != "" {
 				assert.ErrorContains(t, err, c.ExpectedErrorContains)
