From 56121dcab673c304c437334bb3b160dded3db725 Mon Sep 17 00:00:00 2001
From: Jacob Rief <jacob.rief@gmail.com>
Date: Fri, 27 Dec 2019 17:20:13 +0100
Subject: [PATCH 02/11] add user to rrequest object

---
 treebeard/tests/test_treebeard.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/treebeard/tests/test_treebeard.py b/treebeard/tests/test_treebeard.py
index 3185582..5fa2c99 100644
--- a/treebeard/tests/test_treebeard.py
+++ b/treebeard/tests/test_treebeard.py
@@ -8,7 +8,7 @@ import os
 from django import VERSION as DJANGO_VERSION
 from django.contrib.admin.sites import AdminSite
 from django.contrib.admin.views.main import ChangeList
-from django.contrib.auth.models import User
+from django.contrib.auth.models import User, AnonymousUser
 from django.contrib.messages.storage.fallback import FallbackStorage
 from django.db.models import Q
 from django.template import Template, Context
@@ -2367,6 +2367,7 @@ class TestAdminTree(TestNonEmptyTree):
         """
         model = model_without_proxy
         request = RequestFactory().get('/admin/tree/')
+        request.user = AnonymousUser()
         site = AdminSite()
         form_class = movenodeform_factory(model)
         admin_class = admin_factory(form_class)
-- 
2.25.4

