From d28870061d60b609ecab9c9eb5968eee21db46cf Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Thu, 2 Jul 2020 00:45:52 +0200
Subject: [PATCH 10/11] tox.ini: Document state of environment *-dj22-mssql

---
 tox.ini | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tox.ini b/tox.ini
index 48de565..777fae6 100644
--- a/tox.ini
+++ b/tox.ini
@@ -5,6 +5,9 @@
 #
 
 [tox]
+# mssql: Note that django-pyodbc-azure 2.1.0.0 does not support Django 2.2,
+#        see https://github.com/michiya/django-pyodbc-azure/issues/192 ;
+#        as a result, environments *-dj22-mssql are not actually usable.
 envlist =
     {py27,py35,py36,py37}-{dj111,dj20,dj21,dj22}-{sqlite,postgres,mysql,mssql}
 
-- 
2.25.4

