On riscv64, it takes slightly over 0.5 seconds.
diff --git a/tests/test_ltree_utils.py b/tests/test_ltree_utils.py
index 235b898..a01ce03 100644
--- a/tests/test_ltree_utils.py
+++ b/tests/test_ltree_utils.py
@@ -44,11 +44,11 @@ class TestGenerateLabel:
         start = time.perf_counter()
         res = generate_label(before="A" * 59999 + "B", after="A" * 60000, skip=set())
         time_taken = time.perf_counter() - start
-        assert time_taken < 0.5  # Conservative to allow for flakiness on CI
+        assert time_taken < 0.6  # Conservative to allow for flakiness on CI
         assert res == "A" * 60000 + "0"
 
         start = time.perf_counter()
         res = generate_label(before="B", after="A" * 60000, skip=set())
         time_taken = time.perf_counter() - start
-        assert time_taken < 0.05
+        assert time_taken < 0.06
         assert res == "AB"
