diff -rupN a/seccompsandbox.c b/seccompsandbox.c
--- a/seccompsandbox.c	2012-09-18 08:52:30.000000000 +0200
+++ b/seccompsandbox.c	2014-02-05 20:22:15.919607188 +0100
@@ -21,7 +21,7 @@
 
 #include <netinet/in.h>
 
-#include <sys/fcntl.h>
+#include <fcntl.h>
 #include <sys/mman.h>
 #include <sys/prctl.h>
 #include <sys/socket.h>
diff -rupN a/sysdeputil.c b/sysdeputil.c
--- a/sysdeputil.c	2012-09-16 06:18:04.000000000 +0200
+++ b/sysdeputil.c	2014-02-05 20:20:07.212933389 +0100
@@ -1216,7 +1216,9 @@ vsf_insert_uwtmp(const struct mystr* p_u
   setutxent();
   (void) pututxline(&s_utent);
   endutxent();
+#if defined(WTMPX_FILE)
   updwtmpx(WTMPX_FILE, &s_utent);
+#endif
 }
 
 void
@@ -1235,7 +1237,9 @@ vsf_remove_uwtmp(void)
   (void) pututxline(&s_utent);
   endutxent();
   s_utent.ut_tv.tv_sec = vsf_sysutil_get_time_sec();
+#if defined(WTMPX_FILE)
   updwtmpx(WTMPX_FILE, &s_utent);
+#endif
 }
 
 #endif /* !VSF_SYSDEP_HAVE_UTMPX */
