diff --git a/setup.py b/setup.py
index 051e9f6..0421483 100644
--- a/setup.py
+++ b/setup.py
@@ -3,4 +3,3 @@
 from setuptools import setup, find_packages
-from imp import load_source
-
+from importlib.machinery import SourceFileLoader
 
@@ -8,3 +7,3 @@ setup(
     name='Flask-Components',
-    version=load_source('', 'src/flask_components/_version.py').__version__,
+    version=SourceFileLoader('', 'src/flask_components/_version.py').load_module().__version__,
     description='A simple flask extension to discover files in a declared '
