diff --git a/mk/support/pkg/v8.sh b/mk/support/pkg/v8.sh
index 5454e76..58bf70e 100644
--- a/mk/support/pkg/v8.sh
+++ b/mk/support/pkg/v8.sh
@@ -47,6 +47,10 @@ pkg_install () {
         *)      arch=native ;;
     esac
     mode=release
+    pkg_make out/Makefile.$arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
+    pkg_make -C out -f Makefile.$arch.$mode mksnapshot BUILDTYPE=Release builddir="$build_dir/out/$arch.$mode" \
+        CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
+    paxmark -m "$build_dir"/out/$arch.$mode/mksnapshot
     pkg_make $arch.$mode CXX=$CXX LINK=$CXX LINK.target=$CXX GYPFLAGS="-Dwerror= $arch_gypflags" V=1
     for lib in `find "$build_dir/out/$arch.$mode" -maxdepth 1 -name \*.a` `find "$build_dir/out/$arch.$mode/obj.target" -name \*.a`; do
         name=`basename $lib`
