--- a/zstd-ruby.gemspec
+++ b/zstd-ruby.gemspec
@@ -23,11 +23,7 @@
   #    "public gem pushes."
   #end
 
-  spec.files         = `git ls-files -z`.split("\x0").reject do |f|
-    f.match(%r{^(test|spec|features|benchmarks|zstd|.github|examples)/})
-  end
-  spec.bindir        = "exe"
-  spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
+  spec.files         = Dir["ext/**/*", "lib/**/*.rb"]
   spec.require_paths = ["lib"]
   spec.extensions    = ["ext/zstdruby/extconf.rb"]
 
