--- bsnes_v077-source/bsnes/snes/system/serialization.cpp
+++ bsnes_v077-source/bsnes/snes/system/serialization.cpp
@@ -7,7 +7,7 @@
   char profile[16], description[512];
   memset(&profile, 0, sizeof profile);
   memset(&description, 0, sizeof description);
-  strlcpy(profile, Info::Profile, sizeof profile);
+  nall::strlcpy(profile, Info::Profile, sizeof profile);
 
   s.integer(signature);
   s.integer(version);
--- bsnes_v077-source/bsnes/ui/tools/state-manager.cpp
+++ bsnes_v077-source/bsnes/ui/tools/state-manager.cpp
@@ -130,7 +130,7 @@
 string StateManager::slotLoadDescription(unsigned i) {
   if(slot[i].capacity() == 0) return "(empty)";
   char text[512];
-  strlcpy(text, (const char*)slot[i].data() + HeaderLength, 512);
+  nall::strlcpy(text, (const char*)slot[i].data() + HeaderLength, 512);
   return text;
 }
 
@@ -138,7 +138,7 @@
   if(stateList.selected() == false) return;
   string text = descEdit.text();
   if(slot[stateList.selection()].capacity() > 0) {
-    strlcpy((char*)slot[stateList.selection()].data() + HeaderLength, (const char*)text, 512);
+    nall::strlcpy((char*)slot[stateList.selection()].data() + HeaderLength, (const char*)text, 512);
   }
   refresh();
 }
