--- 3.26.1-origin/Makefile.in
+++ 3.26.1/Makefile.in
@@ -492,7 +492,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 ACLOCAL_AMFLAGS = -I support
-SUBDIRS = . man doc tests systemd gznbd
+SUBDIRS = . man doc tests gznbd
 noinst_LTLIBRARIES = libnbdsrv.la libcliserv.la libnbdclt.la
 libcliserv_la_SOURCES = cliserv.h cliserv.c
 libcliserv_la_CFLAGS = @CFLAGS@
--- 3.26.1-origin/configure
+++ 3.26.1/configure
@@ -16775,8 +16775,6 @@
 elif test "x$DB2M" != "x"
 then :
   enable_manpages=yes
-else $as_nop
-  as_fn_error $? "don't know what to do here" "$LINENO" 5
 fi
 { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking whether to build manpages" >&5
 printf %s "checking whether to build manpages... " >&6; }
