--- ./CMakeLists.txt.orig
+++ ./CMakeLists.txt
@@ -19,9 +19,12 @@
 # Build Translations
 CREATE_TRANSLATION_TARGETS()
 
+find_package(Python REQUIRED COMPONENTS Interpreter)
 
-install(DIRECTORY UM DESTINATION "${Python_SITELIB_LOCAL}")
+install(DIRECTORY UM DESTINATION "${Python_SITELIB}")
 
 install(FILES ${CMAKE_SOURCE_DIR}/cmake/UraniumTranslationTools.cmake
         DESTINATION ${CMAKE_INSTALL_DATADIR}/cmake-${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}/Modules/ )
 install(DIRECTORY resources DESTINATION ${CMAKE_INSTALL_DATADIR}/uranium)
+
+include(UraniumPluginInstall)
