From 1271891d88af3e0571f8a10c97f5a78795ea2ee9 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 28 Mar 2014 13:25:35 +0000
Subject: [PATCH] buildrepo: show pkgversion of packages

---
 buildrepo.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/buildrepo.lua b/buildrepo.lua
index ec92572..21b444b 100755
--- a/buildrepo.lua
+++ b/buildrepo.lua
@@ -179,9 +179,9 @@ for _,repo in pairs(args) do
 	for aport in db:each_in_build_order(pkgs) do
 		tried = tried + 1
 		local totally_built = stats[repo].relevant_aports - #pkgs + built
-		io.write(("%d/%d %d/%d %s\n"):format(tried, #pkgs,
+		io.write(("%d/%d %d/%d %s %s\n"):format(tried, #pkgs,
 					totally_built, stats[repo].relevant_aports,
-					aport.pkgname))
+					aport.pkgname, aport.pkgver))
 		if not db:known_deps_exists(aport) then
 			warn("%s: Skipped due to missing dependencies", aport.pkgname)
 		elseif build_aport(aport, repodest, logdir, opts.s) then
-- 
1.9.1

