From e9ec2aa96e4357a71b2579fee7459c943db4fdcb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timo=20Ter=C3=A4s?= <timo.teras@iki.fi>
Date: Tue, 24 Feb 2026 18:58:40 +0200
Subject: [PATCH 1/1] ap: fix apk-list to print subpackage names correctly

---
 bin/ap.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/ap.lua b/bin/ap.lua
index bee4089..66459a7 100755
--- a/bin/ap.lua
+++ b/bin/ap.lua
@@ -102,9 +102,9 @@ subcmd["apk-list"] = {
 	desc = "List all apk files",
 	usage = "",
 	run = function(db)
-		for pkg in db:each() do
+		for pkg, name in db:each() do
 			if pkg:relevant() then
-				print(pkg:get_apk_file_name())
+				print(pkg:get_apk_file_name(name))
 			end
 		end
 	end,
-- 
2.53.0

