From 4efd713435315918becc1a838fa3fb4f29e13d58 Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 24 Jul 2017 21:47:54 +0200
Subject: [PATCH] makefile: add dump.lua

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 22d5f77..22c68ce 100644
--- a/Makefile
+++ b/Makefile
@@ -10,6 +10,7 @@ aportsfiles= \
 	abuild.lua \
 	apkrepo.lua \
 	db.lua \
+	dump.lua \
 	pkg.lua
 
 binfiles=buildrepo.lua ap.lua
