diff --git a/client/sngtc_client/sngtc_client.c b/client/sngtc_client/sngtc_client.c
index b01c106..25e85af 100644
--- a/client/sngtc_client/sngtc_client.c
+++ b/client/sngtc_client/sngtc_client.c
@@ -1134,7 +1134,7 @@ static void *run_transcoding_session(void *arg)
 	/* go! */
 	globals.sessions_streaming++;
 
-	thread_id = syscall(SYS_gettid);
+	thread_id = (pthread_t)syscall(SYS_gettid);
 	pthread_setspecific(globals.thread_id_key, (void *)(long)thread_id);
 	local_sngtc_logger(SNGTC_LOGLEVEL_DEBUG, "Running session %d in thread %d\n", sessindex, thread_id);
 
