diff --git a/client/sngtc_client/sngtc_client.c b/client/sngtc_client/sngtc_client.c
index b01c106..bfc504a 100644
--- a/client/sngtc_client/sngtc_client.c
+++ b/client/sngtc_client/sngtc_client.c
@@ -714,7 +714,7 @@ static int create_rtp_socket(void *usr_priv, sngtc_codec_request_leg_t *codec_re
 
 	rtp_session_set_blocking_mode(session, FALSE);
 
-	rtp_session_set_local_addr(session, globals.bindip_str, local_port);
+	rtp_session_set_local_addr(session, globals.bindip_str, local_port, local_port+1);
 
 	rtp_session_set_connected_mode(session, TRUE);
 
diff --git a/client/sngtc_client/sngtc_client.c b/client/sngtc_client/sngtc_client.c
index bfc504a..8f720e2 100644
--- a/client/sngtc_client/sngtc_client.c
+++ b/client/sngtc_client/sngtc_client.c
@@ -1875,9 +1875,9 @@ int main(int argc, char*argv[])
 	/*ortp_scheduler_init();*/
 
 	if (globals.rtpdebug) {
-		ortp_set_log_level_mask(ORTP_DEBUG|ORTP_MESSAGE|ORTP_WARNING|ORTP_ERROR|ORTP_FATAL);
+		ortp_set_log_level_mask(NULL, ORTP_DEBUG|ORTP_MESSAGE|ORTP_WARNING|ORTP_ERROR|ORTP_FATAL);
 	} else {
-		ortp_set_log_level_mask(ORTP_WARNING|ORTP_ERROR|ORTP_FATAL);
+		ortp_set_log_level_mask(NULL, ORTP_WARNING|ORTP_ERROR|ORTP_FATAL);
 	}
 
 
