diff -Nurp a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2024-01-23 18:24:18.000000000 +0000
+++ b/CMakeLists.txt	2024-01-27 17:17:02.432298551 +0000
@@ -567,11 +567,11 @@ if (NOT DRM_INCLUDE_DIRS)
 endif()
 
 if (DRM_INCLUDE_DIRS)
-  if (EXISTS ${DRM_INCLUDE_DIRS}/i915_drm.h AND EXISTS ${DRM_INCLUDE_DIRS}/amdgpu_drm.h)
+#  if (EXISTS ${DRM_INCLUDE_DIRS}/i915_drm.h AND EXISTS ${DRM_INCLUDE_DIRS}/amdgpu_drm.h)
     include_directories(${DRM_INCLUDE_DIRS})
-  else()
-    unset(DRM_INCLUDE_DIRS CACHE)
-  endif()
+#  else()
+#    unset(DRM_INCLUDE_DIRS CACHE)
+#  endif()
 endif()
 
 # LTTng Tracer support
