--- docopt-ng-0.9.0-orig/tests/conftest.py	2023-05-30 13:38:46.000000000 -0700
+++ docopt-ng-0.9.0/tests/conftest.py	2024-05-22 16:12:53.952007724 -0700
@@ -10,7 +10,7 @@ import pytest
 import docopt
 
 
-def pytest_collect_file(file_path: Path, path, parent):
+def pytest_collect_file(file_path: Path, parent):
     if file_path.suffix == ".docopt" and file_path.stem.startswith("test"):
         return DocoptTestFile.from_parent(path=file_path, parent=parent)
 
