diff -Naur sessreg-1.0.8.orig/sessreg.h sessreg-1.0.8/sessreg.h
--- sessreg-1.0.8.orig/sessreg.h	2013-02-10 21:43:50.000000000 -0500
+++ sessreg-1.0.8/sessreg.h	2014-06-18 10:02:17.769459663 -0400
@@ -106,3 +106,8 @@
 #ifndef UTMPX_FILE
 # define UTMPX_FILE	_PATH_UTMPX
 #endif
+
+#ifndef _PATH_WTMPX
+# define _PATH_WTMPX	"/var/log/wtmp"
+#endif
+
diff --git a/sessreg.h b/sessreg.h
index b3bbac9..d0c0105 100644
--- a/sessreg.h
+++ b/sessreg.h
@@ -114,3 +114,7 @@
 # define _PATH_WTMPX	"/var/log/wtmp"
 #endif
 
+#ifndef _PATH_UTMPX
+# define _PATH_UTMPX    "/var/log/utmp"
+#endif
+
