diff --git a/Makefile b/Makefile
index ddd5f9a..d40fd6a 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -40,7 +40,7 @@ WEBDIR = $(prefix)/www
 # CONFIGURE: The group that the web directory belongs to.  This is so that
 # the makeweb program can be installed set-group-id to that group, and make
 # subdirectories.  If you're not going to use makeweb, ignore this.
-WEBGROUP =	www
+WEBGROUP =	www-data
 
 # CONFIGURE: Directory for CGI executables.
 CGIBINDIR =	$(WEBDIR)/cgi-bin
