diff --git a/test/mitmproxy/io/test_tnetstring.py b/test/mitmproxy/io/test_tnetstring.py
index caf37fe..250bf74 100644
--- a/test/mitmproxy/io/test_tnetstring.py
+++ b/test/mitmproxy/io/test_tnetstring.py
@@ -87,7 +87,7 @@ class Test_Format(unittest.TestCase):
             self.assertEqual((v, b""), tnetstring.pop(tnetstring.dumps(v)))
 
     def test_roundtrip_big_integer(self):
-        i1 = math.factorial(30000)
+        i1 = math.factorial(1557)
         s = tnetstring.dumps(i1)
         i2 = tnetstring.loads(s)
         self.assertEqual(i1, i2)
