Upstream: No
Reason: Use system CA certificates

diff --git a/mitmproxy/net/tls.py b/mitmproxy/net/tls.py
index 8e217ec0c..125d9874f 100644
--- a/mitmproxy/net/tls.py
+++ b/mitmproxy/net/tls.py
@@ -8,7 +8,6 @@ import struct
 import threading
 import typing
 
-import certifi
 from OpenSSL import SSL
 from kaitaistruct import KaitaiStream
 
@@ -178,7 +177,7 @@ def _create_ssl_context(
     if verify is not None:
         context.set_verify(verify, verify_callback)
         if ca_path is None and ca_pemfile is None:
-            ca_pemfile = certifi.where()
+            ca_pemfile = "/etc/ssl/certs/ca-certificates.crt"
         try:
             context.load_verify_locations(ca_pemfile, ca_path)
         except SSL.Error:
