diff --git a/CMakeLists.txt b/CMakeLists.txt
index db7cb9b..3681573 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,9 +190,9 @@ configure_file (VERSION.in VERSION)
 ## Program
 
 if (MINGW)
-  set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2")
+  set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2")
 else (MINGW)
-  set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -D_FORTIFY_SOURCE=2 -fstack-protector")
+  set (HARDENING_FLAGS            "-Wformat -Wformat-security -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -fstack-protector")
   set (LINKER_HARDENING_FLAGS     "-Wl,-z,relro -Wl,-z,now")
 endif (MINGW)
 
