diff --git a/purgatory/arch/arm64/Makefile b/purgatory/arch/arm64/Makefile
index 636abea..9aad157 100644
--- a/purgatory/arch/arm64/Makefile
+++ b/purgatory/arch/arm64/Makefile
@@ -3,6 +3,7 @@ arm64_PURGATORY_EXTRA_CFLAGS = \
 	-mcmodel=large \
 	-fno-stack-protector \
 	-fno-asynchronous-unwind-tables \
+	-fno-PIC \
 	-Wundef \
 	-Werror-implicit-function-declaration \
 	-Wdeclaration-after-statement \
