diff --git a/inc/Module/Install/RTx.pm b/inc/Module/Install/RTx.pm
index 20a354b..7e681c6 100755
--- a/inc/Module/Install/RTx.pm
+++ b/inc/Module/Install/RTx.pm
@@ -83,13 +83,13 @@ sub RTx {
     if ( $RT::LocalPluginPath ) {
         die "Because of bugs in RT 3.8.0 this extension can not be installed.\n"
             ."Upgrade to RT 3.8.1 or newer.\n" if $RT::VERSION =~ /^3\.8\.0/;
-        $path{$_} = $RT::LocalPluginPath . "/$original_name/$_"
+        $path{$_} = '$(DESTDIR)' . $RT::LocalPluginPath . "/$original_name/$_"
             foreach @DIRS;
     } else {
         foreach ( @DIRS ) {
             no strict 'refs';
             my $varname = "RT::Local" . ucfirst($_) . "Path";
-            $path{$_} = ${$varname} || "$RT::LocalPath/$_";
+            $path{$_} = '$(DESTDIR)' . (${$varname} || "$RT::LocalPath/$_");
         }
 
         $path{$_} .= "/$name" for grep $path{$_}, qw(etc po var);

