--- a/pkgcfgs
+++ b/pkgcfgs
@@ -2,34 +2,34 @@
 Name: libmustach
 Version: VERSION
 Description: C Mustach single library
-Cflags: -Imustach
+Cflags: -I/usr/include/mustach
 Libs: -lmustach
 
 ==libmustach-core.pc==
 Name: libmustach-core
 Version: VERSION
 Description: C Mustach core library
-Cflags: -Imustach
+Cflags: -I/usr/include/mustach
 Libs: -lmustach-core
 
 ==libmustach-cjson.pc==
 Name: libmustach-cjson
 Version: VERSION
 Description: C Mustach library for cJSON
-Cflags: -Imustach
+Cflags: -I/usr/include/mustach
 Libs: -lmustach-cjson
 
 ==libmustach-json-c.pc==
 Name: libmustach-json-c
 Version: VERSION
 Description: C Mustach library for json-c
-Cflags: -Imustach
+Cflags: -I/usr/include/mustach
 Libs: -lmustach-json-c
 
 ==libmustach-jansson.pc==
 Name: libmustach-jansson
 Version: VERSION
 Description: C Mustach library for jansson
-Cflags: -Imustach
+Cflags: -I/usr/include/mustach
 Libs: -lmustach-jansson
 
