Requires python2

diff --git a/apps/CMakeLists.txt b/apps/CMakeLists.txt
index e2668eb..c8cc2d2 100644
--- a/apps/CMakeLists.txt
+++ b/apps/CMakeLists.txt
@@ -6,6 +6,7 @@ ADD_SUBDIRECTORY(callback)
 ADD_SUBDIRECTORY(click2dial)
 IF(PYTHONLIBS_FOUND)
 	ADD_SUBDIRECTORY(conf_auth)
+	ADD_SUBDIRECTORY(dsm)
 ENDIF(PYTHONLIBS_FOUND)
 ADD_SUBDIRECTORY(conference)
 IF(MYSQLPP_FOUND)
@@ -14,7 +15,6 @@ ENDIF(MYSQLPP_FOUND)
 IF(OPENSSL_FOUND)
 	ADD_SUBDIRECTORY(diameter_client)
 ENDIF(OPENSSL_FOUND)
-ADD_SUBDIRECTORY(dsm)
 ADD_SUBDIRECTORY(early_announce)
 #ADD_SUBDIRECTORY(examples)
 IF(MISDN_FOUND)

