From 8712daf57f4ec10caadd215dc969b1f261a949b5 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 29 Dec 2010 17:02:49 +0300
Subject: [PATCH 09/11] Don't start py_sems by default

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 core/etc/sems.conf.cmake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/core/etc/sems.conf.cmake b/core/etc/sems.conf.cmake
index eeeeb67..0e0ca62 100644
--- a/core/etc/sems.conf.cmake
+++ b/core/etc/sems.conf.cmake
@@ -140,7 +140,7 @@ plugin_path=${SEMS_EXEC_PREFIX}/${SEMS_LIBDIR}/sems/plug-in/
 #
 # o binrpcctrl and unixsockctrl: excluded as sipctrl is default ctrl
 # o precoded_announce: no precoded sample files present
-exclude_plugins=binrpcctrl;unixsockctrl;precoded_announce;sw_prepaid_sip
+exclude_plugins=binrpcctrl;unixsockctrl;precoded_announce;sw_prepaid_sip;py_sems
 
 # optional parameter: application
 # 
-- 
1.7.3.4

