From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 8 Apr 2016 14:50:23 +0300
Subject: [PATCH] Fix for C++11

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>

diff --git a/apps/sbc/RegisterCache.cpp b/apps/sbc/RegisterCache.cpp
index cbd3c30..5cd1235 100644
--- a/apps/sbc/RegisterCache.cpp
+++ b/apps/sbc/RegisterCache.cpp
@@ -955,7 +955,7 @@ bool _RegisterCache::throttleRegister(RegisterCacheCtx& ctx,
       return false; // fwd
     }
 
-    alias_updates.push_back(make_pair<string,long int>(reg_binding.alias,
+    alias_updates.push_back(make_pair(reg_binding.alias,
 						       contact_expires));
   }
 
diff --git a/apps/sbc/RegisterDialog.cpp b/apps/sbc/RegisterDialog.cpp
index 820e03d..88b94cb 100644
--- a/apps/sbc/RegisterDialog.cpp
+++ b/apps/sbc/RegisterDialog.cpp
@@ -239,7 +239,7 @@ int RegisterDialog::fixUacContacts(const AmSipRequest& req)
 	continue;
       }
 
-      alias_updates.push_back(make_pair<string,long int>(reg_binding.alias,
+      alias_updates.push_back(make_pair(reg_binding.alias,
 							 contact_expires));
     }
 
diff --git a/core/sip/resolver.cpp b/core/sip/resolver.cpp
index 9793069..7bc6ff7 100644
--- a/core/sip/resolver.cpp
+++ b/core/sip/resolver.cpp
@@ -823,7 +823,7 @@ dns_entry_map::insert(const dns_entry_map::value_type& x)
 bool dns_entry_map::insert(const string& key, dns_entry* e)
 {
     std::pair<iterator, bool> res =
-    	insert(make_pair<const key_type&,mapped_type>(key,e));
+	insert(make_pair(key,e));
 
     if(res.second) {
 	inc_ref(e);
