From 7f2c45fc1e9c2c1900244709abfa125c9b7e5755 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 29 Dec 2010 16:02:44 +0300
Subject: [PATCH 07/11] Fix gateway module's version passing in CMake

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 apps/gateway/CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/apps/gateway/CMakeLists.txt b/apps/gateway/CMakeLists.txt
index e18389a..ef0acde 100644
--- a/apps/gateway/CMakeLists.txt
+++ b/apps/gateway/CMakeLists.txt
@@ -6,7 +6,7 @@ mISDNNames.cpp
 mISDNStack.cpp
 )
 
-ADD_DEFINITIONS(-DGW_VERSION="0.1")
+ADD_DEFINITIONS(-DGW_VERSION=\\\"0.1\\\")
 INCLUDE_DIRECTORIES(${MISDN_INCLUDE_DIR}/mISDNuser)
 
 SET(sems_module_name gateway)
-- 
1.7.3.4

