From 57e8f6ccdf5335207f7017414487db6bf81a64f3 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 29 Dec 2010 16:55:50 +0300
Subject: [PATCH 08/11] Install email template

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 apps/voicemail/CMakeLists.txt |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/apps/voicemail/CMakeLists.txt b/apps/voicemail/CMakeLists.txt
index b82736f..38778bc 100644
--- a/apps/voicemail/CMakeLists.txt
+++ b/apps/voicemail/CMakeLists.txt
@@ -25,3 +25,7 @@ SET(sems_module_name voicemail)
 INCLUDE(${CMAKE_SOURCE_DIR}/cmake/module.rules.txt)
 INCLUDE(${CMAKE_SOURCE_DIR}/cmake/audio.rules.txt)
 INCLUDE(${CMAKE_SOURCE_DIR}/cmake/config.rules.txt)
+
+INSTALL(
+	FILES ./default.template  DESTINATION ${SEMS_CFG_PREFIX}/etc/sems/
+)
-- 
1.7.3.4

