--- a/core/AmRtpAudio.cpp	2021-10-21 10:06:12.000000000 +0200
+++ b/core/AmRtpAudio.cpp	2021-12-02 09:48:19.320236030 +0100
@@ -76,7 +76,8 @@ void AmAudioRtpFormat::initCodec()
     } else {
       if (NULL != sdp_format_parameters_out) {
 	DBG("negotiated fmt parameters '%s'\n", sdp_format_parameters_out);
-	log_demangled_stacktrace(L_DBG, 30);
+	// MUSL BACKTRACE NOT WORKING
+	//log_demangled_stacktrace(L_DBG, 30);
       }
 
       if (NULL != fmt_i) {
