From 708b02d96f4a4acb7c4ede0387a84d2f5948e0db Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Mon, 27 Dec 2010 17:18:46 +0300
Subject: [PATCH 06/11] last-minute fix - proper version

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 CMakeLists.txt |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c36ce62..78a6dfb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,7 @@ PROJECT (SEMS)
 
 SET(SEMS_MAJOR_VERSION 1)
 SET(SEMS_MINOR_VERSION 3)
-SET(SEMS_BUILD_VERSION 0)
+SET(SEMS_BUILD_VERSION 1)
 SET(SEMS_EXTRA_VERSION dev)
 
 FIND_PATH(IS_DEVEL ".git" PATHS "${CMAKE_SOURCE_DIR}")
-- 
1.7.3.4

