diff --git a/src/github.com/cockroachdb/cockroach/Makefile b/src/github.com/cockroachdb/cockroach/Makefile
index f988916..5d0bf8d 100644
--- a/src/github.com/cockroachdb/cockroach/Makefile
+++ b/src/github.com/cockroachdb/cockroach/Makefile
@@ -253,18 +253,18 @@ $(foreach v,$(filter-out $(strip $(VALID_VARS)),$(.VARIABLES)),\
 .ALWAYS_REBUILD:
 .PHONY: .ALWAYS_REBUILD
 
-ifneq ($(GIT_DIR),)
+#ifneq ($(GIT_DIR),)
 # If we're in a git worktree, the git hooks directory may not be in our root,
 # so we ask git for the location.
 #
 # Note that `git rev-parse --git-path hooks` requires git 2.5+.
-GITHOOKS := $(subst githooks/,$(GITHOOKSDIR)/,$(wildcard githooks/*))
-$(GITHOOKSDIR)/%: githooks/%
-	@echo installing $<
-	@rm -f $@
-	@mkdir -p $(dir $@)
-	@ln -s ../../$(basename $<) $(dir $@)
-endif
+#GITHOOKS := $(subst githooks/,$(GITHOOKSDIR)/,$(wildcard githooks/*))
+#$(GITHOOKSDIR)/%: githooks/%
+#	@echo installing $<
+#	@rm -f $@
+#	@mkdir -p $(dir $@)
+#	@ln -s ../../$(basename $<) $(dir $@)
+#endif
 
 .SECONDARY: pkg/ui/yarn.installed
 pkg/ui/yarn.installed: pkg/ui/package.json pkg/ui/yarn.lock pkg/ui/yarn.protobufjs-cli.lock | bin/.submodules-initialized
