From c2503017d7f70c83a7bcf25da356db15b5015005 Mon Sep 17 00:00:00 2001
From: Alan Lacerda <alacerda@alpinelinux.org>
Date: Thu, 16 Jul 2015 17:45:10 +0000
Subject: [PATCH] fix accessperms in tests

---
 tests/document-saver.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/document-saver.c b/tests/document-saver.c
index 455ef65..a36700d 100644
--- a/tests/document-saver.c
+++ b/tests/document-saver.c
@@ -27,6 +27,10 @@
 #include <string.h>
 #include <sys/stat.h>
 
+#ifndef ACCESSPERMS
+#define ACCESSPERMS (S_IRWXU|S_IRWXG|S_IRWXO)
+#endif
+
 #define DEFAULT_LOCAL_URI "/tmp/pluma-document-saver-test.txt"
 #define DEFAULT_REMOTE_URI "sftp://localhost/tmp/pluma-document-saver-test.txt"
 #define DEFAULT_CONTENT "hello world!"
-- 
2.4.6

