--- a/Makefile
+++ b/Makefile
@@ -79,10 +79,10 @@
 # Install.
 
 install:
-	rm -rf $(DESTDIR)$(OCAMLLIBDIR)/site-lib/poCaml
-	install -c -m 0755 -d $(DESTDIR)$(OCAMLLIBDIR)/site-lib/poCaml
+	rm -rf $(DESTDIR)$(OCAMLLIBDIR)/poCaml
+	install -c -m 0755 -d $(DESTDIR)$(OCAMLLIBDIR)/poCaml
 	install -c -m 0755 -d $(DESTDIR)$(OCAMLLIBDIR)/stublibs
 	install -c -m 0644 poCaml.cmi poCaml.cma poCaml.cmxa \
 	  poCaml.a libpoCaml.a META \
-	  $(DESTDIR)$(OCAMLLIBDIR)/site-lib/poCaml
-	install -c -m 0644 dllpoCaml.so $(DESTDIR)$(OCAMLLIBDIR)/stublibs
+	  $(DESTDIR)$(OCAMLLIBDIR)/poCaml
+	install -c -m 0755 dllpoCaml.so $(DESTDIR)$(OCAMLLIBDIR)/stublibs
