diff -Nru quagga-0.99.11.orig/lib/sockopt.c quagga-0.99.11/lib/sockopt.c
--- quagga-0.99.11.orig/lib/sockopt.c	2008-09-05 14:27:26.000000000 +0000
+++ quagga-0.99.11/lib/sockopt.c	2008-10-13 21:46:13.000000000 +0000
@@ -530,6 +530,7 @@
           return -1;
         };
       
+#ifdef HAVE_IPV6
       /* If this does not work, then all users of this sockopt will need to
        * differentiate between IPv4 and IPv6, and keep seperate sockets for
        * each. 
@@ -546,6 +547,7 @@
            su2->sin6.sin6_addr.s6_addr32[2] = htonl(0xffff);
            memcpy (&su2->sin6.sin6_addr.s6_addr32[3], &su->sin.sin_addr, 4);
         }
+#endif /* HAVE_IPV6 */
     }
   
   memset (&md5sig, 0, sizeof (md5sig));
