--- a/ox.gemspec
+++ b/ox.gemspec
@@ -18,14 +18,13 @@
 serialization. }
 
   s.licenses = ['MIT']
-  s.files = Dir["{lib,ext}/**/*.{rb,h,c}"] + ['LICENSE', 'README.md', 'CHANGELOG.md']
+  s.files = Dir["{lib,ext}/**/*.{rb,h,c}"]
 
   s.extensions = ["ext/ox/extconf.rb"]
   # s.executables = []
 
-  s.require_paths = ["lib", "ext"]
+  s.require_paths = ["lib"]
 
-  s.extra_rdoc_files = ['README.md', 'CHANGELOG.md']
   s.rdoc_options = ['--main', 'README.md', '--title', 'Ox Documentation', '--exclude', 'extconf.rb']
   
   s.rubyforge_project = 'ox'
