Upstream-Issue: https://github.com/ohler55/ox/issues/335

--- a/test/sax/sax_test.rb
+++ b/test/sax/sax_test.rb
@@ -202,6 +202,8 @@
   end
 
   def test_sax_element_attrs
+    omit "FIXME: error contains '(null)' instead of 'top'"
+
     Ox.default_options = $ox_sax_options
     parse_compare(%{<top x="57" y='42' z=33/>},
                   [
@@ -398,6 +400,8 @@
   end
 
   def test_sax_element_no_term
+    omit "FIXME: error contains '(null)' instead of 'top'"
+
     Ox.default_options = $ox_sax_options
     parse_compare(%{
 <top>
@@ -561,6 +565,8 @@
   end
 
   def test_sax_text_no_term
+    omit "FIXME: error contains '(null)' instead of 'top'"
+
     Ox.default_options = $ox_sax_options
     parse_compare(%{<top>This is some text.},
                   [
