From 4203c73eae6f989d891be2d284b3c1cb4f9aa476 Mon Sep 17 00:00:00 2001
From: Jeff Ruan <cjr@mega.co.nz>
Date: Fri, 9 Jun 2023 07:25:53 +1200
Subject: [PATCH] SDK-2974 fix glibcxx assertions on unit test

---
 tests/unit/utils.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tests/unit/utils.cpp b/tests/unit/utils.cpp
index bd6b236bf..1edc85580 100644
--- a/tests/unit/utils.cpp
+++ b/tests/unit/utils.cpp
@@ -64,8 +64,11 @@ std::shared_ptr<mega::MegaClient> makeClient(mega::MegaApp& app)
         delete httpio;
     };
 
+    using namespace mega;
+    auto waiter = std::make_shared<WAIT_CLASS>();
+
     std::shared_ptr<mega::MegaClient> client{new mega::MegaClient{
-            &app, nullptr, httpio, nullptr, nullptr, "XXX", "unit_test", 0
+            &app, waiter, httpio, nullptr, nullptr, "XXX", "unit_test", 0
         }, deleter};
 
     return client;
