Add missing headers to fix compiling of tests.

--- sdk-4.17.1d-origin/tests/integration/SdkTest_test.cpp
+++ sdk-4.17.1d/tests/integration/SdkTest_test.cpp
@@ -24,6 +24,7 @@
 #include "SdkTest_test.h"
 #include "mega/testhooks.h"
 #include <algorithm>
+#include <iomanip>
 
 #define SSTR( x ) static_cast< const std::ostringstream & >( \
         (  std::ostringstream() << std::dec << x ) ).str()
--- sdk-4.17.1d-origin/tests/unit/utils_test.cpp
+++ sdk-4.17.1d/tests/unit/utils_test.cpp
@@ -29,6 +29,7 @@
 #include <mega/db.h>
 #include <mega/db/sqlite.h>
 #include <mega/json.h>
+#include <iomanip>
 
 TEST(utils, hashCombine_integer)
 {
