--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,6 @@
 #  Run "radiusd -C", looking for errors.
 #
 $(BUILD_DIR)/tests/radiusd-c: ${BUILD_DIR}/bin/radiusd | build.raddb
-	@$(MAKE) -C raddb/certs
 	@if ! ./build/make/jlibtool --mode=execute ./build/bin/radiusd -XCMd ./raddb -n debug -D ./share > $(BUILD_DIR)/tests/radiusd.config.log 2>&1; then \
 		cat $(BUILD_DIR)/tests/radiusd.config.log; \
 		echo "FAIL: radiusd -C"; \
@@ -205,13 +204,6 @@
 .PHONY: TAGS
 TAGS:
 	etags `find src -type f -name '*.[ch]' -print` > $@
-
-#
-#  Make test certificates.
-#
-.PHONY: certs
-certs:
-	@cd raddb/certs && $(MAKE)
 
 ######################################################################
 #
--- a/raddb/all.mk
+++ b/raddb/all.mk
@@ -114,7 +114,6 @@
 
 $(LOCAL_CERT_PRODUCTS):
 	@echo BOOTSTRAP raddb/certs/
-	@$(MAKE) -C $(R)$(raddbdir)/certs/
 
 # Bootstrap is special
 $(R)$(raddbdir)/certs/bootstrap: | raddb/certs/bootstrap $(LOCAL_CERT_PRODUCTS)
