diff --git a/pygraphviz/tests/test_attribute_defaults.py b/pygraphviz/tests/test_attribute_defaults.py
index d18ca67..1d8b14e 100644
--- a/pygraphviz/tests/test_attribute_defaults.py
+++ b/pygraphviz/tests/test_attribute_defaults.py
@@ -1,6 +1,6 @@
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 def test_default_attributes():
diff --git a/pygraphviz/tests/test_edge_attributes.py b/pygraphviz/tests/test_edge_attributes.py
index e436855..55b4202 100644
--- a/pygraphviz/tests/test_edge_attributes.py
+++ b/pygraphviz/tests/test_edge_attributes.py
@@ -1,6 +1,6 @@
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 def test_edge_attributes():
diff --git a/pygraphviz/tests/test_graph.py b/pygraphviz/tests/test_graph.py
index b410d58..29281b5 100644
--- a/pygraphviz/tests/test_graph.py
+++ b/pygraphviz/tests/test_graph.py
@@ -2,7 +2,7 @@ import pytest
 import unittest
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 class TestGraph(unittest.TestCase):
diff --git a/pygraphviz/tests/test_html.py b/pygraphviz/tests/test_html.py
index 507ba3e..93e1f40 100644
--- a/pygraphviz/tests/test_html.py
+++ b/pygraphviz/tests/test_html.py
@@ -1,6 +1,6 @@
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 long_html_string = """<<TABLE BORDER=0>
diff --git a/pygraphviz/tests/test_node_attributes.py b/pygraphviz/tests/test_node_attributes.py
index 15b4e90..bb2f150 100644
--- a/pygraphviz/tests/test_node_attributes.py
+++ b/pygraphviz/tests/test_node_attributes.py
@@ -1,6 +1,6 @@
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 def test_node_attribute():
diff --git a/pygraphviz/tests/test_string.py b/pygraphviz/tests/test_string.py
index 67e5283..3475395 100644
--- a/pygraphviz/tests/test_string.py
+++ b/pygraphviz/tests/test_string.py
@@ -1,7 +1,7 @@
 import pygraphviz as pgv
 import pytest
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 def test_name():
diff --git a/pygraphviz/tests/test_subgraph.py b/pygraphviz/tests/test_subgraph.py
index 642a327..99d7d9d 100644
--- a/pygraphviz/tests/test_subgraph.py
+++ b/pygraphviz/tests/test_subgraph.py
@@ -1,6 +1,6 @@
 import pygraphviz as pgv
 
-stringify = pgv.testing.stringify
+from pygraphviz.testing import stringify
 
 
 def test_subgraph():
