Upstream: https://github.com/miniupnp/miniupnp/pull/554

diff --git a/miniupnpd.c b/miniupnpd.c
index c24ee3b..529da66 100644
--- a/miniupnpd.c
+++ b/miniupnpd.c
@@ -1760,6 +1760,10 @@ init(int argc, char * * argv, struct runtime_vars * v)
 	}
 	if(!ext_if_name || !lan_addrs.lh_first) {
 		/* bad configuration */
+		if(!ext_if_name)
+			fprintf(stderr, "Error: Option -i missing and ext_ifname is not set in config file\n");
+		if (!lan_addrs.lh_first)
+			fprintf(stderr, "Error: Option -a missing and listening_ip is not set in config file\n");
 		goto print_usage;
 	}
 
